diff -Naur mlterm-2.9.4.orig/configure.in mlterm-2.9.4/configure.in
--- mlterm-2.9.4.orig/configure.in	2007-11-30 09:30:02.000000000 +0000
+++ mlterm-2.9.4/configure.in	2009-08-11 07:19:41.000000000 +0000
@@ -31,6 +31,7 @@
 dnl The ugly part should be replaced by:
 dnl  AC_PATH_TOOL(PKG_CONFIG, pkg-config)
 
+m4_ifdef([m4_pattern_allow], [m4_pattern_allow([PKG_CONFIG_.*])])
 if test x"${build}" != x"${host}"; then
 	AC_CHECK_TOOL(CC, gcc)
 	AC_CHECK_TOOL(AR, ar)
diff -Naur mlterm-2.9.4.orig/dummy-package.txt mlterm-2.9.4/dummy-package.txt
--- mlterm-2.9.4.orig/dummy-package.txt	1970-01-01 00:00:00.000000000 +0000
+++ mlterm-2.9.4/dummy-package.txt	2009-08-11 07:23:59.000000000 +0000
@@ -0,0 +1,2 @@
+The mlterm-shlibs package is merged into mlterm.
+You may remove this dummy package safely.
diff -Naur mlterm-2.9.4.orig/kiklib/configure.in mlterm-2.9.4/kiklib/configure.in
--- mlterm-2.9.4.orig/kiklib/configure.in	2006-10-20 09:04:01.000000000 +0000
+++ mlterm-2.9.4/kiklib/configure.in	2009-08-11 07:19:41.000000000 +0000
@@ -251,7 +251,6 @@
 	else
 		AC_CHECK_FUNC( posix_openpt,
 			[
-			AC_DEFINE(HAVE_POSIX_OPENPT,,"HAVE_POSIX_OPENPT")
 			kik_cv_pty=streams
 			], [
 			if test -c /dev/ptmx ; then
@@ -263,6 +262,9 @@
 AC_MSG_RESULT($kik_cv_pty)
 PTY_NAME="${kik_cv_pty}"
 AC_SUBST(PTY_NAME)
+if test "x$ac_cv_func_posix_openpt" = xyes; then
+	AC_DEFINE(HAVE_POSIX_OPENPT,,"HAVE_POSIX_OPENPT")
+fi
 
 
 #
diff -Naur mlterm-2.9.4.orig/kiklib/src/kik_pty_streams.c mlterm-2.9.4/kiklib/src/kik_pty_streams.c
--- mlterm-2.9.4.orig/kiklib/src/kik_pty_streams.c	2006-10-19 14:41:51.000000000 +0000
+++ mlterm-2.9.4/kiklib/src/kik_pty_streams.c	2009-08-11 07:19:41.000000000 +0000
@@ -2,6 +2,9 @@
  *	$Id: kik_pty_streams.c,v 1.21 2006/10/19 14:41:51 h_minami Exp $
  */
 
+#if defined(__APPLE__) && defined(__MACH__) /* MacOS X */
+# define _APPLE_C_SOURCE
+#else
 /* ptsname/grantpt/unlockpt are not available without _XOPEN_SOURCE
  * on GNU libc based systems. (It may have some ill side-effect though) */
 #define  _XOPEN_SOURCE 500
@@ -12,6 +15,7 @@
 
 /* for solaris */
 #define __EXTENSIONS__
+#endif /* not MacOS X */
 
 #include  "kik_pty.h"
 
@@ -245,6 +249,12 @@
 		}
 #endif /*TIOCNOTTY*/
 #endif /*HAVE_SETSID*/
+#ifdef TIOCSCTTY /* BSD (in addition Linux also knows TIOCSCTTY) */
+		if(ioctl(*slave, TIOCSCTTY, NULL) < 0)
+		{
+			return -1 ;
+		}
+#else /* no TIOCSCTTY (SysV) */
 		fd = open("/dev/tty", O_RDWR | O_NOCTTY);
 		if (fd >= 0)
 		{
@@ -261,6 +271,7 @@
 			return -1;
 		}
 		close(fd);
+#endif /* no TIOCSCTTY (SysV) */
 
 		dup2( *slave , 0) ;
 		dup2( *slave , 1) ;
diff -Naur mlterm-2.9.4.orig/tool/mlconfig/po/Makefile.in.in mlterm-2.9.4/tool/mlconfig/po/Makefile.in.in
--- mlterm-2.9.4.orig/tool/mlconfig/po/Makefile.in.in	2003-01-12 10:08:51.000000000 +0000
+++ mlterm-2.9.4/tool/mlconfig/po/Makefile.in.in	2005-04-02 17:29:55.000000000 +0000
@@ -26,8 +26,7 @@
 
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
-MKINSTALLDIRS = @MKINSTALLDIRS@
-mkinstalldirs = $(SHELL) `case "$(MKINSTALLDIRS)" in /*) echo "$(MKINSTALLDIRS)" ;; *) echo "$(top_builddir)/$(MKINSTALLDIRS)" ;; esac`
+mkinstalldirs = $(top_srcdir)/script/install-sh -d
 
 GMSGFMT = @GMSGFMT@
 MSGFMT = @MSGFMT@
