diff -ruN monafont-2.90/Makefile monafont-2.90-patch/Makefile
--- monafont-2.90/Makefile	Tue Sep  9 22:54:10 2003
+++ monafont-2.90-patch/Makefile	Mon Mar  7 21:26:29 2005
@@ -19,7 +19,7 @@
 
 DESTDIR		=
 X11BINDIR	= /usr/X11R6/bin
-X11FONTDIR	= $(DESTDIR)/usr/X11R6/lib/X11/fonts/local
+X11FONTDIR	= $(DESTDIR)/@PREFIX@/lib/X11/fonts/mona
 BDFTOPCF	= $(X11BINDIR)/bdftopcf
 MKDIRHIER	= $(X11BINDIR)/mkdirhier	# 'mkdir -p' may also work.
 MKFONTDIR	= $(X11BINDIR)/mkfontdir
diff -ruN monafont-2.90/ttfsrc/Makefile monafont-2.90-patch/ttfsrc/Makefile
--- monafont-2.90/ttfsrc/Makefile	Tue Sep  9 22:54:11 2003
+++ monafont-2.90-patch/ttfsrc/Makefile	Mon Mar  7 21:27:30 2005
@@ -27,13 +27,13 @@
 DESTDIR		=
 X11BINDIR	= /usr/X11R6/bin
 MKDIRHIER	= $(X11BINDIR)/mkdirhier	# 'mkdir -p' may also work.
-TTFFONTDIR	= $(DESTDIR)/usr/share/fonts/ja/TrueType
+TTFFONTDIR	= $(DESTDIR)/@PREFIX@/lib/X11/fonts/mona
 
 MY_RPM_ROOT	= /tmp/rpm-root
 
 # modify this:
 #BASE_OUTLINE_TTF=/usr/share/fonts/ja/TrueType/kochi-gothic.ttf	# for RedHat
-BASE_OUTLINE_TTF=/src/kochi-substitute/kochi-gothic-subst.ttf
+BASE_OUTLINE_TTF=@PREFIX@/lib/X11/fonts/kochi-subst/kochi-gothic-subst.ttf
 MONA_OUTLINE_TTF=mona-outline.ttf
 
 TTFPACK=$(PERL) $(TTFTOOLS)/ttfpack
@@ -71,7 +71,7 @@
 MKTESTUNICODE=$(PERL) $(TTFTOOLS)/mktestunicode
 GETFIELD=$(AWK) 'BEGIN{IGNORECASE=1;f=ARGV[1];ARGV[1]="";} $$1~f{print $$2;}'
 GETUNIT=$(GETFIELD) unitsPerEm
-GET_NUMHMETRICS=$(GETFIELD) numberOfHMetrics
+GET_NUMHMETRICS=grep numberOfHMetrics | awk '{print $$2;}'
 GET_NUMGLYPHS=$(GETFIELD) numGlyphs
 COUNTLINES=$(AWK) 'BEGIN{n=0;} {n++;} END{print n;}'
 REPEATFILE=$(PERL) $(TTFTOOLS)/repeatfile
