diff -ru Src/Makefile ../wmWeather-1.31-patched/Src/Makefile
--- Src/Makefile	Tue Mar 16 17:59:39 1999
+++ ../wmWeather-1.31-patched/Src/Makefile	Fri Aug 24 10:54:07 2001
@@ -1,7 +1,7 @@
-CC     = gcc
+CC     = cc
 CFLAGS = -O2 -Wall
 INCDIR = -I/usr/X11R6/include/X11 -I/usr/X11R6/include
-DESTDIR= /usr/X11R6
+DESTDIR= @FINKPREFIX@
 LIBDIR = -L/usr/X11R6/lib
 # for linux
 LIBS   = -lXpm -lX11 -lXext
@@ -30,5 +30,5 @@
 install:: wmWeather
 	install -c -s -m 0755 wmWeather $(DESTDIR)/bin
 	install -c    -m 0755 GrabWeather $(DESTDIR)/bin
-	install -c    -m 0644 wmWeather.1 $(DESTDIR)/man/man1 
+	install -c    -m 0644 wmWeather.1 $(DESTDIR)/share/man/man1 
 
