diff -Nurd -x'*~' libXft-2.1.14.orig/src/Makefile.in libXft-2.1.14/src/Makefile.in
--- libXft-2.1.14.orig/src/Makefile.in	2009-10-09 19:46:04.000000000 -0400
+++ libXft-2.1.14/src/Makefile.in	2010-02-10 14:43:22.000000000 -0500
@@ -250,8 +250,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-AM_CFLAGS = $(FONTCONFIG_CFLAGS) $(FREETYPE_CFLAGS) $(XRENDER_CFLAGS) \
-            $(CWARNFLAGS) -I$(top_builddir)/include/X11/Xft
+AM_CPPFLAGS =  -I$(top_builddir)/include/X11/Xft $(FONTCONFIG_CFLAGS) $(FREETYPE_CFLAGS) $(XRENDER_CFLAGS) $(CWARNFLAGS)
 
 lib_LTLIBRARIES = libXft.la
 libXft_la_SOURCES = xftint.h \
diff -Nurd -x'*~' libXft-2.1.14.orig/xft.pc.in libXft-2.1.14/xft.pc.in
--- libXft-2.1.14.orig/xft.pc.in	2006-07-11 14:05:27.000000000 -0400
+++ libXft-2.1.14/xft.pc.in	2010-02-10 14:28:42.000000000 -0500
@@ -6,7 +6,6 @@
 Name: Xft
 Description: X FreeType library
 Version: @VERSION@
-Requires: xproto, xrender, fontconfig, freetype2
-Requires.private: xrender, fontconfig, freetype2
-Cflags: -I${includedir}
+Requires: fontconfig, freetype2
+Cflags: -I${includedir} -I/usr/X11/include -I/usr/X11R6/include
 Libs: -L${libdir} -lXft
