diff -Naur libid3tag-0.15.0b-old/id3tag.pc libid3tag-0.15.0b/id3tag.pc
--- libid3tag-0.15.0b-old/id3tag.pc   Wed Dec 31 19:00:00 1969
+++ libid3tag-0.15.0b/id3tag.pc       Mon Jan 26 18:46:33 2004
@@ -0,0 +1,11 @@
+prefix=FINK_PREFIX
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: id3tag
+Description: ID3 tag reading library
+Requires:
+Version: 0.15.0b
+Libs: -L${libdir} -lid3tag -lm
+Cflags: -I${includedir}
