--- a/doc/Makefile.in	2012-09-10 22:31:54.000000000 -0400
+++ b/doc/Makefile.in	2012-09-25 14:57:24.000000000 -0400
@@ -393,7 +393,8 @@
 
 @HAVE_DOXYGEN_TRUE@install-data-local:
 @HAVE_DOXYGEN_TRUE@	for f in `find html -type f \! -name "installdox"`; do	\
-@HAVE_DOXYGEN_TRUE@		$(INSTALL_DATA) -D $$f $(DESTDIR)$(docdir)/$$f;	\
+@HAVE_DOXYGEN_TRUE@		$(MKDIR_P) -m 0755 $(DESTDIR)$(docdir)/html;		\
+@HAVE_DOXYGEN_TRUE@		$(INSTALL_DATA) $$f $(DESTDIR)$(docdir)/$$f;	\
 @HAVE_DOXYGEN_TRUE@	done
 
 @HAVE_DOXYGEN_TRUE@	$(INSTALL) -d $(DESTDIR)$(mandir)/man3
