diff -ru libxmp-4.2.0.orig/Makefile.in libxmp-4.2.0/Makefile.in
--- libxmp-4.2.0.orig/Makefile.in	2013-09-01 18:14:50.000000000 -0400
+++ libxmp-4.2.0/Makefile.in	2013-11-10 20:54:50.000000000 -0500
@@ -112,7 +112,7 @@
 
 lib/$(DYLIB): $(LOBJS)
 	@mkdir -p lib
-	@CMD='$(LD) $(LDFLAGS) -dynamiclib -Wl,-headerpad_max_install_names,-undefined,dynamic_lookup,-compatibility_version,$(COMPAT_VERSION),-current_version,$(VERSION),-install_name,$(prefix)/lib/$(DYLIB) -o $@ $(LOBJS) $(LIBS)'; \
+	@CMD='$(LD) $(LDFLAGS) -dynamiclib -Wl,-compatibility_version,$(COMPAT_VERSION),-current_version,$(VERSION),-install_name,$(prefix)/lib/libxmp.$(VERSION_MAJOR).dylib -o $@ $(LOBJS) $(LIBS)'; \
 	if [ "$(V)" -gt 0 ]; then echo $$CMD; else echo LD $@ ; fi; \
 	eval $$CMD
 	ln -sf $(DYLIB) lib/libxmp.$(VERSION_MAJOR).dylib
diff -ru libxmp-4.2.0.orig/configure libxmp-4.2.0/configure
--- libxmp-4.2.0.orig/configure	2013-06-08 13:11:50.000000000 -0400
+++ libxmp-4.2.0/configure	2013-11-10 20:54:50.000000000 -0500
@@ -3972,7 +3972,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_c_flag_w_unused_but_set_variable" >&5
 $as_echo "$ac_cv_c_flag_w_unused_but_set_variable" >&6; }
   if test "x$ac_cv_c_flag_w_unused_but_set_variable" = xyes; then :
-  CFLAGS="${CFLAGS} -Wno-unused-but-set-variable"
+  CFLAGS="${CFLAGS} -Wno-unused-variable"
 fi
 
 
diff -ru libxmp-4.2.0.orig/src/common.h libxmp-4.2.0/src/common.h
--- libxmp-4.2.0.orig/src/common.h	2013-11-09 16:38:53.000000000 -0500
+++ libxmp-4.2.0/src/common.h	2013-11-10 20:56:24.000000000 -0500
@@ -113,14 +113,6 @@
 
 #endif	/* !_MSC_VER */
 
-#ifdef HAVE_STRLCPY
-#define strncpy strlcpy
-#endif
-
-#ifdef HAVE_STRLCAT
-#define strncat strlcat
-#endif
-
 #ifdef _MSC_VER
 #define dup _dup
 #define fileno _fileno
