--- portaudio_v18_1/Makefile.in	2003-02-11 23:24:56.000000000 +0100
+++ portaudio_v18_1_pachted/Makefile.in	2007-11-15 17:34:05.000000000 +0100
@@ -6,24 +6,25 @@
 
 PREFIX = @prefix@
 CC = @CC@
-CFLAGS = @CFLAGS@ -Ipa_common
-LIBS = @LIBS@
+CFLAGS = @CFLAGS@ -Ipa_common -Ipablio
+LIBS = @LIBS@ -framework AudioToolbox
 AR = @AR@
 RANLIB = @RANLIB@
 INSTALL = @INSTALL@
-SHARED_FLAGS = @SHARED_FLAGS@
+SHARED_FLAGS = @SHARED_FLAGS@ -framework AudioToolbox -install_name @prefix@/lib/libportaudio.0.dylib -compatibility_version 18.0 -current_version 18.1
 DLL_LIBS = @DLL_LIBS@
 
 OTHER_OBJS = @OTHER_OBJS@
 
 PALIB = libportaudio.a
 PADLL = @PADLL@
-PADLLV = $(PADLL).0.0.18
+PADLLV = libportaudio.0.18.1.dylib
 PAINC = pa_common/portaudio.h
 
 COMMON_OBJS = \
 	pa_common/pa_convert.o \
-	pa_common/pa_lib.o
+	pa_common/pa_lib.o \
+	pablio/ringbuffer.o
 
 TESTS = \
 	bin/patest_buffer \
