diff -ruN quodlibet-2.6.3-orig/exfalso.py quodlibet-2.6.3/exfalso.py
--- quodlibet-2.6.3-orig/exfalso.py	2013-09-12 18:49:22.000000000 -0400
+++ quodlibet-2.6.3/exfalso.py	2013-10-01 15:18:26.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!@FINK_PREFIX@/bin/python@FINK_PYTHON_V@
 # Copyright 2004-2005 Joe Wreschnig, Niklas Janlert
 #           2012 Christoph Reiter
 # <quod-libet-development@googlegroups.com>
diff -ruN quodlibet-2.6.3-orig/operon.py quodlibet-2.6.3/operon.py
--- quodlibet-2.6.3-orig/operon.py	2013-09-12 18:49:22.000000000 -0400
+++ quodlibet-2.6.3/operon.py	2013-10-01 15:18:26.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@FINK_PREFIX@/bin/python@FINK_PYTHON_V@
 # Copyright 2012,2013 Christoph Reiter
 #
 # This program is free software; you can redistribute it and/or modify
diff -ruN quodlibet-2.6.3-orig/quodlibet/util/__init__.py quodlibet-2.6.3/quodlibet/util/__init__.py
--- quodlibet-2.6.3-orig/quodlibet/util/__init__.py	2013-09-25 06:31:19.000000000 -0400
+++ quodlibet-2.6.3/quodlibet/util/__init__.py	2013-10-01 15:18:26.000000000 -0400
@@ -175,6 +175,8 @@
     def parse(self, args=None):
         if args is None:
             args = sys.argv[1:]
+        print "CmdLine is " + " ".join(args)
+        args = [a for a in args if not a.startswith("-psn_")]
         from getopt import getopt, GetoptError
         try:
             opts, args = getopt(args, self.__shorts(), self.__longs())
diff -ruN quodlibet-2.6.3-orig/quodlibet.py quodlibet-2.6.3/quodlibet.py
--- quodlibet-2.6.3-orig/quodlibet.py	2013-09-25 06:31:19.000000000 -0400
+++ quodlibet-2.6.3/quodlibet.py	2013-10-01 15:18:26.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!@FINK_PREFIX@/bin/python@FINK_PYTHON_V@
 # -*- coding: utf-8 -*-
 
 # Copyright 2004-2005 Joe Wreschnig, Michael Urman, Iñigo Serna
diff -ruN quodlibet-2.6.3-orig/setup.py quodlibet-2.6.3/setup.py
--- quodlibet-2.6.3-orig/setup.py	2013-09-25 06:31:19.000000000 -0400
+++ quodlibet-2.6.3/setup.py	2013-10-01 15:19:06.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!@FINK_PREFIX@/bin/python@FINK_PYTHON_V@
 
 0<>0 # Python 3.x not supported. Use 2.6+ instead.
 
