diff -ruN libdvdread-0.9.4.orig/dvdread/dvd_input.c libdvdread-0.9.4/dvdread/dvd_input.c
--- libdvdread-0.9.4.orig/dvdread/dvd_input.c	Thu Feb 13 14:48:24 2003
+++ libdvdread-0.9.4/dvdread/dvd_input.c	Sun Mar  2 14:21:57 2003
@@ -274,7 +274,7 @@
 
 #else
   /* dlopening libdvdcss */
-  dvdcss_library = dlopen("libdvdcss.so.2", RTLD_LAZY);
+  dvdcss_library = dlopen("libdvdcss.2.dylib", RTLD_LAZY);
   
   if(dvdcss_library != NULL) {
 #if defined(__OpenBSD__) && !defined(__ELF__)
@@ -306,7 +306,7 @@
       dvdcss_library = NULL;
     } else if(!DVDcss_open  || !DVDcss_close || !DVDcss_title || !DVDcss_seek
 	      || !DVDcss_read || !DVDcss_error || !dvdcss_version) {
-      fprintf(stderr,  "libdvdread: Missing symbols in libdvdcss.so.2, "
+      fprintf(stderr,  "libdvdread: Missing symbols in libdvdcss.2.dylib, "
 	      "this shouldn't happen !\n");
       dlclose(dvdcss_library);
     }
