diff -uNr libxslt-1.1.8/configure libxslt-1.1.8-new/configure
--- libxslt-1.1.8/configure	2004-07-05 15:48:05.000000000 -0400
+++ libxslt-1.1.8-new/configure	2004-07-07 18:29:10.000000000 -0400
@@ -10173,7 +10173,7 @@
     # Don't fix this by using the ld -exported_symbols_list flag, it doesn't exist in older darwin ld's
 		  archive_expsym_cmds='sed -e "s,#.*,," -e "s,^[    ]*,," -e "s,^\(..*\),_&," < $export_symbols > $output_objdir/${libname}-symbols.expsym~$CC -dynamiclib $allow_undefined_flag  -o $lib $libobjs $deplibs$compiler_flags -install_name $rpath/$soname $verstring~nmedit -s $output_objdir/${libname}-symbols.expsym ${lib}'
 		  module_expsym_cmds='sed -e "s,#.*,," -e "s,^[    ]*,," -e "s,^\(..*\),_&," < $export_symbols > $output_objdir/${libname}-symbols.expsym~$CC -bundle $allow_undefined_flag  -o $lib $libobjs $deplibs$compiler_flags~nmedit -s $output_objdir/${libname}-symbols.expsym ${lib}'
-    hardcode_direct=no
+    hardcode_direct=yes
     hardcode_automatic=yes
     hardcode_shlibpath_var=unsupported
     whole_archive_flag_spec='-all_load $convenience'
