diff -ru ethereal-0.10.10/configure ethereal-0.10.10-patched/configure
--- ethereal-0.10.10/configure	2005-03-10 16:58:39.000000000 +0100
+++ ethereal-0.10.10-patched/configure	2005-03-13 16:28:49.000000000 +0100
@@ -30134,7 +30134,7 @@
 #
 # The plugin dissectors reside in ./plugins/PROTO/
 #
-PLUGIN_LIBS="-L../../epan -lethereal $GLIB_LIBS"
+PLUGIN_LIBS="../../epan/libethereal.la $GLIB_LIBS"
 
 
 #
