# $Id $
diff -ruN giFT/etc/gift.conf giFT.new/etc/gift.conf
--- giFT/etc/gift.conf	2003-06-18 20:50:41.000000000 -0400
+++ giFT.new/etc/gift.conf	2003-07-02 19:48:10.000000000 -0400
@@ -1,4 +1,5 @@
 # gift.conf
+
 ###############################################################################
 
 ###############################################################################
@@ -71,9 +72,9 @@
 #
 #  Protocol names are case sensitive.
 #
-# For example, to use the OpenFT and Gnutella protocols use:
+# For example, to use the OpenFT, Gnutella and FastTrack protocols use:
 #
-#  plugins = OpenFT:Gnutella
+#  plugins = OpenFT:Gnutella:FastTrack
 #
 # Default: none
 #
diff -ruN giFT/src/Makefile.am giFT.new/src/Makefile.am
--- giFT/src/Makefile.am	2003-06-02 00:20:19.000000000 -0400
+++ giFT.new/src/Makefile.am	2003-06-02 19:48:10.000000000 -0400
@@ -58,6 +58,7 @@
 
 giFT_LDFLAGS = \
 	$(GIFT_LDFLAGS) \
+	-L$(top_builddir)/lib/.libs \
 	$(IMAGEMAGICK_LDFLAGS) \
 	-export-dynamic
 
