diff -ruN hxd-0.2.17/hxd/hxd.conf hxd-0.2.17-patched/hxd/hxd.conf
--- hxd-0.2.17/hxd/hxd.conf	Fri May 23 07:25:40 2003
+++ hxd-0.2.17-patched/hxd/hxd.conf	Sat Dec 13 22:08:28 2003
@@ -85,16 +85,16 @@
 };
 
 paths {
-	files "./files";
-	accounts "./accounts";
-	exec "./exec";
-	news "./news";
-	newsdir "./newsdir";
-	agreement "./agreement";
-	log "./log";
-	banlist "./banlist";
-	tracker_banlist "./tracker_banlist";
-	hldump "./hldump";
+	files "<STOWDIR>/share/hxd/files";
+	accounts "<STOWDIR>/share/hxd/accounts";
+	exec "<STOWDIR>/share/hxd/exec";
+	news "<STOWDIR>/share/hxd/news";
+	newsdir ".<STOWDIR>/share/hxd/newsdir";
+	agreement "<STOWDIR>/share/hxd/agreement";
+	log "<STOWDIR>/share/hxd/log";
+	banlist "<STOWDIR>/share/hxd/banlist";
+	tracker_banlist "<STOWDIR>/share/hxd/tracker_banlist";
+	hldump "<STOWDIR>/share/hxd/hldump";
 };
 
 # default permissions of created files
diff -ruN hxd-0.2.17/src/common/htxf.c hxd-0.2.17-patched/src/common/htxf.c
--- hxd-0.2.17/src/common/htxf.c	Thu May 29 08:23:17 2003
+++ hxd-0.2.17-patched/src/common/htxf.c	Sat Dec 13 22:03:24 2003
@@ -1199,7 +1199,7 @@
 #if defined(CONFIG_HTXF_CLONE) || defined(CONFIG_HTXF_PTHREAD)
 				xxx_add(htlcp->htxf_out[i]);
 #endif
-				err = thread_create(&htxf->tid, &htxf->stack, get_thread, htlcp->htxf_out[i]);
+				err = hxd_thread_create(&htxf->tid, &htxf->stack, get_thread, htlcp->htxf_out[i]);
 				if (err) {
 #if defined(CONFIG_HTXF_CLONE) || defined(CONFIG_HTXF_PTHREAD)
 					xxx_remove(htlcp->htxf_out[i]);
@@ -1238,7 +1238,7 @@
 #if defined(CONFIG_HTXF_CLONE) || defined(CONFIG_HTXF_PTHREAD)
 				xxx_add(htlcp->htxf_in[i]);
 #endif
-				err = thread_create(&htxf->tid, &htxf->stack, put_thread, htlcp->htxf_in[i]);
+				err = hxd_thread_create(&htxf->tid, &htxf->stack, put_thread, htlcp->htxf_in[i]);
 				if (err) {
 #if defined(CONFIG_HTXF_CLONE) || defined(CONFIG_HTXF_PTHREAD)
 					xxx_remove(htlcp->htxf_in[i]);
diff -ruN hxd-0.2.17/src/common/hxd.h hxd-0.2.17-patched/src/common/hxd.h
--- hxd-0.2.17/src/common/hxd.h	Sun May 25 22:57:16 2003
+++ hxd-0.2.17-patched/src/common/hxd.h	Sat Dec 13 22:06:33 2003
@@ -138,7 +138,7 @@
 	struct qbuf in;
 	struct timeval start;
 
-	thread_t tid;
+	hxd_thread_t tid;
 	void *stack;
 
 #if defined(CONFIG_HOTLINE_CLIENT)
diff -ruN hxd-0.2.17/src/common/threads.c hxd-0.2.17-patched/src/common/threads.c
--- hxd-0.2.17/src/common/threads.c	Thu May 29 08:05:15 2003
+++ hxd-0.2.17-patched/src/common/threads.c	Sat Dec 13 22:05:30 2003
@@ -5,10 +5,10 @@
 #include "threads.h"
 
 int
-thread_create (thread_t *tidp, void **stackp, thread_return_type (*fn)(void *), void *datap)
+hxd_thread_create (hxd_thread_t *tidp, void **stackp, thread_return_type (*fn)(void *), void *datap)
 {
 	int err;
-	thread_t tid;
+	hxd_thread_t tid;
 #if defined(CONFIG_HTXF_CLONE)
 	void *stack, *hstack;
 #endif
diff -ruN hxd-0.2.17/src/common/threads.h hxd-0.2.17-patched/src/common/threads.h
--- hxd-0.2.17/src/common/threads.h	Thu May 29 08:04:43 2003
+++ hxd-0.2.17-patched/src/common/threads.h	Sat Dec 13 22:06:15 2003
@@ -14,15 +14,15 @@
 
 #if defined(CONFIG_HTXF_PTHREAD)
 typedef void * thread_return_type;
-typedef pthread_t thread_t;
+typedef pthread_t hxd_thread_t;
 #else
 typedef int thread_return_type;
-typedef pid_t thread_t;
+typedef pid_t hxd_thread_t;
 #endif
 
 #include <signal.h>
 
-extern int thread_create (thread_t *tidp, void **stackp, thread_return_type (*fn)(void *), void *datap);
+extern int hxd_thread_create (hxd_thread_t *tidp, void **stackp, thread_return_type (*fn)(void *), void *datap);
 extern void mask_signal (int how, int sig);
 
 //extern void threads_init (struct SOCKADDR_IN *saddr);
diff -ruN hxd-0.2.17/src/conf/hxd.conf hxd-0.2.17-patched/src/conf/hxd.conf
--- hxd-0.2.17/src/conf/hxd.conf	Fri May 23 07:29:35 2003
+++ hxd-0.2.17-patched/src/conf/hxd.conf	Sat Dec 13 22:23:42 2003
@@ -125,8 +125,8 @@
 # do not change the number and order of the %'s
 	str news_divider "_________________________________________________________";
 	str news_time_format "(%c):";
-	str chat_format "\r%13.13s:  %s";
-	str chat_opt_format "\r *** %s %s";
+	str chat_format "\\r%13.13s:  %s";
+	str chat_opt_format "\\r *** %s %s";
 };
 
 # --enable-sql
diff -ruN hxd-0.2.17/src/hx/files.c hxd-0.2.17-patched/src/hx/files.c
--- hxd-0.2.17/src/hx/files.c	Mon May 26 00:19:39 2003
+++ hxd-0.2.17-patched/src/hx/files.c	Sat Dec 13 22:03:24 2003
@@ -1690,14 +1690,14 @@
 #else
 	htxf->pipe = 0;
 #endif
-	err = thread_create(&htxf->tid, &htxf->stack, ((htxf->type & XFER_GET) ? get_thread : put_thread), htxf);
+	err = hxd_thread_create(&htxf->tid, &htxf->stack, ((htxf->type & XFER_GET) ? get_thread : put_thread), htxf);
 #if !defined(__WIN32__)
 	sigaction(SIGTSTP, &tstpact, 0);
 	sigaction(SIGCONT, &contact, 0);
 	unignore_signals(&oldset);
 #endif
 	if (err) {
-		hx_printf_prefix(htlc, 0, INFOPREFIX, "xfer: thread_create: %s\n", strerror(err));
+		hx_printf_prefix(htlc, 0, INFOPREFIX, "xfer: hxd_thread_create: %s\n", strerror(err));
 		goto err_pfds;
 	}
 
diff -ruN hxd-0.2.17/src/hxd/files.c hxd-0.2.17-patched/src/hxd/files.c
--- hxd-0.2.17/src/hxd/files.c	Sun May 25 22:57:47 2003
+++ hxd-0.2.17-patched/src/hxd/files.c	Sat Dec 13 22:03:24 2003
@@ -1469,7 +1469,7 @@
 	strcpy(htxf->path, path);
 
 #if HTXF_THREADS_LISTEN
-	err = thread_create(&htxf->tid, &htxf->stack, get_thread, htxf);
+	err = hxd_thread_create(&htxf->tid, &htxf->stack, get_thread, htxf);
 	if (err) {
 		LOCK_HTXF(htlc);
 		for (i = 0; i < HTXF_GET_MAX; i++) {
@@ -1618,7 +1618,7 @@
 	strcpy(htxf->path, path);
 
 #if HTXF_THREADS_LISTEN
-	err = thread_create(&htxf->tid, &htxf->stack, put_thread, htxf);
+	err = hxd_thread_create(&htxf->tid, &htxf->stack, put_thread, htxf);
 	if (err) {
 		LOCK_HTXF(htlc);
 		for (i = 0; i < HTXF_PUT_MAX; i++) {
@@ -1786,7 +1786,7 @@
 	strcpy(htxf->path, path);
 
 #if HTXF_THREADS_LISTEN
-	err = thread_create(&htxf->tid, &htxf->stack, get_thread, htxf);
+	err = hxd_thread_create(&htxf->tid, &htxf->stack, get_thread, htxf);
 	if (err) {
 		LOCK_HTXF(htlc);
 		for (i = 0; i < HTXF_GET_MAX; i++) {
@@ -1934,7 +1934,7 @@
 	strcpy(htxf->path, path);
 
 #if HTXF_THREADS_LISTEN
-	err = thread_create(&htxf->tid, &htxf->stack, put_thread, htxf);
+	err = hxd_thread_create(&htxf->tid, &htxf->stack, put_thread, htxf);
 	if (err) {
 		LOCK_HTXF(htlc);
 		for (i = 0; i < HTXF_PUT_MAX; i++) {
diff -ruN hxd-0.2.17/src/hxd/rcv.c hxd-0.2.17-patched/src/hxd/rcv.c
--- hxd-0.2.17/src/hxd/rcv.c	Sat May 24 08:58:29 2003
+++ hxd-0.2.17-patched/src/hxd/rcv.c	Sat Dec 13 22:28:03 2003
@@ -5,6 +5,7 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <dirent.h>
+#include <sys/stat.h>
 #include <errno.h>
 #include "sys_net.h"
 #if !defined(__WIN32__)
