diff -urN oinkmaster-0.8.orig/oinkmaster.pl oinkmaster-0.8/oinkmaster.pl
--- oinkmaster-0.8.orig/oinkmaster.pl	2003-09-03 10:45:08.000000000 +0200
+++ oinkmaster-0.8/oinkmaster.pl	2003-09-30 18:01:08.000000000 +0200
@@ -48,7 +48,7 @@
 my $min_rules         = 1;
 my $min_files         = 1;
 
-my $config_file       = "/usr/local/etc/oinkmaster.conf";
+my $config_file       = "@PREFIX@/etc/oinkmaster.conf";
 
 my $verbose           = 0;
 my $careful           = 0;
--- oinkmaster/oinkmaster.conf.orig	2003-09-30 18:50:49.000000000 +0200
+++ oinkmaster/oinkmaster.conf	2003-09-30 18:51:14.000000000 +0200
@@ -26,7 +26,7 @@
 # 'tar' and 'gzip' must be found, and also 'wget' if fetching rules
 # from a remote host (all with optional .exe suffix).
 # Assume UNIX style by default:
-path = /bin:/usr/bin:/usr/local/bin
+path = @PREFIX@/bin:/bin:/usr/bin:/usr/local/bin
 
 # Example if running native Win32 or standalone Cygwin:
 # path = c:\oinkmaster;c:\oinkmaster\bin
