diff -Nurd -x'*~' evolution-data-server-2.26.2.orig/Makefile.am evolution-data-server-2.26.2/Makefile.am
--- evolution-data-server-2.26.2.orig/Makefile.am	2009-05-03 10:55:03.000000000 -0400
+++ evolution-data-server-2.26.2/Makefile.am	2009-06-28 03:52:06.000000000 -0400
@@ -1,9 +1,6 @@
 AUTOMAKE_OPTIONS = 1.6
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
 
-if ! WITH_LIBDB
-LIBDB = libdb
-endif
 
 if ENABLE_CALENDAR
 CALENDAR_DIR = calendar
diff -Nurd -x'*~' evolution-data-server-2.26.2.orig/acinclude.m4 evolution-data-server-2.26.2/acinclude.m4
--- evolution-data-server-2.26.2.orig/acinclude.m4	2009-05-03 10:55:03.000000000 -0400
+++ evolution-data-server-2.26.2/acinclude.m4	2009-06-28 03:52:06.000000000 -0400
@@ -233,11 +233,10 @@
 ])
 dnl -*- mode: autoconf -*-
 
-# serial 1
 
 dnl Usage:
 dnl   GTK_DOC_CHECK([minimum-gtk-doc-version])
-AC_DEFUN([GTK_DOC_CHECK],
+AC_DEFUN([GTK_DOC_CHECK__DISABLED],
 [
   AC_BEFORE([AC_PROG_LIBTOOL],[$0])dnl setup libtool first
   AC_BEFORE([AM_PROG_LIBTOOL],[$0])dnl setup libtool first
@@ -296,7 +295,7 @@
 dnl Optionally, copy this file into acinclude.m4, to avoid the need to have it
 dnl installed when running autoconf on your project.
 
-AC_DEFUN([DOLT], [
+AC_DEFUN([DOLT__DISABLED], [
 AC_REQUIRE([AC_CANONICAL_HOST])
 # dolt, a replacement for libtool
 # Josh Triplett <josh@freedesktop.org>
diff -Nurd -x'*~' evolution-data-server-2.26.3.orig/configure.in evolution-data-server-2.26.3/configure.in
--- evolution-data-server-2.26.3.orig/configure.in	2009-06-26 08:36:57.000000000 -0400
+++ evolution-data-server-2.26.3/configure.in	2009-07-10 14:33:56.000000000 -0400
@@ -191,7 +191,6 @@
 AM_DISABLE_STATIC
 AC_LIBTOOL_WIN32_DLL
 AM_PROG_LIBTOOL
-DOLT
 
 PKG_PROG_PKG_CONFIG
 
@@ -280,16 +279,6 @@
         LIBS=$save_libs
 
         msg_libdb="dynamically linked"
-else
-        DB_CFLAGS="-I\$(top_builddir)/libdb/dist"
-        DB_LIBS="\$(top_builddir)/libdb/dist/libdb-4.1.la"
-	if test $os_win32 = yes; then
-		# Don't pointlessly auto-export the global symbols
-		# from the static libdb
-		DB_LIBS="$DB_LIBS -Wl,--exclude-libs=libdb-4.1.a"
-	fi
-        AC_MSG_NOTICE(Using local libdb)
-        msg_libdb="statically linked to local copy"
 fi
 AM_CONDITIONAL(WITH_LIBDB, test "x$dynamic_libdb" != "xno")
 AC_SUBST(DB_CFLAGS)
@@ -303,7 +292,7 @@
 AC_ARG_WITH(libiconv, 
   AC_HELP_STRING( [--with-libiconv],
 		  [Prefix where libiconv is installed]))
-if test -d $withval; then
+if test -d "$withval"; then
     ICONV_CFLAGS="-I$withval/include"
     ICONV_LIBS="-L$withval/lib"
 fi
@@ -315,13 +304,9 @@
 AC_CACHE_CHECK(for iconv in -liconv, ac_cv_libiconv, AC_TRY_LINK([
 #include <iconv.h>
 #include <stdlib.h>],[
-
-int main (int argc, char **argv)
-{
 	iconv_t cd;
 	cd = iconv_open ("UTF-8", "ISO-8859-1");
 	exit (0);
-}
 ], ac_cv_libiconv=yes, ac_cv_libiconv=no))
 if test $ac_cv_libiconv = yes; then
 	ICONV_LIBS="$ICONV_LIBS -liconv"
@@ -345,7 +330,7 @@
 #include <iconv.h>
 #include <stdlib.h>
 #include <string.h>
-#ifdef HAVE_GNU_GET_LIBC_VERSION
+#if 0
 #include <gnu/libc-version.h>
 #endif
 
@@ -358,7 +343,7 @@
 	size_t from_len = strlen (from), utf8_len = 20;
 	size_t utf8_real_len = strlen (utf8);
 
-#ifdef HAVE_GNU_GET_LIBC_VERSION
+#if 0
 	/* glibc 2.1.2's iconv is broken in hard to test ways. */
 	if (!strcmp (gnu_get_libc_version (), "2.1.2"))
 		exit (1);
@@ -1244,7 +1229,8 @@
 				msg_krb5="yes (Sun)"
 			fi
 		fi
-		KRB5_LDFLAGS="-L$with_krb5_libs $ac_cv_lib_kerberos5"
+		KRB5_CFLAGS=""
+		KRB5_LDFLAGS="$ac_cv_lib_kerberos5"
 		AC_MSG_RESULT(msg_krb5)
 	else
 		dnl AC_MSG_CHECKING([for Kerberos 5])
@@ -1294,15 +1280,15 @@
 
 		if test -z "$with_krb4_includes"; then
 			if test -f "$with_krb4/include/krb.h" -o -f "$with_krb4/include/port-sockets.h"; then
-				KRB4_CFLAGS="-I$with_krb4/include"
+				KRB4_CFLAGS=""
 			fi
 			if test -d "$with_krb4/include/kerberosIV"; then
 				KRB4_CFLAGS="$KRB4_CFLAGS -I$with_krb4/include/kerberosIV"
 			fi
 		else
-			KRB4_CFLAGS="-I$with_krb4_includes"
+			KRB4_CFLAGS=""
 		fi
-		KRB4_LDFLAGS="-L$with_krb4_libs $ac_cv_lib_kerberos4"
+		KRB4_LDFLAGS="$ac_cv_lib_kerberos4"
 		
 		CFLAGS_save="$CFLAGS"
 		CFLAGS="$CFLAGS $KRB4_CFLAGS"
@@ -1487,10 +1473,7 @@
 AC_MSG_RESULT($IDL_INCLUDES)
 AC_SUBST(IDL_INCLUDES)
 
-AC_MSG_CHECKING(for libgnomeui server directory)
-GNOMEUI_SERVERDIR="`$PKG_CONFIG --variable=libgnomeui_serverdir libgnomeui-2.0`"
-AC_MSG_RESULT($GNOMEUI_SERVERDIR)
-AC_DEFINE_UNQUOTED(GNOMEUI_SERVERDIR, "$GNOMEUI_SERVERDIR", [Path where we can find gnome_segv2])
+echo "Disabling check for libgnomeui-2.0: gnome_segv2 removed as of GNOME2.20"
 
 dnl Utility macro to set compiler flags for a specific lib.
 dnl  EVO_SET_COMPILE_FLAGS(VAR-PREFIX, DEPS, EXTRA-CFLAGS, EXTRA-LIBS)
@@ -1730,9 +1713,6 @@
 export privincludedir
 export privdatadir
 
-if test $dynamic_libdb = no; then
-        AC_CONFIG_SUBDIRS(libdb/dist)
-fi
 AC_OUTPUT([
 Makefile
 evolution-data-server-zip
@@ -1792,7 +1772,6 @@
 camel/tests/stream/Makefile
 camel/camel.pc
 camel/camel-provider.pc
-libdb/Makefile
 libebackend/Makefile
 libebackend/libebackend.pc
 libedataserver/Makefile
diff -Nurd -x'*~' evolution-data-server-2.26.2.orig/src/server.c evolution-data-server-2.26.2/src/server.c
--- evolution-data-server-2.26.2.orig/src/server.c	2009-05-03 10:55:04.000000000 -0400
+++ evolution-data-server-2.26.2/src/server.c	2009-06-28 03:52:38.000000000 -0400
@@ -90,9 +90,7 @@
 static void
 gnome_segv_handler (int signo)
 {
-	const char *gnome_segv_path;
 	static int in_segv = 0;
-	char *exec;
 
 	if (pthread_self() != main_thread) {
 		/* deadlock intentionally in the sub-threads */
@@ -110,12 +108,7 @@
                 _exit (1);
         }
 
-	gnome_segv_path = GNOMEUI_SERVERDIR "/gnome_segv2";
-
-	exec = g_strdup_printf ("%s \"" PACKAGE "-" BASE_VERSION "\" %d \"" VERSION "\"",
-				gnome_segv_path, signo);
-	system (exec);
-	g_free (exec);
+	fprintf (stderr, "Going down on segv (signo=%d)\n", signo);
 
 	_exit(1);
 }
