diff -Nurd -x'*~' galeon-2.0.5.orig/libegg/dock/Makefile.in galeon-2.0.5/libegg/dock/Makefile.in
--- galeon-2.0.5.orig/libegg/dock/Makefile.in	2008-02-10 12:44:10.000000000 -0500
+++ galeon-2.0.5/libegg/dock/Makefile.in	2008-05-01 23:00:13.000000000 -0400
@@ -263,11 +263,11 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 INCLUDES = \
-	$(GALEON_DEPENDENCY_CFLAGS) \
 	-DG_LOG_DOMAIN=\"Egg\" \
 	-DEGG_GLADEDIR=\".\" \
 	-I$(top_builddir)/libegg \
-	-I$(top_srcdir)/libegg
+	-I$(top_srcdir)/libegg \
+	$(GALEON_DEPENDENCY_CFLAGS)
 
 noinst_LTLIBRARIES = libegg-dock.la
 BUILT_SOURCES = \
diff -Nurd -x'*~' galeon-2.0.5.orig/mozilla/mozilla-embed-shell.cpp galeon-2.0.5/mozilla/mozilla-embed-shell.cpp
--- galeon-2.0.5.orig/mozilla/mozilla-embed-shell.cpp	2008-02-02 17:26:31.000000000 -0500
+++ galeon-2.0.5/mozilla/mozilla-embed-shell.cpp	2008-05-01 12:24:20.000000000 -0400
@@ -236,6 +236,9 @@
 
 	g_setenv ("MOZ_PLUGIN_PATH", new_path->str, TRUE);
 
+	/* Work-around Flash Player crash */
+	g_setenv ("XLIB_SKIP_ARGB_VISUALS", "1", FALSE);
+
 	g_string_free (new_path, TRUE);
 }
 
diff -Nurd -x'*~' galeon-2.0.5.orig/src/Makefile.in galeon-2.0.5/src/Makefile.in
--- galeon-2.0.5.orig/src/Makefile.in	2008-02-10 12:44:11.000000000 -0500
+++ galeon-2.0.5/src/Makefile.in	2010-02-07 15:24:32.000000000 -0500
@@ -488,7 +488,7 @@
 	$(top_builddir)/libegg/libegg.la \
 	$(top_builddir)/libegg/dock/libegg-dock.la \
 	$(GALEON_DEPENDENCY_LIBS) \
-	$(INTLLIBS)
+	$(INTLLIBS) -lX11
 
 galeon_LDFLAGS = -R$(MOZILLA_HOME) -dlopen self
 @ENABLE_GTKHTML_EMBED_TRUE@gtkhtmlembed_libs = $(top_builddir)/gtkhtml/libgtkhtmlembed.la
diff -Nurd -x'*~' galeon-2.0.5.orig/utils/gul-x11.c galeon-2.0.5/utils/gul-x11.c
--- galeon-2.0.5.orig/utils/gul-x11.c	2007-12-15 01:10:36.000000000 -0500
+++ galeon-2.0.5/utils/gul-x11.c	2008-05-01 12:24:20.000000000 -0400
@@ -52,7 +52,7 @@
 
        XGetWindowProperty (GDK_DISPLAY_XDISPLAY (display), GDK_WINDOW_XID (root_win),
                            gdk_x11_get_xatom_by_name_for_display (display, "_NET_CURRENT_DESKTOP"),
-                           0, G_MAXLONG,
+                           0, G_MAXLONG - 3,
                            False, XA_CARDINAL, &type, &format, &nitems,
                            &bytes_after, (gpointer)&current_desktop);
        if (type == XA_CARDINAL && format == 32 && nitems > 0)
@@ -85,7 +85,7 @@
 
        XGetWindowProperty (GDK_DISPLAY_XDISPLAY (display), GDK_WINDOW_XID (window),
                            gdk_x11_get_xatom_by_name_for_display (display, "_NET_WM_DESKTOP"),
-                           0, G_MAXLONG,
+                           0, G_MAXLONG - 3,
                            False, XA_CARDINAL, &type, &format, &nitems,
                            &bytes_after, (gpointer)&workspace);
        if (type == XA_CARDINAL && format == 32 && nitems > 0)
