diff -Nurd libgnomedb-3.1.2.orig/configure libgnomedb-3.1.2/configure
--- libgnomedb-3.1.2.orig/configure	2007-10-24 15:44:07.000000000 -0400
+++ libgnomedb-3.1.2/configure	2008-03-06 14:51:57.000000000 -0500
@@ -23288,7 +23288,7 @@
 			LIBGCRYPT_LIBS=""
 
 
-if 0; then
+if false; then
   HAVE_LIBGCRYPT_TRUE=
   HAVE_LIBGCRYPT_FALSE='#'
 else
@@ -23316,7 +23316,7 @@
       	LIBGCRYPT_LIBS=""
 
 
-if 0; then
+if false; then
   HAVE_LIBGCRYPT_TRUE=
   HAVE_LIBGCRYPT_FALSE='#'
 else
@@ -23527,6 +23527,7 @@
 
 GRAPHVIZ_MODULES="libgvc"
 
+if false; then
 pkg_failed=no
 { echo "$as_me:$LINENO: checking for GRAPHVIZ" >&5
 echo $ECHO_N "checking for GRAPHVIZ... $ECHO_C" >&6; }
@@ -23597,6 +23598,8 @@
 echo "${ECHO_T}yes" >&6; }
 	have_graphviz=yes
 fi
+fi
+have_graphviz=no
 
 
 if test x"$have_graphviz" = "xyes"; then
@@ -24092,6 +24095,7 @@
 
 
 
+if false; then
 pkg_failed=no
 { echo "$as_me:$LINENO: checking for GDA_EVOLUTION" >&5
 echo $ECHO_N "checking for GDA_EVOLUTION... $ECHO_C" >&6; }
@@ -24167,6 +24171,8 @@
 	EVOLUTION_VERSION=`pkg-config --modversion libedataserver-1.2`
 
 fi
+fi
+have_evolution=no
 
 
 if test x"$have_evolution" = "xyes"; then
diff -Nurd libgnomedb-3.1.2.orig/glade3/Makefile.in libgnomedb-3.1.2/glade3/Makefile.in
--- libgnomedb-3.1.2.orig/glade3/Makefile.in	2007-10-24 16:22:18.000000000 -0400
+++ libgnomedb-3.1.2/glade3/Makefile.in	2008-11-06 04:27:02.000000000 -0500
@@ -62,7 +62,6 @@
 	$(am__DEPENDENCIES_1) \
 	$(top_builddir)/libgnomedb/libgnomedb-3.0.la \
 	$(top_builddir)/libgnomedb-extra/libgnomedb_extra-3.0.la \
-	$(top_builddir)/libgnomedb/plugins/libgnomedb_entry_builtin_plugins.la \
 	$(am__DEPENDENCIES_2)
 am_libgladegnomedb_la_OBJECTS = libgladegnomedb_la-glade3-gnomedb.lo
 libgladegnomedb_la_OBJECTS = $(am_libgladegnomedb_la_OBJECTS)
@@ -319,7 +318,7 @@
 @HAVE_LIBGNOMECANVAS_TRUE@graphlib = $(top_builddir)/libgnomedb-graph/libgnomedb_graph-3.0.la
 gladegnomedb_LTLIBRARIES = libgladegnomedb.la
 libgladegnomedb_la_SOURCES = glade3-gnomedb.c
-libgladegnomedb_la_CFLAGS = -g -Wall -I$(top_srcdir) -I$(top_builddir) \
+AM_CPPFLAGS = -g -Wall -I$(top_srcdir) -I$(top_builddir) \
                             -DG_LOG_DOMAIN=\"GnomeDB\" \
                             $(GLADE3_CFLAGS) $(LIBGNOMEDB_CFLAGS) $(LIBGLADE_CFLAGS)
 
@@ -328,7 +327,6 @@
 			    $(LIBGNOMEDB_LIBS) $(GLADE3_LIBS) \
 			    $(top_builddir)/libgnomedb/libgnomedb-3.0.la \
                             $(top_builddir)/libgnomedb-extra/libgnomedb_extra-3.0.la \
-			    $(top_builddir)/libgnomedb/plugins/libgnomedb_entry_builtin_plugins.la \
 			    $(graphlib)
 
 gladegnomedbdir = @GLADE3_MODULES_DIR@
