diff -Nurd monodevelop-1.0/Makefile.in monodevelop-1.0-new/Makefile.in
--- monodevelop-1.0/Makefile.in	2008-03-10 22:22:06.000000000 -0400
+++ monodevelop-1.0-new/Makefile.in	2008-04-29 21:32:47.000000000 -0400
@@ -263,7 +263,7 @@
 build_test_sources = $(addprefix $(srcdir)/, $(TESTFILES)) $(GENERATED_FILES)
 
 ### SYNCH BLOCK: any changes to this line should be kept in sync with the corresponding block in monodevelop.in and mdtool.in
-MD_PKG_CONFIG_PATH = $(gtksharp_prefix)/lib/pkgconfig/:/usr/lib/pkgconfig/:/usr/local/lib/pkgconfig/:/usr/share/pkgconfig/:/usr/local/share/pkgconfig/$(shell if test -d /usr/lib64; then echo ":/usr/lib64/pkgconfig/:/usr/local/lib64/pkgconfig/"; fi)
+MD_PKG_CONFIG_PATH = @FINKPREFIX@/lib/fontconfig2/lib/pkgconfig/:@FINKPREFIX@/lib/freetype219/lib/pkgconfig/:@FINKPREFIX@/lib/pango-ft219/lib/pkgconfig/:$(gtksharp_prefix)/lib/pkgconfig/:/usr/lib/pkgconfig/:/usr/local/lib/pkgconfig/:/usr/share/pkgconfig/:/usr/local/share/pkgconfig/$(shell if test -d /usr/lib64; then echo ":/usr/lib64/pkgconfig/:/usr/local/lib64/pkgconfig/"; fi)
 MD_BIN_PATH = $(abs_top_builddir)/build/bin
 MD_LAUNCH_SETUP = \
 	MONODEVELOP_LOCALE_PATH=$(abs_top_builddir)/build/locale \
diff -Nurd monodevelop-1.0/monodevelop.in monodevelop-1.0-new/monodevelop.in
--- monodevelop-1.0/monodevelop.in	2008-03-10 22:22:00.000000000 -0400
+++ monodevelop-1.0-new/monodevelop.in	2008-04-29 21:34:33.000000000 -0400
@@ -18,7 +18,7 @@
 
 
 #(NOTE: any changes to this block should be kept in sync with the one in configure.in)
-MOZFILE="libgtkembedmoz.so"
+MOZFILE="libgtkembedmoz.dylib"
 find_libgtkembedmoz ()
 {
 	MOZDIR=$MOZILLA_FIVE_HOME
@@ -55,14 +55,14 @@
 
 MD_BIN_PATH=@prefix@/lib/monodevelop/bin
 
-if [ -n $LD_LIBRARY_PATH ]; then
-	export LD_LIBRARY_PATH=$MOZILLA_HOME:$LD_LIBRARY_PATH
+if [ -n $DYLD_FALLBACK_LIBRARY_PATH ]; then
+	export DYLD_FALLBACK_LIBRARY_PATH=$MOZILLA_HOME:$DYLD_FALLBACK_LIBRARY_PATH
 else
-	export LD_LIBRARY_PATH=$MOZILLA_HOME
+	export DYLD_FALLBACK_LIBRARY_PATH=$MOZILLA_HOME
 fi
 
 ### SYNCH BLOCK: any changes to this block should be kept in sync with the one in Makefile.include and mdtool.in
-MD_PKG_CONFIG_PATH=@gtksharp_prefix@/lib/pkgconfig/:/usr/lib/pkgconfig/:/usr/local/lib/pkgconfig/:/usr/share/pkgconfig/:/usr/local/share/pkgconfig/
+MD_PKG_CONFIG_PATH=@FINKPREFIX@/lib/fontconfig2/lib/pkgconfig/:@FINKPREFIX@/lib/freetype219/lib/pkgconfig/:@FINKPREFIX@/lib/pango-ft219/lib/pkgconfig/:@gtksharp_prefix@/lib/pkgconfig/:/usr/lib/pkgconfig/:/usr/local/lib/pkgconfig/:/usr/share/pkgconfig/:/usr/local/share/pkgconfig/
 if test -d /usr/lib64; then MD_PKG_CONFIG_PATH=$MD_PKG_CONFIG_PATH:/usr/lib64/pkgconfig/:/usr/local/lib64/pkgconfig/; fi
 ### END BLOCK
 
