diff -ruN bowtie2-2.3.4-orig/Makefile bowtie2-2.3.4/Makefile
--- bowtie2-2.3.4-orig/Makefile	2017-12-29 10:08:31.000000000 -0600
+++ bowtie2-2.3.4/Makefile	2018-01-21 06:53:17.000000000 -0600
@@ -31,7 +31,7 @@
 CPP ?= $(GCC_PREFIX)/g++$(GCC_SUFFIX)
 CXX ?= $(CPP)
 
-HEADERS := $(wildcard *.h)
+HEADERS := $(sort $(wildcard *.h))
 BOWTIE_MM := 1
 BOWTIE_SHARED_MEM :=
 
@@ -164,7 +164,7 @@
 BUILD_CPPS := diff_sample.cpp
 BUILD_CPPS_MAIN := $(BUILD_CPPS) bowtie_build_main.cpp
 
-SEARCH_FRAGMENTS := $(wildcard search_*_phase*.c)
+SEARCH_FRAGMENTS := $(sort $(wildcard search_*_phase*.c))
 VERSION := $(shell cat VERSION)
 
 BITS := 32
@@ -226,14 +226,14 @@
                         bowtie2-inspect-s-sanitized \
                         bowtie2-inspect-l-sanitized
 
-GENERAL_LIST := $(wildcard scripts/*.sh) \
-                $(wildcard scripts/*.pl) \
+GENERAL_LIST := $(sort $(wildcard scripts/*.sh)) \
+                $(sort $(wildcard scripts/*.pl)) \
                 doc/manual.html \
                 doc/README \
                 doc/style.css \
-			    $(wildcard example/index/*.bt2) \
-			    $(wildcard example/reads/*.fq) \
-			    $(wildcard example/reads/*.pl) \
+			    $(sort $(wildcard example/index/*.bt2)) \
+			    $(sort $(wildcard example/reads/*.fq)) \
+			    $(sort $(wildcard example/reads/*.pl)) \
 			    example/reference/lambda_virus.fa \
                 $(PTHREAD_PKG) \
 			    bowtie2 \
@@ -260,11 +260,11 @@
 # the Windows FIND tool instead.
 FIND := $(shell which find)
 
-SRC_PKG_LIST = $(wildcard *.h) \
-               $(wildcard *.hh) \
-               $(wildcard *.c) \
-               $(wildcard *.cpp) \
-               $(wildcard third_party/*) \
+SRC_PKG_LIST = $(sort $(wildcard *.h)) \
+               $(sort $(wildcard *.hh)) \
+               $(sort $(wildcard *.c)) \
+               $(sort $(wildcard *.cpp)) \
+               $(sort $(wildcard third_party/*)) \
                Makefile \
                $(GENERAL_LIST)
 
@@ -284,9 +284,9 @@
 
 DEFS := -fno-strict-aliasing \
         -DBOWTIE2_VERSION="\"`cat VERSION`\"" \
-        -DBUILD_HOST="\"`hostname`\"" \
-        -DBUILD_TIME="\"`date`\"" \
-        -DCOMPILER_VERSION="\"`$(CXX) -v 2>&1 | tail -1`\"" \
+        -DBUILD_HOST="\"\"" \
+        -DBUILD_TIME="\"\"" \
+        -DCOMPILER_VERSION="\"\"" \
         $(FILE_FLAGS) \
         $(PREF_DEF) \
         $(MM_DEF) \
@@ -496,12 +496,30 @@
 MANUAL: MANUAL.markdown
 	pandoc -f markdown -t plain $^ -o $@
 
+.PHONY: man
+man:
+	help2man --name="wrapper for bowtie2-align" --no-info \
+		`pwd`/bowtie2 > bowtie2.1
+	help2man --name="building a colorspace index for bowtie2" --no-info \
+		`pwd`/bowtie2-build > bowtie2-build.1
+	help2man --name="extracts information from a bowtie2 index" --no-info \
+		`pwd`/bowtie2-inspect > bowtie2-inspect.1
+	help2man --name="ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences" --no-info \
+	    `pwd`/bowtie2-align-s> bowtie2-align.1
+
 .PHONY: install
-install: all
+install: all doc man
 	mkdir -p $(DESTDIR)$(bindir)
 	for file in $(BOWTIE2_BIN_LIST) bowtie2-inspect bowtie2-build bowtie2 ; do \
 		cp -f $$file $(DESTDIR)$(bindir) ; \
 	done
+	mkdir -p $(DESTDIR)$(prefix)/share/doc/bowtie2
+	cp -R doc/* $(DESTDIR)$(prefix)/share/doc/bowtie2
+	mkdir -p $(DESTDIR)$(prefix)/share/man/man1
+	cp bowtie2{,-build,-inspect,-align}.1 $(DESTDIR)$(prefix)/share/man/man1
+	mkdir -p $(DESTDIR)$(prefix)/share/bowtie2/examples
+	cp -R example/* $(DESTDIR)$(prefix)/share/bowtie2/examples
+	cp -R scripts $(DESTDIR)$(prefix)/share/bowtie2/examples
 
 .PHONY: simple-test
 simple-test: perl-deps both both-debug both-sanitized
diff -ruN bowtie2-2.3.4-orig/bt2_build.cpp bowtie2-2.3.4/bt2_build.cpp
--- bowtie2-2.3.4-orig/bt2_build.cpp	2017-12-29 10:08:31.000000000 -0600
+++ bowtie2-2.3.4/bt2_build.cpp	2018-01-21 06:50:26.000000000 -0600
@@ -548,15 +548,15 @@
 			} else {
 				cout << "Neither 32- nor 64-bit: sizeof(void*) = " << sizeof(void*) << endl;
 			}
-			cout << "Built on " << BUILD_HOST << endl;
-			cout << BUILD_TIME << endl;
-			cout << "Compiler: " << COMPILER_VERSION << endl;
-			cout << "Options: " << COMPILER_OPTIONS << endl;
-			cout << "Sizeof {int, long, long long, void*, size_t, off_t}: {"
-				 << sizeof(int)
-				 << ", " << sizeof(long) << ", " << sizeof(long long)
-				 << ", " << sizeof(void *) << ", " << sizeof(size_t)
-				 << ", " << sizeof(off_t) << "}" << endl;
+			//cout << "Built on " << BUILD_HOST << endl;
+			//cout << BUILD_TIME << endl;
+			//cout << "Compiler: " << COMPILER_VERSION << endl;
+			//cout << "Options: " << COMPILER_OPTIONS << endl;
+			//cout << "Sizeof {int, long, long long, void*, size_t, off_t}: {"
+			//	 << sizeof(int)
+			//	 << ", " << sizeof(long) << ", " << sizeof(long long)
+			//	 << ", " << sizeof(void *) << ", " << sizeof(size_t)
+			//	 << ", " << sizeof(off_t) << "}" << endl;
 			return 0;
 		}
 
diff -ruN bowtie2-2.3.4-orig/bt2_inspect.cpp bowtie2-2.3.4/bt2_inspect.cpp
--- bowtie2-2.3.4-orig/bt2_inspect.cpp	2017-12-29 10:08:31.000000000 -0600
+++ bowtie2-2.3.4/bt2_inspect.cpp	2018-01-21 06:50:12.000000000 -0600
@@ -443,15 +443,15 @@
 			} else {
 				cout << "Neither 32- nor 64-bit: sizeof(void*) = " << sizeof(void*) << endl;
 			}
-			cout << "Built on " << BUILD_HOST << endl;
-			cout << BUILD_TIME << endl;
-			cout << "Compiler: " << COMPILER_VERSION << endl;
-			cout << "Options: " << COMPILER_OPTIONS << endl;
-			cout << "Sizeof {int, long, long long, void*, size_t, off_t}: {"
-				 << sizeof(int)
-				 << ", " << sizeof(long) << ", " << sizeof(long long)
-				 << ", " << sizeof(void *) << ", " << sizeof(size_t)
-				 << ", " << sizeof(off_t) << "}" << endl;
+			//cout << "Built on " << BUILD_HOST << endl;
+			//cout << BUILD_TIME << endl;
+			//cout << "Compiler: " << COMPILER_VERSION << endl;
+			//cout << "Options: " << COMPILER_OPTIONS << endl;
+			//cout << "Sizeof {int, long, long long, void*, size_t, off_t}: {"
+			//	 << sizeof(int)
+			//	 << ", " << sizeof(long) << ", " << sizeof(long long)
+			//	 << ", " << sizeof(void *) << ", " << sizeof(size_t)
+			//	 << ", " << sizeof(off_t) << "}" << endl;
 			return 0;
 		}
 
diff -ruN bowtie2-2.3.4-orig/bt2_search.cpp bowtie2-2.3.4/bt2_search.cpp
--- bowtie2-2.3.4-orig/bt2_search.cpp	2017-12-29 10:08:31.000000000 -0600
+++ bowtie2-2.3.4/bt2_search.cpp	2018-01-21 06:49:50.000000000 -0600
@@ -5006,15 +5006,15 @@
 			} else {
 				cout << "Neither 32- nor 64-bit: sizeof(void*) = " << sizeof(void*) << endl;
 			}
-			cout << "Built on " << BUILD_HOST << endl;
-			cout << BUILD_TIME << endl;
-			cout << "Compiler: " << COMPILER_VERSION << endl;
-			cout << "Options: " << COMPILER_OPTIONS << endl;
-			cout << "Sizeof {int, long, long long, void*, size_t, off_t}: {"
-				 << sizeof(int)
-				 << ", " << sizeof(long) << ", " << sizeof(long long)
-				 << ", " << sizeof(void *) << ", " << sizeof(size_t)
-				 << ", " << sizeof(off_t) << "}" << endl;
+			//cout << "Built on " << BUILD_HOST << endl;
+			//cout << BUILD_TIME << endl;
+			//cout << "Compiler: " << COMPILER_VERSION << endl;
+			//cout << "Options: " << COMPILER_OPTIONS << endl;
+			//cout << "Sizeof {int, long, long long, void*, size_t, off_t}: {"
+			//	 << sizeof(int)
+			//	 << ", " << sizeof(long) << ", " << sizeof(long long)
+			//	 << ", " << sizeof(void *) << ", " << sizeof(size_t)
+			//	 << ", " << sizeof(off_t) << "}" << endl;
 			return 0;
 		}
 		{
