diff -ruN cell_id_1.4.6-orig/Makefile cell_id_1.4.6/Makefile
--- cell_id_1.4.6-orig/Makefile	2013-08-04 20:42:37.000000000 -0400
+++ cell_id_1.4.6/Makefile	2013-08-04 20:42:33.000000000 -0400
@@ -12,11 +12,11 @@
 # the folder\s that contain libtiff.a and libglib-2.0.dll.a 
 # (-L\local\lib).    
 
-CFLAGS = -Wall -static -I/local/include -I/local/include/glib-2.0 -L/local/lib
+CFLAGS = -Wall -I@FINKPREFIX@/include -I@FINKPREFIX@/include/glib-2.0 -I@FINKPREFIX@/lib/glib-2.0/include -L@FINKPREFIX@/lib
 
 # the order of the linker commands is important!
 cell : $(objects)
-	gcc -o cell $(CFLAGS) $(objects) -lm -ltiff -lglib-2.0
+	$(CC) -o cell $(CFLAGS) $(objects) -lm -ltiff -lglib-2.0
 
 cell.o: segment.h tif_routines.h date_and_time.h nums.h point.h image_type.h align_image.h split_and_overlap.h parameters.h file_type.h
 
@@ -48,5 +48,6 @@
 #oif.o: oif.h
 
 clean:
-	rm cell $(objects)
+	rm -f cell $(objects)
+
 
