diff -ruN -x finkbuild cernlib-cernlib-2025.09.18.4-free-orig/geant321/CMakeLists.txt cernlib-cernlib-2025.09.18.4-free/geant321/CMakeLists.txt
--- cernlib-cernlib-2025.09.18.4-free-orig/geant321/CMakeLists.txt	2025-09-21 08:17:35
+++ cernlib-cernlib-2025.09.18.4-free/geant321/CMakeLists.txt	2025-12-21 07:14:05
@@ -480,6 +487,7 @@
 target_include_directories(geant321 PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/")
 target_include_directories(geant321 PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/geant321")
 target_include_directories(geant321 PRIVATE ${FREETYPE_INCLUDE_DIRS})
+target_link_libraries(geant321 PRIVATE kernlib packlib graflib higz pawlib paw_motif code_motif)
 install(TARGETS geant321 DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libs)
 set_target_properties(geant321        PROPERTIES POSITION_INDEPENDENT_CODE ON OUTPUT_NAME geant321 SOVERSION 2_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR})
 install_symlink(libgeant321.so.2_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR} ${CMAKE_INSTALL_LIBDIR}/libgeant321.so.2_${COMPSUFFIX})
diff -ruN -x finkbuild cernlib-cernlib-2025.09.18.4-free-orig/graflib/CMakeLists.txt cernlib-cernlib-2025.09.18.4-free/graflib/CMakeLists.txt
--- cernlib-cernlib-2025.09.18.4-free-orig/graflib/CMakeLists.txt	2025-09-21 08:17:35
+++ cernlib-cernlib-2025.09.18.4-free/graflib/CMakeLists.txt	2025-12-20 21:30:59
@@ -75,7 +75,7 @@
 target_include_directories(graflib PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/dzdoc")
 target_include_directories(graflib PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/hplot")
 target_include_directories(graflib PRIVATE ${FREETYPE_INCLUDE_DIRS})
-target_link_libraries(graflib PRIVATE ${Xm})
+target_link_libraries(graflib PRIVATE ${Xm} kernlib packlib higz)
 install(TARGETS graflib DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libs)
 set_target_properties(graflib        PROPERTIES POSITION_INDEPENDENT_CODE ON OUTPUT_NAME graflib SOVERSION 1_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR})
 install_symlink(libgraflib.so.1_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR} ${CMAKE_INSTALL_LIBDIR}/libgraflib.so.1_${COMPSUFFIX})
diff -ruN -x finkbuild cernlib-cernlib-2025.09.18.4-free-orig/graflib/higz/CMakeLists.txt cernlib-cernlib-2025.09.18.4-free/graflib/higz/CMakeLists.txt
--- cernlib-cernlib-2025.09.18.4-free-orig/graflib/higz/CMakeLists.txt	2025-09-21 08:17:35
+++ cernlib-cernlib-2025.09.18.4-free/graflib/higz/CMakeLists.txt	2025-12-20 21:17:53
@@ -226,6 +226,7 @@
 target_include_directories(higz PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
 target_include_directories(higz PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/../../packlib/kuip")
 target_include_directories(higz PRIVATE ${FREETYPE_INCLUDE_DIRS})
+target_link_libraries(higz PRIVATE kernlib packlib ${X11_LIBRARIES})
 install(TARGETS higz DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libs)
 set_target_properties(higz        PROPERTIES POSITION_INDEPENDENT_CODE ON OUTPUT_NAME grafX11 SOVERSION 1_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR})
 install_symlink(libgrafX11.so.1_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR} ${CMAKE_INSTALL_LIBDIR}/libgrafX11.so.1_${COMPSUFFIX})
diff -ruN -x finkbuild cernlib-cernlib-2025.09.18.4-free-orig/lapack/CMakeLists.txt cernlib-cernlib-2025.09.18.4-free/lapack/CMakeLists.txt
--- cernlib-cernlib-2025.09.18.4-free-orig/lapack/CMakeLists.txt	2025-09-21 08:17:35
+++ cernlib-cernlib-2025.09.18.4-free/lapack/CMakeLists.txt	2025-12-20 20:59:33
@@ -187,7 +187,7 @@
   add_library(lapack SHARED ${lapack_FSRC} )
   target_include_directories(lapack PRIVATE ${PROJECT_SOURCE_DIR}/include)
   target_include_directories(lapack PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
-  target_link_libraries(lapack PRIVATE blas)
+  target_link_libraries(lapack PRIVATE blas lapack_static)
   install(TARGETS lapack DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libs)
   set_target_properties(lapack  PROPERTIES POSITION_INDEPENDENT_CODE ${CERNLIB_POSITION_INDEPENDENT_CODE} OUTPUT_NAME lapack SOVERSION 1_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR})
   install_symlink(liblapack.so.1_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR} ${CMAKE_INSTALL_LIBDIR}/liblapack.so.1_${COMPSUFFIX})
diff -ruN -x finkbuild cernlib-cernlib-2025.09.18.4-free-orig/mathlib/CMakeLists.txt cernlib-cernlib-2025.09.18.4-free/mathlib/CMakeLists.txt
--- cernlib-cernlib-2025.09.18.4-free-orig/mathlib/CMakeLists.txt	2025-09-21 08:17:35
+++ cernlib-cernlib-2025.09.18.4-free/mathlib/CMakeLists.txt	2025-12-20 20:48:15
@@ -418,6 +418,7 @@
 target_include_directories(mathlib PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
 target_include_directories(mathlib PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/gen/")
 target_include_directories(mathlib PRIVATE ${FREETYPE_INCLUDE_DIRS})
+target_link_libraries(mathlib PRIVATE kernlib packlib lapack)
 install(TARGETS mathlib DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libs)
 set_target_properties(mathlib        PROPERTIES POSITION_INDEPENDENT_CODE ON OUTPUT_NAME mathlib SOVERSION 2_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR})
 install_symlink(libmathlib.so.2_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR} ${CMAKE_INSTALL_LIBDIR}/libmathlib.so.2_${COMPSUFFIX})
diff -ruN -x finkbuild cernlib-cernlib-2025.09.18.4-free-orig/packlib/CMakeLists.txt cernlib-cernlib-2025.09.18.4-free/packlib/CMakeLists.txt
--- cernlib-cernlib-2025.09.18.4-free-orig/packlib/CMakeLists.txt	2025-09-21 08:17:35
+++ cernlib-cernlib-2025.09.18.4-free/packlib/CMakeLists.txt	2025-12-20 20:21:54
@@ -936,7 +936,7 @@
 target_include_directories(packlib PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/zbook")
 target_include_directories(packlib PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/zebra")
 target_include_directories(packlib PRIVATE ${FREETYPE_INCLUDE_DIRS})
-target_link_libraries(packlib PRIVATE ${libshift} ${libxrdposix} ${crypto})
+target_link_libraries(packlib PRIVATE ${libshift} ${libxrdposix} ${crypto} kernlib)
 install(TARGETS packlib DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libs)
 set_target_properties(packlib        PROPERTIES POSITION_INDEPENDENT_CODE ON OUTPUT_NAME packlib SOVERSION 1_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR})
 install_symlink(libpacklib.so.1_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR} ${CMAKE_INSTALL_LIBDIR}/libpacklib.so.1_${COMPSUFFIX})
diff -ruN -x finkbuild cernlib-cernlib-2025.09.18.4-free-orig/paw_motif/CMakeLists.txt cernlib-cernlib-2025.09.18.4-free/paw_motif/CMakeLists.txt
--- cernlib-cernlib-2025.09.18.4-free-orig/paw_motif/CMakeLists.txt	2025-09-21 08:17:35
+++ cernlib-cernlib-2025.09.18.4-free/paw_motif/CMakeLists.txt	2025-12-21 08:58:57
@@ -103,7 +103,7 @@
 target_include_directories(paw_motif PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
 target_include_directories(paw_motif PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/")
 target_include_directories(paw_motif PRIVATE ${FREETYPE_INCLUDE_DIRS})
-target_link_libraries(paw_motif PRIVATE ${X11_LIBRARIES})
+target_link_libraries(paw_motif PRIVATE ${X11_Xt_LIB} ${X11_LIBRARIES} kernlib packlib graflib Xm xbae_static higz pawlib code_motif)
 install(TARGETS paw_motif DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libs)
 set_target_properties(paw_motif        PROPERTIES POSITION_INDEPENDENT_CODE ON OUTPUT_NAME pawlib-lesstif SOVERSION 3_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR})
 install_symlink(libpawlib-lesstif.so.3_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR} ${CMAKE_INSTALL_LIBDIR}/libpawlib-lesstif.so.3_${COMPSUFFIX})
diff -ruN -x finkbuild cernlib-cernlib-2025.09.18.4-free-orig/pawlib/CMakeLists.txt cernlib-cernlib-2025.09.18.4-free/pawlib/CMakeLists.txt
--- cernlib-cernlib-2025.09.18.4-free-orig/pawlib/CMakeLists.txt	2025-09-21 08:17:35
+++ cernlib-cernlib-2025.09.18.4-free/pawlib/CMakeLists.txt	2025-12-20 21:48:16
@@ -443,6 +443,7 @@
 target_include_directories(pawlib PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/paw")
 target_include_directories(pawlib PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/sigma")
 target_include_directories(pawlib PRIVATE ${FREETYPE_INCLUDE_DIRS})
+target_link_libraries(pawlib PRIVATE kernlib packlib mathlib graflib lapack higz)
 install(TARGETS pawlib DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libs)
 set_target_properties(pawlib        PROPERTIES POSITION_INDEPENDENT_CODE ON OUTPUT_NAME pawlib SOVERSION 2_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR})
 install_symlink(libpawlib.so.2_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR} ${CMAKE_INSTALL_LIBDIR}/libpawlib.so.2_${COMPSUFFIX})
diff -ruN -x finkbuild cernlib-cernlib-2025.09.18.4-free-orig/phtools/CMakeLists.txt cernlib-cernlib-2025.09.18.4-free/phtools/CMakeLists.txt
--- cernlib-cernlib-2025.09.18.4-free-orig/phtools/CMakeLists.txt	2025-09-21 08:17:35
+++ cernlib-cernlib-2025.09.18.4-free/phtools/CMakeLists.txt	2025-12-21 05:53:47
@@ -57,6 +57,7 @@
 target_include_directories(phtools PRIVATE ${PROJECT_SOURCE_DIR}/include)
 target_include_directories(phtools PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
 target_include_directories(phtools PRIVATE ${FREETYPE_INCLUDE_DIRS})
+target_link_libraries(phtools PRIVATE kernlib mathlib)
 install(TARGETS phtools DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libs)
 set_target_properties(phtools        PROPERTIES POSITION_INDEPENDENT_CODE ON OUTPUT_NAME phtools SOVERSION 2_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR})
 install_symlink(libphtools.so.2_${COMPSUFFIX}.${CERNLIB_VERSION_MAJOR} ${CMAKE_INSTALL_LIBDIR}/libphtools.so.2_${COMPSUFFIX})
