diff -ruN chemtool-1.6.14-orig/Makefile.in chemtool-1.6.14/Makefile.in
--- chemtool-1.6.14-orig/Makefile.in	2013-08-12 16:42:51.000000000 -0500
+++ chemtool-1.6.14/Makefile.in	2018-01-11 05:50:17.000000000 -0600
@@ -81,7 +81,7 @@
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) -o $@
 SOURCES = $(chemtool_SOURCES) src-cht/cht.c
 DIST_SOURCES = $(chemtool_SOURCES) src-cht/cht.c
 RECURSIVE_TARGETS = all-recursive check-recursive dvi-recursive \
@@ -403,13 +403,13 @@
 	-test -z "$(bin_PROGRAMS)" || rm -f $(bin_PROGRAMS)
 chemtool$(EXEEXT): $(chemtool_OBJECTS) $(chemtool_DEPENDENCIES) 
 	@rm -f chemtool$(EXEEXT)
-	$(LINK) $(chemtool_OBJECTS) $(chemtool_LDADD) $(LIBS)
+	$(LINK) $(chemtool_OBJECTS) $(chemtool_LDADD) $(LDFLAGS) $(LIBS)
 src-cht/$(am__dirstamp):
 	@$(MKDIR_P) src-cht
 	@: > src-cht/$(am__dirstamp)
 src-cht/cht$(EXEEXT): $(src_cht_cht_OBJECTS) $(src_cht_cht_DEPENDENCIES) src-cht/$(am__dirstamp)
 	@rm -f src-cht/cht$(EXEEXT)
-	$(LINK) $(src_cht_cht_OBJECTS) $(src_cht_cht_LDADD) $(LIBS)
+	$(LINK) $(src_cht_cht_OBJECTS) $(src_cht_cht_LDADD) $(LDFLAGS) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
