diff -Nurd eccodes-2.21.0-Source-orig/fortran/CMakeLists.txt eccodes-2.21.0-Source/fortran/CMakeLists.txt
--- eccodes-2.21.0-Source-orig/fortran/CMakeLists.txt	2021-03-24 06:58:39.000000000 -0500
+++ eccodes-2.21.0-Source/fortran/CMakeLists.txt	2021-04-17 21:10:11.000000000 -0500
@@ -47,6 +47,8 @@
     ecbuild_add_library( TARGET          eccodes_f90
                          SOURCES         grib_fortran.c grib_f90.f90 eccodes_f90.f90 grib_kinds.h
                          GENERATED       grib_f90.f90 eccodes_f90.f90
+                         VERSION    0
+                         CFLAGS     ${CFLAGS}
                          PUBLIC_INCLUDES $<BUILD_INTERFACE:${CMAKE_Fortran_MODULE_DIRECTORY}>
                                          $<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>
                          PRIVATE_LIBS    eccodes )
diff -Nurd eccodes-2.21.0-Source-orig/src/CMakeLists.txt eccodes-2.21.0-Source/src/CMakeLists.txt
--- eccodes-2.21.0-Source-orig/src/CMakeLists.txt	2021-03-24 06:58:39.000000000 -0500
+++ eccodes-2.21.0-Source/src/CMakeLists.txt	2021-04-17 21:15:45.000000000 -0500
@@ -440,6 +440,8 @@
                               # griby.c gribl.c
                               ${grib_api_srcs}
                      GENERATED grib_api_version.c
+                     VERSION   0
+                     CFLAGS    ${CFLAGS}
                      #PRIVATE_LIBS      ${ECCODES_EXTRA_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${CMATH_LIBRARIES}
                      PRIVATE_INCLUDES "${AEC_INCLUDE_DIRS}"
                      PRIVATE_LIBS ${ECCODES_EXTRA_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${AEC_LIBRARIES} ${PNG_LIBRARIES}
