diff -Nurd eccodes-2.12.0-Source-orig/cmake/FindNetCDF.cmake eccodes-2.12.0-Source/cmake/FindNetCDF.cmake
--- eccodes-2.12.0-Source-orig/cmake/FindNetCDF.cmake	2019-02-15 10:44:50.000000000 +0100
+++ eccodes-2.12.0-Source/cmake/FindNetCDF.cmake	2019-03-25 15:00:24.069712123 +0100
@@ -78,7 +78,7 @@
 
   # Note: Only the HDF5 C-library is required for NetCDF
   #       ( even for Fortan and CXX bindings)
-  find_package( HDF5 COMPONENTS C QUIET )
+  # find_package( HDF5 COMPONENTS C QUIET )
 
   ## netcdf4
 
@@ -111,8 +111,8 @@
 
   if( NETCDF_FOUND AND HDF5_FOUND )
     # list( APPEND NETCDF_DEFINITIONS  ${HDF5_DEFINITIONS} )
-    list( APPEND NETCDF_LIBRARIES    ${HDF5_HL_LIBRARIES} ${HDF5_LIBRARIES}  )
-    list( APPEND NETCDF_INCLUDE_DIRS ${HDF5_INCLUDE_DIRS} )
+    # list( APPEND NETCDF_LIBRARIES    ${HDF5_HL_LIBRARIES} ${HDF5_LIBRARIES}  )
+    # list( APPEND NETCDF_INCLUDE_DIRS ${HDF5_INCLUDE_DIRS} )
   endif()
 
   #ecbuild_debug_var( NETCDF_FOUND )
diff -Nurd eccodes-2.12.0-Source-orig/cmake/contrib/FindNetCDF4.cmake eccodes-2.12.0-Source/cmake/contrib/FindNetCDF4.cmake
--- eccodes-2.12.0-Source-orig/cmake/contrib/FindNetCDF4.cmake	2019-02-15 10:44:50.000000000 +0100
+++ eccodes-2.12.0-Source/cmake/contrib/FindNetCDF4.cmake	2019-03-25 15:29:19.264443319 +0100
@@ -106,7 +106,7 @@
   set(HAS_HDF5 TRUE)
   set(HDF5_FIND_QUIETLY ${NETCDF_FIND_QUIETLY})
   set(HDF5_FIND_REQUIRED ${NETCDF_FIND_REQUIRED})
-  find_package(HDF5)
+#  find_package(HDF5)
 #        list( APPEND NETCDF_LIBRARIES_DEBUG
 #            ${HDF5_LIBRARIES_DEBUG} )
 #        list( APPEND NETCDF_LIBRARIES_RELEASE
diff -Nurd eccodes-2.12.0-Source-orig/fortran/CMakeLists.txt eccodes-2.12.0-Source/fortran/CMakeLists.txt
--- eccodes-2.12.0-Source-orig/fortran/CMakeLists.txt	2019-02-15 10:44:55.000000000 +0100
+++ eccodes-2.12.0-Source/fortran/CMakeLists.txt	2019-03-25 14:51:09.309393934 +0100
@@ -43,6 +43,8 @@
     ecbuild_add_library( TARGET     eccodes_f90
                          SOURCES    grib_fortran.c grib_f90.f90 eccodes_f90.f90 grib_kinds.h
                          GENERATED  grib_f90.f90 eccodes_f90.f90
+                         CFLAGS     ${CFLAGS}
+                         VERSION    0
                          LIBS       eccodes )
     add_custom_command( TARGET     eccodes_f90 POST_BUILD
                         COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_BINARY_DIR}/include
diff -Nurd eccodes-2.12.0-Source-orig/src/CMakeLists.txt eccodes-2.12.0-Source/src/CMakeLists.txt
--- eccodes-2.12.0-Source-orig/src/CMakeLists.txt	2019-02-15 10:44:55.000000000 +0100
+++ eccodes-2.12.0-Source/src/CMakeLists.txt	2019-03-25 14:51:09.310495327 +0100
@@ -435,6 +435,8 @@
                               # griby.c gribl.c
                              ${grib_api_srcs}
                     GENERATED grib_api_version.c
+                    VERSION   0
+                    CFLAGS    ${CFLAGS}
                     LIBS      ${ECCODES_EXTRA_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${CMATH_LIBRARIES}
                     TEMPLATES ${grib_api_extra_srcs})
 
