diff -ruN eccodes-2.30.0-Source-orig/fortran/CMakeLists.txt eccodes-2.30.0-Source/fortran/CMakeLists.txt
--- eccodes-2.30.0-Source-orig/fortran/CMakeLists.txt	2023-04-04 08:09:01.000000000 -0500
+++ eccodes-2.30.0-Source/fortran/CMakeLists.txt	2023-04-06 05:12:06.000000000 -0500
@@ -47,6 +47,8 @@
     ecbuild_add_library( TARGET          eccodes_f90
                          SOURCES         grib_fortran.c grib_f90.f90 eccodes_f90.f90 grib_kinds.h
                          GENERATED       grib_f90.f90 eccodes_f90.f90
+                         VERSION    0
+                         CFLAGS     ${CFLAGS}
                          PUBLIC_INCLUDES $<BUILD_INTERFACE:${CMAKE_Fortran_MODULE_DIRECTORY}>
                                          $<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>
                          PRIVATE_LIBS    eccodes ${ECCODES_PTHREADS_LIBRARIES} )
diff -ruN eccodes-2.30.0-Source-orig/src/CMakeLists.txt eccodes-2.30.0-Source/src/CMakeLists.txt
--- eccodes-2.30.0-Source-orig/src/CMakeLists.txt	2023-04-04 08:09:01.000000000 -0500
+++ eccodes-2.30.0-Source/src/CMakeLists.txt	2023-04-06 05:12:45.000000000 -0500
@@ -433,6 +433,8 @@
                      SOURCES   ${CMAKE_CURRENT_BINARY_DIR}/grib_api_version.cc
                               # griby.cc gribl.cc
                               ${eccodes_src_files}
+                     VERSION   0
+                     CFLAGS    ${CFLAGS}
                      #PRIVATE_LIBS      ${ECCODES_EXTRA_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${CMATH_LIBRARIES}
                      PRIVATE_INCLUDES "${AEC_INCLUDE_DIRS}" "${PNG_INCLUDE_DIRS}"
                      PRIVATE_LIBS ${ECCODES_EXTRA_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${AEC_LIBRARIES} ${PNG_LIBRARIES}
