diff -Nurd eccodes-2.12.0-Source-orig/cmake/FindNetCDF.cmake eccodes-2.12.0-Source/cmake/FindNetCDF.cmake
--- eccodes-2.12.0-Source-orig/cmake/FindNetCDF.cmake	2019-02-15 10:44:50.000000000 +0100
+++ eccodes-2.12.0-Source/cmake/FindNetCDF.cmake	2019-03-19 13:27:22.840759666 +0100
@@ -78,7 +78,7 @@
 
   # Note: Only the HDF5 C-library is required for NetCDF
   #       ( even for Fortan and CXX bindings)
-  find_package( HDF5 COMPONENTS C QUIET )
+  # find_package( HDF5 COMPONENTS C QUIET )
 
   ## netcdf4
 
diff -Nurd eccodes-2.12.0-Source-orig/fortran/CMakeLists.txt eccodes-2.12.0-Source/fortran/CMakeLists.txt
--- eccodes-2.12.0-Source-orig/fortran/CMakeLists.txt	2019-02-15 10:44:55.000000000 +0100
+++ eccodes-2.12.0-Source/fortran/CMakeLists.txt	2019-03-19 13:25:41.226730415 +0100
@@ -43,6 +43,8 @@
     ecbuild_add_library( TARGET     eccodes_f90
                          SOURCES    grib_fortran.c grib_f90.f90 eccodes_f90.f90 grib_kinds.h
                          GENERATED  grib_f90.f90 eccodes_f90.f90
+                         CFLAGS     ${CFLAGS}
+                         VERSION    0
                          LIBS       eccodes )
     add_custom_command( TARGET     eccodes_f90 POST_BUILD
                         COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_BINARY_DIR}/include
diff -Nurd eccodes-2.12.0-Source-orig/src/CMakeLists.txt eccodes-2.12.0-Source/src/CMakeLists.txt
--- eccodes-2.12.0-Source-orig/src/CMakeLists.txt	2019-02-15 10:44:55.000000000 +0100
+++ eccodes-2.12.0-Source/src/CMakeLists.txt	2019-03-19 13:25:41.229526778 +0100
@@ -435,6 +435,8 @@
                               # griby.c gribl.c
                              ${grib_api_srcs}
                     GENERATED grib_api_version.c
+                    VERSION   0
+                    CFLAGS    ${CFLAGS}
                     LIBS      ${ECCODES_EXTRA_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${CMATH_LIBRARIES}
                     TEMPLATES ${grib_api_extra_srcs})
 
