diff -Nurd fftw-3.3.10.orig/kernel/cycle.h fftw-3.3.10/kernel/cycle.h
--- fftw-3.3.10.orig/kernel/cycle.h	2023-07-16 16:02:45
+++ fftw-3.3.10/kernel/cycle.h	2023-07-16 16:04:25
@@ -539,7 +539,7 @@
 #define HAVE_TICK_COUNTER
 #endif
 
-#if defined(__aarch64__) && defined(HAVE_ARMV8_CNTVCT_EL0) && !defined(HAVE_ARMV8_PMCCNTR_EL0)
+#if defined(__aarch64__) && (defined(__APPLE__) || defined(HAVE_ARMV8_CNTVCT_EL0)) && !defined(HAVE_ARMV8_PMCCNTR_EL0)
 typedef uint64_t ticks;
 static inline ticks getticks(void)
 {
