diff --git a/configure b/configure
index 424ab0a..52ab48a 100755
--- a/configure
+++ b/configure
@@ -433,7 +433,7 @@ if [ -z "$FLINT_LIB" ]; then
    case "$OS" in
       Darwin)
          FLINT_LIBNAME="libflint.dylib"
-         FLINT_LIB="libflint-$FLINT_MAJOR.$FLINT_MINOR.$FLINT_PATCH.dylib"
+         FLINT_LIB="libflint-$FLINT_MAJOR.dylib"
          EXTRA_SHARED_FLAGS="-install_name $PREFIX/lib/$FLINT_LIB -compatibility_version $FLINT_MAJOR.$FLINT_MINOR -current_version $FLINT_MAJOR.$FLINT_MINOR.$FLINT_PATCH -Wl,-rpath,$GMP_LIB_DIR -Wl,-rpath,$MPFR_LIB_DIR";;
       CYGWIN* | MINGW*)
          FLINT_LIBNAME="libflint.dll"
