diff -ruN gdcm-2.6.9.orig/Utilities/doxygen/CMakeLists.txt gdcm-2.6.9/Utilities/doxygen/CMakeLists.txt
--- gdcm-2.6.9.orig/Utilities/doxygen/CMakeLists.txt	2017-06-29 03:22:56.000000000 -0500
+++ gdcm-2.6.9/Utilities/doxygen/CMakeLists.txt	2018-03-14 04:55:14.000000000 -0500
@@ -267,8 +267,8 @@
 
       set(DOCBOOK_REWRITE_PREFIX_FOUND FALSE)
       foreach(line ${xml_catalog})
-        string(REGEX MATCH "^.*systemIdStartString=\"http://docbook.sourceforge.net/release/xsl-ns/\".*" out1 ${line})
-        string(REGEX REPLACE "^.*systemIdStartString=\"http://docbook.sourceforge.net/release/xsl-ns/\".*\"([A-Za-z ]*)\".*" "\\1" output_variable ${line})
+        string(REGEX MATCH "^.*systemIdStartString=\"http://docbook.sourceforge.net/release/xsl-ns/current\".*" out1 ${line})
+        string(REGEX REPLACE "^.*systemIdStartString=\"http://docbook.sourceforge.net/release/xsl-ns/current\".*\"([A-Za-z ]*)\".*" "\\1" output_variable ${line})
         if(out1)
           message(STATUS "Your docbook install was found here :${output_variable}")
           set(DOCBOOK_REWRITE_PREFIX_FOUND TRUE)
diff -ruN gdcm-2.6.9.orig/Utilities/gdcmjpeg/jdcoefct.c gdcm-2.6.9/Utilities/gdcmjpeg/jdcoefct.c
--- gdcm-2.6.9.orig/Utilities/gdcmjpeg/jdcoefct.c	2017-06-29 03:22:56.000000000 -0500
+++ gdcm-2.6.9/Utilities/gdcmjpeg/jdcoefct.c	2018-03-14 04:22:09.000000000 -0500
@@ -418,7 +418,7 @@
   int * coef_bits;
   int * coef_bits_latch;
 
-  if (! cinfo->process == JPROC_PROGRESSIVE || cinfo->coef_bits == NULL)
+  if ( cinfo->process != JPROC_PROGRESSIVE || cinfo->coef_bits == NULL)
     return FALSE;
 
   /* Allocate latch area if not already done */
