diff -Nurd -x'*~' glpk-4.55.orig/src/Makefile.in glpk-4.55/src/Makefile.in
--- glpk-4.55.orig/src/Makefile.in	2014-08-22 02:00:00.000000000 -0400
+++ glpk-4.55/src/Makefile.in	2015-05-17 18:20:51.000000000 -0400
@@ -314,6 +314,7 @@
 -I$(srcdir)/zlib
 
 libglpk_la_LDFLAGS = \
+-no-undefined \
 -version-info 37:2:1 \
 -export-symbols-regex '^glp_*'
 
diff -Nurd -x'*~' glpk-4.55.orig/src/env/env.h glpk-4.55/src/env/env.h
--- glpk-4.55.orig/src/env/env.h	2014-08-22 02:00:00.000000000 -0400
+++ glpk-4.55/src/env/env.h	2015-05-17 18:19:30.000000000 -0400
@@ -29,8 +29,6 @@
 typedef struct ENV ENV;
 typedef struct MBD MBD;
 
-#define SIZE_T_MAX (~(size_t)0)
-/* largest value of size_t type */
 
 #define TBUF_SIZE 4096
 /* terminal output buffer size, in bytes */
