diff -ruN grass-7.8.5-orig/gui/wxpython/vnet/dialogs.py grass-7.8.5/gui/wxpython/vnet/dialogs.py
--- grass-7.8.5-orig/gui/wxpython/vnet/dialogs.py	2020-12-21 13:40:15.000000000 -0500
+++ grass-7.8.5/gui/wxpython/vnet/dialogs.py	2021-01-11 12:38:43.000000000 -0500
@@ -1378,7 +1378,8 @@
         gridSizer = wx.GridBagSizer(vgap=1, hgap=1)
 
         row = 0
-        setts = {**self.colorsSetts, **self.sizeSetts}
+        setts = self.colorsSetts.copy()
+        setts.update(self.sizeSetts)
 
         settsOrder = [
             "selected",
diff -ruN grass-7.8.5-orig/include/Make/HtmlRules.make grass-7.8.5/include/Make/HtmlRules.make
--- grass-7.8.5-orig/include/Make/HtmlRules.make	2020-12-21 13:40:15.000000000 -0500
+++ grass-7.8.5/include/Make/HtmlRules.make	2021-01-11 13:45:31.000000000 -0500
@@ -1,7 +1,7 @@
 
 # common html rules (included by Html.make and GuiScript.make)
 
-htmldesc = $(call run_grass,$(1) --html-description < /dev/null | grep -v '</body>\|</html>' > $(2))
+htmldesc = $(call run_grass,$(1) --html-description < /dev/null | grep -v '</body>\|</html>' || > $(2))
 
 IMGSRC := $(wildcard *.png) $(wildcard *.jpg) $(wildcard *.gif)
 IMGDST := $(patsubst %,$(HTMLDIR)/%,$(IMGSRC))
diff -ruN grass-7.8.5-orig/lib/python/ctypes/ctypesgencore/parser/lex.py grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/lex.py
--- grass-7.8.5-orig/lib/python/ctypes/ctypesgencore/parser/lex.py	2020-12-21 13:40:15.000000000 -0500
+++ grass-7.8.5/lib/python/ctypes/ctypesgencore/parser/lex.py	2021-01-11 12:37:28.000000000 -0500
@@ -735,8 +735,8 @@
         if os.sys.version_info.major >= 3:
             f.sort(key=lambda x: get_func_code(x[1]).co_firstlineno)
         else:
-            f.sort(key=lambda x, y: cmp(get_func_code(x[1]).co_firstlineno,
-                                        get_func_code(y[1]).co_firstlineno))
+            f.sort(lambda x, y: cmp(get_func_code(x[1]).co_firstlineno,
+                                    get_func_code(y[1]).co_firstlineno))
 
     # Sort the strings by regular expression length
     for s in strsym.values():
@@ -745,8 +745,8 @@
                                             (len(x[1]) < len(y[1])) -
                                             (len(x[1]) > len(y[1]))))
         else:
-            s.sort(key=lambda x, y: (len(x[1]) < len(y[1])) -
-                                    (len(x[1]) > len(y[1])))
+            s.sort(lambda x, y: (len(x[1]) < len(y[1])) -
+                                (len(x[1]) > len(y[1])))
 
     regexs = {}
 
