diff -ruN grass-7.2.0-orig/configure grass-7.2.0/configure
--- grass-7.2.0-orig/configure	2021-02-04 17:32:12.000000000 +0900
+++ grass-7.2.0/configure	2021-02-04 17:33:29.000000000 +0900
@@ -5712,41 +5712,41 @@
 # nad2bin needs to be in path for installing datum gridshift files
 
 # Extract the first word of "nad2bin", so it can be a program name with args.
-set dummy nad2bin; ac_word=$2
-echo $ac_n "checking for $ac_word""... $ac_c" 1>&6
-echo "configure:5718: checking for $ac_word" >&5
-
-case "$NAD2BIN" in
-  /*)
-  ac_cv_path_NAD2BIN="$NAD2BIN" # Let the user override the test with a path.
-  ;;
-  ?:/*)			 
-  ac_cv_path_NAD2BIN="$NAD2BIN" # Let the user override the test with a dos path.
-  ;;
-  *)
-  IFS="${IFS= 	}"; ac_save_ifs="$IFS"; IFS=":"
-  ac_dummy="$PATH"
-  for ac_dir in $ac_dummy; do 
-    test -z "$ac_dir" && ac_dir=.
-    if test -f $ac_dir/$ac_word; then
-      ac_cv_path_NAD2BIN="$ac_dir/$ac_word"
-      break
-    fi
-  done
-  IFS="$ac_save_ifs"
-  test -z "$ac_cv_path_NAD2BIN" && ac_cv_path_NAD2BIN="no"
-  ;;
-esac
-NAD2BIN="$ac_cv_path_NAD2BIN"
-if test -n "$NAD2BIN"; then
-  echo "$ac_t""$NAD2BIN" 1>&6
-else
-  echo "$ac_t""no" 1>&6
-fi
+#set dummy nad2bin; ac_word=$2
+#echo $ac_n "checking for $ac_word""... $ac_c" 1>&6
+#echo "configure:5718: checking for $ac_word" >&5
+
+#case "$NAD2BIN" in
+#  /*)
+#  ac_cv_path_NAD2BIN="$NAD2BIN" # Let the user override the test with a path.
+#  ;;
+#  ?:/*)			 
+#  ac_cv_path_NAD2BIN="$NAD2BIN" # Let the user override the test with a dos path.
+#  ;;
+#  *)
+#  IFS="${IFS= 	}"; ac_save_ifs="$IFS"; IFS=":"
+#  ac_dummy="$PATH"
+#  for ac_dir in $ac_dummy; do 
+#    test -z "$ac_dir" && ac_dir=.
+#    if test -f $ac_dir/$ac_word; then
+#      ac_cv_path_NAD2BIN="$ac_dir/$ac_word"
+#      break
+#    fi
+#  done
+#  IFS="$ac_save_ifs"
+#  test -z "$ac_cv_path_NAD2BIN" && ac_cv_path_NAD2BIN="no"
+#  ;;
+#esac
+#NAD2BIN="$ac_cv_path_NAD2BIN"
+#if test -n "$NAD2BIN"; then
+#  echo "$ac_t""$NAD2BIN" 1>&6
+#else
+#  echo "$ac_t""no" 1>&6
+#fi
 
-if test "$NAD2BIN" = "no"; then
-    { echo "configure: error: *** Unable to locate nad2bin (needs to be in PATH)." 1>&2; exit 1; }
-fi
+#if test "$NAD2BIN" = "no"; then
+#    { echo "configure: error: *** Unable to locate nad2bin (needs to be in PATH)." 1>&2; exit 1; }
+#fi
 
 # Done checking PROJ
 
diff -ruN grass-7.2.0-orig/configure.in grass-7.2.0/configure.in
--- grass-7.2.0-orig/configure.in	2021-02-04 17:32:12.000000000 +0900
+++ grass-7.2.0/configure.in	2021-02-04 17:32:52.000000000 +0900
@@ -684,10 +684,10 @@
 
 # nad2bin needs to be in path for installing datum gridshift files
 
-AC_PATH_PROG(NAD2BIN, nad2bin, no)
-if test "$NAD2BIN" = "no"; then
-    AC_MSG_ERROR([*** Unable to locate nad2bin (needs to be in PATH).])
-fi
+#AC_PATH_PROG(NAD2BIN, nad2bin, no)
+#if test "$NAD2BIN" = "no"; then
+#    AC_MSG_ERROR([*** Unable to locate nad2bin (needs to be in PATH).])
+#fi
 
 # Done checking PROJ
 
