diff -uNr gromacs-5.0.5.orig/CMakeLists.txt gromacs-5.0.5/CMakeLists.txt
--- gromacs-5.0.5.orig/CMakeLists.txt	2015-05-13 08:43:06.000000000 -0400
+++ gromacs-5.0.5/CMakeLists.txt	2015-05-19 18:54:17.000000000 -0400
@@ -487,6 +487,7 @@
 # tests. This exports LIBXML2_FOUND, which we should not use because
 # it does not tell us that linking will succeed. Instead, we test that
 # next.
+set(LIBXML2_LIBRARIES @FINKPREFIX@/lib/libxml2.dylib)
 if(DEFINED LIBXML2_LIBRARIES)
   set(LibXml2_FIND_QUIETLY TRUE)
 endif()
