diff -wurb HDF4.2r4/hdf/src/hdfi.h HDF4.2r4-rkg/hdf/src/hdfi.h
--- HDF4.2r4/hdf/src/hdfi.h	2007-12-19 14:16:09.000000000 -0600
+++ HDF4.2r4-rkg/hdf/src/hdfi.h	2009-10-28 15:00:10.000000000 -0500
@@ -766,7 +766,7 @@
 #include <sys/types.h>
 #include <sys/file.h>               /* for unbuffered i/o stuff */
 #include <sys/stat.h>
-#ifdef __i386
+#if defined(__i386) || defined(__x86_64__)
 #ifndef INTEL86
 #define INTEL86   /* we need this Intel define or bad things happen later */
 #endif /* INTEL86 */
@@ -775,6 +775,7 @@
 #define DF_MT   DFMT_APPLE
 #endif /* __i386 */
 
+#if defined(__i386) && !defined(__x86_64__)
 typedef void            VOID;
 typedef void            *VOIDP;
 typedef char            *_fcd;
@@ -792,6 +793,25 @@
 typedef double          float64;
 typedef int             intf;     /* size of INTEGERs in Fortran compiler */
 typedef int               hdf_pint_t;   /* an integer the same size as a pointer */
+#elif defined(__x86_64__)
+typedef void              VOID;
+typedef void              *VOIDP;
+typedef char              *_fcd;
+typedef char              char8;
+typedef unsigned char     uchar8;
+typedef char              int8;
+typedef unsigned char     uint8;
+typedef short int         int16;
+typedef unsigned short int uint16;
+typedef int               int32;
+typedef unsigned int      uint32;
+typedef int               intn;
+typedef unsigned int      uintn;
+typedef int               intf;     /* size of INTEGERs in Fortran compiler */
+typedef float             float32;
+typedef double            float64;
+typedef long              hdf_pint_t;   /* an integer the same size as a pointer */
+#endif
 #define _fcdtocp(desc) (desc)
 #ifdef  HAVE_FMPOOL
 #define FILELIB PAGEBUFIO  /* enable page buffering */
@@ -1346,7 +1366,7 @@
 
 #endif /*power PC 5 64 */
 /* Linux 64 */
-#if defined __x86_64__  && !(defined  SUN) /* i.e. not SunOS on Intel */
+#if defined __x86_64__  && !(defined  SUN) && !(defined __APPLE__) /* i.e. not SunOS on Intel */
 
 #ifdef GOT_MACHINE
 If you get an error on this line more than one machine type has been defined.
