diff -ruN hdf5-1.10.0-orig/configure hdf5-1.10.0/configure
--- hdf5-1.10.0-orig/configure	2016-03-28 11:28:21.000000000 -0500
+++ hdf5-1.10.0/configure	2021-10-26 20:32:01.000000000 -0500
@@ -27573,7 +27573,7 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
+#include <stdlib.h>
                 #include "szlib.h"
 
 #ifdef FC_DUMMY_MAIN
@@ -30314,7 +30314,7 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
+#include <stdlib.h>
                 #include <string.h>
 
 #ifdef FC_DUMMY_MAIN
@@ -30432,7 +30432,7 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
+#include <stdlib.h>
                 #include <string.h>
 
 #ifdef FC_DUMMY_MAIN
@@ -30556,6 +30556,8 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <string.h>
+#include <stdlib.h>
 
         int main(void)
         {
@@ -30645,6 +30647,8 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <string.h>
+#include <stdlib.h>
 
         int main(void)
         {
diff -ruN hdf5-1.10.0-orig/src/H5Dio.c hdf5-1.10.0/src/H5Dio.c
--- hdf5-1.10.0-orig/src/H5Dio.c	2016-03-28 11:26:25.000000000 -0500
+++ hdf5-1.10.0/src/H5Dio.c	2021-10-26 20:28:19.000000000 -0500
@@ -74,6 +74,8 @@
 #endif /* H5_HAVE_PARALLEL */
 static herr_t H5D__typeinfo_term(const H5D_type_info_t *type_info);
 
+herr_t
+H5T_patch_vlen_file(H5T_t *dt, H5F_t *f);
 
 /*********************/
 /* Package Variables */
