diff -ruN hdf5-1.8.14-orig/hl/src/H5LTanalyze.c hdf5-1.8.14/hl/src/H5LTanalyze.c
--- hdf5-1.8.14-orig/hl/src/H5LTanalyze.c	2014-11-07 06:53:42.000000000 -0600
+++ hdf5-1.8.14/hl/src/H5LTanalyze.c	2021-10-26 06:49:28.000000000 -0500
@@ -40,6 +40,7 @@
 #include <string.h>
 #include <errno.h>
 #include <stdlib.h>
+#include <unistd.h>
 
 /* end standard C headers. */
 
diff -ruN hdf5-1.8.14-orig/test/cache.c hdf5-1.8.14/test/cache.c
--- hdf5-1.8.14-orig/test/cache.c	2014-10-13 06:33:37.000000000 -0500
+++ hdf5-1.8.14/test/cache.c	2021-10-26 06:27:48.000000000 -0500
@@ -200,6 +200,11 @@
 static unsigned check_auto_cache_resize_aux_fcns(void);
 static unsigned check_metadata_blizzard_absence(hbool_t fill_via_insertion);
 
+hbool_t
+resize_configs_are_equal(const H5C_auto_size_ctl_t *a,
+    const H5C_auto_size_ctl_t *b,
+    hbool_t compare_init);
+
 
 /**************************************************************************/
 /**************************************************************************/
diff -ruN hdf5-1.8.14-orig/test/cache_api.c hdf5-1.8.14/test/cache_api.c
--- hdf5-1.8.14-orig/test/cache_api.c	2014-10-13 06:33:37.000000000 -0500
+++ hdf5-1.8.14/test/cache_api.c	2021-10-26 06:27:19.000000000 -0500
@@ -42,6 +42,11 @@
 
 static void check_file_mdc_api_errs(void);
 
+hbool_t
+resize_configs_are_equal(const H5C_auto_size_ctl_t *a,
+    const H5C_auto_size_ctl_t *b,
+    hbool_t compare_init);
+
 
 /**************************************************************************/
 /**************************************************************************/
