--- lapack-3.7.1-orig/LAPACKE/src/Makefile	2017-06-18 00:46:53.000000000 +0200
+++ lapack-3.7.1/LAPACKE/src/Makefile	2017-10-10 17:22:22.081979028 +0200
@@ -2412,9 +2412,15 @@
 ../../$(LAPACKELIB): $(OBJ_A) $(OBJ_B) $(DEPRECATED) $(EXTENDED) $(MATGEN)
 	$(ARCH) $(ARCHFLAGS) $@ $(OBJ_A)
 	$(ARCH) $(ARCHFLAGS) $@ $(OBJ_B)
+ifdef BUILD_DEPRECATED
 	$(ARCH) $(ARCHFLAGS) $@ $(DEPRECATED)
+endif
+ifdef USEEXBLAS
 	$(ARCH) $(ARCHFLAGS) $@ $(EXTENDED)
+endif
+ifdef LAPACKE_WITH_TMG
 	$(ARCH) $(ARCHFLAGS) $@ $(MATGEN)
+endif
 	$(RANLIB) $@
 
 clean: cleanobj
