diff -Nurd coordgenlibs-3.0.2/CMakeLists.txt coordgenlibs-55d9985/CMakeLists.txt
--- coordgenlibs-3.0.2/CMakeLists.txt	2023-02-01 15:47:00
+++ coordgenlibs-55d9985/CMakeLists.txt	2025-06-26 13:56:58
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.2)
-project(coordgen)
+project(coordgen VERSION 3.0.2)
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
@@ -86,10 +86,14 @@
 
 set_target_properties(coordgen
     PROPERTIES
-        VERSION 3.0.2
+        VERSION ${PROJECT_VERSION}
         SOVERSION 3
 )
 
+target_include_directories(coordgen PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+    $<INSTALL_INTERFACE:include>)
+
 if(MSVC)
     set(CMAKE_INSTALL_LIBDIR lib)
     set(CMAKE_INSTALL_BINDIR bin)
@@ -129,6 +133,12 @@
 
 install(EXPORT coordgen-targets
     FILE ${PROJECT_NAME}-config.cmake
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake)
+
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config-version.cmake
+    COMPATIBILITY AnyNewerVersion)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config-version.cmake
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake)
 
 # Example
diff -Nurd coordgenlibs-3.0.2/CoordgenMacrocycleBuilder.h coordgenlibs-55d9985/CoordgenMacrocycleBuilder.h
--- coordgenlibs-3.0.2/CoordgenMacrocycleBuilder.h	2023-02-01 15:47:00
+++ coordgenlibs-55d9985/CoordgenMacrocycleBuilder.h	2025-06-26 13:56:58
@@ -49,11 +49,11 @@
     }
     bool operator<(const vertexCoords& rhs) const
     {
-        if (x < rhs.x) {
-            return true;
+        if (x != rhs.x) {
+            return (x < rhs.x);
         }
-        if (y < rhs.y) {
-            return true;
+        if (y != rhs.y) {
+            return (y < rhs.y);
         }
         if (z < rhs.z) {
             return true;
diff -Nurd coordgenlibs-3.0.2/test/coordgenBasicSMILES.h coordgenlibs-55d9985/test/coordgenBasicSMILES.h
--- coordgenlibs-3.0.2/test/coordgenBasicSMILES.h	2023-02-01 15:47:00
+++ coordgenlibs-55d9985/test/coordgenBasicSMILES.h	2025-06-26 13:56:58
@@ -30,7 +30,6 @@
     std::unordered_map<char, sketcherMinimizerAtom*> cycles;
     int bond_order = 1;
 
-    size_t idx = 0;
     for (auto c : smiles) {
         auto atomic_number = elements.find(c);
         if (atomic_number != elements.end()) {
@@ -69,7 +68,6 @@
             msg += " in SMILES: " + smiles;
             throw std::runtime_error(msg);
         }
-        ++idx;
     }
 
     sketcherMinimizerMolecule::assignBondsAndNeighbors(mol->getAtoms(),
diff -Nurd coordgenlibs-3.0.2/test/test_coordgen.cpp coordgenlibs-55d9985/test/test_coordgen.cpp
--- coordgenlibs-3.0.2/test/test_coordgen.cpp	2023-02-01 15:47:00
+++ coordgenlibs-55d9985/test/test_coordgen.cpp	2025-06-26 13:56:58
@@ -310,7 +310,6 @@
     minimizer.initialize(mol); // minimizer takes ownership of mol
     minimizer.runGenerateCoordinates();
     //check the length of every non macrocycle-ring bond
-    int bondsN = 0;
     for (auto interaction : minimizer.getStretchInteractions()) {
         auto ring = sketcherMinimizer::sameRing(interaction->atom1, interaction->atom2);
         if (ring && !ring->isMacrocycle()) {
@@ -318,10 +317,9 @@
             auto tolerance = 2.f;
             auto bondLength = (interaction->atom1->coordinates - interaction->atom2->coordinates).length();
             BOOST_REQUIRE ((bondLength > expectedLength-tolerance) && (bondLength < expectedLength+tolerance));
-            bondsN++;
         }
     }
-//    check the angles
+    // check the angles
     int anglesN = 0;
     for (auto interaction : minimizer.getBendInteractions()) {
         if (interaction->isRing) {
