diff -Nurd -x'*~' libdap-3.12.1.orig/DDXParserSAX2.cc libdap-3.12.1/DDXParserSAX2.cc
--- libdap-3.12.1.orig/DDXParserSAX2.cc	2013-10-01 12:05:22.000000000 -0400
+++ libdap-3.12.1/DDXParserSAX2.cc	2017-10-08 22:12:23.000000000 -0400
@@ -1237,7 +1237,7 @@
         context->validate = true;
 
 
-        while ((fgets(chars, size, in) > 0) && !is_boundary(chars, boundary)) {
+        while ((fgets(chars, size, in) != 0) && !is_boundary(chars, boundary)) {
             //chars[size-1] = '\0';
             DBG(cerr << "line (" << strlen(chars) << "): " << chars << endl);
 
diff -Nurd -x'*~' libdap-3.12.1.orig/dap-config-pkgconfig libdap-3.12.1/dap-config-pkgconfig
--- libdap-3.12.1.orig/dap-config-pkgconfig	2013-07-10 07:38:10.000000000 -0400
+++ libdap-3.12.1/dap-config-pkgconfig	2017-10-08 22:11:41.000000000 -0400
@@ -2,11 +2,11 @@
 #
 # Borrowed the idea for this script (and some code) from libcurl.
 #
-
+export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:@FINKPREFIX@/opt/libdap@DAPVERS@/lib/pkgconfig
 usage()
 {
     cat <<EOF
-Usage: dap-config [OPTION]
+Usage: dap-config-pkgconfig [OPTION]
 
 Available values for OPTION include:
 
diff -Nurd -x'*~' libdap-3.12.1.orig/dap-config.in libdap-3.12.1/dap-config.in
--- libdap-3.12.1.orig/dap-config.in	2013-07-10 07:38:10.000000000 -0400
+++ libdap-3.12.1/dap-config.in	2021-01-23 15:59:29.000000000 -0500
@@ -4,9 +4,9 @@
 #
 prefix=@prefix@
 exec_prefix=${prefix}
-libdir=${exec_prefix}/lib
-libdir64=${exec_prefix}/lib64
-includedir=${prefix}/include
+dapdir=@FINKPREFIX@/opt/libdap@DAPVERS@
+libdir=${dapdir}/lib
+includedir=${dapdir}/include
 
 usage()
 {
@@ -60,7 +60,7 @@
 	;;
 
     --libs)
-       	echo "-L${libdir64} -L${libdir} -ldap -ldapserver -ldapclient @CURL_LIBS@ @XML2_LIBS@ @PTHREAD_LIBS@ @UUID_LIBS@ @LIBS@"
+       	echo "-L${libdir} -ldap -ldapserver -ldapclient @CURL_LIBS@ @XML2_LIBS@ @PTHREAD_LIBS@ @UUID_LIBS@ @LIBS@"
         ;;
 #
 #   Changed CURL_STATIC_LIBS to CURL_LIBS because the former was including a
@@ -69,11 +69,11 @@
 #   jhrg 2/7/12
 
     --server-libs)
-       	echo "-L${libdir64} -L${libdir} -ldap -ldapserver @XML2_LIBS@ @PTHREAD_LIBS@ @UUID_LIBS@ @LIBS@"
+       	echo "-L${libdir} -ldap -ldapserver @XML2_LIBS@ @PTHREAD_LIBS@ @UUID_LIBS@ @LIBS@"
        	;;
 
     --client-libs)
-       	echo "-L${libdir64} -L${libdir} -ldap -ldapclient @CURL_LIBS@ @XML2_LIBS@ @PTHREAD_LIBS@ @UUID_LIBS@ @LIBS@"
+       	echo "-L${libdir} -ldap -ldapclient @CURL_LIBS@ @XML2_LIBS@ @PTHREAD_LIBS@ @UUID_LIBS@ @LIBS@"
        	;;
 
     --prefix)
