diff -Nurd openbabel-3.1.1.orig/CMakeLists.txt openbabel-3.1.1/CMakeLists.txt
--- openbabel-3.1.1.orig/CMakeLists.txt	2020-05-08 11:41:00.000000000 -0400
+++ openbabel-3.1.1/CMakeLists.txt	2021-03-04 02:57:13.000000000 -0500
@@ -176,7 +176,7 @@
 
 # wxWidgets instructions based on http://wiki.wxwidgets.org/CMake
 #find_package(wxWidgets COMPONENTS base core REQUIRED)
-find_package(wxWidgets COMPONENTS base core adv)
+#find_package(wxWidgets COMPONENTS base core adv)
 if(wxWidgets_FOUND)
   include(${wxWidgets_USE_FILE})
   add_definitions(-DHAVE_WXWIDGETS)
diff -Nurd openbabel-3.1.1.orig/cmake/modules/Findcoordgen.cmake openbabel-3.1.1/cmake/modules/Findcoordgen.cmake
--- openbabel-3.1.1.orig/cmake/modules/Findcoordgen.cmake	2020-05-08 11:41:00.000000000 -0400
+++ openbabel-3.1.1/cmake/modules/Findcoordgen.cmake	2021-03-05 17:07:27.000000000 -0500
@@ -6,7 +6,6 @@
 #
 # coordgen_INCLUDE_DIRS   - CoordGen's includes directory
 # coordgen_LIBRARIES      - CoordGen's shared libraries
-# coordgen_TEMPLATE_FILE  - CoordGen templates file
 #
 #
 
@@ -28,19 +27,9 @@
 )
 message(STATUS "coordgen libraries set as '${coordgen_LIBRARIES}'")
 
-# Just in case, add parent directory above libraries to templates search hints
-get_filename_component(libs_parent_dir ${coordgen_LIBRARIES} PATH)
-find_file(coordgen_TEMPLATE_FILE
-    NAMES templates.mae
-    HINTS ${coordgen_DIR} ${libs_parent_dir}
-    PATH_SUFFIXES "share" "share/coordgen"
-    DOC "templates file for coordgen"
-)
-message(STATUS "coordgen templates file set as '${coordgen_TEMPLATE_FILE}'")
-
 find_package_handle_standard_args(coordgen FOUND_VAR coordgen_FOUND
                                   REQUIRED_VARS coordgen_INCLUDE_DIRS
-                                  coordgen_LIBRARIES coordgen_TEMPLATE_FILE)
+                                  coordgen_LIBRARIES)
 
 
 
diff -Nurd openbabel-3.1.1.orig/src/CMakeLists.txt openbabel-3.1.1/src/CMakeLists.txt
--- openbabel-3.1.1.orig/src/CMakeLists.txt	2020-05-08 11:41:00.000000000 -0400
+++ openbabel-3.1.1/src/CMakeLists.txt	2021-03-04 03:52:55.000000000 -0500
@@ -298,4 +298,7 @@
   endforeach(plugingroup)
 
   add_subdirectory(formats)
+
+  include_directories(${INCHI_INCLUDE_DIR})
+  set(libs ${libs} ${INCHI_LIBRARY})
 endif(BUILD_SHARED)
