diff -ruN matplotlib-2.2.5-orig/lib/matplotlib/font_manager.py matplotlib-2.2.5/lib/matplotlib/font_manager.py
--- matplotlib-2.2.5-orig/lib/matplotlib/font_manager.py	2020-02-02 00:14:26.000000000 -0600
+++ matplotlib-2.2.5/lib/matplotlib/font_manager.py	2020-10-04 19:11:30.000000000 -0500
@@ -145,6 +145,11 @@
     "/Library/Fonts/",
     "/Network/Library/Fonts/",
     "/System/Library/Fonts/",
+    # new XQuartz font location
+    "/opt/X11/share/fonts",
+    "/opt/X11/share/fonts/TTF",
+    # fonts installed via Fink
+    "@PREFIX@/lib/X11/fonts",
     # fonts installed via MacPorts
     "/opt/local/share/fonts"
     ""
diff -ruN matplotlib-2.2.5-orig/setup.cfg.template matplotlib-2.2.5/setup.cfg.template
--- matplotlib-2.2.5-orig/setup.cfg.template	2020-02-02 00:14:27.000000000 -0600
+++ matplotlib-2.2.5/setup.cfg.template	2020-10-04 20:43:37.000000000 -0500
@@ -64,18 +64,18 @@
 #           otherwise skip silently. This is the default
 #           behavior
 #
-#agg = auto
-#cairo = auto
-#gtk = auto
+agg = True
+cairo = True
+gtk = True
 #gtk3agg = auto
 #gtk3cairo = auto
-#gtkagg = auto
+gtkagg = True
 #macosx = auto
 #pyside = auto
-#qt4agg = auto
-#tkagg = auto
+qt5agg = False
+tkagg = True
 #windowing = auto
-#wxagg = auto
+wxagg = False
 
 [rc_options]
 # User-configurable options
diff -ruN matplotlib-2.2.5-orig/setup.py matplotlib-2.2.5/setup.py
--- matplotlib-2.2.5-orig/setup.py	2020-02-02 00:14:27.000000000 -0600
+++ matplotlib-2.2.5/setup.py	2020-10-04 19:13:00.000000000 -0500
@@ -282,6 +282,10 @@
             fd.write(
                 template.safe_substitute(TEMPLATE_BACKEND=default_backend))
 
+    for mod in ext_modules:
+        mod.include_dirs.append('@PREFIX@/include')
+        mod.library_dirs.append('@PREFIX@/lib')
+
         # Build in verbose mode if requested
         if setupext.options['verbose']:
             for mod in ext_modules:
diff -ruN matplotlib-2.2.5-orig/setupext.py matplotlib-2.2.5/setupext.py
--- matplotlib-2.2.5-orig/setupext.py	2020-02-02 00:14:27.000000000 -0600
+++ matplotlib-2.2.5/setupext.py	2020-10-04 19:14:32.000000000 -0500
@@ -308,8 +308,8 @@
 
     basedir_map = {
         'win32': win_bases,
-        'darwin': ['/usr/local/', '/usr', '/usr/X11',
-                   '/opt/X11', '/opt/local'],
+        'darwin': ['@PREFIX@/lib/freetype219', '@PREFIX@',
+                   '/usr', '/usr/X11', '/opt/X11'],
         'sunos5': [os.getenv('MPLIB_BASE') or '/usr/local', ],
         'gnu0': ['/usr'],
         'aix5': ['/usr/local'],
