diff -Naur --exclude='*~' mizar/emacsen-install mizar.fink/emacsen-install
--- mizar/emacsen-install	1970-01-01 01:00:00.000000000 +0100
+++ mizar.fink/emacsen-install	2010-04-20 15:36:01.000000000 +0100
@@ -0,0 +1,33 @@
+#!/bin/bash -e
+
+set -o posix
+
+FLAVOR=${1}
+
+echo >&2 "install/mizar-mode: Handling install of emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "install/mizar-mode: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+if [ ${FLAVOR} == emacs ]
+then
+    echo "install/mizar-mode: done."
+    exit 0
+fi
+
+echo >&2 -n "install/mizar-mode: Byte-compiling for ${FLAVOR}..."
+mkdir -p @PREFIX@/share/${FLAVOR}/site-lisp/mizar;
+ln -s -f @PREFIX@/share/emacs/site-lisp/mizar/mizar.el \
+         @PREFIX@/share/${FLAVOR}/site-lisp/mizar;
+
+cd @PREFIX@/share/${FLAVOR}/site-lisp/mizar;
+
+(${FLAVOR} --no-init-file --no-site-file --batch -f batch-byte-compile mizar.el >&1 >&1) | gzip -9qf > @PREFIX@/share/doc/mizar-mode/CompilationLog-${FLAVOR}.gz
+
+echo >&2 "done."
+echo >&2 "install/mizar-mode: Compilation log saved in @PREFIX@/share/doc/mizar-ode/CompilationLog-${FLAVOR}.gz."
+
+exit 0
diff -Naur --exclude='*~' mizar/emacsen-remove mizar.fink/emacsen-remove
--- mizar/emacsen-remove	1970-01-01 01:00:00.000000000 +0100
+++ mizar.fink/emacsen-remove	2010-04-20 15:36:10.000000000 +0100
@@ -0,0 +1,14 @@
+#!/bin/bash -e
+
+set -o posix
+
+FLAVOR=${1}
+
+echo "remove/mizar-mode: Handling removal for emacsen flavor ${FLAVOR}"
+
+echo >&2 -n "remove/mizar-mode: Purging compilation log and byte-compiled files r ${FLAVOR}..."
+rm -f @PREFIX@/share/doc/mizar-mode/CompilationLog-${FLAVOR}.gz
+rm -Rf @PREFIX@/share/${FLAVOR}/site-lisp/mizar
+echo >&2 "done."
+    
+exit 0
