diff -Nurd ncview-2.1.7.orig/src/ncview.protos.h ncview-2.1.7/src/ncview.protos.h
--- ncview-2.1.7.orig/src/ncview.protos.h	2016-03-20 17:45:52.000000000 +0000
+++ ncview-2.1.7/src/ncview.protos.h	2020-11-30 10:12:36.000000000 +0000
@@ -34,6 +34,7 @@
 void	process_user_input          ( void );
 void	quit_app		    ( void );
 void	create_default_colormap     ( void );
+int	any_var_in_group	    ( NCVar *var );
 int	check			    ( int value, int min, int max );
 void	print_disclaimer	    ( void );
 void	print_no_warranty	    ( void );
@@ -143,6 +144,7 @@
 void 	cache_scalar_coord_info( NCVar *vars );
 int 	count_nslashes	    ( char *s );
 Stringlist *get_group_list  ( NCVar *vars );
+int 	unpack_groupname    ( char *varname, int ig, char *groupname );
 
 /******************************************************************************
  * in interface.c 
