diff -Nurd numpy-1.22.1.orig/numpy/distutils/exec_command.py numpy-1.22.1/numpy/distutils/exec_command.py
--- numpy-1.22.1.orig/numpy/distutils/exec_command.py	2022-01-10 10:36:38.000000000 -0500
+++ numpy-1.22.1/numpy/distutils/exec_command.py	2022-01-25 06:03:59.000000000 -0500
@@ -261,7 +261,7 @@
 
     if os.name == 'posix' and use_shell:
         # On POSIX, subprocess always uses /bin/sh, override
-        sh = os.environ.get('SHELL', '/bin/sh')
+        sh = '/bin/sh'
         if is_sequence(command):
             command = [sh, '-c', ' '.join(command)]
         else:
diff -Nurd numpy-1.22.1.orig/numpy/distutils/system_info.py numpy-1.22.1/numpy/distutils/system_info.py
--- numpy-1.22.1.orig/numpy/distutils/system_info.py	2022-01-12 13:16:54.000000000 -0500
+++ numpy-1.22.1/numpy/distutils/system_info.py	2022-01-25 06:11:20.000000000 -0500
@@ -131,8 +131,8 @@
 ----------
 [DEFAULT]
 # default section
-library_dirs = /usr/lib:/usr/local/lib:/opt/lib
-include_dirs = /usr/include:/usr/local/include:/opt/include
+library_dirs = @FINKPREFIX@/lib:/usr/lib
+include_dirs = @FINKPREFIX@/include:/usr/include
 src_dirs = /usr/local/src:/opt/src
 # search static libraries (.a) in preference to shared ones (.so)
 search_static_first = 0
@@ -141,7 +141,7 @@
 libraries = rfftw, fftw
 
 [atlas]
-library_dirs = /usr/lib/3dnow:/usr/lib/3dnow/atlas
+library_dirs = 
 # for overriding the names of the atlas libraries
 libraries = lapack, f77blas, cblas, atlas
 
@@ -323,20 +323,16 @@
         add_system_root(os.path.join(conda_dir, 'Library'))
 
 else:
-    default_lib_dirs = libpaths(['/usr/local/lib', '/opt/lib', '/usr/lib',
-                                 '/opt/local/lib', '/sw/lib'], platform_bits)
+    default_lib_dirs = libpaths(['@FINKPREFIX@/lib',
+                                 '/usr/lib'], platform_bits)
     default_runtime_dirs = []
-    default_include_dirs = ['/usr/local/include',
-                            '/opt/include',
-                            # path of umfpack under macports
-                            '/opt/local/include/ufsparse',
-                            '/opt/local/include', '/sw/include',
-                            '/usr/include/suitesparse']
-    default_src_dirs = ['.', '/usr/local/src', '/opt/src', '/sw/src']
+    default_include_dirs = ['@FINKPREFIX@/include',
+                            '/usr/include']
+    default_src_dirs = ['.']
 
-    default_x11_lib_dirs = libpaths(['/usr/X11R6/lib', '/usr/X11/lib',
+    default_x11_lib_dirs = libpaths(['/opt/X11/lib', '/usr/X11/lib',
                                      '/usr/lib'], platform_bits)
-    default_x11_include_dirs = ['/usr/X11R6/include', '/usr/X11/include']
+    default_x11_include_dirs = ['/opt/X11/include', '/usr/X11/include']
 
     if os.path.exists('/usr/lib/X11'):
         globbed_x11_dir = glob('/usr/lib/*/libX11.so')
