diff -Nurd octave-3.8.2.orig/configure.ac octave-3.8.2/configure.ac
--- octave-3.8.2.orig/configure.ac	2014-08-06 15:03:53.000000000 -0400
+++ octave-3.8.2/configure.ac	2019-11-09 10:40:21.000000000 -0500
@@ -720,6 +720,10 @@
   AC_SUBST(F77_FLOAT_STORE_FLAG)
 ])
 
+if test $ac_cv_f77_compiler_gnu = yes; then
+  OCTAVE_F77_FLAG([-std=legacy])
+fi
+
 ### Check for the Qhull library
 
 OCTAVE_CHECK_LIB(qhull, QHull,
