diff -Nurd -x'*~' octave-3.6.4.orig/configure octave-3.6.4/configure
--- octave-3.6.4.orig/configure	2013-02-21 15:21:49.000000000 -0500
+++ octave-3.6.4/configure	2019-11-10 11:35:48.000000000 -0500
@@ -34766,6 +34766,7 @@
 
   fi
 
+FFLAGS="$FFLAGS -std=legacy"
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for IEEE 754 data format" >&5
 $as_echo_n "checking for IEEE 754 data format... " >&6; }
