diff -Nurd octave-@OCTVERSION@/src/DLD-FUNCTIONS/__magick_read__.cc octave-@OCTVERSION@.patched/src/DLD-FUNCTIONS/__magick_read__.cc
--- octave-@OCTVERSION@/src/DLD-FUNCTIONS/__magick_read__.cc	2009-08-25 04:26:01.000000000 -0400
+++ octave-@OCTVERSION@.patched/src/DLD-FUNCTIONS/__magick_read__.cc	2011-10-19 14:27:31.000000000 -0400
@@ -357,6 +357,8 @@
   octave_value_list output;
 
 #ifdef HAVE_MAGICK
+  // Initialize GraphicsMagick
+  InitializeMagick(NULL);
 
   if (args.length () > 2 || args.length () < 1 || ! args(0).is_string ()
       || nargout > 3)
@@ -769,6 +771,8 @@
   octave_value_list retval;
 
 #ifdef HAVE_MAGICK
+  //Initialize GraphicsMagick
+  InitializeMagick(NULL);
   int nargin = args.length ();
 
   if (nargin > 2)
@@ -897,6 +901,8 @@
   octave_value_list output;
 
 #ifdef HAVE_MAGICK
+  // Initialize GraphicsMagick
+  InitializeMagick(NULL);
 
   if (args.length () < 1 || ! args (0).is_string ())
     {
