diff -Nurd -x'*~' octave-3.6.4.orig/liboctave/lo-specfun.cc octave-3.6.4/liboctave/lo-specfun.cc
--- octave-3.6.4.orig/liboctave/lo-specfun.cc	2013-02-21 15:19:24.000000000 -0500
+++ octave-3.6.4/liboctave/lo-specfun.cc	2019-11-09 23:21:48.000000000 -0500
@@ -3226,7 +3226,9 @@
 // Templatized and simplified workflow.
 
 // FIXME: Maybe this should be globally visible.
+#if !defined (HAVE_ERFC)
 static inline float erfc (float x) { return erfcf (x); }
+#endif
 
 template <class T>
 static T
