diff -Naur startup/octaverc.null startup/octaverc
--- startup/octaverc.null	2011-08-12 14:21:47.000000000 -0400
+++ startup/octaverc	2011-08-12 12:32:41.000000000 -0400
@@ -0,0 +1,24 @@
+## System-wide startup file for Octave.
+##
+## This file should contain any commands that should be executed each
+## time Octave-@OCTVERSION@ starts for every user at this site.
+
+## Configure readline using the file inputrc in the Octave startup
+## directory.
+
+readline_read_init_file (sprintf ("%s%s%s",
+				  octave_config_info ("startupfiledir"),
+				  filesep, "inputrc"));
+
+if (strcmp (PAGER (), "less") && isempty (getenv ("LESS")))
+  PAGER_FLAGS ('-e -X -P"-- less ?pB(%pB\\%):--. (f)orward, (b)ack, (q)uit$"');
+endif
+
+## This appears here instead of in the pkg/PKG_ADD file so that --norc
+## will also skip automatic loading of packages.
+
+try 
+	pkg global_list @FINKPREFIX@/var/octave/@OCTVERSION@/octave_packages
+catch
+end_try_catch
+pkg ("load", "auto");
