diff -uNr a/Makefile.system b/Makefile.system
--- a/Makefile.system	2019-04-29 19:21:59.000000000 +0200
+++ b/Makefile.system	2019-05-03 02:07:52.000000000 +0200
@@ -1173,8 +1173,8 @@
 endif


-REVISION = -r$(VERSION)
 MAJOR_VERSION = $(word 1,$(subst ., ,$(VERSION)))
+REVISION = .$(MAJOR_VERSION)

 ifeq ($(DEBUG), 1)
 COMMON_OPT += -g
@@ -1244,11 +1244,11 @@

 ifneq ($(DYNAMIC_ARCH), 1)
 ifndef SMP
-LIBNAME		= $(LIBPREFIX)_$(LIBCORE)$(REVISION).$(LIBSUFFIX)
-LIBNAME_P	= $(LIBPREFIX)_$(LIBCORE)$(REVISION)_p.$(LIBSUFFIX)
+LIBNAME		= $(LIBPREFIX)$(REVISION).$(LIBSUFFIX)
+LIBNAME_P	= $(LIBPREFIX)_p$(REVISION).$(LIBSUFFIX)
 else
-LIBNAME		= $(LIBPREFIX)_$(LIBCORE)p$(REVISION).$(LIBSUFFIX)
-LIBNAME_P	= $(LIBPREFIX)_$(LIBCORE)p$(REVISION)_p.$(LIBSUFFIX)
+LIBNAME		= $(LIBPREFIX)$(REVISION).$(LIBSUFFIX)
+LIBNAME_P	= $(LIBPREFIX)_p$(REVISION).$(LIBSUFFIX)
 endif
 else
 ifndef SMP
diff -uNr a/benchmark/Makefile b/benchmark/Makefile
--- a/benchmark/Makefile   2019-04-29 19:21:59.000000000 +0200
+++ b/benchmark/Makefile   2019-05-03 02:08:58.000000000 +0200
@@ -19,7 +19,8 @@
 #LIBATLAS = -fopenmp $(ATLAS)/liblapack_atlas.a  $(ATLAS)/libptcblas.a  $(ATLAS)/libptf77blas.a  $(ATLAS)/libatlas.a -lgfortran -lm
 
 # Atlas RHEL and Fedora
-ATLAS=/usr/lib64/atlas
+# Atlas MacOS X Fink
+ATLAS=@PREFIX@/lib
 LIBATLAS = -fopenmp $(ATLAS)/liblapack.a  $(ATLAS)/libptcblas.a  $(ATLAS)/libptf77blas.a  $(ATLAS)/libatlas.a -lgfortran -lm
 
 # Intel standard
diff -uNr a/Makefile b/Makefile
--- a/Makefile		2019-08-11 23:23:27.000000000 +0100
+++ b/Makefile		2019-11-06 19:18:15.000000000 +0100
@@ -109,8 +109,10 @@
 ifeq ($(OSNAME), Darwin)
	@$(MAKE) -C exports dyn
	@ln -fs $(LIBDYNNAME) $(LIBPREFIX).dylib
+ifneq ($(LIBDYNNAME), $(LIBPREFIX).$(MAJOR_VERSION).dylib)
	@ln -fs $(LIBDYNNAME) $(LIBPREFIX).$(MAJOR_VERSION).dylib
 endif
+endif
 ifeq ($(OSNAME), WINNT)
	@$(MAKE) -C exports dll
 endif
diff -uNr a/Makefile.install b/Makefile.install
--- a/Makefile.install	2019-08-11 23:23:27.000000000 +0100
+++ b/Makefile.install	2019-11-06 2019-11-06 20:13:02.000000000 +0100
@@ -83,8 +83,7 @@
	@-cp $(LIBDYNNAME) "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)"
	@-install_name_tool -id "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)/$(LIBDYNNAME)" "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)/$(LIBDYNNAME)"
	@cd "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)" ; \
-	ln -fs $(LIBDYNNAME) $(LIBPREFIX).dylib ; \
-	ln -fs $(LIBDYNNAME) $(LIBPREFIX).$(MAJOR_VERSION).dylib
+	ln -fs $(LIBDYNNAME) $(LIBPREFIX).dylib
 endif
 ifeq ($(OSNAME), WINNT)
	@-cp $(LIBDLLNAME) "$(DESTDIR)$(OPENBLAS_BINARY_DIR)"
