diff -c pari-2.1.1/src/gp/gp.c pari-2.1.1-patched/src/gp/gp.c
*** pari-2.1.1/src/gp/gp.c	Mon Apr  2 05:31:12 2001
--- pari-2.1.1-patched/src/gp/gp.c	Thu Aug 23 18:47:13 2001
***************
*** 1908,1913 ****
--- 1908,1914 ----
      f = gprc_chk(str); /* in $HOME */
      if (!f) f = gprc_chk(s); /* in . */
      if (!f) f = gprc_chk("/etc/gprc");
+     if (!f) f = gprc_chk("@FINK@/etc/gprc");
      if (!f) f = gprc_chk("C:/_gprc");
      free(str);
    }
diff -ruN pari-2.1.4/emacsen-install pari-2.1.4-patched/emacsen-install
--- pari-2.1.4/emacsen-install	Wed Dec 31 16:00:00 1969
+++ pari-2.1.4-patched/emacsen-install	Sun Aug  4 15:01:19 2002
@@ -0,0 +1,33 @@
+#!/bin/sh
+# @FINK@/lib/emacsen-common/packages/install/pari-gp
+set -e
+ 
+FLAVOR=$1
+echo install/pari-gp: Handling install of emacsen flavor ${FLAVOR}
+
+byte_compile_options="-batch -f batch-byte-compile"
+el_files="pari.el pari-*.el sli-tools.el"
+el_dir=@FINK@/share/emacs/site-lisp/pari
+elc_dir=@FINK@/share/${FLAVOR}/site-lisp/pari
+el_path_list="${el_dir}/pari.el ${el_dir}/pari-*.el ${el_dir}/sli-tools.el"
+elc_path_list="${elc_dir}/pari.el ${elc_dir}/pari-*.el ${elc_dir}/sli-tools.el"
+
+if [ ${FLAVOR} != emacs ]
+then
+  echo install/pari-gp: byte-compiling for ${FLAVOR}
+
+  # Create the directory
+  mkdir -p ${elc_dir}
+
+  # Copy the temp .el files
+  cp ${el_path_list} ${elc_dir}
+
+  # Byte compile them
+  ${FLAVOR} ${byte_compile_options} ${elc_path_list}
+
+  # remove the redundant .el files
+  # presumes that any .el files in the <flavor> dir are trash.
+  rm ${elc_dir}/*.el
+fi
+exit 0;
+
diff -ruN pari-2.1.4/emacsen-remove pari-2.1.4-patched/emacsen-remove
--- pari-2.1.4/emacsen-remove	Wed Dec 31 16:00:00 1969
+++ pari-2.1.4-patched/emacsen-remove	Sun Aug  4 15:01:30 2002
@@ -0,0 +1,13 @@
+#!/bin/sh
+# @FINK@/lib/emacsen-common/packages/remove/pari-gp
+set -e
+
+FLAVOR=$1
+echo remove/pari-gp: Handling removal of emacsen flavor ${FLAVOR}
+
+if [ ${FLAVOR} != emacs ]
+then
+  echo emacsen-common: purging byte-compiled files for ${FLAVOR}
+  rm -f @FINK@/share/${FLAVOR}/site-lisp/pari/*.elc
+fi
+exit 0;
diff -ruN pari-2.3.0/misc/gprc.dft pari-2.3.0-patched/misc/gprc.dft
--- pari-2.3.0/misc/gprc.dft	2005-09-03 10:26:31.000000000 -0700
+++ pari-2.3.0-patched/misc/gprc.dft	2006-06-30 05:57:30.000000000 -0700
@@ -5,8 +5,8 @@
  * Customize (by uncommenting and modifying the relevant lines) and put in
  * $HOME/.gprc (or $GPRC) or /etc/gprc. Syntax explained at the end. */
 
-\\ read "lib/gpalias"
-/* Read lib/gpalias BEFORE gp prompts you for commands.
+read "@FINK@/gpalias"
+/* Read @FINK@/gpalias BEFORE gp prompts you for commands.
  * The file contains some standard abbreviations (correct the path first !) */
 
 \\ compatible = 0
