diff -Nurd phylip-3.697.orig/src/Makefile.osx phylip-3.697/src/Makefile.osx
--- phylip-3.697.orig/src/Makefile.osx	2014-09-19 10:29:23
+++ phylip-3.697/src/Makefile.osx	2025-11-01 22:17:50
@@ -230,12 +230,12 @@
 	@cp $(JARS) $(EXEDIR)
 	@echo "Installing font files in $(EXEDIR)"
 	@cp font* $(EXEDIR)
-	@echo "Copying testdata to $(EXEDIR)/testdata"
-	mkdir $(EXEDIR)/testdata
-	for i in $(TESTDIR); do \
-      mkdir $(EXEDIR)/testdata/$$i; \
-      cp TestData/$$i/*.txt $(EXEDIR)/testdata/$$i; \
-    done
+#	@echo "Copying testdata to $(EXEDIR)/testdata"
+#	mkdir $(EXEDIR)/testdata
+#	for i in $(TESTDIR); do \
+#      mkdir $(EXEDIR)/testdata/$$i; \
+#      cp TestData/$$i/*.txt $(EXEDIR)/testdata/$$i; \
+#    done
 	@echo "Finished installation."
 	@echo ""
 
@@ -244,8 +244,8 @@
 	@for i in $(APPS) ; do \
       rm -rf "$(EXEDIR)/$$i" ; \
     done
-	@echo "Removing preexisting testdata"
-	@rm -rf $(EXEDIR)/testdata
+#	@echo "Removing preexisting testdata"
+#	@rm -rf $(EXEDIR)/testdata
 	@echo "Ready for new executables"
 	@echo ""
 
@@ -376,7 +376,7 @@
 	
 # needed by java	
 libdrawgram.dylib:     drawgram.o draw.o draw2.o phylip.o
-	$(CC) $(CFLAGS)  -shared drawgram.o draw.o draw2.o phylip.o $(CLIBS) -o libdrawgram.dylib
+	$(CC) $(CFLAGS)  -shared -install_name @FINK_PREFIX@/lib/libdrawgram.dylib drawgram.o draw.o draw2.o phylip.o $(CLIBS) -o libdrawgram.dylib
 
 drawtree.o:     drawtree.c draw.h phylip.h
 	$(CC) $(CFLAGS) -c drawtree.c
@@ -386,7 +386,7 @@
 
 # needed by java	
 libdrawtree.dylib:     drawtree.o draw.o draw2.o phylip.o
-	$(CC) $(CFLAGS)  -shared drawtree.o draw.o draw2.o phylip.o $(CLIBS) -o libdrawtree.dylib
+	$(CC) $(CFLAGS)  -shared -install_name @FINK_PREFIX@/lib/libdrawtree.dylib drawtree.o draw.o draw2.o phylip.o $(CLIBS) -o libdrawtree.dylib
 
 factor.o:       factor.c phylip.h
 
@@ -614,11 +614,11 @@
 	mkdir $(dist_JAVADIR)
 	mkdir $(dist_SRCDIR)/mac
 	mkdir $(dist_SRCDIR)/icons
-	mkdir $(dist_EXEDIR)/testdata
-	for i in $(TESTDIR); do \
-      mkdir $(dist_EXEDIR)/testdata/$$i; \
-      cp TestData/$$i/*.txt $(dist_EXEDIR)/testdata/$$i; \
-    done
+#	mkdir $(dist_EXEDIR)/testdata
+#	for i in $(TESTDIR); do \
+#      mkdir $(dist_EXEDIR)/testdata/$$i; \
+#      cp TestData/$$i/*.txt $(dist_EXEDIR)/testdata/$$i; \
+#    done
 	for i in $(DIST_COMMON) ; do \
 	  cp -r $$i $(DISTDIR) ; \
 	done
diff -Nurd phylip-3.697.orig/src/cons.c phylip-3.697/src/cons.c
--- phylip-3.697.orig/src/cons.c	2017-12-07 20:18:41
+++ phylip-3.697/src/cons.c	2025-11-01 22:27:09
@@ -254,7 +254,7 @@
     if ( c != '\n' && c!= '\r')
       printf("WARNING: Tree weight set to 1.0\n");
     if ( c == '\r' )
-      if ( (c == gettc(intree)) != '\n')
+      if ( (c = gettc(intree)) != '\n')
         ungetc(c, intree);
     break;
   case hsnolength:
diff -Nurd phylip-3.697.orig/src/draw.c phylip-3.697/src/draw.c
--- phylip-3.697.orig/src/draw.c	2014-09-16 23:38:54
+++ phylip-3.697/src/draw.c	2025-11-01 22:24:13
@@ -195,7 +195,6 @@
 void   plotpb(void);
 char  *findXfont(char *, double, double *, int *);
 int    macfontid(char *);
-void   makebox(char *, double *, double *, double *, long);
 
 /* function prototypes */
 #endif
diff -Nurd phylip-3.697.orig/src/draw.h phylip-3.697/src/draw.h
--- phylip-3.697.orig/src/draw.h	2014-09-16 23:38:54
+++ phylip-3.697/src/draw.h	2025-11-01 22:24:11
@@ -87,6 +87,7 @@
 void plotlabels(char *fontname);
 void   pout(long);
 double computeAngle(double oldx, double oldy, double newx, double newy);
+void   makebox(char *, double *, double *, double *, long);
 
 
 /* For povray, added by Dan F. */
diff -Nurd phylip-3.697.orig/src/phylip.h phylip-3.697/src/phylip.h
--- phylip-3.697.orig/src/phylip.h	2017-12-07 20:19:20
+++ phylip-3.697/src/phylip.h	2025-11-01 22:17:50
@@ -96,7 +96,7 @@
 #define EBCDIC false
 #define INFILE "infile"
 #define OUTFILE "outfile"
-#define FONTFILE "fontfile" /* on unix this might be /usr/local/lib/fontfile */
+#define FONTFILE "@FINK_PREFIX@/share/phylip/fontfile" /* on unix this might be /usr/local/lib/fontfile */
 #define PLOTFILE "plotfile"
 #define INTREE "intree"
 #define INTREE2 "intree2"
