--- pythia.orig/Makefile	2017-10-06 10:45:44.000000000 +0200
+++ pythia.new/Makefile	2018-01-21 15:52:05.000000000 +0100
@@ -38,7 +38,7 @@
 	$(sort $(wildcard $(LOCAL_SRC)/*.cc)))
 TARGETS=$(LOCAL_LIB)/libpythia8.a
 ifeq ($(ENABLE_SHARED),true)
-  TARGETS+=$(LOCAL_LIB)/libpythia8$(LIB_SUFFIX)
+  TARGETS+=$(LOCAL_LIB)/libpythia8.8.2$(LIB_SUFFIX)
 endif

 # LHAPDF.
@@ -103,9 +103,12 @@
 $(LOCAL_LIB)/libpythia8.a: $(OBJECTS)
 	rm -f $(LOCAL_LIB)/libpythia8$(LIB_SUFFIX)
 	ar cru $@ $^
-$(LOCAL_LIB)/libpythia8$(LIB_SUFFIX): $(OBJECTS)
+$(LOCAL_LIB)/libpythia8.8.2$(LIB_SUFFIX): $(OBJECTS)
 	$(CXX) $^ -o $@ $(CXX_COMMON) $(CXX_SHARED) $(CXX_SONAME)$(notdir $@)\
-	  $(LIB_COMMON)
+	  $(LIB_COMMON) \
+	 -Wl,-install_name,@FINKPREFIX@/lib/libpythia8.8.2$(LIB_SUFFIX) \
+	 -Wl,-compatibility_version,8.2.0 -Wl,-current_version,8.2.23
+	 @ln -s libpythia8.8.2$(LIB_SUFFIX) $(LOCAL_LIB)/libpythia8$(LIB_SUFFIX)

 # LHAPDF (turn off all warnings for readability).
 $(LOCAL_TMP)/LHAPDF%Plugin.o: $(LOCAL_INCLUDE)/Pythia8Plugins/LHAPDF%.h
