diff -ruN qgis-2.12.2-orig/CMakeLists.txt qgis-2.12.2-framework/CMakeLists.txt
--- qgis-2.12.2-orig/CMakeLists.txt	2015-12-18 22:21:33.000000000 +0900
+++ qgis-2.12.2-framework/CMakeLists.txt	2016-01-17 16:38:13.000000000 +0900
@@ -569,9 +569,9 @@
     # need to find non-system frameworks
     # cmake bug #0007250 - CMAKE_SHARED_LINKER_FLAGS ignored when creating
     # a framework, so these need to be manually handled with LINK_FLAGS options
-    SET (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -F/Library/Frameworks")
-    SET (CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -F/Library/Frameworks")
-    SET (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -F/Library/Frameworks")
+    # SET (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -F/Library/Frameworks")
+    # SET (CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -F/Library/Frameworks")
+    # SET (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -F/Library/Frameworks")
 
   ELSE (APPLE)
     # UNIX
diff -ruN qgis-2.12.2-orig/cmake/FindOSGEARTH.cmake qgis-2.12.2-framework/cmake/FindOSGEARTH.cmake
--- qgis-2.12.2-orig/cmake/FindOSGEARTH.cmake	2015-12-18 22:21:33.000000000 +0900
+++ qgis-2.12.2-framework/cmake/FindOSGEARTH.cmake	2016-01-17 16:37:36.000000000 +0900
@@ -27,15 +27,7 @@
         $ENV{OSGEARTHDIR}
         $ENV{OSGEARTH_DIR}
         $ENV{OSGEO4W_ROOT}
-        /usr/local/
-        /usr/
         /sw/ # Fink
-        /opt/local/ # DarwinPorts
-        /opt/csw/ # Blastwave
-        /opt/
-        [HKEY_LOCAL_MACHINE\\SYSTEM\\CurrentControlSet\\Control\\Session\ Manager\\Environment;OSGEARTH_ROOT]/
-        ~/Library/Frameworks
-        /Library/Frameworks
     PATH_SUFFIXES
         /include/
 )
@@ -108,7 +100,7 @@
     SET(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${OSGEARTHUTIL_LIBRARY})
     IF(APPLE)
         # no extra LDFLAGS used in link test, may fail in OS X SDK
-        SET(CMAKE_REQUIRED_LIBRARIES "-F/Library/Frameworks" ${CMAKE_REQUIRED_LIBRARIES})
+        # SET(CMAKE_REQUIRED_LIBRARIES "-F/Library/Frameworks" ${CMAKE_REQUIRED_LIBRARIES})
     ENDIF(APPLE)
     CHECK_CXX_SOURCE_COMPILES("
 #include <osgEarthUtil/Controls>
diff -ruN qgis-2.12.2-orig/cmake/FindSPATIALITE.cmake qgis-2.12.2-framework/cmake/FindSPATIALITE.cmake
--- qgis-2.12.2-orig/cmake/FindSPATIALITE.cmake	2015-12-18 22:21:33.000000000 +0900
+++ qgis-2.12.2-framework/cmake/FindSPATIALITE.cmake	2016-01-17 16:33:43.000000000 +0900
@@ -1,4 +1,4 @@
-# Find SpatiaLite
+ # Find SpatiaLite
 # ~~~~~~~~~~~~~~~
 # Copyright (c) 2009, Sandro Furieri <a.furieri at lqt.it>
 # Redistribution and use is allowed according to the terms of the BSD license.
@@ -66,7 +66,7 @@
    # Check for symbol gaiaDropTable
    IF(APPLE)
        # no extra LDFLAGS used in link test, may fail in OS X SDK
-       SET(CMAKE_REQUIRED_LIBRARIES "-F/Library/Frameworks" ${CMAKE_REQUIRED_LIBRARIES})
+       # SET(CMAKE_REQUIRED_LIBRARIES "-F/Library/Frameworks" ${CMAKE_REQUIRED_LIBRARIES})
    ENDIF(APPLE)
    check_library_exists("${SPATIALITE_LIBRARY}" gaiaDropTable "" SPATIALITE_VERSION_GE_4_0_0)
    check_library_exists("${SPATIALITE_LIBRARY}" gaiaStatisticsInvalidate "" SPATIALITE_VERSION_G_4_1_1)
