diff -Nurd qtiplot-0.9.8.9.orig/build.conf.example qtiplot-0.9.8.9/build.conf.example
--- qtiplot-0.9.8.9.orig/build.conf.example	2011-02-18 02:16:46.000000000 -0500
+++ qtiplot-0.9.8.9/build.conf.example	2022-02-12 00:08:52.000000000 -0500
@@ -7,39 +7,39 @@
 ##########################################################
 
 # Global include path which is always added at the end of the INCLUDEPATH
-SYS_INCLUDEPATH = /opt/local/include
+SYS_INCLUDEPATH = @FINK_PREFIX@/include
 # Global lib path and libs which is ls always added at the end of LIBS
-SYS_LIBS = -L/opt/local/lib
+SYS_LIBS = -L@FINK_PREFIX@/lib/qt4-@QT4_TYPE@/lib -L@FINK_PREFIX@/lib -lz
 
 ##########################################################
 ## zlib (http://www.zlib.net/)
 ##########################################################
 
 # include path. leave it blank to use SYS_INCLUDE
-ZLIB_INCLUDEPATH = $$QTI_ROOT/3rdparty/zlib/
+#ZLIB_INCLUDEPATH = $$QTI_ROOT/3rdparty/zlib/
 
 ##########################################################
 ## muParser (http://muparser.sourceforge.net/)
 ##########################################################
 
 # include path. leave it blank to use SYS_INCLUDE
-MUPARSER_INCLUDEPATH = $$QTI_ROOT/3rdparty/muparser/include
+#MUPARSER_INCLUDEPATH = $$QTI_ROOT/3rdparty/muparser/include
 # link statically against a copy in 3rdparty/
-MUPARSER_LIBS = $$QTI_ROOT/3rdparty/muparser/lib/libmuparser.a
+#MUPARSER_LIBS = $$QTI_ROOT/3rdparty/muparser/lib/libmuparser.a
 # or dynamically against a system-wide installation
-#MUPARSER_LIBS = -lmuparser
+MUPARSER_LIBS = -lmuparser
 
 ##########################################################
 ## GNU Sientific Library (http://www.gnu.org/software/gsl/)
 ##########################################################
 
 # include path. leave it blank to use SYS_INCLUDE
-GSL_INCLUDEPATH = $$QTI_ROOT/3rdparty/gsl/include
+#GSL_INCLUDEPATH = $$QTI_ROOT/3rdparty/gsl/include
 # link statically against a copy in 3rdparty/
-GSL_LIBS = $$QTI_ROOT/3rdparty/gsl/lib/libgsl.a \
-           $$QTI_ROOT/3rdparty/gsl/lib/libgslcblas.a
+#GSL_LIBS = $$QTI_ROOT/3rdparty/gsl/lib/libgsl.a \
+#           $$QTI_ROOT/3rdparty/gsl/lib/libgslcblas.a
 # or dynamically against a system-wide installation
-#GSL_LIBS = -lgsl -lgslcblas
+GSL_LIBS = -lgsl -lgslcblas
 
 ##########################################################
 ## QWT - use local copy till upstream catches up
@@ -50,6 +50,7 @@
 QWT_INCLUDEPATH = $$QTI_ROOT/3rdparty/qwt/src
 # link locally against a copy in 3rdparty/
 QWT_LIBS = $$QTI_ROOT/3rdparty/qwt/lib/libqwt.a
+qtiplot.depends += 3rdparty/qwt
 
 ##########################################################
 ## QwtPlot3D - use local copy till upstream catches up
@@ -61,17 +62,23 @@
 # link locally against a copy in 3rdparty/
 win32:QWT3D_LIBS = $$QTI_ROOT/3rdparty/qwtplot3d/lib/qwtplot3d.dll
 unix:QWT3D_LIBS = $$QTI_ROOT/3rdparty/qwtplot3d/lib/libqwtplot3d.a
+qtilot.depends += 3rdparty/qwtplot3d
+
+####
+## for fink qtassistant (see qtiplot/qtiplot.pro patch)
+####
+QTASSISTANT_INCLUDEPATH = @FINK_PREFIX@/lib/qt4-@QT4_TYPE@/include/QtAssistant
 
 ##########################################################
 ## libpng - optional. you don't have to set these variables
 ##########################################################
 
 # include path. leave it blank to use SYS_INCLUDE
-LIBPNG_INCLUDEPATH = $$QTI_ROOT/3rdparty/libpng/
+#LIBPNG_INCLUDEPATH = $$QTI_ROOT/3rdparty/libpng/
 # link statically against a copy in 3rdparty/
-LIBPNG_LIBS = $$QTI_ROOT/3rdparty/libpng/libpng.a
+#LIBPNG_LIBS = $$QTI_ROOT/3rdparty/libpng/libpng.a
 # or dynamically against a system-wide installation
-#LIBPNG_LIBS = -lpng
+LIBPNG_LIBS = -lpng
 
 ##########################################################
 ## QTeXEngine - optional. you don't have to set these variables
@@ -79,9 +86,9 @@
 ##########################################################
 
 # include path.
-TEX_ENGINE_INCLUDEPATH = $$QTI_ROOT/3rdparty/QTeXEngine/src
+#TEX_ENGINE_INCLUDEPATH = $$QTI_ROOT/3rdparty/QTeXEngine/src
 # link locally against a copy in 3rdparty/
-TEX_ENGINE_LIBS = $$QTI_ROOT/3rdparty/QTeXEngine/libQTeXEngine.a
+#TEX_ENGINE_LIBS = $$QTI_ROOT/3rdparty/QTeXEngine/libQTeXEngine.a
 
 ##########################################################
 ## ALGLIB (2.6) - optional. you don't have to set these variables
@@ -89,9 +96,9 @@
 ##########################################################
 
 # include path.
-ALGLIB_INCLUDEPATH = $$QTI_ROOT/3rdparty/alglib/
+#ALGLIB_INCLUDEPATH = $$QTI_ROOT/3rdparty/alglib/
 # link locally against a copy in 3rdparty/
-ALGLIB_LIBS = $$QTI_ROOT/3rdparty/alglib/libalglib.a
+#ALGLIB_LIBS = $$QTI_ROOT/3rdparty/alglib/libalglib.a
 
 ##########################################################
 ## TAMUANOVA - optional. you don't have to set these variables
@@ -99,9 +106,9 @@
 ##########################################################
 
 # include path.
-TAMUANOVA_INCLUDEPATH = $$QTI_ROOT/3rdparty/tamu_anova/
+#TAMUANOVA_INCLUDEPATH = $$QTI_ROOT/3rdparty/tamu_anova/
 # link locally against a copy in 3rdparty/
-TAMUANOVA_LIBS = $$QTI_ROOT/3rdparty/tamu_anova/libtamuanova.a
+#TAMUANOVA_LIBS = $$QTI_ROOT/3rdparty/tamu_anova/libtamuanova.a
 
 ##########################################################
 ## python - only used if python is needed
@@ -109,7 +116,7 @@
 
 # the python interpreter to use
 # (unix only, windows will use what ever is configured to execute .py files!)
-PYTHON = python
+PYTHON = @FINK_PREFIX@/bin/python2.7
 
 ##########################################################
 ## Qt tools - allows to use specific versions
@@ -125,16 +132,18 @@
 contains( TARGET, qtiplot ) {
   # building without muParser doesn't work yet
   SCRIPTING_LANGS += muParser
-  SCRIPTING_LANGS += Python
+#  SCRIPTING_LANGS += Python
 
   # a console displaying output of scripts; particularly useful on Windows
   # where running QtiPlot from a terminal is inconvenient
   DEFINES         += SCRIPTING_CONSOLE
 
-  #DEFINES         += QTIPLOT_DEMO
+  DEFINES         += SCRIPTING_DIALOG
+
+#DEFINES         += QTIPLOT_DEMO
 
   # Uncomment the following line if you want to perform a custom installation using the *.path variables defined in ./qtiplot.pro.
-  #CONFIG          += CustomInstall
+  CONFIG          += CustomInstall
 
   # Uncomment the following line if you want to build QtiPlot as a browser plugin (not working on Internet Explorer).
   #CONFIG          += BrowserPlugin
