diff -ruN refmac5.8-orig/gibbs_gm_sampler.f90 refmac5.8/gibbs_gm_sampler.f90
--- refmac5.8-orig/gibbs_gm_sampler.f90	2014-10-06 16:19:13.000000000 -0500
+++ refmac5.8/gibbs_gm_sampler.f90	2020-01-17 05:57:26.000000000 -0600
@@ -53,7 +53,7 @@
     integer, allocatable :: iuniform(:)
     real runiform
     integer isize,clock
-    integer vals(8),puts(12)
+    integer vals(8),puts(33)
     logical first
     !
     !  More locals
@@ -100,7 +100,7 @@
     !call date_and_time(values=vals)
     !call random_seed(put=vals(8:8))
     call system_clock(count=clock)
-    puts = clock + 37*(/(i - 1, i = 1, 12) /)
+    puts = clock + 37*(/(i - 1, i = 1, 33) /)
     call random_seed(put=puts)
 !    call random_seed(size=isize)
 !    write(*,*)isize
diff -ruN refmac5.8-orig/makefile refmac5.8/makefile
--- refmac5.8-orig/makefile	2014-10-06 16:47:05.000000000 -0700
+++ refmac5.8/makefile	2014-10-06 17:44:18.000000000 -0700
@@ -1,31 +1,26 @@
-VERSION = gfortran
-FC      = gfortran
-CC      = gcc
-CPP     = g++
-FOPTIM  = -O2 
-COPTIM  = -O2 
-XFFLAGS = -fno-second-underscore -static
-CLIB = /usr/lib
-#ATLASLIB = /home/matem/Scaricati/lapack-3.3.1
-ATLASLIB = /usr/lib/lapack-3.4.2/
-
-XCFLAGS = -DGFORTRAN -DPROTOTYPE
-FFLAGS = $(FOPTIM) $(XFFLAGS) 
-CFLAGS = $(COPTIM) $(XCFLAGS)
+VERSION = 5 
+# FC      = @FINKPREFIX@/lib/gcc9/bin/gfortran
+# CC      = @FINKPREFIX@/lib/gcc9/bin/gcc-4
+# CXX     = @FINKPREFIX@/lib/gcc9/bin/g++-4
+# CPP     = @FINKPREFIX@/lib/gcc9/bin/g++-4
+FOPTIM  = -O3 
+COPTIM  = -O 
+XFFLAGS = -m64 -fno-second-underscore 
+XCFLAGS = -DGFORTRAN -DPROTOTYPE 
+FFLAGS = $(FOPTIM) $(XFFLAGS)  
+CFLAGS = $(COPTIM) $(XCFLAGS) 
 CPPFLAGS = $(CFLAGS)
+CLIB = @FINKPREFIX@/lib
 
-LLIBCIF = 
-LLIBCCP = $(CLIB)/libccp4f.a $(CLIB)/libccp4c.a
-#LLIBLAPACK = $(CLIB)/liblapack.a $(CLIB)/libblas.a
-#LLIBLAPACK = $(ATLASLIB)/lapack.a $(ATLASLIB)/blas.a
-LLIBLAPACK = $(ATLASLIB)/liblapack.a $(ATLASLIB)/librefblas.a
-LLIBOTHERS =  -lgfortran -lgfortranbegin -lstdc++ 
+# ATLASLIB = /home/matem/Scaricati/lapack-3.3.1
+# ATLASLIB = /usr/lib/lapack-3.4.2/
+BLAS_DIR = /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A
 
-include makefile_refmac.inc1
-#include makefile_nmr_test.inc
+LLIBCCP = /usr/lib/libSystem.B.dylib  ${BLAS_DIR}/libBLAS.dylib  ${BLAS_DIR}/libLAPACK.dylib  
+
+LLIBLAPACK =  
 
+LLIBOTHERS =  -L/usr/lib -L${CLIB} -L@FINKPREFIX@/lib/gcc9/lib -lgfortran  ${CLIB}/libccp4c.dylib ${CLIB}/libccp4f.dylib -lfftw @FINKPREFIX@/lib/libfftw.dylib @FINKPREFIX@/lib/librfftw.dylib -lstdc++  /usr/lib/libSystem.B.dylib  -lSystem -lm -lc++
 
 
-# per le versioni dei file, modifica il numero e la data nel file makecif_version.fh
-# poi comprimi con
-# tar cvfz refmac5.8_v0001.tar.gz *.f *.fh *.f90 *.c *.cpp *.h makefile*
+include makefile_refmac.inc1
