From 4dde8f521add6fbc60641a280e38136c7b9e6ae5 Mon Sep 17 00:00:00 2001
From: Petr Danecek <pd3@sanger.ac.uk>
Date: Fri, 5 Apr 2013 15:31:19 +0100
Subject: [PATCH] bgzip now compiles

---
 bgzip.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bgzip.c b/bgzip.c
index ebcafa2..4789131 100644
--- a/bgzip.c
+++ b/bgzip.c
@@ -72,7 +72,7 @@ static int write_open(const char *fn, int is_forced)
 
 static void fail(BGZF* fp)
 {
-    fprintf(stderr, "Error: %s\n", fp->error);
+    fprintf(stderr, "Error: %d\n", fp->errcode);
     exit(1);
 }
 
-- 
1.8.1.5

