diff -ruN ScientificPython-2.9.4.orig/Scientific/_netcdf.c ScientificPython-2.9.4/Scientific/_netcdf.c
--- ScientificPython-2.9.4.orig/Scientific/_netcdf.c	2013-10-29 07:12:11.000000000 -0500
+++ ScientificPython-2.9.4/Scientific/_netcdf.c	2020-12-22 19:11:44.000000000 -0600
@@ -19,7 +19,7 @@
 #include "Numeric/arrayobject.h"
 #endif
 #endif
-#include "netcdf.h"
+#include "libnc-dap/netcdf.h"
 
 #define _NETCDF_MODULE
 #include "Scientific/netcdfmodule.h"
diff -ruN ScientificPython-2.9.4.orig/setup.py ScientificPython-2.9.4/setup.py
--- ScientificPython-2.9.4.orig/setup.py	2013-10-29 06:57:54.000000000 -0500
+++ ScientificPython-2.9.4/setup.py	2020-12-22 19:14:08.000000000 -0600
@@ -63,7 +63,7 @@
     for netcdf_prefix in ['/usr/local', '/usr', '/sw']:
         netcdf_include = os.path.join(netcdf_prefix, 'include')
         netcdf_lib = os.path.join(netcdf_prefix, 'lib')
-        if os.path.exists(os.path.join(netcdf_include, 'netcdf.h')):
+        if os.path.exists(os.path.join(netcdf_include, 'libnc-dap/netcdf.h')):
             break
     else:
         netcdf_prefix = None
@@ -90,14 +90,14 @@
     else:
         print "Using netCDF installation in ", netcdf_prefix
         netcdf_include = os.path.join(netcdf_prefix, 'include')
-        netcdf_h_file = os.path.join(netcdf_prefix, 'include', 'netcdf.h')
+        netcdf_h_file = os.path.join(netcdf_prefix, 'include', 'libnc-dap/netcdf.h')
         netcdf_lib = os.path.join(netcdf_prefix, 'lib')
     ext_modules = [Extension('Scientific._netcdf',
                              ['Scientific/_netcdf.c'],
                              include_dirs=['Include', netcdf_include]
                                           + numpy_include,
                              library_dirs=[netcdf_lib],
-                             libraries = ['netcdf'],
+                             libraries = ['nc-dap'],
                              extra_compile_args=extra_compile_args)]
 
 try:
