diff -ruN simcoal-orig/Makefile simcoal/Makefile
--- simcoal-orig/Makefile	2014-08-07 15:34:26.000000000 -0500
+++ simcoal/Makefile	2014-08-10 19:41:29.000000000 -0500
@@ -1,8 +1,8 @@
-CC=g++ -O
+CC=g++ -O3
 
 SOURCES=$(shell ls *.cpp)
 OBJECTS=$(shell for file in $(SOURCES);\
-                do echo -n $$file | sed -e "s/\(.*\)\.cpp/\1\.o/"; echo -n " ";\
+                do echo $$file | sed -e "s/\(.*\)\.cpp/\1\.o/"; echo " ";\
                 done)
 
 PRGNAME=simcoal
diff -ruN simcoal-orig/arrays.h simcoal/arrays.h
--- simcoal-orig/arrays.h	2014-08-07 15:34:26.000000000 -0500
+++ simcoal/arrays.h	2014-08-10 19:41:29.000000000 -0500
@@ -1,7 +1,7 @@
 #ifndef _MY_ARRAYS_HPP_
 #define _MY_ARRAYS_HPP_
 
-#include <fstream.h>
+#include <fstream>
 
 #include <limits.h>  //used to define MAX_INT
 ///#include <_null.h>   //definition of NULL //ANSI C++
diff -ruN simcoal-orig/coalmain.cpp simcoal/coalmain.cpp
--- simcoal-orig/coalmain.cpp	2014-08-07 15:34:26.000000000 -0500
+++ simcoal/coalmain.cpp	2014-08-10 19:42:28.000000000 -0500
@@ -9,7 +9,7 @@
 
 //////////////////////////////////////////////////////////////////////////////*/
 
-
+using namespace std;
 
 #include "genealgy.h"
 #include "public.h"
@@ -18,11 +18,12 @@
 #include "deme.h"
 
 #include "cstring.h"
-#include <fstream.h>
+#include <fstream>
 
 #include "coalmain.h"
 #include "cond_var.h"
 
+
 //------------------------------------------------------------------------------
 int
 TMeanMat::compute_mean(const int& n) {
@@ -945,4 +946,4 @@
       resetNode.reset_node_count();
 	}
    return 1;  
-};
\ No newline at end of file
+};
diff -ruN simcoal-orig/cstring.cpp simcoal/cstring.cpp
--- simcoal-orig/cstring.cpp	2014-08-07 15:34:26.000000000 -0500
+++ simcoal/cstring.cpp	2014-08-10 19:42:53.000000000 -0500
@@ -1,3 +1,5 @@
+using namespace std;
+
 #include "cstring.h"
 //for debug
 //#include "a_glbdef.hpp"
diff -ruN simcoal-orig/cstring.h simcoal/cstring.h
--- simcoal-orig/cstring.h	2014-08-07 15:34:26.000000000 -0500
+++ simcoal/cstring.h	2014-08-10 19:41:29.000000000 -0500
@@ -3,8 +3,8 @@
 
 
 #include <string.h>   //for functions strlen strcpy strcat
-#include <iostream.h>  //for << operator
-//#include <fstream.h>   //for read_token
+#include <iostream>  //for << operator
+//#include <fstream>   //for read_token
 #include <stdio.h>    //for sprint
 //#include <strstream.h>
 
diff -ruN simcoal-orig/deme.cpp simcoal/deme.cpp
--- simcoal-orig/deme.cpp	2014-08-07 15:34:26.000000000 -0500
+++ simcoal/deme.cpp	2014-08-10 19:43:31.000000000 -0500
@@ -6,13 +6,14 @@
 //
 ////////////////////////////////////////////////////////////////////////////////
 
+using namespace std;
 
 #include "deme.h"
  
 #include "cond_var.h"
 
 #ifdef _SHOW_MUT_DISTR_
-	#include <fstream.h>
+	#include <fstream>
 	ofstream f_site_hits("mut_hits_distr.sum");
 	static count_simul=0;
 #endif
@@ -686,7 +687,7 @@
          return 0;
       }
    }
-   else MinCoalTimes->set_elems(MAXLONG);
+   else MinCoalTimes->set_elems(LONG_MAX);
 
    //Compute total coalescent times within and among demes
    root.compute_total_coal_times_among_demes(*this,*CoalTimes, *PairDiff, *MinCoalTimes);
diff -ruN simcoal-orig/genealgy.cpp simcoal/genealgy.cpp
--- simcoal-orig/genealgy.cpp	2014-08-07 15:34:26.000000000 -0500
+++ simcoal/genealgy.cpp	2014-08-10 19:44:02.000000000 -0500
@@ -1,8 +1,9 @@
+using namespace std;
+
 #include "public.h"
 #include "genealgy.h"
 #include "deme.h"
 
-
 enum UNITS {WIDTH=6};
 
 //Initialization of static members of TNode
diff -ruN simcoal-orig/genealgy.h simcoal/genealgy.h
--- simcoal-orig/genealgy.h	2014-08-07 15:34:26.000000000 -0500
+++ simcoal/genealgy.h	2014-08-10 19:41:29.000000000 -0500
@@ -1,7 +1,7 @@
 #if !defined(genealogy_h)              // Sentry, use file only if it's not already included.
 #define genealogy_h
 
-#include <fstream.h>
+#include <fstream>
 #include <stdlib.h>
 #include "arrays.h"
 #include "mutation.h"
diff -ruN simcoal-orig/migrmat.cpp simcoal/migrmat.cpp
--- simcoal-orig/migrmat.cpp	2014-08-07 15:34:26.000000000 -0500
+++ simcoal/migrmat.cpp	2014-08-10 19:43:58.000000000 -0500
@@ -1,3 +1,5 @@
+using namespace std;
+
 //------------------------------------------------------------------------------
 // (c) L. Excoffier , LGB, University of Geneva, August 1995
 //
diff -ruN simcoal-orig/migrmat.h simcoal/migrmat.h
--- simcoal-orig/migrmat.h	2014-08-07 15:34:26.000000000 -0500
+++ simcoal/migrmat.h	2014-08-10 19:41:29.000000000 -0500
@@ -2,9 +2,13 @@
 #define _MIGRATION_H
 
 #include "cstring.h"
-#include <fstream.h>
-#include <values.h>
-#include <iomanip.h>
+#include <fstream>
+#if defined(WIN32) || (defined(__APPLE__) && defined(__MACH__))
+#  include <float.h>
+#else
+#  include <values.h>
+#endif
+#include <iomanip>
 #include <float.h>
 #include <math.h>
 
diff -ruN simcoal-orig/mutation.cpp simcoal/mutation.cpp
--- simcoal-orig/mutation.cpp	2014-08-07 15:34:26.000000000 -0500
+++ simcoal/mutation.cpp	2014-08-10 19:43:49.000000000 -0500
@@ -1,10 +1,12 @@
+using namespace std;
+
 #include "mutation.h"
 //#include <except.h>
 
 #include "cond_var.h"
 
 #ifdef _SHOW_GAMMA_DISTR_
-	#include <fstream.h>
+	#include <fstream>
 	ofstream f_gamma_distr("gamma_rates.distr");
 #endif
 
diff -ruN simcoal-orig/mutation.h simcoal/mutation.h
--- simcoal-orig/mutation.h	2014-08-07 15:34:26.000000000 -0500
+++ simcoal/mutation.h	2014-08-10 19:41:29.000000000 -0500
@@ -47,4 +47,4 @@
 };
 
 #endif
- 
\ No newline at end of file
+ 
diff -ruN simcoal-orig/public.cpp simcoal/public.cpp
--- simcoal-orig/public.cpp	2014-08-07 15:34:30.000000000 -0500
+++ simcoal/public.cpp	2014-08-10 19:43:08.000000000 -0500
@@ -1,3 +1,5 @@
+using namespace std;
+
 #include "public.h"
 
 /* (C) Copr. 1986-92 Numerical Recipes Software Y5jc. */
diff -ruN simcoal-orig/public.h simcoal/public.h
--- simcoal-orig/public.h	2014-08-07 15:34:30.000000000 -0500
+++ simcoal/public.h	2014-08-10 19:41:29.000000000 -0500
@@ -2,7 +2,7 @@
 #define __PUBLIC_H__
 
 #include <math.h>
-#include <iostream.h>
+#include <iostream>
 #include <stdlib.h>
 #include "cstring.h"
 #include "arrays.h"
@@ -79,4 +79,4 @@
 extern int abyx (double a, double x[], int n);
 //------------------------------------------------------------------------------
 
-#endif
\ No newline at end of file
+#endif
