diff -Nurd visual-5.74_release.orig/src/Makefile.in visual-5.74_release/src/Makefile.in
--- visual-5.74_release.orig/src/Makefile.in	2012-04-23 13:20:24.000000000 -0400
+++ visual-5.74_release/src/Makefile.in	2021-07-11 18:05:53.000000000 -0400
@@ -200,7 +200,7 @@
   ifeq ($(PYTHON_PLATFORM),darwin)
     # Special rules for OSX
     CVISUAL_LIBS += $(filter-out $(_FILTER_OUT), $(GTK_LIBS) \
-       $(GTHREAD_LIBS) /sw/lib/libboost_python-mt.a -lboost_thread-mt -lboost_signals)
+       $(GTHREAD_LIBS) -lboost_python-mt -lboost_thread-mt -lboost_signals)
     CXX_RULE = $(DEFAULT_CXX_RULE)
     LINK_RULE = $(OSX_SORULE)
     PLATFORM_TARGET = cvisualmodule.so
