--- /dev/null	2015-10-27 20:54:10.000000000 -0500
+++ a/system/Make.macosx-x11	2015-10-27 20:52:57.000000000 -0500
@@ -0,0 +1,55 @@
+#########################################################################
+# System-dependent definitions for Mac OS X
+#
+# David Strubbe 1 Oct. 2015
+# Tested on Yosemite, OSX 10.10.5
+# -----------------------------------------------------------------------
+#
+# !!! EDIT ACCORDING TO YOUR OWN NEEDS !!!
+# In particular, you must specify:
+# LIBPREFIX (root directory for third party [Tcl/Tk & FFTW3] libraries)
+
+# The following can also be modified as needed:
+# FC (Fortran compiler)
+# TK_LIB and TCL_LIB (location of Tcl and Tk)
+# FFTW3_LIB and FFTW3_INCDIR (location of FFTW3 library and header)
+
+# X11 and OpenGL from system are being used.
+# Change GLU_LIB, GL_LIB and GL_INCDIR if not using the system GL
+
+#------------------------------------------------------------------------
+# compilers & flags
+#------------------------------------------------------------------------
+CFLAGS = -O3 -DUSE_INTERP_RESULT
+CC     = cc
+FFLAGS = -O3
+FC     = gfortran
+
+#------------------------------------------------------------------------
+# X-libraries & include files
+#------------------------------------------------------------------------
+
+X_LIB     = /opt/X11/lib/libXmu.dylib /opt/X11/lib/libX11.dylib /opt/X11/lib/libXext.dylib
+X_INCDIR  = -I/opt/X11/include
+
+COMPILE_TCLTK = no # generally will not compile without passing args to configure script
+COMPILE_FFTW  = no # generally will not compile without passing args to configure script
+COMPILE_MESA  = no
+COMPILE_MESCHACH = no
+
+#------------------------------------------------------------------------
+# Libraries and include files
+#------------------------------------------------------------------------
+
+LIBPREFIX      = 
+LIBDIR      = $(LIBPREFIX)/lib
+INCDIR      = $(LIBPREFIX)/include
+
+TK_LIB      = $(LIBDIR)/libtk.dylib
+TCL_LIB     = $(LIBDIR)/libtcl.dylib
+GLU_LIB     = /opt/X11/lib/libGLU.dylib
+GL_LIB      = /opt/X11/lib/libGL.dylib
+FFTW3_LIB   = $(LIBDIR)/libfftw3.dylib
+
+GL_INCDIR    = -I/opt/X11/include
+FFTW3_INCDIR = -I$(INCDIR)
--- a/Makefile.orig	2012-03-12 16:08:34.000000000 -0400
+++ b/Makefile	2012-04-10 20:46:42.000000000 -0400
@@ -234,31 +234,31 @@
 	@echo
 	@echo "#-----"
 	@echo "#                          "
-	@echo "#   Installing XCRYSDEN to: $(prefix)"
+	@echo "#   Installing XCRYSDEN to: $(DESTDIR)$(prefix)"
 	@echo "#                          "
 	@echo "#-----"
 	@echo	
-	install -m755 -d $(prefix)/share/$(xcrysden)
-	cp -a $(IRON_ITEMS) $(prefix)/share/$(xcrysden)
+	install -m755 -d $(DESTDIR)$(prefix)/share/$(xcrysden)
+	cp -a $(IRON_ITEMS) $(DESTDIR)$(prefix)/share/$(xcrysden)
 	\
 	for subdir in Awk examples $(EXTERNAL_LIB_BWIDGET) images scripts Tcl util; do \
 		if test -d $$subdir; then \
-			install -m755 -d $(prefix)/share/$(xcrysden)/$$subdir; \
-			cp -a   $$subdir/* $(prefix)/share/$(xcrysden)/$$subdir; \
+			install -m755 -d $(DESTDIR)$(prefix)/share/$(xcrysden)/$$subdir; \
+			cp -a   $$subdir/* $(DESTDIR)$(prefix)/share/$(xcrysden)/$$subdir; \
 		fi; \
 	done
 	\
-	install -m755 -d $(prefix)/share/man/man1
-	install -m644 $(MAN_FILES) $(prefix)/share/man/man1/
-	gzip -f $(addprefix $(prefix)/share/man/man1/,$(MAN_PAGES))
+	install -m755 -d $(DESTDIR)$(prefix)/share/man/man1
+	install -m644 $(MAN_FILES) $(DESTDIR)$(prefix)/share/man/man1/
+	gzip -f $(addprefix $(DESTDIR)$(prefix)/share/man/man1/,$(MAN_PAGES))
 	\
-	install -m755 -d $(prefix)/lib/$(xcrysden)
-	install -m755 bin/*  $(prefix)/lib/$(xcrysden)/
+	install -m755 -d $(DESTDIR)$(prefix)/lib/$(xcrysden)
+	install -m755 bin/*  $(DESTDIR)$(prefix)/lib/$(xcrysden)/
 	\
 	prefix=$(prefix) xcrysden=$(xcrysden) sh sys_utils/wrappers.sh
-	if test ! -d $(prefix)/bin; then install -m755 -d $(prefix)/bin; fi
+	if test ! -d $(DESTDIR)$(prefix)/bin; then install -m755 -d $(DESTDIR)$(prefix)/bin; fi
 	for prog in $(PROGS); do \
-		install -m755 $$prog.wrapper   $(prefix)/bin/$$prog; \
+		install -m755 $$prog.wrapper   $(DESTDIR)$(prefix)/bin/$$prog; \
 	done
 
 
