diff -Nurd xdrawchem-1.10.2.1.orig/README-docs.txt xdrawchem-1.10.2.1/README-docs.txt
--- xdrawchem-1.10.2.1.orig/README-docs.txt	1969-12-31 19:00:00.000000000 -0500
+++ xdrawchem-1.10.2.1/README-docs.txt	2021-03-06 17:05:14.000000000 -0500
@@ -0,0 +1,3 @@
+Documentation is available in the About and Help menus. You can also find
+copies of those files in the Contents/Resource/doc directory inside the
+%n.app bundle.
diff -Nurd xdrawchem-1.10.2.1.orig/xdrawchem.pro xdrawchem-1.10.2.1/xdrawchem.pro
--- xdrawchem-1.10.2.1.orig/xdrawchem.pro	2017-01-09 07:25:32.000000000 -0500
+++ xdrawchem-1.10.2.1/xdrawchem.pro	2021-03-06 15:36:01.000000000 -0500
@@ -2,27 +2,8 @@
 TARGET = xdrawchem
 
 # edit to match your OB install
-exists(/usr/include/openbabel-2.0/openbabel/mol.h) {
-INCLUDEPATH += /usr/include/openbabel-2.0
-LIBS += -lopenbabel
-}
-exists(/usr/local/include/openbabel-2.0/openbabel/mol.h) {
-INCLUDEPATH += /usr/local/include/openbabel-2.0
-LIBS += -lopenbabel
-}
-exists(/Developer/openbabel-2.4.1/include/openbabel/mol.h) {
-INCLUDEPATH += /Developer/openbabel-2.4.1/include
-LIBS += -L/Developer/openbabel-2.4.1/lib/Debug -lopenbabel
-}
-# exit if openbabel wasn't found
-contains( LIBS, -lopenbabel ) {
-} else {
-error(OpenBabel was not found.  Edit xdrawchem.pro and add valid include and lib values)
-# if you get this error, comment the above with #, uncomment the variables below, and set
-# the paths to match your OpenBabel install
-#INCLUDEPATH += /usr/local/include/openbabel-2.0
-#LIBS += -lopenbabel
-}
+CONFIG += link_pkgconfig
+PKGCONFIG += openbabel-3
 # set PREFIX if not set (it is set when building RPM)
 isEmpty(PREFIX) {
 PREFIX = /usr/local
