diff -Nurd xraylib-2.16.0.orig/example/Makefile.in xraylib-2.16.0/example/Makefile.in
--- xraylib-2.16.0.orig/example/Makefile.in	2012-09-17 11:19:29
+++ xraylib-2.16.0/example/Makefile.in	2025-12-25 20:31:22
@@ -842,7 +842,7 @@
 #test the lua bindings
 xrlexample9.sh:
 	echo "Testing lua bindings"
-	echo "LD_LIBRARY_PATH=\"${top_builddir}/src/.libs\" DYLD_LIBRARY_PATH=\"${top_builddir}/src/.libs\" LUA_CPATH=\"${top_builddir}/lua/?.so\" lua  ${top_srcdir}/example/xrlexample9.lua" >> xrlexample9.sh
+	echo "LD_LIBRARY_PATH=\"${top_builddir}/src/.libs\" DYLD_LIBRARY_PATH=\"${top_builddir}/src/.libs\" LUA_CPATH=\"${top_builddir}/lua/?.so\" ${LUA}  ${top_srcdir}/example/xrlexample9.lua" >> xrlexample9.sh
 	chmod +x xrlexample9.sh
 
 clean-local:
diff -Nurd xraylib-2.16.0.orig/idl/Makefile.in xraylib-2.16.0/idl/Makefile.in
--- xraylib-2.16.0.orig/idl/Makefile.in	2012-09-17 11:19:29
+++ xraylib-2.16.0/idl/Makefile.in	2025-12-25 20:31:22
@@ -478,7 +478,7 @@
 #Maybe this could be rewritten using libtool...
 @ENABLE_IDL_TRUE@idlxrl:	xraylib_idl.c ../src/libxrl.la
 @ENABLE_IDL_TRUE@	$(CC) -c $(AM_CPPFLAGS) -I@RSIIDL_DIR@/external/include $(CFLAGS) @IDL_CFLAGS@ ${srcdir}/xraylib_idl.c
-@ENABLE_IDL_TRUE@	gcc @IDL_LD_FLAGS@ -o libxrlidl.so xraylib_idl.o -L${top_builddir}/src/.libs -lxrl
+@ENABLE_IDL_TRUE@	gcc @IDL_LD_FLAGS@ -o libxrlidl.so xraylib_idl.o ${top_builddir}/src/.libs/libxrl.dylib
 
 #install the IDL bindings specific files
 @ENABLE_IDL_TRUE@install-exec-local:
diff -Nurd xraylib-2.16.0.orig/java/Makefile.in xraylib-2.16.0/java/Makefile.in
--- xraylib-2.16.0.orig/java/Makefile.in	2012-09-17 11:19:29
+++ xraylib-2.16.0/java/Makefile.in	2025-12-25 20:31:22
@@ -516,7 +516,7 @@
 @ENABLE_JAVA_TRUE@xraylib.java: ${top_srcdir}/src/xraylib.i
 @ENABLE_JAVA_TRUE@	$(SWIG) -I${top_srcdir}/include -includeall -java -o xraylib_wrap.c ${top_srcdir}/src/xraylib.i
 @ENABLE_JAVA_TRUE@	$(CC) -I${top_srcdir}/include -c $(CCSHARED) $(CFLAGS) $(JAVACFLAGS) xraylib_wrap.c $(JAVAINC)
-@ENABLE_JAVA_TRUE@	$(JAVALDSHARED) $(CFLAGS) xraylib_wrap.o $(JAVADYNAMICLINKING) -L@abs_top_builddir@/src/.libs -lxrl -o $(JAVALIBRARYPREFIX)xraylib$(JAVASO)
+@ENABLE_JAVA_TRUE@	$(JAVALDSHARED) $(CFLAGS) xraylib_wrap.o $(JAVADYNAMICLINKING) @abs_top_builddir@/src/.libs/libxrl.dylib -o $(JAVALIBRARYPREFIX)xraylib$(JAVASO)
 
 @ENABLE_JAVA_TRUE@install-exec-local:
 @ENABLE_JAVA_TRUE@	$(MKDIR_P) $(DESTDIR)$(xrljavadir)
diff -Nurd xraylib-2.16.0.orig/lua/Makefile.in xraylib-2.16.0/lua/Makefile.in
--- xraylib-2.16.0.orig/lua/Makefile.in	2012-09-17 11:19:29
+++ xraylib-2.16.0/lua/Makefile.in	2025-12-25 20:31:22
@@ -474,7 +474,7 @@
 @ENABLE_LUA_TRUE@luaxrl: ${top_srcdir}/src/xraylib.i
 @ENABLE_LUA_TRUE@	$(SWIG) -I${top_srcdir}/include -includeall -o xraylib_wrap.c -lua ${top_srcdir}/src/xraylib.i
 @ENABLE_LUA_TRUE@	$(CC) $(LUA_CFLAGS) -I${top_srcdir}/include -Wall $(LUA_INCLUDE) -c xraylib_wrap.c -o xraylib_wrap.o
-@ENABLE_LUA_TRUE@	$(CC) -Wall $(LUA_LD_FLAGS) -o xraylib.so xraylib_wrap.o -L../src/.libs -lxrl $(LUA_LIB) 
+@ENABLE_LUA_TRUE@	$(CC) -Wall $(LUA_LD_FLAGS) -o xraylib.so xraylib_wrap.o ../src/.libs/libxrl.dylib $(LUA_LIB) 
 
 @ENABLE_LUA_TRUE@install-exec-local: xraylib.so
 @ENABLE_LUA_TRUE@	$(MKDIR_P) $(DESTDIR)$(libdir)/lua/5.1
diff -Nurd xraylib-2.16.0.orig/perl/Makefile.in xraylib-2.16.0/perl/Makefile.in
--- xraylib-2.16.0.orig/perl/Makefile.in	2012-09-17 11:19:29
+++ xraylib-2.16.0/perl/Makefile.in	2025-12-25 20:31:22
@@ -481,7 +481,7 @@
 @ENABLE_PERL_TRUE@	@echo "'OBJECT'=> 'xraylib_wrap.o'," >> Makefile.PL
 @ENABLE_PERL_TRUE@	@echo "'FIRST_MAKEFILE'=> 'Makefile2'" >> Makefile.PL
 @ENABLE_PERL_TRUE@	@echo ");" >> Makefile.PL
-@ENABLE_PERL_TRUE@	$(PERL) Makefile.PL CCDLFLAGS="$(ARCHFLAGS)" LDDLFLAGS="$(ARCHFLAGS)" INC="-I${top_srcdir}/include" CCFLAGS="$(ARCHFLAGS)" INSTALL_BASE=$(DESTDIR)/${prefix} VERSION=$(VERSION) LIBS="-L../src/.libs -lxrl"
+@ENABLE_PERL_TRUE@	$(PERL) Makefile.PL CCDLFLAGS="$(ARCHFLAGS)" LDDLFLAGS="$(ARCHFLAGS)" INC="-I${top_srcdir}/include" CCFLAGS="$(ARCHFLAGS)" INSTALL_BASE=$(DESTDIR)/${prefix} VERSION=$(VERSION) LIBS="../src/.libs/libxrl.dylib"
 @ENABLE_PERL_TRUE@	$(MAKE) $(AM_MAKEFLAGS)	INSTALL_BASE=$(DESTDIR)/$(prefix) VERSION=$(VERSION) -f Makefile2
 
 @ENABLE_PERL_TRUE@install-exec-local:
