--- xrootd.orig/cmake/XRootDOSDefs.cmake.orig	2011-12-12 17:27:21.000000000 +0100
+++ xrootd.new/cmake/XRootDOSDefs.cmake	2011-12-12 17:28:21.000000000 +0100
@@ -49,9 +49,10 @@
   add_definitions( -DLT_MODULE_EXT=".dylib" )
   set( CMAKE_INSTALL_LIBDIR "lib" )
   set( CMAKE_INSTALL_BINDIR "bin" )
-  set( CMAKE_INSTALL_MANDIR "man" )
+  set( CMAKE_INSTALL_MANDIR "share/man" )
   set( CMAKE_INSTALL_INCLUDEDIR "include" )
   set( CMAKE_INSTALL_DATADIR "share" )
+  set( CMAKE_INSTALL_NAME_DIR "@PREFIX@/lib" )
 endif()

 #-------------------------------------------------------------------------------
