diff -Naur growl-python-1.1.4.orig/Growl.py growl-python-1.1.4/Growl.py
--- growl-python-1.1.4.orig/Growl.py	2007-09-20 12:53:34.000000000 -0400
+++ growl-python-1.1.4/Growl.py	2009-12-18 20:46:35.000000000 -0500
@@ -14,7 +14,7 @@
 import _growl
 import types
 import struct
-import md5
+import hashlib
 import socket
 
 GROWL_UDP_PORT=9887
@@ -129,7 +129,7 @@
 		return self.encodePassword(data)
 
 	def encodePassword(self, data):
-		checksum = md5.new()
+		checksum = hashlib.md5()
 		checksum.update(data)
 		if self.password:
 		   checksum.update(self.password)
@@ -173,11 +173,11 @@
 	def __init__(self, applicationName=None, notifications=None, defaultNotifications=None, applicationIcon=None, hostname=None, password=None):
 		if applicationName:
 			self.applicationName = applicationName
-		assert(self.applicationName, 'An application name is required.')
+		assert self.applicationName, 'An application name is required.'
 
 		if notifications:
 			self.notifications = list(notifications)
-		assert(self.notifications, 'A sequence of one or more notification names is required.')
+		assert self.notifications, 'A sequence of one or more notification names is required.'
 
 		if defaultNotifications is not None:
 			self.defaultNotifications = list(defaultNotifications)
