--- a/pyobjc-core/setup.py	2008-07-02 12:52:52.000000000 +1000
+++ b/pyobjc-core/setup.py	2008-07-03 01:06:55.000000000 +1000
@@ -295,7 +295,7 @@
 Topic :: Software Development :: User Interfaces
 """.splitlines())
 
-install_requires = setup_requires = ['py2app>=0.4.0', 'bdist_mpkg>=0.4.2']
+###install_requires = setup_requires = ['py2app>=0.4.0', 'bdist_mpkg>=0.4.2']
 
 dist = setup(
     name = "pyobjc-core", 
@@ -312,8 +312,8 @@
     packages = [ 'objc', 'objc.test', 'PyObjCTools' ], # 'ExceptionHandling'
     namespace_packages = ['PyObjCTools'],
     package_dir = { '': 'Lib' },
-    install_requires = install_requires,
-    setup_requires = setup_requires,
+###    install_requires = install_requires,
+###    setup_requires = setup_requires,
     extra_path = "PyObjC",
     cmdclass = extra_cmdclass,
     options = OPTIONS,
