diff -ru pysvn-1.7.2.orig/Source/setup_configure.py pysvn-1.7.2/Source/setup_configure.py
--- pysvn-1.7.2.orig/Source/setup_configure.py	2009-12-29 06:31:15.000000000 -0500
+++ pysvn-1.7.2/Source/setup_configure.py	2010-08-08 11:11:46.000000000 -0400
@@ -316,7 +316,7 @@
             # python framework will be used and not the one matching this python
             var_prefix = distutils.sysconfig.get_config_var('prefix')
             var_ldlibrary = distutils.sysconfig.get_config_var('LDLIBRARY')
-            framework_lib = os.path.join( var_prefix, os.path.basename( var_ldlibrary ) )
+            framework_lib = os.path.join( var_prefix, "lib", "python" + distutils.sysconfig.get_python_version(), "config", os.path.basename( var_ldlibrary ) )
 
             if self.is_atleast_mac_os_x_version( (10,4) ) >= 0:
                 if self.verbose:
@@ -664,36 +664,21 @@
 PYTHON=%(python_exe)s
 SVN_INCLUDE=%(svn_include)s
 CCC=g++
-CCCFLAGS=-Wall -Wno-long-double -fPIC -fexceptions -frtti %(includes)s %(py_cflags)s %(debug_cflags)s
+CCCFLAGS=-Wall -fPIC -fexceptions -frtti %(includes)s %(py_cflags)s %(debug_cflags)s
 CC=gcc
-CCFLAGS=-Wall -Wno-long-double -fPIC %(includes)s %(debug_cflags)s
+CCFLAGS=-Wall -fPIC %(includes)s %(debug_cflags)s
 PYCXX=%(pycxx_dir)s
 PYCXXSRC=%(pycxx_src_dir)s
-LDSHARED=g++ -bundle %(debug_cflags)s -u _PyMac_Error %(frameworks)s
-LDLIBS= \
-%(svn_lib_dir)s/libsvn_client-1.a \
-%(svn_lib_dir)s/libsvn_subr-1.a \
-%(svn_lib_dir)s/libsvn_wc-1.a \
-%(svn_lib_dir)s/libsvn_ra-1.a \
-%(svn_lib_dir)s/libsvn_ra_dav-1.a \
-%(svn_lib_dir)s/libsvn_ra_local-1.a \
-%(svn_lib_dir)s/libsvn_ra_svn-1.a \
-%(svn_lib_dir)s/libsvn_delta-1.a \
-%(svn_lib_dir)s/libsvn_repos-1.a \
-%(svn_lib_dir)s/libsvn_fs-1.a \
-%(svn_lib_dir)s/libsvn_fs_fs-1.a \
-%(svn_lib_dir)s/libsvn_fs_base-1.a \
-%(svn_lib_dir)s/libsvn_diff-1.a \
-%(apr_lib_dir)s/libaprutil-0.a \
-%(apr_lib_dir)s/libapr-0.a \
-%(svn_lib_dir)s/libneon.a \
-%(svn_lib_dir)s/libssl.a \
-%(svn_lib_dir)s/libcrypto.a \
-%(svn_lib_dir)s/libexpat.a \
-%(svn_lib_dir)s/libxml2.a \
-%(svn_lib_dir)s/libdb-4.3.a \
- -lintl.3 -liconv.2 -lz
-
+LDSHARED=g++ -bundle %(debug_cflags)s %(frameworks)s
+LDLIBS=-L%(svn_lib_dir)s \
+-L%(apr_lib_dir)s \
+-lsvn_client-1 \
+-lsvn_repos-1 \
+-lsvn_wc-1 \
+-lsvn_fs-1 \
+-lsvn_subr-1 \
+-lsvn_diff-1 \
+-l%(lib_apr)s
 #include pysvn_common.mak
 '''
 
@@ -865,7 +850,7 @@
 
     def find_apr_lib( self ):
         last_exception = None
-        lib_list = [(self.get_lib_name_for_platform( 'libapr-1' ), 'apr-1'),
+        lib_list = [(self.get_lib_name_for_platform( 'libapr' ), 'apr'),
                     (self.get_lib_name_for_platform( 'libapr-0' ), 'apr-0')]
 
         for apr_libname, self.lib_apr in lib_list:
