diff -ru xmlwitch-0.2.orig/xmlwitch.py xmlwitch-0.2/xmlwitch.py
--- xmlwitch-0.2.orig/xmlwitch.py	2011-01-01 16:15:53.000000000 -0500
+++ xmlwitch-0.2/xmlwitch.py	2011-01-12 18:12:39.000000000 -0500
@@ -1,5 +1,5 @@
-from __future__ import with_statement
-from StringIO import StringIO
+
+from io import StringIO
 from xml.sax import saxutils
 from keyword import kwlist as PYTHON_KWORD_LIST
 
@@ -29,14 +29,13 @@
         return Element(name, self)
     
     def __str__(self):
-        return self._document.getvalue().encode(self._encoding).strip()
+        return self._document.getvalue().strip()
         
-    def __unicode__(self):
+    def __byte__(self):
         return self._document.getvalue().decode(self._encoding).strip()
         
     def write(self, content):
         """Write raw content to the document"""
-        content = content.decode(self._encoding)
         self._document.write('%s' % content)
         
     def write_escaped(self, content):
@@ -45,7 +44,6 @@
         
     def write_indented(self, content):
         """Write indented content to the document"""
-        content = content.decode(self._encoding)
         self.write('%s%s\n' % (self._indent * self._indentation, content))
 
 builder = Builder # 0.1 backward compatibility
@@ -92,7 +90,7 @@
     def _serialized_attrs(self):
         """Serialize attributes for element insertion"""
         serialized = []
-        for attr, value in self.attributes.items():
+        for attr, value in list(self.attributes.items()):
             serialized.append(' %s=%s' % (
                 self._nameprep(attr), saxutils.quoteattr(value)
             ))
