--- xforms-1.0-release/fd2ps/Imakefile.orig	Mon Dec  9 06:31:22 2002
+++ xforms-1.0-release/fd2ps/Imakefile	Mon Dec  9 07:05:56 2002
@@ -29,12 +29,10 @@
 
 OBJS	= $(SRCS:.c=.o)
 
-#if !defined( PpcDarwinArchitecture )
 FORMSLIBDIR     = $(TOP)/lib
 FORMSINCDIR     = $(FORMSLIBDIR)       
 FORMSLIB        = -L$(FORMSLIBDIR) -lforms
 DEPFORMSLIB     = $(FORMSLIBDIR)/libforms.a
-#endif
 
 IMAGELIBDIR     = $(TOP)/image
 IMAGEINCDIR     = $(IMAGELIBDIR)       
@@ -94,8 +92,8 @@
 #endif
 
 #if StaticLink
-STATIC		= -Xlinker -Bstatic
-UNSTATIC	= -Xlinker -Bdynamic
+STATIC		= 
+UNSTATIC	= 
 #endif
 
 IMAKE_DEFINES	= \
--- xforms-1.0-release/fdesign/Imakefile.orig	Mon Dec  9 07:23:37 2002
+++ xforms-1.0-release/fdesign/Imakefile	Mon Dec  9 07:24:20 2002
@@ -54,11 +54,11 @@
 
 #if StaticLink
 #if HasGcc2
-STATIC		= -Xlinker -Bstatic
-UNSTATIC	= -Xlinker -Bdynamic
+STATIC		= 
+UNSTATIC	= 
 #else
-STATIC		= -static
-UNSTATIC	= -dynamic
+STATIC		= 
+UNSTATIC	= 
 #endif
 
 #if NonstandardXpm
--- xforms-1.0-release/gl/Imakefile.orig	Mon Dec  9 06:25:49 2002
+++ xforms-1.0-release/gl/Imakefile	Mon Dec  9 06:26:06 2002
@@ -31,7 +31,7 @@
 #endif
 
 #if defined( PpcDarwinArchitecture )
-SHLIBLDFLAGS	+= -lSystem -lcc_dynamic
+SHLIBLDFLAGS	+= -lSystem -lcc_dynamic -flat_namespace -undefined suppress
 #endif
 
 #if Solaris64Bit
--- xforms-1.0-release/image/Imakefile.orig	Mon Dec  9 06:32:18 2002
+++ xforms-1.0-release/image/Imakefile	Mon Dec  9 06:32:43 2002
@@ -80,12 +80,12 @@
 
 EXTRA_INCLUDES	= \
 		  $(XPMINCDIR) \
-		  $(TIFFINCDIR) \
-		  $(JPEGINCDIR) \
 		  -I$(BUILDINCDIR) \
 		  -I$(TOP)/lib \
 		  -I$(TOP)/lib/private \
-		  -I.
+		  -I. \
+		  $(TIFFINCDIR) \
+		  $(JPEGINCDIR) 
 
 DEPEND_DEFINES	= \
 		  -I/usr/local/include
@@ -115,7 +115,7 @@
 #endif
 
 #if defined( PpcDarwinArchitecture )
-SHLIBLDFLAGS	+= -lSystem -lcc_dynamic
+SHLIBLDFLAGS	+= -lSystem -lcc_dynamic -flat_namespace -undefined suppress
 #endif
 
 REQUIREDLIBS	= \
--- xforms-1.0-release/Imakefile.orig	Mon Dec  9 06:27:28 2002
+++ xforms-1.0-release/Imakefile	Mon Dec  9 07:42:57 2002
@@ -2,7 +2,7 @@
 
 /* Fairly important stuff */
 
-#define OptimizationLevel -O4
+#define OptimizationLevel -O2
 #define DoWall NO
 
 #define BuildShared YES
@@ -32,23 +32,23 @@
 
 /* Out in the Tall Weeds */
 
-#define InstallNonstandard NO
-#define XFormsNonstandardInstallDir /usr/local
+#define InstallNonstandard YES
+#define XFormsNonstandardInstallDir @PREFIX@
 
 #define NonstandardXpm NO
 
 #define XpmIncLoc /usr/local/include
 #define XpmLibLoc /usr/local/lib
 
-#define NonstandardTiff NO
+#define NonstandardTiff YES
 
-#define TiffIncLoc /usr/local/include
-#define TiffLibLoc /usr/local/lib
+#define TiffIncLoc @PREFIX@/include
+#define TiffLibLoc @PREFIX@/lib
 
-#define NonstandardJpeg NO
+#define NonstandardJpeg YES
 
-#define JpegIncLoc /usr/local/include
-#define JpegLibLoc /usr/local/lib
+#define JpegIncLoc @PREFIX@/include
+#define JpegLibLoc @PREFIX@/lib
 
 ######################################################################
 #                                                                    #
--- xforms-1.0-release/lib/Imakefile.orig	Mon Dec  9 06:30:19 2002
+++ xforms-1.0-release/lib/Imakefile	Mon Dec  9 06:30:32 2002
@@ -171,7 +171,7 @@
 #endif
 
 #if defined( PpcDarwinArchitecture )
-SHLIBLDFLAGS	+= -flat_namespace
+SHLIBLDFLAGS	+= -flat_namespace -undefined suppress
 #endif
 
 REQUIREDLIBS	= \
