diff -Nurb xplanet-1.0.1/src/libdisplay/DisplayBase.h xplanet-1.0.1-new/src/libdisplay/DisplayBase.h
--- xplanet-1.0.1/src/libdisplay/DisplayBase.h	Sun Mar 16 00:39:26 2003
+++ xplanet-1.0.1-new/src/libdisplay/DisplayBase.h	Wed Apr 16 18:58:10 2003
@@ -9,6 +9,7 @@
 #ifdef HAVE_LIBFREETYPE
 #include <ft2build.h>
 #include FT_FREETYPE_H
+#include <freetype2/freetype/freetype.h>
 #include FT_GLYPH_H
 #endif
 
