diff -Nurd -x'*~' evolution-2.24.4.orig/addressbook/printing/Makefile.am evolution-2.24.4/addressbook/printing/Makefile.am
--- evolution-2.24.4.orig/addressbook/printing/Makefile.am	2008-10-13 04:32:41.000000000 -0400
+++ evolution-2.24.4/addressbook/printing/Makefile.am	2009-02-26 02:26:14.000000000 -0500
@@ -8,7 +8,6 @@
 	e-contact-print.glade
 
 INCLUDES =					\
-	$(GNOME_INCLUDEDIR)			\
 	-DG_LOG_DOMAIN=\"addressbook-printing\"	\
 	-I$(top_srcdir)/addressbook		\
 	-I$(top_srcdir)/addressbook/backend	\
diff -Nurd -x'*~' evolution-2.24.4.orig/calendar/conduits/common/Makefile.am evolution-2.24.4/calendar/conduits/common/Makefile.am
--- evolution-2.24.4.orig/calendar/conduits/common/Makefile.am	2008-10-19 23:28:33.000000000 -0400
+++ evolution-2.24.4/calendar/conduits/common/Makefile.am	2009-02-26 02:38:34.000000000 -0500
@@ -5,17 +5,15 @@
 	$(EVOLUTION_CALENDAR_CONDUIT_CFLAGS)
 
 # Calendar Conduit library
-privconduit_LTLIBRARIES = libecalendar_common_conduit.la
+noinst_LTLIBRARIES = libecalendar_common_conduit.la
 
 ecalendar_common_conduitincludedir = $(privincludedir)/calendar/conduits/common
 
-ecalendar_common_conduitinclude_HEADERS =		\
-	libecalendar-common-conduit.h
 
 libecalendar_common_conduit_la_SOURCES = 	\
+	libecalendar-common-conduit.h		\
 	libecalendar-common-conduit.c
 
-libecalendar_common_conduit_la_LDFLAGS = -module -avoid-version $(NO_UNDEFINED)
 libecalendar_common_conduit_la_LIBADD = 				\
 	$(top_builddir)/e-util/libeutil.la				\
 	$(top_builddir)/e-util/libeconduit.la		 		\
diff -Nurd -x'*~' evolution-2.24.4.orig/configure.in evolution-2.24.4/configure.in
--- evolution-2.24.4.orig/configure.in	2009-01-19 22:18:00.000000000 -0500
+++ evolution-2.24.4/configure.in	2009-02-26 03:01:32.000000000 -0500
@@ -131,7 +131,6 @@
 AM_DISABLE_STATIC
 AC_LIBTOOL_WIN32_DLL
 AM_PROG_LIBTOOL
-DOLT
 
 PKG_PROG_PKG_CONFIG
 
diff -Nurd -x'*~' evolution-2.24.4.orig/mail/em-folder-browser.c evolution-2.24.4/mail/em-folder-browser.c
--- evolution-2.24.4.orig/mail/em-folder-browser.c	2008-10-13 04:37:24.000000000 -0400
+++ evolution-2.24.4/mail/em-folder-browser.c	2009-02-25 01:53:43.000000000 -0500
@@ -98,7 +98,7 @@
 #include <gtkhtml/gtkhtml.h>
 
 extern CamelSession *session;
-CamelStore *vfolder_store; /* the 1 static vfolder store */
+extern CamelStore *vfolder_store; /* the static vfolder store comes from mail-vfolder.c */
 
 #define d(x)
 
diff -Nurd -x'*~' evolution-2.24.4.orig/plugins/caldav/Makefile.am evolution-2.24.4/plugins/caldav/Makefile.am
--- evolution-2.24.4.orig/plugins/caldav/Makefile.am	2008-10-13 04:36:59.000000000 -0400
+++ evolution-2.24.4/plugins/caldav/Makefile.am	2009-02-26 02:26:32.000000000 -0500
@@ -1,8 +1,8 @@
 INCLUDES = 						\
-	$(EVOLUTION_CALENDAR_CFLAGS)			\
 	-I .						\
 	-I$(top_srcdir)	                                \
-	-DCALDAV_GLADEDIR=\""$(gladedir)"\"		
+	-DCALDAV_GLADEDIR=\""$(gladedir)"\"		\
+	$(EVOLUTION_CALENDAR_CFLAGS)
 
 @EVO_PLUGIN_RULE@
 
diff -Nurd -x'*~' evolution-2.24.4.orig/plugins/google-account-setup/Makefile.am evolution-2.24.4/plugins/google-account-setup/Makefile.am
--- evolution-2.24.4.orig/plugins/google-account-setup/Makefile.am	2008-10-13 04:37:00.000000000 -0400
+++ evolution-2.24.4/plugins/google-account-setup/Makefile.am	2009-02-26 02:32:44.000000000 -0500
@@ -1,9 +1,9 @@
 INCLUDES = 						\
-	$(EVOLUTION_CALENDAR_CFLAGS)			\
-	$(EVOLUTION_ADDRESSBOOK_CFLAGS) \
 	-I .						\
 	-I$(top_srcdir)	                                \
-	-DCALDAV_GLADEDIR=\""$(gladedir)"\"		
+	-DCALDAV_GLADEDIR=\""$(gladedir)"\"		\
+	$(EVOLUTION_CALENDAR_CFLAGS)			\
+	$(EVOLUTION_ADDRESSBOOK_CFLAGS)
 
 @EVO_PLUGIN_RULE@
 
diff -Nurd -x'*~' evolution-2.24.4.orig/plugins/mail-to-task/Makefile.am evolution-2.24.4/plugins/mail-to-task/Makefile.am
--- evolution-2.24.4.orig/plugins/mail-to-task/Makefile.am	2008-10-13 04:37:00.000000000 -0400
+++ evolution-2.24.4/plugins/mail-to-task/Makefile.am	2009-02-26 03:06:15.000000000 -0500
@@ -12,7 +12,7 @@
 liborg_gnome_mail_to_task_la_LIBADD =	\
 	$(top_builddir)/e-util/libeutil.la				\
 	$(top_builddir)/calendar/common/libevolution-calendarprivate.la	\
-	$(top_builddir)/mail/libevolution-mail.la			\ 
+	$(top_builddir)/mail/libevolution-mail.la			\
 	$(EVOLUTION_CALENDAR_LIBS)					\
 	$(EVOLUTION_MAIL_LIBS)
 
diff -Nurd -x'*~' evolution-2.24.4.orig/plugins/webdav-account-setup/Makefile.am evolution-2.24.4/plugins/webdav-account-setup/Makefile.am
--- evolution-2.24.4.orig/plugins/webdav-account-setup/Makefile.am	2008-10-13 04:37:00.000000000 -0400
+++ evolution-2.24.4/plugins/webdav-account-setup/Makefile.am	2009-02-26 02:35:17.000000000 -0500
@@ -1,7 +1,7 @@
 INCLUDES = \
-	$(EVOLUTION_ADDRESSBOOK_CFLAGS)      \
 	-I .				     \
-	-I$(top_srcdir)
+	-I$(top_srcdir)				\
+	$(EVOLUTION_ADDRESSBOOK_CFLAGS)
 
 @EVO_PLUGIN_RULE@
 
