diff -Nurd -x'*~' glib-2.54.1.orig/docs/reference/gio/Makefile.in glib-2.54.1/docs/reference/gio/Makefile.in
--- glib-2.54.1.orig/docs/reference/gio/Makefile.in	2017-10-02 11:02:23.000000000 -0400
+++ glib-2.54.1/docs/reference/gio/Makefile.in	2017-10-23 03:04:25.000000000 -0400
@@ -93,17 +93,17 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@ENABLE_MAN_TRUE@am__append_1 = \
-@ENABLE_MAN_TRUE@	gapplication.1			\
-@ENABLE_MAN_TRUE@	gio-querymodules.1		\
-@ENABLE_MAN_TRUE@	glib-compile-schemas.1		\
-@ENABLE_MAN_TRUE@	glib-compile-resources.1	\
-@ENABLE_MAN_TRUE@	gsettings.1			\
-@ENABLE_MAN_TRUE@	gresource.1			\
-@ENABLE_MAN_TRUE@	gdbus.1				\
-@ENABLE_MAN_TRUE@	gdbus-codegen.1			\
-@ENABLE_MAN_TRUE@	gio.1				\
-@ENABLE_MAN_TRUE@	$(NULL)
+am__append_1 = \
+	gapplication.1			\
+	gio-querymodules.1		\
+	glib-compile-schemas.1		\
+	glib-compile-resources.1	\
+	gsettings.1			\
+	gresource.1			\
+	gdbus.1				\
+	gdbus-codegen.1			\
+	gio.1				\
+	$(NULL)
 
 subdir = docs/reference/gio
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
diff -Nurd -x'*~' glib-2.54.1.orig/docs/reference/gio/glib-compile-schemas.1 glib-2.54.1/docs/reference/gio/glib-compile-schemas.1
--- glib-2.54.1.orig/docs/reference/gio/glib-compile-schemas.1	2017-10-02 11:07:53.000000000 -0400
+++ glib-2.54.1/docs/reference/gio/glib-compile-schemas.1	2017-10-23 03:04:25.000000000 -0400
@@ -50,7 +50,7 @@
 subdirectories of all directories specified in the
 \fBXDG_DATA_DIRS\fR
 environment variable\&. The usual location to install schema files is
-/usr/share/glib\-2\&.0/schemas\&.
+@PREFIX@/share/glib\-2\&.0/schemas\&.
 .PP
 In addition to schema files, glib\-compile\-schemas reads \*(Aqvendor override\*(Aq files, which are key files that can override default values for keys in the schemas\&. The group names in the key files are the schema id, and the values are written in serialized GVariant form\&. Vendor override files must have the filename extension
 \&.gschema\&.override\&.
diff -Nurd -x'*~' glib-2.54.1.orig/docs/reference/glib/Makefile.in glib-2.54.1/docs/reference/glib/Makefile.in
--- glib-2.54.1.orig/docs/reference/glib/Makefile.in	2017-10-02 11:02:23.000000000 -0400
+++ glib-2.54.1/docs/reference/glib/Makefile.in	2017-10-23 03:04:25.000000000 -0400
@@ -93,10 +93,10 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@ENABLE_MAN_TRUE@am__append_1 = \
-@ENABLE_MAN_TRUE@	glib-gettextize.1 	\
-@ENABLE_MAN_TRUE@	gtester.1		\
-@ENABLE_MAN_TRUE@	gtester-report.1
+am__append_1 = \
+	glib-gettextize.1 	\
+	gtester.1		\
+	gtester-report.1
 
 subdir = docs/reference/glib
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
diff -Nurd -x'*~' glib-2.54.1.orig/docs/reference/gobject/Makefile.in glib-2.54.1/docs/reference/gobject/Makefile.in
--- glib-2.54.1.orig/docs/reference/gobject/Makefile.in	2017-10-02 11:02:23.000000000 -0400
+++ glib-2.54.1/docs/reference/gobject/Makefile.in	2017-10-23 03:04:25.000000000 -0400
@@ -93,10 +93,10 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@ENABLE_MAN_TRUE@am__append_1 = \
-@ENABLE_MAN_TRUE@	glib-mkenums.1 		\
-@ENABLE_MAN_TRUE@	glib-genmarshal.1 	\
-@ENABLE_MAN_TRUE@	gobject-query.1
+am__append_1 = \
+	glib-mkenums.1 		\
+	glib-genmarshal.1 	\
+	gobject-query.1
 
 subdir = docs/reference/gobject
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
diff -Nurd -x'*~' glib-2.54.1.orig/gio/gdbusaddress.c glib-2.54.1/gio/gdbusaddress.c
--- glib-2.54.1.orig/gio/gdbusaddress.c	2017-08-07 10:58:31.000000000 -0400
+++ glib-2.54.1/gio/gdbusaddress.c	2017-10-23 03:04:25.000000000 -0400
@@ -1614,7 +1614,7 @@
       ret = g_strdup (g_getenv ("DBUS_SYSTEM_BUS_ADDRESS"));
       if (ret == NULL)
         {
-          ret = g_strdup ("unix:path=/var/run/dbus/system_bus_socket");
+          ret = g_strdup ("unix:path=@PREFIX@/var/run/dbus/system_bus_socket");
         }
       break;
 
diff -Nurd -x'*~' glib-2.54.1.orig/gio/gdbusprivate.c glib-2.54.1/gio/gdbusprivate.c
--- glib-2.54.1.orig/gio/gdbusprivate.c	2017-07-13 19:03:38.000000000 -0400
+++ glib-2.54.1/gio/gdbusprivate.c	2017-10-23 03:04:25.000000000 -0400
@@ -2098,7 +2098,7 @@
   /* TODO: use PACKAGE_LOCALSTATEDIR ? */
   ret = NULL;
   first_error = NULL;
-  if (!g_file_get_contents ("/var/lib/dbus/machine-id",
+  if (!g_file_get_contents ("@PREFIX@/var/lib/dbus/machine-id",
                             &ret,
                             NULL,
                             &first_error) &&
@@ -2108,7 +2108,7 @@
                             NULL))
     {
       g_propagate_prefixed_error (error, first_error,
-                                  _("Unable to load /var/lib/dbus/machine-id or /etc/machine-id: "));
+                                  _("Unable to load @PREFIX@/var/lib/dbus/machine-id or /etc/machine-id: "));
     }
   else
     {
diff -Nurd -x'*~' glib-2.54.1.orig/gio/gdesktopappinfo.c glib-2.54.1/gio/gdesktopappinfo.c
--- glib-2.54.1.orig/gio/gdesktopappinfo.c	2017-10-02 10:58:06.000000000 -0400
+++ glib-2.54.1/gio/gdesktopappinfo.c	2017-10-23 03:04:25.000000000 -0400
@@ -3492,7 +3492,7 @@
       g_file_set_contents (filename, contents, -1, NULL);
       g_free (contents);
 
-      run_update_command ("update-mime-database", "mime");
+      run_update_command ("@PREFIX@/bin/update-mime-database", "mime");
     }
   g_free (filename);
 
@@ -3650,7 +3650,7 @@
   info->filename = filename;
   info->desktop_id = desktop_id;
 
-  run_update_command ("update-desktop-database", "applications");
+  run_update_command ("@PREFIX@/bin/update-desktop-database", "applications");
 
   /* We just dropped a file in the user's desktop file directory.  Save
    * the monitor the bother of having to notice it and invalidate
diff -Nurd -x'*~' glib-2.54.1.orig/gio/gunixmounts.c glib-2.54.1/gio/gunixmounts.c
--- glib-2.54.1.orig/gio/gunixmounts.c	2017-10-02 10:58:06.000000000 -0400
+++ glib-2.54.1/gio/gunixmounts.c	2017-10-23 03:04:25.000000000 -0400
@@ -246,6 +246,7 @@
     "/tmp",
     "/usr",
     "/usr/X11R6",
+    "/usr/X11",
     "/usr/local",
     "/usr/obj",
     "/usr/ports",
@@ -263,6 +264,7 @@
     "/sbin",
     "/net",
     "/sys",
+    "@PREFIX@",
     NULL
   };
 
diff -Nurd -x'*~' glib-2.54.1.orig/gio/tests/Makefile.in glib-2.54.1/gio/tests/Makefile.in
--- glib-2.54.1.orig/gio/tests/Makefile.in	2017-10-02 11:02:24.000000000 -0400
+++ glib-2.54.1/gio/tests/Makefile.in	2017-10-23 03:04:25.000000000 -0400
@@ -345,8 +345,6 @@
 @HAVE_DBUS_DAEMON_TRUE@@OS_UNIX_TRUE@	gdbus-connection-loss$(EXEEXT) \
 @HAVE_DBUS_DAEMON_TRUE@@OS_UNIX_TRUE@	gdbus-connection-slow$(EXEEXT) \
 @HAVE_DBUS_DAEMON_TRUE@@OS_UNIX_TRUE@	gdbus-error$(EXEEXT) \
-@HAVE_DBUS_DAEMON_TRUE@@OS_UNIX_TRUE@	gdbus-exit-on-close$(EXEEXT) \
-@HAVE_DBUS_DAEMON_TRUE@@OS_UNIX_TRUE@	gdbus-export$(EXEEXT) \
 @HAVE_DBUS_DAEMON_TRUE@@OS_UNIX_TRUE@	gdbus-introspection$(EXEEXT) \
 @HAVE_DBUS_DAEMON_TRUE@@OS_UNIX_TRUE@	gdbus-names$(EXEEXT) \
 @HAVE_DBUS_DAEMON_TRUE@@OS_UNIX_TRUE@	gdbus-peer$(EXEEXT) \
@@ -2191,6 +2189,7 @@
 
 AM_CPPFLAGS = $(gio_INCLUDES) $(GLIB_DEBUG_FLAGS) -I$(top_builddir)/gio -I$(top_srcdir)/gio
 AM_CFLAGS = $(GLIB_WARN_CFLAGS)
+AM_TESTS_ENVIRONMENT += GIO_MODULE_DIR=
 
 # -----------------------------------------------------------------------------
 #  Test programs buildable on all platforms
diff -Nurd -x'*~' glib-2.54.1.orig/gio/tests/desktop-files/home/applications/epiphany-weather-for-toronto-island-9c6a4e022b17686306243dada811d550d25eb1fb.desktop glib-2.54.1/gio/tests/desktop-files/home/applications/epiphany-weather-for-toronto-island-9c6a4e022b17686306243dada811d550d25eb1fb.desktop
--- glib-2.54.1.orig/gio/tests/desktop-files/home/applications/epiphany-weather-for-toronto-island-9c6a4e022b17686306243dada811d550d25eb1fb.desktop	2016-10-22 01:17:54.000000000 -0400
+++ glib-2.54.1/gio/tests/desktop-files/home/applications/epiphany-weather-for-toronto-island-9c6a4e022b17686306243dada811d550d25eb1fb.desktop	2017-10-23 03:04:25.000000000 -0400
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Name=Weather for Toronto Island
-Exec=/bin/true
+Exec=true
 StartupNotify=true
 Terminal=false
 Type=Application
diff -Nurd -x'*~' glib-2.54.1.orig/gio/xdgmime/xdgmime.c glib-2.54.1/gio/xdgmime/xdgmime.c
--- glib-2.54.1.orig/gio/xdgmime/xdgmime.c	2017-07-13 19:03:39.000000000 -0400
+++ glib-2.54.1/gio/xdgmime/xdgmime.c	2017-10-23 03:04:25.000000000 -0400
@@ -255,7 +255,7 @@
 
   xdg_data_dirs = getenv ("XDG_DATA_DIRS");
   if (xdg_data_dirs == NULL)
-    xdg_data_dirs = "/usr/local/share/:/usr/share/";
+    xdg_data_dirs = "@PREFIX@/share:/usr/local/share/:/usr/share/";
 
   ptr = xdg_data_dirs;
 
diff -Nurd -x'*~' glib-2.54.1.orig/glib/gslice.c glib-2.54.1/glib/gslice.c
--- glib-2.54.1.orig/glib/gslice.c	2017-08-07 10:58:31.000000000 -0400
+++ glib-2.54.1/glib/gslice.c	2017-10-23 03:04:25.000000000 -0400
@@ -1406,19 +1406,18 @@
                     gsize memsize)
 {
   gpointer aligned_memory = NULL;
-  gint err = ENOMEM;
 #if     HAVE_COMPLIANT_POSIX_MEMALIGN
+  gint err;
   err = posix_memalign (&aligned_memory, alignment, memsize);
+  if (!aligned_memory)
+    errno = err;
 #elif   HAVE_MEMALIGN
-  errno = 0;
   aligned_memory = memalign (alignment, memsize);
-  err = errno;
 #elif   HAVE_VALLOC
-  errno = 0;
   aligned_memory = valloc (memsize);
-  err = errno;
 #else
   /* simplistic non-freeing page allocator */
+  gint err = ENOMEM;
   mem_assert (alignment == sys_page_size);
   mem_assert (memsize <= sys_page_size);
   if (!compat_valloc_trash)
@@ -1437,9 +1436,9 @@
         }
     }
   aligned_memory = g_trash_stack_pop (&compat_valloc_trash);
-#endif
   if (!aligned_memory)
     errno = err;
+#endif
   return aligned_memory;
 }
 
diff -Nurd -x'*~' glib-2.54.1.orig/glib/gutils.c glib-2.54.1/glib/gutils.c
--- glib-2.54.1.orig/glib/gutils.c	2017-08-15 16:27:00.000000000 -0400
+++ glib-2.54.1/glib/gutils.c	2017-10-23 03:04:25.000000000 -0400
@@ -2093,7 +2093,7 @@
 	}
 #else
       if (!conf_dirs || !conf_dirs[0])
-          conf_dirs = "/etc/xdg";
+          conf_dirs = "@PREFIX@/etc/xdg";
 
       conf_dir_vector = g_strsplit (conf_dirs, G_SEARCHPATH_SEPARATOR_S, 0);
 #endif
diff -Nurd -x'*~' glib-2.54.1.orig/glib/libcharset/Makefile.in glib-2.54.1/glib/libcharset/Makefile.in
--- glib-2.54.1.orig/glib/libcharset/Makefile.in	2017-10-02 11:02:25.000000000 -0400
+++ glib-2.54.1/glib/libcharset/Makefile.in	2017-10-23 03:04:25.000000000 -0400
@@ -776,7 +776,7 @@
 
 @ENABLE_INSTALLED_TESTS_TRUE@installed_test_meta_DATA = $(installed_testcases:=.test)
 AM_CPPFLAGS = \
-	-DLIBDIR=\"$(libdir)\" 		\
+	-DLIBDIR=\"$(sysconfdir)/glib-2.0\" 		\
 	$(config_h_INCLUDES)
 
 libcharset_la_CFLAGS = $(GLIB_HIDDEN_VISIBILITY_CFLAGS)
@@ -785,8 +785,8 @@
 	localcharset.h		\
 	localcharset.c
 
-charset_alias = $(DESTDIR)$(libdir)/charset.alias
-charset_tmp = $(DESTDIR)$(libdir)/charset.tmp
+charset_alias = $(DESTDIR)$(sysconfdir)/glib-2.0/charset.alias
+charset_tmp = $(DESTDIR)$(sysconfdir)/glib-2.0/charset.tmp
 SUFFIXES = .sed .sin
 all: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) all-am
@@ -1509,7 +1509,7 @@
 @ENABLE_INSTALLED_TESTS_TRUE@	echo 'Exec=$(installed_testdir)/$(notdir $<)' >> $@.tmp; \
 @ENABLE_INSTALLED_TESTS_TRUE@	mv $@.tmp $@)
 install-exec-local: all-local
-	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(mkinstalldirs) $(DESTDIR)$(sysconfdir)/glib-2.0
 	if test -f $(charset_alias); then \
 	  sed -f ref-add.sed $(charset_alias) > $(charset_tmp) ; \
 	  $(INSTALL_DATA) $(charset_tmp) $(charset_alias) ; \
diff -Nurd -x'*~' glib-2.54.1.orig/glib/libcharset/charset.alias glib-2.54.1/glib/libcharset/charset.alias
--- glib-2.54.1.orig/glib/libcharset/charset.alias	1969-12-31 19:00:00.000000000 -0500
+++ glib-2.54.1/glib/libcharset/charset.alias	2017-10-23 03:04:25.000000000 -0400
@@ -0,0 +1,111 @@
+# This file contains a table of character encoding aliases,
+# suitable for operating system 'darwin'.
+C			ASCII
+bg_BG			UTF-8
+bg_BG.CP1251		CP1251
+cs_CZ			UTF-8
+cs_CZ.ISO8859-2		ISO-8859-2
+da_DK			UTF-8
+da_DK.ISO8859-1		ISO-8859-1
+da_DK.ISO8859-15	ISO-8859-15
+de_AT			UTF-8
+de_AT.ISO8859-1		ISO-8859-1
+de_AT.ISO8859-15	ISO-8859-15
+de_CH			UTF-8
+de_CH.ISO8859-1		ISO-8859-1
+de_CH.ISO8859-15	ISO-8859-15
+de_DE			UTF-8
+de_DE.ISO8859-1		ISO-8859-1
+de_DE.ISO8859-15	ISO-8859-15
+en_AU			UTF-8
+en_AU.ISO8859-1		ISO-8859-1
+en_AU.ISO8859-15	ISO-8859-15
+en_AU.US-ASCII		ASCII
+en_CA			UTF-8
+en_CA.ISO8859-1		ISO-8859-1
+en_CA.ISO8859-15	ISO-8859-15
+en_CA.US-ASCII		ASCII
+en_GB			UTF-8
+en_GB.ISO8859-1		ISO-8859-1
+en_GB.ISO8859-15	ISO-8859-15
+en_GB.US-ASCII		ASCII
+en_US			UTF-8
+en_US.ISO8859-1		ISO-8859-1
+en_US.ISO8859-15	ISO-8859-15
+en_US.US-ASCII		ASCII
+es_ES			UTF-8
+es_ES.ISO8859-1		ISO-8859-1
+es_ES.ISO8859-15	ISO-8859-15
+fi_FI			UTF-8
+fi_FI.ISO8859-1		ISO-8859-1
+fi_FI.ISO8859-15	ISO-8859-15
+fr_BE			UTF-8
+fr_BE.ISO8859-1		ISO-8859-1
+fr_BE.ISO8859-15	ISO-8859-15
+fr_CA			UTF-8
+fr_CA.ISO8859-1		ISO-8859-1
+fr_CA.ISO8859-15	ISO-8859-15
+fr_CH			UTF-8
+fr_CH.ISO8859-1		ISO-8859-1
+fr_CH.ISO8859-15	ISO-8859-15
+fr_FR			UTF-8
+fr_FR.ISO8859-1		ISO-8859-1
+fr_FR.ISO8859-15	ISO-8859-15
+hr_HR			UTF-8
+hr_HR.ISO8859-2		ISO-8859-2
+hu_HU			UTF-8
+hu_HU.ISO8859-2		ISO-8859-2
+is_IS			UTF-8
+is_IS.ISO8859-1		ISO-8859-1
+is_IS.ISO8859-15	ISO-8859-15
+it_CH			UTF-8
+it_CH.ISO8859-1		ISO-8859-1
+it_CH.ISO8859-15	ISO-8859-15
+it_IT			UTF-8
+it_IT.ISO8859-1		ISO-8859-1
+it_IT.ISO8859-15	ISO-8859-15
+ja_JP			UTF-8
+ja_JP.EUC		EUC-JP
+ja_JP.SJIS		SHIFT_JIS
+ko_KR			UTF-8
+ko_KR.EUC		EUC-KR
+la_LN			UTF-8
+la_LN.ISO8859-1		ISO-8859-1
+la_LN.ISO8859-2		ISO-8859-2
+la_LN.ISO8859-4		ISO-8859-4
+la_LN.ISO8859-15	ISO-8859-15
+la_LN.US-ASCII		ASCII
+lt_LT			UTF-8
+lt_LT.ISO8859-4		ISO-8859-4
+nl_BE			UTF-8
+nl_BE.ISO8869-1		ISO-8859-1
+nl_BE.ISO8869-15	ISO-8859-15
+nl_NL			UTF-8
+nl_NL.ISO8869-1		ISO-8859-1
+nl_NL.ISO8869-15	ISO-8859-15
+no_NO			UTF-8
+no_NO.ISO8869-1		ISO-8859-1
+no_NO.ISO8869-15	ISO-8859-15
+pl_PL			UTF-8
+pl_PL.ISO8859-2		ISO-8859-2
+pt_PT			UTF-8
+pt_PT.ISO8859-1		ISO-8859-1
+pt_PT.ISO8859-15	ISO-8859-15
+ru_RU			UTF-8
+ru_RU.CP866		CP866
+ru_RU.ISO8859-5		ISO-8859-5
+ru_RU.KOI8-R		KOI8-R
+ru_RU.cp1251		CP1251
+ru_RU.koi8r		KOI8-R
+sl_SI			UTF-8
+sl_SI.ISO8859-2		ISO-8859-2
+sv_SE			UTF-8
+sv_SE.ISO8859-1		ISO-8859-1
+sv_SE.ISO8859-15	ISO-8859-15
+uk_UA			UTF-8
+uk_UA.KOI8-U		KOI8-U
+uk_UA.cp1251		CP1251
+zh_CN			UTF-8
+zh_CN.EUC		GB2312
+zh_TW			UTF-8
+zh_TW.Big5		BIG5
diff -Nurd -x'*~' glib-2.54.1.orig/glib/tests/gdatetime.c glib-2.54.1/glib/tests/gdatetime.c
--- glib-2.54.1.orig/glib/tests/gdatetime.c	2017-10-02 10:58:06.000000000 -0400
+++ glib-2.54.1/glib/tests/gdatetime.c	2017-10-23 03:04:25.000000000 -0400
@@ -1125,14 +1125,14 @@
    */
   TEST_PRINTF ("%a", "\345\234\237");
   TEST_PRINTF ("%A", "\345\234\237\346\233\234\346\227\245");
-#ifndef HAVE_CARBON /* OSX just returns the number */
+#ifndef __APPLE__ /* OSX just returns the number */
   TEST_PRINTF ("%b", "10\346\234\210");
 #endif
   TEST_PRINTF ("%B", "10\346\234\210");
   TEST_PRINTF ("%d", "24");
   TEST_PRINTF_DATE (2009, 1, 1, "%d", "01");
   TEST_PRINTF ("%e", "24"); // fixme
-#ifndef HAVE_CARBON /* OSX just returns the number */
+#ifndef __APPLE__ /* OSX just returns the number */
   TEST_PRINTF ("%h", "10\346\234\210");
 #endif
   TEST_PRINTF ("%H", "00");
@@ -1149,7 +1149,7 @@
   TEST_PRINTF_TIME (10, 13, 13, "%l", "10");
   TEST_PRINTF ("%m", "10");
   TEST_PRINTF ("%M", "00");
-#ifndef HAVE_CARBON /* OSX returns latin "AM", not japanese */
+#ifndef __APPLE__ /* OSX returns latin "AM", not japanese */
   TEST_PRINTF ("%p", "\345\215\210\345\211\215");
   TEST_PRINTF_TIME (13, 13, 13, "%p", "\345\215\210\345\276\214");
   TEST_PRINTF ("%P", "\345\215\210\345\211\215");
@@ -1162,7 +1162,7 @@
   TEST_PRINTF ("%S", "00");
   TEST_PRINTF ("%t", "	");
   TEST_PRINTF ("%u", "6");
-#ifndef HAVE_CARBON /* OSX returns YYYY/MM/DD in ASCII */
+#ifndef __APPLE__ /* OSX returns YYYY/MM/DD in ASCII */
   TEST_PRINTF ("%x", "2009\345\271\26410\346\234\21024\346\227\245");
 #endif
   TEST_PRINTF ("%X", "00\346\231\20200\345\210\20600\347\247\222");
diff -Nurd -x'*~' glib-2.54.1.orig/tests/run-assert-msg-test.sh glib-2.54.1/tests/run-assert-msg-test.sh
--- glib-2.54.1.orig/tests/run-assert-msg-test.sh	2016-10-22 01:17:10.000000000 -0400
+++ glib-2.54.1/tests/run-assert-msg-test.sh	2017-10-23 03:04:25.000000000 -0400
@@ -27,6 +27,11 @@
   fi
 fi
 
+if [ $UID -ne 0 ]; then
+  echo_v "Skipped (cannot reliably run gdb as non-root)"
+  exit 0
+fi
+
 echo_v "Running assert-msg-test"
 OUT=$(./assert-msg-test 2>&1) && fail "assert-msg-test should abort"
 echo "$OUT" | grep -q '^GLib:ERROR:.*assert-msg-test.c:.*:.*main.*: assertion failed: (42 < 0)' || \
@@ -38,7 +43,7 @@
 fi
 
 echo_v "Running gdb on assert-msg-test"
-OUT=$($LIBTOOL --mode=execute gdb --batch -x ${srcdir:-.}/assert-msg-test.gdb ./assert-msg-test 2> $error_out) || fail "failed to run gdb"
+OUT=$($LIBTOOL --mode=execute /usr/bin/gdb --batch -x ${srcdir:-.}/assert-msg-test.gdb ./assert-msg-test 2> $error_out) || fail "failed to run gdb"
 
 echo_v "Checking if assert message is in __glib_assert_msg"
 if ! echo "$OUT" | grep -q '^$1.*"GLib:ERROR:.*assert-msg-test.c:.*:.*main.*: assertion failed: (42 < 0)"'; then
