diff -Nurd -x'*~' gnome-menus-2.10.2.orig/ltmain.sh gnome-menus-2.10.2/ltmain.sh
--- gnome-menus-2.10.2.orig/ltmain.sh	2005-06-28 03:51:53.000000000 -0400
+++ gnome-menus-2.10.2/ltmain.sh	2005-08-04 18:05:32.000000000 -0400
@@ -3887,6 +3887,11 @@
 	  eval test_cmds=\"$archive_expsym_cmds\"
 	  cmds=$archive_expsym_cmds
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+          else
+            tmp_verstring="$verstring"
+          fi
 	  eval test_cmds=\"$archive_cmds\"
 	  cmds=$archive_cmds
 	  fi
