diff -Nurd -x'*~' gnome-apt-0.4.7.orig/src/gadefs.h gnome-apt-0.4.7/src/gadefs.h
--- gnome-apt-0.4.7.orig/src/gadefs.h	2003-10-16 12:57:26.000000000 -0400
+++ gnome-apt-0.4.7/src/gadefs.h	2007-02-14 14:37:38.000000000 -0500
@@ -28,6 +28,7 @@
 
 #include "config.h"
 
+#include <libintl.h>
 #ifndef _
 #if ENABLE_NLS
 #define _(x) gettext(x)
diff -Nurd -x'*~' gnome-apt-0.4.7.orig/src/gdeb/Makefile.in gnome-apt-0.4.7/src/gdeb/Makefile.in
--- gnome-apt-0.4.7.orig/src/gdeb/Makefile.in	2004-07-11 12:53:14.000000000 -0400
+++ gnome-apt-0.4.7/src/gdeb/Makefile.in	2007-02-14 13:52:33.000000000 -0500
@@ -194,7 +194,7 @@
 
 EXTRA_DIST = $(appreg_DATA) $(mimeinfo_DATA)
 
-gdeb_LDADD = $(GKSU_LIBS) -lapt-inst
+gdeb_LDADD = $(GKSU_LIBS) -lapt-inst -lapt-pkg
 subdir = src/gdeb
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
diff -Nurd -x'*~' gnome-apt-0.4.7.orig/src/gdeb/main.cc gnome-apt-0.4.7/src/gdeb/main.cc
--- gnome-apt-0.4.7.orig/src/gdeb/main.cc	2004-07-11 12:53:14.000000000 -0400
+++ gnome-apt-0.4.7/src/gdeb/main.cc	2007-02-14 13:49:22.000000000 -0500
@@ -32,6 +32,8 @@
 #include <gksuui.h>
 #include <gksu.h>
 #include <sys/stat.h>
+#include <libintl.h>
+#include <libgen.h>
 
 #include "gdeb.h"
 
