diff -Nurd -x'*~' libxml2-2.6.20.orig/python/setup.py libxml2-2.6.20/python/setup.py
--- libxml2-2.6.20.orig/python/setup.py	2005-07-10 18:53:45.000000000 -0400
+++ libxml2-2.6.20/python/setup.py	2005-08-25 12:19:56.000000000 -0400
@@ -231,7 +231,7 @@
        author = "Daniel Veillard",
        author_email = "veillard@redhat.com",
        url = "http://xmlsoft.org/python.html",
-       licence="MIT Licence",
+       license="MIT Licence",
        py_modules=modules,
        ext_modules=extens,
        ext_package=ext_package,
diff -Nurd -x'*~' libxml2-2.6.20.orig/python/tests/Makefile libxml2-2.6.20/python/tests/Makefile
--- libxml2-2.6.20.orig/python/tests/Makefile	2005-08-25 12:18:59.000000000 -0400
+++ libxml2-2.6.20/python/tests/Makefile	2005-08-25 12:19:56.000000000 -0400
@@ -53,20 +53,22 @@
 
 EXTRA_DIST = $(PYTESTS) $(XMLS)
 
-if WITH_PYTHON
 tests: $(PYTESTS)
 	@echo "## running Python regression tests"
-	-@(PYTHONPATH="..:../.libs:$(srcdir)/..:$$PYTHONPATH" ; \
-	   export PYTHONPATH; \
-	   LD_LIBRARY_PATH="$(top_builddir)/.libs:$$LD_LIBRARY_PATH" ; \
-	   export LD_LIBRARY_PATH; \
+	@rm -f test.out
+	-@( ; \
 	   for test in $(PYTESTS) ; \
-	   do log=`$(PYTHON) $(srcdir)/$$test` ; \
-	   if [ "`echo $$log | grep OK`" = "" ] ; then \
-	   echo "-- $$test" ; echo "$$log" ; fi ; done)
-else
-tests:
-endif
+	   do log=`$(PYTHON) ./$$test 2>&1` ; \
+	      if [ "`echo $$log | grep OK`" = "" ] ; then \
+	      	 echo "-- FAILED $$test" ; \
+	   	 echo "$$log" ; \
+		 echo "  $$test" >> test.out ; \
+	      else \
+	      	 echo "-- OK $$test" ; \
+	      fi ; \
+	   done; \
+	)
+	@[ ! -s test.out ]
 
 clean:
 	rm -f *.pyc core
