diff -Nurd -x'*~' pygobject-2.16.0.orig/Makefile.am pygobject-2.16.0/Makefile.am
--- pygobject-2.16.0.orig/Makefile.am	2008-12-21 11:30:46.000000000 -0500
+++ pygobject-2.16.0/Makefile.am	2009-01-06 03:13:51.000000000 -0500
@@ -19,7 +19,7 @@
 	m4/jhflags.m4			\
 	m4/python.m4
 
-INCLUDES = $(PYTHON_INCLUDES) $(GLIB_CFLAGS) -I$(top_srcdir)/gobject
+INCLUDES = -I$(top_srcdir)/gobject $(PYTHON_INCLUDES) $(GLIB_CFLAGS)
 
 # pkg-config files
 pkgconfigdir = $(libdir)/pkgconfig
@@ -37,9 +37,7 @@
 
 # linker flags
 common_ldflags = -module -avoid-version
-if PLATFORM_WIN32
 common_ldflags += -no-undefined
-endif
 
 # install pth file.
 install-data-local:
diff -Nurd -x'*~' pygobject-2.16.0.orig/configure.ac pygobject-2.16.0/configure.ac
--- pygobject-2.16.0.orig/configure.ac	2009-01-04 15:17:33.000000000 -0500
+++ pygobject-2.16.0/configure.ac	2009-01-06 04:09:21.000000000 -0500
@@ -67,6 +67,8 @@
 
 AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
+dnl when using libtool 2.x create libtool early, because it's used in configure
+m4_ifdef([LT_OUTPUT], [LT_OUTPUT]) 
 AM_PROG_CC_C_O
 
 JD_PATH_PYTHON(2.3.5)
@@ -96,6 +98,9 @@
 PLATFORM=`$PYTHON -c "import sys; from distutils import util; sys.stdout.write(util.get_platform())"`
 AC_SUBST(PLATFORM)
 
+PYTHON_LIBS="-L${py_prefix}/lib/python${PYTHON_VERSION}/config -lpython${PYTHON_VERSION}"
+AC_SUBST(PYTHON_LIBS)
+
 AC_ARG_ENABLE(thread,
   AC_HELP_STRING([--disable-thread], [Disable pygobject threading support]),,
   enable_thread=yes)
@@ -167,11 +172,13 @@
 fi
 if test x"$have_libffi" = xyes; then
   AC_DEFINE(HAVE_FFI_H,1,[Have libffi include files])
+  LIBFFI_PC=libffi
 fi
 AC_MSG_RESULT([$have_libffi])
 AM_CONDITIONAL(HAVE_LIBFFI, test "$have_libffi" = "yes")
 AC_SUBST(FFI_CFLAGS)
 AC_SUBST(FFI_LIBS)
+AC_SUBST(LIBFFI_PC)
 
 dnl gio
 PKG_CHECK_MODULES(GIO, gio-2.0 >= gio_required_version,
diff -Nurd -x'*~' pygobject-2.16.0.orig/gio/Makefile.am pygobject-2.16.0/gio/Makefile.am
--- pygobject-2.16.0.orig/gio/Makefile.am	2008-12-21 11:30:46.000000000 -0500
+++ pygobject-2.16.0/gio/Makefile.am	2009-01-06 04:31:02.000000000 -0500
@@ -3,11 +3,11 @@
 PLATFORM_VERSION = 2.0
 
 INCLUDES = 				\
+	-I$(top_srcdir)/glib		\
+	-I$(top_srcdir)/gobject		\
 	$(PYTHON_INCLUDES) 		\
 	$(PYGOBJECT_CFLAGS)		\
-	$(GIO_CFLAGS)			\
-	-I$(top_srcdir)/glib		\
-	-I$(top_srcdir)/gobject
+	$(GIO_CFLAGS)
 
 # defs files
 defsdir = $(pkgdatadir)/$(PLATFORM_VERSION)/defs
@@ -26,9 +26,7 @@
 
 # linker flags
 common_ldflags = -module -avoid-version
-if PLATFORM_WIN32
 common_ldflags += -no-undefined
-endif
 
 CLEANFILES =
 EXTRA_DIST =
@@ -72,7 +70,7 @@
 giounix.c: $(GIOUNIX_DEFS) unix.override
 unix_la_CFLAGS = $(GIOUNIX_CFLAGS)
 unix_la_LDFLAGS = $(common_ldflags) -export-symbols-regex initunix
-unix_la_LIBADD = $(GIOUNIX_LIBS)
+unix_la_LIBADD = $(GIOUNIX_LIBS) $(PYTHON_LIBS)
 unix_la_SOURCES = unixmodule.c
 nodist_unix_la_SOURCES = unix.c
 if BUILD_GIOUNIX
diff -Nurd -x'*~' pygobject-2.16.0.orig/glib/Makefile.am pygobject-2.16.0/glib/Makefile.am
--- pygobject-2.16.0.orig/glib/Makefile.am	2008-12-21 11:30:46.000000000 -0500
+++ pygobject-2.16.0/glib/Makefile.am	2009-01-06 04:15:42.000000000 -0500
@@ -15,12 +15,10 @@
 pyglib_LTLIBRARIES = _glib.la
 
 common_ldflags = -module -avoid-version
-if PLATFORM_WIN32
 common_ldflags += -no-undefined
-endif
 
 libpyglib_2_0_la_CFLAGS = $(GLIB_CFLAGS)
-libpyglib_2_0_la_LIBADD = $(GLIB_LIBS) $(FFI_LIBS)
+libpyglib_2_0_la_LIBADD = $(GLIB_LIBS) $(FFI_LIBS) $(PYTHON_LIBS)
 libpyglib_2_0_la_SOURCES = 	\
 	pyglib.c		\
 	pyglib.h		\
diff -Nurd -x'*~' pygobject-2.16.0.orig/gobject/Makefile.am pygobject-2.16.0/gobject/Makefile.am
--- pygobject-2.16.0.orig/gobject/Makefile.am	2008-12-21 11:30:46.000000000 -0500
+++ pygobject-2.16.0/gobject/Makefile.am	2009-01-06 03:14:25.000000000 -0500
@@ -15,9 +15,7 @@
 nodist_pygobject_PYTHON = constants.py
 
 common_ldflags = -module -avoid-version
-if PLATFORM_WIN32
 common_ldflags += -no-undefined
-endif
 
 constants.py: generate-constants$(EXEEXT) constants.py.in
 	rm -f constants.py
@@ -33,11 +31,11 @@
 EXTRA_DIST = constants.py.in
 
 _gobject_la_CFLAGS = \
+	-I$(top_srcdir)/glib \
 	$(PYTHON_INCLUDES) \
 	$(FFI_CFLAGS) \
 	$(GLIB_CFLAGS) \
-	 -DPY_SSIZE_T_CLEAN \
-	-I$(top_srcdir)/glib
+	 -DPY_SSIZE_T_CLEAN
 _gobject_la_LDFLAGS = $(common_ldflags) -export-symbols-regex "_gobject|PyInit__gobject"
 _gobject_la_LIBADD = \
 	$(GLIB_LIBS) \
diff -Nurd -x'*~' pygobject-2.16.0.orig/pygobject-2.0-uninstalled.pc.in pygobject-2.16.0/pygobject-2.0-uninstalled.pc.in
--- pygobject-2.16.0.orig/pygobject-2.0-uninstalled.pc.in	2008-12-21 11:30:46.000000000 -0500
+++ pygobject-2.16.0/pygobject-2.0-uninstalled.pc.in	2009-01-06 03:09:48.000000000 -0500
@@ -1,8 +1,7 @@
 Name: PyGObject
 Description: Python bindings for GObject
-Requires: gobject-2.0
+Requires: gobject-2.0 @LIBFFI_PC@
 Version: @VERSION@
-Libs: @FFI_LIBS@
 Cflags: -I${pc_top_builddir}/${pcfiledir}/gobject
 
 # you can use the --variable=pygtkincludedir argument to
diff -Nurd -x'*~' pygobject-2.16.0.orig/pygobject-2.0.pc.in pygobject-2.16.0/pygobject-2.0.pc.in
--- pygobject-2.16.0.orig/pygobject-2.0.pc.in	2008-12-21 11:30:46.000000000 -0500
+++ pygobject-2.16.0/pygobject-2.0.pc.in	2009-01-06 03:09:48.000000000 -0500
@@ -16,7 +16,6 @@
 
 Name: PyGObject
 Description: Python bindings for GObject
-Requires: gobject-2.0
+Requires: gobject-2.0 @LIBFFI_PC@
 Version: @VERSION@
-Libs: -L${libdir} @FFI_LIBS@
 Cflags: -I${pygtkincludedir}
diff -Nurd -x'*~' pygobject-2.16.0.orig/tests/Makefile.am pygobject-2.16.0/tests/Makefile.am
--- pygobject-2.16.0.orig/tests/Makefile.am	2008-12-21 11:30:45.000000000 -0500
+++ pygobject-2.16.0/tests/Makefile.am	2009-01-06 04:33:21.000000000 -0500
@@ -1,7 +1,7 @@
 INCLUDES = \
+	-I$(top_srcdir)/gobject \
 	$(PYTHON_INCLUDES) \
-	$(GLIB_CFLAGS) \
-	-I$(top_srcdir)/gobject
+	$(GLIB_CFLAGS)
 
 EXTRA_DIST = \
 	$(tests) \
@@ -15,7 +15,7 @@
 linked_LIBS = testhelper.la
 
 testhelper_la_LDFLAGS = -module -avoid-version
-testhelper_la_LIBADD = $(GLIB_LIBS)
+testhelper_la_LIBADD = $(GLIB_LIBS) $(PYTHON_LIBS)
 testhelper_la_SOURCES = \
 	testhelpermodule.c \
 	test-thread.c \
