
diff -aBdruw Eterm-0.9/doc/Makefile.am Eterm-0.9-patched/doc/Makefile.am
--- Eterm-0.9/doc/Makefile.am	Tue Nov 16 01:55:07 1999
+++ Eterm-0.9-patched/doc/Makefile.am	Fri Jul 20 23:12:54 2001
@@ -2,13 +2,10 @@
 
 man_MANS = Eterm.1
 
-all: Makefile Eterm.1.html
+all: Makefile
 
 Eterm.1: $(top_srcdir)/doc/Eterm.1.in
 	$(SED) -e 's%@''VERSION''@%$(VERSION)%g' -e 's%@''DATE''@%$(DATE)%g' -e 's%@THEMEDIR@%$(pkgdatadir)/themes%g' $(top_srcdir)/doc/$@.in > $@
-
-Eterm.1.html: Eterm.1
-	-nroff -man $< | man2html -title Eterm > $@
 
 EXTRA_DIST = Eterm.1.in Eterm_reference.html Eterm.tcap Eterm.ti Makefile.am Makefile.in
 
--- Eterm-0.9/libmej/debug.c	Wed Jan  5 18:34:23 2000
+++ Eterm-0.9-patched/libmej/debug.c	Fri Jul 20 22:38:36 2001
@@ -35,7 +35,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #ifndef WITH_DMALLOC
-# include <malloc.h>
+# include <sys/malloc.h>
 #endif
 #include <stdarg.h>
 #include <string.h>
diff -aBdruw Eterm-0.9/libmej/strings.c Eterm-0.9-patched/libmej/strings.c
--- Eterm-0.9/libmej/strings.c	Wed Jan  5 18:34:23 2000
+++ Eterm-0.9-patched/libmej/strings.c	Fri Jul 20 22:39:17 2001
@@ -38,7 +38,7 @@
 #include <unistd.h>
 #include <errno.h>
 #ifndef WITH_DMALLOC
-# include <malloc.h>
+# include <sys/malloc.h>
 #endif
 #include <ctype.h>
 #include "debug.h"
diff -aBdruw Eterm-0.9/src/utmp.c Eterm-0.9-patched/src/utmp.c
--- Eterm-0.9/src/utmp.c	Wed Jan  5 18:34:31 2000
+++ Eterm-0.9-patched/src/utmp.c	Fri Jul 20 23:15:35 2001
@@ -68,7 +68,7 @@
 # ifdef HAVE_LASTLOG_H
 #  include <lastlog.h>
 # endif
-# if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__bsdi__)
+# if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__bsdi__) || defined(DARWIN)
 #  include <ttyent.h>
 # endif
 
@@ -85,7 +85,7 @@
 
 /* don't go off end of ut_id & remember if an entry has been made */
 # ifndef HAVE_UTEMPTER
-#  if defined(USE_SYSV_UTMP) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__bsdi__)
+#  if defined(USE_SYSV_UTMP) || defined(__FreeBSD__) || defined(DARWIN) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__bsdi__)
 static char ut_id[5];		/* remember if entry to utmp made */
 #  else
 static int utmp_pos;		/* BSD position of utmp-stamp */
@@ -277,7 +277,7 @@
 # else /* USE_SYSV_UTMP */
 /* BSD utmp support */
 
-#  if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__bsdi__)
+#  if defined(__FreeBSD__) || defined(DARWIN) || defined(__NetBSD__) || defined(__bsdi__)
 
 /* used to hold the line we are using */
 static char ut_line[32];
@@ -325,7 +325,7 @@
   }
 }
 
-#  else /* __FreeBSD__ || NetBSD || BSDI */
+#  else /* __FreeBSD__ || DARWIN || NetBSD || BSDI */
 static int utmp_pos = 0;	/* position of utmp-stamp */
 
 /*----------------------------------------------------------------------*
@@ -383,7 +383,7 @@
   return rval;
 }
 
-#  endif /* __FreeBSD__ || NetBSD || BSDI */
+#  endif /* __FreeBSD__ || DARWIN || NetBSD || BSDI */
 
 void
 add_utmp_entry(const char *pty, const char *hostname, int fd)
@@ -403,7 +403,7 @@
     return;
   }
 
-#  if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__bsdi__)
+#  if defined(__FreeBSD__) || defined(DARWIN) || defined(__NetBSD__) || defined(__bsdi__)
   strncpy(ut_line, pty, 31);
 
   strncpy(utmp.ut_line, pty, UT_LINESIZE);
@@ -412,7 +412,7 @@
   utmp.ut_time = time(NULL);
 
   b_login(&utmp);
-#  else /* __FreeBSD__ || NetBSD || BSDI */
+#  else /* __FreeBSD__ DARWIN || || NetBSD || BSDI */
   strncpy(utmp.ut_line, ut_id, sizeof(utmp.ut_line));
   strncpy(utmp.ut_name, pwent->pw_name, sizeof(utmp.ut_name));
   strncpy(utmp.ut_host, hostname, sizeof(utmp.ut_host));
@@ -431,7 +431,7 @@
 void
 remove_utmp_entry(void)
 {
-#  if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__bsdi__)
+#  if defined(__FreeBSD__) || defined(DARWIN) || defined(__NetBSD__) || defined(__bsdi__)
   logout(ut_line);
   logwtmp(ut_line, "", "");
 #  else /* __FreeBSD__ */
@@ -448,7 +448,7 @@
     fclose(fd);
   }
   privileges(REVERT);
-#  endif /* __FreeBSD__ || NetBSD || BSDI */
+#  endif /* __FreeBSD__ || DARWIN || NetBSD || BSDI */
 }
 
 # endif /* USE_SYSV_UTMP */
