diff -ruN src.orig/Makefile.in src/Makefile.in
--- src.orig/Makefile.in	Sat May  3 13:54:53 2003
+++ src/Makefile.in	Sat May  3 13:54:59 2003
@@ -27,9 +27,7 @@
 install: wmweather wmweather.1
 	$(INSTALL) -D -m 0755 wmweather $(DESTDIR)$(bindir)/wmweather
 	$(INSTALL) -D -m 0644 wmweather.1 $(DESTDIR)$(mandir)/man1/wmweather.1
-	test -e $(DESTDIR)$(sysconfig) || $(INSTALL) -D -m 0644 wmweather.conf $(DESTDIR)$(sysconfig)
-	$(LN_S) wmweather $(DESTDIR)$(bindir)/wmWeather
-	$(LN_S) wmweather.1 $(DESTDIR)$(mandir)/man1/wmWeather.1
+	test -e $(prefix)$(sysconfig) || $(INSTALL) -D -m 0644 wmweather.conf $(prefix)$(sysconfig)
 
 uninstall:
 	rm -f $(DESTDIR)$(bindir)/wmweather $(DESTDIR)$(bindir)/wmWeather
