--- Xbae-4.9.5/src/Makefile.in	2002-07-04 19:01:25.000000000 +1000
+++ Xbae-4.9.5-patched/src/Makefile.in	2002-07-10 16:33:51.000000000 +1000
@@ -69,7 +69,7 @@
 OBJEXT = @OBJEXT@
 PATH_SEPARATOR = @PATH_SEPARATOR@
 
-mandir = $(prefix)/man
+mandir = $(prefix)/share/man
 ACLOCALDIR = @ACLOCALDIR@
 AMTAR = @AMTAR@
 AS = @AS@

--- Xbae-4.9.5/Makefile.in	2002-07-04 19:01:11.000000000 +1000
+++ Xbae-4.9.5-patched/Makefile.in	2002-07-10 19:20:57.000000000 +1000
@@ -115,7 +115,7 @@
 x_libraries = @x_libraries@
 
 MAINTAINERCLEANFILES = Makefile.in configure libtool config.guess config.sub ltconfig ltmain.sh aclocal.m4 config.h.in install-sh missing mkinstalldirs
-SUBDIRS = include src examples doc scripts
+SUBDIRS = include src
 
 EXTRA_DIST = configure ac_find_xbae.m4 Imakefile Xbae.tmpl Xbae.rules \
 	CVSMake acconfig.h
@@ -125,8 +125,8 @@
 
 @Aclocal_TRUE@aclocal_DATA = ac_find_xbae.m4
 
-docdir = $(prefix)/Xbae
-doc_DATA = NEWS README INSTALL
+docdir = $(prefix)/share/doc/xbae
+doc_DATA = AUTHORS ChangeLog COPYING NEWS README
 
 #
 # Some convenient targets for users of Exuberant ctags
