diff -ru algae-4.3.6-orig/Makefile.in algae-4.3.6/Makefile.in
--- algae-4.3.6-orig/Makefile.in	2004-06-02 09:20:23.000000000 -0700
+++ algae-4.3.6/Makefile.in	2004-06-02 09:21:05.000000000 -0700
@@ -165,6 +165,12 @@
 	cd lapack; $(MAKE)
 .PHONY: xlapack
 
+vblas:
+.PHONY: vblas
+
+vlapack:
+.PHONY: vlapack
+
 check: source
 times: source
 
diff -ru algae-4.3.6-orig/configure algae-4.3.6/configure
--- algae-4.3.6-orig/configure	2004-06-02 09:20:23.000000000 -0700
+++ algae-4.3.6/configure	2004-06-02 09:21:06.000000000 -0700
@@ -3462,7 +3462,7 @@
 echo "${ECHO_T}no" >&6
 fi
 
-for ac_prog in mozilla netscape mosaic
+for ac_prog in mozilla netscape mosaic open
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
@@ -5279,11 +5279,12 @@
   case "${withval}" in
      y|yes ) ;;
      n|no ) BLAS=xblas ;;
+     veclib ) BLAS=vblas ;;
      * ) LDFLAGS="$LDFLAGS -L${withval}" ;;
   esac
 fi;
 case $BLAS in
-  xblas ) ;;
+  xblas|vblas ) ;;
      *  )
 as_ac_Lib=`echo "ac_cv_lib_blas_$blas_func" | $as_tr_sh`
 echo "$as_me:$LINENO: checking for $blas_func in -lblas" >&5
@@ -5494,7 +5495,7 @@
 # checks (except for LAPACK) don't need it and won't find it.
 
 if test "$BLAS" = xblas; then
-  LDFLAGS="$LDFLAGS -L../blas"
+  LDFLAGS="-L../blas $LDFLAGS"
   LOCAL_LIBS="-lblas $LOCAL_LIBS"
 fi
 
@@ -5511,11 +5512,12 @@
   case "${withval}" in
      y|yes ) ;;
      n|no ) LAPACK=xlapack ;;
+     veclib ) LAPACK=vlapack ;;
      * ) LDFLAGS="$LDFLAGS -L${withval}" ;;
   esac
 fi;
 case $LAPACK in
-  xlapack ) ;;
+  xlapack|vlapack ) ;;
   *       )
 as_ac_Lib=`echo "ac_cv_lib_lapack_$lapack_func" | $as_tr_sh`
 echo "$as_me:$LINENO: checking for $lapack_func in -llapack" >&5
@@ -5590,7 +5592,7 @@
 # checks don't need it and won't find it.
 
 if test "$LAPACK" = xlapack; then
-  LDFLAGS="$LDFLAGS -L../lapack"
+  LDFLAGS="-L../lapack $LDFLAGS"
   LOCAL_LIBS="-llapack $LOCAL_LIBS"
 fi
 
diff -ru algae-4.3.6-orig/doc/Makefile.in algae-4.3.6/doc/Makefile.in
--- algae-4.3.6-orig/doc/Makefile.in	2004-06-02 09:20:22.000000000 -0700
+++ algae-4.3.6/doc/Makefile.in	2004-06-02 09:21:06.000000000 -0700
@@ -174,13 +174,10 @@
 	-if test -f algae.info; then d=.; else d=$(srcdir); fi; \
 	for f in $$d/algae.info $$d/algae.info-*; do \
 	  $(INSTALL_DATA) $$f $(infodir); \
-	done; \
-	if $(SHELL) -c 'install-info --version' > /dev/null 2>&1; then \
-	  install-info --info-dir=$(infodir) $$d/algae.info; else true; fi
+	done;
 	-if test -f algae_toc.html; then d=.; else d=$(srcdir); fi; \
 	for f in $$d/index.html $$d/algae*.html $$d/icon.gif $$d/info.db; do \
 	  $(INSTALL_DATA) $$f $(htmldir); done
-	-rm -f $(datadir)/algae/html; ln -s $(htmldir) $(datadir)/algae/html
 	-$(INSTALL_DATA) algae.1 $(mandir)/algae.$(manext)
 
 uninstall:
diff -ru algae-4.3.6-orig/fpe/darwin.c algae-4.3.6/fpe/darwin.c
--- algae-4.3.6-orig/fpe/darwin.c	2004-06-02 09:20:22.000000000 -0700
+++ algae-4.3.6/fpe/darwin.c	2004-06-02 09:21:06.000000000 -0700
@@ -39,15 +39,15 @@
 	/* First enable the right FP exception conditions */
 	count = PPC_FLOAT_STATE_COUNT;
 	thread_get_state(victim, PPC_FLOAT_STATE, (thread_state_t)&fs, &count);
-	/* Enable VE OE ZE, Disable UE XE */
-	fs.fpscr = (fs.fpscr & ~0x1FFFFF28) | 0x0D0;
+	/* Enable OE ZE, Disable VE UE XE */
+	fs.fpscr = (fs.fpscr & ~0x1FFFFF28) | 0x050;
 	thread_set_state(victim, PPC_FLOAT_STATE, (thread_state_t)&fs, count);
 
 	/* Now, enable FP exceptions as such */
 	count = PPC_THREAD_STATE_COUNT;
 	thread_get_state(victim, PPC_THREAD_STATE, (thread_state_t)&ts, &count);
-	/* Set FE0 = FE1 = 1 */
-	ts.srr1 |= 0x0900;
+	/* Set FE0 = 1 FE1 = 0 */
+	ts.srr1 |= 0x0800;
 	thread_set_state(victim, PPC_THREAD_STATE, (thread_state_t)&ts, count);
 	
 	return 0;
diff -ru algae-4.3.6-orig/src/DEPENDS algae-4.3.6/src/DEPENDS
--- algae-4.3.6-orig/src/DEPENDS	2004-06-02 09:20:21.000000000 -0700
+++ algae-4.3.6/src/DEPENDS	2004-06-03 00:02:03.000000000 -0700
@@ -102,7 +102,7 @@
 unop.o : datum.h not.h matrix_st.h test.h scalar_st.h transpose.h scalar.h sigint.h exception.h vector.h negate.h fortran.h message.h function.h put.h unop.h table_st.h entity.h matrix.h vector_st.h mem.h algae.h ../config.h ptypes.h table.h ok.h
 vector.o : file_io.h get.h matrix_st.h test.h scalar_st.h scalar.h sigint.h exception.h binop.h vector.h fortran.h message.h put.h dense.h table_st.h entity.h matrix.h vector_st.h mem.h algae.h ../config.h table.h ok.h
 zero.o : matrix_st.h scalar_st.h scalar.h sigint.h exception.h cast.h vector.h fortran.h message.h put.h entity.h partition.h two_type.h matrix.h vector_st.h mem.h algae.h ../config.h ok.h
-dlmalloc.o : sigint.h exception.h fortran.h message.h mem.h /usr/include/malloc.h algae.h ../config.h
+dlmalloc.o : sigint.h exception.h fortran.h message.h mem.h algae.h ../config.h
 assign.o : matrix_st.h full.h scalar_st.h transpose.h scalar.h sigint.h exception.h binop.h cast.h vector.h fortran.h message.h put.h dense.h sort.h sparse.h fill.h table_st.h entity.h two_type.h partition.h matrix.h vector_st.h mem.h assign.h algae.h ../config.h find.h table.h ok.h
 parse.o : code.h datum.h psr.h sigint.h exception.h fortran.h for_loop.h message.h function.h put.h error.h pmem.h entity.h vector_st.h mem.h algae.h ../config.h ptypes.h ok.h
 scan.o : code.h datum.h psr.h print.h sigint.h exception.h fortran.h message.h function.h put.h error.h pmem.h parse.h entity.h printf.h mem.h algae.h ../config.h ptypes.h ok.h
diff -ru algae-4.3.6-orig/src/Makefile.in algae-4.3.6/src/Makefile.in
--- algae-4.3.6-orig/src/Makefile.in	2004-06-02 09:20:21.000000000 -0700
+++ algae-4.3.6/src/Makefile.in	2004-06-02 09:21:59.000000000 -0700
@@ -220,7 +220,7 @@
 install: all
 	$(INSTALL_PROGRAM) algae $(bindir)/algae-$(VERSION_NUMBER)
 	rm -f $(bindir)/algae
-	ln -s $(bindir)/algae-$(VERSION_NUMBER) $(bindir)/algae
+	ln -s algae-$(VERSION_NUMBER) $(bindir)/algae
 .PHONY:	install
 
 uninstall:
diff -ru algae-4.3.6-orig/superlu/supermatrix.h algae-4.3.6/superlu/supermatrix.h
--- algae-4.3.6-orig/superlu/supermatrix.h	2004-06-02 09:20:23.000000000 -0700
+++ algae-4.3.6/superlu/supermatrix.h	2004-06-02 09:21:06.000000000 -0700
@@ -15,6 +15,11 @@
     SLU_DN     /* Fortran style column-wise storage for dense matrix */
 } Stype_t;
 
+#undef _S
+#undef _D
+#undef _C
+#undef _Z
+
 typedef enum {
     SLU_S,     /* single */
     SLU_D,     /* double */
