diff -ru2N Crypt-SSLeay-0.45/Makefile.PL Crypt-SSLeay-0.45-patched/Makefile.PL
--- Crypt-SSLeay-0.45/Makefile.PL	Thu Aug  1 17:25:08 2002
+++ Crypt-SSLeay-0.45-patched/Makefile.PL	Mon Dec 16 20:17:09 2002
@@ -9,10 +9,10 @@
 &GetOptions("default", \$opt_default);
 
-$opt_default ||= $ENV{CRYPT_SSLEAY_DEFAULT};
+$opt_default = 1;
 my $currdir = dirname($0) || '.';
 chdir($currdir) || die("can't change to $currdir: $!");
 
 # FIND POSSIBLE SSL INSTALLATIONS
-my @POSSIBLE_SSL_DIRS = qw(/usr/local/openssl /usr/local/ssl /local/ssl /opt/ssl /usr/local /local /usr);
+my @POSSIBLE_SSL_DIRS = qw(@PREFIX@);
 my @CANDIDATE;
 my $open_ssl = 0;
