diff -ruN evolution-1.2.1/configure evolution-1.2.1-patched/configure
--- evolution-1.2.1/configure	2002-12-05 20:01:27.000000000 +0100
+++ evolution-1.2.1-patched/configure	2003-01-19 05:35:25.000000000 +0100
@@ -4183,7 +4183,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname) $tmp_verstring'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
@@ -7260,7 +7260,7 @@
 fi
 rm -f conftest*
 
-ALL_LINGUAS="az bg ca cs da de el en_AU en_GB es et eu fi fr ga gl hu it ja ko lt lv nl nn no pl pt pt_BR ro ru sk sl sv tr uk vi zh_CN zh_TW"
+ALL_LINGUAS="az bg ca cs da de el en_AU en_GB es et eu fi fr ga gl hu it ja ko lt lv nl nn no pl pt pt_BR ro ru sk sl sv tr uk vi zh_CN"
 # Extract the first word of "ranlib", so it can be a program name with args.
 set dummy ranlib; ac_word=$2
 echo $ac_n "checking for $ac_word""... $ac_c" 1>&6
@@ -9636,7 +9636,7 @@
 
 CPPFLAGS_save="$CPPFLAGS"
 CPPFLAGS="$DB3_CFLAGS $CPPFLAGS"
-for ac_hdr in db.h db3/db.h
+for ac_hdr in db31/db.h
 do
 ac_safe=`echo "$ac_hdr" | sed 'y%./+-%__p_%'`
 echo $ac_n "checking for $ac_hdr""... $ac_c" 1>&6
@@ -9688,7 +9688,7 @@
 #include "confdefs.h"
 
 		#ifdef HAVE_DB3_DB_H
-		#include <db3/db.h>
+		#include <db31/db.h>
 		#else
 		#include <db.h>
 		#endif
@@ -9729,14 +9729,14 @@
 	LIBS_save="$LIBS"
 	ac_cv_db3_ldadd=""
 
-	for name in db db3 db-3.1; do
+	for name in db-3.1; do
 		LIBS="$LIBS_save $with_db3_libs/lib${name}.a"
 		cat > conftest.$ac_ext <<EOF
 #line 9736 "configure"
 #include "confdefs.h"
 
 			#ifdef HAVE_DB3_DB_H
-			#include <db3/db.h>
+			#include <db31/db.h>
 			#else
 			#include <db.h>
 			#endif
@@ -9787,7 +9787,7 @@
 #include "confdefs.h"
 
 		#ifdef HAVE_DB3_DB_H
-		#include <db3/db.h>
+		#include <db31/db.h>
 		#else
 		#include <db.h>
 		#endif
diff -ruN evolution-1.2.1/default_user/Makefile.in evolution-1.2.1-patched/default_user/Makefile.in
--- evolution-1.2.1/default_user/Makefile.in	2002-12-06 00:14:16.000000000 +0100
+++ evolution-1.2.1-patched/default_user/Makefile.in	2003-01-16 03:55:26.000000000 +0100
@@ -455,9 +455,9 @@
 
 
 install-data-hook:
-	-mkdir $(defaultdir)/private
-@ENABLE_NNTP_TRUE@	-mkdir $(defaultdir)/news
-@ENABLE_NNTP_TRUE@	-mkdir $(defaultdir)/config
+	-mkdir $(DESTDIR)$(defaultdir)/private
+@ENABLE_NNTP_TRUE@	-mkdir $(DESTDIR)$(defaultdir)/news
+@ENABLE_NNTP_TRUE@	-mkdir $(DESTDIR)$(defaultdir)/config
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
diff -ruN evolution-1.2.1/e-util/Makefile.in evolution-1.2.1-patched/e-util/Makefile.in
--- evolution-1.2.1/e-util/Makefile.in	2002-12-06 00:07:28.000000000 +0100
+++ evolution-1.2.1-patched/e-util/Makefile.in	2003-01-14 22:07:03.000000000 +0100
@@ -252,7 +252,7 @@
 pilot_sources =  	$(econdinclude_HEADERS)		e-pilot-map.c			e-pilot-settings.c		e-pilot-util.c
 
 @ENABLE_PILOT_CONDUITS_TRUE@pilot_compile = $(pilot_sources)
-@ENABLE_PILOT_CONDUITS_FALSE@pilot_compile = 
+@ENABLE_PILOT_CONDUITS_FALSE@pilot_compile = dummy.c
 
 libeconduit_la_SOURCES = $(pilot_compile)
 
@@ -286,7 +286,7 @@
 e-passwords.lo e-path.lo e-proxy.lo e-request.lo e-sexp.lo \
 e-time-utils.lo e-url.lo md5-utils.lo
 libeconduit_static_la_LIBADD = 
-@ENABLE_PILOT_CONDUITS_FALSE@libeconduit_static_la_OBJECTS = 
+@ENABLE_PILOT_CONDUITS_FALSE@libeconduit_static_la_OBJECTS = dummy.lo
 @ENABLE_PILOT_CONDUITS_TRUE@libeconduit_static_la_OBJECTS =  \
 @ENABLE_PILOT_CONDUITS_TRUE@e-pilot-map.lo e-pilot-settings.lo \
 @ENABLE_PILOT_CONDUITS_TRUE@e-pilot-util.lo
@@ -305,7 +305,7 @@
 libedb3util_la_OBJECTS =  e-db3-utils.lo e-dbhash.lo
 libeconduit_la_LDFLAGS = 
 libeconduit_la_LIBADD = 
-@ENABLE_PILOT_CONDUITS_FALSE@libeconduit_la_OBJECTS = 
+@ENABLE_PILOT_CONDUITS_FALSE@libeconduit_la_OBJECTS = dummy.lo
 @ENABLE_PILOT_CONDUITS_TRUE@libeconduit_la_OBJECTS =  e-pilot-map.lo \
 @ENABLE_PILOT_CONDUITS_TRUE@e-pilot-settings.lo e-pilot-util.lo
 CFLAGS = @CFLAGS@
diff -ruN evolution-1.2.1/e-util/dummy.c evolution-1.2.1-patched/e-util/dummy.c
--- evolution-1.2.1/e-util/dummy.c	1970-01-01 01:00:00.000000000 +0100
+++ evolution-1.2.1-patched/e-util/dummy.c	2003-01-14 22:05:46.000000000 +0100
@@ -0,0 +1,6 @@
+/* Empty placeholder, to prevent link error when no conduits are used */
+
+static int foo()
+{
+	return 0;
+}
diff -ruN evolution-1.2.1/libical/ltmain.sh evolution-1.2.1-patched/libical/ltmain.sh
--- evolution-1.2.1/libical/ltmain.sh	2001-03-30 17:47:33.000000000 +0200
+++ evolution-1.2.1-patched/libical/ltmain.sh	2003-01-19 04:40:32.000000000 +0100
@@ -2152,7 +2152,23 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
-	  eval cmds=\"$archive_cmds\"
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+	  else
+	    tmp_verstring="$verstring"
+	  fi
+	  save_deplibs="$deplibs"
+	  for conv in $convenience; do
+	    tmp_deplibs=
+	    for test_deplib in $deplibs; do
+	      if test "$test_deplib" != "$conv"; then
+	        tmp_deplibs="$tmp_deplibs $test_deplib"
+	      fi
+	    done
+	    deplibs="$tmp_deplibs"
+	  done
+ 	  eval cmds=\"$archive_cmds\"
+	  deplibs="$save_deplibs"
 	fi
 	IFS="${IFS= 	}"; save_ifs="$IFS"; IFS='~'
 	for cmd in $cmds; do
@@ -2623,7 +2639,8 @@
 	  # linked before any other PIC object.  But we must not use
 	  # pic_flag when linking with -static.  The problem exists in
 	  # FreeBSD 2.2.6 and is fixed in FreeBSD 3.1.
-	  *-*-freebsd2*|*-*-freebsd3.0*|*-*-freebsdelf3.0*)
+	  *-*-freebsd2*|*-*-freebsd3.0*|*-*-freebsdelf3.0*)
+
 	    case "$compile_command " in
 	    *" -static "*) ;;
 	    *) pic_flag_for_symtable=" $pic_flag -DPIC -DFREEBSD_WORKAROUND";;
diff -ruN evolution-1.2.1/ltmain.sh evolution-1.2.1-patched/ltmain.sh
--- evolution-1.2.1/ltmain.sh	2002-12-05 20:01:28.000000000 +0100
+++ evolution-1.2.1-patched/ltmain.sh	2003-01-24 23:36:08.000000000 +0100
@@ -1001,6 +1001,10 @@
 	continue
 	;;
 
+      -L@PREFIX@/lib)
+	continue
+	;;
+
       # The native IRIX linker understands -LANG:*, -LIST:* and -LNO:*
       # so, if we see these flags be careful not to treat them like -L
       -L[A-Z][A-Z]*:*)
@@ -1446,6 +1450,7 @@
 	    continue
 	  fi
 	  ;; # -l
+	-L@PREFIX@/lib) continue ;;
 	-L*)
 	  case $linkmode in
 	  lib)
@@ -1706,6 +1711,7 @@
 	  tmp_libs=
 	  for deplib in $dependency_libs; do
 	    case $deplib in
+	    -L@PREFIX@/lib) continue ;;
 	    -L*) newlib_search_path="$newlib_search_path "`$echo "X$deplib" | $Xsed -e 's/^-L//'`;; ### testsuite: skip nested quoting test
 	    esac
 	    # Need to link against all dependency_libs?
@@ -1777,7 +1783,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    need_relink=no
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then
@@ -2049,6 +2055,7 @@
 	    # Add the search paths of all dependency libraries
 	    for deplib in $dependency_libs; do
 	      case $deplib in
+	      -L@PREFIX@/lib) continue ;;
 	      -L*) path="$deplib" ;;
 	      *.la)
 		dir=`$echo "X$deplib" | $Xsed -e 's%/[^/]*$%%'`
@@ -2080,6 +2087,9 @@
 		;;
 	      *) continue ;;
 	      esac
+	      if test "$path" = "-L@PREFIX@/lib"; then
+	        continue
+	      fi
 	      case " $deplibs " in
 	      *" $path "*) ;;
 	      *) deplibs="$deplibs $path" ;;
@@ -2136,6 +2146,7 @@
 	  tmp_libs=
 	  for deplib in $new_libs; do
 	    case $deplib in
+	    -L@PREFIX@/lib) continue ;;
 	    -L*)
 	      case " $tmp_libs " in
 	      *" $deplib "*) ;;
@@ -2657,7 +2668,9 @@
 	      fi
 	    else
 	      # Add a -L argument.
-	      newdeplibs="$newdeplibs $a_deplib"
+	      if test "$a_deplib" != "@PREFIX@/lib"; then
+		newdeplibs="$newdeplibs $a_deplib"
+	      fi
 	    fi
 	  done # Gone through all deplibs.
 	  ;;
@@ -2691,7 +2704,9 @@
 	      fi
 	    else
 	      # Add a -L argument.
-	      newdeplibs="$newdeplibs $a_deplib"
+	      if test "$a_deplib" != "@PREFIX@/lib"; then
+		newdeplibs="$newdeplibs $a_deplib"
+	      fi
 	    fi
 	  done # Gone through all deplibs.
 	  ;;
@@ -2952,7 +2967,23 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
-	  eval cmds=\"$archive_cmds\"
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+	  else
+	    tmp_verstring="$verstring"
+	  fi
+	  save_deplibs="$deplibs"
+	  for conv in $convenience; do
+	    tmp_deplibs=
+	    for test_deplib in $deplibs; do
+	      if test "$test_deplib" != "$conv"; then
+	        tmp_deplibs="$tmp_deplibs $test_deplib"
+	      fi
+	    done
+	    deplibs="$tmp_deplibs"
+	  done
+ 	  eval cmds=\"$archive_cmds\"
+	  deplibs="$save_deplibs"
 	fi
 	save_ifs="$IFS"; IFS='~'
 	for cmd in $cmds; do
