Only in ../../m2crypto-0.09: build
diff -rp ./setup.py ../../m2crypto-0.09/setup.py
*** ./setup.py	Wed Jan  8 00:43:54 2003
--- ../../m2crypto-0.09/setup.py	Fri Feb 21 00:28:11 2003
*************** if os.name == 'nt':
*** 22,29 ****
      #libraries = ['ssleay32_bc', 'libeay32_bc']
  
  elif os.name == 'posix':
!     include_dirs = [my_inc, '/usr/local/include']
!     library_dirs = ['/usr/local/lib']
      libraries = ['ssl', 'crypto']
  
  # Describe the module.
--- 22,29 ----
      #libraries = ['ssleay32_bc', 'libeay32_bc']
  
  elif os.name == 'posix':
!     include_dirs = [my_inc, '/usr/local/include', "/sw/include"]
!     library_dirs = ['/usr/local/lib', "/sw/lib"]
      libraries = ['ssl', 'crypto']
  
  # Describe the module.
diff -rp ./swig/_m2crypto.i ../../m2crypto-0.09/swig/_m2crypto.i
*** ./swig/_m2crypto.i	Mon Dec 23 12:53:00 2002
--- ../../m2crypto-0.09/swig/_m2crypto.i	Fri Feb 21 00:27:22 2003
*************** static int thread_mode = 0;
*** 17,36 ****
  %}
  
  %include constraints.i
! %include _threads.i
! %include _lib.i
! %include _bio.i
! %include _rand.i
! %include _evp.i
! %include _rc4.i
! %include _dh.i
! %include _rsa.i
! %include _dsa.i
! %include _ssl.i
! %include _x509.i
! %include _asn1.i
! %include _pkcs7.i
! %include _util.i
  
  #ifdef SWIG_VERSION
  %constant int encrypt = 1;
--- 17,36 ----
  %}
  
  %include constraints.i
! %include "swig/_threads.i"
! %include "swig/_lib.i"
! %include "swig/_bio.i"
! %include "swig/_rand.i"
! %include "swig/_evp.i"
! %include "swig/_rc4.i"
! %include "swig/_dh.i"
! %include "swig/_rsa.i"
! %include "swig/_dsa.i"
! %include "swig/_ssl.i"
! %include "swig/_x509.i"
! %include "swig/_asn1.i"
! %include "swig/_pkcs7.i"
! %include "swig/_util.i"
  
  #ifdef SWIG_VERSION
  %constant int encrypt = 1;
Only in ../../m2crypto-0.09/swig: _m2crypto.py
Only in ../../m2crypto-0.09/swig: _m2crypto_wrap.c
