diff -Naur mozilla-1.1.old/.mozconfig mozilla-1.1.new/.mozconfig
--- mozilla-1.1.old/.mozconfig	Thu Jan  1 09:00:00 1970
+++ mozilla-1.1.new/.mozconfig	Thu Aug 29 17:21:57 2002
@@ -0,0 +1,13 @@
+ac_add_options	--libdir=@PREFIX@/lib/mozilla
+ac_add_options	--includedir=@PREFIX@/include/mozilla
+ac_add_options	--with-pthreads
+ac_add_options	--with-system-jpeg=@PREFIX@
+ac_add_options	--with-system-zlib=@PREFIX@
+ac_add_options	--with-system-png=@PREFIX@
+ac_add_options	--enable-xft
+ac_add_options	--enable-crypto
+ac_add_options	--disable-tests
+ac_add_options	--disable-debug
+ac_add_options	--enable-optimize
+ac_add_options	--with-default-mozilla-five-home=@PREFIX@/lib/mozilla
+mk_add_options	MOZ_MAKE_FLAGS=-j4
diff -Naur mozilla-1.1.old/configure mozilla-1.1.new/configure
--- mozilla-1.1.old/configure	Sat Aug 10 05:05:10 2002
+++ mozilla-1.1.new/configure	Wed Aug 28 23:55:42 2002
@@ -4525,34 +4525,34 @@
 
 *-darwin*) 
     _PEDANTIC=
-    CFLAGS="$CFLAGS -fpascal-strings -traditional-cpp -fno-common"
-    CXXFLAGS="$CXXFLAGS -fpascal-strings -traditional-cpp -fno-common"
+    CFLAGS="$CFLAGS -fpascal-strings -no-cpp-precomp -fno-common"
+    CXXFLAGS="$CXXFLAGS -fpascal-strings -no-cpp-precomp -fno-common"
     DLL_SUFFIX=".dylib"
     DSO_LDOPTS=''
     STRIP="$STRIP -x -S"
     case "${target_cpu}" in
     powerpc*)
-        cat >> confdefs.h <<\EOF
-#define XP_MACOSX 1
-EOF
-
-        cat >> confdefs.h <<\EOF
-#define TARGET_CARBON 1
-EOF
+#        cat >> confdefs.h <<\EOF
+##define XP_MACOSX 1
+#EOF
+
+#        cat >> confdefs.h <<\EOF
+##define TARGET_CARBON 1
+#EOF
+
+#        cat >> confdefs.h <<\EOF
+##define TARGET_API_MAC_CARBON 1
+#EOF
 
-        cat >> confdefs.h <<\EOF
-#define TARGET_API_MAC_CARBON 1
-EOF
-
-        no_x=yes
+#        no_x=yes
         MACOSX=1
-        _PLATFORM_DEFAULT_TOOLKIT='mac'
+#        _PLATFORM_DEFAULT_TOOLKIT='mac'
         DSO_LDOPTS="$DSO_LDOPTS -arch ppc"
-        TK_LIBS='-framework Carbon /System/Library/Frameworks/Carbon.framework/Carbon'
-        TK_CFLAGS="-I/Developer/Headers/FlatCarbon -F/System/Library/Frameworks"
-        CFLAGS="$CFLAGS $TK_CFLAGS"
-        CXXFLAGS="$CXXFLAGS $TK_CFLAGS"
-        MOZ_ENABLE_POSTSCRIPT=
+#        TK_LIBS='-framework Carbon /System/Library/Frameworks/Carbon.framework/Carbon'
+#        TK_CFLAGS="-I/Developer/Headers/FlatCarbon -F/System/Library/Frameworks"
+#        CFLAGS="$CFLAGS $TK_CFLAGS"
+#        CXXFLAGS="$CXXFLAGS $TK_CFLAGS"
+#        MOZ_ENABLE_POSTSCRIPT=
         ;;
     esac
     ;;
@@ -6137,7 +6137,7 @@
 
 echo "$ac_t""$ac_cv_have_usable_wchar_option" 1>&6
 
-    if test "$ac_cv_have_usable_wchar_option" = "yes"; then
+    if test "$ac_cv_have_usable_wchar_option" = "yes" -a "$OS_ARCH" = "Darwin"; then
         cat >> confdefs.h <<\EOF
 #define HAVE_CPP_2BYTE_WCHAR_T 1
 EOF
@@ -14139,9 +14139,9 @@
 #define XP_UNIX 1
 EOF
 
-  cat >> confdefs.h <<\EOF
-#define UNIX_EMBED 1
-EOF
+#  cat >> confdefs.h <<\EOF
+##define UNIX_EMBED 1
+#EOF
 
   cat >> confdefs.h <<\EOF
 #define UNIX_LDAP 1
diff -Naur mozilla-1.1.old/directory/c-sdk/config/config.mk mozilla-1.1.new/directory/c-sdk/config/config.mk
--- mozilla-1.1.old/directory/c-sdk/config/config.mk	Wed Mar 27 06:51:57 2002
+++ mozilla-1.1.new/directory/c-sdk/config/config.mk	Wed Aug 28 23:35:53 2002
@@ -60,10 +60,10 @@
 
 NFSPWD		= $(MOD_DEPTH)/config/nfspwd
 
-CFLAGS		= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(OS_CFLAGS)\
-		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
-CCCFLAGS	= $(CCC_ONLY_FLAGS) $(OPTIMIZER) $(OS_CFLAGS)\
-		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
+CFLAGS		= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(XP_DEFINE)\
+		  $(DEFINES) $(INCLUDES) $(XCFLAGS) $(OS_CFLAGS)
+CCCFLAGS	= $(CCC_ONLY_FLAGS) $(OPTIMIZER) $(XP_DEFINE)\
+		  $(DEFINES) $(INCLUDES) $(XCFLAGS) $(OS_CFLAGS)
 # For purify
 NOMD_CFLAGS	= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(NOMD_OS_CFLAGS)\
 		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
diff -Naur mozilla-1.1.old/directory/c-sdk/configure mozilla-1.1.new/directory/c-sdk/configure
--- mozilla-1.1.old/directory/c-sdk/configure	Wed Jun 19 06:34:56 2002
+++ mozilla-1.1.new/directory/c-sdk/configure	Wed Aug 28 23:35:53 2002
@@ -3347,7 +3347,7 @@
 
         CPU_ARCH=ppc
     fi
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @PREFIX@/lib/mozilla/$@'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS) -o $@'
diff -Naur mozilla-1.1.old/fink/mozilla mozilla-1.1.new/fink/mozilla
--- mozilla-1.1.old/fink/mozilla	Thu Jan  1 09:00:00 1970
+++ mozilla-1.1.new/fink/mozilla	Wed Aug 28 23:35:53 2002
@@ -0,0 +1,99 @@
+#!/bin/sh
+#
+# The contents of this file are subject to the Netscape Public
+# License Version 1.1 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of
+# the License at http://www.mozilla.org/NPL/
+#
+# Software distributed under the License is distributed on an "AS
+# IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# rights and limitations under the License.
+#
+# The Original Code is mozilla.org code.
+#
+# The Initial Developer of the Original Code is Netscape
+# Communications Corporation.  Portions created by Netscape are
+# Copyright (C) 1998 Netscape Communications Corporation. All
+# Rights Reserved.
+#
+# Contributor(s): 
+#
+
+## 
+## Usage:
+##
+## $ mozilla
+##
+## This script is meant to run a mozilla program from the mozilla
+## rpm installation.
+##
+## The script will setup all the environment voodoo needed to make
+## mozilla work.
+
+##
+## Variables
+##
+MOZ_PROGRAM="@PREFIX@/lib/mozilla/mozilla-bin"
+
+##
+## Set MOZILLA_FIVE_HOME
+##
+MOZILLA_FIVE_HOME="@PREFIX@/lib/mozilla"
+export MOZILLA_FIVE_HOME
+
+##
+## Set DYLD_LIBRARY_PATH
+##
+DYLD_LIBRARY_PATH=@PREFIX@/lib/mozilla:$DYLD_LIBRARY_PATH
+export DYLD_LIBRARY_PATH
+
+# If there is no command line argument at all then try to open a new
+# window in an already running instance.
+
+if [ -z "$1" ]; then
+  $MOZ_PROGRAM -remote "openurl(about:blank,new-window)" 2>/dev/null 1>/dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "2" ]; then
+    $MOZ_PROGRAM ${1+"$@"}
+    RETURN_VAL=$?
+  fi
+  exit $RETURN_VAL
+fi
+
+# If there's a command line argument but it doesn't begin with a -
+# it's probably a url.  Try to send it to a running instance.
+
+USE_EXIST=0
+opt="$1"
+case "$opt" in
+  -*) ;;
+  *) USE_EXIST=1 ;;
+esac
+
+if [ "${USE_EXIST}" -eq "1" ]; then
+  # check to make sure that the command contains at least a :/ in it.
+  echo $opt | grep -e ':/' 2>/dev/null > /dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "1" ]; then
+    # does it begin with a / ?
+    echo $opt | grep -e '^/' 2>/dev/null > /dev/null
+    RETURN_VAL=$?
+    if [ "$RETURN_VAL" -eq "0" ]; then
+      opt="file:$opt"
+    elif [ -e `pwd`/$opt ]; then
+      opt="`pwd`/$opt"
+    else
+      opt="http://$opt"
+    fi
+  fi
+  $MOZ_PROGRAM -remote "openurl($opt,new-window)" 2>/dev/null >/dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "2" ]; then
+    $MOZ_PROGRAM ${1+"$@"}
+    RETURN_VAL=$?
+  fi
+  exit $RETURN_VAL
+fi
+
+$MOZ_PROGRAM ${1+"$@"}
diff -Naur mozilla-1.1.old/fink/mozilla-address.desktop mozilla-1.1.new/fink/mozilla-address.desktop
--- mozilla-1.1.old/fink/mozilla-address.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.1.new/fink/mozilla-address.desktop	Wed Aug 28 23:35:53 2002
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Addressbook
+Comment=Addressbook from Mozilla Mailnews
+Exec=mozilla -addressbook
+Icon=gnome-calendar.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.1.old/fink/mozilla-chatzilla.desktop mozilla-1.1.new/fink/mozilla-chatzilla.desktop
--- mozilla-1.1.old/fink/mozilla-chatzilla.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.1.new/fink/mozilla-chatzilla.desktop	Wed Aug 28 23:35:53 2002
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Chatzilla
+Comment=IRC client
+Exec=mozilla -chat
+Icon=gnome-irc.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.1.old/fink/mozilla-editor.desktop mozilla-1.1.new/fink/mozilla-editor.desktop
--- mozilla-1.1.old/fink/mozilla-editor.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.1.new/fink/mozilla-editor.desktop	Wed Aug 28 23:35:53 2002
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Editor
+Comment=HTML Composer
+Exec=mozilla -edit
+Icon=gnome-word.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.1.old/fink/mozilla-mail-composer.desktop mozilla-1.1.new/fink/mozilla-mail-composer.desktop
--- mozilla-1.1.old/fink/mozilla-mail-composer.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.1.new/fink/mozilla-mail-composer.desktop	Wed Aug 28 23:35:53 2002
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Mail Composer
+Comment=Mozilla Mail Composer
+Exec=mozilla -compose
+Icon=mozilla-mail-composer.gif
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.1.old/fink/mozilla-mail.desktop mozilla-1.1.new/fink/mozilla-mail.desktop
--- mozilla-1.1.old/fink/mozilla-mail.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.1.new/fink/mozilla-mail.desktop	Wed Aug 28 23:35:53 2002
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Mail
+Comment=Mozilla Mail
+Exec=mozilla -mail
+Icon=mozilla-mail.gif
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.1.old/fink/mozilla-news.desktop mozilla-1.1.new/fink/mozilla-news.desktop
--- mozilla-1.1.old/fink/mozilla-news.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.1.new/fink/mozilla-news.desktop	Wed Aug 28 23:35:53 2002
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla News
+Comment=Mozilla Mail
+Exec=mozilla -news
+Icon=gnome-news.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.1.old/fink/mozilla.desktop mozilla-1.1.new/fink/mozilla.desktop
--- mozilla-1.1.old/fink/mozilla.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.1.new/fink/mozilla.desktop	Wed Aug 28 23:35:53 2002
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla
+Comment=Mozilla Web Browser
+Exec=mozilla 
+Icon=mozilla.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.1.old/gfx/src/gtk/Makefile.in mozilla-1.1.new/gfx/src/gtk/Makefile.in
--- mozilla-1.1.old/gfx/src/gtk/Makefile.in	Tue May  7 21:02:50 2002
+++ mozilla-1.1.new/gfx/src/gtk/Makefile.in	Wed Aug 28 23:35:53 2002
@@ -177,6 +177,8 @@
 ifdef MOZ_ENABLE_FREETYPE2
 DEFINES         += -DMOZ_ENABLE_FREETYPE2
 INCLUDES        += $(FT2_CFLAGS)
+CC              += $(FT2_CFLAGS)
+CXX             += $(FT2_CFLAGS)
 # due to runtime linking with PR_LoadLibrary
 # EXTRA_DSO_LDOPTS does not need $(FT2_LIBS)
 endif
diff -Naur mozilla-1.1.old/nsprpub/configure mozilla-1.1.new/nsprpub/configure
--- mozilla-1.1.old/nsprpub/configure	Wed Jun 19 06:34:50 2002
+++ mozilla-1.1.new/nsprpub/configure	Wed Aug 28 23:35:53 2002
@@ -3224,7 +3224,7 @@
 
         CPU_ARCH=ppc
     fi
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @PREFIX@/lib/mozilla/$@'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS) -o $@'
diff -Naur mozilla-1.1.old/other-licenses/Xft/Xft/Makefile.in mozilla-1.1.new/other-licenses/Xft/Xft/Makefile.in
--- mozilla-1.1.old/other-licenses/Xft/Xft/Makefile.in	Thu Feb 28 15:59:01 2002
+++ mozilla-1.1.new/other-licenses/Xft/Xft/Makefile.in	Wed Aug 28 23:35:53 2002
@@ -44,4 +44,6 @@
 
 ifdef MOZ_XFT_SYSTEM_FREETYPE2
 CFLAGS		+= $(FT2_CFLAGS)
+CC		+= $(FT2_CFLAGS)
+CXX		+= $(FT2_CFLAGS)
 endif
diff -Naur mozilla-1.1.old/other-licenses/Xft/fontconfig/src/Makefile.in mozilla-1.1.new/other-licenses/Xft/fontconfig/src/Makefile.in
--- mozilla-1.1.old/other-licenses/Xft/fontconfig/src/Makefile.in	Fri Apr 26 04:03:35 2002
+++ mozilla-1.1.new/other-licenses/Xft/fontconfig/src/Makefile.in	Wed Aug 28 23:35:53 2002
@@ -45,4 +45,6 @@
 
 ifdef MOZ_XFT_SYSTEM_FREETYPE2
 CFLAGS		+= $(FT2_CFLAGS)
+CC		+= $(FT2_CFLAGS)
+CXX		+= $(FT2_CFLAGS)
 endif
diff -Naur mozilla-1.1.old/other-licenses/expat/xmlwf/Makefile.in mozilla-1.1.new/other-licenses/expat/xmlwf/Makefile.in
--- mozilla-1.1.old/other-licenses/expat/xmlwf/Makefile.in	Mon Feb 25 15:53:11 2002
+++ mozilla-1.1.new/other-licenses/expat/xmlwf/Makefile.in	Wed Aug 28 23:35:53 2002
@@ -23,7 +23,7 @@
 LIBDIR= ../lib/.libs
 INCDIR= ../lib
 
-LDFLAGS= @LDFLAGS@ -static
+LDFLAGS= @LDFLAGS@
 CFLAGS= @CFLAGS@ -I$(INCDIR)
 CC = @CC@
 
diff -Naur mozilla-1.1.old/security/coreconf/Darwin.mk mozilla-1.1.new/security/coreconf/Darwin.mk
--- mozilla-1.1.old/security/coreconf/Darwin.mk	Wed Jun 12 09:22:41 2002
+++ mozilla-1.1.new/security/coreconf/Darwin.mk	Wed Aug 28 23:35:53 2002
@@ -57,7 +57,7 @@
 # definitions so that the linker can catch multiply-defined symbols.
 # Also, common symbols are not allowed with Darwin dynamic libraries.
 
-OS_CFLAGS	= $(DSO_CFLAGS) $(OS_REL_CFLAGS) -Wmost -fpascal-strings -traditional-cpp -fno-common -pipe -DDARWIN -DHAVE_STRERROR -DHAVE_BSD_FLOCK
+OS_CFLAGS	= $(DSO_CFLAGS) $(OS_REL_CFLAGS) -Wmost -fpascal-strings -no-cpp-precomp -fno-common -pipe -DDARWIN -DHAVE_STRERROR -DHAVE_BSD_FLOCK
 
 ifdef BUILD_OPT
 OPTIMIZER	= -O2
@@ -66,7 +66,7 @@
 ARCH		= darwin
 
 # May override this with -bundle to create a loadable module.
-DSO_LDOPTS	= -dynamiclib -compatibility_version 1 -current_version 1 -install_name @executable_path/$(notdir $@)
+DSO_LDOPTS	= -dynamiclib -compatibility_version 1 -current_version 1 -install_name @PREFIX@/lib/mozilla/$(notdir $@)
 
 MKSHLIB		= $(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS)
 DLL_SUFFIX	= dylib
diff -Naur mozilla-1.1.old/security/nss/cmd/platlibs.mk mozilla-1.1.new/security/nss/cmd/platlibs.mk
--- mozilla-1.1.old/security/nss/cmd/platlibs.mk	Fri Apr  5 12:30:35 2002
+++ mozilla-1.1.new/security/nss/cmd/platlibs.mk	Wed Aug 28 23:35:53 2002
@@ -177,7 +177,7 @@
 endif
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file @PREFIX@/lib/mozilla/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 # $(PROGRAM) has NO explicit dependencies on $(EXTRA_SHARED_LIBS)
diff -Naur mozilla-1.1.old/security/nss/lib/smime/config.mk mozilla-1.1.new/security/nss/lib/smime/config.mk
--- mozilla-1.1.old/security/nss/lib/smime/config.mk	Sat Feb 16 08:38:43 2002
+++ mozilla-1.1.new/security/nss/lib/smime/config.mk	Wed Aug 28 23:35:53 2002
@@ -60,7 +60,7 @@
 	$(NULL)
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file @PREFIX@/lib/mozilla/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 endif
diff -Naur mozilla-1.1.old/security/nss/lib/ssl/config.mk mozilla-1.1.new/security/nss/lib/ssl/config.mk
--- mozilla-1.1.old/security/nss/lib/ssl/config.mk	Fri Feb 22 13:23:22 2002
+++ mozilla-1.1.new/security/nss/lib/ssl/config.mk	Wed Aug 28 23:35:53 2002
@@ -64,7 +64,7 @@
 endif
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file @PREFIX@/lib/mozilla/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 endif
