diff -ru neon-0.18.5/configure neon-0.18.5-patched/configure
--- neon-0.18.5/configure	2002-01-27 12:18:05.000000000 +0100
+++ neon-0.18.5-patched/configure	2002-04-15 13:20:33.000000000 +0200
@@ -4509,7 +4509,7 @@
     ;;
 
   darwin* | rhapsody*)
-    allow_undefined_flag='-undefined suppress'
+    allow_undefined_flag='-undefined suppress -flat_namespace'
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
diff -ru neon-0.18.5/src/ne_xml.c neon-0.18.5-patched/src/ne_xml.c
--- neon-0.18.5/src/ne_xml.c	2002-01-03 15:24:17.000000000 +0100
+++ neon-0.18.5-patched/src/ne_xml.c	2002-04-15 13:20:18.000000000 +0200
@@ -58,7 +58,7 @@
 #endif
 
 /******** libxml **********/
-#  include <parser.h>
+#  include <libxml/parser.h>
 typedef xmlChar ne_xml_char;
 
 # else /* not HAVE_LIBXML */
diff -ru neon-0.19.3/src/http_base64.c libghttp-1.0.9-patched/src/http_base64.c
--- neon-0.19.3/src/http_base64.c	Wed Dec  9 15:26:32 1998
+++ neon-0.19.3-patched/src/base64.c	Sun Jun 30 15:01:27 2002
@@ -38,7 +38,7 @@
 static const char b64_alphabet[65] = { 
     "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
     "abcdefghijklmnopqrstuvwxyz"
-    "0123456789+/=" };
+    "0123456789+/=\0" };
     
 char *ne_base64(const char *text, size_t inlen)
 {
