diff -ru openssh-3.7.1p1/configure openssh-3.7.1p1-patched/configure
--- openssh-3.7.1p1/configure	2003-09-16 23:38:28.000000000 +0200
+++ openssh-3.7.1p1-patched/configure	2003-09-17 14:33:13.000000000 +0200
@@ -14960,7 +14960,7 @@
 
 # Looking for programs, paths and files
 
-PRIVSEP_PATH=/var/empty
+PRIVSEP_PATH='${prefix}/var/empty'
 
 # Check whether --with-privsep-path or --without-privsep-path was given.
 if test "${with_privsep_path+set}" = set; then
