diff -ru openssh-3.7.1p2/configure openssh-3.7.1p2-patched/configure
--- openssh-3.7.1p2/configure	2003-09-27 18:01:49.000000000 +0200
+++ openssh-3.7.1p2-patched/configure	2003-09-27 18:02:08.000000000 +0200
@@ -15140,7 +15140,7 @@
 
 # Looking for programs, paths and files
 
-PRIVSEP_PATH=/var/empty
+PRIVSEP_PATH='${prefix}/var/empty'
 
 # Check whether --with-privsep-path or --without-privsep-path was given.
 if test "${with_privsep_path+set}" = set; then
