diff -Nur silc-client-0.9.5.1/configure silc-client-0.9.5.1-patched/configure
--- silc-client-0.9.5.1/configure	Sat Sep  7 18:04:38 2002
+++ silc-client-0.9.5.1-patched/configure	Mon Sep  9 11:54:54 2002
@@ -5011,7 +5011,7 @@
     ;;
 
   darwin* | rhapsody*)
-    allow_undefined_flag='-undefined suppress'
+    allow_undefined_flag='-flat_namespace -undefined suppress'
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
diff -Nur silc-client-0.9.5.1/irssi/configure silc-client-0.9.5.1-patched/irssi/configure
--- silc-client-0.9.5.1/irssi/configure	Sat Sep  7 18:05:03 2002
+++ silc-client-0.9.5.1-patched/irssi/configure	Mon Sep  9 11:58:04 2002
@@ -5253,11 +5253,11 @@
     ;;
 
   darwin* | rhapsody*)
-    allow_undefined_flag='-undefined suppress'
+    allow_undefined_flag='-flat_namespace -undefined suppress'
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linkopts -install_name $rpath/$soname $(test -n "$verstring" -a x$verstring != x0.0 && echo $verstring)'
+    archive_cmds='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linkopts -install_name $rpath/$soname $tmp_versionstring'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -Nur silc-client-0.9.5.1/irssi/src/perl/Makefile.in silc-client-0.9.5.1-patched/irssi/src/perl/Makefile.in
--- silc-client-0.9.5.1/irssi/src/perl/Makefile.in	Sat Sep  7 18:05:06 2002
+++ silc-client-0.9.5.1-patched/irssi/src/perl/Makefile.in	Mon Sep  9 11:55:12 2002
@@ -574,7 +574,7 @@
 
 install-exec-local:
 	for dir in $(perl_dirs); do \
-	  cd $$dir && $(MAKE) install && cd ..; \
+	  cd $$dir && $(MAKE) pure_install && cd ..; \
 	done
 
 clean-generic:
diff -Nur silc-client-0.9.5.1/ltmain.sh silc-client-0.9.5.1-patched/ltmain.sh
--- silc-client-0.9.5.1/ltmain.sh	Tue Jan 29 20:26:03 2002
+++ silc-client-0.9.5.1-patched/ltmain.sh	Mon Sep  9 11:58:49 2002
@@ -2858,6 +2858,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+	  else
+	    tmp_verstring="$verstring"
+	  fi
 	  eval cmds=\"$archive_cmds\"
 	fi
 	IFS="${IFS= 	}"; save_ifs="$IFS"; IFS='~'
