diff -Naur silc-client-0.9.7/configure silc-client-0.9.7-patched/configure
--- silc-client-0.9.7/configure	Mon Oct 21 19:11:09 2002
+++ silc-client-0.9.7-patched/configure	Wed Oct 23 07:40:05 2002
@@ -12407,6 +12407,9 @@
 #line $LINENO "configure"
 #include "confdefs.h"
 
+      #ifdef HAVE_SYS_TYPES_H
+      #include <sys/types.h>
+      #endif
       #ifdef HAVE_NETINET_TCP_H
       #include <netinet/tcp.h>
       #endif
diff -Naur silc-client-0.9.7/irssi/src/perl/Makefile.in silc-client-0.9.7-patched/irssi/src/perl/Makefile.in
--- silc-client-0.9.7/irssi/src/perl/Makefile.in	Mon Oct 21 19:11:48 2002
+++ silc-client-0.9.7-patched/irssi/src/perl/Makefile.in	Wed Oct 23 07:12:44 2002
@@ -693,7 +693,7 @@
 
 install-exec-local:
 	for dir in $(perl_dirs); do \
-	  cd $$dir && $(MAKE) install && cd ..; \
+	  cd $$dir && $(MAKE) pure_install && cd ..; \
 	done
 
 clean-generic:
diff -Naur silc-client-0.9.7/lib/silcsim/Makefile.in silc-client-0.9.7-patched/lib/silcsim/Makefile.in
--- silc-client-0.9.7/lib/silcsim/Makefile.in	Mon Oct 21 19:11:16 2002
+++ silc-client-0.9.7-patched/lib/silcsim/Makefile.in	Wed Oct 23 11:18:29 2002
@@ -546,12 +546,12 @@
 
 $(SIM_CIPHER_OBJS): ../silccrypt/libsilccrypt.a
 	$(LTCOMPILE) -c $(srcdir)/../silccrypt/$*.c
-	$(LIBTOOL) --mode=link $(LINK) -rpath $(silc_modulesdir) -o lib$*.la $*.lo
+	$(LIBTOOL) --mode=link $(CCLD) -rpath $(silc_modulesdir) -o lib$*.la $*.lo
 	cd $(srcdir) && $(LN_S) -f $(srcdir)/.libs/lib$*.so $(srcdir)/$*.sim.so
 
 $(SIM_HASH_OBJS): ../silccrypt/libsilccrypt.a
 	$(LTCOMPILE) -c $(srcdir)/../silccrypt/$*.c
-	$(LIBTOOL) --mode=link $(LINK) -rpath $(silc_modulesdir) -o lib$*.la $*.lo
+	$(LIBTOOL) --mode=link $(CCLD) -rpath $(silc_modulesdir) -o lib$*.la $*.lo
 	cd $(srcdir) && $(LN_S) -f $(srcdir)/.libs/lib$*.so $(srcdir)/$*.sim.so
 
 @INCLUDE_DEFINES_INT@
                                                                                       
