diff -u -Nar yafc-0.7.9/NEWS yafc-0.7.9.fink/NEWS
--- yafc-0.7.9/NEWS	Thu May  9 09:45:38 2002
+++ yafc-0.7.9.fink/NEWS	Fri May 31 17:04:58 2002
@@ -222,7 +222,7 @@
 
 * added --help option for untag
 * added *.md5 to ascii_transfer_mask
-* now parses system-wide /etc/yafcrc on startup
+* now parses system-wide @PREFIX@/etc/yafcrc on startup
   + doesn't create default ~/.yafc/yafcrc on first run
 * rglob.c: didn't update directory cache properly
 * segfault in get/put if connection timed out
diff -u -Nar yafc-0.7.9/configure yafc-0.7.9.fink/configure
--- yafc-0.7.9/configure	Tue May 28 05:54:40 2002
+++ yafc-0.7.9.fink/configure	Fri May 31 17:04:58 2002
@@ -758,7 +758,7 @@
 
 
 # expand $ac_aux_dir to an absolute path
-am_aux_dir=`CDPATH=:; cd $ac_aux_dir && pwd`
+am_aux_dir=`CDPATH=:; cd $ac_aux_dir`
 
 # Find a good install program.  We prefer a C program (faster),
 # so one script is as good as another.  But avoid the broken or
diff -u -Nar yafc-0.7.9/doc/yafc.info yafc-0.7.9.fink/doc/yafc.info
--- yafc-0.7.9/doc/yafc.info	Wed Apr 24 14:26:05 2002
+++ yafc-0.7.9.fink/doc/yafc.info	Fri May 31 17:04:58 2002
@@ -1497,7 +1497,7 @@
 Configuration
 *************
 
-   Yafc will parse a system-wide configuration file, `/etc/yafcrc',
+   Yafc will parse a system-wide configuration file, `@PREFIX@/etc/yafcrc',
 first. Then, personal settings are read from `~/.yafc/yafcrc'
 (overriding the global ones). If none of those files exists, internal
 defaults will be used.
diff -u -Nar yafc-0.7.9/doc/yafc_config.texi yafc-0.7.9.fink/doc/yafc_config.texi
--- yafc-0.7.9/doc/yafc_config.texi	Tue Dec 11 06:21:49 2001
+++ yafc-0.7.9.fink/doc/yafc_config.texi	Fri May 31 17:04:58 2002
@@ -2,7 +2,7 @@
 @chapter Configuration
 
 Yafc will parse a system-wide configuration file,
-@file{/etc/yafcrc}, first. Then, personal settings are read from
+@file{@PREFIX@/etc/yafcrc}, first. Then, personal settings are read from
 @file{~/.yafc/yafcrc} (overriding the global ones). If none of those
 files exists, internal defaults will be used.
 
diff -u -Nar yafc-0.7.9/src/main.c yafc-0.7.9.fink/src/main.c
--- yafc-0.7.9/src/main.c	Thu May  9 08:28:36 2002
+++ yafc-0.7.9.fink/src/main.c	Fri May 31 17:04:58 2002
@@ -384,7 +384,7 @@
 
 	if(!override_yafcrc) {
 		char *tmp;
-		parse_rc("/etc/yafcrc", false);
+		parse_rc("@PREFIX@/etc/yafcrc", false);
 		parse_rc(configfile, false);
 		asprintf(&tmp, "%s/bookmarks", gvWorkingDirectory);
 		parse_rc(tmp, false);
diff -u -Nar yafc-0.7.9/yafc-import_ncftp.pl yafc-0.7.9.fink/yafc-import_ncftp.pl
--- yafc-0.7.9/yafc-import_ncftp.pl	Wed Dec 31 19:00:00 1969
+++ yafc-0.7.9.fink/yafc-import_ncftp.pl	Fri May 31 17:04:58 2002
@@ -0,0 +1,68 @@
+#!/usr/bin/perl -w
+#
+# A simple ncftp -> yafc bookmark converter by Mike Wayne (arc@cts.com)
+# Sorry if my code sucks!
+#
+# Example usage:
+# yafc-import_ncftp.pl ~/.ncftp/bookmarks >> ~/.yafc/bookmarks
+#
+# If a filename argument is not given, ~/.ncftp/bookmarks will be read.
+
+use strict;
+my $mime;
+
+# set these to their appropriate settings.
+my $anonpass	= 'user@host.com';			# anonymous password
+my $ncftpbm	= "$ENV{HOME}/.ncftp/bookmarks";	# ncftp bookmarks
+
+eval 'use MIME::Base64';
+if ($@ eq '') { $mime = 1 }
+
+chomp($anonpass = encode_base64($anonpass)) if $mime;
+$ncftpbm = $ARGV[0] if $ARGV[0];
+
+open(BOOKMARKS, $ncftpbm) or die "Unable to open ncftp bookmarks file $ncftpbm: $!\n";
+
+print <<NOTICE;
+# this is an automagically created file
+# so don't bother placing comments here, they will be overwritten
+# make sure this file isn't world readable if passwords are stored here
+
+NOTICE
+
+while (<BOOKMARKS>)
+{
+	if ($_ =~ /,/)
+	{
+		my @bookmark = split(/,/, $_);
+
+		$bookmark[2] ||= 'anonymous';
+		$bookmark[7] ||= 21;
+	
+		print "machine $bookmark[1]:$bookmark[7] alias '$bookmark[0]'\n  login ";
+
+		if ($bookmark[2] eq 'anonymous')
+		{
+			print 'anonymous password ';
+			print "[base64]" if $mime;
+			print $anonpass;
+		}
+
+		else
+		{
+			print $bookmark[2];
+
+			$bookmark[3] =~ s/^\*encoded\*/\[base64\]/ if $bookmark[3];
+			print " password $bookmark[3]" if $bookmark[3];
+		}
+
+		$bookmark[5] = "\"$bookmark[5]\"" if ($bookmark[5] =~ m/ /);
+		print " cwd $bookmark[5]" if ($bookmark[5] && $bookmark[5] ne '/');
+
+		print "\n\n";
+	}
+}
+
+print "# end of bookmark file\n";
+
+close(BOOKMARKS);
