Only in monotone-0.31-patched: cmd_automate.cc.orig
diff -ru monotone-0.31/configure monotone-0.31-patched/configure
--- monotone-0.31/configure	2006-11-11 13:08:59.000000000 -0600
+++ monotone-0.31-patched/configure	2006-11-14 00:09:01.000000000 -0600
@@ -3256,13 +3256,13 @@
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g -O0"
   else
     CFLAGS="-g"
   fi
 else
   if test "$GCC" = yes; then
-    CFLAGS="-O2"
+    CFLAGS="-O0"
   else
     CFLAGS=
   fi
@@ -4527,13 +4527,13 @@
   CXXFLAGS=$ac_save_CXXFLAGS
 elif test $ac_cv_prog_cxx_g = yes; then
   if test "$GXX" = yes; then
-    CXXFLAGS="-g -O2"
+    CXXFLAGS="-g -O0"
   else
     CXXFLAGS="-g"
   fi
 else
   if test "$GXX" = yes; then
-    CXXFLAGS="-O2"
+    CXXFLAGS="-O0"
   else
     CXXFLAGS=
   fi
Only in monotone-0.31-patched: configure.orig
diff -ru monotone-0.31/hash_map.hh monotone-0.31-patched/hash_map.hh
--- monotone-0.31/hash_map.hh	2006-11-11 13:08:11.000000000 -0600
+++ monotone-0.31-patched/hash_map.hh	2006-11-14 00:09:01.000000000 -0600
@@ -3,6 +3,7 @@
 
 #include "config.h"
 
+#undef _T
 #include <functional>
 namespace hashmap {
 
