diff -ru SDL_ttf-2.0.8/SDL_ttf.c SDL_ttf-2.0.8-patched/SDL_ttf.c
--- SDL_ttf-2.0.8/SDL_ttf.c	2006-05-01 11:26:17.000000000 +0200
+++ SDL_ttf-2.0.8-patched/SDL_ttf.c	2007-02-13 13:07:22.000000000 +0100
@@ -43,12 +43,6 @@
 #include FT_FREETYPE_H
 #include FT_OUTLINE_H
 #include FT_TRUETYPE_IDS_H
-/*
-#include <freetype/freetype.h>
-#include <freetype/ftoutln.h>
-#include <freetype/ttnameid.h>
-*/
-#include <freetype/internal/ftobjs.h>
 
 #ifndef FT_OPEN_STREAM
 #define FT_OPEN_STREAM ft_open_stream
@@ -278,7 +272,6 @@
 	}
 	memset(stream, 0, sizeof(*stream));
 
-	stream->memory = library->memory;
 	stream->read = RWread;
 	stream->descriptor.pointer = src;
 	stream->pos = (unsigned long)position;
