diff -ru SDL-1.2.14/src/cpuinfo/SDL_cpuinfo.c /Users/mhorn/Projekte/foreign/SDL/SDL-1.2/src/cpuinfo/SDL_cpuinfo.c
--- SDL-1.2.14/src/cpuinfo/SDL_cpuinfo.c	2009-10-13 01:07:14.000000000 +0200
+++ /Users/mhorn/Projekte/foreign/SDL/SDL-1.2/src/cpuinfo/SDL_cpuinfo.c	2009-11-01 20:30:29.000000000 +0100
@@ -148,7 +148,7 @@
 	int features = 0;
 #if defined(__GNUC__) && defined(i386)
 	__asm__ (
-"        movl    %%ebx,%%edi\n"
+"        pushl   %%ebx\n"
 "        xorl    %%eax,%%eax         # Set up for CPUID instruction    \n"
 "        cpuid                       # Get and save vendor ID          \n"
 "        cmpl    $1,%%eax            # Make sure 1 is valid input for CPUID\n"
@@ -158,14 +158,14 @@
 "        cpuid                       # Get family/model/stepping/features\n"
 "        movl    %%edx,%0                                              \n"
 "1:                                                                    \n"
-"        movl    %%edi,%%ebx\n"
+"        popl    %%ebx\n"
 	: "=m" (features)
 	:
-	: "%eax", "%ecx", "%edx", "%edi"
+	: "%eax", "%ecx", "%edx"
 	);
 #elif defined(__GNUC__) && defined(__x86_64__)
 	__asm__ (
-"        movq    %%rbx,%%rdi\n"
+"        pushq   %%rbx\n"
 "        xorl    %%eax,%%eax         # Set up for CPUID instruction    \n"
 "        cpuid                       # Get and save vendor ID          \n"
 "        cmpl    $1,%%eax            # Make sure 1 is valid input for CPUID\n"
@@ -175,10 +175,10 @@
 "        cpuid                       # Get family/model/stepping/features\n"
 "        movl    %%edx,%0                                              \n"
 "1:                                                                    \n"
-"        movq    %%rdi,%%rbx\n"
+"        popq    %%rbx\n"
 	: "=m" (features)
 	:
-	: "%rax", "%rbx", "%rcx", "%rdx", "%rdi"
+	: "%rax", "%rcx", "%rdx"
 	);
 #elif (defined(_MSC_VER) && defined(_M_IX86)) || defined(__WATCOMC__)
 	__asm {
@@ -194,7 +194,7 @@
 	}
 #elif defined(__sun) && (defined(__i386) || defined(__amd64))
 	    __asm(
-"        movl    %ebx,%edi\n"
+"        pushl   %ebx\n"
 "        xorl    %eax,%eax         \n"
 "        cpuid                     \n"
 "        cmpl    $1,%eax           \n"
@@ -208,7 +208,7 @@
 "        movl    %edx,-8(%rbp)     \n"
 #endif
 "1:                                \n"
-"        movl    %edi,%ebx\n" );
+"        popl    %ebx\n" );
 #endif
 	return features;
 }
@@ -218,7 +218,7 @@
 	int features = 0;
 #if defined(__GNUC__) && defined(i386)
 	__asm__ (
-"        movl    %%ebx,%%edi\n"
+"        pushl   %%ebx\n"
 "        movl    $0x80000000,%%eax   # Query for extended functions    \n"
 "        cpuid                       # Get extended function limit     \n"
 "        cmpl    $0x80000001,%%eax                                     \n"
@@ -227,14 +227,14 @@
 "        cpuid                       # and get the information         \n"
 "        movl    %%edx,%0                                              \n"
 "1:                                                                    \n"
-"        movl    %%edi,%%ebx\n"
+"        popl    %%ebx\n"
 	: "=m" (features)
 	:
-	: "%eax", "%ecx", "%edx", "%edi"
+	: "%eax", "%ecx", "%edx"
 	);
 #elif defined(__GNUC__) && defined (__x86_64__)
 	__asm__ (
-"        movq    %%rbx,%%rdi\n"
+"        pushq   %%rbx\n"
 "        movl    $0x80000000,%%eax   # Query for extended functions    \n"
 "        cpuid                       # Get extended function limit     \n"
 "        cmpl    $0x80000001,%%eax                                     \n"
@@ -243,10 +243,10 @@
 "        cpuid                       # and get the information         \n"
 "        movl    %%edx,%0                                              \n"
 "1:                                                                    \n"
-"        movq    %%rdi,%%rbx\n"
+"        popq    %%rbx\n"
 	: "=m" (features)
 	:
-	: "%rax", "%rbx", "%rcx", "%rdx", "%rdi"
+	: "%rax", "%rcx", "%rdx"
 	);
 #elif (defined(_MSC_VER) && defined(_M_IX86)) || defined(__WATCOMC__)
 	__asm {
@@ -261,7 +261,7 @@
 	}
 #elif defined(__sun) && ( defined(__i386) || defined(__amd64) )
 	    __asm (
-"        movl    %ebx,%edi\n"
+"        pushl   %ebx\n"
 "        movl    $0x80000000,%eax \n"
 "        cpuid                    \n"
 "        cmpl    $0x80000001,%eax \n"
@@ -274,7 +274,7 @@
 "        movl    %edx,-8(%rbp)   \n"
 #endif
 "1:                               \n"
-"        movl    %edi,%ebx\n"
+"        popl    %ebx\n"
 	    );
 #endif
 	return features;
diff -ru SDL-1.2.14/src/video/quartz/SDL_QuartzGL.m /Users/mhorn/Projekte/foreign/SDL/SDL-1.2/src/video/quartz/SDL_QuartzGL.m
--- SDL-1.2.14/src/video/quartz/SDL_QuartzGL.m	2009-10-13 01:07:14.000000000 +0200
+++ /Users/mhorn/Projekte/foreign/SDL/SDL-1.2/src/video/quartz/SDL_QuartzGL.m	2009-10-23 14:52:29.000000000 +0200
@@ -153,7 +153,7 @@
      * http://lists.apple.com/archives/mac-opengl/2006/Jan/msg00080.html )
      */
     if ( this->gl_config.swap_control >= 0 ) {
-        long value;
+        GLint value;
         value = this->gl_config.swap_control;
         [ gl_context setValues: &value forParameter: NSOpenGLCPSwapInterval ];
     }
@@ -175,7 +175,7 @@
     #endif
 
     {
-        long cache_max = 64;
+        GLint cache_max = 64;
         CGLContextObj ctx = QZ_GetCGLContextObj(gl_context);
         CGLSetParameter (ctx, GLI_SUBMIT_FUNC_CACHE_MAX, &cache_max);
         CGLSetParameter (ctx, GLI_ARRAY_FUNC_CACHE_MAX, &cache_max);
@@ -261,7 +261,7 @@
         }
         case SDL_GL_ACCELERATED_VISUAL:
         {
-            long val;
+            GLint val;
 	    /* FIXME: How do we get this information here?
             [fmt getValues: &val forAttribute: NSOpenGLPFAAccelerated attr forVirtualScreen: 0];
 	    */
@@ -271,7 +271,7 @@
         }
         case SDL_GL_SWAP_CONTROL:
         {
-            long val;
+            GLint val;
             [ gl_context getValues: &val forParameter: NSOpenGLCPSwapInterval ];
             *value = val;
             return 0;
