diff -Naur teg-old/docs/gnome-help/C/Makefile.in teg-new/docs/gnome-help/C/Makefile.in
--- teg-old/docs/gnome-help/C/Makefile.in	2007-03-14 20:45:39.000000000 -0400
+++ teg-new/docs/gnome-help/C/Makefile.in	2010-10-20 03:37:35.000000000 -0400
@@ -507,10 +507,6 @@
 # Suse Linux 10.0 does not have "db2html", only "docbook2html" which is a script that calls "jw"
 # Check if any of those is available.
 $(docname)/index.html: $(srcdir)/$(docname).sgml
-	-srcdir=`cd $(srcdir) && pwd`; \
-	test ! "x`which db2html`" = "x" && db2html $$srcdir/$(docname).sgml; \
-	test ! "x`which docbook2html`" = "x" && docbook2html $$srcdir/$(docname).sgml; \
-	test ! "x`which jw`" = "x" && jw -f docbook -b html $$srcdir/$(docname).sgml
 
 # remove $(docname) to make installation work
 app-dist-hook: index.html
diff -Naur teg-old/docs/gnome-help/pl/Makefile.in teg-new/docs/gnome-help/pl/Makefile.in
--- teg-old/docs/gnome-help/pl/Makefile.in	2007-03-14 20:45:40.000000000 -0400
+++ teg-new/docs/gnome-help/pl/Makefile.in	2010-10-20 03:37:29.000000000 -0400
@@ -514,16 +514,6 @@
 # Do it in the following steps:
 #	SGML iso8859-2   ->    HTML iso8859-2   -> HTML UTF-8   ->   HTML ASCII with entities
 $(docname)/index.html: $(srcdir)/$(docname).sgml
-	-srcdir=`cd $(srcdir) && pwd`; \
-	test ! "x`which db2html`" = "x" && db2html $$srcdir/$(docname).sgml; \
-	test ! "x`which docbook2html`" = "x" && docbook2html $$srcdir/$(docname).sgml; \
-	test ! "x`which jw`" = "x" && jw -f docbook -b html $$srcdir/$(docname).sgml
-
-	-for file in *.html; do \
-	    iconv -f iso-8859-2 -t utf-8 $$file > utf8_$$file; \
-	    tidy --show-warnings no --quiet yes --input-encoding utf8 utf8_$$file > $$file; \
-	    rm -f utf8_$$file; \
-	done
 
 # removed $(docname) to make thing install
 app-dist-hook: index.html
