--- ./Makefile-orig	2004-12-19 16:12:46.000000000 -0500
+++ ./Makefile	2006-03-21 01:47:34.000000000 -0500
@@ -66,12 +66,12 @@
 install:
 	${INSTALL} -d  $(INSTALL_PREFIX)/bin
-	${INSTALL} -d  $(INSTALL_PREFIX)/man
-	${INSTALL} -d  $(INSTALL_PREFIX)/man/man1
-	${INSTALL} -d  $(INSTALL_PREFIX)/share/vrcon
+	${INSTALL} -d  $(INSTALL_PREFIX)/share/man
+	${INSTALL} -d  $(INSTALL_PREFIX)/share/man/man1
+	${INSTALL} -d  $(INSTALL_PREFIX)/share/doc/vrcon
 	${INSTALL} -c -s -b $(VLOG_EXE) $(INSTALL_PREFIX)/bin
 	${INSTALL} -c -s  -b $(VRCON_EXE) $(INSTALL_PREFIX)/bin
-	${INSTALL} -c -b vrcon.1 $(INSTALL_PREFIX)/man/man1
-	${INSTALL} -c -b vlog.1 $(INSTALL_PREFIX)/man/man1
-	${INSTALL} -c -b vrcon_sample_config $(INSTALL_PREFIX)/share/vrcon
+	${INSTALL} -c -b -m 644 vrcon.1 $(INSTALL_PREFIX)/share/man/man1
+	${INSTALL} -c -b -m 644 vlog.1 $(INSTALL_PREFIX)/share/man/man1
+	${INSTALL} -c -b -m 644 vrcon_sample_config $(INSTALL_PREFIX)/share/doc/vrcon
 
 	@echo

