diff -ruN fink/fink.in fink.test/fink.in
--- fink/fink.in	Mon Apr 15 13:33:34 2002
+++ fink.test/fink.in	Mon Jul  8 20:50:21 2002
@@ -38,8 +38,8 @@
 
 Getopt::Long::Configure(qw(bundling ignore_case require_order));
 GetOptions('yes|y' => \$options{dontask},
-	   'quiet|q' => sub { $options{verbosity}-- },
-	   'verbose|v' => sub { $options{verbosity}++ },
+	   'quiet|q' => sub { $options{verbosity} = -1 },
+	   'verbose|v' => sub { $options{verbosity} = 3 },
 #	   'interactive|i' => \$options{interactive},
 	   'version|V' => \&show_version,
 	   'help|h|?' => \&show_help)
diff -ruN fink/perlmod/Fink/Config.pm fink.test/perlmod/Fink/Config.pm
--- fink/perlmod/Fink/Config.pm	Mon Apr 15 13:33:35 2002
+++ fink.test/perlmod/Fink/Config.pm	Mon Jul  8 21:01:33 2002
@@ -229,17 +229,29 @@
   return $default;
 }
 
-### determine whether verbose mode is on or off
+### determine whether verbose mode is on or off and level -v equals full and 
+### can override
 
 sub is_verbose {
-  my $verbosity;
+  my ($verbosity, $verblevel);
 
+  $verblevel = $config->param("Verbose");
   $verbosity = get_option("verbosity");
-  
-  if ($verbosity != 0) {
-    return ($verbosity>0)
+ 
+  if ($verbosity != -1 && ($verbosity == 3 || $verblevel eq "true" || $verblevel eq "high")) {
+    ### Sets Verbose mode to Full
+    $verbosity = 3;
+  } elsif ($verbosity != -1 && $verblevel eq "medium") {
+    ### Sets Verbose mode to download and tarballs
+    $verbosity = 2;
+  } elsif ($verbosity != -1 && $verblevel eq "low") {
+    ### Sets Verbose mode to download
+    $verbosity = 1;
+  } else {
+    ### Sets Verbose mode to none
+    $verbosity = 0;
   }
-  return $config->param_boolean("Verbose");
+  return $verbosity;
 }
 
 
diff -ruN fink/perlmod/Fink/Configure.pm fink.test/perlmod/Fink/Configure.pm
--- fink/perlmod/Fink/Configure.pm	Mon Apr 15 13:33:35 2002
+++ fink.test/perlmod/Fink/Configure.pm	Mon Jul  8 20:50:21 2002
@@ -65,9 +65,9 @@
   }
 
   $verbose =
-    &prompt_boolean("Always print verbose messages?",
-		    $config->param_boolean("Verbose"));
-  $config->set_param("Verbose", $verbose ? "true" : "false");
+    &prompt("Enter the level to print verbose messages? (none, low, medium, high)",
+		    $config->param_default("Verbose", "false"));
+  $config->set_param("Verbose", $verbose);
 
 
   # proxy settings
diff -ruN fink/perlmod/Fink/Engine.pm fink.test/perlmod/Fink/Engine.pm
--- fink/perlmod/Fink/Engine.pm	Sun Jun  9 11:55:55 2002
+++ fink.test/perlmod/Fink/Engine.pm	Mon Jul  8 20:50:21 2002
@@ -179,10 +179,10 @@
   if (Fink::Config::get_option("dontask")) {
     $cmd .= " --yes";
   }
-  if (Fink::Config::get_option("verbosity")>0) {
+  if (Fink::Config::get_option("verbosity") == 3) {
     $cmd .= " --verbose";
   }
-  elsif (Fink::Config::get_option("verbosity")<0) {
+  elsif (Fink::Config::get_option("verbosity") == -1) {
     $cmd .= " --quiet";
   }
   if (Fink::Config::get_option("interactive")) {
@@ -756,7 +756,7 @@
   my ($answer, $s);
   my (%already_rebuilt, %already_activated);
 
-  if (Fink::Config::is_verbose()) {
+  if (Fink::Config::is_verbose() > -1) {
     $showlist = 1;
   }
 
diff -ruN fink/perlmod/Fink/NetAccess.pm fink.test/perlmod/Fink/NetAccess.pm
--- fink/perlmod/Fink/NetAccess.pm	Mon Apr 15 13:33:35 2002
+++ fink.test/perlmod/Fink/NetAccess.pm	Mon Jul  8 20:50:21 2002
@@ -214,7 +214,7 @@
   # check if we have curl
   if (-x "$basepath/bin/curl" or -x "/usr/bin/curl") {
     $cmd = "curl -f -L";
-    if (not Fink::Config::is_verbose()) {
+    if (Fink::Config::is_verbose() == 0) {
       $cmd .= " -s -S";
     }
     if (not $config->param_boolean("ProxyPassiveFTP")) {
@@ -234,7 +234,7 @@
   if ((!$cmd or $config->param_default("DownloadMethod") eq "wget") and
       (-x "$basepath/bin/wget" or -x "/usr/bin/wget")) {
     $cmd = "wget";
-    if (Fink::Config::is_verbose()) {
+    if (Fink::Config::is_verbose() > 0) {
       $cmd .= " --verbose";
     } else {
       $cmd .= " --non-verbose";
@@ -258,7 +258,7 @@
     if ($config->param_default("DownloadMethod") eq "axelautomirror") {
       $cmd = "axel -S 1";
     }
-    if (Fink::Config::is_verbose()) {
+    if (Fink::Config::is_verbose() > 0) {
       $cmd .= " --verbose";
     }
     if ($file ne &filename($url)) {
diff -ruN fink/perlmod/Fink/PkgVersion.pm fink.test/perlmod/Fink/PkgVersion.pm
--- fink/perlmod/Fink/PkgVersion.pm	Tue May 28 14:36:42 2002
+++ fink.test/perlmod/Fink/PkgVersion.pm	Mon Jul  8 20:50:21 2002
@@ -764,7 +764,7 @@
   my ($pkgname, $package, $version, $pkgversion);
   my ($found, @parts, $i, @vlist, $v, @rlist);
 
-  if (not Fink::Config::is_verbose()) {
+  if (Fink::Config::is_verbose() < 3) {
     $quiet = 1;
   }
 
@@ -921,7 +921,7 @@
   $bdir = $self->get_fullname();
 
   $verbosity = "";
-  if (Fink::Config::is_verbose()) {
+  if (Fink::Config::is_verbose() > 1) {
     $verbosity = "v";
   }
 
diff -ruN fink/perlmod/Fink/SelfUpdate.pm fink.test/perlmod/Fink/SelfUpdate.pm
--- fink/perlmod/Fink/SelfUpdate.pm	Fri Jun  7 19:48:47 2002
+++ fink.test/perlmod/Fink/SelfUpdate.pm	Mon Jul  8 20:50:21 2002
@@ -374,7 +374,7 @@
   }
 
   $verbosity = "";
-  if (Fink::Config::is_verbose()) {
+  if (Fink::Config::is_verbose() > 1) {
     $verbosity = "v";
   }
   $unpack_cmd = "tar -xz${verbosity}f $pkgtarball";
diff -ruN fink/perlmod/Fink/Validation.pm fink.test/perlmod/Fink/Validation.pm
--- fink/perlmod/Fink/Validation.pm	Mon May 27 13:06:29 2002
+++ fink.test/perlmod/Fink/Validation.pm	Mon Jul  8 20:50:21 2002
@@ -171,7 +171,7 @@
   my $looks_good = 1;
   my $error_found = 0;
 
-  if (Fink::Config::is_verbose()) {
+  if (Fink::Config::is_verbose() == 3) {
     print "Validating package file $filename...\n";
   }
   
@@ -292,7 +292,7 @@
     print "Error: Length of package description exceeds 60 characters. ($filename)\n";
     $looks_good = 0;
   }
-  elsif (length($value) > 45 and Fink::Config::is_verbose()) {
+  elsif (length($value) > 45 and Fink::Config::is_verbose() == 3) {
     print "Warning: Length of package description exceeds 45 characters. ($filename)\n";
     $looks_good = 0;
   }
@@ -338,7 +338,7 @@
     }
   }
   
-  if ($looks_good and Fink::Config::is_verbose()) {
+  if ($looks_good and Fink::Config::is_verbose() == 3) {
     print "Package looks good!\n";
   }
 }
