diff -ru fink/perlmod/Fink/PkgVersion.pm fink-thesin/perlmod/Fink/PkgVersion.pm
--- fink/perlmod/Fink/PkgVersion.pm	Tue Feb 12 13:30:52 2002
+++ fink-thesin/perlmod/Fink/PkgVersion.pm	Tue Feb 19 00:07:31 2002
@@ -708,6 +708,8 @@
   my $self = shift;
   my ($archive, $found_archive, $bdir, $destdir, $unpack_cmd);
   my ($i, $verbosity, $answer, $tries);
+  my ($renamefield, @renamefiles, $renamefile, $renamelist, $expand);
+  my ($tar, $bzip2, $unzip);
 
   if ($self->{_type} eq "bundle") {
     return;
@@ -754,16 +756,48 @@
       die "can't find source tarball $archive for package ".$self->get_fullname()."\n";
     }
 
-    # determine unpacking command
+    # Fix to fix multi tarball packages
+    if ($i < 2) {
+      $renamefield = "TarFilesRename";
+    } else {
+      $renamefield = "Tar".$i."FilesRename";
+    }
+
+    # To fix invalid tar files
+    $renamelist = "";
+
+    if ($self->has_param($renamefield)) {
+      @renamefiles = split(/\s+/, $self->param($renamefield));
+      foreach $renamefile (@renamefiles) {
+        $renamefile = &expand_percent($renamefile, $expand);
+        if ($renamefile =~ /^(.+)\:(.+)$/) {
+          $renamelist .= " -s ,$1,$2,";
+        } else {
+          $renamelist .= " -s ,${renamefile},${renamefile}_tmp,";
+        }
+      }
+    }
+
+    # Get compression commands
+    if ($self->has_param($renamefield)) {
+      $tar = "/usr/bin/tar"; # Needs to use Darwin tar not GNU
+    } else {
+      $tar = "$basepath/bin/tar"; # Needs to use Darwin tar not GNU
+    }
+    $bzip2 = "bzip2";
+    $unzip = "unzip";
+
+    # Determine unpack command
     $unpack_cmd = "cp $found_archive .";
+
     if ($archive =~ /[\.\-]tar\.(gz|z|Z)$/ or $archive =~ /\.tgz$/) {
-      $unpack_cmd = "tar -x${verbosity}zf $found_archive";
+      $unpack_cmd = "$tar -x${verbosity}zf $found_archive $renamelist";
     } elsif ($archive =~ /[\.\-]tar\.bz2$/) {
-      $unpack_cmd = "bzip2 -dc $found_archive | tar -x${verbosity}f -";
+      $unpack_cmd = "$bzip2 -dc $found_archive | $tar -x${verbosity}f $renamelist -";
     } elsif ($archive =~ /[\.\-]tar$/) {
-      $unpack_cmd = "tar -x${verbosity}f $found_archive";
+      $unpack_cmd = "$tar -x${verbosity}f $found_archive $renamelist";
     } elsif ($archive =~ /\.zip$/) {
-      $unpack_cmd = "unzip -o $found_archive";
+      $unpack_cmd = "$unzip -o $found_archive";
     }
 
     # calculate destination directory
