diff -ruN kdemultimedia-3.0.1.orig/admin/Makefile.common kdemultimedia-3.0.1/admin/Makefile.common
--- kdemultimedia-3.0.1.orig/admin/Makefile.common	Wed May  8 12:28:50 2002
+++ kdemultimedia-3.0.1/admin/Makefile.common	Tue Jun 11 00:10:45 2002
@@ -1,24 +1,12 @@
 
 SHELL=/bin/sh
 
+ACLOCAL=aclocal-1.5
+AUTOMAKE=automake-1.5
+AUTOCONF=autoconf
+
 cvs:
-	if test -x /usr/bin/autoconf2.50 ; then \
-	   AUTOCONF="/usr/bin/autoconf2.50" ;\
-	else \
-	   AUTOCONF="autoconf" ;\
-	fi; \
-	AUTOCONF_VERSION=`$$AUTOCONF --version` ;\
-	if test "x$$AUTOCONF_VERSION" = "xAutoconf version 2.13"; then \
-	   echo "*** YOU'RE USING AUTOCONF 2.13. We suggest updating." ;\
-	   echo "*** autoconf 2.5x works best for now, we will drop" ;\
-	   echo "*** support for autoconf 2.13 soon."; \
-	   test ! -f admin/libtool.m4.in || mv admin/libtool.m4.in admin/new-libtool.m4.in ;\
-	   test ! -f admin/ltmain.sh || mv admin/ltmain.sh admin/new-ltmain.sh ;\
-	   for file in libtool.m4.in ltcf-c.sh ltcf-cxx.sh ltcf-gcj.sh ltconfig ltmain.sh; do \
-		rm -f admin/$$file ;\
-		ln -s old-$$file admin/$$file ;\
-	   done ;\
-	fi 
+	@chmod a+x admin/*
 	@if grep '$$(top_srcdir)/acinclude.m4:' Makefile.am >/dev/null; then \
 	  echo "*** Creating acinclude.m4" ;\
 	  rm -f acinclude.m4 configure.files ;\
@@ -34,7 +22,7 @@
 	  $(MAKE) -f Makefile.am top_srcdir=. configure.in ;\
 	fi
 	@echo "*** Creating aclocal.m4"
-	@aclocal
+	@$(ACLOCAL)
 	@echo "*** Creating configure"
 	@autoconf
 	@if test -r configure.in.in ; then \
@@ -52,7 +40,7 @@
 		autoheader ;\
 	fi
 	@echo "*** Creating Makefile templates"
-	@automake
+	@$(AUTOMAKE)
 	@echo "*** Postprocessing Makefile templates"
 	@perl admin/am_edit
 	@if egrep "^cvs-local:" Makefile.am >/dev/null; then \
@@ -71,9 +59,9 @@
 	@if test -r configure.in.in; then \
 	  $(MAKE) -f Makefile.am top_srcdir=. subdirs configure.in; \
 	fi ;\
-	aclocal; \
+	$(ACLOCAL); \
 	autoheader; \
-	automake --foreign --include-deps; \
+	$(AUTOMAKE) --foreign --include-deps; \
 	perl admin/am_edit; \
 	autoconf; \
 	if test -r configure.in.in ; then \
@@ -281,7 +269,9 @@
 	done ;\
 	rm -f _SUBDIRS ;\
 	for i in $$dirs; do \
+	if [ $$i != "kdm" ]; then \
 	echo $$i >> ./_SUBDIRS; \
+        fi \
 	done
 	@if test -r subdirs && diff subdirs _SUBDIRS > /dev/null; then \
 	  rm -f _SUBDIRS; \
diff -ruN kdemultimedia-3.0.1.orig/admin/acinclude.m4.in kdemultimedia-3.0.1/admin/acinclude.m4.in
--- kdemultimedia-3.0.1.orig/admin/acinclude.m4.in	Wed May  8 12:28:50 2002
+++ kdemultimedia-3.0.1/admin/acinclude.m4.in	Tue Jun 11 00:11:14 2002
@@ -693,6 +693,19 @@
    AC_CHECK_FUNC(shmat, ,
      AC_CHECK_LIB(ipc, shmat, X_EXTRA_LIBS="$X_EXTRA_LIBS -lipc"))
    
+   # darwin needs this to initialize the environment
+   AC_CHECK_HEADERS([crt_externs.h sys/types.h])
+   AC_CHECK_FUNC(_NSGetEnviron, [AC_DEFINE(HAVE_NSGETENVIRON, 1, [Define if your system needs _NSGetEnviron to set up the environment])])
+
+   # darwin requires a poll.h emulation library
+   AC_CHECK_LIB(poll, poll, LIB_POLL="-lpoll")
+
+   # CoreAudio framework
+   AC_CHECK_HEADER(CoreAudio/CoreAudio.h, [
+     AC_DEFINE(HAVE_COREAUDIO, 1, [Define if you have the CoreAudio API])
+     FRAMEWORK_COREAUDIO="-framework CoreAudio"
+   ])
+
    # Solaris 2.6 and others need -lresolv for res_init
    AC_CHECK_FUNCS(res_init, , [
      kde_libs_safe="$LIBS"
@@ -713,6 +726,8 @@
 
    LIBSOCKET="$X_EXTRA_LIBS"
    AC_SUBST(LIBSOCKET)
+   AC_SUBST(FRAMEWORK_COREAUDIO)
+   AC_SUBST(LIB_POLL)
    AC_SUBST(LIBRESOLV)
    AC_SUBST(X_EXTRA_LIBS)
    AC_CHECK_LIB(ucb, killpg, [LIBUCB="-lucb"]) dnl for Solaris2.4
@@ -1826,6 +1841,32 @@
 if test $kde_qtver = 3; then
    LIB_KDECORE='-lkdecore'
    AC_SUBST(LIB_KDECORE)
+   LIB_KDEFX='-lkdefx'
+   AC_SUBST(LIB_KDEFX)
+   LIB_ARTSKDE='-lartskde'
+   AC_SUBST(LIB_ARTSKDE)
+   LIB_KABC='-lkabc'
+   AC_SUBST(LIB_KABC)
+   LIB_KDESASL='-lkdesasl'
+   AC_SUBST(LIB_KDESASL)
+   LIB_KDESU='-lkdesu'
+   AC_SUBST(LIB_KDESU)
+   LIB_KJAVA='-lkjava'
+   AC_SUBST(LIB_KJAVA)
+   LIB_KJS='-lkjs'
+   AC_SUBST(LIB_KJS)
+   LIB_KMID='-lkmid'
+   AC_SUBST(LIB_KMID)
+   LIB_KSCRIPT='-lkscript'
+   AC_SUBST(LIB_KSCRIPT)
+   LIB_KTEXTEDITOR='-lktexteditor'
+   AC_SUBST(LIB_KTEXTEDITOR)
+   LIB_SHELLSCRIPT='-lshellscript'
+   AC_SUBST(LIB_SHELLSCRIPT)
+   LIB_VCARD='-lvcard'
+   AC_SUBST(LIB_VCARD)
+   LIB_DCOP='-lDCOP'
+   AC_SUBST(LIB_DCOP)
    LIB_KDEUI='-lkdeui'
    AC_SUBST(LIB_KDEUI)
    LIB_KIO='-lkio'
@@ -1842,6 +1883,8 @@
    AC_SUBST(LIB_KPARTS)
    LIB_KDEPRINT='-lkdeprint'
    AC_SUBST(LIB_KDEPRINT)
+   LIB_KDED='@PREFIX@/lib/kded.la'
+   AC_SUBST(LIB_KDED)
 # these are for backward compatibility
    LIB_KSYCOCA='-lkio'
    AC_SUBST(LIB_KSYCOCA)
@@ -4158,7 +4201,7 @@
 AC_DEFUN(KDE_CREATE_SUBDIRSLIST,
 [
 
-DO_NOT_COMPILE="$DO_NOT_COMPILE CVS debian bsd-port admin"
+DO_NOT_COMPILE="$DO_NOT_COMPILE CVS debian bsd-port admin kdm ksysguard kscd"
 
 if test ! -s $srcdir/subdirs; then
   dnl Note: Makefile.common creates subdirs, so this is just a fallback
@@ -4530,9 +4573,12 @@
            [initgroups may exist but not its prototype (e.g. AIX<4.3.3:8)])
 ])
 
-
 AC_DEFUN(KDE_CHECK_JAVA_DIR,
 [
+
+# needed by the MacOS X JVM in certain circumstances
+AC_CHECK_HEADERS([TargetConditionals.h])
+
 AC_MSG_CHECKING([for Java directory])
 
 AC_ARG_WITH(java,
@@ -4541,6 +4587,7 @@
 ], ac_java_dir=""
 )
 
+
 dnl at this point ac_java_dir is either a dir, 'no' to disable, or '' to say look in $PATH
 if test "x$ac_java_dir" = "xno"; then
    kde_cv_java_bindir=no
@@ -4572,10 +4619,20 @@
 dnl At this point kde_cv_java_bindir and kde_cv_java_includedir are either set or "no"
 if test "x$kde_cv_java_bindir" != "xno"; then
 
-  dnl Look for libjvm.so
-  kde_java_libjvmdir=`find $kde_cv_java_bindir/.. -name libjvm.so | sed 's,libjvm.so,,'|head -n 1`
-  dnl Look for libhpi.so and avoid green threads
-  kde_java_libhpidir=`find $kde_cv_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,'`
+  case $host_os in
+    darwin*)
+      dnl Look for libjvm.dylib
+      kde_java_libjvmdir=`find /System/Library/Frameworks/JavaVM.framework/ -name libjvm.dylib | sed 's,libjvm.dylib,,'|head -n 1`
+      kd_java_libhpidir=''
+      dnl Darwin doesn't have libhpi
+      ;;
+    *)
+      dnl Look for libjvm.so
+      kde_java_libjvmdir=`find $kde_cv_java_bindir/.. -name libjvm.so | sed 's,libjvm.so,,'|head -n 1`
+      dnl Look for libhpi.so and avoid green threads
+      kde_java_libhpidir=`find $kde_cv_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,'`
+      ;;
+  esac
 
   dnl Now check everything's fine under there
 
@@ -4591,12 +4648,26 @@
   if test ! -r "$kde_cv_java_includedir/jni.h"; then
     AC_MSG_ERROR([jni.h not found under $kde_cv_java_includedir. Use --with-java or --without-java.])
   fi
-  if test ! -r "$kde_java_libjvmdir/libjvm.so"; then
-    AC_MSG_ERROR([libjvm.so not found under $kde_java_libjvmdir. Use --without-java.])
-  fi
-  if test ! -r "$kde_java_libhpidir/libhpi.so"; then
-    AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
-  fi
+  case $host_os in
+    darwin*)
+      if test ! -r "$kde_java_libjvmdir/libjvm.dylib"; then
+        AC_MSG_ERROR([libjvm.dylib not found under $kde_java_libjvmdir. Use --without-java.])
+      fi
+      dnl Darwin doesn't have libhpi, but it's not fatal
+      LIB_JVM="-framework JavaVM"
+      LIB_HPI=""
+      ;;
+    *)
+      if test ! -r "$kde_java_libjvmdir/libjvm.so"; then
+        AC_MSG_ERROR([libjvm.so not found under $kde_java_libjvmdir. Use --without-java.])
+      fi
+      LIB_JVM="-L$kde_java_libjvmdir -ljvm"
+      if test ! -r "$kde_java_libhpidir/libhpi.so"; then
+        AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
+      fi
+      LIB_HPI="-L$kde_java_libhpidir -lhpi"
+      ;;
+  esac
 
   jni_includes="-I$kde_cv_java_includedir"
   dnl Strange thing, jni.h requires jni_md.h which is under genunix here..
@@ -4643,7 +4714,7 @@
   JAR=$kde_cv_java_bindir/jar
   AC_SUBST(JAR)
   AC_SUBST(jni_includes)
-  JVMLIBS="-L$kde_java_libjvmdir -ljvm -L$kde_java_libhpidir -lhpi"
+  JVMLIBS="$LIB_JVM $LIB_HPI"
   AC_SUBST(JVMLIBS)
 fi
 ])
diff -ruN kdemultimedia-3.0.1.orig/admin/libtool.m4.in kdemultimedia-3.0.1/admin/libtool.m4.in
--- kdemultimedia-3.0.1.orig/admin/libtool.m4.in	Wed May  8 12:28:50 2002
+++ kdemultimedia-3.0.1/admin/libtool.m4.in	Tue Jun 11 00:10:45 2002
@@ -593,6 +593,12 @@
     lt_cv_sys_max_cmd_len=-1;
     ;;
 
+  darwin*)
+    # this should be enough... the length test freaks out some versions
+    # of OSX's `expr`
+    lt_cv_sys_max_cmd_len=16384;
+    ;;
+
   *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -1876,16 +1882,7 @@
   ;;
 
 darwin* | rhapsody*)
-  lt_cv_deplibs_check_method='file_magic Mach-O dynamically linked shared library'
-  lt_cv_file_magic_cmd='/usr/bin/file -L'
-  case "$host_os" in
-  rhapsody* | darwin1.[[012]])
-    lt_cv_file_magic_test_file=`/System/Library/Frameworks/System.framework/System`
-    ;;
-  *) # Darwin 1.3 on
-    lt_cv_file_magic_test_file='/usr/lib/libSystem.dylib'
-    ;;
-  esac
+  lt_cv_deplibs_check_method='pass_all'
   ;;
 
 freebsd*)
@@ -2535,6 +2532,11 @@
 	;;
     esac
     ;;
+  darwin*)
+    shared_flag='-dynamiclib'
+    _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC -dynamiclib -install_name $rpath/$soname $predep_objects ${lib}-master.o $deplibs $postdep_objects $compiler_flags -o $lib'
+    output_verbose_link_cmd='$CC -dynamiclib $CFLAGS -v conftest.$objext 2>&1 | egrep "\-L"'
+    ;;
   dgux*)
     case $cc_basename in
       ec++)
@@ -4725,7 +4727,9 @@
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+	#_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+        # testing with twolevel namespaces
+	_LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
 	;;
       esac
 
@@ -4733,7 +4737,7 @@
       #        cross-compilation, but unfortunately the echo tests do not
       #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
       #	       `"' quotes if we put them in here... so don't!
-      _LT_AC_TAGVAR(archive_cmds, $1)='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+      _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) -install_name $rpath/$soname $allow_undefined_flag -o $lib ${lib}-master.o $deplibs $linker_flags $verstring'
       # We need to add '_' to the symbols in $export_symbols first
       #_LT_AC_TAGVAR(archive_expsym_cmds, $1)="$_LT_AC_TAGVAR(archive_cmds, $1)"' && strip -s $export_symbols'
       _LT_AC_TAGVAR(hardcode_direct, $1)=yes
diff -ruN kdemultimedia-3.0.1.orig/admin/ltmain.sh kdemultimedia-3.0.1/admin/ltmain.sh
--- kdemultimedia-3.0.1.orig/admin/ltmain.sh	Wed May  8 12:28:51 2002
+++ kdemultimedia-3.0.1/admin/ltmain.sh	Tue Jun 11 00:10:45 2002
@@ -3324,7 +3324,18 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+ 	  save_deplibs="$deplibs"
+ 	  for conv in $convenience; do
+            tmp_deplibs=
+            for test_deplib in $deplibs; do
+              if test "$test_deplib" != "$conv"; then
+                tmp_deplibs="$tmp_deplibs $test_deplib"
+              fi
+            done
+            deplibs="$tmp_deplibs"
+ 	  done
 	  eval cmds=\"$archive_cmds\"
+ 	  deplibs="$save_deplibs"
 	fi
 
 	if len=`expr "X$cmds" : ".*"` &&
@@ -3635,6 +3646,14 @@
 	compile_deplibs=`$echo "X $compile_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	finalize_deplibs=`$echo "X $finalize_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	;;
+      esac
+
+      case $host in
+      *darwin*)
+        # Don't allow lazy linking, it breaks C++ global constructors
+        compile_command="$compile_command ${wl}-bind_at_load"
+        finalize_command="$finalize_command ${wl}-bind_at_load"
+        ;;
       esac
 
       compile_command="$compile_command $compile_deplibs"
diff -ruN kdemultimedia-3.0.1.orig/admin/missing kdemultimedia-3.0.1/admin/missing
--- kdemultimedia-3.0.1.orig/admin/missing	Thu Oct  4 18:46:36 2001
+++ kdemultimedia-3.0.1/admin/missing	Tue Jun 11 00:10:45 2002
@@ -87,7 +87,7 @@
     exit 1
     ;;
 
-  aclocal)
+  aclocal*)
     echo 1>&2 "\
 WARNING: \`$1' is missing on your system.  You should only need it if
          you modified \`acinclude.m4' or \`${configure_ac}'.  You might want
@@ -124,7 +124,7 @@
     touch $touch_files
     ;;
 
-  automake)
+  automake*)
     echo 1>&2 "\
 WARNING: \`$1' is missing on your system.  You should only need it if
          you modified \`Makefile.am', \`acinclude.m4' or \`${configure_ac}'.
diff -ruN kdemultimedia-3.0.1.orig/kmix/Makefile.am kdemultimedia-3.0.1/kmix/Makefile.am
--- kdemultimedia-3.0.1.orig/kmix/Makefile.am	Wed May  8 12:29:16 2002
+++ kdemultimedia-3.0.1/kmix/Makefile.am	Tue Jun 11 00:10:45 2002
@@ -15,7 +15,8 @@
 kmix_la_SOURCES = main.cpp kmix.cpp kmixdockwidget.cpp kmixprefdlg.cpp \
 	kmixerwidget.cpp channel.cpp mixer.cpp ksmallslider.cpp \
 	volume.cpp kledbutton.cpp mixdevicewidget.cpp
-kmix_la_LIBADD = $(LIB_KDEUI) $(LIBALIB) $(LIBOSSAUDIO) $(LIBASOUND)
+kmix_la_LIBADD = $(LIB_KDEUI) $(LIBALIB) $(LIBOSSAUDIO) $(LIBASOUND) $(LIB_QT) \
+	$(LIB_KDECORE) $(LIB_DCOP)
 kmix_la_LDFLAGS = $(all_libraries) -module -avoid-version
 
 kmix_SOURCES = dummy.cpp
@@ -27,7 +28,8 @@
 kmixctrl_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
 kmixctrl_la_SOURCES = kmixctrl.cpp channel.cpp mixer.cpp volume.cpp 
-kmixctrl_la_LIBADD = $(LIB_KDEUI) $(LIBALIB) $(LIBOSSAUDIO) $(LIBASOUND)
+kmixctrl_la_LIBADD = $(LIB_KDEUI) $(LIBALIB) $(LIBOSSAUDIO) $(LIBASOUND) \
+	$(LIB_QT) $(LIB_KDECORE)
 kmixctrl_la_LDFLAGS = $(all_libraries) -module -avoid-version
 
 kmix_panelapplet_la_METASOURCES=AUTO
diff -ruN kdemultimedia-3.0.1.orig/mpg123_artsplugin/mpg123PlayObject_impl.cpp kdemultimedia-3.0.1/mpg123_artsplugin/mpg123PlayObject_impl.cpp
--- kdemultimedia-3.0.1.orig/mpg123_artsplugin/mpg123PlayObject_impl.cpp	Thu Aug 23 11:21:36 2001
+++ kdemultimedia-3.0.1/mpg123_artsplugin/mpg123PlayObject_impl.cpp	Tue Jun 11 00:13:29 2002
@@ -22,7 +22,11 @@
 #include <config.h>
 #include <sys/types.h>
 #include <sys/ipc.h>
+#ifdef NOW_USING_SEMAPHORES
+#ifdef HAVE_SYS_SEM_H
 #include <sys/sem.h>
+#endif
+#endif 
 #include <sys/stat.h>
 #include <sys/shm.h>
 #include <sys/wait.h>
diff -ruN kdemultimedia-3.0.1.orig/oggvorbis_artsplugin/oggPlayObject_impl.cpp kdemultimedia-3.0.1/oggvorbis_artsplugin/oggPlayObject_impl.cpp
--- kdemultimedia-3.0.1.orig/oggvorbis_artsplugin/oggPlayObject_impl.cpp	Fri Dec 14 13:54:43 2001
+++ kdemultimedia-3.0.1/oggvorbis_artsplugin/oggPlayObject_impl.cpp	Tue Jun 11 00:13:54 2002
@@ -2,7 +2,11 @@
 
 #include <sys/types.h>
 #include <sys/ipc.h>
+#ifdef NOW_USING_SEMAPHORES
+#ifdef HAVE_SYS_SEM_H
 #include <sys/sem.h>
+#endif
+#endif 
 #include <sys/stat.h>
 #include <sys/shm.h>
 #include <sys/wait.h>
