diff -ruN MPlayer-0.90pre8.orig/Makefile MPlayer-0.90pre8/Makefile
--- MPlayer-0.90pre8.orig/Makefile	2002-09-09 05:47:54.000000000 -0600
+++ MPlayer-0.90pre8/Makefile	2002-09-19 15:51:15.000000000 -0600
@@ -193,36 +193,36 @@
 ifeq ($(VIDIX),yes)
 	$(DO_MAKE)
 endif
-	if test ! -d $(BINDIR) ; then mkdir -p $(BINDIR) ; fi
-	$(INSTALL) -m 755 $(INSTALLSTRIP) $(PRG) $(BINDIR)/$(PRG)
+	if test ! -d $(DESTDIR)$(BINDIR) ; then mkdir -p $(DESTDIR)$(BINDIR) ; fi
+	$(INSTALL) -m 755 $(INSTALLSTRIP) $(PRG) $(DESTDIR)$(BINDIR)/$(PRG)
 ifeq ($(GUI),yes)
-	-ln -sf $(PRG) $(BINDIR)/gmplayer
+	-ln -sf $(BINDIR)/$(PRG) $(DESTDIR)$(BINDIR)/gmplayer
 endif
-	if test ! -d $(MANDIR)/man1 ; then mkdir -p $(MANDIR)/man1; fi
-	$(INSTALL) -c -m 644 DOCS/mplayer.1 $(MANDIR)/man1/mplayer.1
+	if test ! -d $(DESTDIR)$(MANDIR)/man1 ; then mkdir -p $(DESTDIR)$(MANDIR)/man1; fi
+	$(INSTALL) -c -m 644 DOCS/mplayer.1 $(DESTDIR)$(MANDIR)/man1/mplayer.1
 ifeq ($(MENCODER),yes)
-	$(INSTALL) -m 755 $(INSTALLSTRIP) $(PRG_MENCODER) $(BINDIR)/$(PRG_MENCODER)
-	-ln -sf mplayer.1 $(MANDIR)/man1/mencoder.1
+	$(INSTALL) -m 755 $(INSTALLSTRIP) $(PRG_MENCODER) $(DESTDIR)$(BINDIR)/$(PRG_MENCODER)
+	-ln -sf $(MANDIR)/man1/mplayer.1 $(DESTDIR)$(MANDIR)/man1/mencoder.1
 endif
-	@if test ! -d $(DATADIR) ; then mkdir -p $(DATADIR) ; fi
-	@if test ! -d $(DATADIR)/font ; then mkdir -p $(DATADIR)/font ; fi
-	@if test ! -f $(DATADIR)/font/font.desc ; then \
+	@if test ! -d $(DESTDIR)$(DATADIR) ; then mkdir -p $(DESTDIR)$(DATADIR) ; fi
+	@if test ! -d $(DESTDIR)$(DATADIR)/font ; then mkdir -p $(DESTDIR)$(DATADIR)/font ; fi
+	@if test ! -f $(DESTDIR)$(DATADIR)/font/font.desc ; then \
 	echo "*** Download font at http://www.mplayerhq.hu/homepage/dload.html" ; \
 	echo "*** for OSD/Subtitles support and extract to $(DATADIR)/font/" ; \
 	fi
 ifeq ($(GUI),yes)
-	@if test ! -d $(DATADIR)/Skin ; then mkdir -p $(DATADIR)/Skin ; fi
+	@if test ! -d $(DESTDIR)$(DATADIR)/Skin ; then mkdir -p $(DESTDIR)$(DATADIR)/Skin ; fi
 	@echo "*** Download skin(s) at http://www.mplayerhq.hu/homepage/dload.html"
 	@echo "*** for GUI, and extract to $(DATADIR)/Skin/"
 endif
-	@if test ! -d $(CONFDIR) ; then mkdir -p $(CONFDIR) ; fi
-	@if test -f $(CONFDIR)/codecs.conf.old ; then mv -f $(CONFDIR)/codecs.conf.old $(CONFDIR)/codecs.conf.older ; fi
-	@if test -f $(CONFDIR)/codecs.conf ; then mv -f $(CONFDIR)/codecs.conf $(CONFDIR)/codecs.conf.old ; fi
-	$(INSTALL) -c -m 644 etc/codecs.conf $(CONFDIR)/codecs.conf
+	@if test ! -d $(DESTDIR)$(CONFDIR) ; then mkdir -p $(DESTDIR)$(CONFDIR) ; fi
+	@if test -f $(DESTDIR)$(CONFDIR)/codecs.conf.old ; then mv -f $(DESTDIR)$(CONFDIR)/codecs.conf.old $(CONFDIR)/codecs.conf.older ; fi
+	@if test -f $(DESTDIR)$(CONFDIR)/codecs.conf ; then mv -f $(DESTDIR)$(CONFDIR)/codecs.conf $(CONFDIR)/codecs.conf.old ; fi
+	$(INSTALL) -c -m 644 etc/codecs.conf $(DESTDIR)$(CONFDIR)/codecs.conf
 ifeq ($(DVDKIT),yes)
 ifeq ($(DVDKIT_SHARED),yes)
-	if test ! -d $(LIBDIR) ; then mkdir -p $(LIBDIR) ; fi
-	$(INSTALL) -m 755 $(INSTALLSTRIP) libmpdvdkit2/libmpdvdkit.so $(LIBDIR)/libmpdvdkit.so
+	if test ! -d $(DESTDIR)$(LIBDIR) ; then mkdir -p $(DESTDIR)$(LIBDIR) ; fi
+	$(INSTALL) -m 755 $(INSTALLSTRIP) libmpdvdkit2/libmpdvdkit.so $(DESTDIR)$(LIBDIR)/libmpdvdkit.so
 endif
 endif
 ifeq ($(CSS_USE),yes)
@@ -230,7 +230,7 @@
 	@echo "however it means you can't use fibmap_mplayer."
 	@echo "Without this (or without running mplayer as root) you won't be"
 	@echo "able to play encrypted DVDs."
-	-$(INSTALL) -o 0 -g 0 -m 4755 $(INSTALLSTRIP) $(PRG_FIBMAP) $(BINDIR)/$(PRG_FIBMAP)
+	-$(INSTALL) -o 0 -g 0 -m 4755 $(INSTALLSTRIP) $(PRG_FIBMAP) $(DESTDIR)$(BINDIR)/$(PRG_FIBMAP)
 endif
 
 uninstall:
diff -ruN MPlayer-0.90pre8.orig/configure MPlayer-0.90pre8/configure
--- MPlayer-0.90pre8.orig/configure	2002-09-17 13:47:55.000000000 -0600
+++ MPlayer-0.90pre8/configure	2002-09-19 13:34:55.000000000 -0600
@@ -29,13 +29,13 @@
   echo >> "$TMPLOG"
   cat "$TMPC" >> "$TMPLOG"
   echo >> "$TMPLOG"
-  echo "$_cc $_inc_extra $_ld_static $_ld_extra $TMPC -o $TMPO $@" >> "$TMPLOG"
+  echo "$_cc $CFLAGS $_inc_extra $_ld_static $_ld_extra $TMPC -o $TMPO $@" >> "$TMPLOG"
   rm -f "$TMPO"
-  ( "$_cc" $_inc_extra $_ld_static $_ld_extra "$TMPC" -o "$TMPO" "$@" ) >> "$TMPLOG" 2>&1
+  ( "$_cc" $CFLAGS $_inc_extra $_ld_static $_ld_extra "$TMPC" -o "$TMPO" "$@" ) >> "$TMPLOG" 2>&1
   TMP="$?"
   echo >> "$TMPLOG"
-  echo "ldd $TMPO" >> "$TMPLOG"
-  ( ldd "$TMPO" ) >> "$TMPLOG" 2>&1
+  echo "nm $TMPO" >> "$TMPLOG"
+  ( nm "$TMPO" ) >> "$TMPLOG" 2>&1
   echo >> "$TMPLOG"
   return "$TMP"
 }
@@ -1578,6 +1578,10 @@
 cc_check -lsocket -lnsl && _ld_sock="-lsocket -lnsl"
 if test "$_ld_sock" ; then
   echores "yes (using $_ld_sock)"
+elif cc_check $_ld_sock -ldnet ; then
+  # NOTE: needed for Darwin at least
+  _ld_sock="$_ld_sock -ldnet"
+  echores "yes (using $_ld_sock)"
 else
   echores "no"
 fi
@@ -2015,6 +2019,16 @@
 else
   _def_shm='#undef HAVE_SHM'
 fi
+# Darwin Xfree86 has x-shm disabled thus shm has to be disabled
+if darwin ; then
+  if test `uname -r` > 5.9 ; then
+    _def_shm='#define HAVE_SHM 1'
+    _shm='yes (Darwin 6+/OSX 10.2+ supports SHM)'
+  else
+    _def_shm='#undef HAVE_SHM'
+    _shm='no (not supported by Darwin)'
+  fi
+fi
 echores "$_shm"
 
 
diff -ruN MPlayer-0.90pre8.orig/etc/example.conf MPlayer-0.90pre8/etc/example.conf
--- MPlayer-0.90pre8.orig/etc/example.conf	2002-05-04 22:46:09.000000000 -0600
+++ MPlayer-0.90pre8/etc/example.conf	2002-09-19 14:04:30.000000000 -0600
@@ -8,13 +8,13 @@
 ## more flexibly here. See below.
 ##
 
-# vo=xv			# To specify default video driver (see -vo help for
+vo=sdl			# To specify default video driver (see -vo help for
 			# list)
 
-# ao=oss		# To specify default audio driver (see -ao help for
+ao=sdl			# To specify default audio driver (see -ao help for
 			# list)
 
-fs=yes			# Enlarges movie window to your desktop's size.
+fs=no			# Enlarges movie window to your desktop's size.
 			# Used by drivers: all
 
 # fsmode=0		# This option workarounds some problems when using
@@ -48,7 +48,7 @@
 ## Use GUI mode by default
 ##
 
-# gui = yes
+gui = yes
 
 ##
 ## Specify your preferred default skin here
@@ -56,7 +56,7 @@
 ##  and ~/.mplayer/Skin/yourskin)
 ##
 
-# skin = default
+skin = default
 
 ##
 ## Multiple languages are available :)
@@ -145,4 +145,4 @@
 ## Delete this default :)
 ##
 
-include = /home/gabucino/.mplayer/i_did_not_RTFM_carefully_enough...
+#include = /home/gabucino/.mplayer/i_did_not_RTFM_carefully_enough...
