diff -ruN MPlayer-0.90pre6.orig/Makefile MPlayer-0.90pre6/Makefile
--- MPlayer-0.90pre6.orig/Makefile	2002-07-24 21:37:27.000000000 -0600
+++ MPlayer-0.90pre6/Makefile	2002-08-13 10:25:10.000000000 -0600
@@ -26,7 +26,11 @@
 #prefix = /usr/local
 BINDIR = ${prefix}/bin
 # BINDIR = /usr/local/bin
+ifeq ($(TARGET_OS), Darwin)
+MANDIR = ${prefix}/share/man
+else
 MANDIR = ${prefix}/man
+endif
 
 # a BSD compatible 'install' program
 INSTALL = install
diff -ruN MPlayer-0.90pre6.orig/configure MPlayer-0.90pre6/configure
--- MPlayer-0.90pre6.orig/configure	2002-08-05 11:23:22.000000000 -0600
+++ MPlayer-0.90pre6/configure	2002-08-13 11:54:14.000000000 -0600
@@ -30,13 +30,13 @@
   echo >> "$TMPLOG"
   cat "$TMPC" >> "$TMPLOG"
   echo >> "$TMPLOG"
-  echo "$_cc $_inc_extra $_ld_static $_ld_extra $TMPC -o $TMPO $@" >> "$TMPLOG"
+  echo "$_cc $CFLAGS $_inc_extra $_ld_static $_ld_extra $TMPC -o $TMPO $@" >> "$TMPLOG"
   rm -f "$TMPO"
-  ( "$_cc" $_inc_extra $_ld_static $_ld_extra "$TMPC" -o "$TMPO" "$@" ) >> "$TMPLOG" 2>&1
+  ( "$_cc" $CFLAGS $_inc_extra $_ld_static $_ld_extra "$TMPC" -o "$TMPO" "$@" ) >> "$TMPLOG" 2>&1
   TMP="$?"
   echo >> "$TMPLOG"
-  echo "ldd $TMPO" >> "$TMPLOG"
-  ( ldd "$TMPO" ) >> "$TMPLOG" 2>&1
+  echo "nm $TMPO" >> "$TMPLOG"
+  ( nm "$TMPO" ) >> "$TMPLOG" 2>&1
   echo >> "$TMPLOG"
   return "$TMP"
 }
@@ -1533,6 +1533,10 @@
   # NOTE: needed for SunOS at least
   _ld_sock="$_ld_sock -lresolv"
   echores "yes (using $_ld_sock)"
+elif cc_check $_ld_sock -ldnet ; then
+  # NOTE: needed for Darwin at least
+  _ld_sock="$_ld_sock -ldnet"
+  echores "yes (using $_ld_sock)"
 else
   echores "no (=> streaming support disabled)"
   _streaming=no
@@ -1900,6 +1904,16 @@
 else
   _def_shm='#undef HAVE_SHM'
 fi
+# Darwin Xfree86 has x-shm disabled thus shm has to be disabled
+if darwin ; then
+  if test `uname -r` = 6.0 ; then
+    _def_shm='#define HAVE_SHM 1'    
+    _shm='yes (Darwin 6.0/OSX 10.2 supports SHM)' 
+  else
+    _def_shm='#undef HAVE_SHM'
+    _shm='no (not supported by Darwin)'
+  fi
+fi
 echores "$_shm"
 
 
@@ -3452,7 +3466,7 @@
   _iconv=no
   if freebsd ; then
     cc_check  -lm -lgiconv && _iconv=yes
-  elif bsdos || openbsd ; then
+  elif bsdos || openbsd || darwin ; then
     cc_check  -lm -liconv && _iconv=yes
   else
     cc_check -lm && _iconv=yes
@@ -3463,6 +3477,7 @@
   freebsd && _ld_iconv='-lgiconv'
   bsdos && _ld_iconv='-liconv'
   openbsd && _ld_iconv='-liconv'
+  darwin && _ld_iconv='-liconv'
 else
   _def_iconv='#undef USE_ICONV'
 fi
@@ -3580,7 +3595,7 @@
   _def_xvid='#define HAVE_XVID 1'
   _def_divx4_h='#define HAVE_DIVX4_H 1'
   _codecmodules="xvid $_codecmodules"
-else
+else      
   _xvid=no
   _ld_xvid=''
   _def_xvid='#undef HAVE_XVID'
@@ -3613,7 +3628,7 @@
 echocheck "DivX4linux/DivX5linux/OpenDivX decore"
 # DivX5: DEC_OPT_MEMORY_REQS - DivX4: DEC_OPT_FRAME_311
 cat > $TMPC << EOF
-#include <decore.h>
+#include <divx/decore.h>
 int main(void) { (void) decore(0, 0, 0, 0);  return DEC_OPT_FRAME_311; }
 EOF
 if test "$_divx4linux" != no && cc_check -lm -ldivxdecore -lm ; then
@@ -3631,7 +3646,7 @@
 # OdivxPP disabled because of:
 # ld: Warning: type of symbol `dering' changed from 1 to 2 in opendivx/postprocess.o
 cat > $TMPC << EOF
-#include <decore.h>
+#include <divx/decore.h>
 int main(void) { (void) decore(0, 0, 0, 0);  return DEC_OPT_MEMORY_REQS; }
 EOF
 if cc_check -lm -ldivxdecore -lm ; then
diff -ruN MPlayer-0.90pre6.orig/etc/example.conf MPlayer-0.90pre6/etc/example.conf
--- MPlayer-0.90pre6.orig/etc/example.conf	2002-05-04 22:46:09.000000000 -0600
+++ MPlayer-0.90pre6/etc/example.conf	2002-08-13 13:34:03.000000000 -0600
@@ -8,13 +8,13 @@
 ## more flexibly here. See below.
 ##
 
-# vo=xv			# To specify default video driver (see -vo help for
+vo=sdl			# To specify default video driver (see -vo help for
 			# list)
 
-# ao=oss		# To specify default audio driver (see -ao help for
+ao=sdl			# To specify default audio driver (see -ao help for
 			# list)
 
-fs=yes			# Enlarges movie window to your desktop's size.
+fs=no			# Enlarges movie window to your desktop's size.
 			# Used by drivers: all
 
 # fsmode=0		# This option workarounds some problems when using
@@ -48,7 +48,7 @@
 ## Use GUI mode by default
 ##
 
-# gui = yes
+gui = yes
 
 ##
 ## Specify your preferred default skin here
@@ -56,7 +56,7 @@
 ##  and ~/.mplayer/Skin/yourskin)
 ##
 
-# skin = default
+skin = default
 
 ##
 ## Multiple languages are available :)
@@ -112,8 +112,8 @@
 			# for specifying SDL subdrivers
 # ao = sdl:esd		# use SDL audio driver by default
 			# use "ao = sdl:esd" to use SDL's ESD driver
-# noxv = no		# whether to use XVideo hardware acceleration or not
-# forcexv = yes		# force XVideo even if not detected
+noxv = no		# whether to use XVideo hardware acceleration or not
+forcexv = yes		# force XVideo even if not detected
 
 
 ##
@@ -123,7 +123,7 @@
 framedrop 	= yes	# drop frames, when not in sync (slow CPU, videocard,
 			# etc)
 
-# vfm		= 5	# use libavcodec video codec family (you know,
+vfm		= 5	# use libavcodec video codec family (you know,
 			# opensource DivX for example.. So preferred default on
 			# non-x86 boxes)
 
@@ -145,4 +145,4 @@
 ## Delete this default :)
 ##
 
-include = /home/gabucino/.mplayer/i_did_not_RTFM_carefully_enough...
+#include = /home/gabucino/.mplayer/i_did_not_RTFM_carefully_enough...
