diff -Naur frotz-2.44-orig/Makefile frotz-2.44/Makefile
--- frotz-2.44-orig/Makefile	2015-05-20 04:29:14.000000000 -0500
+++ frotz-2.44/Makefile	2015-08-09 16:00:35.000000000 -0500
@@ -14,9 +14,9 @@
 #-malign-jumps=2
 
 # Define where you want Frotz installed.  Usually this is /usr/local
-PREFIX = /usr/local
+PREFIX = @PREFIX@
 
-MAN_PREFIX = $(PREFIX)
+MAN_PREFIX = $(PREFIX)/share
 #MAN_PREFIX = /usr/local/share
 
 CONFIG_DIR = $(PREFIX)/etc
@@ -24,7 +24,7 @@
 
 # Define where you want Frotz to look for frotz.conf.
 #
-CONFIG_DIR = /usr/local/etc
+CONFIG_DIR = @PREFIX@/etc
 #CONFIG_DIR = /etc
 #CONFIG_DIR = /usr/pkg/etc
 #CONFIG_DIR =
@@ -73,15 +74,15 @@
 # curses library won't work, comment out the first option and uncomment
 # the second.
 #
-CURSES = -lcurses
-#CURSES = -lncurses
+#CURSES = -lcurses
+CURSES = -lncurses
 
 # Uncomment this if your need to use ncurses instead of the
 # vendor-supplied curses library.  This just tells the compile process
 # which header to include, so don't worry if ncurses is all you have
 # (like on Linux).  You'll be fine.
 #
-#CURSES_DEF = -DUSE_NCURSES_H
+CURSES_DEF = -DUSE_NCURSES_H
 
 # Uncomment this if you're compiling Unix Frotz on a machine that lacks
 # the memmove(3) system call.  If you don't know what this means, leave it
@@ -214,19 +215,19 @@
 .SUFFIXES: .c .o .h
 
 $(COMMON_OBJECT): %.o: %.c
-	$(CC) $(OPTS) $(COMMON_DEFS) -o $@ -c $<
+	$(CC) $(FLAGS) $(COMMON_DEFS) -o $@ -c $<
 
 $(BLORB_OBJECT): %.o: %.c
-	$(CC) $(OPTS) -o $@ -c $<
+	$(CC) $(FLAGS) -o $@ -c $<
 
 $(DUMB_OBJECT): %.o: %.c
-	$(CC) $(OPTS) -o $@ -c $<
+	$(CC) $(FLAGS) -o $@ -c $<
 
 $(CURSES_OBJECT): %.o: %.c
-	$(CC) $(OPTS) $(CURSES_DEFS) -o $@ -c $<
+	$(CC) $(FLAGS) $(CURSES_DEFS) -o $@ -c $<
 
 $(SDL_OBJECT): %.o: %.c
-	$(CC) $(OPTS) $(SDL_DEFS) -o $@ -c $<
+	$(CC) $(FLAGS) $(SDL_DEFS) -o $@ -c $<
 
 
 # If you're going to make this target manually, you'd better know which
