diff -ru trackballs-1.1.4-orig/share/Makefile.am trackballs-1.1.4/share/Makefile.am
--- trackballs-1.1.4-orig/share/Makefile.am	2007-11-24 21:08:34.000000000 +0100
+++ trackballs-1.1.4/share/Makefile.am	2007-11-24 21:08:56.000000000 +0100
@@ -4,6 +4,7 @@
 EXTRA_DIST = $(pkgdata_DATA) trackballs.6
 
 install-pkgdataDATA:
-	-install -D -m664 -g games highScores		$(DESTDIR)@with_highscores@
-	-chgrp -R games			$(DESTDIR)$(pkgdatadir)/levels
+	mkdir -p $(DESTDIR)/share/trackballs
+	install -m664 -g games highScores		$(DESTDIR)@with_highscores@
+	chgrp -R games			$(DESTDIR)$(pkgdatadir)/levels
 	chmod -R g+r			$(DESTDIR)$(pkgdatadir)/levels
diff -ru trackballs-1.1.4-orig/src/Makefile.am trackballs-1.1.4/src/Makefile.am
--- trackballs-1.1.4-orig/src/Makefile.am	2007-11-24 21:08:34.000000000 +0100
+++ trackballs-1.1.4/src/Makefile.am	2007-11-24 21:40:36.000000000 +0100
@@ -30,7 +30,7 @@
 
 EXTRA_trackballs_SOURCES = trackballsIcon.rc
 
-trackballs_LDADD = @WINRESOURCES@
+trackballs_LDADD = @WINRESOURCES@ $(LTLIBINTL)
 trackballs_DEPENDENCIES = @WINRESOURCES@
 
 trackballsIcon.o : trackballsIcon.rc trackballsIcon.ico
diff -ru trackballs-1.1.4-orig/src/Makefile.in trackballs-1.1.4/src/Makefile.in
--- trackballs-1.1.4-orig/src/Makefile.in	2007-11-24 21:08:34.000000000 +0100
+++ trackballs-1.1.4/src/Makefile.in	2007-11-24 21:12:24.000000000 +0100
@@ -316,7 +316,7 @@
 	-test -z "$(bin_PROGRAMS)" || rm -f $(bin_PROGRAMS)
 trackballs$(EXEEXT): $(trackballs_OBJECTS) $(trackballs_DEPENDENCIES) 
 	@rm -f trackballs$(EXEEXT)
-	$(trackballs_LINK) $(trackballs_OBJECTS) $(trackballs_LDADD) $(LIBS)
+	$(trackballs_LINK) $(trackballs_OBJECTS) $(trackballs_LDADD) $(LIBS) $(LTLIBINTL)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
