--- wesnoth-1.8.5-orig/configure.ac	2010-09-24 10:37:41.000000000 -0700
+++ wesnoth-1.8.5/configure.ac	2010-10-22 15:08:14.000000000 -0700
@@ -755,6 +755,7 @@
 
 AC_LANG([C])
 AC_MSG_CHECKING([for PNG support in SDL_image])
+if test "x$enable_sdltest" = "xyes" ; then
 
 OLD_CPPFLAGS=$CPPFLAGS
 OLD_CFLAGS=$CFLAGS
@@ -789,11 +790,14 @@
 	[AC_MSG_ERROR([*** Either your test image has vanished, or SDL_image has no PNG support!])],
 	[AC_MSG_RESULT([not tested in cross-compiling])])
 
 CPPFLAGS=$OLD_CPPFLAGS
 CFLAGS=$OLD_CFLAGS
 LIBS=$OLD_LIBS
+else
+	AC_MSG_RESULT([test disabled])
+fi
 
 
 #######################################################################
 # Check for OGG support in SDL_mixer                                  #
 #######################################################################
--- wesnoth-1.8.6-orig/src/gui/auxiliary/event/dispatcher_private.hpp	2013-08-09 15:05:49.000000000 -0700
+++ wesnoth-1.8.6/src/gui/auxiliary/event/dispatcher_private.hpp	2013-08-09 15:07:41.000000000 -0700
@@ -319,9 +319,9 @@
 			// MSVC 2008 doesn't like operator() here so changed the name.
 			return functor.template oper<item>(event);
 		} else {
-			typedef typename boost::mpl::next<itor>::type itor;
-			return find<boost::is_same<itor, end>::value>
-				::execute((itor*)0, (end*)0, event, functor);
+			typedef typename boost::mpl::next<itor>::type itor2;
+			return find<boost::is_same<itor2, end>::value>
+				::execute((itor2*)0, (end*)0, event, functor);
 		}
     }
 };
