diff -ru enlightenment-0.16.5/dox/file.c enlightenment-0.16.5-patched/dox/file.c
--- enlightenment-0.16.5/dox/file.c	Thu Jun 29 21:29:04 2000
+++ enlightenment-0.16.5-patched/dox/file.c	Sat Oct 13 19:52:12 2001
@@ -966,7 +966,11 @@
    char               *tmp, *lang, *p[3];
    int                 i;
 
+#ifdef HAVE_LC_MESSAGES
    if (!(lang = setlocale(LC_MESSAGES, NULL)))
+#else
+   if (!(lang = setlocale(LC_ALL, NULL)))
+#endif
      return 0;
 
    tmp = strdup(fname);
diff -ru enlightenment-0.16.5/scripts/enlightenment.install.in enlightenment-0.16.5-patched/scripts/enlightenment.install.in
--- enlightenment-0.16.5/scripts/enlightenment.install.in	Thu Sep 14 16:37:51 2000
+++ enlightenment-0.16.5-patched/scripts/enlightenment.install.in	Sat Oct 13 19:55:17 2001
@@ -24,6 +24,7 @@
 # CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 ##############################################################################
 
+EBIN=@ENLIGHTENMENT_BIN@
 PATH=$PATH:"/usr/local/enlightenment/bin:/usr/enlightenment/bin:/opt/enlightenment/bin:/usr/local/bin:"$EBIN
 export PATH
 
@@ -68,7 +69,6 @@
   fi
 fi
 
-EBIN=@ENLIGHTENMENT_BIN@
 echo "... Generating login startup files ..."
 echo "#!/bin/sh" >> $HOME"/.xsession"
 echo "# Execute enlightenment. ALWAYS make sure this is at the end of this" >> $HOME"/.xsession"
diff -ru enlightenment-0.16.5/src/file.c enlightenment-0.16.5-patched/src/file.c
--- enlightenment-0.16.5/src/file.c	Fri Jul 28 21:13:13 2000
+++ enlightenment-0.16.5-patched/src/file.c	Sat Oct 13 19:52:12 2001
@@ -1087,7 +1087,11 @@
    char               *tmp, *lang, *p[3];
    int                 i;
 
+#ifdef HAVE_LC_MESSAGES
    if (!(lang = setlocale(LC_MESSAGES, NULL)))
+#else
+   if (!(lang = setlocale(LC_ALL, NULL)))
+#endif
       return 0;
 
    tmp = strdup(fname);
diff -ru enlightenment-0.16.5/src/menus.c enlightenment-0.16.5-patched/src/menus.c
--- enlightenment-0.16.5/src/menus.c	Fri Jul 28 19:23:35 2000
+++ enlightenment-0.16.5-patched/src/menus.c	Sat Oct 13 19:52:12 2001
@@ -1694,7 +1694,11 @@
 
    EDBUG(5, "CreateMenuFromGnome");
 
+#ifdef HAVE_LC_MESSAGES
    if ((lang = setlocale(LC_MESSAGES, NULL)) != NULL)
+#else
+   if ((lang = setlocale(LC_ALL, NULL)) != NULL)
+#endif
       Esnprintf(name_buf, sizeof(name_buf), "Name[%s]=", lang);
    else
       name_buf[0] = '\0';
