
--- amarok-1.4.0/amarok/configure.in.bot	2006-05-14 14:11:27.000000000 -0400
+++ amarok-1.4.0-new/amarok/configure.in.bot	2006-05-18 14:16:54.000000000 -0400
@@ -136,17 +136,6 @@
     echo " ="
 fi
 
-if test x$no_engine = xyes; then
-    all_tests=bad
-    echo " =================================="
-    echo " ===  AMAROK WILL NOT BE BUILT  ================================================"
-    echo " =================================="
-    echo " ="
-    echo " = No suitable multimedia framework was detected. You need to install at least"
-    echo " = the Xine or Helix framework as detailed in the amaroK README."
-    echo " ="
-fi
-
 if test x$no_amarok = xyes; then
     all_tests=bad
     echo " =================================="
--- amarok-1.4.0/amarok/configure.in.in	2006-05-14 14:11:27.000000000 -0400
+++ amarok-1.4.0-new/amarok/configure.in.in	2006-05-18 14:29:13.000000000 -0400
@@ -943,18 +943,3 @@
 
 
 
-###############################################################################
-# BEGIN DO_NOT_COMPILE CHECK
-###############################################################################
-
-if test x$build_xine = xno -a x$build_helix = xno; then
-
-    DO_NOT_COMPILE="$DO_NOT_COMPILE amarok"
-    no_engine=yes
-
-fi
-
-###############################################################################
-# END DO_NOT_COMPILE CHECK
-###############################################################################
-
--- amarok-1.4.0/amarok/src/Makefile.am	2006-05-14 14:11:27.000000000 -0400
+++ amarok-1.4.0-new/amarok/src/Makefile.am	2006-05-18 14:16:34.000000000 -0400
@@ -153,7 +153,7 @@
 
 libamarok_la_LDFLAGS = \
     $(all_libraries) \
-    $(KDE_RPATH)
+    $(KDE_RPATH) -Wl,-framework,CoreFoundation
 
 METASOURCES = \
     AUTO
--- amarok-1.4.0/amarok/src/engine/xine/Makefile.am	2006-05-14 14:11:27.000000000 -0400
+++ amarok-1.4.0-new/amarok/src/engine/xine/Makefile.am	2006-05-18 15:21:14.000000000 -0400
@@ -6,9 +6,7 @@
     $(CFLAGS_XINE) \
     $(all_includes)
 
-libamarok_xine_engine_la_LIBADD = \
-    $(top_builddir)/amarok/src/engine/libengine.la \
-    $(top_builddir)/amarok/src/plugin/libplugin.la \
+libamarok_xine_engine_la_LIBADD = -lamarok \
     -lkdeui -lkdecore \
     $(LIBS_XINE)
 
--- amarok-1.4.0/amarok/src/vis/Makefile.am	2006-05-14 14:11:27.000000000 -0400
+++ amarok-1.4.0-new/amarok/src/vis/Makefile.am	2006-05-18 14:16:34.000000000 -0400
@@ -3,7 +3,7 @@
 endif
 
 if with_libvisual
-    LIBVISUAL_BUILD = libvisual
+    LIBVISUAL_BUILD = 
 endif
 
 SUBDIRS = \
