diff -uNr kdeadmin-3.4.2/configure.in.in kdeadmin-3.4.2-patched/configure.in.in
--- kdeadmin-3.4.2/configure.in.in	2005-07-20 12:42:20.000000000 -0400
+++ kdeadmin-3.4.2-patched/configure.in.in	2005-08-30 15:41:28.000000000 -0400
@@ -8,3 +8,5 @@
 AC_CHECK_HEADERS(sys/stropts.h stropts.h)
 AC_CHECK_HEADERS(sys/time.h)
 AC_HEADER_TIME
+
+DO_NOT_COMPILE="$DO_NOT_COMPILE kdat ksysv lilo-config kuser"
diff -uNr kdeadmin-3.4.2/kpackage/debAptInterface.cpp kdeadmin-3.4.2-patched/kpackage/debAptInterface.cpp
--- kdeadmin-3.4.2/kpackage/debAptInterface.cpp	2005-07-20 12:42:16.000000000 -0400
+++ kdeadmin-3.4.2-patched/kpackage/debAptInterface.cpp	2005-08-30 16:26:26.000000000 -0400
@@ -470,8 +470,8 @@
     cmd += "\\";
     cmd += "n";
   }
-  cmd += "> /etc/apt/sources.list.n; if [ $? = 0 ]; then ";
-  cmd += "mv /etc/apt/sources.list /etc/apt/sources.list.b; mv /etc/apt/sources.list.n /etc/apt/sources.list; fi'";
+  cmd += "> @PREFIX@/etc/apt/sources.list.n; if [ $? = 0 ]; then ";
+  cmd += "mv @PREFIX@/etc/apt/sources.list @PREFIX@/etc/apt/sources.list.b; mv @PREFIX@/etc/apt/sources.list.n @PREFIX@/etc/apt/sources.list; fi'";
   
 
  QStringList rlist =  kpty->run(cmd,TRUE,TRUE);
diff -uNr kdeadmin-3.4.2/kpackage/debAptInterface.h kdeadmin-3.4.2-patched/kpackage/debAptInterface.h
--- kdeadmin-3.4.2/kpackage/debAptInterface.h	2005-07-20 12:42:16.000000000 -0400
+++ kdeadmin-3.4.2-patched/kpackage/debAptInterface.h	2005-08-30 16:25:54.000000000 -0400
@@ -33,7 +33,7 @@
 
 #include "debInterface.h"
 
-#define APT_SOURCE "/etc/apt/sources.list"
+#define APT_SOURCE "@PREFIX@/etc/apt/sources.list"
 
 class DEBAPT: public DEB
 {
diff -uNr kdeadmin-3.4.2/kpackage/debInterface.cpp kdeadmin-3.4.2-patched/kpackage/debInterface.cpp
--- kdeadmin-3.4.2/kpackage/debInterface.cpp	2005-07-20 12:42:16.000000000 -0400
+++ kdeadmin-3.4.2-patched/kpackage/debInterface.cpp	2005-08-30 16:26:54.000000000 -0400
@@ -407,7 +407,7 @@
   QStringList ret;
   QString name = p->getProperty("name");
 
-  from = "zcat /usr/share/doc/";
+  from = "zcat @PREFIX@/share/doc/";
   from += name;
   from += "/changelog.Debian.gz";
 
@@ -416,7 +416,7 @@
   if (!kpty->Result)
     return ret;
   else {
-    from = "zcat /usr/share/doc/";
+    from = "zcat @PREFIX@/share/doc/";
     from += name;
     from += "/changelog.gz";
 
diff -uNr kdeadmin-3.4.2/kpackage/debInterface.h kdeadmin-3.4.2-patched/kpackage/debInterface.h
--- kdeadmin-3.4.2/kpackage/debInterface.h	2005-07-20 12:42:16.000000000 -0400
+++ kdeadmin-3.4.2-patched/kpackage/debInterface.h	2005-08-30 16:27:19.000000000 -0400
@@ -45,10 +45,10 @@
 class updateLoc;
 class cacheObj;
 
-#define AVAIL          "/var/lib/dpkg/available"
-#define STATUS          "/var/lib/dpkg/status"
+#define AVAIL          "@PREFIX@/var/lib/dpkg/available"
+#define STATUS          "@PREFIX@/var/lib/dpkg/status"
 
-#define INFODIR         "/var/lib/dpkg/info/"
+#define INFODIR         "@PREFIX@/var/lib/dpkg/info/"
 
 
 class DEB: public pkgInterface
