diff -uNr ../quanta-3.2.0/kommander/widgets/Makefile.am ./kommander/widgets/Makefile.am
--- ../quanta-3.2.0/kommander/widgets/Makefile.am	Sun Jan 25 12:10:35 2004
+++ ./kommander/widgets/Makefile.am	Mon Feb  9 14:06:32 2004
@@ -8,7 +8,7 @@
 libkommanderwidgets_la_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
 # the libraries to link against.
-libkommanderwidgets_la_LIBADD  = $(LIB_KIO) 
+libkommanderwidgets_la_LIBADD  = $(LIB_KIO) $(LIBZ)
 
 # which sources should be compiled for widgets
 libkommanderwidgets_la_SOURCES = assoctextwidget.cpp myprocess.cpp buttongroup.cpp ewidgetfactory.cpp subdialog.cpp checkbox.cpp closebutton.cpp combobox.cpp listbox.cpp wizard.cpp dialog.cpp tabwidget.cpp execbutton.cpp fileselector.cpp groupbox.cpp	lineedit.cpp radiobutton.cpp  spinboxint.cpp textedit.cpp scriptobject.cpp richtexteditor.cpp treewidget.cpp
diff -uNr ../quanta-3.2.0/quanta/parts/kxsldbg/Makefile.am ./quanta/parts/kxsldbg/Makefile.am
--- ../quanta-3.2.0/quanta/parts/kxsldbg/Makefile.am	Sun Nov 30 04:52:04 2003
+++ ./quanta/parts/kxsldbg/Makefile.am	Mon Feb  9 14:06:07 2004
@@ -3,7 +3,7 @@
 kxsldbg_SOURCES = kxsldbg.cpp main.cpp kxsldbgif.skel
 noinst_HEADERS = kxsldbg.h
 include_HEADERS = kxsldbgif.h
-kxsldbg_LDADD   = ./kxsldbgpart/libqtnotfier/libqtnotfier.la ./kxsldbgpart/libkxsldbgpart.la ./kxsldbgpart/libxsldbg/libxsldbg.la  $(LIB_KDEUI) $(LIBXSLT_LIBS) $(LIBXML_LIBS)
+kxsldbg_LDADD   = ./kxsldbgpart/libqtnotfier/libqtnotfier.la ./kxsldbgpart/libkxsldbgcommon.la ./kxsldbgpart/libxsldbg/libxsldbg.la  $(LIB_KDEUI) $(LIBXSLT_LIBS) $(LIBXML_LIBS)
 
 SUBDIRS = kxsldbgpart
 
diff -uNr ../quanta-3.2.0/quanta/parts/kxsldbg/kxsldbgpart/Makefile.am ./quanta/parts/kxsldbg/kxsldbgpart/Makefile.am
--- ../quanta-3.2.0/quanta/parts/kxsldbg/kxsldbgpart/Makefile.am	Sun Nov 30 04:52:04 2003
+++ ./quanta/parts/kxsldbg/kxsldbgpart/Makefile.am	Mon Feb  9 14:06:07 2004
@@ -1,8 +1,10 @@
 kde_module_LTLIBRARIES = libkxsldbgpart.la
 
-libkxsldbgpart_la_METASOURCES = AUTO
+noinst_LTLIBRARIES = libkxsldbgcommon.la
 
-libkxsldbgpart_la_SOURCES = xsldbgwalkspeedimpl.cpp xsldbgtemplatesimpl.cpp xsldbgtemplatelistitem.cpp xsldbgsourcesimpl.cpp xsldbgoutputview.cpp xsldbglocalvariablesimpl.cpp xsldbglocallistitem.cpp xsldbglistitem.cpp xsldbginspector.cpp xsldbgglobalvariablesimpl.cpp xsldbggloballistitem.cpp xsldbgentitiesimpl.cpp xsldbgdoc.cpp xsldbgdialogbase.cpp xsldbgdebugger.cpp xsldbgconfigimpl.cpp xsldbgcallstackimpl.cpp xsldbgbreakpointsimpl.cpp xsldbgbreakpointlistitem.cpp qxsldbgview.cpp qxsldbgdoc.cpp kxsldbg_part.cpp kxsldbg_partif.skel xsldbgwalkspeed.ui xsldbgtemplates.ui xsldbgsources.ui xsldbgmsgdialog.ui xsldbglocalvariables.ui xsldbgglobalvariables.ui xsldbgentities.ui xsldbgconfig.ui xsldbgcallstack.ui xsldbgbreakpoints.ui
+libkxsldbgcommon_la_METASOURCES = AUTO
+
+libkxsldbgcommon_la_SOURCES = xsldbgwalkspeedimpl.cpp xsldbgtemplatesimpl.cpp xsldbgtemplatelistitem.cpp xsldbgsourcesimpl.cpp xsldbgoutputview.cpp xsldbglocalvariablesimpl.cpp xsldbglocallistitem.cpp xsldbglistitem.cpp xsldbginspector.cpp xsldbgglobalvariablesimpl.cpp xsldbggloballistitem.cpp xsldbgentitiesimpl.cpp xsldbgdoc.cpp xsldbgdialogbase.cpp xsldbgdebugger.cpp xsldbgconfigimpl.cpp xsldbgcallstackimpl.cpp xsldbgbreakpointsimpl.cpp xsldbgbreakpointlistitem.cpp qxsldbgview.cpp qxsldbgdoc.cpp kxsldbg_part.cpp kxsldbg_partif.skel xsldbgwalkspeed.ui xsldbgtemplates.ui xsldbgsources.ui xsldbgmsgdialog.ui xsldbglocalvariables.ui xsldbgglobalvariables.ui xsldbgentities.ui xsldbgconfig.ui xsldbgcallstack.ui xsldbgbreakpoints.ui
 include_HEADERS =  kxsldbg_partif.h
 noinst_HEADERS = kxsldbg_part.h qxsldbgdoc.h qxsldbgview.h xsldbgbreakpointlistitem.h xsldbgbreakpointsimpl.h xsldbgcallstackimpl.h xsldbgconfigimpl.h xsldbgdebugger.h xsldbgdialogbase.h xsldbgdoc.h xsldbgdocmap.h xsldbgentitiesimpl.h xsldbggloballistitem.h xsldbgglobalvariablesimpl.h xsldbginspector.h xsldbglistitem.h xsldbglocallistitem.h xsldbglocalvariablesimpl.h xsldbgmsgdialogimpl.h xsldbgoutputview.h xsldbgsourcesimpl.h xsldbgtemplatelistitem.h xsldbgtemplatesimpl.h xsldbgwalkspeedimpl.h
 
@@ -11,9 +13,15 @@
 INCLUDES = $(XSLDBG_CFLAGS) -I.. $(all_includes)
 
 
+libkxsldbgcommon_la_LDFLAGS = -no-undefined $(all_libraries) -version-info 1:0:0
+libkxsldbgcommon_la_LIBADD  =  libxsldbg/libxsldbg.la libqtnotfier/libqtnotfier.la $(LIB_KPARTS)
+
 libkxsldbgpart_la_LDFLAGS = -module -no-undefined $(all_libraries) -version-info 1:0:0
-libkxsldbgpart_la_LIBADD  =  libxsldbg/libxsldbg.la libqtnotfier/libqtnotfier.la $(LIB_KPARTS)
+libkxsldbgpart_la_LIBADD  = libkxsldbgcommon.la
+libkxsldbgpart_la_SOURCES = dummy.cpp
 
+dummy.cpp:
+	echo > dummy.cpp
 
 # this is where the desktop file will go
 partdesktopdir   = $(kde_servicesdir)
