diff -uNr cairo/cairo.pc.in cairo-new/cairo.pc.in
--- cairo/cairo.pc.in	Fri Apr  9 17:19:47 2004
+++ cairo-new/cairo.pc.in	Wed Sep 22 12:32:06 2004
@@ -8,6 +8,6 @@
 Version: @VERSION@
 
 Requires: fontconfig libpixman @XRENDER_REQUIRES@ @PNG_REQUIRES@ @GL_REQUIRES@
-Libs: -L${libdir} -lcairo -lm @XRENDER_LIBS@ @PS_LIBS@ @FREETYPE_LIBS@
-Cflags: -I${includedir} @FREETYPE_CFLAGS@
+Libs: @FREETYPE_LIBS@ -L${libdir} -lcairo -lm @XRENDER_LIBS@ @PS_LIBS@
+Cflags: @FREETYPE_CFLAGS@ -I${includedir}
 
diff -uNr cairo/src/cairo_image_surface.c cairo-new/src/cairo_image_surface.c
--- cairo/src/cairo_image_surface.c	Mon May 24 05:28:05 2004
+++ cairo-new/src/cairo_image_surface.c	Wed Sep 22 12:31:35 2004
@@ -463,7 +463,17 @@
 _cairo_image_surface_set_clip_region (cairo_image_surface_t *surface,
 				      pixman_region16_t *region)
 {
-    pixman_image_set_clip_region (surface->pixman_image, region);
+    if (region) {
+        pixman_region16_t *rcopy;
+
+        rcopy = pixman_region_create();
+        /* pixman_image_set_clip_region expects to take ownership of the
+         * passed-in region, so we create a copy to give it. */
+        pixman_region_copy (rcopy, region);
+        pixman_image_set_clip_region (surface->pixman_image, rcopy);
+    } else {
+        pixman_image_set_clip_region (surface->pixman_image, region);
+    }
 
     return CAIRO_STATUS_SUCCESS;
 }
