--- enblend-2.5/src/Makefile.in	2005-12-11 01:19:46.000000000 -0500
+++ enblend-2.5-new/src/Makefile.in	2006-02-26 22:25:28.000000000 -0500
@@ -161,7 +161,8 @@
 enblend_LDFLAGS = -DENBLEND_CACHE_IMAGES -ffast-math
 #enblend_LDADD = vigra_impex/libvigra_impex.a -lpng -ljpeg -ltiff
 enblend_LDADD = vigra_impex/libvigra_impex.a -ltiff
-enblend_CXXFLAGS = -g -O3 -ffast-math -Wall -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -DENBLEND_CACHE_IMAGES -I../include
+enblend_CXXFLAGS = -g -O3 -ffast-math -Wall -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -DENBLEND_CACHE_IMAGES
+AM_CPPFLAGS = -I../include
 all: all-recursive
 
 .SUFFIXES:
--- enblend-2.5/src/vigra_impex/Makefile.in	2005-12-11 01:19:46.000000000 -0500
+++ enblend-2.5-new/src/vigra_impex/Makefile.in	2006-02-26 22:25:46.000000000 -0500
@@ -172,7 +172,8 @@
 noinst_LIBRARIES = libvigra_impex.a
 libvigra_impex_a_SOURCES = auto_file.hxx bmp.cxx bmp.hxx byteorder.cxx byteorder.hxx codecmanager.cxx codecmanager.hxx error.hxx gif.cxx gif.hxx imageinfo.cxx jpeg.cxx jpeg.hxx png.cxx png.hxx pnm.cxx pnm.hxx sun.cxx sun.hxx tiff.cxx tiff.hxx viff.cxx viff.hxx void_vector.cxx void_vector.hxx
 #libvigra_impex_a_CXXFLAGS = -I../../include -O3 -Wall -DHasPNG -DHasJPEG -DHasTIFF
-libvigra_impex_a_CXXFLAGS = -I../../include -O3 -Wall -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -DHasTIFF
+libvigra_impex_a_CXXFLAGS = -O3 -Wall -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -DHasTIFF
+AM_CPPFLAGS = -I../../include
 all: all-am
 
 .SUFFIXES:
