diff -ru fontforge-20041023.orig/fontforge/splineutil.c fontforge-20041023/fontforge/splineutil.c
--- fontforge-20041023.orig/fontforge/splineutil.c	Sat Oct 16 10:42:06 2004
+++ fontforge-20041023/fontforge/splineutil.c	Tue Nov  9 23:01:01 2004
@@ -1680,6 +1680,11 @@
     for ( i=0; i<encindex; ++i )
 	if ( i!=encindex && strcmp(encoding[i],encoding[encindex])==0 )
     break;
+    /* Encoding calls for a glyph which did not exist. Try .notdef instead */
+    if ( i<encindex )
+	for ( i=0; i<encindex; ++i )
+	    if ( i!=encindex && strcmp(encoding[i],".notdef")==0 )
+	break;
 
     if ( i<encindex )
 	sc = MakeDupRef(sf->chars[i],encindex,-1);
@@ -1906,12 +1911,14 @@
 		sf->chars[i]->changed = false;
 	    used[k] = true;
 	}
-	sf->chars[i]->orig_pos = k;
-	sf->chars[i]->vwidth = sf->ascent+sf->descent;
-	sf->chars[i]->enc = i;
-	sf->chars[i]->unicodeenc = UniFromName(encoding[i],sf->uni_interp,sf->encoding_name);
-	sf->chars[i]->parent = sf;
-	SCLigDefault(sf->chars[i]);		/* Also reads from AFM file, but it probably doesn't exist */
+	if ( sf->chars[i]!=NULL ) {
+	    sf->chars[i]->orig_pos = k;
+	    sf->chars[i]->vwidth = sf->ascent+sf->descent;
+	    sf->chars[i]->enc = i;
+	    sf->chars[i]->unicodeenc = UniFromName(encoding[i],sf->uni_interp,sf->encoding_name);
+	    sf->chars[i]->parent = sf;
+	    SCLigDefault(sf->chars[i]);		/* Also reads from AFM file, but it probably doesn't exist */
+	}
 #if defined(FONTFORGE_CONFIG_GDRAW)
 	GProgressNext();
 #elif defined(FONTFORGE_CONFIG_GTK)
