diff -uNr graphviz-1.16/lefty/common.h graphviz-1.16-new/lefty/common.h
--- graphviz-1.16/lefty/common.h	Tue Aug 24 18:52:28 2004
+++ graphviz-1.16-new/lefty/common.h	Thu Nov  4 16:27:16 2004
@@ -50,6 +50,9 @@
 #endif
 /* */
 
+#ifdef HAVE_SYS_TYPES_H
+#include <sys/types.h>
+#endif
 #ifdef FEATURE_CS
 #include <ast.h>
 #else
@@ -67,9 +70,6 @@
 #endif
 #ifdef HAVE_LIMITS_H
 #include <limits.h>
-#endif
-#ifdef HAVE_SYS_TYPES_H
-#include <sys/types.h>
 #endif
 #ifdef HAVE_SYS_STAT_H
 #include <sys/stat.h>
diff -uNr graphviz-1.16/lefty/os/unix/io.c graphviz-1.16-new/lefty/os/unix/io.c
--- graphviz-1.16/lefty/os/unix/io.c	Wed Jul 21 13:27:56 2004
+++ graphviz-1.16-new/lefty/os/unix/io.c	Thu Nov  4 16:27:16 2004
@@ -1,5 +1,12 @@
 /* Lefteris Koutsofios - AT&T Bell Laboratories */
 
+#ifdef __APPLE__
+#include <AvailabilityMacros.h>
+#ifndef MAC_OS_X_VERSION_10_3
+#define socklen_t int
+#endif
+#endif
+
 #include "common.h"
 #include "io.h"
 #ifdef FEATURE_CS
diff -uNr graphviz-1.16/ns/ns.c graphviz-1.16-new/ns/ns.c
--- graphviz-1.16/ns/ns.c	Fri Jul 16 08:29:12 2004
+++ graphviz-1.16-new/ns/ns.c	Thu Nov  4 16:28:25 2004
@@ -10,18 +10,24 @@
 
 #ifdef HAVE_VALUES_H
 #include <values.h>
-#else
+#endif
+
+#ifdef HAVE_LIMITS_H
 #include <limits.h>
+#endif
+
 #ifndef MAXINT
 #define MAXINT INT_MAX
 #endif
+
 #include <float.h>
+
 #ifndef MAXDOUBLE
 #define MAXDOUBLE DBL_MAX
 #endif
+
 #ifndef MAXFLOAT
 #define MAXFLOAT FLT_MAX
-#endif
 #endif
 
 static int NS_run;
