diff -uNr gtk-sharp-1.9.2/Makefile.in gtk-sharp-1.9.2-new/Makefile.in
--- gtk-sharp-1.9.2/Makefile.in	Thu Feb 10 14:22:47 2005
+++ gtk-sharp-1.9.2-new/Makefile.in	Wed Mar 23 14:57:50 2005
@@ -248,7 +248,7 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-SUBDIRS = sources generator parser glib pango atk gdk gtk glade art gnomevfs gnome gda gnomedb gtkhtml gconf rsvg vte gtkdotnet sample doc
+SUBDIRS = sources generator parser glib pango atk gdk gtk glade art gnomevfs gnome gda gnomedb gtkhtml gconf rsvg vte gtkdotnet sample
 EXTRA_DIST = \
 	gtk-sharp.snk		\
 	makefile.win32		\
diff -uNr gtk-sharp-1.9.2/glib/glue/Makefile.in gtk-sharp-1.9.2-new/glib/glue/Makefile.in
--- gtk-sharp-1.9.2/glib/glue/Makefile.in	Thu Feb 10 14:22:45 2005
+++ gtk-sharp-1.9.2-new/glib/glue/Makefile.in	Wed Mar 23 14:57:34 2005
@@ -143,7 +143,7 @@
 GLADE_CFLAGS = @GLADE_CFLAGS@
 GLADE_LIBS = @GLADE_LIBS@
 GLIB_CFLAGS = @GLIB_CFLAGS@
-GLIB_LIBS = @GLIB_LIBS@
+GLIB_LIBS = @GLIB_LIBS@ -lgobject-2.0
 GNOMEVFS_CFLAGS = @GNOMEVFS_CFLAGS@
 GNOMEVFS_LIBS = @GNOMEVFS_LIBS@
 GNOME_CFLAGS = @GNOME_CFLAGS@
