--- ivtools-1.0/config/rules.def	Tue Feb  5 09:12:01 2002
+++ ivtools-1.0-patched/config/rules.def	Sat Apr  6 23:14:11 2002
@@ -1096,7 +1096,7 @@
 #define InstallSymLink(dir, file)					@@\
 install::								@@\
 	MakeDir(dir)							@@\
-	cp -fd file dir							@@\
+	cp -f file dir							@@\
 									@@\
 uninstall::								@@\
 	$(RM) dir/file	
--- ivtools-1.0/src/include/ivstd/fstream.h.orig	2002-11-23 06:59:49.000000000 -0800
+++ ivtools-1.0/src/include/ivstd/fstream.h	2002-11-23 07:00:02.000000000 -0800
@@ -4,7 +4,7 @@
 #if __GNUC__>=3
 #define input ios_base::in
 #define output ios_base::out
-#if __GNUC__>3 || __GNUC__==3 && __GNUC_MINOR__>0
+#if __GNUC__>3 || __GNUC__==3 && __GNUC_MINOR__>0 && !defined(__APPLE__)
 #include <ext/stdio_filebuf.h>
 #define fileptr_filebuf __gnu_cxx::stdio_filebuf<char>
 #else
--- ivtools-1.0/src/OS/string.c.orig	2002-11-23 07:02:34.000000000 -0800
+++ ivtools-1.0/src/OS/string.c	2002-11-23 07:03:23.000000000 -0800
@@ -33,21 +33,12 @@
  */
 
 extern "C" {
-#ifdef __APPLE__
-#ifndef tolower
-    extern wchar_t tolower(wchar_t);
-#endif
-#ifndef toupper
-    extern wchar_t toupper(wchar_t);
-#endif
-#else
 #ifndef tolower
     extern int tolower(int);
 #endif
 #ifndef toupper
     extern int toupper(int);
 #endif
-#endif
     extern long int strtol(const char*, char**, int);
     extern double strtod(const char*, char**);
 }
--- ivtools-1.0/config/site.def.DARWIN.orig	2002-11-23 07:07:32.000000000 -0800
+++ ivtools-1.0/config/site.def.DARWIN	2002-11-23 07:06:12.000000000 -0800
@@ -13,6 +13,10 @@
 #undef SharedCCLdFlags
 #define SharedCCLdFlags
 
+/* -gdwarf-2 crashes GCC 3.1 on Darwin */
+#undef DefaultDebugCCFlags
+#define DefaultDebugCCFlags
+
 #undef TroffCmd
 #define TroffCmd groff
 
--- ivtools-1.0/src/include/ivstd/iosfwd.orig	2002-11-23 07:26:52.000000000 -0800
+++ ivtools-1.0/src/include/ivstd/iosfwd	2002-11-23 07:27:06.000000000 -0800
@@ -1,6 +1,6 @@
 #ifndef _iv_iosfwd_
 #define _iv_iosfwd_
-#if __GNUG__>=3
+#if __GNUG__>=3 || defined(__APPLE__)
 #include_next <iosfwd>
 using std::ios_base;
 using std::filebuf;
--- ivtools-1.0/src/Time/Time.h.orig	2002-11-23 07:10:56.000000000 -0800
+++ ivtools-1.0/src/Time/Time.h	2002-11-23 07:38:20.000000000 -0800
@@ -48,6 +48,7 @@
 
 #if defined(BSD) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__APPLE__)
 
+#include_next <time.h>
 #include <sys/time.h>
   #if defined(__NetBSD__)
     #include </usr/include/sys/time.h>
@@ -79,8 +80,7 @@
 */
 
 #include <OS/types.h>
-#include <iostream.h>
-#include_next <time.h>
+#include <iostream>
 
 #define NO_NESTED_TYPES 1
 
@@ -141,9 +141,9 @@
 
 	Time max(const Time&) const;
 	Time min(const Time&) const;
-	virtual void printOn(ostream& strm =cout) const;
+	virtual void printOn(std::ostream& strm = std::cout) const;
 
-	friend ostream& operator << (ostream& s, const Time&);
+	friend std::ostream& operator << (std::ostream& s, const Time&);
 };
 
 #endif /* TIMEH */
--- ivtools-1.0/src/Time/Date.h.orig	2002-11-23 07:38:48.000000000 -0800
+++ ivtools-1.0/src/Time/Date.h	2002-11-23 07:36:19.000000000 -0800
@@ -36,6 +36,7 @@
 */
 
 #include <OS/types.h>
+#include <time.h>
 #include <iostream.h>
 
 #define NO_NESTED_TYPES 1

--- ivtools-1.0/config/site.def.DARWIN.orig	2002-11-23 07:07:32.000000000 -0800
+++ ivtools-1.0/config/site.def.DARWIN	2002-11-23 17:47:50.000000000 -0800
@@ -7,11 +7,18 @@
 #undef HasDynamicSharedLibraries
 #define HasDynamicSharedLibraries YES
 
+/* -fPIC is the default on Darwin */
 #undef SharedCCFlags
-#define SharedCCFlags -fPIC
+#define SharedCCFlags
 
 #undef SharedCCLdFlags
-#define SharedCCLdFlags
+#define SharedCCLdFlags -Wl,-bind_at_load
+
+#define CppCmd g++ -E
+
+/* -gdwarf-2 crashes GCC 3.1 on Darwin */
+#undef DefaultDebugCCFlags
+#define DefaultDebugCCFlags
 
 #undef TroffCmd
 #define TroffCmd groff
@@ -25,17 +32,20 @@
 
 /* override -ansi -pedantic from xfree86.cf */
 #undef DefaultCCOptions
-#define DefaultCCOptions /**/
+#define DefaultCCOptions -fno-common /**/
 
 /* machine-specific gcc include directory -- does not change with each new version of gcc */
-#define ToolIncludeDir /usr/local/include
+#define ToolIncludeDir
 
 /* link libraries for the above */
 #undef ExtraCCLdLibs
 #define ExtraCCLdLibs -lstdc++ -lm
 
+#undef ExtraCCLdFlags
+#define ExtraCCLdFlags -Wl,-bind_at_load
+
 /* command to create shared libraries */
-#define SharedLibraryCmd(ldobjs,extra_flags) $(LD) -r extra_flags -o $@~ ldobjs
+#define SharedLibraryCmd(ldobjs,depobjs,rev,libname) $(CXX) -dynamiclib -install_name @PREFIX@/lib/$(AOUT) -current_version rev -compatibility_version rev -o $@~ ldobjs depobjs -flat_namespace -undefined suppress
 
 #define NormalDynamicSharedLibraryTarget(libname,rev,depobjs,ldobjs)	@@\
 AOUT = Concat(lib,libname.rev.dylib)					@@\
@@ -46,7 +56,7 @@
 Concat(lib,libname.rev.dylib): depobjs					@@\
 	@echo "building $@"						@@\
 	$(RM) $@~							@@\
-	SharedLibraryCmd(ldobjs,)					@@\
+	SharedLibraryCmd(ldobjs,depobjs,rev,libname)					@@\
 	$(RM) $@							@@\
 	$(MV) $@~ $@							@@\
 	$(LN) -f $@ Concat(lib,libname.dylib)				@@\
