diff -ur qt-mac-free-3.3.4_old/mkspecs/macx-g++/qmake.conf qt-mac-free-3.3.4/mkspecs/macx-g++/qmake.conf
--- qt-mac-free-3.3.4_old/mkspecs/macx-g++/qmake.conf	Fri Jan 21 18:16:43 2005
+++ qt-mac-free-3.3.4/mkspecs/macx-g++/qmake.conf	Fri Mar 11 00:51:27 2005
@@ -26,6 +26,7 @@
 QMAKE_CFLAGS_DEBUG	= -g
 QMAKE_CFLAGS_SHLIB	= -fPIC
 QMAKE_EXTENSION_SHLIB	= dylib
+QMAKE_EXTENSION_PLUGIN  = so
 QMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
 
 QMAKE_CXX		= c++
@@ -38,9 +39,9 @@
 QMAKE_CXXFLAGS_SHLIB	= $$QMAKE_CFLAGS_SHLIB
 QMAKE_CXXFLAGS_YACC	= $$QMAKE_CFLAGS_YACC
 
-QMAKE_LIBDIR		=
-QMAKE_INCDIR_QT		= $(QTDIR)/include
-QMAKE_LIBDIR_QT		= $(QTDIR)/lib
+QMAKE_LIBDIR		= @PREFIX@/lib/qt3mac/lib
+QMAKE_INCDIR_QT		= @PREFIX@/lib/qt3mac/include
+QMAKE_LIBDIR_QT		= @PREFIX@/lib/qt3mac/lib
 QMAKE_INCDIR_OPENGL	= /System/Library/Frameworks/OpenGL.framework/Headers \
 	/System/Library/Frameworks/AGL.framework/Headers/
 
@@ -51,12 +52,12 @@
 QMAKE_LFLAGS_RELEASE	=
 QMAKE_LFLAGS_DEBUG	=
 QMAKE_LFLAGS_APP	= -prebind
-QMAKE_LFLAGS_SHLIB	= -prebind -dynamiclib 
-QMAKE_LFLAGS_INCREMENTAL = -undefined suppress -flat_namespace
+QMAKE_LFLAGS_SHLIB	= -prebind -dynamiclib -single_module 
+QMAKE_LFLAGS_INCREMENTAL = -undefined dynamic_lookup
 QMAKE_LFLAGS_PLUGIN	= -bundle
 #You probably don't want to mess with this, instead look at install_name_tool(1)
 #QMAKE_LFLAGS_SONAME	= -install_name$${LITERAL_WHITESPACE}@executable_path/../Frameworks/
-QMAKE_LFLAGS_SONAME	= -install_name$${LITERAL_WHITESPACE}
+QMAKE_LFLAGS_SONAME	= -install_name$${LITERAL_WHITESPACE}$$QMAKE_LIBDIR/
 QMAKE_LFLAGS_THREAD	= 
 
 QMAKE_LFLAGS_VERSION    = -current_version$${LITERAL_WHITESPACE}
@@ -64,7 +65,7 @@
 
 QMAKE_RPATH		= 
 
-QMAKE_LIBS_DYNLOAD	=
+QMAKE_LIBS_DYNLOAD	= -ldl
 QMAKE_LIBS_QT		= -lqt
 QMAKE_LIBS_QT_THREAD	= -lqt-mt
 QMAKE_LIBS_OPENGL	= -framework OpenGL -framework AGL
Only in qt-mac-free-3.3.4/mkspecs/macx-g++: qmake.conf~
diff -ur qt-mac-free-3.3.4_old/mkspecs/macx-g++/qplatformdefs.h qt-mac-free-3.3.4/mkspecs/macx-g++/qplatformdefs.h
--- qt-mac-free-3.3.4_old/mkspecs/macx-g++/qplatformdefs.h	Tue May 27 17:19:20 2003
+++ qt-mac-free-3.3.4/mkspecs/macx-g++/qplatformdefs.h	Fri Mar 11 00:52:37 2005
@@ -75,10 +75,11 @@
 #define QT_SIGNAL_ARGS		int
 #define QT_SIGNAL_IGNORE	(void (*)(int))1
 
-#define QT_SOCKLEN_T		int
+#define QT_SOCKLEN_T		socklen_t
 
 #define QT_SNPRINTF		::snprintf
 #define QT_VSNPRINTF		::vsnprintf
 
+#define QT_MITSHM
 
 #endif // QPLATFORMDEFS_H
