diff -ruN transfig.3.2.5-alpha5/fig2dev/Imakefile transfig.3.2.5-alpha5-patched/fig2dev/Imakefile
--- transfig.3.2.5-alpha5/fig2dev/Imakefile	Mon Jan  5 13:40:42 2004
+++ transfig.3.2.5-alpha5-patched/fig2dev/Imakefile	Sat Sep 11 15:33:22 2004
@@ -34,13 +34,13 @@
 XCOMM You should point XFIGLIBDIR to the same directory you did when you compiled
 XCOMM and installed xfig.
 
-XFIGLIBDIR =	/usr/local/lib/X11/xfig
+XFIGLIBDIR =	$(PREFIX)/share/xfig
 
 XCOMM ******
 XCOMM If your system doesn't have the strerror() function (has sys_errlist) then
 XCOMM uncomment NEED_STRERROR.
 
-NEED_STRERROR = -DNEED_STRERROR
+XCOMM NEED_STRERROR = -DNEED_STRERROR
 
 XCOMM The following probably only applies to Windows 9x/NT:
 XCOMM If your system can open files in text and binary modes and has the
@@ -74,9 +74,9 @@
 XCOMM file are in different places
 
 #ifdef USEPNG
-PNGLIBDIR = $(USRLIBDIR)
+PNGLIBDIR = $(PREFIX)/lib
 ZLIBDIR = $(USRLIBDIR)
-PNGINC = -I/usr/include/X11
+PNGINC = -I$(PREFIX)/include
 #endif
 
 XCOMM ****************
@@ -91,8 +91,8 @@
 #define USEXPM
 
 #ifdef USEXPM
-XPMLIBDIR = $(USRLIBDIR)
-XPMINC = -I/usr/include/X11
+XPMLIBDIR = /usr/X11R6/lib
+XPMINC = -I/usr/X11R6/include/X11
 #endif
 
 XCOMM ****************
@@ -117,7 +117,7 @@
 
 DDNFSS = -DNFSS
 DDLATEX2E_GRAPHICS = -DLATEX2E_GRAPHICS
-XCOMM DDEPSFIG = -DEPSFIG
+DDEPSFIG = -DEPSFIG
 
 XCOMM ****************
 XCOMM Comment out the DDDVIPS to NOT use dvips for the DVI-to-PostScript driver
@@ -141,7 +141,7 @@
 
 #ifdef I18N
 I18N_DEFS = -DI18N 
-FIG2DEV_LIBDIR = /usr/local/lib/fig2dev
+FIG2DEV_LIBDIR = @PREFIX@/lib/fig2dev
 I18N_DEV_DEFS = $(I18N_DEFS) -DFIG2DEV_LIBDIR=$(FIG2DEV_LIBDIR) -DFIG2DEV_LIBDIR_STR=\\\"$(FIG2DEV_LIBDIR)\\\"
 
 #endif /* I18N */
diff -ruN transfig.3.2.5-alpha5/transfig/Imakefile transfig.3.2.5-alpha5-patched/transfig/Imakefile
--- transfig.3.2.5-alpha5/transfig/Imakefile	Tue Mar 25 16:52:37 2003
+++ transfig.3.2.5-alpha5-patched/transfig/Imakefile	Sat Sep 11 15:34:05 2004
@@ -32,7 +32,7 @@
 XCOMM \\usepackage{} command for LaTeX2e. 
 XCOMM The default is to use \\documentstyle{} for LaTeX209.
 
-XCOMM USELATEX2E = -DLATEX2E
+USELATEX2E = -DLATEX2E
 
 XCOMM ******* DON'T CHANGE ANYTHING BELOW THIS POINT *******
 
diff -ruN transfig.3.2.5-alpha5/fig2dev/dev/Imakefile transfig.3.2.5-alpha5-patched/fig2dev/dev/Imakefile
--- transfig.3.2.5-alpha5/fig2dev/dev/Imakefile	Tue Sep 30 12:48:08 2003
+++ transfig.3.2.5-alpha5-patched/fig2dev/dev/Imakefile	Sat Sep 11 15:50:30 2004
@@ -50,6 +50,8 @@
 NormalLibraryTarget($(LIB),$(LIBOBJS))
 DependTarget()
 
+FIG2DEV_LIBDIR = $(PREFIX)/lib/fig2dev
+
 #ifdef I18N
 install::
 	InstallNonExecFile(japanese.ps,FIG2DEV_LIBDIR)
