diff -ur abs-src-0.908/Database/cell.c abs-src-0.908-patched/Database/cell.c
--- abs-src-0.908/Database/cell.c	2001-11-11 20:24:33.000000000 +0100
+++ abs-src-0.908-patched/Database/cell.c	2003-11-04 16:25:26.000000000 +0100
@@ -1455,35 +1455,6 @@
   return s;
 }
 
-char *
-strcasestr (char *str1, char *str2)
-{
-  char *s1;
-  char *s2;
-  char *pos = NULL;
-  int l1;
-  s1 = strdup (str1);
-  if (s1 == NULL)
-    return NULL;
-  s2 = strdup (str2);
-  if (s2 == NULL)
-    {
-      free (s1);
-      return NULL;
-    }
-
-  tolower (s1);
-  tolower (s2);
-  pos = strstr (s1, s2);
-  l1 = pos - s1;
-  free (s1);
-  free (s2);
-  if (pos == NULL)
-    return NULL;
-
-  return str1 + l1;
-}
-
 int
 cell_chgwksname (Cell * cell, char *oldname, char *newname)
 {
diff -ur abs-src-0.908/Makefile abs-src-0.908-patched/Makefile
--- abs-src-0.908/Makefile	2001-11-11 20:24:35.000000000 +0100
+++ abs-src-0.908-patched/Makefile	2003-11-04 16:45:45.000000000 +0100
@@ -1,7 +1,7 @@
 CURRENT_DIR = 	.
 
 
-SUBDIRS =	Xpm ABV X11 Kernel Plot Database Fig Interface X11/xedit 
+SUBDIRS =	ABV X11 Kernel Plot Database Fig Interface X11/xedit 
 
 SUBDIRS2 =	ABV X11 Kernel Plot Database Fig Interface 
 MAKE =	make
@@ -10,13 +10,13 @@
 #for silicon: LIBS = -lXaw -lX11 -lm -lXt -lXmu -lXext
 #for hp LIBS = -L /lib/X11 -L /usr/X11R6/lib -L /usr/contrib/X11R6/lib -lXaw -lX11 -lm -lXt -lXmu -lXext
 #for netBSD LIBS = -L /usr/lib -L /usr/X11R6/lib -lXaw -lX11 -lm -lXt -lXmu -lXext
-LIBS = -L /lib/X11 -L /usr/X11R6/lib -L /usr/contrib/X11R6/lib -lXaw -lX11 -lm -lXt -lXmu -lXext
+LIBS = -L/usr/X11R6/lib -lXaw -lX11 -lXt -lXmu -lXext -lXpm -L/usr/lib -lm
 
 #for hp DEFINE = -D_HPUX_SOURCE
 #for AIX DEFINE = -D_AIX_SOURCE
 DEFINE = 
 
-CFLAGS = '-O2 -Wall '
+CFLAGS = '-O2 -Wall -I/usr/X11R6/include'
 
 #for hp IncludeDir = /usr/contrib/X11R6/include
 #for net-BSD IncludeDir = /usr/X11R6/include
@@ -37,7 +37,7 @@
 	done
 	cc -o abs  ./Plot/*.o ./ABV/*.o \
         Fig/*.o Interface/*.o X11/xedit/*.o \
-        ./Kernel/*.o ./X11/*.o ./Database/*.o ./Xpm/libxpm.a \
+        ./Kernel/*.o ./X11/*.o ./Database/*.o \
         $(LIBS) $(DEFINE)
 
 
