diff -urN avl-3.22-1/Avl/bin/Makefile avl-3.22-1mod/Avl/bin/Makefile
--- Avl/bin/Makefile	2005-12-02 20:38:33.000000000 -0500
+++ Avl/bin/Makefile	2005-12-02 20:34:51.000000000 -0500
@@ -2,7 +2,7 @@
 # Makefile for AVL
 #
 
-BINDIR = .
+BINDIR = @PREFIX@/bin
 
 SRC = ../src
 
@@ -22,9 +22,9 @@
 
 ###================================================
 ###  Default compilers and flags
-FC = f77
+FC = g77
-FFLAGS = -O
+FFLAGS = -O2
-PLTLIB = -lX11
+PLTLIB = -L/usr/X11R6/lib -lX11
 FTNLIB =
 ##--------------------------
 
@@ -59,16 +59,16 @@
 #FFLAGS = -O3
 ##--------------------------
 ### Intel Fortran Compiler 8.x
-FC = ifort
-FFLAGS = -O 
-PLTLIB = -L/usr/X11R6/lib -lX11
+#FC = ifort
+#FFLAGS = -O 
+#PLTLIB = -L/usr/X11R6/lib -lX11
 #FTNLIB = -Vaxlib /usr/lib/C-ctype.o /usr/lib/C_name.o /usr/lib/ctype-info.o
 #FTNLIB = -Vaxlib -i_dynamic
 #FTNLIB = -Vaxlib
 ##--------------------------
 ### double-precision option (also need double precision Eispack and libPlt)
-FFLAGS = -O -r8
-PLTOBJ = ../plotlib/libPltDP.a 
+#FFLAGS = -O -r8
+#PLTOBJ = ../plotlib/libPltDP.a 
 
 
 all:    $(PROGS)
diff -urN avl-3.22-1/Avl/eispack/Makefile avl-3.22-1mod/Avl/eispack/Makefile
--- Avl/eispack/Makefile	2005-07-11 11:43:03.000000000 -0400
+++ Avl/eispack/Makefile	2005-12-02 20:35:37.000000000 -0500
@@ -1,7 +1,7 @@
 
-#FC = f77
-FC = ifort
+FC = g77
+#FC = ifort
-FLG = -O
+FLG = -O2
 EISLIB = eispack.a
 AR = ar r
 RAN = ranlib

 
 
diff -urN avl-3.22-1/Avl/plotlib/Makefile avl-3.22-1mod/Avl/plotlib/Makefile
--- Avl/plotlib/Makefile	2005-12-02 20:38:33.000000000 -0500
+++ Avl/plotlib/Makefile	2005-12-02 20:32:10.000000000 -0500
@@ -30,7 +30,7 @@
 #================================#
 
 # Point to your install directory
-#INSTALLDIR= $(HOME)/lib
+INSTALLDIR= @PREFIX@/lib
 #INSTALLDIR= /usr/local/lib
 #INSTALLDIR= .
 
@@ -69,14 +69,14 @@
 
 ###================================================
 ###  Default compilers and flags, install commands
-FC = f77
-CC  = cc
+FC = g77
+CC  = gcc
 # Some fortrans need trailing underscores in C interface symbols (see Xwin.c)
 DEFINE = -DUNDERSCORE
 # Uncomment DP to make double-precision version
 #DP = -r8
-FFLAGS  = -O $(DP)
+FFLAGS  = -O2 $(DP)
-CFLAGS  = -O $(DEFINE)
+CFLAGS  = -O2 -I/usr/X11R6/include/ $(DEFINE)
 AR = ar r 
 RANLIB = ranlib
 LINKLIB =  -lX11 
 
 
 
diff -urN avl-3.22-1/Avl/plotlib/config.make avl-3.22-1mod/Avl/plotlib/config.make
--- Avl/plotlib/config.make	2005-12-02 20:38:33.000000000 -0500
+++ Avl/plotlib/config.make	2005-12-02 20:30:35.000000000 -0500
@@ -8,8 +8,8 @@
 
 ### Use these to set library name 
 ### (you might add DP to name to keep double precision version separate)
-#PLTLIB = libPlt.a
-PLTLIB = libPltDP.a
+PLTLIB = libPlt.a
+#PLTLIB = libPltDP.a
 
 
 # Some fortrans need trailing underscores in C interface symbols (see Xwin.c)
@@ -24,17 +24,17 @@
 ###          or the yaf77 or the old f77 shell script from f2c
 ###   g77    the GNU Fortran compiler 
 #
-#FC = g77-3
+FC = g77
 #FC = fort77
-#CC  = gcc
+CC  = gcc
 # Uncomment DP to make double-precision version
 # (note -r8 does not work in g77, use f2c instead)
 #DP = -r8
 #FFLAGS  = -O2 $(DP)
-#CFLAGS  = -O2 $(DEFINE)
-#AR = ar r
-#RANLIB = ranlib 
-#LINKLIB = -L/usr/X11R6/lib -lX11 
+CFLAGS  = -O2 -I/usr/X11R6/include/ $(DEFINE)
+AR = ar r
+RANLIB = ranlib 
+LINKLIB = -L/usr/X11R6/lib -lX11 
 
 ###-------------------------------------------------------------------------
 ###  Uncomment for Linux, using PGI f77
@@ -51,17 +51,17 @@
 
 ###-------------------------------------------------------------------------
 ###  Uncomment for Linux, using Intel Fortran compiler
-FC = ifort
-CC  = gcc
+#FC = ifort
+#CC  = gcc
  
 # Uncomment to make double-precision version
-DP = -r8
-FFLAGS  = -O3 $(DP)
-CFLAGS  = -O3 $(DEFINE)
-AR = ar r
-RANLIB = ranlib 
+#DP = -r8
+#FFLAGS  = -O3 $(DP)
+#CFLAGS  = -O3 $(DEFINE)
+#AR = ar r
+#RANLIB = ranlib 
 
-LINKLIB = -L/usr/X11R6/lib -lX11
+#LINKLIB = -L/usr/X11R6/lib -lX11
 
 
 ###-------------------------------------------------------------------------


diff -urN avl-3.22-1/Avl/src/aplotmd.f avl-3.22-1mod/Avl/src/aplotmd.f
--- Avl/src/aplotmd.f	2004-11-19 21:35:34.000000000 -0500
+++ Avl/src/aplotmd.f	2005-12-02 20:30:35.000000000 -0500
@@ -1100,14 +1100,14 @@
       REAL*4 SECNDS, TREF, TSEC4
 C
 C---- get CPU time in microseconds
-      CALL CLOCKX(MICSEC)
+C      CALL CLOCKX(MICSEC)
 C---- return seconds
-      TSEC = MICSEC/1.0D6
+C      TSEC = MICSEC/1.0D6
 C
 ccC---- alternative call for some Fortran systems
-cc      TREF = 0.0
-cc      TSEC4 = SECNDS(TREF)
-cc      TSEC = TSEC4
+      TREF = 0.0
+      TSEC4 = SECNDS(TREF)
+      TSEC = TSEC4
 C
      RETURN
      END
