diff -ruN coot-0.6-pre-1-orig/configure coot-0.6-pre-1/configure
--- coot-0.6-pre-1-orig/configure	2009-11-12 09:02:22.000000000 -0800
+++ coot-0.6-pre-1/configure	2009-11-13 21:52:32.000000000 -0800
@@ -23343,14 +23343,14 @@
  # added lz, we should have proper autoconf check for this.
  #
  CLIPPER_LDOPTS="-L$clipper_prefix/lib -lclipper-mtz -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb -lclipper-mmdbold -lclipper-core -lmccp4 $MMDB_LIBS -lrfftw -lfftw -lz -lm"
- CLIPPER_LDOPTS="-L$clipper_prefix/lib -lclipper-ccp4 -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-minimol -lclipper-cns -lclipper-mmdb -lclipper-core -lccp4c $MMDB_LIBS -lrfftw -lfftw -lz -lm"
+ CLIPPER_LDOPTS="-L$clipper_prefix/lib -lclipper-ccp4 -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-minimol -lclipper-cns -lclipper-mmdb -lclipper-core -lgpp4 $MMDB_LIBS -lrfftw -lfftw -lz -lm"
 # -L$clipper_prefix/boost/lib -lclipper-cctbx -L$clipper_prefix/cctbx/lib -lsgtbx -luctbx
 else
  # the compiler looks in the "standard" places for clipper.  In real life,
  # it would be quite unlikely that clipper would be installed in /usr/include,
  # /usr/lib etc. so this code will not usually find the right dependencies.
  CLIPPER_CXXFLAGS=""
- CLIPPER_LDOPTS="-lclipper-ccp4 -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb -lclipper-minimol -lclipper-cns -lclipper-core -lccp4c $MMDB_LIBS -lrfftw -lfftw -lz -lm"
+ CLIPPER_LDOPTS="-lclipper-ccp4 -lclipper-cif -lclipper-phs -lclipper-contrib -lclipper-mmdb -lclipper-minimol -lclipper-cns -lclipper-core -lgpp4 $MMDB_LIBS -lrfftw -lfftw -lz -lm"
 fi
 
 # BL: workaround needed for new MinGW
diff -ruN coot-0.6-pre-1-orig/scheme/group-settings.scm coot-0.6-pre-1/scheme/group-settings.scm
--- coot-0.6-pre-1-orig/scheme/group-settings.scm	2009-07-04 05:57:36.000000000 -0700
+++ coot-0.6-pre-1/scheme/group-settings.scm	2009-11-13 21:52:32.000000000 -0800
@@ -49,3 +49,4 @@
 	 #f)))
 
 
+(set-display-lists-for-maps 0)
diff -ruN coot-0.6-pre-1-orig/scheme/povray.scm coot-0.6-pre-1/scheme/povray.scm
--- coot-0.6-pre-1-orig/scheme/povray.scm	2009-07-04 05:57:34.000000000 -0700
+++ coot-0.6-pre-1/scheme/povray.scm	2009-11-13 21:52:32.000000000 -0800
@@ -14,7 +14,7 @@
   (lambda ()
     
     ; this is directory that contains colors.inc
-    (list (string-append "+L/sw/share/povray-" povray-version "/include")
+    (list (string-append "+L@/share/povray-" povray-version "/include")
 	  "+FN16")))
 
 ;; Run provray using current displayed image and write .pov file to
diff -ruN coot-0.6-pre-1-orig/scheme/raster3d.scm coot-0.6-pre-1/scheme/raster3d.scm
--- coot-0.6-pre-1-orig/scheme/raster3d.scm	2009-07-04 05:57:34.000000000 -0700
+++ coot-0.6-pre-1/scheme/raster3d.scm	2009-11-13 21:52:32.000000000 -0800
@@ -1,7 +1,7 @@
 
 (define coot-r3d-file-name "coot.r3d")
 (define coot-png-file-name "coot.png")
-(define coot-png-display-program "display")
+(define coot-png-display-program "open")
 
 ;; run raster3d
 (define render-image
