--- lib/system-gschemrc.in.bak	Sun Jun  1 19:59:51 2003
+++ lib/system-gschemrc.in	Sat Jun  7 11:57:35 2003
@@ -425,9 +425,11 @@
 ; These numbers are NOT the true size of the window, but of the drawing area.
 ;
 ;(window-size 650 487)  ; Good size for 800x600
-(window-size 900 650)   ; Good size for 1024x768
+;(window-size 900 650)  ; Good size for 1024x768
 ;(window-size 950 712)  ; Good size for 1152x864
 ;(window-size 1100 825) ; Good size for 1280x1024
+
+(window-size 720 540)   ; Good size for 1024x768 OS X screen, minus dock and menu bar
 
 ; world-size width height border
 ;
