--- hdf/config/linux-gnu.orig	2006-04-19 09:47:24.000000000 -0600
+++ hdf/config/linux-gnu	2006-04-19 09:48:03.000000000 -0600
@@ -26,8 +26,8 @@
 fi
 
 if test "X-$F77" = "X-"; then
-  F77=g77
-  F77_BASENAME=g77
+  F77=g95
+  F77_BASENAME=g95
 
 else
     case $F77 in
@@ -99,7 +99,7 @@
 #
 #                     WARNING: flags do not have to be added to the CFLAGS
 #                     or FFLAGS variable if the compiler is the GNU gcc
-#                     and g77 compiler.
+#                     and g95 compiler.
 #
 #                     FFLAGS and CFLAGS should contain *something* or else
 #                     configure will probably add `-g'. For most systems
@@ -138,10 +138,10 @@
 esac
 
 case $F77_BASENAME in 
-  g77)
-    FFLAGS="$FFLAGS -Wsign-compare"
+  g95)
+    FFLAGS="$FFLAGS"
     DEBUG_FFLAGS="-g"
-    PROD_FFLAGS="-O3 -fomit-frame-pointer"
+    PROD_FFLAGS="-O2"
     PROFILE_FFLAGS="-pg"
     ;;
 
--- hdf/hdf/src/hdfi.h.orig	2006-01-02 20:07:53.000000000 -0700
+++ hdf/hdf/src/hdfi.h	2006-01-02 20:08:51.000000000 -0700
@@ -736,7 +736,7 @@
 #include <sys/types.h>
 #include <sys/file.h>               /* for unbuffered i/o stuff */
 #include <sys/stat.h>
-#define DF_MT   DFMT_APPLE 
+#define DF_MT   DFMT_@FMT@
 typedef void            VOID;
 typedef void            *VOIDP;
 typedef char            *_fcd;
@@ -754,8 +754,8 @@
 typedef double          float64;
 typedef int             intf;     /* size of INTEGERs in Fortran compiler */
 typedef int               hdf_pint_t;   /* an integer the same size as a pointer */
+#define FNAME_POST_UNDERSCORE
 #define _fcdtocp(desc) (desc)
-#define FNAME(x) x
 #ifdef H4_ABSOFT
 #define DF_CAPFNAMES
 #endif /*ABSOFT*/
@@ -886,7 +886,7 @@
 #endif /*MAC*/
 
 /* Metrowerks Mac compiler defines some PC stuff so need to exclude this on the Mac */
-#if !(defined(macintosh) || defined(MAC))
+#if !(defined(__APPLE__) || defined(macintosh) || defined(MAC))
 
 #if defined _M_ALPHA || defined _M_IX86 || defined INTEL86 || defined M_I86 || defined M_I386 || defined DOS386 || defined __i386 || defined UNIX386 || defined i386
 #ifndef INTEL86
