--- magic-7.4.59-orig/ext2sim/Makefile	2008-03-29 15:44:18.000000000 -0700
+++ magic-7.4.59/ext2sim/Makefile	2008-03-29 15:45:41.000000000 -0700
@@ -20,10 +20,12 @@
 simwrap.o: ext2sim.c
 	${CC} ${CFLAGS} ${CPPFLAGS} ${DFLAGS} ext2sim.c -c -o simwrap.o
 
+# -install_name is darwin-gcc-specific
 exttosim${SHDLIB_EXT}: simwrap.o ${MAGICDIR}/extflat/libextflat.o
 	@echo --- making exttosim Tcl library \(exttosim${SHDLIB_EXT}\)
 	${RM} exttosim${SHDLIB_EXT}
 	${CC} ${CFLAGS} ${CPPFLAGS} -o $@ ${LDDL_FLAGS} simwrap.o \
+		-install_name ${TCLDIR}/$@ \
 		${MAGICDIR}/extflat/libextflat.o ${LD_SHARED} -lc ${LIBS}
 
 install: $(DESTDIR)${BINDIR}/${MODULE}$(PROGRAM_SUFFIX)${EXEEXT}
--- magic-7.4.59-orig/ext2spice/Makefile	2008-03-29 15:47:30.000000000 -0700
+++ magic-7.4.59/ext2spice/Makefile	2008-03-29 15:48:18.000000000 -0700
@@ -20,10 +20,12 @@
 spicewrap.o: ext2spice.c
 	${CC} ${CFLAGS} ${CPPFLAGS} ${DFLAGS} ext2spice.c -c -o spicewrap.o
 
+# -install_name is darwin-gcc-specific
 exttospice${SHDLIB_EXT}: spicewrap.o ${MAGICDIR}/extflat/libextflat.o
 	@echo --- making exttospice Tcl library \(exttospice${SHDLIB_EXT}\)
 	${RM} exttospice${SHDLIB_EXT}
 	${CC} ${CFLAGS} ${CPPFLAGS} -o $@ ${LDDL_FLAGS} spicewrap.o \
+		-install_name ${TCLDIR}/$@ \
 		${MAGICDIR}/extflat/libextflat.o ${LD_SHARED} -lc ${LIBS}
 
 install: $(DESTDIR)${BINDIR}/${MODULE}$(PROGRAM_SUFFIX)${EXEEXT} $(DESTDIR)${BINDIR}/spice2sim$(PROGRAM_SUFFIX)
--- magic-7.4.59-orig/magic/Makefile	2008-03-29 15:50:01.000000000 -0700
+++ magic-7.4.59/magic/Makefile	2008-01-25 19:58:58.000000000 -0800
@@ -31,10 +31,12 @@
 
 tcl-main: tclmagic${SHDLIB_EXT} proto.magicrc
 
+# -install_name is darwin-gcc-specific
 tclmagic${SHDLIB_EXT}: ${EXTRA_LIBS}
 	@echo --- making magic Tcl library \(tclmagic${SHDLIB_EXT}\)
 	${RM} tclmagic${SHDLIB_EXT}
 	${CC} ${CFLAGS} ${CPPFLAGS} -o $@ ${LDDL_FLAGS} ${LD_RUN_PATH} \
+		-install_name ${TCLDIR}/$@ \
 		${EXTRA_LIBS} -lc ${LIBS}
 
 proto.magicrc: proto.magicrc.m4
