diff -Nurd -x'*~' polyxmass-bin-0.9.2.orig/src/polyxcalc-ui-chempad.c polyxmass-bin-0.9.2/src/polyxcalc-ui-chempad.c
--- polyxmass-bin-0.9.2.orig/src/polyxcalc-ui-chempad.c	2005-12-14 13:31:22.000000000 -0500
+++ polyxmass-bin-0.9.2/src/polyxcalc-ui-chempad.c	2012-06-11 03:40:55.000000000 -0400
@@ -27,6 +27,7 @@
    Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
    Boston, MA 02110-1301, USA.
 */
+#include <gtk/gtktooltips.h>
 #include "polyxcalc-globals.h"
 
 #include "polyxcalc-ui-calculator.h"
@@ -628,7 +629,7 @@
 
       g_signal_connect (G_OBJECT (button),
 			"clicked",
-			GTK_SIGNAL_FUNC 
+			G_CALLBACK
 			(polyxcalc_ui_chempad_wnd_chempadkey_button),
 			calcctxt);
     }
diff -Nurd -x'*~' polyxmass-bin-0.9.2.orig/src/polyxdef-init.c polyxmass-bin-0.9.2/src/polyxdef-init.c
--- polyxmass-bin-0.9.2.orig/src/polyxdef-init.c	2005-12-14 13:31:22.000000000 -0500
+++ polyxmass-bin-0.9.2/src/polyxdef-init.c	2012-06-11 03:27:44.000000000 -0400
@@ -133,7 +133,7 @@
 				   GTK_DIALOG_DESTROY_WITH_PARENT,
 				   GTK_MESSAGE_QUESTION,
 				   GTK_BUTTONS_YES_NO,
-				   help);
+				   "%s", help);
 
   g_free (help);
   				  
diff -Nurd -x'*~' polyxmass-bin-0.9.2.orig/src/polyxdef-ui-atomdef.c polyxmass-bin-0.9.2/src/polyxdef-ui-atomdef.c
--- polyxmass-bin-0.9.2.orig/src/polyxdef-ui-atomdef.c	2005-12-14 13:31:22.000000000 -0500
+++ polyxmass-bin-0.9.2/src/polyxdef-ui-atomdef.c	2012-06-11 03:27:32.000000000 -0400
@@ -2541,7 +2541,7 @@
 				   GTK_DIALOG_DESTROY_WITH_PARENT,
 				   GTK_MESSAGE_QUESTION,
 				   GTK_BUTTONS_YES_NO,
-				   help);
+				   "%s", help);
   g_free (help);
   
   /* Add the CANCEL button, so that the user can simply cancel 
diff -Nurd -x'*~' polyxmass-bin-0.9.2.orig/src/polyxdef-ui-polchemdef.c polyxmass-bin-0.9.2/src/polyxdef-ui-polchemdef.c
--- polyxmass-bin-0.9.2.orig/src/polyxdef-ui-polchemdef.c	2005-12-14 13:31:22.000000000 -0500
+++ polyxmass-bin-0.9.2/src/polyxdef-ui-polchemdef.c	2012-06-11 03:28:30.000000000 -0400
@@ -1290,7 +1290,7 @@
 				   GTK_DIALOG_DESTROY_WITH_PARENT,
 				   GTK_MESSAGE_QUESTION,
 				   GTK_BUTTONS_YES_NO,
-				   help);
+				   "%s", help);
   g_free (help);
   
   /* Add the CANCEL button, so that the user can simply cancel 
diff -Nurd -x'*~' polyxmass-bin-0.9.2.orig/src/polyxedit-init.c polyxmass-bin-0.9.2/src/polyxedit-init.c
--- polyxmass-bin-0.9.2.orig/src/polyxedit-init.c	2005-12-14 13:31:22.000000000 -0500
+++ polyxmass-bin-0.9.2/src/polyxedit-init.c	2012-06-11 03:29:11.000000000 -0400
@@ -165,7 +165,7 @@
 				   GTK_DIALOG_DESTROY_WITH_PARENT,
 				   GTK_MESSAGE_QUESTION,
 				   GTK_BUTTONS_YES_NO,
-				   help);
+				   "%s", help);
 
   g_free (help);
   				  
diff -Nurd -x'*~' polyxmass-bin-0.9.2.orig/src/polyxedit-ui-seqed-wnd.c polyxmass-bin-0.9.2/src/polyxedit-ui-seqed-wnd.c
--- polyxmass-bin-0.9.2.orig/src/polyxedit-ui-seqed-wnd.c	2005-12-14 13:31:22.000000000 -0500
+++ polyxmass-bin-0.9.2/src/polyxedit-ui-seqed-wnd.c	2012-06-11 03:28:51.000000000 -0400
@@ -1503,7 +1503,7 @@
 				       GTK_DIALOG_DESTROY_WITH_PARENT,
 				       GTK_MESSAGE_QUESTION,
 				       GTK_BUTTONS_YES_NO,
-				       help);
+				       "%s", help);
       g_free (help);
       
       gtk_dialog_add_buttons (GTK_DIALOG (dialog),
diff -Nurd -x'*~' polyxmass-bin-0.9.2.orig/src/polyxmass-ui-note.c polyxmass-bin-0.9.2/src/polyxmass-ui-note.c
--- polyxmass-bin-0.9.2.orig/src/polyxmass-ui-note.c	2006-02-20 13:15:00.000000000 -0500
+++ polyxmass-bin-0.9.2/src/polyxmass-ui-note.c	2012-06-11 03:32:31.000000000 -0400
@@ -2370,7 +2370,7 @@
 	      
 	      if (0 == libpolyxmass_prop_cmp 
 		  (prop, prop_iter,
-		   PXM_CMP_1_SUBSET_OF_2 | PXM_CMP_2_SUBSET_OF_1));
+		   PXM_CMP_1_SUBSET_OF_2 | PXM_CMP_2_SUBSET_OF_1))
 		{
 		  /* The iterated prop object is identical to the 
 		     one we are asked to remove from all the monomers
diff -Nurd -x'*~' polyxmass-bin-0.9.2.orig/src/polyxmass-ui-seqed-widget-context-menu.c polyxmass-bin-0.9.2/src/polyxmass-ui-seqed-widget-context-menu.c
--- polyxmass-bin-0.9.2.orig/src/polyxmass-ui-seqed-widget-context-menu.c	2005-12-17 09:14:59.000000000 -0500
+++ polyxmass-bin-0.9.2/src/polyxmass-ui-seqed-widget-context-menu.c	2012-06-11 03:38:59.000000000 -0400
@@ -27,6 +27,7 @@
    Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
    Boston, MA 02110-1301, USA.
 */
+#include <gtk/gtktooltips.h>
 #include "polyxmass-ui-seqed-widget.h"
 #include "polyxmass-ui-seqed-widget-context-menu.h"
 #include "polyxmass-ui-seqed-widget-kbd.h"
