diff -urN qprop-1.16-1/Qprop/bin/Makefile qprop-1.16-1/Qprop/bin/Makefile
--- Qprop/bin/Makefile	2005-12-02 20:38:33.000000000 -0500
+++ Qprop/bin/Makefile	2005-12-02 20:34:51.000000000 -0500
@@ -15,8 +15,8 @@
 #FTNLIB = -Vaxlib /usr/lib/C-ctype.o /usr/lib/C_name.o /usr/lib/ctype-info.o
 
 #------------------------------------
-#FC = g77
-#FFLAGS = -O -dbl
+FC = gfortran
+FFLAGS = -O3 -fdefault-real-8
 
 #------------------------------------
 

 	  	 
