diff -uNr rasmol-2.7.5-23Jul09/src/Imakefile rasmol-2.7.5-23Jul09.patched/src/Imakefile
--- rasmol-2.7.5-23Jul09/src/Imakefile	2009-07-20 10:08:43.000000000 -0400
+++ rasmol-2.7.5-23Jul09.patched/src/Imakefile	2009-08-10 09:27:00.000000000 -0400
@@ -101,6 +101,7 @@
 # USER DEFINITIONS
 #
 #  At this point, you may define the cpp macro PIXELDEPTH as 8, 16 or 32
+#define PIXELDEPTH 8
 
 #ifdef PIXELDEPTH
 #if PIXELDEPTH==8
@@ -276,7 +277,7 @@
 
 # RasMol's on-line help direcory
 #   e.g. /usr/local/lib/rasmol/rasmol.hlp
-RASMOLDIR = $(USRLIBDIR)/rasmol/
+RASMOLDIR = @PREFIX@/lib/rasmol/
 
 
 #ifndef USE_XFORMSLIB
diff -uNr rasmol-2.7.5-23Jul09/src/rasmol.h rasmol-2.7.5-23Jul09.debian/src/rasmol.h
--- rasmol-2.7.5-23Jul09/src/rasmol.h	2009-07-20 10:08:43.000000000 -0400
+++ rasmol-2.7.5-23Jul09.debian/src/rasmol.h	2009-08-11 12:27:48.000000000 -0400
@@ -78,6 +78,7 @@
 /* rasmol.h
  */
 
+#include <stdint.h>
 
 /*===========================*/
 /*  User Definable Options!  */
@@ -202,7 +203,7 @@
 
 typedef double Real;
 #ifndef APPLEMAC
-typedef unsigned char Byte;
+typedef uint8_t Byte;
 #endif
 
 #ifdef __STDC__
@@ -211,21 +212,18 @@
 typedef char Char;
 #endif
 
-#ifdef _LONGLONG
-typedef unsigned int Card;
-typedef int Long;
-#else
-typedef unsigned long Card;
+typedef uint32_t Card;
+/* Long has to be defined as long, instead of int32_t because of
+   multiple printf and scanf calls using %ld in the format. */
 typedef long Long;
-#endif
 
 #ifdef EIGHTBIT
-typedef unsigned char Pixel;
+typedef uint8_t Pixel;
 #else
 #ifdef THIRTYTWOBIT
-typedef Long Pixel;
+typedef int32_t Pixel;
 #else
-typedef short Pixel;
+typedef int16_t Pixel;
 #endif
 #endif
 
@@ -249,7 +247,6 @@
 #define ToUpper(x)   (islower((x))?toupper((x)):(x))
 #endif
 
-
 #if !defined(IBMPC) || defined(_WIN32)
 #ifdef APPLEMAC
 #define _fmalloc   NewPtrSys
