Index: configure
===================================================================
--- configure	(revision 24552)
+++ configure	(working copy)
@@ -2730,21 +2730,21 @@
     # for a system library, then see if we have various headers needed.
     if test "x$enable_builtin_afterimage" = "xyes" ; then
        check_header "jpeglib.h" "" $ASIMAGE $ASIMAGE/include \
-          /usr/local/include /usr/include /opt/include
+          $finkdir/include /usr/local/include /usr/include /opt/include
        asjpegincdir=$found_dir
        check_header "png.h" "" $ASIMAGE $ASIMAGE/include $ASPNG/include \
-          /usr/local/include /usr/include /usr/local/include/libpng \
+          $finkdir/include /usr/local/include /usr/include /usr/local/include/libpng \
           /opt/include
        aspngincdir=$found_dir
        if test ! "x$enable_astiff" = "xno" ; then
            check_header "tiffio.h" "" $ASIMAGE $ASIMAGE/include \
-               /usr/local/include /usr/include /opt/include
+               $finkdir/include /usr/local/include /usr/include /opt/include
            astiffincdir=$found_dir
        else
            astiffincdir="--with-tiff=no"
        fi
        check_header "gif_lib.h" "" $ASIMAGE $ASIMAGE/include \
-          /usr/local/include /usr/include /opt/include
+          $finkdir/include /usr/local/include /usr/include /opt/include
        asgifincdir=$found_dir
 
        asextralib=""
@@ -2762,7 +2762,7 @@
        fi
        for k in $aslibs ; do
            check_library $k "$enable_shared" "" \
-             $ASIMAGE $ASIMAGE/lib $ASPNG/lib /usr/local/lib /usr/lib /opt/lib
+             $ASIMAGE $ASIMAGE/lib $ASPNG/lib $finkdir/lib /usr/local/lib /usr/lib /opt/lib
            if test "x$k" = "xlibz" ; then
                found_libz=$found_lib
                found_dirz=$found_dir
Index: core/base/src/TSystem.cxx
===================================================================
--- core/base/src/TSystem.cxx	(revision 24552)
+++ core/base/src/TSystem.cxx	(working copy)
@@ -29,6 +29,7 @@
 #include <stdlib.h>
 #include <errno.h>
 
+#include "RConfigure.h"
 #include "Riostream.h"
 #include "TSystem.h"
 #include "TApplication.h"
Index: cint/ROOT/cintdlls.mk
===================================================================
--- cint/ROOT/cintdlls.mk	(revision 24552)
+++ cint/ROOT/cintdlls.mk	(working copy)
@@ -158,6 +158,11 @@
   $(CINTDLLS): MAKELIB := $(subst -x,,$(MAKELIB))
 endif
 
+# Filter out version information
+ifneq ($(subst build/unix/makelib.sh,,$(MAKELIB)),$(MAKELIB))
+  $(CINTDLLS): MAKELIB := $(filter-out $(MKLIBOPTIONS),$(MAKELIB))
+endif
+
 $(ALLCINTDLLS): $(ORDER_) $(MAINLIBS)
 
 $(CINTDLLDIRSTL)/%.dll: $(CINTDLLDIRDLLSTL)/G__cpp_%.o 
Index: net/xrootd/src/xrootd/src/XrdPosix/GNUmakefile
===================================================================
--- net/xrootd/src/xrootd/src/XrdPosix/GNUmakefile	(revision 24857)
+++ net/xrootd/src/xrootd/src/XrdPosix/GNUmakefile	(working copy)
@@ -88,7 +88,7 @@
 $(LIBRARY): $(OBJECTP) $(OBJECTS) $(OBJECTT) $(LIBDEP)
 	@echo Creating shared library $(LIBRARY) 
 	$(ECHO)$(CC) $(CFLAGS) $(OBJECTS) $(OBJECTU) $(LDSO) $(MORELIBS) $(LIBS) -o $(LIBRARY)
-ifneq ($(subst $(MACOSX_MINOR),,1234),1234)
+ifneq ($(subst $(MACOSX_MINOR),,12345),12345)
 	$(ECHO)if [ "x$(TYPE)" = "xDarwin" ]; then \
 	   $(CC) $(CFLAGS) $(OBJECTS) $(TYPELDDY) $(MORELIBS) $(LIBS) -o $(LIBDIR)/libXrdPosix.$(TYPEDYLIB); \
 	fi
Index: net/xrootd/src/xrootd/src/XrdClient/GNUmakefile
===================================================================
--- net/xrootd/src/xrootd/src/XrdClient/GNUmakefile	(revision 24857)
+++ net/xrootd/src/xrootd/src/XrdClient/GNUmakefile	(working copy)
@@ -155,7 +155,7 @@
 	@echo Creating shared library $(LIBSHARED) 
 	$(ECHO)rm -f $(LIBSHARED)
 	$(ECHO)$(CC) $(CFLAGS) $(OBJECTS) $(LDSO) $(LIBLIBS) $(LIBS) -o $(LIBSHARED)
-ifneq ($(subst $(MACOSX_MINOR),,1234),1234)
+ifneq ($(subst $(MACOSX_MINOR),,12345),12345)
 	$(ECHO)if [ "x$(TYPE)" = "xDarwin" ]; then \
 	          $(CC) $(CFLAGS) $(OBJECTS) $(TYPELDDY) $(LIBLIBS) $(LIBS) -o $(LIBDIR)/libXrdClient.$(TYPEDYLIB); \
 	fi
Index: net/xrootd/src/xrootd/config/GNUmake.rules.macos
===================================================================
--- net/xrootd/src/xrootd/config/GNUmake.rules.macos	(revision 24857)
+++ net/xrootd/src/xrootd/config/GNUmake.rules.macos	(working copy)
@@ -20,11 +20,11 @@
 TYPELINK   = -undefined dynamic_lookup -multiply_defined suppress 
 TYPEOPT    = $(TYPEMISC) $(TYPELINK) -O2
 TYPEDBG    = $(TYPEMISC) $(TYPELINK) -g
-ifeq ($(subst $(MACOSX_MINOR),,1234),1234)
+ifeq ($(subst $(MACOSX_MINOR),,123),123)
 TYPELDSO   = $(TYPELINK) -dynamiclib -single_module
 TYPELDDY   = $(TYPELINK) -dynamiclib -single_module
-TYPESHLIB  = so
-TYPEDYLIB  = so
+TYPESHLIB  = dylib
+TYPEDYLIB  = dylib
 else
 TYPELDSO   = $(TYPELINK) -bundle
 TYPELDDY   = $(TYPELINK) -dynamiclib -single_module
Index: net/xrootd/Module.mk
===================================================================
--- net/xrootd/Module.mk	(revision 24857)
+++ net/xrootd/Module.mk	(working copy)
@@ -29,7 +29,7 @@
 endif
 endif
 ifeq ($(PLATFORM),macosx)
-XRDSOEXT    = so
+XRDSOEXT    = dylib
 else
 XRDSOEXT    = $(SOEXT)
 endif
Index: build/unix/makelib.sh
===================================================================
--- build/unix/makelib.sh	(revision 24552)
+++ build/unix/makelib.sh	(working copy)
@@ -34,7 +34,7 @@
 if [ $PLATFORM = "macosx" ]; then
    macosx_minor=`sw_vers | sed -n 's/ProductVersion://p' | cut -d . -f 2`
    if [ $macosx_minor -ge 5 ] && [ $LD != "icpc" ]; then
-      soext="so"
+      soext="dylib"
    else
       soext="dylib"
    fi
@@ -209,7 +209,7 @@
 	ln -fs $SONAME.$MAJOR        $LIB
     elif [ -f $LIBVERS ]; then
 	# Versioned library has format foo.3.05.so
-	source_file=`echo $SONAME | sed "s/.*\./&${MINOR}./"`
+	source_file=`/usr/bin/basename $SONAME | sed "s/.*\./&${MINOR}./"`
 	LIBNOMAJORMINOR=`echo $LIB|sed 's,\.'$MAJOR'\.'$MINOR',,'`
 	if [ $LIB != $LIBNOMAJORMINOR ]; then
 	    LIBNOMINOR=`echo $LIB|sed 's,\.'$MINOR',,'`
Index: config/Makefile.macosx
===================================================================
--- config/Makefile.macosx	(revision 24857)
+++ config/Makefile.macosx	(working copy)
@@ -89,7 +89,7 @@
 endif
 endif
 ifeq ($(subst $(MACOSX_MINOR),,1234),1234)
-SOEXT         = so
+SOEXT         = dylib
 else
 SOEXT         = dylib
 endif
@@ -101,21 +101,14 @@
 CILIBS        = -lm $(EXTRA_LDFLAGS) $(FINK_LDFLAGS) -ldl
 
 # Fortran:
-F77           =
-ifeq ($(subst $(MACOSX_MINOR),,123),123)
-ifeq ($(shell which gfortran 2>&1 | sed -ne "s/.*\/gfortran/gfortran/p"),gfortran)
-F77          := gfortran
+F77           = @FORTRAN@
+ifeq ($(F77),gfortran)
 F77LIBS      := $(shell $(F77) -print-file-name=libgfortran.dylib)
 F77LIBS      += $(shell $(F77) -print-file-name=libgfortranbegin.a)
 endif
-endif
-
-ifeq ($(F77),)
-ifeq ($(shell which g77 2>&1 | sed -ne "s/.*\/g77/g77/p"),g77)
-F77          := g77
+ifeq ($(F77),g77)
 F77LIBS      := -L$(FINK_DIR)/lib -lg2c
 endif
-endif
 
 # Override default in Makefile.config
 INSTALL      := /bin/cp -pPR
Index: config/Makefile.depend
===================================================================
--- config/Makefile.depend	(revision 24857)
+++ config/Makefile.depend	(working copy)
@@ -322,88 +322,105 @@
 MATRIXLIBEXTRA          = -Llib -lMathCore
 HISTLIBEXTRA            = -Llib -lMatrix -lMathCore
 GRAFLIBEXTRA            = -Llib -lHist -lMatrix -lRIO -lMathCore
-GPADLIBEXTRA            = -Llib -lGraf -lHist
-G3DLIBEXTRA             = -Llib -lGraf -lHist -lGpad -lMathCore
-GEDLIBEXTRA             = -Llib -lHist -lGpad -lGraf -lGui -lTree -lTreePlayer
-POSTSCRIPTLIBEXTRA      = -Llib -lGraf
-GUILIBEXTRA             = -Llib -lGpad -lGraf -lRIO -lMathCore
-HISTPAINTERLIBEXTRA     = -Llib -lGraf -lHist -lMatrix -lGpad -lMathCore
-SPECTRUMPAINTERLIBEXTRA = -Llib -lGraf -lHist
-HTMLLIBEXTRA            = -Llib -lGraf -lThread
-MINUITLIBEXTRA          = -Llib -lGraf -lHist -lMatrix -lMathCore
-MINUIT2LIBEXTRA         = -Llib -lGraf -lHist -lMatrix -lMathCore
-FUMILILIBEXTRA          = -Llib -lGraf -lHist -lMathCore
-TREELIBEXTRA            = -Llib -lNet -lRIO -lThread
-TREEPLAYERLIBEXTRA      = -Llib -lTree -lGraf3d -lGraf -lHist -lGpad -lRIO -lMathCore
+GPADLIBEXTRA            = -Llib -lGraf -lHist -lMatrix -lRIO -lMathCore
+G3DLIBEXTRA             = -Llib -lGraf -lHist -lGpad -lMathCore -lMatrix -lRIO
+GEDLIBEXTRA             = -Llib -lHist -lGpad -lGraf -lGui -lTree -lTreePlayer \
+                          -lMatrix -lMathCore -lRIO -lNet -lThread -lGraf3d
+POSTSCRIPTLIBEXTRA      = -Llib -lGraf -lHist -lMatrix -lRIO -lMathCore
+GUILIBEXTRA             = -Llib -lGpad -lGraf -lRIO -lMathCore -lHist -lMatrix
+HISTPAINTERLIBEXTRA     = -Llib -lGraf -lHist -lMatrix -lGpad -lMathCore -lRIO
+SPECTRUMPAINTERLIBEXTRA = -Llib -lGraf -lHist -lMatrix -lRIO -lMathCore
+HTMLLIBEXTRA            = -Llib -lGraf -lThread -lHist -lMatrix -lRIO -lMathCore
+MINUITLIBEXTRA          = -Llib -lGraf -lHist -lMatrix -lMathCore -lRIO
+MINUIT2LIBEXTRA         = -Llib -lGraf -lHist -lMatrix -lMathCore -lRIO
+FUMILILIBEXTRA          = -Llib -lGraf -lHist -lMathCore -lMatrix -lRIO
+TREELIBEXTRA            = -Llib -lNet -lRIO -lThread -lMathCore
+TREEPLAYERLIBEXTRA      = -Llib -lTree -lGraf3d -lGraf -lHist -lGpad -lRIO -lMathCore \
+                          -lNet -lThread -lMatrix
 TREEVIEWERLIBEXTRA      = -Llib -lTree -lGpad -lGraf -lHist -lGui -lTreePlayer \
-                          -lGed -lRIO -lMathCore
+                          -lGed -lRIO -lMathCore -lNet -lThread -lMatrix -lGraf3d
 PROOFLIBEXTRA           = -Llib -lNet -lTree -lThread -lRIO -lMathCore
-PROOFPLAYERLIBEXTRA     = -Llib -lProof -lHist -lTree -lRIO -lNet -lThread -lMathCore
+PROOFPLAYERLIBEXTRA     = -Llib -lProof -lHist -lTree -lRIO -lNet -lThread -lMathCore \
+                          -lMatrix
 PROOFDRAWLIBEXTRA       = -Llib -lTreePlayer -lGraf3d -lGraf -lGpad -lProofPlayer \
-                          -lHist -lTree -lProof
-PROOFXLIBEXTRA          = -Llib -lNet -lProof -lThread
-SESSIONVIEWERLIBEXTRA   = -Llib -lProof -lGui -lGpad -lGraf -lTree -lMathCore
+                          -lHist -lTree -lProof -lRIO -lMathCore -lNet -lThread -lMatrix
+PROOFXLIBEXTRA          = -Llib -lNet -lProof -lThread -lRIO -lMathCore -lTree
+SESSIONVIEWERLIBEXTRA   = -Llib -lProof -lGui -lGpad -lGraf -lTree -lMathCore -lNet \
+                          -lThread -lRIO -lHist -lMatrix
 PEACLIBEXTRA            = -Llib -lProof -lClarens
 PEACGUILIBEXTRA         = -Llib -lGui
-X3DLIBEXTRA             = -Llib -lGraf3d -lGui
-EGLIBEXTRA              = -Llib -lGraf3d -lGraf -lGpad -lMathCore
-VMCLIBEXTRA             = -Llib -lEG -lGeom -lMathCore
+X3DLIBEXTRA             = -Llib -lGraf3d -lGui -lGraf -lHist -lGpad -lMathCore -lMatrix -lRIO
+EGLIBEXTRA              = -Llib -lGraf3d -lGraf -lGpad -lMathCore -lHist -lMatrix -lRIO
+VMCLIBEXTRA             = -Llib -lEG -lGeom -lMathCore -lGraf3d -lGraf -lGpad -lHist \
+                          -lMatrix -lRIO
 PHYSICSLIBEXTRA         = -Llib -lMatrix -lMathCore
-PYTHIA6LIBEXTRA         = -Llib -lEG -lGraf -lVMC -lPhysics $(FPYTHIA6LIBDIR) \
-                          $(FPYTHIA6LIB)
-PYTHIA8LIBEXTRA         = -Llib -lEG -lGraf -lVMC -lPhysics $(FPYTHIA8LIBDIR) \
-                          $(FPYTHIA8LIB)
-X11TTFLIBEXTRA          = -Llib -lGX11 -lGraf
-TABLELIBEXTRA           = -Llib -lTree -lGpad -lGraf3d -lGraf -lHist -lRIO -lMathCore
-GLLIBEXTRA              = -Llib -lGpad -lGraf3d -lGui -lGraf -lHist -lGed -lMathCore
+PYTHIA6LIBEXTRA         = -Llib -lEG -lGraf -lVMC -lPhysics \
+                          -lGraf3d -lGpad -lMathCore -lHist -lMatrix -lRIO \
+                          $(FPYTHIA6LIBDIR) $(FPYTHIA6LIB)
+PYTHIA8LIBEXTRA         = -Llib -lEG -lGraf -lVMC -lPhysics -lGeom \
+                          -lGraf3d -lGpad -lMathCore -lHist -lMatrix -lRIO \
+                          $(FPYTHIA8LIBDIR) $(FPYTHIA8LIB)
+X11TTFLIBEXTRA          = -Llib -lGX11 -lGraf -lHist -lMatrix -lRIO -lMathCore
+TABLELIBEXTRA           = -Llib -lTree -lGpad -lGraf3d -lGraf -lHist -lRIO -lMathCore \
+                          -lNet -lThread -lMatrix
+GLLIBEXTRA              = -Llib -lGpad -lGraf3d -lGui -lGraf -lHist -lGed -lMathCore \
+                          -lMatrix -lRIO -lTree -lTreePlayer -lNet -lThread
 HBOOKLIBEXTRA           = -Llib -lHist -lMatrix -lTree -lGraf -lTreePlayer \
-                          -lRIO -lminicern
+                          -lRIO -lminicern -lMathCore -lNet -lThread -lGraf3d -lGpad
 GEOMLIBEXTRA            = -Llib -lRIO -lMathCore
-GEOMPAINTERLIBEXTRA     = -Llib -lGeom -lTree -lGraf3d -lHist -lGpad -lRIO -lMathCore
-GEOMBUILDERLIBEXTRA     = -Llib -lGeom -lGraf3d -lGpad -lGraf -lGui -lGed
-ASIMAGELIBEXTRA         = -Llib -lGraf -lMathCore
-ASIMAGEGUILIBEXTRA      = -Llib -lGraf -lHist -lGui -lASImage -lRIO
+GEOMPAINTERLIBEXTRA     = -Llib -lGeom -lTree -lGraf3d -lHist -lGpad -lRIO -lMathCore \
+                          -lNet -lThread -lGraf -lMatrix
+GEOMBUILDERLIBEXTRA     = -Llib -lGeom -lGraf3d -lGpad -lGraf -lGui -lGed -lRIO \
+                          -lMathCore -lHist -lMatrix -lTree -lTreePlayer -lNet -lThread
+ASIMAGELIBEXTRA         = -Llib -lGraf -lMathCore -lHist -lMatrix -lRIO
+ASIMAGEGUILIBEXTRA      = -Llib -lGraf -lHist -lGui -lASImage -lRIO -lMatrix -lMathCore \
+                          -lGpad
 ASIMAGEGSLIBEXTRA       = -Llib -lGraf -lASImage
 MLPLIBEXTRA             = -Llib -lHist -lMatrix -lTree -lGraf -lGpad \
-                          -lTreePlayer -lMathCore
-SPECTRUMLIBEXTRA        = -Llib -lHist -lMatrix
+                          -lTreePlayer -lMathCore -lNet -lRIO -lThread -lGraf3d
+SPECTRUMLIBEXTRA        = -Llib -lHist -lMatrix -lMathCore
 TMVALIBEXTRA            = -Llib -lRIO -lHist -lMatrix -lTree -lGraf -lGpad \
-                          -lTreePlayer -lMLP -lMinuit -lMathCore
-SPLOTLIBEXTRA           = -Llib -lMatrix -lHist -lTree -lTreePlayer -lGraf3d -lGraf
-QTROOTLIBEXTRA          = -Llib -lGui -lGQt
-GQTLIBEXTRA             = -Llib -lGui -lGpad -lGraf -lRint
-QUADPLIBEXTRA           = -Llib -lMatrix
-RUBYLIBEXTRA            = -Llib -lHist -lMathCore
-GUIBLDLIBEXTRA          = -Llib -lGui -lGraf -lMathCore
-GUIHTMLLIBEXTRA         = -Llib -lGui -lGraf -lNet
+                          -lTreePlayer -lMLP -lMinuit -lMathCore -lNet -lThread -lGraf3d
+SPLOTLIBEXTRA           = -Llib -lMatrix -lHist -lTree -lTreePlayer -lGraf3d -lGraf \
+                          -lMathCore -lNet -lRIO -lThread -lGpad
+QTROOTLIBEXTRA          = -Llib -lGui -lGQt -lGpad -lGraf -lRIO -lMathCore -lHist -lMatrix -lRint
+GQTLIBEXTRA             = -Llib -lGui -lGpad -lGraf -lRint -lRIO -lMathCore -lHist -lMatrix
+QUADPLIBEXTRA           = -Llib -lMatrix -lMathCore
+RUBYLIBEXTRA            = -Llib -lHist -lMathCore -lMatrix
+GUIBLDLIBEXTRA          = -Llib -lGui -lGraf -lMathCore -lGpad -lRIO -lHist -lMatrix
+GUIHTMLLIBEXTRA         = -Llib -lGui -lGraf -lNet -lGpad -lRIO -lMathCore -lHist -lMatrix
 XMLLIBEXTRA             = -Llib -lRIO
-FOAMLIBEXTRA            = -Llib -lHist
+FOAMLIBEXTRA            = -Llib -lHist -lMatrix -lMathCore
 ALIENLIBEXTRA           = -Llib -lXMLIO -lNetx -lTree -lProofPlayer -lNet -lRIO
-ROOFITCORELIBEXTRA      = -Llib -lHist -lGraf -lMatrix -lTree -lMinuit -lRIO -lMathCore
-ROOFITLIBEXTRA          = -Llib -lRooFitCore -lTree -lRIO -lMatrix -lMathCore
+ROOFITCORELIBEXTRA      = -Llib -lHist -lGraf -lMatrix -lTree -lMinuit -lRIO -lMathCore -lNet \
+                          -lThread
+ROOFITLIBEXTRA          = -Llib -lRooFitCore -lTree -lRIO -lMatrix -lMathCore -lHist -lGraf \
+                          -lMinuit -lNet -lThread
 CINT7LIBEXTRA           = -Llib -lReflex
 CINTEXLIBEXTRA          = -Llib -lReflex
 REFLEXDICTLIBEXTRA      = -Llib -lReflex
-RAUTHLIBEXTRA           = -Llib -lNet -lRIO
-KRB5AUTHLIBEXTRA        = -Llib -lRootAuth -lNet
-SRVAUTHLIBEXTRA         = -Llib -lNet
-QTGSILIBEXTRA           = -Llib -lGui -lGpad
-FITPANELLIBEXTRA        = -Llib -lGui -lGpad -lHist -lGraf -lTree -lMathCore
+RAUTHLIBEXTRA           = -Llib -lNet -lRIO -lMathCore
+KRB5AUTHLIBEXTRA        = -Llib -lRootAuth -lNet -lRIO -lMathCore
+SRVAUTHLIBEXTRA         = -Llib -lNet -lRIO -lMathCore
+QTGSILIBEXTRA           = -Llib -lGui -lGpad -lGraf -lRIO -lMathCore -lHist -lMatrix
+FITPANELLIBEXTRA        = -Llib -lGui -lGpad -lHist -lGraf -lTree -lMathCore -lRIO \
+                          -lMatrix -lNet -lThread
 RFIOLIBEXTRA            = -Llib -lRIO
-SQLLIBEXTRA             = -Llib -lNet -lRIO
-ODBCLIBEXTRA            = -Llib -lNet -lRIO
-NETXLIBEXTRA            = -Llib -lNet -lRIO -lThread
+SQLLIBEXTRA             = -Llib -lNet -lRIO -lMathCore
+ODBCLIBEXTRA            = -Llib -lNet -lRIO -lMathCore
+NETXLIBEXTRA            = -Llib -lNet -lRIO -lThread -lMathCore
 ORACLELIBEXTRA          = -Llib -lNet -lRIO
 CASTORLIBEXTRA          = -Llib -lNet -lRIO
 PGSQLLIBEXTRA           = -Llib -lNet -lRIO
-MYSQLLIBEXTRA           = -Llib -lNet -lRIO
-GDMLLIBEXTRA            = -Llib -lGeom -lXMLIO -lHist -lRIO
-UNURANLIBEXTRA          = -Llib -lHist -lMathCore
+MYSQLLIBEXTRA           = -Llib -lNet -lRIO -lMathCore
+GDMLLIBEXTRA            = -Llib -lGeom -lXMLIO -lHist -lRIO -lMathCore -lMatrix
+UNURANLIBEXTRA          = -Llib -lHist -lMathCore -lMatrix
 MONALISALIBEXTRA        = -Llib -lNet -lRIO
 THREADLIBEXTRA          = -Llib -lRIO
 EVELIBEXTRA             = -Llib -lGeom -lGeomPainter -lGraf3d -lGui -lGpad \
                           -lGraf -lHist -lPhysics -lGed -lEG -lTree \
-                          -lTreePlayer -lRGL -lRIO -lRint -lMathCore
+                          -lTreePlayer -lRGL -lRIO -lRint -lMathCore \
+                          -lNet -lThread -lMatrix
 MEMSTATLIBEXTRA         = -Llib -lTree
 MEMSTATGUILIBEXTRA      = -Llib -lMemStat -lTree -lGui
 
