diff -ruN rstudio-0.94.82-orig/CMakeGlobals.txt rstudio-0.94.82/CMakeGlobals.txt
--- rstudio-0.94.82-orig/CMakeGlobals.txt	2014-02-20 15:49:49.000000000 +0900
+++ rstudio-0.94.82/CMakeGlobals.txt	2014-02-20 15:50:50.000000000 +0900
@@ -13,17 +13,17 @@
 
 # version info
 if ("$ENV{RSTUDIO_VERSION_MAJOR}" STREQUAL "")
-  set(CPACK_PACKAGE_VERSION_MAJOR "99")
+  set(CPACK_PACKAGE_VERSION_MAJOR "0")
 else()
   set(CPACK_PACKAGE_VERSION_MAJOR $ENV{RSTUDIO_VERSION_MAJOR})
 endif()
 if ("$ENV{RSTUDIO_VERSION_MINOR}" STREQUAL "")
-  set(CPACK_PACKAGE_VERSION_MINOR "9")
+  set(CPACK_PACKAGE_VERSION_MINOR "94")
 else()
   set(CPACK_PACKAGE_VERSION_MINOR $ENV{RSTUDIO_VERSION_MINOR})
 endif()
 if ("$ENV{RSTUDIO_VERSION_PATCH}" STREQUAL "")
-  set(CPACK_PACKAGE_VERSION_PATCH "9")
+  set(CPACK_PACKAGE_VERSION_PATCH "82")
 else()
   set(CPACK_PACKAGE_VERSION_PATCH $ENV{RSTUDIO_VERSION_PATCH})
 endif()
diff -ruN rstudio-0.94.82-orig/cmake/modules/FindLibR.cmake rstudio-0.94.82/cmake/modules/FindLibR.cmake
--- rstudio-0.94.82-orig/cmake/modules/FindLibR.cmake	2014-02-20 15:49:49.000000000 +0900
+++ rstudio-0.94.82/cmake/modules/FindLibR.cmake	2014-02-20 15:52:07.000000000 +0900
@@ -22,8 +22,8 @@
 
    find_library(LIBR_LIBRARIES R)
    if(LIBR_LIBRARIES)
-      set(LIBR_HOME "${LIBR_LIBRARIES}/Resources" CACHE PATH "R home directory")
-      set(LIBR_INCLUDE_DIRS "${LIBR_HOME}/include" CACHE PATH "R include directory")
+      set(LIBR_HOME "@PREFIX@/Library/Frameworks/R.framework/Resources" CACHE PATH "R home directory")
+      set(LIBR_INCLUDE_DIRS "@PREFIX@/Library/Frameworks/R.framework/Versions/2.15/Headers" CACHE PATH "R include directory")
       set(LIBR_DOC_DIR "${LIBR_HOME}/doc" CACHE PATH "R doc directory")
    endif()
 
diff -ruN rstudio-0.94.82-orig/src/cpp/core/r_util/REnvironmentPosix.cpp rstudio-0.94.82/src/cpp/core/r_util/REnvironmentPosix.cpp
--- rstudio-0.94.82-orig/src/cpp/core/r_util/REnvironmentPosix.cpp	2014-02-20 15:49:50.000000000 +0900
+++ rstudio-0.94.82/src/cpp/core/r_util/REnvironmentPosix.cpp	2014-02-20 15:53:04.000000000 +0900
@@ -78,9 +78,7 @@
 {
    // define potential paths
    std::vector<std::string> rScriptPaths;
-   rScriptPaths.push_back("/usr/bin/R");
-   rScriptPaths.push_back("/usr/local/bin/R");
-   rScriptPaths.push_back("/opt/local/bin/R");
+   rScriptPaths.push_back("@PREFIX@/bin/R");
    return scanForRScript(rScriptPaths, pErrMsg);
 }
 
@@ -140,7 +138,7 @@
                                     std::string* pErrMsg)
 {
    // home path
-   *pHomePath = FilePath("/Library/Frameworks/R.framework/Resources");
+   *pHomePath = FilePath("@PREFIX@/Library/Frameworks/R.framework/Resources");
 
    // lib path
    std::string rLibPath;
diff -ruN rstudio-0.94.82-orig/src/cpp/desktop/CMakeLists.txt rstudio-0.94.82/src/cpp/desktop/CMakeLists.txt
--- rstudio-0.94.82-orig/src/cpp/desktop/CMakeLists.txt	2014-02-20 15:49:50.000000000 +0900
+++ rstudio-0.94.82/src/cpp/desktop/CMakeLists.txt	2014-02-20 15:54:24.000000000 +0900
@@ -251,7 +251,7 @@
             \"\${CMAKE_INSTALL_PREFIX}/${QTPLUGINS_DEST_DIR}/plugins/*${CMAKE_SHARED_LIBRARY_SUFFIX}\")
           include(BundleUtilities)
           file(REMOVE \"\${CMAKE_INSTALL_PREFIX}/RStudio.app/Contents/MacOS/rsession\")
-          fixup_bundle(\"${APPS}\" \"\${QTPLUGINS}\" \"${DIRS}\")
+          # fixup_bundle(\"${APPS}\" \"\${QTPLUGINS}\" \"${DIRS}\")
           ")
 
    # copy qt shared objects directly to bin on unix (fixed up rpath above)
