diff -ru stellarium-0.6.2.orig/configure stellarium-0.6.2/configure
--- stellarium-0.6.2.orig/configure	2004-11-18 17:54:06.000000000 -0500
+++ stellarium-0.6.2/configure	2006-02-01 18:10:57.000000000 -0500
@@ -5653,7 +5653,7 @@
 CFLAGS="$CFLAGS $X_CFLAGS"
 
 case "$host" in
-*-*-darwin*)
+*-*-skip*)
 SDL_VERSION=1.2.0
 ;;
 
@@ -6359,7 +6359,7 @@
 
 
 case "$host" in
-*-*-darwin*)
+*-*-skip*)
 PKGDATADIR="Stellarium.app/Contents/Resources"
 ;;
 *)
@@ -6384,7 +6384,7 @@
 #define MACOSX 1
 _ACEOF
 
-	LDFLAGS="$LDFLAGS -flat_namespace -undefined suppress -lz "
+	LDFLAGS="$LDFLAGS -flat_namespace -undefined suppress -lmx -lz "
 	LIBS="$LIBS -framework OpenGL -lobjc "
 ;;
 
diff -ru stellarium-0.6.2.orig/src/main.cpp stellarium-0.6.2/src/main.cpp
--- stellarium-0.6.2.orig/src/main.cpp	2004-10-08 20:14:04.000000000 -0400
+++ stellarium-0.6.2/src/main.cpp	2006-02-01 18:10:57.000000000 -0500
@@ -142,7 +142,7 @@
 
 	// If the system is non unix (windows) or if it's macosx the config file is in the
 	// config/ directory of the dataRoot directory
-#if defined(WIN32) || defined(CYGWIN) || defined(__MINGW32__) || defined(MACOSX)
+#if defined(WIN32) || defined(CYGWIN) || defined(__MINGW32__)
 	CDIR = DATA_ROOT + "/config/";
 
 	if ((tempFile = fopen((CDIR + "config.ini").c_str(),"r")))
diff -ru stellarium-0.6.2.orig/src/vecmath.h stellarium-0.6.2/src/vecmath.h
--- stellarium-0.6.2.orig/src/vecmath.h	2004-10-08 20:14:04.000000000 -0400
+++ stellarium-0.6.2/src/vecmath.h	2006-02-01 18:22:36.000000000 -0500
@@ -41,7 +41,7 @@
 
 typedef Vector4<double>	Vec4d;
 typedef Vector4<float>	Vec4f;
-typedef Vector4<int>	Vec4i;
+typedef Vector4<long>	Vec4i;
 
 typedef Matrix4<float>	Mat4f;
 typedef Matrix4<double>	Mat4d;
@@ -73,8 +73,8 @@
     inline Vector2 operator-(const Vector2<T>&) const;
     inline Vector2 operator+(const Vector2<T>&) const;
 
-	inline Vector2 Vector2<T>::operator-() const;
-	inline Vector2 Vector2<T>::operator+() const;
+	inline Vector2 operator-() const;
+	inline Vector2 operator+() const;
 
 	inline Vector2 operator*(T) const;
 	inline Vector2 operator/(T) const;
@@ -119,8 +119,8 @@
     inline Vector3 operator-(const Vector3<T>&) const;
     inline Vector3 operator+(const Vector3<T>&) const;
 
-	inline Vector3 Vector3<T>::operator-() const;
-	inline Vector3 Vector3<T>::operator+() const;
+	inline Vector3 operator-() const;
+	inline Vector3 operator+() const;
 
 	inline Vector3 operator*(T) const;
 	inline Vector3 operator/(T) const;
@@ -168,8 +168,8 @@
     inline Vector4 operator-(const Vector4<T>&) const;
     inline Vector4 operator+(const Vector4<T>&) const;
 
-	inline Vector4 Vector4<T>::operator-() const;
-	inline Vector4 Vector4<T>::operator+() const;
+	inline Vector4 operator-() const;
+	inline Vector4 operator+() const;
 
 	inline Vector4 operator*(T) const;
 	inline Vector4 operator/(T) const;
