diff -ruN XCrySDen-1.5.24-src-orig/system/Make.macosx-x11 XCrySDen-1.5.24-src/system/Make.macosx-x11
--- XCrySDen-1.5.24-src-orig/system/Make.macosx-x11	2008-12-22 10:44:52.000000000 -0500
+++ XCrySDen-1.5.24-src/system/Make.macosx-x11	2012-01-10 20:49:09.000000000 -0500
@@ -34,14 +34,13 @@
 #------------------------------------------------------------------------
 # compilers & flags
 #------------------------------------------------------------------------
-CFLAGS = -ansi -funroll-loops -fPIC -DHAVE_NO_GETLINE
-CC     = gcc
+CFLAGS = -ansi -funroll-loops -fPIC -DXC_HAVE_NO_GETLINE -MD
+CC     = gcc-4
 #LDLIB  = -ldl
 MATH   = -lm
 
 FFLAGS = -O2
-FC     = g77
-
+FC     = gfortran
 
 #------------------------------------------------------------------------
 # X-libraries & include files
@@ -51,11 +50,19 @@
 
 # NOTE: Use the XDarwin server instead of Apple's Xserver
 
-DARWIN_X11_PREFIX = /usr/X11R6-darwin
+DARWIN_X11_PREFIX = /usr/X11R6
 X_LIB     = -L$(DARWIN_X11_PREFIX)/lib -lXmu -lX11 -lXext
 X_INCDIR  = -I$(DARWIN_X11_PREFIX)/include
 
+#------------------------------------------------------------------------
+# Flags for external libraries
+#------------------------------------------------------------------------
+
+FFTW3_INCDIR = -I@FINKPREFIX@/include
+MESCHACH_INCDIR = -I@FINKPREFIX@/include
 
+FFTW3_LIB = -L@FINKPREFIX@/lib -lfftw3
+MESCHACH_LIB = -L@FINKPREFIX@/lib -lmeschach
 
 ###
 ### use EITHER THIS:
@@ -75,22 +82,27 @@
 # Libraries (use .a for static and .dylib for dynamic libraries)
 #
 # Note: use the darwin OpenGL libraries
-TCL_LIB     = -ltcl8.5.dylib
-TK_LIB      = -ltk8.5.dylib
-GLU_LIB     = -L$(DARWIN_X11_PREFIX)/lib -lGLU.dylib
-GL_LIB      = -lGL.dylib
+TCL_LIB     = -L@FINKPREFIX@/lib -ltcl@TCL_VERSION@
+TK_LIB      = -L@FINKPREFIX@/lib -ltk@TCL_VERSION@
+## GL that comes with Apple's X11 leads to runtime crash
+#GLU_LIB     = -L$(DARWIN_X11_PREFIX)/lib -lGLU
+#GL_LIB      = -L$(DARWIN_X11_PREFIX)/lib -lGL
+GLU_LIB     = -L@FINKPREFIX@/lib/mesa -lGLU
+GL_LIB      = -L@FINKPREFIX@/lib/mesa -lGL
 
 #
 # Include directories (set accoring to your needs)
 #
-TCL_INCDIR  = \
-	-I/Library/Frameworks/Tcl.framework/Versions/8.5/Headers \
-	-I/Library/Frameworks/Tcl.framework/Versions/8.5/PrivateHeaders
-TK_INCDIR   = \
-	-I/Library/Frameworks/Tk.framework/Versions/8.5/Headers \
-	-I/Library/Frameworks/Tk.framework/Versions/8.5/PrivateHeaders
+TCL_INCDIR  = -I@FINKPREFIX@/include
+TK_INCDIR   = -I@FINKPREFIX@/include
+#TCL_INCDIR  = \
+	-I/System/Library/Frameworks/Tcl.framework/Versions/8.5/Headers \
+	-I/System/Library/Frameworks/Tcl.framework/Versions/8.5/PrivateHeaders
+#TK_INCDIR   = \
+	-I/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers \
+	-I/System/Library/Frameworks/Tk.framework/Versions/8.5/PrivateHeaders
 GL_INCDIR   = \
-	-I$(DARWIN_X11_PREFIX)/include
+	-I@FINKPREFIX@/include/mesa
  
 #------------------------------------------------------------------------
 
