diff -uNr squid-2.5.STABLE11/src/cf.data.pre squid-2.5.STABLE11-new/src/cf.data.pre
--- squid-2.5.STABLE11/src/cf.data.pre	2005-09-13 09:44:35.000000000 -0400
+++ squid-2.5.STABLE11-new/src/cf.data.pre	2005-10-19 09:43:02.000000000 -0400
@@ -2077,6 +2077,9 @@
 	  # to check username/password combinations (see
 	  # auth_param directive).
 	  #
+# Allow proxy connections from localhost
+http_access allow localhost
+
 	  # WARNING: proxy_auth can't be used in a transparent proxy. It
 	  # collides with any authentication done by origin servers. It may
 	  # seem like it works at first, but it doesn't.
@@ -2502,7 +2505,7 @@
 NAME: visible_hostname
 TYPE: string
 LOC: Config.visibleHostname
-DEFAULT: none
+DEFAULT: localhost
 DOC_START
 	If you want to present a special hostname in error messages, etc,
 	define this.  Otherwise, the return value of gethostname()
