--- xtraceroute-0.9.0/po/Makefile.in.in	1999-02-22 13:42:10.000000000 -0800
+++ xtraceroute-0.9.0-patched/po/Makefile.in.in	2002-10-05 21:57:52.000000000 -0700
@@ -119,8 +119,8 @@
 	for cat in $$catalogs; do \
 	  cat=`basename $$cat`; \
 	  case "$$cat" in \
-	    *.gmo) destdir=$(gnulocaledir);; \
-	    *)     destdir=$(localedir);; \
+	    *.gmo) destdir=$(DESTDIR)$(gnulocaledir);; \
+	    *)     destdir=$(DESTDIR)$(localedir);; \
 	  esac; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
 	  dir=$$destdir/$$lang/LC_MESSAGES; \
--- xtraceroute-0.9.0/dbio.c	2001-04-28 18:16:38.000000000 -0700
+++ xtraceroute-0.9.0-patched/dbio.c	2002-10-06 01:30:03.000000000 -0700
@@ -170,8 +170,9 @@
       if(!strcmp(mode, "w"))
         {
           struct stat sb;
+          mkdir(fn, 0700);    // Too paranoid?
           if(!stat(fn, &sb))
-            mkdir(fn, 700);    // Too paranoid?
+            mkdir(fn, 0700);    // Too paranoid?
 	  else if(!(sb.st_mode&S_IFDIR))
             {
              printf("Ouch! $HOME/.xt exists but is not a directory!\n");
