diff -Nurd -x'*~' libsexy-0.1.11.orig/libsexy/Makefile.in libsexy-0.1.11/libsexy/Makefile.in
--- libsexy-0.1.11.orig/libsexy/Makefile.in	2007-03-31 06:15:59.000000000 -0400
+++ libsexy-0.1.11/libsexy/Makefile.in	2011-02-21 03:35:12.000000000 -0500
@@ -232,7 +232,7 @@
 	sexy-url-label.c \
 	sexy-marshal.c
 
-libsexy_la_LIBADD = $(PACKAGE_LIBS)
+libsexy_la_LIBADD = $(PACKAGE_LIBS) -lintl
 libsexy_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(NO_UNDEFINED)
 INCLUDES = \
 	-I$(top_srcdir) \
diff -Nurd -x'*~' libsexy-0.1.11.orig/libsexy.pc.in libsexy-0.1.11/libsexy.pc.in
--- libsexy-0.1.11.orig/libsexy.pc.in	2005-11-10 05:21:18.000000000 -0500
+++ libsexy-0.1.11/libsexy.pc.in	2011-02-21 03:31:05.000000000 -0500
@@ -6,7 +6,8 @@
 Name: libsexy
 Description: Doing naughty things to good widgets
 Version: @VERSION@
-Libs: -L${libdir} -lsexy @PACKAGE_LIBS@
-Cflags: -I${includedir} @PACKAGE_CFLAGS@
+Requires: glib-2.0 >= 2.4.0, gtk+-2.0 >= 2.4.0
+Libs: -L${libdir} -lsexy
+Cflags: -I${includedir}
 
 
