diff -ur xmlbeans-2.2.0.orig/bin/dumpxsb xmlbeans-2.2.0/bin/dumpxsb
--- xmlbeans-2.2.0.orig/bin/dumpxsb	2006-05-05 10:28:16.000000000 -0700
+++ xmlbeans-2.2.0/bin/dumpxsb	2006-09-24 12:45:23.000000000 -0700
@@ -18,16 +18,4 @@
 #XSB file dumper
 #Prints the contents of an xsb file in human-readmble form
 
-echo `dirname $0`
-
-if [ -z "$XMLBEANS_LIB" ]; then . `dirname $0`/_setlib; fi
-
-cp=$XMLBEANS_LIB/xbean.jar:$XMLBEANS_LIB/jsr173_1.0_api.jar
-
-case "`uname`" in
-    CYGWIN*)
-        cp=`cygpath -w -p $cp`
-        ;;
-esac
-
-java -classpath "$cp" org.apache.xmlbeans.impl.tool.XsbDumper "$@"
+java org.apache.xmlbeans.impl.tool.XsbDumper "$@"
diff -ur xmlbeans-2.2.0.orig/bin/inst2xsd xmlbeans-2.2.0/bin/inst2xsd
--- xmlbeans-2.2.0.orig/bin/inst2xsd	2006-05-05 10:28:16.000000000 -0700
+++ xmlbeans-2.2.0/bin/inst2xsd	2006-09-24 12:45:36.000000000 -0700
@@ -18,14 +18,4 @@
 #Instance to Schema tool
 #Builds xsd files from xml instance files.
 
-if [ -z "$XMLBEANS_LIB" ]; then . `dirname $0`/_setlib; fi
-
-cp=$XMLBEANS_LIB/xbean.jar:$XMLBEANS_LIB/jsr173_1.0_api.jar
-
-case "`uname`" in
-    CYGWIN*)
-        cp=`cygpath -w -p $cp`
-        ;;
-esac
-
-java -classpath "$cp" org.apache.xmlbeans.impl.inst2xsd.Inst2Xsd "$@"
+java org.apache.xmlbeans.impl.inst2xsd.Inst2Xsd "$@"
diff -ur xmlbeans-2.2.0.orig/bin/scomp xmlbeans-2.2.0/bin/scomp
--- xmlbeans-2.2.0.orig/bin/scomp	2006-05-05 10:28:16.000000000 -0700
+++ xmlbeans-2.2.0/bin/scomp	2006-09-24 12:47:20.000000000 -0700
@@ -18,14 +18,4 @@
 #Schema compiler
 #Builds XBean types from xsd files.
 
-if [ -z "$XMLBEANS_LIB" ]; then . `dirname $0`/_setlib; fi
-
-cp=$XMLBEANS_LIB/xbean.jar:$XMLBEANS_LIB/jsr173_1.0_api.jar:$JAVA_HOME/lib/tools.jar:$XMLBEANS_LIB/resolver.jar
-
-case "`uname`" in
-    CYGWIN*)
-        cp=`cygpath -w -p $cp`
-        ;;
-esac
-
-java -Xmx256m -classpath "$cp" org.apache.xmlbeans.impl.tool.SchemaCompiler "$@"
+java -Xmx256m org.apache.xmlbeans.impl.tool.SchemaCompiler "$@"
diff -ur xmlbeans-2.2.0.orig/bin/sdownload xmlbeans-2.2.0/bin/sdownload
--- xmlbeans-2.2.0.orig/bin/sdownload	2006-05-05 10:28:16.000000000 -0700
+++ xmlbeans-2.2.0/bin/sdownload	2006-09-24 12:47:20.000000000 -0700
@@ -18,14 +18,4 @@
 #Schema downloader
 #Tool to download Schema files
 
-if [ -z "$XMLBEANS_LIB" ]; then . `dirname $0`/_setlib; fi
-
-cp=$XMLBEANS_LIB/xbean.jar:$XMLBEANS_LIB/jsr173_1.0_api.jar
-
-case "`uname`" in
-    CYGWIN*)
-        cp=`cygpath -w -p $cp`
-        ;;
-esac
-
-java -classpath "$cp" org.apache.xmlbeans.impl.tool.SchemaResourceManager "$@"
+java org.apache.xmlbeans.impl.tool.SchemaResourceManager "$@"
diff -ur xmlbeans-2.2.0.orig/bin/sfactor xmlbeans-2.2.0/bin/sfactor
--- xmlbeans-2.2.0.orig/bin/sfactor	2006-05-05 10:28:16.000000000 -0700
+++ xmlbeans-2.2.0/bin/sfactor	2006-09-24 12:47:20.000000000 -0700
@@ -20,15 +20,5 @@
 ## Factors redundant definitions out of a set of schemas and 
 ## uses imports instead.
 
-if [ -z "$XMLBEANS_LIB" ]; then . `dirname $0`/_setlib; fi
-
-cp=$XMLBEANS_LIB/xbean.jar:$XMLBEANS_LIB/jsr173_1.0_api.jar:$XMLBEANS_LIB/resolver.jar
-
-case "`uname`" in
-    CYGWIN*)
-        cp=`cygpath -w -p $cp`
-        ;;
-esac
-
-java -Xmx256m -classpath "$cp" org.apache.xmlbeans.impl.tool.FactorImports "$@"
+java -Xmx256m org.apache.xmlbeans.impl.tool.FactorImports "$@"
 
diff -ur xmlbeans-2.2.0.orig/bin/svalidate xmlbeans-2.2.0/bin/svalidate
--- xmlbeans-2.2.0.orig/bin/svalidate	2006-05-05 10:28:16.000000000 -0700
+++ xmlbeans-2.2.0/bin/svalidate	2006-09-24 12:47:20.000000000 -0700
@@ -19,14 +19,4 @@
 #
 # Validates an instance against a schema.
 
-if [ -z "$XMLBEANS_LIB" ]; then . `dirname $0`/_setlib; fi
-
-cp=$XMLBEANS_LIB/xbean.jar:$XMLBEANS_LIB/jsr173_1.0_api.jar:$XMLBEANS_LIB/jsr173_1.0_ri.jar
-
-case "`uname`" in
-    CYGWIN*)
-        cp=`cygpath -w -p $cp`
-        ;;
-esac
-
-java -classpath "$cp" org.apache.xmlbeans.impl.tool.StreamInstanceValidator "$@"
+java org.apache.xmlbeans.impl.tool.StreamInstanceValidator "$@"
diff -ur xmlbeans-2.2.0.orig/bin/validate xmlbeans-2.2.0/bin/validate
--- xmlbeans-2.2.0.orig/bin/validate	2006-05-05 10:28:16.000000000 -0700
+++ xmlbeans-2.2.0/bin/validate	2006-09-24 12:47:20.000000000 -0700
@@ -19,14 +19,4 @@
 #
 # Validates an instance against a schema.
 
-if [ -z "$XMLBEANS_LIB" ]; then . `dirname $0`/_setlib; fi
-
-cp=$XMLBEANS_LIB/xbean.jar:$XMLBEANS_LIB/jsr173_1.0_api.jar
-
-case "`uname`" in
-    CYGWIN*)
-        cp=`cygpath -w -p $cp`
-        ;;
-esac
-
-exec java -classpath "$cp" org.apache.xmlbeans.impl.tool.InstanceValidator "$@"
+java org.apache.xmlbeans.impl.tool.InstanceValidator "$@"
diff -ur xmlbeans-2.2.0.orig/bin/xpretty xmlbeans-2.2.0/bin/xpretty
--- xmlbeans-2.2.0.orig/bin/xpretty	2006-05-05 10:28:16.000000000 -0700
+++ xmlbeans-2.2.0/bin/xpretty	2006-09-24 12:47:20.000000000 -0700
@@ -17,14 +17,4 @@
 
 #Invokes pretty printer
 
-if [ -z "$XMLBEANS_LIB" ]; then . `dirname $0`/_setlib; fi
-
-cp=$XMLBEANS_LIB/xbean.jar:$XMLBEANS_LIB/jsr173_1.0_api.jar
-
-case "`uname`" in
-    CYGWIN*)
-        cp=`cygpath -w -p $cp`
-        ;;
-esac
-
-java -classpath "$cp" org.apache.xmlbeans.impl.tool.PrettyPrinter "$@"
+java org.apache.xmlbeans.impl.tool.PrettyPrinter "$@"
diff -ur xmlbeans-2.2.0.orig/bin/xsd2inst xmlbeans-2.2.0/bin/xsd2inst
--- xmlbeans-2.2.0.orig/bin/xsd2inst	2006-05-05 10:28:16.000000000 -0700
+++ xmlbeans-2.2.0/bin/xsd2inst	2006-09-24 12:47:20.000000000 -0700
@@ -17,14 +17,4 @@
 
 #Schema to instance tool
 
-if [ -z "$XMLBEANS_LIB" ]; then . `dirname $0`/_setlib; fi
-
-cp=$XMLBEANS_LIB/xbean.jar:$XMLBEANS_LIB/jsr173_1.0_api.jar
-
-case "`uname`" in
-    CYGWIN*)
-        cp=`cygpath -w -p $cp`
-        ;;
-esac
-
-java -classpath "$cp" org.apache.xmlbeans.impl.xsd2inst.SchemaInstanceGenerator "$@"
+java org.apache.xmlbeans.impl.xsd2inst.SchemaInstanceGenerator "$@"
diff -ur xmlbeans-2.2.0.orig/bin/xsdtree xmlbeans-2.2.0/bin/xsdtree
--- xmlbeans-2.2.0.orig/bin/xsdtree	2006-05-05 10:28:16.000000000 -0700
+++ xmlbeans-2.2.0/bin/xsdtree	2006-09-24 12:47:20.000000000 -0700
@@ -17,14 +17,4 @@
 
 # Invokes type hierarchy printer
 
-if [ -z "$XMLBEANS_LIB" ]; then . `dirname $0`/_setlib; fi
-
-cp=$XMLBEANS_LIB/xbean.jar:$XMLBEANS_LIB/jsr173_1.0_api.jar
-
-case "`uname`" in
-    CYGWIN*)
-        cp=`cygpath -w -p $cp`
-        ;;
-esac
-
-java -Xmx256m -classpath "$cp" org.apache.xmlbeans.impl.tool.TypeHierarchyPrinter "$@"
+java -Xmx256m org.apache.xmlbeans.impl.tool.TypeHierarchyPrinter "$@"
diff -ur xmlbeans-2.2.0.orig/bin/xstc xmlbeans-2.2.0/bin/xstc
--- xmlbeans-2.2.0.orig/bin/xstc	2006-05-05 10:28:16.000000000 -0700
+++ xmlbeans-2.2.0/bin/xstc	2006-09-24 12:54:03.000000000 -0700
@@ -17,14 +17,4 @@
 
 # Invokes XSTC
 
-if [ -z "$XMLBEANS_LIB" ]; then . `dirname $0`/_setlib; fi
-
-cp=$XMLBEANS_LIB/xbean.jar:$XMLBEANS_LIB/jsr173_1.0_api.jar:$XMLBEANS_LIB/resolver.jar
-
-case "`uname`" in
-    CYGWIN*)
-        cp=`cygpath -w -p $cp`
-        ;;
-esac
-
-java -classpath "$cp" org.apache.xmlbeans.impl.tool.XSTCTester "$@"
+java org.apache.xmlbeans.impl.tool.XSTCTester "$@"
Only in xmlbeans-2.2.0/: build
diff -ur xmlbeans-2.2.0.orig/build.xml xmlbeans-2.2.0/build.xml
--- xmlbeans-2.2.0.orig/build.xml	2006-06-09 12:01:23.000000000 -0700
+++ xmlbeans-2.2.0/build.xml	2006-09-24 13:01:49.000000000 -0700
@@ -100,19 +100,6 @@
 
     <target name="saxon8.jar" depends="check.saxon8.jar, dirs"
         unless="saxon8.jar.exists">
-	<!-- other saxon urls
-		http://voxel.dl.sourceforge.net/sourceforge/saxon/saxonb8-1.zip
-	-->
-        <get dest="external/lib/saxonb8-6-1.zip"
-	    src="http://easynews.dl.sourceforge.net/sourceforge/saxon/saxonb8-6-1.zip"
-            verbose="true" usetimestamp="true" ignoreerrors="true"/>
-        <unzip src="external/lib/saxonb8-6-1.zip" dest="external/lib/">
-            <patternset>
-                <include name="saxon8.jar"/>
-                <include name="saxon8-dom.jar"/>
-            </patternset>
-        </unzip>
-        
         <copy file="external/lib/saxon8.jar" tofile="build/lib/saxon8.jar"/>
         
         <!-- saxon8-dom.jar is used to register DOMObjectModel for use in 
@@ -200,23 +187,7 @@
     </target>
 
     <target name="version">
-        <!-- svn must be on the path -->
-        <echo message="Trying to get revision number from Subversion..."/>
-        <exec executable="svn" output="build/private/svninfo.properties" failifexecutionfails="false">
-            <arg line="info ${basedir}"/>
-        </exec>
-        <replaceregexp file="build/private/svninfo.properties"
-            match="Last Changed Rev: (.*)"
-            replace="svn.last.rev=r\1"
-            byline="true"/>
-        <property file="build/private/svninfo.properties"/>
-
-        <!-- just in case svn info or regexp failed -->
-        <condition property="svn.last.rev" value="unknown">
-            <not><isset property="svn.last.rev"/></not>
-        </condition>
-
-        <property name="version.full" value="${version.base}-${svn.last.rev}"/>
+        <property name="version.full" value="${version.base}-unknown"/>
         <property name="manifest.comment" value="Apache XmlBeans version ${version.full}"/>
 
         <echo message="${manifest.comment}"/>
@@ -294,10 +265,6 @@
         <javadoc packagenames="*" sourcepath="src/xmlpublic"
             classpath="${javadoc.path}" destdir="${docs.dir}/reference"
             windowtitle="XMLBeans ${version.base} Documentation" source="${javac.source}" stylesheetfile="docs/stylesheet.css">
-            <link href="http://xmlbeans.apache.org/${version.base}/docs/reference/index.html"/>
-            <link href="http://java.sun.com/j2se/1.4.2/docs/api/"/>
-            <link href="http://java.sun.com/j2ee/1.4/docs/api/"/>
-            <link href="http://xml.apache.org/xerces-j/apiDocs/"/>
             <classpath>
                 <pathelement location="build/ar/xbean.jar"/>
             </classpath>
