diff -Nurd -x'*~' gcalctool-5.6.28.orig/gcalctool/Makefile.in gcalctool-5.6.28/gcalctool/Makefile.in
--- gcalctool-5.6.28.orig/gcalctool/Makefile.in	2005-08-22 10:47:30.000000000 -0400
+++ gcalctool-5.6.28/gcalctool/Makefile.in	2005-08-26 01:34:51.000000000 -0400
@@ -641,7 +641,8 @@
 	lr_parser.tab.o\
 	ce_parser.o\
 	lr_parser.o
-	ar r libparser.a $^
+	$(AR) r libparser.a $^
+	$(RANLIB) libparser.a
 
 ce_parser.o: ce_parser.c ce_parser.h
 	$(COMPILE) -c $(INCLUDES) -o $@ ce_parser.c
diff -Nurd -x'*~' gcalctool-5.6.28.orig/gcalctool/syntax_translation.c gcalctool-5.6.28/gcalctool/syntax_translation.c
--- gcalctool-5.6.28.orig/gcalctool/syntax_translation.c	2005-08-03 15:11:58.000000000 -0400
+++ gcalctool-5.6.28/gcalctool/syntax_translation.c	2005-08-26 01:23:08.000000000 -0400
@@ -21,7 +21,7 @@
 
 #include <stdio.h>
 #include <libintl.h>
-#include <malloc.h>
+#include <stdlib.h>
 #include <assert.h>
 #include <string.h>
 
diff -Nurd -x'*~' gcalctool-5.6.28.orig/po/Makefile.in.in gcalctool-5.6.28/po/Makefile.in.in
--- gcalctool-5.6.28.orig/po/Makefile.in.in	2005-08-22 10:47:18.000000000 -0400
+++ gcalctool-5.6.28/po/Makefile.in.in	2005-08-26 01:23:08.000000000 -0400
@@ -28,7 +28,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 libdir = @libdir@
-localedir = $(libdir)/locale
+localedir = $(datadir)/locale
 gnulocaledir = $(datadir)/locale
 gettextsrcdir = $(datadir)/glib-2.0/gettext/po
 subdir = po
