diff -Nurd -x'*~' gnome-menus-2.14.0.orig/Makefile.am gnome-menus-2.14.0/Makefile.am
--- gnome-menus-2.14.0.orig/Makefile.am	2005-04-11 05:36:03.000000000 -0400
+++ gnome-menus-2.14.0/Makefile.am	2006-06-06 01:08:14.000000000 -0400
@@ -5,11 +5,6 @@
 endif
 
 ALL_SUBDIRS = 			\
-	po 			\
-	libmenu			\
-	layout			\
-	desktop-directories	\
-	util			\
 	$(NULL)
 
 SUBDIRS = $(ALL_SUBDIRS) $(PYTHON_SUBDIRS)
diff -Nurd -x'*~' gnome-menus-2.14.0.orig/ltmain.sh gnome-menus-2.14.0/ltmain.sh
--- gnome-menus-2.14.0.orig/ltmain.sh	2006-01-10 10:23:50.000000000 -0500
+++ gnome-menus-2.14.0/ltmain.sh	2006-06-06 00:05:42.000000000 -0400
@@ -4023,6 +4023,11 @@
 	  eval test_cmds=\"$archive_expsym_cmds\"
 	  cmds=$archive_expsym_cmds
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+          else
+            tmp_verstring="$verstring"
+          fi
 	  eval test_cmds=\"$archive_cmds\"
 	  cmds=$archive_cmds
 	  fi
diff -Nurd -x'*~' gnome-menus-2.14.0.orig/python/Makefile.am gnome-menus-2.14.0/python/Makefile.am
--- gnome-menus-2.14.0.orig/python/Makefile.am	2005-04-22 10:55:57.000000000 -0400
+++ gnome-menus-2.14.0/python/Makefile.am	2006-06-06 01:10:50.000000000 -0400
@@ -2,7 +2,6 @@
 
 INCLUDES =				\
 	-DGMENU_I_KNOW_THIS_IS_UNSTABLE	\
-	-I$(srcdir)/../libmenu		\
 	$(WARN_CFLAGS)			\
 	$(NO_STRICT_ALIASING_CFLAGS)	\
 	$(GLIB_CFLAGS)			\
@@ -15,5 +14,5 @@
 gmenu_LTLIBRARIES = gmenu.la
 
 gmenu_la_LDFLAGS = -module -avoid-version -fPIC -export-symbols-regex initgmenu
-gmenu_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/libmenu/libgnome-menu.la
+gmenu_la_LIBADD  = $(GLIB_LIBS) -lgnome-menu
 gmenu_la_SOURCES = gmenu.c
diff -Nurd -x'*~' gnome-menus-2.14.0.orig/simple-editor/Makefile.am gnome-menus-2.14.0/simple-editor/Makefile.am
--- gnome-menus-2.14.0.orig/simple-editor/Makefile.am	2005-08-02 05:47:50.000000000 -0400
+++ gnome-menus-2.14.0/simple-editor/Makefile.am	2006-06-06 02:13:13.000000000 -0400
@@ -3,7 +3,7 @@
 bin_SCRIPTS = gmenu-simple-editor
 
 gmenu-simple-editor: gmenu-simple-editor.in
-	sed -e s!\@pyexecdir\@!@pyexecdir@! < $< > $@
+	sed -e s!\@pyexecdir\@!@pyexecdir@! -e s!\@PYTHON\@!@PYTHON@! < $< > $@
 	chmod a+x $@
 
 gladedir   = $(pkgdatadir)/glade
diff -Nurd -x'*~' gnome-menus-2.14.0.orig/simple-editor/gmenu-simple-editor.in gnome-menus-2.14.0/simple-editor/gmenu-simple-editor.in
--- gnome-menus-2.14.0.orig/simple-editor/gmenu-simple-editor.in	2005-08-02 05:35:47.000000000 -0400
+++ gnome-menus-2.14.0/simple-editor/gmenu-simple-editor.in	2006-06-06 01:02:19.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!@PYTHON@
 
 #
 # Copyright (C) 2005 Red Hat, Inc.
