diff -Naur gnome-vfs-2.5.91-old/Makefile.in gnome-vfs-2.5.91/Makefile.in
--- gnome-vfs-2.5.91-old/Makefile.in	Mon Mar 15 08:16:08 2004
+++ gnome-vfs-2.5.91/Makefile.in	Sun Mar 21 11:11:22 2004
@@ -206,7 +206,6 @@
 	devel-docs				\
 	doc					\
 	programs				\
-	test					\
 	monikers				\
 	po
 
diff -Naur gnome-vfs-2.5.91-old/configure gnome-vfs-2.5.91/configure
--- gnome-vfs-2.5.91-old/configure	Mon Mar 15 08:16:12 2004
+++ gnome-vfs-2.5.91/configure	Sun Mar 21 11:11:22 2004
@@ -22370,6 +22370,8 @@
 if eval "test \"\${$as_ac_var+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
+  ac_check_lib_save_LIBS=$LIBS
+  LIBS="-lintl  $LIBS"
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -22804,117 +22806,6 @@
    rm -f po/POTFILES
    sed -e "/^#/d" -e "/^\$/d" -e "s,.*,	$posrcprefix& \\\\," -e "\$s/\(.*\) \\\\/\1/" \
 	< $srcdir/po/POTFILES.in > po/POTFILES
-
-
-
-
-for ac_func in ngettext
-do
-as_ac_var=`echo "ac_cv_func_$ac_func" | $as_tr_sh`
-echo "$as_me:$LINENO: checking for $ac_func" >&5
-echo $ECHO_N "checking for $ac_func... $ECHO_C" >&6
-if eval "test \"\${$as_ac_var+set}\" = set"; then
-  echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  cat >conftest.$ac_ext <<_ACEOF
-/* confdefs.h.  */
-_ACEOF
-cat confdefs.h >>conftest.$ac_ext
-cat >>conftest.$ac_ext <<_ACEOF
-/* end confdefs.h.  */
-/* Define $ac_func to an innocuous variant, in case <limits.h> declares $ac_func.
-   For example, HP-UX 11i <limits.h> declares gettimeofday.  */
-#define $ac_func innocuous_$ac_func
-
-/* System header to define __stub macros and hopefully few prototypes,
-    which can conflict with char $ac_func (); below.
-    Prefer <limits.h> to <assert.h> if __STDC__ is defined, since
-    <limits.h> exists even on freestanding compilers.  */
-
-#ifdef __STDC__
-# include <limits.h>
-#else
-# include <assert.h>
-#endif
-
-#undef $ac_func
-
-/* Override any gcc2 internal prototype to avoid an error.  */
-#ifdef __cplusplus
-extern "C"
-{
-#endif
-/* We use char because int might match the return type of a gcc2
-   builtin and then its argument prototype would still apply.  */
-char $ac_func ();
-/* The GNU C library defines this for functions which it implements
-    to always fail with ENOSYS.  Some functions are actually named
-    something starting with __ and the normal name is an alias.  */
-#if defined (__stub_$ac_func) || defined (__stub___$ac_func)
-choke me
-#else
-char (*f) () = $ac_func;
-#endif
-#ifdef __cplusplus
-}
-#endif
-
-int
-main ()
-{
-return f != $ac_func;
-  ;
-  return 0;
-}
-_ACEOF
-rm -f conftest.$ac_objext conftest$ac_exeext
-if { (eval echo "$as_me:$LINENO: \"$ac_link\"") >&5
-  (eval $ac_link) 2>conftest.er1
-  ac_status=$?
-  grep -v '^ *+' conftest.er1 >conftest.err
-  rm -f conftest.er1
-  cat conftest.err >&5
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); } &&
-	 { ac_try='test -z "$ac_c_werror_flag"
-			 || test ! -s conftest.err'
-  { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
-  (eval $ac_try) 2>&5
-  ac_status=$?
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; } &&
-	 { ac_try='test -s conftest$ac_exeext'
-  { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
-  (eval $ac_try) 2>&5
-  ac_status=$?
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; }; then
-  eval "$as_ac_var=yes"
-else
-  echo "$as_me: failed program was:" >&5
-sed 's/^/| /' conftest.$ac_ext >&5
-
-eval "$as_ac_var=no"
-fi
-rm -f conftest.err conftest.$ac_objext \
-      conftest$ac_exeext conftest.$ac_ext
-fi
-echo "$as_me:$LINENO: result: `eval echo '${'$as_ac_var'}'`" >&5
-echo "${ECHO_T}`eval echo '${'$as_ac_var'}'`" >&6
-if test `eval echo '${'$as_ac_var'}'` = yes; then
-  cat >>confdefs.h <<_ACEOF
-#define `echo "HAVE_$ac_func" | $as_tr_cpp` 1
-_ACEOF
- have_ngettext=yes
-fi
-done
-
-
-if test x"$have_ngettext" != "xyes"; then
-	{ { echo "$as_me:$LINENO: error: Your Gettext installation doesn't seem to support ngettext to handle translation of plural forms. Please install GNU Gettext" >&5
-echo "$as_me: error: Your Gettext installation doesn't seem to support ngettext to handle translation of plural forms. Please install GNU Gettext" >&2;}
-   { (exit 1); exit 1; }; }
-fi;
 
 echo "$as_me:$LINENO: checking for library containing gethostent" >&5
 echo $ECHO_N "checking for library containing gethostent... $ECHO_C" >&6
diff -Naur gnome-vfs-2.5.91-old/libgnomevfs/Makefile.in gnome-vfs-2.5.91/libgnomevfs/Makefile.in
--- gnome-vfs-2.5.91-old/libgnomevfs/Makefile.in	Mon Mar 15 08:16:09 2004
+++ gnome-vfs-2.5.91/libgnomevfs/Makefile.in	Sun Mar 21 11:12:53 2004
@@ -395,7 +395,6 @@
 	gnome-vfs-async-ops.c			\
 	gnome-vfs-cancellable-ops.c	 	\
 	gnome-vfs-cancellation.c		\
-	gnome-vfs-cdrom.c			\
 	gnome-vfs-client-call.c			\
 	gnome-vfs-client.c			\
 	gnome-vfs-configuration.c		\
@@ -497,7 +496,7 @@
 am_libgnomevfs_2_la_OBJECTS = $(am__objects_1) \
 	gnome-vfs-application-registry.lo gnome-vfs-async-job-map.lo \
 	gnome-vfs-async-ops.lo gnome-vfs-cancellable-ops.lo \
-	gnome-vfs-cancellation.lo gnome-vfs-cdrom.lo \
+	gnome-vfs-cancellation.lo \
 	gnome-vfs-client-call.lo gnome-vfs-client.lo \
 	gnome-vfs-configuration.lo gnome-vfs-context.lo \
 	gnome-vfs-daemon-method.lo gnome-vfs-directory.lo \
@@ -540,7 +539,6 @@
 @AMDEP_TRUE@	./$(DEPDIR)/gnome-vfs-async-ops.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gnome-vfs-cancellable-ops.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gnome-vfs-cancellation.Plo \
-@AMDEP_TRUE@	./$(DEPDIR)/gnome-vfs-cdrom.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gnome-vfs-client-call.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gnome-vfs-client.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gnome-vfs-configuration.Plo \
@@ -672,7 +670,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gnome-vfs-async-ops.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gnome-vfs-cancellable-ops.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gnome-vfs-cancellation.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gnome-vfs-cdrom.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gnome-vfs-client-call.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gnome-vfs-client.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gnome-vfs-configuration.Plo@am__quote@
diff -Naur gnome-vfs-2.5.91-old/libgnomevfs/gnome-vfs-configuration.c gnome-vfs-2.5.91/libgnomevfs/gnome-vfs-configuration.c
--- gnome-vfs-2.5.91-old/libgnomevfs/gnome-vfs-configuration.c	Fri Oct 17 11:28:59 2003
+++ gnome-vfs-2.5.91/libgnomevfs/gnome-vfs-configuration.c	Sun Mar 21 11:11:22 2004
@@ -24,6 +24,7 @@
 #include <config.h>
 #include "gnome-vfs-configuration.h"
 
+#include <sys/types.h>
 #include <dirent.h>
 #include <errno.h>
 #include <glib/ghash.h>
diff -Naur gnome-vfs-2.5.91-old/libgnomevfs/gnome-vfs-volume-monitor-daemon.c gnome-vfs-2.5.91/libgnomevfs/gnome-vfs-volume-monitor-daemon.c
--- gnome-vfs-2.5.91-old/libgnomevfs/gnome-vfs-volume-monitor-daemon.c	Fri Jan 30 05:51:15 2004
+++ gnome-vfs-2.5.91/libgnomevfs/gnome-vfs-volume-monitor-daemon.c	Sun Mar 21 11:14:47 2004
@@ -31,7 +31,6 @@
 #include "gnome-vfs-volume-monitor-daemon.h"
 #include "gnome-vfs-volume-monitor-private.h"
 #include "gnome-vfs-unix-mounts.h"
-#include "gnome-vfs-cdrom.h"
 #include "gnome-vfs-filesystem-type.h"
 
 static void gnome_vfs_volume_monitor_daemon_class_init (GnomeVFSVolumeMonitorDaemonClass *klass);
@@ -727,35 +726,6 @@
 		vol->priv->device_type = GNOME_VFS_DEVICE_TYPE_HARDDRIVE;
 	}
 
-	if (vol->priv->device_type == GNOME_VFS_DEVICE_TYPE_CDROM) {
-		fd = 0;
-		disctype = _gnome_vfs_get_cdrom_type (mount->device_path, &fd);
-
-		switch (disctype) {
-		case CDS_AUDIO:
-			display_name = g_strdup (_("Audio CD"));
-			vol->priv->device_type = GNOME_VFS_DEVICE_TYPE_AUDIO_CD;
-			break;
-			
-		case CDS_DATA_1:
-		case CDS_DATA_2:
-		case CDS_XA_2_1:
-		case CDS_XA_2_2:
-		case CDS_MIXED:
-			/* Get volume name */
-			tmp_name = _gnome_vfs_get_iso9660_volume_name (fd);
-			display_name = modify_volume_name_for_display (tmp_name);
-			g_free (tmp_name);
-			break;
-			
-		default:
-			break;
-		}
-		
-		if (fd > 0) {
-			close (fd);
-		}
-	}
 
 	if (display_name == NULL) {
 		display_name = make_volume_name_from_path_and_fs (mount->mount_path, mount->filesystem_type);
diff -Naur gnome-vfs-2.5.91-old/ltmain.sh gnome-vfs-2.5.91/ltmain.sh
--- gnome-vfs-2.5.91-old/ltmain.sh	Sun Jan 11 17:28:53 2004
+++ gnome-vfs-2.5.91/ltmain.sh	Sun Mar 21 11:11:22 2004
@@ -5145,10 +5145,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test "$need_relink" = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
diff -Naur gnome-vfs-2.5.91-old/modules/Makefile.in gnome-vfs-2.5.91/modules/Makefile.in
--- gnome-vfs-2.5.91-old/modules/Makefile.in	Mon Mar 15 08:16:09 2004
+++ gnome-vfs-2.5.91/modules/Makefile.in	Sun Mar 21 11:16:55 2004
@@ -261,7 +261,6 @@
 	libvfs-translate.la			\
 	$(CDEMENU_LTLIBS)                       \
 	$(SMB_LTLIBS)	                        \
-	libsftp.la				\
 	libssh.la				\
 	libtar.la				\
 	$(NULL)
@@ -332,12 +331,6 @@
 libsmb_la_LIBADD = $(SAMBA_LIBS) $(MODULES_LIBS) ../libgnomevfs/libgnomevfs-2.la 
 
 
-###  `sftp' method
-libsftp_la_SOURCES = sftp-method.c pty-open.c sftp.h pty-open.h
-libsftp_la_LDFLAGS = $(module_flags)
-libsftp_la_LIBADD = $(MODULES_LIBS) ../libgnomevfs/libgnomevfs-2.la
-
-
 ###  `ssh' method
 libssh_la_SOURCES = ssh-method.c
 libssh_la_LDFLAGS = $(module_flags)
@@ -446,9 +439,6 @@
 libnntp_la_DEPENDENCIES = ../libgnomevfs/libgnomevfs-2.la
 am_libnntp_la_OBJECTS = nntp-method.lo
 libnntp_la_OBJECTS = $(am_libnntp_la_OBJECTS)
-libsftp_la_DEPENDENCIES = ../libgnomevfs/libgnomevfs-2.la
-am_libsftp_la_OBJECTS = sftp-method.lo pty-open.lo
-libsftp_la_OBJECTS = $(am_libsftp_la_OBJECTS)
 libsmb_la_DEPENDENCIES = ../libgnomevfs/libgnomevfs-2.la
 am_libsmb_la_OBJECTS = smb-method.lo
 libsmb_la_OBJECTS = $(am_libsmb_la_OBJECTS)
@@ -489,7 +479,6 @@
 @AMDEP_TRUE@	./$(DEPDIR)/nntp-method.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/pipe-method.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/pty-open.Plo \
-@AMDEP_TRUE@	./$(DEPDIR)/sftp-method.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/smb-method.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/ssh-method.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/tar-method.Plo \
@@ -507,7 +496,7 @@
 	$(libextfs_la_SOURCES) $(libfile_la_SOURCES) \
 	$(libftp_la_SOURCES) $(libgzip_la_SOURCES) \
 	$(libhttp_la_SOURCES) $(libnetwork_la_SOURCES) \
-	$(libnntp_la_SOURCES) $(libsftp_la_SOURCES) \
+	$(libnntp_la_SOURCES) \
 	$(libsmb_la_SOURCES) $(libssh_la_SOURCES) $(libtar_la_SOURCES) \
 	$(libvfs_pipe_la_SOURCES) $(libvfs_test_la_SOURCES) \
 	$(libvfs_translate_la_SOURCES)
@@ -520,7 +509,7 @@
 	uninstall-recursive check-recursive installcheck-recursive
 DIST_COMMON = Makefile.am Makefile.in
 DIST_SUBDIRS = $(SUBDIRS)
-SOURCES = $(libbzip2_la_SOURCES) $(libcdda_la_SOURCES) $(libcdemenu_desktop_la_SOURCES) $(libcomputer_la_SOURCES) $(libextfs_la_SOURCES) $(libfile_la_SOURCES) $(libftp_la_SOURCES) $(libgzip_la_SOURCES) $(libhttp_la_SOURCES) $(libnetwork_la_SOURCES) $(libnntp_la_SOURCES) $(libsftp_la_SOURCES) $(libsmb_la_SOURCES) $(libssh_la_SOURCES) $(libtar_la_SOURCES) $(libvfs_pipe_la_SOURCES) $(libvfs_test_la_SOURCES) $(libvfs_translate_la_SOURCES)
+SOURCES = $(libbzip2_la_SOURCES) $(libcdda_la_SOURCES) $(libcdemenu_desktop_la_SOURCES) $(libcomputer_la_SOURCES) $(libextfs_la_SOURCES) $(libfile_la_SOURCES) $(libftp_la_SOURCES) $(libgzip_la_SOURCES) $(libhttp_la_SOURCES) $(libnetwork_la_SOURCES) $(libnntp_la_SOURCES) $(libsmb_la_SOURCES) $(libssh_la_SOURCES) $(libtar_la_SOURCES) $(libvfs_pipe_la_SOURCES) $(libvfs_test_la_SOURCES) $(libvfs_translate_la_SOURCES)
 
 all: all-recursive
 
@@ -581,8 +570,6 @@
 	$(LINK) -rpath $(modulesdir) $(libnetwork_la_LDFLAGS) $(libnetwork_la_OBJECTS) $(libnetwork_la_LIBADD) $(LIBS)
 libnntp.la: $(libnntp_la_OBJECTS) $(libnntp_la_DEPENDENCIES) 
 	$(LINK) -rpath $(modulesdir) $(libnntp_la_LDFLAGS) $(libnntp_la_OBJECTS) $(libnntp_la_LIBADD) $(LIBS)
-libsftp.la: $(libsftp_la_OBJECTS) $(libsftp_la_DEPENDENCIES) 
-	$(LINK) -rpath $(modulesdir) $(libsftp_la_LDFLAGS) $(libsftp_la_OBJECTS) $(libsftp_la_LIBADD) $(LIBS)
 libsmb.la: $(libsmb_la_OBJECTS) $(libsmb_la_DEPENDENCIES) 
 	$(LINK) -rpath $(modulesdir) $(libsmb_la_LDFLAGS) $(libsmb_la_OBJECTS) $(libsmb_la_LIBADD) $(LIBS)
 libssh.la: $(libssh_la_OBJECTS) $(libssh_la_DEPENDENCIES) 
@@ -619,7 +606,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/nntp-method.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/pipe-method.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/pty-open.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/sftp-method.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/smb-method.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/ssh-method.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/tar-method.Plo@am__quote@
diff -Naur gnome-vfs-2.5.91-old/modules/file-method.c gnome-vfs-2.5.91/modules/file-method.c
--- gnome-vfs-2.5.91-old/modules/file-method.c	Wed Mar  3 05:20:28 2004
+++ gnome-vfs-2.5.91/modules/file-method.c	Sun Mar 21 11:11:22 2004
@@ -38,6 +38,7 @@
 #include <libgnomevfs/gnome-vfs-monitor-private.h>
 #include <libgnomevfs/gnome-vfs-private-utils.h>
 #include <libgnomevfs/gnome-vfs-ops.h>
+#include <sys/types.h>
 #include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
@@ -48,7 +49,6 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
-#include <sys/types.h>
 #include <unistd.h>
 #include <utime.h>
 #include <string.h>
diff -Naur gnome-vfs-2.5.91-old/modules/fstype.c gnome-vfs-2.5.91/modules/fstype.c
--- gnome-vfs-2.5.91-old/modules/fstype.c	Thu Jan 22 10:28:47 2004
+++ gnome-vfs-2.5.91/modules/fstype.c	Sun Mar 21 11:11:22 2004
@@ -363,7 +363,7 @@
   char *p;
 
   if (S_ISLNK (statp->st_mode))
-    p = dirname (relpath);
+    p = g_path_get_dirname (relpath);
   else
     p = relpath;
 
