diff -uNr gnome-vfs-2.15.2/configure gnome-vfs-2.15.2-new/configure
--- gnome-vfs-2.15.2/configure	2006-06-12 07:31:06.000000000 -0400
+++ gnome-vfs-2.15.2-new/configure	2006-07-21 01:49:47.000000000 -0400
@@ -38134,7 +38134,7 @@
 
 
 
-if test $have_posix_acl_support = yes; then
+if test x"$have_posix_acl_support" = x"yes"; then
   HAVE_POSIX_ACL_TRUE=
   HAVE_POSIX_ACL_FALSE='#'
 else
@@ -38144,7 +38144,7 @@
 
 
 
-if test $have_solaris_acl_support = yes; then
+if test x"$have_solaris_acl_support" = x"yes"; then
   HAVE_SOLARIS_ACL_TRUE=
   HAVE_SOLARIS_ACL_FALSE='#'
 else
diff -uNr gnome-vfs-2.15.2/daemon/Makefile.in gnome-vfs-2.15.2-new/daemon/Makefile.in
--- gnome-vfs-2.15.2/daemon/Makefile.in	2006-06-12 07:31:08.000000000 -0400
+++ gnome-vfs-2.15.2-new/daemon/Makefile.in	2006-07-21 01:50:24.000000000 -0400
@@ -61,7 +61,6 @@
 am__DEPENDENCIES_1 =
 gnome_vfs_daemon_DEPENDENCIES = $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) \
-	$(top_builddir)/libgnomevfs/libgnomevfs-2.la \
 	$(top_builddir)/libgnomevfs/libgnomevfsdaemon-2.la
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
@@ -374,7 +373,7 @@
 	-I$(top_builddir)/libgnomevfs           \
 	$(NULL)
 
-gnome_vfs_daemon_LDADD = $(MONIKERS_LIBS) $(LIBGNOMEVFSDAEMON_LIBS) $(top_builddir)/libgnomevfs/libgnomevfs-2.la $(top_builddir)/libgnomevfs/libgnomevfsdaemon-2.la
+gnome_vfs_daemon_LDADD = $(MONIKERS_LIBS) $(LIBGNOMEVFSDAEMON_LIBS) $(top_builddir)/libgnomevfs/libgnomevfsdaemon-2.la
 CORBA_SOURCE = \
 	$(top_builddir)/libgnomevfs/GNOME_VFS_Daemon.h		\
 	$(top_builddir)/libgnomevfs/GNOME_VFS_Daemon-common.c	\
diff -uNr gnome-vfs-2.15.2/modules/file-method-acl.c gnome-vfs-2.15.2-new/modules/file-method-acl.c
--- gnome-vfs-2.15.2/modules/file-method-acl.c	2006-06-08 09:08:11.000000000 -0400
+++ gnome-vfs-2.15.2-new/modules/file-method-acl.c	2006-07-21 02:31:06.000000000 -0400
@@ -34,6 +34,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <sys/types.h>
+#include <sys/time.h>
 #include <sys/resource.h>
 
 #ifdef HAVE_PWD_H
@@ -46,10 +47,12 @@
 
 #ifdef HAVE_POSIX_ACL
 # include <acl/libacl.h>
+#define HAVE_ACL
 #endif
 
 #ifdef HAVE_SOLARIS_ACL
 # include <sys/acl.h>
+#define HAVE_ACL
 #endif
 
 #include "file-method-acl.h"
@@ -59,6 +62,7 @@
 #define CMD_PERM_EXECUTE        1
 #define CMD_PERM_COND_EXECUTE   8
 
+#ifdef HAVE_ACL
 
 static char *
 uid_to_string (uid_t uid) 
@@ -717,6 +721,7 @@
 	return GNOME_VFS_ERROR_GENERIC;
 }
 
+#endif /* HAVE_ACL */
 
 GnomeVFSResult file_get_acl (const char       *path,
                              GnomeVFSFileInfo *info,
