diff -uNr gnomoradio-0.14/docs/C/Makefile.in gnomoradio-0.14-new/docs/C/Makefile.in
--- gnomoradio-0.14/docs/C/Makefile.in	Mon Jun 14 14:55:17 2004
+++ gnomoradio-0.14-new/docs/C/Makefile.in	Fri Sep 10 18:24:12 2004
@@ -422,7 +422,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
@@ -430,7 +429,6 @@
 		rm -f $(omf_dest_dir)/$$basefile; \
 	done
 	-rmdir $(omf_dest_dir)
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
 
 clean-local-omf:
 	-for file in $(omffile); do \
