diff -Naur gnome-spell-1.0.5-old/po/Makefile.in.in gnome-spell-1.0.5/po/Makefile.in.in
--- gnome-spell-1.0.5-old/po/Makefile.in.in	Wed Jul 30 11:43:01 2003
+++ gnome-spell-1.0.5/po/Makefile.in.in	Tue Dec 30 12:28:40 2003
@@ -28,7 +28,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 libdir = @libdir@
-localedir = $(libdir)/locale
+localedir = $(datadir)/locale
 gnulocaledir = $(datadir)/locale
 gettextsrcdir = $(datadir)/glib-2.0/gettext/po
 subdir = po
