diff -ruN gproftpd-8.1.0.orig/configure gproftpd-8.1.0/configure
--- gproftpd-8.1.0.orig/configure	Thu Oct 30 12:49:06 2003
+++ gproftpd-8.1.0/configure	Thu Nov  6 10:54:33 2003
@@ -6610,7 +6610,7 @@
 
 
 cat >>confdefs.h <<_ACEOF
-#define GP_PASSWD "`eval echo "${sysconfdir}/passwd"`"
+#define GP_PASSWD "`eval echo "/etc/passwd"`"
 _ACEOF
 
 
@@ -6620,7 +6620,7 @@
 
 
 cat >>confdefs.h <<_ACEOF
-#define GP_GROUP "`eval echo "${sysconfdir}/group"`"
+#define GP_GROUP "`eval echo "/etc/group"`"
 _ACEOF
 
 
diff -ruN gproftpd-8.1.0.orig/src/gprostats.c gproftpd-8.1.0/src/gprostats.c
--- gproftpd-8.1.0.orig/src/gprostats.c	Wed Oct 15 16:37:55 2003
+++ gproftpd-8.1.0/src/gprostats.c	Thu Nov  6 10:42:53 2003
@@ -36,6 +36,10 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef __APPLE__
+#include <stdint.h>
+#define __uint64_t uint64_t
+#endif
 #include <unistd.h>
 #include <string.h>
 
