diff -uNr gst-plugins-good-0.10.1/docs/plugins/Makefile.in gst-plugins-good-0.10.1-new/docs/plugins/Makefile.in
--- gst-plugins-good-0.10.1/docs/plugins/Makefile.in	2006-01-13 13:56:58.000000000 -0500
+++ gst-plugins-good-0.10.1-new/docs/plugins/Makefile.in	2006-01-19 14:00:59.000000000 -0500
@@ -904,7 +904,7 @@
 
 @ENABLE_GTK_DOC_TRUE@html-build.stamp: sgml.stamp $(DOC_MAIN_SGML_FILE) $(content_files)
 @ENABLE_GTK_DOC_TRUE@	@echo '*** Building HTML ***'
-@ENABLE_GTK_DOC_TRUE@	if test -d html; then rm -rf html; fi
+@ENABLE_GTK_DOC_TRUE@	if test -d html; then rm -rf html || rm -rf html; fi
 @ENABLE_GTK_DOC_TRUE@	mkdir html
 @ENABLE_GTK_DOC_TRUE@	cp $(srcdir)/$(DOC_MAIN_SGML_FILE) html
 @ENABLE_GTK_DOC_TRUE@	@for f in $(content_files); do cp $(srcdir)/$$f html; done
@@ -912,7 +912,7 @@
 @ENABLE_GTK_DOC_TRUE@	cp ../version.entities html
 @ENABLE_GTK_DOC_TRUE@	cd html && gtkdoc-mkhtml $(DOC_MODULE) $(DOC_MAIN_SGML_FILE)
 @ENABLE_GTK_DOC_TRUE@	rm -f html/$(DOC_MAIN_SGML_FILE)
-@ENABLE_GTK_DOC_TRUE@	rm -rf html/xml
+@ENABLE_GTK_DOC_TRUE@	rm -rf html/xml || rm -rf html/xml
 @ENABLE_GTK_DOC_TRUE@	rm -f html/version.entities
 @ENABLE_GTK_DOC_TRUE@	test "x$(HTML_IMAGES)" = "x" || for i in "" $(HTML_IMAGES) ; do \
 @ENABLE_GTK_DOC_TRUE@	    if test "$$i" != ""; then cp $(srcdir)/$$i html ; fi; done
