diff -Nurd -x'*~' gtklp-1.2.5.orig/gtklp/Makefile.in gtklp-1.2.5/gtklp/Makefile.in
--- gtklp-1.2.5.orig/gtklp/Makefile.in	2007-09-15 12:05:23.000000000 -0400
+++ gtklp-1.2.5/gtklp/Makefile.in	2007-11-04 15:40:41.000000000 -0500
@@ -248,7 +248,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-AM_CPPFLAGS = $(FORTE_CCFLAGS) -I$(top_srcdir)/include -I$(top_srcdir)/libgtklp $(GTK2_CFLAGS) $(GTK_CFLAGS) $(CUPS_CFLAGS) -DLOCALEDIR=\"$(datadir)/locale\" $(GTKVER) -DMAX_PRT=$(MAXPRT) $(OLDDIALOG)
+AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/libgtklp $(FORTE_CCFLAGS) $(GTK2_CFLAGS) $(GTK_CFLAGS) $(CUPS_CFLAGS) -DLOCALEDIR=\"$(datadir)/locale\" $(GTKVER) -DMAX_PRT=$(MAXPRT) $(OLDDIALOG)
 gtklp_SOURCES = file.c file.h general.c general.h gtklp.c gtklp.h \
 	gtklp_functions.c gtklp_functions.h gtklptab.c gtklptab.h hpgl2.c \
 	hpgl2.h image.c image.h output.c output.h ppd.c ppd.h special.c \
diff -Nurd -x'*~' gtklp-1.2.5.orig/gtklpq/Makefile.in gtklp-1.2.5/gtklpq/Makefile.in
--- gtklp-1.2.5.orig/gtklpq/Makefile.in	2007-09-15 12:05:23.000000000 -0400
+++ gtklp-1.2.5/gtklpq/Makefile.in	2007-11-04 15:40:04.000000000 -0500
@@ -246,7 +246,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-AM_CPPFLAGS = $(FORTE_CCFLAGS) $(GTK2_CFLAGS) $(GTK_CFLAGS) $(CUPS_CFLAGS) -I$(top_srcdir)/include -I$(top_srcdir)/libgtklp -DLOCALEDIR=\"$(datadir)/locale\" $(GTKVER) -DMAX_PRT=$(MAXPRT)
+AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/libgtklp $(FORTE_CCFLAGS) $(GTK2_CFLAGS) $(GTK_CFLAGS) $(CUPS_CFLAGS) -DLOCALEDIR=\"$(datadir)/locale\" $(GTKVER) -DMAX_PRT=$(MAXPRT)
 gtklpq_SOURCES = gtklpq.c gtklpq.h gtklpq_functions.c gtklpq_functions.h \
 	jobtags.h printer.c printer.h \
 	gtklpq_exit.xpm gtklpq_hold.xpm gtklpq_move.xpm gtklpq_prio.xpm \
