diff -Naur libgnomeprint-2.4.0.old/configure libgnomeprint-2.4.0.new/configure
--- libgnomeprint-2.4.0.old/configure	Fri Sep 26 23:14:50 2003
+++ libgnomeprint-2.4.0.new/configure	Thu Jan 15 00:27:29 2004
@@ -21764,7 +21764,7 @@
 GNOME_PRINT_DATA_DIR=$GNOME_PRINT_DATA_DIR
 
 
-LIBGNOMEPRINT_CFLAGS="$GP_CFLAGS $FREETYPE_CFLAGS"
+LIBGNOMEPRINT_CFLAGS="$FREETYPE_CFLAGS $GP_CFLAGS"
 LIBGNOMEPRINT_LIBS="$GP_LIBS $FREETYPE_LIBS"
 
 
diff -Naur libgnomeprint-2.4.0.old/libgnomeprint/gnome-font.c libgnomeprint-2.4.0.new/libgnomeprint/gnome-font.c
--- libgnomeprint-2.4.0.old/libgnomeprint/gnome-font.c	Wed Jan  8 15:36:16 2003
+++ libgnomeprint-2.4.0.new/libgnomeprint/gnome-font.c	Thu Jan 15 00:32:03 2004
@@ -293,6 +293,8 @@
 	if (!face) {
 		/* fixme: */
 		face = gnome_font_face_find_closest ("Helvetica");
+		if (!face)
+			face = gnome_font_face_find_closest ("Sans Serif");
 		g_return_val_if_fail (face != NULL, NULL);
 	}
 
diff -Naur libgnomeprint-2.4.0.old/libgnomeprint/gnome-fontmap.c libgnomeprint-2.4.0.new/libgnomeprint/gnome-fontmap.c
--- libgnomeprint-2.4.0.old/libgnomeprint/gnome-fontmap.c	Sun Feb 16 02:04:36 2003
+++ libgnomeprint-2.4.0.new/libgnomeprint/gnome-fontmap.c	Thu Jan 15 00:32:26 2004
@@ -194,6 +194,8 @@
 		font_type = GP_FONT_ENTRY_TYPE1;
 	else if (strcasecmp (c, ".ttf") == 0)
 		font_type = GP_FONT_ENTRY_TRUETYPE;
+	else if (strcasecmp (c, ".otf") == 0)
+		font_type = GP_FONT_ENTRY_TRUETYPE;
 	else if (strcasecmp (c, ".font") == 0) /* See #102400 */
 		font_type = GP_FONT_ENTRY_TRUETYPE;
 	else {
diff -Naur libgnomeprint-2.4.0.old/libgnomeprint/gnome-print.c libgnomeprint-2.4.0.new/libgnomeprint/gnome-print.c
--- libgnomeprint-2.4.0.old/libgnomeprint/gnome-print.c	Tue Mar 18 01:58:08 2003
+++ libgnomeprint-2.4.0.new/libgnomeprint/gnome-print.c	Thu Jan 15 00:27:29 2004
@@ -28,6 +28,7 @@
 #include <config.h>
 #include <string.h>
 #include <gmodule.h>
+#include <sys/types.h>
 
 #include <libgnomeprint/gnome-print-i18n.h>
 #include <libgnomeprint/gnome-print-private.h>
diff -Naur libgnomeprint-2.4.0.old/ltmain.sh libgnomeprint-2.4.0.new/ltmain.sh
--- libgnomeprint-2.4.0.old/ltmain.sh	Thu Sep 11 07:55:56 2003
+++ libgnomeprint-2.4.0.new/ltmain.sh	Thu Jan 15 00:27:29 2004
@@ -3794,7 +3794,17 @@
 	  if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	    eval cmds=\"$archive_expsym_cmds\"
 	  else
-	    eval cmds=\"$archive_cmds\"
+	    if test "x$verstring" = "x0.0"; then
+              tmp_verstring=
+            else
+              tmp_verstring="$verstring"
+            fi
+            save_deplibs="$deplibs"
+            for conv in $convenience; do
+              deplibs="${deplibs%$conv*} ${deplibs#*$conv}"
+            done
+	    eval cmds=\"$archive_cmds\"a
+	    deplibs="$save_deplibs"
 	  fi
 
 	  # Append the command to remove the reloadable object files
