diff -Naur libgtkhtml-2.4.0-old/ltmain.sh libgtkhtml-2.4.0/ltmain.sh
--- libgtkhtml-2.4.0-old/ltmain.sh	Fri May 23 04:59:21 2003
+++ libgtkhtml-2.4.0/ltmain.sh	Fri Oct 17 14:22:35 2003
@@ -3783,7 +3783,23 @@
 	  if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	    eval cmds=\"$archive_expsym_cmds\"
 	  else
+           if test "x$verstring" = "x0.0"; then
+            tmp_verstring=
+           else
+             tmp_verstring="$verstring"
+           fi
+           save_deplibs="$deplibs"
+           for conv in $convenience; do
+             tmp_deplibs=
+             for test_deplib in $deplibs; do
+               if test "$test_deplib" != "$conv"; then
+                 tmp_deplibs="$tmp_deplibs $test_deplib"
+               fi
+             done
+             deplibs="$tmp_deplibs"
+            done
 	    eval cmds=\"$archive_cmds\"
+	    deplibs="$save_deplibs"
 	  fi
 
 	  # Append the command to remove the reloadable object files
