diff -Naur nautilus-cd-burner-2.6.0-old/make-iso.c nautilus-cd-burner-2.6.0/make-iso.c
--- nautilus-cd-burner-2.6.0-old/make-iso.c	Tue Mar  2 03:52:11 2004
+++ nautilus-cd-burner-2.6.0/make-iso.c	Thu Apr  8 22:43:36 2004
@@ -24,7 +24,7 @@
 
 #include "config.h"
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__Darwin__)
 #include <sys/param.h>
 #include <sys/mount.h>
 #else
diff -Naur nautilus-cd-burner-2.6.0-old/po/Makefile.in.in nautilus-cd-burner-2.6.0/po/Makefile.in.in
--- nautilus-cd-burner-2.6.0-old/po/Makefile.in.in	Mon Mar 22 09:39:17 2004
+++ nautilus-cd-burner-2.6.0/po/Makefile.in.in	Thu Apr  8 22:39:16 2004
@@ -28,7 +28,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 libdir = @libdir@
-localedir = $(libdir)/locale
+localedir = $(datadir)/locale
 gnulocaledir = $(datadir)/locale
 gettextsrcdir = $(datadir)/glib-2.0/gettext/po
 subdir = po
