diff -Nurd ORBit2-2.14.4.orig/linc2/src/Makefile.in ORBit2-2.14.4/linc2/src/Makefile.in
--- ORBit2-2.14.4.orig/linc2/src/Makefile.in	2006-12-18 03:14:41.000000000 -0500
+++ ORBit2-2.14.4/linc2/src/Makefile.in	2007-01-22 04:33:46.000000000 -0500
@@ -250,7 +250,6 @@
 
 liblinc_la_LDFLAGS = \
 	$(LINC_LIBS)                 \
-	-version-info $(LT_VERSION)  \
 	-no-undefined
 
 linc_cleanup_sockets_SOURCES = \
diff -Nurd ORBit2-2.14.4.orig/ltmain.sh ORBit2-2.14.4/ltmain.sh
--- ORBit2-2.14.4.orig/ltmain.sh	2006-07-13 03:12:13.000000000 -0400
+++ ORBit2-2.14.4/ltmain.sh	2007-01-22 04:14:48.000000000 -0500
@@ -5739,10 +5739,6 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
 	done
       fi
 
