diff -Naur pango-1.2.5-old/Makefile.in pango-1.2.5/Makefile.in
--- pango-1.2.5-old/Makefile.in	Wed Aug 27 11:38:53 2003
+++ pango-1.2.5/Makefile.in	Mon Oct 13 18:34:56 2003
@@ -120,7 +120,7 @@
 XFT_LIBS = @XFT_LIBS@
 ms_librarian = @ms_librarian@
 
-SUBDIRS = pango modules examples docs tools tests
+SUBDIRS = pango modules docs tools
 
 EXTRA_DIST = \
 	pango.pc.in	\
diff -Naur pango-1.2.5-old/ltmain.sh pango-1.2.5/ltmain.sh
--- pango-1.2.5-old/ltmain.sh	Thu Jul 17 00:47:23 2003
+++ pango-1.2.5/ltmain.sh	Mon Oct 13 18:49:51 2003
@@ -5145,10 +5145,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test "$need_relink" = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
diff -Naur pango-1.2.5-old/modules/arabic/Makefile.in pango-1.2.5/modules/arabic/Makefile.in
--- pango-1.2.5-old/modules/arabic/Makefile.in	Wed Aug 27 11:39:03 2003
+++ pango-1.2.5/modules/arabic/Makefile.in	Mon Oct 13 18:34:56 2003
@@ -122,8 +122,8 @@
 
 pangolibs = $(top_builddir)/pango/libpango-$(PANGO_API_VERSION).la $(GLIB_LIBS)
 pangoxlibs = $(top_builddir)/pango/libpangox-$(PANGO_API_VERSION).la $(X_LIBS) $(pangolibs)
-pangoxftlibs = $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(XFT_LIBS) $(pangolibs)
-pangoft2libs = $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(FREETYPE_LIBS) $(pangolibs)
+pangoxftlibs = $(XFTLIBS) $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(pangolibs)
+pangoft2libs = $(FREETYPE_LIBS) $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(pangolibs)
 @HAVE_XFT_TRUE@@INCLUDE_ARABIC_XFT_TRUE@XFT_MODULES = 
 @HAVE_XFT_TRUE@@INCLUDE_ARABIC_XFT_FALSE@XFT_MODULES = @HAVE_XFT_TRUE@@INCLUDE_ARABIC_XFT_FALSE@pango-arabic-xft.la
 @HAVE_XFT_FALSE@XFT_MODULES = 
diff -Naur pango-1.2.5-old/modules/basic/Makefile.in pango-1.2.5/modules/basic/Makefile.in
--- pango-1.2.5-old/modules/basic/Makefile.in	Wed Aug 27 11:39:05 2003
+++ pango-1.2.5/modules/basic/Makefile.in	Mon Oct 13 18:34:56 2003
@@ -124,8 +124,8 @@
 
 pangolibs = $(top_builddir)/pango/libpango-$(PANGO_API_VERSION).la $(GLIB_LIBS)
 pangoxlibs = $(top_builddir)/pango/libpangox-$(PANGO_API_VERSION).la $(X_LIBS) $(pangolibs)
-pangoxftlibs = $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(XFT_LIBS) $(pangolibs)
-pangoft2libs = $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(FREETYPE_LIBS) $(pangolibs)
+pangoxftlibs = $(XFT_LIBS) $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(pangolibs)
+pangoft2libs = $(FREETYPE_LIBS) $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(pangolibs)
 pangowin32libs = $(top_builddir)/pango/libpangowin32-$(PANGO_API_VERSION).la $(pangolibs)
 @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_TRUE@XFT_INCLUDED = @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_TRUE@libpango-basic-xft.la
 @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_FALSE@XFT_INCLUDED = 
diff -Naur pango-1.2.5-old/modules/hangul/Makefile.in pango-1.2.5/modules/hangul/Makefile.in
--- pango-1.2.5-old/modules/hangul/Makefile.in	Wed Aug 27 11:39:06 2003
+++ pango-1.2.5/modules/hangul/Makefile.in	Mon Oct 13 18:34:56 2003
@@ -122,8 +122,8 @@
 
 pangolibs = $(top_builddir)/pango/libpango-$(PANGO_API_VERSION).la $(GLIB_LIBS)
 pangoxlibs = $(top_builddir)/pango/libpangox-$(PANGO_API_VERSION).la $(X_LIBS) $(pangolibs)
-pangoxftlibs = $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(XFT_LIBS) $(pangolibs)
-pangoft2libs = $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(FREETYPE_LIBS) $(pangolibs)
+pangoxftlibs = $(XFT_LIBS) $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(pangolibs)
+pangoft2libs = $(FREETYPE_LIBS) $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(pangolibs)
 
 x_sources = hangul-x.c hangul-defs.h
 xft_sources = hangul-xft.c hangul-defs.h
diff -Naur pango-1.2.5-old/modules/hebrew/Makefile.in pango-1.2.5/modules/hebrew/Makefile.in
--- pango-1.2.5-old/modules/hebrew/Makefile.in	Wed Aug 27 11:39:13 2003
+++ pango-1.2.5/modules/hebrew/Makefile.in	Mon Oct 13 18:34:56 2003
@@ -127,8 +127,8 @@
 
 pangolibs = $(top_builddir)/pango/libpango-$(PANGO_API_VERSION).la $(GLIB_LIBS)
 pangoxlibs = $(top_builddir)/pango/libpangox-$(PANGO_API_VERSION).la $(X_LIBS) $(pangolibs)
-pangoxftlibs = $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(XFT_LIBS) $(pangolibs)
-pangoft2libs = $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(FREETYPE_LIBS) $(pangolibs)
+pangoxftlibs = $(XFT_LIBS) $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(pangolibs)
+pangoft2libs = $(FREETYPE_LIBS) $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(pangolibs)
 @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_TRUE@XFT_INCLUDED = @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_TRUE@libpango-hebrew-xft.la
 @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_FALSE@XFT_INCLUDED = 
 @HAVE_XFT_FALSE@XFT_INCLUDED = 
@@ -138,7 +138,7 @@
 @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_TRUE@XFT_PREFIX = @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_TRUE@-DXFT_MODULE_PREFIX
 @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_FALSE@XFT_PREFIX = 
 @HAVE_XFT_FALSE@XFT_PREFIX = 
-@HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_FALSE@hebrew_xft_libadd = @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_FALSE@$(pangoxlibs) $(pangoxftlibs)
+@HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_FALSE@hebrew_xft_libadd = @HAVE_XFT_TRUE@@INCLUDE_BASIC_XFT_FALSE@$(pangoxftlibs) $(pangoxlibs)
 @HAVE_X_TRUE@@INCLUDE_BASIC_X_TRUE@X_INCLUDED = @HAVE_X_TRUE@@INCLUDE_BASIC_X_TRUE@libpango-hebrew-x.la
 @HAVE_X_TRUE@@INCLUDE_BASIC_X_FALSE@X_INCLUDED = 
 @HAVE_X_FALSE@X_INCLUDED = 
diff -Naur pango-1.2.5-old/modules/indic/Makefile.in pango-1.2.5/modules/indic/Makefile.in
--- pango-1.2.5-old/modules/indic/Makefile.in	Wed Aug 27 11:39:08 2003
+++ pango-1.2.5/modules/indic/Makefile.in	Mon Oct 13 18:34:56 2003
@@ -140,8 +140,8 @@
 
 pangolibs = $(top_builddir)/pango/libpango-$(PANGO_API_VERSION).la $(GLIB_LIBS)
 pangoxlibs = $(top_builddir)/pango/libpangox-$(PANGO_API_VERSION).la $(X_LIBS) $(pangolibs)
-pangoxftlibs = $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(XFT_LIBS) $(pangolibs)
-pangoft2libs = $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(FREETYPE_LIBS) $(pangolibs)
+pangoxftlibs = $(XFT_LIBS) $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(pangolibs)
+pangoft2libs = $(FREETYPE_LIBS) $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(pangolibs)
 @HAVE_XFT_TRUE@@INCLUDE_INDIC_XFT_TRUE@indic_xft_inst = 
 @HAVE_XFT_TRUE@@INCLUDE_INDIC_XFT_FALSE@indic_xft_inst = @HAVE_XFT_TRUE@@INCLUDE_INDIC_XFT_FALSE@pango-indic-xft.la
 @HAVE_XFT_TRUE@@INCLUDE_INDIC_XFT_TRUE@indic_xft_noinst = @HAVE_XFT_TRUE@@INCLUDE_INDIC_XFT_TRUE@libpango-indic-xft.la
diff -Naur pango-1.2.5-old/modules/tamil/Makefile.in pango-1.2.5/modules/tamil/Makefile.in
--- pango-1.2.5-old/modules/tamil/Makefile.in	Wed Aug 27 11:39:10 2003
+++ pango-1.2.5/modules/tamil/Makefile.in	Mon Oct 13 18:34:56 2003
@@ -122,7 +122,7 @@
 
 pangolibs = $(top_builddir)/pango/libpango-$(PANGO_API_VERSION).la $(GLIB_LIBS)
 pangoxlibs = $(top_builddir)/pango/libpangox-$(PANGO_API_VERSION).la $(X_LIBS) $(pangolibs)
-pangoft2libs = $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(FREETYPE_LIBS) $(pangolibs)
+pangoft2libs = $(FREETYPE_LIBS) $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(pangolibs)
 
 x_sources = \
 	tamil-x.c \
diff -Naur pango-1.2.5-old/modules/thai/Makefile.in pango-1.2.5/modules/thai/Makefile.in
--- pango-1.2.5-old/modules/thai/Makefile.in	Wed Aug 27 11:39:11 2003
+++ pango-1.2.5/modules/thai/Makefile.in	Mon Oct 13 18:34:56 2003
@@ -127,8 +127,8 @@
 
 pangolibs = $(top_builddir)/pango/libpango-$(PANGO_API_VERSION).la $(GLIB_LIBS)
 pangoxlibs = $(top_builddir)/pango/libpangox-$(PANGO_API_VERSION).la $(X_LIBS) $(pangolibs)
-pangoxftlibs = $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(XFT_LIBS) $(pangolibs)
-pangoft2libs = $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(FREETYPE_LIBS) $(pangolibs)
+pangoxftlibs = $(XFT_LIBS) $(top_builddir)/pango/libpangoxft-$(PANGO_API_VERSION).la $(pangoxlibs)
+pangoft2libs = $(FREETYPE_LIBS) $(top_builddir)/pango/libpangoft2-$(PANGO_API_VERSION).la $(pangolibs)
 @HAVE_XFT_TRUE@@INCLUDE_THAI_XFT_TRUE@XFT_MODULES = 
 @HAVE_XFT_TRUE@@INCLUDE_THAI_XFT_FALSE@XFT_MODULES = @HAVE_XFT_TRUE@@INCLUDE_THAI_XFT_FALSE@pango-thai-xft.la
 @HAVE_XFT_FALSE@XFT_MODULES = 
diff -Naur pango-1.2.5-old/pango/Makefile.in pango-1.2.5/pango/Makefile.in
--- pango-1.2.5-old/pango/Makefile.in	Wed Aug 27 11:38:57 2003
+++ pango-1.2.5/pango/Makefile.in	Mon Oct 13 18:34:56 2003
@@ -225,7 +225,10 @@
 libpango_1_0_la_LIBADD = mini-fribidi/libmini-fribidi.la $(pango_win32res)
 libpango_1_0_la_DEPENDENCIES = $(pango_win32res)
 
-libpangox_1_0_la_LDFLAGS = -version-info $(LT_VERSION_INFO)
+libpangox_1_0_la_LDFLAGS = \
+	-version-info $(LT_VERSION_INFO)	\
+	$(GLIB_LIBS)				\
+	$(no_undefined)
 libpangox_1_0_la_LIBADD = libpango-$(PANGO_API_VERSION).la $(INCLUDED_X_MODULES) $(X_LIBS) $(X_EXTRA_LIBS)
 libpangox_1_0_la_DEPENDENCIES = $(INCLUDED_X_MODULES) libpango-$(PANGO_API_VERSION).la
 
@@ -252,8 +255,11 @@
 	module-defs-xft.c
 
 
-libpangoxft_1_0_la_LDFLAGS = -version-info $(LT_VERSION_INFO)
-libpangoxft_1_0_la_LIBADD = opentype/libpango-ot.la libpango-$(PANGO_API_VERSION).la $(INCLUDED_XFT_MODULES) $(XFT_LIBS) 
+libpangoxft_1_0_la_LDFLAGS = \
+	-version-info $(LT_VERSION_INFO)	\
+	$(XFT_LIBS) $(X_LIBS) $(GLIB_LIBS)	\
+	$(no_undefined)
+libpangoxft_1_0_la_LIBADD = opentype/libpango-ot.la libpango-$(PANGO_API_VERSION).la $(INCLUDED_XFT_MODULES)
 libpangoxft_1_0_la_DEPENDENCIES = $(INCLUDED_XFT_MODULES) libpango-$(PANGO_API_VERSION).la
 
 libpangoft2_1_0_la_SOURCES = \
@@ -265,7 +271,7 @@
 
 
 libpangoft2_1_0_la_LDFLAGS = -version-info $(LT_VERSION_INFO)	\
-	$(GLIB_LIBS)						\
+	$(FREETYPE_LIBS) $(GLIB_LIBS)				\
 	$(no_undefined) $(pangoft2_export_symbols)
 
 libpangoft2_1_0_la_LIBADD = opentype/libpango-ot.la libpango-$(PANGO_API_VERSION).la $(INCLUDED_FT2_MODULES) $(FREETYPE_LIBS) $(pangoft2_win32res)
diff -Naur pango-1.2.5-old/pango/opentype/pango-ot-info.c pango-1.2.5/pango/opentype/pango-ot-info.c
--- pango-1.2.5-old/pango/opentype/pango-ot-info.c	Wed Apr 16 17:48:20 2003
+++ pango-1.2.5/pango/opentype/pango-ot-info.c	Mon Oct 13 18:34:56 2003
@@ -216,6 +216,8 @@
   /* Collect all the glyphs in the charmap, and guess
    * the appropriate classes for them
    */
+/* Apple's X11 does not have FT_Get_First_Char() */
+#if 0 /* #ifdef HAVE_FT_GET_FIRST_CHAR */
   charcode = FT_Get_First_Char (info->face, &glyph);
   while (glyph != 0)
     {
@@ -231,6 +233,28 @@
       
       charcode = FT_Get_Next_Char (info->face, charcode, &glyph);
     }
+#else
+  {
+    gunichar wc;
+
+    for (wc = 0; wc < G_MAXUSHORT; wc++)
+      {
+        glyph = FT_Get_Char_Index (info->face, wc);
+
+	if (glyph && glyph < info->face->num_glyphs) {
+	  GlyphInfo glyph_info;
+
+	  if (glyph > 65535)
+	    continue;
+
+	  glyph_info.glyph = glyph;
+	  glyph_info.class = get_glyph_class (charcode);
+
+	  g_array_append_val (glyph_infos, glyph_info);
+	}
+      }
+  }
+#endif
 
   /* Sort and remove duplicates
    */
diff -Naur pango-1.2.5-old/update-pango-modules pango-1.2.5/update-pango-modules
--- pango-1.2.5-old/update-pango-modules	Wed Dec 31 19:00:00 1969
+++ pango-1.2.5/update-pango-modules	Mon Oct 13 18:34:56 2003
@@ -0,0 +1,20 @@
+#!/bin/sh -e
+#
+# update-pango-modules
+#
+#   Package which install pango module(s) should call update-pango-modules in
+#   it's PostInstScript/PostRmScript to update pango.module file.
+#
+
+CONFDIR=/opt/etc/pango
+CONFFILE=$CONFDIR/pango.modules
+
+mkdir -p $CONFDIR
+
+TMPFILE=`tempfile`
+pango-querymodules >$TMPFILE
+
+mv -f $TMPFILE $CONFFILE
+chmod 0644 $CONFFILE
+
+exit 0
