diff -Naur pango-1.2.5-old/Makefile.in pango-1.2.5/Makefile.in
--- pango-1.2.5-old/Makefile.in	Wed Aug 27 11:38:53 2003
+++ pango-1.2.5/Makefile.in	Mon Oct 13 18:34:56 2003
@@ -120,7 +120,7 @@
 XFT_LIBS = @XFT_LIBS@
 ms_librarian = @ms_librarian@
 
-SUBDIRS = pango modules examples docs tools tests
+SUBDIRS = pango modules docs tools
 
 EXTRA_DIST = \
 	pango.pc.in	\
diff -Naur pango-1.2.5-old/ltmain.sh pango-1.2.5/ltmain.sh
--- pango-1.2.5-old/ltmain.sh	Thu Jul 17 00:47:23 2003
+++ pango-1.2.5/ltmain.sh	Mon Oct 13 18:49:51 2003
@@ -5145,10 +5145,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test "$need_relink" = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
diff -Naur pango-1.2.5-old/pango/Makefile.in pango-1.2.5/pango/Makefile.in
--- pango-1.2.5-old/pango/Makefile.in	Wed Aug 27 11:38:57 2003
+++ pango-1.2.5/pango/Makefile.in	Mon Oct 13 18:34:56 2003
@@ -253,7 +253,7 @@
 
 
 libpangoxft_1_0_la_LDFLAGS = -version-info $(LT_VERSION_INFO)
-libpangoxft_1_0_la_LIBADD = opentype/libpango-ot.la libpango-$(PANGO_API_VERSION).la $(INCLUDED_XFT_MODULES) $(XFT_LIBS) 
+libpangoxft_1_0_la_LIBADD = opentype/libpango-ot.la libpango-$(PANGO_API_VERSION).la $(INCLUDED_XFT_MODULES) $(XFT_LIBS) $(X_LIBS)
 libpangoxft_1_0_la_DEPENDENCIES = $(INCLUDED_XFT_MODULES) libpango-$(PANGO_API_VERSION).la
 
 libpangoft2_1_0_la_SOURCES = \
diff -Naur pango-1.2.5-old/update-pango-modules pango-1.2.5/update-pango-modules
--- pango-1.2.5-old/update-pango-modules	Wed Dec 31 19:00:00 1969
+++ pango-1.2.5/update-pango-modules	Mon Oct 13 18:34:56 2003
@@ -0,0 +1,20 @@
+#!/bin/sh -e
+#
+# update-pango-modules
+#
+#   Package which install pango module(s) should call update-pango-modules in
+#   it's PostInstScript/PostRmScript to update pango.module file.
+#
+
+CONFDIR=@PREFIX@/etc/pango
+CONFFILE=$CONFDIR/pango.modules
+
+mkdir -p $CONFDIR
+
+TMPFILE=`tempfile`
+pango-querymodules >$TMPFILE
+
+mv -f $TMPFILE $CONFFILE
+chmod 0644 $CONFFILE
+
+exit 0
