diff -Naur zip-2.3/unix/Makefile zip-new/unix/Makefile
--- zip-2.3/unix/Makefile	Sun Nov 28 21:22:42 1999
+++ zip-new/unix/Makefile	Thu May 16 21:19:17 2002
@@ -47,7 +47,7 @@
 #   LFLAGS2   flags after obj file list (libraries, etc)
 CFLAGS = -O2 -I. -DUNIX $(LOCAL_ZIP)
 LFLAGS1 =
-LFLAGS2 = -s
+LFLAGS2 =
 
 # object file lists
 OBJZ = zip.o zipfile.o zipup.o fileio.o util.o globals.o crypt.o ttyio.o \
diff -Naur zip-2.3/unix/configure zip-new/unix/configure
--- zip-2.3/unix/configure	Tue Apr 27 12:49:05 1999
+++ zip-new/unix/configure	Thu May 16 21:42:46 2002
@@ -13,7 +13,7 @@
 
 CC=${1-cc}
 CFLAGS=${2-"-O2 -I. -DUNIX"}
-LFLAGS1="-s"
+LFLAGS1=
 LN="ln -s"
 
 echo Check for the C preprocessor
diff -Naur zip-2.3/zip.c zip-new/zip.c
--- zip-2.3/zip.c	Tue Nov 16 12:08:10 1999
+++ zip-new/zip.c	Thu May 16 21:33:45 2002
@@ -248,7 +248,7 @@
     if (PERR(c))
       perror("zip I/O error");
     fflush(mesg);
-    fprintf(stderr, "\nzip error: %s (%s)\n", errors[c-1], h);
+    fprintf(stderr, "\nzip error: %s (%s)\n", zip_errors[c-1], h);
   }
   if (tempzip != NULL)
   {
diff -Naur zip-2.3/zipcloak.c zip-new/zipcloak.c
--- zip-2.3/zipcloak.c	Sun Nov  7 02:29:59 1999
+++ zip-new/zipcloak.c	Thu May 16 21:48:20 2002
@@ -55,7 +55,7 @@
     char *msg;              /* message about how it happened */
 {
     if (PERR(code)) perror("zipcloak error");
-    fprintf(stderr, "zipcloak error: %s (%s)\n", errors[code-1], msg);
+    fprintf(stderr, "zipcloak error: %s (%s)\n", zip_errors[code-1], msg);
     if (tempzf != NULL) fclose(tempzf);
     if (tempzip != NULL) {
         destroy(tempzip);
diff -Naur zip-2.3/ziperr.h zip-new/ziperr.h
--- zip-2.3/ziperr.h	Sun Nov  7 02:31:27 1999
+++ zip-new/ziperr.h	Thu May 16 21:33:04 2002
@@ -39,7 +39,7 @@
 
 #ifdef GLOBALS
 /* Error messages for the ziperr() function in the zip programs */
-char *errors[ZE_MAXERR] = {
+char *zip_errors[ZE_MAXERR] = {
 /*  1 */  "",
 /*  2 */  "Unexpected end of zip file",
 /*  3 */  "Zip file structure invalid",
@@ -63,5 +63,5 @@
 #  endif
 };
 #else /* !GLOBALS */
-extern char *errors[ZE_MAXERR]; /* Error messages for ziperr() */
+extern char *zip_errors[ZE_MAXERR]; /* Error messages for ziperr() */
 #endif /* ?GLOBALS */
diff -Naur zip-2.3/zipnote.c zip-new/zipnote.c
--- zip-2.3/zipnote.c	Sun Nov  7 02:30:22 1999
+++ zip-new/zipnote.c	Thu May 16 21:45:56 2002
@@ -59,7 +59,7 @@
 {
   if (PERR(c))
     perror("zipnote error");
-  fprintf(stderr, "zipnote error: %s (%s)\n", errors[c-1], h);
+  fprintf(stderr, "zipnote error: %s (%s)\n", zip_errors[c-1], h);
   if (tempzf != NULL)
     fclose(tempzf);
   if (tempzip != NULL)
diff -Naur zip-2.3/zipsplit.c zip-new/zipsplit.c
--- zip-2.3/zipsplit.c	Sun Nov  7 02:30:31 1999
+++ zip-new/zipsplit.c	Thu May 16 21:48:30 2002
@@ -139,7 +139,7 @@
 {
   if (PERR(c))
     perror("zipsplit error");
-  fprintf(stderr, "zipsplit error: %s (%s)\n", errors[c-1], h);
+  fprintf(stderr, "zipsplit error: %s (%s)\n", zip_errors[c-1], h);
   if (indexmade)
   {
     strcpy(name, INDEX);
