--- asr-manpages-1.3.orig/copyright
+++ asr-manpages-1.3/copyright
@@ -0,0 +1,27 @@
+This package was debianized by Pawel Wiecek coven@pwr.wroc.pl on
+Sun, 1 Jun 1997 23:13:56 +0200.
+
+It was downloaded from ftp://ftp.winternet.com/users/eric/asr.pages.tar
+
+Uubp page is missing from distribution - it was reverse-engeneered from
+HTML version found on http://www.bofh.org.pl/man/.
+
+The authors of the pages are:
+ - c - Nick Williams <nickw@iname.com>
+ - lart - Tim Bandy <timn8r@risk.cs.umn.edu>
+ - slave - Ingvar Mattsson
+ - sysadmin - Abby Franquemont-Guillory <abbyfg@tezcat.com>
+ - think - John Guthrie <guthrie@math.upenn.edu>
+ - whack - Eric L. Pederson <eric@winternet.com>
+ - people - Ingvar Mattsson
+ - chastise - Ingvar Mattsson
+ - normality - Ingvar Mattsson
+ - bosskill - Eric L. Pederson <eric@winternet.com>
+ - ctluser - Simon Burr <simes@tcp.co.uk>
+ - guru - unknown author <user@foo.com>
+ - knife - Kurt Hockenbury <kurt@something.com>
+ - luser - Eric L. Pederson <eric@winternet.com>
+ - nuke - Eric S. Raymond <esr@thyrus.com>
+ - pmsd - Eric L. Pederson <eric@winternet.com>
+
+To the best of my knowledge they are all completely free.
diff -ru2N asr-manpages-1.3.orig/c.1fun asr-manpages-1.3/c.1fun
--- asr-manpages-1.3.orig/c.1fun	Sun Jul 27 18:38:16 1997
+++ asr-manpages-1.3/c.1fun	Sat Dec 21 15:48:23 2002
@@ -86,5 +86,5 @@
 
 Use of -S 6 is contraindicated  for use with  n(1), due to  
-possible  damage  to t he  nasal  membranes,  and risk  of  
+possible  damage  to the  nasal  membranes,  and risk  of  
 drowning.  The default is 4; unless a beta version of  the 
 OS is installed, when the default is 5.  If the beta is of
diff -ru2N asr-manpages-1.3.orig/chastise.3fun asr-manpages-1.3/chastise.3fun
--- asr-manpages-1.3.orig/chastise.3fun	Thu May 29 17:37:10 1997
+++ asr-manpages-1.3/chastise.3fun	Sat Dec 21 15:48:23 2002
@@ -19,5 +19,5 @@
 .B Chastise
 normally returns a counter on how many times the luser specified has
-been \fBchastised\fP, but if any error occur, it returns
+been \fBchastised\fP, but if any error occurs, it returns
 .B -1
 and no explanation, what so ever, is made anywhere.
--- asr-manpages-1.3.orig/knife.8fun
+++ asr-manpages-1.3/knife.8fun
@@ -1,4 +1,4 @@
 .TH KNIFE 8 
-.SH KNIFE
+.SH NAME
 knife, axe, cutter, chainsaw \- tools to improve network performance via SNIP
 .SH SYNOPSIS
diff -ru2N asr-manpages-1.3.orig/lart.1fun asr-manpages-1.3/lart.1fun
--- asr-manpages-1.3.orig/lart.1fun	Thu May 29 17:37:10 1997
+++ asr-manpages-1.3/lart.1fun	Sat Dec 21 15:48:23 2002
@@ -30,5 +30,5 @@
 .TP 15
 .BI \-use " tool"
-Choose your weapon.  If this option is not specified, the enviroment
+Choose your weapon.  If this option is not specified, the environment
 variable LART is used.  If neither is set, a default value will be
 used, which is set at compile time
@@ -38,5 +38,5 @@
 .TP
 .BI \-B
-The BOFH option.  Removes a users files and kills all their processes.
+The BOFH option.  Removes a user's files and kills all their processes.
 If used in conjunction with
 .B \-p
--- asr-manpages-1.3.orig/luser.8fun
+++ asr-manpages-1.3/luser.8fun
@@ -3,5 +3,5 @@
 .TH LUSER 8 "25 September 1995"
 .SH NAME
-luser \- process to control the clueless induhviduals
+luser \- process to control the clueless individuals
 who (mis)use computer systems, peripheral devices
 and system administrators. Word play on "loser" and "user".
--- asr-manpages-1.3.orig/people.2fun
+++ asr-manpages-1.3/people.2fun
@@ -28,5 +28,5 @@
   int is_aluser;
   int has_aclue;
-  pid_t prgp_leader;
+  pid_t pgrp_leader;
 };
 .fi
@@ -58,5 +58,5 @@
 .SH EXAMPLE
 .nf
-#include <sys/pople.h>
+#include <sys/people.h>
 #include <asr.h>
 #include <signal.h>
@@ -72,5 +72,5 @@
     for (c=0;c<rv;c++)
       if ((ttys[c].is_amoron)||(ttys[c].is_aluser)) {
-	kill(-(ttys[c].pgrp_leader),SIGKILL);
+        kill(-(ttys[c].pgrp_leader),SIGKILL);
       }
   } else {
--- asr-manpages-1.3.orig/slave.1fun
+++ asr-manpages-1.3/slave.1fun
@@ -19,5 +19,5 @@
 .TP
 \-\fIf\fP
-Send a rapid succession of \fISIHUP\fPs to \fIslave\fP, mainly used
+Send a rapid succession of \fISIGHUP\fPs to \fIslave\fP, mainly used
 when one needs to blow off a little steam.
 .SH THE COMMANDS
--- asr-manpages-1.3.orig/sysadmin.1fun
+++ asr-manpages-1.3/sysadmin.1fun
@@ -1,4 +1,4 @@
-\." Written by Abby Franquement-Guillory (abbyfg@tezcat.com).
-\." Convered to troff by Eric Pederson (eric@winternet.com).
+.\" Written by Abby Franquement-Guillory (abbyfg@tezcat.com).
+.\" Convered to troff by Eric Pederson (eric@winternet.com).
 .TH SYSADMIN 1 "25 September 1995"
 .SH NAME
diff -ru2N asr-manpages-1.3.orig/think.1fun asr-manpages-1.3/think.1fun
--- asr-manpages-1.3.orig/think.1fun	Thu May 29 17:37:10 1997
+++ asr-manpages-1.3/think.1fun	Sat Dec 21 15:48:23 2002
@@ -21,5 +21,5 @@
 user is wearing a brain interface bus (bib) device.  A bib device is normally
 worn on the head, and if being used, then \fIthink\fP will try to see what was
-going through the users head at the time of invocation.  After \fIthink\fP
+going through the user's head at the time of invocation.  After \fIthink\fP
 does this, it will send electric signals to the users brain, causing the
 user to type in whatever keystrokes are necessary to accomplish the task
@@ -42,5 +42,5 @@
 .B /usr/lib/think
 file to indicate various user abilities.  The format of this file is a
-username on each line followed by some whitspace and then a number.  The
+username on each line followed by some whitespace and then a number.  The
 higher the number for a given user, the more likely \fIthink\fP is to assume
 that that user knows what he/she is doing.  Unfortunately, what \fIthink\fP
