--- gtkmonop-0.3.0/src/client.c.orig	Mon Oct 29 01:40:49 2001
+++ gtkmonop-0.3.0/src/client.c	Mon Oct 29 01:48:35 2001
@@ -20,7 +20,9 @@
 /* client.c: connects to the monopd server and adds the gIoChannel to the main
    event loop */
 
+#include <sys/types.h>
 #include <sys/socket.h>
+#include <netinet/in.h>
 #include <sys/wait.h>
 #include <netdb.h>
 #include <unistd.h>
--- gtkmonop-0.3.0/src/malloc.h.orig	Mon Oct 29 01:35:01 2001
+++ gtkmonop-0.3.0/src/malloc.h	Mon Oct 29 01:34:56 2001
@@ -0,0 +1 @@
+/* Dummy file for stupid programmer */
