--- ./wtf-1.7-1.sh	Tue Sep 10 21:28:23 2002
+++ ./wtf-1.7-1.sh-patched	Tue Sep 10 21:30:03 2002
@@ -6,5 +6,5 @@
 #
 
-acronyms=${ACRONYMDB:-/usr/share/misc/acronyms}
+acronyms=${ACRONYMDB:-@PREFIX@/share/wtf/acronyms}
 
 if [ X"$1" = X"is" ] ; then
--- ./wtf-1.7-1.man	Tue Sep 10 21:42:38 2002
+++ ./wtf-1.7-1.man-patched	Tue Sep 10 21:45:00 2002
@@ -38,6 +38,6 @@
 .El
 .Sh FILES
-.Bl -tag -width /usr/share/misc/acronyms -compact
-.It Pa /usr/share/misc/acronyms
+.Bl -tag -width @PREFIX@/share/wtf/acronyms -compact
+.It Pa @PREFIX@/share/wtf/acronyms
 default acronym database.
 .El
