diff -Naur XFrisk/Makefile XFrisk-new/Makefile
--- XFrisk/Makefile	2000-01-23 14:01:55.000000000 -0500
+++ XFrisk-new/Makefile	2002-07-29 16:01:35.000000000 -0400
@@ -54,8 +54,8 @@
 
 ####################
 # C compiler options
-# GNU gcc is recommended.
-CC=gcc
+# GNU cc is recommended.
+CC=cc
 # Use these with gcc
 CFLAGS=-g -Wall -W -fno-common -pedantic
 
@@ -81,7 +81,8 @@
 
 ## Installation prefix
 # Adjust to taste. Stuff gets installed here.
-PREFIX=/usr/local
+PREFIX=@PREFIX@
+DESTDIR=@DESTDIR@
 #PREFIX=/usr/local/X11
 
 ## X11 location and options
@@ -208,16 +209,16 @@
 	$(RANLIB) $@
 
 install: all
-	-(umask 022; mkdir -p -m755 $(LIBDIR) $(BINDIR) $(APPLOADDIR))
-	$(INSTALL_DATA) $(MAP) $(LIBDIR)
-	$(INSTALL_DATA) $(COUNTRY) $(LIBDIR)
-	$(INSTALL_DATA) $(HELP) $(LIBDIR)
-	$(INSTALL_BIN) risk $(BINDIR)
-	$(INSTALL_BIN) xfrisk $(BINDIR)
-	$(INSTALL_BIN) friskserver $(BINDIR)
-	$(INSTALL_BIN) aiDummy $(BINDIR)
-	$(INSTALL_BIN) aiConway $(BINDIR)
-	$(INSTALL_BIN) aiColson $(BINDIR)
+	-(umask 022; mkdir -p -m755 $(DESTDIR)/$(LIBDIR) $(DESTDIR)/$(BINDIR) $(APPLOADDIR))
+	$(INSTALL_DATA) $(MAP) $(DESTDIR)/$(LIBDIR)
+	$(INSTALL_DATA) $(COUNTRY) $(DESTDIR)/$(LIBDIR)
+	$(INSTALL_DATA) $(HELP) $(DESTDIR)/$(LIBDIR)
+	$(INSTALL_BIN) risk $(DESTDIR)/$(BINDIR)
+	$(INSTALL_BIN) xfrisk $(DESTDIR)/$(BINDIR)
+	$(INSTALL_BIN) friskserver $(DESTDIR)/$(BINDIR)
+	$(INSTALL_BIN) aiDummy $(DESTDIR)/$(BINDIR)
+	$(INSTALL_BIN) aiConway $(DESTDIR)/$(BINDIR)
+	$(INSTALL_BIN) aiColson $(DESTDIR)/$(BINDIR)
 
 
 # Other targets
