diff -Nuar sary-ruby-1.2.0.orig/extconf.rb sary-ruby-1.2.0/extconf.rb
--- sary-ruby-1.2.0.orig/extconf.rb	Wed Mar 30 07:48:09 2005
+++ sary-ruby-1.2.0/extconf.rb	Mon Apr  4 15:05:06 2005
@@ -12,7 +12,7 @@
 pkg_config = 'pkg-config'
 
 unless system("#{pkg_config} sary --exists") then
-  ENV['PKG_CONFIG_PATH'] = '/usr/local/lib/pkgconfig/'
+  ENV['PKG_CONFIG_PATH'] = '@PREFIX@/lib/pkgconfig/'
 end
 
 $CFLAGS  += ' ' + `#{pkg_config} sary --cflags`.chomp
diff -Nuar sary-ruby-1.2.0.orig/searcher.c sary-ruby-1.2.0/searcher.c
--- sary-ruby-1.2.0.orig/searcher.c	Wed Mar 30 07:48:09 2005
+++ sary-ruby-1.2.0/searcher.c	Mon Apr  4 15:06:44 2005
@@ -137,7 +137,7 @@
     pat = StringValuePtr(pattern);
     len = RSTRING(pattern)->len;
 #else
-    pat = str2cstr(pattern, &len);
+    pat = rb_str2cstr(pattern, &len);
 #endif
 
     if (sary_searcher_search2(searcher, pat, len))
