--- CLHEP.orig/Random/src/TripleRand.cc	2010-06-16 19:24:53.000000000 +0200
+++ CLHEP.new/Random/src/TripleRand.cc	2011-08-02 15:29:22.000000000 +0200
@@ -200,13 +200,13 @@
 
 TripleRand::operator float() {
   return (float)
-    ( ( integerCong ^ tausworthe ^ hurd ) * twoToMinus_32() 
+    ( ( integerCong ^ tausworthe ^ (unsigned int)hurd ) * twoToMinus_32() 
 					+ nearlyTwoToMinus_54() );
 					// make sure non-zero!
 }
 
 TripleRand::operator unsigned int() {
-  return integerCong ^ tausworthe ^ hurd;
+  return integerCong ^ tausworthe ^ (unsigned int)hurd;
 }
 
 Hurd288Engine & TripleRand::Hurd() 	       { return hurd; }
