diff -Nurd -x'*~' EMBOSS-6.5.7.orig/Makefile.am EMBOSS-6.5.7/Makefile.am
--- EMBOSS-6.5.7.orig/Makefile.am	2012-07-22 07:16:46.000000000 -0400
+++ EMBOSS-6.5.7/Makefile.am	2013-05-25 19:37:43.000000000 -0400
@@ -12,8 +12,11 @@
 ONEWS \
 PROBLEMS
 
+if !ESYSTEMLIBS
+EXTRA_DIRS = plplot
+endif
 SUBDIRS = \
-plplot \
+$(EXTRA_DIRS) \
 ajax \
 nucleus \
 emboss \
@@ -34,4 +37,3 @@
 	tar cBf - jemboss | ( cd $(distdir); tar xBf - ; find jemboss -name CVS | xargs rm -rf; find jemboss -name Makefile | xargs rm -rf; find jemboss -name .cvsignore | xargs rm -rf )
 
 install-exec-hook:
-	$(bindir)/embossupdate
diff -Nurd -x'*~' EMBOSS-6.5.7.orig/ajax/Makefile.am EMBOSS-6.5.7/ajax/Makefile.am
--- EMBOSS-6.5.7.orig/ajax/Makefile.am	2011-08-25 15:45:55.000000000 -0400
+++ EMBOSS-6.5.7/ajax/Makefile.am	2013-05-25 19:40:09.000000000 -0400
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS  = pcre
+SUBDIRS  = 
 if !ESYSTEMLIBS
-SUBDIRS += expat zlib
+SUBDIRS += pcre expat zlib
 endif
 SUBDIRS += core graphics ensembl ajaxdb acd
diff -Nurd -x'*~' EMBOSS-6.5.7.orig/ajax/acd/Makefile.am EMBOSS-6.5.7/ajax/acd/Makefile.am
--- EMBOSS-6.5.7.orig/ajax/acd/Makefile.am	2012-03-23 16:19:27.000000000 -0400
+++ EMBOSS-6.5.7/ajax/acd/Makefile.am	2013-05-25 19:43:49.000000000 -0400
@@ -7,12 +7,13 @@
 libacd_la_CPPFLAGS  = -I$(top_srcdir)/ajax/ajaxdb
 libacd_la_CPPFLAGS += -I$(top_srcdir)/ajax/ensembl
 libacd_la_CPPFLAGS += -I$(top_srcdir)/ajax/graphics
-libacd_la_CPPFLAGS += -I$(top_srcdir)/ajax/pcre
 if !ESYSTEMLIBS
 libacd_la_CPPFLAGS += -I$(top_srcdir)/ajax/expat
 libacd_la_CPPFLAGS += -I$(top_srcdir)/ajax/zlib
-endif
 libacd_la_CPPFLAGS += -I$(top_srcdir)/plplot
+libacd_la_CPPFLAGS += -I$(top_srcdir)/ajax/pcre
+endif
+libacd_la_CPPFLAGS += $(PLPLOT_CFLAGS)
 libacd_la_CPPFLAGS += $(JAVA_CPPFLAGS)
 
 libacd_la_LDFLAGS  = -version-info 6:4:0
@@ -38,6 +39,7 @@
 libacd_la_LDFLAGS += -lepcre
 libacd_la_LDFLAGS += -lezlib
 endif
+libacd_la_LIBADD = ../core/libajax.la ../graphics/libajaxg.la ../ajaxdb/libajaxdb.la
 
 libacd_la_SOURCES = ajacd.c
 
diff -Nurd -x'*~' EMBOSS-6.5.7.orig/ajax/ajaxdb/Makefile.am EMBOSS-6.5.7/ajax/ajaxdb/Makefile.am
--- EMBOSS-6.5.7.orig/ajax/ajaxdb/Makefile.am	2012-03-23 16:19:27.000000000 -0400
+++ EMBOSS-6.5.7/ajax/ajaxdb/Makefile.am	2013-05-25 19:45:08.000000000 -0400
@@ -5,12 +5,12 @@
 libajaxdb_la_CFLAGS = $(WARN_CFLAGS) $(DEVWARN_CFLAGS) $(JAVA_CFLAGS)
 
 libajaxdb_la_CPPFLAGS  = -I$(top_srcdir)/ajax/ensembl
-libajaxdb_la_CPPFLAGS += -I$(top_srcdir)/ajax/pcre
 if !ESYSTEMLIBS
 libajaxdb_la_CPPFLAGS += -I$(top_srcdir)/ajax/expat
 libajaxdb_la_CPPFLAGS += -I$(top_srcdir)/ajax/zlib
-endif
+libajaxdb_la_CPPFLAGS += -I$(top_srcdir)/ajax/pcre
 libajaxdb_la_CPPFLAGS += -I$(top_srcdir)/plplot
+endif
 libajaxdb_la_CPPFLAGS += $(AXIS2C_CPPFLAGS)
 libajaxdb_la_CPPFLAGS += $(JAVA_CPPFLAGS)
 
@@ -34,6 +34,7 @@
 libajaxdb_la_LDFLAGS += -lezlib
 endif
 libajaxdb_la_LDFLAGS += $(AXIS2C_LDFLAGS)
+libajaxdb_la_LIBADD = ../core/libajax.la ../ensembl/libensembl.la
 
 libajaxdb_la_SOURCES = \
 ajassemdb.c \
diff -Nurd -x'*~' EMBOSS-6.5.7.orig/ajax/core/Makefile.am EMBOSS-6.5.7/ajax/core/Makefile.am
--- EMBOSS-6.5.7.orig/ajax/core/Makefile.am	2012-03-23 16:19:27.000000000 -0400
+++ EMBOSS-6.5.7/ajax/core/Makefile.am	2013-05-25 21:34:12.000000000 -0400
@@ -4,12 +4,19 @@
 
 libajax_la_CFLAGS = $(WARN_CFLAGS) $(DEVWARN_CFLAGS) $(JAVA_CFLAGS)
 
-libajax_la_CPPFLAGS  = -I$(top_srcdir)/ajax/pcre
+libajax_la_CPPFLAGS  =
 if !ESYSTEMLIBS
 libajax_la_CPPFLAGS += -I$(top_srcdir)/ajax/expat
 libajax_la_CPPFLAGS += -I$(top_srcdir)/ajax/zlib
-endif
+libajax_la_CPPFLAGS += -I$(top_srcdir)/ajax/pcre
 libajax_la_CPPFLAGS += -I$(top_srcdir)/plplot
+else
+EXTRA_LIBS = -lexpat -lpcre
+endif
+
+EXTRA_DIST = ajax-pcre-config.h.in
+DISTCLEAN = ajax-pcre-config.h
+
 libajax_la_CPPFLAGS += $(JAVA_CPPFLAGS)
 libajax_la_CPPFLAGS += $(MYSQL_CPPFLAGS)
 libajax_la_CPPFLAGS += $(POSTGRESQL_CPPFLAGS)
@@ -35,6 +42,7 @@
 libajax_la_LDFLAGS += -lepcre
 libajax_la_LDFLAGS += -lezlib
 endif
+libajax_la_LIBADD = $(EXTRA_LIBS) $(POSTGRESQL_LDFLAGS) $(MYSQL_LDFLAGS)
 
 libajax_la_SOURCES = \
 ajalign.c \
@@ -172,6 +180,7 @@
 ajquerydata.h \
 ajrange.h \
 ajreg.h \
+ajax-pcre-config.h \
 ajreport.h \
 ajrefseq.h \
 ajrefseqdata.h \
diff -Nurd -x'*~' EMBOSS-6.5.7.orig/ajax/core/ajax-pcre-config.h.in EMBOSS-6.5.7/ajax/core/ajax-pcre-config.h.in
--- EMBOSS-6.5.7.orig/ajax/core/ajax-pcre-config.h.in	1969-12-31 19:00:00.000000000 -0500
+++ EMBOSS-6.5.7/ajax/core/ajax-pcre-config.h.in	2013-05-25 19:50:00.000000000 -0400
@@ -0,0 +1 @@
+@DEFINE_USE_SYSTEM_PCRE@ AJAX_USE_SYSTEM_PCRE
diff -Nurd -x'*~' EMBOSS-6.5.7.orig/ajax/core/ajreg.h EMBOSS-6.5.7/ajax/core/ajreg.h
--- EMBOSS-6.5.7.orig/ajax/core/ajreg.h	2011-10-18 10:23:40.000000000 -0400
+++ EMBOSS-6.5.7/ajax/core/ajreg.h	2013-05-25 19:51:43.000000000 -0400
@@ -39,9 +39,14 @@
 #include "ajdefine.h"
 #include "ajstr.h"
 
+#include "ajax-pcre-config.h"
+#ifndef AJAX_USE_SYSTEM_PCRE
 #include "pcre_config.h"
 #include "pcre_internal.h"
 #include "pcreposix.h"
+#else
+#include <pcre.h>
+#endif
 
 AJ_BEGIN_DECLS
 
@@ -84,7 +88,11 @@
 
 typedef struct AjSRegexp
 {
+#ifndef AJAX_USE_SYSTEM_PCRE
     real_pcre *pcre;
+#else
+	pcre *pcre;
+#endif
     pcre_extra *extra;
     int *ovector;
     const char* orig;
diff -Nurd -x'*~' EMBOSS-6.5.7.orig/ajax/ensembl/Makefile.am EMBOSS-6.5.7/ajax/ensembl/Makefile.am
--- EMBOSS-6.5.7.orig/ajax/ensembl/Makefile.am	2012-03-23 16:19:27.000000000 -0400
+++ EMBOSS-6.5.7/ajax/ensembl/Makefile.am	2013-05-25 21:34:32.000000000 -0400
@@ -4,12 +4,13 @@
 
 libensembl_la_CFLAGS = $(WARN_CFLAGS) $(DEVWARN_CFLAGS) $(JAVA_CFLAGS)
 
-libensembl_la_CPPFLAGS  = -I$(top_srcdir)/ajax/pcre
+libensembl_la_CPPFLAGS  =
 if !ESYSTEMLIBS
 libensembl_la_CPPFLAGS += -I$(top_srcdir)/ajax/expat
 libensembl_la_CPPFLAGS += -I$(top_srcdir)/ajax/zlib
-endif
 libensembl_la_CPPFLAGS += -I$(top_srcdir)/plplot
+libensembl_la_CPPFLAGS += -I$(top_srcdir)/ajax/pcre
+endif
 libensembl_la_CPPFLAGS += $(JAVA_CPPFLAGS)
 
 libensembl_la_LDFLAGS  = -version-info 6:4:0
@@ -27,6 +28,7 @@
 libensembl_la_LDFLAGS += -leexpat
 libensembl_la_LDFLAGS += -lepcre
 endif
+libensembl_la_LIBADD = ../core/libajax.la
 
 libensembl_la_SOURCES = \
 ensalign.c \
diff -Nurd -x'*~' EMBOSS-6.5.7.orig/ajax/graphics/Makefile.am EMBOSS-6.5.7/ajax/graphics/Makefile.am
--- EMBOSS-6.5.7.orig/ajax/graphics/Makefile.am	2012-03-23 16:19:27.000000000 -0400
+++ EMBOSS-6.5.7/ajax/graphics/Makefile.am	2013-05-25 21:35:17.000000000 -0400
@@ -4,11 +4,15 @@
 
 libajaxg_la_CFLAGS = $(WARN_CFLAGS) $(DEVWARN_CFLAGS) $(JAVA_CFLAGS)
 
-libajaxg_la_CPPFLAGS  = -I$(top_srcdir)/ajax/pcre
+libajaxg_la_CPPFLAGS  =
 if !ESYSTEMLIBS
 libajaxg_la_CPPFLAGS += -I$(top_srcdir)/ajax/expat
-endif
+libajaxg_la_CPPFLAGS += -I$(top_srcdir)/ajax/pcre
 libajaxg_la_CPPFLAGS += -I$(top_srcdir)/plplot
+else
+libajaxg_la_CPPFLAGS += $(PLPLOT_CFLAGS) -DUSE_PLXSFNAM_SHIM
+EXTRA_LIBS = $(PLPLOT_LIBS)
+endif
 libajaxg_la_CPPFLAGS += $(JAVA_CPPFLAGS)
 
 libajaxg_la_LDFLAGS  = -version-info 6:4:0
@@ -26,6 +27,7 @@
 libajaxg_la_LDFLAGS += -leexpat
 libajaxg_la_LDFLAGS += -lepcre
 endif
+libajaxg_la_LIBADD = ../core/libajax.la $(EXTRA_LIBS)
 
 libajaxg_la_SOURCES = ajgraph.c ajhist.c
 
diff -Nurd -x'*~' EMBOSS-6.5.7.orig/ajax/graphics/ajgraph.c EMBOSS-6.5.7/ajax/graphics/ajgraph.c
--- EMBOSS-6.5.7.orig/ajax/graphics/ajgraph.c	2011-11-08 10:12:19.000000000 -0500
+++ EMBOSS-6.5.7/ajax/graphics/ajgraph.c	2013-05-25 19:55:59.000000000 -0400
@@ -963,11 +963,20 @@
 {
     if(!thys->ready)
     {
+#ifdef USE_PLXSFNAM_SHIM
+        char *fullname;
+#endif
 #if GRAPH_DEBUG
         ajDebug("=g= plxsfnam ('%S', '%s')\n", txt, ext);
 #endif
-
+#ifdef USE_PLXSFNAM_SHIM
+        fullname = asprintf(fullname,"%s%s", ajStrGetPtr(txt), ext);
+        ajDebug("=g= plsfnam ('%S') instead\n", fullname);
+        plsfnam(fullname);
+        free(fullname);
+#else
         plxsfnam(ajStrGetPtr(txt), ext);
+#endif
         ajStrAssignS(&graphBasename, txt);
         ajStrAssignC(&graphExtension, ext);
         if(ajStrGetCharFirst(graphExtension) == '.')
diff -Nurd -x'*~' EMBOSS-6.5.7.orig/configure.in EMBOSS-6.5.7/configure.in
--- EMBOSS-6.5.7.orig/configure.in	2012-07-24 10:06:02.000000000 -0400
+++ EMBOSS-6.5.7/configure.in	2013-05-25 20:10:04.000000000 -0400
@@ -707,7 +707,7 @@
 
 AC_CHECK_FUNCS([memmove])
 
-AS_IF([test "x${with_x}" != "xno"], [LF_EMBOSS_PATH_XLIB])
+dnl AS_IF([test "x${with_x}" != "xno"], [LF_EMBOSS_PATH_XLIB])
 
 
 # Library checks.
@@ -846,9 +846,14 @@
 
 AC_ARG_ENABLE([systemlibs],
 [AS_HELP_STRING([--enable-systemlibs], [utility for RPM/dpkg bundles])])
+DEFINE_USE_SYSTEM_PCRE="#undef"
 
 AM_CONDITIONAL([ESYSTEMLIBS], [test "x${enable_systemlibs}" = "xyes"])
-
+DEFINE_USE_SYSTEM_PCRE="#define"
+PKG_CHECK_MODULES(PLPLOT, [plplotd]) 
+AC_SUBST(DEFINE_USE_SYSTEM_PCRE)
+AC_SUBST(PLPLOT_CFLAGS)
+AC_SUBST(PLPLOT_LIBS)
 
 
 
@@ -988,6 +991,7 @@
                  ajax/acd/Makefile
                  ajax/ajaxdb/Makefile
                  ajax/core/Makefile
+                 ajax/core/ajax-pcre-config.h
                  ajax/ensembl/Makefile
                  ajax/expat/Makefile
                  ajax/graphics/Makefile
diff -Nurd -x'*~' EMBOSS-6.5.7.orig/emboss/Makefile.am EMBOSS-6.5.7/emboss/Makefile.am
--- EMBOSS-6.5.7.orig/emboss/Makefile.am	2012-07-14 12:46:15.000000000 -0400
+++ EMBOSS-6.5.7/emboss/Makefile.am	2013-05-25 20:13:39.000000000 -0400
@@ -30,13 +30,14 @@
 AM_CPPFLAGS += -I$(top_srcdir)/ajax/ajaxdb
 AM_CPPFLAGS += -I$(top_srcdir)/ajax/ensembl
 AM_CPPFLAGS += -I$(top_srcdir)/ajax/graphics
-AM_CPPFLAGS += -I$(top_srcdir)/ajax/pcre
 if !ESYSTEMLIBS
 AM_CPPFLAGS += -I$(top_srcdir)/ajax/expat
 AM_CPPFLAGS += -I$(top_srcdir)/ajax/zlib
+AM_CPPFLAGS += -I$(top_srcdir)/ajax/pcre
+AM_CPPFLAGS += -I$(top_srcdir)/plplot
 endif
+AM_CPPFLAGS += $(PLPLOT_CFLAGS)
 AM_CPPFLAGS += -I$(top_srcdir)/nucleus
-AM_CPPFLAGS += -I$(top_srcdir)/plplot
 AM_CPPFLAGS += $(AXIS2C_CPPFLAGS)
 AM_CPPFLAGS += $(JAVA_CPPFLAGS)
 
@@ -49,9 +50,9 @@
 if !ESYSTEMLIBS
 LDADD += ../ajax/zlib/libezlib.la
 LDADD += ../ajax/expat/libeexpat.la
-endif
-LDADD += ../ajax/pcre/libepcre.la
 LDADD += ../plplot/libeplplot.la
+LDADD += ../ajax/pcre/libepcre.la
+endif
 LDADD += $(XLIB)
 
 #if PURIFY
diff -Nurd -x'*~' EMBOSS-6.5.7.orig/nucleus/Makefile.am EMBOSS-6.5.7/nucleus/Makefile.am
--- EMBOSS-6.5.7.orig/nucleus/Makefile.am	2011-08-25 17:18:40.000000000 -0400
+++ EMBOSS-6.5.7/nucleus/Makefile.am	2013-05-25 21:35:32.000000000 -0400
@@ -10,12 +10,16 @@
 endif
 endif
 
-libnucleus_la_CPPFLAGS  = -I$(top_srcdir)/plplot
-libnucleus_la_CPPFLAGS += -I$(top_srcdir)/ajax/pcre
+libnucleus_la_CPPFLAGS  =
 if !ESYSTEMLIBS
 libnucleus_la_CPPFLAGS += -I$(top_srcdir)/ajax/expat
 libnucleus_la_CPPFLAGS += -I$(top_srcdir)/ajax/zlib
+libnucleus_la_CPPFLAGS += -I$(top_srcdir)/plplot
+libnucleus_la_CPPFLAGS += -I$(top_srcdir)/ajax/pcre
+else
+libnucleus_la_CPPFLAGS +=  $(PLPLOT_CFLAGS)
 endif
+
 libnucleus_la_CPPFLAGS += -I$(top_srcdir)/ajax/graphics
 libnucleus_la_CPPFLAGS += -I$(top_srcdir)/ajax/ensembl
 libnucleus_la_CPPFLAGS += -I$(top_srcdir)/ajax/ajaxdb
@@ -48,6 +52,7 @@
 libnucleus_la_LDFLAGS += -lezlib
 libnucleus_la_LDFLAGS += -leplplot
 endif
+libnucleus_la_LIBADD = ../ajax/core/libajax.la ../ajax/graphics/libajaxg.la ../ajax/acd/libacd.la
 
 libnucleus_la_SOURCES = \
 embaln.c \
