diff -ur -x Makefile -x 'config*' -x libtool euler-1.61.0/src/command.c euler-1.61.0_corr/src/command.c
--- euler-1.61.0/src/command.c	2005-10-24 02:38:24.000000000 +0200
+++ euler-1.61.0_corr/src/command.c	2007-10-06 09:05:14.000000000 +0200
@@ -235,7 +235,7 @@
 
 extern int builtin_count;
 extern builtintyp builtin_list[];
-extern int command_count;
+static int command_count;
 extern commandtyp command_list[];
 
 static void do_list (void)
@@ -732,7 +732,6 @@
 }
 
 
-static int command_count;
 
 commandtyp command_list[] = {
 	{"quit",c_quit,do_quit},
diff -ur -x Makefile -x 'config*' -x libtool euler-1.61.0/src/main.c euler-1.61.0_corr/src/main.c
--- euler-1.61.0/src/main.c	2005-10-30 23:54:17.000000000 +0100
+++ euler-1.61.0_corr/src/main.c	2007-10-06 09:55:07.000000000 +0200
@@ -439,7 +439,8 @@
 	gettimer(TIMEOFDAY,&t);
 	return (t.tv_sec+t.tv_nsec/1000000000.0);
 #else
-	return ((double)(times(NULL)))/CLK_TCK;
+        static struct tms tp;
+        return ((double)(times(&tp)))/CLK_TCK;
 #endif
 }
 
