--- gaiw/MainDialog.ui.h	Mon Mar  8 21:12:41 2004
+++ gaiw-patched/MainDialog.ui.h	Mon Mar  8 21:12:41 2004
@@ -1557,7 +1557,7 @@
      // Finally, call Grace
      // We open the Grace window in fullscreen mode thanks to a very simple trick:
      // specify a very large geometry!
-     system("xmgrace -geometry 10000x10000 -nosafe -b "+outputFile+"&");
+     system("xmgrace -nosafe -b "+outputFile+"&");
      //system("xmgrace -geometry 10000x10000 -b "+outputFile+"&"); For versions up to 5.1.7!
  }
 
