--- geant4.8.orig/fink.sh	1969-12-31 16:00:00.000000000 -0800
+++ geant4.8.new/fink.sh	2004-01-17 11:44:39.000000000 -0800
@@ -0,0 +1,65 @@
+g4clhep_base_dir='@PREFIX@'
+g4clhep_include_dir='@PREFIX@/include'
+g4clhep_lib='CLHEP'
+g4clhep_lib_dir='@PREFIX@/lib'
+g4compiler='g++'
+g4data='@PREFIX@/share/geant4.8/data'
+g4debug='n'
+d_portable='define'
+g4global='n'
+g4granular='y'
+g4include=''
+g4includes_flag='y'
+g4install='@BUILDDIR@'
+g4lib_build_shared='y'
+g4lib_build_static='y'
+g4lib_use_granular='y'
+g4lib_use_shared='n'
+g4lib_use_static='y'
+g4make='make'
+g4osname='Darwin'
+g4system='Darwin-g++'
+g4ui_build_gag_session='y'
+g4ui_build_terminal_session='y'
+g4ui_build_win32_session='n'
+g4ui_build_xaw_session='n'
+g4ui_build_xm_session='y'
+g4ui_use_gag='y'
+g4ui_use_tcsh='y'
+g4ui_use_terminal='y'
+g4ui_use_win32='n'
+g4ui_use_xaw='n'
+g4ui_use_xm='y'
+g4vis_build_asciitree_driver='y'
+g4vis_build_dawn_driver='y'
+g4vis_build_dawnfile_driver='y'
+g4vis_build_oiwin32_driver='n'
+g4vis_build_oix_driver='n'
+g4vis_build_openglwin32_driver='n'
+g4vis_build_openglx_driver='y'
+g4vis_build_openglxm_driver='y'
+g4vis_build_raytracer_driver='y'
+g4vis_build_vrml_driver='y'
+g4vis_build_vrmlfile_driver='y'
+g4vis_oglhome='/usr/X11R6/'
+g4vis_use_asciitree='y'
+g4vis_use_dawn='y'
+g4vis_use_dawnfile='y'
+g4vis_use_oiwin32='n'
+g4vis_use_oix='n'
+g4vis_use_openglwin32='n'
+g4vis_use_openglx='y'
+g4vis_use_openglxm='y'
+g4vis_use_raytracer='y'
+g4vis_use_vrml='y'
+g4vis_use_vrmlfile='y'
+g4w_use_g3tog4='@BUILDCERN@'
+g4wanalysis_build=''
+g4wanalysis_build_jas=''
+g4wanalysis_build_lab=''
+g4wanalysis_build_lizard=''
+g4wanalysis_use='n'
+g4wanalysis_use_jas=''
+g4wanalysis_use_lab=''
+g4wanalysis_use_lizard=''
+g4wlib_build_g3tog4='@BUILDCERN@'
--- geant4.8.orig/config/sys/Darwin-g++.gmk	2006-12-15 05:16:27.000000000 -0600
+++ geant4.8.new/config/sys/Darwin-g++.gmk	2006-12-22 10:26:22.000000000 -0600
@@ -27,27 +27,27 @@
     FCFLAGS   += -pg
     CCFLAGS   += -pg
   endif
-  FC := g77
+  FC := @FORTRAN@
   FCFLAGS += -fno-automatic -fno-backslash -fno-second-underscore
   LDFLAGS += -bind_at_load
-  FCLIBS := -lg2c -lnsl
+  FCLIBS := @FORTRAN_LIBS@
   SHEXT   := dylib
   ifndef X11FLAGS
-    X11FLAGS  := -I/usr/include/X11/extensions -I/usr/include/X11
+    X11FLAGS  := -I/usr/X11R6/include
   endif
   ifndef X11LIBS
     X11LIBS   := -L/usr/X11R6/lib  -lXmu -lXt -lXext -lX11 -lXi -lSM -lICE
   endif
   ifndef XMFLAGS
-    XMFLAGS   := -I/sw/include
+    XMFLAGS   := -I@PREFIX@/include
 #   XMFLAGS   := -I/usr/X11R6/include          ##### some installations.
   endif
   ifndef XMLIBS
-    XMLIBS    := -L/sw/lib -lXm -lXpm
+    XMLIBS    := -L@PREFIX@/lib -lXm -lXpm
 #   XMLIBS    := -L/usr/X11R6/lib -lXm -lXpm   ##### some installations.
   endif
   ifndef XAWFLAGS
-    XAWFLAGS   := -I/sw/include
+    XAWFLAGS   := -I@PREFIX@/include
 #   XAWFLAGS   := -I/usr/X11R6/include         ##### some installations.
   endif
   ifndef XAWLIBS
@@ -61,20 +61,37 @@
     OGLFLAGS  := -I$(OGLHOME)/include 
   endif
   ifndef OGLLIBS
-    OGLLIBS   := -L$(OGLHOME)/lib -lGLU -lGL
+    OGLLIBS   := -L$(OGLHOME)/lib -lGLU -lGL -Wl,-dylib_file,/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib:/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib
   endif
 
   define build-granular-shared-lib
     @libdir=`(cd $(@D);/bin/pwd)`; \
+     libname=`(echo $(@F) | /usr/bin/sed 's/\.dylib/.@VERSION@.dylib/')`; \
+     compversion=`(echo @VERSION@ | /usr/bin/cut -d'.' -f1-2)`; \
+     libs=`(/bin/ls -1 $$libdir/*$$compversion.dylib | /usr/bin/xargs -IFILE /usr/bin/basename FILE .$$compversion.dylib | /usr/bin/sed -n 's|^lib|-l|p')`; \
      cd $(G4TMPDIR); \
      g++ -dynamiclib -single_module -undefined dynamic_lookup \
-         -o $$libdir/$(@F) $(INTYLIBS) *.o
+         -install_name @PREFIX@/lib/geant4.8/Darwin-g++/`echo $$libname | /usr/bin/cut -d'.' -f1-3`".dylib" \
+         -compatibility_version $$compversion -current_version @VERSION@ \
+         -o $$libdir/$$libname *.o -L$(G4LIB)/$(G4SYSTEM) $$libs \
+         -L$(CLHEP_LIB_DIR) -l$(CLHEP_LIB) $(INTYLIBS); \
+     /bin/ln -s $$libname $$libdir/`echo $$libname | /usr/bin/cut -d'.' -f1-3`".dylib"; \
+     /bin/ln -s $$libname $$libdir/`echo $$libname | /usr/bin/cut -d'.' -f1`".dylib"
   endef
   define build-global-shared-lib
     @libdir=`(cd $(@D);/bin/pwd)`; \
+     libname=`(echo $(@F) | /usr/bin/sed 's/\.dylib/.@VERSION@.dylib/')`; \
+     compversion=`(echo @VERSION@ | /usr/bin/cut -d'.' -f1-2)`; \
+     libs=`(/bin/ls -1 $$libdir/*$$compversion.dylib | /usr/bin/xargs -IFILE /usr/bin/basename FILE .$$compversion.dylib | /usr/bin/sed -n 's|^lib|-l|p')`; \
      cd $(G4TMP)/$(G4SYSTEM); \
      g++ -dynamiclib -single_module -undefined dynamic_lookup \
-         -o $$libdir/$(@F) $(INTYLIBS) $(foreach dir,$(SUBLIBS),$(dir)/*.o);
+         -install_name @PREFIX@/lib/geant4.8/Darwin-g++/`echo $$libname | /usr/bin/cut -d'.' -f1-3`".dylib" \
+         -compatibility_version $$compversion -current_version @VERSION@ \
+         -o $$libdir/$$libname \
+         $(foreach dir,$(SUBLIBS),$(dir)/*.o) \
+         -L$(CLHEP_LIB_DIR) -l$(CLHEP_LIB) $(INTYLIBS); \
+     /bin/ln -s $$libname $$libdir/`echo $$libname | /usr/bin/cut -d'.' -f1-3`".dylib"; \
+     /bin/ln -s $$libname $$libdir/`echo $$libname | /usr/bin/cut -d'.' -f1`".dylib"
   endef
 
 endif
--- geant4.8.orig/Configure	2006-12-23 11:40:51.000000000 -0600
+++ geant4.8.new/Configure	2006-12-23 11:41:25.000000000 -0600
@@ -3291,8 +3291,8 @@
 Do you want to build 'static' (.a) libraries too?
 EOM`
 
-#dflt=$g4lib_build_static
-dflt=n
+dflt=$g4lib_build_static
+#dflt=n
 rp=$g4message
 . ./myread
 gans=$ans
--- geant4.8.orig/source/g3tog4/GNUmakefile	2006-12-15 05:18:26.000000000 -0600
+++ geant4.8.new/source/g3tog4/GNUmakefile	2007-03-01 16:55:03.000000000 -0600
@@ -104,7 +104,7 @@
 
 all: lib
 
-lib: libc libF
+lib: libc libF bin
 
 ifneq ($(G4LIB_BUILD_SHARED),)
 libc : $(G4LIBDIR)/lib$(name).$(SHEXT)
@@ -138,7 +138,7 @@
 	@echo Linking $(G4TARGET)
 	$(RM) $@
 	echo $(G4TARGET) dependencies: $^
-	$(FC) $(Fbinobjects) -o $@ $(G4LIBDIR)/lib$(name)F.a $(shell cernlib -v pro geant321 packlib)
+	$(FC) $(Fbinobjects) -o $@ $(G4LIBDIR)/lib$(name)F.a $(shell cernlib -v pro geant321 packlib) $(FCLIBS)
 	@echo Finished linking $(G4TARGET)
 
 ###-------------------- locally defined pattern rules ------------------------
