diff -Nurd gmsh-2.7.1-source/CMakeLists.txt gmsh-2.7.1-source.patched/CMakeLists.txt
--- gmsh-2.7.1-source/CMakeLists.txt	2013-05-12 05:34:32.000000000 -0700
+++ gmsh-2.7.1-source.patched/CMakeLists.txt	2013-05-12 13:54:03.000000000 -0700
@@ -1258,10 +1258,10 @@
 elseif(APPLE AND ENABLE_OS_SPECIFIC_INSTALL)
   # set these so that the files get installed nicely in the MacOSX
   # .app bundle
-  set(GMSH_BIN ../MacOS)
-  set(GMSH_LIB ../MacOS)
-  set(GMSH_DOC ../../..)
-  set(GMSH_MAN ../../..)
+  set(GMSH_BIN Gmsh.app/Contents/MacOS)
+  set(GMSH_LIB lib)
+  set(GMSH_DOC share/doc/gmsh)
+  set(GMSH_MAN share/man/man1)
 else(WIN32 OR CYGWIN)
   set(GMSH_BIN bin)
   set(GMSH_LIB lib)
@@ -1410,13 +1410,13 @@
   file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/Info.plist "${F1}")
   set(CPACK_BUNDLE_PLIST ${CMAKE_CURRENT_BINARY_DIR}/Info.plist)
   set(CPACK_BUNDLE_ICON ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIcons.icns)
-  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIconsGeo.icns DESTINATION .
+  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIconsGeo.icns DESTINATION Gmsh.app/Contents/Resources
           RENAME GmshGeo.icns)
-  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIconsMsh.icns DESTINATION .
+  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIconsMsh.icns DESTINATION Gmsh.app/Contents/Resources
           RENAME GmshMsh.icns)
-  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIconsSol.icns DESTINATION .
+  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIconsSol.icns DESTINATION Gmsh.app/Contents/Resources
           RENAME GmshSol.icns)
-  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIconsPos.icns DESTINATION .
+  install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIconsPos.icns DESTINATION Gmsh.app/Contents/Resources
           RENAME GmshPos.icns)
   set(CPACK_PACKAGE_ICON ${CMAKE_CURRENT_SOURCE_DIR}/Fltk/MacIcons.icns)
 elseif(WIN32 OR CYGWIN)
