diff -Nurd massxpert-2.3.5/gui/CMakeLists.txt massxpert-2.3.5-patched/gui/CMakeLists.txt
--- massxpert-2.3.5/gui/CMakeLists.txt	2010-04-29 08:21:07.000000000 -0400
+++ massxpert-2.3.5-patched/gui/CMakeLists.txt	2010-05-09 19:32:47.000000000 -0400
@@ -233,19 +233,19 @@
 EXEC_PROGRAM ("cp ${CMAKE_CURRENT_SOURCE_DIR}/images/massXpert.icns ${CMAKE_CURRENT_BINARY_DIR}/massXpert.app/Contents/Resources")
   
 # We want to build universal binaries 
-SET (CMAKE_OSX_ARCHITECTURES "ppc;i386" 
-  CACHE STRING "Build architectures for OSX" FORCE) 
+#SET (CMAKE_OSX_ARCHITECTURES "ppc;i386" 
+#  CACHE STRING "Build architectures for OSX" FORCE) 
 
 # That can be deployed on older system 
 # (Qt 4.4.0 is available starting at 10.3.9) 
-SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mmacosx-version-min=10.3" 
+SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mmacosx-version-min=10.4" 
   CACHE STRING "Flags used by the compiler during all build types." FORCE)
 
 SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility-inlines-hidden" 
   CACHE STRING "Flags used by the compiler during all build types." FORCE)
 
-SET (CMAKE_OSX_SYSROOT "/Developer/SDKs/MacOSX10.4u.sdk" 
-  CACHE STRING "MacOSX10.4u.sdk." FORCE)
+#SET (CMAKE_OSX_SYSROOT "/Developer/SDKs/MacOSX10.4u.sdk" 
+#  CACHE STRING "MacOSX10.4u.sdk." FORCE)
 
 SET_SOURCE_FILES_PROPERTIES (application.qrc ${TARGET}_fr.qm 
   PROPERTIES MACOSX_BUNDLE_LOCATION Resources)
