diff -Nurd -x'*~' pynifti-0.20090205.1.orig/setup.py pynifti-0.20090205.1/setup.py
--- pynifti-0.20090205.1.orig/setup.py	2009-02-05 15:10:32.000000000 -0500
+++ pynifti-0.20090205.1/setup.py	2012-09-07 17:39:36.000000000 -0400
@@ -21,7 +21,7 @@
 # Common configuration #
 ########################
 
-extra_link_args = ['--Wl,--no-undefined']
+extra_link_args = []
 include_dirs = []
 library_dirs = []
 defines = []
@@ -45,11 +45,7 @@
 else:
     # try to look for nifticlibs in some place
     if not sys.platform.startswith('win'):
-        include_dirs += ['/usr/include/nifti',
-                         '/usr/include/nifticlibs',
-                         '/usr/local/include/nifti',
-                         '/usr/local/include/nifticlibs',
-                         '/usr/local/include']
+        include_dirs += ['@PREFIX@/include/nifti']
     else:
         # no clue on windows
         pass
@@ -79,7 +75,7 @@
     define_macros = defines,
     include_dirs = include_dirs,
     library_dirs = library_dirs,
-    libraries = ['niftiio'] + link_libs,
+    libraries = ['niftiio', 'znz'] + link_libs,
     extra_link_args = extra_link_args,
     swig_opts = swig_opts)
 
