diff -uNr R-2.15.0/Makeconf.in R-2.15.0.fink/Makeconf.in
--- R-2.15.0/Makeconf.in	2012-03-01 18:02:30.000000000 -0500
+++ R-2.15.0.fink/Makeconf.in	2012-04-13 20:43:42.000000000 -0400
@@ -83,7 +83,7 @@
 READLINE_LIBS = @READLINE_LIBS@
 SED = @SED@
 SHELL = @R_SHELL@
-SHLIB_EXT = @SHLIB_EXT@
+SHLIB_EXT = .dylib
 SHLIB_CFLAGS = @SHLIB_CFLAGS@
 SHLIB_FFLAGS = @SHLIB_FFLAGS@
 SHLIB_LD = @SHLIB_LD@
diff -uNr R-2.15.0/Makefile.fw R-2.15.0.fink/Makefile.fw
--- R-2.15.0/Makefile.fw	2011-10-02 18:02:39.000000000 -0400
+++ R-2.15.0.fink/Makefile.fw	2012-04-13 20:43:42.000000000 -0400
@@ -8,7 +8,7 @@
 
 install install-strip: install-R-framework
 		@(sed 's|^LIBR =.*|LIBR = -F$(R_FRAMEWORK_DIR)/.. -framework R|' \
-		  $(top_builddir)/etc/Makeconf > "$(rhome)/etc${R_ARCH}/Makeconf")
+		  $(top_builddir)/etc/Makeconf > "$(DESTDIR)$(R_FRAMEWORK_DIR)/Resources/etc${R_ARCH}/Makeconf")
 		@(sed 's/Versions\/$(FW_VERSION)\/Resources/Resources/' \
 		  "$(DESTDIR)$(R_FRAMEWORK_DIR)/Resources/bin/R" > \
 		  "$(DESTDIR)$(R_FRAMEWORK_DIR)/Resources/bin/RR")
@@ -37,7 +37,7 @@
 	  ln -f -s -n Versions/Current/Resources Resources)
 	@## the resulting libR will point dyld to the fat libR regardless of its origin
 	@(cd "$(DESTDIR)$(R_FRAMEWORK_DIR)/Resources/lib" && \
-	  install_name_tool -id "$(R_FRAMEWORK_DIR)/Versions/$(FW_VERSION)/Resources/lib/libR.dylib" $(Rexeclibdir)/libR.dylib)
+	  install_name_tool -id "$(R_FRAMEWORK_DIR)/Versions/$(FW_VERSION)/Resources/lib/libR.dylib" $(DESTDIR)$(Rexeclibdir)/libR.dylib)
 	@( otool=otool ; if otool -D "$(DESTDIR)$(Rexeclibdir)/libR.dylib"|grep 'not an object'>/dev/null; then otool=otool64; fi; \
 	   for lib in Rlapack Rblas R; do \
 	   if test -e "$(DESTDIR)$(Rexeclibdir)/lib$${lib}.dylib"; then $(ECHO) "  lib$${lib}"; \
diff -uNr R-2.15.0/configure R-2.15.0.fink/configure
--- R-2.15.0/configure	2012-03-22 19:05:53.000000000 -0400
+++ R-2.15.0.fink/configure	2012-04-13 20:43:42.000000000 -0400
@@ -25888,7 +25888,7 @@
 main_ld="${CC}"
 shlib_ld="${CC}"
 shlib_cxxld="${CXX}"
-SHLIB_EXT=".so"
+SHLIB_EXT=".dylib"
 SHLIB_LIBADD=
 use_exportfiles=no
 
