diff -ru SNNSv4.3/configuration/Makefile.din SNNSv4.3-patched/configuration/Makefile.din
--- SNNSv4.3/configuration/Makefile.din	2008-06-25 09:37:04.000000000 +0200
+++ SNNSv4.3-patched/configuration/Makefile.din	2012-02-15 13:10:05.000000000 +0100
@@ -72,7 +72,7 @@
 XGUIBINDIR         = $(INSTALLDIR)$(XGUIBINEXT)/bin$(HOSTBINEXT)
 XGUISRCDIR         = xgui/sources
 XGUIICONSDIR       = xgui/iconsXgui
-MANDIR             = $(INSTALLDIR)/man
+MANDIR             = $(INSTALLDIR)/share/man
 MAN1DIR            = $(MANDIR)/man1
 MANSRCDIR          = man
 
diff -ru SNNSv4.3/kernel/sources/kr_mem.c SNNSv4.3-patched/kernel/sources/kr_mem.c
--- SNNSv4.3/kernel/sources/kr_mem.c	2008-04-21 12:15:23.000000000 +0200
+++ SNNSv4.3-patched/kernel/sources/kr_mem.c	2012-02-15 13:10:05.000000000 +0100
@@ -47,7 +47,9 @@
 #include "strdup.h"	/*  include strdup function because strdup is
 			    missing in ULTRIX-32  */
 #endif
+#include <limits.h>
 
+#define MAXSHORT 32767
 
 /*****************************************************************************
   FUNCTION : krm_allocLinks
diff -ru SNNSv4.3/kernel/sources/learn_f.c SNNSv4.3-patched/kernel/sources/learn_f.c
--- SNNSv4.3/kernel/sources/learn_f.c	2008-04-21 12:15:23.000000000 +0200
+++ SNNSv4.3-patched/kernel/sources/learn_f.c	2012-02-15 13:10:05.000000000 +0100
@@ -39,6 +39,8 @@
 #include <limits.h>
 #define MAXINT INT_MAX
 
+#define MAXINT 2147483647
+
 #include "kr_typ.h"		/* Kernel Types and Constants  */
 #include "kr_const.h"		/* Constant Declarators for SNNS-Kernel  */
 #include "kr_def.h"		/* Default Values  */
diff -ru SNNSv4.3/xgui/sources/Makefile SNNSv4.3-patched/xgui/sources/Makefile
--- SNNSv4.3/xgui/sources/Makefile	2008-04-21 09:56:15.000000000 +0200
+++ SNNSv4.3-patched/xgui/sources/Makefile	2012-02-15 13:20:51.000000000 +0100
@@ -203,7 +203,7 @@
 
 xgui: $(LIBXGUI1) $(LIBXGUI2) $(BN3DLIB) $(KERNELLIBS)
 	$(CC) $(LIBXGUI1) $(LIBXGUI2) $(BN3DLIB) \
-	$(X_LIBS) -lXaw3d -lXmu -lXt -lXext $(X_PRE_LIBS) -lX11 $(X_EXTRA_LIBS) \
+	$(X_LIBS) $(LDFLAGS) -lXaw3d -lXmu -lXt -lXext $(X_PRE_LIBS) -lX11 $(X_EXTRA_LIBS) \
 	$(KERNELALLLIBS) -o xgui 
 
 # Dependencies
diff -ru SNNSv4.3/xgui/sources/ui_mainP.c SNNSv4.3-patched/xgui/sources/ui_mainP.c
--- SNNSv4.3/xgui/sources/ui_mainP.c	2008-04-21 09:56:25.000000000 +0200
+++ SNNSv4.3-patched/xgui/sources/ui_mainP.c	2012-03-07 11:53:50.000000000 +0100
@@ -1458,7 +1458,7 @@
 
     struct HelpType *listPtr;
 
-    sprintf(hdoc, "help.hdoc");
+    sprintf(hdoc, "@PREFIX@/share/doc/snns/help.hdoc");
     if (NOT ui_fileExist(hdoc,0)) {
 	if (getenv("XGUILOADPATH") != NULL) 
 	    sprintf(hdoc, "%s/help.hdoc", getenv("XGUILOADPATH"));
