diff -Nurd SuiteSparse/KLU/Include/klu_internal.h SuiteSparse.patched/KLU/Include/klu_internal.h
--- SuiteSparse/KLU/Include/klu_internal.h	2012-01-20 08:29:18.000000000 -0700
+++ SuiteSparse.patched/KLU/Include/klu_internal.h	2012-06-22 09:00:00.000000000 -0700
@@ -9,7 +9,6 @@
 
 #include "klu.h" 
 #include "btf.h"
-#include "klu_version.h"
 
 /* ========================================================================== */
 /* make sure debugging and printing is turned off */
@@ -36,6 +35,8 @@
 #include <limits.h>
 #include <stdlib.h>
 #include <math.h>
+/* moved klu_version.h after math.h to circumvent compile error on ppc */
+#include "klu_version.h"
 
 #undef ASSERT
 #ifndef NDEBUG
--- SuiteSparse/SuiteSparse_config/SuiteSparse_config.mk.orig	2013-03-31 12:47:50.000000000 -0400
+++ SuiteSparse/SuiteSparse_config/SuiteSparse_config.mk	2013-03-31 12:57:59.000000000 -0400
@@ -77,7 +77,7 @@
 F77LIB =
 
 # C and Fortran libraries.  Remove -lrt if you don't have it.
-  LIB = -lm -lrt
+#  LIB = -lm -lrt
 # Using the following requires CF = ... -DNTIMER on POSIX C systems.
 # LIB = -lm
 
@@ -106,8 +106,8 @@
 # naming the BLAS and LAPACK library (*.a or *.so) files.
 
 # This is probably slow ... it might connect to the Standard Reference BLAS:
-BLAS = -lblas -lgfortran
-LAPACK = -llapack
+# BLAS = -lblas -lgfortran
+# LAPACK = -llapack
 
 # NOTE: this next option for the "Goto BLAS" has nothing to do with a "goto"
 # statement.  Rather, the Goto BLAS is written by Dr. Kazushige Goto.
@@ -162,8 +162,8 @@
 # The path is relative to where it is used, in CHOLMOD/Lib, CHOLMOD/MATLAB, etc.
 # You may wish to use an absolute path.  METIS is optional.  Compile
 # CHOLMOD with -DNPARTITION if you do not wish to use METIS.
-METIS_PATH = ../../metis-4.0
-METIS = ../../metis-4.0/libmetis.a
+METIS_PATH =
+METIS =
 
 #------------------------------------------------------------------------------
 # UMFPACK configuration:
@@ -222,10 +222,10 @@
 # -DNSUNPERF	    for Solaris only.  If defined, do not use the Sun
 #			Performance Library
 
-CHOLMOD_CONFIG = $(GPU_CONFIG)
+# CHOLMOD_CONFIG = $(GPU_CONFIG)
 
 # uncomment this line to compile CHOLMOD without METIS:
-# CHOLMOD_CONFIG = -DNPARTITION
+CHOLMOD_CONFIG = -DNPARTITION
 
 #------------------------------------------------------------------------------
 # SuiteSparseQR configuration:
@@ -238,9 +238,10 @@
 # -DHAVE_TBB        enable the use of Intel's Threading Building Blocks (TBB)
 
 # default, without timing, without TBB:
-SPQR_CONFIG =
+# SPQR_CONFIG =
 # with TBB:
 # SPQR_CONFIG = -DHAVE_TBB
+SPQR_CONFIG = -DNPARTITION
 
 # This is needed for IBM AIX: (but not for and C codes, just C++)
 # SPQR_CONFIG = -DBLAS_NO_UNDERSCORE
@@ -314,6 +315,11 @@
 # BLAS = -framework Accelerate
 # LAPACK = -framework Accelerate
 # LIB = -lm
+CC = gcc
+CFLAGS = -O3 -fno-common -no-cpp-precomp -fexceptions
+LIB = -lstdc++
+BLAS = -framework Accelerate
+LAPACK = -framework Accelerate
 
 #------------------------------------------------------------------------------
 # Solaris
