diff -Naur Xfoil_orig/bin/Makefile Xfoil/bin/Makefile
--- Xfoil_orig/bin/Makefile	2008-04-07 17:05:29.000000000 -0400
+++ Xfoil/bin/Makefile	2008-04-09 18:50:59.000000000 -0400
@@ -44,7 +44,7 @@
 ###================================================
 ###  Default compilers and flags
 ###  FFLOPT used for xsolve.f
-FC = f77
+FC = gfortran
 FFLAGS  = -O 
 FFLOPT  = -O
 INSTALLCMD = install -s
@@ -99,7 +99,7 @@
 #FFLAGS = -g -O0 -C
 ##--------------------------
 ### Intel Fortran Compiler
-FC = ifort
+#FC = ifort
 FFLAGS = -O
 FFLOPT = -O
 #FTNLIB = -Vaxlib /usr/lib/C-ctype.o /usr/lib/C_name.o /usr/lib/ctype-info.o
@@ -112,8 +112,8 @@
 FFLOPT = -O -r8 -CB
 PLTOBJ = ../plotlib/libPltDP.a 
 
-FFLAGS = -O -r8 -ftrapuv -fpe0
-FFLOPT = -O -r8 -ftrapuv -fpe0
+FFLAGS = -O0 -fdefault-real-8 -fdefault-integer-8
+FFLOPT = -O0 -fdefault-real-8 -fdefault-integer-8
 
 
 all:	 $(PROGS)
diff -Naur Xfoil_orig/orrs/bin/Makefile_DP Xfoil/orrs/bin/Makefile_DP
--- Xfoil_orig/orrs/bin/Makefile_DP	2007-04-05 17:29:35.000000000 -0400
+++ Xfoil/orrs/bin/Makefile_DP	2008-04-09 18:53:03.000000000 -0400
@@ -1,20 +1,20 @@
 
 SRC = ../src
-PLTOBJ = /var/local/codes/Xplot/libPltDP.a
+PLTOBJ = ../../plotlib/libPltDP.a
 
 #========================================
 #  Default compiler flags
-FC = f77
-FLG = -O -r8
+FC = gfortran
+FLG = -O0 -fdefault-real-8 -fdefault-integer-8
 PLTLIB = -lX11
 FTNLIB =
 
 #========================================
 #  Uncomment for Intel Fortran Compiler
-FC = ifort
-FLG = -O -r8
-PLTLIB = -L/usr/X11R6/lib -lX11
-FTNLIB =
+#FC = ifort
+#FLG = -O -r8
+#PLTLIB = -L/usr/X11R6/lib -lX11
+#FTNLIB =
 
 #========================================
 
diff -Naur Xfoil_orig/orrs/src/osmap.f Xfoil/orrs/src/osmap.f
--- Xfoil_orig/orrs/src/osmap.f	2006-10-10 12:42:45.000000000 -0400
+++ Xfoil/orrs/src/osmap.f	2008-04-09 18:56:01.000000000 -0400
@@ -98,7 +98,7 @@
 C
 C---- set OSFILE to match the absolute OS database filename
       CHARACTER*128 OSFILE
-      DATA OSFILE / '/var/local/codes/orrs/osmapDP.dat' /
+      DATA OSFILE / '@BUILDPREFIX@/orrs/osmap.dat' /
 ccc      DATA OSFILE / '/var/local/codes/orrs/osmap.dat' /
 c
 c      DATA OSFILE
diff -Naur Xfoil_orig/orrs/src/osmap_DP.f Xfoil/orrs/src/osmap_DP.f
--- Xfoil_orig/orrs/src/osmap_DP.f	2003-08-17 22:22:54.000000000 -0400
+++ Xfoil/orrs/src/osmap_DP.f	2008-04-09 18:55:32.000000000 -0400
@@ -94,7 +94,7 @@
 C
 C---- set OSFILE to match the absolute location of the OS database file
       CHARACTER*48 OSFILE
-      DATA OSFILE / '/var/local/codes/orrs/osmap_DP.dat' /
+      DATA OSFILE / '@BUILDPREFIX@/orrs/osmap.dat' /
 C
       DATA LOADED / .FALSE. /
 C
diff -Naur Xfoil_orig/plotlib/Makefile Xfoil/plotlib/Makefile
--- Xfoil_orig/plotlib/Makefile	2004-06-02 12:23:43.000000000 -0400
+++ Xfoil/plotlib/Makefile	2008-04-09 18:58:44.000000000 -0400
@@ -35,8 +35,8 @@
 #INSTALLDIR= .
 
 # Use these to set default library name (overridden in config.make file) 
-PLTLIB = libPlt.a
-#PLTLIB = libPltDP.a
+#PLTLIB = libPlt.a
+PLTLIB = libPltDP.a
 
 
 ###========================================================
@@ -69,14 +69,14 @@
 
 ###================================================
 ###  Default compilers and flags, install commands
-FC = f77
-CC  = cc
+FC = gfortran
+CC  = gcc
 # Some fortrans need trailing underscores in C interface symbols (see Xwin.c)
 DEFINE = -DUNDERSCORE
 # Uncomment DP to make double-precision version
-#DP = -r8
-FFLAGS  = -O $(DP)
-CFLAGS  = -O $(DEFINE)
+DP = -fdefault-real-8 -fdefault-integer-8
+FFLAGS  = -O0 $(DP)
+CFLAGS  = -O0 -I/usr/X11R6/include/ $(DEFINE)
 AR = ar r 
 RANLIB = ranlib
 LINKLIB =  -lX11 
diff -Naur Xfoil_orig/plotlib/config.make.DP Xfoil/plotlib/config.make.DP
--- Xfoil_orig/plotlib/config.make.DP	2004-11-27 01:13:21.000000000 -0500
+++ Xfoil/plotlib/config.make.DP	2008-04-09 19:02:36.000000000 -0400
@@ -24,9 +24,9 @@
 ###          or the yaf77 or the old f77 shell script from f2c
 ###   g77    the GNU Fortran compiler 
 #
-#FC = g77-3
+FC = gfortran
 #FC = fort77
-#CC  = gcc
+CC  = gcc
 # Uncomment DP to make double-precision version
 # (note -r8 does not work in g77, use f2c instead)
 #DP = -r8
@@ -51,13 +51,13 @@
 
 ###-------------------------------------------------------------------------
 ###  Uncomment for Linux, using Intel Fortran compiler
-FC = ifort
+#FC = ifort
 CC  = gcc
  
 # Uncomment to make double-precision version
-DP = -r8
-FFLAGS  = -O3 $(DP)
-CFLAGS  = -O3 $(DEFINE)
+DP = -fdefault-real-8
+FFLAGS  = -O0 $(DP)
+CFLAGS  = -O0 $(DEFINE)
 AR = ar r
 RANLIB = ranlib 
 
@@ -127,3 +127,14 @@
 #RANLIB = ar qs
 #LINKLIB =  -lX11
 
+###-------------------------------------------------------------------------
+###  OS X via Fink packaging system
+#
+FC = gfortran
+CC = gcc
+FFLAGS  = -O0 -fdefault-real-8 -fdefault-integer-8
+CFLAGS  = -O0 -I/usr/X11R6/include/ $(DEFINE)
+AR = ar r
+RANLIB = ranlib 
+
+LINKLIB = -L/usr/X11R6/lib -lX11
diff -Naur Xfoil_orig/src/pplot.f Xfoil/src/pplot.f
--- Xfoil_orig/src/pplot.f	2008-04-07 17:05:29.000000000 -0400
+++ Xfoil/srcpplot.f	2008-04-09 18:50:59.000000000 -0400
@@ -36,7 +36,7 @@
       PROGRAM PPLOT
       INCLUDE 'PPLOT.INC'
 C
-      LOGICAL ERROR, LGETFN
+      LOGICAL ERROR, LGETFN, LERR
       REAL RINP(10)
       REAL CPOLO(NAX,IPTOT,NPX), VPOLO(NAX,2,NPX)
 C
