diff -Nurd -x'*~' xmakemol-5.16.orig/configure xmakemol-5.16/configure
--- xmakemol-5.16.orig/configure	2007-08-30 13:45:18.000000000 -0400
+++ xmakemol-5.16/configure	2007-10-15 00:18:43.000000000 -0400
@@ -6209,9 +6209,9 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
 
-	XWIDGETLIBS="$X_LIBS $X11_LIBS $XT_LIBS $X_LIBS $X_PRE_LIBS $XM_LIBS"
+	XWIDGETLIBS="$X_LIBS $X11_LIBS $XM_LIBS $XT_LIBS $X_LIBS $X_PRE_LIBS"
   	sim_cv_motif_glwidget=UNKNOWN
-  	for lib in GLwM GLw MesaGLwM MesaGLw; do
+  	for lib in GLw; do
     	  if test x"$sim_cv_motif_glwidget" = x"UNKNOWN"; then
       		LIBS="-l$lib $SAVELIBS $XWIDGETLIBS $GL_LIBS -lm"
       		cat >conftest.$ac_ext <<_ACEOF
diff -Nurd -x'*~' xmakemol-5.15.orig/file.c xmakemol-5.15/file.c
--- xmakemol-5.15.orig/file.c	2005-03-06 12:04:32.000000000 -0500
+++ xmakemol-5.15/file.c	2005-12-16 01:38:10.000000000 -0500
@@ -29,6 +29,7 @@
 #include <alloca.h>
 #endif
 #include <ctype.h> /* aro - needed for isalnum() */
+#include <sys/types.h>
 #include <dirent.h>
 #include <locale.h>
 #include <math.h>
@@ -96,6 +97,15 @@
 /* directory that was accessed in any of the file selection dialogs */
 static XmString current_directory = NULL;
 
+  void export_type_cb(Widget, XtPointer, XtPointer);
+  void file_type_cb(Widget, XtPointer, XtPointer);
+  void load_file_cb(Widget, XtPointer, XtPointer);
+  void save_file_cb(Widget, XtPointer, XtPointer);
+  void merge_file_cb(Widget, XtPointer, XtPointer);
+  void export_file_cb(Widget, XtPointer, XtPointer);
+  void print_file_cb(Widget, XtPointer, XtPointer);/* aro */
+  void print_file_select_cb(Widget, XtPointer, XtPointer);/* aro */
+  void cleanup_cb(Widget, XtPointer, XtPointer);/* aro */
 
 void
 file_cb(Widget widget, XtPointer client_data, XtPointer call_data)
@@ -112,15 +122,6 @@
      Widgets p_dialog, print_printer, temp_widget; 
      XmStringTable printer_xmstrings. */
      
-  void export_type_cb(Widget, XtPointer, XtPointer);
-  void file_type_cb(Widget, XtPointer, XtPointer);
-  void load_file_cb(Widget, XtPointer, XtPointer);
-  void save_file_cb(Widget, XtPointer, XtPointer);
-  void merge_file_cb(Widget, XtPointer, XtPointer);
-  void export_file_cb(Widget, XtPointer, XtPointer);
-  void print_file_cb(Widget, XtPointer, XtPointer);/* aro */
-  void print_file_select_cb(Widget, XtPointer, XtPointer);/* aro */
-  void cleanup_cb(Widget, XtPointer, XtPointer);/* aro */
   int item_no=(int) client_data, i, j;/* aro */
   char *printer_name, *line_ptr, line[1024], printer_strings[MAX_PRINTERS+1][80],
        print_command_string[1024];/* aro */
diff -Nurd -x'*~' xmakemol-5.15.orig/xmake_anim.pl xmakemol-5.15/xmake_anim.pl
--- xmakemol-5.15.orig/xmake_anim.pl	2005-03-07 05:10:41.000000000 -0500
+++ xmakemol-5.15/xmake_anim.pl	2005-12-16 01:38:39.000000000 -0500
@@ -27,13 +27,13 @@
 
 # Configure: specify where the binaries are
 
-$convert = "/usr/bin/convert";
+$convert = "@FINKPREFIX@/bin/convert";
 if (! -e $convert)
   {
     die ("Cannot find convert program (part of ImageMagick). Please install it.\n");
   }
 
-$gifsicle = "/usr/bin/gifsicle";
+$gifsicle = "@FINKPREFIX@/bin/gifsicle";
 if (! -e $gifsicle)
   {
     print ("Cannot find gifsicle program, install it for better performance.\n");
diff -Nurd -x'*~' xmakemol-5.15.orig/xmakemol.1 xmakemol-5.15/xmakemol.1
--- xmakemol-5.15.orig/xmakemol.1	2005-01-28 09:34:54.000000000 -0500
+++ xmakemol-5.15/xmakemol.1	2005-12-16 01:39:03.000000000 -0500
@@ -47,7 +47,7 @@
 
 .SH FILES
 .LP 
-\fI/usr/share/xmakemol/elements\fP
+\fI@FINKPREFIX@/share/xmakemol/elements\fP
 List of element properties.
 
 .SH ENVIRONMENT VARIABLES
