diff -ru align-1.7.2/align align-1.7.2-patched/align
--- align-1.7.2/align	2012-04-18 21:02:27.000000000 +0200
+++ align-1.7.2-patched/align	2012-04-26 12:10:06.000000000 +0200
@@ -5,11 +5,13 @@
 # see website http://www.cs.indiana.edu/~kinzler/align/
 # http://www.cs.indiana.edu/~kinzler/home.html#unix
 
-$version = '1.7.2';
+$VERSION = '1.7.2';
 
 require 5.000;
 use Getopt::Std;
 
+$Getopt::Std::STANDARD_HELP_VERSION = 1;
+
 $usage = <<EOF;
 usage: $0 [ -s { s+ | s | t+ | t | _+ | _ | /patt } ]
        [ -j { s | t | _ | /string } ] [ -g # ] [ -t # ]
@@ -60,7 +62,7 @@
 Visual alignment may not be achieved if the column text or join string
 contains tabs, non-printing characters or other characters not occupying
 one character place.
-Version $version
+Version $VERSION
 EOF
 
 die $usage if ! getopts('hs:j:g:t:a:e:E:') || $opt_h;
