diff -ru coreutils-8.19/lib/localcharset.c coreutils-8.19-patched/lib/localcharset.c
--- coreutils-8.19/lib/localcharset.c	2012-09-05 23:21:21.000000000 -0600
+++ coreutils-8.19-patched/lib/localcharset.c	2012-09-05 23:21:04.000000000 -0600
@@ -278,6 +278,7 @@
            "SJIS" "\0" "SHIFT_JIS" "\0"
            "ARMSCII-8" "\0" "ARMSCII-8" "\0"
            "PT154" "\0" "PT154" "\0"
+           "US-ASCII" "\0" "ASCII" "\0"
          /*"ISCII-DEV" "\0" "?" "\0"*/
            "*" "\0" "UTF-8" "\0";
 # endif
