diff -Nurd dc3dd-7.3.1.orig/configure dc3dd-7.3.1/configure
--- dc3dd-7.3.1.orig/configure	2023-04-24 19:15:24
+++ dc3dd-7.3.1/configure	2025-12-19 10:48:34
@@ -54147,6 +54147,7 @@
 #endif
 #include <stdlib.h>
 #include <string.h>
+#include <wctype.h>
 struct tm t;
 char buf[16];
 int main () {
@@ -57902,7 +57903,7 @@
 
 
 cat >>confdefs.h <<\_ACEOF
-#define ARGMATCH_DIE_DECL void usage ()
+#define ARGMATCH_DIE_DECL void usage (int)
 _ACEOF
 
 
diff -Nurd dc3dd-7.3.1.orig/lib/c-strtod.c dc3dd-7.3.1/lib/c-strtod.c
--- dc3dd-7.3.1.orig/lib/c-strtod.c	2012-11-06 08:42:58
+++ dc3dd-7.3.1/lib/c-strtod.c	2025-12-19 10:53:22
@@ -23,6 +23,7 @@
 
 #include <locale.h>
 #include <stdlib.h>
+#include <xlocale.h>
 
 #include "xalloc.h"
 
diff -Nurd dc3dd-7.3.1.orig/lib/posixtm.c dc3dd-7.3.1/lib/posixtm.c
--- dc3dd-7.3.1.orig/lib/posixtm.c	2012-11-06 08:42:58
+++ dc3dd-7.3.1/lib/posixtm.c	2025-12-19 10:55:52
@@ -41,7 +41,7 @@
    of `digit' even when the host does not conform to POSIX.  */
 #define ISDIGIT(c) ((unsigned int) (c) - '0' <= 9)
 
-time_t mktime ();
+time_t mktime (struct tm *tm);
 
 /*
   POSIX requires:
