diff -ruN duplicity-0.7.14-orig/bin/duplicity duplicity-0.7.14/bin/duplicity
--- duplicity-0.7.14-orig/bin/duplicity	2017-09-04 16:48:41.000000000 -0400
+++ duplicity-0.7.14/bin/duplicity	2017-09-04 16:53:08.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!@PREFIX@/bin/python2.7
 # -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
 #
 # duplicity -- Encrypted bandwidth efficient backup
diff -ruN duplicity-0.7.14-orig/bin/rdiffdir duplicity-0.7.14/bin/rdiffdir
--- duplicity-0.7.14-orig/bin/rdiffdir	2017-09-04 16:48:41.000000000 -0400
+++ duplicity-0.7.14/bin/rdiffdir	2017-09-04 16:53:08.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!@PREFIX@/bin/python2.7
 # rdiffdir -- Extend rdiff functionality to directories
 # Version 0.7.14 released August 31, 2017
 #
diff -ruN duplicity-0.7.14-orig/dist/makedist duplicity-0.7.14/dist/makedist
--- duplicity-0.7.14-orig/dist/makedist	2017-09-04 16:48:41.000000000 -0400
+++ duplicity-0.7.14/dist/makedist	2017-09-04 16:53:08.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!@PREFIX@/bin/python2.7
 # -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
 #
 # Copyright 2002 Ben Escoto <ben@emerose.org>
diff -ruN duplicity-0.7.14-orig/duplicity/backends/pyrax_identity/hubic.py duplicity-0.7.14/duplicity/backends/pyrax_identity/hubic.py
--- duplicity-0.7.14-orig/duplicity/backends/pyrax_identity/hubic.py	2017-09-04 16:48:41.000000000 -0400
+++ duplicity-0.7.14/duplicity/backends/pyrax_identity/hubic.py	2017-09-04 16:53:08.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!@PREFIX@/bin/python2.7
 # -*- coding: utf-8 -*-
 # Copyright (c) 2014 Gu1
 # Licensed under the MIT license
diff -ruN duplicity-0.7.14-orig/duplicity/compilec.py duplicity-0.7.14/duplicity/compilec.py
--- duplicity-0.7.14-orig/duplicity/compilec.py	2017-09-04 16:48:41.000000000 -0400
+++ duplicity-0.7.14/duplicity/compilec.py	2017-09-04 16:53:08.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!@PREFIX@/bin/python2.7
 # -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
 #
 # Copyright 2002 Ben Escoto <ben@emerose.org>
diff -ruN duplicity-0.7.14-orig/setup.py duplicity-0.7.14/setup.py
--- duplicity-0.7.14-orig/setup.py	2017-09-04 16:48:41.000000000 -0400
+++ duplicity-0.7.14/setup.py	2017-09-04 16:53:08.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!@PREFIX@/bin/python2.7
 # -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
 #
 # Copyright 2002 Ben Escoto <ben@emerose.org>
diff -ruN duplicity-0.7.14-orig/testing/manual/backendtest duplicity-0.7.14/testing/manual/backendtest
--- duplicity-0.7.14-orig/testing/manual/backendtest	2017-09-04 16:48:41.000000000 -0400
+++ duplicity-0.7.14/testing/manual/backendtest	2017-09-04 16:53:08.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!@PREFIX@/bin/python2.7
 # -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
 #
 # Copyright 2002 Ben Escoto <ben@emerose.org>
diff -ruN duplicity-0.7.14-orig/testing/overrides/bin/hsi duplicity-0.7.14/testing/overrides/bin/hsi
--- duplicity-0.7.14-orig/testing/overrides/bin/hsi	2017-09-04 16:48:41.000000000 -0400
+++ duplicity-0.7.14/testing/overrides/bin/hsi	2017-09-04 16:53:08.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!@PREFIX@/bin/python2.7
 
 import subprocess
 import sys
diff -ruN duplicity-0.7.14-orig/testing/overrides/bin/lftp duplicity-0.7.14/testing/overrides/bin/lftp
--- duplicity-0.7.14-orig/testing/overrides/bin/lftp	2017-09-04 16:48:41.000000000 -0400
+++ duplicity-0.7.14/testing/overrides/bin/lftp	2017-09-04 16:53:08.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!@PREFIX@/bin/python2.7
 
 import os
 import subprocess
diff -ruN duplicity-0.7.14-orig/testing/overrides/bin/ncftpget duplicity-0.7.14/testing/overrides/bin/ncftpget
--- duplicity-0.7.14-orig/testing/overrides/bin/ncftpget	2017-09-04 16:48:41.000000000 -0400
+++ duplicity-0.7.14/testing/overrides/bin/ncftpget	2017-09-04 16:53:08.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!@PREFIX@/bin/python2.7
 
 import subprocess
 import sys
diff -ruN duplicity-0.7.14-orig/testing/overrides/bin/ncftpls duplicity-0.7.14/testing/overrides/bin/ncftpls
--- duplicity-0.7.14-orig/testing/overrides/bin/ncftpls	2017-09-04 16:48:41.000000000 -0400
+++ duplicity-0.7.14/testing/overrides/bin/ncftpls	2017-09-04 16:53:08.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!@PREFIX@/bin/python2.7
 
 from duplicity.backend import ParsedUrl
 import os
diff -ruN duplicity-0.7.14-orig/testing/overrides/bin/ncftpput duplicity-0.7.14/testing/overrides/bin/ncftpput
--- duplicity-0.7.14-orig/testing/overrides/bin/ncftpput	2017-09-04 16:48:41.000000000 -0400
+++ duplicity-0.7.14/testing/overrides/bin/ncftpput	2017-09-04 16:53:08.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!@PREFIX@/bin/python2.7
 
 import subprocess
 import sys
diff -ruN duplicity-0.7.14-orig/testing/overrides/bin/tahoe duplicity-0.7.14/testing/overrides/bin/tahoe
--- duplicity-0.7.14-orig/testing/overrides/bin/tahoe	2017-09-04 16:48:41.000000000 -0400
+++ duplicity-0.7.14/testing/overrides/bin/tahoe	2017-09-04 16:53:08.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!@PREFIX@/bin/python2.7
 
 import subprocess
 import sys
diff -ruN duplicity-0.7.14-orig/testing/run-tests duplicity-0.7.14/testing/run-tests
--- duplicity-0.7.14-orig/testing/run-tests	2017-09-04 16:48:41.000000000 -0400
+++ duplicity-0.7.14/testing/run-tests	2017-09-04 16:53:08.000000000 -0400
@@ -20,4 +20,4 @@
 # along with duplicity; if not, write to the Free Software Foundation,
 # Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 
-tox
+tox-py27
