diff -ruN e2fsprogs-1.42.13.orig/configure e2fsprogs-1.42.13/configure
--- e2fsprogs-1.42.13.orig/configure	2014-11-30 16:34:31.000000000 -0500
+++ e2fsprogs-1.42.13/configure	2018-07-15 09:39:00.000000000 -0400
@@ -5296,8 +5296,15 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_uuid_uuid_generate" >&5
 $as_echo "$ac_cv_lib_uuid_uuid_generate" >&6; }
 if test "x$ac_cv_lib_uuid_uuid_generate" = xyes; then :
-  LIBUUID=`$PKG_CONFIG --libs uuid`;
-		 STATIC_LIBUUID=`$PKG_CONFIG --static --libs uuid`
+# This was a convoluted mess, hardcoding just "-luuid" as neccesary
+# and sufficient, but then if so using flags from uuid.pc instead (and
+# only --libs not --cflags). It's not necessary (functions are in
+# libSystem) and passing it explicitly picks fink's but sources
+# hardcode a specific subdir header path that only picks OS X's.
+# Just..no. We also pass ac_cv_lib_uuid_uuid_generate=yes to
+# ./configure to avoid doing the -luuid test.
+#  LIBUUID=`$PKG_CONFIG --libs uuid`;
+#		 STATIC_LIBUUID=`$PKG_CONFIG --static --libs uuid`
 else
   as_fn_error $? "external uuid library not found" "$LINENO" 5
 fi
@@ -13410,7 +13417,7 @@
 
 
 if test $cross_compiling = no; then
-   BUILD_CFLAGS="$CFLAGS $CPPFLAGS $INCLUDES -DHAVE_CONFIG_H"
+   BUILD_CFLAGS="$CFLAGS $INCLUDES $CPPFLAGS -DHAVE_CONFIG_H"
    BUILD_LDFLAGS="$LDFLAGS"
 fi
 
diff -ruN e2fsprogs-1.42.13.orig/debugfs/Makefile.in e2fsprogs-1.42.13/debugfs/Makefile.in
--- e2fsprogs-1.42.13.orig/debugfs/Makefile.in	2014-07-06 00:13:18.000000000 -0400
+++ e2fsprogs-1.42.13/debugfs/Makefile.in	2018-07-15 07:12:17.000000000 -0400
@@ -81,7 +81,7 @@
 
 e2freefrag.o: $(srcdir)/../misc/e2freefrag.c
 	$(E) "	CC $@"
-	$(Q) $(CC) -c $(ALL_CFLAGS) -I$(srcdir) $< -DDEBUGFS -o $@
+	$(Q) $(CC) -c -I$(srcdir) $(ALL_CFLAGS) $< -DDEBUGFS -o $@
 
 debugfs.8: $(DEP_SUBSTITUTE) $(srcdir)/debugfs.8.in
 	$(E) "	SUBST $@"
diff -ruN e2fsprogs-1.42.13.orig/lib/Makefile.darwin-lib e2fsprogs-1.42.13/lib/Makefile.darwin-lib
--- e2fsprogs-1.42.13.orig/lib/Makefile.darwin-lib	2011-01-14 13:07:50.000000000 -0500
+++ e2fsprogs-1.42.13/lib/Makefile.darwin-lib	2018-07-15 06:52:35.000000000 -0400
@@ -16,25 +16,31 @@
 	$(E) "	MKDIR pic"
 	$(Q) mkdir -p pic
 
-BSD_LIB = $(BSDLIB_IMAGE).$(BSDLIB_VERSION).dylib
+BSD_LIB = $(BSDLIB_IMAGE).$(ELF_VERSION).dylib
+BSD_SO_LIB = $(BSDLIB_IMAGE).$(ELF_SO_VERSION).dylib
 BSDLIB_PIC_FLAG = -fPIC
 
 image:		$(BSD_LIB)
 
 $(BSD_LIB): $(OBJS)
 	$(E) "	GEN_BSD_SOLIB $(BSD_LIB)"
-	$(Q) (cd pic; $(CC) -dynamiclib -compatibility_version 1.0 -current_version $(BSDLIB_VERSION) \
-		-flat_namespace -undefined warning -o $(BSD_LIB) $(OBJS))
+	$(Q) (cd pic; $(CC) -dynamiclib -install_name $(BSDLIB_INSTALL_DIR)/$(BSD_SO_LIB) -compatibility_version 1.0 -current_version $(BSDLIB_VERSION) \
+		-undefined dynamic_lookup -o $(BSD_LIB) $(OBJS))
 	$(Q) $(MV) pic/$(BSD_LIB) .
 	$(Q) $(RM) -f ../$(BSD_LIB)
 	$(Q) (cd ..; $(LN) $(LINK_BUILD_FLAGS) \
 		`echo $(my_dir) | sed -e 's;lib/;;'`/$(BSD_LIB) $(BSD_LIB))
+	$(Q) (cd ..; $(LN) $(LINK_BUILD_FLAGS) $(BSD_LIB) $(BSD_SO_LIB))
 	$(Q) (cd ..; $(LN) $(LINK_BUILD_FLAGS) $(BSD_LIB) $(BSDLIB_IMAGE).dylib)
 
 install-shlibs install:: $(BSD_LIB)
 	$(E) "	INSTALL_PROGRAM $(BSDLIB_INSTALL_DIR)/$(BSD_LIB)"
 	$(Q) $(INSTALL_PROGRAM) $(BSD_LIB) \
 		$(DESTDIR)$(BSDLIB_INSTALL_DIR)/$(BSD_LIB)
+	$(Q) $(RM) -f $(DESTDIR)$(BSDLIB_INSTALL_DIR)/$(BSD_SO_LIB)
+	$(Q) $(LN_S) $(BSDLIB_INSTALL_DIR)/$(BSD_LIB) $(DESTDIR)$(BSDLIB_INSTALL_DIR)/$(BSD_SO_LIB)
+	$(Q) $(RM) -f $(DESTDIR)$(BSDLIB_INSTALL_DIR)/$(BSDLIB_IMAGE).dylib
+	$(Q) $(LN_S) $(BSDLIB_INSTALL_DIR)/$(BSD_LIB) $(DESTDIR)$(BSDLIB_INSTALL_DIR)/$(BSDLIB_IMAGE).dylib
 	-$(Q) $(LDCONFIG)
 
 install-strip: install
