diff -ruN e2fsprogs-1.46.2.orig/configure e2fsprogs-1.46.2/configure
--- e2fsprogs-1.46.2.orig/configure	2021-03-01 02:46:44.000000000 +0000
+++ e2fsprogs-1.46.2/configure	2021-04-16 15:17:33.000000000 +0100
@@ -5335,8 +5335,15 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_uuid_uuid_generate" >&5
 $as_echo "$ac_cv_lib_uuid_uuid_generate" >&6; }
 if test "x$ac_cv_lib_uuid_uuid_generate" = xyes; then :
-  LIBUUID=`$PKG_CONFIG --libs uuid`;
-		 STATIC_LIBUUID=`$PKG_CONFIG --static --libs uuid`
+# This was a convoluted mess, hardcoding just "-luuid" as neccesary
+# and sufficient, but then if so using flags from uuid.pc instead (and
+# only --libs not --cflags). It's not necessary (functions are in
+# libSystem) and passing it explicitly picks fink's but sources
+# hardcode a specific subdir header path that only picks OS X's.
+# Just..no. We also pass ac_cv_lib_uuid_uuid_generate=yes to
+# ./configure to avoid doing the -luuid test.
+#  LIBUUID=`$PKG_CONFIG --libs uuid`;
+#		 STATIC_LIBUUID=`$PKG_CONFIG --static --libs uuid`
 else
   as_fn_error $? "external uuid library not found" "$LINENO" 5
 fi
@@ -12383,7 +12390,7 @@
 fi
 
 if test $cross_compiling = no; then
-   BUILD_CFLAGS="$CFLAGS $CPPFLAGS $INCLUDES -DHAVE_CONFIG_H"
+   BUILD_CFLAGS="$CFLAGS $INCLUDES $CPPFLAGS -DHAVE_CONFIG_H"
    BUILD_LDFLAGS="$LDFLAGS"
 fi
 
diff -ruN e2fsprogs-1.46.2.orig/debugfs/Makefile.in e2fsprogs-1.46.2/debugfs/Makefile.in
--- e2fsprogs-1.46.2.orig/debugfs/Makefile.in	2021-03-01 02:46:44.000000000 +0000
+++ e2fsprogs-1.46.2/debugfs/Makefile.in	2021-04-16 15:21:44.000000000 +0100
@@ -96,21 +96,21 @@
 
 e2freefrag.o: $(srcdir)/../misc/e2freefrag.c
 	$(E) "	CC $@"
-	$(Q) $(CC) -c $(ALL_CFLAGS) -I$(srcdir) $< -o $@
+	$(Q) $(CC) -c -I$(srcdir) $(ALL_CFLAGS) $< -o $@
 
 recovery.o: $(srcdir)/../e2fsck/recovery.c
 	$(E) "	CC $@"
-	$(Q) $(CC) -c $(ALL_CFLAGS) -I$(srcdir) \
+	$(Q) $(CC) -c -I$(srcdir) $(ALL_CFLAGS) \
 		$(srcdir)/../e2fsck/recovery.c -o $@
 
 revoke.o: $(srcdir)/../e2fsck/revoke.c
 	$(E) "	CC $@"
-	$(Q) $(CC) -c $(ALL_CFLAGS) -I$(srcdir) \
+	$(Q) $(CC) -c -I$(srcdir) $(ALL_CFLAGS) \
 		$(srcdir)/../e2fsck/revoke.c -o $@
 
 create_inode.o: $(srcdir)/../misc/create_inode.c
 	$(E) "	CC $@"
-	$(Q) $(CC) -c $(ALL_CFLAGS) -I$(srcdir) \
+	$(Q) $(CC) -c -I$(srcdir) $(ALL_CFLAGS) \
 		 $(srcdir)/../misc/create_inode.c -o $@
 
 debugfs.8: $(DEP_SUBSTITUTE) $(srcdir)/debugfs.8.in
diff -ruN e2fsprogs-1.46.2.orig/lib/Makefile.darwin-lib e2fsprogs-1.46.2/lib/Makefile.darwin-lib
--- e2fsprogs-1.46.2.orig/lib/Makefile.darwin-lib	2021-03-01 02:46:44.000000000 +0000
+++ e2fsprogs-1.46.2/lib/Makefile.darwin-lib	2021-04-16 15:17:33.000000000 +0100
@@ -16,25 +16,31 @@
 	$(E) "	MKDIR pic"
 	$(Q) mkdir -p pic
 
-BSD_LIB = $(BSDLIB_IMAGE).$(BSDLIB_VERSION).dylib
+BSD_LIB = $(BSDLIB_IMAGE).$(ELF_VERSION).dylib
+BSD_SO_LIB = $(BSDLIB_IMAGE).$(ELF_SO_VERSION).dylib
 BSDLIB_PIC_FLAG = -fPIC
 
 image:		$(BSD_LIB)
 
 $(BSD_LIB): $(OBJS)
 	$(E) "	GEN_BSD_SOLIB $(BSD_LIB)"
-	$(Q) (cd pic; $(CC) -dynamiclib -compatibility_version 1.0 -current_version $(BSDLIB_VERSION) \
-		-flat_namespace -undefined warning -o $(BSD_LIB) $(OBJS))
+	$(Q) (cd pic; $(CC) -dynamiclib -install_name $(BSDLIB_INSTALL_DIR)/$(BSD_SO_LIB) -compatibility_version 1.0 -current_version $(BSDLIB_VERSION) \
+		-undefined dynamic_lookup -o $(BSD_LIB) $(OBJS))
 	$(Q) $(MV) pic/$(BSD_LIB) .
 	$(Q) $(RM) -f ../$(BSD_LIB)
 	$(Q) (cd ..; $(LN) $(LINK_BUILD_FLAGS) \
 		`echo $(my_dir) | sed -e 's;lib/;;'`/$(BSD_LIB) $(BSD_LIB))
+	$(Q) (cd ..; $(LN) $(LINK_BUILD_FLAGS) $(BSD_LIB) $(BSD_SO_LIB))
 	$(Q) (cd ..; $(LN) $(LINK_BUILD_FLAGS) $(BSD_LIB) $(BSDLIB_IMAGE).dylib)
 
 install-shlibs install:: $(BSD_LIB)
 	$(E) "	INSTALL_PROGRAM $(BSDLIB_INSTALL_DIR)/$(BSD_LIB)"
 	$(Q) $(INSTALL_PROGRAM) $(BSD_LIB) \
 		$(DESTDIR)$(BSDLIB_INSTALL_DIR)/$(BSD_LIB)
+	$(Q) $(RM) -f $(DESTDIR)$(BSDLIB_INSTALL_DIR)/$(BSD_SO_LIB)
+	$(Q) $(LN_S) $(BSDLIB_INSTALL_DIR)/$(BSD_LIB) $(DESTDIR)$(BSDLIB_INSTALL_DIR)/$(BSD_SO_LIB)
+	$(Q) $(RM) -f $(DESTDIR)$(BSDLIB_INSTALL_DIR)/$(BSDLIB_IMAGE).dylib
+	$(Q) $(LN_S) $(BSDLIB_INSTALL_DIR)/$(BSD_LIB) $(DESTDIR)$(BSDLIB_INSTALL_DIR)/$(BSDLIB_IMAGE).dylib
 	-$(Q) $(LDCONFIG)
 
 install-strip: install
diff -ruN e2fsprogs-1.46.2.orig/lib/blkid/probe.c e2fsprogs-1.46.2/lib/blkid/probe.c
--- e2fsprogs-1.46.2.orig/lib/blkid/probe.c	2021-03-01 02:46:44.000000000 +0000
+++ e2fsprogs-1.46.2/lib/blkid/probe.c	2021-04-16 15:35:18.000000000 +0100
@@ -33,6 +33,7 @@
 #ifdef HAVE_ERRNO_H
 #include <errno.h>
 #endif
+#include <time.h>
 #include "blkidP.h"
 #include "uuid/uuid.h"
 #include "probe.h"
