diff -Nurd expect5.45.4.orig/Dbg.c expect5.45.4/Dbg.c
--- expect5.45.4.orig/Dbg.c	2018-02-02 14:15:52.000000000 -0500
+++ expect5.45.4/Dbg.c	2025-08-20 09:36:52.000000000 -0400
@@ -177,7 +177,7 @@
 static int
 breakpoint_test(interp,cmd,bp)
 Tcl_Interp *interp;
-char *cmd;		/* command about to be executed */
+CONST char *cmd;		/* command about to be executed */
 struct breakpoint *bp;	/* breakpoint to test */
 {
     if (bp->re) {
@@ -472,7 +472,7 @@
 		PrintStackBelow(interp,curf->callerVarPtr,viewf);
 		print(interp,"%c%d: %s\n",ptr,curf->level,
 #if TCL_MAJOR_VERSION >= 8
-	      print_objv(interp,curf->objc,curf->objv)
+	      print_objv(interp,curf->objc,(struct Tcl_Obj **)curf->objv)
 #else
 	      print_argv(interp,curf->argc,curf->argv)
 #endif
@@ -491,7 +491,7 @@
 char *level;
 {
 	PrintStackBelow(interp,curf,viewf);
-    print(interp," %s: %s\n",level,print_objv(interp,objc,objv));
+    print(interp," %s: %s\n",level,print_objv(interp,objc,(struct Tcl_Obj **)objv));
 }
 
 /* return 0 if goal matches current frame or goal can't be found */
@@ -663,7 +663,7 @@
 start_interact:
 	if (print_command_first_time) {
 		print(interp,"%s: %s\n",
-				level_text,print_argv(interp,1,&command));
+				level_text,print_argv(interp,1,(char **)&command));
 		print_command_first_time = FALSE;
 	}
 	/* since user is typing a command, don't interrupt it immediately */
diff -Nurd expect5.45.4.orig/configure.in expect5.45.4/configure.in
--- expect5.45.4.orig/configure.in	2025-08-20 09:36:36.000000000 -0400
+++ expect5.45.4/configure.in	2025-08-20 10:44:10.000000000 -0400
@@ -452,7 +452,7 @@
 # because Unixware 2.0 handles it specially and refuses to compile
 # autoconf's automatic test that is a call with no arguments
 AC_MSG_CHECKING([for memcpy])
-AC_TRY_LINK(,[
+AC_TRY_COMPILE([#include <string.h>],[
 char *s1, *s2;
 memcpy(s1,s2,0);
 ],
@@ -486,7 +486,7 @@
 #include <stdio.h>
 #include <sys/wait.h>
 ],[WNOHANG_BACKUP_VALUE=no])
-if "$WNOHANG_BACKUP_VALUE" = "no"; then
+if [[ "$WNOHANG_BACKUP_VALUE" = "no" ]]; then
 	AC_DEFINE(WNOHANG_BACKUP_VALUE, 1)
 	AC_MSG_RESULT(no)
 else
@@ -520,8 +520,10 @@
 # FIXME: check if alarm exists
 AC_MSG_CHECKING([if signals need to be re-armed])
 AC_TRY_RUN([
+#include <stdlib.h>
+#include <unistd.h>
 #include <signal.h>
-#define RETSIGTYPE $retsigtype
+// #define RETSIGTYPE $retsigtype
 
 int signal_rearms = 0;
 
@@ -538,6 +540,7 @@
 signal_rearms++;
 }
 
+int
 main()
 {
 	signal(SIGINT,parent_sigint_handler);
@@ -866,7 +869,7 @@
     else
         EXP_LIB_FLAG="-lexpect`echo ${EXP_LIB_VERSION} | tr -d .`"
     fi
-    EXP_BUILD_LIB_SPEC="-L`pwd` ${EXP_LIB_FLAG}"
+    EXP_BUILD_LIB_SPEC=`echo ${EXP_LIB_FLAG} | sed "s/-l/lib/; s/$/${SHLIB_SUFFIX}/"`
     EXP_LIB_SPEC="-L${libdir} ${EXP_LIB_FLAG}"
 fi
 
diff -Nurd expect5.45.4.orig/exp_chan.c expect5.45.4/exp_chan.c
--- expect5.45.4.orig/exp_chan.c	2025-08-20 09:36:36.000000000 -0400
+++ expect5.45.4/exp_chan.c	2025-08-20 09:36:52.000000000 -0400
@@ -599,7 +599,7 @@
 	if (esPtr->user_waited) continue;	/* one wait only! */
 	if (esPtr->sys_waited) break;
       restart:
-	result = waitpid(esPtr->pid,&esPtr->wait,WNOHANG);
+	result = waitpid(esPtr->pid,(int *)&esPtr->wait,WNOHANG);
 	if (result == esPtr->pid) break;
 	if (result == 0) continue;	/* busy, try next */
 	if (result == -1) {
@@ -618,7 +618,7 @@
     /* should really be recoded using the common wait code in command.c */
     WAIT_STATUS_TYPE status;
 
-    pid = wait(&status);
+    pid = wait((int *)&status);
     for (esPtr = tsdPtr->firstExpPtr;esPtr;esPtr = esPtr->nextPtr) {
 	if (esPtr->pid == pid) {
 	    esPtr->sys_waited = TRUE;
diff -Nurd expect5.45.4.orig/exp_inter.c expect5.45.4/exp_inter.c
--- expect5.45.4.orig/exp_inter.c	2018-02-04 05:12:26.000000000 -0500
+++ expect5.45.4/exp_inter.c	2025-08-20 09:36:52.000000000 -0400
@@ -404,8 +404,8 @@
     int matchBytes)
 {
     int seenBytes;	/* either printed or echoed */
-    int echoBytes;
-    int offsetBytes;
+    int echoBytes = 0;
+    int offsetBytes = 0;
 
     /* write is unlikely to fail, since we just read from same descriptor */
     seenBytes = esPtr->printed + esPtr->echoed;
@@ -437,7 +437,7 @@
     int key)
 {
     Tcl_UniChar *eobOld;  /* old end of buffer */
-    int cc;
+    int cc = 0;
     int numchars;
     Tcl_UniChar *str;
 
diff -Nurd expect5.45.4.orig/exp_log.c expect5.45.4/exp_log.c
--- expect5.45.4.orig/exp_log.c	2025-08-20 09:36:36.000000000 -0400
+++ expect5.45.4/exp_log.c	2025-08-20 09:36:52.000000000 -0400
@@ -85,7 +85,7 @@
     Tcl_UniChar *buf;
     int lenChars;
 {
-    int wc;
+    int wc = 0;
     ThreadSpecificData *tsdPtr = TCL_TSD_INIT(&dataKey);
 
     if (esPtr->valid)
@@ -176,7 +176,7 @@
 
     if ((!tsdPtr->logUser) && (!force_stdout) && (!tsdPtr->logAll)) return;
 
-    (void) vsprintf(bigbuf,fmt,args);
+    (void) vsnprintf(bigbuf,sizeof(bigbuf),fmt,args);
     expDiagWriteBytes(bigbuf,-1);
     if (tsdPtr->logAll || (LOGUSER && tsdPtr->logChannel)) Tcl_WriteChars(tsdPtr->logChannel,bigbuf,-1);
     if (LOGUSER) fprintf(stdout,"%s",bigbuf);
@@ -222,7 +222,7 @@
     va_list args;
 
     fmt = TCL_VARARGS_START(char *,arg1,args);
-    (void) vsprintf(bigbuf,fmt,args);
+    (void) vsnprintf(bigbuf,sizeof(bigbuf),fmt,args);
 
     expDiagWriteChars(bigbuf,-1);
     fprintf(stderr,"%s",bigbuf);
@@ -264,7 +264,7 @@
 
     fmt = TCL_VARARGS_START(char *,arg1,args);
 
-    (void) vsprintf(bigbuf,fmt,args);
+    (void) vsnprintf(bigbuf,sizeof(bigbuf),fmt,args);
 
     expDiagWriteBytes(bigbuf,-1);
     if (tsdPtr->diagToStderr) {
@@ -307,7 +307,7 @@
   int len, rc;
 
   fmt = TCL_VARARGS_START(char *,arg1,args);
-  len = vsprintf(bigbuf,arg1,args);
+  len = vsnprintf(bigbuf,sizeof(bigbuf),arg1,args);
  retry:
   rc = write(2,bigbuf,len);
   if ((rc == -1) && (errno == EAGAIN)) goto retry;
diff -Nurd expect5.45.4.orig/exp_win.c expect5.45.4/exp_win.c
--- expect5.45.4.orig/exp_win.c	2025-08-20 09:36:36.000000000 -0400
+++ expect5.45.4/exp_win.c	2025-08-20 09:51:04.000000000 -0400
@@ -51,6 +51,9 @@
 #   include <sys/ptem.h>
 #endif /* HAVE_SYS_PTEM_H */
 
+/* Darwin needs ioctl.h for ioctl() */
+#include <sys/ioctl.h>
+
 #include "expect.h"
 #include "exp_tty_in.h"
 #include "exp_win.h"
@@ -88,6 +91,7 @@
 #if defined(TIOCSSIZE) && !defined(TIOCSWINSZ)
 	ioctl(fd,TIOCSSIZE,&winsize);
 #endif
+	return 0;
 }
 
 int exp_window_size_get(fd)
@@ -103,6 +107,7 @@
 	winsize.rows = 0;
 	winsize.columns = 0;
 #endif
+	return 0;
 }
 
 void
@@ -152,6 +157,7 @@
 #if defined(TIOCSSIZE) && !defined(TIOCSWINSZ)
 			ioctl(fd,TIOCSSIZE,&win2size);
 #endif
+	return 0;
 }
 
 int exp_win2_size_get(fd)
@@ -163,6 +169,7 @@
 #if defined(TIOCGSIZE) && !defined(TIOCGWINSZ)
 	ioctl(fd,TIOCGSIZE,&win2size);
 #endif
+	return 0;
 }
 
 void
diff -Nurd expect5.45.4.orig/expect.c expect5.45.4/expect.c
--- expect5.45.4.orig/expect.c	2018-02-04 04:58:22.000000000 -0500
+++ expect5.45.4/expect.c	2025-08-20 09:36:52.000000000 -0400
@@ -185,9 +185,9 @@
 {
     if (ec->i_list->duration == EXP_PERMANENT) {
 	if (ec->pat)  { Tcl_DecrRefCount(ec->pat); }
-	if (ec->gate) { Tcl_DecrRefCount(ec->gate); }
 	if (ec->body) { Tcl_DecrRefCount(ec->body); }
     }
+    if (ec->gate) { Tcl_DecrRefCount(ec->gate); }
 
     if (free_ilist) {
 	ec->i_list->ecount--;
@@ -2359,7 +2359,12 @@
 
 	/* "!e" means no case matched - transfer by default */
 	if (!e || e->transfer) {
-	    int remainder = numchars-match;
+	    int remainder;
+	    if (match > numchars) {
+		match = numchars;
+		eo->matchlen = match;
+	    }
+	    remainder = numchars-match;
 	    /* delete matched chars from input buffer */
 	    esPtr->printed -= match;
 	    if (numchars != 0) {
@@ -2899,7 +2904,7 @@
     int def = FALSE;
     char* chan = NULL;
     int i;
-    ExpState *esPtr;
+    ExpState *esPtr = NULL;
 
     for (i=1; i<objc; i++) {
 	char *name;
diff -Nurd expect5.45.4.orig/pty_termios.c expect5.45.4/pty_termios.c
--- expect5.45.4.orig/pty_termios.c	2025-08-20 09:36:36.000000000 -0400
+++ expect5.45.4/pty_termios.c	2025-08-20 09:36:52.000000000 -0400
@@ -10,7 +10,7 @@
 #include <stdio.h>
 #include <signal.h>
 #include <string.h>
-#include <pty.h>
+#include <util.h>
 #include <utmp.h>
 
 #if defined(SIGCLD) && !defined(SIGCHLD)
@@ -690,7 +690,7 @@
 		fcntl(0,F_DUPFD,2);
 	}
 
-	ttytype(SET_TTYTYPE,slave,ttycopy,ttyinit,stty_args);
+	ttytype(SET_TTYTYPE,slave,ttycopy,ttyinit,(char *)stty_args);
 
 #if 0
 #ifdef HAVE_PTYTRAP
diff -Nurd expect5.45.4.orig/tclconfig/tcl.m4 expect5.45.4/tclconfig/tcl.m4
--- expect5.45.4.orig/tclconfig/tcl.m4	2025-08-20 09:36:36.000000000 -0400
+++ expect5.45.4/tclconfig/tcl.m4	2025-08-20 09:36:52.000000000 -0400
@@ -1139,7 +1139,7 @@
     AS_IF([test "$GCC" = yes], [
 	# TEA specific:
 	CFLAGS_OPTIMIZE=-O2
-	CFLAGS_WARNING="-Wall"
+	CFLAGS_WARNING="-Wall -Wno-unused-variable -Wno-unused-function -Wno-knr-promoted-parameter -Wno-switch"
     ], [CFLAGS_WARNING=""])
 dnl FIXME: Replace AC_CHECK_PROG with AC_CHECK_TOOL once cross compiling is fixed.
 dnl AC_CHECK_TOOL(AR, ar)
@@ -2400,7 +2400,7 @@
 	AC_TRY_COMPILE([#include <time.h>],
 	    [extern long timezone;
 	    timezone += 1;
-	    exit (0);],
+	    ],
 	    tcl_cv_timezone_long=yes, tcl_cv_timezone_long=no)])
     if test $tcl_cv_timezone_long = yes ; then
 	AC_DEFINE(HAVE_TIMEZONE_VAR, 1, [Should we use the global timezone variable?])
@@ -3196,7 +3196,7 @@
 	MAKE_STUB_LIB="\${STLIB_LD} -out:\[$]@ \$(PKG_STUB_OBJECTS)"
     else
 	MAKE_STATIC_LIB="\${STLIB_LD} \[$]@ \$(PKG_OBJECTS)"
-	MAKE_SHARED_LIB="\${SHLIB_LD} -o \[$]@ \$(PKG_OBJECTS) \${SHLIB_LD_LIBS}"
+	MAKE_SHARED_LIB="\${SHLIB_LD} -o \[$]@ \$(PKG_OBJECTS) \${SHLIB_LD_LIBS} -install_name @SONAME@"
 	MAKE_STUB_LIB="\${STLIB_LD} \[$]@ \$(PKG_STUB_OBJECTS)"
     fi
 
@@ -3360,7 +3360,7 @@
     AC_REQUIRE([TEA_PUBLIC_TCL_HEADERS])
     AC_MSG_CHECKING([for Tcl private include files])
 
-    TCL_SRC_DIR_NATIVE=`${CYGPATH} ${TCL_SRC_DIR}`
+    TCL_SRC_DIR_NATIVE=`${CYGPATH} ${TCL_SRC_DIR}|sed s:/private::g`
     TCL_TOP_DIR_NATIVE=\"${TCL_SRC_DIR_NATIVE}\"
 
     # Check to see if tcl<Plat>Port.h isn't already with the public headers
@@ -3943,7 +3943,7 @@
 	eval $1_LIB_FLAG="-l$1`echo ${PACKAGE_VERSION} | tr -d .`${DBGX}"
 	eval $1_STUB_LIB_FLAG="-l$1stub`echo ${PACKAGE_VERSION} | tr -d .`${DBGX}"
     fi
-    $1_BUILD_LIB_SPEC="-L`pwd` ${$1_LIB_FLAG}"
+    $1_BUILD_LIB_SPEC=`echo ${EXP_LIB_FLAG} | sed "s/-l/lib/; s/$/${SHLIB_SUFFIX}/"`
     $1_LIB_SPEC="-L${pkglibdir} ${$1_LIB_FLAG}"
     $1_BUILD_STUB_LIB_SPEC="-L`pwd` [$]{$1_STUB_LIB_FLAG}"
     $1_STUB_LIB_SPEC="-L${pkglibdir} [$]{$1_STUB_LIB_FLAG}"
