diff -bur fhist-1.12/ fhist-1.12-patched/
--- fhist-1.12/Makefile.in	Wed Nov 27 17:19:38 2002
+++ fhist-1.12-patched/Makefile.in	Wed Dec  4 01:23:22 2002
@@ -77,6 +77,11 @@
 mandir = $(RPM_BUILD_ROOT)@mandir@
 
 #
+# where to put the locale-specific messages
+#
+localedir = $(RPM_BUILD_ROOT)@localedir@
+
+#
 # Which yacc to use
 #
 YACC = @YACC@
@@ -842,8 +847,8 @@
 		--output=$@ lib/en/LC_MESSAGES/fcomp.po \
 		lib/en/LC_MESSAGES/common.po
 
-$(libdir)/en/LC_MESSAGES/fcomp.mo: lib/en/LC_MESSAGES/fcomp.mo \
-		lib/en/LC_MESSAGES/.libdir
+$(localedir)/en/LC_MESSAGES/fcomp.mo: lib/en/LC_MESSAGES/fcomp.mo \
+		lib/en/LC_MESSAGES/.localedir
 	$(INSTALL_DATA) lib/en/LC_MESSAGES/fcomp.mo $@
 
 lib/en/LC_MESSAGES/fhist.mo: lib/en/LC_MESSAGES/fhist.po \
@@ -852,8 +857,8 @@
 		--output=$@ lib/en/LC_MESSAGES/fhist.po \
 		lib/en/LC_MESSAGES/common.po
 
-$(libdir)/en/LC_MESSAGES/fhist.mo: lib/en/LC_MESSAGES/fhist.mo \
-		lib/en/LC_MESSAGES/.libdir
+$(localedir)/en/LC_MESSAGES/fhist.mo: lib/en/LC_MESSAGES/fhist.mo \
+		lib/en/LC_MESSAGES/.localedir
 	$(INSTALL_DATA) lib/en/LC_MESSAGES/fhist.mo $@
 
 lib/en/LC_MESSAGES/fmerge.mo: lib/en/LC_MESSAGES/fmerge.po \
@@ -862,8 +867,8 @@
 		--output=$@ lib/en/LC_MESSAGES/fmerge.po \
 		lib/en/LC_MESSAGES/common.po
 
-$(libdir)/en/LC_MESSAGES/fmerge.mo: lib/en/LC_MESSAGES/fmerge.mo \
-		lib/en/LC_MESSAGES/.libdir
+$(localedir)/en/LC_MESSAGES/fmerge.mo: lib/en/LC_MESSAGES/fmerge.mo \
+		lib/en/LC_MESSAGES/.localedir
 	$(INSTALL_DATA) lib/en/LC_MESSAGES/fmerge.mo $@
 
 lib/en/building.ps: lib/en/building/main.man
@@ -1062,19 +1067,19 @@
 	$(SH) test/00/t0026a.sh
 
 
-lib/en/LC_MESSAGES/.libdir: lib/en/.libdir
-	-$(INSTALL) -m 0755 -d $(libdir)/en/LC_MESSAGES
-	@-test -d $(libdir)/en/LC_MESSAGES && touch $@
+lib/en/LC_MESSAGES/.localedir: lib/en/.localedir
+	-$(INSTALL) -m 0755 -d $(localedir)/en/LC_MESSAGES
+	@-test -d $(localedir)/en/LC_MESSAGES && touch $@
 	@sleep 1
 
-lib/en/.libdir: lib/.libdir
-	-$(INSTALL) -m 0755 -d $(libdir)/en
-	@-test -d $(libdir)/en && touch $@
+lib/en/.localedir: lib/.localedir
+	-$(INSTALL) -m 0755 -d $(localedir)/en
+	@-test -d $(localedir)/en && touch $@
 	@sleep 1
 
-lib/.libdir:
-	-$(INSTALL) -m 0755 -d $(libdir)
-	@-test -d $(libdir) && touch $@
+lib/.localedir:
+	-$(INSTALL) -m 0755 -d $(localedir)
+	@-test -d $(localedir) && touch $@
 	@sleep 1
 
 lib/en/man1/.mandir: lib/en/.mandir
@@ -1168,9 +1173,9 @@
 
 install-po: install-po-$(HAVE_I18N)
 
-install-po-yes: $(libdir)/en/LC_MESSAGES/fcomp.mo \
-		$(libdir)/en/LC_MESSAGES/fhist.mo \
-		$(libdir)/en/LC_MESSAGES/fmerge.mo
+install-po-yes: $(localedir)/en/LC_MESSAGES/fcomp.mo \
+		$(localedir)/en/LC_MESSAGES/fhist.mo \
+		$(localedir)/en/LC_MESSAGES/fmerge.mo
 
 install-po-no:
 
@@ -1274,7 +1279,7 @@
 	rm -f fhist/name.$(OBJEXT) fhist/prune.$(OBJEXT)
 	rm -f fhist/subroutine.$(OBJEXT) fhist/update.$(OBJEXT)
 	rm -f fmerge/main.$(OBJEXT) fmerge/work.$(OBJEXT)
-	rm -f lib/en/LC_MESSAGES/.libdir lib/en/.libdir lib/.libdir
+	rm -f lib/en/LC_MESSAGES/.localedir lib/en/.localedir lib/.localedir
 	rm -f lib/en/building.ps lib/en/building.dvi lib/en/building.txt
 	rm -f lib/en/man1/.mandir lib/en/.mandir lib/en/man1/.datadir
 	rm -f lib/en/.datadir lib/.datadir lib/en/readme.ps
@@ -1306,7 +1311,7 @@
 sure: $(TestFiles)
 	@echo Passed All Tests
 
-install: install-bin install-man install-iman install-po install-doc
+install: install-bin install-man install-po install-doc
 	@echo Installation Complete
 
 install-bin: $(bindir)/$(PROGRAM_PREFIX)fcomp$(PROGRAM_SUFFIX)$(EXEEXT) \
--- fhist-1.12/configure	Wed Nov 27 17:19:38 2002
+++ fhist-1.12-patched/configure	Wed Dec  4 01:23:22 2002
@@ -345,6 +345,7 @@
 oldincludedir='/usr/include'
 infodir='${prefix}/info'
 mandir='${prefix}/man'
+localedir='${prefix}/share/locale'
 
 ac_prev=
 for ac_option
@@ -4954,6 +4955,7 @@
 s,@oldincludedir@,$oldincludedir,;t t
 s,@infodir@,$infodir,;t t
 s,@mandir@,$mandir,;t t
+s,@localedir@,$localedir,;t t
 s,@build_alias@,$build_alias,;t t
 s,@host_alias@,$host_alias,;t t
 s,@target_alias@,$target_alias,;t t
