diff -ruN grep-dctrl-1.9/Makefile.in grep-dctrl-1.9.new/Makefile.in
--- grep-dctrl-1.9/Makefile.in	2001-07-30 13:03:08.000000000 -0400
+++ grep-dctrl-1.9.new/Makefile.in	2002-12-26 12:23:20.000000000 -0500
@@ -57,6 +57,8 @@
 NORMAL_UNINSTALL = :
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
+EXTRASRC = @EXTRASRC@
+EXTRAOBJ = @EXTRAOBJ@
 CATALOGS = @CATALOGS@
 CATOBJEXT = @CATOBJEXT@
 CC = @CC@
@@ -87,7 +89,8 @@
 SUBDIRS = intl po
 
 bin_PROGRAMS = grep-dctrl
-grep_dctrl_SOURCES = grep-dctrl.c matcher.c buffer.c strutil.c msg.c 	buffer.h i18n.h matcher.h  msg.h strutil.h rc.c rc.h
+grep_dctrl_SOURCES = grep-dctrl.c matcher.c buffer.c strutil.c msg.c \
+buffer.h i18n.h matcher.h msg.h strutil.h rc.c rc.h $(EXTRASRC)
 
 
 sysconf_DATA = grep-dctrl.rc
@@ -110,7 +113,7 @@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
 grep_dctrl_OBJECTS =  grep-dctrl.o matcher.o buffer.o strutil.o msg.o \
-rc.o
+rc.o $(EXTRAOBJ)
 grep_dctrl_LDADD = $(LDADD)
 grep_dctrl_DEPENDENCIES = 
 grep_dctrl_LDFLAGS = 
diff -ruN grep-dctrl-1.9/configure grep-dctrl-1.9.new/configure
--- grep-dctrl-1.9/configure	2000-02-06 09:04:51.000000000 -0500
+++ grep-dctrl-1.9.new/configure	2002-12-26 12:25:44.000000000 -0500
@@ -3400,6 +3400,10 @@
 s%@USE_INCLUDED_LIBINTL@%$USE_INCLUDED_LIBINTL%g
 s%@CATALOGS@%$CATALOGS%g
 s%@CATOBJEXT@%$CATOBJEXT%g
+
+s%@EXTRASRC@%$EXTRASRC%g
+s%@EXTRAOBJ@%$EXTRAOBJ%g
+
 s%@DATADIRNAME@%$DATADIRNAME%g
 s%@GMOFILES@%$GMOFILES%g
 s%@INSTOBJEXT@%$INSTOBJEXT%g
diff -ruN grep-dctrl-1.9/matcher.c grep-dctrl-1.9.new/matcher.c
--- grep-dctrl-1.9/matcher.c	2005-04-03 16:57:15.000000000 -0700
+++ grep-dctrl-1.9.new/matcher.c	2005-04-03 16:59:17.000000000 -0700
@@ -54,6 +54,7 @@
 }
 
 /* strstr ignoring case */
+#if 0
 static char *
 strcasestr (const char * haystack, const char * needle)
      /* This implemetation is suboptimal.  Ugh. */
@@ -84,6 +85,7 @@
 
   return rv;
 }
+#endif
 
 /* Copy the body of the field "field" from "s" to "body". */
 static void
