diff -ruN jpilot-1.8.2-orig/configure jpilot-1.8.2/configure
--- jpilot-1.8.2-orig/configure	2014-05-17 22:45:22
+++ jpilot-1.8.2/configure	2023-10-20 04:18:12
@@ -8107,7 +8107,7 @@
 	  _lt_dar_allow_undefined='${wl}-undefined ${wl}dynamic_lookup' ;;
 	10.[012]*)
 	  _lt_dar_allow_undefined='${wl}-flat_namespace ${wl}-undefined ${wl}suppress' ;;
-	10.*)
+	*)
 	  _lt_dar_allow_undefined='${wl}-undefined ${wl}dynamic_lookup' ;;
       esac
     ;;
@@ -14866,6 +14866,7 @@
 
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdlib.h>
 #include <pi-version.h>
 int
 main ()
