diff -Nurd -x'*~' dmtx-utils-0.7.4.orig/common/dmtxutil.c dmtx-utils-0.7.4/common/dmtxutil.c
--- dmtx-utils-0.7.4.orig/common/dmtxutil.c	2011-06-03 01:13:10.000000000 -0400
+++ dmtx-utils-0.7.4/common/dmtxutil.c	2015-05-10 03:04:08.000000000 -0400
@@ -27,6 +27,7 @@
 #include <ctype.h>
 #include <errno.h>
 #include <assert.h>
+#include <sys/time.h>
 #include <dmtx.h>
 #include "dmtxutil.h"
 
diff -Nurd -x'*~' dmtx-utils-0.7.4.orig/dmtxread/Makefile.in dmtx-utils-0.7.4/dmtxread/Makefile.in
--- dmtx-utils-0.7.4.orig/dmtxread/Makefile.in	2011-06-03 01:13:55.000000000 -0400
+++ dmtx-utils-0.7.4/dmtxread/Makefile.in	2015-05-10 03:05:42.000000000 -0400
@@ -203,13 +203,13 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = subdir-objects
-AM_CPPFLAGS = -Wshadow -Wall -pedantic
+AM_CPPFLAGS = -Wshadow -Wall -pedantic $(MAGICK_CFLAGS) -D_MAGICK_CONFIG_H
 dmtxread_SOURCES = dmtxread.c dmtxread.h ../common/dmtxutil.c ../common/dmtxutil.h
-dmtxread_CFLAGS = $(DMTX_CFLAGS) $(MAGICK_CFLAGS) -D_MAGICK_CONFIG_H
+dmtxread_CFLAGS = $(DMTX_CFLAGS)
 dmtxread_LDFLAGS = $(DMTX_LIBS) $(MAGICK_LIBS)
 dmtxread_LDADD = $(LIBOBJS)
 dmtxread_debug_SOURCES = dmtxread.c dmtxread.h ../common/dmtxutil.c ../common/dmtxutil.h
-dmtxread_debug_CFLAGS = $(DMTX_CFLAGS) $(MAGICK_CFLAGS) -D_MAGICK_CONFIG_H
+dmtxread_debug_CFLAGS = $(DMTX_CFLAGS)
 dmtxread_debug_LDFLAGS = -static $(DMTX_LIBS) $(MAGICK_LIBS)
 dmtxread_debug_LDADD = $(LIBOBJS)
 all: all-am
diff -Nurd -x'*~' dmtx-utils-0.7.4.orig/dmtxwrite/Makefile.in dmtx-utils-0.7.4/dmtxwrite/Makefile.in
--- dmtx-utils-0.7.4.orig/dmtxwrite/Makefile.in	2011-06-03 01:13:55.000000000 -0400
+++ dmtx-utils-0.7.4/dmtxwrite/Makefile.in	2015-05-10 03:06:19.000000000 -0400
@@ -203,13 +203,13 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = subdir-objects
-AM_CPPFLAGS = -Wshadow -Wall -pedantic
+AM_CPPFLAGS = -Wshadow -Wall -pedantic $(MAGICK_CFLAGS) -D_MAGICK_CONFIG_H
 dmtxwrite_SOURCES = dmtxwrite.c dmtxwrite.h ../common/dmtxutil.c ../common/dmtxutil.h
-dmtxwrite_CFLAGS = $(DMTX_CFLAGS) $(MAGICK_CFLAGS) -D_MAGICK_CONFIG_H
+dmtxwrite_CFLAGS = $(DMTX_CFLAGS)
 dmtxwrite_LDFLAGS = $(DMTX_LIBS) $(MAGICK_LIBS)
 dmtxwrite_LDADD = $(LIBOBJS)
 dmtxwrite_debug_SOURCES = dmtxwrite.c dmtxwrite.h ../common/dmtxutil.c ../common/dmtxutil.h
-dmtxwrite_debug_CFLAGS = $(DMTX_FLAGS) $(MAGICK_CFLAGS) -D_MAGICK_CONFIG_H
+dmtxwrite_debug_CFLAGS = $(DMTX_FLAGS)
 dmtxwrite_debug_LDFLAGS = -static $(DMTX_LIBS) $(MAGICK_LIBS)
 dmtxwrite_debug_LDADD = $(LIBOBJS)
 all: all-am
