diff -ruN pciutils-3.10.0-orig/lib/configure pciutils-3.10.0/lib/configure
--- pciutils-3.10.0-orig/lib/configure	2023-03-05 07:45:24.000000000 -0600
+++ pciutils-3.10.0/lib/configure	2023-07-04 05:55:59.000000000 -0500
@@ -315,6 +315,8 @@
 		echo >>$m 'PCILIB=$(LIBNAME).$(LIBEXT).$(VERSION)'
 	elif [ "$LIBEXT" = dll ]; then
 		echo >>$m 'PCILIB=$(LIBNAME)$(ABI_VERSION).$(LIBEXT)'
+	elif [ "$LIBEXT" = dylib ]; then
+		echo >>$m 'PCILIB=$(LIBNAME).$(ABI_VERSION).$(LIBEXT)'
 	else
 		echo >>$m 'PCILIB=$(LIBNAME).$(VERSION).$(LIBEXT)'
 	fi
@@ -326,7 +328,7 @@
 		echo >>$m 'PCILIB_LDFLAGS+=-Wl,-soname,$(LIBNAME).$(LIBEXT).$(ABI_VERSION)'
 		echo >>$m 'PCILIB_LDFLAGS+=-Wl,--version-script=libpci.ver'
 	elif [ "$LIBEXT" = dylib ]; then
-		echo >>$m 'PCILIB_LDFLAGS+=-Wl,-install_name,$(LIBDIR)/$(PCILIB)'
+		echo >>$m 'PCILIB_LDFLAGS+=-Wl,-install_name,$(LIBDIR)/$(PCILIB),-compatibility_version,$(ABI_VERSION),-current_version,$(VERSION)'
 	elif [ "$LIBEXT" = dll ]; then
 		echo >>$m 'PCIIMPDEF=$(LIBNAME)$(ABI_VERSION).def'
 		# GCC's -fvisibility=hidden is broken for Windows targets, use -Wl,--exclude-all-symbols instead (supported since GNU LD 2.21)
