diff -ruN lv451-orig/src/configure lv451/src/configure
--- lv451-orig/src/configure	2004-01-05 00:37:17.000000000 -0600
+++ lv451/src/configure	2024-06-29 11:17:57.000000000 -0500
@@ -1324,6 +1324,7 @@
   cat > conftest.$ac_ext <<EOF
 #line 1326 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <ctype.h>
 #define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 #define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
@@ -1513,6 +1514,7 @@
  *
  * Snarfed from Chet Ramey's bash pgrp.c test program
  */
+#include <stdlib.h>
 #include <stdio.h>
 #include <sys/types.h>
 
diff -ruN lv451-orig/src/guess.c lv451/src/guess.c
--- lv451-orig/src/guess.c	2003-11-12 21:08:19.000000000 -0600
+++ lv451/src/guess.c	2024-06-29 11:13:34.000000000 -0500
@@ -21,6 +21,7 @@
  */
 
 #include <stdio.h>
+#include <string.h>
 
 #include <import.h>
 #include <decode.h>
diff -ruN lv451-orig/src/guesslocale.c lv451/src/guesslocale.c
--- lv451-orig/src/guesslocale.c	2004-01-05 02:41:22.000000000 -0600
+++ lv451/src/guesslocale.c	2024-06-29 11:19:58.000000000 -0500
@@ -23,6 +23,7 @@
 #ifdef HAVE_SETLOCALE
 
 #include <stdlib.h>
+#include <ctype.h>
 #include <string.h>
 #include <locale.h>
 #if defined(HAVE_LANGINFO_CODESET)
