diff -Nurd -x'*~' MediaInfo.orig/Source/GUI/WxWidgets/GUI_Main_FileDrop.cpp MediaInfo/Source/GUI/WxWidgets/GUI_Main_FileDrop.cpp
--- MediaInfo.orig/Source/GUI/WxWidgets/GUI_Main_FileDrop.cpp	2012-03-11 06:39:30.000000000 -0400
+++ MediaInfo/Source/GUI/WxWidgets/GUI_Main_FileDrop.cpp	2015-10-12 06:07:25.000000000 -0400
@@ -67,9 +67,9 @@
     wxDateTime Begin=wxDateTime::UNow();
     for (size_t Pos=0; Pos<FileNames.size(); Pos++)
     #if defined(__WXMSW__)
-        C->Menu_File_Open_Files_Continue(FileNames[Pos].c_str());
+        C->Menu_File_Open_Files_Continue(String(FileNames[Pos].c_str()));
     #else //defined(__WXMSW__)
-        C->Menu_File_Open_Files_Continue(FileNames2[Pos].c_str());
+        C->Menu_File_Open_Files_Continue(String(FileNames2[Pos].c_str()));
     #endif //defined(__WXMSW__)
     wxTimeSpan Span=wxDateTime::UNow()-Begin;
     ((GUI_Main*)C->GUI_Main_Handler)->View_Refresh();
diff -Nurd -x'*~' MediaInfo.orig/Source/GUI/WxWidgets/GUI_Main_Menu.cpp MediaInfo/Source/GUI/WxWidgets/GUI_Main_Menu.cpp
--- MediaInfo.orig/Source/GUI/WxWidgets/GUI_Main_Menu.cpp	2012-01-22 07:49:16.000000000 -0500
+++ MediaInfo/Source/GUI/WxWidgets/GUI_Main_Menu.cpp	2015-10-12 06:08:51.000000000 -0400
@@ -158,7 +158,7 @@
 void GUI_Main::OnMenu_File_Open_Files(wxCommandEvent& WXUNUSED(event))
 {
     //User interaction
-    wxFileDialog* Dialog=new wxFileDialog(this, _T("Choose a file"), _T(""), _T(""), _T("*.*"), wxOPEN|wxFILE_MUST_EXIST|wxMULTIPLE);
+    wxFileDialog* Dialog=new wxFileDialog(this, _T("Choose a file"), _T(""), _T(""), _T("*.*"), wxFD_OPEN|wxFD_FILE_MUST_EXIST|wxFD_MULTIPLE);
     if (Dialog->ShowModal()!=wxID_OK)
     {
         delete Dialog;
@@ -171,7 +171,7 @@
     //Configuring
     C->Menu_File_Open_Files_Begin();
     for (size_t Pos=0; Pos<FileNames.size(); Pos++)
-        C->Menu_File_Open_Files_Continue(FileNames[Pos].c_str());
+        C->Menu_File_Open_Files_Continue(String(FileNames[Pos].c_str()));
 
     //Showing
     View->GUI_Refresh();
@@ -189,7 +189,7 @@
 
     //Configuring
     C->Menu_File_Open_Files_Begin();
-    C->Menu_File_Open_Files_Continue(DirName.c_str());
+    C->Menu_File_Open_Files_Continue(String(DirName.c_str()));
 
     //Showing
     View->GUI_Refresh();
