diff -ruN mtools-4.0.30-orig/Makefile.in mtools-4.0.30/Makefile.in
--- mtools-4.0.30-orig/Makefile.in	2021-06-06 14:23:24.000000000 -0500
+++ mtools-4.0.30/Makefile.in	2021-06-17 20:13:44.000000000 -0500
@@ -149,8 +149,8 @@
 X_LIBS = @X_LIBS@
 X_EXTRA_LIBS = @X_EXTRA_LIBS@
 X_PRE_LIBS = @X_PRE_LIBS@
-CFLAGS = $(CPPFLAGS) $(DEFS) $(MYCFLAGS) -fno-strict-aliasing -I. @extraincludedir@ -I@srcdir@ $(USERCFLAGS)
-CXXFLAGS  = $(CPPFLAGS) $(DEFS) $(MYCXXFLAGS) -I. @extraincludedir@ -I@srcdir@ $(USERCFLAGS)
+CFLAGS = -I. -I@srcdir@ $(CPPFLAGS) $(DEFS) $(MYCFLAGS) -fno-strict-aliasing @extraincludedir@ $(USERCFLAGS)
+CXXFLAGS  = -I. -I@srcdir@ $(CPPFLAGS) $(DEFS) $(MYCXXFLAGS) @extraincludedir@ $(USERCFLAGS)
 LINK      = $(CC) $(LDFLAGS) $(USERLDFLAGS) @extralibdir@
 ALLLIBS   = $(USERLDLIBS) $(MACHDEPLIBS) $(SHLIB) $(LIBS)
 X_LDFLAGS = $(X_EXTRA_LIBS) $(X_LIBS) -lXau -lX11 $(LIBS)
@@ -165,7 +165,7 @@
 #	$(CXX) $(CXXFLAGS) -c $<
 
 mtools: $(OBJS1)
-	$(LINK) $(OBJS1) -o $@ $(ALLLIBS)
+	$(LINK) $(OBJS1) -o $@ $(ALLLIBS) -liconv
 
 mkmanifest: $(OBJS2)
 	$(LINK) $(OBJS2) -o $@ $(ALLLIBS)
diff -ruN mtools-4.0.30-orig/config.c mtools-4.0.30/config.c
--- mtools-4.0.30-orig/config.c	2021-06-06 03:23:10.000000000 -0500
+++ mtools-4.0.30/config.c	2021-06-17 20:14:26.000000000 -0500
@@ -19,6 +19,8 @@
 #include "mtools.h"
 #include "codepage.h"
 #include "mtoolsPaths.h"
+#include <string.h>
+#include <xlocale.h>
 
 /* global variables */
 /* they are not really harmful here, because there is only one configuration
@@ -836,13 +838,7 @@
 	memcpy(devices, const_devices,
 	       nr_const_devices*sizeof(struct device));
 
-    (void) ((parse(CONF_FILE,1) |
-	     parse(LOCAL_CONF_FILE,1) |
-	     parse(SYS_CONF_FILE,1)) ||
-	    (parse(OLD_CONF_FILE,1) |
-	     parse(OLD_LOCAL_CONF_FILE,1)));
-    /* the old-name configuration files only get executed if none of the
-     * new-name config files were used */
+    parse(SYS_CONF_FILE,1);
 
     homedir = get_homedir();
     if ( homedir ){
diff -ruN mtools-4.0.30-orig/mformat.c mtools-4.0.30/mformat.c
--- mtools-4.0.30-orig/mformat.c	2021-06-16 14:53:47.000000000 -0500
+++ mtools-4.0.30/mformat.c	2021-06-17 20:14:50.000000000 -0500
@@ -476,7 +476,7 @@
 
 	if ( Fs->num_clus < FAT16 && Fs->fat_bits > 16 ){
 		fprintf(stderr,"Too few clusters for this fat size."
-			" Please choose a 16-bit fat in your /etc/mtools.conf"
+			" Please choose a 16-bit fat in your SYSCONFDIR/mtools.conf"
 			" or .mtoolsrc file\n");
 		exit(1);
 	}
diff -ruN mtools-4.0.30-orig/mtools.h mtools-4.0.30/mtools.h
--- mtools-4.0.30-orig/mtools.h	2021-06-09 17:06:31.000000000 -0500
+++ mtools-4.0.30/mtools.h	2021-06-17 20:15:46.000000000 -0500
@@ -289,7 +289,9 @@
 
 #define New(type) ((type*)(calloc(1,sizeof(type))))
 #define Grow(adr,n,type) ((type*)(realloc((char *)adr,n*sizeof(type))))
+#ifndef Free
 #define Free(adr) (free((char *)adr));
+#endif
 #define NewArray(size,type) ((type*)(calloc((size),sizeof(type))))
 
 void mattrib(int argc, char **argv, int type);
diff -ruN mtools-4.0.30-orig/mtools.texi mtools-4.0.30/mtools.texi
--- mtools-4.0.30-orig/mtools.texi	2021-05-24 02:51:17.000000000 -0500
+++ mtools-4.0.30/mtools.texi	2021-06-17 20:16:12.000000000 -0500
@@ -599,9 +599,6 @@
 @file{@value{SYSCONFDIR}mtools.conf} is the system-wide configuration file,
 and @file{~/.mtoolsrc} is the user's private configuration file.
 
-On some systems, the system-wide configuration file is called
-@file{/etc/default/mtools.conf} instead.
-
 
 @node general syntax, default values, configuration file location, Configuration
 @subsection General configuration file syntax
diff -ruN mtools-4.0.30-orig/mtoolsPaths.h mtools-4.0.30/mtoolsPaths.h
--- mtools-4.0.30-orig/mtoolsPaths.h	2018-12-02 10:23:26.000000000 -0600
+++ mtools-4.0.30/mtoolsPaths.h	2021-06-17 20:16:54.000000000 -0500
@@ -22,22 +22,8 @@
  */
 
 
-#define CONF_FILE "/etc/mtools.conf"
-
-
-#define OLD_CONF_FILE "/etc/mtools"
-
-
-
-#define LOCAL_CONF_FILE "/etc/default/mtools.conf"
-/* Use this if you like to keep the configuration file in a non-standard
- * place such as /etc/default, /opt/etc, /usr/etc, /usr/local/etc ...
- */
-
 #define SYS_CONF_FILE SYSCONFDIR "/mtools.conf"
 
-#define OLD_LOCAL_CONF_FILE "/etc/default/mtools"
-
 
 
 #define CFG_FILE1 "/.mtoolsrc"
