diff -Nurd pinentry-1.2.1.orig/configure pinentry-1.2.1/configure
--- pinentry-1.2.1.orig/configure	2022-08-24 06:49:45.000000000 -0400
+++ pinentry-1.2.1/configure	2025-09-14 12:46:12.000000000 -0400
@@ -7433,6 +7433,9 @@
         if test "$req_libassuan_api" -eq "$tmp" ; then
           { $as_echo "$as_me:${as_lineno-$LINENO}: result: okay" >&5
 $as_echo "okay" >&6; }
+        elif test "$req_libassuan_api" -eq 2 -a "$tmp" -eq 3 ; then
+          { $as_echo "$as_me:${as_lineno-$LINENO}: result: okay" >&5
+$as_echo "okay" >&6; }
         else
           ok=no
           { $as_echo "$as_me:${as_lineno-$LINENO}: result: does not match.  want=$req_libassuan_api got=$tmp." >&5
