diff -Nurd -x'*~' rpm-4.16.1.2.orig/scripts/rpm2cpio.sh rpm-4.16.1.2/scripts/rpm2cpio.sh
--- rpm-4.16.1.2.orig/scripts/rpm2cpio.sh	2020-05-28 06:04:25.000000000 -0400
+++ rpm-4.16.1.2/scripts/rpm2cpio.sh	2021-01-03 20:32:41.000000000 -0500
@@ -11,7 +11,7 @@
 
 _dd() {
 	local o="$1"; shift
-	dd if="$pkg" skip="$o" iflag=skip_bytes status=none $*
+	gdd if="$pkg" skip="$o" iflag=skip_bytes status=none $*
 }
 
 calcsize() {
