diff -ru coreutils-8.19/lib/localcharset.c coreutils-8.19-patched/lib/localcharset.c
--- coreutils-8.19/lib/localcharset.c	2012-09-05 23:21:21.000000000 -0600
+++ coreutils-8.19-patched/lib/localcharset.c	2012-09-05 23:21:04.000000000 -0600
@@ -278,6 +278,7 @@
            "SJIS" "\0" "SHIFT_JIS" "\0"
            "ARMSCII-8" "\0" "ARMSCII-8" "\0"
            "PT154" "\0" "PT154" "\0"
+           "US-ASCII" "\0" "ASCII" "\0"
          /*"ISCII-DEV" "\0" "?" "\0"*/
            "*" "\0" "UTF-8" "\0";
 # endif
--- coreutils-8.21/gnulib-tests/test-vasprintf-posix.c.orig	2013-11-09 09:33:24.000000000 -0500
+++ coreutils-8.21/gnulib-tests/test-vasprintf-posix.c	2013-11-09 09:37:11.000000000 -0500
@@ -232,7 +232,8 @@
     int retval =
       my_asprintf (&result, "%.0a %d", 1.5, 33, 44, 55);
     ASSERT (result != NULL);
-    ASSERT (strcmp (result, "0x2p+0 33") == 0
+    ASSERT (strcmp (result, "0x1p+0 33") == 0
+	    || strcmp (result, "0x2p+0 33") == 0
             || strcmp (result, "0x3p-1 33") == 0
             || strcmp (result, "0x6p-2 33") == 0
             || strcmp (result, "0xcp-3 33") == 0);
@@ -245,7 +246,8 @@
     int retval =
       my_asprintf (&result, "%.0a %d", 1.51, 33, 44, 55);
     ASSERT (result != NULL);
-    ASSERT (strcmp (result, "0x2p+0 33") == 0
+    ASSERT (strcmp (result, "0x1p+0 33") == 0
+	    || strcmp (result, "0x2p+0 33") == 0
             || strcmp (result, "0x3p-1 33") == 0
             || strcmp (result, "0x6p-2 33") == 0
             || strcmp (result, "0xcp-3 33") == 0);
@@ -647,7 +649,8 @@
     int retval =
       my_asprintf (&result, "%.0La %d", 1.5L, 33, 44, 55);
     ASSERT (result != NULL);
-    ASSERT (strcmp (result, "0x2p+0 33") == 0
+    ASSERT (strcmp (result, "0x1p+0 33") == 0
+	    || strcmp (result, "0x2p+0 33") == 0
             || strcmp (result, "0x3p-1 33") == 0
             || strcmp (result, "0x6p-2 33") == 0
             || strcmp (result, "0xcp-3 33") == 0);
--- coreutils-8.21/lib/getgroups.c.orig	2013-11-09 09:54:54.000000000 -0500
+++ coreutils-8.21/lib/getgroups.c	2013-11-09 09:55:09.000000000 -0500
@@ -86,7 +86,7 @@
         }
       saved_errno = errno;
       free (gbuf);
-      errno == saved_errno;
+      errno = saved_errno;
       return result;
     }
 
