diff -ruN mpgtx-1.3.1.orig/configure mpgtx-1.3.1/configure
--- mpgtx-1.3.1.orig/configure	2005-05-07 23:55:09.000000000 -0600
+++ mpgtx-1.3.1/configure	2005-05-08 00:05:43.000000000 -0600
@@ -7,9 +7,9 @@
 #################################################
 
 prefix="/usr/local"
-manprefix='$(PREFIX)'
+manprefix='$(PREFIX)/share'
 #compiler flags
-CC="g++"
+CC="g++"
 common_cflags="-Wall"
 devel_cflags="-g -Werror -fno-builtin"
 optimization_cflags="-O3"
@@ -34,29 +34,29 @@
 # some stuff for MacOSX -----------------------------
 HOSTNAME=`./config.guess`
 
-if [ "${HOSTNAME}" = "powerpc-apple-darwin6.0" ]; then
-#compiler flags
-CC="c++"
-common_cflags="-pedantic -Wall"
-#devel_cflags="-g -Werror -fno-builtin"
-devel_cflags="-g -fno-builtin"
-optimization_cflags="-O0"
-release_cflags=$optimization_cflags
-cflags=$common_cflags
-OFLAGS="-D_MACOSX"
-#linker flags
-lflags="-x"
-#state vars
-debug="no"
-devel="no"
-parachute="no"
-static="no"
-LFS="yes"
-
-#cp flags macosx does not know about cp -d !
-cpflags=""
-
-fi
+case $HOSTNAME in
+  powerpc-apple-darwin*)
+    #compiler flags
+    CC="g++"
+    common_cflags="-pedantic -Wall"
+    #devel_cflags="-g -Werror -fno-builtin"
+    devel_cflags="-g -fno-builtin"
+    optimization_cflags="-O0"
+    release_cflags=$optimization_cflags
+    cflags=$common_cflags
+    OFLAGS="-D_MACOSX"
+    #linker flags
+    lflags="-x"
+    #state vars
+    debug="no"
+    devel="no"
+    parachute="yes"
+    static="no"
+    LFS="yes"
+    #cp flags macosx does not know about cp -d !
+    cpflags=""
+  ;;
+esac
 
 # IRIX support (thank to Eric Sokolowsky)
 if [ "${HOSTNAME}" = "mips-sgi-irix6.5" ]; then
@@ -245,7 +245,7 @@
 echo "cpflags=$cpflags" >> $makefilename
 
 cat << EOF >>$makefilename
-INSTALLDIR=\$(PREFIX)/bin
+INSTALLDIR=\$(prefix)/bin
 MANDIR=\$(manprefix)/man/man1
 MANDIRDE=\$(manprefix)/man/de/man1
 # Compilation and linker flags
@@ -322,13 +322,13 @@
 	install -d -m 755  \$(MANDIR)
 	install -d -m 755  \$(MANDIRDE)
 	install -s -m 755 mpgtx \$(INSTALLDIR)
-	cp -\$(cpflags)f mpgdemux mpgjoin mpgcat mpginfo mpgsplit tagmp3 \$(INSTALLDIR)
+	install -m 755 mpgdemux mpgjoin mpgcat mpginfo mpgsplit tagmp3 \$(INSTALLDIR)
 	install -m 644 man/mpgtx.1 man/tagmp3.1 \$(MANDIR)
 	install -m 644 man/de/mpgtx.1 man/de/tagmp3.1 \$(MANDIRDE)
 	cd man ; make
 	cd man/de ; make
-	cp -\$(cpflags)f man/mpgdemux.1 man/mpgjoin.1 man/mpgsplit.1 man/mpgcat.1 man/mpginfo.1 \$(MANDIR)
-	cp -\$(cpflags)f man/de/mpgdemux.1 man/de/mpgjoin.1 man/de/mpgsplit.1 man/de/mpgcat.1 man/de/mpginfo.1 \$(MANDIRDE)
+	install -m 644 man/mpgdemux.1 man/mpgjoin.1 man/mpgsplit.1 man/mpgcat.1 man/mpginfo.1 \$(MANDIR)
+	install -m 644 man/de/mpgdemux.1 man/de/mpgjoin.1 man/de/mpgsplit.1 man/de/mpgcat.1 man/de/mpginfo.1 \$(MANDIRDE)
 
 
 uninstall:
