diff -Naur parallel-20130722/src/Makefile.am parallel-20130722.fink/src/Makefile.am
--- parallel-20130722/src/Makefile.am	2012-03-21 22:01:19.000000000 +0100
+++ parallel-20130722.fink/src/Makefile.am	2013-07-23 07:01:56.000000000 +0200
@@ -19,22 +19,22 @@
 		--section=1 $(srcdir)/niceload.pod > $(srcdir)/niceload.1
 
 parallel.html: parallel.pod Makefile
-	pod2html $(srcdir)/parallel.pod > $(srcdir)/parallel.html
+	pod2html5.10 $(srcdir)/parallel.pod > $(srcdir)/parallel.html
 	rm -f $(srcdir)/pod2htm*
 
 # Depending on parallel.html to avoid stupid pod2html race condition
 sem.html: sem.pod Makefile parallel.html
-	pod2html $(srcdir)/sem.pod > $(srcdir)/sem.html
+	pod2html5.10 $(srcdir)/sem.pod > $(srcdir)/sem.html
 	rm -f $(srcdir)/pod2htm*
 
 # Depending on sem.html to avoid stupid pod2html race condition
 sql.html: sql Makefile sem.html
-	pod2html $(srcdir)/sql > $(srcdir)/sql.html
+	pod2html5.10 $(srcdir)/sql > $(srcdir)/sql.html
 	rm -f $(srcdir)/pod2htm*
 
 # Depending on sql.html to avoid stupid pod2html race condition
 niceload.html: niceload.pod Makefile sql.html
-	pod2html $(srcdir)/niceload.pod > $(srcdir)/niceload.html
+	pod2html5.10 $(srcdir)/niceload.pod > $(srcdir)/niceload.html
 	rm -f $(srcdir)/pod2htm*
 
 parallel.texi: parallel.pod
