diff -ru sed-4.2.1/lib/localcharset.c sed-4.2.1-patched/lib/localcharset.c
--- sed-4.2.1/lib/localcharset.c	2009-06-03 21:10:51.000000000 +0200
+++ sed-4.2.1-patched/lib/localcharset.c	2012-06-06 10:58:36.000000000 +0200
@@ -248,6 +248,7 @@
 	   "SJIS" "\0" "SHIFT_JIS" "\0"
 	   "ARMSCII-8" "\0" "ARMSCII-8" "\0"
 	   "PT154" "\0" "PT154" "\0"
+	   "US-ASCII" "\0" "ASCII" "\0"
 	 /*"ISCII-DEV" "\0" "?" "\0"*/
 	   "*" "\0" "UTF-8" "\0";
 # endif
