--- ./antiword.h	Thu Jun 20 09:16:22 2002
+++ ./antiword.h-patched.h	Tue Aug 20 10:56:27 2002
@@ -177,4 +177,8 @@
 #define ANTIWORD_DIR		"antiword"
 #define FONTNAMES_FILE		"fontnames"
+#elif defined(__macosx_fink)
+#define GLOBAL_ANTIWORD_DIR	"@PREFIX@/share/antiword"
+#define ANTIWORD_DIR		"antiword"
+#define FONTNAMES_FILE		"fontnames"
 #else	/* All others */
 #define GLOBAL_ANTIWORD_DIR	"/usr/share/antiword"
--- ./Makefile.Linux	Fri Jul  5 08:45:13 2002
+++ ./Makefile.Linux-patched	Tue Aug 20 11:51:43 2002
@@ -13,5 +13,5 @@
 LDLIBS	=
 
-CFLAGS	= -Wall -pedantic $(OPT) -D$(DB)
+CFLAGS	= $(OPT) -D$(DB)
 LDFLAGS	=
 
@@ -46,4 +46,5 @@
 global_install:	all
 	@[ `id -u` -eq 0 ] || (echo "You must be root to do this" && false)
+	mkdir -p $(GLOBAL_INSTALL_DIR)
 	cp -f $(PROGS) $(GLOBAL_INSTALL_DIR)
 	cd $(GLOBAL_INSTALL_DIR); chmod 755 $(PROGS)
@@ -52,4 +53,6 @@
 	cp -f Resources/fontnames $(GLOBAL_RESOURCES_DIR)
 	cd $(GLOBAL_RESOURCES_DIR); chmod 644 *.txt fontnames
+	mkdir -p $(DESTDIR)@PREFIX@/share/man/man1
+	cp -f Docs/*.1 $(DESTDIR)@PREFIX@/share/man/man1
 
 clean:
