diff -ruN openjade-1.3.1/include/Text.h openjade-1.3.1-patched/include/Text.h
--- openjade-1.3.1/include/Text.h	Sun May  2 11:47:05 1999
+++ openjade-1.3.1-patched/include/Text.h	Tue Jun 11 15:54:37 2002
@@ -52,6 +52,8 @@
 class SP_API Text {
 public:
   Text();
+  // Dummy Destructor
+  ~Text();
   void clear();
   void swap(Text &to);
   void addChar(Char c, const Location &);
diff -ruN openjade-1.3.1/lib/Text.cxx openjade-1.3.1-patched/lib/Text.cxx
--- openjade-1.3.1/lib/Text.cxx	Sun May  2 11:47:31 1999
+++ openjade-1.3.1-patched/lib/Text.cxx	Tue Jun 11 15:54:37 2002
@@ -17,6 +17,10 @@
 {
 }
 
+Text::~Text()
+{
+}
+
 void Text::addChar(Char c, const Location &loc)
 {
   if (items_.size() == 0
