diff -ur scribus-0.8.ori/scribus/scribusXml.cpp scribus-0.8.corr/scribus/scribusXml.cpp
--- scribus-0.8.ori/scribus/scribusXml.cpp	Sun Sep 22 13:37:21 2002
+++ scribus-0.8.corr/scribus/scribusXml.cpp	Sat Oct 12 15:27:00 2002
@@ -1092,7 +1092,7 @@
 			txnu += tmp.setNum(item->Ptext.at(kt)->xp) + " " + tmp.setNum(item->Ptext.at(kt)->yp) + " ";
 			}
 		ob.setAttribute("TEXTCOOR", txnu);
-		ob.setAttribute("NUMGROUP",item->Groups.count());
+		ob.setAttribute("NUMGROUP",(uint)item->Groups.count());
 		QString glp = "";
 		QValueStack<int>::Iterator nx;
 		for (nx = item->Groups.begin(); nx != item->Groups.end(); ++nx)
@@ -1355,7 +1355,7 @@
 			k--;
 			ob.appendChild(it);
 			}
-		ob.setAttribute("NUMGROUP",item->Groups.count());
+		ob.setAttribute("NUMGROUP",(uint)item->Groups.count());
 		QString glp = "";
 		QValueStack<int>::Iterator nx;
 		for (nx = item->Groups.begin(); nx != item->Groups.end(); ++nx)
diff -ur scribus-0.8.ori/scribus/scribusdoc.h scribus-0.8.corr/scribus/scribusdoc.h
--- scribus-0.8.ori/scribus/scribusdoc.h	Fri Sep 20 15:58:46 2002
+++ scribus-0.8.corr/scribus/scribusdoc.h	Sat Oct 12 15:27:00 2002
@@ -34,6 +34,7 @@
 #include "scfonts.h"
 #include "fpointarray.h"
 #ifdef HAVE_CMS
+        typedef void * HANDLE;
 	#include <lcms.h>
 #endif
 
