diff -ur scribus-0.7.7.ori/scribus/scribusXml.cpp scribus-0.7.7/scribus/scribusXml.cpp
--- scribus-0.7.7.ori/scribus/scribusXml.cpp	Thu Jul 25 09:23:39 2002
+++ scribus-0.7.7/scribus/scribusXml.cpp	Tue Sep  3 07:15:01 2002
@@ -1070,7 +1070,7 @@
 			txnu += tmp.setNum(item->Ptext.at(kt)->xp) + " " + tmp.setNum(item->Ptext.at(kt)->yp) + " ";
 			}
 		ob.setAttribute("TEXTCOOR", txnu);
-		ob.setAttribute("NUMGROUP",item->Groups.count());
+		ob.setAttribute("NUMGROUP",(uint)item->Groups.count());
 		QString glp = "";
 		QValueStack<int>::Iterator nx;
 		for (nx = item->Groups.begin(); nx != item->Groups.end(); ++nx)
@@ -1326,7 +1326,7 @@
 			k--;
 			ob.appendChild(it);
 			}
-		ob.setAttribute("NUMGROUP",item->Groups.count());
+		ob.setAttribute("NUMGROUP",(uint)item->Groups.count());
 		QString glp = "";
 		QValueStack<int>::Iterator nx;
 		for (nx = item->Groups.begin(); nx != item->Groups.end(); ++nx)
Only in scribus-0.7.7/scribus: scribusXml.cpp~
diff -ur scribus-0.7.7.ori/scribus/scribusXml.h scribus-0.7.7/scribus/scribusXml.h
diff -ur scribus-0.7.7.ori/scribus/scribusdoc.h scribus-0.7.7/scribus/scribusdoc.h
--- scribus-0.7.7.ori/scribus/scribusdoc.h	Thu Jul 25 08:09:32 2002
+++ scribus-0.7.7/scribus/scribusdoc.h	Tue Sep  3 06:55:03 2002
@@ -34,6 +34,7 @@
 #include "scfonts.h"
 #include "fpointarray.h"
 #ifdef HAVE_CMS
+        typedef void * HANDLE;
 	#include <lcms.h>
 #endif
 
