diff -ur scribus-0.8.ori/scribus/scribusXml.cpp scribus-0.8.corr/scribus/scribusXml.cpp
--- scribus-0.8.ori/scribus/scribusXml.cpp	Sun Sep 22 13:37:21 2002
+++ scribus-0.8.corr/scribus/scribusXml.cpp	Sat Oct 12 15:27:00 2002
@@ -1092,7 +1092,7 @@
 			txnu += tmp.setNum(item->Ptext.at(kt)->xp) + " " + tmp.setNum(item->Ptext.at(kt)->yp) + " ";
 			}
 		ob.setAttribute("TEXTCOOR", txnu);
-		ob.setAttribute("NUMGROUP",item->Groups.count());
+		ob.setAttribute("NUMGROUP",(uint)item->Groups.count());
 		QString glp = "";
 		QValueStack<int>::Iterator nx;
 		for (nx = item->Groups.begin(); nx != item->Groups.end(); ++nx)
@@ -1355,7 +1355,7 @@
 			k--;
 			ob.appendChild(it);
 			}
-		ob.setAttribute("NUMGROUP",item->Groups.count());
+		ob.setAttribute("NUMGROUP",(uint)item->Groups.count());
 		QString glp = "";
 		QValueStack<int>::Iterator nx;
 		for (nx = item->Groups.begin(); nx != item->Groups.end(); ++nx)
diff -ur scribus-0.8.ori/scribus/scribusdoc.h scribus-0.8.corr/scribus/scribusdoc.h
--- scribus-0.8.ori/scribus/scribusdoc.h	Fri Sep 20 15:58:46 2002
+++ scribus-0.8.corr/scribus/scribusdoc.h	Sat Oct 12 15:27:00 2002
@@ -34,6 +34,7 @@
 #include "scfonts.h"
 #include "fpointarray.h"
 #ifdef HAVE_CMS
+        typedef void * HANDLE;
 	#include <lcms.h>
 #endif
 
diff -ur scribus-0.8/scribus/libabout/Makefile.in scribus-0.8.corr/scribus/libabout/Makefile.in
--- scribus-0.8/scribus/libabout/Makefile.in	Sat Sep 28 12:30:30 2002
+++ scribus-0.8.corr/scribus/libabout/Makefile.in	Sun Apr  6 00:26:38 2003
@@ -227,13 +227,13 @@
 
 libdir = $(prefix)/share/scribus/libs
 ####### kdevelop will overwrite this part!!! (begin)##########
-lib_LTLIBRARIES = libabout.la
+lib_LTLIBRARIES = libabout.so
 
 INCLUDES = $(all_includes)
 
-LDFLAGS = -s        -export-dynamic 
+LDFLAGS = -bundle -flat_namespace -undefined suppress
 
-libabout_la_LDFLAGS = -version-info 0:0:0
+libabout_la_LDFLAGS = 
 
 #>- libabout_la_METASOURCES = AUTO
 
@@ -360,7 +360,7 @@
 
 maintainer-clean-libtool:
 
-libabout.la: $(libabout_la_OBJECTS) $(libabout_la_DEPENDENCIES)
+libabout.so: $(libabout_la_OBJECTS) $(libabout_la_DEPENDENCIES)
 	$(CXXLINK) -rpath $(libdir) $(libabout_la_LDFLAGS) $(libabout_la_OBJECTS) $(libabout_la_LIBADD) $(LIBS)
 .cpp.o:
 	$(CXXCOMPILE) -c $<
diff -ur scribus-0.8/scribus/libpdf/Makefile.in scribus-0.8.corr/scribus/libpdf/Makefile.in
--- scribus-0.8/scribus/libpdf/Makefile.in	Sat Sep 28 12:30:31 2002
+++ scribus-0.8.corr/scribus/libpdf/Makefile.in	Sun Apr  6 00:30:10 2003
@@ -228,13 +228,13 @@
 libdir = $(prefix)/share/scribus/libs
 CPPFLAGS = -I$(top_builddir)/scribus
 ####### kdevelop will overwrite this part!!! (begin)##########
-lib_LTLIBRARIES = libpdf.la
+lib_LTLIBRARIES = libpdf.so
 
 INCLUDES = $(all_includes)
 
-LDFLAGS = -s        -export-dynamic 
+LDFLAGS = -bundle -flat_namespace -undefined suppress
 
-libpdf_la_LDFLAGS = -version-info 0:0:0
+libpdf_la_LDFLAGS =
 
 #>- libpdf_la_METASOURCES = AUTO
 
@@ -360,7 +360,7 @@
 
 maintainer-clean-libtool:
 
-libpdf.la: $(libpdf_la_OBJECTS) $(libpdf_la_DEPENDENCIES)
+libpdf.so: $(libpdf_la_OBJECTS) $(libpdf_la_DEPENDENCIES)
 	$(CXXLINK) -rpath $(libdir) $(libpdf_la_LDFLAGS) $(libpdf_la_OBJECTS) $(libpdf_la_LIBADD) $(LIBS)
 .cpp.o:
 	$(CXXCOMPILE) -c $<
diff -ur scribus-0.8/scribus/libpostscript/Makefile.in scribus-0.8.corr/scribus/libpostscript/Makefile.in
--- scribus-0.8/scribus/libpostscript/Makefile.in	Sat Sep 28 12:30:31 2002
+++ scribus-0.8.corr/scribus/libpostscript/Makefile.in	Sun Apr  6 00:31:47 2003
@@ -228,13 +228,13 @@
 libdir = $(prefix)/share/scribus/libs
 CPPFLAGS = -I$(top_builddir)/scribus
 ####### kdevelop will overwrite this part!!! (begin)##########
-lib_LTLIBRARIES = libpostscript.la
+lib_LTLIBRARIES = libpostscript.so
 
 INCLUDES = $(all_includes)
 
-LDFLAGS = -s        -export-dynamic 
+LDFLAGS = -bundle -flat_namespace -undefined suppress
 
-libpostscript_la_LDFLAGS = -version-info 0:0:0
+libpostscript_la_LDFLAGS = 
 
 #>- libpostscript_la_METASOURCES = AUTO
 
@@ -360,7 +360,7 @@
 
 maintainer-clean-libtool:
 
-libpostscript.la: $(libpostscript_la_OBJECTS) $(libpostscript_la_DEPENDENCIES)
+libpostscript.so: $(libpostscript_la_OBJECTS) $(libpostscript_la_DEPENDENCIES)
 	$(CXXLINK) -rpath $(libdir) $(libpostscript_la_LDFLAGS) $(libpostscript_la_OBJECTS) $(libpostscript_la_LIBADD) $(LIBS)
 .cpp.o:
 	$(CXXCOMPILE) -c $<
diff -ur scribus-0.8/scribus/libprefs/Makefile.in scribus-0.8.corr/scribus/libprefs/Makefile.in
--- scribus-0.8/scribus/libprefs/Makefile.in	Sat Sep 28 12:30:31 2002
+++ scribus-0.8.corr/scribus/libprefs/Makefile.in	Sun Apr  6 00:33:55 2003
@@ -228,13 +228,13 @@
 libdir = $(prefix)/share/scribus/libs
 CPPFLAGS = -I$(top_builddir)/scribus
 ####### kdevelop will overwrite this part!!! (begin)##########
-lib_LTLIBRARIES = libpreferences.la
+lib_LTLIBRARIES = libpreferences.so
 
 INCLUDES = $(all_includes)
 
-LDFLAGS = -s        -export-dynamic 
+LDFLAGS = -bundle -flat_namespace -undefined suppress
 
-libpreferences_la_LDFLAGS = -version-info 0:0:0
+libpreferences_la_LDFLAGS = 
 
 #>- libpreferences_la_METASOURCES = AUTO
 
@@ -360,7 +360,7 @@
 
 maintainer-clean-libtool:
 
-libpreferences.la: $(libpreferences_la_OBJECTS) $(libpreferences_la_DEPENDENCIES)
+libpreferences.so: $(libpreferences_la_OBJECTS) $(libpreferences_la_DEPENDENCIES)
 	$(CXXLINK) -rpath $(libdir) $(libpreferences_la_LDFLAGS) $(libpreferences_la_OBJECTS) $(libpreferences_la_LIBADD) $(LIBS)
 .cpp.o:
 	$(CXXCOMPILE) -c $<
diff -ur scribus-0.8/scribus/plugins/libchar/Makefile.in scribus-0.8.corr/scribus/plugins/libchar/Makefile.in
--- scribus-0.8/scribus/plugins/libchar/Makefile.in	Sat Sep 28 12:30:31 2002
+++ scribus-0.8.corr/scribus/plugins/libchar/Makefile.in	Sun Apr  6 00:35:18 2003
@@ -228,13 +228,13 @@
 libdir = $(prefix)/share/scribus/plugins
 CPPFLAGS = -I$(top_builddir)/scribus
 ####### kdevelop will overwrite this part!!! (begin)##########
-lib_LTLIBRARIES = libcharselect.la
+lib_LTLIBRARIES = libcharselect.so
 
 INCLUDES = $(all_includes)
 
-LDFLAGS = -s        -export-dynamic 
+LDFLAGS = -bundle -flat_namespace -undefined suppress
 
-libcharselect_la_LDFLAGS = -version-info 0:0:0
+libcharselect_la_LDFLAGS = 
 
 #>- libcharselect_la_METASOURCES = AUTO
 
@@ -365,7 +365,7 @@
 
 maintainer-clean-libtool:
 
-libcharselect.la: $(libcharselect_la_OBJECTS) $(libcharselect_la_DEPENDENCIES)
+libcharselect.so: $(libcharselect_la_OBJECTS) $(libcharselect_la_DEPENDENCIES)
 	$(CXXLINK) -rpath $(libdir) $(libcharselect_la_LDFLAGS) $(libcharselect_la_OBJECTS) $(libcharselect_la_LIBADD) $(LIBS)
 .cpp.o:
 	$(CXXCOMPILE) -c $<
diff -ur scribus-0.8/scribus/mpalette.cpp scribus-0.8.02/scribus/mpalette.cpp
--- scribus-0.8/scribus/mpalette.cpp	Sat Sep 14 12:26:16 2002
+++ scribus-0.8.02/scribus/mpalette.cpp	Sun Oct 27 08:15:56 2002
@@ -430,8 +430,8 @@
     connect(FrameScale, SIGNAL(clicked()), this, SLOT(ChangeScaling()));
     connect(Aspect, SIGNAL(clicked()), this, SLOT(ChangeScaling()));
   	connect(this, SIGNAL(placeChanged(QDockWindow::Place)), this, SLOT(Docken(QDockWindow::Place)));
-  	setCloseMode(Always);
-		undock();
+		pa->moveDockWindow(this, DockTornOff);
+		setCloseMode(Always);
 		resize(sizeHint());
 		setResizeEnabled(true);
     Fram->raiseWidget(4);
diff -ur scribus-0.8/scribus/scribus.cpp scribus-0.8.02/scribus/scribus.cpp
--- scribus-0.8/scribus/scribus.cpp	Tue Sep 24 22:09:45 2002
+++ scribus-0.8.02/scribus/scribus.cpp	Sat Oct 26 16:37:53 2002
@@ -4686,7 +4686,7 @@
 			pda.Zeiger = 0;
 			pda.Datei = "";
 			pda.Typ = 0;
-			if (DLLName(d[dc], &nam, &ty, pda.Zeiger))
+			if (DLLName(d[dc], &nam, &ty, &pda.Zeiger))
 				{
 				if (ty == 1)
 					extraMenu->insertItem(tr(nam));
@@ -4698,7 +4698,7 @@
 					helpMenu->insertItem(tr(nam));
 				pda.Datei = d[dc];
 				pda.Typ = ty;
-				PluginMap[tr(nam)] = pda;
+				PluginMap.insert(tr(nam), pda);
 				}
 			}
 	 	connect(extraMenu, SIGNAL(activated(int)), this, SLOT(RunPlug(int)));
@@ -4742,7 +4742,7 @@
 		dlclose(mo);
 }
 
-bool ScribusApp::DLLName(QString name, QString *PName, int *typ, void *Zeig)
+bool ScribusApp::DLLName(QString name, QString *PName, int *typ, void **Zeig)
 {
 	void *mo;
 	char *error;
@@ -4773,7 +4773,7 @@
 		return false;
 		}
 	*typ = (*demo1)();
-	Zeig = mo;
+	*Zeig = mo;
 	if (*typ != 4)
 		dlclose(mo);
 	else
diff -ur scribus-0.8/scribus/scribus.h scribus-0.8.02/scribus/scribus.h
--- scribus-0.8/scribus/scribus.h	Tue Sep 24 22:09:58 2002
+++ scribus-0.8.02/scribus/scribus.h	Sat Oct 26 16:34:00 2002
@@ -98,7 +98,7 @@
     void UpdateRecent(QString fn);
     void InitPlugs();
 		void FinalizePlugs();
-    bool DLLName(QString name, QString *PName, int *typ, void *Zeig);
+    bool DLLName(QString name, QString *PName, int *typ, void **Zeig);
     void CallDLL(QString name);
     PSLib* getPSDriver(bool psart, SCFonts &AllFonts, QMap<QString,QFont> DocFonts, CListe DocColors, bool pdf);
     void closePSDriver();
diff -ur scribus-0.8/scribus/fmitem.cpp scribus-0.8n/scribus/fmitem.cpp
--- scribus-0.8/scribus/fmitem.cpp	Fri Nov 30 21:50:57 2001
+++ scribus-0.8n/scribus/fmitem.cpp	Fri Oct 11 11:16:59 2002
@@ -17,11 +17,13 @@
 
 #include "fmitem.h"
 #include <qfontmetrics.h>
+#include <qapplication.h>
 
 FmItem::FmItem(const QString& s, const QFont& f)
 {
 	string = s;
 	font = f;
+	font.setPointSize(qApp->font().pointSize());
 }
 FmItem::~FmItem()
 {
diff -ur scribus-0.8/scribus/scfonts.cpp scribus-0.8n/scribus/scfonts.cpp
--- scribus-0.8/scribus/scfonts.cpp	Sun Sep 22 10:54:38 2002
+++ scribus-0.8n/scribus/scfonts.cpp	Sat Oct 12 19:41:57 2002
@@ -443,6 +443,7 @@
 					{
 						t->Font.setPointSize(12);
 						t->Font.setRawName(tmp);
+						t->Font.setPointSize(12);
 						t->FontEnc = FoEnc;
 						insert(SCName,t);
 						if(t->HasMetrics)
