--- xpdf-1.00/doc/sample-xpdfrc	Fri Feb  1 16:15:45 2002
+++ xpdf-1.00/doc/sample-xpdfrc	Fri Mar  1 17:06:04 2002
@@ -46,20 +46,20 @@
 # this, but you'll need to adjust this to point to wherever
 # ghostscript is installed on your system.
 
-#displayFontT1 Times-Roman		/usr/local/share/ghostscript/fonts/n021003l.pfb
-#displayFontT1 Times-Italic		/usr/local/share/ghostscript/fonts/n021023l.pfb
-#displayFontT1 Times-Bold		/usr/local/share/ghostscript/fonts/n021004l.pfb
-#displayFontT1 Times-BoldItalic		/usr/local/share/ghostscript/fonts/n021024l.pfb
-#displayFontT1 Helvetica		/usr/local/share/ghostscript/fonts/n019003l.pfb
-#displayFontT1 Helvetica-Oblique	/usr/local/share/ghostscript/fonts/n019023l.pfb
-#displayFontT1 Helvetica-Bold		/usr/local/share/ghostscript/fonts/n019004l.pfb
-#displayFontT1 Helvetica-BoldOblique	/usr/local/share/ghostscript/fonts/n019024l.pfb
-#displayFontT1 Courier			/usr/local/share/ghostscript/fonts/n022003l.pfb
-#displayFontT1 Courier-Oblique		/usr/local/share/ghostscript/fonts/n022023l.pfb
-#displayFontT1 Courier-Bold		/usr/local/share/ghostscript/fonts/n022004l.pfb
-#displayFontT1 Courier-BoldOblique	/usr/local/share/ghostscript/fonts/n022024l.pfb
-#displayFontT1 Symbol			/usr/local/share/ghostscript/fonts/s050000l.pfb
-#displayFontT1 ZapfDingbats		/usr/local/share/ghostscript/fonts/d050000l.pfb
+displayFontT1 Times-Roman		/sw/share/ghostscript/fonts/n021003l.pfb
+displayFontT1 Times-Italic		/sw/share/ghostscript/fonts/n021023l.pfb
+displayFontT1 Times-Bold		/sw/share/ghostscript/fonts/n021004l.pfb
+displayFontT1 Times-BoldItalic		/sw/share/ghostscript/fonts/n021024l.pfb
+displayFontT1 Helvetica		/sw/share/ghostscript/fonts/n019003l.pfb
+displayFontT1 Helvetica-Oblique	/sw/share/ghostscript/fonts/n019023l.pfb
+displayFontT1 Helvetica-Bold		/sw/share/ghostscript/fonts/n019004l.pfb
+displayFontT1 Helvetica-BoldOblique	/sw/share/ghostscript/fonts/n019024l.pfb
+displayFontT1 Courier			/sw/share/ghostscript/fonts/n022003l.pfb
+displayFontT1 Courier-Oblique		/sw/share/ghostscript/fonts/n022023l.pfb
+displayFontT1 Courier-Bold		/sw/share/ghostscript/fonts/n022004l.pfb
+displayFontT1 Courier-BoldOblique	/sw/share/ghostscript/fonts/n022024l.pfb
+displayFontT1 Symbol			/sw/share/ghostscript/fonts/s050000l.pfb
+displayFontT1 ZapfDingbats		/sw/share/ghostscript/fonts/d050000l.pfb
 
 # If you need to display PDF files that refer to non-embedded fonts,
 # you should add one or more fontDir options to point to the
