--- ProofGeneral/Makefile.orig	2005-09-21 14:26:45.000000000 -0400
+++ ProofGeneral/Makefile	2007-08-30 15:05:02.000000000 -0400
@@ -22,6 +22,9 @@
 # Set this to "emacs" or "xemacs" according to your version of Emacs.
 # NB: this is also used to set default install path names below.
 EMACS=$(shell if [ -z "`which emacs`" ]; then echo xemacs; else echo emacs; fi)
+
+# Set this to the path to the install-info program, if necessary.
+INSTALL_INFO=/sbin/install-info
 
 # We default to /usr rather than /usr/local because installs of
 # desktop and doc files under /usr/local are unlikely to work with
@@ -215,8 +218,8 @@
 	cp -pf doc/proofgeneral.1 ${MANDIR}
 	mkdir -p ${INFODIR}
 	cp -pf doc/*.info ${INFODIR}
-	/sbin/install-info ${INFODIR}/ProofGeneral.info* ${INFODIR}/dir
-	/sbin/install-info ${INFODIR}/PG-adapting.info* ${INFODIR}/dir
+	${INSTALL_INFO} ${INFODIR}/ProofGeneral.info* ${INFODIR}/dir
+	${INSTALL_INFO} ${INFODIR}/PG-adapting.info* ${INFODIR}/dir
 	mkdir -p ${DOCDIR}
 	for f in ${DOC_FILES}; do cp -pf $$f ${DOCDIR}; done
 	for f in ${DOC_EXAMPLES}; do mkdir -p ${DOCDIR}/`dirname $$f`; cp -pf $$f ${DOCDIR}/$$f; done
