diff -Naur findutils-4.1.20/findutils.cron findutils-patched/findutils.cron
--- findutils-4.1.20/findutils.cron	Wed Dec 31 19:00:00 1969
+++ findutils-patched/findutils.cron	Tue Dec  2 02:28:21 2003
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+test -x @PREFIX@/bin/updatedb || exit 0
+cd / && @PREFIX@/bin/updatedb
+
+exit 0
diff -Naur findutils-4.1.20/gnulib/lib/regex.c findutils-patched/gnulib/lib/regex.c
--- findutils-4.1.20/gnulib/lib/regex.c	Sun Nov 24 19:17:33 2002
+++ findutils-patched/gnulib/lib/regex.c	Tue Dec  2 02:28:57 2003
@@ -8081,7 +8081,7 @@
 
 /* POSIX.2 functions.  Don't define these for Emacs.  */
 
-#ifndef emacs
+#if !defined(emacs) && !defined(FINK)
 
 /* regcomp takes a regular expression as a string and compiles it.
 
