diff -Nurd -x'*~' umoria.orig/source/config.h umoria/source/config.h
--- umoria.orig/source/config.h	1994-07-21 21:47:13.000000000 -0400
+++ umoria/source/config.h	2006-02-03 12:40:27.000000000 -0500
@@ -14,7 +14,7 @@
 
 /* Person to bother if something goes wrong.  */
 /* Recompile files.c and misc2.c if this changes.  */
-#define WIZARD	"David Grabiner"
+#define WIZARD	"James Gibbs <jimgibbs@users.sourceforge.net>"
 /* The wizard password and wizard uid are no longer used.  */
 
 
@@ -38,7 +38,8 @@
    moria will be linked with -lbsd.  */
 
 /* If you are compiling on a SYS V version of UNIX, define this.  */
-/* #define SYS_V */
+#define SYS_V
+#define unix
 
 /* If you are compiling on a SYS III version of UNIX, define this.
    The SYS_III support may not be complete.  I do not know if this works.  */
@@ -199,8 +200,9 @@
 #else
 
 /* This must be unix; change MORIA_LIB as appropriate.  */
+/* 
 #define MORIA_SAV	"moria.save"
-#define MORIA_LIB(xxx)  "/home/math/grabiner/moria/files/xxx"
+#define MORIA_LIB(xxx)  "@PREFIX@/var/games/moria/files/xxx"
 #define MORIA_HOU	MORIA_LIB(hours)
 #define MORIA_MOR	MORIA_LIB(news)
 #define MORIA_TOP	MORIA_LIB(scores)
@@ -210,6 +212,29 @@
 #define MORIA_OWIZ_HELP	MORIA_LIB(owizcmds.hlp)
 #define MORIA_WELCOME	MORIA_LIB(welcome.hlp)
 #define MORIA_VER	MORIA_LIB(version.hlp)
+*/ 
+
+/* This must be unix; change MORIA_LIB as appropriate.  */
+#define MORIA_SAV	"moria.save"
+#define MORIA_LIB(xxx)  "@PREFIX@/var/games/moria/files/xxx"
+#define MORIA_HOU       "@PREFIX@/var/games/moria/files/hours"
+#define MORIA_MOR       "@PREFIX@/var/games/moria/files/news"
+#define MORIA_TOP       "@PREFIX@/var/games/moria/files/scores"
+#define MORIA_HELP      "@PREFIX@/var/games/moria/files/roglcmds.hlp"
+#define MORIA_ORIG_HELP "@PREFIX@/var/games/moria/files/origcmds.hlp"
+#define MORIA_WIZ_HELP  "@PREFIX@/var/games/moria/files/rwizcmds.hlp"
+#define MORIA_OWIZ_HELP "@PREFIX@/var/games/moria/files/owizcmds.hlp"
+#define MORIA_WELCOME   "@PREFIX@/var/games/moria/files/welcome.hlp"
+#define MORIA_VER       "@PREFIX@/var/games/moria/files/version.hlp"
+
+/* missing headers from various files */
+#include <unistd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 
 #endif
 #endif
diff -Nurd -x'*~' umoria.orig/source/death.c umoria/source/death.c
--- umoria.orig/source/death.c	1994-07-21 21:47:16.000000000 -0400
+++ umoria/source/death.c	2006-02-03 12:29:48.000000000 -0500
@@ -165,7 +165,7 @@
 }
 
 
-#ifndef __TURBOC__
+#if 0
 #if (defined(USG) || defined(atarist) || defined(HPUX)) && !defined(VMS)
 #if !defined(AMIGA) && !defined(MAC) && !defined(ATARIST_TC)
 
diff -Nurd -x'*~' umoria.orig/source/io.c umoria/source/io.c
--- umoria.orig/source/io.c	1994-07-21 21:47:26.000000000 -0400
+++ umoria/source/io.c	2006-02-03 03:28:12.000000000 -0500
@@ -6,6 +6,11 @@
    not for profit purposes provided that this copyright and statement are
    included in all such copies. */
 
+#define termio termios
+#define TCGETA TIOCGETA
+#define TCSETA TIOCSETA
+
+#include <termios.h>
 #include <stdio.h>
 #ifndef STDIO_LOADED
 #define STDIO_LOADED
