diff -Nurd -x'*~' gnome-utils-2.28.3.orig/baobab/src/baobab.c gnome-utils-2.28.3/baobab/src/baobab.c
--- gnome-utils-2.28.3.orig/baobab/src/baobab.c	2010-01-16 11:36:16.000000000 -0500
+++ gnome-utils-2.28.3/baobab/src/baobab.c	2010-04-25 04:53:51.000000000 -0400
@@ -20,6 +20,8 @@
  * Boston, MA  02110-1301  USA
  */
  
+#define NO_EXTERN_BAOBAB 1
+
 #include <config.h>
 
 #include <gtk/gtk.h>
diff -Nurd -x'*~' gnome-utils-2.28.3.orig/baobab/src/baobab.h gnome-utils-2.28.3/baobab/src/baobab.h
--- gnome-utils-2.28.3.orig/baobab/src/baobab.h	2009-10-27 09:16:36.000000000 -0400
+++ gnome-utils-2.28.3/baobab/src/baobab.h	2010-04-25 04:53:42.000000000 -0400
@@ -103,9 +103,15 @@
 };
 
 /* globals */
+#ifdef NO_EXTERN_BAOBAB
 gboolean g_noactivescans;
 baobab_application baobab;
 baobab_fs g_fs;
+#else
+extern gboolean g_noactivescans;
+extern baobab_application baobab;
+extern baobab_fs g_fs;
+#endif
 
 void baobab_set_busy (gboolean busy);
 void baobab_scan_location (GFile *);
diff -Nurd -x'*~' gnome-utils-2.28.3.orig/configure gnome-utils-2.28.3/configure
--- gnome-utils-2.28.3.orig/configure	2010-01-16 12:13:41.000000000 -0500
+++ gnome-utils-2.28.3/configure	2010-04-25 04:48:47.000000000 -0400
@@ -23977,6 +23977,7 @@
    sed -e "/^#/d" -e "/^\$/d" -e "s,.*,	$posrcprefix& \\\\," -e "\$s/\(.*\) \\\\/\1/" \
 	< $srcdir/po/POTFILES.in > po/POTFILES
 
+GNOME_UTILS_LIBS="$GNOME_UTILS_LIBS $INTLLIBS"
 
 # AM_GLIB_GNU_GETTEXT above substs $DATADIRNAME
 # this is the directory where the *.{mo,gmo} files are installed
diff -Nurd -x'*~' gnome-utils-2.28.3.orig/gnome-dictionary/src/Makefile.in gnome-utils-2.28.3/gnome-dictionary/src/Makefile.in
--- gnome-utils-2.28.3.orig/gnome-dictionary/src/Makefile.in	2010-01-16 12:13:44.000000000 -0500
+++ gnome-utils-2.28.3/gnome-dictionary/src/Makefile.in	2010-04-25 04:50:55.000000000 -0400
@@ -319,6 +319,10 @@
 	-DUIDATADIR=\""$(datadir)/gnome-2.0/ui"\" 	\
 	-DPKGDATADIR=\""$(datadir)/gnome-dictionary"\" 	\
 	-DGNOMELOCALEDIR=\""$(prefix)/$(DATADIRNAME)/locale"\" \
+	-I$(top_builddir)/gnome-dictionary \
+	-I$(top_srcdir)/gnome-dictionary \
+	$(GNOME_UTILS_CFLAGS) \
+	$(APPLET_CFLAGS) \
 	$(NULL)
 
 gnome_dictionary_SOURCES = \
@@ -342,9 +346,6 @@
 	$(NULL)
 
 gnome_dictionary_CFLAGS = \
-	-I$(top_builddir)/gnome-dictionary \
-	-I$(top_srcdir)/gnome-dictionary \
-	$(GNOME_UTILS_CFLAGS) \
 	$(NULL)
 
 gnome_dictionary_LDADD = \
@@ -372,10 +373,6 @@
 	$(NULL)
 
 gnome_dictionary_applet_CFLAGS = \
-	-I$(top_builddir)/gnome-dictionary \
-	-I$(top_srcdir)/gnome-dictionary \
-	$(GNOME_UTILS_CFLAGS) \
-	$(APPLET_CFLAGS) \
 	$(NULL)
 
 gnome_dictionary_applet_LDADD = \
diff -Nurd -x'*~' gnome-utils-2.28.3.orig/gnome-screenshot/Makefile.in gnome-utils-2.28.3/gnome-screenshot/Makefile.in
--- gnome-utils-2.28.3.orig/gnome-screenshot/Makefile.in	2010-01-16 12:13:44.000000000 -0500
+++ gnome-utils-2.28.3/gnome-screenshot/Makefile.in	2010-04-25 04:51:50.000000000 -0400
@@ -338,7 +338,7 @@
 
 gnome_screenshot_LDFLAGS = -export-dynamic
 gnome_screenshot_LDADD = \
-	$(XSHAPE_LIBS)			\
+	$(XSHAPE_LIBS) $(X_LIBS)	\
 	$(GNOME_UTILS_LIBS)		\
 	$(LIBCANBERRA_GTK_LIBS)		\
 	$(GTHREAD_LIBS)			\
diff -Nurd -x'*~' gnome-utils-2.28.3.orig/gnome-screenshot/screenshot-utils.c gnome-utils-2.28.3/gnome-screenshot/screenshot-utils.c
--- gnome-utils-2.28.3.orig/gnome-screenshot/screenshot-utils.c	2009-10-27 09:16:36.000000000 -0400
+++ gnome-utils-2.28.3/gnome-screenshot/screenshot-utils.c	2010-04-25 04:52:30.000000000 -0400
@@ -26,6 +26,8 @@
 #include <glib.h>
 #include <glib/gi18n.h>
 
+#define XK_MISCELLANY
+#include <X11/keysymdef.h>
 #ifdef HAVE_X11_EXTENSIONS_SHAPE_H
 #include <X11/extensions/shape.h>
 #endif
diff -Nurd -x'*~' gnome-utils-2.28.3.orig/libeggsmclient/Makefile.in gnome-utils-2.28.3/libeggsmclient/Makefile.in
--- gnome-utils-2.28.3.orig/libeggsmclient/Makefile.in	2010-01-16 12:13:45.000000000 -0500
+++ gnome-utils-2.28.3/libeggsmclient/Makefile.in	2010-04-25 06:12:26.000000000 -0400
@@ -283,7 +283,7 @@
 
 noinst_LTLIBRARIES = libeggsmclient.la
 libeggsmclient_la_LIBADD = $(LIBEGGSMCLIENT_LIBS)           \
-                           -lSM -lICE
+                           $(X_LIBS) -lSM -lICE
 
 libeggsmclient_la_CFLAGS = $(LIBEGGSMCLIENT_CFLAGS)         \
                            -DEGG_SM_CLIENT_BACKEND_XSMP     \
