diff -Nurd -x'*~' gnome-terminal-2.28.2.orig/src/Makefile.in gnome-terminal-2.28.2/src/Makefile.in
--- gnome-terminal-2.28.2.orig/src/Makefile.in	2009-12-14 15:23:46.000000000 -0500
+++ gnome-terminal-2.28.2/src/Makefile.in	2010-02-22 03:59:57.000000000 -0500
@@ -411,11 +411,11 @@
 	-DGDK_DISABLE_SINGLE_INCLUDES \
 	-DGDK_PIXBUF_DISABLE_SINGLE_INCLUDES \
 	-DGTK_DISABLE_SINGLE_INCLUDES $(DISABLE_DEPRECATED) \
-	$(AM_CPPFLAGS) $(am__append_5)
-gnome_terminal_CFLAGS = $(TERM_CFLAGS) $(WARN_CFLAGS) $(AM_CFLAGS) \
+	$(TERM_CFLAGS) $(AM_CPPFLAGS) $(am__append_5)
+gnome_terminal_CFLAGS = $(WARN_CFLAGS) $(AM_CFLAGS) \
 	$(am__append_2)
 gnome_terminal_LDFLAGS = 
-gnome_terminal_LDADD = skey/libskey.la $(TERM_LIBS) $(am__append_3)
+gnome_terminal_LDADD = skey/libskey.la $(TERM_LIBS) $(am__append_3) $(INTLLIBS)
 TYPES_H_FILES = \
 	terminal-profile.h \
 	$(NULL)
diff -Nurd -x'*~' gnome-terminal-2.28.2.orig/src/skey/Makefile.in gnome-terminal-2.28.2/src/skey/Makefile.in
--- gnome-terminal-2.28.2.orig/src/skey/Makefile.in	2009-12-14 15:23:46.000000000 -0500
+++ gnome-terminal-2.28.2/src/skey/Makefile.in	2010-02-22 03:48:21.000000000 -0500
@@ -278,10 +278,10 @@
 	-I$(srcdir)/.. \
 	-DG_DISABLE_SINGLE_INCLUDES \
 	$(DISABLE_DEPRECATED) \
+	$(TERM_CFLAGS) \
 	$(AM_CPPFLAGS)
 
 libskey_la_CFLAGS = \
-	$(TERM_CFLAGS) \
 	$(WARN_CFLAGS) \
 	$(AM_CFLASG)
 
