diff -Nurd -x'*~' libgksu-2.0.12.orig/configure.ac libgksu-2.0.12/configure.ac
--- libgksu-2.0.12.orig/configure.ac	2009-06-29 13:55:08.000000000 -0400
+++ libgksu-2.0.12/configure.ac	2010-04-29 04:21:51.000000000 -0400
@@ -42,7 +42,8 @@
 
 PKG_CHECK_MODULES(LIBGKSU, [gtk+-2.0 >= 2.4.0, gconf-2.0, libstartup-notification-1.0, gnome-keyring-1, libgtop-2.0])
 PKG_CHECK_MODULES(GKSU_PROPERTIES, [gtk+-2.0 >= 2.12, gconf-2.0])
-
+PKG_CHECK_MODULES(LIBGKSUUI, [gtk+-2.0 >= 2.4.0, gconf-2.0, glib-2.0])
+PKG_CHECK_MODULES(GTKDOC_GOBJECT, [gobject-2.0])
 # Checks for library functions.
 ALL_LINGUAS="ca cs da de es eu fr hu it ko lt pl pt_BR ro ru sk sv nb nl th zh_CN"
 
diff -Nurd -x'*~' libgksu-2.0.12.orig/docs/Makefile.am libgksu-2.0.12/docs/Makefile.am
--- libgksu-2.0.12.orig/docs/Makefile.am	2009-05-09 22:04:14.000000000 -0400
+++ libgksu-2.0.12/docs/Makefile.am	2010-04-29 04:22:54.000000000 -0400
@@ -48,8 +48,8 @@
 
 # CFLAGS and LDFLAGS for compiling scan program. Only needed if your app/lib
 # contains GtkObjects/GObjects and you want to document signals and properties.
-GTKDOC_CFLAGS = -I../libgksu `pkg-config --cflags gobject-2.0`
-GTKDOC_LIBS = ../libgksu/.libs/libgksu-1.2.a `pkg-config --libs gobject-2.0`
+GTKDOC_CFLAGS = -I../libgksu $(GTKDOC_GOBJECT_CFLAGS)
+GTKDOC_LIBS = ../libgksu/.libs/libgksu-1.2.a $(GTKDOC_GOBJECT_LIBS)
 
 # GTKDOC_CC=$(LIBTOOL) --mode=compile $(CC)
 # GTKDOC_LD=$(LIBTOOL) --mode=link $(CC)
diff -Nurd -x'*~' libgksu-2.0.12.orig/gksu-properties/Makefile.am libgksu-2.0.12/gksu-properties/Makefile.am
--- libgksu-2.0.12.orig/gksu-properties/Makefile.am	2009-06-29 13:55:26.000000000 -0400
+++ libgksu-2.0.12/gksu-properties/Makefile.am	2010-04-29 04:56:01.000000000 -0400
@@ -1,9 +1,10 @@
-AM_CFLAGS = -g -O2 -Wall -Wl,--export-dynamic
+AM_CFLAGS = -g -O2 -Wall
 INCLUDES = ${GKSU_PROPERTIES_CFLAGS}
 AM_CPPFLAGS = -DLOCALEDIR=\"$(datadir)/locale\" -DDATA_DIR=\"$(datadir)\" -DPREFIX=\"$(prefix)\"
 
 bin_PROGRAMS = gksu-properties
-gksu_properties_LDFLAGS = ${GKSU_PROPERTIES_LIBS}
+gksu_properties_LDFLAGS = -export-dynamic
+gksu_properties_LDADD = ${GKSU_PROPERTIES_LIBS} $(INTLLIBS)
 gksu_properties_SOURCES = gksu-properties.c
 
 uidir = ${prefix}/share/${PACKAGE}
diff -Nurd -x'*~' libgksu-2.0.12.orig/libgksu/Makefile.am libgksu-2.0.12/libgksu/Makefile.am
--- libgksu-2.0.12.orig/libgksu/Makefile.am	2008-05-09 13:39:55.000000000 -0400
+++ libgksu-2.0.12/libgksu/Makefile.am	2010-04-29 04:44:19.000000000 -0400
@@ -8,8 +8,8 @@
 # major -> breaks backward compatibility (changes to existing ABI)
 # minor -> keeps compatibility (additions to the API)
 # micro -> no change to the API/ABI
-libgksu2_la_LIBADD = ../libgksuui/libgksuui1.0.la
-libgksu2_la_LDFLAGS = -version-info 0:2:0 -Wl,-O1 -lutil ${LIBGKSU_LIBS}
+libgksu2_la_LIBADD = ../libgksuui/libgksuui1.0.la ${LIBGKSU_LIBS}
+libgksu2_la_LDFLAGS = -version-info 0:2:0 -no-undefined
 if USE_VERSION_SCRIPT
 libgksu2_la_LDFLAGS += -Wl,--version-script=libgksu.ver
 endif
@@ -24,12 +24,11 @@
 
 pkglibdir = ${libdir}/${PACKAGE}
 pkglib_PROGRAMS = gksu-run-helper
-gksu_run_helper_LDFLAGS = `pkg-config --libs glib-2.0`
+gksu_run_helper_LDADD =  ${LIBGKSU_LIBS} $(INTLLIBS)
 gksu_run_helper_SOURCES = gksu-run-helper.c
 
 noinst_PROGRAMS = test-gksu
 test_gksu_SOURCES = test-gksu.c
-test_gksu_LDADD = libgksu2.la
-test_gksu_LDFLAGS = `pkg-config --libs glib-2.0`
+test_gksu_LDADD = libgksu2.la ${LIBGKSU_LIBS}
 
 EXTRA_DIST = libgksu.ver
diff -Nurd -x'*~' libgksu-2.0.12.orig/libgksu/libgksu.c libgksu-2.0.12/libgksu/libgksu.c
--- libgksu-2.0.12.orig/libgksu/libgksu.c	2009-06-29 13:48:24.000000000 -0400
+++ libgksu-2.0.12/libgksu/libgksu.c	2010-04-29 03:59:23.000000000 -0400
@@ -24,7 +24,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <fcntl.h>
-#include <pty.h>
+#include <util.h>
 #include <pwd.h>
 #include <sys/types.h>
 #include <sys/wait.h>
@@ -1936,7 +1936,7 @@
   GQuark gksu_quark;
   int i = 0;
 
-  gchar auxcommand[] = PREFIX "/lib/" PACKAGE "/gksu-run-helper";
+  gchar auxcommand[] = PREFIX "/lib/" PACKAGE "/gksu-run-helper-2";
 
   int fdpty;
   pid_t pid;
@@ -1980,7 +1980,7 @@
       gchar **cmd = g_malloc (sizeof(gchar*)*7);
 
       setsid();   // make us session leader
-      cmd[i] = g_strdup ("/bin/su"); i++;
+      cmd[i] = g_strdup ("/usr/bin/su"); i++;
       if (context->login_shell)
 	{
 	  cmd[i] = g_strdup ("-"); i++;
@@ -2002,7 +2002,7 @@
       if (execv (cmd[0], cmd) == -1)
 	{
 	  fprintf (stderr,
-		   "Unable to run /bin/su: %s",
+		   "Unable to run /usr/bin/su: %s",
 		   strerror(errno));
 	}
 
diff -Nurd -x'*~' libgksu-2.0.12.orig/libgksu/libgksu2.pc.in libgksu-2.0.12/libgksu/libgksu2.pc.in
--- libgksu-2.0.12.orig/libgksu/libgksu2.pc.in	2008-05-09 13:39:55.000000000 -0400
+++ libgksu-2.0.12/libgksu/libgksu2.pc.in	2010-04-29 05:06:41.000000000 -0400
@@ -6,6 +6,6 @@
 Name: LibGKSu
 Version: @VERSION@
 Description: GKSu library version 2.
-Requires: glib-2.0, libstartup-notification-1.0, gnome-keyring-1, gconf-2.0
+Requires: glib-2.0, libstartup-notification-1.0, gtk+-2.0, gconf-2.0
 Libs: -L${libdir} -lgksu2
 Cflags: -I${includedir}/libgksu
diff -Nurd -x'*~' libgksu-2.0.12.orig/libgksuui/Makefile.am libgksu-2.0.12/libgksuui/Makefile.am
--- libgksu-2.0.12.orig/libgksuui/Makefile.am	2008-05-09 13:39:54.000000000 -0400
+++ libgksu-2.0.12/libgksuui/Makefile.am	2010-04-29 04:39:37.000000000 -0400
@@ -1,15 +1,14 @@
 AM_CFLAGS = -g -O2 -Wall
-INCLUDES = `pkg-config --cflags gtk+-2.0 gconf-2.0`
+INCLUDES = $(LIBGKSUUI_CFLAGS)
 AM_CPPFLAGS = -DLOCALEDIR=\"$(datadir)/locale\" -DDATA_DIR=\"$(datadir)\" -DPREFIX=\"$(prefix)\"
 
 noinst_LTLIBRARIES = libgksuui1.0.la
 libgksuui1_0_la_SOURCES = gksuui-dialog.c
-libgksuui1_0_la_LDFLAGS = -Wl,-O1 `pkg-config --libs gtk+-2.0 gconf-2.0`
+libgksuui1_0_la_LIBADD = $(LIBGKSUUI_LIBS) $(INTLLIBS) -L/usr/X11R6/lib -lX11
 
 noinst_HEADERS = defines.h gksuui.h gksuui-dialog.h
 includedir = ${prefix}/include/$(PACKAGE)
 
 noinst_PROGRAMS = test-gksuui
 test_gksuui_SOURCES = test-gksuui.c
-test_gksuui_LDADD = libgksuui1.0.la
-test_gksuui_LDFLAGS = `pkg-config --libs glib-2.0`
+test_gksuui_LDADD = libgksuui1.0.la $(LIBGKSUUI_LIBS)
