diff -urN cd-discid-0.9.orig/Makefile cd-discid-0.9/Makefile
--- cd-discid-0.9.orig/Makefile	Tue Jun 14 12:24:15 2005
+++ cd-discid-0.9/Makefile	Tue Jun 14 12:24:39 2005
@@ -32,9 +32,9 @@
 
 install: cd-discid
 	$(INSTALL) -d -m 755 $(bindir)
-	$(INSTALL) -s -m 755 -o 0 cd-discid $(bindir)
+	$(INSTALL) -s -m 755 cd-discid $(bindir)
 	$(INSTALL) -d -m 755 $(mandir)
-	$(INSTALL) -m 644 -o 0 cd-discid.1 $(mandir)
+	$(INSTALL) -m 644 cd-discid.1 $(mandir)
 
 tarball:
 	@cd .. && tar czvf cd-discid_$(VERSION).orig.tar.gz \
