diff -ru faad2.orig/common/mp4ff/mp4ff.c faad2/common/mp4ff/mp4ff.c
--- faad2.orig/common/mp4ff/mp4ff.c	2004-01-11 10:52:18.000000000 -0500
+++ faad2/common/mp4ff/mp4ff.c	2006-03-23 21:38:27.000000000 -0500
@@ -87,7 +87,7 @@
     if (ff) free(ff);
 }
 
-static void mp4ff_track_add(mp4ff_t *f)
+void mp4ff_track_add(mp4ff_t *f)
 {
     f->total_tracks++;
 
diff -ru faad2.orig/common/mp4v2/mp4property.h faad2/common/mp4v2/mp4property.h
--- faad2.orig/common/mp4v2/mp4property.h	2003-06-29 17:41:00.000000000 -0400
+++ faad2/common/mp4v2/mp4property.h	2006-03-23 21:30:38.000000000 -0500
@@ -58,7 +58,7 @@
 		return m_name;
 	}
 
-	virtual MP4PropertyType GetType() = NULL; 
+	virtual MP4PropertyType GetType() = 0; 
 
 	bool IsReadOnly() {
 		return m_readOnly;
@@ -74,17 +74,17 @@
 		m_implicit = value;
 	}
 
-	virtual u_int32_t GetCount() = NULL;
-	virtual void SetCount(u_int32_t count) = NULL;
+	virtual u_int32_t GetCount() = 0;
+	virtual void SetCount(u_int32_t count) = 0;
 
 	virtual void Generate() { /* default is a no-op */ };
 
-	virtual void Read(MP4File* pFile, u_int32_t index = 0) = NULL;
+	virtual void Read(MP4File* pFile, u_int32_t index = 0) = 0;
 
-	virtual void Write(MP4File* pFile, u_int32_t index = 0) = NULL;
+	virtual void Write(MP4File* pFile, u_int32_t index = 0) = 0;
 
 	virtual void Dump(FILE* pFile, u_int8_t indent,
-		bool dumpImplicits, u_int32_t index = 0) = NULL;
+		bool dumpImplicits, u_int32_t index = 0) = 0;
 
 	virtual bool FindProperty(const char* name,
 		MP4Property** ppProperty, u_int32_t* pIndex = NULL);
diff -ru faad2.orig/common/mp4v2/rtphint.h faad2/common/mp4v2/rtphint.h
--- faad2.orig/common/mp4v2/rtphint.h	2003-06-29 17:41:00.000000000 -0400
+++ faad2/common/mp4v2/rtphint.h	2006-03-23 21:31:01.000000000 -0500
@@ -35,8 +35,8 @@
 		return m_pPacket;
 	}
 
-	virtual u_int16_t GetDataSize() = NULL;
-	virtual void GetData(u_int8_t* pDest) = NULL;
+	virtual u_int16_t GetDataSize() = 0;
+	virtual void GetData(u_int8_t* pDest) = 0;
 
 	MP4Track* FindTrackFromRefIndex(u_int8_t refIndex);
 
diff -ru faad2.orig/libfaad/common.h faad2/libfaad/common.h
--- faad2.orig/libfaad/common.h	2004-02-06 07:55:24.000000000 -0500
+++ faad2/libfaad/common.h	2006-03-23 21:27:29.000000000 -0500
@@ -307,6 +307,8 @@
         }
         return i;
     }
+  #elif (defined(__i386__) && defined(__APPLE_CC__))
+    #define HAS_LRINTF
   #elif (defined(__i386__) && defined(__GNUC__))
     #define HAS_LRINTF
     // from http://www.stereopsis.com/FPU.html
