diff -ru libogg-1.1.2/doc/Makefile.in libogg-1.1.2-patched/doc/Makefile.in
--- libogg-1.1.2/doc/Makefile.in	2004-09-23 15:31:31.000000000 +0200
+++ libogg-1.1.2-patched/doc/Makefile.in	2005-01-16 13:44:11.000000000 +0100
@@ -169,7 +169,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 SUBDIRS = ogg
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 doc_DATA = framing.html index.html oggstream.html rfc3533.txt rfc3534.txt stream.png vorbisword2.png white-ogg.png white-xifish.png
 EXTRA_DIST = $(doc_DATA)
 all: all-recursive
diff -ru libogg-1.1.2/doc/ogg/Makefile.in libogg-1.1.2-patched/doc/ogg/Makefile.in
--- libogg-1.1.2/doc/ogg/Makefile.in	2004-09-23 15:31:31.000000000 +0200
+++ libogg-1.1.2-patched/doc/ogg/Makefile.in	2005-01-16 13:44:11.000000000 +0100
@@ -159,7 +159,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)/ogg
+docdir = $(datadir)/doc/$(PACKAGE)/ogg
 doc_DATA = bitpacking.html datastructures.html decoding.html encoding.html\
 	general.html index.html ogg_packet.html ogg_packet_clear.html\
 	ogg_page.html ogg_page_bos.html ogg_page_checksum_set.html\
