diff -cr vorbis-tools-1.0.1/ogg123/ogg123.c vorbis-tools-juan/ogg123/ogg123.c
*** vorbis-tools-1.0.1/ogg123/ogg123.c	Tue Sep  2 21:37:05 2003
--- vorbis-tools-juan/ogg123/ogg123.c	Wed Feb 11 03:07:39 2004
***************
*** 377,383 ****
  
    /* Setup signal handlers and callbacks */
  
-   ATEXIT (exit_cleanup);
    signal (SIGINT, signal_handler);
    signal (SIGTSTP, signal_handler);
    signal (SIGCONT, signal_handler);
--- 377,382 ----
***************
*** 392,398 ****
  
    playlist_array_destroy(playlist_array, items);
  
!   exit (0);
  }
  
  
--- 391,397 ----
  
    playlist_array_destroy(playlist_array, items);
  
!   exit_cleanup();
  }
  
  
***************
*** 609,617 ****
    status_reset_output_lock();  /* In case we were killed mid-output */
  
    status_message(1, _("Done."));
-   
-   if (sig_request.exit)
-     exit (0);
  }
  
  
--- 608,613 ----
