--- PyGreSQL-3.5/setup.py	Sun Aug 29 20:12:24 2004
+++ PyGreSQL-3.5-new/setup.py	Mon Oct 25 12:02:33 2004
@@ -36,8 +36,8 @@
 	optional_libs=[ 'libpqdll', 'wsock32', 'advapi32' ]
 	data_files = [ 'libpq.dll' ]
 else:
-	include_dirs=[getconfig ('includedir'),getconfig('includedir-server')]
-	library_dirs=[getconfig('libdir')]
+	include_dirs=['@INSTPREFIX@/include/postgresql','@INSTPREFIX@/include',getconfig ('includedir'),getconfig('includedir-server')]
+	library_dirs=['@INSTPREFIX@/lib/postgresql-7.4', '@INSTPREFIX@/lib',getconfig('libdir')]
 	optional_libs=['pq']
 	data_files = []
 
