diff -Naur inid/inid.cnf inid.fink/inid.cnf
--- inid/inid.cnf	Tue Aug 21 20:05:31 2001
+++ inid.fink/inid.cnf	Sun May  2 16:47:45 2004
@@ -8,7 +8,7 @@
 ##
 
 ;; Message catalog (English, Japanese, ...)
-CatalogFile = inid-ja.cat
+;CatalogFile = inid-ja.cat
 
 ;; Write information and error messages to a file instead of stdout.
 ;SyslogFile = iniderr.log
@@ -30,9 +30,9 @@
 ;;
 
 ;ServerPort = 12345
-;ServerLogDir = ../public_html/italk-log
+ServerLogDir = @PREFIX@/var/log/italk
 				; relative path or absolute path for log spool
-;ServerDBDir = DB		; path for keeping user data
+ServerDBDir = @PREFIX@/var/db/italk		; path for keeping user data
 
 
 ;;
diff -Naur inid/killinid.sh inid.fink/killinid.sh
--- inid/killinid.sh	Mon May  8 04:54:27 2000
+++ inid.fink/killinid.sh	Sun May  2 16:54:05 2004
@@ -5,9 +5,9 @@
 #
 
 # for BSD
-kill -QUIT `ps x | awk '/[^l][i]nid/{p=p " " $1}; END{print "PID =" p >"/dev/tty"; print p}'`
+kill -QUIT `ps ax | awk '/[^l][i]nid/{p=p " " $1}; END{print "PID =" p >"/dev/tty"; print p}'`
 sleep 1
-kill -TERM `ps x | awk '/[^l][i]nid/{p=p " " $1}; END{print "PID =" p >"/dev/tty"; print p}'`
+kill -TERM `ps ax | awk '/[^l][i]nid/{p=p " " $1}; END{print "PID =" p >"/dev/tty"; print p}'`
 
 # for System V
 # kill -QUIT `ps -f | awk '/[^l][i]nid/{p=p " " $2}; END{print "PID =" p > "/dev/tty"; print p}'`
diff -Naur inid/wakeinid.csh inid.fink/wakeinid.csh
--- inid/wakeinid.csh	Mon Jun 18 02:20:51 2001
+++ inid.fink/wakeinid.csh	Sun May  2 16:51:22 2004
@@ -5,9 +5,9 @@
 # logging error messages into a file.
 
 # directory and path settings
-set inidpath=~/bin/inid/		# directory for inid(1) binary
-set confpath=~/etc/inid.cnf		# please modify config file, too
-set corepath=~/etc/			# for coredump
+set inidpath=@PREFIX@/bin/
+set confpath=@PREFIX@/etc/inid/inid.cnf
+set corepath=/tmp
 
 # wakeup inid
 setenv PATH /usr/sbin:$PATH
