diff -Nurd -x'*~' iperf-1.7.0.orig/src/Makefile iperf-1.7.0/src/Makefile
--- iperf-1.7.0.orig/src/Makefile	2003-02-26 17:39:30.000000000 -0500
+++ iperf-1.7.0/src/Makefile	2012-02-08 13:22:07.000000000 -0500
@@ -69,6 +69,6 @@
 # build specific files
 
 $(PROG): $(LIBRARY) $(OBJS)
-	$(LD) -o $(PROG) $(LDFLAGS) $(OBJS) -L$(TOP)/lib -ldast $(LIBS)
+	$(LD) -o $(PROG) $(LDFLAGS) $(OBJS) $(TOP)/lib/libdast.a $(LIBS)
 
 include Makefile.depends
