diff -Nurd -x'*~' libproxy-0.4.7.orig/CMakeLists.txt libproxy-0.4.7/CMakeLists.txt
--- libproxy-0.4.7.orig/CMakeLists.txt	2011-06-06 17:44:37.000000000 -0400
+++ libproxy-0.4.7/CMakeLists.txt	2011-09-16 06:21:31.000000000 -0400
@@ -29,7 +29,6 @@
 ### Subdirectories
 # Conditionally build bundled libmodman
 option(FORCE_SYSTEM_LIBMODMAN "Force using system libmodman" OFF)
-find_package(libmodman QUIET)
 if(LIBMODMAN_FOUND)
     if("${LIBMODMAN_VERSION_MAJOR}" STREQUAL "2")
       message(STATUS "Building with system libmodman")    
@@ -48,7 +47,6 @@
 
 # Conditionally build bindings
 if(NOT WIN32)
-    add_subdirectory(bindings)
 endif()
 
 # Always build these
diff -Nurd -x'*~' libproxy-0.4.7.orig/libproxy/cmake/devfiles.cmk libproxy-0.4.7/libproxy/cmake/devfiles.cmk
--- libproxy-0.4.7.orig/libproxy/cmake/devfiles.cmk	2011-06-06 17:44:37.000000000 -0400
+++ libproxy-0.4.7/libproxy/cmake/devfiles.cmk	2011-09-16 04:58:22.000000000 -0400
@@ -1,4 +1,4 @@
-if(NOT WIN32 AND NOT APPLE)
+if(NOT WIN32)
   # PkgConfig file
   set (PKGCONFIG_PREFIX ${CMAKE_INSTALL_PREFIX})
   
diff -Nurd -x'*~' libproxy-0.4.7.orig/libproxy/cmake/modules/pacrunner_webkit.cmk libproxy-0.4.7/libproxy/cmake/modules/pacrunner_webkit.cmk
--- libproxy-0.4.7.orig/libproxy/cmake/modules/pacrunner_webkit.cmk	2011-06-06 17:44:37.000000000 -0400
+++ libproxy-0.4.7/libproxy/cmake/modules/pacrunner_webkit.cmk	2011-09-16 05:01:43.000000000 -0400
@@ -7,10 +7,6 @@
     include_directories("${WEBKIT_INCLUDE_DIR}")
   endif()
 elseif(APPLE)
-  find_library(WEBKIT_LIBRARIES JavaScriptCore)
-  if(WEBKIT_LIBRARIES)
-    set(WEBKIT_FOUND 1)
-  endif()
 else()
   if(WITH_WEBKIT3)
     px_check_modules(WEBKIT webkitgtk-3.0)
