diff -Nurd -x'*~' mutella-0.4.5.orig/util/sockio.c mutella-0.4.5/util/sockio.c
--- mutella-0.4.5.orig/util/sockio.c	2003-09-29 09:52:22.000000000 -0400
+++ mutella-0.4.5/util/sockio.c	2008-06-10 14:59:33.000000000 -0400
@@ -13,6 +13,7 @@
 #include <sys/un.h>
 #include <unistd.h>
 #include <signal.h>
+#include <string.h>
 
 #define BUFF_SIZE 1024
 int main(int argc, char *argv[])
--- mutella-0.4.5.orig/mutella/gnushare.h	2002/11/15 20:22:57	1.12
+++ mutella-0.4.5/mutella/gnushare.h	2006/06/13 20:28:38	1.13
@@ -26,6 +26,7 @@
 #include "mthread.h"
 
 class MGnuDirector;
+class MShareThread;
 
 class MGnuShare  
 {
