diff -Nurd net-snmp-5.4.2.1/configure.in net-snmp-5.4.2.1-new/configure.in
--- net-snmp-5.4.2.1/configure.in	2008-10-31 11:22:23.000000000 -0400
+++ net-snmp-5.4.2.1-new/configure.in	2009-09-04 16:32:58.000000000 -0400
@@ -1057,7 +1057,7 @@
 	    AC_MSG_RESULT([none])
 	fi
 	;;
-    darwin8*|darwin9*)
+    darwin8*|darwin9*|darwin10*)
         AC_MSG_CHECKING([whether darwin ld might need -flat_namespace])
 	if test -z "$MACOSX_DEPLOYMENT_TARGET"; then
 		LDFLAGS="-flat_namespace $LDFLAGS"
@@ -2986,7 +2986,7 @@
     if test "xenable_perl_cc_checks" != "xno" ; then
       AC_MSG_CHECKING([for Perl cc])
       changequote(, )
-      PERLCC=`$myperl -V:cc | $myperl -n -e 'print if (s/^\s*cc=.([-=\w\s\/]+).;\s*/$1/);'`
+      PERLCC=`$myperl -V:cc | $myperl -n -e 'print if (s/^\s*cc=.([-=\w\s\.\/]+).;\s*/$1/);'`
       changequote([, ])
       if test "x$PERLCC" != "x" ; then
         AC_MSG_RESULT([$PERLCC])
@@ -3072,7 +3072,9 @@
 # check the cflags
 if test "x$embed_perl" != "xno" ; then
     AC_MSG_CHECKING([for Perl CFLAGS])
-    perlcflags=`$myperl -MExtUtils::Embed -e ccopts`
+    changequote(, )
+    perlcflags=`$myperl -MExtUtils::Embed -e ccopts | /usr/bin/sed -e 's,-arch [^ ]*,,g'`
+    changequote([, ])
     if test "x$perlcflags" != "x" ; then
       AC_MSG_RESULT([$perlcflags])
       CFLAGS="$CFLAGS $perlcflags"
@@ -3088,7 +3090,9 @@
 # check the ldflags
 if test "x$embed_perl" != "xno" ; then
     AC_MSG_CHECKING([for Perl LDFLAGS])
-    netsnmp_perlldopts=`$myperl -MExtUtils::Embed -e ldopts`
+    changequote(, )
+    netsnmp_perlldopts=`$myperl -MExtUtils::Embed -e ldopts | /usr/bin/sed -e 's,-arch [^ ]*,,g'`
+    changequote([, ])
     if test "x$netsnmp_perlldopts" != "x" ; then
       AC_MSG_RESULT([$netsnmp_perlldopts])
     else
@@ -3104,7 +3108,7 @@
       # Perl ccdlflags (RPATH to libperl, hopefully)
       AC_MSG_CHECKING([for Perl CCDLFLAGS])
       changequote(, )
-      netsnmp_perlccdlflags=`$myperl -V:ccdlflags | $myperl -n -e 'print $1 '"if (/^\s*ccdlflags='([^']+)';/);"`
+      netsnmp_perlccdlflags=`$myperl -V:ccdlflags | /usr/bin/sed -e 's,-arch [^ ]*,,g'`
       changequote([, ])
       AC_MSG_RESULT([$netsnmp_perlccdlflags])
       PERLLDOPTS_FOR_APPS="$netsnmp_perlccdlflags"
diff -Nurd net-snmp-5.4.2.1/perl/TrapReceiver/Makefile.PL net-snmp-5.4.2.1-new/perl/TrapReceiver/Makefile.PL
--- net-snmp-5.4.2.1/perl/TrapReceiver/Makefile.PL	2007-06-18 10:10:08.000000000 -0400
+++ net-snmp-5.4.2.1-new/perl/TrapReceiver/Makefile.PL	2009-09-04 16:21:14.000000000 -0400
@@ -108,7 +108,7 @@
     else {
 	$opts = NetSNMPGetOpts();
 	$Params{'LDDLFLAGS'} = "$Config{lddlflags} " . `$opts->{'nsconfig'} --ldflags`;
-	$Params{'LIBS'} = '-lnetsnmptrapd -lnetsnmpagent -lnetsnmp';
+	$Params{'LIBS'} = '-lnetsnmptrapd -lnetsnmpagent -lnetsnmp -lnetsnmpmibs -lnetsnmphelpers ';
 	chomp($Params{'LIBS'});
 	if (!$ENV{'NETSNMP_CCFLAGS'}) {
 	    $Params{'CCFLAGS'} = `$opts->{'nsconfig'} --cflags`;
