diff -Nurd -x'*~' pyslsk-1.2.7c.orig/pyslsk pyslsk-1.2.7c/pyslsk
--- pyslsk-1.2.7c.orig/pyslsk	2008-12-19 15:15:18.000000000 -0500
+++ pyslsk-1.2.7c/pyslsk	2012-02-11 17:54:51.000000000 -0500
@@ -35,7 +35,7 @@
 
     major = wxPython.wx.wxMAJOR_VERSION
     minor = wxPython.wx.wxMINOR_VERSION
-    rel = wxPython.wx.wxRELEASE_NUMBER
+    rel = wxPython.wx.wxRELEASE_VERSION
 
     oldwx = """You're using an old version of wxPython libraries (%i.%i.%i).
 You should install wxPython 2.6.0 or newer.""" %(major,minor,rel)
