diff -ruN scanssh/config.h.in scanssh-patched/config.h.in
--- scanssh/config.h.in Tue Nov 27 08:49:51 2001
+++ scanssh-patched/config.h.in Wed Jan 30 05:31:46 2002
@@ -1,5 +1,7 @@
 /* config.h.in.  Generated automatically from configure.in by autoheader.  */
 
+#define IN6ADDRSZ 16
+
 /* Define if you have <sys/wait.h> that is POSIX.1 compatible.  */
 #undef HAVE_SYS_WAIT_H

diff -ruN scanssh/missing/getnameinfo.c scanssh-patched/missing/getnameinfo.c 
--- scanssh/missing/getnameinfo.c       Mon Feb 19 16:13:05 2001
+++ scanssh-patched/missing/getnameinfo.c       Wed Jan 30 04:47:41 2002
@@ -12,6 +12,9 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netdb.h>
+#if __APPLE__
+#include <netinet/in.h>
+#endif
 
 #include "config.h"
 

