--- mutella-0.4.3rc1/mutella/xsleep.h	Tue Dec  3 20:05:35 2002
+++ mutella-0.4.3rc1-patched/mutella/xsleep.h	Mon Dec  9 10:49:57 2002
@@ -62,14 +62,6 @@
 extern "C" {
 #endif
 
-// Support dumb missing headers on OSX.
-#ifdef HAVE_NANOSLEEP
-#if defined HAVE_DECL_NANOSLEEP__ && !HAVE_DECL_NANOSLEEP__
-//#ifdef __APPLE__
-int nanosleep(struct timespec *, struct timespec *);
-#endif
-#endif
-
 int safe_nanosleep(struct timespec *rqtp, struct timespec *rmtp);
 
 #ifdef __cplusplus

--- mutella-0.4.3rc1/mutella/sha1thread.cpp	Wed Nov 27 05:27:37 2002
+++ mutella-0.4.3rc1-patched/mutella/sha1thread.cpp	Tue Dec 10 00:41:37 2002
@@ -36,6 +36,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <stdint.h>
 #include "xsleep.h"
 #include "conversions.h"
 #include "common.h"
