diff -urN dvtm-0.5.2/config.mk dvtm-0.5.2.mod/config.mk
--- dvtm-0.5.2/config.mk	2009-02-08 22:52:37.000000000 +0100
+++ dvtm-0.5.2.mod/config.mk	2009-07-17 23:55:09.000000000 +0200
@@ -3,11 +3,11 @@
 
 # Customize below to fit your system
 
-PREFIX = /usr/local
+PREFIX = @PREFIX@
 MANPREFIX = ${PREFIX}/share/man
 
 INCS = -I. -I/usr/include -I/usr/local/include 
-LIBS = -lc -lutil -lncursesw
+LIBS = -lc -lSystem -lncursesw
 #LIBS = -lc -lutil -lncurses
 
 CFLAGS += -std=c99 -Os ${INCS} -DVERSION=\"${VERSION}\" -DNDEBUG
diff -urN dvtm-0.5.2/Makefile dvtm-0.5.2.mod/Makefile
--- dvtm-0.5.2/Makefile	2009-02-08 22:52:37.000000000 +0100
+++ dvtm-0.5.2.mod/Makefile	2009-07-17 08:56:10.000000000 +0200
@@ -41,7 +41,7 @@
 
 install: dvtm
 	@echo stripping executable
-	@strip -s dvtm
+	@strip dvtm
 	@echo installing executable file to ${DESTDIR}${PREFIX}/bin
 	@mkdir -p ${DESTDIR}${PREFIX}/bin
 	@cp -f dvtm ${DESTDIR}${PREFIX}/bin
