diff -Nur AnnoCPAN-Perldoc-0.10-orig/Makefile.PL AnnoCPAN-Perldoc-0.10/Makefile.PL
--- AnnoCPAN-Perldoc-0.10-orig/Makefile.PL	2005-07-21 21:37:41.000000000 -0500
+++ AnnoCPAN-Perldoc-0.10/Makefile.PL	2005-09-27 10:24:17.000000000 -0500
@@ -10,7 +10,7 @@
         DBD::SQLite  => 1.0,
         IO::String   => 0,
         Digest::MD5  => 0,
-        Pod::Perldoc => 3.14,
+        Pod::Perldoc => 3.13,
     },
 );
 
diff -Nur AnnoCPAN-Perldoc-0.10-orig/README AnnoCPAN-Perldoc-0.10/README
--- AnnoCPAN-Perldoc-0.10-orig/README	2005-07-21 21:22:57.000000000 -0500
+++ AnnoCPAN-Perldoc-0.10/README	2005-09-27 21:32:03.000000000 -0500
@@ -17,6 +17,7 @@
 $HOME
 $USERPROFILE
 $ALLUSERSPROFILE
+@PREFIX@/var/annocpan
 /var/annocpan
 
 It can also be called .annopod.db, to hide it in Unix-like systems. It is your
diff -Nur AnnoCPAN-Perldoc-0.10-orig/annopod AnnoCPAN-Perldoc-0.10/annopod
--- AnnoCPAN-Perldoc-0.10-orig/annopod	2005-07-21 21:36:28.000000000 -0500
+++ AnnoCPAN-Perldoc-0.10/annopod	2005-09-27 21:31:43.000000000 -0500
@@ -43,6 +43,7 @@
 $HOME
 $USERPROFILE
 $ALLUSERSPROFILE
+@PREFIX@/var/annocpan
 /var/annocpan
 
 It can also be called .annopod.db, to hide it in Unix-like systems. It is your
diff -Nur AnnoCPAN-Perldoc-0.10-orig/lib/AnnoCPAN/Perldoc/Filter.pm AnnoCPAN-Perldoc-0.10/lib/AnnoCPAN/Perldoc/Filter.pm
--- AnnoCPAN-Perldoc-0.10-orig/lib/AnnoCPAN/Perldoc/Filter.pm	2005-07-21 21:12:11.000000000 -0500
+++ AnnoCPAN-Perldoc-0.10/lib/AnnoCPAN/Perldoc/Filter.pm	2005-09-27 21:30:51.000000000 -0500
@@ -32,7 +32,7 @@
     my $signature = md5_hex($pod);
     my $db_file;
     DIR: for my $dir (@ENV{qw(HOME USERPROFILE ALLUSERSPROFILE)}, 
-        '/var/annocpan', '.'
+        '@PREFIX@/var/annocpan', '/var/annocpan', '.'
     ) {
         for my $file ('annopod.db', '.annopod.db') {
             no warnings 'uninitialized';
diff -Nur AnnoCPAN-Perldoc-0.10-orig/lib/AnnoCPAN/Perldoc.pm AnnoCPAN-Perldoc-0.10/lib/AnnoCPAN/Perldoc.pm
--- AnnoCPAN-Perldoc-0.10-orig/lib/AnnoCPAN/Perldoc.pm	2005-07-21 21:29:50.000000000 -0500
+++ AnnoCPAN-Perldoc-0.10/lib/AnnoCPAN/Perldoc.pm	2005-09-27 21:31:20.000000000 -0500
@@ -101,6 +101,7 @@
     $HOME
     $USERPROFILE
     $ALLUSERSPROFILE
+    @PREFIX@/var/annocpan
     /var/annocpan
 
 It can also be called .annopod.db, to hide it in Unix-like systems. It is your
