diff -ru BerkeleyDB-0.26.orig/config.in BerkeleyDB-0.26/config.in
--- BerkeleyDB-0.26.orig/config.in	Tue Oct 19 21:55:32 2004
+++ BerkeleyDB-0.26/config.in	Tue Oct 19 22:01:18 2004
@@ -9,7 +9,8 @@
 
 #INCLUDE	= /usr/local/include
 #INCLUDE	= ../..
-INCLUDE	= /usr/local/BerkeleyDB/include
+#INCLUDE	= /usr/local/BerkeleyDB/include
+INCLUDE	= @PREFIX@/include/db4
 
 # 2. Where is libdb?
 #
@@ -18,7 +19,8 @@
 
 #LIB	= /usr/local/lib
 #LIB	= ../..
-LIB	= /usr/local/BerkeleyDB/lib
+#LIB	= /usr/local/BerkeleyDB/lib
+LIB	= @PREFIX@/lib
 
 # 3. Is the library called libdb?
 #
