diff -Nurd -x'*~' Class-Method-Modifiers-2.04.orig/Makefile.PL Class-Method-Modifiers-2.04/Makefile.PL
--- Class-Method-Modifiers-2.04.orig/Makefile.PL	2013-03-30 23:15:02.000000000 -0400
+++ Class-Method-Modifiers-2.04/Makefile.PL	2013-05-02 21:36:51.000000000 -0400
@@ -1,9 +1,5 @@
 # Load the Module::Install bundled in ./inc/
 use inc::Module::Install;
-use Module::Install::AuthorRequires;
-use Module::Install::ReadmeFromPod;
-use Module::Install::GithubMeta;
-use Module::Install::AutoLicense;
 
 # Define metadata
 name           'Class-Method-Modifiers';
