diff -uNr ExtUtils-F77-1.16/F77.pm ExtUtils-F77-1.16-new/F77.pm
--- ExtUtils-F77-1.16/F77.pm	2007-04-01 23:40:23.000000000 -0500
+++ ExtUtils-F77-1.16-new/F77.pm	2007-07-28 14:50:33.000000000 -0500
@@ -173,7 +173,7 @@
     my @libs = ('g2c', 'f2c');
     my ($dir, $lib, $test);
     foreach $test (@libs) {
-      $dir = `g77 -print-file-name=lib$test.a`;
+      $dir = `g95 -print-file-name=lib$test.a`;
       chomp $dir;
       # Note that -print-file-name returns just the library name
       # if it cant be found - make sure that we only accept the
@@ -193,8 +193,8 @@
     return( "-L$dir -L/usr/lib -l$lib -lm" );
 };
 $F77config{Generic}{G77}{Trail_} = 1;
-$F77config{Generic}{G77}{Compiler} = find_in_path('g77','f77','fort77');
-$F77config{Generic}{G77}{Cflags} = '-O';
+$F77config{Generic}{G77}{Compiler} = find_in_path('g95','g77','f77','fort77');
+$F77config{Generic}{G77}{Cflags} = '-Os';
 $F77config{Generic}{DEFAULT} = 'G77';
 $F77config{Generic}{F2c}     = $F77config{Generic}{G77};
 
