diff -Nurd -x'*~' Gnome2-VFS-1.080.orig/Makefile.PL Gnome2-VFS-1.080/Makefile.PL
--- Gnome2-VFS-1.080.orig/Makefile.PL	2006-03-12 17:08:19.000000000 -0500
+++ Gnome2-VFS-1.080/Makefile.PL	2008-02-03 01:21:19.000000000 -0500
@@ -111,7 +111,7 @@
 ###############################################################################
 
 my $vfs = ExtUtils::Depends->new ('Gnome2::VFS', 'Glib');
-$vfs->set_inc ($vfs_pkgcfg{cflags} . " -I build");
+$vfs->set_inc (' -Ibuild ' . $vfs_pkgcfg{cflags});
 $vfs->set_libs ($vfs_pkgcfg{libs});
 $vfs->add_xs (@xs_files);
 $vfs->add_c ("vfs2perl.c");
