diff -Nurd -x'*~' Locale-US-2.112150.orig/Makefile.PL Locale-US-2.112150/Makefile.PL
--- Locale-US-2.112150.orig/Makefile.PL	2011-08-03 10:40:06.000000000 -0400
+++ Locale-US-2.112150/Makefile.PL	2011-09-04 14:03:08.000000000 -0400
@@ -12,7 +12,6 @@
   'ABSTRACT' => 'two letter codes for state identification in the United States and vice versa.',
   'AUTHOR' => 'Terrence Brannon',
   'BUILD_REQUIRES' => {
-    'Test' => '0',
     'Test::More' => '0'
   },
   'CONFIGURE_REQUIRES' => {
