diff -Nurd -x'*~' Module-Release-2.06.orig/lib/Module/Release.pm Module-Release-2.06/lib/Module/Release.pm
--- Module-Release-2.06.orig/lib/Module/Release.pm	2011-12-19 15:47:24.000000000 -0500
+++ Module-Release-2.06/lib/Module/Release.pm	2012-07-23 05:09:54.000000000 -0400
@@ -600,12 +600,13 @@
 		if( $path =~ m/[*?[]/ && $self->config->allow_glob_in_perls )
 			{
 			$self->add_a_perl( $_ ) for glob $path;
+			return;
 			}
-		else
+		if( system( "$path -e 'exit 0' 2>/dev/null" ) )
 			{
-			$self->_warn( "$path is not executable" );
+			$self->_warn( "$path is not executable ($?)" );
+			return;
 			}
-		return;
 		}
 
 	my $version = $self->_looks_like_perl( $path );
