diff -Nurd -x'*~' Pango-1.220.orig/Makefile.PL Pango-1.220/Makefile.PL
--- Pango-1.220.orig/Makefile.PL	2009-03-17 13:58:06.000000000 -0400
+++ Pango-1.220/Makefile.PL	2009-06-03 05:15:29.000000000 -0400
@@ -157,12 +157,12 @@
 #
 
 my @deps = qw/Glib/;
-my $inc = $pango_pkgcfg{cflags} . ' -I./build ';
+my $inc = '-I./build ' . $pango_pkgcfg{cflags};
 my $libs = $pango_pkgcfg{libs};
 if ($have_cairo) {
 	push @deps, qw/Cairo/;
-	$inc .= $pangocairo_pkgcfg{cflags};
-	$libs .= $pangocairo_pkgcfg{libs};
+	$inc .= ' ' . $pangocairo_pkgcfg{cflags};
+	$libs .= ' ' . $pangocairo_pkgcfg{libs};
 }
 
 my @typemaps = map { File::Spec->catfile(cwd(), $_) }
