diff -ruN fink/perlmod/Fink/ChangeLog fink.cvs/perlmod/Fink/ChangeLog
--- fink/perlmod/Fink/ChangeLog	Tue Aug  5 07:09:15 2003
+++ fink.cvs/perlmod/Fink/ChangeLog	Tue Aug  5 07:51:36 2003
@@ -1,3 +1,9 @@
+2003-08-05  Justin F. Hallett  <thesin@users.sourceforge.net>
+
+	* Selfupdate.pm: Added ability to switch CVS servers, this requires
+	  3 new fink.conf fields and 1 optional.  CVSServer,
+	  CVSType (pserver or ext), CVSPath and the optional CVSUser.
+
 2003-07-17  Benjamin Reed  <rangerrick@users.sourceforge.net>
 
 	* VirtPackage.pm: Don't print the output of linking the
diff -ruN fink/perlmod/Fink/SelfUpdate.pm fink.cvs/perlmod/Fink/SelfUpdate.pm
--- fink/perlmod/Fink/SelfUpdate.pm	Thu Jul 17 13:39:57 2003
+++ fink.cvs/perlmod/Fink/SelfUpdate.pm	Tue Aug  5 07:55:05 2003
@@ -59,6 +59,45 @@
 	$finkdir = "$basepath/fink";
 
 	if (-d "$finkdir/CVS") {
+		### Check if the server in the fink.conf matches current
+		### checkout if not, nuke CVS dirs and restart checkout
+		### FIXME
+		my ($servertype, $server, $cvspth, $cvsuser);
+		my ($fullcvs, $currentcvs);
+
+		$servertype = $config->param("CVSType") if ($config->param("CVSType"));
+		$server = $config->param("CVSServer") if ($config->param("CVSServer"));
+		$cvspth = $config->param("CVSPath") if ($config->param("CVSPath"));
+		$cvsuser = $config->param("CVSUser") if ($config->param("CVSUser"));
+		if (!defined $cvsuser) {
+			$cvsuser = "anonymous";
+		}
+		if (defined $server && defined $cvspth &&
+		    defined $servertype && ($servertype eq "pserver" or
+					    $servertype eq "ext")) {
+			### Get current cvs rep info
+			$currentcvs = `cat $finkdir/CVS/Root`;
+			### Make info from conf file
+			$fullcvs = ":".$servertype.":".$cvsuser."@".$server.
+				   ":".$cvspth;
+			if ($fullcvs ne $currentcvs) {
+				$answer = &prompt_boolean(
+					"You are requesting to change CVS ".
+					"servers from $currentcvs to ".
+					"$fullcvs, are you sure this is what ".
+					"you want?", 1);
+				### Add change cvs question
+				if ($answer) {
+					### Nuke CVS dir
+					if (&execute("rm -rf $finkdir/CVS")) {
+						die "Couldn't remove $finkdir/CVS\n";
+					}
+				}
+			}
+		}
+	}
+
+        if (-d "$finkdir/CVS") {
 		if ($usecvs) {
 			$answer = 1;
 		} else {
@@ -162,7 +201,12 @@
 	my ($username, $cvsuser, @testlist);
 	my ($use_hardlinks, $cutoff, $cmd);
 	my ($cmdd);
+	my ($servertype, $server, $cvspth, $serveruser);
 
+	$servertype = $config->param("CVSType") || "pserver";
+	$server = $config->param("CVSServer") || "cvs.sourceforge.net";
+	$cvspth = $config->param("CVSPath") || "/cvsroot/fink";
+	$serveruser = $config->param("CVSUser") || "anonymous";
 
 	$username = "root";
 	if (exists $ENV{SUDO_USER}) {
@@ -186,12 +230,16 @@
 	}
 
 	print "\n";
-	$cvsuser =
-		&prompt("For Fink developers only: ".
-				"Enter your SourceForge login name to set up full CVS access. ".
+	if ($serveruser ne "anonymous") {
+		$cvsuser =
+			&prompt("For Fink developers only: ".
+				"Enter your CVS login name to set up full CVS access. ".
 				"Other users, just press return to set up anonymous ".
-				"read-only access.",
+				"read-only access.  Normally used with Sourceforge.",
 				"anonymous");
+	} else {
+		$cvsuser = $serveruser;
+	}
 	print "\n";
 
 	# start by creating a temporary directory with the right permissions
@@ -232,21 +280,21 @@
 	if (Fink::Config::verbosity_level() > 1) {
 		$verbosity = "";
 	}
-	if ($cvsuser eq "anonymous") {
+	if ($server eq "pserver") {
 		&print_breaking("Now logging into the CVS server. When CVS asks you ".
 						"for a password, just press return (i.e. the password ".
 						"is empty).");
-		$cmd = "cvs -d:pserver:anonymous\@cvs.sourceforge.net:/cvsroot/fink login";
+		$cmd = "cvs -d:$servertype:$cvsuser\@$server:$cvspth login";
 		if ($username ne "root") {
 			$cmd = "su $username -c '$cmd'";
 		}
 		if (&execute($cmd)) {
-			die "Logging into the CVS server for anonymous read-only access failed.\n";
+			die "Logging into the CVS server for $cvsuser read-only access failed.\n";
 		}
 
-		$cmd = "cvs ${verbosity} -z3 -d:pserver:anonymous\@cvs.sourceforge.net:/cvsroot/fink";
+		$cmd = "cvs ${verbosity} -z3 -d:$servertype:$cvsuser\@$server:$cvspth";
 	} else {
-		$cmd = "cvs ${verbosity} -z3 -d:ext:$cvsuser\@cvs.sourceforge.net:/cvsroot/fink";
+		$cmd = "cvs ${verbosity} -z3 -d:$servertype:$cvsuser\@$server:$cvspth";
 		$ENV{CVS_RSH} = "ssh";
 	}
 	$cmdd = "$cmd checkout -d fink dists";
