? fink.dbm.db
Index: perlmod/Fink/ChangeLog
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/ChangeLog,v
retrieving revision 1.315
diff -u -u -r1.315 ChangeLog
--- perlmod/Fink/ChangeLog	15 Aug 2003 02:43:39 -0000	1.315
+++ perlmod/Fink/ChangeLog	17 Aug 2003 03:04:22 -0000
@@ -1,3 +1,8 @@
+2003-08-16  Justin F. Hallett  <thesin@users.sourceforge.net>
+
+	* PkgVersion.pm: Added -j to CFLAGS and CXXFLAGS via fink.conf field
+	  'Processors:'.
+
 2003-08-14  Dave Morrison  <dmrrsn@users.sourceforge.net>
 
 	* PkgVersion.pm: Disable the prebinding code unless the distribution
Index: perlmod/Fink/PkgVersion.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/PkgVersion.pm,v
retrieving revision 1.129
diff -u -u -r1.129 PkgVersion.pm
--- perlmod/Fink/PkgVersion.pm	14 Aug 2003 15:58:56 -0000	1.129
+++ perlmod/Fink/PkgVersion.pm	17 Aug 2003 03:04:23 -0000
@@ -1905,9 +1905,17 @@
 
 sub set_env {
 	my $self = shift;
-	my ($varname, $s, $expand);
+	my ($varname, $s, $expand, $procs);
+	if ($config->has_param("Processors")) {
+		### Add -j <proc number> to CFLAGS and CXXFLAGS
+		$procs = $config->param("Processors");
+	} else {
+		$procs = 1;
+	}
 	my %defaults = ( "CPPFLAGS" => "-I\%p/include",
-					 "LDFLAGS" => "-L\%p/lib" );
+					"LDFLAGS"  => "-L\%p/lib",
+					"CFLAGS"   => "-j $procs",
+					"CXXFLAGS" => "-j $procs" );
 	my $bsbase = Fink::Bootstrap::get_bsbase();
 
 	# clean the environment
