diff -ruN fink/perlmod/Fink/Engine.pm fink-thesin/perlmod/Fink/Engine.pm
--- fink/perlmod/Fink/Engine.pm	Sun May 26 11:01:02 2002
+++ fink-thesin/perlmod/Fink/Engine.pm	Mon May 27 14:28:14 2002
@@ -76,6 +76,7 @@
     'description' => [\&cmd_description, 1, 0],
     'desc' => [\&cmd_description, 1, 0],
     'info' => [\&cmd_description, 1, 0],
+    'devinfo' => [\&cmd_devinfo, 1, 0],
     'scanpackages' => [\&cmd_scanpackages, 1, 1],
     'list' => [\&cmd_list, 0, 0],
     'listpackages' => [\&cmd_listpackages, 1, 0],
@@ -519,6 +520,39 @@
     print $package->get_fullname().": ";
     print $package->get_description();
     print "\n";
+  }
+}
+
+sub cmd_devinfo {  
+  my ($package, @plist);
+
+  @plist = &expand_packages(@_);
+  if ($#plist < 0) { 
+    die "no package specified for command 'devinfo'!\n";
+  } 
+      
+  print "\n";
+  foreach $package (@plist) {
+    print "Package      : " . $package->get_fullname() . "\n";
+    print "Location     : " . $package->get_location() . "\n";
+    if ($package->get_depends() ne "None") {
+      print "Depends      : " . $package->get_depends() . "\n";
+    }
+    if ($package->get_builddepends() ne "None") {
+      print "BuildDepends : " . $package->get_builddepends() . "\n";
+    }
+    if ($package->get_conflicts() ne "None") {
+      print "Conflicts    : " . $package->get_conflicts() . "\n";
+    }
+    if ($package->get_replaces() ne "None") {
+      print "Replaces     : " . $package->get_replaces() . "\n";
+    }
+    if ($package->get_provides() ne "None") {
+      print "Provides     : " . $package->get_provides() . "\n";
+    }
+    print "\n";
+    print "Port Log:\n";
+    print $package->get_portlog();
   }
 }
 
diff -ruN fink/perlmod/Fink/PkgVersion.pm fink-thesin/perlmod/Fink/PkgVersion.pm
--- fink/perlmod/Fink/PkgVersion.pm	Sat May 25 15:40:44 2002
+++ fink-thesin/perlmod/Fink/PkgVersion.pm	Mon May 27 14:28:23 2002
@@ -377,6 +377,56 @@
   return $self->{_tree};
 }
 
+sub get_location {    
+  my $self = shift;
+  return $self->{_filename};        
+}        
+
+sub get_depends {
+  my $self = shift;
+  if ($self->param("Depends")) {
+    return $self->param("Depends");
+  } else {
+    return "None";
+  }
+} 
+
+sub get_builddepends {
+  my $self = shift;
+  if ($self->param("BuildDepends")) {
+    return $self->param("BuildDepends");
+  } else {
+    return "None";
+  }
+} 
+
+sub get_replaces {
+  my $self = shift;
+  if ($self->param("Replaces")) {
+    return $self->param("Replaces");
+  } else {
+    return "None";
+  }
+} 
+
+sub get_conflicts {
+  my $self = shift;
+  if ($self->param("Conflicts")) {
+    return $self->param("Conflicts");
+  } else {
+    return "None";
+  }
+} 
+
+sub get_provides {
+  my $self = shift;
+  if ($self->param("Provides")) {
+    return $self->param("Provides");
+  } else {
+    return "None";
+  }
+} 
+
 ### other accessors
 
 sub is_multisource {
@@ -551,6 +601,22 @@
     }
   }
 
+  return $desc;
+}
+
+### Changelog style log for porting information
+sub get_portlog {
+  my $self = shift;
+  my $style = shift || 0;
+  my ($desc, $s);
+    
+  if ($self->has_param("PortLog")) {
+    $desc = $self->param("PortLog");
+  } else {
+    $desc = "  [Package ".$self->get_name()." version ".$self->get_fullversion()." - Has *NO* PortLog]";
+  } 
+  $desc .= "\n";
+    
   return $desc;
 }
 
diff -ruN fink/perlmod/Fink/Validation.pm fink-thesin/perlmod/Fink/Validation.pm
--- fink/perlmod/Fink/Validation.pm	Mon May 27 13:06:29 2002
+++ fink-thesin/perlmod/Fink/Validation.pm	Mon May 27 14:28:18 2002
@@ -134,6 +134,7 @@
      descusage
      descpackaging
      descport
+     portlog
      homepage
      license
     )
