diff -ruN fink/perlmod/Fink/ChangeLog fink.-j/perlmod/Fink/ChangeLog
--- fink/perlmod/Fink/ChangeLog	Sat Aug 16 21:11:15 2003
+++ fink.-j/perlmod/Fink/ChangeLog	Sat Aug 16 21:11:07 2003
@@ -1,3 +1,8 @@
+2003-08-16  Justin F. Hallett  <thesin@users.sourceforge.net>
+
+	* PkgVersion.pm: Added -j to CFLAGS and CXXFLAGS via fink.conf field
+	  'Processors:'.
+
 2003-08-14  Dave Morrison  <dmrrsn@users.sourceforge.net>
 
 	* PkgVersion.pm: Disable the prebinding code unless the distribution
diff -ruN fink/perlmod/Fink/PkgVersion.pm fink.-j/perlmod/Fink/PkgVersion.pm
--- fink/perlmod/Fink/PkgVersion.pm	Sat Aug 16 21:11:15 2003
+++ fink.-j/perlmod/Fink/PkgVersion.pm	Sat Aug 16 21:12:23 2003
@@ -1905,9 +1905,16 @@
 
 sub set_env {
 	my $self = shift;
-	my ($varname, $s, $expand);
+	my ($varname, $s, $expand, $procs);
+	if ($config->has_param("Processors")) {
+		### Add -j <proc number> to MAKEFLAGS
+		$procs = $config->param("Processors");
+	} else {
+		$procs = 1;
+	}
 	my %defaults = ( "CPPFLAGS" => "-I\%p/include",
-					 "LDFLAGS" => "-L\%p/lib" );
+					"LDFLAGS"   => "-L\%p/lib",
+					"MAKEFLAGS" => "-j $procs" );
 	my $bsbase = Fink::Bootstrap::get_bsbase();
 
 	# clean the environment
@@ -1937,7 +1944,7 @@
 					  "LD", "LDFLAGS", 
 					  "LIBRARY_PATH", "LIBS",
 					  "MACOSX_DEPLOYMENT_TARGET",
-					  "MAKE", "MFLAGS") {
+					  "MAKE", "MFLAGS", "MAKEFLAGS") {
 		if ($self->has_param("Set$varname")) {
 			$s = $self->param("Set$varname");
 			if (exists $defaults{$varname} and
