diff -ru Botan-1.10.8.orig/src/build-data/makefile/unix_shr.in Botan-1.10.8/src/build-data/makefile/unix_shr.in
--- Botan-1.10.8.orig/src/build-data/makefile/unix_shr.in	2014-04-10 10:11:44.000000000 -0400
+++ Botan-1.10.8/src/build-data/makefile/unix_shr.in	2014-04-13 19:11:54.000000000 -0400
@@ -19,7 +19,7 @@
 BINDIR        = $(DESTDIR)/bin
 LIBDIR        = $(DESTDIR)/%{libdir}
 HEADERDIR     = $(DESTDIR)/%{includedir}/botan-$(SERIES)/botan
-DOCDIR        = $(DESTDIR)/%{docdir}/botan-$(VERSION)
+DOCDIR        = $(DESTDIR)/%{docdir}/botan$(SERIES)
 PKGCONF_DIR   = $(LIBDIR)/pkgconfig
 
 CONFIG_SCRIPT = %{botan_config}
@@ -57,8 +57,8 @@
 LIBNAME       = %{lib_prefix}libbotan
 STATIC_LIB    = $(LIBNAME)-$(SERIES).a
 
-SONAME        = $(LIBNAME)-$(SERIES).%{so_suffix}.%{so_abi_rev}
-SHARED_LIB    = $(SONAME).%{version_patch}
+SONAME        = $(LIBNAME)-$(SERIES).0.%{so_suffix}
+SHARED_LIB    = $(SONAME)
 
 SYMLINK       = $(LIBNAME)-$(SERIES).%{so_suffix}
 
@@ -80,7 +80,6 @@
 
 $(SHARED_LIB): $(LIBOBJS)
 	$(SO_LINK_CMD) $(LDFLAGS) $(LIBOBJS) -o $(SHARED_LIB) $(LINK_TO)
-	$(LN) $(SHARED_LIB) $(SONAME)
 	$(LN) $(SHARED_LIB) $(SYMLINK)
 
 # Fake Targets
@@ -116,7 +115,6 @@
 	$(INSTALL_CMD_DATA) $(STATIC_LIB) $(LIBDIR)
 	$(INSTALL_CMD_EXEC) $(SHARED_LIB) $(LIBDIR)
 	$(CD) $(LIBDIR); $(LN) $(SHARED_LIB) $(SYMLINK)
-	$(CD) $(LIBDIR); $(LN) $(SHARED_LIB) $(SONAME)
 
 	$(MKDIR_INSTALL) $(BINDIR)
 	$(INSTALL_CMD_EXEC) $(CONFIG_SCRIPT) $(BINDIR)
