diff -Nurd gsasl-2.2.1.orig/configure.ac gsasl-2.2.1/configure.ac
--- gsasl-2.2.1.orig/configure.ac	2024-01-02 11:40:13.000000000 -0500
+++ gsasl-2.2.1/configure.ac	2024-01-02 18:45:08.000000000 -0500
@@ -239,7 +239,7 @@
 
 # GSSAPI.
 AC_ARG_WITH(gssapi-impl,
-  AS_HELP_STRING([--with-gssapi-impl=no/gss/gssglue/kfw/mit/heimdal/yes],
+  AS_HELP_STRING([--with-gssapi-impl=no/gss/gssglue/kfw/mit/heimdal/framework/yes],
                  [select GSS-API implementation]),
   gssapi_impl=$withval, gssapi_impl=yes)
 if test "$gs2" = no && test "$gssapi" = no; then
@@ -297,6 +297,16 @@
     AC_MSG_WARN([Heimdal krb5-config not found, disabling GSSAPI])
     gssapi_impl=no
   fi
+elif test "$gssapi_impl" = "framework"; then
+  AC_MSG_NOTICE([trying GSS framework])
+  GSS_CFLAGS=""		# <GSS/gssapi.h> in default search-path
+  GSS_LIBS="-framework GSS"
+  AC_CHECK_HEADERS([GSS/gssapi.h])
+  if test "$ac_cv_header_gss_gssapi_h" = "no"; then
+    gssapi_impl=no
+    AC_MSG_WARN([Cannot find GSS/gssapi.h, disabling GSSAPI])
+  fi
+  # linking with -framework flag tested later via function tests
 elif test "$gssapi_impl" != "no"; then
   AC_MSG_NOTICE([auto-detecting GSS-API library GSS/Libgssglue/MIT/Heimdal])
   AC_MSG_NOTICE([use --with-gssapi-impl=IMPL to hard code])
@@ -337,7 +347,8 @@
   AC_CHECK_FUNCS([gss_decapsulate_token])
   AC_CHECK_FUNCS([gss_oid_equal])
   AC_CHECK_FUNCS([gss_inquire_mech_for_saslname])
-  if test "$gssapi_impl" != "gss" && test "$gssapi_impl" != "gssglue"; then
+  # "framework" already verified its own header path
+  if test "$gssapi_impl" != "gss" && test "$gssapi_impl" != "gssglue" && test "$gssapi_impl" != "framework"; then
     AC_CHECK_HEADERS([gssapi.h gssapi/gssapi.h gssapi/gssapi_ext.h])
     if test "$ac_cv_header_gssapi_h$ac_cv_header_gssapi_gssapi_h" = "nono"; then
       gssapi_impl=no
@@ -358,6 +369,9 @@
 #  if HAVE_GSSAPI_GSSAPI_EXT_H
 #   include <gssapi/gssapi_ext.h>
 #  endif
+#  if HAVE_GSS_GSSAPI_H
+#   include <GSS/gssapi.h>
+#  endif
 # endif
 ]])
   CPPFLAGS="$save_CPPFLAGS"
diff -Nurd gsasl-2.2.1.orig/doc/Makefile.am gsasl-2.2.1/doc/Makefile.am
--- gsasl-2.2.1.orig/doc/Makefile.am	2024-01-02 08:03:48.000000000 -0500
+++ gsasl-2.2.1/doc/Makefile.am	2024-01-02 18:50:50.000000000 -0500
@@ -43,9 +43,9 @@
 AM_CPPFLAGS = -I$(top_builddir)/lib/src -I$(top_srcdir)/lib/src
 EXTRA_PROGRAMS = print-errors
 print_errors_LDADD = ../lib/src/libgsasl.la
+print_errors_SOURCES = print-errors.c
 
-gsasl-api-error-labels.texi: $(top_srcdir)/lib/src/error.c $(srcdir)/print-errors.c
-	$(MAKE) $(builddir)/print-errors$(EXEEXT)
+gsasl-api-error-labels.texi: $(top_srcdir)/lib/src/error.c $(builddir)/print-errors$(EXEEXT)
 	$(builddir)/print-errors$(EXEEXT) > $@.new
 	mv $@.new $@
 
diff -Nurd gsasl-2.2.1.orig/lib/gl/gss-extra.h gsasl-2.2.1/lib/gl/gss-extra.h
--- gsasl-2.2.1.orig/lib/gl/gss-extra.h	2024-01-02 12:00:58.000000000 -0500
+++ gsasl-2.2.1/lib/gl/gss-extra.h	2024-01-02 18:45:08.000000000 -0500
@@ -37,6 +37,9 @@
 #  if HAVE_GSSAPI_GSSAPI_EXT_H
 #   include <gssapi/gssapi_ext.h>
 #  endif
+#  if HAVE_GSS_GSSAPI_H
+#   include <GSS/gssapi.h>
+#  endif
 # endif
 
 # if ! HAVE_GSS_OID_EQUAL
