diff -Nurd -x'*~' gssdp-0.8.0.orig/libgssdp/Makefile.in gssdp-0.8.0/libgssdp/Makefile.in
--- gssdp-0.8.0.orig/libgssdp/Makefile.in	2010-09-16 08:10:09.000000000 -0400
+++ gssdp-0.8.0/libgssdp/Makefile.in	2011-03-21 12:42:17.000000000 -0400
@@ -260,7 +260,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 LTVERSION = 2:0:0
-AM_CFLAGS = $(LIBGSSDP_CFLAGS) -I$(top_srcdir)
+AM_CPPFLAGS = -I$(top_srcdir) $(LIBGSSDP_CFLAGS)
 libgssdpincdir = $(includedir)/gssdp-1.0/libgssdp
 lib_LTLIBRARIES = libgssdp-1.0.la
 libgssdpinc_HEADERS = gssdp-client.h		 \
diff -Nurd -x'*~' gssdp-0.8.0.orig/tests/Makefile.in gssdp-0.8.0/tests/Makefile.in
--- gssdp-0.8.0.orig/tests/Makefile.in	2010-09-16 08:10:09.000000000 -0400
+++ gssdp-0.8.0/tests/Makefile.in	2011-03-21 15:02:20.000000000 -0400
@@ -217,7 +217,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-AM_CFLAGS = $(LIBGSSDP_CFLAGS) -I$(top_srcdir)
+AM_CPPFLAGS = -I$(top_srcdir) $(LIBGSSDP_CFLAGS)
 test_browser_SOURCES = test-browser.c
 test_browser_LDADD = $(top_builddir)/libgssdp/libgssdp-1.0.la
 test_publish_SOURCES = test-publish.c
diff -Nurd -x'*~' gssdp-0.8.0.orig/tools/Makefile.in gssdp-0.8.0/tools/Makefile.in
--- gssdp-0.8.0.orig/tools/Makefile.in	2010-09-16 08:10:09.000000000 -0400
+++ gssdp-0.8.0/tools/Makefile.in	2011-03-21 12:49:19.000000000 -0400
@@ -247,7 +247,7 @@
 top_srcdir = @top_srcdir@
 uidir = $(datadir)/gssdp
 dist_ui_DATA = $(srcdir)/gssdp-device-sniffer.ui
-AM_CFLAGS = $(LIBGSSDP_CFLAGS) $(LIBGTK_CFLAGS) -I$(top_srcdir) -DUI_DIR='"$(uidir)"'
+AM_CPPFLAGS = -I$(top_srcdir) $(LIBGSSDP_CFLAGS) $(LIBGTK_CFLAGS) -DUI_DIR='"$(uidir)"'
 gssdp_device_sniffer_SOURCES = gssdp-device-sniffer.c
 gssdp_device_sniffer_LDADD = $(LIBGSSDP_LIBS) $(LIBGTK_LIBS) $(top_builddir)/libgssdp/libgssdp-1.0.la
 gssdp_device_sniffer_LDFLAGS = -export-dynamic
