diff -Nurd gupnp-0.14.1.orig/libgupnp/gupnp-device-info.c gupnp-0.14.1/libgupnp/gupnp-device-info.c
--- gupnp-0.14.1.orig/libgupnp/gupnp-device-info.c	2010-09-21 09:59:15
+++ gupnp-0.14.1/libgupnp/gupnp-device-info.c	2025-10-03 18:12:25
@@ -27,6 +27,7 @@
  * device information.
  */
 
+#include <stdlib.h>
 #include <string.h>
 
 #include "gupnp-device-info.h"
diff -Nurd gupnp-0.14.1.orig/libgupnp/gupnp-service-proxy.c gupnp-0.14.1/libgupnp/gupnp-service-proxy.c
--- gupnp-0.14.1.orig/libgupnp/gupnp-service-proxy.c	2010-11-30 14:00:47
+++ gupnp-0.14.1/libgupnp/gupnp-service-proxy.c	2025-10-03 18:15:30
@@ -30,6 +30,7 @@
 
 #include <libsoup/soup.h>
 #include <gobject/gvaluecollector.h>
+#include <stdlib.h>
 #include <string.h>
 #include <locale.h>
 
diff -Nurd gupnp-0.14.1.orig/libgupnp/xml-util.c gupnp-0.14.1/libgupnp/xml-util.c
--- gupnp-0.14.1.orig/libgupnp/xml-util.c	2010-11-30 14:00:47
+++ gupnp-0.14.1/libgupnp/xml-util.c	2025-10-03 18:13:59
@@ -19,6 +19,7 @@
  * Boston, MA 02111-1307, USA.
  */
 
+#include <stdlib.h>
 #include <string.h>
 
 #include "gvalue-util.h"
