diff -ruN phc-winner-argon2-20190702-orig/Makefile phc-winner-argon2-20190702/Makefile
--- phc-winner-argon2-20190702-orig/Makefile	2019-05-20 04:18:00.000000000 -0500
+++ phc-winner-argon2-20190702/Makefile	2021-03-27 06:45:59.000000000 -0500
@@ -83,7 +83,7 @@
 endif
 ifeq ($(KERNEL_NAME), Darwin)
 	LIB_EXT := $(ABI_VERSION).dylib
-	LIB_CFLAGS := -dynamiclib -install_name @rpath/lib$(LIB_NAME).$(LIB_EXT)
+	LIB_CFLAGS = -dynamiclib -install_name $(PREFIX)/$(LIBRARY_REL)/lib$(LIB_NAME).$(LIB_EXT)
 	LINKED_LIB_EXT := dylib
 	PC_EXTRA_LIBS ?=
 endif
