diff -Nurd gpgme-1.7.1.orig/configure.ac gpgme-1.7.1/configure.ac
--- gpgme-1.7.1.orig/configure.ac	2016-10-18 13:22:22.000000000 -0400
+++ gpgme-1.7.1/configure.ac	2025-07-19 09:26:48.000000000 -0400
@@ -157,6 +157,8 @@
 	have_ld_version_script=yes
         ;;
     *-apple-darwin*)
+        AC_DEFINE(_DARWIN_C_SOURCE, 1,
+                  Expose all libc features (__DARWIN_C_FULL).)
         AC_DEFINE(_XOPEN_SOURCE, 500, Activate POSIX interface on MacOS X)
         ;;
 esac
