diff -Nurd gssdp-0.10.0.orig/libgssdp/Makefile.in gssdp-0.10.0/libgssdp/Makefile.in
--- gssdp-0.10.0.orig/libgssdp/Makefile.in	2011-04-05 11:32:24
+++ gssdp-0.10.0/libgssdp/Makefile.in	2025-12-19 00:28:03
@@ -264,7 +264,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 LTVERSION = 2:0:0
-AM_CFLAGS = $(LIBGSSDP_CFLAGS) -I$(top_srcdir)
+AM_CPPFLAGS = -I$(top_srcdir) $(LIBGSSDP_CFLAGS)
 libgssdpincdir = $(includedir)/gssdp-1.0/libgssdp
 lib_LTLIBRARIES = libgssdp-1.0.la
 libgssdpinc_HEADERS = gssdp-client.h		 \
diff -Nurd gssdp-0.10.0.orig/tests/Makefile.in gssdp-0.10.0/tests/Makefile.in
--- gssdp-0.10.0.orig/tests/Makefile.in	2011-04-05 11:32:24
+++ gssdp-0.10.0/tests/Makefile.in	2025-12-19 00:28:03
@@ -220,7 +220,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-AM_CFLAGS = $(LIBGSSDP_CFLAGS) -I$(top_srcdir)
+AM_CPPFLAGS = -I$(top_srcdir) $(LIBGSSDP_CFLAGS)
 test_browser_SOURCES = test-browser.c
 test_browser_LDADD = $(top_builddir)/libgssdp/libgssdp-1.0.la $(LIBGSSDP_LIBS)
 test_publish_SOURCES = test-publish.c
diff -Nurd gssdp-0.10.0.orig/tools/Makefile.in gssdp-0.10.0/tools/Makefile.in
--- gssdp-0.10.0.orig/tools/Makefile.in	2011-04-05 11:32:24
+++ gssdp-0.10.0/tools/Makefile.in	2025-12-19 00:28:03
@@ -247,7 +247,7 @@
 top_srcdir = @top_srcdir@
 uidir = $(datadir)/gssdp
 dist_ui_DATA = $(srcdir)/gssdp-device-sniffer.ui
-AM_CFLAGS = $(LIBGSSDP_CFLAGS) $(LIBGTK_CFLAGS) -I$(top_srcdir) -DUI_DIR='"$(uidir)"'
+AM_CPPFLAGS = -I$(top_srcdir) $(LIBGSSDP_CFLAGS) $(LIBGTK_CFLAGS) -DUI_DIR='"$(uidir)"'
 gssdp_device_sniffer_SOURCES = gssdp-device-sniffer.c
 gssdp_device_sniffer_LDADD = $(LIBGSSDP_LIBS) $(LIBGTK_LIBS) $(top_builddir)/libgssdp/libgssdp-1.0.la
 gssdp_device_sniffer_LDFLAGS = -export-dynamic
