diff -ruN nettle-3.5.1-orig/configure nettle-3.5.1/configure
--- nettle-3.5.1-orig/configure	2019-06-27 00:35:06.000000000 -0500
+++ nettle-3.5.1/configure	2019-09-04 05:29:08.000000000 -0500
@@ -2806,9 +2806,7 @@
 $as_echo "Searching for libraries" >&6; }
 
 for d in `echo $with_lib_path | sed 's/:/ /g'` \
-    `echo $exec_prefix | sed "s@^NONE@$prefix/lib@g" | sed "s@^NONE@$ac_default_prefix/lib@g"` \
-    /usr/local/lib /sw/local/lib /sw/lib \
-    /usr/gnu/lib /opt/gnu/lib /sw/gnu/lib /usr/freeware/lib /usr/pkg/lib ; do
+    `echo $exec_prefix | sed "s@^NONE@$prefix/lib@g" | sed "s@^NONE@$ac_default_prefix/lib@g"` ; do
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking $d" >&5
 $as_echo_n "checking $d... " >&6; }
 ac_exists=no
