diff -Nurd openssl-3.0.0.orig/crypto/bn/asm/rsaz-avx512.pl openssl-3.0.0/crypto/bn/asm/rsaz-avx512.pl
--- openssl-3.0.0.orig/crypto/bn/asm/rsaz-avx512.pl	2021-09-07 07:46:32.000000000 -0400
+++ openssl-3.0.0/crypto/bn/asm/rsaz-avx512.pl	2021-10-20 22:29:18.000000000 -0400
@@ -49,7 +49,7 @@
 }
 
 if (!$avx512 && `$ENV{CC} -v 2>&1` =~ /((?:clang|LLVM) version|.*based on LLVM) ([0-9]+\.[0-9]+)/) {
-    $avx512ifma = ($2>=7.0);
+    $avx512ifma = 0;
 }
 
 open OUT,"| \"$^X\" \"$xlate\" $flavour \"$output\""
