diff -Nurd pidgin-2.10.11.orig/configure pidgin-2.10.11/configure
--- pidgin-2.10.11.orig/configure	2014-11-23 11:42:00
+++ pidgin-2.10.11/configure	2025-10-03 23:55:35
@@ -8147,16 +8147,15 @@
     darwin1.*)
       _lt_dar_allow_undefined='${wl}-flat_namespace ${wl}-undefined ${wl}suppress' ;;
     darwin*) # darwin 5.x on
-      # if running on 10.5 or later, the deployment target defaults
-      # to the OS version, if on x86, and 10.4, the deployment
-      # target defaults to 10.4. Don't you love it?
-      case ${MACOSX_DEPLOYMENT_TARGET-10.0},$host in
-	10.0,*86*-darwin8*|10.0,*-darwin[91]*)
-	  _lt_dar_allow_undefined='${wl}-undefined ${wl}dynamic_lookup' ;;
-	10.[012]*)
-	  _lt_dar_allow_undefined='${wl}-flat_namespace ${wl}-undefined ${wl}suppress' ;;
-	10.*)
-	  _lt_dar_allow_undefined='${wl}-undefined ${wl}dynamic_lookup' ;;
+      case $MACOSX_DEPLOYMENT_TARGET,$host in
+        10.[[012]],*|,*powerpc*-darwin[[5-8]]*)
+          _lt_dar_allow_undefined='$wl-flat_namespace $wl-undefined ${wl}suppress' ;;
+        *)
+          _lt_dar_allow_undefined='$wl-undefined ${wl}dynamic_lookup'
+          if test yes = "$lt_cv_support_no_fixup_chains"; then
+            _lt_dar_allow_undefined="$_lt_dar_allow_undefined $wl-no_fixup_chains"
+          fi
+        ;;
       esac
     ;;
   esac
@@ -14261,6 +14260,7 @@
 
 
 
+if false ; then
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for fink" >&5
 $as_echo_n "checking for fink... " >&6; }
 	if test -d /sw; then
@@ -14272,6 +14272,7 @@
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: not found" >&5
 $as_echo "not found" >&6; }
 	fi
+fi
 	;;
 *)
 	;;
@@ -22125,7 +22126,7 @@
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for Tk linkability" >&5
 $as_echo_n "checking for Tk linkability... " >&6; }
 		oldCPPFLAGS=$CPPFLAGS
-		CPPFLAGS="$CPPFLAGS $TCL_CFLAGS"
+		CPPFLAGS="$CPPFLAGS $TCL_CFLAGS $X11_CFLAGS"
 		oldLIBS=$LIBS
 		LIBS="$LIBS $TCL_LIB_SPEC $TK_LIB_SPEC"
 		cat confdefs.h - <<_ACEOF >conftest.$ac_ext
@@ -22149,6 +22150,7 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 		CPPFLAGS="$oldCPPFLAGS"
+		TCL_CFLAGS="$TCL_CFLAGS $X11_CFLAGS"
 		LIBS="$oldLIBS"
 	fi
 else
diff -Nurd pidgin-2.10.11.orig/finch/libgnt/gntwm.c pidgin-2.10.11/finch/libgnt/gntwm.c
--- pidgin-2.10.11.orig/finch/libgnt/gntwm.c	2014-11-23 11:41:25
+++ pidgin-2.10.11/finch/libgnt/gntwm.c	2025-10-03 22:31:07
@@ -36,6 +36,12 @@
 #define _XOPEN_SOURCE_EXTENDED
 #endif
 
+/*
+	Don't know why this is avoided when using Python (above),
+	but definition is needed for <ncurses.h> definition of cchar_t.
+ */
+#define _XOPEN_SOURCE_EXTENDED
+
 #include <glib.h>
 #if GLIB_CHECK_VERSION(2,6,0)
 #	include <glib/gstdio.h>
diff -Nurd pidgin-2.10.11.orig/libpurple/protocols/bonjour/jabber.c pidgin-2.10.11/libpurple/protocols/bonjour/jabber.c
--- pidgin-2.10.11.orig/libpurple/protocols/bonjour/jabber.c	2014-11-23 11:41:25
+++ pidgin-2.10.11/libpurple/protocols/bonjour/jabber.c	2025-10-03 22:31:07
@@ -23,9 +23,9 @@
 #include "internal.h"
 
 #ifndef _WIN32
+#include <sys/socket.h>
 #include <net/if.h>
 #include <sys/ioctl.h>
-#include <sys/socket.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #else
diff -Nurd pidgin-2.10.11.orig/libpurple/protocols/bonjour/parser.c pidgin-2.10.11/libpurple/protocols/bonjour/parser.c
--- pidgin-2.10.11.orig/libpurple/protocols/bonjour/parser.c	2014-11-23 11:41:25
+++ pidgin-2.10.11/libpurple/protocols/bonjour/parser.c	2025-10-03 22:53:51
@@ -197,7 +197,7 @@
 	NULL,									/*_private*/
 	bonjour_parser_element_start_libxml,	/*startElementNs*/
 	bonjour_parser_element_end_libxml,		/*endElementNs*/
-	bonjour_parser_structured_error_handler /*serror*/
+	(xmlStructuredErrorFunc)bonjour_parser_structured_error_handler /*serror*/
 };
 
 void
diff -Nurd pidgin-2.10.11.orig/libpurple/protocols/irc/msgs.c pidgin-2.10.11/libpurple/protocols/irc/msgs.c
--- pidgin-2.10.11.orig/libpurple/protocols/irc/msgs.c	2014-11-23 11:41:26
+++ pidgin-2.10.11/libpurple/protocols/irc/msgs.c	2025-10-03 23:28:34
@@ -1596,22 +1596,22 @@
 	irc->sasl_cb = g_new0(sasl_callback_t, 5);
 
 	irc->sasl_cb[id].id = SASL_CB_AUTHNAME;
-	irc->sasl_cb[id].proc = irc_sasl_cb_simple;
+	irc->sasl_cb[id].proc = (void *)irc_sasl_cb_simple;
 	irc->sasl_cb[id].context = (void *)irc;
 	id++;
 
 	irc->sasl_cb[id].id = SASL_CB_USER;
-	irc->sasl_cb[id].proc = irc_sasl_cb_simple;
+	irc->sasl_cb[id].proc = (void *)irc_sasl_cb_simple;
 	irc->sasl_cb[id].context = (void *)irc;
 	id++;
 
 	irc->sasl_cb[id].id = SASL_CB_PASS;
-	irc->sasl_cb[id].proc = irc_sasl_cb_secret;
+	irc->sasl_cb[id].proc = (void *)irc_sasl_cb_secret;
 	irc->sasl_cb[id].context = (void *)irc;
 	id++;
 
 	irc->sasl_cb[id].id = SASL_CB_LOG;
-	irc->sasl_cb[id].proc = irc_sasl_cb_log;
+	irc->sasl_cb[id].proc = (void *)irc_sasl_cb_log;
 	irc->sasl_cb[id].context = (void *)irc;
 	id++;
 
diff -Nurd pidgin-2.10.11.orig/libpurple/protocols/jabber/auth_cyrus.c pidgin-2.10.11/libpurple/protocols/jabber/auth_cyrus.c
--- pidgin-2.10.11.orig/libpurple/protocols/jabber/auth_cyrus.c	2014-11-23 11:41:26
+++ pidgin-2.10.11/libpurple/protocols/jabber/auth_cyrus.c	2025-10-03 23:29:42
@@ -373,30 +373,30 @@
 
 	id = 0;
 	js->sasl_cb[id].id = SASL_CB_GETREALM;
-	js->sasl_cb[id].proc = jabber_sasl_cb_realm;
+	js->sasl_cb[id].proc = (void *)jabber_sasl_cb_realm;
 	js->sasl_cb[id].context = (void *)js;
 	id++;
 
 	js->sasl_cb[id].id = SASL_CB_AUTHNAME;
-	js->sasl_cb[id].proc = jabber_sasl_cb_simple;
+	js->sasl_cb[id].proc = (void *)jabber_sasl_cb_simple;
 	js->sasl_cb[id].context = (void *)js;
 	id++;
 
 	js->sasl_cb[id].id = SASL_CB_USER;
-	js->sasl_cb[id].proc = jabber_sasl_cb_simple;
+	js->sasl_cb[id].proc = (void *)jabber_sasl_cb_simple;
 	js->sasl_cb[id].context = (void *)js;
 	id++;
 
 	account = purple_connection_get_account(js->gc);
 	if (purple_account_get_password(account) != NULL ) {
 		js->sasl_cb[id].id = SASL_CB_PASS;
-		js->sasl_cb[id].proc = jabber_sasl_cb_secret;
+		js->sasl_cb[id].proc = (void *)jabber_sasl_cb_secret;
 		js->sasl_cb[id].context = (void *)js;
 		id++;
 	}
 
 	js->sasl_cb[id].id = SASL_CB_LOG;
-	js->sasl_cb[id].proc = jabber_sasl_cb_log;
+	js->sasl_cb[id].proc = (void *)jabber_sasl_cb_log;
 	js->sasl_cb[id].context = (void*)js;
 	id++;
 
diff -Nurd pidgin-2.10.11.orig/libpurple/protocols/jabber/parser.c pidgin-2.10.11/libpurple/protocols/jabber/parser.c
--- pidgin-2.10.11.orig/libpurple/protocols/jabber/parser.c	2014-11-23 11:41:26
+++ pidgin-2.10.11/libpurple/protocols/jabber/parser.c	2025-10-03 22:54:08
@@ -246,7 +246,7 @@
 	NULL,									/*_private*/
 	jabber_parser_element_start_libxml,		/*startElementNs*/
 	jabber_parser_element_end_libxml,		/*endElementNs*/
-	jabber_parser_structured_error_handler	/*serror*/
+	(xmlStructuredErrorFunc)jabber_parser_structured_error_handler	/*serror*/
 };
 
 void
diff -Nurd pidgin-2.10.11.orig/libpurple/xmlnode.c pidgin-2.10.11/libpurple/xmlnode.c
--- pidgin-2.10.11.orig/libpurple/xmlnode.c	2014-11-23 11:41:26
+++ pidgin-2.10.11/libpurple/xmlnode.c	2025-10-03 22:58:33
@@ -699,7 +699,7 @@
 	NULL, /* _private */
 	xmlnode_parser_element_start_libxml, /* startElementNs */
 	xmlnode_parser_element_end_libxml,   /* endElementNs   */
-	xmlnode_parser_structural_error_libxml, /* serror */
+	(xmlStructuredErrorFunc)xmlnode_parser_structural_error_libxml, /* serror */
 };
 
 xmlnode *
diff -Nurd pidgin-2.10.11.orig/pidgin/gtkidle.c pidgin-2.10.11/pidgin/gtkidle.c
--- pidgin-2.10.11.orig/pidgin/gtkidle.c	2014-11-23 11:41:26
+++ pidgin-2.10.11/pidgin/gtkidle.c	2025-10-03 23:02:47
@@ -74,11 +74,11 @@
 # ifdef HAVE_IOKIT
 	/* Query the IOKit API */
 
-	static io_service_t macIOsrvc = NULL;
+	static io_service_t macIOsrvc = 0;
 	CFTypeRef property;
 	uint64_t idle_time = 0; /* nanoseconds */
 
-	if (macIOsrvc == NULL)
+	if (macIOsrvc == 0)
 	{
 		mach_port_t master;
 		IOMasterPort(MACH_PORT_NULL, &master);
diff -Nurd pidgin-2.10.11.orig/po/POTFILES.in pidgin-2.10.11/po/POTFILES.in
--- pidgin-2.10.11.orig/po/POTFILES.in	2014-11-23 11:41:27
+++ pidgin-2.10.11/po/POTFILES.in	2025-10-03 22:31:07
@@ -174,6 +174,7 @@
 libpurple/protocols/yahoo/libyahoo.c
 libpurple/protocols/yahoo/libyahoojp.c
 libpurple/protocols/yahoo/libymsg.c
+libpurple/protocols/yahoo/util.c
 libpurple/protocols/yahoo/yahoo_aliases.c
 libpurple/protocols/yahoo/yahoo_doodle.c
 libpurple/protocols/yahoo/yahoo_filexfer.c
diff -Nurd pidgin-2.10.11.orig/po/POTFILES.skip pidgin-2.10.11/po/POTFILES.skip
--- pidgin-2.10.11.orig/po/POTFILES.skip	2014-11-23 11:41:27
+++ pidgin-2.10.11/po/POTFILES.skip	2025-10-03 22:31:07
@@ -37,6 +37,7 @@
 libpurple/plugins/perl/common/Whiteboard.c
 libpurple/plugins/perl/common/XMLNode.c
 libpurple/protocols/null/nullprpl.c
+build/pidgin/data/pidgin.desktop.in
 pidgin/data/pidgin.desktop.in.in
 pidgin/plugins/crazychat/cc_pidgin_plugin.c
 pidgin/plugins/perl/common/GtkAccount.c
