diff --git a/src/masternodeman.cpp b/src/masternodeman.cpp
index 060446d31..9a98535b0 100644
--- a/src/masternodeman.cpp
+++ b/src/masternodeman.cpp
@@ -1550,6 +1550,9 @@ void CMasternodeMan::UpdateWatchdogVoteTime(const CTxIn& vin)
 
 bool CMasternodeMan::IsWatchdogActive()
 {
+    if(!sporkManager.IsSporkActive(SPORK_7_REQUIRE_SENTINEL_FLAG))
+        return false;
+
     LOCK(cs);
     // Check if any masternodes have voted recently, otherwise return false
     return (GetTime() - nLastWatchdogVoteTime) <= MASTERNODE_WATCHDOG_MAX_SECONDS;
