diff -Nurd yaz-5.31.1.orig/yaz-icu.pc.in yaz-5.31.1/yaz-icu.pc.in
--- yaz-5.31.1.orig/yaz-icu.pc.in	2021-12-02 09:20:41.000000000 -0500
+++ yaz-5.31.1/yaz-icu.pc.in	2022-05-01 13:03:33.000000000 -0400
@@ -6,6 +6,7 @@
 Name: YAZ ICU
 Version: @VERSION@
 Description: YAZ ICU library for building Z39.50 applications
-Requires: yaz libexslt icu-i18n
+Requires: yaz
+Requires.private: icu-i18n
 Libs: -L${libdir} -lyaz_icu @PTHREAD_LIBS@
 Cflags: @ICU_CPPFLAGS@
diff -Nurd yaz-5.31.1.orig/yaz-server.pc.in yaz-5.31.1/yaz-server.pc.in
--- yaz-5.31.1.orig/yaz-server.pc.in	2021-12-02 09:20:41.000000000 -0500
+++ yaz-5.31.1/yaz-server.pc.in	2022-05-01 12:47:44.000000000 -0400
@@ -6,5 +6,5 @@
 Name: YAZ server
 Version: @VERSION@
 Description: YAZ server library for building Z39.50 applications
-Requires: yaz libexslt
+Requires: yaz
 Libs: -L${libdir} -lyaz_server @PTHREAD_LIBS@
diff -Nurd yaz-5.31.1.orig/yaz.pc.in yaz-5.31.1/yaz.pc.in
--- yaz-5.31.1.orig/yaz.pc.in	2021-12-02 09:20:41.000000000 -0500
+++ yaz-5.31.1/yaz.pc.in	2022-05-01 13:03:07.000000000 -0400
@@ -6,7 +6,6 @@
 Name: YAZ
 Version: @VERSION@
 Description: YAZ library for building Z39.50 applications
-Requires: libexslt
+Requires.private: libxml2
 Libs: -L${libdir} -lyaz
-Libs.private: @LIBS@
 Cflags: -I${includedir} @YAZ_CONFIG_CFLAGS@
