diff -ru xmame-0.55.2/makefile.unix xmame-0.55.2-patched/makefile.unix
--- xmame-0.55.2/makefile.unix	Sun Sep 16 17:52:04 2001
+++ xmame-0.55.2-patched/makefile.unix	Wed Nov 21 16:50:51 2001
@@ -43,9 +43,9 @@
 
 # *** Choose your compiler
 # Use gcc if available
-CC	= gcc
+# CC	= gcc
 # otherwise try the default cc
-# CC    = cc
+CC    = cc
 # Use of `c89' is recommend for ultrix as it generates faster code (which
 # means less frames to be skipped and better graphics) but `gcc' works just
 # as well. However, stay away from the `cc' ultrix compiler if possible.
@@ -74,7 +74,7 @@
 # try using -O instead. 
 #
 ### for normal optimization, full warnings
-CFLAGS    = -O -Wall
+# CFLAGS    = -O -Wall
 ### for common optimizations, full warnings except unused vars
 # CFLAGS    = -O2 -Wall -Wno-unused
 ### this should work with all compilers 
@@ -84,7 +84,7 @@
 #  -fstrength-reduce -fomit-frame-pointer -ffast-math -malign-functions=2 \
 #  -malign-jumps=2 -malign-loops=2
 ### for Linux/PowerPC use following opts
-# CFLAGS    = -O -Wall  -Wno-unused -funroll-loops\
+CFLAGS    = -O -Wall  -Wno-unused -funroll-loops -no-cpp-precomp
 #  -fstrength-reduce -fomit-frame-pointer -ffast-math -fsigned-char
 ### for OpenStep/Intel use following opts
 # CFLAGS    = -O -Wall -Wno-unused -finline-functions -ffast-math \
@@ -116,11 +116,11 @@
 # *** Choose your linker & linker flags (some OSes need native linker
 # instead GNU one)
 # normal
-LD	= $(CC) -Wl,-s
+# LD	= $(CC) -Wl,-s
 # to profile with gcc and gprof
 # LD	= $(CC) -pg
 # no profiling (less strace output), but with debug info
-# LD	= $(CC)
+LD	= $(CC)
 # for debugging only, only works with gnu-binutils !!
 # LD	= $(CC) -Wl,-warn-common
 # for IRIX R5+K MIPSpro, optimization at link time, takes nearly a gig of 
@@ -133,7 +133,7 @@
 # *** Does your platform have a separate libm ? only comment this out if you
 # have maths libraries included within libc. This is very rare, you will
 # probably only need it for OpenStep systems
-SEPARATE_LIBM = 1
+# SEPARATE_LIBM = 1
 
 # *** Does your platform have ranlib (leave this untouched unless you are
 # using irix or Unixware 7
@@ -157,8 +157,8 @@
 # *** Extra include and or library paths, sometimes needed for zlib,
 # with suse linux -L/usr/X11R6/lib is needed for example,
 # and with unixware7 -lsocket is needed
-# LIBS     = -L/usr/X11R6/lib -L/usr/local/lib
-# INCLUDES = -I/usr/X11R6/include -I/usr/local/include
+LIBS     = -L/usr/X11R6/lib -L@FINKPREFIX@/lib
+INCLUDES = -I/usr/X11R6/include -I@FINKPREFIX@/include
 
 # *** If your system doesn't have zlib, or your systems zlib is giving you
 # troubles, you can use the one distributed with xmame.
@@ -181,10 +181,10 @@
 # and binary distribution ( if you want to... )
 # ( only needed to install, not to compile... )
 ifndef PREFIX
-PREFIX = /usr/local
+PREFIX = @FINKPREFIX@
 endif
 BINDIR = $(PREFIX)/bin
-MANDIR = $(PREFIX)/man/man6
+MANDIR = $(PREFIX)/share/man/man6
 
 # *** xmameroot, this is the default rompath, place where the highscores are
 # stored, global rc location etc. Since most of these are configurable through
@@ -197,7 +197,7 @@
 ##############################################################################
 
 ### i386 + gnu-asm
-MY_CPU = i386
+# MY_CPU = i386
 ### i386 no asm
 # MY_CPU = i386_noasm
 ### ia64
@@ -207,7 +207,7 @@
 ### m68k
 # MY_CPU = m68k
 ### generic risc (powerpc, sparc, hppa, ibm)
-# MY_CPU = risc
+MY_CPU = risc
 ### generic risc, lsb-first (RISC (ultrix machines))
 # MY_CPU = risc_lsb
 ### mips (generic risc + SGI compiler bug workarounds)
@@ -218,7 +218,7 @@
 ##############################################################################
 
 ### Linux 
-ARCH  = linux
+# ARCH  = linux
 ### FreeBSD
 # ARCH  = freebsd
 ### NetBSD
@@ -240,7 +240,7 @@
 ### aix ( with sound, you'll need the UMS and SOM lpp's installed ( under AIX4 ))
 # ARCH  = aix
 ### generic unix, no sound
-# ARCH  = generic
+ARCH  = generic
 
 # *) For IRIX 6.5 or higher add -DHAVE_SNPRINTF to CFLAGS.irix(_al) in
 # src/unix/Makefile
@@ -268,7 +268,7 @@
 
 # *** any additonal sound drivers you want to include, besided the native one
 # sound driver for your system
-# SOUND_ESOUND = 1
+SOUND_ESOUND = 1
 # SOUND_ALSA = 1
 # SOUND_ARTS_TEIRA = 1
 # SOUND_ARTS_SMOTEK = 1
@@ -373,12 +373,12 @@
 # And choose elf / or aout
 # ELF settings: no leading underscores + elf object format Tested with
 # linux/i386. Also for e.g. Solaris/x86
-ASM_STRIP = src/unix/contrib/porting/strip_
-NASM_FMT  = -f elf
+# ASM_STRIP = src/unix/contrib/porting/strip_
+# NASM_FMT  = -f elf
 # aout settings: with leading underscores + aout object format. Tested with
 # netBSD i386. Also for older linux installs, freebsd and others.
-# ASM_STRIP = true  # no strip command
-# NASM_FMT  = -f aoutb
+ASM_STRIP = true  # no strip command
+NASM_FMT  = -f aoutb
 
 ifdef MAME_DEBUG
 NAME=x$(TARGET)d
