diff -Naur allegro-4.2.0/include/allegro/platform/alosx.h allegro-new/include/allegro/platform/alosx.h
--- allegro-4.2.0/include/allegro/platform/alosx.h	2004-07-31 13:39:06.000000000 -0400
+++ allegro-new/include/allegro/platform/alosx.h	2005-05-23 14:16:09.000000000 -0400
@@ -66,7 +66,16 @@
    #define ALLEGRO_MAGIC_MAIN
    #define main _mangled_main
    #undef END_OF_MAIN
-   #define END_OF_MAIN() void *_mangled_main_address = (void*) _mangled_main;
+
+   #ifdef ALLEGRO_USE_CONSOLE
+      #define END_OF_MAIN() \
+         void *_mangled_main_address = (void*) _mangled_main; \
+         int allegro_osx_dock = 0;
+   #else
+      #define END_OF_MAIN() \
+         void *_mangled_main_address = (void*) _mangled_main; \
+         int allegro_osx_dock = 1;
+   #endif
 #else
    #undef END_OF_MAIN
    #define END_OF_MAIN() void *_mangled_main_address;
diff -Naur allegro-4.2.0/makefile.osx allegro-new/makefile.osx
--- allegro-4.2.0/makefile.osx	2005-04-23 04:22:46.000000000 -0400
+++ allegro-new/makefile.osx	2005-05-23 15:33:16.000000000 -0400
@@ -51,9 +51,9 @@
 LIB_NAME = lib/macosx/lib$(VERSION)-$(shared_version).dylib
 LIB_MAIN_NAME = lib/macosx/lib$(VERSION)-main.a
 
-DYLINK_FLAGS = -prebind -seg1addr 0x30000000 -dylib_compatibility_version=4.0.0 -dylib_current_version=$(shared_version)
+DYLINK_FLAGS = -prebind -compatibility_version 4.0.0 -current_version $(shared_version)
 
-INSTALL_NAME = -install_name lib$(VERSION)-$(shared_major_minor).dylib
+INSTALL_NAME = -install_name @PREFIX@/lib/lib$(VERSION)-4.dylib
 INSTALL_NAME_EMBED = -install_name "`echo "@executable_path/../Frameworks/$(FRAMEWORK_NAME).framework/Versions/$(shared_version)/$(FRAMEWORK_NAME)" | sed 's!//*!/!g'`"
 
 ifdef EMBED
diff -Naur allegro-4.2.0/misc/allegro-config.in allegro-new/misc/allegro-config.in
--- allegro-4.2.0/misc/allegro-config.in	2005-05-14 13:58:00.000000000 -0400
+++ allegro-new/misc/allegro-config.in	2005-05-23 14:16:09.000000000 -0400
@@ -11,10 +11,10 @@
 
 version=4.2.0
 
-prefix=@prefix@
+prefix=@PREFIX@
 exec_prefix=$prefix
 exec_prefix_set=no
-include_prefix=@INCLUDE_PREFIX@
+include_prefix=@PREFIX@
 
 static_libs=@LINK_WITH_STATIC_LIBS@
 lib_type=@LIB_TO_LINK@
diff -Naur allegro-4.2.0/src/macosx/main.m allegro-new/src/macosx/main.m
--- allegro-4.2.0/src/macosx/main.m	2004-07-11 10:36:07.000000000 -0400
+++ allegro-new/src/macosx/main.m	2005-05-23 14:16:09.000000000 -0400
@@ -33,6 +33,9 @@
 extern char **__crt0_argv;
 extern void *_mangled_main_address;
 
+/* Use the dock? Set based on ALLEGRO_USE_CONSOLE */
+extern int allegro_osx_dock;
+
 static char *arg0, *arg1 = NULL;
 static int refresh_rate = 70;
 
@@ -204,15 +207,17 @@
    __crt0_argv = argv;
    
    [NSApplication sharedApplication];
-      
-   /* Tell the dock about us; the origins of this hack are unknown, but it's
-    * currently the only way to make a Cocoa app to work when started from a
-    * console.
-    */
-   if ((!CPSGetCurrentProcess(&psn)) &&
-       (!CPSEnableForegroundOperation(&psn, 0x03, 0x3C, 0x2C, 0x1103)) &&
-       (!CPSSetFrontProcess(&psn)))
-      [NSApplication sharedApplication];
+         
+   if (allegro_osx_dock) {
+      /* Tell the dock about us; the origins of this hack are unknown, but it's
+       * currently the only way to make a Cocoa app to work when started from a
+       * console.
+       */
+      if ((!CPSGetCurrentProcess(&psn)) &&
+          (!CPSEnableForegroundOperation(&psn, 0x03, 0x3C, 0x2C, 0x1103)) &&
+          (!CPSSetFrontProcess(&psn)))
+         [NSApplication sharedApplication];
+   }
    
    /* Creates a custom application menu */
    [NSApp setMainMenu: [[NSMenu allocWithZone: [NSMenu menuZone]] initWithTitle: @"temp"]];
