diff -uNr cgilib-0.5/Makefile cgilib-0.5-new/Makefile
--- cgilib-0.5/Makefile	Fri Aug 20 17:14:07 1999
+++ cgilib-0.5-new/Makefile	Tue Jun  7 15:33:10 2005
@@ -21,7 +21,11 @@
 OBJS = cgi.o cookies.o
 
 libcgi.a: $(OBJS)
-	ar rc $@ $^
+	ar -rc $@ $^
+	ranlib libcgi.a
+
+libcgi.0.5.dylib: $(OBJS)
+	gcc -dynamiclib -install_name @PREFIX@/lib/libcgi.0.5.dylib -current_version 0.5 -o $@ $^
 
 cgitest: cgitest.o libcgi.a
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS)
@@ -29,13 +33,13 @@
 jumpto: jumpto.o libcgi.a
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS)
 
-all: libcgi.a cgitest jumpto
+all: libcgi.a libcgi.0.5.dylib cgitest jumpto
 
 install: cgitest
 	install -m 755 cgitest /usr/lib/cgi-bin
 
 clean:
-	rm -f cgitest cgitest.o jumpto jumpto.o libcgi.a $(OBJS) *.[35].html
+	rm -f cgitest cgitest.o jumpto jumpto.o libcgi.a libcgi.0.5.dylib $(OBJS) *.[35].html
 
 htmlman:
 	for f in *.[35]; do \
diff -uNr cgilib-0.5/cgi.c cgilib-0.5-new/cgi.c
--- cgilib-0.5/cgi.c	Fri Aug 20 17:14:07 1999
+++ cgilib-0.5-new/cgi.c	Tue Jun  7 15:30:28 2005
@@ -31,7 +31,6 @@
 #include <unistd.h>
 #include <string.h>
 #include <ctype.h>
-#include <malloc.h>
 #include <cgi.h>
 
 int cgiDebugLevel = 0;
diff -uNr cgilib-0.5/cgitest.c cgilib-0.5-new/cgitest.c
--- cgilib-0.5/cgitest.c	Tue Aug 17 09:21:20 1999
+++ cgilib-0.5-new/cgitest.c	Tue Jun  7 15:30:28 2005
@@ -23,6 +23,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include <cgi.h>
 
 s_cgi *cgi;
@@ -113,7 +114,7 @@
             cgiHeader();
 	    printf ("<html>\n<head><title>cgilib</title></title>\n\n<body bgcolor=\"#ffffff\">\n");
 	    printf ("<h1><a href=\"%s\">cgilib</a></h1>\n", URL);
-	    printf ("<h3>Cookie "Library" set</h3>\n");
+	    printf ("<h3>Cookie \"Library\" set</h3>\n");
 	    printf ("<p><br><p><br><a href=\"/cgi-bin/cgitest\">Test</a><p>\n");
 	    printf ("<p><br><p><br><a href=\"/cgi-bin/cgitest/redirect\">Redirect</a><p>\n");
 	    printf ("<p><br><p><br><a href=\"/cgi-bin/cgitest/listall\">List Everything</a><p>\n");
