diff -ruN id3lib-3.7.13/src/tag_parse_lyrics3.cpp id3lib-3.7.13.new/src/tag_parse_lyrics3.cpp
--- id3lib-3.7.13/src/tag_parse_lyrics3.cpp	Mon Sep 11 02:34:06 2000
+++ id3lib-3.7.13.new/src/tag_parse_lyrics3.cpp	Thu Sep 26 23:33:14 2002
@@ -200,7 +200,7 @@
       // reserve enough space for lyrics3 + id3v1 tag
       const size_t max_lyr_size = 11 + 5100 + 9 + 128;
 
-      size_t lyr_buffer_size = MIN(max_lyr_size, file.tellg());
+      size_t lyr_buffer_size = MIN(max_lyr_size, (size_t)file.tellg());
 
       file.seekg(- static_cast<long>(lyr_buffer_size), ios::cur);
       if (!file)
@@ -274,7 +274,7 @@
       
       // Using binary minus rather than unary minus to avoid compiler warning
       
-      file.seekg(- MIN(file.tellg(), 
+      file.seekg(- MIN((size_t)file.tellg(), 
                        (static_cast<long>(lyr_size) + 6 + 9 + 3)), ios::cur);
       if (!file)
       {
diff -ruN id3lib-3.7.13/src/utils.cpp id3lib-3.7.13.new/src/utils.cpp
--- id3lib-3.7.13/src/utils.cpp	Mon Sep 11 03:46:32 2000
+++ id3lib-3.7.13.new/src/utils.cpp	Tue Oct  1 21:47:19 2002
@@ -25,6 +25,15 @@
 // http://download.sourceforge.net/id3lib/
 
 #include <ctype.h>
+#include <fstream>
+using namespace std;
+
+#if (defined(__GNUC__) && __GNUC__ == 2)
+#define NOCREATE ios::nocreate
+#else
+#define NOCREATE ((std::ios_base::openmode)0)
+#endif
+
 #include "utils.h"
 
 #if defined HAVE_CONFIG_H
@@ -231,7 +240,7 @@
 
   bool exists(const char *name)
   {
-    ifstream file(name, ios::nocreate);
+    ifstream file(name, NOCREATE);
     return file.is_open() != 0;
   }
   
@@ -275,7 +284,7 @@
     {
       file.close();
     }
-    file.open(name, ios::in | ios::out | ios::binary | ios::nocreate);
+    file.open(name, ios::in | ios::out | ios::binary | NOCREATE);
     if (!file)
     {
       return ID3E_ReadOnly;
@@ -290,7 +299,7 @@
     {
       file.close();
     }
-    file.open(name, ios::in | ios::binary | ios::nocreate);
+    file.open(name, ios::in | ios::binary | NOCREATE);
     if (!file)
     {
       return ID3E_NoFile;
