--- /sw/src/fink.build/jacorb-2.2.2-1/JacORB/src/org/jacorb/orb/standardInterceptors/SSLComponentInterceptor.java	2004-05-06 05:40:00.000000000 -0700
+++ JacORB/src/org/jacorb/orb/standardInterceptors/SSLComponentInterceptor.java	2005-08-12 21:36:06.000000000 -0700
@@ -122,7 +122,7 @@
 
                 SSLHelper.write( sslDataStream , ssl );
 
-                tc = new TaggedComponent( TAG_SSL_SEC_TRANS.value,
+                tc = new TaggedComponent( org.omg.SSLIOP.TAG_SSL_SEC_TRANS.value,
                                           sslDataStream.getBufferCopy() );
 
                 sslDataStream.close ();
--- /sw/src/fink.build/jacorb-2.2.2-1/JacORB/src/org/jacorb/orb/iiop/IIOPProfile.java	2004-10-21 07:50:44.000000000 -0700
+++ JacORB/src/org/jacorb/orb/iiop/IIOPProfile.java	2005-08-12 21:33:49.000000000 -0700
@@ -195,8 +195,10 @@
             out.beginEncapsulatedArray();
             SSLHelper.write( out, ssl );
 
+            // TAG_SSL_SEC_TRANS must be disambiguated in case OpenORB-generated
+            // OMG classes are in the classpath.
             components.addComponent
-                (new TaggedComponent( TAG_SSL_SEC_TRANS.value,
+                (new TaggedComponent( org.omg.SSLIOP.TAG_SSL_SEC_TRANS.value,
                                       out.getBufferCopy() )
                  );
         }
@@ -444,7 +446,9 @@
 
     public SSL getSSL()
     {
-        return (SSL)components.getComponent( TAG_SSL_SEC_TRANS.value,
+        // TAG_SSL_SEC_TRANS must be disambiguated in case OpenORB-generated
+        // OMG classes are in the classpath.
+        return (SSL)components.getComponent( org.omg.SSLIOP.TAG_SSL_SEC_TRANS.value,
                                              SSLHelper.class );
     }
 
