--- libgdiplus-1.1.13/Makefile.in	2006-01-06 14:01:18.000000000 -0500
+++ libgdiplus-1.1.13-new/Makefile.in	2006-01-12 22:45:55.000000000 -0500
@@ -190,7 +190,7 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-SUBDIRS = cairo src tests
+SUBDIRS = src tests
 #libpixman
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libgdiplus.pc 
--- libgdiplus-1.1.13/configure	2006-01-06 14:01:19.000000000 -0500
+++ libgdiplus-1.1.13-new/configure	2006-01-12 22:47:34.000000000 -0500
@@ -19603,14 +19603,9 @@
 
 
 
-subdirs="$subdirs cairo"
+CAIRO_LIBS="`pkg-config --libs cairo`"
+CAIRO_CFLAGS="`pkg-config --cflags cairo`"
 
-CAIRO_LIBS='$(top_builddir)/cairo/src/libcairo.la'
-CAIRO_CFLAGS='-I$(top_builddir)/cairo/pixman/src -I$(top_srcdir)/cairo/pixman/src -I$(top_builddir)/cairo/src -I$(top_srcdir)/cairo/src'
-
-cat >>confdefs.h <<\_ACEOF
-#define USE_INCLUDED_CAIRO 1
-_ACEOF
 
 
 # Activate this when we support linking to an already exiting Cairo installation
@@ -19635,7 +19630,7 @@
 #	CAIRO_CFLAGS="`pkg-config --cflags cairo`"
 #fi
 
-GDIPLUS_LIBS="$CAIRO_LIBS `pkg-config --libs glib-2.0 ` `freetype-config --libs`"
+GDIPLUS_LIBS="$CAIRO_LIBS `pkg-config --libs glib-2.0 ` `freetype-config --libs` -lfontconfig -lX11"
 GDIPLUS_CFLAGS="$CAIRO_CFLAGS `pkg-config --cflags glib-2.0 ` `freetype-config --cflags`"
 
 
@@ -19994,7 +19989,7 @@
 #define _REENTRANT 1
 _ACEOF
 
-		CFLAGS="-pthreads $CFLAGS"
+		CFLAGS="$CFLAGS"
 		;;
 
 	*-*-darwin*)
@@ -20004,7 +19999,7 @@
 _ACEOF
 
 		CFLAGS="-no-cpp-precomp $CFLAGS"
-		LDFLAGS="-flat_namespace -undefined suppress $LDFLAGS"
+		LDFLAGS="-Xlinker -framework -Xlinker ApplicationServices $LDFLAGS"
 		;;
 
 	*)
@@ -20027,7 +20022,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_save_CFLAGS=$CFLAGS
-   CFLAGS="$CFLAGS -pthread"
+   CFLAGS="$CFLAGS"
 
 
    ac_ext=c
@@ -20096,7 +20091,7 @@
 if test $myapp_cv_gcc_pthread = yes
 then
    myapp_threads=""
-   CFLAGS="$CFLAGS -pthread"
+   CFLAGS="$CFLAGS"
 else
    echo "$as_me:$LINENO: checking for pthread_create in -lpthread" >&5
 echo $ECHO_N "checking for pthread_create in -lpthread... $ECHO_C" >&6
@@ -21534,16 +21529,16 @@
 
 fi
 if test $ac_cv_header_gif_lib_h = yes; then
-  GIF='ungif'; LIBGIF='-lungif'
+  GIF='gif'; LIBGIF='-lgif'
 else
-  { echo "$as_me:$LINENO: WARNING: *** GIF loader will not be built (ungiflibrary not found) ***" >&5
-echo "$as_me: WARNING: *** GIF loader will not be built (ungiflibrary not found) ***" >&2;}
+  { echo "$as_me:$LINENO: WARNING: *** GIF loader will not be built (giflibrary not found) ***" >&5
+echo "$as_me: WARNING: *** GIF loader will not be built (giflibrary not found) ***" >&2;}
 fi
 
 
 else
-  { echo "$as_me:$LINENO: WARNING: *** GIF loader will not be built (ungiflibrary not found) ***" >&5
-echo "$as_me: WARNING: *** GIF loader will not be built (ungiflibrary not found) ***" >&2;}
+  { echo "$as_me:$LINENO: WARNING: *** GIF loader will not be built (giflibrary not found) ***" >&5
+echo "$as_me: WARNING: *** GIF loader will not be built (giflibrary not found) ***" >&2;}
 fi
 
   fi
--- libgdiplus-1.1.13/src/pngcodec.c	2005-11-10 10:23:04.000000000 -0500
+++ libgdiplus-1.1.13-new/src/pngcodec.c	2006-01-12 22:45:55.000000000 -0500
@@ -451,7 +451,7 @@
     GpBitmap *bitmap = (GpBitmap *) image;
     gdip_stream_png_source_ptr pngsrc = NULL;
 
-    int i;
+    int i, j;
 
     int bit_depth;
     int color_type;
