diff -ruN libmovtar-0.1.3.orig/movtar_play.c libmovtar-0.1.3/movtar_play.c
--- libmovtar-0.1.3.orig/movtar_play.c	Fri Jul 13 15:38:03 2001
+++ libmovtar-0.1.3/movtar_play.c	Fri Aug 15 11:38:09 2003
@@ -12,9 +12,11 @@
 #include <SDL/SDL.h>
 #include <SDL/SDL_timer.h>
 
-#ifndef IRIX
+#if !defined(IRIX) && !defined(__APPLE__)
 #define JPEG_INTERNALS
 #include <jinclude.h>
+#else
+#define IRIX
 #endif
 #include <jpeglib.h>
 
@@ -581,7 +583,7 @@
       cconvert = cinfo.cconvert;
       cconvert->color_convert = ycc_rgb16_convert_mmx;
 #else
-      fprintf(stderr, "15/16 bits per pixel can't be decoded by libjpeg on IRIX!");
+      fprintf(stderr, "15/16 bits per pixel can't be decoded by libjpeg on this platform!");
 #endif
       break;
     default: break;
