diff -uNr liboss-0.0.1/include/liboss/soundcard.h liboss-0.0.1-new/include/liboss/soundcard.h
--- liboss-0.0.1/include/liboss/soundcard.h	2003-02-21 22:02:38.000000000 -0500
+++ liboss-0.0.1-new/include/liboss/soundcard.h	2005-04-30 14:13:30.000000000 -0400
@@ -311,66 +311,26 @@
 
 #include <esd.h>
 
-#undef ioctl
-#undef open
-#undef close
-#undef write
-
 #ifndef LIBOSS_IOCTL
-# define LIBOSS_IOCTL ioctl
+# define LIBOSS_IOCTL liboss_ioctl
 #endif
 
 #ifndef LIBOSS_OPEN
-# define LIBOSS_OPEN open
+# define LIBOSS_OPEN liboss_open
 #endif
 
 #ifndef LIBOSS_CLOSE
-# define LIBOSS_CLOSE close
+# define LIBOSS_CLOSE liboss_close
 #endif
 
 #ifndef LIBOSS_WRITE
-# define LIBOSS_WRITE write
+# define LIBOSS_WRITE liboss_write
 #endif
 
-#ifndef LIBOSS_INTERNAL
 extern int liboss_ioctl (int,unsigned long,va_list);
 extern int liboss_open (const char *,int,va_list);
 extern int liboss_close (int);
-extern int liboss_write (int, const void *, size_t);
+extern ssize_t liboss_write (int, const void *, size_t);
 
-static inline int LIBOSS_IOCTL (int x, unsigned long y,...)
-{
-    int result;
-
-    va_list l;
-    va_start(l,y);
-    result = liboss_ioctl(x,y,l);
-    va_end (l);
-    return result;
-}
-
-static inline int LIBOSS_OPEN (const char* x, int y,...)
-{
-    int result;
-
-    va_list l;
-    va_start(l,y);
-    result = liboss_open(x,y,l);
-    va_end (l);
-    return result;
-}
-
-static inline int LIBOSS_CLOSE (int x) {return liboss_close(x);}
-
-static inline int LIBOSS_WRITE (int x, const void *y, size_t l)
-{
-    return liboss_write(x,y,l);
-}
-#else
-extern int liboss_ioctl (int,unsigned long,va_list);
-extern int liboss_open (const char *,int,va_list);
-extern int liboss_close (int);
-extern int liboss_write (int, const void *, size_t);
-#endif
 __END_DECLS
 #endif /* !_SOUNDCARD_H_ */
diff -uNr liboss-0.0.1/lib/esddsp.c liboss-0.0.1-new/lib/esddsp.c
--- liboss-0.0.1/lib/esddsp.c	2003-01-19 14:47:14.000000000 -0500
+++ liboss-0.0.1-new/lib/esddsp.c	2005-04-30 14:19:52.000000000 -0400
@@ -456,7 +456,7 @@
 #endif
 }
 
-int
+ssize_t
 liboss_write (int play, const void *buf, size_t sz)
 {
 #if 0
diff -uNr liboss-0.0.1/lib/ossins.c liboss-0.0.1-new/lib/ossins.c
--- liboss-0.0.1/lib/ossins.c	2003-01-11 14:33:10.000000000 -0500
+++ liboss-0.0.1-new/lib/ossins.c	2005-04-30 14:24:11.000000000 -0400
@@ -23,7 +23,7 @@
 }
 
 int close (int x) {return liboss_close(x);}
-int write (int x, const void *y, size_t l)
+ssize_t write (int x, const void *y, size_t l)
 {
     return liboss_write(x,y,l);
 }
diff -uNr liboss-0.0.1/ltmain.sh liboss-0.0.1-new/ltmain.sh
--- liboss-0.0.1/ltmain.sh	2003-02-22 00:31:16.000000000 -0500
+++ liboss-0.0.1-new/ltmain.sh	2005-04-30 10:42:18.000000000 -0400
@@ -1758,7 +1758,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    need_relink=no
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then
