diff -uNr libvisual-0.2.0/libvisual/lv_cpu.c libvisual-0.2.0-patched/libvisual/lv_cpu.c
--- libvisual-0.2.0/libvisual/lv_cpu.c	Wed Feb  9 14:20:12 2005
+++ libvisual-0.2.0-patched/libvisual/lv_cpu.c	Tue Mar 15 09:33:12 2005
@@ -354,7 +354,7 @@
 #endif
 
 	/* Count the number of CPUs in system */
-#if !defined(VISUAL_OS_WIN32) && !defined(VISUAL_OS_UNKNOWN)
+#if !defined(VISUAL_OS_WIN32) && !defined(VISUAL_OS_UNKNOWN) && !defined(VISUAL_OS_DARWIN)
 	_lv_cpu_caps.nrcpu = sysconf (_SC_NPROCESSORS_ONLN);
 	if (_lv_cpu_caps.nrcpu == -1)
 		_lv_cpu_caps.nrcpu = 1;
@@ -423,7 +423,7 @@
 #endif
 #endif /* VISUAL_ARCH_X86 */
 
-#if VISUAL_ARCH_POWERPC
+#if defined(VISUAL_ARCH_POWERPC) && !defined(VISUAL_OS_DARWIN)
 	check_os_altivec_support ();
 #endif /* VISUAL_ARCH_POWERPC */
 
