diff -Naur openh323/src/Makefile openh323-patched/src/Makefile
--- openh323/src/Makefile	Sun Aug  3 20:03:29 2003
+++ openh323-patched/src/Makefile	Sun Feb 15 00:53:03 2004
@@ -894,7 +894,10 @@
 endif
 
 asnparser.version: $(ASNPARSER)
-	$(ASNPARSER) --version | awk '{print $$1,$$2,$$3}' > asnparser.version.new
+	$(ASNPARSER) --version > tmpversion ; \
+	awk '{print $$1,$$2,$$3}' < tmpversion > asnparser.version.new ; \
+	rm tmpversion ; \
+#	$(ASNPARSER) --version | awk '{print $$1,$$2,$$3}' > asnparser.version.new
 	if test -f asnparser.version && diff asnparser.version.new asnparser.version >/dev/null 2>&1 ; \
 		then rm asnparser.version.new ; \
 		else mv asnparser.version.new asnparser.version ; \
