Index: Package.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Package.pm,v
retrieving revision 1.148
diff -u -r1.148 Package.pm
--- Package.pm	30 Jul 2005 05:45:19 -0000	1.148
+++ Package.pm	3 Aug 2005 00:25:08 -0000
@@ -1023,7 +1023,7 @@
 	
 	# Get data from dpkg's status file. Note that we do *not* store this 
 	# information into the package database.
-	$class->insert_runtime_packages_hash(Fink::Status->list());
+#	$class->insert_runtime_packages_hash(Fink::Status->list());
 
 	# Get data from VirtPackage.pm. Note that we do *not* store this 
 	# information into the package database.
Index: Services.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Services.pm,v
retrieving revision 1.174
diff -u -r1.174 Services.pm
--- Services.pm	30 Jul 2005 05:45:27 -0000	1.174
+++ Services.pm	3 Aug 2005 00:25:08 -0000
@@ -1139,17 +1139,7 @@
 
 sub get_arch {
 	if(not defined $arch) {
-		foreach ('/usr/bin/uname', '/bin/uname') {
-			# check some common places (why aren't we using $ENV{PATH}?)
-			if (-x $_) {
-				chomp($arch = `$_ -p 2>/dev/null`);
-				chomp($arch = `$_ -m 2>/dev/null`) if ($arch eq "");
-				last;
-			}
-		}
-		if (not defined $arch) {
-			die "Could not find an 'arch' executable\n";
-		}
+		$arch = "powerpc";
 	}
 	return $arch;
 }
