? pdb.sf-net.patch
Index: Package.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Package.pm,v
retrieving revision 1.50
diff -u -r1.50 Package.pm
--- Package.pm	18 Nov 2003 09:09:51 -0000	1.50
+++ Package.pm	19 Nov 2003 10:54:11 -0000
@@ -330,7 +330,8 @@
 
 	# Get data from dpkg's status file. Note that we do *not* store this 
 	# information into the package database.
-	$dlist = Fink::Status->list();
+	#$dlist = Fink::Status->list();
+	$dlist = ();
 	foreach $pkgname (keys %$dlist) {
 		$po = Fink::Package->package_by_name_create($pkgname);
 		next if exists $po->{_versions}->{$dlist->{$pkgname}->{version}};
Index: Services.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Services.pm,v
retrieving revision 1.65
diff -u -r1.65 Services.pm
--- Services.pm	15 Nov 2003 11:36:43 -0000	1.65
+++ Services.pm	19 Nov 2003 10:54:11 -0000
@@ -622,11 +622,7 @@
 # Returns the architecture string to be used on this platform.
 # For example, "powerpc" for ppc.
 sub get_arch {
-	if(not defined $arch) {
-	  $arch = `/usr/bin/uname -p`;
-	  chomp $arch;
-	}
-	return $arch;
+	return "powerpc";
 }
 
 # get_sw_vers
