Index: Services.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Services.pm,v
retrieving revision 1.174
diff -u -r1.174 Services.pm
--- Services.pm	30 Jul 2005 05:45:27 -0000	1.174
+++ Services.pm	3 Aug 2005 00:56:29 -0000
@@ -1138,6 +1138,7 @@
 =cut
 
 sub get_arch {
+	$arch = "powerpc";
 	if(not defined $arch) {
 		foreach ('/usr/bin/uname', '/bin/uname') {
 			# check some common places (why aren't we using $ENV{PATH}?)
Index: Status.pm
===================================================================
RCS file: /cvsroot/fink/fink/perlmod/Fink/Status.pm,v
retrieving revision 1.17
diff -u -r1.17 Status.pm
--- Status.pm	2 Jan 2005 15:56:59 -0000	1.17
+++ Status.pm	3 Aug 2005 00:56:29 -0000
@@ -76,9 +76,7 @@
 	$file = $basepath."/var/lib/dpkg/status";
 	$hash = {};
 
-	if (! -f $file) {
-		print "WARNING: can't read dpkg status file \"$file\".\n";
-
+	{
 		$self->{_invalid} = 0;
 		return;
 	}
