diff -Naur advancemame-0.77.2/advance/advance.mak advancemame-0.77.2-new/advance/advance.mak
--- advancemame-0.77.2/advance/advance.mak	Sun Dec 21 16:58:39 2003
+++ advancemame-0.77.2-new/advance/advance.mak	Sun Jan  4 22:14:01 2004
@@ -274,21 +274,21 @@
 # Common install
 
 installdirs:
-	-$(INSTALL_DATA_DIR) $(DATADIR)
-	-$(INSTALL_DATA_DIR) $(DATADIR)/doc
-	-$(INSTALL_MAN_DIR) $(PREFIX)/man/man1
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/doc
+	-$(INSTALL_MAN_DIR) $(DESTDIR)$(PREFIX)/share/man/man1
 ifneq ($(wildcard $(EMUSRC)),)
-	-$(INSTALL_DATA_DIR) $(DATADIR)/rom
-	-$(INSTALL_DATA_DIR) $(DATADIR)/sample
-	-$(INSTALL_DATA_DIR) $(DATADIR)/artwork
-	-$(INSTALL_DATA_DIR) $(DATADIR)/image
-	-$(INSTALL_DATA_DIR) $(DATADIR)/crc
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/rom
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/sample
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/artwork
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/image
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/crc
 endif
 
 install-data: $(INSTALL_DATAFILES)
 ifdef INSTALL_DATAFILES
 	@for i in $(INSTALL_DATAFILES); do \
-		$(INSTALL_DATA) $$i $(DATADIR); \
+		$(INSTALL_DATA) $$i $(DESTDIR)$(DATADIR); \
 	done
 endif
 
@@ -301,7 +301,7 @@
 
 install-bin: $(INSTALL_BINFILES)
 	@for i in $(INSTALL_BINFILES); do \
-		$(INSTALL_PROGRAM) $$i $(PREFIX)/bin; \
+		$(INSTALL_PROGRAM) $$i $(DESTDIR)$(PREFIX)/bin; \
 	done
 
 uninstall-bin:
@@ -312,7 +312,7 @@
 install-doc: $(DOCOBJ) $(INSTALL_DOCFILES)
 ifdef INSTALL_DOCFILES
 	@for i in $(INSTALL_DOCFILES); do \
-		$(INSTALL_DATA) $$i $(DATADIR)/doc; \
+		$(INSTALL_DATA) $$i $(DESTDIR)$(DATADIR)/doc; \
 	done
 endif
 
@@ -326,7 +326,7 @@
 install-man: $(DOCOBJ) $(INSTALL_MANFILES)
 ifdef INSTALL_MANFILES
 	@for i in $(INSTALL_MANFILES); do \
-		$(INSTALL_DATA) $$i $(PREFIX)/man/man1; \
+		$(INSTALL_DATA) $$i $(DESTDIR)$(PREFIX)/share/man/man1; \
 	done
 endif
 
diff -Naur advancemame-0.77.2/advance/linux/itty.c advancemame-0.77.2-new/advance/linux/itty.c
--- advancemame-0.77.2/advance/linux/itty.c	Sun Dec 21 16:58:39 2003
+++ advancemame-0.77.2-new/advance/linux/itty.c	Sun Jan  4 22:14:01 2004
@@ -38,6 +38,8 @@
 #include "ossdl.h"
 #endif
 
+#include <sys/types.h>
+#include <machine/types.h>
 #include <sys/select.h>
 #include <stdio.h>
 #include <unistd.h>
