diff -Naur advancemame-0.80.0/advance/advance.mak advancemame-0.80.0-patched/advance/advance.mak
--- advancemame-0.80.0/advance/advance.mak	Tue Mar 30 11:58:24 2004
+++ advancemame-0.80.0-patched/advance/advance.mak	Fri Apr  2 19:00:21 2004
@@ -289,22 +289,22 @@
 pkgdocdir = $(docdir)/advance
 
 installdirs:
-	-$(INSTALL_DATA_DIR) $(pkgdir)
-	-$(INSTALL_DATA_DIR) $(pkgdocdir)
-	-$(INSTALL_MAN_DIR) $(mandir)/man1
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(pkgdir)
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(pkgdocdir)
+	-$(INSTALL_MAN_DIR) $(DESTDIR)$(mandir)/man1
 ifneq ($(wildcard $(EMUSRC)),)
-	-$(INSTALL_DATA_DIR) $(pkgdir)/rom
-	-$(INSTALL_DATA_DIR) $(pkgdir)/sample
-	-$(INSTALL_DATA_DIR) $(pkgdir)/artwork
-	-$(INSTALL_DATA_DIR) $(pkgdir)/image
-	-$(INSTALL_DATA_DIR) $(pkgdir)/crc
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(pkgdir)/rom
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(pkgdir)/sample
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(pkgdir)/artwork
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(pkgdir)/image
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(pkgdir)/crc
 endif
 
 install-data: $(INSTALL_DATAFILES)
 ifdef INSTALL_DATAFILES
 	@for i in $(INSTALL_DATAFILES); do \
-		echo "$(INSTALL_DATA) $$i $(pkgdir)"; \
-		$(INSTALL_DATA) $$i $(pkgdir); \
+		echo "$(INSTALL_DATA) $$i $(DESTDIR)$(pkgdir)"; \
+		$(INSTALL_DATA) $$i $(DESTDIR)$(pkgdir); \
 	done
 endif
 
@@ -317,8 +317,8 @@
 
 install-bin: $(INSTALL_BINFILES)
 	@for i in $(INSTALL_BINFILES); do \
-		echo "$(INSTALL_PROGRAM) $$i $(bindir)"; \
-		$(INSTALL_PROGRAM) $$i $(bindir); \
+		echo "$(INSTALL_PROGRAM) $$i $(DESTDIR)$(bindir)"; \
+		$(INSTALL_PROGRAM) $$i $(DESTDIR)$(bindir); \
 	done
 
 uninstall-bin:
@@ -329,8 +329,8 @@
 install-doc: $(DOCOBJ) $(INSTALL_DOCFILES)
 ifdef INSTALL_DOCFILES
 	@for i in $(INSTALL_DOCFILES); do \
-		echo "$(INSTALL_DATA) $$i $(pkgdocdir)"; \
-		$(INSTALL_DATA) $$i $(pkgdocdir); \
+		echo "$(INSTALL_DATA) $$i $(DESTDIR)$(pkgdocdir)"; \
+		$(INSTALL_DATA) $$i $(DESTDIR)$(pkgdocdir); \
 	done
 endif
 
@@ -344,8 +344,8 @@
 install-man: $(DOCOBJ) $(INSTALL_MANFILES)
 ifdef INSTALL_MANFILES
 	@for i in $(INSTALL_MANFILES); do \
-		echo "$(INSTALL_DATA) $$i $(mandir)/man1"; \
-		$(INSTALL_DATA) $$i $(mandir)/man1; \
+		echo "$(INSTALL_DATA) $$i $(DESTDIR)$(mandir)/man1"; \
+		$(INSTALL_DATA) $$i $(DESTDIR)$(mandir)/man1; \
 	done
 endif
 
