diff -Naur advancemenu-2.2.17/advance/advance.mak advancemenu-2.2.17-patched/advance/advance.mak
--- advancemenu-2.2.17/advance/advance.mak	Tue Jan  6 13:48:49 2004
+++ advancemenu-2.2.17-patched/advance/advance.mak	Mon Jan 19 14:09:23 2004
@@ -276,21 +276,21 @@
 # Common install
 
 installdirs:
-	-$(INSTALL_DATA_DIR) $(DATADIR)
-	-$(INSTALL_DATA_DIR) $(DATADIR)/doc
-	-$(INSTALL_MAN_DIR) $(PREFIX)/man/man1
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/doc
+	-$(INSTALL_MAN_DIR) $(DESTDIR)$(PREFIX)/share/man/man1
 ifneq ($(wildcard $(EMUSRC)),)
-	-$(INSTALL_DATA_DIR) $(DATADIR)/rom
-	-$(INSTALL_DATA_DIR) $(DATADIR)/sample
-	-$(INSTALL_DATA_DIR) $(DATADIR)/artwork
-	-$(INSTALL_DATA_DIR) $(DATADIR)/image
-	-$(INSTALL_DATA_DIR) $(DATADIR)/crc
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/rom
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/sample
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/artwork
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/image
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/crc
 endif
 
 install-data: $(INSTALL_DATAFILES)
 ifdef INSTALL_DATAFILES
 	@for i in $(INSTALL_DATAFILES); do \
-		$(INSTALL_DATA) $$i $(DATADIR); \
+		$(INSTALL_DATA) $$i $(DESTDIR)$(DATADIR); \
 	done
 endif
 
@@ -303,7 +303,7 @@
 
 install-bin: $(INSTALL_BINFILES)
 	@for i in $(INSTALL_BINFILES); do \
-		$(INSTALL_PROGRAM) $$i $(PREFIX)/bin; \
+		$(INSTALL_PROGRAM) $$i $(DESTDIR)$(PREFIX)/bin; \
 	done
 
 uninstall-bin:
@@ -314,7 +314,7 @@
 install-doc: $(DOCOBJ) $(INSTALL_DOCFILES)
 ifdef INSTALL_DOCFILES
 	@for i in $(INSTALL_DOCFILES); do \
-		$(INSTALL_DATA) $$i $(DATADIR)/doc; \
+		$(INSTALL_DATA) $$i $(DESTDIR)$(DATADIR)/doc; \
 	done
 endif
 
@@ -328,7 +328,7 @@
 install-man: $(DOCOBJ) $(INSTALL_MANFILES)
 ifdef INSTALL_MANFILES
 	@for i in $(INSTALL_MANFILES); do \
-		$(INSTALL_DATA) $$i $(PREFIX)/man/man1; \
+		$(INSTALL_DATA) $$i $(DESTDIR)$(PREFIX)/share/man/man1; \
 	done
 endif
 
diff -Naur advancemenu-2.2.17/advance/linux/itty.c advancemenu-2.2.17-patched/advance/linux/itty.c
--- advancemenu-2.2.17/advance/linux/itty.c	Tue Jan  6 13:48:49 2004
+++ advancemenu-2.2.17-patched/advance/linux/itty.c	Mon Jan 19 14:09:23 2004
@@ -38,6 +38,8 @@
 #include "ossdl.h"
 #endif
 
+#include <sys/types.h>
+#include <machine/types.h>
 #include <sys/select.h>
 #include <stdio.h>
 #include <unistd.h>
diff -Naur advancemenu-2.2.17/advance/menu/mconfig.cc advancemenu-2.2.17-patched/advance/menu/mconfig.cc
--- advancemenu-2.2.17/advance/menu/mconfig.cc	Tue Jan  6 13:48:49 2004
+++ advancemenu-2.2.17-patched/advance/menu/mconfig.cc	Mon Jan 19 14:20:12 2004
@@ -455,8 +455,6 @@
 			e = new wmame(a0, a2, a3);
 		} else if (a1 == "dmame") {
 			e = new dmame(a0, a2, a3);
-		} else if (a1 == "xmame") {
-			e = new xmame(a0, a2, a3);
 		} else if (a1 == "dmess") {
 			e = new dmess(a0, a2, a3);
 		} else if (a1 == "draine") {
@@ -983,18 +981,9 @@
 		}
 		if (target_search(path, FILE_MAXPATH, "advpac") == 0) {
 			conf_set(config_context, "", "emulator", "\"advpac\" advpac \"advpac\" \"\"");
-		}
-		if (target_search(path, FILE_MAXPATH, "xmame") == 0) {
-			conf_set(config_context, "", "emulator", "\"xmame\" xmame \"xmame\" \"\"");
 		} else {
-			if (target_search(path, FILE_MAXPATH, "xmame.x11") == 0) {
-				conf_set(config_context, "", "emulator", "\"xmame.x11\" xmame \"xmame.x11\" \"\"");
-			}
 			if (target_search(path, FILE_MAXPATH, "xmame.SDL") == 0) {
 				conf_set(config_context, "", "emulator", "\"xmame.SDL\" xmame \"xmame.SDL\" \"\"");
-			}
-			if (target_search(path, FILE_MAXPATH, "xmame.svgalib") == 0) {
-				conf_set(config_context, "", "emulator", "\"xmame.svgalib\" xmame \"xmame.svgalib\" \"\"");
 			}
 		}
 #endif
