diff -Naur old/Makefile.in new/Makefile.in
--- old/Makefile.in	Mon Oct  7 19:26:24 2002
+++ new/Makefile.in	Thu Mar 27 04:24:33 2003
@@ -435,7 +435,7 @@
 
 
 install:
-	mkdir -pv $(DESTDIR)/`$(GGZ_CONFIG) -c`
+	mkdir -p $(DESTDIR)/`$(GGZ_CONFIG) -c`
 	touch $(DESTDIR)/`$(GGZ_CONFIG) -c`/ggz.modules
 	make install-recursive # FIXME - kludge
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
diff -Naur old/configure new/configure
--- old/configure	Mon Oct  7 19:28:04 2002
+++ new/configure	Thu Mar 27 04:14:58 2003
@@ -3265,7 +3265,7 @@
 test=NONE
 libggz_libdir=NONE
 for dir in $libggz_libdirs; do
-  try="ls -1 $dir/libggz.so"
+  try="ls -1 $dir/libggz.dylib"
   if test -n "`$try 2> /dev/null`"; then libggz_libdir=$dir; break; else echo "tried $dir" >&5 ; fi
 done
 
@@ -3423,7 +3423,7 @@
 test=NONE
 ggzmod_libdir=NONE
 for dir in $ggzmod_libdirs; do
-  try="ls -1 $dir/libggzmod.so"
+  try="ls -1 $dir/libggzmod.dylib"
   if test -n "`$try 2> /dev/null`"; then ggzmod_libdir=$dir; break; else echo "tried $dir" >&5 ; fi
 done
 
@@ -3481,7 +3481,7 @@
 
 
 
-LIB_GGZMOD='-lggzmod'
+LIB_GGZMOD='-lggzmod -lggz'
 
 
 
