diff -Naur neverball-1.2.2-old/Makefile neverball-1.2.2/Makefile
--- neverball-1.2.2-old/Makefile	Fri Apr 16 17:21:58 2004
+++ neverball-1.2.2/Makefile	Mon Apr 19 09:24:23 2004
@@ -3,8 +3,8 @@
 
 # Maybe you need one of these.  Maybe you don't.
 
-#X11_PATH= -L/usr/X11/lib
-#X11_PATH= -L/usr/X11R6/lib
+X11_PATH= -L/usr/X11/lib
+X11_PATH= -L/usr/X11R6/lib
 
 OGL_LIBS= -lGLU -lGL -lm
 #OGL_LIBS= -lm                                                # Think Different
diff -Naur neverball-1.2.2-old/share/config.h neverball-1.2.2/share/config.h
--- neverball-1.2.2-old/share/config.h	Sat Mar 20 10:42:23 2004
+++ neverball-1.2.2/share/config.h	Mon Apr 19 09:25:42 2004
@@ -15,7 +15,7 @@
  *
  * If the game is to be installed globally, change this value.
  */
-#define CONFIG_PATH "./data"
+#define CONFIG_PATH "@FINK@/share/neverball/data"
 
 /*
  * Global settings are stored in USER_CONFIG_FILE.  Replays are stored
