diff -uNr SDL_ttf-2.0.6/SDL_ttf.c SDL_ttf-2.0.6-new/SDL_ttf.c
--- SDL_ttf-2.0.6/SDL_ttf.c	2003-02-21 12:38:15.000000000 -0500
+++ SDL_ttf-2.0.6-new/SDL_ttf.c	2006-11-14 17:53:21.000000000 -0500
@@ -39,10 +39,8 @@
 #define FREEA(p) free(p)
 #endif
 
-#include <freetype/freetype.h>
-#include <freetype/ftoutln.h>
-#include <freetype/ttnameid.h>
-#include <freetype/internal/ftobjs.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 
 #include "SDL.h"
 #include "SDL_endian.h"
@@ -265,7 +263,7 @@
 	}
 	memset(stream, 0, sizeof(*stream));
 
-	stream->memory = library->memory;
+	stream->memory = NULL; /* set by FT_Open_Face */
 	stream->read = RWread;
 	stream->descriptor.pointer = src;
 	stream->pos = (unsigned long)position;
