diff -ru trackballs-1.0.0-orig/share/Makefile.am trackballs-1.0.0/share/Makefile.am
--- trackballs-1.0.0-orig/share/Makefile.am	Sat Mar 20 02:04:32 2004
+++ trackballs-1.0.0/share/Makefile.am	Sat Mar 20 02:04:48 2004
@@ -4,6 +4,7 @@
 EXTRA_DIST = $(pkgdata_DATA) trackballs.6
 
 install-pkgdataDATA:
-	install -D -m664 -g games highScores		$(DESTDIR)@with_highscores@
+	mkdir -p $(DESTDIR)/share/trackballs
+	install -m664 -g games highScores		$(DESTDIR)@with_highscores@
 	chgrp -R games			$(DESTDIR)$(pkgdatadir)/levels
 	chmod -R g+r			$(DESTDIR)$(pkgdatadir)/levels
diff -ru trackballs-1.0.0-orig/src/player.cc trackballs-1.0.0/src/player.cc
--- trackballs-1.0.0-orig/src/player.cc	Sat Mar 20 02:04:32 2004
+++ trackballs-1.0.0/src/player.cc	Sat Mar 20 02:05:32 2004
@@ -68,6 +68,7 @@
 
   if(!Game::current) return;
   if(!playing) return;
+  dx = dy = 0.0;
   Map *map = Game::current->map;
   health+=t * 0.4; if(health > 1.0) health = 1.0;
 
