diff -Naur xbomb-2.1a/Makefile xbomb-new/Makefile
--- xbomb-2.1a/Makefile	1996-01-20 16:02:46.000000000 -0500
+++ xbomb-new/Makefile	2003-03-21 05:16:44.000000000 -0500
@@ -15,7 +15,7 @@
 CC=gcc
 CFLAGS=-O2
 
-INCLUDES=
+INCLUDES=-I/usr/X11R6/include
 
 LIB=
 
diff -Naur xbomb-2.1a/hiscore.c xbomb-new/hiscore.c
--- xbomb-2.1a/hiscore.c	1998-01-16 14:50:39.000000000 -0500
+++ xbomb-new/hiscore.c	2003-03-21 05:16:24.000000000 -0500
@@ -79,6 +79,9 @@
 static void decrypt_score(struct score_name* sn);
 static void encrypt_score(struct score_name* sn);
 
+static char *cuserid(char *buf) {
+	return strncpy(buf, getlogin(), 19);
+}
 
 /*++++++++++++++++++++++++++++++++++++++
   Prints the high score tables.
