diff -Naur anjuta-0.9.99.orig/launcher.c anjuta-0.9.99/launcher.c
--- anjuta-0.9.99-1/src/launcher.c	Sun Aug  4 19:39:33 2002
+++ anjuta-0.9.99-1/src/launcher.c	Mon Sep  9 06:47:30 2002
@@ -349,9 +349,9 @@
 	/* Set no delays for the write pipes (non_buffered) so
 	  that we get all the outputs immidiately */
 	if ((md = fcntl (launcher.stdout_pipe[1], F_GETFL)) != -1)
-		fcntl (launcher.stdout_pipe[1], F_SETFL, O_SYNC | md);
+		fcntl (launcher.stdout_pipe[1], F_SETFL, O_ASYNC | md);
 	if ((md = fcntl (launcher.stderr_pipe[1], F_GETFL)) != -1)
-		fcntl (launcher.stderr_pipe[1], F_SETFL, O_SYNC | md);
+		fcntl (launcher.stderr_pipe[1], F_SETFL, O_ASYNC | md);
   
 	/* This is a quick hack to get the child's exit code */
 	/* Don't complain!! */
--- njuta-0.9.99-1/data/Makefile.in.orig	2002-10-08 22:20:07.000000000 -0700
+++ njuta-0.9.99-1/data/Makefile.in	2002-10-08 22:24:44.000000000 -0700
@@ -445,9 +445,9 @@
 
 
 install-data-local: system.tags
-	mkdir -p $(MACROS_dir)
-	$(INSTALL_DATA) $(MACROS:%=$(top_srcdir)/macros/%) $(MACROS_dir)
-	$(INSTALL_DATA) system.tags $(anjuta_datadir)
+	mkdir -p $(DESTDIR)$(MACROS_dir)
+	$(INSTALL_DATA) $(MACROS:%=$(top_srcdir)/macros/%) $(DESTDIR)$(MACROS_dir)
+	$(INSTALL_DATA) system.tags $(DESTDIR)$(anjuta_datadir)
 
 uninstall-local:
 	@rm -fv $(MACROS:%=$(MACROS_dir)/%)
diff -Naur anjuta-0.9.99.orig/omf-install/Makefile.in anjuta-0.9.99/omf-install/Makefile.in
--- anjuta-0.9.99.orig/omf-install/Makefile.in	Mon Aug  5 04:45:42 2002
+++ anjuta-0.9.99/omf-install/Makefile.in	Thu Feb 20 00:05:24 2003
@@ -278,7 +278,7 @@
 	-for file in $(srcdir)/*.omf; do \
 		$(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(omf_dest_dir); \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) 
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) 
 
 uninstall-local:
 	-for file in $(srcdir)/*.omf; do \
