diff -Naur cvsgrab_v1.2.old/cvsgrab.sh cvsgrab_v1.2.new/cvsgrab.sh
--- cvsgrab_v1.2.old/cvsgrab.sh	Wed Jul 17 03:41:26 2002
+++ cvsgrab_v1.2.new/cvsgrab.sh	Fri Mar  7 13:52:03 2003
@@ -9,7 +9,7 @@
       JAVACMD="$JAVA_HOME/bin/java"
     fi
   else
-    JAVACMD=java
+    JAVACMD=/usr/bin/java
   fi
 fi
  
@@ -24,7 +24,7 @@
 fi
 
 # add in the dependency .jar files in non-RPM mode (the default)
-DIRLIBS=./lib
+DIRLIBS=@PREFIX@/share/cvsgrab
 for i in "${DIRLIBS}"/*.jar
 do
   # if the directory is empty, then it will return the input string
