+++ ripper/ripper.y	2003-07-22 13:30:22.000000000 +0200
@@ -2805,10 +2805,10 @@
 		  case 'x':
 		    options |= RE_OPTION_EXTENDED;
 		    break;
-		  case 'p':	/* /p is obsolete */
-		    rb_warn("/p option is obsolete; use /m\n\tnote: /m does not change ^, $ behavior");
-		    options |= RE_OPTION_POSIXLINE;
-		    break;
+		  //case 'p':	/* /p is obsolete */
+		  //  rb_warn("/p option is obsolete; use /m\n\tnote: /m does not change ^, $ behavior");
+		  //  options |= RE_OPTION_POSIXLINE;
+		  //  break;
 		  case 'm':
 		    options |= RE_OPTION_MULTILINE;
 		    break;
