diff -Nurd -x'*~' fox-1.6.46.orig/configure fox-1.6.46/configure
--- fox-1.6.46.orig/configure	2012-07-11 20:40:36.000000000 -0400
+++ fox-1.6.46/configure	2012-08-17 13:02:25.000000000 -0400
@@ -16166,7 +16166,7 @@
 
 
 
-X_BASE_LIBS="-lX11 -lXext"
+X_BASE_LIBS="-lX11 -lXext -L/usr/X11R6/lib"
 LIBGLU="-lGLU"
 LIBGL="-lGL"
 
@@ -17902,11 +17902,11 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $with_xft" >&5
 $as_echo "$with_xft" >&6; }
 if test "x$with_xft" != "xno"; then
-XFTCFLAGS="-I/usr/include/freetype2"
-XFTLIBS="-lfreetype -lXft"
+XFTCFLAGS=`pkg-config --cflags xft`
+XFTLIBS=`pkg-config --libs xft`
 saved_cppflags="${CXXFLAGS}"
 CXXFLAGS="${CXXFLAGS} -DHAVE_XFT_H=1 $XFTCFLAGS"
-X_BASE_LIBS="${X_BASE_LIBS} $XFTLIBS"
+X_BASE_LIBS="$XFTLIBS ${X_BASE_LIBS}"
 fi
 
 
diff -Nurd -x'*~' fox-1.6.46.orig/fox.pc.in fox-1.6.46/fox.pc.in
--- fox-1.6.46.orig/fox.pc.in	2012-07-11 20:38:41.000000000 -0400
+++ fox-1.6.46/fox.pc.in	2012-08-17 13:29:25.000000000 -0400
@@ -13,6 +13,6 @@
 Description: The FOX Toolkit
 URL: www.fox-toolkit.org
 Version: @FOX_MAJOR_VERSION@.@FOX_MINOR_VERSION@.@FOX_PATCH_LEVEL@
-Libs: ${FOX_LIBS}
-Libs.private: ${X_LIBS} ${X_BASE_LIBS} ${X_EXTRA_LIBS} ${GL_LIBS} ${LIBS}
+Libs: -I${libdir} ${FOX_LIBS}
 Cflags: -I${includedir}
+Requires: xft
\ No newline at end of file
