--- pcb-20060822/src/hid/gtk/gui-library-window.c.orig	2006-04-17 21:48:10.000000000 -0400
+++ pcb-20060822/src/hid/gtk/gui-library-window.c	2006-10-04 23:11:25.000000000 -0400
@@ -344,7 +344,7 @@
    * don't want.
    */
   gtk_widget_realize (library_window);
-  gdk_window_set_accept_focus (library_window->window, FALSE);
+  //gdk_window_set_accept_focus (library_window->window, FALSE);
 
   gtk_widget_show_all (library_window);
 }
--- pcb-20060822/src/hid/gtk/gui-log-window.c.orig	2006-04-17 21:48:11.000000000 -0400
+++ pcb-20060822/src/hid/gtk/gui-log-window.c	2006-11-15 22:55:19.000000000 -0500
@@ -113,7 +113,7 @@
    * don't want.
    */
   gtk_widget_realize (log_window);
-  gdk_window_set_accept_focus (log_window->window, FALSE);
+  //gdk_window_set_accept_focus (log_window->window, FALSE);
   if (Settings.AutoPlace)
     gtk_widget_set_uposition (GTK_WIDGET (log_window), 10, 10);
   gtk_widget_show_all (log_window);
--- pcb-20060822/src/hid/gtk/gui-netlist-window.c.orig	2006-04-17 21:48:11.000000000 -0400
+++ pcb-20060822/src/hid/gtk/gui-netlist-window.c	2006-11-15 22:55:45.000000000 -0500
@@ -640,7 +640,7 @@
      |  don't want.
    */
   gtk_widget_realize (netlist_window);
-  gdk_window_set_accept_focus (netlist_window->window, FALSE);
+  //gdk_window_set_accept_focus (netlist_window->window, FALSE);
   if (Settings.AutoPlace)
     gtk_widget_set_uposition (GTK_WIDGET (netlist_window), 10, 10);
 
--- pcb-20060822/src/hid/gtk/gui-pinout-window.c.orig	2006-03-22 18:05:41.000000000 -0500
+++ pcb-20060822/src/hid/gtk/gui-pinout-window.c	2006-11-15 22:55:57.000000000 -0500
@@ -217,7 +217,7 @@
      |  don't want.
    */
   gtk_widget_realize (pinout->top_window);
-  gdk_window_set_accept_focus (pinout->top_window->window, FALSE);
+  //gdk_window_set_accept_focus (pinout->top_window->window, FALSE);
   if (Settings.AutoPlace)
     gtk_widget_set_uposition (GTK_WIDGET (pinout->top_window), 10, 10);
   gtk_widget_show_all (pinout->top_window);
