diff -Nurd -x'*~' xforms-1.0.90.orig/configure.ac xforms-1.0.90/configure.ac
--- xforms-1.0.90.orig/configure.ac	2004-03-30 04:48:24.000000000 -0500
+++ xforms-1.0.90/configure.ac	2011-09-26 00:52:48.000000000 -0400
@@ -28,7 +28,6 @@
 AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
 
 ### check for special systems
 AC_ISC_POSIX
@@ -77,11 +76,20 @@
 # Checks for header files.
 AC_CHECK_HEADERS([sys/select.h])
 
-# Check whether we want to build the gl code
+# Check whether we want to build the gl code (headers may be with X11's)
+AC_PATH_XTRA
 AC_ARG_ENABLE(gl,
   [AC_HELP_STRING([--disable-gl], [Do not build GL support library])])
 if test x$enable_gl != xno ; then
-  AC_CHECK_HEADER([GL/glx.h],, [enable_gl=no])
+  save_CPPFLAGS=$CPPFLAGS
+  CPPFLAGS="$X_CFLAGS $CPPFLAGS"
+  AC_CHECK_HEADER([GL/glx.h],, [
+    enable_gl=no
+    if test x$enable_gl != xyes ; then
+      AC_MSG_ERROR([GL requested but not available])
+    fi
+  ])
+  CPPFLAGS=$save_CPPFLAGS
 fi
 AM_CONDITIONAL(BUILD_GL, [test x$enable_gl != xno])
 
@@ -90,7 +98,6 @@
 dnl AC_HEADER_DIRENT
 
 # Check for X, XPM and JPEG
-AC_PATH_XTRA
 XFORMS_PATH_XPM
 XFORMS_CHECK_LIB_JPEG
 
diff -Nurd -x'*~' xforms-1.0.90.orig/fdesign/Makefile.am xforms-1.0.90/fdesign/Makefile.am
--- xforms-1.0.90.orig/fdesign/Makefile.am	2003-11-13 16:46:48.000000000 -0500
+++ xforms-1.0.90/fdesign/Makefile.am	2011-09-25 22:16:57.000000000 -0400
@@ -55,6 +55,3 @@
 	sp_xyplot.c \
 	xpm.h
 
-#	viewforms.c \
-#	viewforms.h \
-#	viewforms_main.c \
diff -Nurd -x'*~' xforms-1.0.90.orig/gl/Makefile.am xforms-1.0.90/gl/Makefile.am
--- xforms-1.0.90.orig/gl/Makefile.am	2003-11-20 04:33:11.000000000 -0500
+++ xforms-1.0.90/gl/Makefile.am	2011-09-26 01:07:32.000000000 -0400
@@ -1,13 +1,14 @@
 include $(top_srcdir)/config/common.am
 
 # forms.h is a generated file
-INCLUDES = -DMAKING_FORMS -I../lib -I$(top_srcdir)/lib
+INCLUDES = -DMAKING_FORMS -I../lib -I$(top_srcdir)/lib $(X_CFLAGS)
 
 include_HEADERS = glcanvas.h
 
 lib_LTLIBRARIES = libformsGL.la
 
-libformsGL_la_LDFLAGS = -version-info 1:0:0
+libformsGL_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+libformsGL_la_LIBADD = $(top_builddir)/lib/libforms.la $(X_LIBS) -lGL
 
 libformsGL_la_SOURCES = \
 	glcanvas.c \
diff -Nurd -x'*~' xforms-1.0.90.orig/image/Makefile.am xforms-1.0.90/image/Makefile.am
--- xforms-1.0.90.orig/image/Makefile.am	2003-11-13 16:46:48.000000000 -0500
+++ xforms-1.0.90/image/Makefile.am	2011-09-26 01:04:11.000000000 -0400
@@ -7,7 +7,8 @@
 
 lib_LTLIBRARIES = libflimage.la
 
-libflimage_la_LDFLAGS = -version-info 1:0:0
+libflimage_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+libflimage_la_LIBADD = $(top_builddir)/lib/libforms.la $(JPEG_LIB)
 
 libflimage_la_SOURCES = \
 	flimage.h \
diff -Nurd -x'*~' xforms-1.0.90.orig/lib/Makefile.am xforms-1.0.90/lib/Makefile.am
--- xforms-1.0.90.orig/lib/Makefile.am	2003-11-20 04:33:16.000000000 -0500
+++ xforms-1.0.90/lib/Makefile.am	2011-09-26 01:01:17.000000000 -0400
@@ -17,7 +17,8 @@
 
 lib_LTLIBRARIES = libforms.la
 
-libforms_la_LDFLAGS = -version-info 1:0:0
+libforms_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+libforms_la_LIBADD = $(X_LIBS) -lX11 -lXpm
 
 libforms_la_SOURCES = \
 	align.c \
@@ -119,5 +120,3 @@
 	xtext.c \
 	xyplot.c
 
-#	dirent_vms.h \
-#	vms_readdir.c \
