diff -Nurd -x'*~' xplanet-1.2.0.orig/src/libprojection/ProjectionIcosagnomonic.cpp xplanet-1.2.0/src/libprojection/ProjectionIcosagnomonic.cpp
--- xplanet-1.2.0.orig/src/libprojection/ProjectionIcosagnomonic.cpp	2005-06-14 14:10:36.000000000 -0400
+++ xplanet-1.2.0/src/libprojection/ProjectionIcosagnomonic.cpp	2007-07-17 18:18:44.000000000 -0400
@@ -26,6 +26,19 @@
 #define THIGH 3
 #define RATIO (((THIGH)*sqrt(3.0)/2)/(TWIDE))
 
+/* signbit() appaired only in FreeBSD 5.1, so we need this hack */
+#ifndef signbit
+static int
+signbit(double x)
+{
+
+	if ((x < 0.0) || (x = -0.0)) 
+		return 1;
+	else
+		return 0;
+}
+#endif /* signbit */
+
 bool
 ProjectionIcosagnomonic::PointXY::sameSide(const PointXY& p1,
                                            const PointXY& p2,
