diff -Nurd -x'*~' control-center-2.12.2.orig/configure control-center-2.12.2/configure
--- control-center-2.12.2.orig/configure	2005-11-29 16:24:15.000000000 -0500
+++ control-center-2.12.2/configure	2006-06-28 02:56:51.000000000 -0400
@@ -21765,7 +21765,7 @@
 if test "x$GCC" = "xyes"; then
   case " $CFLAGS " in
   *[\ \	]-Wall[\ \	]*) ;;
-  *) CFLAGS="$CFLAGS -Wall -Wextra" ;;
+  *) CFLAGS="$CFLAGS -Wall" ;;
   esac
 fi
 
@@ -25036,7 +25036,7 @@
 
 fi
 
-x_libs="$X_PRE_LIBS $X_LIBS -lX11 $X_EXTRA_LIBS"
+x_libs="$X_PRE_LIBS $X_LIBS -lX11 -lXext $X_EXTRA_LIBS"
 
 echo "$as_me:$LINENO: checking for XScreenSaverQueryInfo" >&5
 echo $ECHO_N "checking for XScreenSaverQueryInfo... $ECHO_C" >&6
diff -Nurd -x'*~' control-center-2.12.2.orig/libwindow-settings/Makefile.in control-center-2.12.2/libwindow-settings/Makefile.in
--- control-center-2.12.2.orig/libwindow-settings/Makefile.in	2005-11-29 16:24:23.000000000 -0500
+++ control-center-2.12.2/libwindow-settings/Makefile.in	2006-06-28 02:56:03.000000000 -0400
@@ -381,7 +381,7 @@
 	gnome-window-settings-2.0.pc.in		\
 	ChangeLog
 
-wms_flags = -export_dynamic -avoid-version
+wms_flags = -export_dynamic -avoid-version -module
 wmsdir = $(WM_MODULE_DIR)
 wms_LTLIBRARIES = \
 	libmetacity.la
