diff -Nurd -x'*~' eel-2.14.1.orig/configure eel-2.14.1/configure
--- eel-2.14.1.orig/configure	2006-04-10 16:37:48.000000000 -0400
+++ eel-2.14.1/configure	2006-06-07 18:27:38.000000000 -0400
@@ -22756,7 +22756,7 @@
 CONFIG_STATUS_DEPENDENCIES='$(top_srcdir)/po/LINGUAS'
 
 
-GETTEXT_PACKAGE=eel-2.0
+GETTEXT_PACKAGE=eel-2.2
 
 cat >>confdefs.h <<_ACEOF
 #define GETTEXT_PACKAGE "$GETTEXT_PACKAGE"
diff -Nurd -x'*~' eel-2.14.1.orig/eel/Makefile.in eel-2.14.1/eel/Makefile.in
--- eel-2.14.1.orig/eel/Makefile.in	2006-04-10 19:03:17.000000000 -0400
+++ eel-2.14.1/eel/Makefile.in	2006-06-07 18:27:14.000000000 -0400
@@ -306,7 +306,6 @@
 #	-DBONOBO_DISABLE_DEPRECATED
 libeel_2_la_LDFLAGS = \
 	-version-info @EEL_VERSION_INFO@	\
-	-no-undefined				\
 	$(EEL_LIBS)				\
 	$(RENDER_LIBS)				\
 	$(X_LIBS)				\
diff -Nurd -x'*~' eel-2.14.1.orig/ltmain.sh eel-2.14.1/ltmain.sh
--- eel-2.14.1.orig/ltmain.sh	2006-02-11 19:50:49.000000000 -0500
+++ eel-2.14.1/ltmain.sh	2006-06-07 18:27:14.000000000 -0400
@@ -5696,10 +5696,6 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
 	done
       fi
 
