diff -Naur GConf-2.10.1-old/doc/gconf/Makefile.in GConf-2.10.1/doc/gconf/Makefile.in
--- GConf-2.10.0-old/doc/gconf/Makefile.in	2004-04-19 08:12:05.000000000 -0400
+++ GConf-2.10.0/doc/gconf/Makefile.in	2005-06-09 06:31:22.000000000 -0400
@@ -236,7 +236,7 @@
 
 # CFLAGS and LDFLAGS for compiling scan program. Only needed
 # if $(DOC_MODULE).types is non-empty.
-GTKDOC_CFLAGS = -I$(top_srcdir) $(CFLAGS) $(DEPENDENT_CFLAGS)
+GTKDOC_CFLAGS = -I$(top_srcdir) $(CFLAGS) $(CPPFLAGS) $(DEPENDENT_CFLAGS)
 
 GTKDOC_LIBS = $(LDFLAGS) $(DEPENDENT_LIBS) $(top_builddir)/gconf/libgconf-$(MAJOR_VERSION).la
 @GTK_DOC_USE_LIBTOOL_FALSE@GTKDOC_CC = $(CC) $(INCLUDES) $(AM_CFLAGS) $(CFLAGS)
diff -Naur GConf-2.10.0-old/gconf/Makefile.in GConf-2.10.0/gconf/Makefile.in
--- GConf-2.10.0-old/gconf/Makefile.in	2004-04-19 08:11:03.000000000 -0400
+++ GConf-2.10.0/gconf/Makefile.in	2005-06-09 06:31:22.000000000 -0400
@@ -256,7 +256,7 @@
 	gconfd.c
 
 
-gconfd_2_LDADD = $(EFENCE) $(INTLLIBS) $(DEPENDENT_LIBS) libgconf-$(MAJOR_VERSION).la
+gconfd_2_LDADD = libgconf-$(MAJOR_VERSION).la $(EFENCE) $(INTLLIBS) $(DEPENDENT_LIBS)
 
 
 # gconf_testclient_SOURCES = \
@@ -267,7 +267,7 @@
 	gconftool.c
 
 
-gconftool_2_LDADD = $(EFENCE) $(INTLLIBS) $(DEPENDENT_WITH_XML_LIBS) $(POPT_LIBS) libgconf-$(MAJOR_VERSION).la
+gconftool_2_LDADD = libgconf-$(MAJOR_VERSION).la $(EFENCE) $(INTLLIBS) $(DEPENDENT_WITH_XML_LIBS) $(POPT_LIBS)
 
 gconf_sanity_check_2_SOURCES = \
 	gconf-sanity-check.c
@@ -273,7 +273,7 @@
 	gconf-sanity-check.c
 
 
-gconf_sanity_check_2_LDADD = $(EFENCE) $(INTLLIBS) $(DEPENDENT_WITH_XML_AND_GTK_LIBS) $(POPT_LIBS) libgconf-$(MAJOR_VERSION).la
+gconf_sanity_check_2_LDADD = libgconf-$(MAJOR_VERSION).la $(EFENCE) $(INTLLIBS) $(DEPENDENT_WITH_XML_AND_GTK_LIBS) $(POPT_LIBS)
 
 libgconf_2_la_SOURCES = \
 	gconf-internals.c	\
diff -Naur GConf-2.10.-old/gconf/gconf-internals.c GConf-2.10.0/gconf/gconf-internals.c
--- GConf-2.10.0-old/gconf/gconf-internals.c	2004-04-19 06:08:20.000000000 -0400
+++ GConf-2.10.0/gconf/gconf-internals.c	2005-06-09 06:31:22.000000000 -0400
@@ -2606,6 +2606,9 @@
         }
     }
 
+  /* needed on Darwin */
+  fsync (lock->lock_fd);
+
   return lock;
 }
 
diff -Naur GConf-2.10.0-old/ltmain.sh GConf-2.10.0/ltmain.sh
--- GConf-2.10.0-old/ltmain.sh	2004-04-19 08:03:11.000000000 -0400
+++ GConf-2.10.0/ltmain.sh	2005-06-09 06:31:22.000000000 -0400
@@ -3845,6 +3845,11 @@
 	  eval test_cmds=\"$archive_expsym_cmds\"
 	  cmds=$archive_expsym_cmds
 	else
+	  if test "x$verstring" = "x0.0"; then
+            tmp_verstring=
+          else
+            tmp_verstring="$verstring"
+          fi
 	  eval test_cmds=\"$archive_cmds\"
 	  cmds=$archive_cmds
 	  fi
