diff -urN gnome-python-2.0.0.orig/bonobo/bonobo-arg.c gnome-python-2.0.0/bonobo/bonobo-arg.c
--- gnome-python-2.0.0.orig/bonobo/bonobo-arg.c	Tue Feb 11 00:32:24 2003
+++ gnome-python-2.0.0/bonobo/bonobo-arg.c	Tue Jan 27 22:58:43 2004
@@ -1,5 +1,7 @@
 /* -*- Mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #include <pygobject.h>
 #include <pyorbit.h>
 #include <bonobo/bonobo-types.h>
diff -urN gnome-python-2.0.0.orig/bonobo/bonobo.override gnome-python-2.0.0/bonobo/bonobo.override
--- gnome-python-2.0.0.orig/bonobo/bonobo.override	Tue Aug 12 22:58:35 2003
+++ gnome-python-2.0.0/bonobo/bonobo.override	Wed Jan 28 14:04:08 2004
@@ -3,7 +3,6 @@
 headers
 
 #include <Python.h>
-#define NO_IMPORT_PYGOBJECT
 #include <pygobject.h>
 #include <pyorbit.h>
 #include <libbonobo.h>
diff -urN gnome-python-2.0.0.orig/bonobo/bonobomodule.c gnome-python-2.0.0/bonobo/bonobomodule.c
--- gnome-python-2.0.0.orig/bonobo/bonobomodule.c	Sun Aug 24 22:34:32 2003
+++ gnome-python-2.0.0/bonobo/bonobomodule.c	Tue Jan 27 22:58:30 2004
@@ -1,4 +1,7 @@
 /* -*- Mode: C; c-basic-offset: 4 -*- */
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #ifdef HAVE_CONFIG_H
 #  include "config.h"
 #endif
diff -urN gnome-python-2.0.0.orig/bonobo/bonoboui.override gnome-python-2.0.0/bonobo/bonoboui.override
--- gnome-python-2.0.0.orig/bonobo/bonoboui.override	Wed Jul 16 18:24:35 2003
+++ gnome-python-2.0.0/bonobo/bonoboui.override	Wed Jan 28 14:03:58 2004
@@ -2,7 +2,6 @@
 %%
 headers
 #include <Python.h>
-#define NO_IMPORT_PYGOBJECT
 #include <pygobject.h>
 #include <pyorbit.h>
 #include <bonobo.h>
diff -urN gnome-python-2.0.0.orig/bonobo/bonobouimodule.c gnome-python-2.0.0/bonobo/bonobouimodule.c
--- gnome-python-2.0.0.orig/bonobo/bonobouimodule.c	Sun Aug 24 22:34:43 2003
+++ gnome-python-2.0.0/bonobo/bonobouimodule.c	Wed Jan 28 20:48:45 2004
@@ -1,4 +1,7 @@
 /* -*- Mode: C; c-basic-offset: 4 -*- */
+#define NO_IMPORT_PYGOBJECT 1
+/* #define NO_IMPORT_PYGTK 1 */
+#define NO_IMPORT_PYORBIT 1
 #ifdef HAVE_CONFIG_H
 #  include "config.h"
 #endif
diff -urN gnome-python-2.0.0.orig/configure gnome-python-2.0.0/configure
--- gnome-python-2.0.0.orig/configure	Mon Sep  1 19:55:41 2003
+++ gnome-python-2.0.0/configure	Tue Jan 27 22:12:03 2004
@@ -6833,7 +6833,7 @@
 	allow_undefined_flag='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag='-flat_namespace -undefined suppress'
+	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag='-undefined dynamic_lookup'
 	;;
       esac
       # FIXME: Relying on posixy $() will cause problems for
diff -urN gnome-python-2.0.0.orig/gconf/gconf-fixes.c gnome-python-2.0.0/gconf/gconf-fixes.c
--- gnome-python-2.0.0.orig/gconf/gconf-fixes.c	Sat Jan 19 20:55:11 2002
+++ gnome-python-2.0.0/gconf/gconf-fixes.c	Tue Jan 27 22:46:00 2004
@@ -18,7 +18,9 @@
  * 
  * Author: Johan Dahlin <jdahlin@telia.com>
  */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #include <gconf/gconf-client.h>
 #include <gconf/gconf-value.h>
 
diff -urN gnome-python-2.0.0.orig/gconf/gconfmodule.c gnome-python-2.0.0/gconf/gconfmodule.c
--- gnome-python-2.0.0.orig/gconf/gconfmodule.c	Sun Aug 24 22:34:54 2003
+++ gnome-python-2.0.0/gconf/gconfmodule.c	Tue Jan 27 22:45:34 2004
@@ -1,5 +1,7 @@
 /* -*- Mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
Binary files gnome-python-2.0.0.orig/gnome/__init__.pyc and gnome-python-2.0.0/gnome/__init__.pyc differ
diff -urN gnome-python-2.0.0.orig/gnome/appletmodule.c gnome-python-2.0.0/gnome/appletmodule.c
--- gnome-python-2.0.0.orig/gnome/appletmodule.c	Sun Aug 24 22:35:13 2003
+++ gnome-python-2.0.0/gnome/appletmodule.c	Tue Jan 27 22:53:46 2004
@@ -1,5 +1,7 @@
 /* -*- Mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
diff -urN gnome-python-2.0.0.orig/gnome/canvasmodule.c gnome-python-2.0.0/gnome/canvasmodule.c
--- gnome-python-2.0.0.orig/gnome/canvasmodule.c	Sun Aug 24 22:35:27 2003
+++ gnome-python-2.0.0/gnome/canvasmodule.c	Tue Jan 27 22:51:39 2004
@@ -1,5 +1,7 @@
 /* -*- Mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
diff -urN gnome-python-2.0.0.orig/gnome/gnomemodule.c gnome-python-2.0.0/gnome/gnomemodule.c
--- gnome-python-2.0.0.orig/gnome/gnomemodule.c	Sun Aug 24 22:36:04 2003
+++ gnome-python-2.0.0/gnome/gnomemodule.c	Tue Jan 27 22:50:58 2004
@@ -1,5 +1,7 @@
 /* -*- Mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
diff -urN gnome-python-2.0.0.orig/gnome/nautilusmodule.c gnome-python-2.0.0/gnome/nautilusmodule.c
--- gnome-python-2.0.0.orig/gnome/nautilusmodule.c	Sun Aug 24 22:36:14 2003
+++ gnome-python-2.0.0/gnome/nautilusmodule.c	Tue Jan 27 22:54:01 2004
@@ -1,5 +1,7 @@
 /* -*- Mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
diff -urN gnome-python-2.0.0.orig/gnome/uimodule.c gnome-python-2.0.0/gnome/uimodule.c
--- gnome-python-2.0.0.orig/gnome/uimodule.c	Sun Aug 24 22:37:00 2003
+++ gnome-python-2.0.0/gnome/uimodule.c	Tue Jan 27 22:51:22 2004
@@ -1,5 +1,7 @@
 /* -*- Mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
diff -urN gnome-python-2.0.0.orig/gnome/vfs-dir-handle.c gnome-python-2.0.0/gnome/vfs-dir-handle.c
--- gnome-python-2.0.0.orig/gnome/vfs-dir-handle.c	Thu Jun  5 23:32:22 2003
+++ gnome-python-2.0.0/gnome/vfs-dir-handle.c	Tue Jan 27 22:52:15 2004
@@ -1,5 +1,7 @@
 /* -*- mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #include "pygnomevfs-private.h"
 
 PyObject *
diff -urN gnome-python-2.0.0.orig/gnome/vfs-file-info.c gnome-python-2.0.0/gnome/vfs-file-info.c
--- gnome-python-2.0.0.orig/gnome/vfs-file-info.c	Thu Jun  5 23:32:22 2003
+++ gnome-python-2.0.0/gnome/vfs-file-info.c	Tue Jan 27 22:52:48 2004
@@ -1,5 +1,7 @@
 /* -*- mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #include "pygnomevfs-private.h"
 
 PyObject *
diff -urN gnome-python-2.0.0.orig/gnome/vfs-handle.c gnome-python-2.0.0/gnome/vfs-handle.c
--- gnome-python-2.0.0.orig/gnome/vfs-handle.c	Thu Jun  5 23:32:22 2003
+++ gnome-python-2.0.0/gnome/vfs-handle.c	Tue Jan 27 22:52:58 2004
@@ -1,5 +1,7 @@
 /* -*- mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #include "pygnomevfs-private.h"
 
 PyObject *
diff -urN gnome-python-2.0.0.orig/gnome/vfs-uri.c gnome-python-2.0.0/gnome/vfs-uri.c
--- gnome-python-2.0.0.orig/gnome/vfs-uri.c	Thu Jun  5 23:32:22 2003
+++ gnome-python-2.0.0/gnome/vfs-uri.c	Tue Jan 27 22:53:14 2004
@@ -1,5 +1,7 @@
 /* -*- mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #include "pygnomevfs-private.h"
 
 PyObject *
diff -urN gnome-python-2.0.0.orig/gnome/zvtmodule.c gnome-python-2.0.0/gnome/zvtmodule.c
--- gnome-python-2.0.0.orig/gnome/zvtmodule.c	Sun Aug 24 22:37:12 2003
+++ gnome-python-2.0.0/gnome/zvtmodule.c	Tue Jan 27 22:53:33 2004
@@ -1,5 +1,7 @@
 /* -*- Mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
diff -urN gnome-python-2.0.0.orig/gnomeprint/art-gtype.c gnome-python-2.0.0/gnomeprint/art-gtype.c
--- gnome-python-2.0.0.orig/gnomeprint/art-gtype.c	Wed Jan 15 06:45:24 2003
+++ gnome-python-2.0.0/gnomeprint/art-gtype.c	Tue Jan 27 22:55:28 2004
@@ -1,4 +1,7 @@
 /* -*- Mode: C; c-basic-offset: 4 -*- */
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #include <string.h>
 #include "art-gtype.h"
 
diff -urN gnome-python-2.0.0.orig/gnomeprint/printmodule.c gnome-python-2.0.0/gnomeprint/printmodule.c
--- gnome-python-2.0.0.orig/gnomeprint/printmodule.c	Sun Aug 24 22:36:29 2003
+++ gnome-python-2.0.0/gnomeprint/printmodule.c	Tue Jan 27 22:55:48 2004
@@ -1,5 +1,7 @@
 /* -*- Mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
diff -urN gnome-python-2.0.0.orig/gnomeprint/printuimodule.c gnome-python-2.0.0/gnomeprint/printuimodule.c
--- gnome-python-2.0.0.orig/gnomeprint/printuimodule.c	Sun Aug 24 22:36:37 2003
+++ gnome-python-2.0.0/gnomeprint/printuimodule.c	Tue Jan 27 22:55:59 2004
@@ -1,5 +1,7 @@
 /* -*- Mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
diff -urN gnome-python-2.0.0.orig/gtkhtml2/gtkhtml2module.c gnome-python-2.0.0/gtkhtml2/gtkhtml2module.c
--- gnome-python-2.0.0.orig/gtkhtml2/gtkhtml2module.c	Mon Sep  1 20:00:08 2003
+++ gnome-python-2.0.0/gtkhtml2/gtkhtml2module.c	Tue Jan 27 22:57:10 2004
@@ -1,5 +1,7 @@
 /* -*- Mode: C; c-basic-offset: 4 -*- */
-
+#define NO_IMPORT_PYGOBJECT 1
+#define NO_IMPORT_PYGTK 1
+#define NO_IMPORT_PYORBIT 1
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
diff -urN gnome-python-2.0.0.orig/setup.py gnome-python-2.0.0/setup.py
--- gnome-python-2.0.0.orig/setup.py	Mon Sep  1 22:47:50 2003
+++ gnome-python-2.0.0/setup.py	Wed Jan 28 00:00:45 2004
@@ -40,7 +40,7 @@
 LIBGNOME_REQUIRED_VERSION           = '2.0.0'
 LIBGNOMEUI_REQUIRED_VERSION         = '2.0.0'
 LIBGNOMECANVAS_REQUIRED_VERSION     = '2.0.0'
-LIBGNOMEVFS_REQUIRED_VERSION        = '2.0.0'
+LIBGNOMEVFS_REQUIRED_VERSION        = '22.0.0'
 LIBZVT_REQUIRED_VERSION             = '2.0.0'
 GCONF_REQUIRED_VERSION              = '1.2.0'
 BONOBO_ACTIVATION_REQUIRED_VERSION  = '1.0.0'
