diff -Nurd -x'*~' libgnomecanvas-2.10.2.orig/Makefile.in libgnomecanvas-2.10.2/Makefile.in
--- libgnomecanvas-2.10.2.orig/Makefile.in	2005-05-24 15:19:20.000000000 -0400
+++ libgnomecanvas-2.10.2/Makefile.in	2005-07-17 14:36:45.000000000 -0400
@@ -180,7 +180,7 @@
 
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
 
-SUBDIRS = libgnomecanvas glade demos po tests docs
+SUBDIRS = libgnomecanvas glade demos po docs
 
 EXTRA_DIST = COPYING.LIB NEWS libgnomecanvas-zip.in
 subdir = .
diff -Nurd -x'*~' libgnomecanvas-2.10.2.orig/demos/Makefile.in libgnomecanvas-2.10.2/demos/Makefile.in
--- libgnomecanvas-2.10.2.orig/demos/Makefile.in	2005-05-24 15:19:21.000000000 -0400
+++ libgnomecanvas-2.10.2/demos/Makefile.in	2005-07-17 14:36:45.000000000 -0400
@@ -192,7 +192,7 @@
 
 LDADD = \
 	$(top_builddir)/libgnomecanvas/libgnomecanvas-2.la \
-	$(INTLLIBS)
+	$(LIBGNOMECANVAS_LIBS) $(INTLLIBS)
 
 
 noinst_PROGRAMS = canvas_demo
diff -Nurd -x'*~' libgnomecanvas-2.10.2.orig/ltmain.sh libgnomecanvas-2.10.2/ltmain.sh
--- libgnomecanvas-2.10.2.orig/ltmain.sh	2005-04-26 14:23:39.000000000 -0400
+++ libgnomecanvas-2.10.2/ltmain.sh	2005-07-17 14:36:45.000000000 -0400
@@ -2446,7 +2446,6 @@
 	   { test "$prefer_static_libs" = no || test -z "$old_library"; }; then
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
 	  fi
 	  # This is a shared library
 
@@ -3996,6 +3995,12 @@
 	  if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	    cmds=$archive_expsym_cmds
 	  else
+           if test "x$verstring" = "x0.0"; then
+              tmp_verstring=
+            else
+              tmp_verstring="$verstring"
+            fi
+
 	    cmds=$archive_cmds
 	    fi
 	  fi
