diff -Naur gnome-games-1.4.0.4.old/aisleriot/menu.c gnome-games-1.4.0.4.new/aisleriot/menu.c
--- gnome-games-1.4.0.4.old/aisleriot/menu.c	Wed Feb 28 15:57:43 2001
+++ gnome-games-1.4.0.4.new/aisleriot/menu.c	Sat Jun  8 18:28:09 2002
@@ -18,6 +18,7 @@
  */
 
 #include <config.h>
+#include <sys/types.h>
 #include <dirent.h>
 #include "gnome.h"
 #include "sol.h"
diff -Naur gnome-games-1.4.0.4.old/freecell/gdk-card-image/Makefile.in gnome-games-1.4.0.4.new/freecell/gdk-card-image/Makefile.in
--- gnome-games-1.4.0.4.old/freecell/gdk-card-image/Makefile.in	Sat May 25 12:18:53 2002
+++ gnome-games-1.4.0.4.new/freecell/gdk-card-image/Makefile.in	Sat Jun  8 18:28:09 2002
@@ -163,7 +163,7 @@
 cardslotdir = $(datadir)/pixmaps/cards/slots/
 cardslot_DATA = hand.png plain.png
 
-INCLUDES = -I$(includedir) $(GNOME_INCLUDEDIR)
+INCLUDES = -I$(includedir) $(GNOME_INCLUDEDIR) $(GDK_PIXBUF_CFLAGS)
 
 EXTRA_DIST = $(cardback_DATA) $(cardhonor_DATA) $(cardjoker_DATA) 	$(cardrank_DATA) $(cardsuitsmall_DATA) $(cardsuitlarge_DATA) 	$(cardsuitmedium_DATA) $(cardslot_DATA)
 
diff -Naur gnome-games-1.4.0.4.old/gnect/src/adjmtrx.c gnome-games-1.4.0.4.new/gnect/src/adjmtrx.c
--- gnome-games-1.4.0.4.old/gnect/src/adjmtrx.c	Thu Dec 20 22:24:43 2001
+++ gnome-games-1.4.0.4.new/gnect/src/adjmtrx.c	Sun Jun  9 07:55:51 2002
@@ -20,11 +20,14 @@
 
 */
 
+#include <config.h>
 
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+# include <malloc.h>
+#endif
 
 #include "connect4.h"
 #include "con4vals.h"
diff -Naur gnome-games-1.4.0.4.old/gnect/src/bintree.c gnome-games-1.4.0.4.new/gnect/src/bintree.c
--- gnome-games-1.4.0.4.old/gnect/src/bintree.c	Thu Dec 20 22:24:43 2001
+++ gnome-games-1.4.0.4.new/gnect/src/bintree.c	Sun Jun  9 07:55:51 2002
@@ -20,11 +20,14 @@
 
 */
 
+#include <config.h>
 
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+# include <malloc.h>
+#endif
 
 #include "connect4.h"
 #include "con4vals.h"
diff -Naur gnome-games-1.4.0.4.old/gnect/src/connect4.c gnome-games-1.4.0.4.new/gnect/src/connect4.c
--- gnome-games-1.4.0.4.old/gnect/src/connect4.c	Thu Dec 20 22:24:43 2001
+++ gnome-games-1.4.0.4.new/gnect/src/connect4.c	Sun Jun  9 07:55:51 2002
@@ -23,11 +23,14 @@
 
 */
 
+#include <config.h>
 
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+# include <malloc.h>
+#endif
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
diff -Naur gnome-games-1.4.0.4.old/gnect/src/evaluate.c gnome-games-1.4.0.4.new/gnect/src/evaluate.c
--- gnome-games-1.4.0.4.old/gnect/src/evaluate.c	Thu Dec 20 22:24:43 2001
+++ gnome-games-1.4.0.4.new/gnect/src/evaluate.c	Sun Jun  9 07:55:51 2002
@@ -20,12 +20,14 @@
 
 */
 
-
+#include <config.h>
 
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>
+#endif
 #include <time.h>
 #include <math.h>
 
diff -Naur gnome-games-1.4.0.4.old/gnect/src/heurist.c gnome-games-1.4.0.4.new/gnect/src/heurist.c
--- gnome-games-1.4.0.4.old/gnect/src/heurist.c	Thu Dec 20 22:24:43 2001
+++ gnome-games-1.4.0.4.new/gnect/src/heurist.c	Sun Jun  9 07:55:51 2002
@@ -20,11 +20,14 @@
 
 */
 
+#include <config.h>
 
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+# include <malloc.h>
+#endif
 
 #include "connect4.h"
 #include "con4vals.h"
diff -Naur gnome-games-1.4.0.4.old/gnect/src/ia_main.c gnome-games-1.4.0.4.new/gnect/src/ia_main.c
--- gnome-games-1.4.0.4.old/gnect/src/ia_main.c	Thu Dec 20 22:24:43 2001
+++ gnome-games-1.4.0.4.new/gnect/src/ia_main.c	Sun Jun  9 07:55:51 2002
@@ -20,11 +20,14 @@
 
 */
 
+#include <config.h>
 
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+# include <malloc.h>
+#endif
 #include <time.h>
 
 #include <gtk/gtk.h>
diff -Naur gnome-games-1.4.0.4.old/gnect/src/pbsolver.c gnome-games-1.4.0.4.new/gnect/src/pbsolver.c
--- gnome-games-1.4.0.4.old/gnect/src/pbsolver.c	Thu Dec 20 22:24:43 2001
+++ gnome-games-1.4.0.4.new/gnect/src/pbsolver.c	Sun Jun  9 07:55:51 2002
@@ -20,11 +20,14 @@
 
 */
 
+#include <config.h>
 
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-#include <malloc.h>
+#ifdef HAVE_MALLOC_H
+# include <malloc.h>
+#endif
 
 #include "connect4.h"
 #include "con4vals.h"
diff -Naur gnome-games-1.4.0.4.old/gnome-stones/objects/Makefile.in gnome-games-1.4.0.4.new/gnome-stones/objects/Makefile.in
--- gnome-games-1.4.0.4.old/gnome-stones/objects/Makefile.in	Sat May 25 12:19:11 2002
+++ gnome-games-1.4.0.4.new/gnome-stones/objects/Makefile.in	Sun Jun  9 08:01:17 2002
@@ -140,7 +140,7 @@
 
 objectdir = $(libdir)/gnome-stones/objects
 
-INCLUDES =  	-I$(GSTONES_SRC_DIR)					-DGNOMELOCALEDIR=\""$(datadir)/locale"\" 		-I$(includedir)						$(GNOME_INCLUDEDIR)
+INCLUDES =  	-I$(GSTONES_SRC_DIR)					-DGNOMELOCALEDIR=\""$(datadir)/locale"\" 		-I$(includedir)						$(GNOME_INCLUDEDIR) $(GDK_PIXBUF_CFLAGS)
 
 
 object_LTLIBRARIES = libgnome-stones.la libgnomekoban.la
@@ -155,10 +155,10 @@
 
 DEFS = @DEFS@ -I. -I$(srcdir) -I../..
 LIBS = @LIBS@
-libgnome_stones_la_LDFLAGS = 
+libgnome_stones_la_LDFLAGS = -module -avoid-version
 libgnome_stones_la_LIBADD = 
 libgnome_stones_la_OBJECTS =  gnome-stones.lo
-libgnomekoban_la_LDFLAGS = 
+libgnomekoban_la_LDFLAGS = -module -avoid-version
 libgnomekoban_la_LIBADD = 
 libgnomekoban_la_OBJECTS =  gnomekoban.lo
 COMPILE = $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
diff -Naur gnome-games-1.4.0.4.old/gnotski/Makefile.in gnome-games-1.4.0.4.new/gnotski/Makefile.in
--- gnome-games-1.4.0.4.old/gnotski/Makefile.in	Sat May 25 12:19:13 2002
+++ gnome-games-1.4.0.4.new/gnotski/Makefile.in	Sat Jun  8 18:28:10 2002
@@ -431,7 +431,7 @@
 
 
 install-data-local:
-	-$(mkinstalldirs) $(DESTDIR)$(scoredir)
+	-$(mkinstalldirs) $(scoredir)
 	-chgrp games $(DESTDIR)$(bindir)/gnotski && chmod 2111 $(DESTDIR)$(bindir)/gnotski
 	-touch             $(scoredir)/gnotski.1.scores
 	-chown games.games $(scoredir)/gnotski.1.scores
diff -Naur gnome-games-1.4.0.4.old/iagno/network.c gnome-games-1.4.0.4.new/iagno/network.c
--- gnome-games-1.4.0.4.old/iagno/network.c	Fri Nov 23 03:35:31 2001
+++ gnome-games-1.4.0.4.new/iagno/network.c	Sun Jun  9 07:46:21 2002
@@ -24,6 +24,12 @@
 #include "gnothello.h"
 #include "network.h"
 
+#if defined(__APPLE__)
+# ifndef socklen_t
+#  define socklen_t int
+# endif
+#endif
+
 /* (('g'<<8)+'n' */
 #define GAME_PORT "26478"
 
diff -Naur gnome-games-1.4.0.4.old/ltmain.sh gnome-games-1.4.0.4.new/ltmain.sh
--- gnome-games-1.4.0.4.old/ltmain.sh	Sat May 25 12:17:14 2002
+++ gnome-games-1.4.0.4.new/ltmain.sh	Sat Jun  8 18:28:10 2002
@@ -2947,6 +2947,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+	  else
+	    tmp_verstring="$verstring"
+	  fi
 	  eval cmds=\"$archive_cmds\"
 	fi
 	save_ifs="$IFS"; IFS='~'
diff -Naur gnome-games-1.4.0.4.old/xbill/MCursor.cc gnome-games-1.4.0.4.new/xbill/MCursor.cc
--- gnome-games-1.4.0.4.old/xbill/MCursor.cc	Sat Jul 14 10:47:27 2001
+++ gnome-games-1.4.0.4.new/xbill/MCursor.cc	Sat Jun  8 18:28:10 2002
@@ -10,8 +10,8 @@
 	file = g_strdup_printf ("%s/%s.xpm", dir, name);
 
 	pixmap = gdk_pixmap_colormap_create_from_xpm(ui.display, ui.colormap,
-						     &bitmap, NULL, file);
-	if (pixmap == NULL) {
+						     &bitmap, 0, file);
+	if (pixmap == 0) {
 		printf ("cannot open %s\n", file);
 		exit(1);
 	} else
@@ -21,8 +21,8 @@
 	  file = g_strdup_printf ("%s/%s_mask.xpm", dir, name);
 
 	  pixmap = gdk_pixmap_colormap_create_from_xpm(ui.display, ui.colormap,
-						       &mask, NULL, file);
-	  if (pixmap == NULL) {
+						       &mask, 0, file);
+	  if (pixmap == 0) {
 	    printf("cannot open %s\n", file);
 	    exit(1);
 	  } else
