diff -Naur gstreamer-0.6.4.old/configure gstreamer-0.6.4.new/configure
--- gstreamer-0.6.4.old/configure	Tue Oct 14 03:17:16 2003
+++ gstreamer-0.6.4.new/configure	Tue Dec 23 03:08:41 2003
@@ -5840,7 +5840,7 @@
       allow_undefined_flag='-undefined suppress'
       ;;
     *) # Darwin 1.3 on
-      allow_undefined_flag='-flat_namespace -undefined suppress'
+      allow_undefined_flag='-undefined dynamic_lookup'
       ;;
     esac
     # FIXME: Relying on posixy $() will cause problems for
diff -Naur gstreamer-0.6.4.old/configure.ac gstreamer-0.6.4.new/configure.ac
--- gstreamer-0.6.4.old/configure.ac	Tue Oct 14 00:03:13 2003
+++ gstreamer-0.6.4.new/configure.ac	Tue Dec 23 03:08:24 2003
@@ -128,6 +128,15 @@
 dnl Check for a way to display the function name in debug output
 GST_CHECK_FUNCTION()
 
+dnl check if we should use -Wa,-mregnames on PowerPC, so we can use
+dnl symbolic register names in inline asm
+if test x$HAVE_CPU_PPC = xyes ; then
+  AS_COMPILER_FLAG(["-Wa,-mregnames"],
+    [GST_INT_CFLAGS="$GST_INT_CFLAGS -Wa,-mregnames"],
+    true)
+fi
+
+
 dnl Check for essential libraries first:
 dnl ====================================
 
@@ -558,3 +567,4 @@
 gstreamer.spec,
 echo "$infomessages", infomessages="$infomessages"
 )
+
diff -Naur gstreamer-0.6.4.old/gst/autoplug/gstspideridentity.h gstreamer-0.6.4.new/gst/autoplug/gstspideridentity.h
--- gstreamer-0.6.4.old/gst/autoplug/gstspideridentity.h	Sun Dec  1 10:37:58 2002
+++ gstreamer-0.6.4.new/gst/autoplug/gstspideridentity.h	Tue Dec 23 03:08:24 2003
@@ -30,7 +30,7 @@
 
 G_BEGIN_DECLS
 
-GstElementDetails gst_spider_identity_details;
+extern GstElementDetails gst_spider_identity_details;
 
 
 #define GST_TYPE_SPIDER_IDENTITY \
diff -Naur gstreamer-0.6.4.old/gst/elements/gstfakesink.h gstreamer-0.6.4.new/gst/elements/gstfakesink.h
--- gstreamer-0.6.4.old/gst/elements/gstfakesink.h	Wed Oct 30 18:17:32 2002
+++ gstreamer-0.6.4.new/gst/elements/gstfakesink.h	Tue Dec 23 03:08:24 2003
@@ -30,7 +30,7 @@
 
 G_BEGIN_DECLS
 
-GstElementDetails gst_fakesink_details;
+extern GstElementDetails gst_fakesink_details;
 
 
 #define GST_TYPE_FAKESINK \
diff -Naur gstreamer-0.6.4.old/gst/elements/gstfakesrc.h gstreamer-0.6.4.new/gst/elements/gstfakesrc.h
--- gstreamer-0.6.4.old/gst/elements/gstfakesrc.h	Wed Oct 30 18:17:32 2002
+++ gstreamer-0.6.4.new/gst/elements/gstfakesrc.h	Tue Dec 23 03:08:24 2003
@@ -30,7 +30,7 @@
 
 G_BEGIN_DECLS
 
-GstElementDetails gst_fakesrc_details;
+extern GstElementDetails gst_fakesrc_details;
 
 typedef enum {
   FAKESRC_FIRST_LAST_LOOP = 1,
diff -Naur gstreamer-0.6.4.old/gst/elements/gstfdsink.h gstreamer-0.6.4.new/gst/elements/gstfdsink.h
--- gstreamer-0.6.4.old/gst/elements/gstfdsink.h	Wed Oct 30 18:17:32 2002
+++ gstreamer-0.6.4.new/gst/elements/gstfdsink.h	Tue Dec 23 03:08:24 2003
@@ -30,7 +30,7 @@
 
 G_BEGIN_DECLS
 
-GstElementDetails gst_fdsink_details;
+extern GstElementDetails gst_fdsink_details;
 
 
 #define GST_TYPE_FDSINK \
diff -Naur gstreamer-0.6.4.old/gst/elements/gstfdsrc.h gstreamer-0.6.4.new/gst/elements/gstfdsrc.h
--- gstreamer-0.6.4.old/gst/elements/gstfdsrc.h	Fri Nov 22 08:52:30 2002
+++ gstreamer-0.6.4.new/gst/elements/gstfdsrc.h	Tue Dec 23 03:08:24 2003
@@ -30,7 +30,7 @@
 
 G_BEGIN_DECLS
 
-GstElementDetails gst_fdsrc_details;
+extern GstElementDetails gst_fdsrc_details;
 
 
 #define GST_TYPE_FDSRC \
diff -Naur gstreamer-0.6.4.old/gst/elements/gstfilesink.h gstreamer-0.6.4.new/gst/elements/gstfilesink.h
--- gstreamer-0.6.4.old/gst/elements/gstfilesink.h	Mon May 19 07:22:13 2003
+++ gstreamer-0.6.4.new/gst/elements/gstfilesink.h	Tue Dec 23 03:08:24 2003
@@ -30,7 +30,7 @@
 
 G_BEGIN_DECLS
 
-GstElementDetails gst_filesink_details;
+extern GstElementDetails gst_filesink_details;
 
 
 #define GST_TYPE_FILESINK \
diff -Naur gstreamer-0.6.4.old/gst/elements/gstidentity.h gstreamer-0.6.4.new/gst/elements/gstidentity.h
--- gstreamer-0.6.4.old/gst/elements/gstidentity.h	Wed Oct 30 18:17:32 2002
+++ gstreamer-0.6.4.new/gst/elements/gstidentity.h	Tue Dec 23 03:08:24 2003
@@ -30,7 +30,7 @@
 
 G_BEGIN_DECLS
 
-GstElementDetails gst_identity_details;
+extern GstElementDetails gst_identity_details;
 
 
 #define GST_TYPE_IDENTITY \
diff -Naur gstreamer-0.6.4.old/gst/elements/gstmd5sink.h gstreamer-0.6.4.new/gst/elements/gstmd5sink.h
--- gstreamer-0.6.4.old/gst/elements/gstmd5sink.h	Sat Nov 30 04:08:47 2002
+++ gstreamer-0.6.4.new/gst/elements/gstmd5sink.h	Tue Dec 23 03:08:24 2003
@@ -30,7 +30,7 @@
 
 G_BEGIN_DECLS
 
-GstElementDetails gst_md5sink_details;
+extern GstElementDetails gst_md5sink_details;
 
 
 #define GST_TYPE_MD5SINK \
diff -Naur gstreamer-0.6.4.old/gst/elements/gstmultidisksrc.h gstreamer-0.6.4.new/gst/elements/gstmultidisksrc.h
--- gstreamer-0.6.4.old/gst/elements/gstmultidisksrc.h	Mon May 19 07:22:13 2003
+++ gstreamer-0.6.4.new/gst/elements/gstmultidisksrc.h	Tue Dec 23 03:08:24 2003
@@ -28,7 +28,7 @@
 
 G_BEGIN_DECLS
 
-GstElementDetails gst_multidisksrc_details;
+extern GstElementDetails gst_multidisksrc_details;
 
 #define GST_TYPE_MULTIDISKSRC \
   (gst_multidisksrc_get_type())
diff -Naur gstreamer-0.6.4.old/gst/elements/gstpipefilter.h gstreamer-0.6.4.new/gst/elements/gstpipefilter.h
--- gstreamer-0.6.4.old/gst/elements/gstpipefilter.h	Mon May 19 07:22:13 2003
+++ gstreamer-0.6.4.new/gst/elements/gstpipefilter.h	Tue Dec 23 03:08:24 2003
@@ -31,7 +31,7 @@
 
 G_BEGIN_DECLS
 
-GstElementDetails gst_pipefilter_details;
+extern GstElementDetails gst_pipefilter_details;
 
 #define GST_TYPE_PIPEFILTER \
   (gst_pipefilter_get_type())
diff -Naur gstreamer-0.6.4.old/gst/elements/gstshaper.h gstreamer-0.6.4.new/gst/elements/gstshaper.h
--- gstreamer-0.6.4.old/gst/elements/gstshaper.h	Mon May 19 07:22:13 2003
+++ gstreamer-0.6.4.new/gst/elements/gstshaper.h	Tue Dec 23 03:08:24 2003
@@ -30,7 +30,7 @@
 
 G_BEGIN_DECLS
 
-GstElementDetails gst_shaper_details;
+extern GstElementDetails gst_shaper_details;
 
 
 #define GST_TYPE_SHAPER \
diff -Naur gstreamer-0.6.4.old/gst/elements/gststatistics.h gstreamer-0.6.4.new/gst/elements/gststatistics.h
--- gstreamer-0.6.4.old/gst/elements/gststatistics.h	Wed Oct 30 18:17:33 2002
+++ gstreamer-0.6.4.new/gst/elements/gststatistics.h	Tue Dec 23 03:08:24 2003
@@ -29,7 +29,7 @@
 
 G_BEGIN_DECLS
 
-GstElementDetails gst_statistics_details;
+extern GstElementDetails gst_statistics_details;
 
 
 #define GST_TYPE_STATISTICS \
diff -Naur gstreamer-0.6.4.old/libs/ext/cothreads/cothreads/cothreads.c gstreamer-0.6.4.new/libs/ext/cothreads/cothreads/cothreads.c
--- gstreamer-0.6.4.old/libs/ext/cothreads/cothreads/cothreads.c	Wed Aug 27 07:08:01 2003
+++ gstreamer-0.6.4.new/libs/ext/cothreads/cothreads/cothreads.c	Tue Dec 23 03:08:24 2003
@@ -20,6 +20,7 @@
  */
 
 #include "pth_p.h" /* this pulls in everything */
+#include <sys/time.h>
 #include <stdlib.h>
 #include <sys/mman.h>
 #include <sys/resource.h>
diff -Naur gstreamer-0.6.4.old/ltmain.sh gstreamer-0.6.4.new/ltmain.sh
--- gstreamer-0.6.4.old/ltmain.sh	Tue Oct 14 03:17:09 2003
+++ gstreamer-0.6.4.new/ltmain.sh	Tue Dec 23 03:08:24 2003
@@ -4050,10 +4050,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test $need_relink = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test $need_relink = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
