diff -Naur ORBit2-2.8.1.old/ltmain.sh ORBit2-2.8.1.new/ltmain.sh
--- ORBit2-2.8.1.old/ltmain.sh	Thu Jun  5 21:09:04 2003
+++ ORBit2-2.8.1.new/ltmain.sh	Thu Nov  6 20:01:55 2003
@@ -5145,10 +5145,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test "$need_relink" = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
