diff -Nurd beecrypt-4.1.2/c++/Makefile.in beecrypt-4.1.2-new/c++/Makefile.in
--- beecrypt-4.1.2/c++/Makefile.in	2004-12-21 06:38:25.000000000 -0500
+++ beecrypt-4.1.2-new/c++/Makefile.in	2009-02-11 09:17:13.000000000 -0500
@@ -266,7 +266,7 @@
 bstream.cxx \
 resource.cxx
 
-libbeecrypt_cxx_la_LIBADD = ../libbeecrypt.la beeyond/libcxxbeeyond.la crypto/libcxxcrypto.la io/libcxxio.la lang/libcxxlang.la util/libcxxutil.la security/libcxxsecurity.la -licuuc -licuio
+libbeecrypt_cxx_la_LIBADD = ../libbeecrypt.la beeyond/libcxxbeeyond.la crypto/libcxxcrypto.la io/libcxxio.la lang/libcxxlang.la util/libcxxutil.la security/libcxxsecurity.la -licucore
 libbeecrypt_cxx_la_LDFLAGS = -no-undefined -version-info $(LIBBEECRYPT_CXX_LT_CURRENT):$(LIBBEECRYPT_CXX_LT_REVISION):$(LIBBEECRYPT_CXX_LT_AGE)
 cxx_HEADERS = \
 array.h \
diff -Nurd beecrypt-4.1.2/c++/beeyond/BeeCertificate.h beecrypt-4.1.2-new/c++/beeyond/BeeCertificate.h
--- beecrypt-4.1.2/c++/beeyond/BeeCertificate.h	2004-11-03 04:37:27.000000000 -0500
+++ beecrypt-4.1.2-new/c++/beeyond/BeeCertificate.h	2009-02-11 09:20:59.000000000 -0500
@@ -150,11 +150,11 @@
 			mutable String* str;
 
 			BeeCertificate();
-			BeeCertificate(InputStream& in) throw (IOException);
 
 			bytearray* encodeTBS() const;
 
 		public:
+			BeeCertificate(InputStream& in) throw (IOException);
 			BeeCertificate(const BeeCertificate&);
 			virtual ~BeeCertificate();
 
diff -Nurd beecrypt-4.1.2/c++/beeyond/Makefile.in beecrypt-4.1.2-new/c++/beeyond/Makefile.in
--- beecrypt-4.1.2/c++/beeyond/Makefile.in	2004-12-21 06:38:25.000000000 -0500
+++ beecrypt-4.1.2-new/c++/beeyond/Makefile.in	2009-02-11 09:17:13.000000000 -0500
@@ -140,7 +140,7 @@
 LDFLAGS = @LDFLAGS@
 LIBALT = @LIBALT@
 LIBOBJS = @LIBOBJS@
-LIBS = -licuuc -licuio
+LIBS = -licucore
 LIBTOOL = @LIBTOOL@
 LN_S = @LN_S@
 LTLIBOBJS = @LTLIBOBJS@
diff -Nurd beecrypt-4.1.2/c++/io/Makefile.in beecrypt-4.1.2-new/c++/io/Makefile.in
--- beecrypt-4.1.2/c++/io/Makefile.in	2004-12-21 06:38:26.000000000 -0500
+++ beecrypt-4.1.2-new/c++/io/Makefile.in	2009-02-11 09:17:13.000000000 -0500
@@ -255,7 +255,7 @@
 PrintStream.cxx \
 PushbackInputStream.cxx
 
-libcxxio_la_LIBADD = -licuuc
+libcxxio_la_LIBADD = -licucore
 cxxio_HEADERS = \
 ByteArrayInputStream.h \
 ByteArrayOutputStream.h \
@@ -276,7 +276,7 @@
 
 TESTS = test
 test_SOURCES = test.cxx
-test_LDADD = libcxxio.la ../lang/libcxxlang.la $(top_builddir)/libbeecrypt.la -licuuc -licuio
+test_LDADD = libcxxio.la ../lang/libcxxlang.la $(top_builddir)/libbeecrypt.la -licucore
 all: all-am
 
 .SUFFIXES:
diff -Nurd beecrypt-4.1.2/c++/provider/Makefile.in beecrypt-4.1.2-new/c++/provider/Makefile.in
--- beecrypt-4.1.2/c++/provider/Makefile.in	2004-12-21 06:38:27.000000000 -0500
+++ beecrypt-4.1.2-new/c++/provider/Makefile.in	2009-02-11 09:17:13.000000000 -0500
@@ -160,7 +160,7 @@
 LDFLAGS = @LDFLAGS@
 LIBALT = @LIBALT@
 LIBOBJS = @LIBOBJS@
-LIBS = -licuuc -licuio
+LIBS = -licucore
 LIBTOOL = @LIBTOOL@
 LN_S = @LN_S@
 LTLIBOBJS = @LTLIBOBJS@
diff -Nurd beecrypt-4.1.2/c++/security/Security.h beecrypt-4.1.2-new/c++/security/Security.h
--- beecrypt-4.1.2/c++/security/Security.h	2004-11-02 02:06:58.000000000 -0500
+++ beecrypt-4.1.2-new/c++/security/Security.h	2009-02-11 09:17:13.000000000 -0500
@@ -75,8 +75,6 @@
 		public:
 			typedef vector<const Provider*> provider_vector;
 			typedef provider_vector::iterator provider_vector_iterator;
-
-		private:
 			struct spi
 			{
 				Object* cspi;
@@ -91,6 +89,7 @@
 			static spi* getSpi(const String& algo, const String& type, const Provider&) throw (NoSuchAlgorithmException);
 			static spi* getFirstSpi(const String& type);
 
+		private:
 			static const String& getKeyStoreDefault();
 
 			static bool _init;
