diff -ruN mozilla-orig/.mozconfig mozilla/.mozconfig
--- mozilla-orig/.mozconfig	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/.mozconfig	2007-11-08 09:49:21.000000000 -0500
@@ -0,0 +1,42 @@
+# sh
+# Build configuration script
+#
+# See http://www.mozilla.org/build/unix.html for build instructions.
+#
+
+. $topsrcdir/browser/config/mozconfig
+
+# Options for client.mk.
+# environment variables to set
+#mk_add_options MOZILLA_OFFICIAL=1
+
+# Options for 'configure' (same as command-line options).
+ac_add_options --enable-official-branding
+ac_add_options --enable-default-toolkit=gtk2
+ac_add_options --enable-xft
+#ac_add_options --disable-xft
+#ac_add_options --enable-freetype2
+ac_add_options --disable-freetype2
+ac_add_options --enable-postscript
+ac_add_options --enable-xinerama
+ac_add_options --with-pthreads
+ac_add_options --disable-gnomevfs
+ac_add_options --disable-gnomeui
+ac_add_options --enable-extensions=default,-gnomevfs,xforms,schema-validation
+ac_add_options --enable-canvas
+ac_add_options --enable-svg
+
+# prebinding disabled or otherwise don't get binary
+ac_add_options --disable-prebinding
+ac_add_options --disable-debug
+ac_add_options --disable-tests
+ac_add_options --enable-strip
+ac_add_options --enable-optimize='-O2'
+ac_add_options --enable-macos-target=10.3
+ac_add_options --with-default-mozilla-five-home=@PREFIX@/lib/firefox1.5
+ac_add_options --with-system-jpeg=@PREFIX@
+ac_add_options --with-system-png=@PREFIX@
+ac_add_options --with-freetype-prefix=@PREFIX@/lib/freetype219
+
+#ac_add_options --with-ft-prefix=@PREFIX@/lib/freetype2
+#ac_add_options --with-ft-exec-prefix=@PREFIX@/lib/freetype2/bin
diff -ruN mozilla-orig/browser/components/build/Makefile.in mozilla/browser/components/build/Makefile.in
--- mozilla-orig/browser/components/build/Makefile.in	2005-06-22 22:25:04.000000000 -0400
+++ mozilla/browser/components/build/Makefile.in	2007-11-08 09:49:21.000000000 -0500
@@ -74,6 +74,12 @@
 	$(MOZ_COMPONENT_LIBS) \
 	$(NULL)
 
+# fix for building on Mac/X11 (bug 257773)
+# dependencies for gkgfx
+ifeq ($(MOZ_WIDGET_TOOLKIT), gtk2)
+EXTRA_DSO_LDOPTS += $(MOZ_GTK2_LIBS)
+endif
+
 # Need to link to CoreFoundation for Mac Migrators (PList reading code)
 ifneq (,$(filter mac cocoa,$(MOZ_WIDGET_TOOLKIT)))
 EXTRA_DSO_LDOPTS += \
diff -ruN mozilla-orig/browser/locales/en-US/profile/bookmarks.html mozilla/browser/locales/en-US/profile/bookmarks.html
--- mozilla-orig/browser/locales/en-US/profile/bookmarks.html	2005-11-03 03:30:35.000000000 -0500
+++ mozilla/browser/locales/en-US/profile/bookmarks.html	2007-11-08 09:49:21.000000000 -0500
@@ -12,6 +12,7 @@
     <DL><p>
         <DT><A HREF="http://www.mozilla.com/products/firefox/central.html" ICON="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAABGdBTUEAAK/INwWK6QAAABl0RVh0U29mdHdhcmUAQWRvYmUgSW1hZ2VSZWFkeXHJZTwAAAHWSURBVHjaYvz//z8DJQAggJiQOe/fv2fv7Oz8rays/N+VkfG/iYnJfyD/1+rVq7ffu3dPFpsBAAHEAHIBCJ85c8bN2Nj4vwsDw/8zQLwKiO8CcRoQu0DxqlWrdsHUwzBAAIGJmTNnPgYa9j8UqhFElwPxf2MIDeIrKSn9FwSJoRkAEEAM0DD4DzMAyPi/G+QKY4hh5WAXGf8PDQ0FGwJ22d27CjADAAIIrLmjo+MXA9R2kAHvGBA2wwx6B8W7od6CeQcggKCmCEL8bgwxYCbUIGTDVkHDBia+CuotgACCueD3TDQN75D4xmAvCoK9ARMHBzAw0AECiBHkAlC0Mdy7x9ABNA3obAZXIAa6iKEcGlMVQHwWyjYuL2d4v2cPg8vZswx7gHyAAAK7AOif7SAbOqCmn4Ha3AHFsIDtgPq/vLz8P4MSkJ2W9h8ggBjevXvHDo4FQUQg/kdypqCg4H8lUIACnQ/SOBMYI8bAsAJFPcj1AAEEjwVQqLpAbXmH5BJjqI0gi9DTAAgDBBCcAVLkgmQ7yKCZxpCQxqUZhAECCJ4XgMl493ug21ZD+aDAXH0WLM4A9MZPXJkJIIAwTAR5pQMalaCABQUULttBGCCAGCnNzgABBgAMJ5THwGvJLAAAAABJRU5ErkJggg==" LAST_CHARSET="ISO-8859-1" ID="rdf:#$GvPhC3">Getting Started</A>
         <DT><A HREF="http://fxfeeds.mozilla.com/" LAST_MODIFIED="1094668003" FEEDURL="http://fxfeeds.mozilla.com/rss20.xml" ID="rdf:#$HvPhC3">Latest Headlines</A>
+	 <DT><A HREF="http://www.finkproject.org/">Fink - Home</A>
     </DL><p>
     <HR>
     <DT><H3 ID="rdf:#$6wPhC3">Quick Searches</H3>
diff -ruN mozilla-orig/build/unix/mozilla-config.in mozilla/build/unix/mozilla-config.in
--- mozilla-orig/build/unix/mozilla-config.in	2004-04-22 16:17:59.000000000 -0400
+++ mozilla/build/unix/mozilla-config.in	2007-11-08 09:49:21.000000000 -0500
@@ -115,7 +115,7 @@
 fi
 
 _nspr_libs="%FULL_NSPR_LIBS%"
-_xpcom_libs="-lxpcom $_nspr_libs"
+_xpcom_libs="-lxpcom -lxpcom_core $_nspr_libs"
 _js_libs="-ljs"
 
 if test "$echo_libs" = "yes"; then
diff -ruN mozilla-orig/build/unix/mozilla-xpcom.pc.in mozilla/build/unix/mozilla-xpcom.pc.in
--- mozilla-orig/build/unix/mozilla-xpcom.pc.in	2005-07-05 19:21:52.000000000 -0400
+++ mozilla/build/unix/mozilla-xpcom.pc.in	2007-11-08 09:49:21.000000000 -0500
@@ -8,5 +8,5 @@
 Description: The Mozilla Cross Platform Component Library
 Version: %MOZILLA_VERSION%
 Requires: %NSPR_NAME% >= %NSPR_VERSION%
-Libs: -L${libdir} -lxpcom
+Libs: -L${libdir} -lxpcom -lxpcom_core
 Cflags: -I${includedir} -I${includedir}/xpcom -I${includedir}/string
diff -ruN mozilla-orig/config/autoconf.mk.in mozilla/config/autoconf.mk.in
--- mozilla-orig/config/autoconf.mk.in	2006-04-21 16:50:30.000000000 -0400
+++ mozilla/config/autoconf.mk.in	2007-11-08 09:49:21.000000000 -0500
@@ -56,14 +56,14 @@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-includedir	= @includedir@/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+includedir	= @includedir@/$(MOZ_APP_NAME)1.5
 libdir		= @libdir@
 datadir		= @datadir@
 mandir		= @mandir@
-idldir		= @datadir@/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+idldir		= @datadir@/idl/$(MOZ_APP_NAME)1.5
 
-mozappdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-mredir		= $(libdir)/mre/mre-$(MOZ_APP_VERSION)
+mozappdir	= $(libdir)/$(MOZ_APP_NAME)1.5
+mredir		= $(libdir)/mre/mre1.5
 mrelibdir	= $(mredir)/lib
 
 DIST		= $(DEPTH)/dist
diff -ruN mozilla-orig/config/config.mk mozilla/config/config.mk
--- mozilla-orig/config/config.mk	2006-04-06 12:28:13.000000000 -0400
+++ mozilla/config/config.mk	2007-11-08 09:49:21.000000000 -0500
@@ -679,7 +679,7 @@
 ifeq ($(OS_ARCH),Darwin)
 ifdef USE_PREBINDING
 export LD_PREBIND=1
-export LD_SEG_ADDR_TABLE=$(shell cd $(topsrcdir); pwd)/config/prebind-address-table
+export LD_PREBIND_ALLOW_OVERLAP=1
 endif # USE_PREBINDING
 ifdef NEXT_ROOT
 export NEXT_ROOT
diff -ruN mozilla-orig/configure mozilla/configure
--- mozilla-orig/configure	2007-03-29 20:33:27.000000000 -0400
+++ mozilla/configure	2007-11-08 09:49:21.000000000 -0500
@@ -5516,7 +5516,7 @@
     ;;
 
 *-darwin*)
-    HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX -DXP_MACOSX -DNO_X11"
+    HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX"
     HOST_NSPR_MDCPUCFG='\"md/_darwin.cfg\"'
     HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-O3}"
     MOZ_FIX_LINK_PATHS='-Wl,-executable_path,$(DIST)/bin'
@@ -5802,10 +5802,10 @@
 
 *-darwin*) 
     _PEDANTIC=
-    CFLAGS="$CFLAGS -fpascal-strings -no-cpp-precomp -fno-common"
-    CXXFLAGS="$CXXFLAGS -fpascal-strings -no-cpp-precomp -fno-common"
+    CFLAGS="$CFLAGS -no-cpp-precomp -fno-common"
+    CXXFLAGS="$CXXFLAGS -no-cpp-precomp -fno-common"
     DLL_SUFFIX=".dylib"
-    DSO_LDOPTS=''
+    DSO_LDOPTS='-framework CoreFoundation'
     STRIP="$STRIP -x -S"
     _PLATFORM_DEFAULT_TOOLKIT='mac'
     MOZ_ENABLE_POSTSCRIPT=
diff -ruN mozilla-orig/fink/applications/firefox.desktop mozilla/fink/applications/firefox.desktop
--- mozilla-orig/fink/applications/firefox.desktop	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/fink/applications/firefox.desktop	2007-11-08 09:49:22.000000000 -0500
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Firefox Web Browser
+Comment=Browse the World Wide Web
+Exec=firefox
+Icon=firefox.xpm
+Terminal=false
+Type=Application
+Categories=Application;Network
+StartupNotify=true
diff -ruN mozilla-orig/fink/fink.js mozilla/fink/fink.js
--- mozilla-orig/fink/fink.js	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/fink/fink.js	2007-11-08 09:49:22.000000000 -0500
@@ -0,0 +1,27 @@
+pref("general.useragent.extra.vendorComment", "Fink Community Edition");
+
+// Give a default font set that can do BOLD
+pref("font.name.serif.x-western", "Bitstream Vera Serif");
+pref("font.name.sans-serif.x-western", "Bitstream Vera Sans");
+pref("font.size.variable.x-western", 15);
+
+// TrueType
+//pref("font.FreeType2.enable", true);
+//pref("font.freetype2.shared-library", "@PREFIX@/lib/freetype219/lib/libfreetype.6.dylib");
+//pref("font.freetype2.shared-library", "/usr/X11R6/lib/libfreetype.6.dylib");
+//pref("font.freetype2.autohinted", true);
+//pref("font.freetype2.unhinted", true);
+//pref("font.antialias.min", 10);
+//pref("font.embedded_bitmaps.max", 1000000);
+//pref("font.scale.tt_bitmap.dark_text.min", 64);
+//pref("font.scale.tt_bitmap.dark_text.gain", "0.0");
+//pref("font.directory.truetype.1", "/Library/Fonts");
+//pref("font.directory.truetype.2", "/Network/Library/Fonts");
+//pref("font.directory.truetype.3", "/System/Library/Fonts");
+//pref("font.directory.truetype.4", "/usr/X11R6/lib/X11/fonts/TTF");
+//pref("font.directory.truetype.5", "@PREFIX@/share/fonts/truetype");
+
+// AA with Bitmap scaling.
+//pref("font.scale.aa_bitmap.enable", true);
+//pref("font.scale.aa_bitmap.always", false);
+//pref("font.scale.aa_bitmap.min", 6);
diff -ruN mozilla-orig/modules/libpref/src/nsPrefService.cpp mozilla/modules/libpref/src/nsPrefService.cpp
--- mozilla-orig/modules/libpref/src/nsPrefService.cpp	2005-05-13 20:30:08.000000000 -0400
+++ mozilla/modules/libpref/src/nsPrefService.cpp	2007-11-08 09:49:22.000000000 -0500
@@ -748,6 +748,7 @@
 #elif defined(XP_BEOS)
       "beos.js"
 #endif
+      , "fink.js"
   };
 
   rv = pref_LoadPrefsInDir(defaultPrefDir, specialFiles, NS_ARRAY_LENGTH(specialFiles));
diff -ruN mozilla-orig/nsprpub/config/nspr.m4 mozilla/nsprpub/config/nspr.m4
--- mozilla-orig/nsprpub/config/nspr.m4	2001-05-11 20:36:57.000000000 -0400
+++ mozilla/nsprpub/config/nspr.m4	2007-11-08 09:49:22.000000000 -0500
@@ -5,7 +5,7 @@
 
 dnl AM_PATH_NSPR([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]]])
 dnl Test for NSPR, and define NSPR_CFLAGS and NSPR_LIBS
-AC_DEFUN(AM_PATH_NSPR,
+AC_DEFUN([AM_PATH_NSPR],
 [dnl
 
 AC_ARG_WITH(nspr-prefix,
diff -ruN mozilla-orig/nsprpub/configure mozilla/nsprpub/configure
--- mozilla-orig/nsprpub/configure	2007-04-27 16:01:01.000000000 -0400
+++ mozilla/nsprpub/configure	2007-11-08 09:49:22.000000000 -0500
@@ -3338,9 +3338,6 @@
 
     # Add Mac OS X support for loading CFM & CFBundle plugins
     if test -f /System/Library/Frameworks/Carbon.framework/Carbon; then
-        cat >> confdefs.h <<\EOF
-#define XP_MACOSX 1
-EOF
 
         OS_TARGET=MacOSX
 
@@ -4956,8 +4953,6 @@
 
 
 case $target in
-*-darwin*)
-    ;;
 *)
     echo $ac_n "checking for dlopen in -ldl""... $ac_c" 1>&6
 echo "configure:4964: checking for dlopen in -ldl" >&5
diff -ruN mozilla-orig/nsprpub/pr/include/md/_darwin.h mozilla/nsprpub/pr/include/md/_darwin.h
--- mozilla-orig/nsprpub/pr/include/md/_darwin.h	2006-12-06 13:02:35.000000000 -0500
+++ mozilla/nsprpub/pr/include/md/_darwin.h	2007-11-08 09:49:22.000000000 -0500
@@ -62,7 +62,8 @@
 
 #undef  HAVE_STACK_GROWING_UP
 #define HAVE_DLL
-#define USE_MACH_DYLD
+/* #define USE_MACH_DYLD */
+#define USE_DLFCN
 #define _PR_HAVE_SOCKADDR_LEN  
 #define _PR_STAT_HAS_ST_ATIMESPEC
 #define _PR_HAVE_LARGE_OFF_T
diff -ruN mozilla-orig/nsprpub/pr/src/Makefile.in mozilla/nsprpub/pr/src/Makefile.in
--- mozilla-orig/nsprpub/pr/src/Makefile.in	2006-10-18 20:39:28.000000000 -0400
+++ mozilla/nsprpub/pr/src/Makefile.in	2007-11-08 09:49:22.000000000 -0500
@@ -204,9 +204,9 @@
 endif
 endif
 
-ifeq ($(OS_TARGET),MacOSX)
-OS_LIBS		= -framework CoreServices -framework CoreFoundation
-endif
+#ifeq ($(OS_TARGET),MacOSX)
+#OS_LIBS		= -framework CoreServices -framework CoreFoundation
+#endif
 
 ifdef GC_LEAK_DETECTOR
 EXTRA_LIBS	= -L$(dist_libdir) -lboehm
diff -ruN mozilla-orig/nsprpub/pr/src/linking/prlink.c mozilla/nsprpub/pr/src/linking/prlink.c
--- mozilla-orig/nsprpub/pr/src/linking/prlink.c	2006-10-06 19:36:22.000000000 -0400
+++ mozilla/nsprpub/pr/src/linking/prlink.c	2007-11-08 09:49:22.000000000 -0500
@@ -158,7 +158,8 @@
 #if defined(SUNOS4) || defined(DARWIN) || defined(NEXTSTEP) \
     || defined(WIN16) || defined(XP_OS2) \
     || ((defined(OPENBSD) || defined(NETBSD)) && !defined(__ELF__))
-#define NEED_LEADING_UNDERSCORE
+/* On MacOSX 10.3, dlsym does not accept symbols with '_'. */
+/* #define NEED_LEADING_UNDERSCORE */
 #endif
 
 #define PR_LD_PATHW 0x8000  /* for PR_LibSpec_PathnameU */
diff -ruN mozilla-orig/xpcom/reflect/xptcall/src/md/unix/xptc_platforms_unixish_x86.h mozilla/xpcom/reflect/xptcall/src/md/unix/xptc_platforms_unixish_x86.h
--- mozilla-orig/xpcom/reflect/xptcall/src/md/unix/xptc_platforms_unixish_x86.h	2007-04-09 16:01:13.000000000 -0400
+++ mozilla/xpcom/reflect/xptcall/src/md/unix/xptc_platforms_unixish_x86.h	2007-11-08 09:49:22.000000000 -0500
@@ -167,7 +167,7 @@
 /* Define KEEP_STACK_16_BYTE_ALIGNED if the stack needs to maintain alignment
  * in a CALL for some good reason (like ABI compliance). */
 
-#ifdef XP_MACOSX
+#ifdef __APPLE__
 /* http://developer.apple.com/documentation/DeveloperTools/Conceptual/LowLevelABI/Articles/IA32.html */
 #define KEEP_STACK_16_BYTE_ALIGNED
 #endif
diff -ruN mozilla-orig/xpcom/reflect/xptcall/src/md/unix/xptcstubs_unixish_x86.cpp mozilla/xpcom/reflect/xptcall/src/md/unix/xptcstubs_unixish_x86.cpp
--- mozilla-orig/xpcom/reflect/xptcall/src/md/unix/xptcstubs_unixish_x86.cpp	2005-10-24 01:42:28.000000000 -0400
+++ mozilla/xpcom/reflect/xptcall/src/md/unix/xptcstubs_unixish_x86.cpp	2007-11-08 09:49:22.000000000 -0500
@@ -104,7 +104,7 @@
 
 #ifdef __GNUC__         /* Gnu Compiler. */
 
-#ifdef XP_MACOSX
+#ifdef __APPLE__
 /* Make sure the stack is 16-byte aligned.  Do that by aligning to 16 bytes and
  * then subtracting 4 so the three subsequent pushes result in a 16-byte aligned
  * stack. */
