diff -Nurd gnumeric-1.8.0/plugins/python-loader/gnm-python.c modgnumeric-1.8.0/plugins/python-loader/gnm-python.c
--- gnumeric-1.8.0/plugins/python-loader/gnm-python.c	2007-12-29 01:43:58.000000000 -0500
+++ modgnumeric-1.8.0/plugins/python-loader/gnm-python.c	2008-01-11 20:16:53.000000000 -0500
@@ -7,6 +7,7 @@
 
 #include <gnumeric-config.h>
 #include <Python.h>
+#define NO_IMPORT_PYGOBJECT
 #include <pygobject.h>
 #include <glib.h>
 #include "gnm-py-interpreter.h"
diff -Nurd gnumeric-1.8.0/plugins/python-loader/py-gnumeric.c modgnumeric-1.8.0/plugins/python-loader/py-gnumeric.c
--- gnumeric-1.8.0/plugins/python-loader/py-gnumeric.c	2007-12-29 01:43:54.000000000 -0500
+++ modgnumeric-1.8.0/plugins/python-loader/py-gnumeric.c	2008-01-11 20:19:23.000000000 -0500
@@ -9,6 +9,7 @@
 #include <Python.h>
 #include <gnumeric.h>
 #include <glib.h>
+#define NO_IMPORT_PYGOBJECT
 #include "pygobject.h"
 #include "application.h"
 #include "workbook.h"
diff -Nurd gnumeric-1.8.0/plugins/python-loader/python-loader.c modgnumeric-1.8.0/plugins/python-loader/python-loader.c
--- gnumeric-1.8.0/plugins/python-loader/python-loader.c	2007-12-29 01:43:59.000000000 -0500
+++ modgnumeric-1.8.0/plugins/python-loader/python-loader.c	2008-01-11 22:52:41.000000000 -0500
@@ -10,6 +10,7 @@
 #include "python-loader.h"
 
 #include <Python.h>
+#define NO_IMPORT_PYGOBJECT
 #include "pygobject.h"
 #include "py-gnumeric.h"
 #include "gnm-python.h"
