diff -Nurd -x'*~' kbib-0.6.6.orig/kbib/Makefile.in kbib-0.6.6/kbib/Makefile.in
--- kbib-0.6.6.orig/kbib/Makefile.in	2009-11-01 05:37:00.000000000 -0500
+++ kbib-0.6.6/kbib/Makefile.in	2012-02-14 01:28:33.000000000 -0500
@@ -362,7 +362,7 @@
 	mathscinetsearcher.cpp opticsinfobasesearcher.cpp sciencedirectsearcher.cpp kbibiface_stub.cpp kbibiface_skel.cpp
 
 
-kbib_LDADD = $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI) $(LIB_KFILE)  $(LIBSOCKET) $(LIBXSLT_LIBS) $(YAZ_LIBS) $(YAZPP_LIBS) $(POPPLER_LIBS)
+kbib_LDADD = $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI) $(LIB_KFILE)  $(LIBSOCKET) $(LIBXSLT_LIBS) $(YAZ_LIBS) $(YAZPP_LIBS) $(POPPLER_LIBS) -lDCOP
 KDE_CXXFLAGS = $(USE_EXCEPTIONS)
 
 EXTRA_DIST = main.cpp kbib.cpp kbib.h kbibdoc.cpp kbibdoc.h kbibui.rc kbib.desktop lo32-app-kbib.png lo16-app-kbib.png bibentry.cpp bibentry.h bibentrytable.cpp bibentrytable.h bibfile.cpp bibfile.h entrydef.cpp entrydef.h kbiblistview.cpp kbiblistview.h kbiblistviewitem.cpp kbiblistviewitem.h kbibeditentrydlg.cpp kbibeditentrydlg.h kbibiface.h hi16-app-kbib.png hi32-app-kbib.png kbibcheckkeydlg.cpp kbibcheckkeydlg.h kbibfinddlg.cpp kbibfinddlg.h kbibpreview.cpp kbibpreview.h kbibstart.cpp kbibstart.h kbibnewentrydefdlg.cpp kbibnewentrydefdlg.h kbibentrytypeparser.cpp  kbibentrytypeparser.h kbibselectentrytype.cpp kbibselectentrytype.h kbibinsertfiledlg.cpp kbibinsertfiledlg.h kbibconfigdlg.cpp kbibconfigdlg.h kbibsortrefdlg.cpp kbibsortrefdlg.h kbiblatexbib.h kbiblatexbib.cpp kbibcreatebiliographydlg.h kbibcreatebiliographydlg.cpp kbibsortdlg.h kbibsortdlg.cpp  kbibstringmacrodlg.cpp
