diff -Nurd kdeartwork-4.4.0/cmake/modules/FindXscreensaver.cmake kdeartwork-4.4.0-new/cmake/modules/FindXscreensaver.cmake
--- kdeartwork-4.4.0/cmake/modules/FindXscreensaver.cmake	2009-02-12 05:24:48.000000000 -0500
+++ kdeartwork-4.4.0-new/cmake/modules/FindXscreensaver.cmake	2010-02-09 08:44:14.000000000 -0500
@@ -8,6 +8,7 @@
 if (NOT XSCREENSAVER_FOUND)
   set(xscreensaver_alldirs)
   set(xscreensaver_checkdirs ${KDE4_INCLUDE_DIR}
+   	@FINKPREFIX@/
    	/usr/
    	/usr/local/
    	/opt/local/
diff -Nurd kdeartwork-4.4.0/kscreensaver/xsavers/CMakeLists.txt kdeartwork-4.4.0-new/kscreensaver/xsavers/CMakeLists.txt
--- kdeartwork-4.4.0/kscreensaver/xsavers/CMakeLists.txt	2008-09-13 04:16:47.000000000 -0400
+++ kdeartwork-4.4.0-new/kscreensaver/xsavers/CMakeLists.txt	2010-02-09 08:54:17.000000000 -0500
@@ -10,7 +10,7 @@
 
 kde4_add_executable(kswarm.kss ${kswarm.kss_SRCS})
 
-target_link_libraries(kswarm.kss  ${KDE4_KDEUI_LIBS} m )
+target_link_libraries(kswarm.kss  ${KDE4_KDEUI_LIBS} m ${X11_X11_LIB} )
 
 install(TARGETS kswarm.kss  ${INSTALL_TARGETS_DEFAULT_ARGS} )
 install( FILES KSwarm.desktop  DESTINATION  ${SERVICES_INSTALL_DIR}/ScreenSavers )
@@ -23,7 +23,7 @@
 
 kde4_add_executable(kspace.kss ${kspace_SRCS})
 
-target_link_libraries(kspace.kss  ${KDE4_KDEUI_LIBS} ${OPENGL_LIBRARY} )
+target_link_libraries(kspace.kss  ${KDE4_KDEUI_LIBS} ${OPENGL_LIBRARY} ${X11_X11_LIB} )
 
 install(TARGETS kspace.kss  ${INSTALL_TARGETS_DEFAULT_ARGS} )
 install(FILES KSpace.desktop DESTINATION ${SERVICES_INSTALL_DIR}/ScreenSavers )
