diff -Nurd -x'*~' antiprism-0.15.orig/src/minmax.cc antiprism-0.15/src/minmax.cc
--- antiprism-0.15.orig/src/minmax.cc	2007-07-12 04:24:56.000000000 -0400
+++ antiprism-0.15/src/minmax.cc	2007-08-27 21:50:36.000000000 -0400
@@ -53,7 +53,7 @@
       string ifile;
       string ofile;
 
-      mm_opts(): prog_opts("minmax"), num_iters(1000), algm('a'),
+      mm_opts(): prog_opts("minmax-antiprism"), num_iters(1000), algm('a'),
                  placement('n'), shorten_by(1.0), lengthen_by(0.0)
                  {}
 
