diff -aur cinepaint-0.18-1/configure cinepaint-0.18-1-new/configure
--- cinepaint-0.18-1/configure	Sun Aug  3 19:32:40 2003
+++ cinepaint-0.18-1-patched/configure	Sat Jan 17 18:03:02 2004
@@ -6023,10 +6023,10 @@
   darwin* | rhapsody*)
     case "$host_os" in
     rhapsody* | darwin1.[012])
-      allow_undefined_flag='-undefined suppress'
+      allow_undefined_flag=''
       ;;
     *) # Darwin 1.3 on
-      allow_undefined_flag='-flat_namespace -undefined suppress'
+      allow_undefined_flag=''
       ;;
     esac
     # FIXME: Relying on posixy $() will cause problems for
@@ -10378,10 +10378,10 @@
 
 case $host_os in
 darwin* | rhapsody*)
-CPPFLAGS="$CPPFLAGS -I/sw/include -no-cpp-precomp -D_REGEX_RE_COMP"
-LDFLAGS="$LDFLAGS -L/sw/lib -flat_namespace -undefined suppress"
-cinepaint_save_CPPFLAGS="$cinepaint_save_CPPFLAGS -I/sw/include -no-cpp-precomp -D_REGEX_RE_COMP"
-cinepaint_save_LDFLAGS="$cinepaint_save_LDFLAGS -L/sw/lib -flat_namespace -undefined suppress"
+CPPFLAGS="$CPPFLAGS -no-cpp-precomp -D_REGEX_RE_COMP"
+LDFLAGS="$LDFLAGS $GTK_LIBS"
+cinepaint_save_CPPFLAGS="$cinepaint_save_CPPFLAGS -no-cpp-precomp -D_REGEX_RE_COMP"
+cinepaint_save_LDFLAGS="$cinepaint_save_LDFLAGS $LDFLAGS $GTK_LIBS"
 ;;
 esac
 
