diff -ruN gimp-1.2.3/plug-ins/libgck/gck/Makefile.in gimp-1.2.3-patched/plug-ins/libgck/gck/Makefile.in
--- gimp-1.2.3/plug-ins/libgck/gck/Makefile.in	Mon Feb 11 16:46:23 2002
+++ gimp-1.2.3-patched/plug-ins/libgck/gck/Makefile.in	Sun Mar 10 10:22:57 2002
@@ -193,7 +193,7 @@
 gckinclude_HEADERS =  	gck.h
 
 
-libgck_la_LDFLAGS =  	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -release $(LT_RELEASE) $(no_undefined)
+libgck_la_LDFLAGS =  	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -release 1.2.5 $(no_undefined)
 
 
 INCLUDES =  	-I..				-I$(srcdir)/..			-I$(top_srcdir)			$(GTK_CFLAGS)			-I$(includedir)
