diff -rua devil-1.7.8.orig/docs/DevIL_manual.info devil-1.7.8/docs/DevIL_manual.info
--- devil-1.7.8.orig/docs/DevIL_manual.info	2009-03-08 16:43:50.000000000 -0400
+++ devil-1.7.8/docs/DevIL_manual.info	2009-06-24 10:53:28.000000000 -0400
@@ -1,7 +1,7 @@
 This is DevIL_manual.info, produced by makeinfo version 4.11 from
 DevIL_manual.texi.
 
-This is `DevIL 1.7.8' manual.  Last update is from 8 March 2009.
+This file documents `DevIL 1.7.8'.  Last update is from 8 March 2009.
 
    Copyright (C) 2008,2009 Denton Woods, Matěj Týč
 
diff -rua devil-1.7.8.orig/src-ILU/ilur/ilur.c devil-1.7.8/src-ILU/ilur/ilur.c
--- devil-1.7.8.orig/src-ILU/ilur/ilur.c	2009-03-08 03:10:12.000000000 -0400
+++ devil-1.7.8/src-ILU/ilur/ilur.c	2009-06-24 10:52:35.000000000 -0400
@@ -1,6 +1,6 @@
 #include <string.h>
 #include <stdio.h>
-#include <malloc.h>
+#include <stdlib.h>
 
 #include <IL/il.h>
 #include <IL/ilu.h>
--- devil-1.7.8/src-IL/src/il_icon.c.orig	2012-03-07 19:48:46.000000000 -0500
+++ devil-1.7.8/src-IL/src/il_icon.c	2012-03-07 19:49:28.000000000 -0500
@@ -525,7 +525,7 @@
 
 	// Expand low-bit-depth grayscale images to 8 bits
 	if (ico_color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8) {
-		png_set_gray_1_2_4_to_8(ico_png_ptr);
+		png_set_expand_gray_1_2_4_to_8(ico_png_ptr);
 	}
 
 	// Expand RGB images with transparency to full alpha channels
--- devil-1.7.8/src-IL/src/il_png.c.orig	2012-03-07 19:52:01.000000000 -0500
+++ devil-1.7.8/src-IL/src/il_png.c	2012-03-07 19:52:16.000000000 -0500
@@ -278,7 +278,7 @@
 
 	// Expand low-bit-depth grayscale images to 8 bits
 	if (png_color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8) {
-		png_set_gray_1_2_4_to_8(png_ptr);
+		png_set_expand_gray_1_2_4_to_8(png_ptr);
 	}
 
 	// Expand RGB images with transparency to full alpha channels
--- devil-1.7.8/src-IL/include/il_internal.h.orig	2012-03-08 17:31:13.000000000 -0500
+++ devil-1.7.8/src-IL/include/il_internal.h	2012-03-08 17:32:40.000000000 -0500
@@ -227,7 +227,9 @@
 ILboolean ilLoadBlp(ILconst_string FileName);
 ILboolean ilLoadBlpF(ILHANDLE File);
 ILboolean ilLoadBlpL(const void *Lump, ILuint Size);
+#ifndef __cplusplus
 ILboolean ilIsValidBmp(ILconst_string CONST_RESTRICT FileName);
+#endif
 ILboolean ilIsValidBmpF(ILHANDLE File);
 ILboolean ilIsValidBmpL(const void *Lump, ILuint Size);
 ILboolean ilLoadBmp(ILconst_string FileName);
--- devil-1.7.8/src-IL/include/il_endian.h.orig	2012-03-08 19:00:03.000000000 -0500
+++ devil-1.7.8/src-IL/include/il_endian.h	2012-03-08 18:55:42.000000000 -0500
@@ -56,6 +56,7 @@
 	#define BigDouble(d) iSwapDouble(d)
 #endif
 
+#if 0
 void   iSwapUShort(ILushort *s);
 void   iSwapShort(ILshort *s);
 void   iSwapUInt(ILuint *i);
@@ -86,6 +87,7 @@
 ILubyte SaveBigInt(ILint i);
 ILubyte SaveBigFloat(ILfloat f);
 ILubyte SaveBigDouble(ILdouble d);
+#endif 
 
 #ifdef IL_ENDIAN_C
 #undef NOINLINE
@@ -94,7 +96,7 @@
 #endif
 
 #ifndef NOINLINE
-INLINE void iSwapUShort(ILushort *s)  {
+static inline void iSwapUShort(ILushort *s)  {
 	#ifdef USE_WIN32_ASM
 		__asm {
 			mov ebx, s
@@ -113,11 +115,11 @@
 	#endif //USE_WIN32_ASM
 }
 
-INLINE void iSwapShort(ILshort *s) {
+static inline void iSwapShort(ILshort *s) {
 	iSwapUShort((ILushort*)s);
 }
 
-INLINE void iSwapUInt(ILuint *i) {
+static inline void iSwapUInt(ILuint *i) {
 	#ifdef USE_WIN32_ASM
 		__asm {
 			mov ebx, i
@@ -135,15 +137,15 @@
 	#endif //USE_WIN32_ASM
 }
 
-INLINE void iSwapInt(ILint *i) {
+static inline void iSwapInt(ILint *i) {
 	iSwapUInt((ILuint*)i);
 }
 
-INLINE void iSwapFloat(ILfloat *f) {
+static inline void iSwapFloat(ILfloat *f) {
 	iSwapUInt((ILuint*)f);
 }
 
-INLINE void iSwapDouble(ILdouble *d) {
+static inline void iSwapDouble(ILdouble *d) {
 	#ifdef GCC_X86_ASM
 	int *t = (int*)d;
 	asm("bswap %2    \n"
@@ -164,7 +166,7 @@
 }
 
 
-INLINE ILushort GetLittleUShort() {
+static inline ILushort GetLittleUShort() {
 	ILushort s;
 	iread(&s, sizeof(ILushort), 1);
 #ifdef __BIG_ENDIAN__
@@ -173,7 +175,7 @@
 	return s;
 }
 
-INLINE ILshort GetLittleShort() {
+static inline ILshort GetLittleShort() {
 	ILshort s;
 	iread(&s, sizeof(ILshort), 1);
 #ifdef __BIG_ENDIAN__
@@ -182,7 +184,7 @@
 	return s;
 }
 
-INLINE ILuint GetLittleUInt() {
+static inline ILuint GetLittleUInt() {
 	ILuint i;
 	iread(&i, sizeof(ILuint), 1);
 #ifdef __BIG_ENDIAN__
@@ -191,7 +193,7 @@
 	return i;
 }
 
-INLINE ILint GetLittleInt() {
+static inline ILint GetLittleInt() {
 	ILint i;
 	iread(&i, sizeof(ILint), 1);
 #ifdef __BIG_ENDIAN__
@@ -200,7 +202,7 @@
 	return i;
 }
 
-INLINE ILfloat GetLittleFloat() {
+static inline ILfloat GetLittleFloat() {
 	ILfloat f;
 	iread(&f, sizeof(ILfloat), 1);
 #ifdef __BIG_ENDIAN__
@@ -209,7 +211,7 @@
 	return f;
 }
 
-INLINE ILdouble GetLittleDouble() {
+static inline ILdouble GetLittleDouble() {
 	ILdouble d;
 	iread(&d, sizeof(ILdouble), 1);
 #ifdef __BIG_ENDIAN__
@@ -219,7 +221,7 @@
 }
 
 
-INLINE ILushort GetBigUShort() {
+static inline ILushort GetBigUShort() {
 	ILushort s;
 	iread(&s, sizeof(ILushort), 1);
 #ifdef __LITTLE_ENDIAN__
@@ -229,7 +231,7 @@
 }
 
 
-INLINE ILshort GetBigShort() {
+static inline ILshort GetBigShort() {
 	ILshort s;
 	iread(&s, sizeof(ILshort), 1);
 #ifdef __LITTLE_ENDIAN__
@@ -239,7 +241,7 @@
 }
 
 
-INLINE ILuint GetBigUInt() {
+static inline ILuint GetBigUInt() {
 	ILuint i;
 	iread(&i, sizeof(ILuint), 1);
 #ifdef __LITTLE_ENDIAN__
@@ -249,7 +251,7 @@
 }
 
 
-INLINE ILint GetBigInt() {
+static inline ILint GetBigInt() {
 	ILint i;
 	iread(&i, sizeof(ILint), 1);
 #ifdef __LITTLE_ENDIAN__
@@ -259,7 +261,7 @@
 }
 
 
-INLINE ILfloat GetBigFloat() {
+static inline ILfloat GetBigFloat() {
 	ILfloat f;
 	iread(&f, sizeof(ILfloat), 1);
 #ifdef __LITTLE_ENDIAN__
@@ -269,7 +271,7 @@
 }
 
 
-INLINE ILdouble GetBigDouble() {
+static inline ILdouble GetBigDouble() {
 	ILdouble d;
 	iread(&d, sizeof(ILdouble), 1);
 #ifdef __LITTLE_ENDIAN__
@@ -278,14 +280,14 @@
 	return d;
 }
 
-INLINE ILubyte SaveLittleUShort(ILushort s) {
+static inline ILubyte SaveLittleUShort(ILushort s) {
 #ifdef __BIG_ENDIAN__
 	iSwapUShort(&s);
 #endif
 	return iwrite(&s, sizeof(ILushort), 1);
 }
 
-INLINE ILubyte SaveLittleShort(ILshort s) {
+static inline ILubyte SaveLittleShort(ILshort s) {
 #ifdef __BIG_ENDIAN__
 	iSwapShort(&s);
 #endif
@@ -293,7 +295,7 @@
 }
 
 
-INLINE ILubyte SaveLittleUInt(ILuint i) {
+static inline ILubyte SaveLittleUInt(ILuint i) {
 #ifdef __BIG_ENDIAN__
 	iSwapUInt(&i);
 #endif
@@ -301,14 +303,14 @@
 }
 
 
-INLINE ILubyte SaveLittleInt(ILint i) {
+static inline ILubyte SaveLittleInt(ILint i) {
 #ifdef __BIG_ENDIAN__
 	iSwapInt(&i);
 #endif
 	return iwrite(&i, sizeof(ILint), 1);
 }
 
-INLINE ILubyte SaveLittleFloat(ILfloat f) {
+static inline ILubyte SaveLittleFloat(ILfloat f) {
 #ifdef __BIG_ENDIAN__
 	iSwapFloat(&f);
 #endif
@@ -316,7 +318,7 @@
 }
 
 
-INLINE ILubyte SaveLittleDouble(ILdouble d) {
+static inline ILubyte SaveLittleDouble(ILdouble d) {
 #ifdef __BIG_ENDIAN__
 	iSwapDouble(&d);
 #endif
@@ -324,7 +326,7 @@
 }
 
 
-INLINE ILubyte SaveBigUShort(ILushort s) {
+static inline ILubyte SaveBigUShort(ILushort s) {
 #ifdef __LITTLE_ENDIAN__
 	iSwapUShort(&s);
 #endif
@@ -332,7 +334,7 @@
 }
 
 
-INLINE ILubyte SaveBigShort(ILshort s) {
+static inline ILubyte SaveBigShort(ILshort s) {
 #ifdef __LITTLE_ENDIAN__
 	iSwapShort(&s);
 #endif
@@ -340,7 +342,7 @@
 }
 
 
-INLINE ILubyte SaveBigUInt(ILuint i) {
+static inline ILubyte SaveBigUInt(ILuint i) {
 #ifdef __LITTLE_ENDIAN__
 	iSwapUInt(&i);
 #endif
@@ -348,7 +350,7 @@
 }
 
 
-INLINE ILubyte SaveBigInt(ILint i) {
+static inline ILubyte SaveBigInt(ILint i) {
 #ifdef __LITTLE_ENDIAN__
 	iSwapInt(&i);
 #endif
@@ -356,7 +358,7 @@
 }
 
 
-INLINE ILubyte SaveBigFloat(ILfloat f) {
+static inline ILubyte SaveBigFloat(ILfloat f) {
 #ifdef __LITTLE_ENDIAN__
 	iSwapFloat(&f);
 #endif
@@ -364,7 +366,7 @@
 }
 
 
-INLINE ILubyte SaveBigDouble(ILdouble d) {
+static inline ILubyte SaveBigDouble(ILdouble d) {
 #ifdef __LITTLE_ENDIAN__
 	iSwapDouble(&d);
 #endif
--- devil-1.7.8/src-IL/include/il_manip.h.orig	2012-03-08 19:03:57.000000000 -0500
+++ devil-1.7.8/src-IL/include/il_manip.h	2012-03-08 18:55:42.000000000 -0500
@@ -31,7 +31,7 @@
 	#pragma warning(push)
 	#pragma warning(disable : 4756)  // Disables 'named type definition in parentheses' warning
 #endif
-INLINE ILfloat /*ILAPIENTRY*/ ilFloatToHalfOverflow() {
+static inline ILfloat /*ILAPIENTRY*/ ilFloatToHalfOverflow() {
 	ILfloat f = 1e10;
 	ILint j;
 	for (j = 0; j < 10; j++)
@@ -47,7 +47,7 @@
 // Float-to-half conversion -- general case, including
 // zeroes, denormalized numbers and exponent overflows.
 //-----------------------------------------------------
-INLINE ILushort ILAPIENTRY ilFloatToHalf(ILuint i) {
+static inline ILushort ILAPIENTRY ilFloatToHalf(ILuint i) {
 	//
 	// Our floating point number, f, is represented by the bit
 	// pattern in integer i.  Disassemble that bit pattern into
@@ -175,7 +175,7 @@
 }
 
 // Taken from OpenEXR
-INLINE ILuint ILAPIENTRY ilHalfToFloat (ILushort y) {
+static inline ILuint ILAPIENTRY ilHalfToFloat (ILushort y) {
 
 	int s = (y >> 15) & 0x00000001;
 	int e = (y >> 10) & 0x0000001f;
--- devil-1.7.8/src-ILU/include/ilu_internal.h.orig	2012-03-08 18:49:44.000000000 -0500
+++ devil-1.7.8/src-ILU/include/ilu_internal.h	2012-03-08 18:51:25.000000000 -0500
@@ -71,22 +71,24 @@
 #define INLINE
 #endif
 
+#if 0
 // Internal functions
 ILfloat ilCos(ILfloat Angle);
 ILfloat ilSin(ILfloat Angle);
 ILint ilRound(ILfloat Num);
+#endif
 
 #ifndef NOINLINE
-INLINE ILfloat ilCos(ILfloat Angle) {
+static inline ILfloat ilCos(ILfloat Angle) {
 	return (ILfloat)(cos(Angle * IL_DEGCONV));
 }
 
-INLINE ILfloat ilSin(ILfloat Angle) {
+static inline ILfloat ilSin(ILfloat Angle) {
 	return (ILfloat)(sin(Angle * IL_DEGCONV));
 }
 
 
-INLINE ILint ilRound(ILfloat Num) {
+static inline ILint ilRound(ILfloat Num) {
 	return (ILint)(Num + 0.5); // this is truncating in away-from-0, not rounding
 }
 #endif
--- devil-1.7.8/configure.orig	2012-03-08 20:03:09.000000000 -0500
+++ devil-1.7.8/configure	2012-03-08 20:03:43.000000000 -0500
@@ -16134,7 +16134,7 @@
 	                     { echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
 	                     CFLAGS="$TMP_CFLAGS"
-                             GENERAL_CFLAGS="$GENERAL_CFLAGS -std=gnu99 -fgnu89-inline"
+                             GENERAL_CFLAGS="$GENERAL_CFLAGS -fgnu89-inline"
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
