--- libpng-1.0.60/scripts/makefile.darwin.orig	2013-06-02 15:15:50.000000000 -0400
+++ libpng-1.0.60/scripts/makefile.darwin	2013-06-02 15:17:04.000000000 -0400
@@ -10,7 +10,7 @@
 # and license in png.h
 
 # where "make install" puts libpng.a, libpng10.dylib, png.h and pngconf.h
-prefix=/usr/local
+prefix=@FINKPREFIX@
 exec_prefix=$(prefix)
 
 # Where the zlib library and include files are located
@@ -58,7 +58,7 @@
 # If you're going to install into a temporary location
 # via DESTDIR, $(DESTDIR)$(prefix) must already exist before
 # you execute make install.
-DESTDIR=
+DESTDIR=${DESTDIR}
 
 DB=$(DESTDIR)$(BINPATH)
 DI=$(DESTDIR)$(INCPATH)
@@ -101,21 +101,18 @@
 $(LIBSO): $(LIBSOMAJ)
 	$(LN_SF) $(LIBSOMAJ) $(LIBSO)
 
-$(LIBSOMAJ): $(LIBSOVER)
-	$(LN_SF) $(LIBSOVER) $(LIBSOMAJ)
-
-$(LIBSOVER): $(OBJSDLL)
+$(LIBSOMAJ): $(OBJSDLL)
 	$(CC) -dynamiclib \
 	 -install_name $(LIBPATH)/$(LIBSOMAJ) \
 	 -current_version 0 -compatibility_version 0 \
-	 -o $(LIBSOVER) \
+	 -o $(LIBSOMAJ) \
 	 $(OBJSDLL) -L$(ZLIBLIB) -lz
 
 $(OLDSOVER): $(OBJSDLL)
 	$(CC) -dynamiclib \
 	 -install_name $(LIBPATH)/$(OLDSOMAJ) \
 	 -current_version 2 -compatibility_version 2 \
-	 -o $(OLDSOVER) \
+	 -o $(OLDSOMAJ) \
 	 $(OBJSDLL) -L$(ZLIBLIB) -lz
 
 pngtest: pngtest.o $(LIBSO)
@@ -150,21 +147,10 @@
 	-@$(RM_F) $(DL)/$(OLDSO)
 	-@$(RM_F) $(DL)/$(OLDSOMAJ)
 	-@$(RM_F) $(DL)/libpng.3.$(PNGMIN)*.dylib
-	cp $(LIBSOVER) $(DL)
-	cp $(OLDSOVER) $(DL)
-	chmod 755 $(DL)/$(LIBSOVER)
-	chmod 755 $(DL)/$(OLDSOVER)
-	(cd $(DL); \
-	$(LN_SF) $(OLDSOVER) $(OLDSOMAJ); \
-	$(LN_SF) $(OLDSOMAJ) $(OLDSO); \
-	$(LN_SF) $(LIBSOVER) $(LIBSOMAJ); \
-	$(LN_SF) $(LIBSOMAJ) $(LIBSO))
-	-@if [ ! -d $(DL)/pkgconfig ]; then $(MKDIR_P) $(DL)/pkgconfig; fi
-	-@$(RM_F) $(DL)/pkgconfig/$(LIBNAME).pc
-	-@$(RM_F) $(DL)/pkgconfig/libpng.pc
-	cp libpng.pc $(DL)/pkgconfig/$(LIBNAME).pc
-	chmod 644 $(DL)/pkgconfig/$(LIBNAME).pc
-	(cd $(DL)/pkgconfig; $(LN_SF) $(LIBNAME).pc libpng.pc)
+	cp $(LIBSOMAJ) $(DL)
+	cp $(OLDSOMAJ) $(DL)
+	chmod 755 $(DL)/$(LIBSOMAJ)
+	chmod 755 $(DL)/$(OLDSOMAJ)
 
 install-man: libpng.3 libpngpf.3 png.5
 	-@if [ ! -d $(DM) ]; then $(MKDIR_P) $(DM); fi
