diff -urN Mesa-7.0.2.orig/Makefile Mesa-7.0.2/Makefile
--- Mesa-7.0.2.orig/Makefile	2007-11-10 16:41:24.000000000 +0000
+++ Mesa-7.0.2/Makefile	2008-03-02 00:49:08.000000000 +0000
@@ -27,7 +27,7 @@
 realclean:
 	touch $(TOP)/configs/current
 	$(MAKE) clean
-	-rm -rf lib*
+#	-rm -rf lib*
 	-rm -f $(TOP)/configs/current
 	-rm -f `find . -name \*.o`
 	-rm -f `find . -name \*.a`
diff -urN Mesa-7.0.2.orig/bin/mklib Mesa-7.0.2/bin/mklib
--- Mesa-7.0.2.orig/bin/mklib	2007-11-01 15:30:52.000000000 +0000
+++ Mesa-7.0.2/bin/mklib	2008-03-02 00:55:20.000000000 +0000
@@ -575,6 +575,7 @@
             echo "mklib: Making Darwin static library: " ${LIBNAME}
             LINK="ar"
             OPTS="-ruvs"
+	    echo "${LINK} ${OPTS} ${LIBNAME} ${OBJECTS}"
             ${LINK} ${OPTS} ${LIBNAME} ${OBJECTS}
             FINAL_LIBS=${LIBNAME}
         else
@@ -584,10 +585,11 @@
                 OPTS="${ARCHOPT} -bundle -multiply_defined suppress"
             else
 		LIBSUFFIX="dylib"
-                OPTS="${ARCHOPT} -dynamiclib -multiply_defined suppress -current_version ${MAJOR}.${MINOR}.0 -compatibility_version ${MAJOR}.${MINOR}.0 -install_name lib${LIBNAME}.${MAJOR}.${LIBSUFFIX}"
+                OPTS="${ARCHOPT} -dynamiclib -multiply_defined suppress -headerpad_max_install_names -current_version ${MAJOR}.${MINOR}.0 -compatibility_version ${MAJOR}.${MINOR}.0 -install_name @PREFIX@/lib/mesa/lib${LIBNAME}.${MAJOR}.${LIBSUFFIX}"
             fi
-            LINKNAME="lib${LIBNAME}.${LIBSUFFIX}"
-            LIBNAME="lib${LIBNAME}.${MAJOR}.${LIBSUFFIX}"
+            LINKNAME="lib${LIBNAME}.${MAJOR}.${LIBSUFFIX}"
+	    LINKNAME2="lib${LIBNAME}.${LIBSUFFIX}"
+            LIBNAME="lib${LIBNAME}.${MAJOR}.${MINOR}.${LIBSUFFIX}"
 
 	    # examine first object to determine ABI
     	    set ${OBJECTS}
@@ -611,9 +613,11 @@
 	    fi
 
             echo "mklib: Making Darwin shared library: " ${LIBNAME}
-            ${LINK} ${OPTS} ${LDFLAGS} -o ${LIBNAME} ${OBJECTS} ${DEPS}
+            echo "${LINK} ${OPTS} ${LDFLAGS} -o ${LIBNAME} ${OBJECTS} ${DEPS}"
+	    ${LINK} ${OPTS} ${LDFLAGS} -o ${LIBNAME} ${OBJECTS} ${DEPS}
             ln -s ${LIBNAME} ${LINKNAME}
-            FINAL_LIBS="${LIBNAME} ${LINKNAME}"
+            ln -s ${LIBNAME} ${LINKNAME2}
+            FINAL_LIBS="${LIBNAME} ${LINKNAME} ${LINKNAME2}"
         fi
         ;;
 
