diff -ru nvidia-texture-tools.orig/src/nvcore/CMakeLists.txt nvidia-texture-tools/src/nvcore/CMakeLists.txt
--- nvidia-texture-tools.orig/src/nvcore/CMakeLists.txt	2009-11-06 12:43:09.000000000 +1100
+++ nvidia-texture-tools/src/nvcore/CMakeLists.txt	2010-01-27 23:16:32.000000000 +1100
@@ -34,6 +34,7 @@
 
 IF(NVCORE_SHARED)
 	ADD_LIBRARY(nvcore SHARED ${CORE_SRCS})
+	SET_TARGET_PROPERTIES(nvcore PROPERTIES SOVERSION NVTT_VERSION)
 ELSE(NVCORE_SHARED)
 	ADD_LIBRARY(nvcore ${CORE_SRCS})
 ENDIF(NVCORE_SHARED)
diff -ru nvidia-texture-tools.orig/src/nvimage/CMakeLists.txt nvidia-texture-tools/src/nvimage/CMakeLists.txt
--- nvidia-texture-tools.orig/src/nvimage/CMakeLists.txt	2009-11-06 12:43:03.000000000 +1100
+++ nvidia-texture-tools/src/nvimage/CMakeLists.txt	2010-01-27 23:26:09.000000000 +1100
@@ -54,6 +54,7 @@
 
 IF(NVIMAGE_SHARED)	
 	ADD_LIBRARY(nvimage SHARED ${IMAGE_SRCS})
+	SET_TARGET_PROPERTIES(nvimage PROPERTIES SOVERSION NVTT_VERSION)
 ELSE(NVIMAGE_SHARED)
 	ADD_LIBRARY(nvimage ${IMAGE_SRCS})
 ENDIF(NVIMAGE_SHARED)
diff -ru nvidia-texture-tools.orig/src/nvmath/CMakeLists.txt nvidia-texture-tools/src/nvmath/CMakeLists.txt
--- nvidia-texture-tools.orig/src/nvmath/CMakeLists.txt	2009-11-06 12:43:10.000000000 +1100
+++ nvidia-texture-tools/src/nvmath/CMakeLists.txt	2010-01-27 23:25:57.000000000 +1100
@@ -20,6 +20,7 @@
 
 IF(NVMATH_SHARED)
 	ADD_LIBRARY(nvmath SHARED ${MATH_SRCS})
+	SET_TARGET_PROPERTIES(nvmath PROPERTIES SOVERSION NVTT_VERSION)
 ELSE(NVMATH_SHARED)
 	ADD_LIBRARY(nvmath ${MATH_SRCS})
 ENDIF(NVMATH_SHARED)
diff -ru nvidia-texture-tools.orig/src/nvtt/CMakeLists.txt nvidia-texture-tools/src/nvtt/CMakeLists.txt
--- nvidia-texture-tools.orig/src/nvtt/CMakeLists.txt	2009-11-06 12:43:08.000000000 +1100
+++ nvidia-texture-tools/src/nvtt/CMakeLists.txt	2010-01-27 23:25:43.000000000 +1100
@@ -46,6 +46,7 @@
 IF(NVTT_SHARED)
 	ADD_DEFINITIONS(-DNVTT_SHARED=1)
 	ADD_LIBRARY(nvtt SHARED ${NVTT_SRCS})
+	SET_TARGET_PROPERTIES(nvtt PROPERTIES SOVERSION NVTT_VERSION)
 ELSE(NVTT_SHARED)
 	ADD_LIBRARY(nvtt ${NVTT_SRCS})
 ENDIF(NVTT_SHARED)
