--- OpenCV-2.3.1/modules/flann/include/opencv2/flann/any.h.orig	2012-03-06 20:55:40.000000000 -0500
+++ OpenCV-2.3.1/modules/flann/include/opencv2/flann/any.h	2012-03-06 20:57:58.000000000 -0500
@@ -30,6 +30,12 @@
 {
 };
 
+inline std::ostream& operator <<(std::ostream& out, const empty_any&)
+{
+    out<<"[empty_any]";
+    return out;
+}
+
 struct base_any_policy
 {
     virtual void static_delete(void** x) = 0;
--- OpenCV-2.3.1/modules/flann/include/opencv2/flann/lsh_index.h.orig	2012-03-06 20:55:54.000000000 -0500
+++ OpenCV-2.3.1/modules/flann/include/opencv2/flann/lsh_index.h	2012-03-06 21:00:27.000000000 -0500
@@ -56,7 +56,7 @@
 
 struct LshIndexParams : public IndexParams
 {
-    LshIndexParams(unsigned int table_number, unsigned int key_size, unsigned int multi_probe_level)
+    LshIndexParams(unsigned int table_number = 12, unsigned int key_size = 20, unsigned int multi_probe_level = 2)
     {
         (* this)["algorithm"] = FLANN_INDEX_LSH;
         // The number of hash tables to use
