diff -Nurd -x'*~' tesseract-3.01.orig/api/Makefile.am tesseract-3.01/api/Makefile.am
--- tesseract-3.01.orig/api/Makefile.am	2011-10-21 22:40:50.000000000 -0400
+++ tesseract-3.01/api/Makefile.am	2013-03-23 20:46:37.000000000 -0400
@@ -52,7 +52,6 @@
     $(top_srcdir)/ccutil/libtesseract_ccutil.la
 
 libtesseract_la_LDFLAGS += -version-info $(GENERIC_LIBRARY_VERSION)
-libtesseract_la_LDFLAGS += -L./ -L../ -L../api -L../ccutil -L../viewer -L../cutil -L../image -L../ccstruct -L../dict -L../classify -L../wordrec -L../neural_networks/runtime -L../textord -L../cube -L../ccmain
 
 bin_PROGRAMS = tesseract
 tesseract_SOURCES = $(top_srcdir)/api/tesseractmain.cpp
diff -Nurd -x'*~' tesseract-3.01.orig/configure.ac tesseract-3.01/configure.ac
--- tesseract-3.01.orig/configure.ac	2011-10-21 22:40:55.000000000 -0400
+++ tesseract-3.01/configure.ac	2013-03-22 12:53:40.000000000 -0400
@@ -120,17 +120,6 @@
 fi
 
 # Special cases
-case "$host" in
-  *-darwin* | *-macos10*)
-     if test -d /opt/local ; then
-       CPPFLAGS="$CPPFLAGS -I/opt/local/include"
-       LDFLAGS="$LDFLAGS -L/opt/local/lib"
-     elif test -d /sw ; then
-       CPPFLAGS="$CPPFLAGS -I/sw/include"
-       LDFLAGS="$LDFLAGS -L/sw/lib"
-     fi
-  ;;
-esac
 
 # ----------------------------------------
 # Check Compiler Characteristics and
