diff -cr xfig.3.2.3d/Imakefile xfig.3.2.3d.patched/Imakefile
*** xfig.3.2.3d/Imakefile	Tue May 29 12:32:11 2001
--- xfig.3.2.3d.patched/Imakefile	Mon Jul 16 07:11:26 2001
***************
*** 26,34 ****
  XCOMM app-defaults resource files to go.  You will have to use the environment
  XCOMM variable XAPPLRESDIR to point to this directory.
  
! XCOMM BINDIR = /usr/bin/X11
! XCOMM MKDIRHIER = /bin/sh /usr/bin/X11/mkdirhier
! XCOMM XAPPLOADDIR = /home/user/xfig
  
  XCOMM If don't want JPEG support, comment out the #define USEJPEG line
  XCOMM Uncomment the #define for USEJPEG if you want to be able to import
--- 26,34 ----
  XCOMM app-defaults resource files to go.  You will have to use the environment
  XCOMM variable XAPPLRESDIR to point to this directory.
  
! BINDIR = $(PREFIX)/bin
! MKDIRHIER = /bin/sh @FINKPREFIX@/lib/xmkmf/bin/mkdirhier
! XAPPLOADDIR = $(PREFIX)/etc/app-defaults
  
  XCOMM If don't want JPEG support, comment out the #define USEJPEG line
  XCOMM Uncomment the #define for USEJPEG if you want to be able to import
***************
*** 47,54 ****
  
  #ifdef USEJPEG
  #ifdef USEINSTALLEDJPEG
! JPEGLIBDIR = /usr/local/lib
! JPEGINCDIR = /usr/include/X11
  #else
  JPEGLIBDIR = ../jpeg
  JPEGINCDIR = $(JPEGLIBDIR)
--- 47,54 ----
  
  #ifdef USEJPEG
  #ifdef USEINSTALLEDJPEG
! JPEGLIBDIR = $(PREFIX)/lib
! JPEGINCDIR = $(PREFIX)/include
  #else
  JPEGLIBDIR = ../jpeg
  JPEGINCDIR = $(JPEGLIBDIR)
***************
*** 72,83 ****
  XCOMM Change XPMLIBDIR if necessary to point to the xpm library (libXpm)
  XCOMM Change XPMINC if necessary to point to the include file for xpm (xpm.h)
  
! XCOMM #define USEXPM
! XCOMM #define USEXPM_ICON
  
  #ifdef USEXPM
! XPMLIBDIR = $(USRLIBDIR)
! XPMINC = -I/usr/include/X11
  #endif
  
  XCOMM Uncomment the following definiton if you want to use the small icons
--- 72,83 ----
  XCOMM Change XPMLIBDIR if necessary to point to the xpm library (libXpm)
  XCOMM Change XPMINC if necessary to point to the include file for xpm (xpm.h)
  
! #define USEXPM
! #define USEXPM_ICON
  
  #ifdef USEXPM
! XPMLIBDIR = /usr/X11R6/lib
! XPMINC = -I/usr/X11R6/include/X11
  #endif
  
  XCOMM Uncomment the following definiton if you want to use the small icons
***************
*** 93,99 ****
  #define XAW3D
  
  #ifdef XAW3D
! XAW3DINC = -I/usr/include/X11/Xaw3d
  DUSEXAW3D = -DXAW3D
  XAWLIB = -lXaw3d
  #endif
--- 93,99 ----
  #define XAW3D
  
  #ifdef XAW3D
! XAW3DINC = -I/usr/X11R6/include/X11/Xaw3d
  DUSEXAW3D = -DXAW3D
  XAWLIB = -lXaw3d
  #endif
***************
*** 136,149 ****
  XCOMM use (and change) the following if you want the multi-key data base file
  XCOMM somewhere other than the standard X11 library directory
  XCOMM be sure to comment out the second copy of XFIGLIBDIR if you use this one
! XCOMM XFIGLIBDIR = /usr/local/lib/xfig
  
  XCOMM use this if you want the multi-key data base file in the standard X11 tree
! XFIGLIBDIR = $(LIBDIR)/xfig
  
  XCOMM now you can configure where the documentation should be located
! XFIGDOCDIR = $(LIBDIR)/xfig
! XCOMM XFIGDOCDIR = /usr/share/doc/xfig-doc
  
  XCOMM If your system doesn't have strstr undefine the following definition
  XCOMM HAVE_NO_NOSTRSTR = -DNOSTRSTR
--- 136,149 ----
  XCOMM use (and change) the following if you want the multi-key data base file
  XCOMM somewhere other than the standard X11 library directory
  XCOMM be sure to comment out the second copy of XFIGLIBDIR if you use this one
! XFIGLIBDIR = $(PREFIX)/share/xfig
  
  XCOMM use this if you want the multi-key data base file in the standard X11 tree
! XCOMM XFIGLIBDIR = $(LIBDIR)/xfig
  
  XCOMM now you can configure where the documentation should be located
! XCOMM XFIGDOCDIR = $(LIBDIR)/xfig
! XFIGDOCDIR = $(PREFIX)/share/doc/xfig
  
  XCOMM If your system doesn't have strstr undefine the following definition
  XCOMM HAVE_NO_NOSTRSTR = -DNOSTRSTR
diff -ur xfig.3.2.3d.bak/u_fonts.h xfig.3.2.3d/u_fonts.h
--- xfig.3.2.3d.bak/u_fonts.h	2000-01-03 10:09:39.000000000 -0800
+++ xfig.3.2.3d/u_fonts.h	2005-06-13 17:15:38.000000000 -0700
@@ -30,12 +30,6 @@
 /* font number for the "nil" font (when user wants tiny text) */
 #define NILL_FONT NUM_FONTS
 
-extern int		psfontnum();
-extern int		latexfontnum();
-extern struct _xfstruct	x_fontinfo[], x_backup_fontinfo[];
-extern struct _fstruct	ps_fontinfo[];
-extern struct _fstruct	latex_fontinfo[];
-
 /* element of linked list for each font
    The head of list is for the different font NAMES,
    and the elements of this list are for each different
@@ -61,5 +55,13 @@
 				 * sizes */
 };
 
+
+extern int		psfontnum();
+extern int		latexfontnum();
+extern struct _xfstruct	x_fontinfo[], x_backup_fontinfo[];
+extern struct _fstruct	ps_fontinfo[];
+extern struct _fstruct	latex_fontinfo[];
+
+
 int		x_fontnum();
 #endif /* U_FONTS_H */
