--- denemo-0.8.0/Makefile.in	2008-10-20 16:13:36.000000000 +0200
+++ denemo-0.8.0_corr/Makefile.in	2009-01-22 17:10:59.000000000 +0100
@@ -648,8 +648,8 @@
 
 
 install-data-hook:
-	cp -r actions  $(datadir)/${PACKAGE}/
-	chmod -R 755 $(datadir)/${PACKAGE}/actions
+	cp -r actions  $(DESTDIR)$(datadir)/${PACKAGE}/
+	chmod -R 755 $(DESTDIR)$(datadir)/${PACKAGE}/actions
 dist-hook: @PACKAGE@.spec
 	cp $(srcdir)/@PACKAGE@.spec $(distdir)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
--- denemo-0.8.0/src/file.c	2008-10-20 16:11:04.000000000 +0200
+++ denemo-0.8.0_corr/src/file.c	2009-01-22 17:10:59.000000000 +0100
@@ -33,7 +33,9 @@
 #include <string.h>
 #include <sys/stat.h>		/* check existance and type of files */
 #include <dirent.h>		/* filter and sort filenames */
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>		/* I use free() */
+#endif
 
 
 
--- denemo-0.8.0/src/lyparser.y	2008-09-04 18:33:02.000000000 +0200
+++ denemo-0.8.0_corr/src/lyparser.y	2009-01-22 17:10:59.000000000 +0100
@@ -169,7 +169,9 @@
 
 // needed for bison.simple's malloc () and free ()
 
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>
+#endif
 #include <stdlib.h>
 #include <stdio.h>
 
--- denemo-0.8.0/src/main.c	2008-10-20 16:11:04.000000000 +0200
+++ denemo-0.8.0_corr/src/main.c	2009-01-22 17:10:59.000000000 +0100
@@ -26,6 +26,9 @@
 #include <wait.h>
 #endif
 #ifdef HAVE_SYS_RESOURCE_H
+#ifdef HAVE_SYS_TIME_H
+#include <sys/time.h>
+#endif
 #include <sys/resource.h>
 #endif
 
