diff -u libao-0.8.2/doc/Makefile.am libao-0.8.2-patched/doc/Makefile.am
--- libao-0.8.2/doc/Makefile.am	Tue Nov 20 20:53:11 2001
+++ libao-0.8.2-patched/doc/Makefile.am	Sat Mar 30 23:55:16 2002
@@ -2,7 +2,7 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 
 # We list all of these as opposed to using a wildcard so that
 # building outside the source directory works.
diff -u libao-0.8.2/doc/Makefile.in libao-0.8.2-patched/doc/Makefile.in
--- libao-0.8.2/doc/Makefile.in	Mon Dec 31 17:43:57 2001
+++ libao-0.8.2-patched/doc/Makefile.in	Sat Mar 30 23:54:42 2002
@@ -93,7 +93,7 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 
 # We list all of these as opposed to using a wildcard so that
 # building outside the source directory works.

--- libao-0.8.3/src/plugins/macosx/Makefile.in.orig	2002-11-18 22:28:16.000000000 -0800
+++ libao-0.8.3/src/plugins/macosx/Makefile.in	2002-11-18 22:28:26.000000000 -0800
@@ -94,7 +94,7 @@
 
 @HAVE_MACOSX_TRUE@macosxltlibs = libmacosx.la
 @HAVE_MACOSX_FALSE@macosxltlibs = 
-@HAVE_MACOSX_TRUE@macosxldflags = -export-dynamic -avoid-version
+@HAVE_MACOSX_TRUE@macosxldflags = -module -avoid-version
 @HAVE_MACOSX_FALSE@macosxldflags = 
 @HAVE_MACOSX_TRUE@macosxsources = ao_macosx.c
 @HAVE_MACOSX_FALSE@macosxsources = 
--- libao-0.8.3/src/plugins/esd/Makefile.in.orig	2002-11-18 22:27:47.000000000 -0800
+++ libao-0.8.3/src/plugins/esd/Makefile.in	2002-11-18 22:28:01.000000000 -0800
@@ -94,7 +94,7 @@
 
 @HAVE_ESD_TRUE@esdltlibs = libesd.la
 @HAVE_ESD_FALSE@esdltlibs = 
-@HAVE_ESD_TRUE@esdldflags = -export-dynamic -avoid-version
+@HAVE_ESD_TRUE@esdldflags = -module -avoid-version
 @HAVE_ESD_FALSE@esdldflags = 
 @HAVE_ESD_TRUE@esdsources = ao_esd.c
 @HAVE_ESD_FALSE@esdsources = 
 
--- libao-0.8.3/src/plugins/arts/Makefile.in.orig	2002-11-18 22:27:47.000000000 -0800
+++ libao-0.8.3/src/plugins/arts/Makefile.in	2002-11-18 22:28:01.000000000 -0800
@@ -94,7 +94,7 @@
 
 @HAVE_ARTS_TRUE@artsltlibs = libarts.la
 @HAVE_ARTS_FALSE@artsltlibs = 
-@HAVE_ARTS_TRUE@artsldflags = -export-dynamic -avoid-version
+@HAVE_ARTS_TRUE@artsldflags = -module -avoid-version
 @HAVE_ARTS_FALSE@artsldflags = 
 @HAVE_ARTS_TRUE@artssources = ao_esd.c
 @HAVE_ARTS_FALSE@artssources = 


--- libao-0.8.3/configure	2002-07-19 07:16:55.000000000 -0700
+++ libao-0.8.3/configure	2002-11-18 22:43:21.000000000 -0800
@@ -3479,7 +3479,7 @@
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
     #	     `"' quotes if we put them in here... so don't!
-    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring) -framework CoreAudio'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
@@ -5429,8 +5429,9 @@
         *-darwin*)
                 SILLY_CARBON_PATH="-I/System/Library/Frameworks/Carbon.framework/Versions/A/Headers"
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char $SILLY_CARBON_PATH"
-                CFLAGS="-O20 -D__NO_MATH_INLINES -fsigned-char $SILLY_CARBON_PATH"
-                PROFILE="-O20 -g -pg -D__NO_MATH_INLINES -fsigned-char $SILLY_CARBON_PATH" ;;
+                CFLAGS="-D__NO_MATH_INLINES -fsigned-char -framework CoreAudio -Ddlsym=dlsym_auto_underscore"
+                LDFLAGS="-framework CoreAudio"
+                PROFILE="-g -pg -D__NO_MATH_INLINES -fsigned-char -framework CoreAudio " ;;
         *)
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char"
                 CFLAGS="-O20 -D__NO_MATH_INLINES -fsigned-char"
--- libao-0.8.3/src/plugins/macosx/ao_macosx.c.orig	2001-10-25 02:44:01.000000000 -0700
+++ libao-0.8.3/src/plugins/macosx/ao_macosx.c	2003-03-31 17:24:18.000000000 -0800
@@ -190,6 +190,10 @@
     }
 
     propertySize = sizeof(internal->outputBufferByteCount);
+
+    internal->outputBufferByteCount = 8192;
+    status = AudioDeviceSetProperty(internal->outputDeviceID, 0, 0, false, kAudioDevicePropertyBufferSize, propertySize, &internal->outputBufferByteCount);
+
     status = AudioDeviceGetProperty(internal->outputDeviceID, 0, false, kAudioDevicePropertyBufferSize, &propertySize, &internal->outputBufferByteCount);
     if (status) {
         fprintf(stderr, "ao_macosx_open: AudioDeviceGetProperty returned %d when getting kAudioDevicePropertyBufferSize\n", (int)status);
@@ -200,7 +204,7 @@
 
     // It appears that AudioDeviceGetProperty lies about this property in DP4
     // Set the actual value
-    internal->outputBufferByteCount = 32768;
+    // internal->outputBufferByteCount = 32768;
 
     // Set the IO proc that CoreAudio will call when it needs data, but don't start
     // the stream yet.
@@ -301,7 +305,7 @@
         
         internal->bytesQueued += bytesToCopy;
         
-        //fprintf(stderr, "Copy: %d bytes, %d bytes left\n", bytesToCopy, internal->availableByteCount);
+        // fprintf(stderr, "Copy: %d bytes, %d bytes left\n", bytesToCopy, internal->availableByteCount);
         pthread_mutex_unlock(&internal->mutex);
         
         // We have to wait to start the device until we have some data queued.
@@ -383,9 +387,9 @@
     // Find the first valid frame and the number of valid frames
     pthread_mutex_lock(&internal->mutex);
 
-    bytesToCopy = internal->outputBufferByteCount;
+    bytesToCopy = internal->outputBufferByteCount/2;
     validByteCount = internal->validByteCount;
-    outBuffer = (float *)outOutputData;
+    outBuffer = (float *)outOutputData->mBuffers[0].mData;
 
     if (validByteCount < bytesToCopy && !internal->isStopping) {
         // Not enough data ... let it build up a bit more before we start copying stuff over.
@@ -424,8 +428,8 @@
         sample++;
     }
     
-    pthread_cond_signal(&internal->condition);
     pthread_mutex_unlock(&internal->mutex);
+    pthread_cond_signal(&internal->condition);
     
     return 0;
 }
