diff -Nurd -x'*~' Xlt-9.2.9.orig/Makefile.in Xlt-9.2.9/Makefile.in
--- Xlt-9.2.9.orig/Makefile.in	2001-07-21 11:36:44.000000000 -0400
+++ Xlt-9.2.9/Makefile.in	2006-07-13 13:26:48.000000000 -0400
@@ -97,7 +97,7 @@
 version_suffix = @version_suffix@
 
 MAINTAINERCLEANFILES = Makefile.in configure libtool config.guess config.sub ltconfig ltmain.sh aclocal.m4 config.h.in install-sh missing mkinstalldirs
-SUBDIRS = . lib doc tests
+SUBDIRS = . lib doc
 
 EXTRA_DIST = configure COPYING.LIB ac_find_xlt.m4 ac_find_xpm.m4 CVSMake
 
@@ -401,8 +401,9 @@
 #aclocal_DATA = ac_find_xlt.m4 ac_find_xpm.m4
 
 @Aclocal_TRUE@install-data-local:
-@Aclocal_TRUE@	if test -w $(aclocaldir) ; then $(INSTALL_DATA) $(srcdir)/ac_find_xlt.m4 $(aclocaldir) ; fi
-@Aclocal_TRUE@	if test -w $(aclocaldir) ; then $(INSTALL_DATA) $(srcdir)/ac_find_xpm.m4 $(aclocaldir) ; fi
+@Aclocal_TRUE@	$(mkinstalldirs) $(DESTDIR)$(aclocaldir)
+@Aclocal_TRUE@	$(INSTALL_DATA) $(srcdir)/ac_find_xlt.m4 $(DESTDIR)$(aclocaldir)
+@Aclocal_TRUE@	$(INSTALL_DATA) $(srcdir)/ac_find_xpm.m4 $(DESTDIR)$(aclocaldir)
 
 .PHONY: changelog UpdateCheck
 
diff -Nurd -x'*~' Xlt-9.2.9.orig/configure Xlt-9.2.9/configure
--- Xlt-9.2.9.orig/configure	2001-07-21 11:36:42.000000000 -0400
+++ Xlt-9.2.9/configure	2006-07-13 13:25:03.000000000 -0400
@@ -853,11 +853,8 @@
 
 
 
-if (aclocal --print-ac-dir) </dev/null >/dev/null 2>&1
-then
-    ACLOCALDIR=`aclocal --print-ac-dir`
+    ACLOCALDIR="${prefix}/share/aclocal"
     
-fi
 
 
 if test x$ACLOCALDIR != x ; then
diff -Nurd -x'*~' Xlt-9.2.9.orig/doc/Makefile.in Xlt-9.2.9/doc/Makefile.in
--- Xlt-9.2.9.orig/doc/Makefile.in	2001-07-21 11:36:44.000000000 -0400
+++ Xlt-9.2.9/doc/Makefile.in	2006-07-13 13:13:36.000000000 -0400
@@ -100,7 +100,7 @@
 
 xltdir = $(includedir)/Xlt
 
-mandir = $(prefix)/man
+mandir = $(prefix)/share/man
 man_MANS =  StrokeInstall.3 Strokes.3 Xlt.3 XltAnimatedButton.3 XltAppInitialize.3 XltAppShell.3 XltBubbleButton.3 XltCheckArgSize.3 XltCreateAnimatedButton.3 XltCreateBubbleButton.3 XltCreateFilePicker.3 XltCreateHost.3 XltCreateNumEntry.3 XltDisplayFallbackResources.3 XltDisplayOptions.3 XltEntrySB.3 XltFilePicker.3 XltFilePickerGetChild.3 XltFilePickerGetString.3 XltFilePickerSetString.3 XltFontChooser.3 XltGetValues.3 XltHost.3 XltHostCreateTranscript.3 XltHostDiscardOutput.3 XltHostSelect.3 XltHostSendData.3 XltHostSendString.3 XltNumEntry.3 XltNumEntryGetChild.3 XltRedirectStdErr.3 XltSelectionBox.3 XltSelectDebugLevel.3 XltSetClientIcon.3 XltSetValues.3 XltSlideContext.3 XltToolBarConfigure.3 XltToolBarAddItem.3 XltWaitTillMapped.3 XltWorking.3 XltYesNo.3
 
 
@@ -108,7 +108,7 @@
 
 html_DATA = $(html_HTML)
 
-htmldir = $(prefix)/LessTif/doc/html
+htmldir = $(prefix)/share/doc/xlt
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../config.h
 CONFIG_CLEAN_FILES =  Strokes.3 StrokeInstall.3 Xlt.3 \
