diff -urN bibclean-2.11.4.orig/bibclean.copyright bibclean-2.11.4/bibclean.copyright
--- bibclean-2.11.4.orig/bibclean.copyright	1970-01-01 01:00:00.000000000 +0100
+++ bibclean-2.11.4/bibclean.copyright	2006-12-02 18:41:53.000000000 +0000
@@ -0,0 +1,68 @@
+This package was debianized by Henning Makholm <henning@makholm.net> on
+September 7-21, 2003.
+
+It was downloaded from
+  http://www.math.utah.edu/pub/bibclean/bibclean-2.11.4.tar.gz
+
+Upstream Author:
+  Nelson H. F. Beebe
+
+Copyright:
+  The upstream tarball contains no copyright notices, but the
+  following email from the author makes it clear that it is intended
+  to be "in the public domain, and freely distributable and
+  modifiable":
+
+Date: Sat, 20 Sep 2003 08:43:14 -0600 (MDT)
+From: "Nelson H. F. Beebe" <beebe@math.utah.edu>
+To: Henning Makholm <henning@makholm.net>
+Cc: beebe@math.utah.edu
+X-US-Mail: "Center for Scientific Computing, Department of Mathematics, 110
+        LCB, University of Utah, 155 S 1400 E RM 233, Salt Lake City, UT
+        84112-0090, USA"
+X-Telephone: +1 801 581 5254
+X-FAX: +1 801 585 1640, +1 801 581 4148
+X-URL: http://www.math.utah.edu/~beebe
+Subject: Re: Request for license clarification for bibclean
+In-Reply-To: Your message of Sat, 20 Sep 2003 16:14:26 +0200
+Message-ID: <CMM.0.92.0.1064068994.beebe@psi.math.utah.edu>
+
+Thanks for your query about the licensing issue with bibclean.
+
+I am of course happy to see bibclean used and freely distributed.  In
+the past, I have not placed explicit licenses on my software packages,
+expecting them to be consider in the public domain, and freely
+distributable and modifiable.  However, at the urging of Karl Berry
+and Richard Stallman in their response to instances of unscrupulous
+reuse of free software, I have starting adding explicit GPL licensing
+as new package releases come out.  It is not practical to retrofit
+such licenses into older releases.
+
+The next release of bibclean will have such a license.
+
+I am at the tail end of a year-long book project, so many of my
+software projects have been delayed.
+
+Recently, I found some time to work on the next version of bibclean,
+which is primarily polishing of packaging, and enhancing portability
+even further.  That version will be bibclean-2-14, but it is quite
+likely that it won't be out before late fall this year.
+
+It would therefore make sense to delay your Debian packaging until
+then.  In the meantime, you can continue to point people to the
+official archive at
+
+	ftp://ftp.math.utah.edu/pub/bibclean
+	http://www.math.utah.edu/pub/bibclean
+
+-------------------------------------------------------------------------------
+- Nelson H. F. Beebe                    Tel: +1 801 581 5254                  -
+- Center for Scientific Computing       FAX: +1 801 581 4148                  -
+- University of Utah                    Internet e-mail: beebe@math.utah.edu  -
+- Department of Mathematics, 110 LCB        beebe@acm.org  beebe@computer.org -
+- 155 S 1400 E RM 233                                                         -
+- Salt Lake City, UT 84112-0090, USA    URL: http://www.math.utah.edu/~beebe  -
+-------------------------------------------------------------------------------
+
+
+

 	  	 
