diff -Nurd -x'*~' djvulibre-3.5.24.orig/configure djvulibre-3.5.24/configure
--- djvulibre-3.5.24.orig/configure	2011-03-06 12:43:17.000000000 -0500
+++ djvulibre-3.5.24/configure	2013-07-27 01:45:39.000000000 -0400
@@ -18820,7 +18820,7 @@
 
     NSDEJAVU_CFLAGS=${X_CFLAGS}
     case "$host" in
-        *-cygwin* | *-mingw*)
+        *-cygwin* | *-mingw* | *-darwin*)
             NSDEJAVU_LIBS="`echo ${X_LIBS} | sed -e 's/-lXext/-lXt -lXext/'`"
             NSDEJAVU_LIBS="-no-undefined $NSDEJAVU_LIBS"
             ;;
diff -Nurd -x'*~' djvulibre-3.5.24.orig/gui/indep/utils/execdir.cpp djvulibre-3.5.24/gui/indep/utils/execdir.cpp
--- djvulibre-3.5.24.orig/gui/indep/utils/execdir.cpp	2011-03-06 12:43:17.000000000 -0500
+++ djvulibre-3.5.24/gui/indep/utils/execdir.cpp	2013-07-27 01:18:15.000000000 -0400
@@ -141,22 +141,22 @@
     {
       GURL url = GURL(file, paths[pos]);
       GUTF8String urls = (const char*)url;
-      int pos = urls.search(osi);
-      if (pos >= 0)
+      int ipos = urls.search(osi);
+      if (ipos >= 0)
         {
           GUTF8String urlx;
-          urlx += urls.substr(0, pos);
+          urlx += urls.substr(0, ipos);
           urlx += djview3;
-          urlx += urls.substr(pos+strlen(osi), -1);
+          urlx += urls.substr(ipos+strlen(osi), -1);
           GURL url = GURL::UTF8(urlx);
           if (url.is_file())
             return url;
         }
     }
   // end hack alert
-  for (GPosition pos=paths; pos; ++pos)
+  for (GPosition ipos=paths; ipos; ++ipos)
     {
-      GURL url = GURL(file, paths[pos]);
+      GURL url = GURL(file, paths[ipos]);
       if(url.is_file())
         return url;
     }
diff -Nurd -x'*~' djvulibre-3.5.24.orig/gui/nsdejavu/Makefile.in djvulibre-3.5.24/gui/nsdejavu/Makefile.in
--- djvulibre-3.5.24.orig/gui/nsdejavu/Makefile.in	2011-03-06 12:43:17.000000000 -0500
+++ djvulibre-3.5.24/gui/nsdejavu/Makefile.in	2013-07-27 01:18:15.000000000 -0400
@@ -52,9 +52,9 @@
  -I${top_srcdir}/gui/shared/res \
  -I${top_srcdir}/gui/npsdk 
 
-CFLAGS = @CPPFLAGS@ ${DEFS} ${OPTS} ${INCS} ${NSDEJAVU_CFLAGS} ${GLIB_CFLAGS}
+CFLAGS = ${DEFS} ${OPTS} ${INCS} ${NSDEJAVU_CFLAGS} @CPPFLAGS@ ${GLIB_CFLAGS}
 
-LIBS = @LDFLAGS@ ${NSDEJAVU_LIBS} 
+LIBS = @LDFLAGS@ ${NSDEJAVU_LIBS} -lglib-2.0
 
 all: nsdejavu.la nsdejavu.x nsdejavu.1
 
diff -Nurd -x'*~' djvulibre-3.5.24.orig/libdjvu/GContainer.h djvulibre-3.5.24/libdjvu/GContainer.h
--- djvulibre-3.5.24.orig/libdjvu/GContainer.h	2011-03-06 12:43:17.000000000 -0500
+++ djvulibre-3.5.24/libdjvu/GContainer.h	2013-07-27 01:18:15.000000000 -0400
@@ -962,7 +962,7 @@
   /** Inserts an element before the first element of the list. 
       The new element is initialized with a copy of argument #elt#. */
   void prepend(const TYPE &elt)
-    { GListImpl<TI>::prepend(newnode((const TI&)elt)); }
+    { GListImpl<TI>::prepend(this->newnode((const TI&)elt)); }
   /** Inserts a new element after the list element at position #pos#.  When
       position #pos# is null the element is inserted at the beginning of the
       list.  The new element is initialized with a copy of #elt#. */
diff -Nurd -x'*~' djvulibre-3.5.24.orig/tools/jb2cmp/Makefile.in djvulibre-3.5.24/tools/jb2cmp/Makefile.in
--- djvulibre-3.5.24.orig/tools/jb2cmp/Makefile.in	2011-03-06 12:43:17.000000000 -0500
+++ djvulibre-3.5.24/tools/jb2cmp/Makefile.in	2013-07-27 01:18:15.000000000 -0400
@@ -49,8 +49,8 @@
 INCS = -I${top_builddir} -I${top_srcdir} -I${srcdir}
 FLAGS = ${DEFS} ${INCS} ${OPTS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${THREAD_LIBS} @LIBS@
-CFLAGS = @CPPFLAGS@ @CFLAGS@ ${FLAGS}
-CXXFLAGS = @CPPFLAGS@ @CXXFLAGS@ ${CXXRPOFLAGS} ${FLAGS}
+CFLAGS = ${FLAGS} @CPPFLAGS@ @CFLAGS@
+CXXFLAGS = ${FLAGS} @CPPFLAGS@ @CXXFLAGS@ ${CXXRPOFLAGS}
 
 
 OBJS = classify.o cuts.o frames.o patterns.o
diff -Nurd -x'*~' djvulibre-3.5.24.orig/tools/tiff2pdf.c djvulibre-3.5.24/tools/tiff2pdf.c
--- djvulibre-3.5.24.orig/tools/tiff2pdf.c	2011-03-06 12:43:17.000000000 -0500
+++ djvulibre-3.5.24/tools/tiff2pdf.c	2013-07-27 01:41:20.000000000 -0400
@@ -2116,7 +2116,7 @@
 			buffer= (unsigned char*) _TIFFmalloc(t2p->tiff_datasize);
 			if(buffer==NULL){
 				TIFFError(TIFF2PDF_MODULE, 
-					"Can't allocate %u bytes of memory for t2p_readwrite_pdf_image, %s", 
+					"Can't allocate %lu bytes of memory for t2p_readwrite_pdf_image, %s", 
 					t2p->tiff_datasize, 
 					TIFFFileName(input));
 				t2p->t2p_error = T2P_ERR_ERROR;
@@ -2138,7 +2138,7 @@
                         memset(buffer, 0, t2p->tiff_datasize);
 			if(buffer==NULL){
 				TIFFError(TIFF2PDF_MODULE, 
-					"Can't allocate %u bytes of memory for t2p_readwrite_pdf_image, %s", 
+					"Can't allocate %lu bytes of memory for t2p_readwrite_pdf_image, %s", 
 					t2p->tiff_datasize, 
 					TIFFFileName(input));
 				t2p->t2p_error = T2P_ERR_ERROR;
@@ -2161,7 +2161,7 @@
                                 memset(buffer, 0, t2p->tiff_datasize);
 				if(buffer==NULL){
 					TIFFError(TIFF2PDF_MODULE, 
-						"Can't allocate %u bytes of memory for t2p_readwrite_pdf_image, %s", 
+						"Can't allocate %lu bytes of memory for t2p_readwrite_pdf_image, %s", 
 						t2p->tiff_datasize, 
 						TIFFFileName(input));
 					t2p->t2p_error = T2P_ERR_ERROR;
@@ -2220,7 +2220,7 @@
                                 memset(buffer, 0, t2p->tiff_datasize);
 				if(buffer==NULL){
 					TIFFError(TIFF2PDF_MODULE, 
-						"Can't allocate %u bytes of memory for t2p_readwrite_pdf_image, %s", 
+						"Can't allocate %lu bytes of memory for t2p_readwrite_pdf_image, %s", 
 						t2p->tiff_datasize, 
 						TIFFFileName(input));
 					t2p->t2p_error = T2P_ERR_ERROR;
@@ -2261,7 +2261,7 @@
                         memset(buffer, 0, t2p->tiff_datasize);
 			if(buffer==NULL){
 				TIFFError(TIFF2PDF_MODULE, 
-					"Can't allocate %u bytes of memory for t2p_readwrite_pdf_image, %s", 
+					"Can't allocate %lu bytes of memory for t2p_readwrite_pdf_image, %s", 
 					t2p->tiff_datasize, 
 					TIFFFileName(input));
 				t2p->t2p_error = T2P_ERR_ERROR;
@@ -2322,7 +2322,7 @@
                 memset(buffer, 0, t2p->tiff_datasize);
 		if(buffer==NULL){
 			TIFFError(TIFF2PDF_MODULE, 
-				"Can't allocate %u bytes of memory for t2p_readwrite_pdf_image, %s", 
+				"Can't allocate %lu bytes of memory for t2p_readwrite_pdf_image, %s", 
 				t2p->tiff_datasize, 
 				TIFFFileName(input));
 			t2p->t2p_error = T2P_ERR_ERROR;
@@ -2360,7 +2360,7 @@
                         memset(buffer, 0, t2p->tiff_datasize);
 			if(buffer==NULL){
 				TIFFError(TIFF2PDF_MODULE, 
-					"Can't allocate %u bytes of memory for t2p_readwrite_pdf_image, %s", 
+					"Can't allocate %lu bytes of memory for t2p_readwrite_pdf_image, %s", 
 					t2p->tiff_datasize, 
 					TIFFFileName(input));
 				t2p->t2p_error = T2P_ERR_ERROR;
@@ -2369,7 +2369,7 @@
 			samplebuffer = (unsigned char*) _TIFFmalloc(stripsize);
 			if(samplebuffer==NULL){
 				TIFFError(TIFF2PDF_MODULE, 
-					"Can't allocate %u bytes of memory for t2p_readwrite_pdf_image, %s", 
+					"Can't allocate %lu bytes of memory for t2p_readwrite_pdf_image, %s", 
 					t2p->tiff_datasize, 
 					TIFFFileName(input));
 				t2p->t2p_error = T2P_ERR_ERROR;
@@ -2409,7 +2409,7 @@
                 memset(buffer, 0, t2p->tiff_datasize);
 		if(buffer==NULL){
 			TIFFError(TIFF2PDF_MODULE, 
-				"Can't allocate %u bytes of memory for t2p_readwrite_pdf_image, %s", 
+				"Can't allocate %lu bytes of memory for t2p_readwrite_pdf_image, %s", 
 				t2p->tiff_datasize, 
 				TIFFFileName(input));
 			t2p->t2p_error = T2P_ERR_ERROR;
@@ -2442,7 +2442,7 @@
 				t2p->tiff_datasize * t2p->tiff_samplesperpixel);
 			if(samplebuffer==NULL){
 				TIFFError(TIFF2PDF_MODULE, 
-					"Can't allocate %u bytes of memory for t2p_readwrite_pdf_image, %s", 
+					"Can't allocate %lu bytes of memory for t2p_readwrite_pdf_image, %s", 
 					t2p->tiff_datasize, 
 					TIFFFileName(input));
 				t2p->t2p_error = T2P_ERR_ERROR;
@@ -2472,7 +2472,7 @@
 				t2p->tiff_width*t2p->tiff_length*4);
 			if(samplebuffer==NULL){
 				TIFFError(TIFF2PDF_MODULE, 
-					"Can't allocate %u bytes of memory for t2p_readwrite_pdf_image, %s", 
+					"Can't allocate %lu bytes of memory for t2p_readwrite_pdf_image, %s", 
 					t2p->tiff_datasize, 
 					TIFFFileName(input));
 				t2p->t2p_error = T2P_ERR_ERROR;
@@ -2658,7 +2658,7 @@
 			buffer= (unsigned char*) _TIFFmalloc(t2p->tiff_datasize);
 			if(buffer==NULL){
 				TIFFError(TIFF2PDF_MODULE, 
-					"Can't allocate %u bytes of memory "
+					"Can't allocate %lu bytes of memory "
                                         "for t2p_readwrite_pdf_image_tile, %s", 
 					t2p->tiff_datasize, 
 					TIFFFileName(input));
@@ -2679,7 +2679,7 @@
 			buffer= (unsigned char*) _TIFFmalloc(t2p->tiff_datasize);
 			if(buffer==NULL){
 				TIFFError(TIFF2PDF_MODULE, 
-					"Can't allocate %u bytes of memory "
+					"Can't allocate %lu bytes of memory "
                                         "for t2p_readwrite_pdf_image_tile, %s", 
 					t2p->tiff_datasize, 
 					TIFFFileName(input));
@@ -2708,7 +2708,7 @@
 			buffer=(unsigned char*) _TIFFmalloc(t2p->tiff_datasize);
 			if(buffer==NULL){
 				TIFFError(TIFF2PDF_MODULE, 
-					"Can't allocate %u bytes of memory "
+					"Can't allocate %lu bytes of memory "
                                         "for t2p_readwrite_pdf_image, %s", 
 					t2p->tiff_datasize, 
 					TIFFFileName(input));
@@ -2748,7 +2748,7 @@
 			buffer= (unsigned char*) _TIFFmalloc(t2p->tiff_datasize);
 			if(buffer==NULL){
 				TIFFError(TIFF2PDF_MODULE, 
-					"Can't allocate %u bytes of memory "
+					"Can't allocate %lu bytes of memory "
                                         "for t2p_readwrite_pdf_image_tile, %s", 
 					t2p->tiff_datasize, 
 					TIFFFileName(input));
@@ -2791,7 +2791,7 @@
 		buffer = (unsigned char*) _TIFFmalloc(t2p->tiff_datasize);
 		if(buffer==NULL){
 			TIFFError(TIFF2PDF_MODULE, 
-				"Can't allocate %u bytes of memory for "
+				"Can't allocate %lu bytes of memory for "
                                 "t2p_readwrite_pdf_image_tile, %s", 
 				t2p->tiff_datasize, 
 				TIFFFileName(input));
@@ -2824,7 +2824,7 @@
 			buffer = (unsigned char*) _TIFFmalloc(t2p->tiff_datasize);
 			if(buffer==NULL){
 				TIFFError(TIFF2PDF_MODULE, 
-					"Can't allocate %u bytes of memory "
+					"Can't allocate %lu bytes of memory "
                                         "for t2p_readwrite_pdf_image_tile, %s", 
 					t2p->tiff_datasize, 
 					TIFFFileName(input));
@@ -2834,7 +2834,7 @@
 			samplebuffer = (unsigned char*) _TIFFmalloc(t2p->tiff_datasize);
 			if(samplebuffer==NULL){
 				TIFFError(TIFF2PDF_MODULE, 
-					"Can't allocate %u bytes of memory "
+					"Can't allocate %lu bytes of memory "
                                         "for t2p_readwrite_pdf_image_tile, %s", 
 					t2p->tiff_datasize, 
 					TIFFFileName(input));
@@ -2873,7 +2873,7 @@
 			buffer = (unsigned char*) _TIFFmalloc(t2p->tiff_datasize);
 			if(buffer==NULL){
 				TIFFError(TIFF2PDF_MODULE, 
-					"Can't allocate %u bytes of memory "
+					"Can't allocate %lu bytes of memory "
                                         "for t2p_readwrite_pdf_image_tile, %s", 
 					t2p->tiff_datasize, 
 					TIFFFileName(input));
