diff -Naur ghostscript-9.07/base/gdevpng.c ghostscript-9.07.modified/base/gdevpng.c
--- ghostscript-9.07/base/gdevpng.c	2013-02-13 23:58:13.000000000 -0800
+++ ghostscript-9.07.modified/base/gdevpng.c	2013-05-28 21:14:21.000000000 -0700
@@ -692,6 +692,7 @@
 
 #if PNG_LIBPNG_VER_MINOR < 5
 
+#if 0 /* not required in 1.5 */
 /*
  * Patch around a static reference to a never-used procedure.
  * This could be avoided if we were willing to edit pngconf.h to
@@ -712,6 +713,7 @@
 }
 #endif
 #endif
+#endif
 
 static int
 pngalpha_open(gx_device * pdev)
diff -Naur ghostscript-9.07/base/gxpcolor.h ghostscript-9.07.modified/base/gxpcolor.h
--- ghostscript-9.07/base/gxpcolor.h	2013-02-13 23:58:13.000000000 -0800
+++ ghostscript-9.07.modified/base/gxpcolor.h	2013-05-28 21:14:21.000000000 -0700
@@ -20,6 +20,7 @@
 #ifndef gxpcolor_INCLUDED
 #  define gxpcolor_INCLUDED
 
+#include "gxdcolor.h"
 #include "gspcolor.h"
 #include "gxcspace.h"
 #include "gxdevice.h"
diff -Naur ghostscript-9.07/base/mkromfs.c ghostscript-9.07.modified/base/mkromfs.c
--- ghostscript-9.07/base/mkromfs.c	2013-02-13 23:58:13.000000000 -0800
+++ ghostscript-9.07.modified/base/mkromfs.c	2013-05-28 21:14:21.000000000 -0700
@@ -2297,7 +2297,7 @@
 #endif
     fprintf(out,"\n#include \"stdint_.h\"\n");
     fprintf(out,"\n#include \"time_.h\"\n\n");
-    fprintf(out,"    time_t gs_romfs_buildtime = %ld;\n\n", time(NULL));
+    fprintf(out,"    time_t gs_romfs_buildtime = %lld;\n\n", (long long)time(NULL));
 
     /* process the remaining arguments (options interspersed with paths) */
     for (; atarg < argc; atarg++) {
diff -Naur ghostscript-9.07/configure ghostscript-9.07.modified/configure
--- ghostscript-9.07/configure	2013-02-13 23:58:14.000000000 -0800
+++ ghostscript-9.07.modified/configure	2013-05-28 21:14:21.000000000 -0700
@@ -836,6 +836,7 @@
 LIBS
 CPPFLAGS
 CPP
+CPPFLAGS
 XMKMF'
 
 
diff -Naur ghostscript-9.07/psi/imainarg.c ghostscript-9.07.modified/psi/imainarg.c
--- ghostscript-9.07/psi/imainarg.c	2013-02-13 23:58:13.000000000 -0800
+++ ghostscript-9.07.modified/psi/imainarg.c	2013-05-28 21:22:42.000000000 -0700
@@ -205,6 +205,11 @@
                 return e_Fatal;
         }
     }
+
+	code = gs_main_init1(minst);
+	if (code < 0)
+		return code;
+
     while ((arg = arg_next(&args, &code, minst->heap)) != 0) {
         switch (*arg) {
             case '-':
