diff -ruN zvbi-0.2.34-orig/contrib/Makefile.in zvbi-0.2.34/contrib/Makefile.in
--- zvbi-0.2.34-orig/contrib/Makefile.in	2013-07-04 23:27:56.000000000 -0400
+++ zvbi-0.2.34/contrib/Makefile.in	2013-08-31 17:29:20.000000000 -0400
@@ -101,9 +101,9 @@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
+	$(CPPFLAGS) $(X_CFLAGS) $(AM_CFLAGS) $(CFLAGS)
 LTCOMPILE = $(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
+	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(X_CFLAGS) \
 	$(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
 LINK = $(LIBTOOL) --tag=CC --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
@@ -294,7 +294,7 @@
 
 AM_CPPFLAGS = \
 	-I$(top_srcdir) \
-	$(X_CFLAGS) \
+	\
 	-D_REENTRANT \
 	-D_GNU_SOURCE
 
