diff -Nurd -x'*~' e16-0.16.8.5.orig/src/Makefile.in e16-0.16.8.5/src/Makefile.in
--- e16-0.16.8.5.orig/src/Makefile.in	2006-12-16 09:46:36.000000000 -0500
+++ e16-0.16.8.5/src/Makefile.in	2007-01-12 02:40:35.000000000 -0500
@@ -402,8 +402,8 @@
 	$(LTLIBINTL)		\
 	$(LTLIBICONV)		\
 	@ecore_libs@		\
-	$(ESD_LIBS) 		\
 	$(IMLIB2_LIBS) 		\
+	$(ESD_LIBS) 		\
 	$(XFT_LIBS) 		\
 	$(X_PRE_LIBS) 		\
 	$(X_LIBS) 		\
@@ -412,7 +412,7 @@
 	$(E_X_LIBS) 		\
 	-lX11 -lm
 
-INCLUDES = -I$(top_builddir) -I$(top_srcdir) -I$(top_srcdir)/intl @ecore_cflags@ $(ESD_CFLAGS) $(IMLIB2_CFLAGS) $(XFT_CFLAGS) $(X_CFLAGS)
+INCLUDES = -I$(top_builddir) -I$(top_srcdir) @ecore_cflags@ $(IMLIB2_CFLAGS) $(ESD_CFLAGS) $(XFT_CFLAGS) $(X_CFLAGS)
 all: all-recursive
 
 .SUFFIXES:
diff -Nurd -x'*~' e16-0.16.8.5.orig/themes/Makefile.in e16-0.16.8.5/themes/Makefile.in
--- e16-0.16.8.5.orig/themes/Makefile.in	2006-12-16 09:46:36.000000000 -0500
+++ e16-0.16.8.5/themes/Makefile.in	2007-01-12 02:41:54.000000000 -0500
@@ -387,7 +387,6 @@
 	for tf in $(THEME_FILES); do \
 		gzip -d -c < $(srcdir)/$$tf | (cd $(THEME_DIR); tar -xmf -); \
 	done
-	-if [ -n "$(USER)" ]; then chown -R $(USER) $(THEME_DIR); else chown -R root $(THEME_DIR); fi
 
 uninstall-local:
 	-for tf in $(THEME_FILES); do \
