diff -ruN dpkg-1.10.21.orig/config.h.in dpkg-1.10.21/config.h.in
--- dpkg-1.10.21.orig/config.h.in	2004-04-25 13:01:02.000000000 -0500
+++ dpkg-1.10.21/config.h.in	2020-10-19 19:29:04.000000000 -0500
@@ -203,8 +203,15 @@
 #include <sys/cdefs.h>
 #endif
 
-#ifdef HAVE_STDDEF_H
-#include <stddef.h>
+#ifdef __PTRDIFF_TYPE__
+# define PTR_INT_TYPE __PTRDIFF_TYPE__
+#else
+# ifdef HAVE_STDDEF_H
+#  include <stddef.h>
+#  define PTR_INT_TYPE ptrdiff_t
+# else
+#  define PTR_INT_TYPE long
+# endif
 #endif
 
 /* Use the definitions: */
diff -ruN dpkg-1.10.21.orig/configure dpkg-1.10.21/configure
--- dpkg-1.10.21.orig/configure	2004-04-25 13:01:03.000000000 -0500
+++ dpkg-1.10.21/configure	2020-10-19 19:30:44.000000000 -0500
@@ -7974,6 +7974,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
+#include <stddef.h>
 int
 main ()
 {
@@ -9241,6 +9242,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 
+#include <stdlib.h>
 #include <stdarg.h>
 main(){
 va_list v1,v2;
diff -ruN dpkg-1.10.21.orig/dpkg-deb/main.c dpkg-1.10.21/dpkg-deb/main.c
--- dpkg-1.10.21.orig/dpkg-deb/main.c	2004-04-24 17:18:29.000000000 -0500
+++ dpkg-1.10.21/dpkg-deb/main.c	2020-10-19 19:54:04.000000000 -0500
@@ -40,6 +40,7 @@
 
 #include "dpkg-deb.h"
 
+void nffreeall();
 const char* showformat	= "${Package}\t${Version}\n";
 
 static void printversion(void) {
diff -ruN dpkg-1.10.21.orig/main/enquiry.c dpkg-1.10.21/main/enquiry.c
--- dpkg-1.10.21.orig/main/enquiry.c	2003-10-25 15:03:21.000000000 -0500
+++ dpkg-1.10.21/main/enquiry.c	2020-10-19 19:57:33.000000000 -0500
@@ -33,6 +33,7 @@
 #include <sys/ioctl.h>
 #include <sys/termios.h>
 #include <fcntl.h>
+#include <stddef.h>
 
 #include <dpkg.h>
 #include <dpkg-db.h>
diff -ruN dpkg-1.10.21.orig/split/main.c dpkg-1.10.21/split/main.c
--- dpkg-1.10.21.orig/split/main.c	2003-10-25 15:03:21.000000000 -0500
+++ dpkg-1.10.21/split/main.c	2020-10-19 19:57:49.000000000 -0500
@@ -25,6 +25,7 @@
 #include <stdlib.h>
 #include <limits.h>
 #include <assert.h>
+#include <sys/stat.h>
 
 #include <dpkg.h>
 #include <dpkg-db.h>
