diff -Nurd -x'*~' bullet-2.80-rev2527.orig/configure.ac bullet-2.80-rev2527/configure.ac
--- bullet-2.80-rev2527.orig/configure.ac	2012-03-02 19:07:20.000000000 -0500
+++ bullet-2.80-rev2527/configure.ac	2015-02-17 13:29:31.000000000 -0500
@@ -13,7 +13,7 @@
     [bullet@erwincoumans.com])
 AC_CANONICAL_HOST
 AC_CONFIG_SRCDIR([configure.ac])
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([subdir-objects])
 AM_PROG_CC_C_O
 AC_PROG_CXX
 AC_PROG_LIBTOOL
