diff -Nurd freeciv-2.6.4.orig/client/gui-gtk-2.0/gui_main.c freeciv-2.6.4/client/gui-gtk-2.0/gui_main.c
--- freeciv-2.6.4.orig/client/gui-gtk-2.0/gui_main.c	2021-04-01 07:38:24.000000000 -0400
+++ freeciv-2.6.4/client/gui-gtk-2.0/gui_main.c	2021-04-04 01:54:51.000000000 -0400
@@ -47,6 +47,11 @@
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 
+#if !GTK_CHECK_VERSION(2, 90, 7)
+#define GDK_KEY_minus GDK_minus
+#define GDK_KEY_plus GDK_plus
+#endif
+
 /* utility */
 #include "fc_cmdline.h"
 #include "fciconv.h"
