diff -Nurd allegro-5.0.11.orig/addons/native_dialog/osx_dialog.m allegro-5.0.11/addons/native_dialog/osx_dialog.m
--- allegro-5.0.11.orig/addons/native_dialog/osx_dialog.m	2013-02-06 06:57:46.000000000 -0500
+++ allegro-5.0.11/addons/native_dialog/osx_dialog.m	2021-07-24 20:08:32.000000000 -0400
@@ -4,6 +4,11 @@
 
 #import <Cocoa/Cocoa.h>
 
+/* Normally we should include aintosx.h to get this prototype but it doesn't
+* work. So repeat the prototype here as a stopgap measure.
+*/
+void _al_osx_clear_mouse_state(void);
+
 bool _al_init_native_dialog_addon(void)
 {
    return true;
diff -Nurd allegro-5.0.11.orig/include/allegro5/platform/aintosx.h.rej allegro-5.0.11/include/allegro5/platform/aintosx.h.rej
--- allegro-5.0.11.orig/include/allegro5/platform/aintosx.h.rej	1969-12-31 19:00:00.000000000 -0500
+++ allegro-5.0.11/include/allegro5/platform/aintosx.h.rej	2021-07-24 20:03:43.000000000 -0400
@@ -0,0 +1,22 @@
+***************
+*** 108,118 ****
+     HID_DEVICE* devices;
+  } HID_DEVICE_COLLECTION;
+  
+- 
+  int _al_osx_bootstrap_ok(void);
+  
+- ALLEGRO_PATH *_al_osx_get_path(int id);
+- 
+  void _al_osx_keyboard_handler(int pressed, NSEvent *event, ALLEGRO_DISPLAY*);
+  void _al_osx_keyboard_modifiers(unsigned int new_mods, ALLEGRO_DISPLAY*);
+  void _al_osx_keyboard_focused(int focused, int state);
+--- 108,115 ----
+     HID_DEVICE* devices;
+  } HID_DEVICE_COLLECTION;
+  
+  int _al_osx_bootstrap_ok(void);
+  
+  void _al_osx_keyboard_handler(int pressed, NSEvent *event, ALLEGRO_DISPLAY*);
+  void _al_osx_keyboard_modifiers(unsigned int new_mods, ALLEGRO_DISPLAY*);
+  void _al_osx_keyboard_focused(int focused, int state);
diff -Nurd allegro-5.0.11.orig/include/allegro5/platform/alosx.h allegro-5.0.11/include/allegro5/platform/alosx.h
--- allegro-5.0.11.orig/include/allegro5/platform/alosx.h	2011-03-11 18:30:40.000000000 -0500
+++ allegro-5.0.11/include/allegro5/platform/alosx.h	2021-07-24 20:03:43.000000000 -0400
@@ -24,6 +24,7 @@
 #endif
 
 
+ALLEGRO_PATH *_al_osx_get_path(int id);
 
 #include <stdio.h>
 #include <stdlib.h>
@@ -40,7 +41,6 @@
    #import <CoreAudio/CoreAudio.h>
    #import <AudioUnit/AudioUnit.h>
    #import <AudioToolbox/AudioToolbox.h>
-   #import <QuickTime/QuickTime.h>
    #import <IOKit/IOKitLib.h>
    #import <IOKit/IOCFPlugIn.h>
    #import <IOKit/hid/IOHIDLib.h>
diff -Nurd allegro-5.0.11.orig/src/allegro.c allegro-5.0.11/src/allegro.c
--- allegro-5.0.11.orig/src/allegro.c	2013-04-12 20:54:10.000000000 -0400
+++ allegro-5.0.11/src/allegro.c	2021-07-24 20:03:43.000000000 -0400
@@ -18,6 +18,7 @@
 
 #include "allegro5/allegro.h"
 #include "allegro5/platform/alplatf.h"
+#include ALLEGRO_INTERNAL_HEADER
 
 
 
