diff -Nurd smpeg-0.4.4.orig/MPEGaudio.h smpeg-0.4.4/MPEGaudio.h
--- smpeg-0.4.4.orig/MPEGaudio.h	2001-04-04 16:42:40
+++ smpeg-0.4.4/MPEGaudio.h	2025-11-22 09:58:24
@@ -149,6 +149,14 @@
 };
 
 /* The actual MPEG audio class */
+
+class MPEGaudio;
+void Play_MPEGaudioSDL(void *udata, Uint8 *stream, int len);
+int Play_MPEGaudio(MPEGaudio *audio, Uint8 *stream, int len);
+#ifdef THREADED_AUDIO
+int Decode_MPEGaudio(void *udata);
+#endif
+
 class MPEGaudio : public MPEGerror, public MPEGaudioaction {
 
     friend void Play_MPEGaudioSDL(void *udata, Uint8 *stream, int len);
diff -Nurd smpeg-0.4.4.orig/Makefile.am smpeg-0.4.4/Makefile.am
--- smpeg-0.4.4.orig/Makefile.am	2001-04-04 16:28:14
+++ smpeg-0.4.4/Makefile.am	2025-11-22 10:01:10
@@ -1,6 +1,8 @@
 
 ## Makefile.am for the smpeg library
 
+ACLOCAL_AMFLAGS = -I m4
+
 # These are the subdirectories that are always built
 SUBDIRS = audio video
 
diff -Nurd smpeg-0.4.4.orig/configure.in smpeg-0.4.4/configure.in
--- smpeg-0.4.4.orig/configure.in	2001-04-24 15:25:39
+++ smpeg-0.4.4/configure.in	2025-11-22 10:50:28
@@ -1,6 +1,9 @@
 
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(README)
+AC_INIT(smpeg,0.4.4)
+AC_CONFIG_SRCDIR(README)
+AC_CONFIG_MACRO_DIRS([m4])
+AC_CANONICAL_TARGET
 
 dnl Set various version strings - taken gratefully from the GTk sources
 #
@@ -38,12 +41,8 @@
 AC_SUBST(LT_AGE)
 
 dnl Setup for automake
-AM_INIT_AUTOMAKE(smpeg, $VERSION)
+AM_INIT_AUTOMAKE
 
-dnl Detect the canonical host and target build environment
-AC_CANONICAL_HOST
-AC_CANONICAL_TARGET
-
 dnl Check for tools
 
 AC_PROG_MAKE_SET
@@ -52,6 +51,7 @@
 AC_LIBTOOL_WIN32_DLL
 AM_PROG_LIBTOOL
 AC_PROG_INSTALL
+AM_PROG_AS
 
 dnl Work around the lack of certain typedefs.
 AC_TYPE_SOCKLEN_T
@@ -124,7 +124,7 @@
             :,
 	    AC_MSG_ERROR([*** SDL version $SDL_VERSION not found!])
 )
-CFLAGS="$CFLAGS $SDL_CFLAGS"
+CPPFLAGS="$CPPFLAGS $SDL_CFLAGS"
 LIBS="$LIBS $SDL_LIBS"
 
 dnl See if we need to pass -lm for the math library
@@ -195,13 +195,13 @@
 [  --enable-gtk-player     build a GTk sample SMPEG player [default=yes]],
               , enable_gtk_player=yes)
 have_gtk=no
-if test x$enable_gtk_player = xyes; then
-    AM_PATH_GTK(1.2.1, have_gtk=yes)
-    if test x$have_gtk = xyes; then
-        CFLAGS="$CFLAGS $GTK_CFLAGS"
-    fi
-    AC_SUBST(GTK_LIBS)
-fi
+#if test x$enable_gtk_player = xyes; then
+#    AM_PATH_GTK(1.2.1, have_gtk=yes)
+#    if test x$have_gtk = xyes; then
+#        CFLAGS="$CFLAGS $GTK_CFLAGS"
+#    fi
+#    AC_SUBST(GTK_LIBS)
+#fi
 AM_CONDITIONAL(HAVE_GTK, test x$have_gtk = xyes)
 
 dnl See if we can build the Mesa player
@@ -224,7 +224,7 @@
             AC_PATH_X
             AC_PATH_XTRA
             if test x$have_x = xyes; then
-                CFLAGS="$CFLAGS $X_CFLAGS"
+                CPPFLAGS="$CPPFLAGS $X_CFLAGS"
                 SYS_GL_LIBS="$X_LIBS -lGL -lGLU"
             else
                 SYS_GL_LIBS="-lGL -lGLU"
@@ -259,8 +259,8 @@
 fi
 
 dnl Add the source include directories
-CFLAGS="$CFLAGS -I.. -DNOCONTROLS"
-CFLAGS="$CFLAGS -I\$(top_srcdir) -I\$(top_srcdir)/audio -I\$(top_srcdir)/video"
+CPPFLAGS="-I.. -DNOCONTROLS $CPPFLAGS"
+CPPFLAGS="-I\$(top_srcdir) -I\$(top_srcdir)/audio -I\$(top_srcdir)/video $CPPFLAGS"
 CXXFLAGS="$CFLAGS"
 
 
diff -Nurd smpeg-0.4.4.orig/m4/acinclude-AC_TYPE_SOCKLEN_T.m4 smpeg-0.4.4/m4/acinclude-AC_TYPE_SOCKLEN_T.m4
--- smpeg-0.4.4.orig/m4/acinclude-AC_TYPE_SOCKLEN_T.m4	1969-12-31 19:00:00
+++ smpeg-0.4.4/m4/acinclude-AC_TYPE_SOCKLEN_T.m4	2025-11-22 10:07:53
@@ -0,0 +1,25 @@
+dnl This is just to silence aclocal about the macro not being used
+ifelse([AC_DISABLE_FAST_INSTALL])dnl
+###################################################################
+## To: autoconf@gnu.org 
+## Subject: socklen_t 
+## From: lars brinkhoff <lars@nocrew.org> 
+## Date: 26 Mar 1999 11:38:09 +0100 
+## 
+## Here's an attempt at a check for socklen_t.  AC_CHECK_TYPE doesn't
+## work because it doesn't search <sys/socket.h>.  Maybe that macro
+## should be changed instead.
+## 
+AC_DEFUN(AC_TYPE_SOCKLEN_T,
+[AC_CACHE_CHECK([for socklen_t], ac_cv_type_socklen_t,
+[
+  AC_TRY_COMPILE(
+  [#include <sys/socket.h>],
+  [socklen_t len = 42; return len;],
+  ac_cv_type_socklen_t=yes,
+  ac_cv_type_socklen_t=no)
+])
+  if test $ac_cv_type_socklen_t != yes; then
+    AC_DEFINE(socklen_t, int)
+  fi
+])
