diff -ru trackballs-1.1.4-orig/configure.ac trackballs-1.1.4/configure.ac
--- trackballs-1.1.4-orig/configure.ac	2007-05-25 10:52:10.000000000 -0400
+++ trackballs-1.1.4/configure.ac	2015-09-04 17:17:20.000000000 -0400
@@ -28,8 +28,10 @@
   *)       
 	WINRESOURCES=''
 	AC_CHECK_LIB(z,main,,[AC_MSG_ERROR([zlib should be included in most unix distributions, this is a strange error])])
+	PKG_CHECK_MODULES(X11, x11)
+	LDFLAGS="${LDFLAGS} ${X11_LIBS}"
+	CPPFLAGS="${CPPFLAGS} ${X11_CFLAGS}"
 	AC_CHECK_LIB(GL,main,,[AC_MSG_ERROR([you need OpenGL])])
-	LIBS="${LIBS} -L/usr/X11R6/lib/"     ## Problem with some slack distributions not including all X11R6 stuff properly.
 	AC_CHECK_LIB(GLU,main,,[AC_MSG_ERROR([missing libGLU, it should be a part of your OpenGL installation])])
 ;;
 esac
diff -ru trackballs-1.1.4-orig/share/Makefile.am trackballs-1.1.4/share/Makefile.am
--- trackballs-1.1.4-orig/share/Makefile.am	2005-04-02 05:20:38.000000000 -0500
+++ trackballs-1.1.4/share/Makefile.am	2015-09-04 17:01:32.000000000 -0400
@@ -4,6 +4,5 @@
 EXTRA_DIST = $(pkgdata_DATA) trackballs.6
 
 install-pkgdataDATA:
-	-install -D -m664 -g games highScores		$(DESTDIR)@with_highscores@
-	-chgrp -R games			$(DESTDIR)$(pkgdatadir)/levels
-	chmod -R g+r			$(DESTDIR)$(pkgdatadir)/levels
+	mkdir -p $(DESTDIR)$(pkgdatadir)
+	install -m664 highScores		$(DESTDIR)@with_highscores@
diff -ru trackballs-1.1.4-orig/src/Makefile.am trackballs-1.1.4/src/Makefile.am
--- trackballs-1.1.4-orig/src/Makefile.am	2005-04-02 06:38:29.000000000 -0500
+++ trackballs-1.1.4/src/Makefile.am	2015-09-04 17:01:32.000000000 -0400
@@ -30,7 +30,7 @@
 
 EXTRA_trackballs_SOURCES = trackballsIcon.rc
 
-trackballs_LDADD = @WINRESOURCES@
+trackballs_LDADD = @WINRESOURCES@ $(LTLIBINTL)
 trackballs_DEPENDENCIES = @WINRESOURCES@
 
 trackballsIcon.o : trackballsIcon.rc trackballsIcon.ico
Binary files trackballs-1.1.4-orig/src/.glHelp.cc.un~ and trackballs-1.1.4/src/.glHelp.cc.un~ differ
Binary files trackballs-1.1.4-orig/src/.menuMode.cc.un~ and trackballs-1.1.4/src/.menuMode.cc.un~ differ
diff -ruN trackballs-1.1.4-orig/src/glHelp.cc trackballs-1.1.4/src/glHelp.cc
--- trackballs-1.1.4-orig/src/glHelp.cc	2007-04-07 08:15:09.000000000 -0500
+++ trackballs-1.1.4/src/glHelp.cc	2018-04-29 20:29:16.000000000 -0500
@@ -129,7 +129,7 @@
   // why is this removed? should it not be done???
   glDeleteTextures(1,&texture);  
 }
-inline Real abs(Real v) {return v>0.0?v:-v;}
+
 double mousePointerPhase=0.0;
 
 void tickMouse(Real td) {
diff -ruN trackballs-1.1.4-orig/src/menuMode.cc trackballs-1.1.4/src/menuMode.cc
--- trackballs-1.1.4-orig/src/menuMode.cc	2007-04-07 08:16:26.000000000 -0500
+++ trackballs-1.1.4/src/menuMode.cc	2018-04-29 20:30:09.000000000 -0500
@@ -163,7 +163,6 @@
   case MENU_EDITOR: GameMode::activate(EditMode::editMode); break;
   }
 }
-inline Real abs(Real v) {return v>0.0?v:-v;}
 
 void MenuMode::idle(Real td) {
   int w,h,i,x,y;
