diff -u --unidirectional eject-1.0.orig/README eject-1.0/README
--- eject-1.0.orig/README	Wed Dec 31 16:00:00 1969
+++ eject-1.0/README	Sat Aug 24 18:05:08 2002
@@ -0,0 +1,6 @@
+eject is a trivial script to emulate part of the functionality of
+Jeff Tranter's eject for Linux. Call it as:
+
+eject /dev/disk2
+eject             # Ejects /dev/disk1
+eject /dev/cdrom  # Ejects /dev/disk1
diff -u --unidirectional eject-1.0.orig/eject eject-1.0/eject
--- eject-1.0.orig/eject	Wed Dec 31 16:00:00 1969
+++ eject-1.0/eject	Sat Aug 24 18:07:32 2002
@@ -0,0 +1,6 @@
+#!/usr/bin/perl
+
+$device = $ARGV[0] || "/dev/disk1";
+$device =~ s/cdrom/disk1/;
+
+exec "/usr/bin/hdiutil eject $device";
