diff -uNr findutils-4.2.20/findutils.cron findutils-4.2.20-new/findutils.cron
--- findutils-4.2.20/findutils.cron	1969-12-31 19:00:00.000000000 -0500
+++ findutils-4.2.20-new/findutils.cron	2005-03-21 10:01:54.000000000 -0500
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+test -x @PREFIX@/bin/updatedb || exit 0
+cd / && @PREFIX@/bin/updatedb
+
+exit 0
diff -uNr findutils-4.2.20/gnulib/lib/regex.c findutils-4.2.20-new/gnulib/lib/regex.c
--- findutils-4.2.20/gnulib/lib/regex.c	2005-01-24 12:10:02.000000000 -0500
+++ findutils-4.2.20-new/gnulib/lib/regex.c	2005-03-21 10:01:54.000000000 -0500
@@ -7963,7 +7963,7 @@
 
 /* POSIX.2 functions.  Don't define these for Emacs.  */
 
-#ifndef emacs
+#if !defined(emacs) && !defined(FINK)
 
 /* regcomp takes a regular expression as a string and compiles it.
 
diff -uNr findutils-4.2.20/lib/listfile.c findutils-4.2.20-new/lib/listfile.c
--- findutils-4.2.20/lib/listfile.c	2004-11-01 04:13:38.000000000 -0500
+++ findutils-4.2.20-new/lib/listfile.c	2005-03-21 10:10:20.000000000 -0500
@@ -89,10 +89,6 @@
 #define S_ISLNK(m) (((m) & S_IFMT) == S_IFLNK)
 #endif
 
-#if defined(S_ISLNK)
-int readlink ();
-#endif
-
 /* Get or fake the disk device blocksize.
    Usually defined by sys/param.h (if at all).  */
 #ifndef DEV_BSIZE
