diff -Naur config.h.orig config.h
--- fplan-1.4.2/config.h.orig       Sun Dec 21 08:29:54 2003
+++ fplan-1.4.2/config.h    Sun Dec 21 08:31:16 2003
@@ -62,7 +62,7 @@
 #if defined(OS2) || defined(MSDOS)
 #define DEFAULT_COMMON_DBDIR	"\\fplan\\commondb"
 #else
-#define DEFAULT_COMMON_DBDIR	"/usr/share/fplan"
+#define DEFAULT_COMMON_DBDIR	"@PREFIX@/share/fplan"
 #endif
 #endif

diff -Naur Makefile.orig Makefile
--- fplan-1.4.2/Makefile.orig	Thu Feb  5 18:09:44 2004
+++ fplan-1.4.2/Makefile	Thu Feb  5 18:13:18 2004
@@ -57,32 +57,33 @@
 # Define which graphics implementation to use. Currently, the options are
 # -DGFX_GNOME for Gnome, or -DGFX_XVIEW for XView, or empty for no previewer
 # under OS2, DOS.
-#GFX_OPTIONS = -DGFX_GNOME
-GFX_OPTIONS = -DGFX_XVIEW
+GFX_OPTIONS = -DGFX_GNOME
+#GFX_OPTIONS = -DGFX_XVIEW
 #GFX_OPTIONS =
 
 # ---------------------------------------------------------------------------
 # Define which directories contain the graphics header files. The lines
 # below are for the Gnome, XView and no previewer cases;
-#GFX_INCLUDE = $(shell gnome-config --cflags glib gtk gnome gnomeui)
-GFX_INCLUDE = -I/usr/openwin/include
+GFX_INCLUDE = $(shell gnome-config --cflags glib gtk gnome gnomeui)
+#GFX_INCLUDE = -I/usr/openwin/include
 #GFX_INCLUDE = 
 
 # ---------------------------------------------------------------------------
 # Define which graphics libraries are required when linking the executable.
 # The lines below are for the Gnome, XView, and no previewer cases;
-#GFX_LIBS = $(shell gnome-config --libs glib gtk gnome gnomeui)
-GFX_LIBS = -L/usr/openwin/lib -lxview -lolgx -L/usr/X11R6/lib -lX11
+GFX_LIBS = $(shell gnome-config --libs glib gtk gnome gnomeui)
+#GFX_LIBS = -L/usr/openwin/lib -lxview -lolgx -L/usr/X11R6/lib -lX11
 #GFX_LIBS =
 
 # ---------------------------------------------------------------------------
 # Define any additional flags required by the C compiler during the
 # compilation phase (such as flags for debugging or optimization,
 # include -DOS2 when building under OS2/EMX, and -DDOS under MS-DOS)
-CCOPTS = -O2 -m486 -fomit-frame-pointer
+#CCOPTS = -O2 -m486 -fomit-frame-pointer
 #CCOPTS = -g -Wall
 #CCOPTS = -O -DOS2
 #CCOPTS = -O -DMSDOS
+CCOPTS = -no-cpp-precomp -I@PREFIX@/include
 
 # ---------------------------------------------------------------------------
 # Define any additional flags required by the loader for the link phase,
@@ -107,17 +108,17 @@
 
 # ---------------------------------------------------------------------------
 # Define where 'make install' should put the executable and man pages
-BINDIR = /usr/bin
+BINDIR = 
 
 # ---------------------------------------------------------------------------
 # Define where 'make install-dbs' should put the databases (be *sure*
 # this matches with the value assigned to FPLAN_PUBLIC in config.h)
-DBSDIR = /usr/share/fplan
+DBSDIR = @PREFIX@/share/fplan
 
 # ---------------------------------------------------------------------------
 # Define where 'make install-man' should put the unformatted man pages
 # and the formatted and plain text documentation files respectively
-MANDIR = /usr/man
+MANDIR = 
 
 # ---------------------------------------------------------------------------
 # Define the command to be used by 'make doc' to create formatted man
diff -Naur fplan-1.4.2.orig/compute.c fplan-1.4.2/compute.c
--- fplan-1.4.2.orig/compute.c	1999-04-20 02:18:22.000000000 -0400
+++ fplan-1.4.2/compute.c	2006-02-01 13:48:42.000000000 -0500
@@ -568,6 +568,7 @@
 	    }
 	    break;
 	 default:
+	    break;
 	 }
       }
    }
diff -Naur fplan-1.4.2.orig/output_text.c fplan-1.4.2/output_text.c
--- fplan-1.4.2.orig/output_text.c	1999-04-20 02:18:48.000000000 -0400
+++ fplan-1.4.2/output_text.c	2006-02-01 13:49:28.000000000 -0500
@@ -346,6 +346,7 @@
       break;
    default:
       put_string (out_fp, fw, headers[(int) cell.kind]);
+      break;
    }
 }
 

