diff -ru latexmk-307a/latexmk.pl latexmk-307a-patched/latexmk.pl
--- latexmk-307a/latexmk.pl	2004-06-02 20:55:43.000000000 +0200
+++ latexmk-307a-patched/latexmk.pl	2005-07-24 18:22:17.000000000 +0200
@@ -1,9 +1,8 @@
 eval '(exit $?0)' && eval 'exec perl -x -S $0 ${1+"$@"}' && 
 eval 'exec perl -x -S  $0 $argv:q'
 if 0;
+#!@PREFIX@/bin/perl -w
 #!/usr/bin/perl -w
-#!/opt/local/bin/perl -w
-#!/usr/local/bin/perl -w
 
 #================= TO DO ================
 #
@@ -537,13 +536,11 @@
     ## /usr/local/lib/latexmk/LatexMk is put in the list for
     ## compatibility with older versions of latexmk.
     @rc_system_files = 
-     ( '/opt/local/share/latexmk/LatexMk', 
-       '/usr/local/share/latexmk/LatexMk',
-       '/usr/local/lib/latexmk/LatexMk' );
+     ( '@PREFIX@/share/latexmk/LatexMk' );
 
     $search_path_separator = ':';  # Separator of elements in search_path
     ## default document processing programs.
-    $pdf_previewer = 'start acroread';
+    $pdf_previewer = 'start open';
     ## The following are corrects for the SUNs at phys.psu.edu:
     #$ps_previewer  = 'start ghostview';
     #$ps_previewer_landscape  = 'start ghostview -swap';
@@ -560,10 +557,10 @@
     #         cannot be updated.  (acroread under MSWIN)
     $dvi_update_method = 2;  # xdvi responds to SIGUSR1 to update
     $ps_update_method = 0;   # gv -watch watches the ps file
-    $pdf_update_method = 1;  # acroread under unix needs manual update
+    $pdf_update_method = 0;  # acroread under unix needs manual update
     $lpr = 'lpr';         # Assume lpr command prints postscript files correctly
     $lpr_dvi  = 'NONE lpr_dvi';# Use NONE as flag that I am not implementing this
-    $lpr_pdf  = 'NONE lpr_pdf';# Use NONE as flag that I am not implementing this
+    $lpr_pdf  = 'lpr';# Use NONE as flag that I am not implementing this
     # The $pscmd below holds a command to list running processes.  It
     # is used to find the process ID of the viewer looking at the
     # current output file.  The output of the command must include the
@@ -580,8 +577,8 @@
     # Use -u $ENV{CMD} to get all processes started by current user (not just
     #   those associated with current terminal), but none of other users' 
     #   processes. 
-    $pscmd = "ps -f -u $ENV{USER}"; 
-    $pid_position = 1; # offset of PID in output of pscmd; first item is 0.  
+    $pscmd = "ps -ww -u $ENV{USER}"; 
+    $pid_position = 2; # offset of PID in output of pscmd; first item is 0.  
     if ( $^O eq "linux" ) {
         # Ps on Redhat (at least v. 7.2) appears to truncate its output
         #    at 80 cols, so that a long command string is truncated.
