--- osxutils-1.6/src/fileinfo.c.bak	2005-04-16 12:39:55.000000000 -0700
+++ osxutils-1.6/src/fileinfo.c	2005-04-16 13:06:48.000000000 -0700
@@ -199,6 +199,15 @@
 
 
 
+OSErr RetrieveStatData (FileInfoStruct *file);
+OSErr RetrieveFileInfo (FileInfoStruct *file);
+OSErr ProcessFinderInfo (FileInfoStruct *file);
+char* GetFileNameFromPath (char *name);
+static OSStatus FSMakePath(FSRef fileRef, UInt8 *path, UInt32 maxPathSize);
+static char* GetSizeString( UInt64 size, short sizeFormat);
+OSErr GetDateTimeStringFromUTCDateTime (UTCDateTime *utcDateTime, char *dateTimeString);
+static short GetLabelNumber (short flags);
+
 void OutputFileInfo (char *path)
 {
 	short				i = 0;
