diff -ruN xmltv-0.5.28.orig/Makefile.PL xmltv-0.5.28/Makefile.PL
--- xmltv-0.5.28.orig/Makefile.PL	Sun Feb  1 14:42:30 2004
+++ xmltv-0.5.28/Makefile.PL	Mon Feb  2 13:49:24 2004
@@ -296,16 +296,16 @@
        exes => [ 'grab/dk/tv_grab_dk' ],
        prereqs => { 'HTML::TreeBuilder' => 0 } },
 
-     { name => 'tv_check',
-       blurb => 'Program to report exceptions and changes in a schedule',
-       exes => [ 'choose/tv_check/tv_check' ],
-       docs => [ qw(choose/tv_check/README.tv_check
-		    choose/tv_check/tv_check_doc.html
-		    choose/tv_check/tv_check_doc.jpg
-		   ) ],
-       prereqs => { 'Tk' => 0,
-		    'Tk::TableMatrix' => 0,
-		  } },
+#     { name => 'tv_check',
+#       blurb => 'Program to report exceptions and changes in a schedule',
+#       exes => [ 'choose/tv_check/tv_check' ],
+#       docs => [ qw(choose/tv_check/README.tv_check
+#		    choose/tv_check/tv_check_doc.html
+#		    choose/tv_check/tv_check_doc.jpg
+#		   ) ],
+#       prereqs => { 'Tk' => 0,
+#		    'Tk::TableMatrix' => 0,
+#		  } },
 
      { name => 'tv_pick_cgi',
        blurb => 'CGI program to filter listings (to install manually)',
