--- tests/test_urldispatch.py.orig	2025-02-09 10:44:20.000000000 -0600
+++ tests/test_urldispatch.py	2025-02-09 10:44:59.000000000 -0600
@@ -351,7 +351,7 @@
     """Test that static paths are expanded and absolute."""
     res = router.add_static("/", "~/..")
     directory = str(res.get_info()["directory"])
-    assert directory == str(pathlib.Path.home().parent)
+    assert directory == str(pathlib.Path.home().resolve(strict=True).parent)
 
 
 def test_add_static(router) -> None:
