diff -ru cffi-1.3.1.orig/testing/cffi0/test_verify.py cffi-1.3.1/testing/cffi0/test_verify.py
--- cffi-1.3.1.orig/testing/cffi0/test_verify.py	2015-11-18 09:47:13.000000000 -0500
+++ cffi-1.3.1/testing/cffi0/test_verify.py	2015-11-19 08:44:03.000000000 -0500
@@ -15,7 +15,7 @@
     if (sys.platform == 'darwin' and
           [int(x) for x in os.uname()[2].split('.')] >= [11, 0, 0]):
         # assume a standard clang or gcc
-        extra_compile_args = ['-Werror', '-Wall', '-Wextra', '-Wconversion']
+        extra_compile_args = ['-Werror', '-Wall', '-Wextra', '-Wconversion', '-Wno-unreachable-code', '-Wno-shift-negative-value']
         # special things for clang
         extra_compile_args.append('-Qunused-arguments')
     else:
diff -ru cffi-1.3.1.orig/testing/cffi1/test_verify1.py cffi-1.3.1/testing/cffi1/test_verify1.py
--- cffi-1.3.1.orig/testing/cffi1/test_verify1.py	2015-11-18 09:47:13.000000000 -0500
+++ cffi-1.3.1/testing/cffi1/test_verify1.py	2015-11-19 08:44:25.000000000 -0500
@@ -15,7 +15,7 @@
     if (sys.platform == 'darwin' and
           [int(x) for x in os.uname()[2].split('.')] >= [11, 0, 0]):
         # assume a standard clang or gcc
-        extra_compile_args = ['-Werror', '-Wall', '-Wextra', '-Wconversion']
+        extra_compile_args = ['-Werror', '-Wall', '-Wextra', '-Wconversion', '-Wno-unreachable-code', '-Wno-shift-negative-value']
         # special things for clang
         extra_compile_args.append('-Qunused-arguments')
     else:
@@ -2093,7 +2093,7 @@
         ffi1 = FFI()
         ffi1.cdef("int foo_verify_dlopen_flags_1;")
         sys.setdlopenflags(ffi1.RTLD_GLOBAL | ffi1.RTLD_NOW)
-        lib1 = ffi1.verify("int foo_verify_dlopen_flags_1;")
+        lib1 = ffi1.verify("int foo_verify_dlopen_flags_1;", extra_link_args=["-Wl,-flat_namespace"])
     finally:
         sys.setdlopenflags(old)
 
@@ -2102,7 +2102,7 @@
     lib2 = ffi2.verify("""
         extern int foo_verify_dlopen_flags_1;
         static int *getptr(void) { return &foo_verify_dlopen_flags_1; }
-    """)
+    """, extra_link_args=["-Wl,-flat_namespace", "-Wl,-undefined,suppress"])
     p = lib2.getptr()
     assert ffi1.addressof(lib1, 'foo_verify_dlopen_flags_1') == p
 
