diff -Nurd Fiona-1.7.13.orig/setup.py Fiona-1.7.13/setup.py
--- Fiona-1.7.13.orig/setup.py	2018-07-17 14:09:40.000000000 -0400
+++ Fiona-1.7.13/setup.py	2018-08-05 01:05:53.000000000 -0400
@@ -180,6 +180,9 @@
     libraries=libraries,
     extra_link_args=extra_link_args,
 )
+ext_options_cpp = ext_options.copy()
+# GDAL 2.3+ requires C++11
+ext_options_cpp["extra_compile_args"] = ["-std=c++11"]
 
 # Define the extension modules.
 ext_modules = []
@@ -207,7 +210,7 @@
 
     ext_modules = cythonize([
         Extension('fiona._geometry', ['fiona/_geometry.pyx'], **ext_options),
-        Extension('fiona._transform', ['fiona/_transform.pyx'], **ext_options),
+        Extension('fiona._transform', ['fiona/_transform.pyx'], **ext_options_cpp),
         Extension('fiona._crs', ['fiona/_crs.pyx'], **ext_options),
         Extension('fiona._drivers', ['fiona/_drivers.pyx'], **ext_options),
         Extension('fiona._err', ['fiona/_err.pyx'], **ext_options),
@@ -217,7 +220,7 @@
 # If there's no manifest template, as in an sdist, we just specify .c files.
 elif "clean" not in sys.argv:
     ext_modules = [
-        Extension('fiona._transform', ['fiona/_transform.cpp'], **ext_options),
+        Extension('fiona._transform', ['fiona/_transform.cpp'], **ext_options_cpp),
         Extension('fiona._geometry', ['fiona/_geometry.c'], **ext_options),
         Extension('fiona._crs', ['fiona/_crs.c'], **ext_options),
         Extension('fiona._drivers', ['fiona/_drivers.c'], **ext_options),
