diff -Nurd gdal-2.4.4.orig/swig/python/setup.py gdal-2.4.4/swig/python/setup.py
--- gdal-2.4.4.orig/swig/python/setup.py	2020-01-08 11:50:09.000000000 +0000
+++ gdal-2.4.4/swig/python/setup.py	2020-12-02 13:22:40.000000000 +0000
@@ -60,8 +60,8 @@
 # (may be overridden with setup.cfg or command line switches).
 # ---------------------------------------------------------------------------
 
-include_dirs = ['../../port', '../../gcore', '../../alg', '../../ogr/', '../../ogr/ogrsf_frmts', '../../gnm', '../../apps']
-library_dirs = ['../../.libs', '../../']
+include_dirs = []
+library_dirs = []
 libraries = ['gdal']
 
 
@@ -265,12 +265,14 @@
             self.include_dirs = include_dirs
         # Needed on recent MacOSX
         elif isinstance(self.include_dirs, str) and sys.platform == 'darwin':
-            self.include_dirs += ':' + ':'.join(include_dirs)
+            if include_dirs:
+                self.include_dirs += ':' + ':'.join(include_dirs)
         if self.library_dirs is None:
             self.library_dirs = library_dirs
         # Needed on recent MacOSX
         elif isinstance(self.library_dirs, str) and sys.platform == 'darwin':
-            self.library_dirs += ':' + ':'.join(library_dirs)
+            if library_dirs:
+                self.library_dirs += ':' + ':'.join(library_dirs)
         if self.libraries is None:
             if self.get_compiler() == 'msvc':
                 libraries.remove('gdal')
