diff -ru python-hglib-1.7.orig/tests/test-summary.py python-hglib-1.7/tests/test-summary.py
--- python-hglib-1.7.orig/tests/test-summary.py	2015-03-13 17:46:43.000000000 -0400
+++ python-hglib-1.7/tests/test-summary.py	2015-08-26 17:38:23.000000000 -0400
@@ -18,7 +18,8 @@
         d = {b('parent') : [(0, node[:12], b('tip'), b('first'))],
              b('branch') : b('default'),
              b('commit') : True,
-             b('update') : 0}
+             b('update') : 0,
+             b('phases') : b('1 draft')}
 
         self.assertEquals(self.client.summary(), d)
 
@@ -30,7 +31,8 @@
         d = {b('parent') : [(0, node[:12], b('tip'), b('first'))],
              b('branch') : b('default'),
              b('commit') : False,
-             b('update') : 0}
+             b('update') : 0,
+             b('phases') : b('1 draft')}
 
         self.assertEquals(self.client.summary(), d)
 
@@ -44,7 +46,8 @@
         d = {b('parent') : [(0, node[:12], None, b('first'))],
              b('branch') : b('default'),
              b('commit') : True,
-             b('update') : 1}
+             b('update') : 1,
+             b('phases') : b('2 draft')}
 
         self.assertEquals(self.client.summary(), d)
 
@@ -85,6 +88,7 @@
         rev, node = other.commit(b('second in other'))
 
         d[b('remote')] = (1, 1, 1, 1)
+        d[b('phases')] = b('1 draft')
         if self.client.version < (2, 0, 0):
             tags = b('tip bmother')
         else:
@@ -110,6 +114,7 @@
                          (1, node1[:12], None, b('second'))],
              b('branch') : b('default'),
              b('commit') : False,
-             b('update') : 0}
+             b('update') : 0,
+             b('phases') : b('3 draft')}
 
         self.assertEquals(self.client.summary(), d)
