diff -ruN imagecodecs-2024.12.30-orig/setup.py imagecodecs-2024.12.30/setup.py
--- imagecodecs-2024.12.30-orig/setup.py	2024-12-22 14:14:22
+++ imagecodecs-2024.12.30/setup.py	2025-11-23 21:10:58
@@ -563,15 +563,17 @@
         EXTENSIONS['jpegxr']['libraries'] = ['jpegxr', 'jxrglue']
 
 
-def customize_build_macports(EXTENSIONS, OPTIONS):
-    """Customize build for MacPorts."""
+def customize_build_fink(EXTENSIONS, OPTIONS):
+    """Customize build for Fink."""
 
+    del EXTENSIONS['aec']
+
     del EXTENSIONS['apng']
     del EXTENSIONS['avif']
     del EXTENSIONS['blosc2']
     del EXTENSIONS['brunsli']
-    del EXTENSIONS['deflate']
-    del EXTENSIONS['heif']
+    #del EXTENSIONS['deflate']
+    #del EXTENSIONS['heif']
     # del EXTENSIONS['htj2k']
     del EXTENSIONS['jetraw']
     # del EXTENSIONS['jpegli']
@@ -591,19 +593,22 @@
     del EXTENSIONS['ultrahdr']
     del EXTENSIONS['zfp']
     del EXTENSIONS['zlibng']
+    del EXTENSIONS['zopfli']
 
     EXTENSIONS['jpeg8']['sources'] = []  # use libjpeg-turbo 3
 
-    EXTENSIONS['szip']['library_dirs'] = ['%PREFIX%/lib/libaec/lib']
-    EXTENSIONS['szip']['include_dirs'] = ['%PREFIX%/lib/libaec/include']
-    EXTENSIONS['aec']['library_dirs'] = ['%PREFIX%/lib/libaec/lib']
-    EXTENSIONS['aec']['include_dirs'] = ['%PREFIX%/lib/libaec/include']
-    EXTENSIONS['gif']['include_dirs'] = ['%PREFIX%/include/giflib5']
+    #EXTENSIONS['szip']['library_dirs'] = ['%PREFIX%/lib/libaec/lib']
+    #EXTENSIONS['szip']['include_dirs'] = ['%PREFIX%/lib/libaec/include']
+    #EXTENSIONS['aec']['library_dirs'] = ['%PREFIX%/lib/libaec/lib']
+    #EXTENSIONS['aec']['include_dirs'] = ['%PREFIX%/lib/libaec/include']
+    #EXTENSIONS['gif']['include_dirs'] = ['%PREFIX%/include/giflib5']
+    EXTENSIONS['jpeg8']['library_dirs'] = ['@FINK_PREFIX@/lib/libjpeg8.2-turbo/lib']
+    EXTENSIONS['jpeg8']['include_dirs'] = ['@FINK_PREFIX@/lib/libjpeg8.2-turbo/include']
     EXTENSIONS['jpeg2k']['include_dirs'].extend(
         (
-            '%PREFIX%/include/openjpeg-2.3',
-            '%PREFIX%/include/openjpeg-2.4',
-            '%PREFIX%/include/openjpeg-2.5',
+            '@FINK_PREFIX@/include/openjpeg-2.3',
+            '@FINK_PREFIX@/include/openjpeg-2.4',
+            '@FINK_PREFIX@/include/openjpeg-2.5',
         )
     )
 
@@ -649,8 +654,8 @@
             customize_build = customize_build_cibuildwheel
         elif os.environ.get('CONDA_BUILD', ''):
             customize_build = customize_build_condaforge
-        elif shutil.which('port'):
-            customize_build = customize_build_macports
+        elif shutil.which('fink'):
+            customize_build = customize_build_fink
         elif os.name == 'nt' and 'GCC' in sys.version:
             customize_build = customize_build_mingw
         else:
