diff --git a/docs/source/conf.py b/docs/source/conf.py
--- a/docs/source/conf.py       2018-09-14 22:38:33.000000000 +0200
+++ b/docs/source/conf.py       2019-05-29 23:55:00.000000000 +0200
@@ -43,10 +43,10 @@
 ]
 
 intersphinx_mapping = {
-    'ipython': ('http://ipython.org/ipython-doc/dev/', None),
-    'nbconvert': ('https://nbconvert.readthedocs.io/en/latest/', None),
-    'nbformat': ('https://nbformat.readthedocs.io/en/latest/', None),
-    'jupyter': ('https://jupyter.readthedocs.io/en/latest/', None),
+    'nbconvert': ('@PREFIX@/share/doc/nbconvert-py@PY_PKG@/html', None),
+    'nbformat': ('@PREFIX@/share/doc/nbformat-py@PY_PKG@/html', None),
+    'jupyterclient': ('@PREFIX@/share/doc/jupyter-client-py@PY_PKG@/html', None),
+    'jupyter': ('@PREFIX@/share/doc/jupyter-core-py@PY_PKG@/html', None),
 }

