diff --git a/docs/source/conf.py b/docs/source/conf.py
--- a/docs/source/conf.py       2018-05-10 20:32:43.000000000 +0200
+++ b/docs/source/conf.py       2019-05-29 21:55:00.000000000 +0200
@@ -324,6 +324,6 @@
 
 # Example configuration for intersphinx: refer to the Python standard library.
 intersphinx_mapping = {
-    'python': ('https://docs.python.org/3.5', None),
-    'jinja': ('http://jinja.pocoo.org/docs/dev', None),
+    'jupyterclient': ('@PREFIX@/share/doc/jupyter-client-py@PY_PKG@/html', None),
+    'jupyter': ('@PREFIX@/share/doc/jupyter-core-py@PY_PKG@/html', None),
 }
