diff -Nurd pathlib2-2.3.6.orig/tests/test_pathlib2.py pathlib2-2.3.6/tests/test_pathlib2.py
--- pathlib2-2.3.6.orig/tests/test_pathlib2.py	2021-07-05 07:53:43.000000000 -0400
+++ pathlib2-2.3.6/tests/test_pathlib2.py	2022-01-25 18:55:56.000000000 -0500
@@ -45,6 +45,17 @@
 except ImportError:
     from test import test_support as support  # type: ignore
 
+# shim for API splitting into several sub-modules
+if sys.version_info < (3, 10):
+    def import_module(s):
+        support.import_module(s)
+else:
+    from test.support import os_helper as support
+    from test.support import import_helper
+    def import_module(s):
+        import_helper.import_module(s)
+        
+
 android_not_root = getattr(support, "android_not_root", False)
 
 try:
@@ -2303,7 +2314,7 @@
                          'pwd module does not expose getpwall()')
     def test_expanduser(self):
         P = self.cls
-        support.import_module('pwd')
+        import_module('pwd')
         import pwd
         pwdent = pwd.getpwuid(os.getuid())
         username = pwdent.pw_name
