diff -ruN pexpect-4.8.0-orig/tests/test_misc.py pexpect-4.8.0/tests/test_misc.py
--- pexpect-4.8.0-orig/tests/test_misc.py	2020-01-17 11:08:15.000000000 -0600
+++ pexpect-4.8.0/tests/test_misc.py	2020-11-26 05:57:20.000000000 -0600
@@ -74,25 +74,6 @@
         remaining = child.read().replace(_CAT_EOF, b'')
         self.assertEqual(remaining, b'abc\r\n')
 
-    def test_read_poll(self):
-        " Test spawn.read by calls of various size. "
-        child = pexpect.spawn('cat', use_poll=True)
-        child.sendline("abc")
-        child.sendeof()
-        self.assertEqual(child.read(0), b'')
-        self.assertEqual(child.read(1), b'a')
-        self.assertEqual(child.read(1), b'b')
-        self.assertEqual(child.read(1), b'c')
-        self.assertEqual(child.read(2), b'\r\n')
-        remaining = child.read().replace(_CAT_EOF, b'')
-        self.assertEqual(remaining, b'abc\r\n')
-
-    def test_read_poll_timeout(self):
-        " Test use_poll properly times out "
-        child = pexpect.spawn('sleep 5', use_poll=True)
-        with self.assertRaises(pexpect.TIMEOUT):
-            child.expect(pexpect.EOF, timeout=1)
-
     def test_readline_bin_echo(self):
         " Test spawn('echo'). "
         # given,
