diff -ruN pygraphviz-1.14-orig/setup.py pygraphviz-1.14/setup.py
--- pygraphviz-1.14-orig/setup.py	2024-09-29 13:31:00
+++ pygraphviz-1.14/setup.py	2026-02-03 06:31:54
@@ -10,8 +10,8 @@
         Extension(
             name="pygraphviz._graphviz",
             sources=["pygraphviz/graphviz_wrap.c"],
-            include_dirs=[],
-            library_dirs=[],
+            include_dirs=["@FINK_PREFIX@/include"],
+            library_dirs=["@FINK_PREFIX@/lib/graphviz-2.38"],
             # cdt does not link to cgraph, whereas cgraph links to cdt.
             # thus, cdt needs to come first in the library list to be sure
             # that both libraries are linked in the final built .so (if cgraph
