diff -Nurd -x'*~' egenix-mx-base-3.2.4.orig/mxSetup.py egenix-mx-base-3.2.4/mxSetup.py
--- egenix-mx-base-3.2.4.orig/mxSetup.py	2012-04-23 12:57:37.000000000 -0400
+++ egenix-mx-base-3.2.4/mxSetup.py	2012-08-11 05:42:31.000000000 -0400
@@ -993,22 +993,18 @@
 # Default paths for additional library and include file search (in
 # addition to the standard system directories above); these are always
 # added to the compile and link commands by mxSetup per default.
-LIBPATH = build_path(['/usr/local/lib',
-                      '/opt/local/lib',
+LIBPATH = build_path([
                       os.path.join(sys.prefix, 'lib')])
-INCLPATH = build_path(['/usr/local/include',
-                       '/opt/local/include',
+INCLPATH = build_path([
                        os.path.join(sys.prefix, 'include')])
 
 # Add 32- or 64-bit dirs if needed by the Python version
 if sys.maxint > 2147483647L:
     # 64-bit build
     STDLIBPATH.extend(['/usr/lib64', '/opt/lib64'])
-    LIBPATH.extend(['/usr/local/lib64', '/opt/local/lib64'])
 else:
     # 32-bit build
     STDLIBPATH.extend(['/usr/lib32', '/opt/lib32'])
-    LIBPATH.extend(['/usr/local/lib32', '/opt/local/lib32'])
 
 # Additional paths to scan in order to find third party libs and
 # headers; these are used by mx_autoconf.find_*_file() APIs.
