diff --git a/tox.ini b/tox.ini
--- a/tox.ini	2018-02-22 12:08:28.000000000 +0100
+++ b/tox.ini	2018-05-16 22:15:56.000000000 +0200
@@ -20,7 +20,8 @@
 deps =
 extras = testing
 changedir = {toxinidir}/tests
-commands = pytest {posargs:--cov="{envsitepackagesdir}/tox" --cov-config="{toxinidir}/tox.ini"  --timeout=180 . -n {env:PYTEST_XDIST_PROC_NR:auto} --junitxml={toxworkdir}/test-results.{envname}.xml}
+commands = pytest-PY_RAW {posargs:--cov="{envsitepackagesdir}/tox" --cov-config="{toxinidir}/tox.ini"  --timeout=180 . -n {env:PYTEST_XDIST_PROC_NR:auto} --junitxml={toxworkdir}/test-results.{envname}.xml}
+whitelist_externals: pytest-PY_RAW
 
 [testenv:docs]
 description = invoke sphinx-build to build the HTML docs and check that all links are valid
diff --git a/tests/test_z_cmdline.py b/tests/test_z_cmdline.py
--- a/tests/unit/test_z_cmdline.py	2018-03-24 11:05:06.000000000 +0100
+++ b/tests/unit/test_z_cmdline.py	2018-05-16 22:07:44.000000000 +0200
@@ -447,7 +447,7 @@
             changedir=tests
             commands= pytest --basetemp={envtmpdir} \
                               --junitxml=junit-{envname}.xml
-            deps=pytest
+            sitepackages=True
         """,
         },
     )
@@ -565,7 +565,7 @@
             changedir=tests
             commands=
                 pytest --basetemp={envtmpdir} --junitxml=junit-{envname}.xml []
-            deps=pytest
+            sitepackages=True
         """,
         },
     )
@@ -852,15 +852,17 @@
     )
 
 
+@pytest.mark.skipif(os.environ.get('DEB_SKIP_TOX_TESTS'), reason='Use DEP-8')
 def test_tox_console_script(initproj):
     initproj("help", filedefs={"tox.ini": ""})
-    result = subprocess.check_call(["tox", "--help"])
+    result = subprocess.check_call(["tox-pyPY_PKG", "--help"])
     assert result == 0
 
 
+@pytest.mark.skipif(os.environ.get('DEB_SKIP_TOX_TESTS'), reason='Use DEP-8')
 def test_tox_quickstart_script(initproj):
     initproj("help", filedefs={"tox.ini": ""})
-    result = subprocess.check_call(["tox-quickstart", "--help"])
+    result = subprocess.check_call(["tox-quickstart-pyPY_PKG", "--help"])
     assert result == 0
 
 
