--- sunclock-3.46/Makefile.noimake	Wed Jun 27 08:33:26 2001
+++ sunclock-3.46-patched/Makefile.noimake	Wed May 29 15:37:06 2002
@@ -29,12 +29,16 @@
 #LFLAGS=-L/usr/openwin/lib
 
 # For OpenServer
-CFLAGS=-I/usr/include/X11 -DMAXPATHLEN=1024 -O -DSHAREDIR=\"$(DESTDIR)$(SHAREDIR)"
-LFLAGS=-L/usr/lib/X11 -lXpm -ljpeg -lsocket
+#CFLAGS=-I/usr/include/X11 -DMAXPATHLEN=1024 -O -DSHAREDIR=\"$(DESTDIR)$(SHAREDIR)"
+#LFLAGS=-L/usr/lib/X11 -lXpm -ljpeg -lsocket
 
 # For Linux
-CFLAGS=-I/usr/include/X11 -O -Wall -DSHAREDIR=\"$(DESTDIR)$(SHAREDIR)\"
-LFLAGS=-L/usr/X11/lib -lXpm -ljpeg
+#CFLAGS=-I/usr/include/X11 -O -Wall -DSHAREDIR=\"$(DESTDIR)$(SHAREDIR)\"
+#LFLAGS=-L/usr/X11/lib -lXpm -ljpeg
+
+# For Darwin
+CFLAGS = -I/usr/X11R6/include -O -Wall -DNEW_CTIME -DNO_MALLOC_H -DSHAREDIR=\"$(SHAREDIR)\" $(EXTRA_CFLAGS) 
+LFLAGS	=-L/usr/X11R6/lib -lXpm -ljpeg $(EXTRA_LDFLAGS) 
 
 OFILES=sunclock.o astro.o widgets.o tildepath.o dirlist.o \
        readjpeg.o readvmf.o readxpm.o
@@ -61,7 +65,7 @@
 	cat WARNING
 
 install.man:
-	cp -f sunclock.man $(DESTDIR)$(MANDIR)/sunclock.1x
+	cp -f sunclock.man $(DESTDIR)$(MANDIR)/sunclock.man
 	if [ -r $(DESTDIR)$(MANDIR)/sunclock.1x.gz ] ; \
 	   then gzip -f $(DESTDIR)$(MANDIR)/sunclock.1x ; fi
 
--- sunclock-3.46/sunclock.c	Wed Jul  4 14:27:14 2001
+++ sunclock-3.46-patched/sunclock.c	Wed May 29 15:51:25 2002
@@ -2178,7 +2178,7 @@
         char            s[128];
         char            slat[20], slon[20], slatp[20], slonp[20];
         double          dist;
-#ifdef NEW_CTIME
+#if 0
         struct timeb            tp;
 
         if (ftime(&tp) == -1) {
--- sunclock-3.46/tildepath.c	Fri Sep  1 08:34:51 2000
+++ sunclock-3.46-patched/tildepath.c	Wed May 29 15:18:41 2002
@@ -38,7 +38,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <pwd.h>
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <string.h>
 
 /*
