
--- emboss-2.4.1/jemboss/Makefile.in	Tue May 14 14:07:31 2002
+++ emboss-2.4.1-patched/jemboss/Makefile.in	Tue Jun 18 22:40:36 2002
@@ -396,12 +396,12 @@
 	fi
 
 install-exec-hook:
-	rm -f $(bindir)/runJemboss.csh; \
-	echo "#!/bin/csh" >> $(bindir)/runJemboss.csh; \
-	echo "setenv LD_LIBRARY_PATH $(libdir)" >> $(bindir)/runJemboss.csh; \
-	echo "setenv CLASSPATH $(prefix)/share/$(PACKAGE)/jemboss/lib/soap.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/xerces.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/mail.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/activation.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/jakarta-regexp-1.2.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/jalview.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/jcert.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/jnet.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/jsse.jar:." >> $(bindir)/runJemboss.csh; \
-	echo "(cd $(prefix)/share/$(PACKAGE)/jemboss; java org/emboss/jemboss/Jemboss &)" >> $(bindir)/runJemboss.csh; \
-	chmod 755 $(bindir)/runJemboss.csh;
+	rm -f $(DESTDIR)$(bindir)/runJemboss.csh; \
+	echo "#!/bin/csh" >> $(DESTDIR)$(bindir)/runJemboss.csh; \
+	echo "setenv LD_LIBRARY_PATH $(libdir)" >> $(DESTDIR)$(bindir)/runJemboss.csh; \
+	echo "setenv CLASSPATH $(prefix)/share/$(PACKAGE)/jemboss/lib/soap.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/xerces.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/mail.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/activation.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/jakarta-regexp-1.2.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/jalview.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/jcert.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/jnet.jar:$(prefix)/share/$(PACKAGE)/jemboss/lib/jsse.jar:." >> $(DESTDIR)$(bindir)/runJemboss.csh; \
+	echo "(cd $(prefix)/share/$(PACKAGE)/jemboss; java org/emboss/jemboss/Jemboss &)" >> $(DESTDIR)$(bindir)/runJemboss.csh; \
+	chmod 755 $(DESTDIR)$(bindir)/runJemboss.csh;
 
 # tar to pick up the other directories
 # then remove any CVS subdirectories
--- emboss-2.4.1/jemboss/resources/Makefile.in	Tue May 14 14:07:32 2002
+++ emboss-2.4.1-patched/jemboss/resources/Makefile.in	Wed Jun 19 04:37:55 2002
@@ -250,14 +250,14 @@
 
 install-exec-hook:
 	  @if test "${JAVA_OK}" = "yes" ; then \
-	          (if test -d $(prefix)/share/$(PACKAGE)/jemboss/resources; then \
+	          (if test -d $(DESTDIR)$(pkgdatadir); then \
 			echo "resources exists"; \
 		   else \
-			mkdir $(prefix)/share/$(PACKAGE)/jemboss/resources; \
+			mkdir $(DESTDIR)$(pkgdatadir); \
 		   fi); \
-	          rm -f $(prefix)/share/$(PACKAGE)/jemboss/resources/acdstore.jar; \
+	          rm -f $(DESTDIR)$(pkgdatadir)/acdstore.jar; \
 		  cd $(EMBOSS_TOP)/emboss/acd; \
-		  jar cvf $(prefix)/share/$(PACKAGE)/jemboss/resources/acdstore.jar *.acd; \
+		  jar cvf $(DESTDIR)$(pkgdatadir)/acdstore.jar *.acd; \
 	  fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
--- emboss-2.4.1/jemboss/org/emboss/jemboss/Makefile.in	Tue May 14 14:07:32 2002
+++ emboss-2.4.1-patched/jemboss/org/emboss/jemboss/Makefile.in	Wed Jun 19 03:49:47 2002
@@ -375,9 +375,9 @@
 	@if test "${JAVA_OK}" != "yes" ; then \
 	  echo "No class files to copy"; \
 	else \
-	  $(top_srcdir)/mkinstalldirs $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss; \
+	  $(top_srcdir)/mkinstalldirs $(DESTDIR)$(pkgdatadir); \
 	for i in *.class; do \
-	  ${INSTALL_DATA} $$i $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss; \
+	  ${INSTALL_DATA} $$i $(DESTDIR)$(pkgdatadir); \
 	done; \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
--- emboss-2.4.1/jemboss/org/emboss/jemboss/soap/Makefile.in	Tue May 14 14:07:32 2002
+++ emboss-2.4.1-patched/jemboss/org/emboss/jemboss/soap/Makefile.in	Wed Jun 19 03:52:11 2002
@@ -258,9 +258,9 @@
 	@if test "${JAVA_OK}" != "yes" ; then \
 	  echo "No class files to copy"; \
 	else \
-	  $(top_srcdir)/mkinstalldirs $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/soap; \
+	  $(top_srcdir)/mkinstalldirs $(DESTDIR)$(pkgdatadir); \
 	for i in *.class; do \
-	  ${INSTALL_DATA} $$i $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/soap; \
+	  ${INSTALL_DATA} $$i $(DESTDIR)$(pkgdatadir); \
 	done; \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
--- emboss-2.4.1/jemboss/org/emboss/jemboss/server/Makefile.in	Tue May 14 14:07:32 2002
+++ emboss-2.4.1-patched/jemboss/org/emboss/jemboss/server/Makefile.in	Wed Jun 19 03:51:27 2002
@@ -254,9 +254,9 @@
 	@if test "${JAVA_OK}" != "yes" ; then \
 	  echo "No class files to copy"; \
 	else \
-	  $(top_srcdir)/mkinstalldirs $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/server; \
+	  $(top_srcdir)/mkinstalldirs $(DESTDIR)$(pkgdatadir); \
 	  if test -f JembossServer.class ; then \
-	    ${INSTALL_DATA} *.class $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/server; \
+	    ${INSTALL_DATA} *.class $(DESTDIR)$(pkgdatadir); \
 	  fi \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
--- emboss-2.4.1/jemboss/org/emboss/jemboss/programs/Makefile.in	Tue May 14 14:07:32 2002
+++ emboss-2.4.1-patched/jemboss/org/emboss/jemboss/programs/Makefile.in	Wed Jun 19 03:50:47 2002
@@ -253,9 +253,9 @@
 	@if test "${JAVA_OK}" != "yes" ; then \
 	  echo "No class files to copy"; \
 	else \
-	  $(top_srcdir)/mkinstalldirs $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/programs; \
+	  $(top_srcdir)/mkinstalldirs $(DESTDIR)$(pkgdatadir); \
 	for i in *.class; do \
-	  ${INSTALL_DATA} $$i $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/programs; \
+	  ${INSTALL_DATA} $$i $(DESTDIR)$(pkgdatadir); \
 	done; \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
--- emboss-2.4.1/jemboss/org/emboss/jemboss/parser/Makefile.in	Tue May 14 14:07:32 2002
+++ emboss-2.4.1-patched/jemboss/org/emboss/jemboss/parser/Makefile.in	Wed Jun 19 03:50:10 2002
@@ -378,9 +378,9 @@
 	@if test "${JAVA_OK}" != "yes" ; then \
 	  echo "No class files to copy"; \
 	else \
-	  $(top_srcdir)/mkinstalldirs $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/parser; \
+	  $(top_srcdir)/mkinstalldirs $(DESTDIR)$(pkgdatadir); \
 	for i in *.class; do \
-	  ${INSTALL_DATA} $$i $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/parser; \
+	  ${INSTALL_DATA} $$i $(DESTDIR)$(pkgdatadir); \
 	done; \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
--- emboss-2.4.1/jemboss/org/emboss/jemboss/parser/acd/Makefile.in	Tue May 14 14:07:32 2002
+++ emboss-2.4.1-patched/jemboss/org/emboss/jemboss/parser/acd/Makefile.in	Wed Jun 19 03:50:27 2002
@@ -251,9 +251,9 @@
 	@if test "${JAVA_OK}" != "yes" ; then \
 	  echo "No class files to copy"; \
 	else \
-	  $(top_srcdir)/mkinstalldirs $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/parser/acd; \
+	  $(top_srcdir)/mkinstalldirs $(DESTDIR)$(pkgdatadir); \
 	for i in *.class; do \
-	  ${INSTALL_DATA} $$i $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/parser/acd; \
+	  ${INSTALL_DATA} $$i $(DESTDIR)$(pkgdatadir); \
 	done; \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
--- emboss-2.4.1/jemboss/org/emboss/jemboss/gui/Makefile.in	Tue May 14 14:07:32 2002
+++ emboss-2.4.1-patched/jemboss/org/emboss/jemboss/gui/Makefile.in	Wed Jun 19 03:48:56 2002
@@ -380,9 +380,9 @@
 	@if test "${JAVA_OK}" != "yes" ; then \
 	  echo "No class files to copy"; \
 	else \
-	  $(top_srcdir)/mkinstalldirs $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/gui; \
+	  $(top_srcdir)/mkinstalldirs $(DESTDIR)$(pkgdatadir); \
 	for i in *.class; do \
-	  ${INSTALL_DATA} $$i $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/gui; \
+	  ${INSTALL_DATA} $$i $(DESTDIR)$(pkgdatadir); \
 	done; \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
--- emboss-2.4.1/jemboss/org/emboss/jemboss/gui/startup/Makefile.in	Tue May 14 14:07:32 2002
+++ emboss-2.4.1-patched/jemboss/org/emboss/jemboss/gui/startup/Makefile.in	Wed Jun 19 04:11:03 2002
@@ -251,9 +251,9 @@
 	@if test "${JAVA_OK}" != "yes" ; then \
 	  echo "No class files to copy"; \
 	else \
-	  $(top_srcdir)/mkinstalldirs $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/gui/startup; \
+	  $(top_srcdir)/mkinstalldirs $(DESTDIR)$(pkgdatadir); \
 	for i in *.class; do \
-	  ${INSTALL_DATA} $$i $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/gui/startup; \
+	  ${INSTALL_DATA} $$i $(DESTDIR)$(pkgdatadir); \
 	done; \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
--- emboss-2.4.1/jemboss/org/emboss/jemboss/gui/sequenceChooser/Makefile.in	Tue May 14 14:07:32 2002
+++ emboss-2.4.1-patched/jemboss/org/emboss/jemboss/gui/sequenceChooser/Makefile.in	Wed Jun 19 04:10:24 2002
@@ -254,9 +254,9 @@
 	@if test "${JAVA_OK}" != "yes" ; then \
 	  echo "No class files to copy"; \
 	else \
-	  $(top_srcdir)/mkinstalldirs $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/gui/sequenceChooser; \
+	  $(top_srcdir)/mkinstalldirs $(DESTDIR)$(pkgdatadir); \
 	for i in *.class; do \
-	  ${INSTALL_DATA} $$i $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/gui/sequenceChooser; \
+	  ${INSTALL_DATA} $$i $(DESTDIR)$(pkgdatadir); \
 	done; \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
--- emboss-2.4.1/jemboss/org/emboss/jemboss/gui/form/Makefile.in	Tue May 14 14:07:32 2002
+++ emboss-2.4.1-patched/jemboss/org/emboss/jemboss/gui/form/Makefile.in	Wed Jun 19 03:48:09 2002
@@ -255,9 +255,9 @@
 	@if test "${JAVA_OK}" != "yes" ; then \
 	  echo "No class files to copy"; \
 	else \
-	  $(top_srcdir)/mkinstalldirs $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/gui/form; \
+	  $(top_srcdir)/mkinstalldirs $(DESTDIR)$(pkgdatadir); \
 	for i in *.class; do \
-	  ${INSTALL_DATA} $$i $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/gui/form; \
+	  ${INSTALL_DATA} $$i $(DESTDIR)$(pkgdatadir); \
 	done; \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
--- emboss-2.4.1/jemboss/org/emboss/jemboss/gui/filetree/Makefile.in	Tue May 14 14:07:32 2002
+++ emboss-2.4.1-patched/jemboss/org/emboss/jemboss/gui/filetree/Makefile.in	Wed Jun 19 03:25:48 2002
@@ -254,9 +254,9 @@
 	@if test "${JAVA_OK}" != "yes" ; then \
 	  echo "No class files to copy"; \
 	else \
-	$(top_srcdir)/mkinstalldirs $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/gui/filetree; \
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(pkgdatadir); \
 	for i in *.class; do \
-	  ${INSTALL_DATA} $$i $(prefix)/share/$(PACKAGE)/jemboss/org/emboss/jemboss/gui/filetree; \
+	  ${INSTALL_DATA} $$i $(DESTDIR)$(pkgdatadir); \
 	done; \
 	fi
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
