--- gtkgraph-0.6.2/src/gtkitementry.c	1999-12-23 23:08:28.000000000 +1100
+++ gtkgraph-0.6.2-patched/src/gtkitementry.c	2002-07-09 00:20:17.000000000 +1000
@@ -1918,7 +1918,7 @@
 }
 
 void       
-gtk_entry_select_region  (GtkEntry       *entry,
+gtkgraph_entry_select_region  (GtkEntry       *entry,
 			  gint            start,
 			  gint            end)
 {
@@ -1926,7 +1926,7 @@
 }
 
 void
-gtk_entry_set_max_length (GtkEntry     *entry,
+gtkgraph_entry_set_max_length (GtkEntry     *entry,
                           guint16       max)
 {
   g_return_if_fail (entry != NULL);
