diff -Nurd autotrace-0.31.10.orig/Makefile.am autotrace-0.31.10/Makefile.am
--- autotrace-0.31.10.orig/Makefile.am	2024-01-29 16:37:09.000000000 -0500
+++ autotrace-0.31.10/Makefile.am	2024-12-22 22:31:34.000000000 -0500
@@ -20,7 +20,7 @@
 	   src/output-swf.c \
 	   src/output-swf.h
 
-AUTOMAKE_OPTIONS = foreign no-dependencies
+AUTOMAKE_OPTIONS = foreign
 
 autotraceincludedir=$(includedir)/autotrace
 lib_LTLIBRARIES=libautotrace.la
@@ -127,7 +127,7 @@
 		src/exception.h \
 		src/color.h
 
-libautotrace_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+libautotrace_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -no-undefined
 
 #
 # noinst_HEADERS: headers shared between lib and bin.
@@ -141,7 +141,16 @@
 		src/atou.h				\
 		src/main.c
 
-AM_CPPFLAGS = $(GRAPHICSMAGICK_CFLAGS) $(IMAGEMAGICK_CFLAGS) $(LIBPSTOEDIT_CFLAGS) $(LIBSWF_CFLAGS) $(GLIB2_CFLAGS) -DLOCALEDIR=\""$(datadir)/locale"\"
+AM_CPPFLAGS = $(GRAPHICSMAGICK_CFLAGS) $(IMAGEMAGICK_CFLAGS) $(LIBPSTOEDIT_CFLAGS) $(LIBSWF_CFLAGS) $(GLIB2_CFLAGS) $(LIBPKG_CFLAGS) -DLOCALEDIR=\""$(datadir)/locale"\"
+
+libautotrace_la_LIBADD =				\
+		$(GRAPHICSMAGICK_LIBS)		\
+		$(IMAGEMAGICK_LIBS)		\
+		$(LIBPNG_LIBS)		\
+		$(LIBSWF_LIBS)		\
+		$(LIBPSTOEDIT_LIBS)		\
+		$(GLIB2_LIBS)			\
+		$(INTLLIBS)
 
 autotrace_LDADD =				\
 		libautotrace.la			\
diff -Nurd autotrace-0.31.10.orig/autotrace.pc.in autotrace-0.31.10/autotrace.pc.in
--- autotrace-0.31.10.orig/autotrace.pc.in	2024-01-29 16:37:09.000000000 -0500
+++ autotrace-0.31.10/autotrace.pc.in	2024-12-22 18:07:45.000000000 -0500
@@ -6,6 +6,6 @@
 Name: Autotrace
 Description: a utility that converts bitmap to vector graphics
 Version: @VERSION@
-Requires:
-Libs: -L@libdir@ -lautotrace @LIBPNG_LIBS@ @MAGICK_LIBS@ @LIBSWF_LIBS@ @LIBPSTOEDIT_LIBS@ @GLIB2_LIBS@
-Cflags: -I@includedir@ @MAGICK_CFLAGS@ @LIBPSTOEDIT_CFLAGS@ @GLIB2_CFLAGS@
+Requires.private: glib-2.0
+Libs: -L@libdir@ -lautotrace
+Cflags: -I@includedir@
