diff -Nurd autotrace-travis-20200219.65.orig/Makefile.am autotrace-travis-20200219.65/Makefile.am
--- autotrace-travis-20200219.65.orig/Makefile.am	2020-02-19 11:15:50.000000000 -0500
+++ autotrace-travis-20200219.65/Makefile.am	2021-04-04 03:44:57.000000000 -0400
@@ -20,7 +20,7 @@
 	   src/output-swf.c \
 	   src/output-swf.h
 
-AUTOMAKE_OPTIONS = foreign no-dependencies
+AUTOMAKE_OPTIONS = foreign
 
 autotraceincludedir=$(includedir)/autotrace
 lib_LTLIBRARIES=libautotrace.la
@@ -121,7 +121,7 @@
 		src/exception.h \
 		src/color.h
 
-libautotrace_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
+libautotrace_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -no-undefined
 
 #
 # noinst_HEADERS: headers shared between lib and bin.
@@ -141,6 +141,8 @@
 
 autotrace_LDADD =				\
 		libautotrace.la			\
+		$(INTLLIBS)
+libautotrace_la_LIBADD =			\
 		$(MAGICK_LIBS)		\
 		$(LIBPNG_LIBS)		\
 		$(LIBSWF_LIBS)		\
diff -Nurd autotrace-travis-20200219.65.orig/autotrace.pc.in autotrace-travis-20200219.65/autotrace.pc.in
--- autotrace-travis-20200219.65.orig/autotrace.pc.in	2020-02-19 11:15:50.000000000 -0500
+++ autotrace-travis-20200219.65/autotrace.pc.in	2021-04-04 04:09:47.000000000 -0400
@@ -6,6 +6,6 @@
 Name: Autotrace
 Description: a utility that converts bitmap to vector graphics
 Version: @VERSION@
-Requires:
-Libs: -L@libdir@ -lautotrace @LIBPNG_LIBS@ @MAGICK_LIBS@ @LIBSWF_LIBS@ @LIBPSTOEDIT_LIBS@ @GLIB2_LIBS@
-Cflags: -I@includedir@ @MAGICK_CFLAGS@ @LIBPSTOEDIT_CFLAGS@ @GLIB2_CFLAGS@
+Requires.private: glib-2.0
+Libs: -L@libdir@ -lautotrace
+Cflags: -I@includedir@
