diff -Nurd cairo-1.16.0.orig/Makefile.in cairo-1.16.0/Makefile.in
--- cairo-1.16.0.orig/Makefile.in	2018-10-19 16:20:07.000000000 -0400
+++ cairo-1.16.0/Makefile.in	2025-05-29 09:36:12.000000000 -0400
@@ -107,7 +107,7 @@
 TESTS =
 check_PROGRAMS =
 # libpng is required for our test programs
-@CAIRO_HAS_PNG_FUNCTIONS_TRUE@am__append_1 = boilerplate test perf
+@CAIRO_HAS_PNG_FUNCTIONS_TRUE@am__append_1 = boilerplate
 subdir = .
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/build/aclocal.cairo.m4 \
diff -Nurd cairo-1.16.0.orig/configure cairo-1.16.0/configure
--- cairo-1.16.0.orig/configure	2018-10-19 16:20:08.000000000 -0400
+++ cairo-1.16.0/configure	2025-05-29 09:36:12.000000000 -0400
@@ -22837,6 +22837,8 @@
 
       fi
 
+      old_CPPFLAGS=$CPPFLAGS
+      CPPFLAGS="$CPPFLAGS $xlib_CFLAGS $xlib_NONPKGCONFIG_CFLAGS"
       for ac_header in X11/extensions/XShm.h X11/extensions/shmproto.h X11/extensions/shmstr.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
@@ -22851,6 +22853,7 @@
 fi
 
 done
+CPPFLAGS=$old_CPPFLAGS
 
 
 
diff -Nurd cairo-1.16.0.orig/util/cairo-gobject/Makefile.in cairo-1.16.0/util/cairo-gobject/Makefile.in
--- cairo-1.16.0.orig/util/cairo-gobject/Makefile.in	2018-10-19 16:20:08.000000000 -0400
+++ cairo-1.16.0/util/cairo-gobject/Makefile.in	2025-05-29 09:36:12.000000000 -0400
@@ -150,7 +150,7 @@
 am__installdirs = "$(DESTDIR)$(libdir)" "$(DESTDIR)$(cairoincludedir)"
 LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
-libcairo_gobject_la_DEPENDENCIES = $(top_builddir)/src/libcairo.la \
+libcairo_gobject_la_DEPENDENCIES = \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 am_libcairo_gobject_la_OBJECTS =  \
 	libcairo_gobject_la-cairo-gobject-enums.lo \
@@ -447,7 +447,7 @@
 xlib_xrender_CFLAGS = @xlib_xrender_CFLAGS@
 xlib_xrender_LIBS = @xlib_xrender_LIBS@
 lib_LTLIBRARIES = libcairo-gobject.la
-AM_CPPFLAGS = -I$(top_srcdir)/src -I$(top_builddir)/src
+AM_CPPFLAGS = `$(PKG_CONFIG) --cflags cairo`
 cairoincludedir = $(includedir)/cairo
 cairoinclude_HEADERS = cairo-gobject.h
 libcairo_gobject_la_SOURCES = \
@@ -457,7 +457,7 @@
 
 libcairo_gobject_la_CFLAGS = $(CAIRO_CFLAGS) $(GOBJECT_CFLAGS)
 libcairo_gobject_la_LDFLAGS = -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols)
-libcairo_gobject_la_LIBADD = $(top_builddir)/src/libcairo.la $(CAIRO_LIBS) $(GOBJECT_LIBS)
+libcairo_gobject_la_LIBADD = `$(PKG_CONFIG) --libs cairo` $(GOBJECT_LIBS)
 all: all-am
 
 .SUFFIXES:
diff -Nurd cairo-1.16.0.orig/util/cairo-script/Makefile.in cairo-1.16.0/util/cairo-script/Makefile.in
--- cairo-1.16.0.orig/util/cairo-script/Makefile.in	2018-10-19 16:20:08.000000000 -0400
+++ cairo-1.16.0/util/cairo-script/Makefile.in	2025-05-29 09:36:12.000000000 -0400
@@ -154,7 +154,7 @@
 LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
 libcairo_script_interpreter_la_DEPENDENCIES =  \
-	$(top_builddir)/src/libcairo.la $(am__DEPENDENCIES_1) \
+	$(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1)
 am__objects_1 = libcairo_script_interpreter_la-cairo-script-file.lo \
 	libcairo_script_interpreter_la-cairo-script-hash.lo \
@@ -183,11 +183,11 @@
 am_csi_exec_OBJECTS = csi-exec.$(OBJEXT)
 csi_exec_OBJECTS = $(am_csi_exec_OBJECTS)
 csi_exec_DEPENDENCIES = libcairo-script-interpreter.la \
-	$(top_builddir)/src/libcairo.la $(am__DEPENDENCIES_1)
+	$(am__DEPENDENCIES_1)
 am_csi_replay_OBJECTS = csi_replay-csi-replay.$(OBJEXT)
 csi_replay_OBJECTS = $(am_csi_replay_OBJECTS)
 csi_replay_DEPENDENCIES = libcairo-script-interpreter.la \
-	$(top_builddir)/src/libcairo.la $(am__DEPENDENCIES_1)
+	$(am__DEPENDENCIES_1)
 csi_replay_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(csi_replay_CFLAGS) \
 	$(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
@@ -197,7 +197,7 @@
 csi_trace_OBJECTS = $(am_csi_trace_OBJECTS)
 @CAIRO_HAS_SCRIPT_SURFACE_TRUE@csi_trace_DEPENDENCIES =  \
 @CAIRO_HAS_SCRIPT_SURFACE_TRUE@	libcairo-script-interpreter.la \
-@CAIRO_HAS_SCRIPT_SURFACE_TRUE@	$(top_builddir)/src/libcairo.la \
+@CAIRO_HAS_SCRIPT_SURFACE_TRUE@	 \
 @CAIRO_HAS_SCRIPT_SURFACE_TRUE@	$(am__DEPENDENCIES_1)
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
@@ -543,7 +543,7 @@
 
 SUBDIRS = examples
 lib_LTLIBRARIES = libcairo-script-interpreter.la
-AM_CPPFLAGS = -I$(top_srcdir)/src -I$(top_builddir)/src
+AM_CPPFLAGS = `$(PKG_CONFIG) --cflags cairo`
 cairoincludedir = $(includedir)/cairo
 cairoinclude_HEADERS = cairo-script-interpreter.h
 libcairo_script_interpreter_la_SOURCES = \
@@ -553,14 +553,14 @@
 
 libcairo_script_interpreter_la_CFLAGS = $(CAIRO_CFLAGS)
 libcairo_script_interpreter_la_LDFLAGS = -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols)
-libcairo_script_interpreter_la_LIBADD = $(top_builddir)/src/libcairo.la $(CAIRO_LIBS) $(lzo_LIBS) -lz
+libcairo_script_interpreter_la_LIBADD = `$(PKG_CONFIG) --libs cairo` $(CAIRO_LIBS) $(lzo_LIBS) -lz
 csi_replay_SOURCES = csi-replay.c
 csi_replay_CFLAGS = $(CAIRO_CFLAGS)
-csi_replay_LDADD = libcairo-script-interpreter.la $(top_builddir)/src/libcairo.la $(CAIRO_LIBS)
+csi_replay_LDADD = libcairo-script-interpreter.la `$(PKG_CONFIG) --libs cairo` $(CAIRO_LIBS)
 csi_exec_SOURCES = csi-exec.c
-csi_exec_LDADD = libcairo-script-interpreter.la $(top_builddir)/src/libcairo.la $(CAIRO_LIBS)
+csi_exec_LDADD = libcairo-script-interpreter.la `$(PKG_CONFIG) --libs cairo` $(CAIRO_LIBS)
 @CAIRO_HAS_SCRIPT_SURFACE_TRUE@csi_trace_SOURCES = csi-trace.c
-@CAIRO_HAS_SCRIPT_SURFACE_TRUE@csi_trace_LDADD = libcairo-script-interpreter.la $(top_builddir)/src/libcairo.la $(CAIRO_LIBS)
+@CAIRO_HAS_SCRIPT_SURFACE_TRUE@csi_trace_LDADD = libcairo-script-interpreter.la `$(PKG_CONFIG) --libs cairo` $(CAIRO_LIBS)
 EXTRA_DIST = \
 	COPYING
 
