diff -ruN compface-1.5.2-orig/cmain.c compface-1.5.2/cmain.c
--- compface-1.5.2-orig/cmain.c	2005-10-04 07:28:47.000000000 -0500
+++ compface-1.5.2/cmain.c	2021-01-17 08:34:43.000000000 -0600
@@ -15,6 +15,8 @@
  *  to me, then an attempt will be made to fix them.
  */
 
+#include <unistd.h>
+
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
diff -ruN compface-1.5.2-orig/configure compface-1.5.2/configure
--- compface-1.5.2-orig/configure	2005-10-04 07:57:59.000000000 -0500
+++ compface-1.5.2/configure	2021-01-17 08:34:08.000000000 -0600
@@ -2802,6 +2802,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 #include <ctype.h>
+#include <stdlib.h>
 #if ((' ' & 0x0FF) == 0x020)
 # define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 # define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
diff -ruN compface-1.5.2-orig/uncmain.c compface-1.5.2/uncmain.c
--- compface-1.5.2-orig/uncmain.c	2005-10-04 07:28:47.000000000 -0500
+++ compface-1.5.2/uncmain.c	2021-01-17 08:35:03.000000000 -0600
@@ -15,6 +15,8 @@
  *  to me, then an attempt will be made to fix them.
  */
 
+#include <unistd.h>
+
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
