diff -Nurd -x'*~' compface-1.5.2.orig/Makefile.in compface-1.5.2/Makefile.in
--- compface-1.5.2.orig/Makefile.in	2005-10-04 08:59:49.000000000 -0400
+++ compface-1.5.2/Makefile.in	2008-06-09 15:09:50.000000000 -0400
@@ -17,6 +17,7 @@
 SHELL           = /bin/sh
 
 NAME		= compface
+VERSION		= 1.4
 UNNAME		= uncompface
 EXEEXT		= @EXEEXT@
 NAMEEXE		= $(NAME)$(EXEEXT)
@@ -25,6 +26,8 @@
 UNEXECUTABLE	= $(BINDIR)/$(UNNAMEEXE)
 LIBNAME		= lib$(NAME).a
 LIBRARY		= $(LIBDIR)/$(LIBNAME)
+DYLIBNAME	= lib$(NAME).0.dylib
+DYLIBLDNAME	= lib$(NAME).dylib
 MAN1DIR		= $(MANDIR)/man1
 MAN3DIR		= $(MANDIR)/man3
 OBJECTS		= arith.o file.o compress.o gen.o uncompface.o
@@ -43,18 +46,21 @@
 LDFLAGS		= @LDFLAGS@
 LIBS		= @LIBS@
 
-all:		$(NAMEEXE) $(UNNAMEEXE)
+all:		$(NAMEEXE) $(UNNAMEEXE) $(LIBNAME) $(DYLIBNAME)
 
-$(NAMEEXE) :	cmain.o compface.o $(LIBNAME)
-		$(CC) $(LDFLAGS) -o $@ cmain.o compface.o $(LIBNAME)
+$(NAMEEXE) :	cmain.o compface.o $(DYLIBNAME)
+		$(CC) $(LDFLAGS) -o $@ cmain.o compface.o $(DYLIBNAME)
 
 $(UNNAMEEXE) :	uncmain.o $(LIBNAME)
-		$(CC) $(LDFLAGS) -o $@ uncmain.o $(LIBNAME)
+		$(CC) $(LDFLAGS) -o $@ uncmain.o $(DYLIBNAME)
 
 $(LIBNAME) :	$(OBJECTS)
 		ar rc $(LIBNAME) $(OBJECTS)
 		-$(RANLIB) $(LIBNAME)
 
+$(DYLIBNAME) :	$(OBJECTS)
+		$(CC) -dynamiclib $(CFLAGS) $(LDFLAGS) -install_name $(LIBDIR)/$(DYLIBNAME) -current_version 0.$(VERSION) -compatibility_version 0.$(VERSION) -o $(DYLIBNAME) $(OBJECTS)
+
 lint :
 		lint -abchx $(SOURCES)
 
@@ -87,6 +93,8 @@
 		cd $(MAN3DIR) && $(RM) ./$(UNNAME).3 && $(LN_S) $(NAME).3 $(UNNAME).3
 		for hdr in $(INSTALLHEADERS); do $(INSTALL_DATA) $(srcdir)/$${hdr} $(INCLUDEDIR)/$${hdr}; done
 		$(INSTALL_DATA) $(srcdir)/$(LIBNAME) $(LIBRARY)
+		$(INSTALL_DATA) $(srcdir)/$(DYLIBNAME) $(LIBDIR)
+		$(LN_S) $(DYLIBNAME) $(LIBDIR)/$(DYLIBLDNAME)
 
 shar :
 		shar.script $(OTHERS) $(HDRS) $(SOURCES) > $(NAME).sh
