diff -ruN compface-1.5.2-orig/Makefile.in compface-1.5.2/Makefile.in
--- compface-1.5.2-orig/Makefile.in	2005-10-04 08:59:49
+++ compface-1.5.2/Makefile.in	2024-11-03 10:34:12
@@ -17,14 +17,16 @@
 SHELL           = /bin/sh
 
 NAME		= compface
+VERSION		= 1.4
 UNNAME		= uncompface
 EXEEXT		= @EXEEXT@
 NAMEEXE		= $(NAME)$(EXEEXT)
 UNNAMEEXE	= $(UNNAME)$(EXEEXT)
 EXECUTABLE	= $(BINDIR)/$(NAMEEXE)
 UNEXECUTABLE	= $(BINDIR)/$(UNNAMEEXE)
-LIBNAME		= lib$(NAME).a
+LIBNAME		= lib$(NAME).0.dylib
 LIBRARY		= $(LIBDIR)/$(LIBNAME)
+DYLIBLDNAME	= lib$(NAME).dylib
 MAN1DIR		= $(MANDIR)/man1
 MAN3DIR		= $(MANDIR)/man3
 OBJECTS		= arith.o file.o compress.o gen.o uncompface.o
@@ -43,7 +45,7 @@
 LDFLAGS		= @LDFLAGS@
 LIBS		= @LIBS@
 
-all:		$(NAMEEXE) $(UNNAMEEXE)
+all:		$(NAMEEXE) $(UNNAMEEXE) $(LIBNAME)
 
 $(NAMEEXE) :	cmain.o compface.o $(LIBNAME)
 		$(CC) $(LDFLAGS) -o $@ cmain.o compface.o $(LIBNAME)
@@ -52,8 +54,7 @@
 		$(CC) $(LDFLAGS) -o $@ uncmain.o $(LIBNAME)
 
 $(LIBNAME) :	$(OBJECTS)
-		ar rc $(LIBNAME) $(OBJECTS)
-		-$(RANLIB) $(LIBNAME)
+		$(CC) -dynamiclib $(CFLAGS) $(LDFLAGS) -install_name $(LIBRARY) -current_version 0.$(VERSION) -compatibility_version 0.$(VERSION) -o $(LIBNAME) $(OBJECTS)
 
 lint :
 		lint -abchx $(SOURCES)
@@ -87,6 +88,7 @@
 		cd $(MAN3DIR) && $(RM) ./$(UNNAME).3 && $(LN_S) $(NAME).3 $(UNNAME).3
 		for hdr in $(INSTALLHEADERS); do $(INSTALL_DATA) $(srcdir)/$${hdr} $(INCLUDEDIR)/$${hdr}; done
 		$(INSTALL_DATA) $(srcdir)/$(LIBNAME) $(LIBRARY)
+		$(LN_S) $(LIBNAME) $(LIBDIR)/$(DYLIBLDNAME)
 
 shar :
 		shar.script $(OTHERS) $(HDRS) $(SOURCES) > $(NAME).sh
diff -ruN compface-1.5.2-orig/file.c compface-1.5.2/file.c
--- compface-1.5.2-orig/file.c	2024-11-03 10:37:08
+++ compface-1.5.2/file.c	2024-11-03 10:35:48
@@ -80,7 +80,7 @@
 	static char table_inv[] = { 0,8,4,12,2,10,6,14,1,9, 5,13, 3,11, 7,15 };
 	static char table_nop[] = { 0,1,2, 3,4, 5,6, 7,8,9,10,11,12,13,14,15 };
 	char *table = table_nop;	/* optionally invert bits in nibble */
-	register inc = 0;		/* optionally swap nimmles */
+	register int inc = 0;		/* optionally swap nimmles */
 	int bits;
 	int len;
 
