diff -Nurd converseen-0.9.9.0.orig/CMakeLists.txt converseen-0.9.9.0/CMakeLists.txt
--- converseen-0.9.9.0.orig/CMakeLists.txt	2021-02-15 07:36:48.000000000 -0500
+++ converseen-0.9.9.0/CMakeLists.txt	2021-04-18 11:26:38.000000000 -0400
@@ -135,7 +135,7 @@
 install(FILES res/converseen.desktop DESTINATION ${CMAKE_INSTALL_PREFIX}/share/applications)
 install(FILES res/converseen.png DESTINATION ${CMAKE_INSTALL_PREFIX}/share/pixmaps)
 install(FILES res/converseen_import.desktop DESTINATION ${CMAKE_INSTALL_PREFIX}/share/kservices5/ServiceMenus)
-install(FILES converseen.appdata.xml DESTINATION /usr/share/metainfo/)
+install(FILES converseen.appdata.xml DESTINATION ${CMAKE_INSTALL_PREFIX}/share/metainfo/)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/converseen_cs_CZ.qm DESTINATION ${CMAKE_INSTALL_PREFIX}/share/converseen/loc)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/converseen_de_DE.qm DESTINATION ${CMAKE_INSTALL_PREFIX}/share/converseen/loc)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/converseen_es_CL.qm DESTINATION ${CMAKE_INSTALL_PREFIX}/share/converseen/loc)
diff -Nurd converseen-0.9.9.0.orig/src/mainwindowimpl.cpp converseen-0.9.9.0/src/mainwindowimpl.cpp
--- converseen-0.9.9.0.orig/src/mainwindowimpl.cpp	2021-02-15 07:36:48.000000000 -0500
+++ converseen-0.9.9.0/src/mainwindowimpl.cpp	2021-04-18 11:07:10.000000000 -0400
@@ -1014,7 +1014,6 @@
 void MainWindowImpl::checkForUpdates()
 {
     UpdateChecker *up = new UpdateChecker();
-    up->checkForUpdates();
 }
 
 void MainWindowImpl::bugReport()
