diff -ruN ETL-0.04.16-orig/test/Makefile.in ETL-0.04.16/test/Makefile.in
--- ETL-0.04.16-orig/test/Makefile.in	2013-05-09 12:32:44.000000000 -0400
+++ ETL-0.04.16/test/Makefile.in	2013-08-01 15:29:08.000000000 -0400
@@ -381,6 +381,10 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/surface.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/value.Po@am__quote@
 
+fixed.o:
+@am__fastdepCXX_TRUE@	$(CXXCOMPILE) -O0 -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $(fixed_SOURCES)
+@am__fastdepCXX_TRUE@	$(am__mv) $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
+
 .cpp.o:
 @am__fastdepCXX_TRUE@	$(CXXCOMPILE) -MT $@ -MD -MP -MF $(DEPDIR)/$*.Tpo -c -o $@ $<
 @am__fastdepCXX_TRUE@	$(am__mv) $(DEPDIR)/$*.Tpo $(DEPDIR)/$*.Po
