diff -ruN evolvotron-orig/common.pro evolvotron/common.pro
--- evolvotron-orig/common.pro	2017-11-14 14:40:53.000000000 -0600
+++ evolvotron/common.pro	2020-05-20 06:17:04.000000000 -0500
@@ -9,7 +9,7 @@
 # Control platform specific code
 # (the Qt platform scopes don't seem that useful; might be easier to just add DEFINES+=... explicitly 
 
-unix {
+linux {
   DEFINES+=PLATFORM_LINUX      # of course PLATFORM_BSD is more appropriate to some unices
 }
 
diff -ruN evolvotron-orig/libevolvotron/libevolvotron.pro evolvotron/libevolvotron/libevolvotron.pro
--- evolvotron-orig/libevolvotron/libevolvotron.pro	2017-11-14 14:40:53.000000000 -0600
+++ evolvotron/libevolvotron/libevolvotron.pro	2020-05-20 06:17:20.000000000 -0500
@@ -1,6 +1,6 @@
 TEMPLATE = lib
 
-QT += widgets
+QT += widgets xml
 
 TARGET = evolvotron   # Have to override this or we get "liblibevolvotron"
 
