diff -Nurd -x'*~' feh-2.14.orig/config.mk feh-2.14/config.mk
--- feh-2.14.orig/config.mk	2015-10-04 04:03:43.000000000 -0400
+++ feh-2.14/config.mk	2015-11-07 23:45:32.000000000 -0500
@@ -71,4 +71,5 @@
 CFLAGS += -DPREFIX=\"${PREFIX}\" \
 	-DPACKAGE=\"${PACKAGE}\" -DVERSION=\"${VERSION}\"
 
-LDLIBS += -lm -lpng -lX11 -lImlib2
+LDLIBS += -lm -lpng `pkg-config --libs x11` -lImlib2
+CFLAGS += `pkg-config --cflags imlib2 x11`
