diff -ruN a/fig2dev/tests/output.at b/fig2dev/tests/output.at
--- a/fig2dev/tests/output.at
+++ b/fig2dev/tests/output.at
@@ -160,7 +160,7 @@
 
 AT_SETUP([conditionally allocate \XFigu])
 AT_KEYWORDS(tikz newdimen)
-AT_SKIP_IF([! tex --version || ! latex --version])
+AT_SKIP_IF([! etex --version || ! latex --version])
 AT_DATA(tex.tex, [\input tikz
 \input line.tikz\immediate\write16{\meaning\XFigu}
 \input line.tikz\immediate\write16{\meaning\XFigu}\bye
@@ -172,7 +172,7 @@
 ])
 fig2dev -L tikz $srcdir/data/line.fig line.tikz
 dnl do not count on grep to support -q, or -s
-AT_CHECK([tex tex.tex | $FGREP dimen | uniq | wc -l
+AT_CHECK([etex tex.tex | $FGREP dimen | uniq | wc -l
 ],0,[1
 ],ignore-nolog)
 AT_CHECK([latex latex.tex | $FGREP dimen | uniq | wc -l
diff -ruN a/fig2dev/tests/output.at b/fig2dev/tests/output.at
--- a/fig2dev/tests/output.at
+++ b/fig2dev/tests/output.at
@@ -172,10 +172,10 @@
 ])
 fig2dev -L tikz $srcdir/data/line.fig line.tikz
 dnl do not count on grep to support -q, or -s
-AT_CHECK([etex tex.tex | $FGREP dimen | uniq | wc -l
+AT_CHECK([etex tex.tex | $FGREP dimen | uniq | wc -l | tr -d ' \t'
 ],0,[1
 ],ignore-nolog)
-AT_CHECK([latex latex.tex | $FGREP dimen | uniq | wc -l
+AT_CHECK([latex latex.tex | $FGREP dimen | uniq | wc -l | tr -d ' \t'
 ],0,[1
 ],ignore-nolog)
 AT_CLEANUP
