--- fig2dev-3.2.9-orig/fig2dev/read.c	2023-05-21 11:50:32.000000000 -0500
+++ fig2dev-3.2.9/fig2dev/read.c	2023-09-04 14:31:15.000000000 -0500
@@ -152,7 +152,8 @@
 		status = read_objects(fp, obj);
 	else
 		status = read_1_3_objects(fp, obj);
-	(void)fclose(fp);
+	if (fp != stdin)
+		(void)fclose(fp);
 	return status;
 }
 
https://sourceforge.net/p/mcj/fig2dev/ci/ebbb9e22afa6b2646d8cd9c359d62b604fc7f9d4/tree/fig2dev/tests/bitmaps.at?diff=84322bcb96d5c1bbb5407705d8847a2b8b997615
--- a/fig2dev/tests/bitmaps.at
+++ b/fig2dev/tests/bitmaps.at
@@ -245,8 +245,9 @@
 4 0 0 50 -1 16 12 4.7124 4 210 2100 7225 1575 Included postscript file\001
 EOF`
 # $blackpixels are 1488 on archlinux with ghostscript 9.20,
-#		1433 on ubuntu 16.04.2 with ghostscript 9.18
-AT_CHECK([test $blackpixels -gt 1431 && test $blackpixels -lt 1498])
+#		1433 on ubuntu 16.04.2 with ghostscript 9.18,
+#               1498 on Debian with ghostscript 10.02
+AT_CHECK([test $blackpixels -gt 1431 && test $blackpixels -lt 1500])
 AT_CLEANUP
 
 AT_SETUP([find /MediaBox in pdf file])
