diff -Nurd fontforge-20160404.orig/m4/fontforge_platform_specifics.m4 fontforge-20160404/m4/fontforge_platform_specifics.m4
--- fontforge-20160404.orig/m4/fontforge_platform_specifics.m4	2016-04-04 19:56:57.000000000 -0400
+++ fontforge-20160404/m4/fontforge_platform_specifics.m4	2021-05-29 06:41:13.000000000 -0400
@@ -31,24 +31,7 @@
       dnl but when macport/fink overwrite a standard lib, I still want the standard
       dnl  library to be found (otherwise only portable to machines with macports/fink)
 
-      if test "$CPPFLAGS" \!= "" >/dev/null ; then
        TrustUser="yes"
-      elif test -d /sw/include >/dev/null ; then
-       CPPFLAGS="$CPPFLAGS -I /sw/include"
-       gww_extraincludes="/sw/include"
-      elif test -d /opt/local/include >/dev/null ; then
-       CPPFLAGS="$CPPFLAGS -I /opt/local/include"
-       gww_extraincludes="/opt/local/include";
-      fi
-      if test "$oldLDFLAGS" \!= "" >/dev/null ; then
-       TrustUser="yes"
-      elif test -d /sw/lib >/dev/null ; then
-       LDFLAGS="$LDFLAGS -L/sw/lib"
-       gww_rpath="-rpath /sw/lib"
-      elif test -d /opt/local/lib >/dev/null ; then
-       LDFLAGS="$LDFLAGS -L/opt/local/lib"
-       gww_rpath="-rpath /opt/local/lib"
-      fi
 
       AC_ARG_VAR([SDK],[(Macintosh only) path to software development kit; defaults to ]default_SDK)
       AC_ARG_VAR([CARBON],[(Macintosh only) path to Carbon; defaults to SDK/]default_CARBON)
