diff -ruN gmic-1.6.0.0-orig/src/Makefile gmic-1.6.0.0/src/Makefile
--- gmic-1.6.0.0-orig/src/Makefile	2014-08-21 02:47:10.000000000 -0500
+++ gmic-1.6.0.0/src/Makefile	2014-09-04 18:25:40.000000000 -0500
@@ -48,7 +48,7 @@
 # Set correct variables and paths.
 #----------------------------------
 OS := $(shell uname)
-CC = g++
+CC = g++-fsf-9 -MD
 ifeq ($(OS),Linux)
 OS = Unix
 endif
@@ -58,13 +58,8 @@
 ifeq ($(OS),GNU)
 OS = Unix
 endif
-ifeq ($(OS),Darwin)
-PLUGINDIR = ~/Library/Application\ Support/Gimp/plug-ins
-USR = /opt/local
-else
 PLUGINDIR = `gimptool-2.0 --gimpplugindir`/plug-ins
-USR = /usr
-endif
+USR = @FINKPREFIX@
 ifeq ($(OSTYPE),msys)
 EXE = .exe
 endif
@@ -406,8 +401,7 @@
 	mkdir -p $(DESTDIR)$(USR)/share/man/fr/man1/
 	cp -f ../man/gmic.1.gz $(DESTDIR)$(USR)/share/man/man1/gmic.1.gz
 	cp -f ../man/gmic.1.gz $(DESTDIR)$(USR)/share/man/fr/man1/gmic.1.gz
-	if [ -d /etc/bash_completion.d/ ]; then mkdir -p $(DESTDIR)/etc/bash_completion.d/; cp -f ../resources/gmic_bashcompletion.sh $(DESTDIR)/etc/bash_completion.d/gmic; fi
-	if [ -d /opt/local/etc/bash_completion.d/ ]; then mkdir -p $(DESTDIR)/opt/local/etc/bash_completion.d/; cp -f ../resources/gmic_bashcompletion.sh $(DESTDIR)/opt/local/etc/bash_completion.d/gmic; fi
+	mkdir -p $(DESTDIR)$(USR)/etc/bash_completion.d/; cp -f ../resources/gmic_bashcompletion.sh $(DESTDIR)$(USR)/etc/bash_completion.d/gmic
 
 uninstall:
 	rm -f $(DESTDIR)$(USR)/$(BIN)/gmic
diff -ruN gmic-1.6.0.0-orig/zart/zart.pro gmic-1.6.0.0/zart/zart.pro
--- gmic-1.6.0.0-orig/zart/zart.pro	2014-08-21 02:47:10.000000000 -0500
+++ gmic-1.6.0.0/zart/zart.pro	2014-09-04 18:19:10.000000000 -0500
@@ -4,9 +4,9 @@
 CONFIG	+= warn_on release
 
 !macx {
-	INCLUDEPATH	+= .. ./include /usr/include/opencv ../src/
+	INCLUDEPATH	+= .. ./include ../src/ @FINKPREFIX@/include/opencv @FINKPREFIX@/include
 } else {
-	INCLUDEPATH	+= .. ./include /opt/local/include /opt/local/include/opencv ../src/
+	INCLUDEPATH	+= .. ./include ../src/ @FINKPREFIX@/include/opencv
 }
 
 DEPENDPATH += ./include
