diff -Nurd GraphicsMagick-1.3.47.orig/configure GraphicsMagick-1.3.47/configure
--- GraphicsMagick-1.3.47.orig/configure	2026-05-13 10:14:52
+++ GraphicsMagick-1.3.47/configure	2026-07-11 16:13:56
@@ -36178,7 +36178,7 @@
 
 if test "$build_modules" != 'no' ; then
   MAGICK_DEP_LIBS=''
-  for token in $LIB_FPX $LCMS2_LIBS $FREETYPE_LIBS $LIB_XEXT $LIB_IPC $LIB_X11 $LIB_BZLIB $ZLIB_LIBS $LIB_LTDL $LIB_TRIO $LIB_GDI32 $LIBS_USER $LIB_MATH $LIB_THREAD $LIB_TCMALLOC $LIB_UMEM $LIB_MTMALLOC ; do
+  for token in $LIB_FPX $LIB_LCMS $FREETYPE_LIBS $LIB_XEXT $LIB_IPC $LIB_X11 $LIB_BZLIB $LIB_ZLIB $LIB_LTDL $LIB_TRIO $LIB_GDI32 $LIBS_USER $LIB_MATH $LIB_THREAD $LIB_TCMALLOC $LIB_UMEM $LIB_MTMALLOC ; do
     case $token in
       -l*)
         # Add token to MAGICK_DEP_LIBS if not already present
