diff -Nurd -x'*~' ilmbase-2.2.0.orig/HalfTest/Makefile.in ilmbase-2.2.0/HalfTest/Makefile.in
--- ilmbase-2.2.0.orig/HalfTest/Makefile.in	2014-08-10 00:37:23.000000000 -0400
+++ ilmbase-2.2.0/HalfTest/Makefile.in	2015-05-05 23:43:05.000000000 -0400
@@ -221,7 +221,7 @@
 		   testSize.h
 
 INCLUDES = -I$(top_builddir) -I$(top_srcdir)/Half -I$(top_srcdir)/config
-LDADD = -L$(top_builddir)/Half -lHalf
+LDADD = $(top_builddir)/Half/libHalf.la
 EXTRA_DIST = CMakeLists.txt
 all: all-am
 
diff -Nurd -x'*~' ilmbase-2.2.0.orig/IexTest/Makefile.in ilmbase-2.2.0/IexTest/Makefile.in
--- ilmbase-2.2.0.orig/IexTest/Makefile.in	2014-08-10 00:37:24.000000000 -0400
+++ ilmbase-2.2.0/IexTest/Makefile.in	2015-05-05 23:43:16.000000000 -0400
@@ -212,7 +212,7 @@
 top_srcdir = @top_srcdir@
 IexTest_SOURCES = main.cpp testBaseExc.cpp testBaseExc.h
 INCLUDES = -I$(top_builddir) -I$(top_srcdir)/Iex -I$(top_srcdir)/config
-LDADD = -L$(top_builddir)/Iex -lIex
+LDADD = $(top_builddir)/Iex/libIex.la
 EXTRA_DIST = CMakeLists.txt
 all: all-am
 
diff -Nurd -x'*~' ilmbase-2.2.0.orig/ImathTest/Makefile.in ilmbase-2.2.0/ImathTest/Makefile.in
--- ilmbase-2.2.0.orig/ImathTest/Makefile.in	2014-08-10 00:37:24.000000000 -0400
+++ ilmbase-2.2.0/ImathTest/Makefile.in	2015-05-05 23:42:51.000000000 -0400
@@ -239,8 +239,7 @@
 INCLUDES = -I$(top_builddir) -I$(top_srcdir)/Imath -I$(top_srcdir)/Iex -I$(top_srcdir)/Half \
 	   -I$(top_srcdir)/config
 
-LDADD = -L$(top_builddir)/Imath -L$(top_builddir)/Iex -L$(top_builddir)/Half \
-	-lImath -lIex -lHalf
+LDADD = $(top_builddir)/Imath/libImath.la $(top_builddir)/Iex/libIex.la $(top_builddir)/Half/libHalf.la
 
 EXTRA_DIST = CMakeLists.txt
 all: all-am
