diff -ruN imlib2-1.6.1-orig/src/lib/Makefile.in imlib2-1.6.1/src/lib/Makefile.in
--- imlib2-1.6.1-orig/src/lib/Makefile.in	2019-12-13 12:10:13.000000000 -0600
+++ imlib2-1.6.1/src/lib/Makefile.in	2020-07-19 16:19:06.000000000 -0500
@@ -403,8 +403,8 @@
 AM_CFLAGS = $(CFLAGS_WARNINGS) $(CFLAGS_VISIBILITY)
 AM_CPPFLAGS = -DPACKAGE_LIB_DIR=\"$(libdir)\" \
                        -I$(top_builddir) \
-                       $(X_CFLAGS) \
-                       $(FREETYPE_CFLAGS)
+                       $(FREETYPE_CFLAGS) \
+                       $(X_CFLAGS)
 
 
 # Older automakes (<1.10) do not use INCLUDES when compiling .S files
@@ -482,7 +482,7 @@
 MMX_OBJS = $(MMX_SRCS:.S=.lo)
 AMD64_OBJS = $(AMD64_SRCS:.S=.lo)
 EXTRA_DIST = $(MMX_SRCS) $(AMD64_SRCS) asm_loadimmq.S
-MY_LIBS = $(FREETYPE_LIBS) $(DLOPEN_LIBS) -lm $(am__append_1)
+MY_LIBS = $(FREETYPE_LIBS) $(X_LIBS) $(DLOPEN_LIBS) -lm $(am__append_1)
 @BUILD_AMD64_FALSE@@BUILD_MMX_FALSE@libImlib2_la_LIBADD = $(MY_LIBS)
 @BUILD_AMD64_TRUE@@BUILD_MMX_FALSE@libImlib2_la_LIBADD = $(AMD64_OBJS) $(MY_LIBS)
 @BUILD_MMX_TRUE@libImlib2_la_LIBADD = $(MMX_OBJS) $(MY_LIBS)
@@ -491,7 +491,7 @@
 @BUILD_MMX_TRUE@libImlib2_la_DEPENDENCIES = $(top_builddir)/config.h $(MMX_OBJS)
 @BUILD_AMD64_TRUE@@BUILD_MMX_FALSE@EXTRA_libImlib2_la_SOURCES = $(AMD64_SRCS)
 @BUILD_MMX_TRUE@EXTRA_libImlib2_la_SOURCES = $(MMX_SRCS) 
-libImlib2_la_LDFLAGS = -version-info @lt_version@
+libImlib2_la_LDFLAGS = -version-info @lt_version@ -no-undefined
 all: all-am
 
 .SUFFIXES:
diff -ruN imlib2-1.6.1-orig/src/modules/filters/Makefile.in imlib2-1.6.1/src/modules/filters/Makefile.in
--- imlib2-1.6.1-orig/src/modules/filters/Makefile.in	2019-12-13 12:10:13.000000000 -0600
+++ imlib2-1.6.1/src/modules/filters/Makefile.in	2020-07-19 16:20:09.000000000 -0500
@@ -240,7 +240,7 @@
 CFLAGS_VISIBILITY = @CFLAGS_VISIBILITY@
 CFLAGS_WARNINGS = @CFLAGS_WARNINGS@
 CPP = @CPP@
-CPPFLAGS = @CPPFLAGS@
+CPPFLAGS = @CPPFLAGS@ $(X_CFLAGS)
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
@@ -374,15 +374,15 @@
 pkg_LTLIBRARIES = testfilter.la bumpmap.la colormod.la
 EXTRA_DIST = filter_common.h
 testfilter_la_SOURCES = filter_test.c
-testfilter_la_LDFLAGS = -module -avoid-version
+testfilter_la_LDFLAGS = -module -no-undefined -avoid-version
 testfilter_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 testfilter_la_LIBTOOLFLAGS = --tag=disable-static
 bumpmap_la_SOURCES = filter_bumpmap.c
-bumpmap_la_LDFLAGS = -module -avoid-version
+bumpmap_la_LDFLAGS = -module -no-undefined -avoid-version
 bumpmap_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la -lm
 bumpmap_la_LIBTOOLFLAGS = --tag=disable-static
 colormod_la_SOURCES = filter_colormod.c
-colormod_la_LDFLAGS = -module -avoid-version
+colormod_la_LDFLAGS = -module -no-undefined -avoid-version
 colormod_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 colormod_la_LIBTOOLFLAGS = --tag=disable-static
 all: all-am
diff -ruN imlib2-1.6.1-orig/src/modules/loaders/Makefile.in imlib2-1.6.1/src/modules/loaders/Makefile.in
--- imlib2-1.6.1-orig/src/modules/loaders/Makefile.in	2019-12-13 12:10:13.000000000 -0600
+++ imlib2-1.6.1/src/modules/loaders/Makefile.in	2020-07-19 16:22:04.000000000 -0500
@@ -345,7 +345,7 @@
 CFLAGS_VISIBILITY = @CFLAGS_VISIBILITY@
 CFLAGS_WARNINGS = @CFLAGS_WARNINGS@
 CPP = @CPP@
-CPPFLAGS = @CPPFLAGS@
+CPPFLAGS = @CPPFLAGS@ $(X_CFLAGS)
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
 DEPDIR = @DEPDIR@
@@ -504,69 +504,69 @@
 
 EXTRA_DIST = loader_common.h
 argb_la_SOURCES = loader_argb.c
-argb_la_LDFLAGS = -module -avoid-version
+argb_la_LDFLAGS = -module -no-undefined -avoid-version
 argb_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 argb_la_LIBTOOLFLAGS = --tag=disable-static
 bmp_la_SOURCES = loader_bmp.c
-bmp_la_LDFLAGS = -module -avoid-version
+bmp_la_LDFLAGS = -module -no-undefined -avoid-version
 bmp_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 bmp_la_LIBTOOLFLAGS = --tag=disable-static
 bz2_la_SOURCES = loader_bz2.c
-bz2_la_LDFLAGS = -module -avoid-version
+bz2_la_LDFLAGS = -module -no-undefined -avoid-version
 bz2_la_LIBADD = @BZ2LIBS@ $(top_builddir)/src/lib/libImlib2.la
 bz2_la_LIBTOOLFLAGS = --tag=disable-static
 ff_la_SOURCES = loader_ff.c
-ff_la_LDFLAGS = -module -avoid-version
+ff_la_LDFLAGS = -module -no-undefined -avoid-version
 ff_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 ff_la_LIBTOOLFLAGS = --tag=disable-static
 gif_la_SOURCES = loader_gif.c
-gif_la_LDFLAGS = -module -avoid-version
+gif_la_LDFLAGS = -module -no-undefined -avoid-version
 gif_la_LIBADD = @GIFLIBS@ $(top_builddir)/src/lib/libImlib2.la
 gif_la_LIBTOOLFLAGS = --tag=disable-static
 ico_la_SOURCES = loader_ico.c
-ico_la_LDFLAGS = -module -avoid-version
+ico_la_LDFLAGS = -module -no-undefined -avoid-version
 ico_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 ico_la_LIBTOOLFLAGS = --tag=disable-static
 id3_la_SOURCES = loader_id3.c
-id3_la_LDFLAGS = -module -avoid-version
+id3_la_LDFLAGS = -module -no-undefined -avoid-version
 id3_la_LIBADD = @ID3LIBS@ $(top_builddir)/src/lib/libImlib2.la
 id3_la_LIBTOOLFLAGS = --tag=disable-static
 jpeg_la_SOURCES = loader_jpeg.c
-jpeg_la_LDFLAGS = -module -avoid-version
+jpeg_la_LDFLAGS = -module -no-undefined -avoid-version
 jpeg_la_LIBADD = @JPEGLIBS@ $(top_builddir)/src/lib/libImlib2.la
 jpeg_la_LIBTOOLFLAGS = --tag=disable-static
 lbm_la_SOURCES = loader_lbm.c
-lbm_la_LDFLAGS = -module -avoid-version
+lbm_la_LDFLAGS = -module -no-undefined -avoid-version
 lbm_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 lbm_la_LIBTOOLFLAGS = --tag=disable-static
 png_la_SOURCES = loader_png.c
 png_la_CPPFLAGS = @PNG_CFLAGS@ $(AM_CPPFLAGS)
-png_la_LDFLAGS = -module -avoid-version
+png_la_LDFLAGS = -module -no-undefined -avoid-version
 png_la_LIBADD = @PNG_LIBS@ $(top_builddir)/src/lib/libImlib2.la
 png_la_LIBTOOLFLAGS = --tag=disable-static
 pnm_la_SOURCES = loader_pnm.c
-pnm_la_LDFLAGS = -module -avoid-version
+pnm_la_LDFLAGS = -module -no-undefined -avoid-version
 pnm_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 pnm_la_LIBTOOLFLAGS = --tag=disable-static
 tga_la_SOURCES = loader_tga.c
-tga_la_LDFLAGS = -module -avoid-version
+tga_la_LDFLAGS = -module -no-undefined -avoid-version
 tga_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 tga_la_LIBTOOLFLAGS = --tag=disable-static
 tiff_la_SOURCES = loader_tiff.c
-tiff_la_LDFLAGS = -module -avoid-version
+tiff_la_LDFLAGS = -module -no-undefined -avoid-version
 tiff_la_LIBADD = @TIFFLIBS@ $(top_builddir)/src/lib/libImlib2.la
 tiff_la_LIBTOOLFLAGS = --tag=disable-static
 webp_la_SOURCES = loader_webp.c
 webp_la_CPPFLAGS = @WEBP_CFLAGS@ $(AM_CPPFLAGS)
-webp_la_LDFLAGS = -module -avoid-version
+webp_la_LDFLAGS = -module -no-undefined -avoid-version
 webp_la_LIBADD = @WEBP_LIBS@ $(top_builddir)/src/lib/libImlib2.la
 webp_la_LIBTOOLFLAGS = --tag=disable-static
 xpm_la_SOURCES = loader_xpm.c
-xpm_la_LDFLAGS = -module -avoid-version
+xpm_la_LDFLAGS = -module -no-undefined -avoid-version
 xpm_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 xpm_la_LIBTOOLFLAGS = --tag=disable-static
 zlib_la_SOURCES = loader_zlib.c
-zlib_la_LDFLAGS = -module -avoid-version
+zlib_la_LDFLAGS = -module -no-undefined -avoid-version
 zlib_la_LIBADD = @ZLIBLIBS@ $(top_builddir)/src/lib/libImlib2.la
 zlib_la_LIBTOOLFLAGS = --tag=disable-static
 all: all-am
diff -ruN imlib2-1.6.1-orig/src/modules/loaders/loader_xpm.c imlib2-1.6.1/src/modules/loaders/loader_xpm.c
--- imlib2-1.6.1-orig/src/modules/loaders/loader_xpm.c	2019-12-12 09:33:39.000000000 -0600
+++ imlib2-1.6.1/src/modules/loaders/loader_xpm.c	2020-07-19 16:22:54.000000000 -0500
@@ -55,6 +55,10 @@
 
    /* look in rgb txt database */
    if (!rgb_txt)
+      rgb_txt = fopen("/opt/X11/share/X11/rgb.txt", "r");
+   if (!rgb_txt)
+      rgb_txt = fopen("/usr/X11/share/X11/rgb.txt", "r");
+   if (!rgb_txt)
       rgb_txt = fopen("/usr/share/X11/rgb.txt", "r");
    if (!rgb_txt)
       rgb_txt = fopen("/usr/X11R6/lib/X11/rgb.txt", "r");
