--- a/docs/CMakeLists.txt	2014-09-30 12:37:34.000000000 -0500
+++ b/docs/CMakeLists.txt	2014-11-05 18:16:15.000000000 -0600
@@ -36,5 +36,5 @@
     ${CMAKE_CURRENT_BINARY_DIR}/g-lensfun-update-data.1 
     ${CMAKE_CURRENT_BINARY_DIR}/lensfun-update-data.1 
     ${CMAKE_CURRENT_BINARY_DIR}/lensfun-add-adapter.1 
-    DESTINATION ${CMAKE_INSTALL_PREFIX}/man/man1
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/share/man/man1
 )
