diff -ruN assimp-5.2.3-orig/test/unit/utVersion.cpp assimp-5.2.3/test/unit/utVersion.cpp
--- assimp-5.2.3-orig/test/unit/utVersion.cpp	2022-03-17 05:35:44.000000000 -0500
+++ assimp-5.2.3/test/unit/utVersion.cpp	2022-03-22 06:51:34.000000000 -0500
@@ -68,10 +68,6 @@
     EXPECT_NE( aiGetCompileFlags(), 0U );
 }
 
-TEST_F( utVersion, aiGetVersionRevisionTest ) {
-    EXPECT_NE( aiGetVersionRevision(), 0U );
-}
-
 TEST_F( utVersion, aiGetBranchNameTest ) {
     EXPECT_NE( nullptr, aiGetBranchName() );
 }
