diff -ruN libemf-1.0.13-orig/libemf/libemf.cpp libemf-1.0.13/libemf/libemf.cpp
--- libemf-1.0.13-orig/libemf/libemf.cpp	2020-06-07 08:10:20.000000000 -0500
+++ libemf-1.0.13/libemf/libemf.cpp	2025-04-20 05:28:39.000000000 -0500
@@ -21,6 +21,13 @@
 #include <iostream>
 #include <climits>
 
+#ifdef __APPLE__
+#include <libkern/OSByteOrder.h>
+#define bswap_32(x) OSSwapInt32(x)
+#else
+#include <byteswap.h>
+#endif
+
 #include "libemf.h"
 
 namespace EMF {
@@ -72,7 +79,6 @@
     if ( not bigEndian() ) {
       return a;
     }
-#include <byteswap.h>
     return bswap_32(a);
   }
 
