diff -ruN graphviz-12.2.1-orig/tclpkg/gdtclft/Makefile.in graphviz-12.2.1/tclpkg/gdtclft/Makefile.in
--- graphviz-12.2.1-orig/tclpkg/gdtclft/Makefile.in	2024-12-06 17:56:08
+++ graphviz-12.2.1/tclpkg/gdtclft/Makefile.in	2025-10-22 18:24:00
@@ -1081,7 +1081,7 @@
 @WITH_LIBGD_TRUE@@WITH_TCL_TRUE@all-local: pkgIndex.tcl
 
 @WITH_LIBGD_TRUE@@WITH_TCL_TRUE@pkgIndex.tcl:
-@WITH_LIBGD_TRUE@@WITH_TCL_TRUE@	python3 $(top_srcdir)/tclpkg/mkpkgindex.py \
+@WITH_LIBGD_TRUE@@WITH_TCL_TRUE@	$(PYTHON3) $(top_srcdir)/tclpkg/mkpkgindex.py \
 @WITH_LIBGD_TRUE@@WITH_TCL_TRUE@	  --file libgdtclft$(SHARED_LIBRARY_SUFFIX) \
 @WITH_LIBGD_TRUE@@WITH_TCL_TRUE@	  --name Gdtclft \
 @WITH_LIBGD_TRUE@@WITH_TCL_TRUE@	  --version $(VERSION)
diff -ruN graphviz-12.2.1-orig/tclpkg/gv/Makefile.in graphviz-12.2.1/tclpkg/gv/Makefile.in
--- graphviz-12.2.1-orig/tclpkg/gv/Makefile.in	2024-12-06 17:56:08
+++ graphviz-12.2.1/tclpkg/gv/Makefile.in	2025-10-22 18:29:36
@@ -2917,7 +2917,7 @@
 @WITH_SWIG_TRUE@@WITH_TCL_TRUE@all-local: pkgIndex.tcl
 
 @WITH_SWIG_TRUE@@WITH_TCL_TRUE@pkgIndex.tcl:
-@WITH_SWIG_TRUE@@WITH_TCL_TRUE@	python3 $(top_srcdir)/tclpkg/mkpkgindex.py \
+@WITH_SWIG_TRUE@@WITH_TCL_TRUE@	$(PYTHON3) $(top_srcdir)/tclpkg/mkpkgindex.py \
 @WITH_SWIG_TRUE@@WITH_TCL_TRUE@	  --file libgv_tcl$(SHARED_LIBRARY_SUFFIX) \
 @WITH_SWIG_TRUE@@WITH_TCL_TRUE@	  --name gv \
 @WITH_SWIG_TRUE@@WITH_TCL_TRUE@	  --version 0
diff -ruN graphviz-12.2.1-orig/tclpkg/tcldot/Makefile.in graphviz-12.2.1/tclpkg/tcldot/Makefile.in
--- graphviz-12.2.1-orig/tclpkg/tcldot/Makefile.in	2024-12-06 17:56:09
+++ graphviz-12.2.1/tclpkg/tcldot/Makefile.in	2025-10-22 18:22:04
@@ -1396,7 +1396,7 @@
 @WITH_TCL_TRUE@all-local: pkgIndex.tcl
 
 @WITH_TCL_TRUE@pkgIndex.tcl:
-@WITH_TCL_TRUE@	python3 $(top_srcdir)/tclpkg/mkpkgindex.py \
+@WITH_TCL_TRUE@	$(PYTHON3) $(top_srcdir)/tclpkg/mkpkgindex.py \
 @WITH_TCL_TRUE@	  --file libtcldot$(SHARED_LIBRARY_SUFFIX) \
 @WITH_TCL_TRUE@	  --name Tcldot \
 @WITH_TCL_TRUE@	  --version $(VERSION)
diff -ruN graphviz-12.2.1-orig/tclpkg/tclpathplan/Makefile.in graphviz-12.2.1/tclpkg/tclpathplan/Makefile.in
--- graphviz-12.2.1-orig/tclpkg/tclpathplan/Makefile.in	2024-12-06 17:56:09
+++ graphviz-12.2.1/tclpkg/tclpathplan/Makefile.in	2025-10-22 18:23:26
@@ -1040,7 +1040,7 @@
 @WITH_TCL_TRUE@all-local: pkgIndex.tcl
 
 @WITH_TCL_TRUE@pkgIndex.tcl:
-@WITH_TCL_TRUE@	python3 $(top_srcdir)/tclpkg/mkpkgindex.py \
+@WITH_TCL_TRUE@	$(PYTHON3) $(top_srcdir)/tclpkg/mkpkgindex.py \
 @WITH_TCL_TRUE@	  --file libtclplan$(SHARED_LIBRARY_SUFFIX) \
 @WITH_TCL_TRUE@	  --name Tclpathplan \
 @WITH_TCL_TRUE@	  --version $(VERSION)
