diff -Nurd gegl-0.1.6.orig/configure.ac gegl-0.1.6/configure.ac
--- gegl-0.1.6.orig/configure.ac	2011-02-13 12:54:23
+++ gegl-0.1.6/configure.ac	2025-10-12 22:43:26
@@ -849,7 +849,7 @@
 AC_ARG_WITH(libavformat,  [  --without-libavformat   build without libavformat support])
 
 if test "x$with_libavformat" != xno; then
-  PKG_CHECK_MODULES(AVFORMAT, libavformat libavcodec,
+  PKG_CHECK_MODULES(AVFORMAT, libavformat libavcodec libavutil,
     have_libavformat="yes",
     have_libavformat="no  (libavformat not found)")
   # verify the presence of the avformat.h header
@@ -930,7 +930,7 @@
                 [Define to 1 if the spiro library is available])
       LIBSPIRO='-lspiro',
       spiro_ok="no  (Can't include spiro.h)",
-      [#include <bezctx_intf.h>])
+      [#include <spiroentrypoints.h>])
       AC_MSG_RESULT($spiro_ok)
   fi
 fi
diff -Nurd -x'*~' gegl-0.1.6.orig/docs/Makefile.am gegl-0.1.6/docs/Makefile.am
--- gegl-0.1.6.orig/docs/Makefile.am	2010-10-08 12:51:41.000000000 -0400
+++ gegl-0.1.6/docs/Makefile.am	2011-03-22 08:36:20.000000000 -0400
@@ -6,7 +6,7 @@
 # is very small, and should probably be documented in a tutorial.
 #
 
-gtkdochtmldir = $(DESTDIR)$(datadir)/gtk-doc/html/gegl
+gtkdochtmldir = $(datadir)/gtk-doc/html/gegl
 
 HTML_FILES = \
 	operations.html		\
diff -Nurd -x'*~' gegl-0.1.6.orig/operations/Makefile-common.am gegl-0.1.6/operations/Makefile-common.am
--- gegl-0.1.6.orig/operations/Makefile-common.am	2010-10-08 12:51:42.000000000 -0400
+++ gegl-0.1.6/operations/Makefile-common.am	2011-03-22 08:24:40.000000000 -0400
@@ -21,7 +21,7 @@
 	-I$(top_builddir)/gegl/module \
 	-I$(top_srcdir)/gegl/module
 
-AM_CFLAGS = $(DEP_CFLAGS) $(BABL_CFLAGS)
+AM_CPPFLAGS += $(DEP_CFLAGS) $(BABL_CFLAGS)
 
 AM_LDFLAGS = -avoid-version -export-dynamic -module $(no_undefined)
 
diff -Nurd -x'*~' gegl-0.1.6.orig/tools/create-reference.rb gegl-0.1.6/tools/create-reference.rb
--- gegl-0.1.6.orig/tools/create-reference.rb	2010-10-08 12:51:42.000000000 -0400
+++ gegl-0.1.6/tools/create-reference.rb	2013-07-28 17:48:18.000000000 -0400
@@ -5,6 +5,11 @@
 # Use under a public domain license.
 #
 
+if RUBY_VERSION =~ /^1.9/ or RUBY_VERSION =~ /^[2345]/
+  Encoding.default_external = Encoding::UTF_8
+  Encoding.default_internal = Encoding::UTF_8
+end
+
 class Argument
     attr_accessor :name, :data_type, :doc
     def initialize
