diff -Nurd -x'*~' jasper-1.900.1.orig/Makefile.am jasper-1.900.1/Makefile.am
--- jasper-1.900.1.orig/Makefile.am	2007-01-19 16:43:14.000000000 -0500
+++ jasper-1.900.1/Makefile.am	2015-11-15 22:36:13.000000000 -0500
@@ -56,7 +56,8 @@
 # 
 # __END_OF_JASPER_LICENSE__
 
-AUTOMAKE_OPTIONS = foreign dist-zip dist-tarZ
+AUTOMAKE_OPTIONS = foreign dist-zip
+ACLOCAL_AMFLAGS = -I m4
 
 EXTRA_DIST = README INSTALL COPYRIGHT LICENSE NEWS ChangeLog data doc \
   jasper.spec.in jasper.spec acaux
diff -Nurd -x'*~' jasper-1.900.1.orig/configure.ac jasper-1.900.1/configure.ac
--- jasper-1.900.1.orig/configure.ac	2007-01-19 16:47:11.000000000 -0500
+++ jasper-1.900.1/configure.ac	2016-11-06 23:08:41.000000000 -0500
@@ -67,6 +67,7 @@
 dnl Note: Avoid the temptation to call the auxiliary directory "aux" because
 dnl Microsoft Windows will barf if you try to use "aux" as a directory name. :-(
 AC_CONFIG_AUX_DIR(acaux)
+AC_CONFIG_MACRO_DIRS([m4])
 
 # Detect the canonical host and target build environment.
 # Note: This precedes automake initialization to avoid warnings.
@@ -147,7 +148,6 @@
 
 AC_PROG_CC
 AC_PROG_INSTALL
-AC_PROG_RANLIB
 # AC_PROG_AWK
 # AC_PROG_CPP
 # AC_PROG_LN_S
@@ -245,7 +245,7 @@
 OPENGL_LIBS=""
 if test $ENABLE_OPENGL = yes; then
 	if test $HAVE_OPENGL = no; then
-		TMPLIBS="-lglut -lGL -lGLU $X_PRE_LIBS -lX11 -lXmu -lXi -lXext -lXt $X_EXTRA_LIBS $X_LIBS"
+		TMPLIBS="-lglut -lGL -lGLU $X_LIBS"
 		AC_CHECK_LIB(glut, glutInit, [HAVE_OPENGL=yes;
 		  OPENGL_LIBS=$TMPLIBS], HAVE_OPENGL=no, $TMPLIBS)
 	fi
diff -Nurd -x'*~' jasper-1.900.1.orig/src/libjasper/base/jas_getopt.c jasper-1.900.1/src/libjasper/base/jas_getopt.c
--- jasper-1.900.1.orig/src/libjasper/base/jas_getopt.c	2007-01-19 16:43:05.000000000 -0500
+++ jasper-1.900.1/src/libjasper/base/jas_getopt.c	2016-02-21 15:56:46.000000000 -0500
@@ -76,6 +76,7 @@
 
 #include "jasper/jas_getopt.h"
 #include "jasper/jas_math.h"
+#include "jasper/jas_debug.h"
 
 /******************************************************************************\
 * Global data.
diff -Nurd -x'*~' jasper-1.900.1.orig/src/libjasper/jpc/jpc_t1dec.c jasper-1.900.1/src/libjasper/jpc/jpc_t1dec.c
--- jasper-1.900.1.orig/src/libjasper/jpc/jpc_t1dec.c	2007-01-19 16:43:07.000000000 -0500
+++ jasper-1.900.1/src/libjasper/jpc/jpc_t1dec.c	2016-02-21 16:01:33.000000000 -0500
@@ -78,6 +78,7 @@
 #include "jasper/jas_fix.h"
 #include "jasper/jas_stream.h"
 #include "jasper/jas_math.h"
+#include "jasper/jas_debug.h"
 
 #include "jpc_bs.h"
 #include "jpc_mqdec.h"
diff -Nurd -x'*~' jasper-1.900.1.orig/src/libjasper/jpc/jpc_tsfb.c jasper-1.900.1/src/libjasper/jpc/jpc_tsfb.c
--- jasper-1.900.1.orig/src/libjasper/jpc/jpc_tsfb.c	2007-01-19 16:43:07.000000000 -0500
+++ jasper-1.900.1/src/libjasper/jpc/jpc_tsfb.c	2016-02-21 16:01:33.000000000 -0500
@@ -75,6 +75,7 @@
 
 #include "jasper/jas_malloc.h"
 #include "jasper/jas_seq.h"
+#include "jasper/jas_debug.h"
 
 #include "jpc_tsfb.h"
 #include "jpc_cod.h"
@@ -119,14 +120,6 @@
 	free(tsfb);
 }
 
-int jpc_tsfb_analyze(jpc_tsfb_t *tsfb, jas_seq2d_t *a)
-{
-	return (tsfb->numlvls > 0) ? jpc_tsfb_analyze2(tsfb, jas_seq2d_getref(a,
-	  jas_seq2d_xstart(a), jas_seq2d_ystart(a)), jas_seq2d_xstart(a),
-	  jas_seq2d_ystart(a), jas_seq2d_width(a),
-	  jas_seq2d_height(a), jas_seq2d_rowstep(a), tsfb->numlvls - 1) : 0;
-}
-
 int jpc_tsfb_analyze2(jpc_tsfb_t *tsfb, int *a, int xstart, int ystart,
   int width, int height, int stride, int numlvls)
 {
@@ -146,11 +139,11 @@
 	return 0;
 }
 
-int jpc_tsfb_synthesize(jpc_tsfb_t *tsfb, jas_seq2d_t *a)
+int jpc_tsfb_analyze(jpc_tsfb_t *tsfb, jas_seq2d_t *a)
 {
-	return (tsfb->numlvls > 0) ? jpc_tsfb_synthesize2(tsfb,
-	  jas_seq2d_getref(a, jas_seq2d_xstart(a), jas_seq2d_ystart(a)),
-	  jas_seq2d_xstart(a), jas_seq2d_ystart(a), jas_seq2d_width(a),
+	return (tsfb->numlvls > 0) ? jpc_tsfb_analyze2(tsfb, jas_seq2d_getref(a,
+	  jas_seq2d_xstart(a), jas_seq2d_ystart(a)), jas_seq2d_xstart(a),
+	  jas_seq2d_ystart(a), jas_seq2d_width(a),
 	  jas_seq2d_height(a), jas_seq2d_rowstep(a), tsfb->numlvls - 1) : 0;
 }
 
@@ -174,6 +167,14 @@
 	return 0;
 }
 
+int jpc_tsfb_synthesize(jpc_tsfb_t *tsfb, jas_seq2d_t *a)
+{
+	return (tsfb->numlvls > 0) ? jpc_tsfb_synthesize2(tsfb,
+	  jas_seq2d_getref(a, jas_seq2d_xstart(a), jas_seq2d_ystart(a)),
+	  jas_seq2d_xstart(a), jas_seq2d_ystart(a), jas_seq2d_width(a),
+	  jas_seq2d_height(a), jas_seq2d_rowstep(a), tsfb->numlvls - 1) : 0;
+}
+
 int jpc_tsfb_getbands(jpc_tsfb_t *tsfb, uint_fast32_t xstart,
   uint_fast32_t ystart, uint_fast32_t xend, uint_fast32_t yend,
   jpc_tsfb_band_t *bands)
diff -Nurd -x'*~' jasper-1.900.1.orig/src/libjasper/mif/mif_cod.c jasper-1.900.1/src/libjasper/mif/mif_cod.c
--- jasper-1.900.1.orig/src/libjasper/mif/mif_cod.c	2007-01-19 16:43:05.000000000 -0500
+++ jasper-1.900.1/src/libjasper/mif/mif_cod.c	2016-02-21 16:02:16.000000000 -0500
@@ -70,6 +70,7 @@
 #include "jasper/jas_image.h"
 #include "jasper/jas_string.h"
 #include "jasper/jas_malloc.h"
+#include "jasper/jas_debug.h"
 
 #include "mif_cod.h"
 
diff -Nurd -x'*~' jasper-1.900.1.orig/src/libjasper/pnm/pnm_dec.c jasper-1.900.1/src/libjasper/pnm/pnm_dec.c
--- jasper-1.900.1.orig/src/libjasper/pnm/pnm_dec.c	2007-01-19 16:43:05.000000000 -0500
+++ jasper-1.900.1/src/libjasper/pnm/pnm_dec.c	2016-02-21 16:02:16.000000000 -0500
@@ -79,6 +79,7 @@
 #include "jasper/jas_types.h"
 #include "jasper/jas_stream.h"
 #include "jasper/jas_image.h"
+#include "jasper/jas_debug.h"
 
 #include "pnm_cod.h"
 
