diff -Nurd -x'*~' jasper-1.900.1.orig/Makefile.am jasper-1.900.1/Makefile.am
--- jasper-1.900.1.orig/Makefile.am	2007-01-19 16:43:14.000000000 -0500
+++ jasper-1.900.1/Makefile.am	2015-11-15 22:36:13.000000000 -0500
@@ -56,7 +56,8 @@
 # 
 # __END_OF_JASPER_LICENSE__
 
-AUTOMAKE_OPTIONS = foreign dist-zip dist-tarZ
+AUTOMAKE_OPTIONS = foreign dist-zip
+ACLOCAL_AMFLAGS = -I m4
 
 EXTRA_DIST = README INSTALL COPYRIGHT LICENSE NEWS ChangeLog data doc \
   jasper.spec.in jasper.spec acaux
diff -Nurd -x'*~' jasper-1.900.1.orig/configure.ac jasper-1.900.1/configure.ac
--- jasper-1.900.1.orig/configure.ac	2007-01-19 16:47:11.000000000 -0500
+++ jasper-1.900.1/configure.ac	2015-11-15 22:37:44.000000000 -0500
@@ -67,6 +67,7 @@
 dnl Note: Avoid the temptation to call the auxiliary directory "aux" because
 dnl Microsoft Windows will barf if you try to use "aux" as a directory name. :-(
 AC_CONFIG_AUX_DIR(acaux)
+AC_CONFIG_MACRO_DIRS([m4])
 
 # Detect the canonical host and target build environment.
 # Note: This precedes automake initialization to avoid warnings.
@@ -147,7 +148,6 @@
 
 AC_PROG_CC
 AC_PROG_INSTALL
-AC_PROG_RANLIB
 # AC_PROG_AWK
 # AC_PROG_CPP
 # AC_PROG_LN_S
