diff -Nurd -x'*~' openjpeg-version.2.1.orig/CMakeLists.txt openjpeg-version.2.1/CMakeLists.txt
--- openjpeg-version.2.1.orig/CMakeLists.txt	2014-04-29 08:58:10.000000000 -0400
+++ openjpeg-version.2.1/CMakeLists.txt	2016-01-18 00:30:16.000000000 -0500
@@ -146,6 +146,10 @@
   set(OPENJPEG_INSTALL_PACKAGE_DIR "${OPENJPEG_INSTALL_LIB_DIR}/${OPENJPEG_INSTALL_SUBDIR}")
 endif()
 
+if (APPLE)
+	list(APPEND OPENJPEG_LIBRARY_PROPERTIES INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${OPENJPEG_INSTALL_LIB_DIR}")
+endif()
+
 #-----------------------------------------------------------------------------
 # Big endian test:
 include (${CMAKE_ROOT}/Modules/TestBigEndian.cmake)
