diff -Nurd OpenEXR.orig/OpenEXR.pc.in OpenEXR/OpenEXR.pc.in
--- OpenEXR.orig/OpenEXR.pc.in	2020-06-14 02:30:34.000000000 +0100
+++ OpenEXR/OpenEXR.pc.in	2020-12-28 14:28:49.000000000 +0000
@@ -4,10 +4,10 @@
 ##
 
 prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
-OpenEXR_includedir=@includedir@/OpenEXR
+exec_prefix=@prefix@/@exec_prefix@
+libdir=@prefix@/@libdir@
+includedir=@prefix@/@includedir@
+OpenEXR_includedir=@prefix@/@includedir@/OpenEXR
 libsuffix=@LIB_SUFFIX_DASH@
 
 Name: OpenEXR
