diff -Nurd podofo-0.10.5.orig/CMakeLists.txt podofo-0.10.5/CMakeLists.txt
--- podofo-0.10.5.orig/CMakeLists.txt	2025-05-26 10:32:47
+++ podofo-0.10.5/CMakeLists.txt	2026-08-01 14:35:36
@@ -248,6 +248,7 @@
 
 if(PODOFO_BUILD_TOOLS)
     add_subdirectory(tools)
+    add_subdirectory(man)
 endif()
 
 # Enable packaging
diff -Nurd podofo-0.10.5.orig/man/CMakeLists.txt podofo-0.10.5/man/CMakeLists.txt
--- podofo-0.10.5.orig/man/CMakeLists.txt	2025-05-26 10:32:47
+++ podofo-0.10.5/man/CMakeLists.txt	2026-08-01 14:34:32
@@ -17,5 +17,5 @@
   podofotxtextract.1
   podofouncompress.1
   podofoxmp.1
-  DESTINATION ${MANDIR}man1
+  DESTINATION ${CMAKE_INSTALL_MANDIR}/man1
   )
