diff -Nurd podofo-1.0.1.orig/CMakeLists.txt podofo-1.0.1/CMakeLists.txt
--- podofo-1.0.1.orig/CMakeLists.txt	2025-06-21 17:47:21.000000000 -0400
+++ podofo-1.0.1/CMakeLists.txt	2025-08-16 18:25:39.000000000 -0400
@@ -287,6 +287,7 @@
 
 if(PODOFO_BUILD_UNSUPPORTED_TOOLS)
     add_subdirectory(tools)
+    add_subdirectory(man)
 endif()
 
 # Enable packaging
diff -Nurd podofo-1.0.1.orig/src/podofo/libpodofo.pc.in podofo-1.0.1/src/podofo/libpodofo.pc.in
--- podofo-1.0.1.orig/src/podofo/libpodofo.pc.in	2025-06-21 17:47:21.000000000 -0400
+++ podofo-1.0.1/src/podofo/libpodofo.pc.in	2025-08-16 22:46:38.000000000 -0400
@@ -6,7 +6,5 @@
 Name: @CMAKE_PROJECT_NAME@
 Description: A C++ library to work with the PDF file format
 Version: @PODOFO_VERSION@
-Requires.private: @PODOFO_PKGCONFIG_REQUIRES_PRIVATE@
 Libs: -L${libdir} -lpodofo
-Libs.private: @PODOFO_PACKAGE_CONFIG_LIBS@
 Cflags: -I${includedir}
