diff -Nurd tesseract-5.5.2.orig/Makefile.am tesseract-5.5.2/Makefile.am
--- tesseract-5.5.2.orig/Makefile.am	2025-12-26 12:27:50
+++ tesseract-5.5.2/Makefile.am	2026-05-16 02:36:06
@@ -107,7 +107,8 @@
 libtesseract_la_CPPFLAGS += $(libcurl_CFLAGS)
 
 lib_LTLIBRARIES = libtesseract.la
-libtesseract_la_LDFLAGS = $(LEPTONICA_LIBS)
+libtesseract_la_LDFLAGS = $(AM_LDFLAGS)
+libtesseract_la_LDFLAGS += $(LEPTONICA_LIBS)
 libtesseract_la_LDFLAGS += $(libarchive_LIBS)
 libtesseract_la_LDFLAGS += $(libcurl_LIBS)
 if T_WIN
@@ -143,7 +144,7 @@
 if OPENMP_SIMD
 libtesseract_native_la_CXXFLAGS += -fopenmp-simd -DOPENMP_SIMD
 endif
-libtesseract_native_la_CXXFLAGS += -I$(top_srcdir)/src/ccutil
+libtesseract_native_la_CPPFLAGS = $(AM_CPPFLAGS) -I$(top_srcdir)/src/ccutil
 libtesseract_native_la_SOURCES = src/arch/dotproduct.cpp
 
 if HAVE_AVX
diff -Nurd tesseract-5.5.2.orig/configure.ac tesseract-5.5.2/configure.ac
--- tesseract-5.5.2.orig/configure.ac	2025-12-26 12:27:50
+++ tesseract-5.5.2/configure.ac	2026-05-16 02:04:19
@@ -516,7 +516,8 @@
   fi
 ])
 
-AM_CONDITIONAL([ENABLE_TRAINING], true)
+AM_CONDITIONAL([ENABLE_TRAINING], false)
+AC_MSG_WARN([Package force-disables building Training tools.])
 
 # Check availability of ICU packages.
 PKG_CHECK_MODULES([ICU_UC], [icu-uc >= 52.1], [have_icu_uc=true], [have_icu_uc=false])
diff -Nurd tesseract-5.5.2.orig/src/arch/simddetect.cpp tesseract-5.5.2/src/arch/simddetect.cpp
--- tesseract-5.5.2.orig/src/arch/simddetect.cpp	2025-12-26 12:27:50
+++ tesseract-5.5.2/src/arch/simddetect.cpp	2026-05-16 02:28:12
@@ -273,7 +273,7 @@
     // SSE detected.
     SetDotProduct(DotProductSSE, &IntSimdMatrix::intSimdMatrixSSE);
 #endif
-#if defined(HAVE_NEON) || defined(__aarch64__)
+#if defined(HAVE_NEON) && defined(__aarch64__)
   } else if (neon_available_) {
     // NEON detected.
     SetDotProduct(DotProductNEON, &IntSimdMatrix::intSimdMatrixNEON);
@@ -334,7 +334,7 @@
   } else if (dotproduct == "accelerate") {
     SetDotProduct(DotProductAccelerate, IntSimdMatrix::intSimdMatrix);
 #endif
-#if defined(HAVE_NEON) || defined(__aarch64__)
+#if defined(HAVE_NEON) && defined(__aarch64__)
   } else if (dotproduct == "neon" && neon_available_) {
     // NEON selected by config variable.
     SetDotProduct(DotProductNEON, &IntSimdMatrix::intSimdMatrixNEON);
