diff -ruN tiff-4.5.0-orig/configure tiff-4.5.0/configure
--- tiff-4.5.0-orig/configure	2022-12-13 08:18:30.000000000 -0600
+++ tiff-4.5.0/configure	2023-02-26 11:45:36.000000000 -0600
@@ -20616,7 +20616,7 @@
 CPPFLAGS="${GL_CFLAGS} ${CPPFLAGS}"
 ax_save_LIBS="${LIBS}"
 LIBS=""
-ax_check_libs="-lopengl32 -lGL"
+ax_check_libs="-lopengl32"
 for ax_lib in ${ax_check_libs}; do
   if test X$ax_compiler_ms = Xyes; then :
   ax_try_lib=`echo $ax_lib | sed -e 's/^-l//' -e 's/$/.lib/'`
@@ -20831,7 +20831,7 @@
   ax_cv_check_glu_libglu=yes
 else
   LIBS=""
-ax_check_libs="-lglu32 -lGLU"
+ax_check_libs="-lglu32"
 for ax_lib in ${ax_check_libs}; do
   if test X$ax_compiler_ms = Xyes; then :
   ax_try_lib=`echo $ax_lib | sed -e 's/^-l//' -e 's/$/.lib/'`
@@ -20986,7 +20986,7 @@
 CPPFLAGS="${GLUT_CFLAGS} ${CPPFLAGS}"
 ax_save_LIBS="${LIBS}"
 LIBS=""
-ax_check_libs="-lglut32 -lglut"
+ax_check_libs="-lglut32"
 for ax_lib in ${ax_check_libs}; do
   if test X$ax_compiler_ms = Xyes; then :
   ax_try_lib=`echo $ax_lib | sed -e 's/^-l//' -e 's/$/.lib/'`
@@ -21074,7 +21074,7 @@
 
 
 
-  if test "$no_x" != "yes" -a "$no_gl" != "yes" \
+  if test "$no_gl" != "yes" \
       -a "$no_glu" != "yes" -a "$no_glut" != "yes" ; then
     HAVE_OPENGL=yes
   fi
