diff -Nurd libwebp-1.6.0.orig/configure libwebp-1.6.0/configure
--- libwebp-1.6.0.orig/configure	2025-07-09 16:53:33
+++ libwebp-1.6.0/configure	2025-11-26 19:45:34
@@ -14866,8 +14866,7 @@
 fi
 
 
-if test "x$enable_avx2" != "xno" -a "x$enable_sse4_1" != "xno"
-      -a "x$enable_sse2" != "xno"
+if test "x$enable_avx2" != "xno" -a "x$enable_sse4_1" != "xno" -a "x$enable_sse2" != "xno"
 then :
 
   AVX2_FLAGS="$INTRINSICS_CFLAGS $AVX2_FLAGS"
diff -Nurd libwebp-1.6.0.orig/examples/Makefile.in libwebp-1.6.0/examples/Makefile.in
--- libwebp-1.6.0.orig/examples/Makefile.in	2025-07-09 16:53:34
+++ libwebp-1.6.0/examples/Makefile.in	2025-11-26 20:04:11
@@ -91,11 +91,11 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-bin_PROGRAMS = $(am__EXEEXT_1) $(am__EXEEXT_2) $(am__EXEEXT_3) \
+bin_PROGRAMS = $(am__EXEEXT_anim) $(am__EXEEXT_1) $(am__EXEEXT_2) $(am__EXEEXT_3) \
 	$(am__EXEEXT_4) $(am__EXEEXT_5) $(am__EXEEXT_6)
 @BUILD_DEMUX_TRUE@am__append_1 = dwebp cwebp
-@BUILD_ANIMDIFF_TRUE@noinst_PROGRAMS = anim_diff$(EXEEXT) \
-@BUILD_ANIMDIFF_TRUE@	anim_dump$(EXEEXT)
+@BUILD_ANIMDIFF_TRUE@am__append_anim = anim_diff \
+@BUILD_ANIMDIFF_TRUE@	anim_dump
 @BUILD_GIF2WEBP_TRUE@am__append_2 = gif2webp
 @BUILD_IMG2WEBP_TRUE@am__append_3 = img2webp
 @BUILD_MUX_TRUE@am__append_4 = webpmux
@@ -120,6 +120,7 @@
 CONFIG_HEADER = $(top_builddir)/src/webp/config.h
 CONFIG_CLEAN_FILES =
 CONFIG_CLEAN_VPATH_FILES =
+@BUILD_ANIMDIFF_TRUE@am__EXEEXT_anim = anim_diff$(EXEEXT) anim_dump$(EXEEXT)
 @BUILD_DEMUX_TRUE@am__EXEEXT_1 = dwebp$(EXEEXT) cwebp$(EXEEXT)
 @BUILD_GIF2WEBP_TRUE@am__EXEEXT_2 = gif2webp$(EXEEXT)
 @BUILD_IMG2WEBP_TRUE@am__EXEEXT_3 = img2webp$(EXEEXT)
