diff -Nurd libwmf-0.2.13.orig/configure libwmf-0.2.13/configure
--- libwmf-0.2.13.orig/configure	2023-04-17 07:39:47
+++ libwmf-0.2.13/configure	2025-10-22 00:45:30
@@ -15407,8 +15407,8 @@
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 
-		GDK_PIXBUF_VERSION=`$PKG_CONFIG --variable=gdk_pixbuf_binary_version gdk-pixbuf-2.0`
-		GDK_PIXBUF_DIR="gdk-pixbuf-2.0/$GDK_PIXBUF_VERSION/loaders"
+		GTK_VERSION=`$PKG_CONFIG --variable=gtk_binary_version gtk+-2.0`
+		GDK_PIXBUF_DIR="gtk-2.0/$GTK_VERSION/loaders"
 		wmf_gdk_pixbuf=yes
 
 fi
diff -Nurd libwmf-0.2.13.orig/fonts/Makefile.in libwmf-0.2.13/fonts/Makefile.in
--- libwmf-0.2.13.orig/fonts/Makefile.in	2023-04-17 07:39:47
+++ libwmf-0.2.13/fonts/Makefile.in	2025-10-22 17:31:35
@@ -305,7 +305,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-fontdir = $(datadir)/libwmf/fonts/
+fontdir = @WMF_FONTDIR@
 bin_SCRIPTS = libwmf-fontmap
 FONTS = \
 	NimbusMonoPS-Bold.afm \
diff -Nurd libwmf-0.2.13.orig/libwmf.pc.in libwmf-0.2.13/libwmf.pc.in
--- libwmf-0.2.13.orig/libwmf.pc.in	2023-04-17 07:39:47
+++ libwmf-0.2.13/libwmf.pc.in	2025-10-20 00:29:35
@@ -6,5 +6,5 @@
 Name: libwmf
 Description: A library for reading and converting Windows MetaFile vector graphics (WMF)
 Version: @LIBWMF_VERSION@
-Libs: -lwmf -lwmflite @WMF_LIBFLAGS@
-Cflags: @WMF_CONFIG_CFLAGS@
+Libs: -lwmf -lwmflite -L${libdir}
+Cflags: -I${includedir}
diff -Nurd libwmf-0.2.13.orig/src/convert/Makefile.in libwmf-0.2.13/src/convert/Makefile.in
--- libwmf-0.2.13.orig/src/convert/Makefile.in	2023-04-17 07:39:47
+++ libwmf-0.2.13/src/convert/Makefile.in	2025-10-20 00:23:51
@@ -237,7 +237,7 @@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 LD = @LD@
-LDFLAGS = -L$(top_builddir)/src/.libs @WMF_LIBFLAGS@ @LDFLAGS@
+LDFLAGS = @WMF_LIBFLAGS@ @LDFLAGS@
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LIBTOOL = @LIBTOOL@
diff -Nurd libwmf-0.2.13.orig/src/font.c libwmf-0.2.13/src/font.c
--- libwmf-0.2.13.orig/src/font.c	2023-04-17 07:39:47
+++ libwmf-0.2.13/src/font.c	2025-10-20 00:23:51
@@ -199,7 +199,7 @@
 	unsigned int i = 0;
 
 	while (font_data->cache[i].name)
-	{	if (strcmp (font_data->cache[i].path,entry->path) == 0) break;
+	{	if (strcmp (font_data->cache[i].name,entry->name) == 0) break;
 		i++;
 	}
 
@@ -422,7 +422,6 @@
 
 	fontmap_data->FD.FI = 0;
 
-	if (API->flags & WMF_OPT_SYS_FONTS)
 	{	if (API->flags & WMF_OPT_SYS_FONTMAP)
 		{	wmf_ipa_font_map_xml (API,&(fontmap_data->FD),options->sys_fontmap_file);
 		}
diff -Nurd libwmf-0.2.13.orig/src/player.c libwmf-0.2.13/src/player.c
--- libwmf-0.2.13.orig/src/player.c	2023-04-17 07:39:47
+++ libwmf-0.2.13/src/player.c	2025-10-20 00:23:51
@@ -23,6 +23,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdint.h>
 #include <string.h>
 #include <math.h>
 
