diff -ruN MPlayer-1.5-orig/VERSION MPlayer-1.5/VERSION
--- MPlayer-1.5-orig/VERSION	2022-02-27 09:27:41.000000000 -0600
+++ MPlayer-1.5/VERSION	2022-04-02 05:58:43.000000000 -0500
@@ -1 +1 @@
-1.5
+1.5 (Fink Project)
diff -ruN MPlayer-1.5-orig/configure MPlayer-1.5/configure
--- MPlayer-1.5-orig/configure	2022-02-27 03:09:00.000000000 -0600
+++ MPlayer-1.5/configure	2022-04-02 05:58:43.000000000 -0500
@@ -4932,8 +4932,7 @@
     fi
   done
   if test $_cross_compile = no; then
-    for I in /usr/X11/include /usr/X11R7/include /usr/local/include /usr/X11R6/include \
-             /usr/include/X11R6 /usr/openwin/include ; do
+    for I in /opt/X11/include /usr/X11/include ; do
       if test -f "$I/X11/Xlib.h" ; then
         extra_cflags="$extra_cflags -I$I"
         _x11_headers="yes"
@@ -4947,9 +4946,7 @@
 
 echocheck "X11"
 if test "$_x11" = auto && test "$_x11_headers" = yes ; then
-  for I in "" -L/usr/X11R7/lib -L/usr/local/lib -L/usr/X11R6/lib -L/usr/lib/X11R6 \
-           -L/usr/X11/lib -L/usr/lib32 -L/usr/openwin/lib -L/usr/local/lib64 -L/usr/X11R6/lib64 \
-           -L/usr/lib ; do
+  for I in "" -L/opt/X11/lib -L/usr/X11/lib ; do
     if netbsd; then
       ld_tmp="$I -lXext -lX11 $ld_pthread -Wl,-R$(echo $I | sed s/^-L//)"
     else
@@ -5346,7 +5343,7 @@
 return 0; }
 EOF
   _aa=no
-  for ld_tmp in "-laa" ; do
+  for ld_tmp in "-L@FINK_PREFIX@/lib -laa" ; do
     cc_check $ld_tmp && libs_mplayer="$libs_mplayer $ld_tmp" && _aa=yes && break
   done
 fi
@@ -5619,6 +5616,17 @@
   nocodecmodules="OpenJPEG $nocodecmodules"
 fi
 
+# http://lists.mplayerhq.hu/pipermail/mplayer-dev-eng/2013-January/071352.html
+# to make sure we have the correct OPENJPEG header location
+echocheck "openjpeg-1.5/openjpeg.h"
+_openjpeg_1_5_openjpeg_h=no
+header_check openjpeg-1.5/openjpeg.h && _openjpeg_1_5_openjpeg_h=yes
+if [ $_openjpeg_1_5_openjpeg_h = yes ]; then
+  def_openjpeg_1_5_openjpeg_h='#define HAVE_OPENJPEG_1_5_OPENJPEG_H 1'
+else
+  def_openjpeg_1_5_openjpeg_h='#define HAVE_OPENJPEG_1_5_OPENJPEG_H 0'
+fi
+echores "$_openjpeg_1_5_openjpeg_h"
 
 echocheck "PNM support"
 if test "$_pnm" = yes; then
@@ -9473,6 +9481,7 @@
 $def_arpa_inet_h
 $def_direct_h
 $def_io_h
+$def_openjpeg_1_5_openjpeg_h
 $def_poll_h
 $def_windows_h
 $def_emmintrin_h
diff -ruN MPlayer-1.5-orig/libmpdemux/demux_gif.c MPlayer-1.5/libmpdemux/demux_gif.c
--- MPlayer-1.5-orig/libmpdemux/demux_gif.c	2016-07-21 16:46:42.000000000 -0500
+++ MPlayer-1.5/libmpdemux/demux_gif.c	2022-04-02 05:58:43.000000000 -0500
@@ -316,6 +316,17 @@
     return NULL;
   }
 
+  // Validate image size, most code in this demuxer assumes w*h <= INT_MAX
+  if ((int64_t)gif->SWidth * gif->SHeight > INT_MAX) {
+    mp_msg(MSGT_DEMUX, MSGL_ERR,
+           "[demux_gif] Unsupported picture size %dx%d.\n", gif->SWidth,
+           gif->SHeight);
+    if (DGifCloseFile(gif) == GIF_ERROR)
+      print_gif_error(NULL);
+    free(priv);
+    return NULL;
+  }
+
   // create a new video stream header
   sh_video = new_sh_video(demuxer, 0);
 
