diff -Nurd netpbm-free-11.07.01.orig/Makefile.config.fink netpbm-free-11.07.01/Makefile.config.fink
--- netpbm-free-11.07.01.orig/Makefile.config.fink	1969-12-31 19:00:00.000000000 -0500
+++ netpbm-free-11.07.01/Makefile.config.fink	2024-09-24 02:20:35.000000000 -0400
@@ -0,0 +1,31 @@
+#
+# Configuration for fink
+# Copied from a manual run of ./configure (plus some additions)
+
+DEFAULT_TARGET	= nonmerge
+NETPBMLIBTYPE	= dylib
+NETPBMLIBSUFFIX	= dylib
+STATICLIB_TOO	= n
+NETPBMLIB_RUNTIME_PATH = @PREFIX@/lib
+CC		= cc
+CFLAGS = -O3 -ffast-math  -pedantic -fno-common -Wall -Wno-uninitialized -Wmissing-declarations -Wimplicit -Wwrite-strings -Wmissing-prototypes -Wundef -Wno-unknown-pragmas -MD
+CFLAGS_SHLIB	= -fno-common
+LDSHLIB		= -dynamiclib -install_name $(NETPBMLIB_RUNTIME_PATH)/libnetpbm.$(MAJ).dylib -compatibility_version $$(($(MAJ)+1)) -current_version $$(($(MAJ)+1)).$(MIN)
+LINKERISCOMPILER	= Y
+CFLAGS_SHLIB	+= -fPIC
+TIFFLIB		= libtiff.dylib
+JPEGLIB		= @PREFIX@/lib/libjpeg8.2-turbo/lib/libjpeg.8.dylib
+PNGLIB		= libpng.dylib
+ZLIB		= libz.dylib
+JBIGLIB		= libjbig.dylib
+JASPERLIB       = libjasper.dylib
+TIFFHDR_DIR	= @PREFIX@/include
+JPEGHDR_DIR	= @PREFIX@/lib/libjpeg8.2-turbo/include
+PNGHDR_DIR	= @PREFIX@/include
+ZHDR_DIR	= @PREFIX@/include
+JBIGHDR_DIR	= @PREFIX@/include
+JASPERHDR_DIR	= @PREFIX@/include
+PKGMANDIR	= share/man
+NETPBM_DOCURL	= file:@PREFIX@/share/doc/netpbm
+WANT_SSE	= Y
+RGB_DB_PATH	= @PREFIX@/share/netpbm/rgb.txt:/opt/X11/share/X11/rgb.txt:/usr/lib/X11/rgb.txt:/usr/X11/share/X11/rgb.txt:/usr/X11R6/lib/X11/rgb.txt
diff -Nurd netpbm-free-11.07.01.orig/buildtools/installnetpbm.pl netpbm-free-11.07.01/buildtools/installnetpbm.pl
--- netpbm-free-11.07.01.orig/buildtools/installnetpbm.pl	2024-06-28 19:33:22.000000000 -0400
+++ netpbm-free-11.07.01/buildtools/installnetpbm.pl	2024-09-23 23:47:16.000000000 -0400
@@ -183,7 +183,7 @@
     } elsif ($ENV{OSTYPE} && $ENV{OSTYPE} eq "msdosdjgpp") {
         $default = "/dev/env/DJDIR";
     } else {
-        $default = "/usr/local/netpbm";
+        $default = "@PREFIX@";
     }
 
     my $response = fsObjPrompt("install prefix", $default);
diff -Nurd netpbm-free-11.07.01.orig/buildtools/manpage.mk netpbm-free-11.07.01/buildtools/manpage.mk
--- netpbm-free-11.07.01.orig/buildtools/manpage.mk	2017-09-30 14:51:40.000000000 -0400
+++ netpbm-free-11.07.01/buildtools/manpage.mk	2024-09-23 23:47:16.000000000 -0400
@@ -171,9 +171,9 @@
 .PHONY : installman
 installman: report
 	set -x
-	for f in $(wildcard $(MAN1)); do if [ -f $$f ]; then gzip <$$f >$(MANDIR)/man1/$$f.gz; fi; done
-	for f in $(wildcard $(MAN3)); do if [ -f $$f ]; then gzip <$$f >$(MANDIR)/man3/$$f.gz; fi; done
-	for f in $(wildcard $(MAN5)); do if [ -f $$f ]; then gzip <$$f >$(MANDIR)/man5/$$f.gz; fi; done
+	for f in $(wildcard $(MAN1)); do install -d -m 0755 $(MANDIR)/man1; if [ -f $$f ]; then gzip <$$f >$(MANDIR)/man1/$$f.gz; fi; done
+	for f in $(wildcard $(MAN3)); do install -d -m 0755 $(MANDIR)/man3; if [ -f $$f ]; then gzip <$$f >$(MANDIR)/man3/$$f.gz; fi; done
+	for f in $(wildcard $(MAN5)); do install -d -m 0755 $(MANDIR)/man5; if [ -f $$f ]; then gzip <$$f >$(MANDIR)/man5/$$f.gz; fi; done
 
 
 # This will uninstall the man pages.
diff -Nurd netpbm-free-11.07.01.orig/converter/other/pnmtopng.c netpbm-free-11.07.01/converter/other/pnmtopng.c
--- netpbm-free-11.07.01.orig/converter/other/pnmtopng.c	2024-03-30 14:31:01.000000000 -0400
+++ netpbm-free-11.07.01/converter/other/pnmtopng.c	2024-09-23 23:47:16.000000000 -0400
@@ -551,7 +551,7 @@
                       unsigned int * const bestMatchP) {
 
     unsigned int paletteIndex;
-    unsigned int bestIndex;
+    unsigned int bestIndex = 0;
     unsigned int bestMatch;
 
     assert(paletteSize > 0);
@@ -1635,9 +1635,9 @@
     colorhist_vector chv;
     unsigned int colors;
 
-    gray * alphas_of_color[MAXPALETTEENTRIES];
+    gray * alphas_of_color[MAXPALETTEENTRIES + 1];
     unsigned int alphas_first_index[MAXPALETTEENTRIES];
-    unsigned int alphas_of_color_cnt[MAXPALETTEENTRIES];
+    unsigned int alphas_of_color_cnt[MAXPALETTEENTRIES + 1];
 
     getChv(ifP, rasterPos, cols, rows, maxval, format, MAXCOLORS,
            &chv, &colors);
@@ -2732,7 +2732,7 @@
         /* The color part of the color/alpha palette passed to the PNG
            compressor
         */
-    unsigned int paletteSize;
+    unsigned int paletteSize = MAXCOLORS;
 
     gray transPnm[MAXCOLORS];
     png_byte  trans[MAXCOLORS];
