diff -ruN opencv-4.6.0-orig/3rdparty/ippicv/ippicv.cmake opencv-4.6.0/3rdparty/ippicv/ippicv.cmake
--- opencv-4.6.0-orig/3rdparty/ippicv/ippicv.cmake	2022-06-05 10:32:44.000000000 -0500
+++ opencv-4.6.0/3rdparty/ippicv/ippicv.cmake	2022-11-24 05:41:15.000000000 -0600
@@ -8,7 +8,7 @@
     set(OPENCV_ICV_PLATFORM "macosx")
     set(OPENCV_ICV_PACKAGE_SUBDIR "ippicv_mac")
     set(OPENCV_ICV_NAME "ippicv_2020_mac_intel64_20191018_general.tgz")
-    set(OPENCV_ICV_HASH "1c3d675c2a2395d094d523024896e01b")
+    set(OPENCV_ICV_HASH "d41d8cd98f00b204e9800998ecf8427e")
   elseif((UNIX AND NOT ANDROID) OR (UNIX AND ANDROID_ABI MATCHES "x86"))
     set(OPENCV_ICV_PLATFORM "linux")
     set(OPENCV_ICV_PACKAGE_SUBDIR "ippicv_lnx")
diff -ruN opencv-4.6.0-orig/cmake/OpenCVFindOpenEXR.cmake opencv-4.6.0/cmake/OpenCVFindOpenEXR.cmake
--- opencv-4.6.0-orig/cmake/OpenCVFindOpenEXR.cmake	2022-06-05 10:32:44.000000000 -0500
+++ opencv-4.6.0/cmake/OpenCVFindOpenEXR.cmake	2022-11-24 05:47:38.000000000 -0600
@@ -42,10 +42,7 @@
 
 SET(SEARCH_PATHS
     "${OPENEXR_ROOT}"
-    /usr
-    /usr/local
-    /sw
-    /opt
+    @FINK_PREFIX@
     "${ProgramFiles_ENV_PATH}/OpenEXR")
 
 MACRO(FIND_OPENEXR_LIBRARY LIBRARY_NAME LIBRARY_SUFFIX)
