diff -ruN opencv-3.4.19-orig/cmake/OpenCVFindLibsVideo.cmake opencv-3.4.19/cmake/OpenCVFindLibsVideo.cmake
--- opencv-3.4.19-orig/cmake/OpenCVFindLibsVideo.cmake	2022-12-26 21:50:12.000000000 -0600
+++ opencv-3.4.19/cmake/OpenCVFindLibsVideo.cmake	2023-03-21 04:37:43.000000000 -0500
@@ -67,7 +67,7 @@
 ocv_clear_vars(HAVE_PVAPI)
 if(WITH_PVAPI)
   find_path(PVAPI_INCLUDE_PATH "PvApi.h"
-            PATHS /usr/local /opt /usr ENV ProgramFiles ENV ProgramW6432
+            PATHS @FINK_PREFIX@/lib/ffmpeg-@FFMPEG_VER@ /usr ENV ProgramFiles ENV ProgramW6432
             PATH_SUFFIXES include "Allied Vision Technologies/GigESDK/inc-pc" "AVT GigE SDK/inc-pc" "GigESDK/inc-pc"
             DOC "The path to PvAPI header")
 
@@ -99,7 +99,7 @@
 ocv_clear_vars(HAVE_GIGE_API)
 if(WITH_GIGEAPI)
   find_path(GIGEAPI_INCLUDE_PATH "GigEVisionSDK.h"
-            PATHS /usr/local /var /opt /usr ENV ProgramFiles ENV ProgramW6432
+            PATHS @FINK_PREFIX@/lib/ffmpeg-@FFMPEG_VER@ /usr ENV ProgramFiles ENV ProgramW6432
             PATH_SUFFIXES include "Smartek Vision Technologies/GigEVisionSDK/gige_cpp" "GigEVisionSDK/gige_cpp" "GigEVisionSDK/gige_c"
             DOC "The path to Smartek GigEVisionSDK header")
   FIND_LIBRARY(GIGEAPI_LIBRARIES NAMES GigEVisionSDK)
@@ -114,7 +114,7 @@
   ocv_check_modules(ARAVIS_GLIB glib-2.0)
   if(HAVE_ARAVIS_GLIB)
     find_path(ARAVIS_INCLUDE_PATH "arv.h"
-              PATHS /usr/local /var /opt /usr ENV ProgramFiles ENV ProgramW6432
+              PATHS @FINK_PREFIX@/lib/ffmpeg-@FFMPEG_VER@ /usr ENV ProgramFiles ENV ProgramW6432
               PATH_SUFFIXES include "aravis-0.6" "aravis-0.4"
               DOC "The path to Aravis SDK headers")
     find_library(ARAVIS_LIBRARIES NAMES "aravis-0.6" "aravis-0.4" )
diff -ruN opencv-3.4.19-orig/cmake/OpenCVFindOpenEXR.cmake opencv-3.4.19/cmake/OpenCVFindOpenEXR.cmake
--- opencv-3.4.19-orig/cmake/OpenCVFindOpenEXR.cmake	2022-12-26 21:50:12.000000000 -0600
+++ opencv-3.4.19/cmake/OpenCVFindOpenEXR.cmake	2023-03-21 04:40:19.000000000 -0500
@@ -47,10 +47,7 @@
 
 SET(SEARCH_PATHS
     "${OPENEXR_ROOT}"
-    /usr
-    /usr/local
-    /sw
-    /opt
+    @FINK_PREFIX@
     "${ProgramFiles_ENV_PATH}/OpenEXR")
 
 MACRO(FIND_OPENEXR_LIBRARY LIBRARY_NAME LIBRARY_SUFFIX)
