diff -Nurd -x'*~' PDFlib-Lite-7.0.4.orig/configure PDFlib-Lite-7.0.4/configure
--- PDFlib-Lite-7.0.4.orig/configure	2009-03-18 10:10:04.000000000 -0400
+++ PDFlib-Lite-7.0.4/configure	2009-04-23 02:20:20.000000000 -0400
@@ -3666,8 +3666,8 @@
 	fi
 	CPPFLAGS="$CPPFLAGS -I/Developer/Headers/FlatCarbon"
 	LDFLAGS="$LDFLAGS -framework ApplicationServices"
-	MAC_CFLAGS="$MAC_CFLAGS -isysroot $DEVSDK"
-	MAC_LDFLAGS="$MAC_LDFLAGS -Wl,-syslibroot,$DEVSDK"
+#	MAC_CFLAGS="$MAC_CFLAGS -isysroot $DEVSDK"
+#	MAC_LDFLAGS="$MAC_LDFLAGS -Wl,-syslibroot,$DEVSDK"
 
 	# set PLATFROM ID
 	# will be completed in pc_config.h (32/64 ppc/intel)
