diff -Nurd PDFlib-Lite-7.0.5p3.orig/configure PDFlib-Lite-7.0.5p3/configure
--- PDFlib-Lite-7.0.5p3.orig/configure	2012-06-06 07:58:57
+++ PDFlib-Lite-7.0.5p3/configure	2025-11-09 07:53:58
@@ -3687,8 +3687,8 @@
 	fi
 	CPPFLAGS="$CPPFLAGS -I/Developer/Headers/FlatCarbon"
 	LDFLAGS="$LDFLAGS -framework ApplicationServices"
-	MAC_CFLAGS="$MAC_CFLAGS -isysroot $DEVSDK"
-	MAC_LDFLAGS="$MAC_LDFLAGS -Wl,-syslibroot,$DEVSDK"
+#	MAC_CFLAGS="$MAC_CFLAGS -isysroot $DEVSDK"
+#	MAC_LDFLAGS="$MAC_LDFLAGS -Wl,-syslibroot,$DEVSDK"
 
 	# set PLATFROM ID
 	# will be completed in pc_config.h (32/64 ppc/intel)
@@ -3720,6 +3720,8 @@
 	echo "$as_me:$LINENO: checking Mac build Architecture" >&5
 echo $ECHO_N "checking Mac build Architecture... $ECHO_C" >&6
 	case $MAC_ARCH in
+	    automatic)
+	    ;;
 	    i386)
 		MAC_CFLAGS="$MAC_CFLAGS -arch i386"
 		MAC_LDFLAGS="$MAC_LDFLAGS -arch i386"
@@ -5045,6 +5047,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 #include <ctype.h>
+#include <stdlib.h>
 #if ((' ' & 0x0FF) == 0x020)
 # define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 # define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
@@ -9864,6 +9867,7 @@
 #endif
 
 #include <stdio.h>
+#include <stdlib.h>
 
 #ifdef RTLD_GLOBAL
 #  define LT_DLGLOBAL		RTLD_GLOBAL
@@ -9897,10 +9901,6 @@
 #  endif
 #endif
 
-#ifdef __cplusplus
-extern "C" void exit (int);
-#endif
-
 void fnord() { int i=42;}
 int main ()
 {
@@ -9964,6 +9964,7 @@
 #endif
 
 #include <stdio.h>
+#include <stdlib.h>
 
 #ifdef RTLD_GLOBAL
 #  define LT_DLGLOBAL		RTLD_GLOBAL
@@ -9995,10 +9996,6 @@
 #      endif
 #    endif
 #  endif
-#endif
-
-#ifdef __cplusplus
-extern "C" void exit (int);
 #endif
 
 void fnord() { int i=42;}
diff -Nurd PDFlib-Lite-7.0.5p3.orig/libs/pdcore/pc_config.h PDFlib-Lite-7.0.5p3/libs/pdcore/pc_config.h
--- PDFlib-Lite-7.0.5p3.orig/libs/pdcore/pc_config.h	2012-06-06 07:58:58
+++ PDFlib-Lite-7.0.5p3/libs/pdcore/pc_config.h	2025-11-09 07:41:02
@@ -142,8 +142,7 @@
 
 /* try to identify the Mac OS X command line compiler */
 
-#if defined(__APPLE__) && (defined(__ppc__) || \
-    defined(__i386__) || defined(__ppc64__) || defined(__x86_64__))
+#if defined(__APPLE__) && defined(__MACH__)
 
 #define MACOSX
 
diff -Nurd PDFlib-Lite-7.0.5p3.orig/libs/png/pngconf.h PDFlib-Lite-7.0.5p3/libs/png/pngconf.h
--- PDFlib-Lite-7.0.5p3.orig/libs/png/pngconf.h	2012-06-06 07:59:01
+++ PDFlib-Lite-7.0.5p3/libs/png/pngconf.h	2025-11-09 07:33:55
@@ -382,17 +382,7 @@
  */
 
 #if defined(PNG_FLOATING_POINT_SUPPORTED)
-#  if defined(MACOS)
-     /* We need to check that <math.h> hasn't already been included earlier
-      * as it seems it doesn't agree with <fp.h>, yet we should really use
-      * <fp.h> if possible.
-      */
-#    if !defined(__MATH_H__) && !defined(__MATH_H) && !defined(__cmath__)
-#      include <fp.h>
-#    endif
-#  else
 #    include <math.h>
-#  endif
 #  if defined(_AMIGA) && defined(__SASC) && defined(_M68881)
      /* Amiga SAS/C: We must include builtin FPU functions when compiling using
       * MATH=68881
