diff -ruN pixman-0.34.0-orig/test/utils-prng.c pixman-0.34.0/test/utils-prng.c
--- pixman-0.34.0-orig/test/utils-prng.c	2015-06-30 04:48:31.000000000 -0500
+++ pixman-0.34.0/test/utils-prng.c	2018-04-16 21:22:07.000000000 -0500
@@ -27,6 +27,10 @@
 #include "utils.h"
 #include "utils-prng.h"
 
+#ifndef __has_builtin
+#define __has_builtin(x) 0
+#endif
+
 #if defined(HAVE_GCC_VECTOR_EXTENSIONS) && defined(__SSE2__)
 #include <xmmintrin.h>
 #endif
@@ -199,7 +203,7 @@
         }
         else
         {
-#ifdef HAVE_GCC_VECTOR_EXTENSIONS
+#if defined(HAVE_GCC_VECTOR_EXTENSIONS) && __has_builtin(__builtin_shuffle)
             const uint8x16 bswap_shufflemask =
             {
                 3, 2, 1, 0, 7, 6, 5, 4, 11, 10, 9, 8, 15, 14, 13, 12
