diff -Nurd povray-3.7.0.10.orig/source/base/image/image.cpp povray-3.7.0.10/source/base/image/image.cpp
--- povray-3.7.0.10.orig/source/base/image/image.cpp	2021-07-08 05:56:01.000000000 -0400
+++ povray-3.7.0.10/source/base/image/image.cpp	2021-07-11 11:45:38.000000000 -0400
@@ -2829,7 +2829,7 @@
 			meta[0] = sizeof(pixel_type);
 			meta[1] = m_Width;
 			meta[2] = m_Height;
-			if (lseek64(m_File, pos, SEEK_SET) != pos)
+			if (lseek(m_File, pos, SEEK_SET) != pos)
 				throw POV_EXCEPTION(kFileDataErr, "Intermediate image storage backing file write/seek failed at creation.");
 			if (write(m_File, &meta[0], (int) sizeof(size_type)*3) != (sizeof(size_type)*3))
 				throw POV_EXCEPTION(kFileDataErr, "Intermediate image storage backing file write failed at creation.");
@@ -3000,7 +3000,7 @@
 #endif
 				pos = block * sizeof(pixel_type) * m_Blocksize;
 				int chunk = sizeof(pixel_type) * m_Blocksize;
-				if (lseek64(m_File, pos, SEEK_SET) != pos)
+				if (lseek(m_File, pos, SEEK_SET) != pos)
 					throw POV_EXCEPTION(kFileDataErr, "Intermediate image storage backing file read/seek failed.");
 				int bytes = read(m_File, &m_Buffer[0], chunk);
 				if (bytes != (sizeof(pixel_type) * m_Blocksize))
@@ -3023,7 +3023,7 @@
 
 			if (m_Dirty) {
 				pos = m_CurrentBlock * sizeof(pixel_type) * m_Blocksize;
-				if (lseek64(m_File, pos, SEEK_SET) != pos)
+				if (lseek(m_File, pos, SEEK_SET) != pos)
 					throw POV_EXCEPTION(kFileDataErr, "Intermediate image storage backing file write/seek failed.");
 				if (write(m_File, &m_Buffer[0], (int) sizeof(pixel_type) * m_Blocksize) != (sizeof(pixel_type) * m_Blocksize))
 					throw POV_EXCEPTION(kFileDataErr, "Intermediate image storage backing file write failed.");
diff -Nurd povray-3.7.0.10.orig/unix/prebuild.sh povray-3.7.0.10/unix/prebuild.sh
--- povray-3.7.0.10.orig/unix/prebuild.sh	2021-07-08 05:56:01.000000000 -0400
+++ povray-3.7.0.10/unix/prebuild.sh	2021-07-11 11:45:41.000000000 -0400
@@ -357,7 +357,7 @@
   do
     out=`basename $file`
     echo "Create ../$out`test -d $file && echo /`"
-    $cp_u -f -R $file ../  ||  echo "$file not copied !"
+    $cp_u -f -R $file ../$out  ||  echo "$file not copied !"
     chmod -f -R u+rw ../$out
   done
 
diff -Nurd povray-3.7.0.10.orig/vfe/unix/vfeplatform.h povray-3.7.0.10/vfe/unix/vfeplatform.h
--- povray-3.7.0.10.orig/vfe/unix/vfeplatform.h	2021-07-08 05:56:01.000000000 -0400
+++ povray-3.7.0.10/vfe/unix/vfeplatform.h	2021-07-11 11:45:41.000000000 -0400
@@ -36,6 +36,8 @@
 #ifndef __VFEPLATFORM_H__
 #define __VFEPLATFORM_H__
 
+#include <sys/types.h>
+
 #include <boost/shared_ptr.hpp>
 
 namespace vfePlatform
diff -Nurd povray-3.7.0.10.orig/vfe/vfesession.h povray-3.7.0.10/vfe/vfesession.h
--- povray-3.7.0.10.orig/vfe/vfesession.h	2021-07-08 05:56:01.000000000 -0400
+++ povray-3.7.0.10/vfe/vfesession.h	2021-07-11 11:45:41.000000000 -0400
@@ -36,6 +36,8 @@
 #ifndef __VFESESSION_H__
 #define __VFESESSION_H__
 
+#include <sys/types.h>
+
 #include <queue>
 
 #include "base/image/colourspace.h"
