diff -ruN qpdf-10.5.0-orig/libqpdf/build.mk qpdf-10.5.0/libqpdf/build.mk
--- qpdf-10.5.0-orig/libqpdf/build.mk	2021-12-21 09:21:23.000000000 -0600
+++ qpdf-10.5.0/libqpdf/build.mk	2022-02-01 04:44:24.000000000 -0600
@@ -1,8 +1,8 @@
 TARGETS_libqpdf = libqpdf/$(OUTPUT_DIR)/$(call libname,qpdf)
 
 INCLUDES_libqpdf = include libqpdf
-LDFLAGS_libqpdf = -Llibqpdf/$(OUTPUT_DIR)
-LIBS_libqpdf = -lqpdf
+LDFLAGS_libqpdf = 
+LIBS_libqpdf = libqpdf/$(OUTPUT_DIR)/libqpdf.la
 
 CRYPTO_NATIVE = \
 	libqpdf/AES_PDF_native.cc \
