diff -Nurd sane-backends-1.2.1.orig/configure sane-backends-1.2.1/configure
--- sane-backends-1.2.1.orig/configure	2023-02-05 21:12:47.000000000 -0500
+++ sane-backends-1.2.1/configure	2023-11-11 11:32:13.000000000 -0500
@@ -20138,7 +20138,8 @@
       AM_LDFLAGS="$AM_LDFLAGS -Wl,-brtl"
       ;;
     darwin*) #include frameworks
-      LIBS="$LIBS -framework CoreFoundation -framework IOKit"
+      LIBS="$LIBS -Wl,-framework,CoreFoundation -Wl,-framework,IOKit"
+      STRICT_LDFLAGS=-no-undefined
       ;;
   esac
 
@@ -25824,6 +25825,7 @@
 /* end confdefs.h.  */
 
 	#define INET6
+	#include <stdlib.h>
 	#include <sys/types.h>
 	#include <sys/socket.h>
 int
@@ -25866,6 +25868,7 @@
 /* end confdefs.h.  */
 
 	#define INET6
+	#include <stdlib.h>
 	#include <sys/types.h>
 	#include <sys/socket.h>
 int
@@ -25895,6 +25898,7 @@
 /* end confdefs.h.  */
 
 		#define INET6
+		#include <stdlib.h>
 		#include <sys/types.h>
 		#include <sys/socket.h>
 int
