diff -ruN separate+-0.5.8-orig/Makefile.macosx separate+-0.5.8/Makefile.macosx
--- separate+-0.5.8-orig/Makefile.macosx	2010-08-17 05:14:32.000000000 -0400
+++ separate+-0.5.8/Makefile.macosx	2013-09-16 19:51:36.000000000 -0400
@@ -1,8 +1,8 @@
 GIMP_VER = 2.4
 
-#USE_LCMS2 = yes
+USE_LCMS2 = yes
 
-PREFIX      = /opt/local
+PREFIX      = @FINK_PREFIX@
 INSTALLDIR  = $(PREFIX)/lib/gimp/2.0/plug-ins
 #INSTALLDIR = $(HOME)/.gimp-$(GIMP_VER)/plug-ins
 
@@ -39,7 +39,7 @@
 NLS     = -DENABLE_NLS -DGETTEXT_PACKAGE="\"$(GETTEXT_PACKAGE)\""
 CFLAGS += $(CF_INCLUDE) $(GIMP_INCLUDE) $(GTK_INCLUDE) $(LCMS_INCLUDE) $(CM) $(NLS) \
           -DUSE_ICC_BUTTON
-LIBS    = $(CF_LIB) $(GIMP_LIB) $(GTK_LIB) $(LCMS_LIB) $(TIFF_LIB) $(JPEG_LIB)
+LIBS    = $(CF_LIB) $(GIMP_LIB) $(GTK_LIB) $(LCMS_LIB) $(TIFF_LIB) $(JPEG_LIB) -lintl
 
 SOURCES = $(SEPARATE_SOURCES) $(IMPORT_SOURCES) \
           $(ICC_COLORSPACE_SOURCES) $(EXTRA_SOURCES)
