diff -Nurd -x'*~' tango-icon-theme-0.8.90.orig/configure tango-icon-theme-0.8.90/configure
--- tango-icon-theme-0.8.90.orig/configure	2009-02-25 22:08:00.000000000 -0500
+++ tango-icon-theme-0.8.90/configure	2017-03-26 22:22:52.000000000 -0400
@@ -6454,25 +6454,6 @@
 
 
 if test "x$enable_framing" = "xyes"; then
-   if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"ImageMagick >= 5.5.7\"") >&5
-  ($PKG_CONFIG --exists --print-errors "ImageMagick >= 5.5.7") 2>&5
-  ac_status=$?
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; then
-  enable_framing=yes
-else
-  if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"GraphicsMagick >= 1.1\"") >&5
-  ($PKG_CONFIG --exists --print-errors "GraphicsMagick >= 1.1") 2>&5
-  ac_status=$?
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; then
-  enable_framing=yes
-else
-  enable_framing=no
-fi
-fi
    if test "x$enable_framing" = "xyes"; then
       # Extract the first word of "convert", so it can be a program name with args.
 set dummy convert; ac_word=$2
@@ -6543,20 +6524,8 @@
 { echo "$as_me:$LINENO: checking SVG to PNG conversion method" >&5
 echo $ECHO_N "checking SVG to PNG conversion method... $ECHO_C" >&6; }
 if test "x$enable_large_bitmaps" = "xyes"; then
-   if test -n "$PKG_CONFIG" && \
-    { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"librsvg-2.0 >= 2.12.3\"") >&5
-  ($PKG_CONFIG --exists --print-errors "librsvg-2.0 >= 2.12.3") 2>&5
-  ac_status=$?
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; then
-  enable_large_bitmaps=yes
-else
-  enable_large_bitmaps=no
-fi
    if test "x$enable_large_bitmaps" = "xyes"; then
       svgconvert_prog="rsvg"
-   else
-      svgconvert_prog="ksvgtopng"
    fi
    { echo "$as_me:$LINENO: result: $svgconvert_prog" >&5
 echo "${ECHO_T}$svgconvert_prog" >&6; }
@@ -6601,8 +6570,8 @@
 
 
    if test -z "$SVGCONVERT"; then
-      { { echo "$as_me:$LINENO: error: You need librsvg >= 2.12.3 or ksvgtopng installed to create large scale bitmaps" >&5
-echo "$as_me: error: You need librsvg >= 2.12.3 or ksvgtopng installed to create large scale bitmaps" >&2;}
+      { { echo "$as_me:$LINENO: error: You need librsvg installed to create large scale bitmaps" >&5
+echo "$as_me: error: You need librsvg installed to create large scale bitmaps" >&2;}
    { (exit 1); exit 1; }; }
    else
       enable_large_bitmaps=yes
diff -Nurd -x'*~' tango-icon-theme-0.8.90.orig/index.theme.in tango-icon-theme-0.8.90/index.theme.in
--- tango-icon-theme-0.8.90.orig/index.theme.in	2006-06-07 09:19:56.000000000 -0400
+++ tango-icon-theme-0.8.90/index.theme.in	2017-03-26 22:24:51.000000000 -0400
@@ -457,10 +457,10 @@
 MaxSize=256
 
 [scalable/emblems]
-Size=48
+Size=96
 Context=Emblems
 Type=Scalable
-MinSize=32
+MinSize=1
 MaxSize=256
 
 [scalable/emotes]
