diff -Nurd -x'*~' xsane-0.999.orig/Makefile.in xsane-0.999/Makefile.in
--- xsane-0.999.orig/Makefile.in	2008-03-29 05:36:52.000000000 -0400
+++ xsane-0.999/Makefile.in	2015-12-20 04:13:51.000000000 -0500
@@ -30,6 +30,7 @@
 
 @SET_MAKE@
 
+ACLOCAL_AMFLAGS=-I m4
 SUBDIRS	= lib @INTLSUB@ src @POSUB@ doc
 
 all: all-recursive
diff -Nurd xsane-0.999.orig/configure xsane-0.999/configure
--- xsane-0.999.orig/configure	2013-05-22 17:50:40.000000000 -0400
+++ xsane-0.999/configure	2023-11-11 12:33:16.000000000 -0500
@@ -6631,6 +6631,7 @@
 
 #include <sane/sane.h>
 #include <stdio.h>
+#include <stdlib.h>
 
 int
 main ()
diff -Nurd -x'*~' xsane-0.999.orig/configure.in xsane-0.999/configure.in
--- xsane-0.999.orig/configure.in	2013-05-22 17:16:55.000000000 -0400
+++ xsane-0.999/configure.in	2015-12-20 04:13:18.000000000 -0500
@@ -1,6 +1,7 @@
 dnl Process this file with autoconf to produce a configure script. -*-sh-*-
 AC_INIT(include/config.h.in)
 AC_CONFIG_HEADER(include/config.h)
+AC_CONFIG_MACRO_DIRS([m4])
 # AC_PREREQ(2.10)dnl		dnl Minimum Autoconf version required.
 # AC_ARG_PROGRAM
 
diff -Nurd -x'*~' xsane-0.999.orig/lib/getopt.c xsane-0.999/lib/getopt.c
--- xsane-0.999.orig/lib/getopt.c	1999-03-02 18:14:45.000000000 -0500
+++ xsane-0.999/lib/getopt.c	2015-12-20 04:13:18.000000000 -0500
@@ -75,6 +75,9 @@
 #define getpid() GetCurrentProcessId()
 #endif
 
+#include <string.h>
+#include <unistd.h>
+
 #ifndef _
 /* This is for other GNU distributions with internationalized messages.
    When compiling libc, the _ macro is predefined.  */
diff -Nurd xsane-0.999.orig/m4/sane.m4 xsane-0.999/m4/sane.m4
--- xsane-0.999.orig/m4/sane.m4	2004-04-18 18:43:05.000000000 -0400
+++ xsane-0.999/m4/sane.m4	2023-11-11 12:26:42.000000000 -0500
@@ -44,6 +44,7 @@
       AC_TRY_RUN([
 #include <sane/sane.h>
 #include <stdio.h>
+#include <stdlib.h>
 
 int 
 main ()
