diff -uNr kdegraphics-3.4.0/kcoloredit/Makefile.am kdegraphics-3.4.0-new/kcoloredit/Makefile.am
--- kdegraphics-3.4.0/kcoloredit/Makefile.am	Thu Feb 27 19:25:14 2003
+++ kdegraphics-3.4.0-new/kcoloredit/Makefile.am	Thu Mar 10 16:54:31 2005
@@ -1,21 +1,24 @@
 INCLUDES = $(all_includes)
 
-kcoloredit_SOURCES = kzcolorselector.cpp imageselection.cpp \
+kcoloredit_la_SOURCES = kzcolorselector.cpp imageselection.cpp \
 	texteditselection.cpp gradientselection.cpp colorselector.cpp \
 	kxycolorselector.cpp paletteview.cpp paletteviewscrolledarea.cpp \
 	editablestreamhistory.cpp color.cpp palette.cpp loadpalettedlg.cpp \
 	kcoloreditview.cpp kcoloreditdoc.cpp kcoloredit.cpp main.cpp 
-kcoloredit_METASOURCES = AUTO
-kcoloredit_LDADD = $(LIB_KDEUI) $(LIB_KFILE)
-kcoloredit_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+kcoloredit_la_METASOURCES = AUTO
+kcoloredit_la_LIBADD = $(LIB_KDEUI) $(LIB_KFILE)
+kcoloredit_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 rcdir = $(kde_datadir)/kcoloredit
 rc_DATA = kcoloreditui.rc
 
-bin_PROGRAMS = kcoloredit kcolorchooser
-kcolorchooser_SOURCES = kcolorchooser.cpp
-kcolorchooser_LDADD = $(LIB_KDEUI)
-kcolorchooser_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kcoloredit.la kcolorchooser.la
+
+kcolorchooser_la_SOURCES = kcolorchooser.cpp
+kcolorchooser_la_LIBADD = $(LIB_KDEUI)
+kcolorchooser_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
 
 noinst_HEADERS = color.h colorselector.h editablestreamhistory.h \
 	gradientselection.h imageselection.h kcoloredit.h kcoloreditdoc.h \
diff -uNr kdegraphics-3.4.0/kcoloredit/kcolorchooser.cpp kdegraphics-3.4.0-new/kcoloredit/kcolorchooser.cpp
--- kdegraphics-3.4.0/kcoloredit/kcolorchooser.cpp	Fri Feb  6 18:41:03 2004
+++ kdegraphics-3.4.0-new/kcoloredit/kcolorchooser.cpp	Thu Mar 10 16:52:59 2005
@@ -40,7 +40,7 @@
    KCmdLineLastOption
 };
 	
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
   KLocale::setMainCatalogue("kdelibs"); 
   KAboutData aboutData( "kcolorchooser", I18N_NOOP("KColorChooser"),
diff -uNr kdegraphics-3.4.0/kcoloredit/main.cpp kdegraphics-3.4.0-new/kcoloredit/main.cpp
--- kdegraphics-3.4.0/kcoloredit/main.cpp	Wed Feb 23 05:12:42 2005
+++ kdegraphics-3.4.0-new/kcoloredit/main.cpp	Thu Mar 10 16:53:06 2005
@@ -34,7 +34,7 @@
   // INSERT YOUR COMMANDLINE OPTIONS HERE
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
 
 	KAboutData aboutData( "kcoloredit", I18N_NOOP("KColorEdit"),
diff -uNr kdegraphics-3.4.0/kdvi/Makefile.am kdegraphics-3.4.0-new/kdvi/Makefile.am
--- kdegraphics-3.4.0/kdvi/Makefile.am	Thu Feb  3 17:44:57 2005
+++ kdegraphics-3.4.0-new/kdvi/Makefile.am	Thu Mar 10 16:55:41 2005
@@ -1,11 +1,13 @@
 # set the include path for X, qt and KDE
 INCLUDES= -I$(top_srcdir)/kviewshell \
 	-I$(top_builddir)/kviewshell \
-	$(all_includes) $(LIBFREETYPE_CFLAGS)
+	$(LIBFREETYPE_CFLAGS) $(all_includes)
 # claim, which subdirectories you want to install
 SUBDIRS = . pix
 
-bin_PROGRAMS = kdvi
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kdvi.la
 
 # you can add here more. This one gets installed
 kde_module_LTLIBRARIES= kdvipart.la
@@ -37,9 +39,9 @@
 
 KDE_OPTIONS = nofinal
 
-kdvi_SOURCES = main.cpp 
-kdvi_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kdvi_LDADD = ../kviewshell/libifaces.la ../kviewshell/libkviewshell.la -lkparts
+kdvi_la_SOURCES = main.cpp 
+kdvi_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+kdvi_la_LIBADD = ../kviewshell/libifaces.la ../kviewshell/libkviewshell.la -lkparts
 
 ## this option you can leave out. Just, if you use "make dist", you need it
 noinst_HEADERS = dvi.h dviwin.h xdvi.h
diff -uNr kdegraphics-3.4.0/kdvi/main.cpp kdegraphics-3.4.0-new/kdvi/main.cpp
--- kdegraphics-3.4.0/kdvi/main.cpp	Wed Feb 23 05:12:43 2005
+++ kdegraphics-3.4.0-new/kdvi/main.cpp	Thu Mar 10 16:55:15 2005
@@ -25,7 +25,7 @@
 static const char description[] = I18N_NOOP("A previewer for Device Independent files (DVI files) produced by the TeX typesetting system.");
 
 
-int main(int argc, char** argv)
+extern "C" int kdemain(int argc, char** argv)
 {
   KAboutData about ("kdvi", I18N_NOOP("KDVI"), "1.3",
                     description, KAboutData::License_GPL,
diff -uNr kdegraphics-3.4.0/kfax/Makefile.am kdegraphics-3.4.0-new/kfax/Makefile.am
--- kdegraphics-3.4.0/kfax/Makefile.am	Thu Feb  3 17:44:58 2005
+++ kdegraphics-3.4.0-new/kfax/Makefile.am	Thu Mar 10 16:57:48 2005
@@ -1,16 +1,18 @@
-AM_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-LDADD      = $(LIB_KFILE) $(LIB_KDEPRINT) -lm
 INCLUDES   = $(all_includes)
 
 ####### Files
 noinst_HEADERS = kfax.h faxexpand.h options.h
 
-kfax_SOURCES   = options.cpp kfax.cpp faxexpand.cpp faxinit.cpp faxinput.cpp \
+kfax_la_SOURCES   = options.cpp kfax.cpp faxexpand.cpp faxinit.cpp faxinput.cpp \
 		 viewfax.cpp kfax_printsettings.cpp
+kfax_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+kfax_la_LIBADD  = $(LIB_KFILE) $(LIB_KDEPRINT) -lm
 
 METASOURCES    = AUTO
 
-bin_PROGRAMS   = kfax
+bin_PROGRAMS   = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kfax.la
 
 xdg_apps_DATA = kfax.desktop
 
diff -uNr kdegraphics-3.4.0/kfax/kfax.cpp kdegraphics-3.4.0-new/kfax/kfax.cpp
--- kdegraphics-3.4.0/kfax/kfax.cpp	Thu Feb  3 17:44:58 2005
+++ kdegraphics-3.4.0-new/kfax/kfax.cpp	Thu Mar 10 16:57:01 2005
@@ -1682,7 +1682,7 @@
    KCmdLineLastOption
 };
 
-int main (int argc, char **argv)
+extern "C" int kdemain (int argc, char **argv)
 {
   KAboutData aboutData( "kfax", I18N_NOOP("KFax"),
       KFAXVERSION, description, KAboutData::License_GPL,
diff -uNr kdegraphics-3.4.0/kfile-plugins/png/Makefile.am kdegraphics-3.4.0-new/kfile-plugins/png/Makefile.am
--- kdegraphics-3.4.0/kfile-plugins/png/Makefile.am	Mon Feb  2 16:50:42 2004
+++ kdegraphics-3.4.0-new/kfile-plugins/png/Makefile.am	Thu Mar 10 16:51:52 2005
@@ -10,7 +10,7 @@
 
 kfile_png_la_SOURCES = kfile_png.cpp
 kfile_png_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN)
-kfile_png_la_LIBADD = $(LIB_KIO)
+kfile_png_la_LIBADD = $(LIB_KIO) $(LIBPNG)
 
 # let automoc handle all of the meta source files (moc)
 METASOURCES = AUTO
diff -uNr kdegraphics-3.4.0/kghostview/Makefile.am kdegraphics-3.4.0-new/kghostview/Makefile.am
--- kdegraphics-3.4.0/kghostview/Makefile.am	Fri Oct 15 18:50:46 2004
+++ kdegraphics-3.4.0-new/kghostview/Makefile.am	Thu Mar 10 17:02:26 2005
@@ -4,7 +4,8 @@
 
 ####### Files
 
-bin_PROGRAMS = 	kghostview
+kdeinit_LTLIBRARIES = kghostview.la
+bin_PROGRAMS = 	
 lib_LTLIBRARIES = libkghostviewlib.la
 kde_module_LTLIBRARIES = libkghostviewpart.la
 noinst_LTLIBRARIES = libdscparse.la
@@ -27,9 +28,9 @@
 
 libkghostviewpart_la_SOURCES = part_init.cpp
 
-kghostview_SOURCES = main.cpp
-kghostview_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kghostview_LDADD = libkghostviewlib.la
+kghostview_la_SOURCES = main.cpp
+kghostview_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+kghostview_la_LIBADD = libkghostviewlib.la
 
 libdscparse_la_LDFLAGS = $(all_libraries) -no-undefined
 libdscparse_la_LIBADD = $(LIB_QT)
diff -uNr kdegraphics-3.4.0/kghostview/main.cpp kdegraphics-3.4.0-new/kghostview/main.cpp
--- kdegraphics-3.4.0/kghostview/main.cpp	Mon Jun 21 14:59:04 2004
+++ kdegraphics-3.4.0-new/kghostview/main.cpp	Thu Mar 10 17:02:06 2005
@@ -43,7 +43,7 @@
     KCmdLineLastOption
 };
 
-int main( int argc, char** argv )
+extern "C" int kdemain( int argc, char** argv )
 {
     KCmdLineArgs::init( argc, argv, KGVPart::createAboutData() );
     KCmdLineArgs::addCmdLineOptions( options );
diff -uNr kdegraphics-3.4.0/kiconedit/Makefile.am kdegraphics-3.4.0-new/kiconedit/Makefile.am
--- kdegraphics-3.4.0/kiconedit/Makefile.am	Thu Feb 27 19:25:15 2003
+++ kdegraphics-3.4.0-new/kiconedit/Makefile.am	Thu Mar 10 17:03:56 2005
@@ -1,18 +1,20 @@
 INCLUDES = $(all_includes)
 SUBDIRS = pics
 
-bin_PROGRAMS = kiconedit
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kiconedit.la
 
-kiconedit_SOURCES = utils.cpp main.cpp kiconedit.cpp kicongrid.cpp \
+kiconedit_la_SOURCES = utils.cpp main.cpp kiconedit.cpp kicongrid.cpp \
     kiconcolors.cpp kcolorgrid.cpp palettetoolbar.cpp\
     kicon.cpp kresize.cpp knew.cpp properties.cpp \
     kiconeditslots.cpp kiconconfig.cpp 
 
-kiconedit_METASOURCES = AUTO
+kiconedit_la_METASOURCES = AUTO
 
 # the library search path. 
-kiconedit_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kiconedit_LDADD = $(LIB_KFILE) -lkdeprint
+kiconedit_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+kiconedit_la_LIBADD = $(LIB_KFILE) -lkdeprint
 
 DISTCLEANFILES = *~ .deps/* .libs/*
 
diff -uNr kdegraphics-3.4.0/kiconedit/main.cpp kdegraphics-3.4.0-new/kiconedit/main.cpp
--- kdegraphics-3.4.0/kiconedit/main.cpp	Fri Feb  6 18:41:03 2004
+++ kdegraphics-3.4.0-new/kiconedit/main.cpp	Thu Mar 10 17:03:37 2005
@@ -40,7 +40,7 @@
   KCmdLineLastOption
 };
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int argc, char **argv)
 {
   KAboutData aboutData( "kiconedit", I18N_NOOP("KIconEdit"),
     VERSION, description, KAboutData::License_GPL,
diff -uNr kdegraphics-3.4.0/kooka/Makefile.am kdegraphics-3.4.0-new/kooka/Makefile.am
--- kdegraphics-3.4.0/kooka/Makefile.am	Thu Dec  2 04:41:05 2004
+++ kdegraphics-3.4.0-new/kooka/Makefile.am	Thu Mar 10 17:06:33 2005
@@ -2,10 +2,13 @@
 
 SUBDIRS = pics
 
-bin_PROGRAMS = kooka
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kooka.la
+
 METASOURCES = AUTO
 
-kooka_SOURCES =	main.cpp kooka.cpp kookaview.cpp kookapref.cpp 		\
+kooka_la_SOURCES =	main.cpp kooka.cpp kookaview.cpp kookapref.cpp 		\
 		img_saver.cpp ksaneocr.cpp \
 		kookaimage.cpp kookaimagemeta.cpp scanpackager.cpp 	\
 		imgnamecombo.cpp imageselectline.cpp      \
@@ -16,8 +19,8 @@
 		kocrocrad.cpp
 # 		pagesetup.cpp 
 
-kooka_LDADD   = $(LIB_KFILE) -lkdeprint -lktexteditor $(LIBTIFF) $(top_builddir)/libkscan/libkscan.la $(KADMOS_LIB) $(LIB_KSPELL)
-kooka_LDFLAGS = $(KDE_RPATH) $(all_libraries)
+kooka_la_LIBADD   = $(LIB_KFILE) -lkdeprint -lktexteditor $(LIBTIFF) $(top_builddir)/libkscan/libkscan.la $(KADMOS_LIB) $(LIB_KSPELL)
+kooka_la_LDFLAGS = $(KDE_PLUGIN) $(all_libraries) -module
 
 INCLUDES         = -I$(top_srcdir)/libkscan $(all_includes) $(LIBSANE_INCLUDES) $(KADMOS_INC)
 
diff -uNr kdegraphics-3.4.0/kooka/main.cpp kdegraphics-3.4.0-new/kooka/main.cpp
--- kdegraphics-3.4.0/kooka/main.cpp	Sun Jul  4 08:03:16 2004
+++ kdegraphics-3.4.0-new/kooka/main.cpp	Thu Mar 10 17:06:31 2005
@@ -71,7 +71,7 @@
 
 
 
-int main( int argc, char *argv[] )
+extern "C" int kdemain( int argc, char *argv[] )
 {
    KAboutData about("kooka", I18N_NOOP("Kooka"), KOOKA_VERSION, I18N_NOOP(description),
 		    KAboutData::License_GPL_V2, "(C) 2000 Klaas Freitag", 0,
diff -uNr kdegraphics-3.4.0/kpdf/shell/Makefile.am kdegraphics-3.4.0-new/kpdf/shell/Makefile.am
--- kdegraphics-3.4.0/kpdf/shell/Makefile.am	Mon Jan  3 16:59:57 2005
+++ kdegraphics-3.4.0-new/kpdf/shell/Makefile.am	Thu Mar 10 17:07:58 2005
@@ -1,12 +1,15 @@
-INCLUDES = -I$(srcdir)/xpdf -I$(srcdir)/xpdf/goo -I$(top_builddir)/kpdf $(all_includes) $(FREETYPE_CFLAGS)
+INCLUDES = -I$(srcdir)/xpdf -I$(srcdir)/xpdf/goo -I$(top_builddir)/kpdf $(FREETYPE_CFLAGS) $(all_includes)
+LDFLAGS  = $(FREETYPE_LDFLAGS)
 
 METASOURCES = AUTO
 
-bin_PROGRAMS = kpdf
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kpdf.la
 
-kpdf_SOURCES = main.cpp shell.cpp
-kpdf_LDFLAGS = $(KDE_RPATH) $(all_libraries)
-kpdf_LDADD   = $(LIB_KPARTS)
+kpdf_la_SOURCES = main.cpp shell.cpp
+kpdf_la_LDFLAGS = $(KDE_PLUGIN) $(all_libraries) -module
+kpdf_la_LIBADD  = $(LIB_KPARTS)
 
 EXTRA_DIST = kpdf.desktop
 xdg_apps_DATA = kpdf.desktop
diff -uNr kdegraphics-3.4.0/kpdf/shell/main.cpp kdegraphics-3.4.0-new/kpdf/shell/main.cpp
--- kdegraphics-3.4.0/kpdf/shell/main.cpp	Wed Feb 23 05:12:46 2005
+++ kdegraphics-3.4.0-new/kpdf/shell/main.cpp	Thu Mar 10 17:07:55 2005
@@ -29,7 +29,7 @@
     KCmdLineLastOption
 };
 
-int main(int argc, char** argv)
+extern "C" int kdemain(int argc, char** argv)
 {
     KAboutData about(
         "kpdf",
diff -uNr kdegraphics-3.4.0/kpovmodeler/Makefile.am kdegraphics-3.4.0-new/kpovmodeler/Makefile.am
--- kdegraphics-3.4.0/kpovmodeler/Makefile.am	Thu Oct 28 21:27:16 2004
+++ kdegraphics-3.4.0-new/kpovmodeler/Makefile.am	Thu Mar 10 17:09:26 2005
@@ -224,11 +224,12 @@
 ####
 # the executable
 
-bin_PROGRAMS = kpovmodeler
+bin_PROGRAMS = 
+kdeinit_LTLIBRARIES = kpovmodeler.la
 
-kpovmodeler_SOURCES = main.cpp
-kpovmodeler_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kpovmodeler_LDADD = libkpovmodeler.la
+kpovmodeler_la_SOURCES = main.cpp
+kpovmodeler_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+kpovmodeler_la_LIBADD  = libkpovmodeler.la
 
 dummy.cpp:
 	echo > dummy.cpp
diff -uNr kdegraphics-3.4.0/kpovmodeler/main.cpp kdegraphics-3.4.0-new/kpovmodeler/main.cpp
--- kdegraphics-3.4.0/kpovmodeler/main.cpp	Fri Sep 19 15:48:10 2003
+++ kdegraphics-3.4.0-new/kpovmodeler/main.cpp	Thu Mar 10 17:08:48 2005
@@ -35,7 +35,7 @@
   KCmdLineLastOption
 };
 
-int main( int argc, char* argv[] )
+extern "C" int kdemain( int argc, char* argv[] )
 {
    PMShell* shell = 0;
 
diff -uNr kdegraphics-3.4.0/kruler/Makefile.am kdegraphics-3.4.0-new/kruler/Makefile.am
--- kdegraphics-3.4.0/kruler/Makefile.am	Thu Feb 27 19:25:15 2003
+++ kdegraphics-3.4.0-new/kruler/Makefile.am	Thu Mar 10 17:12:09 2005
@@ -2,11 +2,14 @@
 
 INCLUDES= $(all_includes)
 
-bin_PROGRAMS = kruler
-kruler_METASOURCES = AUTO
-kruler_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kruler_LDADD   = $(LIB_KDEUI)
-kruler_SOURCES = klineal.cpp main.cpp 
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = kruler.la
+
+kruler_la_METASOURCES = AUTO
+kruler_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+kruler_la_LIBADD  = $(LIB_KDEUI)
+kruler_la_SOURCES = klineal.cpp main.cpp 
 
 xdg_apps_DATA = kruler.desktop
 
diff -uNr kdegraphics-3.4.0/kruler/main.cpp kdegraphics-3.4.0-new/kruler/main.cpp
--- kdegraphics-3.4.0/kruler/main.cpp	Sun Jul  4 08:03:16 2004
+++ kdegraphics-3.4.0-new/kruler/main.cpp	Thu Mar 10 17:11:00 2005
@@ -22,7 +22,7 @@
   // INSERT YOUR COMMANDLINE OPTIONS HERE
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
 
 
diff -uNr kdegraphics-3.4.0/ksnapshot/Makefile.am kdegraphics-3.4.0-new/ksnapshot/Makefile.am
--- kdegraphics-3.4.0/ksnapshot/Makefile.am	Sat Sep 13 13:03:57 2003
+++ kdegraphics-3.4.0-new/ksnapshot/Makefile.am	Thu Mar 10 17:14:39 2005
@@ -1,14 +1,17 @@
-bin_PROGRAMS= ksnapshot
+
+bin_PROGRAMS=
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = ksnapshot.la
 
 INCLUDES = -DKSNAPVERSION="\"0.7\"" $(all_includes)
 
-ksnapshot_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-ksnapshot_LDADD = $(LIB_KDEPRINT)
+ksnapshot_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+ksnapshot_la_LIBADD = $(LIB_KDEPRINT)
 
-ksnapshot_SOURCES = ksnapshotiface.skel main.cpp ksnapshot.cpp \
+ksnapshot_la_SOURCES = ksnapshotiface.skel main.cpp ksnapshot.cpp \
     regiongrabber.cpp ksnapshotwidget.ui
 
-ksnapshot_METASOURCES = AUTO
+ksnapshot_la_METASOURCES = AUTO
 
 xdg_apps_DATA = ksnapshot.desktop
 
diff -uNr kdegraphics-3.4.0/ksnapshot/main.cpp kdegraphics-3.4.0-new/ksnapshot/main.cpp
--- kdegraphics-3.4.0/ksnapshot/main.cpp	Wed Feb 23 05:13:01 2005
+++ kdegraphics-3.4.0-new/ksnapshot/main.cpp	Thu Mar 10 17:13:12 2005
@@ -17,7 +17,7 @@
     { 0, 0, 0 }
 };
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int argc, char **argv)
 {
   KAboutData aboutData( "ksnapshot", I18N_NOOP("KSnapshot"),
     KSNAPVERSION, description, KAboutData::License_GPL,
diff -uNr kdegraphics-3.4.0/ksvg/impl/libs/libtext2path/src/Makefile.am kdegraphics-3.4.0-new/ksvg/impl/libs/libtext2path/src/Makefile.am
--- kdegraphics-3.4.0/ksvg/impl/libs/libtext2path/src/Makefile.am	Fri Mar 26 15:21:56 2004
+++ kdegraphics-3.4.0-new/ksvg/impl/libs/libtext2path/src/Makefile.am	Thu Mar 10 16:51:52 2005
@@ -1,10 +1,10 @@
 lib_LTLIBRARIES = libtext2path.la
 
 KDE_CXXFLAGS = $(USE_EXCEPTIONS)
-INCLUDES = $(FRIBIDI_CFLAGS) $(FREETYPE_CFLAGS) $(FONTCONFIG_CFLAGS) $(all_includes)
+INCLUDES = $(FREETYPE_CFLAGS) $(FRIBIDI_CFLAGS) $(FONTCONFIG_CFLAGS) $(all_includes)
 
 libtext2path_includedir=$(includedir)/libtext2path-0.1
 libtext2path_include_HEADERS = BezierPath.h Glyph.h GlyphTracer.h
 
 libtext2path_la_SOURCES = Affine.cpp Rectangle.cpp Font.cpp Glyph.cpp GlyphTracer.cpp Converter.cpp QtUnicode.cpp
-libtext2path_la_LDFLAGS = $(FRIBIDI_LIBS) $(FONTCONFIG_LIBS)
+libtext2path_la_LDFLAGS = $(FRIBIDI_LIBS) $(FONTCONFIG_LIBS) $(FREETYPE_LIBS)
diff -uNr kdegraphics-3.4.0/ksvg/plugin/backends/libart/Makefile.am kdegraphics-3.4.0-new/ksvg/plugin/backends/libart/Makefile.am
--- kdegraphics-3.4.0/ksvg/plugin/backends/libart/Makefile.am	Mon Jan  3 09:53:15 2005
+++ kdegraphics-3.4.0-new/ksvg/plugin/backends/libart/Makefile.am	Thu Mar 10 16:51:52 2005
@@ -8,4 +8,4 @@
 kde_services_DATA = ksvglibartcanvas.desktop
 
 KDE_CXXFLAGS = $(USE_EXCEPTIONS)
-INCLUDES = $(LIBART_CFLAGS) $(FREETYPE_CFLAGS) $(FONTCONFIG_CFLAGS) -I$(top_srcdir)/ksvg/dom -I$(top_srcdir)/ksvg/impl -I$(top_srcdir)/ksvg/ecma -I$(top_srcdir)/ksvg/core -I$(top_srcdir)/ksvg/impl/libs/art_support -I$(top_srcdir)/ksvg/impl/libs/libtext2path/src $(KDE_INCLUDES) $(all_includes)
+INCLUDES = $(FREETYPE_CFLAGS) $(LIBART_CFLAGS) $(FONTCONFIG_CFLAGS) -I$(top_srcdir)/ksvg/dom -I$(top_srcdir)/ksvg/impl -I$(top_srcdir)/ksvg/ecma -I$(top_srcdir)/ksvg/core -I$(top_srcdir)/ksvg/impl/libs/art_support -I$(top_srcdir)/ksvg/impl/libs/libtext2path/src $(KDE_INCLUDES) $(all_includes)
diff -uNr kdegraphics-3.4.0/ksvg/test/external/Makefile.am kdegraphics-3.4.0-new/ksvg/test/external/Makefile.am
--- kdegraphics-3.4.0/ksvg/test/external/Makefile.am	Thu Oct 23 17:18:10 2003
+++ kdegraphics-3.4.0-new/ksvg/test/external/Makefile.am	Thu Mar 10 17:16:38 2005
@@ -1,13 +1,16 @@
 INCLUDES = $(FREETYPE_CFLAGS) -I$(top_srcdir)/ksvg -I$(top_srcdir)/ksvg/dom -I$(top_srcdir)/ksvg/impl -I$(top_srcdir)/ksvg/ecma -I$(top_srcdir)/ksvg/core -I$(top_srcdir)/ksvg/impl/libs/libtext2path/src $(all_includes)
+LDFLAGS = $(FREETYPE_LDFLAGS)
 METASOURCES = AUTO
 KDE_CXXFLAGS = $(USE_EXCEPTIONS)
 
-bin_PROGRAMS = svgdisplay printnodetest
+bin_PROGRAMS = 
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = svgdisplay.la printnodetest.la
 
-svgdisplay_SOURCES = SVGTestWidget.cc svgdisplay.cc
-svgdisplay_LDADD =  $(LIB_KDECORE) ../../libksvg.la
-svgdisplay_LDFLAGS = $(KDE_RPATH) $(all_libraries)
+svgdisplay_la_SOURCES = SVGTestWidget.cc svgdisplay.cc
+svgdisplay_la_LIBADD =  $(LIB_KDECORE) ../../libksvg.la
+svgdisplay_la_LDFLAGS = $(KDE_PLUGIN) $(all_libraries) -module
 
-printnodetest_SOURCES = printnodetest.cpp
-printnodetest_LDADD =  $(LIB_KDECORE) ../../libksvg.la
-printnodetest_LDFLAGS = $(KDE_RPATH) $(all_libraries)
+printnodetest_la_SOURCES = printnodetest.cpp
+printnodetest_la_LIBADD =  $(LIB_KDECORE) ../../libksvg.la
+printnodetest_la_LDFLAGS = $(KDE_PLUGIN) $(all_libraries) -module
diff -uNr kdegraphics-3.4.0/ksvg/test/external/printnodetest.cpp kdegraphics-3.4.0-new/ksvg/test/external/printnodetest.cpp
--- kdegraphics-3.4.0/ksvg/test/external/printnodetest.cpp	Mon Jun 21 14:59:05 2004
+++ kdegraphics-3.4.0-new/ksvg/test/external/printnodetest.cpp	Thu Mar 10 17:16:10 2005
@@ -44,7 +44,7 @@
 	exit(0);
 };
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int argc, char **argv)
 {
 	KAboutData *about = new KAboutData("KSVG printNode test app", I18N_NOOP("KSVG test"), "0.1", I18N_NOOP("KSVG printNode test app"));
 	KCmdLineArgs::init(argc, argv, about);
diff -uNr kdegraphics-3.4.0/ksvg/test/external/svgdisplay.cc kdegraphics-3.4.0-new/ksvg/test/external/svgdisplay.cc
--- kdegraphics-3.4.0/ksvg/test/external/svgdisplay.cc	Mon Jun 21 14:59:05 2004
+++ kdegraphics-3.4.0-new/ksvg/test/external/svgdisplay.cc	Thu Mar 10 17:16:16 2005
@@ -10,7 +10,7 @@
 	KCmdLineLastOption
 };
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int argc, char **argv)
 {
 
 	KAboutData *about = new KAboutData( "KSVG test app", I18N_NOOP("KSVG test"), "0.1", I18N_NOOP("KSVG standalone test app") );
diff -uNr kdegraphics-3.4.0/kviewshell/Makefile.am kdegraphics-3.4.0-new/kviewshell/Makefile.am
--- kdegraphics-3.4.0/kviewshell/Makefile.am	Wed Feb 23 05:13:43 2005
+++ kdegraphics-3.4.0-new/kviewshell/Makefile.am	Thu Mar 10 17:17:46 2005
@@ -3,8 +3,9 @@
 
 ####### Files
 
-bin_PROGRAMS = 	kviewshell
+bin_PROGRAMS = 	
 lib_LTLIBRARIES = libkmultipage.la
+kdeinit_LTLIBRARIES = kviewshell.la
 kde_module_LTLIBRARIES = kviewerpart.la emptymultipagepart.la
 noinst_LTLIBRARIES = libifaces.la libkviewshell.la
 
@@ -33,9 +34,9 @@
 
 libkviewshell_la_SOURCES = kviewshell.cpp
 
-kviewshell_SOURCES = main.cpp
-kviewshell_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kviewshell_LDADD = libkviewshell.la ./libifaces.la -lkparts
+kviewshell_la_SOURCES = main.cpp
+kviewshell_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+kviewshell_la_LIBADD = libkviewshell.la ./libifaces.la -lkparts
 
 METASOURCES = AUTO
 
diff -uNr kdegraphics-3.4.0/kviewshell/main.cpp kdegraphics-3.4.0-new/kviewshell/main.cpp
--- kdegraphics-3.4.0/kviewshell/main.cpp	Wed Feb 23 05:13:45 2005
+++ kdegraphics-3.4.0-new/kviewshell/main.cpp	Thu Mar 10 17:17:30 2005
@@ -26,7 +26,7 @@
 static const char description[] = I18N_NOOP("Generic framework for viewer applications");
 
 
-int main(int argc, char **argv)
+extern "C" int kdemain(int argc, char **argv)
 {
   KAboutData about ("kviewshell", I18N_NOOP("KViewShell"), "0.4",
 		    description, KAboutData::License_GPL,
diff -uNr kdegraphics-3.4.0/libkscan/img_canvas.cpp kdegraphics-3.4.0-new/libkscan/img_canvas.cpp
--- kdegraphics-3.4.0/libkscan/img_canvas.cpp	Tue Nov 16 13:07:25 2004
+++ kdegraphics-3.4.0-new/libkscan/img_canvas.cpp	Thu Mar 10 16:51:52 2005
@@ -700,7 +700,7 @@
                            double(image->width());
             kdDebug(29000) << "FIT WIDTH scrollbar to substract: " << sbWidth << endl;
         }
-        scale_factor = 100*used_xscaler;
+        scale_factor = (int)(100*used_xscaler);
         break;
     case FIT_HEIGHT:
         used_yscaler = used_xscaler = double(noSBSize.height())/double(image->height());
@@ -715,12 +715,12 @@
             kdDebug(29000) << "FIT HEIGHT scrollbar to substract: " << sbWidth << endl;
             // scale = int(100.0*(noSBSize.height() -sbWidth) / image->height());
         }
-        scale_factor = 100*used_xscaler;
+        scale_factor = (int)(100*used_xscaler);
 
         break;
     case ZOOM:
         used_xscaler = used_yscaler = double(getScaleFactor())/100.0;
-        scale_factor = 100*used_xscaler;
+        scale_factor = (int)(100*used_xscaler);
         break;
     default:
         break;
