Index: cervisia/cvsservice/Makefile.am
===================================================================
RCS file: /home/kde/kdesdk/cervisia/cvsservice/Makefile.am,v
retrieving revision 1.11
diff -u -u -r1.11 Makefile.am
--- cervisia/cvsservice/Makefile.am	18 Jul 2003 21:55:21 -0000	1.11
+++ cervisia/cvsservice/Makefile.am	29 Jan 2004 03:45:22 -0000
@@ -5,6 +5,8 @@
 #
 #
 
+CC=g++
+
 INCLUDES = $(all_includes)
 
 # cvs DCOP service
Index: kcachegrind/kcachegrind/sourceitem.h
===================================================================
RCS file: /home/kde/kdesdk/kcachegrind/kcachegrind/sourceitem.h,v
retrieving revision 1.3
diff -u -u -r1.3 sourceitem.h
--- kcachegrind/kcachegrind/sourceitem.h	27 Oct 2003 23:15:20 -0000	1.3
+++ kcachegrind/kcachegrind/sourceitem.h	29 Jan 2004 03:45:22 -0000
@@ -25,6 +25,7 @@
 
 #include <qlistview.h>
 #include "tracedata.h"
+#undef fileno
 
 class SourceItem: public QListViewItem
 {
Index: umbrello/umbrello/classparser/Makefile.am
===================================================================
RCS file: /home/kde/kdesdk/umbrello/umbrello/classparser/Makefile.am,v
retrieving revision 1.4
diff -u -u -r1.4 Makefile.am
--- umbrello/umbrello/classparser/Makefile.am	6 Nov 2003 23:05:50 -0000	1.4
+++ umbrello/umbrello/classparser/Makefile.am	29 Jan 2004 03:45:22 -0000
@@ -1,10 +1,10 @@
 tokenizer.cc: $(srcdir)/tokenizer.l
-	flex -+ -o$@ $(srcdir)/tokenizer.l
+	$(LEX) -+ -o$@ $(srcdir)/tokenizer.l
 
 CLEANFILES = tokenizer.cc
 
 noinst_LTLIBRARIES = libclassparser.la
-INCLUDES = $(all_includes)
+INCLUDES = -I@PREFIX@/lib/flex/include $(all_includes)
 libclassparser_la_METASOURCES = AUTO
 
 libclassparser_la_SOURCES = tokenizer.cc PersistantClassStore.cc ParsedStruct.cc ParsedSignalSlot.cc ParsedScopeContainer.cc ParsedParent.cc ParsedMethod.cc ParsedItem.cc ParsedContainer.cc ParsedClassContainer.cc ParsedClass.cc ParsedAttribute.cc ParsedArgument.cc ClassTreeNode.cc ClassStore.cc ClassParser.cc 
