diff -uNr kmd-0.3/Makefile.am kmd-0.3-new/Makefile.am
--- kmd-0.3/Makefile.am	Sat Mar  8 18:10:13 2003
+++ kmd-0.3-new/Makefile.am	Sun Feb  1 10:35:37 2004
@@ -15,8 +15,8 @@
 $(top_srcdir)/subdirs:
 	cd $(top_srcdir) && $(MAKE) -f admin/Makefile.common subdirs
 
-$(top_srcdir)/acinclude.m4: $(top_srcdir)/admin/acinclude.m4.in $(top_srcdir)/admin/libtool.m4.in
-	@cd $(top_srcdir) && cat admin/acinclude.m4.in admin/libtool.m4.in > acinclude.m4
+$(top_srcdir)/acinclude.m4: $(top_srcdir)/admin/acinclude.m4.in $(top_srcdir)/admin/libtool.m4.in $(top_srcdir)/admin/ltsugar.m4.in $(top_srcdir)/admin/lt~obsolete.m4.in
+	@cd $(top_srcdir) && cat admin/acinclude.m4.in admin/libtool.m4.in admin/ltsugar.m4.in admin/lt~obsolete.m4.in > acinclude.m4
 
 MAINTAINERCLEANFILES = subdirs configure.in acinclude.m4 configure.files 
 
