diff -uNr kxmleditor-1.0.0/Makefile.am kxmleditor-1.0.0-new/Makefile.am
--- kxmleditor-1.0.0/Makefile.am	Thu Dec 13 13:16:30 2001
+++ kxmleditor-1.0.0-new/Makefile.am	Sun Feb  1 19:54:28 2004
@@ -15,8 +15,8 @@
 $(top_srcdir)/subdirs:
 	cd $(top_srcdir) && $(MAKE) -f admin/Makefile.common subdirs
 
-$(top_srcdir)/acinclude.m4: $(top_srcdir)/admin/acinclude.m4.in $(top_srcdir)/admin/libtool.m4.in
-	@cd $(top_srcdir) && cat admin/acinclude.m4.in admin/libtool.m4.in > acinclude.m4
+$(top_srcdir)/acinclude.m4: $(top_srcdir)/admin/acinclude.m4.in $(top_srcdir)/admin/libtool.m4.in $(top_srcdir)/admin/ltsugar.m4.in $(top_srcdir)/admin/lt~obsolete.m4.in
+	@cd $(top_srcdir) && cat admin/acinclude.m4.in admin/libtool.m4.in admin/ltsugar.m4.in admin/lt~obsolete.m4.in > acinclude.m4
 
 MAINTAINERCLEANFILES = subdirs configure.in acinclude.m4 configure.files 
 
diff -uNr kxmleditor-1.0.0/kxmleditor/Makefile.am kxmleditor-1.0.0-new/kxmleditor/Makefile.am
--- kxmleditor-1.0.0/kxmleditor/Makefile.am	Tue Jul 23 15:14:18 2002
+++ kxmleditor-1.0.0-new/kxmleditor/Makefile.am	Sun Feb  1 10:15:30 2004
@@ -13,13 +13,7 @@
 	kxmleditorshell.cpp kxmleditorshell.h kxmleditorshell.rc \
 	dcopiface_shell.cpp dcopiface_shell.h
 
-install-data-local:
-	$(mkinstalldirs) $(kde_appsdir)/Applications/
-	$(INSTALL_DATA) $(srcdir)/kxmleditor.desktop $(kde_appsdir)/Applications/kxmleditor.desktop
-
-uninstall-local:
-	-rm -f $(kde_appsdir)/Applications/kxmleditor.desktop
-	-rmdir $(kde_appsdir)/Applications
+xdg_apps_DATA = kxmleditor.desktop
 
 ####### kdevelop will overwrite this part!!! (end)############
 # this 10 paths are KDE specific. Use them:
