diff -Naur emacs-26.3.orig/configure emacs-26.3/configure
--- emacs-26.3.orig/configure	2019-08-22 07:43:48.000000000 +0900
+++ emacs-26.3/configure	2020-05-06 01:35:58.000000000 +0900
@@ -10687,7 +10687,7 @@
      INSTALL_ARCH_INDEP_EXTRA=
   fi
 
-  NS_OBJC_OBJ="nsterm.o nsfns.o nsmenu.o nsselect.o nsimage.o $ns_fontfile"
+  NS_OBJC_OBJ="nsterm.o nsfns.o nsmenu.o nsselect.o nsimage.o macim.o $ns_fontfile"
 fi
 CFLAGS="$tmp_CFLAGS"
 CPPFLAGS="$tmp_CPPFLAGS"
@@ -16568,7 +16568,7 @@
 
     CPPFLAGS="$SAVE_CPPFLAGS"
     if test "${HAVE_LIBXML2}" = "yes"; then
-      LIBXML2_CFLAGS="-I'$xcsdkdir/usr/include/libxml2'"
+      LIBXML2_CFLAGS="-I'${includedir}/libxml2'"
       LIBXML2_LIBS="-lxml2"
     fi
   fi
@@ -31342,7 +31342,7 @@
    ## only costs about 1.5K of wasted binary space.
    headerpad_extra=1000
    if test "$HAVE_NS" = "yes"; then
-     libs_nsgui="-framework AppKit"
+     libs_nsgui="-framework AppKit -framework Carbon -framework Cocoa"
      if test "$NS_IMPL_COCOA" = "yes"; then
         libs_nsgui="$libs_nsgui -framework IOKit"
      fi
diff -Naur emacs-26.3.orig/nextstep/templates/Info.plist.in emacs-26.3/nextstep/templates/Info.plist.in
--- emacs-26.3.orig/nextstep/templates/Info.plist.in	2019-01-07 23:26:07.000000000 +0900
+++ emacs-26.3/nextstep/templates/Info.plist.in	2020-05-05 15:49:41.000000000 +0900
@@ -677,5 +677,13 @@
 	<string>YES</string>
         <key>NSAppleEventsUsageDescription</key>
         <string>Emacs requires permission to send AppleEvents to other applications.</string>
+        <key>NSDesktopFolderUsageDescription</key>
+        <string>Emacs requires permission to access the Desktop folder.</string>
+        <key>NSDocumentsFolderUsageDescription</key>
+        <string>Emacs requires permission to access the Documents folder.</string>
+        <key>NSDownloadsFolderUsageDescription</key>
+        <string>Emacs requires permission to access the Downloads folder.</string>
+        <key>NSRemovableVolumesUsageDescription</key>
+        <string>Emacs requires permission to access files on Removable Volumes.</string>
 </dict>
 </plist>
diff -Naur emacs-26.3.orig/src/macfont.m emacs-26.3/src/macfont.m
--- emacs-26.3.orig/src/macfont.m	2019-07-26 04:41:28.000000000 +0900
+++ emacs-26.3/src/macfont.m	2020-05-05 15:49:41.000000000 +0900
@@ -2346,9 +2346,9 @@
                   != (spacing >= FONT_SPACING_MONO)))
             continue;
 
-          /* Don't use a color bitmap font until it is supported on
-	     free platforms.  */
-          if (sym_traits & kCTFontTraitColorGlyphs)
+          /* Don't use a color bitmap font unless its family is
+             explicitly specified.  */
+          if ((sym_traits & kCTFontTraitColorGlyphs) && NILP (family))
             continue;
 
           if (j > 0
