diff -Naur emacsen-common-3.0.8-orig/debian-startup.el emacsen-common-3.0.8/debian-startup.el
--- emacsen-common-3.0.8-orig/debian-startup.el	2025-03-19 01:17:25
+++ emacsen-common-3.0.8/debian-startup.el	2025-08-15 21:50:10
@@ -1,10 +1,10 @@
 ;;; -*- lexical-binding: t -*-
-;;; debian-startup.el --- Debian-specific emacsen startup code
+;;; fink-startup.el --- Fink-specific emacsen startup code
 
 ;; Copyright (C) 1998-2017 Rob Browning
 
 ;; Maintainer: Rob Browning <rlb@defaultvalue.org>
-;; Keywords: debian
+;; Keywords: fink
 
 ;; This file is part of the debian release of GNU Emacs, and will
 ;; be contributed to the FSF after testing. It is released under the same
@@ -28,10 +28,10 @@
 ;;; Commentary:
 
 ;; This file contains startup code needed by all the various flavors
-;; of Emacs for a Debian system.
+;; of Emacs for a Fink system.
 
 
-(defun debian-pkg-add-load-path-item (item)
+(defun fink-pkg-add-load-path-item (item)
   "Takes a path item (a string) and adds it to load path in the
 correct position for an add-on package, before the emacs system
 directories, but after the /usr/local/ directories.  After modifying
@@ -63,7 +63,7 @@
       (setq load-path (nreverse result))
       load-path)))
 
-(defun debian-unique-strings (strings) 
+(defun fink-unique-strings (strings) 
   "Takes a list of strings and returns the list with *adjacent*
 duplicates removed."
   (let ((result '()))
@@ -73,7 +73,7 @@
       (setq strings (cdr strings)))
     (nreverse result)))
 
-(defun debian-run-directories (&rest dirs)
+(defun fink-run-directories (&rest dirs)
   "Load each file of the form XXfilename.el or XXfilename.elc in any
 of the dirs, where XX must be a number.  The files will be run in
 alphabetical order.  If a file appears in more than one of the dirs,
@@ -105,7 +105,7 @@
                    files)))
          
          ;; Finally sort them, and delete duplicates
-         (base-names (debian-unique-strings (sort stripped-names 'string<))))
+         (base-names (fink-unique-strings (sort stripped-names 'string<))))
 
     (setq load-path (append paths load-path)) ; Prefix paths temporarily.
     ;; Now load the files.  "load" will make sure we get the byte
@@ -122,5 +122,5 @@
     (dolist (item paths)
       (setq load-path (remq item load-path)))))
 
-(defun debian-startup (flavor)
-  (debian-run-directories (format "/etc/%s/site-start.d" (symbol-name flavor))))
+(defun fink-startup (flavor)
+  (fink-run-directories (format "@PREFIX@/etc/%s/site-start.d" (symbol-name flavor))))
diff -Naur emacsen-common-3.0.8-orig/emacs-install emacsen-common-3.0.8/emacs-install
--- emacsen-common-3.0.8-orig/emacs-install	2025-06-03 18:17:26
+++ emacsen-common-3.0.8/emacs-install	2025-08-15 21:52:19
@@ -5,7 +5,7 @@
 
 $ENV{'EMACS_INHIBIT_AUTOMATIC_NATIVE_COMPILATION'} = 't';
 
-my $lib_dir = "/usr/lib/emacsen-common";
+my $lib_dir = "@PREFIX@/lib/emacsen-common";
 my $invoked_by_old_pkg;
 my $context;
 my $flavor;
diff -Naur emacsen-common-3.0.8-orig/emacs-package-install emacsen-common-3.0.8/emacs-package-install
--- emacsen-common-3.0.8-orig/emacs-package-install	2025-03-19 01:16:59
+++ emacsen-common-3.0.8/emacs-package-install	2025-08-15 21:53:58
@@ -5,7 +5,7 @@
 
 $ENV{'EMACS_INHIBIT_AUTOMATIC_NATIVE_COMPILATION'} = 't';
 
-my $lib_dir = "/usr/lib/emacsen-common";
+my $lib_dir = "@PREFIX@/lib/emacsen-common";
 my $invoked_by_old_pkg;
 my $context;
 my $pkg;
diff -Naur emacsen-common-3.0.8-orig/emacs-package-remove emacsen-common-3.0.8/emacs-package-remove
--- emacsen-common-3.0.8-orig/emacs-package-remove	2025-03-19 01:16:59
+++ emacsen-common-3.0.8/emacs-package-remove	2025-08-15 21:57:12
@@ -5,7 +5,7 @@
 
 $ENV{'EMACS_INHIBIT_AUTOMATIC_NATIVE_COMPILATION'} = 't';
 
-my $lib_dir = "/usr/lib/emacsen-common";
+my $lib_dir = "@PREFIX@/lib/emacsen-common";
 my $invoked_by_old_pkg;
 my $pkg;
 
diff -Naur emacsen-common-3.0.8-orig/emacs-remove emacsen-common-3.0.8/emacs-remove
--- emacsen-common-3.0.8-orig/emacs-remove	2025-06-03 18:17:26
+++ emacsen-common-3.0.8/emacs-remove	2025-08-15 23:08:51
@@ -5,7 +5,7 @@
 
 $ENV{'EMACS_INHIBIT_AUTOMATIC_NATIVE_COMPILATION'} = 't';
 
-my $lib_dir = "/usr/lib/emacsen-common";
+my $lib_dir = "@PREFIX@/lib/emacsen-common";
 my $invoked_by_old_pkg;
 my $flavor;
 
@@ -71,7 +71,7 @@
   }
 }
 
-my $pending_flavor = "/var/lib/emacsen-common/state/flavor/pending/$flavor";
+my $pending_flavor = "@PREFIX@/var/lib/emacsen-common/state/flavor/pending/$flavor";
 die "ERROR: cannot unlink $pending_flavor: $!, "
     unless (unlink($pending_flavor) or $!{ENOENT});
 
diff -Naur emacsen-common-3.0.8-orig/emacsen-common.install emacsen-common-3.0.8/emacsen-common.install
--- emacsen-common-3.0.8-orig/emacsen-common.install	2018-05-13 18:12:48
+++ emacsen-common-3.0.8/emacsen-common.install	2025-08-15 23:55:18
@@ -3,16 +3,48 @@
 set -e
 
 flavor="$1"
+byte_compile_options="-batch -f batch-byte-compile"
 
 echo "emacsen-common: Handling install of emacsen flavor $flavor"
 
+# Try compiling a trivial testfile  {cswinehart  |  22 june 2004}
+echo ';;; -*- lexical-binding: t -*-' > /tmp/emacsen-common-testfile.el
+echo '(defun a-fake-func()  "This function does nothing"  )' >> /tmp/emacsen-common-testfile.el
+if ${flavor} ${byte_compile_options} /tmp/emacsen-common-testfile.el
+then
+    # It worked. Clean up the testfile and be done with it.
+    rm -f /tmp/emacsen-common-testfile.el{,c}
+else 
+    # It failed. Stop trying to install and print a meaningful error message
+    printf "\n\n\n"
+    echo "  +-------------------------------------------------------------------+"      
+    echo "  |  Fink had a problem installing this version of Emacs because      |"
+    echo "  |  ${flavor} couldn't be launched successfully.                       |"
+    echo "  |                                                                   |"
+    echo "  |  This typically happens when the package was compiled on a        |"
+    echo "  |  system with an older version of Mac OS X installed. At this      |"
+    echo "  |  point you should try recompiling the emacs version that caused   |"
+    echo "  |  this error with a command like:                                  |"
+    echo "  |    fink rebuild <flavor>                                          |"
+    echo "  |  where <flavor> is a package name like \`${flavor}' or \`${flavor}-nox' |"
+    echo "  |                                                                   |"
+    echo "  |  If this doesn't solve the problem, please notify the maintainer  |"
+    echo "  |  of the emacs package in question.                                |"
+    echo "  +-------------------------------------------------------------------+"
+    printf "\n\n\n\n"
+
+    # Clean up and return the error code
+    rm -f /tmp/emacsen-common-testfile.el{,c}
+    exit 1
+fi
+
 rm -f \
-   "/usr/share/$flavor/site-lisp/debian-startup.el" \
-   "/usr/share/$flavor/site-lisp/debian-startup.elc"
+   "@PREFIX@/share/$flavor/site-lisp/fink-startup.el" \
+   "@PREFIX@/share/$flavor/site-lisp/fink-startup.elc"
 
 # Create symlinks to the .el files (see section 5E in debian-emacs polcy).
-(cd "/usr/share/$flavor/site-lisp"
- ln -s ../../emacsen-common/debian-startup.el .)
+(cd "@PREFIX@/share/$flavor/site-lisp"
+ ln -s ../../emacs/site-lisp/fink-startup.el .)
 
 "$flavor" --no-init-file --no-site-file -batch -f batch-byte-compile \
-          "/usr/share/$flavor/site-lisp/debian-startup.el"
+          "@PREFIX@/share/$flavor/site-lisp/fink-startup.el"
diff -Naur emacsen-common-3.0.8-orig/emacsen-common.remove emacsen-common-3.0.8/emacsen-common.remove
--- emacsen-common-3.0.8-orig/emacsen-common.remove	2018-05-13 18:11:54
+++ emacsen-common-3.0.8/emacsen-common.remove	2025-08-15 22:05:50
@@ -7,5 +7,5 @@
 echo "emacsen-common: Handling removal of emacsen flavor $flavor"
 
 rm -f \
-   "/usr/share/$flavor/site-lisp/debian-startup.el" \
-   "/usr/share/$flavor/site-lisp/debian-startup.elc"
+   "@PREFIX@/share/$flavor/site-lisp/fink-startup.el" \
+   "@PREFIX@/share/$flavor/site-lisp/fink-startup.elc"
diff -Naur emacsen-common-3.0.8-orig/lib.pl emacsen-common-3.0.8/lib.pl
--- emacsen-common-3.0.8-orig/lib.pl	2025-03-19 01:16:59
+++ emacsen-common-3.0.8/lib.pl	2025-08-15 21:55:50
@@ -5,8 +5,8 @@
 
 # depends on: dpkg, tsort, perl
 
-my $lib_dir = "/usr/lib/emacsen-common";
-my $var_dir = "/var/lib/emacsen-common";
+my $lib_dir = "@PREFIX@/lib/emacsen-common";
+my $var_dir = "@PREFIX@/var/lib/emacsen-common";
 
 $::installed_package_state_dir = "${var_dir}/state/package/installed";
 $::installed_flavor_state_dir = "${var_dir}/state/flavor/installed";
diff -Naur emacsen-common-3.0.8-orig/sample-package-install-foo emacsen-common-3.0.8/sample-package-install-foo
--- emacsen-common-3.0.8-orig/sample-package-install-foo	2025-03-19 01:16:59
+++ emacsen-common-3.0.8/sample-package-install-foo	2025-08-15 22:07:30
@@ -1,5 +1,5 @@
 #!/bin/sh
-# /usr/lib/emacsen-common/packages/install/foo
+# @PREFIX@/lib/emacsen-common/packages/install/foo
 # [ Sample only -- this script hasn't been tested, so be careful. ]
 set -e
 
@@ -7,7 +7,7 @@
 echo "install/foo: Handling install of emacsen flavor $flavor"
 
 byte_compile_options='-batch -f batch-byte-compile'
-elc_dir="/usr/share/$flavor/site-lisp/foo/"
+elc_dir="@PREFIX@/share/$flavor/site-lisp/foo/"
 
 mkdir -p "$elc_dir"
 
diff -Naur emacsen-common-3.0.8-orig/sample-package-remove-foo emacsen-common-3.0.8/sample-package-remove-foo
--- emacsen-common-3.0.8-orig/sample-package-remove-foo	2018-05-13 18:09:48
+++ emacsen-common-3.0.8/sample-package-remove-foo	2025-08-15 22:07:54
@@ -1,10 +1,10 @@
 #!/bin/sh
-# /usr/lib/emacsen-common/packages/remove/foo
+# @PREFIX@/lib/emacsen-common/packages/remove/foo
 # [ Sample only -- this script hasn't been tested, so be careful. ]
 set -e
 
 flavor="$1"
-elc_dir="/usr/share/$flavor/site-lisp/foo"
+elc_dir="@PREFIX@/share/$flavor/site-lisp/foo"
 
 echo "remove/foo: Handling removal of emacsen flavor $flavor"
 
