diff -ruN emacs-26.3-orig/src/nsfns.m emacs-26.3/src/nsfns.m
--- emacs-26.3-orig/src/nsfns.m	2019-01-07 08:26:07.000000000 -0600
+++ emacs-26.3/src/nsfns.m	2021-06-25 07:02:24.000000000 -0500
@@ -49,6 +49,7 @@
 #include "macfont.h"
 #endif
 
+void mac_init_input_method ();
 
 #ifdef HAVE_NS
 
diff -ruN emacs-26.3-orig/src/nsterm.m emacs-26.3/src/nsterm.m
--- emacs-26.3-orig/src/nsterm.m	2019-01-23 07:53:23.000000000 -0600
+++ emacs-26.3/src/nsterm.m	2021-06-25 07:00:58.000000000 -0500
@@ -73,6 +73,10 @@
 static EmacsMenu *mainMenu;
 #endif
 
+int mac_pass_key_to_system (int code, int modifiers);
+int mac_pass_key_directly_to_emacs ();
+int mac_store_change_input_method_event ();
+
 /* ==========================================================================
 
    NSTRACE, Trace support.
