diff -Nurd -x'*~' ghex-2.24.0.orig/Makefile.am ghex-2.24.0/Makefile.am
--- ghex-2.24.0.orig/Makefile.am	2008-08-12 23:55:41.000000000 -0400
+++ ghex-2.24.0/Makefile.am	2016-02-28 06:06:18.000000000 -0500
@@ -1,6 +1,7 @@
 ## Process this file with automake to produce Makefile.in.
 
 DISTCHECK_CONFIGURE_FLAGS = --disable-scrollkeeper --disable-schemas-install
+ACLOCAL_AMFLAGS = -I m4
 
 SUBDIRS = po src help icons
 
diff -Nurd -x'*~' ghex-2.24.0.orig/configure.ac ghex-2.24.0/configure.ac
--- ghex-2.24.0.orig/configure.ac	2009-02-18 23:56:58.000000000 -0500
+++ ghex-2.24.0/configure.ac	2016-02-28 06:05:58.000000000 -0500
@@ -2,6 +2,7 @@
 AC_INIT([ghex], [2.24.0],
 	[http://bugzilla.gnome.org/enter_bug.cgi?product=ghex])
 AC_CONFIG_SRCDIR(src/ghex-window.c)
+AC_CONFIG_MACRO_DIRS([m4])
 AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER(config.h)
 
@@ -27,6 +28,7 @@
                    [GtkHex library release string.])
 
 PKG_CHECK_MODULES(LIBGTKHEX,
+		  gio-2.0		\
 		  gtk+-2.0 >= 2.4.0	\
 		  atk >= 1.0.0)
 
@@ -36,6 +38,7 @@
 AC_SUBST(LIBGTKHEX_CFLAGS)
 
 PKG_CHECK_MODULES(GHEX,
+		  gio-2.0			\
 		  gtk+-2.0 >= 2.4.0 		\
 		  libgnomeui-2.0 >= 2.6.0	\
 		  libgnomeprintui-2.2)
diff -Nurd -x'*~' ghex-2.24.0.orig/src/Makefile.am ghex-2.24.0/src/Makefile.am
--- ghex-2.24.0.orig/src/Makefile.am	2008-08-13 00:23:17.000000000 -0400
+++ ghex-2.24.0/src/Makefile.am	2016-02-28 05:35:29.000000000 -0500
@@ -33,9 +33,12 @@
 
 libgtkhex_la_LIBADD = \
 	$(LIBGTKHEX_LIBS)	\
-	$(GAILUTIL_LIBS)
+	$(GAILUTIL_LIBS)	\
+	$(INTLLIBS)
 
-libgtkhex_la_CFLAGS = -DG_LOG_DOMAIN=\"libgtkhex\"
+libgtkhex_la_CPPFLAGS = -DG_LOG_DOMAIN=\"libgtkhex\"
+
+libgtkhex_la_LDFLAGS = -no-undefined
 
 libgtkhex_HEADERS = \
 	hex-document.h 			\
diff -Nurd -x'*~' ghex-2.24.0.orig/src/ghex-window.c ghex-2.24.0/src/ghex-window.c
--- ghex-2.24.0.orig/src/ghex-window.c	2009-02-06 23:05:35.000000000 -0500
+++ ghex-2.24.0/src/ghex-window.c	2016-02-28 05:32:55.000000000 -0500
@@ -20,7 +20,7 @@
 #include <bonobo.h>
 #include <bonobo/bonobo-ui-main.h>
 
-#include <libgnomevfs/gnome-vfs-ops.h>
+#include <gio/gio.h>
 
 #include <math.h>
 #include <ctype.h>
@@ -1183,15 +1183,15 @@
 gboolean
 ghex_window_uri_exists (const gchar* text_uri)
 {
-	GnomeVFSURI *uri;
+	GFile *uri;
 	gboolean res;
 
 	g_return_val_if_fail (text_uri != NULL, FALSE);
-	uri = gnome_vfs_uri_new (text_uri);
+	uri = g_file_new_for_uri (text_uri);
 	g_return_val_if_fail (uri != NULL, FALSE);
-	res = gnome_vfs_uri_exists (uri);
+	res = g_file_query_exists (uri, NULL);
 
-	gnome_vfs_uri_unref (uri);
+	g_object_unref (uri);
 	return res;
 }
 
