diff -Nurd -x'*~' gobby-0.4.94.orig/code/Makefile.in gobby-0.4.94/code/Makefile.in
--- gobby-0.4.94.orig/code/Makefile.in	2011-03-27 13:34:03.000000000 -0400
+++ gobby-0.4.94/code/Makefile.in	2014-03-24 10:22:13.000000000 -0400
@@ -269,8 +269,7 @@
 noinst_HEADERS = window.hpp
 appicondir = $(datadir)/pixmaps
 pixmapdir = $(datadir)/pixmaps/gobby-0.5
-AM_CPPFLAGS = $(gobby_CFLAGS) $(unique_CFLAGS) $(infinote_CFLAGS) \
-              -I$(top_srcdir)/code
+AM_CPPFLAGS = -I$(top_srcdir)/code $(gobby_CFLAGS) $(unique_CFLAGS) $(infinote_CFLAGS)
 
 all: features.hpp
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
diff -Nurd -x'*~' gobby-0.4.94.orig/code/commands/Makefile.in gobby-0.4.94/code/commands/Makefile.in
--- gobby-0.4.94.orig/code/commands/Makefile.in	2011-03-27 13:34:03.000000000 -0400
+++ gobby-0.4.94/code/commands/Makefile.in	2014-03-24 10:22:47.000000000 -0400
@@ -298,8 +298,7 @@
 
 appicondir = $(datadir)/pixmaps
 pixmapdir = $(datadir)/pixmaps/gobby-0.5
-AM_CPPFLAGS = $(gobby_CFLAGS) $(unique_CFLAGS) $(infinote_CFLAGS) \
-              -I$(top_srcdir)/code
+AM_CPPFLAGS = -I$(top_srcdir)/code $(gobby_CFLAGS) $(unique_CFLAGS) $(infinote_CFLAGS)
 
 all: all-recursive
 
diff -Nurd -x'*~' gobby-0.4.94.orig/code/commands/file-tasks/Makefile.in gobby-0.4.94/code/commands/file-tasks/Makefile.in
--- gobby-0.4.94.orig/code/commands/file-tasks/Makefile.in	2011-03-27 13:34:03.000000000 -0400
+++ gobby-0.4.94/code/commands/file-tasks/Makefile.in	2014-03-24 10:22:53.000000000 -0400
@@ -249,8 +249,7 @@
 
 appicondir = $(datadir)/pixmaps
 pixmapdir = $(datadir)/pixmaps/gobby-0.5
-AM_CPPFLAGS = $(gobby_CFLAGS) $(unique_CFLAGS) $(infinote_CFLAGS) \
-              -I$(top_srcdir)/code
+AM_CPPFLAGS = -I$(top_srcdir)/code $(gobby_CFLAGS) $(unique_CFLAGS) $(infinote_CFLAGS)
 
 all: all-am
 
diff -Nurd -x'*~' gobby-0.4.94.orig/code/core/Makefile.in gobby-0.4.94/code/core/Makefile.in
--- gobby-0.4.94.orig/code/core/Makefile.in	2011-03-27 13:34:03.000000000 -0400
+++ gobby-0.4.94/code/core/Makefile.in	2014-03-24 10:22:40.000000000 -0400
@@ -290,8 +290,7 @@
 # private icons that are installed into prefix/share/gobby-0.5/icons instead
 # of prefix/share/icons. We also set PUBLIC_ICONS_DIR explicitely so that
 # we find the application icon if it is installed in a non-standard prefix.
-AM_CPPFLAGS = $(gobby_CFLAGS) $(unique_CFLAGS) $(infinote_CFLAGS) \
-              -I$(top_srcdir)/code \
+AM_CPPFLAGS = -I$(top_srcdir)/code $(gobby_CFLAGS) $(unique_CFLAGS) $(infinote_CFLAGS) \
 	      -DPRIVATE_ICONS_DIR=\""$(datadir)/gobby-0.5/icons"\" \
 	      -DPUBLIC_ICONS_DIR=\""$(datadir)/icons"\"
 
diff -Nurd -x'*~' gobby-0.4.94.orig/code/dialogs/Makefile.in gobby-0.4.94/code/dialogs/Makefile.in
--- gobby-0.4.94.orig/code/dialogs/Makefile.in	2011-03-27 13:34:03.000000000 -0400
+++ gobby-0.4.94/code/dialogs/Makefile.in	2014-03-24 10:22:19.000000000 -0400
@@ -251,8 +251,7 @@
 
 appicondir = $(datadir)/pixmaps
 pixmapdir = $(datadir)/pixmaps/gobby-0.5
-AM_CPPFLAGS = $(gobby_CFLAGS) $(unique_CFLAGS) $(infinote_CFLAGS) \
-              -I$(top_srcdir)/code
+AM_CPPFLAGS = -I$(top_srcdir)/code $(gobby_CFLAGS) $(unique_CFLAGS) $(infinote_CFLAGS)
 
 all: all-am
 
diff -Nurd -x'*~' gobby-0.4.94.orig/code/operations/Makefile.in gobby-0.4.94/code/operations/Makefile.in
--- gobby-0.4.94.orig/code/operations/Makefile.in	2011-03-27 13:34:03.000000000 -0400
+++ gobby-0.4.94/code/operations/Makefile.in	2014-03-24 10:22:07.000000000 -0400
@@ -246,8 +246,7 @@
 
 appicondir = $(datadir)/pixmaps
 pixmapdir = $(datadir)/pixmaps/gobby-0.5
-AM_CPPFLAGS = $(gobby_CFLAGS) $(unique_CFLAGS) $(infinote_CFLAGS) \
-              -I$(top_srcdir)/code
+AM_CPPFLAGS = -I$(top_srcdir)/code $(gobby_CFLAGS) $(unique_CFLAGS) $(infinote_CFLAGS)
 
 all: all-am
 
diff -Nurd -x'*~' gobby-0.4.94.orig/code/util/Makefile.in gobby-0.4.94/code/util/Makefile.in
--- gobby-0.4.94.orig/code/util/Makefile.in	2011-03-27 13:34:03.000000000 -0400
+++ gobby-0.4.94/code/util/Makefile.in	2014-03-24 10:21:58.000000000 -0400
@@ -247,8 +247,7 @@
         resolv.hpp \
         serialize.hpp
 
-AM_CPPFLAGS = $(gobby_CFLAGS) $(unique_CFLAGS) $(infinote_CFLAGS) \
-              -I$(top_srcdir)/code
+AM_CPPFLAGS = -I$(top_srcdir)/code $(gobby_CFLAGS) $(unique_CFLAGS) $(infinote_CFLAGS)
 
 all: all-am
 
