diff -ru sam-orig/libXg/Makefile sam/libXg/Makefile
--- sam-orig/libXg/Makefile	2011-07-23 15:51:24.000000000 +0200
+++ sam/libXg/Makefile	2011-07-23 15:57:46.000000000 +0200
@@ -8,7 +8,7 @@
 #	
 #	Additionally, -D_POSIX_SOURCE (or its equivalent) may be specified
 #	if your compiler supports posix-compatible compilation
-OS=-DIRIX -ansiposix
+OS=-D_POSIX_SOURCE -DSOLARIS
 
 #	add -Iincludedir for any include directories that need to be searched
 INCS=-I../include
diff -ru sam-orig/libframe/Makefile sam/libframe/Makefile
--- sam-orig/libframe/Makefile	2011-07-23 15:51:24.000000000 +0200
+++ sam/libframe/Makefile	2011-07-23 15:56:33.000000000 +0200
@@ -8,7 +8,7 @@
 #	
 #	Additionally, -D_POSIX_SOURCE (or its equivalent) may be specified
 #	if your compiler supports posix-compatible compilation
-OS=-DIRIX -ansiposix
+OS=-D_POSIX_SOURCE -DSOLARIS
 
 #	add -Iincludedir for any include directories that need to be searched
 #	for posix header files (for UMIPS, add -I/usr/include/posix)
diff -ru sam-orig/sam/Makefile sam/sam/Makefile
--- sam-orig/sam/Makefile	2011-07-23 15:51:24.000000000 +0200
+++ sam/sam/Makefile	2011-07-23 16:02:19.000000000 +0200
@@ -8,7 +8,7 @@
 #	
 #	Additionally, -D_POSIX_SOURCE (or its equivalent) may be specified
 #	if your compiler supports posix-compatible compilation
-OS=-DIRIX -ansiposix
+OS=-D_POSIX_SOURCE -DSOLARIS
 
 #	add -Iincludedir for any include directories that need to be searched
 #	for posix header files (for UMIPS, add -I/usr/include/posix)
diff -ru sam-orig/sam/sam.c sam/sam/sam.c
--- sam-orig/sam/sam.c	2011-07-23 15:51:24.000000000 +0200
+++ sam/sam/sam.c	2011-07-23 16:03:21.000000000 +0200
@@ -30,7 +30,7 @@
 
 void	usage(void);
 
-void
+int
 main(int argc, char *argv[])
 {
 	int i;
@@ -124,6 +124,8 @@
 	cmdloop();
 	trytoquit();	/* if we already q'ed, quitok will be TRUE */
 	exits(0);
+	
+	return 0;
 }
 
 void
diff -ru sam-orig/samterm/Makefile sam/samterm/Makefile
--- sam-orig/samterm/Makefile	2011-07-23 15:51:24.000000000 +0200
+++ sam/samterm/Makefile	2011-07-23 15:58:18.000000000 +0200
@@ -8,7 +8,7 @@
 #	
 #	Additionally, -D_POSIX_SOURCE (or its equivalent) may be specified
 #	if your compiler supports posix-compatible compilation
-OS=-DIRIX -ansiposix
+OS=-D_POSIX_SOURCE -DSOLARIS
 
 #	add -Iincludedir for any include directories that need to be searched
 #	for posix header files (for UMIPS, add -I/usr/include/posix)
@@ -21,7 +21,7 @@
 
 #	set this if your X libraries are in different locations
 #	or if you need extra libraries to load with X11 applications
-XLIBS=/usr/local/X11R5/lib/libXt.a /usr/local/X11R5/lib/libX11.a
+XLIBS=-L/opt/X11/lib -lXt -lX11 -lSM -lICE
 
 CFLAGS=$(OS) $(INCS) -D_LIBXG_EXTENSION
 
diff -ru sam-orig/samterm/main.c sam/samterm/main.c
--- sam-orig/samterm/main.c	2011-07-23 15:51:24.000000000 +0200
+++ sam/samterm/main.c	2011-07-23 16:04:26.000000000 +0200
@@ -22,7 +22,7 @@
 char	lock = 1;
 char	hasunlocked = 0;
 
-void
+int
 main(int argc, char *argv[])
 {
 	int i, got, scr;
@@ -106,6 +106,7 @@
 			mouseunblock();
 		}
 	}
+	exit(0);
 }
 
 
