diff -Nurd scite344-orig/scite/gtk/deps.mak scite344/scite/gtk/deps.mak
--- scite344-orig/scite/gtk/deps.mak	2016-10-05 21:35:16.000000000 -0500
+++ scite344/scite/gtk/deps.mak	2016-10-05 21:36:32.000000000 -0500
@@ -53,9 +53,7 @@
 LuaExtension.o: ../src/LuaExtension.cxx \
  ../../scintilla/include/Scintilla.h ../src/GUI.h ../src/SString.h \
  ../src/FilePath.h ../src/StyleWriter.h ../src/Extender.h \
- ../src/LuaExtension.h ../src/IFaceTable.h ../src/SciTEKeys.h \
- ../lua/include/lua.h ../lua/include/luaconf.h ../lua/include/lualib.h \
- ../lua/include/lua.h ../lua/include/lauxlib.h
+ ../src/LuaExtension.h ../src/IFaceTable.h ../src/SciTEKeys.h
 MatchMarker.o: ../src/MatchMarker.cxx ../../scintilla/include/Scintilla.h \
  ../src/GUI.h ../src/MatchMarker.h
 MultiplexExtension.o: ../src/MultiplexExtension.cxx \
diff -Nurd scite344-orig/scite/gtk/makefile scite344/scite/gtk/makefile
--- scite344-orig/scite/gtk/makefile	2016-10-05 21:35:16.000000000 -0500
+++ scite344/scite/gtk/makefile	2016-10-05 21:38:32.000000000 -0500
@@ -59,18 +59,15 @@
 endif
 
 ifndef NO_LUA
-LUA_CORE_OBJS = lapi.o lcode.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o \
-		lmem.o lobject.o lopcodes.o lparser.o lstate.o lstring.o \
-		ltable.o ltm.o lundump.o lvm.o lzio.o
+LUA_CORE_OBJS = 
 
-LUA_LIB_OBJS =	lauxlib.o lbaselib.o ldblib.o liolib.o lmathlib.o ltablib.o \
-		lstrlib.o loadlib.o loslib.o linit.o
+LUA_LIB_OBJS = 
 
 LUA_OBJS = LuaExtension.o $(LUA_CORE_OBJS) $(LUA_LIB_OBJS)
 
-vpath %.c ../lua/src ../lua/src/lib
+vpath %.c 
 
-INCLUDEDIRS=-I ../../scintilla/include -I ../src -I../lua/include
+INCLUDEDIRS=-I ../../scintilla/include -I ../src 
 
 .c.o:
 	$(CC) $(CFLAGS) $(INCLUDEDIRS) -DLUA_USE_POSIX $(CXXTFLAGS) -c $< -o $@
@@ -114,7 +111,7 @@
 Exporters.o  MatchMarker.o StringHelpers.o \
 PropSetFile.o MultiplexExtension.o DirectorExtension.o SciTEProps.o StyleDefinition.o StyleWriter.o Utf8_16.o \
 	JobQueue.o GTKMutex.o IFaceTable.o $(COMPLIB) $(LUA_OBJS)
-	$(CXX) `$(CONFIGTHREADS)` -rdynamic -Wl,--as-needed -Wl,--version-script lua.vers -DGTK $^ -o $@ $(CONFIGLIB) $(LIBDL) -lm -lstdc++
+	$(CXX) `$(CONFIGTHREADS)` -DGTK $^ -o $@ $(CONFIGLIB) $(LIBDL) -lm -llua
 
 # Automatically generate header dependencies with "make deps"
 include deps.mak
diff -ruN scite344-orig/scintilla/gtk/ScintillaGTK.cxx scite344/scintilla/gtk/ScintillaGTK.cxx
--- scite344-orig/scintilla/gtk/ScintillaGTK.cxx	2014-06-15 20:39:28.000000000 -0500
+++ scite344/scintilla/gtk/ScintillaGTK.cxx	2018-04-19 22:46:34.000000000 -0500
@@ -1578,7 +1578,7 @@
 		drop.push_back('\0');
 		NotifyURIDropped(&drop[0]);
 	} else if ((TypeOfGSD(selection_data) == GDK_TARGET_STRING) || (TypeOfGSD(selection_data) == atomUTF8)) {
-		if (TypeOfGSD(selection_data) > 0) {
+		if (LengthOfGSD(selection_data) > 0) {
 			SelectionText selText;
 			GetGtkSelectionText(selection_data, selText);
 			DropAt(posDrop, selText.Data(), selText.Length(), false, selText.rectangular);
