diff -uNr xemacs-21.4.22.orig/lwlib/Makefile.in.in xemacs-21.4.22/lwlib/Makefile.in.in
--- xemacs-21.4.22.orig/lwlib/Makefile.in.in	2001-04-12 14:21:51.000000000 -0400
+++ xemacs-21.4.22/lwlib/Makefile.in.in	2021-06-07 09:37:09.000000000 -0400
@@ -54,7 +54,7 @@
 
 c_switch_all=@c_switch_all@
 
-cppflags = $(CPPFLAGS) -I. $(c_switch_all)
+cppflags = -I. $(CPPFLAGS) $(c_switch_all)
 cflags = $(CFLAGS) $(cppflags)
 
 #ifdef SOLARIS2
diff -uNr xemacs-21.4.22.orig/lwlib/xlwgauge.c xemacs-21.4.22/lwlib/xlwgauge.c
--- xemacs-21.4.22.orig/lwlib/xlwgauge.c	2007-08-20 15:51:30.000000000 -0400
+++ xemacs-21.4.22/lwlib/xlwgauge.c	2016-06-01 16:53:41.000000000 -0400
@@ -839,7 +839,7 @@
 	if(( gw->gauge.autoScaleUp && value > gw->gauge.v1) ||
 	   (gw->gauge.autoScaleDown && value < gw->gauge.v1/3 ))
 	{
-	  XtVaSetValues(w, XtNvalue, value, 0) ;
+	  XtVaSetValues(w, XtNvalue, value, 0, NULL) ;
 	  return ;
 	}
 
diff -uNr xemacs-21.4.22.orig/nt/xemacs.mak xemacs-21.4.22/nt/xemacs.mak
--- xemacs-21.4.22.orig/nt/xemacs.mak	2008-01-23 07:14:02.000000000 -0500
+++ xemacs-21.4.22/nt/xemacs.mak	2021-06-07 09:20:26.000000000 -0400
@@ -502,7 +502,7 @@
 
 # Generic variables
 
-INCLUDES=$(X_INCLUDES) -I$(NT)\inc -I$(SRC) $(MSW_INCLUDES) -I$(LWLIB_SRCDIR)
+INCLUDES=-I$(SRC) $(X_INCLUDES) -I$(NT)\inc $(MSW_INCLUDES) -I$(LWLIB_SRCDIR)
 
 DEFINES=$(X_DEFINES) $(MSW_DEFINES) $(MULE_DEFINES) $(UNION_DEFINES) \
 	$(DUMPER_DEFINES) $(MALLOC_DEFINES) $(QUICK_DEFINES) \
diff -uNr xemacs-21.4.22.orig/src/Makefile.in.in xemacs-21.4.22/src/Makefile.in.in
--- xemacs-21.4.22.orig/src/Makefile.in.in	2005-01-30 21:55:02.000000000 -0500
+++ xemacs-21.4.22/src/Makefile.in.in	2021-06-07 09:19:32.000000000 -0400
@@ -149,7 +149,7 @@
 ## -Demacs is needed to make some files produce the correct version
 ## for use in Emacs.
 
-cppflags = $(CPPFLAGS) -Demacs -I. $(c_switch_all)
+cppflags = -I. $(CPPFLAGS) -Demacs $(c_switch_all)
 cflags   = $(CFLAGS) $(cppflags)
 #if defined (WIN32_NATIVE)
 ldflags  = $(LDFLAGS) -mwindows -e _mainCRTStartup $(ld_switch_all) $(ld_dynamic_link_flags)
diff -uNr xemacs-21.4.22.orig/src/frame-x.c xemacs-21.4.22/src/frame-x.c
--- xemacs-21.4.22.orig/src/frame-x.c	2005-01-30 21:55:16.000000000 -0500
+++ xemacs-21.4.22/src/frame-x.c	2016-06-01 16:52:32.000000000 -0400
@@ -777,12 +777,12 @@
 				  Qctext);
 	      XtVaSetValues (w, XtVaTypedArg, extprop,
 			     XtRString, extval, extvallen + 1,
-			     (XtArgVal) NULL);
+			     (XtArgVal) NULL, NULL);
 	    }
 	  else
 	    XtVaSetValues (w, XtVaTypedArg, extprop, XtRInt,
 			   XINT (val), sizeof (int),
-			   (XtArgVal) NULL);
+			   (XtArgVal) NULL, NULL);
 	}
       else if (SYMBOLP (prop))
 	{
@@ -874,7 +874,7 @@
 			     XtRString,
 			     XSTRING_DATA (val),
 			     XSTRING_LENGTH (val) + 1,
-			     (XtArgVal) NULL);
+			     (XtArgVal) NULL, NULL);
 	    }
 
 #ifdef HAVE_SCROLLBARS
diff -uNr xemacs-21.4.22.orig/src/glyphs-eimage.c xemacs-21.4.22/src/glyphs-eimage.c
--- xemacs-21.4.22.orig/src/glyphs-eimage.c	2007-08-20 15:52:28.000000000 -0400
+++ xemacs-21.4.22/src/glyphs-eimage.c	2016-06-01 16:50:51.000000000 -0400
@@ -937,8 +937,8 @@
   {
     int y;
     unsigned char **row_pointers;
-    height = info_ptr->height;
-    width = info_ptr->width;
+    height = png_get_image_height(png_ptr, info_ptr);
+    width = png_get_image_width(png_ptr, info_ptr);
 
     /* Wow, allocate all the memory.  Truly, exciting. */
     unwind.eimage = xnew_array_and_zero (unsigned char, width * height * 3);
@@ -990,22 +990,22 @@
     /* Now that we're using EImage, ask for 8bit RGB triples for any type
        of image*/
     /* convert palette images to full RGB */
-    if (info_ptr->color_type == PNG_COLOR_TYPE_PALETTE)
+    if (png_get_color_type(png_ptr, info_ptr) == PNG_COLOR_TYPE_PALETTE)
       png_set_expand (png_ptr);
     /* send grayscale images to RGB too */
-    if (info_ptr->color_type == PNG_COLOR_TYPE_GRAY ||
-        info_ptr->color_type == PNG_COLOR_TYPE_GRAY_ALPHA)
+    if (png_get_color_type(png_ptr, info_ptr) == PNG_COLOR_TYPE_GRAY ||
+        png_get_color_type(png_ptr, info_ptr) == PNG_COLOR_TYPE_GRAY_ALPHA)
       png_set_gray_to_rgb (png_ptr);
     /* we can't handle alpha values */
-    if (info_ptr->color_type & PNG_COLOR_MASK_ALPHA)
+    if (png_get_color_type(png_ptr, info_ptr) & PNG_COLOR_MASK_ALPHA)
       png_set_strip_alpha (png_ptr);
     /* tell libpng to strip 16 bit depth files down to 8 bits */
-    if (info_ptr->bit_depth == 16)
+    if (png_get_bit_depth(png_ptr, info_ptr) == 16)
       png_set_strip_16 (png_ptr);
     /* if the image is < 8 bits, pad it out */
-    if (info_ptr->bit_depth < 8)
+    if (png_get_bit_depth(png_ptr, info_ptr) < 8)
       {
-	if (info_ptr->color_type == PNG_COLOR_TYPE_GRAY)
+	if (png_get_color_type(png_ptr, info_ptr) == PNG_COLOR_TYPE_GRAY)
 	  png_set_expand (png_ptr);
 	else
 	  png_set_packing (png_ptr);
@@ -1023,17 +1023,22 @@
      * from lisp anyway. - WMP
      */
     {
-      int i;
+      int i, num_text = 0;
+      png_textp text_ptr = NULL;
 
-      for (i = 0 ; i < info_ptr->num_text ; i++)
+      if (png_get_text (png_ptr, info_ptr, &text_ptr, &num_text) > 0)
 	{
-	  /* How paranoid do I have to be about no trailing NULLs, and
-	     using (int)info_ptr->text[i].text_length, and strncpy and a temp
-	     string somewhere? */
-
-	  warn_when_safe (Qpng, Qinfo, "%s - %s",
-			  info_ptr->text[i].key,
-			  info_ptr->text[i].text);
+	  for (i = 0 ; i < num_text ; i++)
+	    {
+
+	      /* How paranoid do I have to be about no trailing NULLs, and
+		 using (int)info_ptr->text[i].text_length, and strncpy and a temp
+		 string somewhere? */
+
+	      warn_when_safe (Qpng, Qinfo, "%s - %s",
+			      text_ptr[i].key, text_ptr[i].text);
+	    }
+
 	}
     }
 #endif
