diff -uNr xemacs-21.4.22.orig/lwlib/Makefile.in.in xemacs-21.4.22/lwlib/Makefile.in.in
--- xemacs-21.4.22.orig/lwlib/Makefile.in.in	2001-04-12 14:21:51.000000000 -0400
+++ xemacs-21.4.22/lwlib/Makefile.in.in	2021-06-07 09:37:09.000000000 -0400
@@ -54,7 +54,7 @@
 
 c_switch_all=@c_switch_all@
 
-cppflags = $(CPPFLAGS) -I. $(c_switch_all)
+cppflags = -I. $(CPPFLAGS) $(c_switch_all)
 cflags = $(CFLAGS) $(cppflags)
 
 #ifdef SOLARIS2
diff -uNr xemacs-21.4.22.orig/lwlib/xlwgauge.c xemacs-21.4.22/lwlib/xlwgauge.c
--- xemacs-21.4.22.orig/lwlib/xlwgauge.c	2007-08-20 15:51:30.000000000 -0400
+++ xemacs-21.4.22/lwlib/xlwgauge.c	2016-06-01 16:53:41.000000000 -0400
@@ -839,7 +839,7 @@
 	if(( gw->gauge.autoScaleUp && value > gw->gauge.v1) ||
 	   (gw->gauge.autoScaleDown && value < gw->gauge.v1/3 ))
 	{
-	  XtVaSetValues(w, XtNvalue, value, 0) ;
+	  XtVaSetValues(w, XtNvalue, value, 0, NULL) ;
 	  return ;
 	}
 
diff -uNr xemacs-21.4.22.orig/nt/xemacs.mak xemacs-21.4.22/nt/xemacs.mak
--- xemacs-21.4.22.orig/nt/xemacs.mak	2008-01-23 07:14:02.000000000 -0500
+++ xemacs-21.4.22/nt/xemacs.mak	2021-06-07 09:20:26.000000000 -0400
@@ -502,7 +502,7 @@
 
 # Generic variables
 
-INCLUDES=$(X_INCLUDES) -I$(NT)\inc -I$(SRC) $(MSW_INCLUDES) -I$(LWLIB_SRCDIR)
+INCLUDES=-I$(SRC) $(X_INCLUDES) -I$(NT)\inc $(MSW_INCLUDES) -I$(LWLIB_SRCDIR)
 
 DEFINES=$(X_DEFINES) $(MSW_DEFINES) $(MULE_DEFINES) $(UNION_DEFINES) \
 	$(DUMPER_DEFINES) $(MALLOC_DEFINES) $(QUICK_DEFINES) \
diff -uNr xemacs-21.4.22.orig/src/Makefile.in.in xemacs-21.4.22/src/Makefile.in.in
--- xemacs-21.4.22.orig/src/Makefile.in.in	2005-01-30 21:55:02.000000000 -0500
+++ xemacs-21.4.22/src/Makefile.in.in	2021-06-07 09:19:32.000000000 -0400
@@ -149,7 +149,7 @@
 ## -Demacs is needed to make some files produce the correct version
 ## for use in Emacs.
 
-cppflags = $(CPPFLAGS) -Demacs -I. $(c_switch_all)
+cppflags = -I. $(CPPFLAGS) -Demacs $(c_switch_all)
 cflags   = $(CFLAGS) $(cppflags)
 #if defined (WIN32_NATIVE)
 ldflags  = $(LDFLAGS) -mwindows -e _mainCRTStartup $(ld_switch_all) $(ld_dynamic_link_flags)
diff -uNr xemacs-21.4.22.orig/src/frame-x.c xemacs-21.4.22/src/frame-x.c
--- xemacs-21.4.22.orig/src/frame-x.c	2005-01-30 21:55:16.000000000 -0500
+++ xemacs-21.4.22/src/frame-x.c	2016-06-01 16:52:32.000000000 -0400
@@ -777,12 +777,12 @@
 				  Qctext);
 	      XtVaSetValues (w, XtVaTypedArg, extprop,
 			     XtRString, extval, extvallen + 1,
-			     (XtArgVal) NULL);
+			     (XtArgVal) NULL, NULL);
 	    }
 	  else
 	    XtVaSetValues (w, XtVaTypedArg, extprop, XtRInt,
 			   XINT (val), sizeof (int),
-			   (XtArgVal) NULL);
+			   (XtArgVal) NULL, NULL);
 	}
       else if (SYMBOLP (prop))
 	{
@@ -874,7 +874,7 @@
 			     XtRString,
 			     XSTRING_DATA (val),
 			     XSTRING_LENGTH (val) + 1,
-			     (XtArgVal) NULL);
+			     (XtArgVal) NULL, NULL);
 	    }
 
 #ifdef HAVE_SCROLLBARS
