diff -ruN xemacs-21.4.24-orig/configure xemacs-21.4.24/configure
--- xemacs-21.4.24-orig/configure	2015-03-25 06:25:33.000000000 -0500
+++ xemacs-21.4.24/configure	2023-11-12 04:49:09.000000000 -0600
@@ -3658,6 +3658,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 3660 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <ctype.h>
 #define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 #define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
@@ -4515,6 +4516,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 4517 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <stdio.h>
 #include <sys/types.h>
 main()
@@ -4558,6 +4560,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 4560 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <stdio.h>
 #include <sys/types.h>
 main()
@@ -4595,6 +4598,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 4597 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <stdio.h>
 #include <sys/types.h>
 main()
@@ -4632,6 +4636,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 4634 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <stdio.h>
 #include <sys/types.h>
 main()
@@ -4669,6 +4674,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 4671 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <stdio.h>
 #include <sys/types.h>
 main()
@@ -7155,6 +7161,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 7157 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <stdio.h>
 #include <sys/types.h>
 main()
@@ -8973,6 +8980,7 @@
     cat > conftest.$ac_ext <<EOF
 #line 8975 "configure"
 #include "confdefs.h"
+#include <string.h>
 #include <png.h>
     int main(int c, char **v) {
     if (c == 1) return 0;
@@ -12493,6 +12501,7 @@
 #line 12494 "configure"
 #include "confdefs.h"
 /* Thanks to Paul Eggert for this test.  */
+#include <stdlib.h>
 #include <stdio.h>
 #include <sys/types.h>
 #include <sys/stat.h>
@@ -12616,6 +12625,7 @@
 cat > conftest.$ac_ext <<EOF
 #line 12618 "configure"
 #include "confdefs.h"
+#include <stdlib.h>
 #include <string.h>
 main ()
 {
@@ -12718,8 +12728,10 @@
  *
  * Snarfed from Chet Ramey's bash pgrp.c test program
  */
+#include <stdlib.h>
 #include <stdio.h>
 #include <sys/types.h>
+#include <unistd.h>
 
 int     pid;
 int     pg1, pg2, pg3, pg4;
