diff -Naur latex2html-2002-2-1.orig/config/install.pl latex2html-2002-2-1/config/install.pl
--- latex2html-2002-2-1.orig/config/install.pl	2002-11-29 20:49:01.000000000 +0900
+++ latex2html-2002-2-1/config/install.pl	2006-04-02 22:46:31.000000000 +0900
@@ -294,12 +294,12 @@
 my $iconrx = join('|', @icon_types);
 
 my $dest1 = "$cfg{'SHLIBDIR'}${dd}icons";
-if((-d $dest1 && !-w _) || (-d $cfg{'SHLIBDIR'} && !-w _)) {
+if((-d "$ENV{'DESTDIR'}$dest1" && !-w _) || (-d "$ENV{'DESTDIR'}$cfg{'SHLIBDIR'}" && !-w _)) {
   print STDERR "Error: Cannot install icons in '$dest1': No write permission.\n";
   $dest1 = '';
 }
 my $dest2 = $cfg{'ICONSTORAGE'} || '';
-if(-d $dest2 && !-w $dest2) {
+if(-d "$ENV{'DESTDIR'}$dest2" && !-w _) {
   print STDERR "Error: Cannot install icons in '$dest2': No write permission.\n";
   $dest2 = '';
 }
diff -Naur latex2html-2002-2-1.orig/prefs.pm latex2html-2002-2-1/prefs.pm
--- latex2html-2002-2-1.orig/prefs.pm	2006-04-02 21:04:18.000000000 +0900
+++ latex2html-2002-2-1/prefs.pm	2006-04-02 21:04:55.000000000 +0900
@@ -220,7 +220,7 @@
 # $prefs{'DVIPSOPT'} = ' -E';  # create encapsulated (EPS) images
 # $prefs{'DVIPSOPT'} = ' -Ppdf'; # use Type 1 fonts, as for PDF
 # $prefs{'DVIPSOPT'} = ' -Pcmz -Pamz'; # use CM and AMS Type 1 fonts
-$prefs{'DVIPSOPT'} = ' -Ppdf';
+$prefs{'DVIPSOPT'} = ' ';
 
 # set this to 1 if you want dvips to create special fonts for better
 # image generation. You need to set the following two items to correct
diff -Naur latex2html-2002-2-1.orig/pstoimg.pin latex2html-2002-2-1/pstoimg.pin
--- latex2html-2002-2-1.orig/pstoimg.pin	2004-01-03 15:14:59.000000000 +0900
+++ latex2html-2002-2-1/pstoimg.pin	2006-04-02 21:20:50.000000000 +0900
@@ -330,7 +330,7 @@
 #-
 #if @IMAGE_TYPES@ =~ /gif/i
 # GIF support
-my $PPMTOGIF = '@PPMTOGIF@';
+my $PPMTOGIF = '@PPMTOGIF@ -nolzw';
 #if @GIFTRANS@
 my $GIFTRANS = '@GIFTRANS@';
 #fi
diff -Naur latex2html-2002-2-1.orig/styles/japanese.perl latex2html-2002-2-1/styles/japanese.perl
--- latex2html-2002-2-1.orig/styles/japanese.perl	2006-04-02 21:04:18.000000000 +0900
+++ latex2html-2002-2-1/styles/japanese.perl	2006-04-02 21:04:44.000000000 +0900
@@ -20,14 +20,42 @@
     $lof_title = "図目次";
     $lot_title = "表目次";
     $idx_title = "索引";
-    $bib_title = "参考文献";
+    $ref_title = "参考文献";
+    $bib_title = "文献目録";
     $abs_title = "概要";
+    $app_title = "補遺";
+    $pre_title = "Preface";
+    $foot_title = "脚注";
+    $thm_title = "定理";
     $fig_name  = "図";
     $tab_name  = "表";
+    $prf_name = "証明";
+    $date_name = "日付";
+    $page_name = "ページ";
+  #  Sectioning-level titles
+    $part_name = "Part";
+    $chapter_name = "章";
+    $section_name = "節";
+    $subsection_name = "節";
+    $subsubsection_name = "節";
+    $paragraph_name = "節";
+  #  Misc. strings
+    $child_name = "Subsections";
     $info_title = "この文書について..."; 
+    $also_name = "see also";
+    $see_name = "see";
+  #  names in navigation panels
+    $next_name = "次へ";
+    $up_name = "上へ";
+    $prev_name = "戻る";
+  #  field names in email headers
+    $encl_name = "encl";
+    $headto_name = "To";
+    $cc_name = "cc";
     @Month = ('', '1月', '2月', '3月', '4月', '5月',
               '6月', '7月', '8月', '9月', '10月',
               '11月', '12月');
+    $GENERIC_WORDS = "and|the|of|for|by|a|an|to";
 #    @Month = ('', '睦月', '如月', '弥生', '卯月', '皐月',
 #              '水無月', '文月', '葉月', '長月', '神無月',
 #              '霜月', '師走');
@@ -83,6 +111,12 @@
         sprintf("平成%s年%s%d日", $y, @Month[$m+1], $d);
 }
 
+sub japanese_today {
+    local($today) = &get_date();
+    $today =~ s|(\d+)/0?(\d+)/|$2 $Month[$1] |;
+    join('',$today,$_[0]);
+}
+
 &japanese_titles;
 $default_language = 'japanese';
 $TITLES_LANGUAGE = 'japanese';
