diff -Nurd miktex-2.8.3541/BuildUtilities/c4p/gram.y miktex-new-2.8.3541/BuildUtilities/c4p/gram.y
--- miktex-2.8.3541/BuildUtilities/c4p/gram.y	2009-01-20 17:41:55.000000000 +0100
+++ miktex-new-2.8.3541/BuildUtilities/c4p/gram.y	2011-03-23 12:55:57.000000000 +0100
@@ -23,7 +23,6 @@
 #include <cstring>
 #include <cstdlib>
 #include <cctype>
-#include <malloc.h>
 
 #include "common.h"
 
diff -Nurd miktex-2.8.3541/CMakeLists.txt miktex-new-2.8.3541/CMakeLists.txt
--- miktex-2.8.3541/CMakeLists.txt	2009-02-24 12:37:30.000000000 +0100
+++ miktex-new-2.8.3541/CMakeLists.txt	2011-03-23 13:01:15.000000000 +0100
@@ -249,13 +249,14 @@
   set(HAVE_MPM_QT_SOURCES 1)
 endif(EXISTS ${CMAKE_SOURCE_DIR}/${rel_mpm_qt_dir}/)
 
+# Fink on Mac OS X 10.4, use flex-devel flex
+SET( FLEX_EXE /usr/bin/flex )
+
 set(required_programs
   bison
   cat
   date
   diff
-  flex
-  lynx
   perl
   sed
   tr
@@ -505,6 +506,9 @@
     "${CMAKE_RC_FLAGS} -I${CMAKE_SOURCE_DIR}/Resources/Videos")
 else(NATIVE_WINDOWS)
   add_definitions(-DMIKTEX_UNIX)
+
+  link_libraries(z)
+  link_libraries(png)
 endif(NATIVE_WINDOWS)
 
 ###############################################################################
diff -Nurd miktex-2.8.3541/Libraries/MiKTeX/KPathSeaEmulation/StdAfx.h miktex-new-2.8.3541/Libraries/MiKTeX/KPathSeaEmulation/StdAfx.h
--- miktex-2.8.3541/Libraries/MiKTeX/KPathSeaEmulation/StdAfx.h	2008-12-21 20:30:05.000000000 +0100
+++ miktex-new-2.8.3541/Libraries/MiKTeX/KPathSeaEmulation/StdAfx.h	2011-03-23 12:55:57.000000000 +0100
@@ -22,7 +22,6 @@
 #pragma once
 
 #include <cstdlib>
-#include <malloc.h>
 #if defined(MIKTEX_UNIX)
 #include <sys/time.h>
 #endif
diff -Nurd miktex-2.8.3541/Libraries/MiKTeX/KPathSeaEmulation/kpsemu.cpp miktex-new-2.8.3541/Libraries/MiKTeX/KPathSeaEmulation/kpsemu.cpp
--- miktex-2.8.3541/Libraries/MiKTeX/KPathSeaEmulation/kpsemu.cpp	2009-01-26 16:23:49.000000000 +0100
+++ miktex-new-2.8.3541/Libraries/MiKTeX/KPathSeaEmulation/kpsemu.cpp	2011-03-23 12:55:57.000000000 +0100
@@ -527,6 +527,13 @@
       FATAL_CRT_ERROR ("fseeko64", lpszFileName);
     }
   return (0);
+#elif defined(__APPLE__)
+  // Mac OS X fseeko is already 64-bits
+  if (fseeko(pfile, offset, where) != 0)
+    {
+      FATAL_CRT_ERROR ("fseeko", lpszFileName);
+    }
+  return (0);
 #else
 #  error Unimplemented: KPSE::FSeek64()
 #endif
@@ -573,6 +580,14 @@
     }
   return (pos);
   
+#elif defined(__APPLE__)
+  // Mac OS X ftello is already 64-bits
+  MIKTEX_INT64 pos = ftello(pfile);
+  if (pos < 0)
+    {
+      FATAL_CRT_ERROR ("ftello", lpszFileName);
+    }
+  return (pos);
 #else
 #  error Unimplemented: KPSE::FTell64
 #endif
diff -Nurd miktex-2.8.3541/Programs/MiKTeX/makex/internal.h miktex-new-2.8.3541/Programs/MiKTeX/makex/internal.h
--- miktex-2.8.3541/Programs/MiKTeX/makex/internal.h	2009-01-04 21:48:59.000000000 +0100
+++ miktex-new-2.8.3541/Programs/MiKTeX/makex/internal.h	2011-03-23 12:55:57.000000000 +0100
@@ -37,8 +37,6 @@
 #include <cstdlib>
 #include <string>
 
-#include <malloc.h>
-
 #include <getopt.h>
 #include <miktex/App/Application>
 #include <miktex/Core/Core>
