Index: contrib/Makefile
===================================================================
RCS file: /net/pauillac/caml/repository/camllight/sources/contrib/Makefile,v
retrieving revision 1.28
diff -c -r1.28 Makefile
*** contrib/Makefile	2002/02/11 08:33:24	1.28
--- contrib/Makefile	2002/10/02 13:45:12
***************
*** 4,11 ****
  # See the file INDEX for a description of the packages and their requirements.
  # Remember that "libunix" is required for
  # "debugger", "libgraph", "camltk", "camltk4", and "search_isos".
! PACKAGES=libunix libgraph debugger libnum libstr mletags \
!   camlmode lorder profiler camltk4 camlsearch
  # caml-tex
  # caml-latex2e
  # camltk
--- 4,11 ----
  # See the file INDEX for a description of the packages and their requirements.
  # Remember that "libunix" is required for
  # "debugger", "libgraph", "camltk", "camltk4", and "search_isos".
! PACKAGES=libunix debugger libnum libstr mletags \
!   camlmode lorder profiler camlsearch
  # caml-tex
  # caml-latex2e
  # camltk
Index: src/Makefile
===================================================================
RCS file: /net/pauillac/caml/repository/camllight/sources/src/Makefile,v
retrieving revision 1.13
diff -c -r1.13 Makefile
*** src/Makefile	1999/02/28 14:07:26	1.13
--- src/Makefile	2002/10/02 13:45:15
***************
*** 9,15 ****
  # This option circumvents a gcc bug on some platforms (680x0, 80386).
  # If you are using Linux with libc6 (RedHat 5, Debian 2), add -D__FAVOR_BSD
  # This option avoids signal-related problems.
! OPTS=-fno-defer-pop -D__FAVOR_BSD
  
  # Extra libraries that have to be linked with the runtime system.
  # The math library "-lm" is linked by default.
--- 9,15 ----
  # This option circumvents a gcc bug on some platforms (680x0, 80386).
  # If you are using Linux with libc6 (RedHat 5, Debian 2), add -D__FAVOR_BSD
  # This option avoids signal-related problems.
! OPTS=-fno-defer-pop -D__FAVOR_BSD -no-cpp-precomp
  
  # Extra libraries that have to be linked with the runtime system.
  # The math library "-lm" is linked by default.
***************
*** 26,32 ****
  # not all Unix C preprocessors define it.
  # If your cpp is too fussy, make tools/clprepro and use this:
  # CPP=../../src/tools/clprepro -Dunix
! CPP=/lib/cpp -P -traditional -Dunix
  
  # The directory where public executables will be installed
  BINDIR=/usr/local/bin
--- 26,32 ----
  # not all Unix C preprocessors define it.
  # If your cpp is too fussy, make tools/clprepro and use this:
  # CPP=../../src/tools/clprepro -Dunix
! CPP=/usr/bin/cpp -P -traditional -Dunix
  
  # The directory where public executables will be installed
  BINDIR=/usr/local/bin
Index: src/runtime/io.c
===================================================================
RCS file: /net/pauillac/caml/repository/camllight/sources/src/runtime/io.c,v
retrieving revision 1.11
diff -c -r1.11 io.c
*** src/runtime/io.c	1997/04/24 14:51:06	1.11
--- src/runtime/io.c	2002/10/02 13:45:16
***************
*** 1,5 ****
--- 1,9 ----
  /* Buffered input/output. */
  
+ #include "../../config/s.h"
+ #ifdef HAS_UNISTD
+ #include <unistd.h>
+ #endif
  #include <errno.h>
  #ifdef __MWERKS__
  #include "myfcntl.h"
Index: src/runtime/main.c
===================================================================
RCS file: /net/pauillac/caml/repository/camllight/sources/src/runtime/main.c,v
retrieving revision 1.22
diff -c -r1.22 main.c
*** src/runtime/main.c	1997/04/24 14:51:06	1.22
--- src/runtime/main.c	2002/10/02 13:45:16
***************
*** 1,5 ****
--- 1,9 ----
  /* Start-up code */
  
+ #include "../../config/s.h"
+ #ifdef HAS_UNISTD
+ #include <unistd.h>
+ #endif
  #include <stdio.h>
  #ifdef __MWERKS__
  #include "myfcntl.h"
