diff -uNr mono-1.1.8.3/data/config.in mono-1.1.8.3-new/data/config.in
--- mono-1.1.8.3/data/config.in	Mon Jun 13 10:00:53 2005
+++ mono-1.1.8.3-new/data/config.in	Tue Aug 16 16:09:07 2005
@@ -9,6 +9,10 @@
 	<dllmap dll="db2cli" target="libdb2_36@libsuffix@"/>
 	<dllmap dll="msvcrt" target="@LIBC@"/>
 	<dllmap dll="MonoPosixHelper" target="libMonoPosixHelper@libsuffix@"/>
+	<dllmap dll="libgtk-win32-2.0-0.dll" target="libgtk-x11-2.0@libsuffix@"/>
+	<dllmap dll="glib-2.0" target="libglib-2.0@libsuffix@"/>
+	<dllmap dll="gnomevfs-2" target="libgnomevfs-2@libsuffix@"/>
+	<dllmap dll="gtksourceview-1.0" target="libgtksourceview-1.0@libsuffix@"/>
 	<dllmap dll="sqlite" target="@SQLITE@"/>
 	<dllmap dll="sqlite3" target="@SQLITE3@"/>
 	<dllmap dll="libX11" target="@X11@"/>
diff -uNr mono-1.1.8.3/mono/metadata/Makefile.in mono-1.1.8.3-new/mono/metadata/Makefile.in
--- mono-1.1.8.3/mono/metadata/Makefile.in	Tue Jul 26 20:08:30 2005
+++ mono-1.1.8.3-new/mono/metadata/Makefile.in	Tue Aug 16 16:09:07 2005
@@ -68,7 +68,7 @@
 am_libmonoruntime_static_la_OBJECTS = $(am__objects_1)
 libmonoruntime_static_la_OBJECTS =  \
 	$(am_libmonoruntime_static_la_OBJECTS)
-libmonoruntime_la_LIBADD =
+libmonoruntime_la_LIBADD = $(ICU_LIBS)
 am_libmonoruntime_la_OBJECTS = reflection.lo object.lo icall.lo \
 	decimal.lo boehm-gc.lo null-gc.lo gc.lo marshal.lo monitor.lo \
 	threads.lo threadpool.lo file-io.lo socket-io.lo exception.lo \
diff -uNr mono-1.1.8.3/mono/metadata/loader.c mono-1.1.8.3-new/mono/metadata/loader.c
--- mono-1.1.8.3/mono/metadata/loader.c	Mon Jun 13 10:11:20 2005
+++ mono-1.1.8.3-new/mono/metadata/loader.c	Tue Aug 16 16:10:16 2005
@@ -736,6 +736,19 @@
 		}
 
 		if (!gmodule) {
+			full_name = g_module_build_path ("@PREFIX@/lib", file_name);
+			mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_DLLIMPORT,
+					"DllImport loading location: '%s'.", full_name);
+			gmodule = g_module_open (full_name, G_MODULE_BIND_LAZY);
+			if (!gmodule) {
+				mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_DLLIMPORT,
+						"DllImport error loading library: '%s'.",
+						g_module_error ());
+			}
+			g_free (full_name);
+		}
+
+		if (!gmodule) {
 			full_name = g_module_build_path (".", file_name);
 			mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_DLLIMPORT,
 					"DllImport loading library: '%s'.", full_name);
diff -uNr mono-1.1.8.3/mono/metadata/mono-debug-debugger.c mono-1.1.8.3-new/mono/metadata/mono-debug-debugger.c
--- mono-1.1.8.3/mono/metadata/mono-debug-debugger.c	Thu May 12 11:36:50 2005
+++ mono-1.1.8.3-new/mono/metadata/mono-debug-debugger.c	Tue Aug 16 16:09:07 2005
@@ -65,9 +65,12 @@
 
 #endif
 
+static int initialized = 0;
+
 void
 mono_debugger_lock (void)
 {
+	g_assert (initialized);
 	EnterCriticalSection (&debugger_lock_mutex);
 	debugger_lock_level++;
 }
@@ -75,6 +78,7 @@
 void
 mono_debugger_unlock (void)
 {
+	g_assert (initialized);
 	if (debugger_lock_level == 1) {
 		if (must_reload_symtabs && mono_debugger_use_debugger) {
 			mono_debugger_event (MONO_DEBUGGER_EVENT_RELOAD_SYMTABS, 0, 0);
@@ -95,6 +99,7 @@
 
 	InitializeCriticalSection (&debugger_lock_mutex);
 	mono_debugger_use_debugger = use_debugger;
+	initialized = 1;
 }
 
 void
diff -uNr mono-1.1.8.3/mono/metadata/mono-debug.c mono-1.1.8.3-new/mono/metadata/mono-debug.c
--- mono-1.1.8.3/mono/metadata/mono-debug.c	Tue Jul 26 14:26:52 2005
+++ mono-1.1.8.3-new/mono/metadata/mono-debug.c	Tue Aug 16 16:09:07 2005
@@ -771,6 +771,9 @@
 	char *res = NULL;
 	gint32 offset;
 
+	if (mono_debug_format == MONO_DEBUG_FORMAT_NONE)
+		return NULL;
+
 	mono_debugger_lock ();
 	minfo = _mono_debug_lookup_method (method);
 	if (!minfo || !minfo->handle || !minfo->handle->symfile || !minfo->handle->symfile->offset_table) {
@@ -808,6 +811,9 @@
 	char *res;
 	MonoDebugMethodInfo *minfo;
 
+	if (mono_debug_format == MONO_DEBUG_FORMAT_NONE)
+		return NULL;
+
 	mono_debugger_lock ();
 	minfo = _mono_debug_lookup_method (method);
 	if (!minfo || !minfo->handle || !minfo->handle->symfile) {
@@ -835,7 +841,7 @@
 	MonoDebugMethodInfo *minfo;
 	gint32 res;
 
-	if (address < 0)
+	if ((address < 0) || (mono_debug_format == MONO_DEBUG_FORMAT_NONE))
 		return -1;
 
 	mono_debugger_lock ();
@@ -867,7 +873,7 @@
 	MonoDebugMethodJitInfo *jit;
 	gint32 res;
 
-	if (il_offset < 0)
+	if ((il_offset < 0) || (mono_debug_format == MONO_DEBUG_FORMAT_NONE))
 		return -1;
 
 	mono_debugger_lock ();
diff -uNr mono-1.1.8.3/mono/metadata/process.c mono-1.1.8.3-new/mono/metadata/process.c
--- mono-1.1.8.3/mono/metadata/process.c	Sun Mar 13 21:18:24 2005
+++ mono-1.1.8.3-new/mono/metadata/process.c	Tue Aug 16 16:09:07 2005
@@ -854,7 +854,8 @@
 		process_info->process_handle=procinfo.hProcess;
 		/*process_info->thread_handle=procinfo.hThread;*/
 		process_info->thread_handle=NULL;
-		CloseHandle(procinfo.hThread);
+		if (procinfo.hThread != NULL)
+			CloseHandle(procinfo.hThread);
 		process_info->pid=procinfo.dwProcessId;
 		process_info->tid=procinfo.dwThreadId;
 	} else {
