--- Pyfort-8.2/fortran_compiler.py.orig	Tue Sep 10 06:58:35 2002
+++ Pyfort-8.2/fortran_compiler.py	Tue Sep 10 06:59:15 2002
@@ -347,14 +347,14 @@
         self.typedict = typedict
     
     def executable_name (self):
-        return "fort77"
+        return "g77 -O3 -funroll-all-loops"
 
     def header(self):
         "Return a string containing anything needed to add to the header."
         text = \
 """
 /* 
-    Built by PyFort for the fort77 (GNU Darwin) compiler.
+    Built by PyFort for the g77 compiler.
 */
 """
         return text 
