diff -Naur sbcl-0.9.2.orig/install.sh sbcl-0.9.2/install.sh
--- sbcl-0.9.2.orig/install.sh	2005-06-21 08:26:45.000000000 -0700
+++ sbcl-0.9.2/install.sh	2005-07-26 17:03:03.000000000 -0700
@@ -99,18 +99,9 @@
 do
   cp $info $BUILD_ROOT$INFO_DIR/ \
       && echo -n " info $BUILD_ROOT$INFO_DIR/`basename $info` $BUILD_ROOT$INFO_DIR/dir" \
-      && ( install-info $BUILD_ROOT$INFO_DIR/`basename $info` $BUILD_ROOT$INFO_DIR/dir > /dev/null 2>&1 \
-           || echo -n " (could not add to system catalog)" ) \
       && echo
 done
 
-# pdf
-for pdf in doc/manual/*.pdf
-do
-  cp $pdf $BUILD_ROOT$DOC_DIR/ \
-      && echo " pdf $BUILD_ROOT$DOC_DIR/`basename $pdf`"
-done
-
 # html
 for html in doc/manual/sbcl doc/manual/asdf
 do
