--- wxPython/config.py.orig	Thu Apr 29 06:01:37 2004
+++ wxPython/config.py	Thu Apr 29 06:05:14 2004
@@ -194,8 +194,8 @@
 
 
 # change the PORT default for wxMac
-if sys.platform[:6] == "darwin":
-    WXPORT = 'mac'
+#if sys.platform[:6] == "darwin":
+#    WXPORT = 'mac'
 
 # and do the same for wxMSW, just for consistency
 if os.name == 'nt':
@@ -408,8 +408,8 @@
             return
 
         root = self.root
-        if WXPREFIX.startswith(root):
-            root = ''
+#       if WXPREFIX.startswith(root):
+        root = ''
         for header, location in headers:
             install_dir = os.path.normpath(root + WXPREFIX + location)
             self.mkpath(install_dir)
@@ -650,7 +650,7 @@
     WXPREFIX   = os.popen(WX_CONFIG + ' --prefix').read()[:-1]
 
 
-    if sys.platform[:6] == "darwin":
+    if sys.platform[:6] == "darwinx":
         # Flags and such for a Darwin (Max OS X) build of Python
         WXPLAT = '__WXMAC__'
         GENDIR = 'mac'
