--- autogen-5.17.4/configure.orig	2013-05-21 15:39:04.000000000 -0400
+++ autogen-5.17.4/configure	2013-05-21 15:42:18.000000000 -0400
@@ -11564,7 +11564,7 @@
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-ldl  $LIBS"
+LIBS="$LIBS -ldl"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -11660,7 +11660,7 @@
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-ldl  $LIBS"
+LIBS="$LIBS -ldl"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -14700,7 +14700,7 @@
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lintl  $LIBS"
+LIBS="$LIBS -lintl"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -14735,7 +14735,7 @@
 #define HAVE_LIBINTL 1
 _ACEOF
 
-  LIBS="-lintl $LIBS"
+  LIBS="$LIBS -lintl"
 
 fi
 
@@ -17063,7 +17063,7 @@
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lm  $LIBS"
+LIBS="$LIBS -lm "
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -17098,7 +17098,7 @@
 #define HAVE_LIBM 1
 _ACEOF
 
-  LIBS="-lm $LIBS"
+  LIBS="$LIBS -lm"
 
 fi
 
