diff -wrauN -x .svn -x .gitignore nano-2.3.2/ChangeLog nano/ChangeLog
--- nano-2.3.2/ChangeLog	2013-03-22 15:16:07.000000000 +0100
+++ nano/ChangeLog	2013-11-29 16:24:18.000000000 +0100
@@ -1,3 +1,25 @@
+2013-06-13 Matthew Fischer <mfisch@Savannah>
+	* doc/syntax/c.nanorc - Add auto keyword
+
+2013-06-13  David Lawrence Ramsey <pooka109@gmail.com>
+	* src/global.c (first_sc_for) - try and more consistently display keystrokes,
+	  useful when the user has re-binded a bunch of them.
+
+2013-06-13 Kamil Dudka <kdudka@redhat.com>
+	* doc/man/nano.1 - Actually document the -P (--poslog) option.
+
+2013-04-12 Chris Allegretta <chrisa@asty.org>
+	* src/files.c (do_insertfile): Check for saved cursor position when inserting a
+	  file as well.  Fixes Savannah bug 38600 reported by Craig Barnes.
+	* src/files.c (write_file): Don't re-stat() the file if we're writing out
+	  a marked file (especially that would give it the wrong stat info).  Part
+	  2 of issue reported by Benno Schulenberg.
+
+2013-04-07 Michael Berg <mike@berg-net.us>
+	* do_cut_text - Fix copying (not cutting) text setting Modified state.  Fixes
+	  issue reported by Benno Schulenberg.
+
+GNU nano 2.3.2 - 2011.03.23
 2013-03-17 Chris Allegretta <chrisa@asty.org>
         * Revert r4547 as we should have a new release and the overlap code is not yet
           ready for public consumption.
diff -wrauN -x .svn -x .gitignore nano-2.3.2/README nano/README
--- nano-2.3.2/README	2013-01-01 03:53:40.000000000 +0100
+++ nano/README	2013-11-29 16:24:18.000000000 +0100
@@ -65,8 +65,8 @@
 
 Current Status
 
-	GNU nano has reached its third stable milestone, 2.0.x.
-	Development of new features continues in the 2.1.x branch, while
-	2.0.x versions are dedicated to bug-fixing and polishing.
+	GNU nano has reached its fourth stable milestone, 2.2.x.
+	Development of new features continues in the 2.3.x branch, while
+	2.2.x versions are dedicated to bug-fixing and polishing.
 
-$Id: README 4057 2007-01-29 12:40:48Z dolorous $
+$Id: README 4576 2013-06-14 02:17:57Z astyanax $
diff -wrauN -x .svn -x .gitignore nano-2.3.2/autogen.sh nano/autogen.sh
--- nano-2.3.2/autogen.sh	1970-01-01 01:00:00.000000000 +0100
+++ nano/autogen.sh	2013-11-29 16:24:20.000000000 +0100
@@ -0,0 +1,4 @@
+#!/bin/sh
+# Generate configure & friends for CVS users.
+
+autoreconf -f -i -s
diff -wrauN -x .svn -x .gitignore nano-2.3.2/configure.ac nano/configure.ac
--- nano-2.3.2/configure.ac	2013-03-22 15:16:07.000000000 +0100
+++ nano/configure.ac	2013-11-29 16:24:18.000000000 +0100
@@ -18,9 +18,9 @@
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301,
 # USA.
 #
-# $Id: configure.ac 4570 2013-03-22 14:06:58Z astyanax $
+# $Id: configure.ac 4572 2013-03-24 12:38:51Z astyanax $
 
-AC_INIT([GNU nano], [2.3.2], [nano-devel@gnu.org], [nano])
+AC_INIT([GNU nano], [2.3.2-svn], [nano-devel@gnu.org], [nano])
 AC_CONFIG_SRCDIR([src/nano.c])
 AC_CANONICAL_HOST
 AM_INIT_AUTOMAKE
diff -wrauN -x .svn -x .gitignore nano-2.3.2/doc/man/nano.1 nano/doc/man/nano.1
--- nano-2.3.2/doc/man/nano.1	2013-03-21 14:17:24.000000000 +0100
+++ nano/doc/man/nano.1	2013-11-29 16:24:19.000000000 +0100
@@ -18,7 +18,7 @@
 .\"   Documentation License along with this program.  If not, see
 .\"   <http://www.gnu.org/licenses/>.
 .\"
-.\" $Id: nano.1 4549 2013-01-01 03:24:39Z astyanax $
+.\" $Id: nano.1 4578 2013-06-14 02:31:38Z astyanax $
 .TH NANO 1 "version 2.2.0" "November 30, 2009"
 .\" Please adjust this date whenever revising the manpage.
 .\"
@@ -102,6 +102,9 @@
 .B \-O (\-\-morespace)
 Use the blank line below the titlebar as extra editing space.
 .TP
+.B \-P (\-\-poslog)
+Log & read location of cursor position.
+.TP
 .B \-Q \fIstr\fP (\-\-quotestr=\fIstr\fP)
 Set the quoting string for justifying.  The default is
 "\fI^([\ \\t]*[#:>\\|}])+\fP" if extended regular expression support is
diff -wrauN -x .svn -x .gitignore nano-2.3.2/doc/syntax/c.nanorc nano/doc/syntax/c.nanorc
--- nano-2.3.2/doc/syntax/c.nanorc	2013-01-01 03:53:33.000000000 +0100
+++ nano/doc/syntax/c.nanorc	2013-11-29 16:24:18.000000000 +0100
@@ -3,7 +3,7 @@
 syntax "c" "\.(c(c|pp|xx)?|C)$" "\.(h(h|pp|xx)?|H)$" "\.ii?$"
 magic "ASCII C(\+\+)? program text"
 color brightred "\<[A-Z_][0-9A-Z_]+\>" 
-color green "\<(float|double|bool|char|int|short|long|sizeof|enum|void|static|const|struct|union|typedef|extern|(un)?signed|inline)\>"
+color green "\<(float|double|bool|char|int|short|long|sizeof|enum|void|auto|static|const|struct|union|typedef|extern|(un)?signed|inline)\>"
 color green "\<((s?size)|((u_?)?int(8|16|32|64|ptr)))_t\>"
 color green "\<(class|namespace|template|public|protected|private|typename|this|friend|virtual|using|mutable|volatile|register|explicit)\>"
 color brightyellow "\<(for|if|while|do|else|case|default|switch)\>"
diff -wrauN -x .svn -x .gitignore nano-2.3.2/doc/texinfo/nano.html nano/doc/texinfo/nano.html
--- nano-2.3.2/doc/texinfo/nano.html	2013-01-22 08:56:27.000000000 +0100
+++ nano/doc/texinfo/nano.html	2013-11-29 17:04:08.000000000 +0100
@@ -3,7 +3,7 @@
 <title>nano Command Manual</title>
 <meta http-equiv="Content-Type" content="text/html">
 <meta name="description" content="nano Command Manual">
-<meta name="generator" content="makeinfo 4.13">
+<meta name="generator" content="makeinfo 4.8">
 <link title="Top" rel="top" href="#Top">
 <link href="http://www.gnu.org/software/texinfo/" rel="generator-home" title="Texinfo Homepage">
 <meta http-equiv="Content-Style-Type" content="text/css">
@@ -22,8 +22,8 @@
 <body>
 <h1 class="settitle">nano Command Manual</h1>
 <div class="node">
-<a name="Top"></a>
 <p><hr>
+<a name="Top"></a>
 Next:&nbsp;<a rel="next" accesskey="n" href="#Introduction">Introduction</a>,
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#dir">(dir)</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#dir">(dir)</a>
@@ -45,8 +45,8 @@
 </ul>
 
 <div class="node">
-<a name="Introduction"></a>
 <p><hr>
+<a name="Introduction"></a>
 Next:&nbsp;<a rel="next" accesskey="n" href="#Editor-Basics">Editor Basics</a>,
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#Top">Top</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>
@@ -67,8 +67,8 @@
 </ul>
 
 <div class="node">
-<a name="Overview"></a>
 <p><hr>
+<a name="Overview"></a>
 Next:&nbsp;<a rel="next" accesskey="n" href="#Command-Line-Options">Command Line Options</a>,
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#Introduction">Introduction</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Introduction">Introduction</a>
@@ -104,8 +104,8 @@
    <p>Email bug reports to <a href="mailto:nano@nano-editor.org">nano@nano-editor.org</a>.
 
 <div class="node">
-<a name="Command-Line-Options"></a>
 <p><hr>
+<a name="Command-Line-Options"></a>
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#Overview">Overview</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Introduction">Introduction</a>
 
@@ -242,7 +242,8 @@
      <br><dt><code>-w, --nowrap</code><dd>Don't wrap long lines at any length.  This option overrides any value
 for -r.
 
-     <p><a name="Expert-Mode"></a><br><dt><code>-x, --nohelp</code><dd>Expert Mode: don't show the Shortcut Lists at the bottom of the screen. 
+     <p><a name="Expert-Mode"></a>
+<br><dt><code>-x, --nohelp</code><dd>Expert Mode: don't show the Shortcut Lists at the bottom of the screen. 
 This affects the location of the statusbar as well, as in Expert Mode it
 is located at the very bottom of the editor.
 
@@ -262,8 +263,8 @@
 </dl>
 
 <div class="node">
-<a name="Editor-Basics"></a>
 <p><hr>
+<a name="Editor-Basics"></a>
 Next:&nbsp;<a rel="next" accesskey="n" href="#Online-Help">Online Help</a>,
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#Introduction">Introduction</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>
@@ -282,8 +283,8 @@
 </ul>
 
 <div class="node">
-<a name="Entering-Text"></a>
 <p><hr>
+<a name="Entering-Text"></a>
 Next:&nbsp;<a rel="next" accesskey="n" href="#Special-Functions">Special Functions</a>,
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#Editor-Basics">Editor Basics</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Editor-Basics">Editor Basics</a>
@@ -298,8 +299,8 @@
 edited.
 
 <div class="node">
-<a name="Special-Functions"></a>
 <p><hr>
+<a name="Special-Functions"></a>
 Next:&nbsp;<a rel="next" accesskey="n" href="#The-Titlebar">The Titlebar</a>,
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#Entering-Text">Entering Text</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Editor-Basics">Editor Basics</a>
@@ -327,8 +328,8 @@
 </ul>
 
 <div class="node">
-<a name="The-Titlebar"></a>
 <p><hr>
+<a name="The-Titlebar"></a>
 Next:&nbsp;<a rel="next" accesskey="n" href="#The-Statusbar">The Statusbar</a>,
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#Special-Functions">Special Functions</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Editor-Basics">Editor Basics</a>
@@ -349,8 +350,8 @@
 See <a href="#The-File-Browser">The File Browser</a>, for more info.
 
 <div class="node">
-<a name="The-Statusbar"></a>
 <p><hr>
+<a name="The-Statusbar"></a>
 Next:&nbsp;<a rel="next" accesskey="n" href="#Shortcut-Lists">Shortcut Lists</a>,
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#The-Titlebar">The Titlebar</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Editor-Basics">Editor Basics</a>
@@ -367,8 +368,8 @@
 input (search strings, filenames, etc.) will be input on the statusbar.
 
 <div class="node">
-<a name="Shortcut-Lists"></a>
 <p><hr>
+<a name="Shortcut-Lists"></a>
 Next:&nbsp;<a rel="next" accesskey="n" href="#Using-the-Mouse">Using the Mouse</a>,
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#The-Statusbar">The Statusbar</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Editor-Basics">Editor Basics</a>
@@ -381,8 +382,8 @@
 show some of the more commonly used functions in the editor.
 
 <div class="node">
-<a name="Using-the-Mouse"></a>
 <p><hr>
+<a name="Using-the-Mouse"></a>
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#Shortcut-Lists">Shortcut Lists</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Editor-Basics">Editor Basics</a>
 
@@ -399,8 +400,8 @@
 is running.
 
 <div class="node">
-<a name="Online-Help"></a>
 <p><hr>
+<a name="Online-Help"></a>
 Next:&nbsp;<a rel="next" accesskey="n" href="#Feature-Toggles">Feature Toggles</a>,
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#Editor-Basics">Editor Basics</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>
@@ -415,8 +416,8 @@
 and ^V (Page Down) keys.  ^X exits the help system.
 
 <div class="node">
-<a name="Feature-Toggles"></a>
 <p><hr>
+<a name="Feature-Toggles"></a>
 Next:&nbsp;<a rel="next" accesskey="n" href="#Nanorc-Files">Nanorc Files</a>,
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#Online-Help">Online Help</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>
@@ -470,8 +471,8 @@
    </dl>
 
 <div class="node">
-<a name="Nanorc-Files"></a>
 <p><hr>
+<a name="Nanorc-Files"></a>
 Next:&nbsp;<a rel="next" accesskey="n" href="#The-File-Browser">The File Browser</a>,
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#Feature-Toggles">Feature Toggles</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>
@@ -675,8 +676,8 @@
    </dl>
 
 <div class="node">
-<a name="The-File-Browser"></a>
 <p><hr>
+<a name="The-File-Browser"></a>
 Next:&nbsp;<a rel="next" accesskey="n" href="#Pico-Compatibility">Pico Compatibility</a>,
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#Nanorc-Files">Nanorc Files</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>
@@ -699,8 +700,8 @@
 browser exits.
 
 <div class="node">
-<a name="Pico-Compatibility"></a>
 <p><hr>
+<a name="Pico-Compatibility"></a>
 Next:&nbsp;<a rel="next" accesskey="n" href="#Building-and-Configure-Options">Building and Configure Options</a>,
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#The-File-Browser">The File Browser</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>
@@ -752,8 +753,8 @@
 </dl>
 
 <div class="node">
-<a name="Building-and-Configure-Options"></a>
 <p><hr>
+<a name="Building-and-Configure-Options"></a>
 Previous:&nbsp;<a rel="previous" accesskey="p" href="#Pico-Compatibility">Pico Compatibility</a>,
 Up:&nbsp;<a rel="up" accesskey="u" href="#Top">Top</a>
 
diff -wrauN -x .svn -x .gitignore nano-2.3.2/nano-regress nano/nano-regress
--- nano-2.3.2/nano-regress	1970-01-01 01:00:00.000000000 +0100
+++ nano/nano-regress	2013-11-29 16:24:18.000000000 +0100
@@ -0,0 +1,29 @@
+#!/usr/bin/perl
+use strict;
+
+sub combinations {
+  return [] unless @_;
+  my $first = shift;
+  my @rest = combinations(@_);
+  return @rest, map { [$first, @$_] } @rest;
+} 
+
+my @allargs=("--enable-debug", "--disable-wrapping", "--disable-justify", "--disable-extra", "--enable-tiny", "--disable-browser --disable-help --disable-mouse --disable-operatingdir --disable-speller", "--disable-multibuffer", "--disable-nanorc", "--with-slang");
+my @combos =  combinations(@allargs);
+
+my $i = 0;
+foreach my $name (@combos) {
+    my @args = @$name;
+    my $pct = $i / $#combos * 100;
+    printf "Trying with options: @args, %d%% done...\n", $pct;
+    my $cmd = "./configure @args && make clean all";
+    system("($cmd) >/dev/null 2>&1");
+    if ($? != 0)  {
+	print "Build failed for args: @args\n";
+	print "To reproduce, run:\n $cmd\n";
+	exit(1);
+    }
+    $i++;
+}
+
+print "All options completed successfully!\n";
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/bg.po nano/po/bg.po
--- nano-2.3.2/po/bg.po	2013-03-22 15:18:14.000000000 +0100
+++ nano/po/bg.po	2013-11-29 16:24:20.000000000 +0100
@@ -118,7 +118,7 @@
 #: src/files.c:271
 #, fuzzy, c-format
 msgid "Error opening lockfile %s: %s"
-msgstr "–ì—Ä–µ—à–∫–∞ –ø—Ä–∏ —á–µ—Ç–µ–Ω–µ –Ω–∞ %s: %s"
+msgstr "–ì—Ä–µ—à–∫–∞ –ø—Ä–∏ –∑–∞–ø–∏—Å–∞ –Ω–∞ –≤—Ä–µ–º–µ–Ω–µ–Ω —Ñ–∞–π–ª: %s"
 
 #: src/files.c:281
 #, c-format
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/es.po nano/po/es.po
--- nano-2.3.2/po/es.po	2013-03-22 15:18:14.000000000 +0100
+++ nano/po/es.po	2013-11-29 16:24:20.000000000 +0100
@@ -1,7 +1,7 @@
 # translation of nano-1.9.99pre0.es.po to Spanish
 # Mensajes en espaÒol para GNU nano.
 # Copyright (C) 2000, 2001, 2002, 2003, 2004, 2005, 2006 Free Software Foundation, Inc.
-# $Id: es.po,v 1.127 2002/10/26 19:56:43 jordi Exp $
+# $Id: es.po 4570 2013-03-22 14:06:58Z astyanax $
 #
 # Jordi Mallach <jordi@sindominio.net>, 2000, 2001, 2002.
 # Ricardo C·rdenes <ricardo@conysis.com>, 2002, 2003, 2004, 2005.
@@ -119,7 +119,7 @@
 #: src/files.c:271
 #, fuzzy, c-format
 msgid "Error opening lockfile %s: %s"
-msgstr "Error leyendo '%s': %s"
+msgstr "Error escribiendo en fichero temporal: %s"
 
 #: src/files.c:281
 #, c-format
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/eu.po nano/po/eu.po
--- nano-2.3.2/po/eu.po	2013-03-22 15:18:15.000000000 +0100
+++ nano/po/eu.po	2013-11-29 16:24:20.000000000 +0100
@@ -115,7 +115,7 @@
 #: src/files.c:271
 #, fuzzy, c-format
 msgid "Error opening lockfile %s: %s"
-msgstr "Errorea %s irakurtzen: %s"
+msgstr "Errorea aldi baterako fitxategia idazten: %s"
 
 #: src/files.c:281
 #, c-format
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/id.po nano/po/id.po
--- nano-2.3.2/po/id.po	2013-03-22 15:18:15.000000000 +0100
+++ nano/po/id.po	2013-11-29 16:24:20.000000000 +0100
@@ -104,17 +104,17 @@
 #: src/files.c:169 src/files.c:207 src/files.c:217
 #, fuzzy, c-format
 msgid "Error writing lock file %s: %s"
-msgstr "Kesalahan dalam %s pada baris %d: "
+msgstr "Quote string buruk %s: %s"
 
 #: src/files.c:234
 #, fuzzy, c-format
 msgid "Error deleting lock file %s: %s"
-msgstr "Kesalahan dalam %s pada baris %d: "
+msgstr "Quote string buruk %s: %s"
 
 #: src/files.c:271
 #, fuzzy, c-format
 msgid "Error opening lockfile %s: %s"
-msgstr "Kesalahan dalam %s pada baris %d: "
+msgstr "Quote string buruk %s: %s"
 
 #: src/files.c:281
 #, c-format
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/it.po nano/po/it.po
--- nano-2.3.2/po/it.po	2013-03-22 15:18:15.000000000 +0100
+++ nano/po/it.po	2013-11-29 16:24:20.000000000 +0100
@@ -2,7 +2,7 @@
 # Copyright (C) 2001-2006, 2008, 2009, 2010, 2011 Free Software Foundation, Inc.
 # This file is distributed under the same license as the nano package.
 # Marco Colombo <m.colombo@ed.ac.uk>, 2001-2011.
-# $Id: it.po,v 1.116 2006/06/09 09:37:06 jordi Exp $
+# $Id: it.po 4570 2013-03-22 14:06:58Z astyanax $
 #
 msgid ""
 msgstr ""
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/nb.po nano/po/nb.po
--- nano-2.3.2/po/nb.po	2013-03-22 15:18:15.000000000 +0100
+++ nano/po/nb.po	2013-11-29 16:24:20.000000000 +0100
@@ -3,7 +3,7 @@
 # Eivind Kj¯rstad <ekj@vestdata.no>, 2001.
 # Stig E Sand¯ <stig@ii.uib.no>, 2002.
 # Geir Helland <pjallabais@users.sourceforge.net>, 2004
-# $Id: nb.po,v 1.3 2002/01/23 01:06:20 astyanax Exp $
+# $Id: nb.po 4570 2013-03-22 14:06:58Z astyanax $
 #
 msgid ""
 msgstr ""
@@ -115,7 +115,7 @@
 #: src/files.c:271
 #, fuzzy, c-format
 msgid "Error opening lockfile %s: %s"
-msgstr "Lesefeil pÂ %s: %s"
+msgstr "Feil ved skriving til midlertidlig fil: %s"
 
 #: src/files.c:281
 #, c-format
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/nn.po nano/po/nn.po
--- nano-2.3.2/po/nn.po	2013-03-22 15:18:15.000000000 +0100
+++ nano/po/nn.po	2013-11-29 16:24:20.000000000 +0100
@@ -112,9 +112,9 @@
 msgstr "Feil i %s pÂ linje %d: "
 
 #: src/files.c:271
-#, fuzzy, c-format
+#, c-format
 msgid "Error opening lockfile %s: %s"
-msgstr "Feil i %s pÂ linje %d: "
+msgstr ""
 
 #: src/files.c:281
 #, c-format
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/ro.po nano/po/ro.po
--- nano-2.3.2/po/ro.po	2013-03-22 15:18:15.000000000 +0100
+++ nano/po/ro.po	2013-11-29 16:24:20.000000000 +0100
@@ -114,7 +114,7 @@
 #: src/files.c:271
 #, fuzzy, c-format
 msgid "Error opening lockfile %s: %s"
-msgstr "Eroare citind %s: %s"
+msgstr "Eroare la scrierea fi∫ierului temporar: %s"
 
 #: src/files.c:281
 #, c-format
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/sr.po nano/po/sr.po
--- nano-2.3.2/po/sr.po	2013-03-22 15:18:15.000000000 +0100
+++ nano/po/sr.po	2013-11-29 16:24:20.000000000 +0100
@@ -104,17 +104,17 @@
 #: src/files.c:169 src/files.c:207 src/files.c:217
 #, fuzzy, c-format
 msgid "Error writing lock file %s: %s"
-msgstr "–ì—Ä–µ—à–∫–∞ —É %s —É %d. —Ä–µ–¥—É: "
+msgstr "–õ–æ—à–∞ –Ω–∏—Å–∫–∞ –∑–∞ —Ü–∏—Ç–∏—Ä–∞—ö–µ %s: %s"
 
 #: src/files.c:234
 #, fuzzy, c-format
 msgid "Error deleting lock file %s: %s"
-msgstr "–ì—Ä–µ—à–∫–∞ —É %s —É %d. —Ä–µ–¥—É: "
+msgstr "–õ–æ—à–∞ –Ω–∏—Å–∫–∞ –∑–∞ —Ü–∏—Ç–∏—Ä–∞—ö–µ %s: %s"
 
 #: src/files.c:271
 #, fuzzy, c-format
 msgid "Error opening lockfile %s: %s"
-msgstr "–ì—Ä–µ—à–∫–∞ —É %s —É %d. —Ä–µ–¥—É: "
+msgstr "–õ–æ—à–∞ –Ω–∏—Å–∫–∞ –∑–∞ —Ü–∏—Ç–∏—Ä–∞—ö–µ %s: %s"
 
 #: src/files.c:281
 #, c-format
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/sv.po nano/po/sv.po
--- nano-2.3.2/po/sv.po	2013-03-22 15:18:15.000000000 +0100
+++ nano/po/sv.po	2013-11-29 16:24:20.000000000 +0100
@@ -113,7 +113,7 @@
 #: src/files.c:271
 #, fuzzy, c-format
 msgid "Error opening lockfile %s: %s"
-msgstr "Fel vid inl‰sning av %s: %s"
+msgstr "Fel vid skrivning av tempor‰r fil: %s"
 
 #: src/files.c:281
 #, c-format
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/tr.po nano/po/tr.po
--- nano-2.3.2/po/tr.po	2013-03-22 15:18:15.000000000 +0100
+++ nano/po/tr.po	2013-11-29 16:24:20.000000000 +0100
@@ -105,17 +105,17 @@
 #: src/files.c:169 src/files.c:207 src/files.c:217
 #, fuzzy, c-format
 msgid "Error writing lock file %s: %s"
-msgstr "%s'nin %d. satƒ±rƒ±nda hata: "
+msgstr "Hatalƒ± alƒ±ntƒ± dizgesi %s: %s"
 
 #: src/files.c:234
 #, fuzzy, c-format
 msgid "Error deleting lock file %s: %s"
-msgstr "%s'nin %d. satƒ±rƒ±nda hata: "
+msgstr "Hatalƒ± alƒ±ntƒ± dizgesi %s: %s"
 
 #: src/files.c:271
 #, fuzzy, c-format
 msgid "Error opening lockfile %s: %s"
-msgstr "%s'nin %d. satƒ±rƒ±nda hata: "
+msgstr "Hatalƒ± alƒ±ntƒ± dizgesi %s: %s"
 
 #: src/files.c:281
 #, c-format
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/uk.po nano/po/uk.po
--- nano-2.3.2/po/uk.po	2013-03-22 15:18:16.000000000 +0100
+++ nano/po/uk.po	2013-11-29 16:24:20.000000000 +0100
@@ -105,17 +105,17 @@
 #: src/files.c:169 src/files.c:207 src/files.c:217
 #, fuzzy, c-format
 msgid "Error writing lock file %s: %s"
-msgstr "œÕ…ÃÀ¡ ◊ %s ’ “—ƒÀ’ %d: "
+msgstr "œ«¡Œ…  “—ƒœÀ √…‘’◊¡ŒŒ— %s: %s"
 
 #: src/files.c:234
 #, fuzzy, c-format
 msgid "Error deleting lock file %s: %s"
-msgstr "œÕ…ÃÀ¡ ◊ %s ’ “—ƒÀ’ %d: "
+msgstr "œ«¡Œ…  “—ƒœÀ √…‘’◊¡ŒŒ— %s: %s"
 
 #: src/files.c:271
 #, fuzzy, c-format
 msgid "Error opening lockfile %s: %s"
-msgstr "œÕ…ÃÀ¡ ◊ %s ’ “—ƒÀ’ %d: "
+msgstr "œ«¡Œ…  “—ƒœÀ √…‘’◊¡ŒŒ— %s: %s"
 
 #: src/files.c:281
 #, c-format
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/update_linguas.sh nano/po/update_linguas.sh
--- nano-2.3.2/po/update_linguas.sh	1970-01-01 01:00:00.000000000 +0100
+++ nano/po/update_linguas.sh	2013-11-29 16:24:20.000000000 +0100
@@ -0,0 +1,21 @@
+#!/bin/sh
+echo "Updating translations via TP" && \
+rsync -Lrtvz  translationproject.org::tp/latest/nano/ .
+#
+NEWSTUFF=`svn status | grep "^\? .*.po$" | awk '{ print $NF }'`
+if [ "x$NEWSTUFF" != "x" ]; then
+    echo "Adding new languaes to SVN"
+    svn add $NEWSTUFF
+fi
+#
+echo "Updating LINGUAS for all translations"
+FILE=LINGUAS
+echo "# List of available languages." >$FILE
+/bin/ls *.po | tr '\n' ' ' | sed 's/\.po//g' >>$FILE
+echo >> $FILE
+#
+if [ "x$NEWSTUFF" != "x" ]; then
+    echo -n "New langs found, re-running configure and make at top level (silently)..."
+    (cd .. && ./configure  && make) >/dev/null
+    echo "done"
+fi
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/update_sources.pl nano/po/update_sources.pl
--- nano-2.3.2/po/update_sources.pl	1970-01-01 01:00:00.000000000 +0100
+++ nano/po/update_sources.pl	2013-11-29 16:24:20.000000000 +0100
@@ -0,0 +1,368 @@
+#!/usr/bin/perl -w
+
+#
+#  GNOME PO Update Utility
+#
+#  Copyright (C) 2000, 2003 Free Software Foundation, Inc.
+#
+#  This script is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or (at
+#  your option) any later version.
+#
+#  This script is distributed in the hope that it will be useful, but
+#  WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this library; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301,
+#  USA.
+#
+#  Authors: Kenneth Christiansen <kenneth@gnu.org>
+#
+
+#  NOTICE: Please remember to change the variable $PACKAGE to reflect 
+#  the package this script is used within.
+
+$PACKAGE="nano";
+
+use File::Basename;
+
+# Declare global variables
+#-------------------------
+my $VERSION = "1.5beta5";
+my $LANG    = $ARGV[0];
+
+# Always print as the first thing
+#--------------------------------
+$| = 1;
+
+# Figure out what package that is in use
+#---------------------------------------
+#open FILE, "../configure.in";
+#    while (<FILE>) {
+#	next if /^dnl/; #ignore comments
+#        if ($_=~/AM_INIT_AUTOMAKE\((.*),(.*)\)/o){
+#            $PACKAGE=$1;
+#	    last; #stop when found
+#            }
+#	if ($_=~/PACKAGE\((.*)\)/o){
+#            $PACKAGE=$1;
+#            last; #stop when found
+#            }
+#        }   
+#close FILE;
+
+
+# Give error if script is run without an argument
+#------------------------------------------------
+if (! $LANG){
+    print "update.pl:  missing file arguments\n";
+    print "Try `update.pl --help' for more information.\n";
+    exit;
+}
+
+# Use the supplied arguments
+#---------------------------
+if ($LANG=~/^-(.)*/){
+
+    if ("$LANG" eq "--version"   || "$LANG" eq "-V"){
+        &Version;
+    }
+    elsif ($LANG eq "--help"     || "$LANG" eq "-H"){
+	&Help;
+    }
+    elsif ($LANG eq "--dist"     || "$LANG" eq "-D"){
+        &Merging;
+    }
+    elsif ($LANG eq "--pot"      || "$LANG" eq "-P"){
+
+ 	# Check for .headerlock file, so the Makefile
+        # will not generate the header files twise 
+	#--------------------------------------------
+	if (-e ".headerlock"){
+   	&GeneratePot;
+	}else{
+        &GenHeaders;
+	&GeneratePot;}
+        exit;
+    }
+    elsif ($LANG eq "--headers"  || "$LANG" eq "-S"){
+        &GenHeaders;
+        exit;
+    }
+    elsif ($LANG eq "--maintain" || "$LANG" eq "-M"){
+        &Maintain;
+    }
+    else {
+        &InvalidOption;
+    }
+
+} else {
+   
+   # Run standard procedure
+   #-----------------------
+   if(-s "$LANG.po"){
+        &GenHeaders; 
+	&GeneratePot;
+	&Merging;
+	&Status;
+   }  
+
+   # Report error if the language file supplied
+   # to the command line is non-existent
+   #-------------------------------------------
+   else {
+	&NotExisting;       
+   }
+}
+
+sub Version{
+
+    # Print version information
+    #--------------------------
+    print "GNOME PO Updater $VERSION\n";
+    print "Written by Kenneth Christiansen <kenneth\@gnome.org>, 2000.\n\n";
+    print "Copyright (C) 2000 Free Software Foundation, Inc.\n";
+    print "This is free software; see the source for copying conditions.  There is NO\n";
+    print "warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.\n";
+    exit;
+}
+
+sub Help{
+
+    # Print usage information
+    #------------------------
+    print "Usage: ./update.pl [OPTIONS] ...LANGCODE\n";
+    print "Updates pot files and merge them with the translations.\n\n";
+    print "  -H, --help                   shows this help page\n";
+    print "  -P, --pot                    generate the pot file only\n";
+    print "  -S, --headers                generate the XML headerfiles in POTFILES.in\n";
+    print "  -M, --maintain               search for missing files in POTFILES.in\n";
+    print "  -V, --version                shows the version\n";
+    print "\nExamples of use:\n";
+    print "update.sh --pot    just creates a new pot file from the source\n";
+    print "update.sh da       created new pot file and updated the da.po file\n\n";
+    print "Report bugs to <kenneth\@gnome.org>.\n";
+    exit;
+}
+
+sub Maintain{
+   
+    # Search and fine, all translatable files
+    # ---------------------------------------
+    $i18nfiles="find ../ -print | egrep '.*\\.(c|y|cc|c++|h|gob)' ";
+
+    open(BUF2, "POTFILES.in") || die "update.pl:  there's no POTFILES.in!!!\n";
+    
+    print "Searching for missing _(\" \") entries...\n";
+    
+    open(BUF1, "$i18nfiles|");
+
+    @buf1_1 = <BUF1>;
+    @buf1_2 = <BUF2>;
+
+    # Check if we should ignore some found files, when 
+    # comparing with POTFILES.in
+    #-------------------------------------------------
+    if (-s ".potignore"){
+        open FILE, ".potignore";
+        while (<FILE>) {
+            if ($_=~/^[^#]/o){
+                push @bup, $_;
+            }
+        }
+        print "Found .potignore: Ignoring files...\n";
+        @buf1_2 = (@bup, @buf1_2);
+    }
+
+    foreach my $file (@buf1_1){
+        open FILE, "<$file";
+        while (<FILE>) {
+            if ($_=~/_\(\"/o){
+                $file = unpack("x3 A*",$file) . "\n";
+                push @buf2_1, $file;
+                last;
+            }
+        }
+    }
+
+    @buf3_1 = sort (@buf2_1);
+    @buf3_2 = sort (@buf1_2);
+
+    my %in2;
+    foreach (@buf3_2) {
+        $in2{$_} = 1;
+    }
+
+    foreach (@buf3_1){
+        if (!exists($in2{$_})){
+            push @result, $_ 
+        }
+    }
+
+    # Save file with information about the files missing
+    # if any, and give information about this proceedier
+    #---------------------------------------------------
+    if(@result){
+        open OUT, ">missing";
+        print OUT @result;
+        print "\nHere is the result:\n\n", @result, "\n";
+        print "The file \"missing\" has been placed in the current directory.\n";
+        print "Files supposed to be ignored should be placed in \".potignore\"\n";
+    }
+
+    # If there is nothing to complain about, notice the user
+    #-------------------------------------------------------
+    else{
+        print "\nWell, it's all perfect! Congratulation!\n";
+    }         
+}
+
+sub InvalidOption{
+
+    # Handle invalid arguments
+    #-------------------------
+    print "update.pl: invalid option -- $LANG\n";
+    print "Try `update.pl --help' for more information.\n";
+}
+ 
+sub GenHeaders{
+
+    # Generate the .h header files, so we can allow glade and
+    # xml translation support
+    #--------------------------------------------------------
+    if(-s "ui-extract.pl"){
+
+        print "Found ui-extract.pl script\nRunning ui-extract...\n";
+
+        open FILE, "<POTFILES.in";
+        while (<FILE>) {
+
+           # Find .xml files in POTFILES.in and generate the
+           # files with help from the ui-extract.pl script
+           #--------------------------------------------------
+           if ($_=~ /(.*)(\.xml)/o){
+              $filename = "../$1.xml";
+              $xmlfiles="perl \.\/ui-extract.pl --local $filename";
+              system($xmlfiles);
+           }
+      
+           # Find .glade files in POTFILES.in and generate
+           # the files with help from the ui-extract.pl script
+           #--------------------------------------------------
+           elsif ($_=~ /(.*)(\.glade)/o){
+              $filename = "../$1.glade";
+              $xmlfiles="perl \.\/ui-extract.pl --local $filename";
+              system($xmlfiles);  
+           }
+       }
+       close FILE;
+
+       # Create .headerlock file, so the script will know 
+       # that we already passed this section. This is required 
+       # since the individual sections can be reaced at different
+       # times by the Makefile
+       #--------------------------------------------------------- 
+       system("touch .headerlock");
+   }
+}
+
+sub GeneratePot{
+
+    # Generate the potfiles from the POTFILES.in file
+    #------------------------------------------------
+
+    print "Building the $PACKAGE.pot...\n";
+
+    system ("mv POTFILES.in POTFILES.in.old");    
+
+    open INFILE, "<POTFILES.in.old";
+    open OUTFILE, ">POTFILES.in";
+    while (<INFILE>) {
+	if ($_ =~ /\.(glade|xml)$/) {
+        s/\.glade$/\.glade\.h/;
+        s/\.xml$/\.xml\.h/;
+        $_ = basename($_);
+	$_ = "po/tmp/$_\n";
+        }
+        print OUTFILE $_;        
+    }
+    close OUTFILE;
+    close INFILE;
+
+    $GETTEXT ="xgettext --default-domain\=$PACKAGE --directory\=\.\."
+             ." --add-comments --keyword\=\_ --keyword\=N\_ --keyword\=P_\:1,2"
+             ." --files-from\=\.\/POTFILES\.in ";  
+    $GTEST   ="test \! -f $PACKAGE\.po \|\| \( rm -f \.\/$PACKAGE\.pot "
+             ."&& mv $PACKAGE\.po \.\/$PACKAGE\.pot \)";
+
+    system($GETTEXT);
+    system($GTEST);
+    print "Wrote $PACKAGE.pot\n";
+    system("mv POTFILES.in.old POTFILES.in");
+
+    # If .headerlock file is found, it means that the potfiles
+    # already has been generated. If so delete the generated 
+    # .h header files. The reason for this approach with a 
+    # file as a marker is due to that the Makefile runs the
+    # scripts in turns
+    #---------------------------------------------------------
+
+    if(-e ".headerlock"){
+        unlink(".headerlock");
+
+        print "Removing generated header (.h) files...";
+	system("rm ./tmp/ -rf");
+    }
+    print "done\n";
+}
+
+sub Merging{
+
+    if ($ARGV[1]){
+        $LANG   = $ARGV[1]; 
+    } else {
+	$LANG   = $ARGV[0];
+    }
+
+    if ($ARGV[0] ne "--dist" && $ARGV[0] ne "-D") {
+        print "Merging $LANG.po with $PACKAGE.pot...";
+    }
+
+    $MERGE="cp $LANG.po $LANG.po.old && msgmerge $LANG.po.old $PACKAGE.pot -o $LANG.po";
+
+    system($MERGE);
+    
+    if ($ARGV[0] ne "--dist" && $ARGV[0] ne "-D") {
+        print "\n\n";
+    }
+
+    # Remove the "messages" trash file generated
+    # by gettext, aswell as the backup file
+    #-------------------------------------------
+    unlink "messages";
+    unlink "$LANG.po.old";
+}
+
+sub NotExisting{
+
+    # Report error if supplied language 
+    # file is non-existant
+    #----------------------------------
+    print "update.pl:  sorry, $LANG.po does not exist!\n";
+    print "Try `update.pl --help' for more information.\n";    
+    exit;
+}
+
+sub Status{
+
+    # Print status information about the po file
+    #-------------------------------------------
+    $STATUS="msgfmt --statistics $LANG.po";
+    
+    system($STATUS);
+    print "\n";   
+}
diff -wrauN -x .svn -x .gitignore nano-2.3.2/po/zh_TW.po nano/po/zh_TW.po
--- nano-2.3.2/po/zh_TW.po	2013-03-22 15:18:16.000000000 +0100
+++ nano/po/zh_TW.po	2013-11-29 16:24:20.000000000 +0100
@@ -113,7 +113,7 @@
 #: src/files.c:271
 #, fuzzy, c-format
 msgid "Error opening lockfile %s: %s"
-msgstr "ËÆÄÂèñ %s ÊôÇÁôºÁîüÈåØË™§: %s"
+msgstr "ÂØ´ÂÖ•Êö´Â≠òÊ™îÊ°à %s ÊôÇÁôºÁîüÈåØË™§"
 
 #: src/files.c:281
 #, c-format
diff -wrauN -x .svn -x .gitignore nano-2.3.2/src/cut.c nano/src/cut.c
--- nano-2.3.2/src/cut.c	2013-01-01 03:53:32.000000000 +0100
+++ nano/src/cut.c	2013-11-29 16:24:18.000000000 +0100
@@ -1,4 +1,4 @@
-/* $Id: cut.c 4453 2009-12-02 03:36:22Z astyanax $ */
+/* $Id: cut.c 4573 2013-04-08 03:19:21Z astyanax $ */
 /**************************************************************************
  *   cut.c                                                                *
  *                                                                        *
@@ -209,7 +209,9 @@
 #endif
 	/* Leave the text in the cutbuffer, and mark the file as
 	 * modified. */
+  if (!copy_text) {
 	set_modified();
+  }
 
     /* Update the screen. */
     edit_refresh_needed = TRUE;
diff -wrauN -x .svn -x .gitignore nano-2.3.2/src/files.c nano/src/files.c
--- nano-2.3.2/src/files.c	2013-03-21 14:17:24.000000000 +0100
+++ nano/src/files.c	2013-11-29 16:24:18.000000000 +0100
@@ -1,4 +1,4 @@
-/* $Id: files.c 4558 2013-01-03 04:50:49Z astyanax $ */
+/* $Id: files.c 4575 2013-04-13 02:56:03Z astyanax $ */
 /**************************************************************************
  *   files.c                                                              *
  *                                                                        *
@@ -1195,11 +1195,16 @@
 #endif
 
 #ifdef ENABLE_MULTIBUFFER
-	    if (ISSET(MULTIBUFFER))
+	    if (ISSET(MULTIBUFFER)) {
 		/* Update the screen to account for the current
 		 * buffer. */
+		ssize_t savedposline, savedposcol;
+
 		display_buffer();
-	    else
+		if (!execute && ISSET(POS_HISTORY)
+			&& check_poshistory(answer, &savedposline, &savedposcol))
+		    do_gotolinecolumn(savedposline, savedposcol, FALSE, FALSE, FALSE, FALSE);
+	    } else
 #endif
 	    {
 		filestruct *top_save = openfile->fileage;
@@ -2134,6 +2139,7 @@
 	if (openfile->current_stat == NULL)
 	    openfile->current_stat =
 		(struct stat *)nmalloc(sizeof(struct stat));
+	if (!openfile->mark_set)
 	stat(realname, openfile->current_stat);
 #endif
 
@@ -3195,7 +3201,7 @@
 	if (hist == NULL) {
 	    if (errno != ENOENT) {
 		/* Don't save history when we quit. */
-		UNSET(HISTORYLOG);
+		UNSET(POS_HISTORY);
 		history_error(N_("Error reading %s: %s"), nanohist,
 			strerror(errno));
 	    }
diff -wrauN -x .svn -x .gitignore nano-2.3.2/src/global.c nano/src/global.c
--- nano-2.3.2/src/global.c	2013-03-21 14:17:24.000000000 +0100
+++ nano/src/global.c	2013-11-29 16:24:18.000000000 +0100
@@ -1,4 +1,4 @@
-/* $Id: global.c 4556 2013-01-03 04:36:39Z astyanax $ */
+/* $Id: global.c 4579 2013-06-14 02:44:54Z astyanax $ */
 /**************************************************************************
  *   global.c                                                             *
  *                                                                        *
@@ -308,26 +308,41 @@
 #endif
 }
 
-const sc *first_sc_for(int menu, void (*func)(void)) {
+const sc *first_sc_for(int menu, void (*func)(void))
+{
     const sc *s;
+    const sc *fkeysc = NULL;
     const sc *metasc = NULL;
 
     for (s = sclist; s != NULL; s = s->next) {
 	if ((s->menu & menu) && s->scfunc == func) {
-	    /* try to use a meta sequence as a last resort.  Otherwise
-	       we will run into problems when we try and handle things like
-	       the arrow keys, home, etc, if for some reason the user bound
-	       them to a meta sequence first *shrug* */
-	    if (s->type == META) {
+	    /* Try to use function keys and meta sequences as last
+	     * resorts.  Otherwise, we will run into problems when we
+	     * try and handle things like the arrow keys, Home, etc., if
+	     * for some reason the user bound them to a function key or
+	     * meta sequence first *shrug*. */
+	    if (s->type == FKEY) {
+		if (!fkeysc)
+		    fkeysc = s;
+		continue;
+	    } else if (s->type == META) {
+		if (!metasc)
 		metasc = s;
 		continue;
-	    } /* otherwise it was something else, use it */
+	    }
+
+	    /* Otherwise, it was something else, so use it. */
 	    return s;
 	}
     }
 
-    /* If we're here we may have found only meta sequences, if so use one */
-    if (metasc)
+    /* If we're here, we may have found only function keys or meta
+     * sequences.  If so, use one, with the same priority as in the
+     * help browser: function keys come first, unless meta sequences are
+     * available, in which case meta sequences come first. */
+    if (fkeysc && !metasc)
+	return fkeysc;
+    else if (metasc)
 	return metasc;
 
 #ifdef DEBUG
