diff -Naur BerkeleyDB-0.23.orig/config.in BerkeleyDB-0.23/config.in
--- BerkeleyDB-0.23.orig/config.in	2003-06-15 17:12:42.000000000 -0400
+++ BerkeleyDB-0.23/config.in	2003-07-10 14:43:30.000000000 -0400
@@ -8,7 +8,7 @@
 #    installed on your system.
 
 #INCLUDE	= /usr/local/include
-INCLUDE	= /usr/local/BerkeleyDB/include
+INCLUDE	= @PREFIX@/include/db4
 
 # 2. Where is libdb?
 #
@@ -16,7 +16,7 @@
 #    installed on your system.
 
 #LIB	= /usr/local/lib
-LIB	= /usr/local/BerkeleyDB/lib
+LIB	= @PREFIX@/lib
 
 # 3. Is the library called libdb?
 #
