diff -uNr rrdtool-1.0.49/Makefile.in rrdtool-1.0.49-new/Makefile.in
--- rrdtool-1.0.49/Makefile.in	Tue Apr  6 17:30:22 2004
+++ rrdtool-1.0.49-new/Makefile.in	Wed Jul 21 16:43:03 2004
@@ -172,7 +172,7 @@
 
 # build the following subdirectories
 SUBDIRS = cgilib-0.4 gd1.3 $(SUBDIRS_1) $(SUBDIRS_2) \
-          src doc examples contrib tcl
+          src
 
 
 # the following files are not mentioned in any other Makefile
@@ -214,8 +214,7 @@
 	config/config.sub config/depcomp config/install-sh \
 	config/ltmain.sh config/missing config/mkinstalldirs configure \
 	configure.ac
-DIST_SUBDIRS = cgilib-0.4 gd1.3 zlib-1.1.4 libpng-1.0.9 src doc examples \
-	contrib tcl
+DIST_SUBDIRS = cgilib-0.4 gd1.3 zlib-1.1.4 libpng-1.0.9 src
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
diff -uNr rrdtool-1.0.49/configure rrdtool-1.0.49-new/configure
--- rrdtool-1.0.49/configure	Tue Apr  6 17:30:38 2004
+++ rrdtool-1.0.49-new/configure	Wed Jul 21 16:09:43 2004
@@ -18840,7 +18840,7 @@
 
 if test x$local_libpng = xyes
 then
-  PNG_LINK=-lpng
+  PNG_LINK="-L@PREFIX@/lib -lpng"
 
 else
   PNG_LIB_DIR=libpng-1.0.9
diff -uNr rrdtool-1.0.49/src/Makefile.in rrdtool-1.0.49-new/src/Makefile.in
--- rrdtool-1.0.49/src/Makefile.in	Tue Apr  6 17:30:25 2004
+++ rrdtool-1.0.49-new/src/Makefile.in	Wed Jul 21 16:08:56 2004
@@ -235,7 +235,7 @@
 librrd_private_la_SOURCES = $(RRD_C_FILES)
 
 librrd_la_LIBADD = $(RRD_LIBS)
-librrd_la_LDFLAGS = -version-info 0:0:0
+librrd_la_LDFLAGS = -version-info 1:49
 
 include_HEADERS = rrd.h
 
diff -uNr rrdtool-1.0.49/tcl/Makefile.in rrdtool-1.0.49-new/tcl/Makefile.in
--- rrdtool-1.0.49/tcl/Makefile.in	Tue Apr  6 17:30:25 2004
+++ rrdtool-1.0.49-new/tcl/Makefile.in	Wed Jul 21 16:08:56 2004
@@ -320,7 +320,7 @@
 
 
 @COMP_TCL_TRUE@$(TCL_LIB_FILE): tclrrd.o
-@COMP_TCL_TRUE@	$(TCL_SHLIB_LD) $(LIBDIRS) $< -o $@ @RRD_PRIVATE_LINK@ -lm $(TCL_LIB_SPEC)
+@COMP_TCL_TRUE@	$(TCL_SHLIB_LD) $(LIBDIRS) $< -o $@ @RRD_PRIVATE_LINK@ -ltcl -lm $(TCL_LIB_SPEC)
 
 @COMP_TCL_TRUE@tclrrd.o: tclrrd.c
 @COMP_TCL_TRUE@	$(CC) $(CFLAGS) $(TCL_SHLIB_CFLAGS) $(INCLUDES) $(AM_CPPFLAGS) -c $< -DVERSION=\"$(VERSION)\"
