diff -ruN dopewars-1.5.9.orig/ltmain.sh dopewars-1.5.9/ltmain.sh
--- dopewars-1.5.9.orig/ltmain.sh	Mon Jun  2 11:40:22 2003
+++ dopewars-1.5.9/ltmain.sh	Thu Nov  6 14:18:00 2003
@@ -1772,7 +1772,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    need_relink=no
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then
diff -ruN dopewars-1.5.9.orig/src/Makefile.in dopewars-1.5.9/src/Makefile.in
--- dopewars-1.5.9.orig/src/Makefile.in	Fri Jun  6 13:19:22 2003
+++ dopewars-1.5.9/src/Makefile.in	Thu Nov  6 14:09:52 2003
@@ -245,7 +245,7 @@
 	  echo " rm -f $$p $$f"; \
 	  rm -f $$p $$f ; \
 	done
-dopewars$(EXEEXT): $(dopewars_OBJECTS) $(dopewars_DEPENDENCIES) 
+dopewars$(EXEEXT): $(dopewars_OBJECTS)
 	@rm -f dopewars$(EXEEXT)
 	$(LINK) $(dopewars_LDFLAGS) $(dopewars_OBJECTS) $(dopewars_LDADD) $(LIBS)
 
diff -ruN dopewars-1.5.9.orig/src/plugins/Makefile.in dopewars-1.5.9/src/plugins/Makefile.in
--- dopewars-1.5.9.orig/src/plugins/Makefile.in	Fri Jun  6 13:19:23 2003
+++ dopewars-1.5.9/src/plugins/Makefile.in	Thu Nov  6 14:21:50 2003
@@ -152,14 +152,14 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(noinst_LTLIBRARIES)
 
-libsound_esd_la_LIBADD =
+libsound_esd_la_LIBADD = -module -avoid-version
 am_libsound_esd_la_OBJECTS = sound_esd.lo
 libsound_esd_la_OBJECTS = $(am_libsound_esd_la_OBJECTS)
-libsound_sdl_la_LIBADD =
+libsound_sdl_la_LIBADD = -module -avoid-version
 am_libsound_sdl_la_OBJECTS = sound_sdl.lo
 libsound_sdl_la_OBJECTS = $(am_libsound_sdl_la_OBJECTS)
 libsound_winmm_la_LDFLAGS =
-libsound_winmm_la_LIBADD =
+libsound_winmm_la_LIBADD = -module -avoid-version
 am_libsound_winmm_la_OBJECTS = sound_winmm.lo
 libsound_winmm_la_OBJECTS = $(am_libsound_winmm_la_OBJECTS)
 
