diff -Naur galeon-1.2.8/configure galeon-new/configure
--- galeon-1.2.8/configure	2003-02-15 20:03:13.000000000 -0500
+++ galeon-new/configure	2003-03-13 05:53:55.000000000 -0500
@@ -2871,13 +2871,13 @@
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g -O3"
   else
     CFLAGS="-g"
   fi
 else
   if test "$GCC" = yes; then
-    CFLAGS="-O2"
+    CFLAGS="-O3"
   else
     CFLAGS=
   fi
@@ -8449,13 +8449,13 @@
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g -O3"
   else
     CFLAGS="-g"
   fi
 else
   if test "$GCC" = yes; then
-    CFLAGS="-O2"
+    CFLAGS="-O3"
   else
     CFLAGS=
   fi
@@ -8973,13 +8973,13 @@
   CXXFLAGS=$ac_save_CXXFLAGS
 elif test $ac_cv_prog_cxx_g = yes; then
   if test "$GXX" = yes; then
-    CXXFLAGS="-g -O2"
+    CXXFLAGS="-g -O3"
   else
     CXXFLAGS="-g"
   fi
 else
   if test "$GXX" = yes; then
-    CXXFLAGS="-O2"
+    CXXFLAGS="-O3"
   else
     CXXFLAGS=
   fi
diff -Naur galeon-1.2.8/src/galeon.in galeon-new/src/galeon.in
--- galeon-1.2.8/src/galeon.in	2001-12-27 19:13:42.000000000 -0500
+++ galeon-new/src/galeon.in	2003-03-13 05:53:55.000000000 -0500
@@ -21,8 +21,8 @@
 	exit
 fi
 
-LD_LIBRARY_PATH=$MOZILLA_HOME:$LD_LIBRARY_PATH
-export LD_LIBRARY_PATH
+DYLD_LIBRARY_PATH=$MOZILLA_HOME:$DYLD_LIBRARY_PATH
+export DYLD_LIBRARY_PATH
 
 MOZILLA_FIVE_HOME=$MOZILLA_HOME
 export MOZILLA_FIVE_HOME
diff -Naur galeon-1.2.8/src/mozilla/Makefile.in galeon-new/src/mozilla/Makefile.in
--- galeon-1.2.8/src/mozilla/Makefile.in	2003-02-15 20:02:49.000000000 -0500
+++ galeon-new/src/mozilla/Makefile.in	2003-03-13 05:53:55.000000000 -0500
@@ -314,7 +314,7 @@
 	ProgressCallbacks.h
 
 
-libgaleon_xpcom_la_LDFLAGS = -avoid-version
+libgaleon_xpcom_la_LDFLAGS = -avoid-version -module
 
 galeon_libdir = $(pkglibdir)/components
 galeon_lib_LTLIBRARIES = libgaleon-xpcom.la
