diff -uNr kdebase-3.3.0/configure kdebase-3.3.0-new/configure
--- kdebase-3.3.0/configure	Mon Aug 16 05:24:19 2004
+++ kdebase-3.3.0-new/configure	Sun Sep  5 14:31:36 2004
@@ -7993,6 +7993,19 @@
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
+  netbsd* | freebsd* | openbsd* | darwin* )
+    # This has been around since 386BSD, at least.  Likely further.
+    if test -x /sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/sbin/sysctl -n kern.argmax`
+    elif test -x /usr/sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/usr/sbin/sysctl -n kern.argmax`
+    else
+      lt_cv_sys_max_cmd_len=65536 # usable default for *BSD
+    fi
+    # And add a safety zone
+    lt_cv_sys_max_cmd_len=`expr $lt_cv_sys_max_cmd_len \/ 4`
+    ;;
+
  *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -9636,10 +9649,21 @@
       archive_cmds_need_lc=no
       case "$host_os" in
       rhapsody* | darwin1.[012])
-	allow_undefined_flag='-undefined suppress'
+	allow_undefined_flag='-Wl,-undefined -Wl,suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -11486,10 +11510,21 @@
     archive_cmds_need_lc=no
     case "$host_os" in
     rhapsody* | darwin1.[012])
-      allow_undefined_flag='-undefined suppress'
+      allow_undefined_flag='-Wl,-undefined -Wl,suppress'
       ;;
     *) # Darwin 1.3 on
-      test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
       ;;
     esac
     output_verbose_link_cmd='echo'
@@ -12537,10 +12572,21 @@
     archive_cmds_need_lc_CXX=no
     case "$host_os" in
     rhapsody* | darwin1.[012])
-      allow_undefined_flag_CXX='-undefined suppress'
+      allow_undefined_flag_CXX='-Wl,-undefined -Wl,suppress'
       ;;
     *) # Darwin 1.3 on
-      test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_CXX='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_CXX='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_CXX='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_CXX='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
       ;;
     esac
     lt_int_apple_cc_single_mod=no
@@ -16360,10 +16406,21 @@
       archive_cmds_need_lc_F77=no
       case "$host_os" in
       rhapsody* | darwin1.[012])
-	allow_undefined_flag_F77='-undefined suppress'
+	allow_undefined_flag_F77='-Wl,-undefined -Wl,suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_F77='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_F77='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_F77='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_F77='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -18642,10 +18699,21 @@
       archive_cmds_need_lc_GCJ=no
       case "$host_os" in
       rhapsody* | darwin1.[012])
-	allow_undefined_flag_GCJ='-undefined suppress'
+	allow_undefined_flag_GCJ='-Wl,-undefined -Wl,suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_GCJ='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_GCJ='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_GCJ='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_GCJ='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
diff -uNr kdebase-3.3.0/kdeprint/kprinter/printwrapper.cpp kdebase-3.3.0-new/kdeprint/kprinter/printwrapper.cpp
--- kdebase-3.3.0/kdeprint/kprinter/printwrapper.cpp	Sat May 22 16:26:46 2004
+++ kdebase-3.3.0-new/kdeprint/kprinter/printwrapper.cpp	Sun Sep  5 14:31:29 2004
@@ -137,10 +137,6 @@
 {
 	KCmdLineArgs	*args = KCmdLineArgs::parsedArgs();
 
-#if defined(HAVE_SIGACTION) && !defined(HAVE_SIGSET)
-	struct sigaction action;
-#endif /* HAVE_SIGACTION && !HAVE_SIGSET*/
-
 	// read variables from command line
 	QString	printer = args->getOption("d");
 	QString	title = args->getOption("t");
@@ -320,6 +316,10 @@
 	if (filestoprint.count() == 0)
 	{
 		// print from stdin
+
+#if defined(HAVE_SIGACTION) && !defined(HAVE_SIGSET)
+	struct sigaction action;
+#endif /* HAVE_SIGACTION && !HAVE_SIGSET*/
 
 #  if defined(HAVE_SIGSET)
 		sigset(SIGHUP, signal_handler);
diff -uNr kdebase-3.3.0/kioslave/Makefile.in kdebase-3.3.0-new/kioslave/Makefile.in
--- kdebase-3.3.0/kioslave/Makefile.in	Mon Aug 16 05:23:45 2004
+++ kdebase-3.3.0-new/kioslave/Makefile.in	Sun Sep  5 14:31:29 2004
@@ -64,7 +64,7 @@
 	uninstall-recursive nmcheck-recursive bcheck-recursive
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = about cgi floppy filter fish info mac man nfs nntp pop3 \
+DIST_SUBDIRS = about cgi floppy filter fish info mac man nntp pop3 \
 	imap4 smtp sftp tar finger thumbnail ldap smb devices settings
 #>- DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 #>+ 1
@@ -496,7 +496,7 @@
 @include_kioslave_ldap_TRUE@LDAP_SUBDIR = ldap
 @include_kioslave_smb_TRUE@SMB_SUBDIR = smb
 @include_devicesdir_TRUE@DEVICES_SUBDIR = devices
-SUBDIRS = about cgi floppy filter fish info mac man nfs nntp pop3 imap4 smtp \
+SUBDIRS = about cgi floppy filter fish info mac man nntp pop3 imap4 smtp \
   sftp tar finger thumbnail $(LDAP_SUBDIR) $(SMB_SUBDIR) $(DEVICES_SUBDIR) settings
 
 #>- all: all-recursive
diff -uNr kdebase-3.3.0/kioslave/info/kde-info2html.conf kdebase-3.3.0-new/kioslave/info/kde-info2html.conf
--- kdebase-3.3.0/kioslave/info/kde-info2html.conf	Sat May 22 16:26:52 2004
+++ kdebase-3.3.0-new/kioslave/info/kde-info2html.conf	Sun Sep  5 14:31:29 2004
@@ -25,6 +25,7 @@
 
 #-- location of info files.
 our @INFODIR = (
+	    "@PREFIX@/share/info",
 	    "/usr/share/info",
 	    "/usr/info",
 	    "/usr/lib/info",
diff -uNr kdebase-3.3.0/kioslave/man/kio_man.cpp kdebase-3.3.0-new/kioslave/man/kio_man.cpp
--- kdebase-3.3.0/kioslave/man/kio_man.cpp	Thu Aug  5 14:57:13 2004
+++ kdebase-3.3.0-new/kioslave/man/kio_man.cpp	Sun Sep  5 14:31:29 2004
@@ -50,7 +50,7 @@
 
 MANProtocol *MANProtocol::_self = 0;
 
-#define SGML2ROFF_DIRS "/usr/lib/sgml"
+#define SGML2ROFF_DIRS "@PREFIX@/share/sgml"
 
 /*
  * Drop trailing ".section[.gz]" from name
@@ -862,6 +862,7 @@
                         "/usr/sunpc/man",
                         "/usr/ncd/man",
                         "/usr/newsprint/man",
+                        "@PREFIX@/share/man",
                         NULL };
 
 
diff -uNr kdebase-3.3.0/ksysguard/ksysguardd/ksysguardd.c kdebase-3.3.0-new/ksysguard/ksysguardd/ksysguardd.c
--- kdebase-3.3.0/ksysguard/ksysguardd/ksysguardd.c	Sat Jul 17 13:52:09 2004
+++ kdebase-3.3.0-new/ksysguard/ksysguardd/ksysguardd.c	Sun Sep  5 14:31:29 2004
@@ -433,7 +433,7 @@
     if ( FD_ISSET( socketNo, fds ) ) {
       int clientsocket;
       struct sockaddr addr;
-      socklen_t addr_len = sizeof( struct sockaddr );
+      kde_socklen_t addr_len = sizeof( struct sockaddr );
 
       /* a new connection */
       if ( ( clientsocket = accept( socketNo, &addr, &addr_len ) ) < 0 ) {
diff -uNr kdebase-3.3.0/libkonq/Makefile.in kdebase-3.3.0-new/libkonq/Makefile.in
--- kdebase-3.3.0/libkonq/Makefile.in	Mon Aug 16 05:24:15 2004
+++ kdebase-3.3.0-new/libkonq/Makefile.in	Sun Sep  5 14:31:29 2004
@@ -588,7 +588,7 @@
 #>- libkonq_la_LDFLAGS = $(all_libraries) -version-info 6:0:2 -no-undefined
 #>+ 1
 libkonq_la_LDFLAGS = $(all_libraries) -version-info 6:0:2 -no-undefined $(KDE_NO_UNDEFINED)
-libkonq_la_LIBADD = $(LIB_KPARTS)
+libkonq_la_LIBADD = $(LIB_KPARTS) $(LIBZ)
 #>- libkonq_la_SOURCES = konq_popupmenu.cc knewmenu.cc \
 #>-    konq_xmlguiclient.cc\
 #>-    kfileivi.cc konq_iconviewwidget.cc konq_settings.cc konq_drag.cc  \
diff -uNr kdebase-3.3.0/startkde kdebase-3.3.0-new/startkde
--- kdebase-3.3.0/startkde	Thu Aug  5 14:56:53 2004
+++ kdebase-3.3.0-new/startkde	Sun Sep  5 14:31:29 2004
@@ -3,6 +3,8 @@
 #  DEFAULT KDE STARTUP SCRIPT ( KDE-3.3 )
 #
 
+source "@PREFIX@/bin/init.sh"
+
 # When the X server dies we get a HUP signal from xinit. We must ignore it
 # because we still need to do some cleanup.
 trap 'echo GOT SIGHUP' HUP
@@ -192,7 +194,7 @@
     done
     dcopserver_shutdown
     # shutdown will also make kwin quit, give it time to do so
-    sleep 1
+    sleep 5
 fi
 
 # the splashscreen and progress indicator
@@ -216,7 +218,29 @@
 # We only check for 255 which means that the ksmserver process could not be 
 # started, any problems thereafter, e.g. ksmserver failing to initialize, 
 # will remain undetected.
-test -n "$KDEWM" && KDEWM="--windowmanager $KDEWM"
+
+USE_PROXY=0
+if test -n "$KDEWM"; then
+  if test -z `expr "$KDEWM" : '\(.*quartz-wm.*\)'`; then
+    USE_PROXY=1
+  fi
+  KDEWM="--windowmanager $KDEWM"
+else
+  KDEWM="--windowmanager kwin"
+  USE_PROXY=1
+fi
+
+if test "$USE_PROXY" = "1"; then
+  if test -x /usr/X11R6/bin/quartz-wm; then
+    /usr/X11R6/bin/quartz-wm --only-proxy >/dev/null 2>&1 &
+  else
+    AUTOCUTSEL=`which autocutsel 2>/dev/null`
+    if test -n "$AUTOCUTSEL" && test -x "$AUTOCUTSEL"; then
+      $AUTOCUTSEL >/dev/null 2>&1 &
+    fi
+  fi
+fi
+
 kwrapper ksmserver $KDEWM 
 if test $? -eq 255; then
   # Startup error
--- kdelibs-3.3.1/admin/ltmain.sh	Tue Jun 15 12:47:06 2004
+++ kdelibs-3.3.1-new/admin/ltmain.sh	Wed Oct 13 11:34:12 2004
@@ -4397,6 +4397,7 @@
 	# We should set the shlibpath_var
 	rpath=
 	for dir in $temp_rpath; do
+     if test "$dir" != "/sw/lib"; then
 	  case $dir in
 	  [\\/]* | [A-Za-z]:[\\/]*)
 	    # Absolute path.
@@ -4407,6 +4408,7 @@
 	    rpath="$rpath\$thisdir/$dir:"
 	    ;;
 	  esac
+     fi
 	done
 	temp_rpath="$rpath"
       fi
