diff -uNr postgresql-7.3.6/configure postgresql-7.3.6-new/configure
--- postgresql-7.3.6/configure	Mon Mar  1 19:44:51 2004
+++ postgresql-7.3.6-new/configure	Fri Jun 18 13:08:58 2004
@@ -4222,7 +4222,7 @@
 
 
 # Extract the first word of "perl", so it can be a program name with args.
-set dummy perl; ac_word=$2
+set dummy perl; ac_word=perl5.6.0
 echo "$as_me:$LINENO: checking for $ac_word" >&5
 echo $ECHO_N "checking for $ac_word... $ECHO_C" >&6
 if test "${ac_cv_path_PERL+set}" = set; then
@@ -4289,7 +4289,7 @@
 
 if test "$with_python" = yes; then
   # Extract the first word of "python", so it can be a program name with args.
-set dummy python; ac_word=$2
+set dummy python; ac_word=python2.1
 echo "$as_me:$LINENO: checking for $ac_word" >&5
 echo $ECHO_N "checking for $ac_word... $ECHO_C" >&6
 if test "${ac_cv_path_PYTHON+set}" = set; then
@@ -8397,8 +8397,8 @@
 
 if test "$with_pam" = yes ; then
   if test "${ac_cv_header_security_pam_appl_h+set}" = set; then
-  echo "$as_me:$LINENO: checking for security/pam_appl.h" >&5
-echo $ECHO_N "checking for security/pam_appl.h... $ECHO_C" >&6
+  echo "$as_me:$LINENO: checking for pam/pam_appl.h" >&5
+echo $ECHO_N "checking for pam/pam_appl.h... $ECHO_C" >&6
 if test "${ac_cv_header_security_pam_appl_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 fi
@@ -8406,13 +8406,13 @@
 echo "${ECHO_T}$ac_cv_header_security_pam_appl_h" >&6
 else
   # Is the header compilable?
-echo "$as_me:$LINENO: checking security/pam_appl.h usability" >&5
-echo $ECHO_N "checking security/pam_appl.h usability... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking pam/pam_appl.h usability" >&5
+echo $ECHO_N "checking pam/pam_appl.h usability... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 #include "confdefs.h"
 $ac_includes_default
-#include <security/pam_appl.h>
+#include <pam/pam_appl.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
@@ -8437,12 +8437,12 @@
 echo "${ECHO_T}$ac_header_compiler" >&6
 
 # Is the header present?
-echo "$as_me:$LINENO: checking security/pam_appl.h presence" >&5
-echo $ECHO_N "checking security/pam_appl.h presence... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking pam/pam_appl.h presence" >&5
+echo $ECHO_N "checking pam/pam_appl.h presence... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 #include "confdefs.h"
-#include <security/pam_appl.h>
+#include <pam/pam_appl.h>
 _ACEOF
 if { (eval echo "$as_me:$LINENO: \"$ac_cpp conftest.$ac_ext\"") >&5
   (eval $ac_cpp conftest.$ac_ext) 2>conftest.er1
@@ -8474,20 +8474,20 @@
 # So?  What about this header?
 case $ac_header_compiler:$ac_header_preproc in
   yes:no )
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&5
-echo "$as_me: WARNING: security/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&2;};;
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&2;};;
   no:yes )
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: present but cannot be compiled" >&5
-echo "$as_me: WARNING: security/pam_appl.h: present but cannot be compiled" >&2;}
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: check for missing prerequisite headers?" >&5
-echo "$as_me: WARNING: security/pam_appl.h: check for missing prerequisite headers?" >&2;}
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&2;};;
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: present but cannot be compiled" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: present but cannot be compiled" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: check for missing prerequisite headers?" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: check for missing prerequisite headers?" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&2;};;
 esac
-echo "$as_me:$LINENO: checking for security/pam_appl.h" >&5
-echo $ECHO_N "checking for security/pam_appl.h... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking for pam/pam_appl.h" >&5
+echo $ECHO_N "checking for pam/pam_appl.h... $ECHO_C" >&6
 if test "${ac_cv_header_security_pam_appl_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
@@ -8500,8 +8500,8 @@
 if test $ac_cv_header_security_pam_appl_h = yes; then
   :
 else
-  { { echo "$as_me:$LINENO: error: header file <security/pam_appl.h> is required for PAM" >&5
-echo "$as_me: error: header file <security/pam_appl.h> is required for PAM" >&2;}
+  { { echo "$as_me:$LINENO: error: header file <pam/pam_appl.h> is required for PAM" >&5
+echo "$as_me: error: header file <pam/pam_appl.h> is required for PAM" >&2;}
    { (exit 1); exit 1; }; }
 fi
 
diff -uNr postgresql-7.3.6/configure.in postgresql-7.3.6-new/configure.in
--- postgresql-7.3.6/configure.in	Mon Mar  1 19:44:53 2004
+++ postgresql-7.3.6-new/configure.in	Fri Jun 18 13:08:58 2004
@@ -729,7 +729,12 @@
 fi
 
 if test "$with_pam" = yes ; then
-  AC_CHECK_HEADER([security/pam_appl.h], [], [AC_MSG_ERROR([header file <security/pam_appl.h> is required for PAM])])
+  AC_CHECK_HEADERS([security/pam_appl.h], [], [
+    AC_CHECK_HEADERS([pam/pam_appl.h], [
+      AC_DEFINE([HAVE_PAM_PAM_APPL_H], 1, [Define if pam_appl.h is in pam/ instead of security/])
+      ], [AC_MSG_ERROR([header file <security/pam_appl.h> or <pam/pam_appl.h> is required for PAM])]
+    )
+  ])
 fi
 
 
diff -uNr postgresql-7.3.6/pgsql.sh postgresql-7.3.6-new/pgsql.sh
--- postgresql-7.3.6/pgsql.sh	Wed Dec 31 19:00:00 1969
+++ postgresql-7.3.6-new/pgsql.sh	Fri Jun 18 13:08:58 2004
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+for flag in "-n" "-d" "-s" "-u" "-v"; do
+	ulimit $flag 65536 >/dev/null 2>&1 || :
+done
+
+PREFIX="@INSTPREFIX@"
+DATADIR="${PREFIX}/var/postgresql/data"
+LOGFILE="${PREFIX}/var/postgresql/pgsql.log"
+
+case "$1" in
+	start)
+		sudo -u postgres "${PREFIX}/bin/pg_ctl" start -D "$DATADIR" -l "$LOGFILE"
+		;;
+	restart)
+		sudo -u postgres "${PREFIX}/bin/pg_ctl" restart -D "$DATADIR" -m fast
+		;;
+	stop)
+		sudo -u postgres "${PREFIX}/bin/pg_ctl" stop -D "$DATADIR" -m fast
+		;;
+esac
diff -uNr postgresql-7.3.6/src/Makefile.shlib postgresql-7.3.6-new/src/Makefile.shlib
--- postgresql-7.3.6/src/Makefile.shlib	Sat Jan 11 12:22:34 2003
+++ postgresql-7.3.6-new/src/Makefile.shlib	Fri Jun 18 13:08:58 2004
@@ -81,8 +81,24 @@
 endif
 
 ifeq ($(PORTNAME), darwin)
-  shlib			:= lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)
-  LINK.shared		= $(COMPILER) $(DARWIN_NAMESPACE_SPEC) -bundle -undefined suppress
+  ifeq (,$(filter $(host_os), darwin1.0 darwin1.1 darwin1.2))
+     MULTIPLY_DEFINED	=
+  else
+     MULTIPLY_DEFINED	= -multiply_defined suppress
+  endif
+
+  shlib			:= lib$(NAME).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)$(DLSUFFIX)
+  ifneq ($(SO_MAJOR_VERSION), 0)
+    version_link	:= -compatibility_version $(SO_MAJOR_VERSION) -current_version $(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)
+  endif
+  ifeq ($(DLTYPE), bundle)
+    DLSUFFIX		:= .so
+    LINK.shared		= $(COMPILER) $(DARWIN_NAMESPACE_SPEC) -bundle -undefined suppress -flat_namespace
+  else
+    DLSUFFIX		:= .dylib
+    LINK.shared		= $(COMPILER) $(DARWIN_NAMESPACE_SPEC) -install_name $(libdir)/lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX) $(version_link) -dynamiclib $(MULTIPLY_DEFINED)
+  endif
+  SHLIB_LINK		:= -L$(libpq_builddir) $(SHLIB_LINK)
 endif
 
 ifeq ($(PORTNAME), openbsd)
@@ -246,6 +262,7 @@
 ifneq ($(PORTNAME), beos)
 ifneq ($(PORTNAME), win)
 ifneq ($(PORTNAME), aix)
+ifneq ($(PORTNAME), darwin)
 
 # Normal case
 $(shlib): $(OBJS)
@@ -261,6 +278,25 @@
 	$(LN_S) $(shlib) lib$(NAME)$(DLSUFFIX)
 endif
 
+else # PORTNAME == darwin
+
+# Darwin case
+# (same as normal case, but libfoo.version.suffix)
+
+$(shlib): $(OBJS)
+	$(LINK.shared) $(OBJS) $(SHLIB_LINK) -o $@
+ifneq ($(shlib), lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX))
+	rm -f lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX)
+	$(LN_S) $(shlib) lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX)
+endif
+
+ifneq ($(shlib), lib$(NAME)$(DLSUFFIX))
+	rm -f lib$(NAME)$(DLSUFFIX)
+	$(LN_S) $(shlib) lib$(NAME)$(DLSUFFIX)
+endif
+
+endif # PORTNAME == darwin
+
 else # PORTNAME == aix
 
 # AIX case
@@ -308,7 +344,10 @@
 ifeq ($(enable_shared), yes)
 install-lib-shared: $(shlib)
 	$(INSTALL_SHLIB) $< $(DESTDIR)$(libdir)/$(shlib)
+
 ifneq ($(PORTNAME), win)
+ifneq ($(PORTNAME), darwin)
+
 ifneq ($(shlib), lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION))
 	cd $(DESTDIR)$(libdir) && \
 	rm -f lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION) && \
@@ -320,7 +359,23 @@
 	$(LN_S) $(shlib) lib$(NAME)$(DLSUFFIX)
 endif
 
-endif # not win
+else # PORTNAME == darwin
+
+ifneq ($(shlib), lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX))
+	cd $(DESTDIR)$(libdir) && \
+	rm -f lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX) && \
+	$(LN_S) $(shlib) lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX)
+endif
+ifneq ($(shlib), lib$(NAME)$(DLSUFFIX))
+	cd $(DESTDIR)$(libdir) && \
+	rm -f lib$(NAME)$(DLSUFFIX) && \
+	$(LN_S) $(shlib) lib$(NAME)$(DLSUFFIX)
+endif
+
+endif # PORTNAME == darwin
+
+endif # PORTNAME == win
+
 endif # enable_shared
 
 
@@ -334,7 +389,9 @@
 ifeq ($(enable_shared), yes)
 	rm -f $(DESTDIR)$(libdir)/lib$(NAME)$(DLSUFFIX) \
 	  $(DESTDIR)$(libdir)/lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION) \
-	  $(DESTDIR)$(libdir)/lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)
+	  $(DESTDIR)$(libdir)/lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION) \
+	  $(DESTDIR)$(libdir)/lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX) \
+	  $(DESTDIR)$(libdir)/lib$(NAME).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)$(DLSUFFIX)
 endif # enable_shared
 
 
@@ -346,7 +403,7 @@
 clean-lib:
 	rm -f lib$(NAME).a
 ifeq ($(enable_shared), yes)
-	rm -f lib$(NAME)$(DLSUFFIX) lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION) lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)
+	rm -f lib$(NAME)$(DLSUFFIX) lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION) lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION) lib$(NAME).$(SO_MAJOR_VERSION)$(DLSUFFIX) lib$(NAME).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)$(DLSUFFIX)
 ifdef EXPSUFF
 	rm -f lib$(NAME)$(EXPSUFF)
 endif
diff -uNr postgresql-7.3.6/src/backend/libpq/auth.c postgresql-7.3.6-new/src/backend/libpq/auth.c
--- postgresql-7.3.6/src/backend/libpq/auth.c	Wed Sep  4 19:31:34 2002
+++ postgresql-7.3.6-new/src/backend/libpq/auth.c	Fri Jun 18 13:08:58 2004
@@ -42,7 +42,12 @@
 char	   *pg_krb_server_keyfile;
 
 #ifdef USE_PAM
+
+#ifdef HAVE_PAM_PAM_APPL_H
+#include <pam/pam_appl.h>
+#else
 #include <security/pam_appl.h>
+#endif
 
 #define PGSQL_PAM_SERVICE "postgresql"	/* Service name passed to PAM */
 
diff -uNr postgresql-7.3.6/src/backend/postmaster/postmaster.c postgresql-7.3.6-new/src/backend/postmaster/postmaster.c
--- postgresql-7.3.6/src/backend/postmaster/postmaster.c	Sat Feb 22 23:48:38 2003
+++ postgresql-7.3.6-new/src/backend/postmaster/postmaster.c	Fri Jun 18 13:08:58 2004
@@ -85,6 +85,10 @@
 #include <getopt.h>
 #endif
 
+#ifdef HAVE_RENDEZVOUS
+#include <DNSServiceDiscovery/DNSServiceDiscovery.h>
+#endif
+
 #include "catalog/pg_database.h"
 #include "commands/async.h"
 #include "lib/dllist.h"
@@ -203,6 +207,10 @@
 bool		Log_connections = false;
 bool		Db_user_namespace = false;
 
+#ifdef HAVE_RENDEZVOUS
+/* service name */
+char		*service_name = NULL;
+#endif
 
 /* Startup/shutdown state */
 static pid_t StartupPID = 0,
@@ -354,6 +362,17 @@
 }
 
 
+#ifdef HAVE_RENDEZVOUS
+
+/* reg_reply -- empty callback function for DNSServiceRegistrationCreate() */
+static void
+reg_reply(DNSServiceRegistrationReplyErrorType errorCode, void *context)
+{
+
+}
+
+#endif
+
 int
 PostmasterMain(int argc, char *argv[])
 {
@@ -684,6 +703,21 @@
 		{
 			postmaster_error("cannot create INET stream port");
 			ExitPostmaster(1);
+                } else {
+#ifdef HAVE_RENDEZVOUS
+                    /* Rendezvous, baby! */
+                    
+                    if (service_name != NULL)
+                    {
+                        DNSServiceRegistrationCreate(service_name,
+                                "_pgsql._tcp.",
+                                "",
+                                htonl(PostPortNumber),
+                                "",
+                                (DNSServiceRegistrationReply)reg_reply,
+                                NULL);
+                    }
+#endif
 		}
 	}
 
diff -uNr postgresql-7.3.6/src/backend/utils/fmgr/dfmgr.c postgresql-7.3.6-new/src/backend/utils/fmgr/dfmgr.c
--- postgresql-7.3.6/src/backend/utils/fmgr/dfmgr.c	Sun Sep  1 22:47:05 2002
+++ postgresql-7.3.6-new/src/backend/utils/fmgr/dfmgr.c	Fri Jun 18 13:08:58 2004
@@ -287,6 +287,26 @@
 		pfree(full);
 	}
 
+	new = palloc(strlen(name) + strlen(".so") + 1);
+	strcpy(new, name);
+	strcat(new, ".so");
+
+	if (!have_slash)
+	{
+		full = find_in_dynamic_libpath(new);
+		pfree(new);
+		if (full)
+			return full;
+	}
+	else
+	{
+		full = substitute_libpath_macro(new);
+		pfree(new);
+		if (file_exists(full))
+			return full;
+		pfree(full);
+	}
+
 	new = palloc(strlen(name) + strlen(DLSUFFIX) + 1);
 	strcpy(new, name);
 	strcat(new, DLSUFFIX);
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/ascii_and_mic/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/ascii_and_mic/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/ascii_and_mic/Makefile	Wed Aug 21 22:18:45 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/ascii_and_mic/Makefile	Fri Jun 18 13:08:58 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.3 2002/08/22 02:18:45 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/ascii_and_mic
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/cyrillic_and_mic/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/cyrillic_and_mic/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/cyrillic_and_mic/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/cyrillic_and_mic/Makefile	Fri Jun 18 13:08:58 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/cyrillic_and_mic
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/euc_cn_and_mic/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/euc_cn_and_mic/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/euc_cn_and_mic/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/euc_cn_and_mic/Makefile	Fri Jun 18 13:08:58 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/euc_cn_and_mic
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/euc_jp_and_sjis/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/euc_jp_and_sjis/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/euc_jp_and_sjis/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/euc_jp_and_sjis/Makefile	Fri Jun 18 13:08:58 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/euc_jp_and_sjis
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/euc_kr_and_mic/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/euc_kr_and_mic/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/euc_kr_and_mic/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/euc_kr_and_mic/Makefile	Fri Jun 18 13:08:58 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/euc_kr_and_mic
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/euc_tw_and_big5/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/euc_tw_and_big5/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/euc_tw_and_big5/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/euc_tw_and_big5/Makefile	Fri Jun 18 13:08:58 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/euc_tw_and_big5
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/latin2_and_win1250/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/latin2_and_win1250/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/latin2_and_win1250/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/latin2_and_win1250/Makefile	Fri Jun 18 13:08:58 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/latin2_and_win1250
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/latin_and_mic/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/latin_and_mic/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/latin_and_mic/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/latin_and_mic/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/latin_and_mic
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/proc.mk postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/proc.mk
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/proc.mk	Thu Sep  5 14:28:46 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/proc.mk	Fri Jun 18 13:08:59 2004
@@ -1,6 +1,7 @@
 SRCS		+= $(NAME).c
 OBJS		+= $(NAME).o
 
+DLTYPE		= bundle
 SHLIB_LINK 	:= $(BE_DLLLIBS)
 
 SO_MAJOR_VERSION := 0
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_ascii/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_ascii/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_ascii/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_ascii/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_ascii
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_big5/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_big5/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_big5/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_big5/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_big5
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_cyrillic/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_cyrillic/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_cyrillic/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_cyrillic/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_cyrillic
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_euc_cn/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_euc_cn/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_euc_cn/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_euc_cn/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_euc_cn
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_euc_jp/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_euc_jp/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_euc_jp/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_euc_jp/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_euc_jp
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_euc_kr/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_euc_kr/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_euc_kr/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_euc_kr/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_euc_kr
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_euc_tw/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_euc_tw/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_euc_tw/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_euc_tw/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_euc_tw
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_gb18030/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_gb18030/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_gb18030/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_gb18030/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_gb18030
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_gbk/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_gbk/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_gbk/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_gbk/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_gbk
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_iso8859/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_iso8859/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_iso8859/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_iso8859/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_iso8859
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_iso8859_1/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_iso8859_1/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_iso8859_1/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_iso8859_1/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_iso8859_1
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_johab/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_johab/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_johab/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_johab/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_johab
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_sjis/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_sjis/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_sjis/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_sjis/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_sjis
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_tcvn/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_tcvn/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_tcvn/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_tcvn/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_tcvn
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_uhc/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_uhc/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_uhc/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_uhc/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_uhc
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_win1250/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_win1250/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_win1250/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_win1250/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_win1250
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_win1256/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_win1256/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_win1256/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_win1256/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_win1256
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_win874/Makefile postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_win874/Makefile
--- postgresql-7.3.6/src/backend/utils/mb/conversion_procs/utf8_and_win874/Makefile	Wed Aug 21 17:33:55 2002
+++ postgresql-7.3.6-new/src/backend/utils/mb/conversion_procs/utf8_and_win874/Makefile	Fri Jun 18 13:08:59 2004
@@ -3,6 +3,9 @@
 # $Id: Makefile,v 1.2 2002/08/21 21:33:55 momjian Exp $
 #
 #-------------------------------------------------------------------------
+
+DLTYPE = bundle
+
 subdir = src/backend/utils/mb/conversion_procs/utf8_and_win874
 top_builddir = ../../../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/backend/utils/misc/guc.c postgresql-7.3.6-new/src/backend/utils/misc/guc.c
--- postgresql-7.3.6/src/backend/utils/misc/guc.c	Thu Apr  3 19:32:57 2003
+++ postgresql-7.3.6-new/src/backend/utils/misc/guc.c	Fri Jun 18 13:08:59 2004
@@ -798,6 +798,13 @@
 		"$user,public", assign_search_path, NULL
 	},
 
+#ifdef HAVE_RENDEZVOUS
+	{
+		{"service_name", PGC_POSTMASTER}, &service_name,
+		NULL, NULL, NULL
+	},
+#endif
+
 	{
 		{"server_encoding", PGC_USERSET}, &server_encoding_string,
 		"SQL_ASCII", assign_server_encoding, show_server_encoding
@@ -838,12 +845,12 @@
 
 	{
 		{"unix_socket_group", PGC_POSTMASTER}, &Unix_socket_group,
-		"", NULL, NULL
+		"postgres", NULL, NULL
 	},
 
 	{
 		{"unix_socket_directory", PGC_POSTMASTER}, &UnixSocketDir,
-		"", NULL, NULL
+		"/tmp", NULL, NULL
 	},
 
 	{
diff -uNr postgresql-7.3.6/src/backend/utils/misc/postgresql.conf.sample postgresql-7.3.6-new/src/backend/utils/misc/postgresql.conf.sample
--- postgresql-7.3.6/src/backend/utils/misc/postgresql.conf.sample	Sun Mar 30 16:38:13 2003
+++ postgresql-7.3.6-new/src/backend/utils/misc/postgresql.conf.sample	Fri Jun 18 13:08:59 2004
@@ -27,10 +27,10 @@
 #
 #	Connection Parameters
 #
-#tcpip_socket = false
-#ssl = false
+tcpip_socket = true
+ssl = false
 
-#max_connections = 32
+max_connections = 96
 #superuser_reserved_connections = 2
 
 #port = 5432 
@@ -49,7 +49,7 @@
 #
 #	Shared Memory Size
 #
-#shared_buffers = 64		# min max_connections*2 or 16, 8KB each
+shared_buffers = 192		# min max_connections*2 or 16, 8KB each
 #max_fsm_relations = 1000	# min 10, fsm is free space map, ~40 bytes
 #max_fsm_pages = 10000		# min 1000, fsm is free space map, ~6 bytes
 #max_locks_per_transaction = 64	# min 10
diff -uNr postgresql-7.3.6/src/bin/initdb/initdb.sh postgresql-7.3.6-new/src/bin/initdb/initdb.sh
--- postgresql-7.3.6/src/bin/initdb/initdb.sh	Mon Nov 25 16:42:11 2002
+++ postgresql-7.3.6-new/src/bin/initdb/initdb.sh	Fri Jun 18 13:08:59 2004
@@ -122,20 +122,20 @@
 # their trees to backup places, so $bindir might be inaccurate.
 
 if [ x"$self_path" != x"" ] \
-  && [ -x "$self_path/postgres" ] \
-  && [ x"`$self_path/postgres -V 2>/dev/null`" = x"postgres (PostgreSQL) $VERSION" ]
+  && [ -x "$self_path/postgres-7.3" ] \
+  && [ x"`$self_path/postgres-7.3 -V 2>/dev/null`" = x"postgres (PostgreSQL) $VERSION" ]
 then
     PGPATH=$self_path
-elif [ -x "$bindir/postgres" ]; then
-    if [ x"`$bindir/postgres -V 2>/dev/null`" = x"postgres (PostgreSQL) $VERSION" ]
+elif [ -x "$bindir/postgres-7.3" ]; then
+    if [ x"`$bindir/postgres-7.3 -V 2>/dev/null`" = x"postgres (PostgreSQL) $VERSION" ]
     then
         PGPATH=$bindir
     else
         # Maybe there was an error message?
-        errormsg=`$bindir/postgres -V 2>&1 >/dev/null`
+        errormsg=`$bindir/postgres-7.3 -V 2>&1 >/dev/null`
       (
         echo "The program "
-        echo "    '$bindir/postgres'"
+        echo "    '$bindir/postgres-7.3'"
         echo "needed by $CMDNAME does not belong to PostgreSQL version $VERSION, or"
         echo "there may be a configuration problem."
         if test x"$errormsg" != x""; then
@@ -147,7 +147,7 @@
         exit 1
     fi
 else
-    echo "The program 'postgres' is needed by $CMDNAME but was not found in" 1>&2
+    echo "The program 'postgres-7.3' is needed by $CMDNAME but was not found in" 1>&2
     echo "the directory '$bindir'.  Check your installation." 1>&2
     exit 1
 fi
@@ -155,20 +155,20 @@
 
 # Now we can assume that 'pg_id' belongs to the same version as the
 # verified 'postgres' in the same directory.
-if [ ! -x "$PGPATH/pg_id" ]; then
-    echo "The program 'pg_id' is needed by $CMDNAME but was not found in" 1>&2
+if [ ! -x "$PGPATH/pg_id-7.3" ]; then
+    echo "The program 'pg_id-7.3' is needed by $CMDNAME but was not found in" 1>&2
     echo "the directory '$PGPATH'.  Check your installation." 1>&2
     exit 1
 fi
 
 
-EffectiveUser=`$PGPATH/pg_id -n -u`
+EffectiveUser=`$PGPATH/pg_id-7.3 -n -u`
 if [ -z "$EffectiveUser" ]; then
     echo "$CMDNAME: could not determine current user name" 1>&2
     exit 1
 fi
 
-if [ `$PGPATH/pg_id -u` -eq 0 ]
+if [ `$PGPATH/pg_id-7.3 -u` -eq 0 ]
 then
     echo "You cannot run $CMDNAME as root. Please log in (using, e.g., 'su')" 1>&2
     echo "as the (unprivileged) user that will own the server process." 1>&2
@@ -370,7 +370,7 @@
 
 if [ "$ENCODING" ]
 then
-	ENCODINGID=`$PGPATH/pg_encoding -b $ENCODING`
+	ENCODINGID=`$PGPATH/pg_encoding-7.3 -b $ENCODING`
 	if [ "$?" -ne 0 ]
 	then
               (
@@ -577,7 +577,7 @@
   export LC_COLLATE
   export LC_CTYPE
   unset LC_ALL
-  "$PGPATH"/postgres -boot -x1 $PGSQL_OPT $BACKEND_TALK_ARG template1
+  "$PGPATH"/postgres-7.3 -boot -x1 $PGSQL_OPT $BACKEND_TALK_ARG template1
 ) \
 || exit_nicely
 
@@ -625,7 +625,7 @@
 
 $ECHO_N "initializing pg_shadow... "$ECHO_C
 
-"$PGPATH"/postgres $PGSQL_OPT template1 >/dev/null <<EOF
+"$PGPATH"/postgres-7.3 $PGSQL_OPT template1 >/dev/null <<EOF
 -- Create a trigger so that direct updates to pg_shadow will be written
 -- to the flat password/group files pg_pwd and pg_group
 CREATE TRIGGER pg_sync_pg_pwd AFTER INSERT OR UPDATE OR DELETE ON pg_shadow \
@@ -658,7 +658,7 @@
         exit_nicely
     fi
     $ECHO_N "setting password... "$ECHO_C
-    "$PGPATH"/postgres $PGSQL_OPT template1 >/dev/null <<EOF
+    "$PGPATH"/postgres-7.3 $PGSQL_OPT template1 >/dev/null <<EOF
 	ALTER USER "$POSTGRES_SUPERUSERNAME" WITH PASSWORD '$FirstPw';
 EOF
     if [ "$?" -ne 0 ]; then
@@ -674,7 +674,7 @@
 
 $ECHO_N "enabling unlimited row size for system tables... "$ECHO_C
 
-"$PGPATH"/postgres $PGSQL_OPT template1 >/dev/null <<EOF
+"$PGPATH"/postgres-7.3 $PGSQL_OPT template1 >/dev/null <<EOF
 ALTER TABLE pg_attrdef CREATE TOAST TABLE;
 ALTER TABLE pg_constraint CREATE TOAST TABLE;
 ALTER TABLE pg_database CREATE TOAST TABLE;
@@ -693,7 +693,7 @@
 
 $ECHO_N "initializing pg_depend... "$ECHO_C
 
-"$PGPATH"/postgres $PGSQL_OPT template1 >/dev/null <<EOF
+"$PGPATH"/postgres-7.3 $PGSQL_OPT template1 >/dev/null <<EOF
 -- Make PIN entries in pg_depend for all objects made so far in the tables
 -- that the dependency code handles.  This is overkill (the system doesn't
 -- really depend on having every last weird datatype, for instance)
@@ -725,7 +725,7 @@
 
 $ECHO_N "creating system views... "$ECHO_C
 
-"$PGPATH"/postgres $PGSQL_OPT template1 >/dev/null <<EOF
+"$PGPATH"/postgres-7.3 $PGSQL_OPT template1 >/dev/null <<EOF
 
 CREATE VIEW pg_user AS \
     SELECT \
@@ -1014,12 +1014,12 @@
     FROM tmp_pg_description t, pg_class c WHERE c.relname = t.classname;
 EOF
 ) \
-	| "$PGPATH"/postgres $PGSQL_OPT template1 > /dev/null || exit_nicely
+	| "$PGPATH"/postgres-7.3 $PGSQL_OPT template1 > /dev/null || exit_nicely
 echo "ok"
 
 # Create pg_conversion and support functions
 $ECHO_N "creating conversions... "$ECHO_C
-grep -v '^DROP CONVERSION' $datadir/conversion_create.sql | "$PGPATH"/postgres $PGSQL_OPT template1 > /dev/null || exit_nicely
+grep -v '^DROP CONVERSION' $datadir/conversion_create.sql | "$PGPATH"/postgres-7.3 $PGSQL_OPT template1 > /dev/null || exit_nicely
 echo "ok"
 
 # Set most system catalogs and built-in functions as world-accessible.
@@ -1039,12 +1039,12 @@
         WHERE NOT lanpltrusted;
 EOF
 ) \
-	| "$PGPATH"/postgres $PGSQL_OPT template1 > /dev/null || exit_nicely
+	| "$PGPATH"/postgres-7.3 $PGSQL_OPT template1 > /dev/null || exit_nicely
 echo "ok"
 
 $ECHO_N "vacuuming database template1... "$ECHO_C
 
-"$PGPATH"/postgres $PGSQL_OPT template1 >/dev/null <<EOF
+"$PGPATH"/postgres-7.3 $PGSQL_OPT template1 >/dev/null <<EOF
 ANALYZE;
 VACUUM FULL FREEZE;
 EOF
@@ -1055,7 +1055,7 @@
 
 $ECHO_N "copying template1 to template0... "$ECHO_C
 
-"$PGPATH"/postgres $PGSQL_OPT template1 >/dev/null <<EOF
+"$PGPATH"/postgres-7.3 $PGSQL_OPT template1 >/dev/null <<EOF
 CREATE DATABASE template0;
 
 UPDATE pg_database SET \
@@ -1091,11 +1091,8 @@
 echo
 echo "Success. You can now start the database server using:"
 echo ""
-echo "    $PGPATH/postmaster -D $PGDATA"
-echo "or"
-# (Advertise -l option here, otherwise we have a background
-#  process writing to the terminal.)
-echo "    $PGPATH/pg_ctl -D $PGDATA -l logfile start"
-echo
+echo "    sudo -u root sudo -u postgres $PGPATH/pg_ctl-7.3 \\"
+echo "        -D $PGDATA -l @INSTPREFIX@/var/postgresql/pgsql.log start"
+echo ""
 
 exit 0
diff -uNr postgresql-7.3.6/src/bin/initlocation/initlocation.sh postgresql-7.3.6-new/src/bin/initlocation/initlocation.sh
--- postgresql-7.3.6/src/bin/initlocation/initlocation.sh	Fri Oct 18 18:05:35 2002
+++ postgresql-7.3.6-new/src/bin/initlocation/initlocation.sh	Fri Jun 18 13:08:59 2004
@@ -162,7 +162,7 @@
     echo "You can now create a database using"
     echo "  CREATE DATABASE <name> WITH LOCATION = '$Location'"
     echo "in SQL, or"
-    echo "  createdb -D '$Location' <name>"
+    echo "  createdb-7.3 -D '$Location' <name>"
     echo "from the shell."
 fi
 echo
diff -uNr postgresql-7.3.6/src/bin/pg_ctl/pg_ctl.sh postgresql-7.3.6-new/src/bin/pg_ctl/pg_ctl.sh
--- postgresql-7.3.6/src/bin/pg_ctl/pg_ctl.sh	Fri Oct 18 18:05:35 2002
+++ postgresql-7.3.6-new/src/bin/pg_ctl/pg_ctl.sh	Fri Jun 18 13:08:59 2004
@@ -14,6 +14,10 @@
 
 CMDNAME=`basename $0`
 
+ulimit -d 24000 >/dev/null 2>&1
+ulimit -n 512   >/dev/null 2>&1
+ulimit -s 4096  >/dev/null 2>&1
+
 help="\
 $CMDNAME is a utility to start, stop, restart, reload configuration files,
 or report the status of a PostgreSQL server.
@@ -97,18 +101,18 @@
 fi
 
 # Check if needed programs actually exist in path
-if [ -x "$self_path/postmaster" ] && [ -x "$self_path/psql" ]; then
+if [ -x "$self_path/postmaster-7.3" ] && [ -x "$self_path/psql-7.3" ]; then
     PGPATH="$self_path"
-elif [ -x "$bindir/postmaster" ] && [ -x "$bindir/psql" ]; then
+elif [ -x "$bindir/postmaster-7.3" ] && [ -x "$bindir/psql-7.3" ]; then
     PGPATH="$bindir"
 else
-    echo "The programs 'postmaster' and 'psql' are needed by $CMDNAME but" 1>&2
+    echo "The programs 'postmaster-7.3' and 'psql-7.3' are needed by $CMDNAME but" 1>&2
     echo "were not found in the directory '$bindir'." 1>&2
     echo "Check your installation." 1>&2
     exit 1
 fi
 
-po_path="$PGPATH/postmaster"
+po_path="$PGPATH/postmaster-7.3"
 
 wait=
 wait_seconds=60
@@ -243,15 +247,15 @@
 	PID=`sed -n 1p $PIDFILE`
 	if [ "$PID" -lt 0 ];then
 	    PID=`expr 0 - $PID`
-	    echo "$CMDNAME: postgres is running (pid: $PID)"
+	    echo "$CMDNAME: postgres-7.3 is running (pid: $PID)"
 	else
-	    echo "$CMDNAME: postmaster is running (pid: $PID)"
+	    echo "$CMDNAME: postmaster-7.3 is running (pid: $PID)"
 	    echo "Command line was:"
 	    cat "$POSTOPTSFILE"
 	fi
 	exit 0
     else
-	echo "$CMDNAME: postmaster or postgres is not running"
+	echo "$CMDNAME: postmaster-7.3 or postgres-7.3 is not running"
 	exit 1
     fi
 fi
@@ -261,8 +265,8 @@
 	PID=`sed -n 1p $PIDFILE`
 	if [ "$PID" -lt 0 ];then
 	    PID=`expr 0 - $PID`
-	    echo "$CMDNAME: Cannot restart postmaster.  postgres is running (pid: $PID)" 1>&2
-	    echo "Please terminate postgres and try again." 1>&2
+	    echo "$CMDNAME: Cannot restart postmaster-7.3.  postgres-7.3 is running (pid: $PID)" 1>&2
+	    echo "Please terminate postgres-7.3 and try again." 1>&2
 	    exit 1
 	fi
 
@@ -271,7 +275,7 @@
 	# wait for postmaster to shut down
 	if [ "$wait" = yes -o "$op" = restart ];then
 	    cnt=0
-	    $silence_echo $ECHO_N "waiting for postmaster to shut down..."$ECHO_C
+	    $silence_echo $ECHO_N "waiting for postmaster-7.3 to shut down..."$ECHO_C
 
 	    while :
 	    do
@@ -280,7 +284,7 @@
 		    cnt=`expr $cnt + 1`
 		    if [ "$cnt" -gt "$wait_seconds" ];then
 			$silence_echo echo " failed"
-			echo "$CMDNAME: postmaster does not shut down" 1>&2
+			echo "$CMDNAME: postmaster-7.3 does not shut down" 1>&2
 			exit 1
 		    fi
 		else
@@ -292,16 +296,16 @@
 	fi
 
 	if [ "$op" = "reload" ];then
-	    $silence_echo echo "postmaster successfully signaled"
+	    $silence_echo echo "postmaster-7.3 successfully signaled"
 	else
-	    $silence_echo echo "postmaster successfully shut down"
+	    $silence_echo echo "postmaster-7.3 successfully shut down"
 	fi
 
     else # ! -f $PIDFILE
 	echo "$CMDNAME: cannot find $PIDFILE" 1>&2
-	echo "Is postmaster running?" 1>&2
+	echo "Is postmaster-7.3 running?" 1>&2
 	if [ "$op" = "restart" ];then
-	    echo "starting postmaster anyway" 1>&2
+	    echo "starting postmaster-7.3 anyway" 1>&2
 	else
 	    exit 1
 	fi
@@ -311,7 +315,7 @@
 if [ "$op" = "start" -o "$op" = "restart" ];then
     oldpid=""
     if [ -f "$PIDFILE" ];then
-	echo "$CMDNAME: Another postmaster may be running.  Trying to start postmaster anyway." 1>&2
+	echo "$CMDNAME: Another postmaster-7.3 may be running.  Trying to start postmaster-7.3 anyway." 1>&2
 	oldpid=`sed -n 1p $PIDFILE`
     fi
 
@@ -346,7 +350,7 @@
 	sleep 1
 	if [ -f "$PIDFILE" ];then
 	    if [ "`sed -n 1p $PIDFILE`" = "$oldpid" ];then
-		echo "$CMDNAME: cannot start postmaster" 1>&2
+		echo "$CMDNAME: cannot start postmaster-7.3" 1>&2
 		echo "Examine the log output." 1>&2
 		exit 1
 	    fi
@@ -356,7 +360,7 @@
     # wait for postmaster to start
     if [ "$wait" = yes ];then
 	cnt=0
-	$silence_echo $ECHO_N "waiting for postmaster to start..."$ECHO_C
+	$silence_echo $ECHO_N "waiting for postmaster-7.3 to start..."$ECHO_C
 	while :
 	do
 # FIXME:  This is horribly misconceived.
@@ -371,7 +375,7 @@
 # 6) If the dynamic loader is not set up correctly (for this user/at
 #    this time), psql will fail (to find libpq).
 # 7) If psql is misconfigured, this may fail.
-	    if "$PGPATH/psql" -l >/dev/null 2>&1
+	    if "$PGPATH/psql-7.3" -l >/dev/null 2>&1
 	    then
 		break;
 	    else
@@ -379,7 +383,7 @@
 		cnt=`expr $cnt + 1`
 		if [ "$cnt" -gt "$wait_seconds" ];then
 		    $silence_echo echo "failed"
-		    echo "$CMDNAME: postmaster does not start" 1>&2
+		    echo "$CMDNAME: postmaster-7.3 does not start" 1>&2
 		    exit 1
 		fi
 		sleep 1
@@ -387,7 +391,7 @@
 	done
 	$silence_echo echo "done"
     fi
-    $silence_echo echo "postmaster successfully started"
+    $silence_echo echo "postmaster-7.3 successfully started"
 fi # start or restart
 
 exit 0
diff -uNr postgresql-7.3.6/src/bin/scripts/clusterdb postgresql-7.3.6-new/src/bin/scripts/clusterdb
--- postgresql-7.3.6/src/bin/scripts/clusterdb	Mon Sep 29 14:53:08 2003
+++ postgresql-7.3.6-new/src/bin/scripts/clusterdb	Fri Jun 18 13:09:38 2004
@@ -140,7 +140,7 @@
 		echo "$CMDNAME: cannot cluster all databases and a specific one at the same time" 1>&2
 		exit 1
 	fi
-	dbname=`${PATHNAME}psql $PSQLOPT -q -t -A -d template1 -c 'SELECT datname FROM pg_database WHERE datallowconn'`
+	dbname=`${PATHNAME}psql-7.3 $PSQLOPT -q -t -A -d template1 -c 'SELECT datname FROM pg_database WHERE datallowconn'`
 	[ "$?" -ne 0 ] && exit 1
 
 elif [ -z "$dbname" ]; then
@@ -149,7 +149,7 @@
 	elif [ "$PGUSER" ]; then
 		dbname="$PGUSER"
 	else
-		dbname=`${PATHNAME}pg_id -u -n`
+		dbname=`${PATHNAME}pg_id-7.3 -u -n`
 	fi
 	[ "$?" -ne 0 ] && exit 1
 fi
@@ -161,7 +161,7 @@
 	query="SELECT nspname, pg_class.relname, pg_class_2.relname FROM pg_class, pg_class AS pg_class_2 JOIN pg_namespace ON (pg_namespace.oid=relnamespace), pg_index WHERE pg_class.oid=pg_index.indrelid AND pg_class_2.oid=pg_index.indexrelid AND pg_index.indisclustered AND pg_class.relowner=(SELECT usesysid FROM pg_user WHERE usename=current_user)"
 
 	if [ -z "$table" ]; then
-		tables=`${PATHNAME}psql $PSQLOPT -F: -P format=unaligned -t -c "$query" $db`
+		tables=`${PATHNAME}psql-7.3 $PSQLOPT -F: -P format=unaligned -t -c "$query" $db`
 		[ "$?" -ne 0 ] && exit 1
 	else
 		# if tablename has a dot, use it as namespace separator
@@ -169,10 +169,10 @@
 		then
 			tbl=`echo $table | cut -d. -f2`
 			nspc=`echo $table | cut -d. -f1`
-			tables=`${PATHNAME}psql $PSQLOPT -F: -P format=unaligned -t -c "$query AND pg_class.relname='$tbl' AND nspname='$nspc'" $db`
+			tables=`${PATHNAME}psql-7.3 $PSQLOPT -F: -P format=unaligned -t -c "$query AND pg_class.relname='$tbl' AND nspname='$nspc'" $db`
 			echo $tables
 		else
-			tables=`${PATHNAME}psql $PSQLOPT -F: -P format=unaligned -t -c "$query AND pg_class.relname='$table'" $db`
+			tables=`${PATHNAME}psql-7.3 $PSQLOPT -F: -P format=unaligned -t -c "$query AND pg_class.relname='$table'" $db`
 		fi
 	fi
 	query=
@@ -183,7 +183,7 @@
 		idx=`echo $tabs | cut -d: -f3`
 		query="$query CLUSTER $idx ON $nspc.$tab;"
 	done
-	${PATHNAME}psql $PSQLOPT $ECHOOPT -c "SET autocommit TO 'on';$query" -d $db
+	${PATHNAME}psql-7.3 $PSQLOPT $ECHOOPT -c "SET autocommit TO 'on';$query" -d $db
 	if [ "$?" -ne 0 ]
 	then
 		echo "$CMDNAME: While clustering $db, the following failed: $query" 1>&2
diff -uNr postgresql-7.3.6/src/bin/scripts/createdb postgresql-7.3.6-new/src/bin/scripts/createdb
--- postgresql-7.3.6/src/bin/scripts/createdb	Fri Oct 18 18:05:36 2002
+++ postgresql-7.3.6-new/src/bin/scripts/createdb	Fri Jun 18 13:08:59 2004
@@ -160,7 +160,7 @@
 
 if [ -n "$MB" ]
 then
-        mbcode=`${PATHNAME}pg_encoding "$MB"`
+        mbcode=`${PATHNAME}pg_encoding-7.3 "$MB"`
         if [ -z "$mbcode" ]
 	then
 		echo "$CMDNAME: \"$MB\" is not a valid encoding name" 1>&2
@@ -174,7 +174,7 @@
         elif [ "$PGUSER" ]; then
                 dbname="$PGUSER"
         else
-                dbname=`${PATHNAME}pg_id -u -n`
+                dbname=`${PATHNAME}pg_id-7.3 -u -n`
         fi
         [ "$?" -ne 0 ] && exit 1
 fi
@@ -192,7 +192,7 @@
 [ "$TEMPLATE" ] &&   withstring="$withstring TEMPLATE = \"$TEMPLATE\""
 [ "$withstring" ] && withstring=" WITH$withstring"
 
-${PATHNAME}psql $PSQLOPT -d template1 -c "SET autocommit TO 'on';CREATE DATABASE \"$dbname\"$withstring"
+${PATHNAME}psql-7.3 $PSQLOPT -d template1 -c "SET autocommit TO 'on';CREATE DATABASE \"$dbname\"$withstring"
 if [ "$?" -ne 0 ]; then
 	echo "$CMDNAME: database creation failed" 1>&2
 	exit 1
@@ -203,7 +203,7 @@
 
 dbcomment=`echo "$dbcomment" | sed "s/'/\\\\\'/g"`
 
-${PATHNAME}psql $PSQLOPT -d "$dbname" -c "SET autocommit TO 'on';COMMENT ON DATABASE \"$dbname\" IS '$dbcomment'"
+${PATHNAME}psql-7.3 $PSQLOPT -d "$dbname" -c "SET autocommit TO 'on';COMMENT ON DATABASE \"$dbname\" IS '$dbcomment'"
 if [ "$?" -ne 0 ]; then
 	echo "$CMDNAME: comment creation failed (database was created)" 1>&2
 	exit 1
diff -uNr postgresql-7.3.6/src/bin/scripts/createlang.sh postgresql-7.3.6-new/src/bin/scripts/createlang.sh
--- postgresql-7.3.6/src/bin/scripts/createlang.sh	Sat Apr 26 11:19:05 2003
+++ postgresql-7.3.6-new/src/bin/scripts/createlang.sh	Fri Jun 18 13:08:59 2004
@@ -153,7 +153,7 @@
         elif [ "$PGUSER" ]; then
                 dbname="$PGUSER"
         else
-                dbname=`${PATHNAME}pg_id -u -n`
+                dbname=`${PATHNAME}pg_id-7.3 -u -n`
         fi
         [ "$?" -ne 0 ] && exit 1
 fi
@@ -167,7 +167,7 @@
 	if [ "$showsql" = yes ]; then
 		echo "$sqlcmd"
 	fi
-        ${PATHNAME}psql $PSQLOPT -d "$dbname" -P 'title=Procedural languages' -c "$sqlcmd"
+        ${PATHNAME}psql-7.3 $PSQLOPT -d "$dbname" -P 'title=Procedural languages' -c "$sqlcmd"
         exit $?
 fi
 
@@ -233,7 +233,7 @@
 esac
 
 
-PSQL="${PATHNAME}psql -A -t -q $PSQLOPT -d $dbname -c"
+PSQL="${PATHNAME}psql-7.3 -A -t -q $PSQLOPT -d $dbname -c"
 
 # ----------
 # Make sure the language isn't already installed
diff -uNr postgresql-7.3.6/src/bin/scripts/createuser postgresql-7.3.6-new/src/bin/scripts/createuser
--- postgresql-7.3.6/src/bin/scripts/createuser	Fri May 16 11:17:02 2003
+++ postgresql-7.3.6-new/src/bin/scripts/createuser	Fri Jun 18 13:08:59 2004
@@ -256,7 +256,7 @@
 [ "$CanAddUser" = t ] &&  QUERY="$QUERY CREATEUSER"
 [ "$CanAddUser" = f ] &&  QUERY="$QUERY NOCREATEUSER"
 
-${PATHNAME}psql -c "SET autocommit TO 'on';$QUERY" -d template1 $PSQLOPT
+${PATHNAME}psql-7.3 -c "SET autocommit TO 'on';$QUERY" -d template1 $PSQLOPT
 if [ "$?" -ne 0 ]; then
 	echo "$CMDNAME: creation of user \"$NewUser\" failed" 1>&2
 	exit 1
diff -uNr postgresql-7.3.6/src/bin/scripts/dropdb postgresql-7.3.6-new/src/bin/scripts/dropdb
--- postgresql-7.3.6/src/bin/scripts/dropdb	Fri Oct 18 18:05:36 2002
+++ postgresql-7.3.6-new/src/bin/scripts/dropdb	Fri Jun 18 13:08:59 2004
@@ -139,7 +139,7 @@
 
 dbname=`echo "$dbname" | sed 's/\"/\\\"/g'`
 
-${PATHNAME}psql $PSQLOPT -d template1 -c "SET autocommit TO 'on';DROP DATABASE \"$dbname\""
+${PATHNAME}psql-7.3 $PSQLOPT -d template1 -c "SET autocommit TO 'on';DROP DATABASE \"$dbname\""
 if [ "$?" -ne 0 ]; then
 	echo "$CMDNAME: database removal failed" 1>&2
 	exit 1
diff -uNr postgresql-7.3.6/src/bin/scripts/droplang postgresql-7.3.6-new/src/bin/scripts/droplang
--- postgresql-7.3.6/src/bin/scripts/droplang	Fri Oct 18 18:05:36 2002
+++ postgresql-7.3.6-new/src/bin/scripts/droplang	Fri Jun 18 13:08:59 2004
@@ -142,7 +142,7 @@
         elif [ "$PGUSER" ]; then
                 dbname="$PGUSER"
         else
-                dbname=`${PATHNAME}pg_id -u -n`
+                dbname=`${PATHNAME}pg_id-7.3 -u -n`
         fi
         [ "$?" -ne 0 ] && exit 1
 fi
@@ -156,7 +156,7 @@
 	if [ "$showsql" = yes ]; then
 		echo "$sqlcmd"
 	fi
-        ${PATHNAME}psql $PSQLOPT -d "$dbname" -P 'title=Procedural languages' -c "$sqlcmd"
+        ${PATHNAME}psql-7.3 $PSQLOPT -d "$dbname" -P 'title=Procedural languages' -c "$sqlcmd"
         exit $?
 fi
 
@@ -170,7 +170,7 @@
 	exit 1
 fi
 
-PSQL="${PATHNAME}psql -A -t -q $PSQLOPT -d $dbname -c"
+PSQL="${PATHNAME}psql-7.3 -A -t -q $PSQLOPT -d $dbname -c"
 
 
 # ----------
diff -uNr postgresql-7.3.6/src/bin/scripts/dropuser postgresql-7.3.6-new/src/bin/scripts/dropuser
--- postgresql-7.3.6/src/bin/scripts/dropuser	Fri Oct 18 18:05:36 2002
+++ postgresql-7.3.6-new/src/bin/scripts/dropuser	Fri Jun 18 13:08:59 2004
@@ -149,7 +149,7 @@
 
 DelUser=`echo "$DelUser" | sed 's/\"/\\\"/g'`
 
-${PATHNAME}psql $PSQLOPT -d template1 -c "SET autocommit TO 'on';DROP USER \"$DelUser\""
+${PATHNAME}psql-7.3 $PSQLOPT -d template1 -c "SET autocommit TO 'on';DROP USER \"$DelUser\""
 
 if [ "$?" -ne 0 ]; then
 	echo "$CMDNAME: deletion of user \"$DelUser\" failed" 1>&2
diff -uNr postgresql-7.3.6/src/bin/scripts/vacuumdb postgresql-7.3.6-new/src/bin/scripts/vacuumdb
--- postgresql-7.3.6/src/bin/scripts/vacuumdb	Fri Oct 18 18:05:36 2002
+++ postgresql-7.3.6-new/src/bin/scripts/vacuumdb	Fri Jun 18 13:08:59 2004
@@ -156,7 +156,7 @@
                 echo "$CMDNAME: cannot vacuum all databases and a specific one at the same time" 1>&2
                 exit 1
         fi
-	dbname=`${PATHNAME}psql $PSQLOPT -q -t -A -d template1 -c 'SELECT datname FROM pg_database WHERE datallowconn'`
+	dbname=`${PATHNAME}psql-7.3 $PSQLOPT -q -t -A -d template1 -c 'SELECT datname FROM pg_database WHERE datallowconn'`
 
 elif [ -z "$dbname" ]; then
         if [ "$PGDATABASE" ]; then
@@ -164,7 +164,7 @@
         elif [ "$PGUSER" ]; then
                 dbname="$PGUSER"
         else
-                dbname=`${PATHNAME}pg_id -u -n`
+                dbname=`${PATHNAME}pg_id-7.3 -u -n`
         fi
         [ "$?" -ne 0 ] && exit 1
 fi
@@ -172,7 +172,7 @@
 for db in $dbname
 do
         [ "$alldb" -a "$quiet" -ne 1 ] && echo "Vacuuming $db"
-	${PATHNAME}psql $PSQLOPT $ECHOOPT -c "SET autocommit TO 'on';VACUUM $full $verbose $analyze $table" -d $db
+	${PATHNAME}psql-7.3 $PSQLOPT $ECHOOPT -c "SET autocommit TO 'on';VACUUM $full $verbose $analyze $table" -d $db
 	if [ "$?" -ne 0 ]; then
 	    echo "$CMDNAME: vacuum $table $db failed" 1>&2
 	    exit 1
diff -uNr postgresql-7.3.6/src/include/miscadmin.h postgresql-7.3.6-new/src/include/miscadmin.h
--- postgresql-7.3.6/src/include/miscadmin.h	Thu Oct  3 13:07:53 2002
+++ postgresql-7.3.6-new/src/include/miscadmin.h	Fri Jun 18 13:08:59 2004
@@ -182,6 +182,7 @@
 extern char *Unix_socket_group;
 extern char *UnixSocketDir;
 extern char *VirtualHost;
+extern char *service_name;
 
 
 /*****************************************************************************
diff -uNr postgresql-7.3.6/src/include/pg_config.h.in postgresql-7.3.6-new/src/include/pg_config.h.in
--- postgresql-7.3.6/src/include/pg_config.h.in	Fri Nov  8 00:23:09 2002
+++ postgresql-7.3.6-new/src/include/pg_config.h.in	Fri Jun 18 13:08:59 2004
@@ -57,6 +57,9 @@
 /* Define to build with PAM Support */
 #undef USE_PAM
 
+/* Define if pam_appl.h is in pam/ instead of security/ */
+#undef HAVE_PAM_PAM_APPL_H
+
 /* 
  * DEF_PGPORT is the TCP port number on which the Postmaster listens and
  * which clients will try to connect to.  This is just a default value;
diff -uNr postgresql-7.3.6/src/include/port/darwin.h postgresql-7.3.6-new/src/include/port/darwin.h
--- postgresql-7.3.6/src/include/port/darwin.h	Sun Oct 28 01:26:08 2001
+++ postgresql-7.3.6-new/src/include/port/darwin.h	Fri Jun 18 13:08:59 2004
@@ -1,5 +1,7 @@
 #define __darwin__	1
 
+#define HAVE_RENDEZVOUS
+
 #if defined(__ppc__)
 #define HAS_TEST_AND_SET
 #endif
diff -uNr postgresql-7.3.6/src/interfaces/libpgtcl/Makefile postgresql-7.3.6-new/src/interfaces/libpgtcl/Makefile
--- postgresql-7.3.6/src/interfaces/libpgtcl/Makefile	Tue Dec 10 23:08:05 2002
+++ postgresql-7.3.6-new/src/interfaces/libpgtcl/Makefile	Fri Jun 18 13:08:59 2004
@@ -20,7 +20,7 @@
 
 OBJS= pgtcl.o pgtclCmds.o pgtclId.o
 
-SHLIB_LINK = $(libpq)
+SHLIB_LINK = $(libpq) $(TCL_LIB_SPEC)
 
 # If crypt is a separate library, rather than part of libc, it may need
 # to be referenced separately to keep (broken) linkers happy.  (This is
diff -uNr postgresql-7.3.6/src/interfaces/python/GNUmakefile postgresql-7.3.6-new/src/interfaces/python/GNUmakefile
--- postgresql-7.3.6/src/interfaces/python/GNUmakefile	Thu Dec 13 13:39:04 2001
+++ postgresql-7.3.6-new/src/interfaces/python/GNUmakefile	Fri Jun 18 13:08:59 2004
@@ -1,5 +1,7 @@
 # $Header: /cvsroot/pgsql-server/src/interfaces/python/Attic/GNUmakefile,v 1.13 2001/12/13 18:39:04 petere Exp $
 
+DLTYPE = bundle
+
 subdir = src/interfaces/python
 top_builddir = ../../..
 include $(top_builddir)/src/Makefile.global
@@ -13,7 +15,9 @@
 override CPPFLAGS += -DUSE_DL_IMPORT
 SHLIB_LINK += $(python_libspec)
 endif
-
+ifeq ($(PORTNAME), darwin)
+SHLIB_LINK += -bundle_loader @INSTPREFIX@/bin/python2.1
+endif
 
 include $(top_srcdir)/src/Makefile.shlib
 
diff -uNr postgresql-7.3.6/src/makefiles/Makefile.darwin postgresql-7.3.6-new/src/makefiles/Makefile.darwin
--- postgresql-7.3.6/src/makefiles/Makefile.darwin	Fri Oct  5 17:15:38 2001
+++ postgresql-7.3.6-new/src/makefiles/Makefile.darwin	Fri Jun 18 13:08:59 2004
@@ -1,13 +1,23 @@
 AROPT = cr
 AWK= awk
  
-DLSUFFIX = .so
+ifeq ($(DLTYPE), bundle)
+	DLSUFFIX := .so
+else
+	DLSUFFIX := .dylib
+endif
+
 CFLAGS_SL =
+CFLAGS += -fno-common
 ifeq (,$(filter $(host_os), darwin1.0 darwin1.1 darwin1.2))
-DARWIN_NAMESPACE_SPEC = -flat_namespace
+  MULTIPLY_DEFINED=
+else
+  MULTIPLY_DEFINED=-multiply_defined suppress
 endif
  
 %.so: %.o
-	$(CC) $(DARWIN_NAMESPACE_SPEC) -bundle -undefined suppress -o $@ $<
+	$(CC) -bundle $(MULTIPLY_DEFINED) -o $@ $<
+%.dylib: %.o
+	$(CC) -dynamiclib $(MULTIPLY_DEFINED) -o $@ $<
 
 sqlmansect = 7
diff -uNr postgresql-7.3.6/src/pl/plperl/GNUmakefile postgresql-7.3.6-new/src/pl/plperl/GNUmakefile
--- postgresql-7.3.6/src/pl/plperl/GNUmakefile	Tue May 28 12:57:53 2002
+++ postgresql-7.3.6-new/src/pl/plperl/GNUmakefile	Fri Jun 18 13:08:59 2004
@@ -20,10 +20,10 @@
 
 override CPPFLAGS := -I$(srcdir) -I$(perl_archlibexp)/CORE $(CPPFLAGS)
 
-
 NAME = plperl
 SO_MAJOR_VERSION = 0
 SO_MINOR_VERSION = 0
+DLTYPE = bundle
 
 OBJS = plperl.o eloglvl.o SPI.o
 SHLIB_LINK = $(perl_embed_ldflags) $(BE_DLLLIBS)
diff -uNr postgresql-7.3.6/src/pl/plpgsql/src/Makefile postgresql-7.3.6-new/src/pl/plpgsql/src/Makefile
--- postgresql-7.3.6/src/pl/plpgsql/src/Makefile	Fri Nov  1 17:52:34 2002
+++ postgresql-7.3.6-new/src/pl/plpgsql/src/Makefile	Fri Jun 18 13:08:59 2004
@@ -6,6 +6,8 @@
 #
 #-------------------------------------------------------------------------
 
+DLTYPE = bundle
+
 subdir = src/pl/plpgsql/src
 top_builddir = ../../../..
 include $(top_builddir)/src/Makefile.global
diff -uNr postgresql-7.3.6/src/pl/plpython/Makefile postgresql-7.3.6-new/src/pl/plpython/Makefile
--- postgresql-7.3.6/src/pl/plpython/Makefile	Thu Sep  5 14:28:46 2002
+++ postgresql-7.3.6-new/src/pl/plpython/Makefile	Fri Jun 18 13:08:59 2004
@@ -1,5 +1,7 @@
 # $Header: /cvsroot/pgsql-server/src/pl/plpython/Makefile,v 1.11 2002/09/05 18:28:46 petere Exp $
 
+DLTYPE = bundle
+
 subdir = src/pl/plpython
 top_builddir = ../../..
 include $(top_builddir)/src/Makefile.global
@@ -8,7 +10,7 @@
 # On some platforms we can only build PL/Python if libpython is a
 # shared library.  Since there is no official way to determine this,
 # we see if there is a file that is named like a shared library.
-ifneq (,$(wildcard $(python_configdir)/libpython*$(DLSUFFIX)*))
+ifneq (,$(wildcard $(python_configdir)/libpython*.a*))
 shared_libpython = yes
 endif
 
@@ -24,7 +26,7 @@
 SO_MINOR_VERSION = 0
 OBJS = plpython.o
 
-SHLIB_LINK = $(BE_DLLLIBS) $(python_libspec)
+SHLIB_LINK = $(BE_DLLLIBS) -bundle_loader @INSTPREFIX@/bin/python2.1
 
 include $(top_srcdir)/src/Makefile.shlib
 
diff -uNr postgresql-7.3.6/src/pl/tcl/Makefile postgresql-7.3.6-new/src/pl/tcl/Makefile
--- postgresql-7.3.6/src/pl/tcl/Makefile	Fri May 24 14:10:17 2002
+++ postgresql-7.3.6-new/src/pl/tcl/Makefile	Fri Jun 18 13:08:59 2004
@@ -25,7 +25,6 @@
 endif
 endif
 
-
 # The following attempts to figure out what libraries need to be
 # linked with pltcl.  The information comes from the tclConfig.sh
 # file, but it's mostly bogus.  This just might work.
@@ -48,6 +47,7 @@
 NAME = pltcl
 SO_MAJOR_VERSION = 2
 SO_MINOR_VERSION = 0
+DLTYPE = bundle
 OBJS = pltcl.o
 
 include $(top_srcdir)/src/Makefile.shlib
diff -uNr postgresql-7.3.6/src/template/darwin postgresql-7.3.6-new/src/template/darwin
--- postgresql-7.3.6/src/template/darwin	Sat May  4 20:03:29 2002
+++ postgresql-7.3.6-new/src/template/darwin	Fri Jun 18 13:08:59 2004
@@ -1,6 +1,6 @@
 # -traditional-cpp means "don't use apple's cpp-precomp" on darwin
 # this should change to -no-cpp-precomp when that flag is implemented
-CC="$CC -traditional-cpp"
+CC="$CC -no-cpp-precomp"
 
 # Select appropriate semaphore support
 USE_NAMED_POSIX_SEMAPHORES=1
