diff -ru sitecopy-0.12.1/configure sitecopy-0.12.1-patched/configure
--- sitecopy-0.12.1/configure	2003-02-19 22:52:30.000000000 +0100
+++ sitecopy-0.12.1-patched/configure	2003-03-06 18:12:25.000000000 +0100
@@ -9486,7 +9486,7 @@
 
 
     CFLAGS="$CFLAGS -I$neon_bundled_srcdir"
-    NEON_LIBS="-L$neon_bundled_builddir -lneon $NEON_LIBS"
+    NEON_LIBS="$neon_bundled_builddir/libneon.a $NEON_LIBS"
     NEON_NEED_XML_PARSER=yes
     neon_library_message="included libneon (${NEON_VERSION})"
 else
