diff -ru xchat-2.0.4/configure xchat-2.0.4-patched/configure
--- xchat-2.0.4/configure	2003-07-31 07:24:30.000000000 +0200
+++ xchat-2.0.4-patched/configure	2003-08-27 19:26:27.000000000 +0200
@@ -23276,7 +23276,7 @@
 echo "${ECHO_T}ok" >&6
 
 		oldLIBS=$LIBS
-		LIBS="$LIBS $PERL_LDFLAGS"
+		LIBS="$PERL_LDFLAGS $LIBS"
 
 for ac_func in eval_pv
 do
@@ -23468,7 +23468,6 @@
                 echo "$as_me:$LINENO: checking Python version" >&5
 echo $ECHO_N "checking Python version... $ECHO_C" >&6
                                 PY_VERSION=`$pythonpath -c 'import sys; print sys.version[:3]'`
-                PY_PATH=`$pythonpath -c 'import sys ; print sys.path[1][5:]'`
                 $pythonpath -c "import sys; map(int,sys.version[:3].split('.')) >= [2,2] or sys.exit(1)"
                                 echo "$as_me:$LINENO: result: $PY_VERSION" >&5
 echo "${ECHO_T}$PY_VERSION" >&6
@@ -23478,7 +23477,7 @@
                         PY_PREFIX=`$pythonpath -c 'import sys; print sys.prefix'`
                         PY_EXEC_PREFIX=`$pythonpath -c 'import sys; print sys.exec_prefix'`
                         if test -f $PY_PREFIX/include/python$PY_VERSION/Python.h; then
-                                PY_LIBS="-L$PY_EXEC_PREFIX/$PY_PATH/config -lpython$PY_VERSION -lpthread -lutil"
+                                PY_LIBS="-L$PY_PREFIX/lib/python$PY_VERSION/config -lpython$PY_VERSION"
                                 PY_CFLAGS="-I$PY_PREFIX/include/python$PY_VERSION"
                                 echo "$as_me:$LINENO: result: ok" >&5
 echo "${ECHO_T}ok" >&6
diff -ru xchat-2.0.4/ltmain.sh xchat-2.0.4-patched/ltmain.sh
--- xchat-2.0.4/ltmain.sh	2003-06-12 07:09:04.000000000 +0200
+++ xchat-2.0.4-patched/ltmain.sh	2003-08-27 19:25:39.000000000 +0200
@@ -1918,20 +1918,7 @@
 	  fi
 	  case $linkmode in
 	  lib)
-	    if test "$deplibs_check_method" != pass_all; then
-	      $echo
-	      $echo "*** Warning: Trying to link with static lib archive $deplib."
-	      $echo "*** I have the capability to make that library automatically link in when"
-	      $echo "*** you link to this library.  But I can only do this if you have a"
-	      $echo "*** shared version of the library, which you do not appear to have"
-	      $echo "*** because the file extensions .$libext of this argument makes me believe"
-	      $echo "*** that it is just a static archive that I should not used here."
-	    else
-	      $echo
-	      $echo "*** Warning: Linking the shared library $output against the"
-	      $echo "*** static library $deplib is not portable!"
-	      deplibs="$deplib $deplibs"
-	    fi
+	    deplibs="$deplib $deplibs"
 	    continue
 	    ;;
 	  prog)
