--- sv7/mpp.h.orig	Sun Dec 19 23:14:56 2004
+++ sv7/mpp.h	Sun Dec 19 23:16:03 2004
@@ -43,6 +43,7 @@
 #define USE_ESD_AUDIO
 
 #endif
+#define USE_ESD_AUDIO
 
 // native Sun Onboard-Audio support (only SunOS)
 // If you have a Sun Workstation with Onboard-Audio, you can output to /dev/audio and
@@ -54,12 +55,12 @@
 // Sound support for SGI Irix
 // If you have a SGI Workstation running IRIX, you can output to /dev/audio and
 // instead of writing a file the program plays the file via this sound device.
-#define USE_IRIX_AUDIO
+//#define USE_IRIX_AUDIO
 
 // Audio support for Windows (WAVE OUT) (only Windows)
 // If you have a Windows based system and if you want also play files directly instead of only writing audio files,
 // then define the next item
-#define USE_WIN_AUDIO
+//#define USE_WIN_AUDIO
 
 // Buffersize for Windows Audio in 4.5 KByte units
 // Only needed for Windows+USE_WIN_AUDIO
