--- mppdec-1.1/mppdec.h	2002-07-30 15:19:57.000000000 -0700
+++ mppdec-1.1/mppdec.h	2003-05-28 17:56:37.000000000 -0700
@@ -108,6 +108,11 @@
 # undef USE_REALTIME
 #endif
 
+#if defined __APPLE__
+# undef USE_SUN_AUDIO
+# undef USE_REALTIME
+#endif
+
 #if defined USE_DIET  ||  defined MAKE_24BIT  ||  defined MAKE_32BIT
 # undef USE_ESD_AUDIO
 #endif
@@ -963,10 +968,12 @@
 #define LITTLE                  0
 #define BIG                     1
 extern Bool_t                   output_endianess;
-#if   defined HAVE_LITTLE_ENDIAN
+#if   ENDIAN == HAVE_LITTLE_ENDIAN
 # define machine_endianess      LITTLE
-#elif defined HAVE_BIG_ENDIAN
+#elif ENDIAN == HAVE_BIG_ENDIAN
 # define machine_endianess      BIG
+#else
+#error nothing is defined
 #endif
 
 // requant.c

--- mppdec-1.1/mpp.h.orig	2003-04-02 21:27:10.000000000 -0800
+++ mppdec-1.1/mpp.h	2003-04-02 21:48:37.000000000 -0800
@@ -8,13 +8,13 @@
 // use optimized assembler routines for Pentium III/K6-2/Athlon (only 32 bit OS, Intel x86 and no MAKE_xxBITS)
 // you need the NASM assembler on your system, the program becomes a little bit larger and decoding
 // on AMD K6-2 (x3), AMD K6-III (x3), AMD Duron (x1.7), AMD Athlon (x1.7), Pentium III (x2) and Pentium 4 (x1.8) becomes faster
-#define USE_ASM
+//#define USE_ASM 0
 
 // Open Sound System support (only Unix with OSS support)
 // If your Operating System supports the Open Sound System, you can output to /dev/dsp* and
 // instead of writing a file the program plays the file via this sound device.
 // on some systems you also must link the libossaudio library, so maybe you also must edit the Makefile
-#define USE_OSS_AUDIO
+//#define USE_OSS_AUDIO 0
 
 // Enlightenment Sound Daemon support (only Unix with ESD support)
 // If your Operating System supports the Enlightenment Sound Daemon you can output to /dev/esd and
--- mppdec-1.1/Makefile.nol.orig	2003-04-02 21:30:52.000000000 -0800
+++ mppdec-1.1/Makefile.nol	2003-04-02 21:46:45.000000000 -0800
@@ -14,7 +14,7 @@
 
 WARN     =
 
-CFLAGS   = -DMPP_DECODER $(OPTIM) $(WARN) -DMPPDEC_VERSION=\"$(MPPDEC_VERSION)\"
+CFLAGS   = -DMPP_DECODER $(OPTIM) $(WARN) -DMPPDEC_VERSION=\"$(MPPDEC_VERSION)\" $(EXTRA_CFLAGS)
 
 TARGETS  = mppdec
 
@@ -40,7 +40,6 @@
 	_setargv.o       \
 	dump.o           \
 	gain_analysis.o  \
-	http.o           \
 	pipeopen.o       \
 	stderr.o         \
 
@@ -49,10 +48,9 @@
 
 mppdec: $(OBJ)
 	$(CC)         $(CFLAGS) $(OBJ) -o $(TARGETS)        $(LDADD)
-	$(CC) -static $(CFLAGS) $(OBJ) -o $(TARGETS)-static $(LDADD)
 
 config.h: mpp.h config.c
-	$(CC)         $(OPTIM)    -DMPP_DECODER     -o config config.c   $(LDADD)
+	$(CC)         $(CFLAGS) $(OPTIM)    -DMPP_DECODER     -o config config.c   $(LDADD)
 	@./config "$(CC) $(OPTIM) -o <<EXE>> <<SRC>> $(LDADD)" "./<<EXE>>"
 	@$(RM_F) config
 
