--- portaudio/configure	2009-12-10 02:37:02.000000000 +0100
+++ portaudio-patched/configure	2010-02-01 20:09:48.000000000 +0100
@@ -21448,6 +21448,7 @@
               mac_sysroot="-isysroot /Developer/SDKs/MacOSX10.4u.sdk"
            fi
         else
+           mac_version_min="-mmacosx-version-min=10.4"
            mac_arches=""
            mac_sysroot=""
            mac_version=""
--- portaudio/src/hostapi/coreaudio/pa_mac_core.c	2008-03-11 16:20:38.000000000 +0100
+++ portaudio-patched/src/hostapi/coreaudio/pa_mac_core.c	2009-10-21 20:52:13.000000000 +0200
@@ -71,6 +71,7 @@
 #include "pa_mac_core_utilities.h"
 #include "pa_mac_core_blocking.h"
 
+#include <CoreServices/CoreServices.h>
 
 #ifdef __cplusplus
 extern "C"
