diff -Nurd -x'*~' sphinxbase-0.3.orig/configure sphinxbase-0.3/configure
--- sphinxbase-0.3.orig/configure	2007-08-08 15:34:44.000000000 -0400
+++ sphinxbase-0.3/configure	2012-02-14 03:58:28.000000000 -0500
@@ -7239,7 +7239,7 @@
                         ad_files="ad_portaudio.lo pa_convert.lo pa_lib.lo \
 			          pa_mac_core.lo pa_trace.lo pablio.lo ringbuffer.lo"
                         ad_backend="AD_BACKEND_PORTAUDIO"
-			ad_libs="-framework CoreAudio -framework AudioToolbox"
+			ad_libs="-Wl,-framework,CoreAudio -Wl,-framework,AudioToolbox"
 
 cat >>confdefs.h <<\_ACEOF
 #define AD_BACKEND_PORTAUDIO
