--- ./config.makefile.orig	Fri Aug 15 09:09:51 2003
+++ ./config.makefile	Fri Aug 15 09:17:07 2003
@@ -13,9 +13,9 @@
 # SEER = 1
 
 #********* LINUX / gcc g++ / static ********
-MACH = LINUX# Linus g++  and gcc
-LINUX = 1
-ECGS = 1# for newer Linux-versions
+# MACH = LINUX# Linus g++  and gcc
+# LINUX = 1
+# ECGS = 1# for newer Linux-versions
 
 # export DEBIAN = 1
 
@@ -24,8 +24,8 @@
 
 #********* Mac OSX ********
 
-# MACH   = DARWIN#
-# DARWIN = 1
+ MACH   = DARWIN#
+ DARWIN = 1
 
 #********* SUN4 / acc CC / dynamic ********
 
--- AWT/AWT_input_mask.cxx.orig	Fri Aug 15 08:29:32 2003
+++ AWT/AWT_input_mask.cxx	Fri Aug 15 08:30:32 2003
@@ -692,7 +692,10 @@
 
     for (; b != buttons.end() && v != values.end(); ++b, ++v, ++pos) {
         void (AW_window::*ins_togg)(AW_label, const char*, const char*);
-        ins_togg = (pos == default_position) ? &AW_window::insert_default_toggle : &AW_window::insert_toggle;
+        if(pos == default_position)
+           ins_togg = &AW_window::insert_default_toggle;
+        else
+           ins_togg = &AW_window::insert_toggle;
 
         (aws->*ins_togg)(b->c_str(), mask_global()->hotkey(*b), b->c_str());
     }
--- ./Makefile~	Wed Aug 13 21:52:41 2003
+++ ./Makefile	Fri Aug 15 08:43:40 2003
@@ -848,7 +848,7 @@
 # MOTIF_LIBPATH
 
 ifndef MOTIF_LIBNAME
-MOTIF_LIBNAME=libXm.so.2
+MOTIF_LIBNAME=libXm.so.3
 endif
 ifndef MOTIF_LIBPATH
 MOTIF_LIBPATH=Motif/$(MOTIF_LIBNAME)
@@ -1197,7 +1197,7 @@
 arbshared: dball aw dp awt
 arbapplications: nt pa ed e4 we pr pg na al nal di ph ds trs
 
-arb: arbbasic arbshared arbapplications help arbxtras
+arb: arbbasic arbshared arbapplications arbxtras
 #arb: arbbasic db aw dp awt dbs nt pa ed e4 we pr pg na al nal di db2 ph ds trs help arbxtras
 
 save:  rmbak
--- SECEDIT/SEC_main.cxx.orig   Mon Jun 23 22:08:23 2003
+++ SECEDIT/SEC_main.cxx        Mon Jun 23 22:06:24 2003
@@ -489,10 +489,10 @@
 
     for (pos=0; ; pos++) {
         BI_helix::BI_helix_entry *entry = &(helix->entries[pos]);
-        //char *helix_nr = 0;
+        char *helix_nr = 0;
 
         if (entry->pair_type!=HELIX_NONE) {
-            char *helix_nr = entry->helix_nr;
+            helix_nr = entry->helix_nr;
 
             if (helix_nr==start_helix_nr) { // same helix_nr as last
                 end_helix = pos;
@@ -545,10 +545,10 @@
 
     for (pos=0; ; pos++) {
         BI_helix::BI_helix_entry *entry = &(helix->entries[pos]);
-        //char *helix_nr;
+        char *helix_nr = 0;
 
         if (entry->pair_type!=HELIX_NONE) {
-            char *helix_nr = entry->helix_nr;
+            helix_nr = entry->helix_nr;
 
             if (helix_nr==start_helix_nr) { // same helix as last
                 end_helix = pos;
--- PROBE_GROUP/pg_main.cxx.orig        Mon Jun 23 22:11:39 2003
+++ PROBE_GROUP/pg_main.cxx     Fri Jul  5 02:56:58 2002
@@ -38,6 +38,7 @@
 #define UU     0
 
 #include <arbdbt.h>
+#include <stdarg.h>
 
 using namespace std;
