--- cdat-4.0/install.py.orig	Fri Feb 20 06:20:14 2004
+++ cdat-4.0/install.py	Fri Feb 20 06:20:01 2004
@@ -6,10 +6,10 @@
 if not hasattr(sys, 'version_info') or sys.version_info < minimum_python_version:
     sys.stderr.write("Your Python is too old; please see README.txt.\n")
     trouble = 1
-for x in string.split("PYTHONHOME PYTHONPATH"):
-    if os.environ.has_key(x):
-        sys.stderr.write('Please undefine ' + x + ' before installation.\n')
-        trouble = 1
+#for x in string.split("PYTHONHOME PYTHONPATH"):
+#    if os.environ.has_key(x):
+#        sys.stderr.write('Please undefine ' + x + ' before installation.\n')
+#        trouble = 1
 if not os.environ.has_key('HOME'):
     sys.stderr.write(\
 "Caution: You'll need to set environment variable HOME before using CDAT.\n")
--- cdat-4.0/installation/contrib.py.orig	Fri Feb 20 09:59:54 2004
+++ cdat-4.0/installation/contrib.py	Fri Feb 20 10:01:43 2004
@@ -5,9 +5,9 @@
     dostandard = 1
 
 clist = [
-    'contrib/SP',
+#   'contrib/SP',
     'contrib/cssgrid',
-    'contrib/lmoments',
+#   'contrib/lmoments',
     'contrib/ort',
     'contrib/spherepack',
     'contrib/asciidata',
@@ -19,3 +19,3 @@
     'contrib/pyncl',
-    'contrib/f2py',
+#   'contrib/f2py',
     'contrib/IaGraph',
--- cdat-4.0/installation/myconfig.py.orig	Fri Feb 20 07:30:26 2004
+++ cdat-4.0/installation/myconfig.py	Fri Feb 20 07:30:19 2004
@@ -0,0 +1,2 @@
+netcdf_library_directory = os.path.join('@PREFIX@', 'lib')
+netcdf_include_directory= os.path.join('@PREFIX@', 'include')
--- cdat-4.0/installation/install.py.orig	Fri Feb 20 06:38:55 2004
+++ cdat-4.0/installation/install.py	Fri Feb 20 06:38:39 2004
@@ -1,7 +1,8 @@
 import sys, getopt, os, shutil, string, glob, tempfile
 from distutils.core import setup
 here = os.getcwd()
-logdir = os.path.join(here, 'logs')
+logdir = os.path.join(here, 'logs')
+#logdir = os.path.join(os.environ.get('HOME')+'/cdat/', 'logs')
 execfile(os.path.join("installation", "control.py"), globals(), globals())
 cdms_include_directory = os.path.join(sys.prefix, 'include', 'cdms')
 cdms_library_directory = os.path.join(sys.exec_prefix, 'lib')
--- cdat-4.0/Packages/vcs/setup.py.orig	Mon Feb 23 13:07:28 2004
+++ cdat-4.0/Packages/vcs/setup.py	Mon Feb 23 13:08:42 2004
@@ -28,7 +28,7 @@
                    cdat_info.cdunif_include_directories +\
                    [ x11include,
                      py+'/include', 
-                     ]
+                     ]+['@PREFIX@/include']
 vcs_library_dirs = cdat_info.cdunif_library_directories + [ x11libdir,]
 vcs_libraries= x_libraries + cdat_info.cdunif_libraries
 
--- cdat-4.0/installation/standard.py.orig	Sat Mar 13 10:23:56 2004
+++ cdat-4.0/installation/standard.py	Sat Mar 13 10:25:45 2004
@@ -43,7 +43,7 @@
 # Build actions
 action = {}
 action['*.pfp'] = os.path.join(sys.exec_prefix, 'bin', 'pyfort') + " -i %(file)s"
-action['setup.py'] = sys.executable + ' setup.py install'
+action['setup.py'] = sys.executable + ' setup.py install --prefix=' + sys.exec_prefix
 action['install_script'] = './install_script ' + sys.exec_prefix
 for k in ['makefile','Makefile','MAKEFILE']:
     action[k] = make_code + " PREFIX='%s' install " % sys.exec_prefix
--- cdat-4.0/contrib/pyncl/install_script.orig	Sat Mar 13 17:19:43 2004
+++ cdat-4.0/contrib/pyncl/install_script	Sat Mar 13 17:20:51 2004
@@ -52,5 +52,5 @@
 
 #PyNCL
 (cd pyncl*; \
-   ${prefix}/bin/python setup.py build ${D} install \
+   ${prefix}/bin/python setup.py build ${D} install --prefix=${prefix}\
 )
--- cdat-4.0/contrib/IaGraph/install_script.orig	Sat Mar 13 17:21:39 2004
+++ cdat-4.0/contrib/IaGraph/install_script	Sat Mar 13 17:22:03 2004
@@ -52,5 +52,5 @@
 
 #IaGraph
 (cd IaGraph*; \
-   ${prefix}/bin/python setup.py build ${D} install \
+   ${prefix}/bin/python setup.py build ${D} install --prefix=${prefix}\
 )
--- cdat-4.0/contrib/pyclimate/install_script.orig	Sat Mar 13 17:22:24 2004
+++ cdat-4.0/contrib/pyclimate/install_script	Sat Mar 13 17:22:45 2004
@@ -45,5 +45,5 @@
 
 #Scientific Python
 (cd PyClimate*; \
-   ${prefix}/bin/python setup.py build ${D} install \
+   ${prefix}/bin/python setup.py build ${D} install --prefix=${prefix}\
 )
