--- DODS/src/packages-3.4.2/Makefile.in.orig	Sun Feb 22 09:42:01 2004
+++ DODS/src/packages-3.4.2/Makefile.in	Mon Feb 23 15:55:29 2004
@@ -84,7 +84,7 @@
 	-$(MAKE) -C $(ZLIB) $(MFLAGS) clean
 
 config-libcurl: 
-	cd $(CURL) && ./configure --prefix=$(dods_root) --disable-shared;
+	cd $(CURL) && ./configure --prefix=$(dods_root) --disable-shared --without-ssl --without-ca-bundle;
 
 libcurl: .FORCE
 	cd $(CURL) && $(MAKE) $(MFLAGS)
--- DODS/src/packages-3.4.2/libcurl/lib/Makefile.in.orig	Mon Feb 23 15:55:45 2004
+++ DODS/src/packages-3.4.2/libcurl/lib/Makefile.in	Mon Feb 23 16:02:44 2004
@@ -644,7 +644,7 @@
 $(srcdir)/ca-bundle.h: Makefile.in Makefile
 	cd $(srcdir) && \
 	echo "/* The file is generated automaticly */" > $@
-	echo '#define CURL_CA_BUNDLE @CURL_CA_BUNDLE@' >> $@
+	echo '#undef CURL_CA_BUNDLE @CURL_CA_BUNDLE@' >> $@
 
 install-data-hook:
 	@if test -n "@CURL_CA_BUNDLE@"; then \
