diff -ruN eden-orig/python/eden.py eden/python/eden.py
--- eden-orig/python/eden.py	2004-06-25 00:56:17.000000000 -0700
+++ eden/python/eden.py	2004-08-28 08:39:03.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/swprefix/bin/python
 
 # File: eden.py
 from random import randint
@@ -187,7 +187,7 @@
         self.root.mainloop()
 
     def browse(self):
-        system("mozilla http://www.edencrystallography.org &")
+        system("open http://www.edencrystallography.org &")
         
     def run_cb(self,k):
         # Here we need to insert code to generate the correct input
diff -ruN eden-orig/source/Makefile eden/source/Makefile
--- eden-orig/source/Makefile	2004-08-19 10:48:16.000000000 -0700
+++ eden/source/Makefile	2004-08-28 17:44:33.000000000 -0700
@@ -56,14 +56,14 @@
 #######################################################################
 # 	Directory where the FFTW include and lib are
 #######################################################################
-FFTW		= /sw
+FFTW		= /swprefix
 LIB  		= $(FFTW)/lib
 INC  		= $(FFTW)/include
 
 #######################################################################
 # 	Directory to install the binary, version
 #######################################################################
-BIN 		= $(HOME)/bin
+BIN 		= /swinstallprefix/bin
 VERSION		=V5.2
 
 #######################################################################
@@ -74,7 +74,8 @@
 # -O3 on HP, -O2 on SGI or DEC.
 #
 #######################################################################
-CFLAGS		=   -Wall -pipe -g3
+#CFLAGS		=   -Wall -O3 
+CFLAGS		=   -Wall -pipe -g3 -Wno-long-double
 #CFLAGS		=   -Wall -pipe -marchi686 -pedantic  -gdwarf-2 -g3
 #CFLAGS		=   -Wall -O3 -march=i686 
 ifneq ($(FFTW),/usr/local)
@@ -88,7 +89,7 @@
 # Use different libraries and include files for single and
 # double precision versions of EDEN.
 #######################################################################
-DOUBLESWITCH 	= ON
+DOUBLESWITCH = ON
 
 ifeq ($(DOUBLESWITCH),ON)
 LDFLAGS	 = -L$(LIB) -ldfftw -lm  
@@ -126,8 +127,8 @@
 		$(CC) -o $@ $^
 
 install:	all
-		mv fbyteswap $(BIN)
-		mv eden $(BIN)/$(TARGET)
+		cp fbyteswap $(BIN)
+		cp eden $(BIN)/$(TARGET)
 		chmod 751 $(BIN)/$(TARGET)
 		chmod +x mkeden.csh
 		./mkeden.csh $(BIN) $(BIN)/$(TARGET)
