diff -ruN -x '*~' gdl-0.8.11/src/basic_fun.cpp gdl-0.8.11.patched/src/basic_fun.cpp
--- gdl-0.8.11/src/basic_fun.cpp	2005-10-17 04:22:04.000000000 -0400
+++ gdl-0.8.11.patched/src/basic_fun.cpp	2006-09-17 12:11:19.000000000 -0400
@@ -33,6 +33,13 @@
 #include <unistd.h>
 #endif
 
+// on OS X isfinite is not defined
+#if defined(__APPLE__) && !defined(isfinite)
+#define isfinite( x ) ( ( sizeof ( (x) ) == sizeof( float) ) ? \
+                         __isfinitef ( (float)(x) ) : \
+                         __isfinited  ( (double)(x) ) )
+#endif
+
 #include "datatypes.hpp"
 #include "envt.hpp"
 #include "dpro.hpp"
diff -ruN -x '*~' gdl-0.8.11/src/math_fun_jmg.cpp gdl-0.8.11.patched/src/math_fun_jmg.cpp
--- gdl-0.8.11/src/math_fun_jmg.cpp	2005-10-17 04:22:04.000000000 -0400
+++ gdl-0.8.11.patched/src/math_fun_jmg.cpp	2006-09-17 12:09:37.000000000 -0400
@@ -31,20 +31,10 @@
 
 // on OS X isfinite is not defined
 #if defined(__APPLE__) && !defined(isfinite)
-
-#ifdef __cplusplus
-extern "C" {
-#endif
-#define      isfinite( x )      ( ( sizeof ( x ) == sizeof(double) ) ?       \
-                              __isfinited ( x ) :                            \
-                                ( sizeof ( x ) == sizeof( float) ) ?         \
-                              __isfinitef ( x ) :                            \
-                              __isfinite  ( x ) )
-#ifdef __cplusplus
-}
+#define isfinite( x ) ( ( sizeof ( (x) ) == sizeof( float) ) ? \
+                       __isfinitef ( (float)(x) ) : \
+                       __isfinited  ( (double)(x) ) )
 #endif
-#endif
-
 
 using namespace std;
 
