diff -ruN ghemical-1.01/src/graphics/objects.h ghemical-1.01-patched/src/graphics/objects.h
--- ghemical-1.01/src/graphics/objects.h	2001-12-29 09:56:49.000000000 -0500
+++ ghemical-1.01-patched/src/graphics/objects.h	2005-03-18 22:02:59.000000000 -0500
@@ -177,7 +177,7 @@
 {
 	protected:
 	
-	friend obj_loc_data * dummy_object::GetLocDataRW(void) const;
+	friend class dummy_object;
 	
 	obj_loc_data * data;
 	
