diff -Nurd -x'*~' gOpenMol-3.00.orig/environment.txt gOpenMol-3.00/environment.txt
--- gOpenMol-3.00.orig/environment.txt	2005-08-29 10:04:23.000000000 -0400
+++ gOpenMol-3.00/environment.txt	2007-02-01 01:29:25.000000000 -0500
@@ -16,4 +16,4 @@
 # GOM_HOME    ""
 ### On Windows, the default value is system temp directory.
 ### On UNIX, the default is $TMPDIR, $TMP or /tmp.
-# GOM_TEMP    temp
+GOM_TEMP    /tmp
diff -Nurd -x'*~' gOpenMol-3.00.orig/src/config.mk.ac gOpenMol-3.00/src/config.mk.ac
--- gOpenMol-3.00.orig/src/config.mk.ac	2004-09-23 17:24:29.000000000 -0400
+++ gOpenMol-3.00/src/config.mk.ac	2007-02-01 01:30:10.000000000 -0500
@@ -18,7 +18,7 @@
 gomdataroot      = $(datadir)/$(gomsubdir)
 gombindir        = $(gomroot)/bin
 gomincdir        = $(includedir)/$(gomsubdir)
-gomlibdir        = $(gomroot)/bin
+gomlibdir        = $(gomroot)/lib
 # Tcl/Tk entries.
 TCL_LIBRARY      = @TCL_LIBRARY@
 TK_LIBRARY       = @TK_LIBRARY@
@@ -64,7 +64,7 @@
 CCTAGS           = @CCTAGS@
 CCDEPMODE        = @CCDEPMODE@
 CCDEPCOMP        = @CCDEPCOMP@
-GOM_CPPFLAGS     = @CPPFLAGS@ -I$(incdir) -Iinclude -I.
+GOM_CPPFLAGS     = -I$(incdir) -Iinclude -I. @CPPFLAGS@
 GOM_CFLAGS       = @ALL_CFLAGS@
 GOM_FFLAGS       = @ALL_FFLAGS@
 GOM_LDFLAGS      = @ALL_LDFLAGS@
diff -Nurd -x'*~' gOpenMol-3.00.orig/src/parser/p_showc.c gOpenMol-3.00/src/parser/p_showc.c
--- gOpenMol-3.00.orig/src/parser/p_showc.c	2005-08-29 10:05:21.000000000 -0400
+++ gOpenMol-3.00/src/parser/p_showc.c	2007-02-01 01:29:25.000000000 -0500
@@ -42,21 +42,21 @@
 
 #define ALIAS GOM_PARSER_CMD_ALIAS
 
-int gomp_ParserReturnFloatValueFromFunc(GOM_PARSER_ARGLIST,ptrdiff_t Offset)
+int gomp_ParserReturnFloatValueFromFunc(GOM_PARSER_ARGLIST,intptr_t Offset)
 {
     FloatFunc *floatf = GOM_PARSER_GET_POINTER_VALUE(Offset);
     GOM_PARSER_RETURN_DOUBLE(floatf->Func());
     GOM_PARSER_SUCCEEDED;
 }
 
-int gomp_ParserReturnIntValueFromFunc(GOM_PARSER_ARGLIST,ptrdiff_t Offset)
+int gomp_ParserReturnIntValueFromFunc(GOM_PARSER_ARGLIST,intptr_t Offset)
 {
     IntFunc *intf = GOM_PARSER_GET_POINTER_VALUE(Offset);
     GOM_PARSER_RETURN_INT(intf->Func());
     GOM_PARSER_SUCCEEDED;
 }
 
-int gomp_ParserReturnBooleanValueFromFunc(GOM_PARSER_ARGLIST,ptrdiff_t Offset)
+int gomp_ParserReturnBooleanValueFromFunc(GOM_PARSER_ARGLIST,intptr_t Offset)
 {
     IntFunc *intf = GOM_PARSER_GET_POINTER_VALUE(Offset);
     GOM_PARSER_RETURN_BOOLEAN(intf->Func());
@@ -64,7 +64,7 @@
 }
 
 int gomp_ParserReturnBooleanEnumValueFromFunc(GOM_PARSER_ARGLIST,
-                                              ptrdiff_t Offset)
+                                              intptr_t Offset)
 {
     BooleanEnumFunc *booleanf = GOM_PARSER_GET_POINTER_VALUE(Offset);
     GOM_PARSER_RETURN_STRING(
@@ -74,14 +74,14 @@
     GOM_PARSER_SUCCEEDED;
 }
 
-int gomp_ParserReturnStringValueFromFunc(GOM_PARSER_ARGLIST,ptrdiff_t Offset)
+int gomp_ParserReturnStringValueFromFunc(GOM_PARSER_ARGLIST,intptr_t Offset)
 {
     StringFunc *stringf = GOM_PARSER_GET_POINTER_VALUE(Offset);
     GOM_PARSER_RETURN_STRING(stringf->Func());
     GOM_PARSER_SUCCEEDED;
 }
 
-int gomp_ParserVerifyFuncResult(GOM_PARSER_ARGLIST,ptrdiff_t Offset)
+int gomp_ParserVerifyFuncResult(GOM_PARSER_ARGLIST,intptr_t Offset)
 {
     IntFunc *intf = GOM_PARSER_GET_POINTER_VALUE(Offset);
     GOM_PARSER_VERIFY( intf->Func() == 0 );
diff -Nurd -x'*~' gOpenMol-3.00.orig/src/plugins/Makefile gOpenMol-3.00/src/plugins/Makefile
--- gOpenMol-3.00.orig/src/plugins/Makefile	2005-08-29 10:05:21.000000000 -0400
+++ gOpenMol-3.00/src/plugins/Makefile	2007-02-01 01:29:25.000000000 -0500
@@ -22,7 +22,7 @@
 build compile clean clean-libs clean-objs: FORCE
 	-@for dir in *; do \
 	    if test -f "$$dir/Makefile"; then \
-		( cd "$$dir" && $(MAKE) $(MFLAGS) $@ ) ;\
+		( cd "$$dir" && $(MAKE) $(MFLAGS) $@ ) || exit 1;\
 	    fi ;\
 	done
 
diff -Nurd -x'*~' gOpenMol-3.00.orig/src/plugins/config.mk.ac gOpenMol-3.00/src/plugins/config.mk.ac
--- gOpenMol-3.00.orig/src/plugins/config.mk.ac	2004-09-23 17:24:54.000000000 -0400
+++ gOpenMol-3.00/src/plugins/config.mk.ac	2007-02-01 01:30:34.000000000 -0500
@@ -11,7 +11,7 @@
 gombindir        = ${gomroot}/bin
 gomincdir        = ${gomsrcdir}/include/gomlib
 #INS gomincdir        = ${includedir}/${gomsubdir}
-gomlibdir        = ${gomroot}/bin
+gomlibdir        = ${gomroot}/lib
 gomsrcdir        = @abs_top_srcdir@
 gombltdir        = @abs_top_builddir@
 #INS gomsrcdir        = ${gomroot}/src
@@ -59,7 +59,7 @@
 COMMON_LTLDFLAGS = @COMMON_LTLDFLAGS@
 # Values used by gOpenMol.
 # Not quite... Don't show private gOpenMol headers to plugins.
-GOM_CPPFLAGS     = @CPPFLAGS@ -I${gomincdir} -I${gombltdir}/include/gomlib
+GOM_CPPFLAGS     = -I${gomincdir} -I${gombltdir}/include/gomlib @CPPFLAGS@
 GOM_CFLAGS       = @CFLAGS@
 GOM_FFLAGS       = @FFLAGS@
 GOM_LDFLAGS      = @LDFLAGS@
diff -Nurd -x'*~' gOpenMol-3.00.orig/src/unix/make/gomapi.mk gOpenMol-3.00/src/unix/make/gomapi.mk
--- gOpenMol-3.00.orig/src/unix/make/gomapi.mk	2005-08-29 10:05:30.000000000 -0400
+++ gOpenMol-3.00/src/unix/make/gomapi.mk	2007-02-01 01:29:25.000000000 -0500
@@ -47,6 +47,7 @@
 # (see case statement above).
 graphics/gomapi.exp: include/gomext/*.h
 	{ cat include/gomext/*.h | sed 's/^.*gom_/gom_/'; echo gom_main; } >$@
+	perl -ni.bak -e 'print unless /dummy_func/' $@
 
 clean: clean-gomapi clean-stamps FORCE
 
diff -Nurd -x'*~' gOpenMol-3.00.orig/src/unix/make/install.mk gOpenMol-3.00/src/unix/make/install.mk
--- gOpenMol-3.00.orig/src/unix/make/install.mk	2005-08-29 10:05:31.000000000 -0400
+++ gOpenMol-3.00/src/unix/make/install.mk	2007-02-01 02:24:45.000000000 -0500
@@ -26,6 +26,7 @@
 	-e 's%^TCL_LIBRARY=.*%TCL_LIBRARY='"'"'$(TCL_LIBRARY)'"'%" \
 	-e 's%^TK_LIBRARY=.*%TK_LIBRARY='"'"'$(TK_LIBRARY)'"'%" \
 	-e 's%^GOM_ROOT=.*%GOM_ROOT='"'"'$(gomroot)'"'%" \
+	-e 's%__fink_d__%%' \
 	-e 's%@PATH_SEPARATOR@%$(PATH_SEPARATOR)%' \
 	-e "s%@SHLIBPATHVAR@%$${shlibpath_var}%g" )
 
@@ -36,7 +37,7 @@
 mkinstalldirs-local: FORCE
 	$(MKDIR_P) '$(gomroot)' || test -d '$(gomroot)'
 	@$(set_show); set -e; \
-	for dir in bin src/plugins src/unix; do \
+	for dir in bin lib src/plugins src/unix; do \
 	    dir='$(gomroot)'/$$dir; \
 	    $$show $(MKDIR_P) "$$dir"; \
 	    $(MKDIR_P) "$$dir" || test -d "$$dir"; \
@@ -85,7 +86,6 @@
 
 install-local-local: install-local-generated \
                      install-local-environment \
-                     install-local-rungopenmol \
                      install-local-data \
                      FORCE ;
 
diff -Nurd -x'*~' gOpenMol-3.00.orig/src/unix/make/utility.mk.ac gOpenMol-3.00/src/unix/make/utility.mk.ac
--- gOpenMol-3.00.orig/src/unix/make/utility.mk.ac	2005-08-29 10:05:31.000000000 -0400
+++ gOpenMol-3.00/src/unix/make/utility.mk.ac	2007-02-01 01:29:25.000000000 -0500
@@ -47,8 +47,7 @@
                xmol2bamber$(EXEEXT) \
                xplor2charmm$(EXEEXT) \
                gridasc2plt$(EXEEXT) \
-               gridbin2plt$(EXEEXT) \
-               icon8$(EXEEXT)
+               gridbin2plt$(EXEEXT)
 
 BINARIES_JAR = xvibs.jar
 
