diff -Nurd -x'*~' gperiodic-2.0.8.orig/Makefile gperiodic-2.0.8/Makefile
--- gperiodic-2.0.8.orig/Makefile	2003-08-09 15:33:27.000000000 -0400
+++ gperiodic-2.0.8/Makefile	2007-06-13 18:40:08.000000000 -0400
@@ -1,5 +1,5 @@
 CC=gcc
-CFLAGS=`pkg-config --cflags gtk+-2.0` -I. -DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGDK_PIXBUF_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED
+CFLAGS=-I. `pkg-config --cflags gtk+-2.0` -DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGDK_PIXBUF_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED
 LIBS=`pkg-config --libs gtk+-2.0`
 bindir ?= /usr/bin
 datadir ?= /usr/share
@@ -17,8 +17,11 @@
 
 install:
 	make -C po/ install enable_nls=1 datadir=$(datadir)
+	install -d        $(bindir)
 	install gperiodic $(bindir)
+	install -d                $(datadir)/applications
 	install gperiodic.desktop $(datadir)/applications
+	install -d            $(datadir)/pixmaps
 	install gperiodic.png $(datadir)/pixmaps
 	install gperiodic-crystal.png $(datadir)/pixmaps
 	make clean
