diff -ru kseg-0.4/G_pointLocus.cpp kseg-0.4-patched/G_pointLocus.cpp
--- kseg-0.4/G_pointLocus.cpp	2003-04-19 04:50:13.000000000 +0200
+++ kseg-0.4-patched/G_pointLocus.cpp	2003-12-30 04:02:02.000000000 +0100
@@ -283,7 +283,7 @@
     }
     else discontCount = 0; //number of times the goodness decreased--for finding discontinuities
 
-    if(discontCount > log(maxSamples) / 1.5) {
+    if(discontCount > log(double(maxSamples)) / 1.5) {
       points[0].loc = G_point::inValid();
       points[points[0].next].loc = G_point::inValid();
       points[points[0].prev].loc = G_point::inValid();
diff -ru kseg-0.4/Makefile kseg-0.4-patched/Makefile
--- kseg-0.4/Makefile	2003-04-19 04:50:13.000000000 +0200
+++ kseg-0.4-patched/Makefile	2003-12-30 04:23:49.000000000 +0100
@@ -1,8 +1,8 @@
 # Makefile for KSEG
 CC = g++
-QTINCLUDE = -I$(QTDIR)/include
+QTINCLUDE = -I$(QTDIR)/include/qt
 CCFLAGS = -c -Wall -O2 -fomit-frame-pointer -ffast-math
-LIBS = -lm -L$(QTDIR)/lib -lqt -lz
+LIBS = -lm -L$(QTDIR)/lib -lqt-mt -lz
 MOC = $(QTDIR)/bin/moc
 LUPDATE = $(QTDIR)/bin/lupdate
 LRELEASE = $(QTDIR)/bin/lrelease
diff -ru kseg-0.4/formula/Makefile kseg-0.4-patched/formula/Makefile
--- kseg-0.4/formula/Makefile	2003-04-19 04:50:13.000000000 +0200
+++ kseg-0.4-patched/formula/Makefile	2003-12-30 03:28:33.000000000 +0100
@@ -1,6 +1,6 @@
 # Makefile for formula
 CC = g++
-QTINCLUDE = -I$(QTDIR)/include
+QTINCLUDE = -I$(QTDIR)/include/qt
 CCFLAGS = -c -Wall -O2 -fomit-frame-pointer -ffast-math
 LIBS = 
 MOC = $(QTDIR)/bin/moc
diff -ru kseg-0.4/main.cpp kseg-0.4-patched/main.cpp
--- kseg-0.4/main.cpp	2003-04-19 04:50:13.000000000 +0200
+++ kseg-0.4-patched/main.cpp	2003-12-30 03:28:17.000000000 +0100
@@ -37,7 +37,7 @@
 
   //set some default properties
   KSegProperties::setPropertyDefault("Language", "");
-  KSegProperties::setPropertyDefault("LanguageDir", ".");
+  KSegProperties::setPropertyDefault("LanguageDir", "@PREFIX@/share/qt3/translations");
   KSegProperties::setPropertyDefault("SelectType", "BorderSelect");
   KSegProperties::setPropertyDefault("QuickPlayDirectory", ".");
   KSegProperties::setPropertyDefault("RecentListSize", "4");
