diff -u libpolyxmass-0.9.0/configure libpolyxmass-0.9.0-fink/configure
--- libpolyxmass-0.9.0/configure	2006-01-06 13:43:45.000000000 -0800
+++ libpolyxmass-0.9.0-fink/configure	2006-01-26 23:00:28.000000000 -0800
@@ -19625,7 +19625,7 @@
 fi
 
 
-CONFDIR="${sysconfdir}/polyxmass"
+CONFDIR="${prefix}/etc/polyxmass"
 
 cat >>confdefs.h <<_ACEOF
 #define CONFDIR "$CONFDIR"
Common subdirectories: libpolyxmass-0.9.0/intl and libpolyxmass-0.9.0-fink/intl
Common subdirectories: libpolyxmass-0.9.0/m4 and libpolyxmass-0.9.0-fink/m4
Common subdirectories: libpolyxmass-0.9.0/po and libpolyxmass-0.9.0-fink/po
Common subdirectories: libpolyxmass-0.9.0/src and libpolyxmass-0.9.0-fink/src
