diff -Nurd -x'*~' mathomatic-12.1b.orig/makefile mathomatic-12.1b/makefile
--- mathomatic-12.1b.orig/makefile	2005-02-12 20:54:58.000000000 -0500
+++ mathomatic-12.1b/makefile	2005-02-20 23:59:52.000000000 -0500
@@ -1,8 +1,8 @@
 # Makefile for compiling Mathomatic for UNIX or Linux or Mac OS X.
 
 VERSION		= `cat VERSION`
-CFLAGS		= -O -DUNIX -DVERSION=\"$(VERSION)\"
-LDFLAGS		= -s
+CFLAGS		+= -O -DUNIX -DVERSION=\"$(VERSION)\"
+LDFLAGS		+= -s
 LIBS		= -lm
 
 CFLAGS		+= $(READLINE:1=-DREADLINE)
@@ -10,7 +10,8 @@
 
 prefix		= /usr/local
 bindir		= $(prefix)/bin
-mandir		= $(prefix)/man
+mandir		= $(prefix)/share/man
+docdir		= $(prefix)/share/doc
 
 AOUT	=mathomatic
 OBJECTS	=main.o am.o parse.o cmds.o simplify.o factor.o super.o \
@@ -38,10 +39,10 @@
 install: $(AOUT)
 	install -d $(bindir)
 	install -d $(mandir)/man1
-	install -d $(prefix)/doc/mathomatic
+	install -d $(docdir)/mathomatic
 	install -m 0755 $(AOUT) $(bindir)
 	install -m 0644 doc/mathomatic.1 $(mandir)/man1
-	install -m 0644 doc/*.htm $(prefix)/doc/mathomatic
+	install -m 0644 doc/*.htm $(docdir)/mathomatic
 	@echo Install completed.
 
 clean:
