--- mbsystem-5.0.6/install_makefiles.orig	2005-02-21 07:55:15.000000000 -0800
+++ mbsystem-5.0.6/install_makefiles	2005-02-21 07:39:12.000000000 -0800
@@ -405,8 +405,8 @@
 if ($preset eq "macosxfink")
 	{
 	print "\nUsing preset configuration for MacOs X by fink in /sw...\n";
-	$FINKPREFIX = "/sw";
-	$GMTVERSION = "3.4";
+	$FINKPREFIX = "@FINKPREFIX@";
+	$GMTVERSION = "4.0";
 	$MBSYSTEM_HOME = "$FINKPREFIX";
 	$OS = "DARWIN";
 	$CC = "cc";
