diff -Naur ncview-1.92e/Makefile.in ncview-1.92enew/Makefile.in
--- ncview-1.92e/Makefile.in	Wed Aug 13 19:29:29 2003
+++ ncview-1.92e/Makefile.in	Wed Feb  4 16:44:44 2004
@@ -36,14 +36,14 @@
 ## installed when you run "make install".
 ################################################################
 BINDIR = @prefix@/bin
-MANDIR = @prefix@/man/man1
+MANDIR = @prefix@/share/man/man1
 
 #####################################################################
 ## Set this to the directory where system-wide .ncmap (colormap) 
 ## files will be kept and installed.  Comment this line out if
 ## no such directory will be specified.
 #####################################################################
-NCVIEW_LIB_DIR = @prefix@/lib/ncview
+NCVIEW_LIB_DIR = @prefix@/share/ncview
 
 ######################################################################
 ## If needed, *add* here the names of any additional include or library 
@@ -135,17 +135,20 @@
 	@echo "-----------------------------------------------------------"
 	@echo "Installing colormaps in directory $(NCVIEW_LIB_DIR)"
 	@echo "-----------------------------------------------------------"
-	if test $(NCVIEW_LIB_DIR)x != x; then if test ! -d $(NCVIEW_LIB_DIR); then mkdir -p $(NCVIEW_LIB_DIR); fi; for i in $(COLORMAPS); do chown `whoami` $$i; $(INSTALL) $(INSTDATFLAGS) $$i $(NCVIEW_LIB_DIR); done; fi
+	if test $(NCVIEW_LIB_DIR)x != x; then if test ! -d $(BUILDPREFIX)$(NCVIEW_LIB_DIR); then mkdir -p $(BUILDPREFIX)$(NCVIEW_LIB_DIR); fi; for i in $(COLORMAPS); do chown `whoami` $$i; $(INSTALL) $(INSTDATFLAGS) $$i $(BUILDPREFIX)$(NCVIEW_LIB_DIR); done; fi
 	@echo "----------------------------------------------------"
 	@echo "Installing binary in directory $(BINDIR)"
 	@echo "----------------------------------------------------"
-	if test ! -d $(BINDIR); then  mkdir -p $(BINDIR); fi; $(INSTALL) $(INSTBINFLAGS) ncview   $(BINDIR)/ncview
+	if test ! -d $(BUILDPREFIX)$(BINDIR); then  mkdir -p $(BUILDPREFIX)$(BINDIR); fi; $(INSTALL) $(INSTBINFLAGS) ncview   $(BUILDPREFIX)$(BINDIR)/ncview
 	@echo "--------------------------------------------------------"
 	@echo "Installing manual in directory $(MANDIR)"
 	@echo "--------------------------------------------------------"
-	if test ! -d $(MANDIR); then mkdir -p $(MANDIR); fi; $(INSTALL) $(INSTMANFLAGS) ncview.1 $(MANDIR)/ncview.1
-	@echo "------------------------------------"
-	@echo "Installing application defaults file"
-	@echo "------------------------------------"
-	/bin/sh ./install-appdef "$(INSTALL) $(INSTMANFLAGS)"
-
+	if test ! -d $(BUILDPREFIX)$(MANDIR); then mkdir -p $(BUILDPREFIX)$(MANDIR); fi; $(INSTALL) $(INSTMANFLAGS) ncview.1 $(BUILDPREFIX)$(MANDIR)/ncview.1
+	@echo "--------------------------------------------------------"
+	@echo "Installing app-defaults file in $(XAPPLRESDIR)"
+	@echo "--------------------------------------------------------"
+	mv ncview ncview.exe
+	cp ./Ncview-appdefaults ./Ncview
+	if test ! -d $(BUILDPREFIX)$(XAPPLRESDIR); then mkdir -p $(BUILDPREFIX)$(XAPPLRESDIR); fi; $(INSTALL) $(INSTMANFLAGS) Ncview $(BUILDPREFIX)$(XAPPLRESDIR)
+	rm -f ./Ncview
+	mv ncview.exe ncview
diff -Naur ncview-1.92e/configure ncview-1.92enew/configure
--- ncview-1.92e/configure	Wed Aug 13 19:04:36 2003
+++ ncview-1.92e/configure	Mon Feb  2 19:02:12 2004
@@ -2651,14 +2651,13 @@
 fi
 if test $do_udunits = true; then
 	echo "****************************************************************************"
-	echo "Udunits support enabled. ***NOTE THIS WILL >>NOT<< WORK WITH MAC OS X!!! ***"
-	echo "If you are using Mac OS X, remove the udunits library and remake ncview  ***"
-	echo "udunits dirs: include: $UDUNITS_INCDIR  library: $UDUNITS_LIBDIR  libname: $UDUNITS_LIBNAME"
+	echo "Udunits support enabled."
+	echo "udunits dirs: include: $UDUNITS_INCDIR  library: $UDUNITS_LIBDIR  libname: $UDUNITS_LIBNAME & udport"
 	echo "****************************************************************************"
 	DO_UDUNITS_LINE="INC_UDUNITS   = -DINC_UDUNITS"
 	DO_UDUNITS_INCDIR="UDUNITSINCDIR = -I$UDUNITS_INCDIR"
 	DO_UDUNITS_LIBDIR="UDUNITSLIBDIR = -L$UDUNITS_LIBDIR"
-	DO_UDUNITS_LIBNAME="UDUNITSLIB = -l$UDUNITS_LIBNAME"
+	DO_UDUNITS_LIBNAME="UDUNITSLIB = -l$UDUNITS_LIBNAME -ludport"
 else
 	echo "************************************************************************"
 	echo "Note: udunits support is NOT enabled, because I could not find the "
@@ -2674,7 +2673,7 @@
 	DO_UDUNITS_LINE="#INC_UDUNITS   = -DINC_UDUNITS"
 	DO_UDUNITS_INCDIR="#UDUNITSINCDIR = -I$UDUNITS_INCDIR"
 	DO_UDUNITS_LIBDIR="#UDUNITSLIBDIR = -I$UDUNITS_LIBDIR"
-	DO_UDUNITS_LIBNAME="#UDUNITSLIB = -l$UDUNITS_LIBNAME"
+	DO_UDUNITS_LIBNAME="#UDUNITSLIB = -l$UDUNITS_LIBNAME -ludport"
 fi
 
 
