--- numarray-1.1/addons.py.orig	Fri Oct  1 04:50:03 2004
+++ numarray-1.1/addons.py	Fri Oct  1 04:51:52 2004
@@ -31,11 +31,11 @@
 
     # Set to list off libraries to link against.
     # (only the basenames, e.g. 'lapack')
-    lapack_libs = ['lapack', 'f77blas', 'cblas', 'atlas', 'f90math', 'fio', 'f77math', 'm']
+    lapack_libs = []
 
 
 # Set to list directories to be searched for BLAS and LAPACK libraries
-lapack_dirs = ['/usr/local/lib/atlas', '/opt/absoft/lib']
+lapack_dirs = []
 
 ADDONS_DATA_FILES = [('numarray/ma', ['Packages/MA/Legal.htm'])]
 
@@ -85,6 +85,7 @@
     Extension('numarray.linear_algebra.lapack_lite2',
               sourcelist,
               include_dirs = ["Packages/LinearAlgebra2/Src", 'Include/numarray'],
+              extra_link_args = ["-framework", "vecLib"],
               library_dirs = lapack_dirs,
               libraries = lapack_libs),
     Extension("numarray.random_array.ranlib2",
