--- Numeric/setup.py.orig	Thu Oct 14 08:14:48 2004
+++ Numeric/setup.py	Thu Oct 14 08:45:26 2004
@@ -20,6 +20,7 @@
 
 headers = glob (os.path.join ("Include","Numeric","*.h"))
 extra_compile_args = []  # You could put "-O4" etc. here.
+extra_link_args = ["-framework", "vecLib"]
 mathlibs = ['m']
 define_macros = [('HAVE_INVERSE_HYPERBOLIC',None)]
 undef_macros = []
@@ -40,14 +41,14 @@
              ]
 # set these to use your own BLAS;
 
-library_dirs_list = ['/usr/lib/atlas']
-libraries_list = ['lapack', 'cblas', 'f77blas', 'atlas', 'g2c'] 
+library_dirs_list = []
+libraries_list = [] 
                    # if you also set `use_dotblas` (see below), you'll need:
                    # ['lapack', 'cblas', 'f77blas', 'atlas', 'g2c']
 
 # set to true (1), if you also want BLAS optimized matrixmultiply/dot/innerproduct
 use_dotblas = 1
-include_dirs = ['/usr/include/atlas']  # You may need to set this to find cblas.h
+include_dirs =  ['/System/Library/Frameworks/vecLib.framework/Headers/']
                    #  e.g. on UNIX using ATLAS this should be ['/usr/include/atlas'] 
 
 # The packages are split in this way to allow future optional inclusion
@@ -81,6 +82,7 @@
               extra_compile_args = extra_compile_args),
     Extension('lapack_lite', sourcelist,
               library_dirs = library_dirs_list,
+              extra_link_args = extra_link_args,
               libraries = libraries_list,
               extra_compile_args = extra_compile_args) 
     ]
@@ -114,6 +116,7 @@
     ext_modules.append(Extension('_dotblas',
                                  [os.path.join('Packages', 'dotblas', 'dotblas', '_dotblas.c')],
                                  library_dirs = library_dirs_list,
+                                 extra_link_args = extra_link_args,
                                  libraries = libraries_list,
                                  extra_compile_args=extra_compile_args))
 
