diff -ru octave-forge-2004.09.09/main/comm/Makefile octave-forge-2004.09.09-patched/main/comm/Makefile
--- octave-forge-2004.09.09/main/comm/Makefile	Wed Sep  8 02:41:48 2004
+++ octave-forge-2004.09.09-patched/main/comm/Makefile	Fri Oct  8 18:36:54 2004
@@ -30,7 +30,7 @@
 DELETES = $(OBJECTS) $(GALOISDEPENDS) $(OTHERDEPENDS) *~ $(TARGETS) core octave-core
 
 DEFINES = -DGALOIS_DISP_PRIVATES $(HAVE_DO_FORTRAN_INDEXING) $(HAVE_PROPAGATE_EMPTY_MATRICES) $(HAVE_ND_ARRAYS) $(TYPEID_HAS_CLASS) $(CLASS_HAS_LOAD_SAVE) $(HAVE_OCTAVE_CONCAT) $(HAVE_SWAP_BYTES)
-MOFLAGS =
+MOFLAGS = $(LDFLAGS) -lhdf5
 
 .PHONY: all dist clean realclean count $(SUBDIRS)
 .SUFFIXES:
diff -ru octave-forge-2004.09.09/main/image/Makefile octave-forge-2004.09.09-patched/main/image/Makefile
--- octave-forge-2004.09.09/main/image/Makefile	Wed Sep  1 03:52:24 2004
+++ octave-forge-2004.09.09-patched/main/image/Makefile	Fri Oct  8 18:36:45 2004
@@ -13,13 +13,13 @@
 	$(JPEG) $(PNG)
 
 jpgread.oct: jpgread.cc
-	$(MKOCTFILE) $< -ljpeg
+	$(MKOCTFILE) $< $(LDFLAGS) -ljpeg
 
 jpgwrite.oct: jpgwrite.cc
-	$(MKOCTFILE) $< -ljpeg
+	$(MKOCTFILE) $< $(LDFLAGS) -ljpeg
 
 pngread.oct: pngread.cc
-	$(MKOCTFILE) $< -lpng
+	$(MKOCTFILE) $< $(LDFLAGS) -lpng
 
 pngwrite.oct:
 	$(LN_S) pngread.oct $@
