--- oleo-1.99.16/Makefile.in	Wed Jan 23 15:52:55 2002
+++ oleo-1.99.16-patched/Makefile.in	Wed Jan 23 15:54:04 2002
@@ -131,7 +131,7 @@
 
 EXTRA_DIST = oleobug.in FAQ oleo.spec.in oleo.spec
 
-docdir = $(prefix)/Oleo
+docdir = $(prefix)/share/doc/oleo
 doc_DATA = AUTHORS FAQ
 
 # remove all the symlinks that come from automake

--- oleo-1.99.16/src/cmd.c	Wed Feb 14 20:54:50 2001
+++ oleo-1.99.16-patched/src/cmd.c	Sun Mar 17 23:34:47 2002
@@ -31,7 +31,7 @@
 #include <stdarg.h>
 #include "sysdef.h"
 #include <termios.h>
-#include <malloc.h>
+#include <stdlib.h>
 
 #ifdef	HAVE_MOTIF
 #include "io-motif.h"

--- oleo-1.99.16/configure	2001-03-10 14:32:48.000000000 +1100
+++ oleo-1.99.16-patched/configure	2002-08-15 01:26:49.000000000 +1000
@@ -5233,7 +5233,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lgsl -lgslblas $LIBS"
+LIBS="-lgsl -lgslcblas $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 5239 "configure"
 #include "confdefs.h"
@@ -5261,7 +5261,7 @@
 fi
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  LIBS="$LIBS -lgsl -lgslblas"
+  LIBS="$LIBS -lgsl -lgslcblas"
 	cat >> confdefs.h <<\EOF
 #define HAVE_GSL 1
 EOF

diff -ru oleo-1.99.16/src/global.h oleo-1.99.16-patched/src/global.h
--- oleo-1.99.16/src/global.h   Wed Feb 14 20:54:50 2001
+++ oleo-1.99.16-patched/src/global.h   Wed Sep  4 21:19:03 2002
@@ -208,7 +208,7 @@
 extern char nname[];
 
 extern VOIDSTAR parse_hash;
-extern double __plinf, __neinf, __nan;
+extern double __plinf, __neinf, int__nan;
 
 /* These have two uses.  During parsing, these contain the 
  * base address of all relative references.  During evaluation,

--- oleo-1.99.16/src/eval.c	Tue Feb 13 23:38:05 2001
+++ oleo-1.99.16-patched/src/eval.c	Thu Sep  5 09:22:04 2002
@@ -565,7 +565,7 @@
 	case CONST_NINF:
 	case CONST_NAN:
 	  p->type = TYP_FLT;
-	  p->Float = (byte == CONST_INF) ? __plinf : ((byte == CONST_NINF) ? __neinf : __nan);
+	  p->Float = (byte == CONST_INF) ? __plinf : ((byte == CONST_NINF) ? __neinf : int__nan);
 	  break;
 
 	case VAR:

--- oleo-1.99.16/src/ref.c	Tue Feb 13 23:38:06 2001
+++ oleo-1.99.16-patched/src/ref.c	Thu Sep  5 10:10:15 2002
@@ -263,7 +263,7 @@
 read_new_value (CELLREF row, CELLREF col, char *form, char *val)
 {
   unsigned char *new_bytes;
-  extern double __plinf, __neinf, __nan;
+  extern double __plinf, __neinf, int__nan;
 
   cur_row = row;
   cur_col = col;
@@ -348,7 +348,7 @@
 	  else if (!stricmp (nname, val))
 	    {
 	      SET_TYP (my_cell, TYP_FLT);
-	      my_cell->cell_flt = __nan;
+	      my_cell->cell_flt = int__nan;
 	    }
 	  else
 	    {

--- oleo-1.99.16/src/utils.c	Tue Feb 13 23:38:06 2001
+++ oleo-1.99.16-patched/src/utils.c	Thu Sep  5 10:36:45 2002
@@ -66,7 +66,7 @@
 #define _IOSTRG 0
 #endif
 
-extern int sys_nerr;
+/*extern int sys_nerr; */
 
 struct id
   {

--- oleo-1.99.16/src/io-utils.c	Tue Feb 13 23:38:06 2001
+++ oleo-1.99.16-patched/src/io-utils.c	Thu Sep 26 14:31:19 2002
@@ -71,7 +71,7 @@
 
 double __plinf;
 double __neinf;
-double __nan;
+double int__nan;
 
 char nname[] = "#NOT_A_NUMBER";
 char iname[] = "#INFINITY";
@@ -125,7 +125,7 @@
   __neinf = divide (-1., 0.);
   (void) signal (SIGFPE, ignore_sig);
 #endif
-  __nan = __plinf + __neinf;
+  int__nan = __plinf + __neinf;
 }
 
 
--- oleo-1.99.16/src/io-motif.c	Fri Mar  9 11:33:29 2001
+++ oleo-1.99.16-patched/src/io-motif.c	Thu Feb  7 01:18:02 2002
@@ -3296,7 +3296,7 @@
 #define	HELP_FILE_LENGTH	200000
 
 #ifndef	HTMLDIR
-#define	HTMLDIR			"/usr/local/share/oleo"
+#define	HTMLDIR			"@prefix@/share/oleo"
 #endif
 
 static void

