diff -N -r -u phylip.1/src/Makefile phylip/src/Makefile
--- phylip.1/src/Makefile	Sat Jan  3 01:06:00 2004
+++ phylip/src/Makefile	Sun Feb 15 19:49:51 2004
@@ -62,13 +62,13 @@
 #
 #
 #A minimal one
-CFLAGS =
+#CFLAGS =
 #
 # A basic one for debugging
 #CFLAGS  = -g 
 #
 # An optimized one for gcc
-#CFLAGS  = -O3 -fomit-frame-pointer
+CFLAGS  = -O3 -fomit-frame-pointer
 #
 # For some serious debugging using Gnu gcc
 #
@@ -119,16 +119,16 @@
 # ordinary compiles and the tree-drawing programs, CC and DC
 #
 # This one specifies the "cc" C compiler
-CC        = cc $(CFLAGS)
+#CC        = cc $(CFLAGS)
 #
 #  To use GCC instead, if it is not the compiler that "cc" invokes
-#CC        = gcc $(CFLAGS)
+CC        = gcc $(CFLAGS)
 #
 # This one specifies the "cc" C compiler for the Draw programs
-DC        = cc $(DFLAGS)
+#DC        = cc $(DFLAGS)
 #
 # To use GCC instead, if it is not the compiler that "cc" invokes
-#DC        = gcc $(DFLAGS)
+DC        = gcc $(DFLAGS)
 #
 # ----------------------------------------------------------------------------
 #  (After this point there should not be any reason to change anything)
diff -N -r -u phylip.1/src/phylip.h phylip/src/phylip.h
--- phylip.1/src/phylip.h	Sat Jan  3 01:06:00 2004
+++ phylip/src/phylip.h	Sun Feb 15 19:51:37 2004
@@ -67,7 +67,7 @@
 #define EBCDIC false
 #define INFILE "infile"
 #define OUTFILE "outfile"
-#define FONTFILE "fontfile" /* on unix this might be /usr/local/lib/fontfile */
+#define FONTFILE "@PREFIX@/share/phylip/fontfile" /* on unix this might be /usr/local/lib/fontfile */
 #define PLOTFILE "plotfile"
 #define INTREE "intree"
 #define INTREE2 "intree2"
