diff -u polyxmass-bin-0.8.4/configure polyxmass-bin-0.8.4.fink/configure
--- polyxmass-bin-0.8.4/configure	Mon Jan 24 06:45:52 2005
+++ polyxmass-bin-0.8.4.fink/configure	Mon Jan 24 23:01:55 2005
@@ -1844,11 +1844,11 @@
 
 
 GLIB_REQUIRED=2.4.6
-GTK_REQUIRED=2.4.10
+GTK_REQUIRED=2.4.9
 LIB_XML_REQUIRED=2.6.11
 LIBGNOMECANVAS_REQUIRED=2.6.1
-LIBRSVG_REQUIRED=2.7.2
-LIBGLADE_REQUIRED=2.4.0
+LIBRSVG_REQUIRED=2.6.5
+LIBGLADE_REQUIRED=2.3.6
 LIBPOLYXMASS_REQUIRED=0.8.4
 POLYXMASS_COMMON_REQUIRED=0.8.3
 
Common subdirectories: polyxmass-bin-0.8.4/glade and polyxmass-bin-0.8.4.fink/glade
Common subdirectories: polyxmass-bin-0.8.4/intl and polyxmass-bin-0.8.4.fink/intl
Common subdirectories: polyxmass-bin-0.8.4/m4 and polyxmass-bin-0.8.4.fink/m4
Common subdirectories: polyxmass-bin-0.8.4/po and polyxmass-bin-0.8.4.fink/po
Common subdirectories: polyxmass-bin-0.8.4/src and polyxmass-bin-0.8.4.fink/src
