--- R-1.8.1/configure.orig	Thu Jan 29 10:30:56 2004
+++ R-1.8.1/configure	Thu Jan 29 10:45:54 2004
@@ -2219,7 +2219,14 @@
 # Check whether --with-aqua or --without-aqua was given.
 if test "${with_aqua+set}" = set; then
   withval="$with_aqua"
-  want_aqua=yes
+  if test "${withval}" = no; then
+  use_aqua=no
+else
+  use_aqua=yes
+fi
+
+else
+  use_aqua=yes
 fi;
 
 ## Tcl/Tk.
--- R-1.8.1/src/modules/X11/dataentry.c.orig	Fri Jan 30 05:08:22 2004
+++ R-1.8.1/src/modules/X11/dataentry.c	Fri Jan 30 05:05:11 2004
@@ -29,7 +29,8 @@
 #include "Print.h"
 
 /* don't use X11 function prototypes (which tend to ...): */
-#define NeedFunctionPrototypes 0
+#define NeedFunctionPrototypes 1
+#include <X11/Xfuncproto.h>
 #include <X11/X.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
