--- root.orig/config/Makefile.macosx	Wed Jul  7 17:30:19 2004
+++ root.new/config/Makefile.macosx	Fri Aug 13 14:59:52 2004
@@ -23,9 +23,11 @@
 CXX           = g++
 CC            = gcc
 CXXFLAGS      = -pipe -Wall -W -Woverloaded-virtual -fsigned-char -fno-common \
-                -I/usr/X11R6/include $(EXTRA_CXXFLAGS) $(FINK_CXXFLAGS)
+                -I/usr/X11R6/include -I/usr/X11R6/include/freetype2 \
+		$(EXTRA_CXXFLAGS) $(FINK_CXXFLAGS)
 CFLAGS        = -pipe -W -Wall -fsigned-char -fno-common \
-                -I/usr/X11R6/include $(EXTRA_CFLAGS) $(FINK_CFLAGS)
+                -I/usr/X11R6/include -I/usr/X11R6/include/freetype2 \
+		$(EXTRA_CFLAGS) $(FINK_CFLAGS)
 CINTCXXFLAGS  = -pipe -Wall -W -Woverloaded-virtual -fsigned-char -fno-common \
                 -DG__REGEXP $(EXTRA_CXXFLAGS) $(FINK_CXXFLAGS) \
 		-DG__UNIX -DG__SHAREDLIB \
@@ -62,7 +64,7 @@
 
 # System libraries:
 SYSLIBS       = -lm $(EXTRA_LDFLAGS) $(FINK_LDFLAGS) $(OSTHREADLIBDIR) \
-                $(OSTHREADLIB) -ldl
+                $(OSTHREADLIB) -ldl -L/usr/X11R6/lib -lfreetype
 XLIBS         = $(XPMLIBDIR) $(XPMLIB) $(X11LIBDIR) -lX11
 CILIBS        = -lm $(EXTRA_LDFLAGS) $(FINK_LDFLAGS) -ldl
 
--- root.orig/Makefile	Fri Jul  9 05:32:58 2004
+++ root.new/Makefile	Fri Aug 13 15:04:52 2004
@@ -43,7 +43,7 @@
 ##### Modules to build #####
 
 MODULES       = build cint metautils utils base cont meta net zip clib matrix \
-                newdelete hist tree freetype graf g3d gpad gui minuit \
+                newdelete hist tree graf g3d gpad gui minuit \
                 histpainter treeplayer treeviewer proof physics postscript \
                 rint html eg geom geompainter vmc fumili mlp gedold ged quadp
 
@@ -71,6 +71,9 @@
 SYSTEMDO      = $(UNIXDO)
 endif
 endif
+endif
+ifneq ($(ARCH),macosx)
+MODULES      += freetype
 endif
 ifneq ($(OPENGLINCDIR),)
 ifneq ($(OPENGLULIB),)
--- root.orig/config/root-config.in	Wed Jul  7 17:30:19 2004
+++ root.new/config/root-config.in	Fri Aug 13 16:50:35 2004
@@ -251,7 +251,7 @@
    macosx_minor=`sw_vers | sed -n 's/ProductVersion://p' | cut -d . -f 2`
    finkdir=`which fink 2>&1 | sed -ne "s/\/bin\/fink//p"`
    auxcflags=`[ -d ${finkdir}/include ] && echo -I${finkdir}/include`
-   auxlibs="-lm `[ -d ${finkdir}/lib ] && echo -L${finkdir}/lib` -ldl"
+   auxlibs="-lm `[ -d ${finkdir}/lib ] && echo -L${finkdir}/lib` -ldl -L/usr/X11R6/lib -lfreetype"
    if [ $macosx_minor -le 3 ]; then
       forcelibs=$rootulibs
       forceglibs=$rootuglibs
--- root.orig/utils/src/rlibmap.cxx	Wed Jun  2 06:58:36 2004
+++ root.new/utils/src/rlibmap.cxx	Fri Aug 13 17:22:45 2004
@@ -341,7 +341,9 @@
    }
 
    if (replace) {
-#if !defined(WIN32) && !defined(__CYGWIN__) && !defined(__FreeBSD__)
+#if !defined(WIN32) && !defined(__CYGWIN__) && !defined(__FreeBSD__) && \
+    !(defined(__APPLE__) && (!defined(MAC_OS_X_VERSION_10_3) || \
+     (MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_3)))
       // lock file
       if (lockf(fileno(fp), F_LOCK, (off_t)1) == -1) {
          fprintf(stderr, "rlibmap: error locking output file\n");
@@ -357,7 +359,9 @@
    LibMap(solib, solibdeps, linkdefs, fullpath, fp);
 
    if (replace) {
-#if !defined(WIN32) && !defined(__CYGWIN__) && !defined(__FreeBSD__)
+#if !defined(WIN32) && !defined(__CYGWIN__) && !defined(__FreeBSD__) && \
+    !(defined(__APPLE__) && (!defined(MAC_OS_X_VERSION_10_3) || \
+     (MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_3)))
       // remove lock
       lseek(fileno(fp), 0, SEEK_SET);
       if (lockf(fileno(fp), F_ULOCK, (off_t)1) == -1) {
