--- scilab-3.0_ori/configure	Fri Jul  9 10:28:05 2004
+++ scilab-3.0/configure	Wed Aug 11 17:20:47 2004
@@ -22460,7 +22460,7 @@
      FC=g77
      FC_OPTIONS='-O2'
      FC_OPTIONS_O0=
-     FC_LDFLAGS='-L/sw/lib -ldl'
+     FC_LDFLAGS='-L/sw/lib -ldl -lcc_dynamic'
      FC_PICFLAGS='-fPIC'
      LD=ld
      LD_LDFLAGS=
--- scilab-3.0/routines/console/zzledt.c~	Wed Oct  6 18:10:01 2004
+++ scilab-3.0/routines/console/zzledt.c	Wed Jan  5 23:09:05 2005
@@ -68,7 +68,7 @@
 #if defined(netbsd) || defined(freebsd)
 #define TERMCAP
 #endif
-
+#define B42UNIX 
 #include <stdio.h>
 #include <ctype.h>
 #include "../machine.h"
