--- Scilab-3.0/configure.orig	Thu Jun 17 06:14:38 2004
+++ Scilab-3.0/configure	Thu Jun 17 06:17:23 2004
@@ -22449,19 +22449,19 @@
      STRIP=echo
      GCC=yes
      CC=cc
-     CC_OPTIONS='-no-cpp-precomp -O2 -I/sw/include -L/sw/lib'
+     CC_OPTIONS='-no-cpp-precomp -O2 -I@PREFIX@/include -L@PREFIX@/lib'
      CC_LDFLAGS=
      CC_PICFLAGS='-fPIC'
      # script shell f77 have bugs, get ours
      FC=g77
      FC_OPTIONS='-O2'
      FC_OPTIONS_O0=
-     FC_LDFLAGS='-L/sw/lib -ldl'
+     FC_LDFLAGS='-L@PREFIX@/lib -ldl -lcc_dynamic'
      FC_PICFLAGS='-fPIC'
      LD=ld
      LD_LDFLAGS=
      if test "$enable_debug" = yes; then
-       CC_OPTIONS='-g -I/sw/include '
+       CC_OPTIONS='-g -I@PREFIX@/include '
        FC_OPTIONS=-g
      fi
      if test "$WITH_G77" = yes; then
