--- scipy/scipy_core/setup.py.orig	Mon Feb  9 09:52:47 2004
+++ scipy/scipy_core/setup.py	Mon Feb  9 09:53:24 2004
@@ -18,7 +18,7 @@
 from scipy_distutils.misc_util import default_config_dict
 from scipy_distutils.misc_util import get_path, merge_config_dicts
 
-bundle_packages = ['scipy_distutils','scipy_test','scipy_base','weave']
+bundle_packages = ['scipy_test','scipy_base','weave']
 
 def setup_package():
     old_path = os.getcwd()
--- scipy/scipy_core/scipy_distutils/command/build_flib.py.orig	Tue Feb 10 13:29:07 2004
+++ scipy/scipy_core/scipy_distutils/command/build_flib.py	Tue Feb 10 13:30:36 2004
@@ -1041,7 +1041,7 @@
             self.libraries = ['gcc',g2c]
             self.library_dirs = gcc_lib_dir
         elif sys.platform == 'darwin':
-            self.libraries = [g2c]
+            self.libraries = [g2c,'cc_dynamic']
             self.library_dirs = gcc_lib_dir
         else:
             # On linux g77 does not need lib_directories to be specified.
@@ -1065,12 +1065,12 @@
         # only check for more optimization if g77 can handle it.
         if self.get_version():
             if sys.platform=='darwin':
-                if os.name != 'posix':
-                    # this should presumably correspond to Apple
-                    if cpu.is_ppc():
-                        opt = opt + ' -arch ppc '
-                    elif cpu.is_i386():
-                        opt = opt + ' -arch i386 '
+#               if os.name != 'posix':
+#                   # this should presumably correspond to Apple
+#                   if cpu.is_ppc():
+#                       opt = opt + ' -arch ppc '
+#                   elif cpu.is_i386():
+#                       opt = opt + ' -arch i386 '
                 for a in '601 602 603 603e 604 604e 620 630 740 7400 7450 750'\
                     '403 505 801 821 823 860'.split():
                     if getattr(cpu,'is_ppc%s'%a)():
--- scipy/setup.py.orig	Tue Feb 10 15:44:03 2004
+++ scipy/setup.py	Tue Feb 10 15:44:30 2004
@@ -119,7 +119,7 @@
         #'sparse',
         #'kiva','freetype','chaco','traits',
         ]
-    if sys.platform in ['win32','cygwin']:
+    if sys.platform in ['win32','cygwin','darwin']:
         # Fix sparse on windows.
         ignore_packages.append('sparse')
     setup_package(ignore_packages)
