diff -ru xcircuit-3.1.32/prototypes.h xcircuit-3.1.32-patched/prototypes.h
--- xcircuit-3.1.32/prototypes.h	2003-12-10 17:30:27.000000000 +0100
+++ xcircuit-3.1.32-patched/prototypes.h	2003-12-29 22:53:11.000000000 +0100
@@ -453,6 +453,8 @@
 extern void setjust(xcWidget, pointertype, labelptr, short);
 extern void setvjust(xcWidget, pointertype, caddr_t);
 extern void sethjust(xcWidget, pointertype, caddr_t);
+extern void setjustbit(xcWidget, pointertype, caddr_t);
+extern void setpinjustbit(xcWidget, pointertype, caddr_t);
 extern void setflipinv(xcWidget, pointertype, caddr_t);
 extern void setpinvisible(xcWidget, pointertype, caddr_t);
 extern void boxedit(xcWidget, pointertype, caddr_t);
