diff -ruN xcircuit-3.3.20/configure xcircuit-3.3.20.new/configure
--- xcircuit-3.3.20/configure 2005-07-11 13:29:44.000000000 -0400
+++ xcircuit-3.3.20.new/configure     2005-07-12 12:08:01.000000000 -0400
@@ -8875,7 +8875,7 @@
         LDDL_FLAGS="${LDDL_FLAGS} -undefined suppress -noprebind"
         SHLIB_CFLAGS="-I/sw/include -fno-common"
       fi
-      LDFLAGS="${LDFLAGS} ${LIB_SPEC} -bind_at_load"
+      LDFLAGS="${LDFLAGS} ${LIB_SPEC}"
       ;;

     *cygwin*)
