diff -urN xephem-3.6/GUI/xephem/Imakefile xephem-3.6.new/GUI/xephem/Imakefile
--- xephem-3.6/GUI/xephem/Imakefile	Sun Apr 18 16:00:57 2004
+++ xephem-3.6.new/GUI/xephem/Imakefile	Thu Jun 10 07:54:10 2004
@@ -80,8 +80,8 @@
 XCOMM Sometimes you need -lXp after -lXm
           XMLIB = -lXm
         DEPLIBS = $(DEPXMLIB) $(DEPXTOOLLIB) $(DEPXLIB)
- EXTRA_INCLUDES = -I../../libastro -I../../libip -I../../liblilxml -I../../libjpegd
-  LOCAL_LDFLAGS = -L../../libastro -L../../libip -L../../liblilxml -L../../libjpegd
+ EXTRA_INCLUDES = -I../../libastro -I../../libip -I../../liblilxml -I../../libjpegd  -I@FINKPREFIX@/include
+  LOCAL_LDFLAGS = -L../../libastro -L../../libip -L../../liblilxml -L../../libjpegd -L@FINKPREFIX@/lib
 LOCAL_LIBRARIES = $(XMLIB) $(XTOOLLIB) $(XLIB) -lastro -lip -llilxml -ljpegd
   SYS_LIBRARIES = -lm
        PROGRAMS = xephem
diff -urN xephem-3.6/xephem xephem-3.6.new/xephem
--- xephem-3.6/xephem	Thu Jan  1 09:00:00 1970
+++ xephem-3.6.new/xephem	Thu Jun 10 08:00:29 2004
@@ -0,0 +1,2 @@
+#!/bin/sh
+@FINKPREFIX@/lib/xephem/xephem -env TELHOME=@FINKPREFIX@/lib
