diff -Naur xephem-old/GUI/xephem/Imakefile xephem-3.5.2/GUI/xephem/Imakefile
--- xephem-old/GUI/xephem/Imakefile	Wed Oct 10 15:07:58 2001
+++ xephem-3.5.2/GUI/xephem/Imakefile	Sat Jan 19 21:09:55 2002
@@ -73,8 +73,8 @@
 XCOMM Sometimes you need -lXp after -lXm
           XMLIB = -lXm
         DEPLIBS = $(DEPXMLIB) $(DEPXTOOLLIB) $(DEPXLIB)
- EXTRA_INCLUDES = -I../../libastro -I../../libip
-  LOCAL_LDFLAGS = -L../../libastro -L../../libip
+ EXTRA_INCLUDES = -I../../libastro -I../../libip -I@FINKPREFIX@/include
+  LOCAL_LDFLAGS = -force_flat_namespace -L../../libastro -L../../libip -L@FINKPREFIX@/lib
 LOCAL_LIBRARIES = $(XMLIB) $(XTOOLLIB) $(XLIB) -lastro -lip
   SYS_LIBRARIES = -lm
        PROGRAMS = xephem
diff -Naur xephem-old/xephem xephem-3.5.2/xephem
--- xephem-old/xephem	Thu Jan  1 09:00:00 1970
+++ xephem-3.5.2/xephem	Sat Jan 19 21:15:50 2002
@@ -0,0 +1,2 @@
+#!/bin/sh
+@FINKPREFIX@/lib/xephem/xephem -env TELHOME=@FINKPREFIX@/lib
