--- scribus-1.0/scribus/scfonts_encoding.cpp~	Sun Mar  2 13:40:24 2003
+++ scribus-1.0/scribus/scfonts_encoding.cpp	Fri Aug  8 02:29:36 2003
@@ -14,6 +14,12 @@
 
 bool SCFonts_Encoding::read(QString name)
 {
+       QCString lang = getenv( "LANG" );
+	unsetenv( "LANG" );
+       QCString lc_ctype = getenv( "LC_CTYPE" );
+       unsetenv( "LC_CTYPE" );
+       QCString lc_all = getenv( "LC_ALL" );
+       unsetenv( "LC_ALL" );
 	QString ISO;
 	QString pfad = PREL;
 	pfad += "/lib/scribus/" + name;
@@ -23,6 +25,9 @@
 		QTextStream t(&tf);
 		ISO=t.read();
 		tf.close();
+		if (lang) setenv( "LANG", lang, 1);
+		if (lc_ctype) setenv( "LC_CTYPE", lc_ctype, 1);
+		if (lc_all) setenv( "LC_ALL", lc_all, 1);
 	}
 	else
 		return(false);
