diff -uNbr DBD-Pg-1.22/Pg.pm DBD-Pg-1.22-new/Pg.pm
--- DBD-Pg-1.22/Pg.pm	Wed Mar 26 23:28:37 2003
+++ DBD-Pg-1.22-new/Pg.pm	Sat Jun 14 21:56:39 2003
@@ -908,7 +908,7 @@
             # by pg_constraint. To maintain compatibility, check 
             # version number and execute appropriate query.
 	
-            my $version = pg_server_version( $dbh );
+            my $version = DBD::Pg::pg_server_version( $dbh );
             
             my $con_query = $version < 7.3
              ? "SELECT rcsrc FROM pg_relcheck WHERE rcname = '${table}_$col_name'"
diff -uNbr DBD-Pg-1.22/dbdimp.c DBD-Pg-1.22-new/dbdimp.c
--- DBD-Pg-1.22/dbdimp.c	Wed Mar 26 22:14:19 2003
+++ DBD-Pg-1.22-new/dbdimp.c	Sat Jun 14 22:35:13 2003
@@ -1483,8 +1483,10 @@
 		SvUTF8_off(sv);
 		/* XXX Is this all the character data types? */
 		if (18 == type || 25 == type || 1042 ==type || 1043 == type) {
+#if !(PERL_VERSION == 6 && PERL_SUBVERSION == 0)
 		    if (is_high_bit_set(val) && is_utf8_string(val, val_len))
 			SvUTF8_on(sv);
+#endif
 		}
 	    }
 #endif
