--- DODS/src/packages-3.4.2/libcurl/configure.in.orig	Wed May 26 20:12:33 2004
+++ DODS/src/packages-3.4.2/libcurl/configure.in	Wed May 26 20:13:34 2004
@@ -433,7 +433,7 @@
 
   case "$OPT_SSL" in
   yes)
-    EXTRA_SSL=/usr/local/ssl ;;
+    EXTRA_SSL=@PREFIX@/ssl ;;
   off)
     EXTRA_SSL= ;;
   *)
--- DODS/src/dap-3.4.5/configure	Thu May 27 06:51:08 2004
+++ DODS/src/dap-3.4.5/configure.orig	Thu May 27 06:50:37 2004
@@ -6570,7 +6570,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lrx  $LIBS"
+#LIBS="-lrx  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -6619,9 +6619,9 @@
 echo "$as_me:$LINENO: result: $ac_cv_lib_rx_rx_version_string" >&5
 echo "${ECHO_T}$ac_cv_lib_rx_rx_version_string" >&6
 if test $ac_cv_lib_rx_rx_version_string = yes; then
-  HAVE_RX=1; LIBS="$LIBS -lrx"
+  HAVE_RX=1; LIBS="$LIBS"
 else
-  packages="$packages librx"; HAVE_RX=1; LIBS="$LIBS -lrx"
+  packages="$packages librx"; HAVE_RX=1; LIBS="$LIBS"
 fi
 
 
--- DODS/src/nc3-dods-3.4.5/configure	Thu May 27 06:49:56 2004
+++ DODS/src/nc3-dods-3.4.5/configure.orig	Thu May 27 05:43:31 2004
@@ -3924,7 +3924,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lrx  $LIBS"
+#LIBS="-lrx  $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 3920 "configure"
 #include "confdefs.h"
@@ -3952,10 +3952,10 @@
 fi
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  HAVE_RX=1; LIBS="$LIBS -lrx"
+  HAVE_RX=1; LIBS="$LIBS"
 else
   echo "$ac_t""no" 1>&6
-packages="$packages librx"; HAVE_RX=1; LIBS="$LIBS -lrx"
+packages="$packages librx"; HAVE_RX=1; LIBS="$LIBS"
 fi
 
     
--- DODS/src/packages-3.4.2/libcurl/curl-config.in.orig	Thu May 27 10:29:13 2004
+++ DODS/src/packages-3.4.2/libcurl/curl-config.in	Thu May 27 10:29:28 2004
@@ -115,7 +115,7 @@
        	;;
 
     --libs)
-       	echo -L@libdir@ -lcurl @LDFLAGS@ @LIBS@
+       	echo -L@libdir@ -lcurl -lrx @LDFLAGS@ @LIBS@
        	;;
 
     *)
