diff -uNr ekg-1.3/examples/Makefile.in ekg-1.3-patched/examples/Makefile.in
--- ekg-1.3/examples/Makefile.in	Tue Jun 24 16:56:52 2003
+++ ekg-1.3-patched/examples/Makefile.in	Tue Nov  4 17:11:18 2003
@@ -5,30 +5,30 @@
 LIBS = @LIBS_LIBGADU@
 MAKE = @MAKE@
 
-OBJS = ../lib/libgadu.a
+OBJS = ../lib/libgadu.la
 
 all:	register-sync register-async conn-async send remind-async remind-sync httphash
 
 register-sync: register.c $(OBJS)
-	$(CC) $(CFLAGS) register.c -o register-sync $(OBJS) $(LIBS)
+	glibtool --tag=CC --mode=link $(CC) $(CFLAGS) register.c -o register-sync $(OBJS) $(LIBS)
 
 register-async: register.c $(OBJS)
-	$(CC) $(CFLAGS) register.c -o register-async -DASYNC $(OBJS) $(LIBS)
+	glibtool --tag=CC --mode=link $(CC) $(CFLAGS) register.c -o register-async -DASYNC $(OBJS) $(LIBS)
 
 remind-sync: remind.c $(OBJS)
-	$(CC) $(CFLAGS) remind.c -o remind-sync $(OBJS) $(LIBS)
+	glibtool --tag=CC --mode=link $(CC) $(CFLAGS) remind.c -o remind-sync $(OBJS) $(LIBS)
 
 remind-async: remind.c $(OBJS)
-	$(CC) $(CFLAGS) remind.c -o remind-async -DASYNC $(OBJS) $(LIBS)
+	glibtool --tag=CC --mode=link $(CC) $(CFLAGS) remind.c -o remind-async -DASYNC $(OBJS) $(LIBS)
 
 send:	send.c $(OBJS)
-	$(CC) $(CFLAGS) send.c -o send $(OBJS) $(LIBS)
+	glibtool --tag=CC --mode=link $(CC) $(CFLAGS) send.c -o send $(OBJS) $(LIBS)
 
 httphash: httphash.c $(OBJS)
-	$(CC) $(CFLAGS) httphash.c -o httphash $(OBJS) $(LIBS)
+	glibtool --tag=CC --mode=link $(CC) $(CFLAGS) httphash.c -o httphash $(OBJS) $(LIBS)
 
 conn-async: conn-async.c $(OBJS)
-	$(CC) $(CFLAGS) conn-async.c -o conn-async $(OBJS) $(LIBS)
+	glibtool --tag=CC --mode=link $(CC) $(CFLAGS) conn-async.c -o conn-async $(OBJS) $(LIBS)
 
 $(OBJS):
 	cd ../lib && $(MAKE)
diff -uNr ekg-1.3/lib/Makefile.in ekg-1.3-patched/lib/Makefile.in
--- ekg-1.3/lib/Makefile.in	Thu May 15 07:59:09 2003
+++ ekg-1.3-patched/lib/Makefile.in	Tue Nov  4 17:09:17 2003
@@ -25,9 +25,7 @@
 LOBJS = common.lo libgadu.lo search.lo pubdir.lo http.lo dcc.lo userlist.lo events.lo pubdir50.lo
 SRCS = $(OBJS:.o=.c)
 
-SHAREDMM = libgadu.so
-SHAREDM = $(SHAREDMM).$(VERSION_MAJOR)
-SHARED = $(SHAREDM).$(VERSION_MINOR)
+SHARED = libgadu.la
 STATIC = libgadu.a
 
 all:	@static@ @shared@
@@ -50,15 +48,16 @@
 
 #
 
-%.lo:	%.c
+%.o:	%.c
 	$(CC) -c $(CPPFLAGS) $(CFLAGS) -fPIC $< -o $@
 
+%.lo:	%.c
+	glibtool --tag=CC --mode=compile $(CC) -fno-common -c $(CPPFLAGS) $(CFLAGS) -fPIC $< -o $@
+
 shared:	dep $(SHARED)
 
 $(SHARED): $(LOBJS)
-	$(CC) -shared -Wl,-soname,libgadu.so.$(VERSION_MAJOR) -o $(SHARED) $(LOBJS) $(LIBS) -lc
-	$(LN_S) -f $(SHARED) $(SHAREDM)
-	$(LN_S) -f $(SHARED) $(SHAREDMM)
+	glibtool --tag=CC --mode=link $(CC) -o libgadu.la -rpath $(libdir) -version-info $(VERSION_MAJOR):$(VERSION_MINOR) $(LOBJS) $(LIBS)
 
 #
 
@@ -81,15 +80,13 @@
 	$(INSTALL) -m 644 libgadu.pc $(DESTDIR)$(libdir)/pkgconfig
 
 install-shared:	install-common @strip_libgadu_so@
-	$(INSTALL) -m 755 $(SHARED) $(DESTDIR)$(libdir)
-	cd $(DESTDIR)$(libdir) && $(LN_S) -f $(SHARED) $(SHAREDM)
-	cd $(DESTDIR)$(libdir) && $(LN_S) -f $(SHARED) $(SHAREDMM)
+	$(INSTALL) -m 755 -d $(DESTDIR)$(libdir)
+        glibtool --tag=CC --mode=install $(INSTALL) -c -p libgadu.la $(DESTDIR)$(libdir)/libgadu.la
 
 install-static:	install-common
 	$(INSTALL) -m 755 $(STATIC) $(DESTDIR)$(libdir)
 
 strip-libgadu-so:
-	$(STRIP) $(SHARED)
 
 #
 
diff -uNr ekg-1.3/src/Makefile.in ekg-1.3-patched/src/Makefile.in
--- ekg-1.3/src/Makefile.in	Mon May 19 09:36:41 2003
+++ ekg-1.3-patched/src/Makefile.in	Tue Nov  4 17:13:02 2003
@@ -25,11 +25,11 @@
 
 all:	dep ekg @ioctld@
 
-ekg:	$(OBJS) @libgadu_a@
-	$(CC) $(CFLAGS) -o ekg $(OBJS) -L../lib @lgadu@ @libgadu_a@ $(LDFLAGS) $(LIBS)
+ekg:	$(OBJS) ../lib/libgadu.la
+	glibtool --tag=CC --mode=link $(CC) $(CFLAGS) -o ekg $(OBJS) -L../lib ../lib/libgadu.la $(LDFLAGS) $(LIBS)
 
 ioctld:	ioctld.c
-	$(CC) $(CFLAGS) ioctld.c -o ioctld $(LIBS) @IOCTLD_OBJS@
+	glibtool --tag=CC --mode=link $(CC) $(CFLAGS) ioctld.c -o ioctld $(LIBS) @IOCTLD_OBJS@
 
 dep:	.depend
 
@@ -40,19 +40,17 @@
 
 install: all install-ekg @install_ioctld@
 
-install-ekg:	@strip_ekg@
+install-ekg:
 	$(INSTALL) -d $(DESTDIR)$(bindir)
-	$(INSTALL) -m 755 ekg $(DESTDIR)$(bindir)
+	glibtool --tag=CC --mode=install $(INSTALL) -m 755 ekg $(DESTDIR)$(bindir)
 
-install-ioctld:	@strip_ioctld@
+install-ioctld:
 	$(INSTALL) -d $(DESTDIR)$(libexecdir)
-	$(INSTALL) -m 755 ioctld $(DESTDIR)$(libexecdir)
+	glibtool --tag=CC --mode=install $(INSTALL) -m 755 ioctld $(DESTDIR)$(libexecdir)
 
 strip-ekg:
-	$(STRIP) ekg
 
 strip-ioctld:
-	$(STRIP) ioctld
 
 #
 
