diff -Nurd -x'*~' elinks-0.10.3.orig/configure elinks-0.10.3/configure
--- elinks-0.10.3.orig/configure	2005-02-28 16:17:42.000000000 -0500
+++ elinks-0.10.3/configure	2005-03-29 18:05:07.000000000 -0500
@@ -7269,6 +7269,9 @@
 
 if test "$disable_openssl" = yes; then
 	cf_result="not used"
+	LIBS="-L@PREFIX@/lib $LIBS_X"
+	CFLAGS="$CFLAGS_X -I@PREFIX@/include"
+	CPPFLAGS="$CPPFLAGS_X -I@PREFIX@/include"
 else
 	for ssldir in "$openssl_withval" "" /usr /usr/local/openssl \
 			 /usr/lib/openssl /usr/local/ssl \
