diff -Naur ettercap-0.6.b.orig/plugins/H03_hydra1/Makefile.in ettercap-0.6.b/plugins/H03_hydra1/Makefile.in
--- ettercap-0.6.b.orig/plugins/H03_hydra1/Makefile.in	Thu Oct 30 19:03:57 2003
+++ ettercap-0.6.b/plugins/H03_hydra1/Makefile.in	Fri Oct 31 11:49:22 2003
@@ -3,7 +3,8 @@
 DEFS     = @PICFLAG@ @DEFS@ -D@OS@
 
 CFLAG    = @CFLAGS@
-COPTS    = -I../..
+COPTS    = -I../.. -I@FINK_PREFIX@/include
+LIBS     = -lcrypto
 
 PROG     = H03_hydra1
 
@@ -13,7 +14,7 @@
 all: $(PROG)
 
 $(PROG):
-	$(CC) $(ALLOPT) $(DEFS) $(PROG).c @CWPLUGINLIBS@ -o ec_$(PROG).so
+	$(CC) $(ALLOPT) $(DEFS) $(LIBS) $(PROG).c @CWPLUGINLIBS@ -o ec_$(PROG).so
 	@echo
 	@echo "@SB@$(PROG) plugin is ready...@EB@"
 	@echo
diff -Naur ettercap-0.6.b.orig/plugins/H10_phantom/H10_phantom.c ettercap-0.6.b/plugins/H10_phantom/H10_phantom.c
--- ettercap-0.6.b.orig/plugins/H10_phantom/H10_phantom.c	Thu Oct 30 19:03:57 2003
+++ ettercap-0.6.b/plugins/H10_phantom/H10_phantom.c	Fri Oct 31 12:15:35 2003
@@ -41,7 +41,7 @@
 
 #include <stdlib.h>
 #include <sys/param.h>
-#include <arpa/nameser.h>
+#include <arpa/nameser_compat.h>
 #include <resolv.h>
 
 #ifndef HAVE_NS_GET
diff -Naur ettercap-0.6.b.orig/plugins/H10_phantom/Makefile.in ettercap-0.6.b/plugins/H10_phantom/Makefile.in
--- ettercap-0.6.b.orig/plugins/H10_phantom/Makefile.in	Thu Oct 30 19:03:57 2003
+++ ettercap-0.6.b/plugins/H10_phantom/Makefile.in	Fri Oct 31 12:11:18 2003
@@ -4,6 +4,7 @@
 
 CFLAG    = @CFLAGS@
 COPTS    = -I../..
+LIBS     = -lresolv
 
 PROG     = H10_phantom
 
@@ -13,7 +14,7 @@
 all: $(PROG)
 
 $(PROG):
-	$(CC) $(ALLOPT) $(DEFS) $(PROG).c @CWPLUGINLIBS@ -o ec_$(PROG).so
+	$(CC) $(ALLOPT) $(DEFS) $(LIBS) $(PROG).c @CWPLUGINLIBS@ -o ec_$(PROG).so
 	@echo
 	@echo "@SB@$(PROG) plugin is ready...@EB@"
 	@echo
diff -Naur ettercap-0.6.b.orig/plugins/Makefile.in ettercap-0.6.b/plugins/Makefile.in
--- ettercap-0.6.b.orig/plugins/Makefile.in	Thu Oct 30 19:03:58 2003
+++ ettercap-0.6.b/plugins/Makefile.in	Thu Oct 30 19:08:58 2003
@@ -4,7 +4,7 @@
 
 prefix      = @prefix@
 exec_prefix = @exec_prefix@
-PLUGDIR     = @libdir@/@PROG@
+PLUGDIR     = @FINK_PREFIX@/share/@PROG@/plugins/
 
 all: plugins
 
diff -Naur ettercap-0.6.b.orig/plugins/phantom/Makefile.in ettercap-0.6.b/plugins/phantom/Makefile.in
--- ettercap-0.6.b.orig/plugins/phantom/Makefile.in	Thu Oct 30 19:03:57 2003
+++ ettercap-0.6.b/plugins/phantom/Makefile.in	Fri Oct 31 11:45:33 2003
@@ -4,6 +4,7 @@
 
 CFLAG    = @CFLAGS@
 COPTS    = -I../..
+LIBS     = -lresolv
 
 PROG     = phantom
 
@@ -13,7 +14,7 @@
 all: $(PROG)
 
 $(PROG):
-	$(CC) $(ALLOPT) $(DEFS) $(PROG).c @CWPLUGINLIBS@ -o ec_$(PROG).so
+	$(CC) $(ALLOPT) $(DEFS) $(LIBS) $(PROG).c @CWPLUGINLIBS@ -o ec_$(PROG).so
 	@echo
 	@echo "@SB@$(PROG) plugin is ready...@EB@"
 	@echo
diff -Naur ettercap-0.6.b.orig/plugins/phantom/phantom.c ettercap-0.6.b/plugins/phantom/phantom.c
--- ettercap-0.6.b.orig/plugins/phantom/phantom.c	Thu Oct 30 19:03:57 2003
+++ ettercap-0.6.b/plugins/phantom/phantom.c	Fri Oct 31 09:38:13 2003
@@ -43,7 +43,7 @@
 
 #include <stdlib.h>
 #include <sys/param.h>
-#include <arpa/nameser.h>
+#include <arpa/nameser_compat.h>
 #include <resolv.h>
 
 #ifndef HAVE_NS_GET
diff -Naur ettercap-0.6.b.orig/src/ec_plugins.c ettercap-0.6.b/src/ec_plugins.c
--- ettercap-0.6.b.orig/src/ec_plugins.c	Thu Oct 30 19:03:57 2003
+++ ettercap-0.6.b/src/ec_plugins.c	Fri Oct 31 11:50:09 2003
@@ -50,7 +50,7 @@
    #define RTLD_NOW DL_LAZY
 #endif
 
-#if defined(OPENBSD) || defined(MACOSX)
+#if defined(OPENBSD)
    #define SYM_PREFIX "_"     // for the symbols loaded with dlsym
 #else
    #define SYM_PREFIX ""
--- ettercap-0.6.b.orig/Makefile.in	Thu Oct 30 19:03:58 2003
+++ ettercap-0.6.b/Makefile.in	Fri Oct 31 12:48:04 2003
@@ -20,16 +20,16 @@
 BINDIR   = @sbindir@
 MANDIR   = @mandir@
 DATADIR  = @datadir@/@PROG@
-PLUGDIR  = @libdir@/@PROG@
+PLUGDIR  = @FINK_PREFIX@/share/@PROG@/plugins/
 DOCDIR   = ${prefix}/share/doc/@PROG@
 DOC      = COPYING README README.PLUGINS README.WIN32 CHANGELOG HISTORY \
            AUTHORS TODO TODO.WIN32 THANKS KNOWN-BUGS PORTINGS INSTALL \
            ./plugins/H01_zaratan/ZARATAN.HOWTO ./plugins/H09_roper/ROPER.HOWTO \
-	   ./plugins/H03_hydra1/HYDRA.HOWTO    
+           ./plugins/confusion/CONFUSION.HOWTO
 
 DEBUG    = @DEBUG@
 CFLAG    = @E_CFLAGS@
-COPTS    = -I. -Isrc/include @COPTS@ @GTK_CFLAGS@
+COPTS    = -I. -I@FINK_PREFIX@/include -Isrc/include @COPTS@ @GTK_CFLAGS@
 
 PROG     = @PROG@
 VERSION  = @VERSION@
