diff -Naur evolution-1.4.5-old/e-util/Makefile.in evolution-1.4.5/e-util/Makefile.in
--- evolution-1.4.5-old/e-util/Makefile.in	Tue Jan  6 14:32:26 2004
+++ evolution-1.4.5/e-util/Makefile.in	Tue Jan  6 14:44:06 2004
@@ -236,7 +236,7 @@
 	$(E_UTIL_CFLAGS)
 
 
-privlib_LTLIBRARIES = libeutil.la libedb3util.la libeconduit.la
+privlib_LTLIBRARIES = libeutil.la libedb3util.la
 noinst_LTLIBRARIES = libeutil-static.la
 
 eutilinclude_HEADERS = \
diff -Naur evolution-1.4.5-old/ltmain.sh evolution-1.4.5/ltmain.sh
--- evolution-1.4.5-old/ltmain.sh	Tue Jan  6 14:32:29 2004
+++ evolution-1.4.5/ltmain.sh	Tue Jan  6 14:43:29 2004
@@ -4050,10 +4050,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test $need_relink = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test $need_relink = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
diff -Naur evolution-1.4.5-old/po/Makefile.in.in evolution-1.4.5/po/Makefile.in.in
--- evolution-1.4.5-old/po/Makefile.in.in	Tue Jan  6 14:32:33 2004
+++ evolution-1.4.5/po/Makefile.in.in	Tue Jan  6 14:43:29 2004
@@ -28,7 +28,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 libdir = @libdir@
-localedir = $(libdir)/locale
+localedir = $(datadir)/locale
 gnulocaledir = $(datadir)/locale
 gettextsrcdir = $(datadir)/glib-2.0/gettext/po
 subdir = po
