diff -Naur galeon-1.3.11a/mozilla/Makefile.in galeon-new/mozilla/Makefile.in
--- galeon-1.3.11a/mozilla/Makefile.in	Fri Dec 19 11:57:30 2003
+++ galeon-new/mozilla/Makefile.in	Fri Jan 23 00:09:18 2004
@@ -254,7 +254,7 @@
 
 libgaleon_xpcom_la_LIBADD = $(MOZILLA_COMPONENT_LIBS)
 
-libgaleon_xpcom_la_LDFLAGS = -avoid-version
+libgaleon_xpcom_la_LDFLAGS = -avoid-version -module
 
 galeon_libdir = $(pkglibdir)/components
 galeon_lib_LTLIBRARIES = libgaleon-xpcom.la
@@ -434,7 +434,9 @@
 	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
 	$(AM_CXXFLAGS) $(CXXFLAGS)
 CXXLD = $(CXX)
-CXXLINK = $(LIBTOOL) --mode=link $(CXXLD) $(AM_CXXFLAGS) $(CXXFLAGS) \
+CXXLINK = if [ ! -x gcc ]; then ln -s `which g++` gcc; fi; \
+	PATH=".:$(PATH)" \
+	$(LIBTOOL) --mode=link $(CXXLD) $(AM_CXXFLAGS) $(CXXFLAGS) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
 DIST_SOURCES = $(libgaleon_xpcom_la_SOURCES) \
 	$(libmozillaembed_la_SOURCES)
