diff -Naur galeon-old/doc/C/Makefile.in galeon-new/doc/C/Makefile.in
--- galeon-old/doc/C/Makefile.in	Fri Dec 19 11:57:30 2003
+++ galeon-new/doc/C/Makefile.in	Mon Jan 26 01:45:10 2004
@@ -462,7 +462,7 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -Naur galeon-old/mozilla/Makefile.in galeon-new/mozilla/Makefile.in
--- galeon-old/mozilla/Makefile.in	Fri Dec 19 11:57:30 2003
+++ galeon-new/mozilla/Makefile.in	Mon Jan 26 01:31:58 2004
@@ -254,7 +254,7 @@
 
 libgaleon_xpcom_la_LIBADD = $(MOZILLA_COMPONENT_LIBS)
 
-libgaleon_xpcom_la_LDFLAGS = -avoid-version
+libgaleon_xpcom_la_LDFLAGS = -avoid-version -module
 
 galeon_libdir = $(pkglibdir)/components
 galeon_lib_LTLIBRARIES = libgaleon-xpcom.la
@@ -434,7 +434,9 @@
 	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
 	$(AM_CXXFLAGS) $(CXXFLAGS)
 CXXLD = $(CXX)
-CXXLINK = $(LIBTOOL) --mode=link $(CXXLD) $(AM_CXXFLAGS) $(CXXFLAGS) \
+CXXLINK = if [ ! -x gcc ]; then ln -s `which g++` gcc; fi; \
+	PATH=".:$(PATH)" \
+	$(LIBTOOL) --mode=link $(CXXLD) $(AM_CXXFLAGS) $(CXXFLAGS) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
 DIST_SOURCES = $(libgaleon_xpcom_la_SOURCES) \
 	$(libmozillaembed_la_SOURCES)
