diff -uNr gpgme-1.0.3/gpgme/Makefile.in gpgme-1.0.3-new/gpgme/Makefile.in
--- gpgme-1.0.3/gpgme/Makefile.in	2005-06-20 15:35:24.000000000 -0400
+++ gpgme-1.0.3-new/gpgme/Makefile.in	2006-08-08 21:37:49.000000000 -0400
@@ -335,13 +335,14 @@
 libgpgme_pth_la_SOURCES = ath.h ath-pth.c
 AM_CPPFLAGS = $(assuan_cppflags) @GPG_ERROR_CFLAGS@
 libgpgme_la_LDFLAGS = $(libgpgme_version_script_cmd) -version-info \
-	@LIBGPGME_LT_CURRENT@:@LIBGPGME_LT_REVISION@:@LIBGPGME_LT_AGE@
+	@LIBGPGME_LT_CURRENT@:@LIBGPGME_LT_REVISION@:@LIBGPGME_LT_AGE@ \
+	@PTH_LDFLAGS@
 
 libgpgme_la_DEPENDENCIES = libgpgme-real.la $(assuan_libobjs) \
 	@LTLIBOBJS@ $(srcdir)/libgpgme.vers
 
 libgpgme_la_LIBADD = libgpgme-real.la $(assuan_libobjs) @LTLIBOBJS@ \
-	@GPG_ERROR_LIBS@
+	@PTH_LIBS@ @GPG_ERROR_LIBS@
 
 libgpgme_pthread_la_LDFLAGS = $(libgpgme_version_script_cmd) -version-info \
 	@LIBGPGME_LT_CURRENT@:@LIBGPGME_LT_REVISION@:@LIBGPGME_LT_AGE@
diff -uNr gpgme-1.0.3/gpgme/gpgme-config.in gpgme-1.0.3-new/gpgme/gpgme-config.in
--- gpgme-1.0.3/gpgme/gpgme-config.in	2004-05-21 15:49:38.000000000 -0400
+++ gpgme-1.0.3-new/gpgme/gpgme-config.in	2005-07-05 09:43:29.000000000 -0400
@@ -21,6 +21,9 @@
 # Configure thread packages.
 thread_modules=""
 
+# default to pthread
+thread_module="pthread"
+
 @HAVE_PTH_TRUE@thread_modules="$thread_modules pth"
 libs_pth="@PTH_LDFLAGS@ @PTH_LIBS@"
 cflags_pth="@PTH_CFLAGS@"
diff -Nurd -x'*~' gpgme-1.0.3.orig/configure gpgme-1.0.3/configure
--- gpgme-1.0.3.orig/configure	2005-06-20 15:35:26.000000000 -0400
+++ gpgme-1.0.3/configure	2005-11-28 13:06:27.000000000 -0500
@@ -19490,8 +19490,6 @@
 int
 main ()
 {
-        int main(int argc, char *argv)
-        {
             FILE *fp;
             if (!(fp = fopen("conftestval", "w")))
                 exit(1);
@@ -19504,8 +19502,6 @@
             fprintf(fp, "yes");
             fclose(fp);
             exit(0);
-        }
-
   ;
   return 0;
 }
