Index: kopete/protocols/irc/libkirc/dcchandler.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/irc/libkirc/dcchandler.cpp,v
retrieving revision 1.22
diff -u -r1.22 dcchandler.cpp
--- kopete/protocols/irc/libkirc/dcchandler.cpp	29 Aug 2003 11:09:13 -0000	1.22
+++ kopete/protocols/irc/libkirc/dcchandler.cpp	4 Nov 2003 20:25:03 -0000
@@ -19,6 +19,7 @@
     *************************************************************************
 */
 
+#include "config.h"
 #include "dcchandler.h"
 
 #include <kglobal.h>
@@ -28,7 +29,7 @@
 #include <qregexp.h>
 #include <qtextcodec.h>
 
-#ifdef NEED_TYPES_H
+#ifdef HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif
 #ifdef HAVE_INTTYPES_H
