diff -Nurd -x'*~' libfwbuilder-2.1.9.orig/configure libfwbuilder-2.1.9/configure
--- libfwbuilder-2.1.9.orig/configure	2007-02-10 22:38:57.000000000 -0500
+++ libfwbuilder-2.1.9/configure	2007-02-14 01:01:01.000000000 -0500
@@ -5891,7 +5891,7 @@
 
   if test Z$openssl_prefix != Z ; then
 	OPENSSL_CFLAGS="-I${openssl_prefix}/include"
-	OPENSSL_LIBS="-L${openssl_prefix}/lib -R${openssl_prefix}/lib -lssl -lcrypto"
+	OPENSSL_LIBS="-L${openssl_prefix}/lib -lssl -lcrypto"
 	OPENSSL_RPATH="${openssl_prefix}/lib"
   else
 
@@ -5901,7 +5901,7 @@
 		echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6
 		OPENSSL_CFLAGS="-I/usr/local/ssl/include"
-		OPENSSL_LIBS="-L/usr/local/ssl/lib -R/usr/local/ssl/lib -lssl -lcrypto"
+		OPENSSL_LIBS="-L/usr/local/ssl/lib -lssl -lcrypto"
 		OPENSSL_RPATH="/usr/local/ssl/lib"
 	else
 	        echo "$as_me:$LINENO: result: no" >&5
@@ -5912,7 +5912,7 @@
 			echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6
 			OPENSSL_CFLAGS="-I/usr/local/include"
-			OPENSSL_LIBS="-L/usr/local/lib -R/usr/local/lib -lssl -lcrypto"
+			OPENSSL_LIBS="-L/usr/local/lib -lssl -lcrypto"
 			OPENSSL_RPATH="/usr/local/lib"
 		else
 			echo "$as_me:$LINENO: result: no. Assuming it is in /usr" >&5
diff -Nurd -x'*~' libfwbuilder-2.1.9.orig/qmake.inc.in libfwbuilder-2.1.9/qmake.inc.in
--- libfwbuilder-2.1.9.orig/qmake.inc.in	2006-06-22 00:50:45.000000000 -0400
+++ libfwbuilder-2.1.9/qmake.inc.in	2007-02-14 01:43:31.000000000 -0500
@@ -36,7 +36,7 @@
 }
 
 unix {
-	INCLUDEPATH            += /usr/include @XML_CFLAGS_Q@ @XSLT_CFLAGS_Q@ @PTHREAD_CFLAGS_Q@ @OPENSSL_CFLAGS_Q@ @LWRES_CFLAGS_Q@
+	INCLUDEPATH            += @XML_CFLAGS_Q@ @XSLT_CFLAGS_Q@ @PTHREAD_CFLAGS_Q@ @OPENSSL_CFLAGS_Q@ @LWRES_CFLAGS_Q@
 	LIBS                   += @PTHREAD_LIBS@ @XML_LIBS@ @XSLT_LIBS@ @OPENSSL_LIBS@ @LIBSNMP_LIBS@ @LIB_RESOLV@ @LWRES_LIBS@ @LIBS@
 }
 
diff -Nurd -x'*~' libfwbuilder-2.1.9.orig/src/fwcompiler/fwcompiler.pro libfwbuilder-2.1.9/src/fwcompiler/fwcompiler.pro
--- libfwbuilder-2.1.9.orig/src/fwcompiler/fwcompiler.pro	2006-03-20 11:47:23.000000000 -0500
+++ libfwbuilder-2.1.9/src/fwcompiler/fwcompiler.pro	2007-02-14 00:20:42.000000000 -0500
@@ -24,9 +24,7 @@
 headers.files =$$HEADERS
 headers.path  =$(DDIR)$$prefix/include/fwb-2.1/fwcompiler
 
-macx {
-        LIBS += -L../fwbuilder -lfwbuilder
-}
+LIBS += ../fwbuilder/libfwbuilder.dylib
 
 TARGET    = fwcompiler
 
diff -Nurd -x'*~' libfwbuilder-2.1.9.orig/src/test/test.pro libfwbuilder-2.1.9/src/test/test.pro
--- libfwbuilder-2.1.9.orig/src/test/test.pro	2004-09-08 01:34:53.000000000 -0400
+++ libfwbuilder-2.1.9/src/test/test.pro	2007-02-14 01:46:33.000000000 -0500
@@ -21,7 +21,7 @@
 	CONFIG   += warn_on debug
 	INCLUDEPATH += ../
 
-	LIBS     += -L../fwbuilder -lfwbuilder -pthread -lxslt -lxml2 -lz -lm -lssl -lcrypto /usr/lib/libresolv.a
+	LIBS     += ../fwbuilder/libfwbuilder.dylib -pthread -lxslt -lxml2 -lz -lm -lssl -lcrypto /usr/lib/libresolv.a
 }
 
 win32 {
diff -Nurd -x'*~' libfwbuilder-2.1.9.orig/src/unit_tests/unit-tests.pro libfwbuilder-2.1.9/src/unit_tests/unit-tests.pro
--- libfwbuilder-2.1.9.orig/src/unit_tests/unit-tests.pro	2005-10-13 11:52:51.000000000 -0400
+++ libfwbuilder-2.1.9/src/unit_tests/unit-tests.pro	2007-02-14 01:46:23.000000000 -0500
@@ -19,7 +19,7 @@
     CONFIG   += warn_on debug
     INCLUDEPATH += ../
 
-    LIBS     += -L../fwbuilder -lfwbuilder
+    LIBS     += ../fwbuilder/libfwbuilder.dylib
 }
 
 win32 {
