--- nessus-libraries/libnessus/harglists.c	2005-04-04 01:31:05.000000000 -0700
+++ nessus-libraries-patched/libnessus/harglists.c	2005-04-04 01:31:24.000000000 -0700
@@ -503,6 +503,7 @@
 }
 
 
+static void **harg_walk_next_ptr (hargwalk*);
 static void
 do_harg_dump
   (harglst *a,
@@ -510,7 +511,6 @@
 {
   hargwalk *w ;
   harg **R, *r ;
-  static void **harg_walk_next_ptr (hargwalk*);
 
   if(a == 0 || (w = harg_walk_init (a)) == 0) {
     do_printf ("-error; no such list!\n",0,0,0,0,0);
