diff -uNr libpqxx-2.4.0/configure libpqxx-2.4.0-new/configure
--- libpqxx-2.4.0/configure	Thu Oct 14 07:55:55 2004
+++ libpqxx-2.4.0-new/configure	Mon Oct 25 10:03:32 2004
@@ -20195,7 +20195,8 @@
 { echo "$as_me:$LINENO: using PostgreSQL libraries at $with_postgres_lib" >&5
 echo "$as_me: using PostgreSQL libraries at $with_postgres_lib" >&6;}
 
-
+with_postgres_include="@INSTPREFIX@/include/postgresql"
+with_postgres_lib="@INSTPREFIX@/lib"
 
 POSTGRES_LIB="-L${with_postgres_lib} -R${with_postgres_lib} -lpq"
 
@@ -23495,8 +23496,8 @@
 # 1. Name of the configuration item, e.g. PQXX_HAVE_PTRDIFF_T
 # 2. Publication marker: public or internal
 # 3. A single environmental factor determining the item, e.g. libpq or compiler
-PUBLICATIONS=`cut ${srcdir}/configitems -f 2 | sort -u`
-FACTORS=`cut ${srcdir}/configitems -f 3 | sort -u`
+PUBLICATIONS=`cut -f 2 ${srcdir}/configitems | sort -u`
+FACTORS=`cut -f 3 ${srcdir}/configitems | sort -u`
 for publication in $PUBLICATIONS ; do
 	for factor in $FACTORS ; do
 		CONFIGFILE="include/pqxx/config-$publication-$factor.h"
