diff -ru monotone-0.29/cmd_automate.cc monotone-0.29-patched/cmd_automate.cc
--- monotone-0.29/cmd_automate.cc	2006-08-20 20:46:05.000000000 -0500
+++ monotone-0.29-patched/cmd_automate.cc	2006-09-02 21:11:50.000000000 -0500
@@ -13,6 +13,7 @@
 
 #include <boost/function.hpp>
 #include <boost/bind.hpp>
+#include <boost/function.hpp>
 
 #include "cmd.hh"
 
diff -ru monotone-0.29/configure monotone-0.29-patched/configure
--- monotone-0.29/configure	2006-08-20 20:46:30.000000000 -0500
+++ monotone-0.29-patched/configure	2006-09-02 21:13:49.000000000 -0500
@@ -3254,13 +3254,13 @@
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g -O0"
   else
     CFLAGS="-g"
   fi
 else
   if test "$GCC" = yes; then
-    CFLAGS="-O2"
+    CFLAGS="-O0"
   else
     CFLAGS=
   fi
@@ -4525,13 +4525,13 @@
   CXXFLAGS=$ac_save_CXXFLAGS
 elif test $ac_cv_prog_cxx_g = yes; then
   if test "$GXX" = yes; then
-    CXXFLAGS="-g -O2"
+    CXXFLAGS="-g -O0"
   else
     CXXFLAGS="-g"
   fi
 else
   if test "$GXX" = yes; then
-    CXXFLAGS="-O2"
+    CXXFLAGS="-O0"
   else
     CXXFLAGS=
   fi
diff -ru monotone-0.29/hash_map.hh monotone-0.29-patched/hash_map.hh
--- monotone-0.29/hash_map.hh	2006-08-20 20:46:04.000000000 -0500
+++ monotone-0.29-patched/hash_map.hh	2006-09-02 21:11:50.000000000 -0500
@@ -3,6 +3,7 @@
 
 #include "config.h"
 
+#undef _T
 #include <functional>
 namespace hashmap {
 
