diff -Naur p2_v123/p2/conf.php p2_v123.fink/p2/conf.php
--- p2_v123/p2/conf.php	Fri Dec 24 01:33:58 2004
+++ p2_v123.fink/p2/conf.php	Sat Dec 25 02:16:53 2004
@@ -3,7 +3,7 @@
 
 $_conf['p2version'] = '1.2.3';
 
-$_conf['p2name'] = 'p2';	// p2の名前。
+$_conf['p2name'] = 'Fink/p2';	// p2の名前。
 
 // $_conf['p2name'] = 'P2';	// p2の名前。
 
diff -Naur p2_v123/p2/conf_user.php p2_v123.fink/p2/conf_user.php
--- p2_v123/p2/conf_user.php	Thu Dec 23 04:19:54 2004
+++ p2_v123.fink/p2/conf_user.php	Sat Dec 25 02:21:18 2004
@@ -12,10 +12,10 @@
 
 /* PATH =========================================================== */
 // 取得スレッドのdat & idx データ保存ディレクトリ(パーミッションは707に) 
-$datdir = "./data";	// ("./data")
+$datdir = "@PREFIX@/var/p2";	// ("./data")
 
 // 初期設定データ保存ディレクトリ(パーミッションは707に) 
-$prefdir = "./data";	// ("./data")
+$prefdir = "@PREFIX@/var/p2";	// ("./data")
 
 $first_page = "first_cont.php";	// ("first_cont.php") 右下部分に最初に表示されるページ。オンラインURLも可。
 
diff -Naur p2_v123/p2.conf p2_v123.fink/p2.conf
--- p2_v123/p2.conf	Thu Jan  1 09:00:00 1970
+++ p2_v123.fink/p2.conf	Sat Dec 25 02:17:27 2004
@@ -0,0 +1,6 @@
+Alias "/p2" "@PREFIX@/share/p2/php"
+<Directory "@PREFIX@/share/p2/php">
+	AddType application/x-httpd-php .php
+	AddType application/x-httpd-php-source .phps
+	DirectoryIndex index.html index.php
+</Directory>
