diff -uNr PyGreSQL-3.4/setup.py PyGreSQL-3.4-new/setup.py
--- PyGreSQL-3.4/setup.py	Tue Jan 13 07:29:57 2004
+++ PyGreSQL-3.4-new/setup.py	Thu Jun  3 20:39:08 2004
@@ -30,8 +30,8 @@
 	optional_libs=[ 'libpqdll', 'wsock32', 'advapi32' ]
 	data_files = [ 'libpq.dll' ]
 else:
-	include_dirs=['/usr/include/pgsql']
-	library_dirs=['/usr/lib/pgsql']
+	include_dirs=['@INSTPREFIX@/include/postgresql','@INSTPREFIX@/include']
+	library_dirs=['@INSTPREFIX@/lib/postgresql-7.4', '@INSTPREFIX@/lib']
 	optional_libs=['pq']
 	data_files = []
 
