diff -Naur -x'*~' pwlib/include/ptlib/unix/ptlib/pmachdep.h pwlib.new/include/ptlib/unix/ptlib/pmachdep.h
--- pwlib/include/ptlib/unix/ptlib/pmachdep.h	2003-05-06 01:59:12.000000000 -0500
+++ pwlib.new/include/ptlib/unix/ptlib/pmachdep.h	2006-01-16 15:54:52.000000000 -0600
@@ -445,8 +445,6 @@
 #include <netinet/tcp.h>
 #include <sys/ioctl.h>
  
-typedef int socklen_t;
- 
 #define HAS_IFREQ
  
 #define PSETPGRP()  setpgrp(0, 0)
diff -Naur -x'*~' pwlib/make/unix.mak pwlib.new/make/unix.mak
--- pwlib/make/unix.mak	2003-07-24 17:01:42.000000000 -0500
+++ pwlib.new/make/unix.mak	2006-01-16 15:57:40.000000000 -0600
@@ -627,6 +627,7 @@
 # it typically is /usr/local
 
 #ESDDIR =/sw
+SYSLIBDIR	= $(PWLIBDIR)/lib
 
 # Quicktime support is still a long way off. But for development purposes,
 # I am inluding the flags to allow QuickTime to be linked.
@@ -970,8 +971,6 @@
 
 
 # add library directory to library path and include the library
-LDFLAGS		+= -L$(PW_LIBDIR)
-
-LDLIBS		+= -l$(PTLIB_BASE)$(LIB_TYPE)
+LDLIBS		+= $(PW_LIBDIR)/lib$(PTLIB_BASE)$(LIB_TYPE).dylib
 
 # End of unix.mak
diff -Naur -x'*~' pwlib/src/ptclib/pdns.cxx pwlib.new/src/ptclib/pdns.cxx
--- pwlib/src/ptclib/pdns.cxx	2003-04-28 18:57:40.000000000 -0500
+++ pwlib.new/src/ptclib/pdns.cxx	2006-01-15 11:03:16.000000000 -0600
@@ -72,6 +72,7 @@
 #if P_HAS_RESOLVER
 #include <arpa/nameser.h>
 #include <resolv.h>
+#include <arpa/nameser_compat.h>
 #endif
 
 #endif
diff -Naur -x'*~' pwlib/src/ptclib/pssl.cxx pwlib.new/src/ptclib/pssl.cxx
--- pwlib/src/ptclib/pssl.cxx	2003-04-16 03:00:19.000000000 -0500
+++ pwlib.new/src/ptclib/pssl.cxx	2006-01-15 12:13:16.000000000 -0600
@@ -751,7 +751,8 @@
 
   switch (fileType) {
     case PSSLFileTypeASN1 :
-      dh = d2i_DHparams_bio(in, NULL);
+      dh = (DH *)ASN1_d2i_bio((char *(*)())DH_new,
+                (char *(*)())d2i_DHparams,in,NULL);
       if (dh != NULL)
         return TRUE;
 
diff -Naur -x'*~' pwlib/src/ptclib/udp.cxx pwlib.new/src/ptclib/udp.cxx
--- pwlib/src/ptclib/udp.cxx	2003-06-23 11:23:57.000000000 -0500
+++ pwlib.new/src/ptclib/udp.cxx	2006-01-15 11:46:12.000000000 -0600
@@ -215,11 +215,7 @@
    int originalSize = *len;
 
    struct sockaddr_in from;
-#ifdef P_MACOSX
-   int fromLen = sizeof(from);
-#else
    socklen_t fromLen = sizeof(from);
-#endif
 
    *len = recvfrom(fd,
                    buf,
diff -Naur -x'*~' pwlib/src/ptlib/unix/udll.cxx pwlib.new/src/ptlib/unix/udll.cxx
--- pwlib/src/ptlib/unix/udll.cxx	2003-07-09 06:37:13.000000000 -0500
+++ pwlib.new/src/ptlib/unix/udll.cxx	2006-01-15 14:27:45.000000000 -0600
@@ -80,7 +80,11 @@
 
 #include <ptlib.h>
 
-#ifdef P_MACOSX
+#if 1
+#include <dlfcn.h>
+#endif
+
+#if 0
 
 /*
 Copyright (c) 2002 Peter O'Gorman <ogorman@users.sourceforge.net>
