diff -Naur silc-client-0.9.16/Makefile.in silc-client-0.9.16-patched/Makefile.in
--- silc-client-0.9.16/Makefile.in	Mon Nov 17 18:39:52 2003
+++ silc-client-0.9.16-patched/Makefile.in	Fri Nov 21 09:20:49 2003
@@ -701,9 +701,7 @@
 	-rm -rf $(distdir)
 
 install-dirs:
-	-mkdir -p $(etcdir)
 	-mkdir -p $(modulesdir)
-	-mkdir -p $(docdir)
 
 install-dirs-client: install-dirs
 	-mkdir -p $(helpdir)
diff -Naur silc-client-0.9.16/includes/Makefile.in silc-client-0.9.16-patched/includes/Makefile.in
--- silc-client-0.9.16/includes/Makefile.in	Mon Nov 17 18:39:52 2003
+++ silc-client-0.9.16-patched/includes/Makefile.in	Fri Nov 21 09:21:07 2003
@@ -381,7 +381,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-0.9.16/irssi/Makefile.defines_int.in silc-client-0.9.16-patched/irssi/Makefile.defines_int.in
--- silc-client-0.9.16/irssi/Makefile.defines_int.in	Mon Nov 17 18:39:39 2003
+++ silc-client-0.9.16-patched/irssi/Makefile.defines_int.in	Fri Nov 21 09:21:30 2003
@@ -30,7 +30,7 @@
 #
 # Common libraries that are linked against the created executable
 #
-SILC_COMMON_LIBS= @LIBS@ -L$(silc_top_srcdir)/lib -lsilc
+SILC_COMMON_LIBS= @LIBS@ $(silc_top_srcdir)/lib/.libs/libsilc.a
 
 #
 # Common compilation flags
diff -Naur silc-client-0.9.16/irssi/Makefile.in silc-client-0.9.16-patched/irssi/Makefile.in
--- silc-client-0.9.16/irssi/Makefile.in	Mon Nov 17 18:40:25 2003
+++ silc-client-0.9.16-patched/irssi/Makefile.in	Fri Nov 21 09:21:42 2003
@@ -231,7 +231,7 @@
 confdir = $(silc_etcdir)
 conffile = silc.conf
 
-themedir = $(datadir)/silc/themes
+themedir = $(datadir)/silc-client/themes
 theme_DATA = default.theme
 
 noinst_HEADERS = irssi-version.h.in
diff -Naur silc-client-0.9.16/irssi/scripts/Makefile.in silc-client-0.9.16-patched/irssi/scripts/Makefile.in
--- silc-client-0.9.16/irssi/scripts/Makefile.in	Mon Nov 17 18:40:26 2003
+++ silc-client-0.9.16-patched/irssi/scripts/Makefile.in	Fri Nov 21 09:21:55 2003
@@ -163,7 +163,7 @@
 silc_MODULES = @silc_MODULES@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-scriptdir = $(datadir)/silc/scripts
+scriptdir = $(datadir)/silc-client/scripts
 
 script_DATA = \
 	autoop.pl \
diff -Naur silc-client-0.9.16/irssi/src/fe-text/Makefile.in silc-client-0.9.16-patched/irssi/src/fe-text/Makefile.in
--- silc-client-0.9.16/irssi/src/fe-text/Makefile.in	Mon Nov 17 18:40:29 2003
+++ silc-client-0.9.16-patched/irssi/src/fe-text/Makefile.in	Fri Nov 21 09:22:18 2003
@@ -224,7 +224,7 @@
 
 silc_LDADD = \
 	@COMMON_LIBS@ \
-	-L../../../lib -lsilcclient \
+	../../../lib/.libs/libsilcclient.a \
 	@PERL_LINK_LIBS@ \
 	@PERL_FE_LINK_LIBS@ \
 	@PERL_LINK_FLAGS@ \
diff -Naur silc-client-0.9.16/irssi/src/perl/Makefile.in silc-client-0.9.16-patched/irssi/src/perl/Makefile.in
--- silc-client-0.9.16/irssi/src/perl/Makefile.in	Mon Nov 17 18:40:30 2003
+++ silc-client-0.9.16-patched/irssi/src/perl/Makefile.in	Fri Nov 21 09:22:35 2003
@@ -694,7 +694,7 @@
 
 install-exec-local:
 	for dir in $(perl_dirs); do \
-	  cd $$dir && $(MAKE) install && cd ..; \
+	  cd $$dir && $(MAKE) pure_install && cd ..; \
 	done
 
 clean-generic:
diff -Naur silc-client-0.9.16/lib/Makefile.in silc-client-0.9.16-patched/lib/Makefile.in
--- silc-client-0.9.16/lib/Makefile.in	Mon Nov 17 18:39:53 2003
+++ silc-client-0.9.16-patched/lib/Makefile.in	Fri Nov 21 09:22:50 2003
@@ -444,7 +444,6 @@
 uninstall: uninstall-recursive
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
 
 installcheck: installcheck-recursive
 install-strip:
diff -Naur silc-client-0.9.16/lib/contrib/Makefile.in silc-client-0.9.16-patched/lib/contrib/Makefile.in
--- silc-client-0.9.16/lib/contrib/Makefile.in	Mon Nov 17 18:39:53 2003
+++ silc-client-0.9.16-patched/lib/contrib/Makefile.in	Fri Nov 21 09:23:06 2003
@@ -467,7 +467,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-0.9.16/lib/silcclient/Makefile.in silc-client-0.9.16-patched/lib/silcclient/Makefile.in
--- silc-client-0.9.16/lib/silcclient/Makefile.in	Mon Nov 17 18:39:53 2003
+++ silc-client-0.9.16-patched/lib/silcclient/Makefile.in	Fri Nov 21 09:23:16 2003
@@ -474,7 +474,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-0.9.16/lib/silccore/Makefile.in silc-client-0.9.16-patched/lib/silccore/Makefile.in
--- silc-client-0.9.16/lib/silccore/Makefile.in	Mon Nov 17 18:39:54 2003
+++ silc-client-0.9.16-patched/lib/silccore/Makefile.in	Fri Nov 21 09:23:26 2003
@@ -477,7 +477,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-0.9.16/lib/silccrypt/Makefile.in silc-client-0.9.16-patched/lib/silccrypt/Makefile.in
--- silc-client-0.9.16/lib/silccrypt/Makefile.in	Mon Nov 17 18:39:55 2003
+++ silc-client-0.9.16-patched/lib/silccrypt/Makefile.in	Fri Nov 21 09:23:40 2003
@@ -492,7 +492,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-0.9.16/lib/silcmath/Makefile.in silc-client-0.9.16-patched/lib/silcmath/Makefile.in
--- silc-client-0.9.16/lib/silcmath/Makefile.in	Mon Nov 17 18:39:55 2003
+++ silc-client-0.9.16-patched/lib/silcmath/Makefile.in	Fri Nov 21 09:23:50 2003
@@ -558,7 +558,7 @@
 uninstall: uninstall-recursive
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-recursive
 install-strip:
diff -Naur silc-client-0.9.16/lib/silcsftp/Makefile.in silc-client-0.9.16-patched/lib/silcsftp/Makefile.in
--- silc-client-0.9.16/lib/silcsftp/Makefile.in	Mon Nov 17 18:39:56 2003
+++ silc-client-0.9.16-patched/lib/silcsftp/Makefile.in	Fri Nov 21 09:23:59 2003
@@ -460,7 +460,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-0.9.16/lib/silcsim/Makefile.in silc-client-0.9.16-patched/lib/silcsim/Makefile.in
--- silc-client-0.9.16/lib/silcsim/Makefile.in	Mon Nov 17 18:39:56 2003
+++ silc-client-0.9.16-patched/lib/silcsim/Makefile.in	Fri Nov 21 09:24:10 2003
@@ -481,7 +481,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-0.9.16/lib/silcske/Makefile.in silc-client-0.9.16-patched/lib/silcske/Makefile.in
--- silc-client-0.9.16/lib/silcske/Makefile.in	Mon Nov 17 18:39:57 2003
+++ silc-client-0.9.16-patched/lib/silcske/Makefile.in	Fri Nov 21 09:24:18 2003
@@ -460,7 +460,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-0.9.16/lib/silcutil/Makefile.in silc-client-0.9.16-patched/lib/silcutil/Makefile.in
--- silc-client-0.9.16/lib/silcutil/Makefile.in	Mon Nov 17 18:39:57 2003
+++ silc-client-0.9.16-patched/lib/silcutil/Makefile.in	Fri Nov 21 09:24:27 2003
@@ -588,7 +588,7 @@
 uninstall: uninstall-recursive
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-recursive
 install-strip:
