diff -Naur silc-toolkit-0.9.11/Makefile.in silc-toolkit-0.9.11-patched/Makefile.in
--- silc-toolkit-0.9.11/Makefile.in	Thu Nov 13 16:25:40 2003
+++ silc-toolkit-0.9.11-patched/Makefile.in	Thu Nov 20 11:44:56 2003
@@ -701,7 +701,6 @@
 	-rm -rf $(distdir)
 
 install-dirs:
-	-mkdir -p $(etcdir)
 	-mkdir -p $(modulesdir)
 	-mkdir -p $(docdir)
 
@@ -764,7 +763,7 @@
 	-mkdir -p $(docdir)/toolkit/
 	-$(INSTALL_DATA) $(srcdir)/doc/toolkit/* $(docdir)/toolkit
 	-$(INSTALL_DATA) $(srcdir)/lib/doc/*.gif $(docdir)/toolkit
-	-cp -R $(srcdir)/tutorial $(prefix)
+	-cp -R $(srcdir)/tutorial $(DESTDIR)$(datadir)/
 
 examples-install:
 	-mkdir -p $(docdir)/examples/
@@ -777,7 +776,7 @@
 	fi
 
 @SILC_DIST_CLIENT_TRUE@install-data-hook: install-dirs-client sim-install doc-install-client etc-install-client
-@SILC_DIST_CLIENT_FALSE@@SILC_DIST_TOOLKIT_TRUE@install-data-hook: install-dirs-client install-dirs-server sim-install doc-install-client doc-install-server toolkit-install examples-install etc-install-client etc-install-server
+@SILC_DIST_CLIENT_FALSE@@SILC_DIST_TOOLKIT_TRUE@install-data-hook: install-dirs sim-install toolkit-install examples-install
 @SILC_DIST_CLIENT_FALSE@@SILC_DIST_TOOLKIT_FALSE@install-data-hook: install-dirs-server sim-install doc-install-server examples-install etc-install-server generate-server-key
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
diff -Naur silc-toolkit-0.9.11/lib/silcsim/Makefile.in silc-toolkit-0.9.11-patched/lib/silcsim/Makefile.in
--- silc-toolkit-0.9.11/lib/silcsim/Makefile.in	Thu Nov 13 16:25:45 2003
+++ silc-toolkit-0.9.11-patched/lib/silcsim/Makefile.in	Thu Nov 20 11:45:22 2003
@@ -565,7 +565,7 @@
 
 $(SIM_CIPHER_OBJS):
 	@if test '!' -f lib$*.la ; then \
-	  $(LIBTOOL) --mode=link $(CCLD) -rpath $(silc_modulesdir) \
+	  $(LIBTOOL) --mode=link $(CCLD) -rpath $(silc_modulesdir) -module \
 		../silccrypt/$*.lo -o lib$*.la $(LTFLAGS); \
 	  cd $(srcdir) && $(LN_S) -f $(srcdir)/.libs/lib$*.so \
 		$(srcdir)/$*.sim.so; \
