diff -Naur silc-client-1.0/Makefile.in silc-client-1.0-patched/Makefile.in
--- silc-client-1.0/Makefile.in	Thu Nov 27 10:07:16 2003
+++ silc-client-1.0-patched/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -711,9 +711,7 @@
 	-rm -rf $(distdir)
 
 install-dirs:
-	-mkdir -p $(etcdir)
 	-mkdir -p $(modulesdir)
-	-mkdir -p $(docdir)
 
 install-dirs-client: install-dirs
 	-mkdir -p $(helpdir)
diff -Naur silc-client-1.0/includes/Makefile.in silc-client-1.0-patched/includes/Makefile.in
--- silc-client-1.0/includes/Makefile.in	Thu Nov 27 10:07:17 2003
+++ silc-client-1.0-patched/includes/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -384,7 +384,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-1.0/irssi/Makefile.defines_int.in silc-client-1.0-patched/irssi/Makefile.defines_int.in
--- silc-client-1.0/irssi/Makefile.defines_int.in	Thu Nov 27 10:07:00 2003
+++ silc-client-1.0-patched/irssi/Makefile.defines_int.in	Sat Nov 29 11:35:43 2003
@@ -30,7 +30,7 @@
 #
 # Common libraries that are linked against the created executable
 #
-SILC_COMMON_LIBS= @LIBS@ -L$(silc_top_srcdir)/lib -lsilc
+SILC_COMMON_LIBS= @LIBS@ $(silc_top_srcdir)/lib/.libs/libsilc.a
 
 #
 # Common compilation flags
diff -Naur silc-client-1.0/irssi/Makefile.in silc-client-1.0-patched/irssi/Makefile.in
--- silc-client-1.0/irssi/Makefile.in	Thu Nov 27 10:07:59 2003
+++ silc-client-1.0-patched/irssi/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -232,7 +232,7 @@
 confdir = $(silc_etcdir)
 conffile = silc.conf
 
-themedir = $(datadir)/silc/themes
+themedir = $(datadir)/silc-client/themes
 theme_DATA = default.theme
 
 noinst_HEADERS = irssi-version.h.in
diff -Naur silc-client-1.0/irssi/scripts/Makefile.in silc-client-1.0-patched/irssi/scripts/Makefile.in
--- silc-client-1.0/irssi/scripts/Makefile.in	Thu Nov 27 10:08:00 2003
+++ silc-client-1.0-patched/irssi/scripts/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -164,7 +164,7 @@
 silc_MODULES = @silc_MODULES@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-scriptdir = $(datadir)/silc/scripts
+scriptdir = $(datadir)/silc-client/scripts
 
 script_DATA = \
 	autoop.pl \
diff -Naur silc-client-1.0/irssi/src/fe-text/Makefile.in silc-client-1.0-patched/irssi/src/fe-text/Makefile.in
--- silc-client-1.0/irssi/src/fe-text/Makefile.in	Thu Nov 27 10:08:03 2003
+++ silc-client-1.0-patched/irssi/src/fe-text/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -225,7 +225,7 @@
 
 silc_LDADD = \
 	@COMMON_LIBS@ \
-	-L../../../lib -lsilcclient \
+	../../../lib/.libs/libsilcclient.a \
 	@PERL_LINK_LIBS@ \
 	@PERL_FE_LINK_LIBS@ \
 	@PERL_LINK_FLAGS@ \
diff -Naur silc-client-1.0/irssi/src/perl/Makefile.in silc-client-1.0-patched/irssi/src/perl/Makefile.in
--- silc-client-1.0/irssi/src/perl/Makefile.in	Thu Nov 27 10:08:04 2003
+++ silc-client-1.0-patched/irssi/src/perl/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -697,7 +697,7 @@
 
 install-exec-local:
 	for dir in $(perl_dirs); do \
-	  cd $$dir && $(MAKE) install && cd ..; \
+	  cd $$dir && $(MAKE) pure_install && cd ..; \
 	done
 
 clean-generic:
diff -Naur silc-client-1.0/lib/Makefile.in silc-client-1.0-patched/lib/Makefile.in
--- silc-client-1.0/lib/Makefile.in	Thu Nov 27 10:07:17 2003
+++ silc-client-1.0-patched/lib/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -454,7 +454,6 @@
 uninstall: uninstall-recursive
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
 
 installcheck: installcheck-recursive
 install-strip:
diff -Naur silc-client-1.0/lib/contrib/Makefile.in silc-client-1.0-patched/lib/contrib/Makefile.in
--- silc-client-1.0/lib/contrib/Makefile.in	Thu Nov 27 10:07:17 2003
+++ silc-client-1.0-patched/lib/contrib/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -471,7 +471,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-1.0/lib/silcclient/Makefile.in silc-client-1.0-patched/lib/silcclient/Makefile.in
--- silc-client-1.0/lib/silcclient/Makefile.in	Thu Nov 27 10:07:17 2003
+++ silc-client-1.0-patched/lib/silcclient/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -477,7 +477,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-1.0/lib/silccore/Makefile.in silc-client-1.0-patched/lib/silccore/Makefile.in
--- silc-client-1.0/lib/silccore/Makefile.in	Thu Nov 27 10:07:17 2003
+++ silc-client-1.0-patched/lib/silccore/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -480,7 +480,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-1.0/lib/silccrypt/Makefile.in silc-client-1.0-patched/lib/silccrypt/Makefile.in
--- silc-client-1.0/lib/silccrypt/Makefile.in	Thu Nov 27 10:07:18 2003
+++ silc-client-1.0-patched/lib/silccrypt/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -495,7 +495,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-1.0/lib/silcmath/Makefile.in silc-client-1.0-patched/lib/silcmath/Makefile.in
--- silc-client-1.0/lib/silcmath/Makefile.in	Thu Nov 27 10:07:19 2003
+++ silc-client-1.0-patched/lib/silcmath/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -570,7 +570,7 @@
 uninstall: uninstall-recursive
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-recursive
 install-strip:
diff -Naur silc-client-1.0/lib/silcsftp/Makefile.in silc-client-1.0-patched/lib/silcsftp/Makefile.in
--- silc-client-1.0/lib/silcsftp/Makefile.in	Thu Nov 27 10:07:19 2003
+++ silc-client-1.0-patched/lib/silcsftp/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -463,7 +463,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-1.0/lib/silcsim/Makefile.in silc-client-1.0-patched/lib/silcsim/Makefile.in
--- silc-client-1.0/lib/silcsim/Makefile.in	Thu Nov 27 10:07:19 2003
+++ silc-client-1.0-patched/lib/silcsim/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -484,7 +484,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-1.0/lib/silcske/Makefile.in silc-client-1.0-patched/lib/silcske/Makefile.in
--- silc-client-1.0/lib/silcske/Makefile.in	Thu Nov 27 10:07:20 2003
+++ silc-client-1.0-patched/lib/silcske/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -463,7 +463,7 @@
 uninstall: uninstall-am
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-am
 install-strip:
diff -Naur silc-client-1.0/lib/silcutil/Makefile.in silc-client-1.0-patched/lib/silcutil/Makefile.in
--- silc-client-1.0/lib/silcutil/Makefile.in	Thu Nov 27 10:07:20 2003
+++ silc-client-1.0-patched/lib/silcutil/Makefile.in	Sat Nov 29 11:35:43 2003
@@ -602,7 +602,7 @@
 uninstall: uninstall-recursive
 
 install-am: all-am
-	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am
 
 installcheck: installcheck-recursive
 install-strip:
