--- snort/src/snort.c.orig	2003-04-22 20:27:16.000000000 +0200
+++ snort/src/snort.c	2003-04-22 20:27:40.000000000 +0200
@@ -1729,7 +1729,7 @@
 {
     struct stat st;
     int i;
-    char *conf_files[]={"/etc/snort.conf", "./snort.conf", NULL};
+    char *conf_files[]={"@PREFIX@/etc/snort.conf", "./snort.conf", NULL};
     char *fname = NULL;
     char *home_dir = NULL;
     char *rval = NULL;
--- snort/etc/snort.conf.orig	2003-04-22 20:59:34.000000000 +0200
+++ snort/etc/snort.conf	2003-04-22 21:01:23.000000000 +0200
@@ -99,7 +99,7 @@
 var AIM_SERVERS [64.12.24.0/24,64.12.25.0/24,64.12.26.14/24,64.12.28.0/24,64.12.29.0/24,64.12.161.0/24,64.12.163.0/24,205.188.5.0/24,205.188.9.0/24]
 
 # Path to your rules files (this can be a relative path)
-var RULE_PATH ../rules
+var RULE_PATH .
 
 # Configure the snort decoder:
 # ============================
--- snort/src/parser.c.orig	2003-04-22 21:03:50.000000000 +0200
+++ snort/src/parser.c	2003-04-22 21:04:11.000000000 +0200
@@ -3210,7 +3210,7 @@
         }
         else
         {
-            strlcpy(buffer, "/var/log/snort", STD_BUF);
+            strlcpy(buffer, "@PREFIX@/var/log/snort", STD_BUF);
         }
 
         strlcat(buffer, "/", STD_BUF - strlen(buffer));
--- snort/src/snort.h.orig	2003-04-22 21:05:21.000000000 +0200
+++ snort/src/snort.h	2003-04-22 21:05:40.000000000 +0200
@@ -68,7 +68,7 @@
 #endif
 
 #ifndef WIN32
-    #define DEFAULT_LOG_DIR            "/var/log/snort"
+    #define DEFAULT_LOG_DIR            "@PREFIX@/var/log/snort"
     #define DEFAULT_DAEMON_ALERT_FILE  "alert"
 #else
     #define DEFAULT_LOG_DIR            "log"
--- snort/src/util.c.orig	2003-04-22 21:14:10.000000000 +0200
+++ snort/src/util.c	2003-04-22 21:14:51.000000000 +0200
@@ -713,7 +713,7 @@
         }
 #ifndef _PATH_VARRUN
 #ifndef WIN32
-        strlcpy(_PATH_VARRUN, "/var/run/", 10);
+        strlcpy(_PATH_VARRUN, "@PREFIX@/var/run/", 10);
 #else
         if (GetCurrentDirectory(sizeof (dir)-1, dir))
             strncpy (_PATH_VARRUN, dir, sizeof(dir)-1);
@@ -731,8 +731,8 @@
         if(!S_ISDIR(pt.st_mode) || access(_PATH_VARRUN, W_OK) == -1)
         {
             LogMessage("WARNING: _PATH_VARRUN is invalid, trying "
-                    "/var/log...\n");
-            strncpy(pv.pid_path, "/var/log/", strlen("/var/log/"));
+                    "@PREFIX@/var/log...\n");
+            strncpy(pv.pid_path, "@PREFIX@/var/log/", strlen("@PREFIX@/var/log/"));
             stat(pv.pid_path, &pt);
 
             if(!S_ISDIR(pt.st_mode) || access(pv.pid_path, W_OK) == -1)
--- snort-2.0.2/src/util.h.orig 2003-09-30 17:25:13.000000000 +0200
+++ snort-2.0.2/src/util.h      2003-09-30 17:25:31.000000000 +0200
@@ -39,6 +39,8 @@
 #include "strlcpyu.h"
 #endif
 
+#include <sys/types.h>
+
 extern u_long netmasks[33];
 
 /* Self preservation memory control struct */

