diff -urN subversion-0.37.0.orig/ac-helpers/ltmain.sh subversion-0.37.0/ac-helpers/ltmain.sh
--- subversion-0.37.0.orig/ac-helpers/ltmain.sh	Sat Dec 20 12:02:12 2003
+++ subversion-0.37.0/ac-helpers/ltmain.sh	Thu Dec 25 20:21:10 2003
@@ -1395,6 +1395,8 @@
 	lib=
 	found=no
 	case $deplib in
+	-ldb-4.1*) continue ;;
+	libdb-4.1*) continue ;;
 	-l*)
 	  if test $linkmode = oldlib && test $linkmode = obj; then
 	    $echo "$modename: warning: \`-l' is ignored for archives/objects: $deplib" 1>&2
@@ -3649,7 +3651,7 @@
 # if CDPATH is set.
 if test \"\${CDPATH+set}\" = set; then CDPATH=:; export CDPATH; fi
 
-relink_command=\"$relink_command\"
+relink_command=""
 
 # This environment variable determines our operation mode.
 if test \"\$libtool_install_magic\" = \"$magic\"; then
@@ -4020,7 +4022,7 @@
 libdir='$install_libdir'"
 	  if test "$installed" = no && test $need_relink = yes; then
 	    $echo >> $output "\
-relink_command=\"$relink_command\""
+relink_command=\"\""
 	  fi
 	done
       fi
diff -urN subversion-0.37.0.orig/subversion/bindings/swig/perl/Makefile.PL subversion-0.37.0/subversion/bindings/swig/perl/Makefile.PL
--- subversion-0.37.0.orig/subversion/bindings/swig/perl/Makefile.PL 2004-02-06 12:07:58.000000000 +0100
+++ subversion-0.37.0/subversion/bindings/swig/perl/Makefile.PL    2004-01-30 09:51:16.000000000 +0100
@@ -20,8 +20,8 @@
 
 my @modules = qw/client delta fs ra repos wc/;
 my @ldpaths = ('../.libs',
-               map {"../../../libsvn_$_/.libs"} (@modules, qw/diff subr/));
-my @ldmodules = map {"-lsvn_$_-1"} (@modules, qw/diff subr/);
+               map {"../../../libsvn_$_/.libs"} (@modules, qw/diff subr ra_local ra_dav ra_svn/));
+my @ldmodules = map {"-lsvn_$_-1"} (@modules, qw/diff subr ra_local ra_dav ra_svn/);
 
 my $apr_config = $ENV{APR_CONFIG} || '/usr/local/lib/apache2/apr-config';
 $apr_config = 'apr-config' unless -e $apr_config;
