diff -Nurd -x'*~' gnome-menus-2.20.3.orig/Makefile.in gnome-menus-2.20.3/Makefile.in
--- gnome-menus-2.20.3.orig/Makefile.in	2008-01-07 17:29:24.000000000 -0500
+++ gnome-menus-2.20.3/Makefile.in	2008-01-09 10:16:32.000000000 -0500
@@ -245,11 +245,6 @@
 NULL = 
 @HAVE_PYTHON_TRUE@PYTHON_SUBDIRS = python simple-editor
 ALL_SUBDIRS = \
-	po 			\
-	libmenu			\
-	layout			\
-	desktop-directories	\
-	util			\
 	$(NULL)
 
 SUBDIRS = $(ALL_SUBDIRS) $(PYTHON_SUBDIRS)
diff -Nurd -x'*~' gnome-menus-2.14.3.orig/ltmain.sh gnome-menus-2.14.3/ltmain.sh
--- gnome-menus-2.14.3.orig/ltmain.sh	2006-04-29 04:39:27.000000000 -0400
+++ gnome-menus-2.14.3/ltmain.sh	2006-08-11 02:26:59.000000000 -0400
@@ -4015,6 +4015,11 @@
 	  eval test_cmds=\"$archive_expsym_cmds\"
 	  cmds=$archive_expsym_cmds
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+          else
+            tmp_verstring="$verstring"
+          fi
 	  eval test_cmds=\"$archive_cmds\"
 	  cmds=$archive_cmds
 	  fi
diff -Nurd -x'*~' gnome-menus-2.20.3.orig/python/Makefile.in gnome-menus-2.20.3/python/Makefile.in
--- gnome-menus-2.20.3.orig/python/Makefile.in	2008-01-07 17:29:23.000000000 -0500
+++ gnome-menus-2.20.3/python/Makefile.in	2008-01-09 10:16:32.000000000 -0500
@@ -52,8 +52,7 @@
 gmenuLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(gmenu_LTLIBRARIES)
 am__DEPENDENCIES_1 =
-gmenu_la_DEPENDENCIES = $(am__DEPENDENCIES_1) \
-	$(top_builddir)/libmenu/libgnome-menu.la
+gmenu_la_DEPENDENCIES = $(am__DEPENDENCIES_1)
 am_gmenu_la_OBJECTS = gmenu.lo
 gmenu_la_OBJECTS = $(am_gmenu_la_OBJECTS)
 gmenu_la_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
@@ -250,7 +249,6 @@
 NULL = 
 INCLUDES = \
 	-DGMENU_I_KNOW_THIS_IS_UNSTABLE	\
-	-I$(srcdir)/../libmenu		\
 	$(WARN_CFLAGS)			\
 	$(NO_STRICT_ALIASING_CFLAGS)	\
 	$(GLIB_CFLAGS)			\
@@ -262,7 +260,7 @@
 gmenudir = $(pyexecdir)
 gmenu_LTLIBRARIES = gmenu.la
 gmenu_la_LDFLAGS = -module -avoid-version -fPIC -export-symbols-regex initgmenu
-gmenu_la_LIBADD = $(GLIB_LIBS) $(top_builddir)/libmenu/libgnome-menu.la
+gmenu_la_LIBADD = $(GLIB_LIBS)
 gmenu_la_SOURCES = gmenu.c
 all: all-am
 
