diff -Nurd -x'*~' gnome-desktop-2.28.2.orig/configure gnome-desktop-2.28.2/configure
--- gnome-desktop-2.28.2.orig/configure	2009-12-17 06:22:33.000000000 -0500
+++ gnome-desktop-2.28.2/configure	2010-09-30 03:32:20.000000000 -0400
@@ -13622,6 +13622,7 @@
 
 
 
+if false; then
 GDK_PIXBUF_REQUIRED=2.4.0
 GTK_REQUIRED=2.14.0
 GLIB_REQUIRED=2.19.1
@@ -16370,6 +16371,7 @@
 
 
 
+fi
         # Find any Python interpreter.
     if test -z "$PYTHON"; then
       for ac_prog in python python2 python3 python3.0 python2.5 python2.4 python2.3 python2.2 python2.1 python2.0
@@ -16621,7 +16623,7 @@
 
 
 
-GETTEXT_PACKAGE=gnome-desktop-2.0
+GETTEXT_PACKAGE=gnome-desktop-2.11
 
 cat >>confdefs.h <<_ACEOF
 #define GETTEXT_PACKAGE "$GETTEXT_PACKAGE"
