diff -Nurd -x'*~' gtksourceview-2.8.2.orig/gtksourceview/Makefile.in gtksourceview-2.8.2/gtksourceview/Makefile.in
--- gtksourceview-2.8.2.orig/gtksourceview/Makefile.in	2009-12-19 13:20:08.000000000 -0500
+++ gtksourceview-2.8.2/gtksourceview/Makefile.in	2010-02-25 04:13:34.000000000 -0500
@@ -383,7 +383,7 @@
 nodist_libgtksourceview_2_0_la_SOURCES = \
 	$(BUILT_SOURCES)
 
-libgtksourceview_2_0_la_LIBADD = $(DEP_LIBS)
+libgtksourceview_2_0_la_LIBADD = $(DEP_LIBS) $(INTLLIBS)
 libgtksourceview_2_0_la_LDFLAGS = -no-undefined -export-symbols-regex "^gtk_source_.*"
 libgtksourceview_2_0_includedir = $(includedir)/gtksourceview-2.0/gtksourceview
 libgtksourceview_2_0_include_HEADERS = \
