diff -Nurd -x'*~' rarian-0.8.1.orig/librarian/rarian-info.c rarian-0.8.1/librarian/rarian-info.c
--- rarian-0.8.1.orig/librarian/rarian-info.c	2008-09-01 14:40:21.000000000 -0400
+++ rarian-0.8.1/librarian/rarian-info.c	2011-11-06 02:39:30.000000000 -0500
@@ -477,7 +477,7 @@
     if ((*line != '*') && !isspace(*line)) {
       /* New category */
       set_category (line);
-    } else if ((*line == '*')) {
+    } else if (*line == '*') {
       /* New entry */
       if (current_entry) {
 	if (process_check_file()) {
diff -Nurd -x'*~' rarian-0.8.1.orig/librarian/rarian-reg-full.c rarian-0.8.1/librarian/rarian-reg-full.c
--- rarian-0.8.1.orig/librarian/rarian-reg-full.c	2008-09-01 14:40:21.000000000 -0400
+++ rarian-0.8.1/librarian/rarian-reg-full.c	2011-11-06 03:08:53.000000000 -0500
@@ -164,7 +164,7 @@
     while (sect) {
     	if (rrn_reg_add_sect_full (reg, sect) == 1) {
     		fprintf (stderr, "Error: Orphaned section %s not added.\n",
-    				 sect->name);
+    				 sect->name->text);
     	}
     	sect = sect->next;
     }
diff -Nurd -x'*~' rarian-0.8.1.orig/librarian/rarian-reg-utils.c rarian-0.8.1/librarian/rarian-reg-utils.c
--- rarian-0.8.1.orig/librarian/rarian-reg-utils.c	2008-09-01 14:40:21.000000000 -0400
+++ rarian-0.8.1/librarian/rarian-reg-utils.c	2011-11-06 03:03:18.000000000 -0500
@@ -430,6 +430,8 @@
 	sect->children = NULL;
 	sect->owner = NULL;
 	sect->priority = 0;
+
+	return sect;
 }
 
 RrnSect *
diff -Nurd -x'*~' rarian-0.8.1.orig/util/rarian-example.c rarian-0.8.1/util/rarian-example.c
--- rarian-0.8.1.orig/util/rarian-example.c	2008-09-01 14:40:21.000000000 -0400
+++ rarian-0.8.1/util/rarian-example.c	2011-11-06 02:44:29.000000000 -0500
@@ -77,6 +77,7 @@
     return TRUE;
 }
 
+int
 info_for_each (RrnInfoEntry *entry, void *data)
 {
   if (entry->section)
@@ -90,6 +91,7 @@
   return TRUE;
 }
 
+int
 man_for_each (RrnManEntry *entry, void *data)
 {
   printf ("Man page %s\n", entry->name);
diff -Nurd -x'*~' rarian-0.8.1.orig/util/rarian-sk-migrate.cpp rarian-0.8.1/util/rarian-sk-migrate.cpp
--- rarian-0.8.1.orig/util/rarian-sk-migrate.cpp	2008-09-01 14:40:21.000000000 -0400
+++ rarian-0.8.1/util/rarian-sk-migrate.cpp	2011-11-06 02:42:37.000000000 -0500
@@ -305,7 +305,7 @@
   		time(&t1);
 
         new_series = (char *) malloc (sizeof (char) * 255);
-        sprintf (new_series, "org.scrollkeeper.undefined.%d", (long) t1);
+        sprintf (new_series, "org.scrollkeeper.undefined.%ld", (long) t1);
     }
     dump_extended_keyfile (argv[1], argv[2]);
 
diff -Nurd -x'*~' rarian-0.8.1.orig/util/rarian-sk-update.in rarian-0.8.1/util/rarian-sk-update.in
--- rarian-0.8.1.orig/util/rarian-sk-update.in	2008-09-01 14:40:21.000000000 -0400
+++ rarian-0.8.1/util/rarian-sk-update.in	2008-09-27 14:49:44.000000000 -0400
@@ -57,7 +57,7 @@
 	echo -e "\t\t\t\tscrolls.  If the directory doesn't exist, it"
 	echo -e "\t\t\t\twill be created at run time."
 	echo -e "\t\t\t\t Note: Multiple output paths are not supported"
-	echo -e "--c\t\t\t\tRebuild the index entirely.  This will rebuild"
+	echo -e "-c\t\t\t\tRebuild the index entirely.  This will rebuild"
 	echo -e "\t\t\t\tall scroll files and may take a long time."
 	echo -e "-v\t\t\t\tTurn Verbosity on."
 	echo -e "-p\t\t\t\tSpecify a different place to store the mtimes file."
@@ -264,7 +264,7 @@
 # They are inherited from scrollkeeper :(
 
 # We use TEMP as set -- seems to nuke the return value of getopt
-TEMP=`getopt -u -n$(basename $0) -o "o:r:p:vqnhV" \
+TEMP=`getopt "o:r:p:vqnhV" \
     	     -- "$@"` \
     || print_usage
 
diff -Nurd -x'*~' rarian-0.8.1.orig/util/tinyxmlparser.cpp rarian-0.8.1/util/tinyxmlparser.cpp
--- rarian-0.8.1.orig/util/tinyxmlparser.cpp	2008-09-01 14:40:21.000000000 -0400
+++ rarian-0.8.1/util/tinyxmlparser.cpp	2011-11-06 02:13:49.000000000 -0500
@@ -353,7 +353,7 @@
 	}
 	else
 	{
-		while ( *p && IsWhiteSpace( *p ) || *p == '\n' || *p =='\r' )
+		while ( *p && IsWhiteSpace( *p ) )
 			++p;
 	}
 
