diff -Nurd -x'*~' rep-gtk-0.18.orig/Makefile.in rep-gtk-0.18/Makefile.in
--- rep-gtk-0.18.orig/Makefile.in	2002-02-13 02:50:27.000000000 -0500
+++ rep-gtk-0.18/Makefile.in	2007-06-14 01:52:05.000000000 -0400
@@ -32,11 +32,11 @@
 include $(repexecdir)/rules.mk
 
 CC=@CC@
-CPPFLAGS=@CPPFLAGS@
 CFLAGS=@CFLAGS@
 LDFLAGS=@LDFLAGS@
-override CPPFLAGS += -DHAVE_CONFIG_H -I.
-override CFLAGS += @GLIB_CFLAGS@ @GTK_CFLAGS@ @REP_CFLAGS@ @LIBGLADE_CFLAGS@ @GNOME_CFLAGS@ @GNOMEUI_CFLAGS@ @GNOME_CANVAS_CFLAGS@ @GDK_PIXBUF_CFLAGS@
+CPPFLAGS = -DHAVE_CONFIG_H -I.
+CPPFLAGS += @GLIB_CFLAGS@ @GTK_CFLAGS@ @REP_CFLAGS@ @LIBGLADE_CFLAGS@ @GNOME_CFLAGS@ @GNOMEUI_CFLAGS@ @GNOME_CANVAS_CFLAGS@ @GDK_PIXBUF_CFLAGS@
+CPPFLAGS += @CPPFLAGS@
 REP_SGTK_LIBS=@GLIB_LIBS@ @LIBS@
 REP_GTK_LIBS=@GTK_LIBS@ @LIBS@
 REP_LIBGLADE_LIBS=@LIBGLADE_LIBS@ @LIBS@
@@ -60,22 +60,22 @@
 all : build_dirs $(OUT)
 
 gtk-2/types.la : $(SGTK_OBJS)
-	$(rep_DL_LD) $(CPPFLAGS) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(REP_SGTK_LIBS)
+	$(rep_DL_LD) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(REP_SGTK_LIBS)
 
 gtk-2/gtk.la : $(OBJS)
-	$(rep_DL_LD) $(CPPFLAGS) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(REP_GTK_LIBS)
+	$(rep_DL_LD) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(REP_GTK_LIBS)
 
 gtk-2/libglade.la : $(LIBGLADE_OBJS)
-	$(rep_DL_LD) $(CPPFLAGS) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(REP_LIBGLADE_LIBS)
+	$(rep_DL_LD) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(REP_LIBGLADE_LIBS)
 
 gtk-2/gnome-lib.la : $(GNOMELIB_OBJS)
-	$(rep_DL_LD) $(CPPFLAGS) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(REP_GNOMELIB_LIBS)
+	$(rep_DL_LD) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(REP_GNOMELIB_LIBS)
 
 gtk-2/gnome-ui.la : $(GNOMEUI_OBJS)
-	$(rep_DL_LD) $(CPPFLAGS) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(REP_GNOMEUI_LIBS)
+	$(rep_DL_LD) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(REP_GNOMEUI_LIBS)
 
 gtk-2/gnome-canvas.la : $(GNOME_CANVAS_OBJS)
-	$(rep_DL_LD) $(CPPFLAGS) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(REP_GNOME_CANVAS_LIBS)
+	$(rep_DL_LD) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(REP_GNOME_CANVAS_LIBS)
 
 %-glue.c : %.defs build-gtk.jlc
 	rep build-gtk.jlc -f build-gtk-batch $< $@
diff -ruN rep-gtk-0.18/gtk-compat.c rep-gtk-0.18.patched/gtk-compat.c
--- rep-gtk-0.18/gtk-compat.c	Fri Jul 12 00:19:22 2002
+++ rep-gtk-0.18.patched/gtk-compat.c	Sat Jul 24 08:04:58 2004
@@ -62,6 +62,7 @@
 		  func, func_data, button, activate_time);
 }
 
+#if GTK_MAJOR_VERSION == 2 && GTK_MINOR_VERSION < 3
 GtkWidget*
 gtk_radio_menu_item_new_with_label_from_widget (GtkRadioMenuItem *group,
 						gchar            *label)
@@ -77,6 +78,7 @@
   GSList *g = group? gtk_radio_menu_item_group (group) : NULL;
   return gtk_radio_menu_item_new_with_mnemonic (g, label);
 }
+#endif
 
 GtkWidget*
 gtk_radio_menu_item_new_from_widget (GtkRadioMenuItem *group)
@@ -101,6 +103,8 @@
 }
 
 #ifndef HAVE_GDK_COLOR_COPY
+#if GTK_MAJOR_VERSION == 2 && GTK_MINOR_VERSION < 3
+
 /*
  *--------------------------------------------------------------
  * gdk_color_copy
@@ -156,6 +160,7 @@
 
   g_mem_chunk_free (color_chunk, color);
 }
+#endif
 #endif
 
 GdkColor*
diff -ruN rep-gtk-0.18/rep-gtk.h rep-gtk-0.18.patched/rep-gtk.h
--- rep-gtk-0.18/rep-gtk.h	Wed Feb 12 03:51:45 2003
+++ rep-gtk-0.18.patched/rep-gtk.h	Sat Jul 24 08:02:32 2004
@@ -288,19 +288,23 @@
 			    gint button,
 			    guint32 activate_time,
 			    repv position);
-
+#if GTK_MAJOR_VERSION == 2 && GTK_MINOR_VERSION < 3
 GtkWidget*
 gtk_radio_menu_item_new_with_label_from_widget (GtkRadioMenuItem *group,
 						gchar            *label);
 GtkWidget*
 gtk_radio_menu_item_new_with_mnemonic_from_widget (GtkRadioMenuItem *group,
 						   gchar            *label);
+#endif
+
 GtkWidget* gtk_radio_menu_item_new_from_widget (GtkRadioMenuItem *group);
 GtkWidget* gtk_pixmap_new_interp (char *file, GtkWidget *intended_parent);
 
 #ifndef HAVE_GDK_COLOR_COPY
+#if GTK_MAJOR_VERSION == 2 && GTK_MINOR_VERSION < 3
 GdkColor*    gdk_color_copy     (GdkColor *);
 void         gdk_color_free     (GdkColor *);
+#endif
 #endif
 
 GdkColor *gdk_color_parse_interp (char *spec);
