diff -Nurd '-x*~' watray-0.1.1.orig/watraysimple/simpledocumentmanager.c watray-0.1.1/watraysimple/simpledocumentmanager.c
--- watray-0.1.1.orig/watraysimple/simpledocumentmanager.c	2009-02-09 09:21:48.000000000 -0500
+++ watray-0.1.1/watraysimple/simpledocumentmanager.c	2012-02-19 22:49:36.000000000 -0500
@@ -24,14 +24,14 @@
 #include <gtk/gtk.h>
 #include <stdlib.h>
 #include <string.h>
-#include <gee/hashmap.h>
+#include <gee.h>
 #include <watray/watrayproject.h>
 #include <glib/gi18n-lib.h>
-#include <gee/map.h>
+#include <gee.h>
 #include <watray/watraydocumentview.h>
 #include <config.h>
-#include <gee/iterable.h>
-#include <gee/iterator.h>
+#include <gee.h>
+#include <gee.h>
 
 
 
@@ -424,7 +424,7 @@
 	gtk_file_filter_add_pattern (filter, "*");
 	gtk_file_chooser_add_filter ((GtkFileChooser*) dialog, filter);
 	if (gtk_dialog_run ((GtkDialog*) dialog) == GTK_RESPONSE_OK) {
-		gee_map_remove ((GeeMap*) self->priv->_views, simple_document_view_get_filename (view));
+		gee_map_remove ((GeeMap*) self->priv->_views, simple_document_view_get_filename (view), NULL);
 		simple_document_view_set_filename (view, gtk_file_chooser_get_filename ((GtkFileChooser*) dialog));
 		gee_map_set ((GeeMap*) self->priv->_views, simple_document_view_get_filename (view), view);
 		simple_document_view_save (view);
@@ -513,7 +513,7 @@
 			return;
 		}
 	}
-	gee_map_remove ((GeeMap*) self->priv->_views, simple_document_view_get_filename (document_view));
+	gee_map_remove ((GeeMap*) self->priv->_views, simple_document_view_get_filename (document_view), NULL);
 }
 
 
