diff -Nurd nexus-webapp-1.5.0/bin/jsw/macosx-universal-32/nexus nexus-webapp-1.5.0-new/bin/jsw/macosx-universal-32/nexus
--- nexus-webapp-1.5.0/bin/jsw/macosx-universal-32/nexus	2010-01-26 17:58:28.000000000 -0500
+++ nexus-webapp-1.5.0-new/bin/jsw/macosx-universal-32/nexus	2010-02-25 11:54:28.000000000 -0500
@@ -30,7 +30,7 @@
 PRIORITY=
 
 # Location of the pid file.
-PIDDIR="."
+PIDDIR="@FINKPREFIX@/var/run/nexus"
 
 # If uncommented, causes the Wrapper to be shutdown using an anchor file.
 #  When launched with the 'start' command, it will also ignore all INT and
diff -Nurd nexus-webapp-1.5.0/conf/plexus.properties nexus-webapp-1.5.0-new/conf/plexus.properties
--- nexus-webapp-1.5.0/conf/plexus.properties	2010-01-26 17:41:25.000000000 -0500
+++ nexus-webapp-1.5.0-new/conf/plexus.properties	2010-02-25 11:17:08.000000000 -0500
@@ -2,13 +2,13 @@
 application-host=0.0.0.0
 runtime=${basedir}/runtime
 apps=${runtime}/apps
-nexus-work=${basedir}/../sonatype-work/nexus
+nexus-work=@FINKPREFIX@/var/run/nexus
 nexus-app=${runtime}/apps/nexus
 webapp=${runtime}/apps/nexus/webapp
 webapp-context-path=/nexus
 security-xml-file=${nexus-work}/conf/security.xml
 application-conf=${nexus-work}/conf
-runtime-tmp=${runtime}/tmp
+runtime-tmp=/tmp/nexus
 
 # If this file is present, it will be used to configure Jetty.
 jetty.xml=${basedir}/conf/jetty.xml
diff -Nurd nexus-webapp-1.5.0/conf/wrapper.conf nexus-webapp-1.5.0-new/conf/wrapper.conf
--- nexus-webapp-1.5.0/conf/wrapper.conf	2010-01-26 17:41:25.000000000 -0500
+++ nexus-webapp-1.5.0-new/conf/wrapper.conf	2010-02-25 11:49:27.000000000 -0500
@@ -7,6 +7,8 @@
 # Wait a bit longer than default before assuming jvm hasn't started properly
 wrapper.startup.timeout=90
 
+wrapper.java.pidfile=@FINKPREFIX@/var/run/nexus/wrapper-java.pid
+wrapper.pidfile=@FINKPREFIX@/var/run/nexus/wrapper.pid
 #wrapper.debug=true
 #This should be set to the base of your system, which is one folder below the container and nexus folders
 #wrapper.working.dir=../../../../
@@ -21,18 +23,18 @@
 #  needed starting from 1
 #wrapper.java.classpath.1=../lib/wrapper.jar
 #wrapper.java.classpath.2=../lib/my-app-1.0-SNAPSHOT.jar
-wrapper.java.classpath.1=../../../lib/*.jar
-wrapper.java.classpath.2=../../../conf/
+wrapper.java.classpath.1=@FINKPREFIX@/lib/nexus/lib/*.jar
+wrapper.java.classpath.2=@FINKPREFIX@/lib/nexus/conf/
 
 # Java Library Path (location of Wrapper.DLL or libwrapper.so)
-wrapper.java.library.path.1=./lib
+wrapper.java.library.path.1=@FINKPREFIX@/lib/nexus/lib
 
 # Java Additional Parameters
 #wrapper.java.additional.1=-d64
 #wrapper.java.additional.2=-Dcom.sun.management.jmxremote
 #wrapper.java.additional.3=-Dorg.tanukisoftware.wrapper.WrapperManager.mbean=true
-wrapper.java.additional.1=-Dbasedir=../../..
-wrapper.java.additional.2=-Djava.io.tmpdir=../../../runtime/tmp
+wrapper.java.additional.1=-Dbasedir=@FINKPREFIX@/lib/nexus
+wrapper.java.additional.2=-Djava.io.tmpdir=/tmp/nexus
 wrapper.java.additional.3=-Dsun.net.inetaddr.ttl=3600
 #wrapper.java.additional.4=-Xdebug
 #wrapper.java.additional.5=-Xnoagent
@@ -66,7 +68,7 @@
 wrapper.console.loglevel=INFO
 
 # Log file to use for wrapper output logging.
-wrapper.logfile=../../../logs/wrapper.log
+wrapper.logfile=@FINKPREFIX@/var/log/nexus/wrapper.log
 
 # Format of output for the log file.  (See docs for formats)
 wrapper.logfile.format=LPTM
@@ -82,7 +84,7 @@
 
 # Maximum number of rolled log files which will be allowed before old
 #  files are deleted.  The default value of 0 implies no limit.
-wrapper.logfile.maxfiles=2
+wrapper.logfile.maxfiles=4
 
 # Log Level for sys/event log output.  (See docs for log levels)
 wrapper.syslog.loglevel=NONE
