diff -Nurd -x'*~' sendfile-2.1b.orig/develop/configure sendfile-2.1b/develop/configure
--- sendfile-2.1b.orig/develop/configure	2009-10-07 14:20:14.000000000 -0400
+++ sendfile-2.1b/develop/configure	2010-06-12 11:29:35.000000000 -0400
@@ -4104,7 +4104,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lreadline -ltermcap -L/usr/lib/termcap $LIBS"
+LIBS="-lreadline $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
diff -Nurd -x'*~' sendfile-2.1b.orig/makeconfig sendfile-2.1b/makeconfig
--- sendfile-2.1b.orig/makeconfig	2006-06-07 09:57:47.000000000 -0400
+++ sendfile-2.1b/makeconfig	2010-06-12 11:25:23.000000000 -0400
@@ -11,16 +11,16 @@
 
 # default compiling settings
 CC=gcc
-CFLAGS=-O2
-LDFLAGS=-s
+CFLAGS="-O2 -I@PREFIX@/include -DHAVE_STRERROR"
+LDFLAGS=-L@PREFIX@/lib
 
 # default installation settings
-SPOOL=/var/spool/sendfile
-BINDIR=/usr/local/bin
-MANDIR=/usr/local/man
-CONFIG=/usr/local/etc
-SERVERDIR=/usr/local/sbin
-INETDCONF=/etc/inetd.conf # xinetd will be automaticly recognized
+SPOOL=@PREFIX@/var/spool/sendfile
+BINDIR=@PREFIX@/bin
+MANDIR=@PREFIX@/share/man
+CONFIG=@PREFIX@/etc
+SERVERDIR=@PREFIX@/sbin
+INETDCONF=@PREFIX@/etc/inetd.conf # xinetd will be automaticly recognized
 SERVICES=/etc/services
 
 ############################# OPTIONAL SETTINGS ##############################
@@ -82,7 +82,7 @@
     --sysconfdir=$CONFIG \
   || exit $?
 
-if [ -d /var/run ]; then FIFODIR=/var/run/sfm; fi
+if [ -d @PREFIX@/var/run ]; then FIFODIR=@PREFIX@/var/run/sfm; fi
 
 SYSTEM=`./config.guess`
 
@@ -117,8 +117,12 @@
 for i in nsl socket readline; do
   if grep -i "have_lib$i 1" config.h >/dev/null; then LIBS="$LIBS -l$i"; fi
 done
-case "$LIBS" in
-  *readline*) LIBS="$LIBS -L/usr/lib/termcap -ltermcap";;
+case "$SYSTEM" in
+  *DARWIN*)	;;
+  *)
+      case "$LIBS" in
+        *readline*) LIBS="$LIBS -L/usr/lib/termcap -ltermcap";;
+      esac
 esac
 
 echo creating globals.h
