diff -ru slrn-0.9.9/autoconf/Makefile.in slrn-0.9.9-patched/autoconf/Makefile.in
--- slrn-0.9.9/autoconf/Makefile.in	2008-02-11 19:04:23.000000000 +0100
+++ slrn-0.9.9-patched/autoconf/Makefile.in	2008-09-10 22:21:28.000000000 +0200
@@ -27,3 +27,5 @@
 # directory as a symbolic link to a local disk instead of an NFS mounted one.
 symlinks:
 	cd src; $(MAKE) symlinks
+
+.PHONY: all clean install
diff -ru slrn-0.9.9/src/Makefile.in slrn-0.9.9-patched/src/Makefile.in
--- slrn-0.9.9/src/Makefile.in	2008-08-09 18:36:02.000000000 +0200
+++ slrn-0.9.9-patched/src/Makefile.in	2008-09-10 22:22:40.000000000 +0200
@@ -88,9 +88,9 @@
 EXEC = slrn
 OTHER_EXECS = @SLRNPULL@
 CONFIG_H = config.h
-INCS = $(SLANG_INC) $(CANLOCK_INC) $(ICONV_INC) $(UUDEVIEW_INC) $(SSL_INC) $(MISCINCLUDES)
+INCS = $(SSL_INC) $(SLANG_INC) $(CANLOCK_INC) $(ICONV_INC) $(UUDEVIEW_INC) $(MISCINCLUDES)
 ALL_CFLAGS = $(CFLAGS) -Dunix $(INCS) $(DEFINES)
-EXECLIBS = $(SLANG_LIB) $(UUDEVIEW_LIB) $(CANLOCK_LIB) $(SSL_LIB) $(ICONV_LIB) $(SYSLIBS)
+EXECLIBS = $(SSL_LIB) $(SLANG_LIB) $(UUDEVIEW_LIB) $(CANLOCK_LIB) $(ICONV_LIB) $(SYSLIBS)
 DOT_O_DEPS = slrnfeat.h
 server_O_DEP = clientlib.c spool.c nntp.c sltcp.c nntplib.c
 chmap_O_DEP = chmap.h charmaps.h
