diff -aur tcpslice-1.1a3.orig/Makefile.in tcpslice-1.1a3/Makefile.in
--- tcpslice-1.1a3.orig/Makefile.in	1997-01-23 22:25:22.000000000 -0800
+++ tcpslice-1.1a3/Makefile.in	2008-02-05 18:22:33.000000000 -0800
@@ -94,10 +94,12 @@
 
 
 install: force
-	$(INSTALL) -m 555 -o bin -g bin tcpslice $(DESTDIR)$(BINDEST)
+	$(INSTALL) -d $(DESTDIR)$(BINDEST)
+	$(INSTALL) tcpslice $(DESTDIR)$(BINDEST)
 
 install-man: force
-	$(INSTALL) -m 444 -o bin -g bin $(srcdir)/$(PROG).1 \
+	$(INSTALL) -d $(DESTDIR)$(MANDEST)/man1
+	$(INSTALL) -m 644 $(srcdir)/$(PROG).1 \
 	    $(DESTDIR)$(MANDEST)/man1
 
 lint: $(GENSRC) force
