diff -ruN libxfce4util-4.2.0/configure libxfce4util-4.2.0-mod/configure
--- libxfce4util-4.2.0/configure	Wed Jan  5 15:33:56 2005
+++ libxfce4util-4.2.0-mod/configure	Thu Jan 20 11:24:17 2005
@@ -3935,11 +3935,11 @@
 
 
 if test "`uname | grep \"CYGWIN\"`" != ""; then
-  HAVE_CYGWIN_TRUE=
-  HAVE_CYGWIN_FALSE='#'
-else
   HAVE_CYGWIN_TRUE='#'
   HAVE_CYGWIN_FALSE=
+else
+  HAVE_CYGWIN_TRUE=
+  HAVE_CYGWIN_FALSE='#'
 fi
 
 
diff -ruN libxfce4util-4.2.0/libxfce4util/Makefile.in libxfce4util-4.2.0-mod/libxfce4util/Makefile.in
--- libxfce4util-4.2.0/libxfce4util/Makefile.in	Thu Dec 23 02:56:25 2004
+++ libxfce4util-4.2.0-mod/libxfce4util/Makefile.in	Thu Jan 20 11:24:53 2005
@@ -40,7 +40,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@HAVE_CYGWIN_TRUE@am__append_1 = -no-undefined
+@HAVE_CYGWIN_FALSE@am__append_1 = -no-undefined
 subdir = libxfce4util
 DIST_COMMON = $(libxfce4utilinclude_HEADERS) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in $(srcdir)/libxfce4util-1.0.pc.in \
