diff -ru oroborus-2.0.12/example.oroborusrc oroborus-2.0.12-patched/example.oroborusrc
--- oroborus-2.0.12/example.oroborusrc	Sun Jul  7 13:17:40 2002
+++ oroborus-2.0.12-patched/example.oroborusrc	Mon Feb 24 09:18:29 2003
@@ -11,10 +11,10 @@
 ##########################################
 
 # theme=<directory>
-theme=/usr/share/oroborus/themes/pillage
+theme=@FINK_PREFIX@/share/oroborus/themes/pillage
 
 # scheme=<scheme file>
-scheme=/usr/share/oroborus/schemes/oroborus
+scheme=@FINK_PREFIX@/share/oroborus/schemes/oroborus
 
 # font=<x font name>
 font=-adobe-helvetica-medium-r-*-*-*-100-*-*-*-*-iso8859-*
