diff -ruN git2r.orig/configure git2r/configure
--- git2r.orig/configure	2019-03-19 14:56:18.000000000 -0500
+++ git2r/configure	2019-04-06 08:34:22.000000000 -0500
@@ -3246,7 +3246,7 @@
 
     if test "x${PKGCONFIG_CFLAGS}${PKGCONFIG_LIBS}" = x; then
         case "${host_os}" in
-            darwin*)
+            darwin_nonfink*)
                 # Extract the first word of "brew", so it can be a program name with args.
 set dummy brew; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
@@ -4807,7 +4807,7 @@
 # Check for SSL for https transport
 have_ssl=no
 case "${host_os}" in
-    darwin*)
+    darwin_notfink*)
         # On macOS, use the Security and CoreFoundation framework
         have_ssl=yes
         CPPFLAGS="${CPPFLAGS} -DGIT_SECURE_TRANSPORT=1 -DGIT_HTTPS=1"
@@ -6699,7 +6699,7 @@
 
 # Specify sha1 implementation
 case "${host_os}" in
-    darwin*)
+    darwin_notfink*)
         CPPFLAGS="${CPPFLAGS} -DGIT_SHA1_COLLISIONDETECT=1 -DSHA1DC_NO_STANDARD_INCLUDES=1"
         CPPFLAGS="${CPPFLAGS} -DSHA1DC_CUSTOM_INCLUDE_SHA1_C=\\\"common.h\\\" -DSHA1DC_CUSTOM_INCLUDE_UBC_CHECK_C=\\\"common.h\\\""
         GIT2R_SRC_SHA1DC="libgit2/src/hash/sha1dc/sha1.o libgit2/src/hash/sha1dc/ubc_check.o"
