diff -ruN RGtk2-orig/src/gtkManuals.c RGtk2/src/gtkManuals.c
--- RGtk2-orig/src/gtkManuals.c	2017-05-07 08:31:44.000000000 -0500
+++ RGtk2/src/gtkManuals.c	2017-08-13 16:32:16.000000000 -0500
@@ -1210,7 +1210,7 @@
 
   *s_length = GET_LENGTH(s_ans);
   
-  guint8 *ans = g_malloc_n(*s_length, sizeof(guint8));
+  guint8 *ans = g_malloc((*s_length) * (sizeof(guint8)));
   memcpy(ans, RAW(s_ans), *s_length);
 
   UNPROTECT(1);
