--- Rserve/src/Makevars.in.orig	2014-05-06 22:52:06.000000000 -0400
+++ Rserve/src/Makevars.in	2014-05-06 22:55:00.000000000 -0400
@@ -12,8 +12,8 @@
 	$(CC) -DSTANDALONE_RSERVE -DRSERV_DEBUG -DNODAEMON -I. -Iinclude $(ALL_CPPFLAGS) $(ALL_CFLAGS) $(PKG_CPPFLAGS) $(PKG_CFLAGS) -o Rserve.dbg $(SERVER_SRC) $(ALL_LIBS) $(PKG_LIBS)
 
 # merging to bin/Rserve works only if installed from sources, won't work for binary
-	-./mergefat Rserve "$(R_HOME)/bin/Rserve"
-	-./mergefat Rserve.dbg "$(R_HOME)/bin/Rserve.dbg"
+#	-./mergefat Rserve "$(R_HOME)/bin/Rserve"
+#	-./mergefat Rserve.dbg "$(R_HOME)/bin/Rserve.dbg"
 
 client: config.h
 	cp config.h client/cxx/
