--- Rgraphviz/src/common.h.orig	2014-05-07 18:07:50.000000000 -0400
+++ Rgraphviz/src/common.h	2014-05-07 18:08:00.000000000 -0400
@@ -4,7 +4,6 @@
 #include <Rinternals.h>
 #include <Rdefines.h>
 #include <Rmath.h>
-#include <R_ext/RConverters.h>
 #include <R_ext/Rdynload.h>
 #include <R_ext/RS.h>		/* CallocCharBuf */
 
--- Rgraphviz/NAMESPACE.orig	2014-05-07 18:13:01.000000000 -0400
+++ Rgraphviz/NAMESPACE	2009-04-25 02:03:53.000000000 -0400
@@ -27,7 +27,7 @@
 importFrom(grid, convertHeight, convertWidth, current.viewport,
            current.vpTree, popViewport, pushViewport, unit)
 
+importFrom(methods, is, slot)
-importFrom(methods, "@<-", is, slot)
 
 importFrom(utils, head, tail)
 
