--- graph/src/graph.c.orig	2014-05-07 12:30:45.000000000 -0400
+++ graph/src/graph.c	2014-05-07 12:31:06.000000000 -0400
@@ -1,7 +1,6 @@
 #include <Rinternals.h>
 #include <Rdefines.h>
 #include <Rmath.h>
-#include <R_ext/RConverters.h>
 #include <R_ext/Rdynload.h>
 
 SEXP R_scalarString(const char *);
--- graph/NAMESPACE.orig	2014-05-07 12:34:10.000000000 -0400
+++ graph/NAMESPACE	2014-05-07 12:35:45.000000000 -0400
@@ -3,7 +3,7 @@
 importClassesFrom(methods, ANY, character, list, logical, matrix,
                   missing, "NULL", numeric, oldClass, vector)
 importMethodsFrom(methods, coerce, initialize, show)
-importFrom(methods, "@<-", as, callGeneric, is, isVirtualClass, new,
+importFrom(methods, as, callGeneric, is, isVirtualClass, new,
            slot, "slot<-", slotNames, validObject)
 importFrom(stats, as.dist, dhyper, phyper, runif)
 importFrom(tools, package.dependencies)

