diff -Nurd classicladder/Makefile classicladder.patched/Makefile
--- classicladder/Makefile	2012-08-17 05:27:03.000000000 -0700
+++ classicladder.patched/Makefile	2012-08-23 07:09:54.000000000 -0700
@@ -29,7 +29,7 @@
 ###ifndef WINDOWS
 # Comment if you do not want the I/O access (not x86 platform)
 # Used in the hardware.c file.
-MAKE_IO_ACCESS = 1
+#MAKE_IO_ACCESS = 1
 ###endif
 
 # Uncomment the following flag to make the MAT connection version
@@ -71,8 +71,8 @@
 endif
 
 ifdef GTK_INTERFACE
-OWN_CFLAGS += -Wall `pkg-config gtk+-2.0 --cflags` `pkg-config pango --cflags` `pkg-config gthread-2.0 --cflags` -DGTK_INTERFACE -DGTK2
-OWN_LIBS += `pkg-config gtk+-2.0 --cflags --libs` `pkg-config pango --cflags --libs` `pkg-config gthread-2.0 --cflags --libs`
+OWN_CFLAGS += -Wall `pkg-config --cflags fontconfig` `pkg-config gtk+-2.0 --cflags` `pkg-config pango --cflags` `pkg-config gthread-2.0 --cflags` -DGTK_INTERFACE -DGTK2
+OWN_LIBS += `pkg-config --cflags --libs fontconfig` `pkg-config gtk+-2.0 --cflags --libs` `pkg-config pango --cflags --libs` `pkg-config gthread-2.0 --cflags --libs`
 ###ifdef GNOME_PRINT
 ###OWN_CFLAGS += `pkg-config --cflags libgnomeprintui-2.2` -DGNOME_PRINT_USE
 ###OWN_LIBS += `pkg-config --libs libgnomeprintui-2.2`
@@ -176,7 +176,7 @@
 endif
 
 classicladder: $(OBJ_CLASSICLADDER) classicladder_.o hardware_.o
-	$(CC) -o classicladder $(OBJ_CLASSICLADDER) hardware_.o classicladder_.o $(OWN_LIBS) $(COMEDI_LIB)
+	$(CC) -o classicladder $(OBJ_CLASSICLADDER) hardware_.o classicladder_.o $(OWN_LIBS) $(COMEDI_LIB) -lz
 
 classicladder-mat: $(OBJ_CLASSICLADDER_MAT) classicladder_mat.o
 	libtool $(CC) -o classicladder-mat $(OBJ_CLASSICLADDER_MAT) classicladder_mat.o $(OWN_LIBS)
diff -Nurd classicladder/tasks.h classicladder.patched/tasks.h
--- classicladder/tasks.h	2012-08-02 06:08:36.000000000 -0700
+++ classicladder.patched/tasks.h	2012-08-23 06:59:39.000000000 -0700
@@ -1,4 +1,4 @@
-
+#include <time.h>
 #define NBR_TASKS 10
 
 #define ID_TASK_SCAN_INPUTS 0
