diff -Naur --exclude='*~' E/HEURISTICS/che_axfilter.c E.fink/HEURISTICS/che_axfilter.c
--- E/HEURISTICS/che_axfilter.c	2012-07-10 00:07:04.000000000 +0200
+++ E.fink/HEURISTICS/che_axfilter.c	2012-07-30 17:15:05.000000000 +0200
@@ -5,7 +5,7 @@
 Author: Stephan Schulz (schulz@eprover.org)
 
 Contents
- 
+
   Code implementing axiom filters as a data type.
 
   Copyright 2011 by the author.
@@ -86,7 +86,7 @@
 // Function: get_gen_measure()
 //
 //   Given a string, return the corresponding GenMeasure, or 0 on
-//   failure. 
+//   failure.
 //
 // Global Variables: -
 //
@@ -115,16 +115,16 @@
 //
 //   Allocate an empty, initialized axiom filter description.
 //
-// Global Variables: 
+// Global Variables:
 //
-// Side Effects    : 
+// Side Effects    :
 //
 /----------------------------------------------------------------------*/
 
 AxFilter_p AxFilterAlloc()
 {
    AxFilter_p handle = AxFilterCellAlloc();
-   
+
    handle->name                 = NULL;
    handle->type                 = AFNoFilter;
    handle->gen_measure          = GMNoMeasure;
@@ -171,9 +171,9 @@
 //
 //   The preliminary syntax is:
 //
-//    GSinE(<g-measure:type>, <[no]hypos>,<benvolvence:double>, 
+//    GSinE(<g-measure:type>, <[no]hypos>,<benvolvence:double>,
 //    <generosity:int>, <rec-depth:int>, <set-size:int>,
-//    <set-fraction:double>) 
+//    <set-fraction:double>)
 //
 //    where "GSinE" represents AFGSineE, "Generalized SinE", currently
 //    the only support filter type. Other filter types can support
@@ -195,20 +195,20 @@
    AcceptInpId(in, "GSinE");
    res->type = AFGSinE;
    AcceptInpTok(in, OpenBracket);
-   
+
    res->gen_measure = get_gen_measure(DStrView(AktToken(in)->literal));
 
    if(!res->gen_measure)
    {
-      AktTokenError(in, "Unknown generality measure", false);      
+      AktTokenError(in, "Unknown generality measure", false);
    }
    if(res->gen_measure!=GMTerms && res->gen_measure!=GMFormulas)
    {
-      AktTokenError(in, "Generality measure not yet implemented", false);      
+      AktTokenError(in, "Generality measure not yet implemented", false);
    }
    NextToken(in);
    AcceptInpTok(in, Comma);
-   
+
    if(!TestInpTok(in, Comma))
    {
       CheckInpId(in, "hypos|nohypos");
@@ -216,7 +216,7 @@
       {
          res->use_hypotheses = true;
       }
-      NextToken(in);         
+      NextToken(in);
    }
    AcceptInpTok(in, Comma);
    if(!TestInpTok(in, Comma))
@@ -224,7 +224,7 @@
       res->benevolence = ParseFloat(in);
    }
    AcceptInpTok(in, Comma);
-   
+
    if(!TestInpTok(in, Comma))
    {
       res->generosity = AktToken(in)->numval;
@@ -242,18 +242,18 @@
       res->max_set_size = AktToken(in)->numval;
       AcceptInpTok(in, PosInt);
    }
-   AcceptInpTok(in, Comma); 
+   AcceptInpTok(in, Comma);
    if(!TestInpTok(in, CloseBracket|Comma))
    {
       res->max_set_fraction = ParseFloat(in);
    }
    if(TestInpTok(in, Comma))
    {
-      AcceptInpTok(in, Comma);       
+      AcceptInpTok(in, Comma);
       res->add_no_symbol_axioms = TestInpId(in, "addnosymb");
       AcceptInpId(in, "addnosymb|ignorenosymb");
    }
-  
+
 
    AcceptInpTok(in, CloseBracket);
 
@@ -285,7 +285,7 @@
    AcceptInpId(in, "Threshold");
    res->type = AFThreshold;
    AcceptInpTok(in, OpenBracket);
-   
+
    res->threshold = AktToken(in)->numval;
    AcceptInpTok(in, PosInt);
    AcceptInpTok(in, CloseBracket);
@@ -310,7 +310,7 @@
 AxFilter_p AxFilterParse(Scanner_p in)
 {
    CheckInpId(in, "GSinE|Threshold");
-   
+
    if(TestInpId(in, "GSinE"))
    {
       return GSinEParse(in);
@@ -334,7 +334,7 @@
 //   generated name of the form "axfilter_auto%4udd" is
 //   generated. This name is unique among auto-generated names (up to
 //   the period of unsigned long, but not checked against manually
-//   given names. 
+//   given names.
 //
 // Global Variables: -
 //
@@ -352,9 +352,9 @@
    if(TestTok(LookToken(in,1), EqualSign))
    {
       CheckInpTok(in, Identifier);
-      name = SecureStrdup(DStrView(AktToken(in)->literal)); 
+      name = SecureStrdup(DStrView(AktToken(in)->literal));
       NextToken(in);
-      AcceptInpTok(in, EqualSign);      
+      AcceptInpTok(in, EqualSign);
    }
    else
    {
@@ -363,7 +363,7 @@
    }
    res = AxFilterParse(in);
    res->name = name;
-   
+
    return res;
 }
 
@@ -390,7 +390,7 @@
    {
    case AFGSinE:
          res = snprintf(buf, buflen, "%s(%s, %s, %f, %ld, %ld, %lld, %f)",
-                        "GSinE", 
+                        "GSinE",
                         GeneralityMeasureNames[filter->gen_measure],
                         filter->use_hypotheses?"hypos":"nohypos",
                         filter->benevolence,
@@ -400,7 +400,7 @@
                         filter->max_set_fraction);
          break;
    case AFThreshold:
-         res = snprintf(buf, buflen, "Threshold(%ld)", 
+         res = snprintf(buf, buflen, "Threshold(%ld)",
                         filter->threshold);
          break;
    default:
@@ -474,11 +474,11 @@
 /----------------------------------------------------------------------*/
 
 AxFilterSet_p AxFilterSetAlloc()
-{   
+{
    AxFilterSet_p set = AxFilterSetCellAlloc();
-   
+
    set->set = PStackAlloc();
-   return set;   
+   return set;
 }
 
 
@@ -513,7 +513,7 @@
 // Function: AxFilterSetParse()
 //
 //   Parse a set of axfilter definitions. Returns number of filters
-//   parsed. 
+//   parsed.
 //
 // Global Variables: -
 //
@@ -525,7 +525,7 @@
 {
    long res = 0;
    AxFilter_p filter;
-   
+
    while(TestInpTok(in, Identifier))
    {
       filter = AxFilterDefParse(in);
@@ -540,7 +540,7 @@
 // Function: AxFilterSetCreateInternal()
 //
 //   Create and return an AxFilterSet from a provided string
-//   description. 
+//   description.
 //
 // Global Variables: -
 //
@@ -550,10 +550,10 @@
 
 AxFilterSet_p AxFilterSetCreateInternal(char* str)
 {
-   AxFilterSet_p filters = AxFilterSetAlloc();   
+   AxFilterSet_p filters = AxFilterSetAlloc();
    Scanner_p in = CreateScanner(StreamTypeInternalString, str, true, NULL);
    AxFilterSetParse(in, filters);
-   DestroyScanner(in); 
+   DestroyScanner(in);
    return filters;
 }
 
@@ -638,12 +638,10 @@
       DStrAppendStr(res, tmp->name);
       sep = ", ";
    }
-   return res;
+   // return res;
 }
 
 
 /*---------------------------------------------------------------------*/
 /*                        End of File                                  */
 /*---------------------------------------------------------------------*/
-
-
diff -Naur --exclude='*~' E/Makefile E.fink/Makefile
--- E/Makefile	2012-07-10 01:14:00.000000000 +0200
+++ E.fink/Makefile	2012-07-30 17:04:04.000000000 +0200
@@ -5,7 +5,7 @@
 # Author: Stephan Schulz
 #
 # Top level make file. Check Makefile.vars for system-dependend tool
-# selection and compilation options. 
+# selection and compilation options.
 #
 # Changes
 #
@@ -37,7 +37,7 @@
 
 # Generate dependencies
 
-depend: 
+depend:
 	for subdir in $(CODE); do\
 	   cd $$subdir; touch Makefile.dependencies; $(MAKE) depend; cd ..;\
 	done;
@@ -78,7 +78,7 @@
 	@cd ..; find $(PROJECT) -name ".git" -print >> $(PROJECT)/etc/NO_DISTRIB;\
          $(TAR) cfX - $(PROJECT)/etc/NO_DISTRIB $(PROJECT) |$(GZIP) - -c > $(PROJECT).tgz
 
-# Include proprietary code not part of the GPL'ed version, 
+# Include proprietary code not part of the GPL'ed version,
 # as well as CVS subdirecctories
 
 fulldistrib: man documentation cleandist default_config
@@ -104,7 +104,7 @@
            $(LN) ../$$subdir/$$subdir.a .;\
 	done;
 
-tags: 
+tags:
 	etags */*.c */*.h
 	cd PYTHON; make ptags
 
@@ -115,7 +115,7 @@
 # Rebuilding from scratch
 
 rebuild:
-	echo 'Rebuilding with debug options $(DEBUGFLAGS)'	
+	echo 'Rebuilding with debug options $(DEBUGFLAGS)'
 	$(MAKE) clean
 	$(MAKE) config
 	$(MAKE) depend
@@ -123,7 +123,7 @@
 
 # Configure the whole package
 
-config: 
+config:
 	echo 'Configuring build system and tools'
 	$(MAKE) links
 	$(MAKE) tools
@@ -133,32 +133,32 @@
 # Configure and copy executables to the installation directory
 
 install: top
-	-sh -c 'mkdir -p $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/eprover      $(EXECPATH)' 
-	-sh -c 'development_tools/e_install PROVER/epclextract  $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/eproof       $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/eproof_ram   $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/eground      $(EXECPATH)'	
-	-sh -c 'development_tools/e_install PROVER/e_ltb_runner $(EXECPATH)'	
-	-sh -c 'development_tools/e_install PROVER/e_axfilter   $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/checkproof   $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/ekb_create   $(EXECPATH)'	
-	-sh -c 'development_tools/e_install PROVER/ekb_delete   $(EXECPATH)'	
-	-sh -c 'development_tools/e_install PROVER/ekb_ginsert  $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/ekb_insert   $(EXECPATH)'
-	-sh -c 'mkdir -p $(MANPATH)'
-	-sh -c 'development_tools/e_install DOC/man/eprover.1      $(MANPATH)'
-	-sh -c 'development_tools/e_install DOC/man/epclextract.1  $(MANPATH)'
-	-sh -c 'development_tools/e_install DOC/man/eproof.1       $(MANPATH)'
-	-sh -c 'development_tools/e_install DOC/man/eproof_ram.1   $(MANPATH)'
-	-sh -c 'development_tools/e_install DOC/man/eground.1      $(MANPATH)'
-	-sh -c 'development_tools/e_install DOC/man/e_ltb_runner.1 $(MANPATH)'
-	-sh -c 'development_tools/e_install DOC/man/e_axfilter.1   $(MANPATH)'
-	-sh -c 'development_tools/e_install DOC/man/checkproof.1   $(MANPATH)'
-	-sh -c 'development_tools/e_install DOC/man/ekb_create.1   $(MANPATH)'
-	-sh -c 'development_tools/e_install DOC/man/ekb_delete.1   $(MANPATH)'
-	-sh -c 'development_tools/e_install DOC/man/ekb_ginsert.1  $(MANPATH)'
-	-sh -c 'development_tools/e_install DOC/man/ekb_insert.1   $(MANPATH)'
+	-sh -c 'mkdir -p $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/eprover      $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/epclextract  $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/eproof       $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/eproof_ram   $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/eground      $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/e_ltb_runner $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/e_axfilter   $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/checkproof   $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/ekb_create   $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/ekb_delete   $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/ekb_ginsert  $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/ekb_insert   $(DESTDIR)$(EXECPATH)'
+	-sh -c 'mkdir -p $(DESTDIR)$(MANPATH)'
+	-sh -c 'development_tools/e_install DOC/man/eprover.1      $(DESTDIR)$(MANPATH)'
+	-sh -c 'development_tools/e_install DOC/man/epclextract.1  $(DESTDIR)$(MANPATH)'
+	-sh -c 'development_tools/e_install DOC/man/eproof.1       $(DESTDIR)$(MANPATH)'
+	-sh -c 'development_tools/e_install DOC/man/eproof_ram.1   $(DESTDIR)$(MANPATH)'
+	-sh -c 'development_tools/e_install DOC/man/eground.1      $(DESTDIR)$(MANPATH)'
+	-sh -c 'development_tools/e_install DOC/man/e_ltb_runner.1 $(DESTDIR)$(MANPATH)'
+	-sh -c 'development_tools/e_install DOC/man/e_axfilter.1   $(DESTDIR)$(MANPATH)'
+	-sh -c 'development_tools/e_install DOC/man/checkproof.1   $(DESTDIR)$(MANPATH)'
+	-sh -c 'development_tools/e_install DOC/man/ekb_create.1   $(DESTDIR)$(MANPATH)'
+	-sh -c 'development_tools/e_install DOC/man/ekb_delete.1   $(DESTDIR)$(MANPATH)'
+	-sh -c 'development_tools/e_install DOC/man/ekb_ginsert.1  $(DESTDIR)$(MANPATH)'
+	-sh -c 'development_tools/e_install DOC/man/ekb_insert.1   $(DESTDIR)$(MANPATH)'
 
 
 # Also remake documentation
@@ -189,4 +189,3 @@
 	for subdir in $(CODE); do\
 	   cd $$subdir;touch Makefile.dependencies;$(MAKE);cd ..;\
 	done;
-
diff -Naur --exclude='*~' E/Makefile.vars E.fink/Makefile.vars
--- E/Makefile.vars	2012-07-11 08:39:07.000000000 +0200
+++ E.fink/Makefile.vars	2012-07-30 17:10:44.000000000 +0200
@@ -17,7 +17,7 @@
 
 
 # EXECPATH is where make install-exec will move the more important
-# executables. Edit it to point to wherever you want them to live. 
+# executables. Edit it to point to wherever you want them to live.
 # Note that ./configure takes care of this automatically.
 
 EXECPATH = /Users/schulz/E/PROVER
@@ -50,7 +50,7 @@
 # ======================
 
 # Remove the comment from the following to enable gprof-style
-#  profiling 
+#  profiling
 #  -> This will break some stuff on at least some Solaris 2.6
 #     machines, which seem to lack a profiling-enabled libm. I have a
 #     simple sqrt() imlementation to circumvent the problem...
@@ -65,8 +65,8 @@
 # BUILDFLAGS:
 #
 # PRINT_SOMEERRORS_STDOUT: Print various error messages (out of
-#                          memory, empty input file) to stdout  
-#                          (otherwise only to stderr) 
+#                          memory, empty input file) to stdout
+#                          (otherwise only to stderr)
 # USE_NEWMEM: Use a memory management system like everybody else,
 #             using free lists filled up by allocating large blocks
 #             and hacking them  into suitabe pieces. Contrary to
@@ -75,22 +75,22 @@
 #             compared to its native memory management. It's
 #             left in as a warning reminder only.
 # USE_SYSTEM_MEM: Use normal malloc()/free instead of the build-in
-#                  memory management. Does not combine with USE_NEWMEM! 
+#                  memory management. Does not combine with USE_NEWMEM!
 # CLAUSE_PERM_IDENT: Clauses have an extra unchanging identifier. Useful
-#                 for testing some proerties  
+#                 for testing some proerties
 #
 # MEASURE_EXPENSIVE: Compile counting operations and things into the
 #                    code even in time-critical sections.
 # PRINT_SHARING: Determine and print the sharing factor of the proof
-#                state for each clause activation. 
+#                state for each clause activation.
 # PRINT_RW_STATE: Dump R,E,NEW in each loop traversal.
 #
 # FULL_MEM_STATS: Print size of the most important data types and
-#                 information about allocated memory. 
+#                 information about allocated memory.
 # CONSTANT_MEM_ESTIMATE: Use normalized portable data type estimates
-#                        instead of sizeof() to get actual machine data 
+#                        instead of sizeof() to get actual machine data
 #                        sizes. Necessary to make the prover behave _exactly_
-#                        the same on different machines, but makes memory 
+#                        the same on different machines, but makes memory
 #                        estimation worse on most machines!
 # STACK_SIZE=VALUE: Increase the system stacksize to value and run a
 #                   freshly exec()'ed copy of E. This seems to work
@@ -118,22 +118,22 @@
 
 ARCHFLAGS  =  # -mdynamic-no-pic # Helps a tiny bit on Mac
               # -mcpu=v8 -mtune=ultrasparc # Somewhat better than
-                                           # without, v9 is 
+                                           # without, v9 is
                                            # worse (!), ultrasparc does
                                            # not work at all (probably
                                            # due to 64 bit include files
-                                           # or something...) 
+                                           # or something...)
 # ARCHFLAGS   = -march=i686  # No detectable improvement
 
 # CFLAGS for different setups:
-# 
+#
 # Solaris/Linux with gcc
 #
 # CFLAGS     = -O6 -Wall -Wno-char-subscripts -Wshadow -ansi \
 #             -pedantic -I../include $(DEBUGFLAGS) \
 #             $(BUILDFLAGS) $(ARCHFLAGS)
 
-CFLAGS     = -O6 -fomit-frame-pointer -Wall -Wno-char-subscripts -std=gnu99 \
+CFLAGS     = -fomit-frame-pointer -Wall -Wno-char-subscripts -std=gnu99 \
              -I../include $(DEBUGFLAGS) \
              $(BUILDFLAGS) $(ARCHFLAGS)
 
@@ -172,11 +172,7 @@
 
 # Both makedepend and a good C compiler can be used to generate
 # dependencies. gcc does a better job than makedepend, so it is the
-# default. 
+# default.
 #
 # MAKEDEPEND = echo "" > Makefile.dependencies; makedepend -- $(CFLAGS) -- -f Makefile.dependencies *.c
 MAKEDEPEND = gcc -M $(CFLAGS) *.c > Makefile.dependencies
-
-
-
-
diff -Naur --exclude='*~' E/SIMPLE_APPS/eprove E.fink/SIMPLE_APPS/eprove
--- E/SIMPLE_APPS/eprove	1970-01-01 01:00:00.000000000 +0100
+++ E.fink/SIMPLE_APPS/eprove	2012-07-30 18:10:58.000000000 +0200
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+@PREFIX@/bin/eprover -l4 -xAuto -tAuto -R --tptp3-in $1 | @PREFIX@/bin/epclextract
