diff -Naur --ignore-space-change '--exclude=*~' E/Makefile E.fink/Makefile
--- E/Makefile	2012-11-07 23:31:30.000000000 +0100
+++ E.fink/Makefile	2013-03-28 00:13:06.000000000 +0100
@@ -133,19 +133,19 @@
 # Configure and copy executables to the installation directory
 
 install: top
-	-sh -c 'mkdir -p $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/eprover      $(EXECPATH)' 
-	-sh -c 'development_tools/e_install PROVER/epclextract  $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/eproof       $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/eproof_ram   $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/eground      $(EXECPATH)'	
-	-sh -c 'development_tools/e_install PROVER/e_ltb_runner $(EXECPATH)'	
-	-sh -c 'development_tools/e_install PROVER/e_axfilter   $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/checkproof   $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/ekb_create   $(EXECPATH)'	
-	-sh -c 'development_tools/e_install PROVER/ekb_delete   $(EXECPATH)'	
-	-sh -c 'development_tools/e_install PROVER/ekb_ginsert  $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/ekb_insert   $(EXECPATH)'
+	-sh -c 'mkdir -p $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/eprover      $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/epclextract  $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/eproof       $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/eproof_ram   $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/eground      $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/e_ltb_runner $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/e_axfilter   $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/checkproof   $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/ekb_create   $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/ekb_delete   $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/ekb_ginsert  $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/ekb_insert   $(DESTDIR)$(EXECPATH)'
 	-sh -c 'mkdir -p $(MANPATH)'
 	-sh -c 'development_tools/e_install DOC/man/eprover.1      $(MANPATH)'
 	-sh -c 'development_tools/e_install DOC/man/epclextract.1  $(MANPATH)'
@@ -189,4 +189,3 @@
 	for subdir in $(CODE); do\
 	   cd $$subdir;touch Makefile.dependencies;$(MAKE);cd ..;\
 	done;
-
diff -Naur --ignore-space-change '--exclude=*~' E/Makefile.vars E.fink/Makefile.vars
--- E/Makefile.vars	2013-03-24 01:31:07.000000000 +0100
+++ E.fink/Makefile.vars	2013-03-28 00:00:26.000000000 +0100
@@ -133,7 +133,7 @@
 #             -pedantic -I../include $(DEBUGFLAGS) \
 #             $(BUILDFLAGS) $(ARCHFLAGS)
 
-CFLAGS     = -O6 -fomit-frame-pointer -Wall -Wno-char-subscripts -std=gnu99 \
+CFLAGS     = -fomit-frame-pointer -Wall -Wno-char-subscripts -std=gnu99 \
              -I../include $(DEBUGFLAGS) \
              $(BUILDFLAGS) $(ARCHFLAGS)
 
@@ -176,7 +176,3 @@
 #
 # MAKEDEPEND = echo "" > Makefile.dependencies; makedepend -- $(CFLAGS) -- -f Makefile.dependencies *.c
 MAKEDEPEND = gcc -M $(CFLAGS) *.c > Makefile.dependencies
-
-
-
-
