--- grib_api-1.12.0.orig/configure	2014-03-05 16:36:51.000000000 +0100
+++ grib_api-1.12.0/configure	2014-05-05 15:57:31.000000000 +0200
@@ -699,10 +699,10 @@
 MAKEFLAGS=
 
 # Identity of this package.
-PACKAGE_NAME='grib_api'
-PACKAGE_TARNAME='grib_api'
+PACKAGE_NAME='grib-api'
+PACKAGE_TARNAME='grib-api'
 PACKAGE_VERSION=' '
-PACKAGE_STRING='grib_api  '
+PACKAGE_STRING='grib-api  '
 PACKAGE_BUGREPORT='Software.Support@ecmwf.int'
 PACKAGE_URL=''
 
@@ -10806,9 +10806,9 @@
 
 
 
-definition_files_path="${datadir}/grib_api/definitions"
-samples_files_path="${datadir}/grib_api/samples"
-ifs_samples_files_path="${datadir}/grib_api/ifs_samples"
+definition_files_path="${datadir}/${PACKAGE}/definitions"
+samples_files_path="${datadir}/${PACKAGE}/samples"
+ifs_samples_files_path="${datadir}/${PACKAGE}/ifs_samples"
 default_perl_install="${prefix}/perl"
 
 
--- grib_api-1.12.0.orig/version.sh	2014-01-03 16:52:56.000000000 +0100
+++ grib_api-1.12.0/version.sh	2014-05-05 15:58:09.000000000 +0200
@@ -10,7 +10,7 @@
 #
 #
 # Package base name
-PACKAGE_NAME='grib_api'
+PACKAGE_NAME='grib-api'
 
 # Package version
 GRIB_API_MAJOR_VERSION=1
--- grib_api-1.12.0.orig/config/ltmain.sh	2014-03-05 16:36:49.000000000 +0100
+++ grib_api-1.12.0/config/ltmain.sh	2014-05-05 14:37:26.000000000 +0200
@@ -6256,13 +6256,13 @@
 	  # Like Linux, but with the current version available in
 	  # verstring for coding it into the library header
 	  func_arith $current - $age
-	  major=.$func_arith_result
+	  major=.1
 	  versuffix="$major.$age.$revision"
 	  # Darwin ld doesn't like 0 for these options...
 	  func_arith $current + 1
 	  minor_current=$func_arith_result
 	  xlcverstring="${wl}-compatibility_version ${wl}$minor_current ${wl}-current_version ${wl}$minor_current.$revision"
-	  verstring="-compatibility_version $minor_current -current_version $minor_current.$revision"
+	  verstring="-compatibility_version 1.9.0 -current_version 1.12.0"
 	  ;;
 
 	freebsd-aout)
--- grib_api-1.12.0.orig/fortran/Makefile.in	2014-03-05 16:36:53.000000000 +0100
+++ grib_api-1.12.0/fortran/Makefile.in	2014-05-05 14:42:27.000000000 +0200
@@ -172,7 +172,7 @@
 LEXLIB = @LEXLIB@
 LEX_OUTPUT_ROOT = @LEX_OUTPUT_ROOT@
 LIBOBJS = @LIBOBJS@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lgrib_api
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIB_JASPER = @LIB_JASPER@
@@ -296,10 +296,10 @@
 lib_LTLIBRARIES = libgrib_api_f77.la libgrib_api_f90.la
 include_HEADERS = grib_api_f77.h
 libgrib_api_f77_la_SOURCES = grib_fortran.c grib_f77.c 
-libgrib_api_f77_la_DEPENDENCIES = $(top_builddir)/src/libgrib_api.la
+libgrib_api_f77_la_DEPENDENCIES =
 libgrib_api_f77_la_LDFLAGS = -version-info $(GRIB_ABI_CURRENT):$(GRIB_ABI_REVISION):$(GRIB_ABI_AGE)
 libgrib_api_f90_la_SOURCES = grib_fortran.c grib_f90.f90 
-libgrib_api_f90_la_DEPENDENCIES = $(top_builddir)/src/libgrib_api.la grib_api_externals.h grib_api_visibility.h grib_api_constants.h grib_kinds.h
+libgrib_api_f90_la_DEPENDENCIES = grib_api_externals.h grib_api_visibility.h grib_api_constants.h grib_kinds.h
 libgrib_api_f90_la_LDFLAGS = -version-info $(GRIB_ABI_CURRENT):$(GRIB_ABI_REVISION):$(GRIB_ABI_AGE)
 libgrib_api_fortran_prototypes = grib_fortran.c
 @UPPER_CASE_MOD_FALSE@nodist_include_HEADERS = grib_api.mod
