diff -uNr gromacs-4.6/CMakeLists.txt gromacs-4.6.fink/CMakeLists.txt
--- gromacs-4.6/CMakeLists.txt	2013-01-18 15:46:08.000000000 -0500
+++ gromacs-4.6.fink/CMakeLists.txt	2013-02-12 18:23:54.000000000 -0500
@@ -64,6 +64,7 @@
     set(PROJECT_VERSION ${CUSTOM_VERSION_STRING})
 endif (CUSTOM_VERSION_STRING)
 set(SOVERSION 6)
+set(VERSION 6.0.0)
 # It is a bit irritating, but this has to be set separately for now!
 SET(CPACK_PACKAGE_VERSION_MAJOR "4")
 SET(CPACK_PACKAGE_VERSION_MINOR "6")
diff -uNr gromacs-4.6/src/gmxlib/CMakeLists.txt gromacs-4.6.fink/src/gmxlib/CMakeLists.txt
--- gromacs-4.6/src/gmxlib/CMakeLists.txt	2013-01-18 15:38:57.000000000 -0500
+++ gromacs-4.6.fink/src/gmxlib/CMakeLists.txt	2013-02-12 18:25:19.000000000 -0500
@@ -117,7 +117,7 @@
 if(USE_VERSION_H)
 	add_dependencies(gmx gmx_version) 
 endif()
-set_target_properties(gmx PROPERTIES OUTPUT_NAME "gmx${GMX_LIBS_SUFFIX}" SOVERSION ${SOVERSION} INSTALL_NAME_DIR "${LIB_INSTALL_DIR}"
+set_target_properties(gmx PROPERTIES OUTPUT_NAME "gmx${GMX_LIBS_SUFFIX}" SOVERSION ${SOVERSION} VERSION ${VERSION} INSTALL_NAME_DIR "${LIB_INSTALL_DIR}"
     COMPILE_FLAGS "${OpenMP_C_FLAGS}" COMPILE_DEFINITIONS "TMPI_EXPORTS")
 
 install(TARGETS gmx DESTINATION ${LIB_INSTALL_DIR} COMPONENT libraries)
diff -uNr gromacs-4.6/src/kernel/CMakeLists.txt gromacs-4.6.fink/src/kernel/CMakeLists.txt
--- gromacs-4.6/src/kernel/CMakeLists.txt	2013-01-18 15:38:58.000000000 -0500
+++ gromacs-4.6.fink/src/kernel/CMakeLists.txt	2013-02-12 18:26:51.000000000 -0500
@@ -76,7 +76,7 @@
 
 add_library(gmxpreprocess ${GMXPREPROCESS_SOURCES})
 target_link_libraries(gmxpreprocess md)
-set_target_properties(gmxpreprocess PROPERTIES OUTPUT_NAME "gmxpreprocess${GMX_LIBS_SUFFIX}" SOVERSION ${SOVERSION} INSTALL_NAME_DIR "${LIB_INSTALL_DIR}"
+set_target_properties(gmxpreprocess PROPERTIES OUTPUT_NAME "gmxpreprocess${GMX_LIBS_SUFFIX}" SOVERSION ${SOVERSION} VERSION ${VERSION} INSTALL_NAME_DIR "${LIB_INSTALL_DIR}"
     COMPILE_FLAGS "${OpenMP_C_FLAGS}")
 
 
diff -uNr gromacs-4.6/src/mdlib/CMakeLists.txt gromacs-4.6.fink/src/mdlib/CMakeLists.txt
--- gromacs-4.6/src/mdlib/CMakeLists.txt	2013-01-18 15:38:58.000000000 -0500
+++ gromacs-4.6.fink/src/mdlib/CMakeLists.txt	2013-02-12 18:25:52.000000000 -0500
@@ -73,7 +73,7 @@
 else()
   set(MD_PREFIX)
 endif()
-set_target_properties(md PROPERTIES OUTPUT_NAME "${MD_PREFIX}md${GMX_LIBS_SUFFIX}" SOVERSION ${SOVERSION} INSTALL_NAME_DIR "${LIB_INSTALL_DIR}"
+set_target_properties(md PROPERTIES OUTPUT_NAME "${MD_PREFIX}md${GMX_LIBS_SUFFIX}" SOVERSION ${SOVERSION} VERSION ${VERSION} INSTALL_NAME_DIR "${LIB_INSTALL_DIR}"
     COMPILE_FLAGS "${OpenMP_C_FLAGS}")
 
 install(TARGETS md DESTINATION ${LIB_INSTALL_DIR} COMPONENT libraries)
diff -uNr gromacs-4.6/src/tools/CMakeLists.txt gromacs-4.6.fink/src/tools/CMakeLists.txt
--- gromacs-4.6/src/tools/CMakeLists.txt	2013-01-18 15:38:58.000000000 -0500
+++ gromacs-4.6.fink/src/tools/CMakeLists.txt	2013-02-12 18:28:59.000000000 -0500
@@ -71,7 +71,7 @@
 
 
 target_link_libraries(gmxana md gmx ${GSL_LIBRARIES})
-set_target_properties(gmxana PROPERTIES OUTPUT_NAME "gmxana${GMX_LIBS_SUFFIX}" SOVERSION ${SOVERSION} INSTALL_NAME_DIR "${LIB_INSTALL_DIR}"
+set_target_properties(gmxana PROPERTIES OUTPUT_NAME "gmxana${GMX_LIBS_SUFFIX}" SOVERSION ${SOVERSION} VERSION ${VERSION} INSTALL_NAME_DIR "${LIB_INSTALL_DIR}"
     COMPILE_FLAGS "${OpenMP_C_FLAGS}")
 
 # List of programs with single corresponding .c source file,
