diff -Nurd '-x*~' libghemical-2.99.1.orig/src/Makefile.in libghemical-2.99.1/src/Makefile.in
--- libghemical-2.99.1.orig/src/Makefile.in	2009-09-11 08:25:41.000000000 -0400
+++ libghemical-2.99.1/src/Makefile.in	2012-02-25 22:29:31.000000000 -0500
@@ -55,7 +55,7 @@
 	"$(DESTDIR)$(libghemicalincludedir)"
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libghemical_la_LIBADD =
+libghemical_la_LIBADD = $(INTLLIBS)
 am_libghemical_la_OBJECTS = atom.lo bond.lo constraint.lo model.lo \
 	engine.lo typerule.lo utility.lo chn_info.lo seqbuild.lo \
 	mfinder.lo eng1_qm.lo eng1_qm_mpqc.lo eng1_qm_mopac.lo \
