diff -Nurd linear-algebra/src/Makefile linear-algebra.patched/src/Makefile
--- linear-algebra/src/Makefile	2011-11-02 14:36:05.000000000 -0400
+++ linear-algebra.patched/src/Makefile	2012-01-20 08:07:25.000000000 -0500
@@ -4,6 +4,8 @@
 MKOCTFILE = mkoctfile
 endif
 
+LAPACK_LIBS := $(shell mkoctfile -p LAPACK_LIBS)
+
 DEFINES = -DHAVE_CONFIG_H -Wall
 GSVD_OBJECTS = gsvd.o dbleGSVD.o CmplxGSVD.o 
 GSVD_TARGET = gsvd.oct
@@ -23,7 +25,7 @@
 all : $(TARGETS) pgmres.oct
 
 $(GSVD_TARGET) : $(GSVD_DEPENDS) $(GSVD_OBJECTS)
-	$(MKOCTFILE) $(DEFINES) $(GSVD_OBJECTS) -o $@
+	$(MKOCTFILE) $(DEFINES) $(GSVD_OBJECTS) -o $@ ${LAPACK_LIBS}
 
 $(GSVD_TEST) : $(GSVD_TARGET)
 
@@ -32,19 +34,19 @@
 endif
 
 %.d:%.cc
-	$(MKOCTFILE) $(DEFINES) -M $<
+	$(MKOCTFILE) $(DEFINES) -M $< 
 
 %.o:%.cc
 %.o:%.cc %.d
-	$(MKOCTFILE) $(DEFINES) -c $<
+	$(MKOCTFILE) $(DEFINES) -c $< 
 
 %.o:%.f
-	$(MKOCTFILE) $(DEFINES) -c $<
+	$(MKOCTFILE) $(DEFINES) -c $< 
 %.oct:%.o
 	$(MKOCTFILE) $(DEFINES) $< -o $@
 
 %.oct: %.cc
-	mkoctfile $(DEFINES) $< -o $@
+	mkoctfile $(DEFINES) $< -o $@ ${LAPACK_LIBS}
 
 .phony: test
 test: $(GSVD_TEST)
