diff -Nurd -x'*~' mathgl-2.0.3.orig/src/data_new.cpp mathgl-2.0.3/src/data_new.cpp
--- mathgl-2.0.3.orig/src/data_new.cpp	2012-07-27 03:27:00.000000000 -0400
+++ mathgl-2.0.3/src/data_new.cpp	2014-10-26 02:20:19.000000000 -0400
@@ -24,6 +24,12 @@
 #include <gsl/gsl_fft_complex.h>
 #include <gsl/gsl_dht.h>
 #include <gsl/gsl_sf.h>
+
+// compensate for deprecated-API changes in gsl-1.16 based on
+// http://lists.opensuse.org/archive/opensuse-commit/2013-09/msg01121.html
+#define forward gsl_fft_forward
+#define backward gsl_fft_backward
+
 #endif
 //-----------------------------------------------------------------------------
 void mgl_data_envelop(HMDT d, char dir)
diff -Nurd -x'*~' mathgl-2.0.3.orig/src/pde.cpp mathgl-2.0.3/src/pde.cpp
--- mathgl-2.0.3.orig/src/pde.cpp	2012-07-27 03:27:00.000000000 -0400
+++ mathgl-2.0.3/src/pde.cpp	2014-10-26 02:22:15.000000000 -0400
@@ -23,6 +23,12 @@
 #define GAMMA	0.1
 #if MGL_HAVE_GSL
 #include <gsl/gsl_fft_complex.h>
+
+// compensate for deprecated-API changes in gsl-1.16 based on
+// http://lists.opensuse.org/archive/opensuse-commit/2013-09/msg01121.html
+#define forward gsl_fft_forward
+#define backward gsl_fft_backward
+
 #endif
 //-----------------------------------------------------------------------------
 struct mgl_pde_ham
