diff -ruNd matplotlib-1.4.3.orig/setup.cfg.template matplotlib-1.4.3/setup.cfg.template
--- matplotlib-1.4.3.orig/setup.cfg.template	2015-02-15 20:50:55.000000000 -0700
+++ matplotlib-1.4.3/setup.cfg.template	2015-03-10 17:45:52.000000000 +0100
@@ -57,18 +57,19 @@
 #           otherwise skip silently. This is the default
 #           behavior
 #
-#agg = auto
-#cairo = auto
-#gtk = auto
+agg = True
+cairo = True
+gtk = True
 #gtk3agg = auto
 #gtk3cairo = auto
-#gtkagg = auto
+gtkagg = True
 #macosx = auto
 #pyside = auto
-#qt4agg = auto
-#tkagg = auto
+qt4agg = True
+qt5agg = True
+tkagg = True
 #windowing = auto
-#wxagg = auto
+wxagg = True
 
 [rc_options]
 # User-configurable options
diff -ruNd matplotlib-1.3.1.orig/setup.py matplotlib-1.3.1/setup.py
--- matplotlib-1.3.1.orig/setup.py	2013-10-10 05:42:48.000000000 -0700
+++ matplotlib-1.3.1/setup.py	2013-11-29 10:48:34.000000000 -0800
@@ -210,6 +210,9 @@
     with open('lib/matplotlib/mpl-data/matplotlibrc', 'w') as fd:
         fd.write(template % {'backend': default_backend})
 
+    for mod in ext_modules:
+        mod.include_dirs.append('@PREFIX@/include')
+        mod.library_dirs.append('@PREFIX@/lib')
 
     # Build in verbose mode if requested
     if setupext.options['verbose']:
diff -ruNd matplotlib-1.4.2.orig/setupext.py matplotlib-1.4.2/setupext.py
--- matplotlib-1.4.2.orig/setupext.py	2013-10-10 05:42:48.000000000 -0700
+++ matplotlib-1.4.2/setupext.py	2015-01-29 10:58:07.000000000 -0800
@@ -147,8 +147,8 @@
 
     basedir_map = {
         'win32': ['win32_static', ],
-        'darwin': ['/usr/local/', '/usr', '/usr/X11',
-                   '/opt/X11', '/opt/local'],
+        'darwin': ['@PREFIX@/lib/freetype219', '@PREFIX@',
+                   '/usr', '/usr/X11', '/opt/X11'],
         'sunos5': [os.getenv('MPLIB_BASE') or '/usr/local', ],
         'gnu0': ['/usr'],
         'aix5': ['/usr/local'],
@@ -1053,14 +1053,14 @@ class Qhull(SetupPackage):
         self.__class__.found_external = True
         try:
             return self._check_for_pkg_config(
-                'qhull', 'qhull/qhull_a.h', min_version='2003.1')
+                'qhull', 'libqhull/qhull_a.h', min_version='2003.1')
         except CheckFailed as e:
             self.__class__.found_pkgconfig = False
             # Qhull may not be in the pkg-config system but may still be
             # present on this system, so check if the header files can be
             # found.
             include_dirs = [
-                os.path.join(x, 'qhull') for x in get_include_dirs()]
+                os.path.join(x, 'libqhull') for x in get_include_dirs()]
             if has_include_file(include_dirs, 'qhull_a.h'):
                 return 'Using system Qhull (version unknown, no pkg-config info)'
             else:
@@ -1543,6 +1543,17 @@
             # this config section lifted directly from Imaging - thanks to
             # the effbot!
 
+            tcl_inc = "@PREFIX@/include"
+            tk_inc = "@PREFIX@/include"
+            tcl_lib = "@PREFIX@/lib"
+            tk_lib = "@PREFIX@/lib"
+            tk_ver = ""
+            # Add final versions of directories and libraries to module lists
+            ext.include_dirs.extend([tcl_inc, tk_inc])
+            ext.library_dirs.extend([tcl_lib, tk_lib])
+            ext.libraries.extend(['tk' + tk_ver, 'tcl' + tk_ver])
+
+        elif False:
             # First test for a MacOSX/darwin framework install
             from os.path import join, exists
             framework_dirs = [
diff -ruNd matplotlib-1.4.2.orig/lib/matplotlib/font_manager.py matplotlib-1.4.2/lib/matplotlib/font_manager.py
--- matplotlib-1.4.2.orig/lib/matplotlib/font_manager.py	2013-10-10 05:42:48.000000000 -0700
+++ matplotlib-1.4.2/lib/matplotlib/font_manager.py	2015-02-03 20:58:07.000000000 -0800
@@ -138,6 +138,11 @@ OSXFontDirectories = [
     "/Library/Fonts/",
     "/Network/Library/Fonts/",
     "/System/Library/Fonts/",
+    # new XQuartz font location
+    "/opt/X11/share/fonts",
+    "/opt/X11/share/fonts/TTF",
+    # fonts installed via Fink
+    "@PREFIX@/lib/X11/fonts",
     # fonts installed via MacPorts
     "/opt/local/share/fonts"
     ""
diff -ruNd matplotlib-1.4.2.orig/src/qhull_wrap.c matplotlib-1.4.2/src/qhull_wrap.c
--- matplotlib-1.4.2.orig/src/qhull_wrap.c	2014-10-21 01:14:48.000000000 -0700
+++ matplotlib-1.4.2/src/qhull_wrap.c	2015-02-03 10:58:07.000000000 -0800
@@ -7,7 +7,7 @@
  */
 #include "Python.h"
 #include "numpy/noprefix.h"
-#include "qhull/qhull_a.h"
+#include "libqhull/qhull_a.h"
 #include <stdio.h>
 
 
--- setup.cfg.template	2015-02-02 22:21:21.000000000 +0100
+++ setup.cfg	2015-03-11 20:05:20.000000000 +0100
