--- Numeric/setup.py.orig	Mon Aug 15 23:20:24 2005
+++ Numeric/setup.py	Tue Oct 25 15:55:40 2005
@@ -83,17 +83,17 @@
     extra_compile_args = []
     extra_link_args = []
     include_dirs = []
-    use_system_lapack = 0
+    use_system_lapack = 1
     use_system_blas = 0
     lapack_library_dirs = []
     lapack_libraries = []
     lapack_extra_link_args = []
-    use_dotblas = 0
+    use_dotblas = 1
     dotblas_include_dirs = []
     dotblas_cblas_header = '<cblas.h>'
     dotblas_library_dirs = []
     dotblas_extra_link_args = []
-execfile('customize.py', customize.__dict__)
+#execfile('customize.py', customize.__dict__)
 
 # Get all version numbers
 execfile(path('Lib/numeric_version.py'))
@@ -140,10 +140,10 @@
 
 lapack_source = [path('Src/lapack_litemodule.c')]
 lapack_link_args = customize.lapack_extra_link_args
-if customize.use_system_lapack:
-    pass
-elif have_veclib:
+if have_veclib:
     lapack_link_args.extend(['-framework', 'vecLib'])
+elif customize.use_system_lapack:
+    pass
 else:
     lapack_source.extend([path('Src/f2c_lite.c'),
                           path('Src/zlapack_lite.c'),
@@ -159,7 +159,13 @@
                        extra_link_args=lapack_link_args)
 
 dotblas_source = [path('Packages/dotblas/dotblas/_dotblas.c')]
-if customize.use_dotblas:
+if have_veclib:
+    dotblas_ext = extension('_dotblas', dotblas_source,
+                            include_dirs=[os.path.join(VECLIB_PATH, 'Headers')],
+                            define_macros=[('CBLAS_HEADER', '<cblas.h>')],
+                            extra_link_args=['-framework', 'vecLib'],
+                           )
+elif customize.use_dotblas:
     dotblas_ext = extension('_dotblas', dotblas_source,
                             include_dirs=customize.dotblas_include_dirs,
                             library_dirs=customize.dotblas_library_dirs,
@@ -167,12 +173,6 @@
                             define_macros=[('CBLAS_HEADER',
                                             customize.dotblas_cblas_header)],
                             extra_link_args=customize.dotblas_extra_link_args,
-                           )
-elif have_veclib:
-    dotblas_ext = extension('_dotblas', dotblas_source,
-                            include_dirs=[os.path.join(VECLIB_PATH, 'Headers')],
-                            define_macros=[('CBLAS_HEADER', '<cblas.h>')],
-                            extra_link_args=['-framework', 'vecLib'],
                            )
 else:
     dotblas_ext = None
diff -ruN Numeric-24.2/Packages/RNG/Src/ranf.c Numeric-24.2-patched/Packages/RNG/Src/ranf.c
--- Numeric-24.2/Packages/RNG/Src/ranf.c	2005-04-03 05:23:06.000000000 -0700
+++ Numeric-24.2-patched/Packages/RNG/Src/ranf.c	2007-08-06 14:05:16.000000000 -0700
@@ -150,7 +150,7 @@
 	struct timeval tv;
 	struct timezone tz;
 #if !defined(__sgi)
-	int gettimeofday(struct timeval *, struct timezone *);
+#include <sys/time.h>
 #endif
 
 	(void)gettimeofday(&tv,&tz);
