diff -Nurd -x'*~' plotutils-2.6.orig/libplot/Makefile.am plotutils-2.6/libplot/Makefile.am
--- plotutils-2.6.orig/libplot/Makefile.am	2009-07-28 22:16:02.000000000 -0400
+++ plotutils-2.6/libplot/Makefile.am	2011-07-10 10:43:38.000000000 -0400
@@ -6,7 +6,7 @@
 
 libplot_la_LDFLAGS = -version-info 4:4:2
 
-INCLUDES = $(X_CFLAGS) -I$(srcdir)/../include -DLIBPLOT
+INCLUDES = -I$(srcdir)/../include $(CPPFLAGS) $(X_CFLAGS) -DLIBPLOT
 
 MISRC = mi_alloc.c mi_api.c mi_arc.c mi_canvas.c mi_fllarc.c mi_fllrct.c \
 mi_fplycon.c mi_gc.c mi_ply.c mi_plycon.c mi_plygen.c mi_plypnt.c	 \
diff -Nurd -x'*~' plotutils-2.6.orig/libplotter/Makefile.am plotutils-2.6/libplotter/Makefile.am
--- plotutils-2.6.orig/libplotter/Makefile.am	2009-07-28 22:16:18.000000000 -0400
+++ plotutils-2.6/libplotter/Makefile.am	2011-07-10 10:48:44.000000000 -0400
@@ -4,7 +4,7 @@
 
 libplotter_la_LDFLAGS = -version-info 4:4:2
 
-INCLUDES = $(X_CFLAGS) -I$(srcdir)/../include -DLIBPLOT -DLIBPLOTTER
+INCLUDES = -I$(srcdir)/../include $(CPPFLAGS) $(X_CFLAGS) -DLIBPLOT -DLIBPLOTTER
 
 MISRC = mi_alloc.cc mi_api.cc mi_arc.cc mi_canvas.cc mi_fllarc.cc	\
 mi_fllrct.cc mi_fplycon.cc mi_gc.cc mi_ply.cc mi_plycon.cc mi_plygen.cc	\
