diff -Naur --exclude='*~' LADR-2009-11A/Makefile LADR-2009-11A.fink/Makefile
--- LADR-2009-11A/Makefile	2007-10-22 22:33:12.000000000 +0100
+++ LADR-2009-11A.fink/Makefile	2010-03-08 15:33:57.000000000 +0000
@@ -6,7 +6,7 @@
 	cd mace4.src    && $(MAKE) all
 	cd provers.src  && $(MAKE) all
 	cd apps.src     && $(MAKE) all
-	/bin/cp -p utilities/* bin
+	find utilities -type f -exec cp {} @PREFIX@/bin ";";
 	@echo ""
 	@echo "**** Now try 'make test1'. ****"
 	@echo ""
@@ -33,7 +33,7 @@
 	@echo ""
 	@echo "*** If you see 5 interpretations, the apps are probably okay. ***"
 	@echo ""
-	@echo "*** All of the programs are in ./bin, and they can be copied anywhere you like. ***"
+	@echo "*** All of the programs are in @PREXI@/bin, and they can be copied anywhere you like. ***"
 	@echo ""
 
 clean:	
diff -Naur --exclude='*~' LADR-2009-11A/provers.src/ladr_to_tptp.c LADR-2009-11A.fink/provers.src/ladr_to_tptp.c
--- LADR-2009-11A/provers.src/ladr_to_tptp.c	2009-06-29 18:51:25.000000000 +0100
+++ LADR-2009-11A.fink/provers.src/ladr_to_tptp.c	2010-03-08 15:20:34.000000000 +0000
@@ -40,7 +40,7 @@
 
   /* This is similar to the way Prover9 initializes and reads its input. */
 
-  input = std_prover_init_and_input(0, NULL,
+  input = std_prover_init_and_input(1, argv,  // ignore command-line args
 			    FALSE,            // clausify formulas
 			    FALSE,            // echo input to stdout
 			    IGNORE_UNKNOWN);  // for unknown flags/parms
