diff -Naur '--exclude=*~' LADR-2009-11A/Makefile LADR-2009-11A.fink/Makefile
--- LADR-2009-11A/Makefile	2007-10-22 23:33:12.000000000 +0200
+++ LADR-2009-11A.fink/Makefile	2012-10-10 16:38:54.000000000 +0200
@@ -2,11 +2,11 @@
 	@echo See README.make
 
 all:
-	cd ladr         && $(MAKE) lib
-	cd mace4.src    && $(MAKE) all
-	cd provers.src  && $(MAKE) all
-	cd apps.src     && $(MAKE) all
-	/bin/cp -p utilities/* bin
+	$(MAKE) -C ladr lib
+	$(MAKE) -C mace4.src all
+	$(MAKE) -C provers.src all
+	$(MAKE) -C apps.src all
+	find utilities -type f -exec cp {} @PREFIX@/bin ';'
 	@echo ""
 	@echo "**** Now try 'make test1'. ****"
 	@echo ""
@@ -33,10 +33,8 @@
 	@echo ""
 	@echo "*** If you see 5 interpretations, the apps are probably okay. ***"
 	@echo ""
-	@echo "*** All of the programs are in ./bin, and they can be copied anywhere you like. ***"
-	@echo ""
 
-clean:	
+clean:
 	cd ladr             && $(MAKE) realclean
 	cd apps.src         && $(MAKE) realclean
 	cd mace4.src        && $(MAKE) realclean
diff -Naur '--exclude=*~' LADR-2009-11A/mace4.src/Makefile LADR-2009-11A.fink/mace4.src/Makefile
--- LADR-2009-11A/mace4.src/Makefile	2009-04-17 16:46:46.000000000 +0200
+++ LADR-2009-11A.fink/mace4.src/Makefile	2012-10-10 16:46:35.000000000 +0200
@@ -1,5 +1,7 @@
 # LADR/mace4.src
 
+.PHONY: all lib install clean realclean
+
 CC = gcc
 
 # XFLAGS can be specified on the command line (see XFLAGS below)
@@ -43,5 +45,5 @@
 realclean:
 	/bin/rm -f *.o *.a mace4
 
-install:
-	/bin/mv mace4 ../bin
+install: mace4
+	cp mace4 ../bin
diff -Naur '--exclude=*~' LADR-2009-11A/provers.src/Makefile LADR-2009-11A.fink/provers.src/Makefile
--- LADR-2009-11A/provers.src/Makefile	2009-10-28 15:22:15.000000000 +0100
+++ LADR-2009-11A.fink/provers.src/Makefile	2012-10-10 16:35:39.000000000 +0200
@@ -1,5 +1,7 @@
 # LADR/prover9.src
 
+.PHONY: all install clean libs
+
 CC = gcc
 
 # XFLAGS can be specified on the command line (see XFLAGS below)
@@ -33,14 +35,14 @@
 libs: ladr libmace4
 
 ladr libladr:
-	cd ../ladr && $(MAKE) libladr
+	$(MAKE) -C ../ladr libladr
 	$(MAKE) clean
 
 libmace libmace4:
-	cd ../mace4.src && $(MAKE) libmace4
+	$(MAKE) -C ../mace4.src libmace4
 	$(MAKE) clean
 
-install:
+install: $(PROGRAMS)
 	/bin/cp -p $(PROGRAMS) ../bin
 
 clean:
diff -Naur '--exclude=*~' LADR-2009-11A/provers.src/ladr_to_tptp.c LADR-2009-11A.fink/provers.src/ladr_to_tptp.c
--- LADR-2009-11A/provers.src/ladr_to_tptp.c	2009-06-29 19:51:25.000000000 +0200
+++ LADR-2009-11A.fink/provers.src/ladr_to_tptp.c	2012-10-10 15:54:53.000000000 +0200
@@ -40,7 +40,7 @@
 
   /* This is similar to the way Prover9 initializes and reads its input. */
 
-  input = std_prover_init_and_input(0, NULL,
+  input = std_prover_init_and_input(1, argv,
 			    FALSE,            // clausify formulas
 			    FALSE,            // echo input to stdout
 			    IGNORE_UNKNOWN);  // for unknown flags/parms
@@ -89,10 +89,10 @@
     printf("\n");
     for (p = all; p; p = p->next) {
       Term t = p->v;
-      ARG(t,2) = replace_bad_tptp_syms_form(ARG(t,2), 
+      ARG(t,2) = replace_bad_tptp_syms_form(ARG(t,2),
 				  str_ident(sn_to_str(SYMNUM(t)),"cnf"),
 				  map);
-						      
+
     }
   }
 
