--- pymol-1.5/setup.py.orig	2012-02-18 13:25:51.000000000 -0500
+++ pymol-1.5/setup.py	2012-02-18 13:29:45.000000000 -0500
@@ -43,7 +43,7 @@
     ext_link_args=[]
 #============================================================================
 elif sys.platform=='darwin':
-    using_fink = "'/sw/" in str(sys.path)
+    using_fink = "'@FINKPREFIX@/" in str(sys.path)
     if using_fink:
         # under Fink, with the following packages installed:
         #
@@ -84,9 +84,9 @@
         inc_dirs=["ov/src",
                   "layer0","layer1","layer2",
                   "layer3","layer4","layer5", 
-                  "/sw/include/freetype2/freetype",
-                  "/sw/include/freetype2",
-                  "/sw/include",
+                  "@FINKPREFIX@/include/freetype2/freetype",
+                  "@FINKPREFIX@/include/freetype2",
+                  "@FINKPREFIX@/include",
                   "/usr/X11/include",
 		  "modules/cealign/src",
 		  "modules/cealign/src/tnt",
@@ -113,12 +113,12 @@
                     ]
         ext_comp_args=[]
         ext_link_args=[
-                       "-L/sw/lib", "-lpng",
+                       "-L@FINKPREFIX@/lib", "-lpng",
                        "/usr/X11/lib/libGL.dylib",
                        "/usr/X11/lib/libGLU.dylib",
                        "-lfreeglut",
                        "-lglew",
-                       "-L/sw/lib/freetype219/lib", "-lfreetype"
+                       "-L@FINKPREFIX@/lib/freetype219/lib", "-lfreetype"
                         ]
     else:
         # Not using Fink -- building as if we are on Linux/X11 with
--- /dev/null	2012-02-14 20:24:44.000000000 -0500
+++ pymol-1.5.0.1/pymol.cmd	2012-02-14 19:31:08.000000000 -0500
@@ -0,0 +1,16 @@
+#!/bin/sh
+#
+# PyMOL startup script
+#
+# ==============================================================
+# Set PYMOL_PATH to point at the correct location on your system
+#
+PYMOL_PATH=@FINKPREFIX@/lib/python2.X/site-packages/pymol
+export PYMOL_PATH
+#
+# ==============================================================
+# (no changes usually required below here)
+#
+# python modules
+#
+exec @FINKPREFIX@/bin/python2.X  @FINKPREFIX@/lib/python2.X/site-packages/pymol/__init__.py -d "_ set stereo_double_pump_mono,quiet=1" "$@"
--- pymol-1.5/modules/pmg_tk/startup/apbs_tools.py.orig	2012-02-18 14:50:35.000000000 -0500
+++ pymol-1.5/modules/pmg_tk/startup/apbs_tools.py	2012-02-18 14:51:33.000000000 -0500
@@ -141,7 +141,7 @@
 # APBS_BINARY_LOCATION = '/opt/bin/apbs'
 #
 APBS_BINARY_LOCATION = None
-APBS_PSIZE_LOCATION = None
+APBS_PSIZE_LOCATION = '@FINKPREFIX@/share/apbs/tools/manip'
 APBS_PDB2PQR_LOCATION = None
 TEMPORARY_FILE_DIR = None
 
--- pymol-1.5/modules/pmg_tk/PMGApp.py.orig	2012-02-18 13:31:34.000000000 -0500
+++ pymol-1.5/modules/pmg_tk/PMGApp.py	2012-02-18 13:32:15.000000000 -0500
@@ -219,7 +219,7 @@
 
 
     def my_show(self,win,center=1):
-        if sys.platform!='linux2':
+        if (sys.platform!='linux2') and (sys.platform!='darwin'):
             win.show()
         else: # autocenter, deiconify, and run mainloop
             # this is a workaround for a bug in the
@@ -237,7 +237,7 @@
             win.deiconify()
 
     def my_withdraw(self,win):
-        if sys.platform!='linux2':
+        if (sys.platform!='linux2') or (sys.platform!='darwin'):
             win.withdraw()
         else: 
             win.destroy()
--- pymol-1.5/modules/pmg_tk/skins/normal/__init__.py.orig	2012-02-18 13:31:47.000000000 -0500
+++ pymol-1.5/modules/pmg_tk/skins/normal/__init__.py	2012-02-18 13:32:15.000000000 -0500
@@ -250,7 +250,7 @@
         self.buttonArea.destroy()
 
     def my_show(self,win,center=1):
-        if sys.platform!='linux2':
+        if (sys.platform!='linux2') and (sys.platform!='darwin'):
             win.show()
         else: # autocenter, deiconify, and run mainloop
             # this is a workaround for a bug in the
@@ -269,13 +269,13 @@
 #         win.show()
             
     def my_withdraw(self,win):
-        if sys.platform!='linux2':
+        if (sys.platform!='linux2') or (sys.platform!='darwin'):
             win.withdraw()
         else: 
             win.destroy()
 
     def my_activate(self,win,center=1,focus=None):
-        if sys.platform!='linux2':
+        if (sys.platform!='linux2') and (sys.platform!='darwin'):
             win.activate()
         else: # autocenter, deiconify, and run mainloop
             # this is a workaround for a bug in the
@@ -296,7 +296,7 @@
             win.mainloop()
             
     def my_deactivate(self,win):
-        if sys.platform!='linux2':
+        if (sys.platform!='linux2') or (sys.platform!='darwin'):
             win.deactivate()
         else: # autocenter, deiconify, and run mainloop
             win.destroy()
