--- pymol-1.7.4.0/layer0/Field.cpp.orig	2014-12-11 10:46:41.000000000 -0500
+++ pymol-1.7.4.0/layer0/Field.cpp	2014-12-11 11:52:23.000000000 -0500
@@ -73,7 +73,7 @@
 
   if(copy) {
     if((result = PyArray_SimpleNew(field->n_dim, dims, typenum)))
-      memcpy(PyArray_DATA(result), field->data, field->size);
+      memcpy(PyArray_DATA((PyArrayObject *)result), field->data, field->size);
   } else {
     result = PyArray_SimpleNewFromData(field->n_dim, dims, typenum, field->data);
   }
--- pymol-1.7.4.0/layer2/CoordSet.cpp.orig	2014-12-11 11:59:47.000000000 -0500
+++ pymol-1.7.4.0/layer2/CoordSet.cpp	2014-12-11 12:00:19.000000000 -0500
@@ -198,7 +198,7 @@
 
   if(copy) {
     if((result = PyArray_SimpleNew(2, dims, typenum)))
-      memcpy(PyArray_DATA(result), cs->Coord, cs->NIndex * 3 * base_size);
+      memcpy(PyArray_DATA((PyArrayObject *)result), cs->Coord, cs->NIndex * 3 * base_size);
   } else {
     result = PyArray_SimpleNewFromData(2, dims, typenum, cs->Coord);
   }
--- pymol-1.7.4.0/layer3/Selector.cpp.orig	2014-12-11 12:06:08.000000000 -0500
+++ pymol-1.7.4.0/layer3/Selector.cpp	2014-12-11 12:16:13.000000000 -0500
@@ -6608,7 +6608,7 @@
   }
 
   result = PyArray_SimpleNew(2, dims, typenum);
-  dataptr = (float*) PyArray_DATA(result);
+  dataptr = (float*) PyArray_DATA((PyArrayObject *)result);
 
   for(i = 0, iter.reset(); iter.next(); i++) {
     v_ptr = iter.getCoord();
@@ -6677,11 +6677,11 @@
   import_array1(false);
 
   if(PyArray_Check(coords)) {
-    if(PyArray_NDIM(coords) != 2 || PyArray_DIM(coords, 1) != 3) {
+    if(PyArray_NDIM((const PyArrayObject *)coords) != 2 || PyArray_DIM((const PyArrayObject *)coords, 1) != 3) {
       ErrMessage(G, "LoadCoords", "numpy array shape mismatch");
       return false;
     }
-    itemsize = PyArray_ITEMSIZE(coords);
+    itemsize = PyArray_ITEMSIZE((const PyArrayObject *)coords);
     switch(itemsize) {
       case sizeof(double):
       case sizeof(float):
@@ -6700,7 +6700,7 @@
       // fast implementation for numpy arrays only
 #ifdef _PYMOL_NUMPY
       for(b = 0; b < 3; b++) {
-        ptr = PyArray_GETPTR2(coords, a, b);
+        ptr = PyArray_GETPTR2((PyArrayObject *)coords, a, b);
 
         switch(itemsize) {
           case sizeof(double):
