diff -ruN qgis-2.6.1-orig/python/console/console.py qgis-2.6.1-python/python/console/console.py
--- qgis-2.6.1-orig/python/console/console.py	2015-02-01 15:14:37.000000000 +0900
+++ qgis-2.6.1-python/python/console/console.py	2015-02-01 15:41:05.000000000 +0900
@@ -21,7 +21,6 @@
 
 from PyQt4.QtCore import *
 from PyQt4.QtGui import *
-from PyQt4 import pyqtconfig
 from qgis.utils import iface
 from console_sci import ShellScintilla
 from console_output import ShellOutputScintilla
@@ -472,10 +471,7 @@
         self.lineEditFind = QgsFilterLineEdit()
         placeHolderTxt = QCoreApplication.translate("PythonConsole", "Enter text to find...")
 
-        if pyqtconfig.Configuration().qt_version >= 0x40700:
-          self.lineEditFind.setPlaceholderText(placeHolderTxt)
-        else:
-          self.lineEditFind.setToolTip(placeHolderTxt)
+        self.lineEditFind.setPlaceholderText(placeHolderTxt)
         self.findNextButton = QToolButton()
         self.findNextButton.setEnabled(False)
         toolTipfindNext = QCoreApplication.translate("PythonConsole", "Find Next")
