diff -ruN refmac-orig/covmat.h refmac/covmat.h
--- refmac-orig/covmat.h	2011-11-30 17:03:44.000000000 -0800
+++ refmac/covmat.h	2013-08-14 15:13:23.000000000 -0700
@@ -44,7 +44,7 @@
 	~covar_matrix();
     void Make_matrix();						// computes re and im from sigmas and D's
 	void Print();
-	void SetZeroRows(int,int,int,int,int,int);	// specifies the order numbers of rows(=columns) for which the covariances should be 0.
+	void SetZeroRows(int = -1,int = -1,int = -1,int = -1,int = -1,int = -1);	// specifies the order numbers of rows(=columns) for which the covariances should be 0.
 	int CheckZeroRow(int row);				// returns 1 if row has been set to be covariance zeroed, 0 otherwise
 	int GetNumZeroRows();					// returns the number of rows with covariaces zeroed
 
@@ -2106,7 +2106,7 @@
 
 // defines zero rows. If the first argument is negative, resets all no non-zero.
 template <typename realnum>
-void covar_matrix<realnum>::SetZeroRows( int r1=-1, int r2=-1, int r3=-1, int r4=-1, int r5=-1, int r6=-1 )
+void covar_matrix<realnum>::SetZeroRows( int r1, int r2, int r3, int r4, int r5, int r6 )
 {
   int r[6] = {r1,r2,r3,r4,r5,r6};
   int i=0;
diff -ruN refmac-orig/makefile refmac/makefile
--- refmac-orig/makefile	2011-07-19 03:41:56.000000000 -0700
+++ refmac/makefile	2013-08-14 15:15:22.000000000 -0700
@@ -1,19 +1,29 @@
-VERSION = gfortran
-FC      = gfortran
-CC      = gcc
-CPP     = g++
-FOPTIM  = -O2 
-COPTIM  = -O2 
-XFFLAGS = -fno-second-underscore -static
-
-XCFLAGS = -DGFORTRAN -DPROTOTYPE
-FFLAGS = $(FOPTIM) $(XFFLAGS) 
-CFLAGS = $(COPTIM) $(XCFLAGS)
+VERSION = 5 
+FC      = @FINKPREFIX@/lib/gcc4.8/bin/gfortran
+CC      = gcc
+CXX     = g++
+FOPTIM  = -O3 
+COPTIM  = -O 
+XFFLAGS = -m64 -fno-second-underscore 
+ 
+
+XCFLAGS = -DGFORTRAN -DPROTOTYPE 
+FFLAGS = $(FOPTIM) $(XFFLAGS)  
+ 
+
+CFLAGS = $(COPTIM) $(XCFLAGS) 
 CPPFLAGS = $(CFLAGS)
 
+CLIB = @FINKPREFIX@/lib
+
 LLIBCIF = 
-LLIBCCP = $(CLIB)/libccp4f.a $(CLIB)/libccp4c.a 
-LLIBLAPACK = $(CLIB)/liblapack.a $(CLIB)/libblas.a
-LLIBOTHERS =  -lgfortran -lgfortranbegin -lstdc++ 
 
+BLAS_DIR = /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A
+
+LLIBCCP = /usr/lib/libSystem.B.dylib  ${BLAS_DIR}/libBLAS.dylib  ${BLAS_DIR}/libLAPACK.dylib  
+
+LLIBLAPACK =  
+
+LLIBOTHERS =  -L/usr/lib -L${CLIB} -L@FINKPREFIX@/lib/gcc4.8/lib -lgfortran  ${CLIB}/libccp4c.dylib ${CLIB}/libccp4f.dylib  -lstdc++  /usr/lib/libSystem.B.dylib  -lSystem -lm -lc++
 include makefile_refmac.inc1
+
diff -ruN refmac-orig/makefile refmac/makefile
--- refmac-orig/bridge.cpp	2014-05-15 19:57:55.000000000 -0400
+++ refmac/bridge.cpp	2014-05-15 20:01:33.000000000 -0400
@@ -1711,7 +1711,7 @@
   if (*ID==2) lF=&lF3;
   if (*ID==3) lF=&lF4;
 
-  freal DPAR[6] = { D_PAR[0], D_PAR[1] };
+  freal DPAR[7] = { D_PAR[0], D_PAR[1] };
   if ( lF->cov->N_part > 2 ) 	DPAR[2]=D_PAR[2];
   if ( lF->cov->N_part > 3 ) 	DPAR[3]=D_PAR[3];
   if ( lF->cov->N_part > 4 ) 	DPAR[4]=D_PAR[4];

