diff -uNr relax-1.3.14/relax relax-1.3.14.fink/relax
--- relax-1.3.14/relax	2012-03-19 12:05:34.000000000 -0400
+++ relax-1.3.14.fink/relax	2012-04-07 15:02:12.000000000 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python2.X
 
 # Import the relax module.
 import relax
diff -uNr relax-1.3.14/sconstruct relax-1.3.14.fink/sconstruct
--- relax-1.3.14/sconstruct	2012-03-19 12:05:34.000000000 -0400
+++ relax-1.3.14.fink/sconstruct	2012-04-07 15:02:12.000000000 -0400
@@ -99,7 +99,7 @@
     SYS = SYSTEM
 
     # Mac OS X installation path.
-    INSTALL_PATH = sys.prefix + sep + 'local'
+    INSTALL_PATH = 'FINK_INSTALL_DIR' + sep + 'lib'
 
 
 # All other operating systems.
@@ -116,10 +116,10 @@
 ###############
 
 # Relax installation directory.
-RELAX_PATH = INSTALL_PATH + sep + 'relax'
+RELAX_PATH = INSTALL_PATH + sep + 'relax-py2X'
 
 # Installation path for binaries.
-BIN_PATH = INSTALL_PATH + sep + 'bin'
+BIN_PATH = 'FINK_INSTALL_DIR' + sep + 'bin'
 
 # Symbolic link installation path.
 SYMLINK = BIN_PATH + sep + 'relax'
@@ -497,11 +497,11 @@
                  'relax_fit.c']
 
         # Construct the python include path (for Python.h).
-        py_include_minpath = sys.prefix + path.sep + 'include'
+        py_include_minpath = 'FINK_PREFIX' + path.sep + 'include'
         py_include_fullpath = py_include_minpath + path.sep + 'python' + `sys.version_info[0]` + '.' + `sys.version_info[1]`
 
         # Construct the python bin path.
-        py_bin_minpath = sys.prefix + path.sep + 'bin'
+        py_bin_minpath = 'FINK_PREFIX' + path.sep + 'bin'
         py_bin_fullpath = py_bin_minpath + path.sep + 'python' + `sys.version_info[0]` + '.' + `sys.version_info[1]`
 
         # Get the numpy include path (for numpy/arrayobject.h).
--- relax-2.0.0/scons/install.py.orig	2012-06-11 05:50:51.000000000 -0400
+++ relax-2.0.0/scons/install.py	2012-06-14 08:49:48.000000000 -0400
@@ -112,11 +112,11 @@
     ###############
 
     # Run relax to create the *.pyc files.
-    print("\nCreating the byte-compiled *.pyc files.")
-    python_path = sys.prefix + path.sep + 'bin' + path.sep + 'python' + `sys.version_info[0]` + '.' + `sys.version_info[1]`
-    cmd = "cd %s; %s -m compileall . ; %s -O -m compileall ." % (env['RELAX_PATH'], python_path, python_path)
-    print(cmd)
-    system(cmd)
+    #print("\nCreating the byte-compiled *.pyc files.")
+    #python_path = sys.prefix + path.sep + 'bin' + path.sep + 'python' + `sys.version_info[0]` + '.' + `sys.version_info[1]`
+    #cmd = "cd %s; %s -m compileall . ; %s -O -m compileall ." % (env['RELAX_PATH'], python_path, python_path)
+    #print(cmd)
+    #system(cmd)
 
     # Final printout.
     print("\n\n\n")
--- trunk/dep_check.py	2012/10/17 16:51:49	17910
+++ trunk/dep_check.py	2012/10/17 20:43:22	17911
@@ -217,12 +217,19 @@
 # ctypes module.
 try:
     import ctypes
-    from ctypes import Structure
     ctypes_module = True
 except ImportError:
     message = sys.exc_info()[1]
     ctypes_module = False
     ctypes_module_message = message.args[0]
+try:
+    from ctypes import Structure
+    ctypes_structure_module = True
+except ImportError:
+    message = sys.exc_info()[1]
+    ctypes_structure_module = False
+    ctypes_structure_module_message = message.args[0]
+
 
 
 
--- trunk/info.py	2012/10/17 16:51:49	17910
+++ trunk/info.py	2012/10/17 20:43:22	17911
@@ -28,11 +28,13 @@
 # Python module imports.
 if dep_check.ctypes_module:
     import ctypes
-    from ctypes import Structure
     if hasattr(ctypes, 'windll'):
         import ctypes.wintypes
 else:
     ctypes = None
+if dep_check.ctypes_structure_module:
+    from ctypes import Structure
+else:
     Structure = object
 import numpy
 from os import environ, waitpid
--- trunk/dep_check.py	2012/10/17 20:43:22	17911
+++ trunk/dep_check.py	2012/10/17 20:53:41	17912
@@ -142,7 +142,7 @@
 try:
     import scipy
     scipy_module = True
-except ImportError:
+except:
     scipy_module = False
 
 # Numeric python package check.
--- relax-2.1.2/dep_check.py.orig	2012-10-19 15:24:10.000000000 -0400
+++ relax-2.1.2/dep_check.py	2012-10-19 15:24:31.000000000 -0400
@@ -157,7 +157,7 @@
     from Scientific.Visualization import VMD    # This requires Numeric to be installed (at least in Scientific 2.7.8).
     del VMD
     vmd_module = True
-except ImportError:
+except:
     vmd_module = False
 
 # mpi4py.
