diff -ruN rstudio-0.98.653-orig/CMakeGlobals.txt rstudio-0.98.653/CMakeGlobals.txt
--- rstudio-0.98.653-orig/CMakeGlobals.txt	2014-02-20 12:20:13.000000000 +0900
+++ rstudio-0.98.653/CMakeGlobals.txt	2014-02-20 12:26:48.000000000 +0900
@@ -15,18 +15,18 @@
 
 # version info
 if ("$ENV{RSTUDIO_VERSION_MAJOR}" STREQUAL "")
-  set(CPACK_PACKAGE_VERSION_MAJOR "99")
+  set(CPACK_PACKAGE_VERSION_MAJOR "0")
   set(RSTUDIO_UNVERSIONED_BUILD TRUE)
 else()
   set(CPACK_PACKAGE_VERSION_MAJOR $ENV{RSTUDIO_VERSION_MAJOR})
 endif()
 if ("$ENV{RSTUDIO_VERSION_MINOR}" STREQUAL "")
-  set(CPACK_PACKAGE_VERSION_MINOR "9")
+  set(CPACK_PACKAGE_VERSION_MINOR "98")
 else()
   set(CPACK_PACKAGE_VERSION_MINOR $ENV{RSTUDIO_VERSION_MINOR})
 endif()
 if ("$ENV{RSTUDIO_VERSION_PATCH}" STREQUAL "")
-  set(CPACK_PACKAGE_VERSION_PATCH "9")
+  set(CPACK_PACKAGE_VERSION_PATCH "653")
 else()
   set(CPACK_PACKAGE_VERSION_PATCH $ENV{RSTUDIO_VERSION_PATCH})
 endif()
diff -ruN rstudio-0.98.653-orig/cmake/modules/FindLibR.cmake rstudio-0.98.653/cmake/modules/FindLibR.cmake
--- rstudio-0.98.653-orig/cmake/modules/FindLibR.cmake	2014-02-20 12:20:13.000000000 +0900
+++ rstudio-0.98.653/cmake/modules/FindLibR.cmake	2014-02-20 13:51:14.000000000 +0900
@@ -22,10 +22,10 @@
 
    find_library(LIBR_LIBRARIES R)
    if(LIBR_LIBRARIES)
-      set(LIBR_HOME "${LIBR_LIBRARIES}/Resources" CACHE PATH "R home directory")
-      set(LIBR_INCLUDE_DIRS "${LIBR_HOME}/include" CACHE PATH "R include directory")
+      set(LIBR_HOME "@PREFIX@/Library/Frameworks/R.framework/Resources" CACHE PATH "R home directory")
+      set(LIBR_INCLUDE_DIRS "@PREFIX@/Library/Frameworks/R.framework/Versions/3.0/Headers" CACHE PATH "R include directory")
       set(LIBR_DOC_DIR "${LIBR_HOME}/doc" CACHE PATH "R doc directory")
-      set(LIBR_EXECUTABLE "${LIBR_HOME}/R" CACHE PATH "R executable")
+      set(LIBR_EXECUTABLE "@PREFIX@/bin/R" CACHE PATH "R executable")
    endif()
 
 # detection for UNIX & Win32
diff -ruN rstudio-0.98.653-orig/src/cpp/core/r_util/REnvironmentPosix.cpp rstudio-0.98.653/src/cpp/core/r_util/REnvironmentPosix.cpp
--- rstudio-0.98.653-orig/src/cpp/core/r_util/REnvironmentPosix.cpp	2014-02-20 12:20:13.000000000 +0900
+++ rstudio-0.98.653/src/cpp/core/r_util/REnvironmentPosix.cpp	2014-02-20 12:30:49.000000000 +0900
@@ -84,9 +84,7 @@
 {
    // define potential paths
    std::vector<std::string> rScriptPaths;
-   rScriptPaths.push_back("/usr/bin/R");
-   rScriptPaths.push_back("/usr/local/bin/R");
-   rScriptPaths.push_back("/opt/local/bin/R");
+   rScriptPaths.push_back("@PREFIX@/bin/R");
    return scanForRScript(rScriptPaths, pErrMsg);
 }
 
@@ -146,7 +144,7 @@
                                     std::string* pErrMsg)
 {
    // home path
-   *pHomePath = FilePath("/Library/Frameworks/R.framework/Resources");
+   *pHomePath = FilePath("@PREFIX@/Library/Frameworks/R.framework/Resources");
 
    // lib path
    std::string rLibPath;
diff -ruN rstudio-0.98.653-orig/src/cpp/desktop/CMakeLists.txt rstudio-0.98.653/src/cpp/desktop/CMakeLists.txt
--- rstudio-0.98.653-orig/src/cpp/desktop/CMakeLists.txt	2014-02-20 12:20:13.000000000 +0900
+++ rstudio-0.98.653/src/cpp/desktop/CMakeLists.txt	2014-02-20 13:25:11.000000000 +0900
@@ -354,7 +354,7 @@
           file(REMOVE \"\${CMAKE_INSTALL_PREFIX}/RStudio.app/Contents/MacOS/rsession\")
           file(REMOVE \"\${CMAKE_INSTALL_PREFIX}/RStudio.app/Contents/MacOS/rserver\")
           file(REMOVE \"\${CMAKE_INSTALL_PREFIX}/RStudio.app/Contents/MacOS/rserver-pam\")
-          fixup_bundle(\"${APPS}\" \"\${QTPLUGINS}\" \"${DIRS}\")
+          # fixup_bundle(\"${APPS}\" \"\${QTPLUGINS}\" \"${DIRS}\")
           ")
 
    # copy qt shared objects directly to bin on unix (fixed up rpath above)
