diff -uNr rstudio-0.98.1103.orig/CMakeGlobals.txt rstudio-0.98.1103/CMakeGlobals.txt
--- rstudio-0.98.1103.orig/CMakeGlobals.txt	2015-03-06 17:12:51.000000000 -0500
+++ rstudio-0.98.1103/CMakeGlobals.txt	2015-04-17 23:25:56.000000000 -0400
@@ -15,18 +15,18 @@
 
 # version info
 if ("$ENV{RSTUDIO_VERSION_MAJOR}" STREQUAL "")
-  set(CPACK_PACKAGE_VERSION_MAJOR "99")
+  set(CPACK_PACKAGE_VERSION_MAJOR "0")
   set(RSTUDIO_UNVERSIONED_BUILD TRUE)
 else()
   set(CPACK_PACKAGE_VERSION_MAJOR $ENV{RSTUDIO_VERSION_MAJOR})
 endif()
 if ("$ENV{RSTUDIO_VERSION_MINOR}" STREQUAL "")
-  set(CPACK_PACKAGE_VERSION_MINOR "9")
+  set(CPACK_PACKAGE_VERSION_MINOR "@MAJOR@")
 else()
   set(CPACK_PACKAGE_VERSION_MINOR $ENV{RSTUDIO_VERSION_MINOR})
 endif()
 if ("$ENV{RSTUDIO_VERSION_PATCH}" STREQUAL "")
-  set(CPACK_PACKAGE_VERSION_PATCH "9")
+  set(CPACK_PACKAGE_VERSION_PATCH "@MINOR@")
 else()
   set(CPACK_PACKAGE_VERSION_PATCH $ENV{RSTUDIO_VERSION_PATCH})
 endif()
diff -uNr rstudio-0.98.1103.orig/cmake/modules/FindLibR.cmake rstudio-0.98.1103/cmake/modules/FindLibR.cmake
--- rstudio-0.98.1103.orig/cmake/modules/FindLibR.cmake	2015-03-06 17:12:51.000000000 -0500
+++ rstudio-0.98.1103/cmake/modules/FindLibR.cmake	2015-04-17 23:25:56.000000000 -0400
@@ -22,10 +22,10 @@
 
    find_library(LIBR_LIBRARIES R)
    if(LIBR_LIBRARIES)
-      set(LIBR_HOME "${LIBR_LIBRARIES}/Resources" CACHE PATH "R home directory")
-      set(LIBR_INCLUDE_DIRS "${LIBR_HOME}/include" CACHE PATH "R include directory")
+      set(LIBR_HOME "@PREFIX@/Library/Frameworks/R.framework/Resources" CACHE PATH "R home directory")
+      set(LIBR_INCLUDE_DIRS "@PREFIX@/Library/Frameworks/R.framework/Versions/3.2/Resources/include" CACHE PATH "R include directory")
       set(LIBR_DOC_DIR "${LIBR_HOME}/doc" CACHE PATH "R doc directory")
-      set(LIBR_EXECUTABLE "${LIBR_HOME}/R" CACHE PATH "R executable")
+      set(LIBR_EXECUTABLE "${LIBR_HOME}/bin/R" CACHE PATH "R executable")
    endif()
 
 # detection for UNIX & Win32
diff -uNr rstudio-0.98.1103.orig/src/cpp/CMakeLists.txt rstudio-0.98.1103/src/cpp/CMakeLists.txt
--- rstudio-0.98.1103.orig/src/cpp/CMakeLists.txt	2015-03-06 17:12:51.000000000 -0500
+++ rstudio-0.98.1103/src/cpp/CMakeLists.txt	2015-04-17 23:25:56.000000000 -0400
@@ -107,7 +107,9 @@
 # unless we are building a non-packaged build on linux (in which case
 # boost dynamic libraries are presumed to be installed on the system ldpath)
 if(APPLE OR WIN32 OR RSTUDIO_PACKAGE_BUILD)
-   set(Boost_USE_STATIC_LIBS ON)
+# not for fink...
+#   set(Boost_USE_STATIC_LIBS ON)
+# ...bloats the build and fink no longer supplies most static libs
 endif()
 
 # require boost 1.50
diff -uNr rstudio-0.98.1103.orig/src/cpp/core/r_util/REnvironmentPosix.cpp rstudio-0.98.1103/src/cpp/core/r_util/REnvironmentPosix.cpp
--- rstudio-0.98.1103.orig/src/cpp/core/r_util/REnvironmentPosix.cpp	2015-03-06 17:12:51.000000000 -0500
+++ rstudio-0.98.1103/src/cpp/core/r_util/REnvironmentPosix.cpp	2015-04-17 23:27:08.000000000 -0400
@@ -86,10 +86,7 @@
 {
    // define potential paths
    std::vector<std::string> rScriptPaths;
-   rScriptPaths.push_back("/usr/bin/R");
-   rScriptPaths.push_back("/usr/local/bin/R");
-   rScriptPaths.push_back("/opt/local/bin/R");
-   rScriptPaths.push_back("/Library/Frameworks/R.framework/Resources/bin/R");
+   rScriptPaths.push_back("@PREFIX@/Library/Frameworks/R.framework/Versions/3.2/Resources/bin/R");
    return scanForRScript(rScriptPaths, pErrMsg);
 }
 
@@ -149,7 +146,7 @@
                                     std::string* pErrMsg)
 {
    // home path
-   *pHomePath = FilePath("/Library/Frameworks/R.framework/Resources");
+   *pHomePath = FilePath("@PREFIX@/Library/Frameworks/R.framework/Resources");
 
    // lib path
    std::string rLibPath;
diff -uNr rstudio-0.98.1103.orig/src/cpp/desktop/CMakeLists.txt rstudio-0.98.1103/src/cpp/desktop/CMakeLists.txt
--- rstudio-0.98.1103.orig/src/cpp/desktop/CMakeLists.txt	2015-03-06 17:12:51.000000000 -0500
+++ rstudio-0.98.1103/src/cpp/desktop/CMakeLists.txt	2015-04-17 23:25:56.000000000 -0400
@@ -356,7 +356,7 @@
           file(REMOVE \"\${CMAKE_INSTALL_PREFIX}/RStudio.app/Contents/MacOS/rsession\")
           file(REMOVE \"\${CMAKE_INSTALL_PREFIX}/RStudio.app/Contents/MacOS/rserver\")
           file(REMOVE \"\${CMAKE_INSTALL_PREFIX}/RStudio.app/Contents/MacOS/rserver-pam\")
-          fixup_bundle(\"${APPS}\" \"\${QTPLUGINS}\" \"${DIRS}\")
+          # fixup_bundle(\"${APPS}\" \"\${QTPLUGINS}\" \"${DIRS}\")
           ")
 
    # copy qt shared objects directly to bin on unix (fixed up rpath above)
diff -uNr rstudio-0.98.1103.orig/src/cpp/desktop-mac/MainFrameController.mm rstudio-0.98.1103/src/cpp/desktop-mac/MainFrameController.mm
--- rstudio-0.98.1103.orig/src/cpp/desktop-mac/MainFrameController.mm	2015-03-06 17:12:51.000000000 -0500
+++ rstudio-0.98.1103/src/cpp/desktop-mac/MainFrameController.mm	2015-04-17 23:25:56.000000000 -0400
@@ -13,6 +13,10 @@
  *
  */
 
+// dmacks: fix build against boost1.53
+// see: https://svn.boost.org/trac/boost/ticket/6219
+#define __ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES 0
+
 #import "MainFrameController.h"
 
 #include <boost/regex.hpp>
