diff -ruN simcoal-orig/Makefile simcoal/Makefile
--- simcoal-orig/Makefile	2001-12-10 09:54:28.000000000 -0500
+++ simcoal/Makefile	2013-09-15 17:45:46.000000000 -0400
@@ -1,8 +1,8 @@
-CC=g++ -O
+CC=g++ -O3
 
 SOURCES=$(shell ls *.cpp)
 OBJECTS=$(shell for file in $(SOURCES);\
-                do echo -n $$file | sed -e "s/\(.*\)\.cpp/\1\.o/"; echo -n " ";\
+                do echo $$file | sed -e "s/\(.*\)\.cpp/\1\.o/"; echo " ";\
                 done)
 
 PRGNAME=simcoal
diff -ruN simcoal-orig/deme.cpp simcoal/deme.cpp
--- simcoal-orig/deme.cpp	2004-06-08 10:40:34.000000000 -0400
+++ simcoal/deme.cpp	2013-09-15 17:45:59.000000000 -0400
@@ -686,7 +686,7 @@
          return 0;
       }
    }
-   else MinCoalTimes->set_elems(MAXLONG);
+   else MinCoalTimes->set_elems(LONG_MAX);
 
    //Compute total coalescent times within and among demes
    root.compute_total_coal_times_among_demes(*this,*CoalTimes, *PairDiff, *MinCoalTimes);
diff -ruN simcoal-orig/migrmat.h simcoal/migrmat.h
--- simcoal-orig/migrmat.h	1999-09-16 17:54:56.000000000 -0400
+++ simcoal/migrmat.h	2013-09-15 17:46:45.000000000 -0400
@@ -3,7 +3,11 @@
 
 #include "cstring.h"
 #include <fstream.h>
-#include <values.h>
+#if defined(WIN32) || (defined(__APPLE__) && defined(__MACH__))
+#  include <float.h>
+#else
+#  include <values.h>
+#endif
 #include <iomanip.h>
 #include <float.h>
 #include <math.h>
