diff -Nurd tsa/src/Makefile tsa.patched/src/Makefile
--- tsa/src/Makefile	2012-02-13 05:49:19.000000000 -0500
+++ tsa.patched/src/Makefile	2012-02-14 19:31:59.000000000 -0500
@@ -28,10 +28,10 @@
 # Instead of building "mex shortpath.c" and "mex uigetpath.c", I used empty m-functions within argout=argin; 
 ####################################################
 
-CC  	= gcc
-CXX 	= g++
+CC  	= gcc-fsf-4.6
+CXX 	= g++-fsf-4.6
 CFLAGS 	= -fopenmp -Wall -Wextra -Wconversion -O2 -fPIC 
-OCTMEX	= mkoctfile$(OCTAVE_VERSION) --mex
+OCTMEX	= mkoctfile-@OCTVERSION@ --mex
 RM      = rm
 
 MEX_OPTION  += -lgomp CC\#$(CXX) CXX\#$(CXX) CFLAGS\#"$(CFLAGS)" CXXFLAGS\#"$(CFLAGS)"
@@ -58,7 +58,7 @@
 #	Octave, MATLAB on Linux 
 #########################################################
 %.oct: %.cc
-	mkoctfile$(OCTAVE_VERSION) $<
+	mkoctfile-@OCTVERSION@-gcc4.6 --mex $<
 
 %.mex: %.cpp
 	$(OCTMEX) $<  -lgomp         ## Octave
diff -Nurd tsa/src/covm_mex.cpp tsa.patched/src/covm_mex.cpp
--- tsa/src/covm_mex.cpp	2012-02-14 20:16:37.000000000 -0500
+++ tsa.patched/src/covm_mex.cpp	2012-02-14 20:19:03.000000000 -0500
@@ -40,9 +40,9 @@
 //-------------------------------------------------------------------
 */
 
-#ifdef __GNUC__ 
+/*#ifdef __GNUC__ 
         #include <stdint.h>
-#endif
+#endif*/
 #include <math.h>
 #include "mex.h"
 
