diff -ruN vcftools_0.1.12b-orig/Makefile vcftools_0.1.12b/Makefile
--- vcftools_0.1.12b-orig/Makefile	2014-08-01 14:11:22.000000000 -0500
+++ vcftools_0.1.12b/Makefile	2014-09-24 19:50:01.000000000 -0500
@@ -20,14 +20,19 @@
     export PREFIX = $(dir $(realpath $(lastword $(MAKEFILE_LIST))))
 endif
 export BINDIR = ${PREFIX}/bin
-export MANDIR = ${PREFIX}/bin/man1
-export MODDIR = ${PREFIX}/lib/perl5/site_perl
+export MANDIR = ${PREFIX}/share/man1
+export MODDIR = ${PREFIX}/lib/perl5
+export DESTDIR = 
 
 DIRS = cpp perl
+
+vcftools:
+	for dir in $(DIRS); do cd $$dir && $(MAKE) && cd ..; done
+
 install:
-	    @mkdir -p $(BINDIR); mkdir -p $(MODDIR); mkdir -p $(MANDIR); \
-	    cp ${PREFIX}/cpp/vcftools.1 $(MANDIR); \
-        for dir in $(DIRS); do cd $$dir && $(MAKE) $(MAKEFLAGS) && cd ..; done
+	mkdir -p $(DESTDIR)$(BINDIR); mkdir -p $(DESTDIR)$(MODDIR); mkdir -p $(DESTDIR)$(MANDIR); \
+	cp cpp/vcftools.1.gz $(DESTDIR)$(MANDIR); \
+	for dir in $(DIRS); do cd $$dir && $(MAKE) install && cd ..; done
 
 clean:
 		@for dir in $(DIRS); do cd $$dir && $(MAKE) clean && cd ..; done
diff -ruN vcftools_0.1.12b-orig/cpp/Makefile vcftools_0.1.12b/cpp/Makefile
--- vcftools_0.1.12b-orig/cpp/Makefile	2014-08-01 14:11:22.000000000 -0500
+++ vcftools_0.1.12b/cpp/Makefile	2014-09-24 19:50:57.000000000 -0500
@@ -36,8 +36,9 @@
 
 vcftools: $(OBJS)
 	$(CPP) $(CPPFLAGS) $(OBJS) -o vcftools $(LIB)
+install:
 ifdef BINDIR
-	cp $(CURDIR)/$@ $(BINDIR)/$@
+	cp $(CURDIR)/vcftools $(DESTDIR)$(BINDIR)/vcftools
 endif
 
 bgzf: bgzf.c
diff -ruN vcftools_0.1.12b-orig/perl/Makefile vcftools_0.1.12b/perl/Makefile
--- vcftools_0.1.12b-orig/perl/Makefile	2014-08-01 14:11:22.000000000 -0500
+++ vcftools_0.1.12b/perl/Makefile	2014-09-24 19:52:22.000000000 -0500
@@ -4,9 +4,11 @@
     vcf-query vcf-shuffle-cols vcf-sort vcf-stats vcf-subset vcf-to-tab vcf-tstv vcf-validator
 MOD = FaSlice.pm Vcf.pm VcfStats.pm
 
+vcftools:
+
 install:
-	    @for i in $(BIN); do cp $(CURDIR)/$$i $(BINDIR)/$$i; done; \
-        for i in $(MOD); do cp $(CURDIR)/$$i $(MODDIR)/$$i; done; 
+	@for i in $(BIN); do cp $(CURDIR)/$$i $(DESTDIR)$(BINDIR)/$$i; done; \
+	 for i in $(MOD); do cp $(CURDIR)/$$i $(DESTDIR)$(MODDIR)/$$i; done;
 
 clean:
 		@for i in $(BIN); do rm -f $(BINDIR)/$$i; done; \
