diff -ruN vcftools_0.1.11-orig/Makefile vcftools_0.1.11/Makefile
--- vcftools_0.1.11-orig/Makefile	2013-06-13 10:40:56.000000000 -0400
+++ vcftools_0.1.11/Makefile	2013-08-15 21:47:45.000000000 -0400
@@ -17,12 +17,17 @@
     export PREFIX = $(dir $(realpath $(lastword $(MAKEFILE_LIST))))
 endif
 export BINDIR = ${PREFIX}/bin
-export MODDIR = ${PREFIX}/lib/perl5/site_perl
+export MODDIR = ${PREFIX}/lib/perl5
+export DESTDIR = 
 
 DIRS = cpp perl
+
+vcftools:
+	for dir in $(DIRS); do cd $$dir && $(MAKE) && cd ..; done
+
 install:
-	    @mkdir -p $(BINDIR); mkdir -p $(MODDIR); \
-        for dir in $(DIRS); do cd $$dir && $(MAKE) $(MAKEFLAGS) && cd ..; done
+	    mkdir -p $(DESTDIR)$(BINDIR); mkdir -p $(DESTDIR)$(MODDIR); \
+        for dir in $(DIRS); do cd $$dir && $(MAKE) install && cd ..; done
 
 clean:
 		@for dir in $(DIRS); do cd $$dir && $(MAKE) clean && cd ..; done
diff -ruN vcftools_0.1.11-orig/cpp/Makefile vcftools_0.1.11/cpp/Makefile
--- vcftools_0.1.11-orig/cpp/Makefile	2013-06-13 10:40:54.000000000 -0400
+++ vcftools_0.1.11/cpp/Makefile	2013-08-15 21:49:08.000000000 -0400
@@ -39,8 +39,10 @@
 
 vcftools: $(OBJS)
 	$(CPP) $(CPPFLAGS) $(OBJS) -o vcftools $(LIB)
+
+install:
 ifdef BINDIR
-	cp $(CURDIR)/$@ $(BINDIR)/$@
+	cp $(CURDIR)/vcftools $(DESTDIR)$(BINDIR)/vcftools
 endif
 
 bgzf: bgzf.c
diff -ruN vcftools_0.1.11-orig/perl/Makefile vcftools_0.1.11/perl/Makefile
--- vcftools_0.1.11-orig/perl/Makefile	2013-06-13 10:40:56.000000000 -0400
+++ vcftools_0.1.11/perl/Makefile	2013-08-15 21:50:26.000000000 -0400
@@ -4,9 +4,11 @@
     vcf-query vcf-shuffle-cols vcf-sort vcf-stats vcf-subset vcf-to-tab vcf-tstv vcf-validator
 MOD = FaSlice.pm Vcf.pm VcfStats.pm
 
+vcftools:
+
 install:
-	    @for i in $(BIN); do cp $(CURDIR)/$$i $(BINDIR)/$$i; done; \
-        for i in $(MOD); do cp $(CURDIR)/$$i $(MODDIR)/$$i; done; 
+	    @for i in $(BIN); do cp $(CURDIR)/$$i $(DESTDIR)$(BINDIR)/$$i; done; \
+        for i in $(MOD); do cp $(CURDIR)/$$i $(DESTDIR)$(MODDIR)/$$i; done; 
 
 clean:
 		@for i in $(BIN); do rm -f $(BINDIR)/$$i; done; \
