diff -Nurd xcircuit-3.9.15/functions.c xcircuit-3.9.15.patched/functions.c
--- xcircuit-3.9.15/functions.c	2014-06-28 12:17:40.000000000 -0700
+++ xcircuit-3.9.15.patched/functions.c	2014-06-28 12:17:57.000000000 -0700
@@ -2218,7 +2218,7 @@
 
    if (!areawin->redraw_ongoing) {
       areawin->redraw_needed = True;
-      return;
+      return 0.0;
    }
 
    if (areawin->selects == 0) return 0.0;
