--- nedit-5.3/makefiles/Makefile.macosx.orig	 Mon Jul  1 05:47:10 2002
+++ nedit-5.3/makefiles/Makefile.macosx		Mon Jul  1 05:42:43 2002
@@ -2,16 +2,16 @@
 # Makefile for MacOS X, using Tenon's Xtools
 # Contributed from hopperj@macconnect.com
 CC=cc
-AR=Libtool
+AR=libtool
 # For editres, add -DEDITRES to CFLAGS and -lXmu to LIBS\\
-CFLAGS=-O -no-cpp-precomp -DNO_XMIM -I${PREFIX}/include -DUSE_DIRENT -DUSE_LPR_PRINT_CMD -DUSE_XMIM
-LIBS= ${MOTIFLIB} -bind_at_load -L/usr/X11R6/lib /usr/X11R6/lib/libXm.2.dylib /usr/X11R6/lib/libXp.6.dylib -lm -lXpm \
--lXext -lXt -lSM -lICE -lX11
+CFLAGS=-O -no-cpp-precomp -DNO_XMIM -I${PREFIX}/include -I/usr/X11R6/include -DUSE_DIRENT -DUSE_LPR_PRINT_CMD -DUSE_XMIM
+#LIBS= ${MOTIFLIB} -bind_at_load -L/usr/X11R6/lib /usr/X11R6/lib/libXm.2.dylib /usr/X11R6/lib/libXp.6.dylib -lm -lXpm \
+#-lXext -lXt -lSM -lICE -lX11
 #
 # That would be an alternative set of flags using XFree86 with LessTif:
 #
-#LIBS=-flat-namespace -force_flat_namespace -L/usr/local/lib\
-#     -L/usr/X11R6/lib -lm -lXm -lXt -lXpm -lXext -lSM -lICE -lX11
+LIBS=-force_flat_namespace -L${PREFIX}/lib\
+     -L/usr/X11R6/lib -lXm -lXt -lXpm -lXext -lSM -lICE -lX11
 
 #
 # generic part
