diff -urN abcde-2.1.19.orig/abcde abcde-2.1.19/abcde
--- abcde-2.1.19.orig/abcde	Fri Apr 16 15:40:58 2004
+++ abcde-2.1.19/abcde	Fri Apr 16 15:53:43 2004
@@ -1445,7 +1445,7 @@
 		cddafs)
 			# Find the track's mounted path
 			REALTRACKNUM=$(expr $UTRACKNUM + 0)
-			FILEPATH=$(mount | grep "$CDROM on" | sed 's/^[^ ]* on \([^(]*\) (.*/\1/')
+			FILEPATH=$(mount | grep "$CDROM on" | sed 's/^[^ ]* on \(.*\) (.*/\1/')
 			FILEPATH=$(find "$FILEPATH" | grep "/$REALTRACKNUM ");
 			# If the file exists, copy it
 			if [ -e "$FILEPATH" ] ; then
@@ -1553,7 +1553,7 @@
 CDDBPROTO=5
 HELLOINFO="$(whoami)@$(hostname)"
 INTERACTIVE=y
-CDROMREADERSYNTAX=cdparanoia
+CDROMREADERSYNTAX=cddafs
 OUTPUTTYPE=ogg
 ENCODERSYNTAX=default
 
@@ -1562,10 +1562,9 @@
 FLACENCODERSYNTAX=default
 SPEEXENCODERSYNTAX=default
 
-OUTPUTFORMAT='${ARTISTFILE}-${ALBUMFILE}/${TRACKNUM}.${TRACKFILE}'
-# Use the following VAOUTPUTFORMAT to revert to 2.0.x VA format:
-#VAOUTPUTFORMAT=${OUTPUTFORMAT}
-VAOUTPUTFORMAT='Various-${ALBUMFILE}/${TRACKNUM}.${ARTISTFILE}-${TRACKFILE}'
+# These output formats are similar to iTunes
+OUTPUTFORMAT='${ARTISTFILE}/${ALBUMFILE}/$TRACKNUM ${TRACKFILE}'
+VAOUTPUTFORMAT=${OUTPUTFORMAT}
 ONETRACKOUTPUTFORMAT='${ARTISTFILE}-${ALBUMFILE}/${ALBUMFILE}'
 VAONETRACKOUTPUTFORMAT='Various-${ALBUMFILE}/${ALBUMFILE}'
 PLAYLISTFORMAT='${ARTISTFILE}-${ALBUMFILE}.${OUTPUT}.m3u'
@@ -1662,7 +1661,7 @@
 # Custom filename munging:
 mungefilename ()
 {
-	echo "$@" | sed s,:,\ -,g | tr \ / __ | tr -d \'\"\?\[:cntrl:\]
+	echo "$@" | sed s,:,\ -,g | tr / __ | tr -d \'\"\?\[:cntrl:\]
 }
 
 # Pre-read execution:
@@ -1695,7 +1694,7 @@
 USEID3=y
 
 if [ -z "$OUTPUTDIR" ]; then
-	OUTPUTDIR=$(pwd)
+	OUTPUTDIR="$HOME/Music"
 fi
 
 if [ -z "$WAVOUTPUTDIR" ]; then
@@ -1703,8 +1702,8 @@
 fi
 
 # Load system defaults
-if [ -r /etc/abcde.conf ]; then
-	. /etc/abcde.conf
+if [ -r @PREFIX@/etc/abcde.conf ]; then
+	. @PREFIX@/etc/abcde.conf
 fi
 # Load user preference defaults
 if [ -r $HOME/.abcde.conf ]; then
@@ -1851,6 +1850,10 @@
 		CDROMREADER="$CDDAFS"
 		CDROMREADEROPTS="$CDDAFSOPTS"
 		;;
+        cddafs)
+                CDROMREADER="$CDDAFS"
+                CDROMREADEROPTS="$CDDAFSOPTS"
+                ;;
 esac
 
 # There's only one normalize...
diff -urN abcde-2.1.19.orig/abcde.conf abcde-2.1.19/abcde.conf
--- abcde-2.1.19.orig/abcde.conf	Fri Apr 16 15:40:58 2004
+++ abcde-2.1.19/abcde.conf	Fri Apr 16 15:43:39 2004
@@ -48,7 +48,7 @@
 
 # CD reader program to use - currently recognized options are 'cdparanoia',
 # 'cdda2wav', 'dagrab' and 'cddafs' (Mac OS X only).
-#CDROMREADERSYNTAX=cdparanoia
+#CDROMREADERSYNTAX=cddafs
 
 # Keep the wav files after encoding. Set it to "y" and remove "clean" from
 # the list of default actions, since we purge the temp directory as default.
@@ -99,11 +99,11 @@
 # If HTTPGET is modified, the HTTPGETOPTS options should also be defined 
 # accordingly. If HTTPGET is changed, the default options will be set,
 # if HTTPGETOPTS is empty or not defined.
-#HTTPGET=wget
+#HTTPGET=curl
 # for fetch (FreeBSD): HTTPGETOPTS="-q -o -"
 # for wget: HTTPGETOPTS="-q -O -"
 # for curl (MacOSX): HTTPGETOPTS="-f -s"
-#HTTPGETOPTS="-q -O -"
+#HTTPGETOPTS="-f -s"
 
 # MP3:
 #LAMEOPTS=
@@ -157,15 +157,15 @@
 #NOCDDBQUERY=n
 
 # CD device you want to read from
-#CDROM=/dev/cdrom
+#CDROM=/dev/disk1
 
-# If you'd like to make a default location that overrides the current
-# directory for putting mp3's, uncomment this.
-#OUTPUTDIR=`pwd`
+# If you'd like to make a default location that overrides your Music
+# directory in your home directory for putting mp3's, uncomment this.
+#OUTPUTDIR="$HOME/Music"
 
 # Or if you'd just like to put the temporary .wav files somewhere else
 # you can specify that here
-#WAVOUTPUTDIR=`pwd`
+#OUTPUTDIR="$HOME/Music"
 
 # OUTPUTTYPE can be either "ogg", "mp3", "flac" or "spx", or a combination
 # of them separated with ",": "ogg,mp3".
@@ -181,10 +181,10 @@
 # Since multiple-output was integrated we always append the file type
 # to the files. Remove it from your user defined string if you are getting
 # files like ".ogg.ogg".
-#OUTPUTFORMAT='${ARTISTFILE}-${ALBUMFILE}/${TRACKNUM}.${TRACKFILE}'
+#OUTPUTFORMAT='${ARTISTFILE}/${ALBUMFILE}/$TRACKNUM ${TRACKFILE}'
 
 # Like OUTPUTFORMAT but for Various Artists discs.
-#VAOUTPUTFORMAT='${ARTISTFILE}/${TRACKFILE}'
+#VAOUTPUTFORMAT=${OUTPUTFORMAT}
 
 # Define how many encoders to run at once. This makes for huge speedups
 # on SMP systems. Defaults to 1. Equivalent to -j.
@@ -223,7 +223,7 @@
 # as $1 and outputs it on stdout.
 #mungefilename ()
 #{
-#	echo "$@" | sed s,:,\ -,g | tr \ / __ | tr -d \'\"\?\[:cntrl:\]
+#	echo "$@" | sed s,:,\ -,g | tr / __ | tr -d \'\"\?\[:cntrl:\]
 #}
 
 # Custom pre-read function
