--- ladspa_sdk/src/makefile	Sat Jul  6 13:34:20 2002
+++ ladspa_sdk-new/src/makefile	Sun Oct 26 14:04:12 2003
@@ -6,7 +6,7 @@
 # you may wish to remove the middle "local/" part of each entry.
 
 INSTALL_PLUGINS_DIR	=	/usr/local/lib/ladspa/
-INSTALL_INCLUDE_DIR	=	/usr/include/
+INSTALL_INCLUDE_DIR	=	/usr/local/include/
 INSTALL_BINARY_DIR	=	/usr/local/bin/
 
 ###############################################################################
@@ -16,7 +16,7 @@
 
 INCLUDES	=	-I.
 LIBRARIES	=	-ldl -lm
-CFLAGS		=	$(INCLUDES) -Wall -Werror -O3 -fPIC
+CFLAGS		=	$(INCLUDES) -Wall -Werror -O3 -fPIC -fno-common
 CXXFLAGS	=	$(CFLAGS)
 PLUGINS		=	../plugins/amp.so				\
 			../plugins/delay.so				\
@@ -28,6 +28,7 @@
 			../bin/listplugins
 CC		=	cc
 CPP		=	c++
+LD		= $(CC)
 
 ###############################################################################
 #
@@ -36,27 +37,16 @@
 
 ../plugins/%.so:	plugins/%.c ladspa.h
 	$(CC) $(CFLAGS) -o plugins/$*.o -c plugins/$*.c
-	$(LD) -o ../plugins/$*.so plugins/$*.o -shared
+	$(LD) -bundle -o ../plugins/$*.so plugins/$*.o
 
 ../plugins/%.so:	plugins/%.cpp ladspa.h
 	$(CPP) $(CXXFLAGS) -o plugins/$*.o -c plugins/$*.cpp
-	$(CPP) -o ../plugins/$*.so plugins/$*.o -shared
+	$(CPP) -bundle -o ../plugins/$*.so plugins/$*.o 
 
 ###############################################################################
 #
 # TARGETS
 #
-
-test:	/tmp/test.wav ../snd/noise.wav always
-	@echo ---------------------------------------------
-	@echo First listen to the white noise input signal:
-	@echo ---------------------------------------------
-	play ../snd/noise.wav
-	@echo -------------------------
-	@echo Compare to plugin output.
-	@echo -------------------------
-	@echo Should be a noise band around 6000Hz, repeated quietly after 1s.
-	play /tmp/test.wav
 
 install:	targets
 	-mkdirhier	$(INSTALL_PLUGINS_DIR)				\
--- ladspa_sdk/src/applyplugin.c	Sat May 18 13:01:36 2002
+++ ladspa_sdk-new/src/applyplugin.c	Sun Oct 26 14:00:33 2003
@@ -6,7 +6,7 @@
 /*****************************************************************************/
 
 #include <dlfcn.h>
-#include <endian.h>
+#include <machine/endian.h>
 #include <errno.h>
 #include <math.h>
 #include <stdlib.h>
