--- cloog-ppl/configure.in	2009-03-29 15:05:36.000000000 -0400
+++ cloog-ppl.autogen/configure.in	2009-04-01 09:51:20.000000000 -0400
@@ -103,8 +103,9 @@
 dnl Checks for library functions.
 AC_CHECK_FUNCS(strtol)
 
-AC_CHECK_HEADER(sys/resource.h,[AC_DEFINE(HAS_SYS_RESOURCE_H)],
-	        [AC_MSG_RESULT([no])])
+if test x$has_sys_resource = xyes; then
+  AC_DEFINE(HAS_SYS_RESOURCE_H, 1, [whether we have sys/resource.h])
+fi
 
 dnl /**************************************************************************
 dnl  *                             Option setting                             *
