diff -Nurd -x'*~' doxygen-1.5.6.orig/Makefile.in doxygen-1.5.6/Makefile.in
--- doxygen-1.5.6.orig/Makefile.in	2008-04-20 03:59:24.000000000 -0400
+++ doxygen-1.5.6/Makefile.in	2008-06-13 14:48:14.000000000 -0400
@@ -50,7 +50,7 @@
 
 DATE=$(shell date "+%B %Y")
 
-MAN1DIR = man/man1
+MAN1DIR = share/man/man1
 
 install: doxywizard_install 
 	$(INSTTOOL) -d $(INSTALL)/bin
diff -Nurd -x'*~' doxygen-1.5.6.orig/addon/doxywizard/doxywizard.pro.in doxygen-1.5.6/addon/doxywizard/doxywizard.pro.in
--- doxygen-1.5.6.orig/addon/doxywizard/doxywizard.pro.in	2008-01-01 05:41:47.000000000 -0500
+++ doxygen-1.5.6/addon/doxywizard/doxywizard.pro.in	2008-06-26 16:26:48.000000000 -0400
@@ -52,7 +52,7 @@
 MOC_DIR     =   moc
 
 # extra C++ compiler options
-INCLUDEPATH             += ../../src
+INCLUDEPATH             += ../../src @PREFIX@/include
 win32:TMAKE_CXXFLAGS    += -DQT_DLL
 win32:INCLUDEPATH       += .
 macosx-c++:LIBS+=-framework CoreFoundation -framework ApplicationServices
diff -Nurd -x'*~' doxygen-1.5.6.orig/configure doxygen-1.5.6/configure
--- doxygen-1.5.6.orig/configure	2008-05-17 13:51:10.000000000 -0400
+++ doxygen-1.5.6/configure	2008-06-13 14:45:41.000000000 -0400
@@ -569,7 +569,7 @@
     TMAKE_CXXFLAGS += -D_LARGEFILE_SOURCE
 EOF
 fi
-if test "$f_platform" = "macosx-uni-c++"; then
+if false; then
   if test -n "`ls /Developer/SDKs/MacOSX10.*.sdk 2>/dev/null`"; then
     mac_sdk=MacOSX10.4u.sdk
   fi
@@ -613,7 +613,6 @@
        echo "" >> $DST
        echo "all: src/version.cpp " >> $DST
        echo "	\$(MAKE) -C qtools" >> $DST
-       echo "	\$(MAKE) -C libpng" >> $DST
        echo "	\$(MAKE) -C libmd5" >> $DST
        echo "	\$(MAKE) -C src" >> $DST
        if test $f_wizard = YES; then
diff -Nurd -x'*~' doxygen-1.5.6.orig/qtools/qtools.pro.in doxygen-1.5.6/qtools/qtools.pro.in
--- doxygen-1.5.6.orig/qtools/qtools.pro.in	2006-09-04 16:53:41.000000000 -0400
+++ doxygen-1.5.6/qtools/qtools.pro.in	2008-06-26 16:27:08.000000000 -0400
@@ -77,7 +77,7 @@
 		  qdir_win32.cpp \
 		  qfileinfo_win32.cpp
 
-INCLUDEPATH = .
+INCLUDEPATH = . @PREFIX@/include
 TMAKE_CXXFLAGS += -DQT_NO_CODECS -DQT_LITE_UNICODE
 win32:TMAKE_CXXFLAGS += -DQT_NODLL
 win32-g++:TMAKE_CXXFLAGS += -D__CYGWIN__ -DALL_STATIC
diff -Nurd -x'*~' doxygen-1.5.6.orig/src/doxygen.pro.in doxygen-1.5.6/src/doxygen.pro.in
--- doxygen-1.5.6.orig/src/doxygen.pro.in	2008-01-01 05:40:58.000000000 -0500
+++ doxygen-1.5.6/src/doxygen.pro.in	2008-06-13 14:47:46.000000000 -0400
@@ -18,7 +18,7 @@
 CONFIG       =	console warn_on $extraopts
 HEADERS      =	doxygen.h 
 SOURCES      =	main.cpp 
-unix:LIBS                  += -L../lib -ldoxygen -ldoxycfg -lqtools -lpng -lmd5 
+unix:LIBS                  += -L../lib -L@PREFIX@/lib -ldoxygen -ldoxycfg -lqtools -lpng -lmd5 
 win32:INCLUDEPATH          += .
 win32-mingw:LIBS           += -L../lib -ldoxygen -ldoxycfg -lqtools -lpng -lmd5 
 win32-msvc:LIBS            += qtools.lib png.lib md5.lib doxygen.lib doxycfg.lib shell32.lib iconv.lib
@@ -28,7 +28,7 @@
 win32:TMAKE_CXXFLAGS       += -DQT_NODLL
 win32-g++:LIBS             = -L../lib -ldoxygen -ldoxycfg -lqtools -lpng -lmd5 -liconv
 win32-g++:TMAKE_CXXFLAGS   += -fno-exceptions -fno-rtti
-INCLUDEPATH                += ../qtools ../libpng ../libmd5 .
+INCLUDEPATH                += ../qtools ../libmd5 . @PREFIX@/include
 #win32-g++:INCLUDEPATH      -= ../libpng
 DESTDIR                    =  ../bin
 TARGET                     =  doxygen
diff -Nurd -x'*~' doxygen-1.5.6.orig/src/doxytag.pro.in doxygen-1.5.6/src/doxytag.pro.in
--- doxygen-1.5.6.orig/src/doxytag.pro.in	2008-01-01 05:40:58.000000000 -0500
+++ doxygen-1.5.6/src/doxytag.pro.in	2008-06-13 15:36:28.000000000 -0400
@@ -18,7 +18,7 @@
 CONFIG       =	console warn_on $extraopts
 HEADERS      =  logos.h version.h
 SOURCES      =	doxytag.cpp logos.cpp version.cpp 
-unix:LIBS                  += -L../lib -lqtools
+unix:LIBS                  += -L../lib -lqtools -L@PREFIX@/lib
 win32:INCLUDEPATH          += .
 win32-mingw:LIBS           += -L../lib -lqtools
 win32-msvc:LIBS            += qtools.lib shell32.lib 
@@ -26,6 +26,6 @@
 win32-borland:LIBS         += qtools.lib shell32.lib
 win32-borland:TMAKE_LFLAGS += -L..\lib -L$(BCB)\lib\psdk
 win32:TMAKE_CXXFLAGS       += -DQT_NODLL
-INCLUDEPATH                += ../qtools
+INCLUDEPATH                += ../qtools @PREFIX@/include
 OBJECTS_DIR                =  ../objects
 TARGET                     =  ../bin/doxytag
diff -Nurd -x'*~' doxygen-1.5.6.orig/src/libdoxycfg.pro.in doxygen-1.5.6/src/libdoxycfg.pro.in
--- doxygen-1.5.6.orig/src/libdoxycfg.pro.in	2008-01-01 05:40:58.000000000 -0500
+++ doxygen-1.5.6/src/libdoxycfg.pro.in	2008-06-26 15:42:44.000000000 -0400
@@ -20,7 +20,7 @@
 SOURCES      =	config.cpp portable.cpp
 win32:TMAKE_CXXFLAGS       += -DQT_NODLL
 win32-g++:TMAKE_CXXFLAGS   += -fno-exceptions -fno-rtti
-INCLUDEPATH                += ../qtools
+INCLUDEPATH                += ../qtools @PREFIX@/include
 win32:INCLUDEPATH          += .
 DESTDIR                    =  ../lib
 TARGET                     =  doxycfg
diff -Nurd -x'*~' doxygen-1.5.6.orig/src/libdoxygen.pro.in doxygen-1.5.6/src/libdoxygen.pro.in
--- doxygen-1.5.6.orig/src/libdoxygen.pro.in	2008-01-01 05:40:58.000000000 -0500
+++ doxygen-1.5.6/src/libdoxygen.pro.in	2008-06-13 14:46:38.000000000 -0400
@@ -224,8 +224,8 @@
 win32-g++:TMAKE_CXXFLAGS   += -fno-exceptions -fno-rtti
 linux-g++:TMAKE_CXXFLAGS    += -fno-exceptions -fno-rtti
 INCLUDEPATH                += ../qtools
-INCLUDEPATH                += ../libpng
 INCLUDEPATH                += ../libmd5
+INCLUDEPATH                += @PREFIX@/include
 win32:INCLUDEPATH          += .
 win32-g++:INCLUDEPATH      = ../qtools /usr/include/libpng12 ../libmd5
 DESTDIR                    =  ../lib
diff -Nurd -x'*~' doxygen-1.5.6.orig/src/pngenc.cpp doxygen-1.5.6/src/pngenc.cpp
--- doxygen-1.5.6.orig/src/pngenc.cpp	2008-01-01 05:41:08.000000000 -0500
+++ doxygen-1.5.6/src/pngenc.cpp	2008-06-13 15:17:57.000000000 -0400
@@ -25,7 +25,7 @@
 #endif
 
 #define ALL_STATIC
-#include <../libpng/png.h>
+#include <png.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -Nurd -x'*~' doxygen-1.5.6.orig/tmake/lib/macosx-uni-c++/tmake.conf doxygen-1.5.6/tmake/lib/macosx-uni-c++/tmake.conf
--- doxygen-1.5.6.orig/tmake/lib/macosx-uni-c++/tmake.conf	2007-11-18 07:10:01.000000000 -0500
+++ doxygen-1.5.6/tmake/lib/macosx-uni-c++/tmake.conf	2008-06-13 15:31:23.000000000 -0400
@@ -8,7 +8,7 @@
 CONFIG			= qt warn_on release
 
 TMAKE_CC		= cc
-TMAKE_CFLAGS		= -pipe -arch i386 -arch ppc
+TMAKE_CFLAGS		= -pipe 
 TMAKE_CFLAGS_WARN_ON	= -Wall -W
 TMAKE_CFLAGS_WARN_OFF	=
 TMAKE_CFLAGS_RELEASE	= -O2
@@ -36,7 +36,7 @@
 
 TMAKE_LINK		= c++
 TMAKE_LINK_SHLIB	= c++
-TMAKE_LFLAGS		= -Wl,-search_paths_first -arch i386 -arch ppc
+TMAKE_LFLAGS		= -Wl,-search_paths_first 
 TMAKE_LFLAGS_RELEASE	=
 TMAKE_LFLAGS_DEBUG	=
 TMAKE_LFLAGS_SHLIB	= -shared
