--- doxygen-1.4.3/src/libdoxygen.pro.in	2005-05-02 20:24:10.000000000 +0200
+++ doxygen-1.4.3/src/libdoxygen.pro-patched.in	2005-05-31 07:48:11.000000000 +0200
@@ -193,6 +193,7 @@
 win32:TMAKE_CXXFLAGS       += -DQT_NODLL 
 win32-msvc:TMAKE_CXXFLAGS  += -Zm200
 win32-g++:TMAKE_CXXFLAGS   += -fno-exceptions -fno-rtti
+INCLUDEPATH                += @PREFIX@/include
 INCLUDEPATH                += ../qtools
 INCLUDEPATH                += ../libpng
 INCLUDEPATH                += ../libmd5
--- doxygen-1.4.3/src/doxygen.pro.in	2005-01-03 20:17:55.000000000 +0100
+++ doxygen-1.4.3/src/doxygen.pro-patched.in	2005-05-31 07:50:38.000000000 +0200
@@ -28,7 +28,7 @@
 win32:TMAKE_CXXFLAGS       += -DQT_NODLL
 win32-g++:LIBS             = -L../lib -ldoxygen -ldoxycfg -lqtools -lpng -lmd5
 win32-g++:TMAKE_CXXFLAGS   += -fno-exceptions -fno-rtti
-INCLUDEPATH                += ../qtools ../libpng ../libmd5 .
+INCLUDEPATH                += @PREFIX@/include ../qtools ../libpng ../libmd5 .
 #win32-g++:INCLUDEPATH      -= ../libpng
 DESTDIR                    =  ../bin
 TARGET                     =  doxygen
