--- gtk-doc-1.5/omf.make	2005-08-15 18:13:31.000000000 +0200
+++ gtk-doc-1.5/omf-patched.make	2006-03-07 19:28:50.000000000 +0100
@@ -42,7 +42,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
@@ -50,4 +49,3 @@
 		rm -f $(omf_dest_dir)/$$basefile; \
 	done
 	-rmdir $(omf_dest_dir)
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
--- gtk-doc-1.5/help/manual/C/Makefile.in	2006-03-07 12:43:38.000000000 +0100
+++ gtk-doc-1.5/help/manual/C/Makefile-patched.in	2006-03-07 21:39:18.000000000 +0100
@@ -389,7 +389,6 @@
 @ENABLE_SCROLLKEEPER_TRUE@	for file in $(omffile); do \
 @ENABLE_SCROLLKEEPER_TRUE@		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 @ENABLE_SCROLLKEEPER_TRUE@	done
-@ENABLE_SCROLLKEEPER_TRUE@	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 @ENABLE_SCROLLKEEPER_TRUE@uninstall-local-omf:
 @ENABLE_SCROLLKEEPER_TRUE@	-for file in $(srcdir)/*.omf; do \
@@ -397,7 +396,6 @@
 @ENABLE_SCROLLKEEPER_TRUE@		rm -f $(omf_dest_dir)/$$basefile; \
 @ENABLE_SCROLLKEEPER_TRUE@	done
 @ENABLE_SCROLLKEEPER_TRUE@	-rmdir $(omf_dest_dir)
-@ENABLE_SCROLLKEEPER_TRUE@	-scrollkeeper-update -p $(scrollkeeper_localstate_dir)
 
 # If the following file is in a subdir (like help/) you need to add that to the path
 
