--- lam-7.1.2/config/lam_memory_setup.m4.org	2006-07-07 22:31:19.000000000 -0400
+++ lam-7.1.2/config/lam_memory_setup.m4	2006-07-07 22:31:38.000000000 -0400
@@ -56,7 +56,7 @@
 AC_DEFUN([LAM_DARWIN_MALLOC_SETUP],[
 case "$host" in
   *apple-darwin*)
-    WRAPPER_EXTRA_LDFLAGS="-Wl,-u -Wl,_lam_darwin_malloc_linker_hack -Wl,-multiply_defined,suppress -Wl,-force_flat_namespace -Wl,-flat_namespace $WRAPPER_EXTRA_LDFLAGS"
+    WRAPPER_EXTRA_LDFLAGS="-Wl,-u -Wl,_lam_darwin_malloc_linker_hack -Wl,-multiply_defined,suppress -Wl,-flat_namespace $WRAPPER_EXTRA_LDFLAGS"
     LDFLAGS="-Wl,-multiply_defined,suppress $LDFLAGS"
     ;;
   *)
