diff -Nurd -x'*~' m4-1.4.5.orig/src/builtin.c m4-1.4.5/src/builtin.c
--- m4-1.4.5.orig/src/builtin.c	2006-07-14 16:15:46.000000000 -0400
+++ m4-1.4.5/src/builtin.c	2006-08-08 01:06:26.000000000 -0400
@@ -1657,7 +1657,7 @@
 	case '\0':
 	  M4ERROR ((warning_status, 0, "\
 Warning: trailing \\ ignored in replacement"));
-	  break;
+	  return;
 
 	default:
 	  obstack_1grow (obs, ch);
diff -Nurd -x'*~' m4-1.4.5.orig/src/m4.h m4-1.4.5/src/m4.h
--- m4-1.4.5.orig/src/m4.h	2006-07-11 08:10:21.000000000 -0400
+++ m4-1.4.5/src/m4.h	2006-08-08 01:07:20.000000000 -0400
@@ -27,7 +27,9 @@
 
 /* Canonicalize UNIX recognition macros.  */
 #if defined unix || defined __unix || defined __unix__ \
-  || defined _POSIX_VERSION || defined _POSIX2_VERSION
+  || defined _POSIX_VERSION || defined _POSIX2_VERSION \
+  || defined __NetBSD__ || defined __OpenBSD__ \
+  || defined __APPLE__ || defined __APPLE_CC__
 # define UNIX 1
 #endif
 
