diff --git a/THANKS b/THANKS
index e127c26..6f40291 100644
--- a/THANKS
+++ b/THANKS
@@ -97,6 +97,7 @@ Stepan Kasal		kasal@ucw.cz
 Stephen Perkins		perkins@cps.msu.edu
 Steve Williamson	willy@uinpla.npl.uiuc.edu
 Steven Augart		saugart@yahoo.com
+Thomas Klausner		tk@giga.or.at
 Thorsten Ohl		ohl@physics.harvard.edu
 Tom McConnell		tmcconne@sedona.intel.com
 Tom Quinn		trq@dionysos.thphys.ox.ac.uk
diff --git a/src/output.c b/src/output.c
index 8c38ecb..617c939 100644
--- a/src/output.c
+++ b/src/output.c
@@ -252,6 +252,11 @@ m4_tmpopen (int divnum)
   else if (set_cloexec_flag (fileno (file), true) != 0)
     M4ERROR ((warning_status, errno,
 	      "Warning: cannot protect diversion across forks"));
+  /* POSIX states that it is undefined whether an append stream starts
+     at offset 0 or at the end.  We want the beginning.  */
+  else if (fseeko (file, 0, SEEK_SET) != 0)
+    M4ERROR ((EXIT_FAILURE, errno,
+	      "cannot seek to beginning of diversion"));
   return file;
 }
 
