diff -Nur makepatch-2.00_09.old/Makefile.PL makepatch-2.00_09/Makefile.PL
--- makepatch-2.00_09.old/Makefile.PL	Fri Sep 20 05:49:18 2002
+++ makepatch-2.00_09/Makefile.PL	Tue Nov  4 23:54:56 2003
@@ -55,7 +55,7 @@
 EOD
 }
 
-my $TMPDIR = $ENV{"TMPDIR"} || $ENV{TEMP} || "/usr/tmp";
+my $TMPDIR = $ENV{"TMPDIR"} || $ENV{TEMP} || "/tmp";
 unless ( -d $TMPDIR && -w $TMPDIR ) {
 print STDERR <<EOD;
 The default location for temporary files ($TMPDIR) does not
diff -Nur makepatch-2.00_09.old/script/applypatch makepatch-2.00_09/script/applypatch
--- makepatch-2.00_09.old/script/applypatch	Thu Nov 22 19:22:18 2001
+++ makepatch-2.00_09/script/applypatch	Tue Nov  4 23:55:40 2003
@@ -13,7 +13,7 @@
 use Getopt::Long 2.00;
 use File::Basename;
 use File::Spec;
-use IO;
+use IO qw(File); 
 use Text::ParseWords;
 #use Diff;
 #use Patch;
diff -Nur makepatch-2.00_09.old/script/makepatch makepatch-2.00_09/script/makepatch
--- makepatch-2.00_09.old/script/makepatch	Thu Nov 22 19:22:18 2001
+++ makepatch-2.00_09/script/makepatch	Tue Nov  4 23:54:56 2003
@@ -62,7 +62,7 @@
 
 my $exclude_pat;		# regex to exclude
 my @workq = ();			# pre/post work
-my $TMPDIR = $ENV{"TMPDIR"} || $ENV{TEMP} || "/usr/tmp";
+my $TMPDIR = $ENV{"TMPDIR"} || $ENV{TEMP} || "/tmp";
 my $dot_u = File::Spec::Unix->curdir; # UNIX current dir
 my $dot = File::Spec->curdir;	# current dir
 my $dotdot = File::Spec->updir;	# parent dir
