diff -urN memcached-1.1.12_old/memcached.c memcached-1.1.12/memcached.c
--- memcached-1.1.12_old/memcached.c    2005-04-05 02:10:26.000000000 +0200
+++ memcached-1.1.12/memcached.c        2006-03-11 16:08:08.000000000 +0100
@@ -102,6 +102,7 @@
 void set_cork (conn *c, int val) {
     if (c->is_corked == val) return;
     c->is_corked = val;
+#undef TCP_NOPUSH
 #ifdef TCP_NOPUSH
     setsockopt(c->sfd, IPPROTO_TCP, TCP_NOPUSH, &val, sizeof(val));
 #endif
