diff -bur nana-2.5/ nana-2.5-patched/
--- nana-2.5/man/Makefile.in	1999-09-13 15:58:15.000000000 -0700
+++ nana-2.5-patched/man/Makefile.in	2005-06-09 17:34:17.000000000 -0700
@@ -266,16 +266,6 @@
 maintainer-clean-generic clean mostlyclean distclean maintainer-clean
 
 
-# We seem to have a problem with automake, if I include nana.1 in the
-# man_MANS variable we don't install it. The following dodgey aims
-# to fix it.
-
-install-data-local:
-	@sect=1;				\
-	inst=`echo "nana" | sed '$(transform)'`.1; \
-	echo installing nana.1 as $(mandir)/man$$sect/$$inst; \
-	$(INSTALL_DATA) $(srcdir)/nana.1 $(mandir)/man$$sect/$$inst
-
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
