diff -Nurd qtscriptgenerator-src-0.1.0/generator/typesystem_core.xml qtscriptgenerator-src-0.1.0-new/generator/typesystem_core.xml
--- qtscriptgenerator-src-0.1.0/generator/typesystem_core.xml	2009-02-20 06:42:24.000000000 -0500
+++ qtscriptgenerator-src-0.1.0-new/generator/typesystem_core.xml	2009-03-31 16:20:21.000000000 -0400
@@ -2090,6 +2090,7 @@
   
   <namespace-type name="QtConcurrent" target-type="class">
     <extra-includes>
+        <include file-name="QThreadPool" location="global"/>
         <include file-name="qtconcurrentreducekernel.h" location="global"/>
         <include file-name="qtconcurrentthreadengine.h" location="global"/>
     </extra-includes>
diff -Nurd qtscriptgenerator-src-0.1.0/qtbindings/qtbindings.pro qtscriptgenerator-src-0.1.0-new/qtbindings/qtbindings.pro
--- qtscriptgenerator-src-0.1.0/qtbindings/qtbindings.pro	2009-02-20 06:42:24.000000000 -0500
+++ qtscriptgenerator-src-0.1.0-new/qtbindings/qtbindings.pro	2009-03-31 16:19:39.000000000 -0400
@@ -1,3 +1,4 @@
+CONFIG += debug_and_release
 TEMPLATE = subdirs
 SUBDIRS = qtscript_core \
 	  qtscript_gui \
diff -Nurd qtscriptgenerator-src-0.1.0/qtbindings/qtbindingsbase.pri qtscriptgenerator-src-0.1.0-new/qtbindings/qtbindingsbase.pri
--- qtscriptgenerator-src-0.1.0/qtbindings/qtbindingsbase.pri	2009-02-20 06:42:24.000000000 -0500
+++ qtscriptgenerator-src-0.1.0-new/qtbindings/qtbindingsbase.pri	2009-03-31 16:19:24.000000000 -0400
@@ -3,6 +3,6 @@
 INCLUDEPATH += .
 DESTDIR = $$PWD/../plugins/script
 QT += script
-CONFIG += debug_and_release
+CONFIG += debug_and_release plugin hide_symbols
 GENERATEDCPP = $$PWD/../generated_cpp
 TARGET=$$qtLibraryTarget($$TARGET)
