diff -Naur tkcvs_8_0_3/tkdiff-svn-wrapper tkcvs_8_0_3-patched/tkdiff-svn-wrapper
--- tkcvs_8_0_3/tkdiff-svn-wrapper	1970-01-01 01:00:00.000000000 +0100
+++ tkcvs_8_0_3-patched/tkdiff-svn-wrapper	2004-06-29 17:06:06.000000000 +0200
@@ -0,0 +1,9 @@
+#!/bin/bash 
+ARGS=("$@") # creates an array variable 
+LABEL1="${ARGS[$(($#-5))]}"; unset ARGS[$(($#-5))] 
+DUMMY="${ARGS[$(($#-4))]}"; unset ARGS[$(($#-4))] 
+LABEL2="${ARGS[$(($#-3))]}"; unset ARGS[$(($#-3))] 
+FILE1="${ARGS[$(($#-2))]}"; unset ARGS[$(($#-2))] 
+FILE2="${ARGS[$(($#-1))]}"; unset ARGS[$(($#-1))] 
+
+tkdiff -L "$LABEL1" -L "$LABEL2" "$FILE1" "$FILE2" 
