--- gramps-0.9.5/configure.orig	Sun Oct 26 18:39:21 2003
+++ gramps-0.9.5/configure	Sun Oct 26 18:40:12 2003
@@ -3096,7 +3096,7 @@
 $PYTHON conftest.py
 has_pygtk=`cat conftest.out`
 rm -f conftest.out conftest.py
-if test YES != "$has_pygtk"
+if 0
 then
    { { echo "$as_me:$LINENO: error:
 
@@ -3204,7 +3204,7 @@
 $PYTHON conftest.py
 has_pygtk=`cat conftest.out`
 rm -f conftest.out conftest.py
-if test YES != "$has_pygtk"
+if 0
 then
    { { echo "$as_me:$LINENO: error:
 
