diff -urN mysql-4.0.17.orig/configure mysql-4.0.17/configure
--- mysql-4.0.17.orig/configure	Mon Dec 15 06:24:06 2003
+++ mysql-4.0.17/configure	Fri Jan  9 07:58:24 2004
@@ -23695,6 +23695,15 @@
     fi
     ;;
   *darwin6*)
+    if test "$ac_cv_c_compiler_gnu" = "yes"
+    then
+      FLAGS="-DHAVE_DARWIN_THREADS -D_P1003_1B_VISIBLE -DSIGNAL_WITH_VIO_CLOSE -DSIGNALS_DONT_BREAK_READ -DHAVE_BROKEN_REALPATH -DFN_NO_CASE_SENCE"
+      CFLAGS="$CFLAGS $FLAGS"
+      CXXFLAGS="$CXXFLAGS $FLAGS"
+      MAX_C_OPTIMIZE="-O"
+    fi
+    ;;
+  *darwin8*)
     if test "$ac_cv_c_compiler_gnu" = "yes"
     then
       FLAGS="-DHAVE_DARWIN_THREADS -D_P1003_1B_VISIBLE -DSIGNAL_WITH_VIO_CLOSE -DSIGNALS_DONT_BREAK_READ -DHAVE_BROKEN_REALPATH -DFN_NO_CASE_SENCE"
