diff -ru biabconverter-0.6-orig/biabconverter biabconverter-0.6/biabconverter
--- biabconverter-0.6-orig/biabconverter	Fri Feb 18 00:21:46 2005
+++ biabconverter-0.6/biabconverter	Fri Feb 18 00:26:09 2005
@@ -89,6 +89,7 @@
 use File::Spec;
 @lilyPath = split(':', $ENV{BIABCONVERTER_TEMPLATES} || "");
  unless ($0 =~ /cvs/) { 
+   push(@lilyPath, "$ENV{HOME}/.biabconverter/templates");
    push(@lilyPath, "$FindBin::Bin/../share/biabconverter/templates");
  }
 $lilyTmpl = "default.lyt";
diff -ru biabconverter-0.6-orig/lilywrite.pm biabconverter-0.6/lilywrite.pm
--- biabconverter-0.6-orig/lilywrite.pm	Fri Feb 18 00:21:46 2005
+++ biabconverter-0.6/lilywrite.pm	Fri Feb 18 00:24:50 2005
@@ -577,6 +577,11 @@
   my $offset = shift();
   $currentNote++; #global var
   my $note = $aMelody[$currentNote];
+  unless ((defined $note) and ($note->channel>0)) { 
+      print "last note processed\n" if ($debug);
+      $note=new melodyNote($lastOnset, $barDur,0,0,144,1);
+      $lastNoteProcessed=1;
+  }
   $lastOnset=$note->onset;
     #check if note is valid
   #while (($note->channel != 144 ) and ($note->channel != 147 ))  { 
@@ -591,11 +596,6 @@
   #  }
   #} 
   print "lilyWrite:getNextNote: Nr $currentNote RawData: ".$note->info if ($debug);
-  unless ((defined $note) and ($note->channel>0)) { 
-      print "last note processed\n" if ($debug);
-      $note=new melodyNote($lastOnset, $barDur,0,0,144,1);
-      $lastNoteProcessed=1;
-  }
 
     # check if note is on a "rest helper channel"
   if (($note->channel == 176 ) or ($note->channel == 179 ))  {
