--- icecast-1.3.12/configure	2002-04-11 00:50:20.000000000 +0200
+++ icecast-1.3.12_corr/configure	2005-11-28 10:47:00.000000000 +0100
@@ -874,18 +874,18 @@
 	ICECAST_DOCDIR=/usr/doc/icecast-${ICECAST_VERSION_SMALL}
 	USE_FSSTD=yes
 else
-	ICECAST_BINDIR=${prefix}/bin
-	ICECAST_SBINDIR=${prefix}/bin
-	ICECAST_ETCDIR=conf
-	ICECAST_ETCDIR_INST=${prefix}/conf
-	ICECAST_SHAREDIR=${prefix}
-	ICECAST_LOGDIR=logs
-	ICECAST_LOGDIR_INST=${prefix}/logs
-	ICECAST_TEMPLATEDIR=templates
-	ICECAST_TEMPLATEDIR_INST=${prefix}/templates
-	ICECAST_STATICDIR=static
-	ICECAST_STATICDIR_INST=${prefix}/static
-	ICECAST_DOCDIR=${prefix}/doc
+        ICECAST_BINDIR=${prefix}/bin
+        ICECAST_SBINDIR=${prefix}/bin
+        ICECAST_ETCDIR=${prefix}/etc/icecast       
+        ICECAST_ETCDIR_INST=${prefix}/etc/icecast          
+        ICECAST_SHAREDIR=${prefix}/share/icecast 
+        ICECAST_LOGDIR=${prefix}/var/log/icecast             
+        ICECAST_LOGDIR_INST=${prefix}/var/log/icecast           
+        ICECAST_TEMPLATEDIR=${prefix}/share/icecast/templates
+        ICECAST_TEMPLATEDIR_INST=${prefix}/share/icecast/templates
+        ICECAST_STATICDIR=${prefix}/share/icecast/static
+        ICECAST_STATICDIR_INST=${prefix}/share/icecast/static
+        ICECAST_DOCDIR=${prefix}/share/doc/icecast
 	USE_FSSTD=no
 fi
 
@@ -2489,7 +2489,7 @@
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   ac_save_LIBS="$LIBS"
-LIBS="-lcrypt  $LIBS"
+LIBS="$LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 2495 "configure"
 #include "confdefs.h"
@@ -2517,7 +2517,7 @@
 fi
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  CRYPTLIB="-lcrypt"
+  CRYPTLIB=""
 else
   echo "$ac_t""no" 1>&6
 fi
@@ -2775,7 +2775,7 @@
   echo "$ac_t""no" 1>&6
 fi
 
-	for lib in python python2.0 python1.6 python1.5; do
+	for lib in python${PYTHON_V}; do
 		echo $ac_n "checking for Py_Initialize in -l$lib""... $ac_c" 1>&6
 echo "configure:2781: checking for Py_Initialize in -l$lib" >&5
 ac_lib_var=`echo $lib'_'Py_Initialize | sed 'y%./+-%__p_%'`
@@ -3459,9 +3459,9 @@
 #include "confdefs.h"
 #include <sys/types.h>
 		#include <sys/socket.h>
+typedef socklen_t interim; 
+	    interim apan (void) {return 1;}
 int main() {
-typedef socklen_t int; 
-	    int apan (void) {return 1;}
 ; return 0; }
 EOF
 if { (eval echo configure:3468: \"$ac_compile\") 1>&5; (eval $ac_compile) 2>&5; }; then
--- icecast-1.3.12/src/sock.c	2001-03-28 01:41:18.000000000 +0200
+++ icecast-1.3.12_corr/src/sock.c	2005-11-28 10:14:04.000000000 +0100
@@ -28,12 +28,14 @@
 #endif
 
 #include "definitions.h"
+#undef _POSIX_C_SOURCE
 
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdarg.h>
 
 #include "definitions.h"
+#undef _POSIX_C_SOURCE
 
 #include <sys/types.h>
 #include <ctype.h>
