diff -Nurd -x'*~' libcanberra-0.6.orig/src/Makefile.in libcanberra-0.6/src/Makefile.in
--- libcanberra-0.6.orig/src/Makefile.in	2008-08-04 14:00:20.000000000 -0400
+++ libcanberra-0.6/src/Makefile.in	2008-08-08 17:38:01.000000000 -0400
@@ -455,11 +455,10 @@
 libcanberra_la_CFLAGS = $(AM_CFLAGS) $(VORBIS_CFLAGS) $(am__append_2) \
 	$(am__append_6) $(am__append_10)
 libcanberra_la_LIBADD = $(VORBIS_LIBS) $(am__append_3) $(am__append_7) \
-	$(am__append_11)
+	$(am__append_11) -lpub
 libcanberra_la_LDFLAGS = \
 	-export-dynamic \
-	-version-info $(LIBCANBERRA_VERSION_INFO) \
-	-Wl,-version-script=$(srcdir)/map-file
+	-version-info $(LIBCANBERRA_VERSION_INFO)
 
 plugin_LTLIBRARIES = $(am__append_4) $(am__append_8) $(am__append_12) \
 	$(am__append_14)
diff -Nurd -x'*~' libcanberra-0.6.orig/src/sound-theme-spec.c libcanberra-0.6/src/sound-theme-spec.c
--- libcanberra-0.6.orig/src/sound-theme-spec.c	2008-07-25 10:50:15.000000000 -0400
+++ libcanberra-0.6/src/sound-theme-spec.c	2008-08-08 16:57:55.000000000 -0400
@@ -26,6 +26,7 @@
 #include <unistd.h>
 
 #include <locale.h>
+#include <publib.h>
 
 #include "sound-theme-spec.h"
 #include "malloc.h"
