diff -Nurd -x'*~' libcanberra-0.6.orig/libcanberra.pc.in libcanberra-0.6/libcanberra.pc.in
--- libcanberra-0.6.orig/libcanberra.pc.in	2008-06-10 10:33:40.000000000 -0400
+++ libcanberra-0.6/libcanberra.pc.in	2008-08-09 15:44:50.000000000 -0400
@@ -7,5 +7,6 @@
 Description: Event Sound API
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -lcanberra @PTHREAD_LIBS@
+Libs.private: -lpub
 Cflags: -D_REENTRANT -I${includedir}
 Requires:
diff -Nurd -x'*~' libcanberra-0.7.orig/src/Makefile.in libcanberra-0.7/src/Makefile.in
--- libcanberra-0.7.orig/src/Makefile.in	2008-08-14 16:55:22.000000000 -0400
+++ libcanberra-0.7/src/Makefile.in	2008-08-26 03:33:48.000000000 -0400
@@ -458,7 +458,7 @@
 libcanberra_la_CFLAGS = $(AM_CFLAGS) $(VORBIS_CFLAGS) $(am__append_3) \
 	$(am__append_7) $(am__append_11)
 libcanberra_la_LIBADD = $(VORBIS_LIBS) $(am__append_4) $(am__append_8) \
-	$(am__append_12)
+	$(am__append_12) -lpub
 libcanberra_la_LDFLAGS = -export-dynamic -version-info \
 	$(LIBCANBERRA_VERSION_INFO) $(am__append_1)
 plugin_LTLIBRARIES = $(am__append_5) $(am__append_9) $(am__append_13) \
diff -Nurd -x'*~' libcanberra-0.6.orig/src/sound-theme-spec.c libcanberra-0.6/src/sound-theme-spec.c
--- libcanberra-0.6.orig/src/sound-theme-spec.c	2008-07-25 10:50:15.000000000 -0400
+++ libcanberra-0.6/src/sound-theme-spec.c	2008-08-08 16:57:55.000000000 -0400
@@ -26,6 +26,7 @@
 #include <unistd.h>
 
 #include <locale.h>
+#include <publib.h>
 
 #include "sound-theme-spec.h"
 #include "malloc.h"
