diff -Nurd -x'*~' libcanberra-0.10.orig/Makefile.am libcanberra-0.10/Makefile.am
--- libcanberra-0.10.orig/Makefile.am	2008-10-05 22:07:43.000000000 -0400
+++ libcanberra-0.10/Makefile.am	2008-11-04 01:25:41.000000000 -0500
@@ -29,10 +29,10 @@
 
 if HAVE_GTK
 pkgconfig_DATA += libcanberra-gtk.pc
+schemas_DATA     = libcanberra.schemas
 endif
 
 schemasdir       = $(GCONF_SCHEMA_FILE_DIR)
-schemas_DATA     = libcanberra.schemas
 
 if GCONF_SCHEMAS_INSTALL
 install-data-local:
diff -Nurd -x'*~' libcanberra-0.10.orig/configure.ac libcanberra-0.10/configure.ac
--- libcanberra-0.10.orig/configure.ac	2008-10-05 21:56:57.000000000 -0400
+++ libcanberra-0.10/configure.ac	2008-11-04 01:08:35.000000000 -0500
@@ -153,7 +153,7 @@
 
 #### [lib]iconv ####
 
-AM_ICONV
+#AM_ICONV
 
 ###################################
 #      External libraries         #
@@ -339,6 +339,7 @@
         ],
         [gtk=auto])
 
+AM_CONDITIONAL([GCONF_SCHEMAS_INSTALL], [false])
 if test "x${gtk}" != xno ; then
     PKG_CHECK_MODULES(GTK, [ gtk+-2.0 >= 2.13.4 gthread-2.0 ],
         [
@@ -346,6 +347,7 @@
             AC_DEFINE([HAVE_GTK], 1, [Have GTK?])
 
             AC_PATH_PROG(GCONFTOOL, gconftool-2)
+	    AM_GCONF_SOURCE_2
 
             GTK_VERSION=`$PKG_CONFIG --variable=gtk_binary_version gtk+-2.0`
             GTK_MODULES_DIR="${libdir}/gtk-2.0/modules"
@@ -360,7 +362,6 @@
     HAVE_GTK=0
 fi
 
-AM_GCONF_SOURCE_2
 
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
