diff -Nurd -x'*~' libcanberra-0.10.orig/src/Makefile.am libcanberra-0.10/src/Makefile.am
--- libcanberra-0.10.orig/src/Makefile.am	2008-10-05 14:43:59.000000000 -0400
+++ libcanberra-0.10/src/Makefile.am	2008-11-04 02:48:38.000000000 -0500
@@ -31,6 +31,10 @@
 EXTRA_DIST = \
 	map-file
 
+if HAVE_GTK
+lib_LTLIBRARIES =
+include_HEADERS =
+else
 lib_LTLIBRARIES = \
         libcanberra.la
 
@@ -270,6 +274,7 @@
 endif
 endif
 
+endif
 if HAVE_GTK
 
 lib_LTLIBRARIES += \
@@ -291,7 +296,7 @@
 	$(GTK_CFLAGS)
 libcanberra_gtk_la_LIBADD = \
 	$(GTK_LIBS) \
-	libcanberra.la
+	-lcanberra
 libcanberra_gtk_la_LDFLAGS = \
 	-export-dynamic -version-info $(LIBCANBERRA_GTK_VERSION_INFO)
 
@@ -301,7 +306,7 @@
 	$(GTK_CFLAGS)
 libcanberra_gtk_module_la_LIBADD = \
 	$(GTK_LIBS) \
-	libcanberra.la \
+	-lcanberra -lX11 \
 	libcanberra-gtk.la
 libcanberra_gtk_module_la_LDFLAGS = \
 	-avoid-version -module -export-dynamic
@@ -310,7 +315,7 @@
 	canberra-gtk-play.c
 canberra_gtk_play_LDADD = \
 	$(GTK_LIBS) \
-	libcanberra.la \
+	-lcanberra \
 	libcanberra-gtk.la
 canberra_gtk_play_CFLAGS = \
 	$(GTK_CFLAGS)
@@ -338,8 +343,11 @@
 
 endif
 
+if HAVE_GTK
+else
 test_canberra_SOURCES = \
         test-canberra.c
 test_canberra_LDADD = \
         $(AM_LDADD) \
         libcanberra.la
+endif
diff -Nurd -x'*~' libcanberra-0.10.orig/src/canberra-gtk-module.c libcanberra-0.10/src/canberra-gtk-module.c
--- libcanberra-0.10.orig/src/canberra-gtk-module.c	2008-10-03 15:47:32.000000000 -0400
+++ libcanberra-0.10/src/canberra-gtk-module.c	2008-11-04 02:48:38.000000000 -0500
@@ -693,12 +693,12 @@
     return TRUE;
 }
 
-static void install_hook(GType type, const char *signal, guint *sn) {
+static void install_hook(GType type, const char *signal_, guint *sn) {
     GTypeClass *type_class;
 
     type_class = g_type_class_ref(type);
 
-    *sn = g_signal_lookup(signal, type);
+    *sn = g_signal_lookup(signal_, type);
     g_signal_add_emission_hook(*sn, 0, emission_hook_cb, NULL, NULL);
 
     g_type_class_unref(type_class);
