diff -Nurd -x'*~' libcanberra-0.10.orig/src/Makefile.in libcanberra-0.10/src/Makefile.in
--- libcanberra-0.10.orig/src/Makefile.in	2008-10-05 21:57:34.000000000 -0400
+++ libcanberra-0.10/src/Makefile.in	2008-11-26 04:15:49.000000000 -0500
@@ -54,7 +54,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-noinst_PROGRAMS = test-canberra$(EXEEXT)
+noinst_PROGRAMS = 
 @USE_VERSION_SCRIPT_TRUE@am__append_1 = -Wl,-version-script=$(srcdir)/map-file
 @HAVE_CACHE_TRUE@am__append_2 = \
 @HAVE_CACHE_TRUE@	cache.c cache.h
@@ -527,8 +527,8 @@
 AM_CXXFLAGS = $(PTHREAD_CFLAGS)
 AM_LDADD = $(PTHREAD_LIBS)
 EXTRA_DIST = map-file $(am__append_27)
-lib_LTLIBRARIES = libcanberra.la $(am__append_25)
-include_HEADERS = canberra.h $(am__append_26)
+lib_LTLIBRARIES = $(am__append_25)
+include_HEADERS = $(am__append_26)
 libcanberra_la_SOURCES = canberra.h common.c common.h mutex-posix.c \
 	mutex.h proplist.c proplist.h driver.h read-sound-file.c \
 	read-sound-file.h read-vorbis.c read-vorbis.h read-wav.c \
@@ -848,7 +848,9 @@
 	$(libcanberra_oss_la_LINK) $(am_libcanberra_oss_la_rpath) $(libcanberra_oss_la_OBJECTS) $(libcanberra_oss_la_LIBADD) $(LIBS)
 libcanberra-pulse.la: $(libcanberra_pulse_la_OBJECTS) $(libcanberra_pulse_la_DEPENDENCIES) 
 	$(libcanberra_pulse_la_LINK) $(am_libcanberra_pulse_la_rpath) $(libcanberra_pulse_la_OBJECTS) $(libcanberra_pulse_la_LIBADD) $(LIBS)
-libcanberra.la: $(libcanberra_la_OBJECTS) $(libcanberra_la_DEPENDENCIES) 
+libcanberra.la:
+	ln -s $(libdir)/$@ .
+_libcanberra.la: $(libcanberra_la_OBJECTS) $(libcanberra_la_DEPENDENCIES) 
 	$(libcanberra_la_LINK) -rpath $(libdir) $(libcanberra_la_OBJECTS) $(libcanberra_la_LIBADD) $(LIBS)
 install-binPROGRAMS: $(bin_PROGRAMS)
 	@$(NORMAL_INSTALL)
diff -Nurd -x'*~' libcanberra-0.10.orig/src/canberra-gtk-module.c libcanberra-0.10/src/canberra-gtk-module.c
--- libcanberra-0.10.orig/src/canberra-gtk-module.c	2008-10-03 15:47:32.000000000 -0400
+++ libcanberra-0.10/src/canberra-gtk-module.c	2008-11-04 02:48:38.000000000 -0500
@@ -693,12 +693,12 @@
     return TRUE;
 }
 
-static void install_hook(GType type, const char *signal, guint *sn) {
+static void install_hook(GType type, const char *signal_, guint *sn) {
     GTypeClass *type_class;
 
     type_class = g_type_class_ref(type);
 
-    *sn = g_signal_lookup(signal, type);
+    *sn = g_signal_lookup(signal_, type);
     g_signal_add_emission_hook(*sn, 0, emission_hook_cb, NULL, NULL);
 
     g_type_class_unref(type_class);
