diff -ru libsndfile-1.0.24.orig/src/binheader_writef_check.py libsndfile-1.0.24/src/binheader_writef_check.py
--- libsndfile-1.0.24.orig/src/binheader_writef_check.py	2011-03-23 05:01:02.000000000 -0400
+++ libsndfile-1.0.24/src/binheader_writef_check.py	2011-03-30 19:41:16.000000000 -0400
@@ -42,7 +42,8 @@
 _whitespace_re = re.compile ("\s+", re.MULTILINE)
 
 def find_binheader_writefs (data):
-	lst = re.findall ('psf_binheader_writef\s*\(\s*[a-zA-Z_]+\s*,\s*\"[^;]+;', data, re.MULTILINE)
+	prog = re.compile ('psf_binheader_writef\s*\(\s*[a-zA-Z_]+\s*,\s*\"[^;]+;', re.MULTILINE)
+	lst = prog.findall(data)
 	return [_whitespace_re.sub (" ", x) for x in lst]
 
 def find_format_string (s):
