diff -ru libvorbis-1.1.2/doc/Makefile.in libvorbis-1.1.2-patched/doc/Makefile.in
--- libvorbis-1.1.2/doc/Makefile.in	2004-09-21 01:22:03.000000000 +0200
+++ libvorbis-1.1.2-patched/doc/Makefile.in	2005-01-16 13:48:19.000000000 +0100
@@ -194,7 +184,7 @@
 target_os = @target_os@
 target_vendor = @target_vendor@
 SUBDIRS = vorbisfile vorbisenc
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 
 ### all of the static docs, commited to SVN and included as is
 static_docs = \
diff -ru libvorbis-1.1.2/doc/vorbisenc/Makefile.in libvorbis-1.1.2-patched/doc/vorbisenc/Makefile.in
--- libvorbis-1.1.2/doc/vorbisenc/Makefile.in	2004-09-21 01:22:03.000000000 +0200
+++ libvorbis-1.1.2-patched/doc/vorbisenc/Makefile.in	2005-01-16 13:48:13.000000000 +0100
@@ -184,7 +174,7 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)/vorbisenc
+docdir = $(datadir)/doc/$(PACKAGE)/vorbisenc
 doc_DATA = index.html overview.html reference.html style.css\
 	 vorbis_encode_ctl.html vorbis_encode_init.html vorbis_info.html\
 	vorbis_encode_init_vbr.html
diff -ru libvorbis-1.1.2/doc/vorbisfile/Makefile.in libvorbis-1.1.2-patched/doc/vorbisfile/Makefile.in
--- libvorbis-1.1.2/doc/vorbisfile/Makefile.in	2004-09-21 01:22:03.000000000 +0200
+++ libvorbis-1.1.2-patched/doc/vorbisfile/Makefile.in	2005-01-16 13:48:07.000000000 +0100
@@ -184,7 +174,7 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)/vorbisfile
+docdir = $(datadir)/doc/$(PACKAGE)/vorbisfile
 doc_DATA = OggVorbis_File.html callbacks.html chaining_example_c.html\
 	chainingexample.html crosslap.html datastructures.html decoding.html\
 	example.html exampleindex.html fileinfo.html index.html\
