--- scizzor/scizzor.c.orig	Sat Dec 18 23:45:01 2004
+++ scizzor/scizzor.c	Sat Dec 18 23:45:05 2004
@@ -11,7 +11,7 @@
 #include <stdio.h>
 #include <gtk/gtk.h>
 #include <sys/ioctl.h>
-#include <sys/soundcard.h>
+#include <liboss/soundcard.h>
 #ifdef BMP
 #include <bmp/beepctrl.h>
 #endif
--- scizzor/Makefile.orig	Sat Dec 18 23:45:24 2004
+++ scizzor/Makefile	Sat Dec 18 23:50:18 2004
@@ -14,7 +14,7 @@
 xmms:
 	$(CC) $(XMMSCFLAGS) -fPIC -c scizzor.c -o scizzor.lo
 	$(CC) $(XMMSCFLAGS) -fPIC -c scizzor_snd.c -o scizzor_snd.lo
-	$(CC) -shared scizzor.lo scizzor_snd.lo $(XMMSLDFLAGS) -lc -o scizzor-xmms.so
+	$(CC) $(LDFLAGS) -bundle scizzor.lo scizzor_snd.lo $(XMMSLDFLAGS) -lc -o scizzor-xmms.so
 
 bmp:	
 	$(CC) $(BEEPCFLAGS) -fPIC -c scizzor.c -o scizzor.lo
