diff -uN autoconf-2.59/emacs-install autoconf-2.59-patched/emacs-install
--- autoconf-2.59/emacs-install	Wed Dec 31 19:00:00 1969
+++ autoconf-2.59-patched/emacs-install	Fri Jun 18 15:08:42 2004
@@ -0,0 +1,30 @@
+#!/bin/sh
+# /sw/lib/emacsen-common/packages/install/autoconf2.5
+set -e
+ 
+FLAVOR=$1
+echo install/autoconf2.5: Handling install of emacsen flavor ${FLAVOR}
+
+byte_compile_options="-batch -f batch-byte-compile"
+el_files="autoconf-mode.el autotest-mode.el"
+el_dir=/sw/share/emacs/site-lisp
+elc_dir=/sw/share/${FLAVOR}/site-lisp
+el_path_list="${el_dir}/autoconf-mode.el ${el_dir}/autotest-mode.el"
+elc_path_list="${elc_dir}/autoconf-mode.el ${elc_dir}/autotest-mode.el"
+
+if [ ${FLAVOR} != emacs ]
+then
+  echo install/autoconf2.5: byte-compiling for ${FLAVOR}
+
+  # Copy the temp .el files
+  cp ${el_path_list} ${elc_dir}
+
+  # Byte compile them
+  ${FLAVOR} ${byte_compile_options} ${elc_path_list}
+
+  # remove the redundant .el files
+  # presumes that any .el files in the <flavor> dir are trash.
+  rm ${elc_dir}/*.el
+fi
+exit 0;
+
diff -uN autoconf-2.59/emacs-remove autoconf-2.59-patched/emacs-remove
--- autoconf-2.59/emacs-remove	Wed Dec 31 19:00:00 1969
+++ autoconf-2.59-patched/emacs-remove	Fri Jun 18 15:06:57 2004
@@ -0,0 +1,14 @@
+#!/bin/sh
+# /sw/lib/emacsen-common/packages/remove/autoconf2.5
+set -e
+
+FLAVOR=$1
+echo remove/autoconf2.5: Handling removal of emacsen flavor ${FLAVOR}
+
+if [ ${FLAVOR} != emacs ]
+then
+  echo emacsen-common: purging byte-compiled files for ${FLAVOR}
+  rm -f /sw/share/${FLAVOR}/site-lisp/autoconf-mode.elc
+  rm -f /sw/share/${FLAVOR}/site-lisp/autotest-mode.elc
+fi
+exit 0;
