--- avr-libc-1.0.2/doc/api/Makefile.in	Thu Nov 27 06:32:57 2003
+++ avr-libc-1.0.2-patch/doc/api/Makefile.in	Tue Jan  6 11:51:34 2004
@@ -301,7 +301,7 @@
 ps: demo $(TARGET_DOX_PS)
 
 # Rule for generating html output.
-html: demo $(TARGET_DOX_HTML) fix_png fix_anchors fix_javascript
+html: demo $(TARGET_DOX_HTML) fix_png fix_anchors #fix_javascript
 
 # Make the background color of the png's transparent.
 fix_png: $(wildcard $(USER_MANUAL)/*.png)
