diff -Naur thelove@canonical.com---dists--bazaar--1.2.orig/src/baz/baz/Makefile.in thelove@canonical.com---dists--bazaar--1.2/src/baz/baz/Makefile.in
--- thelove@canonical.com---dists--bazaar--1.2.orig/src/baz/baz/Makefile.in	2005-03-02 05:36:42.000000000 -0800
+++ thelove@canonical.com---dists--bazaar--1.2/src/baz/baz/Makefile.in	2005-03-02 20:22:21.000000000 -0800
@@ -1,4 +1,4 @@
-EXTRA_CFLAGS:=$(EXTRA_CFLAGS) -I $(srcroot)/$(thispath)/..
+EXTRA_CFLAGS:=$(EXTRA_CFLAGS) -I $(srcroot)/$(thispath)/.. -L@PREFIX@/lib
 
 libs			:=	../commands/libcommands.a \
 							../libarch/libarch.a \
@@ -6,11 +6,11 @@
 							../libawk/libawk.a \
 							../libdate/libdate.a \
 							$(objroot)/hackerlab/libhackerlab/libhackerlab.a \
+							-lintl \
 							-lneon \
 							$(cfg__socket_library) \
 							$(cfg__resolv_library) \
-							$(cfg__net_library) \
-							-lm 
+							$(cfg__net_library)
 
 mains			:=	baz.c
 
diff -Naur thelove@canonical.com---dists--bazaar--1.2.orig/src/baz/commands/tests/Makefile.in thelove@canonical.com---dists--bazaar--1.2/src/baz/commands/tests/Makefile.in
--- thelove@canonical.com---dists--bazaar--1.2.orig/src/baz/commands/tests/Makefile.in	2005-03-02 05:36:31.000000000 -0800
+++ thelove@canonical.com---dists--bazaar--1.2/src/baz/commands/tests/Makefile.in	2005-03-02 23:05:59.000000000 -0800
@@ -1,3 +1,4 @@
+EXTRA_CFLAGS:=$(EXTRA_CFLAGS) -L@PREFIX@/lib
 
 libs			:=	../../commands/libcommands.a \
 				../../libarch/libarch.a \
@@ -8,8 +9,8 @@
                                 $(cfg__socket_library) \
                                 $(cfg__resolv_library) \
                                 $(cfg__net_library) \
-				-lneon \
-				-lm
+				-lintl \
+				-lneon
 
 include $(makefiles)/unit-tests.mk
 
diff -Naur thelove@canonical.com---dists--bazaar--1.2.orig/src/baz/libarch/tests/Makefile.in thelove@canonical.com---dists--bazaar--1.2/src/baz/libarch/tests/Makefile.in
--- thelove@canonical.com---dists--bazaar--1.2.orig/src/baz/libarch/tests/Makefile.in	2005-03-02 05:36:31.000000000 -0800
+++ thelove@canonical.com---dists--bazaar--1.2/src/baz/libarch/tests/Makefile.in	2005-03-02 23:05:14.000000000 -0800
@@ -1,3 +1,4 @@
+EXTRA_CFLAGS:=$(EXTRA_CFLAGS) -L@PREFIX@/lib
 
 # do not add libcommands.a here - commands may depend on libarch, but not 
 # vice verca.
@@ -9,8 +10,8 @@
 				$(cfg__socket_library) \
 				$(cfg__resolv_library) \
 				$(cfg__net_library) \
-				-lneon \
-				-lm
+				-lintl \
+				-lneon 
 
 include $(makefiles)/unit-tests.mk
 
