diff -u -r cscope-15.4/src/constants.h cscope-15.4-patched/src/constants.h
--- cscope-15.4/src/constants.h	2002-07-28 11:22:38.000000000 -0400
+++ cscope-15.4-patched/src/constants.h	2002-09-08 16:52:20.000000000 -0400
@@ -102,7 +102,7 @@
 #endif
 
 
-#ifndef __FreeBSD__	/* Prevent search issues in cscope.out */
+#if !defined(__FreeBSD__) && !defined(__APPLE__)	/* Prevent search issues in cscope.out */
 #if !TERMINFO
 #ifndef KEY_BREAK
 #define	KEY_BREAK	0400	/* easier to define than to add #if around the use */
@@ -127,6 +127,6 @@
 #endif	/* if UNIXPC */
 
 #endif	/* if !TERMINFO */
-#endif	/* ifndef __FreeBSD__ */
+#endif	/* if !defined(__FreeBSD__) && !defined(__APPLE__) */
 
 #endif /* CSCOPE_CONSTANTS_H */
diff -u -r cscope-15.4/src/global.h cscope-15.4-patched/src/global.h
--- cscope-15.4/src/global.h	2002-07-28 11:40:07.000000000 -0400
+++ cscope-15.4-patched/src/global.h	2002-09-08 16:52:38.000000000 -0400
@@ -90,11 +90,11 @@
 # define	tolower(c)	(islower(c) ? (c) : (c) - 'A' + 'a')	
 # define	toupper(c)	(isupper(c) ? (c) : (c) - 'a' + 'A')	
 # if !sun 
-#  if !__FreeBSD__
+#  if !__FreeBSD__ && !__APPLE__
 /* in traditional BSD, *printf() doesn't return the number of bytes
  * written */
 #   define PRINTF_RETVAL_BROKEN 1
-#  endif /* !FreeBSD */
+#  endif /* !FreeBSD && !__APPLE__ */
 # endif /* !sun */
 #endif
