diff -uNr distcc-2.15/src/dparent.c distcc-2.15-new/src/dparent.c
--- distcc-2.15/src/dparent.c	Tue Jul  6 08:39:38 2004
+++ distcc-2.15-new/src/dparent.c	Wed Jul  7 13:02:23 2004
@@ -154,13 +154,8 @@
         int sig = WTERMSIG(status);
         int severity = sig == SIGTERM ? RS_LOG_INFO : RS_LOG_ERR;
         
-#ifdef HAVE_STRSIGNAL
-        /* take off every sig! for great justice. */
-        rs_log(severity, "child %d: %s", (int) kid, strsignal(sig));
-#else
         rs_log(severity, "child %d: signal %d (%s)", (int) kid, sig,
-               WCOREDUMP(status) ? : "core dumped" : "no core");
-#endif
+               WCOREDUMP(status) ? "core dumped" : "no core");
     } else if (WIFEXITED(status)) {
         rs_log_info("child %d exited: exit status %d",
                     (int) kid, WEXITSTATUS(status));
