diff -Naur findbugs-0.6.0.orig/bin/findbugs findbugs-0.6.0/bin/findbugs
--- findbugs-0.6.0.orig/bin/findbugs	Sat Jun  7 06:44:56 2003
+++ findbugs-0.6.0/bin/findbugs	Sat Jun  7 12:28:10 2003
@@ -63,36 +63,7 @@
 	exit 0;
 }
 
-# Find findbugs_home.
-# This code shamelessly stolen from Ant's "bin/ant" script.
-
-## resolve links - $0 may be a link to ant's home
-PRG="$0"
-progname=`basename "$0"`
-saveddir=`pwd`
-
-# need this for relative symlinks
-dirname_prg=`dirname "$PRG"`
-cd "$dirname_prg"
-  
-while [ -h "$PRG" ] ; do
-	ls=`ls -ld "$PRG"`
-	link=`expr "$ls" : '.*-> \(.*\)$'`
-	if expr "$link" : '/.*' > /dev/null; then
-		PRG="$link"
-	else
-		PRG=`dirname "$PRG"`"/$link"
-	fi
-done
-  
-findbugs_home=`dirname "$PRG"`/..
-
-cd "$saveddir"
-
-# make it fully qualified
-findbugs_home=`cd "$findbugs_home" && pwd`
-
-# End of code shamelessly stolen from Ant.
+findbugs_home=@PREFIX@/share/java/findbugs
 
 if [ ! -d "$findbugs_home" ]; then
 	echo "The path $findbugs_home,"
