diff -Nurd robodoc-4.99.26-orig/Contributions/robodoc.kaptn robodoc-4.99.26/Contributions/robodoc.kaptn
--- robodoc-4.99.26-orig/Contributions/robodoc.kaptn	2003-10-01 03:50:54.000000000 +0200
+++ robodoc-4.99.26/Contributions/robodoc.kaptn	2006-07-30 00:27:45.000000000 +0200
@@ -39,7 +39,7 @@
 #
 intro "robodoc.rc builder" -> intro1 intro2;
 intro1 -> text1 introa;
-text1 -> @text("This is a graphical front-end for creating robodoc.rc \
+text1 -> @text("This is a graphical front-end for creating robodoc.rc \
 configuration files made using Kaptain, the Universal graphical front-end.");
 intro2 "" -> @;
 introa "" -> workdir @fill;
@@ -51,8 +51,8 @@
 # finish page, where I write the robodoc.rc file
 #
 finish "Finish" -> text3 other_settings;
-text3 -> @text("Now press the Preview button to see the settings you have made.\
-Don't forget to press Write button in order to save the setting in the given file.");
+text3 -> @text("Now press the Preview button to see the settings you have made.\
+ Don't forget to press Write button in order to save the setting in the given file.");
 other_settings -> filename @fill @preview(allfile)="Preview" @action(writetofile)="Write";
 writetofile-> "echo '" allfile "' > " filename ;
 filename -> @outfile="$PWD/robodoc.rc";
diff -Nurd robodoc-4.99.26-orig/Docs/makefile.in robodoc-4.99.26/Docs/makefile.in
--- robodoc-4.99.26-orig/Docs/makefile.in	2006-06-27 17:08:22.000000000 +0200
+++ robodoc-4.99.26/Docs/makefile.in	2006-07-29 20:36:38.000000000 +0200
@@ -47,9 +47,7 @@
 SOURCES =
 DIST_SOURCES =
 man1dir = $(mandir)/man1
-am__installdirs = "$(DESTDIR)$(man1dir)" "$(DESTDIR)$(manhdir)" \
-	"$(DESTDIR)$(docdir)"
-manhdir = $(mandir)/manh
+am__installdirs = "$(DESTDIR)$(man1dir)" "$(DESTDIR)$(docdir)"
 NROFF = nroff
 MANS = $(man_MANS)
 am__vpath_adj_setup = srcdirstrip=`echo "$(srcdir)" | sed 's|.|.|g'`;
@@ -132,10 +130,10 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-man_MANS = robodoc.1 robohdrs.1 manual.html
+man_MANS = robodoc.1 robohdrs.1
 # MAINTAINERCLEANFILES = robodoc.html stoc.html stoc.m4
-docdir = $(prefix)/doc/$(PACKAGE)-$(VERSION)
-doc_DATA = manual.html
+docdir = $(prefix)/share/doc/$(PACKAGE)
+doc_DATA = manual.html manual.css
 all: all-am
 
 .SUFFIXES:
@@ -214,51 +212,6 @@
 	  echo " rm -f '$(DESTDIR)$(man1dir)/$$inst'"; \
 	  rm -f "$(DESTDIR)$(man1dir)/$$inst"; \
 	done
-install-manh: $(manh_MANS) $(man_MANS)
-	@$(NORMAL_INSTALL)
-	test -z "$(manhdir)" || $(mkdir_p) "$(DESTDIR)$(manhdir)"
-	@list='$(manh_MANS) $(dist_manh_MANS) $(nodist_manh_MANS)'; \
-	l2='$(man_MANS) $(dist_man_MANS) $(nodist_man_MANS)'; \
-	for i in $$l2; do \
-	  case "$$i" in \
-	    *.h*) list="$$list $$i" ;; \
-	  esac; \
-	done; \
-	for i in $$list; do \
-	  if test -f $(srcdir)/$$i; then file=$(srcdir)/$$i; \
-	  else file=$$i; fi; \
-	  ext=`echo $$i | sed -e 's/^.*\\.//'`; \
-	  case "$$ext" in \
-	    h*) ;; \
-	    *) ext='h' ;; \
-	  esac; \
-	  inst=`echo $$i | sed -e 's/\\.[0-9a-z]*$$//'`; \
-	  inst=`echo $$inst | sed -e 's/^.*\///'`; \
-	  inst=`echo $$inst | sed '$(transform)'`.$$ext; \
-	  echo " $(INSTALL_DATA) '$$file' '$(DESTDIR)$(manhdir)/$$inst'"; \
-	  $(INSTALL_DATA) "$$file" "$(DESTDIR)$(manhdir)/$$inst"; \
-	done
-uninstall-manh:
-	@$(NORMAL_UNINSTALL)
-	@list='$(manh_MANS) $(dist_manh_MANS) $(nodist_manh_MANS)'; \
-	l2='$(man_MANS) $(dist_man_MANS) $(nodist_man_MANS)'; \
-	for i in $$l2; do \
-	  case "$$i" in \
-	    *.h*) list="$$list $$i" ;; \
-	  esac; \
-	done; \
-	for i in $$list; do \
-	  ext=`echo $$i | sed -e 's/^.*\\.//'`; \
-	  case "$$ext" in \
-	    h*) ;; \
-	    *) ext='h' ;; \
-	  esac; \
-	  inst=`echo $$i | sed -e 's/\\.[0-9a-z]*$$//'`; \
-	  inst=`echo $$inst | sed -e 's/^.*\///'`; \
-	  inst=`echo $$inst | sed '$(transform)'`.$$ext; \
-	  echo " rm -f '$(DESTDIR)$(manhdir)/$$inst'"; \
-	  rm -f "$(DESTDIR)$(manhdir)/$$inst"; \
-	done
 install-docDATA: $(doc_DATA)
 	@$(NORMAL_INSTALL)
 	test -z "$(docdir)" || $(mkdir_p) "$(DESTDIR)$(docdir)"
@@ -314,7 +267,7 @@
 check: check-am
 all-am: makefile $(MANS) $(DATA)
 installdirs:
-	for dir in "$(DESTDIR)$(man1dir)" "$(DESTDIR)$(manhdir)" "$(DESTDIR)$(docdir)"; do \
+	for dir in "$(DESTDIR)$(man1dir)" "$(DESTDIR)$(docdir)"; do \
 	  test -z "$$dir" || $(mkdir_p) "$$dir"; \
 	done
 install: install-am
@@ -365,7 +318,7 @@
 
 install-info: install-info-am
 
-install-man: install-man1 install-manh
+install-man: install-man1
 
 installcheck-am:
 
@@ -387,27 +340,22 @@
 
 uninstall-am: uninstall-docDATA uninstall-info-am uninstall-man
 
-uninstall-man: uninstall-man1 uninstall-manh
+uninstall-man: uninstall-man1
 
 .PHONY: all all-am check check-am clean clean-generic distclean \
 	distclean-generic distdir dvi dvi-am html html-am info info-am \
 	install install-am install-data install-data-am \
 	install-docDATA install-exec install-exec-am install-info \
-	install-info-am install-man install-man1 install-manh \
+	install-info-am install-man install-man1 \
 	install-strip installcheck installcheck-am installdirs \
 	maintainer-clean maintainer-clean-generic mostlyclean \
 	mostlyclean-generic pdf pdf-am ps ps-am uninstall uninstall-am \
 	uninstall-docDATA uninstall-info-am uninstall-man \
-	uninstall-man1 uninstall-manh
+	uninstall-man1
 
 
 # doc_DATA = main.css 
 
-manual.html : manual.xml
-	if [ -x "`which xmlto`" ] ; then \
-		xmlto html-nochunks manual.xml ; \
-	fi ;
-
 #
 # End of automake
 #
diff -Nurd robodoc-4.99.26-orig/Docs/manual.css robodoc-4.99.26/Docs/manual.css
--- robodoc-4.99.26-orig/Docs/manual.css	1970-01-01 01:00:00.000000000 +0100
+++ robodoc-4.99.26/Docs/manual.css	2006-07-29 19:57:01.000000000 +0200
@@ -0,0 +1,57 @@
+body
+{
+    background-color: #ffffff;
+    color: #000000;
+    font-family: 'Lucida Grande', Verdana, 
+                     Geneva, Lucida, Arial, 
+                     Helvetica, sans-serif;
+    font-size: 10pt;
+    margin: 2% 5%;
+}
+h1.title
+{
+    color: #000000;
+    text-align: left;
+    font-size: 16pt;
+    margin-top: 10%;
+}
+
+h2.title
+{
+    color: #000000;
+    text-align: left;
+    font-size: 14pt;
+    margin-top: 3%;
+}
+h3.title
+{
+    color: #000000;
+    text-align: left;
+    font-size: 11pt;
+    margin-top: 3%;
+}
+
+h3.author
+{
+    color: #000000;
+    text-align: left;
+    font-size: 10pt;
+}
+td.even, td.uneven
+{
+    color: #000000;
+    font-size: 10pt;
+}
+td.even
+{
+    background-color: #eeeeee;
+}
+pre.programlisting, pre.literallayout
+{
+    background:#ffe;
+    font-family: monospace;
+    border:solid #aa9 2px;
+    margin:4px;
+    white-space: pre;
+}
+
diff -Nurd robodoc-4.99.26-orig/Docs/manual.html robodoc-4.99.26/Docs/manual.html
--- robodoc-4.99.26-orig/Docs/manual.html	2006-06-23 23:57:28.000000000 +0200
+++ robodoc-4.99.26/Docs/manual.html	2006-07-29 01:30:19.000000000 +0200
@@ -835,7 +835,7 @@
     <code class="filename">robodoc.rc</code> file. If it can't find
     it there it will search in <code class="literal">$HOME/</code>, then
     <code class="filename">$HOMEDRIVE$HOMEPATH/</code>, and finally in
-    <code class="filename">/usr/share/robodoc/</code>.</p><p>With the <code class="option">--rc</code> option you can tell ROBODoc to
+    <code class="filename">@PREFIX@/share/robodoc/</code>.</p><p>With the <code class="option">--rc</code> option you can tell ROBODoc to
     use a different file than <code class="filename">robodoc.rc</code> as
     configuration file.  This is handy if you want to create
     documentation in different formats.  For instance: </p><pre class="programlisting">
diff -Nurd robodoc-4.99.26-orig/Docs/robodoc.1 robodoc-4.99.26/Docs/robodoc.1
--- robodoc-4.99.26-orig/Docs/robodoc.1	2006-06-23 22:30:36.000000000 +0200
+++ robodoc-4.99.26/Docs/robodoc.1	2006-07-29 01:32:02.000000000 +0200
@@ -165,7 +165,7 @@
 .SS "Example headers"
 .PP
 .nf
-\fI/usr/share/robodoc/Examples\fR
+\fI@PREFIX@/share/robodoc/Examples\fR
 .fi
 
 .SH COPYRIGHT
diff -Nurd robodoc-4.99.26-orig/Source/roboconfig.c robodoc-4.99.26/Source/roboconfig.c
--- robodoc-4.99.26-orig/Source/roboconfig.c	2006-06-23 22:52:10.000000000 +0200
+++ robodoc-4.99.26/Source/roboconfig.c	2006-07-29 01:33:30.000000000 +0200
@@ -421,7 +421,7 @@
             }
             else
             {
-                char* sitespecific = "/usr/share/robodoc/";
+                char* sitespecific = "@prefix@/share/robodoc/";
 
                 if ( path ) {
                    free( path );
diff -Nurd robodoc-4.99.26-orig/makefile.in robodoc-4.99.26/makefile.in
--- robodoc-4.99.26-orig/makefile.in	2006-06-27 17:08:24.000000000 +0200
+++ robodoc-4.99.26/makefile.in	2006-07-29 20:23:34.000000000 +0200
@@ -158,13 +158,14 @@
 target_alias = @target_alias@
 AUTOMAKE_OPTIONS = dist-zip
 SUBDIRS = Source Docs
-docdir = $(prefix)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(prefix)/share/doc/$(PACKAGE)
 doc_DATA = AUTHORS COPYING ChangeLog INSTALL NEWS README
 EXTRA_DIST = \
     Docs/robodoc.1 \
     Docs/robohdrs.1 \
     Docs/robodoc.rc \
     Docs/manual.html \
+    Docs/manual.css \
     Contributions/robodoc2pod.pl \
     Contributions/robodoc.kaptn \
     Headers/assembler.sample \
