diff -ru sitecopy-0.16.6/Makefile.in sitecopy-0.16.6-patched/Makefile.in
--- sitecopy-0.16.6/Makefile.in	2008-07-07 22:55:29.000000000 +0200
+++ sitecopy-0.16.6-patched/Makefile.in	2011-05-12 08:40:04.000000000 +0200
@@ -27,7 +27,7 @@
 INCLUDES = -I$(top_srcdir)/src -I$(top_srcdir)/lib
 CPPFLAGS = @CPPFLAGS@ @DEFS@ -DLOCALEDIR=\"$(localedir)\"
 CFLAGS = @CFLAGS@
-ALL_CFLAGS = $(CPPFLAGS) $(INCLUDES) $(CFLAGS)
+ALL_CFLAGS = $(INCLUDES) $(CFLAGS) $(CPPFLAGS)
 LIBS = @NEON_LIBS@ @LIBINTL@ @LIBS@
 CC = @CC@
 
Nur in sitecopy-0.16.6-patched: Makefile.in.orig.
diff -ru sitecopy-0.16.6/configure sitecopy-0.16.6-patched/configure
--- sitecopy-0.16.6/configure	2008-07-16 23:15:52.000000000 +0200
+++ sitecopy-0.16.6-patched/configure	2011-05-12 08:40:51.000000000 +0200
@@ -8456,7 +8456,7 @@
 echo "${ECHO_T}$ne_cv_lib_neon" >&6; }
     if test "$ne_cv_lib_neon" = "yes"; then
        ne_cv_lib_neonver=no
-       for v in 24 25 26 27 28; do
+       for v in 24 25 26 27 28 29; do
           case $ne_libver in
           0.$v.*) ne_cv_lib_neonver=yes ;;
           esac
@@ -9174,8 +9174,8 @@
     fi
 
 else
-    { echo "$as_me:$LINENO: incompatible neon library version $ne_libver: wanted 0.24 25 26 27 28" >&5
-echo "$as_me: incompatible neon library version $ne_libver: wanted 0.24 25 26 27 28" >&6;}
+    { echo "$as_me:$LINENO: incompatible neon library version $ne_libver: wanted 0.24 25 26 27 28 29" >&5
+echo "$as_me: incompatible neon library version $ne_libver: wanted 0.24 25 26 27 28 29" >&6;}
     neon_got_library=no
 fi
 
@@ -9256,7 +9256,7 @@
 echo "${ECHO_T}$ne_cv_lib_neon" >&6; }
     if test "$ne_cv_lib_neon" = "yes"; then
        ne_cv_lib_neonver=no
-       for v in 24 25 26 27 28; do
+       for v in 24 25 26 27 28 29; do
           case $ne_libver in
           0.$v.*) ne_cv_lib_neonver=yes ;;
           esac
@@ -9974,8 +9974,8 @@
     fi
 
 else
-    { echo "$as_me:$LINENO: incompatible neon library version $ne_libver: wanted 0.24 25 26 27 28" >&5
-echo "$as_me: incompatible neon library version $ne_libver: wanted 0.24 25 26 27 28" >&6;}
+    { echo "$as_me:$LINENO: incompatible neon library version $ne_libver: wanted 0.24 25 26 27 28 29" >&5
+echo "$as_me: incompatible neon library version $ne_libver: wanted 0.24 25 26 27 28 29" >&6;}
     neon_got_library=no
 fi
 
diff -ru sitecopy-0.16.6/lib/neon/Makefile.in sitecopy-0.16.6-patched/lib/neon/Makefile.in
--- sitecopy-0.16.6/lib/neon/Makefile.in	2008-02-06 22:37:26.000000000 +0100
+++ sitecopy-0.16.6-patched/lib/neon/Makefile.in	2011-05-12 08:40:04.000000000 +0200
@@ -32,7 +32,7 @@
 # build of this directory, @LIBS@ may include -lneon.
 LIBS = @NEON_LIBS@ @NEON_LTLIBS@
 
-COMPILE = $(CC) $(CPPFLAGS) $(CFLAGS)
+COMPILE = $(CC) $(CFLAGS) $(CPPFLAGS)
 LINK = $(LIBTOOL) --quiet --mode=link $(CC) -no-undefined $(LDFLAGS)
 
 NEON_BASEOBJS = ne_request.@NEON_OBJEXT@ ne_session.@NEON_OBJEXT@ 	    \
