--- djvulibre-3.5.24/configure.orig	2011-03-06 12:43:17.000000000 -0500
+++ djvulibre-3.5.24/configure	2012-03-02 19:47:30.000000000 -0500
@@ -18820,7 +18820,7 @@
 
     NSDEJAVU_CFLAGS=${X_CFLAGS}
     case "$host" in
-        *-cygwin* | *-mingw*)
+        *-cygwin* | *-mingw* | *-darwin*)
             NSDEJAVU_LIBS="`echo ${X_LIBS} | sed -e 's/-lXext/-lXt -lXext/'`"
             NSDEJAVU_LIBS="-no-undefined $NSDEJAVU_LIBS"
             ;;
--- djvulibre-3.5.24/gui/nsdejavu/Makefile.in.orig	2011-03-06 12:43:17.000000000 -0500
+++ djvulibre-3.5.24/gui/nsdejavu/Makefile.in	2012-03-02 19:49:20.000000000 -0500
@@ -52,7 +52,7 @@
  -I${top_srcdir}/gui/shared/res \
  -I${top_srcdir}/gui/npsdk 
 
-CFLAGS = @CPPFLAGS@ ${DEFS} ${OPTS} ${INCS} ${NSDEJAVU_CFLAGS} ${GLIB_CFLAGS}
+CFLAGS = ${DEFS} ${OPTS} ${INCS} ${NSDEJAVU_CFLAGS} @CPPFLAGS@ ${GLIB_CFLAGS}
 
 LIBS = @LDFLAGS@ ${NSDEJAVU_LIBS} 
 
diff -Nurd -x'*~' djvulibre-3.5.21.orig/tools/jb2cmp/Makefile.in djvulibre-3.5.21/tools/jb2cmp/Makefile.in
--- djvulibre-3.5.21.orig/tools/jb2cmp/Makefile.in	2007-11-28 18:55:29.000000000 -0500
+++ djvulibre-3.5.21/tools/jb2cmp/Makefile.in	2008-11-12 09:05:28.000000000 -0500
@@ -49,8 +49,8 @@
 INCS = -I${top_builddir} -I${top_srcdir} -I${srcdir}
 FLAGS = ${DEFS} ${INCS} ${OPTS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${THREAD_LIBS} @LIBS@
-CFLAGS = @CPPFLAGS@ @CFLAGS@ ${FLAGS}
-CXXFLAGS = @CPPFLAGS@ @CXXFLAGS@ ${CXXRPOFLAGS} ${FLAGS}
+CFLAGS = ${FLAGS} @CPPFLAGS@ @CFLAGS@
+CXXFLAGS = ${FLAGS} @CPPFLAGS@ @CXXFLAGS@ ${CXXRPOFLAGS}
 
 
 OBJS = classify.o cuts.o frames.o patterns.o
--- djvulibre-3.5.24/gui/indep/utils/execdir.cpp.orig	2012-03-02 20:13:00.000000000 -0500
+++ djvulibre-3.5.24/gui/indep/utils/execdir.cpp	2012-03-02 20:14:34.000000000 -0500
@@ -141,22 +141,22 @@
     {
       GURL url = GURL(file, paths[pos]);
       GUTF8String urls = (const char*)url;
-      int pos = urls.search(osi);
-      if (pos >= 0)
+      int ipos = urls.search(osi);
+      if (ipos >= 0)
         {
           GUTF8String urlx;
-          urlx += urls.substr(0, pos);
+          urlx += urls.substr(0, ipos);
           urlx += djview3;
-          urlx += urls.substr(pos+strlen(osi), -1);
+          urlx += urls.substr(ipos+strlen(osi), -1);
           GURL url = GURL::UTF8(urlx);
           if (url.is_file())
             return url;
         }
     }
   // end hack alert
-  for (GPosition pos=paths; pos; ++pos)
+  for (GPosition ipos=paths; ipos; ++ipos)
     {
-      GURL url = GURL(file, paths[pos]);
+      GURL url = GURL(file, paths[ipos]);
       if(url.is_file())
         return url;
     }
--- djvulibre-3.5.24/libdjvu/GContainer.h.orig	2012-03-02 20:24:55.000000000 -0500
+++ djvulibre-3.5.24/libdjvu/GContainer.h	2012-03-02 20:27:59.000000000 -0500
@@ -962,7 +962,7 @@
   /** Inserts an element before the first element of the list. 
       The new element is initialized with a copy of argument #elt#. */
   void prepend(const TYPE &elt)
-    { GListImpl<TI>::prepend(newnode((const TI&)elt)); }
+    { GListImpl<TI>::prepend(this->newnode((const TI&)elt)); }
   /** Inserts a new element after the list element at position #pos#.  When
       position #pos# is null the element is inserted at the beginning of the
       list.  The new element is initialized with a copy of #elt#. */
--- djvulibre-3.5.24/gui/nsdejavu/Makefile.in.orig	2012-03-02 20:49:11.000000000 -0500
+++ djvulibre-3.5.24/gui/nsdejavu/Makefile.in	2012-03-02 20:51:34.000000000 -0500
@@ -54,7 +54,7 @@
 
 CFLAGS = ${DEFS} ${OPTS} ${INCS} ${NSDEJAVU_CFLAGS} @CPPFLAGS@ ${GLIB_CFLAGS}
 
-LIBS = @LDFLAGS@ ${NSDEJAVU_LIBS} 
+LIBS = @LDFLAGS@ ${NSDEJAVU_LIBS} -lglib-2.0
 
 all: nsdejavu.la nsdejavu.x nsdejavu.1
 
