diff -ru doclifter-2.1/doclifter doclifter-2.1-patched/doclifter
--- doclifter-2.1/doclifter	2005-01-14 18:50:42.000000000 +0100
+++ doclifter-2.1-patched/doclifter	2007-02-28 14:15:56.000000000 +0100
@@ -125,7 +125,7 @@
 $Id: doclifter,v 1.1294 2005/01/14 06:35:12 esr Exp $
 """
 # Requires Python 2.2a
-import sys, os, glob, re, sre, string, exceptions, copy, tempfile, time, pprint
+import sys, os, glob, re, string, exceptions, copy, tempfile, time, pprint
 
 # This is a speed hack recommended by Armin Rigo.  It cuts runtime by about 33%
 # and makes it possible for psyco 1.2 to reduce runtime another 33%.
@@ -6301,8 +6301,8 @@
         return result
     def styleargs(self, args, tag, attribute="", prefix=""):
         return self.process_punct(args, lambda x: self.stylehook(x, tag, attribute, prefix), 1)
-    def replacemacro(self, args, with):
-        return self.process_punct(args, lambda x: [with] + x, 1)
+    def replacemacro(self, args, replacement):
+        return self.process_punct(args, lambda x: [replacement] + x, 1)
     def eval(self, args):
         "Evaluate a macro, returning a list."
         if args[0][0] == '.':
