diff -uNr ghostscript-9.10/base/gxpcolor.h ghostscript-9.10.fink/base/gxpcolor.h
--- ghostscript-9.10/base/gxpcolor.h	2013-08-30 06:37:28.000000000 -0400
+++ ghostscript-9.10.fink/base/gxpcolor.h	2013-09-10 15:51:36.000000000 -0400
@@ -20,6 +20,7 @@
 #ifndef gxpcolor_INCLUDED
 #  define gxpcolor_INCLUDED
 
+#include "gxdcolor.h"
 #include "gspcolor.h"
 #include "gxcspace.h"
 #include "gxdevice.h"
diff -uNr ghostscript-9.10/base/mkromfs.c ghostscript-9.10.fink/base/mkromfs.c
--- ghostscript-9.10/base/mkromfs.c	2013-08-30 06:37:28.000000000 -0400
+++ ghostscript-9.10.fink/base/mkromfs.c	2013-09-10 15:51:36.000000000 -0400
@@ -2302,7 +2302,7 @@
 #endif
     fprintf(out,"\n#include \"stdint_.h\"\n");
     fprintf(out,"\n#include \"time_.h\"\n\n");
-    fprintf(out,"    time_t gs_romfs_buildtime = %ld;\n\n", time(NULL));
+    fprintf(out,"    time_t gs_romfs_buildtime = %lld;\n\n", (long long)time(NULL));
 
     /* process the remaining arguments (options interspersed with paths) */
     for (; atarg < argc; atarg++) {
diff -uNr ghostscript-9.10/configure ghostscript-9.10.fink/configure
--- ghostscript-9.10/configure	2013-08-30 06:37:28.000000000 -0400
+++ ghostscript-9.10.fink/configure	2013-09-10 15:51:36.000000000 -0400
@@ -841,6 +841,7 @@
 LIBS
 CPPFLAGS
 CPP
+CPPFLAGS
 XMKMF'
 
 
diff -uNr ghostscript-9.10/devices/gdevpng.c ghostscript-9.10.fink/devices/gdevpng.c
--- ghostscript-9.10/devices/gdevpng.c	2013-08-30 06:37:28.000000000 -0400
+++ ghostscript-9.10.fink/devices/gdevpng.c	2013-09-10 15:51:36.000000000 -0400
@@ -716,6 +716,7 @@
 
 #if PNG_LIBPNG_VER_MINOR < 5
 
+#if 0 /* not required in 1.5 */
 /*
  * Patch around a static reference to a never-used procedure.
  * This could be avoided if we were willing to edit pngconf.h to
@@ -736,6 +737,7 @@
 }
 #endif
 #endif
+#endif
 
 static int
 pngalpha_open(gx_device * pdev)
diff -uNr ghostscript-9.10/psi/imainarg.c ghostscript-9.10.fink/psi/imainarg.c
--- ghostscript-9.10/psi/imainarg.c	2013-08-30 06:37:28.000000000 -0400
+++ ghostscript-9.10.fink/psi/imainarg.c	2013-09-10 15:51:36.000000000 -0400
@@ -212,6 +212,11 @@
 #endif
         }
     }
+
+	code = gs_main_init1(minst);
+	if (code < 0)
+		return code;
+
     while ((arg = arg_next(&args, &code, minst->heap)) != 0) {
         switch (*arg) {
             case '-':
