--- gv/src/file.c.orig	Sun Feb  6 12:25:51 2005
+++ gv/src/file.c	Sun Feb  6 12:26:06 2005
@@ -296,7 +296,7 @@
     errno = EISDIR;
   } else if (s.st_size == 0) {
     r = 1;
-    errno = ENODATA;
+    errno = 61;
   }
   IMESSAGE(r)
   ENDMESSAGE(file_fileIsNotUseful)
--- gv/src/Makefile.in.orig	Fri Feb 11 05:37:27 2005
+++ gv/src/Makefile.in	Fri Feb 11 05:42:31 2005
@@ -234,14 +234,14 @@
 target_alias = @target_alias@
 dist_pkglib_DATA = gv_system.ad gv_user.ad gv_class.ad gv_spartan.dat gv_user_res.dat gv_copyright.dat
 appdefaultsdir = $(pkglibdir)
-appdefaults_DATA = GV
+appdefaults_DATA = GV.addata
 EXTRA_DIST = ad2c gv_font_res.dat gv_layout_res.dat gv_misc_res.dat Aaa_bison.yacc \
            gv_current.xbm gv_doc.xbm gv_empty.xbm gv_even.xbm gv_icon.xbm gv_odd.xbm \
            gv_selected.xbm gv_unmark.xbm strcasecmp.c strcasecmp.h vms_dir.h vms_types.h 
 
 
 # Clean files
-CLEANFILES = $(srcdir)/GV \
+CLEANFILES = $(srcdir)/GV.addata \
              $(srcdir)/GV.ad \
              $(srcdir)/gv_system.ad \
              $(srcdir)/gv_user.ad \
@@ -726,9 +726,9 @@
 
 # Additional rules for generating styles
 
-GV : $(srcdir)/GV.ad
-	@echo "creating GV"
-	@cp $(srcdir)/GV.ad $(srcdir)/GV
+GV.addata : $(srcdir)/GV.ad
+	@echo "creating GV.addata"
+	@cp $(srcdir)/GV.ad $(srcdir)/GV.addata
 
 $(srcdir)/GV.ad : $(srcdir)/gv_system.ad
 	@echo "creating GV.ad"
