diff -ruN gv-3.6.2/src/Makefile.in gv-3.6.2-patched/src/Makefile.in
--- gv-3.6.2/src/Makefile.in	2006-09-14 12:48:06.000000000 -0700
+++ gv-3.6.2-patched/src/Makefile.in	2007-05-01 21:12:57.000000000 -0700
@@ -223,14 +223,14 @@
 top_srcdir = @top_srcdir@
 dist_pkglib_DATA = gv_system.ad gv_user.ad gv_class.ad gv_spartan.dat gv_user_res.dat gv_copyright.dat gv_widgetless.dat
 appdefaultsdir = $(pkglibdir)
-appdefaults_DATA = GV
+appdefaults_DATA = GV.addata
 EXTRA_DIST = ad2c gv_font_res.dat gv_layout_res.dat gv_misc_res.dat Aaa_bison.yacc \
            gv_current.xbm gv_doc.xbm gv_empty.xbm gv_even.xbm gv_icon.xbm gv_odd.xbm \
            gv_selected.xbm gv_unmark.xbm
 
 
 # Clean files
-CLEANFILES = $(srcdir)/GV \
+CLEANFILES = $(srcdir)/GV.addata \
              $(srcdir)/GV.ad \
              $(srcdir)/gv_system.ad \
              $(srcdir)/gv_user.ad \
@@ -624,9 +624,9 @@
 
 # Additional rules for generating styles
 
-GV : $(srcdir)/GV.ad
-	@echo "creating GV"
-	sed -e 's/^[^!]/!&/' $(srcdir)/GV.ad > $(srcdir)/GV
+GV.addata : $(srcdir)/GV.ad
+	@echo "creating GV.addata"
+	sed -e 's/^[^!]/!&/' $(srcdir)/GV.ad > $(srcdir)/GV.addata
 
 $(srcdir)/GV.ad : $(srcdir)/gv_system.ad
 	@echo "creating GV.ad"
--- gv-3.6.5-orig/src/gv_misc_res.dat	2008-07-05 05:57:14.000000000 -0400
+++ gv-3.6.5/src/gv_misc_res.dat	2011-08-09 12:19:21.000000000 -0400
@@ -468,7 +468,6 @@
  <Key>BackSpace:	GV_Page(up,left-bottom,!page-1,bottom-rightedge)\n\
  <Key>Insert:		GV_Page(page-5)		\n\
  <Key>Delete:		GV_Page(page+5)		\n\
- <Key>apLineDel:	GV_Page(page+5)		\n\
  <Key>Home:		GV_Page(page=0)		\n\
  <Key>End:		GV_Page(page=99999)	\n\
  <Key>Prior:		GV_Page(page-1)		\n\
