diff --git a/lib/stdint.mini.h b/lib/stdint.mini.h
index d6f2cb0..3c0acc8 100644
--- a/lib/stdint.mini.h
+++ b/lib/stdint.mini.h
@@ -118,11 +118,6 @@ typedef unsigned int unistring_uint32_t;
 #define int32_t unistring_int32_t
 #define uint32_t unistring_uint32_t
 
-/* Avoid collision with Solaris 2.5.1 <pthread.h> etc.  */
-#define _UINT8_T
-#define _UINT32_T
-#define _UINT64_T
-
 
 #endif /* _UNISTRING_STDINT_H */
 #endif /* !defined _UNISTRING_STDINT_H && !defined _GL_JUST_INCLUDE_SYSTEM_STDINT_H */
