diff -ru lilypond-2.11.45-orig/Documentation/user/lilypond.tely lilypond-2.11.45/Documentation/user/lilypond.tely
--- lilypond-2.11.45-orig/Documentation/user/lilypond.tely	2008-05-03 00:50:50.000000000 +0200
+++ lilypond-2.11.45/Documentation/user/lilypond.tely	2008-05-03 00:51:20.000000000 +0200
@@ -34,7 +34,7 @@
 
   * List all commands in direntry.
 
-@c  * lilypond: (lilypond/lilypond)Running LilyPond.      Invoking the
+@c  * lilypond: (lilypond)Running LilyPond.      Invoking the
 @c    LilyPond  program.
 @end ignore
 
diff -ru lilypond-2.11.45-orig/Documentation/user/music-glossary.tely lilypond-2.11.45/Documentation/user/music-glossary.tely
--- lilypond-2.11.45-orig/Documentation/user/music-glossary.tely	2008-05-03 00:50:50.000000000 +0200
+++ lilypond-2.11.45/Documentation/user/music-glossary.tely	2008-05-03 00:54:33.000000000 +0200
@@ -10,7 +10,7 @@
 @c see lilypond.tely for info installation note
 @dircategory LilyPond
 @direntry
-* Glossary: (lilypond/music-glossary).  Glossary of music terms.
+* Glossary: (music-glossary).  Glossary of music terms.
 @end direntry
 
 @ifhtml
diff -ru lilypond-2.11.45-orig/config.make.in lilypond-2.11.45/config.make.in
--- lilypond-2.11.45-orig/config.make.in	2008-05-03 00:50:50.000000000 +0200
+++ lilypond-2.11.45/config.make.in	2008-05-03 00:51:20.000000000 +0200
@@ -28,7 +28,7 @@
 FREETYPE2_LIBS = @FREETYPE2_LIBS@
 PANGO_FT2_LIBS = @PANGO_FT2_LIBS@
 
-CONFIG_LIBS = @LIBS@ @EXTRA_LIBS@ $(GUILE_LIBS) $(PANGO_FT2_LIBS) $(FONTCONFIG_LIBS) $(FREETYPE2_LIBS)
+CONFIG_LIBS = @LIBS@ @EXTRA_LIBS@ $(PANGO_FT2_LIBS) $(FONTCONFIG_LIBS) $(FREETYPE2_LIBS) $(GUILE_LIBS)
 CONFIG_LDFLAGS = @LDFLAGS@
 
 PACKAGE = @PACKAGE@
@@ -66,7 +66,7 @@
 stepmake = @stepmake@
 
 package_datadir = $(datadir)/$(package)
-package_infodir = $(infodir)/$(package)
+package_infodir = $(infodir)
 package_libdir = $(libdir)/$(package)
 package_docdir = $(docdir)/$(package)
 package_omfdir = $(omfdir)/$(package)
