diff -Nurd -x'*~' talkfilters-2.3.8.orig/talkfilters.info talkfilters-2.3.8/talkfilters.info
--- talkfilters-2.3.8.orig/talkfilters.info	2005-06-27 00:59:03.000000000 -0400
+++ talkfilters-2.3.8/talkfilters.info	2012-02-13 23:04:26.000000000 -0500
@@ -1,4 +1,4 @@
-This is talkfilters.info, produced by makeinfo version 4.8 from
+This file documents `talkfilters', produced by makeinfo version 4.8 from
 talkfilters.texi.
 
 
