--- unrtf-0.19.3/Makefile	Thu Feb 19 00:35:03 2004
+++ unrtf-patched-0.19.3/Makefile	Thu Nov 18 12:32:56 2004
@@ -26,7 +26,7 @@
 
 
 CFLAGS=-g -Wall
-CC=gcc
+CC=cc
 
 OBJS=convert.o word.o error.o main.o hash.o \
 	parse.o malloc.o attr.o util.o \
@@ -35,7 +35,7 @@
 TARGET=unrtf
 
 ${TARGET}:	${OBJS}
-	gcc ${OBJS} -o ${TARGET}
+	cc ${OBJS} -o ${TARGET}
 
 clean:
 	rm -f ${OBJS} ${TARGET} bcount bcount.o
@@ -51,7 +51,7 @@
 	${TARGET} test/fontsize.rtf > test/fontsize.html
 
 bcount:	bcount.c
-	gcc bcount.c -o bcount
+	cc bcount.c -o bcount
 
 all:	${TARGET} bcount
 	rm -f ${OBJS}

--- unrtf-0.19.3/error.c	Thu Feb 19 00:35:03 2004
+++ unrtf-patched-0.19.3/error.c	Thu Nov 18 12:22:39 2004
@@ -38,9 +38,7 @@
 
 
 #include <stdio.h>
-#if linux /* daved - 0.19.1 */
 #include <stdlib.h>
-#endif
 
 #include "defs.h"
 #include "main.h"

--- unrtf-0.19.3/main.c	Thu Feb 19 00:35:03 2004
+++ unrtf-patched-0.19.3/main.c	Thu Nov 18 12:28:20 2004
@@ -43,9 +43,7 @@
 
 #include <stdio.h>
 #include <string.h>
-#if linux /* daved - 0.19.0 */
 #include <stdlib.h>
-#endif
 
 #include "defs.h"
 #include "error.h"

--- unrtf-0.19.3/malloc.c	Thu Nov 18 20:30:10 2004
+++ unrtf-patched-0.19.3/malloc.c	Thu Nov 18 21:04:17 2004
@@ -39,16 +39,7 @@
 
 #include <stdio.h>
 #include <string.h>
-#if linux /* daved - 0.19.0 */
 #include <stdlib.h>
-#endif
-
-#if AMIGA
-#include <stdlib.h>
-#else
-#include <malloc.h>
-#endif
-
 #include "error.h"
 
 

--- unrtf-0.19.3/output.c	Thu Feb 19 00:35:03 2004
+++ unrtf-patched-0.19.3/output.c	Thu Nov 18 12:33:35 2004
@@ -36,9 +36,7 @@
 
 #include <stdio.h>
 #include <string.h>
-#if linux /* daved - 0.19.0 */
 #include <stdlib.h>
-#endif
 #include "malloc.h"
 #include "defs.h"
 #include "error.h"
