
--- libpq++-4.0/Makefile	Tue Nov 26 23:46:08 2002
+++ libpq++-4.0-patched/Makefile	Mon Dec 30 21:29:40 2002
@@ -11,33 +11,35 @@
 SO_MAJOR_VERSION= 4
 SO_MINOR_VERSION= 0
 
-POSTGRES_HOME=/usr/local/pgsql
-CXXOPTS= -DHAVE_NAMESPACE_STD -DHAVE_CXX_STRING_HEADER -DDLLIMPORT=""
+POSTGRES_HOME=@INSTPREFIX@
+CXXOPTS= -DHAVE_NAMESPACE_STD -DHAVE_CXX_STRING_HEADER -DDLLIMPORT="" -Wall -Wmissing-prototypes -Wmissing-declarations -fno-common -traditional-cpp
 
 TARGET= libpq++
-CFLAGS= -O -Wall
+CFLAGS= -Os -Wall
 LIBS=pq
 OBJS= pgconnection.o pgdatabase.o pgtransdb.o pgcursordb.o pglobject.o
 LIBPGXXHEADERS = pgconnection.h pgdatabase.h pgtransdb.h pgcursordb.h pglobject.h
 
-SHARED= -fpic -shared -Wl,-x,-soname,$(TARGET).so.$(SO_MAJOR_VERSION) -Wl,-rpath,$(POSTGRES_HOME)/lib
-soname=$(TARGET).so.$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)
+SHARED= -install_name $(POSTGRES_HOME)/lib/$(TARGET).$(SO_MAJOR_VERSION).dylib -compatibility_version $(SO_MAJOR_VERSION) -current_version $(SO_MAJOR_VERSION).$(SO_MINOR_VERSION) -dynamiclib -multiply_defined suppress
+
+soname=$(TARGET).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION).dylib
 
 all:  $(soname)
 
 $(soname) : $(OBJS)
-	$(CXX) $(CXXFLAGS) $(OBJS) -L$(POSTGRES_HOME)/lib -l$(LIBS) $(SHARED) -o $(soname)
+	$(CXX) $(CXXFLAGS) $(OBJS) -L$(POSTGRES_HOME)/lib -l$(LIBS) $(SHARED) -o $(soname)
 
 %.o	: %.cc
-	$(CXX) -c $(CXXFLAGS) $(CXXOPTS) -I$(POSTGRES_HOME)/include $<
+	$(CXX) -c $(CXXFLAGS) $(CXXOPTS) -I$(POSTGRES_HOME)/include/postgresql -I$(POSTGRES_HOME)/include $<
 
 clean:
 	rm -f $(OBJS) $(soname)
 
 install:
-	install $(soname) $(POSTGRES_HOME)/lib
-	rm -f $(POSTGRES_HOME)/lib/$(TARGET).so
-	ln -s $(POSTGRES_HOME)/lib/$(soname) $(POSTGRES_HOME)/lib/$(TARGET).so
-	install -c $(TARGET).h $(POSTGRES_HOME)/include
-	-mkdir $(POSTGRES_HOME)/include/libpq++
-	for i in $(LIBPGXXHEADERS); do install -c $$i $(POSTGRES_HOME)/include/libpq++ || exit 1; done
+	install $(soname) $(DESTDIR)$(POSTGRES_HOME)/lib
+	rm -f $(DESTDIR)$(POSTGRES_HOME)/lib/$(TARGET).dylib
+	ln -sf $(soname) $(DESTDIR)$(POSTGRES_HOME)/lib/$(TARGET).dylib
+	ln -sf $(soname) $(DESTDIR)$(POSTGRES_HOME)/lib/$(TARGET).$(SO_MAJOR_VERSION).dylib
+	install -c -m 644 $(TARGET).h $(DESTDIR)$(POSTGRES_HOME)/include/postgresql
+	install -d -m 755 $(DESTDIR)$(POSTGRES_HOME)/include/postgresql/libpq++
+	for i in $(LIBPGXXHEADERS); do install -c $$i $(DESTDIR)$(POSTGRES_HOME)/include/postgresql/libpq++ || exit 1; done
