
--- libpqxx-2.2.3/configure	Mon May 10 08:37:31 2004
+++ libpqxx-2.2.3-new/configure	Sun May 16 13:09:30 2004
@@ -19832,7 +19832,8 @@
 { echo "$as_me:$LINENO: using PostgreSQL libraries at $with_postgres_lib" >&5
 echo "$as_me: using PostgreSQL libraries at $with_postgres_lib" >&6;}
 
-
+with_postgres_include="@INSTPREFIX@/include/postgresql"
+with_postgres_lib="@INSTPREFIX@/lib"
 
 POSTGRES_LIB="-L${with_postgres_lib} -R${with_postgres_lib} -lpq"
 
