diff -Naur gnome-desktop-2.0.10.old/libgnome-desktop/gnome-theme-parser.c gnome-desktop-2.0.10.new/libgnome-desktop/gnome-theme-parser.c
--- gnome-desktop-2.0.10.old/libgnome-desktop/gnome-theme-parser.c	Fri May 17 03:12:46 2002
+++ gnome-desktop-2.0.10.new/libgnome-desktop/gnome-theme-parser.c	Thu Feb 27 15:31:13 2003
@@ -763,7 +763,11 @@
 {
   char *p, *lang;
 
+#ifdef LC_MESSAGES
   lang = g_strdup (setlocale (LC_MESSAGES, NULL));
+#else
+  lang = g_strdup (setlocale (LC_ALL, NULL));
+#endif
   
   if (lang)
     {
diff -Naur gnome-desktop-2.0.10.old/ltmain.sh gnome-desktop-2.0.10.new/ltmain.sh
--- gnome-desktop-2.0.10.old/ltmain.sh	Fri Nov 15 12:27:26 2002
+++ gnome-desktop-2.0.10.new/ltmain.sh	Thu Feb 27 15:31:34 2003
@@ -3948,10 +3948,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test $need_relink = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test $need_relink = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
