diff -ruN gnome-apt-0.4.1.orig/src/index.cc gnome-apt-0.4.1/src/index.cc
--- gnome-apt-0.4.1.orig/src/index.cc	Sat May  3 09:57:02 2003
+++ gnome-apt-0.4.1/src/index.cc	Fri May 16 10:04:33 2003
@@ -54,7 +54,7 @@
 	GtkCellRenderer* renderer;
 	GtkTreeSelection* select;
 	GtkTreeViewColumn* column;
-	GtkWidget* tree;
+	GtkWidget* trees;
 
   g_return_if_fail(cache_ != 0);
   g_return_if_fail(filter_ != 0);
@@ -81,7 +81,7 @@
   gtk_container_add(GTK_CONTAINER(dialog_), vbox);
 
 	rlist = gtk_list_store_new (N_COLS, G_TYPE_STRING, G_TYPE_POINTER);
-	tree = gtk_tree_view_new_with_model (GTK_TREE_MODEL (rlist));
+	trees = gtk_tree_view_new_with_model (GTK_TREE_MODEL (rlist));
 	gtk_tree_view_set_headers_visible (GTK_TREE_VIEW (tree), FALSE);
 	renderer = gtk_cell_renderer_text_new();
 	column = gtk_tree_view_column_new_with_attributes ("Package", renderer, "text", C_NAME, NULL);
@@ -95,7 +95,7 @@
   gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(sw),
                                  GTK_POLICY_NEVER,
                                  GTK_POLICY_AUTOMATIC);
-	gtk_container_add (GTK_CONTAINER (sw), tree);
+	gtk_container_add (GTK_CONTAINER (sw), trees);
 
   gtk_box_pack_start(GTK_BOX(vbox),
                      sw,
