diff -Naur gtop-1.0.13.old/config.h.in gtop-1.0.13.new/config.h.in
--- gtop-1.0.13.old/config.h.in	Thu Mar 15 06:23:35 2001
+++ gtop-1.0.13.new/config.h.in	Wed Oct 16 01:09:48 2002
@@ -86,7 +86,7 @@
 #undef NEED_LIBGTOP
 
 /* Define if LibGTop has support for multiple processors. */
-#undef HAVE_LIBGTOP_SMP
+#define HAVE_LIBGTOP_SMP
 
 /* Define if you have the LibGTop Documentation. */
 #undef HAVE_LIBGTOP_DOCU
diff -Naur gtop-1.0.13.old/fsusage.c gtop-1.0.13.new/fsusage.c
--- gtop-1.0.13.old/fsusage.c	Tue Feb 27 23:41:28 2001
+++ gtop-1.0.13.new/fsusage.c	Wed Oct 16 02:14:04 2002
@@ -153,17 +153,17 @@
 		case GTOP_FSUSAGE_TOTAL:
 			d->graph_head = _("Total Filesystem Sizes "
 					  "(including reserved blocks)");
-			d->graph_tail = _("Sum of Total Sizes: %ldk");
+			d->graph_tail = _("Sum of Total Sizes: %lldk");
 			break;
 		case GTOP_FSUSAGE_USED:
 			d->graph_head = _("Used Space on Filesystems "
 					  "(including reserved blocks)");
-			d->graph_tail = _("Sum of Uses Space: %ldk");
+			d->graph_tail = _("Sum of Uses Space: %lldk");
 			break;
 		case GTOP_FSUSAGE_FREE:
 			d->graph_head = _("Free Space on Filesystems "
 					  "(including reserved blocks)");
-			d->graph_tail = _("Sum of Free Space: %ldk");
+			d->graph_tail = _("Sum of Free Space: %lldk");
 			break;
 		}
 		break;
@@ -171,15 +171,15 @@
 		switch (d->ftype) {
 		case GTOP_FSUSAGE_TOTAL:
 			d->graph_head = _("Total Filesystem Sizes");
-			d->graph_tail = _("Sum of Total Sizes: %ldk");
+			d->graph_tail = _("Sum of Total Sizes: %lldk");
 			break;
 		case GTOP_FSUSAGE_USED:
 			d->graph_head = _("Used Space on Filesystems");
-			d->graph_tail = _("Sum of Uses Space: %ldk");
+			d->graph_tail = _("Sum of Uses Space: %lldk");
 			break;
 		case GTOP_FSUSAGE_FREE:
 			d->graph_head = _("Free Space on Filesystems");
-			d->graph_tail = _("Sum of Free Space: %ldk");
+			d->graph_tail = _("Sum of Free Space: %lldk");
 			break;
 		}
 		break;
@@ -187,15 +187,15 @@
 		switch (d->ftype) {
 		case GTOP_FSUSAGE_TOTAL:
 			d->graph_head = _("Total Number of Inodes");
-			d->graph_tail = _("Sum of Total Number of Inodes: %ld");
+			d->graph_tail = _("Sum of Total Number of Inodes: %lld");
 			break;
 		case GTOP_FSUSAGE_USED:
 			d->graph_head = _("Used Inodes on Filesystems");
-			d->graph_tail = _("Sum of Used Inodes: %ld");
+			d->graph_tail = _("Sum of Used Inodes: %lld");
 			break;
 		case GTOP_FSUSAGE_FREE:
 			d->graph_head = _("Free Inodes on Filesystems");
-			d->graph_tail = _("Sum of Free Inodes: %ld");
+			d->graph_tail = _("Sum of Free Inodes: %lld");
 			break;
 		}
 		break;
diff -Naur gtop-1.0.13.old/memusage.c gtop-1.0.13.new/memusage.c
--- gtop-1.0.13.old/memusage.c	Thu Feb 17 02:28:02 2000
+++ gtop-1.0.13.new/memusage.c	Wed Oct 16 02:14:04 2002
@@ -131,23 +131,23 @@
 	switch (d->ftype) {
 	case GTOP_MEMUSAGE_RESIDENT:
 		d->graph_head = _("Resident Sizes of Processes");
-		d->graph_tail = _("Sum of Resident Sizes: %ldk");
+		d->graph_tail = _("Sum of Resident Sizes: %lldk");
 		break;
 	case GTOP_MEMUSAGE_SHARED:
 		d->graph_head = _("Shared Sizes of Processes");
-		d->graph_tail = _("Sum of Shared Sizes: %ldk");
+		d->graph_tail = _("Sum of Shared Sizes: %lldk");
 		break;
 	case GTOP_MEMUSAGE_SIZE:
 		d->graph_head = _("Total Sizes of Processes");
-		d->graph_tail = _("Sum of Total Sizes: %ldk");
+		d->graph_tail = _("Sum of Total Sizes: %lldk");
 		break;
 	case GTOP_MEMUSAGE_VIRTUAL:
 		d->graph_head = _("Virtual Sizes of Processes");
-		d->graph_tail = _("Sum of Virtual Sizes: %ldk");
+		d->graph_tail = _("Sum of Virtual Sizes: %lldk");
 		break;
 	case GTOP_MEMUSAGE_SWAP:
 		d->graph_head = _("Swapped Sizes of Processes");
-		d->graph_tail = _("Sum of Swapped Sizes: %ldk");
+		d->graph_tail = _("Sum of Swapped Sizes: %lldk");
 		break;
 	}
 }
