diff -Naur oaf-0.6.10.old/configure oaf-0.6.10.new/configure
--- oaf-0.6.10.old/configure	Tue Apr 16 19:27:55 2002
+++ oaf-0.6.10.new/configure	Tue Jun  4 22:35:25 2002
@@ -4374,11 +4374,11 @@
     ;;
 
   darwin* | rhapsody*)
-    allow_undefined_flag='-undefined suppress'
+    allow_undefined_flag='-flat_namespace -undefined suppress'
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linkopts -install_name $rpath/$soname $(test -n "$verstring" -a x$verstring != x0.0 && echo $verstring)'
+    archive_cmds='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linkopts -install_name $rpath/$soname $tmp_verstring'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -Naur oaf-0.6.10.old/ltmain.sh oaf-0.6.10.new/ltmain.sh
--- oaf-0.6.10.old/ltmain.sh	Tue Apr 16 19:27:41 2002
+++ oaf-0.6.10.new/ltmain.sh	Tue Jun  4 22:58:40 2002
@@ -2888,6 +2888,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+	  else
+	    tmp_verstring="$verstring"
+	  fi
 	  eval cmds=\"$archive_cmds\"
 	fi
 	IFS="${IFS= 	}"; save_ifs="$IFS"; IFS='~'
diff -Naur oaf-0.6.10.old/oaf-config.in oaf-0.6.10.new/oaf-config.in
--- oaf-0.6.10.old/oaf-config.in	Mon Apr 15 19:37:09 2002
+++ oaf-0.6.10.new/oaf-config.in	Mon May  6 13:14:28 2002
@@ -83,7 +83,7 @@
     shift
 done
 
-the_libs="$the_libs -L$libdir -loaf `@ORBIT_CONFIG@ --use-service=name --libs server`"
+the_libs="$the_libs -L$libdir -loaf -lgnorba `@ORBIT_CONFIG@ --use-service=name --libs server`"
 
 if test $includedir != /usr/include ; then
   the_flags="$the_flags -I$includedir `@ORBIT_CONFIG@ --cflags server`"
diff -Naur oaf-0.6.10.old/test/oaf-slay.in oaf-0.6.10.new/test/oaf-slay.in
--- oaf-0.6.10.old/test/oaf-slay.in	Thu Jan 17 00:10:35 2002
+++ oaf-0.6.10.new/test/oaf-slay.in	Mon May  6 13:15:09 2002
@@ -46,8 +46,8 @@
 $username = $ENV{USER} || $ENV{LOGNAME} || `logname`;
 chomp($username);
 
-@IF_POSIX_PS@ $ps_cmd = "@PS_PATH@ -U $username -opid,args";
-@IF_BSD_PS@   $ps_cmd = "@PS_PATH@ -U $username -xww -opid,command";
+@IF_POSIX_PS@ $ps_cmd = "@PS_PATH@ -U$username -opid,args";
+@IF_BSD_PS@   $ps_cmd = "@PS_PATH@ -U$username -xww -opid,command";
 
 # get OAF files
 #
