diff -ruN gimp-1.2.3/app/paint_funcs.c gimp-1.2.3-patched/app/paint_funcs.c
--- gimp-1.2.3/app/paint_funcs.c	Thu Nov 22 17:10:40 2001
+++ gimp-1.2.3-patched/app/paint_funcs.c	Sat Mar  9 15:24:15 2002
@@ -1106,6 +1106,17 @@
     }
 }
 
+static int do_rand(unsigned long *ctx)
+{
+	return ((*ctx = *ctx * 1103515245 + 12345) % ((u_long)RAND_MAX + 1));
+}
+
+static int rand_r(unsigned int *ctx)
+{
+	u_long val = (u_long) *ctx;
+	*ctx = do_rand(&val);
+	return (int) *ctx;
+}
 
 void
 dissolve_pixels (const unsigned char *src,
diff -ruN gimp-1.2.3/plug-ins/libgck/gck/Makefile.in gimp-1.2.3-patched/plug-ins/libgck/gck/Makefile.in
--- gimp-1.2.3/plug-ins/libgck/gck/Makefile.in	Mon Feb 11 16:46:23 2002
+++ gimp-1.2.3-patched/plug-ins/libgck/gck/Makefile.in	Sun Mar 10 10:22:57 2002
@@ -193,7 +193,7 @@
 gckinclude_HEADERS =  	gck.h
 
 
-libgck_la_LDFLAGS =  	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -release $(LT_RELEASE) $(no_undefined)
+libgck_la_LDFLAGS =  	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -release 1.2.3 $(no_undefined)
 
 
 INCLUDES =  	-I..				-I$(srcdir)/..			-I$(top_srcdir)			$(GTK_CFLAGS)			-I$(includedir)
diff -ruN gimp-1.2.3/po-libgimp/Makefile.in.in gimp-1.2.3-patched/po-libgimp/Makefile.in.in
--- gimp-1.2.3/po-libgimp/Makefile.in.in	Thu Jan 27 05:46:58 2000
+++ gimp-1.2.3-patched/po-libgimp/Makefile.in.in	Sat Mar  9 15:24:16 2002
@@ -111,9 +111,9 @@
 install-data-no: all
 install-data-yes: all
 	if test -r "$(MKINSTALLDIRS)"; then \
-	  $(MKINSTALLDIRS) $(datadir); \
+	  $(MKINSTALLDIRS) $(DESTDIR)$(datadir); \
 	else \
-	  $(SHELL) $(top_srcdir)/mkinstalldirs $(datadir); \
+	  $(SHELL) $(top_srcdir)/mkinstalldirs $(DESTDIR)$(datadir); \
 	fi
 	@catalogs='$(CATALOGS)'; \
 	for cat in $$catalogs; do \
@@ -123,7 +123,7 @@
 	    *)     destdir=$(localedir);; \
 	  esac; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
-	  dir=$$destdir/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)$$destdir/$$lang/LC_MESSAGES; \
 	  if test -r "$(MKINSTALLDIRS)"; then \
 	    $(MKINSTALLDIRS) $$dir; \
 	  else \
diff -ruN gimp-1.2.3/po-plug-ins/Makefile.in.in gimp-1.2.3-patched/po-plug-ins/Makefile.in.in
--- gimp-1.2.3/po-plug-ins/Makefile.in.in	Thu Feb  3 19:19:34 2000
+++ gimp-1.2.3-patched/po-plug-ins/Makefile.in.in	Sat Mar  9 15:24:16 2002
@@ -111,9 +111,9 @@
 install-data-no: all
 install-data-yes: all
 	if test -r "$(MKINSTALLDIRS)"; then \
-	  $(MKINSTALLDIRS) $(datadir); \
+	  $(MKINSTALLDIRS) $(DESTDIR)$(datadir); \
 	else \
-	  $(SHELL) $(top_srcdir)/mkinstalldirs $(datadir); \
+	  $(SHELL) $(top_srcdir)/mkinstalldirs $(DESTDIR)$(datadir); \
 	fi
 	@catalogs='$(CATALOGS)'; \
 	for cat in $$catalogs; do \
diff -ruN gimp-1.2.3/po-script-fu/Makefile.in.in gimp-1.2.3-patched/po-script-fu/Makefile.in.in
--- gimp-1.2.3/po-script-fu/Makefile.in.in	Sat Jun 24 06:46:57 2000
+++ gimp-1.2.3-patched/po-script-fu/Makefile.in.in	Sat Mar  9 15:24:16 2002
@@ -118,9 +118,9 @@
 install-data-no: all
 install-data-yes: all
 	if test -r "$(MKINSTALLDIRS)"; then \
-	  $(MKINSTALLDIRS) $(datadir); \
+	  $(MKINSTALLDIRS) $(DESTDIR)$(datadir); \
 	else \
-	  $(SHELL) $(top_srcdir)/mkinstalldirs $(datadir); \
+	  $(SHELL) $(top_srcdir)/mkinstalldirs $(DESTDIR)$(datadir); \
 	fi
 	@catalogs='$(CATALOGS)'; \
 	for cat in $$catalogs; do \
@@ -130,7 +130,7 @@
 	    *)     destdir=$(localedir);; \
 	  esac; \
 	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
-	  dir=$$destdir/$$lang/LC_MESSAGES; \
+	  dir=$(DESTDIR)$$destdir/$$lang/LC_MESSAGES; \
 	  if test -r "$(MKINSTALLDIRS)"; then \
 	    $(MKINSTALLDIRS) $$dir; \
 	  else \
diff -ru gimp-1.2.3/configure gimp-1.2.3-patched/configure
--- gimp-1.2.3/configure	Sun Feb 10 01:21:30 2002
+++ gimp-1.2.3-patched/configure	Sat Jun 15 11:32:40 2002
@@ -3688,7 +3688,7 @@
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
     #	     `"' quotes if we put them in here... so don't!
-    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
