diff -ru biopython-1.59/setup.py biopython-1.59-patched/setup.py
--- biopython-1.59/setup.py	2012-02-24 14:31:34.000000000 +0100
+++ biopython-1.59-patched/setup.py	2012-02-29 18:09:01.000000000 +0100
@@ -236,7 +236,8 @@
         os.chdir("Tests")
         sys.path.insert(0, '')
         import run_tests   
-        run_tests.main([])
+        if run_tests.main(['--offline']):
+            sys.exit(1)
 
         # change back to the current directory
         os.chdir(this_dir)
@@ -360,12 +360,12 @@
               include_dirs=["Bio"]
               ),
 #Commented out due to the build dependency on flex, see Bug 2619
-#   Extension('Bio.PDB.mmCIF.MMCIFlex',
-#              ['Bio/PDB/mmCIF/lex.yy.c',
-#               'Bio/PDB/mmCIF/MMCIFlexmodule.c'],
-#              include_dirs=["Bio"],
-#              libraries=["fl"]
-#              ),
+   Extension('Bio.PDB.mmCIF.MMCIFlex',
+              ['Bio/PDB/mmCIF/lex.yy.c',
+               'Bio/PDB/mmCIF/MMCIFlexmodule.c'],
+              include_dirs=["Bio"],
+              libraries=["fl"]
+              ),
     Extension('Bio.Nexus.cnexus',
               ['Bio/Nexus/cnexus.c']
               ),
