diff -ru durin42-hg-git-ef41e87ea11a.orig/tests/test-branch-bookmark-suffix.t durin42-hg-git-ef41e87ea11a/tests/test-branch-bookmark-suffix.t
--- durin42-hg-git-ef41e87ea11a.orig/tests/test-branch-bookmark-suffix.t	2013-12-14 12:03:54.000000000 -0500
+++ durin42-hg-git-ef41e87ea11a/tests/test-branch-bookmark-suffix.t	2013-12-14 22:38:48.000000000 -0500
@@ -84,6 +84,7 @@
   $ cd gitrepo2
   $ git checkout branch1
   Already on 'branch1'
+  Your branch is up-to-date with 'origin/branch1'.
   $ echo g1 >> f1
   $ git add f1
   $ commit -m "append f1"
diff -ru durin42-hg-git-ef41e87ea11a.orig/tests/test-git-submodules.t durin42-hg-git-ef41e87ea11a/tests/test-git-submodules.t
--- durin42-hg-git-ef41e87ea11a.orig/tests/test-git-submodules.t	2013-12-14 12:03:54.000000000 -0500
+++ durin42-hg-git-ef41e87ea11a/tests/test-git-submodules.t	2013-12-14 22:39:24.000000000 -0500
@@ -28,10 +28,12 @@
   $ clonefilt='s/Cloning into/Initialized empty Git repository in/;s/in .*/in .../'
 
   $ git clone ../gitrepo1 . | python -c "$rmpwd" | sed "$clonefilt" | egrep -v '^done\.$'
-  Initialized empty Git repository in ...
+  Cloning into '.'...
+  done.
   
   $ git submodule add ../gitsubrepo subrepo | python -c "$rmpwd" | sed "$clonefilt" | egrep -v '^done\.$'
-  Initialized empty Git repository in ...
+  Cloning into 'subrepo'...
+  done.
   
   $ git commit -m 'add subrepo' | sed 's/, 0 deletions(-)//'
   [master e42b08b] add subrepo
diff -ru durin42-hg-git-ef41e87ea11a.orig/tests/test-subrepos.t durin42-hg-git-ef41e87ea11a/tests/test-subrepos.t
--- durin42-hg-git-ef41e87ea11a.orig/tests/test-subrepos.t	2013-12-14 12:03:54.000000000 -0500
+++ durin42-hg-git-ef41e87ea11a/tests/test-subrepos.t	2013-12-14 22:40:11.000000000 -0500
@@ -29,6 +29,10 @@
   $ cd ..
 Ensure gitlinks are transformed to .hgsubstate on hg pull from git
   $ hg clone gitrepo1 hgrepo
+  Cloning into '$TESTTMP/hgrepo/subrepo1'...
+  done.
+  Cloning into '$TESTTMP/hgrepo/xyz/subrepo2'...
+  done.
   importing git objects into hg
   updating to branch default
   cloning subrepo subrepo1 from $TESTTMP/gitsubrepo
