--- a/setup.py	2013-06-27 11:02:48.000000000 -0400
+++ b/setup.py	2013-10-15 09:35:13.000000000 -0400
@@ -126,11 +126,12 @@
 # cp samtools/*.h pysam/*.h pysam/include
 # cp samtools/win32/*.h pysam/include/win32
 
-
-from distribute_setup import use_setuptools
-use_setuptools()
-
-from setuptools import Extension, setup, find_packages
+try:
+    from setuptools import Extension, setup, find_packages
+except ImportError:
+    from ez_setup import use_setuptools
+    use_setuptools()
+    from setuptools import Extension, setup, find_packages
 
 #######################################################
 #######################################################
