--- clucene-core-0.9.21b/src/CLucene/util/Misc.cpp.orig	2014-05-10 19:46:12.000000000 -0400
+++ clucene-core-0.9.21b/src/CLucene/util/Misc.cpp	2014-05-10 19:47:36.000000000 -0400
@@ -120,6 +118,8 @@
     return ret;
 }
 
+using ::wcschr;
+
 TCHAR* cur = ret; //position of return buffer
 const TCHAR* lst = val; //position of value buffer
 pos = val; //searched position of value buffer
--- clucene-core-0.9.21b/src/CLucene/search/FieldCacheImpl.cpp.orig	2014-05-10 20:17:17.000000000 -0400
+++ clucene-core-0.9.21b/src/CLucene/search/FieldCacheImpl.cpp	2014-05-10 20:18:18.000000000 -0400
@@ -404,6 +404,8 @@
     return ret;
   }
 
+using ::wcschr;
+
   // inherit javadocs
   FieldCacheAuto* FieldCacheImpl::getAuto (IndexReader* reader, const TCHAR* field) {
 	field = CLStringIntern::intern(field CL_FILELINE);
