Index: trunk/src/exif.cpp
===================================================================
--- trunk/src/exif.cpp	(revision 2046)
+++ trunk/src/exif.cpp	(revision 2047)
@@ -705,8 +705,8 @@
         }
 
         // IPTC and XMP are stored elsewhere, not in the Exif APP1 segment.
-        const IptcData emptyIptc;
-        const XmpData  emptyXmp;
+        IptcData emptyIptc;
+        XmpData  emptyXmp;
 
         // Encode and check if the result fits into a JPEG Exif APP1 segment
         MemIo mio1;
@@ -891,8 +891,8 @@
         }
 
         Exiv2::MemIo io;
-        const Exiv2::IptcData emptyIptc;
-        const Exiv2::XmpData  emptyXmp;
+        Exiv2::IptcData emptyIptc;
+        Exiv2::XmpData  emptyXmp;
         Exiv2::TiffParser::encode(io, 0, 0, Exiv2::littleEndian, thumb, emptyIptc, emptyXmp);
         return io.read(io.size());
     }
Index: trunk/src/preview.cpp
===================================================================
--- trunk/src/preview.cpp	(revision 2046)
+++ trunk/src/preview.cpp	(revision 2047)
@@ -647,8 +647,8 @@
 
         // write new image
         MemIo mio;
-        const IptcData emptyIptc;
-        const XmpData  emptyXmp;
+        IptcData emptyIptc;
+        XmpData  emptyXmp;
         TiffParser::encode(mio, 0, 0, Exiv2::littleEndian, preview, emptyIptc, emptyXmp);
         return DataBuf(mio.mmap(), mio.size());
     }
