diff -ruN id3lib-3.8.3-orig/configure id3lib-3.8.3/configure
--- id3lib-3.8.3-orig/configure	2003-03-01 19:23:00.000000000 -0500
+++ id3lib-3.8.3/configure	2011-07-27 17:24:03.000000000 -0400
@@ -5581,7 +5581,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle $allow_undefined_flag || echo -dynamiclib) -o $lib $libobjs $deplibs$linker_flags $(test "x$module" != xyes && echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -ruN id3lib-3.8.3-orig/examples/demo_convert.cpp id3lib-3.8.3/examples/demo_convert.cpp
--- id3lib-3.8.3-orig/examples/demo_convert.cpp	2003-03-01 19:23:00.000000000 -0500
+++ id3lib-3.8.3/examples/demo_convert.cpp	2011-07-27 17:24:17.000000000 -0400
@@ -84,7 +84,7 @@
   }
 }
 
-int main( unsigned int argc, char * const argv[])
+int main( int argc, char * const argv[])
 {
   flags_t ulFlag = ID3TT_ALL;
   gengetopt_args_info args;
diff -ruN id3lib-3.8.3-orig/examples/demo_copy.cpp id3lib-3.8.3/examples/demo_copy.cpp
--- id3lib-3.8.3-orig/examples/demo_copy.cpp	2003-03-01 19:23:00.000000000 -0500
+++ id3lib-3.8.3/examples/demo_copy.cpp	2011-07-27 17:24:17.000000000 -0400
@@ -81,7 +81,7 @@
   }
 }
 
-int main( unsigned int argc, char * const argv[])
+int main( int argc, char * const argv[])
 {
   int ulFlag = ID3TT_ID3;
   ID3D_INIT_DOUT();
diff -ruN id3lib-3.8.3-orig/examples/demo_info.cpp id3lib-3.8.3/examples/demo_info.cpp
--- id3lib-3.8.3-orig/examples/demo_info.cpp	2003-03-01 19:23:00.000000000 -0500
+++ id3lib-3.8.3/examples/demo_info.cpp	2011-07-27 17:24:17.000000000 -0400
@@ -309,7 +309,7 @@
 
 #define DEBUG
 
-int main( unsigned int argc, char * const argv[])
+int main( int argc, char * const argv[])
 {
   ID3D_INIT_DOUT();
 
diff -ruN id3lib-3.8.3-orig/examples/demo_tag.cpp id3lib-3.8.3/examples/demo_tag.cpp
--- id3lib-3.8.3-orig/examples/demo_tag.cpp	2003-03-01 19:23:00.000000000 -0500
+++ id3lib-3.8.3/examples/demo_tag.cpp	2011-07-27 17:24:17.000000000 -0400
@@ -46,7 +46,7 @@
     os << "v2";
 }
 
-int main( unsigned int argc, char * const argv[])
+int main( int argc, char * const argv[])
 {
   int ulFlag = ID3TT_ID3;
   ID3D_INIT_DOUT();
diff -ruN id3lib-3.8.3-orig/examples/findeng.cpp id3lib-3.8.3/examples/findeng.cpp
--- id3lib-3.8.3-orig/examples/findeng.cpp	2003-03-01 19:23:00.000000000 -0500
+++ id3lib-3.8.3/examples/findeng.cpp	2011-07-27 17:24:17.000000000 -0400
@@ -9,7 +9,7 @@
 using std::cout;
 using std::endl;
 
-int main(unsigned argc, char* argv[])
+int main(int argc, char* argv[])
 {
   ID3D_INIT_DOUT();
   ID3D_INIT_WARNING();
diff -ruN id3lib-3.8.3-orig/examples/findstr.cpp id3lib-3.8.3/examples/findstr.cpp
--- id3lib-3.8.3-orig/examples/findstr.cpp	2003-03-01 19:23:00.000000000 -0500
+++ id3lib-3.8.3/examples/findstr.cpp	2011-07-27 17:24:17.000000000 -0400
@@ -9,7 +9,7 @@
 using std::cout;
 using std::endl;
 
-int main(unsigned argc, char* argv[])
+int main(int argc, char* argv[])
 {
   ID3D_INIT_DOUT();
   ID3D_INIT_WARNING();
diff -ruN id3lib-3.8.3-orig/examples/test_io.cpp id3lib-3.8.3/examples/test_io.cpp
--- id3lib-3.8.3-orig/examples/test_io.cpp	2003-03-01 19:23:00.000000000 -0500
+++ id3lib-3.8.3/examples/test_io.cpp	2011-07-27 17:24:17.000000000 -0400
@@ -4,6 +4,8 @@
 #  include <config.h>
 #endif
 
+#include <iostream.h>
+#include <iomanip.h>
 #include <id3/readers.h>
 #include <id3/writers.h>
 #include <id3/io_decorators.h>
@@ -18,7 +18,7 @@
 using namespace dami;
 
 int
-main(size_t argc, const char** argv)
+main(int argc, const char** argv)
 {
   ID3D_INIT_DOUT();
   ID3D_INIT_WARNING();
diff -ruN id3lib-3.8.3-orig/ltmain.sh id3lib-3.8.3/ltmain.sh
--- id3lib-3.8.3-orig/ltmain.sh	2003-03-01 19:23:00.000000000 -0500
+++ id3lib-3.8.3/ltmain.sh	2011-07-27 17:23:53.000000000 -0400
@@ -1739,7 +1739,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    need_relink=no
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then
diff -ruN id3lib-3.8.3-orig/src/Makefile.in id3lib-3.8.3/src/Makefile.in
--- id3lib-3.8.3-orig/src/Makefile.in	2003-03-01 19:23:00.000000000 -0500
+++ id3lib-3.8.3/src/Makefile.in	2011-07-27 17:24:13.000000000 -0400
@@ -183,7 +183,7 @@
 
 libid3_la_LDFLAGS = \
   -version-info $(LT_VERSION) \
-  -release $(LT_RELEASE) \
+    \
   -export-dynamic
 
 subdir = src
--- a/src/mp3_parse.cpp	2003-03-02 01:23:00.000000000 +0100
+++ b/src/mp3_parse.cpp	2009-09-27 19:44:18.000000000 +0200
@@ -465,7 +465,7 @@
   // from http://www.xingtech.com/developer/mp3/
 
   const size_t VBR_HEADER_MIN_SIZE = 8;     // "xing" + flags are fixed
-  const size_t VBR_HEADER_MAX_SIZE = 116;   // frames, bytes, toc and scale are optional
+  const size_t VBR_HEADER_MAX_SIZE = 120;   // frames, bytes, toc and scale are optional
 
   if (mp3size >= vbr_header_offest + VBR_HEADER_MIN_SIZE) 
   {
