diff -ruN libcdio-0.83-orig/configure libcdio-0.83/configure
--- libcdio-0.83-orig/configure	2011-10-26 22:23:40.000000000 -0400
+++ libcdio-0.83/configure	2013-11-05 20:58:54.000000000 -0500
@@ -16675,7 +16675,7 @@
        ## AC_DEFINE([HAVE_AIX_CDROM], [1],
        ##     [Define 1 if you have AIX CD-ROM support])
        ;;
-     darwin6*|darwin7*|darwin8*|darwin9*)
+     darwin[6-9].*|darwin1[0-2].*)
        for ac_header in IOKit/IOKitLib.h CoreFoundation/CFBase.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
diff -ruN libcdio-0.83-orig/lib/driver/osx.c libcdio-0.83/lib/driver/osx.c
--- libcdio-0.83-orig/lib/driver/osx.c	2011-10-20 19:16:20.000000000 -0400
+++ libcdio-0.83/lib/driver/osx.c	2013-11-05 21:03:30.000000000 -0500
@@ -56,7 +56,11 @@
 
 #include <mach/mach.h>
 #include <Carbon/Carbon.h>
-#include <IOKit/scsi-commands/SCSITaskLib.h>
+#if __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 1030
+# include <IOKit/scsi/SCSITaskLib.h>
+#else
+# include <IOKit/scsi-commands/SCSITaskLib.h>
+#endif
 #include <IOKit/IOCFPlugIn.h>
 #include <mach/mach_error.h>
 
@@ -75,7 +79,11 @@
 #include <CoreFoundation/CoreFoundation.h>
 #include <IOKit/IOKitLib.h>
 #include <IOKit/IOBSD.h>
-#include <IOKit/scsi-commands/IOSCSIMultimediaCommandsDevice.h>
+#if __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 1030
+# include <IOKit/scsi/IOSCSIMultimediaCommandsDevice.h>
+#else
+# include <IOKit/scsi-commands/IOSCSIMultimediaCommandsDevice.h>
+#endif
 #include <IOKit/storage/IOCDTypes.h>
 #include <IOKit/storage/IODVDTypes.h>
 #include <IOKit/storage/IOMedia.h>
