diff -Nurd -x'*~' exempi-2.2.1.orig/configure exempi-2.2.1/configure
--- exempi-2.2.1.orig/configure	2013-05-25 17:49:55.000000000 -0400
+++ exempi-2.2.1/configure	2013-12-09 20:08:40.000000000 -0500
@@ -16115,7 +16115,7 @@
 case $build_vendor in
 	apple)
 		EXEMPI_PLATFORM_DEF=MAC_ENV
-		LDFLAGS="$LDFLAGS -framework CoreServices"
+		LDFLAGS="$LDFLAGS -Wl,-framework,CoreServices"
 		;;
 	*)
 		EXEMPI_PLATFORM_DEF=UNIX_ENV
diff -Nurd -x'*~' exempi-2.2.1.orig/exempi/tests/Makefile.in exempi-2.2.1/exempi/tests/Makefile.in
--- exempi-2.2.1.orig/exempi/tests/Makefile.in	2013-05-25 17:49:53.000000000 -0400
+++ exempi-2.2.1/exempi/tests/Makefile.in	2013-12-09 20:08:03.000000000 -0500
@@ -89,8 +89,6 @@
 @WITH_UNIT_TEST_TRUE@check_PROGRAMS = testexempicore$(EXEEXT) \
 @WITH_UNIT_TEST_TRUE@	testserialise$(EXEEXT) \
 @WITH_UNIT_TEST_TRUE@	testwritenewprop$(EXEEXT) \
-@WITH_UNIT_TEST_TRUE@	testtiffleak$(EXEEXT) \
-@WITH_UNIT_TEST_TRUE@	testxmpfiles$(EXEEXT) \
 @WITH_UNIT_TEST_TRUE@	testxmpfileswrite$(EXEEXT) test3$(EXEEXT) \
 @WITH_UNIT_TEST_TRUE@	testinit$(EXEEXT) testfdo18635$(EXEEXT) \
 @WITH_UNIT_TEST_TRUE@	testcpp$(EXEEXT)
diff -Nurd -x'*~' exempi-2.2.1.orig/source/common/LargeFileAccess.hpp exempi-2.2.1/source/common/LargeFileAccess.hpp
--- exempi-2.2.1.orig/source/common/LargeFileAccess.hpp	2013-05-25 17:48:43.000000000 -0400
+++ exempi-2.2.1/source/common/LargeFileAccess.hpp	2014-07-29 06:52:01.000000000 -0400
@@ -12,6 +12,7 @@
 #include "XMP_Environment.h"	// ! This must be the first include.
 
 #include <stdio.h>  // The assert macro needs printf.
+#include <stdlib.h>
 #include <string>
 
 #if XMP_WinBuild
