diff -Nurd libidn-1.22/Makefile.in libidn-1.22-new/Makefile.in
--- libidn-1.22/Makefile.in	2011-05-04 14:38:11.000000000 -0400
+++ libidn-1.22-new/Makefile.in	2011-05-13 09:33:10.000000000 -0400
@@ -976,7 +976,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = gl lib/gl lib po src doc gltests lib/gltests tests examples	\
+SUBDIRS = gl lib/gl lib po src gltests lib/gltests tests examples	\
 	java csharp
 
 
diff -Nurd libidn-1.31/java/src/main/java/gnu/inet/encoding/Makefile.in libidn-1.31-new/java/src/main/java/gnu/inet/encoding/Makefile.in
--- libidn-1.31/java/src/main/java/gnu/inet/encoding/Makefile.in	2015-07-08 16:01:48.000000000 -0500
+++ libidn-1.22-new/java/src/main/java/gnu/inet/encoding/Makefile.in	2015-07-15 22:30:04.000000000 -0500
@@ -30,6 +30,7 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
+AM_JAVACFLAGS = -target 1.5 -source 1.5
 VPATH = @srcdir@
 am__is_gnu_make = test -n '$(MAKEFILE_LIST)' && test -n '$(MAKELEVEL)'
 am__make_running_with_option = \
