--- a/test/Font/Makefile.in	2011-10-26 03:30:22.000000000 -0400
+++ b/test/Font/Makefile.in	2013-03-16 10:57:20.000000000 -0400
@@ -53,11 +53,11 @@
 CONFIG_HEADER = $(top_builddir)/src/ming_config.h
 CONFIG_CLEAN_FILES =
 CONFIG_CLEAN_VPATH_FILES =
-am__EXEEXT_1 = test01$(EXEEXT) test02$(EXEEXT) test03$(EXEEXT) \
+am__EXEEXT_1 = test01$(EXEEXT) test02$(EXEEXT) \
 	test04$(EXEEXT) test05$(EXEEXT) test06$(EXEEXT) \
 	test07$(EXEEXT)
 @USE_CXX_TRUE@am__EXEEXT_2 = test01-cxx$(EXEEXT) test02-cxx$(EXEEXT) \
-@USE_CXX_TRUE@	test03-cxx$(EXEEXT) test04-cxx$(EXEEXT) \
+@USE_CXX_TRUE@	test04-cxx$(EXEEXT) \
 @USE_CXX_TRUE@	test05-cxx$(EXEEXT) test06-cxx$(EXEEXT) \
 @USE_CXX_TRUE@	test07-cxx$(EXEEXT)
 am__EXEEXT_3 = $(am__EXEEXT_1) $(am__EXEEXT_2)
