diff -Nurd -x'*~' libnotifymm-0.6.1.orig/libnotify/libnotifymm/Makefile.in libnotifymm-0.6.1/libnotify/libnotifymm/Makefile.in
--- libnotifymm-0.6.1.orig/libnotify/libnotifymm/Makefile.in	2008-03-26 06:46:42.000000000 -0400
+++ libnotifymm-0.6.1/libnotify/libnotifymm/Makefile.in	2014-04-09 11:28:25.000000000 -0400
@@ -242,7 +242,7 @@
 sublib_include_HEADERS = $(files_built_h) $(files_extra_h)
 lib_LTLIBRARIES = libnotifymm-1.0.la
 libnotifymm_1_0_la_SOURCES = $(files_all_cc)
-libnotifymm_1_0_la_LDFLAGS = $(common_ldflags)
+libnotifymm_1_0_la_LDFLAGS = $(common_ldflags) -no-undefined
 libnotifymm_1_0_la_LIBADD = $(LIBNOTIFYMM_LIBS)
 all: all-recursive
 
diff -Nurd -x'*~' libnotifymm-0.6.1.orig/libnotify/libnotifymm-1.0.pc.in libnotifymm-0.6.1/libnotify/libnotifymm-1.0.pc.in
--- libnotifymm-0.6.1.orig/libnotify/libnotifymm-1.0.pc.in	2007-09-25 06:00:51.000000000 -0400
+++ libnotifymm-0.6.1/libnotify/libnotifymm-1.0.pc.in	2011-02-21 02:36:05.000000000 -0500
@@ -5,7 +5,7 @@
 
 Name: libnotifymm
 Description: C++ wrapper for libnotify
-Requires: glibmm-2.4 libnotify
+Requires: glibmm-2.4 libnotify gtkmm-2.4
 Version: @VERSION@
 Libs: -L${libdir} -lnotifymm-1.0
 Cflags: -I${includedir}/libnotifymm-1.0 -I${libdir}/libnotifymm-1.0/include
