diff -ruN libofa-0.9.3-orig/configure.in libofa-0.9.3/configure.in
--- libofa-0.9.3-orig/configure.in	2006-05-10 14:00:19.000000000 -0400
+++ libofa-0.9.3/configure.in	2013-09-06 08:01:01.000000000 -0400
@@ -74,7 +74,7 @@
 dnl  FFTW for FFTW v2. FFTW3 for FFTW v3. VDSP for vDSP. MKL for MKL
 dnl TODO: PREANSI for win32
 
-if test x$os = xdarwin; then
+if test x$os = xBROKENdarwin; then
     AC_MSG_NOTICE([Using vDSP on OS X])
     LIBS="$LIBS -framework Accelerate"
     FFT_WRAPPER="fftlibvdsp_op.cpp"
diff -ruN libofa-0.9.3-orig/examples/protocol.cpp libofa-0.9.3/examples/protocol.cpp
--- libofa-0.9.3-orig/examples/protocol.cpp	2006-05-10 14:05:42.000000000 -0400
+++ libofa-0.9.3/examples/protocol.cpp	2013-09-06 08:02:03.000000000 -0400
@@ -12,8 +12,6 @@
 #include <map>
 #include <expat.h>
 #include <curl/curl.h>
-#include <curl/types.h>
-#include <curl/easy.h>
 
 using namespace std;
 
diff -ruN libofa-0.9.3-orig/lib/JAMA/tnt_math_utils.h libofa-0.9.3/lib/JAMA/tnt_math_utils.h
--- libofa-0.9.3-orig/lib/JAMA/tnt_math_utils.h	2006-05-10 13:58:50.000000000 -0400
+++ libofa-0.9.3/lib/JAMA/tnt_math_utils.h	2013-09-06 08:00:24.000000000 -0400
@@ -19,6 +19,7 @@
 
 namespace TNT
 {
+template <class Real> Real abs(const Real &a);
 /**
 	@returns hypotenuse of real (non-complex) scalars a and b by 
 	avoiding underflow/overflow
