diff -Nurd -x'*~' uriparser-0.8.1.orig/Makefile.in uriparser-0.8.1/Makefile.in
--- uriparser-0.8.1.orig/Makefile.in	2014-10-20 11:36:56.000000000 -0400
+++ uriparser-0.8.1/Makefile.in	2015-04-13 00:34:46.000000000 -0400
@@ -57,7 +57,7 @@
 @URI_TEST_ENABLED_TRUE@check_PROGRAMS = test/uriparser_test$(EXEEXT)
 @URI_TEST_ENABLED_TRUE@TESTS = test/uriparser_test$(EXEEXT)
 bin_PROGRAMS = uriparse$(EXEEXT)
-@WIN32_TRUE@am__append_1 = -no-undefined
+am__append_1 = -no-undefined
 subdir = .
 DIST_COMMON = $(am__configure_deps) $(pkginclude_HEADERS) \
 	$(srcdir)/Makefile.am $(srcdir)/Makefile.in \
