diff -ruN xapian-core-1.2.15/tests/api_serialise.cc xapian-core-1.2.15/tests/api_serialise.cc 
--- xapian-core-1.2.15/tests/api_serialise.cc	2013-10-06 18:20:29.000000000 -0400
+++ xapian-core-1.2.15/tests/api_serialise.cc	2013-10-06 18:25:56.000000000 -0400
@@ -347,6 +347,7 @@
 
 /// Check that exceptions when registering a postingsource are handled well.
 DEFINE_TESTCASE(registry1, !backend) {
+    SKIP_TEST_FOR_BACKEND("none");
     // Test that a replacement object throwing bad_alloc is handled.
     {
 	Xapian::Registry reg;
@@ -448,6 +449,7 @@
 
 /// Check that exceptions when registering are handled well.
 DEFINE_TESTCASE(registry2, !backend) {
+    SKIP_TEST_FOR_BACKEND("none");
     // Test that a replacement object throwing bad_alloc is handled.
     {
 	Xapian::Registry reg;
@@ -540,6 +542,7 @@
 
 /// Check that exceptions when registering are handled well.
 DEFINE_TESTCASE(registry3, !backend) {
+    SKIP_TEST_FOR_BACKEND("none");
     // Test that a replacement object throwing bad_alloc is handled.
     {
 	Xapian::Registry reg;
