diff -Nurd -x'*~' nspr-4.10.7.orig/nspr/config/rules.mk nspr-4.10.7/nspr/config/rules.mk
--- nspr-4.10.7.orig/nspr/config/rules.mk	2014-08-12 11:40:57.000000000 -0400
+++ nspr-4.10.7/nspr/config/rules.mk	2014-09-14 22:16:43.000000000 -0400
@@ -99,7 +99,8 @@
 SHARED_LIBRARY	= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION)_shr.a
 else
 ifdef MKSHLIB
-SHARED_LIBRARY	= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
+SO_VERSION=0e
+SHARED_LIBRARY	= $(OBJDIR)/lib$(LIBRARY_NAME)$(LIBRARY_VERSION).$(SO_VERSION).$(DLL_SUFFIX)
 endif
 endif
 
diff -Nurd -x'*~' nspr-4.10.7.orig/nspr/configure nspr-4.10.7/nspr/configure
--- nspr-4.10.7.orig/nspr/configure	2014-08-12 11:40:57.000000000 -0400
+++ nspr-4.10.7/nspr/configure	2014-09-14 15:38:41.000000000 -0400
@@ -6589,7 +6589,7 @@
 
 
     DSO_CFLAGS=-fPIC
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@ -headerpad_max_install_names'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name $(libdir)/$@ -headerpad_max_install_names'
     _OPTIMIZE_FLAGS=-O2
     MKSHLIB='$(CC) $(DSO_LDOPTS) -o $@'
     STRIP="$STRIP -x -S"
@@ -6598,6 +6598,8 @@
     MDCPUCFG_H=_darwin.cfg
     PR_MD_CSRCS=darwin.c
     PR_MD_ASFILES=os_Darwin.s
+    LIBNSPR='$(dist_libdir)/libnspr$(MOD_MAJOR_VERSION).$(SO_VERSION).dylib'
+    LIBPLC='$(dist_libdir)/libplc$(MOD_MAJOR_VERSION).$(SO_VERSION).dylib'
 
     # Add Mac OS X support for loading CFM & CFBundle plugins
     if test -f "${MACOS_SDK_DIR}/System/Library/Frameworks/Carbon.framework/Carbon"; then
diff -Nurd -x'*~' nspr-4.10.7.orig/nspr/pr/src/md/unix/unix.c nspr-4.10.7/nspr/pr/src/md/unix/unix.c
--- nspr-4.10.7.orig/nspr/pr/src/md/unix/unix.c	2014-08-12 11:40:57.000000000 -0400
+++ nspr-4.10.7/nspr/pr/src/md/unix/unix.c	2014-09-14 22:16:10.000000000 -0400
@@ -3658,7 +3658,7 @@
  * that calls poll(), we emulate poll() using select().
  */
 
-int poll(struct pollfd *filedes, unsigned long nfds, int timeout)
+int __attribute__((visibility("hidden"))) poll(struct pollfd *filedes, unsigned long nfds, int timeout)
 {
     int i;
     int rv;
diff -Nurd -x'*~' nspr-4.10.7.orig/nspr/pr/src/misc/prtime.c nspr-4.10.7/nspr/pr/src/misc/prtime.c
--- nspr-4.10.7.orig/nspr/pr/src/misc/prtime.c	2014-08-12 11:40:57.000000000 -0400
+++ nspr-4.10.7/nspr/pr/src/misc/prtime.c	2014-09-14 15:42:24.000000000 -0400
@@ -1495,7 +1495,7 @@
                 case TT_EET: zone_offset =  2 * 60; break;
                 case TT_JST: zone_offset =  9 * 60; break;
                 default:
-                  PR_ASSERT (0);
+                  return PR_FAILURE;
                   break;
                 }
         }
@@ -1543,11 +1543,12 @@
           struct tm localTime;
           time_t secs;
 
-          PR_ASSERT(result->tm_month > -1 &&
+          if(!(result->tm_month > -1 &&
                     result->tm_mday > 0 &&
                     result->tm_hour > -1 &&
                     result->tm_min > -1 &&
-                    result->tm_sec > -1);
+                    result->tm_sec > -1))
+               return PR_FAILURE;
 
             /*
              * To obtain time_t from a tm structure representing the local
