diff -ruNp soprano-2.7.6-orig/CMakeLists.txt soprano-2.7.6/CMakeLists.txt
--- soprano-2.7.6-orig/CMakeLists.txt	2012-05-16 04:36:35.000000000 -0400
+++ soprano-2.7.6/CMakeLists.txt	2012-10-12 10:16:57.000000000 -0400
@@ -221,7 +221,7 @@ endif(NOT WIN32)
 ##################  apidox ################################
 find_package(Doxygen)
 
-if(DOXYGEN_EXECUTABLE)
+if(FALSE)
   set(QTDOCS_ONLINE "http://qt-project.org/doc/qt-4.8/")
 
   if(EXISTS ${QT_DOC_DIR}/html)
@@ -264,7 +264,7 @@ if(DOXYGEN_EXECUTABLE)
     COMMAND rsync -zrtq docs/html/ $ENV{USER},soprano@web.sf.net:/home/project-web/soprano/htdocs/apidox/trunk
     DEPENDS onlineapidox
     COMMENT "Uploading trunk API docs to Sourceforge...")
-endif(DOXYGEN_EXECUTABLE)
+endif(FALSE)
 
 
 ##################  status messages ################################
diff -ruNp soprano-2.7.6-orig/Doxyfile.cmake soprano-2.7.6/Doxyfile.cmake
--- soprano-2.7.6-orig/Doxyfile.cmake	2012-05-16 04:36:35.000000000 -0400
+++ soprano-2.7.6/Doxyfile.cmake	2012-10-12 10:17:55.000000000 -0400
@@ -239,7 +239,7 @@ SKIP_FUNCTION_MACROS   = YES
 # Configuration::additions related to external references   
 #---------------------------------------------------------------------------
 TAGFILES               = ${soprano_SOURCE_DIR}/qt4.tag=${QTDOCS}
-GENERATE_TAGFILE       = 
+GENERATE_TAGFILE       = docs/soprano.tag
 ALLEXTERNALS           = NO
 EXTERNAL_GROUPS        = YES
 PERL_PATH              = /usr/bin/perl
diff -ruNp soprano-2.7.6-orig/index/CMakeLists.txt soprano-2.7.6/index/CMakeLists.txt
--- soprano-2.7.6-orig/index/CMakeLists.txt	2012-05-16 04:36:35.000000000 -0400
+++ soprano-2.7.6/index/CMakeLists.txt	2012-10-12 10:15:49.000000000 -0400
@@ -11,6 +11,8 @@ include_directories(
   ${CMAKE_CURRENT_BINARY_DIR}
 )
 
+add_definitions(-I${CLUCENE_LIBRARY_DIR})
+
 set( SOPRANO_INDEX_SRCS
   cluceneindex.cpp
   clucenedocumentwrapper.cpp
diff -ruNp soprano-2.7.6-orig/soprano/qhashqurlcompat.cpp soprano-2.7.6/soprano/qhashqurlcompat.cpp
--- soprano-2.7.6-orig/soprano/qhashqurlcompat.cpp	2012-05-16 04:36:35.000000000 -0400
+++ soprano-2.7.6/soprano/qhashqurlcompat.cpp	2012-10-12 10:15:49.000000000 -0400
@@ -46,7 +46,7 @@ namespace Soprano {
     uint soprano_qHash_QUrl(const QUrl &);
 }
 
-#if !((defined(Q_OS_MACX) || defined(_WIN32) ) && QT_VERSION >= 0x040700)
+#if !(defined(_WIN32) && QT_VERSION >= 0x040700)
 uint qHash(const QUrl &url)
 {
     return Soprano::soprano_qHash_QUrl(url);
