diff -Nurd ztex/libusbJava-src/Makefile.macosx ztex-patched/libusbJava-src/Makefile.macosx
--- ztex/libusbJava-src/Makefile.macosx	2011-09-26 12:15:21.000000000 +0200
+++ ztex-patched/libusbJava-src/Makefile.macosx	2011-09-27 19:35:02.000000000 +0200
@@ -60,7 +60,7 @@
 	$(CHMOD) $(LIBTARGET_SH)
 
 $(LIBTARGET_64): $(LIBSRCS64)
-	$(GCC) -fPIC -m64 -shared -std=c99 -Wall -Wno-pointer-to-int-cast -Wl,-soname,$(LIBTARGET_64) $(LIBINCS) $(LIBSRCS64) $(LIBLIBS) -o $(LIBTARGET_64)
+	$(GCC) -fPIC -m64 -std=c99 -Wall -Wno-pointer-to-int-cast -bundle $(LIBINCS) $(LIBSRCS64) $(LIBLIBS) -o $(LIBTARGET_64)
 	$(STRIP) $(LIBTARGET_64)
 	$(CHMOD) $(LIBTARGET_64)
 
