diff -Nurd ztex/Makefile ztex-patched/Makefile
--- ztex/Makefile	2009-12-06 16:50:17.000000000 +0100
+++ ztex-patched/Makefile	2011-09-20 14:28:53.000000000 +0200
@@ -7,6 +7,10 @@
 	@echo "Usage: make all | docs | clean | distclean"
 
 all:
+	make -C bmp all
+	make -C bmp install2
+	make -C libusbJava-src all
+	make -C libusbJava-src install2
 	make -C java all
 	make -C examples all
 
diff -Nurd ztex/bmp/Makefile ztex-patched/bmp/Makefile
--- ztex/bmp/Makefile	2009-04-24 00:01:13.000000000 +0200
+++ ztex-patched/bmp/Makefile	2011-09-20 14:29:59.000000000 +0200
@@ -114,3 +114,4 @@
 	
 distclean: clean resetconf
 	- $(RM) bmp bmp.exe bmp.made bmp.pas
+	- $(RM) ../bin/bmp ../bin/bmp.exe ../bin/bmp.made
diff -Nurd ztex/bmp/configure ztex-patched/bmp/configure
--- ztex/bmp/configure	2003-08-07 21:06:17.000000000 +0200
+++ ztex-patched/bmp/configure	2011-09-20 14:28:53.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 # Makefile.config -- script for automatic configuration of Makefile variables
 # Copyright (C) 2002-2003 Stefan Ziegenbalg
 
diff -Nurd ztex/libusbJava-src/Makefile ztex-patched/libusbJava-src/Makefile
--- ztex/libusbJava-src/Makefile	2011-02-02 13:13:45.000000000 +0100
+++ ztex-patched/libusbJava-src/Makefile	2011-09-20 14:30:56.000000000 +0200
@@ -19,7 +19,7 @@
 # this should not be modified #
 ###############################
 GCC=gcc
-STRIP=strip
+STRIP=strip -S
 CHMOD=chmod -x
 JAVAC=javac
 RM=rm -f
@@ -32,11 +32,11 @@
 LIBSRCS64=LibusbJava.c
 JAVASRCS=$(shell echo ch/ntb/usb/*.java) $(shell echo ch/ntb/usb/logger/*.java)
 
-LIBTARGET_SH=libusbJavaSh.so
-LIBTARGET_ST=libusbJavaSt.so
-LIBTARGET_64=libusbJava64.so
-LIBLIBS=-lusb
-LIBINCS=-I $(JAVAPREFIX)/include -I $(JAVAPREFIX)/include/linux
+LIBTARGET_SH=libusbJavaSh.jnilib
+LIBTARGET_ST=libusbJavaSt.jnilib
+LIBTARGET_64=libusbJava64.jnilib
+LIBLIBS=-lusb -L@@PREFIX@@/lib
+LIBINCS=-I $(JAVAPREFIX)/include -I $(JAVAPREFIX)/include/linux -I @@PREFIX@@/include
 
 .PHONY:	all libs classes install2 clean distclean
 
@@ -51,17 +51,17 @@
 
 $(LIBTARGET_ST): $(LIBSRCS)
 #	$(GCC) -shared -Wl,-static,-soname,$(LIBTARGET_ST) $(LIBINCS) -static $(LIBSRCS) -o $(LIBTARGET_ST) $(LIBLIBS)
-	$(GCC) -shared -Wl,-soname,$(LIBTARGET_ST),-static $(LIBINCS) $(LIBSRCS) -static -o $(LIBTARGET_ST) $(LIBLIBS)
+	$(GCC) -bundle $(LIBINCS) $(LIBSRCS) -o $(LIBTARGET_ST) $(LIBLIBS)
 	$(STRIP) $(LIBTARGET_ST)
 	$(CHMOD) $(LIBTARGET_ST)
 
 $(LIBTARGET_SH): $(LIBSRCS)
-	$(GCC) -fPIC -shared -Wl,-soname,$(LIBTARGET_SH) $(LIBINCS) $(LIBSRCS) -o $(LIBTARGET_SH) $(LIBLIBS)
+	$(GCC) -fPIC -bundle $(LIBINCS) $(LIBSRCS) -o $(LIBTARGET_SH) $(LIBLIBS)
 	$(STRIP) $(LIBTARGET_SH)
 	$(CHMOD) $(LIBTARGET_SH)
 
 $(LIBTARGET_64): $(LIBSRCS64)
-	$(GCC) -fPIC -m64 -shared -std=c99 -Wall -Wno-pointer-to-int-cast -Wl,-soname,$(LIBTARGET_64) $(LIBINCS) $(LIBSRCS64) $(LIBLIBS) -o $(LIBTARGET_64)
+	$(GCC) -fPIC -m64 -bundle -std=c99 -Wall -Wno-pointer-to-int-cast $(LIBINCS) $(LIBSRCS64) $(LIBLIBS) -o $(LIBTARGET_64)
 	$(STRIP) $(LIBTARGET_64)
 	$(CHMOD) $(LIBTARGET_64)
 
@@ -84,3 +84,5 @@
 
 distclean:
 	$(RM) *.o $(LIBTARGET_SH) $(LIBTARGET_ST) $(LIBTARGET_64) libusbJava*.dll ch/ntb/usb/*.class ch/ntb/usb/logger/*.class classes.made
+	$(RM) ../libusbJava/$(LIBTARGET_SH) ../libusbJava/$(LIBTARGET_ST) ../libusbJava/$(LIBTARGET_64) ../libusbJava/ch/ntb/usb/*.class ../libusbJava/ch/ntb/usb/logger/*.class
+	$(RM) -r ../libusbJava/ch/ntb/usb/logger
diff -Nurd ztex/libusbJava-src/ch/ntb/usb/LibusbJava.java ztex-patched/libusbJava-src/ch/ntb/usb/LibusbJava.java
--- ztex/libusbJava-src/ch/ntb/usb/LibusbJava.java	2011-02-02 22:57:04.000000000 +0100
+++ ztex-patched/libusbJava-src/ch/ntb/usb/LibusbJava.java	2011-09-20 14:28:53.000000000 +0200
@@ -370,7 +370,8 @@
 	    		LibLoader.load( "libusbJava32" ); 
 	    	    }
 	    	}
-		else if ( System.getProperty("os.arch").equalsIgnoreCase("amd64") ) {
+		else if ( System.getProperty("os.arch").equalsIgnoreCase("amd64")
+		       || System.getProperty("os.arch").equalsIgnoreCase("x86_64") ) {
 	    		LibLoader.load( "usbJava64" );
 //	    		System.err.println("loaded libusbJava64");
 		}
