Index: admin/Makefile.common
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/Makefile.common,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
Index: admin/acinclude.m4.in
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/acinclude.m4.in,v
retrieving revision 1.1.1.3
diff -u -b -u -r1.1.1.3 acinclude.m4.in
--- admin/acinclude.m4.in	14 Aug 2002 13:18:45 -0000	1.1.1.3
+++ admin/acinclude.m4.in	25 Sep 2002 00:28:53 -0000
@@ -699,13 +699,10 @@
      AC_CHECK_LIB(ipc, shmat, X_EXTRA_LIBS="$X_EXTRA_LIBS -lipc"))
    
    # darwin needs this to initialize the environment
-   AC_CHECK_HEADERS(crt_externs.h)
+   AC_CHECK_HEADERS([crt_externs.h sys/types.h])
    AC_CHECK_FUNC(_NSGetEnviron, [AC_DEFINE(HAVE_NSGETENVIRON, 1, [Define if your system needs _NSGetEnviron to set up the environment])])
  
-   # more headers that need to be explicitly included on darwin
-   AC_CHECK_HEADERS(sys/types.h stdint.h)
-
-   # darwin requires a poll emulation library
+   # darwin requires a poll.h emulation library
    AC_CHECK_LIB(poll, poll, LIB_POLL="-lpoll")
 
    # CoreAudio framework
@@ -714,6 +711,12 @@
      FRAMEWORK_COREAUDIO="-framework CoreAudio"
    ])
 
+   # CoreServices framework
+   AC_CHECK_HEADER(CoreServices/CoreServices.h, [
+     AC_DEFINE(HAVE_CORESERVICES, 1, [Define if you have the CoreServices API])
+     FRAMEWORK_CORESERVICES="-framework CoreServices"
+   ])
+
    # Solaris 2.6 and others need -lresolv for res_init
    AC_CHECK_FUNCS(res_init, , [
      kde_libs_safe="$LIBS"
@@ -736,6 +739,8 @@
    AC_SUBST(FRAMEWORK_COREAUDIO)
    LIBSOCKET="$X_EXTRA_LIBS"
    AC_SUBST(LIBSOCKET)
+   AC_SUBST(FRAMEWORK_COREAUDIO)
+   AC_SUBST(LIB_POLL)
    AC_SUBST(LIBRESOLV)
    AC_SUBST(X_EXTRA_LIBS)
    AC_CHECK_LIB(ucb, killpg, [LIBUCB="-lucb"]) dnl for Solaris2.4
@@ -905,7 +910,7 @@
 kde_ldflags_safe="$LDFLAGS"
 kde_libs_safe="$LIBS"
 
-LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS"
+LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS"
 LIBS="-lXext -lX11 $LIBSOCKET"
 
 AC_TRY_LINK([
@@ -947,9 +952,9 @@
 kde_save_LDFLAGS="$LDFLAGS"
 kde_save_CFLAGS="$CFLAGS"
 kde_save_LIBS="$LIBS"
-LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS"
+LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS"
 CFLAGS="$CFLAGS -I$x_includes"
-LIBS="-lXinerama -lXext"
+LIBS="-lXinerama -lXext -lX11"
 
 if test "x$no_xinerama" = "xno"; then
 
@@ -1846,15 +1851,29 @@
    AC_REQUIRE([K_PATH_X])
 
 if test $kde_qtver = 3; then
+   AC_SUBST(LIB_ARTSKDE, "-lartskde")
+   AC_SUBST(LIB_DCOP, "-lDCOP")
+   AC_SUBST(LIB_KAB, "-lkab")
+   AC_SUBST(LIB_KABC, "-lkabc")
    AC_SUBST(LIB_KDECORE, "-lkdecore")
+   AC_SUBST(LIB_KDED, "@PREFIX@/lib/kded.la")
+   AC_SUBST(LIB_KDEFX, "-lkdefx")
+   AC_SUBST(LIB_KDEPRINT, "-lkdeprint")
+   AC_SUBST(LIB_KDESASL, "-lkdesasl")
+   AC_SUBST(LIB_KDESU, "-lkdesu")
    AC_SUBST(LIB_KDEUI, "-lkdeui")
-   AC_SUBST(LIB_KIO, "-lkio")
-   AC_SUBST(LIB_SMB, "-lsmb")
-   AC_SUBST(LIB_KAB, "-lkab")
    AC_SUBST(LIB_KHTML, "-lkhtml")
-   AC_SUBST(LIB_KSPELL, "-lkspell")
+   AC_SUBST(LIB_KIO, "-lkio")
    AC_SUBST(LIB_KPARTS, "-lkparts")
-   AC_SUBST(LIB_KDEPRINT, "-lkdeprint")
+   AC_SUBST(LIB_KJAVA, "-lkjava")
+   AC_SUBST(LIB_KJS, "-lkjs")
+   AC_SUBST(LIB_KMID, "-lkmid")
+   AC_SUBST(LIB_KSCRIPT, "-lkscript")
+   AC_SUBST(LIB_KSPELL, "-lkspell")
+   AC_SUBST(LIB_KTEXTEDITOR, "-lktexteditor")
+   AC_SUBST(LIB_SHELLSCRIPT, "-lshellscript")
+   AC_SUBST(LIB_SMB, "-lsmb")
+   AC_SUBST(LIB_VCARD, "-lvcard")
 # these are for backward compatibility
    AC_SUBST(LIB_KSYCOCA, "-lkio")
    AC_SUBST(LIB_KFILE, "-lkio")
@@ -2544,10 +2563,10 @@
       if test $kde_use_debug_code = "full"; then
         CFLAGS="-g3 $CFLAGS"
       else
-        CFLAGS="-g -O2 $CFLAGS"
+        CFLAGS="-g -O $CFLAGS"
       fi
     else
-      CFLAGS="-O2 $CFLAGS"
+      CFLAGS="-Os -Ddlsym=dlsym_auto_underscore -frename-registers -mmultiple -mcpu=750 -mtune=750 $CFLAGS"
     fi
   fi
 
@@ -2577,7 +2596,7 @@
         if test "$kde_use_debug_code" = "full"; then
           CXXFLAGS="-g3 $CXXFLAGS"
         else
-          CXXFLAGS="-g -O2 $CXXFLAGS"
+          CXXFLAGS="-g -O $CXXFLAGS"
         fi
       fi
       KDE_CHECK_COMPILER_FLAG(fno-builtin,[CXXFLAGS="-fno-builtin $CXXFLAGS"])
@@ -2589,7 +2608,7 @@
       if test "$CXX" = "KCC"; then
         CXXFLAGS="+K3 $CXXFLAGS"
       else
-        CXXFLAGS="-O2 $CXXFLAGS"
+        CXXFLAGS="-Os -Ddlsym=dlsym_auto_underscore -frename-registers -mmultiple -mcpu=750 -mtune=750 $CXXFLAGS"
       fi  
     fi
   fi
@@ -3059,9 +3078,9 @@
     ac_save_ldflags="$LDFLAGS"
     ac_save_cflags="$CFLAGS"
     if test "x$kde_use_qt_emb" != "xyes"; then
-      LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm -lX11 -lXext $LIBZ $LIBSOCKET"
+      LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm -lX11 -lXext $LIBZ $LIBSOCKET"
     else
-      LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm $LIBZ $LIBSOCKET"
+      LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm $LIBZ $LIBSOCKET"
     fi
     CFLAGS="$CFLAGS $X_INCLUDES $USER_INCLUDES"
     test -n "$XPM_INCLUDE" && CFLAGS="-I$XPM_INCLUDE $CFLAGS"
@@ -4553,6 +4572,7 @@
       KDE_JAVA_PREFIX(/usr/lib/IBMJava2*)
       KDE_JAVA_PREFIX(/usr/lib/IBMJava*)
       KDE_JAVA_PREFIX(/opt/java*)
+      KDE_JAVA_PREFIX(/Library/Java/Home)
     
       kde_cv_path="NONE"
       kde_save_IFS=$IFS
@@ -4604,10 +4624,20 @@
 dnl At this point kde_java_bindir and kde_java_includedir are either set or "no"
 if test "x$kde_java_bindir" != "xno"; then
 
+  case $host_os in
+    darwin*)
+      dnl Look for libjvm.dylib
+      kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.dylib | sed 's,libjvm.dylib,,'|head -n 1`
+      kd_java_libhpidir=''
+      dnl Darwin doesn't have libhpi
+      ;;
+    *)
   dnl Look for libjvm.so
   kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.so | sed 's,libjvm.so,,'|head -n 1`
   dnl Look for libhpi.so and avoid green threads
   kde_java_libhpidir=`find $kde_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,' | tail -n 1`
+      ;;
+  esac
 
   dnl Now check everything's fine under there
   dnl the include dir is our flag for having the JDK
@@ -4635,17 +4665,29 @@
     JAVAC=
     jni_includes=
   fi
-
+  case $host_os in
+    darwin*)
+      if test ! -r "$kde_java_libjvmdir/libjvm.dylib"; then
+        AC_MSG_ERROR([libjvm.dylib not found under $kde_java_libjvmdir. Use --without-java.])
+      fi
+      dnl Darwin doesn't have libhpi, but it's not fatal
+      LIB_JVM="-framework JavaVM"
+      LIB_HPI=""
+      ;;
+    *)
   if test ! -r "$kde_java_libjvmdir/libjvm.so"; then
      AC_MSG_ERROR([libjvm.so not found under $kde_java_libjvmdir. Use --without-java.])
   fi 
+      LIB_JVM="-L$kde_java_libjvmdir -ljvm"
+      if test ! -r "$kde_java_libhpidir/libhpi.so"; then
+        AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
+      fi
+      LIB_HPI="-L$kde_java_libhpidir -lhpi"
+      ;;
+  esac
 
   if test ! -x "$kde_java_bindir/java"; then
       AC_MSG_ERROR([java not found under $kde_java_bindir. javac was found though! Use --with-java or --without-java.])
-  fi
-
-  if test ! -r "$kde_java_libhpidir/libhpi.so"; then
-    AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
   fi
 
   if test -n "$jni_includes"; then
Index: admin/detect-autoconf.sh
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/detect-autoconf.sh,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- admin/detect-autoconf.sh	10 Jul 2002 02:47:38 -0000	1.1.1.1
+++ admin/detect-autoconf.sh	15 Jul 2002 22:55:00 -0000	1.2
@@ -42,12 +42,12 @@
 checkAutomakeAclocal ()
 {
   if test -z "$UNSERMAKE"; then
-    if test -x "`$WHICH automake-1.5`" ; then
-      AUTOMAKE="`$WHICH automake-1.5`"
-      ACLOCAL="`$WHICH aclocal-1.5`"
-    elif test -x "`$WHICH automake-1.6`" ; then
+    if test -x "`$WHICH automake-1.6`" ; then
       AUTOMAKE="`$WHICH automake-1.6`"
       ACLOCAL="`$WHICH aclocal-1.6`"
+    elif test -x "`$WHICH automake-1.5`" ; then
+      AUTOMAKE="`$WHICH automake-1.5`"
+      ACLOCAL="`$WHICH aclocal-1.5`"
     fi
   else
      AUTOMAKE="$UNSERMAKE"
Index: admin/libtool.m4.in
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/libtool.m4.in,v
retrieving revision 1.1.1.3
retrieving revision 1.7
diff -u -b -u -r1.1.1.3 -r1.7
--- admin/libtool.m4.in	14 Aug 2002 13:18:45 -0000	1.1.1.3
+++ admin/libtool.m4.in	16 Sep 2002 14:31:34 -0000	1.7
@@ -609,6 +609,12 @@
     lt_cv_sys_max_cmd_len=-1;
     ;;
 
+  darwin*)
+    # this should be enough... the length test freaks out some versions
+    # of OSX's `expr`
+    lt_cv_sys_max_cmd_len=16384;
+    ;;
+
   *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -1132,7 +1138,7 @@
 darwin* | rhapsody*)
   dynamic_linker="$host_os dyld"
   version_type=darwin
-  need_lib_prefix=no
+  need_lib_prefix=yes
   need_version=no
   # FIXME: Relying on posixy $() will cause problems for
   #        cross-compilation, but unfortunately the echo tests do not
@@ -1901,16 +1907,7 @@
   ;;
 
 darwin* | rhapsody*)
-  lt_cv_deplibs_check_method='file_magic Mach-O dynamically linked shared library'
-  lt_cv_file_magic_cmd='/usr/bin/file -L'
-  case "$host_os" in
-  rhapsody* | darwin1.[[012]])
-    lt_cv_file_magic_test_file=`/System/Library/Frameworks/System.framework/System`
-    ;;
-  *) # Darwin 1.3 on
-    lt_cv_file_magic_test_file='/usr/lib/libSystem.dylib'
-    ;;
-  esac
+  lt_cv_deplibs_check_method='pass_all'
   ;;
 
 freebsd*)
@@ -2560,6 +2557,11 @@
 	;;
     esac
     ;;
+  darwin*)
+    shared_flag='-dynamiclib'
+    _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC -dynamiclib -install_name $rpath/$soname $predep_objects ${lib}-master.o $deplibs $postdep_objects $compiler_flags -o $lib'
+    output_verbose_link_cmd='$CC -dynamiclib $CFLAGS -v conftest.$objext 2>&1 | egrep "\-L"'
+    ;;
   dgux*)
     case $cc_basename in
       ec++)
@@ -4750,7 +4752,9 @@
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+	#_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+        # testing with twolevel namespaces
+	_LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
 	;;
       esac
 
@@ -4758,7 +4762,7 @@
       #        cross-compilation, but unfortunately the echo tests do not
       #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
       #	       `"' quotes if we put them in here... so don't!
-      _LT_AC_TAGVAR(archive_cmds, $1)='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+      _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs $linker_flags $(test x$module != xyes && echo -install_name $rpath/$soname $verstring)'
       # We need to add '_' to the symbols in $export_symbols first
       #_LT_AC_TAGVAR(archive_expsym_cmds, $1)="$_LT_AC_TAGVAR(archive_cmds, $1)"' && strip -s $export_symbols'
       _LT_AC_TAGVAR(hardcode_direct, $1)=yes
Index: admin/ltconfig
===================================================================
RCS file: admin/ltconfig
diff -N admin/ltconfig
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ admin/ltconfig	25 Jun 2002 03:34:27 -0000	1.1.1.1
@@ -0,0 +1,4 @@
+#! /bin/sh
+# Written by <matz@kde.org> and placed into the public domain ;-)
+echo "This file shouldn't be called.  It exists solely for the purpose"
+echo "to make automake 1.4 cope with newer libtools."
Index: admin/ltmain.sh
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/ltmain.sh,v
retrieving revision 1.1.1.2
retrieving revision 1.4
diff -u -b -u -r1.1.1.2 -r1.4
--- admin/ltmain.sh	10 Jul 2002 02:47:39 -0000	1.1.1.2
+++ admin/ltmain.sh	13 Jul 2002 04:33:25 -0000	1.4
@@ -3321,7 +3321,23 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+          if test "x$verstring" = "x0.0"; then
+                tmp_verstring=
+          else
+                tmp_verstring="$verstring"
+          fi
+ 	  save_deplibs="$deplibs"
+ 	  for conv in $convenience; do
+            tmp_deplibs=
+            for test_deplib in $deplibs; do
+              if test "$test_deplib" != "$conv"; then
+                tmp_deplibs="$tmp_deplibs $test_deplib"
+              fi
+            done
+            deplibs="$tmp_deplibs"
+ 	  done
 	  eval cmds=\"$archive_cmds\"
+ 	  deplibs="$save_deplibs"
 	fi
 
 	if len=`expr "X$cmds" : ".*"` &&
@@ -3419,7 +3435,12 @@
 	  if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	    eval cmds=\"$archive_expsym_cmds\"
 	  else
+            save_deplibs="$deplibs"
+            for conv in $convenience; do
+              deplibs="${deplibs%$conv*} ${deplibs#*$conv}"
+            done
 	    eval cmds=\"$archive_cmds\"
+            deplibs="$save_deplibs"
 	  fi
 
 	  # Append the command to remove the reloadable object files
@@ -3632,6 +3653,14 @@
 	compile_deplibs=`$echo "X $compile_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	finalize_deplibs=`$echo "X $finalize_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	;;
+      esac
+
+      case $host in
+      *darwin*)
+        # Don't allow lazy linking, it breaks C++ global constructors
+        compile_command="$compile_command ${wl}-bind_at_load"
+        finalize_command="$finalize_command ${wl}-bind_at_load"
+        ;;
       esac
 
       compile_command="$compile_command $compile_deplibs"
Index: admin/missing
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/missing,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- admin/missing	10 Jul 2002 02:47:39 -0000	1.1.1.2
+++ admin/missing	10 Jul 2002 02:50:11 -0000	1.3
@@ -106,7 +106,7 @@
     exit 1
     ;;
 
-  aclocal)
+  aclocal*)
     echo 1>&2 "\
 WARNING: \`$1' is missing on your system.  You should only need it if
          you modified \`acinclude.m4' or \`${configure_ac}'.  You might want
@@ -143,7 +143,7 @@
     touch $touch_files
     ;;
 
-  automake)
+  automake*)
     echo 1>&2 "\
 WARNING: \`$1' is missing on your system.  You should only need it if
          you modified \`Makefile.am', \`acinclude.m4' or \`${configure_ac}'.
Index: kweather/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdetoys/kweather/Makefile.am,v
retrieving revision 1.1.1.3
retrieving revision 1.2
diff -u -b -u -r1.1.1.3 -r1.2
--- kweather/Makefile.am	14 Aug 2002 21:08:25 -0000	1.1.1.3
+++ kweather/Makefile.am	15 Aug 2002 21:07:41 -0000	1.2
@@ -2,7 +2,7 @@
 
 INCLUDES = $(all_includes)
 
-weather_panelapplet_la_SOURCES = reportview.cpp prefdialogview.cpp weatherlib.cpp prefsDialog.cpp kweather.cpp dockwidget.cpp 
+weather_panelapplet_la_SOURCES = prefdialogview.cpp weatherIface.skel weatherlib.cpp prefsDialog.cpp kweather.cpp dockwidget.cpp
 
 SUBDIRS = graphics
 
@@ -17,11 +17,8 @@
 bin_PROGRAMS = kweatherservice reportview
 
 kweatherservice_LDFLAGS = $(all_libraries) $(KDE_RPATH)
-kweatherservice_LDADD = $(LIB_KIO)
-kweatherservice_SOURCES = main.cpp weatherlib.cpp weatherservice.cpp weatherservice.skel
-
-service_DATA = kweatherservice.desktop
-servicedir = $(kde_servicesdir)
+kweatherservice_LDADD = $(LIB_KIO) weatherlib.lo weatherservice_skel.lo
+kweatherservice_SOURCES = main.cpp weatherservice.cpp
 
 METASOURCES = AUTO
 
Index: kweather/configure.in.in
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdetoys/kweather/configure.in.in,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
--- kweather/configure.in.in	14 Aug 2002 21:08:25 -0000	1.1.1.2
+++ kweather/configure.in.in	15 Aug 2002 21:07:41 -0000	1.2
@@ -1 +1 @@
-#DO_NOT_COMPILE="$DO_NOT_COMPILE kweather"
+DO_NOT_COMPILE="$DO_NOT_COMPILE kweather"
Index: kweather/dockwidget.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdetoys/kweather/dockwidget.cpp,v
retrieving revision 1.1.1.3
retrieving revision 1.2
diff -u -b -u -r1.1.1.3 -r1.2
--- kweather/dockwidget.cpp	14 Aug 2002 21:08:25 -0000	1.1.1.3
+++ kweather/dockwidget.cpp	15 Aug 2002 21:07:41 -0000	1.2
@@ -40,9 +40,11 @@
     isSmall = small;
 }
 
-void dockwidget::setWeatherIcon(const QPixmap &icon)
+void dockwidget::setWeatherIcon(QString icon)
 {
-    theIcon = QPixmap(icon);
+    QString theImage = "kweather/" + icon + ".png";
+    kdDebug(12004) << "Icon: " << theImage << endl;
+    theIcon = QImage(locate("data", theImage));
 }
 
 /** Change the button icon */
Index: kweather/dockwidget.h
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdetoys/kweather/dockwidget.h,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
--- kweather/dockwidget.h	14 Aug 2002 21:08:25 -0000	1.1.1.2
+++ kweather/dockwidget.h	15 Aug 2002 21:07:41 -0000	1.2
@@ -39,7 +39,7 @@
     /** Set the current wind speed and direction data. */
     void setWind( QString s);
     void setSmall(bool small);
-    void setWeatherIcon(const QPixmap &icon);
+    void setWeatherIcon(QString icon);
     void setFont(QFont newFont);
     void showWeather();
     /** resize the view **/
Index: kweather/kweather.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdetoys/kweather/kweather.cpp,v
retrieving revision 1.1.1.3
retrieving revision 1.2
diff -u -b -u -r1.1.1.3 -r1.2
--- kweather/kweather.cpp	14 Aug 2002 21:08:25 -0000	1.1.1.3
+++ kweather/kweather.cpp	15 Aug 2002 21:07:41 -0000	1.2
@@ -19,7 +19,6 @@
 #include <qcursor.h>
 #include <kaboutdata.h>
 #include <kaboutapplication.h>
-#include <kapplication.h>
 #include <kdebug.h>
 #include <klocale.h>
 #include <kiconloader.h>
@@ -50,9 +49,6 @@
 	firstrun = false;
 	loadPrefs();
 	
-	client = KApplication::kApplication()->dcopClient();
-	client->attach();
-
 	p = new dockwidget(this, "test");
 	connect(p, SIGNAL(buttonEvent()), SLOT(getButtonEvent()));
 	p->setFont(theFont);
@@ -135,17 +131,6 @@
 void kweather::doReport(){
     kdDebug(12004) << "Printing out the report" << endl;
     reportView *theReport = new reportView(reportLocation);
-	theReport->setTemp(temperature(reportLocation) );
-	theReport->setDewPoint(dewPoint(reportLocation) );
-	theReport->setRelHumidity( relHumidity(reportLocation));
-	theReport->setHeatIndex( heatIndex(reportLocation) );
-	theReport->setWindChill( windChill(reportLocation));
-	theReport->setWind( wind(reportLocation) );
-	theReport->setDate( date(reportLocation) );
-	theReport->setPressure( pressure(reportLocation) );
-	theReport->setCover( cover(reportLocation) );
-	theReport->setWeather( weather(reportLocation) );
-	theReport->setWeatherIcon(currentIconString(reportLocation)  );
 	theReport->render();
 	theReport->show();
 }
@@ -186,11 +171,11 @@
 
 void kweather::showWeather()
 {
-     p->setPressure(pressure(reportLocation));
-     p->setWind(wind(reportLocation));
-     p->setTemperature(temperature(reportLocation));
-     p->setSmall(smallviewMode);
-     p->setWeatherIcon(currentIcon(reportLocation));
+//     p->setPressure(weather->pressure());
+//     p->setWind(weather->wind());
+//     p->setTemperature(weather->temp());
+//     p->setSmall(smallviewMode);
+//     p->setWeatherIcon(weather->currentIcon());
 
     p->showWeather();
 
@@ -207,20 +192,20 @@
         QTextStream logFileStream(&logFile);
         if (logFile.open(IO_Append | IO_ReadWrite))
         {
-            logFileStream << date(reportLocation);
+/*            logFileStream << weather->date();
             logFileStream << ",";
-            logFileStream << wind(reportLocation);
+            logFileStream << weather->wind();
             logFileStream << ",";
-            logFileStream << temperature(reportLocation);
+            logFileStream << weather->temp();
             logFileStream << ",";
-            logFileStream << pressure(reportLocation);
+            logFileStream << weather->pressure();
             logFileStream << ",";
-            logFileStream << cover(reportLocation).join(";");
+            logFileStream << weather->cover().join(";");
             logFileStream << ",";
-            logFileStream << visibility(reportLocation);
+            logFileStream << weather->visibility();
             logFileStream << ",";
-            logFileStream << weather(reportLocation).join(";");
-            logFileStream << endl;
+            logFileStream << weather->weather().join(";");
+            logFileStream << endl;*/
         }
         else
         {
@@ -243,10 +228,11 @@
 
     if ( !offlineMode )
     {
-        // Make a dcop connection to the weather service and update the UI
-	update(reportLocation);
-	showWeather();
-	writeLogEntry();
+        //url = "ftp://weather.noaa.gov/data/observations/metar/stations/";
+        QString url = "http://weather.noaa.gov/pub/data/observations/metar/stations/";
+        url += reportLocation.upper().stripWhiteSpace();
+        url += ".TXT";
+        metarData = QString::null;
     }
 }
 
@@ -330,328 +316,4 @@
    mPrefs = 0L;
 }
 
-QString kweather::temperature(QString stationID)
-{
-	if (attach())
-	{
-		QCString replyType;
-		QByteArray data,replyData;
-		QDataStream arg(data, IO_WriteOnly);
-		arg << stationID;
-		if(client->call("KWeatherService", "WeatherService", "temperature(QString)", data,replyType, replyData, true))
-		{
-			QDataStream reply(replyData, IO_ReadOnly);
-			if (replyType == "QString") 
-			{
-				QString result;
-				reply >> result;
-				kdDebug() << result << endl;
-				return result;
-			} 
-		}	
-		else
-			kdDebug() << "failed " << endl;
-	}
-	return "";
-}
-
-QString kweather::dewPoint(QString stationID)
-{
-	if (attach())
-	{
-		QCString replyType;
-		QByteArray data,replyData;
-		QDataStream arg(data, IO_WriteOnly);
-		arg << stationID;
-		if(client->call("KWeatherService", "WeatherService", "dewPoint(QString)", data,replyType, replyData, true))
-		{
-			QDataStream reply(replyData, IO_ReadOnly);
-			if (replyType == "QString") 
-			{
-				QString result;
-				reply >> result;
-				return result;
-			} 
-		}	
-	}
-	return "";
-}
-QString kweather::relHumidity(QString stationID)
-{
-	if (attach())
-	{
-
-		QCString replyType;
-		QByteArray data,replyData;
-		QDataStream arg(data, IO_WriteOnly);
-		arg << stationID;
-		if(client->call("KWeatherService", "WeatherService", "relitiveHumidity(QString)", data,replyType, replyData, true))
-		{
-			QDataStream reply(replyData, IO_ReadOnly);
-			if (replyType == "QString") 
-			{
-				QString result;
-				reply >> result;
-				return result;
-			} 
-		}
-			
-	}
-	
-	return "";
-}
-QString kweather::heatIndex(QString stationID)
-{
-	if (attach())
-	{
-
-		QCString replyType;
-		QByteArray data,replyData;
-		QDataStream arg(data, IO_WriteOnly);
-		arg << stationID;
-		if(client->call("KWeatherService", "WeatherService", "heatIndex(QString)", data,replyType, replyData, true))
-		{
-			QDataStream reply(replyData, IO_ReadOnly);
-			if (replyType == "QString") 
-			{
-				QString result;
-				reply >> result;
-				return result;
-			} 
-		}
-	}
-	return "";	
-}
-QString kweather::windChill(QString stationID)
-{
-	if (attach())
-	{
-
-		QCString replyType;
-		QByteArray data,replyData;
-		QDataStream arg(data, IO_WriteOnly);
-		arg << stationID;
-		if(client->call("KWeatherService", "WeatherService", "windChill(QString)", data,replyType, replyData, true))
-		{
-			QDataStream reply(replyData, IO_ReadOnly);
-			if (replyType == "QString") 
-			{
-				QString result;
-				reply >> result;
-				return result;
-			} 
-		}
-	}
-	return "";	
-}
-QString kweather::wind(QString stationID)
-{
-	if (attach())
-	{
-
-		QCString replyType;
-		QByteArray data,replyData;
-		QDataStream arg(data, IO_WriteOnly);
-		arg << stationID;
-		if(client->call("KWeatherService", "WeatherService", "wind(QString)", data,replyType, replyData, true))
-		{
-			QDataStream reply(replyData, IO_ReadOnly);
-			if (replyType == "QString") 
-			{
-				QString result;
-				reply >> result;
-				return result;
-			} 
-		}
-	}
-	return "";	
-}
-QString kweather::pressure(QString stationID)
-{
-	if (attach())
-	{
-
-		QCString replyType;
-		QByteArray data,replyData;
-		QDataStream arg(data, IO_WriteOnly);
-		arg << stationID;
-		if(client->call("KWeatherService", "WeatherService", "pressure(QString)", data,replyType, replyData, true))
-		{
-			QDataStream reply(replyData, IO_ReadOnly);
-			if (replyType == "QString") 
-			{
-				QString result;
-				reply >> result;
-				return result;
-			} 
-		}	
-	}
-	return "";
-}
-QPixmap kweather::currentIcon(QString stationID)
-{
-	if (attach())
-	{
-
-		QCString replyType;
-		QByteArray data,replyData;
-		QDataStream arg(data, IO_WriteOnly);
-		arg << stationID;
-		if(client->call("KWeatherService", "WeatherService", "currentIcon(QString)", data,replyType, replyData, true))
-		{
-			QDataStream reply(replyData, IO_ReadOnly);
-			if (replyType == "QPixmap") 
-			{
-				QPixmap result;
-				reply >> result;
-				return result;
-			} 
-		}	
-	}
-	return QPixmap();
-}
-QString kweather::currentIconString(QString stationID)
-{
-	if (attach())
-	{
-
-		QCString replyType;
-		QByteArray data,replyData;
-		QDataStream arg(data, IO_WriteOnly);
-		arg << stationID;
-		if(client->call("KWeatherService", "WeatherService", "currentIconString(QString)", data,replyType, replyData, true))
-		{
-			QDataStream reply(replyData, IO_ReadOnly);
-			if (replyType == "QString") 
-			{
-				QString result;
-				reply >> result;
-				return result;
-			} 
-		}	
-	}
-	return "";
-}
-
-QString kweather::date(QString stationID)
-{
-	if (attach())
-	{
-
-		QCString replyType;
-		QByteArray data,replyData;
-		QDataStream arg(data, IO_WriteOnly);
-		arg << stationID;
-		if(client->call("KWeatherService", "WeatherService", "date(QString)", data, replyType, replyData, true))
-		{
-			QDataStream reply(replyData, IO_ReadOnly);
-			if (replyType == "QString") 
-			{
-				QString result;
-				reply >> result;
-				return result;
-			} 
-		}
-			
-	}
-	return "";
-}
-QStringList kweather::weather(QString stationID)
-{
-	if (attach())
-	{
-
-		QCString replyType;
-		QByteArray data,replyData;
-		QDataStream arg(data, IO_WriteOnly);
-		arg << stationID;
-		if(client->call("KWeatherService", "WeatherService", "weather(QString)", data,replyType, replyData, true))
-		{
-			QDataStream reply(replyData, IO_ReadOnly);
-			if (replyType == "QStringList") 
-			{
-				QStringList result;
-				reply >> result;
-				return result;
-			} 
-		}
-	}
-	return "";	
-
-}
-QString kweather::visibility(QString stationID)
-{
-	if (attach())
-	{
-
-		QCString replyType;
-		QByteArray data,replyData;
-		QDataStream arg(data, IO_WriteOnly);
-		arg << stationID;
-		if(client->call("KWeatherService", "WeatherService", "visibility(QString)", data,replyType, replyData, true))
-		{
-			QDataStream reply(replyData, IO_ReadOnly);
-			if (replyType == "QString") 
-			{
-				QString result;
-				reply >> result;
-				return result;
-			} 
-		}
-	}
-	return "";	
-}
-QStringList kweather::cover(QString stationID)
-{
-	if (attach())
-	{
-
-		QCString replyType;
-		QByteArray data,replyData;
-		QDataStream arg(data, IO_WriteOnly);
-		arg << stationID;
-		if(client->call("KWeatherService", "WeatherService", "cover(QString)", data,replyType, replyData, true))
-		{
-			QDataStream reply(replyData, IO_ReadOnly);
-			if (replyType == "QStringList") 
-			{
-				QStringList result;
-				reply >> result;
-				return result;
-			} 
-		}
-	}
-	return "";	
-}
-void kweather::update(QString stationID)
-{
-	if (attach())
-	{
-
-		QCString replyType;
-		QByteArray data,replyData;
-		QDataStream arg(data, IO_WriteOnly);
-		arg << stationID;
-		client->call("KWeatherService", "WeatherService", "forceUpdate(QString)", data,replyType, replyData, true);
-	
-	}
-}
-	
-bool kweather::attach()
-{
-               QString error;
-                QCString appID;
-                kdDebug() << "Looking for dcop service..." << endl;
-                if (!client->isApplicationRegistered("KWeatherService"))
-                {
-                        kdDebug() << "Could not find service so I am starting it..." << endl;
-                        if(KApplication::startServiceByName("kweatherservice",QStringList(), &error,&appID ))
-
-                        {
-                                kdDebug() << "Starting KWeatherService failed with message: " << error << endl;
-                                return false;
-                        }
-                }
-                return true;
-}
 #include "kweather.moc"
Index: kweather/kweather.desktop
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdetoys/kweather/kweather.desktop,v
retrieving revision 1.1.1.4
retrieving revision 1.2
diff -u -b -u -r1.1.1.4 -r1.2
--- kweather/kweather.desktop	14 Aug 2002 21:08:25 -0000	1.1.1.4
+++ kweather/kweather.desktop	15 Aug 2002 21:07:41 -0000	1.2
@@ -3,31 +3,23 @@
 Encoding=UTF-8
 Comment=A weather reporting panel applet
 Comment[bs]=Applet za panel sa prognozom vremena
-Comment[da]=Et vejrrapporteringspanelprogram
-Comment[de]=Wetterberichte für die Kontrollleiste
+Comment[da]=Et vejrrapport panelprogram
 Comment[el]=Ένα εφαρμογίδιο του πίνακα για την αναφορά των καιρικών συνθηκών
 Comment[es]=Un applet informador de meteorología para el panel
-Comment[fi]=Sääsovelma
 Comment[he]=יישומון דיווח מזג אוויר עבור הלוח
 Comment[hr]=Applet za ploču sa vremenskom prognozom
-Comment[hu]=Időjárásfigyelő panel-kisalkalmazás
-Comment[ja]=天気をレポートするパネルアプレット
-Comment[pt]=Uma 'applet' de painel com boletins metereológicos
-Comment[pt_BR]=Um mini-aplicativo de painel com o relatório do tempo
+Comment[pt_BR]=Um miniaplicativo de painel com o relatório do tempo
 Comment[ro]=O aplicaţie de panou pentru raportarea vremii
-Comment[sl]=Vstavek za poročanje o vremenu
 Comment[sv]=Ett miniprogram till panelen för väderrapporter
-Comment[th]=แอพเพล็ตรายงานอากาศบนถาดพาเนล
 Comment[tr]=Bir hava durumu raporlayıcı programcığı
 Name=KWeather
 Name[af]=k-weer
-Name[de]=KWetter
+Name[be]=K Надвор'е
 Name[es]=Kweather
+Name[fo]=KVeður
 Name[he]=צג מזג אוויר
 Name[hr]=Vremenska Prognoza
 Name[ro]=Kweather
-Name[sl]=KVreme
 Name[sv]=Kweather
-Name[th]=รายงานอากาศ
 X-KDE-Library=weather_panelapplet
 X-KDE-UniqueApplet=true
Index: kweather/kweather.h
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdetoys/kweather/kweather.h,v
retrieving revision 1.1.1.3
retrieving revision 1.2
diff -u -b -u -r1.1.1.3 -r1.2
--- kweather/kweather.h	14 Aug 2002 21:08:25 -0000	1.1.1.3
+++ kweather/kweather.h	15 Aug 2002 21:07:41 -0000	1.2
@@ -31,9 +31,10 @@
 #include <klocale.h>
 #include <qpixmap.h>
 #include <qfont.h>
-#include <dcopclient.h>
 
 #include "prefsDialog.h"
+#include "weatherIface.h"
+#include "weatherlib.h"
 #include "reportview.h"
 
 class prefsDialog;
@@ -66,23 +67,6 @@
 	void writeLogEntry();
 	void mousePressEvent(QMouseEvent * ev);
 	
-	// DCOP service wrapper
-	QString temperature(QString stationID);
-	QString dewPoint(QString stationID);
-	QString relHumidity(QString stationID);
-	QString heatIndex(QString stationID);
-	QString windChill(QString stationID);
-	QString wind(QString stationID);
-	QString pressure(QString stationID);
-	QPixmap currentIcon(QString stationID);
-	QString currentIconString(QString stationID);
-	QString date(QString stationID);
-	QStringList weather(QString stationID);
-	QString visibility(QString stationID);
-	QStringList cover(QString stationID);
-	void update(QString stationID);
-	bool attach();
-		
 	QString reportLocation;
 	int interval;
 	QString fileName;
@@ -98,7 +82,6 @@
 	KIO::TransferJob *theJob;
 	QString metarData;
 	prefsDialog *mPrefs;
-	DCOPClient *client;
 };
 
 #endif
Index: kweather/kweatherservice.desktop
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdetoys/kweather/kweatherservice.desktop,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
--- kweather/kweatherservice.desktop	14 Aug 2002 21:08:25 -0000	1.1.1.2
+++ kweather/kweatherservice.desktop	15 Aug 2002 21:07:41 -0000	1.2
@@ -1,13 +1,12 @@
 [Desktop Entry]
 Type=Service
 Name=kweatherservice
-Name[de]=KWetter-Service
+Name[be]=K Сэрвіс надвор'я
+Name[fo]=KVeðurtíðindi
 Name[hu]=KWeatherService
 Name[pt_BR]=Serviço Kweather
 Name[ro]=Serviciu Kweather
-Name[sl]=KVremenskiServis
 Name[sv]=Kweather-tjänst
-Name[th]=รายงานอากาศ
 Exec=kweatherservice
 X-DCOP-ServiceType=Unique
 X-KDE-StartupNotify=false
Index: kweather/main.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdetoys/kweather/main.cpp,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
--- kweather/main.cpp	14 Aug 2002 21:08:25 -0000	1.1.1.2
+++ kweather/main.cpp	15 Aug 2002 21:07:41 -0000	1.2
@@ -1,4 +1,4 @@
-/* $Id: main.cpp,v 1.2 2002/08/01 19:45:52 geiseri Exp $ */
+/* $Id: main.cpp,v 1.1 2002/04/05 20:07:04 geiseri Exp $ */
 
 #include <kuniqueapplication.h>
 #include <kaboutdata.h>
@@ -12,8 +12,8 @@
 int main (int argc, char *argv[])
 {
 	KLocale::setMainCatalogue("kdelibs");
-	KAboutData aboutdata("KWeatherService", I18N_NOOP("KDE"),
-				"0.8", I18N_NOOP("KWeather DCOP Service"),
+	KAboutData aboutdata("dcopservice", I18N_NOOP("KDE"),
+				"0.8", I18N_NOOP("Template DCOP Service"),
 				KAboutData::License_GPL, "(C) 2002, Ian Reinhart Geiser");
 	aboutdata.addAuthor("Ian Reinhart Geiser",I18N_NOOP("Developer"),"geiseri@kde.org");
 	aboutdata.addAuthor("Nadeem Hasan",I18N_NOOP("Developer"),"nhasan@kde.org");
Index: kweather/reportview.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdetoys/kweather/reportview.cpp,v
retrieving revision 1.1.1.3
retrieving revision 1.2
diff -u -b -u -r1.1.1.3 -r1.2
--- kweather/reportview.cpp	14 Aug 2002 21:08:25 -0000	1.1.1.3
+++ kweather/reportview.cpp	15 Aug 2002 21:07:41 -0000	1.2
@@ -160,8 +160,8 @@
 }
 
 void reportView::setWeatherIcon( const QString  &icon ){
-	QString theImage = "kweather/" + icon + ".png";
-	currentIcon = locate("data", theImage);
+    currentIcon = locate("data", "kweather/" + icon + ".png");
+    setIcon(QPixmap(currentIcon));
 }
 
 #include "reportview.moc"
Index: kweather/weatherlib.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdetoys/kweather/weatherlib.cpp,v
retrieving revision 1.1.1.3
retrieving revision 1.2
diff -u -b -u -r1.1.1.3 -r1.2
--- kweather/weatherlib.cpp	14 Aug 2002 21:08:25 -0000	1.1.1.3
+++ kweather/weatherlib.cpp	15 Aug 2002 21:07:41 -0000	1.2
@@ -16,10 +16,8 @@
 ***************************************************************************/
 
 #include <iostream.h>
-#include <qfile.h>
 #include <klocale.h>
 #include <qdatetime.h>
-#include <qtextstream.h>
 #include <kglobal.h>
 #include <klocale.h>
 #include <kdebug.h>
@@ -28,7 +26,6 @@
 #include <kio/netaccess.h>
 #include <kurl.h>
 #include <knotifyclient.h>
-#include <ktempfile.h>
 
 #include "weatherlib.h"
 
@@ -43,12 +40,12 @@
 			tempF = 0;
 			qsCoverList.clear();
 			qsCurrentList.clear();
-			qsDate = QDate::currentDate();
+			qsDate = QString::null;
 			qsPressure = QString::null;
 			qsTemperature = QString::null;
 			qsDewPoint = QString::null;
 			qsRelHumidity = QString::null;
-			qsTime = QTime::currentTime();
+			qsTime = QString::null;
 			qsVisibility = QString::null;
 			qsWindSpeed = QString::null;
 			qsWindChill = QString::null;
@@ -67,12 +64,12 @@
 		float tempF;
 		QStringList qsCoverList;
 		QStringList qsCurrentList;
-		QDate qsDate;
+		QString qsDate;
 		QString qsPressure;
 		QString qsTemperature;
 		QString qsDewPoint;
 		QString qsRelHumidity;
-		QTime qsTime;
+		QString qsTime;
 		QString qsVisibility;
 		QString qsWindSpeed;
 		QString qsWindChill;
@@ -87,11 +84,11 @@
     qsCoverList.clear();
     qsCurrentList.clear();
     qsWindDirection = "";
-    qsTime = QTime::currentTime();
+    qsTime = "";
     qsTemperature = "";
     theWeather = "";
     qsWindSpeed = "";
-    qsDate = QDate::currentDate();
+    qsDate = "";
     qsPressure = "";
     qsVisibility = "";
 }
@@ -124,51 +121,40 @@
 		QString sClouds;
 		QString skycondition = "";
 
-		if (KGlobal::locale()->measureSystem() == KLocale::Metric)
-		{
-			fFeet = fFeet * 0.3048;  // Check this!
-		}
-		
 		if (fFeet >= 10)
 		{
 			fFeet /= 10;
-			if (KGlobal::locale()->measureSystem() == KLocale::Metric)
-				sClouds = QString("%1 %2").arg(fFeet).arg(i18n(" kilometers"));
-			else
-				sClouds = QString("%1 %2").arg(fFeet).arg(i18n(" thousand feet"));
+			sClouds = QString("%1 %2").arg(fFeet).arg(i18n(" thousand"));
 		}
 		else
 		{
 			fFeet *= 100;
-			if (KGlobal::locale()->measureSystem() == KLocale::Metric)
-				sClouds = QString("%1 %2").arg(fFeet).arg(i18n("meters"));
-			else
-				sClouds = QString("%1 %2").arg(fFeet).arg(i18n("feet"));
+			sClouds = QString("%1").arg(fFeet);
 		}
 
 		if (sCode == "FEW")
 		{
-			skycondition = i18n( "Few clouds at %1." ).arg(sClouds);
+			skycondition = i18n( "Few clouds at %1 feet" ).arg(sClouds);
 			d->clouds += 2;
 		}
 		else if (sCode == "SCT")
 		{
-			skycondition = i18n( "Scattered clouds at %1." ).arg(sClouds);
+			skycondition = i18n( "Scattered clouds at %1 feet" ).arg(sClouds);
 			d->clouds += 4;
 		}
 		else if (sCode == "BKN")
 		{
-			skycondition = i18n( "Broken clouds at %1." ).arg(sClouds);
+			skycondition = i18n( "Broken clouds at %1 feet" ).arg(sClouds);
 			d->clouds += 8;
 		}
 		else if (sCode == "OVC")
 		{
-			skycondition = i18n( "Overcast clouds at %1." ).arg(sClouds);
+			skycondition = i18n( "Overcast clouds at %1 feet" ).arg(sClouds);
 			d->clouds += 64;
 		}
 		else if ((sCode == "CLR") || (sCode == "SKC") || (sCode == "CAVOK"))
 		{
-			skycondition = i18n("Clear skies.");
+			skycondition = i18n("Clear skies");
 		}
 
 		kdDebug(12004) << "*** Clouds: " << d->clouds << endl;
@@ -344,7 +330,7 @@
 				- (0.00000199*SQR(d->tempF)*SQR(fRelHumidity));
 		}
 
-		if (KGlobal::locale()->measureSystem() == KLocale::Metric)
+		if (MetricMode)
 		{
 			unit = i18n("°C");
 			d->qsTemperature.setNum(fTemp);
@@ -375,17 +361,8 @@
 	if (DateRegExp.search(s) > -1)
 	{
 		kdDebug(12004) << "Date: " << DateRegExp.capturedTexts().join("-") << endl;
-		QString dateString = DateRegExp.cap(1);
-		QString day, month, year;
-
-		day = dateString.mid(8,2);
-		month = dateString.mid(5,2);
-		year = dateString.mid(0,4);
-		
-		QDate theDate(year.toInt(), month.toInt(), day.toInt());
 
-		
-		d->qsDate = theDate;
+		d->qsDate = DateRegExp.cap(1);
 		return true;
 	}
 	return false;
@@ -397,14 +374,7 @@
 	{
 		kdDebug(12004) << "Time: " << TimeRegExp.capturedTexts().join("-") << endl;
 
-		QString hour, minute, dateString;
-
-		dateString = TimeRegExp.cap(1);
-		hour = dateString.mid(0,2);
-		minute = dateString.mid(3,2);
-		QTime theTime(hour.toInt(), minute.toInt());
-
-		d->qsTime = theTime;
+		d->qsTime = TimeRegExp.cap(1);
 		return true;
 	}
 	return false;
@@ -418,7 +388,7 @@
 
 		kdDebug(12004) << "Visibility: " << VisRegExp.capturedTexts().join("-") << endl;
 
-		if (KGlobal::locale()->measureSystem() == KLocale::Metric)
+		if (MetricMode)
 		{
 			fVisibility *= 1.6;
 			d->qsVisibility.setNum(fVisibility);
@@ -443,7 +413,7 @@
 
 		kdDebug(12004) << "Pressure: " << PressRegExp.capturedTexts().join("-") << endl;
 
-		if (KGlobal::locale()->measureSystem() == KLocale::Metric)
+		if (MetricMode)
 		{
 			if (type == "A")
 				fPressure *= (33.8639/100);
@@ -476,7 +446,7 @@
 
 		kdDebug(12004) << "Wind: " << WindRegExp.capturedTexts().join("-") << endl;
 
-		if (KGlobal::locale()->measureSystem() == KLocale::Metric)
+		if (MetricMode)
 		{
 			if (sWindUnit == "KT")
 			{
@@ -615,7 +585,7 @@
 
 	if (windChill < 48)
 	{
-		if (KGlobal::locale()->measureSystem() == KLocale::Metric)
+		if (MetricMode)
 		{
 			d->qsWindChill.setNum((windChill-32)*5/9, 'f', 1);
 			d->qsWindChill += i18n("°C");
@@ -675,44 +645,17 @@
 
 bool WeatherLib::getData(Data *d)
 {
-	QString u = "http://weather.noaa.gov/pub/data/observations/metar/stations/";
-        u += d->reportLocation.upper().stripWhiteSpace();
-        u += ".TXT";
-	
-	KTempFile *target = new KTempFile(QString::null, "-weather");
-	target->setAutoDelete(false);
-	KURL url(u);
-	QString local(target->name());
+	KURL url("some url here");
+	QString target;
 	bool isGood = true;
 
-    if (!KIO::NetAccess::download(url, local))
+    if (!KIO::NetAccess::download(url, target))
 	{
 		KNotifyClient::event("cannotopenfile");
 		kdDebug() << "Could not read: " << url.prettyURL() << endl;
 		isGood = false;
 	}
-	else
-	{
-		kdDebug() << "Copied " << url.prettyURL() << " to " << local << endl;
-		QFile f(target->name());
-		if ( f.open(IO_ReadOnly) )
-		{
-			QTextStream t(&f);
-			QString s;
-			while ( !t.eof() ) 
-			{ 
-				s += " " + t.readLine();
-				kdDebug() << s << endl;
-			}
-			processData(s, d);
-		}
-		else
-		{
-			isGood = false;
-		}
-		f.close();
-	}
-	delete target;
+
 	return isGood;
 }
 
@@ -722,9 +665,6 @@
 	if (!d)
 	{
 		d = new Data();
-		d->reportLocation = stationID;
-		data.insert(stationID, d);
-		getData(d);
 	}
 
 	return d;
@@ -773,11 +713,22 @@
 
 QString WeatherLib::date(QString stationID){
 	Data *d = findData(stationID);
+	QString day, month, year, hour, minute;
+
+	day = d->qsDate.mid(8,2);
+	month = d->qsDate.mid(5,2);
+	year = d->qsDate.mid(0,4);
 	
-	QDateTime gmtDateTime(d->qsDate, d->qsTime);
+	hour = d->qsTime.mid(0,2);
+	minute = d->qsTime.mid(3,2);
+
+	QDate theDate(year.toInt(), month.toInt(), day.toInt());
+	QTime theTime(hour.toInt(), minute.toInt());
+	QDateTime gmtDateTime(theDate, theTime);
 	QDateTime localDateTime = gmtDateTime.addSecs(KRFCDate::localUTCOffset() * 60);
-	//KLocale *loc = KGlobal::locale();
-	return KGlobal::locale()->formatDateTime(localDateTime, false, false);
+
+	KLocale *loc = KGlobal::locale();
+	return loc->formatDateTime(localDateTime, false, false);
 }
 
 /** Returns the current cover */
@@ -796,10 +747,4 @@
 QStringList WeatherLib::weather(QString stationID){
 	Data *d = findData(stationID);
 	return d->qsCurrentList;
-}
-
-void WeatherLib::update(QString stationID)
-{
-	Data *d = findData(stationID);
-	getData(d);
 }
Index: kweather/weatherlib.h
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdetoys/kweather/weatherlib.h,v
retrieving revision 1.1.1.3
retrieving revision 1.2
diff -u -b -u -r1.1.1.3 -r1.2
--- kweather/weatherlib.h	14 Aug 2002 21:08:25 -0000	1.1.1.3
+++ kweather/weatherlib.h	15 Aug 2002 21:07:41 -0000	1.2
@@ -42,17 +42,16 @@
 		QStringList weather(QString stationID);
 		QString visibility(QString stationID);
 		QStringList cover(QString stationID);
-		void update(QString stationID);
 
 	private:
-		bool parsePressure(QString tag, Data *d);
-		bool parseVisibility(QString tag, Data *d);
-		bool parseTime(QString tag, Data *d);
-		bool parseTemperature(QString tag, Data *d);
-		bool parseDate(QString tag, Data *d);
-		bool parseCurrent(QString tag, Data *d);
-		bool parseCover(QString tag, Data *d);
-		bool parseWindSpeed(QString tag, Data *d);
+		bool parsePressure(QString stationID, Data *d);
+		bool parseVisibility(QString stationID, Data *d);
+		bool parseTime(QString stationID, Data *d);
+		bool parseTemperature(QString stationID, Data *d);
+		bool parseDate(QString stationID, Data *d);
+		bool parseCurrent(QString stationID, Data *d);
+		bool parseCover(QString stationID, Data *d);
+		bool parseWindSpeed(QString stationID, Data *d);
 		void calcCurrentIcon(Data *d);
 		void calcWindChill(Data *d);
 
Index: kweather/weatherservice.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdetoys/kweather/weatherservice.cpp,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
--- kweather/weatherservice.cpp	14 Aug 2002 21:08:25 -0000	1.1.1.2
+++ kweather/weatherservice.cpp	15 Aug 2002 21:07:41 -0000	1.2
@@ -1,4 +1,4 @@
-/* $Id: weatherservice.cpp,v 1.9 2002/08/05 01:00:00 geiseri Exp $ */
+/* $Id: weatherservice.cpp,v 1.6 2002/04/15 20:51:35 geiseri Exp $ */
 #include <kdebug.h>
 #include <kapplication.h>
 #include <kconfig.h>
@@ -25,38 +25,37 @@
 
 QString WeatherService::temperature(QString stationID)
 {
-	kdDebug () << "Returning " << stationID << endl;
-	return m_weatherLib->temperature(stationID);
+
 }
 
 QString WeatherService::dewPoint(QString stationID)
 {
-	return m_weatherLib->dewPoint(stationID);
+
 }
 
 QString WeatherService::relitiveHumidity(QString stationID)
 {
-	return m_weatherLib->relHumidity(stationID);
+
 }
 
 QString WeatherService::heatIndex(QString stationID)
 {
-	return m_weatherLib->heatIndex(stationID);
+
 }
 
 QString WeatherService::windChill(QString stationID)
 {
-	return m_weatherLib->windChill(stationID);
+
 }
 
 QString WeatherService::wind(QString stationID)
 {
-	return m_weatherLib->wind(stationID);
+
 }
 
 QString WeatherService::pressure(QString stationID)
 {
-	return m_weatherLib->pressure(stationID);
+
 }
 
 QPixmap WeatherService::currentIcon(QString stationID)
@@ -69,34 +68,29 @@
 	return theIcon;
 }
 
-QString WeatherService::currentIconString(QString stationID)
-{
-	return m_weatherLib->currentIcon(stationID);
-}
-
 QString WeatherService::date(QString stationID)
 {
-	return m_weatherLib->date(stationID);
+
 }
 
 QString WeatherService::visability(QString stationID)
 {
-	return m_weatherLib->visibility(stationID);
+
 }
 
 QStringList WeatherService::cover(QString stationID)
 {
-	return m_weatherLib->cover(stationID);
+
 }
 
 QStringList WeatherService::weather(QString stationID)
 {
-	return m_weatherLib->weather(stationID);
+
 }
 
 bool WeatherService::forceUpdate(QString stationID)
 {
-	m_weatherLib->update(stationID);
+
 }
 
 bool WeatherService::removeStation(QString stationID)
@@ -106,7 +100,7 @@
 
 bool WeatherService::addStation(QString stationID)
 {
-	m_weatherLib->update(stationID);
+
 }
 
 void WeatherService::setMaxAge(int age)
@@ -121,6 +115,6 @@
 
 bool WeatherService::exit()
 {
-	return kapp->quit();
+	kapp->quit();
 }
 
Index: kweather/weatherservice.h
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdetoys/kweather/weatherservice.h,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
--- kweather/weatherservice.h	14 Aug 2002 21:08:25 -0000	1.1.1.2
+++ kweather/weatherservice.h	15 Aug 2002 21:07:41 -0000	1.2
@@ -1,4 +1,4 @@
-/* $Id: weatherservice.h,v 1.7 2002/08/05 01:00:00 geiseri Exp $ */
+/* $Id: weatherservice.h,v 1.6 2002/04/15 20:51:35 geiseri Exp $ */
 #ifndef _WEATHER_SERVICE
 #define _WEATHER_SERVICE
 
@@ -30,7 +30,6 @@
 		QString wind(QString stationID);
 		QString pressure(QString stationID);
 		QPixmap currentIcon(QString stationID);
-		QString currentIconString(QString stationID);
 		QString date(QString stationID);
 		QString visability(QString stationID);
 		QStringList cover(QString stationID);
Index: kweather/weathersig.sh
===================================================================
RCS file: kweather/weathersig.sh
diff -N kweather/weathersig.sh
--- kweather/weathersig.sh	14 Aug 2002 21:08:25 -0000	1.1.1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,9 +0,0 @@
-#!/bin/sh
-echo +KWeather+===========================================
-LOC=KPNE
-TEMP=`dcop KWeatherService WeatherService temperature $LOC`
-WIND=`dcop KWeatherService WeatherService wind $LOC`
-DATE=`dcop KWeatherService WeatherService date $LOC`
-echo "	Report for $DATE"
-echo "	$TEMP with winds at $WIND"
-echo =====================================================
