Index: Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/Makefile.am,v
retrieving revision 1.1.1.1
retrieving revision 1.3
diff -u -b -u -r1.1.1.1 -r1.3
Index: admin/Makefile.common
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/admin/Makefile.common,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
Index: admin/acinclude.m4.in
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/admin/acinclude.m4.in,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
--- admin/acinclude.m4.in	10 Aug 2002 13:45:46 -0000	1.1.1.3
+++ admin/acinclude.m4.in	10 Aug 2002 14:07:14 -0000	1.4
@@ -699,13 +699,10 @@
      AC_CHECK_LIB(ipc, shmat, X_EXTRA_LIBS="$X_EXTRA_LIBS -lipc"))
    
    # darwin needs this to initialize the environment
-   AC_CHECK_HEADERS(crt_externs.h)
+   AC_CHECK_HEADERS([crt_externs.h sys/types.h])
    AC_CHECK_FUNC(_NSGetEnviron, [AC_DEFINE(HAVE_NSGETENVIRON, 1, [Define if your system needs _NSGetEnviron to set up the environment])])
  
-   # more headers that need to be explicitly included on darwin
-   AC_CHECK_HEADERS(sys/types.h stdint.h)
-
-   # darwin requires a poll emulation library
+   # darwin requires a poll.h emulation library
    AC_CHECK_LIB(poll, poll, LIB_POLL="-lpoll")
 
    # CoreAudio framework
@@ -736,6 +733,8 @@
    AC_SUBST(FRAMEWORK_COREAUDIO)
    LIBSOCKET="$X_EXTRA_LIBS"
    AC_SUBST(LIBSOCKET)
+   AC_SUBST(FRAMEWORK_COREAUDIO)
+   AC_SUBST(LIB_POLL)
    AC_SUBST(LIBRESOLV)
    AC_SUBST(X_EXTRA_LIBS)
    AC_CHECK_LIB(ucb, killpg, [LIBUCB="-lucb"]) dnl for Solaris2.4
@@ -1846,15 +1845,29 @@
    AC_REQUIRE([K_PATH_X])
 
 if test $kde_qtver = 3; then
+   AC_SUBST(LIB_ARTSKDE, "-lartskde")
+   AC_SUBST(LIB_DCOP, "-lDCOP")
+   AC_SUBST(LIB_KAB, "-lkab")
+   AC_SUBST(LIB_KABC, "-lkabc")
    AC_SUBST(LIB_KDECORE, "-lkdecore")
+   AC_SUBST(LIB_KDED, "@PREFIX@/lib/kded.la")
+   AC_SUBST(LIB_KDEFX, "-lkdefx")
+   AC_SUBST(LIB_KDEPRINT, "-lkdeprint")
+   AC_SUBST(LIB_KDESASL, "-lkdesasl")
+   AC_SUBST(LIB_KDESU, "-lkdesu")
    AC_SUBST(LIB_KDEUI, "-lkdeui")
-   AC_SUBST(LIB_KIO, "-lkio")
-   AC_SUBST(LIB_SMB, "-lsmb")
-   AC_SUBST(LIB_KAB, "-lkab")
    AC_SUBST(LIB_KHTML, "-lkhtml")
-   AC_SUBST(LIB_KSPELL, "-lkspell")
+   AC_SUBST(LIB_KIO, "-lkio")
    AC_SUBST(LIB_KPARTS, "-lkparts")
-   AC_SUBST(LIB_KDEPRINT, "-lkdeprint")
+   AC_SUBST(LIB_KJAVA, "-lkjava")
+   AC_SUBST(LIB_KJS, "-lkjs")
+   AC_SUBST(LIB_KMID, "-lkmid")
+   AC_SUBST(LIB_KSCRIPT, "-lkscript")
+   AC_SUBST(LIB_KSPELL, "-lkspell")
+   AC_SUBST(LIB_KTEXTEDITOR, "-lktexteditor")
+   AC_SUBST(LIB_SHELLSCRIPT, "-lshellscript")
+   AC_SUBST(LIB_SMB, "-lsmb")
+   AC_SUBST(LIB_VCARD, "-lvcard")
 # these are for backward compatibility
    AC_SUBST(LIB_KSYCOCA, "-lkio")
    AC_SUBST(LIB_KFILE, "-lkio")
@@ -4106,7 +4119,7 @@
 AC_DEFUN(KDE_CREATE_SUBDIRSLIST,
 [
 
-DO_NOT_COMPILE="$DO_NOT_COMPILE CVS debian bsd-port admin"
+DO_NOT_COMPILE="$DO_NOT_COMPILE CVS debian bsd-port admin kdm ksysguard kscd"
 
 if test ! -s $srcdir/subdirs; then
   dnl Note: Makefile.common creates subdirs, so this is just a fallback
@@ -4339,6 +4352,12 @@
   # And delete superflous '/' to make compares easier
   prefix=`echo "$prefix" | sed 's,//*,/,g' | sed -e 's,/$,,'`
   exec_prefix=`echo "$exec_prefix" | sed 's,//*,/,g' | sed -e 's,/$,,'`
+  # For unsermake we let top_builddir be absolute, instead of a
+  # ../.. cascade to make dependencies for better from subdirectories
+  # This doesn't change the behaviour of automake, as there the
+  # top_builddir line is hardcoded into the Makefile.in's.
+  top_builddir=`pwd`
+  AC_SUBST(top_builddir)
   KDE_FAST_CONFIGURE
   KDE_CONF_FILES
 ])
@@ -4553,6 +4572,7 @@
       KDE_JAVA_PREFIX(/usr/lib/IBMJava2*)
       KDE_JAVA_PREFIX(/usr/lib/IBMJava*)
       KDE_JAVA_PREFIX(/opt/java*)
+      KDE_JAVA_PREFIX(/Library/Java/Home)
     
       kde_cv_path="NONE"
       kde_save_IFS=$IFS
@@ -4604,10 +4624,20 @@
 dnl At this point kde_java_bindir and kde_java_includedir are either set or "no"
 if test "x$kde_java_bindir" != "xno"; then
 
+  case $host_os in
+    darwin*)
+      dnl Look for libjvm.dylib
+      kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.dylib | sed 's,libjvm.dylib,,'|head -n 1`
+      kd_java_libhpidir=''
+      dnl Darwin doesn't have libhpi
+      ;;
+    *)
   dnl Look for libjvm.so
   kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.so | sed 's,libjvm.so,,'|head -n 1`
   dnl Look for libhpi.so and avoid green threads
   kde_java_libhpidir=`find $kde_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,' | tail -n 1`
+      ;;
+  esac
 
   dnl Now check everything's fine under there
   dnl the include dir is our flag for having the JDK
@@ -4635,17 +4665,29 @@
     JAVAC=
     jni_includes=
   fi
-
+  case $host_os in
+    darwin*)
+      if test ! -r "$kde_java_libjvmdir/libjvm.dylib"; then
+        AC_MSG_ERROR([libjvm.dylib not found under $kde_java_libjvmdir. Use --without-java.])
+      fi
+      dnl Darwin doesn't have libhpi, but it's not fatal
+      LIB_JVM="-framework JavaVM"
+      LIB_HPI=""
+      ;;
+    *)
   if test ! -r "$kde_java_libjvmdir/libjvm.so"; then
      AC_MSG_ERROR([libjvm.so not found under $kde_java_libjvmdir. Use --without-java.])
   fi 
+      LIB_JVM="-L$kde_java_libjvmdir -ljvm"
+      if test ! -r "$kde_java_libhpidir/libhpi.so"; then
+        AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
+      fi
+      LIB_HPI="-L$kde_java_libhpidir -lhpi"
+      ;;
+  esac
 
   if test ! -x "$kde_java_bindir/java"; then
       AC_MSG_ERROR([java not found under $kde_java_bindir. javac was found though! Use --with-java or --without-java.])
-  fi
-
-  if test ! -r "$kde_java_libhpidir/libhpi.so"; then
-    AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
   fi
 
   if test -n "$jni_includes"; then
Index: admin/am_edit
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/admin/am_edit,v
retrieving revision 1.1.1.3
retrieving revision 1.5
diff -u -b -u -r1.1.1.3 -r1.5
Index: admin/cvs.sh
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/admin/cvs.sh,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
Index: admin/debianrules
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/admin/debianrules,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
Index: admin/depcomp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/admin/depcomp,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
Index: admin/libtool.m4.in
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/admin/libtool.m4.in,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
--- admin/libtool.m4.in	10 Aug 2002 13:45:40 -0000	1.1.1.3
+++ admin/libtool.m4.in	10 Aug 2002 14:07:14 -0000	1.4
@@ -609,6 +609,12 @@
     lt_cv_sys_max_cmd_len=-1;
     ;;
 
+  darwin*)
+    # this should be enough... the length test freaks out some versions
+    # of OSX's `expr`
+    lt_cv_sys_max_cmd_len=16384;
+    ;;
+
   *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -1901,16 +1907,7 @@
   ;;
 
 darwin* | rhapsody*)
-  lt_cv_deplibs_check_method='file_magic Mach-O dynamically linked shared library'
-  lt_cv_file_magic_cmd='/usr/bin/file -L'
-  case "$host_os" in
-  rhapsody* | darwin1.[[012]])
-    lt_cv_file_magic_test_file=`/System/Library/Frameworks/System.framework/System`
-    ;;
-  *) # Darwin 1.3 on
-    lt_cv_file_magic_test_file='/usr/lib/libSystem.dylib'
-    ;;
-  esac
+  lt_cv_deplibs_check_method='pass_all'
   ;;
 
 freebsd*)
@@ -2560,6 +2557,11 @@
 	;;
     esac
     ;;
+  darwin*)
+    shared_flag='-dynamiclib'
+    _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC -dynamiclib -install_name $rpath/$soname $predep_objects ${lib}-master.o $deplibs $postdep_objects $compiler_flags -o $lib'
+    output_verbose_link_cmd='$CC -dynamiclib $CFLAGS -v conftest.$objext 2>&1 | egrep "\-L"'
+    ;;
   dgux*)
     case $cc_basename in
       ec++)
@@ -4750,7 +4752,9 @@
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+	#_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+        # testing with twolevel namespaces
+	_LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
 	;;
       esac
 
@@ -4758,7 +4762,7 @@
       #        cross-compilation, but unfortunately the echo tests do not
       #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
       #	       `"' quotes if we put them in here... so don't!
-      _LT_AC_TAGVAR(archive_cmds, $1)='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+      _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) -install_name $rpath/$soname $allow_undefined_flag -o $lib ${lib}-master.o $deplibs $linker_flags $verstring'
       # We need to add '_' to the symbols in $export_symbols first
       #_LT_AC_TAGVAR(archive_expsym_cmds, $1)="$_LT_AC_TAGVAR(archive_cmds, $1)"' && strip -s $export_symbols'
       _LT_AC_TAGVAR(hardcode_direct, $1)=yes
Index: admin/ltmain.sh
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/admin/ltmain.sh,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- admin/ltmain.sh	20 Jun 2002 04:48:38 -0000	1.1.1.2
+++ admin/ltmain.sh	10 Jul 2002 21:18:18 -0000	1.3
@@ -3321,7 +3321,18 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+ 	  save_deplibs="$deplibs"
+ 	  for conv in $convenience; do
+            tmp_deplibs=
+            for test_deplib in $deplibs; do
+              if test "$test_deplib" != "$conv"; then
+                tmp_deplibs="$tmp_deplibs $test_deplib"
+              fi
+            done
+            deplibs="$tmp_deplibs"
+ 	  done
 	  eval cmds=\"$archive_cmds\"
+ 	  deplibs="$save_deplibs"
 	fi
 
 	if len=`expr "X$cmds" : ".*"` &&
@@ -3632,6 +3643,14 @@
 	compile_deplibs=`$echo "X $compile_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	finalize_deplibs=`$echo "X $finalize_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	;;
+      esac
+
+      case $host in
+      *darwin*)
+        # Don't allow lazy linking, it breaks C++ global constructors
+        compile_command="$compile_command ${wl}-bind_at_load"
+        finalize_command="$finalize_command ${wl}-bind_at_load"
+        ;;
       esac
 
       compile_command="$compile_command $compile_deplibs"
Index: admin/missing
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/admin/missing,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- admin/missing	20 Jun 2002 04:48:38 -0000	1.1.1.2
+++ admin/missing	10 Jul 2002 21:18:18 -0000	1.3
@@ -106,7 +106,7 @@
     exit 1
     ;;
 
-  aclocal)
+  aclocal*)
     echo 1>&2 "\
 WARNING: \`$1' is missing on your system.  You should only need it if
          you modified \`acinclude.m4' or \`${configure_ac}'.  You might want
@@ -143,7 +143,7 @@
     touch $touch_files
     ;;
 
-  automake)
+  automake*)
     echo 1>&2 "\
 WARNING: \`$1' is missing on your system.  You should only need it if
          you modified \`Makefile.am', \`acinclude.m4' or \`${configure_ac}'.
Index: doc/kformula/index.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kformula/index.docbook,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
Index: doc/koffice/index.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/koffice/index.docbook,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
Index: doc/kontour/index.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kontour/index.docbook,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
Index: doc/koshell/index.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/koshell/index.docbook,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
Index: doc/kpresenter/index.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kpresenter/index.docbook,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
Index: doc/kspread/index.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kspread/index.docbook,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
Index: doc/kugar/index.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kugar/index.docbook,v
retrieving revision 1.1.1.3
retrieving revision 1.3
diff -u -b -u -r1.1.1.3 -r1.3
Index: doc/kword/bultxtbut.png
===================================================================
RCS file: doc/kword/bultxtbut.png
diff -N doc/kword/bultxtbut.png
Binary files /tmp/cvs011343 and /dev/null differ
Index: doc/kword/columns.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kword/columns.docbook,v
retrieving revision 1.1.1.3
retrieving revision 1.3
diff -u -b -u -r1.1.1.3 -r1.3
Index: doc/kword/delcoldlg.png
===================================================================
RCS file: doc/kword/delcoldlg.png
diff -N doc/kword/delcoldlg.png
Binary files /tmp/cvs011343 and /dev/null differ
Index: doc/kword/delrowdlg.png
===================================================================
RCS file: doc/kword/delrowdlg.png
diff -N doc/kword/delrowdlg.png
Binary files /tmp/cvs011343 and /dev/null differ
Index: doc/kword/editing.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kword/editing.docbook,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
Index: doc/kword/formatchar.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kword/formatchar.docbook,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
Index: doc/kword/formatpara.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kword/formatpara.docbook,v
retrieving revision 1.1.1.3
retrieving revision 1.3
diff -u -b -u -r1.1.1.3 -r1.3
Index: doc/kword/headerfooter.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kword/headerfooter.docbook,v
retrieving revision 1.1.1.3
retrieving revision 1.3
diff -u -b -u -r1.1.1.3 -r1.3
Index: doc/kword/index.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kword/index.docbook,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
Index: doc/kword/instab2.png
===================================================================
RCS file: doc/kword/instab2.png
diff -N doc/kword/instab2.png
Binary files /tmp/cvs011343 and /dev/null differ
Index: doc/kword/lists.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kword/lists.docbook,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
Index: doc/kword/mbtb.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kword/mbtb.docbook,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
Index: doc/kword/opt.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kword/opt.docbook,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
Index: doc/kword/styldlg3.png
===================================================================
RCS file: doc/kword/styldlg3.png
diff -N doc/kword/styldlg3.png
Binary files /tmp/cvs011343 and /dev/null differ
Index: doc/kword/styldlg4.png
===================================================================
RCS file: doc/kword/styldlg4.png
diff -N doc/kword/styldlg4.png
Binary files /tmp/cvs011343 and /dev/null differ
Index: doc/kword/styldlg5.png
===================================================================
RCS file: doc/kword/styldlg5.png
diff -N doc/kword/styldlg5.png
Binary files /tmp/cvs011343 and /dev/null differ
Index: doc/kword/styldlg6.png
===================================================================
RCS file: doc/kword/styldlg6.png
diff -N doc/kword/styldlg6.png
Binary files /tmp/cvs011343 and /dev/null differ
Index: doc/kword/styldlg7.png
===================================================================
RCS file: doc/kword/styldlg7.png
diff -N doc/kword/styldlg7.png
Binary files /tmp/cvs011343 and /dev/null differ
Index: doc/kword/styldlg8.png
===================================================================
RCS file: doc/kword/styldlg8.png
diff -N doc/kword/styldlg8.png
Binary files /tmp/cvs011343 and /dev/null differ
Index: doc/kword/styles.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kword/styles.docbook,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
Index: doc/kword/table.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/kword/table.docbook,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
Index: doc/thesaurus/index.docbook
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/doc/thesaurus/index.docbook,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
Index: filters/karbon/ai/ailexer.cc
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/filters/karbon/ai/ailexer.cc,v
retrieving revision 1.1.1.3
retrieving revision 1.2
diff -u -b -u -r1.1.1.3 -r1.2
--- filters/karbon/ai/ailexer.cc	10 Aug 2002 13:48:20 -0000	1.1.1.3
+++ filters/karbon/ai/ailexer.cc	12 Aug 2002 01:08:39 -0000	1.2
@@ -38,9 +38,11 @@
   return (c==' ')||(c=='\n')||(c=='\t')||(c=='\r');
 }
 
+#ifndef isspecial
 int isspecial(char c){
   return (c=='*')||(c=='_')||(c=='?')||(c=='~')||(c=='-')||(c=='^')||(c=='`')||(c=='!')||(c=='.')||(c=='@')||(c=='&')||(c=='$')||(c=='=');
 }
+#endif
 
 int isletterhex(char c){
   return (c=='A')||(c=='B')||(c=='C')||(c=='D')||(c=='E')||(c=='F');
Index: filters/kword/latex/import/parser/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/filters/kword/latex/import/parser/Makefile.am,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- filters/kword/latex/import/parser/Makefile.am	20 Jun 2002 04:49:28 -0000	1.1.1.1
+++ filters/kword/latex/import/parser/Makefile.am	22 Jun 2002 19:05:09 -0000	1.2
@@ -2,7 +2,7 @@
 
 INCLUDES= -I$(srcdir) $(all_includes)
 liblatexparser_la_LDFLAGS = $(all_libraries) -module -avoid-version -no-undefined
-liblatexparser_la_LIBADD  = -lfl -lm $(KOFFICE_LIBS)
+liblatexparser_la_LIBADD  = -ll -lm $(KOFFICE_LIBS)
 
 ####### Files
 
Index: filters/kword/mswrite/filtertest.wri
===================================================================
RCS file: filters/kword/mswrite/filtertest.wri
diff -N filters/kword/mswrite/filtertest.wri
Binary files /tmp/cvs011343 and /dev/null differ
Index: filters/xsltfilter/export/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/filters/xsltfilter/export/Makefile.am,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
--- filters/xsltfilter/export/Makefile.am	20 Jun 2002 04:49:37 -0000	1.1.1.2
+++ filters/xsltfilter/export/Makefile.am	20 Jun 2002 04:48:28 -0000	1.2
@@ -1,9 +1,9 @@
 ####### General stuff
-SUBDIRS = xsl
 
-INCLUDES= -I$(srcdir) $(KOFFICE_INCLUDES) $(all_includes) $(LIBXML_CFLAGS) $(LIBXSLT_CFLAGS)
+KDE_INCLUDES=$(KDEDIR)/include
+INCLUDES= -I$(srcdir) $(KOFFICE_INCLUDES) -I$(KDE_INCLUDES)/libxml2 -I$(KDE_INCLUDES)/libxslt $(all_includes) $(LIBXML_CFLAGS) $(LIBXSLT_CFLAGS)
 
-AM_CPPFLAGS = -DSRCDIR="\"$(shell (cd $(top_srcdir)/filters/xsltfilter/export && pwd))\"" 
+AM_CPPFLAGS = -DSRCDIR="\"$(shell (cd $(srcdir) && pwd))\"" 
 
 ####### Files
 
Index: filters/xsltfilter/import/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/filters/xsltfilter/import/Makefile.am,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
--- filters/xsltfilter/import/Makefile.am	20 Jun 2002 04:49:38 -0000	1.1.1.2
+++ filters/xsltfilter/import/Makefile.am	20 Jun 2002 04:48:28 -0000	1.2
@@ -1,7 +1,8 @@
 ####### General stuff
 
-INCLUDES= -I$(srcdir) $(KOFFICE_INCLUDES)  $(all_includes) $(LIBXML_CFLAGS) $(LIBXSLT_CFLAGS)
-AM_CPPFLAGS = -DSRCDIR="\"$(shell (cd $(top_srcdir)/filters/xsltfilter/import && pwd))\"" 
+KDE_INCLUDES=$(KDEDIR)/include
+INCLUDES= -I$(srcdir) $(KOFFICE_INCLUDES) -I$(KDE_INCLUDES)/libxml2 -I$(KDE_INCLUDES)/libxslt $(all_includes) $(LIBXML_CFLAGS) $(LIBXSLT_CFLAGS)
+AM_CPPFLAGS = -DSRCDIR="\"$(shell (cd $(srcdir) && pwd))\"" 
 
 ####### Files
 
Index: karbon/pics/hi16-action-scale.png
===================================================================
RCS file: karbon/pics/hi16-action-scale.png
diff -N karbon/pics/hi16-action-scale.png
Binary files /tmp/cvs011343 and /dev/null differ
Index: karbon/pics/hi22-action-14_scale.png
===================================================================
RCS file: karbon/pics/hi22-action-14_scale.png
diff -N karbon/pics/hi22-action-14_scale.png
Binary files /tmp/cvs011343 and /dev/null differ
Index: karbon/pics/hi22-action-select.png
===================================================================
RCS file: karbon/pics/hi22-action-select.png
diff -N karbon/pics/hi22-action-select.png
Binary files /tmp/cvs011343 and /dev/null differ
Index: karbon/render/xrgbrender/gdk-pixbuf-xlib-private.h
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/karbon/render/xrgbrender/gdk-pixbuf-xlib-private.h,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- karbon/render/xrgbrender/gdk-pixbuf-xlib-private.h	20 Jun 2002 04:43:15 -0000	1.1.1.1
+++ karbon/render/xrgbrender/gdk-pixbuf-xlib-private.h	20 Jun 2002 04:48:28 -0000	1.2
@@ -31,5 +31,11 @@
 #define KSVG_BIG_ENDIAN 2
 
 /* FIXME: Make this auto-detectable */
+/* FIXME AGAIN: probably only Apple Compilers define this */
+#ifdef __BIG_ENDIAN__
+#define KSVG_BYTE_ORDER KSVG_BIG_ENDIAN
+#else
 #define KSVG_BYTE_ORDER KSVG_LITTLE_ENDIAN
+#endif
+
 #endif
Index: kchart/kdchart/KDChart.h
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/kchart/kdchart/KDChart.h,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
Index: kchart/kdchart/KDChartTable.h
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/kchart/kdchart/KDChartTable.h,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
Index: kchart/kdchart/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/kchart/kdchart/Makefile.am,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- kchart/kdchart/Makefile.am	20 Jun 2002 04:49:58 -0000	1.1.1.2
+++ kchart/kdchart/Makefile.am	22 Jun 2002 14:26:30 -0000	1.3
@@ -1,5 +1,7 @@
 lib_LTLIBRARIES = libkdchart.la
 
+libkdchart_la_LIBADD = $(all_libraries) $(LIB_QT)
+
 libkdchart_la_SOURCES = KDChart.cpp 		\
 			KDChartAreaPainter.cpp	\
 			KDChartAxesPainter.cpp	\
Index: kivio/kiviopart/ui/gridsetupdialogbase.ui
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/kivio/kiviopart/ui/gridsetupdialogbase.ui,v
retrieving revision 1.1.1.4
retrieving revision 1.2
diff -u -b -u -r1.1.1.4 -r1.2
Index: kivio/plugins/kivioconnectortool/straight_connector/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/kivio/plugins/kivioconnectortool/straight_connector/Makefile.am,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
--- kivio/plugins/kivioconnectortool/straight_connector/Makefile.am	20 Jun 2002 04:50:13 -0000	1.1.1.2
+++ kivio/plugins/kivioconnectortool/straight_connector/Makefile.am	20 Jun 2002 04:48:28 -0000	1.2
@@ -3,6 +3,7 @@
 #### WARNING! All changes made in this part will be lost! ####
 ##############################################################
 
+libdir = $(kde_datadir)/kivio/autoloadStencils
 lib_LTLIBRARIES =libstraight_connector.la
 
 
@@ -30,14 +31,14 @@
 
 ## This is a very bad hack..... why is this needed ? (David)
 
-install-libLTLIBRARIES: $(lib_LTLIBRARIES)
-	@$(NORMAL_INSTALL)
-	$(mkinstalldirs) $(DESTDIR)$(libdir)
-	$(LIBTOOL)  --mode=install $(INSTALL) ./.libs/libstraight_connector.so $(DESTDIR)$(libdir)/straight_connector.ksp;
-
-uninstall-libLTLIBRARIES:
-	@$(NORMAL_UNINSTALL)
-	$(LIBTOOL)  --mode=uninstall rm -f $(DESTDIR)$(libdir)/straight_connector.ksp;
+#install-libLTLIBRARIES: $(lib_LTLIBRARIES)
+#	@$(NORMAL_INSTALL)
+#	$(mkinstalldirs) $(DESTDIR)$(libdir)
+#	$(LIBTOOL)  --mode=install $(INSTALL) ./.libs/libstraight_connector.dylib $(DESTDIR)$(libdir)/straight_connector.ksp;
+#
+#uninstall-libLTLIBRARIES:
+#	@$(NORMAL_UNINSTALL)
+#	$(LIBTOOL)  --mode=uninstall rm -f $(DESTDIR)$(libdir)/straight_connector.ksp;
 
 stencil_DATA = straight_connector.xpm
 stencildir = $(kde_datadir)/kivio/autoloadStencils
Index: kpresenter/kpresenter_view.cc
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/kpresenter/kpresenter_view.cc,v
retrieving revision 1.1.1.3
retrieving revision 1.3
diff -u -b -u -r1.1.1.3 -r1.3
--- kpresenter/kpresenter_view.cc	10 Aug 2002 13:49:48 -0000	1.1.1.3
+++ kpresenter/kpresenter_view.cc	10 Aug 2002 14:07:16 -0000	1.3
@@ -4458,7 +4458,7 @@
     //(Laurent) deselect object when we change page.
     //otherwise you can change object properties on other page
     m_canvas->deSelectAllObj();
-    m_pKPresenterDoc->repaint( FALSE );
+    m_pKPresenterDoc->repaint( (bool)FALSE );
 
     m_pKPresenterDoc->displayActivePage(m_pKPresenterDoc->pageList().at(currPg));
 
Index: kspread/plugins/calculator/kcalc_core.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/kspread/plugins/calculator/kcalc_core.cpp,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- kspread/plugins/calculator/kcalc_core.cpp	20 Jun 2002 04:43:37 -0000	1.1.1.1
+++ kspread/plugins/calculator/kcalc_core.cpp	11 Aug 2002 16:42:11 -0000	1.2
@@ -51,7 +51,7 @@
 #undef HAVE_LONG_DOUBLE
 #endif
 
-#ifndef HAVE_FUNC_ISINF
+#if !defined(HAVE_FUNC_ISINF) || defined(__APPLE__)
 #ifdef HAVE_IEEEFP_H
 #include <ieeefp.h>
 #else
Index: kugar/part/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/koffice/kugar/part/Makefile.am,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- kugar/part/Makefile.am	20 Jun 2002 04:43:38 -0000	1.1.1.1
+++ kugar/part/Makefile.am	20 Jun 2002 04:48:29 -0000	1.2
@@ -3,7 +3,7 @@
 lib_LTLIBRARIES = libkugarpart.la
 libkugarpart_la_SOURCES = kugar_part.cpp
 libkugarpart_la_LDFLAGS = $(all_libraries) -module -avoid-version
-libkugarpart_la_LIBADD = ../lib/libkugar.la $(LIB_KPARTS) $(LIB_KDEPRINT)
+libkugarpart_la_LIBADD = ../lib/libkugar.la $(LIB_KPARTS) $(LIB_KDEPRINT) $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI) $(LIB_KIO)
 
 libkugarpart_la_METASOURCES = AUTO
 
