diff -uNbr kxine/kxine/Makefile.am kxine-new/kxine/Makefile.am
--- kxine/kxine/Makefile.am	Tue Sep  3 18:07:57 2002
+++ kxine-new/kxine/Makefile.am	Thu Nov 28 13:59:49 2002
@@ -26,18 +26,10 @@
 	     kxinefiledialog.cpp kxinefiledialog.h \
              kscrollinglabel.cpp kscrollinglabel.h
 
-install-data-local:
-	$(mkinstalldirs) $(kde_appsdir)/Applications/
-	$(INSTALL_DATA) $(srcdir)/kxine.desktop $(kde_appsdir)/Applications/kxine.desktop
-	$(mkinstalldirs) $(kde_icondir)/locolor/32x32/apps/
-	$(INSTALL_DATA) $(srcdir)/lo32-app-kxine.png $(kde_icondir)/locolor/32x32/apps/kxine.png
-	$(mkinstalldirs) $(kde_icondir)/locolor/16x16/apps/
-	$(INSTALL_DATA) $(srcdir)/lo16-app-kxine.png $(kde_icondir)/locolor/16x16/apps/kxine.png
+appsdir = $(kde_appsdir)/Graphics
+apps_DATA = kxine.desktop
 
-uninstall-local:
-	-rm -f $(kde_appsdir)/Applications/kxine.desktop
-	-rm -f $(kde_icondir)/locolor/32x32/apps/kxine.png
-	-rm -f $(kde_icondir)/locolor/16x16/apps/kxine.png
+KDE_ICON = kxine
 
 ####### kdevelop will overwrite this part!!! (end)############
 # These paths are KDE specific. Use them:
