diff -uNbr kxmleditor-0.8.1/kxmleditor/Makefile.am kxmleditor-0.8.1-new/kxmleditor/Makefile.am
--- kxmleditor-0.8.1/kxmleditor/Makefile.am	Tue Jul 23 15:14:18 2002
+++ kxmleditor-0.8.1-new/kxmleditor/Makefile.am	Sun Feb 16 12:53:07 2003
@@ -14,8 +14,8 @@
 	dcopiface_shell.cpp dcopiface_shell.h
 
 install-data-local:
-	$(mkinstalldirs) $(kde_appsdir)/Applications/
-	$(INSTALL_DATA) $(srcdir)/kxmleditor.desktop $(kde_appsdir)/Applications/kxmleditor.desktop
+	$(mkinstalldirs) $(kde_appsdir)/Editors/
+	$(INSTALL_DATA) $(srcdir)/kxmleditor.desktop $(kde_appsdir)/Editors/kxmleditor.desktop
 
 uninstall-local:
 	-rm -f $(kde_appsdir)/Applications/kxmleditor.desktop
diff -uNbr kxmleditor-0.8.1/part/Makefile.am kxmleditor-0.8.1-new/part/Makefile.am
--- kxmleditor-0.8.1/part/Makefile.am	Fri Jul 12 15:02:34 2002
+++ kxmleditor-0.8.1-new/part/Makefile.am	Sun Feb 16 12:26:09 2003
@@ -16,7 +16,7 @@
 	kxmleditorpart.cpp kxmleditorfactory.cpp \
 	dcopiface_part_ro.skel dcopiface_part_ro.cpp
 libkxmleditorpart_la_LDFLAGS = $(all_libraries) -version-info 1:0:0 -module
-libkxmleditorpart_la_LIBADD   = $(LIB_KPARTS) $(LIB_KFILE) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT) $(LIBSOCKET)
+libkxmleditorpart_la_LIBADD   = $(LIB_KPARTS) $(LIB_KFILE) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT) $(LIBSOCKET) -lDCOP
 
 SUBDIRS = pics icons
 
