diff -ur fluxbox-0.1.10.orig/util/Makefile.in fluxbox-0.1.10/util/Makefile.in
--- fluxbox-0.1.10.orig/util/Makefile.in	2002-07-13 21:00:52.000000000 -0400
+++ fluxbox-0.1.10/util/Makefile.in	2002-07-14 06:24:18.000000000 -0400
@@ -78,12 +78,12 @@
 gencat_cmd = @gencat_cmd@
 regex_cmd = @regex_cmd@
 
-bin_SCRIPTS = bsetbg
-bin_PROGRAMS = bsetroot
+bin_SCRIPTS = bsetbg-flux
+bin_PROGRAMS = bsetroot-flux
 bsetroot_SOURCES = bsetroot.cc bsetroot.hh
 bsetroot_LDADD = ../src/BaseDisplay.o ../src/Timer.o ../src/Image.o ../src/i18n.o ../src/FbAtoms.o
 MAINTAINERCLEANFILES = Makefile.in
-EXTRA_DIST = bsetbg
+EXTRA_DIST = bsetbg-flux
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../config.h
 CONFIG_CLEAN_FILES = 
@@ -177,8 +177,8 @@
 
 maintainer-clean-compile:
 
-bsetroot: $(bsetroot_OBJECTS) $(bsetroot_DEPENDENCIES)
-	@rm -f bsetroot
+bsetroot-flux: $(bsetroot_OBJECTS) $(bsetroot_DEPENDENCIES)
+	@rm -f bsetroot-flux
 	$(CXXLINK) $(bsetroot_LDFLAGS) $(bsetroot_OBJECTS) $(bsetroot_LDADD) $(LIBS)
 
 install-binSCRIPTS: $(bin_SCRIPTS)
