--- jwhois-3.2.2/src/dns.c.orig	Thu Nov 27 22:16:51 2003
+++ jwhois-3.2.2/src/dns.c	Thu Nov 27 22:17:06 2003
@@ -111,7 +111,7 @@
 lookup_host_addrinfo(struct addrinfo **res, const char *host, int port)
 {
   struct addrinfo hints;
-  char ascport[10] = "whois";
+  char ascport[10] = "nicname";
   int error;
 
   memset(&hints, 0, sizeof(hints));
--- jwhois-3.2.3/src/Makefile.in.orig	2005-07-09 14:52:02.000000000 -0400
+++ jwhois-3.2.3/src/Makefile.in	2013-06-04 10:42:11.000000000 -0400
@@ -67,7 +67,7 @@
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) $(INTLLIBS) -o $@
 SOURCES = $(jwhois_SOURCES)
 DIST_SOURCES = $(jwhois_SOURCES)
 ETAGS = etags
