diff -Nurd -x'*~' libmms-0.6.2.orig/src/Makefile.in libmms-0.6.2/src/Makefile.in
--- libmms-0.6.2.orig/src/Makefile.in	2011-01-19 10:18:55.000000000 -0500
+++ libmms-0.6.2/src/Makefile.in	2014-02-18 01:34:08.000000000 -0500
@@ -239,7 +239,7 @@
 
 # noinst_HEADERS = 
 libmms_la_LDFLAGS = -version-info 0:2:0 -no-undefined -export-dynamic
-libmms_la_LIBADD = $(GLIB_LIBS)
+libmms_la_LIBADD = $(GLIB_LIBS) -liconv
 libmmsincdir = $(includedir)/libmms
 libmmsinc_HEADERS = \
 	mms.h		\
