--- a/configure.ac
+++ b/configure.ac
@@ -561,6 +561,9 @@ case "$host" in
       AC_CHECK_LIB(nsl, main)
       CURSES=-lcurses
   ;;
+  *-darwin*)
+      opsys=darwin
+  ;;
   *-linux*)
       opsys=gnu-linux
       AC_DEFINE(GNU_LINUX,,GNU Linux)
@@ -643,6 +643,12 @@ if test "$with_libpam" = "yes"; then
      pam_conv_func="misc_conv"
     ],
     [], QUAGGA_INCLUDES)
+  AC_CHECK_HEADER([pam/pam_misc.h],
+    [AC_DEFINE(HAVE_PAM_MISC_H,,Have pam_misc.h)
+     AC_DEFINE(PAM_CONV_FUNC,misc_conv,Have misc_conv)
+     pam_conv_func="misc_conv"
+    ],
+    [], QUAGGA_INCLUDES)
   AC_CHECK_HEADER([security/openpam.h],
     [AC_DEFINE(HAVE_OPENPAM_H,,Have openpam.h)
      AC_DEFINE(PAM_CONV_FUNC,openpam_ttyconv,Have openpam_ttyconv)
@@ -966,6 +966,7 @@ AC_MSG_RESULT(no))
 dnl ---------------------------------------------------------------
 dnl figure out how to check link-state
 dnl ---------------------------------------------------------------
+if test "$opsys" != "darwin"; then
 AC_CHECK_HEADER([net/if.h],
   [AC_CHECK_HEADER( [net/if_media.h],
     [m4_define([LINK_DETECT_INCLUDES],
@@ -978,6 +978,7 @@ AC_CHECK_HEADER([net/if.h],
     [],
     QUAGGA_INCLUDES)], 
   [], QUAGGA_INCLUDES )
+fi
 
 dnl ------------------------
 dnl TCP_MD5SIG socket option

--- a/vtysh/vtysh_user.c
+++ b/vtysh/vtysh_user.c
@@ -25,10 +25,18 @@
 #include <pwd.h>
 
 #ifdef USE_PAM
+#ifdef __APPLE__
+#include <pam/pam_appl.h>
+#else
 #include <security/pam_appl.h>
+#endif
 #ifdef HAVE_PAM_MISC_H
+#ifdef __APPLE__
+#include <pam/pam_misc.h>
+#else
 #include <security/pam_misc.h>
 #endif
+#endif
 #ifdef HAVE_OPENPAM_H
 #include <security/openpam.h>
 #endif
--- a/lib/zebra.h.orig 2009-12-09 23:37:12.000000000 +1100
+++ b/lib/zebra.h 2012-07-05 00:54:00.000000000 +1000
@@ -143,6 +143,7 @@
 #endif /* HAVE_SYS_SOCKIO_H */
 
 #ifdef HAVE_NETINET_IN_H
+#define __APPLE_USE_RFC_3542
 #include <netinet/in.h>
 #endif /* HAVE_NETINET_IN_H */
 #include <netinet/in_systm.h>
