--- licq-1.8.1-orig/src/mainloop.cpp	2013-10-28 08:26:38.000000000 -0400
+++ licq-1.8.1/src/mainloop.cpp	2014-02-08 20:14:40.000000000 -0500
@@ -22,6 +22,10 @@
 #include <cassert>
 #include <cerrno>
 #include <ctime>
+#ifdef __MACH__
+#include <mach/clock.h>
+#include <mach/mach.h>
+#endif
 #include <map>
 #include <unistd.h>
 
@@ -93,6 +97,14 @@
 #ifdef CLOCK_MONOTONIC_RAW
   // CLOCK_MONOTONIC_RAW is Linux specific but isn't affected by NTP adjustments
   clock_gettime(CLOCK_MONOTONIC_RAW, &ts);
+#elif __MACH__ // OS X does not have clock_gettime, use clock_get_time
+  clock_serv_t cclock;
+  mach_timespec_t mts;
+  host_get_clock_service(mach_host_self(), CALENDAR_CLOCK, &cclock);
+  clock_get_time(cclock, &mts);
+  mach_port_deallocate(mach_task_self(), cclock);
+  ts.tv_sec = mts.tv_sec;
+  ts.tv_nsec = mts.tv_nsec;
 #else
   clock_gettime(CLOCK_MONOTONIC, &ts);
 #endif
