diff -rauN openvpn-2.2.1/socket.c openvpn-2.2.1-p1/socket.c
--- openvpn-2.2.1/socket.c	2011-06-24 08:13:39.000000000 +0200
+++ openvpn-2.2.1-p1/socket.c	2011-08-20 11:19:46.000000000 +0200
@@ -35,6 +35,10 @@
 
 #include "memdbg.h"
 
+#ifndef SOL_IP
+#define SOL_IP IPPROTO_IP
+#endif
+
 const int proto_overhead[] = { /* indexed by PROTO_x */
   IPv4_UDP_HEADER_SIZE,
   IPv4_TCP_HEADER_SIZE,
diff -rauN openvpn-2.2.1/configure.ac openvpn-2.2.1-p1/configure.ac
--- openvpn-2.2.1/configure.ac	2011-06-24 08:13:38.000000000 +0200
+++ openvpn-2.2.1-p1/configure.ac	2011-08-20 11:44:17.000000000 +0200
@@ -313,6 +313,10 @@
 *-*-netbsd*)
 	AC_DEFINE(TARGET_NETBSD, 1, [Are we running NetBSD?])
 	;;
+*-*-darwin11*)
+	dnl some Mac OS X tendering (we use vararg macros...)
+	AC_DEFINE(TARGET_DARWIN, 1, [Are we running on Mac OS X?])
+	;;
 *-*-darwin*)
 	dnl some Mac OS X tendering (we use vararg macros...)
 	AC_DEFINE(TARGET_DARWIN, 1, [Are we running on Mac OS X?])
