diff -Naur squid-3.1.14.orig/errors/Makefile.in squid-3.1.14/errors/Makefile.in
--- squid-3.1.14.orig/errors/Makefile.in	2011-07-24 15:11:59.000000000 -0700
+++ squid-3.1.14/errors/Makefile.in	2011-07-24 17:23:45.000000000 -0700
@@ -474,7 +474,7 @@
 
 install-dvi-am:
 
-install-exec-am: install-exec-local
+install-exec-am:
 
 install-html: install-html-am
 
@@ -590,7 +590,7 @@
 	done; \
 	$(INSTALL_DATA) $(srcdir)/TRANSLATORS $(DESTDIR)$(DEFAULT_ERROR_DIR)/TRANSLATORS; \
 	$(INSTALL_DATA) $(srcdir)/COPYRIGHT $(DESTDIR)$(DEFAULT_ERROR_DIR)/COPYRIGHT; \
-	$(INSTALL_DATA) $(srcdir)/errorpage.css $(DESTDIR)$(DEFAULT_STYLESHEET).default; \
+#	$(INSTALL_DATA) $(srcdir)/errorpage.css $(DESTDIR)$(DEFAULT_STYLESHEET).default; \
 	$(SHELL) $(srcdir)/alias-link.sh "$(LN)" "$(RM)" "$(DESTDIR)$(DEFAULT_ERROR_DIR)" "$(srcdir)/aliases" || exit 1 ;
 
 uninstall-local:
diff -Naur squid-3.1.14.orig/helpers/basic_auth/MSNT/Makefile.in squid-3.1.14/helpers/basic_auth/MSNT/Makefile.in
--- squid-3.1.14.orig/helpers/basic_auth/MSNT/Makefile.in	2011-07-24 15:11:59.000000000 -0700
+++ squid-3.1.14/helpers/basic_auth/MSNT/Makefile.in	2011-07-24 17:23:45.000000000 -0700
@@ -718,13 +718,13 @@
 
 info-am:
 
-install-data-am: install-data-local
+install-data-am:
 
 install-dvi: install-dvi-am
 
 install-dvi-am:
 
-install-exec-am: install-libexecPROGRAMS install-sysconfDATA
+install-exec-am: install-libexecPROGRAMS
 
 install-html: install-html-am
 
diff -Naur squid-3.1.14.orig/src/ChunkedCodingParser.cc squid-3.1.14/src/ChunkedCodingParser.cc
--- squid-3.1.14.orig/src/ChunkedCodingParser.cc	2011-07-24 15:11:59.000000000 -0700
+++ squid-3.1.14/src/ChunkedCodingParser.cc	2011-07-24 17:31:57.000000000 -0700
@@ -143,9 +143,9 @@
     size_t crlfEnd = 0;
 
     if (findCrlf(crlfBeg, crlfEnd)) {
-        if (crlfBeg > 0)
-
-            ; //theTrailer.append(theIn->content(), crlfEnd);
+//        if (crlfBeg > 0)
+//
+//            ; //theTrailer.append(theIn->content(), crlfEnd);
 
         theIn->consume(crlfEnd);
 
diff -Naur squid-3.1.14.orig/src/Makefile.in squid-3.1.14/src/Makefile.in
--- squid-3.1.14.orig/src/Makefile.in	2011-07-24 15:11:59.000000000 -0700
+++ squid-3.1.14/src/Makefile.in	2011-07-24 17:23:45.000000000 -0700
@@ -4722,14 +4722,14 @@
 
 info-am:
 
-install-data-am: install-data-local install-dataDATA
+install-data-am: install-dataDATA
 
 install-dvi: install-dvi-recursive
 
 install-dvi-am:
 
 install-exec-am: install-binPROGRAMS install-libexecPROGRAMS \
-	install-sbinPROGRAMS install-sysconfDATA
+	install-sbinPROGRAMS
 
 install-html: install-html-recursive
 
diff -Naur squid-3.1.14.orig/src/base/TidyPointer.h squid-3.1.14/src/base/TidyPointer.h
--- squid-3.1.14.orig/src/base/TidyPointer.h	2011-07-24 15:11:59.000000000 -0700
+++ squid-3.1.14/src/base/TidyPointer.h	2011-07-24 17:23:45.000000000 -0700
@@ -5,6 +5,10 @@
 #ifndef SQUID_BASE_TIDYPOINTER_H
 #define SQUID_BASE_TIDYPOINTER_H
 
+#ifndef NULL
+#define NULL ((void *)0)
+#endif
+
 /**
  * A pointer that deletes the object it points to when the pointer's owner or
  * context is gone. Similar to std::auto_ptr but without confusing assignment
diff -Naur squid-3.1.14.orig/src/event.cc squid-3.1.14/src/event.cc
--- squid-3.1.14.orig/src/event.cc	2011-07-24 15:11:59.000000000 -0700
+++ squid-3.1.14/src/event.cc	2011-07-24 17:37:52.000000000 -0700
@@ -238,7 +238,7 @@
 EventScheduler::checkEvents(int timeout)
 {
 
-    struct ev_entry *event = NULL;
+    class ev_entry *event = NULL;
 
     if (NULL == tasks)
         return checkDelay();
@@ -291,7 +291,7 @@
 EventScheduler::dump(StoreEntry * sentry)
 {
 
-    struct ev_entry *e = tasks;
+    class ev_entry *e = tasks;
 
     if (last_event_ran)
         storeAppendPrintf(sentry, "Last event to run: %s\n\n", last_event_ran);
@@ -314,7 +314,7 @@
 EventScheduler::find(EVH * func, void * arg)
 {
 
-    struct ev_entry *event;
+    class ev_entry *event;
 
     for (event = tasks; event != NULL; event = event->next) {
         if (event->func == func && event->arg == arg)
@@ -337,9 +337,9 @@
     // must fire in the submission order. We cannot use current_dtime for them
     // because it may decrease if system clock is adjusted backwards.
     const double timestamp = when > 0.0 ? current_dtime + when : 0;
-    struct ev_entry *event = new ev_entry(name, func, arg, timestamp, weight, cbdata);
+    class ev_entry *event = new ev_entry(name, func, arg, timestamp, weight, cbdata);
 
-    struct ev_entry **E;
+    class ev_entry **E;
     debugs(41, 7, HERE << "schedule: Adding '" << name << "', in " << when << " seconds");
     /* Insert after the last event with the same or earlier time */
 
diff -Naur squid-3.1.14.orig/src/gopher.cc squid-3.1.14/src/gopher.cc
--- squid-3.1.14.orig/src/gopher.cc	2011-07-24 15:11:59.000000000 -0700
+++ squid-3.1.14/src/gopher.cc	2011-07-24 17:42:20.000000000 -0700
@@ -822,7 +822,7 @@
         IOStats.Gopher.read_hist[bin]++;
     }
 
-    if (flag != COMM_OK || len < 0) {
+    if (flag != COMM_OK) {
         debugs(50, 1, "gopherReadReply: error reading: " << xstrerror());
 
         if (ignoreErrno(errno)) {
diff -Naur squid-3.1.14.orig/src/ssl/gadgets.cc squid-3.1.14/src/ssl/gadgets.cc
--- squid-3.1.14.orig/src/ssl/gadgets.cc	2011-07-24 15:11:59.000000000 -0700
+++ squid-3.1.14/src/ssl/gadgets.cc	2011-07-24 17:23:45.000000000 -0700
@@ -2,7 +2,7 @@
  * $Id$
  */
 
-#include "config.h"
+#include "include/config.h"
 #include "ssl/gadgets.h"
 
 /**
diff -Naur squid-3.1.14.orig/src/wccp2.cc squid-3.1.14/src/wccp2.cc
--- squid-3.1.14.orig/src/wccp2.cc	2011-07-24 15:11:59.000000000 -0700
+++ squid-3.1.14/src/wccp2.cc	2011-07-24 17:49:15.000000000 -0700
@@ -301,7 +301,7 @@
     uint32_t received_id;
 };
 
-static struct wccp2_router_id_element_t wccp2_router_id_element;
+// static struct wccp2_router_id_element_t wccp2_router_id_element;
 
 /** \interface WCCpv2_Protocol
  * Sect 5.6.9 Capabilities Info Component
@@ -834,7 +834,7 @@
         wccp2_cache_view_header.cache_view_type = htons(WCCP2_WC_VIEW_INFO);
 
         wccp2_cache_view_header.cache_view_length = htons(sizeof(wccp2_cache_view_header) - 4 +
-                sizeof(wccp2_cache_view_info) + (wccp2_numrouters * sizeof(wccp2_router_id_element)));
+                sizeof(wccp2_cache_view_info) + (wccp2_numrouters * sizeof(wccp2_router_id_element_t)));
 
         wccp2_cache_view_header.cache_view_version = htonl(1);
 
diff -Naur squid-3.1.14.orig/tools/Makefile.in squid-3.1.14/tools/Makefile.in
--- squid-3.1.14.orig/tools/Makefile.in	2011-07-24 15:11:59.000000000 -0700
+++ squid-3.1.14/tools/Makefile.in	2011-07-24 17:23:45.000000000 -0700
@@ -1008,7 +1008,7 @@
 
 info-am:
 
-install-data-am: install-data-local install-man
+install-data-am: install-man
 
 install-dvi: install-dvi-recursive
 
