diff -Naur --exclude='*~' ebib-1.5.2/fink/50ebib.el ebib-1.5.2.fink/fink/50ebib.el
--- ebib-1.5.2/fink/50ebib.el	1969-12-31 18:00:00.000000000 -0600
+++ ebib-1.5.2.fink/fink/50ebib.el	2007-12-26 15:29:44.000000000 -0600
@@ -0,0 +1 @@
+(require 'ebib)
diff -Naur --exclude='*~' ebib-1.5.2/fink/README.fink ebib-1.5.2.fink/fink/README.fink
--- ebib-1.5.2/fink/README.fink	1969-12-31 18:00:00.000000000 -0600
+++ ebib-1.5.2.fink/fink/README.fink	2007-12-10 13:25:27.000000000 -0600
@@ -0,0 +1,5 @@
+This package was finkied by Jesse Alama <alama@stanford.edu>.  The
+upstream maintainer is Joost Kremers
+<joostkremers@users.sourceforge.net>.  It was downloaded from the ebib
+homepage: http://ebib.sourceforge.net .  The license is contained in
+the file `ebib.el' in @PREFIX@/share/emacs/site-lisp/ebib .
\ No newline at end of file
diff -Naur --exclude='*~' ebib-1.5.2/fink/emacsen-install ebib-1.5.2.fink/fink/emacsen-install
--- ebib-1.5.2/fink/emacsen-install	1969-12-31 18:00:00.000000000 -0600
+++ ebib-1.5.2.fink/fink/emacsen-install	2007-12-10 13:26:10.000000000 -0600
@@ -0,0 +1,35 @@
+#!/bin/bash -e
+#
+# install file for the fink ebib emacs package.
+
+set -o posix
+
+FLAVOR=${1}
+
+echo >&2 "install/ebib: Handling install of emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "install/ebib: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+if [ ${FLAVOR} == emacs ]
+then
+    echo "install/ebib: Done."
+    exit 0
+fi
+
+echo >&2 -n "install/ebib: Byte-compiling for ${FLAVOR}..."
+mkdir -p @PREFIX@/share/${FLAVOR}/site-lisp/ebib;
+ln -fs @PREFIX@/share/emacs/site-lisp/ebib/ebib.el @PREFIX@/share/${FLAVOR}/site-lisp/ebib;
+
+
+cd @PREFIX@/share/${FLAVOR}/site-lisp/ebib;
+
+(${FLAVOR} -batch -f batch-byte-compile 2>&1 ebib.el) | gzip -9qf > @PREFIX@/share/doc/ebib/CompilationLog-${FLAVOR}.gz;
+
+echo >&2 "done."
+echo >&2 "install/ebib: Compilation log saved in @PREFIX@/share/doc/ebib/CompilationLog-${FLAVOR}.gz."
+
+exit 0
diff -Naur --exclude='*~' ebib-1.5.2/fink/emacsen-remove ebib-1.5.2.fink/fink/emacsen-remove
--- ebib-1.5.2/fink/emacsen-remove	1969-12-31 18:00:00.000000000 -0600
+++ ebib-1.5.2.fink/fink/emacsen-remove	2007-12-10 13:27:10.000000000 -0600
@@ -0,0 +1,21 @@
+#!/bin/bash -e
+#
+# This is the Fink version of the ebib emacs package.
+set -o posix
+
+FLAVOR=${1}
+
+echo "remove/ebib: Handling removal for emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "remove/ebib: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+echo >&2 -n "remove/ebib: Purging compilation log and byte-compiled files for ${FLAVOR}..."
+rm -f @PREFIX@/share/doc/ebib/CompilationLog-${FLAVOR}.gz
+rm -Rf @PREFIX@/share/${FLAVOR}/site-lisp/ebib
+echo >&2 "done."
+    
+exit 0
diff -Naur --exclude='*~' ebib-1.5.2/info/ebib-manual.info ebib-1.5.2.fink/info/ebib-manual.info
--- ebib-1.5.2/info/ebib-manual.info	2007-10-22 15:00:25.000000000 -0500
+++ ebib-1.5.2.fink/info/ebib-manual.info	2007-12-10 13:43:48.000000000 -0600
@@ -1,6 +1,12 @@
 This is ebib-manual.info, produced by makeinfo version 4.8 from
 ebib-manual.texi.
 
+INFO-DIR-SECTION Emacs
+
+START-INFO-DIR-ENTRY
+* Ebib: (ebib).           Maintain BibTeX databases in emacs
+END-INFO-DIR-ENTRY
+
 
 File: ebib-manual.info,  Node: Top,  Next: Installation,  Prev: (dir),  Up: (dir)
 
