diff -Naur easyedit.old/make.darwin easyedit.new/make.darwin
--- easyedit.old/make.darwin	Thu Jan  1 09:00:00 1970
+++ easyedit.new/make.darwin	Mon Jun 18 02:57:36 2001
@@ -0,0 +1,11 @@
+CPPFLAGS = -DCAP -DNO_CATGETS -DHAS_UNISTD -DHAS_STDARG -DHAS_STDLIB \
+	   -DHAS_CTYPE -DHAS_SYS_IOCTL -DHAS_SYS_WAIT -DSLCT_HDR \
+	   -I$(PREFIX)/include
+CFLAGS = -Os
+LDLIBS = -lcurses
+LDFLAGS = -L$(PREFIX)/lib
+
+all : ee
+
+ee : ee.c
+
