diff -ruN --exclude '*~' --exclude '*.orig' emacs-22.1/Makefile.in emacs-22.1.patched/Makefile.in
--- emacs-22.1/Makefile.in	2007-05-14 10:56:28.000000000 -0400
+++ emacs-22.1.patched/Makefile.in	2007-11-14 21:42:38.000000000 -0500
@@ -437,7 +437,7 @@
 	      (cd $${dir}; tar -chf - . ) \
 		| (cd $${dest}; umask 022; \
                    tar -xvf - && cat > /dev/null) || exit 1; \
-	      find $${dest} -exec chown $${LOGNAME:-$$USERNAME} {} ';' ;\
+	      find $${dest} -exec chown root:admin {} ';' ;\
 	      for subdir in `find $${dest} -type d ! -name RCS ! -name CVS -print` ; do \
 		chmod a+rx $${subdir} ; \
 		rm -rf $${subdir}/RCS ; \
@@ -477,7 +477,7 @@
 	   echo "Copying etc/DOC-* to $(DESTDIR)${docdir} ..." ; \
 	   (cd ./etc; tar -chf - DOC*) \
 	     |(cd $(DESTDIR)${docdir}; umask 022; tar -xvf - && cat > /dev/null) || exit 1; \
-	   (cd $(DESTDIR)$(docdir); chown $${LOGNAME:-$$USERNAME} DOC*; chmod a+r DOC*; \
+	   (cd $(DESTDIR)$(docdir); chown root:admin DOC*; chmod a+r DOC*; \
 	    if test "`echo DOC-*`" != "DOC-*"; then rm DOC; fi); \
 	else true; fi
 	-unset CDPATH; \
@@ -489,7 +489,7 @@
 	   echo "Copying lisp/*.el and lisp/*.elc to $(DESTDIR)${lispdir} ..." ; \
 	   (cd lisp; tar -chf - *.el *.elc) \
 	     |(cd $(DESTDIR)${lispdir}; umask 022; tar -xvf - && cat > /dev/null) || exit 1; \
-	   (cd $(DESTDIR)${lispdir}; find . -exec chown $${LOGNAME:-$$USERNAME} {} ';') ; \
+	   (cd $(DESTDIR)${lispdir}; find . -exec chown root:admin {} ';') ; \
 	else true; fi
 	-unset CDPATH; \
 	if [ -n "${GZIP_PROG}" ]; \
diff -ruN --exclude '*~' --exclude '*.orig' emacs-22.1/configure emacs-22.1.patched/configure
--- emacs-22.1/configure	2007-05-25 08:43:29.000000000 -0400
+++ emacs-22.1.patched/configure	2007-11-14 21:42:38.000000000 -0500
@@ -1807,7 +1807,7 @@
 etcdir='${datadir}/emacs/${version}/etc'
 archlibdir='${libexecdir}/emacs/${version}/${configuration}'
 docdir='${datadir}/emacs/${version}/etc'
-gamedir='${localstatedir}/games/emacs'
+gamedir='${localstatedir}/games/emacs22'
 
 gameuser=games
 
@@ -13917,9 +13917,9 @@
 fi
 { echo "$as_me:$LINENO: result: $ac_cv_lib_ungif_EGifPutExtensionLast" >&5
 echo "${ECHO_T}$ac_cv_lib_ungif_EGifPutExtensionLast" >&6; }
-if test $ac_cv_lib_ungif_EGifPutExtensionLast = yes; then
+#if test $ac_cv_lib_ungif_EGifPutExtensionLast = yes; then
   HAVE_GIF=yes
-fi
+#fi
 
 fi
 
diff -ruN --exclude '*~' --exclude '*.orig' emacs-22.1/fix-info emacs-22.1.patched/fix-info
--- emacs-22.1/fix-info	1969-12-31 19:00:00.000000000 -0500
+++ emacs-22.1.patched/fix-info	2007-11-14 21:42:38.000000000 -0500
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+THE_INFO_PREFIX=$1
+
+for file in ada-mode autotype calc ccmode cl dired-x ebrowse ediff efaq \
+           elisp eintr emacs emacs-mime eshell eudc flymake \
+           forms gnus idlwave info message mh-e newsticker org pcl-cvs \
+           pgg reftex sc ses sieve speedbar tramp vip viper widget \
+           woman smtpmail url rcirc erc
+do
+    ./mangle-info ${THE_INFO_PREFIX}/${file}
+done
\ No newline at end of file
diff -ruN --exclude '*~' --exclude '*.orig' emacs-22.1/lisp/startup.el emacs-22.1.patched/lisp/startup.el
--- emacs-22.1/lisp/startup.el	2007-05-14 10:56:29.000000000 -0400
+++ emacs-22.1.patched/lisp/startup.el	2007-11-14 21:42:38.000000000 -0500
@@ -288,6 +288,10 @@
 
 (defvar normal-top-level-add-subdirs-inode-list nil)
 
+(defconst fink-emacs-flavor 'emacs22
+  "A symbol representing the particular fink flavor of emacs running.
+Something like 'emacs20, 'xemacs20, etc.")
+
 (defvar no-blinking-cursor nil)
 
 (defvar default-frame-background-mode)
@@ -815,9 +819,19 @@
     ;; Run the site-start library if it exists.  The point of this file is
     ;; that it is run before .emacs.  There is no point in doing this after
     ;; .emacs; that is useless.
-    (if site-run-file
-	(load site-run-file t t))
-
+    ;; (if site-run-file                                                       
+    ;;     (load site-run-file t t))                                           
+                                                                              
+    ;; Fink version of site-start.                                           
+    (if site-run-file                                                          
+	(progn                                                                 
+	  ;; Load all the fink package snippets.                             
+	  ;; It's in here because we want -q to kill it too.                   
+	  (if (load "fink-startup" t t nil)                                  
+	      (fink-startup fink-emacs-flavor))                            
+	  ;; Now the normal site file...                                       
+	  (load site-run-file t t nil)))       
+    
     ;; Sites should not disable this.  Only individuals should disable
     ;; the startup message.
     (setq inhibit-startup-message nil)
diff -ruN --exclude '*~' --exclude '*.orig' emacs-22.1/mangle-info emacs-22.1.patched/mangle-info
--- emacs-22.1/mangle-info	1969-12-31 19:00:00.000000000 -0500
+++ emacs-22.1.patched/mangle-info	2007-11-14 21:42:38.000000000 -0500
@@ -0,0 +1,18 @@
+#!/usr/bin/perl -w -i
+
+use English;
+use strict;
+
+$RS = undef;
+
+my $prefix = "emacs-22";
+
+my $x = <>;
+if(!($x =~ m/^(\s*START-INFO-DIR-ENTRY\s+\*\s*[^:]+:\s*)\(([^\)]+)\)/mo)) {
+  die "Couldn't find START-INFO-DIR-ENTRY.";
+}
+   
+print ${PREMATCH};
+print "$1($prefix/$2)";
+print ${POSTMATCH};
+
diff -ruN --exclude '*~' --exclude '*.orig' emacs-22.1/src/Makefile.in emacs-22.1.patched/src/Makefile.in
--- emacs-22.1/src/Makefile.in	2007-03-21 10:31:24.000000000 -0400
+++ emacs-22.1.patched/src/Makefile.in	2007-11-14 21:42:38.000000000 -0500
@@ -440,7 +440,7 @@
 
 #if HAVE_GIF
 #ifndef LIBGIF
-#define LIBGIF -lungif
+#define LIBGIF -lgif
 #endif /* not defined LIBGIF */
 #else /* not HAVE_GIF */
 #define LIBGIF
diff -ruN --exclude '*~' --exclude '*.orig' emacs-22.1/src/unexmacosx.c emacs-22.1.patched/src/unexmacosx.c
--- emacs-22.1/src/unexmacosx.c	2007-01-20 23:18:15.000000000 -0500
+++ emacs-22.1.patched/src/unexmacosx.c	2007-11-14 21:44:17.000000000 -0500
@@ -443,15 +443,13 @@
 
   while (num && num_unexec_regions < MAX_UNEXEC_REGIONS)
     {
-      /* Subtract the size of trailing null pages from filesize.  It
+      /* Subtract the size of trailing null bytes from filesize.  It
 	 can be smaller than vmsize in segment commands.  In such a
-	 case, trailing pages are initialized with zeros.  */
-      for (p = ranges->address + ranges->size; p > ranges->address;
-	   p -= sizeof (int))
-	if (*(((int *) p)-1))
-	  break;
-      filesize = ROUNDUP_TO_PAGE_BOUNDARY (p - ranges->address);
-      assert (filesize <= ranges->size);
+	 case, trailing bytes are initialized with zeros.  */
+      for (p = ranges->address + ranges->size; p > ranges->address; p--)
+      	if (*(((char *) p)-1))
+      	  break;
+      filesize = p - ranges->address;
 
       unexec_regions[num_unexec_regions].filesize = filesize;
       unexec_regions[num_unexec_regions++].range = *ranges;
@@ -503,11 +501,19 @@
 {
   int i, n;
   unexec_region_info r;
+  vm_size_t padsize;
 
   qsort (unexec_regions, num_unexec_regions, sizeof (unexec_regions[0]),
 	 &unexec_regions_sort_compare);
   n = 0;
   r = unexec_regions[0];
+  padsize = r.range.address & (pagesize - 1);
+  if (padsize)
+    {
+      r.range.address -= padsize;
+      r.range.size += padsize;
+      r.filesize += padsize;
+    }
   for (i = 1; i < num_unexec_regions; i++)
     {
       if (r.range.address + r.range.size == unexec_regions[i].range.address
@@ -520,6 +526,17 @@
 	{
 	  unexec_regions[n++] = r;
 	  r = unexec_regions[i];
+	  padsize = r.range.address & (pagesize - 1);
+	  if (padsize)
+	    {
+	      if ((unexec_regions[n-1].range.address
+		   + unexec_regions[n-1].range.size) == r.range.address)
+		unexec_regions[n-1].range.size -= padsize;
+
+	      r.range.address -= padsize;
+	      r.range.size += padsize;
+	      r.filesize += padsize;
+	    }
 	}
     }
   unexec_regions[n++] = r;
@@ -562,6 +579,11 @@
     case LC_TWOLEVEL_HINTS:
       printf ("LC_TWOLEVEL_HINTS");
       break;
+#ifdef LC_UUID
+    case LC_UUID:
+      printf ("LC_UUID          ");
+      break;
+#endif
     default:
       printf ("unknown          ");
     }
