diff --exclude='*~' -ruN emacs-23.1/configure emacs-23.1.patched/configure
--- emacs-23.1/configure	2009-07-29 17:12:00.000000000 +0200
+++ emacs-23.1.patched/configure	2009-08-26 16:55:27.000000000 +0200
@@ -1854,7 +1854,7 @@
 etcdir='${datadir}/emacs/${version}/etc'
 archlibdir='${libexecdir}/emacs/${version}/${configuration}'
 docdir='${datadir}/emacs/${version}/etc'
-gamedir='${localstatedir}/games/emacs'
+gamedir='${localstatedir}/games/emacs23'
 
 gameuser=games
 
@@ -2489,12 +2489,6 @@
     opsys=darwin
     # Define CPP as follows to make autoconf work correctly.
     CPP="${CC-cc} -E -no-cpp-precomp"
-    # Use fink packages if available.
-    if test -d /sw/include && test -d /sw/lib; then
-      GCC_TEST_OPTIONS="-I/sw/include -L/sw/lib"
-      CPP="${CPP} ${GCC_TEST_OPTIONS}"
-      NON_GCC_TEST_OPTIONS=${GCC_TEST_OPTIONS}
-    fi
   ;;
 
   ## HP 9000 series 700 and 800, running HP/UX
@@ -10942,7 +10942,7 @@
 HAVE_GTK=no
 if test "${with_gtk}" = "yes" || test "$USE_X_TOOLKIT" = "maybe"; then
   GLIB_REQUIRED=2.6
-  GTK_REQUIRED=2.6
+  GTK_REQUIRED=2.17.9
   GTK_MODULES="gtk+-2.0 >= $GTK_REQUIRED glib-2.0 >= $GLIB_REQUIRED"
 
 
@@ -14610,9 +14610,9 @@
 fi
 { echo "$as_me:$LINENO: result: $ac_cv_lib_ungif_EGifPutExtensionLast" >&5
 echo "${ECHO_T}$ac_cv_lib_ungif_EGifPutExtensionLast" >&6; }
-if test $ac_cv_lib_ungif_EGifPutExtensionLast = yes; then
+#if test $ac_cv_lib_ungif_EGifPutExtensionLast = yes; then
   HAVE_GIF=yes
-fi
+#fi
 
 
     if test "$HAVE_GIF" = yes; then
diff --exclude='*~' -ruN emacs-23.1/fix-info emacs-23.1.patched/fix-info
--- emacs-23.1/fix-info	1970-01-01 01:00:00.000000000 +0100
+++ emacs-23.1.patched/fix-info	2009-08-26 17:44:01.000000000 +0200
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+THE_INFO_PREFIX=$1
+
+for file in ada-mode auth autotype calc ccmode cl dbus dired-x \
+           ebrowse ediff efaq eintr elisp emacs emacs-mime epa erc \
+           eshell eudc flymake forms gnus idlwave info mairix-el \
+           message mh-e newsticker nxml-mode org pcl-cvs pgg rcirc \
+           reftex remember sasl sc ses sieve smtpmail speedbar tramp \
+           url vip viper widget woman
+do
+    ./mangle-info ${THE_INFO_PREFIX}/${file}
+done
\ No newline at end of file
diff --exclude='*~' -ruN emacs-23.1/lib-src/Makefile.in emacs-23.1.patched/lib-src/Makefile.in
--- emacs-23.1/lib-src/Makefile.in	2009-06-21 00:37:34.000000000 -0400
+++ emacs-23.1.patched/lib-src/Makefile.in	2010-07-18 03:29:30.000000000 -0400
@@ -306,13 +306,6 @@
 	$(top_srcdir)/mkinstalldirs $(DESTDIR)${gamedir}
 	touch $(DESTDIR)${gamedir}/snake-scores
 	touch $(DESTDIR)${gamedir}/tetris-scores
-/* If the following commands fail, that is not a big deal.
-   update-game-score will detect at runtime that it is not setuid,
-   and handle things accordingly. */
-	-if chown ${gameuser} $(DESTDIR)${archlibdir}/update-game-score && chmod u+s $(DESTDIR)${archlibdir}/update-game-score; then \
-	  chown ${gameuser} $(DESTDIR)${gamedir}; \
-	  chmod u=rwx,g=rwx,o=rx $(DESTDIR)${gamedir}; \
-	fi
         if [ `(cd $(DESTDIR)${archlibdir} && /bin/pwd)` \
              != `(cd ${srcdir} && /bin/pwd)` ]; then \
 	  for file in ${SCRIPTS}; do \
diff --exclude='*~' -ruN emacs-23.1/lisp/startup.el emacs-23.1.patched/lisp/startup.el
--- emacs-23.1/lisp/startup.el	2009-06-21 06:37:46.000000000 +0200
+++ emacs-23.1.patched/lisp/startup.el	2009-08-26 16:57:34.000000000 +0200
@@ -375,6 +375,10 @@
 
 (defvar normal-top-level-add-subdirs-inode-list nil)
 
+(defconst fink-emacs-flavor 'emacs23
+  "A symbol representing the particular fink flavor of emacs running.
+Something like 'emacs20, 'xemacs20, etc.")
+
 (defvar no-blinking-cursor nil)
 
 (defvar default-frame-background-mode)
@@ -977,9 +981,19 @@
     ;; be loaded from site-run-file and wants to test if -q was given
     ;; should check init-file-user instead, since that is already set.
     ;; See cus-edit.el for an example.
-    (if site-run-file
-	(load site-run-file t t))
-
+    ;; (if site-run-file                                                       
+    ;;     (load site-run-file t t))                                           
+                                                                               
+    ;; Fink version of site-start.                                           
+    (if site-run-file                                                          
+ 	(progn                                                                 
+ 	  ;; Load all the fink package snippets.                             
+ 	  ;; It's in here because we want -q to kill it too.                   
+ 	  (if (load "fink-startup" t t nil)                                  
+ 	      (fink-startup fink-emacs-flavor))                            
+ 	  ;; Now the normal site file...                                       
+ 	  (load site-run-file t t nil)))
+    
     ;; Sites should not disable this.  Only individuals should disable
     ;; the startup screen.
     (setq inhibit-startup-screen nil)
diff --exclude='*~' -ruN emacs-23.1/mangle-info emacs-23.1.patched/mangle-info
--- emacs-23.1/mangle-info	1970-01-01 01:00:00.000000000 +0100
+++ emacs-23.1.patched/mangle-info	2009-08-26 16:56:06.000000000 +0200
@@ -0,0 +1,18 @@
+#!/usr/bin/perl -w -i
+
+use English;
+use strict;
+
+$RS = undef;
+
+my $prefix = "emacs-23";
+
+my $x = <>;
+if(!($x =~ m/^(\s*START-INFO-DIR-ENTRY\s+\*\s*[^:]+:\s*)\(([^\)]+)\)/mo)) {
+  die "Couldn't find START-INFO-DIR-ENTRY.";
+}
+   
+print ${PREMATCH};
+print "$1($prefix/$2)";
+print ${POSTMATCH};
+
diff --exclude='*~' -ruN emacs-23.1/src/Makefile.in emacs-23.1.patched/src/Makefile.in
--- emacs-23.1/src/Makefile.in	2009-07-01 08:24:14.000000000 -0400
+++ emacs-23.1.patched/src/Makefile.in	2010-07-21 04:27:48.000000000 -0400
@@ -253,8 +253,8 @@
 
 /* C_SWITCH_X_SITE must come before C_SWITCH_X_MACHINE and C_SWITCH_X_SYSTEM
    since it may have -I options that should override those two.  */
-ALL_CFLAGS=-Demacs -DHAVE_CONFIG_H $(MYCPPFLAGS) -I. -I${srcdir} C_SWITCH_MACHINE C_SWITCH_SYSTEM C_SWITCH_X_SITE C_SWITCH_X_MACHINE C_SWITCH_X_SYSTEM C_SWITCH_SYSTEM_TEMACS ${CFLAGS_SOUND} ${RSVG_CFLAGS} ${DBUS_CFLAGS} ${CFLAGS} @FREETYPE_CFLAGS@ @FONTCONFIG_CFLAGS@ @LIBOTF_CFLAGS@ @M17N_FLT_CFLAGS@
+ALL_CFLAGS=-Demacs -DHAVE_CONFIG_H $(MYCPPFLAGS) -I. -I${srcdir} C_SWITCH_MACHINE C_SWITCH_SYSTEM C_SWITCH_SYSTEM_TEMACS @XFT_CFLAGS@ @FREETYPE_CFLAGS@ @FONTCONFIG_CFLAGS@ ${CFLAGS_SOUND} ${RSVG_CFLAGS} ${DBUS_CFLAGS} ${CFLAGS} @LIBOTF_CFLAGS@ @M17N_FLT_CFLAGS@ C_SWITCH_X_SITE C_SWITCH_X_MACHINE C_SWITCH_X_SYSTEM
 ALL_OBJC_CFLAGS=$(ALL_CFLAGS) @GNU_OBJC_CFLAGS@
 
 .SUFFIXES: .m
 #endif /* not HAVE_X11 */
--- emacs-23.1/src/unexmacosx.c	2009-09-02 12:46:14.000000000 +0200
+++ emacs-23.1.patched/src/unexmacosx.c	2009-09-02 12:53:34.000000000 +0200
@@ -582,6 +582,14 @@
       printf ("LC_UUID          ");
       break;
 #endif
+#ifdef LC_DYLD_INFO
+    case LC_DYLD_INFO:
+      printf ("LC_DYLD_INFO ");
+      break;
+    case LC_DYLD_INFO_ONLY:
+      printf ("LC_DYLD_INFO_ONLY");
+      break;
+#endif
     default:
       printf ("unknown          ");
     }
@@ -819,6 +827,7 @@
 	       || strncmp (sectp->sectname, "__const", 16) == 0
 	       || strncmp (sectp->sectname, "__cfstring", 16) == 0
 	       || strncmp (sectp->sectname, "__gcc_except_tab", 16) == 0
+	       || strncmp (sectp->sectname, "__program_vars", 16) == 0
 	       || strncmp (sectp->sectname, "__objc_", 7) == 0)
 	{
 	  if (!unexec_copy (sectp->offset, old_file_offset, sectp->size))
@@ -1086,6 +1095,36 @@
   curr_header_offset += lc->cmdsize;
 }
 
+#ifdef LC_DYLD_INFO
+/* Copy a LC_DYLD_INFO(_ONLY) load command from the input file to the output
+   file, adjusting the file offset fields. */
+static void
+copy_dyld_info (struct load_command *lc, long delta)
+{
+  struct dyld_info_command *dip = (struct dyld_info_command *) lc;
+
+  if (dip->rebase_off > 0)
+    dip->rebase_off += delta;
+  if (dip->bind_off > 0)
+    dip->bind_off += delta;
+  if (dip->weak_bind_off > 0)
+    dip->weak_bind_off += delta;
+  if (dip->lazy_bind_off > 0)
+    dip->lazy_bind_off += delta;
+  if (dip->export_off > 0)
+    dip->export_off += delta;
+
+  printf ("Writing ");
+  print_load_command_name (lc->cmd);
+  printf (" command\n");
+
+  if (!unexec_write (curr_header_offset, lc, lc->cmdsize))
+    unexec_error ("cannot write dyld info command to header");
+
+  curr_header_offset += lc->cmdsize;
+}
+#endif
+
 /* Copy other kinds of load commands from the input file to the output
    file, ones that do not require adjustments of file offsets.  */
 static void
@@ -1152,6 +1191,12 @@
       case LC_TWOLEVEL_HINTS:
 	copy_twolevelhints (lca[i], linkedit_delta);
 	break;
+#ifdef LC_DYLD_INFO
+      case LC_DYLD_INFO:
+      case LC_DYLD_INFO_ONLY:
+        copy_dyld_info (lca[i], linkedit_delta);
+        break;
+#endif
       default:
 	copy_other (lca[i]);
 	break;
--- emacs-23.1/src/s/darwin.h	2009-09-02 12:54:24.000000000 +0200
+++ emacs-23.1.patched/src/s/darwin.h	2009-09-02 12:56:35.000000000 +0200
@@ -171,6 +171,9 @@
 #define malloc unexec_malloc
 #define realloc unexec_realloc
 #define free unexec_free
+/* Don't use posix_memalign because it is not compatible with
+   unexmacosx.c. */
+#undef HAVE_POSIX_MEMALIGN
 #endif
 
 /* The ncurses library has been moved out of the System framework in
--- emacs-23.1/src/xfns.c	2009-08-31 00:03:02.000000000 +0100
+++ emacs-23.1.patched/src/xfns.c	2009-08-31 00:21:39.000000000 +0100
@@ -468,8 +468,10 @@
       if (!FRAME_X_P (f) || FRAME_X_DISPLAY_INFO (f) != dpyinfo)
 	continue;
       x = f->output_data.x;
-      /* Match if the window is this frame's menubar.  */
 #ifdef USE_GTK
+      /* Match if the window is this frame's window (button presses are no longer
+         caught by the menubar) or a menu in the menubar.
+       */
       if (x->menubar_widget)
         {
           GtkWidget *gwdesc = xg_win_to_widget (dpyinfo->display, wdesc);
@@ -477,13 +479,14 @@
 
           BLOCK_INPUT;
           if (gwdesc != 0
-              && (gwdesc == x->menubar_widget
+              && (gwdesc == gtk_widget_get_parent (gtk_widget_get_parent (x->menubar_widget))
                   || gtk_widget_get_parent (gwdesc) == x->menubar_widget))
             found = 1;
           UNBLOCK_INPUT;
           if (found) return f;
         }
 #else
+      /* Match if the window is in this frame's menubar */
       if (x->menubar_widget
 	  && lw_window_is_in_menubar (wdesc, x->menubar_widget))
 	return f;
