diff -x '*~' -urN /Users/ridgway/Desktop/cedet-1.0pre4/fink/cedet-install /Users/ridgway/Desktop/cedet-1.0pre4.copy/fink/cedet-install
--- jdee-2.3.5.1/fink/jdee-install	1969-12-31 19:00:00.000000000 -0500
+++ jdee-2.3.5.1/fink/jdee-install	2007-09-05 00:50:27.000000000 -0400
@@ -0,0 +1,59 @@
+#! /bin/bash -e
+#
+# <% elisp %> Emacs install script for jdee
+# Adapted from the Emacs install script for auctex by John Ridgway
+#
+# Original Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04, 05, 06
+# by Davide G. M. Salvetti.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
+
+set -o posix
+
+FLAVOR=${1}
+
+INSTALL="install -o root -g admin -m 644"
+INSTDIR="${INSTALL} -m 755 -d"
+
+do_install () {
+    /bin/echo >&2 "install/jdee: Setting up for ${FLAVOR}... " \
+    "(log file: @PREFIX@/share/${FLAVOR}/site-lisp/jdee/CompilationLog)... \c"
+    cd @PREFIX@/share/emacs/site-lisp/jdee/; umask 0022;
+    ${INSTDIR} @PREFIX@/share/${FLAVOR}/site-lisp/jdee/java
+    cp -pr @PREFIX@/share/emacs/site-lisp/jdee/* @PREFIX@/share/${FLAVOR}/site-lisp/jdee/
+    cd @PREFIX@/share/${FLAVOR}/site-lisp/jdee
+    make EMACS=${FLAVOR} ELIB="../elib" CEDET="../cedet" \
+	>> @PREFIX@/share/${FLAVOR}/site-lisp/jdee/CompilationLog 2>&1
+     rm -f @PREFIX@/share/${FLAVOR}/site-lisp/jdee/java/lib \
+	>> @PREFIX@/share/${FLAVOR}/site-lisp/jdee/CompilationLog 2>&1
+     ln -s @PREFIX@/share/java/jdee @PREFIX@/share/${FLAVOR}/site-lisp/jdee/java/lib \
+	>> @PREFIX@/share/${FLAVOR}/site-lisp/jdee/CompilationLog 2>&1
+     ln -s @PREFIX@/share/doc/jdee @PREFIX@/share/${FLAVOR}/site-lisp/jdee/doc \
+	>> @PREFIX@/share/${FLAVOR}/site-lisp/jdee/CompilationLog 2>&1
+    gzip -9fq @PREFIX@/share/${FLAVOR}/site-lisp/jdee/CompilationLog
+    echo >&2 "done."
+    return 0
+}
+
+case "${FLAVOR}" in
+    (emacs) : ;;
+    (emacs21 | emacs22)
+        do_install ;;
+    (*) echo >&2 "install/jdee:" \
+	"Ignoring emacsen flavor: \"${FLAVOR}\"."
+    ;;
+esac
+
+exit 0
diff -x '*~' -urN /Users/ridgway/Desktop/jdee-1.0pre4/fink/jdee-remove /Users/ridgway/Desktop/jdee-1.0pre4.copy/fink/jdee-remove
--- jdee-1.0pre4/fink/jdee-remove	1969-12-31 19:00:00.000000000 -0500
+++ jdee-1.0pre4.copy/fink/jdee-remove	2007-09-05 00:50:13.000000000 -0400
@@ -0,0 +1,46 @@
+#! /bin/bash -e
+#
+# <% elisp %> Emacs remove script for jdee
+# adapted from
+# <% elisp %> Emacs remove script for auctex
+# by John Ridgway, 2007
+#
+# Original
+# Copyright (C) 1997, 98, 99, 2000, 01, 02, 03, 04, 05, 06
+# by Davide G. M. Salvetti.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to: The Free Software Foundation, Inc.,
+# 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
+
+set -o posix
+
+FLAVOR=${1}
+
+undo_install () {
+    /bin/echo >&2 "remove/jdee: Removing for ${FLAVOR}... \c"
+    rm -rf @PREFIX@/share/${FLAVOR}/site-lisp/jdee
+    echo >&2 "done."
+    return 0
+}
+
+case "${FLAVOR}" in
+    (emacs) : ;;
+    (emacs21 | emacs22)
+        undo_install ;;
+    (*) echo >&2 "remove/jdee:" \
+	"Ignoring emacsen flavor: \"${FLAVOR}\"."
+    ;;
+esac
+
+exit 0
diff -x '*~' -urN /Users/ridgway/Desktop/jdee-1.0pre4/fink/jdee.el /Users/ridgway/Desktop/jdee-1.0pre4.copy/fink/jdee.el
--- jdee-1.0pre4/fink/jdee.el	1969-12-31 19:00:00.000000000 -0500
+++ jdee-1.0pre4.copy/fink/jdee.el	2007-09-05 00:50:13.000000000 -0400
@@ -0,0 +1,49 @@
+;; This .emacs file illustrates the minimal setup required to run the JDE.
+
+;; Set the debug option to enable a backtrace when a
+;; problem occurs.
+(setq debug-on-error t)
+
+;; Update the Emacs load-path to include the path to
+;; the JDE and its required packages. 
+;;(add-to-list 'load-path (expand-file-name "~/emacs/site/jde/lisp"))
+;;(add-to-list 'load-path (expand-file-name "~/emacs/site/cedet/common"))
+;;(add-to-list 'load-path (expand-file-name "~/emacs/site/elib"))
+
+;; Initialize CEDET.
+(load-file "@PREFIX@/share/emacs/site-lisp/cedet/common/cedet.el")
+
+;; If you want Emacs to defer loading the JDE until you open a 
+;; Java file, edit the following line
+(setq defer-loading-jde nil)
+;; to read:
+;;
+;;  (setq defer-loading-jde t)
+;;
+
+(if defer-loading-jde
+    (progn
+      (autoload 'jde-mode "jde" "JDE mode." t)
+      (setq auto-mode-alist
+	    (append
+	     '(("\\.java\\'" . jde-mode))
+	     auto-mode-alist)))
+  (require 'jde))
+
+
+;; Sets the basic indentation for Java source files
+;; to two spaces.
+(defun my-jde-mode-hook ()
+  (setq c-basic-offset 2))
+
+(add-hook 'jde-mode-hook 'my-jde-mode-hook)
+
+;; Include the following only if you want to run
+;; bash as your shell.
+
+;; Setup Emacs to run bash as its primary shell.
+(setq shell-file-name "bash")
+(setq shell-command-switch "-c")
+(setq explicit-shell-file-name shell-file-name)
+(setenv "SHELL" shell-file-name)
+(setq explicit-sh-args '("-login" "-i"))
diff -x '*~' -urN /Users/ridgway/Desktop/jdee-1.0pre4/fink/jdee.el /Users/ridgway/Desktop/jdee-1.0pre4.copy/fink/jdee.el
--- jdee-1.0pre4/lisp/beanshell.el	1969-12-31 19:00:00.000000000 -0500
+++ jdee-1.0pre4.copy/lisp/beanshell.el	2007-09-05 00:50:13.000000000 -0400
@@ -84,7 +84,7 @@
   :group 'tools
   :prefix "bsh-")
 
-(defcustom bsh-jar "bsh.jar"
+(defcustom bsh-jar "@PREFIX@/share/java/jdee/bsh.jar"
   "Path to the jar file containing the BeanShell classes."
   :group 'bsh
   :type 'file)
@@ -503,7 +503,7 @@
   (assert 
    (file-exists-p (oref this jar))
    nil 
-   "Specified BeanShell jar filed does not exist: %s" (oref this jar))
+   "Specified BeanShell jar file does not exist: %s" (oref this jar))
 
 
   (if (not (bsh-running-p this))
diff -x '*~' -urN /Users/ridgway/Desktop/jdee-1.0pre4/lisp/jde-util.el /Users/ridgway/Desktop/jdee-1.0pre4.copy/lisp/jde-util.el
--- jdee-1.0pre4/lisp/jde-util.el	1969-12-31 19:00:00.000000000 -0500
+++ jdee-1.0pre4.copy/lisp/jde-util.el	2007-09-05 00:50:13.000000000 -0400
@@ -232,11 +232,10 @@
 directory that results from unpacking the distributable.
 On installations based on the version of the JDEE 
 packaged with XEmacs, the root directory is 
-xemacs-packages/lisp."
-  (let ((directory-sep-char ?/))
-    (expand-file-name 
-	     "../" 
-	     (file-name-directory (locate-library "jde")))))
+xemacs-packages/lisp.  On fink installations the root
+directory is the lisp directory rather than the 
+directory above it."
+  (file-name-directory (locate-library "jde")))
 
 (defun jde-find-jde-data-directory ()
   "Return the path of the JDE data directory.
@@ -248,8 +247,8 @@
   (let ((directory-sep-char ?/))
     (if (featurep 'xemacs)
 	(let ((dir (locate-data-directory "jde")))
-	  (if dir dir (jde-root))))
-    (jde-root)))
+ 	   (if dir dir (jde-root)))
+        (jde-root))))
 
 (defun jde-temp-directory ()
 "Get the location used by the host system to store temporary files."
diff -x '*~' -urN /Users/ridgway/Desktop/jdee-1.0pre4/lisp/jde.el /Users/ridgway/Desktop/jdee-1.0pre4.copy/lisp/jde.el
