diff --unidirectional-new-file -U 0 -rx .svn tags/nano_2_2_5/NEWS trunk/nano/NEWS
--- tags/nano_2_2_5/NEWS	2010-08-15 00:17:22.000000000 +0000
+++ trunk/nano/NEWS	2010-08-15 00:16:55.000000000 +0000
@@ -7 +7 @@
-		"allow_insecure_backup" to be even more permission and
+		"allow_insecure_backup" to be even more permissive and
diff --unidirectional-new-file -U 0 -rx .svn tags/nano_2_2_5/configure.ac trunk/nano/configure.ac
--- tags/nano_2_2_5/configure.ac	2010-08-15 00:17:22.000000000 +0000
+++ trunk/nano/configure.ac	2010-08-15 00:16:55.000000000 +0000
@@ -23 +23 @@
-AC_INIT([GNU nano], [2.2.5], [nano-devel@gnu.org], [nano])
+AC_INIT([GNU nano], [2.2.99-svn], [nano-devel@gnu.org], [nano])
