diff -Naur --exclude='*~' org-6.35g/Makefile org-6.35g.fink/Makefile
--- org-6.35g/Makefile	2010-04-08 19:16:20.000000000 +0100
+++ org-6.35g.fink/Makefile	2010-04-10 14:52:48.000000000 +0100
@@ -29,7 +29,7 @@
 # Using emacs in batch mode.
 
 BATCH=$(EMACS) -batch -q -no-site-file -eval                             \
-  "(progn (add-to-list (quote load-path) \"$(lispdir)\") (add-to-list (quote load-path) (expand-file-name \"./lisp/\")))"
+  "(progn (add-to-list (quote load-path) \"$(lispdir)\") (add-to-list (quote load-path) (expand-file-name \".\")))"
 
 # Specify the byte-compiler for compiling org-mode files
 ELC= $(BATCH) -f batch-byte-compile
@@ -112,8 +112,8 @@
 		org-wl.el		\
 		org-xoxo.el
 
-LISPFILES0 = $(LISPF:%=lisp/%)
-LISPFILES  = $(LISPFILES0) lisp/org-install.el
+LISPFILES0 = $(LISPF)
+LISPFILES  = $(LISPFILES0) org-install.el
 ELCFILES0  = $(LISPFILES0:.el=.elc)
 ELCFILES   = $(LISPFILES:.el=.elc)
 DOCFILES   = doc/org.texi doc/org.pdf doc/org doc/dir doc/.nosearch
@@ -167,9 +167,9 @@
 	if [ ! -d $(lispdir) ]; then $(MKDIR) $(lispdir); else true; fi ;
 	$(CP) xemacs/noutline.el xemacs/noutline.elc $(lispdir)
 
-autoloads: lisp/org-install.el
+autoloads: org-install.el
 
-lisp/org-install.el: $(LISPFILES0) Makefile
+org-install.el: $(LISPFILES0) Makefile
 	$(BATCH) --eval "(require 'autoload)" \
 		--eval '(find-file "org-install.el")'  \
 		--eval '(erase-buffer)' \
@@ -225,7 +225,7 @@
 	${MAKE} cleancontrib
 	${MAKE} info
 	${MAKE} doc
-	${MAKE} lisp/org-install.el
+	${MAKE} org-install.el
 	rm -rf org-$(TAG) org-$(TAG).zip
 	$(MKDIR) org-$(TAG)
 	$(MKDIR) org-$(TAG)/xemacs
@@ -269,7 +269,7 @@
 	${MAKE} upload_manual
 
 db:
-	grep -e '(debug)' lisp/*el
+	grep -e '(debug)' *el
 
 cleancontrib:
 	find contrib -name \*~ -exec rm {} \;
@@ -296,7 +296,7 @@
 
 cleanall:
 	${MAKE} clean
-	rm -f lisp/org-install.el
+	rm -f org-install.el
 
 .el.elc:
 	$(ELC) $<
@@ -318,57 +318,57 @@
 
 # Dependencies
 
-lisp/org.elc:		lisp/org-macs.el lisp/org-compat.el lisp/org-faces.el
-lisp/org-agenda.elc:	lisp/org.el
-lisp/org-ascii.elc:	lisp/org-exp.el
-lisp/org-attach.elc:	lisp/org.el lisp/org-id.el
-lisp/org-archive.elc:	lisp/org.el
-lisp/org-bbdb.elc:	lisp/org.el
-lisp/org-beamer.elc:	lisp/org.el
-lisp/org-bibtex.elc:	lisp/org.el
-lisp/org-clock.elc:	lisp/org.el
-lisp/org-colview.elc:	lisp/org.el
-lisp/org-colview-xemacs.elc:	lisp/org.el
-lisp/org-compat.elc:	lisp/org-macs.el
-lisp/org-crypt.elc:	lisp/org-crypt.el lisp/org.el
-lisp/org-ctags.elc:	lisp/org.el
-lisp/org-datetree.elc:	lisp/org.el
-lisp/org-docview.elc:	lisp/org.el
-lisp/org-entities.elc:	
-lisp/org-exp.elc:	lisp/org.el lisp/org-agenda.el
-lisp/org-exp-blocks.elc: lisp/org.el
-lisp/org-latex.elc:	lisp/org.el lisp/org-exp.el lisp/org-beamer.el
-lisp/org-docbook.elc:	lisp/org.el lisp/org-exp.el
-lisp/org-faces.elc:	lisp/org-macs.el lisp/org-compat.el
-lisp/org-feed.elc:	lisp/org.el
-lisp/org-footnotes.elc:	lisp/org-macs.el lisp/org-compat.el
-lisp/org-freemind.elc:	lisp/org.el
-lisp/org-gnus.elc:	lisp/org.el
-lisp/org-html.elc:	lisp/org-exp.el
-lisp/org-habit.elc:	lisp/org.el lisp/org-agenda.el
-lisp/org-icalendar.elc:	lisp/org-exp.el
-lisp/org-id.elc:	lisp/org.el
-lisp/org-indent.elc:	lisp/org.el lisp/org-macs.el lisp/org-compat.el
-lisp/org-info.elc:	lisp/org.el
-lisp/org-inlinetask.elc:
-lisp/org-irc.elc:	lisp/org.el
-lisp/org-jsinfo.elc:	lisp/org.el lisp/org-exp.el
-lisp/org-list.elc:	lisp/org-macs.el lisp/org-compat.el
-lisp/org-mac-message.elc:	lisp/org.el
-lisp/org-macs.elc:
-lisp/org-mew.elc:	lisp/org.el
-lisp/org-mhe.elc:	lisp/org.el
-lisp/org-mobile.elc:	lisp/org.el
-lisp/org-mouse.elc:	lisp/org.el
-lisp/org-plot.elc:	lisp/org.el lisp/org-exp.el lisp/org-table.el
-lisp/org-publish.elc:
-lisp/org-protocol.elc:	lisp/org.el
-lisp/org-remember.elc:	lisp/org.el
-lisp/org-rmail.elc:	lisp/org.el
-lisp/org-src.elc:	lisp/org-macs.el lisp/org-compat.el
-lisp/org-table.elc:	lisp/org.el
-lisp/org-timer.elc:	lisp/org.el
-lisp/org-vm.elc:	lisp/org.el
-lisp/org-w3m.elc:	lisp/org.el
-lisp/org-wl.elc:	lisp/org.el
-lisp/org-xoxo.elc:	lisp/org-exp.el
\ No newline at end of file
+org.elc:		org-macs.el org-compat.el org-faces.el
+org-agenda.elc:	org.el
+org-ascii.elc:	org-exp.el
+org-attach.elc:	org.el org-id.el
+org-archive.elc:	org.el
+org-bbdb.elc:	org.el
+org-beamer.elc:	org.el
+org-bibtex.elc:	org.el
+org-clock.elc:	org.el
+org-colview.elc:	org.el
+org-colview-xemacs.elc:	org.el
+org-compat.elc:	org-macs.el
+org-crypt.elc:	org-crypt.el org.el
+org-ctags.elc:	org.el
+org-datetree.elc:	org.el
+org-docview.elc:	org.el
+org-entities.elc:	
+org-exp.elc:	org.el org-agenda.el
+org-exp-blocks.elc: org.el
+org-latex.elc:	org.el org-exp.el org-beamer.el
+org-docbook.elc:	org.el org-exp.el
+org-faces.elc:	org-macs.el org-compat.el
+org-feed.elc:	org.el
+org-footnotes.elc:	org-macs.el org-compat.el
+org-freemind.elc:	org.el
+org-gnus.elc:	org.el
+org-html.elc:	org-exp.el
+org-habit.elc:	org.el org-agenda.el
+org-icalendar.elc:	org-exp.el
+org-id.elc:	org.el
+org-indent.elc:	org.el org-macs.el org-compat.el
+org-info.elc:	org.el
+org-inlinetask.elc:
+org-irc.elc:	org.el
+org-jsinfo.elc:	org.el org-exp.el
+org-list.elc:	org-macs.el org-compat.el
+org-mac-message.elc:	org.el
+org-macs.elc:
+org-mew.elc:	org.el
+org-mhe.elc:	org.el
+org-mobile.elc:	org.el
+org-mouse.elc:	org.el
+org-plot.elc:	org.el org-exp.el org-table.el
+org-publish.elc:
+org-protocol.elc:	org.el
+org-remember.elc:	org.el
+org-rmail.elc:	org.el
+org-src.elc:	org-macs.el org-compat.el
+org-table.elc:	org.el
+org-timer.elc:	org.el
+org-vm.elc:	org.el
+org-w3m.elc:	org.el
+org-wl.elc:	org.el
+org-xoxo.elc:	org-exp.el
\ No newline at end of file
diff -Naur --exclude='*~' org-6.35g/fink/50org.el org-6.35g.fink/fink/50org.el
--- org-6.35g/fink/50org.el	1970-01-01 01:00:00.000000000 +0100
+++ org-6.35g.fink/fink/50org.el	2010-04-10 14:38:45.000000000 +0100
@@ -0,0 +1,3 @@
+(require 'org-install)
+(add-to-list 'auto-mode-alist '("\\.org\\'" . org-mode))
+(add-hook 'org-mode-hook 'turn-on-font-lock)  ; org-mode buffers only
diff -Naur --exclude='*~' org-6.35g/fink/emacsen-install org-6.35g.fink/fink/emacsen-install
--- org-6.35g/fink/emacsen-install	1970-01-01 01:00:00.000000000 +0100
+++ org-6.35g.fink/fink/emacsen-install	2010-04-10 14:55:51.000000000 +0100
@@ -0,0 +1,54 @@
+#!/bin/bash -e
+
+set -o posix
+
+FLAVOR=${1}
+
+if [ ${FLAVOR} == emacs ]
+then
+    # This isn't really an emacs.  No need to print anything
+    exit 0
+fi
+
+echo >&2 "install/org-mode: Handling install of emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "install/org-mode: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+if [ ${FLAVOR} == emacs21 ]
+then
+    echo "install/org-mode: Warning: org-mode may not work properly with emacs21"
+    echo "install/org-mode: Warning: consider updating to emacs version 22 or hiher;"
+    exit 0
+fi
+
+if [ ${FLAVOR} == xemacs ]
+then
+    echo "install/org-mode: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+# Install base elisp
+mkdir -p @PREFIX@/share/${FLAVOR}/site-lisp/org
+for i in org-agenda.el org-archive.el org-ascii.el org-attach.el org-bbdb.el org-beamer.el org-bibtex.el org-clock.el org-colview-xemacs.el org-colview.el org-compat.el org-crypt.el org-ctags.el org-datetree.el org-docbook.el org-docview.el org-entities.el org-exp-blocks.el org-exp.el org-faces.el org-feed.el org-footnote.el org-freemind.el org-gnus.el org-habit.el org-html.el org-icalendar.el org-id.el org-indent.el org-info.el org-inlinetask.el org-install.el org-irc.el org-jsinfo.el org-latex.el org-list.el org-mac-message.el org-macs.el org-mew.el org-mhe.el org-mobile.el org-mouse.el org-plot.el org-protocol.el org-publish.el org-remember.el org-rmail.el org-src.el org-table.el org-timer.el org-vm.el org-w3m.el org-wl.el org-xoxo.el org.el Makefile; do
+  ln -s -f @PREFIX@/share/emacs/site-lisp/org/${i} @PREFIX@/share/${FLAVOR}/site-lisp/org;
+done
+
+echo >&2 -n "install/org-mode: Byte-compiling for ${FLAVOR}..."
+
+cd @PREFIX@/share/${FLAVOR}/site-lisp/org;
+
+(make EMACS=${FLAVOR} PREFIX="@PREFIX@" default --keep-going 2>&1) \
+  > @PREFIX@/share/doc/org-mode/CompilationLog-${FLAVOR}
+
+rm -f Makefile
+
+gzip -9qf @PREFIX@/share/doc/org-mode/CompilationLog-${FLAVOR};
+
+echo >&2 "done."
+echo >&2 "install/org-mode: Compilation log saved in @PREFIX@/share/doc/org-modeCompilationLog-${FLAVOR}.gz."
+
+exit 0
diff -Naur --exclude='*~' org-6.35g/fink/emacsen-install-babel org-6.35g.fink/fink/emacsen-install-babel
--- org-6.35g/fink/emacsen-install-babel	1970-01-01 01:00:00.000000000 +0100
+++ org-6.35g.fink/fink/emacsen-install-babel	2010-04-10 14:57:16.000000000 +0100
@@ -0,0 +1,70 @@
+#!/bin/bash -e
+
+set -o posix
+
+FLAVOR=${1}
+
+if [ ${FLAVOR} == emacs ]
+then
+    # No compilation done; no need to print anything
+    exit 0
+fi
+
+echo >&2 "install/org-babel: Handling install of emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "install/org-babel: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+if [ ${FLAVOR} == emacs21 ]
+then
+    echo "install/org-babel: Warning: org-babel may not work properly with emacs1;"
+    echo "install/org-babel: Warning: consider updating to emacs version 22 or hgher;"
+    exit 0
+fi
+
+if [ ${FLAVOR} == xemacs ]
+then
+    echo "install/org-babel: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+# Install babel core
+mkdir -p @PREFIX@/share/${FLAVOR}/site-lisp/org-babel;
+for i in org-babel-comint.el org-babel-exp.el org-babel-keys.el org-babel-lob.el org-babel-ref.el org-babel-table.el org-babel-tangle.el org-babel.el;
+  ln -s -f @PREFIX@/share/emacs/site-lisp/org-babel/${i} @PREFIX@/share/${FLAVOR/site-lisp/org-babel;
+done
+
+# Install babel languages
+mkdir -p @PREFIX@/share/${FLAVOR}/site-lisp/org-babel/langs
+for i in org-babel-C.el org-babel-R.el org-babel-asymptote.el org-babel-clojure.el org-babel-css.el org-babel-ditaa.el org-babel-dot.el org-babel-emacs-lisp.el org-babel-gnuplot.el org-babel-haskell.el org-babel-latex.el org-babel-matlab.el org-babel-ocaml.el org-babel-octave.el org-babel-oz.el org-babel-perl.el org-babel-python.el org-babel-ruby.el org-babel-sass.el org-babel-screen.el org-babel-sh.el org-babel-sql.el;
+  ln -s -f @PREFIX@/share/emacs/site-lisp/org-babel/langs/${i} @PREFIX@/share/${LAVOR}/site-lisp/org-babel/langs;
+done
+
+echo >&2 -n "install/org-babel: Byte-compiling for ${FLAVOR}..."
+
+cd @PREFIX@/share/${FLAVOR}/site-lisp/org-babel;
+
+# Compile core
+(${FLAVOR} -batch --no-init-file --no-site-file \
+  --execute '(push "@PREFIX@/share/${FLAVOR}/site-lisp/org" load-path)' \
+  -f batch-byte-compile *.el 2>&1) \
+  > @PREFIX@/share/doc/org-babel/CompilationLog-${FLAVOR}
+
+cd langs;
+
+# Compile languages
+(${FLAVOR} -batch --no-init-file --no-site-file \
+  --execute '(push ".." load-path)' \
+  --execute '(push "@PREFIX@/share/${FLAVOR}/site-lisp/org" load-path)' \
+  -f batch-byte-compile *.el 2>&1) \
+  >> @PREFIX@/share/doc/org-babel/CompilationLog-${FLAVOR}
+
+gzip -9qf @PREFIX@/share/doc/org-babel/CompilationLog-${FLAVOR};
+
+echo >&2 "done."
+echo >&2 "install/org-babel: Compilation log saved in @PREFIX@/share/doc/org-babl/CompilationLog-${FLAVOR}.gz."
+
+exit 0
diff -Naur --exclude='*~' org-6.35g/fink/emacsen-install-contrib org-6.35g.fink/fink/emacsen-install-contrib
--- org-6.35g/fink/emacsen-install-contrib	1970-01-01 01:00:00.000000000 +0100
+++ org-6.35g.fink/fink/emacsen-install-contrib	2010-04-15 22:36:50.000000000 +0100
@@ -0,0 +1,54 @@
+#!/bin/bash -e
+
+set -o posix
+
+FLAVOR=${1}
+
+if [ ${FLAVOR} == emacs ]
+then
+    # No compilation done; no need to print anything
+    exit 0
+fi
+
+echo >&2 "install/org-mode-contrib: Handling install of emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "install/org-mode-contrib: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+if [ ${FLAVOR} == emacs21 ]
+then
+    echo "install/org-mode-contrib: Warning: org-mode-contrib may not work propely with emacs21;"
+    echo "install/org-mode-contrib: Warning: consider updating to emacs version 2 or higher;"
+    exit 0
+fi
+
+if [ ${FLAVOR} == xemacs ]
+then
+    echo "install/org-mode-contrib: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+# Install contributed code
+mkdir -p @PREFIX@/share/${FLAVOR}/site-lisp/org/contrib;
+for i in org-R.el org-annotate-file.el org-babel-init.el org-bookmark.el org-cheklist.el org-choose.el org-collector.el org-contribdir.el org-depend.el org-elis-symbol.el org-eval-light.el org-eval.el org-exp-bibtex.el org-expiry.el org-exprt-generic.el org-git-link.el org-interactive-query.el org-invoice.el org-jira.e org-learn.el org-mac-iCal.el org-mairix.el org-man.el org-mtags.el org-panel.elorg-registry.el org-screen.el org-special-blocks.el org-toc.el org-track.el org2em.el orgtbl-sqlinsert.el; do
+  ln -s -f @PREFIX@/share/emacs/site-lisp/org/contrib/${i} @PREFIX@/share/${FLAVOR}/site-lisp/org/contrib;
+done
+
+echo >&2 -n "install/org-mode-contrib: Byte-compiling for ${FLAVOR}..."
+
+cd @PREFIX@/share/${FLAVOR}/site-lisp/org;
+
+(${FLAVOR} -batch --no-init-file --no-site-file \
+  --execute '(push "@PREFIX@/share/${FLAVOR}/site-lisp/org" load-path)' \
+  -f batch-byte-compile *.el 2>&1) \
+  > @PREFIX@/share/doc/org-mode-contrib/CompilationLog-${FLAVOR}
+
+gzip -9qf @PREFIX@/share/doc/org-mode-contrib/CompilationLog-${FLAVOR};
+
+echo >&2 "done."
+echo >&2 "install/org-mode-contrib: Compilation log saved in @PREFIX@/share/doc/org-mode-contrib/CompilationLog-${FLAVOR}.gz."
+
+exit 0
diff -Naur --exclude='*~' org-6.35g/fink/emacsen-remove org-6.35g.fink/fink/emacsen-remove
--- org-6.35g/fink/emacsen-remove	1970-01-01 01:00:00.000000000 +0100
+++ org-6.35g.fink/fink/emacsen-remove	2010-04-10 14:42:25.000000000 +0100
@@ -0,0 +1,31 @@
+#!/bin/bash -e
+
+set -o posix
+
+FLAVOR=${1}
+
+if [ ${FLAVOR} == emacs ]
+then
+  exit 0
+fi
+
+echo "remove/org-mode: Handling removal for emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "remove/org-mode: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+if [ ${FLAVOR} == xemacs ]
+then
+    echo "remove/org-mode: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+echo >&2 -n "remove/org-mode: Purging compilation log and byte-compiled files fo ${FLAVOR}..."
+rm -f @PREFIX@/share/doc/org-mode/CompilationLog-${FLAVOR}.gz
+rm -Rf @PREFIX@/share/${FLAVOR}/site-lisp/org
+echo >&2 "done."
+    
+exit 0
diff -Naur --exclude='*~' org-6.35g/fink/emacsen-remove-babel org-6.35g.fink/fink/emacsen-remove-babel
--- org-6.35g/fink/emacsen-remove-babel	1970-01-01 01:00:00.000000000 +0100
+++ org-6.35g.fink/fink/emacsen-remove-babel	2010-04-10 14:42:52.000000000 +0100
@@ -0,0 +1,31 @@
+#!/bin/bash -e
+
+set -o posix
+
+FLAVOR=${1}
+
+if [ ${FLAVOR} == emacs ]
+then
+  exit 0
+fi
+
+echo "remove/org-babel: Handling removal for emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "remove/org-babel: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+if [ ${FLAVOR} == xemacs ]
+then
+    echo "remove/org-babel: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+echo >&2 -n "remove/org-mode-babel: Purging compilation log and byte-compiled fies for ${FLAVOR}..."
+rm -f @PREFIX@/share/doc/org-babel/CompilationLog-${FLAVOR}.gz
+rm -Rf @PREFIX@/share/${FLAVOR}/site-lisp/org-babel
+echo >&2 "done."
+    
+exit 0
diff -Naur --exclude='*~' org-6.35g/fink/emacsen-remove-contrib org-6.35g.fink/fink/emacsen-remove-contrib
--- org-6.35g/fink/emacsen-remove-contrib	1970-01-01 01:00:00.000000000 +0100
+++ org-6.35g.fink/fink/emacsen-remove-contrib	2010-04-10 14:43:15.000000000 +0100
@@ -0,0 +1,31 @@
+#!/bin/bash -e
+
+set -o posix
+
+FLAVOR=${1}
+
+if [ ${FLAVOR} == emacs ]
+then
+  exit 0
+fi
+
+echo "remove/org-mode-contrib: Handling removal for emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "remove/org-mode-contrib: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+if [ ${FLAVOR} == xemacs ]
+then
+    echo "remove/org-mode-contrib: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+echo >&2 -n "remove/org-mode-contrib: Purging compilation log and byte-compiled iles for ${FLAVOR}..."
+rm -f @PREFIX@/share/doc/org-mode-contrib/CompilationLog-${FLAVOR}.gz
+rm -Rf @PREFIX@/share/${FLAVOR}/site-lisp/org/contrib
+echo >&2 "done."
+    
+exit 0
