diff -Naur semi-1.14.6.orig/fink/semi-install semi-1.14.6/fink/semi-install
--- semi-1.14.6.orig/fink/semi-install	Thu Jan  1 09:00:00 1970
+++ semi-1.14.6/fink/semi-install	Mon Mar 14 04:43:21 2005
@@ -0,0 +1,178 @@
+#!/bin/sh
+set -e
+
+FLAVOR=$1
+PACKAGE=semi
+VERSION=@VERSION@
+
+if [ "X${FLAVOR}" = "X" ]; then
+  echo Need argument to determin FLAVOR of emacsen
+  exit 1
+fi
+if [ "X${PACKAGE}" = "X" ]; then
+  echo Internal error: need package name
+  exit 1
+fi
+if [ "X${FLAVOR}" = Xemacs ]; then exit 0; fi
+
+if [ -e @PREFIX@/etc/${FLAVOR}/site-start.d/51semi-init.el ]; then
+  # remove 51semi-init.el which was left by the wemi package
+  rm -f @PREFIX@/etc/${FLAVOR}/site-start.d/51semi-init.el
+fi
+
+ELDIR=@PREFIX@/share/emacs/site-lisp/${PACKAGE}
+SITELISP=@PREFIX@/share/${FLAVOR}/site-lisp
+EFLAGS="-no-site-file -batch -q -l ./SEMI-MK.comp"
+ELCDIR_BASE=@PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}
+APEL_DIR_BASE=${SITELISP}/apel
+FLIM_DIR_BASE=${SITELISP}/flim
+BITMAP_MULE_DIR_BASE=${SITELISP}/bitmap-mule
+
+EMACSEN_MULE=${FLAVOR}
+EMACSEN_NOMULE=""
+
+FLAVOR_SUFFIX=`echo ${FLAVOR} | cut -s -d - -f 2`
+FLAVORTEST=`echo ${FLAVOR} | cut -c-6`
+if [ "X${FLAVORTEST}" = "Xxemacs" ]; then
+  FLAVOR_BODY=`echo ${FLAVOR} | cut -d - -f 1`
+  if [ "X${FLAVOR_SUFFIX}" = "Xmule" ]; then
+    EMACSEN_MULE=`ls @PREFIX@/bin/${FLAVOR_BODY}*-mule* 2>/dev/null | head -1`
+    if [ -n "${EMACSEN_MULE}" ]; then
+      EMACSEN_MULE=`basename ${EMACSEN_MULE}`
+    fi
+    EMACSEN_NOMULE=""
+  elif [ "X${FLAVOR_SUFFIX}" = "Xnomule" ]; then
+    EMACSEN_MULE=""
+    EMACSEN_NOMULE=`ls @PREFIX@/bin/${FLAVOR_BODY}*-nomule* 2>/dev/null | head -1`
+    if [ -n "${EMACSEN_NOMULE}" ]; then
+      EMACSEN_NOMULE=`basename ${EMACSEN_NOMULE}`
+    fi
+  else
+    FLAVOR_SUFFIX=""
+    EMACSEN_MULE=`ls @PREFIX@/bin/${FLAVOR}*-mule* 2>/dev/null | head -1`
+    if [ -n "${EMACSEN_MULE}" ]; then
+      EMACSEN_MULE=`basename ${EMACSEN_MULE}`
+    fi
+    EMACSEN_NOMULE=`ls @PREFIX@/bin/${FLAVOR}*-nomule* 2>/dev/null | head -1`
+    if [ -n "${EMACSEN_NOMULE}" ]; then
+      EMACSEN_NOMULE=`basename ${EMACSEN_NOMULE}`
+    fi
+  fi
+fi
+
+COMPILED=no
+for EMACSEN in "${EMACSEN_MULE}" "${EMACSEN_NOMULE}"; do
+  if [ -n "${EMACSEN}" ]; then
+    ELCDIR=${ELCDIR_BASE}
+    APEL_DIR=${APEL_DIR_BASE}
+    FLIM_DIR=${FLIM_DIR_BASE}
+    BITMAP_MULE_DIR=${BITMAP_MULE_DIR_BASE}
+    if [ "X${FLAVORTEST}" = "Xxemacs" -a -z "${FLAVOR_SUFFIX}" ]; then
+      if [ "X${EMACSEN}" = "X${EMACSEN_MULE}" ]; then
+	/bin/echo "install/${PACKAGE}: Byte-compiling for ${FLAVOR} (mule) ...\c"
+        ELCDIR=${ELCDIR}/mule
+	if [ -d ${APEL_DIR_BASE}/mule ]; then
+	  APEL_DIR=${APEL_DIR_BASE}/mule
+	fi
+	if [ -d ${FLIM_DIR_BASE}/mule ]; then
+	  FLIM_DIR=${FLIM_DIR_BASE}/mule
+	fi
+	if [ -d ${BITMAP_MULE_DIR_BASE}/mule ]; then
+	  BITMAP_MULE_DIR=${BITMAP_MULE_DIR_BASE}/mule
+	fi
+      else
+	/bin/echo "install/${PACKAGE}: Byte-compiling for ${FLAVOR} (nomule) ...\c"
+        ELCDIR=${ELCDIR}/nomule
+	if [ -d ${APEL_DIR_BASE}/nomule ]; then
+	  APEL_DIR=${APEL_DIR_BASE}/nomule
+	fi
+	if [ -d ${FLIM_DIR_BASE}/nomule ]; then
+	  FLIM_DIR=${FLIM_DIR_BASE}/nomule
+	fi
+	if [ -d ${BITMAP_MULE_DIR_BASE}/nomule ]; then
+	  BITMAP_MULE_DIR=${BITMAP_MULE_DIR_BASE}/nomule
+	fi
+      fi
+    else
+      /bin/echo "install/${PACKAGE}: Byte-compiling for ${FLAVOR} ...\c"
+    fi
+    STAMP="${ELCDIR}/compile-stamp"
+    if [ -e "${STAMP}" ]; then
+      if [ "X${VERSION}" = X"`cat ${STAMP}`" ]; then
+	echo " skipped. (already compiled)" 
+      else
+	rm -f ${STAMP}
+      fi
+    fi
+    if [ ! -e "${STAMP}" ]; then
+      if [ "X${FLAVORTEST}" = "Xxemacs" -a -z "${FLAVOR_SUFFIX}" ]; then
+	rm -f ${ELCDIR_BASE}/*.el ${ELCDIR_BASE}/*.elc
+      fi
+      rm -rf ${ELCDIR}
+      install -m 755 -d ${ELCDIR}/tmp
+      cp -r ${ELDIR}/* ${ELCDIR}/tmp/
+      (cd ${ELCDIR}/tmp
+	cat << EOF > SEMI-MK.comp
+(setq load-path (cons "${APEL_DIR}" load-path))
+(setq load-path (cons "${FLIM_DIR}" load-path))
+(setq load-path (cons "${BITMAP_MULE_DIR}" load-path))
+EOF
+	cat SEMI-MK >> SEMI-MK.comp
+	make FLAGS="${EFLAGS}" EMACS=${EMACSEN} LISPDIR=${ELCDIR} install \
+	  > ${ELCDIR}/CompilationLog 2>&1
+	cd ${ELCDIR}
+	echo "rm -rf ${ELCDIR}/semi/*.el ${ELCDIR}/tmp" >> ${ELCDIR}/CompilationLog 2>&1
+	rm -rf ${ELCDIR}/semi/*.el ${ELCDIR}/tmp
+	echo "mv ${ELCDIR}/semi/* ${ELCDIR}/" >> ${ELCDIR}/CompilationLog 2>&1
+	mv ${ELCDIR}/semi/* ${ELCDIR}/
+	echo "rm -rf ${ELCDIR}/semi" >> ${ELCDIR}/CompilationLog 2>&1
+	rm -rf ${ELCDIR}/semi
+	cd ${ELDIR}
+	for f in *.el; do
+	  if [ -f ${ELCDIR}/${f}c ]; then
+	    ln -sf ${ELDIR}/${f} ${ELCDIR}/${f}
+	  fi
+	done
+      )
+      gzip -9 ${ELCDIR}/CompilationLog
+      chmod 644 ${ELCDIR}/CompilationLog.gz
+      if [ -f ${ELCDIR}/mime-play.elc ]; then
+	touch ${ELCDIR}/*.elc
+	chmod 644 ${ELCDIR}/*.elc
+	echo "${VERSION}" > ${STAMP}
+	chmod 644 ${STAMP}
+	COMPILED=yes
+      fi
+      if [ "X${FLAVORTEST}" = "Xxemacs" -a -z "${FLAVOR_SUFFIX}" ]; then
+	if [ -e ${ELCDIR_BASE}/mule/compile-stamp ]; then
+	  ln -sf ${ELCDIR_BASE}/mule/*.elc ${ELCDIR_BASE}/mule/*.el ${ELCDIR_BASE}/
+	elif [ -e ${ELCDIR_BASE}/nomule/compile-stamp ]; then
+	  ln -sf ${ELCDIR_BASE}/nomule/*.elc ${ELCDIR_BASE}/nomule/*.el ${ELCDIR_BASE}/
+	fi
+      fi
+      echo " done."
+    fi
+  fi
+done
+
+if [ "X${COMPILED}" = Xyes ]; then
+  # recompile wl and wl-beta
+  if [ -d @PREFIX@/share/${FLAVOR}/site-lisp/wl ]; then
+    for pkg in wl wl-beta; do
+      if [ -f @PREFIX@/lib/emacsen-common/packages/install/${pkg} ]; then
+	@PREFIX@/lib/emacsen-common/packages/remove/${pkg} ${FLAVOR}
+	@PREFIX@/lib/emacsen-common/packages/install/${pkg} ${FLAVOR}
+      fi
+    done
+  fi
+  # recompile t-gnus
+  pkg=t-gnus
+  if [ -d @PREFIX@/share/${FLAVOR}/site-lisp/${pkg} ]; then
+    if [ -f @PREFIX@/lib/emacsen-common/packages/install/${pkg} ]; then
+      @PREFIX@/lib/emacsen-common/packages/remove/${pkg} ${FLAVOR}
+      @PREFIX@/lib/emacsen-common/packages/install/${pkg} ${FLAVOR}
+    fi
+  fi
+fi
+
+exit 0
diff -Naur semi-1.14.6.orig/fink/semi-remove semi-1.14.6/fink/semi-remove
--- semi-1.14.6.orig/fink/semi-remove	Thu Jan  1 09:00:00 1970
+++ semi-1.14.6/fink/semi-remove	Mon Mar 14 04:19:27 2005
@@ -0,0 +1,23 @@
+#!/bin/sh
+set -e
+
+FLAVOR=$1
+PACKAGE="semi"
+
+if [ "X${FLAVOR}" = "X" ]; then
+  echo Need argument to determin FLAVOR of emacsen
+  exit 1
+fi
+if [ "X${PACKAGE}" = "X" ]; then
+  echo Internal error: need package name
+  exit 1
+fi
+if [ "X${FLAVOR}" = Xemacs ]; then exit 0; fi
+
+ELCDIR=@PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}
+
+/bin/echo "remove/${PACKAGE}: Handling removal of emacsen flavor ${FLAVOR} ...\c"
+rm -rf ${ELCDIR}
+echo " done."
+
+exit 0
diff -Naur semi-1.14.6.orig/fink/semi-startup semi-1.14.6/fink/semi-startup
--- semi-1.14.6.orig/fink/semi-startup	Thu Jan  1 09:00:00 1970
+++ semi-1.14.6/fink/semi-startup	Thu Mar 24 16:18:35 2005
@@ -0,0 +1,39 @@
+;;; 51semi.el --- Fink semi startup file  -*-mode: emacs-lisp;-*-
+
+;;; Code:
+
+(let ((elc-dir (concat "@PREFIX@/share/" (symbol-name fink-emacs-flavor) "/site-lisp/semi")))
+  (if (not (file-directory-p elc-dir))
+      ()
+    (if (featurep 'xemacs)
+	(if (featurep 'mule)
+	    (and (file-directory-p (concat elc-dir "/mule"))
+		 (setq elc-dir (concat elc-dir "/mule")))
+	  (and (file-directory-p (concat elc-dir "/nomule"))
+	       (setq elc-dir (concat elc-dir "/nomule")))))
+    (if (and nil (fboundp 'fink-pkg-add-load-path-item)) ; fink-pkg-* is not
+	(fink-pkg-add-load-path-item elc-dir)            ; correctly working.
+      (setq load-path (cons elc-dir load-path)))
+    (setq mime-edit-split-message nil)
+    (setq pgg-encrypt-for-me t)
+    ;;(setq pgg-messages-locale "C") ;; default value is nil
+    ;;(setq pgg-gpg-messages-locale pgg-messages-locale)
+    (require 'mime-setup)
+    (if (module-installed-p 'mime-w3m)
+	(progn
+	  (setq mime-setup-enable-inline-html nil)
+	  (eval-after-load "mime-view"
+	    '(progn
+	       (autoload 'mime-w3m-preview-text/html "mime-w3m")
+	       (ctree-set-calist-strictly
+		'mime-preview-condition
+		'((type . text)
+		  (subtype . html)
+		  (body . visible)
+		  (body-presentation-method . mime-w3m-preview-text/html)))
+	       (set-alist 'mime-view-type-subtype-score-alist
+			  '(text . html) 3)))))
+    ;;
+    ))
+
+;;; 51semi.el ends here
diff -Naur semi-1.14.6.orig/mime-edit.el semi-1.14.6/mime-edit.el
--- semi-1.14.6.orig/mime-edit.el	Wed Dec 17 08:36:13 2003
+++ semi-1.14.6/mime-edit.el	Thu Aug 12 00:12:56 2004
@@ -236,6 +236,9 @@
     ("application"
      ("octet-stream" ("type" "" "tar" "shar"))
      ("postscript")
+     ("pdf")
+     ("msword")
+     ("vnd.ms-excel")
      ("vnd.ms-powerpoint")
      ("x-kiss" ("x-cnf")))
     ("image"
@@ -321,6 +324,11 @@
      "base64"
      "attachment" (("filename" . file))
      )
+    ("\\.xls$"				; MS Excel
+     "application" "vnd.ms-excel" nil
+     "base64"
+     "attachment" (("filename" . file))
+     )
     ("\\.ppt$"				; MS Power Point
      "application" "vnd.ms-powerpoint" nil
      "base64"
@@ -334,7 +342,12 @@
      )
     ("\\.ps$"
      "application" "postscript"	nil
-     "quoted-printable"
+     "base64"
+     "attachment"	(("filename" . file))
+     )
+    ("\\.pdf$"
+     "application" "pdf"	nil
+     "base64"
      "attachment"	(("filename" . file))
      )
 
@@ -725,7 +738,7 @@
   (concat "MIME-Version:"
 	  (mime-encode-field-body
 	   (concat " 1.0 (split by " mime-edit-version ")\n")
-	   "MIME-Version:"))
+	   "MIME-Version"))
   "MIME version field for message/partial.")
 
 
@@ -2302,9 +2315,10 @@
 		    (narrow-to-region beg (mime-edit-content-end))
 		    (goto-char beg)
 		    (while (re-search-forward "\\(\\=\\|[^\r]\\)\n" nil t)
-		      ;; Don't use this in the multibyte buffer since it may
-		      ;; convert the unibyte string into multibyte.
-		      ;;;;(replace-match "\\1\r\n"))))
+		      ;; In a certain period, `replace-match' with "\\N"
+		      ;; converted 8-bit characters into multibyte string,
+		      ;; but it has been fixed at 2004-01-15.
+		      ;;(replace-match "\\1\r\n"))))
 		      (backward-char 1)
 		      (insert "\r")
 		      (forward-char 1))))
diff -Naur semi-1.14.6.orig/mime-image.el semi-1.14.6/mime-image.el
--- semi-1.14.6.orig/mime-image.el	Mon Nov 11 13:39:54 2002
+++ semi-1.14.6/mime-image.el	Thu Aug 12 00:12:56 2004
@@ -7,7 +7,6 @@
 ;;	Dan Rich <drich@morpheus.corp.sgi.com>
 ;;	Daiki Ueno <ueno@ueda.info.waseda.ac.jp>
 ;;	Katsumi Yamaoka  <yamaoka@jpl.org>
-;; Maintainer: MORIOKA Tomohiko <morioka@jaist.ac.jp>
 ;; Created: 1995/12/15
 ;;	Renamed: 1997/2/21 from tm-image.el
 
@@ -50,12 +49,14 @@
     (let ((case-fold-search t) width height xbytes right margin)
       (goto-char (point-min))
       (or (re-search-forward "_width[\t ]+\\([0-9]+\\)" nil t)
-	  (error "!! Illegal xbm file format" (current-buffer)))
+	  (error "!! Illegal xbm file format in the buffer: %s"
+		 (current-buffer)))
       (setq width (string-to-int (match-string 1))
 	    xbytes (/ (+ width 7) 8))
       (goto-char (point-min))
       (or (re-search-forward "_height[\t ]+\\([0-9]+\\)" nil t)
-	  (error "!! Illegal xbm file format" (current-buffer)))
+	  (error "!! Illegal xbm file format in the buffer: %s"
+		 (current-buffer)))
       (setq height (string-to-int (match-string 1)))
       (goto-char (point-min))
       (re-search-forward "0x[0-9a-f][0-9a-f],")
@@ -188,15 +189,20 @@
 
 (defun mime-display-image (entity situation)
   (message "Decoding image...")
-  (let ((format (cdr (assq 'image-format situation)))
-	image)
-    (setq image (mime-image-create (mime-entity-content entity) format 'data))
-    (if (null image)
-	(message "Invalid glyph!")
-      (save-excursion
-	(mime-image-insert image)
-	(insert "\n")
-	(message "Decoding image...done")))))
+  (condition-case err
+      (let ((format (cdr (assq 'image-format situation)))
+	    image)
+	(setq image
+	      (mime-image-create (mime-entity-content entity)
+				 format 'data))
+	(if (null image)
+	    (message "Invalid glyph!")
+	  (save-excursion
+	    (mime-image-insert image)
+	    (insert "\n")
+	    (message "Decoding image...done"))))
+    (error nil err)))
+
 
 ;;; @ end
 ;;;
diff -Naur semi-1.14.6.orig/mime-view.el semi-1.14.6/mime-view.el
--- semi-1.14.6.orig/mime-view.el	Mon May 12 23:17:48 2003
+++ semi-1.14.6/mime-view.el	Thu Aug 12 00:12:56 2004
@@ -599,24 +599,22 @@
 		  )))
 	    )
 	   (t
-	    (let ((media-type (mime-entity-media-type entity))
-		  (media-subtype (mime-entity-media-subtype entity))
-		  (charset (cdr (assoc "charset" params)))
-		  (encoding (mime-entity-encoding entity)))
+	    (let* ((charset (cdr (assoc "charset" params)))
+		   (encoding (mime-entity-encoding entity))
+		   (rest (format " <%s/%s%s%s>"
+				 (mime-entity-media-type entity)
+				 (mime-entity-media-subtype entity)
+				 (if charset
+				     (concat "; " charset)
+				   "")
+				 (if encoding
+				     (concat " (" encoding ")")
+				   ""))))
 	      (concat
 	       num " " subject
-	       (let ((rest
-		      (format " <%s/%s%s%s>"
-			      media-type media-subtype
-			      (if charset
-				  (concat "; " charset)
-				"")
-			      (if encoding
-				  (concat " (" encoding ")")
-				""))))
-		 (if (>= (+ (current-column)(length rest))(window-width))
-		     "\n\t")
-		 rest)))
+	       (if (>= (+ (current-column)(length rest))(window-width))
+		   "\n\t")
+	       rest))
 	    )))
      (function mime-preview-play-current-entity))
     ))
@@ -941,10 +939,11 @@
 		    (car (mime-entity-children entity))))
 	 (original-major-mode-cell (assq 'major-mode situation))
 	 (default-situation (cdr (assq 'childrens-situation situation))))
-    (if original-major-mode-cell
-	(setq default-situation
-	      (cons original-major-mode-cell default-situation)))
-    (mime-display-entity start nil default-situation)))
+    (when start
+      (if original-major-mode-cell
+	  (setq default-situation
+		(cons original-major-mode-cell default-situation)))
+      (mime-display-entity start nil default-situation))))
 
 ;;; @ acting-condition
 ;;;
diff -Naur semi-1.14.6.orig/mime-ui-en.texi semi-1.14.6/mime-ui-en.texi
--- semi-1.14.6.orig/mime-ui-en.texi	2002-10-15 15:59:21.000000000 +0900
+++ semi-1.14.6/mime-ui-en.texi	2007-05-08 10:40:25.000000000 +0900
@@ -1,5 +1,9 @@
 \input texinfo.tex
 @setfilename mime-ui-en.info
+@dircategory Emacs
+@direntry
+* SEMI-en: (mime-ui-en).   SEMI, a MIME user interface for GNU Emacs.
+@end direntry
 @settitle{SEMI 1.14 Manual}
 @titlepage
 @title SEMI 1.14 Manual
diff -Naur semi-1.14.6.orig/mime-ui-ja.texi semi-1.14.6/mime-ui-ja.texi
--- semi-1.14.6.orig/mime-ui-ja.texi	2002-10-15 15:59:21.000000000 +0900
+++ semi-1.14.6/mime-ui-ja.texi	2007-05-08 10:41:03.000000000 +0900
@@ -1,5 +1,9 @@
 \input texinfo.tex
 @setfilename mime-ui-ja.info
+@dircategory Emacs
+@direntry
+* SEMI-ja: (mime-ui-ja).   SEMI, a MIME user interface for GNU Emacs.
+@end direntry
 @settitle{SEMI 1.14 説明書}
 @titlepage
 @title SEMI 1.14 説明書
