diff -Nurd -x'*~' tea-17.6.1.orig/configure tea-17.6.1/configure
--- tea-17.6.1.orig/configure	2008-06-03 06:01:16.000000000 -0400
+++ tea-17.6.1/configure	2008-06-03 23:09:55.000000000 -0400
@@ -5813,6 +5813,7 @@
 		echo "yes"
 	else
 		echo "no"
+		exit 1
 	fi
 
 else
@@ -6257,7 +6258,7 @@
 	echo -n "checking for gtk+-unix-print-2.0... "
 	if pkg-config --exists gtk+-unix-print-2.0  ; then
 		LIBS="$LIBS `pkg-config --libs gtk+-unix-print-2.0 `"
-		CFLAGS="$CFLAGS `pkg-config --cflags gtk+-unix-print-2.0 `"
+		CPPFLAGS="$CPPFLAGS `pkg-config --cflags gtk+-unix-print-2.0 `"
 		echo "yes"
 
 cat >>confdefs.h <<\_ACEOF
@@ -6298,7 +6299,8 @@
 	echo "yes"
 
 else
-	echo -n "GtkSourceView2 not found, now we search GtkSourceView1"
+	echo -n "Fink package enforces GtkSourceView2 usage, not falling back to  GtkSourceView1"
+	exit 1
 
 	if pkg-config --exists gtksourceview-1.0 ; then
 		LIBS="$LIBS `pkg-config --libs gtksourceview-1.0`"
diff -Nurd -x'*~' tea-17.6.1.orig/src/Makefile.in tea-17.6.1/src/Makefile.in
--- tea-17.6.1.orig/src/Makefile.in	2008-06-03 14:33:00.000000000 -0400
+++ tea-17.6.1/src/Makefile.in	2010-01-15 11:11:19.000000000 -0500
@@ -264,7 +264,7 @@
 @DEB_TRUE@	expr.c expr.h \
 @DEB_TRUE@	grabber.c grabber.h 
 
-@DEB_TRUE@teaed_LDADD = @PACKAGE_LIBS@ $(INTLLIBS)
+@DEB_TRUE@teaed_LDADD = @PACKAGE_LIBS@ $(INTLLIBS) -L/usr/X11R6/lib -lX11
 @DEB_FALSE@tea_SOURCES = \
 @DEB_FALSE@	main.c \
 @DEB_FALSE@	interface.c interface.h \
@@ -294,7 +294,7 @@
 @DEB_FALSE@	expr.c expr.h \
 @DEB_FALSE@	grabber.c grabber.h 
 
-@DEB_FALSE@tea_LDADD = @PACKAGE_LIBS@ $(INTLLIBS)
+@DEB_FALSE@tea_LDADD = @PACKAGE_LIBS@ $(INTLLIBS) -L/usr/X11R6/lib -lX11
 all: all-am
 
 .SUFFIXES:
diff -Nurd -x'*~' tea-17.6.1.orig/src/rox_strings.c tea-17.6.1/src/rox_strings.c
--- tea-17.6.1.orig/src/rox_strings.c	2008-05-07 05:14:23.000000000 -0400
+++ tea-17.6.1/src/rox_strings.c	2008-06-03 20:36:43.000000000 -0400
@@ -34,6 +34,7 @@
 
 #include <ctype.h>
 #include <sys/stat.h>
+#include <sys/types.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
