diff -Naur vim7.1.orig/src/auto/configure vim7.1.not-fink/src/auto/configure
--- vim7.1.orig/src/auto/configure	2007-07-11 14:02:21.000000000 -0700
+++ vim7.1.not-fink/src/auto/configure	2007-08-21 19:24:00.000000000 -0700
@@ -3427,7 +3427,7 @@
 if test -z "$have_local_include" -a -d /usr/local/include; then
   tt=`echo "$CPPFLAGS" | sed -e 's+-I/usr/local/include ++g' -e 's+-I/usr/local/include$++g'`
   if test "$tt" = "$CPPFLAGS"; then
-    CPPFLAGS="$CPPFLAGS -I/usr/local/include"
+    CPPFLAGS="$CPPFLAGS"
   fi
 fi
 
diff -Naur vim7.1.orig/src/feature.h vim7.1.not-fink/src/feature.h
--- vim7.1.orig/src/feature.h	2007-08-05 11:35:48.000000000 -0700
+++ vim7.1.not-fink/src/feature.h	2007-08-21 19:49:53.000000000 -0700
@@ -1202,6 +1202,7 @@
  *			debugger and for tooltips.
  *			Only for GUIs where it was implemented.
  */
+#if 0
 #if (defined(FEAT_GUI_MOTIF) || defined(FEAT_GUI_ATHENA) \
 	|| defined(FEAT_GUI_GTK) || defined(FEAT_GUI_W32)) \
 	&& (   ((defined(FEAT_TOOLBAR) || defined(FEAT_GUI_TABLINE)) \
@@ -1214,6 +1215,7 @@
 #  define FEAT_XFONTSET
 # endif
 #endif
+#endif
 
 #if defined(FEAT_BEVAL) && (defined(FEAT_GUI_MOTIF) || defined(FEAT_GUI_ATHENA))
 # define FEAT_BEVAL_TIP		/* balloon eval used for toolbar tooltip */
diff -Naur vim7.1.orig/src/vim.h vim7.1.not-fink/src/vim.h
--- vim7.1.orig/src/vim.h	2007-08-11 09:44:57.000000000 -0700
+++ vim7.1.not-fink/src/vim.h	2007-08-21 20:02:24.000000000 -0700
@@ -201,9 +201,11 @@
 #endif
 
 /* The Mac conversion stuff doesn't work under X11. */
+#if 0
 #if defined(FEAT_MBYTE) && defined(MACOS_X)
 # define MACOS_CONVERT
 #endif
+#endif
 
 /* Can't use "PACKAGE" here, conflicts with a Perl include file. */
 #ifndef VIMPACKAGE
