diff -urN lynx2-8-5/WWW/Library/Implementation/HTTCP.c lynx2-8-5.patched/WWW/Library/Implementation/HTTCP.c
--- lynx2-8-5/WWW/Library/Implementation/HTTCP.c	2004-01-24 17:51:04.000000000 -0800
+++ lynx2-8-5.patched/WWW/Library/Implementation/HTTCP.c	2006-06-07 22:30:28.000000000 -0700
@@ -1326,7 +1326,11 @@
 	    if (phost->h_length != sizeof soc_in->sin_addr) {
 		HTAlwaysAlert(host, gettext("Address length looks invalid"));
 	    }
+#ifdef _POSIX_C_SOURCE
+	    memcpy((void *)&soc_in->sin_addr, phost->h_addr_list[0], phost->h_length);
+#else
 	    memcpy((void *)&soc_in->sin_addr, phost->h_addr, phost->h_length);
+#endif /* _POSIX_C_SOURCE */
 #endif /* VMS && CMU_TCP */
 	}
 #endif /* _WINDOWS_NSL */
diff -urN lynx2-8-5/WWW/Library/Implementation/makefile.in lynx2-8-5.patched/WWW/Library/Implementation/makefile.in
--- lynx2-8-5/WWW/Library/Implementation/makefile.in	2004-01-24 20:51:04.000000000 -0500
+++ lynx2-8-5.patched/WWW/Library/Implementation/makefile.in	2008-06-08 18:56:59.000000000 -0400
@@ -25,12 +25,12 @@
 INTLDIR_CPPFLAGS= @INTLDIR_CPPFLAGS@ -I$(top_srcdir)/intl
 
 CPP		= @CPP@
-CPPOPTS		= $(DEFS) $(CPPFLAGS) $(LYFLAGS) \
+CPPOPTS		= $(DEFS) $(LYFLAGS) \
 		-I../../.. \
 		-I../../../src \
 		-I$(top_srcdir) \
 		-I$(top_srcdir)/src \
-		$(INTLDIR_CPPFLAGS) -I$(WWWINC)
+		-I$(WWWINC) $(INTLDIR_CPPFLAGS) $(CPPFLAGS)
 
 LY_CFLAGS	= @CFLAGS@
 CFLAGS		= $(CPPOPTS) $(LY_CFLAGS)
diff -urN lynx2-8-5/configure lynx2-8-5.patched/configure
--- lynx2-8-5/configure	2004-02-04 07:07:09.000000000 -0500
+++ lynx2-8-5.patched/configure	2008-06-08 18:52:38.000000000 -0400
@@ -5860,11 +5860,11 @@
 *)
   if test -d $cf_cv_use_libssl ; then
     if test -d $cf_cv_use_libssl/include ; then
-      CPPFLAGS="$CPPFLAGS -I$cf_cv_use_libssl/include"
+      CPPFLAGS="-I$cf_cv_use_libssl/include $CPPFLAGS"
       cf_ssl_library="-L$cf_cv_use_libssl/lib $cf_ssl_library"
     else
       cf_ssl_library="-L$cf_cv_use_libssl $cf_ssl_library"
-      test -d $cf_cv_use_libssl/../include && CPPFLAGS="$CPPFLAGS -I$cf_cv_use_libssl/../include"
+      test -d $cf_cv_use_libssl/../include && CPPFLAGS="-I$cf_cv_use_libssl/../include $CPPFLAGS"
     fi
   else
     echo "configure: warning: expected a directory: $cf_cv_use_libssl" 1>&2
@@ -9595,7 +9595,7 @@
 	;;
 esac
 
-eval cf_libdir=$libdir
+eval cf_libdir=$sysconfdir
 cf_libdir=`echo $cf_libdir | sed -e s%'^NONE/'%$prefix/% -e s%'^NONE/'%$ac_default_prefix/%`
 
 echo $ac_n "checking for location of config-file""... $ac_c" 1>&6
diff -urN lynx2-8-5/makefile.in lynx2-8-5.patched/makefile.in
--- lynx2-8-5/makefile.in	2004-02-04 04:07:09.000000000 -0800
+++ lynx2-8-5.patched/makefile.in	2006-06-07 22:30:28.000000000 -0700
@@ -305,7 +305,7 @@
 	fi'
 	$(SHELL) -c 'for name in COPYING COPYHEADER; do \
 		cd $(HELPDIR) && rm -f $$name ;\
-		cd $(HELPDIR) && ( $(LN_S) $(DOCDIR)/$$name . || cp $(DOCDIR)/$$name . );\
+		cd $(HELPDIR) && cp $(DOCDIR)/$$name . ;\
 		done'
 	-$(SHELL) -c 'case `id|sed -e "s/(.*//"` in uid=0) chown -R root $(DOCDIR);; esac'
 
diff -urN lynx2-8-5/src/LYMain.c lynx2-8-5.patched/src/LYMain.c
--- lynx2-8-5/src/LYMain.c	2004-01-28 14:30:38.000000000 -0500
+++ lynx2-8-5.patched/src/LYMain.c	2008-06-08 18:32:29.000000000 -0400
@@ -48,7 +48,6 @@
 #endif
 
 #ifdef LOCALE
-#undef gettext		/* Solaris locale.h prototypes gettext() */
 #include <locale.h>
 #ifndef HAVE_GETTEXT
 #define gettext(s) s
diff -urN lynx2-8-5/src/chrtrans/makefile.in lynx2-8-5.patched/src/chrtrans/makefile.in
--- lynx2-8-5/src/chrtrans/makefile.in	2004-01-07 21:03:09.000000000 -0500
+++ lynx2-8-5.patched/src/chrtrans/makefile.in	2008-06-08 22:19:13.000000000 -0400
@@ -31,13 +31,13 @@
 x		= @EXEEXT@
 o		= .@OBJEXT@
 
-CPP_OPTS	= @DEFS@ @CPPFLAGS@ \
+CPP_OPTS	= @DEFS@ \
 		-I$(top_builddir) \
 		-I$(top_srcdir)/src \
 		-I$(top_srcdir)/src/chrtrans \
 		-I$(top_srcdir)/$(WWWINC) \
 		-I$(top_srcdir)/ \
-		$(INTLDIR_CPPFLAGS) $(SITE_DEFS)
+		$(INTLDIR_CPPFLAGS) $(SITE_DEFS) @CPPFLAGS@
 CC_OPTS		= $(CPP_OPTS) $(CFLAGS)
 
 #
diff -urN lynx2-8-5/src/makefile.in lynx2-8-5.patched/src/makefile.in
--- lynx2-8-5/src/makefile.in	2004-01-07 21:03:09.000000000 -0500
+++ lynx2-8-5.patched/src/makefile.in	2008-06-08 18:17:34.000000000 -0400
@@ -41,7 +41,7 @@
 INTLLIB		= @INTLLIBS@
 INTLDIR_CPPFLAGS= @INTLDIR_CPPFLAGS@-I$(top_srcdir)/intl
 
-CPP_OPTS	= $(CPPFLAGS) $(DEFS) $(CHARSET_DEFS) \
+CPP_OPTS	= $(DEFS) $(CHARSET_DEFS) \
 		-DLOCALEDIR=\"$(localedir)\" \
 		-I. \
 		-I$(top_builddir) \
@@ -50,7 +50,7 @@
 		-I$(top_srcdir) \
 		-I$(top_srcdir)/src \
 		-I$(top_srcdir)/$(WWWINC) \
-		$(INTLDIR_CPPFLAGS) $(SITE_DEFS)
+		$(INTLDIR_CPPFLAGS) $(SITE_DEFS) $(CPPFLAGS)
 CC_OPTS		= $(CPP_OPTS) $(CFLAGS)
 
 LINT		= @LINT@
