diff -Nurd w3m-0.5.3+git20230121.orig/Makefile.in w3m-0.5.3+git20230121/Makefile.in
--- w3m-0.5.3+git20230121.orig/Makefile.in	2023-01-20 21:01:52.000000000 -0500
+++ w3m-0.5.3+git20230121/Makefile.in	2023-11-13 20:34:51.000000000 -0500
@@ -34,6 +34,7 @@
 RC_DIR = @RC_DIR@
 ETC_DIR = $(sysconfdir)
 CONF_DIR = $(sysconfdir)/$(PACKAGE)
+DOCDIRS = @DOCDIRS@
 
 WARNINGS= -Wall -Wnull-dereference
 CFLAGS = $(WARNINGS) -I. -I$(top_srcdir) @CFLAGS@ $(CPPFLAGS) $(DEFS) $(OPTS)
@@ -47,10 +48,11 @@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
 GC_LIBS = @LIBGC@
-EXT_LIBS = -L. -lindep @LIBINTL@ $(GC_LIBS)
+EXT_LIBS = libindep.a @LIBINTL@ $(GC_LIBS)
 W3M_LIBS = @W3M_LIBS@
 WCTARGET = @WCTARGET@
 NLSTARGET = @NLSTARGET@
+HELP_TARGETS	= w3mhelp-funcname.pl w3mhelp-funcdesc-stamp
 
 MAKE_ARGS = PERL='$(PERL)' MKDIR='$(MKDIR)' \
 	BIN_DIR='$(bindir)' AUXBIN_DIR='$(AUXBIN_DIR)' \
@@ -126,7 +128,7 @@
 SUBDIRS = $(SCRIPTSUBDIRS) w3mimg libwc po
 .PHONY: $(SUBDIRS)
 
-all: $(TARGETS) all-scripts $(NLSTARGET)
+all: funcname.tab $(TARGETS) all-scripts $(NLSTARGET) $(HELP_TARGETS)
 
 $(TARGET): $(ALLOBJS) $(ALIB) $(WCTARGET)
 	$(CC) $(CFLAGS) -o $(TARGET) $(ALLOBJS) $(LDFLAGS) $(LIBS) $(EXT_LIBS) $(W3M_LIBS)
@@ -178,6 +180,31 @@
 	./mktable$(EXT) 100 functable.tab > $@
 	-rm -f functable.tab
 
+w3mhelp-funcname.pl: scripts/w3mhelp-funcname.pl.in funcname.tab $(top_srcdir)/doc/keymap.default $(top_srcdir)/doc/keymap.lynx
+	@echo "generating w3mhelp-funcname.pl..."
+	@echo '%funcname = (' > scripts/w3mhelp-funcname.pl
+	@sed -ne "/^[a-zA-Z@]/s/\([a-zA-Z@][a-zA-Z0-9@_]*\)[ 	][ 	]*\([^ 	]*\)/'\1', '\2',/p" < $(top_srcdir)/funcname.tab >> scripts/w3mhelp-funcname.pl
+	@echo ');' >> scripts/w3mhelp-funcname.pl
+	@echo '%keyfunc = (' >> scripts/w3mhelp-funcname.pl
+	@case "$(KEYBIND_SRC)" in *lynx*) keymap=keymap.lynx;; *) keymap=keymap.default;; esac; \
+	sed -ne "/^keymap/s/^keymap[ 	][ 	]*\([^ 	]*\)[ 	][ 	]*\([a-zA-Z@][a-zA-Z0-9@_]*\)/'\1', '\2', /p" < $(top_srcdir)/doc/$$keymap >> scripts/w3mhelp-funcname.pl
+	@echo ');' >> scripts/w3mhelp-funcname.pl
+	@cat scripts/w3mhelp-funcname.pl.in >> scripts/w3mhelp-funcname.pl
+	@echo "done"
+
+w3mhelp-funcdesc-stamp: $(top_srcdir)/doc/README.func $(top_srcdir)/doc-jp/README.func scripts/w3mhelp-funcdesc.en.pl.in scripts/w3mhelp-funcdesc.ja.pl.in
+	@echo "generating w3mhelp-funcdesc*.pl..."
+	@for dirlang in $(DOCDIRS); do \
+	  dir=`expr "$$dirlang" : "\(.*\):.*"`; \
+	  lang=`expr "$$dirlang" : ".*:\(.*\)_.*"`; \
+	  echo '%funcdesc = (' > scripts/w3mhelp-funcdesc.$$lang.pl; \
+	  sed -ne "/^[a-zA-Z@]/s/\([a-zA-Z@][a-zA-Z0-9@_]*\)[ 	][ 	]*\(.*\)/'\1', '\2',/p" < $(top_srcdir)/$$dir/README.func >> scripts/w3mhelp-funcdesc.$$lang.pl; \
+	  echo ');' >> scripts/w3mhelp-funcdesc.$$lang.pl; \
+	  cat scripts/w3mhelp-funcdesc.$$lang.pl.in >> scripts/w3mhelp-funcdesc.$$lang.pl; \
+	done
+	@echo done
+	@touch scripts/w3mhelp-funcdesc-stamp
+
 mktable$(EXT): mktable.o dummy.o Str.o hash.o myctype.o
 	$(CC) $(CFLAGS) -o mktable mktable.o dummy.o Str.o hash.o myctype.o $(LDFLAGS) $(LIBS) $(GC_LIBS)
 
@@ -215,7 +242,7 @@
 
 install: install-core install-scripts install-po
 
-install-core: $(TARGETS)
+install-core: $(TARGETS) $(HELP_TARGETS)
 	-$(MKDIR) $(DESTDIR)$(bindir)
 	-$(MKDIR) $(DESTDIR)$(AUXBIN_DIR)
 	-$(MKDIR) $(DESTDIR)$(CGIBIN_DIR)
@@ -240,6 +267,10 @@
 	do \
 		$(INSTALL_PROGRAM) $$file $(DESTDIR)$(CGIBIN_DIR)/$$file; \
 	done
+	for file in scripts/w3mhelp-*.pl; \
+	do \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(HELP_DIR); \
+	done
 
 install-helpfile:
 	-$(MKDIR) $(DESTDIR)$(HELP_DIR)
@@ -281,6 +312,10 @@
 	do \
 		$(RM) -f $(HELP_DIR)/$$file; \
 	done
+	-for file in scripts/w3mhelp-*.pl; \
+	do \
+		rm -f $(HELP_DIR)/$$file; \
+	done
 	-$(RM) -f $(HELP_DIR)/$(HELP_TARGET)
 	-$(RM) -f $(mandir)/man1/$(MAN1_TARGET)
 	-$(RM) -f $(mandir)/ja/man1/$(MAN1_TARGET)
@@ -294,6 +329,7 @@
 	-$(RM) -f *.o *.a $(TARGETS) mktable$(EXT)
 	-$(RM) -f funcname.tab
 	-$(RM) -f entity.h funcname.c funcname1.h funcname2.h tagtable.c functable.c
+	-$(RM) -f $(HELP_TARGETS) w3mhelp-*.pl
 	-for dir in w3mimg libwc; \
 	do \
 		(cd $$dir && $(MAKE) clean RM="$(RM)"); \
diff -Nurd w3m-0.5.3+git20230121.orig/configure w3m-0.5.3+git20230121/configure
--- w3m-0.5.3+git20230121.orig/configure	2023-01-20 21:01:52.000000000 -0500
+++ w3m-0.5.3+git20230121/configure	2023-11-13 20:33:14.000000000 -0500
@@ -5062,7 +5062,7 @@
  WCTARGET="libwc/libwc.a"
  WCCFLAGS='-I$(srcdir) -I$(srcdir)/..'
  wcinclude='-I$(srcdir)/libwc'
- wclib="-L./libwc -lwc"
+ wclib="libwc/libwc.a"
  { $as_echo "$as_me:${as_lineno-$LINENO}: checking if Unicode support is enabled" >&5
 $as_echo_n "checking if Unicode support is enabled... " >&6; }
  # Check whether --enable-unicode was given.
@@ -7689,6 +7689,8 @@
      { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: unable to build w3mimgdisplay with X11 support" >&5
 $as_echo "$as_me: WARNING: unable to build w3mimgdisplay with X11 support" >&2;}
    fi
+   IMGX11CFLAGS="${IMGX11CFLAGS} `pkg-config x11 --cflags`"
+   IMGX11LDFLAGS="${IMGX11LDFLAGS} `pkg-config x11 --libs`"
   fi
   if test x"$fb" = xyes; then
    if test x"$have_imlib2" = xyes; then
diff -Nurd w3m-0.5.3+git20230121.orig/scripts/Makefile.in w3m-0.5.3+git20230121/scripts/Makefile.in
--- w3m-0.5.3+git20230121.orig/scripts/Makefile.in	2023-01-20 21:01:52.000000000 -0500
+++ w3m-0.5.3+git20230121/scripts/Makefile.in	2023-11-13 20:36:20.000000000 -0500
@@ -31,11 +31,9 @@
 RC_DIR = @RC_DIR@
 ETC_DIR = $(sysconfdir)
 CONF_DIR = $(sysconfdir)/$(PACKAGE)
-DOCDIRS = @DOCDIRS@
 
 AUXBIN_TARGETS	= xface2xpm
 LIB_TARGETS	= dirlist.cgi w3mhelp.cgi w3mmail.cgi
-HELP_TARGETS	= w3mhelp-funcname.pl w3mhelp-funcdesc-stamp
 SUBDIRS = multipart w3mman
 .PHONY: $(SUBDIRS)
 
@@ -46,32 +44,7 @@
 
 PERL		= @PERL@
 
-all: $(LIB_TARGETS) $(AUXBIN_TARGETS) $(HELP_TARGETS) $(SUBDIRS)
-
-w3mhelp-funcname.pl: w3mhelp-funcname.pl.in ../funcname.tab $(top_srcdir)/doc/keymap.default $(top_srcdir)/doc/keymap.lynx
-	@echo "generating w3mhelp-funcname.pl..."
-	@echo '%funcname = (' > w3mhelp-funcname.pl
-	@sed -ne "/^[a-zA-Z@]/s/\([a-zA-Z@][a-zA-Z0-9@_]*\)[ 	][ 	]*\([^ 	]*\)/'\1', '\2',/p" < ../funcname.tab >> w3mhelp-funcname.pl
-	@echo ');' >> w3mhelp-funcname.pl
-	@echo '%keyfunc = (' >> w3mhelp-funcname.pl
-	@case "$(KEYBIND_SRC)" in *lynx*) keymap=keymap.lynx;; *) keymap=keymap.default;; esac; \
-	sed -ne "/^keymap/s/^keymap[ 	][ 	]*\([^ 	]*\)[ 	][ 	]*\([a-zA-Z@][a-zA-Z0-9@_]*\)/'\1', '\2', /p" < $(top_srcdir)/doc/$$keymap >> w3mhelp-funcname.pl
-	@echo ');' >> w3mhelp-funcname.pl
-	@cat $(srcdir)/w3mhelp-funcname.pl.in >> w3mhelp-funcname.pl
-	@echo "done"
-
-w3mhelp-funcdesc-stamp: $(top_srcdir)/doc/README.func $(top_srcdir)/doc-jp/README.func $(top_srcdir)/doc-de/README.func w3mhelp-funcdesc.en.pl.in w3mhelp-funcdesc.ja.pl.in w3mhelp-funcdesc.de.pl.in
-	@echo "generating w3mhelp-funcdesc*.pl..."
-	@for dirlang in $(DOCDIRS); do \
-	  dir=`expr "$$dirlang" : "\(.*\):.*"`; \
-	  lang=`expr "$$dirlang" : ".*:\(.*\)_.*"`; \
-	  echo '%funcdesc = (' > w3mhelp-funcdesc.$$lang.pl; \
-	  sed -ne "/^[a-zA-Z@]/s/\([a-zA-Z@][a-zA-Z0-9@_]*\)[ 	][ 	]*\(.*\)/'\1', '\2',/p" < $(top_srcdir)/$$dir/README.func >> w3mhelp-funcdesc.$$lang.pl; \
-	  echo ');' >> w3mhelp-funcdesc.$$lang.pl; \
-	  cat $(srcdir)/w3mhelp-funcdesc.$$lang.pl.in >> w3mhelp-funcdesc.$$lang.pl; \
-	done
-	@echo done
-	@touch w3mhelp-funcdesc-stamp
+all: $(LIB_TARGETS) $(AUXBIN_TARGETS) $(SUBDIRS)
 
 $(SUBDIRS):
 	for subdir in $(SUBDIRS); \
@@ -82,7 +55,6 @@
 install: $(LIB_TARGETS) $(HELP_TARGETS)
 	-$(MKDIR) $(DESTDIR)$(AUXBIN_DIR)
 	-$(MKDIR) $(DESTDIR)$(CGIBIN_DIR)
-	-$(MKDIR) $(DESTDIR)$(HELP_DIR)
 	for file in $(AUXBIN_TARGETS);     \
 	do      \
 		$(INSTALL_SCRIPT) $$file $(DESTDIR)$(AUXBIN_DIR);  \
@@ -91,10 +63,6 @@
 	do      \
 		$(INSTALL_SCRIPT) $$file $(DESTDIR)$(CGIBIN_DIR);  \
 	done
-	for file in w3mhelp-*.pl; \
-	do \
-		$(INSTALL_DATA) $$file $(DESTDIR)$(HELP_DIR); \
-	done
 	for subdir in $(SUBDIRS); \
 	do \
 		(cd $$subdir && $(MAKE) install); \
@@ -109,17 +77,12 @@
 	do      \
 		rm -f $(LIB_DIR)/$$file; \
 	done
-	-for file in w3mhelp-*.pl; \
-	do \
-		rm -f $(HELP_DIR)/$$file; \
-	done
 	-for subdir in $(SUBDIRS); \
 	do \
 		(cd $$subdir && $(MAKE) uninstall); \
 	done
 
 clean:
-	-rm -f $(HELP_TARGETS) w3mhelp-*.pl
 	-for subdir in $(SUBDIRS); \
 	do \
 	  (cd $$subdir && $(MAKE) clean); \
diff -Nurd w3m-0.5.3+git20230121.orig/url.c w3m-0.5.3+git20230121/url.c
--- w3m-0.5.3+git20230121.orig/url.c	2023-01-20 21:01:52.000000000 -0500
+++ w3m-0.5.3+git20230121/url.c	2023-11-13 20:29:59.000000000 -0500
@@ -83,11 +83,11 @@
     {"ftp", SCM_FTP},
     {"local", SCM_LOCAL},
     {"file", SCM_LOCAL},
-    /*  {"exec", SCM_EXEC}, */
+    {"exec", SCM_EXEC}, 
     {"nntp", SCM_NNTP},
-    /*  {"nntp", SCM_NNTP_GROUP}, */
+    {"nntp", SCM_NNTP_GROUP}, 
     {"news", SCM_NEWS},
-    /*  {"news", SCM_NEWS_GROUP}, */
+    {"news", SCM_NEWS_GROUP}, 
     {"data", SCM_DATA},
 #ifndef USE_W3MMAILER
     {"mailto", SCM_MAILTO},
