diff -Nurd -x'*~' w3m-0.5.3.orig/Makefile.in w3m-0.5.3/Makefile.in
--- w3m-0.5.3.orig/Makefile.in	2011-01-04 04:22:18.000000000 -0500
+++ w3m-0.5.3/Makefile.in	2015-05-16 23:57:21.000000000 -0400
@@ -34,6 +34,7 @@
 RC_DIR = @RC_DIR@
 ETC_DIR = $(sysconfdir)
 CONF_DIR = $(sysconfdir)/$(PACKAGE)
+DOCDIRS = @DOCDIRS@
 
 CFLAGS = $(OPTS) -I. -I$(top_srcdir) @CFLAGS@ $(CPPFLAGS) $(DEFS)
 WCCFLAGS = @WCCFLAGS@
@@ -46,10 +47,11 @@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
 GC_LIBS = @LIBGC@
-EXT_LIBS = -L. -lindep @LIBINTL@ $(GC_LIBS)
+EXT_LIBS = libindep.a @LIBINTL@ $(GC_LIBS)
 W3M_LIBS = @W3M_LIBS@
 WCTARGET = @WCTARGET@
 NLSTARGET = @NLSTARGET@
+HELP_TARGETS	= w3mhelp-funcname.pl w3mhelp-funcdesc-stamp
 
 MAKE_ARGS = PERL='$(PERL)' MKDIR='$(MKDIR)' \
 	BIN_DIR='$(bindir)' AUXBIN_DIR='$(AUXBIN_DIR)' \
@@ -124,7 +126,7 @@
 SUBDIRS = $(SCRIPTSUBDIRS) w3mimg libwc po
 .PHONY: $(SUBDIRS)
 
-all: $(TARGETS) all-scripts $(NLSTARGET)
+all: funcname.tab $(TARGETS) all-scripts $(NLSTARGET) $(HELP_TARGETS)
 
 $(TARGET): $(ALLOBJS) $(ALIB) $(WCTARGET)
 	$(CC) $(CFLAGS) -o $(TARGET) $(ALLOBJS) $(LDFLAGS) $(LIBS) $(EXT_LIBS) $(W3M_LIBS)
@@ -176,6 +178,31 @@
 	./mktable$(EXT) 100 functable.tab > $@
 	-rm -f functable.tab
 
+w3mhelp-funcname.pl: scripts/w3mhelp-funcname.pl.in funcname.tab $(top_srcdir)/doc/keymap.default $(top_srcdir)/doc/keymap.lynx
+	@echo "generating w3mhelp-funcname.pl..."
+	@echo '%funcname = (' > scripts/w3mhelp-funcname.pl
+	@sed -ne "/^[a-zA-Z@]/s/\([a-zA-Z@][a-zA-Z0-9@_]*\)[ 	][ 	]*\([^ 	]*\)/'\1', '\2',/p" < $(top_srcdir)/funcname.tab >> scripts/w3mhelp-funcname.pl
+	@echo ');' >> scripts/w3mhelp-funcname.pl
+	@echo '%keyfunc = (' >> scripts/w3mhelp-funcname.pl
+	@case "$(KEYBIND_SRC)" in *lynx*) keymap=keymap.lynx;; *) keymap=keymap.default;; esac; \
+	sed -ne "/^keymap/s/^keymap[ 	][ 	]*\([^ 	]*\)[ 	][ 	]*\([a-zA-Z@][a-zA-Z0-9@_]*\)/'\1', '\2', /p" < $(top_srcdir)/doc/$$keymap >> scripts/w3mhelp-funcname.pl
+	@echo ');' >> scripts/w3mhelp-funcname.pl
+	@cat scripts/w3mhelp-funcname.pl.in >> scripts/w3mhelp-funcname.pl
+	@echo "done"
+
+w3mhelp-funcdesc-stamp: $(top_srcdir)/doc/README.func $(top_srcdir)/doc-jp/README.func scripts/w3mhelp-funcdesc.en.pl.in scripts/w3mhelp-funcdesc.ja.pl.in
+	@echo "generating w3mhelp-funcdesc*.pl..."
+	@for dirlang in $(DOCDIRS); do \
+	  dir=`expr "$$dirlang" : "\(.*\):.*"`; \
+	  lang=`expr "$$dirlang" : ".*:\(.*\)_.*"`; \
+	  echo '%funcdesc = (' > scripts/w3mhelp-funcdesc.$$lang.pl; \
+	  sed -ne "/^[a-zA-Z@]/s/\([a-zA-Z@][a-zA-Z0-9@_]*\)[ 	][ 	]*\(.*\)/'\1', '\2',/p" < $(top_srcdir)/$$dir/README.func >> scripts/w3mhelp-funcdesc.$$lang.pl; \
+	  echo ');' >> scripts/w3mhelp-funcdesc.$$lang.pl; \
+	  cat scripts/w3mhelp-funcdesc.$$lang.pl.in >> scripts/w3mhelp-funcdesc.$$lang.pl; \
+	done
+	@echo done
+	@touch scripts/w3mhelp-funcdesc-stamp
+
 mktable$(EXT): mktable.o dummy.o Str.o hash.o myctype.o
 	$(CC) $(CFLAGS) -o mktable mktable.o dummy.o Str.o hash.o myctype.o $(LDFLAGS) $(LIBS) $(GC_LIBS)
 
@@ -213,7 +240,7 @@
 
 install: install-core install-scripts install-po
 
-install-core: $(TARGETS)
+install-core: $(TARGETS) $(HELP_TARGETS)
 	-$(MKDIR) $(DESTDIR)$(bindir)
 	-$(MKDIR) $(DESTDIR)$(AUXBIN_DIR)
 	-$(MKDIR) $(DESTDIR)$(CGIBIN_DIR)
@@ -236,6 +263,10 @@
 	do \
 		$(INSTALL_PROGRAM) $$file $(DESTDIR)$(CGIBIN_DIR)/$$file; \
 	done
+	for file in scripts/w3mhelp-*.pl; \
+	do \
+		$(INSTALL_DATA) $$file $(DESTDIR)$(HELP_DIR); \
+	done
 
 install-helpfile:
 	-$(MKDIR) $(DESTDIR)$(HELP_DIR)
@@ -277,6 +308,10 @@
 	do \
 		$(RM) -f $(HELP_DIR)/$$file; \
 	done
+	-for file in scripts/w3mhelp-*.pl; \
+	do \
+		rm -f $(HELP_DIR)/$$file; \
+	done
 	-$(RM) -f $(HELP_DIR)/$(HELP_TARGET)
 	-$(RM) -f $(mandir)/man1/$(MAN1_TARGET)
 	-$(RM) -f $(mandir)/ja/man1/$(MAN1_TARGET)
@@ -289,6 +324,7 @@
 	-$(RM) -f *.o *.a $(TARGETS) mktable$(EXT)
 	-$(RM) -f funcname.tab
 	-$(RM) -f funcname.c funcname1.h funcname2.h tagtable.c functable.c
+	-rm -f $(HELP_TARGETS) w3mhelp-*.pl
 	-for dir in w3mimg libwc; \
 	do \
 		(cd $$dir && $(MAKE) clean RM="$(RM)"); \
diff -Nurd -x'*~' w3m-0.5.3.orig/configure w3m-0.5.3/configure
--- w3m-0.5.3.orig/configure	2011-01-14 22:04:22.000000000 -0500
+++ w3m-0.5.3/configure	2015-05-16 23:57:01.000000000 -0400
@@ -4329,7 +4329,7 @@
  WCTARGET="libwc/libwc.a"
  WCCFLAGS='-I$(srcdir) -I$(srcdir)/..'
  wcinclude='-I$(srcdir)/libwc'
- wclib="-L./libwc -lwc"
+ wclib="libwc/libwc.a"
  { $as_echo "$as_me:${as_lineno-$LINENO}: checking if unicode support is enabled" >&5
 $as_echo_n "checking if unicode support is enabled... " >&6; }
  # Check whether --enable-unicode was given.
@@ -6960,6 +6960,8 @@
      { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: unable to build w3mimgdisplay with X11 support" >&5
 $as_echo "$as_me: WARNING: unable to build w3mimgdisplay with X11 support" >&2;}
    fi
+   IMGX11CFLAGS="${IMGX11CFLAGS} `pkg-config x11 --cflags`"
+   IMGX11LDFLAGS="${IMGX11LDFLAGS} `pkg-config x11 --libs`"
   fi
   if test x"$fb" = xyes; then
    if test x"$have_gtk2" = xyes; then
diff -Nurd -x'*~' w3m-0.5.3.orig/etc.c w3m-0.5.3/etc.c
--- w3m-0.5.3.orig/etc.c	2011-01-04 04:22:20.000000000 -0500
+++ w3m-0.5.3/etc.c	2015-05-16 23:57:21.000000000 -0400
@@ -634,24 +634,6 @@
 }
 #endif				/* not HAVE_STRERROR */
 
-#ifndef HAVE_SYS_ERRLIST
-char **sys_errlist;
-
-prepare_sys_errlist()
-{
-    int i, n;
-
-    i = 1;
-    while (strerror(i) != NULL)
-	i++;
-    n = i;
-    sys_errlist = New_N(char *, n);
-    sys_errlist[0] = "";
-    for (i = 1; i < n; i++)
-	sys_errlist[i] = strerror(i);
-}
-#endif				/* not HAVE_SYS_ERRLIST */
-
 int
 next_status(char c, int *status)
 {
diff -Nurd -x'*~' w3m-0.5.3.orig/fm.h w3m-0.5.3/fm.h
--- w3m-0.5.3.orig/fm.h	2011-01-04 04:22:21.000000000 -0500
+++ w3m-0.5.3/fm.h	2015-05-16 23:57:41.000000000 -0400
@@ -1135,7 +1135,7 @@
 #endif
 
 #if defined(USE_SSL) && defined(USE_SSL_VERIFY)
-global int ssl_verify_server init(FALSE);
+global int ssl_verify_server init(TRUE);
 global char *ssl_cert_file init(NULL);
 global char *ssl_key_file init(NULL);
 global char *ssl_ca_path init(NULL);
@@ -1144,7 +1144,7 @@
 #endif				/* defined(USE_SSL) &&
 				 * defined(USE_SSL_VERIFY) */
 #ifdef USE_SSL
-global char *ssl_forbid_method init(NULL);
+global char *ssl_forbid_method init("2");
 #endif
 
 global int is_redisplay init(FALSE);
diff -Nurd -x'*~' w3m-0.5.3.orig/main.c w3m-0.5.3/main.c
--- w3m-0.5.3.orig/main.c	2011-01-04 04:42:19.000000000 -0500
+++ w3m-0.5.3/main.c	2015-05-16 23:57:21.000000000 -0400
@@ -406,9 +406,6 @@
     textdomain(PACKAGE);
 #endif
 
-#ifndef HAVE_SYS_ERRLIST
-    prepare_sys_errlist();
-#endif				/* not HAVE_SYS_ERRLIST */
 
     NO_proxy_domains = newTextList();
     fileToDelete = newTextList();
@@ -833,7 +830,8 @@
     mySignal(SIGPIPE, SigPipe);
 #endif
 
-    orig_GC_warn_proc = GC_set_warn_proc(wrap_GC_warn_proc);
+    orig_GC_warn_proc = GC_get_warn_proc();
+    GC_set_warn_proc(wrap_GC_warn_proc);
     err_msg = Strnew();
     if (load_argc == 0) {
 	/* no URL specified */
diff -Nurd -x'*~' w3m-0.5.3.orig/scripts/Makefile.in w3m-0.5.3/scripts/Makefile.in
--- w3m-0.5.3.orig/scripts/Makefile.in	2011-01-04 04:22:28.000000000 -0500
+++ w3m-0.5.3/scripts/Makefile.in	2015-05-16 23:57:21.000000000 -0400
@@ -31,11 +31,9 @@
 RC_DIR = @RC_DIR@
 ETC_DIR = $(sysconfdir)
 CONF_DIR = $(sysconfdir)/$(PACKAGE)
-DOCDIRS = @DOCDIRS@
 
 AUXBIN_TARGETS	= xface2xpm
 LIB_TARGETS	= dirlist.cgi w3mhelp.cgi w3mmail.cgi
-HELP_TARGETS	= w3mhelp-funcname.pl w3mhelp-funcdesc-stamp
 SUBDIRS = multipart w3mman
 .PHONY: $(SUBDIRS)
 
@@ -46,32 +44,7 @@
 
 PERL		= @PERL@
 
-all: $(LIB_TARGETS) $(AUXBIN_TARGETS) $(HELP_TARGETS) $(SUBDIRS)
-
-w3mhelp-funcname.pl: w3mhelp-funcname.pl.in ../funcname.tab $(top_srcdir)/doc/keymap.default $(top_srcdir)/doc/keymap.lynx
-	@echo "generating w3mhelp-funcname.pl..."
-	@echo '%funcname = (' > w3mhelp-funcname.pl
-	@sed -ne "/^[a-zA-Z@]/s/\([a-zA-Z@][a-zA-Z0-9@_]*\)[ 	][ 	]*\([^ 	]*\)/'\1', '\2',/p" < ../funcname.tab >> w3mhelp-funcname.pl
-	@echo ');' >> w3mhelp-funcname.pl
-	@echo '%keyfunc = (' >> w3mhelp-funcname.pl
-	@case "$(KEYBIND_SRC)" in *lynx*) keymap=keymap.lynx;; *) keymap=keymap.default;; esac; \
-	sed -ne "/^keymap/s/^keymap[ 	][ 	]*\([^ 	]*\)[ 	][ 	]*\([a-zA-Z@][a-zA-Z0-9@_]*\)/'\1', '\2', /p" < $(top_srcdir)/doc/$$keymap >> w3mhelp-funcname.pl
-	@echo ');' >> w3mhelp-funcname.pl
-	@cat $(srcdir)/w3mhelp-funcname.pl.in >> w3mhelp-funcname.pl
-	@echo "done"
-
-w3mhelp-funcdesc-stamp: $(top_srcdir)/doc/README.func $(top_srcdir)/doc-jp/README.func w3mhelp-funcdesc.en.pl.in w3mhelp-funcdesc.ja.pl.in
-	@echo "generating w3mhelp-funcdesc*.pl..."
-	@for dirlang in $(DOCDIRS); do \
-	  dir=`expr "$$dirlang" : "\(.*\):.*"`; \
-	  lang=`expr "$$dirlang" : ".*:\(.*\)_.*"`; \
-	  echo '%funcdesc = (' > w3mhelp-funcdesc.$$lang.pl; \
-	  sed -ne "/^[a-zA-Z@]/s/\([a-zA-Z@][a-zA-Z0-9@_]*\)[ 	][ 	]*\(.*\)/'\1', '\2',/p" < $(top_srcdir)/$$dir/README.func >> w3mhelp-funcdesc.$$lang.pl; \
-	  echo ');' >> w3mhelp-funcdesc.$$lang.pl; \
-	  cat $(srcdir)/w3mhelp-funcdesc.$$lang.pl.in >> w3mhelp-funcdesc.$$lang.pl; \
-	done
-	@echo done
-	@touch w3mhelp-funcdesc-stamp
+all: $(LIB_TARGETS) $(AUXBIN_TARGETS) $(SUBDIRS)
 
 $(SUBDIRS):
 	for subdir in $(SUBDIRS); \
@@ -82,7 +55,6 @@
 install: $(LIB_TARGETS) $(HELP_TARGETS)
 	-$(MKDIR) $(DESTDIR)$(AUXBIN_DIR)
 	-$(MKDIR) $(DESTDIR)$(CGIBIN_DIR)
-	-$(MKDIR) $(DESTDIR)$(HELP_DIR)
 	for file in $(AUXBIN_TARGETS);     \
 	do      \
 		$(INSTALL_SCRIPT) $$file $(DESTDIR)$(AUXBIN_DIR);  \
@@ -91,10 +63,6 @@
 	do      \
 		$(INSTALL_SCRIPT) $$file $(DESTDIR)$(CGIBIN_DIR);  \
 	done
-	for file in w3mhelp-*.pl; \
-	do \
-		$(INSTALL_DATA) $$file $(DESTDIR)$(HELP_DIR); \
-	done
 	for subdir in $(SUBDIRS); \
 	do \
 		(cd $$subdir && $(MAKE) install); \
@@ -109,17 +77,12 @@
 	do      \
 		rm -f $(LIB_DIR)/$$file; \
 	done
-	-for file in w3mhelp-*.pl; \
-	do \
-		rm -f $(HELP_DIR)/$$file; \
-	done
 	-for subdir in $(SUBDIRS); \
 	do \
 		(cd $$subdir && $(MAKE) uninstall); \
 	done
 
 clean:
-	-rm -f $(HELP_TARGETS) w3mhelp-*.pl
 	-for subdir in $(SUBDIRS); \
 	do \
 	  (cd $$subdir && $(MAKE) clean); \
diff -Nurd -x'*~' w3m-0.5.3.orig/url.c w3m-0.5.3/url.c
--- w3m-0.5.3.orig/url.c	2011-01-04 04:22:24.000000000 -0500
+++ w3m-0.5.3/url.c	2015-05-16 23:57:21.000000000 -0400
@@ -82,11 +82,11 @@
     {"ftp", SCM_FTP},
     {"local", SCM_LOCAL},
     {"file", SCM_LOCAL},
-    /*  {"exec", SCM_EXEC}, */
+    {"exec", SCM_EXEC}, 
     {"nntp", SCM_NNTP},
-    /*  {"nntp", SCM_NNTP_GROUP}, */
+    {"nntp", SCM_NNTP_GROUP}, 
     {"news", SCM_NEWS},
-    /*  {"news", SCM_NEWS_GROUP}, */
+    {"news", SCM_NEWS_GROUP}, 
     {"data", SCM_DATA},
 #ifndef USE_W3MMAILER
     {"mailto", SCM_MAILTO},
diff -Nurd -x'*~' w3m-0.5.3.orig/w3mbookmark.c w3m-0.5.3/w3mbookmark.c
--- w3m-0.5.3.orig/w3mbookmark.c	2011-01-04 04:22:24.000000000 -0500
+++ w3m-0.5.3/w3mbookmark.c	2015-05-16 23:57:21.000000000 -0400
@@ -99,7 +99,7 @@
 	fprintf(f, "<body>\n<h1>Bookmarks</h1>\n");
 	fprintf(f, "<h2>%s</h2>\n<ul>\n", section);
 	fprintf(f, "<li><a href=\"%s\">%s</a>\n", url, title);
-	fprintf(f, end_section);
+	fprintf(f, "%s", end_section);
 	fprintf(f, "</ul>\n</body>\n</html>\n");
 	fclose(f);
     }
diff -Nurd -x'*~' w3m-0.5.3.orig/w3mimg/Makefile.in w3m-0.5.3/w3mimg/Makefile.in
--- w3m-0.5.3.orig/w3mimg/Makefile.in	2011-01-04 04:22:28.000000000 -0500
+++ w3m-0.5.3/w3mimg/Makefile.in	2015-05-16 23:57:21.000000000 -0400
@@ -18,6 +18,8 @@
 .PHONY: $(SUBDIRS)
 all: @IMGTARGETS@ w3mimg.a
 
+$(IMGOBJS): @IMGTARGETS@
+
 w3mimg.a: $(IMGOBJS)
 	$(AR) rv $@ $(IMGOBJS)
 	$(RANLIB) $@
diff -ruN w3m-0.5.3.orig/configure w3m-0.5.3/configure
--- w3m-0.5.3.orig/configure	2011-01-14 21:04:22.000000000 -0600
+++ w3m-0.5.3/configure	2020-11-20 03:52:19.000000000 -0600
@@ -9006,6 +9006,7 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <setjmp.h>
+#include <stdlib.h>
 int
 main ()
 {
