### Patch 001-tcltk-paths.patch
--- ruby-1.8.4/ext/tk/extconf.rb.orig	2005-05-23 00:27:37.000000000 -0700
+++ ruby-1.8.4/ext/tk/extconf.rb	2005-10-26 22:55:16.000000000 -0700
@@ -41,14 +41,14 @@
 
 dir_config("tk")
 dir_config("tcl")
-dir_config("X11")
+dir_config("X11", "/usr/X11R6/include", "/usr/X11R6/lib")
 
 tklib = with_config("tklib")
 tcllib = with_config("tcllib")
 stubs = enable_config("tcltk_stubs") || with_config("tcltk_stubs")
 
 def find_tcl(tcllib, stubs)
-  paths = ["/usr/local/lib", "/usr/pkg/lib", "/usr/lib"]
+  paths = ["@PREFIX@/lib", "/usr/local/lib", "/usr/pkg/lib", "/usr/lib"]
   if stubs
     func = "Tcl_InitStubs"
     lib = "tclstub"
@@ -71,7 +71,7 @@
 end
 
 def find_tk(tklib, stubs)
-  paths = ["/usr/local/lib", "/usr/pkg/lib", "/usr/lib"]
+  paths = ["@PREFIX@/lib", "/usr/local/lib", "/usr/pkg/lib", "/usr/lib"]
   if stubs
     func = "Tk_InitStubs"
     lib = "tkstub"


--- ruby-1.8.4/ext/tk/tkutil/extconf.rb~	2005-11-02 12:19:30.000000000 +0100
+++ ruby-1.8.4/ext/tk/tkutil/extconf.rb	2006-07-10 16:11:47.000000000 +0200
@@ -1,4 +1,4 @@
-if compiled?('tk')
+#if compiled?('tk')
   require 'mkmf'
   create_makefile('tkutil')
-end
+  #end
