diff -uNbwr bash_completion/bash-completion.sh bash_completion-new/bash-completion.sh
--- bash_completion/bash-completion.sh	1969-12-31 19:00:00.000000000 -0500
+++ bash_completion-new/bash-completion.sh	2008-01-14 15:55:38.000000000 -0500
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+if [ -n "$BASH_VERSION" ]; then
+	bash=${BASH_VERSION%.*}; bmajor=${bash%.*}; bminor=${bash#*.}
+	if [ $bmajor -eq 1 ]; then
+		exit 0
+	fi
+	if [ $bmajor -eq 2 ] && [ $bminor '<' 05 ]; then
+		exit 0
+	fi
+
+	if [ "$PS1" ] && [ -f @PREFIX@/etc/bash_completion ]; then # interactive shell
+		# Source completion code
+		. @PREFIX@/etc/bash_completion
+		if [ -f @PREFIX@/share/init/bash/completions ]; then
+			. @PREFIX@/share/init/bash/completions
+		fi
+	fi
+	unset bash bmajor bminor
+fi
diff -uNbwr bash_completion/bash_completion bash_completion-new/bash_completion
--- bash_completion/bash_completion	2006-03-01 11:20:18.000000000 -0500
+++ bash_completion-new/bash_completion	2008-01-14 15:51:52.000000000 -0500
@@ -32,8 +32,8 @@
 {
   # These declarations must go within braces in order to be able to silence
   # readonly variable errors.
-  BASH_COMPLETION="${BASH_COMPLETION:-/etc/bash_completion}"
-  BASH_COMPLETION_DIR="${BASH_COMPLETION_DIR:=/etc/bash_completion.d}"
+  BASH_COMPLETION="${BASH_COMPLETION:-@PREFIX@/etc/bash_completion}"
+  BASH_COMPLETION_DIR="${BASH_COMPLETION_DIR:=@PREFIX@/etc/bash_completion.d}"
 } 2>/dev/null || :
 readonly BASH_COMPLETION BASH_COMPLETION_DIR
 
@@ -89,7 +89,7 @@
 complete -f -X '!*.@(dvi|DVI)?(.@(gz|Z|bz2))' xdvi
 complete -f -X '!*.@(dvi|DVI)' dvips dviselect dvitype kdvi dvipdf advi
 complete -f -X '!*.@(pdf|PDF)' acroread gpdf xpdf kpdf
-complete -f -X '!*.@(@(?(e)ps|?(E)PS)?(.gz|.GZ)|pdf|PDF|gif|jp?(e)g|miff|tif?(f)|pn[gm]|p[bgp]m|bmp|xpm|ico|xwd|tga|pcx|GIF|JP?(E)G|MIFF|TIF?(F)|PN[GM]|P[BGP]M|BMP|XPM|ICO|XWD|TGA|PCX)' evince
+complete -f -X '!*.@(@(?(e)ps|?(E)PS)?(.gz|.GZ)|pdf|PDF|dvi|DVI|gif|jp?(e)g|miff|tif?(f)|pn[gm]|p[bgp]m|bmp|xpm|ico|xwd|tga|pcx|GIF|JP?(E)G|MIFF|TIF?(F)|PN[GM]|P[BGP]M|BMP|XPM|ICO|XWD|TGA|PCX)' evince
 complete -f -X '!*.@(?(e)ps|?(E)PS)' ps2pdf
 complete -f -X '!*.texi*' makeinfo texi2html
 complete -f -X '!*.@(?(la)tex|?(LA)TEX|texi|TEXI|dtx|DTX|ins|INS)' tex latex slitex jadetex pdfjadetex pdftex pdflatex texi2dvi
@@ -169,7 +169,7 @@
 
 # use GNU sed if we have it, since its extensions are still used in our code
 #
-[ $UNAME != Linux ] && have gsed && alias sed=gsed
+[ $UNAME != Linux ] && [ $UNAME != Darwin ] && have gsed && alias sed=gsed
 
 # This function checks whether a given readline variable
 # is `on'.
@@ -226,7 +226,9 @@
 #
 _configured_interfaces()
 {
-	if [ -f /etc/debian_version ]; then
+	if [ $UNAME eq "Darwin" ]; then
+		COMPREPLY=( $( /sbin/ifconfig | grep -E '^[^[:space:]]+:' | cut -d: -f1 ) )
+	elif [ -f /etc/debian_version ]; then
 		# Debian system
 		COMPREPLY=( $( sed -ne 's|^iface \([^ ]\+\).*$|\1|p' \
 			       /etc/network/interfaces ) )
@@ -346,8 +348,8 @@
 _services()
 {
 	local sysvdir famdir
-	[ -d /etc/rc.d/init.d ] && sysvdir=/etc/rc.d/init.d || sysvdir=/etc/init.d
-	famdir=/etc/xinetd.d
+	[ -d @PREFIX@/etc/rc.d/init.d ] && sysvdir=@PREFIX@/etc/rc.d/init.d || sysvdir=@PREFIX@/etc/init.d
+	famdir=@PREFIX@/etc/xinetd.d
 	COMPREPLY=( $( builtin echo $sysvdir/!(*.rpmsave|*.rpmorig|*~|functions)) )
 
 	if [ -d $famdir ]; then
@@ -535,7 +537,7 @@
 # 'service' command and/or the SysV init.d directory, followed by
 # that script's available commands
 #
-{ have service || [ -d /etc/init.d/ ]; } &&
+{ have service || [ -d @PREFIX@/etc/init.d/ ]; } &&
 _service()
 {
 	local cur sysvdir
@@ -551,8 +553,8 @@
 	# don't complete past 2nd token
 	[ $COMP_CWORD -gt 2 ] && return 0
 
-	[ -d /etc/rc.d/init.d ] && sysvdir=/etc/rc.d/init.d \
-				|| sysvdir=/etc/init.d
+	[ -d @PREFIX@/etc/rc.d/init.d ] && sysvdir=@PREFIX@/etc/rc.d/init.d \
+				|| sysvdir=@PREFIX@/etc/init.d
 
 	if [[ $COMP_CWORD -eq 1 ]] && [[ $prev == "service" ]]; then
 		_services
@@ -565,8 +567,8 @@
 	return 0
 } &&
 complete -F _service service
-[ -d /etc/init.d/ ] && complete -F _service $default \
-	$(for i in /etc/init.d/*; do echo ${i##*/}; done)
+[ -d @PREFIX@/etc/init.d/ ] && complete -F _service $default \
+	$(for i in @PREFIX@/etc/init.d/*; do echo ${i##*/}; done)
 
 # chown(1) completion
 #
@@ -806,7 +808,7 @@
 [ $UNAME = GNU -o $UNAME = Linux -o $UNAME = Darwin \
   -o $UNAME = FreeBSD -o $UNAME = SunOS -o $UNAME = Cygwin \
   -o $UNAME = OpenBSD ] && \
-complete -F _man $filenames man
+complete -F _man $filenames man apropos whatis
 
 # renice(8) completion
 #
@@ -859,7 +861,7 @@
 
 # Linux and FreeBSD killall(1) completion.
 #
-[ $UNAME = Linux -o $UNAME = FreeBSD ] &&
+[ $UNAME = Linux -o $UNAME = FreeBSD -o $UNAME = Darwin ] &&
 _killall()
 {
 	local cur
@@ -877,7 +879,7 @@
 
 	return 0
 }
-[ $UNAME = Linux -o $UNAME = FreeBSD ] && complete -F _killall killall pkill
+[ $UNAME = Linux -o $UNAME = FreeBSD -o $UNAME = Darwin ] && complete -F _killall killall pkill
 
 # Linux and FreeBSD pgrep(1) completion.
 #
@@ -2034,7 +2036,7 @@
 	if [ -n "$special" ]; then
 		case $special in
 		remove)
-			if [ -f /etc/debian_version ]; then
+			if [ -f /etc/debian_version ] || [ -x "@PREFIX@/bin/fink" ]; then
 				# Debian system
 				COMPREPLY=( $( _comp_dpkg_installed_packages \
 						$cur ) )
@@ -2061,7 +2063,7 @@
 	    -@(t|-target-release|-default-release))
 		     COMPREPLY=( $( apt-cache policy | \
 				    grep "release.o=Debian,a=$cur" | \
-				    sed -e "s/.*a=\(\w*\).*/\1/" | uniq ) )
+				    sed -e "s/.*a=\(\w*\).*/\1/" | uniq 2> /dev/null) )
 		     return 0
 		     ;;
  
@@ -2108,7 +2110,7 @@
 	
 	if [ "$cur" != show ]; then
 	    for (( i=0; i < ${#COMP_WORDS[@]}-1; i++ )); do
-		if [[ ${COMP_WORDS[i]} == @(add|depends|dotty|policy|rdepends|show?(pkg|src|)) ]]; then
+		if [[ ${COMP_WORDS[i]} == @(add|depends|dotty|policy|rdepends|madison|show?(pkg|src|)) ]]; then
 		    special=${COMP_WORDS[i]}
 		fi
 	    done
@@ -2156,7 +2158,7 @@
 		COMPREPLY=( $( compgen -W 'add gencaches show showpkg showsrc \
 				stats dump dumpavail unmet search search \
 				depends rdepends pkgnames dotty xvcg \
-				policy' -- $cur ) )
+				policy madison' -- $cur ) )
 
 	fi
 
@@ -2177,7 +2179,7 @@
 } || {
 _comp_dpkg_hold_packages()
 {
-	grep -B 2 'hold' /var/lib/dpkg/status | grep "Package: $1" \
+	grep -B 2 'hold' @PREFIX@/var/lib/dpkg/status | grep "Package: $1" \
 		| cut -d\  -f2
 }
 }
@@ -2200,7 +2202,7 @@
 		     -Z -v --verbose'
 
 	for (( i=0; i < ${#COMP_WORDS[@]}-1; i++ )); do
-	    if [[ ${COMP_WORDS[i]} == @(install|reinstall|hold|unhold|markauto|unmarkauto|dist-upgrade|download|show|forbid-version|purge|remove) ]]; then
+	    if [[ ${COMP_WORDS[i]} == @(install|reinstall|hold|unhold|markauto|unmarkauto|dist-upgrade|download|show|forbid-version|purge|remove|changelog) ]]; then
 		special=${COMP_WORDS[i]}
 	    fi
 	    #exclude some mutually exclusive options
@@ -2210,7 +2212,7 @@
 
 	if [[ -n "$special" ]]; then
 	   case $special in
-	       @(install|hold|markauto|unmarkauto|dist-upgrade|download|show))
+	       @(install|hold|markauto|unmarkauto|dist-upgrade|download|show|changelog))
 		   COMPREPLY=( $( apt-cache pkgnames $cur 2> /dev/null ) )
 		   return 0
 		   ;;
@@ -2240,7 +2242,7 @@
 	    -@(t|-target-release|-default-release))
 		COMPREPLY=( $( apt-cache policy | \
 		    grep "release.o=Debian,a=$cur" | \
-		    sed -e "s/.*a=\(\w*\).*/\1/" | uniq ) )
+		    sed -e "s/.*a=\(\w*\).*/\1/" | uniq 2> /dev/null ) )
 		return 0
 		;;
 
@@ -2253,7 +2255,7 @@
 				       autoclean install reinstall remove \
 				       hold unhold purge markauto unmarkauto \
 				       dist-upgrade download search show \
-				       forbid-version' -- $cur ) )
+				       forbid-version changelog' -- $cur ) )
 	fi
 
 
@@ -2339,7 +2341,7 @@
 	prev=${COMP_WORDS[COMP_CWORD-1]}
 
 	if [ "$prev" = "-s" ]; then
-	  if [ -f /etc/debian_version ]; then
+	  if [ -f /etc/debian_version ] || [ -x @PREFIX@/bin/fink ]; then
 	    COMPREPLY=( $( </etc/shells ) )
 	  else
 	    COMPREPLY=( $( chsh -l | grep "^$cur" ) )
@@ -2422,6 +2424,8 @@
 	kh=()
 
 	# ssh config files
+	[ -r @PREFIX@/etc/ssh/ssh_config ] &&
+	  config=( ${config[@]} @PREFIX@/etc/ssh/ssh_config )
 	[ -r /etc/ssh/ssh_config ] &&
 	  config=( ${config[@]} /etc/ssh/ssh_config )
 	[ -r ~/.ssh/config ] &&
@@ -2440,12 +2444,20 @@
 	if [ -r "$global_kh" ]; then
 	    kh=( "$global_kh" )
 	else
+	    [ -r @PREFIX@/etc/ssh/ssh_known_hosts ] &&
+	      kh=( ${kh[@]} @PREFIX@/etc/ssh/ssh_known_hosts )
 	    [ -r /etc/ssh/ssh_known_hosts ] &&
 	      kh=( ${kh[@]} /etc/ssh/ssh_known_hosts )
+	    [ -r @PREFIX@/etc/ssh/ssh_known_hosts2 ] &&
+	      kh=( ${kh[@]} @PREFIX@/etc/ssh/ssh_known_hosts2 )
 	    [ -r /etc/ssh/ssh_known_hosts2 ] &&
 	      kh=( ${kh[@]} /etc/ssh/ssh_known_hosts2 )
+	    [ -r @PREFIX@/etc/known_hosts ] &&
+	      kh=( ${kh[@]} @PREFIX@/etc/known_hosts )
 	    [ -r /etc/known_hosts ] &&
 	      kh=( ${kh[@]} /etc/known_hosts )
+	    [ -r @PREFIX@/etc/known_hosts2 ] &&
+	      kh=( ${kh[@]} @PREFIX@/etc/known_hosts2 )
 	    [ -r /etc/known_hosts2 ] &&
 	      kh=( ${kh[@]} /etc/known_hosts2 )
 	    [ -d /etc/ssh2/knownhosts ] &&
@@ -3323,7 +3335,7 @@
 have bzip2 &&
 _bzip2()
 {
-	local cur prev xspec
+	local cur prev xspec IFS=$'\t\n'
 
 	COMPREPLY=()
 	cur=${COMP_WORDS[COMP_CWORD]}
@@ -3363,8 +3375,8 @@
 {
 	local config
 
-	config=/etc/ssl/openssl.cnf
-	[ ! -f $config ] && config=/usr/share/ssl/openssl.cnf
+	config=@PREFIX@/etc/ssl/openssl.cnf
+	[ ! -f $config ] && config=/System/Library/OpenSSL/openssl.cnf
 	for (( i=2; i < COMP_CWORD; i++ )); do
 		if [[ "${COMP_WORDS[i]}" == -config ]]; then
 			config=${COMP_WORDS[i+1]}
@@ -4206,7 +4218,7 @@
 } || {
 _comp_dpkg_installed_packages()
 {
-	grep -A 2 "Package: $1" /var/lib/dpkg/status | \
+	grep -A 2 "Package: $1" @PREFIX@/var/lib/dpkg/status | \
 		grep -B 2 'ok installed' | grep "Package: $1" | cut -d\  -f2
 }
 }
@@ -4981,7 +4993,7 @@
 			--no-cc --package-version -z --no-compress \
 			--ui --interface -u \
 			wnpp boot-floppies kernel-image' -- $cur ) \
-	    		$( apt-cache pkgnames -- $cur ) )
+	    		$( apt-cache pkgnames -- $cur 2> /dev/null ) )
 	_filedir
 	return 0
 } &&
@@ -5016,7 +5028,7 @@
 			-B --bts -l --ldap --no-ldap --proxy= --http_proxy= \
 			-s --source -w --web -u --ui --interface \
 			wnpp boot-floppies' -- $cur ) \
-	    		$( apt-cache pkgnames -- $cur ) )
+	    		$( apt-cache pkgnames -- $cur 2> /dev/null ) )
 } &&
 complete -F _querybts $filenames querybts
 
@@ -5029,6 +5041,7 @@
 	# find the admin dir
 	for i in alternatives dpkg/alternatives rpm/alternatives; do
 		[ -d /var/lib/$i ] && admindir=/var/lib/$i && break
+      [ -d "@PREFIX@/etc/$i" ] && admindir="@PREFIX@/etc/$i" && break
 	done
 	for (( i=1; i < COMP_CWORD; i++ )); do
 		if [[ "${COMP_WORDS[i]}" == --admindir ]]; then
@@ -5888,7 +5901,7 @@
 					-xvidencopts -of --verbose' -- $cur) )
 			;;
 		*)
-			_filedir '@(mp?(e)g|MP?(E)G|wm[av]|WM[AV]|avi|AVI|asf|ASF|vob|VOB|bin|BIN|dat|DAT|vcd|VCD|ps|PS|pes|PES|fli|FLI|viv|VIV|rm?(j)|RM?(J)|ra?(m)|RA?(M)|yuv|YUV|mov|MOV|qt|QT|mp[34]|MP[34]|og[gm]|OG[GM]|wav|WAV|dump|DUMP|mkv|MKV|m4a|M4A|aac|AAC|m2v|M2V|dv|DV|rmvb|RMVB|mid|MID|ts|TS|3gp|mpc|MPC|flac|FLAC)'
+			_filedir '@(mp?(e)g|MP?(E)G|wm[av]|WM[AV]|avi|AVI|asf|ASF|vob|VOB|bin|BIN|dat|DAT|vcd|VCD|ps|PS|pes|PES|fli|FLI|viv|VIV|rm?(j)|RM?(J)|ra?(m)|RA?(M)|yuv|YUV|mov|MOV|qt|QT|mp[34]|MP[34]|og[gm]|OG[GM]|wav|WAV|dump|DUMP|mkv|MKV|m4a|M4A|aac|AAC|m2v|M2V|dv|DV|rmvb|RMVB|mid|MID|ts|TS|3gp|mpc|MPC|flac|FLAC|flv|FLV)'
 			;;
 	esac
 
@@ -8803,7 +8816,7 @@
 		COMPREPLY=( $( compgen -W '-s -o -m -M -z -l -L -w -a -t \
 			-c -S -d -p -C -T -8' -- $cur ) )
 	else
-		COMPREPLY=( $( command ls /etc/minirc.* | sed -e 's|/etc/minirc.||' | grep "^$cur" ) )
+		COMPREPLY=( $( command ls /etc/minicom/minirc.* 2>/dev/null | sed -e 's|/etc/minicom/minirc.||' | grep "^$cur" ) )
 	fi
 } &&
 complete -F _minicom minicom
@@ -9253,6 +9266,7 @@
 		    \"${cur#[\`\"\']}\" 2>/dev/null ) \
 		    $( compgen -d -- $cur ) )
 }
+
 list=( $( sed -ne '/^# START exclude/,/^# FINISH exclude/p' \
 	  $BASH_COMPLETION | \
 	# read exclusion compspecs
diff -uNbwr bash_completion/fink.sh bash_completion-new/fink.sh
--- bash_completion/fink.sh	1969-12-31 19:00:00.000000000 -0500
+++ bash_completion-new/fink.sh	2008-01-14 15:51:52.000000000 -0500
@@ -0,0 +1,128 @@
+#!/bin/bash
+
+# 
+# Copyright (c) 2004, Lukas Westermann <contact AT openos DOT ch>
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without 
+# modification, are permitted provided that the following conditions 
+# are met:
+#   - Redistributions of source code must retain the above copyright 
+#     notice, this list of conditions and the following disclaimer.
+#   - Redistributions in binary form must reproduce the above 
+#     copyright notice, this list of conditions and the following 
+#     disclaimer in the documentation and/or other materials 
+#     provided with the distribution.
+#   - The names of its contributors may not be used to endorse or 
+#     promote products derived from this software without specific 
+#     prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
+# CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+# INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR 
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT 
+# NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) 
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR 
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, 
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+#
+# Completion for Fink command
+#
+# History
+# ---------------------------------
+# Version 1.0.0  - initial release
+# Version 1.0.1 - added test to check fink availability
+# Version 1.0.2 - simplified checks for fink availabiltiy
+# Version 1.0.3 - fixed problems with spaces in completion
+#
+
+[ $UNAME = Darwin ] && have fink && {
+_fink_installed() {
+    fink list -i --tab "$*" | cut -f2 2>/dev/null
+}
+
+_fink_all() {
+    apt-cache pkgnames "$*" 2>/dev/null
+}
+
+_fink_trees() {
+    cat @PREFIX@/etc/fink.conf | sed -e '/^Trees/!d; s/^Trees://' 2>/dev/null
+}
+
+_fink()
+{
+	local cur prev special i
+	
+	COMPREPLY=()
+	cur=${COMP_WORDS[COMP_CWORD]}
+	prev=${COMP_WORDS[COMP_CWORD-1]}
+
+	for (( i=0; i < ${#COMP_WORDS}-1; i++ )); do
+		if [[ ${COMP_WORDS[i]} == @(install|update|enable|activate|use|list|remove|disable|deactivate|unuse|delete \
+		                            |purge|apropos|describe|desc|description|info|fetch-missing|fetch \
+		                            |build|rebuild|reinstall|validate|check|scanpackages) ]]; then
+			special=${COMP_WORDS[i]}
+		fi
+	done
+
+	if [ -n "${special}" ]; then
+		case $special in
+		    install|enable|activate|use|apropos|describe|desc|description|info|fetch|build)
+			COMPREPLY=( $( _fink_all $cur ) )
+			;;
+		    remove|disable|deactivate|unuse|delete|purge|update|rebuild|reinstall)
+			COMPREPLY=( $( _fink_installed $cur ) )
+			;;		    
+		    fetch-missing)
+			COMPREPLY=( $( compgen -W '-i --ignore-restrictive' -- $cur ) )
+			;;
+		    validate|check)
+			local IFS=$'\t\n'
+			_filedir '@(deb|info)'
+			#COMPREPLY=( $( compgen -f -- $cur ) )
+			;;
+		    scanpackages)
+			COMPREPLY=( $( compgen -W '$( _fink_trees )' -- $cur ) )
+			;;
+		    list)
+			if [[ "$cur" == -* ]]; then
+			    # TODO: Fix problems with --tree=
+			    #if [[ $cur == --tree=* || $cur == -t=* ]]; then
+				#local pfx las #res
+                                #pfx=$( echo $cur | sed -e 's/^\([^=]*\)=.*$/\1/' )
+				#las=$( echo "$cur" | sed -e 's/^[^=]*=//' )
+				#[[ $las == $cur ]] && las=()
+				#res=$( cat @PREFIX@/etc/fink.conf | sed -e '/^Trees/!d; s/^Trees://' 2>/dev/null )
+                                #COMPREPLY=( $( compgen -W '$( _fink_trees )' -- "${las}" ) )
+			    #else
+				COMPREPLY=( $( compgen -W '-h --help -t --tab -i --installed -o --outdated -u --uptodate -n --notinstalled \
+                                                       -s --section -m --maintainer --tree -w --width' -- $cur ) )
+			    #fi
+			else
+			    COMPREPLY=( $( _fink_all $cur ) )
+			fi
+			;;
+		esac
+
+		return 0
+	fi
+
+	if [[ "$cur" == -* ]]; then
+		COMPREPLY=( $( compgen -W '-h -q -V -v -y --help --quite --version --verbose --yes' -- $cur ) )
+	else
+		COMPREPLY=( $( compgen -W 'install remove purge update selfupdate update-all configure list \
+		                           apropos describe fetch fetch-all fetch-missing index validate scanpackages \
+                                           build rebuild reinstall checksums cleanup' -- $cur ) )
+	fi
+
+	return 0
+}
+complete $filenames -F _fink fink
+}
\ No newline at end of file
diff -uNbwr bash_completion/open.sh bash_completion-new/open.sh
--- bash_completion/open.sh	1969-12-31 19:00:00.000000000 -0500
+++ bash_completion-new/open.sh	2008-01-14 15:51:52.000000000 -0500
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+# 
+# Copyright (c) 2004, Lukas Westermann <contact AT openos DOT ch>
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without 
+# modification, are permitted provided that the following conditions 
+# are met:
+#   - Redistributions of source code must retain the above copyright 
+#     notice, this list of conditions and the following disclaimer.
+#   - Redistributions in binary form must reproduce the above 
+#     copyright notice, this list of conditions and the following 
+#     disclaimer in the documentation and/or other materials 
+#     provided with the distribution.
+#   - The names of its contributors may not be used to endorse or 
+#     promote products derived from this software without specific 
+#     prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
+# CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+# INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR 
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT 
+# NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) 
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR 
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, 
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+#
+# Bash Auto-Completion for the NextStep command 'open'
+#
+# History
+# ----------------------------------
+# Version 1.0.0 - initial release
+# Version 1.0.1 - added test for availabilty (Darwin only
+#                 and check for open)
+# Version 1.0.2 - fixed problems with IFS for -a and -e
+# Version 1.0.3 - simplified availability check
+# Version 1.0.4 - fixed problem with spaces and completion
+#
+
+[ $UNAME = Darwin ] && have open && {
+_list_apps() {
+    local IFS=$'\n'
+    find '/Applications' "${HOME}"'/Applications' '/Developer/Applications' -name "${1}*.app" -maxdepth 2 2>/dev/null
+    #TODO: MS Word and other carbon apps...HOW ???
+}
+
+_open()
+{
+
+    local cur prev IFS=$'\t\n'
+    COMPREPLY=()
+    cur=${COMP_WORDS[COMP_CWORD]}
+    prev=${COMP_WORDS[COMP_CWORD-1]}
+
+    if [[ $prev == -a ]]; then
+	local res ncur
+	ncur=$( echo "$cur" | sed -e 's/\\//g' 2>/dev/null )
+	res=$( _list_apps "${ncur}" | sed -e 's/.*\/\([^\/]*\)\.app/\1/' )
+	COMPREPLY=( $( compgen -W "${res}" -- "${ncur}" ) )
+    elif [[ $prev == -e ]]; then
+	_filedir '@(txt|doc|rtf|rtfd|pdf|png|jpg|gif)'
+    elif [[ $cur == -* ]]; then
+	local IFS=$' \t\n'
+	COMPREPLY=( $( compgen -W '-a -e' -- "$cur" ) )
+    else
+	local first="${COMP_WORDS[1]}" second="${COMP_WORDS[2]}"
+	if [[ $first == -a ]]; then
+	    case "$second" in
+		QuickTime*)
+		    _filedir '@(mp3|mpg|mpeg|avi|m4*|mov|alc)'
+		    ;;
+		iTunes)
+		    _filedir '@(mp3|m4*|ogg|alc)'
+		    ;;
+		Safari|Mozilla|Camino|Opera|Firebird)
+		    # TODO: complete from history/bookmarks ?
+		    _filedir '@(htm|html|xhtm|xhtml|php|php3|php3|xml)'
+		    ;;
+		VLC)
+		    # TODO: complete for CD/DVD too...
+		    _filedir '@(avi|mpg)'
+		    ;;
+		BitTorrent*)
+		    _filedir '@(torrent)'
+		    ;;
+		*)
+		    # others, complete all
+		    _filedir
+		    ;;
+	    esac
+	else
+	    _filedir
+	fi
+    fi
+
+}
+complete $filenames -F _open open
+}
\ No newline at end of file
