diff -Nurd -x'*~' astyle.orig/astyle_main.cpp astyle/astyle_main.cpp
--- astyle.orig/astyle_main.cpp	2002-03-07 08:22:30.000000000 -0500
+++ astyle/astyle_main.cpp	2006-01-17 23:24:36.000000000 -0500
@@ -151,47 +151,6 @@
 
 
 
-template<class ITER>
-bool parseOptions(ASFormatter &formatter,
-                  const ITER &optionsBegin,
-                  const ITER &optionsEnd,
-                  const string &errorInfo)
-{
-    ITER option;
-    bool ok = true;
-    string arg, subArg;
-
-    for (option = optionsBegin; option != optionsEnd; ++option)
-    {
-        arg = *option; //string(*option);
-
-        if (arg.COMPARE(0, 2, string("--")) == 0)
-            ok &= parseOption(formatter, arg.substr(2), errorInfo);
-        else if (arg[0] == '-')
-        {
-            int i;
-
-            for (i=1; i < arg.length(); ++i)
-            {
-                if (isalpha(arg[i]) && i > 1)
-                {
-                    ok &= parseOption(formatter, subArg, errorInfo);
-                    subArg = "";
-                }
-                subArg.append(1, arg[i]);
-            }
-            ok &= parseOption(formatter, subArg, errorInfo);
-            subArg = "";
-        }
-        else
-        {
-            ok &= parseOption(formatter, arg, errorInfo);
-            subArg = "";
-        }
-    }
-
-    return ok;
-}
 
 void manuallySetJavaStyle(ASFormatter &formatter)
 {
@@ -425,6 +384,47 @@
     return true; //o.k.
 }
 
+template<class ITER>
+bool parseOptions(ASFormatter &formatter,
+                  const ITER &optionsBegin,
+                  const ITER &optionsEnd,
+                  const string &errorInfo)
+{
+    ITER option;
+    bool ok = true;
+    string arg, subArg;
+
+    for (option = optionsBegin; option != optionsEnd; ++option)
+    {
+        arg = *option; //string(*option);
+
+        if (arg.COMPARE(0, 2, string("--")) == 0)
+            ok &= parseOption(formatter, arg.substr(2), errorInfo);
+        else if (arg[0] == '-')
+        {
+            int i;
+
+            for (i=1; i < arg.length(); ++i)
+            {
+                if (isalpha(arg[i]) && i > 1)
+                {
+                    ok &= parseOption(formatter, subArg, errorInfo);
+                    subArg = "";
+                }
+                subArg.append(1, arg[i]);
+            }
+            ok &= parseOption(formatter, subArg, errorInfo);
+            subArg = "";
+        }
+        else
+        {
+            ok &= parseOption(formatter, arg, errorInfo);
+            subArg = "";
+        }
+    }
+
+    return ok;
+}
 
 
 void importOptions(istream &in, vector<string> &optionsVector)
