diff -Nurd -x'*~' cscope-15.5.orig/configure cscope-15.5/configure
--- cscope-15.5.orig/configure	2003-09-04 12:39:53.000000000 -0400
+++ cscope-15.5/configure	2005-12-08 21:22:10.000000000 -0500
@@ -4370,13 +4370,8 @@
 
 
 
-if pkg-config --exists gtk+-2.0; then
-  USING_GNOME2_TRUE=
-  USING_GNOME2_FALSE='#'
-else
   USING_GNOME2_TRUE='#'
   USING_GNOME2_FALSE=
-fi
 
 
 
diff -Nurd -x'*~' cscope-15.5.orig/src/constants.h cscope-15.5/src/constants.h
--- cscope-15.5.orig/src/constants.h	2003-09-04 11:54:02.000000000 -0400
+++ cscope-15.5/src/constants.h	2005-12-08 20:50:11.000000000 -0500
@@ -102,7 +102,7 @@
 #endif
 
 
-#ifndef __FreeBSD__	/* Prevent search issues in cscope.out */
+#if !defined(__FreeBSD__) && !defined(__APPLE__)	/* Prevent search issues in cscope.out */
 #if !TERMINFO
 #ifndef KEY_BREAK
 #define	KEY_BREAK	0400	/* easier to define than to add #if around the use */
@@ -127,6 +127,6 @@
 #endif	/* if UNIXPC */
 
 #endif	/* if !TERMINFO */
-#endif	/* ifndef __FreeBSD__ */
+#endif	/* if !defined(__FreeBSD__) && !defined(__APPLE__) */
 
 #endif /* CSCOPE_CONSTANTS_H */
diff -Nurd -x'*~' cscope-15.5.orig/src/egrep.y cscope-15.5/src/egrep.y
--- cscope-15.5.orig/src/egrep.y	2002-12-11 09:39:10.000000000 -0500
+++ cscope-15.5/src/egrep.y	2005-12-08 21:27:12.000000000 -0500
@@ -605,7 +605,7 @@
 
 /* FIXME HBB: should export this to a separate file and use
  * AC_REPLACE_FUNCS() */
-#if BSD
+#ifndef HAVE_MEMSET
 /*LINTLIBRARY*/
 /*
  * Set an array of n chars starting at sp to the character c.
diff -Nurd -x'*~' cscope-15.5.orig/src/exec.c cscope-15.5/src/exec.c
--- cscope-15.5.orig/src/exec.c	2002-07-29 08:37:49.000000000 -0400
+++ cscope-15.5/src/exec.c	2005-12-08 21:15:31.000000000 -0500
@@ -57,7 +57,7 @@
 
 #ifndef __MSDOS__ /* none of these is needed, there */
 static	int	join(pid_t p);
-static	int	myexecvp(char *a, char **args);
+static	void	myexecvp(char *a, char **args);
 static	pid_t	myfork(void);
 #endif
 
@@ -87,7 +87,7 @@
         exitcode = spawnvp(P_WAIT, a, argv);
 #else
 	if ((p = myfork()) == 0) {
-		(void) myexecvp(a, argv);	/* child */
+		myexecvp(a, argv);	/* child (myexecvp doesn't return) */
 	}
 	else {
 		exitcode = join(p);	/* parent */
@@ -114,7 +114,7 @@
  * modify argv[0] to reference the last component of its path-name.
  */
 
-static int
+static void
 myexecvp(char *a, char **args)
 {
 	char	msg[MSGLEN + 1];
diff -Nurd -x'*~' cscope-15.5.orig/src/global.h cscope-15.5/src/global.h
--- cscope-15.5.orig/src/global.h	2003-09-04 11:54:03.000000000 -0400
+++ cscope-15.5/src/global.h	2005-12-08 20:48:00.000000000 -0500
@@ -90,11 +90,11 @@
 # define	tolower(c)	(islower(c) ? (c) : (c) - 'A' + 'a')	
 # define	toupper(c)	(isupper(c) ? (c) : (c) - 'a' + 'A')	
 # if !sun 
-#  if !__FreeBSD__
+#  if !__FreeBSD__ && !__APPLE__
 /* in traditional BSD, *printf() doesn't return the number of bytes
  * written */
 #   define PRINTF_RETVAL_BROKEN 1
-#  endif /* !FreeBSD */
+#  endif /* !FreeBSD && !__APPLE__ */
 # endif /* !sun */
 #endif
 
