diff -ru ddd-3.3.9/configure ddd-3.3.9-patched/configure
--- ddd-3.3.9/configure	2004-06-20 20:40:09.000000000 +0200
+++ ddd-3.3.9-patched/configure	2004-08-17 00:57:21.000000000 +0200
@@ -3619,7 +3619,7 @@
 rm -f conftest$ac_cv_exeext
 echo "$as_me:$LINENO: result: $ac_cv_exeext" >&5
 echo "${ECHO_T}$ac_cv_exeext" >&6
-
+ac_cv_exeext=".exe"
 rm -f conftest.$ac_ext
 EXEEXT=$ac_cv_exeext
 ac_exeext=$EXEEXT
