diff -uNr maven-2.0.6/bin/mvn maven-2.0.6-new/bin/mvn
--- maven-2.0.6/bin/mvn	2007-03-26 22:39:29.000000000 -0400
+++ maven-2.0.6-new/bin/mvn	2007-05-17 13:30:28.000000000 -0400
@@ -48,6 +48,10 @@
 if [ -f /etc/mavenrc ] ; then
   . /etc/mavenrc
 fi
+if [ -f @FINKPREFIX@/etc/maven/mavenrc ] ; then
+  . @FINKPREFIX@/etc/maven/mavenrc
+fi
+
 
 if [ -f "$HOME/.mavenrc" ] ; then
   . "$HOME/.mavenrc"
@@ -146,6 +150,8 @@
   echo "Warning: JAVA_HOME environment variable is not set."
 fi
 
+[ -z "$CLASSWORLDS_CONF" ] && CLASSWORLDS_CONF="${M2_HOME}/bin/m2.conf"
+
 CLASSWORLDS_LAUNCHER=org.codehaus.classworlds.Launcher
 
 # For Cygwin, switch paths to Windows format before running java
@@ -161,7 +167,7 @@
 exec "$JAVACMD" \
   $MAVEN_OPTS \
   -classpath "${M2_HOME}"/boot/classworlds-*.jar \
-  "-Dclassworlds.conf=${M2_HOME}/bin/m2.conf" \
+  "-Dclassworlds.conf=${CLASSWORLDS_CONF}" \
   "-Dmaven.home=${M2_HOME}"  \
   ${CLASSWORLDS_LAUNCHER} $QUOTED_ARGS
 
