diff -urN xerces-c-src_2_6_0-orig/samples/runConfigure xerces-c-src_2_6_0/samples/runConfigure
--- xerces-c-src_2_6_0-orig/samples/runConfigure	Fri Sep 10 09:52:24 2004
+++ xerces-c-src_2_6_0/samples/runConfigure	Mon Mar 14 09:38:18 2005
@@ -93,17 +93,18 @@
 transcoder=native        # by default use native transcoder
 thread=none              # by default no need to have threads
 bitsToBuild=32           # by default 32 bit build assumed
+configureoptions=""
 
 # Check the command line parameters
 if test -x /usr/bin/getopt; then
-   getoptErr=`getopt p:c:x:dr:b:l:z:h $*`
+   getoptErr=`getopt p:c:x:dr:b:l:z:C:h $*`
    if [ $? != 0 ]
       then
       usage
       exit ${ERROR_EXIT_CODE}
    fi
    # Now get the command line parameters
-   set -- `getopt p:c:x:dr:b:l:z:h $*`
+   set -- `getopt p:c:x:dr:b:l:z:C:h $*`
    while [ $# -gt 0 ]
       do
       case $1 in
@@ -131,6 +132,9 @@
       -l)
          linkeroptions="$linkeroptions $2"; shift 2;;
 
+      -C)
+	 configureoptions="$configureoptions $2"; shift 2;;
+
       -h)
          usage
          exit ${ERROR_EXIT_CODE};;
@@ -145,7 +149,7 @@
       esac
    done
 else
-   while getopts "p:c:x:dr:b:l:z:h" switch; do
+   while getopts "p:c:x:dr:b:l:z:C:h" switch; do
       case $switch in
       p)
          platform=$OPTARG;;
@@ -191,6 +195,7 @@
 echo "bitsToBuild option: $bitsToBuild"
 echo "Extra compile options: $compileroptions"
 echo "Extra link options: $linkeroptions"
+echo "Extra configure options: $configureoptions"
 
 #
 # Now check if the options are correct or not, bail out if incorrect
@@ -495,11 +500,11 @@
 rm -f config.log
 rm -f config.status
 if test $platform = "os400"; then
-./configure --host AS400-OS400
+./configure --host AS400-OS400 $configureoptions
 elif test $platform = "ptx"; then
-./configure --prefix=$XMLINSTALL
+./configure --prefix=$XMLINSTALL $configureoptions
 else
-./configure
+./configure $configureoptions
 fi
 # exit if configure failed
 if test $? != 0; then
@@ -518,7 +523,7 @@
 echo export CFLAGS=\"$CFLAGS\"
 echo export LDFLAGS=\"$LDFLAGS\"
 echo export EXTRA_LIBS=\"$EXTRA_LIBS\"
-echo configure
+echo configure $configureoptions
 
 echo
 echo If the result of the above commands look OK to you, go to the directory
