--- lyx-1.2.1/boost/boost/detail/limits.hpp.orig	Tue Apr  9 19:21:23 2002
+++ lyx-1.2.1/boost/boost/detail/limits.hpp	Tue Apr  9 19:21:43 2002
@@ -35,7 +35,9 @@
 
 #include <climits>
 #include <cfloat>
+#ifndef __APPLE__
 #include <cwchar>             // for WCHAR_MIN and WCHAR_MAX
+#endif
 #include <boost/config.hpp>
 
 #if defined(__sparc) || defined(__sparc__) || defined(__powerpc__) || defined(__ppc__) || defined(__hppa) || defined(_MIPSEB)
--- lyx-1.2.1/src/frontends/xforms/xformsGImage.C	31 Oct 2002 11:14:13 -0000	1.12
+++ lyx-1.2.1/src/frontends/xforms/xformsGImage.C	7 Nov 2002 09:36:23 -0000
@@ -424,7 +424,6 @@
 	flimage_enable_tiff();
 	flimage_enable_xbm();
 	flimage_enable_xwd();
-	flimage_enable_xpm();
 
 	// xforms stores this permanently (does not make a copy) so
 	// this should never be destroyed.
--- lyx-1.2.1/lib/lyxrc.example	Thu Jan 10 08:14:20 2002
+++ lyx-1.2.1/lib/lyxrc.example	Thu Nov  7 09:04:58 2002
@@ -611,3 +611,10 @@
 #\screen_font_encoding iso8859-6.8x
 #\screen_font_encoding_menu iso8859-1
 #\screen_font_roman "-*-naskhi"
+#
+# CONVERTERS SECTION ##########################
+#
+
+\converter "epsi" "ppm" "convert EPS:$$i PPM:$$o" ""
+\converter "eps" "ppm" "convert EPS:$$i PPM:$$o" ""
+\converter "ps" "ppm" "convert PS:$$i PPM:$$o" ""
