--- agqt.ori/ag/main.cc	Wed Feb 20 11:26:44 2002
+++ agqt/ag/main.cc	Thu Aug  8 17:02:02 2002
@@ -6,6 +6,10 @@
 #include "myqueue.h"
 #include "printproc.h"
 
+#include <sys/types.h>
+#include <sys/time.h>
+#include <sys/resource.h>
+
 int
 main(int argc, char *argv[])
 {
@@ -23,6 +27,11 @@
 		errorprint("and download the second, versionnum should be 2.\n");
 		exit(-1);
 	}
+
+        struct rlimit rlim;
+        getrlimit(RLIMIT_STACK, &rlim);
+        rlim.rlim_cur=600000;
+        setrlimit(RLIMIT_STACK, &rlim);
 
 	int timeWanted = 0;
 	int versionNum = 1;
--- agqt.ori/am/main.cc	Tue Apr 30 08:06:59 2002
+++ agqt/am/main.cc	Thu Aug  8 17:02:02 2002
@@ -4,6 +4,10 @@
 #include "web.h"
 #include "common.h"
 #include "cfgfile.h"
+ 
+#include <sys/types.h>
+#include <sys/time.h>
+#include <sys/resource.h>
 
 int
 main(int argc, char *argv[])
@@ -31,6 +35,11 @@
 		fprintf(stderr, "   %s \"vic chesnutt\" \\\n\t\"http://www.allmusic.com/cg/amg.dll?p=amg&sql=Ayf9fs30ia3mg\"\n", argv[0]);
 		exit(-1);
 	}
+
+        struct rlimit rlim;
+        getrlimit(RLIMIT_STACK, &rlim);
+        rlim.rlim_cur=600000;
+        setrlimit(RLIMIT_STACK, &rlim);
 
 	readConfig();
 
--- agqt.ori/fdb/main.cc	Tue Apr 30 08:06:48 2002
+++ agqt/fdb/main.cc	Thu Aug  8 17:02:02 2002
@@ -5,6 +5,10 @@
 #include "common.h"
 #include "cfgfile.h"
 
+#include <sys/types.h>
+#include <sys/time.h>
+#include <sys/resource.h>
+
 int
 main(int argc, char *argv[])
 {
@@ -27,6 +31,11 @@
                 fprintf(stderr, "   %s \"vic chesnutt\" \\\n\t\"http://www.allmusic.com/cg/amg.dll?p=amg&sql=Ayf9fs30ia3mg\"\n", argv[0]);
                 exit(-1);
         }
+
+        struct rlimit rlim;
+        getrlimit(RLIMIT_STACK, &rlim);
+        rlim.rlim_cur=600000;
+        setrlimit(RLIMIT_STACK, &rlim);
 
 	readConfig();
 
--- agqt.ori/gq/main.cc	Tue Apr 30 21:06:24 2002
+++ agqt/gq/main.cc	Thu Aug  8 22:28:39 2002
@@ -4,6 +4,10 @@
 #include "client.h"
 #include "common.h"
 
+#include <sys/types.h>
+#include <sys/time.h>
+#include <sys/resource.h>
+
 int
 main(int argc, char *argv[])
 {
@@ -16,6 +20,11 @@
 		fprintf(stderr, "Usage: %s \"query\"\n", argv[0]);
 		exit(-1);
 	}
+
+        struct rlimit rlim;
+        getrlimit(RLIMIT_STACK, &rlim);
+        rlim.rlim_cur=rlim.rlim_max;
+        setrlimit(RLIMIT_STACK, &rlim);
 
 	server *S = new server("localhost", 1213);
 
diff -ur agqt.ori/ag/myqueue.h agqt/ag/myqueue.h
--- agqt.ori/ag/myqueue.h	Fri Jan  4 19:22:09 2002
+++ agqt/ag/myqueue.h	Fri Aug  9 10:33:06 2002
@@ -3,6 +3,9 @@
 
 #include <queue>
 
+using std::queue;
+using std::priority_queue;
+
 struct fileInfo
 {
         fileInfo(char *n, char *u)
diff -ur agqt.ori/am/myqueue.h agqt/am/myqueue.h
--- agqt.ori/am/myqueue.h	Fri Jan  4 19:22:09 2002
+++ agqt/am/myqueue.h	Fri Aug  9 10:33:22 2002
@@ -3,6 +3,9 @@
 
 #include <queue>
 
+using std::queue;
+using std::priority_queue;
+
 void 			q_push		(char *);
 void			q_pop		();
 char *			q_front		();
diff -ur agqt.ori/gq/client.h agqt/gq/client.h
--- agqt.ori/gq/client.h	Wed May  1 04:21:16 2002
+++ agqt/gq/client.h	Fri Aug  9 10:34:22 2002
@@ -2,6 +2,10 @@
 #define CLIENT_H
 
 #include <queue>
+
+using std::queue;
+using std::priority_queue;
+
 #include "parse_xml.h"
 #include "server.h"
 #include "hashlist.h"
