diff -Nru3 dante-1.1.14/configure dante-patched/configure
--- dante-1.1.13/configure	2002-06-24 10:48:56.000000000 +0200
+++ dante-patched/configure	2003-04-13 19:07:21.000000000 +0200
@@ -16423,7 +16423,7 @@
 
 #look for PAM header and lib
 
-for ac_header in security/pam_appl.h
+for ac_header in pam/pam_appl.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 echo "$as_me:16429: checking for $ac_header" >&5
diff -Nru3 dante-1.1.14/include/common.h dante-patched/include/common.h
--- dante-1.1.14/include/common.h	2002-06-17 19:37:33.000000000 +0200
+++ dante-patched/include/common.h	2003-04-13 19:08:06.000000000 +0200
@@ -175,7 +175,7 @@
 #include <ifaddrs.h>
 #endif /* HAVE_IFADDRS_H */
 #if HAVE_PAM
-#include <security/pam_appl.h>
+#include <pam/pam_appl.h>
 #endif /* HAVE_PAM */
 
 #include "yacconfig.h"
diff -Nru3 dante-1.1.14/ltmain.sh dante-patched/ltmain.sh
--- dante-1.1.4/ltmain.sh	2001-12-12 12:23:20.000000000 +0100
+++ dante-patched/ltmain.sh	2003-04-13 18:57:05.000000000 +0200
@@ -2899,6 +2899,16 @@
 	  $run eval '(cd $output_objdir && $rm ${realname}U && $mv $realname ${realname}U)' || exit $?
 	fi
 
+       tmp_deplibs=
+       for test_deplib in $deplibs; do
+               case " $convenience " in
+               *" $test_deplib "*) ;;
+               *)
+                       tmp_deplibs="$tmp_deplibs $test_deplib"
+                       ;;
+               esac
+       done
+       deplibs="$tmp_deplibs" 
 	# Do each of the archive commands.
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
