diff -ruN fidelio-1.0b1/src/connection.c fidelio-1.0b1.new/src/connection.c
--- fidelio-1.0b1/src/connection.c	2003-04-10 09:57:23.000000000 -0400
+++ fidelio-1.0b1.new/src/connection.c	2003-04-10 10:03:21.000000000 -0400
@@ -25,6 +25,12 @@
 #include <gnome.h>
 #include <glib.h>
 
+#ifndef _BSD_SOCKLEN_T_
+#define _BSD_SOCKLEN_T_ unsigned int
+#endif
+
+#include <sys/socket.h>
+
 #include "network.h"
 #include "protocol.h"
 #include "messages.h"
