diff -ruN fidelio-1.0b1/fidelio.default fidelio-1.0b1.new/fidelio.default
--- fidelio-1.0b1/fidelio.default	2000-09-10 18:00:50.000000000 -0400
+++ fidelio-1.0b1.new/fidelio.default	2003-04-26 13:30:47.000000000 -0400
@@ -6,18 +6,19 @@
   </bookmarks>
   <trackers>
     <t name="tracked.group.org" host="tracked.group.org"/>
-    <t name="public.hltracker.com" host="public.hltracker.com"/>
-    <t name="science.hltracker.com" host="science.hltracker.com"/>
-    <t name="travel.hltracker.com" host="travel.hltracker.com"/>
-    <t name="poland.hltracker.com" host="poland.hltracker.com"/>
-    <t name="vertigobox.dhs.org" host="vertigobox.dhs.org"/>
-    <t name="japan.hltracker.com" host="japan.hltracker.com"/>
-    <t name="saliconhub.dhs.org" host="saliconhub.dhs.org"/>
-    <t name="saga-city.com" host="saga-city.com"/>
-    <t name="cyromgdg4.dyndns.org" host="cyromgdg4.dyndns.org"/>
-    <t name="games.hltracker.com" host="games.hltracker.com"/>
-    <t name="teamjesus.dyn.dhs.org" host="teamjesus.dyn.dhs.org"/>
-    <t name="technology.hltracker.com" host="technology.hltracker.com"/>
-    <t name="science.hltracker.com" host="science.hltracker.com"/>
+    <t name="sunwave" host="tracker.sunwave.com"/>
+    <t name="revolution" host="revolution.frenzy.ca"/>
+    <t name="dmp" host="dmp.fimble.com"/>
+    <t name="freefilebackup" host="tracker.freefilebackup.com"/>
+    <t name="french tracker" host="frenchtracker.mine.nu"/>
+    <t name="Sample Tracker" host="samples.dyndns.org"/>
+    <t name="saddle" host="saddle.dyndns.org"/>
+    <t name="pltracker.dyndns.org" host="pltracker.dyndns.org"/>
+    <t name="lovetrain.nu" host="lovetrain.nu"/>
+    <t name="noonane" host="noonane.mine.nu"/>
+    <t name="Anime Heaven ^_^" host="animeparadise.mine.nu"/>
+    <t name="Tracker for LSD Servers, Only Free Downloads + Guest Access" host="lsd.dyndns.biz"/>
+    <t name="hljapan" host="hljapan.ns.tc"/>
+    <t name="Admin Tracker" host="admintracker.nedline.com"/>
   </trackers>
 </fidelio>
diff -ruN fidelio-1.0b1/src/connection.c fidelio-1.0b1.new/src/connection.c
--- fidelio-1.0b1/src/connection.c	2003-04-10 09:57:23.000000000 -0400
+++ fidelio-1.0b1.new/src/connection.c	2003-04-10 10:03:21.000000000 -0400
@@ -25,6 +25,12 @@
 #include <gnome.h>
 #include <glib.h>
 
+#ifndef _BSD_SOCKLEN_T_
+#define _BSD_SOCKLEN_T_ unsigned int
+#endif
+
+#include <sys/socket.h>
+
 #include "network.h"
 #include "protocol.h"
 #include "messages.h"
