diff -ruN gaim-0.59.8/configure gaim-0.59.8-patched/configure
--- gaim-0.59.8/configure	2003-01-08 03:21:24.000000000 +0100
+++ gaim-0.59.8-patched/configure	2003-01-25 18:03:46.000000000 +0100
@@ -7581,7 +7581,6 @@
 # Prevent multiple expansion
 
 
-LIBTOOL="$LIBTOOL --silent"
 # Find a good install program.  We prefer a C program (faster),
 # so one script is as good as another.  But avoid the broken or
 # incompatible versions:
diff -ruN gaim-0.59.8/plugins/Makefile.in gaim-0.59.8-patched/plugins/Makefile.in
--- gaim-0.59.8/plugins/Makefile.in	2003-01-08 03:21:48.000000000 +0100
+++ gaim-0.59.8-patched/plugins/Makefile.in	2003-01-25 18:20:40.000000000 +0100
@@ -286,7 +286,7 @@
 	$(LIBTOOL) --mode=compile $(CC) $(CFLAGS) -c $< -o tmp$@.lo $(PLUGIN_CFLAGS)
 	$(LIBTOOL) --mode=link    $(CC) $(CFLAGS) -o libtmp$@.la -rpath $(plugindir) tmp$@.lo $(LIBS) $(LDFLAGS) -module -avoid-version $(PLUGIN_LIBS)
 	@rm -f tmp$@.lo tmp$@.o libtmp$@.la
-	@cp .libs/libtmp$@.so* $@
+	@mv .libs/libtmp$@.so $@
 	@rm -f .libs/libtmp$@.*
 
 @PLUGINS_TRUE@$(plugin_DATA): $(top_srcdir)/src/gaim.h
