diff -ruN giFT/autogen.macosx.sh giFT.new/autogen.macosx.sh
--- giFT/autogen.macosx.sh	Thu Aug  8 09:28:33 2002
+++ giFT.new/autogen.macosx.sh	Sat Nov  2 17:55:05 2002
@@ -3,7 +3,6 @@
 
 srcdir=`dirname $0`
 PKG_NAME="giFT"
-export LDFLAGS="-ldl -L/sw/lib -lltdl -L../OpenFT/.libs"
 
 DIE=0
 
@@ -50,26 +49,6 @@
   exit 1
 fi
 
-if test -z "$*"; then
-  echo "I am going to run \`configure' with the default arguments."
-  echo "If you wish to pass any other to it, please specify them on the"
-  echo \`$0\'" command line."
-  echo
-  1="ppc"
-  2="--disable-libdl"
-  3="--with-zlib=/sw"
-  echo "Default is:"
-  echo "./configure $*"
-  echo
-else
-  echo "**Warning**: To compile giFT on Mac OS X you must specify"
-  echo "\"ppc\" as the host and "--disable-libdl" as parameters"
-  echo "for configure. Otherwise giFT will not compile."
-  echo
-  echo "Parameters are:"
-  echo "./configure $*"
-fi
-
 case $CC in
 xlc )
   am_opt=--include-deps;;
@@ -92,12 +71,13 @@
         ##  echo "**Warning**: No such directory \`$k'.  Ignored."
         fi
       done
-      if grep "^AM_PROG_LIBTOOL" configure.ac >/dev/null; then
         ln -s configure.ac configure.in 
         echo "Running glibtoolize..."
+        glibtoolize --automake --copy|| exit 1
+	 test -f ltmain.sh || \
+		{ echo "libtoolize failed to generate ltmain.sh" 2>&1; exit 1; }
         glibtoolize --force --copy
         rm -f configure.in
-      fi
       echo "Running aclocal $aclocalinclude ..."
       aclocal $aclocalinclude
       if grep "^AM_CONFIG_HEADER" configure.ac >/dev/null; then
@@ -111,14 +91,3 @@
     )
   fi
 done
-
-#conf_flags="--enable-maintainer-mode --enable-compile-warnings"
-#--enable-iso-c
-
-if test x$NOCONFIGURE = x; then
-  echo Running $srcdir/configure $conf_flags "$@" ...
-  $srcdir/configure $conf_flags "$@" \
-  && echo Now type \`make\' to compile $PKG_NAME
-else
-  echo Skipping configure process.
-fi
diff -ruN giFT/configure.ac giFT.new/configure.ac
--- giFT/configure.ac	Sat Nov  2 17:06:45 2002
+++ giFT.new/configure.ac	Sat Nov  2 17:09:10 2002
@@ -471,7 +471,7 @@
 dnl Check for the presence of ZLIB libraries and headers
 dnl ###########################################################################
 
-ZLIB_VER=1.1.4
+ZLIB_VER=1.1.3
 
 dnl Default to compiler & linker defaults for files & libraries.
 OPT_ZLIB=
diff -ruN giFT/etc/gift.conf giFT.new/etc/gift.conf
--- giFT/etc/gift.conf	Fri Oct 11 22:53:12 2002
+++ giFT.new/etc/gift.conf	Sat Nov  2 17:09:10 2002
@@ -56,7 +56,7 @@
 #
 # Default: none
 #
-plugins = libOpenFT.so
+plugins = libOpenFT.dylib
 
 ###############################################################################
 # DOWNLOAD CONTROLS
diff -ruN giFT/ltmain.patch giFT.new/ltmain.patch
--- giFT/ltmain.patch	Wed Dec 31 19:00:00 1969
+++ giFT.new/ltmain.patch	Sat Nov  2 17:40:23 2002
@@ -0,0 +1,62 @@
+--- ltmain.sh.orig	2002-06-22 21:27:18.000000000 -0400
++++ ltmain.sh	2002-06-22 21:27:52.000000000 -0400
+@@ -1739,7 +1739,7 @@
+ 
+ 	  if test "$installed" = no; then
+ 	    notinst_deplibs="$notinst_deplibs $lib"
+-	    need_relink=yes
++	    need_relink=no
+ 	  fi
+ 
+ 	  if test -n "$old_archive_from_expsyms_cmds"; then
+@@ -1884,7 +1884,11 @@
+ 	      add="-l$name"
+ 	    else
+ 	      # We cannot seem to hardcode it, guess we'll fake it.
+-	      add_dir="-L$libdir"
++	      if test "X$installed" = Xyes; then
++		add_dir="-L$libdir"
++	      else
++		add_dir="-L$DESTDIR$libdir"
++	      fi
+ 	      add="-l$name"
+ 	    fi
+ 
+@@ -4146,12 +4150,21 @@
+ 	esac
+ 
+ 	# Add the libdir to current_libdirs if it is the destination.
++	DESTDIR=
+ 	if test "X$destdir" = "X$libdir"; then
+ 	  case "$current_libdirs " in
+ 	  *" $libdir "*) ;;
+ 	  *) current_libdirs="$current_libdirs $libdir" ;;
+ 	  esac
+ 	else
++	  case "$destdir" in
++	    *"$libdir")
++	      DESTDIR=`$echo "$destdir" | sed -e 's!'"$libdir"'$!!'`
++	      if test "X$destdir" != "X$DESTDIR$libdir"; then
++		DESTDIR=
++	      fi
++	      ;;
++	  esac
+ 	  # Note the libdir as a future libdir.
+ 	  case "$future_libdirs " in
+ 	  *" $libdir "*) ;;
+@@ -4165,6 +4178,7 @@
+ 
+ 	if test -n "$relink_command"; then
+ 	  $echo "$modename: warning: relinking \`$file'" 1>&2
++	  export DESTDIR
+ 	  $show "$relink_command"
+ 	  if $run eval "$relink_command"; then :
+ 	  else
+@@ -4172,6 +4186,7 @@
+ 	    continue
+ 	  fi
+ 	fi
++	unset DESTDIR
+ 
+ 	# See the names of the shared library.
+ 	set dummy $library_names
diff -ruN giFT/src/Makefile.am giFT.new/src/Makefile.am
--- giFT/src/Makefile.am	Sat Oct 19 21:32:26 2002
+++ giFT.new/src/Makefile.am	Sat Nov  2 17:09:10 2002
@@ -52,6 +52,7 @@
 
 giFT_LDFLAGS = \
 	$(GIFT_LDFLAGS) \
+	-L$(top_builddir)/lib/.libs \
 	$(IMAGEMAGICK_LDFLAGS) \
 	-export-dynamic
 
