diff -ruN giFT/FastTrack/Makefile.am giFT.new/FastTrack/Makefile.am
--- giFT/FastTrack/Makefile.am	1969-12-31 19:00:00.000000000 -0500
+++ giFT.new/FastTrack/Makefile.am	2003-05-20 22:31:18.000000000 -0400
@@ -0,0 +1,36 @@
+AM_CFLAGS = $(GIFT_CFLAGS)
+
+INCLUDES = -I$(top_srcdir)/lib \
+	   -I$(top_srcdir)/src \
+	   -I$(top_srcdir)/plugin \
+	   -I$(top_srcdir)/FastTrack/crypt
+# libdir = $(plugindir)
+
+#
+# The rest of this file is conditional on whether we're 
+# compiling the gnutella plugin at all
+#
+
+plugin_LTLIBRARIES = libFastTrack.la
+
+libFastTrack_la_SOURCES = \
+fst_download.c	fst_download.h \
+fst_fasttrack.c	fst_fasttrack.h \
+fst_hash.c	fst_hash.h \
+fst_http.c	fst_http.h \
+fst_meta.c	fst_meta.h \
+fst_node.c	fst_node.h \
+fst_packet.c	fst_packet.h \
+fst_search.c	fst_search.h \
+fst_session.c	fst_session.h \
+fst_stats.c	fst_stats.h \
+fst_utils.c	fst_utils.h \
+md5.c		md5.h \
+crypt/enc_type_1.c \
+crypt/enc_type_2.c \
+crypt/enc_type_20.c \
+crypt/fst_crypt.c	crypt/fst_crypt.h
+
+libFastTrack_la_LIBADD = \
+	$(top_builddir)/lib/libgiFT.la \
+	$(top_builddir)/plugin/libgiFTproto.la
 
diff -ruN giFT/Makefile.am giFT.new/Makefile.am
--- giFT/Makefile.am	2003-05-20 19:51:25.000000000 -0400
+++ giFT.new/Makefile.am	2003-05-20 21:22:41.000000000 -0400
@@ -1,4 +1,4 @@
-SUBDIRS = lib plugin OpenFT Gnutella src data
+SUBDIRS = lib plugin OpenFT Gnutella FastTrack src data
 
 man_MANS = giFT.1 gift.conf.5 OpenFT.conf.5

diff -ruN giFT/configure.ac giFT.new/configure.ac
--- giFT/configure.ac	2003-05-18 20:50:38.000000000 -0400
+++ giFT.new/configure.ac	2003-05-20 21:45:18.000000000 -0400
@@ -37,6 +37,9 @@
 AH_VERBATIM([PLUGIN_OPENFT], [
 #undef PLUGIN_OPENFT
 ])
+AH_VERBATIM([PLUGIN_FASTTRACK], [
+#undef PLUGIN_FASTTRACK
+])
 AH_VERBATIM([PLUGIN_GNUTELLA], [
 #undef PLUGIN_GNUTELLA
 ])
@@ -241,6 +244,8 @@
 AM_CONDITIONAL(DAEMON,      test x$enable_daemon   = xyes)
 AC_ARG_ENABLE(openft,      [  --disable-openft        disable compiling the OpenFT plugin],,[enable_openft=yes])
 AM_CONDITIONAL(OPENFT,      test x$enable_openft   = xyes)
+AC_ARG_ENABLE(fasttrack,    [  --enable-fasttrack       enable compiling the FastTrack plugin],[enable_fasttrack=yes],[enable_fasttrack=no])
+AM_CONDITIONAL(FASTTRACK,    test x$enable_fasttrack = xyes)
 AC_ARG_ENABLE(gnutella,    [  --enable-gnutella       enable compiling the Gnutella plugin],[enable_gnutella=yes],[enable_gnutella=no])
 AM_CONDITIONAL(GNUTELLA,    test x$enable_gnutella = xyes)
 AC_ARG_ENABLE(perl,        [  --enable-perl           enable perl support],[enable_perl=yes],[enable_perl=no])
@@ -268,6 +273,11 @@
   AC_SUBST(PLUGIN_GNUTELLA)
 fi
 
+if test x$enable_fasttrack = xyes; then
+  AC_DEFINE(PLUGIN_FASTTRACK)
+  AC_SUBST(PLUGIN_FASTTRACK)
+fi
+
 dnl ###########################################################################
 dnl Overrides
 dnl ###########################################################################
@@ -331,6 +341,10 @@
     GIFT_LIBS="$GIFT_LIBS \$(top_builddir)/Gnutella/libGnutella.la"
 fi
 
+if test x$enable_fasttrack = xyes; then
+    GIFT_LIBS="$GIFT_LIBS \$(top_builddir)/FastTrack/libFastTrack.la"
+fi
+
 GIFT_LDFLAGS=""
 
 if test x$enable_ltdl = xyes; then
@@ -644,6 +658,7 @@
 OpenFT/Makefile
 OpenFT/proto/Makefile
 Gnutella/Makefile
+FastTrack/Makefile
 data/Makefile
 data/OpenFT/Makefile
 data/Gnutella/Makefile
@@ -670,6 +685,7 @@
 echo "--------------- protocols ---"
 echo
 echo "OpenFT..................: $enable_openft"
+echo "FastTrack...............: $enable_fasttrack"
 echo "Gnutella................: $enable_gnutella"
 echo
 echo "-------------- extensions ---"
diff -ruN giFT/etc/gift.conf giFT.new/etc/gift.conf
--- giFT/etc/gift.conf	2003-05-18 20:50:41.000000000 -0400
+++ giFT.new/etc/gift.conf	2003-05-20 21:47:15.000000000 -0400
@@ -1,4 +1,5 @@
 # gift.conf
+
 ###############################################################################
 
 ###############################################################################
@@ -71,9 +72,9 @@
 #
 #  Protocol names are case sensitive.
 #
-# For example, to use the OpenFT and Gnutella protocols use:
+# For example, to use the OpenFT, Gnutella and FastTrack protocols use:
 #
-#  plugins = OpenFT:Gnutella
+#  plugins = OpenFT:Gnutella:FastTrack
 #
 # Default: none
 #
diff -ruN giFT/src/Makefile.am giFT.new/src/Makefile.am
--- giFT/src/Makefile.am	2003-05-03 13:03:59.000000000 -0400
+++ giFT.new/src/Makefile.am	2003-05-20 19:54:45.000000000 -0400
@@ -60,6 +60,7 @@
 
 giFT_LDFLAGS = \
 	$(GIFT_LDFLAGS) \
+	-L$(top_builddir)/lib/.libs \
 	$(IMAGEMAGICK_LDFLAGS) \
 	-export-dynamic
 
diff -ruN giFT/src/main.c giFT.new/src/main.c
--- giFT/src/main.c	2003-05-18 20:50:42.000000000 -0400
+++ giFT.new/src/main.c	2003-05-21 10:44:09.000000000 -0400
@@ -241,6 +241,9 @@
 	string_appendf (s, "  OpenFT %i.%i.%i-%i\n", OPENFT_MAJOR, OPENFT_MINOR,
 	                OPENFT_MICRO, OPENFT_REV);
 # endif /* PLUGIN_OPENFT */
+# ifdef PLUGIN_FASTTRACK
+	string_appendf (s, "  FastTrack Enabled\n");
+# endif /* PLUGIN_FASTTRACK */
 # ifdef PLUGIN_GNUTELLA
 	string_appendf (s, "  Gnutella %s\n", GT_VERSION);
 # endif /* PLUGIN_GNUTELLA */
diff -ruN giFT/src/plugin.c giFT.new/src/plugin.c
--- giFT/src/plugin.c	2003-05-18 20:50:42.000000000 -0400
+++ giFT.new/src/plugin.c	2003-05-20 21:54:49.000000000 -0400
@@ -35,6 +35,9 @@
 # ifdef PLUGIN_OPENFT
 int OpenFT_init (Protocol *p);
 # endif /* PLUGIN_OPENFT */
+# ifdef PLUGIN_FASTTRACK
+int FastTrack_init (Protocol *p);
+# endif /* PLUGIN_FASTTRACK */
 # ifdef PLUGIN_GNUTELLA
 int Gnutella_init (Protocol *p);
 # endif /* PLUGIN_GNUTELLA */
@@ -224,6 +227,10 @@
 	if (!strcmp (pname, "OpenFT"))
 		init = OpenFT_init;
 # endif /* PLUGIN_OPENFT */
+# ifdef PLUGIN_FASTTRACK
+	if (!strcmp (pname, "FastTrack"))
+		init = FastTrack_init;
+# endif /* PLUGIN_FASTTRACK */
 # ifdef PLUGIN_GNUTELLA
 	if (!strcmp (pname, "Gnutella"))
 		init = Gnutella_init;
