diff -ruN giFT/autogen.sh giFT.new/autogen.sh
--- giFT/autogen.sh	2002-11-02 17:06:45.000000000 -0500
+++ giFT.new/autogen.sh	2003-01-21 21:45:09.000000000 -0500
@@ -9,7 +9,7 @@
 : ${AUTOHEADER=autoheader}
 : ${AUTOMAKE=automake}
 : ${ACLOCAL=aclocal}
-: ${LIBTOOLIZE=libtoolize}
+: ${LIBTOOLIZE=glibtoolize}
 
 srcdir=`dirname $0`
 test -z "$srcdir" && srcdir=.
@@ -57,6 +57,6 @@
 ) || exit 1
 
 #conf_flags="--enable-maintainer-mode --enable-compile-warnings"
-echo Running $srcdir/configure $conf_flags "$@" ...
-$srcdir/configure --cache-file=config.cache $conf_flags "$@" && \
-  echo "Now type \`make' to compile giFT."
+#echo Running $srcdir/configure $conf_flags "$@" ...
+#$srcdir/configure --cache-file=config.cache $conf_flags "$@" && \
+#  echo "Now type \`make' to compile giFT."
diff -ruN giFT/configure.ac giFT.new/configure.ac
--- giFT/configure.ac	2003-01-21 21:36:06.000000000 -0500
+++ giFT.new/configure.ac	2003-01-21 21:45:09.000000000 -0500
@@ -477,7 +477,7 @@
 dnl Check for the presence of ZLIB libraries and headers
 dnl ###########################################################################
 
-ZLIB_VER=1.1.4
+ZLIB_VER=1.1.3
 
 dnl Default to compiler & linker defaults for files & libraries.
 OPT_ZLIB=
diff -ruN giFT/etc/gift.conf giFT.new/etc/gift.conf
--- giFT/etc/gift.conf	2003-01-21 21:36:38.000000000 -0500
+++ giFT.new/etc/gift.conf	2003-01-21 21:45:09.000000000 -0500
@@ -53,10 +53,11 @@
 #
 # *nix users: use the extension '.so', or whatever is appropriate for your OS.
 # Windows users: use the extension '.dll'.
+# Mac users: use the extension '.dylib'.
 #
 # Default: none
 #
-plugins = libOpenFT.so
+plugins = libOpenFT.dylib
 
 ###############################################################################
 # DOWNLOAD CONTROLS
diff -ruN giFT/ltmain.patch giFT.new/ltmain.patch
--- giFT/ltmain.patch	1969-12-31 19:00:00.000000000 -0500
+++ giFT.new/ltmain.patch	2003-01-21 21:45:10.000000000 -0500
@@ -0,0 +1,62 @@
+--- ltmain.sh.orig	2002-06-22 21:27:18.000000000 -0400
++++ ltmain.sh	2002-06-22 21:27:52.000000000 -0400
+@@ -1739,7 +1739,7 @@
+ 
+ 	  if test "$installed" = no; then
+ 	    notinst_deplibs="$notinst_deplibs $lib"
+-	    need_relink=yes
++	    need_relink=no
+ 	  fi
+ 
+ 	  if test -n "$old_archive_from_expsyms_cmds"; then
+@@ -1884,7 +1884,11 @@
+ 	      add="-l$name"
+ 	    else
+ 	      # We cannot seem to hardcode it, guess we'll fake it.
+-	      add_dir="-L$libdir"
++	      if test "X$installed" = Xyes; then
++		add_dir="-L$libdir"
++	      else
++		add_dir="-L$DESTDIR$libdir"
++	      fi
+ 	      add="-l$name"
+ 	    fi
+ 
+@@ -4146,12 +4150,21 @@
+ 	esac
+ 
+ 	# Add the libdir to current_libdirs if it is the destination.
++	DESTDIR=
+ 	if test "X$destdir" = "X$libdir"; then
+ 	  case "$current_libdirs " in
+ 	  *" $libdir "*) ;;
+ 	  *) current_libdirs="$current_libdirs $libdir" ;;
+ 	  esac
+ 	else
++	  case "$destdir" in
++	    *"$libdir")
++	      DESTDIR=`$echo "$destdir" | sed -e 's!'"$libdir"'$!!'`
++	      if test "X$destdir" != "X$DESTDIR$libdir"; then
++		DESTDIR=
++	      fi
++	      ;;
++	  esac
+ 	  # Note the libdir as a future libdir.
+ 	  case "$future_libdirs " in
+ 	  *" $libdir "*) ;;
+@@ -4165,6 +4178,7 @@
+ 
+ 	if test -n "$relink_command"; then
+ 	  $echo "$modename: warning: relinking \`$file'" 1>&2
++	  export DESTDIR
+ 	  $show "$relink_command"
+ 	  if $run eval "$relink_command"; then :
+ 	  else
+@@ -4172,6 +4186,7 @@
+ 	    continue
+ 	  fi
+ 	fi
++	unset DESTDIR
+ 
+ 	# See the names of the shared library.
+ 	set dummy $library_names
diff -ruN giFT/src/Makefile.am giFT.new/src/Makefile.am
--- giFT/src/Makefile.am	2002-11-23 13:57:15.000000000 -0500
+++ giFT.new/src/Makefile.am	2003-01-21 21:45:10.000000000 -0500
@@ -56,6 +56,7 @@
 
 giFT_LDFLAGS = \
 	$(GIFT_LDFLAGS) \
+	-L$(top_builddir)/lib/.libs \
 	$(IMAGEMAGICK_LDFLAGS) \
 	-export-dynamic
 
