diff -ru ../gnomemeeting-0.98.0/configure gnomemeeting-0.98.0/configure
--- ../gnomemeeting-0.98.0/configure	Sun Jun 22 18:51:07 2003
+++ gnomemeeting-0.98.0/configure	Tue Jun 24 17:40:48 2003
@@ -9587,11 +9587,11 @@
     ;;
 
   darwin* )
-    H323_CFLAGS="-DP_MACOSX $ARCH_H323_CFLAGS -O2 -Wall -DNO_LONG_DOUBLE -DPHAS_TEMPLATES -DP_USE_PRAGMA -DPTRACING -DSTATIC_LIBS_USED"
-    H323_LIBS="-multiply_defined suppress -lh323_Darwin_ppc_r_s -lpt_Darwin_ppc_r_s -lssl -lcrypto -lX11 -framework AudioToolbox -framework CoreAudio -framework CoreServices"
-    LIBPT_FILE="libpt_Darwin_ppc_r_s.a"
-    LIBOPENH323_FILE="libh323_Darwin_ppc_r_s.a"
-    HAS_SPEEX="enabled"
+    H323_CFLAGS="-DP_MACOSX=66 $ARCH_H323_CFLAGS -O2 -Wall -DNO_LONG_DOUBLE -D_REENTRANT -fno-common -dynamic -DPHAS_TEMPLATES -DPTRACING -DSTATIC_LIBS_USED"
+    H323_LIBS="-multiply_defined suppress -lh323_Darwin_ppc_r_s -lpt_Darwin_ppc_r_s -lexpat -lssl -lcrypto -lX11 -framework AudioToolbox -framework CoreAudio -framework CoreServices"
+    LIBPT_FILE="libpt_Darwin_ppc_r_s.a"
+    LIBOPENH323_FILE="libh323_Darwin_ppc_r_s.a"
+    HAS_SPEEX="enabled"
     echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6
     gm_platform="macosx"
diff -ru ../gnomemeeting-0.98.0/configure.in gnomemeeting-0.98.0/configure.in
--- ../gnomemeeting-0.98.0/configure.in	Sun Jun 22 18:46:05 2003
+++ gnomemeeting-0.98.0/configure.in	Tue Jun 24 17:40:48 2003
@@ -102,11 +102,11 @@
     ;;
 
   darwin* )
-    H323_CFLAGS="-DP_MACOSX $ARCH_H323_CFLAGS -O2 -Wall -DNO_LONG_DOUBLE -DPHAS_TEMPLATES -DP_USE_PRAGMA -DPTRACING -DSTATIC_LIBS_USED"
-    H323_LIBS="-multiply_defined suppress -lh323_Darwin_ppc_r_s -lpt_Darwin_ppc_r_s -lssl -lcrypto -lX11 -framework AudioToolbox -framework CoreAudio -framework CoreServices"
-    LIBPT_FILE="libpt_Darwin_ppc_r_s.a"
-    LIBOPENH323_FILE="libh323_Darwin_ppc_r_s.a"
-    HAS_SPEEX="enabled"
+    H323_CFLAGS="-DP_MACOSX=66 $ARCH_H323_CFLAGS -O2 -Wall -DNO_LONG_DOUBLE -D_REENTRANT -fno-common -dynamic -DPHAS_TEMPLATES -DPTRACING -DSTATIC_LIBS_USED"
+    H323_LIBS="-multiply_defined suppress -lh323_Darwin_ppc_r_s -lpt_Darwin_ppc_r_s -lexpat -lssl -lcrypto -lX11 -framework AudioToolbox -framework CoreAudio -framework CoreServices"
+    LIBPT_FILE="libpt_Darwin_ppc_r_s.a"
+    LIBOPENH323_FILE="libh323_Darwin_ppc_r_s.a"
+    HAS_SPEEX="enabled"
     AC_MSG_RESULT([yes])
     gm_platform="macosx"
     ;;
diff -ru ../gnomemeeting-0.98.0/src/config.cpp gnomemeeting-0.98.0/src/config.cpp
--- ../gnomemeeting-0.98.0/src/config.cpp	Mon Jun 23 05:16:33 2003
+++ gnomemeeting-0.98.0/src/config.cpp	Tue Jun 24 17:40:48 2003
@@ -928,8 +928,8 @@
   int vol = 0;
   char *mixer = NULL;
   GmWindow *gw = NULL;
-#ifdef HAS_IXJ
   GMH323EndPoint *endpoint = MyApp->Endpoint ();
+#ifdef HAS_IXJ
   GMLid *lid = NULL;
 #endif
   
diff -ru ../gnomemeeting-0.98.0/src/endpoint.cpp gnomemeeting-0.98.0/src/endpoint.cpp
--- ../gnomemeeting-0.98.0/src/endpoint.cpp	Sun Jun 22 05:35:25 2003
+++ gnomemeeting-0.98.0/src/endpoint.cpp	Wed Jun 25 11:25:55 2003
@@ -175,7 +175,7 @@
   ils_client = new GMILSClient ();
 
   /* Use IPv6 address family by default if available. */
-#ifdef P_HAS_IPV6
+#if 0 // for Darwin
   if (PIPSocket::IsIpAddressFamilyV6Supported())
     PIPSocket::SetDefaultIpAddressFamilyV6();
 #endif
Only in gnomemeeting-0.98.0/src: gnomemeeting-config-tool
diff -ru ../gnomemeeting-0.98.0/src/gnomemeeting.cpp gnomemeeting-0.98.0/src/gnomemeeting.cpp
--- ../gnomemeeting-0.98.0/src/gnomemeeting.cpp	Sun Jun 22 18:46:14 2003
+++ gnomemeeting-0.98.0/src/gnomemeeting.cpp	Tue Jun 24 17:40:48 2003
@@ -126,7 +126,9 @@
     
     endpoint->ClearAllCalls (H323Connection::EndedByLocalUser, TRUE);
     endpoint->RemoveVideoGrabber (true);
+#ifdef HAS_IXJ
     endpoint->RemoveLid ();
+#endif
   }
   RemoveEndpoint ();
 
