diff -ruN lftp-2.6.4.orig/src/CmdExec.cc lftp-2.6.4/src/CmdExec.cc
--- lftp-2.6.4.orig/src/CmdExec.cc	Wed Jul 31 08:40:02 2002
+++ lftp-2.6.4/src/CmdExec.cc	Wed Feb 26 14:41:22 2003
@@ -369,7 +369,7 @@
 	       eprintf(_("cd ok, cwd=%s\n"),cwd?cwd:"~");
 	       cwd_history.Set(session,old_cwd);
 	       if(slot)
-		  ConnectionSlot::SetCwd(slot,cwd);
+		  ConnectionSlot::SetCwd(slot,session->GetCwd());
 	    }
 	    session->Close();
 	    exit_code=0;
diff -ruN lftp-2.6.4.orig/src/NetAccess.cc lftp-2.6.4/src/NetAccess.cc
--- lftp-2.6.4.orig/src/NetAccess.cc	Wed Dec 25 06:44:31 2002
+++ lftp-2.6.4/src/NetAccess.cc	Wed Feb 26 14:38:41 2003
@@ -22,6 +22,7 @@
 
 #include <config.h>
 
+#include <sys/types.h>
 #include <errno.h>
 #include <assert.h>
 #include <math.h>
