diff -ru net-snmp-5.0.7/Makefile.in net-snmp-5.0.7-patched/Makefile.in
--- net-snmp-5.0.7/Makefile.in	Tue Jan  7 03:16:00 2003
+++ net-snmp-5.0.7-patched/Makefile.in	Fri Jan 17 15:54:16 2003
@@ -53,7 +53,7 @@
 # local build rules
 #
 sedscript: sedscript.in include/net-snmp/net-snmp-config.h $(srcdir)/agent/mibgroup/mibdefs.h
-	$(CPP) -DPREFIX=$(prefix) -DLIBDIR=$(libdir) -DDATADIR=$(datadir) $(srcdir)/sedscript.in | egrep '^s[/#]' | sed 's/REMOVEME//g;s# */#/#g;s/ *#/#/g;s#/ *#/#g;s/# g/#g/;' > sedscript
+	$(CPP) -no-cpp-precomp -DPREFIX=$(prefix) -DLIBDIR=$(libdir) -DDATADIR=$(datadir) $(srcdir)/sedscript.in | egrep '^s[/#]' | sed 's/REMOVEME//g;s# */#/#g;s/ *#/#/g;s#/ *#/#g;s/# g/#g/;' > sedscript
 
 EXAMPLE.conf: sedscript EXAMPLE.conf.def
 	$(SED) -f sedscript $(srcdir)/EXAMPLE.conf.def > EXAMPLE.conf
diff -ru net-snmp-5.0.7/agent/mibgroup/mibII/tcp.c net-snmp-5.0.7-patched/agent/mibgroup/mibII/tcp.c
--- net-snmp-5.0.7/agent/mibgroup/mibII/tcp.c	Sat May 18 02:05:13 2002
+++ net-snmp-5.0.7-patched/agent/mibgroup/mibII/tcp.c	Sun Jan 19 16:05:39 2003
@@ -101,6 +101,7 @@
 #include <netinet/tcp.h>
 #endif
 #if HAVE_NETINET_TCPIP_H
+#include <netinet/tcp_var.h>
 #include <netinet/tcpip.h>
 #endif
 #if HAVE_NETINET_TCP_TIMER_H
diff -ru net-snmp-5.0.7/configure net-snmp-5.0.7-patched/configure
--- net-snmp-5.0.7/configure	Tue Jan 14 13:56:39 2003
+++ net-snmp-5.0.7-patched/configure	Fri Jan 17 15:56:07 2003
@@ -4269,11 +4269,11 @@
     ;;
 
   darwin* | rhapsody*)
-    allow_undefined_flag='-undefined suppress'
+    allow_undefined_flag='-flat_namespace -undefined suppress'
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linkopts -install_name $rpath/$soname $(test -n "$verstring" -a x$verstring != x0.0 && echo $verstring)'
+    archive_cmds='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linkopts $(test .$module != .yes && echo -install_name $rpath/$soname $tmp_verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -ru net-snmp-5.0.7/ltmain.sh net-snmp-5.0.7-patched/ltmain.sh
--- net-snmp-5.0.7/ltmain.sh	Sat Apr 20 17:29:58 2002
+++ net-snmp-5.0.7-patched/ltmain.sh	Fri Jan 17 15:57:41 2003
@@ -2858,6 +2858,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	  if test "x$verstring" = "x0.0"; then
+            tmp_verstring=
+          else
+            tmp_verstring="$verstring"
+          fi
 	  eval cmds=\"$archive_cmds\"
 	fi
 	IFS="${IFS= 	}"; save_ifs="$IFS"; IFS='~'
