diff -ru slrn-0.9.7.2/configure slrn-0.9.7.2-patched/configure
--- slrn-0.9.7.2/configure	Wed Aug 15 00:00:36 2001
+++ slrn-0.9.7.2-patched/configure	Mon Sep  3 20:53:14 2001
@@ -2099,7 +2099,7 @@
                  /usr/share/lib/terminfo \
 		 /usr/local/lib/terminfo"
 
-TERMCAP=-ltermcap
+TERMCAP=
 
 echo $ac_n "checking for Terminfo""... $ac_c" 1>&6
 echo "configure:2106: checking for Terminfo" >&5
Only in slrn-0.9.7.2-patched: configure~
