diff -uNr squid-2.5.STABLE3/src/cf.data.pre squid-2.5.STABLE3-new/src/cf.data.pre
--- squid-2.5.STABLE3/src/cf.data.pre	Wed May 21 10:34:38 2003
+++ squid-2.5.STABLE3-new/src/cf.data.pre	Mon Aug 18 22:22:36 2003
@@ -2077,6 +2077,9 @@
 #
 # INSERT YOUR OWN RULE(S) HERE TO ALLOW ACCESS FROM YOUR CLIENTS
 
+# Allow proxy connections from localhost
+http_access allow localhost
+
 # Exampe rule allowing access from your local networks. Adapt
 # to list your (internal) IP networks from where browsing should
 # be allowed
@@ -2338,7 +2341,7 @@
 NAME: visible_hostname
 TYPE: string
 LOC: Config.visibleHostname
-DEFAULT: none
+DEFAULT: localhost
 DOC_START
 	If you want to present a special hostname in error messages, etc,
 	then define this.  Otherwise, the return value of gethostname()
