diff -ur whois-4.5.29-dist/Makefile whois-4.5.29/Makefile
--- whois-4.5.29-dist/Makefile	2002-06-22 06:47:33.000000000 -0400
+++ whois-4.5.29/Makefile	2002-08-28 18:06:26.000000000 -0400
@@ -35,7 +35,7 @@
 install: whois
 	install -m 0755 whois $(BASEDIR)$(prefix)/bin/
 	strip $(BASEDIR)$(prefix)/bin/whois
-	install -m 0644 whois.1 $(BASEDIR)$(prefix)/man/man1/
+	install -m 0644 whois.1 $(BASEDIR)$(prefix)/share/man/man1/
 	cd po && $(MAKE) $@
 
 
diff -ur whois-4.5.29-dist/po/Makefile whois-4.5.29/po/Makefile
--- whois-4.5.29-dist/po/Makefile	2001-09-23 12:26:04.000000000 -0400
+++ whois-4.5.29/po/Makefile	2002-08-28 18:07:37.000000000 -0400
@@ -1,5 +1,5 @@
 INSTALL= /usr/bin/install -c
-INSTALLNLSDIR=$(BASEDIR)/usr/share/locale
+INSTALLNLSDIR=$(BASEDIR)$(prefix)/share/locale
 
 PACKAGE = whois
 
