--- wwwoffle-2.7d/doc/Makefile.in	2002-07-28 15:20:24.000000000 +1000
+++ wwwoffle-2.7d-patched/doc/Makefile.in	2002-07-30 20:17:11.000000000 +1000
@@ -21,8 +21,8 @@
 # The installation locations
 
 prefix=@prefix@
-mandir=$(DESTDIR)@mandir@
-docdir=$(DESTDIR)${prefix}/doc/wwwoffle
+mandir=$(DESTDIR)${prefix}/share/man
+docdir=$(DESTDIR)${prefix}/share/doc/wwwoffle
 
 SPOOLDIR=@SPOOLDIR@
 CONFDIR=@CONFDIR@
