diff -ru xchat-2.0.0/configure xchat-2.0.0-patched/configure
--- xchat-2.0.0/configure	2003-02-09 05:37:50.000000000 +0100
+++ xchat-2.0.0-patched/configure	2003-02-13 04:03:52.000000000 +0100
@@ -12059,7 +12059,7 @@
 fi
 
 
-GUI_LIBS="$GUI_LIBS $GTK_LIBS"
+GUI_LIBS="$GUI_LIBS $GTK_LIBS -L/usr/X11R6/lib -lX11"
 GUI_CFLAGS="$GUI_CFLAGS $GTK_CFLAGS"
 
 gnome=no
@@ -12157,7 +12157,7 @@
 echo "${ECHO_T}ok" >&6
 
 		oldLIBS=$LIBS
-		LIBS="$LIBS $PERL_LDFLAGS"
+		LIBS="$PERL_LDFLAGS $LIBS"
 
 for ac_func in eval_pv
 do
@@ -12941,9 +12941,9 @@
 		# we just need the -Wl,--export-dynamic, but not -lgmodule-2.0
 		RDYNAMIC_FLAGS="`$pkgconfigpath gmodule-2.0 --libs | $sedpath 's/ -lgmodule-2.0//'`"
 		LIBS="$LIBS $RDYNAMIC_FLAGS"
-		if test "x$GCC" = "xyes"; then
-			GUI_LIBS="$GUI_LIBS -Wl,--version-script,\$(srcdir)/../version-script"
-		fi
+#		if test "x$GCC" = "xyes"; then
+#			GUI_LIBS="$GUI_LIBS -Wl,--version-script,\$(srcdir)/../version-script"
+#		fi
 	else
 		plugin=no
 	fi
