diff -Naur cvs2svn-2.0.1-orig/svntest/main.py cvs2svn-2.0.1/svntest/main.py
--- cvs2svn-2.0.1-orig/svntest/main.py	2007-10-05 07:35:27.000000000 +0200
+++ cvs2svn-2.0.1/svntest/main.py	2007-10-05 07:42:49.000000000 +0200
@@ -325,21 +325,24 @@
   exit_code = 0
 
   if platform_with_os_wait:
-    pid, wait_code = os.wait()
+    try:
+      pid, wait_code = os.wait()
 
-    if os.WIFSIGNALED(wait_code):
-      exit_signal = os.WTERMSIG(wait_code)
-      sys.stdout.write("".join(stdout_lines))
-      sys.stderr.write("".join(stderr_lines))
-      if verbose_mode:
-        # show the whole path to make it easier to start a debugger
-        sys.stderr.write("CMD: %s terminated by signal %d\n"
-                         % (command, exit_signal))
-      raise SVNProcessTerminatedBySignal
-    else:
-      exit_code = os.WEXITSTATUS(wait_code)
-      if exit_code and verbose_mode:
-        sys.stderr.write("CMD: %s exited with %d\n" % (command, exit_code))
+      if os.WIFSIGNALED(wait_code):
+        exit_signal = os.WTERMSIG(wait_code)
+        sys.stdout.write("".join(stdout_lines))
+        sys.stderr.write("".join(stderr_lines))
+        if verbose_mode:
+          # show the whole path to make it easier to start a debugger
+          sys.stderr.write("CMD: %s terminated by signal %d\n"
+                           % (command, exit_signal))
+        raise SVNProcessTerminatedBySignal
+      else:
+        exit_code = os.WEXITSTATUS(wait_code)
+        if exit_code and verbose_mode:
+          sys.stderr.write("CMD: %s exited with %d\n" % (command, exit_code))
+    except OSError:
+      pass
 
   return exit_code, stdout_lines, stderr_lines
 
diff -Naur cvs2svn-2.0.1_orig/run-tests.py cvs2svn-2.0.1/run-tests.py
--- cvs2svn-2.0.1_orig/run-tests.py	2007-10-09 22:38:22.000000000 +0200
+++ cvs2svn-2.0.1/run-tests.py	2007-10-09 22:40:22.000000000 +0200
@@ -2696,7 +2696,12 @@
     raise Failure()
   if kk_ic != kk_cvs:
     raise Failure()
-  if kv_ic != kv_cvs:
+
+  # The date format changed between cvs and co ('/' instead of '-').
+  # Accept either one:
+  date_substitution_re = re.compile(r' ([0-9]*)-([0-9]*)-([0-9]*) ')
+  if kv_ic != kv_cvs \
+         and date_substitution_re.sub(r' \1/\2/\3 ', kv_ic) != kv_cvs:
     raise Failure()
 
 
