diff -Nurd -x'*~' dbus-glib-0.84.orig/dbus/Makefile.in dbus-glib-0.84/dbus/Makefile.in
--- dbus-glib-0.84.orig/dbus/Makefile.in	2010-01-27 13:47:43.000000000 -0500
+++ dbus-glib-0.84/dbus/Makefile.in	2010-01-27 16:29:06.000000000 -0500
@@ -401,7 +401,7 @@
         dbus-glib-error-enum.h
 
 libdbus_glibdir = $(includedir)/dbus-1.0/dbus
-libdbus_glib_1_la_LIBADD = $(DBUS_LIBS) $(DBUS_GLIB_LIBS)
+libdbus_glib_1_la_LIBADD = $(DBUS_LIBS) $(DBUS_GLIB_LIBS) $(INTLLIBS)
 libdbus_glib_1_la_LDFLAGS = -export-symbols-regex "^[^_].*" -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -no-undefined
 
 # convenience lib used here and by dbus-viewer
diff -Nurd -x'*~' dbus-glib-0.84.orig/doc/reference/Makefile.in dbus-glib-0.84/doc/reference/Makefile.in
--- dbus-glib-0.84.orig/doc/reference/Makefile.in	2010-01-27 13:47:45.000000000 -0500
+++ dbus-glib-0.84/doc/reference/Makefile.in	2010-01-27 16:30:34.000000000 -0500
@@ -612,7 +612,7 @@
 	cd $(srcdir)/html && gtkdoc-mkhtml $(mkhtml_options) $(MKHTML_OPTIONS) $(DOC_MODULE) ../$(DOC_MAIN_SGML_FILE)
 	test "x$(HTML_IMAGES)" = "x" || ( cd $(srcdir) && cp $(HTML_IMAGES) html )
 	@echo 'gtk-doc: Fixing cross-references'
-	cd $(srcdir) && gtkdoc-fixxref --module-dir=html --html-dir=$(HTML_DIR) $(FIXXREF_OPTIONS)
+	cd $(srcdir) && gtkdoc-fixxref --module=$(DOC_MODULE) --module-dir=html --html-dir=$(HTML_DIR) $(FIXXREF_OPTIONS)
 	touch html-build.stamp
 
 ##############
