diff -ruN etm-orig/Makefile etm/Makefile
--- etm-orig/Makefile	Thu Apr 10 16:15:36 1997
+++ etm/Makefile	Tue Jan 16 16:55:42 2007
@@ -65,7 +65,7 @@
  CONFIGMAJORRELEASE = 2
  CONFIGMINORRELEASE = 10
       CONFIGRELEASE = $(CONFIGMAJORRELEASE).$(CONFIGMINORRELEASE)
-      CONFIGROOTDIR = /usr/local/lib/config
+      CONFIGROOTDIR = @PREFIX@/lib/config
 
 # Project identfication - name and release level
 
@@ -83,7 +83,7 @@
            TMPDIR = /tmp
      SPOOLROOTDIR = /usr/spool
 
-     LOCALROOTDIR = /usr/local
+     LOCALROOTDIR = @PREFIX@
       LOCALBINDIR = $(LOCALROOTDIR)/bin
    LOCALUSRLIBDIR = $(LOCALROOTDIR)/lib
   LOCALLINTLIBDIR = $(LOCALUSRLIBDIR)/lint
@@ -91,7 +91,7 @@
       LOCALETCDIR = $(LOCALROOTDIR)/etc
  LOCALINCLUDEROOT = $(LOCALROOTDIR)/include
 
-          MANROOT = /usr/local/man
+          MANROOT = @PREFIX@/share/man
     MANSOURCEPATH = $(MANROOT)/man
        MAN1SUFFIX = 1
        MAN2SUFFIX = 2
@@ -427,7 +427,7 @@
 
 # test program
 
-all:: etmtest
+all:: 
 
 help_aux::
 	@echo "'make "etmtest"' to "build etmtest
@@ -540,8 +540,5 @@
 
 # DO NOT DELETE
 
-etm.o: /usr/include/stdio.h /usr/include/sys/stdsyms.h /usr/include/signal.h
-etm.o: /usr/include/sys/signal.h /usr/include/sys/types.h
-etm.o: /usr/include/stdarg.h /usr/include/varargs.h etm.internal.h etm.h
-etm.o: /usr/include/stdlib.h
-etmtest.o: /usr/include/stdio.h /usr/include/sys/stdsyms.h etm.h
+etm.o: etm.internal.h etm.h
+etmtest.o: etm.h
diff -ruN etm-orig/msub etm/msub
--- etm-orig/msub	Wed Dec 31 19:00:00 1969
+++ etm/msub	Tue Jan 16 16:50:56 2007
@@ -0,0 +1,29 @@
+#!/usr/bin/perl
+
+use Data::Dumper;
+
+my $file = shift;
+my $vars;
+
+if (open(FILEIN, 'Makefile'))
+{
+	while (<FILEIN>)
+	{
+		if (/^\s*([^\=]+?)\s*=\s*(.*?)\s*$/)
+		{
+			$vars->{$1} = $2;
+		}
+	}
+}
+
+#print Dumper($vars);
+
+if (open(FILEIN, $file))
+{
+	while (<FILEIN>)
+	{
+		s/\$\{([^\}]+)\}/$vars->{$1}/gs;
+		print;
+	}
+	close(FILEIN);
+}

 	  	 
