diff -Nurd -x'*~' exiv2-0.19.orig/config/config.mk.in exiv2-0.19/config/config.mk.in
--- exiv2-0.19.orig/config/config.mk.in	2009-11-27 06:39:59.000000000 -0500
+++ exiv2-0.19/config/config.mk.in	2010-08-22 23:49:35.000000000 -0400
@@ -95,8 +95,7 @@
         XMPSDK_LIBRARY = xmpsdk
         XMPSDK_DIR = $(top_srcdir)/xmpsdk
         XMPSDK_CPPFLAGS = -I$(XMPSDK_DIR)/include
-        XMPSDK_LDFLAGS = -L$(XMPSDK_DIR)/src
-        XMPSDK_LIBS = -l$(XMPSDK_LIBRARY)
+        XMPSDK_LDFLAGS = $(XMPSDK_DIR)/src/lib$(XMPSDK_LIBRARY).la
 else
         # Enable additional warnings. XMP Toolkit doesn't compile
         # with these.
diff -Nurd -x'*~' exiv2-0.19.orig/configure exiv2-0.19/configure
--- exiv2-0.19.orig/configure	2009-12-30 08:39:52.000000000 -0500
+++ exiv2-0.19/configure	2010-08-22 19:22:21.000000000 -0400
@@ -5080,7 +5080,7 @@
   fi
   INTL_MACOSX_LIBS=
   if test $gt_cv_func_CFPreferencesCopyAppValue = yes || test $gt_cv_func_CFLocaleCopyCurrent = yes; then
-    INTL_MACOSX_LIBS="-Wl,-framework -Wl,CoreFoundation"
+    INTL_MACOSX_LIBS="-Wl,-framework,CoreFoundation"
   fi
 
 
