diff -Nurd -x'*~' fltk-1.1.6.orig/Makefile fltk-1.1.6/Makefile
--- fltk-1.1.6.orig/Makefile	2004-10-18 16:22:21.000000000 -0400
+++ fltk-1.1.6/Makefile	2005-07-02 01:30:15.000000000 -0400
@@ -25,7 +25,7 @@
 
 include makeinclude
 
-DIRS	=	$(IMAGEDIRS) src fluid test documentation
+DIRS	=	$(IMAGEDIRS) src fluid documentation
 
 all: makeinclude
 	for dir in $(DIRS); do\
diff -Nurd -x'*~' fltk-1.1.6.orig/configure fltk-1.1.6/configure
--- fltk-1.1.6.orig/configure	2004-11-23 15:35:43.000000000 -0500
+++ fltk-1.1.6/configure	2006-02-22 23:42:39.000000000 -0500
@@ -856,7 +856,6 @@
   --enable-gl             turn on OpenGL support default=yes
   --enable-shared         turn on shared libraries default=no
   --enable-threads        enable multi-threading support
-  --enable-quartz         use Quartz instead of Quickdraw (default=no)
   --enable-localjpeg      use local JPEG library, default=auto
   --enable-localzlib      use local ZLIB library, default=auto
   --enable-localpng       use local PNG library, default=auto
@@ -1388,7 +1387,7 @@
             FLDSONAME="libfltk_forms.$FL_API_VERSION.dylib"
             GLDSONAME="libfltk_gl.$FL_API_VERSION.dylib"
             IMGDSONAME="libfltk_images.$FL_API_VERSION.dylib"
-	    DSOCOMMAND="\$(CC) $DSOFLAGS -dynamiclib -lc -o"
+	    DSOCOMMAND="\$(CXX) $DSOFLAGS -single_module -dynamiclib -lc -o"
 	    ;;
 
 	SunOS* | UNIX_S*)
@@ -1501,31 +1500,6 @@
 
 fi;
 
-case $uname in
-    Darwin*)
-        # Check whether --enable-quartz or --disable-quartz was given.
-if test "${enable_quartz+set}" = set; then
-  enableval="$enable_quartz"
-
-fi;
-        if test "x$enable_quartz" = "xyes"; then
-            cat >>confdefs.h <<\_ACEOF
-#define USE_QUARTZ 1
-_ACEOF
-
-            cat >>confdefs.h <<\_ACEOF
-#define __APPLE_QUARTZ__ 1
-_ACEOF
-
-        else
-            cat >>confdefs.h <<\_ACEOF
-#define __APPLE_QD__ 1
-_ACEOF
-
-        fi
-        ;;
-esac
-
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
@@ -6700,7 +6674,17 @@
     fi
 fi
 
-HLINKS=
+# Don't make symlinks if filesystem is not case sensitive...
+case $uname in
+    CYGWIN* | MINGW* | Darwin*)
+	HLINKS="#"
+    ;;
+    *)
+	HLINKS=
+    ;;
+esac
+
+
 POSTBUILD=:
 THREADS=
 
@@ -7011,48 +6995,6 @@
 	    THREADS="threads.exe"
 	fi
 
-	# Don't make symlinks since Windows is not case sensitive.
-	if test "x$with_links" != xyes; then
-		HLINKS="#"
-	fi
-	;;
-
-    Darwin*)
-        # MacOS X uses Carbon for graphics...
-        LIBS="$LIBS -framework Carbon -framework ApplicationServices"
-
-	if test x$have_pthread = xyes; then
-	    cat >>confdefs.h <<\_ACEOF
-#define HAVE_PTHREAD 1
-_ACEOF
-
-	    THREADS="threads"
-	fi
-
-	if test x$enable_gl != xno; then
-            cat >>confdefs.h <<\_ACEOF
-#define HAVE_GL 1
-_ACEOF
-
-            cat >>confdefs.h <<\_ACEOF
-#define HAVE_GL_GLU_H 1
-_ACEOF
-
-            GLLIB="-framework AGL -framework OpenGL"
-        else
-	    LINKFLTKGL=""
-	    GLLIBNAME=""
-	    GLDSONAME=""
-	    GLDEMOS=""
-        fi
-
-	# Don't make symlinks because HFS+ is not case sensitive...
-	if test "x$with_links" != xyes; then
-		HLINKS="#"
-	fi
-
-	# Add a postbuild step after linking applications
-	POSTBUILD="/Developer/Tools/Rez -t APPL -o"
 	;;
 
     *)
@@ -9230,7 +9172,7 @@
 	    ;;
     esac
 
-    CXX="$CC"
+    CXX="${CXX:-$CC}"
 
     # Show all standard warnings + unused variables, conversion errors,
     # and inlining problems when compiling...
diff -Nurd -x'*~' fltk-1.1.6.orig/fluid/fluid.cxx fltk-1.1.6/fluid/fluid.cxx
--- fltk-1.1.6.orig/fluid/fluid.cxx	2004-09-08 11:42:47.000000000 -0400
+++ fltk-1.1.6/fluid/fluid.cxx	2005-07-02 01:18:57.000000000 -0400
@@ -617,11 +617,7 @@
   fl_filename_absolute(absolute, sizeof(absolute), flname);
 
   for (i = 0; i < 10; i ++)
-#if defined(WIN32) || defined(__APPLE__)
     if (!strcasecmp(absolute, absolute_history[i])) break;
-#else
-    if (!strcmp(absolute, absolute_history[i])) break;
-#endif // WIN32 || __APPLE__
 
   if (i == 0) return;
 
diff -Nurd -x'*~' fltk-1.1.6.orig/src/Makefile fltk-1.1.6/src/Makefile
--- fltk-1.1.6.orig/src/Makefile	2004-11-19 22:19:58.000000000 -0500
+++ fltk-1.1.6/src/Makefile	2005-07-02 01:18:57.000000000 -0400
@@ -217,7 +217,7 @@
 		-install_name $(libdir)/$@ \
 		-current_version 1.1.4 \
 		-compatibility_version 1.1.0 \
-		$(OBJECTS) $(LDLIBS)
+		$(OBJECTS) $(LDLIBS) $(FONTCONFIGLIB)
 	$(RM) libfltk.dylib
 	$(LN) libfltk.1.1.dylib libfltk.dylib
 
diff -Nurd -x'*~' fltk-1.1.6.orig/src/filename_absolute.cxx fltk-1.1.6/src/filename_absolute.cxx
--- fltk-1.1.6.orig/src/filename_absolute.cxx	2004-04-11 00:38:59.000000000 -0400
+++ fltk-1.1.6/src/filename_absolute.cxx	2005-07-02 01:18:57.000000000 -0400
@@ -154,11 +154,7 @@
        *slash != '\0' && *newslash != '\0';
        slash ++, newslash ++)
     if (isdirsep(*slash) && isdirsep(*newslash)) continue;
-#if defined(WIN32) || defined(__EMX__) || defined(__APPLE__)
     else if (tolower(*slash) != tolower(*newslash)) break;
-#else
-    else if (*slash != *newslash) break;
-#endif // WIN32 || __EMX__ || __APPLE__
 
   if (*newslash == '\0' && *slash != '\0' && !isdirsep(*slash))
     newslash--;
diff -Nurd -x'*~' fltk-1.1.6.orig/test/colbrowser.cxx fltk-1.1.6/test/colbrowser.cxx
--- fltk-1.1.6.orig/test/colbrowser.cxx	2004-04-11 00:39:01.000000000 -0400
+++ fltk-1.1.6/test/colbrowser.cxx	2005-07-02 01:18:57.000000000 -0400
@@ -42,21 +42,7 @@
 static int load_browser(char *);
 
 /* the RGB data file does not have a standard location on unix. */
-
-#ifdef __VMS
-  static const char *rgbfile = "SYS$MANAGER:DECW$RGB.DAT";
-#else
-#ifdef __EMX__   /* OS2 */
-#include <X11/XlibInt.h>
-  static const char *rgbfile = "/XFree86/lib/X11/rgb.txt";
-#else
-#ifdef __FreeBSD__
-  static const char *rgbfile = "/usr/X11R6/lib/X11/rgb.txt";
-#else
-   static const char *rgbfile = "/usr/lib/X11/rgb.txt";
-#endif
-#endif
-#endif
+static const char *rgbfile = "/usr/X11R6/lib/X11/rgb.txt";
 
 typedef struct { int r, g, b; } RGBdb;
 
