diff -uwr ./configure.ac ../gmerlin-avdecoder-1.0.0new/configure.ac
--- ./configure.ac	2008-11-08 09:06:26.000000000 -0500
+++ ../gmerlin-avdecoder-1.0.0new/configure.ac	2009-05-02 15:28:44.000000000 -0400
@@ -306,14 +306,25 @@
 
 AH_TEMPLATE([HAVE_W32DLL], [ Win32 dll loader enabled ])
 
+AC_ARG_ENABLE(win32,
+[AC_HELP_STRING([--disable-win32],[Disable win32 DLLs (default: autodetect)])],
+[case "${enableval}" in
+   yes) test_w32dll=true ;;
+   no)  test_w32dll=false ;;
+esac],[test_w32dll=true])
+
 enable_w32dll="false"
 
+if test "x$test_w32dll" = "true"; then
+
 if test x"$GCC" = x"yes"; then
  case "$host_cpu" in
     i?86 | k?-* | athlon | pentium) enable_w32dll="true"
  esac
 fi
 
+fi
+
 if test "x$enable_w32dll" = "xtrue"; then
 AC_DEFINE([HAVE_W32DLL])
 fi
diff -uwr ./include/rtp.h ../gmerlin-avdecoder-1.0.0new/include/rtp.h
--- ./include/rtp.h	2008-11-21 19:51:26.000000000 -0500
+++ ../gmerlin-avdecoder-1.0.0new/include/rtp.h	2009-05-02 15:28:51.000000000 -0400
@@ -20,6 +20,7 @@
  * *****************************************************************/
 
 #include <sdp.h>
+#include <pthread.h>
 
 extern bgav_demuxer_t bgav_demuxer_rtp;
 
diff -uwr ./lib/tcp.c ../gmerlin-avdecoder-1.0.0new/lib/tcp.c
--- ./lib/tcp.c	2008-10-15 17:39:36.000000000 -0400
+++ ../gmerlin-avdecoder-1.0.0new/lib/tcp.c	2009-05-02 15:29:00.000000000 -0400
@@ -45,6 +45,12 @@
 #define MSG_NOSIGNAL 0
 #endif
 
+/* for MacOSX */
+#ifndef AI_NUMERICSERV
+#define AI_NUMERICSERV 0
+#endif
+
+
 /* Utility functions */
 
 static int create_socket(int domain, int type, int protocol)
