diff -Nurd libidn-1.12/Makefile.in libidn-1.12-new/Makefile.in
--- libidn-1.12/Makefile.in	2009-01-23 08:31:49.000000000 -0500
+++ libidn-1.12-new/Makefile.in	2009-02-11 10:18:38.000000000 -0500
@@ -493,7 +493,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = gl lib/gl lib po src doc gltests lib/gltests tests examples	\
+SUBDIRS = gl lib/gl lib po src gltests lib/gltests tests examples	\
 	java csharp
 
 
diff -Nurd libidn-1.12/java/misc/Makefile.in libidn-1.12-new/java/misc/Makefile.in
--- libidn-1.12/java/misc/Makefile.in	2009-01-23 08:31:48.000000000 -0500
+++ libidn-1.12-new/java/misc/Makefile.in	2009-02-11 10:18:54.000000000 -0500
@@ -455,7 +455,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 JAVAROOT = $(top_builddir)/java/misc
-AM_JAVACFLAGS = -classpath $(top_builddir)/java/libidn-@VERSION@.jar:$(top_srcdir)/java/libidn-@VERSION@.jar
+AM_JAVACFLAGS = -classpath $(top_builddir)/java/libidn-@VERSION@.jar:$(top_srcdir)/java/libidn-@VERSION@.jar -target 1.5 -source 1.5
 @JAVA_TRUE@dist_noinst_JAVA = GenerateRFC3454.java GenerateNFKC.java	\
 @JAVA_TRUE@			TestIDNA.java TestNFKC.java
 
