diff -Nurd -x'*~' libmaa-1.3.0.orig/Makefile.am libmaa-1.3.0/Makefile.am
--- libmaa-1.3.0.orig/Makefile.am	1969-12-31 19:00:00.000000000 -0500
+++ libmaa-1.3.0/Makefile.am	2011-09-25 09:57:53.000000000 -0400
@@ -0,0 +1,122 @@
+VERSION=$(MAA_MAJOR).$(MAA_MINOR).$(MAA_TEENY)
+VERCFLAGS=       -DMAA_MAJOR=$(MAA_MAJOR) -DMAA_MINOR=$(MAA_MINOR) -DMAA_TEENY=$(MAA_TEENY)
+AM_CPPFLAGS=$(VERCFLAGS)
+
+lib_LTLIBRARIES = libmaa.la
+include_HEADERS= maa.h
+libmaa_la_SOURCES = maa.h obstack.h \
+	 obstack.c xmalloc.c \
+	 hash.c set.c stack.c list.c error.c memory.c string.c \
+	 debug.c flags.c maa.c prime.c bit.c timer.c \
+	 arg.c pr.c sl.c base64.c base26.c source.c parse-concrete.c \
+	 text.c log.c arggram.c
+libmaa_la_LIBADD  = $(LIBOBJS) -lm
+libmaa_la_LDFLAGS = -export-symbols ${srcdir}/export.sym \
+	   -version-info 3:0 -no-undefined
+
+noinst_LTLIBRARIES = librnd.la
+librnd_la_SOURCES = rnd.c
+
+TESTLIBS = librnd.la libmaa.la
+TESTS=	 settest hashtest stringtest debugtest \
+	 primetest listtest bittest argtest sltest prtest basetest
+TESTS_SH= logtest.sh
+EXES=    memtest prtest logtest
+noinst_PROGRAMS = $(TESTS) $(EXES)
+
+test: check logtest
+	@ex=0; \
+	if test "${srcdir}" != .; then \
+		cp ${srcdir}/argtest.in ${srcdir}/logtest.sh .; \
+	fi; \
+	for i in $(TESTS) $(TESTS_SH); do \
+		./$$i | fgrep -v ' at ' > ./$$i.out.new; \
+		if cmp @srcdir@/$$i.out $$i.out.new; \
+		then \
+			echo PASSED $$i; rm -f $$i.out.new; \
+		else \
+			echo FAILED $$i; ex=1; \
+		fi; \
+	done; \
+	exit $$ex
+
+.PHONY: test-update
+test-update: $(TESTS)
+	@for i in $(TESTS) $(TESTS_SH); do \
+		echo Writing $$i.out; \
+		./$$i | fgrep -v ' at ' > $$i.out; \
+	done
+
+BUILT_SOURCES = arggram.c
+arggram.c: ${srcdir}/arggram.txt
+	@AWK@ -f ./arggram2c < ${srcdir}/arggram.txt > $@
+
+settest_SOURCES = settest.c
+settest_LDADD = $(TESTLIBS)
+hashtest_SOURCES = hashtest.c
+hashtest_LDADD = $(TESTLIBS)
+stringtest_SOURCES = stringtest.c
+stringtest_LDADD = $(TESTLIBS)
+debugtest_SOURCES = debugtest.c
+debugtest_LDADD = $(TESTLIBS)
+primetest_SOURCES = primetest.c
+primetest_LDADD = $(TESTLIBS)
+listtest_SOURCES = listtest.c
+listtest_LDADD = $(TESTLIBS)
+bittest_SOURCES = bittest.c
+bittest_LDADD = $(TESTLIBS)
+argtest_SOURCES = argtest.c
+argtest_LDADD = $(TESTLIBS)
+sltest_SOURCES = sltest.c
+sltest_LDADD = $(TESTLIBS)
+prtest_SOURCES = prtest.c
+prtest_LDADD = $(TESTLIBS)
+basetest_SOURCES = basetest.c
+basetest_LDADD = $(TESTLIBS)
+logtest_SOURCES = logtest.c
+logtest_LDADD = $(TESTLIBS)
+memtest_SOURCES = memtest.c
+memtest_LDADD = $(TESTLIBS)
+
+.PHONY: ChangeLog
+ChangeLog:
+	(echo "***** Making new ChangeLog..."; \
+	rm -f ChangeLog; \
+	AWK=@AWK@ rcs2log -i 2 | \
+	sed -e 's,/cvsroot/dict/dictd1/,,g' \
+	      -e 's,/cvs/dict/dictd1/,,g' \
+	      -e 's,@[^>]*,@,g' \
+	      -e 's,cheusov@[^>]*,vle@gmx.net,g' \
+	      -e 's,faith@[^>]*,faith@dict.org,g' \
+	      -e 's,hilliard...hilliard@.*,Bob Hilliard  <hilliard@debian.org>,g' \
+	      -e 's,tek...tek@.*,Julian Squires  <tek@wiw.org>,g' \
+              -e 's,\(.*\)<\([^@<>]\+\)@\([^@<>]\+\)>\(.*\),\1<\2 at \3}\4,g' \
+	> ChangeLog)
+
+.PHONY: cvsdist
+cvsdist:
+	@( CVSROOT=`cat CVS/Root` && \
+	export CVSROOT && \
+	VERSION=$(VERSION) && \
+	VERSION_CVS=`echo $${VERSION} | tr . -` && \
+	export VERSION VERSION_CVS && \
+	$(MAKE) ChangeLog && \
+	cp ChangeLog /tmp/libmaa.ChangeLog.$${VERSION} && \
+	echo "***** Exporting files for libmaa-$${VERSION}..." && \
+	cd /tmp && \
+	rm -rf /tmp/libmaa-$${VERSION} && \
+	cvs export -fd libmaa-$${VERSION} \
+	   -r libmaa-$${VERSION_CVS} dictd1/libmaa && \
+	cd libmaa-$${VERSION} && \
+	mv /tmp/libmaa.ChangeLog.$${VERSION} ChangeLog && \
+	autoheader && autoconf && \
+	rm -rf autom4te.cache libmaa/autom4te.cache && \
+	chmod -R a+rX,g-w . && \
+	cd .. && \
+	echo "***** Taring and zipping libmaa-$${VERSION}.tar.gz..." && \
+	tar cvvf libmaa-$${VERSION}.tar libmaa-$${VERSION} && \
+	gzip -9f libmaa-$${VERSION}.tar && \
+	echo "***** Done making /tmp/libmaa-$${VERSION}.tar.gz")
+
+tags:
+	etags: *.c
diff -Nurd -x'*~' libmaa-1.3.0.orig/configure.in libmaa-1.3.0/configure.in
--- libmaa-1.3.0.orig/configure.in	2011-04-10 11:06:20.000000000 -0400
+++ libmaa-1.3.0/configure.in	2011-09-25 09:35:34.000000000 -0400
@@ -24,9 +24,10 @@
 
 AC_PREREQ(2.53)
 AC_REVISION($Revision: 1.59 $)
-AC_INIT
+AC_INIT([libmaa],[1.3.0])
 AC_CONFIG_SRCDIR([maa.h])
 AC_CONFIG_HEADER(config.h)
+AM_INIT_AUTOMAKE
 
 MAA_MAJOR=1
 MAA_MINOR=3
@@ -38,7 +39,7 @@
 AC_CANONICAL_HOST
 AC_PROG_AWK
 AC_PROG_CC
-#AC_PROG_LIBTOOL
+LT_INIT
 AC_ISC_POSIX
 
 if test "$CC" = gcc; then
@@ -57,7 +58,6 @@
 AC_CHECK_PROGS(REFBIBTEX,refbibtex)
 AC_CHECK_PROGS(BIBTEX,bibtex)
 
-AC_CHECK_PROG(LIBTOOL,libtool,libtool)
 
 echo .
 echo Checking for header file support
@@ -115,8 +115,5 @@
 AC_CONFIG_FILES([Makefile doc/Makefile arggram2c])
 
 AC_OUTPUT
-
-touch stamp-h.in
-touch stamp-h
 echo .
 echo Done
