diff -uNr liboil-0.3.9/liboil/liboilcpu.c liboil-0.3.9-new/liboil/liboilcpu.c
--- liboil-0.3.9/liboil/liboilcpu.c	2006-05-22 20:07:56.000000000 -0400
+++ liboil-0.3.9-new/liboil/liboilcpu.c	2006-07-02 22:40:40.000000000 -0400
@@ -42,7 +42,7 @@
 #include <sys/time.h>
 #include <time.h>
 
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD__) || defined(__APPLE__)
 #include <sys/types.h>
 #include <sys/sysctl.h>
 #endif
@@ -531,7 +531,7 @@
 static void
 oil_cpu_detect_kernel_support (void)
 {
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__APPLE__)
   int ret, enabled;
   size_t len;
 
@@ -542,7 +542,7 @@
 		       OIL_IMPL_FLAG_MMXEXT | OIL_IMPL_FLAG_SSE3);
   }
 #endif
-#if !defined(__linux__) && !defined(__FreeBSD__)
+#if !defined(__linux__) && !defined(__FreeBSD__) && !defined(__APPLE__)
   /* If we don't know that the operating system supports SSE, don't trust that
    * it will properly support it.
    */
@@ -587,25 +587,17 @@
 }
 
 static void
-test_altivec (void * ignored)
-{
-  char x[16] = { 0, };
-
-  asm volatile (
-      "  lvx %%v0, %0, %%r0  \n"
-      :: "r" (x));
-}
-
-static void
 oil_cpu_detect_powerpc(void)
 {
+  int sels[2] = { CTL_HW, HW_VECTORUNIT };
+  int vType = 0; //0 == scalar only
+  size_t length = sizeof(vType);
+  int error = sysctl(sels, 2, &vType, &length, NULL, 0);
 
-  oil_cpu_fault_check_enable ();
-  if (oil_cpu_fault_check_try(test_altivec, NULL)) {
+  if ( 0 == error ) {
     OIL_DEBUG ("cpu flag altivec");
     oil_cpu_flags |= OIL_IMPL_FLAG_ALTIVEC;
   }
-  oil_cpu_fault_check_disable ();
 
   _oil_profile_stamp = oil_profile_stamp_tb;
 }
diff -uNr liboil-0.3.9/liboil/powerpc/Makefile.in liboil-0.3.9-new/liboil/powerpc/Makefile.in
--- liboil-0.3.9/liboil/powerpc/Makefile.in	2006-05-22 20:20:18.000000000 -0400
+++ liboil-0.3.9-new/liboil/powerpc/Makefile.in	2006-07-02 22:07:21.000000000 -0400
@@ -56,7 +56,7 @@
 libpowerpc_la_LIBADD =
 am__objects_1 = libpowerpc_la-abs.lo libpowerpc_la-clip.lo \
 	libpowerpc_la-conv.lo libpowerpc_la-copy.lo \
-	libpowerpc_la-md5.lo libpowerpc_la-mix.lo \
+	libpowerpc_la-mix.lo \
 	libpowerpc_la-multsum.lo libpowerpc_la-resample.lo \
 	libpowerpc_la-rgb2bgr.lo libpowerpc_la-rgb2rgba.lo \
 	libpowerpc_la-sad8x8.lo libpowerpc_la-splat.lo \
@@ -249,7 +249,6 @@
 	clip.c \
 	conv.c \
 	copy.c \
-	md5.c \
 	mix.c \
 	multsum.c \
 	resample.c \
@@ -318,7 +317,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libpowerpc_la-clip.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libpowerpc_la-conv.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libpowerpc_la-copy.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libpowerpc_la-md5.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libpowerpc_la-mix.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libpowerpc_la-multsum.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libpowerpc_la-resample.Plo@am__quote@
@@ -377,13 +375,6 @@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCC_FALSE@	$(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libpowerpc_la_CFLAGS) $(CFLAGS) -c -o libpowerpc_la-copy.lo `test -f 'copy.c' || echo '$(srcdir)/'`copy.c
 
-libpowerpc_la-md5.lo: md5.c
-@am__fastdepCC_TRUE@	if $(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libpowerpc_la_CFLAGS) $(CFLAGS) -MT libpowerpc_la-md5.lo -MD -MP -MF "$(DEPDIR)/libpowerpc_la-md5.Tpo" -c -o libpowerpc_la-md5.lo `test -f 'md5.c' || echo '$(srcdir)/'`md5.c; \
-@am__fastdepCC_TRUE@	then mv -f "$(DEPDIR)/libpowerpc_la-md5.Tpo" "$(DEPDIR)/libpowerpc_la-md5.Plo"; else rm -f "$(DEPDIR)/libpowerpc_la-md5.Tpo"; exit 1; fi
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	source='md5.c' object='libpowerpc_la-md5.lo' libtool=yes @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libpowerpc_la_CFLAGS) $(CFLAGS) -c -o libpowerpc_la-md5.lo `test -f 'md5.c' || echo '$(srcdir)/'`md5.c
-
 libpowerpc_la-mix.lo: mix.c
 @am__fastdepCC_TRUE@	if $(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libpowerpc_la_CFLAGS) $(CFLAGS) -MT libpowerpc_la-mix.lo -MD -MP -MF "$(DEPDIR)/libpowerpc_la-mix.Tpo" -c -o libpowerpc_la-mix.lo `test -f 'mix.c' || echo '$(srcdir)/'`mix.c; \
 @am__fastdepCC_TRUE@	then mv -f "$(DEPDIR)/libpowerpc_la-mix.Tpo" "$(DEPDIR)/libpowerpc_la-mix.Plo"; else rm -f "$(DEPDIR)/libpowerpc_la-mix.Tpo"; exit 1; fi
diff -uNr liboil-0.3.9/testsuite/mmx_engine.c liboil-0.3.9-new/testsuite/mmx_engine.c
--- liboil-0.3.9/testsuite/mmx_engine.c	2006-02-15 16:05:48.000000000 -0500
+++ liboil-0.3.9-new/testsuite/mmx_engine.c	2006-07-02 22:41:01.000000000 -0400
@@ -118,7 +118,7 @@
     oil_profile_start(&prof); \
     asm volatile ( \
         "  mov $1000, %%ecx\n" \
-        ".align 16\n" \
+        ".align 4\n" \
         "1:\n" \
         "  " #insn " %%mm0, %%mm1\n" \
         "  " #insn " %%mm1, %%mm2\n" \
@@ -187,7 +187,7 @@
     oil_profile_start(&prof); \
     asm volatile ( \
         "  mov $1000, %%ecx\n" \
-        ".align 16\n" \
+        ".align 4\n" \
         "1:\n" \
         "  " #insn " %%mm0, %%mm1\n" \
         "  " #insn " %%mm0, %%mm2\n" \
@@ -279,7 +279,7 @@
     oil_profile_start(&prof); \
     asm volatile ( \
         "  mov $1000, %%ecx\n" \
-        ".align 16\n" \
+        ".align 4\n" \
         "1:\n" \
         "  " #insn " %%xmm0, %%xmm1\n" \
         "  " #insn " %%xmm1, %%xmm2\n" \
@@ -347,7 +347,7 @@
     oil_profile_start(&prof); \
     asm volatile ( \
         "  mov $1000, %%ecx\n" \
-        ".align 16\n" \
+        ".align 4\n" \
         "1:\n" \
         "  " #insn " %%xmm0, %%xmm1\n" \
         "  " #insn " %%xmm0, %%xmm2\n" \
