diff -urN ogdi-3.1.4-orig/config/common.mak ogdi-3.1.4/config/common.mak
--- ogdi-3.1.4-orig/config/common.mak	2001-06-26 10:14:46.000000000 +0900
+++ ogdi-3.1.4/config/common.mak	2005-08-15 08:02:49.000000000 +0900
@@ -55,7 +55,7 @@
 #
 #	Final OS installation location.
 #
-prefix		= /usr/local
+prefix		= @PREFIX@
 exec_prefix 	= ${prefix}
 INST_INCLUDE	= ${prefix}/include
 INST_LIB	= ${exec_prefix}/lib
diff -urN ogdi-3.1.4-orig/config/darwin.mak ogdi-3.1.4/config/darwin.mak
--- ogdi-3.1.4-orig/config/darwin.mak	2003-12-16 03:08:00.000000000 +0900
+++ ogdi-3.1.4/config/darwin.mak	2005-08-15 08:04:20.000000000 +0900
@@ -33,7 +33,7 @@
 SHLIB_CFLAGS	= -fno-common
 COMMON_CFLAGS	= $(OPTIMIZATION) -fno-common
 
-SHLIB_LDFLAGS	= -dynamiclib
+SHLIB_LDFLAGS	= -dynamiclib -compatibility_version 3.1.0 -current_version 3.1.5
 COMMON_LDFLAGS  = $(OPTIMIZATION)
 
 
@@ -50,6 +50,16 @@
 
 BIG_ENDIAN          = 1
 
+RPC_LINKLIB	=	-ldl 
+PROJ_SETTING=external
+PROJ_INCLUDE :=	-I@PREFIX@/include
+PROJ_STATICLIB := -L@PREFIX@/lib -lproj
+ZLIB_SETTING=external
+ZLIB_INCLUDE :=	
+ZLIB_LINKLIB := -lz
+EXPAT_SETTING=external
+EXPAT_INCLUDE := -I@PREFIX@/include
+EXPAT_LINKLIB := -L@PREFIX@/lib -lexpat
 
 #
 # make rules
@@ -62,5 +72,5 @@
 	
 $(DYNAGEN): $(OBJECTS)
 	@echo Making shared library: $@
-	$(SHLIB_LD) $(SHLIB_LDFLAGS) $(COMMON_LDFLAGS) -o $@ $^ $(LINK_LIBS) 
+	$(SHLIB_LD) $(SHLIB_LDFLAGS) -install_name $(INST_LIB)/$(notdir $@) $(COMMON_LDFLAGS) -o $@ $^ $(LINK_LIBS) 
 	@echo $@ made successfully ...
diff -urN ogdi-3.1.4-orig/ogdi/c-api/ecs_dyna.c ogdi-3.1.4/ogdi/c-api/ecs_dyna.c
--- ogdi-3.1.4-orig/ogdi/c-api/ecs_dyna.c	2004-02-19 06:50:21.000000000 +0900
+++ ogdi-3.1.4/ogdi/c-api/ecs_dyna.c	2005-08-15 08:02:05.000000000 +0900
@@ -82,11 +82,11 @@
   if (handle == NULL) {
     /* Try with the .so extension */
 
-    if ((temp = (char *) malloc(strlen(libname)+7)) == NULL)
+    if ((temp = (char *) malloc(strlen(libname)+10)) == NULL)
       return NULL;
     strcpy(temp,"lib");
     strcat(temp,libname);
-    strcat(temp,".so");
+    strcat(temp,".dylib");
 
     handle = dlopen(temp,RTLD_LAZY);
 #ifdef notdef
diff -ruN ogdi-3.1.5/ogdi/driver/rpf/rpf.h ogdi-3.1.5-patched/ogdi/driver/rpf/rpf.h
--- ogdi-3.1.5/ogdi/driver/rpf/rpf.h	2001-04-12 12:22:46.000000000 -0700
+++ ogdi-3.1.5-patched/ogdi/driver/rpf/rpf.h	2007-08-19 09:46:05.000000000 -0700
@@ -52,7 +52,7 @@
 typedef unsigned short ushort;
 typedef unsigned long  uint;
 */
-typedef unsigned char  bool;
+typedef unsigned char  bbool;
 typedef unsigned char  uchar;
 typedef unsigned char* ucharp;
 typedef char           ascii;
@@ -114,7 +114,7 @@
 
 typedef struct header
 {
-  bool     endian;
+  bbool    endian;
   ushort   hdr_sec_len;       /* header section length */
   ascii    filename[12];
   uchar    new;
@@ -330,8 +330,8 @@
 
 typedef struct display1
 {
-  bool      all_subframes;
-  bool      no_transparent;
+  bbool     all_subframes;
+  bbool     no_transparent;
   ushort    subframe_seq_length;
   ushort    transparent_seq_length;
   uint      image_rows;
@@ -355,7 +355,7 @@
   RPF_image    img;
   Attribute    att;
   Replace      repupd;
-  bool         all_subframes;
+  bbool        all_subframes;
   int         NITF_hdr_len;
 } Frame_file;
 
