--- libplist-1.3/CMakeLists.orig.txt	2011-01-13 22:02:54.000000000 +0100
+++ libplist-1.3/CMakeLists.txt	2011-01-13 21:57:30.000000000 +0100
@@ -15,6 +15,19 @@
    SET(CMAKE_INSTALL_LIBDIR lib CACHE PATH "Output directory for libraries")
 ENDIF(NOT DEFINED CMAKE_INSTALL_LIBDIR)
 
+# Set up RPATH handling, so the libs are found if they are installed to a non-standard location.
+# By default cmake builds the targets with full RPATH to everything in the build directory,
+# but then removes the RPATH when installing.
+# These two options below make it set the RPATH of the installed targets to all
+# RPATH directories outside the current CMAKE_BINARY_DIR and also the library
+# install directory.
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH  TRUE)
+set(CMAKE_INSTALL_RPATH                ${LIB_DESTINATION} )
+
+if(APPLE)
+   set(CMAKE_INSTALL_NAME_DIR ${LIB_DESTINATION})
+endif(APPLE)
+
 FIND_PACKAGE( LibXml2 REQUIRED )
 FIND_PACKAGE( GLIB2 REQUIRED )
 
