diff -urN libsigsegv-2.6.orig/build-aux/ltmain.sh libsigsegv-2.6/build-aux/ltmain.sh
--- libsigsegv-2.6.orig/build-aux/ltmain.sh	2008-05-18 03:42:00.000000000 -0700
+++ libsigsegv-2.6/build-aux/ltmain.sh	2009-01-18 23:58:29.000000000 -0800
@@ -5728,11 +5728,11 @@
 	darwin)
 	  # Like Linux, but with the current version available in
 	  # verstring for coding it into the library header
-	  func_arith $current - $age
+	  func_arith $current - $age + 1
 	  major=.$func_arith_result
 	  versuffix="$major.$age.$revision"
 	  # Darwin ld doesn't like 0 for these options...
-	  func_arith $current + 1
+	  func_arith $current + 2
 	  minor_current=$func_arith_result
 	  xlcverstring="${wl}-compatibility_version ${wl}$minor_current ${wl}-current_version ${wl}$minor_current.$revision"
 	  verstring="-compatibility_version $minor_current -current_version $minor_current.$revision"
--- libsigsegv-2.6/src/stackvma-mach.c.orig	2010-02-14 20:34:05.000000000 -0200
+++ libsigsegv-2.6/src/stackvma-mach.c	2010-02-14 20:37:32.000000000 -0200
@@ -53,11 +53,11 @@
   for (address = VM_MIN_ADDRESS; more; address += size)
     {
 #ifdef VM_REGION_BASIC_INFO
-      more = (vm_region (task, &address, &size, VM_REGION_BASIC_INFO,
+      more = (mach_vm_region (task, &address, &size, VM_REGION_BASIC_INFO,
                          (vm_region_info_t)&info, &info_count, &object_name)
               == KERN_SUCCESS);
 #else
-      more = (vm_region (task, &address, &size, &protection, &max_protection,
+      more = (mach_vm_region (task, &address, &size, &protection, &max_protection,
                          &inheritance, &shared, &object_name, &offset)
               == KERN_SUCCESS);
 #endif
