diff -ru v8-2.0.3.orig/SConstruct v8-2.0.3/SConstruct
--- v8-2.0.3.orig/SConstruct	2009-12-10 01:58:33.000000000 +1100
+++ v8-2.0.3/SConstruct	2009-12-10 02:09:43.000000000 +1100
@@ -143,6 +143,10 @@
     },
     'os:macos': {
       'CCFLAGS':      ['-ansi', '-mmacosx-version-min=10.4'],
+      'library:shared': {
+        'CPPDEFINES': ['V8_SHARED'],
+		'LINKFLAGS': ['-Wl,-install_name,INSTALL_NAME']
+      }
     },
     'os:freebsd': {
       'CPPPATH' : ['/usr/local/include'],
