diff -uNr libvisual-plugins-0.4.0/plugins/actor/G-Force/Makefile.in libvisual-plugins-0.4.0-patched/plugins/actor/G-Force/Makefile.in
--- libvisual-plugins-0.4.0/plugins/actor/G-Force/Makefile.in	2006-03-20 13:48:53.000000000 -0500
+++ libvisual-plugins-0.4.0-patched/plugins/actor/G-Force/Makefile.in	2011-04-17 18:30:57.000000000 -0400
@@ -202,7 +202,7 @@
 	Common/UI/libUI.la \
 	Common/GeneralTools/libgt.la \
 	unix/libmfl/libmfl.la \
-	unix/libvisual/libactor_gforce.la
+	unix/libvisual/libactor_gforce.la $(INTLLIBS)
 
 
 EXTRA_DIST = deffont
diff -uNr libvisual-plugins-0.4.0/plugins/actor/JESS/Makefile.in libvisual-plugins-0.4.0-patched/plugins/actor/JESS/Makefile.in
--- libvisual-plugins-0.4.0/plugins/actor/JESS/Makefile.in	2006-03-20 13:48:54.000000000 -0500
+++ libvisual-plugins-0.4.0-patched/plugins/actor/JESS/Makefile.in	2011-04-17 18:30:08.000000000 -0400
@@ -208,7 +208,7 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(actor_plugin_LTLIBRARIES)
 
-actor_JESS_la_LIBADD =
+actor_JESS_la_LIBADD = $(INTLLIBS)
 am_actor_JESS_la_OBJECTS = analyser.lo distorsion.lo draw.lo \
 	draw_low_level.lo jess.lo pal.lo projection.lo renderer.lo
 actor_JESS_la_OBJECTS = $(am_actor_JESS_la_OBJECTS)
diff -uNr libvisual-plugins-0.4.0/plugins/actor/bumpscope/Makefile.in libvisual-plugins-0.4.0-patched/plugins/actor/bumpscope/Makefile.in
--- libvisual-plugins-0.4.0/plugins/actor/bumpscope/Makefile.in	2006-03-20 13:48:54.000000000 -0500
+++ libvisual-plugins-0.4.0-patched/plugins/actor/bumpscope/Makefile.in	2011-04-17 18:22:26.000000000 -0400
@@ -203,7 +203,7 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(actor_plugin_LTLIBRARIES)
 
-actor_bumpscope_la_LIBADD =
+actor_bumpscope_la_LIBADD = $(INTLLIBS)
 am_actor_bumpscope_la_OBJECTS = actor_bumpscope.lo bump_scope.lo
 actor_bumpscope_la_OBJECTS = $(am_actor_bumpscope_la_OBJECTS)
 
diff -uNr libvisual-plugins-0.4.0/plugins/actor/corona/Makefile.in libvisual-plugins-0.4.0-patched/plugins/actor/corona/Makefile.in
--- libvisual-plugins-0.4.0/plugins/actor/corona/Makefile.in	2006-03-20 13:48:54.000000000 -0500
+++ libvisual-plugins-0.4.0-patched/plugins/actor/corona/Makefile.in	2011-04-17 18:22:41.000000000 -0400
@@ -206,7 +206,7 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(actor_plugin_LTLIBRARIES)
 
-actor_corona_la_LIBADD =
+actor_corona_la_LIBADD = $(INTLLIBS)
 am_actor_corona_la_OBJECTS = actor_corona.lo corona.lo palette.lo
 actor_corona_la_OBJECTS = $(am_actor_corona_la_OBJECTS)
 
diff -uNr libvisual-plugins-0.4.0/plugins/actor/infinite/Makefile.in libvisual-plugins-0.4.0-patched/plugins/actor/infinite/Makefile.in
--- libvisual-plugins-0.4.0/plugins/actor/infinite/Makefile.in	2006-03-20 13:48:55.000000000 -0500
+++ libvisual-plugins-0.4.0-patched/plugins/actor/infinite/Makefile.in	2011-04-17 18:23:05.000000000 -0400
@@ -204,7 +204,7 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(actor_plugin_LTLIBRARIES)
 
-actor_infinite_la_LIBADD =
+actor_infinite_la_LIBADD = $(INTLLIBS)
 am_actor_infinite_la_OBJECTS = compute.lo display.lo renderer.lo file.lo \
 	lv_infinite.lo
 actor_infinite_la_OBJECTS = $(am_actor_infinite_la_OBJECTS)
diff -uNr libvisual-plugins-0.4.0/plugins/actor/jakdaw/Makefile.in libvisual-plugins-0.4.0-patched/plugins/actor/jakdaw/Makefile.in
--- libvisual-plugins-0.4.0/plugins/actor/jakdaw/Makefile.in	2006-03-20 13:48:55.000000000 -0500
+++ libvisual-plugins-0.4.0-patched/plugins/actor/jakdaw/Makefile.in	2011-04-17 18:23:20.000000000 -0400
@@ -205,7 +205,7 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(actor_plugin_LTLIBRARIES)
 
-actor_jakdaw_la_LIBADD =
+actor_jakdaw_la_LIBADD = $(INTLLIBS)
 am_actor_jakdaw_la_OBJECTS = actor_jakdaw.lo feedback.lo plotter.lo
 actor_jakdaw_la_OBJECTS = $(am_actor_jakdaw_la_OBJECTS)
 
diff -uNr libvisual-plugins-0.4.0/plugins/actor/lv_gltest/Makefile.in libvisual-plugins-0.4.0-patched/plugins/actor/lv_gltest/Makefile.in
--- libvisual-plugins-0.4.0/plugins/actor/lv_gltest/Makefile.in	2006-03-20 13:48:55.000000000 -0500
+++ libvisual-plugins-0.4.0-patched/plugins/actor/lv_gltest/Makefile.in	2011-04-17 18:24:20.000000000 -0400
@@ -186,7 +186,7 @@
 
 actor_lv_gltest_la_CFLAGS = -I$(top_srcdir) $(LIBVISUAL_CFLAGS) $(X_CFLAGS)
 actor_lv_gltest_la_LDFLAGS = -module -avoid-version 
-actor_lv_gltest_la_LIBS = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU 
+actor_lv_gltest_la_LIBADD = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU $(INTLLIBS)
 actor_lv_gltest_la_SOURCES = actor_lv_gltest.c
 subdir = plugins/actor/lv_gltest
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
@@ -195,7 +195,6 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(actor_plugin_LTLIBRARIES)
 
-actor_lv_gltest_la_LIBADD =
 am_actor_lv_gltest_la_OBJECTS = actor_lv_gltest_la-actor_lv_gltest.lo
 actor_lv_gltest_la_OBJECTS = $(am_actor_lv_gltest_la_OBJECTS)
 
diff -uNr libvisual-plugins-0.4.0/plugins/actor/lv_scope/Makefile.in libvisual-plugins-0.4.0-patched/plugins/actor/lv_scope/Makefile.in
--- libvisual-plugins-0.4.0/plugins/actor/lv_scope/Makefile.in	2006-03-20 13:48:56.000000000 -0500
+++ libvisual-plugins-0.4.0-patched/plugins/actor/lv_scope/Makefile.in	2011-04-17 18:23:55.000000000 -0400
@@ -199,7 +199,7 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(actor_plugin_LTLIBRARIES)
 
-actor_lv_scope_la_LIBADD =
+actor_lv_scope_la_LIBADD = $(INTLLIBS)
 am_actor_lv_scope_la_OBJECTS = actor_lv_scope.lo
 actor_lv_scope_la_OBJECTS = $(am_actor_lv_scope_la_OBJECTS)
 
diff -uNr libvisual-plugins-0.4.0/plugins/actor/madspin/Makefile.in libvisual-plugins-0.4.0-patched/plugins/actor/madspin/Makefile.in
--- libvisual-plugins-0.4.0/plugins/actor/madspin/Makefile.in	2006-03-20 13:48:56.000000000 -0500
+++ libvisual-plugins-0.4.0-patched/plugins/actor/madspin/Makefile.in	2011-04-17 18:21:56.000000000 -0400
@@ -191,9 +191,9 @@
 
 actor_plugin_LTLIBRARIES = actor_madspin.la
 
-actor_madspin_la_CFLAGS = -I$(top_srcdir) $(LIBVISUAL_CFLAGS) -DSTAR_DIR=\""$(imagesdir)/"\"
+actor_madspin_la_CFLAGS = -I$(top_srcdir) $(LIBVISUAL_CFLAGS) -DSTAR_DIR=\""$(imagesdir)/"\" $(X_CFLAGS)
 actor_madspin_la_LDFLAGS = -module -avoid-version 
-actor_madspin_la_LIBS = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL 
+actor_madspin_la_LIBADD = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL $(INTLLIBS)
 actor_madspin_la_SOURCES = madspin.c
 subdir = plugins/actor/madspin
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
@@ -202,7 +202,6 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(actor_plugin_LTLIBRARIES)
 
-actor_madspin_la_LIBADD =
 am_actor_madspin_la_OBJECTS = actor_madspin_la-madspin.lo
 actor_madspin_la_OBJECTS = $(am_actor_madspin_la_OBJECTS)
 
diff -uNr libvisual-plugins-0.4.0/plugins/actor/nastyfft/Makefile.in libvisual-plugins-0.4.0-patched/plugins/actor/nastyfft/Makefile.in
--- libvisual-plugins-0.4.0/plugins/actor/nastyfft/Makefile.in	2006-03-20 13:48:56.000000000 -0500
+++ libvisual-plugins-0.4.0-patched/plugins/actor/nastyfft/Makefile.in	2011-04-17 18:06:43.000000000 -0400
@@ -186,7 +186,7 @@
 
 actor_nastyfft_la_CFLAGS = -I$(top_srcdir) $(LIBVISUAL_CFLAGS) $(X_CFLAGS)
 actor_nastyfft_la_LDFLAGS = -module -avoid-version 
-actor_nastyfft_la_LIBS = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU
+actor_nastyfft_la_LIBADD = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU
 actor_nastyfft_la_SOURCES = actor_nastyfft.c
 subdir = plugins/actor/nastyfft
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
@@ -195,7 +195,6 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(actor_plugin_LTLIBRARIES)
 
-actor_nastyfft_la_LIBADD =
 am_actor_nastyfft_la_OBJECTS = actor_nastyfft_la-actor_nastyfft.lo
 actor_nastyfft_la_OBJECTS = $(am_actor_nastyfft_la_OBJECTS)
 
diff -uNr libvisual-plugins-0.4.0/plugins/actor/oinksie/table.c libvisual-plugins-0.4.0-patched/plugins/actor/oinksie/table.c
--- libvisual-plugins-0.4.0/plugins/actor/oinksie/table.c	2006-01-22 08:25:26.000000000 -0500
+++ libvisual-plugins-0.4.0-patched/plugins/actor/oinksie/table.c	2011-04-18 04:35:42.000000000 -0400
@@ -26,6 +26,11 @@
 #include "common.h"
 #include "table.h"
 
+float _oink_table_sin[OINK_TABLE_NORMAL_SIZE];
+float _oink_table_cos[OINK_TABLE_NORMAL_SIZE];
+float _oink_table_sinlarge[OINK_TABLE_LARGE_SIZE];
+float _oink_table_coslarge[OINK_TABLE_LARGE_SIZE];
+
 void _oink_table_init ()
 {
 	float adder;
diff -uNr libvisual-plugins-0.4.0/plugins/actor/oinksie/table.h libvisual-plugins-0.4.0-patched/plugins/actor/oinksie/table.h
--- libvisual-plugins-0.4.0/plugins/actor/oinksie/table.h	2006-01-22 08:25:26.000000000 -0500
+++ libvisual-plugins-0.4.0-patched/plugins/actor/oinksie/table.h	2011-04-17 13:50:37.000000000 -0400
@@ -27,10 +27,10 @@
 #define OINK_TABLE_NORMAL_SIZE	1200
 #define OINK_TABLE_LARGE_SIZE	12000
 
-float _oink_table_sin[OINK_TABLE_NORMAL_SIZE];
-float _oink_table_cos[OINK_TABLE_NORMAL_SIZE];
-float _oink_table_sinlarge[OINK_TABLE_LARGE_SIZE];
-float _oink_table_coslarge[OINK_TABLE_LARGE_SIZE];
+extern float _oink_table_sin[OINK_TABLE_NORMAL_SIZE];
+extern float _oink_table_cos[OINK_TABLE_NORMAL_SIZE];
+extern float _oink_table_sinlarge[OINK_TABLE_LARGE_SIZE];
+extern float _oink_table_coslarge[OINK_TABLE_LARGE_SIZE];
 
 void _oink_table_init ();
 
diff -uNr libvisual-plugins-0.4.0/plugins/actor/pseudotoad_flower/Makefile.in libvisual-plugins-0.4.0-patched/plugins/actor/pseudotoad_flower/Makefile.in
--- libvisual-plugins-0.4.0/plugins/actor/pseudotoad_flower/Makefile.in	2006-03-20 13:48:57.000000000 -0500
+++ libvisual-plugins-0.4.0-patched/plugins/actor/pseudotoad_flower/Makefile.in	2011-04-17 18:24:42.000000000 -0400
@@ -186,7 +186,7 @@
 
 actor_flower_la_LDFLAGS = -module -avoid-version 
 actor_flower_la_CFLAGS = -I$(top_srcdir) $(LIBVISUAL_CFLAGS) $(X_CFLAGS)
-actor_flower_la_LIBS = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU 
+actor_flower_la_LIBADD = $(LIBVISUAL_LIBS) $(X_LIBS) -lGL -lGLU $(INTLLIBS)
 
 actor_flower_la_SOURCES = \
 	actor_flower.c \
@@ -202,7 +202,6 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(actor_plugin_LTLIBRARIES)
 
-actor_flower_la_LIBADD =
 am_actor_flower_la_OBJECTS = actor_flower_la-actor_flower.lo \
 	actor_flower_la-main.lo actor_flower_la-notch.lo
 actor_flower_la_OBJECTS = $(am_actor_flower_la_OBJECTS)
diff -uNr libvisual-plugins-0.4.0/plugins/input/mplayer/Makefile.in libvisual-plugins-0.4.0-patched/plugins/input/mplayer/Makefile.in
--- libvisual-plugins-0.4.0/plugins/input/mplayer/Makefile.in	2006-03-20 13:48:58.000000000 -0500
+++ libvisual-plugins-0.4.0-patched/plugins/input/mplayer/Makefile.in	2011-04-17 18:25:22.000000000 -0400
@@ -198,7 +198,7 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(input_plugin_LTLIBRARIES)
 
-input_mplayer_la_LIBADD =
+input_mplayer_la_LIBADD = $(INTLLIBS)
 am_input_mplayer_la_OBJECTS = input_mplayer.lo
 input_mplayer_la_OBJECTS = $(am_input_mplayer_la_OBJECTS)
 
