diff -Nurd -x'*~' libxslt-1.1.26.orig/doc/Makefile.in libxslt-1.1.26/doc/Makefile.in
--- libxslt-1.1.26.orig/doc/Makefile.in	2009-09-24 10:43:14.000000000 -0400
+++ libxslt-1.1.26/doc/Makefile.in	2011-03-10 00:21:41.000000000 -0500
@@ -670,13 +670,18 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(HTML_DIR)
 	-@INSTALL@ -m 0644 $(srcdir)/*.html $(DESTDIR)$(HTML_DIR)
+	-@INSTALL@ -m 0644 $(srcdir)/*.xml $(DESTDIR)$(HTML_DIR)
+	-@INSTALL@ -m 0644 $(srcdir)/*.xsl $(DESTDIR)$(HTML_DIR)
 	-@INSTALL@ -m 0644 $(srcdir)/*.gif $(DESTDIR)$(HTML_DIR)
 	$(mkinstalldirs) $(DESTDIR)$(HTML_DIR)/html
 	-@INSTALL@ -m 0644 $(srcdir)/html/*.html $(DESTDIR)$(HTML_DIR)/html
 	-@INSTALL@ -m 0644 $(srcdir)/html/*.png $(DESTDIR)$(HTML_DIR)/html
-	-@INSTALL@ -m 0644 $(srcdir)/html/index.sgml $(DESTDIR)$(HTML_DIR)/html
 	$(mkinstalldirs) $(DESTDIR)$(HTML_DIR)/EXSLT
 	-@INSTALL@ -m 0644 $(srcdir)/EXSLT/*.html $(DESTDIR)$(HTML_DIR)/EXSLT
+	-@INSTALL@ -m 0644 $(srcdir)/EXSLT/*.xml $(DESTDIR)$(HTML_DIR)/EXSLT
+	$(mkinstalldirs) $(DESTDIR)$(HTML_DIR)/EXSLT/html
+	-@INSTALL@ -m 0644 $(srcdir)/EXSLT/html/*.html $(DESTDIR)$(HTML_DIR)/EXSLT/html
+	-@INSTALL@ -m 0644 $(srcdir)/EXSLT/html/*.png $(DESTDIR)$(HTML_DIR)/EXSLT/html
 	$(mkinstalldirs) $(DESTDIR)$(HTML_DIR)/tutorial
 	-@INSTALL@ -m 0644 $(srcdir)/tutorial/* $(DESTDIR)$(HTML_DIR)/tutorial
 	$(mkinstalldirs) $(DESTDIR)$(HTML_DIR)/tutorial2
diff -Nurd -x'*~' libxslt-1.1.26.orig/libexslt/date.c libxslt-1.1.26/libexslt/date.c
--- libxslt-1.1.26.orig/libexslt/date.c	2009-09-17 10:51:10.000000000 -0400
+++ libxslt-1.1.26/libexslt/date.c	2011-03-10 00:19:04.000000000 -0500
@@ -28,11 +28,6 @@
 #include "config.h"
 #endif
 
-#if HAVE_LOCALTIME_R	/* _POSIX_SOURCE required by gnu libc */
-#ifndef _AIX51		/* but on AIX we're not using gnu libc */
-#define _POSIX_SOURCE
-#endif
-#endif
 
 #include <libxml/tree.h>
 #include <libxml/xpath.h>
