diff -Nurd xine-lib-1.1.13/configure xine-lib-1.1.13-new/configure
--- xine-lib-1.1.13/configure	2008-06-14 19:16:03.000000000 -0400
+++ xine-lib-1.1.13-new/configure	2009-10-09 23:33:06.000000000 -0400
@@ -30693,10 +30693,10 @@
 #define LIBX11_SO "${x_lib_location:-libX11.so}"
 _ACEOF
 
-x_lib_location="`ls -1 "${x_libraries:-/usr/local/lib}/libXv.so"*  "${x_libraries:-/usr/lib}/libXv.so"*  2>/dev/null | sed -e \"${soname_script}\"`"
+x_lib_location="`ls -1 "${x_libraries:-/usr/local/lib}/libXv.dylib"*  "${x_libraries:-/usr/lib}/libXv.dylib"*  2>/dev/null | sed -e \"${soname_script}\"`"
 
 cat >>confdefs.h <<_ACEOF
-#define LIBXV_SO "${x_lib_location:-libXv.so}"
+#define LIBXV_SO "${x_lib_location:-libXv.dylib}"
 _ACEOF
 
 
@@ -32278,20 +32278,20 @@
 
   if test "x$xv_prefer_shared" = "xyes"; then
 
-  { echo "$as_me:$LINENO: checking for libXv.so" >&5
-echo $ECHO_N "checking for libXv.so... $ECHO_C" >&6; }
-  if test -f "$xv_path/libXv.so"; then
-    { echo "$as_me:$LINENO: result: found libXv.so in $xv_path" >&5
-echo "${ECHO_T}found libXv.so in $xv_path" >&6; }
-    XV_LIBS="libXv.so"
+  { echo "$as_me:$LINENO: checking for libXv.dylib" >&5
+echo $ECHO_N "checking for libXv.dylib... $ECHO_C" >&6; }
+  if test -f "$xv_path/libXv.dylib"; then
+    { echo "$as_me:$LINENO: result: found libXv.dylib in $xv_path" >&5
+echo "${ECHO_T}found libXv.dylib in $xv_path" >&6; }
+    XV_LIBS="libXv.dylib"
   else
-    if test -f "/usr/lib/libXv.so"; then
-      { echo "$as_me:$LINENO: result: found libXv.so in /usr/lib" >&5
-echo "${ECHO_T}found libXv.so in /usr/lib" >&6; }
-      XV_LIBS="libXv.so"
+    if test -f "/usr/lib/libXv.dylib"; then
+      { echo "$as_me:$LINENO: result: found libXv.dylib in /usr/lib" >&5
+echo "${ECHO_T}found libXv.dylib in /usr/lib" >&6; }
+      XV_LIBS="libXv.dylib"
     else
-      { echo "$as_me:$LINENO: result: libXv.so not found in $xv_path" >&5
-echo "${ECHO_T}libXv.so not found in $xv_path" >&6; }
+      { echo "$as_me:$LINENO: result: libXv.dylib not found in $xv_path" >&5
+echo "${ECHO_T}libXv.dylib not found in $xv_path" >&6; }
     fi
   fi
 
@@ -32320,20 +32320,20 @@
   if test x$XV_LIBS = x; then
     if ! test "x$xv_prefer_shared" = "xyes"; then
 
-  { echo "$as_me:$LINENO: checking for libXv.so" >&5
-echo $ECHO_N "checking for libXv.so... $ECHO_C" >&6; }
-  if test -f "$xv_path/libXv.so"; then
-    { echo "$as_me:$LINENO: result: found libXv.so in $xv_path" >&5
-echo "${ECHO_T}found libXv.so in $xv_path" >&6; }
-    XV_LIBS="libXv.so"
+  { echo "$as_me:$LINENO: checking for libXv.dylib" >&5
+echo $ECHO_N "checking for libXv.dylib... $ECHO_C" >&6; }
+  if test -f "$xv_path/libXv.dylib"; then
+    { echo "$as_me:$LINENO: result: found libXv.dylib in $xv_path" >&5
+echo "${ECHO_T}found libXv.dylib in $xv_path" >&6; }
+    XV_LIBS="libXv.dylib"
   else
-    if test -f "/usr/lib/libXv.so"; then
-      { echo "$as_me:$LINENO: result: found libXv.so in /usr/lib" >&5
-echo "${ECHO_T}found libXv.so in /usr/lib" >&6; }
-      XV_LIBS="libXv.so"
+    if test -f "/usr/lib/libXv.dylib"; then
+      { echo "$as_me:$LINENO: result: found libXv.dylib in /usr/lib" >&5
+echo "${ECHO_T}found libXv.dylib in /usr/lib" >&6; }
+      XV_LIBS="libXv.dylib"
     else
-      { echo "$as_me:$LINENO: result: libXv.so not found in $xv_path" >&5
-echo "${ECHO_T}libXv.so not found in $xv_path" >&6; }
+      { echo "$as_me:$LINENO: result: libXv.dylib not found in $xv_path" >&5
+echo "${ECHO_T}libXv.dylib not found in $xv_path" >&6; }
     fi
   fi
 
@@ -32441,8 +32441,8 @@
         ac_have_xv_static="yes"
         XV_LIBS="$xv_path/$XV_LIBS"
         ;;
-      x*.so)
-        XV_LIBS=`echo $XV_LIBS | sed 's/^lib/-l/; s/\.so$//'`
+      x*.dylib)
+        XV_LIBS=`echo $XV_LIBS | sed 's/^lib/-l/; s/.dylib$//'`
         ;;
       *)
         { { echo "$as_me:$LINENO: error: sorry, I don't know about $XV_LIBS" >&5
@@ -32475,20 +32475,20 @@
 
   if test "x$xv_prefer_shared" = "xyes"; then
 
-  { echo "$as_me:$LINENO: checking for libXv.so" >&5
-echo $ECHO_N "checking for libXv.so... $ECHO_C" >&6; }
-  if test -f "$xv_path/libXv.so"; then
-    { echo "$as_me:$LINENO: result: found libXv.so in $xv_path" >&5
-echo "${ECHO_T}found libXv.so in $xv_path" >&6; }
-    XV_LIBS="libXv.so"
+  { echo "$as_me:$LINENO: checking for libXv.dylib" >&5
+echo $ECHO_N "checking for libXv.dylib... $ECHO_C" >&6; }
+  if test -f "$xv_path/libXv.dylib"; then
+    { echo "$as_me:$LINENO: result: found libXv.dylib in $xv_path" >&5
+echo "${ECHO_T}found libXv.dylib in $xv_path" >&6; }
+    XV_LIBS="libXv.dylib"
   else
-    if test -f "/usr/lib/libXv.so"; then
-      { echo "$as_me:$LINENO: result: found libXv.so in /usr/lib" >&5
-echo "${ECHO_T}found libXv.so in /usr/lib" >&6; }
-      XV_LIBS="libXv.so"
+    if test -f "/usr/lib/libXv.dylib"; then
+      { echo "$as_me:$LINENO: result: found libXv.dylib in /usr/lib" >&5
+echo "${ECHO_T}found libXv.dylib in /usr/lib" >&6; }
+      XV_LIBS="libXv.dylib"
     else
-      { echo "$as_me:$LINENO: result: libXv.so not found in $xv_path" >&5
-echo "${ECHO_T}libXv.so not found in $xv_path" >&6; }
+      { echo "$as_me:$LINENO: result: libXv.dylib not found in $xv_path" >&5
+echo "${ECHO_T}libXv.dylib not found in $xv_path" >&6; }
     fi
   fi
 
@@ -32517,20 +32517,20 @@
   if test x$XV_LIBS = x; then
     if ! test "x$xv_prefer_shared" = "xyes"; then
 
-  { echo "$as_me:$LINENO: checking for libXv.so" >&5
-echo $ECHO_N "checking for libXv.so... $ECHO_C" >&6; }
-  if test -f "$xv_path/libXv.so"; then
-    { echo "$as_me:$LINENO: result: found libXv.so in $xv_path" >&5
-echo "${ECHO_T}found libXv.so in $xv_path" >&6; }
-    XV_LIBS="libXv.so"
+  { echo "$as_me:$LINENO: checking for libXv.dylib" >&5
+echo $ECHO_N "checking for libXv.dylib... $ECHO_C" >&6; }
+  if test -f "$xv_path/libXv.dylib"; then
+    { echo "$as_me:$LINENO: result: found libXv.dylib in $xv_path" >&5
+echo "${ECHO_T}found libXv.dylib in $xv_path" >&6; }
+    XV_LIBS="libXv.dylib"
   else
-    if test -f "/usr/lib/libXv.so"; then
-      { echo "$as_me:$LINENO: result: found libXv.so in /usr/lib" >&5
-echo "${ECHO_T}found libXv.so in /usr/lib" >&6; }
-      XV_LIBS="libXv.so"
+    if test -f "/usr/lib/libXv.dylib"; then
+      { echo "$as_me:$LINENO: result: found libXv.dylib in /usr/lib" >&5
+echo "${ECHO_T}found libXv.dylib in /usr/lib" >&6; }
+      XV_LIBS="libXv.dylib"
     else
-      { echo "$as_me:$LINENO: result: libXv.so not found in $xv_path" >&5
-echo "${ECHO_T}libXv.so not found in $xv_path" >&6; }
+      { echo "$as_me:$LINENO: result: libXv.dylib not found in $xv_path" >&5
+echo "${ECHO_T}libXv.dylib not found in $xv_path" >&6; }
     fi
   fi
 
@@ -32638,8 +32638,8 @@
         ac_have_xv_static="yes"
         XV_LIBS="$xv_path/$XV_LIBS"
         ;;
-      x*.so)
-        XV_LIBS=`echo $XV_LIBS | sed 's/^lib/-l/; s/\.so$//'`
+      x*.dylib)
+        XV_LIBS=`echo $XV_LIBS | sed 's/^lib/-l/; s/.dylib$//'`
         ;;
       *)
         { { echo "$as_me:$LINENO: error: sorry, I don't know about $XV_LIBS" >&5
@@ -32695,7 +32695,7 @@
   hppa*)
 	if test "x$ac_have_xv_static" = "xyes"; then
 	  echo "warning: hppa linker - disabling static libXv"
-	  XV_LIBS="libXv.so"
+	  XV_LIBS="libXv.dylib"
 	fi
 	;;
 
diff -Nurd xine-lib-1.1.13/src/audio_out/Makefile.in xine-lib-1.1.13-new/src/audio_out/Makefile.in
--- xine-lib-1.1.13/src/audio_out/Makefile.in	2008-06-14 19:15:47.000000000 -0400
+++ xine-lib-1.1.13-new/src/audio_out/Makefile.in	2009-10-09 23:33:13.000000000 -0400
@@ -639,11 +639,11 @@
 # The "-Wl,-framework -Wl,..." is needed for libtool versions before
 # 1.5.x (1.257): the default version that ships with Mac OS X is 1.5 (1.1220)
 xineplug_ao_out_coreaudio_la_LDFLAGS = \
-	-Wl,-framework -Wl,Cocoa -framework CoreAudio \
+	-Wl,-framework -Wl,Cocoa -isysroot /Developer/SDKs/MacOSX10.5.sdk -framework CoreAudio \
 	-Wl,-framework -Wl,AudioUnit -framework AudioUnit \
 	-avoid-version -module
 
-xineplug_ao_out_coreaudio_la_CFLAGS = -framework CoreAudio -framework AudioUnit $(VISIBILITY_FLAG)
+xineplug_ao_out_coreaudio_la_CFLAGS = -isysroot /Developer/SDKs/MacOSX10.5.sdk -framework CoreAudio -framework AudioUnit $(VISIBILITY_FLAG)
 xineplug_ao_out_pulseaudio_la_SOURCES = audio_pulse_out.c
 xineplug_ao_out_pulseaudio_la_LIBADD = $(XINE_LIB) $(PULSEAUDIO_LIBS) $(LTLIBINTL)
 xineplug_ao_out_pulseaudio_la_CFLAGS = $(VISIBILITY_FLAG) $(PULSEAUDIO_CFLAGS)
diff -Nurd xine-lib-1.1.13/src/post/deinterlace/plugins/Makefile.in xine-lib-1.1.13-new/src/post/deinterlace/plugins/Makefile.in
--- xine-lib-1.1.13/src/post/deinterlace/plugins/Makefile.in	2008-06-14 19:15:57.000000000 -0400
+++ xine-lib-1.1.13-new/src/post/deinterlace/plugins/Makefile.in	2009-10-09 23:30:57.000000000 -0400
@@ -66,9 +66,7 @@
 	libdeinterlaceplugins_la-linearblend.lo \
 	libdeinterlaceplugins_la-vfir.lo \
 	libdeinterlaceplugins_la-weave.lo \
-	libdeinterlaceplugins_la-greedy2frame.lo \
 	libdeinterlaceplugins_la-scalerbob.lo \
-	libdeinterlaceplugins_la-kdetv_greedyh.lo \
 	libdeinterlaceplugins_la-kdetv_tomsmocomp.lo
 libdeinterlaceplugins_la_OBJECTS =  \
 	$(am_libdeinterlaceplugins_la_OBJECTS)
@@ -418,7 +416,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
-EXTRA_DIST = greedy2frame_template.c greedyh.asm \
+EXTRA_DIST =  greedyh.asm \
 	tomsmocomp/SearchLoop0A.inc tomsmocomp/SearchLoopBottom.inc \
 	tomsmocomp/SearchLoopEdgeA.inc tomsmocomp/SearchLoopEdgeA8.inc \
 	tomsmocomp/SearchLoopOddA.inc tomsmocomp/SearchLoopOddA2.inc \
@@ -442,9 +440,7 @@
 	linearblend.c \
 	vfir.c \
 	weave.c \
-	greedy2frame.c \
 	scalerbob.c \
-	kdetv_greedyh.c \
 	kdetv_tomsmocomp.c
 
 libdeinterlaceplugins_la_CFLAGS = $(VISIBILITY_FLAG) $(AM_CFLAGS)
@@ -503,8 +499,6 @@
 
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdeinterlaceplugins_la-double.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdeinterlaceplugins_la-greedy.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdeinterlaceplugins_la-greedy2frame.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdeinterlaceplugins_la-kdetv_greedyh.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdeinterlaceplugins_la-kdetv_tomsmocomp.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdeinterlaceplugins_la-linear.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdeinterlaceplugins_la-linearblend.Plo@am__quote@
