diff -Nurd xine-lib-1.1.19/configure xine-lib-1.1.19-new/configure
--- xine-lib-1.1.19/configure	2010-07-25 10:40:24.000000000 -0400
+++ xine-lib-1.1.19-new/configure	2010-10-03 18:15:19.000000000 -0400
@@ -19871,7 +19871,7 @@
 
 
 
-host_or_hostalias="$host"
+host_or_hostalias="$target"
 if test "$host_or_hostalias" = ""; then
                             host_or_hostalias="$host_alias"
 fi
diff -Nurd xine-lib-1.1.19/src/audio_out/Makefile.in xine-lib-1.1.19-new/src/audio_out/Makefile.in
--- xine-lib-1.1.19/src/audio_out/Makefile.in	2010-07-25 10:40:01.000000000 -0400
+++ xine-lib-1.1.19-new/src/audio_out/Makefile.in	2010-10-03 18:38:36.000000000 -0400
@@ -691,11 +691,12 @@
 # The "-Wl,-framework -Wl,..." is needed for libtool versions before
 # 1.5.x (1.257): the default version that ships with Mac OS X is 1.5 (1.1220)
 xineplug_ao_out_coreaudio_la_LDFLAGS = \
-	-Wl,-framework -Wl,Cocoa -framework CoreAudio \
-	-Wl,-framework -Wl,AudioUnit -framework AudioUnit \
-	-avoid-version -module
+	-F/Developer/SDKs/MacOSX10.5.sdk/System/Library/Frameworks -F/Developer/SDKs/MacOSX10.4u.sdk/System/Library/Frameworks \
+	-Wl,-framework,Cocoa -Wl,-framework,CoreAudio \
+	-Wl,-framework,AudioUnit -avoid-version -module
 
-xineplug_ao_out_coreaudio_la_CFLAGS = -framework CoreAudio -framework AudioUnit $(VISIBILITY_FLAG)
+xineplug_ao_out_coreaudio_la_CFLAGS = -F/Developer/SDKs/MacOSX10.5.sdk/System/Library/Frameworks -F/Developer/SDKs/MacOSX10.4u.sdk/System/Library/Frameworks \
+	-Wl,-framework,CoreAudio -Wl,-framework,AudioUnit $(VISIBILITY_FLAG)
 xineplug_ao_out_pulseaudio_la_SOURCES = audio_pulse_out.c
 xineplug_ao_out_pulseaudio_la_LIBADD = $(XINE_LIB) $(PULSEAUDIO_LIBS) $(LTLIBINTL)
 xineplug_ao_out_pulseaudio_la_CFLAGS = $(VISIBILITY_FLAG) $(PULSEAUDIO_CFLAGS)
diff -Nurd xine-lib-1.1.19/src/libxinevdec/Makefile.in xine-lib-1.1.19-new/src/libxinevdec/Makefile.in
--- xine-lib-1.1.19/src/libxinevdec/Makefile.in	2010-07-25 10:40:16.000000000 -0400
+++ xine-lib-1.1.19-new/src/libxinevdec/Makefile.in	2010-10-03 18:18:03.000000000 -0400
@@ -483,7 +483,7 @@
 xinepostdir = $(XINE_PLUGINDIR)/post
 vidixdir = $(XINE_PLUGINDIR)/vidix
 AM_CFLAGS = $(VISIBILITY_FLAG)
-AM_LDFLAGS = $(xineplug_ldflags)
+AM_LDFLAGS = $(xineplug_ldflags) $(X_LIBS)
 EXTRA_DIST = foovideo.c
 @HAVE_WAND_TRUE@image_module = xineplug_decode_image.la
 @HAVE_GDK_PIXBUF_TRUE@gdkpixbuf_module = xineplug_decode_gdk_pixbuf.la
diff -Nurd xine-lib-1.1.19/src/post/deinterlace/plugins/Makefile.in xine-lib-1.1.19-new/src/post/deinterlace/plugins/Makefile.in
--- xine-lib-1.1.19/src/post/deinterlace/plugins/Makefile.in	2010-07-25 10:40:17.000000000 -0400
+++ xine-lib-1.1.19-new/src/post/deinterlace/plugins/Makefile.in	2010-10-03 18:18:47.000000000 -0400
@@ -72,9 +72,7 @@
 	libdeinterlaceplugins_la-linearblend.lo \
 	libdeinterlaceplugins_la-vfir.lo \
 	libdeinterlaceplugins_la-weave.lo \
-	libdeinterlaceplugins_la-greedy2frame.lo \
 	libdeinterlaceplugins_la-scalerbob.lo \
-	libdeinterlaceplugins_la-kdetv_greedyh.lo \
 	libdeinterlaceplugins_la-kdetv_tomsmocomp.lo
 libdeinterlaceplugins_la_OBJECTS =  \
 	$(am_libdeinterlaceplugins_la_OBJECTS)
@@ -463,9 +461,7 @@
 	linearblend.c \
 	vfir.c \
 	weave.c \
-	greedy2frame.c \
 	scalerbob.c \
-	kdetv_greedyh.c \
 	kdetv_tomsmocomp.c
 
 libdeinterlaceplugins_la_CFLAGS = $(VISIBILITY_FLAG) $(AM_CFLAGS)
@@ -525,8 +521,6 @@
 
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdeinterlaceplugins_la-double.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdeinterlaceplugins_la-greedy.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdeinterlaceplugins_la-greedy2frame.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdeinterlaceplugins_la-kdetv_greedyh.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdeinterlaceplugins_la-kdetv_tomsmocomp.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdeinterlaceplugins_la-linear.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libdeinterlaceplugins_la-linearblend.Plo@am__quote@
diff -Nurd xine-lib-1.1.19/src/xine-engine/Makefile.in xine-lib-1.1.19-new/src/xine-engine/Makefile.in
--- xine-lib-1.1.19/src/xine-engine/Makefile.in	2010-07-25 10:40:20.000000000 -0400
+++ xine-lib-1.1.19-new/src/xine-engine/Makefile.in	2010-10-03 21:22:15.000000000 -0400
@@ -441,7 +441,7 @@
 vidixdir = $(XINE_PLUGINDIR)/vidix
 LIBXINEPOSIX = $(top_builddir)/lib/libxineposix.la
 AM_CFLAGS = $(X_CFLAGS) $(FT2_CFLAGS) $(FONTCONFIG_CFLAGS) $(VISIBILITY_FLAG)
-AM_CPPFLAGS = $(ZLIB_CPPFLAGS) -DXINE_LIBRARY_COMPILE
+AM_CPPFLAGS = $(ZLIB_CPPFLAGS) -DXINE_LIBRARY_COMPILE $(FT2_CFLAGS) $(FONTCONFIG_CFLAGS)
 lib_LTLIBRARIES = libxine.la
 XINEUTILS_LIB = $(top_builddir)/src/xine-utils/libxineutils.la
 DEF_FILE = libxine-$(XINE_MAJOR).def
diff -Nurd xine-lib-1.1.19/src/xine-engine/demux.c xine-lib-1.1.19-new/src/xine-engine/demux.c
--- xine-lib-1.1.19/src/xine-engine/demux.c	2010-07-21 15:10:49.000000000 -0400
+++ xine-lib-1.1.19-new/src/xine-engine/demux.c	2010-10-03 18:19:08.000000000 -0400
@@ -50,6 +50,27 @@
 #include <winsock.h>
 #endif
 
+#ifndef CLOCK_REALTIME
+#include <time.h>
+
+#define  CLOCK_REALTIME 0 
+static int clock_gettime(int clk_id /*ignored*/, struct timespec *tp)
+{
+  struct timeval now;
+  
+  int rv = gettimeofday(&now, NULL);
+  
+  if (rv != 0) {
+    return rv;
+  }
+  
+  tp->tv_sec = now.tv_sec;
+  tp->tv_nsec = now.tv_usec * 1000;
+  
+  return 0;
+}
+#endif
+
 /*
  *  Flush audio and video buffers. It is called from demuxers on
  *  seek/stop, and may be useful when user input changes a stream and
