diff -uNr xine-lib-1.1.1/configure xine-lib-1.1.1-new/configure
--- xine-lib-1.1.1/configure	2005-11-15 06:40:27.000000000 -0500
+++ xine-lib-1.1.1-new/configure	2006-07-02 12:17:45.000000000 -0400
@@ -29793,15 +29793,15 @@
 
   if test "x$xv_prefer_shared" = "xyes"; then
 
-  echo "$as_me:$LINENO: checking for libXv.so" >&5
-echo $ECHO_N "checking for libXv.so... $ECHO_C" >&6
-  if test -f "$xv_path/libXv.so"; then
-    echo "$as_me:$LINENO: result: found libXv.so in $xv_path" >&5
-echo "${ECHO_T}found libXv.so in $xv_path" >&6
-    XV_LIB="libXv.so"
+  echo "$as_me:$LINENO: checking for libXv.dylib" >&5
+echo $ECHO_N "checking for libXv.dylib... $ECHO_C" >&6
+  if test -f "$xv_path/libXv.dylib"; then
+    echo "$as_me:$LINENO: result: found libXv.dylib in $xv_path" >&5
+echo "${ECHO_T}found libXv.dylib in $xv_path" >&6
+    XV_LIB="libXv.dylib"
   else
-    echo "$as_me:$LINENO: result: libXv.so not found in $xv_path" >&5
-echo "${ECHO_T}libXv.so not found in $xv_path" >&6
+    echo "$as_me:$LINENO: result: libXv.dylib not found in $xv_path" >&5
+echo "${ECHO_T}libXv.dylib not found in $xv_path" >&6
   fi
 
   else
@@ -29823,15 +29823,15 @@
   if test x$XV_LIB = x; then
     if ! test "x$xv_prefer_shared" = "xyes"; then
 
-  echo "$as_me:$LINENO: checking for libXv.so" >&5
-echo $ECHO_N "checking for libXv.so... $ECHO_C" >&6
-  if test -f "$xv_path/libXv.so"; then
-    echo "$as_me:$LINENO: result: found libXv.so in $xv_path" >&5
-echo "${ECHO_T}found libXv.so in $xv_path" >&6
-    XV_LIB="libXv.so"
+  echo "$as_me:$LINENO: checking for libXv.dylib" >&5
+echo $ECHO_N "checking for libXv.dylib... $ECHO_C" >&6
+  if test -f "$xv_path/libXv.dylib"; then
+    echo "$as_me:$LINENO: result: found libXv.dylib in $xv_path" >&5
+echo "${ECHO_T}found libXv.dylib in $xv_path" >&6
+    XV_LIB="libXv.dylib"
   else
-    echo "$as_me:$LINENO: result: libXv.so not found in $xv_path" >&5
-echo "${ECHO_T}libXv.so not found in $xv_path" >&6
+    echo "$as_me:$LINENO: result: libXv.dylib not found in $xv_path" >&5
+echo "${ECHO_T}libXv.dylib not found in $xv_path" >&6
   fi
 
     else
@@ -29935,8 +29935,8 @@
         ac_have_xv_static="yes"
         XV_LIB="$xv_path/$XV_LIB"
         ;;
-      x*.so)
-        XV_LIB=`echo $XV_LIB | sed 's/^lib/-l/; s/\.so$//'`
+      x*.dylib)
+        XV_LIB=`echo $XV_LIB | sed 's/^lib/-l/; s/\.dylib$//'`
         ;;
       *)
         { { echo "$as_me:$LINENO: error: sorry, I don't know about $XV_LIB" >&5
@@ -29980,7 +29980,7 @@
   hppa*)
 	if test x$ac_have_xv_static = "xyes"; then
 	  echo "warning: hppa linker - disabling static libXv"
-	  XV_LIB="libXv.so"
+	  XV_LIB="libXv.dylib"
 	fi
 	;;
 
@@ -46268,7 +46268,7 @@
         DEBUG_CFLAGS="-O3 -mieee $DEBUG_CFLAGS"
         ;;
       *darwin*)
-        CFLAGS="-O3 -pipe -fomit-frame-pointer $m_wm $m_psb -fexpensive-optimizations $f_si $f_nsa -ffast-math $INLINE_FUNCTIONS -no-cpp-precomp -D_INTL_REDIRECT_MACROS $CFLAGS"
+        CFLAGS="-Os -pipe -fno-strict-aliasing -ffast-math $INLINE_FUNCTIONS -no-cpp-precomp -D_INTL_REDIRECT_MACROS $CFLAGS"
         DEBUG_CFLAGS="-O3 $DEBUG_CFLAGS"
         ;;
       ppc-*-linux* | powerpc-*)
@@ -46339,7 +46339,7 @@
 enable_armv4l="no"
 
 case "$host_or_hostalias" in
-  i?86-* | k?-* | athlon-* | pentium*-)
+  i?86-*-linux* | k?-* | athlon-* | pentium*-)
 
 cat >>confdefs.h <<_ACEOF
 #define ARCH_X86
@@ -46354,6 +46354,10 @@
     enable_w32dll="yes"
 
     case "$host_or_hostalias" in
+      *-*-darwin)
+        enable_ffmmx="no"
+        enable_w32dll="no"
+        ;;
       *-*-mingw*)
         enable_w32dll="no"
         ;;
diff -uNr xine-lib-1.1.1/misc/cdda_server.c xine-lib-1.1.1-new/misc/cdda_server.c
--- xine-lib-1.1.1/misc/cdda_server.c	2004-12-12 17:00:50.000000000 -0500
+++ xine-lib-1.1.1-new/misc/cdda_server.c	2006-07-02 12:17:43.000000000 -0400
@@ -335,7 +335,9 @@
 
   /* dlopening libdvdcss */
 
-#ifndef _MSC_VER
+#if defined(__APPLE__)
+  dvdcss_library = dlopen("libdvdcss.2.dylib", RTLD_LAZY);
+#elif !defined(_MSC_VER)
   dvdcss_library = dlopen("libdvdcss.so.2", RTLD_LAZY);
 #else
   dvdcss_library = dlopen("libdvdcss.dll", RTLD_LAZY);
@@ -371,7 +373,7 @@
       dvdcss_library = NULL;
     } else if(!dvd_open  || !dvd_close || !dvd_title || !dvd_seek
 	      || !dvd_read || !dvd_error || !dvdcss_version) {
-      fprintf(stderr,  "libdvdread: Missing symbols in libdvdcss.so.2, "
+      fprintf(stderr,  "libdvdread: Missing symbols in libdvdcss, "
 	      "this shouldn't happen !\n");
       dlclose(dvdcss_library);
     }
diff -uNr xine-lib-1.1.1/src/input/input_dvd.c xine-lib-1.1.1-new/src/input/input_dvd.c
--- xine-lib-1.1.1/src/input/input_dvd.c	2005-10-14 17:02:16.000000000 -0400
+++ xine-lib-1.1.1-new/src/input/input_dvd.c	2006-07-02 12:17:43.000000000 -0400
@@ -1784,7 +1784,11 @@
 					       "DVD drive, which you intend to use for playing DVDs."),
 					     10, device_change_cb, (void *)this);
   
+#ifdef __APPLE__
+  if ((dvdcss = dlopen("libdvdcss.2.dylib", RTLD_LAZY)) != NULL) {
+#else
   if ((dvdcss = dlopen("libdvdcss.so.2", RTLD_LAZY)) != NULL) {
+#endif
     /* we have found libdvdcss, enable the specific config options */
     char *raw_device;
     static char *decrypt_modes[] = { "key", "disc", "title", NULL };
diff -uNr xine-lib-1.1.1/src/input/libdvdnav/bswap.h xine-lib-1.1.1-new/src/input/libdvdnav/bswap.h
--- xine-lib-1.1.1/src/input/libdvdnav/bswap.h	2003-05-26 19:11:44.000000000 -0400
+++ xine-lib-1.1.1-new/src/input/libdvdnav/bswap.h	2006-07-02 12:17:43.000000000 -0400
@@ -41,6 +41,12 @@
 #define B2N_32(x) x = bswap_32(x)
 #define B2N_64(x) x = bswap_64(x)
 
+#elif defined(__APPLE__)
+#include <libkern/OSByteOrder.h>
+#define B2N_16(x) x = OSSwapBigToHostInt16(x)
+#define B2N_32(x) x = OSSwapBigToHostInt32(x)
+#define B2N_64(x) x = OSSwapBigToHostInt64(x)
+
 #elif defined(__NetBSD__)
 #include <sys/endian.h>
 #define B2N_16(x) BE16TOH(x)
diff -uNr xine-lib-1.1.1/src/input/libdvdnav/dvd_input.c xine-lib-1.1.1-new/src/input/libdvdnav/dvd_input.c
--- xine-lib-1.1.1/src/input/libdvdnav/dvd_input.c	2004-04-05 14:01:09.000000000 -0400
+++ xine-lib-1.1.1-new/src/input/libdvdnav/dvd_input.c	2006-07-02 12:17:43.000000000 -0400
@@ -287,7 +287,9 @@
 #else
   /* dlopening libdvdcss */
 
-#ifndef WIN32
+#ifdef __APPLE__
+  dvdcss_library = dlopen("libdvdcss.2.dylib", RTLD_LAZY);
+#elif !defined(WIN32)
   dvdcss_library = dlopen("libdvdcss.so.2", RTLD_LAZY);
 #else
   dvdcss_library = dlopen("libdvdcss.dll", RTLD_LAZY);
@@ -323,7 +325,7 @@
       dvdcss_library = NULL;
     } else if(!DVDcss_open  || !DVDcss_close || !DVDcss_title || !DVDcss_seek
 	      || !DVDcss_read || !DVDcss_error || !dvdcss_version) {
-      fprintf(stderr,  "libdvdread: Missing symbols in libdvdcss.so.2, "
+      fprintf(stderr,  "libdvdread: Missing symbols in libdvdcss, "
 	      "this shouldn't happen !\n");
       dlclose(dvdcss_library);
     }
diff -uNr xine-lib-1.1.1/src/libffmpeg/Makefile.in xine-lib-1.1.1-new/src/libffmpeg/Makefile.in
--- xine-lib-1.1.1/src/libffmpeg/Makefile.in	2005-11-15 06:39:53.000000000 -0500
+++ xine-lib-1.1.1-new/src/libffmpeg/Makefile.in	2006-07-02 12:17:43.000000000 -0400
@@ -568,11 +568,11 @@
 @HAVE_DXR3_TRUE@xineplug_decode_ff_la_SOURCES = xine_decoder.c audio_decoder.c video_decoder.c \
 @HAVE_DXR3_TRUE@                                xine_encoder.c mpeg_parser.c
 
-@HAVE_DXR3_FALSE@xineplug_decode_ff_la_LDFLAGS = -avoid-version -module @XINE_PLUGIN_MIN_SYMS@
+@HAVE_DXR3_FALSE@xineplug_decode_ff_la_LDFLAGS =  -Wl,-read_only_relocs,warning -avoid-version -module @XINE_PLUGIN_MIN_SYMS@
 # The dxr3 uses ffmpegs MPEG encoder by dlopen()ing the ffmpeg plugin and
 # dlsym()ing the necessary function. Therefore we must allow more exported
 # symbols and cannot use @XINE_PLUGIN_MIN_SYMS@
-@HAVE_DXR3_TRUE@xineplug_decode_ff_la_LDFLAGS = -avoid-version -module
+@HAVE_DXR3_TRUE@xineplug_decode_ff_la_LDFLAGS =  -Wl,-read_only_relocs,warning -avoid-version -module
 xineplug_decode_ff_la_LIBADD = $(MLIB_LIBS) $(XINE_LIB) -lm $(ZLIB_LIBS) \
 	$(link_ffmpeg) $(THREAD_LIBS)
 
diff -uNr xine-lib-1.1.1/src/post/deinterlace/Makefile.in xine-lib-1.1.1-new/src/post/deinterlace/Makefile.in
--- xine-lib-1.1.1/src/post/deinterlace/Makefile.in	2005-11-15 06:40:13.000000000 -0500
+++ xine-lib-1.1.1-new/src/post/deinterlace/Makefile.in	2006-07-02 12:17:43.000000000 -0400
@@ -537,7 +537,7 @@
 xineplug_post_tvtime_la_LIBADD = $(XINE_LIB) \
 	$(top_builddir)/src/post/deinterlace/plugins/libdeinterlaceplugins.la
 
-xineplug_post_tvtime_la_LDFLAGS = -avoid-version -module @XINE_PLUGIN_MIN_SYMS@
+xineplug_post_tvtime_la_LDFLAGS = -avoid-version -Wl,-read_only_relocs,warning -module @XINE_PLUGIN_MIN_SYMS@
 noinst_HEADERS = deinterlace.h pulldown.h speedtools.h speedy.h tvtime.h
 all: all-recursive
 
diff -uNr xine-lib-1.1.1/src/post/planar/Makefile.in xine-lib-1.1.1-new/src/post/planar/Makefile.in
--- xine-lib-1.1.1/src/post/planar/Makefile.in	2005-11-15 06:40:15.000000000 -0500
+++ xine-lib-1.1.1-new/src/post/planar/Makefile.in	2006-07-02 12:17:43.000000000 -0400
@@ -537,7 +537,7 @@
 
 xineplug_post_planar_la_DEPENDENCIES = $(postproc_dep)
 xineplug_post_planar_la_LIBADD = $(XINE_LIB) $(postproc_lib) -lm $(THREAD_LIBS)
-xineplug_post_planar_la_LDFLAGS = -avoid-version -module @XINE_PLUGIN_MIN_SYMS@
+xineplug_post_planar_la_LDFLAGS = -avoid-version -Wl,-read_only_relocs,warning -module @XINE_PLUGIN_MIN_SYMS@
 all: all-am
 
 .SUFFIXES:
diff -uNr xine-lib-1.1.1/src/post/planar/eq.c xine-lib-1.1.1-new/src/post/planar/eq.c
--- xine-lib-1.1.1/src/post/planar/eq.c	2004-04-17 15:54:32.000000000 -0400
+++ xine-lib-1.1.1-new/src/post/planar/eq.c	2006-07-02 12:17:43.000000000 -0400
@@ -27,6 +27,7 @@
 #include "post.h"
 #include "xineutils.h"
 #include <pthread.h>
+#undef ARCH_X86
 
 
 #ifdef ARCH_X86
diff -uNr xine-lib-1.1.1/src/post/planar/eq2.c xine-lib-1.1.1-new/src/post/planar/eq2.c
--- xine-lib-1.1.1/src/post/planar/eq2.c	2004-04-17 15:54:32.000000000 -0400
+++ xine-lib-1.1.1-new/src/post/planar/eq2.c	2006-07-02 12:17:43.000000000 -0400
@@ -32,6 +32,7 @@
 #include "xineutils.h"
 #include <math.h>
 #include <pthread.h>
+#undef ARCH_X86
 
 
 /* Per channel parameters */
diff -uNr xine-lib-1.1.1/src/video_out/Makefile.in xine-lib-1.1.1-new/src/video_out/Makefile.in
--- xine-lib-1.1.1/src/video_out/Makefile.in	2005-11-15 06:40:17.000000000 -0500
+++ xine-lib-1.1.1-new/src/video_out/Makefile.in	2006-07-02 12:17:43.000000000 -0400
@@ -338,7 +338,7 @@
 CYGPATH_W = @CYGPATH_W@
 DATADIRNAME = @DATADIRNAME@
 DEBUG_CFLAGS = @DEBUG_CFLAGS@
-DEFS = @DEFS@
+DEFS = @DEFS@ -DXINE_COMPILE
 DEPCOMP = @DEPCOMP@
 DEPDIR = @DEPDIR@
 DEPMOD = @DEPMOD@
diff -uNr xine-lib-1.1.1/src/video_out/macosx/Makefile.in xine-lib-1.1.1-new/src/video_out/macosx/Makefile.in
--- xine-lib-1.1.1/src/video_out/macosx/Makefile.in	2005-11-15 06:40:20.000000000 -0500
+++ xine-lib-1.1.1-new/src/video_out/macosx/Makefile.in	2006-07-02 12:17:44.000000000 -0400
@@ -153,7 +153,7 @@
 CYGPATH_W = @CYGPATH_W@
 DATADIRNAME = @DATADIRNAME@
 DEBUG_CFLAGS = @DEBUG_CFLAGS@
-DEFS = @DEFS@
+DEFS = @DEFS@ -DXINE_COMPILE
 DEPCOMP = @DEPCOMP@
 DEPDIR = @DEPDIR@
 DEPMOD = @DEPMOD@
diff -uNr xine-lib-1.1.1/src/xine-utils/xine_check.c xine-lib-1.1.1-new/src/xine-utils/xine_check.c
--- xine-lib-1.1.1/src/xine-utils/xine_check.c	2005-03-08 21:05:16.000000000 -0500
+++ xine-lib-1.1.1-new/src/xine-utils/xine_check.c	2006-07-02 12:17:44.000000000 -0400
@@ -369,7 +369,11 @@
   /* Majority of thi code was taken from or inspired by the xvinfo.c file of XFree86 */
 
   dlerror(); /* clear error code */
+#ifdef __APPLE__
+  x11_handle = dlopen("libX11.6.dylib", RTLD_LAZY);
+#else
   x11_handle = dlopen("libX11.so", RTLD_LAZY);
+#endif
   if(!x11_handle) {
     hc->msg = dlerror();
     hc->status = XINE_HEALTH_CHECK_FAIL;
@@ -395,7 +399,11 @@
   }
   
   dlerror(); /* clear error code */
+#ifdef __APPLE__
+  xv_handle = dlopen("libXv.1.dylib", RTLD_LAZY);
+#else
   xv_handle = dlopen("libXv.so", RTLD_LAZY);
+#endif
   if(!xv_handle) {
     hc->msg = dlerror();
     /* Xv might still work when linked statically into the output plugin,
