diff -Nurd phonon-4.3.0/CMakeLists.txt phonon-4.3.0-patched/CMakeLists.txt
--- phonon-4.3.0/CMakeLists.txt	2009-01-22 03:20:44.000000000 -0500
+++ phonon-4.3.0-patched/CMakeLists.txt	2009-01-31 11:23:05.000000000 -0500
@@ -196,6 +196,8 @@
 if(APPLE)
    set(INSTALL_TARGETS_DEFAULT_ARGS  ${INSTALL_TARGETS_DEFAULT_ARGS}
                                      BUNDLE DESTINATION "${BUNDLE_INSTALL_DIR}" )
+	set(CMAKE_SHARED_MODULE_CREATE_C_FLAGS   "${CMAKE_SHARED_MODULE_CREATE_C_FLAGS}   -flat_namespace -undefined dynamic_lookup")
+	set(CMAKE_SHARED_MODULE_CREATE_CXX_FLAGS "${CMAKE_SHARED_MODULE_CREATE_CXX_FLAGS} -flat_namespace -undefined dynamic_lookup")
 endif(APPLE)
 
 if (CMAKE_SYSTEM_NAME MATCHES Linux)
@@ -260,7 +262,7 @@
 endif (Q_WS_WIN)
 
 if (Q_WS_X11)
-   add_subdirectory(gstreamer)
+   #add_subdirectory(gstreamer)
    add_subdirectory(xine)
 endif (Q_WS_X11)
 
diff -Nurd phonon-4.3.0/qt7/CMakeLists.txt phonon-4.3.0-patched/qt7/CMakeLists.txt
--- phonon-4.3.0/qt7/CMakeLists.txt	2008-11-24 12:12:03.000000000 -0500
+++ phonon-4.3.0-patched/qt7/CMakeLists.txt	2009-01-31 11:24:44.000000000 -0500
@@ -14,6 +14,8 @@
 project(phonon-qt7)
 include(ConfigureChecks.cmake)
 
+include_directories(${OPENGL_INCLUDE_DIR} ${QT_PRIVATE_INCLUDEDIR} ${QT_PRIVATE_INCLUDEDIR}/Qt)
+
 if (BUILD_PHONON_QT7)
     set(phonon_qt7_SRCS
         medianode.cpp
@@ -37,12 +39,15 @@
    	videoframe.cpp
         audiosplitter.cpp
         mediaobjectaudionode.cpp
+		  audioeffects.cpp
+		  quicktimestreamreader.cpp
+		  quicktimestreamreader_objc.mm
        )
 
    automoc4_add_library(phonon_qt7 MODULE ${phonon_qt7_SRCS})
    target_link_libraries(phonon_qt7
        ${QT_QTGUI_LIBRARY} ${QT_QTOPENGL_LIBRARY}
-       "-framework phonon"
+       ${PHONON_LIBS}
        "-framework QuickTime"
        "-framework OpenGL"
        "-framework AudioUnit"
diff -Nurd phonon-4.3.0/qt7/ConfigureChecks.cmake phonon-4.3.0-patched/qt7/ConfigureChecks.cmake
--- phonon-4.3.0/qt7/ConfigureChecks.cmake	2008-11-24 12:12:03.000000000 -0500
+++ phonon-4.3.0-patched/qt7/ConfigureChecks.cmake	2009-01-31 11:24:58.000000000 -0500
@@ -14,3 +14,4 @@
 
 # Find the frameworks if necessary
 set(BUILD_PHONON_QT7 TRUE)
+find_path(QT_PRIVATE_INCLUDEDIR QtOpenGL/private/qgl_p.h)
diff -Nurd phonon-4.3.0/xine/CMakeLists.txt phonon-4.3.0-patched/xine/CMakeLists.txt
--- phonon-4.3.0/xine/CMakeLists.txt	2009-01-08 09:47:18.000000000 -0500
+++ phonon-4.3.0-patched/xine/CMakeLists.txt	2009-01-31 11:25:09.000000000 -0500
@@ -102,7 +102,7 @@
     configure_file(config-xine-widget.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-xine-widget.h )
     
     add_definitions(-DPHONON_BACKEND_VERSION_4_2)
-    automoc4_add_library(phonon_xine SHARED ${phonon_xine_SRCS})
+    automoc4_add_library(phonon_xine MODULE ${phonon_xine_SRCS})
     set_target_properties(phonon_xine PROPERTIES PREFIX "")
     target_link_libraries(phonon_xine ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${PHONON_LIBS} ${XINE_LIBRARY})
     if(XCB_FOUND AND XINE_XCB_FOUND)
