diff -Nurd qt-assistant-qassistantclient-library-compat-version-4.6.3/lib/headers/QAssistantClient qt-assistant-qassistantclient-library-compat-version-4.6.3-patched/lib/headers/QAssistantClient
--- qt-assistant-qassistantclient-library-compat-version-4.6.3/lib/headers/QAssistantClient	1969-12-31 19:00:00.000000000 -0500
+++ qt-assistant-qassistantclient-library-compat-version-4.6.3-patched/lib/headers/QAssistantClient	2011-07-19 12:45:50.000000000 -0400
@@ -0,0 +1 @@
+#include "qassistantclient.h"
diff -Nurd qt-assistant-qassistantclient-library-compat-version-4.6.3/lib/headers/QtAssistant qt-assistant-qassistantclient-library-compat-version-4.6.3-patched/lib/headers/QtAssistant
--- qt-assistant-qassistantclient-library-compat-version-4.6.3/lib/headers/QtAssistant	1969-12-31 19:00:00.000000000 -0500
+++ qt-assistant-qassistantclient-library-compat-version-4.6.3-patched/lib/headers/QtAssistant	2011-07-19 12:45:50.000000000 -0400
@@ -0,0 +1,5 @@
+#ifndef QT_QTASSISTANT_MODULE_H
+#define QT_QTASSISTANT_MODULE_H
+#include <QtNetwork/QtNetwork>
+#include "qassistantclient.h"
+#endif
diff -Nurd qt-assistant-qassistantclient-library-compat-version-4.6.3/lib/lib.pro qt-assistant-qassistantclient-library-compat-version-4.6.3-patched/lib/lib.pro
--- qt-assistant-qassistantclient-library-compat-version-4.6.3/lib/lib.pro	2010-06-18 05:01:10.000000000 -0400
+++ qt-assistant-qassistantclient-library-compat-version-4.6.3-patched/lib/lib.pro	2011-07-19 12:46:04.000000000 -0400
@@ -16,8 +16,8 @@
                   qassistantclient_global.h
 SOURCES         = qassistantclient.cpp
 
-DESTDIR                = ../../../../lib
-DLLDESTDIR             = ../../../../bin
+DESTDIR                = ../lib
+DLLDESTDIR             = ../bin
 
 unix {
         QMAKE_CFLAGS += $$QMAKE_CFLAGS_SHLIB
