diff -Nurd -x'*~' readline-5.0.orig/Makefile.in readline-5.0/Makefile.in
--- readline-5.0.orig/Makefile.in	2004-04-22 16:44:37.000000000 -0400
+++ readline-5.0/Makefile.in	2006-03-29 01:05:33.000000000 -0500
@@ -218,10 +218,8 @@
 install:	$(INSTALL_TARGETS)
 
 install-static: installdirs $(STATIC_LIBS) install-headers install-doc
-	-$(MV) $(DESTDIR)$(libdir)/libreadline.a $(DESTDIR)$(libdir)/libreadline.old
 	$(INSTALL_DATA) libreadline.a $(DESTDIR)$(libdir)/libreadline.a
 	-test -n "$(RANLIB)" && $(RANLIB) $(DESTDIR)$(libdir)/libreadline.a
-	-$(MV) $(DESTDIR)$(libdir)/libhistory.a $(DESTDIR)$(libdir)/libhistory.old
 	$(INSTALL_DATA) libhistory.a $(DESTDIR)$(libdir)/libhistory.a
 	-test -n "$(RANLIB)" && $(RANLIB) $(DESTDIR)$(libdir)/libhistory.a
 
diff -Nurd -x'*~' readline-5.0.orig/doc/Makefile.in readline-5.0/doc/Makefile.in
--- readline-5.0.orig/doc/Makefile.in	2004-01-08 10:42:54.000000000 -0500
+++ readline-5.0/doc/Makefile.in	2006-03-29 00:53:48.000000000 -0500
@@ -211,14 +211,6 @@
 	else \
 		${INSTALL_DATA} $(srcdir)/history.info $(DESTDIR)$(infodir)/history.info; \
 	fi
-	-if $(SHELL) -c 'install-info --version' >/dev/null 2>&1; then \
-		install-info --dir-file=$(DESTDIR)$(infodir)/dir \
-			$(DESTDIR)$(infodir)/readline.info ; \
-		install-info --dir-file=$(DESTDIR)$(infodir)/dir \
-			$(DESTDIR)$(infodir)/history.info ; \
-		install-info --dir-file=$(DESTDIR)$(infodir)/dir \
-			$(DESTDIR)$(infodir)/rluserman.info ; \
-	else true; fi
 	-${INSTALL_DATA} $(srcdir)/readline.3 $(DESTDIR)$(man3dir)/readline$(man3ext)
 	-${INSTALL_DATA} $(srcdir)/history.3 $(DESTDIR)$(man3dir)/history$(man3ext)
 	-if test -n "${htmldir}" ; then \
diff -ruN readline-5.0/support/shobj-conf readline-5.0-patched/support/shobj-conf
--- readline-5.0/support/shobj-conf	Thu Nov 13 09:36:19 2003
+++ readline-5.0-patched/support/shobj-conf	Sat Dec 11 11:22:19 2004
@@ -146,8 +146,8 @@
 	SHLIB_LIBSUFF='dylib'
 
 	case "${host_os}" in
-	darwin7*)	SHOBJ_LDFLAGS=''
-			SHLIB_XLDFLAGS='-dynamiclib -arch_only `/usr/bin/arch` -install_name $(libdir)/$@ -current_version $(SHLIB_MAJOR)$(SHLIB_MINOR) -compatibility_version $(SHLIB_MAJOR) -v'
+	darwin7*|darwin8*|darwin9*|*-darwin1?.*)	SHOBJ_LDFLAGS=''
+			SHLIB_XLDFLAGS='-dynamiclib -arch_only `/usr/bin/arch` -install_name $(libdir)/`echo $@ | sed "s:\\(.*\\.[0-9]\\)\\.[0-9]:\\1:"` -current_version $(SHLIB_MAJOR)$(SHLIB_MINOR) -compatibility_version $(SHLIB_MAJOR) -v'
 			;;
 	*)		SHOBJ_LDFLAGS='-dynamic'
 			SHLIB_XLDFLAGS='-arch_only `/usr/bin/arch` -install_name $(libdir)/$@ -current_version $(SHLIB_MAJOR)$(SHLIB_MINOR) -compatibility_version $(SHLIB_MAJOR) -v'
