diff -Nurd soprano-2.1.1/backends/redland/CMakeLists.txt soprano-2.1.1-new/backends/redland/CMakeLists.txt
--- soprano-2.1.1/backends/redland/CMakeLists.txt	2008-03-21 13:33:44.000000000 -0400
+++ soprano-2.1.1-new/backends/redland/CMakeLists.txt	2008-11-24 14:34:36.000000000 -0500
@@ -33,7 +33,7 @@
 
 add_library(soprano_redlandcore STATIC ${redland_core_SRC})
 
-add_library(soprano_redlandbackend SHARED ${redland_backend_SRC})
+add_library(soprano_redlandbackend MODULE ${redland_backend_SRC})
 
 ADD_DEFINITIONS(${REDLAND_CFLAGS})
 
diff -Nurd soprano-2.1.1/parsers/nquads/CMakeLists.txt soprano-2.1.1-new/parsers/nquads/CMakeLists.txt
--- soprano-2.1.1/parsers/nquads/CMakeLists.txt	2007-11-23 04:56:23.000000000 -0500
+++ soprano-2.1.1-new/parsers/nquads/CMakeLists.txt	2008-11-24 14:35:30.000000000 -0500
@@ -13,7 +13,7 @@
 
 qt4_automoc(${nquadparser_SRC})
 
-add_library(soprano_nquadparser SHARED ${nquadparser_SRC})
+add_library(soprano_nquadparser MODULE ${nquadparser_SRC})
 
 TARGET_LINK_LIBRARIES(soprano_nquadparser soprano)
 
diff -Nurd soprano-2.1.1/parsers/raptor/CMakeLists.txt soprano-2.1.1-new/parsers/raptor/CMakeLists.txt
--- soprano-2.1.1/parsers/raptor/CMakeLists.txt	2007-10-15 14:23:31.000000000 -0400
+++ soprano-2.1.1-new/parsers/raptor/CMakeLists.txt	2008-11-24 14:34:36.000000000 -0500
@@ -17,7 +17,7 @@
 
 qt4_automoc(${raptor_parser_SRC})
 
-add_library(soprano_raptorparser SHARED ${raptor_parser_SRC})
+add_library(soprano_raptorparser MODULE ${raptor_parser_SRC})
 
 TARGET_LINK_LIBRARIES(soprano_raptorparser soprano ${RAPTOR_LIBRARIES})
 
diff -Nurd soprano-2.1.1/serializers/nquads/CMakeLists.txt soprano-2.1.1-new/serializers/nquads/CMakeLists.txt
--- soprano-2.1.1/serializers/nquads/CMakeLists.txt	2007-11-23 04:56:23.000000000 -0500
+++ soprano-2.1.1-new/serializers/nquads/CMakeLists.txt	2008-11-24 14:35:56.000000000 -0500
@@ -13,7 +13,7 @@
 
 qt4_automoc(${nquadserializer_SRC})
 
-add_library(soprano_nquadserializer SHARED ${nquadserializer_SRC})
+add_library(soprano_nquadserializer MODULE ${nquadserializer_SRC})
 
 TARGET_LINK_LIBRARIES(soprano_nquadserializer soprano)
 
diff -Nurd soprano-2.1.1/serializers/raptor/CMakeLists.txt soprano-2.1.1-new/serializers/raptor/CMakeLists.txt
--- soprano-2.1.1/serializers/raptor/CMakeLists.txt	2008-07-10 03:41:35.000000000 -0400
+++ soprano-2.1.1-new/serializers/raptor/CMakeLists.txt	2008-11-24 14:34:36.000000000 -0500
@@ -25,7 +25,7 @@
 
 qt4_automoc(${raptor_serializer_SRC})
 
-add_library(soprano_raptorserializer SHARED ${raptor_serializer_SRC})
+add_library(soprano_raptorserializer MODULE ${raptor_serializer_SRC})
 
 TARGET_LINK_LIBRARIES(soprano_raptorserializer soprano soprano_redlandcore ${RAPTOR_LIBRARIES})
 
