diff -Nurd -x'*~' tre-0.7.5.orig/configure tre-0.7.5/configure
--- tre-0.7.5.orig/configure	2006-12-10 01:04:32.000000000 -0500
+++ tre-0.7.5/configure	2009-03-29 13:05:09.000000000 -0400
@@ -9502,7 +9502,7 @@
   fi
   INTL_MACOSX_LIBS=
   if test $gt_cv_func_CFPreferencesCopyAppValue = yes || test $gt_cv_func_CFLocaleCopyCurrent = yes; then
-    INTL_MACOSX_LIBS="-Wl,-framework -Wl,CoreFoundation"
+    INTL_MACOSX_LIBS="-Wl,-framework,CoreFoundation"
   fi
 
 
diff -Nurd -x'*~' tre-0.7.5.orig/doc/agrep.1.in tre-0.7.5/doc/agrep.1.in
--- tre-0.7.5.orig/doc/agrep.1.in	2006-12-09 15:33:37.000000000 -0500
+++ tre-0.7.5/doc/agrep.1.in	2009-03-29 13:09:48.000000000 -0400
@@ -172,6 +172,9 @@
 .PP
 .I PATTERN
 is a POSIX extended regular expression (ERE) with the TRE extensions.
+See \fB@prefix@/share/doc/tre/tre-syntax.html\fR in the \fBtre\fR
+package for full documentation of the regular-expression language.
+
 .SH "REPORTING BUGS"
 Report bugs to the TRE mailing list
 .BR "" < tre-general@lists.laurikari.net >.
