diff -ruN VFlib3-3.6.14/Makefile.in VFlib3-3.6.14.new/Makefile.in
--- VFlib3-3.6.14/Makefile.in	2002-12-09 19:52:10.000000000 -0500
+++ VFlib3-3.6.14.new/Makefile.in	2008-12-04 12:20:06.000000000 -0500
@@ -13,7 +13,7 @@
 
 default all:
 	for d in $(MAKE_SUBDIRS) ; do \
-	  (cd $$d ; $(MAKE) $@); done
+	  (cd $$d ; $(MAKE) $@) || exit 1; done
 
 install:
 	@if [ ! -d $(bindir) ] ; then \
@@ -25,7 +25,7 @@
 	@if [ ! -d $(runtimedir) ] ; then \
 	    ./recmkdir $(runtimedir); fi
 	@for d in $(INST_SUBDIRS) ; do \
-	  (cd $$d ; $(MAKE) $@); done
+	  (cd $$d ; $(MAKE) $@) || exit 1; done
 	@echo "Installing vflibcaps..."
 	@for f in `ls ./vflibcaps/vflibcap*`; do \
 	   ./$(INSTALL_DATA) $$f $(runtimedir); done
diff -ruN VFlib3-3.6.13/VFlib3-config.in VFlib3-3.6.13.new/VFlib3-config.in
--- VFlib3-3.6.13/VFlib3-config.in	2001-05-30 00:40:15.000000000 -0400
+++ VFlib3-3.6.13.new/VFlib3-config.in	2003-06-09 11:00:32.000000000 -0400
@@ -158,15 +158,15 @@
 fi
 
 if test "$echo_runtime_dir" = "yes"; then
-  S="$S $datadir/VFlib/$version"
+  S="$S $datadir/vflib/"
 fi
 
 if test "$echo_runtime_site_dir" = "yes"; then
-  S="$S $datadir/VFlib/site"
+  S="$S $datadir/vflib/site"
 fi      
 
 if test "$echo_vflibcap" = "yes"; then
-  S="$S $datadir/VFlib/$version/vflibcap"
+  S="$S $datadir/vflib/$version/vflibcap"
 fi      
 
 echo $S
diff -ruN VFlib3-3.6.13/configure VFlib3-3.6.13.new/configure
--- VFlib3-3.6.13/configure	2001-11-30 22:24:33.000000000 -0500
+++ VFlib3-3.6.13.new/configure	2003-06-09 11:48:35.000000000 -0400
@@ -1187,7 +1187,7 @@
 
 # Actually configure libtool.  ac_aux_dir is where install-sh is found.
 CC="$CC" CFLAGS="$CFLAGS" CPPFLAGS="$CPPFLAGS" \
-LD="$LD" NM="$NM" RANLIB="$RANLIB" LN_S="$LN_S" \
+LD="$LD" LDFLAGS="$LDFLAGS" NM="$NM" RANLIB="$RANLIB" LN_S="$LN_S" \
 ${CONFIG_SHELL-/bin/sh} $ac_aux_dir/ltconfig \
 $libtool_flags --no-verify $ac_aux_dir/ltmain.sh $host \
 || { echo "configure: error: libtool configure failed" 1>&2; exit 1; }
@@ -2356,9 +2356,9 @@
 
 
 
-default_vflibcap=${datadir}/VFlib/${vflib_version}/vflibcap
-runtimesitedir=${datadir}/VFlib/site
-runtimedir=${datadir}/VFlib/${vflib_version}
+default_vflibcap=${datadir}/vflib/vflibcap
+runtimesitedir=${datadir}/vflib/site
+runtimedir=${datadir}/vflib/${vflib_version}
 subdir_ccv=ccv
 subdir_t1enc=t1lib
 subdir_ascii_jtex=ascii-jtex
diff -ruN VFlib3-3.6.13/src/vflmkcaptex VFlib3-3.6.13.new/src/vflmkcaptex
--- VFlib3-3.6.13/src/vflmkcaptex	2002-11-04 19:06:33.000000000 -0500
+++ VFlib3-3.6.13.new/src/vflmkcaptex	2003-06-09 10:56:11.000000000 -0400
@@ -30,7 +30,7 @@
 version="1.0"
 
 vflibversion="3.6.14"
-prefix=/usr/local
+prefix=@FINKPREFIX@
 datadir=${prefix}/share
 
 
@@ -607,9 +607,9 @@
     opt="${comopt1} ${comopt2}"
     opt="${opt} -x .pk"
     opt="${opt} -a TEXMF"
-    opt="${opt} -e ${datadir}/VFlib/${vflibversion}/t1lib"
-    opt="${opt} -e ${datadir}/VFlib/site"
-    opt="${opt} -e ${datadir}/VFlib/site/t1lib"
+    opt="${opt} -e ${datadir}/vflib/t1lib"
+    opt="${opt} -e ${datadir}/vflib/site"
+    opt="${opt} -e ${datadir}/vflib/site/t1lib"
     opt="${opt} -f -t"
     type1_opt=""
     type1_gs_opt=""
diff -ruN VFlib3-3.6.13/src/vflmkcaptex.in VFlib3-3.6.13.new/src/vflmkcaptex.in
--- VFlib3-3.6.13/src/vflmkcaptex.in	2001-12-10 06:33:22.000000000 -0500
+++ VFlib3-3.6.13.new/src/vflmkcaptex.in	2003-06-09 10:58:08.000000000 -0400
@@ -607,9 +607,9 @@
     opt="${comopt1} ${comopt2}"
     opt="${opt} -x .pk"
     opt="${opt} -a TEXMF"
-    opt="${opt} -e ${datadir}/VFlib/${vflibversion}/t1lib"
-    opt="${opt} -e ${datadir}/VFlib/site"
-    opt="${opt} -e ${datadir}/VFlib/site/t1lib"
+    opt="${opt} -e ${datadir}/vflib/t1lib"
+    opt="${opt} -e ${datadir}/vflib/site"
+    opt="${opt} -e ${datadir}/vflib/site/t1lib"
     opt="${opt} -f -t"
     type1_opt=""
     type1_gs_opt=""
diff -ruN VFlib3-3.6.13/src/vflserver.c VFlib3-3.6.13.new/src/vflserver.c
--- VFlib3-3.6.13/src/vflserver.c	2001-07-15 22:42:14.000000000 -0400
+++ VFlib3-3.6.13.new/src/vflserver.c	2003-06-09 10:59:31.000000000 -0400
@@ -142,7 +142,7 @@
 	&& ((strcmp(argv[0], "-h") == 0) || (strcmp(argv[0], "-help") == 0))){
       printf("VFLSERVER - a VFlib server\n");
       printf("Usage: vflserver [-v vflibcap] [cmd-file ... ]\n"); 
-      printf("  Example:   ./vflserver -v /usr/local/lib/VFlib3/vflibcap\n");
+      printf("  Example:   ./vflserver -v @FINKPREFIX@/lib/vflib3/vflibcap\n");
       exit(0);
     } else if ((argc >= 2) && (strcmp(argv[0], "-v") == 0)){
       --argc; argv++;
diff -ruN VFlib3-3.6.14/utils/Makefile.in VFlib3-3.6.14.new/utils/Makefile.in
--- VFlib3-3.6.14/utils/Makefile.in	1999-10-31 19:24:11.000000000 -0500
+++ VFlib3-3.6.14.new/utils/Makefile.in	2008-12-04 12:19:12.000000000 -0500
@@ -7,12 +7,12 @@
 
 default all: 
 	@for d in $(SUBDIRS) ; do \
-	  (cd $$d ; $(MAKE) $@); done 
+	  (cd $$d ; $(MAKE) $@) || exit 1; done 
 
 install uninstall::
 	@echo "Installing utility programs..."
 	@for d in $(SUBDIRS) ; do \
-	  (cd $$d ; $(MAKE) $@); done 
+	  (cd $$d ; $(MAKE) $@) || exit 1; done 
 
 clean veryclean::
 	for d in $(SUBDIRS) ; do \
diff -ru VFlib3-3.6.13/ltconfig VFlib3-3.6.13-patched/ltconfig
--- VFlib3-3.6.13/ltconfig	Tue Nov 25 17:04:40 2003
+++ VFlib3-3.6.13-patched/ltconfig	Tue Nov 25 17:02:38 2003
@@ -1379,11 +1379,11 @@
       ;;
     *) # Darwin 1.3 on
     if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
-      allow_undefined_flag='-flat_namespace -undefined suppress'
+      allow_undefined_flag='-flat_namespace -undefined suppress -read_only_relocs suppress'
     else
       case ${MACOSX_DEPLOYMENT_TARGET} in
         10.[012])
-          allow_undefined_flag='-flat_namespace -undefined suppress'
+          allow_undefined_flag='-flat_namespace -undefined suppress -read_only_relocs suppress'
           ;;
         10.*)
           allow_undefined_flag='-undefined dynamic_lookup'
