diff -rauN lynx2-8-8/lynx.cfg lynx2-8-8-p1/lynx.cfg
--- lynx2-8-8/lynx.cfg	2012-11-19 01:20:40.000000000 +0100
+++ lynx2-8-8-p1/lynx.cfg	2013-04-16 00:56:56.000000000 +0200
@@ -90,7 +90,7 @@
 #
 # Normally we expect you will connect to a remote site, e.g., the Lynx starting
 # site:
-STARTFILE:http://lynx.isc.org/
+#STARTFILE:http://lynx.isc.org/
 #
 # As an alternative, you may want to use a local URL.  A good choice for this is
 # the user's home directory:
@@ -114,9 +114,9 @@
 # This definition will be overridden if the "LYNX_HELPFILE" environment
 # variable has been set.
 #
-HELPFILE:http://lynx.isc.org/release/lynx2-8-7/lynx_help/lynx_help_main.html
+#HELPFILE:http://lynx.isc.org/release/lynx2-8-7/lynx_help/lynx_help_main.html
 .ex
-#HELPFILE:file://localhost/PATH_TO/lynx_help/lynx_help_main.html
+HELPFILE:file://localhostFINKPREFIX/share/doc/lynx-cur/help/lynx_help_main.html
 
 .h2 DEFAULT_INDEX_FILE
 # DEFAULT_INDEX_FILE is the default file retrieved when the
@@ -436,7 +436,7 @@
 # Find RFC 1345 at
 .url http://tools.ietf.org/html/rfc1345
 #
-#CHARACTER_SET:iso-8859-1
+CHARACTER_SET:iso-8859-1
 
 .h2 LOCALE_CHARSET
 # LOCALE_CHARSET overrides CHARACTER_SET if true, using the current locale to
@@ -446,7 +446,7 @@
 # values and their relationship to the locale value is not.  GNU libiconv
 # happens to give useful values, but other implementations are not guaranteed
 # to do this.
-#LOCALE_CHARSET:FALSE
+LOCALE_CHARSET:TRUE
 
 .h2 HTML5_CHARSETS
 # HTML5_CHARSETS is an alternative to ASSUME_CHARSET and ASSUME_LOCAL_CHARSET.
@@ -544,7 +544,7 @@
 # compatibility problems with other browsers, see also PREPEND_BASE_TO_SOURCE.
 # Note that the prepending is not done for -source dumps.
 #
-#PREPEND_CHARSET_TO_SOURCE:TRUE
+PREPEND_CHARSET_TO_SOURCE:FALSE
 
 .h2 NCR_IN_BOOKMARKS
 # NCR_IN_BOOKMARKS:TRUE allows you to save 8-bit characters in bookmark titles
@@ -596,7 +596,7 @@
 # the 'o'ptions menu and save that preference in their RC file.
 # This may be a comma-separated list of languages in decreasing preference.
 #
-#PREFERRED_LANGUAGE:en
+PREFERRED_LANGUAGE:en
 
 .h2 PREFERRED_CHARSET
 # PREFERRED_CHARSET specifies the character set in MIME notation (e.g.,
@@ -788,7 +788,7 @@
 # 'o'ptions menu and saved in the RC file, and always can be toggled
 # via the -show_cursor command line switch.
 #
-#SHOW_CURSOR:FALSE
+SHOW_CURSOR:TRUE
 
 .h2 UNDERLINE_LINKS
 # UNDERLINE_LINKS controls whether links are underlined by default, or shown
@@ -973,8 +973,8 @@
 # executed.  This is much less dangerous than enabling all execution
 # links, but can still be dangerous.
 #
-#LOCAL_EXECUTION_LINKS_ALWAYS_ON:FALSE
-#LOCAL_EXECUTION_LINKS_ON_BUT_NOT_REMOTE:FALSE
+LOCAL_EXECUTION_LINKS_ALWAYS_ON:FALSE
+LOCAL_EXECUTION_LINKS_ON_BUT_NOT_REMOTE:FALSE
 
 .h2 TRUSTED_EXEC
 # If LOCAL_EXECUTION_LINK_ON_BUT_NOT_REMOTE is TRUE, and no TRUSTED_EXEC
@@ -1021,7 +1021,7 @@
 # as well (i.e., they will only be able to access lynxexec or lynxprog
 # URLs which meet any ALWAYS_TRUSTED_EXEC rules).
 #
-#TRUSTED_EXEC:none
+TRUSTED_EXEC:none
 
 .h2 ALWAYS_TRUSTED_EXEC
 # If EXEC_LINKS was defined, any lynxexec or lynxprog URL can be made
@@ -1042,7 +1042,7 @@
 #
 # The default ALWAYS_TRUSTED_EXEC rule is "none".
 #
-#ALWAYS_TRUSTED_EXEC:none
+ALWAYS_TRUSTED_EXEC:none
 
 .h2 TRUSTED_LYNXCGI
 # Unix:
@@ -1061,7 +1061,7 @@
 #
 # The default TRUSTED_LYNXCGI rule is "none".
 #
-#TRUSTED_LYNXCGI:none
+TRUSTED_LYNXCGI:none
 
 .h2 LYNXCGI_ENVIRONMENT
 # Unix:
@@ -1381,7 +1381,7 @@
 # The default is FALSE, so that the feature needs to be enabled here
 # explicitly if you want it.
 #
-#PERSISTENT_COOKIES:FALSE
+PERSISTENT_COOKIES:FALSE
 
 .h2 COOKIE_FILE
 # COOKIE_FILE is the default file from which persistent cookies are read
@@ -1828,7 +1828,7 @@
 # and not forced TRUE, the user can regulate it via the 'o'ptions menu (and
 # may save the preference in the RC file).
 #
-#NO_DOT_FILES:TRUE
+NO_DOT_FILES:FALSE
 
 .h1 Internal Behavior
 
@@ -2093,13 +2093,13 @@
 #
 # Unix:
 # ====
-#GLOBAL_EXTENSION_MAP:/usr/local/lib/mosaic/mime.types
+GLOBAL_EXTENSION_MAP:FINKPREFIX/etc/mime.types
 # VMS:
 # ===
 #GLOBAL_EXTENSION_MAP:Lynx_Dir:mime.types
 #
 #	Unix (sought in user's home directory):
-#PERSONAL_EXTENSION_MAP:.mime.types
+PERSONAL_EXTENSION_MAP:.mime.types
 #	VMS (sought in user's sys$login directory):
 #PERSONAL_EXTENSION_MAP:mime.types
 
@@ -2234,6 +2234,8 @@
 #SUFFIX:.dms:application/x-dms
 #SUFFIX:.html:text/html
 #SUFFIX:.txt:text/plain
+SUFFIX:.tgz:application/octet-stream
+SUFFIX:.deb:application/octet-stream
 
 .h2 XLOADIMAGE_COMMAND
 # VMS:
@@ -2337,13 +2339,13 @@
 #
 # Unix:
 # ====
-#GLOBAL_MAILCAP:/usr/local/lib/mosaic/mailcap
+GLOBAL_MAILCAP:FINKPREFIX/etc/mailcap
 # VMS:
 # ===
 #GLOBAL_MAILCAP:Lynx_Dir:mailcap
 #
 # 	Sought in user's home (Unix) or sys$login (VMS) directory.
-#PERSONAL_MAILCAP:.mailcap
+PERSONAL_MAILCAP:.mailcap
 
 .h2 PREFERRED_MEDIA_TYPES
 # When doing a GET, lynx lists the MIME types which it knows how to present
@@ -3530,7 +3532,7 @@
 # e.g.,
 #
 #SSL_CERT_FILE:/etc/ssl/certs/ca-certificates.crt
-#SSL_CERT_FILE:NULL
+SSL_CERT_FILE:NULL
 
 .h1 Appearance
 
@@ -3553,7 +3555,7 @@
 
 .h2 SYSLOG_REQUESTED_URLS
 # Log the requested URLs using the syslog interface.
-#SYSLOG_REQUESTED_URLS:TRUE
+SYSLOG_REQUESTED_URLS:FALSE
 
 .h2 SYSLOG_TEXT
 # Add the given text to calls made to syslog, to distinguish Lynx from other
@@ -3650,3 +3652,19 @@
 #
 # See also BLAT_MAIL and SYSTEM_MAIL flags.
 #ALT_BLAT_MAIL:FALSE
+
+# external
+EXTERNAL:ftp:w3m %s:TRUE
+EXTERNAL:file:w3m %s:TRUE
+EXTERNAL:http:w3m %s:TRUE
+EXTERNAL:http:wget %s:TRUE
+EXTERNAL:http:wget -r %s:TRUE
+EXTERNAL:ftp:x-www-browser %s:TRUE
+EXTERNAL:file:x-www-browser %s:TRUE
+EXTERNAL:http:x-www-browser %s:TRUE
+# include
+INCLUDE:FINKPREFIX/etc/lynx-cur/local.cfg
+INCLUDE:~/.lynx/colors:COLOR
+INCLUDE:~/.lynx/keymap:KEYMAP
+INCLUDE:~/.lynx/viewers:VIEWER
+INCLUDE:~/.lynx/external:EXTERNAL
diff -rauN lynx2-8-8/lynx_help/about_lynx.html lynx2-8-8-p1/lynx_help/about_lynx.html
--- lynx2-8-8/lynx_help/about_lynx.html	2012-02-13 03:46:17.000000000 +0100
+++ lynx2-8-8-p1/lynx_help/about_lynx.html	2013-04-16 00:51:10.000000000 +0200
@@ -60,9 +60,9 @@
   Kansas. However most of the release (and corresponding copyright)
   is the work of developers on the <a href=
   "#lynx-dev_list"><em>lynx-dev mailing list</em></a>. It is
-  distributed <a href="COPYHEADER">without restrictions on usage or
-  redistribution</a> under the <a href="COPYING">GNU General Public
-  License (Version 2)</a>.</p>
+  distributed without restrictions on usage or
+  redistribution under the GNU General Public
+  License (Version 2).</p>
 
   <p>Lynx was built over an early version of the Common Code
   Library developed by the CERN WWW Project. That code is
diff -rauN lynx2-8-8/src/chrtrans/makefile.in lynx2-8-8-p1/src/chrtrans/makefile.in
--- lynx2-8-8/src/chrtrans/makefile.in	2012-08-16 02:47:03.000000000 +0200
+++ lynx2-8-8-p1/src/chrtrans/makefile.in	2013-04-16 00:51:39.000000000 +0200
@@ -123,7 +123,7 @@
 C_SRC		= $(OBJS:$o=.c)
 
 $(MAKEUCTB) : $(OBJS)
-	$(BUILD_CC) $(CC_OPTS) $(BUILD_LDFLAGS) -o $@ $(OBJS) $(INTLLIB) $(BUILD_LIBS)
+	$(BUILD_CC) $(CC_OPTS) $(LDFLAGS) $(BUILD_LDFLAGS) -o $@ $(OBJS) $(INTLLIB) $(BUILD_LIBS)
 
 makeuctb$o : $(srcdir)/UCkd.h $(srcdir)/makeuctb.c
 
