diff -urN -x'*~' ghostscript-8.71/base/gdevpdfm.c ghostscript-8.71-1/base/gdevpdfm.c
--- ghostscript-8.71/base/gdevpdfm.c	2009-10-06 00:33:33.000000000 -0700
+++ ghostscript-8.71-1/base/gdevpdfm.c	2010-08-23 21:24:00.000000000 -0700
@@ -125,7 +125,7 @@
 static int
 pdfmark_make_dest(char dstr[MAX_DEST_STRING], gx_device_pdf * pdev,
 		  const char *Page_key, const char *View_key,
-		  const gs_param_string * pairs, uint count)
+		  const gs_param_string * pairs, uint count, uint RequirePage)
 {
     gs_param_string page_string, view_string;
     int present =
@@ -135,7 +135,7 @@
     gs_param_string action;
     int len;
 
-    if (present)
+    if (present || RequirePage)
 	page = pdfmark_page_number(pdev, &page_string);
     if (view_string.size == 0)
 	param_string_from_string(view_string, "[/XYZ null null null]");
@@ -485,7 +485,7 @@
 	    /* Make a destination even if this is for an outline. */
 	    if (Dest.data == 0) {
 		code = pdfmark_make_dest(dest, params->pdev, "/Page", "/View",
-					 pairs, count);
+					 pairs, count, 0);
 		if (code < 0)
 		    return code;
 		param_string_from_string(Dest, dest);
@@ -1100,7 +1100,7 @@
 
     if (!pdfmark_find_key("/Dest", pairs, count, &key) ||
 	(present =
-	 pdfmark_make_dest(dest, pdev, "/Page", "/View", pairs, count)) < 0
+	 pdfmark_make_dest(dest, pdev, "/Page", "/View", pairs, count, 1)) < 0
 	)
 	return_error(gs_error_rangecheck);
     cos_string_value(&value, (byte *)dest, strlen(dest));
@@ -1523,7 +1523,7 @@
 
     if (count & 1)
 	return_error(gs_error_rangecheck);
-    if (pdfmark_make_dest(dest, pdev, "/Page", "/View", pairs, count)) {
+    if (pdfmark_make_dest(dest, pdev, "/Page", "/View", pairs, count, 0)) {
 	int i;
 
 	code = cos_dict_put_c_key_string(pdev->Catalog, "/OpenAction",
