--- yamcha-0.33/training/Makefile.in.orig	2004-09-20 02:59:16.000000000 -0700
+++ yamcha-0.33/training/Makefile.in	2007-06-04 21:34:55.000000000 -0700
@@ -2,7 +2,7 @@
 #  $Id: Makefile.in,v 1.12 2004/09/20 09:59:16 taku-ku Exp $;
 #
 prefix     = @prefix@
-TOOLDIR    = @prefix@/libexec/yamcha
+TOOLDIR    = @prefix@/share/libexec/yamcha
 SVM_PARAM  = -t 1 -d 2 -c 1
 FEATURE    = F:-2..2:0.. T:-2..-1
 DIRECTION  =
@@ -53,7 +53,7 @@
 	$(PERL) $(TOOLDIR)/mkmodel -t $(TOOLDIR) $(MODEL).txtmodel.gz $(MODEL).model
 
 install:
-	@INSTALL@ -m 644 Makefile $(DESTDIR)$(prefix)/libexec/yamcha/Makefile
+	@INSTALL@ -m 644 Makefile $(DESTDIR)$(prefix)/share/libexec/yamcha/Makefile
 
 check:
 clean:
--- yamcha-0.33/yamcha-config.in.orig	2002-10-14 10:34:51.000000000 -0700
+++ yamcha-0.33/yamcha-config.in	2007-06-04 23:58:10.000000000 -0700
@@ -83,7 +83,7 @@
        	;;
 
     --libexecdir)
-       	echo @prefix@/libexec/yamcha
+       	echo @prefix@/share/libexec/yamcha
        	;;
 
     *)
--- yamcha-0.33/libexec/mkmodel.in.orig	2004-10-02 20:16:26.000000000 -0700
+++ yamcha-0.33/libexec/mkmodel.in	2007-06-05 00:08:39.000000000 -0700
@@ -41,7 +41,7 @@
 my $SIGMA   = $arg{"s"} || 0.001;
 my $MODEL   = shift @ARGV;
 my $OUT     = shift @ARGV;
-my $TOOLDIR = $arg{"t"} || "@prefix@/libexec/yamcha";
+my $TOOLDIR = $arg{"t"} || "@prefix@/share/libexec/yamcha";
 
 die "Usage: $0 textmodel binmodel\n" if (! defined $MODEL || ! defined $OUT);
 
--- yamcha-0.33/Makefile.in.orig	2005-09-05 10:52:31.000000000 -0400
+++ yamcha-0.33/Makefile.in	2007-06-20 18:33:13.000000000 -0400
@@ -716,7 +716,7 @@
 	(cd python; $(MAKE) clean; rm -f build setup.py)
 
 install-exec-local:
-	(cd $(DESTDIR)@prefix@/bin; @LN_S@ -f ../libexec/yamcha/mkmodel yamcha-mkmodel)
+	(cd $(DESTDIR)@prefix@/bin; @LN_S@ -f ../share/libexec/yamcha/mkmodel yamcha-mkmodel)
 
 uninstall-local:	
 	rm -f $(DESTDIR)@prefix@/bin/yamcha-mkmodel
