diff -Nru Canna37p3.orig/Canna.conf Canna37p3/Canna.conf
--- Canna37p3.orig/Canna.conf	Thu May 20 00:48:47 2004
+++ Canna37p3/Canna.conf	Wed Mar  9 02:04:46 2005
@@ -140,19 +140,23 @@
 # define ModernElfLinkAvailable NO
 #endif
 
-cannaPrefix = DefCannaPrefix
+#define MyCCOptions -no-cpp-precomp -Wall -Wpointer-arith -Wno-implicit-int -Wno-return-type
+#undef LibraryCCOptions
+#define LibraryCCOptions MyCCOptions -fno-common
+CCOPTIONS = MyCCOptions
+
+cannaPrefix = @PREFIX@
 cannaExecPrefix = $(cannaPrefix)
 cannaBinDir = $(cannaExecPrefix)/bin
 cannaSrvDir = DefCannaSrvDir
 XCOMM cannaLibDir = /var/lib/canna
 XCOMM cannaLibDir = /var/db/canna
 cannaLibDir = DefCannaLibDir
-XCOMM cannaManDir = $(cannaPrefix)/share/man
-cannaManDir = $(cannaPrefix)/man
+cannaManDir = $(cannaPrefix)/share/man
 cannaIncDir = $(cannaPrefix)/include/canna
 
 libCannaDir = DefLibCannaDir
-ErrDir  = DefErrDir
+ErrDir  = $(cannaPrefix)/var/log/canna
 
 /* 旧バージョンとの互換APIを無効にする場合は0と定義してください */
 #define SupportOldWchar 1
@@ -169,8 +173,8 @@
 #define ModernElfLink NO /* experimental */
 
 
-cannaOwner = bin
-cannaGroup = bin
+cannaOwner = canna
+cannaGroup = canna
 
 #ifdef InstallAsUser
 cannaOwnerGroup =
@@ -199,13 +203,13 @@
 CHGRP = :
 CHMOD = :
 #else
-CHOWN = chown
-CHGRP = chgrp
-CHMOD = chmod
+CHOWN = /usr/sbin/chown
+CHGRP = /usr/bin/chgrp
+CHMOD = /bin/chmod
 #endif
 
 /* 日本語マニュアルを使わないのであればコメントアウト */
-#define JAPANESEMAN
+/* #define JAPANESEMAN */
 
 /* #define engineSwitch */
 
@@ -237,7 +241,7 @@
 /* #define UseInstalledLibCanna YES */
 #define UseInstalledLibCanna NO
 
-DicDir   = $(cannaLibDir)/dic
+DicDir   = $(cannaPrefix)/var/lib/canna/dic
 
 /* ここから下は変更不要です */
 
@@ -343,7 +347,7 @@
 AccessFile = $(cannaPrefix)/etc/hosts.canna
 #else
 UnixSockDir = /tmp/.iroha_unix
-AccessFile = /etc/hosts.canna
+AccessFile = $(cannaPrefix)/etc/canna/hosts.canna
 #endif
 UnixSockName = IROHA
 
diff -Nru Canna37p3.orig/dic/ideo/words/chimei.t Canna37p3/dic/ideo/words/chimei.t
--- Canna37p3.orig/dic/ideo/words/chimei.t	Sat Sep 27 15:18:39 2003
+++ Canna37p3/dic/ideo/words/chimei.t	Wed Mar  9 02:05:13 2005
@@ -714,7 +714,7 @@
 じょうせき #CN 城跡
 じょうとうく #CNS 城東区
 じょうなんく #CNS 城南区
-じょうばんせん #CNS 常盤線
+じょうばんせん #CNS 常磐線
 じょうようし #CNS 城陽市
 じょほーる #CN ジョホール
 じんぐうまえ #CN 神宮前
diff -Nru Canna37p3.orig/dic/phono/Imakefile Canna37p3/dic/phono/Imakefile
--- Canna37p3.orig/dic/phono/Imakefile	Sat Oct 19 17:27:40 2002
+++ Canna37p3/dic/phono/Imakefile	Wed Mar  9 02:04:46 2005
@@ -25,7 +25,11 @@
                 KPDIC = $(CMDDIR)/kpdic/kpdic
 #endif
 
+#ifdef HasGcc
+       ROMAJI_DIC_DEF = -DSHIFT -traditional
+#else
        ROMAJI_DIC_DEF = -DSHIFT
+#endif
 
                DICDIR = $(cannaLibDir)/dic
             SAMPLEDIR = $(cannaLibDir)/sample
diff -Nru Canna37p3.orig/hosts.canna Canna37p3/hosts.canna
--- Canna37p3.orig/hosts.canna	Thu Jan  1 09:00:00 1970
+++ Canna37p3/hosts.canna	Wed Mar  9 02:04:46 2005
@@ -0,0 +1,2 @@
+unix
+localhost
diff -Nru Canna37p3.orig/update-canna-dics-dir Canna37p3/update-canna-dics-dir
--- Canna37p3.orig/update-canna-dics-dir	Thu Jan  1 09:00:00 1970
+++ Canna37p3/update-canna-dics-dir	Wed Mar  9 02:04:46 2005
@@ -0,0 +1,15 @@
+#!/bin/sh -e
+
+PATH=/bin:/sbin:/usr/bin:/usr/sbin:@PREFIX@/bin:@PREFIX@/sbin
+export PATH
+
+LIST_FILE=@PREFIX@/var/lib/canna/dic/canna/dics.dir
+LIST_DIR=@PREFIX@/var/lib/canna/dics.d
+
+TMPFILE=`tempfile`
+cat ${LIST_DIR}/* >>${TMPFILE}
+mv ${TMPFILE} ${LIST_FILE}
+chmod 0644 ${LIST_FILE}
+chown canna:canna ${LIST_FILE}
+
+exit 0
