--- gkrellkam-0.3.4/gkrellkam.c	Fri Mar  8 10:47:39 2002
+++ gkrellkam-0.3.4-patched/gkrellkam.c	Tue Jul  2 22:58:39 2002
@@ -28,7 +28,7 @@
 */
 
 #include <gkrellm.h>
-#include <libgen.h>
+//#include <libgen.h>
 #include <unistd.h>
 #include <errno.h>
 #include <stdlib.h>
@@ -1772,7 +1772,7 @@
      its parameters. If it was using a different script, put it with
      the appropriate type into the source list */
 
-  if (!strcmp (basename (firstword), "krellkam_load"))
+  if (!strcmp (firstword, "krellkam_load"))
     update_source_config (p, rest);
   else
   {
