diff -ru tree-1.4/Makefile tree-1.4-patched/Makefile
--- tree-1.4/Makefile	2003-02-06 17:25:14.000000000 +0100
+++ tree-1.4-patched/Makefile	2003-09-07 14:27:32.000000000 +0200
@@ -6,13 +6,13 @@
 # warranties, including, without limitation, the implied warranties
 # of merchant-ability and fitness for a particular purpose.
 
-CC=gcc
-CFLAGS=-O2 -Wall -fomit-frame-pointer -DLINUX_BIGFILE #-m486
-LDFLAGS=-s
+#CC=gcc
+#CFLAGS=-O2 -Wall -fomit-frame-pointer -DLINUX_BIGFILE #-m486
+#LDFLAGS=-s
 # Uncomment for OS X:
-#CC=cc
-#CFLAGS=-O2 -Wall -fomit-frame-pointer -no-cpp-precomp
-#LDFLAGS=
+CC=cc
+CFLAGS=-O2 -Wall -fomit-frame-pointer -no-cpp-precomp
+LDFLAGS=
 # Uncomment for HP/UX:
 #CC=cc
 #CFLAGS=-Ae +O2 +DAportable -Wall
@@ -24,7 +24,7 @@
 TREE_DEST=tree
 BINDIR=${prefix}/bin
 MAN=tree.1
-MANDIR=${prefix}/man/man1
+MANDIR=${prefix}/share/man/man1
 
 all:	tree
 
