diff -Naur uim-1.4.2.orig/xinitrc.uim.sh uim-1.4.2/xinitrc.uim.sh
--- uim-1.4.2.orig/xinitrc.uim.sh	1970-01-01 09:00:00.000000000 +0900
+++ uim-1.4.2/xinitrc.uim.sh	2008-03-31 02:51:31.000000000 +0900
@@ -0,0 +1,22 @@
+# If you don't want to launch uim, put xinitrc_uim_enable=NO
+# in your .xinitrc
+: ${xinitrc_uim_enable=YES}
+: ${xinitrc_uim_toolbar_enable=NO}
+
+# 'anthy', 'canna' and 'skk' are available at this time.
+: ${xinitrc_uim_engine=anthy}
+
+case $xinitrc_uim_enable in
+    [Yy][Ee][Ss])
+    if test -x "$fink_prefix/bin/uim-xim"; then
+        "$fink_prefix/bin/uim-xim" --engine=$xinitrc_uim_engine &
+    fi
+    ;;
+esac
+case $xinitrc_uim_toolbar_enable in
+    [Yy][Ee][Ss])
+    if test -x "$fink_prefix/bin/uim-toolbar-gtk"; then
+        "$fink_prefix/bin/uim-toolbar-gtk" &
+    fi
+    ;;
+esac
diff -Naur uim-1.4.2.orig/configure uim-1.4.2/configure
--- uim-1.4.2.orig/configure	2008-02-20 21:25:46.000000000 +0900
+++ uim-1.4.2/configure	2008-05-07 02:13:12.000000000 +0900
@@ -22157,6 +22157,7 @@
 /* end confdefs.h.  */
  #undef _WCHAR_T
 
+#define CANNA_NEW_WCHAR_AWARE
 #include <$ac_header>
 _ACEOF
 rm -f conftest.$ac_objext
diff -Naur uim-1.4.2.orig/helper/dict-canna.c uim-1.4.2/helper/dict-canna.c
--- uim-1.4.2.orig/helper/dict-canna.c	2008-02-19 18:16:27.000000000 +0900
+++ uim-1.4.2/helper/dict-canna.c	2008-05-07 02:13:21.000000000 +0900
@@ -36,6 +36,7 @@
 #include <unistd.h>
 #include <sys/types.h>
 
+#define CANNA_NEW_WCHAR_AWARE
 #include <canna/RK.h>
 
 #include <stdlib.h>
diff -Naur uim-1.4.2.orig/uim/canna.c uim-1.4.2/uim/canna.c
--- uim-1.4.2.orig/uim/canna.c	2008-02-19 18:16:27.000000000 +0900
+++ uim-1.4.2/uim/canna.c	2008-05-07 02:25:33.000000000 +0900
@@ -36,6 +36,7 @@
 #include <config.h>
 
 #ifdef HAVE_CANNA_RK_H
+#define CANNA_NEW_WCHAR_AWARE
 #include <canna/RK.h>
 #include <stdlib.h>
 #include <errno.h>
