diff -Nurd -x*~ elinks-0.9.1.orig/configure elinks-0.9.1/configure
--- elinks-0.9.1.orig/configure	Thu Jan 22 17:28:22 2004
+++ elinks-0.9.1/configure	Thu Feb  5 07:19:18 2004
@@ -6707,7 +6707,7 @@
       BUILD_INCLUDED_LIBINTL=yes
       USE_INCLUDED_LIBINTL=yes
       CATOBJEXT=.gmo
-      INTLLIBS="\$(top_builddir)/src/intl/gettext/libintl.a $LIBICONV"
+      INTLLIBS="\$(top_builddir)/src/intl/gettext/libintl.a"
       LIBS=`echo " $LIBS " | sed -e 's/ -lintl / /' -e 's/^ //' -e 's/ $//'`
 
                   if test "$GMSGFMT" != ":"; then
diff -Nurd -x*~ elinks-0.9.1.orig/src/Makefile.in elinks-0.9.1/src/Makefile.in
--- elinks-0.9.1.orig/src/Makefile.in	Thu Jan 22 17:28:45 2004
+++ elinks-0.9.1/src/Makefile.in	Thu Feb  5 07:19:57 2004
@@ -120,7 +120,7 @@
 elinks_DEPENDENCIES =   @INTLLIBS@  modules/libmodules.a  protocol/rewrite/liburirewrite.a  cookies/libcookies.a  viewer/text/libviewer_text.a  viewer/dump/libviewer_dump.a  cache/libcache.a  document/libdocument.a  intl/libintl.a  sched/libsched.a  lowlevel/liblowlevel.a  terminal/libterminal.a  scripting/libscripting.a  scripting/guile/libscriptingguile.a  scripting/lua/libscriptinglua.a  osdep/libosdep.a  protocol/libprotocol.a  protocol/file/libfile.a  protocol/ftp/libftp.a  protocol/http/libhttp.a  protocol/smb/libsmb.a  protocol/auth/libauth.a  ssl/libssl.a  bookmarks/libbookmarks.a  formhist/libformhist.a  globhist/libglobhist.a  config/libconfig.a  dialogs/libdialogs.a  mime/libmime.a  bfu/libbfu.a  encoding/libencoding.a  util/libutil.a
 
 
-elinks_LDADD = $(elinks_DEPENDENCIES)
+elinks_LDADD = $(elinks_DEPENDENCIES) @LIBICONV@
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../config.h
 CONFIG_CLEAN_FILES = 
diff -Nurd -x*~ elinks-0.9.1.orig/src/document/html/tables.c elinks-0.9.1/src/document/html/tables.c
--- elinks-0.9.1.orig/src/document/html/tables.c	Thu Jan  1 18:36:44 2004
+++ elinks-0.9.1/src/document/html/tables.c	Fri Feb  6 05:40:16 2004
@@ -813,8 +813,7 @@
 	if (max) *max = p->max_width;
 	if (n_links) *n_links = p->link_num;
 
-	assertm(!((min && max && *min > *max)), "get_cell_width: %d > %d",
-		*min, *max);
+	assertm(!((min && max && *min > *max)), "get_cell_width: min > max");
 
 	mem_free(p);
 }
@@ -1472,7 +1471,7 @@
 static inline int
 get_frame_pos(int a, int a_size, int b, int b_size)
 {
-	assertm(a >= -1 || a < a_size + 2 || b >= 0 || b <= b_size);
+	assertm(a >= -1 || a < a_size + 2 || b >= 0 || b <= b_size, "frame position out of bounds");
 	if_assert_failed return 0;
 	return a + 1 + (a_size + 2) * b;
 }
