--- ecl-0.9i.orig/src/doc/Makefile.in	2007-07-02 08:00:31.000000000 +0200
+++ ecl-0.9i/src/doc/Makefile.in	2007-07-02 08:44:49.000000000 +0200
@@ -47,9 +47,6 @@
 	$(mkinstalldirs) $(DESTDIR)$(infodir)
 	for k in $(INFO_FILES); do \
 	  $(INSTALL_DATA) $$k $(DESTDIR)$(infodir); \
-	  if [ -x $(INSTALL_INFO) ]; then \
-	    $(INSTALL_INFO) --info-dir=$(DESTDIR)$(infodir) $$k; \
-	  fi; \
 	done
 	$(mkinstalldirs) $(DESTDIR)$(docdir)
 	for i in Copyright LGPL; do \
