--- erlang-otp.info.orig	2007-06-19 18:15:14.000000000 +0200
+++ erlang-otp.info	2007-06-19 23:35:50.000000000 +0200
@@ -1,19 +1,23 @@
+Info2: <<
 Package: erlang-otp
-Version: 10b-10
+
+Version: 11b-5
+License: OSI-Approved
 Revision: 1
-Source: http://www.erlang.org/download/otp_src_R10B-10.tar.gz
-Source-MD5: c1405c885f07d661b7362b822d571586
-Source2: http://www.erlang.org/download/otp_doc_man_R10B-10.tar.gz
-Source2-MD5: 207e00bcaf5a9428bd86e3973f9b699f
+Description: General-purpose programming language
+Maintainer: None <fink-devel@lists.sourceforge.net>
+
+Source: http://www.erlang.org/download/otp_src_R11B-5.tar.gz
+Source-MD5: 96acec41da87d6ee0ef18e1aab36ffdd
+Source2: http://www.erlang.org/download/otp_doc_man_R11B-5.tar.gz
+Source2-MD5: 5c387162ae6d255f525d72a91481d135
 Source2ExtractDir: docs-man
-Source3: http://www.erlang.org/download/otp_doc_html_R10B-10.tar.gz
-Source3-MD5: d6e9655ff07af642025b0465c5e7e1a6
+Source3: http://www.erlang.org/download/otp_doc_html_R11B-5.tar.gz
+Source3-MD5: 6a40146d855f98d024682cd24e9bb5bc
 Source3ExtractDir: docs-html
-Description: General-purpose programming language
-License: OSI-Approved
+
 DocFiles: README EPLICENCE
-NoSetMAKEFLAGS: true
-SetMAKEFLAGS: -j1
+
 InstallScript: <<
  #!/bin/sh -ev
 
@@ -24,15 +28,17 @@
  make install prefix=%i
 
  # Fix symlinks
- rm -f %i/bin/ear %i/bin/ecc %i/bin/elink %i/bin/erl %i/bin/erlc %i/bin/escript %i/lib/erlang/bin/epmd
+ rm -f %i/bin/dialyzer %i/bin/epmd %i/bin/erl %i/bin/erlc %i/bin/escript %i/bin/run_erl %i/bin/to_erl %i/bin/typer %i/lib/erlang/bin/epmd
+
+ ln -s %p/lib/erlang/bin/dialyzer %i/bin/dialyzer
+ ln -s %p/lib/erlang/erts-${ertsver}/bin/epmd %i/lib/erlang/bin/epmd
+ ln -s %p/lib/erlang/bin/epmd %i/bin/epmd
  ln -s %p/lib/erlang/bin/erl %i/bin/erl
  ln -s %p/lib/erlang/bin/erlc %i/bin/erlc
- ln -s %p/lib/erlang/erts-${ertsver}/bin/epmd %i/lib/erlang/bin/epmd
-
- #ln -s %p/lib/erlang/bin/ear %i/bin/ear
- #ln -s %p/lib/erlang/bin/ecc %i/bin/ecc
- #ln -s %p/lib/erlang/bin/elink %i/bin/elink
- #ln -s %p/lib/erlang/bin/escript %i/bin/escript
+ ln -s %p/lib/erlang/bin/escript %i/bin/escript
+ ln -s %p/lib/erlang/bin/run_erl %i/bin/run_erl
+ ln -s %p/lib/erlang/bin/to_erl %i/bin/to_erl
+ ln -s %p/lib/erlang/bin/typer %i/bin/typer
 
  # Fix paths in scripts
  perl -i -pe "s:%d::g" \
@@ -42,17 +48,6 @@
    %i/lib/erlang/erts-${ertsver}/bin/start \
    %i/lib/erlang/releases/RELEASES
 
-   #%i/lib/erlang/bin/ecc \
-   #%i/lib/erlang/bin/ear \
-   #%i/lib/erlang/bin/elink \
-   #%i/lib/erlang/bin/escript \
-   #%i/lib/erlang/bin/esh \
-   #%i/lib/erlang/erts-${ertsver}/bin/ecc \
-   #%i/lib/erlang/erts-${ertsver}/bin/ear \
-   #%i/lib/erlang/erts-${ertsver}/bin/elink \
-   #%i/lib/erlang/erts-${ertsver}/bin/escript \
-   #%i/lib/erlang/erts-${ertsver}/bin/esh \
-
  # Move documentation to erlang-aware location
  mv %b/../docs-man/* %i/lib/erlang/
 
@@ -60,8 +55,7 @@
  mkdir -p %i/share/doc/erlang-otp/html
  mv %b/../docs-html/* %i/share/doc/erlang-otp/html/
 <<
-Maintainer: None <fink-devel@lists.sourceforge.net>
-Homepage: http://www.erlang.org/
+
 DescDetail: <<
 Erlang is a general-purpose programming language and runtime environment.
 Erlang has built-in support for concurrency, distribution and fault
@@ -69,12 +63,11 @@
 from Ericsson.  The most popular implementation of Erlang is available as
 open source from the open source erlang site.
 <<
-DescPort: <<
-I left some symlinks in but commented out.  It seems these links don't work
-on my Linux install as well, so I'm not being upset that they're not there.
-I suspect they existed in older versions and just weren't removed from
-the install.
-<<
+
 DescPackaging: <<
  Maintainer (<= 10b-10-1): Daniel Henninger
 <<
+
+Homepage: http://www.erlang.org/
+
+<<
