--- gcc/mklibgcc.in	9 Feb 2005 20:58:02 -0000	1.81
+++ gcc/mklibgcc.in	10 Feb 2005 13:52:44 -0000
@@ -102,8 +102,9 @@ if [ "$SHLIB_LINK" ]; then
   echo "vis_hide := \$(strip \$(subst @,-,\\"
   echo "    \$(shell if echo 'void foo(void); void foo(void) {}' | \\"
   echo "          $gcc_compile -fvisibility=hidden -Werror \\"
-  echo "          -c -xc - -o /dev/null 2> /dev/null; \\"
+  echo "          -c -xc - -o ./vis_test___out.o 2> /dev/null; \\"
   echo "          then echo @fvisibility=hidden @DHIDE_EXPORTS; \\"
+  echo "          rm ./vis_test___out.o 2> /dev/null; \\"
   echo "          fi)))"
   echo
