diff -Naur Gauche-0.8.14.orig/doc/gauche-ref.texi Gauche-0.8.14/doc/gauche-ref.texi
--- Gauche-0.8.14.orig/doc/gauche-ref.texi	2008-10-06 19:30:27.000000000 +0000
+++ Gauche-0.8.14/doc/gauche-ref.texi	2008-12-30 03:45:54.000000000 +0000
@@ -5,14 +5,14 @@
 @settitle Gauche Users' Reference
 @dircategory The Algorithmic Language Scheme
 @direntry
-* Gauche: (gauche-refe.info).	        An R5RS Scheme implementation.
+* Gauche: (gauche-refe).	        An R5RS Scheme implementation.
 @end direntry
 @c JP
 @setfilename gauche-refj.info
 @settitle Gauche ユーザリファレンス
 @dircategory The Algorithmic Language Scheme
 @direntry
-* Gauche (ja): (gauche-refj.info).	An R5RS Scheme implementation.
+* Gauche (ja): (gauche-refj).	An R5RS Scheme implementation.
 @end direntry
 @documentencoding euc-jp
 @c COMMON
diff -Naur Gauche-0.8.14.orig/ext/Makefile.ext.in Gauche-0.8.14/ext/Makefile.ext.in
--- Gauche-0.8.14.orig/ext/Makefile.ext.in	2008-09-18 15:43:11.000000000 +0000
+++ Gauche-0.8.14/ext/Makefile.ext.in	2008-12-30 03:47:10.000000000 +0000
@@ -22,7 +22,7 @@
 LIBS         = $(XLIBS) @LIBS@ 
 CFLAGS       = @CFLAGS@ @SHLIB_SO_CFLAGS@ $(XCFLAGS)
 CPPFLAGS     = @CPPFLAGS@ $(XCPPFLAGS)
-LDFLAGS      = $(LOCAL_LFLAGS) $(XLDFLAGS) @SHLIB_SO_LDFLAGS@
+LDFLAGS      = @LDFLAGS@ $(LOCAL_LFLAGS) $(XLDFLAGS) @SHLIB_SO_LDFLAGS@
 
 # These are set by configure
 DEFS     = @DEFS@
diff -Naur Gauche-0.8.14.orig/src/Makefile.in Gauche-0.8.14/src/Makefile.in
--- Gauche-0.8.14.orig/src/Makefile.in	2008-10-06 09:11:41.000000000 +0000
+++ Gauche-0.8.14/src/Makefile.in	2008-12-30 03:45:54.000000000 +0000
@@ -92,7 +92,7 @@
 BUILD_GOSH  = GAUCHE_LOAD_PATH="" GAUCHE_DYNLOAD_PATH="" \
 	    gosh -l./preload -I$(top_srcdir)/src -I$(top_srcdir)/lib
 
-LIB_INSTALL_DIR  = @libdir@
+LIB_INSTALL_DIR  = $(ARCH_INSTALL_DIR)
 BIN_INSTALL_DIR  = @bindir@
 DATA_INSTALL_DIR = $(datadir)
 
@@ -344,7 +344,6 @@
 	$(INSTALL) -m 444 $(INSTALL_HEADERS) $(DESTDIR)$(HEADER_INSTALL_DIR)
 	$(INSTALL) -m 444 $(INSTALL_SUBHEADERS) $(DESTDIR)$(HEADER_INSTALL_DIR)/gauche
 	$(INSTALL)        $(INSTALL_LIBS) $(DESTDIR)$(LIB_INSTALL_DIR)
-	$(INSTALL)        $(INSTALL_LIBS) $(DESTDIR)$(ARCH_INSTALL_DIR)
 	$(INSTALL) -m 555 $(INSTALL_BINS) $(DESTDIR)$(BIN_INSTALL_DIR)
 	$(INSTALL) -m 555 $(INSTALL_BINS) $(DESTDIR)$(ARCH_INSTALL_DIR)
 	$(INSTALL) -m 444 $(INSTALL_SCMS) $(DESTDIR)$(SCM_INSTALL_DIR)
diff -Naur Gauche-0.8.14.orig/gc/mach_dep.c Gauche-0.8.14/gc/mach_dep.c
--- Gauche-0.8.14.orig/gc/mach_dep.c	2008-09-19 00:43:38.000000000 +0900
+++ Gauche-0.8.14/gc/mach_dep.c	2009-12-28 02:05:30.000000000 +0900
@@ -160,7 +160,7 @@
 #endif
 
 #if !defined(HAVE_PUSH_REGS) && defined(UNIX_LIKE)
-# include <ucontext.h>
+# include <sys/ucontext.h>
 #endif
 
 /* Ensure that either registers are pushed, or callee-save registers	*/
diff -Naur Gauche-0.8.14.orig/gc/os_dep.c Gauche-0.8.14/gc/os_dep.c
--- Gauche-0.8.14.orig/gc/os_dep.c	2008-09-19 00:43:38.000000000 +0900
+++ Gauche-0.8.14/gc/os_dep.c	2009-12-28 02:05:45.000000000 +0900
@@ -2752,7 +2752,7 @@
 # if defined(MSWIN32) || defined(MSWINCE)
     LONG WINAPI GC_write_fault_handler(struct _EXCEPTION_POINTERS *exc_info)
 # else
-#   include <ucontext.h>
+#   include <sys/ucontext.h>
     /*ARGSUSED*/
     void GC_write_fault_handler(int sig, siginfo_t *si, void *raw_sc)
 # endif /* MSWIN32 || MSWINCE */
