
--- mono-1.2.5/configure	2007-08-27 15:33:48.000000000 -0400
+++ mono-1.2.5-new/configure	2007-09-19 10:01:57.000000000 -0400
@@ -40012,11 +40012,11 @@
 case "$host" in
      *-*-darwin*)
 	libsuffix=".dylib"
-	LIBC="libc.dylib"
-	INTL="libintl.dylib"
-	SQLITE="libsqlite.0.dylib"
-	SQLITE3="libsqlite3.0.dylib"
-	X11="libX11.dylib"
+	LIBC="/usr/lib/libSystem.dylib"
+	INTL="@FINKPREFIX@/lib/libintl.dylib"
+	SQLITE="@FINKPREFIX@/lib/libsqlite.0.dylib"
+	SQLITE3="@FINKPREFIX@/lib/libsqlite3.0.dylib"
+	X11="/usr/X11R6/lib/libX11.6.dylib"
 	;;
      *-*-*netbsd*)
 	LIBC="libc.so.12"
--- mono-1.2.5/data/config.in	2007-07-24 17:47:31.000000000 -0400
+++ mono-1.2.5-new/data/config.in	2007-09-19 10:03:29.000000000 -0400
@@ -3,18 +3,18 @@
 	<dllmap dll="libc" target="@LIBC@" os="!windows"/>
 	<dllmap dll="intl" target="@INTL@" os="!windows"/>
 	<dllmap dll="libintl" target="@INTL@" os="!windows"/>
-	<dllmap dll="i:libxslt.dll" target="libxslt@libsuffix@" os="!windows"/>
-	<dllmap dll="i:odbc32.dll" target="libodbc@libsuffix@" os="!windows"/>
+	<dllmap dll="i:libxslt.dll" target="@FINKPREFIX@/lib/libxslt@libsuffix@" os="!windows"/>
+	<dllmap dll="i:odbc32.dll" target="@FINKPREFIX@/lib/libodbc@libsuffix@" os="!windows"/>
 	<dllmap dll="i:odbc32.dll" target="libiodbc.dylib" os="osx"/>
-	<dllmap dll="oci" target="libclntsh@libsuffix@" os="!windows"/>
-	<dllmap dll="db2cli" target="libdb2_36@libsuffix@" os="!windows"/>
-	<dllmap dll="MonoPosixHelper" target="libMonoPosixHelper@libsuffix@" os="!windows" />
+	<dllmap dll="oci" target="@FINKPREFIX@/lib/libclntsh@libsuffix@" os="!windows"/>
+	<dllmap dll="db2cli" target="@FINKPREFIX@/lib/libdb2_36@libsuffix@" os="!windows"/>
+	<dllmap dll="MonoPosixHelper" target="@FINKPREFIX@/lib/libMonoPosixHelper@libsuffix@" os="!windows" />
 	<dllmap dll="i:msvcrt" target="@LIBC@" os="!windows"/>
 	<dllmap dll="i:msvcrt.dll" target="@LIBC@" os="!windows"/>
 	<dllmap dll="sqlite" target="@SQLITE@" os="!windows"/>
 	<dllmap dll="sqlite3" target="@SQLITE3@" os="!windows"/>
 	<dllmap dll="libX11" target="@X11@" os="!windows" />
-	<dllmap dll="libcairo-2.dll" target="libcairo.so.2" os="!windows"/>
+	<dllmap dll="libcairo-2.dll" target="@FINKPREFIX@/lib/libcairo.2@libsuffix@" os="!windows"/>
 	<dllmap dll="i:kernel32.dll">
 		<dllentry dll="__Internal" name="CopyMemory" target="mono_win32_compat_CopyMemory"/>
 		<dllentry dll="__Internal" name="FillMemory" target="mono_win32_compat_FillMemory"/>
