diff -Nurd mono-2.0/configure mono-2.0-new/configure
--- mono-2.0/configure	2008-10-01 17:07:24.000000000 -0400
+++ mono-2.0-new/configure	2008-10-07 16:52:15.000000000 -0400
@@ -3066,7 +3066,7 @@
 		;;
 	*-*-darwin*)
 		platform_win32=no
-		CPPFLAGS="$CPPFLAGS -no-cpp-precomp -D_THREAD_SAFE -DGC_MACOSX_THREADS -DPLATFORM_MACOSX -DUSE_MMAP -DUSE_MUNMAP"
+		CPPFLAGS="$CPPFLAGS -D_THREAD_SAFE -DGC_MACOSX_THREADS -DPLATFORM_MACOSX -DUSE_MMAP -DUSE_MUNMAP"
 		CPPFLAGS="$CPPFLAGS -DGetCurrentProcess=MonoGetCurrentProcess -DGetCurrentThread=MonoGetCurrentThread -DCreateEvent=MonoCreateEvent"
 		libmono_cflags="-D_THREAD_SAFE"
 		LDFLAGS="$LDFLAGS -pthread"
@@ -38516,11 +38516,11 @@
 case "$host" in
      *-*-darwin*)
 	libsuffix=".dylib"
-	LIBC="libc.dylib"
-	INTL="libintl.dylib"
-	SQLITE="libsqlite.0.dylib"
-	SQLITE3="libsqlite3.0.dylib"
-	X11="libX11.dylib"
+	LIBC="/usr/lib/libSystem.dylib"
+	INTL="@FINKPREFIX@/lib/libintl.dylib"
+	SQLITE="@FINKPREFIX@/lib/libsqlite.0.dylib"
+	SQLITE3="@FINKPREFIX@/lib/libsqlite3.0.dylib"
+	X11="/usr/X11R6/lib/libX11.6.dylib"
 	;;
      *-*-*netbsd*)
 	LIBC="libc.so.12"
diff -Nurd mono-2.0/data/config.in mono-2.0-new/data/config.in
--- mono-2.0/data/config.in	2008-09-02 11:42:56.000000000 -0400
+++ mono-2.0-new/data/config.in	2008-10-07 16:52:15.000000000 -0400
@@ -5,18 +5,18 @@
 	<dllmap dll="intl" name="bind_textdomain_codeset" target="@LIBC@" os="solaris"/>
 	<dllmap dll="libintl" name="bind_textdomain_codeset" target="@LIBC@" os="solaris"/>
 	<dllmap dll="libintl" target="@INTL@" os="!windows"/>
-	<dllmap dll="i:libxslt.dll" target="libxslt@libsuffix@" os="!windows"/>
-	<dllmap dll="i:odbc32.dll" target="libodbc@libsuffix@" os="!windows"/>
-	<dllmap dll="i:odbc32.dll" target="libiodbc.dylib" os="osx"/>
-	<dllmap dll="oci" target="libclntsh@libsuffix@" os="!windows"/>
-	<dllmap dll="db2cli" target="libdb2_36@libsuffix@" os="!windows"/>
-	<dllmap dll="MonoPosixHelper" target="libMonoPosixHelper@libsuffix@" os="!windows" />
+	<dllmap dll="i:libxslt.dll" target="@FINKPREFIX@/lib/libxslt.1@libsuffix@" os="!windows"/>
+	<dllmap dll="i:odbc32.dll" target="@FINKPREFIX@/lib/libodbc.1@libsuffix@" os="!windows"/>
+	<dllmap dll="i:odbc32.dll" target="/usr/lib/libiodbc.2.dylib" os="osx"/>
+	<dllmap dll="oci" target="@FINKPREFIX@/lib/libclntsh@libsuffix@" os="!windows"/>
+	<dllmap dll="db2cli" target="@FINKPREFIX@/lib/libdb2_36@libsuffix@" os="!windows"/>
+	<dllmap dll="MonoPosixHelper" target="@FINKPREFIX@/lib/libMonoPosixHelper@libsuffix@" os="!windows" />
 	<dllmap dll="i:msvcrt" target="@LIBC@" os="!windows"/>
 	<dllmap dll="i:msvcrt.dll" target="@LIBC@" os="!windows"/>
 	<dllmap dll="sqlite" target="@SQLITE@" os="!windows"/>
 	<dllmap dll="sqlite3" target="@SQLITE3@" os="!windows"/>
 	<dllmap dll="libX11" target="@X11@" os="!windows" />
-	<dllmap dll="libcairo-2.dll" target="libcairo.so.2" os="!windows"/>
+	<dllmap dll="libcairo-2.dll" target="@FINKPREFIX@/lib/libcairo.2.dylib" os="!windows"/>
 	<dllmap dll="i:kernel32.dll">
 		<dllentry dll="__Internal" name="CopyMemory" target="mono_win32_compat_CopyMemory"/>
 		<dllentry dll="__Internal" name="FillMemory" target="mono_win32_compat_FillMemory"/>
diff -Nurd mono-2.0/mono/metadata/loader.c mono-2.0-new/mono/metadata/loader.c
--- mono-2.0/mono/metadata/loader.c	2008-09-02 11:42:41.000000000 -0400
+++ mono-2.0-new/mono/metadata/loader.c	2008-10-07 16:52:15.000000000 -0400
@@ -1171,6 +1171,24 @@
 
 		if (!module) {
 			void *iter = NULL;
+			while ((full_name = mono_dl_build_path ("@FINKPREFIX@/lib", file_name, &iter))) {
+				mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_DLLIMPORT,
+					"DllImport loading library: '%s'.", full_name);
+				module = cached_module_load (full_name, MONO_DL_LAZY, &error_msg);
+				if (!module) {
+					mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_DLLIMPORT,
+						"DllImport error loading library '%s'.",
+						error_msg);
+					g_free (error_msg);
+				}
+				g_free (full_name);
+				if (module)
+					break;
+			}
+		}
+
+		if (!module) {
+			void *iter = NULL;
 			while ((full_name = mono_dl_build_path (".", file_name, &iter))) {
 				mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_DLLIMPORT,
 					"DllImport loading library: '%s'.", full_name);
