diff -u configure
--- a/ocaml-3.10.2/configure
+++ b/ocaml-3.10.2/configure
@@ -306,6 +306,8 @@
 
 # Configure compiler to use in further tests
 
+bytecccompopts="$CPPFLAGS $bytecclinkopts"
+bytecclinkopts="$LDFLAGS $bytecclinkopts"
 cc="$bytecc -O $bytecclinkopts"
 export cc cclibs verbose
 
diff -u infoman/ocaml.info
--- a/infoman/ocaml.info
+++ b/infoman/ocaml.info
@@ -1,4 +1,10 @@ This file has been translated from LaTeX
 This file has been translated from LaTeX by HeVeA.
+
+INFO-DIR-SECTION Programming Languages
+START-INFO-DIR-ENTRY
+* ocaml: (ocaml).               The Objective Caml compiler and programming environment
+END-INFO-DIR-ENTRY
+
 
 Indirect:
 ocaml.info-1: 51
