diff -ruN php-5.1.4.orig/ext/gd/gdcache.c php-5.1.4/ext/gd/gdcache.c
--- php-5.1.4.orig/ext/gd/gdcache.c	2005-01-09 14:05:05.000000000 -0700
+++ php-5.1.4/ext/gd/gdcache.c	2006-06-15 12:27:43.000000000 -0600
@@ -52,8 +52,21 @@
 /* implementation                                        */
 /*********************************************************/
 
-
 /* create a new cache */
+#ifdef __APPLE__
+extern gdCache_head_t *
+gdCacheCreate(
+        int                                     size,
+        gdCacheTestFn_t         gdCacheTest,
+        gdCacheFetchFn_t        gdCacheFetch,
+        gdCacheReleaseFn_t      gdCacheRelease );
+
+extern void
+gdCacheDelete( gdCache_head_t *head );
+
+extern void *
+gdCacheGet( gdCache_head_t *head, void *keydata );
+#else
 gdCache_head_t *
 gdCacheCreate(
 	int					size,
@@ -130,7 +143,7 @@
 	elem->userdata = userdata;
 	return userdata;
 }
-
+#endif
 
 
 /*********************************************************/
diff -ruN php-5.1.4.orig/fink/php5.conf php-5.1.4/fink/php5.conf
--- php-5.1.4.orig/fink/php5.conf	1969-12-31 17:00:00.000000000 -0700
+++ php-5.1.4/fink/php5.conf	2006-06-15 12:28:34.000000000 -0600
@@ -0,0 +1,4 @@
+<IfModule mod_php5.c>
+  AddType application/x-httpd-php .php .phtml .php3
+  AddType application/x-httpd-php-source .phps
+</IfModule>
diff -ruN php-5.1.4.orig/fink/php5.load php-5.1.4/fink/php5.load
--- php-5.1.4.orig/fink/php5.load	1969-12-31 17:00:00.000000000 -0700
+++ php-5.1.4/fink/php5.load	2006-06-15 12:28:59.000000000 -0600
@@ -0,0 +1 @@
+LoadModule php5_module @FINKPREFIX@/lib/apache2/modules/libphp5.so
diff -ruN php-5.1.4.orig/fink/fixpearconf php-5.1.4/fink/fixpearconf
--- php-5.1.4.orig/fink/fixpearconf	1969-12-31 17:00:00.000000000 -0700
+++ php-5.1.4/fink/fixpearconf	2006-06-15 12:28:59.000000000 -0600
@@ -0,0 +1,35 @@
+#!/usr/bin/perl -w
+
+$| = 1;
+use strict;
+
+my $arg = $ARGV[0];
+my $val = $ARGV[1];
+
+my ($count, $value, $line);
+
+if ($arg eq "conf") {
+	open(CONF, "<$val") or die "Couldn't read $val: $!\n";
+	while (<CONF>) {
+		if (/^.*(s\:\d+:\".+\/bin\/php\").*$/) {
+			$line = $1;
+			last;
+		}
+	}
+	close(CONF);
+} elsif ($arg eq "new") {
+	$value = $val."/bin/php5";
+	$count = length($value);
+	$line = 's:'.$count.':"'.$value.'"';
+} else {
+	print "First argument should be \"new\" or \"conf\"!\n";
+	exit;
+}
+
+if ($line) {
+	print "$line\n";
+} else {
+	die "Not found!\n";
+}
+
+exit;
