diff -ruN php-5.1.4.orig/ext/curl/interface.c php-5.1.4/ext/curl/interface.c
--- php-5.1.4.orig/ext/curl/interface.c	2006-04-13 07:26:10.000000000 -0400
+++ php-5.1.4/ext/curl/interface.c	2008-03-17 00:47:39.000000000 -0400
@@ -266,7 +266,9 @@
 	REGISTER_CURL_CONSTANT(CURLOPT_FTPAPPEND);
 	REGISTER_CURL_CONSTANT(CURLOPT_NETRC);
 	REGISTER_CURL_CONSTANT(CURLOPT_FOLLOWLOCATION);
+#if CURLOPT_FTPASCII != 0
 	REGISTER_CURL_CONSTANT(CURLOPT_FTPASCII);
+#endif
 	REGISTER_CURL_CONSTANT(CURLOPT_PUT);
 #if CURLOPT_MUTE != 0
 	REGISTER_CURL_CONSTANT(CURLOPT_MUTE);
@@ -306,7 +308,9 @@
 	REGISTER_CURL_CONSTANT(CURLOPT_FILETIME);
 	REGISTER_CURL_CONSTANT(CURLOPT_WRITEFUNCTION);
 	REGISTER_CURL_CONSTANT(CURLOPT_READFUNCTION);
+#if CURLOPT_PASSWDFUNCTION != 0 
 	REGISTER_CURL_CONSTANT(CURLOPT_PASSWDFUNCTION);
+#endif
 	REGISTER_CURL_CONSTANT(CURLOPT_HEADERFUNCTION);
 	REGISTER_CURL_CONSTANT(CURLOPT_MAXREDIRS);
 	REGISTER_CURL_CONSTANT(CURLOPT_MAXCONNECTS);
diff -ruN php-5.1.4.orig/ext/gd/gdcache.c php-5.1.4/ext/gd/gdcache.c
--- php-5.1.4.orig/ext/gd/gdcache.c	2005-01-09 14:05:05.000000000 -0700
+++ php-5.1.4/ext/gd/gdcache.c	2006-06-15 12:27:43.000000000 -0600
@@ -52,8 +52,21 @@
 /* implementation                                        */
 /*********************************************************/
 
-
 /* create a new cache */
+#ifdef __APPLE__
+extern gdCache_head_t *
+gdCacheCreate(
+        int                                     size,
+        gdCacheTestFn_t         gdCacheTest,
+        gdCacheFetchFn_t        gdCacheFetch,
+        gdCacheReleaseFn_t      gdCacheRelease );
+
+extern void
+gdCacheDelete( gdCache_head_t *head );
+
+extern void *
+gdCacheGet( gdCache_head_t *head, void *keydata );
+#else
 gdCache_head_t *
 gdCacheCreate(
 	int					size,
@@ -130,7 +143,7 @@
 	elem->userdata = userdata;
 	return userdata;
 }
-
+#endif
 
 
 /*********************************************************/
diff -ruN php-5.1.4.orig/ext/imap/php_imap.c php-5.1.4/ext/imap/php_imap.c
--- php-5.1.4.orig/ext/imap/php_imap.c	2006-01-28 03:07:20.000000000 -0500
+++ php-5.1.4/ext/imap/php_imap.c	2008-03-17 01:06:16.000000000 -0400
@@ -71,6 +71,7 @@
 /* the gets we use */
 static char *php_mail_gets(readfn_t f, void *stream, unsigned long size, GETS_DATA *md);
 
+#if 0
 /* These function declarations are missing from the IMAP header files... */
 void rfc822_date(char *date);
 char *cpystr(const char *str);
@@ -79,6 +80,7 @@
 unsigned long find_rightmost_bit(unsigned long *valptr);
 void fs_give(void **block);
 void *fs_get(size_t size);
+#endif
 
 
 /* {{{ imap_functions[]
@@ -2160,7 +2162,7 @@
 	dest.size = 0;
 
 	cpytxt(&src, Z_STRVAL_PP(str), Z_STRLEN_PP(str));
-	utf8_mime2text(&src, &dest);
+	utf8_mime2text(&src, &dest, U8T_CANONICAL);
 	RETURN_STRINGL(dest.data, strlen(dest.data), 1);
 }
 /* }}} */
diff -ruN php-5.1.4.orig/fink/php5.conf php-5.1.4/fink/php5.conf
--- php-5.1.4.orig/fink/php5.conf	1969-12-31 17:00:00.000000000 -0700
+++ php-5.1.4/fink/php5.conf	2006-06-15 12:28:34.000000000 -0600
@@ -0,0 +1,4 @@
+<IfModule mod_php5.c>
+  AddType application/x-httpd-php .php .phtml .php3
+  AddType application/x-httpd-php-source .phps
+</IfModule>
diff -ruN php-5.1.4.orig/fink/php5.load php-5.1.4/fink/php5.load
--- php-5.1.4.orig/fink/php5.load	1969-12-31 17:00:00.000000000 -0700
+++ php-5.1.4/fink/php5.load	2006-06-15 12:28:59.000000000 -0600
@@ -0,0 +1 @@
+LoadModule php5_module @FINKPREFIX@/lib/apache2/modules/libphp5.so
diff -ruN php-5.1.4.orig/fink/fixpearconf php-5.1.4/fink/fixpearconf
--- php-5.1.4.orig/fink/fixpearconf	1969-12-31 17:00:00.000000000 -0700
+++ php-5.1.4/fink/fixpearconf	2006-06-15 12:28:59.000000000 -0600
@@ -0,0 +1,35 @@
+#!/usr/bin/perl -w
+
+$| = 1;
+use strict;
+
+my $arg = $ARGV[0];
+my $val = $ARGV[1];
+
+my ($count, $value, $line);
+
+if ($arg eq "conf") {
+	open(CONF, "<$val") or die "Couldn't read $val: $!\n";
+	while (<CONF>) {
+		if (/^.*(s\:\d+:\".+\/bin\/php\").*$/) {
+			$line = $1;
+			last;
+		}
+	}
+	close(CONF);
+} elsif ($arg eq "new") {
+	$value = $val."/bin/php5";
+	$count = length($value);
+	$line = 's:'.$count.':"'.$value.'"';
+} else {
+	print "First argument should be \"new\" or \"conf\"!\n";
+	exit;
+}
+
+if ($line) {
+	print "$line\n";
+} else {
+	die "Not found!\n";
+}
+
+exit;
