diff -Naur --exclude='*~' scheme48-1.8/Makefile.in scheme48-1.8.fink/Makefile.in
--- scheme48-1.8/Makefile.in	2008-01-26 07:06:27.000000000 -0800
+++ scheme48-1.8.fink/Makefile.in	2008-04-21 20:33:00.000000000 -0700
@@ -409,7 +409,7 @@
 	done
 
 inst-config-script:
-	config_script=$(bindir)/$(CONFIG_SCRIPT) &&			\
+	config_script=$(DESTDIR)$(bindir)/$(CONFIG_SCRIPT) &&			\
 	sed -e 's,@incdir\@,$(incdir),g'				\
 	    -e 's,@libdir\@,$(libdir),g'				\
 	    -e 's,@CC\@,$(CC),g'					\
