diff -Nurd -x'*~' tcltk-8.4.19.orig/tcl8.4.19/unix/configure tcltk-8.4.19/tcl8.4.19/unix/configure
--- tcltk-8.4.19.orig/tcl8.4.19/unix/configure	2008-04-17 16:29:49.000000000 -0400
+++ tcltk-8.4.19/tcl8.4.19/unix/configure	2008-07-28 18:15:21.000000000 -0400
@@ -9246,7 +9246,7 @@
         else
             TCL_LIB_FLAG="-ltcl`echo ${TCL_VERSION} | tr -d .`\${TCL_DBGX}"
         fi
-        TCL_BUILD_LIB_SPEC="-L`pwd` ${TCL_LIB_FLAG}"
+        TCL_BUILD_LIB_SPEC="`pwd`/${TCL_LIB_FILE}"
         TCL_LIB_SPEC="-L${libdir} ${TCL_LIB_FLAG}"
     else
         TCL_BUILD_EXP_FILE="lib.exp"
@@ -9303,7 +9303,7 @@
     TCL_STUB_LIB_FLAG="-ltclstub`echo ${TCL_VERSION} | tr -d .`\${TCL_DBGX}"
 fi
 
-TCL_BUILD_STUB_LIB_SPEC="-L`pwd` ${TCL_STUB_LIB_FLAG}"
+TCL_BUILD_STUB_LIB_SPEC="`pwd`/${TCL_STUB_LIB_FILE}"
 TCL_STUB_LIB_SPEC="-L${TCL_STUB_LIB_DIR} ${TCL_STUB_LIB_FLAG}"
 TCL_BUILD_STUB_LIB_PATH="`pwd`/${TCL_STUB_LIB_FILE}"
 TCL_STUB_LIB_PATH="${TCL_STUB_LIB_DIR}/${TCL_STUB_LIB_FILE}"
@@ -9372,7 +9372,7 @@
 
 
 
-CFLAGS="${CFLAGS} ${CPPFLAGS}"; CPPFLAGS=""
+EXTRA_CC_SWITCHES="${EXTRA_CC_SWITCHES} ${CPPFLAGS}"; CPPFLAGS=""
 
 
 tcl_config_files="${tcl_config_files} Makefile dltest/Makefile tclConfig.sh"
diff -Nurd -x'*~' tcltk-8.4.19.orig/tk8.4.19/unix/configure tcltk-8.4.19/tk8.4.19/unix/configure
--- tcltk-8.4.19.orig/tk8.4.19/unix/configure	2008-04-17 11:14:08.000000000 -0400
+++ tcltk-8.4.19/tk8.4.19/unix/configure	2008-07-28 18:15:21.000000000 -0400
@@ -6012,7 +6012,7 @@
         else
             eval TK_LIB_FLAG="-ltk`echo ${VERSION} | tr -d .`\${TK_DBGX}"
         fi
-        TK_BUILD_LIB_SPEC="-L`pwd` ${TK_LIB_FLAG}"
+        TK_BUILD_LIB_SPEC="`pwd`/${TK_LIB_FILE}"
         TK_LIB_SPEC="-L${libdir} ${TK_LIB_FLAG}"
     else
         TK_BUILD_EXP_FILE="lib.exp"
@@ -6058,7 +6058,7 @@
     eval TK_STUB_LIB_FLAG="-ltkstub`echo ${TK_VERSION} | tr -d .`\${TK_DBGX}"
 fi
 
-TK_BUILD_STUB_LIB_SPEC="-L`pwd` ${TK_STUB_LIB_FLAG}"
+TK_BUILD_STUB_LIB_SPEC="`pwd`/${TK_STUB_LIB_FILE}"
 TK_STUB_LIB_SPEC="-L${TK_STUB_LIB_DIR} ${TK_STUB_LIB_FLAG}"
 TK_BUILD_STUB_LIB_PATH="`pwd`/${TK_STUB_LIB_FILE}"
 TK_STUB_LIB_PATH="${TK_STUB_LIB_DIR}/${TK_STUB_LIB_FILE}"
@@ -6126,7 +6126,7 @@
 
 
 
-CFLAGS="${CFLAGS} ${CPPFLAGS}"; CPPFLAGS=""
+EXTRA_CC_SWITCHES="${EXTRA_CC_SWITCHES} ${CPPFLAGS}"; CPPFLAGS=""
 
 
 tk_config_files="${tk_config_files} Makefile tkConfig.sh"
diff -Nurd -x'*~' tcltk-8.4.19.orig/tk8.4.19/generic/tk.h tcltk-8.4.19/tk8.4.19/generic/tk.h
--- tcltk-8.4.19.orig/tk8.4.19/generic/tk.h	2008-04-07 15:17:54.000000000 -0400
+++ tcltk-8.4.19/tk8.4.19/generic/tk.h	2008-07-28 18:15:21.000000000 -0400
@@ -635,17 +635,18 @@
  *
  *---------------------------------------------------------------------------
  */
-#define VirtualEvent	    (LASTEvent)
-#define ActivateNotify	    (LASTEvent + 1)
-#define DeactivateNotify    (LASTEvent + 2)
-#define MouseWheelEvent     (LASTEvent + 3)
-#define TK_LASTEVENT	    (LASTEvent + 4)
+
+#define VirtualEvent        35 
+#define ActivateNotify	    (VirtualEvent + 1)
+#define DeactivateNotify    (VirtualEvent + 2)
+#define MouseWheelEvent     (VirtualEvent + 3)
+#define TK_LASTEVENT	    (VirtualEvent + 4)
 
 #define MouseWheelMask	    (1L << 28)
 
 #define ActivateMask	    (1L << 29)
 #define VirtualEventMask    (1L << 30)
-#define TK_LASTEVENT	    (LASTEvent + 4)
+
 
 
 /*
