--- tcltk-8.5.6.orig/tcl8.5.6/generic/tclPort.h	2008-11-04 18:56:59.000000000 -0500
+++ tcltk-8.5.6/tcl8.5.6/generic/tclPort.h	2009-04-15 01:52:52.000000000 -0400
@@ -24,7 +24,7 @@
 #if defined(__WIN32__)
 #   include "tclWinPort.h"
 #else
-#   include "tclUnixPort.h"
+#   include "../unix/tclUnixPort.h"
 #endif
 
 #if !defined(LLONG_MIN)
--- tcltk-8.5.6.orig/tcl8.5.6/generic/tclInt.h	2008-11-13 19:22:39.000000000 -0500
+++ tcltk-8.5.6/tcl8.5.6/generic/tclInt.h	2009-04-15 01:52:52.000000000 -0400
@@ -2773,7 +2773,7 @@
 MODULE_SCOPE int	TclClockOldscanObjCmd(
 			    ClientData clientData, Tcl_Interp *interp,
 			    int objc, Tcl_Obj *const objv[]);
-MODULE_SCOPE int	Tcl_CloseObjCmd(ClientData clientData,
+extern int	Tcl_CloseObjCmd(ClientData clientData,
 			    Tcl_Interp *interp, int objc,
 			    Tcl_Obj *const objv[]);
 MODULE_SCOPE int	Tcl_ConcatObjCmd(ClientData clientData,
@@ -2934,7 +2934,7 @@
 MODULE_SCOPE int	Tcl_RenameObjCmd(ClientData clientData,
 			    Tcl_Interp *interp, int objc,
 			    Tcl_Obj *const objv[]);
-MODULE_SCOPE int	Tcl_ReturnObjCmd(ClientData clientData,
+extern int	Tcl_ReturnObjCmd(ClientData clientData,
 			    Tcl_Interp *interp, int objc,
 			    Tcl_Obj *const objv[]);
 MODULE_SCOPE int	Tcl_ScanObjCmd(ClientData clientData,
