diff -ru ucblogo-5.5-orig/emacs/makefile ucblogo-5.5/emacs/makefile
--- ucblogo-5.5-orig/emacs/makefile	2007-11-23 23:48:50.000000000 +0100
+++ ucblogo-5.5/emacs/makefile	2007-11-23 23:50:17.000000000 +0100
@@ -1,6 +1,6 @@
 BUILDIR	      = `pwd`
 EMACSDIR      = $(LIBLOC)/emacs
-INFODIR	      = $(prefix)/info
+INFODIR	      = $(prefix)/share/info
 LOADPATH      = $(subst X,$(EMACSDIR),(setq load-path (cons "\"X\"" load-path)))
 LOGOBIN	      = $(subst X,$(BINDIR),(setq logo-binary-name \"X/logo\"))
 INFOPATH      = $(subst X,$(INFODIR),(setq logo-info-file \"X/ucblogo.info\"))
diff -ru ucblogo-5.5-orig/makefile.in ucblogo-5.5/makefile.in
--- ucblogo-5.5-orig/makefile.in	2007-11-23 23:48:50.000000000 +0100
+++ ucblogo-5.5/makefile.in	2007-11-23 23:51:20.000000000 +0100
@@ -4,7 +4,7 @@
 LIBS  = @X_PRE_LIBS@ @X_LIBS@ @LIBS@ @X_EXTRA_LIBS@
 prefix = @prefix@
 BINDIR        = $(prefix)/bin
-LIBLOC        = $(prefix)/lib/logo
+LIBLOC        = $(prefix)/lib/ucblogo
 MAKE = @MAKE@
 
 # LIBLOC      = `pwd`
