diff -Naur Canna36p1.old/Canna.conf Canna36p1.new/Canna.conf
--- Canna36p1.old/Canna.conf	Mon Dec  2 09:44:59 2002
+++ Canna36p1.new/Canna.conf	Thu Dec  5 00:58:03 2002
@@ -132,29 +132,29 @@
 
 /* rootにならずにインストールする時はこれを定義して下さい */
 /* #define InstallAsUser */
-cannaPrefix = /usr/local/canna
+cannaPrefix = @PREFIX@
 cannaExecPrefix = $(cannaPrefix)
 cannaBinDir = $(cannaExecPrefix)/bin
-cannaSrvDir = $(cannaExecPrefix)/bin
-cannaLibDir = $(cannaPrefix)/lib
-cannaManDir = $(cannaPrefix)/man
+cannaSrvDir = $(cannaExecPrefix)/sbin
+cannaLibDir = $(cannaPrefix)/var/lib/canna
+cannaManDir = $(cannaPrefix)/share/man
 cannaIncDir = $(cannaPrefix)/include/canna
 
 #ifdef InstallAsUser
 libCannaDir = $(cannaExecPrefix)/lib
 ErrDir  = $(cannaExecPrefix)/log
 #else
-libCannaDir = /usr/lib
-ErrDir  = /usr/spool/canna
+libCannaDir = $(cannaPrefix)/lib
+ErrDir  =  $(cannaPrefix)/var/log/canna
 #endif
 
 /* 次の3行は変更しないでください */
 wcharDefinition = -DCANNA_WCHAR
 Wlib =
-JapaneseLocale = japanese
+JapaneseLocale = ja
 
-cannaOwner = bin
-cannaGroup = bin
+cannaOwner = canna
+cannaGroup = canna
 
 #ifdef InstallAsUser
 cannaOwnerGroup =
@@ -164,7 +164,7 @@
 
 pointerIntegerDef =
 
-cannaDsoRev = 1
+cannaDsoRev = 1.0
     /* SunOS 4.x をお使いの方で shared object を利用する場合には
             cannaDsoRev = 1.1
     と指定してください。また、FreeBSD を利用する場合は、
@@ -177,7 +177,7 @@
 #elif defined(hpux)
 sharedLibExtension = sl.$(cannaDsoRev)
 #else
-sharedLibExtension = so.$(cannaDsoRev)
+sharedLibExtension = $(cannaDsoRev).dylib
 #endif
 
 /* iroha.t の場所を指定。pubdic+ があればそちらを指定します。 */
@@ -188,16 +188,16 @@
 CHGRP = :
 CHMOD = :
 #else
-CHOWN = chown
-CHGRP = chgrp
-CHMOD = chmod
+CHOWN = /usr/sbin/chown
+CHGRP = /usr/bin/chgrp
+CHMOD = /bin/chmod
 #endif
 
 /* 日本語マニュアルを使わないのであればコメントアウト */
-#define JAPANESEMAN
+/* #define JAPANESEMAN */
 
 /* locale ディレクトリを持っている場合は以下を記述する */
-/* #define LOCALE_DIR /usr/lib/locale */
+#define LOCALE_DIR /usr/share/locale
 
 /* エンジンスイッチ機能を使うのであればコメントをはずす */
 /* ※注 Canna version 3.5 ではコメントをはずさないでください */
@@ -231,7 +231,7 @@
 /* #define UseInstalledLibCanna YES */
 #define UseInstalledLibCanna NO
 
-DicDir   = $(cannaLibDir)/dic
+DicDir   = $(cannaPrefix)/var/lib/canna/dic
 
 #ifdef NeXT
 #define NO_X_ENV
diff -Naur Canna36p1.old/canna/protodefs.h Canna36p1.new/canna/protodefs.h
--- Canna36p1.old/canna/protodefs.h	Sun Oct 20 17:20:33 2002
+++ Canna36p1.new/canna/protodefs.h	Thu Dec  5 00:58:03 2002
@@ -78,7 +78,7 @@
 #define I16toI32(x) (x)
 #define I8toI32(x) (((x) & 0x80) ? ((x) | 0xffffff80) : (x))
 
-#if defined(__bsdi__) || defined(__FreeBSD__)
+#if defined(__bsdi__) || defined(__FreeBSD__) || defined(__APPLE__)
 #include <machine/endian.h>
 #endif
 
diff -Naur Canna36p1.old/canna/widedef.h Canna36p1.new/canna/widedef.h
--- Canna36p1.old/canna/widedef.h	Sat Oct 19 17:27:28 2002
+++ Canna36p1.new/canna/widedef.h	Thu Dec  5 00:58:04 2002
@@ -35,7 +35,7 @@
 #define WCHAR16
 #endif
      
-#if defined(__FreeBSD__) || defined(__NetBSD__)
+#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__APPLE__)
 # include <machine/ansi.h>
 #endif
 
@@ -61,7 +61,7 @@
 #endif /* !nec_ews_svr2 */
 #endif /* HAVE_WCHAR_OPERATION */
 
-#if defined(__FreeBSD__) || defined(__NetBSD__)
+#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__APPLE__)
 # ifdef _BSD_WCHAR_T_
 #  undef _BSD_WCHAR_T_
 #  ifdef WCHAR16
diff -Naur Canna36p1.old/cmd/catdic/Imakefile Canna36p1.new/cmd/catdic/Imakefile
--- Canna36p1.old/cmd/catdic/Imakefile	Thu Nov  7 23:46:02 2002
+++ Canna36p1.new/cmd/catdic/Imakefile	Thu Dec  5 01:03:11 2002
@@ -85,14 +85,14 @@
 	$(RM) $(DESTDIR)$(cannaBinDir)/addwords
 	$(RM) $(DESTDIR)$(cannaBinDir)/delwords
 	$(RM) $(DESTDIR)$(cannaBinDir)/cannakill
-	$(LN) $(DESTDIR)$(cannaBinDir)/catdic $(DESTDIR)$(cannaBinDir)/cpdic
-	$(LN) $(DESTDIR)$(cannaBinDir)/catdic $(DESTDIR)$(cannaBinDir)/lsdic
-	$(LN) $(DESTDIR)$(cannaBinDir)/catdic $(DESTDIR)$(cannaBinDir)/mkdic
-	$(LN) $(DESTDIR)$(cannaBinDir)/catdic $(DESTDIR)$(cannaBinDir)/mvdic
-	$(LN) $(DESTDIR)$(cannaBinDir)/catdic $(DESTDIR)$(cannaBinDir)/rmdic
-	$(LN) $(DESTDIR)$(cannaBinDir)/catdic $(DESTDIR)$(cannaBinDir)/syncdic
-	$(LN) $(DESTDIR)$(cannaBinDir)/catdic $(DESTDIR)$(cannaBinDir)/chmoddic
-	$(LN) $(DESTDIR)$(cannaBinDir)/catdic $(DESTDIR)$(cannaBinDir)/addwords
-	$(LN) $(DESTDIR)$(cannaBinDir)/catdic $(DESTDIR)$(cannaBinDir)/delwords
-	$(LN) $(DESTDIR)$(cannaBinDir)/catdic $(DESTDIR)$(cannaBinDir)/cannakill
+	(cd $(DESTDIR)$(cannaBinDir); $(LN) catdic cpdic)
+	(cd $(DESTDIR)$(cannaBinDir); $(LN) catdic lsdic)
+	(cd $(DESTDIR)$(cannaBinDir); $(LN) catdic mkdic)
+	(cd $(DESTDIR)$(cannaBinDir); $(LN) catdic mvdic)
+	(cd $(DESTDIR)$(cannaBinDir); $(LN) catdic rmdic)
+	(cd $(DESTDIR)$(cannaBinDir); $(LN) catdic syncdic)
+	(cd $(DESTDIR)$(cannaBinDir); $(LN) catdic chmoddic)
+	(cd $(DESTDIR)$(cannaBinDir); $(LN) catdic addwords)
+	(cd $(DESTDIR)$(cannaBinDir); $(LN) catdic delwords)
+	(cd $(DESTDIR)$(cannaBinDir); $(LN) catdic cannakill)
 #endif
diff -Naur Canna36p1.old/dic/ideo/grammar/Imakefile Canna36p1.new/dic/ideo/grammar/Imakefile
--- Canna36p1.old/dic/ideo/grammar/Imakefile	Sat Oct 19 17:27:34 2002
+++ Canna36p1.new/dic/ideo/grammar/Imakefile	Thu Dec  5 00:58:04 2002
@@ -85,6 +85,6 @@
 InstallMultipleFlags($(ALLDIC),$(CANNADICDIR),-m 0664 $(cannaOwnerGroup))
 
 install::
-	$(CHGRP) $(cannaGroup) $(CANNADICDIR)
-	$(CHOWN) $(cannaOwner) $(CANNADICDIR)
-	$(CHMOD) ug+w $(CANNADICDIR)
+	$(CHGRP) $(cannaGroup) $(DESTDIR)$(CANNADICDIR)
+	$(CHOWN) $(cannaOwner) $(DESTDIR)$(CANNADICDIR)
+	$(CHMOD) ug+w $(DESTDIR)$(CANNADICDIR)
diff -Naur Canna36p1.old/lib/RKC/rkc.c Canna36p1.new/lib/RKC/rkc.c
--- Canna36p1.old/lib/RKC/rkc.c	Sun Oct 20 23:29:59 2002
+++ Canna36p1.new/lib/RKC/rkc.c	Thu Dec  5 00:58:18 2002
@@ -241,7 +241,7 @@
 
 int RkwSetUserInfo pro((char *, char *, char *));
 
-RkUserInfo *uinfo;
+static RkUserInfo *uinfo;
 
 int
 RkwSetUserInfo(user, group, topdir)
diff -Naur Canna36p1.old/server/misc.c Canna36p1.new/server/misc.c
--- Canna36p1.old/server/misc.c	Mon Oct 21 03:00:22 2002
+++ Canna36p1.new/server/misc.c	Thu Dec  5 00:58:18 2002
@@ -291,15 +291,12 @@
 	exit( 1 ) ;
     }
     if ( parent ) {
-        signal(SIGTERM, parQUIT);
-	pause() ;
 	exit( 0 ) ;
 	/* wait( (int *)0 ) ;	*/
     }
-    return parentid;
-#else
-    return 0;
 #endif
+
+    return 0;
 }
 
 void
diff -Naur Canna36p1.old/update-canna-dics-dir Canna36p1.new/update-canna-dics-dir
--- Canna36p1.old/update-canna-dics-dir	Thu Jan  1 09:00:00 1970
+++ Canna36p1.new/update-canna-dics-dir	Thu Dec  5 00:58:18 2002
@@ -0,0 +1,14 @@
+#!/bin/sh -e
+
+PATH=/bin:/sbin:/usr/bin:/usr/sbin:@PREFIX@/bin:@PREFIX@/sbin
+
+LIST_FILE=@PREFIX@/var/lib/canna/dic/canna/dics.dir
+LIST_DIR=@PREFIX@/var/lib/canna/dics.d
+
+TMPFILE=`tempfile`
+cat ${LIST_DIR}/* >>${TMPFILE}
+mv ${TMPFILE} ${LIST_FILE}
+chmod 0644 ${LIST_FILE}
+chown canna:canna ${LIST_FILE}
+
+exit 0
