diff -Naur Canna35b2.old/Canna.conf Canna35b2.new/Canna.conf
--- Canna35b2.old/Canna.conf	Thu Nov 28 19:27:01 1996
+++ Canna35b2.new/Canna.conf	Sat Mar 16 03:05:12 2002
@@ -128,49 +128,49 @@
  * $Id: Canna.conf,v 3.13 1996/11/28 10:27:01 kon Exp $
  */
 
-cannaBinDir = /usr/local/canna/bin
-cannaSrvDir = /usr/local/canna/bin
-cannaLibDir = /usr/local/canna/lib
-cannaManDir = /usr/local/canna/man
-cannaIncDir = /usr/local/canna/include/canna
-libCannaDir = /usr/lib
+cannaBinDir = @PREFIX@/bin
+cannaSrvDir = @PREFIX@/sbin
+cannaLibDir = @PREFIX@/etc/canna
+cannaManDir = @PREFIX@/share/man
+cannaIncDir = @PREFIX@/include/canna
+libCannaDir = @PREFIX@/lib
 
-ErrDir  = /usr/spool/canna
+ErrDir  = @PREFIX@/var/log/canna
 
 wcharDefinition = -DCANNA_WCHAR
 Wlib =
-JapaneseLocale = japanese
+JapaneseLocale = ja
 
-cannaOwner = bin
-cannaGroup = bin
+cannaOwner = canna
+cannaGroup = canna
 
 cannaOwnerGroup = -o $(cannaOwner) -g $(cannaGroup)
 
 pointerIntegerDef =
 
-cannaDsoRev = 1
+cannaDsoRev = 1.0
     /* SunOS 4.x をお使いの方で shared object を利用する場合には
             cannaDsoRev = 1.1
     と指定してください。また、FreeBSD を利用する場合は、
             cannaDsoRev = 1.0
     と指定してください。 */
 
-sharedLibExtension = so.$(cannaDsoRev)
+sharedLibExtension = $(cannaDsoRev).dylib
 
 /* iroha.t の場所を指定。pubdic+ があればそちらを指定します。 */
 pubdicDir = $(CANNAROOT)/dic/ideo/pubdic
 
 /* 以下の３つは root にならないでインストールするのであれば echo に
    変えてください。 */
-CHOWN = chown
-CHGRP = chgrp
-CHMOD = chmod
+CHOWN = /usr/sbin/chown
+CHGRP = /usr/bin/chgrp
+CHMOD = /bin/chmod
 
 /* 日本語マニュアルを使わないのであればコメントアウト */
-#define JAPANESEMAN
+/* #define JAPANESEMAN */
 
 /* locale ディレクトリを持っている場合は以下を記述する */
-/* #define LOCALE_DIR /usr/lib/locale */
+#define LOCALE_DIR /usr/share/locale
 
 /* エンジンスイッチ機能を使うのであればコメントをはずす */
 /* ※注 Canna version 3.5 ではコメントをはずさないでください */
@@ -204,7 +204,7 @@
 /* #define UseInstalledLibCanna YES */
 #define UseInstalledLibCanna NO
 
-DicDir   = $(cannaLibDir)/dic
+DicDir   = @PREFIX@/var/lib/canna/dic
 
 #ifdef NeXT
 #define NO_X_ENV
diff -Naur Canna35b2.old/Imakefile Canna35b2.new/Imakefile
--- Canna35b2.old/Imakefile	Wed Nov 27 16:01:34 1996
+++ Canna35b2.new/Imakefile	Sat Mar 16 02:40:30 2002
@@ -13,7 +13,7 @@
 clean::
 	$(RM) -r include cannaconf.h
 
-includes::
+includes:: cannaconf.h
 	$(RM) -r include
 	MakeDir($(INCLUDE))
 
diff -Naur Canna35b2.old/canna/protodefs.h Canna35b2.new/canna/protodefs.h
--- Canna35b2.old/canna/protodefs.h	Thu Oct 24 13:20:29 1996
+++ Canna35b2.new/canna/protodefs.h	Sat Mar 16 01:36:33 2002
@@ -79,7 +79,7 @@
 #define I16toI32(x) (x)
 #define I8toI32(x) (((x) & 0x80) ? ((x) | 0xffffff80) : (x))
 
-#if defined(__bsdi__) || defined(__FreeBSD__)
+#if defined(__bsdi__) || defined(__FreeBSD__) || defined(__APPLE__)
 #include <machine/endian.h>
 #endif
 
diff -Naur Canna35b2.old/canna/widedef.h Canna35b2.new/canna/widedef.h
--- Canna35b2.old/canna/widedef.h	Fri Nov 29 22:00:55 1996
+++ Canna35b2.new/canna/widedef.h	Sat Mar 16 01:36:33 2002
@@ -35,7 +35,7 @@
 #define WCHAR16
 #endif
      
-#if defined(__FreeBSD__) || defined(__NetBSD__)
+#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__APPLE__)
 # include <machine/ansi.h>
 #endif
 
@@ -61,7 +61,7 @@
 #endif /* !nec_ews_svr2 */
 #endif /* HAVE_WCHAR_OPERATION */
 
-#if defined(__FreeBSD__) || defined(__NetBSD__)
+#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__APPLE__)
 # ifdef _BSD_WCHAR_T_
 #  undef _BSD_WCHAR_T_
 #  ifdef WCHAR16
diff -Naur Canna35b2.old/dic/ideo/grammar/Imakefile Canna35b2.new/dic/ideo/grammar/Imakefile
--- Canna35b2.old/dic/ideo/grammar/Imakefile	Wed Nov 27 16:14:45 1996
+++ Canna35b2.new/dic/ideo/grammar/Imakefile	Sat Mar 16 01:36:33 2002
@@ -85,6 +85,6 @@
 InstallMultipleFlags($(ALLDIC),$(CANNADICDIR),-m 0664 $(cannaOwnerGroup))
 
 install::
-	$(CHGRP) $(cannaGroup) $(CANNADICDIR)
-	$(CHOWN) $(cannaOwner) $(CANNADICDIR)
-	$(CHMOD) ug+w $(CANNADICDIR)
+	$(CHGRP) $(cannaGroup) $(DESTDIR)$(CANNADICDIR)
+	$(CHOWN) $(cannaOwner) $(DESTDIR)$(CANNADICDIR)
+	$(CHMOD) ug+w $(DESTDIR)$(CANNADICDIR)
diff -Naur Canna35b2.old/dic/ideo/words/Imakefile Canna35b2.new/dic/ideo/words/Imakefile
--- Canna35b2.old/dic/ideo/words/Imakefile	Wed Nov 27 16:14:55 1996
+++ Canna35b2.new/dic/ideo/words/Imakefile	Sat Mar 16 01:36:33 2002
@@ -69,13 +69,13 @@
 #endif
 
 install::
-	$(CHGRP) $(cannaGroup) $(CANNADICDIR)
-	$(CHOWN) $(cannaOwner) $(CANNADICDIR)
-	$(CHMOD) ug+w $(CANNADICDIR)
-	$(CHGRP) $(cannaGroup) $(DicDir)
-	$(CHOWN) $(cannaOwner) $(DicDir)
-	$(CHMOD) ug+w $(DicDir)
-	$(CHMOD) -w $(CANNADICDIR)/dics.dir
+	$(CHGRP) $(cannaGroup) $(DESTDIR)$(CANNADICDIR)
+	$(CHOWN) $(cannaOwner) $(DESTDIR)$(CANNADICDIR)
+	$(CHMOD) ug+w $(DESTDIR)$(CANNADICDIR)
+	$(CHGRP) $(cannaGroup) $(DESTDIR)$(DicDir)
+	$(CHOWN) $(cannaOwner) $(DESTDIR)$(DicDir)
+	$(CHMOD) ug+w $(DESTDIR)$(DicDir)
+	$(CHMOD) -w $(DESTDIR)$(CANNADICDIR)/dics.dir
 
 #ifndef USE_OBSOLETE_STYLE_FILENAME
 InstallNamedNonExec(necgaiji.t,necgaiji.ctd,$(CANNADICDIR))
diff -Naur Canna35b2.old/lib/RKC/rkc.c Canna35b2.new/lib/RKC/rkc.c
--- Canna35b2.old/lib/RKC/rkc.c	Wed Nov 27 16:21:11 1996
+++ Canna35b2.new/lib/RKC/rkc.c	Sat Mar 16 01:36:33 2002
@@ -253,7 +253,7 @@
 
 int RkwSetUserInfo pro((char *, char *, char *));
 
-RkUserInfo *uinfo;
+static RkUserInfo *uinfo;
 
 int
 RkwSetUserInfo(user, group, topdir)
diff -Naur Canna35b2.old/lib/canna/Imakefile Canna35b2.new/lib/canna/Imakefile
--- Canna35b2.old/lib/canna/Imakefile	Wed Nov 27 16:23:42 1996
+++ Canna35b2.new/lib/canna/Imakefile	Sat Mar 16 02:41:11 2002
@@ -116,6 +116,8 @@
 #endif /* ProjectX */
 #endif /* DoSharedLib */
 
+includes:: $(HEADERS)
+
 sglobal.h: GSYM.awk globnames
 	awk -f GSYM.awk < globnames > $@
 
