diff -bur glark-1.6.5/ glark-1.6.5-patched/
--- glark-1.6.5/Makefile	Thu Aug 21 10:00:26 2003
+++ glark-1.6.5-patched/Makefile	Wed Sep  3 14:37:57 2003
@@ -6,6 +6,8 @@
 DIST_SOURCES  = glark.ps glark.pdf glark.1 glark Makefile ChangeLog
 INSTALL       = /usr/bin/install -c
 PREFIX       ?= /usr
+bindir        = $(PREFIX)/bin
+mandir        = $(PREFIX)/share/man
 
 distdir       = ${PACKAGE}-${VERSION}
 srcdir        = .
@@ -74,13 +76,17 @@
 	  fi; \
 	done
 
-install: glark.1
-	${INSTALL} glark $(PREFIX)/bin
-	${INSTALL} glark.1 $(PREFIX)/man/man1
+installdirs:
+	${INSTALL} -d $(bindir)
+	${INSTALL} -d $(mandir)/man1
+
+install: glark.1 installdirs
+	${INSTALL} -m 755 glark $(bindir)
+	${INSTALL} -m 644 glark.1 $(mandir)/man1
 
 uninstall: glark.1
-	-rm -f $(PREFIX)/bin/glark
-	-rm -f $(PREFIX)/man/man1/glark.1
+	-rm -f $(bindir)/glark
+	-rm -f $(mandir)/man1/glark.1
 
 clean:
 	rm -f glark.ps glark.pdf glark.1 glark.dvi glark.tex
