diff -Naur kinput2-v3.old/Kinput2.conf kinput2-v3.new/Kinput2.conf
--- kinput2-v3.old/Kinput2.conf	Thu May 27 08:55:18 1999
+++ kinput2-v3.new/Kinput2.conf	Tue Oct  9 00:45:02 2001
@@ -18,7 +18,7 @@
 /* #define UseWnn */	/* define if you are going to use Wnn (v4 or v6) */
 /* #define UseWnn6 */	/* define if you are going to use Wnn6 */
 #define UseCanna	/* define if you are going to use Canna */
-#define UseSj3		/* define if you are going to use Sj3 */
+/* #define UseSj3 */	/* define if you are going to use Sj3 */
 
 
 XCOMM Wnn configuration
@@ -77,17 +77,17 @@
  * If you have already installed Canna header files and libraries..
  */
 XCOMM use installed headers/libraries
-XCOMM CANNAINSTDIR = /usr/local/canna
-XCOMM CANNASRC = $(CANNAINSTDIR)/include
-XCOMM CANNALIB = -lcanna16
+CANNAINSTDIR = $(PREFIX)
+CANNASRC = $(PREFIX)/include
+CANNALIB = -L$(PREFIX)/lib -lcanna16
 
 /*
  * If you have compiled Canna that came with X11R6 (contrib/programs/Canna),
  * but not installed yet..
  */
 XCOMM use headers/libraries in the source tree
-CANNASRC = $(CONTRIBSRC)/programs/Canna32
-CANNALIB = -L$(CANNASRC)/lib/canna16 -lcanna16
+XCOMM CANNASRC = $(CONTRIBSRC)/programs/Canna32
+XCOMM CANNALIB = -L$(CANNASRC)/lib/canna16 -lcanna16
 
 
 XCOMM SJ3 configuration
@@ -116,8 +116,8 @@
  * Use compiled library in the source tree..
  */
 XCOMM use library in the source tree
-SJ3SRC = $(CONTRIBSRC)/programs/sj3
-SJ3LIB = $(SJ3SRC)/sj3lib/libsj3lib.a
+XCOMM SJ3SRC = $(CONTRIBSRC)/programs/sj3
+XCOMM SJ3LIB = $(SJ3SRC)/sj3lib/libsj3lib.a
 
 
 /**
diff -Naur kinput2-v3.old/cmd/Kinput2.ad kinput2-v3.new/cmd/Kinput2.ad
--- kinput2-v3.old/cmd/Kinput2.ad	Thu May 27 08:55:18 1999
+++ kinput2-v3.new/cmd/Kinput2.ad	Tue Oct  9 00:45:02 2001
@@ -96,7 +96,7 @@
 ! XimpProtocol
 *XimpProtocol.ForceDefaultServer: true
 ! X Input Method Protocol
-*IMProtocol.locales: ja_JP.SJIS, ja_JP.EUC, ja_JP, japanese, japan, ja
+*IMProtocol.locales: ja_JP.SJIS, ja_JP.EUC, ja_JP, japanese, japan, ja, C
 ! common to XimpProtocol and IMProtocol
 *defaultFontList: -misc-fixed-medium-r-normal--14-*-*-*-c-*
 *ConversionStartKeys: \
diff -Naur kinput2-v3.old/copyright kinput2-v3.new/copyright
--- kinput2-v3.old/copyright	Thu Jan  1 09:00:00 1970
+++ kinput2-v3.new/copyright	Tue Oct  9 00:42:46 2001
@@ -0,0 +1,21 @@
+/*
+ *	kinput2
+ */
+
+/*
+ * Copyright (C) 1991 by Software Research Associates, Inc.
+ *
+ * Permission to use, copy, modify, and distribute this software and its
+ * documentation for any purpose and without fee is hereby granted, provided
+ * that the above copyright notice appear in all copies and that both that
+ * copyright notice and this permission notice appear in supporting
+ * documentation, and that the name of Software Research Associates not be
+ * used in advertising or publicity pertaining to distribution of the
+ * software without specific, written prior permission.  Software Research
+ * Associates makes no representations about the suitability of this software
+ * for any purpose.  It is provided "as is" without express or implied
+ * warranty.
+ *
+ * Author:  Makoto Ishisone, Software Research Associates, Inc., Japan
+ *		ishisone@sra.co.jp
+ */
