diff -ruN mpgtx-1.3.orig/configure mpgtx-1.3/configure
--- mpgtx-1.3.orig/configure	Fri Aug 30 10:03:12 2002
+++ mpgtx-1.3/configure	Wed Feb 26 08:50:53 2003
@@ -7,7 +7,7 @@
 #################################################
 
 prefix="/usr/local"
-manprefix="/usr/local"
+manprefix="@FINKINSTALL@/share"
 #compiler flags
 CC="g++"
 common_cflags="-pedantic -Wall"
@@ -34,29 +34,30 @@
 # some stuff for MacOSX -----------------------------
 HOSTNAME=$(./config.guess)
 
-if [ "${HOSTNAME}" = "powerpc-apple-darwin6.0" ]; then
-#compiler flags
-CC="c++"
-common_cflags="-pedantic -Wall"
-#devel_cflags="-g -Werror -fno-builtin"
-devel_cflags="-g -fno-builtin"
-optimization_cflags="-O0"
-release_cflags=$optimization_cflags
-cflags=$common_cflags
-OFLAGS="-D_MACOSX"
-#linker flags
-lflags="-x"
-#state vars
-debug="no"
-devel="no"
-parachute="no"
-static="no"
-LFS="yes"
-
-#cp flags macosx does not know about cp -d !
-cpflags=""
+case $HOSTNAME in
+  powerpc-apple-darwin*)
+    #compiler flags
+    CC="g++"
+    common_cflags="-pedantic -Wall"
+    #devel_cflags="-g -Werror -fno-builtin"
+    devel_cflags="-g -fno-builtin"
+    optimization_cflags="-O0"
+    release_cflags=$optimization_cflags
+    cflags=$common_cflags
+    OFLAGS="-D_MACOSX"
+    #linker flags
+    lflags="-x"
+    #state vars
+    debug="no"
+    devel="no"
+    parachute="yes"
+    static="no"
+    LFS="yes"
+    #cp flags macosx does not know about cp -d !
+    cpflags=""
+  ;;
+esac
 
-fi
 
 #################################################
 #                                               #
@@ -230,7 +231,7 @@
 echo "cpflags=$cpflags" >> $makefilename
 
 cat << EOF >>$makefilename
-INSTALLDIR=\$(PREFIX)/bin
+INSTALLDIR=\$(prefix)/bin
 MANDIR=\$(manprefix)/man/man1
 # Compilation and linker flags
 
@@ -302,11 +303,11 @@
 	install -d -m 755  \$(INSTALLDIR)
 	install -d -m 755  \$(MANDIR)
 	install -s -m 755 mpgtx \$(INSTALLDIR)
-	cp -\$(cpflags)f mpgdemux mpgjoin mpgcat mpginfo mpgsplit tagmp3 \$(INSTALLDIR)
+	install -m 755  mpgdemux mpgjoin mpgcat mpginfo mpgsplit tagmp3 \$(INSTALLDIR)
 	install -m 644 man/mpgtx.1 man/tagmp3.1 \$(MANDIR) 
 	cd man ; make
 	cd ..
-	cp -\$(cpflags)f ./man/mpgdemux.1 ./man/mpgjoin.1 man/mpgsplit.1 man/mpgcat.1 man/mpginfo.1 \$(MANDIR)
+	install -m 644 ./man/mpgdemux.1 ./man/mpgjoin.1 man/mpgsplit.1 man/mpgcat.1 man/mpginfo.1 \$(MANDIR)
 
 
 uninstall:
