diff -Nurd -x'*~' multitail-2.4.orig/mt.h multitail-2.4/mt.h
--- multitail-2.4.orig/mt.h	2003-05-08 05:09:38.000000000 -0400
+++ multitail-2.4/mt.h	2003-05-08 05:11:22.000000000 -0400
@@ -10,11 +10,7 @@
 			 */
 #define __USE_BSD	/* manpage says _BSD_SOURCE, stdlib.h says __USE_BSD */
 
-#if defined(__FreeBSD__) || defined(linux) || defined(__OpenBSD__) || defined(__NetBSD__)
 #include <ncurses.h>
 #define N_CURSES
-#else
-#include <curses.h>
-#endif
 
 #endif
