diff -ru appconfig-1.52/AppConfig.pm appconfig-1.52-patched/AppConfig.pm 
--- AppConfig-1.52/AppConfig.pm Thu Oct 29 13:08:55 1998
+++ AppConfig-1.52-patched/AppConfig.pm Sun Dec 16 16:17:56 2001
@@ -578,7 +578,7 @@
 Used to provide an argument specification string to pass to Getopt::Long 
 via AppConfig::Getopt.  E.g. "=i", ":s", "=s@".  This can also be used to 
 implicitly set the ARGCOUNT value (C</^!/> = ARGCOUNT_NONE, C</@/> = 
-ARGCOUNT_LIST, C</%/ = ARGCOUNT_HASH, C</[=:].*/> = ARGCOUNT_ONE)
+ARGCOUNT_LIST, C</%/> = ARGCOUNT_HASH, C</[=:].*/> = ARGCOUNT_ONE)
