diff -Naur Exception-Class-1.07/t/basic.t Exception-Class-1.07-patched/t/basic.t
--- Exception-Class-1.07/t/basic.t	Wed Dec 18 18:40:29 2002
+++ Exception-Class-1.07-patched/t/basic.t	Wed Dec 18 18:45:22 2002
@@ -174,7 +174,7 @@
     my $re;
     if ($] == 5.006)
     {
-	$re = qr/overloaded again.+eval {...}\((?:'Exception::Class::Base', )?'error', 'overloaded again'\)/s;
+	$re = qr/overloaded again.+eval {...}\('Exception::Class::Base', 'Exception::Class::Base', 'Exception::Class::Base', 'error', 'overloaded again'\)/s;
     }
     else
     {
