diff -ru GDTextUtil-0.83/t/align.t GDTextUtil-0.83-patched/t/align.t
--- GDTextUtil-0.83/t/align.t	Wed Nov 20 11:21:50 2002
+++ GDTextUtil-0.83-patched/t/align.t	Wed Nov 20 11:21:50 2002
@@ -96,7 +96,8 @@
 printf "ok %d\n", $i++;
 
 # TTF fonts
-if ($t->can_do_ttf)
+if (0)
+#if ($t->can_do_ttf)
 {
     $t = GD::Text::Align->new($gd,
         valign => 'top',
diff -ru GDTextUtil-0.83/t/text.t GDTextUtil-0.83-patched/t/text.t
--- GDTextUtil-0.83/t/text.t	Wed Nov 20 11:21:50 2002
+++ GDTextUtil-0.83-patched/t/text.t	Wed Nov 20 11:21:50 2002
@@ -59,7 +59,8 @@
 print 'not ' unless $rc;
 printf "ok %d\n", $i++;
 
-if ($t->can_do_ttf)
+if (0)
+#if ($t->can_do_ttf)
 {
 	# Test loading of TTF
 	$rc = $t->set_font('cetus.ttf', 18);
diff -ru GDTextUtil-0.83/t/wrap.t GDTextUtil-0.83-patched/t/wrap.t
--- GDTextUtil-0.83/t/wrap.t	Wed Nov 20 11:21:50 2002
+++ GDTextUtil-0.83-patched/t/wrap.t	Wed Nov 20 11:21:50 2002
@@ -69,7 +69,8 @@
 $wp->set(preserve_nl => 0);
 
 # TTF
-if ($wp->can_do_ttf)
+if (0)
+#if ($wp->can_do_ttf)
 {
 	$rc = $wp->set_font('cetus.ttf', 10);
 	print 'not ' unless $rc;
