diff -ruN libwww-perl-5.69.orig/Makefile.PL libwww-perl-5.69/Makefile.PL
--- libwww-perl-5.69.orig/Makefile.PL	Thu Jan 23 23:13:42 2003
+++ libwww-perl-5.69/Makefile.PL	Mon Jan 27 09:05:21 2003
@@ -57,7 +57,7 @@
 
 EOT
 
-        if (prompt("Do you want to enable these tests?", "y") =~ /^y/i) {
+#        if (prompt("Do you want to enable these tests?", "y") =~ /^y/i) {
 	    open(ENABLED, ">t/live/ENABLED") || die "Can't enable: $!";
 	    close(ENABLED);
 
@@ -76,10 +76,10 @@
 		#warn $@;
 		unlink("t/live/ZLIB_OK");
 	    }
-	}
-	else {
-	    unlink("t/live/ENABLED");
-	}
+#	}
+#	else {
+#	    unlink("t/live/ENABLED");
+#	}
     }
 }
 
@@ -95,9 +95,9 @@
 EOT
     my @tmp;
     for (@programs_to_install) {
-	if (prompt("Do you want to install $_?", "y") =~ /^y/) {
+#	if (prompt("Do you want to install $_?", "y") =~ /^y/) {
 	    push(@tmp, $_);
-	}
+#	}
     }
     @programs_to_install = @tmp;
 }
@@ -126,9 +126,9 @@
 	    }
 	    close(PROG);
 	}
-	if (prompt("Do you want to install the $alias alias?", $default) =~ /^y/) {
-	    push(@tmp, $alias);
-	}
+#	if (prompt("Do you want to install the $alias alias?", $default) =~ /^y/) {
+#	    push(@tmp, $alias);
+#	}
     }
     @request_aliases = @tmp;
 }
