diff -ruN libwww-perl-5.64.orig/Makefile.PL libwww-perl-5.64/Makefile.PL
--- libwww-perl-5.64.orig/Makefile.PL	Wed Jan  2 18:51:36 2002
+++ libwww-perl-5.64/Makefile.PL	Sat Mar 23 13:33:27 2002
@@ -9,7 +9,7 @@
 #--- Configuration section ---
 
 my @programs_to_install = qw(lwp-request lwp-mirror lwp-rget lwp-download);
-my @request_aliases     = qw(GET HEAD POST);
+my @request_aliases     = qw();
 
 #--- End Configuration - You should not have to change anything below this line
 
@@ -57,7 +57,7 @@
 
 EOT
 
-        if (prompt("Do you want to enable these tests?", "y") =~ /^y/i) {
+        if ($s) {
 	    open(ENABLED, ">t/live/ENABLED") || die "Can't enable: $!";
 	    close(ENABLED);
 
