diff -Naur Params-Validate-0.66/t/06-options.t Params-Validate-0.66-patched/t/06-options.t
--- Params-Validate-0.66/t/06-options.t	2003-09-30 17:45:41.000000000 +0200
+++ Params-Validate-0.66-patched/t/06-options.t	2003-10-10 09:57:18.000000000 +0200
@@ -10,7 +10,7 @@
 }
 
 use Test;
-plan test => $] == 5.006 ? 4 : 7;
+plan test => $] == 5.006 ? 3 : 7;
 
 Params::Validate::validation_options( stack_skip => 2 );
 
