--- PerlMagick/Makefile.PL.orig	Fri May  3 14:07:50 2002
+++ PerlMagick/Makefile.PL	Fri May  3 14:09:42 2002
@@ -60,13 +60,16 @@
    'DEFINE'	=> '-DHAVE_CONFIG_H',     # e.g., '-DHAVE_SOMETHING' 
 
    # Header search specfication and preprocessor flags
-   'INC'	=> '-I../ -I.. -I/usr/include/freetype2 -D_FILE_OFFSET_BITS=64 -D_REENTRANT -I/usr/X11R6/include -I/usr/include/libxml2',
+#  'INC'	=> '-I../ -I.. -I/usr/include/freetype2 -D_FILE_OFFSET_BITS=64 -D_REENTRANT -I/usr/X11R6/include -I/usr/include/libxml2',
+   'INC'	=> '-I../ -I.. -I@PREFIX@/include/freetype2 -D_FILE_OFFSET_BITS=64 -D_REENTRANT -I/usr/X11R6/include -I@PREFIX@/include/libxml2 -I@PREFIX@/include',
 
    # Install PerlMagick binary into ImageMagick bin directory
-   'INSTALLBIN'	=> '/usr/local/bin',
+#  'INSTALLBIN'	=> '/usr/local/bin',
+   'INSTALLBIN'	=> '@PREFIX@/bin',
 
    # Library specification
-   'LIBS'	=> ['-L/usr/X11R6/lib -L/usr/lib -L/usr/lib -L/home/cristy/ImageMagick/magick -lMagick -llcms -lfreetype -lXext -lSM -lICE -lX11 -lbz2 -lz -lpthread -lm'],
+#  'LIBS'	=> ['-L/usr/X11R6/lib -L/usr/lib -L/usr/lib -L/home/cristy/ImageMagick/magick -lMagick -llcms -lfreetype -lXext -lSM -lICE -lX11 -lbz2 -lz -lpthread -lm'],
+   'LIBS'	=> ['-L/usr/X11R6/lib -L@PREFIX@/lib  -lMagick -llcms -ltiff -lfreetype -ljasper -ljpeg -lpng -lXext -lSM -lICE -lX11 -lbz2 -lxml2 -lz -lpthread -lm -ldpstk'],
 
    # Perl binary name (if a Perl binary is built)
    'MAP_TARGET'	=> 'PerlMagick',
