diff -ruN WWW-Search-2.38.orig/lib/WWW/Search/Test.pm WWW-Search-2.38/lib/WWW/Search/Test.pm
--- WWW-Search-2.38.orig/lib/WWW/Search/Test.pm	Thu Jul 18 15:41:49 2002
+++ WWW-Search-2.38/lib/WWW/Search/Test.pm	Tue Feb 25 14:16:43 2003
@@ -52,14 +52,18 @@
 ($TEST_DUMMY, $TEST_EXACTLY, $TEST_BY_COUNTING, $TEST_GREATER_THAN, $TEST_RANGE) = (1..10);
 
 # At the time this module is loaded, try to find a working WebSearch:
-my @as = split(/\s/, eval{`WebSearch --VERSION`});
-my $websearch = shift @as;
+my ($websearch, @as);
+if (-f './WebSearch') {
 # Try local directory, in case . is not in the path:
-(-f './WebSearch') && (@as = split(/\s/, eval{`./WebSearch --VERSION`}));
+@as = split(/\s/, eval{`./WebSearch --VERSION`});                        
 $websearch ||= shift @as;
 $websearch ||= 'not in the path';
 undef $websearch unless $websearch =~ m/WebSearch/;
 # print STDERR "in WWW::Search::Test, websearch is $websearch\n";
+} else {
+@as = split(/\s/, eval{`WebSearch --VERSION`});
+$websearch = shift @as;
+}
 
 =head2 new
 
