diff -ru fontforge-20060125.orig/Makefile.dynamic.in fontforge-20060125/Makefile.dynamic.in
--- fontforge-20060125.orig/Makefile.dynamic.in	2006-01-04 15:20:59.000000000 -0500
+++ fontforge-20060125/Makefile.dynamic.in	2006-02-22 20:16:24.000000000 -0500
@@ -86,7 +86,7 @@
 	-cp $(srcdir)/fontforge/sfddiff.1 $(mandir)/man1
 #explicit use of sh to make sure we don't invoke csh with different "if" syntax
 	sh -c "if test -d htdocs >/dev/null ; then $(MAKE) install_docs ; fi"
-	sh < systemspecific
+#	sh < systemspecific
 
 uninstall:
 	-rm $(bindir)/fontforge
diff -ru fontforge-20060125.orig/configure fontforge-20060125/configure
--- fontforge-20060125.orig/configure	2005-12-24 19:46:20.000000000 -0500
+++ fontforge-20060125/configure	2006-02-22 20:16:24.000000000 -0500
@@ -2652,9 +2652,7 @@
  gww_define_caps_for_alt="1"
 
 
- CPPFLAGS="$CPPFLAGS -I/sw/include"
- LDFLAGS="$LDFLAGS -L/sw/lib"
- gww_rpath="-rpath /sw/lib"
+ gww_rpath=""
  LIBS="/System/Library/Frameworks/CoreServices.framework/CoreServices $LIBS"
  ;;
 powerpc-*-*linux*)
@@ -8267,7 +8265,7 @@
 echo "$as_me:$LINENO: result: $ac_cv_file__usr_include_libxml2_libxml" >&5
 echo "${ECHO_T}$ac_cv_file__usr_include_libxml2_libxml" >&6
 if test $ac_cv_file__usr_include_libxml2_libxml = yes; then
-  CFLAGS="$CFLAGS -I/usr/include/libxml2/"
+  true
 else
 
  echo "$as_me:$LINENO: checking for /usr/local/include/libxml2/libxml" >&5
@@ -8288,7 +8286,7 @@
 echo "$as_me:$LINENO: result: $ac_cv_file__usr_local_include_libxml2_libxml" >&5
 echo "${ECHO_T}$ac_cv_file__usr_local_include_libxml2_libxml" >&6
 if test $ac_cv_file__usr_local_include_libxml2_libxml = yes; then
-  CFLAGS="$CFLAGS -I/usr/local/include -I/usr/local/include/libxml2/"
+  true
 else
   cat >>confdefs.h <<\_ACEOF
 #define _NO_LIBXML 1
@@ -8300,13 +8298,13 @@
 
 
 
-echo "$as_me:$LINENO: checking for DGifOpenFileName in -lungif" >&5
-echo $ECHO_N "checking for DGifOpenFileName in -lungif... $ECHO_C" >&6
-if test "${ac_cv_lib_ungif_DGifOpenFileName+set}" = set; then
+echo "$as_me:$LINENO: checking for DGifOpenFileName in -lgif" >&5
+echo $ECHO_N "checking for DGifOpenFileName in -lgif... $ECHO_C" >&6
+if test "${ac_cv_lib_gif_DGifOpenFileName+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lungif  $LIBS"
+LIBS="-lgif  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -8351,20 +8349,20 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_cv_lib_ungif_DGifOpenFileName=yes
+  ac_cv_lib_gif_DGifOpenFileName=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-ac_cv_lib_ungif_DGifOpenFileName=no
+ac_cv_lib_gif_DGifOpenFileName=no
 fi
 rm -f conftest.err conftest.$ac_objext \
       conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-echo "$as_me:$LINENO: result: $ac_cv_lib_ungif_DGifOpenFileName" >&5
-echo "${ECHO_T}$ac_cv_lib_ungif_DGifOpenFileName" >&6
-if test $ac_cv_lib_ungif_DGifOpenFileName = yes; then
+echo "$as_me:$LINENO: result: $ac_cv_lib_gif_DGifOpenFileName" >&5
+echo "${ECHO_T}$ac_cv_lib_gif_DGifOpenFileName" >&6
+if test $ac_cv_lib_gif_DGifOpenFileName = yes; then
 
 echo -n checking for ExtensionBlock.Function in gif_lib.h...
 cat >conftest.$ac_ext <<_ACEOF
@@ -8410,7 +8408,7 @@
 sed 's/^/| /' conftest.$ac_ext >&5
 
 echo " " no
-    ac_cv_lib_ungif_DGifOpenFileName=no
+    ac_cv_lib_gif_DGifOpenFileName=no
     cat >>confdefs.h <<\_ACEOF
 #define _NO_LIBUNGIF 1
 _ACEOF
@@ -12048,8 +12046,8 @@
  if test "$gww_needsstatic" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -static"
  fi
- if test "$ac_cv_lib_ungif_DGifOpenFileName" = "yes"; then
-  STATIC_LIBS="$STATIC_LIBS -lungif"
+ if test "$ac_cv_lib_gif_DGifOpenFileName" = "yes"; then
+  STATIC_LIBS="$STATIC_LIBS -lgif"
  fi
  if test "$ac_cv_lib_png_png_create_read_struct" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -lpng -lz"
@@ -12224,8 +12222,8 @@
  if test "$gww_needsstatic" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -static"
  fi
- if test "$ac_cv_lib_ungif_DGifOpenFileName" = "yes"; then
-  STATIC_LIBS="$STATIC_LIBS -lungif"
+ if test "$ac_cv_lib_gif_DGifOpenFileName" = "yes"; then
+  STATIC_LIBS="$STATIC_LIBS -lgif"
   cat >>confdefs.h <<\_ACEOF
 #define _STATIC_LIBUNGIF 1
 _ACEOF
diff -ru fontforge-20060125.orig/fontforge/Makefile.dynamic.in fontforge-20060125/fontforge/Makefile.dynamic.in
--- fontforge-20060125.orig/fontforge/Makefile.dynamic.in	2006-01-15 09:26:05.000000000 -0500
+++ fontforge-20060125/fontforge/Makefile.dynamic.in	2006-02-22 20:16:24.000000000 -0500
@@ -50,11 +50,11 @@
     @CPPFLAGS@ '-DPREFIX="@prefix@"'
 CFLAGS = @CFLAGS@ $(_CFLAGS)
 
-LIBS = -rpath $(libdir) $(X_LIBS) ../libgdraw.la ../libgunicode.la \
-    $(X_PRE_LIBS) $(X_11LIB) $(X_EXTRA_LIBS) @LIBS@ $(STATIC_LIBS) -L/usr/lib -lm
-DLIBS = -rpath $(libdir) ../libgunicode.la @LIBS@ $(STATIC_LIBS) -lm
+LIBS = -rpath $(libdir) $(STATIC_LIBS) $(X_LIBS) ../libgdraw.la ../libgunicode.la \
+    $(X_PRE_LIBS) $(X_11LIB) $(X_EXTRA_LIBS) @LIBS@ -L/usr/lib -lm
+DLIBS = -rpath $(libdir) -L$(libdir) -liconv ../libgunicode.la @LIBS@ $(STATIC_LIBS) -lm
 
-all: fontforge
+all: fontforge sfddiff
 
 fontforge: main.o $(fontforge_LIBOBJECTS)
 	$(LIBTOOL) --mode=link $(CC) -export-dynamic -o fontforge main.o $(fontforge_LIBOBJECTS) $(LIBS)
diff -ru fontforge-20060125.orig/fontforge/start.c fontforge-20060125/fontforge/start.c
--- fontforge-20060125.orig/fontforge/start.c	2006-01-16 10:10:22.000000000 -0500
+++ fontforge-20060125/fontforge/start.c	2006-02-22 20:17:47.000000000 -0500
@@ -202,7 +202,6 @@
     /*  otherwise leave alone. On the mac people often use fink to */
     /*  install image libs. For some reason fink installs in a place */
     /*  the dynamic loader doesn't find */
-    setenv("DYLD_LIBRARY_PATH","/sw/lib",0);
 #endif
 }
 
