--- Raster3D_2.7b/label3d	2003-06-20 07:52:12.000000000 +1000
+++ Raster3D_2.7b-patched/label3d	2003-06-27 15:29:48.000000000 +1000
@@ -39,7 +39,7 @@
 if [ "$TMPDIR" ]; then
     tmp=$TMPDIR/$$
 else
-    tmp=/usr/tmp/$$
+    tmp=/tmp/$$
 fi
 #echo "DEBUG scratch files to $tmp"  1>&2
 

--- Raster3D_2.7b/Makefile	2003-06-20 07:52:12.000000000 +1000
+++ Raster3D_2.7b-patched/Makefile	2003-06-27 15:31:42.000000000 +1000
@@ -50,9 +50,9 @@
 	@cp Makefile.template Makefile.incl
 	@echo OS = linux                      >> Makefile.incl
 	@echo CC = gcc                        >> Makefile.incl
-	@echo CFLAGS = -g -m486 -w            >> Makefile.incl
+	@echo CFLAGS = -g -dynamic            >> Makefile.incl
 	@echo FC = g77                        >> Makefile.incl
-	@echo FFLAGS = -g -O -w -malign-double>> Makefile.incl
+	@echo FFLAGS = -g -O -w               >> Makefile.incl
 	@echo RM = /bin/rm -f                 >> Makefile.incl
 	@echo OSDEFS =  -DLINUX -DNETWORKBYTEORDER       >> Makefile.incl
 	@echo include Makefile.package        >> Makefile.incl

--- Raster3D_2.7b/Makefile.template	2003-06-20 07:52:12.000000000 +1000
+++ Raster3D_2.7b-patched/Makefile.template	2003-06-27 15:28:24.000000000 +1000
@@ -8,15 +8,15 @@
 #
 # Where to install the program, data files, and manuals
 #
-prefix  = /usr/local
+prefix  = $(prefix)
 bindir  = $(prefix)/bin
 datadir = $(prefix)/share/Raster3D/materials
-mandir  = $(prefix)/man/manl
+mandir  = $(prefix)/share/man/manl
 htmldir = $(prefix)/share/Raster3D/html
 examdir = $(prefix)/share/Raster3D/examples
 
-INCDIRS  =	-I/usr/include -I/usr/local/include
-LIBDIRS  =	-L/usr/local/lib
+INCDIRS  =	-I/usr/include -I@FINKPREFIX@/include
+LIBDIRS  =	-L@FINKPREFIX@/lib -lcc_dynamic
 
 #
 # TIFF support (enabled by default)

--- Raster3D_2.7b/stereo3d	2003-06-20 07:52:13.000000000 +1000
+++ Raster3D_2.7b-patched/stereo3d	2003-06-27 15:30:09.000000000 +1000
@@ -42,7 +42,7 @@
 if [ "$TMPDIR" ]; then
     tmp=$TMPDIR/$$
 else
-    tmp=/usr/tmp/$$
+    tmp=/tmp/$$
 fi
 # echo "DEBUG scratch files to $tmp"  1>&2
 
