diff -Nurd -x'*~' Raster3D_2.7d.orig/Makefile Raster3D_2.7d/Makefile
--- Raster3D_2.7d.orig/Makefile	2006-04-12 14:46:10.000000000 -0400
+++ Raster3D_2.7d/Makefile	2006-06-26 14:33:01.000000000 -0400
@@ -50,9 +50,9 @@
 	@cp Makefile.template Makefile.incl
 	@echo OS = linux                      >> Makefile.incl
 	@echo CC = gcc                        >> Makefile.incl
-	@echo CFLAGS = -g -w                  >> Makefile.incl
+	@echo CFLAGS = -g -dynamic            >> Makefile.incl
 	@echo FC = g77                        >> Makefile.incl
-	@echo FFLAGS = -g -O -w -malign-double>> Makefile.incl
+	@echo FFLAGS = -g -O -w               >> Makefile.incl
 	@echo RM = /bin/rm -f                 >> Makefile.incl
 	@echo OSDEFS =  -DLINUX -DNETWORKBYTEORDER       >> Makefile.incl
 	@echo include Makefile.package        >> Makefile.incl
diff -Nurd -x'*~' Raster3D_2.7d.orig/Makefile.template Raster3D_2.7d/Makefile.template
--- Raster3D_2.7d.orig/Makefile.template	2003-10-20 18:50:11.000000000 -0400
+++ Raster3D_2.7d/Makefile.template	2006-06-26 14:39:18.000000000 -0400
@@ -11,12 +11,12 @@
 prefix  = /usr/local
 bindir  = $(prefix)/bin
 datadir = $(prefix)/share/Raster3D/materials
-mandir  = $(prefix)/man/manl
+mandir  = $(prefix)/share/man/manl
 htmldir = $(prefix)/share/Raster3D/html
 examdir = $(prefix)/share/Raster3D/examples
 
-INCDIRS  =	-I/usr/include -I/usr/local/include
-LIBDIRS  =	-L/usr/local/lib
+INCDIRS  =	-I$(prefix)/include
+LIBDIRS  =	-L$(prefix)/lib
 
 #
 # TIFF support (enabled by default)
