diff -ru terraform-0.9.0/configure terraform-0.9.0-patched/configure
--- terraform-0.9.0/configure	2002-04-24 17:34:16.000000000 +0200
+++ terraform-0.9.0-patched/configure	2002-06-02 19:36:06.000000000 +0200
@@ -4665,26 +4665,9 @@
   cat > conftest.$ac_ext <<EOF
 #line 4667 "configure"
 #include "confdefs.h"
-/* System header to define __stub macros and hopefully few prototypes,
-    which can conflict with char vfscanf(); below.  */
-#include <assert.h>
-/* Override any gcc2 internal prototype to avoid an error.  */
-/* We use char because int might match the return type of a gcc2
-    builtin and then its argument prototype would still apply.  */
-char vfscanf();
 
 int main() {
-
-/* The GNU C library defines this for functions which it implements
-    to always fail with ENOSYS.  Some functions are actually named
-    something starting with __ and the normal name is an alias.  */
-#if defined (__stub_vfscanf) || defined (__stub___vfscanf)
-choke me
-#else
-vfscanf();
-#endif
-
-; return 0; }
+return 0; }
 EOF
 if { (eval echo configure:4690: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   rm -rf conftest*
diff -ru terraform-0.9.0/docs/Makefile.in terraform-0.9.0-patched/docs/Makefile.in
--- terraform-0.9.0/docs/Makefile.in	2002-04-24 18:16:48.000000000 +0200
+++ terraform-0.9.0-patched/docs/Makefile.in	2002-06-02 19:36:19.000000000 +0200
@@ -123,7 +123,7 @@
 cflags_set = @cflags_set@
 cxxflags_set = @cxxflags_set@
 
-datadir = @prefix@/doc/terraform
+datadir = @prefix@/share/doc/terraform
 
 EXTRA_DIST = FAQ.sgml 		i18n.txt 		README.sgml 		UsersGuide.sgml
 
diff -ru terraform-0.9.0/src/xmlsupport.c terraform-0.9.0-patched/src/xmlsupport.c
--- terraform-0.9.0/src/xmlsupport.c	2002-04-24 17:27:11.000000000 +0200
+++ terraform-0.9.0-patched/src/xmlsupport.c	2002-06-02 19:36:06.000000000 +0200
@@ -17,8 +17,8 @@
  */
 
 #include <string.h>
+#include <stdio.h>
 
-#include <malloc.h>
 #include "xmlsupport.h"
 #include "clocale.h"
 
