diff -Nurd -x'*~' XaoS-3.2.orig/Makefile.in XaoS-3.2/Makefile.in
--- XaoS-3.2.orig/Makefile.in	2005-02-11 04:37:47.000000000 -0500
+++ XaoS-3.2/Makefile.in	2006-04-26 21:24:38.000000000 -0400
@@ -10,15 +10,13 @@
 
 @SET_MAKE@
 
-default clean realclean echo help depend xaos:
-	@cd src ; $(MAKE) $@
-
-all:
-	@cd src ; $(MAKE) all
+default clean realclean echo help depend xaos all:
+	$(MAKE) -C src $@
+	@echo "done in $@"
 
 distclean:
 	#@$(MAKE) -C src $@
-	@$(MAKE) -C src distclean
+	$(MAKE) -C src distclean
 	rm -f libtool config.cache config.status config.log src/include/config.h src/include/aconfig.h
 	rm -f src/Makefile.common src/arch/linux/Makefile.main
 	rm -f `find . -name Makefile`
@@ -36,6 +34,7 @@
 	@INSTALL@ -d $(DESTDIR)$(datadir)/XaoS/help
 	@INSTALL@ -d $(DESTDIR)$(bindir)
 	@INSTALL@ -d $(DESTDIR)$(mandir)/man6
+	@INSTALL@ -d $(DESTDIR)$(infodir)
 	@INSTALL@ -s bin/xaos $(DESTDIR)$(bindir)
 	@INSTALL@ -m 444 tutorial/*.x[ah]f $(DESTDIR)$(datadir)/XaoS/tutorial
 
diff -Nurd -x'*~' XaoS-3.2.orig/configure XaoS-3.2/configure
--- XaoS-3.2.orig/configure	2006-01-22 17:48:38.000000000 -0500
+++ XaoS-3.2/configure	2006-04-26 22:45:05.000000000 -0400
@@ -18892,7 +18892,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lggi $LIBS  $LIBS"
+LIBS="-lggi -L/usr/X11R6/lib $LIBS  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
diff -Nurd -x'*~' XaoS-3.2.orig/src/Makefile.in XaoS-3.2/src/Makefile.in
--- XaoS-3.2.orig/src/Makefile.in	2003-02-10 17:42:28.000000000 -0500
+++ XaoS-3.2/src/Makefile.in	2006-04-26 20:36:51.000000000 -0400
@@ -1,6 +1,6 @@
 CC     = @CC@
 CFLAGS = @CFLAGS@
-LIBS   = @LIBS@ 
+LIBS   = @LIBS@ @INTLLIBS@
 LFLAGS = @LDFLAGS@
 AR     = @AR@
 RANLIB = @RANLIB@
@@ -31,7 +31,7 @@
 	mkdir $@
 
 libs:
-	@for name in $(DIRS) ; do sh -c "cd $$name ; $(MAKE) all"; done
+	@for name in $(DIRS) ; do echo "$(MAKE) -C $$name all"; $(MAKE) -C $$name all || exit 1; echo "done in $$name"; done
 
 $(PROG): @BINPATH@ @LIBPATH@ include/config.h libs
 	$(CC) $(CFLAGS) $(LFLAGS) -o $@ $(TLIBS) $(LIBS)
diff -Nurd -x'*~' XaoS-3.2.orig/src/engine/3d.c XaoS-3.2/src/engine/3d.c
--- XaoS-3.2.orig/src/engine/3d.c	2003-02-03 11:16:33.000000000 -0500
+++ XaoS-3.2/src/engine/3d.c	2006-04-26 17:46:36.000000000 -0400
@@ -3,6 +3,7 @@
 #ifndef NO_MALLOC_H
 #include <malloc.h>
 #endif
+#include <stdlib.h>
 #include <stdio.h>		/*for NULL */
 #include <math.h>
 #else
@@ -117,7 +118,7 @@
 getinstance (CONST struct filteraction *a)
 {
   struct filter *f = createfilter (a);
-  struct threeddata *d = calloc (sizeof (*d), 1);
+  struct threeddata *d = calloc (1, sizeof (*d));
   f->data = d;
   f->name = "3d";
   return (f);
diff -Nurd -x'*~' XaoS-3.2.orig/src/engine/btrace.c XaoS-3.2/src/engine/btrace.c
--- XaoS-3.2.orig/src/engine/btrace.c	2004-11-24 02:11:43.000000000 -0500
+++ XaoS-3.2/src/engine/btrace.c	2006-04-26 17:47:02.000000000 -0400
@@ -3,6 +3,7 @@
 #ifndef NO_MALLOC_H
 #include <malloc.h>
 #endif
+#include <stdlib.h>
 #include <math.h>
 #include <stdio.h>
 #include <string.h>
diff -Nurd -x'*~' XaoS-3.2.orig/src/engine/fractal.c XaoS-3.2/src/engine/fractal.c
--- XaoS-3.2.orig/src/engine/fractal.c	2003-02-03 11:16:37.000000000 -0500
+++ XaoS-3.2/src/engine/fractal.c	2006-04-26 17:44:54.000000000 -0400
@@ -65,6 +65,7 @@
 #define M_PI 3.1415
 #endif
 #include <xerror.h>
+#include <stdlib.h>
 
 struct symetryinfo2 cursymetry;
 struct palette cpalette;
diff -Nurd -x'*~' XaoS-3.2.orig/src/include/gccaccel.h XaoS-3.2/src/include/gccaccel.h
--- XaoS-3.2.orig/src/include/gccaccel.h	2002-10-02 07:30:21.000000000 -0400
+++ XaoS-3.2/src/include/gccaccel.h	2006-04-26 22:56:52.000000000 -0400
@@ -36,3 +36,6 @@
 #ifndef myfabs
 #define myfabs(x) ((x)>0?(x):-(x))
 #endif
+#ifndef CONST
+#define CONST const
+#endif
diff -Nurd -x'*~' XaoS-3.2.orig/src/include/xerror.h XaoS-3.2/src/include/xerror.h
--- XaoS-3.2.orig/src/include/xerror.h	2002-09-25 11:52:09.000000000 -0400
+++ XaoS-3.2/src/include/xerror.h	2006-04-26 22:58:23.000000000 -0400
@@ -4,6 +4,9 @@
 extern "C"
 {
 #endif
+
+#include "gccaccel.h"
+
   void x_message (CONST char *text, ...);
   void x_error (CONST char *text, ...);
   void x_fatalerror (CONST char *text, ...) NORETURN;
diff -Nurd -x'*~' XaoS-3.2.orig/src/ui/Makefile.in XaoS-3.2/src/ui/Makefile.in
--- XaoS-3.2.orig/src/ui/Makefile.in	2001-02-06 08:00:40.000000000 -0500
+++ XaoS-3.2/src/ui/Makefile.in	2006-04-26 21:45:18.000000000 -0400
@@ -1,5 +1,5 @@
 CC     = @CC@
-CFLAGS = @CFLAGS@
+CFLAGS = @CFLAGS@ -DLOCALEDIR=\"@LOCALEDIR@\"
 LIBS   = @LIBS@ -lm
 LFLAGS = @LDFLAGS@
 AR     = @AR@
diff -Nurd -x'*~' XaoS-3.2.orig/src/ui/ui-drv/dga/ui_dga.c XaoS-3.2/src/ui/ui-drv/dga/ui_dga.c
--- XaoS-3.2.orig/src/ui/ui-drv/dga/ui_dga.c	2002-09-25 11:52:09.000000000 -0400
+++ XaoS-3.2/src/ui/ui-drv/dga/ui_dga.c	2006-04-26 17:26:31.000000000 -0400
@@ -601,7 +601,7 @@
       XCloseDisplay (display);
       return 0;
     }
-  names = calloc (sizeof (*names), xf86ctx.modecount + 1);
+  names = calloc (xf86ctx.modecount + 1, sizeof (*names));
   for (i = 0; i < xf86ctx.modecount; i++)
     {
       char c[256];
diff -Nurd -x'*~' XaoS-3.2.orig/src/ui/ui-drv/ggi/ui_ggi.c XaoS-3.2/src/ui/ui-drv/ggi/ui_ggi.c
--- XaoS-3.2.orig/src/ui/ui-drv/ggi/ui_ggi.c	2002-09-25 11:52:09.000000000 -0400
+++ XaoS-3.2/src/ui/ui-drv/ggi/ui_ggi.c	2006-04-26 23:05:52.000000000 -0400
@@ -333,8 +333,10 @@
 	    case GIIK_Delete:
 #endif
 #ifdef GIIUC_Delete
+#if GIIUC_Delete != GIIK_Delete
 	    case GIIUC_Delete:
 #endif
+#endif
 	    case GIIUC_BackSpace:
 	      ui_key (UIKEY_BACKSPACE);
 	      break;
@@ -397,7 +399,7 @@
   "32",
   NULL
 };
-static CONST menudialog uih_resizedialog[] = {
+static menudialog uih_resizedialog[] = {
   DIALOGINT ("X:", 0),
   DIALOGINT ("Y:", 0),
   DIALOGCHOICE ("Depth", names, 0),
diff -Nurd -x'*~' XaoS-3.2.orig/src/ui/ui-drv/x11/xlib.c XaoS-3.2/src/ui/ui-drv/x11/xlib.c
--- XaoS-3.2.orig/src/ui/ui-drv/x11/xlib.c	2004-11-24 02:11:49.000000000 -0500
+++ XaoS-3.2/src/ui/ui-drv/x11/xlib.c	2006-04-26 22:17:17.000000000 -0400
@@ -31,6 +31,7 @@
 #ifndef NO_MALLOC_H
 #include <malloc.h>
 #endif
+#include <sys/param.h>
 #include "xlib.h"
 #ifdef AMIGA
 #define XFlush(x) while(0)
@@ -286,7 +287,6 @@
   XDestroyImage (d->image[1]);
 }
 
-#define MAX(x,y) ((x)>(y)?(x):(y))
 
 
 xdisplay *
@@ -302,7 +302,7 @@
   XWMHints *hints;
   char **faked_argv;
 
-  new = (xdisplay *) calloc (sizeof (xdisplay), 1);
+  new = (xdisplay *) calloc (1, sizeof (xdisplay));
   chkalloc (new);
   new->display = XOpenDisplay (params->display);
   if (!new->display)
diff -Nurd -x'*~' XaoS-3.2.orig/src/ui/ui.c XaoS-3.2/src/ui/ui.c
--- XaoS-3.2.orig/src/ui/ui.c	2004-11-24 02:11:46.000000000 -0500
+++ XaoS-3.2/src/ui/ui.c	2006-04-26 21:46:57.000000000 -0400
@@ -1201,7 +1201,7 @@
     )
 #endif
 #else
-			  "/usr/share/locale")
+			  LOCALEDIR)
 #ifdef DEBUG
     )
 #endif
diff -Nurd -x'*~' XaoS-3.2.orig/src/ui-hlp/save.c XaoS-3.2/src/ui-hlp/save.c
--- XaoS-3.2.orig/src/ui-hlp/save.c	2004-11-24 02:11:50.000000000 -0500
+++ XaoS-3.2/src/ui-hlp/save.c	2006-04-26 17:48:43.000000000 -0400
@@ -6,6 +6,7 @@
 #ifndef NO_MALLOC_H
 #include <malloc.h>
 #endif
+#include <stdlib.h>
 #else
 #include <u.h>
 #include <libc.h>
diff -Nurd -x'*~' XaoS-3.2.orig/src/ui-hlp/ui_helper.c XaoS-3.2/src/ui-hlp/ui_helper.c
--- XaoS-3.2.orig/src/ui-hlp/ui_helper.c	2003-02-03 11:16:44.000000000 -0500
+++ XaoS-3.2/src/ui-hlp/ui_helper.c	2006-04-26 17:26:56.000000000 -0400
@@ -2193,7 +2193,7 @@
 	       void (*upd) (struct uih_context *, CONST char *))
 {
   uih_context *uih;
-  uih = (uih_context *) calloc (sizeof (*uih), 1);	/*setup parameters */
+  uih = (uih_context *) calloc (1, sizeof (*uih));	/*setup parameters */
   uih->updatemenus = upd;
   uih->autopilot = 0;
   uih->flags = flags;
diff -Nurd -x'*~' XaoS-3.2.orig/src/util/catalog.c XaoS-3.2/src/util/catalog.c
--- XaoS-3.2.orig/src/util/catalog.c	2003-02-03 11:16:46.000000000 -0500
+++ XaoS-3.2/src/util/catalog.c	2006-04-26 17:49:06.000000000 -0400
@@ -4,6 +4,7 @@
 #ifndef NO_MALLOC_H
 #include <malloc.h>
 #endif
+#include <stdlib.h>
 #include <string.h>
 #else
 #include <u.h>
diff -Nurd -x'*~' XaoS-3.2.orig/src/util/thread.c XaoS-3.2/src/util/thread.c
--- XaoS-3.2.orig/src/util/thread.c	2004-11-24 02:11:51.000000000 -0500
+++ XaoS-3.2/src/util/thread.c	2006-04-26 17:49:34.000000000 -0400
@@ -1,6 +1,7 @@
 #ifndef _plan9_
 #include <signal.h>
 #include <stdio.h>
+#include <stdlib.h>
 #endif
 #include <xthread.h>
 #ifndef __BEOS__
diff -Nurd -x'*~' XaoS-3.2.orig/src/util/timers.c XaoS-3.2/src/util/timers.c
--- XaoS-3.2.orig/src/util/timers.c	2004-11-24 02:11:51.000000000 -0500
+++ XaoS-3.2/src/util/timers.c	2006-04-26 17:50:08.000000000 -0400
@@ -49,6 +49,7 @@
 #ifndef NO_MALLOC_H
 #include <malloc.h>
 #endif
+#include <stdlib.h>
 #include <stdio.h>
 #include <unistd.h>
 #ifndef _MAC
