diff -Nur AnnoCPAN-Perldoc-SyncDB-0.11-orig/cron.weekly AnnoCPAN-Perldoc-SyncDB-0.11/cron.weekly
--- AnnoCPAN-Perldoc-SyncDB-0.11-orig/cron.weekly	1969-12-31 18:00:00.000000000 -0600
+++ AnnoCPAN-Perldoc-SyncDB-0.11/cron.weekly	2005-10-12 13:17:18.000000000 -0500
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+. '@PREFIX@/bin/init.sh'
+test -x '@PREFIX@/bin/syncannopod' && '@PREFIX@/bin/syncannopod' --dest '@PREFIX@/var/annocpan/annopod.db'
diff -Nur AnnoCPAN-Perldoc-SyncDB-0.11-orig/lib/AnnoCPAN/Perldoc/SyncDB.pm AnnoCPAN-Perldoc-SyncDB-0.11/lib/AnnoCPAN/Perldoc/SyncDB.pm
--- AnnoCPAN-Perldoc-SyncDB-0.11-orig/lib/AnnoCPAN/Perldoc/SyncDB.pm	2005-09-27 22:11:25.000000000 -0500
+++ AnnoCPAN-Perldoc-SyncDB-0.11/lib/AnnoCPAN/Perldoc/SyncDB.pm	2005-10-12 13:15:58.000000000 -0500
@@ -137,7 +137,7 @@
       # This algorithm is duplicated from AnnoCPAN::Perldoc
       # Future versions should access that module's algorithm directly
       DIR: foreach my $dir (@ENV{qw(HOME USERPROFILE ALLUSERSPROFILE)},
-                            '/var/annocpan')
+                            '@PREFIX@/var/annocpan', '/var/annocpan')
       {
          if ($dir && -d $dir)
          {
