diff -Naur Bio-Emboss-5.0.0.1/Makefile.PL Bio-Emboss-5.0.0.1-patched/Makefile.PL
--- Bio-Emboss-5.0.0.1/Makefile.PL	2007-08-08 10:52:08.000000000 -0400
+++ Bio-Emboss-5.0.0.1-patched/Makefile.PL	2007-09-20 18:23:42.000000000 -0400
@@ -11,7 +11,7 @@
 #        (if you did no "make install" of EMBOSS,
 #         put here the path to the source code)
 
-$EMB_ROOT     = "";
+$EMB_ROOT     = "@PREFIX@";
 #$EMB_ROOT     = "/usr/local";
 
 
@@ -19,7 +19,7 @@
 #     XLIB and LIBS from the EMBOSS Makefile 
 #     (<emboss-build-dir>/emboss/Makefile)
 
-$EMB_EXT_LIBS = "";
+$EMB_EXT_LIBS = "@PREFIX@/lib";
 #$EMB_EXT_LIBS = "-L/usr/X11R6/lib -lX11 -lm  -lgd -lpng -lz";
 #$EMB_EXT_LIBS = "-L/usr/openwin/lib -lX11 -lsocket -lm -lnsl -lgd -lpng -lz";
 
@@ -89,7 +89,8 @@
 }
 
 if (-d "$EMB_ROOT/include") {
-    $EMB_INC = "-I$EMB_ROOT/include";
+#    $EMB_INC = "-I$EMB_ROOT/include";
+    $EMB_INC = "-I@PREFIX@/include/EMBOSS -I@PREFIX@/include/eplplot";
 } else {
     $EMB_INC =  join " ", map { "-I$EMB_ROOT/$_" } grep { -d "$EMB_ROOT/$_"} @emb_lib_dirs;
 }
@@ -101,7 +102,7 @@
 # --- look for EMBOSS in $EMB_ROOT and $EMB_SRC_ROOT
 #     (disable if you change $EMB_LIBS or $EMB_INC manually)
 # ---- check function needs update
-$check_installation = 0;
+#$check_installation = 0;
 
 
 # --- trying to get the EMBOSS version from the application 'embossversion'
