diff -ru Crypt-SSLeay-0.51/Makefile.PL Crypt-SSLeay-0.51-patched/Makefile.PL
--- Crypt-SSLeay-0.51/Makefile.PL	2003-04-22 07:38:06.000000000 +0200
+++ Crypt-SSLeay-0.51-patched/Makefile.PL	2006-10-09 10:07:29.000000000 +0200
@@ -8,12 +8,12 @@
 
 &GetOptions("default", \$opt_default);
 
-$opt_default ||= $ENV{CRYPT_SSLEAY_DEFAULT};
+$opt_default = 1;
 my $currdir = dirname($0) || '.';
 chdir($currdir) || die("can't change to $currdir: $!");
 
 # FIND POSSIBLE SSL INSTALLATIONS
-my @POSSIBLE_SSL_DIRS = qw(/usr/local/openssl /usr/local/ssl /local/ssl /opt/ssl /usr/local /local /usr);
+my @POSSIBLE_SSL_DIRS = qw(@PREFIX@/lib/system-openssl);
 if($^O eq 'MSWin32') {
     unshift(@POSSIBLE_SSL_DIRS, 'c:/openssl');
 }
