diff -Naur DBIx-SearchBuilder-1.36-orig/Makefile.PL DBIx-SearchBuilder-1.36/Makefile.PL
--- DBIx-SearchBuilder-1.36-orig/Makefile.PL	2005-12-14 18:42:45.000000000 +0100
+++ DBIx-SearchBuilder-1.36/Makefile.PL	2005-12-14 18:43:03.000000000 +0100
@@ -9,7 +9,6 @@
 requires('Class::ReturnValue', 0.40);
 requires('Cache::Simple::TimedExpiry' => '0.21');
 requires('Clone');
-build_requires('Test::More' => 0.52);
 build_requires('DBD::SQLite');
 
 features(
