diff -ru Image-Xbm-1.08/Xbm.pm Image-Xbm-1.08-patched/Xbm.pm
--- Image-Xbm-1.08/Xbm.pm	2000-11-09 19:05:51.000000000 +0000
+++ Image-Xbm-1.08-patched/Xbm.pm	2004-06-08 01:27:20.000000000 +0100
@@ -140,7 +140,7 @@
     }
 
     my $file = $self->get( '-file' ) ;
-    $self->load if defined $file and -r $file and not $self->{'-bits'} ;
+    $self->load if defined $file and not $self->{'-bits'} ;
 
     croak "new() `$file' not found or unreadable" 
     if defined $file and not defined $self->get( '-width' ) ;
