diff -Nurd Inline-0.46/C/C.pm Inline-0.46-new/C/C.pm
--- Inline-0.46/C/C.pm	2010-02-11 05:36:12.000000000 -0500
+++ Inline-0.46-new/C/C.pm	2010-06-01 17:04:47.000000000 -0400
@@ -704,10 +704,7 @@
 
 sub makefile_pl {
     my ($o) = @_;
-    my $perl;
-    -f ($perl = $Config::Config{perlpath})
-      or ($perl = $^X)
-      or croak "Can't locate your perl binary";
+    my $perl = sprintf "perl%vd", $^V;
     $o->system_call("$perl Makefile.PL", 'out.Makefile_PL');
     $o->fix_make;
 }
diff -Nurd Inline-0.46/C/Makefile.PL Inline-0.46-new/C/Makefile.PL
--- Inline-0.46/C/Makefile.PL	2010-01-27 06:03:44.000000000 -0500
+++ Inline-0.46-new/C/Makefile.PL	2010-06-01 17:06:18.000000000 -0400
@@ -71,12 +71,14 @@
 # '
 }
 
-my $answer = '';
+my $answer = 'yes';
+if (0) {
 my $default = $found ? "y" : "n";
 while (1) {
     $answer = prompt ('Do you want to install Inline::C?', $default);
     last if $answer =~ /^(y|yes|n|no)$/i;
 }
+}
 
 if ($answer =~ /^(y|yes)$/i) {
     WriteMakefile(
diff -Nurd Inline-0.46/Inline.pm Inline-0.46-new/Inline.pm
--- Inline-0.46/Inline.pm	2010-02-11 05:33:48.000000000 -0500
+++ Inline-0.46-new/Inline.pm	2010-06-01 17:07:06.000000000 -0400
@@ -757,8 +757,7 @@
     # That way the queried modules don't stay loaded.
     if (defined $o) {
 	($dir) = $dir =~ /(.*)/s if UNTAINT;
-	my $perl = $Config{perlpath};
-        $perl = $^X unless -f $perl;
+	my $perl = sprintf "perl%vd", $^V;
 	($perl) = $perl =~ /(.*)/s if UNTAINT;
 	local $ENV{PERL5LIB} if defined $ENV{PERL5LIB};
 	local $ENV{PERL5OPT} if defined $ENV{PERL5OPT};
