
--- Inline-0.44/C/C.pm	2002-10-18 15:05:48.000000000 -0400
+++ Inline-0.44-new/C/C.pm	2006-08-31 12:26:01.000000000 -0400
@@ -695,10 +695,7 @@
 
 sub makefile_pl {
     my ($o) = @_;
-    my $perl;
-    -f ($perl = $Config::Config{perlpath})
-      or ($perl = $^X)
-      or croak "Can't locate your perl binary";
+    my $perl = sprintf "perl%vd", $^V;
     $o->system_call("$perl Makefile.PL", 'out.Makefile_PL');
     $o->fix_make;
 }
--- Inline-0.44/C/Makefile.PL	2002-10-28 13:26:08.000000000 -0500
+++ Inline-0.44-new/C/Makefile.PL	2006-08-31 12:27:11.000000000 -0400
@@ -34,7 +34,7 @@
 
 if ($found) {
     print <<END;
-I have located this compiler on your system:
+I have located this compiler on your system.
 
 END
 }
@@ -55,12 +55,7 @@
 # '
 }
 
-my $answer = '';
-my $default = $found ? "y" : "n";
-while (1) {
-    $answer = prompt ('Do you want to install Inline::C?', $default); 
-    last if $answer =~ /^(y|yes|n|no)$/i;
-}
+my $answer = 'yes';
 
 if ($answer =~ /^(y|yes)$/i) {
     WriteMakefile(
--- Inline-0.44/Inline.pm	2002-10-28 02:25:19.000000000 -0500
+++ Inline-0.44-new/Inline.pm	2006-08-31 12:27:52.000000000 -0400
@@ -730,7 +730,7 @@
     # That way the queried modules don't stay loaded.
     if (defined $o) {
 	($dir) = $dir =~ /(.*)/s if UNTAINT;
-	my $perl = $Config{perlpath};
+	my $perl = sprintf "perl%vd", $^V;
         $perl = $^X unless -f $perl;
 	($perl) = $perl =~ /(.*)/s if UNTAINT;
 	local $ENV{PERL5LIB} if defined $ENV{PERL5LIB};
