diff -Nurd libintl-perl-1.20.orig/gettext_xs/Makefile.PL libintl-perl-1.20/gettext_xs/Makefile.PL
--- libintl-perl-1.20.orig/gettext_xs/Makefile.PL	2009-06-23 03:04:48.000000000 -0400
+++ libintl-perl-1.20/gettext_xs/Makefile.PL	2010-06-16 04:26:23.000000000 -0400
@@ -38,7 +38,7 @@
     ),
     PREREQ_PM => { File::Spec => 0 },
 	PL_FILES        => {},
-	LIBS => ['-lintl -liconv'],
+	LIBS => ['-LFINKPATH/lib -lintl -liconv'],
 );
 
 __END__
