diff -Nurd -x'*~' local-lib-1.006005.orig/Makefile.PL local-lib-1.006005/Makefile.PL
--- local-lib-1.006005.orig/Makefile.PL	2010-06-08 19:01:42.000000000 -0400
+++ local-lib-1.006005/Makefile.PL	2010-08-01 03:20:30.000000000 -0400
@@ -192,7 +192,7 @@
 
 # and make sure that the user doesn't have any existing CPAN config that'll
 # cause us problems for the next few steps.
-{
+if (0) {
     local $@;
     eval { require CPAN::HandleConfig; };
     # Need newish CPAN.pm for this, ergo skip it if that version of CPAN isn't
@@ -232,13 +232,11 @@
 }
 
 if ($bootstrapping) {
-  auto_install_now;
   postamble <<"END";
 PERL     += $bootstrapping_args
 FULLPERL += $bootstrapping_args
 END
 } else {
-  auto_install;
 }
 chdir($cwd);
 WriteAll;
