diff -Nurd Log-Log4perl-1.01.orig/t/048lwp.t Log-Log4perl-1.01/t/048lwp.t
--- Log-Log4perl-1.01.orig/t/048lwp.t	2004-08-17 02:08:24.000000000 -0400
+++ Log-Log4perl-1.01/t/048lwp.t	2010-05-25 01:32:35.000000000 -0400
@@ -11,11 +11,12 @@
 BEGIN {
     eval {
         require LWP::UserAgent;
-        die "Skip tests" if $LWP::UserAgent::VERSION < 2.0;
+        die "Skip tests" if $LWP::UserAgent::VERSION <  2.0;
+        die "Skip tests" if $LWP::UserAgent::VERSION >= 5.822;
     };
 
     if($@) {
-        plan skip_all => "Only with LWP::UserAgent 2.0";
+        plan skip_all => "Only with 2.0 < LWP::UserAgent <= 5.822";
     } else {
         plan tests => 3;
     }
