diff -uNr rrdtool-1.2.12/bindings/perl-piped/t/base.t rrdtool-1.2.12-new/bindings/perl-piped/t/base.t
--- rrdtool-1.2.12/bindings/perl-piped/t/base.t	2005-12-18 15:30:46.000000000 -0500
+++ rrdtool-1.2.12-new/bindings/perl-piped/t/base.t	2006-02-27 21:23:35.000000000 -0500
@@ -23,7 +23,7 @@
 
 print "ok 1 module load\n";
 
-ok("RRDp::start", RRDp::start "../../src/rrdtool" > 0);
+ok("RRDp::start", RRDp::start "rrdtool" > 0);
 
 $now=time();
 RRDp::cmd qw(create demo.rrd --start ), $now, qw(--step 100 ),
