diff -Nurd rrdtool-1.2.30/bindings/perl-piped/t/base.t rrdtool-1.2.30-patched/bindings/perl-piped/t/base.t
--- rrdtool-1.2.30/bindings/perl-piped/t/base.t	2009-01-19 09:29:14.000000000 -0500
+++ rrdtool-1.2.30-patched/bindings/perl-piped/t/base.t	2009-02-20 08:45:31.000000000 -0500
@@ -23,7 +23,7 @@
 
 print "ok 1 module load\n";
 
-ok("RRDp::start", RRDp::start "../../src/rrdtool" > 0);
+ok("RRDp::start", RRDp::start "rrdtool" > 0);
 
 $now=time();
 RRDp::cmd qw(create demo.rrd --start ), $now, qw(--step 100 ),
diff -Nurd rrdtool-1.2.30/bindings/perl-shared/Makefile.PL rrdtool-1.2.30-patched/bindings/perl-shared/Makefile.PL
--- rrdtool-1.2.30/bindings/perl-shared/Makefile.PL	2009-01-19 09:29:14.000000000 -0500
+++ rrdtool-1.2.30-patched/bindings/perl-shared/Makefile.PL	2009-02-20 08:46:26.000000000 -0500
@@ -30,10 +30,9 @@
     'VERSION_FROM' => 'RRDs.pm', # finds $VERSION
     'DEFINE'	   => "-DPERLPATCHLEVEL=$Config{PATCHLEVEL}",
     'INC'          => '-I../../src',
-    'LIBS'         => '-lm -lpng -lz -lfreetype',
+    'LIBS'         => '-lm -lpng -lz -lfreetype -lrrd',
     # Perl will figure out which one is valid
     #'dynamic_lib'  => {'OTHERLDFLAGS' => "$librrd -lm"},
-    'depend'       => {'RRDs.c' => "../../src/librrd.la"},
     'LDFROM'       => '$(OBJECT) '.$librrd,        
     'realclean'    => {FILES => 't/demo?.rrd t/demo?.png' }
 );
