diff -u -r scsh-0.6.6-orig/c/unix/io.c scsh-0.6.6/c/unix/io.c
--- scsh-0.6.6-orig/c/unix/io.c	Tue Sep 14 14:44:54 1999
+++ scsh-0.6.6/c/unix/io.c	Tue May 17 20:26:47 2005
@@ -146,12 +146,12 @@
       return 0; }
 }
 
+static long write_integer(unsigned long n, FILE *port);
+
 long
 ps_write_integer(long n, FILE *port)
 {
   int status;
-
-  static long write_integer(unsigned long n, FILE *port);
 
   if (n == 0) {
     WRITE_CHAR('0', port, status);
