diff -Naur --exclude='*~' ccl/scripts/ccl ccl-fink/scripts/ccl
--- ccl/scripts/ccl	2014-12-27 09:10:48.000000000 +0100
+++ ccl-fink/scripts/ccl	2014-12-27 09:21:48.000000000 +0100
@@ -9,14 +9,14 @@
 # environment takes precedence over definition made below.
 
 if [ -z "$CCL_DEFAULT_DIRECTORY" ]; then
-  CCL_DEFAULT_DIRECTORY=/usr/local/src/ccl
+  CCL_DEFAULT_DIRECTORY=@PREFIX@/bin
 fi
 
 # If you don't want to guess the name of the lisp kernel on
 # every invocation (or if you want to use a kernel with a
 # non-default name), you might want to uncomment and change
 # the following line:
-#OPENMCL_KERNEL=some_name
+OPENMCL_KERNEL=dx86cl
 
 if [ -z "$OPENMCL_KERNEL" ]; then
   case `uname -s` in
@@ -46,4 +46,3 @@
 
 export CCL_DEFAULT_DIRECTORY
 exec ${CCL_DEFAULT_DIRECTORY}/${OPENMCL_KERNEL} "$@"
-
diff -Naur --exclude='*~' ccl/scripts/ccl64 ccl-fink/scripts/ccl64
--- ccl/scripts/ccl64	2014-12-27 09:10:48.000000000 +0100
+++ ccl-fink/scripts/ccl64	2014-12-27 09:22:14.000000000 +0100
@@ -9,14 +9,14 @@
 # environment takes precedence over definition made below.
 
 if [ -z "$CCL_DEFAULT_DIRECTORY" ]; then
-  CCL_DEFAULT_DIRECTORY=/usr/local/src/ccl
+  CCL_DEFAULT_DIRECTORY=@PREFIX@/bin
 fi
 
 # If you don't want to guess the name of the OpenMCL kernel on
 # every invocation (or if you want to use a kernel with a
 # non-default name), you might want to uncomment and change
 # the following line:
-#OPENMCL_KERNEL=some_name
+OPENMCL_KERNEL=dx86cl64
 
 if [ -z "$OPENMCL_KERNEL" ]; then
   case `uname -s` in
@@ -79,4 +79,3 @@
 
 export CCL_DEFAULT_DIRECTORY
 exec ${CCL_DEFAULT_DIRECTORY}/${OPENMCL_KERNEL} "$@"
-
