--- cfe-3.4.1.src/lib/Driver/Tools.cpp.orig	2014-05-20 16:54:39.000000000 -0400
+++ cfe-3.4.1.src/lib/Driver/Tools.cpp	2014-05-20 17:01:28.000000000 -0400
@@ -5123,9 +5123,12 @@
 
   Args.AddAllArgs(CmdArgs, options::OPT_L);
 
-  if (Args.hasArg(options::OPT_fopenmp))
+  if (Args.hasArg(options::OPT_fopenmp)) {
+    // Help clang find libiomp5.dylib
+    CmdArgs.push_back("-L@FINK_PREFIX@/opt/llvm-3.4/lib");
     // This is more complicated in gcc...
     CmdArgs.push_back("-liomp5");
+  }
 
   AddLinkerInputs(getToolChain(), Inputs, Args, CmdArgs);
   
@@ -6690,7 +6693,9 @@
 
       bool OpenMP = Args.hasArg(options::OPT_fopenmp);
       if (OpenMP) {
-        CmdArgs.push_back("-liomp5");
+	// Help clang find libiomp5.dylib
+	CmdArgs.push_back("-L@FINK_PREFIX@/opt/llvm-3.4/lib");
+	CmdArgs.push_back("-liomp5");
       }
 
       AddLibgcc(ToolChain.getTriple(), D, CmdArgs, Args);
