diff -ruN php-5.4.3.orig/configure php-5.4.3/configure
--- php-5.4.3.orig/configure	2012-05-08 00:19:35.000000000 -0600
+++ php-5.4.3/configure	2012-05-09 15:45:14.000000000 -0600
@@ -1997,7 +1997,7 @@
 
 PEAR:
 
-  --with-pear=DIR         Install PEAR in DIR [PREFIX/lib/php]
+  --with-pear=DIR         Install PEAR in DIR [PREFIX/lib/php5]
   --without-pear          Do not install PEAR
 
 Zend:
@@ -67393,7 +67393,7 @@
 
   if test "$ext_shared" = "yes" ; then
     case $host_alias in
-      *darwin*)
+      *darwin7*)
           as_fn_error "
 Due to the way that loadable modules work on OSX/Darwin, you need to
 compile the PDO package statically into the PHP core.
@@ -102635,7 +102635,8 @@
   fi
 
   if test -n "$recode_conflict"; then
-    as_fn_error "recode extension can not be configured together with:$recode_conflict" "$LINENO" 5
+    $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: recode extension can not be configured together with:$recode_conflict" >&5
+    $as_echo "$as_me: WARNING: recode extension can not be configured together with:$recode_conflict" >&2;
   fi
 fi
 
@@ -102762,7 +102763,7 @@
   if test "$PHP_PEAR" = "DEFAULT" || test "$PHP_PEAR" = "yes"; then
     case $PHP_LAYOUT in
       GNU) PEAR_INSTALLDIR=$datadir/pear;;
-      *)   PEAR_INSTALLDIR=$libdir/php;;
+      *)   PEAR_INSTALLDIR=$libdir/php5;;
     esac
   fi
 
@@ -104644,12 +104645,12 @@
 
 case $libdir in
   '${exec_prefix}/lib')
-    libdir=$libdir/php
+    libdir=$libdir/php5
     ;;
 esac
 case $datadir in
   '${prefix}/share')
-    datadir=$datadir/php
+    datadir=$datadir/php5
     ;;
 esac
 
@@ -104714,7 +104715,7 @@
 EXPANDED_DATADIR=$datadir
 EXPANDED_PHP_CONFIG_FILE_PATH=`eval echo "$PHP_CONFIG_FILE_PATH"`
 EXPANDED_PHP_CONFIG_FILE_SCAN_DIR=`eval echo "$PHP_CONFIG_FILE_SCAN_DIR"`
-INCLUDE_PATH=.:$EXPANDED_PEAR_INSTALLDIR
+INCLUDE_PATH=.:$EXPANDED_PEAR_INSTALLDIR:@FINKPREFIX@/share/pear
 
 exec_prefix=$old_exec_prefix
 libdir=$old_libdir
diff -ruN php-5.4.3.orig/ext/ext_skel php-5.4.3/ext/ext_skel
--- php-5.4.3.orig/ext/ext_skel	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/ext/ext_skel	2012-05-09 15:45:14.000000000 -0600
@@ -70,7 +70,7 @@
 fi
 
 if test -z "$skel_dir"; then
-  skel_dir="skeleton"
+  skel_dir="@FINKPREFIX@/lib/php5/skeleton"
 fi
 
 ## convert skel_dir to full path
diff -ruN php-5.4.3.orig/ext/mysql/config.m4 php-5.4.3/ext/mysql/config.m4
--- php-5.4.3.orig/ext/mysql/config.m4	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/ext/mysql/config.m4	2012-05-09 15:45:14.000000000 -0600
@@ -76,7 +76,7 @@
 Note that the MySQL client library is not bundled anymore!])
   fi
 
-  if test "$enable_maintainer_zts" = "yes"; then
+  if true || test "$enable_maintainer_zts" = "yes"; then
     MYSQL_LIBNAME=mysqlclient_r
   else
     MYSQL_LIBNAME=mysqlclient
diff -ruN php-5.4.3.orig/ext/mysqli/config.m4 php-5.4.3/ext/mysqli/config.m4
--- php-5.4.3.orig/ext/mysqli/config.m4	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/ext/mysqli/config.m4	2012-05-09 15:45:14.000000000 -0600
@@ -25,7 +25,7 @@
     MYSQL_LIB_CFG='--libmysqld-libs'
     dnl mysqlnd doesn't support embedded, so we have to add some extra stuff
     mysqli_extra_sources="mysqli_embedded.c"
-  elif test "$enable_maintainer_zts" = "yes"; then
+  elif true || test "$enable_maintainer_zts" = "yes"; then
     MYSQL_LIB_CFG='--libs_r'
     MYSQL_LIB_NAME='mysqlclient_r'
   else
diff -ruN php-5.4.3.orig/ext/pdo_mysql/config.m4 php-5.4.3/ext/pdo_mysql/config.m4
--- php-5.4.3.orig/ext/pdo_mysql/config.m4	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/ext/pdo_mysql/config.m4	2012-05-09 15:45:14.000000000 -0600
@@ -55,7 +55,7 @@
       if test "x$SED" = "x"; then
         AC_PATH_PROG(SED, sed)
       fi
-      if test "$enable_maintainer_zts" = "yes"; then
+      if true || test "$enable_maintainer_zts" = "yes"; then
         PDO_MYSQL_LIBNAME=mysqlclient_r
         PDO_MYSQL_LIBS=`$PDO_MYSQL_CONFIG --libs_r | $SED -e "s/'//g"`
       else
diff -ruN php-5.4.3.orig/ext/recode/recode.c php-5.4.3/ext/recode/recode.c
--- php-5.4.3.orig/ext/recode/recode.c	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/ext/recode/recode.c	2012-05-09 15:45:14.000000000 -0600
@@ -149,7 +149,7 @@
 	int req_len, str_len;
 	char *req, *str;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "ss", &req, &req_len, &str, &str_len) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "ss", &req, &req_len, &str, &str_len) == FAILURE || str_len < 0) {
 		return;
 	}
 
diff -ruN php-5.4.3.orig/ext/session/session.c php-5.4.3/ext/session/session.c
--- php-5.4.3.orig/ext/session/session.c	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/ext/session/session.c	2012-05-09 15:45:14.000000000 -0600
@@ -705,7 +705,7 @@
 /* {{{ PHP_INI
  */
 PHP_INI_BEGIN()
-	STD_PHP_INI_ENTRY("session.save_path",          "",          PHP_INI_ALL, OnUpdateSaveDir,save_path,          php_ps_globals,    ps_globals)
+	STD_PHP_INI_ENTRY("session.save_path",          "@FINKPREFIX@/var/lib/php5",          PHP_INI_ALL, OnUpdateSaveDir,save_path,          php_ps_globals,    ps_globals)
 	STD_PHP_INI_ENTRY("session.name",               "PHPSESSID", PHP_INI_ALL, OnUpdateString, session_name,       php_ps_globals,    ps_globals)
 	PHP_INI_ENTRY("session.save_handler",           "files",     PHP_INI_ALL, OnUpdateSaveHandler)
 	STD_PHP_INI_BOOLEAN("session.auto_start",       "0",         PHP_INI_ALL, OnUpdateBool,   auto_start,         php_ps_globals,    ps_globals)
diff -ruN php-5.4.3.orig/ext/standard/datetime.c php-5.4.3/ext/standard/datetime.c
--- php-5.4.3.orig/ext/standard/datetime.c	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/ext/standard/datetime.c	2012-05-09 15:45:14.000000000 -0600
@@ -20,6 +20,9 @@
 
 /* $Id$ */
 
+#define _XOPEN_SOURCE	/* needed to get strptime() declared */
+#define _BSD_SOURCE		/* needed to get ulong declared */
+
 #include "php.h"
 #include "zend_operators.h"
 #include "datetime.h"
diff -ruN php-5.4.3.orig/ext/standard/dl.c php-5.4.3/ext/standard/dl.c
--- php-5.4.3.orig/ext/standard/dl.c	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/ext/standard/dl.c	2012-05-09 15:45:14.000000000 -0600
@@ -74,12 +74,7 @@
 		(strcmp(sapi_module.name, "cli") != 0) &&
 		(strncmp(sapi_module.name, "embed", 5) != 0)
 	) {
-#ifdef ZTS
-		php_error_docref(NULL TSRMLS_CC, E_WARNING, "Not supported in multithreaded Web servers - use extension=%s in your php.ini", filename);
-		RETURN_FALSE;
-#else
 		php_error_docref(NULL TSRMLS_CC, E_DEPRECATED, "dl() is deprecated - use extension=%s in your php.ini", filename);
-#endif
 	}
 
 	php_dl(filename, MODULE_TEMPORARY, return_value, 0 TSRMLS_CC);
diff -ruN php-5.4.3.orig/fink/crontab php-5.4.3/fink/crontab
--- php-5.4.3.orig/fink/crontab	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/crontab	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1,7 @@
+# /etc/cron.d/php5: crontab fragment for php5
+#  This purges session files older than X, where X is defined in seconds
+#  as the largest value of session.gc_maxlifetime from all your php.ini
+#  files, or 24 minutes if not defined.  See @FINKPREFIX@/lib/php5/maxlifetime
+
+# Look for and purge old sessions every 30 minutes
+09,39 *     * * *     root   [ -x @FINKPREFIX@/lib/php5/maxlifetime ] && [ -d @FINKPREFIX@/var/lib/php5 ] && find @FINKPREFIX@/var/lib/php5/ -depth -mindepth 1 -maxdepth 1 -type f -ignore_readdir_race -cmin +$(@FINKPREFIX@/lib/php5/maxlifetime) ! -execdir fuser -s {} 2>/dev/null \; -delete
diff -ruN php-5.4.3.orig/fink/extramodulelist php-5.4.3/fink/extramodulelist
--- php-5.4.3.orig/fink/extramodulelist	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/extramodulelist	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1,10 @@
+mysql MySQL mysqli
+mysql MySQL pdo_mysql
+mysqlnd MySQL mysql
+mysqlnd MySQL mysqli
+mysqlnd MySQL pdo_mysql
+interbase InterBase/Firebird pdo_firebird
+odbc ODBC pdo_odbc
+pgsql PostgreSQL pdo_pgsql
+sqlite SQLite pdo_sqlite
+sybase Sybase pdo_dblib
diff -ruN php-5.4.3.orig/fink/libapache2-mod-php5.conf php-5.4.3/fink/libapache2-mod-php5.conf
--- php-5.4.3.orig/fink/libapache2-mod-php5.conf	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/libapache2-mod-php5.conf	2012-06-07 12:59:45.000000000 -0600
@@ -0,0 +1,16 @@
+<IfModule mod_php5.c>
+    <FilesMatch "\.ph(p3?|tml)$">
+	SetHandler application/x-httpd-php
+    </FilesMatch>
+    <FilesMatch "\.phps$">
+	SetHandler application/x-httpd-php-source
+    </FilesMatch>
+    # To disable php in user directories uncomment the following lines
+    # (from <IfModule ...> to </IfModule>.) Do NOT set it to On as it
+    # prevents .htaccess files from disabling it.
+    #<IfModule mod_userdir.c>
+    #    <Directory /Users/*/Sites>
+    #        php_admin_value engine Off
+    #    </Directory>
+    #</IfModule>
+</IfModule>
diff -ruN php-5.4.3.orig/fink/libapache2-mod-php5.load php-5.4.3/fink/libapache2-mod-php5.load
--- php-5.4.3.orig/fink/libapache2-mod-php5.load	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/libapache2-mod-php5.load	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1 @@
+LoadModule php5_module @FINKPREFIX@/lib/apache2/modules/libphp5.so
diff -ruN php-5.4.3.orig/fink/libapache2-mod-php5filter.conf php-5.4.3/fink/libapache2-mod-php5filter.conf
--- php-5.4.3.orig/fink/libapache2-mod-php5filter.conf	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/libapache2-mod-php5filter.conf	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1,6 @@
+<IfModule mod_php5.c>
+    <FilesMatch "\.ph(p3?|tml)$">
+	SetInputFilter PHP
+	SetOutputFilter PHP
+    </FilesMatch>
+</IfModule>
diff -ruN php-5.4.3.orig/fink/libapache2-mod-php5filter.load php-5.4.3/fink/libapache2-mod-php5filter.load
--- php-5.4.3.orig/fink/libapache2-mod-php5filter.load	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/libapache2-mod-php5filter.load	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1 @@
+LoadModule php5_module @FINKPREFIX@/lib/apache2/modules/libphp5filter.so
diff -ruN php-5.4.3.orig/fink/maxlifetime php-5.4.3/fink/maxlifetime
--- php-5.4.3.orig/fink/maxlifetime	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/maxlifetime	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1,23 @@
+#!/bin/sh -e
+
+max=1440
+
+if which php5 >/dev/null 2>&1; then
+    for sapi in apache2 apache2filter cgi fpm; do
+	if [ -e /etc/php5/${sapi}/php.ini ]; then
+	    cur=$(php5 -c /etc/php5/${sapi}/php.ini -d "error_reporting='~E_ALL'" -r 'print ini_get("session.gc_maxlifetime");')
+	    [ -z "$cur" ] && cur=0
+	    [ "$cur" -gt "$max" ] && max=$cur
+	fi
+    done
+else
+    for ini in /etc/php5/*/php.ini /etc/php5/conf.d/*.ini; do
+        cur=$(sed -n -e 's/^[[:space:]]*session.gc_maxlifetime[[:space:]]*=[[:space:]]*\([0-9]\+\).*$/\1/p' $ini 2>/dev/null || true);
+        [ -z "$cur" ] && cur=0
+        [ "$cur" -gt "$max" ] && max=$cur
+    done
+fi
+
+echo $(($max/60))
+
+exit 0
diff -ruN php-5.4.3.orig/fink/modulelist php-5.4.3/fink/modulelist
--- php-5.4.3.orig/fink/modulelist	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/modulelist	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1,22 @@
+curl CURL
+common PDO pdo 10
+enchant Enchant
+gd GD
+gmp GMP
+imap IMAP
+interbase Interbase
+intl Internationalisation
+ldap LDAP
+mcrypt MCrypt
+mysql MySQL
+mysqlnd MySQL mysqlnd 10
+odbc ODBC
+pgsql PostgreSQL
+pspell pspell
+recode recode
+snmp SNMP
+sqlite SQLite sqlite3
+sybase Sybase mssql
+tidy tidy
+xmlrpc XML-RPC
+xsl XSL
diff -ruN php-5.4.3.orig/fink/patches/PEAR-Builder-print-info-about-php5-dev.patch php-5.4.3/fink/patches/PEAR-Builder-print-info-about-php5-dev.patch
--- php-5.4.3.orig/fink/patches/PEAR-Builder-print-info-about-php5-dev.patch	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/patches/PEAR-Builder-print-info-about-php5-dev.patch	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1,10 @@
+--- a/PEAR/Builder.php	2011-05-14 20:43:01.000000000 +0000
++++ b/PEAR/Builder.php		2011-05-26 15:56:41.096485701 +0000
+@@ -309,6 +309,8 @@ class PEAR_Builder extends PEAR_Common
+         }
+ 
+         if (!$err) {
++             print "If the command failed with 'phpize: not found' then you need to install php5-dev package";
++             print "You can do it by running 'fink install php5-dev' as a root user";
+             return $this->raiseError("`phpize' failed");
+         }
diff -ruN php-5.4.3.orig/fink/patches/deprecate_short_open_tag.patch php-5.4.3/fink/patches/deprecate_short_open_tag.patch
--- php-5.4.3.orig/fink/patches/deprecate_short_open_tag.patch	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/patches/deprecate_short_open_tag.patch	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1,22 @@
+diff -ruN php-5.4.1.orig/Zend/zend_language_scanner.c php-5.4.1/Zend/zend_language_scanner.c
+--- php-5.4.1.orig/Zend/zend_language_scanner.c	2012-04-24 10:47:33.000000000 -0600
++++ php-5.4.1/Zend/zend_language_scanner.c	2012-05-09 11:19:57.000000000 -0600
+@@ -1171,6 +1171,7 @@
+ #line 1771 "Zend/zend_language_scanner.l"
+ 		{
+ 	if (CG(short_tags)) {
++		zend_error(E_DEPRECATED, "Usage of short open tag <? is not recommended and will be disabled by default in future versions of PHP");
+ 		zendlval->value.str.val = yytext; /* no copying - intentional */
+ 		zendlval->value.str.len = yyleng;
+ 		zendlval->type = IS_STRING;
+diff -ruN php-5.4.1.orig/Zend/zend_language_scanner.l php-5.4.1/Zend/zend_language_scanner.l
+--- php-5.4.1.orig/Zend/zend_language_scanner.l	2012-04-24 10:47:33.000000000 -0600
++++ php-5.4.1/Zend/zend_language_scanner.l	2012-05-09 11:19:26.000000000 -0600
+@@ -1770,6 +1770,7 @@
+ 
+ <INITIAL>"<?" {
+ 	if (CG(short_tags)) {
++		zend_error(E_DEPRECATED, "Usage of short open tag <? is not recommended and will be disabled by default in future versions of PHP");
+ 		zendlval->value.str.val = yytext; /* no copying - intentional */
+ 		zendlval->value.str.len = yyleng;
+ 		zendlval->type = IS_STRING;
diff -ruN php-5.4.3.orig/fink/patches/fpm-config.patch php-5.4.3/fink/patches/fpm-config.patch
--- php-5.4.3.orig/fink/patches/fpm-config.patch	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/patches/fpm-config.patch	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1,54 @@
+Description: Add major version number to paths and allow process pools
+ to be configured in individual files in /etc/php5/fpm/pool.d/
+Origin: vendor
+Forwarded: not-needed
+Last-Update: 2010-07-30
+
+--- a/sapi/fpm/php-fpm.conf.in
++++ b/sapi/fpm/php-fpm.conf.in
+@@ -12,7 +12,7 @@
+ ; Relative path can also be used. They will be prefixed by:
+ ;  - the global prefix if it's been set (-p arguement)
+ ;  - @prefix@ otherwise
+-;include=etc/fpm.d/*.conf
++;include=@FINKPREFIX@/etc/php5/fpm/*.conf
+ 
+ ;;;;;;;;;;;;;;;;;;
+ ; Global Options ;
+@@ -22,14 +22,14 @@
+ ; Pid file
+ ; Note: the default prefix is @EXPANDED_LOCALSTATEDIR@
+ ; Default Value: none
+-;pid = run/php-fpm.pid
++pid = @FINKPREFIX@/var/run/php5-fpm.pid
+ 
+ ; Error log file
+ ; If it's set to "syslog", log is sent to syslogd instead of being written
+ ; in a local file.
+ ; Note: the default prefix is @EXPANDED_LOCALSTATEDIR@
+ ; Default Value: log/php-fpm.log
+-;error_log = log/php-fpm.log
++error_log = @FINKPREFIX@/var/log/php5-fpm.log
+ 
+ ; syslog_facility is used to specify what type of program is logging the
+ ; message. This lets syslogd specify that messages from different facilities
+@@ -108,6 +108,10 @@
+ ; used in logs and stats. There is no limitation on the number of pools which
+ ; FPM can handle. Your system will tell you anyway :)
+ 
++; To configure the pools it is recommended to have one .conf file per
++; pool in the following directory:
++include=@FINKPREFIX@/etc/php5/fpm/pool.d/*.conf
++
+ ; Start a new pool named 'www'.
+ ; the variable $pool can we used in any directive and will be replaced by the
+ ; pool name ('www' here)
+@@ -442,7 +446,7 @@ pm.max_spare_servers = 3
+ ; Chdir to this directory at the start.
+ ; Note: relative path can be used.
+ ; Default Value: current directory or / when chroot
+-;chdir = /var/www
++chdir = /
+  
+ ; Redirect worker stdout and stderr into main error log. If not set, stdout and
+ ; stderr will be redirected to /dev/null according to FastCGI specs.
diff -ruN php-5.4.3.orig/fink/patches/gd-libJPEG-Version.patch php-5.4.3/fink/patches/gd-libJPEG-Version.patch
--- php-5.4.3.orig/fink/patches/gd-libJPEG-Version.patch	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/patches/gd-libJPEG-Version.patch	2012-06-07 11:57:52.000000000 -0600
@@ -0,0 +1,13 @@
+diff -ruN php-5.4.3.orig/ext/gd/libgd/gd_compat.c php-5.4.3.patched/ext/gd/libgd/gd_compat.c
+--- php-5.4.3.orig/ext/gd/libgd/gd_compat.c	2012-05-07 23:22:56.000000000 -0600
++++ php-5.4.3.patched/ext/gd/libgd/gd_compat.c	2012-06-07 11:57:03.000000000 -0600
+@@ -20,6 +20,9 @@
+ 		case 62:
+ 			return "6b";
+ 			break;
++		case 80:
++			return "8";
++			break;
+ 		default:
+ 			return "unknown";
+ 	}
diff -ruN php-5.4.3.orig/fink/patches/gdIOCtx.patch php-5.4.3/fink/patches/gdIOCtx.patch
--- php-5.4.3.orig/fink/patches/gdIOCtx.patch	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/patches/gdIOCtx.patch	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1,124 @@
+--- a/ext/gd/gd_ctx.c
++++ b/ext/gd/gd_ctx.c
+@@ -46,33 +46,6 @@ static void _php_image_output_ctxfree(st
+ 	}
+ }
+ 
+-static void _php_image_stream_putc(struct gdIOCtx *ctx, int c)  {
+-	char ch = (char) c;
+-	php_stream * stream = (php_stream *)ctx->data;
+-	TSRMLS_FETCH();
+-	php_stream_write(stream, &ch, 1);
+-}
+-
+-static int _php_image_stream_putbuf(struct gdIOCtx *ctx, const void* buf, int l)
+-{
+-	php_stream * stream = (php_stream *)ctx->data;
+-	TSRMLS_FETCH();
+-	return php_stream_write(stream, (void *)buf, l);
+-}
+-
+-static void _php_image_stream_ctxfree(struct gdIOCtx *ctx)
+-{
+-	TSRMLS_FETCH();
+-
+-	if(ctx->data) {
+-		php_stream_close((php_stream *) ctx->data);
+-		ctx->data = NULL;
+-	}
+-	if(ctx) {
+-		efree(ctx);
+-	}
+-}
+-
+ /* {{{ _php_image_output_ctx */
+ static void _php_image_output_ctx(INTERNAL_FUNCTION_PARAMETERS, int image_type, char *tn, void (*func_p)())
+ {
+@@ -81,12 +54,11 @@ static void _php_image_output_ctx(INTERN
+ 	int file_len = 0;
+ 	long quality, basefilter;
+ 	gdImagePtr im;
++	FILE *fp = NULL;
+ 	int argc = ZEND_NUM_ARGS();
+ 	int q = -1, i;
+ 	int f = -1;
+ 	gdIOCtx *ctx = NULL;
+-	zval *to_zval = NULL;
+-	php_stream *stream;
+ 
+ 	/* The third (quality) parameter for Wbmp stands for the threshold when called from image2wbmp().
+ 	 * The third (quality) parameter for Wbmp and Xbm stands for the foreground color index when called
+@@ -103,7 +75,7 @@ static void _php_image_output_ctx(INTERN
+ 		 * PHP_GDIMG_TYPE_WBM 
+ 		 * PHP_GDIMG_TYPE_WEBP 
+ 		 * */
+-		if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "r|z/!ll", &imgind, &to_zval, &quality, &basefilter) == FAILURE) {
++		if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "r|p!ll", &imgind, &file, &file_len, &quality, &basefilter) == FAILURE) {
+ 			return;
+ 		}
+ 	}
+@@ -117,21 +89,19 @@ static void _php_image_output_ctx(INTERN
+ 		}
+ 	}
+ 
+-	if (argc > 1 && to_zval != NULL) {
+-		if (Z_TYPE_P(to_zval) == IS_RESOURCE) {
+-			php_stream_from_zval_no_verify(stream, &to_zval);
+-			if (stream == NULL) {
+-				RETURN_FALSE;
+-			}
+-		} else if (Z_TYPE_P(to_zval) == IS_STRING) {
+-			stream = php_stream_open_wrapper(Z_STRVAL_P(to_zval), "wb", REPORT_ERRORS|IGNORE_PATH|IGNORE_URL_WIN, NULL);
+-			if (stream == NULL) {
+-				RETURN_FALSE;
+-			}
+-		} else {
+-			php_error_docref(NULL TSRMLS_CC, E_WARNING, "Invalid 2nd parameter, it must a filename or a stream");
++	if (argc > 1 && file_len) {
++		if (strlen(file) != file_len) {
++			RETURN_FALSE;
++		}
++		PHP_GD_CHECK_OPEN_BASEDIR(file, "Invalid filename");
++
++		fp = VCWD_FOPEN(file, "wb");
++		if (!fp) {
++			php_error_docref(NULL TSRMLS_CC, E_WARNING, "Unable to open '%s' for writing: %s", file, strerror(errno));
+ 			RETURN_FALSE;
+ 		}
++
++		ctx = gdNewFileCtx(fp);
+ 	} else {
+ 		ctx = emalloc(sizeof(gdIOCtx));
+ 		ctx->putC = _php_image_output_putc;
+@@ -145,14 +115,6 @@ static void _php_image_output_ctx(INTERN
+ #endif
+ 	}
+ 
+-	if (!ctx)	{
+-		ctx = emalloc(sizeof(gdIOCtx));
+-		ctx->putC = _php_image_stream_putc;
+-		ctx->putBuf = _php_image_stream_putbuf;
+-		ctx->gd_free = _php_image_stream_ctxfree;
+-		ctx->data = (void *)stream;
+-	}
+-
+ 	switch(image_type) {
+ 		case PHP_GDIMG_CONVERT_WBM:
+ 			if(q<0||q>255) {
+@@ -189,11 +151,12 @@ static void _php_image_output_ctx(INTERN
+ 			break;
+ 	}
+ 
+-#if HAVE_LIBGD204
+ 	ctx->gd_free(ctx);
+-#else
+-	ctx->free(ctx);
+-#endif
++
++	if(fp) {
++		fflush(fp);
++		fclose(fp);
++	}
+ 
+ 	RETURN_TRUE;
+ }
diff -ruN php-5.4.3.orig/fink/patches/ldap_fix.patch php-5.4.3/fink/patches/ldap_fix.patch
--- php-5.4.3.orig/fink/patches/ldap_fix.patch	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/patches/ldap_fix.patch	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1,27 @@
+Description: Prevent null dereferencing in ldap_explode_dn()
+Origin: vendor
+Bug-Debian: http://bugs.debian.org/205405
+Forwarded: no
+Last-Update: 2010-01-18
+
+--- a/ext/ldap/ldap.c
++++ b/ext/ldap/ldap.c
+@@ -1212,7 +1212,7 @@ PHP_FUNCTION(ldap_explode_dn)
+ 	}
+ 
+ 	i=0;
+-	while (ldap_value[i] != NULL) i++;
++	while (ldap_value && ldap_value[i] != NULL) i++;
+ 	count = i;
+ 
+ 	array_init(return_value);
+@@ -1222,7 +1222,8 @@ PHP_FUNCTION(ldap_explode_dn)
+ 		add_index_string(return_value, i, ldap_value[i], 1);
+ 	}
+ 
+-	ldap_value_free(ldap_value);
++	if (ldap_value)
++		ldap_value_free(ldap_value);
+ }
+ /* }}} */
+ 
diff -ruN php-5.4.3.orig/fink/patches/phpinfo_no_configure.patch php-5.4.3/fink/patches/phpinfo_no_configure.patch
--- php-5.4.3.orig/fink/patches/phpinfo_no_configure.patch	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/patches/phpinfo_no_configure.patch	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1,33 @@
+Description: Disable configure parameters on phpinfo() output
+ .
+ Patch needs to be discussed with upstream and the issues that lead to
+ its addition re-checked.  Quoting changelog entry:
+ .
+ Add [...], which disables the display of our "Configure Command" in
+ phpinfo(), which was the source of many bogus bug reports over the
+ years, due to people misinterpreting its meaning.
+Origin: vendor
+Forwarded: no
+Last-Update: 2010-01-18
+
+--- a/ext/standard/info.c
++++ b/ext/standard/info.c
+@@ -704,7 +704,7 @@ PHPAPI void php_print_info(int flag TSRM
+ #ifdef ARCHITECTURE
+ 		php_info_print_table_row(2, "Architecture", ARCHITECTURE);
+ #endif
+-#ifdef CONFIGURE_COMMAND
++#if 0
+ 		php_info_print_table_row(2, "Configure Command", CONFIGURE_COMMAND );
+ #endif
+ 
+--- a/ext/standard/tests/general_functions/phpinfo.phpt
++++ b/ext/standard/tests/general_functions/phpinfo.phpt
+@@ -20,7 +20,6 @@ PHP Version => %s
+ 
+ System => %s
+ Build Date => %s%a
+-Configure Command => %s
+ Server API => Command Line Interface
+ Virtual Directory Support => %s
+ Configuration File (php.ini) Path => %s
diff -ruN php-5.4.3.orig/fink/patches/sybase-alias.patch php-5.4.3/fink/patches/sybase-alias.patch
--- php-5.4.3.orig/fink/patches/sybase-alias.patch	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/patches/sybase-alias.patch	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1,41 @@
+--- a/ext/mssql/php_mssql.c
++++ b/ext/mssql/php_mssql.c
+@@ -178,6 +178,38 @@ const zend_function_entry mssql_function
+  	PHP_FE(mssql_execute,				arginfo_mssql_execute)
+ 	PHP_FE(mssql_free_statement,		arginfo_mssql_free_statement)
+  	PHP_FE(mssql_guid_string,			arginfo_mssql_guid_string)
++#if !defined(PHP_WIN32) && !defined(HAVE_SYBASE_CT)
++        PHP_FALIAS(sybase_connect, mssql_connect,		arginfo_mssql_connect)
++        PHP_FALIAS(sybase_pconnect, mssql_pconnect,		arginfo_mssql_connect)
++        PHP_FALIAS(sybase_close, mssql_close,			arginfo_mssql_close)
++        PHP_FALIAS(sybase_select_db, mssql_select_db,		arginfo_mssql_select_db)
++        PHP_FALIAS(sybase_query, mssql_query,			arginfo_mssql_query)
++        PHP_FALIAS(sybase_fetch_batch, mssql_fetch_batch,	arginfo_mssql_fetch_batch)
++        PHP_FALIAS(sybase_affected_rows, mssql_rows_affected,	arginfo_mssql_rows_affected)
++        PHP_FALIAS(sybase_free_result, mssql_free_result,	arginfo_mssql_fetch_batch)
++        PHP_FALIAS(sybase_get_last_message, mssql_get_last_message,	arginfo_mssql_get_last_message)
++        PHP_FALIAS(sybase_num_rows, mssql_num_rows,		arginfo_mssql_fetch_batch)
++        PHP_FALIAS(sybase_num_fields, mssql_num_fields,		arginfo_mssql_fetch_batch)
++        PHP_FALIAS(sybase_fetch_field, mssql_fetch_field,	arginfo_mssql_fetch_field)
++        PHP_FALIAS(sybase_fetch_row, mssql_fetch_row,		arginfo_mssql_fetch_batch)
++        PHP_FALIAS(sybase_fetch_array, mssql_fetch_array,	arginfo_mssql_fetch_array)
++        PHP_FALIAS(sybase_fetch_assoc, mssql_fetch_assoc,	arginfo_mssql_fetch_assoc)
++        PHP_FALIAS(sybase_fetch_object, mssql_fetch_object,	arginfo_mssql_fetch_batch)
++        PHP_FALIAS(sybase_field_length, mssql_field_length,	arginfo_mssql_field_length)
++        PHP_FALIAS(sybase_field_name, mssql_field_name,		arginfo_mssql_field_length)
++        PHP_FALIAS(sybase_field_type, mssql_field_type,		arginfo_mssql_field_length)
++        PHP_FALIAS(sybase_data_seek, mssql_data_seek,		arginfo_mssql_data_seek)
++        PHP_FALIAS(sybase_field_seek, mssql_field_seek,		arginfo_mssql_fetch_field)
++        PHP_FALIAS(sybase_result, mssql_result,			arginfo_mssql_result)
++        PHP_FALIAS(sybase_next_result, mssql_next_result,	arginfo_mssql_fetch_assoc)
++        PHP_FALIAS(sybase_min_error_severity, mssql_min_error_severity,	arginfo_mssql_min_error_severity)
++        PHP_FALIAS(sybase_min_message_severity, mssql_min_message_severity,	arginfo_mssql_min_error_severity)
++        PHP_FALIAS(sybase_init, mssql_init,			arginfo_mssql_init)
++        PHP_FALIAS(sybase_bind, mssql_bind,			arginfo_mssql_bind)
++        PHP_FALIAS(sybase_execute, mssql_execute,		arginfo_mssql_execute)
++        PHP_FALIAS(sybase_free_statement, mssql_free_statement,	arginfo_mssql_free_statement)
++        PHP_FALIAS(sybase_guid_string, mssql_guid_string,	arginfo_mssql_guid_string)
++#endif
+ 	PHP_FE_END
+ };
+ /* }}} */
diff -ruN php-5.4.3.orig/fink/php5-module.ini php-5.4.3/fink/php5-module.ini
--- php-5.4.3.orig/fink/php5-module.ini	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/php5-module.ini	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1,3 @@
+; configuration for php @extname@ module
+; priority=@priority@
+extension=@dsoname@.so
diff -ruN php-5.4.3.orig/fink/php5enmod php-5.4.3/fink/php5enmod
--- php-5.4.3.orig/fink/php5enmod	1969-12-31 17:00:00.000000000 -0700
+++ php-5.4.3/fink/php5enmod	2012-05-09 15:45:14.000000000 -0600
@@ -0,0 +1,126 @@
+#!/bin/sh
+#
+#  php5enmod - a php5 module manager for Debian
+#
+#  Copyright 2012 Canonical Ltd., All Rights Reserved.
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  On Debian systems the full text of the GPL version 3 is available at
+#  /usr/share/common-licenses/GPL-3.
+#
+set -ue
+
+SCRIPT_NAME=${0##*/}
+ENABLED=0
+DISABLED=0
+VERBOSE=no
+
+cleanup() {
+    if [ "${VERBOSE}" != "no" ]; then
+	if [ ${ENABLED} -gt 0 ] ; then
+            echo "Enabled ${ENABLED} module(s), you may need to restart any running PHP processes."
+	fi
+	if [ ${DISABLED} -gt 0 ] ; then
+            echo "Disabled ${DISABLED} module(s), you may need to restart any running PHP processes."
+	fi
+    fi
+}
+
+trap cleanup EXIT
+
+usage() {
+    echo "usage: ${SCRIPT_NAME} module_name [ module_name_2 ]"
+    exit 1
+}
+
+error() {
+    echo "ERROR: ${1}" >&2
+    shift
+    local ecode=${1:-1}
+    exit ${ecode}
+}
+
+enmods() {
+    local modname=""
+    ENABLED=0
+    for modname in ${@} ; do
+        enmod ${modname}
+    done
+}
+
+dismods() {
+    DISABLED=0
+    local modname=""
+    for modname in ${@} ; do
+        dismod "${modname}"
+    done
+}
+
+enmod() {
+    local modname="$(basename "${1%/[0-9]*}")"
+    local priority="$(basename "${1#[a-z]*/}")"
+    [ "${modname}" = "${priority}" ] && priority=20
+    # assert $modname is in @FINKPREFIX@/etc/php5/mods-available
+    local source_ini="@FINKPREFIX@/etc/php5/mods-available/${modname}.ini"
+    [ -e "${source_ini}" ] || error "${source_ini} does not exist" 2
+    [ -z "${priority}" ] && priority=20
+        
+    # assert $modname is not present in @FINKPREFIX@/etc/php5/conf.d, or already symlink to @FINKPREFIX@/etc/php5/mods-available
+    local live_link="@FINKPREFIX@/etc/php5/conf.d/$priority-$modname.ini"
+    local live_link_content="../mods-available/$modname.ini"
+    if [ -e "${live_link}" ] ; then
+        if [ -h "${live_link}" ] ; then
+            local content="$(readlink "${live_link}")"
+            if [ "${content}" = "${live_link_content}" ] ; then
+                return
+            fi
+        fi
+        error "${modname} module symlink already exists in @FINKPREFIX@/etc/php5/conf.d with different content"
+    fi
+    ln -s "${live_link_content}" "${live_link}"
+    ENABLED=$((${ENABLED}+1))
+}
+
+dismod() {
+    local modname="$(basename "${1%/[0-9]*}")"
+    local live_link=""
+
+    local live_link_content="../mods-available/${modname}.ini"
+    for live_link in $(ls -1 @FINKPREFIX@/etc/php5/conf.d/*.ini); do
+        # assert $modname is in @FINKPREFIX@/etc/php5/conf.d
+	[ -e "${live_link}" ] || continue
+	[ -h "${live_link}" ] || continue
+        # assert $modname is a symlink to @FINKPREFIX@/etc/php5/mods-available
+	[ "$(readlink "${live_link}")" != "${live_link_content}" ] && continue
+	# remove the symlink
+	rm -f "${live_link}"
+	FOUND=1
+    done
+    [ "${FOUND}" -gt 0 ] && DISABLED=$(($DISABLED+1))
+}
+
+# parse args
+# modname
+[ -n ${1:-""} ] || usage
+
+case "${SCRIPT_NAME}" in
+php5enmod)
+    enmods $@
+    ;;
+php5dismod)
+    dismods $@
+    ;;
+*)
+    usage
+    ;;
+esac
+exit 0
diff -ruN php-5.4.3.orig/php.ini-development php-5.4.3/php.ini-development
--- php-5.4.3.orig/php.ini-development	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/php.ini-development	2012-05-09 15:45:14.000000000 -0600
@@ -702,7 +702,7 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;
 
 ; UNIX: "/path1:/path2"
-;include_path = ".:/php/includes"
+;include_path = ".:@FINKPREFIX@/share/php"
 ;
 ; Windows: "\path1;\path2"
 ;include_path = ".;c:\php\includes"
@@ -1391,7 +1391,7 @@
 ; where MODE is the octal representation of the mode. Note that this
 ; does not overwrite the process's umask.
 ; http://php.net/session.save-path
-;session.save_path = "/tmp"
+;session.save_path = "@FINKPREFIX@/var/lib/php5"
 
 ; Whether to use cookies.
 ; http://php.net/session.use-cookies
diff -ruN php-5.4.3.orig/php.ini-production php-5.4.3/php.ini-production
--- php-5.4.3.orig/php.ini-production	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/php.ini-production	2012-05-09 15:45:14.000000000 -0600
@@ -702,7 +702,7 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;
 
 ; UNIX: "/path1:/path2"
-;include_path = ".:/php/includes"
+;include_path = ".:@FINKPREFIX@/share/php"
 ;
 ; Windows: "\path1;\path2"
 ;include_path = ".;c:\php\includes"
@@ -859,51 +859,6 @@
 ; If you only provide the name of the extension, PHP will look for it in its
 ; default extension directory.
 ;
-; Windows Extensions
-; Note that ODBC support is built in, so no dll is needed for it.
-; Note that many DLL files are located in the extensions/ (PHP 4) ext/ (PHP 5)
-; extension folders as well as the separate PECL DLL download (PHP 5).
-; Be sure to appropriately set the extension_dir directive.
-;
-;extension=php_bz2.dll
-;extension=php_curl.dll
-;extension=php_fileinfo.dll
-;extension=php_gd2.dll
-;extension=php_gettext.dll
-;extension=php_gmp.dll
-;extension=php_intl.dll
-;extension=php_imap.dll
-;extension=php_interbase.dll
-;extension=php_ldap.dll
-;extension=php_mbstring.dll
-;extension=php_exif.dll      ; Must be after mbstring as it depends on it
-;extension=php_mysql.dll
-;extension=php_mysqli.dll
-;extension=php_oci8.dll      ; Use with Oracle 10gR2 Instant Client
-;extension=php_oci8_11g.dll  ; Use with Oracle 11gR2 Instant Client
-;extension=php_openssl.dll
-;extension=php_pdo_firebird.dll
-;extension=php_pdo_mysql.dll
-;extension=php_pdo_oci.dll
-;extension=php_pdo_odbc.dll
-;extension=php_pdo_pgsql.dll
-;extension=php_pdo_sqlite.dll
-;extension=php_pgsql.dll
-;extension=php_pspell.dll
-;extension=php_shmop.dll
-
-; The MIBS data available in the PHP distribution must be installed. 
-; See http://www.php.net/manual/en/snmp.installation.php 
-;extension=php_snmp.dll
-
-;extension=php_soap.dll
-;extension=php_sockets.dll
-;extension=php_sqlite3.dll
-;extension=php_sybase_ct.dll
-;extension=php_tidy.dll
-;extension=php_xmlrpc.dll
-;extension=php_xsl.dll
-;extension=php_zip.dll
 
 ;;;;;;;;;;;;;;;;;;;
 ; Module Settings ;
@@ -1391,7 +1346,7 @@
 ; where MODE is the octal representation of the mode. Note that this
 ; does not overwrite the process's umask.
 ; http://php.net/session.save-path
-;session.save_path = "/tmp"
+;session.save_path = "@FINKPREFIX@/var/lib/php5"
 
 ; Whether to use cookies.
 ; http://php.net/session.use-cookies
diff -ruN php-5.4.3.orig/sapi/caudium/config.m4 php-5.4.3/sapi/caudium/config.m4
--- php-5.4.3.orig/sapi/caudium/config.m4	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/sapi/caudium/config.m4	2012-05-09 15:45:14.000000000 -0600
@@ -26,8 +26,8 @@
     AC_MSG_ERROR([Could not find a pike in $PHP_CAUDIUM/bin/])
   fi
   if $PIKE -e 'float v; int rel;sscanf(version(), "Pike v%f release %d", v, rel);v += rel/10000.0; if(v < 7.0268) exit(1); exit(0);'; then
-    PIKE_MODULE_DIR=`$PIKE --show-paths 2>&1| grep '^Module' | sed -e 's/.*: //'`
-    PIKE_INCLUDE_DIR=`echo $PIKE_MODULE_DIR | sed -e 's,lib/pike/modules,include/pike,' -e 's,lib/modules,include/pike,' `
+    PIKE_MODULE_DIR=`$PIKE --show-paths 2>&1| grep '^Master file' | sed -e 's/.*: //' -e 's/master.pike/modules/'`
+    PIKE_INCLUDE_DIR=`echo $PIKE_MODULE_DIR | sed -e 's,lib/modules,,' -e 's,modules,include,' `
     if test -z "$PIKE_INCLUDE_DIR" || test -z "$PIKE_MODULE_DIR"; then
       AC_MSG_ERROR(Failed to figure out Pike module and include directories)
     fi
@@ -84,7 +84,9 @@
   PIKE_VERSION=`$PIKE -e 'string v; int rel;sscanf(version(), "Pike v%s release %d", v, rel); write(v+"."+rel);'`   
   AC_DEFINE(HAVE_CAUDIUM,1,[Whether to compile with Caudium support])
   PHP_SELECT_SAPI(caudium, shared, caudium.c)
-  INSTALL_IT="\$(INSTALL) -m 0755 $SAPI_SHARED $PHP_CAUDIUM/lib/$PIKE_VERSION/PHP5.so"
+  dnl FIXME: This is the ugliest hack in the world!
+  dnl INSTALL_IT="\$(mkinstalldirs) \$(INSTALL_ROOT)$PHP_CAUDIUM/lib/$PIKE_VERSION/ && \$(INSTALL) -m 0755 $SAPI_SHARED \$(INSTALL_ROOT)$PHP_CAUDIUM/lib/$PIKE_VERSION/php5.so"
+  INSTALL_IT="\$(mkinstalldirs) \$(INSTALL_ROOT)$PHP_CAUDIUM/lib/$PIKE_VERSION/ && \$(INSTALL) -m 0755 .$SAPI_SHARED \$(INSTALL_ROOT)$PHP_CAUDIUM/lib/$PIKE_VERSION/PHP5.so"
   RESULT="  *** Pike binary used:         $PIKE
   *** Pike include dir(s) used: $PIKE_INCLUDE_DIR
   *** Pike version:             $PIKE_VERSION"
diff -ruN php-5.4.3.orig/sapi/cli/php.1.in php-5.4.3/sapi/cli/php.1.in
--- php-5.4.3.orig/sapi/cli/php.1.in	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/sapi/cli/php.1.in	2012-05-09 15:45:14.000000000 -0600
@@ -346,13 +346,14 @@
 Show configuration file names
 .SH FILES
 .TP 15
-.B php\-cli.ini
+.B @FINKPREFIX@/etc/php5/cli/php.ini
 The configuration file for the CLI version of PHP.
 .TP
-.B php.ini
-The standard configuration file will only be used when 
-.B php\-cli.ini
-cannot be found.
+.B @FINKPREFIX@/etc/php5/cgi/php.ini
+The configuration file for the CGI version of PHP.
+.TP
+.B @FINKPREFIX@/etc/php5/apache2/php.ini
+The configuration file for the version of PHP that apache2 uses.
 .SH EXAMPLES
 .TP 5
 \fIphp \-r 'echo "Hello World\\n";'\fP
@@ -395,7 +396,7 @@
 .P
 .PD 0
 .RS
-#!/bin/php
+#!@FINKPREFIX@/bin/php
 .P
 <?php
 .P
diff -ruN php-5.4.3.orig/sapi/fpm/fpm/fpm_conf.c php-5.4.3/sapi/fpm/fpm/fpm_conf.c
--- php-5.4.3.orig/sapi/fpm/fpm/fpm_conf.c	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/sapi/fpm/fpm/fpm_conf.c	2012-05-09 15:45:14.000000000 -0600
@@ -1562,7 +1562,7 @@
 		char *tmp;
 
 		if (fpm_globals.prefix == NULL) {
-			spprintf(&tmp, 0, "%s/php-fpm.conf", PHP_SYSCONFDIR);
+			spprintf(&tmp, 0, "%s/php5/fpm/php-fpm.conf", PHP_SYSCONFDIR);
 		} else {
 			spprintf(&tmp, 0, "%s/etc/php-fpm.conf", fpm_globals.prefix);
 		}
diff -ruN php-5.4.3.orig/scripts/Makefile.frag php-5.4.3/scripts/Makefile.frag
--- php-5.4.3.orig/scripts/Makefile.frag	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/scripts/Makefile.frag	2012-05-09 15:45:14.000000000 -0600
@@ -3,8 +3,8 @@
 # Build environment install
 #
 
-phpincludedir = $(includedir)/php
-phpbuilddir = $(libdir)/build
+phpincludedir = $(includedir)/php5
+phpbuilddir = $(prefix)/lib/php5/build
 
 BUILD_FILES = \
 	scripts/phpize.m4 \
diff -ruN php-5.4.3.orig/scripts/php-config.in php-5.4.3/scripts/php-config.in
--- php-5.4.3.orig/scripts/php-config.in	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/scripts/php-config.in	2012-05-09 15:45:14.000000000 -0600
@@ -6,8 +6,8 @@
 exec_prefix="@exec_prefix@"
 version="@PHP_VERSION@"
 vernum="@PHP_VERSION_ID@"
-include_dir="@includedir@/php"
-includes="-I$include_dir -I$include_dir/main -I$include_dir/TSRM -I$include_dir/Zend -I$include_dir/ext -I$include_dir/ext/date/lib"
+include_dir="@includedir@/php5"
+includes="-I$include_dir -I$include_dir/main -I$include_dir/TSRM -I$include_dir/Zend -I$include_dir/ext -I$include_dir/ext/date/lib $(getconf LFS_CFLAGS)"
 ldflags="@PHP_LDFLAGS@"
 libs="@EXTRA_LIBS@"
 extension_dir='@EXTENSION_DIR@'
diff -ruN php-5.4.3.orig/scripts/phpize.in php-5.4.3/scripts/phpize.in
--- php-5.4.3.orig/scripts/phpize.in	2012-05-07 23:22:56.000000000 -0600
+++ php-5.4.3/scripts/phpize.in	2012-05-09 15:45:14.000000000 -0600
@@ -4,8 +4,8 @@
 prefix='@prefix@'
 datarootdir='@datarootdir@'
 exec_prefix="`eval echo @exec_prefix@`"
-phpdir="`eval echo @libdir@`/build"
-includedir="`eval echo @includedir@`/php"
+phpdir="$prefix/lib/php5/build"
+includedir="$prefix/include/php5"
 builddir="`pwd`"
 SED="@SED@"
 
