diff -Nurd -x'*~' tcllib-1.10.orig/configure tcllib-1.10/configure
--- tcllib-1.10.orig/configure	2005-01-27 13:22:56.000000000 -0500
+++ tcllib-1.10/configure	2008-08-27 12:52:05.000000000 -0400
@@ -583,9 +583,9 @@
 
 # ### ######### ###########################
 
-PACKAGE=`$TCLSH_PROG ${srcdir}/sak.tcl name`
-MAJOR_VERSION=`$TCLSH_PROG ${srcdir}/sak.tcl major`
-MINOR_VERSION=`$TCLSH_PROG ${srcdir}/sak.tcl minor`
+PACKAGE="tcllib"
+MAJOR_VERSION="1"
+MINOR_VERSION="10"
 PATCHLEVEL=""
 
 VERSION=${MAJOR_VERSION}.${MINOR_VERSION}${PATCHLEVEL}
