diff -Nurd -x'*~' LinkChecker-8.0.orig/linkcheck/__init__.py LinkChecker-8.0/linkcheck/__init__.py
--- LinkChecker-8.0.orig/linkcheck/__init__.py	2012-08-11 15:19:17.000000000 -0400
+++ LinkChecker-8.0/linkcheck/__init__.py	2012-09-20 09:17:22.000000000 -0400
@@ -173,9 +173,9 @@
 
 def find_third_party_modules ():
     """Find third party modules and add them to the python path."""
-    parent = os.path.dirname(os.path.dirname(__file__))
+    parent = os.path.dirname(__file__)
     third_party = os.path.join(parent, "third_party")
     if os.path.isdir(third_party):
-        sys.path.append(os.path.join(third_party, "dnspython"))
+        sys.path.insert(0,os.path.join(third_party, "dnspython"))
 
 find_third_party_modules()
diff -Nurd -x'*~' LinkChecker-8.0.orig/linkchecker-gui LinkChecker-8.0/linkchecker-gui
--- LinkChecker-8.0.orig/linkchecker-gui	2012-08-26 03:55:52.000000000 -0400
+++ LinkChecker-8.0/linkchecker-gui	2012-09-20 10:29:48.000000000 -0400
@@ -26,7 +26,14 @@
     import os
     from linkcheck import get_install_data
     sys.path.append(os.path.join(get_install_data(), 'Microsoft.VC90.CRT'))
-from PyQt4.QtGui import QApplication
+sys.path.insert(1,'@PREFIX@/lib/qt4-x11/lib/python2.7/site-packages/')
+try:
+    from PyQt4.QtGui import QApplication
+except ImportError:
+    # warn of likely cause, but also display actual error message
+    print "To use this program, you must install fink's 'pyqt4-py27' package\n"
+    from PyQt4.QtGui import QApplication
+    exit(1)
 from linkcheck.fileutil import is_readable
 from linkcheck.gui import LinkCheckerMain, get_app_style
 from linkcheck.gui.projects import ProjectExt
diff -Nurd -x'*~' LinkChecker-8.0.orig/setup.py LinkChecker-8.0/setup.py
--- LinkChecker-8.0.orig/setup.py	2012-06-29 03:50:06.000000000 -0400
+++ LinkChecker-8.0/setup.py	2012-09-20 09:51:52.000000000 -0400
@@ -868,9 +868,6 @@
         'py2app': MyPy2app,
         'register': MyRegister,
     },
-    package_dir = {
-        'dns': 'third_party/dnspython/dns',
-    },
     packages = [
         'linkcheck',
         'linkcheck.bookmarks',
@@ -883,10 +880,11 @@
         'linkcheck.HtmlParser',
         'linkcheck.logger',
         'linkcheck.network',
-        'dns',
-        'dns.rdtypes',
-        'dns.rdtypes.ANY',
-        'dns.rdtypes.IN',
+        'linkcheck.third_party.dnspython.dns',
+        'linkcheck.third_party.dnspython.dns.rdtypes',
+        'linkcheck.third_party.dnspython.dns.rdtypes.ANY',
+        'linkcheck.third_party.dnspython.dns.rdtypes.IN',
+        'linkcheck.third_party.dnspython.tests',
     ],
     ext_modules = [
         Extension('linkcheck.HtmlParser.htmlsax',
