diff -Nurd -x'*~' webkit-1.1.15.4.orig/GNUmakefile.in webkit-1.1.15.4/GNUmakefile.in
--- webkit-1.1.15.4.orig/GNUmakefile.in	2009-11-27 08:54:41.000000000 -0500
+++ webkit-1.1.15.4/GNUmakefile.in	2010-03-13 13:42:52.000000000 -0500
@@ -8400,7 +8400,7 @@
 	$(GEOCLUE_LIBS) \
 	$(GLIB_LIBS) \
 	$(GSTREAMER_LIBS) \
-	$(GTK_LIBS) \
+	$(GTK_LIBS) -lintl \
 	$(HILDON_LIBS) \
 	$(JPEG_LIBS) \
 	$(LIBSOUP_LIBS) \
@@ -8707,8 +8707,8 @@
 	$(LIBSOUP_CFLAGS)
 
 Programs_DumpRenderTree_LDADD = \
-	libwebkit-1.0.la \
 	libJavaScriptCore.la \
+	libwebkit-1.0.la \
 	$(GLOBALDEPS_LIBS) \
 	$(CAIRO_LIBS) \
 	$(GTK_LIBS) \
@@ -8727,7 +8727,7 @@
 @TARGET_X11_TRUE@	-I$(srcdir)/WebCore/plugins \
 @TARGET_X11_TRUE@	-I$(srcdir)/WebKitTools/DumpRenderTree/TestNetscapePlugIn.subproj \
 @TARGET_X11_TRUE@	$(global_cppflags) \
-@TARGET_X11_TRUE@	$(javascriptcore_cppflags)
+@TARGET_X11_TRUE@	$(javascriptcore_cppflags) -DXP_UNIX
 
 @TARGET_X11_TRUE@TestNetscapePlugin_libtestnetscapeplugin_la_SOURCES = \
 @TARGET_X11_TRUE@	WebKitTools/DumpRenderTree/gtk/TestNetscapePlugin/ForwardingHeaders/WebKit/npapi.h \
diff -Nurd -x'*~' webkit-1.1.15.4.orig/configure webkit-1.1.15.4/configure
--- webkit-1.1.15.4.orig/configure	2009-11-27 08:54:42.000000000 -0500
+++ webkit-1.1.15.4/configure	2010-03-12 02:29:19.000000000 -0500
@@ -16722,7 +16722,7 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking if dolt supports this host" >&5
 $as_echo_n "checking if dolt supports this host... " >&6; }
-dolt_supported=yes
+dolt_supported=no
 if test x$DOLT_BASH = x; then
     dolt_supported=no
 fi
@@ -17726,7 +17726,7 @@
 # keep the ICU dependency. That's why we temporarily add icu headers and libs for glib config case as well.
 if test "$with_unicode_backend" = "icu" -o "$with_unicode_backend" = "glib"; then
         case "$host" in
-            *-*-darwin*)
+            *-*-darwin*-but-not-fink-because-it-has-icu-config)
 		UNICODE_CFLAGS="-I$srcdir/JavaScriptCore/icu -I$srcdir/WebCore/icu"
 		UNICODE_LIBS="-licucore"
                 ;;
@@ -17903,7 +17903,7 @@
 done
 
 
-GETTEXT_PACKAGE=$PACKAGE
+GETTEXT_PACKAGE="${PACKAGE}-1.0.2"
 
 
 cat >>confdefs.h <<_ACEOF
