diff --git a/src/BtLeecherStateChoke.cc b/src/BtLeecherStateChoke.cc
index ad9c264..125657f 100644
--- a/src/BtLeecherStateChoke.cc
+++ b/src/BtLeecherStateChoke.cc
@@ -239,13 +239,3 @@ const Timer& BtLeecherStateChoke::getLastRound() const
 }
 
 } // namespace aria2
-
-namespace std {
-template<>
-void swap<aria2::BtLeecherStateChoke::PeerEntry>
-(aria2::BtLeecherStateChoke::PeerEntry& a,
- aria2::BtLeecherStateChoke::PeerEntry& b)
-{
-  a.swap(b);
-}
-} // namespace std;
diff --git a/src/BtLeecherStateChoke.h b/src/BtLeecherStateChoke.h
index d9e8985..cb1ca6d 100644
--- a/src/BtLeecherStateChoke.h
+++ b/src/BtLeecherStateChoke.h
@@ -118,7 +118,6 @@ public:
   const Timer& getLastRound() const;
 
   friend void swap(PeerEntry& a, PeerEntry& b);
-  friend void std::swap<PeerEntry>(PeerEntry& a, PeerEntry& b);
 };
 
 void swap
@@ -127,11 +126,4 @@ void swap
 
 } // namespace aria2
 
-namespace std {
-template<>
-void swap<aria2::BtLeecherStateChoke::PeerEntry>
-(aria2::BtLeecherStateChoke::PeerEntry& a,
- aria2::BtLeecherStateChoke::PeerEntry& b);
-} // namespace std
-
 #endif // D_BT_LEECHER_STATE_CHOKE_H
diff --git a/src/BtSeederStateChoke.cc b/src/BtSeederStateChoke.cc
index 0867107..c0c8284 100644
--- a/src/BtSeederStateChoke.cc
+++ b/src/BtSeederStateChoke.cc
@@ -194,13 +194,3 @@ void swap
 }
 
 } // namespace aria2
-
-namespace std {
-template<>
-void swap<aria2::BtSeederStateChoke::PeerEntry>
-(aria2::BtSeederStateChoke::PeerEntry& a,
- aria2::BtSeederStateChoke::PeerEntry& b)
-{
-  a.swap(b);
-}
-} // namespace std
diff --git a/src/BtSeederStateChoke.h b/src/BtSeederStateChoke.h
index dacfd46..c87fd55 100644
--- a/src/BtSeederStateChoke.h
+++ b/src/BtSeederStateChoke.h
@@ -103,7 +103,6 @@ public:
   const Timer& getLastRound() const { return lastRound_; }
 
   friend void swap(PeerEntry& a, PeerEntry& b);
-  friend void std::swap<PeerEntry>(PeerEntry& a, PeerEntry& b);
 };
 
 void swap
@@ -112,11 +111,4 @@ void swap
 
 } // namespace aria2
 
-namespace std {
-template<>
-void swap<aria2::BtSeederStateChoke::PeerEntry>
-(aria2::BtSeederStateChoke::PeerEntry& a,
- aria2::BtSeederStateChoke::PeerEntry& b);
-} // namespace std
-
 #endif // D_BT_SEEDER_STATE_CHOKE_H
