diff -uNr avahi-0.6.20/avahi-core/fdutil.c avahi-0.6.20-patched/avahi-core/fdutil.c
--- avahi-0.6.20/avahi-core/fdutil.c	2006-08-31 12:18:03.000000000 -0500
+++ avahi-0.6.20-patched/avahi-core/fdutil.c	2007-08-04 10:46:58.000000000 -0500
@@ -23,6 +23,11 @@
 #include <config.h>
 #endif
 
+#ifdef __APPLE__
+/** hack! **/
+#undef _XOPEN_SOURCE
+#endif
+
 #include <unistd.h>
 #include <fcntl.h>
 #include <assert.h>
diff -uNr avahi-0.6.20/avahi-core/iface-pfroute.c avahi-0.6.20-patched/avahi-core/iface-pfroute.c
--- avahi-0.6.20/avahi-core/iface-pfroute.c	2007-02-05 07:44:19.000000000 -0600
+++ avahi-0.6.20-patched/avahi-core/iface-pfroute.c	2007-08-04 10:48:29.000000000 -0500
@@ -19,6 +19,11 @@
   USA.
 ***/
 
+#ifdef __APPLE__
+/** hack!  **/
+#undef _XOPEN_SOURCE
+#endif
+
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
diff -uNr avahi-0.6.20/avahi-core/socket.c avahi-0.6.20-patched/avahi-core/socket.c
--- avahi-0.6.20/avahi-core/socket.c	2007-01-06 10:10:05.000000000 -0600
+++ avahi-0.6.20-patched/avahi-core/socket.c	2007-08-04 10:45:16.000000000 -0500
@@ -19,6 +19,11 @@
   USA.
 ***/
 
+#ifdef __APPLE__
+/** hack!  **/
+#undef _XOPEN_SOURCE
+#endif
+
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
