diff -Nurd avahi-0.6.25/avahi-discover-standalone/Makefile.in avahi-0.6.25-new/avahi-discover-standalone/Makefile.in
--- avahi-0.6.25/avahi-discover-standalone/Makefile.in	2009-04-13 21:57:14.000000000 -0400
+++ avahi-0.6.25-new/avahi-discover-standalone/Makefile.in	2009-06-12 11:06:48.000000000 -0400
@@ -80,9 +80,7 @@
 avahi_discover_standalone_OBJECTS =  \
 	$(am_avahi_discover_standalone_OBJECTS)
 am__DEPENDENCIES_1 =
-@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@avahi_discover_standalone_DEPENDENCIES = ../avahi-common/libavahi-common.la \
-@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	../avahi-glib/libavahi-glib.la \
-@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	../avahi-core/libavahi-core.la \
+@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@avahi_discover_standalone_DEPENDENCIES = \
 @HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	$(am__DEPENDENCIES_1) \
 @HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	$(am__DEPENDENCIES_1) \
 @HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	$(am__DEPENDENCIES_1)
@@ -360,9 +358,9 @@
 
 @HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@avahi_discover_standalone_LDADD = \
 @HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	$(AM_LDADD) \
-@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	../avahi-common/libavahi-common.la \
-@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	../avahi-glib/libavahi-glib.la \
-@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	../avahi-core/libavahi-core.la  \
+@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	-lavahi-common \
+@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	-lavahi-glib \
+@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	-lavahi-core  \
 @HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	$(GLIB20_LIBS) $(GTK20_LIBS) $(GLADE20_LIBS)
 
 @HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@interfaces_DATA = $(interfaces)
diff -Nurd avahi-0.6.25/avahi-glib/Makefile.in avahi-0.6.25-new/avahi-glib/Makefile.in
--- avahi-0.6.25/avahi-glib/Makefile.in	2009-04-13 21:57:14.000000000 -0400
+++ avahi-0.6.25-new/avahi-glib/Makefile.in	2009-06-12 11:06:48.000000000 -0400
@@ -87,7 +87,7 @@
 LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
 @HAVE_GLIB_TRUE@libavahi_glib_la_DEPENDENCIES =  \
-@HAVE_GLIB_TRUE@	../avahi-common/libavahi-common.la \
+@HAVE_GLIB_TRUE@	\
 @HAVE_GLIB_TRUE@	$(am__DEPENDENCIES_1)
 am__libavahi_glib_la_SOURCES_DIST = glib-watch.c glib-watch.h \
 	glib-malloc.h glib-malloc.c
@@ -107,7 +107,7 @@
 @HAVE_GLIB_TRUE@	glib_watch_test-glib-watch-test.$(OBJEXT)
 glib_watch_test_OBJECTS = $(am_glib_watch_test_OBJECTS)
 @HAVE_GLIB_TRUE@glib_watch_test_DEPENDENCIES =  \
-@HAVE_GLIB_TRUE@	../avahi-common/libavahi-common.la \
+@HAVE_GLIB_TRUE@	-lavahi-common \
 @HAVE_GLIB_TRUE@	$(am__DEPENDENCIES_1)
 glib_watch_test_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(glib_watch_test_CFLAGS) \
@@ -378,14 +378,14 @@
 @HAVE_GLIB_TRUE@	glib-malloc.h glib-malloc.c
 
 @HAVE_GLIB_TRUE@libavahi_glib_la_CFLAGS = $(AM_CFLAGS) $(GLIB20_CFLAGS)
-@HAVE_GLIB_TRUE@libavahi_glib_la_LIBADD = $(AM_LDADD) ../avahi-common/libavahi-common.la $(GLIB20_LIBS)
+@HAVE_GLIB_TRUE@libavahi_glib_la_LIBADD = $(AM_LDADD) -lavahi-common $(GLIB20_LIBS)
 @HAVE_GLIB_TRUE@libavahi_glib_la_LDFLAGS = $(AM_LDFLAGS)  -version-info $(LIBAVAHI_GLIB_VERSION_INFO)
 @HAVE_GLIB_TRUE@glib_watch_test_SOURCES = \
 @HAVE_GLIB_TRUE@	glib-watch.c glib-watch.h \
 @HAVE_GLIB_TRUE@	glib-watch-test.c
 
 @HAVE_GLIB_TRUE@glib_watch_test_CFLAGS = $(AM_CFLAGS) $(GLIB20_CFLAGS)
-@HAVE_GLIB_TRUE@glib_watch_test_LDADD = $(AM_LDADD) ../avahi-common/libavahi-common.la $(GLIB20_LIBS)
+@HAVE_GLIB_TRUE@glib_watch_test_LDADD = $(AM_LDADD) -lavahi-common $(GLIB20_LIBS)
 all: all-am
 
 .SUFFIXES:
diff -Nurd avahi-0.6.25/avahi-gobject/Makefile.in avahi-0.6.25-new/avahi-gobject/Makefile.in
--- avahi-0.6.25/avahi-gobject/Makefile.in	2009-04-13 21:57:14.000000000 -0400
+++ avahi-0.6.25-new/avahi-gobject/Makefile.in	2009-06-12 11:06:48.000000000 -0400
@@ -83,9 +83,7 @@
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
-@HAVE_DBUS_TRUE@@HAVE_GOBJECT_TRUE@libavahi_gobject_la_DEPENDENCIES = ../avahi-common/libavahi-common.la \
-@HAVE_DBUS_TRUE@@HAVE_GOBJECT_TRUE@	../avahi-client/libavahi-client.la \
-@HAVE_DBUS_TRUE@@HAVE_GOBJECT_TRUE@	../avahi-glib/libavahi-glib.la \
+@HAVE_DBUS_TRUE@@HAVE_GOBJECT_TRUE@libavahi_gobject_la_DEPENDENCIES = \
 @HAVE_DBUS_TRUE@@HAVE_GOBJECT_TRUE@	$(am__DEPENDENCIES_1)
 am__libavahi_gobject_la_SOURCES_DIST = ga-client.c ga-client.h \
 	ga-entry-group.c ga-entry-group.h ga-enums.h ga-error.c \
@@ -408,7 +406,7 @@
 @HAVE_DBUS_TRUE@@HAVE_GOBJECT_TRUE@	$(BUILT_SOURCES)
 
 @HAVE_DBUS_TRUE@@HAVE_GOBJECT_TRUE@libavahi_gobject_la_CFLAGS = $(AM_CFLAGS) $(GOBJECT_CFLAGS)
-@HAVE_DBUS_TRUE@@HAVE_GOBJECT_TRUE@libavahi_gobject_la_LIBADD = $(AM_LDADD) ../avahi-common/libavahi-common.la ../avahi-client/libavahi-client.la ../avahi-glib/libavahi-glib.la $(GOBJECT_LIBS)
+@HAVE_DBUS_TRUE@@HAVE_GOBJECT_TRUE@libavahi_gobject_la_LIBADD = $(AM_LDADD) -lavahi-common -lavahi-client -lavahi-glib $(GOBJECT_LIBS)
 @HAVE_DBUS_TRUE@@HAVE_GOBJECT_TRUE@libavahi_gobject_la_LDFLAGS = $(AM_LDFLAGS) -version-info $(LIBAVAHI_GOBJECT_VERSION_INFO) -export-symbols-regex '^ga_'
 
 # correctly clean the generated source files
diff -Nurd avahi-0.6.25/avahi-ui/Makefile.in avahi-0.6.25-new/avahi-ui/Makefile.in
--- avahi-0.6.25/avahi-ui/Makefile.in	2009-04-13 21:57:15.000000000 -0400
+++ avahi-0.6.25-new/avahi-ui/Makefile.in	2009-06-12 11:16:35.000000000 -0400
@@ -94,9 +94,7 @@
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
-@HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@libavahi_ui_la_DEPENDENCIES = ../avahi-common/libavahi-common.la \
-@HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	../avahi-client/libavahi-client.la \
-@HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	../avahi-glib/libavahi-glib.la \
+@HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@libavahi_ui_la_DEPENDENCIES = \
 @HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	$(am__DEPENDENCIES_1) \
 @HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	$(am__DEPENDENCIES_1)
 am__libavahi_ui_la_SOURCES_DIST = avahi-ui.h avahi-ui.c \
@@ -119,8 +117,6 @@
 @HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@am_bssh_OBJECTS = bssh-bssh.$(OBJEXT)
 bssh_OBJECTS = $(am_bssh_OBJECTS)
 @HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@bssh_DEPENDENCIES = $(am__DEPENDENCIES_1) \
-@HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	../avahi-client/libavahi-client.la \
-@HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	../avahi-common/libavahi-common.la \
 @HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	libavahi-ui.la
 bssh_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=link $(CCLD) $(bssh_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) \
@@ -406,15 +402,15 @@
 @HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	$(am__append_6)
 @HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@libavahi_ui_la_LIBADD =  \
 @HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	$(AM_LDADD) \
-@HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	../avahi-common/libavahi-common.la \
-@HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	../avahi-client/libavahi-client.la \
-@HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	../avahi-glib/libavahi-glib.la \
+@HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	-lavahi-common \
+@HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	-lavahi-client \
+@HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	-lavahi-glib \
 @HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	$(GTK20_LIBS) \
 @HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@	$(am__append_4)
 @HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@libavahi_ui_la_LDFLAGS = $(AM_LDFLAGS)  -version-info $(LIBAVAHI_UI_VERSION_INFO)
 @HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@bssh_SOURCES = bssh.c
 @HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@bssh_CFLAGS = $(AM_CFLAGS) $(GTK20_CFLAGS)
-@HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@bssh_LDADD = $(AM_LDADD) $(GTK20_LIBS) ../avahi-client/libavahi-client.la ../avahi-common/libavahi-common.la libavahi-ui.la
+@HAVE_DBUS_TRUE@@HAVE_GLIB_TRUE@@HAVE_GTK_TRUE@bssh_LDADD = $(AM_LDADD) $(GTK20_LIBS) -lavahi-client -lavahi-common libavahi-ui.la
 CLEANFILES = $(desktop_DATA) $(desktop_DATA_in)
 all: all-am
 
