diff -Naur BloGTK-1.0-old/Makefile BloGTK-1.0/Makefile
--- BloGTK-1.0-old/Makefile	Fri Mar 12 15:17:53 2004
+++ BloGTK-1.0/Makefile	Sun Mar 14 08:31:16 2004
@@ -8,7 +8,8 @@
 
 #PREFIX = /usr
 # autodetect GNOME prefix, change this if you want it elsewhere
-PREFIX ?= `pkg-config libgnome-2.0 --variable=prefix || echo /usr`
+#PREFIX ?= `pkg-config libgnome-2.0 --variable=prefix || echo /usr`
+PREFIX = @PREFIX@
 
 LIBDIR = $(PREFIX)/lib/blogtk
 BINDIR = $(PREFIX)/bin
@@ -22,13 +23,9 @@
 	mv src/BloGTK.py src/BloGTK.py.old
 	sed -e "s|/usr/share|$(PREFIX)/share|g" src/BloGTK.py.old > src/BloGTK.py
 	rm src/BloGTK.py.old
-	$(INSTALL) -m 755 -d $(BINDIR) $(LIBDIR) $(DATADIR) $(APPLICATIONSDIR)	$(ICONDIR) 
-	$(INSTALL) -m 755 src/* $(LIBDIR)
-	$(INSTALL) -m 644 pixmaps/* $(DATADIR)
-	$(INSTALL) -m 644 data/blogtk-icon.svgz $(ICONDIR)
-	$(INSTALL) -m 644 data/blogtk.desktop $(APPLICATIONSDIR)
-
-	if test -f /usr/bin/BloGTK; then rm /usr/bin/BloGTK; fi	
-
-	ln -s $(LIBDIR)/BloGTK.py $(BINDIR)/BloGTK
-	chmod +x $(BINDIR)/BloGTK
+	$(INSTALL) -m 755 -d $(DESTDIR)$(BINDIR) $(DESTDIR)$(LIBDIR) $(DESTDIR)$(DATADIR) $(DESTDIR)$(APPLICATIONSDIR) $(DESTDIR)$(ICONDIR) 
+	$(INSTALL) -m 755 src/* $(DESTDIR)$(LIBDIR)
+	$(INSTALL) -m 644 pixmaps/* $(DESTDIR)$(DATADIR)
+	$(INSTALL) -m 644 data/blogtk-icon.svgz $(DESTDIR)$(ICONDIR)
+	$(INSTALL) -m 644 data/blogtk.desktop $(DESTDIR)$(APPLICATIONSDIR)
+	ln -s $(LIBDIR)/BloGTK.py $(DESTDIR)$(BINDIR)/BloGTK
