Only in dhcp-3.0-new: #Makefile.dist#
diff -ru dhcp-3.0/Makefile.conf dhcp-3.0-new/Makefile.conf
--- dhcp-3.0/Makefile.conf	Tue Jun 26 14:31:36 2001
+++ dhcp-3.0-new/Makefile.conf	Mon Jan 28 14:06:10 2002
@@ -19,16 +19,16 @@
 
 ## Defaults...
 SCRIPT = none
-USERBINDIR = /usr/bin
-BINDIR = /usr/sbin
-CLIENTBINDIR=/sbin
+USERBINDIR = @FINK@/bin
+BINDIR = @FINK@/sbin
+CLIENTBINDIR=@FINK@/sbin
-ADMMANDIR = /usr/share/man/cat8
+ADMMANDIR = @FINK@/share/man/man8
 ADMMANEXT = .0
-FFMANDIR = /usr/share/man/cat5
+FFMANDIR = @FINK@/share/man/man5
 FFMANEXT = .0
-LIBMANDIR = /usr/share/man/cat3
+LIBMANDIR = @FINK@/share/man/man3
 LIBMANEXT = .0
-USRMANDIR = /usr/share/man/cat1
+USRMANDIR = @FINK@/share/man/man1
 USRMANEXT = .0
 MANCAT = cat
 INSTALL = install -c -m 444
@@ -36,16 +36,16 @@
 CHMOD = chmod
 CATMANPAGES =
-ETC = /etc
-VARRUN = /var/run
-VARDB = /var/db
-LIBDIR=/usr/local/lib
-INCDIR=/usr/local/include
+ETC = @FINK@/etc
+VARRUN = @FINK@/var/run
+VARDB = @FINK@/var/db
+LIBDIR=@FINK@/lib
+INCDIR=@FINK@/include
 LIBS =
 COPTS = $(BINDDEF) $(CC_OPTIONS)
 DEBUG = -g
 RANLIB = ranlib
 MKDEP = mkdep
-CLIENT_PATH = '"PATH=/usr/ucb:/usr/bin:/usr/sbin:/bin:/sbin"'
+CLIENT_PATH = '"PATH=@FINK@/bin:@FINK@/sbin:/usr/bin:/usr/sbin:/bin:/sbin"'
 
 BINDLIB = ../minires/libres.a
 BINDINC =


