diff -uNr ekg-1.6/configure ekg-1.6-new/configure
--- ekg-1.6/configure	2005-10-29 15:37:31.000000000 -0400
+++ ekg-1.6-new/configure	2008-08-11 15:14:44.000000000 -0400
@@ -4312,7 +4312,7 @@
 		# FIXME: -fPIC is required for -shared on many architectures,
 		# so we specify it here, but the right way would probably be to
 		# properly detect whether it is actually required.
-		CFLAGS="-shared -fPIC -Wl,-z,defs $CFLAGS $PTHREAD_CFLAGS"
+		CFLAGS="-dynamiclib -fPIC $CFLAGS $PTHREAD_CFLAGS"
 		LIBS="$PTHREAD_LIBS $LIBS"
 		CC="$PTHREAD_CC"
 
@@ -7420,7 +7420,7 @@
    { (exit 1); exit 1; }; }
 	fi
 
-	LDFLAGS="$READLINE_LIBS $LDFLAGS"
+	LDFLAGS="$LDFLAGS $READLINE_LIBS"
 	CFLAGS="$CFLAGS $READLINE_INCLUDES"
 
 
@@ -8446,7 +8446,7 @@
 			PY_OTHER_LIBS=`sed -n -e 's/^LIBS=\(.*\)/\1/p' $PY_MAKEFILE`
 			PY_OTHER_LIBM=`sed -n -e 's/^LIBC=\(.*\)/\1/p' $PY_MAKEFILE`
 			PY_OTHER_LIBC=`sed -n -e 's/^LIBM=\(.*\)/\1/p' $PY_MAKEFILE`
-			PY_LINKFORSHARED="$(python -c "import distutils.sysconfig; print distutils.sysconfig.get_config_var('LINKFORSHARED')")"
+			PY_LINKFORSHARED="$($PYTHON -c "import distutils.sysconfig; print distutils.sysconfig.get_config_var('LINKFORSHARED')")"
 			PY_LIBS="$PY_LOCALMODLIBS $PY_BASEMODLIBS $PY_OTHER_LIBS $PY_OTHER_LIBC $PY_OTHER_LIBM $PY_LINKFORSHARED"
 
 			PYTHON_LIBS="-L$PY_EXEC_PREFIX/lib $PY_LIB_LOC -lpython$PY_VERSION $PY_LIBS"
@@ -8469,7 +8469,7 @@
 
 if test "x$have_python" = "xyes"; then
 	OBJS="$OBJS python.o"
-	LDFLAGS="$PYTHON_LIBS $LDFLAGS"
+	LDFLAGS="$LDFLAGS $PYTHON_LIBS"
 	CFLAGS="$CFLAGS $PYTHON_INCLUDES"
 fi
 
diff -uNr ekg-1.6rc3/examples/Makefile.in ekg-1.6rc3-new/examples/Makefile.in
--- ekg-1.6rc3/examples/Makefile.in	2003-06-24 16:56:52.000000000 -0400
+++ ekg-1.6rc3-new/examples/Makefile.in	2005-08-22 14:21:48.000000000 -0400
@@ -1,6 +1,6 @@
 # $Id: Makefile.in,v 1.13 2003/06/24 20:56:52 wojtekka Exp $
 
-CC = @CC@
+CC = glibtool --tag=CC --mode=link @CC@
 CFLAGS = @CFLAGS_LIBGADU@ @CFLAGS@ -I.. -I../lib -I../src
 LIBS = @LIBS_LIBGADU@
 MAKE = @MAKE@
diff -uNr ekg-1.6/lib/Makefile.in ekg-1.6-new/lib/Makefile.in
--- ekg-1.6/lib/Makefile.in	2005-06-25 02:41:07.000000000 -0400
+++ ekg-1.6-new/lib/Makefile.in	2008-08-11 20:22:19.000000000 -0400
@@ -11,7 +11,7 @@
 VERSION_MINOR=3
 
 CC = @CC@ 
-CFLAGS = @CFLAGS_LIBGADU@ -I. -I.. -Wall
+CFLAGS = -I. -I.. @CFLAGS_LIBGADU@ -Wall
 LIBS = @LIBS_LIBGADU@
 
 MAKE = @MAKE@
@@ -25,7 +25,7 @@
 LOBJS = common.lo libgadu.lo pubdir.lo http.lo dcc.lo events.lo pubdir50.lo obsolete.lo
 SRCS = $(OBJS:.o=.c)
 
-SHAREDMM = libgadu.so
+SHAREDMM = libgadu.la
 SHAREDM = $(SHAREDMM).$(VERSION_MAJOR)
 SHARED = $(SHAREDM).$(VERSION_MINOR)
 STATIC = libgadu.a
@@ -51,14 +51,12 @@
 #
 
 %.lo:	%.c
-	$(CC) -c $(CPPFLAGS) $(CFLAGS) -fPIC $< -o $@
+	glibtool --tag=CC --mode=compile $(CC) -c $(CFLAGS) $(CPPFLAGS) -fPIC $< -o $@
 
-shared:	dep $(SHARED)
+shared:	dep $(SHAREDMM)
 
-$(SHARED): $(LOBJS)
-	$(CC) -shared -Wl,-soname,libgadu.so.$(VERSION_MAJOR) -o $(SHARED) $(LOBJS) $(LIBS) -lc
-	$(LN_S) -f $(SHARED) $(SHAREDM)
-	$(LN_S) -f $(SHARED) $(SHAREDMM)
+$(SHAREDMM): $(LOBJS)
+	glibtool --tag=CC --mode=link $(CC) -o $(SHAREDMM) -rpath $(libdir) -version-info $(VERSION_MAJOR):$(VERSION_MINOR) $(LOBJS) $(LIBS) $(LDFLAGS)
 
 #
 
@@ -81,15 +79,12 @@
 	$(INSTALL) -m 644 libgadu.pc $(DESTDIR)$(libdir)/pkgconfig
 
 install-shared:	install-common @strip_libgadu_so@
-	$(INSTALL) -m 755 $(SHARED) $(DESTDIR)$(libdir)
-	cd $(DESTDIR)$(libdir) && $(LN_S) -f $(SHARED) $(SHAREDM)
-	cd $(DESTDIR)$(libdir) && $(LN_S) -f $(SHARED) $(SHAREDMM)
+	glibtool --tag=CC --mode=install $(INSTALL) -m 755 $(SHAREDMM) $(DESTDIR)$(libdir)
 
 install-static:	install-common
 	$(INSTALL) -m 755 $(STATIC) $(DESTDIR)$(libdir)
 
 strip-libgadu-so:
-	$(STRIP) $(SHARED)
 
 #
 
diff -uNr ekg-1.6/src/Makefile.in ekg-1.6-new/src/Makefile.in
--- ekg-1.6/src/Makefile.in	2003-05-19 09:36:41.000000000 -0400
+++ ekg-1.6-new/src/Makefile.in	2008-08-11 20:22:13.000000000 -0400
@@ -25,11 +25,11 @@
 
 all:	dep ekg @ioctld@
 
-ekg:	$(OBJS) @libgadu_a@
-	$(CC) $(CFLAGS) -o ekg $(OBJS) -L../lib @lgadu@ @libgadu_a@ $(LDFLAGS) $(LIBS)
+ekg:	$(OBJS) ../lib/libgadu.la
+	glibtool --tag=CC --mode=link $(CC) $(CFLAGS) -o ekg $(OBJS) ../lib/libgadu.la $(LIBS) $(LDFLAGS)
 
 ioctld:	ioctld.c
-	$(CC) $(CFLAGS) ioctld.c -o ioctld $(LIBS) @IOCTLD_OBJS@
+	glibtool --tag=CC --mode=link $(CC) $(CFLAGS) ioctld.c -o ioctld $(LIBS) $(LDFLAGS) @IOCTLD_OBJS@
 
 dep:	.depend
 
@@ -42,17 +42,15 @@
 
 install-ekg:	@strip_ekg@
 	$(INSTALL) -d $(DESTDIR)$(bindir)
-	$(INSTALL) -m 755 ekg $(DESTDIR)$(bindir)
+	glibtool --tag=CC --mode=install $(INSTALL) -m 755 ekg $(DESTDIR)$(bindir)
 
 install-ioctld:	@strip_ioctld@
 	$(INSTALL) -d $(DESTDIR)$(libexecdir)
-	$(INSTALL) -m 755 ioctld $(DESTDIR)$(libexecdir)
+	glibtool --tag=CC --mode=install $(INSTALL) -m 755 ioctld $(DESTDIR)$(libexecdir)
 
 strip-ekg:
-	$(STRIP) ekg
 
 strip-ioctld:
-	$(STRIP) ioctld
 
 #
 
