diff -Nurd -x'*~' gnet-1.1.9.orig/doc/Makefile.in gnet-1.1.9/doc/Makefile.in
--- gnet-1.1.9.orig/doc/Makefile.in	2003-04-29 13:57:00.000000000 -0400
+++ gnet-1.1.9/doc/Makefile.in	2007-05-17 00:43:56.000000000 -0400
@@ -268,7 +268,7 @@
 
 html:
 	if ! test -d html ; then mkdir html ; fi
-	-cd html && gtkdoc-mkhtml $(DOC_MODULE) ../$(DOC_MAIN_SGML_FILE)
+	cd html && gtkdoc-mkhtml $(DOC_MODULE) ../$(DOC_MAIN_SGML_FILE)
 
 clean-local:
 	rm -f *~ *.bak *.hierarchy *.signals *-unused.txt
@@ -285,12 +285,12 @@
 
 install-data-local: 
 	$(mkinstalldirs) $(DESTDIR)$(TARGET_DIR)
-	-@INSTALL@ -m 0644 $(srcdir)/html/*.html $(DESTDIR)$(TARGET_DIR)
-	-@INSTALL@ -m 0644 $(srcdir)/html/index.sgml $(DESTDIR)$(TARGET_DIR)
-	-(cd $(DESTDIR); gtkdoc-fixxref --module=$(DOC_MODULE) --html-dir=$(HTML_DIR))
-	-mv $(HTML_DIR)/$(DOC_MODULE)/* $(HTML_DIR)
-	-rm -rf $(HTML_DIR)/$(DOC_MODULE)
-	-(cd $(HTML_DIR); ln -s book1.html index.html)
+	@INSTALL@ -m 0644 $(srcdir)/html/*.html $(DESTDIR)$(TARGET_DIR)
+	@INSTALL@ -m 0644 $(srcdir)/html/index.sgml $(DESTDIR)$(TARGET_DIR)
+	(cd $(DESTDIR); gtkdoc-fixxref --module=$(DOC_MODULE) --html-dir=$(DESTDIR)$(HTML_DIR)) || exit 1
+	mv $(DESTDIR)$(HTML_DIR)/$(DOC_MODULE)/* $(DESTDIR)$(HTML_DIR)
+	rm -rf $(DESTDIR)$(HTML_DIR)/$(DOC_MODULE)
+	(cd $(DESTDIR)$(HTML_DIR); ln -s book1.html index.html) || exit 1
 
 dist-hook:
 	(cd $(srcdir) ; tar cvf - html/*.html html/*.sgml) | (cd $(distdir); tar xf -)
