diff -ru2N hotway-0.51/inet.c hotway-0.51-patched/inet.c
--- hotway-0.51/inet.c	Tue Sep 24 07:19:18 2002
+++ hotway-0.51-patched/inet.c	Tue Jan  7 04:41:54 2003
@@ -25,4 +25,5 @@
 #include <sys/socket.h>
 #include <stdlib.h>
+#include <netinet/in.h>
 #include <arpa/inet.h>
 #include <netdb.h>
diff -ru2N hotway-0.51/libghttp-1.0.9-mod/http_base64.c hotway-0.51-patched/libghttp-1.0.9-mod/http_base64.c
--- hotway-0.51/libghttp-1.0.9-mod/http_base64.c	Tue Sep 24 07:35:50 2002
+++ hotway-0.51-patched/libghttp-1.0.9-mod/http_base64.c	Tue Jan  7 07:35:36 2003
@@ -25,8 +25,8 @@
 #include <string.h>
 
-const char b64_alphabet[65] = { 
+const char b64_alphabet[66] = {
     "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
     "abcdefghijklmnopqrstuvwxyz"
-    "0123456789+/=" };
+    "0123456789+/=\0" }; 
     
 char *
