diff -Nurd -x'*~' kismet-3.0.1.orig/configure kismet-3.0.1/configure
--- kismet-3.0.1.orig/configure	2003-08-07 00:24:06.000000000 -0400
+++ kismet-3.0.1/configure	2008-05-03 18:44:23.000000000 -0400
@@ -6785,10 +6785,14 @@
 
 	if test "$magickmajor" -lt 5; then
 		magickold="yes";
-	elif test "$magickmajor" -eq 5 -a "$magickminor" -lt 4; then
-		magickold="yes";
-	elif test "$magickmajor" -eq 5 -a "$magickminor" -eq 4 -a "$magicktiny" -lt 7; then
+	elif test "$magickmajor" -eq 5; then
+	    if test "$magickminor" -lt 4; then
 		magickold="yes";
+	    elif test "$magickminor" -eq 4; then
+		if test "$magicktiny" -lt 7; then
+		    magickold="yes";
+		fi
+	    fi
 	fi
 fi
 
@@ -7091,13 +7095,13 @@
 	fi
 
 	if test "$magickhdr" = "yes"; then
-		echo "$as_me:$LINENO: checking for WriteImage in -lMagick" >&5
-echo $ECHO_N "checking for WriteImage in -lMagick... $ECHO_C" >&6
+		echo "$as_me:$LINENO: checking for WriteImage in -lMagickCore" >&5
+echo $ECHO_N "checking for WriteImage in -lMagickCore... $ECHO_C" >&6
 if test "${ac_cv_lib_Magick_WriteImage+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lMagick  $LIBS"
+LIBS="-lMagickCore  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
diff -Nurd -x'*~' kismet-3.0.1.orig/gpsmap.cc kismet-3.0.1/gpsmap.cc
--- kismet-3.0.1.orig/gpsmap.cc	2003-07-04 00:05:07.000000000 -0400
+++ kismet-3.0.1/gpsmap.cc	2003-10-31 06:06:18.000000000 -0500
@@ -742,11 +742,11 @@
             }
 
             if (in_printstats)
-                printf("Track %d: %d samples.\n", vec, track_vec[vec].size());
+                printf("Track %d: %ld samples.\n", vec, track_vec[vec].size());
         }
     }
 
-    printf("Processing %d networks.\n", bssid_gpsnet_map.size());
+    printf("Processing %ld networks.\n", bssid_gpsnet_map.size());
 
     for (map<string, gps_network *>::const_iterator x = bssid_gpsnet_map.begin();
          x != bssid_gpsnet_map.end(); ++x) {
diff -Nurd -x'*~' kismet-3.0.1.orig/panelfront_display.cc kismet-3.0.1/panelfront_display.cc
--- kismet-3.0.1.orig/panelfront_display.cc	2003-08-05 02:04:06.000000000 -0400
+++ kismet-3.0.1/panelfront_display.cc	2003-10-31 03:51:03.000000000 -0500
@@ -208,7 +208,7 @@
             snprintf(element, 4, "%3d", (idle_time < (decay * 2)) ? net->noise : 0);
             len = 3;
         } else if (colindex == mcol_clients) {
-            snprintf(element, 5, "%4d", net->client_map.size());
+            snprintf(element, 5, "%4ld", net->client_map.size());
             len = 4;
         } else if (colindex == mcol_datasize) {
             if (net->datasize < 1024) // Less than 1k gets raw bytes
@@ -1368,7 +1368,7 @@
     kwin->text.push_back(output);
 
     if (details_network->networks.size() > 1) {
-        snprintf(output, print_width, "Networks: %d", details_network->networks.size());
+        snprintf(output, print_width, "Networks: %ld", details_network->networks.size());
         kwin->text.push_back(output);
 
         if (details_network->virtnet->gps_fixed != -1) {
@@ -1540,7 +1540,7 @@
         snprintf(output, print_width, "Latest  : %.24s", ctime((const time_t *) &dnet->last_time));
         kwin->text.push_back(output);
 
-        snprintf(output, print_width, "Clients : %d", dnet->client_map.size());
+        snprintf(output, print_width, "Clients : %ld", dnet->client_map.size());
         kwin->text.push_back(output);
 
         switch (dnet->type) {
@@ -2429,7 +2429,7 @@
     snprintf(output, print_width, "Start   : %.24s", ctime((const time_t *) &start_time));
     details_text.push_back(output);
 
-    snprintf(output, print_width, "Servers : %d", context_list.size());
+    snprintf(output, print_width, "Servers : %ld", context_list.size());
     details_text.push_back(output);
 
     snprintf(output, print_width, "Networks: %d", num_networks);
diff -Nurd -x'*~' kismet-3.0.1.orig/tcpclient.cc kismet-3.0.1/tcpclient.cc
--- kismet-3.0.1.orig/tcpclient.cc	2003-08-02 14:21:46.000000000 -0400
+++ kismet-3.0.1/tcpclient.cc	2003-10-31 03:46:56.000000000 -0500
@@ -516,7 +516,7 @@
         char alrmstr[2048];
         char atype[128];
         alert_info alrm;
-        if (sscanf(in_data+hdrlen, "%ld %ld %128s \001%2047[^\001]\001\n", &alrm.alert_ts.tv_sec,
+        if (sscanf(in_data+hdrlen, "%d %d %128s \001%2047[^\001]\001\n", &alrm.alert_ts.tv_sec,
                    &alrm.alert_ts.tv_usec, atype, alrmstr) < 3)
             return 0;
         alrm.alert_text = string(atype) + string(" ") + string(alrmstr);
