diff -Nurd libmsn-4.0-beta4/CMakeLists.txt libmsn-4.0-beta4-patched/CMakeLists.txt
--- libmsn-4.0-beta4/CMakeLists.txt	2009-01-04 08:17:18.000000000 -0500
+++ libmsn-4.0-beta4-patched/CMakeLists.txt	2009-03-25 12:11:55.000000000 -0400
@@ -6,6 +6,12 @@
 find_package(OpenSSL REQUIRED)
 
 set (LIB_SUFFIX "" CACHE STRING "Define suffix of directory name (32/64)" )
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH  TRUE)
+set(CMAKE_INSTALL_RPATH "" )
+
+if(APPLE)
+	set(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
+endif(APPLE)
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${OPENSSL_INCLUDE_DIR})
 add_subdirectory(msn)
