diff -ruN linuxdcpp-1.0.1-orig/client/Text.cpp linuxdcpp-1.0.1/client/Text.cpp
--- linuxdcpp-1.0.1-orig/client/Text.cpp	2007-12-07 21:18:31.000000000 -0500
+++ linuxdcpp-1.0.1/client/Text.cpp	2008-04-08 18:09:48.000000000 -0400
@@ -336,7 +336,7 @@
 	char *outbuf = (char *)tmp.data();
 
 	while(inleft > 0) {
-		rv = iconv(cd, (ICONV_CONST char **)&inbuf, &inleft, &outbuf, &outleft);
+		rv = iconv(cd, &inbuf, &inleft, &outbuf, &outleft);
 		if(rv == (size_t)-1) {
 			size_t used = outbuf - tmp.data();
 			if(errno == E2BIG) {
