--- a/Makefile.rules.orig	2008-11-17 18:27:03.000000000 +0000
+++ b/Makefile.rules	2008-11-17 18:27:33.000000000 +0000
@@ -123,7 +123,7 @@
 .PHONY: install-findlib
 install-findlib: META
 	files=`$(COLLECT_FILES) *.mli *.cmi *.cma *.cmxa *.a dll* META $(INSTALL_EXTRA)` && \
-	$(OCAMLFIND) install $(PKGNAME) $$files
+	$(OCAMLFIND) install -ldconf ignore $(PKGNAME) $$files
 
 .PHONY: uninstall-findlib
 uninstall-findlib:
