diff -ruN oidentd-2.0.7.orig/configure oidentd-2.0.7/configure
--- oidentd-2.0.7.orig/configure	Fri Jul 11 09:50:57 2003
+++ oidentd-2.0.7/configure	Mon Jul 21 14:34:07 2003
@@ -5831,7 +5831,7 @@
 	;;
 
 	*darwin* )
-		CFLAGS="-traditional-cpp $CFLAGS"
+		CFLAGS="-no-cpp-precomp $CFLAGS"
 		use_kmem=yes
 		os_src=darwin.c
 		if test "$ac_cv_ipf_nat_p" = "no"; then
diff -ruN oidentd-2.0.7.orig/fink/oidentd.conf oidentd-2.0.7/fink/oidentd.conf
--- oidentd-2.0.7.orig/fink/oidentd.conf	Wed Dec 31 17:00:00 1969
+++ oidentd-2.0.7/fink/oidentd.conf	Mon Jul 21 14:33:50 2003
@@ -0,0 +1,22 @@
+# Configuration for oidentd
+# see oidentd.conf(5)
+# 
+default {
+	default {
+		deny spoof
+		deny spoof_all
+		deny spoof_privport
+		allow random
+		allow random_numeric
+		allow numeric
+		deny hide
+	}
+}
+
+# you may want to hide root connections
+#user "root" {
+#	default {
+#		force reply "UNKNOWN"
+#	}
+#}
+
diff -ruN oidentd-2.0.7.orig/src/kernel/darwin.c oidentd-2.0.7/src/kernel/darwin.c
--- oidentd-2.0.7.orig/src/kernel/darwin.c	Thu Feb 13 21:33:32 2003
+++ oidentd-2.0.7/src/kernel/darwin.c	Mon Jul 21 14:32:31 2003
@@ -46,9 +46,9 @@
 #undef KERNEL
 
 #ifdef WANT_IPV6
-#	include <netinet/ip_var.h>
+/* #	include <netinet/ip_var.h> */
 #	include <netinet/tcp_timer.h>
-#	include <netinet/tcp_var.h>
+/* #	include <netinet/tcp_var.h> */
 #	include <netinet/ip6.h>
 #	ifdef HAVE_NETINET6_IN6_PCB_H
 #		include <netinet6/in6_pcb.h>
