diff -uNr opennms-1.3.2-1/source/maven-settings.xml opennms-1.3.2-1-new/source/maven-settings.xml
--- opennms-1.3.2-1/source/maven-settings.xml	1969-12-31 19:00:00.000000000 -0500
+++ opennms-1.3.2-1-new/source/maven-settings.xml	2007-05-19 12:44:23.000000000 -0400
@@ -0,0 +1,5 @@
+<settings>
+ <localRepository>@REPOROOT@</localRepository>
+ <interactiveMode>false</interactiveMode>
+ <offline>true</offline>
+</settings>
diff -uNr opennms-1.3.2-1/source/opennms-daemon/pom.xml opennms-1.3.2-1-new/source/opennms-daemon/pom.xml
--- opennms-1.3.2-1/source/opennms-daemon/pom.xml	2007-01-06 00:08:42.000000000 -0500
+++ opennms-1.3.2-1-new/source/opennms-daemon/pom.xml	2007-05-19 12:45:28.000000000 -0400
@@ -61,13 +61,6 @@
     </dependency>
     <dependency>
       <groupId>org.opennms</groupId>
-      <artifactId>opennms-iplike-${platform}</artifactId>
-      <type>${platform.shlib}</type>
-      <version>${project.version}</version>
-      <scope>compile</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.opennms</groupId>
       <artifactId>opennms-icmp-jni-${platform}</artifactId>
       <type>${platform.jni}</type>
       <scope>compile</scope>
diff -uNr opennms-1.3.2-1/source/opennms-daemon/src/assembly/daemon.xml opennms-1.3.2-1-new/source/opennms-daemon/src/assembly/daemon.xml
--- opennms-1.3.2-1/source/opennms-daemon/src/assembly/daemon.xml	2006-10-17 15:16:06.000000000 -0400
+++ opennms-1.3.2-1-new/source/opennms-daemon/src/assembly/daemon.xml	2007-05-19 12:45:50.000000000 -0400
@@ -69,17 +69,6 @@
        </includes>
     </dependencySet>
     <dependencySet>
-       <outputFileNameMapping>iplike.${extension}</outputFileNameMapping>
-       <outputDirectory>lib</outputDirectory>
-       <!-- all the platforms need to be listed until the IncludesArtifactFiler supports wildcards -->
-       <includes>
-         <include>org.opennms:opennms-iplike-macosx</include>
-         <include>org.opennms:opennms-iplike-linux</include>
-         <include>org.opennms:opennms-iplike-freebsd</include>
-         <include>org.opennms:opennms-iplike-solaris</include>
-       </includes>
-    </dependencySet>
-    <dependencySet>
       <outputDirectory>lib</outputDirectory>
       <excludes>
          <exclude>org.opennms:opennms-bootstrap</exclude>
diff -uNr opennms-1.3.2-1/source/pom.xml opennms-1.3.2-1-new/source/pom.xml
--- opennms-1.3.2-1/source/pom.xml	2007-01-09 22:04:27.000000000 -0500
+++ opennms-1.3.2-1-new/source/pom.xml	2007-05-19 12:46:10.000000000 -0400
@@ -85,7 +85,6 @@
     <module>opennms-dao</module>
     <module>opennms-icmp</module>
     <module>opennms-rrd</module>
-    <module>opennms-iplike</module>
     <module>opennms-snmp</module>
     <module>opennms-config</module>
     <module>opennms-services</module>
@@ -933,11 +932,6 @@
       </dependency>
       <dependency>
         <groupId>org.opennms</groupId>
-        <artifactId>opennms-iplike</artifactId>
-        <version>${project.version}</version>
-      </dependency>
-      <dependency>
-        <groupId>org.opennms</groupId>
         <artifactId>opennms-joesnmp</artifactId>
         <version>${project.version}</version>
       </dependency>
diff -uNr opennms-1.3.2-1/source/tools/packages/fink/README.Darwin opennms-1.3.2-1-new/source/tools/packages/fink/README.Darwin
--- opennms-1.3.2-1/source/tools/packages/fink/README.Darwin	2005-03-22 00:48:57.000000000 -0500
+++ opennms-1.3.2-1-new/source/tools/packages/fink/README.Darwin	2007-05-19 12:48:44.000000000 -0400
@@ -3,17 +3,15 @@
 ==============================================================================
 
 ------------------------------------------------------------------------------
-System Limits -- Mac OS X 10.2 and Lower
+System Limits -- Mac OS X 10.3
 ------------------------------------------------------------------------------
 
-On MacOSX 10.2 and lower, to be able to give PostgreSQL enough buffers and
-such to run properly, you will need to change the shared memory buffers in
-the startup item:
-
-  /System/Library/StartupItems/SystemTuning/SystemTuning
+On MacOSX 10.3, to be able to give PostgreSQL enough buffers and such to
+run properly, you will need to change the shared memory buffers in the
+/etc/rc file.
 
 On our test system we multiplied the supplied values by 4 and all seems to
-run OK.  Just look for the sysctl lines near the bottom and set them to:
+run OK.  Just look for the sysctl lines and set them to:
 
   sysctl -w kern.maxfiles=65536
   sysctl -w kern.sysv.shmmax=16777216
@@ -22,47 +20,58 @@
   sysctl -w kern.sysv.shmseg=32
   sysctl -w kern.sysv.shmall=4096
 
-Be warned, these settings can get changed back on OS upgrade!
+Your mileage may vary, you may need to raise them higher, depending on how
+large your network is.
 
 ------------------------------------------------------------------------------
-System Limits -- Mac OS X 10.3 and Higher
+System Limits -- Mac OS X 10.4 and Higher
 ------------------------------------------------------------------------------
 
-On recent versions of Mac OS X, these settings are now in:
-
-  /etc/rc
+On recent versions of Mac OS X, these settings are now properly read from
+/etc/sysctl.conf.  Create an /etc/sysctl.conf file with the following lines:
 
-You should probably do something similar to the limits above (multiply by 4
-and reboot) for OpenNMS to run properly, although it seems to work out of the
-box on a reasonably full class-C network with recent Fink packages.  Your
-mileage may vary, you may need to raise them higher, depending on how large
-your network is.
+  kern.maxfiles=65536
+  kern.sysv.shmmax=16777216
+  kern.sysv.shmmin=1
+  kern.sysv.shmmni=128
+  kern.sysv.shmseg=32
+  kern.sysv.shmall=4096
 
 ------------------------------------------------------------------------------
 PostgreSQL Limits -- All Platforms
 ------------------------------------------------------------------------------
 
-Once you do that, you will need to edit
-@PREFIX@/var/postgresql-7.4/data/postgresql.conf (as root, or postgres), and
+Once you do that, you will want to edit
+@PREFIX@/var/postgresql-8.2/data/postgresql.conf (as root, or postgres), and
 make sure that the following values are set:
 
-  tcpip_socket    = true
-  max_connections = 192
-  shared_buffers  = 384
-
-Again, on a reasonably small network, the defaults from Fink seem to work
-alright now.
+  listen_addresses = 'localhost'
+  max_connections  = 192
+  shared_buffers   = 3072kB
+
+Or, if you're using an older PostgreSQL, you want:
+
+  tcpip_socket     = true
+  max_connections  = 192
+  shared_buffers   = 384
+
+On a reasonably small network, the defaults from Fink's PostgreSQL for
+max_connections and shared_buffers seem to work alright, but your mileage
+may vary.
 
 ------------------------------------------------------------------------------
 Database Setup -- All Platforms
 ------------------------------------------------------------------------------
 
 Make sure PostgreSQL is started before you install.  If it isn't, you will
-have to start it ("sudo @PREFIX@/bin/pgsql.sh start") and try again.  To run
-the installer manually, run:
+have to start it ("sudo @PREFIX@/bin/pgsql.sh start") and try again.  To
+finish the installation, run:
 
   sudo @PREFIX@/var/opennms/bin/install -disU
 
+Note that the "opennms-pgsqlXX" package must match the version of PostgreSQL
+you have installed through fink, or this will fail!
+
 ------------------------------------------------------------------------------
 Startup
 ------------------------------------------------------------------------------
@@ -73,5 +82,5 @@
 
   sudo @PREFIX@/bin/pgsql.sh start
   sudo @PREFIX@/bin/opennms start
-  sudo @PREFIX@/bin/tomcat4 start
+  sudo @PREFIX@/bin/tomcat5 start
 
