diff -uNbr opennap-0.44/config.cache opennap-0.44-new/config.cache
--- opennap-0.44/config.cache	1969-12-31 19:00:00.000000000 -0500
+++ opennap-0.44-new/config.cache	2002-07-28 18:47:52.000000000 -0500
@@ -0,0 +1 @@
+ac_cv_func_mlockall=${ac_cv_func_mlockall=no}
diff -uNbr opennap-0.44/init.c opennap-0.44-new/init.c
--- opennap-0.44/init.c	2001-09-23 18:09:04.000000000 -0500
+++ opennap-0.44-new/init.c	2002-07-28 18:48:00.000000000 -0500
@@ -9,10 +9,10 @@
 #endif
 
 #ifndef WIN32
+#include <sys/types.h>
 #include <grp.h>
 #include <pwd.h>
 #include <unistd.h>
-#include <sys/types.h>
 #include <netdb.h>
 #include <limits.h>
 #endif /* !WIN32 */
