diff -ru wireshark-0.99.6/configure wireshark-0.99.6-patched/configure
--- wireshark-0.99.6/configure	2007-07-05 21:26:25.000000000 +0200
+++ wireshark-0.99.6-patched/configure	2007-07-08 19:21:08.000000000 +0200
@@ -20149,7 +20149,7 @@
        LIBS="$ac_save_LIBS"
   fi
 
-  if test "x$no_libgnutls" = x ; then
+  if test "x$no_libgnutls" = xXXXX ; then
      { echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
 
@@ -20388,7 +20388,7 @@
         fi
      fi
   fi
-  if test $ok = yes; then
+  if test $ok = XXXX; then
     LIBGCRYPT_CFLAGS=`$LIBGCRYPT_CONFIG --cflags`
     LIBGCRYPT_LIBS=`$LIBGCRYPT_CONFIG --libs`
 
@@ -33019,7 +33019,7 @@
 #
 # The plugin dissectors reside in ./plugins/PROTO/
 #
-PLUGIN_LIBS="-L../../epan -lwireshark $GLIB_LIBS"
+PLUGIN_LIBS="../../epan/libwireshark.la $GLIB_LIBS"
 
 
 #
