diff -Nurd -x'*~' PodToHTML-0.08.orig/t/script.t PodToHTML-0.08/t/script.t
--- PodToHTML-0.08.orig/t/script.t	2008-07-27 17:55:23.000000000 -0400
+++ PodToHTML-0.08/t/script.t	2010-08-09 06:39:23.000000000 -0400
@@ -1,11 +1,27 @@
 #!perl 
 use Test;                              
+use Config;
 plan test => 1;
 mkdir("t/html",0777) unless -d "t/html";
-my $code = system($^X,"-Mblib","podtohtml",'-q',
+my @command;
+if ($] >= 5.010 && $Config{"byteorder"} =~ /^1/) {
+    # little-endian means intel (not powerpc), for which fink has
+    # multiple single-arch distros that are seen by a single (fat)
+    # perl interp from apple for a perl version supported by fink
+    # (i.e., starting with 5.10.0). This is the logic taken from
+    # Fink::PkgVersion as of fink-0.29.13.
+    push @command, '/usr/bin/arch';
+    if ( $Config{"longsize"} == 4 ) {
+        push @command, '-i386';   # 32-bit
+    } else {
+        push @command, '-x86_64'; # 64-bit
+    }
+}
+push @command, $^X;
+my $code = system(@command,"-Mblib","podtohtml",'-q',
              '-d' => "t/html",-i => "t/html/index.html",
              "podtohtml","Pod");
 ok($code,0,"Error from script");
 
 unlink glob( "t/html/*.html" );
-rmdir "t/html";
\ No newline at end of file
+rmdir "t/html";
