diff -Nurd -x'*~' threads-1.86.orig/t/test.pl threads-1.86/t/test.pl
--- threads-1.86.orig/t/test.pl	2011-12-14 16:44:44.000000000 -0500
+++ threads-1.86/t/test.pl	2012-08-09 16:17:16.000000000 -0400
@@ -716,7 +716,11 @@
 	    $Perl = $Perl . $exe;
 	}
 
-	warn "which_perl: cannot find $Perl from $^X" unless -f $Perl;
+	# The perl interp could be multiword (due to wrapper) or
+	# simple name to be searched in PATH, so "-f" could fail.
+	# Tests look at stderr, so silence this "not a symptom of
+	# anything that matters" warning...
+	#warn "which_perl: cannot find $Perl from $^X" unless -f $Perl;
 
 	# For subcommands to use.
 	$ENV{PERLEXE} = $Perl;
