diff -Nurd -x'*~' Variable-Magic-0.50.orig/t/17-ctl.t Variable-Magic-0.50/t/17-ctl.t
--- Variable-Magic-0.50.orig/t/17-ctl.t	2012-06-24 18:48:51.000000000 -0400
+++ Variable-Magic-0.50/t/17-ctl.t	2012-07-06 00:32:18.000000000 -0400
@@ -2,6 +2,7 @@
 
 use strict;
 use warnings;
+use Config;
 
 use Test::More tests => 4 * 8 + 10 + 1 + 1;
 
@@ -221,7 +222,22 @@
  $ENV{SystemRoot} = $SystemRoot if $^O eq 'MSWin32' and defined $SystemRoot;
  $ENV{PATH}       = $PATH       if $^O eq 'cygwin'  and defined $PATH;
 
- system { $^X } $^X, '-T', map("-I$_", @INC), '-e', $code;
+ my @command;
+ if ($] >= 5.010 && $Config{"byteorder"} =~ /^1/) {
+     # little-endian means intel (not powerpc), for which fink has
+     # multiple single-arch distros that are seen by a single (fat)
+     # perl interp from apple for a perl version supported by fink
+     # (i.e., starting with 5.10.0). This is the logic taken from
+     # Fink::PkgVersion as of fink-0.29.13.
+     push @command, '/usr/bin/arch';
+     if ( $Config{"longsize"} == 4 ) {
+	 push @command, '-i386';   # 32-bit
+     } else {
+	 push @command, '-x86_64'; # 64-bit
+     }
+ }
+ push @command, $^X;
+ system { $command[0] } @command, '-T', map("-I$_", @INC), '-e', $code;
 }
 
 my $has_capture_tiny = do {
