diff -u -Naur bash-3.0/Makefile.in bash-3.0.fink/Makefile.in
--- bash-3.0/Makefile.in	Wed Mar 17 08:34:39 2004
+++ bash-3.0.fink/Makefile.in	Fri Nov 19 14:28:02 2004
@@ -376,7 +376,7 @@
 		 $(BASHINCDIR)/shtty.h $(BASHINCDIR)/typemax.h \
 		 $(BASHINCDIR)/ocache.h
 
-LIBRARIES = $(SHLIB_LIB) $(READLINE_LIB) $(HISTORY_LIB) $(TERMCAP_LIB) $(GLOB_LIB) \
+LIBRARIES = $(SHLIB_LIB) ./lib/readline/libreadline.a ./lib/readline/libhistory.a $(TERMCAP_LIB) $(GLOB_LIB) \
 	    $(TILDE_LIB) $(MALLOC_LIB) $(INTL_LIB) $(LOCAL_LIBS)
 
 LIBDEP = $(SHLIB_DEP) $(INTL_DEP) $(READLINE_DEP) $(HISTORY_DEP) $(TERMCAP_DEP) $(GLOB_DEP) \
diff -u -Naur bash-3.0/README.Fink bash-3.0.fink/README.Fink
--- bash-3.0/README.Fink	Wed Dec 31 19:00:00 1969
+++ bash-3.0.fink/README.Fink	Fri Nov 19 14:57:47 2004
@@ -0,0 +1,32 @@
+This installation of bash has been set up to use an init system similar
+to the tcsh installed by OS X.
+
+In order to use this configuration:
+
+	echo "source ~/.bashrc"                  > ~/.bash_profile
+	echo "source @PREFIX@/share/init/bash/rc"     > ~/.bashrc
+	echo "source @PREFIX@/bin/init.sh"            >> ~/.bashrc
+
+In order to customize this setup:
+
+	mkdir ~/Library/init/bash
+
+and create the following files there as necessary:
+
+	aliases.mine		- shell aliases
+	completions.mine	- completions
+	environment.mine	- environment
+	rc.mine			- run commands
+	path			- command search path
+
+See the corresponding file in @PREFIX@/share/init/bash for more
+information about the role of each file. You can easily extend or
+override the configuration provided by the default file. For example,
+you can add more aliases by adding the appropriate commands in
+aliases.mine.
+
+The init files were originally written by Wilfredo Sanchez, but they contained
+a few errors which I've taken the liberty of correcting.
+
+-Christian Swinehart <cswinehart@users.sourceforge.net>
+ 16 Jan 2002
diff -u -Naur bash-3.0/doc/Makefile.in bash-3.0.fink/doc/Makefile.in
--- bash-3.0/doc/Makefile.in	Tue Jul 27 08:57:48 2004
+++ bash-3.0.fink/doc/Makefile.in	Fri Nov 19 15:17:46 2004
@@ -228,7 +228,7 @@
 	-$(INSTALL_DATA) $(srcdir)/bash.info $(DESTDIR)$(infodir)/bash.info
 # run install-info if it is present to update the info directory
 	if $(SHELL) -c 'install-info --version' >/dev/null 2>&1; then \
-		install-info --dir-file=$(DESTDIR)$(infodir)/dir $(DESTDIR)$(infodir)/bash.info; \
+		echo "Not installing info now, will wait until package is installed." \
 	else true; fi
 # if htmldir is set, install the html files into that directory
 	-if test -n "${htmldir}" ; then \
diff -u -Naur bash-3.0/doc/bash.1 bash-3.0.fink/doc/bash.1
--- bash-3.0/doc/bash.1	Mon Jul 12 11:27:08 2004
+++ bash-3.0.fink/doc/bash.1	Fri Nov 19 14:57:47 2004
@@ -197,7 +197,7 @@
 .TP
 .B \-\-noprofile
 Do not read either the system-wide startup file
-.FN /etc/profile
+.FN @PREFIX@/etc/profile
 or any of the personal initialization files
 .IR ~/.bash_profile ,
 .IR ~/.bash_login ,
@@ -303,7 +303,7 @@
 .B bash
 is invoked as an interactive login shell, or as a non-interactive shell
 with the \fB\-\-login\fP option, it first reads and
-executes commands from the file \fI/etc/profile\fP, if that
+executes commands from the file \fI@PREFIX@/etc/profile\fP, if that
 file exists.
 After reading that file, it looks for \fI~/.bash_profile\fP,
 \fI~/.bash_login\fP, and \fI~/.profile\fP, in that order, and reads
@@ -359,7 +359,7 @@
 When invoked as an interactive login shell, or a non-interactive
 shell with the \fB\-\-login\fP option, it first attempts to
 read and execute commands from
-.I /etc/profile
+.I @PREFIX@/etc/profile
 and
 .IR ~/.profile ,
 in that order.
@@ -8636,10 +8636,10 @@
 .SH FILES
 .PD 0
 .TP
-.FN /bin/bash
+.FN @PREFIX@/bin/bash
 The \fBbash\fP executable
 .TP
-.FN /etc/profile
+.FN @PREFIX@/etc/profile
 The systemwide initialization file, executed for login shells
 .TP
 .FN ~/.bash_profile
diff -u -Naur bash-3.0/doc/bashref.texi bash-3.0.fink/doc/bashref.texi
--- bash-3.0/doc/bashref.texi	Sat Jun 26 14:26:07 2004
+++ bash-3.0.fink/doc/bashref.texi	Fri Nov 19 14:57:47 2004
@@ -4806,7 +4806,7 @@
 to read  command lines when the shell is interactive.
 
 @item --noprofile
-Don't load the system-wide startup file @file{/etc/profile}
+Don't load the system-wide startup file @file{@PREFIX@/etc/profile}
 or any of the personal initialization files
 @file{~/.bash_profile}, @file{~/.bash_login}, or @file{~/.profile}
 when Bash is invoked as a login shell.
@@ -4931,7 +4931,7 @@
 
 When Bash is invoked as an interactive login shell, or as a
 non-interactive shell with the @option{--login} option, it first reads and
-executes commands from the file @file{/etc/profile}, if that file exists.
+executes commands from the file @file{@PREFIX@/etc/profile}, if that file exists.
 After reading that file, it looks for @file{~/.bash_profile},
 @file{~/.bash_login}, and @file{~/.profile}, in that order, and reads
 and executes commands from the first one that exists and is readable.
@@ -4982,7 +4982,7 @@
 
 When invoked as an interactive login shell, or as a non-interactive
 shell with the @option{--login} option, it first attempts to read
-and execute commands from @file{/etc/profile} and @file{~/.profile}, in
+and execute commands from @file{@PREFIX@/etc/profile} and @file{~/.profile}, in
 that order.
 The @option{--noprofile} option may be used to inhibit this behavior.
 When invoked as an interactive shell with the name @code{sh}, Bash
diff -u -Naur bash-3.0/doc/rbash.1 bash-3.0.fink/doc/rbash.1
--- bash-3.0/doc/rbash.1	Mon May 24 10:19:39 2004
+++ bash-3.0.fink/doc/rbash.1	Fri Nov 19 14:57:47 2004
@@ -3,6 +3,6 @@
 rbash \- restricted bash, see \fBbash\fR(1)
 .SH RESTRICTED SHELL
 .nr zY 1
-.so bash.1
+.so man1/bash.1
 .SH SEE ALSO
 bash(1)
diff -u -Naur bash-3.0/init/README bash-3.0.fink/init/README
--- bash-3.0/init/README	Wed Dec 31 19:00:00 1969
+++ bash-3.0.fink/init/README	Fri Nov 19 14:57:47 2004
@@ -0,0 +1,28 @@
+This directory contains some useful bash files.
+
+In order to use this configuration:
+
+	echo "source ~/.bashrc"                  > ~/.bash_profile
+	echo "source @PREFIX@/share/init/bash/rc"     > ~/.bashrc
+	echo "source @PREFIX@/bin/init.sh"            >> ~/.bashrc
+
+In order to customize this setup:
+
+	mkdir ~/Library/init/bash
+
+and create the following files there as necessary:
+
+	aliases.mine		- shell aliases
+	completions.mine	- completions
+	environment.mine	- environment
+	rc.mine			- run commands
+	path			- command search path
+
+See the corresponding file in @PREFIX@/share/init/bash for more
+information about the role of each file. You can easily extend or
+override the configuration provided by the default file. For example,
+you can add more aliases by adding the appropriate commands in
+aliases.mine.
+
+	-christian
+	 cds@users.sourceforge.net
diff -u -Naur bash-3.0/init/aliases bash-3.0.fink/init/aliases
--- bash-3.0/init/aliases	Wed Dec 31 19:00:00 1969
+++ bash-3.0.fink/init/aliases	Fri Nov 19 14:57:47 2004
@@ -0,0 +1,39 @@
+##
+# Bash aliases file
+#
+# Wilfredo Sanchez Jr. | tritan@mit.edu
+##
+
+##
+# Aliases
+##
+
+#alias         .='cwd'
+alias        ..='cd ..'
+alias      cd..='cd ..'
+alias      cdwd='cd $(/bin/pwd)'
+alias       cwd='echo $PWD'
+alias         l='ls -lg'
+
+##
+# Functions
+##
+
+function files     () { find ${1} -type f -print ; }
+function ff        () { find . -name ${1} -print ; }
+function ll        () { ls -lag "$@" | more ; }
+function word      () { fgrep -i "$*" /usr/dict/web2 ; }
+function wordcount () { cat "${1}" | tr -s ' 	.,;:?\!()[]"' '\012' | \
+	       		cat -n | tail -1 | awk '{print $1}' ; }
+
+##
+# Read user's aliases
+##
+if [ -r ${bash_initdir}/aliases.mine ]; then
+  source ${bash_initdir}/aliases.mine
+fi
+
+# Local Variables:
+# mode:shell-script
+# sh-shell:bash
+# End:
diff -u -Naur bash-3.0/init/bash.defaults bash-3.0.fink/init/bash.defaults
--- bash-3.0/init/bash.defaults	Wed Dec 31 19:00:00 1969
+++ bash-3.0.fink/init/bash.defaults	Fri Nov 19 14:57:47 2004
@@ -0,0 +1,27 @@
+##
+# Bash
+# User preferences file
+# Override these in rc.mine
+#
+# Wilfredo Sanchez Jr. | tritan@mit.edu
+# July 09, 1992
+#
+# MIT Project Athena
+##
+
+if [ -n "$PS1" ]; then
+
+  # Prompts
+  PS1='[\h:\w] \u\$ '
+  PS2=' -> '
+  #PS3=
+  #PS4=
+
+  set -o emacs
+
+fi
+
+# Local Variables:
+# mode:shell-script
+# sh-shell:bash
+# End:
diff -u -Naur bash-3.0/init/completions bash-3.0.fink/init/completions
--- bash-3.0/init/completions	Wed Dec 31 19:00:00 1969
+++ bash-3.0.fink/init/completions	Fri Nov 19 14:57:47 2004
@@ -0,0 +1,17 @@
+##
+# Add completions for the fink command
+##
+complete -W "--help --version apropos configure describe install list selfupdate selfupdate-cvs remove update-all" fink
+
+##
+# Read user's completions
+##
+if [ -r ${bash_initdir}/completions.mine ]; then
+  source ${bash_initdir}/completions.mine
+fi
+
+
+# Local Variables:
+# mode:shell-script
+# sh-shell:bash
+# End:
diff -u -Naur bash-3.0/init/environment bash-3.0.fink/init/environment
--- bash-3.0/init/environment	Wed Dec 31 19:00:00 1969
+++ bash-3.0.fink/init/environment	Fri Nov 19 14:57:47 2004
@@ -0,0 +1,29 @@
+##
+# Bourne Again Shell environment file
+# Global environment setup
+#
+# Wilfredo Sanchez Jr. | tritan@mit.edu
+# July 09, 1992
+#
+# MIT Project Athena
+#
+# ORIGINAL SOURCES: /usr/athena/lib/init/cshrc (ATHENA REL 7.3P)
+##
+
+export ENV_SET="YES"		# avoid repeat
+
+# File creation mask
+umask 022			# all files created are -rw-r--r--
+
+##
+# Load user environment
+##
+
+if [ -f ${bash_initdir}/environment.mine ]; then
+    source ${bash_initdir}/environment.mine
+fi
+
+# Local Variables:
+# mode:shell-script
+# sh-shell:bash
+# End:
diff -u -Naur bash-3.0/init/logout bash-3.0.fink/init/logout
--- bash-3.0/init/logout	Wed Dec 31 19:00:00 1969
+++ bash-3.0.fink/init/logout	Fri Nov 19 14:57:47 2004
@@ -0,0 +1,15 @@
+##
+# Destroy credentials
+##
+
+if [ -z "${TERM_PROGRAM}" ]; then
+  # Don't run these commands if the shell is launched by Terminal,
+  # even if it's a login shell.
+
+  if klist -s; then kdestroy; fi
+fi
+
+# Local Variables:
+# mode:shell-script
+# sh-shell:bash
+# End:
diff -u -Naur bash-3.0/init/profile bash-3.0.fink/init/profile
--- bash-3.0/init/profile	Wed Dec 31 19:00:00 1969
+++ bash-3.0.fink/init/profile	Fri Nov 19 14:57:47 2004
@@ -0,0 +1,31 @@
+##
+# Set path
+##
+
+export PATH="${HOME}/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
+export MANPATH="${HOME}/man:/usr/local/share/man:/usr/share/man"
+
+##
+# Read user's .profile
+##
+     default_initdir=@PREFIX@/share/init
+default_bash_initdir=${default_initdir}/bash
+        user_initdir=~/Library/init
+   user_bash_initdir=${user_initdir}/bash
+
+if [ -r ${user_bash_initdir} ]; then
+       initdir=${user_initdir}
+  bash_initdir=${user_bash_initdir}
+else
+       initdir=${default_initdir}
+  bash_initdir=${default_bash_initdir}
+fi
+
+if [ -r ${bash_initdir}/profile.mine ]; then
+  source ${bash_initdir}/profile.mine
+fi
+
+# Local Variables:
+# mode:shell-script
+# sh-shell:bash
+# End:
diff -u -Naur bash-3.0/init/rc bash-3.0.fink/init/rc
--- bash-3.0/init/rc	Wed Dec 31 19:00:00 1969
+++ bash-3.0.fink/init/rc	Fri Nov 19 14:57:47 2004
@@ -0,0 +1,71 @@
+##
+# Bourne Again Shell config file
+#
+# Wilfredo Sanchez Jr. | tritan@mit.edu
+# July 09, 1992
+#
+# MIT Project Athena
+#
+# ORIGINAL SOURCES: /usr/athena/lib/init/cshrc (ATHENA REL 7.3P)
+##
+
+     default_initdir=@PREFIX@/share/init
+default_bash_initdir=${default_initdir}/bash
+        user_initdir=~/Library/init
+   user_bash_initdir=${user_initdir}/bash
+
+if [ -r ${user_bash_initdir} ]; then
+       initdir=${user_initdir}
+  bash_initdir=${user_bash_initdir}
+else
+       initdir=${default_initdir}
+  bash_initdir=${default_bash_initdir}
+fi
+
+# SET UP HOST-DEPENDANT VARIABLES, ETC.
+
+host=$(echo $(hostname) | tr A-Z a-z)
+
+user=`whoami`
+
+export HOST=${host}
+export USER=${user}
+
+# User ID
+if [ -z "${uid}" ]; then uid=$(id | cut -d = -f 2 | cut -d \( -f 1); fi
+
+# SET COMMAND SEARCH PATH AND MAN PATH
+if [ -f ${bash_initdir}/path ]; then source ${bash_initdir}/path; fi
+
+# ENVIRONMENT SETUP
+
+if [ -n "${PS1}" ]; then interactive="YES"; fi
+
+if [ -z "${ENV_SET}" ]; then
+    if [ -f ${default_bash_initdir}/environment ]; then
+	#echo "Initializing environment..."
+	source ${default_bash_initdir}/environment
+    fi
+fi
+
+if [ -r ${default_bash_initdir}/bash.defaults ]; then
+    source ${default_bash_initdir}/bash.defaults
+fi
+
+if [ "${interactive}" = "YES" ]; then
+    # These aren't useful for non-interactive sessions
+    if [ -f ${default_bash_initdir}/aliases ]; then
+        source ${default_bash_initdir}/aliases
+    fi
+    if [ -f ${default_bash_initdir}/completions ]; then
+        source ${default_bash_initdir}/completions
+    fi
+fi
+
+# DEFAULT LOGIN SOURCES
+if [ -f ${bash_initdir}/rc.mine ]; then source ${bash_initdir}/rc.mine; fi
+
+# Local Variables:
+# mode:shell-script
+# sh-shell:bash
+# End:
diff -u -Naur bash-3.0/pathnames.h.in bash-3.0.fink/pathnames.h.in
--- bash-3.0/pathnames.h.in	Tue Feb 25 11:31:46 2003
+++ bash-3.0.fink/pathnames.h.in	Fri Nov 19 14:57:47 2004
@@ -25,7 +25,7 @@
 #define DEFAULT_HOSTS_FILE "/etc/hosts"
 
 /* The default login shell startup file. */
-#define SYS_PROFILE "/etc/profile"
+#define SYS_PROFILE "@PREFIX@/etc/profile"
 
 /* The default location of the bash debugger initialization/startup file. */
 #define DEBUGGER_START_FILE	"@DEBUGGER_START_FILE@"
diff -u -Naur bash-3.0/profile.fink bash-3.0.fink/profile.fink
--- bash-3.0/profile.fink	Wed Dec 31 19:00:00 1969
+++ bash-3.0.fink/profile.fink	Fri Nov 19 14:57:47 2004
@@ -0,0 +1 @@
+source @PREFIX@/share/init/bash/profile
