diff -ruN bash-completion-1.3/bash_completion bash-completion-1.3-patched/bash_completion
--- bash-completion-1.3/bash_completion	2011-02-06 20:16:00.000000000 +0100
+++ bash-completion-1.3-patched/bash_completion	2011-04-27 12:50:57.000000000 +0200
@@ -39,9 +39,9 @@
 
 # Alter the following to reflect the location of this file.
 #
-[ -n "$BASH_COMPLETION" ] || BASH_COMPLETION=/etc/bash_completion
-[ -n "$BASH_COMPLETION_DIR" ] || BASH_COMPLETION_DIR=/etc/bash_completion.d
-[ -n "$BASH_COMPLETION_COMPAT_DIR" ] || BASH_COMPLETION_COMPAT_DIR=/etc/bash_completion.d
+[ -n "$BASH_COMPLETION" ] || BASH_COMPLETION=@PREFIX@/etc/bash_completion
+[ -n "$BASH_COMPLETION_DIR" ] || BASH_COMPLETION_DIR=@PREFIX@/etc/bash_completion.d
+[ -n "$BASH_COMPLETION_COMPAT_DIR" ] || BASH_COMPLETION_COMPAT_DIR=@PREFIX@/etc/bash_completion.d
 readonly BASH_COMPLETION BASH_COMPLETION_DIR BASH_COMPLETION_COMPAT_DIR
 
 # Set a couple of useful vars
@@ -735,7 +735,11 @@
 #
 _configured_interfaces()
 {
-    if [ -f /etc/debian_version ]; then
+    if [ $UNAME eq "Darwin" ]; then
+        # TODO/FIXME: Update this to new bash_completion code
+        COMPREPLY=( $( compgen -W "$( /sbin/ifconfig | \
+            grep -E '^[^[:space:]]+:' | cut -d: -f1 )" -- "$cur" ) )
+    elif [ -f /etc/debian_version ]; then
         # Debian system
         COMPREPLY=( $( compgen -W "$( sed -ne 's|^iface \([^ ]\{1,\}\).*$|\1|p'\
             /etc/network/interfaces )" -- "$cur" ) )
@@ -1245,8 +1249,8 @@
         [ -r "$configfile" ] &&
         config=( "${config[@]}" "$configfile" )
     else
-        for i in /etc/ssh/ssh_config "${HOME}/.ssh/config" \
-            "${HOME}/.ssh2/config"; do
+        for i in @PREFIX@/etc/ssh/ssh_config /etc/ssh_config \
+            "${HOME}/.ssh/config" "${HOME}/.ssh2/config"; do
             [ -r $i ] && config=( "${config[@]}" "$i" )
         done
     fi
@@ -1272,9 +1276,11 @@
 
     if [ -z "$configfile" ]; then
         # Global and user known_hosts files
-        for i in /etc/ssh/ssh_known_hosts /etc/ssh/ssh_known_hosts2 \
-            /etc/known_hosts /etc/known_hosts2 ~/.ssh/known_hosts \
-            ~/.ssh/known_hosts2; do
+        for i in @PREFIX@/etc/ssh/ssh_known_hosts @PREFIX@/etc/ssh/ssh_known_hosts2 \
+            /etc/ssh/ssh_known_hosts /etc/ssh/ssh_known_hosts2 \
+            @PREFIX@/etc/known_hosts @PREFIX@/etc/known_hosts2 \
+            /etc/known_hosts /etc/known_hosts2 \
+            ~/.ssh/known_hosts ~/.ssh/known_hosts2; do
             [ -r $i ] && kh=( "${kh[@]}" $i )
         done
         for i in /etc/ssh2/knownhosts ~/.ssh2/hostkeys; do
diff -ruN bash-completion-1.3/completions/Makefile.in bash-completion-1.3-patched/completions/Makefile.in
--- bash-completion-1.3/completions/Makefile.in	2011-02-06 20:16:21.000000000 +0100
+++ bash-completion-1.3-patched/completions/Makefile.in	2011-04-27 11:17:18.000000000 +0200
@@ -224,6 +224,7 @@
 		dvd+rw-tools \
 		e2fsprogs \
 		findutils \
+		fink \
 		freeciv \
 		freerdp \
 		fuse \
@@ -288,6 +289,7 @@
 		net-tools \
 		nmap \
 		ntpdate \
+		open \
 		openldap \
 		openssl \
 		open-iscsi \
diff -ruN bash-completion-1.3/completions/apt bash-completion-1.3-patched/completions/apt
--- bash-completion-1.3/completions/apt	2011-01-21 10:36:11.000000000 +0100
+++ bash-completion-1.3-patched/completions/apt	2011-04-27 11:16:22.000000000 +0200
@@ -17,7 +17,7 @@
     if [ -n "$special" ]; then
         case $special in
             remove|autoremove|purge)
-                if [ -f /etc/debian_version ]; then
+                if [ -f /etc/debian_version ] || [ -x "@PREFIX@/bin/fink" ]; then
                     # Debian system
                     COMPREPLY=( $( _comp_dpkg_installed_packages $cur ) )
                 else
diff -ruN bash-completion-1.3/completions/aptitude bash-completion-1.3-patched/completions/aptitude
--- bash-completion-1.3/completions/aptitude	2011-01-21 10:36:11.000000000 +0100
+++ bash-completion-1.3-patched/completions/aptitude	2011-04-27 11:16:22.000000000 +0200
@@ -9,7 +9,7 @@
 } || {
 _comp_dpkg_hold_packages()
 {
-    command grep -B 2 'hold' /var/lib/dpkg/status | \
+    command grep -B 2 'hold' @PREFIX@/var/lib/dpkg/status | \
         command grep "Package: $1" | cut -d\  -f2
 }
 }
diff -ruN bash-completion-1.3/completions/dpkg bash-completion-1.3-patched/completions/dpkg
--- bash-completion-1.3/completions/dpkg	2011-01-21 10:36:11.000000000 +0100
+++ bash-completion-1.3-patched/completions/dpkg	2011-04-27 11:16:22.000000000 +0200
@@ -8,7 +8,7 @@
 } || {
 _comp_dpkg_installed_packages()
 {
-    command grep -A 1 "Package: $1" /var/lib/dpkg/status | \
+    command grep -A 1 "Package: $1" @PREFIX@/var/lib/dpkg/status | \
         command grep -B 1 -Ee "ok installed|half-installed|unpacked| \
             half-configured|config-files" \
             -Ee "^Essential: yes" | \
diff -ruN bash-completion-1.3/completions/fink bash-completion-1.3-patched/completions/fink
--- bash-completion-1.3/completions/fink	1970-01-01 01:00:00.000000000 +0100
+++ bash-completion-1.3-patched/completions/fink	2011-04-27 11:16:22.000000000 +0200
@@ -0,0 +1,128 @@
+#!/bin/bash
+
+# 
+# Copyright (c) 2004, Lukas Westermann <contact AT openos DOT ch>
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without 
+# modification, are permitted provided that the following conditions 
+# are met:
+#   - Redistributions of source code must retain the above copyright 
+#     notice, this list of conditions and the following disclaimer.
+#   - Redistributions in binary form must reproduce the above 
+#     copyright notice, this list of conditions and the following 
+#     disclaimer in the documentation and/or other materials 
+#     provided with the distribution.
+#   - The names of its contributors may not be used to endorse or 
+#     promote products derived from this software without specific 
+#     prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
+# CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+# INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR 
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT 
+# NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) 
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR 
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, 
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+#
+# Completion for Fink command
+#
+# History
+# ---------------------------------
+# Version 1.0.0  - initial release
+# Version 1.0.1 - added test to check fink availability
+# Version 1.0.2 - simplified checks for fink availabiltiy
+# Version 1.0.3 - fixed problems with spaces in completion
+#
+
+[ $UNAME = Darwin ] && have fink && {
+_fink_installed() {
+    fink list -i --tab "$*" | cut -f2 2>/dev/null
+}
+
+_fink_all() {
+    apt-cache pkgnames "$*" 2>/dev/null
+}
+
+_fink_trees() {
+    cat @PREFIX@/etc/fink.conf | sed -e '/^Trees/!d; s/^Trees://' 2>/dev/null
+}
+
+_fink()
+{
+	local cur prev special i
+	
+	COMPREPLY=()
+	cur=${COMP_WORDS[COMP_CWORD]}
+	prev=${COMP_WORDS[COMP_CWORD-1]}
+
+	for (( i=0; i < ${#COMP_WORDS}-1; i++ )); do
+		if [[ ${COMP_WORDS[i]} == @(install|update|enable|activate|use|list|remove|disable|deactivate|unuse|delete \
+		                            |purge|apropos|describe|desc|description|info|fetch-missing|fetch \
+		                            |build|rebuild|reinstall|validate|check|scanpackages) ]]; then
+			special=${COMP_WORDS[i]}
+		fi
+	done
+
+	if [ -n "${special}" ]; then
+		case $special in
+		    install|enable|activate|use|apropos|describe|desc|description|info|fetch|build)
+			COMPREPLY=( $( _fink_all $cur ) )
+			;;
+		    remove|disable|deactivate|unuse|delete|purge|update|rebuild|reinstall)
+			COMPREPLY=( $( _fink_installed $cur ) )
+			;;		    
+		    fetch-missing)
+			COMPREPLY=( $( compgen -W '-i --ignore-restrictive' -- $cur ) )
+			;;
+		    validate|check)
+			local IFS=$'\t\n'
+			_filedir '@(deb|info)'
+			#COMPREPLY=( $( compgen -f -- $cur ) )
+			;;
+		    scanpackages)
+			COMPREPLY=( $( compgen -W '$( _fink_trees )' -- $cur ) )
+			;;
+		    list)
+			if [[ "$cur" == -* ]]; then
+			    # TODO: Fix problems with --tree=
+			    #if [[ $cur == --tree=* || $cur == -t=* ]]; then
+				#local pfx las #res
+                                #pfx=$( echo $cur | sed -e 's/^\([^=]*\)=.*$/\1/' )
+				#las=$( echo "$cur" | sed -e 's/^[^=]*=//' )
+				#[[ $las == $cur ]] && las=()
+				#res=$( cat @PREFIX@/etc/fink.conf | sed -e '/^Trees/!d; s/^Trees://' 2>/dev/null )
+                                #COMPREPLY=( $( compgen -W '$( _fink_trees )' -- "${las}" ) )
+			    #else
+				COMPREPLY=( $( compgen -W '-h --help -t --tab -i --installed -o --outdated -u --uptodate -n --notinstalled \
+                                                       -s --section -m --maintainer --tree -w --width' -- $cur ) )
+			    #fi
+			else
+			    COMPREPLY=( $( _fink_all $cur ) )
+			fi
+			;;
+		esac
+
+		return 0
+	fi
+
+	if [[ "$cur" == -* ]]; then
+		COMPREPLY=( $( compgen -W '-h -q -V -v -y --help --quite --version --verbose --yes' -- $cur ) )
+	else
+		COMPREPLY=( $( compgen -W 'install remove purge update selfupdate update-all configure list \
+		                           apropos describe fetch fetch-all fetch-missing index validate scanpackages \
+                                           build rebuild reinstall checksums cleanup' -- $cur ) )
+	fi
+
+	return 0
+}
+complete $filenames -F _fink fink
+}
diff -ruN bash-completion-1.3/completions/open bash-completion-1.3-patched/completions/open
--- bash-completion-1.3/completions/open	1970-01-01 01:00:00.000000000 +0100
+++ bash-completion-1.3-patched/completions/open	2011-04-27 11:16:22.000000000 +0200
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+# 
+# Copyright (c) 2004, Lukas Westermann <contact AT openos DOT ch>
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without 
+# modification, are permitted provided that the following conditions 
+# are met:
+#   - Redistributions of source code must retain the above copyright 
+#     notice, this list of conditions and the following disclaimer.
+#   - Redistributions in binary form must reproduce the above 
+#     copyright notice, this list of conditions and the following 
+#     disclaimer in the documentation and/or other materials 
+#     provided with the distribution.
+#   - The names of its contributors may not be used to endorse or 
+#     promote products derived from this software without specific 
+#     prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
+# CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+# INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR 
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT 
+# NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) 
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR 
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, 
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+#
+# Bash Auto-Completion for the NextStep command 'open'
+#
+# History
+# ----------------------------------
+# Version 1.0.0 - initial release
+# Version 1.0.1 - added test for availabilty (Darwin only
+#                 and check for open)
+# Version 1.0.2 - fixed problems with IFS for -a and -e
+# Version 1.0.3 - simplified availability check
+# Version 1.0.4 - fixed problem with spaces and completion
+#
+
+[ $UNAME = Darwin ] && have open && {
+_list_apps() {
+    local IFS=$'\n'
+    find '/Applications' "${HOME}"'/Applications' '/Developer/Applications' -name "${1}*.app" -maxdepth 2 2>/dev/null
+    #TODO: MS Word and other carbon apps...HOW ???
+}
+
+_open()
+{
+
+    local cur prev IFS=$'\t\n'
+    COMPREPLY=()
+    cur=${COMP_WORDS[COMP_CWORD]}
+    prev=${COMP_WORDS[COMP_CWORD-1]}
+
+    if [[ $prev == -a ]]; then
+	local res ncur
+	ncur=$( echo "$cur" | sed -e 's/\\//g' 2>/dev/null )
+	res=$( _list_apps "${ncur}" | sed -e 's/.*\/\([^\/]*\)\.app/\1/' )
+	COMPREPLY=( $( compgen -W "${res}" -- "${ncur}" ) )
+    elif [[ $prev == -e ]]; then
+	_filedir '@(txt|doc|rtf|rtfd|pdf|png|jpg|gif)'
+    elif [[ $cur == -* ]]; then
+	local IFS=$' \t\n'
+	COMPREPLY=( $( compgen -W '-a -e' -- "$cur" ) )
+    else
+	local first="${COMP_WORDS[1]}" second="${COMP_WORDS[2]}"
+	if [[ $first == -a ]]; then
+	    case "$second" in
+		QuickTime*)
+		    _filedir '@(mp3|mpg|mpeg|avi|m4*|mov|alc)'
+		    ;;
+		iTunes)
+		    _filedir '@(mp3|m4*|ogg|alc)'
+		    ;;
+		Safari|Mozilla|Camino|Opera|Firebird)
+		    # TODO: complete from history/bookmarks ?
+		    _filedir '@(htm|html|xhtm|xhtml|php|php3|php3|xml)'
+		    ;;
+		VLC)
+		    # TODO: complete for CD/DVD too...
+		    _filedir '@(avi|mpg)'
+		    ;;
+		BitTorrent*)
+		    _filedir '@(torrent)'
+		    ;;
+		*)
+		    # others, complete all
+		    _filedir
+		    ;;
+	    esac
+	else
+	    _filedir
+	fi
+    fi
+
+}
+complete $filenames -F _open open
+}
diff -ruN bash-completion-1.3/completions/openssl bash-completion-1.3-patched/completions/openssl
--- bash-completion-1.3/completions/openssl	2011-01-21 10:36:11.000000000 +0100
+++ bash-completion-1.3-patched/completions/openssl	2011-04-27 11:16:22.000000000 +0200
@@ -15,8 +15,8 @@
 
     # if no config given, check some usual default locations
     if [ -z "$config" ]; then
-        for f in /etc/ssl/openssl.cnf /etc/pki/tls/openssl.cnf \
-            /usr/share/ssl/openssl.cnf; do
+        for f in @PREFIX@/etc/ssl/openssl.cnf \
+            /System/Library/OpenSSL/openssl.cnf; do
             [ -f $f ] && config=$f && break
         done
     fi
diff -ruN bash-completion-1.3/completions/procps bash-completion-1.3-patched/completions/procps
--- bash-completion-1.3/completions/procps	2011-01-21 10:36:11.000000000 +0100
+++ bash-completion-1.3-patched/completions/procps	2011-04-27 11:16:22.000000000 +0200
@@ -2,7 +2,7 @@
 
 # killall(1) (Linux and FreeBSD) and pkill(1) completion.
 #
-[[ $UNAME == Linux || $UNAME == FreeBSD ]] || have pkill &&
+[[ $UNAME == Linux || $UNAME == FreeBSD || $UNAME == Darwin ]] || have pkill &&
 _killall()
 {
     local cur
@@ -18,7 +18,7 @@
 
     return 0
 }
-[[ $UNAME == Linux || $UNAME == FreeBSD ]] && complete -F _killall killall
+[[ $UNAME == Linux || $UNAME == FreeBSD || $UNAME == Darwin ]] && complete -F _killall killall
 have pkill && complete -F _killall pkill
 
 # pgrep(1) completion.
diff -ruN bash-completion-1.3/completions/update-alternatives bash-completion-1.3-patched/completions/update-alternatives
--- bash-completion-1.3/completions/update-alternatives	2011-01-21 10:36:11.000000000 +0100
+++ bash-completion-1.3-patched/completions/update-alternatives	2011-04-27 11:16:22.000000000 +0200
@@ -7,6 +7,7 @@
     # find the admin dir
     for i in alternatives dpkg/alternatives rpm/alternatives; do
         [ -d /var/lib/$i ] && admindir=/var/lib/$i && break
+        [ -d "@PREFIX@/etc/$i" ] && admindir="@PREFIX@/etc/$i" && break
     done
     for (( i=1; i < COMP_CWORD; i++ )); do
         if [[ "${COMP_WORDS[i]}" == --admindir ]]; then
