diff -ruN gengetopt-2.8.1.orig/txtc.sh.in gengetopt-2.8.1/txtc.sh.in
--- gengetopt-2.8.1.orig/txtc.sh.in	Tue Mar 20 16:16:38 2001
+++ gengetopt-2.8.1/txtc.sh.in	Sun Feb 23 11:03:32 2003
@@ -89,7 +89,7 @@
   # Then pipe through awk to make the C program
   # FIXME: perhaps another substitution needed for tabs?
   rm -f $c_file
-  cat $file | sed 's/\\/\\\\/g
+  cat $file | /usr/bin/sed 's/\\/\\\\/g
                    s/\"/\\\"/g
                    s/[[:blank:]]*$//g' | 
               $AWK -v symbol_name="$symbol_name" -v file_length="$file_length" \
