diff -Nurd control-center-2.18.1.orig/libwindow-settings/Makefile.am control-center-2.18.1/libwindow-settings/Makefile.am
--- control-center-2.18.1.orig/libwindow-settings/Makefile.am	2007-03-18 13:04:18.000000000 -0400
+++ control-center-2.18.1/libwindow-settings/Makefile.am	2008-03-13 16:00:31.000000000 -0400
@@ -36,7 +36,7 @@
 	gnome-window-settings-2.0.pc.in		\
 	ChangeLog
 
-wms_flags = -export_dynamic -avoid-version
+wms_flags = -export_dynamic -avoid-version -module
 wmsdir = $(WM_MODULE_DIR)
 
 wms_LTLIBRARIES =			\
