diff -Nurd gnome-control-center-2.20.3.orig/capplets/common/theme-thumbnail.c gnome-control-center-2.20.3/capplets/common/theme-thumbnail.c
--- gnome-control-center-2.20.3.orig/capplets/common/theme-thumbnail.c	2008-01-08 08:18:46.000000000 -0500
+++ gnome-control-center-2.20.3/capplets/common/theme-thumbnail.c	2008-04-24 14:08:15.000000000 -0400
@@ -19,8 +19,6 @@
 #include "gtkrc-utils.h"
 #include "capplet-util.h"
 
-static gint child_pid;
-
 typedef struct
 {
   gboolean set;
@@ -1109,7 +1107,15 @@
   pipe (pipe_to_factory_fd);
   pipe (pipe_from_factory_fd);
 
-  child_pid = fork ();
+/* Apple's CoreFoundation classes must not be used from forked
+ * processes. Since freetype (and thus GTK) uses them, we simply
+ * disable the thumbnailer on MacOS for now. That means no thumbs
+ * until the thumbnailing process is rewritten, but at least we won't
+ * make apps crash. */
+#ifndef __APPLE__
+  gint child_pid;
+
+child_pid = fork ();
   if (child_pid == 0)
   {
     ThemeThumbnailData data;
@@ -1147,6 +1153,8 @@
   /* Parent */
   close (pipe_to_factory_fd[0]);
   close (pipe_from_factory_fd[1]);
+#endif /* __APPLE__ */
+
   async_data.set = FALSE;
   async_data.thumbnail_width = 0;
   async_data.thumbnail_height = 0;
diff -Nurd gnome-control-center-2.20.3.orig/libwindow-settings/Makefile.in gnome-control-center-2.20.3/libwindow-settings/Makefile.in
--- gnome-control-center-2.20.3.orig/libwindow-settings/Makefile.in	2008-01-08 08:28:18.000000000 -0500
+++ gnome-control-center-2.20.3/libwindow-settings/Makefile.in	2008-04-24 14:03:24.000000000 -0400
@@ -375,7 +375,7 @@
 	gnome-window-settings-2.0.pc.in		\
 	ChangeLog
 
-wms_flags = -export_dynamic -avoid-version
+wms_flags = -export_dynamic -avoid-version -module
 wmsdir = $(WM_MODULE_DIR)
 wms_LTLIBRARIES = \
 	libmetacity.la
