diff -uNr dasher-4.3.3/Src/Gtk2/DasherControl.cpp dasher-4.3.3-new/Src/Gtk2/DasherControl.cpp
--- dasher-4.3.3/Src/Gtk2/DasherControl.cpp	2006-12-18 06:07:31.000000000 -0500
+++ dasher-4.3.3-new/Src/Gtk2/DasherControl.cpp	2007-01-08 23:19:43.000000000 -0500
@@ -45,8 +45,8 @@
   RegisterFactory(new CWrapperFactory(m_pEventHandler, m_pSettingsStore, new CDasherMouseInput(m_pEventHandler, m_pSettingsStore)));
   RegisterFactory(new CWrapperFactory(m_pEventHandler, m_pSettingsStore, new CSocketInput(m_pEventHandler, m_pSettingsStore)));
   RegisterFactory(new CWrapperFactory(m_pEventHandler, m_pSettingsStore, new CDasher1DMouseInput(m_pEventHandler, m_pSettingsStore)));
-  RegisterFactory(new CWrapperFactory(m_pEventHandler, m_pSettingsStore, new CDasherJoystickInput(m_pEventHandler, m_pSettingsStore, this)));
-  RegisterFactory(new CWrapperFactory(m_pEventHandler, m_pSettingsStore, new CDasherJoystickInputDiscrete(m_pEventHandler, m_pSettingsStore, this)));
+  // RegisterFactory(new CWrapperFactory(m_pEventHandler, m_pSettingsStore, new CDasherJoystickInput(m_pEventHandler, m_pSettingsStore, this)));
+  // RegisterFactory(new CWrapperFactory(m_pEventHandler, m_pSettingsStore, new CDasherJoystickInputDiscrete(m_pEventHandler, m_pSettingsStore, this)));
   
   // Create locally cached copies of the mouse input objects, as we
   // need to pass coordinates to them from the timer callback
diff -uNr dasher-4.3.3/Src/Gtk2/DasherControl.h dasher-4.3.3-new/Src/Gtk2/DasherControl.h
--- dasher-4.3.3/Src/Gtk2/DasherControl.h	2006-12-18 06:07:31.000000000 -0500
+++ dasher-4.3.3-new/Src/Gtk2/DasherControl.h	2007-01-08 23:19:34.000000000 -0500
@@ -4,7 +4,6 @@
 #include "PangoCache.h"
 #include "Canvas.h"
 #include "../DasherCore/SocketInput.h"
-#include "joystick_input.h"
 #include "mouse_input.h"
 
 #include "GtkDasherControl.h"
