diff -uNr eog-2.17.92/help/eu/Makefile.in eog-2.17.92-new/help/eu/Makefile.in
--- eog-2.17.92/help/eu/Makefile.in	2007-02-26 12:36:56.000000000 -0500
+++ eog-2.17.92-new/help/eu/Makefile.in	2007-03-04 00:41:22.000000000 -0500
@@ -466,7 +466,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(DESTDIR)$(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -uNr eog-2.17.92/help/it/Makefile.in eog-2.17.92-new/help/it/Makefile.in
--- eog-2.17.92/help/it/Makefile.in	2007-02-26 12:36:57.000000000 -0500
+++ eog-2.17.92-new/help/it/Makefile.in	2007-03-04 00:41:25.000000000 -0500
@@ -466,7 +466,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(DESTDIR)$(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -uNr eog-2.17.92/help/ja/Makefile.in eog-2.17.92-new/help/ja/Makefile.in
--- eog-2.17.92/help/ja/Makefile.in	2007-02-26 12:36:57.000000000 -0500
+++ eog-2.17.92-new/help/ja/Makefile.in	2007-03-04 00:41:28.000000000 -0500
@@ -466,7 +466,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(DESTDIR)$(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -uNr eog-2.17.92/help/ko/Makefile.in eog-2.17.92-new/help/ko/Makefile.in
--- eog-2.17.92/help/ko/Makefile.in	2007-02-26 12:36:57.000000000 -0500
+++ eog-2.17.92-new/help/ko/Makefile.in	2007-03-04 00:41:35.000000000 -0500
@@ -466,7 +466,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(DESTDIR)$(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -uNr eog-2.17.92/help/zh_CN/Makefile.in eog-2.17.92-new/help/zh_CN/Makefile.in
--- eog-2.17.92/help/zh_CN/Makefile.in	2007-02-26 12:36:58.000000000 -0500
+++ eog-2.17.92-new/help/zh_CN/Makefile.in	2007-03-04 00:41:38.000000000 -0500
@@ -466,7 +466,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(DESTDIR)$(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -uNr eog-2.17.92/help/zh_TW/Makefile.in eog-2.17.92-new/help/zh_TW/Makefile.in
--- eog-2.17.92/help/zh_TW/Makefile.in	2007-02-26 12:36:58.000000000 -0500
+++ eog-2.17.92-new/help/zh_TW/Makefile.in	2007-03-04 00:41:42.000000000 -0500
@@ -466,7 +466,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(DESTDIR)$(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -uNr eog-2.17.92/omf.make eog-2.17.92-new/omf.make
--- eog-2.17.92/omf.make	2007-02-26 12:36:35.000000000 -0500
+++ eog-2.17.92-new/omf.make	2007-03-04 00:40:46.000000000 -0500
@@ -45,7 +45,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(DESTDIR)$(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
