diff -uNr gdm-2.20.0/daemon/gdm-xdmcp-manager.c gdm-2.20.0-new/daemon/gdm-xdmcp-manager.c
--- gdm-2.20.0/daemon/gdm-xdmcp-manager.c	2007-09-17 10:45:50.000000000 -0400
+++ gdm-2.20.0-new/daemon/gdm-xdmcp-manager.c	2007-09-25 15:07:01.000000000 -0400
@@ -366,7 +366,7 @@
 		if (ai->ai_flags & AI_NUMERICHOST) {
 			g_string_append (str, "numhost ");
 		}
-		if (ai->ai_flags & AI_NUMERICSERV) {
+		if (ai->ai_flags & NI_NUMERICSERV) {
 			g_string_append (str, "numserv ");
 		}
 		if (ai->ai_flags & AI_V4MAPPED) {
diff -uNr gdm-2.20.0/daemon/slave.c gdm-2.20.0-new/daemon/slave.c
--- gdm-2.20.0/daemon/slave.c	2007-09-17 10:45:50.000000000 -0400
+++ gdm-2.20.0-new/daemon/slave.c	2007-09-25 15:04:26.000000000 -0400
@@ -4407,8 +4407,10 @@
 
 			u->ut_type = DEAD_PROCESS;
 			u->ut_tv.tv_sec = record.ut_tv.tv_sec;
+#ifndef __APPLE__
 			u->ut_exit.e_termination = 0;
 			u->ut_exit.e_exit = 0;
+#endif
 
 			pututxline (u);
 
