diff -Nurd -x'*~' gdm-2.8.0.5.orig/daemon/xdmcp.c gdm-2.8.0.5/daemon/xdmcp.c
--- gdm-2.8.0.5.orig/daemon/xdmcp.c	2005-09-14 21:51:18.000000000 -0400
+++ gdm-2.8.0.5/daemon/xdmcp.c	2005-11-03 02:16:29.000000000 -0500
@@ -2467,6 +2467,13 @@
     }
 }
 
+#ifdef HAVE_TCPWRAPPERS
+	
+/* avoids a warning, my tcpd.h file doesn't include this prototype, even
+ * though the library does include the function and the manpage mentions it
+ */
+extern int hosts_ctl (char *daemon, char *client_name, char *client_addr, char *client_user);
+#endif
 
 static gboolean
 gdm_xdmcp_host_allow (
@@ -2478,12 +2485,6 @@
 		     )
 {
 #ifdef HAVE_TCPWRAPPERS
-	
-    /* avoids a warning, my tcpd.h file doesn't include this prototype, even
-     * though the library does include the function and the manpage mentions it
-     */
-    extern int hosts_ctl (char *daemon, char *client_name, char *client_addr, char *client_user);
-
     GdmHostent *client_he;
     char *client;
     gboolean ret;
