diff -uNr gdm-2.17.8/daemon/xdmcp.c gdm-2.17.8-new/daemon/xdmcp.c
--- gdm-2.17.8/daemon/xdmcp.c	2007-02-26 03:59:35.000000000 -0500
+++ gdm-2.17.8-new/daemon/xdmcp.c	2007-03-10 14:32:01.000000000 -0500
@@ -2658,6 +2658,14 @@
     }
 }
 
+#ifdef HAVE_TCPWRAPPERS
+    /*
+     * Avoids a warning, my tcpd.h file doesn't include this prototype, even
+     * though the library does include the function and the manpage mentions it
+     */
+    extern int hosts_ctl (char *daemon, char *client_name,
+                          char *client_addr, char *client_user);
+#endif
 
 static gboolean
 gdm_xdmcp_host_allow (
@@ -2670,12 +2678,6 @@
 {
 #ifdef HAVE_TCPWRAPPERS
 	
-    /*
-     * Avoids a warning, my tcpd.h file doesn't include this prototype, even
-     * though the library does include the function and the manpage mentions it
-     */
-    extern int hosts_ctl (char *daemon, char *client_name,
-                          char *client_addr, char *client_user);
 
     GdmHostent *client_he;
     char *client;
