diff -Nurd -x'*~' gedit-2.24.1.orig/configure gedit-2.24.1/configure
--- gedit-2.24.1.orig/configure	2008-11-02 18:18:39.000000000 -0500
+++ gedit-2.24.1/configure	2008-11-20 17:35:40.000000000 -0500
@@ -23903,7 +23903,7 @@
 echo "${ECHO_T}yes" >&6; }
 	:
 fi
-GEDIT_LIBS="${GEDIT_LIBS}"
+GEDIT_LIBS="${GEDIT_LIBS} -lSM"
 
 
 
diff -Nurd -x'*~' gedit-2.24.1.orig/gedit/gedit-python-module.c gedit-2.24.1/gedit/gedit-python-module.c
--- gedit-2.24.1.orig/gedit/gedit-python-module.c	2008-09-07 12:13:19.000000000 -0400
+++ gedit-2.24.1/gedit/gedit-python-module.c	2008-11-20 18:06:56.000000000 -0500
@@ -25,6 +25,8 @@
 #endif
 
 #include <Python.h>
+#undef NO_IMPORT_PYGOBJECT
+#undef NO_IMPORT_PYGTK
 #include <pygobject.h>
 #include <pygtk/pygtk.h>
 
diff -Nurd -x'*~' gedit-2.24.1.orig/plugins/externaltools/data/Makefile.in gedit-2.24.1/plugins/externaltools/data/Makefile.in
--- gedit-2.24.1.orig/plugins/externaltools/data/Makefile.in	2008-11-02 18:18:43.000000000 -0500
+++ gedit-2.24.1/plugins/externaltools/data/Makefile.in	2008-11-20 17:28:16.000000000 -0500
@@ -471,7 +471,7 @@
 # (.desktop), which happens to be translated using intltool.
 $(tools_SCRIPTS): %: %.tool.in %.desktop $(TOOL_MERGE)
 	perl $(TOOL_MERGE) -o $@ $< $(word 2,$^)
-	chmod --reference=$< $@
+	chmod 755 $@
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
