diff -uNr gedit-2.20.0/gedit/gedit-python-module.c gedit-2.20.0-new/gedit/gedit-python-module.c
--- gedit-2.20.0/gedit/gedit-python-module.c	2007-07-21 06:31:18.000000000 -0400
+++ gedit-2.20.0-new/gedit/gedit-python-module.c	2007-09-26 10:55:41.000000000 -0400
@@ -24,6 +24,9 @@
 #include <config.h>
 #endif
 
+#undef NO_IMPORT_PYGOBJECT
+#undef NO_IMPORT_PYGTK
+
 #include <pygobject.h>
 #include <pygtk/pygtk.h>
 
diff -uNr gedit-2.20.0/plugins/externaltools/data/Makefile.in gedit-2.20.0-new/plugins/externaltools/data/Makefile.in
--- gedit-2.20.0/plugins/externaltools/data/Makefile.in	2007-09-17 06:51:11.000000000 -0400
+++ gedit-2.20.0-new/plugins/externaltools/data/Makefile.in	2007-09-26 10:56:03.000000000 -0400
@@ -465,7 +465,7 @@
 # (.desktop), which happens to be translated using intltool.
 $(tools_SCRIPTS): %: %.tool.in %.desktop $(TOOL_MERGE)
 	perl $(TOOL_MERGE) -o $@ $< $(word 2,$^)
-	chmod --reference=$< $@
+	chmod 755 $@
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
