diff -Nurd -x'*~' gksu-2.0.2.orig/gksu/Makefile.in gksu-2.0.2/gksu/Makefile.in
--- gksu-2.0.2.orig/gksu/Makefile.in	2009-03-04 14:22:25.000000000 -0500
+++ gksu-2.0.2/gksu/Makefile.in	2010-04-29 05:29:25.000000000 -0400
@@ -241,7 +241,7 @@
 INCLUDES = @GKSU_CFLAGS@
 AM_CPPFLAGS = -DLOCALEDIR=\"$(datadir)/locale\" -DDATA_DIR=\"$(datadir)\" -DPREFIX=\"$(prefix)\"
 noinst_HEADERS = defines.h
-gksu_LDFLAGS = @GKSU_LIBS@ -lutil
+gksu_LDFLAGS = @GKSU_LIBS@ $(INTLLIBS)
 gksu_SOURCES = gksu.c
 all: all-am
 
diff -Nurd -x'*~' gksu-2.0.2.orig/nautilus-gksu/Makefile.in gksu-2.0.2/nautilus-gksu/Makefile.in
--- gksu-2.0.2.orig/nautilus-gksu/Makefile.in	2009-03-04 14:22:25.000000000 -0500
+++ gksu-2.0.2/nautilus-gksu/Makefile.in	2010-04-29 05:34:01.000000000 -0400
@@ -247,7 +247,7 @@
 nautilus_extension_LTLIBRARIES = libnautilus-gksu.la
 libnautilus_gksu_la_SOURCES = libnautilus-gksu.c
 libnautilus_gksu_la_LDFLAGS = LDFLAGS = -module -avoid-version -no-undefined
-libnautilus_gksu_la_LIBADD = $(NAUTILUS_LIBS)
+libnautilus_gksu_la_LIBADD = $(NAUTILUS_LIBS) $(INTLLIBS)
 all: all-am
 
 .SUFFIXES:
