diff -Nurd -x'*~' glade3-3.6.4.orig/plugins/gnome/Makefile.am glade3-3.6.4/plugins/gnome/Makefile.am
--- glade3-3.6.4.orig/plugins/gnome/Makefile.am	2009-04-18 13:52:05.000000000 -0400
+++ glade3-3.6.4/plugins/gnome/Makefile.am	2009-06-06 16:52:00.000000000 -0400
@@ -11,8 +11,7 @@
 gladegnomedir = $(pkglibdir)/modules
 
 libgladegnome_la_SOURCES = glade-gnome.c
-libgladegnome_la_CPPFLAGS = $(AM_CPPFLAGS)
-libgladegnome_la_CFLAGS = \
+libgladegnome_la_CPPFLAGS = \
         -DG_LOG_DOMAIN=\"GladeUI-GNOME\" \
 	-I$(top_srcdir)                  \
 	-I$(top_builddir)                \
@@ -20,6 +19,8 @@
 	$(GNOME_CFLAGS)                  \
 	$(PLUGINS_WARN_CFLAGS)           \
 	$(AM_CPPFLAGS)
+libgladegnome_la_CFLAGS = \
+	$(AM_CFLAGS)
 libgladegnome_la_LDFLAGS = -module -avoid-version $(GNOME_LIBS) $(AM_LDFLAGS)
 libgladegnome_la_LIBADD = $(libgladeui) $(GLADE_LIBS)
 
diff -Nurd -x'*~' glade3-3.6.4.orig/plugins/gtk+/Makefile.am glade3-3.6.4/plugins/gtk+/Makefile.am
--- glade3-3.6.4.orig/plugins/gtk+/Makefile.am	2009-04-18 13:52:05.000000000 -0400
+++ glade3-3.6.4/plugins/gtk+/Makefile.am	2009-06-06 16:52:00.000000000 -0400
@@ -10,17 +10,15 @@
 gladegtk_LTLIBRARIES = libgladegtk.la
 gladegtkdir = $(pkglibdir)/modules
 
-libgladegtk_la_CPPLAGS =   \
+libgladegtk_la_CPPFLAGS =   \
 	-I$(top_srcdir)    \
 	-I$(top_builddir)  \
-	$(AM_CPPFLAGS)
-
-libgladegtk_la_CFLAGS =  \
 	-DG_LOG_DOMAIN=\"GladeUI-GTK\" \
-	-I$(top_srcdir)                \
-	-I$(top_builddir)              \
 	$(GTK_CFLAGS)	               \
 	$(PLUGINS_WARN_CFLAGS)         \
+	$(AM_CPPFLAGS)
+
+libgladegtk_la_CFLAGS =  \
 	$(AM_CFLAGS)
 
 libgladegtk_la_SOURCES     = glade-gtk.c glade-accels.c glade-attributes.c glade-convert.c fixed-bg.xpm \
