diff -Nurd -x'*~' glib-2.14.6.orig/configure glib-2.14.6/configure
--- glib-2.14.6.orig/configure	2008-02-07 00:41:30.000000000 -0500
+++ glib-2.14.6/configure	2008-02-23 16:34:41.000000000 -0500
@@ -33808,10 +33808,11 @@
 { echo "$as_me:$LINENO: result: $glib_cv___va_copy" >&5
 echo "${ECHO_T}$glib_cv___va_copy" >&6; }
 
-if test "x$glib_cv_va_copy" = "xyes"; then
-  g_va_copy_func=va_copy
-else if test "x$glib_cv___va_copy" = "xyes"; then
+# Prefer __va_copy, available on both gcc2 and gcc3.
+if test "x$glib_cv___va_copy" = "xyes"; then
   g_va_copy_func=__va_copy
+else if test "x$glib_cv_va_copy" = "xyes"; then
+  g_va_copy_func=va_copy
 fi
 fi
 
@@ -35219,6 +35220,9 @@
 			G_THREAD_CFLAGS="-D_THREAD_SAFE"
 		fi
 		;;
+	*-darwin*)
+		# Nothing needed.
+		;;
 	*-dg-dgux*)  # DG/UX
 		G_THREAD_CFLAGS="-D_REENTRANT -D_POSIX4A_DRAFT10_SOURCE"
 		;;
@@ -41793,9 +41797,10 @@
 #define GLIB_MICRO_VERSION $GLIB_MICRO_VERSION
 "
 
+# Prefer __va_copy, available on both gcc2 and gcc3.
 case xyes in
-x$glib_cv_va_copy)	glib_vacopy='#define G_VA_COPY	va_copy' ;;
 x$glib_cv___va_copy)	glib_vacopy='#define G_VA_COPY	__va_copy' ;;
+x$glib_cv_va_copy)	glib_vacopy='#define G_VA_COPY	va_copy' ;;
 *)			glib_vacopy=''
 esac
 
@@ -43072,8 +43077,11 @@
 /* gcc-2.95.x supports both gnu style and ISO varargs, but if -ansi
  * is passed ISO vararg support is turned off, and there is no work
  * around to turn it on, so we unconditionally turn it off.
+ *
+ * On darwin, use gcc macro to avoid incompatibility between gcc2
+ * and gcc3.
  */
-#if __GNUC__ == 2 && __GNUC_MINOR__ == 95
+#if (__GNUC__ == 2 && __GNUC_MINOR__ == 95) || defined (__APPLE__)
 #  undef G_HAVE_ISO_VARARGS
 #endif
 
diff -Nurd -x'*~' glib-2.14.6.orig/glib/gutils.h glib-2.14.6/glib/gutils.h
--- glib-2.14.6.orig/glib/gutils.h	2008-02-07 00:24:59.000000000 -0500
+++ glib-2.14.6/glib/gutils.h	2008-02-23 20:56:44.000000000 -0500
@@ -109,12 +109,7 @@
 #  define G_INLINE_FUNC
 #  undef  G_CAN_INLINE
 #elif defined (__GNUC__) 
-#  if __GNUC_PREREQ (4,2) && defined (__STDC_VERSION__) \
-   && __STDC_VERSION__ >= 199901L
-#    define G_INLINE_FUNC extern __inline __attribute__ ((__gnu_inline__))
-#  else
-#    define G_INLINE_FUNC extern __inline
-#  endif
+#  define G_INLINE_FUNC static inline
 #elif defined (G_CAN_INLINE) 
 #  define G_INLINE_FUNC static inline
 #else /* can't inline */
diff -Nurd -x'*~' glib-2.14.6.orig/glib/libcharset/Makefile.in glib-2.14.6/glib/libcharset/Makefile.in
--- glib-2.14.6.orig/glib/libcharset/Makefile.in	2008-02-07 00:41:28.000000000 -0500
+++ glib-2.14.6/glib/libcharset/Makefile.in	2008-02-23 16:34:41.000000000 -0500
@@ -252,7 +252,7 @@
 target_alias = @target_alias@
 
 INCLUDES = \
-	-DLIBDIR=\"$(libdir)\" -I$(top_srcdir)
+	-DLIBDIR=\"$(sysconfdir)/glib-2.0\" -I$(top_srcdir)
 
 
 noinst_LTLIBRARIES = libcharset.la
@@ -274,8 +274,8 @@
 	libcharset-glib.patch
 
 
-charset_alias = $(DESTDIR)$(libdir)/charset.alias
-charset_tmp = $(DESTDIR)$(libdir)/charset.tmp
+charset_alias = $(DESTDIR)$(sysconfdir)/glib-2.0/charset.alias
+charset_tmp = $(DESTDIR)$(sysconfdir)/glib-2.0/charset.tmp
 
 SUFFIXES = .sed .sin
 
@@ -556,7 +556,7 @@
 	uninstall-am uninstall-info-am uninstall-local
 
 install-exec-local: all-local
-	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(mkinstalldirs) $(DESTDIR)$(sysconfdir)/glib-2.0
 	if test -f $(charset_alias); then \
 	  sed -f ref-add.sed $(charset_alias) > $(charset_tmp) ; \
 	  $(INSTALL_DATA) $(charset_tmp) $(charset_alias) ; \
diff -Nurd -x'*~' glib-2.14.6.orig/glib/libcharset/charset.alias glib-2.14.6/glib/libcharset/charset.alias
--- glib-2.14.6.orig/glib/libcharset/charset.alias	1969-12-31 19:00:00.000000000 -0500
+++ glib-2.14.6/glib/libcharset/charset.alias	2008-02-23 16:34:41.000000000 -0500
@@ -0,0 +1,111 @@
+# This file contains a table of character encoding aliases,
+# suitable for operating system 'darwin'.
+C			ASCII
+bg_BG			UTF-8
+bg_BG.CP1251		CP1251
+cs_CZ			UTF-8
+cs_CZ.ISO8859-2		ISO-8859-2
+da_DK			UTF-8
+da_DK.ISO8859-1		ISO-8859-1
+da_DK.ISO8859-15	ISO-8859-15
+de_AT			UTF-8
+de_AT.ISO8859-1		ISO-8859-1
+de_AT.ISO8859-15	ISO-8859-15
+de_CH			UTF-8
+de_CH.ISO8859-1		ISO-8859-1
+de_CH.ISO8859-15	ISO-8859-15
+de_DE			UTF-8
+de_DE.ISO8859-1		ISO-8859-1
+de_DE.ISO8859-15	ISO-8859-15
+en_AU			UTF-8
+en_AU.ISO8859-1		ISO-8859-1
+en_AU.ISO8859-15	ISO-8859-15
+en_AU.US-ASCII		ASCII
+en_CA			UTF-8
+en_CA.ISO8859-1		ISO-8859-1
+en_CA.ISO8859-15	ISO-8859-15
+en_CA.US-ASCII		ASCII
+en_GB			UTF-8
+en_GB.ISO8859-1		ISO-8859-1
+en_GB.ISO8859-15	ISO-8859-15
+en_GB.US-ASCII		ASCII
+en_US			UTF-8
+en_US.ISO8859-1		ISO-8859-1
+en_US.ISO8859-15	ISO-8859-15
+en_US.US-ASCII		ASCII
+es_ES			UTF-8
+es_ES.ISO8859-1		ISO-8859-1
+es_ES.ISO8859-15	ISO-8859-15
+fi_FI			UTF-8
+fi_FI.ISO8859-1		ISO-8859-1
+fi_FI.ISO8859-15	ISO-8859-15
+fr_BE			UTF-8
+fr_BE.ISO8859-1		ISO-8859-1
+fr_BE.ISO8859-15	ISO-8859-15
+fr_CA			UTF-8
+fr_CA.ISO8859-1		ISO-8859-1
+fr_CA.ISO8859-15	ISO-8859-15
+fr_CH			UTF-8
+fr_CH.ISO8859-1		ISO-8859-1
+fr_CH.ISO8859-15	ISO-8859-15
+fr_FR			UTF-8
+fr_FR.ISO8859-1		ISO-8859-1
+fr_FR.ISO8859-15	ISO-8859-15
+hr_HR			UTF-8
+hr_HR.ISO8859-2		ISO-8859-2
+hu_HU			UTF-8
+hu_HU.ISO8859-2		ISO-8859-2
+is_IS			UTF-8
+is_IS.ISO8859-1		ISO-8859-1
+is_IS.ISO8859-15	ISO-8859-15
+it_CH			UTF-8
+it_CH.ISO8859-1		ISO-8859-1
+it_CH.ISO8859-15	ISO-8859-15
+it_IT			UTF-8
+it_IT.ISO8859-1		ISO-8859-1
+it_IT.ISO8859-15	ISO-8859-15
+ja_JP			UTF-8
+ja_JP.EUC		EUC-JP
+ja_JP.SJIS		SHIFT_JIS
+ko_KR			UTF-8
+ko_KR.EUC		EUC-KR
+la_LN			UTF-8
+la_LN.ISO8859-1		ISO-8859-1
+la_LN.ISO8859-2		ISO-8859-2
+la_LN.ISO8859-4		ISO-8859-4
+la_LN.ISO8859-15	ISO-8859-15
+la_LN.US-ASCII		ASCII
+lt_LT			UTF-8
+lt_LT.ISO8859-4		ISO-8859-4
+nl_BE			UTF-8
+nl_BE.ISO8869-1		ISO-8859-1
+nl_BE.ISO8869-15	ISO-8859-15
+nl_NL			UTF-8
+nl_NL.ISO8869-1		ISO-8859-1
+nl_NL.ISO8869-15	ISO-8859-15
+no_NO			UTF-8
+no_NO.ISO8869-1		ISO-8859-1
+no_NO.ISO8869-15	ISO-8859-15
+pl_PL			UTF-8
+pl_PL.ISO8859-2		ISO-8859-2
+pt_PT			UTF-8
+pt_PT.ISO8859-1		ISO-8859-1
+pt_PT.ISO8859-15	ISO-8859-15
+ru_RU			UTF-8
+ru_RU.CP866		CP866
+ru_RU.ISO8859-5		ISO-8859-5
+ru_RU.KOI8-R		KOI8-R
+ru_RU.cp1251		CP1251
+ru_RU.koi8r		KOI8-R
+sl_SI			UTF-8
+sl_SI.ISO8859-2		ISO-8859-2
+sv_SE			UTF-8
+sv_SE.ISO8859-1		ISO-8859-1
+sv_SE.ISO8859-15	ISO-8859-15
+uk_UA			UTF-8
+uk_UA.KOI8-U		KOI8-U
+uk_UA.cp1251		CP1251
+zh_CN			UTF-8
+zh_CN.EUC		GB2312
+zh_TW			UTF-8
+zh_TW.Big5		BIG5
diff -Nurd -x'*~' glib-2.14.6.orig/glib-gettextize.in glib-2.14.6/glib-gettextize.in
--- glib-2.14.6.orig/glib-gettextize.in	2008-02-07 00:25:02.000000000 -0500
+++ glib-2.14.6/glib-gettextize.in	2008-02-23 16:40:48.000000000 -0500
@@ -49,8 +49,8 @@
     ;;
 esac
 
-datadir=@datadir@
 datarootdir=@datarootdir@
+datadir=@datadir@
 
 gettext_dir=$prefix/share/glib-2.0/gettext
 
diff -Nurd -x'*~' glib-2.14.6.orig/ltmain.sh glib-2.14.6/ltmain.sh
--- glib-2.14.6.orig/ltmain.sh	2008-01-30 15:26:55.000000000 -0500
+++ glib-2.14.6/ltmain.sh	2008-02-23 16:34:41.000000000 -0500
@@ -5759,10 +5759,6 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
 	done
       fi
 
diff -Nurd -x'*~' glib-2.14.6.orig/tests/Makefile.in glib-2.14.6/tests/Makefile.in
--- glib-2.14.6.orig/tests/Makefile.in	2008-02-07 00:41:29.000000000 -0500
+++ glib-2.14.6/tests/Makefile.in	2008-02-23 16:34:41.000000000 -0500
@@ -330,7 +330,6 @@
 	module-test				\
 	node-test				\
 	onceinit				\
-	option-test				\
 	patterntest				\
 	printf-test				\
 	queue-test				\
@@ -363,7 +362,7 @@
 	regex-test
 
 
-test_scripts = run-markup-tests.sh run-collate-tests.sh run-bookmark-test.sh
+test_scripts = run-markup-tests.sh run-bookmark-test.sh
 
 test_script_support_programs = markup-test unicode-collate bookmarkfile-test
 
diff -Nurd -x'*~' glib-2.14.6.orig/tests/array-test.c glib-2.14.6/tests/array-test.c
--- glib-2.14.6.orig/tests/array-test.c	2008-02-07 00:24:55.000000000 -0500
+++ glib-2.14.6/tests/array-test.c	2008-02-23 16:34:41.000000000 -0500
@@ -92,10 +92,14 @@
 
   for (i = 0; i < 10000; i++)
     {
-      g_assert (gbarray->data[4*i] == 'a');
-      g_assert (gbarray->data[4*i+1] == 'b');
-      g_assert (gbarray->data[4*i+2] == 'c');
-      g_assert (gbarray->data[4*i+3] == 'd');
+      int idx = 4*i;
+      g_assert (gbarray->data[idx] == 'a');
+      idx++;
+      g_assert (gbarray->data[idx] == 'b');
+      idx++;
+      g_assert (gbarray->data[idx] == 'c');
+      idx++;
+      g_assert (gbarray->data[idx] == 'd');
     }
 
   g_byte_array_free (gbarray, TRUE);
diff -Nurd -x'*~' glib-2.14.6.orig/tests/child-test.c glib-2.14.6/tests/child-test.c
--- glib-2.14.6.orig/tests/child-test.c	2008-02-07 00:24:55.000000000 -0500
+++ glib-2.14.6/tests/child-test.c	2008-02-23 16:34:41.000000000 -0500
@@ -175,7 +175,7 @@
 #ifdef G_OS_WIN32
   system ("ipconfig /all");
 #else
-  system ("/bin/true");
+  system ("/usr/bin/true");
 #endif
 
   alive = 2;
