diff -Nurd -x'*~' glib-2.18.1.orig/ChangeLog glib-2.18.1/ChangeLog
--- glib-2.18.1.orig/ChangeLog	2008-09-17 19:20:31.000000000 -0400
+++ glib-2.18.1/ChangeLog	2008-10-05 10:33:20.000000000 -0400
@@ -1,3 +1,20 @@
+2008-09-30  Behdad Esfahbod  <behdad@gnome.org>
+
+	Bug 554092 – glib doesn't return G_FILE_ERROR_NOENT et al on OS X
+
+	* glib/giounix.c (g_io_unix_read), (g_io_unix_write),
+	(g_io_unix_seek), (g_io_unix_close), (g_io_unix_set_flags),
+	(g_io_unix_get_flags), (g_io_channel_new_file):
+	Like mclasen says: "well, thats the way errno works...,
+	save it or loose it".  Save errno.
+
+2008-09-30  Dan Winship  <danw@gnome.org>
+
+	Bug 554092 – glib doesn't return G_FILE_ERROR_NOENT et al on OS X
+
+	* glib/gslice.c (allocator_memalign): Don't clear errno on
+	non-posix_memalign() platforms.
+
 2008-09-17  Matthias Clasen  <mclasen@redhat.com>
 
 	* === Released 2.18.1 ===
diff -Nurd -x'*~' glib-2.18.1.orig/configure glib-2.18.1/configure
--- glib-2.18.1.orig/configure	2008-09-17 18:54:36.000000000 -0400
+++ glib-2.18.1/configure	2008-09-25 01:46:38.000000000 -0400
@@ -37092,10 +37092,11 @@
 { $as_echo "$as_me:$LINENO: result: $glib_cv___va_copy" >&5
 $as_echo "$glib_cv___va_copy" >&6; }
 
-if test "x$glib_cv_va_copy" = "xyes"; then
-  g_va_copy_func=va_copy
-else if test "x$glib_cv___va_copy" = "xyes"; then
+# Prefer __va_copy, available on both gcc2 and gcc3.
+if test "x$glib_cv___va_copy" = "xyes"; then
   g_va_copy_func=__va_copy
+else if test "x$glib_cv_va_copy" = "xyes"; then
+  g_va_copy_func=va_copy
 fi
 fi
 
@@ -39945,6 +39946,9 @@
 			G_THREAD_CFLAGS="-D_THREAD_SAFE"
 		fi
 		;;
+	*-darwin*)
+		# Nothing needed.
+		;;
 	*-dg-dgux*)  # DG/UX
 		G_THREAD_CFLAGS="-D_REENTRANT -D_POSIX4A_DRAFT10_SOURCE"
 		;;
@@ -46939,9 +46943,10 @@
 #define GLIB_MICRO_VERSION $GLIB_MICRO_VERSION
 "
 
+# Prefer __va_copy, available on both gcc2 and gcc3.
 case xyes in
-x$glib_cv_va_copy)	glib_vacopy='#define G_VA_COPY	va_copy' ;;
 x$glib_cv___va_copy)	glib_vacopy='#define G_VA_COPY	__va_copy' ;;
+x$glib_cv_va_copy)	glib_vacopy='#define G_VA_COPY	va_copy' ;;
 *)			glib_vacopy=''
 esac
 
@@ -48098,8 +48103,11 @@
 /* gcc-2.95.x supports both gnu style and ISO varargs, but if -ansi
  * is passed ISO vararg support is turned off, and there is no work
  * around to turn it on, so we unconditionally turn it off.
+ *
+ * On darwin, use gcc macro to avoid incompatibility between gcc2
+ * and gcc3.
  */
-#if __GNUC__ == 2 && __GNUC_MINOR__ == 95
+#if (__GNUC__ == 2 && __GNUC_MINOR__ == 95) || defined (__APPLE__)
 #  undef G_HAVE_ISO_VARARGS
 #endif
 
diff -Nurd -x'*~' glib-2.18.1.orig/glib/giounix.c glib-2.18.1/glib/giounix.c
--- glib-2.18.1.orig/glib/giounix.c	2008-09-17 18:33:57.000000000 -0400
+++ glib-2.18.1/glib/giounix.c	2008-10-05 10:27:14.000000000 -0400
@@ -190,9 +190,10 @@
 
   if (result < 0)
     {
+      int err = errno;
       *bytes_read = 0;
 
-      switch (errno)
+      switch (err)
         {
 #ifdef EINTR
           case EINTR:
@@ -204,8 +205,8 @@
 #endif
           default:
             g_set_error_literal (err, G_IO_CHANNEL_ERROR,
-                                 g_io_channel_error_from_errno (errno),
-                                 g_strerror (errno));
+                                 g_io_channel_error_from_errno (err),
+                                 g_strerror (err));
             return G_IO_STATUS_ERROR;
         }
     }
@@ -230,9 +231,10 @@
 
   if (result < 0)
     {
+      int err = errno;
       *bytes_written = 0;
 
-      switch (errno)
+      switch (err)
         {
 #ifdef EINTR
           case EINTR:
@@ -244,8 +246,8 @@
 #endif
           default:
             g_set_error_literal (err, G_IO_CHANNEL_ERROR,
-                                 g_io_channel_error_from_errno (errno),
-                                 g_strerror (errno));
+                                 g_io_channel_error_from_errno (err),
+                                 g_strerror (err));
             return G_IO_STATUS_ERROR;
         }
     }
@@ -295,9 +297,10 @@
 
   if (result < 0)
     {
+      int err = errno;
       g_set_error_literal (err, G_IO_CHANNEL_ERROR,
-                           g_io_channel_error_from_errno (errno),
-                           g_strerror (errno));
+                           g_io_channel_error_from_errno (err),
+                           g_strerror (err));
       return G_IO_STATUS_ERROR;
     }
 
@@ -313,9 +316,10 @@
 
   if (close (unix_channel->fd) < 0)
     {
+      int err = errno;
       g_set_error_literal (err, G_IO_CHANNEL_ERROR,
-                           g_io_channel_error_from_errno (errno),
-                           g_strerror (errno));
+                           g_io_channel_error_from_errno (err),
+                           g_strerror (err));
       return G_IO_STATUS_ERROR;
     }
 
@@ -376,9 +380,10 @@
 
   if (fcntl (unix_channel->fd, F_SETFL, fcntl_flags) == -1)
     {
+      int err = errno;
       g_set_error_literal (err, G_IO_CHANNEL_ERROR,
-                           g_io_channel_error_from_errno (errno),
-                           g_strerror (errno));
+                           g_io_channel_error_from_errno (err),
+                           g_strerror (err));
       return G_IO_STATUS_ERROR;
     }
 
@@ -396,8 +401,9 @@
 
   if (fcntl_flags == -1)
     {
+      int err = errno;
       g_warning (G_STRLOC "Error while getting flags for FD: %s (%d)\n",
-		 g_strerror (errno), errno);
+		 g_strerror (err), err);
       return 0;
     }
 
@@ -512,18 +518,20 @@
   fid = open (filename, flags, create_mode);
   if (fid == -1)
     {
+      int err = errno;
       g_set_error_literal (error, G_FILE_ERROR,
-                           g_file_error_from_errno (errno),
-                           g_strerror (errno));
+                           g_file_error_from_errno (err),
+                           g_strerror (err));
       return (GIOChannel *)NULL;
     }
 
   if (fstat (fid, &buffer) == -1) /* In case someone opens a FIFO */
     {
+      int err = errno;
       close (fid);
       g_set_error_literal (error, G_FILE_ERROR,
-                           g_file_error_from_errno (errno),
-                           g_strerror (errno));
+                           g_file_error_from_errno (err),
+                           g_strerror (err));
       return (GIOChannel *)NULL;
     }
 
diff -Nurd -x'*~' glib-2.18.1.orig/glib/gmacros.h glib-2.18.1/glib/gmacros.h
--- glib-2.18.1.orig/glib/gmacros.h	2008-09-17 18:33:57.000000000 -0400
+++ glib-2.18.1/glib/gmacros.h	2008-09-25 04:57:43.000000000 -0400
@@ -120,7 +120,8 @@
 #define G_GNUC_WARN_UNUSED_RESULT
 #endif /* __GNUC__ */
 
-#ifndef G_DISABLE_DEPRECATED
+#if 1
+/* don't supress (if G_DISABLE_DEPRECATED) for fink yet */
 /* Wrap the gcc __PRETTY_FUNCTION__ and __FUNCTION__ variables with
  * macros, so we can refer to them as strings unconditionally.
  * usage not-recommended since gcc-3.0
diff -Nurd -x'*~' glib-2.18.1.orig/glib/gmessages.h glib-2.18.1/glib/gmessages.h
--- glib-2.18.1.orig/glib/gmessages.h	2008-09-17 18:33:57.000000000 -0400
+++ glib-2.18.1/glib/gmessages.h	2008-09-25 04:56:51.000000000 -0400
@@ -121,7 +121,8 @@
                                int             line,
                                const char     *func,
                                const char     *warnexpr);
-#ifndef G_DISABLE_DEPRECATED
+#if 1
+/* don't supress (if G_DISABLE_DEPRECATED) for fink yet */
 void g_assert_warning         (const char *log_domain,
 			       const char *file,
 			       const int   line,
diff -Nurd -x'*~' glib-2.18.1.orig/glib/gslice.c glib-2.18.1/glib/gslice.c
--- glib-2.18.1.orig/glib/gslice.c	2008-09-17 18:33:57.000000000 -0400
+++ glib-2.18.1/glib/gslice.c	2008-10-05 10:30:48.000000000 -0400
@@ -1131,19 +1131,18 @@
                     gsize memsize)
 {
   gpointer aligned_memory = NULL;
-  gint err = ENOMEM;
 #if     HAVE_COMPLIANT_POSIX_MEMALIGN
+  gint err;
   err = posix_memalign (&aligned_memory, alignment, memsize);
+  if (!aligned_memory)
+    errno = err;
 #elif   HAVE_MEMALIGN
-  errno = 0;
   aligned_memory = memalign (alignment, memsize);
-  err = errno;
 #elif   HAVE_VALLOC
-  errno = 0;
   aligned_memory = valloc (memsize);
-  err = errno;
 #else
   /* simplistic non-freeing page allocator */
+  gint err = ENOMEM;
   mem_assert (alignment == sys_page_size);
   mem_assert (memsize <= sys_page_size);
   if (!compat_valloc_trash)
@@ -1162,9 +1161,9 @@
         }
     }
   aligned_memory = g_trash_stack_pop (&compat_valloc_trash);
-#endif
   if (!aligned_memory)
     errno = err;
+#endif
   return aligned_memory;
 }
 
diff -Nurd -x'*~' glib-2.18.1.orig/glib/gtester-report glib-2.18.1/glib/gtester-report
--- glib-2.18.1.orig/glib/gtester-report	2008-09-17 18:33:57.000000000 -0400
+++ glib-2.18.1/glib/gtester-report	2008-09-25 03:23:51.000000000 -0400
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/env python
 # GLib Testing Framework Utility			-*- Mode: python; -*-
 # Copyright (C) 2007 Imendio AB
 # Authors: Tim Janik
diff -Nurd -x'*~' glib-2.18.1.orig/glib/gutils.h glib-2.18.1/glib/gutils.h
--- glib-2.18.1.orig/glib/gutils.h	2008-09-17 18:33:57.000000000 -0400
+++ glib-2.18.1/glib/gutils.h	2008-09-25 03:59:41.000000000 -0400
@@ -101,7 +101,7 @@
 #  define G_INLINE_FUNC
 #  undef  G_CAN_INLINE
 #elif defined (__GNUC__) 
-#  define G_INLINE_FUNC static __inline __attribute__ ((unused))
+#  define G_INLINE_FUNC static inline
 #elif defined (G_CAN_INLINE) 
 #  define G_INLINE_FUNC static inline
 #else /* can't inline */
diff -Nurd -x'*~' glib-2.18.1.orig/glib/libcharset/Makefile.in glib-2.18.1/glib/libcharset/Makefile.in
--- glib-2.18.1.orig/glib/libcharset/Makefile.in	2008-09-17 18:54:31.000000000 -0400
+++ glib-2.18.1/glib/libcharset/Makefile.in	2008-09-25 01:54:37.000000000 -0400
@@ -280,7 +280,7 @@
 	codeset.m4 update.sh make-patch.sh libcharset-glib.patch
 TEST_PROGS = 
 INCLUDES = \
-	-DLIBDIR=\"$(libdir)\" -I$(top_srcdir)
+	-DLIBDIR=\"$(sysconfdir)/glib-2.0\" -I$(top_srcdir)
 
 noinst_LTLIBRARIES = libcharset.la
 libcharset_la_SOURCES = \
@@ -288,8 +288,8 @@
 	localcharset.h		\
 	localcharset.c
 
-charset_alias = $(DESTDIR)$(libdir)/charset.alias
-charset_tmp = $(DESTDIR)$(libdir)/charset.tmp
+charset_alias = $(DESTDIR)$(sysconfdir)/glib-2.0/charset.alias
+charset_tmp = $(DESTDIR)$(sysconfdir)/glib-2.0/charset.tmp
 SUFFIXES = .sed .sin
 CLEANFILES = charset.alias ref-add.sed ref-del.sed
 all: all-am
@@ -601,7 +601,7 @@
 # run make test as part of make check
 check-local: test
 install-exec-local: all-local
-	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(mkinstalldirs) $(DESTDIR)$(sysconfdir)/glib-2.0
 	if test -f $(charset_alias); then \
 	  sed -f ref-add.sed $(charset_alias) > $(charset_tmp) ; \
 	  $(INSTALL_DATA) $(charset_tmp) $(charset_alias) ; \
diff -Nurd -x'*~' glib-2.18.1.orig/glib/libcharset/charset.alias glib-2.18.1/glib/libcharset/charset.alias
--- glib-2.18.1.orig/glib/libcharset/charset.alias	1969-12-31 19:00:00.000000000 -0500
+++ glib-2.18.1/glib/libcharset/charset.alias	2008-09-25 01:46:38.000000000 -0400
@@ -0,0 +1,111 @@
+# This file contains a table of character encoding aliases,
+# suitable for operating system 'darwin'.
+C			ASCII
+bg_BG			UTF-8
+bg_BG.CP1251		CP1251
+cs_CZ			UTF-8
+cs_CZ.ISO8859-2		ISO-8859-2
+da_DK			UTF-8
+da_DK.ISO8859-1		ISO-8859-1
+da_DK.ISO8859-15	ISO-8859-15
+de_AT			UTF-8
+de_AT.ISO8859-1		ISO-8859-1
+de_AT.ISO8859-15	ISO-8859-15
+de_CH			UTF-8
+de_CH.ISO8859-1		ISO-8859-1
+de_CH.ISO8859-15	ISO-8859-15
+de_DE			UTF-8
+de_DE.ISO8859-1		ISO-8859-1
+de_DE.ISO8859-15	ISO-8859-15
+en_AU			UTF-8
+en_AU.ISO8859-1		ISO-8859-1
+en_AU.ISO8859-15	ISO-8859-15
+en_AU.US-ASCII		ASCII
+en_CA			UTF-8
+en_CA.ISO8859-1		ISO-8859-1
+en_CA.ISO8859-15	ISO-8859-15
+en_CA.US-ASCII		ASCII
+en_GB			UTF-8
+en_GB.ISO8859-1		ISO-8859-1
+en_GB.ISO8859-15	ISO-8859-15
+en_GB.US-ASCII		ASCII
+en_US			UTF-8
+en_US.ISO8859-1		ISO-8859-1
+en_US.ISO8859-15	ISO-8859-15
+en_US.US-ASCII		ASCII
+es_ES			UTF-8
+es_ES.ISO8859-1		ISO-8859-1
+es_ES.ISO8859-15	ISO-8859-15
+fi_FI			UTF-8
+fi_FI.ISO8859-1		ISO-8859-1
+fi_FI.ISO8859-15	ISO-8859-15
+fr_BE			UTF-8
+fr_BE.ISO8859-1		ISO-8859-1
+fr_BE.ISO8859-15	ISO-8859-15
+fr_CA			UTF-8
+fr_CA.ISO8859-1		ISO-8859-1
+fr_CA.ISO8859-15	ISO-8859-15
+fr_CH			UTF-8
+fr_CH.ISO8859-1		ISO-8859-1
+fr_CH.ISO8859-15	ISO-8859-15
+fr_FR			UTF-8
+fr_FR.ISO8859-1		ISO-8859-1
+fr_FR.ISO8859-15	ISO-8859-15
+hr_HR			UTF-8
+hr_HR.ISO8859-2		ISO-8859-2
+hu_HU			UTF-8
+hu_HU.ISO8859-2		ISO-8859-2
+is_IS			UTF-8
+is_IS.ISO8859-1		ISO-8859-1
+is_IS.ISO8859-15	ISO-8859-15
+it_CH			UTF-8
+it_CH.ISO8859-1		ISO-8859-1
+it_CH.ISO8859-15	ISO-8859-15
+it_IT			UTF-8
+it_IT.ISO8859-1		ISO-8859-1
+it_IT.ISO8859-15	ISO-8859-15
+ja_JP			UTF-8
+ja_JP.EUC		EUC-JP
+ja_JP.SJIS		SHIFT_JIS
+ko_KR			UTF-8
+ko_KR.EUC		EUC-KR
+la_LN			UTF-8
+la_LN.ISO8859-1		ISO-8859-1
+la_LN.ISO8859-2		ISO-8859-2
+la_LN.ISO8859-4		ISO-8859-4
+la_LN.ISO8859-15	ISO-8859-15
+la_LN.US-ASCII		ASCII
+lt_LT			UTF-8
+lt_LT.ISO8859-4		ISO-8859-4
+nl_BE			UTF-8
+nl_BE.ISO8869-1		ISO-8859-1
+nl_BE.ISO8869-15	ISO-8859-15
+nl_NL			UTF-8
+nl_NL.ISO8869-1		ISO-8859-1
+nl_NL.ISO8869-15	ISO-8859-15
+no_NO			UTF-8
+no_NO.ISO8869-1		ISO-8859-1
+no_NO.ISO8869-15	ISO-8859-15
+pl_PL			UTF-8
+pl_PL.ISO8859-2		ISO-8859-2
+pt_PT			UTF-8
+pt_PT.ISO8859-1		ISO-8859-1
+pt_PT.ISO8859-15	ISO-8859-15
+ru_RU			UTF-8
+ru_RU.CP866		CP866
+ru_RU.ISO8859-5		ISO-8859-5
+ru_RU.KOI8-R		KOI8-R
+ru_RU.cp1251		CP1251
+ru_RU.koi8r		KOI8-R
+sl_SI			UTF-8
+sl_SI.ISO8859-2		ISO-8859-2
+sv_SE			UTF-8
+sv_SE.ISO8859-1		ISO-8859-1
+sv_SE.ISO8859-15	ISO-8859-15
+uk_UA			UTF-8
+uk_UA.KOI8-U		KOI8-U
+uk_UA.cp1251		CP1251
+zh_CN			UTF-8
+zh_CN.EUC		GB2312
+zh_TW			UTF-8
+zh_TW.Big5		BIG5
diff -Nurd -x'*~' glib-2.18.1.orig/glib-gettextize.in glib-2.18.1/glib-gettextize.in
--- glib-2.18.1.orig/glib-gettextize.in	2008-09-17 18:34:01.000000000 -0400
+++ glib-2.18.1/glib-gettextize.in	2008-09-25 01:46:38.000000000 -0400
@@ -49,8 +49,8 @@
     ;;
 esac
 
-datadir=@datadir@
 datarootdir=@datarootdir@
+datadir=@datadir@
 
 gettext_dir=$prefix/share/glib-2.0/gettext
 
diff -Nurd -x'*~' glib-2.18.1.orig/ltmain.sh glib-2.18.1/ltmain.sh
--- glib-2.18.1.orig/ltmain.sh	2008-08-29 18:27:25.000000000 -0400
+++ glib-2.18.1/ltmain.sh	2008-09-25 01:46:38.000000000 -0400
@@ -5785,10 +5785,6 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
 	done
       fi
 
diff -Nurd -x'*~' glib-2.18.1.orig/tests/Makefile.in glib-2.18.1/tests/Makefile.in
--- glib-2.18.1.orig/tests/Makefile.in	2008-09-17 18:54:33.000000000 -0400
+++ glib-2.18.1/tests/Makefile.in	2008-09-25 01:57:51.000000000 -0400
@@ -678,7 +678,7 @@
 	uri-test				\
 	regex-test
 
-test_scripts = run-markup-tests.sh run-collate-tests.sh run-bookmark-test.sh
+test_scripts = run-markup-tests.sh run-bookmark-test.sh
 test_script_support_programs = markup-test unicode-collate bookmarkfile-test
 TESTS_ENVIRONMENT = srcdir=$(srcdir) \
 	LIBCHARSET_ALIAS_DIR=$(top_builddir)/glib/libcharset \
diff -Nurd -x'*~' glib-2.18.1.orig/tests/child-test.c glib-2.18.1/tests/child-test.c
--- glib-2.18.1.orig/tests/child-test.c	2008-09-17 18:33:52.000000000 -0400
+++ glib-2.18.1/tests/child-test.c	2008-09-25 01:48:08.000000000 -0400
@@ -175,7 +175,7 @@
 #ifdef G_OS_WIN32
   system ("ipconfig /all");
 #else
-  system ("/bin/true");
+  system ("/usr/bin/true");
 #endif
 
   alive = 2;
