diff -Naur gnome-system-monitor-2.6.0-old/src/Makefile.in gnome-system-monitor-2.6.0/src/Makefile.in
--- gnome-system-monitor-2.6.0-old/src/Makefile.in	Sun Mar 21 13:37:22 2004
+++ gnome-system-monitor-2.6.0/src/Makefile.in	Mon Apr 19 17:58:41 2004
@@ -128,7 +128,7 @@
 	defaulttable.h
 
 
-gnome_system_monitor_LDADD = @PROCMAN_LIBS@
+gnome_system_monitor_LDADD = @PROCMAN_LIBS@ -lXau -lX11
 
 specdir = $(datadir)/procman
 
@@ -151,14 +151,14 @@
 
 DEFS = @DEFS@ -I. -I$(srcdir) -I..
 CPPFLAGS = @CPPFLAGS@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ -undefined dynamic_lookup
 LIBS = @LIBS@
 gnome_system_monitor_OBJECTS =  cellrenderer.o procman.o interface.o \
 callbacks.o load-graph.o proctable.o prettytable.o util.o infoview.o \
 procactions.o procdialogs.o memmaps.o favorites.o
 gnome_system_monitor_DEPENDENCIES = 
 gnome_system_monitor_LDFLAGS = 
-CFLAGS = @CFLAGS@
+CFLAGS = @CFLAGS@ -D__APPLE__
 COMPILE = $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
 LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(LDFLAGS) -o $@
diff -Naur gnome-system-monitor-2.6.0-old/src/util.c gnome-system-monitor-2.6.0/src/util.c
--- gnome-system-monitor-2.6.0-old/src/util.c	Sun Feb  8 10:04:13 2004
+++ gnome-system-monitor-2.6.0/src/util.c	Mon Apr 19 17:58:41 2004
@@ -49,7 +49,7 @@
  */
 #define OPEN_TTY() getpt()
 
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__APPLE__)
 /* FreeBSD doesn't have getpt(). This function emulates it's behaviour. */
 int getpt (void);
 
@@ -242,8 +242,15 @@
 		}
 
 		if ((env = getenv ("XAUTHORITY")) == NULL) {
+#ifdef HAVE_ASPRINTF
 			if (asprintf (&env, "XAUTHORITY=%s/.Xauthority", home) > 0)
 				putenv (env);
+#else
+			if ((env = malloc (1024))) {
+				snprintf (env, 1024, "XAUTHORITY=%s/.Xauthority", home);
+				env[1023] = '\0';
+			}
+#endif
 			else {
 				perror ("Unable to allocate memory chunk");
 				_exit (-1);
@@ -251,8 +258,15 @@
 		}
 
 		if ((env = getenv ("ICEAUTHORITY")) == NULL) {
+#ifdef HAVE_ASPRINTF
 			if (asprintf (&env, "ICEAUTHORITY=%s/.ICEauthority", home) > 0)
 				putenv (env);
+#else
+			if ((env = malloc (1024))) {
+				snprintf (env, 1024, "ICEAUTHORITY=%s/.ICEauthority", home);
+				env[1023] = '\0';
+			}
+#endif
 			else {
 				perror ("Unable to allocate memory chunk");
 				_exit (-1);
