diff -uNr gnome-vfs-2.18.0/modules/file-method-acl.c gnome-vfs-2.18.0-new/modules/file-method-acl.c
--- gnome-vfs-2.18.0/modules/file-method-acl.c	2007-03-05 04:05:05.000000000 -0500
+++ gnome-vfs-2.18.0-new/modules/file-method-acl.c	2007-03-12 15:52:59.000000000 -0400
@@ -34,6 +34,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <sys/types.h>
+#include <sys/time.h>
 #ifdef HAVE_SYS_RESOURCE_H
 #include <sys/resource.h>
 #endif
@@ -49,10 +50,12 @@
 #ifdef HAVE_POSIX_ACL
 # include <acl/libacl.h>
 #define HAVE_ACL 1
+#define HAVE_ACL
 #endif
 
 #ifdef HAVE_SOLARIS_ACL
 # include <sys/acl.h>
+#define HAVE_ACL
 #define HAVE_ACL 1
 #endif
 
@@ -63,6 +66,7 @@
 #define CMD_PERM_EXECUTE        1
 #define CMD_PERM_COND_EXECUTE   8
 
+#ifdef HAVE_ACL
 
 #ifdef HAVE_ACL
 
@@ -693,6 +697,7 @@
 #endif /* HAVE_SOLARIS_ACL */
 
 
+#endif /* HAVE_ACL */
 
 /* ************************************************************************** */
 /* Common */
