diff -Nurd -x'*~' gnome-mag-0.12.4.orig/configure gnome-mag-0.12.4/configure
--- gnome-mag-0.12.4.orig/configure	2006-03-13 10:54:51.000000000 -0500
+++ gnome-mag-0.12.4/configure	2006-04-06 00:58:04.000000000 -0400
@@ -23913,7 +23913,7 @@
 
 if test "x$XDAMAGE_LIBS" = x; then
   save_LIBS="$LIBS"
-  for xdamagepath in /usr/X11R6/lib /usr/openwin/lib /usr/openwin/sfw/lib /opt/fdo/lib $x_libraries; do
+  for xdamagepath in ; do
     LIBS="-L$xdamagepath -lXdamage -lX11"
     case "$host" in
     *solaris*) XDAMAGE_RPATH_FLAGS="-R$xdamagepath" ;;
@@ -24344,7 +24344,7 @@
 
 if test "x$XFIXES_LIBS" = x; then
   save_LIBS="$LIBS"
-  for xfixespath in /usr/X11R6/lib /usr/openwin/lib /usr/openwin/sfw/lib /opt/fdo/lib $x_libraries; do
+  for xfixespath in ; do
     LIBS="-L$xfixespath -lXfixes -lX11"
     case "$host" in
     *solaris*) XFIXES_RPATH_FLAGS="-R$xfixespath" ;;
