diff -uNr gnome-print-0.37/libgnomeprint/gnome-print-pclc.c gnome-print-0.37-new/libgnomeprint/gnome-print-pclc.c
--- gnome-print-0.37/libgnomeprint/gnome-print-pclc.c	Sat Jul 22 15:59:09 2000
+++ gnome-print-0.37-new/libgnomeprint/gnome-print-pclc.c	Mon Nov  3 10:49:28 2003
@@ -16,6 +16,8 @@
 4.  The application and the print dialog will load the
     job specific parameters */
 
+PCLJobData * jobdata;
+
 void
 pclc_new_job_data (void)
 {
diff -uNr gnome-print-0.37/libgnomeprint/gnome-print-pclr.c gnome-print-0.37-new/libgnomeprint/gnome-print-pclr.c
--- gnome-print-0.37/libgnomeprint/gnome-print-pclr.c	Thu May 24 16:35:07 2001
+++ gnome-print-0.37-new/libgnomeprint/gnome-print-pclr.c	Mon Nov  3 10:50:11 2003
@@ -62,7 +62,6 @@
 #include <libgnomeprint/gnome-print-encode.h>
 #include <libgnomeprint/gnome-print-encode-private.h>
 
-PCLJobData * jobdata;
 static GnomePrintRGBPClass *pclr_parent_class;
 
 static int
diff -uNr gnome-print-0.37/libgnomeprint/gnome-print-pclv.c gnome-print-0.37-new/libgnomeprint/gnome-print-pclv.c
--- gnome-print-0.37/libgnomeprint/gnome-print-pclv.c	Thu May 24 16:35:07 2001
+++ gnome-print-0.37-new/libgnomeprint/gnome-print-pclv.c	Mon Nov  3 10:50:23 2003
@@ -53,7 +53,6 @@
 #include <libgnomeprint/gnome-print-pclc.h>
 #include <libgnomeprint/gnome-print-encode-private.h>
 
-PCLJobData * jobdata;
 static GnomePrintRGBPClass *pclv_parent_class;
 
 static int
diff -uNr gnome-print-0.37/run-gnome-font-install gnome-print-0.37-new/run-gnome-font-install
--- gnome-print-0.37/run-gnome-font-install	Mon Jan 14 10:43:06 2002
+++ gnome-print-0.37-new/run-gnome-font-install	Mon Nov  3 10:29:55 2003
@@ -10,7 +10,7 @@
 my $datadir = $ARGV[1];
 my $srcdir = $ARGV[2];
 my $sysconfdir = $ARGV[3];
-if (!$sysconfdir) {$sysconfdir="/etc";}
+if (!$sysconfdir) {$sysconfdir="@PREFIX@/etc";}
 
 die "Error: installer not executable.\n" unless -f $installer && -x $installer;
 die "Error: datadir is not a directory.\n" unless -d $datadir;
@@ -18,6 +18,10 @@
 
 my @default_paths = 
     (
+     '@PREFIX@/share/ghostscript/fonts',
+     '@PREFIX@/lib/ghostscript/fonts',
+     '@PREFIX@/share/fonts/default/Type1',
+     '@PREFIX@/share/fonts/default/ghostscript',
      '/usr/share/ghostscript/fonts',
      '/usr/lib/ghostscript/fonts',
      '/usr/share/fonts/default/Type1',
@@ -143,6 +147,12 @@
 	       "--aliases=$datadir/fonts/adobe-urw.font",
 	       "--aliases=$datadir/gnome/fonts/adobe-urw.font",
 	       "$datadir/fonts/afms",
+	       '@PREFIX@/lib/texmf/fonts/afm',
+	       '@PREFIX@/lib/texmf/fonts/type1/adobe',
+	       '@PREFIX@/lib/texmf/fonts/type1/omega',
+	       '@PREFIX@/share/texmf/fonts/afm',
+	       '@PREFIX@/share/texmf/fonts/type1/adobe',
+	       '@PREFIX@/share/texmf/fonts/type1/omega',
 	       '/usr/lib/texmf/fonts/afm',
 	       '/usr/lib/texmf/fonts/type1/adobe',
 	       '/usr/lib/texmf/fonts/type1/omega',
diff -ruN gnome-print-0.37/installer/gf-pfb.c gnome-print-0.37-patched/installer/gf-pfb.c
--- gnome-print-0.37/installer/gf-pfb.c	2002-01-18 16:23:00.000000000 -0800
+++ gnome-print-0.37-patched/installer/gf-pfb.c	2007-07-14 10:36:51.000000000 -0700
@@ -12,6 +12,7 @@
 #include <fcntl.h>
 #include <string.h>
 #include <ctype.h>
+#include <ft2build.h>
 #include <freetype/freetype.h>
 #include "gf-pfb.h"
 
diff -ruN gnome-print-0.37/installer/gf-ttf.c gnome-print-0.37-patched/installer/gf-ttf.c
--- gnome-print-0.37/installer/gf-ttf.c	2001-11-05 06:07:28.000000000 -0800
+++ gnome-print-0.37-patched/installer/gf-ttf.c	2007-07-14 10:37:10.000000000 -0700
@@ -32,6 +32,7 @@
 #include <fcntl.h>
 #include <string.h>
 #include <ctype.h>
+#include <ft2build.h>
 #include <freetype/freetype.h>
 
 #include "gf-ttf.h"
diff -ruN gnome-print-0.37/installer/gnome-font-install.c gnome-print-0.37-patched/installer/gnome-font-install.c
--- gnome-print-0.37/installer/gnome-font-install.c	2002-01-22 13:44:55.000000000 -0800
+++ gnome-print-0.37-patched/installer/gnome-font-install.c	2007-07-14 10:33:26.000000000 -0700
@@ -26,6 +26,7 @@
 #include <parser.h>
 #include <xmlmemory.h>
 /* End of ugly thing */
+#include <ft2build.h>
 #include <freetype/freetype.h>
 #include <libgnome/gnome-defs.h>
 #include <libgnome/gnome-util.h>
diff -ruN gnome-print-0.37/libgnomeprint/gnome-font-private.h gnome-print-0.37-patched/libgnomeprint/gnome-font-private.h
--- gnome-print-0.37/libgnomeprint/gnome-font-private.h	2001-12-05 14:07:58.000000000 -0800
+++ gnome-print-0.37-patched/libgnomeprint/gnome-font-private.h	2007-07-14 10:16:36.000000000 -0700
@@ -22,6 +22,7 @@
 
 BEGIN_GNOME_DECLS
 
+#include <ft2build.h>
 #include <freetype/freetype.h>
 #include <libgnomeprint/gp-character-block.h>
 #include <libgnomeprint/gp-fontmap.h>
diff -ruN gnome-print-0.37/libgnomeprint/parseTT.h gnome-print-0.37-patched/libgnomeprint/parseTT.h
--- gnome-print-0.37/libgnomeprint/parseTT.h	2001-11-05 11:27:09.000000000 -0800
+++ gnome-print-0.37-patched/libgnomeprint/parseTT.h	2007-07-14 10:29:33.000000000 -0700
@@ -22,6 +22,7 @@
 BEGIN_GNOME_DECLS
 
 #include <glib.h>
+#include <ft2build.h>
 #include <freetype/freetype.h>
 
 /*
