diff -Nurd -x'*~' gnome-python-desktop-2.26.0.orig/configure gnome-python-desktop-2.26.0/configure
--- gnome-python-desktop-2.26.0.orig/configure	2009-03-14 15:19:55.000000000 -0400
+++ gnome-python-desktop-2.26.0/configure	2009-06-09 18:59:10.000000000 -0400
@@ -25058,8 +25058,11 @@
 $build_evince            || echo     evince
 
 
-echo "*****************************************************************************"
-echo "*** WARNING: You are using a deprecated build system (autotools)."
-echo "***          You should be using WAF instead; see the file INSTALL.WAF."
-echo "***          The autotools build system will be removed in a future version."
-echo "*****************************************************************************"
+# make sure all expected modules will be built
+# (missing deps just disable modules instead of crashing ./configure)
+if $build_applet && $build_gnomeprint && $build_gnomeprintui && $build_gtksourceview && $build_wnck && $build_gtop && $build_rsvg && $build_gnomekeyring && $build_gnomedesktop; then
+  :
+else
+  echo "Incomplete package configuration."
+  exit 1
+fi
