diff -Naur gnome-panel-2.6.0-old/applets/clock/clock.c gnome-panel-2.6.0/applets/clock/clock.c
--- gnome-panel-2.6.0-old/applets/clock/clock.c	Sat Mar 13 15:34:14 2004
+++ gnome-panel-2.6.0/applets/clock/clock.c	Mon Apr  5 13:07:38 2004
@@ -47,7 +47,9 @@
 #include <dirent.h>
 #include <string.h>
 #include <time.h>
+#if !defined(__APPLE__)
 #include <langinfo.h>
+#endif
 
 #include <panel-applet.h>
 #include <panel-applet-gconf.h>
@@ -524,9 +526,14 @@
 clock_locale_format (void)
 {
         const char *am;
-
-        am = nl_langinfo (AM_STR);
-        return (am[0] == '\0') ? CLOCK_FORMAT_24 : CLOCK_FORMAT_12;
+	#if !defined(__APPLE__)
+        	am = nl_langinfo (AM_STR);
+        	return (am[0] == '\0') ? CLOCK_FORMAT_24 : CLOCK_FORMAT_12;
+	#else
+		/* on Darwin, just set it to 24 because there is no nl_langinfo
+*/
+		return CLOCK_FORMAT_24;
+	#endif
 }
 
 #ifdef HAVE_LIBECAL
diff -Naur gnome-panel-2.6.0-old/help/C/clock/clock-C.omf gnome-panel-2.6.0/help/C/clock/clock-C.omf
--- gnome-panel-2.6.0-old/help/C/clock/clock-C.omf	Thu Feb 12 04:35:35 2004
+++ gnome-panel-2.6.0/help/C/clock/clock-C.omf	Mon Apr  5 13:05:39 2004
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="UTF-8"?>
+<?xml version="1.0"?>
 <omf>
   <resource>
     <creator>
diff -Naur gnome-panel-2.6.0-old/ltmain.sh gnome-panel-2.6.0/ltmain.sh
--- gnome-panel-2.6.0-old/ltmain.sh	Sun Feb 15 06:51:13 2004
+++ gnome-panel-2.6.0/ltmain.sh	Mon Apr  5 13:05:39 2004
@@ -3745,6 +3745,11 @@
 	  eval test_cmds=\"$archive_expsym_cmds\"
 	  cmds=$archive_expsym_cmds
 	else
+	  if test "x$verstring" = "x0.0"; then
+            tmp_verstring=
+          else
+            tmp_verstring="$verstring"
+          fi
 	  eval test_cmds=\"$archive_cmds\"
 	  cmds=$archive_cmds
 	  fi
@@ -5247,10 +5252,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test "$need_relink" = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
