
--- gst-plugins-bad-0.10.3/ext/sdl/sdlaudiosink.c	Tue May  2 12:16:32 2006
+++ gst-plugins-bad-0.10.3-new/ext/sdl/sdlaudiosink.c	Fri May  5 10:23:09 2006
@@ -389,19 +389,19 @@
 unable_open:
   {
     GST_ELEMENT_ERROR (sdlaudio, RESOURCE, OPEN_READ,
-        ("Unable to open audio: %s", SDL_GetError ()), NULL);
+        ("Unable to open audio: %s", SDL_GetError ()), (NULL));
     return FALSE;
   }
 wrong_format:
   {
     GST_ELEMENT_ERROR (sdlaudio, RESOURCE, OPEN_READ,
-        ("Unable to get format %d", spec->format), NULL);
+        ("Unable to get format %d", spec->format), (NULL));
     return FALSE;
   }
 dodgy_width:
   {
     GST_ELEMENT_ERROR (sdlaudio, RESOURCE, OPEN_READ,
-        ("unexpected width %d", spec->width), NULL);
+        ("unexpected width %d", spec->width), (NULL));
     return FALSE;
   }
 }
