diff -Nurd gst-plugins-bad-0.10.17/ext/mpeg2enc/gstmpeg2encoptions.cc gst-plugins-bad-0.10.17-new/ext/mpeg2enc/gstmpeg2encoptions.cc
--- gst-plugins-bad-0.10.17/ext/mpeg2enc/gstmpeg2encoptions.cc	2009-05-11 17:18:13.000000000 -0400
+++ gst-plugins-bad-0.10.17-new/ext/mpeg2enc/gstmpeg2encoptions.cc	2010-02-11 12:53:44.000000000 -0500
@@ -30,6 +30,8 @@
 
 #include <gst/gst.h>
 
+int force_cbr = 0;
+
 /*
  * Property enumeration.
  */
diff -Nurd gst-plugins-bad-0.10.17/tests/examples/camerabin/Makefile.am gst-plugins-bad-0.10.17-new/tests/examples/camerabin/Makefile.am
--- gst-plugins-bad-0.10.17/tests/examples/camerabin/Makefile.am	2009-09-11 11:35:40.000000000 -0400
+++ gst-plugins-bad-0.10.17-new/tests/examples/camerabin/Makefile.am	2010-02-11 12:54:04.000000000 -0500
@@ -15,7 +15,7 @@
         $(GST_PLUGINS_BASE_LIBS) \
         -lgstinterfaces-@GST_MAJORMINOR@ \
         $(GST_LIBS) \
-        $(GLADE_LIBS) 
+        $(GLADE_LIBS) $(X_LIBS)
 
 gladedir = $(datadir)/gstreamer-@GST_MAJORMINOR@/camera-apps
 glade_DATA = $(GST_CAMERABIN_GLADE_FILES)
@@ -31,7 +31,7 @@
 
 gst_camera_perf_SOURCES = gst-camera-perf.c
 gst_camera_perf_CFLAGS  = $(GST_CFLAGS)
-gst_camera_perf_LDADD = $(GST_LIBS) 
+gst_camera_perf_LDADD = $(GST_LIBS) $(X_LIBS)
 
 bin_PROGRAMS = gst-camera-perf $(GST_CAMERABIN_GTK_EXAMPLES)
 
diff -Nurd gst-plugins-bad-0.10.17/tests/examples/camerabin/Makefile.in gst-plugins-bad-0.10.17-new/tests/examples/camerabin/Makefile.in
--- gst-plugins-bad-0.10.17/tests/examples/camerabin/Makefile.in	2009-11-16 15:35:38.000000000 -0500
+++ gst-plugins-bad-0.10.17-new/tests/examples/camerabin/Makefile.in	2010-02-11 13:03:58.000000000 -0500
@@ -522,14 +522,14 @@
 @HAVE_GLADE_TRUE@@HAVE_GTK_TRUE@        $(GST_PLUGINS_BASE_LIBS) \
 @HAVE_GLADE_TRUE@@HAVE_GTK_TRUE@        -lgstinterfaces-@GST_MAJORMINOR@ \
 @HAVE_GLADE_TRUE@@HAVE_GTK_TRUE@        $(GST_LIBS) \
-@HAVE_GLADE_TRUE@@HAVE_GTK_TRUE@        $(GLADE_LIBS) 
+@HAVE_GLADE_TRUE@@HAVE_GTK_TRUE@        $(GLADE_LIBS) $(X_LIBS)
 
 @HAVE_GLADE_TRUE@@HAVE_GTK_TRUE@gladedir = $(datadir)/gstreamer-@GST_MAJORMINOR@/camera-apps
 @HAVE_GLADE_TRUE@@HAVE_GTK_TRUE@glade_DATA = $(GST_CAMERABIN_GLADE_FILES)
 @HAVE_GLADE_TRUE@@HAVE_GTK_TRUE@INCLUDES = -DCAMERA_APPS_GLADEDIR=\""$(gladedir)"\" 
 gst_camera_perf_SOURCES = gst-camera-perf.c
 gst_camera_perf_CFLAGS = $(GST_CFLAGS)
-gst_camera_perf_LDADD = $(GST_LIBS) 
+gst_camera_perf_LDADD = $(GST_LIBS) $(X_LIBS)
 EXTRA_DIST = $(GST_CAMERABIN_GLADE_FILES)
 all: all-am
 
