--- gst-plugins-0.8.8/ext/cdparanoia/gstcdparanoia.c	Sat Mar  5 11:33:30 2005
+++ gst-plugins-0.8.8-new/ext/cdparanoia/gstcdparanoia.c	Sun Mar 20 18:25:09 2005
@@ -43,8 +43,9 @@
 #define CD_SECS              60 /* seconds per minute */
 #define CD_FRAMES            75 /* frames per second */
 
-#include "gstcdparanoia.h"
+#undef VERSION
 
+#include "gstcdparanoia.h"
 
 static GstElementDetails cdparanoia_details = {
   "CD Audio (cdda) Source, Paranoia IV",
@@ -1227,4 +1228,4 @@
     GST_VERSION_MINOR,
     "cdparanoia",
     "Read audio from CD in paranoid mode",
-    plugin_init, VERSION, "GPL", GST_PACKAGE, GST_ORIGIN)
+    plugin_init, GST_PLUGINS_VERSION, "GPL", GST_PACKAGE, GST_ORIGIN)
--- gst-plugins-0.8.8/ext/dvdread/dvdreadsrc.h	Sat Nov 27 12:20:26 2004
+++ gst-plugins-0.8.8-new/ext/dvdread/dvdreadsrc.h	Sun Mar 20 19:31:07 2005
@@ -24,7 +24,7 @@
 
 G_BEGIN_DECLS
 
-GstElementDetails dvdreadsrc_details;
+extern GstElementDetails dvdreadsrc_details;
 
 #define GST_TYPE_DVDREADSRC \
   (dvdreadsrc_get_type())
--- gst-plugins-0.8.8/ext/mpeg2enc/gstmpeg2encoptions.cc	Mon Mar 15 14:31:53 2004
+++ gst-plugins-0.8.8-new/ext/mpeg2enc/gstmpeg2encoptions.cc	Sun Mar 20 19:49:22 2005
@@ -25,6 +25,12 @@
 
 #include "gstmpeg2encoptions.hh"
 
+#ifdef __APPLE__
+#include <sys/types.h>
+#include <sys/param.h>
+#include <sys/sysctl.h>
+#endif
+
 /*
  * Property enumeration.
  */
@@ -271,7 +277,17 @@
 MPEG2EncOptions ()
 {
   /* autodetect number of CPUs */
+#ifdef __APPLE__
+    int mib[2];
+    size_t len = sizeof(num_cpus);
+    mib[0] = CTL_HW;
+    mib[1] = HW_NCPU;
+    if (! sysctl(mib, 2, (void *) &num_cpus, &len, NULL, 0) == 0) {
+        num_cpus = 1;
+    }
+#else
   num_cpus = sysconf (_SC_NPROCESSORS_ONLN);
+#endif
   if (num_cpus < 0)
     num_cpus = 1;
   if (num_cpus > 32)
--- gst-plugins-0.8.8/gst/audioconvert/Makefile.in	Mon Mar  7 11:43:21 2005
+++ gst-plugins-0.8.8-new/gst/audioconvert/Makefile.in	Sun Mar 20 18:25:09 2005
@@ -41,7 +41,7 @@
 build_triplet = @build@
 host_triplet = @host@
 target_triplet = @target@
-noinst_PROGRAMS = channelmixtest$(EXEEXT)
+noinst_PROGRAMS = 
 subdir = gst/audioconvert
 DIST_COMMON = $(noinst_HEADERS) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in
--- gst-plugins-0.8.10/gst/librfb/gstrfbsrc.c.orig	Thu Jul  7 13:04:55 2005
+++ gst-plugins-0.8.10/gst/librfb/gstrfbsrc.c	Thu Jul  7 13:04:10 2005
@@ -31,6 +31,8 @@
 
 #include <librfb/rfb.h>
 
+#include <sys/time.h>
+
 GST_DEBUG_CATEGORY (gst_debug_rfbsrc);
 #define GST_CAT_DEFAULT gst_debug_rfbsrc
 
