diff -uNr gtk+-2.12.11/Makefile.in gtk+-2.12.11-new/Makefile.in
--- gtk+-2.12.11/Makefile.in	2008-07-01 09:44:00.000000000 -0400
+++ gtk+-2.12.11-new/Makefile.in	2008-08-01 18:20:55.000000000 -0400
@@ -332,7 +332,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules       tests perf contrib
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
 
 # require automake 1.4
diff -uNr gtk+-2.12.0/configure.in gtk+-2.12.0-new/configure.in
--- gtk+-2.12.0/configure.in	2007-09-13 16:39:37.000000000 -0400
+++ gtk+-2.12.0-new/configure.in	2007-09-17 22:24:37.000000000 -0400
@@ -1112,7 +1112,7 @@
 
     GDK_PIXBUF_XLIB_EXTRA_CFLAGS="$x_cflags"
     GDK_PIXBUF_XLIB_EXTRA_LIBS="$X_LIBS -lX11 $X_EXTRA_LIBS"
-    GTK_DEP_LIBS_FOR_X="$X_LIBS -lXrender -lX11 $X_EXTRA_LIBS"
+    GTK_DEP_LIBS_FOR_X="$X_LIBS -lXrender -lX11 -lfontconfig $X_EXTRA_LIBS"
   fi
 
   # Extra libraries found during checks (-lXinerama, etc), not from pkg-config.
diff -uNr gtk+-2.12.0/docs/tools/shooter.c gtk+-2.12.0-new/docs/tools/shooter.c
--- gtk+-2.12.0/docs/tools/shooter.c	2007-09-13 16:35:44.000000000 -0400
+++ gtk+-2.12.0-new/docs/tools/shooter.c	2007-09-17 22:24:37.000000000 -0400
@@ -16,7 +16,11 @@
 #include <stdlib.h>
 #include <fcntl.h>
 #include <errno.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 #include "widgets.h"
 #include "shadow.h"
 
diff -uNr gtk+-2.12.0/gdk/directfb/gdkim-directfb.c gtk+-2.12.0-new/gdk/directfb/gdkim-directfb.c
--- gtk+-2.12.0/gdk/directfb/gdkim-directfb.c	2007-09-13 16:36:06.000000000 -0400
+++ gtk+-2.12.0-new/gdk/directfb/gdkim-directfb.c	2007-09-17 22:24:37.000000000 -0400
@@ -33,7 +33,11 @@
 #include "config.h"
 
 #include <string.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 
 #include "gdkdirectfb.h"
 
diff -uNr gtk+-2.12.0/gdk/x11/gdkdrawable-x11.c gtk+-2.12.0-new/gdk/x11/gdkdrawable-x11.c
--- gtk+-2.12.0/gdk/x11/gdkdrawable-x11.c	2007-09-13 16:36:08.000000000 -0400
+++ gtk+-2.12.0-new/gdk/x11/gdkdrawable-x11.c	2007-09-17 22:24:37.000000000 -0400
@@ -959,11 +959,19 @@
    * RENDER's preferred order
    */
   pf.direct.alphaMask = 0;
-  /* ARGB */
-  pf.direct.red = 16;
-  pf.direct.green = 8;
-  pf.direct.blue = 0;
-  
+  if (ImageByteOrder (xdisplay) == LSBFirst)
+    {
+      /* ARGB */
+      pf.direct.red = 16;
+      pf.direct.green = 8;
+      pf.direct.blue = 0;
+    } else {
+      /* BGRA */
+      pf.direct.red = 8;
+      pf.direct.green = 16;
+      pf.direct.blue = 24;
+    }
+
   *format = XRenderFindFormat (xdisplay,
 			       (PictFormatType | PictFormatDepth |
 				PictFormatRedMask | PictFormatRed |
@@ -974,7 +982,14 @@
 			       0);
 
   pf.direct.alphaMask = 0xff;
-  pf.direct.alpha = 24;
+  if (ImageByteOrder (xdisplay) == LSBFirst)
+    {
+      /* ARGB */
+      pf.direct.alpha = 24;
+    } else {
+      /* BGRA */
+      pf.direct.alpha = 0;
+    }
   
   *mask = XRenderFindFormat (xdisplay,
 			     (PictFormatType | PictFormatDepth |
@@ -991,11 +1006,19 @@
 
   pf.type = PictTypeDirect;
   pf.depth = 32;
-  pf.direct.red = 16;
-  pf.direct.green = 8;
-  pf.direct.blue = 0;
+  if (ImageByteOrder (xdisplay) == LSBFirst)
+    {
+      pf.direct.red = 16;
+      pf.direct.green = 8;
+      pf.direct.blue = 0;
+      pf.direct.alpha = 24;
+    } else {
+      pf.direct.red = 8;
+      pf.direct.green = 16;
+      pf.direct.blue = 24;
+      pf.direct.alpha = 0;
+    }
   pf.direct.alphaMask = 0xff;
-  pf.direct.alpha = 24;
 
   *format = XRenderFindFormat (xdisplay,
 			       (PictFormatType | PictFormatDepth |
diff -uNr gtk+-2.12.0/gdk/x11/gdkfont-x11.c gtk+-2.12.0-new/gdk/x11/gdkfont-x11.c
--- gtk+-2.12.0/gdk/x11/gdkfont-x11.c	2007-09-13 16:36:08.000000000 -0400
+++ gtk+-2.12.0-new/gdk/x11/gdkfont-x11.c	2007-09-17 22:24:37.000000000 -0400
@@ -29,7 +29,11 @@
 #include <config.h>
 #include <X11/Xlib.h>
 #include <X11/Xos.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 
 #include "gdkx.h"
 #include "gdkfont.h"
diff -uNr gtk+-2.12.0/gdk/x11/gdkim-x11.c gtk+-2.12.0-new/gdk/x11/gdkim-x11.c
--- gtk+-2.12.0/gdk/x11/gdkim-x11.c	2007-09-13 16:36:08.000000000 -0400
+++ gtk+-2.12.0-new/gdk/x11/gdkim-x11.c	2007-09-17 22:24:37.000000000 -0400
@@ -26,7 +26,11 @@
 
 #include <config.h>
 
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 
diff -uNr gtk+-2.12.0/gtk/gtkimmulticontext.c gtk+-2.12.0-new/gtk/gtkimmulticontext.c
--- gtk+-2.12.0/gtk/gtkimmulticontext.c	2007-09-13 16:35:41.000000000 -0400
+++ gtk+-2.12.0-new/gtk/gtkimmulticontext.c	2007-09-17 22:24:37.000000000 -0400
@@ -20,7 +20,11 @@
 #include <config.h>
 
 #include <string.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 
 #include "gtkimmulticontext.h"
 #include "gtkimmodule.h"
diff -uNr gtk+-2.12.0/gtk/gtkmain.c gtk+-2.12.0-new/gtk/gtkmain.c
--- gtk+-2.12.0/gtk/gtkmain.c	2007-09-13 16:35:41.000000000 -0400
+++ gtk+-2.12.0-new/gtk/gtkmain.c	2007-09-17 22:24:37.000000000 -0400
@@ -29,7 +29,11 @@
 #include <glib.h>
 #include "gdkconfig.h"
 
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 
 #ifdef HAVE_BIND_TEXTDOMAIN_CODESET
 #include <libintl.h>
diff -uNr gtk+-2.12.0/gtk/gtkpagesetupunixdialog.c gtk+-2.12.0-new/gtk/gtkpagesetupunixdialog.c
--- gtk+-2.12.0/gtk/gtkpagesetupunixdialog.c	2007-09-13 16:35:41.000000000 -0400
+++ gtk+-2.12.0-new/gtk/gtkpagesetupunixdialog.c	2007-09-17 22:24:37.000000000 -0400
@@ -20,7 +20,11 @@
 
 #include "config.h"
 #include <string.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 
 #ifdef HAVE__NL_MEASUREMENT_MEASUREMENT
 #include <langinfo.h>
diff -uNr gtk+-2.12.0/gtk/gtkpapersize.c gtk+-2.12.0-new/gtk/gtkpapersize.c
--- gtk+-2.12.0/gtk/gtkpapersize.c	2007-09-13 16:35:41.000000000 -0400
+++ gtk+-2.12.0-new/gtk/gtkpapersize.c	2007-09-17 22:24:37.000000000 -0400
@@ -22,7 +22,11 @@
 #include "config.h"
 #include <string.h>
 #include <stdlib.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 #if defined(HAVE__NL_PAPER_HEIGHT) && defined(HAVE__NL_PAPER_WIDTH)
 #include <langinfo.h>
 #endif
diff -uNr gtk+-2.12.0/gtk/gtkrc.c gtk+-2.12.0-new/gtk/gtkrc.c
--- gtk+-2.12.0/gtk/gtkrc.c	2007-09-13 16:35:42.000000000 -0400
+++ gtk+-2.12.0-new/gtk/gtkrc.c	2007-09-17 22:24:37.000000000 -0400
@@ -26,7 +26,11 @@
 
 #include <config.h>
 
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
diff -uNr gtk+-2.12.0/gtk/gtkspinbutton.c gtk+-2.12.0-new/gtk/gtkspinbutton.c
--- gtk+-2.12.0/gtk/gtkspinbutton.c	2007-09-13 16:35:40.000000000 -0400
+++ gtk+-2.12.0-new/gtk/gtkspinbutton.c	2007-09-17 22:24:37.000000000 -0400
@@ -32,7 +32,11 @@
 #include <stdlib.h>
 #include <math.h>
 #include <string.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 #include "gdk/gdkkeysyms.h"
 #include "gtkbindings.h"
 #include "gtkspinbutton.h"
diff -uNr gtk+-2.12.0/gtk/gtkwidget.c gtk+-2.12.0-new/gtk/gtkwidget.c
--- gtk+-2.12.0/gtk/gtkwidget.c	2007-09-13 16:35:40.000000000 -0400
+++ gtk+-2.12.0-new/gtk/gtkwidget.c	2007-09-17 22:24:37.000000000 -0400
@@ -27,7 +27,11 @@
 #include <config.h>
 #include <stdarg.h>
 #include <string.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 #include "gtkcontainer.h"
 #include "gtkaccelmap.h"
 #include "gtkclipboard.h"
diff -uNr gtk+-2.12.0/gtk/xdgmime/xdgmimecache.c gtk+-2.12.0-new/gtk/xdgmime/xdgmimecache.c
--- gtk+-2.12.0/gtk/xdgmime/xdgmimecache.c	2007-09-13 16:35:37.000000000 -0400
+++ gtk+-2.12.0-new/gtk/xdgmime/xdgmimecache.c	2007-09-17 22:24:37.000000000 -0400
@@ -46,6 +46,10 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 
+#ifdef HAVE_STDINT_H
+#include <stdint.h>
+#endif
+
 #include "xdgmimecache.h"
 #include "xdgmimeint.h"
 
diff -uNr gtk+-2.12.11/modules/input/Makefile.in gtk+-2.12.11-new/modules/input/Makefile.in
--- gtk+-2.12.11/modules/input/Makefile.in	2008-07-01 09:44:05.000000000 -0400
+++ gtk+-2.12.11-new/modules/input/Makefile.in	2008-08-01 18:20:56.000000000 -0400
@@ -970,25 +970,13 @@
 # install-data-am, and not install-exec-am. We need to ensure this gets run
 # after the libraries are installed in their final locations.
 install-data-hook:
-	@if $(RUN_QUERY_IMMODULES_TEST) ; then \
-	  echo $(mkinstalldirs) $(DESTDIR)$(sysconfdir)/gtk-2.0 ; \
-	  $(mkinstalldirs) $(DESTDIR)$(sysconfdir)/gtk-2.0 ; \
-	  echo "$(top_builddir)/gtk/gtk-query-immodules-2.0 > $(DESTDIR)$(sysconfdir)/gtk-2.0/gtk.immodules" ; \
-	  $(top_builddir)/gtk/gtk-query-immodules-2.0 > $(DESTDIR)$(sysconfdir)/gtk-2.0/gtk.immodules ; \
-	else \
-	  echo "***" ; \
-	  echo "*** Warning: gtk.immodules not built" ; \
-	  echo "***" ; \
-	  echo "*** Generate this file manually on host" ; \
-	  echo "*** system using gtk-query-immodules-2.0" ; \
-	  echo "***" ; \
-        fi
+	$(mkinstalldirs) $(DESTDIR)$(sysconfdir)/gtk-2.0
 
 uninstall-local:
 	rm -f $(DESTDIR)$(sysconfdir)/gtk-2.0/gtk.immodules
 
 gtk.immodules: Makefile.am $(module_LTLIBRARIES)
-	$(top_builddir)/gtk/gtk-query-immodules-2.0 $(module_LTLIBRARIES) > gtk.immodules
+#	$(top_builddir)/gtk/gtk-query-immodules-2.0 $(module_LTLIBRARIES) > gtk.immodules
 
 @CROSS_COMPILING_FALSE@all-local: gtk.immodules
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
diff -uNr gtk+-2.12.0/modules/input/gtkimcontextxim.c gtk+-2.12.0-new/modules/input/gtkimcontextxim.c
--- gtk+-2.12.0/modules/input/gtkimcontextxim.c	2007-09-13 16:35:49.000000000 -0400
+++ gtk+-2.12.0-new/modules/input/gtkimcontextxim.c	2007-09-17 22:24:37.000000000 -0400
@@ -18,7 +18,11 @@
  */
 
 #include <config.h>
-#include "locale.h"
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
+#include <locale.h>
+#endif
 #include <string.h>
 #include <stdlib.h>
 
diff -uNr gtk+-2.12.0/po/Makefile.in.in gtk+-2.12.0-new/po/Makefile.in.in
--- gtk+-2.12.0/po/Makefile.in.in	2007-09-13 16:36:05.000000000 -0400
+++ gtk+-2.12.0-new/po/Makefile.in.in	2007-09-17 22:24:38.000000000 -0400
@@ -27,7 +27,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 libdir = @libdir@
-localedir = $(libdir)/locale
+localedir = $(datadir)/locale
 gnulocaledir = $(datadir)/locale
 gettextsrcdir = $(datadir)/glib-2.0/gettext/po
 subdir = po
diff -uNr gtk+-2.12.0/tests/testicontheme.c gtk+-2.12.0-new/tests/testicontheme.c
--- gtk+-2.12.0/tests/testicontheme.c	2007-09-13 16:35:48.000000000 -0400
+++ gtk+-2.12.0-new/tests/testicontheme.c	2007-09-17 22:24:38.000000000 -0400
@@ -21,7 +21,11 @@
 #include <gtk/gtk.h>
 #include <stdlib.h>
 #include <string.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 
 static void
 usage (void)
diff -uNr gtk+-2.12.0/update-gdk-pixbuf-loaders gtk+-2.12.0-new/update-gdk-pixbuf-loaders
--- gtk+-2.12.0/update-gdk-pixbuf-loaders	1969-12-31 19:00:00.000000000 -0500
+++ gtk+-2.12.0-new/update-gdk-pixbuf-loaders	2007-09-17 22:24:38.000000000 -0400
@@ -0,0 +1,21 @@
+#!/bin/sh -e
+#
+# update-gdk-pixbuf-loaders
+#
+#   Packages which install gdk-pixbuf loader(s) should call update-gdk-
+#   pixbuf-loaders in their PostInstScript and PostRmScript to update the
+#   gdk-pixbuf.loaders file.
+#
+
+CONFDIR=@PREFIX@/etc/gtk-2.0
+CONFFILE=$CONFDIR/gdk-pixbuf.loaders
+
+mkdir -p $CONFDIR
+
+TMPFILE=`tempfile`
+gdk-pixbuf-query-loaders >$TMPFILE
+
+mv -f $TMPFILE $CONFFILE
+chmod 0644 $CONFFILE
+
+exit 0
diff -uNr gtk+-2.12.0/update-gtk-immodules gtk+-2.12.0-new/update-gtk-immodules
--- gtk+-2.12.0/update-gtk-immodules	1969-12-31 19:00:00.000000000 -0500
+++ gtk+-2.12.0-new/update-gtk-immodules	2007-09-17 22:24:38.000000000 -0400
@@ -0,0 +1,20 @@
+#!/bin/sh -e
+#
+# update-gtk-immodules
+#
+#   Packages which install gtk immodule(s) should call update-gtk-immodules
+#   in their PostInstScript and PostRmScript to update the gtk.immodules file.
+#
+
+CONFDIR=@PREFIX@/etc/gtk-2.0
+CONFFILE=$CONFDIR/gtk.immodules
+
+mkdir -p $CONFDIR
+
+TMPFILE=`tempfile`
+gtk-query-immodules-2.0 >$TMPFILE
+
+mv -f $TMPFILE $CONFFILE
+chmod 0644 $CONFFILE
+
+exit 0
