diff -Nurd -x'*~' gtk+-2.14.7.orig/Makefile.in gtk+-2.14.7/Makefile.in
--- gtk+-2.14.7.orig/Makefile.in	2009-01-07 16:52:00.000000000 -0500
+++ gtk+-2.14.7/Makefile.in	2009-01-09 01:45:25.000000000 -0500
@@ -516,7 +516,7 @@
 	&& DISPLAY=:$$XID && export DISPLAY
 
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules             perf contrib
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
 
 # require automake 1.4
diff -Nurd -x'*~' gtk+-2.14.7.orig/configure gtk+-2.14.7/configure
--- gtk+-2.14.7.orig/configure	2009-01-07 16:52:17.000000000 -0500
+++ gtk+-2.14.7/configure	2009-01-09 01:45:25.000000000 -0500
@@ -33739,16 +33739,6 @@
 fi
 
 
-  # Check for the RANDR extension
-  if $PKG_CONFIG --exists "xrandr >= 1.2" ; then
-
-cat >>confdefs.h <<\_ACEOF
-#define HAVE_RANDR 1
-_ACEOF
-
-
-     X_PACKAGES="$X_PACKAGES xrandr"
-  fi
 
   # Checks for Xcursor library
 
diff -Nurd -x'*~' gtk+-2.14.7.orig/docs/tools/shooter.c gtk+-2.14.7/docs/tools/shooter.c
--- gtk+-2.14.7.orig/docs/tools/shooter.c	2009-01-07 11:33:00.000000000 -0500
+++ gtk+-2.14.7/docs/tools/shooter.c	2009-01-09 01:45:25.000000000 -0500
@@ -16,7 +16,11 @@
 #include <stdlib.h>
 #include <fcntl.h>
 #include <errno.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 #include "widgets.h"
 #include "shadow.h"
 
diff -Nurd -x'*~' gtk+-2.14.7.orig/gdk/Makefile.in gtk+-2.14.7/gdk/Makefile.in
--- gtk+-2.14.7.orig/gdk/Makefile.in	2009-01-07 16:52:06.000000000 -0500
+++ gtk+-2.14.7/gdk/Makefile.in	2009-01-09 01:45:25.000000000 -0500
@@ -577,7 +577,7 @@
 libgdk_x11_2_0_la_LIBADD = x11/libgdk-x11.la $(GDK_DEP_LIBS) \
   $(top_builddir)/gdk-pixbuf/libgdk_pixbuf-$(GTK_API_VERSION).la
 
-libgdk_x11_2_0_la_LDFLAGS = $(LDADD)
+libgdk_x11_2_0_la_LDFLAGS = $(LDADD) -lXrender
 
 libgdk_quartz_2_0_la_SOURCES = $(common_sources) gdkkeynames.c
 libgdk_quartz_2_0_la_LIBADD = quartz/libgdk-quartz.la $(GDK_DEP_LIBS) \
diff -Nurd -x'*~' gtk+-2.14.7.orig/gdk/directfb/gdkim-directfb.c gtk+-2.14.7/gdk/directfb/gdkim-directfb.c
--- gtk+-2.14.7.orig/gdk/directfb/gdkim-directfb.c	2009-01-07 11:33:47.000000000 -0500
+++ gtk+-2.14.7/gdk/directfb/gdkim-directfb.c	2009-01-09 01:45:25.000000000 -0500
@@ -33,7 +33,11 @@
 #include "config.h"
 
 #include <string.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 
 #include "gdkdirectfb.h"
 
diff -Nurd -x'*~' gtk+-2.14.7.orig/gdk/x11/gdkdrawable-x11.c gtk+-2.14.7/gdk/x11/gdkdrawable-x11.c
--- gtk+-2.14.7.orig/gdk/x11/gdkdrawable-x11.c	2009-01-07 11:33:56.000000000 -0500
+++ gtk+-2.14.7/gdk/x11/gdkdrawable-x11.c	2009-01-09 01:45:25.000000000 -0500
@@ -959,11 +959,19 @@
    * RENDER's preferred order
    */
   pf.direct.alphaMask = 0;
-  /* ARGB */
-  pf.direct.red = 16;
-  pf.direct.green = 8;
-  pf.direct.blue = 0;
-  
+  if (ImageByteOrder (xdisplay) == LSBFirst)
+    {
+      /* ARGB */
+      pf.direct.red = 16;
+      pf.direct.green = 8;
+      pf.direct.blue = 0;
+    } else {
+      /* BGRA */
+      pf.direct.red = 8;
+      pf.direct.green = 16;
+      pf.direct.blue = 24;
+    }
+
   *format = XRenderFindFormat (xdisplay,
 			       (PictFormatType | PictFormatDepth |
 				PictFormatRedMask | PictFormatRed |
@@ -974,7 +982,14 @@
 			       0);
 
   pf.direct.alphaMask = 0xff;
-  pf.direct.alpha = 24;
+  if (ImageByteOrder (xdisplay) == LSBFirst)
+    {
+      /* ARGB */
+      pf.direct.alpha = 24;
+    } else {
+      /* BGRA */
+      pf.direct.alpha = 0;
+    }
   
   *mask = XRenderFindFormat (xdisplay,
 			     (PictFormatType | PictFormatDepth |
@@ -991,11 +1006,19 @@
 
   pf.type = PictTypeDirect;
   pf.depth = 32;
-  pf.direct.red = 16;
-  pf.direct.green = 8;
-  pf.direct.blue = 0;
+  if (ImageByteOrder (xdisplay) == LSBFirst)
+    {
+      pf.direct.red = 16;
+      pf.direct.green = 8;
+      pf.direct.blue = 0;
+      pf.direct.alpha = 24;
+    } else {
+      pf.direct.red = 8;
+      pf.direct.green = 16;
+      pf.direct.blue = 24;
+      pf.direct.alpha = 0;
+    }
   pf.direct.alphaMask = 0xff;
-  pf.direct.alpha = 24;
 
   *format = XRenderFindFormat (xdisplay,
 			       (PictFormatType | PictFormatDepth |
diff -Nurd -x'*~' gtk+-2.14.7.orig/gdk/x11/gdkfont-x11.c gtk+-2.14.7/gdk/x11/gdkfont-x11.c
--- gtk+-2.14.7.orig/gdk/x11/gdkfont-x11.c	2009-01-07 11:33:56.000000000 -0500
+++ gtk+-2.14.7/gdk/x11/gdkfont-x11.c	2009-01-09 01:45:25.000000000 -0500
@@ -29,7 +29,11 @@
 #include "config.h"
 #include <X11/Xlib.h>
 #include <X11/Xos.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 
 #include "gdkx.h"
 #include "gdkfont.h"
diff -Nurd -x'*~' gtk+-2.14.7.orig/gdk/x11/gdkim-x11.c gtk+-2.14.7/gdk/x11/gdkim-x11.c
--- gtk+-2.14.7.orig/gdk/x11/gdkim-x11.c	2009-01-07 11:33:56.000000000 -0500
+++ gtk+-2.14.7/gdk/x11/gdkim-x11.c	2009-01-09 01:45:25.000000000 -0500
@@ -26,7 +26,11 @@
 
 #include "config.h"
 
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 
diff -Nurd -x'*~' gtk+-2.14.7.orig/gtk/gtk-builder-convert gtk+-2.14.7/gtk/gtk-builder-convert
--- gtk+-2.14.7.orig/gtk/gtk-builder-convert	2009-01-07 11:32:57.000000000 -0500
+++ gtk+-2.14.7/gtk/gtk-builder-convert	2009-01-13 01:09:10.000000000 -0500
@@ -269,9 +269,14 @@
 
         # Output the newly created root objects and sort them
         # by attribute id
-        for obj in sorted(self.root_objects,
-                          key=lambda n: n.getAttribute('id'),
-                          reverse=True):
+        # FIXME: Use sorted(self.root_objects,
+        #                   key=lambda n: n.getAttribute('id'),
+        #                   reverse=True):
+        # when we can depend on python 2.4 or higher
+        root_objects = self.root_objects[:]
+        root_objects.sort(lambda a, b: cmp(b.getAttribute('id'),
+                                           a.getAttribute('id')))
+        for obj in root_objects:
             self._interface.childNodes.insert(0, obj)
 
     def _convert(self, klass, node):
@@ -520,12 +525,12 @@
         if prop.childNodes:
             data = prop.childNodes[0].data
             value, lower, upper, step, page, page_size = data.split(' ')
-            properties.update(value=value,
-                              lower=lower,
-                              upper=upper,
-                              step_increment=step,
-                              page_increment=page,
-                              page_size=page_size)
+            properties.update({"value":value,
+                               "lower":lower,
+                               "upper":upper,
+                               "step_increment":step,
+                               "page_increment":page,
+                               "page_size":page_size})
         else:
             prop.appendChild(self._dom.createTextNode(""))
 
diff -Nurd -x'*~' gtk+-2.14.7.orig/gtk/gtkimmulticontext.c gtk+-2.14.7/gtk/gtkimmulticontext.c
--- gtk+-2.14.7.orig/gtk/gtkimmulticontext.c	2009-01-07 11:32:57.000000000 -0500
+++ gtk+-2.14.7/gtk/gtkimmulticontext.c	2009-01-09 01:45:25.000000000 -0500
@@ -20,7 +20,11 @@
 #include "config.h"
 
 #include <string.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 
 #include "gtkimmulticontext.h"
 #include "gtkimmodule.h"
diff -Nurd -x'*~' gtk+-2.14.7.orig/gtk/gtkmain.c gtk+-2.14.7/gtk/gtkmain.c
--- gtk+-2.14.7.orig/gtk/gtkmain.c	2009-01-07 11:32:57.000000000 -0500
+++ gtk+-2.14.7/gtk/gtkmain.c	2009-01-09 01:45:25.000000000 -0500
@@ -29,7 +29,11 @@
 #include <glib.h>
 #include "gdkconfig.h"
 
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 
 #include <stdio.h>
 #include <stdlib.h>
diff -Nurd -x'*~' gtk+-2.14.7.orig/gtk/gtkpagesetupunixdialog.c gtk+-2.14.7/gtk/gtkpagesetupunixdialog.c
--- gtk+-2.14.7.orig/gtk/gtkpagesetupunixdialog.c	2009-01-07 11:32:57.000000000 -0500
+++ gtk+-2.14.7/gtk/gtkpagesetupunixdialog.c	2009-01-09 01:45:25.000000000 -0500
@@ -21,7 +21,11 @@
 
 #include "config.h"
 #include <string.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 
 #ifdef HAVE__NL_MEASUREMENT_MEASUREMENT
 #include <langinfo.h>
diff -Nurd -x'*~' gtk+-2.14.7.orig/gtk/gtkpapersize.c gtk+-2.14.7/gtk/gtkpapersize.c
--- gtk+-2.14.7.orig/gtk/gtkpapersize.c	2009-01-07 11:32:57.000000000 -0500
+++ gtk+-2.14.7/gtk/gtkpapersize.c	2009-01-09 01:45:25.000000000 -0500
@@ -22,7 +22,11 @@
 #include "config.h"
 #include <string.h>
 #include <stdlib.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 #if defined(HAVE__NL_PAPER_HEIGHT) && defined(HAVE__NL_PAPER_WIDTH)
 #include <langinfo.h>
 #endif
diff -Nurd -x'*~' gtk+-2.14.7.orig/gtk/gtkrc.c gtk+-2.14.7/gtk/gtkrc.c
--- gtk+-2.14.7.orig/gtk/gtkrc.c	2009-01-07 11:32:57.000000000 -0500
+++ gtk+-2.14.7/gtk/gtkrc.c	2009-01-09 01:45:25.000000000 -0500
@@ -26,7 +26,11 @@
 
 #include "config.h"
 
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
diff -Nurd -x'*~' gtk+-2.14.7.orig/gtk/gtkspinbutton.c gtk+-2.14.7/gtk/gtkspinbutton.c
--- gtk+-2.14.7.orig/gtk/gtkspinbutton.c	2009-01-07 11:32:57.000000000 -0500
+++ gtk+-2.14.7/gtk/gtkspinbutton.c	2009-01-09 01:45:25.000000000 -0500
@@ -32,7 +32,11 @@
 #include <stdlib.h>
 #include <math.h>
 #include <string.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 #include "gdk/gdkkeysyms.h"
 #include "gtkbindings.h"
 #include "gtkspinbutton.h"
diff -Nurd -x'*~' gtk+-2.14.7.orig/gtk/gtkwidget.c gtk+-2.14.7/gtk/gtkwidget.c
--- gtk+-2.14.7.orig/gtk/gtkwidget.c	2009-01-07 11:32:57.000000000 -0500
+++ gtk+-2.14.7/gtk/gtkwidget.c	2009-01-09 01:45:25.000000000 -0500
@@ -27,7 +27,11 @@
 #include "config.h"
 #include <stdarg.h>
 #include <string.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 #include "gtkcontainer.h"
 #include "gtkaccelmap.h"
 #include "gtkclipboard.h"
diff -Nurd -x'*~' gtk+-2.14.7.orig/ltmain.sh gtk+-2.14.7/ltmain.sh
--- gtk+-2.14.7.orig/ltmain.sh	2008-08-29 18:27:25.000000000 -0400
+++ gtk+-2.14.7/ltmain.sh	2009-02-17 02:09:19.000000000 -0500
@@ -2924,100 +2924,72 @@
 	    fi
 	    tmp_libs="$tmp_libs $deplib"
 	  done
-
-	  if test "$link_all_deplibs" != no; then
-	    # Add the search paths of all dependency libraries
-	    for deplib in $dependency_libs; do
-	      case $deplib in
-	      -L*) path="$deplib" ;;
-	      *.la)
-		dir=`$echo "X$deplib" | $Xsed -e 's%/[^/]*$%%'`
-		test "X$dir" = "X$deplib" && dir="."
-		# We need an absolute path.
-		case $dir in
-		[\\/]* | [A-Za-z]:[\\/]*) absdir="$dir" ;;
-		*)
-		  absdir=`cd "$dir" && pwd`
-		  if test -z "$absdir"; then
-		    $echo "$modename: warning: cannot determine absolute directory name of \`$dir'" 1>&2
-		    absdir="$dir"
-		  fi
-		  ;;
-		esac
-		if grep "^installed=no" $deplib > /dev/null; then
-		  path="$absdir/$objdir"
-		else
-		  eval libdir=`${SED} -n -e 's/^libdir=\(.*\)$/\1/p' $deplib`
-		  if test -z "$libdir"; then
-		    $echo "$modename: \`$deplib' is not a valid libtool archive" 1>&2
-		    exit $EXIT_FAILURE
-		  fi
-		  if test "$absdir" != "$libdir"; then
-		    $echo "$modename: warning: \`$deplib' seems to be moved" 1>&2
-		  fi
-		  path="$absdir"
-		fi
-		depdepl=
-		case $host in
-		*-*-darwin*)
-		  # we do not want to link against static libs,
-		  # but need to link against shared
-		  eval deplibrary_names=`${SED} -n -e 's/^library_names=\(.*\)$/\1/p' $deplib`
-		  eval deplibdir=`${SED} -n -e 's/^libdir=\(.*\)$/\1/p' $deplib`
-		  if test -n "$deplibrary_names" ; then
-		    for tmp in $deplibrary_names ; do
-		      depdepl=$tmp
-		    done
-		    if test -f "$deplibdir/$depdepl" ; then
-		      depdepl="$deplibdir/$depdepl"
-	      	    elif test -f "$path/$depdepl" ; then
-		      depdepl="$path/$depdepl"
-		    else
-		      # Can't find it, oh well...
-		      depdepl=
-		    fi
-		    # do not add paths which are already there
-		    case " $newlib_search_path " in
-		    *" $path "*) ;;
-		    *) newlib_search_path="$newlib_search_path $path";;
-		    esac
-		  fi
-		  path=""
-		  ;;
-		*)
-		  path="-L$path"
-		  ;;
-		esac
-		;;
-	      -l*)
-		case $host in
-		*-*-darwin*)
-		  # Again, we only want to link against shared libraries
-		  eval tmp_libs=`$echo "X$deplib" | $Xsed -e "s,^\-l,,"`
-		  for tmp in $newlib_search_path ; do
-		    if test -f "$tmp/lib$tmp_libs.dylib" ; then
-		      eval depdepl="$tmp/lib$tmp_libs.dylib"
-		      break
-		    fi
-		  done
-		  path=""
-		  ;;
-		*) continue ;;
-		esac
-		;;
-	      *) continue ;;
-	      esac
-	      case " $deplibs " in
-	      *" $path "*) ;;
-	      *) deplibs="$path $deplibs" ;;
-	      esac
-	      case " $deplibs " in
-	      *" $depdepl "*) ;;
-	      *) deplibs="$depdepl $deplibs" ;;
-	      esac
-	    done
-	  fi # link_all_deplibs != no
-	fi # linkmode = lib
+          if test "$link_all_deplibs" != no; then
+            # Add the search paths of all dependency libraries
+            for deplib in $dependency_libs; do
+              case $deplib in
+              -L*) path="$deplib" ;;
+              *.la)
+#               func_dirname "$deplib" "" "."
+#               dir="$func_dirname_result"
+                case "$deplib" in
+                  */*) dir="${deplib%/*}" ;;
+                  *  ) dir="." ;;
+                esac
+                # We need an absolute path.
+                case $dir in
+                [\\/]* | [A-Za-z]:[\\/]*) absdir="$dir" ;;
+                *)
+                  absdir=`cd "$dir" && pwd`
+                  if test -z "$absdir"; then
+                    $echo "$modename: warning: cannot determine absolute directory name of \`$dir'" 1>&2
+                    absdir="$dir"
+                  fi
+                  ;;
+                esac
+                if grep "^installed=no" $deplib > /dev/null; then
+                case $host in
+                *-*-darwin*)
+                  depdepl=
+                  eval deplibrary_names=`${SED} -n -e 's/^library_names=\(.*\)$/\1/p' $deplib`
+                  if test -n "$deplibrary_names" ; then
+                    for tmp in $deplibrary_names ; do
+                      depdepl=$tmp
+                    done
+                    if test -f "$absdir/$objdir/$depdepl" ; then
+                      depdepl="$absdir/$objdir/$depdepl"
+                      darwin_install_name=`otool -L $depdepl | awk '{if (NR == 2) {print $1;exit}}'`
+                      if test -z "$darwin_install_name"; then
+                          darwin_install_name=`otool64 -L $depdepl  | awk '{if (NR == 2) {print $1;exit}}'`
+                      fi
+                      compiler_flags="$compiler_flags ${wl}-dylib_file ${wl}${darwin_install_name}:${depdepl}"
+                      linker_flags="$linker_flags -dylib_file ${darwin_install_name}:${depdepl}"
+                      path=
+                    fi
+                  fi
+                  ;;
+                *)
+                  path="-L$absdir/$objdir"
+                  ;;
+                esac
+                else
+                  eval libdir=`${SED} -n -e 's/^libdir=\(.*\)$/\1/p' $deplib`
+                  test -z "$libdir" && \
+                    $echo "\`$deplib' is not a valid libtool archive" 1>&2
+                  test -z "$libdir" && exit 1
+                  test "$absdir" != "$libdir" && \
+                    $echo "\`$deplib' seems to be moved" 1>&2
+                  path="-L$absdir"
+                fi
+                ;;
+              esac
+              case " $deplibs " in
+              *" $path "*) ;;
+              *) deplibs="$path $deplibs" ;;
+              esac
+            done
+          fi # link_all_deplibs != no
+        fi # linkmode = lib
       done # for deplib in $libs
       dependency_libs="$newdependency_libs"
       if test "$pass" = dlpreopen; then
diff -Nurd -x'*~' gtk+-2.14.7.orig/modules/input/gtkimcontextxim.c gtk+-2.14.7/modules/input/gtkimcontextxim.c
--- gtk+-2.14.7.orig/modules/input/gtkimcontextxim.c	2009-01-07 11:33:31.000000000 -0500
+++ gtk+-2.14.7/modules/input/gtkimcontextxim.c	2009-01-09 01:45:25.000000000 -0500
@@ -18,7 +18,11 @@
  */
 
 #include "config.h"
-#include "locale.h"
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
+#include <locale.h>
+#endif
 #include <string.h>
 #include <stdlib.h>
 
diff -Nurd -x'*~' gtk+-2.14.7.orig/tests/testicontheme.c gtk+-2.14.7/tests/testicontheme.c
--- gtk+-2.14.7.orig/tests/testicontheme.c	2009-01-07 11:33:12.000000000 -0500
+++ gtk+-2.14.7/tests/testicontheme.c	2009-01-09 01:45:25.000000000 -0500
@@ -21,7 +21,11 @@
 #include <gtk/gtk.h>
 #include <stdlib.h>
 #include <string.h>
+#if defined(X_LOCALE)
+#include <X11/Xlocale.h>
+#else
 #include <locale.h>
+#endif
 
 static void
 usage (void)
diff -Nurd -x'*~' gtk+-2.14.7.orig/update-gdk-pixbuf-loaders gtk+-2.14.7/update-gdk-pixbuf-loaders
--- gtk+-2.14.7.orig/update-gdk-pixbuf-loaders	1969-12-31 19:00:00.000000000 -0500
+++ gtk+-2.14.7/update-gdk-pixbuf-loaders	2009-01-09 01:45:25.000000000 -0500
@@ -0,0 +1,21 @@
+#!/bin/sh -e
+#
+# update-gdk-pixbuf-loaders
+#
+#   Packages which install gdk-pixbuf loader(s) should call update-gdk-
+#   pixbuf-loaders in their PostInstScript and PostRmScript to update the
+#   gdk-pixbuf.loaders file.
+#
+
+CONFDIR=@PREFIX@/etc/gtk-2.0
+CONFFILE=$CONFDIR/gdk-pixbuf.loaders
+
+mkdir -p $CONFDIR
+
+TMPFILE=`tempfile`
+gdk-pixbuf-query-loaders >$TMPFILE
+
+mv -f $TMPFILE $CONFFILE
+chmod 0644 $CONFFILE
+
+exit 0
diff -Nurd -x'*~' gtk+-2.14.7.orig/update-gtk-immodules gtk+-2.14.7/update-gtk-immodules
--- gtk+-2.14.7.orig/update-gtk-immodules	1969-12-31 19:00:00.000000000 -0500
+++ gtk+-2.14.7/update-gtk-immodules	2009-01-09 01:45:25.000000000 -0500
@@ -0,0 +1,20 @@
+#!/bin/sh -e
+#
+# update-gtk-immodules
+#
+#   Packages which install gtk immodule(s) should call update-gtk-immodules
+#   in their PostInstScript and PostRmScript to update the gtk.immodules file.
+#
+
+CONFDIR=@PREFIX@/etc/gtk-2.0
+CONFFILE=$CONFDIR/gtk.immodules
+
+mkdir -p $CONFDIR
+
+TMPFILE=`tempfile`
+gtk-query-immodules-2.0 >$TMPFILE
+
+mv -f $TMPFILE $CONFFILE
+chmod 0644 $CONFFILE
+
+exit 0
