diff -Nurd -x'*~' gucharmap-2.28.2.orig/gucharmap/Makefile.in gucharmap-2.28.2/gucharmap/Makefile.in
--- gucharmap-2.28.2.orig/gucharmap/Makefile.in	2009-12-14 15:23:15.000000000 -0500
+++ gucharmap-2.28.2/gucharmap/Makefile.in	2010-02-21 00:10:17.000000000 -0500
@@ -423,7 +423,7 @@
 
 libgucharmap_la_LIBADD = \
 	$(GTK_LIBS)	\
-	$(INTL_LIBS)
+	$(INTLLIBS)
 
 libgucharmapincludedir = $(includedir)/gucharmap-2/gucharmap
 libgucharmapinclude_HEADERS = \
