diff -Nurd -x'*~' libbonobo-2.20.3.orig/configure libbonobo-2.20.3/configure
--- libbonobo-2.20.3.orig/configure	2007-12-22 15:35:08.000000000 -0500
+++ libbonobo-2.20.3/configure	2008-01-21 03:10:57.000000000 -0500
@@ -24062,6 +24062,8 @@
 
 
 # popt
+# See: http://bugzilla.gnome.org/show_bug.cgi?id=357145
+if false; then
 { echo "$as_me:$LINENO: checking for poptStrippedArgv in -lpopt" >&5
 echo $ECHO_N "checking for poptStrippedArgv in -lpopt... $ECHO_C" >&6; }
 if test "${ac_cv_lib_popt_poptStrippedArgv+set}" = set; then
@@ -24130,6 +24132,7 @@
 echo "$as_me: error: You must have popt version 1.5 or greater installed." >&2;}
    { (exit 1); exit 1; }; }
 fi
+fi
 
 
 if test "${ac_cv_header_popt_h+set}" = set; then
diff -Nurd -x'*~' libbonobo-2.20.3.orig/doc/api/Makefile.in libbonobo-2.20.3/doc/api/Makefile.in
--- libbonobo-2.20.3.orig/doc/api/Makefile.in	2007-12-22 15:45:16.000000000 -0500
+++ libbonobo-2.20.3/doc/api/Makefile.in	2008-01-21 03:05:35.000000000 -0500
@@ -301,6 +301,7 @@
 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"	\
 	$(LIBBONOBO_CFLAGS)				\
 	-I$(top_srcdir)					\
+	-I$(top_srcdir)/bonobo				\
 	-I$(top_builddir)
 
 GTKDOC_LIBS = $(top_builddir)/bonobo/libbonobo-2.la $(LIBBONOBO_LIBS)
diff -Nurd -x'*~' libbonobo-2.20.3.orig/doc/api/libbonobo.types libbonobo-2.20.3/doc/api/libbonobo.types
--- libbonobo-2.20.3.orig/doc/api/libbonobo.types	2007-12-22 15:46:37.000000000 -0500
+++ libbonobo-2.20.3/doc/api/libbonobo.types	2008-01-21 03:05:35.000000000 -0500
@@ -1,4 +1,8 @@
 #include <libbonobo.h>
+#include "bonobo-running-context.h"
+#include "bonobo-moniker-extender.h"
+#include "bonobo-item-handler.h"
+#include "bonobo-shlib-factory.h"
 
 bonobo_generic_factory_get_type
 bonobo_storage_mem_get_type
