diff -Naur libbonobo-2.5.4-old/bonobo-activation/bonobo-activation.h libbonobo-2.5.4/bonobo-activation/bonobo-activation.h
--- libbonobo-2.5.4-old/bonobo-activation/bonobo-activation.h	Sat Mar  6 19:38:09 2004
+++ libbonobo-2.5.4/bonobo-activation/bonobo-activation.h	Sat Mar  6 19:49:59 2004
@@ -24,6 +24,9 @@
 #ifndef BONOBO_ACTIVATION_H
 #define BONOBO_ACTIVATION_H
 
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+
 #include <bonobo-activation/Bonobo_Unknown.h>
 #include <bonobo-activation/Bonobo_GenericFactory.h>
 
--- libbonobo-2.8.1/doc/api/libbonobo.types	2003-01-21 11:54:30.000000000 +0100
+++ libbonobo-2.8.1/doc/api/libbonobo-patched.types	2005-07-06 07:48:10.000000000 +0200
@@ -1,4 +1,8 @@
 #include <libbonobo.h>
+#include "bonobo-running-context.h"
+#include "bonobo-moniker-extender.h"
+#include "bonobo-item-handler.h"
+#include "bonobo-shlib-factory.h"
 
 bonobo_generic_factory_get_type
 bonobo_storage_mem_get_type
--- libbonobo-2.8.1/doc/api/Makefile.am	2003-06-02 16:25:33.000000000 +0200
+++ libbonobo-2.8.1/doc/api/Makefile-patched.am	2005-07-06 07:50:16.000000000 +0200
@@ -52,6 +52,7 @@
 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"	\
 	$(LIBBONOBO_CFLAGS)				\
 	-I$(top_srcdir)					\
+	-I$(top_srcdir)/bonobo				\
 	-I$(top_builddir)
 
 GTKDOC_LIBS = $(top_builddir)/bonobo/libbonobo-2.la $(LIBBONOBO_LIBS)
--- libbonobo-2.8.1/doc/api/Makefile.in	2005-02-05 15:39:55.000000000 +0100
+++ libbonobo-2.8.1/doc/api/Makefile-patched.in	2005-07-06 07:51:38.000000000 +0200
@@ -272,6 +272,7 @@
 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"	\
 	$(LIBBONOBO_CFLAGS)				\
 	-I$(top_srcdir)					\
+	-I$(top_srcdir)/bonobo				\
 	-I$(top_builddir)
 
 
