diff -Nurd -x'*~' libbonobo-2.20.3.orig/doc/api/Makefile.in libbonobo-2.20.3/doc/api/Makefile.in
--- libbonobo-2.20.3.orig/doc/api/Makefile.in	2007-12-22 15:45:16.000000000 -0500
+++ libbonobo-2.20.3/doc/api/Makefile.in	2008-01-21 03:05:35.000000000 -0500
@@ -301,6 +301,7 @@
 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"	\
 	$(LIBBONOBO_CFLAGS)				\
 	-I$(top_srcdir)					\
+	-I$(top_srcdir)/bonobo				\
 	-I$(top_builddir)
 
 GTKDOC_LIBS = $(top_builddir)/bonobo/libbonobo-2.la $(LIBBONOBO_LIBS)
diff -Nurd -x'*~' libbonobo-2.20.3.orig/doc/api/libbonobo.types libbonobo-2.20.3/doc/api/libbonobo.types
--- libbonobo-2.20.3.orig/doc/api/libbonobo.types	2007-12-22 15:46:37.000000000 -0500
+++ libbonobo-2.20.3/doc/api/libbonobo.types	2008-01-21 03:05:35.000000000 -0500
@@ -1,4 +1,8 @@
 #include <libbonobo.h>
+#include "bonobo-running-context.h"
+#include "bonobo-moniker-extender.h"
+#include "bonobo-item-handler.h"
+#include "bonobo-shlib-factory.h"
 
 bonobo_generic_factory_get_type
 bonobo_storage_mem_get_type
