diff -Nurd -x'*~' libbonobo-2.14.0.orig/bonobo-activation/bonobo-activation-fork-server.c libbonobo-2.14.0/bonobo-activation/bonobo-activation-fork-server.c
--- libbonobo-2.14.0.orig/bonobo-activation/bonobo-activation-fork-server.c	2006-03-14 09:28:07.000000000 -0500
+++ libbonobo-2.14.0/bonobo-activation/bonobo-activation-fork-server.c	2006-05-17 02:24:04.000000000 -0400
@@ -364,11 +364,19 @@
                 g_source_destroy (source);
                 g_source_unref (source);
                 
+                g_io_channel_shutdown (ai.ioc, FALSE, NULL);
                 g_io_channel_unref (ai.ioc);
                 
                 if (use_new_loop)
                         g_main_context_unref (context);
 
+                close (iopipes[1]);
+
+                g_strfreev(newenv);
+
+                if (fd_arg != 0)
+                        g_free ((char *) cmd[fd_arg]);
+
                 g_strfreev (cmd);
 
 		return CORBA_OBJECT_NIL;
diff -Nurd -x'*~' libbonobo-2.14.0.orig/doc/api/Makefile.in libbonobo-2.14.0/doc/api/Makefile.in
--- libbonobo-2.14.0.orig/doc/api/Makefile.in	2006-03-14 09:25:43.000000000 -0500
+++ libbonobo-2.14.0/doc/api/Makefile.in	2006-05-15 21:54:30.000000000 -0400
@@ -297,6 +297,7 @@
 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"	\
 	$(LIBBONOBO_CFLAGS)				\
 	-I$(top_srcdir)					\
+	-I$(top_srcdir)/bonobo				\
 	-I$(top_builddir)
 
 GTKDOC_LIBS = $(top_builddir)/bonobo/libbonobo-2.la $(LIBBONOBO_LIBS)
diff -Nurd -x'*~' libbonobo-2.14.0.orig/doc/api/libbonobo.types libbonobo-2.14.0/doc/api/libbonobo.types
--- libbonobo-2.14.0.orig/doc/api/libbonobo.types	2006-03-14 09:29:59.000000000 -0500
+++ libbonobo-2.14.0/doc/api/libbonobo.types	2006-05-15 21:54:30.000000000 -0400
@@ -1,4 +1,8 @@
 #include <libbonobo.h>
+#include "bonobo-running-context.h"
+#include "bonobo-moniker-extender.h"
+#include "bonobo-item-handler.h"
+#include "bonobo-shlib-factory.h"
 
 bonobo_generic_factory_get_type
 bonobo_storage_mem_get_type
