diff -Nurd -x'*~' libbonobo-2.16.0.orig/doc/api/Makefile.in libbonobo-2.16.0/doc/api/Makefile.in
--- libbonobo-2.16.0.orig/doc/api/Makefile.in	2006-09-04 11:09:02.000000000 -0400
+++ libbonobo-2.16.0/doc/api/Makefile.in	2006-09-19 23:06:29.000000000 -0400
@@ -279,6 +279,7 @@
 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"	\
 	$(LIBBONOBO_CFLAGS)				\
 	-I$(top_srcdir)					\
+	-I$(top_srcdir)/bonobo				\
 	-I$(top_builddir)
 
 
diff -Nurd -x'*~' libbonobo-2.16.0.orig/doc/api/libbonobo.types libbonobo-2.16.0/doc/api/libbonobo.types
--- libbonobo-2.16.0.orig/doc/api/libbonobo.types	2006-09-04 11:11:51.000000000 -0400
+++ libbonobo-2.16.0/doc/api/libbonobo.types	2006-09-19 23:06:29.000000000 -0400
@@ -1,4 +1,8 @@
 #include <libbonobo.h>
+#include "bonobo-running-context.h"
+#include "bonobo-moniker-extender.h"
+#include "bonobo-item-handler.h"
+#include "bonobo-shlib-factory.h"
 
 bonobo_generic_factory_get_type
 bonobo_storage_mem_get_type
