diff -Nurd -x'*~' libgnomeprint-2.18.2.orig/configure libgnomeprint-2.18.2/configure
--- libgnomeprint-2.18.2.orig/configure	2007-09-17 09:13:10.000000000 -0400
+++ libgnomeprint-2.18.2/configure	2007-11-03 13:36:39.000000000 -0400
@@ -24779,8 +24779,8 @@
    { (exit 1); exit 1; }; }
 fi
 
-LIBGNOMEPRINT_CFLAGS="$GP_CFLAGS $FREETYPE_CFLAGS"
-LIBGNOMEPRINT_LIBS="$GP_LIBS $FREETYPE_LIBS $Z_LIBS"
+LIBGNOMEPRINT_CFLAGS="$FREETYPE_CFLAGS $GP_CFLAGS"
+LIBGNOMEPRINT_LIBS="$FREETYPE_LIBS $GP_LIBS $Z_LIBS"
 
 
 
