diff -uNr libgksu-2.0.3/libgksu/Makefile.in libgksu-2.0.3-new/libgksu/Makefile.in
--- libgksu-2.0.3/libgksu/Makefile.in	2006-10-23 22:27:25.000000000 -0400
+++ libgksu-2.0.3-new/libgksu/Makefile.in	2007-09-26 16:41:04.000000000 -0400
@@ -273,7 +273,7 @@
 # minor -> keeps compatibility (additions to the API)
 # micro -> no change to the API/ABI
 libgksu2_la_LIBADD = ../libgksuui/libgksuui1.0.la
-libgksu2_la_LDFLAGS = -version-info 0:1:0 -Wl,-O1 -lutil \
+libgksu2_la_LDFLAGS = -version-info 0:1:0 \
 	${LIBGKSU_LIBS} $(am__append_1)
 noinst_HEADERS = defines.h
 include_HEADERS = gksu.h libgksu.h
diff -uNr libgksu-2.0.3/libgksu/libgksu.c libgksu-2.0.3-new/libgksu/libgksu.c
--- libgksu-2.0.3/libgksu/libgksu.c	2006-10-23 22:16:19.000000000 -0400
+++ libgksu-2.0.3-new/libgksu/libgksu.c	2007-09-26 16:36:07.000000000 -0400
@@ -23,7 +23,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <fcntl.h>
-#include <pty.h>
+#include <util.h>
 #include <pwd.h>
 #include <sys/types.h>
 #include <sys/wait.h>
@@ -1806,7 +1806,7 @@
       gchar **cmd = g_malloc (sizeof(gchar*)*7);
 
       setsid();   // make us session leader
-      cmd[i] = g_strdup ("/bin/su"); i++;
+      cmd[i] = g_strdup ("/usr/bin/su"); i++;
       if (context->login_shell)
 	{
 	  cmd[i] = g_strdup ("-"); i++;
@@ -1828,7 +1828,7 @@
       if (execv (cmd[0], cmd) == -1)
 	{
 	  fprintf (stderr,
-		   "Unable to run /bin/su: %s",
+		   "Unable to run /usr/bin/su: %s",
 		   strerror(errno));
 	}
 
diff -uNr libgksu-2.0.3/libgksuui/Makefile.in libgksu-2.0.3-new/libgksuui/Makefile.in
--- libgksu-2.0.3/libgksuui/Makefile.in	2006-10-23 22:27:25.000000000 -0400
+++ libgksu-2.0.3-new/libgksuui/Makefile.in	2007-09-26 16:38:28.000000000 -0400
@@ -246,7 +246,7 @@
 AM_CPPFLAGS = -DLOCALEDIR=\"$(datadir)/locale\" -DDATA_DIR=\"$(datadir)\" -DPREFIX=\"$(prefix)\"
 noinst_LTLIBRARIES = libgksuui1.0.la
 libgksuui1_0_la_SOURCES = gksuui-dialog.c
-libgksuui1_0_la_LDFLAGS = -Wl,-O1 `pkg-config --libs gtk+-2.0 gconf-2.0`
+libgksuui1_0_la_LDFLAGS = `pkg-config --libs gtk+-2.0 gconf-2.0`
 noinst_HEADERS = defines.h gksuui.h gksuui-dialog.h
 test_gksuui_SOURCES = test-gksuui.c
 test_gksuui_LDADD = libgksuui1.0.la
