diff -uNr libgail-gnome-1.18.0/Makefile.in libgail-gnome-1.18.0-new/Makefile.in
--- libgail-gnome-1.18.0/Makefile.in	2007-03-12 07:37:04.000000000 -0400
+++ libgail-gnome-1.18.0-new/Makefile.in	2007-03-13 09:12:50.000000000 -0400
@@ -214,7 +214,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-SUBDIRS = gail-gnome test
+SUBDIRS = gail-gnome
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libgail-gnome.pc
 EXTRA_DIST = libgail-gnome.pc.in
