diff -Nurd -x'*~' libgnomeprint-2.10.3.orig/configure libgnomeprint-2.10.3/configure
--- libgnomeprint-2.10.3.orig/configure	2005-03-30 17:28:44.000000000 -0500
+++ libgnomeprint-2.10.3/configure	2005-07-16 03:49:07.000000000 -0400
@@ -23958,7 +23958,7 @@
 GNOME_PRINT_DATA_DIR=$GNOME_PRINT_DATA_DIR
 
 
-LIBGNOMEPRINT_CFLAGS="$GP_CFLAGS $FREETYPE_CFLAGS"
+LIBGNOMEPRINT_CFLAGS="$FREETYPE_CFLAGS $GP_CFLAGS"
 LIBGNOMEPRINT_LIBS="$GP_LIBS $FREETYPE_LIBS $Z_LIBS"
 
 
diff -Nurd -x'*~' libgnomeprint-2.10.3.orig/libgnomeprint/gnome-fontmap.c libgnomeprint-2.10.3/libgnomeprint/gnome-fontmap.c
--- libgnomeprint-2.10.3.orig/libgnomeprint/gnome-fontmap.c	2005-03-30 17:05:33.000000000 -0500
+++ libgnomeprint-2.10.3/libgnomeprint/gnome-fontmap.c	2005-07-16 03:49:07.000000000 -0400
@@ -223,6 +223,8 @@
 		font_type = GP_FONT_ENTRY_TRUETYPE;
 	else if (g_ascii_strcasecmp (c, ".ttc") == 0)
 		font_type = GP_FONT_ENTRY_TRUETYPE;
+        else if (g_ascii_strcasecmp (c, ".otf") == 0)
+                font_type = GP_FONT_ENTRY_TRUETYPE;
 	else if (g_ascii_strcasecmp (c, ".font") == 0) /* See #102400 */
 		font_type = GP_FONT_ENTRY_TRUETYPE;
 	else {
diff -Nurd -x'*~' libgnomeprint-2.10.3.orig/libgnomeprint/gnome-print.c libgnomeprint-2.10.3/libgnomeprint/gnome-print.c
--- libgnomeprint-2.10.3.orig/libgnomeprint/gnome-print.c	2004-12-31 12:25:25.000000000 -0500
+++ libgnomeprint-2.10.3/libgnomeprint/gnome-print.c	2005-07-16 03:49:07.000000000 -0400
@@ -28,6 +28,7 @@
 #include <config.h>
 #include <string.h>
 #include <gmodule.h>
+#include <sys/types.h>
 #include <errno.h>
 #include <fcntl.h>
 
