diff -uNr libgtop-2.20.0/lib/Makefile.in libgtop-2.20.0-patched/lib/Makefile.in
--- libgtop-2.20.0/lib/Makefile.in	2007-09-16 09:05:59.000000000 -0400
+++ libgtop-2.20.0-patched/lib/Makefile.in	2007-09-26 17:18:07.000000000 -0400
@@ -286,7 +286,7 @@
 				  read_data.c write.c lib.c parameter.c \
 				  sysdeps.c
 
-libgtop_2_0_la_LDFLAGS = $(LT_VERSION_INFO) -export-symbols $(srcdir)/libgtop.sym
+libgtop_2_0_la_LDFLAGS = $(LT_VERSION_INFO)
 libgtop_2_0_la_LIBADD = $(GLIB_LIBS) $(top_builddir)/sysdeps/common/libgtop_common-2.0.la $(top_builddir)/sysdeps/@sysdeps_dir@/libgtop_sysdeps-2.0.la $(sysdeps_suid_lib)
 BUILT_SOURCES = lib.c
 EXTRA_DIST = lib.pl libgtop.sym
diff -uNr libgtop-2.20.0/sysdeps/stub/glibtop_server.h libgtop-2.20.0-patched/sysdeps/stub/glibtop_server.h
--- libgtop-2.20.0/sysdeps/stub/glibtop_server.h	2007-04-26 18:27:38.000000000 -0400
+++ libgtop-2.20.0-patched/sysdeps/stub/glibtop_server.h	2007-09-26 17:15:55.000000000 -0400
@@ -45,6 +45,7 @@
 #define GLIBTOP_SUID_NETLOAD		0
 #define GLIBTOP_SUID_NETLIST		0
 #define GLIBTOP_SUID_PPP		0
+#define GLIBTOP_SUID_PROC_WD		0
 
 G_END_DECLS
 
diff -uNr libgtop-2.20.0/sysdeps/stub/proclist.c libgtop-2.20.0-patched/sysdeps/stub/proclist.c
--- libgtop-2.20.0/sysdeps/stub/proclist.c	2007-04-26 18:27:38.000000000 -0400
+++ libgtop-2.20.0-patched/sysdeps/stub/proclist.c	2007-09-26 17:11:51.000000000 -0400
@@ -42,7 +42,7 @@
  *   On success, it returnes a pointer to a list of buf->number elements
  *   each buf->size big. The total size is stored in buf->total. */
 
-unsigned *
+pid_t *
 glibtop_get_proclist_s (glibtop *server, glibtop_proclist *buf,
 			gint64 which, gint64 arg)
 {
