diff -Nurd -x'*~' libgnomeui-2.20.1.1.orig/doc/reference/Makefile.in libgnomeui-2.20.1.1/doc/reference/Makefile.in
--- libgnomeui-2.20.1.1.orig/doc/reference/Makefile.in	2007-10-17 17:50:54.000000000 -0400
+++ libgnomeui-2.20.1.1/doc/reference/Makefile.in	2008-01-23 08:09:03.000000000 -0500
@@ -277,8 +277,8 @@
 
 # CFLAGS and LDFLAGS for compiling scan program. Only needed if your app/lib
 # contains GtkObjects/GObjects and you want to document signals and properties.
-INCLUDES = -I$(top_builddir) -I$(top_srcdir) $(LIBGNOMEUI_CFLAGS)
-GTKDOC_LIBS = $(top_builddir)/libgnomeui/libgnomeui-2.la
+INCLUDES = -I$(top_builddir) -I$(top_srcdir) $(LIBGNOMEUI_CFLAGS) $(CFLAGS) $(CPPFLAGS)
+GTKDOC_LIBS = $(top_builddir)/libgnomeui/libgnomeui-2.la $(LIBGNOMEUI_LIBS) $(SM_LIBS) $(LDFLAGS)
 
 # Extra options to supply to gtkdoc-mkdb.
 MKDB_OPTIONS = --output-format=xml
