diff -Nurd -x'*~' libgda-4.0.4.orig/configure libgda-4.0.4/configure
--- libgda-4.0.4.orig/configure	2009-08-30 06:45:08.000000000 -0400
+++ libgda-4.0.4/configure	2009-09-11 02:24:25.000000000 -0400
@@ -20671,7 +20671,7 @@
 #define HAVE_CARBON 1
 _ACEOF
 
-    LIBGDA_LIBS="$LIBGDA_LIBS -framework Carbon"
+    LIBGDA_LIBS="$LIBGDA_LIBS -Wl,-framework,Carbon"
       ;;
 *)
     { $as_echo "$as_me:$LINENO: result: Unix" >&5
diff -Nurd -x'*~' libgda-4.0.4.orig/libgda-report/RML/gda-report-rml-document.c libgda-4.0.4/libgda-report/RML/gda-report-rml-document.c
--- libgda-4.0.4.orig/libgda-report/RML/gda-report-rml-document.c	2009-07-05 15:27:21.000000000 -0400
+++ libgda-4.0.4/libgda-report/RML/gda-report-rml-document.c	2009-09-11 02:17:58.000000000 -0400
@@ -198,7 +198,7 @@
 	if (!converter) {
 		converter = g_find_program_in_path ("trml2html.py");
 		if (!converter) {
-			converter = gda_gbr_get_file_path (GDA_BIN_DIR, "gda_trml2html", "trml2html.py", NULL);
+			converter = gda_gbr_get_file_path (GDA_LIBEXEC_DIR, "gda_trml2html", "trml2html.py", NULL);
 			if (!g_file_test (converter, G_FILE_TEST_IS_EXECUTABLE)) {
 				g_free (converter);
 				converter = NULL;
@@ -229,7 +229,7 @@
 	if (!converter) {
 		converter = g_find_program_in_path ("trml2pdf.py");
 		if (!converter) {
-			converter = gda_gbr_get_file_path (GDA_BIN_DIR, "gda_trml2pdf", "trml2pdf.py", NULL);
+			converter = gda_gbr_get_file_path (GDA_LIBEXEC_DIR, "gda_trml2pdf", "trml2pdf.py", NULL);
 			if (!g_file_test (converter, G_FILE_TEST_IS_EXECUTABLE)) {
 				g_free (converter);
 				converter = NULL;
diff -Nurd -x'*~' libgda-4.0.4.orig/libgda-report/RML/trml2html/Makefile.in libgda-4.0.4/libgda-report/RML/trml2html/Makefile.in
--- libgda-4.0.4.orig/libgda-report/RML/trml2html/Makefile.in	2009-08-30 14:59:25.000000000 -0400
+++ libgda-4.0.4/libgda-report/RML/trml2html/Makefile.in	2009-09-11 02:25:31.000000000 -0400
@@ -365,7 +365,7 @@
 target_alias = @target_alias@
 trml2html_SCRIPTS = trml2html.py
 trml2html_DATA = __init__.py utils.py
-trml2htmldir = $(bindir)/gda_trml2html
+trml2htmldir = $(libexecdir)/gda_trml2html
 EXTRA_DIST = \
 	$(trml2html_SCRIPTS) \
 	$(trml2html_DATA)
diff -Nurd -x'*~' libgda-4.0.4.orig/libgda-report/RML/trml2pdf/Makefile.in libgda-4.0.4/libgda-report/RML/trml2pdf/Makefile.in
--- libgda-4.0.4.orig/libgda-report/RML/trml2pdf/Makefile.in	2009-08-30 14:59:25.000000000 -0400
+++ libgda-4.0.4/libgda-report/RML/trml2pdf/Makefile.in	2009-09-11 02:25:57.000000000 -0400
@@ -365,7 +365,7 @@
 target_alias = @target_alias@
 trml2pdf_SCRIPTS = trml2pdf.py
 trml2pdf_DATA = __init__.py color.py utils.py
-trml2pdfdir = $(bindir)/gda_trml2pdf
+trml2pdfdir = $(libexecdir)/gda_trml2pdf
 EXTRA_DIST = \
 	$(trml2pdf_SCRIPTS) \
 	$(trml2pdf_DATA)
diff -Nurd -x'*~' libgda-4.0.4.orig/libgda-report/engine/Makefile.in libgda-4.0.4/libgda-report/engine/Makefile.in
--- libgda-4.0.4.orig/libgda-report/engine/Makefile.in	2009-08-30 14:59:25.000000000 -0400
+++ libgda-4.0.4/libgda-report/engine/Makefile.in	2009-09-11 02:27:20.000000000 -0400
@@ -400,8 +400,8 @@
 	$(gda_report_headers)	\
 	gda-report-engine.c	
 
-libgda_report_engine_4_0_la_LIBADD = $(LIBGDA_LIBS) \
-	$(top_builddir)/libgda/libgda-4.0.la 
+libgda_report_engine_4_0_la_LIBADD = \
+	$(top_builddir)/libgda/libgda-4.0.la $(LIBGDA_LIBS)
 
 gdareportincludedir = $(includedir)/libgda-$(GDA_ABI_MAJOR_VERSION).$(GDA_ABI_MINOR_VERSION)/libgda-report
 gdareportinclude_HEADERS = $(gda_report_headers)
diff -Nurd -x'*~' libgda-4.0.4.orig/libgda-xslt/Makefile.in libgda-4.0.4/libgda-xslt/Makefile.in
--- libgda-4.0.4.orig/libgda-xslt/Makefile.in	2009-08-30 14:59:26.000000000 -0400
+++ libgda-4.0.4/libgda-xslt/Makefile.in	2009-09-11 02:28:37.000000000 -0400
@@ -405,8 +405,8 @@
 	sql_backend.c
 
 libgda_xslt_4_0_la_LDFLAGS = -version-info $(GDA_CURRENT):$(GDA_REVISION):$(GDA_AGE) $(NO_UNDEFINED)
-libgda_xslt_4_0_la_LIBADD = $(LIBGDA_LIBS) $(LIBXSLT_LIBS)\
-        $(top_builddir)/libgda/libgda-4.0.la
+libgda_xslt_4_0_la_LIBADD = \
+        $(top_builddir)/libgda/libgda-4.0.la $(LIBGDA_LIBS) $(LIBXSLT_LIBS)
 
 gdaxsltincludedir = $(includedir)/libgda-$(GDA_ABI_MAJOR_VERSION).$(GDA_ABI_MINOR_VERSION)/libgda-xslt
 gdaxsltinclude_HEADERS = $(gda_xslt_headers)
diff -Nurd -x'*~' libgda-4.0.4.orig/providers/bdb/Makefile.in libgda-4.0.4/providers/bdb/Makefile.in
--- libgda-4.0.4.orig/providers/bdb/Makefile.in	2009-08-30 14:59:28.000000000 -0400
+++ libgda-4.0.4/providers/bdb/Makefile.in	2009-09-11 02:29:42.000000000 -0400
@@ -413,9 +413,9 @@
 
 gda_bdb_test_SOURCES = gda-bdb-test.c
 gda_bdb_test_LDADD = \
+	$(top_builddir)/libgda/libgda-4.0.la \
 	$(LIBGDA_LIBS) \
-	$(BDB_LIBS) \
-	$(top_builddir)/libgda/libgda-4.0.la 
+	$(BDB_LIBS)
 
 xmldir = $(datadir)/libgda-4.0
 xml_in_files = bdb_specs_dsn.xml.in
