diff -Nurd -x'*~' libgnomecups-0.2.3.orig/configure libgnomecups-0.2.3/configure
--- libgnomecups-0.2.3.orig/configure	2008-01-27 07:07:47.000000000 -0500
+++ libgnomecups-0.2.3/configure	2008-08-14 13:32:55.000000000 -0400
@@ -20310,7 +20310,7 @@
 fi
 
 CUPS_CFLAGS=`cups-config --cflags | sed 's/-O0-9*//' | sed 's/-m^\t*//g'`
-CUPS_LIBS=`cups-config --libs`
+CUPS_LIBS=`cups-config --libs |  perl -p -e 's/(-\S*framework)\s+(\S+)/-Wl,\1,\2/g'`
 
 
 
diff -Nurd -x'*~' libgnomecups-0.2.3.orig/libgnomecups/gnome-cups-request.c libgnomecups-0.2.3/libgnomecups/gnome-cups-request.c
--- libgnomecups-0.2.3.orig/libgnomecups/gnome-cups-request.c	2007-01-31 13:49:17.000000000 -0500
+++ libgnomecups-0.2.3/libgnomecups/gnome-cups-request.c	2008-01-28 23:04:53.000000000 -0500
@@ -393,15 +393,22 @@
 	return ret;
 }
 
+// ippNewRequest not available in cups on 10.4.11 (but maybe in 10.4.0uSDK?)
+// revive libgnomecups-0.2.2 incantation for some of gnome_cups_request_new()
 ipp_t *
 gnome_cups_request_new (int operation_id)
 {
 	cups_lang_t *language;
 	ipp_t *request;
 	
-	request = ippNewRequest (operation_id);
+	request = ippNew ();
+	request->request.op.operation_id = operation_id;
+	request->request.op.request_id = 1;
 	
 	language = cupsLangDefault ();
+	ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_CHARSET,
+		     "attributes-charset", 
+		     NULL, "utf-8");
 	ippAddString (request, IPP_TAG_OPERATION, IPP_TAG_LANGUAGE,
 		      "attributes-natural-language", 
 		      NULL, language->language);
