diff -Nurd -x'*~' libglade-2.6.4.orig/glade/Makefile.in libglade-2.6.4/glade/Makefile.in
--- libglade-2.6.4.orig/glade/Makefile.in	2009-03-17 10:25:37.000000000 -0400
+++ libglade-2.6.4/glade/Makefile.in	2010-02-16 01:36:44.000000000 -0500
@@ -260,7 +260,7 @@
   glade-gtk.c \
   glade-private.h
 
-libglade_2_0_la_LIBADD = $(LIBGLADE_LIBS)
+libglade_2_0_la_LIBADD = $(LIBGLADE_LIBS) $(INTLLIBS)
 libglade_2_0_la_LDFLAGS = -version-info $(LTVERSION) $(export_symbols) -no-undefined
 EXTRA_DIST = $(gladeinclude_HEADERS) \
 	glade.def
diff -ruN libglade-2.3.6/glade/glade-gtk.c libglade-2.3.6-patched/glade/glade-gtk.c
--- libglade-2.3.6/glade/glade-gtk.c	2004-03-10 00:13:29.000000000 -0800
+++ libglade-2.3.6-patched/glade/glade-gtk.c	2007-07-08 13:43:59.000000000 -0700
@@ -26,6 +26,7 @@
 #include <string.h>
 #include <stdlib.h>
 
+#define G_IMPLEMENT_INLINES
 /* for GtkText et all */
 #ifndef GTK_ENABLE_BROKEN
 #define GTK_ENABLE_BROKEN
diff -ruN libglade-2.3.6/glade/glade-parser.c libglade-2.3.6-patched/glade/glade-parser.c
--- libglade-2.3.6/glade/glade-parser.c	2004-02-03 01:40:37.000000000 -0800
+++ libglade-2.3.6-patched/glade/glade-parser.c	2007-07-08 13:43:32.000000000 -0700
@@ -36,6 +36,7 @@
 
 #include <libxml/parser.h>
 
+#define G_IMPLEMENT_INLINES
 #include "glade-parser.h"
 #include "glade-private.h"
 
diff -ruN libglade-2.3.6/glade/glade-xml.c libglade-2.3.6-patched/glade/glade-xml.c
--- libglade-2.3.6/glade/glade-xml.c	2004-02-03 01:40:38.000000000 -0800
+++ libglade-2.3.6-patched/glade/glade-xml.c	2007-07-08 13:43:32.000000000 -0700
@@ -25,6 +25,7 @@
 
 #include <stdlib.h>
 #include <string.h>
+#define G_IMPLEMENT_INLINES
 #include <glade/glade-xml.h>
 #include <glade/glade-init.h>
 #include <glade/glade-build.h>
