diff -Nurd -x'*~' librsvg-2.20.0.orig/Makefile.in librsvg-2.20.0/Makefile.in
--- librsvg-2.20.0.orig/Makefile.in	2008-01-19 11:48:27.000000000 -0500
+++ librsvg-2.20.0/Makefile.in	2008-01-26 00:21:34.000000000 -0500
@@ -371,7 +371,7 @@
 	rsvg-file-util.c
 
 librsvg_2_la_LDFLAGS = -version-info @VERSION_INFO@ -export-dynamic -no-undefined -export-symbols $(srcdir)/librsvg.def
-librsvg_2_la_LIBADD = $(LIBGNOME_VFS_LIBS) $(LIBGSF_LIBS) $(LIBCROCO_LIBS) $(LIBRSVG_LIBS) $(FREETYPE_LIBS)
+librsvg_2_la_LIBADD = $(FREETYPE_LIBS) $(LIBGNOME_VFS_LIBS) $(LIBGSF_LIBS) $(LIBCROCO_LIBS) $(LIBRSVG_LIBS)
 librsvgincdir = $(includedir)/librsvg-2/librsvg
 librsvginc_HEADERS = \
 	$(headers)	\
@@ -379,6 +379,7 @@
 	librsvg-enum-types.h
 
 INCLUDES = \
+	$(FREETYPE_INCLUDES)		\
 	$(LIBGNOME_VFS_CFLAGS)		\
 	$(LIBGSF_CFLAGS)		\
 	$(LIBCROCO_CFLAGS)		\
