diff -uNr librsvg-2.16.1/Makefile.in librsvg-2.16.1-new/Makefile.in
--- librsvg-2.16.1/Makefile.in	2006-11-02 20:57:13.000000000 -0500
+++ librsvg-2.16.1-new/Makefile.in	2006-12-20 10:04:30.000000000 -0500
@@ -368,8 +368,8 @@
 	rsvg-gobject.c		\
 	rsvg-file-util.c
 
-librsvg_2_la_LDFLAGS = -version-info @VERSION_INFO@ -export-dynamic -no-undefined -export-symbols $(srcdir)/librsvg.def
-librsvg_2_la_LIBADD = $(LIBGNOME_VFS_LIBS) $(LIBGSF_LIBS) $(LIBCROCO_LIBS) $(LIBRSVG_LIBS) $(FREETYPE_LIBS)
+librsvg_2_la_LDFLAGS = $(FREETYPE_LDFLAGS) -version-info @VERSION_INFO@ -export-dynamic -no-undefined -export-symbols $(srcdir)/librsvg.def
+librsvg_2_la_LIBADD = $(FREETYPE_LIBS) $(LIBGNOME_VFS_LIBS) $(LIBGSF_LIBS) $(LIBCROCO_LIBS) $(LIBRSVG_LIBS)
 librsvgincdir = $(includedir)/librsvg-2/librsvg
 librsvginc_HEADERS = \
 	$(headers)	\
@@ -377,6 +377,7 @@
 	librsvg-enum-types.h
 
 INCLUDES = \
+	$(FREETYPE_INCLUDES)		\
 	$(LIBGNOME_VFS_CFLAGS)		\
 	$(LIBGSF_CFLAGS)		\
 	$(LIBCROCO_CFLAGS)		\
