diff -Nurd -x'*~' rarian-0.8.1.orig/util/rarian-sk-update.in rarian-0.8.1/util/rarian-sk-update.in
--- rarian-0.8.1.orig/util/rarian-sk-update.in	2008-09-01 14:40:21.000000000 -0400
+++ rarian-0.8.1/util/rarian-sk-update.in	2008-09-27 14:49:44.000000000 -0400
@@ -57,7 +57,7 @@
 	echo -e "\t\t\t\tscrolls.  If the directory doesn't exist, it"
 	echo -e "\t\t\t\twill be created at run time."
 	echo -e "\t\t\t\t Note: Multiple output paths are not supported"
-	echo -e "--c\t\t\t\tRebuild the index entirely.  This will rebuild"
+	echo -e "-c\t\t\t\tRebuild the index entirely.  This will rebuild"
 	echo -e "\t\t\t\tall scroll files and may take a long time."
 	echo -e "-v\t\t\t\tTurn Verbosity on."
 	echo -e "-p\t\t\t\tSpecify a different place to store the mtimes file."
@@ -264,7 +264,7 @@
 # They are inherited from scrollkeeper :(
 
 # We use TEMP as set -- seems to nuke the return value of getopt
-TEMP=`getopt -u -n$(basename $0) -o "o:r:p:vqnhV" \
+TEMP=`getopt "o:r:p:vqnhV" \
     	     -- "$@"` \
     || print_usage
 
