diff -Nurd -x'*~' libxklavier-3.8.orig/configure libxklavier-3.8/configure
--- libxklavier-3.8.orig/configure	2008-11-16 19:47:46.000000000 -0500
+++ libxklavier-3.8/configure	2009-02-21 14:09:19.000000000 -0500
@@ -2316,7 +2316,7 @@
 
 
 # Define the identity of the package.
- PACKAGE='libxklavier'
+ PACKAGE='libxklavier12'
  VERSION='3.8'
 
 
diff -Nurd -x'*~' libxklavier-3.8.orig/doc/reference/Makefile.in libxklavier-3.8/doc/reference/Makefile.in
--- libxklavier-3.8.orig/doc/reference/Makefile.in	2008-11-16 19:47:45.000000000 -0500
+++ libxklavier-3.8/doc/reference/Makefile.in	2009-02-21 14:09:19.000000000 -0500
@@ -247,7 +247,7 @@
 # e.g. INCLUDES=-I$(top_srcdir) -I$(top_builddir) $(GTK_DEBUG_FLAGS)
 # e.g. GTKDOC_LIBS=$(top_builddir)/gtk/$(gtktargetlib)
 INCLUDES = -I$(top_srcdir) -I$(top_builddir) $(GLIB_CFLAGS)
-GTKDOC_LIBS = $(GLIB_LIBS) -L$(top_builddir)/libxklavier -lxklavier
+GTKDOC_LIBS = $(GLIB_LIBS) $(top_builddir)/libxklavier/libxklavier.la
 @GTK_DOC_USE_LIBTOOL_FALSE@GTKDOC_CC = $(CC) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 @GTK_DOC_USE_LIBTOOL_TRUE@GTKDOC_CC = $(LIBTOOL) --mode=compile $(CC) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 @GTK_DOC_USE_LIBTOOL_FALSE@GTKDOC_LD = $(CC) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS)
diff -Nurd -x'*~' libxklavier-3.8.orig/libxklavier/Makefile.in libxklavier-3.8/libxklavier/Makefile.in
--- libxklavier-3.8.orig/libxklavier/Makefile.in	2008-11-16 19:47:46.000000000 -0500
+++ libxklavier-3.8/libxklavier/Makefile.in	2009-02-21 14:13:59.000000000 -0500
@@ -222,8 +222,8 @@
 
 # GLIB_GENMARSHAL = $(shell $PKG_CONFIG --variable=glib_genmarshal glib-2.0)
 GLIB_GENMARSHAL = `$PKG_CONFIG --variable=glib_genmarshal glib-2.0`
-AM_CFLAGS = -Wall -DDATA_DIR=\"$(datadir)/$(PACKAGE)\" \
-  -I. -I$(includedir) $(X_CFLAGS) -I$(top_srcdir) \
+AM_CPPFLAGS = -Wall -DDATA_DIR=\"$(datadir)/$(PACKAGE)\" \
+  -I. -I$(top_srcdir) $(X_CFLAGS) \
   $(XML_CFLAGS) $(GLIB_CFLAGS) $(XINPUT_CFLAGS) \
   $(LIBXKBFILE_PRESENT_CFLAG) \
   $(ENABLE_XKB_SUPPORT_CFLAG) \
diff -Nurd -x'*~' libxklavier-3.8.orig/tests/Makefile.in libxklavier-3.8/tests/Makefile.in
--- libxklavier-3.8.orig/tests/Makefile.in	2008-11-16 19:47:46.000000000 -0500
+++ libxklavier-3.8/tests/Makefile.in	2009-02-21 14:09:19.000000000 -0500
@@ -196,7 +196,7 @@
 top_srcdir = @top_srcdir@
 test_config_SOURCES = test_config.c
 test_monitor_SOURCES = test_monitor.c
-AM_CFLAGS = -Wall -Werror -I$(includedir) $(X_CFLAGS) -I$(top_srcdir) $(GLIB_CFLAGS)
+AM_CPPFLAGS = -Wall -Werror -I$(top_srcdir) $(X_CFLAGS) $(GLIB_CFLAGS)
 AM_LDFLAGS = $(top_builddir)/libxklavier/libxklavier.la $(X_LIBS) -lX11 $(GLIB_LIBS)
 all: all-am
 
