diff -Nurd -x'*~' libxklavier-3.7.orig/configure libxklavier-3.7/configure
--- libxklavier-3.7.orig/configure	2008-09-05 15:50:11.000000000 -0400
+++ libxklavier-3.7/configure	2008-11-10 03:05:53.000000000 -0500
@@ -1964,7 +1964,7 @@
 
 
 
-PACKAGE=libxklavier
+PACKAGE=libxklavier12
 MAJOR_VERSION=3
 MINOR_VERSION=7
 VERSION=$MAJOR_VERSION.$MINOR_VERSION
diff -Nurd -x'*~' libxklavier-3.7.orig/doc/reference/Makefile.in libxklavier-3.7/doc/reference/Makefile.in
--- libxklavier-3.7.orig/doc/reference/Makefile.in	2008-09-05 15:50:10.000000000 -0400
+++ libxklavier-3.7/doc/reference/Makefile.in	2008-11-10 03:05:40.000000000 -0500
@@ -257,7 +257,7 @@
 # e.g. INCLUDES=-I$(top_srcdir) -I$(top_builddir) $(GTK_DEBUG_FLAGS)
 # e.g. GTKDOC_LIBS=$(top_builddir)/gtk/$(gtktargetlib)
 INCLUDES = -I$(top_srcdir) -I$(top_builddir) $(GLIB_CFLAGS)
-GTKDOC_LIBS = $(GLIB_LIBS) -L$(top_builddir)/libxklavier -lxklavier
+GTKDOC_LIBS = $(GLIB_LIBS) $(top_builddir)/libxklavier/libxklavier.la
 @GTK_DOC_USE_LIBTOOL_FALSE@GTKDOC_CC = $(CC) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 @GTK_DOC_USE_LIBTOOL_TRUE@GTKDOC_CC = $(LIBTOOL) --mode=compile $(CC) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 @GTK_DOC_USE_LIBTOOL_FALSE@GTKDOC_LD = $(CC) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS)
diff -Nurd -x'*~' libxklavier-3.7.orig/libxklavier/Makefile.in libxklavier-3.7/libxklavier/Makefile.in
--- libxklavier-3.7.orig/libxklavier/Makefile.in	2008-09-05 15:50:10.000000000 -0400
+++ libxklavier-3.7/libxklavier/Makefile.in	2008-11-10 03:05:54.000000000 -0500
@@ -226,8 +226,8 @@
 @ENABLE_XMODMAP_SUPPORT_TRUE@ENABLE_XMODMAP_SUPPORT_CFLAG = -DENABLE_XMODMAP_SUPPORT=1
 EXTRA_DIST = marshal.list
 GLIB_GENMARSHAL = $(shell pkg-config --variable=glib_genmarshal glib-2.0)
-AM_CFLAGS = -Wall -DDATA_DIR=\"$(datadir)/$(PACKAGE)\" \
-  -I. -I$(includedir) $(X_CFLAGS) -I$(top_srcdir) \
+AM_CPPFLAGS = -Wall -DDATA_DIR=\"$(datadir)/$(PACKAGE)\" \
+  -I. -I$(top_srcdir) -I$(includedir) $(X_CFLAGS) \
   $(XML_CFLAGS) $(GLIB_CFLAGS) \
   $(LIBXKBFILE_PRESENT_CFLAG) \
   $(ENABLE_XKB_SUPPORT_CFLAG) \
diff -Nurd -x'*~' libxklavier-3.7.orig/tests/Makefile.in libxklavier-3.7/tests/Makefile.in
--- libxklavier-3.7.orig/tests/Makefile.in	2008-09-05 15:50:10.000000000 -0400
+++ libxklavier-3.7/tests/Makefile.in	2008-11-10 03:05:54.000000000 -0500
@@ -205,7 +205,7 @@
 target_alias = @target_alias@
 test_config_SOURCES = test_config.c
 test_monitor_SOURCES = test_monitor.c
-AM_CFLAGS = -Wall -Werror -I$(includedir) $(X_CFLAGS) -I$(top_srcdir) $(GLIB_CFLAGS)
+AM_CPPFLAGS = -Wall -Werror -I$(top_srcdir) -I$(includedir) $(X_CFLAGS) $(GLIB_CFLAGS)
 AM_LDFLAGS = $(top_builddir)/libxklavier/libxklavier.la $(X_LIBS) -lX11 $(GLIB_LIBS)
 all: all-am
 
