diff -Nurd -x'*~' meld-1.1.4.orig/GNUmakefile meld-1.1.4/GNUmakefile
--- meld-1.1.4.orig/GNUmakefile	2006-06-10 15:52:38.000000000 -0400
+++ meld-1.1.4/GNUmakefile	2006-08-10 14:00:35.000000000 -0400
@@ -57,8 +57,8 @@
 		$(DESTDIR)$(sharedir)/application-registry/meld.applications
 	install -m 644 meld.desktop \
 		$(DESTDIR)$(sharedir)/applications
-	$(PYTHON) -c 'import compileall; compileall.compile_dir("$(DESTDIR)$(libdir_)")'
-	$(PYTHON) -O -c 'import compileall; compileall.compile_dir("$(DESTDIR)$(libdir_)")'
+	$(PYTHON)    -c 'import compileall; compileall.compile_dir("$(DESTDIR)$(libdir_)",10,"$(libdir_)")'
+	$(PYTHON) -O -c 'import compileall; compileall.compile_dir("$(DESTDIR)$(libdir_)",10,"$(libdir_)")'
 	install -m 644 \
 		glade2/*.glade \
 		$(DESTDIR)$(sharedir_)/glade2
diff -Nurd -x'*~' meld-1.2.orig/help/fr/GNUmakefile meld-1.2/help/fr/GNUmakefile
--- meld-1.2.orig/help/fr/GNUmakefile	2008-03-31 16:44:53.000000000 -0400
+++ meld-1.2/help/fr/GNUmakefile	2008-06-29 23:16:01.000000000 -0400
@@ -17,7 +17,6 @@
 install : $(OMF_NAME).install
 	-mkdir -m 755 -p $(OMF_DIR_) $(XML_DIR_) $(XML_DIR_)/figures
 	install -m 644 meld.$(LANG).xml $(XML_DIR_)/meld.xml
-	install -m 644 figures/*.png $(XML_DIR_)/figures
 	-install -m 644 $< $(OMF_DIR_)/$(OMF_NAME)
 	-scrollkeeper-update -p $(OMF_STATE) -o $(OMF_DIR_)
 
