diff -Nurd -x'*~' nautilus-cd-burner-2.14.2.orig/dvd_plus_rw_utils.cpp nautilus-cd-burner-2.14.2/dvd_plus_rw_utils.cpp
--- nautilus-cd-burner-2.14.2.orig/dvd_plus_rw_utils.cpp	2005-12-16 11:00:22.000000000 -0500
+++ nautilus-cd-burner-2.14.2/dvd_plus_rw_utils.cpp	2006-07-27 16:18:15.000000000 -0400
@@ -34,7 +34,7 @@
 #include <signal.h>
 #include <glib.h>
 
-#include "transport.hxx"
+//#include "transport.hxx"
 
 #ifdef LIBSTDCXX_HACK
 /* Some C++ stuff needed when we not link to libstdc++ */
@@ -70,6 +70,8 @@
 extern "C"
 int get_dvd_r_rw_profile (const char *name)
 {
+  return -1;
+# if 0
   Scsi_Command cmd;
   int retval = -1;
   unsigned char page[20];
@@ -154,11 +156,14 @@
 bail:
 
   return retval;
+#endif
 }
 
 extern "C"
 int get_mmc_profile (int fd)
 {
+  return -1;
+#if 0
   Scsi_Command cmd(fd);
   int retval = -1;
   unsigned char formats[260],disc_info[32];
@@ -297,11 +302,14 @@
 
   return retval;
 
+#endif
 }
 
 extern "C"
 int get_disc_status (int fd, int *empty, int *is_rewritable, int *is_blank)
 {
+  return -1;
+#if 0
   Scsi_Command cmd(fd);
   int retval = -1;
   unsigned char page[32];
@@ -344,11 +352,14 @@
 
   return retval;
 
+#endif
 }
 
 extern "C"
 int get_disc_size_cdr (int fd)
 {
+  return -1;
+#if 0
   Scsi_Command cmd(fd);
   int retval = -1;
   unsigned char toc[4], *atip;
@@ -386,8 +397,10 @@
 bail:
 
   return retval;
+#endif
 }
 
+#if 0
 static gint64
 get_read_capacity (Scsi_Command &cmd)
 {
@@ -417,15 +430,15 @@
 
   return retval;
 }
+#endif
 
 extern "C"
 gint64 get_disc_size_cd (int fd)
 {
-  Scsi_Command cmd (fd);
-
-  return get_read_capacity (cmd);
+  return -1;
 }
 
+#if 0
 static unsigned int get_2k_capacity (Scsi_Command &cmd, int mmc_profile)
 { unsigned char	buf[32];
   unsigned int	ret=0;
@@ -539,28 +552,27 @@
 
   return ret;
 }
+#endif
 
 /* Returns the size of a DVD in bytes */
 extern "C"
 gint64 get_disc_size_dvd (int fd)
 {
-  Scsi_Command cmd(fd);
-
-  return get_read_capacity (cmd);
+  return -1;
 }
 
 /* Returns the size of a DVD+R[W] or DVD-R[W] in bytes */
 extern "C"
 gint64 get_disc_size_dvdr (int fd, int mmc_profile)
 {
-  Scsi_Command cmd(fd);
-
-  return (gint64)get_2k_capacity(cmd, mmc_profile) * 2048;
+  return 0;
 }
 
 extern "C"
 int get_read_write_speed (int fd, int *read_speed, int *write_speed)
 {
+  return -1;
+#if 0
   Scsi_Command cmd(fd);
   int retval = -1;
   unsigned char header[12], *page2A, *p;
@@ -642,5 +654,6 @@
 
   return -1;
 
+#endif
 }
 
diff -Nurd -x'*~' nautilus-cd-burner-2.14.2.orig/make-iso.c nautilus-cd-burner-2.14.2/make-iso.c
--- nautilus-cd-burner-2.14.2.orig/make-iso.c	2005-11-30 12:04:44.000000000 -0500
+++ nautilus-cd-burner-2.14.2/make-iso.c	2006-07-27 17:37:02.000000000 -0400
@@ -26,12 +26,6 @@
 
 #include "config.h"
 
-#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
-#include <sys/param.h>
-#include <sys/mount.h>
-#else
-#include <sys/vfs.h>
-#endif /* __FreeBSD__ || __NetBSD__ || __OpenBSD__ */
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <time.h>
diff -Nurd -x'*~' nautilus-cd-burner-2.14.2.orig/mapping-protocol.c nautilus-cd-burner-2.14.2/mapping-protocol.c
--- nautilus-cd-burner-2.14.2.orig/mapping-protocol.c	2005-12-20 17:21:47.000000000 -0500
+++ nautilus-cd-burner-2.14.2/mapping-protocol.c	2006-07-27 16:13:13.000000000 -0400
@@ -28,6 +28,7 @@
 #include <string.h>
 #include <fcntl.h>
 #include <errno.h>
+#include <sys/time.h>
 
 #include "mapping-protocol.h"
 
