diff -uNr pango-1.16.0/Makefile.in pango-1.16.0-new/Makefile.in
--- pango-1.16.0/Makefile.in	2007-02-26 14:13:26.000000000 -0500
+++ pango-1.16.0-new/Makefile.in	2007-03-02 11:20:00.000000000 -0500
@@ -365,7 +365,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-SUBDIRS = pango modules pango-view examples docs tools tests
+SUBDIRS = pango modules pango-view          docs tools tests
 EXTRA_DIST = \
 	autogen.sh		\
 	pango.pc.in		\
diff -uNr pango-1.16.0/docs/Makefile.in pango-1.16.0-new/docs/Makefile.in
--- pango-1.16.0/docs/Makefile.in	2007-02-26 14:13:21.000000000 -0500
+++ pango-1.16.0-new/docs/Makefile.in	2007-03-02 11:20:30.000000000 -0500
@@ -376,9 +376,9 @@
 	-DPANGO_ENABLE_ENGINE	\
 	-I$(top_srcdir)		\
 	$(GLIB_CFLAGS)		\
+	$(FREETYPE_CFLAGS)	\
 	$(XFT_CFLAGS)		\
 	$(CAIRO_CFLAGS)		\
-	$(FREETYPE_CFLAGS)	\
 	$(X_CFLAGS)
 
 
diff -uNr pango-1.16.0/docs/html/pango-Miscellaneous-Utilities.html pango-1.16.0-new/docs/html/pango-Miscellaneous-Utilities.html
--- pango-1.16.0/docs/html/pango-Miscellaneous-Utilities.html	2007-02-26 14:20:26.000000000 -0500
+++ pango-1.16.0-new/docs/html/pango-Miscellaneous-Utilities.html	2007-03-02 11:22:27.000000000 -0500
@@ -427,7 +427,7 @@
 <a class="indexterm" name="id2928513"></a><pre class="programlisting">char*       pango_config_key_get            (const char *key);</pre>
 <p>
 Looks up a key in the Pango config database
-(pseudo-win.ini style, read from $sysconfdir/pango/pangorc,
+(pseudo-win.ini style, read from sysconfdir/pango/pangorc,
  ~/.pangorc, and getenv (PANGO_RC_FILE).)</p>
 <p>
 
diff -uNr pango-1.16.0/docs/pango-querymodules.1 pango-1.16.0-new/docs/pango-querymodules.1
--- pango-1.16.0/docs/pango-querymodules.1	2007-02-26 14:20:26.000000000 -0500
+++ pango-1.16.0-new/docs/pango-querymodules.1	2007-03-02 11:22:11.000000000 -0500
@@ -40,7 +40,7 @@
 .SH "ENVIRONMENT"
 
 .PP
-The Pango module path is specified by the key Pango/ModulesPath in the Pango config database, which is read from \fI\fIsysconfdir\fR/pango/pangorc\fR, \fI~/\&.pangorc\fR and the file specified in the environment variable \fBPANGO_RC_FILE\fR\&.
+The Pango module path is specified by the key Pango/ModulesPath in the Pango config database, which is read from \fI\fI@SYSCONFDIR@\fR/pango/pangorc\fR, \fI~/\&.pangorc\fR and the file specified in the environment variable \fBPANGO_RC_FILE\fR\&.
 
 .SH "BUGS"
 
diff -uNr pango-1.16.0/pango/Makefile.in pango-1.16.0-new/pango/Makefile.in
--- pango-1.16.0/pango/Makefile.in	2007-02-26 14:13:25.000000000 -0500
+++ pango-1.16.0-new/pango/Makefile.in	2007-03-02 11:23:33.000000000 -0500
@@ -515,11 +515,11 @@
 	-DSYSCONFDIR=\"$(sysconfdir)\"			\
 	-DLIBDIR=\"$(libdir)\"				\
 	-I$(top_srcdir)					\
+	$(FREETYPE_CFLAGS)				\
 	$(PANGO_DEBUG_FLAGS)				\
 	$(GLIB_CFLAGS)					\
 	$(XFT_CFLAGS)					\
 	$(CAIRO_CFLAGS)					\
-	$(FREETYPE_CFLAGS)				\
 	$(X_CFLAGS)
 
 BUILT_SOURCES = pango-enum-types.h pango-enum-types.c pango-features.h
@@ -623,9 +623,9 @@
 libpangoft2_1_0_la_LIBADD = \
 	opentype/libharfbuzz-1.la		\
 	libpango-$(PANGO_API_VERSION).la	\
+	$(FREETYPE_LIBS)			\
 	$(INCLUDED_FC_MODULES)			\
 	$(GLIB_LIBS)				\
-	$(FREETYPE_LIBS)			\
 	$(libm)
 
 libpangoft2_1_0_la_DEPENDENCIES = opentype/libharfbuzz-1.la \
diff -uNr pango-1.16.0/tests/Makefile.in pango-1.16.0-new/tests/Makefile.in
--- pango-1.16.0/tests/Makefile.in	2007-02-26 14:13:25.000000000 -0500
+++ pango-1.16.0-new/tests/Makefile.in	2007-03-02 11:24:30.000000000 -0500
@@ -385,7 +385,7 @@
 CLEANFILES = pangorc
 DISTCLEANFILES = all-unicode.txt runtests.log
 INCLUDES = -I$(top_srcdir) $(PANGO_DEBUG_FLAGS) $(GLIB_CFLAGS) \
-	$(XFT_CFLAGS) $(CAIRO_CFLAGS) $(FREETYPE_CFLAGS) $(X_CFLAGS) \
+	$(FREETYPE_CFLAGS) $(XFT_CFLAGS) $(CAIRO_CFLAGS) $(X_CFLAGS) \
 	$(am__append_1) $(am__append_2) $(am__append_3)
 TESTS = runtests.sh
 TESTS_ENVIRONMENT = \
diff -uNr pango-1.16.0/update-pango-modules pango-1.16.0-new/update-pango-modules
--- pango-1.16.0/update-pango-modules	1969-12-31 19:00:00.000000000 -0500
+++ pango-1.16.0-new/update-pango-modules	2007-03-02 11:24:47.000000000 -0500
@@ -0,0 +1,20 @@
+#!/bin/sh -e
+#
+# update-pango-modules
+#
+#   Packages which install pango module(s) should call update-pango-modules
+#   in their PostInstScript and PostRmScript to update the pango.module file.
+#
+
+CONFDIR=@SYSCONFDIR@/pango
+CONFFILE=$CONFDIR/pango.modules
+
+mkdir -p $CONFDIR
+
+TMPFILE=`tempfile`
+pango-querymodules >$TMPFILE
+
+mv -f $TMPFILE $CONFFILE
+chmod 0644 $CONFFILE
+
+exit 0
