diff -Nurd -x'*~' pygobject-2.20.0.orig/gobject/Makefile.in pygobject-2.20.0/gobject/Makefile.in
--- pygobject-2.20.0.orig/gobject/Makefile.in	2009-09-23 16:43:51.000000000 -0400
+++ pygobject-2.20.0/gobject/Makefile.in	2010-02-26 02:01:03.000000000 -0500
@@ -274,7 +274,8 @@
 generate_constants_CFLAGS = $(GLIB_CFLAGS) $(PYTHON_INCLUDES)
 CLEANFILES = constants.py
 EXTRA_DIST = constants.py.in
-_gobject_la_CFLAGS = -I$(top_srcdir)/glib $(PYTHON_INCLUDES) \
+INCLUDES = -I$(top_srcdir)/glib
+_gobject_la_CFLAGS = $(PYTHON_INCLUDES) \
 	$(FFI_CFLAGS) $(GLIB_CFLAGS) -DPY_SSIZE_T_CLEAN \
 	$(am__append_3)
 _gobject_la_LDFLAGS = $(common_ldflags) -export-symbols-regex "_gobject|PyInit__gobject"
