diff -uNr pygtk-2.10.4/py-compile pygtk-2.10.4-new/py-compile
--- pygtk-2.10.4/py-compile	2007-02-05 15:55:36.000000000 -0500
+++ pygtk-2.10.4-new/py-compile	2007-03-04 16:56:35.000000000 -0500
@@ -33,11 +33,24 @@
   PYTHON=python
 fi
 
+destdir=
 basedir=
 
+do_opts=1
+while [ $do_opts -ne 0 ]; do
+do_opts=0
 case "$1" in
+  --destdir)
+    shift
+    destdir=$1
+    shift
+    do_opts=1
+    ;;
   --basedir)
-    basedir=$2
+    shift
+    basedir=$1
+    shift
+    do_opts=1
     if test -z "$basedir"; then
       echo "$0: Missing argument to --basedir." 1>&2
       exit 1
@@ -60,6 +73,7 @@
     exit 0
     ;;
 esac
+done
 
 if [ $# = 0 ]; then
     echo "$0: No files given.  Try \`$0 --help' for more information." 1>&2
@@ -73,6 +87,11 @@
 else
     trans="path = os.path.join('$basedir', file)"
 fi
+if [ -z "$destdir$basedir" ]; then
+  trans_dest="dest = file"
+else
+  trans_dest="dest = os.path.join('$destdir$basedir', file)"
+fi
 
 $PYTHON -c "
 import sys, os, string, py_compile
@@ -81,11 +100,12 @@
 print 'Byte-compiling python modules...'
 for file in string.split(files):
     $trans
+    $trans_dest
     if not os.path.exists(path) or not (len(path) >= 3 and path[-3:] == '.py'):
 	continue
     print file,
     sys.stdout.flush()
-    py_compile.compile(path)
+    py_compile.compile(dest,None,path)
 print" || exit $?
 
 # this will fail for python < 1.5, but that doesn't matter ...
@@ -96,11 +116,12 @@
 print 'Byte-compiling python modules (optimized versions) ...'
 for file in string.split(files):
     $trans
+    $trans_dest
     if not os.path.exists(path) or not (len(path) >= 3 and path[-3:] == '.py'):
 	continue
     print file,
     sys.stdout.flush()
-    py_compile.compile(path)
+    py_compile.compile(dest,None,path)
 print" 2>/dev/null || :
 
 # Local Variables:
