diff -Nurd -x'*~' pyorbit-2.0.1.orig/py-compile pyorbit-2.0.1/py-compile
--- pyorbit-2.0.1.orig/py-compile	2004-02-15 07:22:50.000000000 -0500
+++ pyorbit-2.0.1/py-compile	2006-01-05 21:57:50.000000000 -0500
@@ -29,12 +29,24 @@
   PYTHON=python
 fi
 
+destdir=
 basedir=
 
+do_opts=1
+while [ $do_opts -ne 0 ]; do
+do_opts=0
 case "$1" in
+    --destdir)
+	shift
+	destdir=$1
+	shift
+	do_opts=1
+	;;
     --basedir)
-	basedir=$2
-	shift 2
+	shift
+	basedir=$1
+	shift
+	do_opts=1
 	;;
     --help)
 	echo "Usage: py-compile [--basedir DIR] PY_FILES ..."
@@ -47,6 +59,7 @@
 	exit 0
 	;;
 esac
+done
 
 if [ $# = 0 ]; then
     echo "No files given to $0" 1>&2
@@ -60,6 +73,11 @@
 else
     trans="path = os.path.join('$basedir', file)"
 fi
+if [ -z "$destdir$basedir" ]; then
+    trans_dest="dest = file"
+else
+    trans_dest="dest = os.path.join('$destdir$basedir', file)"
+fi
 
 $PYTHON -c "
 import sys, os, string, py_compile
@@ -68,11 +86,12 @@
 print 'Byte-compiling python modules...'
 for file in string.split(files):
     $trans
+    $trans_dest
     if not os.path.exists(path) or not (len(path) >= 3 and path[-3:] == '.py'):
 	continue
     print file,
     sys.stdout.flush()
-    py_compile.compile(path)
+    py_compile.compile(dest,None,path)
 print" || exit $?
 
 # this will fail for python < 1.5, but that doesn't matter ...
@@ -83,10 +102,11 @@
 print 'Byte-compiling python modules (optimized versions) ...'
 for file in string.split(files):
     $trans
+    $trans_dest
     if not os.path.exists(path) or not (len(path) >= 3 and path[-3:] == '.py'):
 	continue
     print file,
     sys.stdout.flush()
-    py_compile.compile(path)
+    py_compile.compile(dest,None,path)
 print" 2>/dev/null || :
 
