diff -uNr vte-0.15.3/src/pty.c vte-0.15.3-new/src/pty.c
--- vte-0.15.3/src/pty.c	2007-02-13 02:26:18.000000000 -0500
+++ vte-0.15.3-new/src/pty.c	2007-02-13 11:37:46.000000000 -0500
@@ -69,6 +69,12 @@
 static GTree *_vte_pty_helper_map = NULL;
 #endif
 
+#ifdef __APPLE__
+#include <crt_externs.h>
+#undef environ
+#define environ *_NSGetEnviron()
+#endif
+
 /* Reset the handlers for all known signals to their defaults.  The parent
  * (or one of the libraries it links to) may have changed one to be ignored. */
 static void
diff -uNr vte-0.15.3/src/vtefc.c vte-0.15.3-new/src/vtefc.c
--- vte-0.15.3/src/vtefc.c	2007-02-09 17:12:40.000000000 -0500
+++ vte-0.15.3-new/src/vtefc.c	2007-02-13 12:29:36.000000000 -0500
@@ -434,11 +434,13 @@
 			save = FcPatternDuplicate(tmp);
 			FcPatternDestroy(tmp);
 			g_ptr_array_add(pattern_array, save);
+#ifndef __APPLE__
 			_VTE_DEBUG_IF(VTE_DEBUG_MISC) {
 				FcChar8 *name = FcNameUnparse (save);
 				g_printerr("Added '%s' to fontset\n", name);
 				FcStrFree (name);
 			}
+#endif
 		}
 		FcFontSetDestroy(fontset);
 		ret = TRUE;
@@ -454,11 +456,13 @@
 			save = FcPatternDuplicate(tmp);
 			FcPatternDestroy(tmp);
 			g_ptr_array_add(pattern_array, save);
+#ifndef __APPLE__
 			_VTE_DEBUG_IF(VTE_DEBUG_MISC) {
 				FcChar8 *name = FcNameUnparse (save);
 				g_printerr("Added '%s' to fontset\n", name);
 				FcStrFree (name);
 			}
+#endif
 			ret = TRUE;
 		} else {
 			ret = FALSE;
