diff -Nurd -x'*~' vte-0.11.16.orig/po/Makefile.in.in vte-0.11.16/po/Makefile.in.in
--- vte-0.11.16.orig/po/Makefile.in.in	2005-04-12 13:27:50.000000000 -0400
+++ vte-0.11.16/po/Makefile.in.in	2005-07-19 02:48:05.000000000 -0400
@@ -27,7 +27,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 libdir = @libdir@
-localedir = $(libdir)/locale
+localedir = $(datadir)/locale
 gnulocaledir = $(datadir)/locale
 gettextsrcdir = $(datadir)/glib-2.0/gettext/po
 subdir = po
