diff -uNr vte-0.16.6/src/pty.c vte-0.16.6-new/src/pty.c
--- vte-0.16.6/src/pty.c	2007-02-26 16:05:14.000000000 -0500
+++ vte-0.16.6-new/src/pty.c	2007-07-06 09:34:27.000000000 -0400
@@ -69,7 +69,13 @@
 static GTree *_vte_pty_helper_map = NULL;
 #endif
 
+#ifdef __APPLE__
+#include <crt_externs.h>
+#undef environ
+#define environ *_NSGetEnviron()
+#else
 extern char **environ;
+#endif
 
 /* Reset the handlers for all known signals to their defaults.  The parent
  * (or one of the libraries it links to) may have changed one to be ignored. */
diff -uNr vte-0.16.6/src/vtefc.c vte-0.16.6-new/src/vtefc.c
--- vte-0.16.6/src/vtefc.c	2007-02-26 16:05:14.000000000 -0500
+++ vte-0.16.6-new/src/vtefc.c	2007-07-06 09:34:27.000000000 -0400
@@ -435,12 +435,14 @@
 			FcPatternDestroy(tmp);
 			g_ptr_array_add(pattern_array, save);
 #if defined(HAVE_FCSTRFREE) && defined(HAVE_FCNAMEUNPARSE)
+#ifndef __APPLE__
 			_VTE_DEBUG_IF(VTE_DEBUG_MISC) {
 				FcChar8 *name = FcNameUnparse (save);
 				g_printerr("Added '%s' to fontset\n", name);
 				FcStrFree (name);
 			}
 #endif
+#endif
 		}
 		FcFontSetDestroy(fontset);
 		ret = TRUE;
@@ -457,12 +459,14 @@
 			FcPatternDestroy(tmp);
 			g_ptr_array_add(pattern_array, save);
 #if defined(HAVE_FCSTRFREE) && defined(HAVE_FCNAMEUNPARSE)
+#ifndef __APPLE__
 			_VTE_DEBUG_IF(VTE_DEBUG_MISC) {
 				FcChar8 *name = FcNameUnparse (save);
 				g_printerr("Added '%s' to fontset\n", name);
 				FcStrFree (name);
 			}
 #endif
+#endif
 			ret = TRUE;
 		} else {
 			ret = FALSE;
