diff -ruN gcc-4.9.2-orig/gcc/config/avr/gen-avr-mmcu-texi.c gcc-4.9.2/gcc/config/avr/gen-avr-mmcu-texi.c
--- gcc-4.9.2-orig/gcc/config/avr/gen-avr-mmcu-texi.c	2014-01-02 16:23:26.000000000 -0600
+++ gcc-4.9.2/gcc/config/avr/gen-avr-mmcu-texi.c	2024-09-17 04:53:20.000000000 -0500
@@ -19,6 +19,7 @@
 
 #include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 
 #define IN_GEN_AVR_MMCU_TEXI
 
diff -ruN gcc-4.9.2-orig/gcc/system.h gcc-4.9.2/gcc/system.h
--- gcc-4.9.2-orig/gcc/system.h	2014-01-02 16:23:26.000000000 -0600
+++ gcc-4.9.2/gcc/system.h	2024-09-17 05:02:46.000000000 -0500
@@ -202,6 +202,7 @@
 #endif
 
 #ifdef __cplusplus
+# include <map>
 # include <cstring>
 #endif
 
