diff -Nurd bison-2.7.1.orig/configure bison-2.7.1/configure
--- bison-2.7.1.orig/configure	2013-04-15 05:07:35
+++ bison-2.7.1/configure	2025-12-03 23:02:33
@@ -37582,6 +37582,9 @@
 
 
 
+printf "%s\n" "disabling java by packager fiat" >&6
+CONF_JAVA=
+CONF_JAVAC=
 
 
 
diff -Nurd bison-2.7.1.orig/lib/vasnprintf.c bison-2.7.1/lib/vasnprintf.c
--- bison-2.7.1.orig/lib/vasnprintf.c	2013-04-08 05:19:44
+++ bison-2.7.1/lib/vasnprintf.c	2025-12-03 20:34:38
@@ -4870,7 +4870,7 @@
 #endif
                   *fbp = dp->conversion;
 #if USE_SNPRINTF
-# if !(((__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 3)) && !defined __UCLIBC__) || ((defined _WIN32 || defined __WIN32__) && ! defined __CYGWIN__))
+# if !defined(__APPLE__) && !(((__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 3)) && !defined __UCLIBC__) || ((defined _WIN32 || defined __WIN32__) && ! defined __CYGWIN__))
                 fbp[1] = '%';
                 fbp[2] = 'n';
                 fbp[3] = '\0';
