diff -ruN doxygen-1.8.9.1-orig/Makefile.in doxygen-1.8.9.1/Makefile.in
--- doxygen-1.8.9.1-orig/Makefile.in	2014-11-13 14:22:20.000000000 -0600
+++ doxygen-1.8.9.1/Makefile.in	2017-06-11 16:14:31.000000000 -0500
@@ -73,7 +73,7 @@
 
 DATE=$(shell date "+%B %Y")
 
-MAN1DIR = man/man1
+MAN1DIR = share/man/man1
 
 install: doxywizard_install doxysearch_install
 	$(INSTTOOL) -d $(DESTDIR)$(INSTALL)/bin
diff -ruN doxygen-1.8.9.1-orig/addon/doxywizard/doxywizard.pro.in doxygen-1.8.9.1/addon/doxywizard/doxywizard.pro.in
--- doxygen-1.8.9.1-orig/addon/doxywizard/doxywizard.pro.in	2014-12-27 08:28:27.000000000 -0600
+++ doxygen-1.8.9.1/addon/doxywizard/doxywizard.pro.in	2017-06-11 16:15:25.000000000 -0500
@@ -6,7 +6,7 @@
 DESTDIR      = ../../bin
 TARGET       = 
 DEPENDPATH  += .
-INCLUDEPATH += . ../../generated_src/doxywizard
+INCLUDEPATH += . ../../generated_src/doxywizard @PREFIX@/include
 QT          += xml
 CONFIG      += $extraopts
 OBJECTS_DIR  = ../../objects/doxywizard
@@ -14,10 +14,6 @@
 RCC_DIR      = ../../rcc/doxywizard
 DEFINES     += QT_NO_CAST_FROM_ASCII QT_NO_CAST_TO_ASCII
 
-macx-g++ {
-  CONFIG += x86 x86_64
-}
-
 # Input
 HEADERS     += doxywizard.h version.h expert.h config.h helplabel.h \
                inputbool.h inputstring.h inputint.h inputstrlist.h wizard.h docintf.h
diff -ruN doxygen-1.8.9.1-orig/configure doxygen-1.8.9.1/configure
--- doxygen-1.8.9.1-orig/configure	2015-01-04 05:00:55.000000000 -0600
+++ doxygen-1.8.9.1/configure	2017-06-11 16:18:19.000000000 -0500
@@ -845,7 +845,7 @@
     TMAKE_CXXFLAGS += -D_LARGEFILE_SOURCE
 EOF
 fi
-if test "$f_platform" = "macosx-uni-c++"; then
+if false; then
   if test -n "`ls /Developer/SDKs/MacOSX10.*.sdk 2>/dev/null`"; then
     mac_sdk=MacOSX10.4u.sdk
   fi
diff -ruN doxygen-1.8.9.1-orig/qtools/qtools.pro.in doxygen-1.8.9.1/qtools/qtools.pro.in
--- doxygen-1.8.9.1-orig/qtools/qtools.pro.in	2014-10-24 04:51:53.000000000 -0500
+++ doxygen-1.8.9.1/qtools/qtools.pro.in	2017-06-11 16:18:41.000000000 -0500
@@ -92,7 +92,7 @@
 		  qmutex_win32.cpp \
 		  qwaitcondition_win32.cpp
 
-INCLUDEPATH = .
+INCLUDEPATH = . @PREFIX@/include
 #TMAKE_CXXFLAGS += -DQT_NO_CODECS -DQT_LITE_UNICODE
 TMAKE_CXXFLAGS += -DQT_LITE_UNICODE
 win32:TMAKE_CXXFLAGS += -DQT_NODLL
diff -ruN doxygen-1.8.9.1-orig/src/doxygen.pro.in doxygen-1.8.9.1/src/doxygen.pro.in
--- doxygen-1.8.9.1-orig/src/doxygen.pro.in	2014-10-19 09:39:17.000000000 -0500
+++ doxygen-1.8.9.1/src/doxygen.pro.in	2017-06-11 16:19:47.000000000 -0500
@@ -18,7 +18,7 @@
 CONFIG       =	console warn_on $extraopts
 HEADERS      =	doxygen.h 
 SOURCES      =	main.cpp 
-unix:LIBS                  += -L../lib -ldoxygen -lvhdlparser -ldoxycfg -lqtools -lmd5 -lpthread %%SQLITE3_LIBS%% %%LIBCLANG_LIBS%%
+unix:LIBS                  += -L../lib -L@PREFIX@/lib -ldoxygen -lvhdlparser -ldoxycfg -lqtools ../lib/libmd5.a -lpthread %%SQLITE3_LIBS%% %%LIBCLANG_LIBS%%
 win32:INCLUDEPATH          += .
 win32-mingw:LIBS           += -L../lib -ldoxygen -ldoxycfg -lvhdlparser -lqtools -lmd5 -lpthread -llibiconv -lole32 %%SQLITE3_LIBS%% %%LIBCLANG_LIBS%%
 win32-msvc:LIBS            += qtools.lib md5.lib doxygen.lib doxycfg.lib vhdlparser.lib shell32.lib iconv.lib
@@ -29,7 +29,7 @@
 win32-g++:LIBS             = -L../lib -ldoxygen -ldoxycfg -lvhdlparser -lqtools -lmd5 -liconv -lpthread %%SQLITE3_LIBS%% %%LIBCLANG_LIBS%% -Wl,--as-needed -lole32
 win32-g++:TMAKE_CXXFLAGS   += -fno-exceptions -fno-rtti
 DEPENDPATH                 += ../generated_src/doxygen
-INCLUDEPATH                += ../qtools ../libmd5 . ../vhdlparser
+INCLUDEPATH                += ../qtools ../libmd5 . ../vhdlparser @PREFIX@/include
 DESTDIR                    =  ../bin
 TARGET                     =  doxygen
 unix:TARGETDEPS            =  ../lib/libdoxygen.a ../lib/libdoxycfg.a
diff -ruN doxygen-1.8.9.1-orig/src/libdoxycfg.pro.in doxygen-1.8.9.1/src/libdoxycfg.pro.in
--- doxygen-1.8.9.1-orig/src/libdoxycfg.pro.in	2014-10-19 09:39:17.000000000 -0500
+++ doxygen-1.8.9.1/src/libdoxycfg.pro.in	2017-06-11 21:08:20.000000000 -0500
@@ -21,7 +21,7 @@
 win32:TMAKE_CXXFLAGS       += -DQT_NODLL
 win32-g++:TMAKE_CXXFLAGS   += -fno-exceptions -fno-rtti
 DEPENDPATH                 += ../generated_src/doxygen
-INCLUDEPATH                += ../generated_src/doxygen . ../qtools
+INCLUDEPATH                += ../generated_src/doxygen . ../qtools @PREFIX@/include
 DESTDIR                    =  ../lib
 TARGET                     =  doxycfg
 OBJECTS_DIR                =  ../objects/doxygen
diff -ruN doxygen-1.8.9.1-orig/src/libdoxygen.pro.in doxygen-1.8.9.1/src/libdoxygen.pro.in
--- doxygen-1.8.9.1-orig/src/libdoxygen.pro.in	2014-11-13 14:22:20.000000000 -0600
+++ doxygen-1.8.9.1/src/libdoxygen.pro.in	2017-06-11 16:20:41.000000000 -0500
@@ -234,6 +234,7 @@
 INCLUDEPATH                += ../generated_src/doxygen ../src ../qtools ../libmd5 ../vhdlparser
 INCLUDEPATH                += %%SQLITE3_INC%%
 INCLUDEPATH                += %%LIBCLANG_INC%%
+INCLUDEPATH                += @PREFIX@/include
 DEPENDPATH                 += ../generated_src/doxygen ../qtools ../libmd5 ../vhdlparser
 win32:INCLUDEPATH          += .
 DESTDIR                    =  ../lib
diff -ruN doxygen-1.8.9.1-orig/src/portable_c.c doxygen-1.8.9.1/src/portable_c.c
--- doxygen-1.8.9.1-orig/src/portable_c.c	2014-10-19 09:39:17.000000000 -0500
+++ doxygen-1.8.9.1/src/portable_c.c	2017-06-11 17:50:10.000000000 -0500
@@ -1,9 +1,10 @@
-#if (defined(__APPLE__) || defined(macintosh)) && !defined(DMG_BUILD)
+//#if (defined(__APPLE__) || defined(macintosh)) && !defined(DMG_BUILD)
 // define this before including iconv.h to avoid a mapping of
 // iconv_open and friends to libicon_open (done by mac ports),
 // while the symbols without 'lib' are linked from /usr/lib/libiconv
-#define LIBICONV_PLUG
-#endif
+// Hide LIBICONV_PLUG if using non-system iconv
+//#define LIBICONV_PLUG
+//#endif
 #include <iconv.h>
 
 // These functions are implemented in a C file, because there are different
diff -ruN doxygen-1.8.9.1-orig/tmake/lib/macosx-c++/tmake.conf doxygen-1.8.9.1/tmake/lib/macosx-c++/tmake.conf
--- doxygen-1.8.9.1-orig/tmake/lib/macosx-c++/tmake.conf	2014-10-24 04:51:54.000000000 -0500
+++ doxygen-1.8.9.1/tmake/lib/macosx-c++/tmake.conf	2017-06-11 21:03:43.000000000 -0500
@@ -7,7 +7,7 @@
 TEMPLATE		= app
 CONFIG			= qt warn_on release
 
-TMAKE_CC		= cc
+TMAKE_CC		= cc -MD
 TMAKE_CFLAGS		= -pipe 
 TMAKE_CFLAGS_WARN_ON	= -Wall -W -Wno-deprecated-declarations 
 TMAKE_CFLAGS_WARN_OFF	=
@@ -16,8 +16,8 @@
 TMAKE_CFLAGS_SHLIB	= -fPIC
 TMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
 
-TMAKE_CXX		= c++
-TMAKE_CXXFLAGS		= $$TMAKE_CFLAGS -mmacosx-version-min=10.5 -DYY_TYPEDEF_YY_SIZE_T -Dyy_size_t=int
+TMAKE_CXX		= c++ -MD
+TMAKE_CXXFLAGS		= $$TMAKE_CFLAGS -DYY_TYPEDEF_YY_SIZE_T -Dyy_size_t=int
 TMAKE_CXXFLAGS_WARN_ON	= $$TMAKE_CFLAGS_WARN_ON
 TMAKE_CXXFLAGS_WARN_OFF	= $$TMAKE_CFLAGS_WARN_OFF
 TMAKE_CXXFLAGS_RELEASE	= $$TMAKE_CFLAGS_RELEASE
@@ -31,8 +31,8 @@
 TMAKE_LIBDIR_X11	= 
 TMAKE_INCDIR_QT		= $(QTDIR)/include
 TMAKE_LIBDIR_QT		= $(QTDIR)/lib
-TMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-TMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
+TMAKE_INCDIR_OPENGL	= /opt/X11/include
+TMAKE_LIBDIR_OPENGL	= /opt/X11/lib
 
 TMAKE_LINK		= c++
 TMAKE_LINK_SHLIB	= c++
diff -ruN doxygen-1.8.9.1-orig/tmake/lib/macosx-intel-c++/tmake.conf doxygen-1.8.9.1/tmake/lib/macosx-intel-c++/tmake.conf
--- doxygen-1.8.9.1-orig/tmake/lib/macosx-intel-c++/tmake.conf	2014-10-19 09:39:17.000000000 -0500
+++ doxygen-1.8.9.1/tmake/lib/macosx-intel-c++/tmake.conf	2017-06-11 16:16:19.000000000 -0500
@@ -8,7 +8,7 @@
 CONFIG			= qt warn_on release
 
 TMAKE_CC		= cc
-TMAKE_CFLAGS		= -pipe -arch i386 -arch x86_64 
+TMAKE_CFLAGS		= -pipe -arch @FINK_ARCH@
 TMAKE_CFLAGS_WARN_ON	= -Wall -W -Wno-deprecated-declarations 
 TMAKE_CFLAGS_WARN_OFF	=
 TMAKE_CFLAGS_RELEASE	= -O2
@@ -36,7 +36,7 @@
 
 TMAKE_LINK		= c++
 TMAKE_LINK_SHLIB	= c++
-TMAKE_LFLAGS		= -Wl,-search_paths_first -arch i386 -arch x86_64 
+TMAKE_LFLAGS		= -Wl,-search_paths_first -arch @FINK_ARCH@
 TMAKE_LFLAGS_RELEASE	=
 TMAKE_LFLAGS_DEBUG	=
 TMAKE_LFLAGS_SHLIB	= -shared
diff -ruN doxygen-1.8.9.1-orig/tmake/lib/macosx-uni-c++/tmake.conf doxygen-1.8.9.1/tmake/lib/macosx-uni-c++/tmake.conf
--- doxygen-1.8.9.1-orig/tmake/lib/macosx-uni-c++/tmake.conf	2014-10-19 09:39:17.000000000 -0500
+++ doxygen-1.8.9.1/tmake/lib/macosx-uni-c++/tmake.conf	2017-06-11 16:16:51.000000000 -0500
@@ -8,7 +8,7 @@
 CONFIG			= qt warn_on release
 
 TMAKE_CC		= cc
-TMAKE_CFLAGS		= -pipe -arch i386 -arch ppc 
+TMAKE_CFLAGS		= -pipe -arch @FINK_ARCH@
 TMAKE_CFLAGS_WARN_ON	= -Wall -W -Wno-deprecated-declarations
 TMAKE_CFLAGS_WARN_OFF	=
 TMAKE_CFLAGS_RELEASE	= -O2
@@ -36,7 +36,7 @@
 
 TMAKE_LINK		= c++
 TMAKE_LINK_SHLIB	= c++
-TMAKE_LFLAGS		= -Wl,-search_paths_first -arch i386 -arch ppc 
+TMAKE_LFLAGS		= -Wl,-search_paths_first -arch @FINK_ARCH@
 TMAKE_LFLAGS_RELEASE	=
 TMAKE_LFLAGS_DEBUG	=
 TMAKE_LFLAGS_SHLIB	= -shared
