diff --git a/bin/git-cola b/bin/git-cola
index 3b4c8fd..7cc15bf 100755
--- a/bin/git-cola
+++ b/bin/git-cola
@@ -36,6 +36,8 @@ def setup_environment():
     # It is assumed that the user installed Cola using the --prefix= option
     python2 = sys.version_info[0] == 2
     prefix = dirname(dirname(realpath(abspath(__file__))))
+    if os.path.islink(__file__):
+        prefix = dirname(dirname(os.readlink(__file__)))
     if python2:
         cola_mod = join(prefix, str('cola'), str('__init__.py'))
         install_lib = join(prefix, str('share'), str('git-cola'), str('lib'))
@@ -50,6 +52,8 @@ def setup_environment():
     else:
         # Install tree
         sys.path.insert(1, install_lib)
+        fink_lib = os.path.join(prefix, 'lib', '@PYTHON@', 'site-packages')
+        sys.path.insert(1, fink_lib)
 setup_environment()
 
 from cola.app import add_common_arguments
diff --git a/bin/git-dag b/bin/git-dag
index f849510..79be316 100755
--- a/bin/git-dag
+++ b/bin/git-dag
@@ -35,6 +35,8 @@ def setup_environment():
     # It is assumed that the user installed Cola using the --prefix= option
     python2 = sys.version_info[0] == 2
     prefix = dirname(dirname(realpath(abspath(__file__))))
+    if os.path.islink(__file__):
+        prefix = dirname(dirname(os.readlink(__file__)))
     if python2:
         cola_mod = join(prefix, str('cola'), str('__init__.py'))
         install_lib = join(prefix, str('share'), str('git-cola'), str('lib'))
@@ -49,6 +51,8 @@ def setup_environment():
     else:
         # Install tree
         sys.path.insert(1, install_lib)
+        fink_lib = os.path.join(prefix, 'lib', '@PYTHON@', 'site-packages')
+        sys.path.insert(1, fink_lib)
 setup_environment()
 
 
diff --git a/share/git-cola/bin/git-xbase b/share/git-cola/bin/git-xbase
index c8e56bb..00e702c 100755
--- a/share/git-cola/bin/git-xbase
+++ b/share/git-cola/bin/git-xbase
@@ -11,12 +11,16 @@ from os.path import dirname
 
 def setup_environment():
     prefix = dirname(dirname(dirname(dirname(abspath(__file__)))))
+    if os.path.islink(__file__):
+        prefix = dirname(dirname(os.readlink(__file__)))
     source_tree = os.path.join(prefix, 'cola', '__init__.py')
     if os.path.exists(source_tree):
         modules = prefix
     else:
         modules = os.path.join(prefix, 'share', 'git-cola', 'lib')
     sys.path.insert(1, modules)
+    fink_lib = os.path.join(prefix, 'lib', '@PYTHON@', 'site-packages')
+    sys.path.insert(1, fink_lib)
 setup_environment()
 
 from cola import app
