diff -ru git-2.18.0.orig/contrib/completion/git-completion.bash git-2.18.0/contrib/completion/git-completion.bash
--- git-2.18.0.orig/contrib/completion/git-completion.bash	2018-06-21 13:18:50.000000000 -0400
+++ git-2.18.0/contrib/completion/git-completion.bash	2018-06-30 21:28:23.000000000 -0400
@@ -39,6 +39,9 @@
 #     When set to "1", do not include "DWIM" suggestions in git-checkout
 #     completion (e.g., completing "foo" when "origin/foo" exists).
 
+# Make sure we're not in posix mode, otherwise process substitution is invalid.
+set +o posix
+
 case "$COMP_WORDBREAKS" in
 *:*) : great ;;
 *)   COMP_WORDBREAKS="$COMP_WORDBREAKS:"
