diff -ruN gnulib-ac9dd00-orig/build-aux/vcs-to-changelog.py gnulib-ac9dd00/build-aux/vcs-to-changelog.py
--- gnulib-ac9dd00-orig/build-aux/vcs-to-changelog.py	2025-03-03 15:02:04.000000000 -0600
+++ gnulib-ac9dd00/build-aux/vcs-to-changelog.py	2025-04-17 06:29:03.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!@FINK_PREFIX@/bin/python3.10
 # Main VCSToChangeLog script.
 # Copyright (C) 2019-2025 Free Software Foundation, Inc.
 #
diff -ruN gnulib-ac9dd00-orig/gnulib-tool gnulib-ac9dd00/gnulib-tool
--- gnulib-ac9dd00-orig/gnulib-tool	2025-03-03 15:02:04.000000000 -0600
+++ gnulib-ac9dd00/gnulib-tool	2025-04-17 06:28:55.000000000 -0500
@@ -157,10 +157,10 @@
   '')
     # Use the Python implementation if a suitable Python version is found
     # in $PATH. This is the same Python version test as in gnulib-tool.py.
-    if (python3 -c 'import sys; sys.exit(not sys.version_info >= (3,7))') 2>/dev/null; then
+    if (@FINK_PREFIX@/bin/python3.10 -c 'import sys; sys.exit(not sys.version_info >= (3,7))') 2>/dev/null; then
       exec "$prog.py" "$@"
     else
-      echo "gnulib-tool: warning: python3 not found or too old, using the slow shell-based implementation" 1>&2
+      echo "gnulib-tool: warning: @FINK_PREFIX@/bin/python3.10 not found or too old, using the slow shell-based implementation" 1>&2
       exec "$prog.sh" "$@"
     fi
     ;;
diff -ruN gnulib-ac9dd00-orig/gnulib-tool.py gnulib-ac9dd00/gnulib-tool.py
--- gnulib-ac9dd00-orig/gnulib-tool.py	2025-03-03 15:02:04.000000000 -0600
+++ gnulib-ac9dd00/gnulib-tool.py	2025-04-17 06:28:46.000000000 -0500
@@ -143,20 +143,20 @@
 fi
 
 # Check the Python version.
-if (python3 -c 'import sys; sys.exit(not sys.version_info >= (3,7))') 2>/dev/null; then
+if (@FINK_PREFIX@/bin/python3.10 -c 'import sys; sys.exit(not sys.version_info >= (3,7))') 2>/dev/null; then
   :
 else
-  if (python3 --version) >/dev/null 2>/dev/null; then
-    case `python3 --version 2>&1` in
+  if (@FINK_PREFIX@/bin/python3.10 --version) >/dev/null 2>/dev/null; then
+    case `@FINK_PREFIX@/bin/python3.10 --version 2>&1` in
       Python\ 3.[0-6] | Python\ 3.[0-6].*)
         func_fatal_error "python3 is too old (minimum required version is 3.7); try setting GNULIB_TOOL_IMPL=sh" ;;
       Python\ 3.*)
         ;;
       *)
-        func_fatal_error "python3 version is unsupported" ;;
+        func_fatal_error "@FINK_PREFIX@/bin/python3.10 version is unsupported" ;;
     esac
   else
-    func_fatal_error "python3 not found; try setting GNULIB_TOOL_IMPL=sh"
+    func_fatal_error "@FINK_PREFIX@/bin/python3.10 not found; try setting GNULIB_TOOL_IMPL=sh"
   fi
 fi
 
@@ -169,4 +169,4 @@
 profiler_args=
 # For profiling, cf. <https://docs.python.org/3/library/profile.html>.
 #profiler_args="-m cProfile -s tottime"
-exec python3 $profiler_args -- "$prog" "$@"
+exec @FINK_PREFIX@/bin/python3.10 $profiler_args -- "$prog" "$@"
