diff -ruN apache-maven-3.6.3-orig/bin/mvn apache-maven-3.6.3/bin/mvn
--- apache-maven-3.6.3-orig/bin/mvn	2019-11-07 06:32:18.000000000 -0600
+++ apache-maven-3.6.3/bin/mvn	2021-12-05 05:46:54.000000000 -0600
@@ -33,6 +33,10 @@
     . /etc/mavenrc
   fi
 
+  if [ -f @FINKPREFIX@/etc/maven/mavenrc ] ; then
+    . @FINKPREFIX@/etc/maven/mavenrc
+  fi
+
   if [ -f "$HOME/.mavenrc" ] ; then
     . "$HOME/.mavenrc"
   fi
@@ -104,6 +108,7 @@
 
 CLASSWORLDS_JAR=`echo "${MAVEN_HOME}"/boot/plexus-classworlds-*.jar`
 CLASSWORLDS_LAUNCHER=org.codehaus.plexus.classworlds.launcher.Launcher
+[ -z "$CLASSWORLDS_CONF" ] && CLASSWORLDS_CONF="${M2_HOME}/bin/m2.conf"
 
 # For Cygwin, switch paths to Windows format before running java
 if $cygwin ; then
@@ -192,7 +197,7 @@
   $MAVEN_OPTS \
   $MAVEN_DEBUG_OPTS \
   -classpath "${CLASSWORLDS_JAR}" \
-  "-Dclassworlds.conf=${MAVEN_HOME}/bin/m2.conf" \
+  "-Dclassworlds.conf=${CLASSWORLDS_CONF}" \
   "-Dmaven.home=${MAVEN_HOME}" \
   "-Dlibrary.jansi.path=${MAVEN_HOME}/lib/jansi-native" \
   "-Dmaven.multiModuleProjectDirectory=${MAVEN_PROJECTBASEDIR}" \
