diff -ruN apache-maven-3.2.3-orig/bin/mvn apache-maven-3.2.3/bin/mvn
--- apache-maven-3.2.3-orig/bin/mvn	2014-08-11 15:59:27.000000000 -0500
+++ apache-maven-3.2.3/bin/mvn	2014-08-23 08:55:40.000000000 -0500
@@ -40,6 +40,10 @@
     . /etc/mavenrc
   fi
 
+  if [ -f @FINKPREFIX@/etc/maven/mavenrc ] ; then
+    . @FINKPREFIX@/etc/maven/mavenrc
+  fi
+
   if [ -f "$HOME/.mavenrc" ] ; then
     . "$HOME/.mavenrc"
   fi
@@ -178,6 +182,7 @@
 fi
 
 CLASSWORLDS_LAUNCHER=org.codehaus.plexus.classworlds.launcher.Launcher
+[ -z "$CLASSWORLDS_CONF" ] && CLASSWORLDS_CONF="${M2_HOME}/bin/m2.conf"
 
 # For Cygwin, switch paths to Windows format before running java
 if $cygwin; then
@@ -197,6 +202,6 @@
 exec "$JAVACMD" \
   $MAVEN_OPTS \
   -classpath "${M2_HOME}"/boot/plexus-classworlds-*.jar \
-  "-Dclassworlds.conf=${M2_HOME}/bin/m2.conf" \
+  "-Dclassworlds.conf=${CLASSWORLDS_CONF}" \
   "-Dmaven.home=${M2_HOME}"  \
-  ${CLASSWORLDS_LAUNCHER} "$@"
\ No newline at end of file
+  ${CLASSWORLDS_LAUNCHER} "$@"
diff -ruN apache-maven-3.2.3-orig/bin/mvnDebug apache-maven-3.2.3/bin/mvnDebug
--- apache-maven-3.2.3-orig/bin/mvnDebug	2014-08-11 15:59:27.000000000 -0500
+++ apache-maven-3.2.3/bin/mvnDebug	2014-08-23 08:55:54.000000000 -0500
@@ -44,6 +44,10 @@
     . /etc/mavenrc
   fi
 
+  if [ -f @FINKPREFIX@/etc/maven/mavenrc ] ; then
+    . @FINKPREFIX@/etc/maven/mavenrc
+  fi
+
   if [ -f "$HOME/.mavenrc" ] ; then
     . "$HOME/.mavenrc"
   fi
@@ -175,6 +179,7 @@
 fi
 
 CLASSWORLDS_LAUNCHER=org.codehaus.plexus.classworlds.launcher.Launcher
+[ -z "$CLASSWORLDS_CONF" ] && CLASSWORLDS_CONF="${M2_HOME}/bin/m2.conf"
 
 # For Cygwin, switch paths to Windows format before running java
 if $cygwin; then
@@ -195,6 +200,6 @@
   $MAVEN_OPTS \
   $MAVEN_DEBUG_OPTS \
   -classpath "${M2_HOME}"/boot/plexus-classworlds-*.jar \
-  "-Dclassworlds.conf=${M2_HOME}/bin/m2.conf" \
+  "-Dclassworlds.conf=${CLASSWORLDS_CONF}" \
   "-Dmaven.home=${M2_HOME}"  \
-  ${CLASSWORLDS_LAUNCHER} "$@"
\ No newline at end of file
+  ${CLASSWORLDS_LAUNCHER} "$@"
diff -ruN apache-maven-3.2.3-orig/bin/mvnyjp apache-maven-3.2.3/bin/mvnyjp
--- apache-maven-3.2.3-orig/bin/mvnyjp	2014-08-11 15:59:27.000000000 -0500
+++ apache-maven-3.2.3/bin/mvnyjp	2014-08-23 08:56:59.000000000 -0500
@@ -48,6 +48,10 @@
     . /etc/mavenrc
   fi
 
+  if [ -f @FINKPREFIX@/etc/maven/mavenrc ] ; then
+    . @FINKPREFIX@/etc/maven/mavenrc
+  fi
+
   if [ -f "$HOME/.mavenrc" ] ; then
     . "$HOME/.mavenrc"
   fi
@@ -179,6 +183,7 @@
 fi
 
 CLASSWORLDS_LAUNCHER=org.codehaus.plexus.classworlds.launcher.Launcher
+[ -z "$CLASSWORLDS_CONF" ] && CLASSWORLDS_CONF="${M2_HOME}/bin/m2.conf"
 
 # For Cygwin, switch paths to Windows format before running java
 if $cygwin; then
@@ -200,7 +205,7 @@
 exec "$JAVACMD" \
   $MAVEN_OPTS \
   -classpath "${M2_HOME}"/boot/plexus-classworlds-*.jar \
-  "-Dclassworlds.conf=${M2_HOME}/bin/m2.conf" \
+  "-Dclassworlds.conf=${CLASSWORLDS_CONF}" \
   "-Dmaven.home=${M2_HOME}"  \
   ${CLASSWORLDS_LAUNCHER} $QUOTED_ARGS
 
