diff -ru mercurial-3.7.1.orig/hgext/zeroconf/__init__.py mercurial-3.7.1/hgext/zeroconf/__init__.py
--- mercurial-3.7.1.orig/hgext/zeroconf/__init__.py	2016-02-03 16:59:58.000000000 -0500
+++ mercurial-3.7.1/hgext/zeroconf/__init__.py	2016-02-20 19:11:35.000000000 -0500
@@ -163,8 +163,8 @@
                 return path
     return orig(self, section, key, default, untrusted)
 
-def configitems(orig, self, section, untrusted=False):
-    repos = orig(self, section, untrusted)
+def configitems(orig, self, section, untrusted=False, ignoresub=False):
+    repos = orig(self, section, untrusted, ignoresub)
     if section == "paths":
         repos += getzcpaths()
     return repos
