--- doc/exts/pylint_options.py-orig	2022-07-17 14:44:17.000000000 -0500
+++ doc/exts/pylint_options.py	2024-09-10 04:18:19.000000000 -0500
@@ -68,8 +68,10 @@
 ) -> str:
     checker_string = get_rst_title(f"``{checker.capitalize()}`` Checker", "^")
     toml_doc = tomlkit.document()
+    tool_table = tomlkit.table(is_super_table=True)
+    toml_doc.add(tomlkit.key("tool"), tool_table)
     pylint_tool_table = tomlkit.table(is_super_table=True)
-    toml_doc.add(tomlkit.key(["tool", "pylint"]), pylint_tool_table)
+    tool_table.add(tomlkit.key("pylint"), pylint_tool_table)
 
     checker_table = tomlkit.table()
 
--- doc/user_guide/configuration/all-options.rst-orig	2022-07-17 14:44:17.000000000 -0500
+++ doc/user_guide/configuration/all-options.rst	2024-09-10 04:20:16.000000000 -0500
@@ -228,7 +228,7 @@
   *Minimum Python version to use for version dependent checks. Will default to the version used to run pylint.*
 
 Default:
-  ``(3, 10)``
+  ``(3, 11)``
 
 
 --recursive
@@ -334,7 +334,7 @@
 
    persistent = true
 
-   py-version = [3, 10]
+   py-version = [3, 11]
 
    recursive = false
 
--- pylint/config/arguments_manager.py-orig	2022-07-17 14:44:17.000000000 -0500
+++ pylint/config/arguments_manager.py	2024-09-10 04:21:52.000000000 -0500
@@ -666,8 +666,11 @@
         stdout.
         """
         toml_doc = tomlkit.document()
+        tool_table = tomlkit.table(is_super_table=True)
+        toml_doc.add(tomlkit.key("tool"), tool_table)
+
         pylint_tool_table = tomlkit.table(is_super_table=True)
-        toml_doc.add(tomlkit.key(["tool", "pylint"]), pylint_tool_table)
+        tool_table.add(tomlkit.key("pylint"), pylint_tool_table)
 
         for group in sorted(
             self._arg_parser._action_groups,
