diff -ruN valkyrie-2.0.0-orig/src/objects/tool_object.cpp valkyrie-2.0.0/src/objects/tool_object.cpp
--- valkyrie-2.0.0-orig/src/objects/tool_object.cpp	2010-10-20 07:23:29.000000000 -0500
+++ valkyrie-2.0.0/src/objects/tool_object.cpp	2015-03-01 08:50:29.000000000 -0600
@@ -47,6 +47,8 @@
 #include "utils/vk_config.h"
 #include "utils/vk_utils.h"
 
+#include <unistd.h>               // usleep
+
 #include <QKeySequence>
 #include <QString>
 #include <QStringList>
diff -ruN valkyrie-2.0.0-orig/src/utils/vk_config.cpp valkyrie-2.0.0/src/utils/vk_config.cpp
--- valkyrie-2.0.0-orig/src/utils/vk_config.cpp	2010-10-20 07:23:29.000000000 -0500
+++ valkyrie-2.0.0/src/utils/vk_config.cpp	2015-03-01 08:47:10.000000000 -0600
@@ -25,6 +25,7 @@
 #include "utils/vk_utils.h"
 
 #include <pwd.h>     // getpwuid
+#include <unistd.h>  // getuid
 
 #include <QColor>
 #include <QDir>
diff -ruN valkyrie-2.0.0-orig/src/utils/vk_utils.cpp valkyrie-2.0.0/src/utils/vk_utils.cpp
--- valkyrie-2.0.0-orig/src/utils/vk_utils.cpp	2010-10-20 07:23:29.000000000 -0500
+++ valkyrie-2.0.0/src/utils/vk_utils.cpp	2015-03-01 08:48:53.000000000 -0600
@@ -22,6 +22,7 @@
 #include "utils/vk_config.h"        // vkname()
 
 #include <cstdlib>                  // exit, mkstemp, free/malloc, etc
+#include <unistd.h>                 // getpid
 
 #include <QDateTime>
 #include <QFile>
