diff -ruN imake-1.0.7-orig/imakemdep.h imake-1.0.7/imakemdep.h
--- imake-1.0.7-orig/imakemdep.h	2013-08-17 05:11:06.000000000 -0500
+++ imake-1.0.7/imakemdep.h	2018-09-27 22:38:22.000000000 -0500
@@ -259,7 +259,7 @@
 #  if !defined (CROSSCOMPILE) || defined (CROSSCOMPILE_CPP)
 
 #   if defined(__APPLE__)
-#    define DEFAULT_CPP "/usr/bin/cpp"
+#    define DEFAULT_CPP "@FINK_CPP_CMD@"
 #    define DEFAULT_CC "cc"
 #   endif
 #   if defined(Lynx) || defined(__Lynx__)
@@ -324,7 +324,7 @@
 #     define DEFAULT_CPP "/usr/X11R6/bin/cpp"
 #    endif
 #   endif
-#   if defined(__GNUC__) && !defined(USE_CC_E)
+#   if defined(__GNUC__) && !defined(__clang__) && !defined(USE_CC_E)
 #    define USE_CC_E
 #    ifndef DEFAULT_CC
 #     define DEFAULT_CC "gcc"
