diff -Nurd Cairo-1.109.orig/t/CairoFt.t Cairo-1.109/t/CairoFt.t
--- Cairo-1.109.orig/t/CairoFt.t	2011-11-27 02:56:45.000000000 -0500
+++ Cairo-1.109/t/CairoFt.t	2025-05-29 18:40:47.000000000 -0400
@@ -19,8 +19,8 @@
 }
 
 my @files_to_try = qw(
-  /usr/share/fonts/truetype/ttf-bitstream-vera/Vera.ttf
-  /usr/share/fonts/truetype/ttf-dejavu/DejaVuSerif.ttf
+  /opt/X11/share/fonts/TTF/Vera.ttf
+  /usr/X11/share/fonts/TTF/Vera.ttf
 );
 my @files_found = grep { -r $_ } @files_to_try;
 my $file = $files_found[0];
