diff -Nurd -x'*~' CGI-Compress-Gzip-1.03.orig/t/gzip.t CGI-Compress-Gzip-1.03/t/gzip.t
--- CGI-Compress-Gzip-1.03.orig/t/gzip.t	2008-10-18 00:31:54.000000000 -0400
+++ CGI-Compress-Gzip-1.03/t/gzip.t	2010-07-30 02:58:47.000000000 -0400
@@ -12,6 +12,7 @@
 use IO::Zlib;
 use Compress::Zlib;
 use English qw(-no_match_vars);
+use Config;
 
 BEGIN
 {
@@ -269,6 +270,18 @@
 my $redir = 'http://www.foo.com/';
 
 my $interp = "$^X -Iblib/arch -Iblib/lib";
+if ($] >= 5.010 && $Config{"byteorder"} =~ /^1/) {
+    # little-endian means intel (not powerpc), for which fink has
+    # multiple single-arch distros that are seen by a single (fat)
+    # perl interp from apple for a perl version supported by fink
+    # (i.e., starting with 5.10.0). This is the logic taken from
+    # Fink::PkgVersion as of fink-0.29.13.
+    if ( $Config{"longsize"} == 4 ) {
+	$interp = "/usr/bin/arch -i386 $interp";   # 32-bit
+    } else {
+	$interp = "/usr/bin/arch -x86_64 $interp"; # 64-bit
+    }
+}
 if (defined $Devel::Cover::VERSION) {
    $interp .= ' -MDevel::Cover';
 }
