diff -Nurd CGI-Compress-Gzip-1.03.orig/t/gzip.t CGI-Compress-Gzip-1.03/t/gzip.t
--- CGI-Compress-Gzip-1.03.orig/t/gzip.t	2008-10-18 00:31:54
+++ CGI-Compress-Gzip-1.03/t/gzip.t	2025-09-30 19:22:31
@@ -268,7 +268,7 @@
 
 my $redir = 'http://www.foo.com/';
 
-my $interp = "$^X -Iblib/arch -Iblib/lib";
+my $interp = "$ENV{FULLPERL} -Iblib/arch -Iblib/lib";
 if (defined $Devel::Cover::VERSION) {
    $interp .= ' -MDevel::Cover';
 }
