diff -Nurd CPAN-Checksums-2.14.orig/t/00signature.t CPAN-Checksums-2.14/t/00signature.t
--- CPAN-Checksums-2.14.orig/t/00signature.t	2021-11-20 12:09:00.000000000 -0500
+++ CPAN-Checksums-2.14/t/00signature.t	2023-07-01 10:13:38.000000000 -0400
@@ -50,21 +50,7 @@
         }
     }
     unless ($exit_message) {
-        if (!eval {
-                use Socket qw(AF_INET SOCK_STREAM pack_sockaddr_in inet_aton);
-                my $socket;
-                socket($socket, AF_INET, SOCK_STREAM, 0) and
-                connect(
-                    $socket,
-                    pack_sockaddr_in(
-                        scalar getservbyname('hkp', 'tcp'),
-                        inet_aton('pgpkeys.eu')
-                    )
-                ) and
-                close($socket)
-            }) {
-            $exit_message = "Cannot connect to the keyserver";
-        }
+            $exit_message = "Network access prohibited in fink sandbox";
     }
     unless ($exit_message) {
         require Config;
