diff -ruN Crypt-OpenSSL-Guess-0.15-orig/lib/Crypt/OpenSSL/Guess.pm Crypt-OpenSSL-Guess-0.15/lib/Crypt/OpenSSL/Guess.pm
--- Crypt-OpenSSL-Guess-0.15-orig/lib/Crypt/OpenSSL/Guess.pm	2022-02-11 02:52:45.000000000 -0600
+++ Crypt-OpenSSL-Guess-0.15/lib/Crypt/OpenSSL/Guess.pm	2025-05-17 08:19:05.000000000 -0500
@@ -118,18 +118,8 @@
         return $ENV{OPENSSL_PREFIX};
     }
 
-    # Homebrew (macOS) or LinuxBrew
-    if ($^O ne 'MSWin32' and my $prefix = `brew --prefix --installed openssl 2>@{[File::Spec->devnull]}`) {
-        chomp $prefix;
-        return $prefix;
-    }
-
     my @guesses = (
-        '/home/linuxbrew/.linuxbrew/opt/openssl/bin/openssl' => '/home/linuxbrew/.linuxbrew/opt/openssl', # LinuxBrew openssl
-        '/opt/homebrew/opt/openssl/bin/openssl' => '/opt/homebrew/opt/openssl', # macOS ARM homebrew
-        '/usr/local/opt/openssl/bin/openssl' => '/usr/local/opt/openssl', # OSX homebrew openssl
-        '/usr/local/bin/openssl'         => '/usr/local', # OSX homebrew openssl
-        '/opt/local/bin/openssl'         => '/opt/local', # Macports openssl
+        '@FINK_PREFIX@/bin/openssl'         => '@FINK_PREFIX@', # Fink openssl
         '/usr/bin/openssl'               => '/usr',
         '/usr/sbin/openssl'              => '/usr',
         '/opt/ssl/bin/openssl'           => '/opt/ssl',
