diff -Nurd -x'*~' Crypt-OpenSSL-Bignum-0.06.orig/Makefile.PL Crypt-OpenSSL-Bignum-0.06/Makefile.PL
--- Crypt-OpenSSL-Bignum-0.06.orig/Makefile.PL	2015-02-04 17:10:31.000000000 -0500
+++ Crypt-OpenSSL-Bignum-0.06/Makefile.PL	2016-01-18 05:02:41.000000000 -0500
@@ -7,10 +7,10 @@
     'PREREQ_PM'         => {},
     'ABSTRACT_FROM'     => 'Bignum.pm', # retrieve abstract from module
     'AUTHOR'            => 'Ian Robertson <iroberts@cpan.org>',
-    'LIBS'              => ['-lcrypto'],   # e.g., '-lm'
+    'LIBS'              => ['-L@PREFIX@/lib -lcrypto'],   # e.g., '-lm'
     'DEFINE'	=> '-DPERL5 -DOPENSSL_NO_KRB5',
     # perl-5.8/gcc-3.2 needs -DPERL5, and redhat9 likes -DOPENSSL_NO_KRB5
-    'INC'               => '',
+    'INC'               => '-I@PREFIX@/include',
     'MIN_PERL_VERSION'  => '5.006002',
     'META_MERGE'        => {
       resources => {
