diff -ruN Crypt-OpenSSL-RSA-0.35-orig/Makefile.PL Crypt-OpenSSL-RSA-0.35/Makefile.PL
--- Crypt-OpenSSL-RSA-0.35-orig/Makefile.PL	2025-05-03 03:16:05.000000000 -0500
+++ Crypt-OpenSSL-RSA-0.35/Makefile.PL	2025-05-17 08:00:29.000000000 -0500
@@ -30,12 +30,12 @@
         'Test::More'             => 0,
     },
     'OBJECT' => 'RSA.o',
-    'LIBS'   => [openssl_lib_paths() . $libs],
+    'LIBS'   => ['-L@FINK_PREFIX@/lib' . $libs],
     'LDDLFLAGS' => openssl_lib_paths() . ' ' . $Config{lddlflags},
     'DEFINE' => '-DPERL5 -DOPENSSL_NO_KRB5',
 
     # perl-5.8/gcc-3.2 needs -DPERL5, and redhat9 likes -DOPENSSL_NO_KRB5
-    'INC' => openssl_inc_paths(),    # e.g., '-I/usr/include/other'
+    'INC' => '-I@FINK_PREFIX@/include',    # e.g., '-I/usr/include/other'
 
     'dist'  => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
     'clean' => { FILES    => 'Crypt-OpenSSL-RSA-*' },
