diff -Nurd -x'*~' Data-Alias-1.16.orig/Alias.xs Data-Alias-1.16/Alias.xs
--- Data-Alias-1.16.orig/Alias.xs	2011-11-17 16:57:41.000000000 -0500
+++ Data-Alias-1.16/Alias.xs	2012-07-23 05:24:21.000000000 -0400
@@ -28,9 +28,14 @@
 #if defined(USE_DTRACE) && defined(PERL_CORE)
 #undef ENTRY_PROBE
 #undef RETURN_PROBE
+#if (PERL_COMBI_VERSION < 5013008)
+#define ENTRY_PROBE(func, file, line)
+#define RETURN_PROBE(func, file, line)
+#else
 #define ENTRY_PROBE(func, file, line, stash)
 #define RETURN_PROBE(func, file, line, stash)
 #endif
+#endif
 
 #if defined(PERL_CORE) && defined(MULTIPLICITY) && \
 		(PERL_COMBI_VERSION < 5013006)
