diff -Nurd -x'*~' DBD-CSV-0.48.orig/Makefile.PL DBD-CSV-0.48/Makefile.PL
--- DBD-CSV-0.48.orig/Makefile.PL	2014-12-31 09:49:48.000000000 -0500
+++ DBD-CSV-0.48/Makefile.PL	2015-08-23 02:50:47.000000000 -0400
@@ -54,7 +54,7 @@
 use vars qw( $DBI_INC_DIR );
 
 {   my $tmp_dir = File::Spec->tmpdir ();
-    if (!$ENV{AUTOMATED_TESTING} &&
+    if (0 && !$ENV{AUTOMATED_TESTING} &&
          prompt ("Enable the use of $tmp_dir for tests?", "y") =~ m/[Yy]/) {
 	unlink "tests.skip";
 	}
