diff -ru ExtUtils-MakeMaker-6.68.orig/lib/ExtUtils/MakeMaker.pm ExtUtils-MakeMaker-6.68/lib/ExtUtils/MakeMaker.pm
--- ExtUtils-MakeMaker-6.68.orig/lib/ExtUtils/MakeMaker.pm	2013-06-14 18:26:40.000000000 -0400
+++ ExtUtils-MakeMaker-6.68/lib/ExtUtils/MakeMaker.pm	2013-07-06 22:55:48.000000000 -0400
@@ -209,8 +209,8 @@
 
 sub eval_in_subdirs {
     my($self) = @_;
-    use Cwd qw(cwd abs_path);
-    my $pwd = cwd() || die "Can't figure out your cwd!";
+    use Cwd qw(getcwd abs_path);
+    my $pwd = getcwd() || die "Can't figure out your cwd!";
 
     local @INC = map eval {abs_path($_) if -e} || $_, @INC;
     push @INC, '.';     # '.' has to always be at the end of @INC
@@ -363,6 +363,11 @@
         $Recognized_Att_Keys{$item} = 1;
     }
     foreach my $item (@Get_from_Config) {
+	if ($item eq 'lddlflags' && '@PREFIX@' ne '/usr/local') {
+	    $Config{$item} =~ s,-L/usr/local/lib,,;
+	    print "# removed -L/usr/local/lib from '\U$item\E'\n"
+		if ($Verbose >= 2);
+	}
         $Recognized_Att_Keys{uc $item} = $Config{$item};
         print "Attribute '\U$item\E' => '$Config{$item}'\n"
             if ($Verbose >= 2);
diff --git a/lib/ExtUtils/Liblist/Kid.pm b/lib/ExtUtils/Liblist/Kid.pm
index 39ae5622..c9e70782 100644
--- a/lib/ExtUtils/Liblist/Kid.pm
+++ b/lib/ExtUtils/Liblist/Kid.pm
@@ -57,6 +57,11 @@ sub _unix_os2_ext {
     my ( $pwd )   = cwd();    # from Cwd.pm
     my ( $found ) = 0;
 
+    if ( $^O eq 'darwin' or $^O eq 'next' )  {
+        # 'escape' Mach-O ld -framework flags, so they aren't dropped later on
+        $potential_libs =~ s/(^|\s)(-(?:weak_|reexport_|lazy_)?framework)\s+(\S+)/$1-Wl,$2 -Wl,$3/g;
+    }
+
     foreach my $thislib ( split ' ', $potential_libs ) {
         my ( $custom_name ) = '';
 
