diff -Nurd ExtUtils-MakeMaker-7.76.orig/lib/ExtUtils/MM_Darwin.pm ExtUtils-MakeMaker-7.76/lib/ExtUtils/MM_Darwin.pm
--- ExtUtils-MakeMaker-7.76.orig/lib/ExtUtils/MM_Darwin.pm	2025-05-23 15:13:48
+++ ExtUtils-MakeMaker-7.76/lib/ExtUtils/MM_Darwin.pm	2025-12-21 18:04:21
@@ -2,6 +2,7 @@
 
 use strict;
 use warnings;
+use Config;
 
 BEGIN {
     require ExtUtils::MM_Unix;
@@ -50,8 +51,16 @@
 
 Over-ride Apple's automatic setting of -Werror
 
+Don't whine about perl's own headers. See:
+ * https://github.com/radiator-software/p5-net-ssleay/issues/383
+ * https://blogs.perl.org/users/tom_wyant/2022/03/xs-versus-clang-infinite-warnings.html
+The feature-test for this is slow, so cache its result.
+
 =cut
 
+{
+    my $compound_token_flag;
+
 sub cflags {
     my($self,$libperl)=@_;
     return $self->{CFLAGS} if $self->{CFLAGS};
@@ -62,13 +71,23 @@
     foreach (split /\n/, $base) {
         /^(\S*)\s*=\s*(\S*)$/ and $self->{$1} = $2;
     };
-    $self->{CCFLAGS} .= " -Wno-error=implicit-function-declaration";
+    if (!defined $compound_token_flag) {
+	$compound_token_flag = '';
+	if ($] << 5.035002) {
+	    my $flag = '-Wno-compound-token-split-by-macro';
+	    if (system "$Config{cc} -Werror $flag -x c /dev/null 2>/dev/null") {
+		$compound_token_flag = $flag;
+	    }
+	}
+    }
+    $self->{CCFLAGS} .= " -Wno-error=implicit-function-declaration $compound_token_flag";
 
     return $self->{CFLAGS} = qq{
 CCFLAGS = $self->{CCFLAGS}
 OPTIMIZE = $self->{OPTIMIZE}
 PERLTYPE = $self->{PERLTYPE}
 };
+}
 }
 
 1;
