diff -Nurd ExtUtils-Install-2.22.orig/lib/ExtUtils/Install.pm ExtUtils-Install-2.22/lib/ExtUtils/Install.pm
--- ExtUtils-Install-2.22.orig/lib/ExtUtils/Install.pm	2022-12-24 15:09:24.000000000 -0500
+++ ExtUtils-Install-2.22/lib/ExtUtils/Install.pm	2023-09-15 23:10:16.000000000 -0400
@@ -750,7 +750,9 @@
             }
             # we have to do this for back compat with old File::Finds
             # and because the target is relative
-            my $save_cwd = File::Spec->catfile($cwd, $sourcedir);
+            my $save_cwd = File::Spec->file_name_is_absolute($sourcedir)
+			? $sourcedir
+			: File::Spec->catfile($cwd, $sourcedir);
             _chdir($cwd);
             my $diff = $always_copy || _compare($sourcefile, $targetfile);
             $check_dirs{$targetdir}++
