diff -Nurd ExtUtils-MakeMaker-7.76.orig/t/04-xs-rpath-darwin.t ExtUtils-MakeMaker-7.76/t/04-xs-rpath-darwin.t
--- ExtUtils-MakeMaker-7.76.orig/t/04-xs-rpath-darwin.t	2025-03-30 06:36:58
+++ ExtUtils-MakeMaker-7.76/t/04-xs-rpath-darwin.t	2025-12-21 09:50:59
@@ -74,7 +74,7 @@
 sub run_test_script {
     my ($self) = @_;
 
-    my @cmd = ($^X, '-Mblib', $self->{test_script_name});
+    my @cmd = ('/usr/bin/arch', '@ARCH@', $^X, '-Mblib', $self->{test_script_name});
     my $out = _capture_stdout(\@cmd);
     like( $out, qr{\Qcalling foo()\E\s+\QHello from foo()\E});
 }
@@ -122,7 +122,7 @@
 sub run_make {
     my ($self) = @_;
 
-    my @cmd = ($^X, '-I'. $self->{ext_utils_lib_dir}, 'Makefile.PL');
+    my @cmd = ('/usr/bin/arch', '@ARCH@', $^X, '-I'. $self->{ext_utils_lib_dir}, 'Makefile.PL');
     _run_system_cmd(\@cmd);
     _run_system_cmd(['make']);
 }
@@ -165,7 +165,7 @@
   NAME          => '<<module_name_colon>>',
   VERSION_FROM  => 'lib/<<module_name_slash>>.pm',
   ABSTRACT_FROM => 'lib/<<module_name_slash>>.pm',
-  PERL          => "$^X -w",
+  PERL          => "/usr/bin/arch @ARCH@ $^X -w",
   LIBS          => ['-L./<<lib_dir>> -l<<lib_name>>'],
   INC           => '-I. -I./<<lib_dir>>',
 );
diff -Nurd ExtUtils-MakeMaker-7.76.orig/t/lib/MakeMaker/Test/Setup/XS.pm ExtUtils-MakeMaker-7.76/t/lib/MakeMaker/Test/Setup/XS.pm
--- ExtUtils-MakeMaker-7.76.orig/t/lib/MakeMaker/Test/Setup/XS.pm	2025-03-30 06:36:58
+++ ExtUtils-MakeMaker-7.76/t/lib/MakeMaker/Test/Setup/XS.pm	2025-12-21 09:49:23
@@ -69,7 +69,7 @@
   NAME          => 'XS::%s',
   VERSION_FROM  => '%s',
   TYPEMAPS      => [ %s ],
-  PERL          => "$^X -w",
+  PERL          => "/usr/bin/arch @ARCH@ $^X -w",
   %s
 );
 END
diff -Nurd ExtUtils-MakeMaker-7.76.orig/t/lib/MakeMaker/Test/Utils.pm ExtUtils-MakeMaker-7.76/t/lib/MakeMaker/Test/Utils.pm
--- ExtUtils-MakeMaker-7.76.orig/t/lib/MakeMaker/Test/Utils.pm	2025-03-30 06:36:58
+++ ExtUtils-MakeMaker-7.76/t/lib/MakeMaker/Test/Utils.pm	2025-12-21 09:49:50
@@ -124,6 +124,7 @@
 =cut
 
 sub which_perl {
+    return "/usr/bin/arch @ARCH@ $^X";
     my $perl = $^X;
     $perl ||= 'perl';
 
