diff -Nurd File-Rename-1.13.orig/Makefile.PL File-Rename-1.13/Makefile.PL
--- File-Rename-1.13.orig/Makefile.PL	2018-06-25 18:47:16.000000000 -0400
+++ File-Rename-1.13/Makefile.PL	2021-01-18 19:51:04.000000000 -0500
@@ -3,7 +3,7 @@
 use strict;
 use File::Spec;
 my $script = File::Spec->catfile( 'script',
-		    $^O =~ /win/i 
+		    $^O =~ /Win/ 
 			? 'file-rename' : 'rename' );
 
 use ExtUtils::MakeMaker;
