diff -Nurd -x'*~' IO-Compress-2.061.orig/lib/IO/Uncompress/AnyUncompress.pm IO-Compress-2.061/lib/IO/Uncompress/AnyUncompress.pm
--- IO-Compress-2.061.orig/lib/IO/Uncompress/AnyUncompress.pm	2013-05-19 07:45:24.000000000 -0400
+++ IO-Compress-2.061/lib/IO/Uncompress/AnyUncompress.pm	2013-06-18 20:57:03.000000000 -0400
@@ -83,6 +83,21 @@
 
     my $magic ;
 
+     if (defined $IO::Uncompress::Bunzip2::VERSION and
+         $magic = $self->ckMagic('Bunzip2')) {
+        *$self->{Info} = $self->readHeader($magic)
+            or return undef ;
+
+        my ($obj, $errstr, $errno) = IO::Uncompress::Adapter::Bunzip2::mkUncompObject();
+
+        return $self->saveErrorString(undef, $errstr, $errno)
+            if ! defined $obj;
+
+        *$self->{Uncomp} = $obj;
+
+         return 1;
+     }
+
     # try zlib first
     if (defined $IO::Uncompress::RawInflate::VERSION )
     {
@@ -142,21 +157,6 @@
          return 1;
      }
 
-     if (defined $IO::Uncompress::Bunzip2::VERSION and
-         $magic = $self->ckMagic('Bunzip2')) {
-        *$self->{Info} = $self->readHeader($magic)
-            or return undef ;
-
-        my ($obj, $errstr, $errno) = IO::Uncompress::Adapter::Bunzip2::mkUncompObject();
-
-        return $self->saveErrorString(undef, $errstr, $errno)
-            if ! defined $obj;
-
-        *$self->{Uncomp} = $obj;
-
-         return 1;
-     }
-
      if (defined $IO::Uncompress::UnLzop::VERSION and
             $magic = $self->ckMagic('UnLzop')) {
 
