diff -Nurd -x'*~' Lingua-EN-Inflect-1.895.orig/Makefile.PL Lingua-EN-Inflect-1.895/Makefile.PL
--- Lingua-EN-Inflect-1.895.orig/Makefile.PL	2013-05-30 14:16:47.000000000 -0400
+++ Lingua-EN-Inflect-1.895/Makefile.PL	2013-05-30 14:16:58.000000000 -0400
@@ -10,7 +10,6 @@
     PL_FILES            => {},
     PREREQ_PM => {
         'Test::More' => 0,
-        'version'    => 0,
     },
     dist                => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
     clean               => { FILES => 'Lingua-EN-Inflect-*' },
