diff -Nurd Mac-FSEvents-0.14/MacVersion.pm.orig Mac-FSEvents-0.14/MacVersion.pm
--- Mac-FSEvents-0.14/MacVersion.pm.orig	2017-11-30 17:17:30
+++ Mac-FSEvents-0.14/MacVersion.pm	2024-11-08 19:22:04
@@ -9,7 +9,7 @@
 
 sub osx_version {
     my $os_version = qx(system_profiler SPSoftwareDataType);
-    if($os_version =~ /System Version:.+(?:(10)\.(\d+)(?:\.(\d+))?)/) {
+    if($os_version =~ /System Version:.+(?:(1[0-5])\.(\d+)(?:\.(\d+))?)/) {
         return ($1, $2, $3 || 0);
     } else {
         $os_version =~ s/^/> /gm;
diff -Nurd Mac-FSEvents-0.14/Makefile.PL.orig Mac-FSEvents-0.14/Makefile.PL
--- Mac-FSEvents-0.14/Makefile.PL.orig	2017-11-30 17:17:30
+++ Mac-FSEvents-0.14/Makefile.PL	2024-11-08 19:54:40
@@ -84,10 +84,10 @@
 
 my ( $major, $minor, $release ) = osx_version();
 
-if($minor >= 6) {
+if($major >= 11 or $minor >= 6) {
     push @names, 'kFSEventStreamCreateFlagIgnoreSelf';
 
-    if($minor >= 7) {
+    if($major >= 11 or $minor >= 7) {
         push @names, 'kFSEventStreamCreateFlagFileEvents';
     }
 }

diff -Nurd Mac-FSEvents-0.14/hints/darwin.pl.orig Mac-FSEvents-0.14/hints/darwin.pl
--- Mac-FSEvents-0.14/hints/darwin.pl.orig	2017-11-30 17:17:30
+++ Mac-FSEvents-0.14/hints/darwin.pl	2024-11-08 19:17:49
@@ -7,7 +7,7 @@
     my ( $major, $minor, $release ) = osx_version();
     my $os_version = join('.', $major, $minor);
 
-    if($minor >= 5) { # Leopard and up
+    if($major > 10 or $minor >= 5) { # Leopard and up
         my @directories = (
             "/Developer/SDKs/MacOSX$os_version.sdk",
             "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX$os_version.sdk",
@@ -27,7 +27,11 @@
             die "No SDK found for your version of OS X.  Please install Xcode.\n";
         }
 
-        $arch = "-arch x86_64 -arch i386 -isysroot $sysroot -mmacosx-version-min=$os_version";
+        if($major > 10) { # Big Sur and up
+                $arch = "-arch x86_64 -arch arm64 -isysroot $sysroot -mmacosx-version-min=$os_version";
+        } else {
+                $arch = "-arch x86_64 -arch i386 -isysroot $sysroot -mmacosx-version-min=$os_version";
+        }
     } else {
         $arch = "-arch i386 -arch ppc";
     }
