diff -Nurd Module-Release-2.136.orig/lib/Module/Release.pm Module-Release-2.136/lib/Module/Release.pm
--- Module-Release-2.136.orig/lib/Module/Release.pm	2025-01-03 01:51:24
+++ Module-Release-2.136/lib/Module/Release.pm	2025-12-23 01:14:21
@@ -529,11 +529,12 @@
 	unless( -x $path ) {
 		if( $path =~ m/[*?[]/ && $self->config->allow_glob_in_perls ) {
 			$self->add_a_perl( $_ ) for glob $path;
+			return;
 			}
-		else {
-			$self->_warn( "$path is not executable" );
+		if( system( "$path -e 'exit 0' 2>/dev/null" ) ) {
+			$self->_warn( "$path is not executable ($?)" );
+			return;
 			}
-		return;
 		}
 
 	my $version = $self->_looks_like_perl( $path );
