diff -Nurd -x'*~' Module-Install-1.06.orig/t/lib/Test.pm Module-Install-1.06/t/lib/Test.pm
--- Module-Install-1.06.orig/t/lib/Test.pm	2012-03-01 08:00:48.000000000 -0500
+++ Module-Install-1.06/t/lib/Test.pm	2013-05-30 05:03:10.000000000 -0400
@@ -165,7 +165,7 @@
 	local $ENV{X_MYMETA} = $X_MYMETA;
 
 	my @run_params=@{ $params{run_params} || [] };
-	my $ret = system($^X, "-I../../lib", "-I../../blib/lib", "Makefile.PL",@run_params);
+	my $ret = system(split(/\s+/,$ENV{FULLPERL}), "-I../../lib", "-I../../blib/lib", "Makefile.PL",@run_params);
 	chdir $home or return 0;
 	return $ret ? 0 : 1;
 }
@@ -180,7 +180,7 @@
 	local $ENV{X_MYMETA} = $X_MYMETA;
 
 	my $run_params=join(' ',@{ $params{run_params} || [] });
-	my $ret = system("$^X -I../../lib -I../../blib/lib Makefile.PL $run_params");
+	my $ret = system("$ENV{FULLPERL} -I../../lib -I../../blib/lib Makefile.PL $run_params");
 	#my $result=qx();
 	chdir $home or return 0;
 	return $ret ? 0 : 1;
@@ -215,7 +215,7 @@
 	local $ENV{X_MYMETA} = $X_MYMETA;
 
 	my @run_params=@{ $params{run_params} || [] };
-	my $command = join ' ', $^X, "-I../../lib", "-I../../blib/lib", "Makefile.PL", @run_params;
+	my $command = join ' ', split(/\s+/,$ENV{FULLPERL}), "-I../../lib", "-I../../blib/lib", "Makefile.PL", @run_params;
 	my $ret = `$command 2>&1`;
 	chdir $home;
 	return $ret;
