diff -Nurd Net-SSLeay-1.90.orig/Makefile.PL Net-SSLeay-1.90/Makefile.PL
--- Net-SSLeay-1.90.orig/Makefile.PL	2021-01-21 10:14:11.000000000 -0500
+++ Net-SSLeay-1.90/Makefile.PL	2021-11-25 00:53:49.000000000 -0500
@@ -154,7 +154,7 @@
     for ("$prefix/include", "$prefix/inc32", '/usr/kerberos/include') {
       push @{$opts->{inc_paths}}, $_ if -f "$_/openssl/ssl.h";
     }
-    for ($prefix, "$prefix/lib64", "$prefix/lib", "$prefix/out32dll") {
+    for ("$prefix/lib") {
       push @{$opts->{lib_paths}}, $_ if -d $_;
     }
 
