diff -Nurd -x'*~' Poppler-0.04.orig/Makefile.PL Poppler-0.04/Makefile.PL
--- Poppler-0.04.orig/Makefile.PL	2011-01-13 05:30:31.000000000 -0500
+++ Poppler-0.04/Makefile.PL	2013-03-19 08:04:56.000000000 -0400
@@ -10,7 +10,7 @@
 
 my %poppler_cfg;
 print "Checking poppler lib\n";
-eval { %poppler_cfg = ExtUtils::PkgConfig->find ("poppler >= $extreqs{poppler}"); };
+eval { %poppler_cfg = ExtUtils::PkgConfig->find ("poppler-glib >= $extreqs{poppler}"); };
 if ($@ ne '') {
     warn $@; 
     WriteMakefile(
@@ -30,8 +30,8 @@
       (ABSTRACT_FROM  => 'lib/Poppler.pm', # retrieve abstract from module
        AUTHOR         => 'Cornelius <cornelius.howl@gmail.com>') : ()),
     DEFINE            => '', # e.g., '-DHAVE_SOMETHING'
-    INC     => `pkg-config --cflags poppler-glib`,
-    LIBS    =>  `pkg-config --libs poppler-glib`,
+    INC     => $poppler_cfg{cflags},
+    LIBS    => $poppler_cfg{libs},
     XS      => {
         'Poppler.xs' => 'Poppler.c',
         # 'Goo.xs' => 'Goo.c',
