diff -Nurd -x'*~' Sort-Packed-0.06.orig/t/pods.t Sort-Packed-0.06/t/pods.t
--- Sort-Packed-0.06.orig/t/pods.t	2009-11-17 07:48:26.000000000 -0500
+++ Sort-Packed-0.06/t/pods.t	2010-08-14 11:44:06.000000000 -0400
@@ -3,8 +3,7 @@
 use strict;
 use Test::More;
 
-plan skip_all => "Only the author needs to check that POD docs are right"
-    unless eval "no warnings; getlogin eq 'salva'";
+plan skip_all => "Only the author needs to check that POD docs are right";
 
 eval "use Test::Pod 1.00";
 plan skip_all => "Test::Pod 1.00 required for testing POD" if $@;
diff -ruN Sort-Packed-0.08-orig/ppport.h Sort-Packed-0.08/ppport.h
--- Sort-Packed-0.08-orig/ppport.h	2009-11-17 06:48:26
+++ Sort-Packed-0.08/ppport.h	2024-11-12 19:23:11
@@ -3683,17 +3683,9 @@
 
 #undef STMT_START
 #undef STMT_END
-#ifdef PERL_USE_GCC_BRACE_GROUPS
-#  define STMT_START	(void)(	/* gcc supports ``({ STATEMENTS; })'' */
-#  define STMT_END	)
-#else
-#  if defined(VOIDFLAGS) && (VOIDFLAGS) && (defined(sun) || defined(__sun__)) && !defined(__GNUC__)
-#    define STMT_START	if (1)
-#    define STMT_END	else (void)0
-#  else
-#    define STMT_START	do
-#    define STMT_END	while (0)
-#  endif
+#if !(defined(STMT_START) && defined(STMT_END))
+#  define STMT_START	do
+#  define STMT_END	while (0)
 #endif
 #ifndef boolSV
 #  define boolSV(b)                      ((b) ? &PL_sv_yes : &PL_sv_no)
