diff -Nurd -x'*~' Term-ReadLine-Gnu-1.35.orig/Gnu.pm Term-ReadLine-Gnu-1.35/Gnu.pm
--- Term-ReadLine-Gnu-1.35.orig/Gnu.pm	2018-06-11 12:59:32.000000000 -0400
+++ Term-ReadLine-Gnu-1.35/Gnu.pm	2018-06-11 13:17:03.000000000 -0400
@@ -80,6 +80,13 @@
 END
     }
 }
+# use Term::ReadLine::Stub on a dumb terminal.
+# https://rt.cpan.org/Ticket/Display.html?id=123398
+BEGIN {
+    if (!defined($ENV{TERM}) || $ENV{TERM} =~ /^(dumb|emacs|unknown)$/) {
+	croak "dumb terminal.";
+    }
+}
 
 {
     use Exporter ();
@@ -245,6 +252,7 @@
     my $this = shift;		# Package
     my $class = ref($this) || $this;
 
+    croak "Wrong number of arguments" unless @_==1 or @_==3;
     my $name = shift;
 
     my $self = \%Attribs;
