diff -Nurd -x'*~' Test-AutoLoader-0.03.orig/t/00basic.t Test-AutoLoader-0.03/t/00basic.t
--- Test-AutoLoader-0.03.orig/t/00basic.t	2005-07-22 01:57:25.000000000 -0400
+++ Test-AutoLoader-0.03/t/00basic.t	2012-07-25 01:52:44.000000000 -0400
@@ -7,7 +7,7 @@
     eval "use Test::Pod 0.95";
     $NO_TEST_POD = $@;
 }
-use Test::More tests=>54;
+use Test::More tests=>64;
 use POSIX;
 use strict;
 
@@ -35,6 +35,7 @@
 my @unreadable = (File::Spec->catdir(qw(tlib auto TestBusted1)),
                   File::Spec->catfile(qw(tlib auto TestBusted2 no_ready.al)));
 my $CAN_CHMOD =  chmod 0000, @unreadable;
+my $POSIX_USES_AUTOLOAD = $^V lt v5.16.0;
 my $file_errors;
 $file_errors .=<<DIAG if $CAN_CHMOD;
     couldn't load no_ready.al: Permission denied
@@ -46,8 +47,6 @@
 
 
 my @tests = (
-  [ ['POSIX'],{ok=>1,name=>"Autoload of POSIX (all files)"},"Standard-distribution module, all files"],
-  [ [qw(POSIX strcpy)],{ok=>1,name=>"Autoload of POSIX (listed subroutines)"},"Standard-distribution module, one file"],
   [ [qw(POSIX no_such_function)], {ok=>0,diag=>"    couldn't load no_such_function.al: No such file or directory"}, "Standard-distribution, bad subroutine name"],
   [ [qw(strict)], {ok=>0,diag=>"Unable to find valid autoload directory for strict"}, "Non-existent auto directory"],
   [ [qw(EmptyModule)], {ok=>0,diag=>"No autoloaded files found"}, "No files in auto directory"],
@@ -59,6 +58,14 @@
 
 );
 
+if ($POSIX_USES_AUTOLOAD) {
+    push @tests,
+      [ ['POSIX'],{ok=>1,name=>"Autoload of POSIX (all files)"},"Standard-distribution module, all files"],
+      [ [qw(POSIX strcpy)],{ok=>1,name=>"Autoload of POSIX (listed subroutines)"},"Standard-distribution module, one file"];
+} else {
+  SKIP:{skip "POSIX module does not use AutoLoader",12}
+}
+
 if ($CAN_CHMOD) {
     push @tests, 
       [ [qw(TestBusted1)], {ok=>0,diag=>"Unable to find valid autoload directory for TestBusted1"}, "Unreadable auto directory"],
@@ -71,5 +78,5 @@
 if ($CAN_CHMOD) {
     chmod 0755, @unreadable or warn "Couldn't chmod @unreadable back: $!\n";
 } else {
-  SKIP:{skip "Couldn't set up unreadable directory for test",5}
+  SKIP:{skip "Couldn't set up unreadable directory for test",6}
 }
