diff -ruN Tk-TableMatrix-1.25-orig/Makefile.PL Tk-TableMatrix-1.25/Makefile.PL
--- Tk-TableMatrix-1.25-orig/Makefile.PL	2020-10-26 20:29:13.000000000 -0500
+++ Tk-TableMatrix-1.25/Makefile.PL	2020-10-26 20:28:25.000000000 -0500
@@ -25,7 +25,10 @@
 use Tk::MMutil;
 
 # Get existing tk version for figuring
-$tkversion = eval 'use Tk; $Tk::VERSION;';
+{
+    require Tk;
+    $tkversion = $Tk::VERSION;
+}
 #print "################tkversion = $tkversion\n";
 
 my $mydefine ='';
diff -ruN Tk-TableMatrix-1.25-orig/myConfig Tk-TableMatrix-1.25/myConfig
--- Tk-TableMatrix-1.25-orig/myConfig	2020-10-26 20:29:09.000000000 -0500
+++ Tk-TableMatrix-1.25/myConfig	2020-10-26 20:28:25.000000000 -0500
@@ -286,7 +286,9 @@
     #
     unless (defined $xlib)
     {
-        $xlib = &lX11(0,chooseX11(</usr/X11*/lib>),chooseX11(</usr/lib/X11*>),</usr/Xfree*/lib>,'/usr/X386/lib')
+        $xlib = &lX11(0,chooseX11(</usr/X11*/lib>),chooseX11(</usr/lib/X11*>),</usr/Xfree*/lib>,'/usr/X386/lib',
+                    '/opt/X11/lib')
+
     }
 
     #
