diff -Nurd -x'*~' Tree-XPathEngine-0.05.orig/lib/Tree/XPathEngine.pm Tree-XPathEngine-0.05/lib/Tree/XPathEngine.pm
--- Tree-XPathEngine-0.05.orig/lib/Tree/XPathEngine.pm	2006-02-15 10:42:44.000000000 -0500
+++ Tree-XPathEngine-0.05/lib/Tree/XPathEngine.pm	2016-12-06 03:11:44.000000000 -0500
@@ -986,7 +986,7 @@
 
 The tree should share some characteristics with a XML tree: it is made of nodes,
 there are 2 kinds of nodes, document (the whole tree, the root of the tree is 
-a child of this node), elements(regular nodes in the tree) and attributes. 
+a child of this node), elements (regular nodes in the tree) and attributes. 
 
 Nodes in the tree are expected to provide methods that will be called by the
 XPath engine to resolve the query. Not all of the possible methods need be 
@@ -1307,7 +1307,7 @@
 
 L<Tree::DAG_Node::XPath> for an exemple of using this module
 
-L<http://www.xmltwig.com/article/extending_xml_xpath/ > for background information
+L<http://www.xmltwig.com/article/extending_xml_xpath/> for background information
 
 L<Class::XPath>, which is probably easier to use, but at this point supports much
 less of XPath that Tree::XPathEngine.
