--- a/Makefile.PL.orig	2017-07-03 05:27:58.000000000 -0700
+++ b/Makefile.PL	2017-07-03 05:28:33.000000000 -0700
@@ -1,6 +1,7 @@
 # Makefile.PL for Perl module WWW::Curl
 # Check out the README file for more information.
 
+use lib '.';
 use inc::Module::Install;
 
 name			'WWW-Curl';
--- a/Curl.xs
+++ b/Curl.xs
@@ -18,6 +18,10 @@
 #include <curl/easy.h>
 #include <curl/multi.h>
 
+#ifdef CURLINC_MULTI_H
+#define __CURL_MULTI_H
+#endif
+
 #define header_callback_func writeheader_callback_func
 
 /* Do a favor for older perl versions */
--- a/Makefile.PL.orig	2016-09-24 11:42:08.000000000 -0700
+++ b/Makefile.PL	2016-09-24 11:54:24.000000000 -0700
@@ -127,7 +127,7 @@
     close H;
 
     for my $e (sort @syms) {
-       if($e =~ /(OBSOLETE|^CURL_EXTERN|_LAST\z|_LASTENTRY\z)/) {
+       if($e =~ /(OBSOLETE|^CURL_EXTERN|^CURL_WIN32\z|^CURLOPT\z|^CURL_STRICTER\z|^CURL_DID_MEMORY_FUNC_TYPEDEFS\z|_LAST\z|_LASTENTRY\z|^CURLINC_)/) {
           next;
        }
        my ($group) = $e =~ m/^([^_]+_)/;
