diff -Nurd -x'*~' X11-WindowHierarchy-0.004.orig/Makefile.PL X11-WindowHierarchy-0.004/Makefile.PL
--- X11-WindowHierarchy-0.004.orig/Makefile.PL	2012-08-04 17:34:46.000000000 -0400
+++ X11-WindowHierarchy-0.004/Makefile.PL	2012-08-08 06:04:51.000000000 -0400
@@ -12,7 +12,7 @@
   "ABSTRACT" => "Retrieve information from X11 windows",
   "AUTHOR" => "Tom Molesworth <cpan\@entitymodel.com>",
   "BUILD_REQUIRES" => {
-    "Test::More" => "0.98"
+    "Test::More" => 0
   },
   "CONFIGURE_REQUIRES" => {
     "ExtUtils::MakeMaker" => "6.30"
