diff -ruN lilypond-2.7.26-orig/Documentation/user/lilypond.tely lilypond-2.7.26/Documentation/user/lilypond.tely
--- lilypond-2.7.26-orig/Documentation/user/lilypond.tely	2006-01-02 00:22:56.000000000 -0800
+++ lilypond-2.7.26/Documentation/user/lilypond.tely	2006-01-02 00:23:15.000000000 -0800
@@ -33,13 +33,13 @@
 @c    LilyPond  program.
   
 @end ignore
-* LilyPond: (lilypond/lilypond).           The GNU music typesetter.
-* abc2ly: (lilypond/lilypond)Invoking abc2ly.          Importing ABC.          
-* convert-ly: (lilypond/lilypond)Invoking convert-ly.  Older LilyPond versions.
-* etf2ly: (lilypond/lilypond)Invoking etf2ly.          Importing Finale.
-* lilypond-book: (lilypond/lilypond)LilyPond-book.     Itegrating text and music.
-* midi2ly: (lilypond/lilypond)Invoking midi2ly.        Importing MIDI.
-* mup2ly: (lilypond/lilypond)Invoking mup2ly.          Importing Mup. 
+* LilyPond: (lilypond).           The GNU music typesetter.
+* abc2ly: (lilypond)Invoking abc2ly.          Importing ABC.          
+* convert-ly: (lilypond)Invoking convert-ly.  Older LilyPond versions.
+* etf2ly: (lilypond)Invoking etf2ly.          Importing Finale.
+* lilypond-book: (lilypond)LilyPond-book.     Itegrating text and music.
+* midi2ly: (lilypond)Invoking midi2ly.        Importing MIDI.
+* mup2ly: (lilypond)Invoking mup2ly.          Importing Mup. 
 @end direntry
 
 
diff -ruN lilypond-2.7.26-orig/Documentation/user/music-glossary.tely lilypond-2.7.26/Documentation/user/music-glossary.tely
--- lilypond-2.7.26-orig/Documentation/user/music-glossary.tely	2006-01-02 00:22:56.000000000 -0800
+++ lilypond-2.7.26/Documentation/user/music-glossary.tely	2006-01-02 00:23:15.000000000 -0800
@@ -15,6 +15,11 @@
 and as @uref{source/Documentation/user/music-glossary.html,one big page}.
 @end ifhtml
 
+@dircategory GNU music project
+@direntry
+* Glossary: (music-glossary).                  Glossary of music terms.
+@end direntry
+
 @titlepage
 @title Music Glossary
 @author Christian Mondrup   @c Original author of LilyPond glossary
diff -ruN lilypond-2.7.26-orig/config.make.in lilypond-2.7.26/config.make.in
--- lilypond-2.7.26-orig/config.make.in	2006-01-02 00:22:59.000000000 -0800
+++ lilypond-2.7.26/config.make.in	2006-01-02 00:23:59.000000000 -0800
@@ -24,7 +24,7 @@
 FREETYPE2_LIBS = @FREETYPE2_LIBS@
 PANGO_FT2_LIBS = @PANGO_FT2_LIBS@
 
-CONFIG_LIBS = @LIBS@ @EXTRA_LIBS@ $(GUILE_LIBS) $(PANGO_FT2_LIBS) $(FONTCONFIG_LIBS) $(FREETYPE2_LIBS)
+CONFIG_LIBS = @LIBS@ @EXTRA_LIBS@ $(PANGO_FT2_LIBS) $(FONTCONFIG_LIBS) $(FREETYPE2_LIBS) $(GUILE_LIBS)
 CONFIG_LDFLAGS = @LDFLAGS@
 
 PACKAGE = @PACKAGE@
@@ -55,7 +55,7 @@
 stepmake = @stepmake@
 
 package_datadir = $(datadir)/$(package)
-package_infodir = $(infodir)/$(package)
+package_infodir = $(infodir)
 package_libdir = $(libdir)/$(package)
 package_docdir = $(docdir)/$(package)
 package_omfdir = $(omfdir)/$(package)
diff -ruN lilypond-2.7.26-orig/stepmake/stepmake/c++-rules.make lilypond-2.7.26/stepmake/stepmake/c++-rules.make
--- lilypond-2.7.26-orig/stepmake/stepmake/c++-rules.make	2006-01-02 00:22:57.000000000 -0800
+++ lilypond-2.7.26/stepmake/stepmake/c++-rules.make	2006-01-02 00:23:15.000000000 -0800
@@ -26,6 +26,7 @@
 
 $(outdir)/%.cc: %.ll
 	$(FLEX) -Cfe -p -p -o$@ $< 
+	perl -i.bak -pe 's/^class istream;/#include <iostream>;\nusing namespace std;/' $@
 
 $(outdir)/%-rc.o: $(outdir)/%.rc
 	$(WINDRES) $(WINDRES_FLAGS) -o$@ $<
diff -ruN lilypond-2.7.26-orig/stepmake/stepmake/compile-vars.make lilypond-2.7.26/stepmake/stepmake/compile-vars.make
--- lilypond-2.7.26-orig/stepmake/stepmake/compile-vars.make	2006-01-02 00:22:57.000000000 -0800
+++ lilypond-2.7.26/stepmake/stepmake/compile-vars.make	2006-01-02 00:23:15.000000000 -0800
@@ -4,7 +4,7 @@
 
 PIC_FLAGS = -fpic -fPIC
 ifeq ($(MINGW_BUILD),)
-SHARED_FLAGS = -shared
+SHARED_FLAGS = -bundle -flat_namespace -undefined suppress
 else
 SHARED_FLAGS = -mdll
 endif
