diff -ruN lilypond-2.5.28-orig/Documentation/user/GNUmakefile lilypond-2.5.28/Documentation/user/GNUmakefile
--- lilypond-2.5.28-orig/Documentation/user/GNUmakefile	2005-06-07 00:20:25.000000000 -0700
+++ lilypond-2.5.28/Documentation/user/GNUmakefile	2005-06-07 00:20:57.000000000 -0700
@@ -277,7 +277,7 @@
 local-clean: local-delete
 
 local-delete:
-	-for i in $$(seq 0 9); do \
+	-for i in $$(jot 9 0 9); do \
 		for j in dvi eps eps.bbox eps.trans eps.trans.ps ps log png tex; do \
 			rm -f $(outdir)/lily-$$i*$$j; \
 		done; \
diff -ruN lilypond-2.5.28-orig/Documentation/user/lilypond.tely lilypond-2.5.28/Documentation/user/lilypond.tely
--- lilypond-2.5.28-orig/Documentation/user/lilypond.tely	2005-06-07 00:20:25.000000000 -0700
+++ lilypond-2.5.28/Documentation/user/lilypond.tely	2005-06-07 00:20:57.000000000 -0700
@@ -7,7 +7,6 @@
 
 @dircategory GNU music project
 @direntry
-
 @ignore
 (I think)
 Current version of the manual: 2.5.18
@@ -32,15 +31,14 @@
 @c    LilyPond  program.
   
 @end ignore
-
-* LilyPond: (lilypond/lilypond).           The GNU music typesetter.
-* abc2ly: (lilypond/lilypond)Invoking abc2ly.          Importing ABC.          
-* convert-ly: (lilypond/lilypond)Invoking convert-ly.  Older LilyPond versions.
-* etf2ly: (lilypond/lilypond)Invoking etf2ly.          Importing Finale.
-* lilypond-book: (lilypond/lilypond)LilyPond-book.     Itegrating text and music.
-* midi2ly: (lilypond/lilypond)Invoking midi2ly.        Importing MIDI.
-* mup2ly: (lilypond/lilypond)Invoking mup2ly.          Importing Mup. 
-* musedata2ly: (lilypond/lilypond)Invoking musedata2ly.        Importing MuseData.
+* LilyPond: (lilypond).           The GNU music typesetter.
+* abc2ly: (lilypond)Invoking abc2ly.          Importing ABC.          
+* convert-ly: (lilypond)Invoking convert-ly.  Older LilyPond versions.
+* etf2ly: (lilypond)Invoking etf2ly.          Importing Finale.
+* lilypond-book: (lilypond)LilyPond-book.     Itegrating text and music.
+* midi2ly: (lilypond)Invoking midi2ly.        Importing MIDI.
+* mup2ly: (lilypond)Invoking mup2ly.          Importing Mup. 
+* musedata2ly: (lilypond)Invoking musedata2ly.        Importing MuseData.
 @end direntry
 
 
diff -ruN lilypond-2.5.28-orig/Documentation/user/music-glossary.tely lilypond-2.5.28/Documentation/user/music-glossary.tely
--- lilypond-2.5.28-orig/Documentation/user/music-glossary.tely	2005-06-07 00:20:25.000000000 -0700
+++ lilypond-2.5.28/Documentation/user/music-glossary.tely	2005-06-07 00:20:57.000000000 -0700
@@ -16,6 +16,11 @@
 and as @uref{source/Documentation/user/out-www/music-glossary.html,one big page}.
 @end ifhtml
 
+@dircategory GNU music project
+@direntry
+* Glossary: (music-glossary).                  Glossary of music terms.
+@end direntry
+
 @titlepage
 @title Music Glossary
 @author Christian Mondrup   @c Original author of LilyPond glossary
diff -ruN lilypond-2.5.28-orig/config.make.in lilypond-2.5.28/config.make.in
--- lilypond-2.5.28-orig/config.make.in	2005-06-07 00:20:29.000000000 -0700
+++ lilypond-2.5.28/config.make.in	2005-06-07 00:20:57.000000000 -0700
@@ -41,7 +41,7 @@
 elispdir = $(datadir)/emacs/site-lisp
 
 package_datadir = $(datadir)/$(package)
-package_infodir = $(infodir)/$(package)
+package_infodir = $(infodir)
 package_libdir = $(libdir)/$(package)
 package_docdir = $(docdir)/$(package)
 package_omfdir = $(omfdir)/$(package)
diff -ruN lilypond-2.5.28-orig/stepmake/stepmake/c++-rules.make lilypond-2.5.28/stepmake/stepmake/c++-rules.make
--- lilypond-2.5.28-orig/stepmake/stepmake/c++-rules.make	2005-06-07 00:20:28.000000000 -0700
+++ lilypond-2.5.28/stepmake/stepmake/c++-rules.make	2005-06-07 00:20:57.000000000 -0700
@@ -26,6 +26,7 @@
 
 $(outdir)/%.cc: %.ll
 	$(FLEX) -Cfe -p -p -o$@ $< 
+	perl -i.bak -pe 's/^class istream;/#include <iostream>;\nusing namespace std;/' $@
 
 $(outdir)/%-rc.o: $(outdir)/%.rc
 	$(WINDRES) $(WINDRES_FLAGS) -o$@ $<
diff -ruN lilypond-2.5.28-orig/stepmake/stepmake/c-vars.make lilypond-2.5.28/stepmake/stepmake/c-vars.make
--- lilypond-2.5.28-orig/stepmake/stepmake/c-vars.make	2005-06-07 00:20:28.000000000 -0700
+++ lilypond-2.5.28/stepmake/stepmake/c-vars.make	2005-06-07 00:20:57.000000000 -0700
@@ -14,4 +14,4 @@
 
 ALL_C_SOURCES += $(H_FILES) $(C_FILES) $(Y_FILES) $(L_FILES)
 
-ALL_CFLAGS = $(CFLAGS) $(ICFLAGS) $(DEFINES) $(addprefix -I,$(INCLUDES)) $(USER_CFLAGS) $(EXTRA_CFLAGS) $(MODULE_CFLAGS)
+ALL_CFLAGS = $(CFLAGS) $(CPPFLAGS) $(ICFLAGS) $(DEFINES) $(addprefix -I,$(INCLUDES)) $(USER_CFLAGS) $(EXTRA_CFLAGS) $(MODULE_CFLAGS)
diff -ruN lilypond-2.5.28-orig/stepmake/stepmake/compile-vars.make lilypond-2.5.28/stepmake/stepmake/compile-vars.make
--- lilypond-2.5.28-orig/stepmake/stepmake/compile-vars.make	2005-06-07 00:20:28.000000000 -0700
+++ lilypond-2.5.28/stepmake/stepmake/compile-vars.make	2005-06-07 00:20:57.000000000 -0700
@@ -3,7 +3,7 @@
 ALL_LDFLAGS = $(LDFLAGS) $(ILDFLAGS) $(EXTRA_LDFLAGS) $($(PACKAGE)_LDFLAGS) $(MODULE_LDFLAGS) $(USER_LDFLAGS)
 
 PIC_FLAGS = -fpic -fPIC
-SHARED_FLAGS = -shared
+SHARED_FLAGS = -bundle -flat_namespace -undefined suppress
 
 o-dep-out = $(outdir)/$(subst .o,.dep,$(notdir $@))#
 DO_O_DEP = rm -f $(o-dep-out); DEPENDENCIES_OUTPUT="$(o-dep-out) $(outdir)/$(notdir $@)"
