--- scribus-1.1.4/scribus/scpainter.cpp	Tue Dec 16 00:09:54 2003
+++ scribus-1.1.4.corr/scribus/scpainter.cpp	Sat Dec 20 16:50:15 2003
@@ -948,7 +948,11 @@
 	ArtPathStrokeJoinType joinStyle = ART_PATH_STROKE_JOIN_MITER;
 	QColor color = m_stroke;
 	as = qRound( 255 * stroke_trans );
+#ifdef WORDS_BIGENDIAN
+        strokeColor = ( color.red() << 24 ) | ( color.green() << 16 ) | ( color.blue() << 8 );
+#else
 	strokeColor = ( 0 << 24 ) | ( color.blue() << 16 ) | ( color.green() << 8 ) | color.red();
+#endif
 	double ratio = m_zoomFactor;
 	ArtSvpWriter *swr;
 	ArtSVP *temp;
