--- lifelines-3.0.17/configure.orig	2002-08-31 23:26:27.000000000 -0700
+++ lifelines-3.0.17-patched/configure	2002-08-31 23:27:07.000000000 -0700
@@ -1424,7 +1424,7 @@
 
 
 # expand $ac_aux_dir to an absolute path
-am_aux_dir=`CDPATH=:; cd $ac_aux_dir && pwd`
+am_aux_dir=`unset CDPATH=; cd $ac_aux_dir && pwd`
 
 test x"${MISSING+set}" = xset || MISSING="\${SHELL} $am_aux_dir/missing"
 # Use eval to expand $SHELL
