diff -Nurd -x'*~' libcanberra-0.23.orig/Makefile.am libcanberra-0.23/Makefile.am
--- libcanberra-0.23.orig/Makefile.am	2009-09-11 17:59:30.000000000 -0400
+++ libcanberra-0.23/Makefile.am	2010-02-23 00:50:16.000000000 -0500
@@ -26,7 +26,7 @@
 	vala/libcanberra.vapi \
 	vala/libcanberra-gtk.vapi
 
-SUBDIRS = src gtkdoc doc
+SUBDIRS = src doc
 
 MAINTAINERCLEANFILES = README
 noinst_DATA = README
diff -Nurd -x'*~' libcanberra-0.23.orig/configure.ac libcanberra-0.23/configure.ac
--- libcanberra-0.23.orig/configure.ac	2010-02-20 11:38:00.000000000 -0500
+++ libcanberra-0.23/configure.ac	2010-02-23 00:50:30.000000000 -0500
@@ -375,6 +375,7 @@
     HAVE_GTK=0
 fi
 
+PKG_CHECK_MODULES(LIBCANBERRA, [ libcanberra >= 0.15 ])
 
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
diff -Nurd -x'*~' libcanberra-0.23.orig/src/Makefile.am libcanberra-0.23/src/Makefile.am
--- libcanberra-0.23.orig/src/Makefile.am	2009-10-06 18:38:46.000000000 -0400
+++ libcanberra-0.23/src/Makefile.am	2010-02-23 01:35:06.000000000 -0500
@@ -34,14 +34,10 @@
 EXTRA_DIST = \
 	map-file
 
-lib_LTLIBRARIES = \
-        libcanberra.la
+lib_LTLIBRARIES =
 
-include_HEADERS = \
-	canberra.h
+include_HEADERS =
 
-noinst_PROGRAMS = \
-	test-canberra
 
 libcanberra_la_SOURCES = \
 	canberra.h \
@@ -290,20 +286,22 @@
 	canberra-gtk.h \
 	canberra-gtk.c
 libcanberra_gtk_la_CFLAGS = \
-	$(GTK_CFLAGS)
+	$(GTK_CFLAGS) \
+	$(LIBCANBERRA_CFLAGS)
 libcanberra_gtk_la_LIBADD = \
 	$(GTK_LIBS) \
-	libcanberra.la
+	$(LIBCANBERRA_LIBS) -lX11 -L/usr/X11/lib -L/usr/X11R6/lib
 libcanberra_gtk_la_LDFLAGS = \
 	-export-dynamic -version-info $(LIBCANBERRA_GTK_VERSION_INFO)
 
 libcanberra_gtk_module_la_SOURCES = \
 	canberra-gtk-module.c
 libcanberra_gtk_module_la_CFLAGS = \
-	$(GTK_CFLAGS)
+	$(GTK_CFLAGS) \
+	$(LIBCANBERRA_CFLAGS)
 libcanberra_gtk_module_la_LIBADD = \
 	$(GTK_LIBS) \
-	libcanberra.la \
+	$(LIBCANBERRA_LIBS) \
 	libcanberra-gtk.la
 libcanberra_gtk_module_la_LDFLAGS = \
 	-avoid-version -module -export-dynamic
@@ -312,10 +310,11 @@
 	canberra-gtk-play.c
 canberra_gtk_play_LDADD = \
 	$(GTK_LIBS) \
-	libcanberra.la \
+	$(LIBCANBERRA_LIBS) \
 	libcanberra-gtk.la
 canberra_gtk_play_CFLAGS = \
-	$(GTK_CFLAGS)
+	$(GTK_CFLAGS) \
+	$(LIBCANBERRA_CFLAGS)
 
 EXTRA_DIST += \
 	libcanberra-login-sound.desktop.in \
