diff -Nurd -x'*~' nautilus-2.28.4.orig/eel/Makefile.in nautilus-2.28.4/eel/Makefile.in
--- nautilus-2.28.4.orig/eel/Makefile.in	2009-12-15 04:12:58.000000000 -0500
+++ nautilus-2.28.4/eel/Makefile.in	2010-03-08 01:30:59.000000000 -0500
@@ -52,7 +52,7 @@
 CONFIG_CLEAN_FILES =
 CONFIG_CLEAN_VPATH_FILES =
 LTLIBRARIES = $(noinst_LTLIBRARIES)
-libeel_2_la_LIBADD =
+libeel_2_la_LIBADD = $(INTLLIBS)
 am__objects_1 =
 am_libeel_2_la_OBJECTS = eel-accessibility.lo eel-alert-dialog.lo \
 	eel-art-extensions.lo eel-art-gtk-extensions.lo \
