diff -Nurd -x'*~' libapreq2-2.08.orig/configure libapreq2-2.08/configure
--- libapreq2-2.08.orig/configure	2006-08-09 00:29:00.000000000 -0400
+++ libapreq2-2.08/configure	2011-06-03 14:20:55.000000000 -0400
@@ -19050,7 +19050,7 @@
         if test "x$PERL_GLUE" != "xno"; then
             echo "$as_me:$LINENO: checking for perl" >&5
 echo $ECHO_N "checking for perl... $ECHO_C" >&6
-            if test -z "`$prereq_check perl $PERL`"; then
+            if test -z "`$prereq_check perl "$PERL"`"; then
                 { { echo "$as_me:$LINENO: error: Bad perl version" >&5
 echo "$as_me: error: Bad perl version" >&2;}
    { (exit 1); exit 1; }; }
