diff -Nurd qt-kde-qt-mac/src/3rdparty/webkit/WebCore/WebCore.pro qt-kde-qt-mac-new/src/3rdparty/webkit/WebCore/WebCore.pro
--- qt-kde-qt-mac/src/3rdparty/webkit/WebCore/WebCore.pro	2011-05-08 23:25:04.000000000 -0400
+++ qt-kde-qt-mac-new/src/3rdparty/webkit/WebCore/WebCore.pro	2011-05-17 00:11:07.000000000 -0400
@@ -2,6 +2,8 @@
 CONFIG += building-libs
 CONFIG += depend_includepath
 
+contains(QT_CONFIG, reduce_exports):CONFIG += hide_symbols
+
 symbian: {
     TARGET.EPOCALLOWDLLDATA=1
     TARGET.CAPABILITY = All -Tcb
diff -Nurd qt-kde-qt-mac/src/3rdparty/webkit/WebCore/platform/sql/SQLiteDatabase.cpp qt-kde-qt-mac-new/src/3rdparty/webkit/WebCore/platform/sql/SQLiteDatabase.cpp
--- qt-kde-qt-mac/src/3rdparty/webkit/WebCore/platform/sql/SQLiteDatabase.cpp	2010-12-27 15:22:06.000000000 -0500
+++ qt-kde-qt-mac-new/src/3rdparty/webkit/WebCore/platform/sql/SQLiteDatabase.cpp	2011-05-16 23:24:29.000000000 -0400
@@ -374,9 +374,9 @@
     m_lockingMutex.unlock();
 }
 
-bool SQLiteDatabase::isAutoCommitOn() const
+bool SQLiteDatabase::isAutoCommitOn()
 {
-    return sqlite3_get_autocommit(m_db);
+    return (sqlite3_get_autocommit(m_db) != 0);
 }
 
 } // namespace WebCore
diff -Nurd qt-kde-qt-mac/src/3rdparty/webkit/WebCore/platform/sql/SQLiteDatabase.h qt-kde-qt-mac-new/src/3rdparty/webkit/WebCore/platform/sql/SQLiteDatabase.h
--- qt-kde-qt-mac/src/3rdparty/webkit/WebCore/platform/sql/SQLiteDatabase.h	2010-12-27 15:22:06.000000000 -0500
+++ qt-kde-qt-mac-new/src/3rdparty/webkit/WebCore/platform/sql/SQLiteDatabase.h	2011-05-16 23:24:29.000000000 -0400
@@ -106,7 +106,7 @@
     // (un)locks the database like a mutex
     void lock();
     void unlock();
-    bool isAutoCommitOn() const;
+    bool isAutoCommitOn();
 
     // Set this flag to allow access from multiple threads.  Not all multi-threaded accesses are safe!
     // See http://www.sqlite.org/cvstrac/wiki?p=MultiThreading for more info.
