diff -Nurd -x'*~' wsmake6.orig/libdba/dba_db.c wsmake6/libdba/dba_db.c
--- wsmake6.orig/libdba/dba_db.c	2001-09-03 22:41:40.000000000 -0400
+++ wsmake6/libdba/dba_db.c	2010-03-17 08:06:16.000000000 -0400
@@ -21,6 +21,7 @@
  ***********************************************************************/
 #include <stddef.h>
 #include <assert.h>
+#include <stdlib.h>
 
 #include  "dba_db.h"
 
@@ -46,6 +47,7 @@
   error = db_create(&tdb, NULL, 0);
   
   error = tdb->open(tdb, 
+		  NULL, /* added txnid, new parameter in newer db */
 		  dba->filename,
 		  NULL,
                   DB_HASH,
