diff -ru gaim-0.63/src/gaimrc.c gaim-0.63-patched/src/gaimrc.c
--- gaim-0.63/src/gaimrc.c	2003-05-19 21:21:21.000000000 +0200
+++ gaim-0.63-patched/src/gaimrc.c	2003-05-19 13:00:58.000000000 +0200
@@ -1440,8 +1440,8 @@
 #else
 	report_idle = IDLE_GAIM;
 #endif
-	web_browser = BROWSER_NETSCAPE;
-	g_snprintf(web_command, sizeof(web_command), "xterm -e lynx %%s");
+	web_browser = BROWSER_MANUAL;
+	g_snprintf(web_command, sizeof(web_command), "open %%s");
 
 	auto_away = 10;
 	a = g_new0(struct away_message, 1);
