diff -urN -x *~ gtk-sharp-1.0.10/Makefile.in gtk-sharp-1.0.10-new/Makefile.in
--- gtk-sharp-1.0.10/Makefile.in	Mon May 16 19:19:16 2005
+++ gtk-sharp-1.0.10-new/Makefile.in	Sat Aug  6 10:19:14 2005
@@ -250,7 +250,7 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-SUBDIRS = sources generator parser glib pango atk gdk gtk glade art gnome gda gnomedb gtkhtml gconf rsvg vte sample doc
+SUBDIRS = sources generator parser glib pango atk gdk gtk glade art gnome gda gnomedb gtkhtml gconf rsvg vte sample
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = gtk-sharp.pc
 EXTRA_DIST = \
diff -urN -x *~ gtk-sharp-1.0.10/configure.in gtk-sharp-1.0.10-new/configure.in
--- gtk-sharp-1.0.10/configure.in	Sat May 14 19:05:17 2005
+++ gtk-sharp-1.0.10-new/configure.in	Sat Aug  6 10:19:39 2005
@@ -47,19 +47,6 @@
 AC_SUBST(BUILD_GTK_CFLAGS)
 AC_SUBST(BUILD_GTK_LIBS)
 
-PKG_PATH=
-AC_ARG_WITH(crosspkgdir, [  --with-crosspkgdir=/path/to/pkg-config/dir],
-	if test x$with_crosspkgdir = "x"; then
-		if test -s $PKG_CONFIG_PATH; then
-			PKG_PATH=$PKG_CONFIG_PATH
-		fi
-	else
-		PKG_PATH=$with_crosspkgdir
-		PKG_CONFIG_PATH=$PKG_PATH
-		export PKG_CONFIG_PATH
-	fi
-)
-
 MONO_REQUIRED_VERSION=0.96
 PKG_CHECK_MODULES(MONO_DEPENDENCY, mono >= $MONO_REQUIRED_VERSION, has_mono=true, has_mono=false)
 
diff -urN -x *~ gtk-sharp-1.0.10/glib/glue/Makefile.in gtk-sharp-1.0.10-new/glib/glue/Makefile.in
--- gtk-sharp-1.0.10/glib/glue/Makefile.in	Mon May 16 19:19:13 2005
+++ gtk-sharp-1.0.10-new/glib/glue/Makefile.in	Sat Aug  6 10:19:14 2005
@@ -137,7 +137,7 @@
 GLADE_CFLAGS = @GLADE_CFLAGS@
 GLADE_LIBS = @GLADE_LIBS@
 GLIB_CFLAGS = @GLIB_CFLAGS@
-GLIB_LIBS = @GLIB_LIBS@
+GLIB_LIBS = @GLIB_LIBS@ -lgobject-2.0
 GNOME_CFLAGS = @GNOME_CFLAGS@
 GNOME_LIBS = @GNOME_LIBS@
 GTKHTML_DEPENDENCIES_CFLAGS = @GTKHTML_DEPENDENCIES_CFLAGS@
