diff -ru monotone-viz-0.14/Makefile monotone-viz-0.14-patched/Makefile
--- monotone-viz-0.14/Makefile	2006-02-27 15:35:36.000000000 -0600
+++ monotone-viz-0.14-patched/Makefile	2006-03-20 16:55:29.000000000 -0600
@@ -140,7 +140,7 @@
 	cd glib   && rm -f *.o *.cm*
 	cd crypto && rm -f *.o *.cm*
 
-install :
+install : $(NAME)
 	install -d $(DESTDIR)$(bindir)
 	install -m 0755 monotone-viz \
 		$(DESTDIR)$(bindir)/`echo monotone-viz | sed '$(transform)'`
