--- silky-0.5.4/src/support.c	Fri Jan 28 17:16:10 2005
+++ silky-0.5.4.patched/src/support.c	Fri Mar 25 22:04:38 2005
@@ -523,6 +523,7 @@
     NULL,
     "/etc/mime.types",		/* default	*/
     "/usr/pkg/etc/mime.types",	/* pkgsrc	*/
+    "/etc/httpd/mime.types",	/* OS X		*/
     "/usr/lib/mime.types",		/* IRIX 6.5	*/
     "/usr/local/etc/mime.types",	/* default?	*/
     "mime.types",			/* Current dir	*/
