diff -Nurd -x'*~' dxpc-3.9.2.orig/Makefile.in dxpc-3.9.2/Makefile.in
--- dxpc-3.9.2.orig/Makefile.in	2007-02-02 09:25:48.000000000 -0500
+++ dxpc-3.9.2/Makefile.in	2015-11-09 01:31:42.000000000 -0500
@@ -1,5 +1,5 @@
 CXX      = @CXX@
-CXXFLAGS = @CXXFLAGS@ @DEFS@
+CXXFLAGS = @CXXFLAGS@ @CPPFLAGS@ @DEFS@ @X_CFLAGS@
 LDFLAGS  = @LDFLAGS@
 LIBS     = @LIBS@ @X_LIBS@ @X_EXTRA_LIBS@
 EXEEXT   = @EXEEXT@
