diff -Nurd Eterm-0.9.6.orig/src/eterm_utmp.h Eterm-0.9.6/src/eterm_utmp.h
--- Eterm-0.9.6.orig/src/eterm_utmp.h	2009-01-06 04:05:36.000000000 -0500
+++ Eterm-0.9.6/src/eterm_utmp.h	2021-11-13 14:32:36.000000000 -0500
@@ -69,7 +69,7 @@
 # ifdef HAVE_LASTLOG_H
 #  include <lastlog.h>
 # endif
-# if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__bsdi__) || defined(__DragonFly__)
+# if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__bsdi__) || defined(__DragonFly__) || defined(__APPLE__)
 #  include <ttyent.h>
 #  define NEW_BSD_UTMP
 # endif
