--- Eterm-0.9.4/src/eterm_utmp.h.orig   2006-05-11 04:33:39.000000000 +0100
+++ Eterm-0.9.4/src/eterm_utmp.h        2007-10-04 13:55:55.000000000 +0100
@@ -67,11 +67,11 @@
 #  include <fcntl.h>
 # endif
 # ifdef HAVE_LASTLOG_H
 #  include <lastlog.h>
 # endif
-# if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__bsdi__) || defined(__DragonFly__)
+# if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__bsdi__) || defined(__DragonFly__) || defined(__APPLE__)
 #  include <ttyent.h>
 #  define NEW_BSD_UTMP
 # endif

 # ifdef HAVE_LIBUTEMPTER
