diff -Nurd fontconfig-2.14.2.orig/fontconfig.pc.in fontconfig-2.14.2/fontconfig.pc.in
--- fontconfig-2.14.2.orig/fontconfig.pc.in	2019-10-16 06:05:28.000000000 -0400
+++ fontconfig-2.14.2/fontconfig.pc.in	2023-03-26 22:45:26.000000000 -0400
@@ -12,7 +12,5 @@
 Description: Font configuration and customization library
 Version: @VERSION@
 Requires: @PKGCONFIG_REQUIRES@
-Requires.private: @PKGCONFIG_REQUIRES_PRIVATELY@
 Libs: -L${libdir} -lfontconfig
-Libs.private: @ICONV_LIBS@ @PKG_EXPAT_LIBS@
 Cflags: -I${includedir} @ICONV_CFLAGS@ @PKG_EXPAT_CFLAGS@
diff -Nurd fontconfig-2.14.2.orig/fonts.conf.in fontconfig-2.14.2/fonts.conf.in
--- fontconfig-2.14.2.orig/fonts.conf.in	2021-07-20 07:02:19.000000000 -0400
+++ fontconfig-2.14.2/fonts.conf.in	2023-03-26 22:38:04.000000000 -0400
@@ -78,16 +78,28 @@
 	</match>
 
 <!--
+  Load per-user customization file
+-->
+	<include ignore_missing="yes">~/.fonts.conf</include>
+
+<!--
   Load local system customization file
 -->
 	<include ignore_missing="yes">@CONFIGDIR@</include>
+ 
+<!--
+  Load local (apple) system customization file
+-->
+	<include ignore_missing="yes">/opt/X11/lib/X11/fonts/local.conf</include>
+	<include ignore_missing="yes">/usr/X11/lib/X11/fonts/local.conf</include>
+	<include ignore_missing="yes">/etc/fonts/local.conf</include>
 
 <!-- Font cache directory list -->
 
 	<cachedir>@FC_CACHEDIR@</cachedir>
 	<cachedir prefix="xdg">fontconfig</cachedir>
 	<!-- the following element will be removed in the future -->
-	<cachedir>~/.fontconfig</cachedir>
+	<cachedir>~/.fontconfig-2.4</cachedir>
 
 	<config>
 <!--
