diff -Nurd libXaw3dXft-1.6.2/src/Xaw3dXft.c libXaw3dXft-1.6.2.patched/src/Xaw3dXft.c
--- libXaw3dXft-1.6.2/src/Xaw3dXft.c	2012-03-04 08:43:44.000000000 -0700
+++ libXaw3dXft-1.6.2.patched/src/Xaw3dXft.c	2012-06-08 15:42:18.000000000 -0700
@@ -46,9 +46,6 @@
 #include <X11/Xaw3d/Xaw3dXft.h>
 
 #define XAW3DXFTDEFAULTFONT "Liberation-9"
-#ifndef Pixel
-typedef unsigned long Pixel;
-#endif
 
 char Xaw3dXftEncoding = 0;
 char Xaw3dXftStringHilight = 0;
