diff -ruN motif-2.3.4-orig/clients/mwm/WmCmd.c motif-2.3.4/clients/mwm/WmCmd.c
--- motif-2.3.4-orig/clients/mwm/WmCmd.c	2012-10-22 09:50:39.000000000 -0500
+++ motif-2.3.4/clients/mwm/WmCmd.c	2020-09-20 05:57:52.000000000 -0500
@@ -48,7 +48,7 @@
 #include "WmWsm.h"
 #include "WmDebug.h"
 #include "WmWinConf.h"
-
+#include "WmFeedback.h"
 
 
 #define MAX_CLIENT_CMDS 1000
diff -ruN motif-2.3.4-orig/clients/mwm/WmFunction.c motif-2.3.4/clients/mwm/WmFunction.c
--- motif-2.3.4-orig/clients/mwm/WmFunction.c	2012-10-22 09:50:39.000000000 -0500
+++ motif-2.3.4/clients/mwm/WmFunction.c	2020-09-20 05:38:22.000000000 -0500
@@ -84,6 +84,7 @@
 /*
  * include extern functions
  */
+#include "WmCmd.h"
 #include "WmFunction.h"
 #include "WmCEvent.h"
 #ifdef WSM
diff -ruN motif-2.3.4-orig/clients/mwm/WmInitWs.c motif-2.3.4/clients/mwm/WmInitWs.c
--- motif-2.3.4-orig/clients/mwm/WmInitWs.c	2012-10-22 09:50:39.000000000 -0500
+++ motif-2.3.4/clients/mwm/WmInitWs.c	2020-09-20 05:55:09.000000000 -0500
@@ -139,6 +139,9 @@
  * Function Declarations:
  */
 
+/* Internal routines "borrowed" from libXm.  Don't try this at home! */
+extern int _XmVirtKeysLoadFallbackBindings(Display *display, String *binding);
+
 #include "WmInitWs.h"
 
 #if ((!defined(WSM)) || defined(MWM_QATS_PROTOCOL))
diff -ruN motif-2.3.4-orig/clients/mwm/WmProtocol.c motif-2.3.4/clients/mwm/WmProtocol.c
--- motif-2.3.4-orig/clients/mwm/WmProtocol.c	2012-10-22 09:50:39.000000000 -0500
+++ motif-2.3.4/clients/mwm/WmProtocol.c	2020-09-20 06:21:49.000000000 -0500
@@ -93,6 +93,8 @@
 static void    WMiConvertCB         (Widget, XtPointer, XtPointer);
 #endif /* !defined(WSM) || defined(MWM_QATS_PROTOCOL) */
 
+GetAutomationData (XtPointer input, Atom *outputType, XtPointer *output, unsigned long *outputLen, int *outputFmt);
+
 /*
  * Global Variables:
  */
diff -ruN motif-2.3.4-orig/clients/mwm/WmResParse.c motif-2.3.4/clients/mwm/WmResParse.c
--- motif-2.3.4-orig/clients/mwm/WmResParse.c	2012-10-22 09:50:39.000000000 -0500
+++ motif-2.3.4/clients/mwm/WmResParse.c	2020-09-20 05:57:52.000000000 -0500
@@ -122,6 +122,10 @@
 #include "WmImage.h"
 #include "WmXSMP.h"
 
+/* Internal routines "borrowed" from libXm.  Don't try this at home! */
+extern Boolean _XmVirtKeysLoadFileBindings(char *fileName, String *binding);
+extern int _XmVirtKeysLoadFallbackBindings(Display *display, String *binding);
+
 #ifdef MOTIF_ONE_DOT_ONE
 extern char   *getenv ();
 #endif
diff -ruN motif-2.3.4-orig/clients/mwm/WmWinList.c motif-2.3.4/clients/mwm/WmWinList.c
--- motif-2.3.4-orig/clients/mwm/WmWinList.c	2012-10-22 09:50:39.000000000 -0500
+++ motif-2.3.4/clients/mwm/WmWinList.c	2020-09-20 05:47:09.000000000 -0500
@@ -64,7 +64,8 @@
 #include "WmWrkspace.h"
 #endif /* WSM */
 
-
+/* avoid implicit declaration */
+extern Time GetTimestamp ();
 
 
 /*
diff -ruN motif-2.3.4-orig/clients/mwm/WmWsmLib/recv.c motif-2.3.4/clients/mwm/WmWsmLib/recv.c
--- motif-2.3.4-orig/clients/mwm/WmWsmLib/recv.c	2012-10-22 09:50:39.000000000 -0500
+++ motif-2.3.4/clients/mwm/WmWsmLib/recv.c	2020-09-20 05:33:54.000000000 -0500
@@ -31,6 +31,9 @@
 #include <X11/Xatom.h>
 #include <Xm/TransferP.h>	/* for XmeNamedSource() */
 
+/* avoid implicit declaration */
+extern Time GetTimestamp ();
+
 #ifdef JUNK
 static Boolean ConvertProc(
 Widget, Atom *, Atom *, XtPointer, unsigned long, int, Atom *,
diff -ruN motif-2.3.4-orig/clients/mwm/WmWsmLib/send.c motif-2.3.4/clients/mwm/WmWsmLib/send.c
--- motif-2.3.4-orig/clients/mwm/WmWsmLib/send.c	2012-10-22 09:50:39.000000000 -0500
+++ motif-2.3.4/clients/mwm/WmWsmLib/send.c	2020-09-20 05:34:21.000000000 -0500
@@ -30,6 +30,9 @@
 #include "wsm_proto.h"
 #include "utm_send.h"
 
+/* avoid implict declaration */
+extern Time GetTimestamp ();
+
 typedef struct _RequestInfo {
     WSMReplyCallbackFunc reply_callback; /* The reply callback func */
     XtPointer reply_data;	/* The user data for the callback func */
diff -ruN motif-2.3.4-orig/clients/uil/Uil.h motif-2.3.4/clients/uil/Uil.h
--- motif-2.3.4-orig/clients/uil/Uil.h	2012-10-22 09:50:39.000000000 -0500
+++ motif-2.3.4/clients/uil/Uil.h	2020-09-20 06:38:36.000000000 -0500
@@ -45,6 +45,8 @@
 #ifndef Uil_h
 #define Uil_h
 
+int yyerror();
+
 /*
 **
 **  INCLUDE FILES
diff -ruN motif-2.3.4-orig/clients/uil/UilDefI.h motif-2.3.4/clients/uil/UilDefI.h
--- motif-2.3.4-orig/clients/uil/UilDefI.h	2012-10-22 09:50:39.000000000 -0500
+++ motif-2.3.4/clients/uil/UilDefI.h	2020-09-20 14:22:33.000000000 -0500
@@ -411,6 +411,9 @@
 /* uilp2reslv.c */
 extern void sem_resolve_forward_refs  _ARGUMENTS(( void ));
 
+/* uilparser.c */
+int yyparse ();
+
 /* uilsarcomp.c */
 extern sym_value_entry_type *sem_create_cstr  _ARGUMENTS(( void ));
 extern sym_value_entry_type *sem_create_wchar_str  _ARGUMENTS(( void ));
@@ -453,6 +456,7 @@
 extern void sar_set_object_flags  _ARGUMENTS(( yystype *current_frame , unsigned char mask ));
 extern void sar_unset_object_flags  _ARGUMENTS(( yystype *current_frame , unsigned char mask ));
 extern void sar_set_list_type  _ARGUMENTS(( yystype *current_frame ));
+extern int sar_get_units_type _ARGUMENTS(( yystype *current_frame ));
 extern void sar_set_object_class  _ARGUMENTS(( yystype *current_frame ));
 extern void sar_set_object_variant  _ARGUMENTS(( yystype *current_frame ));
 extern yystype *sem_find_object  _ARGUMENTS(( yystype *current_frame ));
diff -ruN motif-2.3.4-orig/clients/xmbind/xmbind.c motif-2.3.4/clients/xmbind/xmbind.c
--- motif-2.3.4-orig/clients/xmbind/xmbind.c	2012-10-22 09:50:39.000000000 -0500
+++ motif-2.3.4/clients/xmbind/xmbind.c	2020-09-20 04:50:45.000000000 -0500
@@ -29,6 +29,7 @@
 #endif
 #endif
 #include <stdio.h>
+#include <stdlib.h>
 #include <Xm/Xm.h>
 
 /* Internal routines "borrowed" from libXm.  Don't try this at home! */
diff -ruN motif-2.3.4-orig/lib/Xm/TextF.c motif-2.3.4/lib/Xm/TextF.c
--- motif-2.3.4-orig/lib/Xm/TextF.c	2012-10-22 09:50:34.000000000 -0500
+++ motif-2.3.4/lib/Xm/TextF.c	2020-09-20 05:57:52.000000000 -0500
@@ -39,6 +39,7 @@
 #include <limits.h>		/* required for MB_LEN_MAX definition */
 #include <string.h>
 #include <ctype.h>
+#include <wchar.h>
 #include "XmI.h"
 #include <X11/ShellP.h>
 #include <X11/VendorP.h>
@@ -95,8 +96,6 @@
  * the other missing functions as _Xmwc... routines.  The new functions are
  * added static to this file.
  */
-#define wcslen(c) _Xwcslen(c)
-#define wcscpy(d,s) _Xwcscpy(d,s)
 #define wcsncpy(d,s,l) _Xwcsncpy(d,s,l)
 
 static wchar_t* _Xmwcschr(const wchar_t *ws, wchar_t wc)
diff -ruN motif-2.3.4-orig/tools/wml/wml.h motif-2.3.4/tools/wml/wml.h
--- motif-2.3.4-orig/tools/wml/wml.h	2012-10-22 09:50:39.000000000 -0500
+++ motif-2.3.4/tools/wml/wml.h	2020-09-20 04:51:39.000000000 -0500
@@ -630,7 +630,7 @@
 
 extern DynamicHandleListDefPtr	wml_tok_sens_ptr;
 extern DynamicHandleListDefPtr	wml_tok_insens_ptr;
-
+extern int dup2 ();
 
 
 
@@ -664,6 +664,10 @@
 extern void wmlAddClassResource ();
 extern void wmlAddClassResourceAttribute ();
 extern void wmlAddClassControl ();
+extern void wmlAddClassChild ();
+extern void wmlCreateChild ();
+extern void wmlCreateOrAppendCtrlList ();
+extern void wmlAddCtrlListControl ();
 extern void wmlAddCtrList ();
 extern void wmlCreateResource ();
 extern void wmlCreateDatatype ();
@@ -677,7 +681,9 @@
 extern void wmlCreateCharset ();
 extern void wmlAddCharsetAttribute ();
 extern void LexIssueError ();
-
+int yylex ();
+int yyerror ();
+int yyparse ();
 
 /*
  * Defined in wmlresolve.c
@@ -701,3 +707,5 @@
  * Define in wmloutp1 or wmloutp2
  */
 extern void wmlOutput ();
+extern void wmlOutputKeyWordFiles ();
+extern void wmlOutputMmFiles ();
diff -ruN motif-2.3.4-orig/tools/wml/wmlresolve.c motif-2.3.4/tools/wml/wmlresolve.c
--- motif-2.3.4-orig/tools/wml/wmlresolve.c	2012-10-22 09:50:39.000000000 -0500
+++ motif-2.3.4/tools/wml/wmlresolve.c	2020-09-20 04:47:57.000000000 -0500
@@ -58,6 +58,7 @@
 #include <stdlib.h>
 #endif
 #include <stdio.h>
+#include <string.h>
 
 /*
  * Routines used only in this module
