diff -uNr rxvt-unicode-9.21/Makefile.in rxvt-unicode-9.21.fink/Makefile.in
--- rxvt-unicode-9.21/Makefile.in	2014-11-16 07:22:34.000000000 -0500
+++ rxvt-unicode-9.21.fink/Makefile.in	2015-01-25 21:46:24.000000000 -0500
@@ -31,6 +31,8 @@
 
 RECURSIVE_TARGETS = all allbin alldoc tags clean distclean realclean install
 
+.PHONY: Makefiles all allbin alldoc check clean distclean distclean-local distdir install realclean tags tar.bz2 tar.gz 
+
 #-------------------------------------------------------------------------
 
 $(RECURSIVE_TARGETS):
diff -uNr rxvt-unicode-9.21/configure rxvt-unicode-9.21.fink/configure
--- rxvt-unicode-9.21/configure	2014-12-31 08:51:22.000000000 -0500
+++ rxvt-unicode-9.21.fink/configure	2015-01-25 21:38:02.000000000 -0500
@@ -7899,8 +7899,8 @@
 
      save_CXXFLAGS="$CXXFLAGS"
      save_LIBS="$LIBS"
-     CXXFLAGS="$CXXFLAGS `$PERL -MExtUtils::Embed -e ccopts`"
-     LIBS="$LIBS `$PERL -MExtUtils::Embed -e ldopts`"
+     CXXFLAGS="$CXXFLAGS `$PERL -MExtUtils::Embed -e ccopts | sed -e 's|-arch i386||g' -e 's|-arch ppc||g' -e 's|-arch x86_64||g'`"
+     LIBS="$LIBS `$PERL -MExtUtils::Embed -e ldopts | sed -e 's|-arch i386||g' -e 's|-arch ppc||g' -e 's|-arch x86_64||g'`"
      cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -7936,8 +7936,8 @@
 
         IF_PERL=
         PERL_O=rxvtperl.o
-        PERLFLAGS="`$PERL -MExtUtils::Embed -e ccopts`"
-        PERLLIB="`$PERL -MExtUtils::Embed -e ldopts`"
+        PERLFLAGS="`$PERL -MExtUtils::Embed -e ccopts  | sed -e 's|-arch i386||g' -e 's|-arch ppc||g' -e 's|-arch x86_64||g'`"
+        PERLLIB="`$PERL -MExtUtils::Embed -e ldopts | sed -e 's|-arch i386||g' -e 's|-arch ppc||g' -e 's|-arch x86_64||g'`"
         PERLPRIVLIBEXP="`$PERL -MConfig -e 'print $Config{privlibexp}'`"
      else
         as_fn_error $? "no, unable to link" "$LINENO" 5
diff -uNr rxvt-unicode-9.21/doc/Makefile.in rxvt-unicode-9.21.fink/doc/Makefile.in
--- rxvt-unicode-9.21/doc/Makefile.in	2014-12-31 09:09:25.000000000 -0500
+++ rxvt-unicode-9.21.fink/doc/Makefile.in	2015-01-25 21:39:29.000000000 -0500
@@ -98,7 +98,7 @@
 	$(INSTALL_DATA) rxvtc.1.man $(DESTDIR)$(man1dir)/$(RXVTNAME)c.$(man1ext)
 	$(INSTALL_DATA) rxvtd.1.man $(DESTDIR)$(man1dir)/$(RXVTNAME)d.$(man1ext)
 	$(INSTALL_DATA) rxvt.7.man  $(DESTDIR)$(man7dir)/$(RXVTNAME).$(man7ext)
-	@TIC@ $(srcdir)/etc/rxvt-unicode.terminfo
+	@TIC@ -o $(DESTDIR)$(prefix)/share/terminfo $(srcdir)/etc/rxvt-unicode.terminfo
 
 distdepend: alldoc
 
diff -uNr rxvt-unicode-9.21/doc/rxvt-tabbed rxvt-unicode-9.21.fink/doc/rxvt-tabbed
--- rxvt-unicode-9.21/doc/rxvt-tabbed	2014-05-22 14:53:16.000000000 -0400
+++ rxvt-unicode-9.21.fink/doc/rxvt-tabbed	2015-01-25 21:38:25.000000000 -0500
@@ -9,7 +9,7 @@
 # Also see the Gtk2::URxvt module family, which does this much cleaner
 # and provides a real tabbed-terminal.
 
-my $RXVT_BASENAME = "rxvt";
+my $RXVT_BASENAME = "urxvt";
 
 use Gtk2;
 use Encode;
