diff -uNr xforms-1.2.3/gl/Makefile.am xforms-1.2.3.fink/gl/Makefile.am
--- xforms-1.2.3/gl/Makefile.am	2011-11-29 10:35:02.000000000 -0500
+++ xforms-1.2.3.fink/gl/Makefile.am	2014-06-05 20:51:56.000000000 -0400
@@ -7,7 +7,7 @@
 
 lib_LTLIBRARIES = libformsGL.la
 
-libformsGL_la_LDFLAGS = -no-undefined -Wl,--no-undefined -version-info @SO_VERSION@
+libformsGL_la_LDFLAGS = -no-undefined -version-info @SO_VERSION@
 
 libformsGL_la_LIBADD = ../lib/libforms.la $(X_LIBS) -lGL -lX11
 
diff -uNr xforms-1.2.3/gl/Makefile.in xforms-1.2.3.fink/gl/Makefile.in
--- xforms-1.2.3/gl/Makefile.in	2014-01-15 08:53:55.000000000 -0500
+++ xforms-1.2.3.fink/gl/Makefile.in	2014-06-05 20:54:19.000000000 -0400
@@ -245,7 +245,7 @@
 INCLUDES = -DMAKING_FORMS -I../lib -I$(top_srcdir)/lib $(X_CFLAGS)
 include_HEADERS = glcanvas.h
 lib_LTLIBRARIES = libformsGL.la
-libformsGL_la_LDFLAGS = -no-undefined -Wl,--no-undefined -version-info @SO_VERSION@
+libformsGL_la_LDFLAGS = -no-undefined -Wl,-undefined -Wl,dynamic_lookup -version-info @SO_VERSION@
 libformsGL_la_LIBADD = ../lib/libforms.la $(X_LIBS) -lGL -lX11
 libformsGL_la_SOURCES = \
 	glcanvas.c \
diff -uNr xforms-1.2.3/image/Makefile.am xforms-1.2.3.fink/image/Makefile.am
--- xforms-1.2.3/image/Makefile.am	2013-12-10 18:00:45.000000000 -0500
+++ xforms-1.2.3.fink/image/Makefile.am	2014-06-05 20:51:56.000000000 -0400
@@ -6,7 +6,7 @@
 
 lib_LTLIBRARIES = libflimage.la
 
-libflimage_la_LDFLAGS = -no-undefined -Wl,--no-undefined -version-info @SO_VERSION@
+libflimage_la_LDFLAGS = -no-undefined -version-info @SO_VERSION@
 
 libflimage_la_LIBADD = ../lib/libforms.la $(JPEG_LIB) $(X_LIBS) -lX11
 
diff -uNr xforms-1.2.3/image/Makefile.in xforms-1.2.3.fink/image/Makefile.in
--- xforms-1.2.3/image/Makefile.in	2014-01-15 08:53:55.000000000 -0500
+++ xforms-1.2.3.fink/image/Makefile.in	2014-06-05 20:54:19.000000000 -0400
@@ -252,7 +252,7 @@
 INCLUDES = -DMAKING_FORMS $(X_CFLAGS)
 include_HEADERS = flimage.h
 lib_LTLIBRARIES = libflimage.la
-libflimage_la_LDFLAGS = -no-undefined -Wl,--no-undefined -version-info @SO_VERSION@
+libflimage_la_LDFLAGS = -no-undefined -Wl,-undefined -Wl,dynamic_lookup -version-info @SO_VERSION@
 libflimage_la_LIBADD = ../lib/libforms.la $(JPEG_LIB) $(X_LIBS) -lX11
 libflimage_la_SOURCES = \
 	flimage.h \
diff -uNr xforms-1.2.3/lib/Makefile.am xforms-1.2.3.fink/lib/Makefile.am
--- xforms-1.2.3/lib/Makefile.am	2013-12-19 06:04:57.000000000 -0500
+++ xforms-1.2.3.fink/lib/Makefile.am	2014-06-05 20:51:56.000000000 -0400
@@ -16,7 +16,7 @@
 
 lib_LTLIBRARIES = libforms.la
 
-libforms_la_LDFLAGS = -no-undefined -Wl,--no-undefined -version-info @SO_VERSION@
+libforms_la_LDFLAGS = -no-undefined -version-info @SO_VERSION@
 
 libforms_la_LIBADD =  $(X_LIBS) $(XPM_LIB) -lX11
 
diff -uNr xforms-1.2.3/lib/Makefile.in xforms-1.2.3.fink/lib/Makefile.in
--- xforms-1.2.3/lib/Makefile.in	2014-01-15 08:53:55.000000000 -0500
+++ xforms-1.2.3.fink/lib/Makefile.in	2014-06-05 20:54:19.000000000 -0400
@@ -305,7 +305,7 @@
 SUBDIRS = bitmaps fd include private
 INCLUDES = -DMAKING_FORMS $(X_CFLAGS) $(BWC)
 lib_LTLIBRARIES = libforms.la
-libforms_la_LDFLAGS = -no-undefined -Wl,--no-undefined -version-info @SO_VERSION@
+libforms_la_LDFLAGS = -no-undefined -Wl,-undefined -Wl,dynamic_lookup -version-info @SO_VERSION@
 libforms_la_LIBADD = $(X_LIBS) $(XPM_LIB) -lX11
 nodist_libforms_la_SOURCES = config.h
 man_MANS = xforms.5
diff -uNr xforms-1.2.3/lib/flinternal.h xforms-1.2.3.fink/lib/flinternal.h
--- xforms-1.2.3/lib/flinternal.h	2013-12-10 01:12:13.000000000 -0500
+++ xforms-1.2.3.fink/lib/flinternal.h	2014-06-05 20:53:43.000000000 -0400
@@ -138,8 +138,8 @@
 
 /* events.c or event related */
 
-FL_OBJECT * fli_handled_obj;
-FL_OBJECT * fli_handled_parent;
+extern FL_OBJECT * fli_handled_obj;
+extern FL_OBJECT * fli_handled_parent;
 
 void fli_obj_queue_delete( void );
 
