diff -Nurd xinitrc-1.5/sedsrc/xinitrc.sh.in xinitrc-1.5.patched/sedsrc/xinitrc.sh.in
--- xinitrc-1.5/sedsrc/xinitrc.sh.in	2008-02-21 09:24:08.000000000 -0700
+++ xinitrc-1.5.patched/sedsrc/xinitrc.sh.in	2013-01-27 18:36:14.000000000 -0700
@@ -38,6 +38,11 @@
     unset xinitrc__f
 fi
 
+# Read in user scripts via stock X11 script
+if test -f "$x_xinitdir/xinitrc.d/98-user.sh"; then
+	. "$x_xinitdir/xinitrc.d/98-user.sh"
+fi
+
 # Hook for system administrators.
 if test -f "$fink_sysconfdir/xinitrc-last-hook"; then
     . "$fink_sysconfdir/xinitrc-last-hook"
