diff -cPr fvwm-2.4.2/fvwm2.desktop fvwm-2.4.2-patched/fvwm2.desktop
*** fvwm-2.4.2/fvwm2.desktop	Wed Dec 31 19:00:00 1969
--- fvwm-2.4.2-patched/fvwm2.desktop	Sat Sep 22 14:34:26 2001
***************
*** 0 ****
--- 1,7 ----
+ [Desktop Entry]
+ Name=Fvwm2
+ Exec=fvwm2
+ TryExec=fvwm2
+ 
+ [Window Manager]
+ SessionManaged=true
diff -ruN fvwm-2.6.1-orig/modules/FvwmButtons/FvwmButtons.c fvwm-2.6.1/modules/FvwmButtons/FvwmButtons.c
--- fvwm-2.6.1-orig/modules/FvwmButtons/FvwmButtons.c	2007-07-22 19:01:50.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmButtons/FvwmButtons.c	2012-11-30 21:09:36.000000000 -0500
@@ -41,6 +41,7 @@
 #include <sys/wait.h>
 #include "libs/ftime.h"
 #include <sys/stat.h>
+#include <locale.h>
 
 #include <X11/keysym.h>
 #include <X11/Xlib.h>
diff -ruN fvwm-2.6.1-orig/modules/FvwmButtons/Makefile.in fvwm-2.6.1/modules/FvwmButtons/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmButtons/Makefile.in	2011-04-16 16:35:29.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmButtons/Makefile.in	2012-11-30 21:06:27.000000000 -0500
@@ -341,7 +341,7 @@
 	$(iconv_LIBS) $(Xrender_LIBS) $(Xcursor_LIBS) $(Bidi_LIBS) \
 	$(png_LIBS) $(rsvg_LIBS)
 
-INCLUDES = -I$(top_srcdir) $(xpm_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS) \
+INCLUDES = -I$(top_srcdir) $(xpm_CFLAGS) $(FT2_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS) \
 	$(iconv_CFLAGS) $(Xrender_CFLAGS) $(fribidi_CFLAGS) $(png_CFLAGS)
 
 all: all-am
diff -ruN fvwm-2.6.1-orig/modules/FvwmCommand/Makefile.in fvwm-2.6.1/modules/FvwmCommand/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmCommand/Makefile.in	2011-04-16 16:35:30.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmCommand/Makefile.in	2012-11-30 21:06:12.000000000 -0500
@@ -379,7 +379,7 @@
 # FIXME:
 # Despite not using X functions explicitly, the code includes
 # fvwmlib.h, which *does* include X headers and xpm.h!
-INCLUDES = -I$(top_srcdir) $(xpm_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS)
+INCLUDES = -I$(top_srcdir) $(xpm_CFLAGS) $(FT2_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS)
 all: all-recursive
 
 .SUFFIXES:
diff -ruN fvwm-2.6.1-orig/modules/FvwmForm/FvwmForm.c fvwm-2.6.1/modules/FvwmForm/FvwmForm.c
--- fvwm-2.6.1-orig/modules/FvwmForm/FvwmForm.c	2009-12-18 20:22:20.000000000 -0500
+++ fvwm-2.6.1/modules/FvwmForm/FvwmForm.c	2012-11-30 21:09:14.000000000 -0500
@@ -26,6 +26,7 @@
 #include <stdio.h>
 #include <ctype.h>
 #include <signal.h>
+#include <locale.h>
 
 #include "libs/ftime.h"
 #include <fcntl.h>
diff -ruN fvwm-2.6.1-orig/modules/FvwmForm/Makefile.in fvwm-2.6.1/modules/FvwmForm/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmForm/Makefile.in	2011-04-16 16:35:32.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmForm/Makefile.in	2012-11-30 21:05:34.000000000 -0500
@@ -358,7 +358,7 @@
 	$(X_PRE_LIBS) $(Xinerama_LIBS) -lXext -lX11 $(X_EXTRA_LIBS) \
 	-lm $(Xrender_LIBS) $(rsvg_LIBS) $(iconv_LIBS) $(Bidi_LIBS)
 
-INCLUDES = -I$(top_srcdir) $(Xft_CFLAGS) $(X_CFLAGS) $(iconv_CFLAGS) \
+INCLUDES = -I$(top_srcdir) $(FT2_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS) $(iconv_CFLAGS) \
 	$(Bidi_CFLAGS) $(Xrender_CFLAGS)
 
 CLEANFILES = $(module_SCRIPTS) FvwmForm-Setup
diff -ruN fvwm-2.6.1-orig/modules/FvwmGtk/Makefile.in fvwm-2.6.1/modules/FvwmGtk/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmGtk/Makefile.in	2011-04-16 16:35:32.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmGtk/Makefile.in	2012-11-30 21:05:17.000000000 -0500
@@ -341,7 +341,7 @@
 	$(GTK_LIBS) $(GDK_IMLIB_LIBS) \
 	$(GNOME_LIBDIR) $(GNOMEUI_LIBS)
 
-INCLUDES = -I$(top_srcdir) $(Xft_CFLAGS) $(X_CFLAGS) \
+INCLUDES = -I$(top_srcdir) $(FT2_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS) \
 	$(GTK_CFLAGS) $(GDK_IMLIB_CFLAGS) $(GNOME_INCLUDEDIR)
 
 all: all-am
diff -ruN fvwm-2.6.1-orig/modules/FvwmIconBox/FvwmIconBox.c fvwm-2.6.1/modules/FvwmIconBox/FvwmIconBox.c
--- fvwm-2.6.1-orig/modules/FvwmIconBox/FvwmIconBox.c	2007-02-04 10:10:29.000000000 -0500
+++ fvwm-2.6.1/modules/FvwmIconBox/FvwmIconBox.c	2012-11-30 21:08:56.000000000 -0500
@@ -44,6 +44,7 @@
 
 #include <unistd.h>
 #include <ctype.h>
+#include <locale.h>
 #include "libs/Module.h"
 #include "libs/fvwmsignal.h"
 
diff -ruN fvwm-2.6.1-orig/modules/FvwmIconBox/Makefile.in fvwm-2.6.1/modules/FvwmIconBox/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmIconBox/Makefile.in	2011-04-16 16:35:32.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmIconBox/Makefile.in	2012-11-30 21:04:30.000000000 -0500
@@ -337,7 +337,7 @@
 	$(iconv_LIBS) $(Xrender_LIBS) $(Xcursor_LIBS) $(Bidi_LIBS) \
 	$(png_LIBS) $(rsvg_LIBS)
 
-INCLUDES = -I$(top_srcdir) $(Xft_CFLAGS) $(xpm_CFLAGS) $(X_CFLAGS) \
+INCLUDES = -I$(top_srcdir) $(FT2_CFLAGS) $(Xft_CFLAGS) $(xpm_CFLAGS) $(X_CFLAGS) \
 	$(iconv_CFLAGS) $(Xrender_CFLAGS) $(Bidi_CFLAGS) $(png_CFLAGS)
 
 all: all-am
diff -ruN fvwm-2.6.1-orig/modules/FvwmIconMan/FvwmIconMan.c fvwm-2.6.1/modules/FvwmIconMan/FvwmIconMan.c
--- fvwm-2.6.1-orig/modules/FvwmIconMan/FvwmIconMan.c	2010-10-15 01:02:19.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmIconMan/FvwmIconMan.c	2012-11-30 21:08:30.000000000 -0500
@@ -19,6 +19,7 @@
 #include <unistd.h>
 #include <signal.h>
 #include <errno.h>
+#include <locale.h>
 #include "FvwmIconMan.h"
 #include "readconfig.h"
 #include "x.h"
diff -ruN fvwm-2.6.1-orig/modules/FvwmIconMan/Makefile.in fvwm-2.6.1/modules/FvwmIconMan/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmIconMan/Makefile.in	2011-04-16 16:35:32.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmIconMan/Makefile.in	2012-11-30 21:04:17.000000000 -0500
@@ -342,7 +342,7 @@
 	$(X_PRE_LIBS) $(Xinerama_LIBS) -lXext -lX11 $(X_EXTRA_LIBS) \
 	$(iconv_LIBS) $(Xrender_LIBS) $(rsvg_LIBS) -lm $(Bidi_LIBS)
 
-INCLUDES = -I$(top_srcdir)  $(Xft_CFLAGS) $(X_CFLAGS) \
+INCLUDES = -I$(top_srcdir) $(FT2_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS) \
 	$(iconv_CFLAGS) $(Xrender_CFLAGS) $(Bidi_CFLAGS)
 
 all: all-am
diff -ruN fvwm-2.6.1-orig/modules/FvwmIdent/FvwmIdent.c fvwm-2.6.1/modules/FvwmIdent/FvwmIdent.c
--- fvwm-2.6.1-orig/modules/FvwmIdent/FvwmIdent.c	2007-02-04 10:10:29.000000000 -0500
+++ fvwm-2.6.1/modules/FvwmIdent/FvwmIdent.c	2012-11-30 21:08:09.000000000 -0500
@@ -39,6 +39,7 @@
 
 #include <unistd.h>
 #include <ctype.h>
+#include <locale.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 #include <X11/Xproto.h>
diff -ruN fvwm-2.6.1-orig/modules/FvwmIdent/Makefile.in fvwm-2.6.1/modules/FvwmIdent/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmIdent/Makefile.in	2011-04-16 16:35:33.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmIdent/Makefile.in	2012-11-30 21:03:59.000000000 -0500
@@ -336,7 +336,7 @@
 	$(png_LIBS) $(X_PRE_LIBS) $(Xinerama_LIBS) -lXext -lX11 -lm \
 	$(X_EXTRA_LIBS) $(Xrender_LIBS) $(rsvg_LIBS) $(iconv_LIBS) $(Bidi_LIBS)
 
-INCLUDES = -I$(top_srcdir) $(Xft_CFLAGS) $(xpm_CFLAGS) $(png_CFLAGS) \
+INCLUDES = -I$(top_srcdir) $(FT2_CFLAGS) $(Xft_CFLAGS) $(xpm_CFLAGS) $(png_CFLAGS) \
 	$(X_CFLAGS) $(Xrender_CFLAGS) $(iconv_CFLAGS) $(Bidi_CFLAGS)
 
 all: all-am
diff -ruN fvwm-2.6.1-orig/modules/FvwmPager/FvwmPager.c fvwm-2.6.1/modules/FvwmPager/FvwmPager.c
--- fvwm-2.6.1-orig/modules/FvwmPager/FvwmPager.c	2008-01-21 19:12:07.000000000 -0500
+++ fvwm-2.6.1/modules/FvwmPager/FvwmPager.c	2012-11-30 21:07:39.000000000 -0500
@@ -31,6 +31,7 @@
 #include <sys/wait.h>
 #include "libs/ftime.h"
 #include <ctype.h>
+#include <locale.h>
 
 #ifdef HAVE_SYS_BSDTYPES_H
 #include <sys/bsdtypes.h> /* Saul */
diff -ruN fvwm-2.6.1-orig/modules/FvwmPager/Makefile.in fvwm-2.6.1/modules/FvwmPager/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmPager/Makefile.in	2011-04-16 16:35:33.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmPager/Makefile.in	2012-11-30 21:03:44.000000000 -0500
@@ -337,7 +337,7 @@
 	$(X_EXTRA_LIBS) -lm $(Xrender_LIBS) $(Xcursor_LIBS) $(iconv_LIBS) \
 	$(Bidi_LIBS)
 
-INCLUDES = -I$(top_srcdir) $(xpm_CFLAGS) $(png_CFLAGS) $(Xft_CFLAGS) \
+INCLUDES = -I$(top_srcdir) $(xpm_CFLAGS) $(png_CFLAGS) $(FT2_CFLAGS) $(Xft_CFLAGS) \
 	$(X_CFLAGS) $(Xrender_CFLAGS) $(iconv_CFLAGS) $(Bidi_CFLAGS)
 
 all: all-am
diff -ruN fvwm-2.6.1-orig/modules/FvwmProxy/FvwmProxy.c fvwm-2.6.1/modules/FvwmProxy/FvwmProxy.c
--- fvwm-2.6.1-orig/modules/FvwmProxy/FvwmProxy.c	2009-08-26 06:21:51.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmProxy/FvwmProxy.c	2012-11-30 21:07:21.000000000 -0500
@@ -23,6 +23,7 @@
 #include "config.h"
 #include <stdio.h>
 #include <limits.h>
+#include <locale.h>
 
 #include "libs/Module.h"
 #include "libs/fvwmlib.h"
diff -ruN fvwm-2.6.1-orig/modules/FvwmProxy/Makefile.in fvwm-2.6.1/modules/FvwmProxy/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmProxy/Makefile.in	2011-04-16 16:35:34.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmProxy/Makefile.in	2012-11-30 21:03:31.000000000 -0500
@@ -337,7 +337,7 @@
 	$(X_EXTRA_LIBS) -lm $(Xrender_LIBS) $(Xcursor_LIBS) $(iconv_LIBS) \
 	$(Bidi_LIBS)
 
-INCLUDES = -I$(top_srcdir) $(xpm_CFLAGS) $(png_CFLAGS) $(Xft_CFLAGS) \
+INCLUDES = -I$(top_srcdir) $(xpm_CFLAGS) $(png_CFLAGS) $(FT2_CFLAGS) $(Xft_CFLAGS) \
 	$(X_CFLAGS) $(Xrender_CFLAGS) $(iconv_CFLAGS) $(Bidi_CFLAGS)
 
 all: all-am
diff -ruN fvwm-2.6.1-orig/modules/FvwmRearrange/Makefile.in fvwm-2.6.1/modules/FvwmRearrange/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmRearrange/Makefile.in	2011-04-16 16:35:34.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmRearrange/Makefile.in	2012-11-30 21:03:14.000000000 -0500
@@ -333,7 +333,7 @@
 LDADD = -L$(top_builddir)/libs $(X_LIBS) -lm -lfvwm  \
 	$(X_PRE_LIBS) $(Xinerama_LIBS) -lXext -lX11 $(X_EXTRA_LIBS)
 
-INCLUDES = -I$(top_srcdir) $(Xft_CFLAGS) $(X_CFLAGS)
+INCLUDES = -I$(top_srcdir) $(FT2_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS)
 all: all-am
 
 .SUFFIXES:
diff -ruN fvwm-2.6.1-orig/modules/FvwmSave/Makefile.in fvwm-2.6.1/modules/FvwmSave/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmSave/Makefile.in	2011-04-16 16:35:34.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmSave/Makefile.in	2012-11-30 21:02:56.000000000 -0500
@@ -328,7 +328,7 @@
 LDADD = -L$(top_builddir)/libs $(X_LIBS) -lfvwm  \
 	$(X_PRE_LIBS) -lXext -lX11 $(X_EXTRA_LIBS)
 
-INCLUDES = -I$(top_srcdir) $(Xft_CFLAGS) $(X_CFLAGS)
+INCLUDES = -I$(top_srcdir) $(FT2_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS)
 all: all-am
 
 .SUFFIXES:
diff -ruN fvwm-2.6.1-orig/modules/FvwmSaveDesk/Makefile.in fvwm-2.6.1/modules/FvwmSaveDesk/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmSaveDesk/Makefile.in	2011-04-16 16:35:35.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmSaveDesk/Makefile.in	2012-11-30 21:02:40.000000000 -0500
@@ -328,7 +328,7 @@
 LDADD = -L$(top_builddir)/libs $(X_LIBS) -lfvwm  \
 	$(X_PRE_LIBS) -lXext -lX11 $(X_EXTRA_LIBS)
 
-INCLUDES = -I$(top_srcdir) $(Xft_CFLAGS) $(X_CFLAGS)
+INCLUDES = -I$(top_srcdir) $(FT2_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS)
 all: all-am
 
 .SUFFIXES:
diff -ruN fvwm-2.6.1-orig/modules/FvwmScript/Makefile.in fvwm-2.6.1/modules/FvwmScript/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmScript/Makefile.in	2011-04-16 16:35:35.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmScript/Makefile.in	2012-11-30 21:02:18.000000000 -0500
@@ -373,7 +373,7 @@
 	$(X_EXTRA_LIBS) -lm $(Xrender_LIBS) $(Xcursor_LIBS) $(iconv_LIBS) \
 	$(Bidi_LIBS) $(png_LIBS) $(rsvg_LIBS) $(intl_LIBS)
 
-INCLUDES = -I$(top_srcdir) $(xpm_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS) \
+INCLUDES = -I$(top_srcdir) $(xpm_CFLAGS) $(FT2_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS) \
 	$(Xrender_CFLAGS) $(iconv_CFLAGS) $(Bidi_CFLAGS) $(png_CFLAGS)
 
 
diff -ruN fvwm-2.6.1-orig/modules/FvwmScript/Widgets/Makefile.in fvwm-2.6.1/modules/FvwmScript/Widgets/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmScript/Widgets/Makefile.in	2011-04-16 16:35:36.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmScript/Widgets/Makefile.in	2012-11-30 21:02:02.000000000 -0500
@@ -306,7 +306,7 @@
 	RadioButton.c Rectangle.c Swallow.c TextField.c Tools.c \
 	Tools.h VDipstick.c VScrollBar.c Widget.c
 
-INCLUDES = -I$(top_srcdir) $(xpm_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS)
+INCLUDES = -I$(top_srcdir) $(xpm_CFLAGS) $(FT2_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS)
 all: all-am
 
 .SUFFIXES:
diff -ruN fvwm-2.6.1-orig/modules/FvwmTaskBar/Makefile.in fvwm-2.6.1/modules/FvwmTaskBar/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmTaskBar/Makefile.in	2011-04-16 16:35:36.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmTaskBar/Makefile.in	2012-11-30 21:01:38.000000000 -0500
@@ -342,7 +342,7 @@
 	$(X_EXTRA_LIBS) $(Xrender_LIBS) $(Xcursor_LIBS) $(iconv_LIBS) \
 	$(Bidi_LIBS) $(intl_LIBS)
 
-INCLUDES = -I$(top_srcdir) $(xpm_CFLAGS) $(png_CFLAGS) $(Xft_CFLAGS) \
+INCLUDES = -I$(top_srcdir) $(xpm_CFLAGS) $(png_CFLAGS) $(FT2_CFLAGS) $(Xft_CFLAGS) \
 	$(X_CFLAGS) $(Xrender_CFLAGS) $(iconv_CFLAGS) $(Bidi_CFLAGS) \
 	$(intl_CFLAGS)
 
diff -ruN fvwm-2.6.1-orig/modules/FvwmWinList/FvwmWinList.c fvwm-2.6.1/modules/FvwmWinList/FvwmWinList.c
--- fvwm-2.6.1-orig/modules/FvwmWinList/FvwmWinList.c	2007-01-27 06:57:08.000000000 -0500
+++ fvwm-2.6.1/modules/FvwmWinList/FvwmWinList.c	2012-11-30 21:07:02.000000000 -0500
@@ -50,6 +50,7 @@
 #include <stdio.h>
 #include <signal.h>
 #include <fcntl.h>
+#include <locale.h>
 #include "libs/ftime.h"
 
 #ifdef HAVE_SYS_BSDTYPES_H
diff -ruN fvwm-2.6.1-orig/modules/FvwmWinList/Makefile.in fvwm-2.6.1/modules/FvwmWinList/Makefile.in
--- fvwm-2.6.1-orig/modules/FvwmWinList/Makefile.in	2011-04-16 16:35:37.000000000 -0400
+++ fvwm-2.6.1/modules/FvwmWinList/Makefile.in	2012-11-30 21:01:07.000000000 -0500
@@ -340,7 +340,7 @@
 	$(X_PRE_LIBS) $(Xinerama_LIBS) -lXext -lX11 -lm $(X_EXTRA_LIBS) \
 	$(Xrender_LIBS) $(rsvg_LIBS) $(iconv_LIBS) $(Bidi_LIBS)
 
-INCLUDES = -I$(top_srcdir) $(Xft_CFLAGS) $(X_CFLAGS) $(Xrender_CFLAGS) \
+INCLUDES = -I$(top_srcdir) $(FT2_CFLAGS) $(Xft_CFLAGS) $(X_CFLAGS) $(Xrender_CFLAGS) \
 	$(iconv_CFLAGS) $(Bidi_CFLAGS)
 
 all: all-am
