Find python library with distutils.sysconfig

diff -r 313a2911ca6e py2app/build_app.py
--- a/py2app/build_app.py	Fri Jan 19 00:14:30 2007 -0800
+++ b/py2app/build_app.py	Fri Jan 19 00:24:43 2007 -0800
@@ -15,7 +15,7 @@ from cStringIO import StringIO
 
 from setuptools import Command
 from distutils.util import convert_path
-from distutils import log
+from distutils import log, sysconfig
 from distutils.errors import *
 
 from altgraph.compat import *
@@ -350,7 +350,7 @@ class py2app(Command):
         # XXX - this is a bit of a hack!
         #       ideally we'd use dylib functions to figure this out
         if prefix is None:
-            prefix = sys.prefix
+            prefix = sysconfig.get_config_var('LIBPL')
         if version is None:
             version = sys.version
         version = version[:3]
@@ -366,7 +366,7 @@ class py2app(Command):
             runtime = os.path.join(info['location'], info['name'])
         else:
             dylib = 'libpython%s.dylib' % (sys.version[:3],)
-            runtime = os.path.join(prefix, 'lib', dylib)
+            runtime = os.path.join(prefix, dylib)
         return dylib, runtime
 
     def symlink(self, src, dst):
