diff -Nurd -x'*~' control.orig/src/Makefile control/src/Makefile
--- control.orig/src/Makefile	2015-10-01 08:32:55.000000000 -0400
+++ control/src/Makefile	2019-11-06 10:20:35.000000000 -0500
@@ -30,6 +30,8 @@
 # the developer makefile makefile_control.m
 slicotlibrary.a: slicot.tar.gz
 	tar -xzf slicot.tar.gz
+	/usr/bin/sed -i '.orig' 's/DGEGS/DGGES/g' slicot/src/SG03AD.f slicot/src/SG03BD.f
+	/usr/bin/sed -i '.orig' 's/DLATZM/DORMRZ/g' slicot/src/AB08NX.f slicot/src/AG08BY.f slicot/src/SB01BY.f slicot/src/SB01FY.f
 	mkdir sltmp
 	mv slicot/src/*.f ./sltmp
 	mv slicot/src_aux/*.f ./sltmp
