diff -Naur avidemux-2.0.40/avidemux/ADM_colorspace/yv12_rgb.cpp avidemux-new/avidemux/ADM_colorspace/yv12_rgb.cpp
--- avidemux-2.0.40/avidemux/ADM_colorspace/yv12_rgb.cpp	2005-05-08 11:42:50.000000000 -0400
+++ avidemux-new/avidemux/ADM_colorspace/yv12_rgb.cpp	2006-03-27 02:08:20.000000000 -0500
@@ -45,7 +45,7 @@
  * the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
  *
 
-  2²21=													2097152
+  2ï¿½21=													2097152
 R = Y + 1.402V              2940207
 G = Y - 0.344U - 0.714V     721420 1497366
 B = Y + 1.772U              3716153
@@ -233,7 +233,7 @@
 
 void fastYV12RGB_mmx ( uint8_t * py,
 			       uint8_t * pu, uint8_t * pv,  uint8_t * image,
-			       int h_size)
+			       uint32_t h_size)
 {
     int i_x = 0; //, i_y = 0; //,even=1;
 	volatile uint8_t *cy,*cu,*cv,*o; //,*line;
diff -Naur avidemux-2.0.40/avidemux/ADM_colorspace/yv12rgb_altivec.cpp avidemux-new/avidemux/ADM_colorspace/yv12rgb_altivec.cpp
--- avidemux-2.0.40/avidemux/ADM_colorspace/yv12rgb_altivec.cpp	2004-09-26 07:30:42.000000000 -0400
+++ avidemux-new/avidemux/ADM_colorspace/yv12rgb_altivec.cpp	2006-03-27 02:08:20.000000000 -0500
@@ -6,7 +6,7 @@
 this is my first altivec simd, probably less than optimal
 
 
-    2²21=2097152
+ï¿½   2ï¿½21=2097152
 R = Y + 1.402V              2940207
 G = Y - 0.344U - 0.714V     721420 1497366
 B = Y + 1.772U              3716153
@@ -53,7 +53,7 @@
 #define vecbyte vector unsigned char
 #define vecshort vector unsigned short
 
-uint8_t altivecYV12RGB(uint8_t * ptr_y,
+void altivecYV12RGB(uint8_t * ptr_y,
 		    uint8_t * ptr_u,
 		    uint8_t * ptr_v, uint8_t * ptr2, uint32_t w);
 
@@ -78,7 +78,7 @@
 //	We handle 8 pixels at a time
 //	My first altivec code :)
 //____________________________________
-uint8_t altivecYV12RGB(uint8_t * ptr_y,
+void altivecYV12RGB(uint8_t * ptr_y,
 		    uint8_t * ptr_u,
 		    uint8_t * ptr_v, uint8_t * ptr2, uint32_t w)
 {
@@ -280,7 +280,6 @@
 
 	}
 	
-	return 1;
 }
 //
 
diff -Naur avidemux-2.0.40/avidemux/ADM_colorspace/yv2bmp.cpp avidemux-new/avidemux/ADM_colorspace/yv2bmp.cpp
--- avidemux-2.0.40/avidemux/ADM_colorspace/yv2bmp.cpp	2005-05-08 11:42:50.000000000 -0400
+++ avidemux-new/avidemux/ADM_colorspace/yv2bmp.cpp	2006-03-27 02:08:20.000000000 -0500
@@ -29,7 +29,7 @@
 #include "colorspace.h"
 #include "ADM_toolkit/ADM_cpuCap.h"
 #ifdef HAVE_ALTIVEC
-extern uint8_t altivecYV12RGB(uint8_t * ptr_y,
+extern void altivecYV12RGB(uint8_t * ptr_y,
 		    uint8_t * ptr_u,
 		    uint8_t * ptr_v, uint8_t * ptr2, uint32_t w);
 #endif		    
@@ -42,7 +42,7 @@
 
 extern void fastYV12RGB_mmx ( uint8_t * py,
 			       uint8_t * pu, uint8_t * pv,  uint8_t * image,
-			       int h_size);
+			       uint32_t h_size);
 static uint8_t Clip[256+64+64];
 
 /**
@@ -115,7 +115,7 @@
 		Y 4 bits U 4 bits V 4 bits
 		
 
-    2²21=2097152
+ï¿½   2ï¿½21=2097152
 R = Y + 1.402V              2940207
 G = Y - 0.344U - 0.714V     721420 1497366
 B = Y + 1.772U              3716153
@@ -204,7 +204,7 @@
 //____________________________________________________________
 // YUYV Version
 //____________________________________________________________
-typedef void (proto_yuv) ( uint8_t * py, uint8_t * pu, uint8_t * pv,  uint8_t * image, int h_size);
+typedef void (proto_yuv) ( uint8_t * py, uint8_t * pu, uint8_t * pv,  uint8_t * image, uint32_t h_size);
 
 uint8_t  COL_yv12rgb(uint32_t w, uint32_t h,uint8_t * ptr, uint8_t * ptr2 )
 {
diff -Naur avidemux-2.0.40/configure avidemux-new/configure
--- avidemux-2.0.40/configure	2005-05-25 04:32:18.000000000 -0400
+++ avidemux-new/configure	2006-03-27 02:07:58.000000000 -0500
@@ -21660,20 +21660,20 @@
 		have_little_endian=yes;
 
 cat >>confdefs.h <<\_ACEOF
-#define ARCH_X86 1
+#undef ARCH_X86
 _ACEOF
 
 
 cat >>confdefs.h <<\_ACEOF
-#define HAVE_MMX 1
+#undef HAVE_MMX
 _ACEOF
 
 
 cat >>confdefs.h <<\_ACEOF
-#define RUNTIME_CPUDETECT 1
+#undef RUNTIME_CPUDETECT
 _ACEOF
 
-                use_mmx=1
+                use_mmx=0
 		echo "X86 is ON"
 		;;
 	x86_64|amd64)
--- avidemux-2.0.42/avidemux/ADM_gui2/GUI_accelRender.h.orig	2006-03-20 20:45:07.000000000 +0100
+++ avidemux-2.0.42/avidemux/ADM_gui2/GUI_accelRender.h	2006-03-20 20:45:23.000000000 +0100
@@ -12,7 +12,7 @@
 class AccelRender
 {
 	public:
-		AccelRender::AccelRender( void);
+		AccelRender( void);
 		virtual	uint8_t init(GtkWidget * window, uint32_t w, uint32_t h)=0;
 		virtual	uint8_t end(void)=0;
 		virtual uint8_t display(uint8_t *ptr, uint32_t w, uint32_t h)=0;
--- avidemux-2.0.42/avidemux/ADM_inpics/ADM_pics.h.orig	2006-03-20 20:37:14.000000000 +0100
+++ avidemux-2.0.42/avidemux/ADM_inpics/ADM_pics.h	2006-03-20 20:37:36.000000000 +0100
@@ -38,9 +38,9 @@
 
 					uint32_t 		_type;
 
-					uint32_t    picHeader::read32(FILE *fd);
-					uint16_t    picHeader::read16(FILE *fd);
-					uint8_t    picHeader::read8(FILE *fd);
+					uint32_t    read32(FILE *fd);
+					uint16_t    read16(FILE *fd);
+					uint8_t    read8(FILE *fd);
 
 public:
 //  static int checkFourCC(uint8_t *in, uint8_t *fourcc);
--- avidemux-2.0.42/avidemux/ADM_video/ADM_vidDGbob.cpp.orig	2006-03-20 20:43:19.000000000 +0100
+++ avidemux-2.0.42/avidemux/ADM_video/ADM_vidDGbob.cpp	2006-03-20 20:43:40.000000000 +0100
@@ -55,8 +55,8 @@
         void            update(void); 
 public:
                                 
-                        DGbob::DGbob(AVDMGenericVideoStream *in,CONFcouple *couples);    
-                        DGbob::~DGbob(void);
+                        DGbob(AVDMGenericVideoStream *in,CONFcouple *couples);    
+                        ~DGbob(void);
         uint8_t         getFrameNumberNoAlloc(uint32_t frame, uint32_t *len,
                                 ADMImage *data,uint32_t *flags);
         
--- avidemux-2.0.42/avidemux/ADM_video/ADM_vidDecDec.cpp.orig	2006-03-20 20:42:14.000000000 +0100
+++ avidemux-2.0.42/avidemux/ADM_video/ADM_vidDecDec.cpp	2006-03-20 20:42:31.000000000 +0100
@@ -147,8 +147,8 @@
 	
 public:
 				
-			Decimate::Decimate(AVDMGenericVideoStream *in,CONFcouple *couples);    
-			Decimate::~Decimate(void);
+			Decimate(AVDMGenericVideoStream *in,CONFcouple *couples);    
+			~Decimate(void);
 	uint8_t  	getFrameNumberNoAlloc(uint32_t frame, uint32_t *len,
 				ADMImage *data,uint32_t *flags);
 
--- avidemux-2.0.42/avidemux/ADM_video/ADM_vidDecTelecide.h.orig	2006-03-20 20:39:18.000000000 +0100
+++ avidemux-2.0.42/avidemux/ADM_video/ADM_vidDecTelecide.h	2006-03-20 20:40:49.000000000 +0100
@@ -112,32 +112,32 @@
 
 public:
 	
-	void Telecide::CalculateMetrics(int n, unsigned char *crp, unsigned char *crpU, unsigned char *crpV, 
+	void CalculateMetrics(int n, unsigned char *crp, unsigned char *crpU, unsigned char *crpV, 
 				unsigned char *prp, unsigned char *prpU, unsigned char *prpV);
-	void Telecide::Show(ADMImage *dst, int frame);
-	void Telecide::Debug(int frame);
+	void Show(ADMImage *dst, int frame);
+	void Debug(int frame);
 
-	Telecide::Telecide(AVDMGenericVideoStream *in,CONFcouple *couples);		
+	Telecide(AVDMGenericVideoStream *in,CONFcouple *couples);		
 			
-	Telecide::~Telecide();
+	~Telecide();
 
-	void Telecide::PutChosen(int frame, unsigned int chosen);
+	void PutChosen(int frame, unsigned int chosen);
 	
 
-	void Telecide::CacheInsert(int frame, unsigned int p, unsigned int pblock,
+	void CacheInsert(int frame, unsigned int p, unsigned int pblock,
 				unsigned int c, unsigned int cblock);
 	
-	bool Telecide::CacheQuery(int frame, unsigned int *p, unsigned int *pblock,
+	bool CacheQuery(int frame, unsigned int *p, unsigned int *pblock,
 				unsigned int *c, unsigned int *cblock);	
 
-	bool Telecide::PredictHardYUY2(int frame, unsigned int *predicted, unsigned int *predicted_metric) ;
+	bool PredictHardYUY2(int frame, unsigned int *predicted, unsigned int *predicted_metric) ;
 	
-	struct PREDICTION *Telecide::PredictSoftYUY2(int frame);
+	struct PREDICTION *PredictSoftYUY2(int frame);
 
-	void Telecide::WriteHints(unsigned char *dst, bool film, bool inpattern);
-	virtual uint8_t Telecide::getFrameNumberNoAlloc(uint32_t frame, uint32_t *len,
+	void WriteHints(unsigned char *dst, bool film, bool inpattern);
+	virtual uint8_t getFrameNumberNoAlloc(uint32_t frame, uint32_t *len,
 				ADMImage *data,uint32_t *flags);
-	char *Telecide::printConf( void );
-	uint8_t Telecide::configure(AVDMGenericVideoStream *in);
-	uint8_t	Telecide::getCoupledConf( CONFcouple **couples);
+	char *printConf( void );
+	uint8_t configure(AVDMGenericVideoStream *in);
+	uint8_t	getCoupledConf( CONFcouple **couples);
 };
