diff -ruN freetype-2.2.1-orig/modules.cfg freetype-2.2.1/modules.cfg
--- freetype-2.2.1-orig/modules.cfg	2006-03-01 02:01:21.000000000 +0900
+++ freetype-2.2.1/modules.cfg	2006-11-19 15:40:45.000000000 +0900
@@ -124,7 +124,7 @@
 
 # OpenType table validation.  Needs ftotval.c below.
 #
-# AUX_MODULES += otvalid
+AUX_MODULES += otvalid
 
 # Auxiliary PostScript driver component to share common code.
 #
