--- gd-2.0.35.orig/config/gdlib-config.in	2006-10-11 05:46:22.000000000 -0400
+++ gd-2.0.35/config/gdlib-config.in	2011-08-09 21:02:10.000000000 -0400
@@ -68,10 +68,10 @@
 	echo @GDLIB_REVISION@
 	;;
     --ldflags)
-	echo @LDFLAGS@
+	echo -L@libdir@
 	;;
     --libs)
-	echo @LIBS@ @LIBICONV@
+	echo 
 	;;
     --cflags|--includes)
 	echo -I@includedir@
@@ -83,8 +83,8 @@
 	echo "GD library  @VERSION@"
 	echo "includedir: $includedir"
 	echo "cflags:     -I@includedir@"
-	echo "ldflags:    @LDFLAGS@"
-	echo "libs:       @LIBS@ @LIBICONV@"
+	echo "ldflags:    -L@libdir@"
+	echo "libs:       "
 	echo "libdir:     $libdir"
 	echo "features:   @FEATURES@"
 	;;
--- ./gd_png.c.orig	2007-06-14 14:51:41.000000000 -0500
+++ ./gd_png.c	2010-02-12 01:43:36.000000000 -0600
@@ -149,7 +149,7 @@
 	return NULL;
   }
 
-	if (!png_check_sig (sig, 8)) { /* bad signature */
+	if (!(png_sig_cmp(sig, 0, 8) == 0)) { /* bad signature */
     	return NULL;		/* bad signature */
 	}
 
