diff -Nurd -x'*~' libopenraw-0.0.9.orig/configure libopenraw-0.0.9/configure
--- libopenraw-0.0.9.orig/configure	2011-11-13 16:04:00.000000000 -0500
+++ libopenraw-0.0.9/configure	2013-04-19 09:17:34.000000000 -0400
@@ -16412,16 +16412,6 @@
 fi
 
 
-SHAVE_SAVED_LIBTOOL=$LIBTOOL
-
-LIBTOOL="\$(SHELL) \$(shavedir)/shave-libtool '\$(SHAVE_SAVED_LIBTOOL)'"
-
-SHAVE_SAVED_CC=$CC
-SHAVE_SAVED_CXX=$CXX
-
-
-CC="\$(SHELL) \$(shavedir)/shave cc '\$(SHAVE_SAVED_CC)'"
-CXX="\$(SHELL) \$(shavedir)/shave cxx '\$(SHAVE_SAVED_CXX)'"
 
 
 
diff -Nurd -x'*~' libopenraw-0.0.9.orig/test/Makefile.in libopenraw-0.0.9/test/Makefile.in
--- libopenraw-0.0.9.orig/test/Makefile.in	2011-11-13 16:04:00.000000000 -0500
+++ libopenraw-0.0.9/test/Makefile.in	2012-01-27 14:45:26.000000000 -0500
@@ -268,7 +268,7 @@
 
 EXTRA_DIST = ljpegtest1.jpg
 fileio_LDFLAGS = -L$(top_builddir)/lib/io
-fileio_LDADD = -lio
+fileio_LDADD = ../lib/libopenraw.la
 fileio_SOURCES = fileio.t.c
 extensions_SOURCES = extensions.cpp
 extensions_LDFLAGS = -static $(OPENRAW_LIB)
