--- ./metapixel-0.7/libzoom/filt.c.bak	2005-04-03 17:06:35.000000000 -0700
+++ ./metapixel-0.7/libzoom/filt.c	2005-04-03 17:07:04.000000000 -0700
@@ -73,10 +73,10 @@
     Filt *window;
 } window_data;
 
-void mitchell_init(), mitchell_print();
-void kaiser_init(), kaiser_print();
-void window_print();
-double window_func();
+static void mitchell_init(), mitchell_print();
+static void kaiser_init(), kaiser_print();
+static void window_print();
+static double window_func();
 static mitchell_data md;
 static kaiser_data kd;
 
--- ./metapixel-0.7/libzoom/zoom.c.bak	2005-04-03 17:01:21.000000000 -0700
+++ ./metapixel-0.7/libzoom/zoom.c	2005-04-03 17:06:08.000000000 -0700
@@ -58,6 +58,16 @@
 #include "scanline.h"
 #include "zoom.h"
 
+typedef struct {	/* ZOOM-SPECIFIC FILTER PARAMETERS */
+    double scale;	/* filter scale (spacing between centers in a space) */
+    double supp;	/* scaled filter support radius */
+    int wid;		/* filter width: max number of nonzero samples */
+} Filtpar;
+
+static make_weighttab(int b, double cen, Filt *filter, Filtpar *ap, int len, int trimzeros, Weighttab *wtab);
+static make_map_table(double scale, double tran, double asupp, int a0, int b0, int bn, int overlap, short *map);
+static statistics();
+
 #define EPSILON 1e-7			/* error tolerance */
 #define UNDEF PIC_UNDEFINED
 
@@ -68,12 +78,6 @@
 #define INTEGER(x) (fabs((x)-floor((x)+.5)) < EPSILON)	/* is x an integer? */
 #define FRAC(x) fabs((x)-floor((x)+.5))		/* diff from closest integer */
 
-typedef struct {	/* ZOOM-SPECIFIC FILTER PARAMETERS */
-    double scale;	/* filter scale (spacing between centers in a space) */
-    double supp;	/* scaled filter support radius */
-    int wid;		/* filter width: max number of nonzero samples */
-} Filtpar;
-
 int zoom_debug = 0;	/* debug level: 0=none, 1=scanline i/o, 2=filters */
 int zoom_coerce = 1;	/* simplify filters if possible? */
 int zoom_xy = UNDEF;	/* filter x before y (1) or vice versa (0)? */
