diff -Nurd -x'*~' PDFlib-Lite-7.0.3.orig/configure PDFlib-Lite-7.0.3/configure
--- PDFlib-Lite-7.0.3.orig/configure	2008-03-25 10:19:46.000000000 -0400
+++ PDFlib-Lite-7.0.3/configure	2008-05-22 14:43:51.000000000 -0400
@@ -3335,8 +3335,8 @@
 	fi
 	CPPFLAGS="$CPPFLAGS -I/Developer/Headers/FlatCarbon"
 	LDFLAGS="$LDFLAGS -framework ApplicationServices"
-	MAC_CFLAGS="$MAC_CFLAGS -isysroot $DEVSDK"
-	MAC_LDFLAGS="$MAC_LDFLAGS -Wl,-syslibroot,$DEVSDK"
+#	MAC_CFLAGS="$MAC_CFLAGS -isysroot $DEVSDK"
+#	MAC_LDFLAGS="$MAC_LDFLAGS -Wl,-syslibroot,$DEVSDK"
 
 	# set PLATFROM ID
 	# will be completed in pc_config.h (32/64 ppc/intel)
