diff -Nurd pixman-0.11.8/configure pixman-0.11.8-new/configure
--- pixman-0.11.8/configure	2008-07-17 13:39:07.000000000 -0400
+++ pixman-0.11.8-new/configure	2008-07-22 17:44:38.000000000 -0400
@@ -19917,12 +19917,18 @@
 
 cat >conftest.$ac_ext <<_ACEOF
 
+#ifdef __APPLE__
+#ifndef __SSE__
+#error "Need __SSE___ defined on Mac OS X"
+#endif
+#else
 #if defined(__GNUC__) && (__GNUC__ < 3 || (__GNUC__ == 3 && __GNUC_MINOR__ < 4))
 #error "Need GCC >= 3.4 for SSE intrinsics"
 #endif
 #if !defined(__amd64__) && !defined(__x86_64__)
 #error "Need x86-64 for SSE"
 #endif
+#endif
 #include <mmintrin.h>
 #include <xmmintrin.h>
 int main () {
@@ -19995,9 +20001,15 @@
 
 cat >conftest.$ac_ext <<_ACEOF
 
+#ifdef __APPLE__
+#ifndef __SSE2__
+#error "Need __SSE2__ definition on Mac OS X"
+#endif
+#else
 #if defined(__GNUC__) && (__GNUC__ < 4 || (__GNUC__ == 4 && __GNUC_MINOR__ < 2))
 #error "Need GCC >= 4.2 for SSE2 intrinsics"
 #endif
+#endif
 #include <mmintrin.h>
 #include <xmmintrin.h>
 #include <emmintrin.h>
