diff -Nurd -x'*~' povray-3.6.1.orig/configure povray-3.6.1/configure
--- povray-3.6.1.orig/configure	2004-08-02 19:20:13.000000000 -0400
+++ povray-3.6.1/configure	2012-09-03 12:29:26.000000000 -0400
@@ -1384,10 +1384,10 @@
 
 # Required versions of the supporting libraries.
 # Must be declared after AC_INIT.
-required_libz_version="1.2.1"
-required_libpng_version="1.2.5"
-required_libjpeg_version="6b"
-required_libtiff_version="3.6.1"
+required_libz_version=""
+required_libpng_version=""
+required_libjpeg_version=""
+required_libtiff_version=""
 
 # Supported options.
 
@@ -9821,9 +9821,9 @@
 if test $ac_cv_lib_X11_XFlush = yes; then
 
         # X_CFLAGS should actually be X_CPPFLAGS since it only adds -Idir
-        CPPFLAGS="$X_CFLAGS $CPPFLAGS"
-        LDFLAGS="$X_LIBS $LDFLAGS"
-        LIBS="$X_EXTRA_LIBS $X_PRE_LIBS -lX11 $LIBS"
+        CPPFLAGS="$CPPFLAGS $X_CFLAGS"
+        LDFLAGS="$LDFLAGS $X_LIBS"
+        LIBS="$LIBS $X_EXTRA_LIBS $X_PRE_LIBS -lX11"
         pov_xwin_msg="enabled"
 
 else
@@ -10442,7 +10442,7 @@
 fi
 rm -f conftest.$ac_objext conftest$ac_exeext conftest.$ac_ext
 if test "$ac_cv_search_png_get_libpng_ver" = no; then
-  for ac_lib in png12 png; do
+  for ac_lib in png; do
     LIBS="-l$ac_lib
    $ac_func_search_save_LIBS"
     cat >conftest.$ac_ext <<_ACEOF
