diff -ruN Chromium-0.9-orig/configure Chromium-0.9/configure
--- Chromium-0.9-orig/configure	2001-05-11 01:17:12.000000000 -0400
+++ Chromium-0.9/configure	2007-07-02 11:03:09.000000000 -0400
@@ -14,8 +14,8 @@
 
 ## Where do we look for stuff?
 ###################################
-std_bin_dirs="/usr/bin /usr/X11R6/bin /usr/local/bin $HOME/bin"
-std_lib_dirs="/usr/lib /usr/X11R6/lib /usr/local/lib $HOME/lib"
+std_bin_dirs="/usr/bin @FINKPREFIX@/bin $HOME/bin"
+std_lib_dirs="/usr/lib @FINKPREFIX@/lib $HOME/lib"
 
 ## Bring in the functions
 ###################################
@@ -157,9 +157,9 @@
 
 ## set up other config.mak variables
 ######################################################################
-GL_LIBS="-lglpng -lGL -lGLU -lm"
-AL_LIBS="-L../support/openal/lib -lopenal -ldl" 
-AL_CFLAGS="-I../support/openal/linux/include -I../support/openal/include"
+GL_LIBS="-lglpng -framework OpenGL -lm"
+AL_LIBS="-framework OpenAL -ldl"
+AL_CFLAGS="-F/System/Library/Frameworks/OpenAL.framework -I../support/openal/include"
 
 ## write out config.mak
 ######################################################################
diff -ruN Chromium-0.9-orig/scripts/Makefile.no-setup Chromium-0.9/scripts/Makefile.no-setup
--- Chromium-0.9-orig/scripts/Makefile.no-setup	2001-02-09 21:32:31.000000000 -0500
+++ Chromium-0.9/scripts/Makefile.no-setup	2007-07-02 09:56:25.000000000 -0400
@@ -1,5 +1,4 @@
 SUPPORT_DIRS	=	 \
-			./support/openal/ \
 			./support/glpng/src
 
 CHROMIUM_DIRS	=	\
diff -ruN Chromium-0.9-orig/scripts/configure_functions Chromium-0.9/scripts/configure_functions
--- Chromium-0.9-orig/scripts/configure_functions	2001-03-31 16:21:24.000000000 -0500
+++ Chromium-0.9/scripts/configure_functions	2007-07-02 09:56:25.000000000 -0400
@@ -191,7 +191,7 @@
 	local LIBVORBISFILE
 	
 	for directory in $std_lib_dirs; do
-		if [ -f $directory/libvorbis.so ]; then
+		if [ -f $directory/libvorbis.dylib ]; then
 			LIBVORBISDIR=$directory
 			LIBVORBIS="-lvorbis"
 		fi
diff -ruN Chromium-0.9-orig/src/MainSDL_Event.cpp Chromium-0.9/src/MainSDL_Event.cpp
--- Chromium-0.9-orig/src/MainSDL_Event.cpp	2001-05-20 21:36:48.000000000 -0400
+++ Chromium-0.9/src/MainSDL_Event.cpp	2007-07-02 09:56:25.000000000 -0400
@@ -325,10 +325,10 @@
 			game->hero->moveEvent( 10, 0);
 			break;
 	    case SDLK_UP:
-			game->hero->moveEvent(0, -10);
+			game->hero->moveEvent(0, 10);
 			break;
 	    case SDLK_DOWN:
-			game->hero->moveEvent(0,  10);
+			game->hero->moveEvent(0, -10);
 			break;
 //	    case SDLK_SPACE:
 //			game->hero->fireGun(true);
diff -ruN Chromium-0.9-orig/src/Makefile Chromium-0.9/src/Makefile
--- Chromium-0.9-orig/src/Makefile	2001-05-22 01:25:50.000000000 -0400
+++ Chromium-0.9/src/Makefile	2007-07-02 11:12:45.000000000 -0400
@@ -12,10 +12,10 @@
 CXX	=	g++
 CFLAGS	=	-pipe $(PKG_CFLAGS) $(AL_CFLAGS) $(SDL_CFLAGS) $(SMPEG_CFLAGS) -O2 -DOLD_OPENAL -DAUDIO_OPENAL -D_REENTRANT
 CXXFLAGS=	-pipe $(PKG_CFLAGS) $(AL_CFLAGS) $(SDL_CFLAGS) $(SMPEG_CFLAGS) -O2 -DOLD_OPENAL -DAUDIO_OPENAL -D_REENTRANT
-INCPATH	=	-I../support/include -I../support/glpng/include -I/usr/X11R6/include
+INCPATH	=	-I../support/include -I../support/glpng/include -I@FINKPREFIX@/include
 LINK	=	g++
 LFLAGS	=	
-LIBS	=	$(SUBLIBS) -L../support/glpng/lib -L/usr/lib -L/usr/X11R6/lib -L/usr/local/lib $(GL_LIBS) $(AL_LIBS) $(SDL_LIBS) $(SMPEG_LIBS) $(VORBIS_LIBS)
+LIBS	=	$(SUBLIBS) -L../support/glpng/lib -L/usr/lib -L@FINKPREFIX@/lib $(GL_LIBS) $(AL_LIBS) $(SDL_LIBS) $(SMPEG_LIBS) $(VORBIS_LIBS) -undefined suppress -flat_namespace
 MOC	=	$(QTDIR)/bin/moc
 UIC	=	$(QTDIR)/bin/uic
 
diff -ruN Chromium-0.9-orig/src/NCString.cpp Chromium-0.9/src/NCString.cpp
--- Chromium-0.9-orig/src/NCString.cpp	2001-05-20 13:04:58.000000000 -0400
+++ Chromium-0.9/src/NCString.cpp	2007-07-02 09:56:25.000000000 -0400
@@ -1055,7 +1055,7 @@
 uint NCString::toULong(const char *str, bool *ok)
 {
 	int n;
-	ulong tmp = 0;
+	unsigned long tmp = 0;
 	if(str)
 	{
 		n = sscanf(str, "%lu", &tmp);
diff -ruN Chromium-0.9-orig/src/define.h Chromium-0.9/src/define.h
--- Chromium-0.9-orig/src/define.h	2001-02-09 21:33:24.000000000 -0500
+++ Chromium-0.9/src/define.h	2007-07-02 09:56:25.000000000 -0400
@@ -51,7 +51,7 @@
 #ifdef PKGDATADIR
 #define CHROMIUM_DATA_DIR PKGDATADIR"/data"
 #else
-#define CHROMIUM_DATA_DIR "../data"
+#define CHROMIUM_DATA_DIR "@FINKPREFIX@/share/chromium"
 #endif //PKGDATADIR
 
 #endif // define_h
diff -ruN Chromium-0.9-orig/src/main.cpp Chromium-0.9/src/main.cpp
--- Chromium-0.9-orig/src/main.cpp	2001-05-20 21:48:37.000000000 -0400
+++ Chromium-0.9/src/main.cpp	2007-07-02 10:33:58.000000000 -0400
@@ -31,7 +31,7 @@
 
 #include "EnemyAircraft.h"
 
-static char dataDir[256] = "../data";
+static char dataDir[256] = "@FINKPREFIX@/share/chromium";
 
 //----------------------------------------------------------
 int main(int argc, char **argv)
@@ -121,7 +121,7 @@
 		#ifdef macintosh
 		strcpy(dataDir, "::data");	
 		#else
-		strcpy(dataDir, "../data");	
+		strcpy(dataDir, "@FINKPREFIX@/share/chromium");	
 		#endif
 		fprintf(stderr, "!!\n");
 		fprintf(stderr, "!! WARNING - CHROMIUM_DATA environment variable is not set!\n");
diff -ruN Chromium-0.9-orig/support/glpng/src/Makefile Chromium-0.9/support/glpng/src/Makefile
--- Chromium-0.9-orig/support/glpng/src/Makefile	2001-02-08 00:03:24.000000000 -0500
+++ Chromium-0.9/support/glpng/src/Makefile	2007-07-02 11:11:48.000000000 -0400
@@ -11,7 +11,7 @@
 CXX	=	g++
 CFLAGS	=	-pipe -O2
 CXXFLAGS=	-pipe -O2
-INCPATH	=	-I../include -I/usr/include -I/usr/X11R6/include
+INCPATH	=	-I../include -I/usr/include -I@FINKPREFIX@/include
 AR	=	ar cqs
 RANLIB	=	
 MOC	=	$(QTDIR)/bin/moc
