--- glob2-0.9.1/SConstruct~	2007-09-10 00:17:39.000000000 +0200
+++ glob2-0.9.1/SConstruct	2007-09-14 03:23:22.000000000 +0200
@@ -4,12 +4,19 @@
     opts.Add("LINKFLAGS", "Manually add to the LINKFLAGS", "")
     opts.Add("INSTALLDIR", "Installation Directory", "/usr/local/share")
     opts.Add("BINDIR", "Binary Installation Directory", "/usr/local/bin")
+    opts.Add("DATADIR", "Final destination of data files", "/usr/local/share")
+    opts.Add("FINKPREFIX", "Fink directory", "/sw")
     opts.Add(BoolOption("release", "Build for release", 0))
     opts.Add(BoolOption("mingw", "Build with mingw enabled if not auto-detected", 0))
     opts.Add(BoolOption("osx", "Build for OSX", 0))
     Help(opts.GenerateHelpText(env))
     opts.Update(env)
     opts.Save('options_cache.py', env)
+
+    env.Append(CPPPATH = [ env["FINKPREFIX"] + "/include" ])
+    env.Append(CPPPATH = [ "/usr/X11R6/include" ])
+    env.Append(LIBPATH = [ env["FINKPREFIX"] + "/lib"])
+    env.Append(LIBPATH = [ "/usr/X11R6/lib" ])
     
     
 class Configuration:
@@ -29,7 +36,7 @@
     configfile = Configuration()
     configfile.add("PACKAGE", "Name of package", "\"glob2\"")
     configfile.add("PACKAGE_BUGREPORT", "Define to the address where bug reports for this package should be sent.", "\"glob2-devel@nongnu.org\"")
-    configfile.add("PACKAGE_DATA_DIR", "data directory", "\"" + env["INSTALLDIR"] + "\"")
+    configfile.add("PACKAGE_DATA_DIR", "data directory", "\"" + env["DATADIR"] + "\"")
     configfile.add("PACKAGE_SOURCE_DIR", "source directory", "\"" +env.Dir("#").abspath.replace("\\", "\\\\") + "\"")
     configfile.add("PACKAGE_NAME", "Define to the full name of this package.", "\"Globulation 2\"")
     configfile.add("PACKAGE_TARNAME", "Define to the one symbol short name of this package.", "\"glob2\"")
@@ -94,10 +101,8 @@
     elif conf.CheckLib('opengl32') and conf.CheckCHeader('GL/gl.h'):
         gl_libraries.append("opengl32")
     else:
-        #Quick fix for OSX, ignore libraries not found
-        if not env['osx']:
-            print "Could not find libGL or opengl32, or could not find GL/gl.h or OpenGL/gl.h"
-            Exit(1)
+        print "Could not find libGL or opengl32, or could not find GL/gl.h or OpenGL/gl.h"
+        Exit(1)
     
     #Do checks for GLU, which is different on every system
     if conf.CheckLib('GLU') and conf.CheckCHeader("GL/glu.h"):
@@ -147,8 +152,6 @@
     else:
         env.ParseConfig("sdl-config --cflags")
         env.ParseConfig("sdl-config --libs")
-    if env['osx']:
-        env.Append(CXXFLAGS="-framework OpenGL")
     env.Append(LIBS=['vorbisfile', 'SDL_ttf', 'SDL_image', 'SDL_net', 'speex', 'boost_thread'])
     
     env["TARFILE"] = env.Dir("#").abspath + "/glob2-" + env["VERSION"] + ".tar.gz"
