diff -Naur grustibus-0.43/src/grustibus.c grustibus-0.43-new/src/grustibus.c
--- grustibus-0.43/src/grustibus.c	Thu May 10 08:18:21 2001
+++ grustibus-0.43-new/src/grustibus.c	Tue Jun  3 14:17:55 2003
@@ -694,12 +694,12 @@
 
         *exec = g_strconcat("\"", general_prefs.xmame_exe, "\"",
                             " -rompath ", "\"", general_prefs.rom_dir,
-                            "\"", " -cheatfile ", "\"",
+                            "\"", " -cheat_file ", "\"",
                             general_prefs.cheat_file, "\"",
-                            " -historyfile ", "\"",
+                            " -history_file ", "\"",
                             general_prefs.game_history_file, "\"",
-                            " -screenshotdir ", "\"", screenshotdir, "\"",
-                            " -spooldir ", "\"",
+                            " -snapshot_directory ", "\"", screenshotdir, "\"",
+                            " -hiscore_directory ", "\"",
                             general_prefs.highscore_dir, "\"",
                             game_settings.fullscreen ? fullscreen :
                             windowed,
