--- ./unixio.c	Thu Oct  5 10:08:10 2000
+++ ./unixio.c-darwin	Wed May 14 12:19:13 2003
@@ -53,4 +53,9 @@
 #include <sys/select.h>
 #endif
+/* needed by Darwin */
+#if defined(__APPLE__)
+#include <unistd.h> /* select */
+#include <ncurses.h> /* tgetflag */
+#endif
 
 #define EXTENDED 1
--- ./fileio.c	Thu Oct  5 08:08:10 2000
+++ ./fileio.c-fink	Wed May 14 12:46:39 2003
@@ -230,5 +230,7 @@
    {
       fprintf( stderr, "%s ", tmp );
-      fatal( "open_story(): Zcode file not found" );
+      /* If INFOCOM_PATH env var not set, try fink infocom dir */
+      path = "@PREFIX@/share/zcode-games";
+      /* fatal( "open_story(): Zcode file not found" ); */
    }
    else if ( STANDALONE_FLAG && ( path = getenv( "PATH" ) ) == NULL )
--- ./osdepend.c	2005-05-12 01:25:50.000000000 -0700
+++ ./osdepend.c-fink	2005-05-12 01:26:29.000000000 -0700
@@ -87,7 +87,7 @@
 /* getopt linkages */
 
 extern int optind;
-extern const char *optarg;
+extern char *optarg;
 extern ZINT16 default_fg, default_bg;
 
 #endif /* !defined(AMIGA) */

