diff -ru SDL_mixer-1.2.10/native_midi/native_midi_macosx.c SDL_mixer-1.2.10-patched/native_midi/native_midi_macosx.c
--- SDL_mixer-1.2.10/native_midi/native_midi_macosx.c	2009-11-05 01:37:12.000000000 +0100
+++ SDL_mixer-1.2.10-patched/native_midi/native_midi_macosx.c	2009-11-12 20:08:08.000000000 +0100
@@ -107,7 +107,7 @@
         if (AUGraphGetIndNode(graph, i, &node) != noErr)
             continue;  /* better luck next time. */
 
-#if MAC_OS_X_VERSION_MIN_REQUIRED <= MAC_OS_X_VERSION_10_4 /* this is deprecated, but works back to 10.0 */
+#if !__LP64__ /* this is deprecated, but works back to 10.0 */
         {
             struct ComponentDescription desc;
             UInt32 classdatasize = 0;
@@ -197,7 +197,7 @@
     free(buf);
     buf = NULL;
 
-    #if MAC_OS_X_VERSION_MIN_REQUIRED <= MAC_OS_X_VERSION_10_4 /* this is deprecated, but works back to 10.3 */
+    #if !__LP64__ /* this is deprecated, but works back to 10.3 */
     if (MusicSequenceLoadSMFDataWithFlags(retval->sequence, data, 0) != noErr)
         goto fail;
     #else  /* not deprecated, but requires 10.5 or later */
