diff -Nurd -x'*~' warp7.orig/term.h warp7/term.h
--- warp7.orig/term.h	2001-05-27 14:30:05.000000000 -0400
+++ warp7/term.h	2006-01-09 22:42:04.000000000 -0500
@@ -115,7 +115,7 @@
 #		endif /* lint */
 #	    else /* RDCHK */
 #		ifndef O_NDELAY	/* assert O_NDELAY */
-		    ??? PENDING isn't defined correctly in warp.h
+		    ??? PENDING is not defined correctly in warp.h
 #		endif
 		EXT int devtty INIT(0);
 #		ifndef lint
@@ -126,7 +126,7 @@
 #	    endif /* RDCHK */
 #	endif /* FIONREAD */
 #   else /* PENDING */
-	??? warp won't work without PENDING
+	??? warp will not work without PENDING
 #	ifndef lint
 #	    define input_pending() (nextin!=nextout)
 #	else
@@ -154,7 +154,7 @@
 #		endif /* lint */
 #	    else /* RDCHK */
 #		ifndef O_NDELAY	/* assert O_NDELAY */
-		    ??? PENDING isn't defined correctly in warp.h
+		    ??? PENDING is not defined correctly in warp.h
 #		endif
 		EXT int devtty INIT(0);
 		EXT bool is_input INIT(FALSE);
@@ -168,7 +168,7 @@
 #	    endif /* RDCHK */
 #	endif /* FIONREAD */
 #   else /* PENDING */
-	??? warp won't work without PENDING
+	??? warp will not work without PENDING
 #	define read_tty(addr,size) read(0,addr,size)
 #	define input_pending() (FALSE)
 #   endif /* PENDING */
diff -Nurd -x'*~' warp7.orig/warp.h warp7/warp.h
--- warp7.orig/warp.h	2001-05-21 21:46:36.000000000 -0400
+++ warp7/warp.h	2006-01-09 22:37:53.000000000 -0500
@@ -322,16 +322,8 @@
 
 /* various things of type char */
 
-char	*index();
-char	*rindex();
-char	*getenv();
-char	*strcat();
-char	*strcpy();
-#ifdef CHARSPRINTF
-char	*sprintf();
-#else
-int	sprintf();
-#endif
+#include <stdio.h>
+#include <string.h>
 
 EXT char buf[LBUFLEN+1];	/* general purpose line buffer */
 
