diff -Naur --exclude='*~' wesnoth-1.3.10/configure wesnoth-1.3.10.fink/configure
--- wesnoth-1.3.10/configure	2007-11-09 16:06:12.000000000 -0600
+++ wesnoth-1.3.10.fink/configure	2007-12-08 20:57:48.000000000 -0600
@@ -2534,75 +2534,9 @@
 { echo "$as_me:$LINENO: checking how to create a ustar tar archive" >&5
 echo $ECHO_N "checking how to create a ustar tar archive... $ECHO_C" >&6; }
 # Loop over all known methods to create a tar archive until one works.
-_am_tools='gnutar plaintar pax cpio none'
-_am_tools=${am_cv_prog_tar_ustar-$_am_tools}
-# Do not fold the above two line into one, because Tru64 sh and
-# Solaris sh will not grok spaces in the rhs of `-'.
-for _am_tool in $_am_tools
-do
-  case $_am_tool in
-  gnutar)
-    for _am_tar in tar gnutar gtar;
-    do
-      { echo "$as_me:$LINENO: $_am_tar --version" >&5
-   ($_am_tar --version) >&5 2>&5
-   ac_status=$?
-   echo "$as_me:$LINENO: \$? = $ac_status" >&5
-   (exit $ac_status); } && break
-    done
-    am__tar="$_am_tar --format=ustar -chf - "'"$$tardir"'
-    am__tar_="$_am_tar --format=ustar -chf - "'"$tardir"'
-    am__untar="$_am_tar -xf -"
-    ;;
-  plaintar)
-    # Must skip GNU tar: if it does not support --format= it doesn't create
-    # ustar tarball either.
-    (tar --version) >/dev/null 2>&1 && continue
-    am__tar='tar chf - "$$tardir"'
-    am__tar_='tar chf - "$tardir"'
-    am__untar='tar xf -'
-    ;;
-  pax)
-    am__tar='pax -L -x ustar -w "$$tardir"'
-    am__tar_='pax -L -x ustar -w "$tardir"'
-    am__untar='pax -r'
-    ;;
-  cpio)
-    am__tar='find "$$tardir" -print | cpio -o -H ustar -L'
-    am__tar_='find "$tardir" -print | cpio -o -H ustar -L'
-    am__untar='cpio -i -H ustar -d'
-    ;;
-  none)
-    am__tar=false
-    am__tar_=false
-    am__untar=false
-    ;;
-  esac
-
-  # If the value was cached, stop now.  We just wanted to have am__tar
-  # and am__untar set.
-  test -n "${am_cv_prog_tar_ustar}" && break
-
-  # tar/untar a dummy directory, and stop if the command works
-  rm -rf conftest.dir
-  mkdir conftest.dir
-  echo GrepMe > conftest.dir/file
-  { echo "$as_me:$LINENO: tardir=conftest.dir && eval $am__tar_ >conftest.tar" >&5
-   (tardir=conftest.dir && eval $am__tar_ >conftest.tar) >&5 2>&5
-   ac_status=$?
-   echo "$as_me:$LINENO: \$? = $ac_status" >&5
-   (exit $ac_status); }
-  rm -rf conftest.dir
-  if test -s conftest.tar; then
-    { echo "$as_me:$LINENO: $am__untar <conftest.tar" >&5
-   ($am__untar <conftest.tar) >&5 2>&5
-   ac_status=$?
-   echo "$as_me:$LINENO: \$? = $ac_status" >&5
-   (exit $ac_status); }
-    grep GrepMe conftest.dir/file >/dev/null 2>&1 && break
-  fi
-done
-rm -rf conftest.dir
+am__tar="tar --format=ustar -chf - "'"$$tardir"'
+am__tar_="tar --format=ustar -chf - "'"$tardir"'
+am__untar="tar -xf -"
 
 if test "${am_cv_prog_tar_ustar+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
@@ -13250,6 +13184,7 @@
 { echo "$as_me:$LINENO: checking for PNG support in SDL_image" >&5
 echo $ECHO_N "checking for PNG support in SDL_image... $ECHO_C" >&6; }
 
+if test "x$enable_sdltest" = "xyes" ; then
 OLD_CPPFLAGS=$CPPFLAGS
 OLD_CFLAGS=$CFLAGS
 OLD_LIBS=$LIBS
@@ -13328,7 +13263,7 @@
 CPPFLAGS=$OLD_CPPFLAGS
 CFLAGS=$OLD_CFLAGS
 LIBS=$OLD_LIBS
-
+fi
 
 #######################################################################
 # Check for OGG support in SDL_mixer                                  #
@@ -13345,6 +13280,7 @@
 	{ echo "$as_me:$LINENO: checking for OGG support in SDL_mixer" >&5
 echo $ECHO_N "checking for OGG support in SDL_mixer... $ECHO_C" >&6; }
 
+	if test "x$enable_sdltest" = "xyes" ; then
         OLD_CPPFLAGS=$CPPFLAGS
 	OLD_CFLAGS=$CFLAGS
         OLD_LIBS=$LIBS
@@ -13421,6 +13357,7 @@
         CPPFLAGS=$OLD_CPPFLAGS
 	CFLAGS=$OLD_CFLAGS
         LIBS=$OLD_LIBS
+	fi
     fi
 fi
 
