diff -Naur --exclude='*~' wesnoth-1.4.6/configure wesnoth-1.4.6.fink/configure
--- wesnoth-1.4.6/configure	2008-11-02 12:20:54.000000000 -0800
+++ wesnoth-1.4.6.fink/configure	2008-12-06 14:34:20.000000000 -0800
@@ -13961,6 +13961,7 @@
 { $as_echo "$as_me:$LINENO: checking for PNG support in SDL_image" >&5
 $as_echo_n "checking for PNG support in SDL_image... " >&6; }
 
+if test "x$enable_sdltest" = "xyes" ; then
 OLD_CPPFLAGS=$CPPFLAGS
 OLD_CFLAGS=$CFLAGS
 OLD_LIBS=$LIBS
@@ -14042,7 +14043,7 @@
 CPPFLAGS=$OLD_CPPFLAGS
 CFLAGS=$OLD_CFLAGS
 LIBS=$OLD_LIBS
-
+fi
 
 #######################################################################
 # Check for OGG support in SDL_mixer                                  #
@@ -14059,6 +14060,7 @@
 	{ $as_echo "$as_me:$LINENO: checking for OGG support in SDL_mixer" >&5
 $as_echo_n "checking for OGG support in SDL_mixer... " >&6; }
 
+	if test "x$enable_sdltest" = "xyes" ; then
         OLD_CPPFLAGS=$CPPFLAGS
 	OLD_CFLAGS=$CFLAGS
         OLD_LIBS=$LIBS
@@ -14138,6 +14140,7 @@
         CPPFLAGS=$OLD_CPPFLAGS
         CFLAGS=$OLD_CFLAGS
         LIBS=$OLD_LIBS
+	fi
     fi
 fi
 
