diff -Nurd -x'*~' xscorch-0.2.0.orig/sgtk/Makefile.in xscorch-0.2.0/sgtk/Makefile.in
--- xscorch-0.2.0.orig/sgtk/Makefile.in	2004-03-25 00:24:11.000000000 -0500
+++ xscorch-0.2.0/sgtk/Makefile.in	2007-05-14 22:42:59.000000000 -0400
@@ -146,7 +146,7 @@
 libscorch_gtk_a_SOURCES =  	sai-gtk.c		sautodef-gtk.c		scolor-gtk.c		scolor-gtk.h		sdialog-gtk.h		seconomy-gtk.c		send-gtk.c		sexplosion-gtk.c 	sexplosion-gtk.h 	sfile-gtk.c		sfont-gtk.c		sgraphics-gtk.c		sgtk.h			simage-gtk.c		simage-gtk.h		sinventory-gtk.c 	sland-gtk.c		slottery-gtk.c		smainmenu-gtk.c		smenu-gtk.c		smenu-gtk.h		snet-gtk.c		snet-gtk.h		soptions-gtk.c		spaint-gtk.c		sphysics-gtk.c		splayer-gtk.c		ssetup-gtk.h		ssound-gtk.c		sspill-gtk.c		sstatus-gtk.c		sstatus-gtk.h		ssystem-gtk.c		ssystem-gtk.h		stank-gtk.c		stank-gtk.h		sweapons-gtk.c		swindow-gtk.c		swindow-gtk.h
 
 
-INCLUDES = $(GTK_CFLAGS) -Iswidgets -I../libj
+INCLUDES = -Iswidgets -I../libj $(GTK_CFLAGS)
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../config.h
 CONFIG_CLEAN_FILES = 
diff -Nurd -x'*~' xscorch-0.2.0.orig/sgtk/sdialog-gtk.h xscorch-0.2.0/sgtk/sdialog-gtk.h
--- xscorch-0.2.0.orig/sgtk/sdialog-gtk.h	2004-02-26 10:27:56.000000000 -0500
+++ xscorch-0.2.0/sgtk/sdialog-gtk.h	2007-05-14 22:27:45.000000000 -0400
@@ -31,6 +31,8 @@
 #include <slabel.h>
 #include <sgame/shelp.h>
 #include <libj/jstr/libjstr.h>
+#undef GTK_DISABLE_DEPRECATED
+#include <gtk/gtkcombo.h>
 
 
 /* Tooltips and dialogue helpers */
diff -Nurd -x'*~' xscorch-0.2.0.orig/sgtk/swidgets/Makefile.in xscorch-0.2.0/sgtk/swidgets/Makefile.in
--- xscorch-0.2.0.orig/sgtk/swidgets/Makefile.in	2004-03-25 00:24:12.000000000 -0500
+++ xscorch-0.2.0/sgtk/swidgets/Makefile.in	2007-05-14 22:42:46.000000000 -0400
@@ -142,7 +142,7 @@
 libscorch_widget_gtk_a_SOURCES =  	sactiveconsole.c 	sactiveconsole.h 	sactoggle.c		sactoggle.h		sconsole.c		sconsole.h		sdialog.c		sdialog.h		sdisplay.c		sdisplay.h		sdrawbuf.c		sdrawbuf.h		slabel.c		slabel.h		slinkcheck.c		slinkcheck.h		slinkcombo.c		slinkcombo.h		slinkentry.c		slinkentry.h		slinkspin.c		slinkspin.h
 
 
-INCLUDES = $(GTK_CFLAGS) -I../ -I../../sutil -I../../libj
+INCLUDES = -I../ -I../../sutil -I../../libj $(GTK_CFLAGS)
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../../config.h
 CONFIG_CLEAN_FILES = 
diff -Nurd -x'*~' xscorch-0.2.0.orig/sgtk/swidgets/sdialog.h xscorch-0.2.0/sgtk/swidgets/sdialog.h
--- xscorch-0.2.0.orig/sgtk/swidgets/sdialog.h	2004-02-26 10:27:58.000000000 -0500
+++ xscorch-0.2.0/sgtk/swidgets/sdialog.h	2007-05-14 22:38:13.000000000 -0400
@@ -26,6 +26,7 @@
 
 
 #include <sgtk.h>
+#undef GTK_DISABLE_DEPRECATED
 #include <gtk/gtk.h>
 #include <gtk/gtkwindow.h>
 
diff -Nurd -x'*~' xscorch-0.2.0.orig/sgtk/swidgets/slinkcombo.h xscorch-0.2.0/sgtk/swidgets/slinkcombo.h
--- xscorch-0.2.0.orig/sgtk/swidgets/slinkcombo.h	2004-02-26 10:27:58.000000000 -0500
+++ xscorch-0.2.0/sgtk/swidgets/slinkcombo.h	2007-05-14 22:26:51.000000000 -0400
@@ -27,6 +27,7 @@
 
 #include <sgtk.h>
 #include <gtk/gtk.h>
+#undef GTK_DISABLE_DEPRECATED
 #include <gtk/gtkcombo.h>
 
 
