diff -urN -x '*~' -x '*.orig' kakasi-020928.orig/extconf.rb kakasi-020928/extconf.rb
--- kakasi-020928.orig/extconf.rb	2002-09-28 13:20:42.000000000 +0900
+++ kakasi-020928/extconf.rb	2008-03-20 16:01:02.000000000 +0900
@@ -1,5 +1,4 @@
 require "mkmf"
 
-$CFLAGS += " -I /usr/local/include "
-$LOCAL_LIBS += " -L /usr/local/lib -lkakasi "
+$LOCAL_LIBS += " -lkakasi "
 create_makefile("kakasi")
diff -urN -x '*~' -x '*.orig' kakasi-020928.orig/test.rb kakasi-020928/test.rb
--- kakasi-020928.orig/test.rb	2002-09-28 13:20:42.000000000 +0900
+++ kakasi-020928/test.rb	2008-03-20 16:09:03.000000000 +0900
@@ -1,4 +1,4 @@
-require "./kakasi.so"
+require "kakasi.so"
 include Kakasi
 
 puts KAKASI_VERSION
