diff -Naur denemo-0.6.0.orig/Makefile.in denemo-0.6.0/Makefile.in
--- denemo-0.6.0.orig/Makefile.in	Tue May 28 04:11:28 2002
+++ denemo-0.6.0/Makefile.in	Fri Aug  2 00:40:54 2002
@@ -153,7 +153,7 @@
 
 denemo_LDADD = $(INTLLIBS) 
 
-pkgdata_DATA = denemo.conf denemo.keymaprc english.keymaprc french.keymaprc
+pkgdata_DATA = denemo.conf denemo.keymaprc english.keymaprc french.keymaprc macosx.keymaprc
 
 EXTRA_DIST = DESIGN GOALS @PACKAGE@.spec.in denemo.conf frogparser.y \
 froglexer.l $(pkgdata_DATA)
diff -Naur denemo-0.6.0.orig/aclocal.m4 denemo-0.6.0/aclocal.m4
--- denemo-0.6.0.orig/aclocal.m4	Tue May 28 04:11:23 2002
+++ denemo-0.6.0/aclocal.m4	Fri Aug  2 00:42:29 2002
@@ -987,7 +987,7 @@
    AC_REQUIRE([AC_FUNC_MMAP])dnl
    AC_REQUIRE([jm_GLIBC21])dnl
 
-   AC_CHECK_HEADERS([argz.h limits.h locale.h nl_types.h malloc.h stddef.h \
+   AC_CHECK_HEADERS([argz.h limits.h locale.h nl_types.h stddef.h \
 stdlib.h string.h unistd.h sys/param.h])
    AC_CHECK_FUNCS([feof_unlocked fgets_unlocked getcwd getegid geteuid \
 getgid getuid mempcpy munmap putenv setenv setlocale stpcpy strchr strcasecmp \
diff -Naur denemo-0.6.0.orig/analysis_highlighting.c denemo-0.6.0/analysis_highlighting.c
--- denemo-0.6.0.orig/analysis_highlighting.c	Mon Feb  4 04:23:00 2002
+++ denemo-0.6.0/analysis_highlighting.c	Fri Aug  2 00:40:54 2002
@@ -17,7 +17,7 @@
 #include <unistd.h>
 #include <sys/types.h>
 #ifndef G_OS_WIN32
-# include <wait.h>
+# include <sys/wait.h>
 #endif
 #include <errno.h>
 
diff -Naur denemo-0.6.0.orig/csoundplayback.c denemo-0.6.0/csoundplayback.c
--- denemo-0.6.0.orig/csoundplayback.c	Fri May 24 14:36:41 2002
+++ denemo-0.6.0/csoundplayback.c	Fri Aug  2 00:40:54 2002
@@ -17,7 +17,7 @@
 #include <unistd.h>
 #include <sys/types.h>
 #ifdef G_OS_WIN32
-#include <wait.h>
+#include <sys/wait.h>
 #endif
 #include <errno.h>
 
diff -Naur denemo-0.6.0.orig/exportxml.c denemo-0.6.0/exportxml.c
--- denemo-0.6.0.orig/exportxml.c	Tue May 28 03:58:55 2002
+++ denemo-0.6.0/exportxml.c	Fri Aug  2 00:40:54 2002
@@ -15,7 +15,7 @@
 #include <string.h>
 
 /* libxml includes: for libxml2 this should be <libxml/tree.h> */
-#include <tree.h>
+#include <libxml/tree.h>
 
 
 #define XML_COMPRESSION_RATIO 3
diff -Naur denemo-0.6.0.orig/exportxml.h denemo-0.6.0/exportxml.h
--- denemo-0.6.0.orig/exportxml.h	Tue Jun 26 04:04:47 2001
+++ denemo-0.6.0/exportxml.h	Fri Aug  2 00:40:54 2002
@@ -14,7 +14,7 @@
 #include <glib.h>
 
 /* libxml includes: for libxml2 this should be <libxml/tree.h> */
-#include <tree.h>
+#include <libxml/tree.h>
 
 
 /*
diff -Naur denemo-0.6.0.orig/file.c denemo-0.6.0/file.c
--- denemo-0.6.0.orig/file.c	Tue May 28 03:58:55 2002
+++ denemo-0.6.0/file.c	Fri Aug  2 00:43:10 2002
@@ -27,7 +27,7 @@
 #include <sys/stat.h> /* check existance and type of files */
 #include <dirent.h>   /* filter and sort filenames */
 #include <fnmatch.h>  /* find filenames matches */
-#include <malloc.h>   /* I use free() */
+					  /* I use free() */
 
 
 typedef enum {DENEMO_FORMAT = 0,
diff -Naur denemo-0.6.0.orig/frogparser.c denemo-0.6.0/frogparser.c
--- denemo-0.6.0.orig/frogparser.c	Tue May 28 03:58:56 2002
+++ denemo-0.6.0/frogparser.c	Fri Aug  2 00:44:35 2002
@@ -432,10 +432,6 @@
 /* This used to test MSDOS, but that is a bad idea
    since that symbol is in the user namespace.  */
 #if (defined (_MSDOS) || defined (_MSDOS_)) && !defined (__TURBOC__)
-#if 0 /* No need for malloc.h, which pollutes the namespace;
-	 instead, just don't use alloca.  */
-#include <malloc.h>
-#endif
 #else /* not MSDOS, or __TURBOC__ */
 #if defined(_AIX)
 /* I don't know what this was needed for, but it pollutes the namespace.
diff -Naur denemo-0.6.0.orig/importxml.c denemo-0.6.0/importxml.c
--- denemo-0.6.0.orig/importxml.c	Tue May 28 03:58:56 2002
+++ denemo-0.6.0/importxml.c	Fri Aug  2 00:40:54 2002
@@ -18,8 +18,8 @@
 #include <glib.h>
 
 /* libxml includes: for libxml2 this should be <libxml/{parser,tree}.h> */
-#include <parser.h>
-#include <tree.h>
+#include <libxml/parser.h>
+#include <libxml/tree.h>
 
 
 /* Defines for making traversing XML trees easier */
diff -Naur denemo-0.6.0.orig/importxml.h denemo-0.6.0/importxml.h
--- denemo-0.6.0.orig/importxml.h	Tue Jun 26 04:07:19 2001
+++ denemo-0.6.0/importxml.h	Fri Aug  2 00:40:54 2002
@@ -14,7 +14,7 @@
 #include <glib.h>
 
 /* libxml includes: for libxml2 this should be <libxml/tree.h> */
-#include <tree.h>
+#include <libxml/tree.h>
 
 
 /*
diff -Naur denemo-0.6.0.orig/intl/plural.c denemo-0.6.0/intl/plural.c
--- denemo-0.6.0.orig/intl/plural.c	Wed Nov 14 04:27:25 2001
+++ denemo-0.6.0/intl/plural.c	Fri Aug  2 00:45:01 2002
@@ -350,10 +350,6 @@
 /* This used to test MSDOS, but that is a bad idea
    since that symbol is in the user namespace.  */
 #if (defined (_MSDOS) || defined (_MSDOS_)) && !defined (__TURBOC__)
-#if 0 /* No need for malloc.h, which pollutes the namespace;
-	 instead, just don't use alloca.  */
-#include <malloc.h>
-#endif
 #else /* not MSDOS, or __TURBOC__ */
 #if defined(_AIX)
 /* I don't know what this was needed for, but it pollutes the namespace.
diff -Naur denemo-0.6.0.orig/macosx.keymaprc denemo-0.6.0/macosx.keymaprc
--- denemo-0.6.0.orig/macosx.keymaprc	Wed Dec 31 19:00:00 1969
+++ denemo-0.6.0/macosx.keymaprc	Fri Aug  2 00:40:54 2002
@@ -0,0 +1,74 @@
+CursorLeft h Left 
+CursorDown Down 
+CursorUp Up 
+CursorRight Right 
+StaffUp Up(4) 
+StaffDown Down(4) 
+MeasureLeft Left(4) 
+MeasureRight Right(4) 
+ToNearestA a 
+ToNearestB b 
+ToNearestC c 
+ToNearestD d 
+ToNearestE e 
+ToNearestF f 
+ToNearestG g 
+OctaveUp apostrophe(1) 
+OctaveDown comma 
+InsertWholeNote 1 
+InsertHalfNote 2 
+InsertQuarterNote 3 
+InsertEighthNote 4 
+InsertSixteenthNote 5 
+InsertThirtysecondNote 6 
+InsertSixtyfourthNote 7 
+ToggleRestMode Escape 
+InsertWholeRest 1(4) 
+InsertHalfRest 2(4) 
+InsertQuarterRest 3(4) 
+InsertEighthRest 4(4) 
+InsertSixteenthRest 5(4) 
+InsertThirtysecondRest 6(4) 
+InsertSixtyfourthRest 7(4) 
+InsertArbitraryTuplet t
+Insert2/3Tuplet numbersign(1) 
+Insert4/5Tuplet percent(1) 
+Insert4/6Tuplet question(1) 
+Insert4/7Or8/7Tuplet ampersand(1) 
+Insert8/9Tuplet asterisk(1) 
+AddTone Return 
+RemoveTone Return(1) 
+Sharpen/StemDown equal 
+Flatten/StemUp minus 
+ChangeToWholeNote F1 
+ChangeToHalfNote F2 
+ChangeToQuarterNote F3 
+ChangeToEighthNote F4 
+ChangeToSixteenthNote F5 
+ChangeToThirtysecondNote F6 
+ChangeToSixtyfourthNote F7 
+AddDot . 
+RemoveDot quotedbl(1) 
+ToggleTie T(1) 
+DeleteObject Delete 
+DeletePreviousObject BackSpace 
+InsertMeasure m 
+AppendMeasure M(1) 
+DeleteMeasure Delete(4) 
+DeleteStaff Delete(1) 
+ShrinkMeasures Left(4) 
+WidenMeasures Right(4) 
+ShorterStaffs Up(4) 
+TallerStaffs Down(4) 
+SetMark space 
+UnsetMark space(4) 
+ToggleBeginSlur parenleft 
+ToggleEndSlur parenright 
+ToggleAccent exclam(1) 
+ToggleFermata Mode_switch 
+ToggleStaccato period(4) 
+ToggleTenuto underscore(1) 
+ToggleTrill ccedilla
+ToggleTurn ccedilla(4)
+ToggleMordent w 
+ToggleStaccatissimo semicolon 
\ No newline at end of file
diff -Naur denemo-0.6.0.orig/main.c denemo-0.6.0/main.c
--- denemo-0.6.0.orig/main.c	Tue May 28 03:58:56 2002
+++ denemo-0.6.0/main.c	Mon Aug 12 12:23:31 2002
@@ -5,7 +5,7 @@
  * (c) 1999, 2000, 2001 Matthew Hiller
  */
 
-#include <getopt.h>
+#include <gnugetopt/getopt.h>
 #include <signal.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -Naur denemo-0.6.0.orig/playback.c denemo-0.6.0/playback.c
--- denemo-0.6.0.orig/playback.c	Wed May  1 06:13:27 2002
+++ denemo-0.6.0/playback.c	Fri Aug  2 00:40:54 2002
@@ -20,7 +20,7 @@
 #include <unistd.h>
 #include <sys/types.h>
 #ifndef G_OS_WIN32
-# include <wait.h>
+# include <sys/wait.h>
 #endif
 #include <errno.h>
 
diff -Naur denemo-0.6.0.orig/print.c denemo-0.6.0/print.c
--- denemo-0.6.0.orig/print.c	Mon Nov 19 04:02:04 2001
+++ denemo-0.6.0/print.c	Fri Aug  2 00:40:54 2002
@@ -5,7 +5,7 @@
 #include <sys/types.h>
 
 #ifndef G_OS_WIN32
-#include <wait.h>
+#include <sys/wait.h>
 #endif
 
 #include <errno.h>

