diff -Nur clusterssh-3.22.orig/src/cssh.pl clusterssh-3.22/src/cssh.pl
--- clusterssh-3.22.orig/src/cssh.pl	2008-01-23 16:18:34.000000000 -0600
+++ clusterssh-3.22/src/cssh.pl	2008-08-09 21:33:41.000000000 -0500
@@ -531,7 +531,7 @@
 sub get_clusters() {
 
   # first, read in global file
-  my $cluster_file = '/etc/clusters';
+  my $cluster_file = $sysconfigdir . '/clusters';
 
   logmsg( 3, "Logging for $cluster_file" );
 
@@ -2065,7 +2065,7 @@
 
 =item <tag> ...
 
-Open a series of xterms defined by <tag> within either /etc/clusters or
+Open a series of xterms defined by <tag> within either SYSCONFIGDIR/clusters or
 F<$HOME/.csshrc> (see L<"FILES">).
 
 =back
@@ -2103,7 +2103,7 @@
 
 =over
 
-=item /etc/clusters
+=item SYSCONFIGDIR/clusters
 
 This file contains a list of tags to server names mappings.  When any name
 is used on the command line it is checked to see if it is a tag.
@@ -2124,7 +2124,7 @@
 options) or indirectly (see C<extra_cluster_file> configuration option) 
 in the users F<$HOME/.csshrc> file.
 
-=item F</etc/csshrc> & F<$HOME/.csshrc>
+=item F<SYSCONFIGDIR/csshrc> & F<$HOME/.csshrc>
 
 This file contains configuration overrides - the defaults are as marked.
 Default options are overwritten first by the global file, and then by the
@@ -2153,14 +2153,14 @@
 =item clusters = <blank>
 
 Define a number of cluster tags in addition to (or to replace) tags defined
-in the F</etc/clusters> file.  The format is:
+in the F<SYSCONFIGDIR/clusters> file.  The format is:
 
  clusters = <tag1> <tag2> <tag3>
  <tag1> = host1 host2 host3
  <tag2> = user@host4 user@host5 host6
  <tag3> = <tag1> <tag2>
 
-As with the F</etc/clusters> file, be sure not to create recursivly nested tags.
+As with the F<SYSCONFIGDIR/clusters> file, be sure not to create recursivly nested tags.
 
 =item comms = ssh
 
@@ -2175,7 +2175,7 @@
 
 =item extra_cluster_file = <null>
 
-Define an extra cluster file in the format of F</etc/clusters>.  Multiple
+Define an extra cluster file in the format of F<SYSCONFIGDIR/clusters>.  Multiple
 files can be specified, seperated by commas.  Both ~ and $HOME are acceptable
 as a to reference the users home directory, i.e.
 
@@ -2377,7 +2377,7 @@
 
 C<< perl -MX11::Protocol -e 'print $X11::Protocol::VERSION,$/' >>
 
-C<< cat /etc/csshrc $HOME/.csshrc >>
+C<< cat SYSCONFIGDIR/csshrc $HOME/.csshrc >>
 
 =item *
 
diff -Nur clusterssh-3.22.orig/src/Makefile.in clusterssh-3.22/src/Makefile.in
--- clusterssh-3.22.orig/src/Makefile.in	2008-01-23 16:18:53.000000000 -0600
+++ clusterssh-3.22/src/Makefile.in	2008-08-09 21:34:11.000000000 -0500
@@ -373,15 +373,17 @@
 
 
 cssh: $(srcdir)/cssh.pl
-	rm -f cssh
-	echo "#!"$(PERL) > cssh
-	sed 's!$$sysconfigdir = "/etc";!$$sysconfigdir = "$(sysconfdir)";!' $(srcdir)/cssh.pl >> cssh
+	rm -f cssh.tmp cssh
+	echo "#!"$(PERL) > cssh.tmp
+	sed 's!$$sysconfigdir = "/etc";!$$sysconfigdir = "$(sysconfdir)";!' $(srcdir)/cssh.pl >> cssh.tmp
+	sed 's!SYSCONFIGDIR!$(sysconfdir)!' cssh.tmp > cssh
+	rm -f cssh.tmp
 	chmod ugo+x cssh
 
-cssh.1: $(srcdir)/cssh.pl
+cssh.1: cssh
 	rm -f cssh.1
-	podchecker $(srcdir)/cssh.pl
-	pod2man $(srcdir)/cssh.pl > cssh.1
+	podchecker cssh
+	pod2man cssh > cssh.1
 	chmod 444 cssh.1
 
 #

 	  	 
