diff -Nurd -x'*~' lesstif-0.95.0.orig/clients/Motif-2.1/mwm/Makefile.in lesstif-0.95.0/clients/Motif-2.1/mwm/Makefile.in
--- lesstif-0.95.0.orig/clients/Motif-2.1/mwm/Makefile.in	2006-06-10 05:35:34.000000000 -0400
+++ lesstif-0.95.0/clients/Motif-2.1/mwm/Makefile.in	2008-06-08 03:35:38.000000000 -0400
@@ -300,7 +300,7 @@
 LDADD = ../../../lib/Xm-2.1/libXm.la \
 		$(X_LIBS) $(XPLIB) -lXt $(X_PRE_LIBS) -lXext -lX11 $(X_EXTRA_LIBS)
 
-appdir = $(libdir)/X11/app-defaults
+appdir = $(prefix)/etc/app-defaults
 mwmddir = $(libdir)/X11/mwm
 mwmd_DATA = system.mwmrc alt.map README
 AM_YFLAGS = -d
diff -Nurd -x'*~' lesstif-0.95.0.orig/configure lesstif-0.95.0/configure
--- lesstif-0.95.0.orig/configure	2006-06-10 05:35:28.000000000 -0400
+++ lesstif-0.95.0/configure	2008-09-04 00:26:41.000000000 -0400
@@ -2150,7 +2150,7 @@
 
 echo "$as_me:$LINENO: checking where to store autoconf macro files" >&5
 echo $ECHO_N "checking where to store autoconf macro files... $ECHO_C" >&6
-ACLOCALDIR=`aclocal --print-ac-dir`
+ACLOCALDIR=@PREFIX@/share/aclocal
 
 
 if test x$ACLOCALDIR != x ; then
@@ -28766,9 +28766,9 @@
 esac
 
 saved_LIBS="$LIBS"
-LIBS="$LIBS $freetype_lib"
+LIBS="$freetype_lib $LIBS"
 saved_CPPFLAGS="$CPPFLAGS"
-CPPFLAGS="$CPPFLAGS $FREETYPE_CFLAGS"
+CPPFLAGS="$FREETYPE_CFLAGS $CPPFLAGS"
 
 for ac_header in freetype/freetype.h
 do
