--- Mesa-6.5.1/configs/darwin.orig	2008-02-26 21:47:32.000000000 -0500
+++ Mesa-6.5.1/configs/darwin	2008-02-26 21:49:02.000000000 -0500
@@ -7,6 +7,12 @@
 # Compiler and flags
 CC = cc
 CXX = cc
+
+COPY_LIBS = cp -f
+
+MKDEP = gcc -M -MG
+MKDEP_OPTIONS = -MF depend
+
 CFLAGS = -I/usr/X11R6/include -O3 -fPIC -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -no-cpp-precomp -dynamic -Ddarwin
 CXXFLAGS = -I/usr/X11R6/include -O3 -fPIC -fno-common -ffast-math -funroll-loops -fexpensive-optimizations -no-cpp-precomp -dynamic -Ddarwin
 
@@ -25,9 +32,6 @@
 OSMESA_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -lGL
 GLU_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -lGL
 GLUT_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -lGL -lGLU -L/usr/X11R6/lib -lX11 -lXmu -lXi -lXext
-GLW_LIB_DEPS = -L/usr/X11R6/lib -lX11 -lXt $(TOP)/lib/GL.dylib
+GLW_LIB_DEPS = -L/usr/X11R6/lib -lX11 -lXt -l$(GL_LIB) /System/Library/Frameworks/OpenGL.framework/Libraries/libGL.dylib -L/@FINKPREFIX@/lib
 APP_LIB_DEPS = -L$(TOP)/lib -l$(GLUT_LIB) -l$(GLU_LIB) -l$(GL_LIB) -L/usr/X11R6/lib -lX11 -lXmu -lXt -lXi -lm
 
-# omit glw lib for now:
-SRC_DIRS = mesa glu glut/glx
-
--- Mesa-6.5.1/bin/mklib.org	2008-02-27 00:02:50.000000000 -0500
+++ Mesa-6.5.1/bin/mklib	2008-02-27 00:06:25.000000000 -0500
@@ -584,7 +584,7 @@
                 OPTS="${ARCHOPT} -bundle -multiply_defined suppress"
             else
 		LIBSUFFIX="dylib"
-                OPTS="${ARCHOPT} -dynamiclib -multiply_defined suppress -current_version ${MAJOR}.${MINOR}.0 -compatibility_version ${MAJOR}.${MINOR}.0 -install_name lib${LIBNAME}.${MAJOR}.${LIBSUFFIX}"
+                OPTS="${ARCHOPT} -dynamiclib -multiply_defined suppress -current_version ${MAJOR}.${MINOR}.0 -compatibility_version ${MAJOR}.${MINOR}.0 -install_name @FINKPREFIX@/lib/lib${LIBNAME}.${MAJOR}.${LIBSUFFIX}"
             fi
             LINKNAME="lib${LIBNAME}.${LIBSUFFIX}"
             LIBNAME="lib${LIBNAME}.${MAJOR}.${LIBSUFFIX}"
--- Mesa-7.0.2/configs/default.orig	2007-11-01 11:30:52.000000000 -0400
+++ Mesa-7.0.2/configs/default	2008-02-27 14:54:21.000000000 -0500
@@ -55,12 +55,12 @@
 
 # GLw widget sources (Append "GLwMDrawA.c" here and add -lXm to GLW_LIB_DEPS in
 # order to build the Motif widget too)
-GLW_SOURCES = GLwDrawA.c
+GLW_SOURCES = GLwDrawA.c GLwMDrawA.c
 
 
 # Directories to build
 LIB_DIR = lib
-SRC_DIRS = mesa glu glut/glx glw
+SRC_DIRS = glw
 GLU_DIRS = sgi
 DRIVER_DIRS = x11 osmesa
 # Which subdirs under $(TOP)/progs/ to enter:
@@ -73,7 +73,7 @@
 OSMESA_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -l$(GL_LIB)
 GLU_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -l$(GL_LIB) -lm
 GLUT_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -l$(GLU_LIB) -l$(GL_LIB) $(EXTRA_LIB_PATH) -lX11 -lXmu -lXt -lXi -lm
-GLW_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -l$(GL_LIB) $(EXTRA_LIB_PATH) -lXt -lX11
+GLW_LIB_DEPS = -l$(GL_LIB) -lXt -lX11
 
 # Program dependencies - specific GL/glut libraries added in Makefiles
 APP_LIB_DEPS = -lm
--- Mesa-7.0.2/src/glw/Makefile.orig	2007-09-28 20:43:28.000000000 -0400
+++ Mesa-7.0.2/src/glw/Makefile	2008-02-27 15:13:25.000000000 -0500
@@ -7,7 +7,7 @@
 MINOR = 0
 TINY = 0
 
-INCDIRS = -I$(TOP)/include -I/usr/include/Motif1.2 $(X11_INCLUDES)
+INCDIRS = -I$(TOP)/include -I@FINKPREFIX@/include $(X11_INCLUDES)
 
 
 OBJECTS = $(GLW_SOURCES:.c=.o)
@@ -33,13 +33,11 @@
 glw.pc: glw.pc.in
 	$(pcedit) $< > $@
 
-install: glw.pc
-	$(INSTALL) -d $(DESTDIR)$(INSTALL_DIR)/include/GL
-	$(INSTALL) -d $(DESTDIR)$(INSTALL_DIR)/$(LIB_DIR)
-	$(INSTALL) -d $(DESTDIR)$(INSTALL_DIR)/$(LIB_DIR)/pkgconfig
-	$(INSTALL) -m 644 *.h $(DESTDIR)$(INSTALL_DIR)/include/GL
-	$(INSTALL) $(TOP)/$(LIB_DIR)/libGLw.* $(DESTDIR)$(INSTALL_DIR)/$(LIB_DIR)
-	$(INSTALL) -m 644 glw.pc $(DESTDIR)$(INSTALL_DIR)/$(LIB_DIR)/pkgconfig
+install: 
+	$(INSTALL) -d $(DESTDIR)$(INSTALL_DIR)@FINKPREFIX@/include/GL
+	$(INSTALL) -d $(DESTDIR)$(INSTALL_DIR)@FINKPREFIX@/$(LIB_DIR)
+	$(INSTALL) -m 644 *.h $(DESTDIR)$(INSTALL_DIR)@FINKPREFIX@/include/GL
+	$(INSTALL) $(TOP)/$(LIB_DIR)/libGLw.1.* $(DESTDIR)$(INSTALL_DIR)@FINKPREFIX@/$(LIB_DIR)
 
 clean:
 	-rm depend depend.bak
