--- openmotif-2.3.0.orig/Makefile.in	2007-08-09 03:27:57.000000000 -0400
+++ openmotif-2.3.0/Makefile.in	2008-04-25 22:31:30.000000000 -0400
@@ -249,8 +249,7 @@
              include \
              tools \
              clients \
-             doc \
-             demos
+             doc
 
 AUTOMAKE_OPTIONS = 1.4
 all: all-recursive
--- openmotif-2.3.0.orig/tools/wml/Makefile.in	2007-08-09 03:27:56.000000000 -0400
+++ openmotif-2.3.0/tools/wml/Makefile.in	2008-04-25 22:34:46.000000000 -0400
@@ -270,7 +270,7 @@
 wmluiltok_SOURCES = wmluiltok.l
 wmluiltok_LDADD = @LEXLIB@
 wml_SOURCES = wml.c
-wml_LDADD = -L. -lwml
+wml_LDADD = libwml.a
 wmldbcreate_LDADD = ../../lib/Xm/libXm.la
 INCLUDES = -I../../lib -I$(srcdir)/../../lib -I$(srcdir)/../../include ${X_CFLAGS} 
 noinst_LIBRARIES = libwml.a
