--- xscreensaver-5.10/configure	2009-09-08 06:54:40.000000000 +0200
+++ xscreensaver-5.10_corr/configure	2010-01-31 12:24:52.000000000 +0100
@@ -3534,8 +3534,6 @@
    ac_gcc_accepts_std="$ac_cv_gcc_accepts_std"
   fi
 
-   ;;
-  esac
 
    { echo "$as_me:$LINENO: result: Disabling C++ comments in ANSI C code." >&5
 echo "${ECHO_T}Disabling C++ comments in ANSI C code." >&6; }
@@ -3564,7 +3562,11 @@
      # The old way:
      CC="$CC -Wp,-lang-c89"
    fi
-  fi
+
+   ;;
+  esac
+
+fi
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
