diff -Nurd -x'*~' Alien-SDL-1.425.orig/Build.PL Alien-SDL-1.425/Build.PL
--- Alien-SDL-1.425.orig/Build.PL	2011-01-26 16:10:46.000000000 -0500
+++ Alien-SDL-1.425/Build.PL	2011-04-14 00:42:12.000000000 -0400
@@ -111,13 +111,7 @@
     push @candidates, $rv;
   };
 
-  if($rv=check_prebuilt_binaries($build->os_type)) {
-    push @candidates, @{$rv};
-  };
 
-  if($build->can_build_binaries_from_sources() && ($rv=check_src_build($build->os_type))) {
-    push @candidates, @{$rv};
-  };
 
   push @candidates, { title => 'Quit installation' };
 
