diff -ruN Crypt-SSLeay-0.57/Makefile.PL Crypt-SSLeay-0.57-patched/Makefile.PL
--- Crypt-SSLeay-0.57/Makefile.PL	2007-09-17 16:36:59.000000000 -0400
+++ Crypt-SSLeay-0.57-patched/Makefile.PL	2009-03-13 13:42:10.000000000 -0400
@@ -17,7 +17,7 @@
     "static",  \$opt_static,
 );
 
-$opt_default ||= $ENV{CRYPT_SSLEAY_DEFAULT};
+$opt_default = 1;
 
 # FIND POSSIBLE SSL INSTALLATIONS
 
@@ -35,15 +35,7 @@
 }
 else {
     # Unix and the rest
-    @POSSIBLE_SSL_DIRS = qw(
-        /local
-        /local/ssl
-        /opt/ssl
-        /usr
-        /usr/local
-        /usr/local/ssl
-        /usr/local/openssl
-    );
+    @POSSIBLE_SSL_DIRS = qw(@PREFIX@/lib/system-openssl);
 }
 
 my @CANDIDATE = do {
@@ -267,28 +259,6 @@
     @license,
 );
 
-if (open OUT, '> test.config') {
-    print OUT <<"INFO";
-ssl $pkg_config->{type} $pkg_config->{ver} in $SSL_DIR
-lib @LIB_FLAGS
-inc @INC_FLAGS
-cc $Config{cc}
-INFO
-
-    print <<"INFO";
-The test suite can attempt to connect to public servers
-to ensure that the code is working properly. If you are
-behind a strict firewall or have no network connectivity,
-these tests may fail (through no fault of the code).
-
-INFO
-    my $network_tests = prompt
-        "Do you want to run the live tests (y/N) ?",
-        'N';
-    print OUT "network_tests ", ($network_tests =~ /y/i) ? 1 : 0, "\n";
-    close OUT;
-}
-
 ## HELPERS
 
 sub Candidate {
