diff -ruN xfce4-iconbox-4.2.0/configure xfce4-iconbox-4.2.0-mod/configure
--- xfce4-iconbox-4.2.0/configure	Wed Jan  5 16:19:12 2005
+++ xfce4-iconbox-4.2.0-mod/configure	Thu Jan 20 12:17:17 2005
@@ -3894,11 +3894,11 @@
 
 
 if test "`uname | grep \"CYGWIN\"`" != ""; then
-  HAVE_CYGWIN_TRUE=
-  HAVE_CYGWIN_FALSE='#'
-else
   HAVE_CYGWIN_TRUE='#'
   HAVE_CYGWIN_FALSE=
+else
+  HAVE_CYGWIN_TRUE=
+  HAVE_CYGWIN_FALSE='#'
 fi
 
 
diff -ruN xfce4-iconbox-4.2.0/settings/Makefile.in xfce4-iconbox-4.2.0-mod/settings/Makefile.in
--- xfce4-iconbox-4.2.0/settings/Makefile.in	Wed Dec 22 14:54:27 2004
+++ xfce4-iconbox-4.2.0-mod/settings/Makefile.in	Thu Jan 20 12:18:23 2005
@@ -40,8 +40,8 @@
 build_triplet = @build@
 host_triplet = @host@
 @HAVE_CYGWIN_TRUE@am__append_1 = \
-@HAVE_CYGWIN_TRUE@	-no-undefined							\
-@HAVE_CYGWIN_TRUE@	-export-symbols $(datadir)/xfce4/devel/mcs-manager.def		\
+@HAVE_CYGWIN_FALSE@	-no-undefined							\
+@HAVE_CYGWIN_FALSE@	-export-symbols $(datadir)/xfce4/devel/mcs-manager.def		\
 @HAVE_CYGWIN_TRUE@	@XFCE_MCS_MANAGER_LDFLAGS@
 
 subdir = settings
