--- xc/include/Xos_r.h.orig	Sun Jan 20 13:06:48 2002
+++ xc/include/Xos_r.h	Sun Jan 20 12:55:07 2002
@@ -254,7 +254,7 @@
  * fields.
  */
    
-#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__)
+#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
 static __inline__ void _Xpw_copyPasswd(_Xgetpwparams p)
 {
    memcpy(&(p).pws, (p).pwp, sizeof(struct passwd));
