diff -ruN boost_1_46_1/tools/build/v2/tools/darwin.jam boost_1_46_1_corr/tools/build/v2/tools/darwin.jam
--- boost_1_46_1/tools/build/v2/tools/darwin.jam	2010-07-12 03:37:43.000000000 -0400
+++ boost_1_46_1_corr/tools/build/v2/tools/darwin.jam	2011-06-06 11:18:25.000000000 -0400
@@ -20,6 +20,8 @@
 import property-set ;
 import regex ;
 import errors ;
+import os ; 
+import option ; 
 
 ## Use a framework.
 feature framework : : free ;
@@ -536,11 +538,16 @@
 {
     setup-address-model $(targets) : $(sources) : $(properties) ;
     prepare-framework-path $(<) ;
+    local prefix = [ option.get prefix ] ; 
+    LIB_INSTALL_DIR = [ option.get libdir : $(prefix)/lib ] ;
+    VERSIONINFO_COMP = [ option.get compatibility_version ] ; 
+    VERSIONINFO_CUR = [ option.get current_version ] ; 
+    UNDEFINED = [ option.get undefined ] ;
 }
 
 actions link.dll bind LIBRARIES
 {
-    "$(CONFIG_COMMAND)" -dynamiclib -Wl,-single_module -install_name "$(<:B)$(<:S)" -L"$(LINKPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -l$(FINDLIBS-SA) -l$(FINDLIBS-ST) $(FRAMEWORK_PATH) -framework$(_)$(FRAMEWORK:D=:S=) $(OPTIONS) $(USER_OPTIONS)
+    "$(CONFIG_COMMAND)" -dynamiclib $(UNDEFINED) -Wl,-single_module -install_name "$(LIB_INSTALL_DIR)/$(<:B)-1_@BOOST_MINOR@$(<:S)" -compatibility_version $(VERSIONINFO_COMP) -current_version $(VERSIONINFO_CUR) -L"$(LINKPATH)" -o "$(<)" "$(>)" "$(LIBRARIES)" -l$(FINDLIBS-SA) -l$(FINDLIBS-ST) $(FRAMEWORK_PATH) -framework$(_)$(FRAMEWORK:D=:S=) $(OPTIONS) $(USER_OPTIONS)
 }
 
 # We use libtool instead of ar to support universal binary linking
