diff -Nurd attica-0.1.1/lib/CMakeLists.txt attica-0.1.1-new/lib/CMakeLists.txt
--- attica-0.1.1/lib/CMakeLists.txt	2009-12-08 07:49:32.000000000 -0500
+++ attica-0.1.1-new/lib/CMakeLists.txt	2010-01-12 12:47:16.000000000 -0500
@@ -7,6 +7,12 @@
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH  TRUE)
 set(CMAKE_INSTALL_RPATH "${LIB_DESTINATION}" )
 
+if(APPLE)
+   set(CPACK_SET_DESTDIR ON)
+   set(CPACK_PACKAGE_RELOCATABLE OFF)
+   set(CMAKE_INSTALL_NAME_DIR ${LIB_DESTINATION})
+endif(APPLE)
+
 find_package(Qt4 REQUIRED)
 include_directories( ${QT_INCLUDES} ${CMAKE_CURRENT_SOURCE_DIR} )
 
