diff -uNr libgpod-0.3.2/src/db-parse-context.c libgpod-0.3.2-new/src/db-parse-context.c
--- libgpod-0.3.2/src/db-parse-context.c	Thu Nov 24 06:31:49 2005
+++ libgpod-0.3.2-new/src/db-parse-context.c	Sat Sep  9 15:15:16 2006
@@ -23,9 +23,9 @@
  */
 
 
+#include <sys/types.h>
 #include <sys/mman.h>
 #include <sys/stat.h>
-#include <sys/types.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <string.h>
diff -uNr libgpod-0.3.2/src/itdb_itunesdb.c libgpod-0.3.2-new/src/itdb_itunesdb.c
--- libgpod-0.3.2/src/itdb_itunesdb.c	Sun Feb 19 05:14:33 2006
+++ libgpod-0.3.2-new/src/itdb_itunesdb.c	Sat Sep  9 15:15:02 2006
@@ -3861,15 +3861,8 @@
     return result;
 }
 
-
-/* Do the actual writing to the iTunesSD */
-/* If @filename cannot be NULL */
-gboolean itdb_shuffle_write_file (Itdb_iTunesDB *itdb,
-				  const gchar *filename, GError **error)
+gboolean haystack (gchar *filetype, gchar **desclist)
 {
-    auto gboolean haystack (gchar *filetype, gchar **desclist);
-    gboolean haystack (gchar *filetype, gchar **desclist)
-    {
 	gchar **dlp;
 	if (!filetype || !desclist) return FALSE;
 	for (dlp=desclist; *dlp; ++dlp)
@@ -3877,8 +3870,14 @@
 	    if (strstr (filetype, *dlp)) return TRUE;
 	}
 	return FALSE;
-    }
+}
+
 
+/* Do the actual writing to the iTunesSD */
+/* If @filename cannot be NULL */
+gboolean itdb_shuffle_write_file (Itdb_iTunesDB *itdb,
+				  const gchar *filename, GError **error)
+{
     FExport *fexp;
     GList *gl;
     WContents *cts;
diff -uNr libgpod-0.3.2/src/itdb_track.c libgpod-0.3.2-new/src/itdb_track.c
--- libgpod-0.3.2/src/itdb_track.c	Sun Dec  4 05:26:40 2005
+++ libgpod-0.3.2-new/src/itdb_track.c	Sat Sep  9 15:15:02 2006
@@ -56,20 +56,11 @@
     return track;
 }
 
+extern gboolean haystack (gchar *filetype, gchar **desclist);
+
 /* Attempt to set some of the unknowns to reasonable defaults */
 static void itdb_track_set_defaults (Itdb_Track *tr)
 {
-    auto gboolean haystack (gchar *filetype, gchar **desclist);
-    gboolean haystack (gchar *filetype, gchar **desclist)
-    {
-	gchar **dlp;
-	if (!filetype || !desclist) return FALSE;
-	for (dlp=desclist; *dlp; ++dlp)
-	{
-	    if (strstr (filetype, *dlp)) return TRUE;
-	}
-	return FALSE;
-    }
 
     gchar *mp3_desc[] = {"MPEG", "MP3", "mpeg", "mp3", NULL};
     gchar *mp4_desc[] = {"AAC", "MP4", "aac", "mp4", NULL};
