diff -Nurd libircclient-qt-0.3.2/icu.pri libircclient-qt-0.3.2-new/icu.pri
--- libircclient-qt-0.3.2/icu.pri	2009-08-30 11:35:41.000000000 +0200
+++ libircclient-qt-0.3.2-new/icu.pri	2010-04-30 03:20:28.000000000 +0200
@@ -15,7 +15,7 @@
     return($$RET)
 }
 
-DEFINES += HAVE_ICU
+# DEFINES += HAVE_ICU
 contains(DEFINES, HAVE_ICU) {
     win32:INCLUDEPATH += C:\ICU\include
     win32:LIBS += -LC:\ICU\lib
diff -Nurd libircclient-qt-0.3.2/libircclient-qt.pro libircclient-qt-0.3.2-new/libircclient-qt.pro
--- libircclient-qt-0.3.2/libircclient-qt.pro	2010-04-30 14:08:21.000000000 +0200
+++ libircclient-qt-0.3.2-new/libircclient-qt.pro	2010-04-30 14:09:03.000000000 +0200
@@ -21,11 +21,11 @@
 }
 
 macx {
-    CONFIG += lib_bundle
-    FRAMEWORK_HEADERS.version = Versions
-    FRAMEWORK_HEADERS.files = $$HEADERS
-    FRAMEWORK_HEADERS.path = Headers
-    QMAKE_BUNDLE_DATA += FRAMEWORK_HEADERS
+    # CONFIG += lib_bundle
+    # FRAMEWORK_HEADERS.version = Versions
+    # FRAMEWORK_HEADERS.files = $$HEADERS
+    # FRAMEWORK_HEADERS.path = Headers
+    # QMAKE_BUNDLE_DATA += FRAMEWORK_HEADERS
 }
 
 CONV_HEADERS += include/Irc include/IrcGlobal include/IrcSession include/IrcUtil
