diff -ru libuninameslist/Makefile.in libuninameslist.patched/Makefile.in
--- libuninameslist/Makefile.in	Fri Jul 11 00:36:26 2003
+++ libuninameslist.patched/Makefile.in	Sat Feb 21 13:21:00 2004
@@ -21,7 +21,7 @@
 CFLAGS = @CFLAGS@ $(WFLAGS)
 
 libuninameslist.la: nameslist.lo
-	$(LIBTOOL) $(CC) $(CFLAGS) -o libuninameslist.la nameslist.lo \
+	$(LIBTOOL) $(CC) $(CFLAGS) -no-undefined -o libuninameslist.la nameslist.lo \
 	    -rpath $(libdir) -version-info $(UN_VERSION):$(UN_REVISION):$(UN_AGE)
 
 
