diff -Nurd -x'*~' tclxml-3.1.orig/Makefile.in tclxml-3.1/Makefile.in
--- tclxml-3.1.orig/Makefile.in	2005-05-20 08:02:01.000000000 -0400
+++ tclxml-3.1/Makefile.in	2008-08-28 00:27:48.000000000 -0400
@@ -123,6 +123,7 @@
 pkgdatadir	= $(datadir)/$(PKG_DIR)
 pkglibdir	= $(libdir)/$(PKG_DIR)
 pkgincludedir	= $(includedir)/$(PKG_DIR)
+LIB_RUNTIME_DIR = $(pkglibdir)
 
 top_builddir	= .
 
diff -Nurd -x'*~' tclxml-3.1.orig/configure tclxml-3.1/configure
--- tclxml-3.1.orig/configure	2005-11-04 01:26:33.000000000 -0500
+++ tclxml-3.1/configure	2008-08-28 00:29:17.000000000 -0400
@@ -4573,13 +4573,13 @@
 	Rhapsody-*|Darwin-*)
 	    SHLIB_CFLAGS="-fno-common"
 	    SHLIB_LD="cc -dynamiclib \${LDFLAGS}"
-	    TCL_SHLIB_LD_EXTRAS="-compatibility_version ${TCL_MAJOR_VERSION} -current_version \${VERSION} -install_name \${LIB_RUNTIME_DIR}/\${TCL_LIB_FILE} -prebind -seg1addr 0xa000000"
+	    TCL_SHLIB_LD_EXTRAS="-compatibility_version ${TCL_MAJOR_VERSION} -current_version \${VERSION} -install_name \${LIB_RUNTIME_DIR}/\$@ -prebind -seg1addr 0xa000000"
 	    SHLIB_LD_LIBS='${LIBS}'
 	    SHLIB_SUFFIX=".dylib"
 	    DL_OBJS="tclLoadDyld.o"
 	    DL_LIBS=""
 	    LDFLAGS="-prebind"
-	    LD_SEARCH_FLAGS=""
+	    LD_SEARCH_FLAGS="${TCL_SHLIB_LD_EXTRAS}"
 	    CFLAGS_OPTIMIZE="-O3"
 	    EXTRA_CFLAGS="-arch ppc -pipe"
 	    ;;
@@ -5086,7 +5086,7 @@
     
     
 
-    SHLIB_LDFLAGS='$(LDFLAGS_DEFAULT)'
+    SHLIB_LDFLAGS="\$(LDFLAGS_DEFAULT) ${LD_SEARCH_FLAGS}"
     
     
     
