--- xml-batik/build.sh	2001-10-23 15:42:28.000000000 +0200
+++ xml-batik/build.sh	2003-03-05 18:38:06.000000000 +0100
@@ -18,7 +18,7 @@
 
 # ----- Set Up The Runtime Classpath ------------------------------------------
 
-CP=$JAVA_HOME/lib/tools.jar:$ANT_HOME/lib/build/ant_1_4_1.jar:./lib/build/crimson-ant.jar:./lib/build/jaxp.jar
+CP=@PREFIX@/lib/ant/lib/ant.jar:@PREFIX@/lib/ant/lib/xml-apis.jar
  
 # ----- Execute The Requested Build -------------------------------------------
 
