diff -uNrb commons-pool-old/scripts/commons-pool.csh commons-pool/scripts/commons-pool.csh
--- commons-pool-old/scripts/commons-pool.csh	Wed Dec 31 19:00:00 1969
+++ commons-pool/scripts/commons-pool.csh	Sat Jan  5 18:46:09 2002
@@ -0,0 +1,8 @@
+#!/bin/csh
+set chemin = "@PREFIX@/share/java/commons-pool/commons-pool.jar"
+if ($?CLASSPATH) then
+	setenv CLASSPATH "${CLASSPATH}:$chemin"
+else
+	setenv CLASSPATH "$chemin"
+endif
+unset chemin
diff -uNrb commons-pool-old/scripts/commons-pool.sh commons-pool/scripts/commons-pool.sh
--- commons-pool-old/scripts/commons-pool.sh	Wed Dec 31 19:00:00 1969
+++ commons-pool/scripts/commons-pool.sh	Sat Jan  5 18:46:20 2002
@@ -0,0 +1,9 @@
+#!/bin/sh
+chemin="@PREFIX@/share/java/commons-pool/commons-pool.jar"
+if (test "$CLASSPATH")
+then
+	export  CLASSPATH="${CLASSPATH}:$chemin"
+else
+	export  CLASSPATH="$chemin"
+fi
+
