diff -uNrb commons-lang-old/scripts/commons-lang.csh commons-lang/scripts/commons-lang.csh
--- commons-lang-old/scripts/commons-lang.csh	Wed Dec 31 19:00:00 1969
+++ commons-lang/scripts/commons-lang.csh	Sat Jan  5 18:46:09 2002
@@ -0,0 +1,8 @@
+#!/bin/csh
+set chemin = "@PREFIX@/share/java/commons-lang/commons-lang.jar"
+if ($?CLASSPATH) then
+	setenv CLASSPATH "${CLASSPATH}:$chemin"
+else
+	setenv CLASSPATH "$chemin"
+endif
+unset chemin
diff -uNrb commons-lang-old/scripts/commons-lang.sh commons-lang/scripts/commons-lang.sh
--- commons-lang-old/scripts/commons-lang.sh	Wed Dec 31 19:00:00 1969
+++ commons-lang/scripts/commons-lang.sh	Sat Jan  5 18:46:20 2002
@@ -0,0 +1,9 @@
+#!/bin/sh
+chemin="@PREFIX@/share/java/commons-lang/commons-lang.jar"
+if (test "$CLASSPATH")
+then
+	export  CLASSPATH="${CLASSPATH}:$chemin"
+else
+	export  CLASSPATH="$chemin"
+fi
+
