diff -uNrb commons-fileupload-old/scripts/commons-fileupload.csh commons-fileupload/scripts/commons-fileupload.csh
--- commons-fileupload-old/scripts/commons-fileupload.csh	Wed Dec 31 19:00:00 1969
+++ commons-fileupload/scripts/commons-fileupload.csh	Sat Jan  5 18:46:09 2002
@@ -0,0 +1,8 @@
+#!/bin/csh
+set chemin = "@PREFIX@/share/java/commons-fileupload/commons-fileupload.jar"
+if ($?CLASSPATH) then
+	setenv CLASSPATH "${CLASSPATH}:$chemin"
+else
+	setenv CLASSPATH "$chemin"
+endif
+unset chemin
diff -uNrb commons-fileupload-old/scripts/commons-fileupload.sh commons-fileupload/scripts/commons-fileupload.sh
--- commons-fileupload-old/scripts/commons-fileupload.sh	Wed Dec 31 19:00:00 1969
+++ commons-fileupload/scripts/commons-fileupload.sh	Sat Jan  5 18:46:20 2002
@@ -0,0 +1,9 @@
+#!/bin/sh
+chemin="@PREFIX@/share/java/commons-fileupload/commons-fileupload.jar"
+if (test "$CLASSPATH")
+then
+	export  CLASSPATH="${CLASSPATH}:$chemin"
+else
+	export  CLASSPATH="$chemin"
+fi
+
