diff -Naur dlcompat-20020709.old/dlfcn.h dlcompat-20020709/dlfcn.h
--- dlcompat-20020709.old/dlfcn.h	Tue Jul  2 11:21:05 2002
+++ dlcompat-20020709/dlfcn.h	Tue Jul 30 12:21:59 2002
@@ -24,6 +24,8 @@
 OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
+#ifndef _DLFCN_H_
+#define _DLFCN_H_
 
 #ifdef __cplusplus
 extern "C" {
@@ -58,4 +60,6 @@
 
 #ifdef __cplusplus
 }
+#endif
+
 #endif
