diff -Nru gdal-1.1.5.orig/configure gdal-1.1.5/configure
--- gdal-1.1.5.orig/configure	Fri Sep 14 12:50:51 2001
+++ gdal-1.1.5/configure	Sat Jan 19 00:25:53 2002
@@ -947,9 +947,9 @@
 fi
 
 
-LIBS="$LIBS -L/usr/local/lib"
-CFLAGS="$CFLAGS -I/usr/local/include"
-CPPFLAGS="$CPPFLAGS -I/usr/local/include"
+LIBS="$LIBS $LDFLAGS"
+CFLAGS="$CFLAGS"
+CPPFLAGS="$CPPFLAGS"
 
 echo $ac_n "checking for dlopen in -ldl""... $ac_c" 1>&6
 echo "configure:956: checking for dlopen in -ldl" >&5
