diff -uNbr jaf-1.0.1/scripts/jaf.csh jaf-1.0.1-new/scripts/jaf.csh
--- jaf-1.0.1/scripts/jaf.csh	Wed Dec 31 19:00:00 1969
+++ jaf-1.0.1-new/scripts/jaf.csh	Sat Jan  5 19:29:48 2002
@@ -0,0 +1,8 @@
+#!/bin/csh
+set chemin = "@PREFIX@/share/java/jaf/activation.jar"
+if ($?CLASSPATH) then
+	setenv CLASSPATH "${CLASSPATH}:$chemin"
+else
+	setenv CLASSPATH "$chemin"
+endif
+unset chemin
diff -uNbr jaf-1.0.1/scripts/jaf.sh jaf-1.0.1-new/scripts/jaf.sh
--- jaf-1.0.1/scripts/jaf.sh	Wed Dec 31 19:00:00 1969
+++ jaf-1.0.1-new/scripts/jaf.sh	Sat Jan  5 19:30:07 2002
@@ -0,0 +1,9 @@
+#!/bin/sh
+chemin="@PREFIX@/share/java/jaf/activation.jar"
+if (test "$CLASSPATH")
+then
+	export  CLASSPATH="${CLASSPATH}:$chemin"
+else
+	export  CLASSPATH="$chemin"
+fi
+
