diff -uNbr jakarta-servletapi-4/src/scripts/jakarta-servletapi.csh jakarta-servletapi-4-new/src/scripts/jakarta-servletapi.csh
--- jakarta-servletapi-4/src/scripts/jakarta-servletapi.csh	Wed Dec 31 19:00:00 1969
+++ jakarta-servletapi-4-new/src/scripts/jakarta-servletapi.csh	Sat Jan  5 15:43:25 2002
@@ -0,0 +1,8 @@
+#!/bin/csh
+set chemin = "@PREFIX@/share/java/jakarta-servletapi/servlet.jar"
+if ($?CLASSPATH) then
+	setenv CLASSPATH "${CLASSPATH}:$chemin"
+else
+	setenv CLASSPATH "$chemin"
+endif
+unset chemin
diff -uNbr jakarta-servletapi-4/src/scripts/jakarta-servletapi.sh jakarta-servletapi-4-new/src/scripts/jakarta-servletapi.sh
--- jakarta-servletapi-4/src/scripts/jakarta-servletapi.sh	Wed Dec 31 19:00:00 1969
+++ jakarta-servletapi-4-new/src/scripts/jakarta-servletapi.sh	Sat Jan  5 15:43:17 2002
@@ -0,0 +1,9 @@
+#!/bin/sh
+chemin="@PREFIX@/share/java/jakarta-servletapi/servlet.jar"
+if (test "$CLASSPATH")
+then
+	export  CLASSPATH="${CLASSPATH}:$chemin"
+else
+	export  CLASSPATH="$chemin"
+fi
+
