diff -uNrb jakarta-oro-old/scripts/jakarta-oro.csh jakarta-oro/scripts/jakarta-oro.csh
--- jakarta-oro-old/scripts/jakarta-oro.csh	Wed Dec 31 19:00:00 1969
+++ jakarta-oro/scripts/jakarta-oro.csh	Sat Jan  5 18:46:09 2002
@@ -0,0 +1,8 @@
+#!/bin/csh
+set chemin = "@PREFIX@/share/java/jakarta-oro/jakarta-oro.jar"
+if ($?CLASSPATH) then
+	setenv CLASSPATH "${CLASSPATH}:$chemin"
+else
+	setenv CLASSPATH "$chemin"
+endif
+unset chemin
diff -uNrb jakarta-oro-old/scripts/jakarta-oro.sh jakarta-oro/scripts/jakarta-oro.sh
--- jakarta-oro-old/scripts/jakarta-oro.sh	Wed Dec 31 19:00:00 1969
+++ jakarta-oro/scripts/jakarta-oro.sh	Sat Jan  5 18:46:20 2002
@@ -0,0 +1,9 @@
+#!/bin/sh
+chemin="@PREFIX@/share/java/jakarta-oro/jakarta-oro.jar"
+if (test "$CLASSPATH")
+then
+	export  CLASSPATH="${CLASSPATH}:$chemin"
+else
+	export  CLASSPATH="$chemin"
+fi
+
