diff -uNbr javamail-1.2/scripts/javamail.csh javamail-1.2-new/scripts/javamail.csh
--- javamail-1.2/scripts/javamail.csh	Wed Dec 31 19:00:00 1969
+++ javamail-1.2-new/scripts/javamail.csh	Sat Jan  5 19:45:50 2002
@@ -0,0 +1,8 @@
+#!/bin/csh
+set chemin = "@PREFIX@/share/java/javamail/imap.jar:@PREFIX@/share/java/javamail/mail.jar:@PREFIX@/share/java/javamail/mailapi.jar:@PREFIX@/share/java/javamail/pop3.jar:@PREFIX@/share/java/javamail/smtp.jar"
+if ($?CLASSPATH) then
+	setenv CLASSPATH "${CLASSPATH}:$chemin"
+else
+	setenv CLASSPATH "$chemin"
+endif
+unset chemin
diff -uNbr javamail-1.2/scripts/javamail.sh javamail-1.2-new/scripts/javamail.sh
--- javamail-1.2/scripts/javamail.sh	Wed Dec 31 19:00:00 1969
+++ javamail-1.2-new/scripts/javamail.sh	Sat Jan  5 19:46:09 2002
@@ -0,0 +1,9 @@
+#!/bin/sh
+chemin="@PREFIX@/share/java/javamail/imap.jar:@PREFIX@/share/java/javamail/mail.jar:@PREFIX@/share/java/javamail/mailapi.jar:@PREFIX@/share/java/javamail/pop3.jar:@PREFIX@/share/java/javamail/smtp.jar"
+if (test "$CLASSPATH")
+then
+	export  CLASSPATH="${CLASSPATH}:$chemin"
+else
+	export  CLASSPATH="$chemin"
+fi
+
