diff -uNbr jta-spec1_0_1/scripts/jta-spec.csh jta-spec1_0_1-new/scripts/jta-spec.csh
--- jta-spec1_0_1/scripts/jta-spec.csh	Wed Dec 31 19:00:00 1969
+++ jta-spec1_0_1-new/scripts/jta-spec.csh	Sat Jan  5 16:12:36 2002
@@ -0,0 +1,8 @@
+#!/bin/csh
+set chemin = "@PREFIX@/share/java/jta-spec/jta-spec1_0_1.jar"
+if ($?CLASSPATH) then
+	setenv CLASSPATH "${CLASSPATH}:$chemin"
+else
+	setenv CLASSPATH "$chemin"
+endif
+unset chemin
diff -uNbr jta-spec1_0_1/scripts/jta-spec.sh jta-spec1_0_1-new/scripts/jta-spec.sh
--- jta-spec1_0_1/scripts/jta-spec.sh	Wed Dec 31 19:00:00 1969
+++ jta-spec1_0_1-new/scripts/jta-spec.sh	Sat Jan  5 16:12:47 2002
@@ -0,0 +1,9 @@
+#!/bin/sh
+chemin="@PREFIX@/share/java/jta-spec/jta-spec1_0_1.jar"
+if (test "$CLASSPATH")
+then
+	export  CLASSPATH="${CLASSPATH}:$chemin"
+else
+	export  CLASSPATH="$chemin"
+fi
+
