diff -Nurd libdivxdecore-0.4.7/src/yuv2rgb.h libdivxdecore-0.4.7-patched/src/yuv2rgb.h
--- libdivxdecore-0.4.7/src/yuv2rgb.h	Fri Jan 19 12:55:28 2001
+++ libdivxdecore-0.4.7-patched/src/yuv2rgb.h	Fri Feb  1 19:03:12 2002
@@ -37,25 +37,25 @@
 #ifndef _YUVRGB_H_
 #define _YUVRGB_H_
 
-void (*convert_yuv)(unsigned char *puc_y, int stride_y,
+extern void (*convert_yuv)(unsigned char *puc_y, int stride_y,
 	unsigned char *puc_u, unsigned char *puc_v, int stride_uv,
 	unsigned char *bmp, int width_y, int height_y);
 
 #ifndef LINUX
 
-void yuv2rgb_16(
+extern void yuv2rgb_16(
 	uint8_t *puc_y, int stride_y, 
 	uint8_t *puc_u, 
 	uint8_t *puc_v, int stride_uv, 
   uint8_t *puc_out, 
 	int width_y, int height_y);
-void yuv2rgb_24(
+extern void yuv2rgb_24(
 	uint8_t *puc_y, int stride_y, 
 	uint8_t *puc_u, 
 	uint8_t *puc_v, int stride_uv, 
   uint8_t *puc_out, 
 	int width_y, int height_y);
-void yuv2rgb_32(
+extern void yuv2rgb_32(
 	uint8_t *puc_y, int stride_y, 
   uint8_t *puc_u, uint8_t *puc_v, int stride_uv, 
   uint8_t *puc_out, 
