diff -ruN libdv-0.98.orig/Makefile.in libdv-0.98/Makefile.in
--- libdv-0.98.orig/Makefile.in	2002-07-30 10:36:54.000000000 -0600
+++ libdv-0.98/Makefile.in	2002-09-05 11:41:26.000000000 -0600
@@ -94,7 +94,7 @@
 am__quote = @am__quote@
 install_sh = @install_sh@
 
-SUBDIRS = libdv playdv encodedv
+SUBDIRS = libdv encodedv #playdv
 
 AUX_DIST = $(ac_aux_dir)/config.guess \
                                    $(ac_aux_dir)/config.sub \
diff -ruN libdv-0.98.orig/encodedv/dvavi.c libdv-0.98/encodedv/dvavi.c
--- libdv-0.98.orig/encodedv/dvavi.c	2002-02-24 01:38:07.000000000 -0700
+++ libdv-0.98/encodedv/dvavi.c	2002-09-05 11:35:45.000000000 -0600
@@ -41,7 +41,11 @@
 #include <popt.h>
 #endif
 
+#ifndef __APPLE__
 #include <malloc.h>
+#else
+#include <stdlib.h>
+#endif
 #include <stdarg.h>
 
 #include <libdv/headers.h>
diff -ruN libdv-0.98.orig/encodedv/dvconnect.c libdv-0.98/encodedv/dvconnect.c
--- libdv-0.98.orig/encodedv/dvconnect.c	2002-04-03 22:42:11.000000000 -0700
+++ libdv-0.98/encodedv/dvconnect.c	2002-09-05 11:35:45.000000000 -0600
@@ -44,7 +44,11 @@
 #include <popt.h>
 #endif
 
+#ifndef __APPLE__
 #include <malloc.h>
+#else
+#include <stdlib.h>
+#endif
 
 static long cip_n_ntsc = 2436;
 static long cip_d_ntsc = 38400;
@@ -862,6 +866,7 @@
 int rt_raisepri (int pri)
 {
 	struct sched_param scp;
+#ifndef __APPLE__
 
 	/*
 	 * Verify that scheduling is available
@@ -879,6 +884,9 @@
 		}
 	}
 	return (0);
+#else
+	return (-1);
+#endif
 }
 
 /* ------------------------------------------------------------------------
