diff -uNbr libxslt-1.0.30/doc/Makefile.in libxslt-1.0.30-new/doc/Makefile.in
--- libxslt-1.0.30/doc/Makefile.in	Sun May  4 11:47:54 2003
+++ libxslt-1.0.30-new/doc/Makefile.in	Wed Jun 11 19:10:44 2003
@@ -356,7 +356,7 @@
 libxslt-sections.txt : scan
 	cp libxslt-decl-list.txt libxslt-sections.txt
 
-rebuild: libxslt-sections.txt templates xml html
+rebuild: libxslt-sections.txt templates html
 
 install-data-local: 
 	$(mkinstalldirs) $(DESTDIR)$(TARGET_DIR)
@@ -371,7 +371,7 @@
 dist-hook:
 	(cd $(srcdir) ; tar cvf - *.1 *.html site.xsl news.xsl xsa.xsl *.gif html/*.html html/*.png tutorial/libxslt*) | (cd $(distdir); tar xf -)
 
-.PHONY : html xml templates scan
+.PHONY : html templates scan
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
