diff -uNrb mx4j-old/scripts/mx4j.csh mx4j/scripts/mx4j.csh
--- mx4j-old/scripts/mx4j.csh	Wed Dec 31 19:00:00 1969
+++ mx4j/scripts/mx4j.csh	Sat Jan  5 18:46:09 2002
@@ -0,0 +1,8 @@
+#!/bin/csh
+set chemin = "@PREFIX@/share/java/mx4j/mx4j-jmx.jar:@PREFIX@/share/java/mx4j/mx4j-tools.jar"
+if ($?CLASSPATH) then
+	setenv CLASSPATH "${CLASSPATH}:$chemin"
+else
+	setenv CLASSPATH "$chemin"
+endif
+unset chemin
diff -uNrb mx4j-old/scripts/mx4j.sh mx4j/scripts/mx4j.sh
--- mx4j-old/scripts/mx4j.sh	Wed Dec 31 19:00:00 1969
+++ mx4j/scripts/mx4j.sh	Sat Jan  5 18:46:20 2002
@@ -0,0 +1,9 @@
+#!/bin/sh
+chemin="@PREFIX@/share/java/mx4j/mx4j-jmx.jar:@PREFIX@/share/java/mx4j/mx4j-tools.jar"
+if (test "$CLASSPATH")
+then
+	export  CLASSPATH="${CLASSPATH}:$chemin"
+else
+	export  CLASSPATH="$chemin"
+fi
+
