Only in .: .DS_Store
diff -ru ../python-pam-0.4.2/PAMmodule.c ./PAMmodule.c
--- ../python-pam-0.4.2/PAMmodule.c	Thu Oct 17 07:10:37 2002
+++ ./PAMmodule.c	Fri Jan 24 15:53:49 2003
@@ -9,8 +9,8 @@
 
 static char revision[] = "$Id: PAMmodule.c,v 0.8 1999/08/17 06:27:27 cvs Exp $";
 
-#include <security/pam_appl.h>
-#include <security/pam_misc.h>
+#include <pam/pam_appl.h>
+#include <pam/pam_misc.h>
 #include <Python.h>
 #include <stdio.h>
 #include <dlfcn.h>
Only in .: build
Only in .: patch.patch
diff -ru ../python-pam-0.4.2/setup.py ./setup.py
--- ../python-pam-0.4.2/setup.py	Thu Oct 17 11:58:47 2002
+++ ./setup.py	Fri Jan 24 19:37:38 2003
@@ -8,8 +8,9 @@
 
 ext = Extension(
     name="PAMmodule",
-    libraries=["pam","pam_misc"],
-    sources=["PAMmodule.c"]
+    libraries=["pam","pam_misc", "dl"],
+    sources=["PAMmodule.c"],
+    include_dirs=["/usr/include", "/sw/include"]
 )
 ##print ext.__dict__; sys.exit(1)
 
