diff -ru slang-1.4.5/configure slang-1.4.5-patched/configure
--- slang-1.4.5/configure	Sun Feb 10 08:39:19 2002
+++ slang-1.4.5-patched/configure	Tue Feb 12 19:53:02 2002
@@ -2028,9 +2028,9 @@
      ELF_LINK_CMD="\$(ELF_LINK) -install_name \$(install_lib_dir)/\$(ELFLIB_MAJOR) -compatibility_version \$(ELF_MAJOR_VERSION) -current_version \$(ELF_MAJOR_VERSION).\$(ELF_MINOR_VERSION)"
      ELF_DEP_LIBS="$LDFLAGS \$(DL_LIB)"
      CC_SHARED="cc -bundle -flat_namespace -undefined suppress \$(CFLAGS) -fno-common"
-     ELFLIB="lib$(THIS_LIB).dylib"
-     ELFLIB_MAJOR="lib$(THIS_LIB).$(ELF_MAJOR_VERSION).dylib"
-     ELFLIB_MAJOR_MINOR="lib$(THIS_LIB).$(ELF_MAJOR_VERSION).$(ELF_MINOR_VERSION).dylib"
+     ELFLIB="lib\$(THIS_LIB).dylib"
+     ELFLIB_MAJOR="lib\$(THIS_LIB).\$(ELF_MAJOR_VERSION).dylib"
+     ELFLIB_MAJOR_MINOR="lib\$(THIS_LIB).\$(ELF_MAJOR_VERSION).\$(ELF_MINOR_VERSION).dylib"
      ;;
   * )
     echo "Note: ELF compiler for host_os=$host_os may be wrong"
