diff -bur xotcl-full-0.9.4/ xotcl-full-0.9.4-patched/
--- xotcl-full-0.9.4/Makefile.in	Sun Apr  7 09:40:16 2002
+++ xotcl-full-0.9.4-patched/Makefile.in	Thu Aug 15 22:51:09 2002
@@ -47,7 +47,6 @@
 DOCCMD = $(XOTCL_LIBRARY)/makeDoc.xotcl $(DOCDIR)
 DOCFILES = doc/langRef.xotcl \
 	$(XOTCL_LIBRARY)/*.xotcl \
-	$(XOTCL_TEST)/*.xotcl \
 	$(ACTIWEB_APPS)/scripts/*.xotcl \
 	$(ACTIWEB_APPS)/comm/[flsw]*.xotcl \
 	$(ACTIWEB_APPS)/actiweb/univ/UNIVERSAL.xotcl \
@@ -78,7 +77,7 @@
 everything: config xotcl libs packages doc end
 
 doc: xotcl $(DOCFILES)
-	$(XOTCLSH) $(DOCCMD) $(DOCFILES)
+	DYLD_LIBRARY_PATH=$(XOTCL_BUILD_DIR) $(XOTCLSH) $(DOCCMD) $(DOCFILES)
 
 clean::
 	rm -f $(DOCDIR)/*-xotcl.html
@@ -89,8 +88,7 @@
 	rm -rf apps/actiweb/receiver
 
 packages: xotcl libs
-	(cd $(XOTCL_DIR)/library; ../../$(XOTCLSH) ../../$(MAKE_XOTCL) -target $(INST_XOLIBPKG) -all)
-	(./$(XOTCLSH) ./$(MAKE_XOTCL) -target $(INST_XOLIBPKG) -all)
+	(cd $(XOTCL_DIR)/library; DYLD_LIBRARY_PATH=../../$(XOTCL_BUILD_DIR) ../../$(XOTCLSH) ../../$(MAKE_XOTCL) -target $(INST_XOLIBPKG) -all)
 
 end:
 	@echo "" 
@@ -166,10 +164,10 @@
 
 
 test: xotclsh
-	$(XOTCLSH) $(XOTCL_DIR)/tests/speedtest.xotcl
-	$(XOTCLSH) $(XOTCL_DIR)/tests/testx.xotcl
-	$(XOTCLSH) $(XOTCL_DIR)/tests/testo.xotcl
-	(./$(XOTCLSH) ./$(MAKE_XOTCL) -target $(INST_XOLIBPKG) -test)
+	DYLD_LIBRARY_PATH=$(XOTCL_BUILD_DIR) $(XOTCLSH) $(XOTCL_DIR)/tests/speedtest.xotcl
+	DYLD_LIBRARY_PATH=$(XOTCL_BUILD_DIR) $(XOTCLSH) $(XOTCL_DIR)/tests/testx.xotcl
+	DYLD_LIBRARY_PATH=$(XOTCL_BUILD_DIR) $(XOTCLSH) $(XOTCL_DIR)/tests/testo.xotcl
+	(DYLD_LIBRARY_PATH=$(XOTCL_BUILD_DIR) ./$(XOTCLSH) ./$(MAKE_XOTCL) -target $(INST_XOLIBPKG) -test)
 
 tar: distclean
 	tar zcvf ../xotcl-$(FULLVERSION).tar.gz \
@@ -182,27 +180,27 @@
 	if (cd $(XOTCL_BUILD_DIR); $(MAKE) install-xotclshells) ; then true ; else exit 1 ; fi ;
 	@for dir in $(subdirs) ; do \
 		if test -d $$dir ; then \
-		    echo "$$dir: $(MAKE) $@" ;\
-		    if (cd $$dir; $(MAKE) $@) ; then true ; else exit 1 ; fi ; \
+		    echo "$$dir: $(MAKE) DESTDIR=$(DESTDIR) $@" ;\
+		    if (cd $$dir; $(MAKE) DESTDIR=$(DESTDIR) $@) ; then true ; else exit 1 ; fi ; \
 		else \
 		    exit 1; \
 		fi \
 	done;
-	(cd library; ../$(XOTCLSH) ../$(MAKE_XOTCL) -target $(INST_XOLIBPKG) -install)
+	(cd library; export DYLD_LIBRARY_PATH=`pwd`/../$(XOTCL_BUILD_DIR); ../$(XOTCLSH) ../$(MAKE_XOTCL) -target $(DESTDIR)$(INST_XOLIBPKG) -install)
 
 
 install-doc:
-	$(mkinstalldirs) $(prefix)/doc/xotcl$(VERSION)
-	(cd ./doc; for i in *; \
+	$(mkinstalldirs) $(DESTDIR)$(prefix)/share/doc/xotcl
+	(cd ./doc; for i in *.html *.gif *.jpg; \
 		do \
-		../$(INSTALL) $$i $(prefix)/doc/xotcl$(VERSION); \
+		../$(INSTALL) $$i $(DESTDIR)$(prefix)/share/doc/xotcl; \
 		done)
 
 install-binaries:
 	@for dir in $(subdirs) ; do \
 		if test -d $$dir ; then \
-		    echo "$$dir: $(MAKE) $@" ;\
-		    if (cd $$dir; $(MAKE) $@) ; then true ; else exit 1 ; fi ; \
+		    echo "$$dir: $(MAKE) DESTDIR=$(DESTDIR) $@" ;\
+		    if (cd $$dir; $(MAKE) DESTDIR=$(DESTDIR) $@) ; then true ; else exit 1 ; fi ; \
 		else \
 		    exit 1; \
 		fi \
@@ -211,8 +209,8 @@
 install-libraries:
 	@for dir in $(subdirs) ; do \
 		if test -d $$dir ; then \
-		    echo "$$dir: $(MAKE) $@" ;\
-		    if (cd $$dir; $(MAKE) $@) ; then true ; else exit 1 ; fi ; \
+		    echo "$$dir: $(MAKE) DESTDIR=$(DESTDIR) $@" ;\
+		    if (cd $$dir; $(MAKE) DESTDIR=$(DESTDIR) $@) ; then true ; else exit 1 ; fi ; \
 		else \
 		    exit 1; \
 		fi \
@@ -262,6 +260,6 @@
 	done;
 
 versionchange: xotclsh
-	$(XOTCLSH) $(XOTCL_DIR)/changeXOTclVersion.xotcl
+	DYLD_LIBRARY_PATH=$(XOTCL_BUILD_DIR) $(XOTCLSH) $(XOTCL_DIR)/changeXOTclVersion.xotcl
 
 # DO NOT DELETE THIS LINE -- make depend depends on it.
--- xotcl-full-0.9.4/configure	Fri Dec 28 08:46:15 2001
+++ xotcl-full-0.9.4-patched/configure	Wed Jul 24 22:27:13 2002
@@ -807,6 +807,8 @@
 s%@DEFS@%$DEFS%g
 s%@LDFLAGS@%$LDFLAGS%g
 s%@LIBS@%$LIBS%g
+s%@X11_INCLUDE_DIR@%$x_includes%g
+s%@X11_LIBRARY_DIR@%$x_libraries%g
 s%@exec_prefix@%$exec_prefix%g
 s%@prefix@%$prefix%g
 s%@program_transform_name@%$program_transform_name%g
--- xotcl-full-0.9.4/library/store/XOTclGdbm/Makefile.in	Sun Dec  9 10:46:58 2001
+++ xotcl-full-0.9.4-patched/library/store/XOTclGdbm/Makefile.in	Tue Aug  6 10:36:59 2002
@@ -138,7 +138,7 @@
 SHLIB_CFLAGS = @SHLIB_CFLAGS@
 SHLIB_LD = @SHLIB_LD@
 SHLIB_LDFLAGS = @SHLIB_LDFLAGS@
-SHLIB_LD_LIBS = @SHLIB_LD_LIBS@
+SHLIB_LD_LIBS = @SHLIB_LD_LIBS@ @XOTCL_BUILD_LIB_SPEC@
 STLIB_LD = @STLIB_LD@
 TCL_BIN_DIR = @TCL_BIN_DIR@
 TCL_DEFS = @TCL_DEFS@
@@ -252,7 +252,7 @@
 
 $(xotclgdbm_LIB_FILE): $(xotclgdbm_OBJECTS)
 	-rm -f ../$(xotclgdbm_LIB_FILE)
-	@MAKE_LIB@ -lgdbm
+	@MAKE_LIB@ -L@libdir@ -lgdbm
 	$(RANLIB) $(xotclgdbm_LIB_FILE)
 	mv $(xotclgdbm_LIB_FILE) ..
 
--- xotcl-full-0.9.4/library/store/XOTclGdbm/configure	Fri Dec 28 08:46:15 2001
+++ xotcl-full-0.9.4-patched/library/store/XOTclGdbm/configure	Tue Aug 13 00:13:00 2002
@@ -1597,6 +1597,7 @@
 LDFLAGS_DEBUG=${TCL_LDFLAGS_DEBUG}
 LDFLAGS_OPTIMIZE=${TCL_LDFLAGS_OPTIMIZE}
 SHLIB_LD=${TCL_SHLIB_LD}
+SHLIB_LDFLAGS="-current_version ${VERSION} -compatibility_version ${VERSION}"
 STLIB_LD=${TCL_STLIB_LD}
 SHLIB_CFLAGS=${TCL_SHLIB_CFLAGS}
 
@@ -1727,8 +1728,8 @@
 	;;
     *)
 	if test "${SHARED_BUILD}" = "1" ; then
-            SHLIB_LD_LIBS="${TCL_BUILD_STUB_LIB_SPEC}"
-	    #SHLIB_LD_LIBS="${TCL_STUB_LIB_SPEC}"
+            #SHLIB_LD_LIBS="${TCL_BUILD_STUB_LIB_SPEC}"
+	    SHLIB_LD_LIBS="${TCL_STUB_LIB_SPEC}"
 	    eval "${PACKAGE}_LIB_FILE=lib${PACKAGE}${SHARED_LIB_SUFFIX}"
 	    RANLIB=:
 	else
--- xotcl-full-0.9.4/library/store/XOTclSdbm/Makefile.in	Sun Dec  9 10:46:48 2001
+++ xotcl-full-0.9.4-patched/library/store/XOTclSdbm/Makefile.in	Tue Aug  6 01:54:56 2002
@@ -139,7 +139,7 @@
 SHLIB_CFLAGS = @SHLIB_CFLAGS@
 SHLIB_LD = @SHLIB_LD@
 SHLIB_LDFLAGS = @SHLIB_LDFLAGS@
-SHLIB_LD_LIBS = @SHLIB_LD_LIBS@
+SHLIB_LD_LIBS = @SHLIB_LD_LIBS@ @XOTCL_BUILD_LIB_SPEC@
 STLIB_LD = @STLIB_LD@
 TCL_BIN_DIR = @TCL_BIN_DIR@
 TCL_DEFS = @TCL_DEFS@
--- xotcl-full-0.9.4/library/store/XOTclSdbm/configure	Fri Dec 28 08:46:15 2001
+++ xotcl-full-0.9.4-patched/library/store/XOTclSdbm/configure	Tue Aug 13 00:12:40 2002
@@ -1597,6 +1597,7 @@
 LDFLAGS_DEBUG=${TCL_LDFLAGS_DEBUG}
 LDFLAGS_OPTIMIZE=${TCL_LDFLAGS_OPTIMIZE}
 SHLIB_LD=${TCL_SHLIB_LD}
+SHLIB_LDFLAGS="-current_version ${VERSION} -compatibility_version ${VERSION}"
 STLIB_LD=${TCL_STLIB_LD}
 SHLIB_CFLAGS=${TCL_SHLIB_CFLAGS}
 
@@ -1728,8 +1729,8 @@
 	;;
     *)
 	if test "${SHARED_BUILD}" = "1" ; then
-            SHLIB_LD_LIBS="${TCL_BUILD_STUB_LIB_SPEC}"
-	    #SHLIB_LD_LIBS="${TCL_STUB_LIB_SPEC}"
+            #SHLIB_LD_LIBS="${TCL_BUILD_STUB_LIB_SPEC}"
+	    SHLIB_LD_LIBS="${TCL_STUB_LIB_SPEC}"
 	    eval "${PACKAGE}_LIB_FILE=lib${PACKAGE}${SHARED_LIB_SUFFIX}"
 	    RANLIB=:
 	else
--- xotcl-full-0.9.4/library/store/XOTclSdbm/xotclsdbm.c	Fri Nov  9 04:01:48 2001
+++ xotcl-full-0.9.4-patched/library/store/XOTclSdbm/xotclsdbm.c	Tue Aug  6 00:23:04 2002
@@ -56,11 +56,7 @@
     }
    */
   /* Storage interface at the moment assumes mode=rwc */
-#ifdef _WIN32
   mode = O_CREAT | O_RDWR;
-#else
-  mode = O_CREAT | O_RDWR | O_SYNC;
-#endif
 
   /* name not in hashtab - create new db */
   if (XOTclGetObjClientData(obj))
--- xotcl-full-0.9.4/library/xml/TclExpat-1.1/configure	Fri Dec 28 08:46:16 2001
+++ xotcl-full-0.9.4-patched/library/xml/TclExpat-1.1/configure	Tue Aug 13 00:13:14 2002
@@ -1474,6 +1474,7 @@
 LDFLAGS_DEBUG=${TCL_LDFLAGS_DEBUG}
 LDFLAGS_OPTIMIZE=${TCL_LDFLAGS_OPTIMIZE}
 SHLIB_LD=${TCL_SHLIB_LD}
+SHLIB_LDFLAGS="-current_version ${VERSION} -compatibility_version ${VERSION}"
 STLIB_LD=${TCL_STLIB_LD}
 SHLIB_CFLAGS=${TCL_SHLIB_CFLAGS}
 
@@ -1604,8 +1605,8 @@
 	;;
     *)
 	if test "${SHARED_BUILD}" = "1" ; then
-            SHLIB_LD_LIBS="${TCL_BUILD_STUB_LIB_SPEC}"
-	    #SHLIB_LD_LIBS="${TCL_STUB_LIB_SPEC}"
+            #SHLIB_LD_LIBS="${TCL_BUILD_STUB_LIB_SPEC}"
+	    SHLIB_LD_LIBS="${XOTCL_BUILD_LIB_SPEC} ${TCL_STUB_LIB_SPEC}"
 	    eval "${PACKAGE}_LIB_FILE=lib${PACKAGE}${SHARED_LIB_SUFFIX}"
 	    RANLIB=:
 	else
--- xotcl-full-0.9.4/library/xml/TclExpat-1.1/xmlparse.c	Fri Nov  9 03:57:22 2001
+++ xotcl-full-0.9.4-patched/library/xml/TclExpat-1.1/xmlparse.c	Tue Aug  6 12:04:53 2002
@@ -720,7 +720,9 @@
 					    const char **endPtr)
 {
   const char *next;
-  int tok = XmlContentTok(encoding, start, end, &next);
+  // HACK: gcc doesn't like the macro to refer to something global
+  const ENCODING *enc = encoding;
+  int tok = XmlContentTok(enc, start, end, &next);
   switch (tok) {
   case XML_TOK_BOM:
     start = next;
@@ -751,7 +753,9 @@
 					    const char **endPtr)
 {
   const char *next;
-  int tok = XmlContentTok(encoding, start, end, &next);
+  // HACK: gcc doesn't like the macro to refer to something global
+  const ENCODING *enc = encoding;
+  int tok = XmlContentTok(enc, start, end, &next);
   switch (tok) {
   case XML_TOK_XML_DECL:
     {
@@ -1504,9 +1508,11 @@
 		const char *end,
 		const char **nextPtr)
 {
+  // HACK: gcc doesn't like the macro to refer to something global
+  const ENCODING *enc = encoding;
   for (;;) {
     const char *next;
-    int tok = XmlPrologTok(encoding, s, end, &next);
+    int tok = XmlPrologTok(enc, s, end, &next);
     if (tok <= 0) {
       if (nextPtr != 0 && tok != XML_TOK_INVALID) {
 	*nextPtr = s;
@@ -1802,8 +1808,10 @@
   processor = epilogProcessor;
   eventPtr = s;
   for (;;) {
+    /* HACK: gcc doesn't like the macro to refer to something global */
+    const ENCODING *enc = encoding;
     const char *next;
-    int tok = XmlPrologTok(encoding, s, end, &next);
+    int tok = XmlPrologTok(enc, s, end, &next);
     eventEndPtr = next;
     switch (tok) {
     case XML_TOK_TRAILING_CR:
@@ -2002,8 +2010,10 @@
   entityTextPtr += encoding->minBytesPerChar;
   entityTextEnd -= encoding->minBytesPerChar;
   for (;;) {
+    /* HACK: gcc doesn't like the macro to refer to something global */
+    const ENCODING *enc = encoding;
     const char *next;
-    int tok = XmlEntityValueTok(encoding, entityTextPtr, entityTextEnd, &next);
+    int tok = XmlEntityValueTok(enc, entityTextPtr, entityTextEnd, &next);
     switch (tok) {
     case XML_TOK_PARAM_ENTITY_REF:
       eventPtr = entityTextPtr;
--- xotcl-full-0.9.4/xotcl-0.9.4/library/make.xotcl	Fri Mar  8 17:26:35 2002
+++ xotcl-full-0.9.4-patched/xotcl-0.9.4/library/make.xotcl	Mon Aug 12 11:07:03 2002
@@ -35,7 +35,7 @@
 
   set so [glob -nocomplain *[info sharedlibextension]]
   # loading libxotcl into xotclsh crashes on some systems
-  set p [lsearch -exact $so libxotcl$::xotcl::version.so]
+  set p [lsearch -exact $so libxotcl$::xotcl::version[info sharedlibextension]]
   if {$p != -1} {
     set so [lreplace $so $p $p]
     #puts stderr "new so=<$so>"
@@ -43,7 +43,7 @@
   set p [lsearch -exact $so xotcl$::xotcl::version.dll]
   if {$p != -1} {
     set so [lreplace $so $p $p]
-    puts stderr "new dll=<$so> p=$p"
+    #puts stderr "new dll=<$so> p=$p"
   }
   set fls [concat $fls $so]
   
@@ -107,7 +107,7 @@
   }
 }
 make proc inEachDir {path cmd} {
-  #puts stderr "inEachDir $path  [file isdirectory $path]"
+  #puts stderr "inEachDir $path ([file isdirectory $path]) $cmd"
   if { [file isdirectory $path] 
        && ![string match *CVS $path]
        && ![string match *SCCS $path]
--- xotcl-full-0.9.4/xotcl-0.9.4/unix/Makefile.in	Wed Feb 27 03:08:28 2002
+++ xotcl-full-0.9.4-patched/xotcl-0.9.4/unix/Makefile.in	Thu Aug 15 22:47:31 2002
@@ -220,13 +220,15 @@
 TCL_PLATFORM_DIR_NATIVE = @TCL_PLATFORM_DIR_NATIVE@
 TCL_GENERIC_DIR_NATIVE = @TCL_GENERIC_DIR_NATIVE@
 TCLSH_PROG = @TCLSH_PROG@
+X11_INCLUDES = -I@X11_INCLUDE_DIR@
+X11_LIB_SWITCHES = -L@X11_LIBRARY_DIR@ -lX11
 XOTCL_INCLUDES = @XOTCL_INCLUDES@
 
 AUTOCONF = autoconf
 
 LDFLAGS = $(LDFLAGS_DEFAULT)
 
-INCLUDES = @TCL_INCLUDES@ @XOTCL_INCLUDES@ @TK_INCLUDES@
+INCLUDES = @TCL_INCLUDES@ @XOTCL_INCLUDES@ @TK_INCLUDES@ $(X11_INCLUDES)
 
 EXTRA_CFLAGS = $(AC_FLAGS) $(PROTO_FLAGS) $(MEM_DEBUG_FLAGS) $(NO_DEPRECATED_FLAGS) $(TCL_EXTRA_CFLAGS)
 
@@ -289,16 +291,16 @@
 #========================================================================
 
 install-libraries: libraries
-	$(mkinstalldirs) $(includedir)
-	@echo "Installing header files in $(includedir)"
+	$(mkinstalldirs) $(DESTDIR)$(includedir)
+	@echo "Installing header files in $(DESTDIR)$(includedir)"
 	@for i in $(GENERIC_HDRS) ; do \
 	    echo "Installing $$i" ; \
-	    $(INSTALL_DATA) $$i $(includedir) ; \
+	    $(INSTALL_DATA) $$i $(DESTDIR)$(includedir) ; \
 	done;
 	@echo "Installing library files in $(XOTCL_LIBRARY)"
-	@for i in $(TOP_DIR)/library/*{tcl,add} ; do \
+	@for i in $(TOP_DIR)/library/*tcl $(TOP_DIR)/library/*add ; do \
 	    echo "Installing $$i" ; \
-	    $(INSTALL_DATA) $$i $(XOTCL_LIBRARY) ; \
+	    $(INSTALL_DATA) $$i $(DESTDIR)$(XOTCL_LIBRARY) ; \
 	done;
 
 #========================================================================
@@ -307,17 +309,17 @@
 #========================================================================
 
 install-doc: doc
-	$(mkinstalldirs) $(mandir)/man1
-	$(mkinstalldirs) $(mandir)/man3
-	$(mkinstalldirs) $(mandir)/mann
+	$(mkinstalldirs) $(DESTDIR)$(mandir)/man1
+#	$(mkinstalldirs) $(DESTDIR)$(mandir)/man3
+#	$(mkinstalldirs) $(DESTDIR)$(mandir)/mann
 	(cd ../man ; \
 	 for i in *.1; \
 	    do \
 	    echo "Installing $$i"; \
-	    rm -f $(mandir)/man1/$$i; \
+	    rm -f $(DESTDIR)$(mandir)/man1/$$i; \
 	    sed -e '/man\.macros/r man.macros' -e '/man\.macros/d' \
-		    $$i > $(mandir)/man1/$$i; \
-	    chmod 444 $(mandir)/man1/$$i; \
+		    $$i > $(DESTDIR)$(mandir)/man1/$$i; \
+	    chmod 444 $(DESTDIR)$(mandir)/man1/$$i; \
 	    done)	
 #	$(mkinstalldirs) $(mandir)/mann
 #	@echo "Installing man pages in $(mandir)"
@@ -360,7 +362,7 @@
 	@if test ! -d so; then mkdir so; fi
 	-rm -f $(xotcl_LIB_FILE)
 	@MAKE_LIB@
-	$(RANLIB) $(xotcl_LIB_FILE)
+#	$(RANLIB) $(xotcl_LIB_FILE)
 
 $(xotclstub_LIB_FILE): $(xotclstub_OBJECTS)
 	@if test ! -d so; then mkdir so; fi
@@ -381,15 +383,15 @@
 	#echo package ifneeded XOTcl $(VERSION) [list load [file join \$$dir ../.. "$(xotcl_LIB_FILE)"] XOTcl] > \
 	#	"$(pkglibdir)/libxotcl/pkgIndex.tcl"
 	echo package ifneeded XOTcl $(VERSION) [list load [file join \$$dir .. "$(xotcl_LIB_FILE)"] XOTcl] > \
-		"$(pkglibdir)/pkgIndex.tcl"
+		"$(DESTDIR)$(pkglibdir)/pkgIndex.tcl"
 
-../xotclsh: o/xotclAppInit.$(OBJEXT) o/findXOTcl.$(OBJEXT)
+../xotclsh: o/xotclAppInit.$(OBJEXT) o/findXOTcl.$(OBJEXT) $(xotcl_LIB_FILE)
 	@if test ! -d o; then mkdir o; fi
-	${CC} $(TCL_LD_FLAGS) o/xotclAppInit.$(OBJEXT) o/findXOTcl.$(OBJEXT) @TCL_BUILD_LIB_SPEC@ $(TCL_LIBS) -lc -o ../xotclsh
+	${CC} $(TCL_LD_FLAGS) o/xotclAppInit.$(OBJEXT) o/findXOTcl.$(OBJEXT) @XOTCL_BUILD_LIB_SPEC@ @TCL_LIB_SPEC@ $(TCL_LIBS) -lc -o ../xotclsh
 
-../xowish: o/xotkAppInit.$(OBJEXT) o/findXOTcl.$(OBJEXT)
+../xowish: o/xotkAppInit.$(OBJEXT) o/findXOTcl.$(OBJEXT) $(xotcl_LIB_FILE)
 	@if test ! -d o; then mkdir o; fi
-	${CC} $(TCL_LD_FLAGS) o/xotkAppInit.$(OBJEXT) o/findXOTcl.$(OBJEXT) @TK_LIB_SPEC@ @TCL_BUILD_LIB_SPEC@ $(X11_LIB_SWITCHES) $(TK_LIBS) -lc -o ../xowish
+	${CC} $(TCL_LD_FLAGS) o/xotkAppInit.$(OBJEXT) o/findXOTcl.$(OBJEXT) @XOTCL_BUILD_LIB_SPEC@ @TK_LIB_SPEC@ @TCL_LIB_SPEC@ $(X11_LIB_SWITCHES) $(TK_LIBS) -lc -o ../xowish
 
 xotclsh: ../xotclsh
 
@@ -398,14 +400,14 @@
 
 xotclshells: xotclsh xowish
 
-install-xotclshells:
+install-xotclshells: installdirs
 	@if test -e ../xotclsh; then \
-		$(INSTALL_PROGRAM) ../xotclsh $(bindir)/xotclsh$(VERSION); \
-		ln -sf $(bindir)/xotclsh$(VERSION) $(bindir)/xotclsh; \
+		$(INSTALL_PROGRAM) ../xotclsh $(DESTDIR)$(bindir)/xotclsh$(VERSION); \
+		ln -sf $(bindir)/xotclsh$(VERSION) $(DESTDIR)$(bindir)/xotclsh; \
 	fi
 	@if test -e ../xowish; then \
-		$(INSTALL_PROGRAM) ../xowish $(bindir)/xowish$(VERSION); \
-		ln -sf $(bindir)/xowish$(VERSION) $(bindir)/xowish; \
+		$(INSTALL_PROGRAM) ../xowish $(DESTDIR)$(bindir)/xowish$(VERSION); \
+		ln -sf $(bindir)/xowish$(VERSION) $(DESTDIR)$(bindir)/xowish; \
 	fi
 
 #========================================================================
@@ -538,8 +540,11 @@
 	done
 	@list='$(lib_BINARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
+	    ext=`echo $$p|sed -e "s/.*\.//"`; \
+	    if test "x$$ext" != "xdylib"; then \
 	    echo " $(RANLIB) $(DESTDIR)$(libdir)/$$p"; \
 	    $(RANLIB) $(DESTDIR)$(libdir)/$$p; \
+	    fi; \
 	  else :; fi; \
 	done
 
@@ -579,7 +584,7 @@
 	$(mkinstalldirs)  $(DESTDIR)$(libdir)
 	$(mkinstalldirs)  $(DESTDIR)$(bindir)
 	$(mkinstalldirs)  $(DESTDIR)$(pkglibdir)
-	$(mkinstalldirs)  $(XOTCL_LIBRARY)
+	$(mkinstalldirs)  $(DESTDIR)$(XOTCL_LIBRARY)
 
 .PHONY: all binaries clean depend distclean doc install installdirs \
 libraries test
--- xotcl-full-0.9.4/xotcl-0.9.4/unix/configure	Fri Dec 28 08:46:16 2001
+++ xotcl-full-0.9.4-patched/xotcl-0.9.4/unix/configure	Tue Aug 13 00:11:15 2002
@@ -1707,6 +1707,7 @@
 LDFLAGS_DEBUG=${TCL_LDFLAGS_DEBUG}
 LDFLAGS_OPTIMIZE=${TCL_LDFLAGS_OPTIMIZE}
 SHLIB_LD=${TCL_SHLIB_LD}
+SHLIB_LDFLAGS="-current_version ${VERSION} -compatibility_version ${VERSION}"
 STLIB_LD=${TCL_STLIB_LD}
 SHLIB_CFLAGS=${TCL_SHLIB_CFLAGS}
 
@@ -1716,7 +1717,6 @@
 
 
 
-
 #--------------------------------------------------------------------
 # Set the default compiler switches based on the --enable-symbols 
 # option.
@@ -1851,9 +1851,9 @@
 	;;
     *)
 	if test "${SHARED_BUILD}" = "1" ; then
-	    SHLIB_LD_LIBS="${TCL_BUILD_STUB_LIB_SPEC}"
+	    SHLIB_LD_LIBS="${TCL_LIB_SPEC} ${TCL_STUB_LIB_SPEC}"
 	    eval "${PACKAGE}_LIB_FILE=lib${PACKAGE}${SHARED_LIB_SUFFIX}"
-	    RANLIB=:
+#	    RANLIB=:
 	else
 	    eval "${PACKAGE}_LIB_FILE=lib${PACKAGE}${UNSHARED_LIB_SUFFIX}"
 	fi
@@ -2083,6 +2083,8 @@
 s%@DEFS@%$DEFS%g
 s%@LDFLAGS@%$LDFLAGS%g
 s%@LIBS@%$LIBS%g
+s%@X11_INCLUDE_DIR@%$x_includes%g
+s%@X11_LIBRARY_DIR@%$x_libraries%g
 s%@exec_prefix@%$exec_prefix%g
 s%@prefix@%$prefix%g
 s%@program_transform_name@%$program_transform_name%g
--- xotcl-full-0.9.4/xotcl-0.9.4/unix/xotclAppInit.c	Sun Mar 24 03:10:23 2002
+++ xotcl-full-0.9.4-patched/xotcl-0.9.4/unix/xotclAppInit.c	Tue Aug  6 13:21:28 2002
@@ -122,18 +122,20 @@
      * }
      *
      * where "Mod" is the name of the module.
+     */
     
     if (Xotcl_Init(interp) == TCL_ERROR) {
         return TCL_ERROR;
     }
      Tcl_StaticPackage(interp, "XOTcl", Xotcl_Init, 0);
-    */
 
+    /*
     XOTclCorrectAutoPath(interp);
 
     if (Tcl_PkgRequire(interp, "XOTcl", XOTCLVERSION, 1) == NULL) {
       return TCL_ERROR;
     }
+     */
 
     /*
      *  This is xotclsh, so import all xotcl commands by
