diff -ru dblatex-0.2.2/latex/misc/passivetex/unicode.sty dblatex-0.2.2-patched/latex/misc/passivetex/unicode.sty
--- dblatex-0.2.2/latex/misc/passivetex/unicode.sty	2006-12-22 22:43:47.000000000 +0100
+++ dblatex-0.2.2-patched/latex/misc/passivetex/unicode.sty	2007-01-04 01:00:13.000000000 +0100
@@ -50,7 +50,7 @@
 
 \IfFileExists{elfonts.sty}
  {
-  \usepackage{elfonts}
+  %\usepackage{elfonts}
   \def\ElsevierGlyph##1{\csname U##1\endcsname}%
   \def\textpertenthousand{\Pisymbol{ppi031}{33}}
   \let\overrightharpoon\undefined
diff -ru dblatex-0.2.2/setup.py dblatex-0.2.2-patched/setup.py
--- dblatex-0.2.2/setup.py	2006-11-23 23:15:14.000000000 +0100
+++ dblatex-0.2.2-patched/setup.py	2007-01-05 20:03:59.000000000 +0100
@@ -16,15 +16,12 @@
 #
 class BuildScripts(build_scripts):
 
-    SHELL_SCRIPT = """#!%(env_executable)s%(env_args)s%(py_executable)s
+    SHELL_SCRIPT = """#!@PREFIX@/bin/python2.4
 import sys
 import os
 
 package_base = r"%(package_base)s"
 
-%(lib_path)s
-%(catalogs)s
-%(style_set)s
 from %(package_path)s import %(package)s
 %(package)s.main(base=package_base)
 """
