diff -Nurd -x'*~' dictd-1.12.0.orig/Makefile.in dictd-1.12.0/Makefile.in
--- dictd-1.12.0.orig/Makefile.in	2011-01-09 14:34:55.000000000 -0500
+++ dictd-1.12.0/Makefile.in	2011-09-26 11:28:56.000000000 -0400
@@ -67,7 +67,7 @@
 
 LEXLIB= 	@LEXLIB@
 
-CFLAGS= @CPPFLAGS@ @DEFS@ @CFLAGS@ -DDICT_PLUGIN_PATH=\"$(PLUGIN_DIR)/\" -DDICT_DICTIONARY_PATH=\"$(DICT_DIR)/\" -DDICT_VERSION=\"$(DICT_VERSION)\" -DDICT_CONFIG_PATH=\"$(CONF_DIR)/\" -I. -I$(srcdir)
+CFLAGS= @DEFS@ @CFLAGS@ -DDICT_PLUGIN_PATH=\"$(PLUGIN_DIR)/\" -DDICT_DICTIONARY_PATH=\"$(DICT_DIR)/\" -DDICT_VERSION=\"$(DICT_VERSION)\" -DDICT_CONFIG_PATH=\"$(CONF_DIR)/\" -I. -I$(srcdir) @CPPFLAGS@
 
 LIBMAA=		@LIBMAA@
 LIBS=		@LIBS@ ${LIBMAA}
@@ -122,7 +122,7 @@
 	$(LIBTOOL) --tag=CXX --mode=compile $(CXX) -c $(CFLAGS) $< -o $@
 
 %: %.o
-	$(LIBTOOL) --tag=CC --mode=link $(CC) -o $@ -static \
+	$(LIBTOOL) --tag=CC --mode=link $(CC) -o $@ \
 		$^ $(OBJS) $(LDFLAGS) -lz ${LIBS}
 
 include $(srcdir)/deps
diff -Nurd -x'*~' dictd-1.12.0.orig/dict.conf dictd-1.12.0/dict.conf
--- dictd-1.12.0.orig/dict.conf	1969-12-31 19:00:00.000000000 -0500
+++ dictd-1.12.0/dict.conf	2011-09-26 11:28:56.000000000 -0400
@@ -0,0 +1,16 @@
+#  /etc/dict.conf Written by Bob Hilliard <hilliard@debian.org>
+#  1998/03/20.  Last revised Sun, 22 Nov 1998 18:10:04 -0500 This is
+#  the configuration file for /usr/bin/dict.  In most cases only the
+#  server keyword need be specified.
+
+#  This default configuration will try to access a dictd server on the
+#  local host, failing that, it will try the public server.  In many
+#  cases this will be slow, so you should comment out the line for the
+#  server that you don't want to use. To use any other server, enter
+#  its IP address in place of "dict.org".
+
+#  Refer to the dict manpage (man dict) for other options that could
+#  be inserted in here.
+
+server localhost
+server dict.org
diff -Nurd -x'*~' dictd-1.12.0.orig/dictzip.c dictd-1.12.0/dictzip.c
--- dictd-1.12.0.orig/dictzip.c	2011-01-08 10:43:30.000000000 -0500
+++ dictd-1.12.0/dictzip.c	2011-09-26 11:28:56.000000000 -0400
@@ -21,6 +21,7 @@
 #include "dictzip.h"
 #include "data.h"
 
+#include <sys/types.h>
 #include <sys/stat.h>
 
 void dict_data_print_header( FILE *str, dictData *header )
