diff -Nurd -x'*~' djvulibre-3.5.21.orig/configure djvulibre-3.5.21/configure
--- djvulibre-3.5.21.orig/configure	2008-07-02 18:43:52.000000000 -0400
+++ djvulibre-3.5.21/configure	2008-11-12 09:30:17.000000000 -0500
@@ -27343,12 +27343,8 @@
 if test x$ac_enable_djview != xno
 then
  NSDEJAVU_CFLAGS=${X_CFLAGS}
- case "$host" in
- *-cygwin* | *-mingw*)
    NSDEJAVU_LIBS="`echo ${X_LIBS} | sed -e 's/-lXext/-lXt -lXext/'`"
    NSDEJAVU_LIBS="-no-undefined $NSDEJAVU_LIBS"
-   ;;
- esac
 fi
 
 
diff -Nurd -x'*~' djvulibre-3.5.21.orig/gui/nsdejavu/Makefile.in djvulibre-3.5.21/gui/nsdejavu/Makefile.in
--- djvulibre-3.5.21.orig/gui/nsdejavu/Makefile.in	2007-04-20 12:44:23.000000000 -0400
+++ djvulibre-3.5.21/gui/nsdejavu/Makefile.in	2008-11-12 09:09:27.000000000 -0500
@@ -51,7 +51,7 @@
  -I${top_srcdir}/gui/shared/res \
  -I${top_srcdir}/gui/npsdk 
 
-CFLAGS = @CPPFLAGS@ ${DEFS} ${OPTS} ${INCS} ${NSDEJAVU_CFLAGS} 
+CFLAGS = ${DEFS} ${OPTS} ${INCS} ${NSDEJAVU_CFLAGS} @CPPFLAGS@
 
 LIBS = @LDFLAGS@ ${NSDEJAVU_LIBS} 
 
diff -Nurd -x'*~' djvulibre-3.5.21.orig/tools/jb2cmp/Makefile.in djvulibre-3.5.21/tools/jb2cmp/Makefile.in
--- djvulibre-3.5.21.orig/tools/jb2cmp/Makefile.in	2007-11-28 18:55:29.000000000 -0500
+++ djvulibre-3.5.21/tools/jb2cmp/Makefile.in	2008-11-12 09:05:28.000000000 -0500
@@ -49,8 +49,8 @@
 INCS = -I${top_builddir} -I${top_srcdir} -I${srcdir}
 FLAGS = ${DEFS} ${INCS} ${OPTS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${THREAD_LIBS} @LIBS@
-CFLAGS = @CPPFLAGS@ @CFLAGS@ ${FLAGS}
-CXXFLAGS = @CPPFLAGS@ @CXXFLAGS@ ${CXXRPOFLAGS} ${FLAGS}
+CFLAGS = ${FLAGS} @CPPFLAGS@ @CFLAGS@
+CXXFLAGS = ${FLAGS} @CPPFLAGS@ @CXXFLAGS@ ${CXXRPOFLAGS}
 
 
 OBJS = classify.o cuts.o frames.o patterns.o
