--- dvipdfmx-20071115.orig/data/cid-x.map	2007-11-16 02:08:52.000000000 +0900
+++ dvipdfmx-20071115/data/cid-x.map	2008-05-23 19:48:29.000000000 +0900
@@ -71,10 +71,10 @@
 %%
 
 %% Ryumin and GothicBBB found in PostScript printers:
-rml  H Ryumin-Light
-gbm  H GothicBBB-Medium
-rmlv V Ryumin-Light
-gbmv V GothicBBB-Medium
+%rml  H Ryumin-Light
+%gbm  H GothicBBB-Medium
+%rmlv V Ryumin-Light
+%gbmv V GothicBBB-Medium
 
 %% Synthetic font used for futo* family.
 %% jun101 not available.
--- dvipdfmx-20071115.orig/data/dvipdfmx.cfg	2007-11-16 02:08:52.000000000 +0900
+++ dvipdfmx-20071115/data/dvipdfmx.cfg	2008-05-20 05:56:48.000000000 +0900
@@ -134,7 +134,8 @@
 %% than A0, and their llx=lly=0, you can use "-dEPSCrop" instead of
 %% "-sPAPERSIZE=a0"
 %%
-D  "ps2pdf -dCompatibilityLevel=%v -sPAPERSIZE=a0 -dAutoFilterGrayImages=false -dGrayImageFilter=/FlateEncode -dAutoFilterColorImages=false -dColorImageFilter=/FlateEncode %i %o"
+%D  "ps2pdf -dCompatibilityLevel=%v -sPAPERSIZE=a0 -dAutoFilterGrayImages=false -dGrayImageFilter=/FlateEncode -dAutoFilterColorImages=false -dColorImageFilter=/FlateEncode %i %o"
+D  "gs -q -dNOPAUSE -dBATCH -sPAPERSIZE=a0 -sDEVICE=pdfwrite -dCompatibilityLevel=1.3 -dAutoFilterGrayImages=false -dGrayImageFilter=/FlateEncode -dAutoFilterColorImages=false -dColorImageFilter=/FlateEncode -dUseFlateCompression=true -sOutputFile=%o -c '.setpdfwrite << /NeverEmbed [/Courier /Courier-Bold /Courier-Oblique /Courier-BoldOblique /Helvetica /Helvetica-Bold /Helvetica-Oblique /Helvetica-BoldOblique /Times-Roman /Times-Bold /Times-Italic /Times-BoldItalic /Symbol /ZapfDingbats /Ryumin-Light /GothicBBB-Medium] >> setdistillerparams' -f %i -c quit"
 
 %% Frank Siegert's PStill:
 %D  "/usr/local/bin/pstill -c -o %o %i"
@@ -193,3 +193,4 @@
 
 %% Put additonal fontmap files here (usually for Type0 fonts)
 f  cid-x.map
+f  kanjix.map
--- dvipdfmx-20070518.orig/fink/out2uni	1970-01-01 09:00:00.000000000 +0900
+++ dvipdfmx-20070518/fink/out2uni	2008-03-09 02:54:25.000000000 +0900
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+trap 'test -z "$tmpf" || rm -f "$tmpf"' 0
+test -z "$1" && { echo "Usage: out2uni filename" 1>&2; exit 1; }
+tmpf=`mktemp /tmp/out2uni.XXXXXX` || { echo "$0: mktemp failed" 1>&2; exit 1; }
+
+if NKF="`which nkf 2>/dev/null`"
+then
+	$NKF -e ${1}.out > ${tmpf}
+	/bin/cp ${tmpf} ${1}.out
+fi
+
+${0}.bin ${1}
