diff -ru lilypond-2.12.1-orig/config.make.in lilypond-2.12.1/config.make.in
--- lilypond-2.12.1-orig/config.make.in	2009-01-09 19:09:22.000000000 +0100
+++ lilypond-2.12.1/config.make.in	2009-01-09 19:11:33.000000000 +0100
@@ -28,7 +28,7 @@
 FREETYPE2_LIBS = @FREETYPE2_LIBS@
 PANGO_FT2_LIBS = @PANGO_FT2_LIBS@
 
-CONFIG_LIBS = @LIBS@ @EXTRA_LIBS@ $(GUILE_LIBS) $(PANGO_FT2_LIBS) $(FONTCONFIG_LIBS) $(FREETYPE2_LIBS)
+CONFIG_LIBS = @LIBS@ @EXTRA_LIBS@ $(PANGO_FT2_LIBS) $(FONTCONFIG_LIBS) $(FREETYPE2_LIBS) $(GUILE_LIBS)
 CONFIG_LDFLAGS = @LDFLAGS@
 
 PACKAGE = @PACKAGE@
@@ -66,7 +66,7 @@
 stepmake = @stepmake@
 
 package_datadir = $(datadir)/$(package)
-package_infodir = $(infodir)/$(package)
+package_infodir = $(infodir)
 package_libdir = $(libdir)/$(package)
 package_docdir = $(docdir)/$(package)
 package_omfdir = $(omfdir)/$(package)
diff -ru lilypond-2.12.1-orig/flower/include/std-vector.hh lilypond-2.12.1/flower/include/std-vector.hh
--- lilypond-2.12.1-orig/flower/include/std-vector.hh	2009-01-09 19:09:23.000000000 +0100
+++ lilypond-2.12.1/flower/include/std-vector.hh	2009-01-09 19:13:01.000000000 +0100
@@ -63,7 +63,7 @@
 #if HAVE_STL_DATA_METHOD
 #include <vector>
 #else /* !HAVE_STL_DATA_METHOD */
-#define vector __vector
+#define vector lilypond_vector
 #include <vector>
 #undef vector
 
@@ -71,21 +71,21 @@
 
   /* Interface without pointer arithmetic (iterator) semantics.  */
   template<typename T, typename A=std::allocator<T> >
-  class vector : public __vector<T, A>
+  class vector : public lilypond_vector<T, A>
   {
   public:
-    typedef typename __vector<T>::iterator iterator;
-    typedef typename __vector<T>::const_iterator const_iterator;
+    typedef typename lilypond_vector<T>::iterator iterator;
+    typedef typename lilypond_vector<T>::const_iterator const_iterator;
 
-    vector<T, A> () : __vector<T, A> ()
+    vector<T, A> () : lilypond_vector<T, A> ()
     {
     }
 
-    vector<T, A> (vector<T, A> const& v) : __vector<T, A> (v)
+    vector<T, A> (vector<T, A> const& v) : lilypond_vector<T, A> (v)
     {
     }
 
-    vector<T, A> (const_iterator b, const_iterator e) : __vector<T, A> (b, e)
+    vector<T, A> (const_iterator b, const_iterator e) : lilypond_vector<T, A> (b, e)
     {
     }
 
