diff -Nurd -x'*~' mpage-2.5.3.orig/mpage.h mpage-2.5.3/mpage.h
--- mpage-2.5.3.orig/mpage.h	2002-10-20 14:51:15.000000000 -0400
+++ mpage-2.5.3/mpage.h	2006-01-11 11:37:33.000000000 -0500
@@ -167,6 +167,15 @@
 # define	PS_FLAG2	"-Adobe-"
 
 /*
+ * Structure to describe a physical piece of paper, e.g. A4 or Letter
+ */
+struct page_desc {
+    char *media;
+    int width;
+    int height;
+};
+
+/*
  * some basic PS parameters
  */
 extern int ps_width;	/* number of points in the X direction (8.5 inches) */
@@ -213,14 +222,6 @@
 };
 
 
-/*
- * Structure to describe a physical piece of paper, e.g. A4 or Letter
- */
-struct page_desc {
-    char *media;
-    int width;
-    int height;
-};
     
 
 /* array of sheets where pages are ordered for coli ??? */
