diff -Naur --exclude='*~' muse-3.02.6b/fink/README.fink muse-3.02.6b.fink/fink/README.fink
--- muse-3.02.6b/fink/README.fink	1969-12-31 16:00:00.000000000 -0800
+++ muse-3.02.6b.fink/fink/README.fink	2006-01-21 18:14:47.000000000 -0800
@@ -0,0 +1,44 @@
+To use muse mode, include the form
+
+  (require 'muse)
+
+in your emacs initialization file.  If you wish to publish your muse
+documents, look at the list of output formats; for each scuh format,
+there is a corresponding emacs lisp file which provides the ability
+to transform muse files to that format.  For example, to output muse
+documents to HTML format, add the form
+
+  (require 'muse-html)
+
+to your emacs initialization file.  To build a wiki with muse, add the
+form
+
+  (require 'muse-wiki)
+
+to your emacs initialization file.  See the info docs for more
+information about the output formats supported by muse.
+
+The customization group for muse is simply `muse'.  Thus, to begin
+customization, execute the key sequence
+
+  M-x customize-group RET muse RET
+
+The example in @PREFIX@/share/doc/muse-el/examples/Quickstart.muse is
+a muse document describing muse itself.  Try opening it in emacs and
+publishing it in various formats using the command
+
+  M-x muse-publish-this-file RET
+
+to get a sense of what muse can do and what its syntax looks like.
+
+The directory @PREFIX@/share/doc/muse-el/examples also contains other
+useful stuff, such as muse configuration files designed by John
+Wiegley, inventor of muse, and Michael Olson, the current maintainer
+of muse (not to be confused with the current Fink maintainer, Jesse
+Alama).  Include the contents of either of these files (but not both)
+in your emacs initialization file to get a sense of how muse can be
+configured.  The support files included in Michael Olson's directory
+(i.e., @PREFIX@/share/doc/muse-el/examples/mwolson) also contains CSS
+stylesheets as well as header and footer files that Michael Olson
+actually uses to maintain his website (http://www.mwolson.org) with
+muse.
diff -Naur --exclude='*~' muse-3.02.6b/fink/emacsen-install muse-3.02.6b.fink/fink/emacsen-install
--- muse-3.02.6b/fink/emacsen-install	1969-12-31 16:00:00.000000000 -0800
+++ muse-3.02.6b.fink/fink/emacsen-install	2006-06-09 18:53:36.000000000 -0700
@@ -0,0 +1,52 @@
+#!/bin/bash -e
+#
+# install file for the fink muse-el emacs package.
+
+set -o posix
+
+FLAVOR=${1}
+
+echo >&2 "install/muse-el: Handling install of emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "install/muse-el: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+if [ ${FLAVOR} == emacs ]
+then
+    echo "install/muse-el: Done."
+    exit 0
+fi
+
+echo >&2 -n "install/muse-el: Byte-compiling for ${FLAVOR}..."
+
+mkdir -p @PREFIX@/share/${FLAVOR}/site-lisp/muse;
+for i in muse-blosxom.el muse-book.el muse-colors.el muse-convert.el muse-docbook.el muse-html.el muse-http.el muse-journal.el muse-latex.el muse-mode.el muse-poem.el muse-project.el muse-protocols.el muse-publish.el muse-regexps.el muse-texinfo.el muse-wiki.el muse.el Makefile Makefile.defs muse-build.el; do 
+    ln -s -f @PREFIX@/share/emacs/site-lisp/muse/${i} @PREFIX@/share/${FLAVOR}/site-lisp/muse/${i};
+done
+
+mkdir -p @PREFIX@/share/${FLAVOR}/site-lisp/muse/contrib;
+for i in cgi.el httpd.el; do
+    ln -s -f @PREFIX@/share/emacs/site-lisp/muse/contrib/${i} @PREFIX@/share/${FLAVOR}/site-lisp/muse/contrib/${i};
+done
+
+mkdir -p @PREFIX@/share/${FLAVOR}/site-lisp/muse/scripts;
+ln -s -f @PREFIX@/share/emacs/site-lisp/muse/scripts/muse-build.el @PREFIX@/share/${FLAVOR}/site-lisp/muse/scripts;
+
+cd @PREFIX@/share/${FLAVOR}/site-lisp/muse;
+
+(make EMACS=${FLAVOR} PREFIX="@PREFIX@" INFODIR="@PREFIX@/share/info" lisp --keep-going 2>&1) | gzip -9qf > @PREFIX@/share/doc/muse-el/CompilationLog-${FLAVOR}.gz
+
+rm @PREFIX@/share/${FLAVOR}/site-lisp/muse/scripts/muse-build.el
+rmdir @PREFIX@/share/${FLAVOR}/site-lisp/muse/scripts
+
+rm @PREFIX@/share/${FLAVOR}/site-lisp/muse/Makefile;
+rm @PREFIX@/share/${FLAVOR}/site-lisp/muse/Makefile.defs;
+
+
+echo >&2 "done."
+echo >&2 "install/muse-el: Compilation log saved in @PREFIX@/share/doc/muse-el/CompilationLog-${FLAVOR}.gz."
+
+exit 0
diff -Naur --exclude='*~' muse-3.02.6b/fink/emacsen-remove muse-3.02.6b.fink/fink/emacsen-remove
--- muse-3.02.6b/fink/emacsen-remove	1969-12-31 16:00:00.000000000 -0800
+++ muse-3.02.6b.fink/fink/emacsen-remove	2006-01-05 15:19:56.000000000 -0800
@@ -0,0 +1,21 @@
+#!/bin/bash -e
+#
+# This is the Fink version of the muse-el emacs package.
+set -o posix
+
+FLAVOR=${1}
+
+echo >&2 "remove/muse-el: Handling removal for emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "remove/muse-el: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+echo >&2 -n "remove/muse-el: Purging compilation log and byte-compiled files for ${FLAVOR}..."
+rm -f @PREFIX@/share/doc/muse-el/CompilationLog-${FLAVOR}.gz
+rm -Rf @PREFIX@/share/${FLAVOR}/site-lisp/muse
+echo >&2 "done."
+    
+exit 0
diff -Naur --exclude='*~' muse-3.02.6b/fink/muse-el.copyright muse-3.02.6b.fink/fink/muse-el.copyright
--- muse-3.02.6b/fink/muse-el.copyright	1969-12-31 16:00:00.000000000 -0800
+++ muse-3.02.6b.fink/fink/muse-el.copyright	2006-01-07 00:03:57.000000000 -0800
@@ -0,0 +1,37 @@
+This package was finkified by Jesse Alama <alama@stanford.edu>.
+
+It was downloaded from
+http://www.mwolson.org/projects/MuseMode.html
+
+Upstream Authors: John Wiegley <johnw@gnu.org>
+                  Michael Olson <mwolson@gnu.org>
+                  Sacha Chua <sacha@free.net.ph>
+                  Brad Collins <brad@chenla.org>
+                  Li Daobing <lidaobing@gmail.com>
+                  Yann Hodique <yann.hodique@gmail.com>
+                  Andrea Riciputi <ariciputi@pito.com>
+                  John Sullivan <john@wjsullivan.net>
+                  Gary Vaughan <gary@gnu.org>
+                  Lan Yufeng <nlany.web@gmail.com>
+
+Copyright:
+
+   Copyright (C) 2004, 2005 Free Software Foundation, Inc.
+
+   This package is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This package is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this package; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA,
+   02110-1301, USA.
+
+On Fink systems, the complete text of the GNU General Public License
+can be found in `@PREFIX@/share/doc/fink/COPYING'.
diff -Naur --exclude='*~' muse-3.02.6b/lisp/Makefile muse-3.02.6b.fink/lisp/Makefile
--- muse-3.02.6b/lisp/Makefile	2006-01-16 18:14:23.000000000 -0800
+++ muse-3.02.6b.fink/lisp/Makefile	2006-06-09 18:18:04.000000000 -0700
@@ -1,7 +1,7 @@
 .PHONY: all lisp autoloads clean realclean distclean fullclean install test
 .PRECIOUS: %.elc
 
-include ../Makefile.defs
+include Makefile.defs
 
 EL  = $(filter-out muse-autoloads.el,$(wildcard *.el))
 ELC = $(patsubst %.el,%.elc,$(EL))
@@ -10,17 +10,17 @@
 
 lisp: $(ELC)
 
-muse-build.elc: ../scripts/muse-build.el
+muse-build.elc: scripts/muse-build.el
 	@echo muse-build.el is not byte-compiled
 
 autoloads: muse-autoloads.el
 
 muse-autoloads.el: $(EL)
-	@$(EMACS) -q $(SITEFLAG) -batch -l ../scripts/muse-build.el \
+	@$(EMACS) -q $(SITEFLAG) -batch -l scripts/muse-build.el \
 		-f muse-generate-autoloads . ../contrib ../experimental
 
 %.elc: %.el
-	@$(EMACS) -q $(SITEFLAG) -batch -l ../scripts/muse-build.el \
+	@$(EMACS) -q $(SITEFLAG) -batch -l scripts/muse-build.el \
 		-f batch-byte-compile $<
 
 clean:
@@ -34,5 +34,5 @@
 	install -m 0644 muse-autoloads.el $(EL) $(ELC) $(ELISPDIR)
 
 test: $(ELC)
-	$(EMACS) -q $(SITEFLAG) -batch -l ../scripts/muse-build.el \
+	$(EMACS) -q $(SITEFLAG) -batch -l scripts/muse-build.el \
 		-f muse-elint-files muse-*.el
diff -Naur --exclude='*~' muse-3.02.6b/muse.texi muse-3.02.6b.fink/muse.texi
--- muse-3.02.6b/muse.texi	2006-03-14 11:18:43.000000000 -0800
+++ muse-3.02.6b.fink/muse.texi	2006-06-09 18:45:35.000000000 -0700
@@ -6,7 +6,7 @@
 
 @dircategory Emacs
 @direntry
-* Muse: (muse). Authoring and publishing environment for Emacs.
+* Muse: (muse-el). Authoring and publishing environment for Emacs.
 @end direntry
 
 @syncodeindex fn cp
