--- OpenSP-1.5.1/include/RangeMap.cxx~	2005-03-14 00:10:36.000000000 +0900
+++ OpenSP-1.5.1/include/RangeMap.cxx	2005-03-14 00:11:17.000000000 +0900
@@ -7,6 +7,7 @@
 #include "RangeMap.h"
 #include "ISet.h"
 #include "types.h"
+#include "constant.h"
 
 #ifdef SP_NAMESPACE
 namespace SP_NAMESPACE {
--- OpenSP-1.5.1/include/InternalInputSource.h~	2008-12-09 19:57:29.000000000 -0600
+++ OpenSP-1.5.1/include/InternalInputSource.h	2008-12-09 20:05:18.000000000 -0600
@@ -34,7 +34,7 @@
   void pushCharRef(Char ch, const NamedCharRef &);
   Boolean rewind(Messenger &);
   const StringC *contents();
-  InternalInputSource *InternalInputSource::asInternalInputSource();
+  InternalInputSource *asInternalInputSource();
   ~InternalInputSource();
 private:
   InternalInputSource(const InternalInputSource &); // undefined
