--- tex4ht-20050228-3/bin/unix/mk4ht	2005-08-21 17:27:18.000000000 +0200
+++ tex4ht-20050228-3/bin/unix/mk4ht	2005-09-08 10:45:19.000000000 +0200
@@ -32,11 +32,11 @@
  "uxh", "uxhlatex",  "latex",   "xhtml,uni-html4", " -cunihtf", "-cvalidate",
  "uxh", "uxhtex",    "tex",     "xhtml,uni-html4", " -cunihtf", "-cvalidate",
  "uxh", "uxhtexi",   "texi",    "xhtml,uni-html4", " -cunihtf", "-cvalidate",
- "uxh", "uxhtexexec","texexec,  "xhtml,uni-html4", " -cunihtf", "-cvalidate",
+ "uxh", "uxhtexexec","texexec",  "xhtml,uni-html4", " -cunihtf", "-cvalidate",
  "xhm", "xhmlatex", "latex",  "xhtml,mathml", " -cunihtf", "-cvalidate",
  "xhm", "xhmtex",   "tex",    "xhtml,mathml", " -cunihtf", "-cvalidate",
  "xhm", "xhmtexi",  "texi",   "xhtml,mathml", " -cunihtf", "-cvalidate",
- "xhm", "xhmtexexec,"texexec", "xhtml,mathml", " -cunihtf", "-cvalidate"
+ "xhm", "xhmtexexec","texexec", "xhtml,mathml", " -cunihtf", "-cvalidate",
  "mz", "mzlatex",   "latex",   "xhtml,mozilla", " -cmozhtf",  "-cvalidate",
  "mz", "mztex",     "tex",     "xhtml,mozilla", " -cmozhtf",  "-cvalidate",
  "mz", "mztexi",    "texi",    "xhtml,mozilla", " -cmozhtf",  "-cvalidate",
@@ -49,7 +49,7 @@
  "tei",  "teitex",    "tex",     "xhtml,tei",    " -cunihtf",  "-cvalidate",
  "tei",  "teitexi",   "texi",    "xhtml,tei",    " -cunihtf",  "-cvalidate",
  "tei",  "teitexexec","texexec", "xhtml,tei",    " -cunihtf",  "-cvalidate",
- "teim", "teimlatex", "latex", " "xhtml,tei-mml"," -cunihtf",  "-cvalidate",
+ "teim", "teimlatex", "latex",  "xhtml,tei-mml"," -cunihtf",  "-cvalidate",
  "teim", "teimtex",   "tex",     "xhtml,tei-mml"," -cunihtf",  "-cvalidate",
  "teim", "teimtexi",  "texi",    "xhtml,tei-mml"," -cunihtf",  "-cvalidate",
  "teim", "teimtexexec","texexec","xhtml,tei-mml"," -cunihtf",  "-cvalidate",
