--- texmf/tex4ht/base/unix/tex4ht.env~	2005-08-21 16:57:21.000000000 +0200
+++ texmf/tex4ht/base/unix/tex4ht.env	2005-08-21 16:06:08.000000000 +0200
@@ -132,7 +131,7 @@
 %   magnification: -x mag  (e.g., -x 1200)                          %
 %   page size:     -T x,y  (e.g., -T 14in,14in)                     %
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-<default>
+ <convert>
  G.png
  Gdvips -E -q -Ppdf -f %%1 -pp %%2 > zz%%4.ps
  Ggs -sDEVICE=pngalpha -sOutputFile=%%3 -r110x110 -dEPSCrop -dBackgroundColor=16#ffffff -dTextAlphaBits=2 -dGraphicsAlphaBits=2 -q -dbatch -dNOPAUSE zz%%4.ps -c quit
@@ -152,14 +151,14 @@
   Grm zz%%4.svg           
 G.
 Gdvips -E -Ppdf -mode ibmvga -D 110 -f %%1 -pp %%2  > zz%%4.ps
-Gconvert zz%%4.ps -trim +repage -density 110x110 -transparent '#FFFFFF' %%3
+Gconvert  -trim +repage -density 110x110 zz%%4.ps -transparent '#FFFFFF' %%3
 Grm zz%%4.ps 
 %--------- alternative instructions for old versions of convert
 % Gconvert zz%%4.ps -trim -density 110x110 -transparent "#FFFFFF" %%3
 % Gconvert zz%%4.ps -crop 0x0 +repage -density 110x110 -transparent "#FFFFFF" %%3
 % Gconvert zz%%4.ps -crop 0x0 +page -density 110x110 -transparent "#FFFFFF" %%3
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%  
-</default>
+ </convert>
 <netpbm>
 Gdvips -Ppdf -mode ibmvga -D 110 -f %%1 -pp %%2  > zz%%4.ps
 Ggs -sDEVICE=ppm -r110 -sOutputFile=zz%%4.ppm -q -dbatch -dTextAlphaBits=4 -dGraphicsAlphaBits=4 -dNOPAUSE zz%%4.ps -c quit
@@ -171,6 +170,17 @@
 Grm temp.ppm
 % Gppmtogif -interlace -transparent 1,1,1 zz%%4.ppm > %%3 
 </netpbm>
+<dvipng>
+G.png
+Gdvipng -T tight -x 1400 -D 72 -bg Transparent -pp %%2:%%2 %%1 -o %%3
+G.gif
+Gdvipng -T tight -x 1400 -D 72 -bg Transparent -pp -gif %%2:%%2 %%1 -o %%3
+G.
+Gdvips -Ppdf -mode ibmvga -D 110 -f   > zz%%4.ps
+Gconvert -crop 0x0 -density 110x110 -transparent '#FFFFFF' zz%%4.ps %%3
+Grm zz%%4.ps
+</dvipng>
+
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 % Default scripts, shifted rightward 1 pos
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
