diff -Naur texlive-20080816-source.orig/libs/lua51/lcoco.c texlive-20080816-source/libs/lua51/lcoco.c
--- texlive-20080816-source.orig/texk/web2c/luatexdir/lua51/lcoco.c	2008-03-25 00:47:15.000000000 +0900
+++ texlive-20080816-source/texk/web2c/luatexdir/lua51/lcoco.c	2009-10-11 04:14:04.000000000 +0900
@@ -331,7 +331,7 @@
 /* Else fallback to ucontext. Slower, because it saves/restores signals. */
 #else /* !defined(COCO_MAKECTX) */
 
-#include <ucontext.h>
+#include <sys/ucontext.h>
 
 #define COCO_CTX		ucontext_t
 
diff -Naur texlive-20080816-source.orig/texk/kpathsea/texmf.cnf texlive-20080816-source/texk/kpathsea/texmf.cnf
--- texlive-20080816-source.orig/texk/kpathsea/texmf.cnf	2008-07-27 06:39:36.000000000 +0900
+++ texlive-20080816-source/texk/kpathsea/texmf.cnf	2009-05-10 21:31:18.000000000 +0900
@@ -44,15 +44,15 @@
 
 % The tree containing the runtime files closely related to the specific
 % program version used:
-TEXMFMAIN = $SELFAUTOPARENT/texmf
+TEXMFMAIN = @PREFIX@/share/texmf
 
 % The main distribution tree:
-TEXMFDIST = $SELFAUTOPARENT/texmf-dist
+TEXMFDIST = @PREFIX@/share/texmf-dist
 
 % A place for local additions to a "standard" texmf tree.
 % This tree is not used for local configuration maintained by
 % texconfig, it uses TEXMFCONFIG below.
-TEXMFLOCAL = $SELFAUTOPARENT/../texmf-local
+TEXMFLOCAL = @PREFIX@/share/texmf-local
 
 % TEXMFSYSVAR, where texconfig-sys stores variable runtime data.
 % With teTeX-3.0 or later, this must be set.
@@ -63,7 +63,7 @@
 %   TEXMFSYSVAR = $TEXMFMAIN
 % For using a separate tree:
 %   TEXMFSYSVAR = $SELFAUTOPARENT/texmf-var
-TEXMFSYSVAR = $SELFAUTOPARENT/texmf-var
+TEXMFSYSVAR = @PREFIX@/var/lib/texmf
 
 % TEXMFSYSCONFIG, where texconfig-sys stores configuration data.
 % With teTeX-3.0 or later, this must be set.
@@ -71,11 +71,11 @@
 %   TEXMFSYSCONFIG = $TEXMFMAIN
 % For using a separate tree:
 %   TEXMFSYSCONFIG = $SELFAUTOPARENT/texmf-config
-TEXMFSYSCONFIG = $SELFAUTOPARENT/texmf-config
+TEXMFSYSCONFIG = @PREFIX@/etc/texmf-config
 
 % User texmf trees are allowed as follows.
 % This used to be HOMETEXMF.
-TEXMFHOME = ~/texmf
+TEXMFHOME = ~/Library/texmf
 
 % TEXMFVAR, where texconfig stores variable runtime data.
 % With teTeX-3.0 or later, this must be set.
@@ -83,7 +83,7 @@
 %   TEXMFVAR = $TEXMFMAIN
 % For using a separate tree:
 %   TEXMFVAR = ~/.texmf-var  # teTeX 3.0 default
-TEXMFVAR = ~/.texlive2010/texmf-var
+TEXMFVAR = ~/.texmf-var
 
 % TEXMFCONFIG, where texconfig stores configuration data.
 % With teTeX-3.0 or later, this must be set.
@@ -93,7 +93,7 @@
 %   TEXMFCONFIG = ~/.texmf-config  # teTeX 3.0 default
 % For using a separate tree:
 %   TEXMFCONFIG = $SELFAUTOPARENT/texmf-config
-TEXMFCONFIG = ~/.texlive2010/texmf-config
+TEXMFCONFIG = ~/.texmf-config
 
 % Now, list all the texmf trees. If you have multiple trees you can
 % use shell brace notation, like this:
@@ -111,7 +111,7 @@
 % versions should take precedence over those (although it is generally a
 % source of confusion to have different versions of a package installed,
 % whatever the trees, so try to avoid it).
-TEXMF = {$TEXMFCONFIG,$TEXMFVAR,$TEXMFHOME,!!$TEXMFSYSCONFIG,!!$TEXMFSYSVAR,!!$TEXMFMAIN,!!$TEXMFLOCAL,!!$TEXMFDIST}
+TEXMF = {$TEXMFCONFIG,$TEXMFVAR,$TEXMFHOME,!!$TEXMFSYSCONFIG,!!$TEXMFSYSVAR,!!$TEXMFLOCAL,!!$TEXMFMAIN,!!$TEXMFDIST}
 
 % Where to look for ls-R files.  There need not be an ls-R in the
 % directories in this path, but if there is one, Kpathsea will use it.
@@ -125,18 +125,18 @@
 % does not create ls-R files in the non-!! elements -- because if an
 % ls-R is present, it will be used, and the disk will not be searched.
 % This is arguably a bug in kpathsea.
-TEXMFDBS = {!!$TEXMFSYSCONFIG,!!$TEXMFSYSVAR,!!$TEXMFMAIN,!!$TEXMFLOCAL,!!$TEXMFDIST}
+TEXMFDBS = {!!$TEXMFSYSCONFIG,!!$TEXMFSYSVAR,!!$TEXMFLOCAL,!!$TEXMFMAIN,!!$TEXMFDIST}
 
 % The system trees.  These are the trees that are shared by all the users.
 % If a tree appears in this list, the mktex* scripts will use
 % VARTEXFONTS for generated files, if the original tree isn't writable;
 % otherwise the current working directory is used.
-SYSTEXMF = $TEXMFSYSVAR;$TEXMFMAIN;$TEXMFLOCAL;$TEXMFDIST
+SYSTEXMF = $TEXMFSYSVAR;$TEXMFLOCAL;$TEXMFMAIN;$TEXMFDIST
 
 % Where generated fonts may be written.  This tree is used when the sources
 % were found in a system tree and either that tree wasn't writable, or the
 % varfonts feature was enabled in MT_FEATURES in mktex.cnf.
-VARTEXFONTS = $TEXMFVAR/fonts
+VARTEXFONTS = $TEXMFSYSVAR/fonts
 
 % On some systems, there will be a system tree which contains all the font
 % files that may be created as well as the formats.  For example
@@ -233,13 +233,21 @@
 TEXINPUTS.lamed         = .;$TEXMF/tex/{lamed,lambda,latex,generic,}//
 TEXINPUTS.omega         = .;$TEXMF/tex/{plain,generic,}//
 
-% pTeX
+% pTeX family
 TEXINPUTS.ptex          = .;$TEXMF/tex/{ptex,ptexgeneric,plain,generic,}//
 TEXINPUTS.platex        = .;$TEXMF/tex/{platex,ptexgeneric,latex,generic,}//
+TEXINPUTS.eptex          = .;$TEXMF/tex/{ptex,ptexgeneric,plain,generic,}//
+TEXINPUTS.eplatex        = .;$TEXMF/tex/{platex,ptexgeneric,latex,generic,}//
+TEXINPUTS.uptex          = .;$TEXMF/tex/{ptex,ptexgeneric,plain,generic,}//
+TEXINPUTS.uplatex        = .;$TEXMF/tex/{platex,ptexgeneric,latex,generic,}//
+TEXINPUTS.euptex          = .;$TEXMF/tex/{ptex,ptexgeneric,plain,generic,}//
+TEXINPUTS.euplatex        = .;$TEXMF/tex/{platex,ptexgeneric,latex,generic,}//
 
 % pBibTeX bibliographies and style files
 BIBINPUTS.pbibtex       = .;$TEXMF/{pbibtex,bibtex}/bib//
 BSTINPUTS.pbibtex       = .;$TEXMF/{pbibtex,bibtex}/bst//
+BIBINPUTS.upbibtex       = .;$TEXMF/{pbibtex,bibtex}/bib//
+BSTINPUTS.upbibtex       = .;$TEXMF/{pbibtex,bibtex}/bst//
 
 % ConTeXt.
 TEXINPUTS.context       = .;$TEXMF/tex/{context,plain,generic,}//
@@ -390,8 +390,8 @@
 % For xdvi to find mime.types and .mailcap, if they do not exist in
 % ~.  These are single directories, not paths.
 % (But the default mime.types, at least, may well suffice.)
-MIMELIBDIR = $SELFAUTOPARENT/etc
-MAILCAPLIBDIR = $SELFAUTOPARENT/etc
+MIMELIBDIR = @PREFIX@/etc
+MAILCAPLIBDIR = @PREFIX@/etc
 
 % Default settings for fontconfig library, used by Windows versions of
 % xetex/xdvipdfmx.  On Unixish systems, fontconfig ignores this.
@@ -447,7 +455,7 @@
 % 
 % For security reasons, it is better not to include . in this path.
 %
-TEXMFCNF = {$SELFAUTOLOC,$SELFAUTODIR,$SELFAUTOPARENT}{,{/share,}/texmf{-local,}/web2c}
+TEXMFCNF = {$SELFAUTOLOC,$SELFAUTODIR,$SELFAUTOPARENT,@PREFIX@}{,{/share,}/texmf{-local,}/web2c}
 
 % kpathsea 3.5.3 and later sets these at runtime. To avoid empty
 % expansions from binaries linked against an earlier version of the
@@ -645,3 +653,6 @@
 line_length.gftype = 500
 max_rows.gftype = 8191
 max_cols.gftype = 8191
+
+% for ptex filter
+PTEX_IN_FILTER = @PREFIX@/bin/nkf -w
diff -Naur texlive-20100720-source.orig/texk/tetex/fmtutil.cnf texlive-20100720-source/texk/tetex/fmtutil.cnf
--- texlive-20100720-source.orig/texk/tetex/fmtutil.cnf	2011-01-15 20:50:23.000000000 +0900
+++ texlive-20100720-source/texk/tetex/fmtutil.cnf	2011-01-16 10:52:00.000000000 +0900
@@ -110,6 +110,12 @@
 # from ptex:
 ptex ptex - ptex.ini
 platex ptex language.ptx platex.ini
+#! eptex eptex - -etex eptex.src
+#! eplatex eptex language.ptx -etex platex.ini
+#! uptex uptex - uptex.ini
+#! uplatex uptex language.ptx uplatex.ini
+#! euptex euptex - -etex euptex.src
+#! euplatex euptex language.ptx -etex uplatex.ini
 #
 # from t2:
 #! cyramstex pdftex language.dat -translate-file=cp227.tcx *cyramstx.ini
