diff -u uni2ascii-3.9.5.orig/Makefile.am uni2ascii-3.9.5/Makefile.am
--- uni2ascii-3.9.5.orig/Makefile.am	2006-05-25 21:48:45.000000000 -0700
+++ uni2ascii-3.9.5/Makefile.am	2006-08-11 17:10:14.000000000 -0700
@@ -4,7 +4,7 @@
 uni2ascii_SOURCES = Get_UTF32_From_UTF8.c uni2ascii.c enttbl.c endian.c
 ascii2uni_SOURCES = ascii2uni.c GetWord.c putu8.c enttbl.c
 noinst_HEADERS = enttbl.h exitcode.h unicode.h utf8error.h endian.h
-LOCALEDIR=/usr/local/share/locale
+LOCALEDIR=@PREFIX@/share/locale
 AM_CFLAGS = -DLOCALEDIR=\"${LOCALEDIR}\"
 AUTOMAKE_OPTIONS = dist-zip dist-bzip2
 EXTRA_DIST = TestSuiteAscii2Uni CREDITS uni2html.py ascii2uni.py u2a.tcl Equivalences
diff -u uni2ascii-3.9.5.orig/Makefile.in uni2ascii-3.9.5/Makefile.in
--- uni2ascii-3.9.5.orig/Makefile.in	2006-07-05 16:49:57.000000000 -0700
+++ uni2ascii-3.9.5/Makefile.in	2006-08-11 17:10:51.000000000 -0700
@@ -168,7 +168,7 @@
 uni2ascii_SOURCES = Get_UTF32_From_UTF8.c uni2ascii.c enttbl.c endian.c
 ascii2uni_SOURCES = ascii2uni.c GetWord.c putu8.c enttbl.c
 noinst_HEADERS = enttbl.h exitcode.h unicode.h utf8error.h endian.h
-LOCALEDIR = /usr/local/share/locale
+LOCALEDIR = @PREFIX@/share/locale
 AM_CFLAGS = -DLOCALEDIR=\"${LOCALEDIR}\"
 AUTOMAKE_OPTIONS = dist-zip dist-bzip2
 EXTRA_DIST = TestSuiteAscii2Uni CREDITS uni2html.py ascii2uni.py u2a.tcl Equivalences
