--- yamcha-0.33/training/Makefile.in.orig	2004-09-20 02:59:16.000000000 -0700
+++ yamcha-0.33/training/Makefile.in	2007-06-04 21:34:55.000000000 -0700
@@ -2,7 +2,7 @@
 #  $Id: Makefile.in,v 1.12 2004/09/20 09:59:16 taku-ku Exp $;
 #
 prefix     = @prefix@
-TOOLDIR    = @prefix@/libexec/yamcha
+TOOLDIR    = @libexecdir@/yamcha
 SVM_PARAM  = -t 1 -d 2 -c 1
 FEATURE    = F:-2..2:0.. T:-2..-1
 DIRECTION  =
@@ -53,7 +53,7 @@
 	$(PERL) $(TOOLDIR)/mkmodel -t $(TOOLDIR) $(MODEL).txtmodel.gz $(MODEL).model
 
 install:
-	@INSTALL@ -m 644 Makefile $(DESTDIR)$(prefix)/libexec/yamcha/Makefile
+	@INSTALL@ -m 644 Makefile $(DESTDIR)$(TOOLDIR)/Makefile
 
 check:
 clean:
