--- barnyard/src/barnyard.c.orig	2003-04-22 20:06:33.000000000 +0200
+++ barnyard/src/barnyard.c	2003-04-22 20:07:20.000000000 +0200
@@ -227,7 +227,7 @@
      * one is needed earlier ;)
      */
     if(pv.config_file == NULL)
-        pv.config_file = strdup("/etc/snort/barnyard.conf");
+        pv.config_file = strdup("@PREFIX@/etc/snort/barnyard.conf");
  
     /* attempt to extract config dir from the -c arg */
      if(strchr(pv.config_file, '/'))
