diff -ruN gftp-2.0.14.orig/lib/gftp.h gftp-2.0.14/lib/gftp.h
--- gftp-2.0.14.orig/lib/gftp.h	2002-12-03 19:35:44.000000000 -0700
+++ gftp-2.0.14/lib/gftp.h	2002-12-06 12:35:24.000000000 -0700
@@ -35,7 +35,9 @@
 #endif
 #include <sys/wait.h>
 #include <sys/utsname.h>
-#include <sys/sysmacros.h>
+#ifndef __APPLE__ && __GNUC
+# include <sys/sysmacros.h>
+#endif
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <glib.h>
