--- gnomeicu-0.99/src/userserver.c.orig	Fri Apr 18 00:22:00 2003
+++ gnomeicu-0.99/src/userserver.c	Fri Apr 18 00:22:18 2003
@@ -28,12 +28,12 @@
 #include <gdk/gdk.h>
 #include <gdk/gdkkeysyms.h>
 #include <errno.h>
+#include <sys/types.h>
 #include <netinet/in.h>
 #include <pwd.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
 #include <sys/un.h>
--- gnomeicu-0.99/src/Makefile.in.orig	Fri Apr 18 00:30:57 2003
+++ gnomeicu-0.99/src/Makefile.in	Fri Apr 18 00:40:49 2003
@@ -272,7 +272,8 @@
 	$(GNOMEICU_LIBS)	\
 	$(SOCKS_LIBDIR)		\
 	$(SOCKS_LIBS)		\
-	$(LIBDBM)
+	$(LIBDBM) \
+	/usr/X11R6/lib/libX11.6.dylib
 
 subdir = src
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
--- gnomeicu-0.99/src/Makefile.in.orig	Fri Apr 18 01:07:03 2003
+++ gnomeicu-0.99/src/Makefile.in	Fri Apr 18 01:07:48 2003
@@ -304,7 +304,7 @@
 gnomeicu_DEPENDENCIES =
 am_gnomeicu_client_OBJECTS = gnomeicu-client.$(OBJEXT)
 gnomeicu_client_OBJECTS = $(am_gnomeicu_client_OBJECTS)
-gnomeicu_client_LDADD = $(LDADD)
+gnomeicu_client_LDADD = $(LDADD) -lintl
 gnomeicu_client_DEPENDENCIES =
 gnomeicu_client_LDFLAGS =
 
