diff -ru nmap-3.26/configure nmap-3.26-patched/configure
--- nmap-3.26/configure	Fri Apr 25 14:46:45 2003
+++ nmap-3.26-patched/configure	Fri Apr 25 17:28:39 2003
@@ -5528,7 +5528,7 @@
                  You need at least version $GTK_MINVERSION.\n Subdir X-windows \
                  will not be made" >&2;}
   else
-    subdirs="$subdirs nmapfe"
+    subdirs="$subdirs"
                       fi
 fi
 fi
