diff -u -Naur noffle-1.1.3/Makefile.in noffle-1.1.3.fink/Makefile.in
--- noffle-1.1.3/Makefile.in	Fri Jan 31 10:41:19 2003
+++ noffle-1.1.3.fink/Makefile.in	Mon Feb 10 12:17:26 2003
@@ -369,7 +369,8 @@
 	$(INSTALL_DATA) -o 0 -g 0 $(srcdir)/NEWS $(DESTDIR)$(DOCDIR)
 	$(INSTALL_DATA) -o 0 -g 0 $(srcdir)/README $(DESTDIR)$(DOCDIR)
 	$(INSTALL_DATA) -o 0 -g 0 $(srcdir)/TODO $(DESTDIR)$(DOCDIR)
-	$(INSTALL_DATA) -D -o 0 -g 0 $(srcdir)/noffle.conf.example \
+	mkdir -p `dirname $(DESTDIR)$(CONFIGFILE)`;
+	$(INSTALL_DATA) -o 0 -g 0 $(srcdir)/noffle.conf.example \
                                   $(DESTDIR)$(CONFIGFILE).example
 	chown news.news $(DESTDIR)$(CONFIGFILE).example
 	chmod 0640 $(DESTDIR)$(CONFIGFILE).example
diff -u -Naur noffle-1.1.3/src/authenticate.c noffle-1.1.3.fink/src/authenticate.c
--- noffle-1.1.3/src/authenticate.c	Sun Jan 12 12:05:49 2003
+++ noffle-1.1.3.fink/src/authenticate.c	Mon Feb 10 10:57:44 2003
@@ -11,6 +11,7 @@
 #endif
 
 #include <errno.h>
+#include <sys/types.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
