diff -ru -x *~ ntop-1.1/Makefile.in ntop-1.1-patched/Makefile.in
--- ntop-1.1/Makefile.in	Mon May  3 10:03:10 1999
+++ ntop-1.1-patched/Makefile.in	Tue Nov  6 22:56:58 2001
@@ -104,10 +104,10 @@
 http.c: ntop.h
 report.c: ntop.h
 
-install:
+install: $(PROG)
 	$(INSTALL) -m 6550 $(PROG) $(DESTDIR)$(BINDEST)/$(PROG)
 
-install-man:
+install-man: $(PROG).8
 	$(INSTALL) -m 444 $(PROG).8 $(DESTDIR)$(MANDEST)/man8/$(PROG).8
 
 clean:
