diff -uNr opennms-1.1.5-1/source/etc/rrd-configuration.properties opennms-1.1.5-1-new/source/etc/rrd-configuration.properties
--- opennms-1.1.5-1/source/etc/rrd-configuration.properties	Wed Jan 26 17:49:01 2005
+++ opennms-1.1.5-1-new/source/etc/rrd-configuration.properties	Thu Jan 27 22:02:28 2005
@@ -13,7 +13,7 @@
 # to Tomcat start up so that the JRobin code can properly generate the graphs.
 #
 # The default setting is true (use the JNI code)
-#org.opennms.rrd.usejni=true
+org.opennms.rrd.usejni=false
 
 #
 # This property defines whether or not to use the write queuing system.  
diff -uNr opennms-1.1.5-1/source/src/services/org/opennms/protocols/icmp/IcmpSocket.c opennms-1.1.5-1-new/source/src/services/org/opennms/protocols/icmp/IcmpSocket.c
--- opennms-1.1.5-1/source/src/services/org/opennms/protocols/icmp/IcmpSocket.c	Wed Jan 26 17:49:02 2005
+++ opennms-1.1.5-1-new/source/src/services/org/opennms/protocols/icmp/IcmpSocket.c	Thu Jan 27 22:02:28 2005
@@ -10,6 +10,7 @@
 
  Modifications:
 
+ 2004 Oct 27: Handle Darwin 10.2 gracefully.
  2003 Sep 07: More Darwin tweaks.
  2003 Apr 26: Fixes byteswap issues on Solaris x86.
  2003 Mar 25: Used unt64_t instead of unsigned long long.
@@ -50,9 +51,14 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <sys/types.h>
+#include <stdint.h>
 #include <sys/socket.h>
 #if defined(__DARWIN__) || defined(__SOLARIS__) || defined (__FreeBSD__)
-#include <netinet/in_systm.h>
+# include <netinet/in_systm.h>
+# include <AvailabilityMacros.h>
+# ifndef MAC_OS_X_VERSION_10_3
+#  define socklen_t int
+# endif
 #endif
 #include <netinet/in.h>
 #include <netinet/ip.h>
diff -uNr opennms-1.1.5-1/source/tools/packages/fink/README.Darwin opennms-1.1.5-1-new/source/tools/packages/fink/README.Darwin
--- opennms-1.1.5-1/source/tools/packages/fink/README.Darwin	Wed Jan 26 17:49:02 2005
+++ opennms-1.1.5-1-new/source/tools/packages/fink/README.Darwin	Fri Jan 28 15:23:11 2005
@@ -43,7 +43,7 @@
 ------------------------------------------------------------------------------
 
 Once you do that, you will need to edit
-@PREFIX@/var/postgresql/data/postgresql.conf (as root, or postgres), and
+@PREFIX@/var/postgresql-7.4/data/postgresql.conf (as root, or postgres), and
 make sure that the following values are set:
 
   tcpip_socket    = true
@@ -61,7 +61,7 @@
 have to start it ("sudo @PREFIX@/bin/pgsql.sh start") and try again.  To run
 the installer manually, run:
 
-  sudo @PREFIX@/var/opennms/bin/install -r -disU
+  sudo @PREFIX@/var/opennms/bin/install -disU
 
 ------------------------------------------------------------------------------
 Startup
