--- p0f/config.h	Wed Sep  3 14:56:43 2003
+++ p0f-patched/config.h	Fri Sep  5 14:23:50 2003
@@ -13,13 +13,17 @@
 #define _HAVE_CONFIG_H
 
-#define VER		"2.0.1"
+#ifndef VER
+#  define VER		"2.0.1"
+#endif /* VER */
 
 /* Paths and names to config files */
 
-#ifdef WIN32
-#  define CONFIG_DIR	"."
-#else
-#  define CONFIG_DIR	"/etc"
-#endif /* WIN32 */
+#ifndef CONFIG_DIR
+#  ifdef WIN32
+#    define CONFIG_DIR	"."
+#  else
+#    define CONFIG_DIR	"/etc"
+#  endif /* WIN32 */
+#endif /* CONFIG_DIR */
 
 #define SYN_DB		"p0f.fp"
