diff -ru2 p0f-1.8.2.2/Makefile p0f-1.8.2.2-patched/Makefile
--- p0f-1.8.2.2/Makefile	Mon May 13 17:20:19 2002
+++ p0f-1.8.2.2-patched/Makefile	Fri Oct 11 16:19:05 2002
@@ -4,8 +4,8 @@
 # (C) Copyright 2001 by William Stearns <wstearns@pobox.com>
 #
-
-CC      = gcc
-CLIBS	= -lpcap -I/usr/include/pcap
-MYSQLCLIBS	= -L/usr/local/mysql/lib -lmysqlclient -lpcap -I/usr/include/pcap -I/usr/local/mysql/include
+#DESTDIR=@PREFIX@
+CC      = cc
+CLIBS	= -lpcap -L@PREFIX@/lib -I@PREFIX@/include -I/usr/include/pcap
+MYSQLCLIBS	=  -bind_at_load -lmysqlclient -L@PREFIX@/lib/mysql -L/usr/local/mysql/lib -I@PREFIX@/include/mysql -I/usr/local/mysql/include
 SUNLIBS	= -lsocket -lnsl -D_SUN_=1
 STRIP	= strip
@@ -23,5 +23,5 @@
 
 mysql: p0f.c
-	$(CC) $(CFLAGS) -DVER=\"$(VERSION)\" -o $@ p0f.c $(MYSQLCLIBS) \
+	$(CC) $(CFLAGS) -DVER=\"$(VERSION)\" -o p0f p0f.c $(CLIBS) $(MYSQLCLIBS) \
 	`uname|egrep -i 'sunos|solar' >/dev/null && echo "$(SUNLIBS)"` -D__MYSQL__
 
@@ -48,19 +48,26 @@
 
 install: $(FILE)
-	mkdir -p $(DESTDIR)/usr/bin
-	mkdir -p $(DESTDIR)/usr/sbin
-	mkdir -p $(DESTDIR)/usr/share/doc/p0f-$(VERSION)
-	mkdir -p $(DESTDIR)/usr/share/man/man1
+	mkdir -p $(DESTDIR)/bin
+	mkdir -p $(DESTDIR)/sbin
+	mkdir -p $(DESTDIR)/share/doc/p0f-$(VERSION)
+	mkdir -p $(DESTDIR)/share/man/man1
+	mkdir -p $(DESTDIR)/etc/init.d/p0f
 	cp -p p0f.fp $(DESTDIR)/etc
 	cp -p p0f.init $(DESTDIR)/etc/init.d/p0f
-	cp -p COPYING README CREDITS ChangeLog $(DESTDIR)/usr/share/doc/p0f-$(VERSION)
-	cp -p p0f $(DESTDIR)/usr/sbin
-	cp -p p0frep $(DESTDIR)/usr/bin
+	cp -p COPYING README CREDITS ChangeLog $(DESTDIR)/share/doc/p0f-$(VERSION)
+	cp -p p0f $(DESTDIR)/sbin
+	cp -p p0frep $(DESTDIR)/bin
 	cp -p p0f.1 p0f.1.orig
 	rm -f p0f.1.gz
 	gzip -9 p0f.1
 	mv p0f.1.orig p0f.1
-	mv p0f.1.gz $(DESTDIR)/usr/share/man/man1
-	chmod 755 $(DESTDIR)/etc/init.d/p0f $(DESTDIR)/usr/sbin/p0f $(DESTDIR)/usr/bin/p0frep
+	mv p0f.1.gz $(DESTDIR)/share/man/man1
+	chmod 755 $(DESTDIR)/etc/init.d/p0f $(DESTDIR)/sbin/p0f $(DESTDIR)/bin/p0frep
+
+mysqlinstall: install
+	mkdir -p $(DESTDIR)/etc
+	mkdir -p $(DESTDIR)/share/doc/p0f-$(VERSION)
+	cp -p mysql/p0f-mysql.conf $(DESTDIR)/etc
+	cp -p mysql/db.sql $(DESTDIR)/share/doc/p0f-$(VERSION)
 
 distribs:
diff -ru2 p0f-1.8.2.2/p0f.1 p0f-1.8.2.2-patched/p0f.1
--- p0f-1.8.2.2/p0f.1	Sun May 12 01:30:06 2002
+++ p0f-1.8.2.2-patched/p0f.1	Fri Oct 11 16:19:05 2002
@@ -62,5 +62,5 @@
 .SH FILES
 .TP
-.BI /etc/p0f.fp
+.BI @PREFIX@/etc/p0f.fp
 default Operating System fingerprint file
 .SH AUTHOR
diff -ru2 p0f-1.8.2.2/p0f.init p0f-1.8.2.2-patched/p0f.init
--- p0f-1.8.2.2/p0f.init	Mon May 13 17:30:33 2002
+++ p0f-1.8.2.2-patched/p0f.init	Fri Oct 11 16:52:03 2002
@@ -10,11 +10,21 @@
 # active scanners (nmap, queSO) - without sending anything to this host.
 # processname: p0f
-# pidfile: /var/run/p0f.pid
+prefix=@PREFIX@
+pidfile=${prefix}/var/run/p0f.pid
+lockfile=${prefix}/var/run/p0f.lock
+logfile=${prefix}/var/log/p0f.log
+p0f=${prefix}/sbin/p0f
+p0fmysqlconf='${prefix}/etc/p0f-mysql.conf'
+fingerprintfile=${prefix}/etc/p0f.fp
 
-PATH=/usr/bin:/sbin:/bin:/usr/sbin
+PATH=${prefix}/bin:${prefix}/sbin:/usr/bin:/sbin:/bin:/usr/sbin
 export PATH
 
 # Source function library.
-. /etc/rc.d/init.d/functions
+if [ -f @PREFIX@/etc/init.d/functions ] ; then
+   . @PREFIX@/etc/init.d/functions
+elif [ -f @PREFIX@/etc/rc.d/init.d/functions ] ; then 
+   . @PREFIX@/etc/rc.d/init.d/functions
+fi
 
 case "$1" in
@@ -31,26 +41,26 @@
 
 	#The command in backticks returns all the local IP addresses on this machine.
-	for OneIP in `/sbin/ifconfig 2>/dev/null | grep 'inet addr' | sed -e 's/.*addr://' -e 's/ .*//'` ; do
+	for OneIP in `/sbin/ifconfig 2>/dev/null | grep 'inet ' | sed -e 's/.*inet //' -e 's/ .*//'` ; do
 		BpfFilter="$BpfFilter and not src host $OneIP"
 	done
-	rm -f /var/run/p0f.pid
+	rm -f ${pidfile}
 	#Start up p0f and filter out all packets originating from any of this machines IP's.
-	if [ -e /etc/p0f-mysql.conf ]; then
-		MysqlParam="-m /etc/p0f-mysql.conf"
+	if [ -e ${p0fmysqlconf} ]; then
+		MysqlParam="-m ${p0fmysqlconf}"
 	else
-		MysqlParam=''
+		MysqlParam="-f ${fingerprintfile}"
 	fi
-	nohup /usr/sbin/p0f $MysqlParam -v "$BpfFilter" >>/var/log/p0f 2>&1 &
-	echo $! >/var/run/p0f.pid
-	touch /var/lock/subsys/p0f
+	nohup ${p0f} $MysqlParam -vt "$BpfFilter" >> ${logfile} 2>&1 &
+	echo $! > ${pidfile}
+	touch ${lockfile}
 	echo "done"
 	;;
 
 stop)
-	if [ -f /var/run/p0f.pid ]; then
+	if [ -f ${pidfile} ]; then
 		echo -n "Stopping p0f: "
-		kill -TERM `cat /var/run/p0f.pid`
-		rm -f /var/run/p0f.pid
-		rm -f /var/lock/subsys/p0f
+		kill -TERM `cat ${pidfile}`
+		rm -f ${pidfile}
+		rm -f ${lockfile}
 		echo "done"
 	fi
@@ -64,6 +74,8 @@
 
 status)
-	status p0f
-	exit $?
+#	bsd's rc init status is not available in Darwin
+#	status p0f
+#	exit $?
+	exit 0
 	;;
 
@@ -73,5 +85,6 @@
 
 *)
-	echo "Usage: $0 {start|stop|status|restart}"
+#	echo "Usage: $0 {start|stop|status|restart}"
+	echo "Usage: $0 {start|stop|restart}"
 	exit 1
 	;;
