diff -uNbr sockstat.pl patched/sockstat.pl > /sw/fink/dists/unstable/main/finkin
fo/net/sockstat-1.6.2.5-1.patch
--- sockstat.pl Fri May  2 17:11:51 2003
+++ patched/sockstat.pl Fri May  2 17:12:59 2003
@@ -26,7 +26,7 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
 # THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #
-# $FreeBSD: src/usr.bin/sockstat/sockstat.pl,v 1.6.2.6 2003/05/05 06:44:26 murray Exp $
+# $FreeBSD: src/usr.bin/sockstat/sockstat.pl,v $
 #
 
 use strict;
@@ -53,13 +53,14 @@
     if (!defined($pid = open(PIPE, "-|"))) {
 	die("open(netstat): $!\n");
     } elsif ($pid == 0) {
-	exec("/usr/bin/netstat", "-AanW");
+	exec("/usr/sbin/netstat", "-AanW");
 	die("exec(netstat): $!\n");
     }
     while ($line = <PIPE>) {
-	next unless ($line =~ m/^[0-9a-f]{8} /) || ($line =~ m/^[0-9a-f]{16} /);
+	$line =~ s/^\s+//;
+	next unless ($line =~ m/^[0-9a-f]{7,8} /) || ($line =~ m/^[0-9a-f]{16} /);
 	chomp($line);
-	@fields = split(' ', $line);
+	@fields = split(/\s+/, $line);
 	$netstat{$fields[0]} = [ @fields ];
     }
     close(PIPE)
@@ -74,7 +75,7 @@
     }
     while ($line = <PIPE>) {
 	chomp($line);
-	@fields = split(' ', $line);
+	@fields = split(/\s+/, $line);
 	next if ($fields[4] eq "-");
 	push(@{$fstat{$fields[4]}}, [ @fields ]);
     }
@@ -128,12 +129,12 @@
     my $fsd;			# Fstat data
     my $nsd;			# Netstat data
 
-    foreach $fsd (@{$fstat{"local"}}) {
+    foreach $fsd (@{$fstat{"unix"}}) {
 	$endpoint{$fsd->[6]} = "$fsd->[1]\[$fsd->[2]\]:" .
 	    substr($fsd->[3], 0, -1);
     }
     printf($unix_fmt, "USER", "COMMAND", "PID", "FD", "PROTO", "ADDRESS");
-    foreach $fsd (@{$fstat{"local"}}) {
+    foreach $fsd (@{$fstat{"unix"}}) {
 	next unless defined($fsd->[6]);
 	next if (!$conn && defined($fsd->[8]));
 	next if (!$listen && !defined($fsd->[8]));
