diff -ru2N whois-4.6.0/Makefile whois-4.6.0-patched/Makefile
--- whois-4.6.0/Makefile	Fri Oct 11 06:13:59 2002
+++ whois-4.6.0-patched/Makefile	Mon Dec  9 20:53:20 2002
@@ -1,3 +1,3 @@
-prefix=/usr/local
+prefix=@PREFIX@
 
 OPTS=-O2
diff -ru2N whois-4.6.0/config.h whois-4.6.0-patched/config.h
--- whois-4.6.0/config.h	Thu Nov 28 15:20:58 2002
+++ whois-4.6.0-patched/config.h	Mon Dec  9 20:53:26 2002
@@ -15,7 +15,5 @@
 
 /* Configuration file */
-/*
-#define CONFIG_FILE "/etc/whois.conf"
-*/
+#define CONFIG_FILE "@PREFIX@/etc/whois.conf"
 
 /* autoconf in cpp macros */
diff -ru2N whois-4.6.0/po/Makefile whois-4.6.0-patched/po/Makefile
--- whois-4.6.0/po/Makefile	Sun Sep 23 12:26:04 2001
+++ whois-4.6.0-patched/po/Makefile	Mon Dec  9 20:53:35 2002
@@ -1,4 +1,4 @@
 INSTALL= /usr/bin/install -c
-INSTALLNLSDIR=$(BASEDIR)/usr/share/locale
+INSTALLNLSDIR=$(BASEDIR)$(prefix)/share/locale
 
 PACKAGE = whois
diff -ru2N whois-4.6.0/whois.1 whois-4.6.0-patched/whois.1
--- whois-4.6.0/whois.1	Wed Dec 11 20:44:03 2002
+++ whois-4.6.0-patched/whois.1	Thu Dec 12 21:26:57 2002
@@ -66,5 +66,5 @@
 text and the whois server to use, separated by white space.
 .SH "Files"
-/etc/whois.conf
+@PREFIX@/etc/whois.conf
 .SH "ENVIRONMENT"
 .IP LANG
