diff -udrN xinetd-2.3.11/xinetd/signals.c xinetd-2.3.11-1/xinetd/signals.c
--- xinetd-2.3.11/xinetd/signals.c	Fri Mar  7 13:30:08 2003
+++ xinetd-2.3.11-1/xinetd/signals.c	Wed Apr 30 21:34:04 2003
@@ -368,15 +368,21 @@
 
          case SIGBUS:
             switch(siginfo->si_code) {
+#ifdef BUS_ADRALN
                case BUS_ADRALN:
                   msg(LOG_CRIT, func, "invalid address alignment");
                   break;
+#endif
+#ifdef BUS_ADRERR
                case BUS_ADRERR:
                   msg(LOG_CRIT, func, "nonexistent physical address");
                   break;
+#endif
+#ifdef BUS_OBJERR
                case BUS_OBJERR:
                   msg(LOG_CRIT, func, "object-specific hardware error");
                   break;
+#endif
                default:
                   msg(LOG_CRIT, func, "unknown fault code");
             }
diff -udrN xinetd-2.3.11/xinetd.conf xinetd-2.3.11-1/xinetd.conf
--- xinetd-2.3.11/xinetd.conf	Wed Dec 31 16:00:00 1969
+++ xinetd-2.3.11-1/xinetd.conf	Thu May  1 10:31:04 2003
@@ -0,0 +1,12 @@
+# man xinetd.conf for more information
+
+defaults
+{
+	instances               = 60
+        log_type                = SYSLOG daemon
+        log_on_success		= HOST PID
+        log_on_failure		= HOST
+	cps			= 25 30
+}
+
+includedir @FINKPREFIX@/etc/xinetd.d
