diff -ruNp gwenview-4.13.1-orig/importer/fileutils.cpp gwenview-4.13.1/importer/fileutils.cpp
--- gwenview-4.13.1-orig/importer/fileutils.cpp	2014-05-05 02:21:49.000000000 -0500
+++ gwenview-4.13.1/importer/fileutils.cpp	2014-06-10 06:10:49.000000000 -0500
@@ -39,6 +39,7 @@ Foundation, Inc., 51 Franklin Street, Fi
 #include <errno.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 
 namespace Gwenview
 {
diff -ruNp gwenview-4.13.1-orig/lib/CMakeLists.txt gwenview-4.13.1/lib/CMakeLists.txt
--- gwenview-4.13.1-orig/lib/CMakeLists.txt	2014-05-05 02:21:49.000000000 -0500
+++ gwenview-4.13.1/lib/CMakeLists.txt	2014-06-10 06:09:32.000000000 -0500
@@ -28,7 +28,7 @@ if ("${jpeglib_version}" LESS 80)
     set(GV_JPEG_DIR libjpeg-62)
 endif()
 
-if ("${jpeglib_version}" EQUAL 80)
+if ("${jpeglib_version}" EQUAL 80 OR "${jpeglib_version}" GREATER 80)
     set(GV_JPEG_DIR libjpeg-80)
 endif()
 
