diff -ruN kalgebra-18.08.3-orig/calgebra/CMakeLists.txt kalgebra-18.08.3/calgebra/CMakeLists.txt
--- kalgebra-18.08.3-orig/calgebra/CMakeLists.txt	2018-09-11 00:06:57.000000000 -0500
+++ kalgebra-18.08.3/calgebra/CMakeLists.txt	2021-07-18 08:28:03.000000000 -0500
@@ -13,6 +13,12 @@
 endif()
 
 add_executable(calgebra main.cpp)
+if (APPLE)
+    # calgebra is a CLT non-.app
+    include(ECMMarkNonGuiExecutable)
+    ecm_mark_nongui_executable(calgebra)
+endif ()
+
 target_link_libraries(calgebra Qt5::Core ${READLINE_LIBRARY} KF5::Analitza)
 
 install(TARGETS calgebra ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
