diff -Nurd kalzium-14.12.3.orig/src/kalziumschemetype.cpp kalzium-14.12.3/src/kalziumschemetype.cpp
--- kalzium-14.12.3.orig/src/kalziumschemetype.cpp	2015-02-21 01:35:39.000000000 -0500
+++ kalzium-14.12.3/src/kalziumschemetype.cpp	2021-03-06 23:18:00.000000000 -0500
@@ -23,7 +23,7 @@
 #include "prefs.h"
 
 #ifdef HAVE_OPENBABEL2
-#include <openbabel/mol.h>
+#include <openbabel/elements.h>
 #endif
 
 #include <QBrush>
@@ -407,8 +407,9 @@
 {
     QColor c;
 
-    std::vector<double> color = OpenBabel::etab.GetRGB(el);
-    c.setRgbF(color[0], color[1], color[2]);
+    double red, green, blue;
+    OpenBabel::OBElements::GetRGB(el, &red, &green, &blue);
+    c.setRgbF(red, green, blue);
 
     return QBrush(c);
 }
