diff -ruN kbruch-18.08.3-orig/src/CMakeLists.txt kbruch-18.08.3/src/CMakeLists.txt
--- kbruch-18.08.3-orig/src/CMakeLists.txt	2018-03-12 21:21:23.000000000 -0500
+++ kbruch-18.08.3/src/CMakeLists.txt	2021-10-13 04:09:11.000000000 -0500
@@ -3,6 +3,7 @@
 ########### next target ###############
 
 include_directories( ${CMAKE_CURRENT_BINARY_DIR} )
+include(ECMAddAppIcon)
 
 set(kbruch_SRCS
    PrimeFactorsLineEdit.cpp
@@ -33,6 +34,8 @@
 
 kconfig_add_kcfg_files(kbruch_SRCS settingsclass.kcfgc )
 
+file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/*-apps-kbruch.png")
+ecm_add_app_icon(kbruch_SRCS ICONS ${ICONS_SRCS})
 add_executable(kbruch ${kbruch_SRCS})
 
 target_link_libraries(kbruch
