diff -ruN kdoctools-5.29.0-orig/cmake/FindDocBookXML4.cmake kdoctools-5.29.0/cmake/FindDocBookXML4.cmake
--- kdoctools-5.29.0-orig/cmake/FindDocBookXML4.cmake	2016-12-03 15:01:17.000000000 -0600
+++ kdoctools-5.29.0/cmake/FindDocBookXML4.cmake	2016-12-25 19:08:26.000000000 -0600
@@ -37,7 +37,7 @@
 
 
 if (NOT DocBookXML_FIND_VERSION)
-     set(DocBookXML_FIND_VERSION "4.5")
+     set(DocBookXML_FIND_VERSION "4.5.0")
 endif ()
 
 set (DocBookXML4_DTD_VERSION ${DocBookXML_FIND_VERSION}
@@ -51,6 +51,7 @@
 function (locate_version version found_dir)
 
     set (DTD_PATH_LIST
+        share/xml/dtd/docbookx/${version}
         ${CMAKE_INSTALL_DATAROOTDIR}/xml/docbook/schema/dtd/${version}
         ${CMAKE_INSTALL_DATAROOTDIR}/xml/docbook/xml-dtd-${version}
         ${CMAKE_INSTALL_DATAROOTDIR}/sgml/docbook/xml-dtd-${version}
