diff -ruN kile-2.1.3-orig/src/editorkeysequencemanager.cpp kile-2.1.3/src/editorkeysequencemanager.cpp
--- kile-2.1.3-orig/src/editorkeysequencemanager.cpp	2012-09-22 15:58:08.000000000 -0500
+++ kile-2.1.3/src/editorkeysequencemanager.cpp	2019-09-14 04:57:41.000000000 -0500
@@ -203,7 +203,7 @@
 	{
 		for(int i = 0; i < s.length(); ++i) {
 			QString toCheck = s.right(s.length() - i);
-			if(m_watchedKeySequencesList.contains(toCheck) > 0) {
+			if(m_watchedKeySequencesList.contains(toCheck) != NULL) {
  				m_view->document()->removeText(KTextEditor::Range(m_oldLine, m_oldCol - (s.length() - i - 1), m_oldLine, m_oldCol));
 				m_typedSequence.clear(); // clean m_typedSequence to avoid wrong action triggering if one presses keys without printable character
 				emit detectedTypedKeySequence(toCheck);
