diff -ruN kolf-14.12.3-orig/kcomboboxdialog.cpp kolf-14.12.3/kcomboboxdialog.cpp
--- kolf-14.12.3-orig/kcomboboxdialog.cpp	2014-12-30 02:54:24.000000000 -0600
+++ kolf-14.12.3/kcomboboxdialog.cpp	2020-07-07 07:00:38.000000000 -0500
@@ -94,7 +94,7 @@
 		KConfigGroup *configGroup = new KConfigGroup(config->group("Notification Messages"));
 		prevAnswer = configGroup->readEntry( dontAskAgainName,QString() );
 		if ( !prevAnswer.isEmpty() )
-			if ( _items.contains( prevAnswer ) > 0 )
+			if ( _items.contains( prevAnswer ) != NULL )
 				return prevAnswer;
 	}
 
diff -ruN kolf-14.12.3-orig/newgame.cpp kolf-14.12.3/newgame.cpp
--- kolf-14.12.3-orig/newgame.cpp	2014-12-30 02:54:24.000000000 -0600
+++ kolf-14.12.3/newgame.cpp	2020-07-07 07:00:14.000000000 -0500
@@ -285,7 +285,7 @@
 
 	for (QStringList::const_iterator fileIt = files.begin(); fileIt != files.end(); ++fileIt)
 	{
-		if (names.contains(*fileIt) > 0)
+		if (names.contains(*fileIt) != NULL)
 		{
 			hasDuplicates = true;
 			continue;
