diff -ruN kopete-14.12.3-orig/protocols/jabber/libiris/src/xmpp/xmpp-im/filetransfer.cpp kopete-14.12.3/protocols/jabber/libiris/src/xmpp/xmpp-im/filetransfer.cpp
--- kopete-14.12.3-orig/protocols/jabber/libiris/src/xmpp/xmpp-im/filetransfer.cpp	2015-02-21 14:26:22.000000000 -0600
+++ kopete-14.12.3/protocols/jabber/libiris/src/xmpp/xmpp-im/filetransfer.cpp	2020-07-08 04:03:07.000000000 -0500
@@ -430,7 +430,7 @@
 
 bool FileTransferManager::isActive(const FileTransfer *ft) const
 {
-	return d->list.contains(const_cast<FileTransfer*>(ft)) > 0;
+	return d->list.contains(const_cast<FileTransfer*>(ft)) != NULL;
 }
 
 void FileTransferManager::setDisabled(const QString &ns, bool state)
