diff -ruNp ktouch-4.12.2-orig/src/CMakeLists.txt ktouch-4.12.2/src/CMakeLists.txt
--- ktouch-4.12.2-orig/src/CMakeLists.txt	2014-01-04 06:58:42.000000000 -0500
+++ ktouch-4.12.2/src/CMakeLists.txt	2014-02-18 21:58:05.000000000 -0500
@@ -5,8 +5,8 @@ include(MacroLogFeature)
 find_package(KDeclarative REQUIRED)
 
 # Currently it is called by FindQt4, but call it explicitly
-find_package(X11)
-macro_log_feature(X11_Xkbfile_FOUND "libxkbfile" "X Keyboard Extension" "http://www.x.org/wiki/XKB" FALSE "" "Support for keyboard layout auto-detection in KTouch")
+#find_package(X11)
+#macro_log_feature(X11_Xkbfile_FOUND "libxkbfile" "X Keyboard Extension" "http://www.x.org/wiki/XKB" FALSE "" "Support for keyboard layout auto-detection in KTouch")
 
 add_subdirectory(qml)
 add_subdirectory(schemata)
