diff -ruNp analitza-4.13.1-orig/analitzaplot/plotter3d.h analitza-4.13.1/analitzaplot/plotter3d.h
--- analitza-4.13.1-orig/analitzaplot/plotter3d.h	2014-02-15 11:03:35.000000000 -0600
+++ analitza-4.13.1/analitzaplot/plotter3d.h	2014-06-09 15:21:22.000000000 -0500
@@ -34,7 +34,7 @@
 #include <GL/glew.h>
 #endif
 
-#include <GL/glu.h>
+#include <OpenGL/glu.h>
 
 class QAbstractItemModel;
 class QPainter;
