--- solid-5.4.0-orig/src/imports/CMakeLists.txt	2014-11-02 06:26:53.000000000 -0600
+++ solid-5.4.0/src/imports/CMakeLists.txt	2014-11-20 05:32:30.000000000 -0600
@@ -18,7 +18,7 @@
     devices.cpp
     )
 
-add_library(solidextensionplugin SHARED ${solidextensionplugin_SRCS})
+add_library(solidextensionplugin MODULE ${solidextensionplugin_SRCS})
 
 target_link_libraries(
     solidextensionplugin
