diff -ruN khtml-5.29.0-orig/src/html/kopenssl.cpp khtml-5.29.0/src/html/kopenssl.cpp
--- khtml-5.29.0-orig/src/html/kopenssl.cpp	2016-12-03 15:02:35.000000000 -0600
+++ khtml-5.29.0/src/html/kopenssl.cpp	2017-01-20 07:23:28.000000000 -0600
@@ -333,10 +333,9 @@
 #endif
             << "/usr/lib" KDELIBSUFF "/"
             << "/usr/ssl/lib" KDELIBSUFF "/"
-            << "/usr/local/lib" KDELIBSUFF "/"
-            << "/usr/local/openssl/lib" KDELIBSUFF "/"
-            << "/usr/local/ssl/lib" KDELIBSUFF "/"
-            << "/opt/openssl/lib" KDELIBSUFF "/"
+            << "@FINK_PREFIX@/lib" KDELIBSUFF "/"
+            << "@FINK_PREFIX@/openssl/lib" KDELIBSUFF "/"
+            << "@FINK_PREFIX@/ssl/lib" KDELIBSUFF "/"
             << "/lib" KDELIBSUFF "/"
             << "";
 
@@ -347,7 +346,7 @@
 #elif defined(_AIX)
             << "libssl.a(libssl.so.0)"
 #elif defined(__APPLE__)
-            << "libssl.dylib"
+            << "libcrypto.1.1.dylib"
             << "libssl.0.9.dylib"
 #else
 #ifdef SHLIB_VERSION_NUMBER
@@ -364,7 +363,7 @@
 #elif defined(_AIX)
             << "libcrypto.a(libcrypto.so.0)"
 #elif defined(__APPLE__)
-            << "libcrypto.dylib"
+            << "libcrypto.1.1.dylib"
             << "libcrypto.0.9.dylib"
 #else
 #ifdef SHLIB_VERSION_NUMBER
diff -ruN khtml-5.29.0-orig/src/misc/gennames.py khtml-5.29.0/src/misc/gennames.py
--- khtml-5.29.0-orig/src/misc/gennames.py	2016-12-03 15:02:35.000000000 -0600
+++ khtml-5.29.0/src/misc/gennames.py	2017-01-20 07:20:05.000000000 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@FINK_PREFIX@/bin/python3.7
 # -*- coding: utf-8 -*-
 
 # List of namespaces.The 2nd part in the array states whether the attributes
diff -ruN khtml-5.29.0-orig/src/svg/generateSVGNames.py khtml-5.29.0/src/svg/generateSVGNames.py
--- khtml-5.29.0-orig/src/svg/generateSVGNames.py	2016-12-03 15:02:35.000000000 -0600
+++ khtml-5.29.0/src/svg/generateSVGNames.py	2017-01-20 07:19:33.000000000 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@FINK_PREFIX@/bin/python3.7
 # -*- coding: utf-8 -*-
 # misc/gennames.py generates the appropriate ID_ and ATTR_ constants
 # as well as static mappings. This generates WC compatibility stuff ---
