diff -ruN kactivities-5.29.0-orig/src/cli/CMakeLists.txt kactivities-5.29.0/src/cli/CMakeLists.txt
--- kactivities-5.29.0-orig/src/cli/CMakeLists.txt	2016-12-03 04:07:36.000000000 -0600
+++ kactivities-5.29.0/src/cli/CMakeLists.txt	2017-01-14 09:39:05.000000000 -0600
@@ -1,6 +1,8 @@
 # vim:set softtabstop=3 shiftwidth=3 tabstop=3 expandtab:
 project (KActivitiesCLI)
 
+include(ECMMarkNonGuiExecutable)
+
 find_package (Qt5 REQUIRED NO_MODULE COMPONENTS Core Gui Widgets)
 find_package (Qt5 REQUIRED NO_MODULE COMPONENTS Core Gui Widgets)
 find_package (KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS WindowSystem)
@@ -23,6 +25,10 @@
    kactivities-cli
    ${KActivitiesCLI_SRCS}
    )
+if (APPLE)
+   ecm_mark_nongui_executable(kactivities-cli)
+endif (APPLE)
+
 
 target_link_libraries (
    kactivities-cli
diff -ruN kactivities-5.29.0-orig/src/imports/CMakeLists.txt kactivities-5.29.0/src/imports/CMakeLists.txt
--- kactivities-5.29.0-orig/src/imports/CMakeLists.txt	2016-12-03 04:07:36.000000000 -0600
+++ kactivities-5.29.0/src/imports/CMakeLists.txt	2017-01-11 05:38:36.000000000 -0600
@@ -20,7 +20,7 @@
    ${KACTIVITIES_CURRENT_ROOT_SOURCE_DIR}/src/utils/dbusfuture_p.cpp
    )
 
-add_library (kactivitiesextensionplugin SHARED ${kactivities_imports_LIB_SRCS})
+add_library (kactivitiesextensionplugin MODULE ${kactivities_imports_LIB_SRCS})
 
 target_link_libraries (
    kactivitiesextensionplugin
@@ -36,6 +36,10 @@
    KF5::KIOCore
    )
 
+include_directories(
+   ${BOOST_INCLUDEDIR}
+)
+
 ## install
 
 install (TARGETS kactivitiesextensionplugin DESTINATION ${KDE_INSTALL_QMLDIR}/org/kde/activities)
