diff -ruN minuet-18.08.3-orig/src/app/CMakeLists.txt minuet-18.08.3/src/app/CMakeLists.txt
--- minuet-18.08.3-orig/src/app/CMakeLists.txt	2018-11-02 10:10:22.000000000 -0500
+++ minuet-18.08.3/src/app/CMakeLists.txt	2021-10-10 08:49:18.000000000 -0500
@@ -69,5 +69,6 @@
 IF(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Android")
     install(TARGETS minuet ${INSTALL_TARGETS_DEFAULT_ARGS})
     install(PROGRAMS org.kde.minuet.desktop DESTINATION ${XDG_APPS_INSTALL_DIR})
-    install(TARGETS minuet RUNTIME DESTINATION bin)
+    install(TARGETS minuet RUNTIME DESTINATION bin
+                           BUNDLE DESTINATION @FINK_PREFIX@/Applications/KF5)
 ENDIF(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Android")
