diff -ruNp okular-4.12.2-orig/CMakeLists.txt okular-4.12.2/CMakeLists.txt
--- okular-4.12.2-orig/CMakeLists.txt	2014-01-28 17:01:01.000000000 -0500
+++ okular-4.12.2/CMakeLists.txt	2014-02-16 19:49:20.000000000 -0500
@@ -204,7 +204,7 @@ kde4_add_kcfg_files(okularpart_SRCS conf
 
 qt4_add_dbus_interfaces(okularpart_SRCS ${KDE4_DBUS_INTERFACES_DIR}/org.kde.KSpeech.xml)
 
-kde4_add_plugin(okularpart SHARED ${okularpart_SRCS})
+kde4_add_plugin(okularpart MODULE ${okularpart_SRCS})
 
 target_link_libraries(okularpart okularcore ${KDE4_KPARTS_LIBS} ${KDE4_KPRINTUTILS_LIBS} ${MATH_LIB} ${QIMAGEBLITZ_LIBRARIES} ${KDE4_PHONON_LIBRARY} ${KDE4_SOLID_LIBRARY})
 
diff -ruNp okular-4.12.2-orig/active/components/CMakeLists.txt okular-4.12.2/active/components/CMakeLists.txt
--- okular-4.12.2-orig/active/components/CMakeLists.txt	2014-01-28 17:01:01.000000000 -0500
+++ okular-4.12.2/active/components/CMakeLists.txt	2014-02-16 19:49:43.000000000 -0500
@@ -22,7 +22,7 @@ kde4_add_kcfg_files(okular_SRCS ${CMAKE_
 
 qt4_automoc(${okular_SRCS})
 
-kde4_add_library(okularplugin SHARED ${okular_SRCS})
+kde4_add_library(okularplugin MODULE ${okular_SRCS})
 target_link_libraries(okularplugin
         ${QT_QTCORE_LIBRARY}
         ${QT_QTGUI_LIBRARY}
diff -ruNp okular-4.12.2-orig/tests/CMakeLists.txt okular-4.12.2/tests/CMakeLists.txt
--- okular-4.12.2-orig/tests/CMakeLists.txt	2014-01-28 17:01:01.000000000 -0500
+++ okular-4.12.2/tests/CMakeLists.txt	2014-02-16 20:10:25.000000000 -0500
@@ -5,8 +5,8 @@ include_directories(${CMAKE_CURRENT_BINA
 kde4_add_unit_test( shelltest shelltest.cpp ../shell/shellutils.cpp )
 target_link_libraries( shelltest ${KDE4_KDECORE_LIBS} ${QT_QTTEST_LIBRARY} )
 
-kde4_add_unit_test( parttest parttest.cpp )
-target_link_libraries( parttest ${KDE4_KDECORE_LIBS} ${KDE4_KPARTS_LIBS} ${QT_QTGUI_LIBRARY} ${QT_QTTEST_LIBRARY} okularpart okularcore )
+#kde4_add_unit_test( parttest parttest.cpp )
+#target_link_libraries( parttest ${KDE4_KDECORE_LIBS} ${KDE4_KPARTS_LIBS} ${QT_QTGUI_LIBRARY} ${QT_QTTEST_LIBRARY} okularpart okularcore )
 
 kde4_add_unit_test( documenttest documenttest.cpp )
 target_link_libraries( documenttest ${KDE4_KDECORE_LIBS} ${KDE4_THREADWEAVER_LIBRARY} ${QT_QTGUI_LIBRARY} ${QT_QTTEST_LIBRARY} okularcore )
@@ -35,5 +35,5 @@ target_link_libraries( modifyannotationp
 kde4_add_unit_test( editformstest editformstest.cpp )
 target_link_libraries( editformstest ${KDE4_KDECORE_LIBS} ${QT_QTGUI_LIBRARY} ${QT_QTTEST_LIBRARY} ${QT_QTXML_LIBRARY} okularcore )
 
-kde4_add_unit_test( mainshelltest mainshelltest.cpp ../shell/okular_main.cpp ../shell/shellutils.cpp ../shell/shell.cpp )
-target_link_libraries( mainshelltest ${KDE4_KPARTS_LIBS} ${QT_QTTEST_LIBRARY} okularpart okularcore )
+#kde4_add_unit_test( mainshelltest mainshelltest.cpp ../shell/okular_main.cpp ../shell/shellutils.cpp ../shell/shell.cpp )
+#target_link_libraries( mainshelltest ${KDE4_KPARTS_LIBS} ${QT_QTTEST_LIBRARY} okularpart okularcore )
