diff -ruNp pairs-4.13.1-orig/CMakeLists.txt pairs-4.13.1/CMakeLists.txt
--- pairs-4.13.1-orig/CMakeLists.txt	2014-03-22 23:43:37.000000000 -0500
+++ pairs-4.13.1/CMakeLists.txt	2014-06-05 09:34:41.000000000 -0500
@@ -32,6 +32,7 @@ add_definitions( ${KDE4_DEFINITIONS} )
 add_subdirectory( game )
 add_subdirectory( pairseditor )
 add_subdirectory( icons )
+add_subdirectory( iconseditor )
 add_subdirectory( players )
 add_subdirectory( gameicons )
 add_subdirectory( themes )
diff -ruNp pairs-4.13.1-orig/pairseditor/CMakeLists.txt pairs-4.13.1/pairseditor/CMakeLists.txt
--- pairs-4.13.1-orig/pairseditor/CMakeLists.txt	2014-03-22 23:43:37.000000000 -0500
+++ pairs-4.13.1/pairseditor/CMakeLists.txt	2014-06-05 09:35:27.000000000 -0500
@@ -5,7 +5,7 @@ set(pairseditor_SOURCES  mainwindow.cpp 
 
 kde4_add_ui_files( pairseditor_SOURCES mainwindowview.ui )
 
-kde4_add_app_icon(pairs_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../icons/hi*-apps-pairseditor*")
+kde4_add_app_icon(pairs_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/../iconseditor/hi*-apps-pairseditor*")
 
 kde4_add_executable( pairseditor ${pairseditor_SOURCES} )
 
