diff -ur cmake-2.0.2/Modules/FindOpenGL.cmake cmake-2.0.2.corr/Modules/FindOpenGL.cmake
--- cmake-2.0.2/Modules/FindOpenGL.cmake	Sat May  1 16:07:41 2004
+++ cmake-2.0.2.corr/Modules/FindOpenGL.cmake	Mon Jun 28 13:39:54 2004
@@ -97,9 +97,9 @@
   IF (OPENGL_gl_LIBRARY)
     INCLUDE( ${CMAKE_ROOT}/Modules/FindX11.cmake )
     IF (X11_FOUND)
-      IF (NOT APPLE)
+#      IF (NOT APPLE)
         SET (OPENGL_LIBRARIES ${X11_LIBRARIES})
-      ENDIF (NOT APPLE)
+#      ENDIF (NOT APPLE)
     ENDIF (X11_FOUND)
   ENDIF (OPENGL_gl_LIBRARY)
 
diff -ur cmake-2.0.2/Modules/Platform/Darwin.cmake cmake-2.0.2.corr/Modules/Platform/Darwin.cmake
--- cmake-2.0.2/Modules/Platform/Darwin.cmake	Wed Apr 28 19:21:56 2004
+++ cmake-2.0.2.corr/Modules/Platform/Darwin.cmake	Mon Jun 28 13:38:54 2004
@@ -10,6 +10,9 @@
 IF("${CMAKE_BACKWARDS_COMPATIBILITY}" MATCHES "^1\\.[0-6]$")
   SET(CMAKE_SHARED_MODULE_CREATE_C_FLAGS
     "${CMAKE_SHARED_MODULE_CREATE_C_FLAGS} -flat_namespace -undefined suppress")
+ELSE("${CMAKE_BACKWARDS_COMPATIBILITY}" MATCHES "^1\\.[0-6]$")
+  SET(CMAKE_SHARED_MODULE_CREATE_C_FLAGS
+    "${CMAKE_SHARED_MODULE_CREATE_C_FLAGS} -undefined dynamic_lookup -bind_at_load")
 ENDIF("${CMAKE_BACKWARDS_COMPATIBILITY}" MATCHES "^1\\.[0-6]$")
 
 SET(CMAKE_CXX_CREATE_SHARED_LIBRARY
--- cmake-2.0.5/Source/cmDynamicLoader.h	2002-10-24 00:03:26.000000000 +0200
+++ cmake-2.0.5_corr/Source/cmDynamicLoader.h	2005-03-18 12:11:53.000000000 +0100
@@ -33,6 +33,9 @@
 #elif defined(_WIN32)
   #include <windows.h>
   typedef HMODULE cmLibHandle;
+#elif defined(__APPLE__)
+  #include <mach-o/dyld.h>
+  typedef NSModule cmLibHandle;
 #else
   typedef void* cmLibHandle;
 #endif
