diff -Nurd gdal-1.11.5.orig/port/cpl_xml_validate.cpp gdal-1.11.5/port/cpl_xml_validate.cpp
--- gdal-1.11.5.orig/port/cpl_xml_validate.cpp	2016-07-01 04:55:03.000000000 -0400
+++ gdal-1.11.5/port/cpl_xml_validate.cpp	2025-08-24 15:04:56.000000000 -0400
@@ -847,7 +847,11 @@
 
     if (strstr(pszStr, "since this namespace was already imported") == NULL)
     {
+#if LIBXML_VERSION >= 21200
+        const xmlError *pErrorPtr = xmlGetLastError();
+#else
         xmlErrorPtr pErrorPtr = xmlGetLastError();
+#endif
         const char* pszFilename = (const char*)ctx;
         char* pszStrDup = CPLStrdup(pszStr);
         int nLen = (int)strlen(pszStrDup);
