diff -ruN gdbm-1.13-orig/compat/Makefile.in gdbm-1.13/compat/Makefile.in
--- gdbm-1.13-orig/compat/Makefile.in	2017-03-11 10:13:28.000000000 -0600
+++ gdbm-1.13/compat/Makefile.in	2017-03-11 16:06:55.000000000 -0600
@@ -404,7 +404,7 @@
  dbmrdonly.c
 
 libgdbm_compat_la_SOURCES = $(DBM_CF) $(NDBM_CF)
-libgdbm_compat_la_LDFLAGS = -version-info $(VI_CURRENT):$(VI_REVISION):$(VI_AGE)
+libgdbm_compat_la_LDFLAGS = -version-info $(VI_CURRENT):$(VI_REVISION):$(VI_AGE) -no-undefined
 all: all-am
 
 .SUFFIXES:
diff -ruN gdbm-1.13-orig/src/Makefile.in gdbm-1.13/src/Makefile.in
--- gdbm-1.13-orig/src/Makefile.in	2017-03-11 10:13:28.000000000 -0600
+++ gdbm-1.13/src/Makefile.in	2017-03-11 16:07:08.000000000 -0600
@@ -459,7 +459,7 @@
 	gdbmsetopt.c gdbmstore.c gdbmsync.c base64.c bucket.c falloc.c \
 	findkey.c fullio.c hash.c lock.c mmap.c recover.c update.c \
 	version.c $(am__append_1)
-libgdbm_la_LDFLAGS = -version-info $(VI_CURRENT):$(VI_REVISION):$(VI_AGE)
+libgdbm_la_LDFLAGS = -version-info $(VI_CURRENT):$(VI_REVISION):$(VI_AGE) -no-undefined
 noinst_LIBRARIES = libgdbmapp.a
 libgdbmapp_a_SOURCES = \
  err.c\
