diff -Nurd gdbm-1.21.orig/src/gdbmshell.c gdbm-1.21/src/gdbmshell.c
--- gdbm-1.21.orig/src/gdbmshell.c	2021-09-02 08:39:55.000000000 -0400
+++ gdbm-1.21/src/gdbmshell.c	2021-09-06 11:32:46.000000000 -0400
@@ -1010,7 +1010,7 @@
       fprintf (fp, "%s: ", snapname);
       fprintf (fp, "%03o %s ", st.st_mode & 0777,
 	       decode_mode (st.st_mode, buf));
-      fprintf (fp, "%ld.%09ld", st.st_mtim.tv_sec, st.st_mtim.tv_nsec);
+      fprintf (fp, "%ld.%09ld", st.st_mtimespec.tv_sec, st.st_mtimespec.tv_nsec);
       if (S_ISREG (st.st_mode))
 	{
 	  GDBM_FILE dbf;
