--- apr-1.7.0/build/apr_network.m4	2020/10/29 20:16:55	1882979
+++ apr-1.7.0/build/apr_network.m4	2020/10/29 20:23:11	1882980
@@ -63,6 +63,9 @@
 #ifdef HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
 #endif
+#ifdef HAVE_STDLIB_H
+#include <stdlib.h>
+#endif
 
 int main(void) {
     struct addrinfo hints, *ai;
@@ -151,6 +154,9 @@
 #ifdef HAVE_NETINET_IN_H
 #include <netinet/in.h>
 #endif
+#ifdef HAVE_STDLIB_H
+#include <stdlib.h>
+#endif
 
 int main(void) {
     struct sockaddr_in sa;
@@ -194,6 +200,9 @@
 #ifdef HAVE_NETDB_H
 #include <netdb.h>
 #endif
+#ifdef HAVE_STDLIB_H
+#include <stdlib.h>
+#endif
 
 int main(void) {
     if (EAI_ADDRFAMILY < 0) {
@@ -403,6 +412,9 @@
 #ifndef HAVE_SOCKLEN_T
 typedef int socklen_t;
 #endif
+#ifdef HAVE_STDLIB_H
+#include <stdlib.h>
+#endif
 int main(void) {
     int listen_s, connected_s, client_s;
     int listen_port, rc;
@@ -588,6 +600,9 @@
 #ifdef HAVE_FCNTL_H
 #include <fcntl.h>
 #endif
+#ifdef HAVE_STDLIB_H
+#include <stdlib.h>
+#endif
 int main(void) {
     int listen_s, connected_s, client_s;
     int listen_port, rc;
