diff -ruN c-blosc-1.14.4-orig/internal-complibs/zlib-1.2.8/gzlib.c c-blosc-1.14.4/internal-complibs/zlib-1.2.8/gzlib.c
--- c-blosc-1.14.4-orig/internal-complibs/zlib-1.2.8/gzlib.c	2018-07-30 05:56:38.000000000 -0500
+++ c-blosc-1.14.4/internal-complibs/zlib-1.2.8/gzlib.c	2022-02-15 21:57:15.000000000 -0600
@@ -3,6 +3,7 @@
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
 
+#include <unistd.h>
 #include "gzguts.h"
 
 #if defined(_WIN32) && !defined(__BORLANDC__)
diff -ruN c-blosc-1.14.4-orig/internal-complibs/zlib-1.2.8/gzread.c c-blosc-1.14.4/internal-complibs/zlib-1.2.8/gzread.c
--- c-blosc-1.14.4-orig/internal-complibs/zlib-1.2.8/gzread.c	2018-07-30 05:56:38.000000000 -0500
+++ c-blosc-1.14.4/internal-complibs/zlib-1.2.8/gzread.c	2022-02-15 21:59:06.000000000 -0600
@@ -3,6 +3,7 @@
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
 
+#include <unistd.h>
 #include "gzguts.h"
 
 /* Local functions */
diff -ruN c-blosc-1.14.4-orig/internal-complibs/zlib-1.2.8/gzwrite.c c-blosc-1.14.4/internal-complibs/zlib-1.2.8/gzwrite.c
--- c-blosc-1.14.4-orig/internal-complibs/zlib-1.2.8/gzwrite.c	2018-07-30 05:56:38.000000000 -0500
+++ c-blosc-1.14.4/internal-complibs/zlib-1.2.8/gzwrite.c	2022-02-15 22:00:06.000000000 -0600
@@ -3,6 +3,7 @@
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
 
+#include <unistd.h>
 #include "gzguts.h"
 
 /* Local functions */
