diff -Nurd -x'*~' libdatrie-0.2.9.orig/tools/trietool.c libdatrie-0.2.9/tools/trietool.c
--- libdatrie-0.2.9.orig/tools/trietool.c	2015-02-01 09:34:01.000000000 -0500
+++ libdatrie-0.2.9/tools/trietool.c	2016-09-26 00:14:43.000000000 -0400
@@ -125,7 +125,7 @@
     res = iconv (env->to_alpha_conv, (char **) &in_p, &in_left,
                  &out_p, &out_left);
 
-    if (res < 0)
+    if (res == (size_t) -1)
         return res;
 
     /* convert UCS-4LE to AlphaChar string */
