diff -Nurd libevent-2.1.12-stable.orig/Makefile.am libevent-2.1.12-stable/Makefile.am
--- libevent-2.1.12-stable.orig/Makefile.am	2020-07-05 08:01:34
+++ libevent-2.1.12-stable/Makefile.am	2025-12-02 07:06:16
@@ -251,13 +251,8 @@
 	evrpc.c					\
 	http.c
 
-if BUILD_WITH_NO_UNDEFINED
 NO_UNDEFINED = -no-undefined
 MAYBE_CORE = libevent_core.la
-else
-NO_UNDEFINED =
-MAYBE_CORE =
-endif
 
 AM_CFLAGS = $(LIBEVENT_CFLAGS)
 AM_CPPFLAGS = -I$(srcdir)/compat -I./include -I$(srcdir)/include $(SYS_INCLUDES) $(LIBEVENT_CPPFLAGS)
