diff -Nurd exempi-2.6.6.orig/configure exempi-2.6.6/configure
--- exempi-2.6.6.orig/configure	2025-04-01 20:42:02
+++ exempi-2.6.6/configure	2026-07-29 00:02:52
@@ -19134,7 +19134,7 @@
 case $build_vendor in
 	apple)
 		EXEMPI_PLATFORM_DEF=MAC_ENV
-		LDFLAGS="$LDFLAGS -framework CoreServices -stdlib=libc++"
+		LDFLAGS="$LDFLAGS -Wl,-framework,CoreServices -stdlib=libc++"
 		CPPFLAGS="$CPPFLAGS -std=c++11 -stdlib=libc++"
 		;;
 	*)
diff -Nurd exempi-2.6.6.orig/exempi/Makefile.in exempi-2.6.6/exempi/Makefile.in
--- exempi-2.6.6.orig/exempi/Makefile.in	2025-04-01 20:42:02
+++ exempi-2.6.6/exempi/Makefile.in	2026-07-29 00:03:54
@@ -131,8 +131,6 @@
 @WITH_UNIT_TEST_TRUE@check_PROGRAMS = testexempicore$(EXEEXT) \
 @WITH_UNIT_TEST_TRUE@	testserialise$(EXEEXT) \
 @WITH_UNIT_TEST_TRUE@	testwritenewprop$(EXEEXT) \
-@WITH_UNIT_TEST_TRUE@	testtiffleak$(EXEEXT) \
-@WITH_UNIT_TEST_TRUE@	testxmpfiles$(EXEEXT) \
 @WITH_UNIT_TEST_TRUE@	testxmpfileswrite$(EXEEXT) \
 @WITH_UNIT_TEST_TRUE@	testparse$(EXEEXT) testiterator$(EXEEXT) \
 @WITH_UNIT_TEST_TRUE@	testinit$(EXEEXT) testfdo18635$(EXEEXT) \
