diff -Nurd libevent-2.1.8-stable.orig/Makefile.am libevent-2.1.8-stable/Makefile.am
--- libevent-2.1.8-stable.orig/Makefile.am	2017-01-29 13:41:27
+++ libevent-2.1.8-stable/Makefile.am	2025-12-02 07:11:17
@@ -220,13 +220,8 @@
 	evrpc.c					\
 	http.c
 
-if BUILD_WITH_NO_UNDEFINED
 NO_UNDEFINED = -no-undefined
 MAYBE_CORE = libevent_core.la
-else
-NO_UNDEFINED =
-MAYBE_CORE =
-endif
 
 GENERIC_LDFLAGS = -version-info $(VERSION_INFO) $(RELEASE) $(NO_UNDEFINED)
 
diff -Nurd libevent-2.1.8-stable.orig/configure.ac libevent-2.1.8-stable/configure.ac
--- libevent-2.1.8-stable.orig/configure.ac	2017-01-29 12:51:00
+++ libevent-2.1.8-stable/configure.ac	2025-12-02 07:11:00
@@ -508,6 +508,8 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <stdlib.h>
+#include <string.h>
 
 int
 main(int argc, char **argv)
