diff -ruN iODBC-3.52.13-orig/include/iodbc.h iODBC-3.52.13/include/iodbc.h
--- iODBC-3.52.13-orig/include/iodbc.h	2019-07-23 06:37:26.000000000 -0500
+++ iODBC-3.52.13/include/iodbc.h	2020-12-12 03:23:37.000000000 -0600
@@ -148,7 +148,6 @@
 # 		define SYS_ODBC_INI "Boot:System Folder:Preferences:ODBC Preferences"
 # 	endif
 # elif defined(__APPLE__)
-# 	define SYS_ODBC_INI "/etc/odbc.ini"
 # 	define ODBC_INI_APP "/Library/ODBC/odbc.ini"
 # else
 # 	define SYS_ODBC_INI "/etc/odbc.ini"
@@ -160,7 +159,6 @@
 #    define SYS_ODBCINST_INI	"/boot/beos/etc/odbcinst.ini"
 #  elif defined(macintosh)
 #  elif defined(__APPLE__)
-#    define SYS_ODBCINST_INI	"/etc/odbcinst.ini"
 #    define ODBCINST_INI_APP	"/Library/ODBC/odbcinst.ini"
 #  else
 #    define SYS_ODBCINST_INI	"/etc/odbcinst.ini"
diff -ruN iODBC-3.52.13-orig/include/odbcinst.h iODBC-3.52.13/include/odbcinst.h
--- iODBC-3.52.13-orig/include/odbcinst.h	2019-07-23 06:37:26.000000000 -0500
+++ iODBC-3.52.13/include/odbcinst.h	2020-12-12 03:23:37.000000000 -0600
@@ -111,7 +111,6 @@
 #      define SYS_ODBCINST_INI "Boot:System Folder:Preferences:ODBC Installer Preferences"
 #    endif
 #  else
-#    define SYS_ODBCINST_INI	"/etc/odbcinst.ini"
 #  endif
 #endif
 
@@ -126,7 +125,6 @@
 #      define SYS_ODBC_INI "Boot:System Folder:Preferences:ODBC Preferences"
 #    endif
 #  else
-#    define SYS_ODBC_INI	"/etc/odbc.ini"
 #  endif
 #endif
 
@@ -140,9 +138,7 @@
 #      define DEFAULT_FILEDSNPATH "Boot:System Folder:Preferences:ODBC Preferences:ODBCDataSources"
 #    endif
 #  elif defined (__APPLE__)
-#    define DEFAULT_FILEDSNPATH	"/Library/ODBC/ODBCDataSources"
 #  else
-#    define DEFAULT_FILEDSNPATH	"/etc/ODBCDataSources"
 #  endif
 #endif
 
diff -ruN iODBC-3.52.13-orig/include/sqltypes.h iODBC-3.52.13/include/sqltypes.h
--- iODBC-3.52.13-orig/include/sqltypes.h	2019-07-23 06:37:26.000000000 -0500
+++ iODBC-3.52.13/include/sqltypes.h	2020-12-12 03:23:37.000000000 -0600
@@ -96,6 +96,9 @@
 extern "C" {
 #endif
 
+#ifndef _IODBCUNIX_H
+#include <iodbcunix.h>
+#endif
 
 /*
  *  Environment specific definitions
diff -ruN iODBC-3.52.13-orig/iodbcadm/drvconn.c iODBC-3.52.13/iodbcadm/drvconn.c
--- iODBC-3.52.13-orig/iodbcadm/drvconn.c	2019-07-23 06:37:26.000000000 -0500
+++ iODBC-3.52.13/iodbcadm/drvconn.c	2020-12-12 10:19:15.000000000 -0600
@@ -89,7 +89,7 @@
 typedef SQLRETURN SQL_API (*pDriverConnWFunc) (HWND hwnd, LPWSTR szInOutConnStr,
     DWORD cbInOutConnStr, int FAR * sqlStat, SQLUSMALLINT fDriverCompletion, UWORD *config);
 
-#if defined (__APPLE__)
+#if !defined (__APPLE__)
 
 #define CALL_DRVCONN_DIALBOXW(path, a) \
     if (path) \
diff -ruN iODBC-3.52.13-orig/iodbcinst/unicode.h iODBC-3.52.13/iodbcinst/unicode.h
--- iODBC-3.52.13-orig/iodbcinst/unicode.h	2019-07-23 06:37:26.000000000 -0500
+++ iODBC-3.52.13/iodbcinst/unicode.h	2020-12-12 03:23:37.000000000 -0600
@@ -76,6 +76,7 @@
 
 #ifndef _UNICODE_H
 #define _UNICODE_H
+#include <iodbc.h>
 
 
 #if defined (__APPLE__) && !defined (MACOSX102) && !defined (HAVE_CONFIG_H)
