diff -Nurd noise/lib/Makefile noise-patched/lib/Makefile
--- noise/lib/Makefile	2004-10-24 21:21:12.000000000 +0200
+++ noise-patched/lib/Makefile	2012-06-28 22:14:42.000000000 +0200
@@ -1,7 +1,7 @@
 VPATH=../src/
 
 .PHONY: all clean
-all: libnoise.a libnoise.la libnoise.so.0.3
+all: libnoise.a libnoise.la libnoise.0.3.dylib
 	-cp $? .
 
 clean:
diff -Nurd noise/src/Makefile noise-patched/src/Makefile
--- noise/src/Makefile	2004-10-24 21:21:12.000000000 +0200
+++ noise-patched/src/Makefile	2012-06-28 22:47:45.000000000 +0200
@@ -1,4 +1,4 @@
-LIBTOOL=libtool
+LIBTOOL=glibtool
 
 # defines source files and vpaths
 include Sources
@@ -11,15 +11,16 @@
 .PHONY: all clean cleandeps cleanobjs cleanlib libnoise libnoise.so libnoise.so.0
 
 # hooks for future makefiles being able to make multiple SOs, or older SOs
-libnoise: libnoise.so libnoise.a libnoise.la
-libnoise.so: libnoise.so.0
-libnoise.so.0: libnoise.so.0.3
+libnoise: libnoise.dylib libnoise.a libnoise.la
+libnoise.dylib: libnoise.0.dylib
+libnoise.0.dylib: libnoise.0.3.dylib
 
 # Real build targets
-libnoise.so.0.3: $(OBJECTS)
-	$(LIBTOOL) --mode=link $(CXX) $(LDFLAGS) -shared -Wl,-soname=libnoise.so.0 -o $@ $(OBJECTS:.o=.lo)
+libnoise.0.3.dylib: $(OBJECTS)
+	$(LIBTOOL) --mode=link $(CXX) $(LDFLAGS) -shared -dynamiclib -o $@ $(OBJECTS:.o=.lo)
 
 libnoise.a: $(OBJECTS)
+	touch libnoise.a
 	$(LIBTOOL) --mode=link $(CXX) $(LDFLAGS) -o $@ $(OBJECTS)
 libnoise.la: $(OBJECTS)
 	$(LIBTOOL) --mode=link $(CXX) $(LDFLAGS) -o $@ $(OBJECTS:.o=.lo)
@@ -32,7 +33,7 @@
 	-rm $(OBJECTS:.o=.lo) #clean up after libtool
 	-rm -rf .libs model/.libs module/.libs
 cleanlib:
-	-rm libnoise.so.0.3
+	-rm libnoise.0.3.dylib
 	-rm libnoise.a
 	-rm libnoise.la
 
