diff -Naur libotf-0.9.13.orig/configure libotf-0.9.13/configure
--- libotf-0.9.13.orig/configure	2012-09-07 09:54:17.000000000 -0400
+++ libotf-0.9.13/configure	2013-06-10 14:56:06.000000000 -0400
@@ -12147,6 +12147,8 @@
 done
 
 
+save_cppflags=$CPPFLAGS
+CPPFLAGS="$CPPFLAGS $X_CFLAGS"
 for ac_header in X11/Xaw/Command.h
 do :
   ac_fn_c_check_header_mongrel "$LINENO" "X11/Xaw/Command.h" "ac_cv_header_X11_Xaw_Command_h" "$ac_includes_default"
@@ -12157,6 +12159,7 @@
 
 else
   no_x=yes
+  CPPFLAGS=$save_cppflags
 fi
 
 done
diff -Naur libotf-0.9.13.orig/example/Makefile.in libotf-0.9.13/example/Makefile.in
--- libotf-0.9.13.orig/example/Makefile.in	2012-09-07 09:54:16.000000000 -0400
+++ libotf-0.9.13/example/Makefile.in	2013-06-10 14:42:14.000000000 -0400
@@ -85,8 +85,8 @@
 otfview_OBJECTS = otfview.$(OBJEXT)
 otfview_DEPENDENCIES = $(CommonLDADD)
 otfview_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
-	--mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(otfview_LDFLAGS) \
-	$(LDFLAGS) -o $@
+	--mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(LDFLAGS) \
+	$(otfview_LDFLAGS) -o $@
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/src
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
diff -Naur libotf-0.9.13.orig/libotf.pc.in libotf-0.9.13/libotf.pc.in
--- libotf-0.9.13.orig/libotf.pc.in	2006-07-19 01:17:27.000000000 -0400
+++ libotf-0.9.13/libotf.pc.in	2013-06-10 14:42:34.000000000 -0400
@@ -7,5 +7,5 @@
 Description: OpenType font library
 Version: @PACKAGE_VERSION@
 Requires: freetype2
-Libs: @FREETYPE_LD_FLAGS@ -L${libdir} -lotf
-Cflags: @FREETYPE_INC@ -I${includedir}
+Libs: -L${libdir} -lotf
+Cflags: -I${includedir}
