diff -ru serf-1.3.8.orig/SConstruct serf-1.3.8/SConstruct
--- serf-1.3.8.orig/SConstruct	2014-02-04 15:11:10.000000000 -0500
+++ serf-1.3.8/SConstruct	2015-12-14 18:56:44.000000000 -0500
@@ -210,9 +210,9 @@
 # Unfortunately we can't set the .dylib compatibility_version option separately
 # from current_version, so don't use the PATCH level to avoid that build and
 # runtime patch levels have to be identical.
-env['SHLIBVERSION'] = '%d.%d.%d' % (MAJOR, MINOR, 0)
+env['SHLIBVERSION'] = '%d.%d.%d' % (MAJOR+3, MINOR, 0)
 
-LIBNAME = 'libserf-%d' % (MAJOR,)
+LIBNAME = 'libserf-%d.0' % (MAJOR,)
 if sys.platform != 'win32':
   LIBNAMESTATIC = LIBNAME
 else:
@@ -223,7 +223,7 @@
 
 if sys.platform == 'darwin':
 #  linkflags.append('-Wl,-install_name,@executable_path/%s.dylib' % (LIBNAME,))
-  env.Append(LINKFLAGS='-Wl,-install_name,%s/%s.dylib' % (thisdir, LIBNAME,))
+  env.Append(LINKFLAGS='-Wl,-install_name,%s/%s.dylib,-compatibility_version,%s,-current_version,%s' % (thisdir, LIBNAME, env['SHLIBVERSION'], env['SHLIBVERSION'],))
 
 if sys.platform != 'win32':
   ### gcc only. figure out appropriate test / better way to check these
