diff -ruNd tinyxml.orig/CMakeLists.txt tinyxml/CMakeLists.txt
--- tinyxml.orig/CMakeLists.txt	1970-01-01 09:00:00.000000000 +0900
+++ tinyxml/CMakeLists.txt	2015-01-15 14:58:19.000000000 +0900
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.1)
+project(TinyXML CXX)
+add_library(tinyxml SHARED tinystr.cpp tinyxml.cpp tinyxmlerror.cpp tinyxmlparser.cpp)
+set_target_properties(tinyxml PROPERTIES VERSION @@VERSION@@ INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
+add_executable(xmltest xmltest.cpp tinystr.cpp tinyxml.cpp tinyxmlerror.cpp tinyxmlparser.cpp)
+install(TARGETS tinyxml LIBRARY DESTINATION lib)
+install(FILES tinystr.h tinyxml.h DESTINATION include)
+install(FILES tinyxml.pc DESTINATION lib/pkgconfig)
diff -ruNd tinyxml.orig/tinyxml.pc tinyxml/tinyxml.pc
--- tinyxml.orig/tinyxml.pc	1970-01-01 09:00:00.000000000 +0900
+++ tinyxml/tinyxml.pc	2015-01-10 22:27:51.000000000 +0900
@@ -0,0 +1,10 @@
+prefix=@@PREFIX@@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: TinyXml
+Description: simple, small, C++ XML parser
+Version: @@VERSION@@
+Libs: -L${libdir} -ltinyxml
+Cflags: -I${includedir}
