diff -Nurd userspace-rcu-0.14.0.orig/configure userspace-rcu-0.14.0/configure
--- userspace-rcu-0.14.0.orig/configure	2023-02-14 10:42:11.000000000 -0500
+++ userspace-rcu-0.14.0/configure	2023-10-22 21:44:39.000000000 -0400
@@ -3391,13 +3391,7 @@
 ## OS and Arch specific defaults ##
 ##                               ##
 
-case $host in #(
-  *-cygwin* | *-msys*) :
     LT_NO_UNDEFINED="-no-undefined"
- ;; #(
-  *) :
-     ;;
-esac
 
 
 ##                   ##
diff -Nurd userspace-rcu-0.14.0.orig/doc/examples/Makefile.in userspace-rcu-0.14.0/doc/examples/Makefile.in
--- userspace-rcu-0.14.0.orig/doc/examples/Makefile.in	2023-02-14 10:42:11.000000000 -0500
+++ userspace-rcu-0.14.0/doc/examples/Makefile.in	2023-10-22 21:53:20.000000000 -0400
@@ -878,7 +878,7 @@
 @ENABLE_EXAMPLES_TRUE@	else \
 @ENABLE_EXAMPLES_TRUE@		rel_build_subdir="../"; \
 @ENABLE_EXAMPLES_TRUE@	fi; \
-@ENABLE_EXAMPLES_TRUE@	$(MAKE) -f dist-files/Makefile CC="$(CC)" CPPFLAGS="$(CPPFLAGS)" AM_CPPFLAGS="$(AM_CPPFLAGS) -I"$${rel_src_subdir}/$(top_srcdir)/include/" -I"$${rel_src_subdir}/$(top_srcdir)/src/" -I"$${rel_build_subdir}$(top_builddir)/include/" -I"$${rel_build_subdir}$(top_builddir)/include/src/"" CFLAGS="$(CFLAGS)" AM_CFLAGS="$(AM_CFLAGS)" LDFLAGS="$(LDFLAGS)" AM_LDFLAGS="$(AM_LDFLAGS) -L../../../src/.libs/ -Wl,-rpath "$(PWD)/../../src/.libs/"" AM_V_P="$(AM_V_P)" AM_V_at="$(AM_V_at)" $(AM_MAKEFLAGS) all;
+@ENABLE_EXAMPLES_TRUE@	$(MAKE) -f dist-files/Makefile CC="$(CC)" CPPFLAGS="$(CPPFLAGS)" AM_CPPFLAGS="$(AM_CPPFLAGS) -I"$${rel_src_subdir}/$(top_srcdir)/include/" -I"$${rel_src_subdir}/$(top_srcdir)/src/" -I"$${rel_build_subdir}$(top_builddir)/include/" -I"$${rel_build_subdir}$(top_builddir)/include/src/"" CFLAGS="$(CFLAGS)" AM_CFLAGS="$(AM_CFLAGS)" LDFLAGS="$(LDFLAGS)" AM_LDFLAGS="$(AM_LDFLAGS)" AM_V_P="$(AM_V_P)" AM_V_at="$(AM_V_at)" $(AM_MAKEFLAGS) all;
 
 @ENABLE_EXAMPLES_TRUE@clean-local:
 @ENABLE_EXAMPLES_TRUE@	$(AM_V_at)$(MAKE) -f dist-files/Makefile $(AM_MAKEFLAGS) clean; \
