diff -ruN xine-lib-1.2.6-orig/src/xine-engine/Makefile.in xine-lib-1.2.6/src/xine-engine/Makefile.in
--- xine-lib-1.2.6-orig/src/xine-engine/Makefile.in	2014-07-06 07:55:30.000000000 -0500
+++ xine-lib-1.2.6/src/xine-engine/Makefile.in	2015-05-10 14:41:59.000000000 -0500
@@ -697,7 +697,11 @@
 	}
 
 libxine-interface.la: $(libxine_interface_la_OBJECTS) $(libxine_interface_la_DEPENDENCIES) $(EXTRA_libxine_interface_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(libxine_interface_la_LINK)  $(libxine_interface_la_OBJECTS) $(libxine_interface_la_LIBADD) $(LIBS)
+	if test -n "$(libxine_interface_la_OBJECTS)" ; then \
+	    $(AM_V_CCLD)$(libxine_interface_la_LINK)  $(libxine_interface_la_OBJECTS) $(libxine_interface_la_LIBADD) $(LIBS); \
+	else \
+	    touch $@; \
+	fi
 
 libxine.la: $(libxine_la_OBJECTS) $(libxine_la_DEPENDENCIES) $(EXTRA_libxine_la_DEPENDENCIES) 
 	$(AM_V_CCLD)$(libxine_la_LINK) -rpath $(libdir) $(libxine_la_OBJECTS) $(libxine_la_LIBADD) $(LIBS)
diff -ruN xine-lib-1.2.8-orig/src/xine-engine/builtins.h xine-lib-1.2.8/src/xine-engine/builtins.h
--- xine-lib-1.2.8-orig/src/xine-engine/builtins.h	2017-02-21 05:13:13.000000000 -0600
+++ xine-lib-1.2.8/src/xine-engine/builtins.h	2017-03-04 15:17:43.000000000 -0600
@@ -31,7 +31,7 @@
 
 #include <xine/xine_internal.h>
 
-const plugin_info_t xine_builtin_plugin_info[7];
+extern const plugin_info_t xine_builtin_plugin_info[7];
 
 #endif
 
