diff -Nurd libxml2-2.9.14.orig/python/Makefile.in libxml2-2.9.14/python/Makefile.in
--- libxml2-2.9.14.orig/python/Makefile.in	2022-05-02 07:42:31.000000000 -0400
+++ libxml2-2.9.14/python/Makefile.in	2022-05-24 04:03:40.000000000 -0400
@@ -161,7 +161,7 @@
 am__v_at_ = $(am__v_at_@AM_DEFAULT_V@)
 am__v_at_0 = @
 am__v_at_1 = 
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
+DEFAULT_INCLUDES = -I.@am__isrc@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/libxml.Plo \
@@ -508,14 +508,12 @@
 	libxml2class.txt
 
 @WITH_PYTHON_TRUE@AM_CPPFLAGS = \
-@WITH_PYTHON_TRUE@	-I$(top_builddir)/include \
-@WITH_PYTHON_TRUE@	-I$(top_srcdir)/include	\
 @WITH_PYTHON_TRUE@	-I$(PYTHON_INCLUDES)
 
 @WITH_PYTHON_TRUE@python_LTLIBRARIES = libxml2mod.la
 @WITH_PYTHON_TRUE@libxml2mod_la_SOURCES = libxml.c libxml_wrap.h libxml2-py.h libxml2-py.c types.c
 @WITH_PYTHON_TRUE@libxml2mod_la_LDFLAGS = $(CYGWIN_EXTRA_LDFLAGS) $(WIN32_EXTRA_LDFLAGS) -module -avoid-version \
-@WITH_PYTHON_TRUE@        $(top_builddir)/libxml2.la
+@WITH_PYTHON_TRUE@        -lxml2
 
 @WITH_PYTHON_TRUE@libxml2mod_la_LIBADD = $(CYGWIN_EXTRA_PYTHON_LIBADD) $(WIN32_EXTRA_PYTHON_LIBADD) $(PYTHON_LIBS)
 @WITH_PYTHON_TRUE@BUILT_SOURCES = libxml2-export.c libxml2-py.h libxml2-py.c
diff -Nurd libxml2-2.9.14.orig/python/setup.py libxml2-2.9.14/python/setup.py
--- libxml2-2.9.14.orig/python/setup.py	2022-05-02 07:42:37.000000000 -0400
+++ libxml2-2.9.14/python/setup.py	2022-05-24 04:01:24.000000000 -0400
@@ -259,7 +259,7 @@
        author = "Daniel Veillard",
        author_email = "veillard@redhat.com",
        url = "https://gitlab.gnome.org/GNOME/libxml2",
-       licence="MIT Licence",
+       license="MIT Licence",
        py_modules=modules,
        ext_modules=extens,
        ext_package=ext_package,
diff -Nurd libxml2-2.9.14.orig/python/tests/Makefile.in libxml2-2.9.14/python/tests/Makefile.in
--- libxml2-2.9.14.orig/python/tests/Makefile.in	2022-05-02 07:42:31.000000000 -0400
+++ libxml2-2.9.14/python/tests/Makefile.in	2022-05-24 04:00:24.000000000 -0400
@@ -669,17 +669,19 @@
 @WITH_PYTHON_TRUE@tests: $(PYTESTS)
 @WITH_PYTHON_TRUE@	@for f in $(XMLS) ; do test -f $$f || $(LN_S) $(srcdir)/$$f . ; done
 @WITH_PYTHON_TRUE@	@echo "## running Python regression tests"
+@WITH_PYTHON_TRUE@	-@rm -f test.out
 @WITH_PYTHON_TRUE@	@(export PYTHONPATH="..:../.libs:$(srcdir)/..:$$PYTHONPATH" ; \
-@WITH_PYTHON_TRUE@	  export LD_LIBRARY_PATH="$(top_builddir)/.libs:$$LD_LIBRARY_PATH" ; \
 @WITH_PYTHON_TRUE@	  export PATH="$(top_builddir)/.libs:$$PATH" ; \
 @WITH_PYTHON_TRUE@	  for test in $(PYTESTS) ; do \
-@WITH_PYTHON_TRUE@	    log=`$(PYTHON) $(srcdir)/$$test` ; \
+@WITH_PYTHON_TRUE@	    log=`$(PYTHON) $(srcdir)/$$test 2>&1` ; \
 @WITH_PYTHON_TRUE@	    if [ "$$?" -ne 0 ] ; then \
-@WITH_PYTHON_TRUE@	      echo "-- $$test" ; \
+@WITH_PYTHON_TRUE@	      echo "-- FAILED $$test" ; \
 @WITH_PYTHON_TRUE@	      echo "$$log" ; \
+@WITH_PYTHON_TRUE@	      echo "  $$test" >> test.out ; \
 @WITH_PYTHON_TRUE@	      exit 1 ; \
 @WITH_PYTHON_TRUE@	    fi ; \
 @WITH_PYTHON_TRUE@	  done)
+@WITH_PYTHON_TRUE@	  @[ ! -s test.out ]
 @WITH_PYTHON_FALSE@tests:
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
