diff -Pur x264-snapshot-20111013-2245-orig/common/osdep.h x264-snapshot-20111013-2245/common/osdep.h
--- x264-snapshot-20111013-2245-orig/common/osdep.h	2011-10-13 16:45:02.000000000 -0400
+++ x264-snapshot-20111013-2245/common/osdep.h	2014-04-17 22:48:52.000000000 -0400
@@ -40,6 +40,12 @@
 #include <inttypes.h>
 #endif
 
+#ifdef __INTEL_COMPILER
+#include <mathimf.h>
+#else
+#include <math.h>
+#endif
+
 #if !HAVE_LOG2F
 #define log2f(x) (logf(x)/0.693147180559945f)
 #define log2(x) (log(x)/0.693147180559945)
@@ -59,12 +65,6 @@
 #define S_ISREG(x) (((x) & S_IFMT) == S_IFREG)
 #endif
 
-#ifdef __INTEL_COMPILER
-#include <mathimf.h>
-#else
-#include <math.h>
-#endif
-
 #if (defined(__GNUC__) || defined(__INTEL_COMPILER)) && (ARCH_X86 || ARCH_X86_64)
 #define HAVE_X86_INLINE_ASM 1
 #endif
