From 12ce9b5ffeba776ede786c075795a4dbae94bfa1 Mon Sep 17 00:00:00 2001
From: Nick Wellnhofer <wellnhofer@aevum.de>
Date: Thu, 25 Jan 2024 12:52:14 +0100
Subject: [PATCH] dict: Fix warnings with older clang

---
 libxml.h   | 11 ++++++++---
 testdict.c | 11 ++++++++---
 2 files changed, 16 insertions(+), 6 deletions(-)

diff --git a/libxml.h b/libxml.h
index 00404d2f..d94677b5 100644
--- a/libxml.h
+++ b/libxml.h
@@ -55,9 +55,14 @@
 #endif
 
 #ifdef __clang__
-  #define ATTRIBUTE_NO_SANITIZE_INTEGER \
-    ATTRIBUTE_NO_SANITIZE("unsigned-integer-overflow") \
-    ATTRIBUTE_NO_SANITIZE("unsigned-shift-base")
+  #if __clang_major__ >= 12
+    #define ATTRIBUTE_NO_SANITIZE_INTEGER \
+      ATTRIBUTE_NO_SANITIZE("unsigned-integer-overflow") \
+      ATTRIBUTE_NO_SANITIZE("unsigned-shift-base")
+  #else
+    #define ATTRIBUTE_NO_SANITIZE_INTEGER \
+      ATTRIBUTE_NO_SANITIZE("unsigned-integer-overflow")
+  #endif
 #else
   #define ATTRIBUTE_NO_SANITIZE_INTEGER
 #endif
diff --git a/testdict.c b/testdict.c
index ffbbc947..94f18c2c 100644
--- a/testdict.c
+++ b/testdict.c
@@ -7,9 +7,14 @@
 /**** dictionary tests ****/
 
 #ifdef __clang__
-  #define ATTRIBUTE_NO_SANITIZE_INTEGER \
-    __attribute__ ((no_sanitize("unsigned-integer-overflow"))) \
-    __attribute__ ((no_sanitize("unsigned-shift-base")))
+  #if __clang_major__ >= 12
+    #define ATTRIBUTE_NO_SANITIZE_INTEGER \
+      __attribute__ ((no_sanitize("unsigned-integer-overflow"))) \
+      __attribute__ ((no_sanitize("unsigned-shift-base")))
+  #else
+    #define ATTRIBUTE_NO_SANITIZE_INTEGER \
+      __attribute__ ((no_sanitize("unsigned-integer-overflow")))
+  #endif
 #else
   #define ATTRIBUTE_NO_SANITIZE_INTEGER
 #endif
-- 
GitLab

