diff -ruN xine-lib-1.2.10-orig/configure.ac xine-lib-1.2.10/configure.ac
--- xine-lib-1.2.10-orig/configure.ac	2019-12-13 14:47:50.000000000 -0600
+++ xine-lib-1.2.10/configure.ac	2019-12-28 10:19:10.000000000 -0600
@@ -1260,12 +1260,13 @@
 dnl Even as of 2.61, autoconf is not smart enough to find the X include
 dnl and library paths on Mac OS X, so seed them automatically if they're
 dnl not specified on the configure command-line.
-case "$host_os" in
-    darwin*)
-        test x"$x_includes" = x"NONE" && x_includes="/usr/X11R6/include"
-        test x"$x_libraries" = x"NONE" && x_libraries="/usr/X11R6/lib"
-        ;;
-esac
+#case "$host_os" in
+#    darwin*)
+#        test x"$x_includes" = x"NONE" && x_includes="/usr/X11R6/include"
+#        test x"$x_libraries" = x"NONE" && x_libraries="/usr/X11R6/lib"
+#        ;;
+#esac
+dnl but fink's autoconf has been patched for years.
 
 dnl Check for Xwindows using the autoconf AC_PATH_XTRA macro, which is an
 dnl extension of AC_PATH_X that sets X_CFLAGS and X_LIBS.  It will also set
diff -ruN xine-lib-1.2.10-orig/include/xine/xineutils.h xine-lib-1.2.10/include/xine/xineutils.h
--- xine-lib-1.2.10-orig/include/xine/xineutils.h	2019-12-13 14:47:50.000000000 -0600
+++ xine-lib-1.2.10/include/xine/xineutils.h	2019-12-28 10:19:10.000000000 -0600
@@ -206,12 +206,21 @@
 static inline void *xine_small_memcpy (void *to, const void *from, size_t len) {
   void *t2 = to;
   size_t l2 = len;
+#    if !defined(__clang__)
   __asm__ __volatile__ (
     "cld\n\trep movsb"
     : "=S" (from), "=D" (t2), "=c" (l2), "=m" (*(struct {char foo[len];} *)to)
     : "0"  (from), "1"  (t2), "2"  (l2)
     : "cc"
   );
+#    else /* clang dislikes virtual variable size struct */
+  __asm__ __volatile__ (
+      "cld\n\trep movsb"
+      : "=S" (from), "=D" (t2), "=c" (l2)
+      : "0"  (from), "1"  (t2), "2"  (l2)
+      : "cc", "memory"
+  );
+#    endif
   (void)from;
   (void)t2;
   (void)l2;
diff -ruN xine-lib-1.2.10-orig/src/audio_out/Makefile.am xine-lib-1.2.10/src/audio_out/Makefile.am
--- xine-lib-1.2.10-orig/src/audio_out/Makefile.am	2019-12-13 14:47:50.000000000 -0600
+++ xine-lib-1.2.10/src/audio_out/Makefile.am	2019-12-28 10:19:10.000000000 -0600
@@ -109,7 +109,7 @@
 
 xineplug_ao_out_coreaudio_la_SOURCES = audio_coreaudio_out.c
 xineplug_ao_out_coreaudio_la_LIBADD = $(XINE_LIB)
-xineplug_ao_out_coreaudio_la_LDFLAGS = $(AM_LDFLAGS) -framework CoreAudio -framework AudioUnit
+xineplug_ao_out_coreaudio_la_LDFLAGS = $(AM_LDFLAGS) -framework CoreAudio -framework AudioUnit -framework CoreServices
 
 xineplug_ao_out_pulseaudio_la_SOURCES = audio_pulse_out.c
 xineplug_ao_out_pulseaudio_la_LIBADD = $(XINE_LIB) $(PULSEAUDIO_LIBS) $(LTLIBINTL)
diff -ruN xine-lib-1.2.10-orig/src/demuxers/demux_ts.c xine-lib-1.2.10/src/demuxers/demux_ts.c
--- xine-lib-1.2.10-orig/src/demuxers/demux_ts.c	2019-12-13 14:47:50.000000000 -0600
+++ xine-lib-1.2.10/src/demuxers/demux_ts.c	2019-12-28 10:36:26.000000000 -0600
@@ -3238,7 +3238,6 @@
   this->bpts               = 0;
   this->last_pts[0]        = 0;
   this->last_pts[1]        = 0;
-  this->newpts_fifo        = NULL;
 #  if TS_PACKET_READER == 2
   this->buf_pos            = 0;
   this->buf_size           = 0;
diff -ruN xine-lib-1.2.10-orig/src/input/http_helper.c xine-lib-1.2.10/src/input/http_helper.c
--- xine-lib-1.2.10-orig/src/input/http_helper.c	2019-12-13 14:47:50.000000000 -0600
+++ xine-lib-1.2.10/src/input/http_helper.c	2019-12-28 10:19:58.000000000 -0600
@@ -24,7 +24,9 @@
 #include "config.h"
 #endif
 
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>
+#endif
 #include <string.h>
 #include <stdint.h>
 
diff -ruN xine-lib-1.2.10-orig/src/input/input_http.c xine-lib-1.2.10/src/input/input_http.c
--- xine-lib-1.2.10-orig/src/input/input_http.c	2019-12-13 14:47:50.000000000 -0600
+++ xine-lib-1.2.10/src/input/input_http.c	2019-12-28 10:21:06.000000000 -0600
@@ -30,6 +30,9 @@
 #ifdef HAVE_NETDB_H
 #include <netdb.h>
 #endif
+#ifdef HAVE_SYS_SOCKET_H
+#include <sys/socket.h>
+#endif
 #include <ctype.h>
 #include <errno.h>
 #include <zlib.h>
diff -ruN xine-lib-1.2.10-orig/src/input/tls/tls_gnutls.c xine-lib-1.2.10/src/input/tls/tls_gnutls.c
--- xine-lib-1.2.10-orig/src/input/tls/tls_gnutls.c	2019-12-13 14:47:50.000000000 -0600
+++ xine-lib-1.2.10/src/input/tls/tls_gnutls.c	2019-12-28 10:21:36.000000000 -0600
@@ -41,7 +41,9 @@
 #endif
 
 #ifndef XINE_GNUTLS_3
-#  include <malloc.h>
+#  ifdef HAVE_MALLOC_H
+#    include <malloc.h>
+#  endif
 #  include <string.h>
 #  include <sys/stat.h>
 #  include <dirent.h>
