diff -Nurd libxslt-1.1.39.orig/configure libxslt-1.1.39/configure
--- libxslt-1.1.39.orig/configure	2023-11-16 07:58:37.000000000 -0500
+++ libxslt-1.1.39/configure	2024-01-24 20:57:15.000000000 -0500
@@ -16203,7 +16203,7 @@
 
 EXSLT_LIBDIR='-L${libdir}'
 EXSLT_INCLUDEDIR='-I${includedir}'
-EXSLT_PRIVATE_LIBS="$XSLT_PRIVATE_LIBS $LIBGCRYPT_LIBS"
+EXSLT_PRIVATE_LIBS="$XSLT_PRIVATE_LIBS"
 
 
 
diff -Nurd libxslt-1.1.39.orig/tests/Makefile.in libxslt-1.1.39/tests/Makefile.in
--- libxslt-1.1.39.orig/tests/Makefile.in	2023-11-16 07:58:36.000000000 -0500
+++ libxslt-1.1.39/tests/Makefile.in	2024-01-24 22:29:42.000000000 -0500
@@ -435,7 +435,7 @@
 @WITH_MODULES_TRUE@plugindir = $(abs_builddir)/.libs
 @WITH_MODULES_TRUE@xmlsoft_org_xslt_testplugin_la_CFLAGS = $(AM_CFLAGS) -DMODULE_COMPILE
 @WITH_MODULES_TRUE@xmlsoft_org_xslt_testplugin_la_SOURCES = testplugin.c
-@WITH_MODULES_TRUE@xmlsoft_org_xslt_testplugin_la_LIBADD = $(top_builddir)/libxslt/libxslt.la $(EXTRA_LIBS)
+@WITH_MODULES_TRUE@xmlsoft_org_xslt_testplugin_la_LIBADD = $(top_builddir)/libxslt/libxslt.la $(LIBXML_LIBS) $(EXTRA_LIBS)
 @WITH_MODULES_TRUE@xmlsoft_org_xslt_testplugin_la_LDFLAGS = \
 @WITH_MODULES_TRUE@	$(AM_LDFLAGS) -no-undefined \
 @WITH_MODULES_TRUE@	-module -avoid-version -rpath $(plugindir)
diff -Nurd libxslt-1.1.39.orig/xslt-config.in libxslt-1.1.39/xslt-config.in
--- libxslt-1.1.39.orig/xslt-config.in	2023-11-16 07:36:28.000000000 -0500
+++ libxslt-1.1.39/xslt-config.in	2024-01-24 20:57:15.000000000 -0500
@@ -15,8 +15,8 @@
 
   --prefix=DIR		change XSLT prefix [default $prefix]
   --exec-prefix=DIR	change XSLT executable prefix [default $exec_prefix]
-  --libs		print library linking information
-                        add --dynamic to print only shared libraries
+  --libs		print library dynamic linking information
+                        (--dynamic is a no-op, always assumed)
   --cflags		print pre-processor and compiler flags
   --plugins		print plugin directory
   --help		display this help and exit
@@ -85,7 +85,7 @@
             shift
             libs="@XSLT_LIBS@"
         else
-            libs="@XSLT_LIBS@ @XSLT_PRIVATE_LIBS@"
+            libs="@XSLT_LIBS@"
         fi
 
         if [ "@XSLT_LIBDIR@" != "-L/usr/lib" -a "@XSLT_LIBDIR@" != "-L/usr/lib64" ]; then
