diff -Nurd -x'*~' m17n-lib-1.6.2.orig/README m17n-lib-1.6.2/README
--- m17n-lib-1.6.2.orig/README	2010-09-21 23:26:06.000000000 -0400
+++ m17n-lib-1.6.2/README	2015-09-07 03:17:43.000000000 -0400
@@ -110,17 +110,18 @@
 	% ./configure --without-gui
 
 This file under `example' sub-directory is a Japanese resource file
-for m17n-edit.  It is not installed but useful in Japanese locale.  Copy
-it to your home directory (or, for instance,
-/usr/X11R6/lib/X11/ja/app-defaults) and rename it to "M17NEdit" if you
-want to see labels in Japanese:
+for m17n-edit:
 	M17NEdit.ja
+It is installed as (the latter is converted to UTF-8 before):
+	@PREFIX@/etc/app-defaults/ja_JP.eucJP/M17NEdit
+	@PREFIX@/etc/app-defaults/ja_JP.UTF-8/M17NEdit
 
-These text files under `example' sub-directory are not installed but
-useful for testing the rendering engine of the m17n library:
+These text files under `example' sub-directory are useful for testing
+the rendering engine of the m17n library:
 	HELLO.utf8 HELLO.xml HELLO-ja.utf8 HELLO-ja.xml HELLO.html
 XXX.xml are generated from XXX.utf8 by attaching text property
-`language' and serializing.
+`language' and serializing. They are installed into:
+	@PREFIX@/share/m17n-lib/examples
 
 Please read also INSTALL for the generic installation instructions.
 
diff -Nurd -x'*~' m17n-lib-1.6.2.orig/configure m17n-lib-1.6.2/configure
--- m17n-lib-1.6.2.orig/configure	2010-10-04 20:19:51.000000000 -0400
+++ m17n-lib-1.6.2/configure	2015-09-07 03:17:43.000000000 -0400
@@ -16866,10 +16866,13 @@
 
 done
 
+save_CPPFLAGS=$CPPFLAGS
+CPPFLAGS="$CPPFLAGS $X_CFLAGS"
 ac_fn_c_check_header_mongrel "$LINENO" "X11/Xaw/Command.h" "ac_cv_header_X11_Xaw_Command_h" "$ac_includes_default"
 if test "x$ac_cv_header_X11_Xaw_Command_h" = x""yes; then :
   HAVE_XAW=yes
 fi
+CPPFLAGS=$save_CPPFLAGS
 
 
 
@@ -19587,11 +19590,7 @@
 CPPFLAGS="$CPPFLAGS $X_CFLAGS"
 
 
-SHLIB_EXT=".so"
-case $host_os in
-darwin* | rhapsody*)
-  SHLIB_EXT=".dylib";;
-esac
+eval "SHLIB_EXT=$shrext_cmds"
 
 cat >>confdefs.h <<_ACEOF
 #define DLOPEN_SHLIB_EXT "$SHLIB_EXT"
diff -Nurd -x'*~' m17n-lib-1.6.2.orig/src/Makefile.in m17n-lib-1.6.2/src/Makefile.in
--- m17n-lib-1.6.2.orig/src/Makefile.in	2010-10-04 20:19:54.000000000 -0400
+++ m17n-lib-1.6.2/src/Makefile.in	2015-09-07 03:17:43.000000000 -0400
@@ -488,7 +488,7 @@
 @WITH_GUI_TRUE@module_LTLIBRARIES = libm17n-X.la libm17n-gd.la
 @WITH_GUI_TRUE@X_LD_FLAGS = ${X_PRE_LIBS} ${X_LIBS} @X11_LD_FLAGS@ ${X_EXTRA_LIBS}
 @WITH_GUI_TRUE@libm17n_X_la_SOURCES = m17n-X.h m17n-X.c
-@WITH_GUI_TRUE@libm17n_X_la_LIBADD = ${X_LD_FLAGS} @XFT2_LD_FLAGS@ ${top_builddir}/src/libm17n-core.la ${top_builddir}/src/libm17n.la ${top_builddir}/src/libm17n-flt.la ${top_builddir}/src/libm17n-gui.la
+@WITH_GUI_TRUE@libm17n_X_la_LIBADD = @XFT2_LD_FLAGS@ ${top_builddir}/src/libm17n-core.la ${top_builddir}/src/libm17n.la ${top_builddir}/src/libm17n-flt.la ${top_builddir}/src/libm17n-gui.la ${X_LD_FLAGS}
 @WITH_GUI_TRUE@libm17n_X_la_LDFLAGS = -avoid-version -module
 @WITH_GUI_TRUE@libm17n_gd_la_SOURCES = m17n-gd.c
 @WITH_GUI_TRUE@libm17n_gd_la_LIBADD = @GD_LD_FLAGS@ @FREETYPE_LD_FLAGS@ ${top_builddir}/src/libm17n-core.la ${top_builddir}/src/libm17n.la ${top_builddir}/src/libm17n-flt.la ${top_builddir}/src/libm17n-gui.la
