diff -Nurd neon-0.30.2.orig/configure neon-0.30.2/configure
--- neon-0.30.2.orig/configure	2018-02-27 02:07:49.000000000 -0500
+++ neon-0.30.2/configure	2018-02-27 02:09:17.000000000 -0500
@@ -16356,7 +16356,7 @@
 fi
 
 
-CPPFLAGS="$CPPFLAGS -I\${top_builddir}"
+CPPFLAGS="-I\${top_builddir} $CPPFLAGS"
 
 # Check whether --enable-memleak was given.
 if test "${enable_memleak+set}" = set; then :
@@ -16365,7 +16365,7 @@
 
 
 if test "$enable_memleak" = "yes"; then
-  CPPFLAGS="$CPPFLAGS -DNEON_MEMLEAK -I\$(top_srcdir)/src"
+  CPPFLAGS="-DNEON_MEMLEAK -I\$(top_srcdir)/src $CPPFLAGS"
   # disable 'make install'
   ALLOW_INSTALL=memleak
 fi
diff -Nurd neon-0.30.2.orig/test/Makefile.in neon-0.30.2/test/Makefile.in
--- neon-0.30.2.orig/test/Makefile.in	2018-02-27 02:07:49.000000000 -0500
+++ neon-0.30.2/test/Makefile.in	2018-02-27 02:08:11.000000000 -0500
@@ -1,7 +1,7 @@
 # Makefile for neon test suite.
 
 SHELL = @SHELL@
-CPPFLAGS = @CPPFLAGS@ -I. -I$(top_srcdir)/src -I$(top_srcdir)/test/common
+CPPFLAGS = -I. -I$(top_srcdir)/src -I$(top_srcdir)/test/common @CPPFLAGS@
 CFLAGS = @CFLAGS@ @NEON_CFLAGS@
 LDFLAGS = @LDFLAGS@
 DEFS = @DEFS@
