diff -ruN gsl-ocaml-0.6.3-orig/Makefile gsl-ocaml-0.6.3/Makefile
--- gsl-ocaml-0.6.3-orig/Makefile	2012-05-02 12:37:18.000000000 -0500
+++ gsl-ocaml-0.6.3/Makefile	2020-09-13 05:46:47.000000000 -0500
@@ -148,8 +148,19 @@
 	ocamlmktop -I . -o gsl-ocaml$(EXE) bigarray.cma gsl.cma
 
 install : all
-	$(OCAMLFIND) install gsl META \
-          libmlgsl.$(A) dllmlgsl.$(D) gsl.cma gsl.cmxa gsl.$(A) $(CMI) $(MLOPTOBJ) 
+	mkdir -p $(DESTDIR)$(INSTALLDIR)
+	cp $(STUBS) gsl.cma gsl.cmxa gsl.$(A) \
+          $(CMI) $(MLOPTOBJ) $(DESTDIR)$(INSTALLDIR)
+	mkdir -p $(DESTDIR)$(OCAMLDIR)/stublibs
+	if test -w $(DESTDIR)$(OCAMLDIR)/stublibs ; then \
+          ln -sf $(INSTALLDIR)/dllmlgsl.$(D) $(DESTDIR)$(OCAMLDIR)/stublibs ; fi
+
+install-findlib : all
+	export FINDLIBDIR=$$($(OCAMLFIND) printconf destdir) ; \
+	test -d "$${FINDLIBDIR}/stublibs" && mkdir -p $(DESTDIR)$${FINDLIBDIR}/stublibs ; \
+	OCAMLFIND_DESTDIR=$(DESTDIR)$${FINDLIBDIR} \
+        $(OCAMLFIND) install gsl META \
+          libmlgsl.$(A) dllmlgsl.$(D) gsl.cma gsl.cmxa gsl.$(A) $(CMI) $(MLOPTOBJ)
 
 gsl-ocaml.odoc : $(MLOBJ) $(CMI)
 	$(OCAMLDOC) -v -dump $@ $(filter-out gsl_misc.%, $(filter %.mli, $(SRC)))
diff -ruN gsl-ocaml-0.6.3-orig/io.h gsl-ocaml-0.6.3/io.h
--- gsl-ocaml-0.6.3-orig/io.h	2012-05-02 12:37:18.000000000 -0500
+++ gsl-ocaml-0.6.3/io.h	2020-09-13 05:48:29.000000000 -0500
@@ -19,6 +19,8 @@
 #include <caml/misc.h>
 #include <caml/mlvalues.h>
 
+typedef uint32_t uint32;
+
 #ifndef IO_BUFFER_SIZE
 #define IO_BUFFER_SIZE 4096
 #endif
