diff -ruN openldap-2.6.4-orig/build/ltmain.sh openldap-2.6.4/build/ltmain.sh
--- openldap-2.6.4-orig/build/ltmain.sh	2023-02-08 12:53:35.000000000 -0600
+++ openldap-2.6.4/build/ltmain.sh	2023-03-09 06:34:53.000000000 -0600
@@ -8140,7 +8140,7 @@
 	  *)
 	    if test no = "$installed"; then
 	      func_append notinst_deplibs " $lib"
-	      need_relink=yes
+	      need_relink=no
 	    fi
 	    ;;
 	  esac
diff -ruN openldap-2.6.4-orig/configure openldap-2.6.4/configure
--- openldap-2.6.4-orig/configure	2023-02-08 12:53:35.000000000 -0600
+++ openldap-2.6.4/configure	2023-03-09 06:34:53.000000000 -0600
@@ -9210,9 +9210,9 @@
       case ${MACOSX_DEPLOYMENT_TARGET-10.0},$host in
 	10.0,*86*-darwin8*|10.0,*-darwin[91]*)
 	  _lt_dar_allow_undefined='$wl-undefined ${wl}dynamic_lookup' ;;
-	10.[012][,.]*)
+	10.[012][,.]*|,*powerpc*)
 	  _lt_dar_allow_undefined='$wl-flat_namespace $wl-undefined ${wl}suppress' ;;
-	10.*)
+	*)
 	  _lt_dar_allow_undefined='$wl-undefined ${wl}dynamic_lookup' ;;
       esac
     ;;
@@ -14977,6 +14977,7 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdlib.h>
 #include <ctype.h>
 #ifndef HAVE_EBCDIC
 #	define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
@@ -20853,6 +20854,7 @@
 
 #include <sys/types.h>
 #include <sys/time.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <pthread.h>
 #ifndef NULL
@@ -23240,6 +23242,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+#include <stdlib.h>
 #include <ctype.h>
 main()
 {
