diff -ruN openldap-2.4.56.orig/libraries/libldap/tls2.c openldap-2.4.56/libraries/libldap/tls2.c
--- openldap-2.4.56.orig/libraries/libldap/tls2.c	2020-11-10 12:22:13.000000000 -0600
+++ openldap-2.4.56/libraries/libldap/tls2.c	2020-11-13 19:39:54.000000000 -0600
@@ -74,6 +74,9 @@
 	{ BER_BVNULL, BER_BVNULL }
 };
 
+static int
+ldap_pvt_tls_check_hostname( LDAP *ld, void *s, const char *name_in );
+
 #ifdef HAVE_TLS
 
 void
@@ -503,7 +506,7 @@
 	return rc;
 }
 
-int
+static int
 ldap_pvt_tls_check_hostname( LDAP *ld, void *s, const char *name_in )
 {
 	tls_session *session = s;
diff -ruN openldap-2.4.56.orig/servers/slapd/back-ldap/bind.c openldap-2.4.56/servers/slapd/back-ldap/bind.c
--- openldap-2.4.56.orig/servers/slapd/back-ldap/bind.c	2020-11-10 12:22:13.000000000 -0600
+++ openldap-2.4.56/servers/slapd/back-ldap/bind.c	2020-11-14 04:09:02.000000000 -0600
@@ -145,6 +145,9 @@
 }
 #endif /* LDAP_BACK_PRINT_CONNTREE */
 
+void
+slap_client_keepalive(LDAP *ld, slap_keepalive *sk);
+
 static int
 ldap_back_freeconn( ldapinfo_t *li, ldapconn_t *lc, int dolock );
 
