diff -ruN osxfuse-fuse-98b9960-orig/lib/fuse.c osxfuse-fuse-98b9960/lib/fuse.c
--- osxfuse-fuse-98b9960-orig/lib/fuse.c	2013-12-28 14:32:40.000000000 -0600
+++ osxfuse-fuse-98b9960/lib/fuse.c	2021-02-07 14:56:31.000000000 -0600
@@ -43,6 +43,9 @@
 #define FUSE_UNKNOWN_INO 0xffffffff
 #define OFFSET_MAX 0x7fffffffffffffffLL
 
+int fuse_reply_xtimes(fuse_req_t req, const struct timespec *bkuptime,
+		      const struct timespec *crtime);
+
 struct fuse_config {
 	unsigned int uid;
 	unsigned int gid;
diff -ruN osxfuse-fuse-98b9960-orig/lib/fuse_darwin.c osxfuse-fuse-98b9960/lib/fuse_darwin.c
--- osxfuse-fuse-98b9960-orig/lib/fuse_darwin.c	2013-12-28 14:32:40.000000000 -0600
+++ osxfuse-fuse-98b9960/lib/fuse_darwin.c	2021-02-07 16:09:09.000000000 -0600
@@ -8,6 +8,7 @@
 #include <unistd.h>
 #include <pthread.h>
 #include <stdbool.h>
+#include <stdlib.h>
 
 #include <fuse_lowlevel.h>
 #include "fuse_darwin_private.h"
diff -ruN osxfuse-fuse-98b9960-orig/modules/subdir.c osxfuse-fuse-98b9960/lib/modules/subdir.c
--- osxfuse-fuse-98b9960-orig/lib/modules/subdir.c	2021-04-02 13:05:07.000000000 -0500
+++ osxfuse-fuse-98b9960/lib/modules/subdir.c	2021-04-02 13:07:00.000000000 -0500
@@ -306,6 +306,10 @@
 }
 
 #ifdef __APPLE__
+int fuse_fs_setattr_x(struct fuse_fs *fs, const char *path,
+		      struct setattr_x *attr);
+int fuse_fs_fsetattr_x(struct fuse_fs *fs, const char *path,
+		       struct setattr_x *attr, struct fuse_file_info *fi);
 static int subdir_setattr_x(const char *path, struct setattr_x *attr)
 {
 	struct subdir *d = subdir_get();
diff -ruN osxfuse-fuse-98b9960-orig/lib/mount_darwin.c osxfuse-fuse-98b9960/lib/mount_darwin.c
--- osxfuse-fuse-98b9960-orig/lib/mount_darwin.c	2013-12-28 14:32:40.000000000 -0600
+++ osxfuse-fuse-98b9960/lib/mount_darwin.c	2021-04-02 07:23:13.000000000 -0500
@@ -127,7 +127,7 @@
         goto Return;
     }
 
-    require_action(pid != -1, Return, result = errno);
+    __Require_Action(pid != -1, Return, result = errno);
 
     while ((terminated_pid = wait4(pid, (int *)&status, 0, NULL)) < 0) {
         /* retry if EINTR, else break out with error */
@@ -143,7 +143,7 @@
     }
 
 Return:
-    check_noerr_string(result, strerror(errno));
+    __Check_noErr_String(result, strerror(errno));
     
     return result;
 }
diff -ruN osxfuse-fuse-98b9960-orig/lib/modules/threadid.c osxfuse-fuse-98b9960/lib/modules/threadid.c
--- osxfuse-fuse-98b9960-orig/lib/modules/threadid.c	2021-04-02 13:16:53.000000000 -0500
+++ osxfuse-fuse-98b9960/lib/modules/threadid.c	2021-04-02 13:17:48.000000000 -0500
@@ -17,6 +17,10 @@
 #include <fuse.h>
 
 extern int pthread_setugid_np(uid_t, gid_t);
+int fuse_fs_setattr_x(struct fuse_fs *fs, const char *path,
+		      struct setattr_x *attr);
+int fuse_fs_fsetattr_x(struct fuse_fs *fs, const char *path,
+		       struct setattr_x *attr, struct fuse_file_info *fi);
 
 #define THREADID_PRE()                                      \
                                                             \
diff -ruN osxfuse-fuse-98b9960-orig/lib/modules/volicon.c osxfuse-fuse-98b9960/lib/modules/volicon.c
--- osxfuse-fuse-98b9960-orig/lib/modules/volicon.c.orig	2013-12-28 14:32:40.000000000 -0600
+++ osxfuse-fuse-98b9960/lib/modules/volicon.c	2021-04-02 13:20:44.000000000 -0500
@@ -31,6 +31,11 @@
 #define VOLICON_ICON_MAGIC_PATH    "/.VolumeIcon.icns"
 #define VOLICON_ICON_MAXSIZE       (4 * 1024 * 1024)
 
+int fuse_fs_setattr_x(struct fuse_fs *fs, const char *path,
+		      struct setattr_x *attr);
+int fuse_fs_fsetattr_x(struct fuse_fs *fs, const char *path,
+		       struct setattr_x *attr, struct fuse_file_info *fi);
+
 struct FndrGenericInfo {
     u_int32_t   ignored0;
     u_int32_t   ignored1;
