diff -ruN phonon-4.10.3-orig/CMakeLists.txt phonon-4.10.3/CMakeLists.txt
--- phonon-4.10.3-orig/CMakeLists.txt	2019-07-19 06:42:16.000000000 -0500
+++ phonon-4.10.3/CMakeLists.txt	2021-10-08 11:51:34.000000000 -0500
@@ -67,20 +67,24 @@
     macro_log_feature(Qt5DBus_FOUND "Qt5 DBus (qtbase)" ${DBUS_DESC} "http://www.qt.gitorious.net/qt/qtbase" FALSE)
     set(QT_QTDBUS_FOUND ${Qt5DBus_FOUND})
 
-    # Designer Plugin
-    find_package(Qt5Designer)
-    macro_log_feature(Qt5Designer_FOUND "Qt5 Designer (qttools)" ${DESIGNER_DESC} "http://www.qt.gitorious.net/qt/qttools" FALSE)
-    set(QT_QTDESIGNER_FOUND ${Qt5Designer_FOUND})
+    if (PHONON_BUILD_DESIGNER_PLUGIN)
+        # Designer Plugin
+        find_package(Qt5Designer)
+        macro_log_feature(Qt5Designer_FOUND "Qt5 Designer (qttools)" ${DESIGNER_DESC} "http://www.qt.gitorious.net/qt/qttools" FALSE)
+        set(QT_QTDESIGNER_FOUND ${Qt5Designer_FOUND})
+    endif (PHONON_BUILD_DESIGNER_PLUGIN)
 
     # VideoGraphicsObject GL Painting
     find_package(Qt5OpenGL)
     macro_log_feature(Qt5OpenGL_FOUND "Qt5 OpenGL (qtbase)" ${GL_DESC} "http://www.qt.gitorious.net/qt/qtbase" FALSE)
     set(QT_QTOPENGL_FOUND ${Qt5OpenGL_FOUND})
 
-    # Declarative/QML1
-    find_package(Qt5Declarative)
-    macro_log_feature(Qt5Declarative_FOUND "Qt5 Declarative (qtquick1)" ${DECL_DESC} "http://www.qt.gitorious.net/qt/qtquick1" FALSE)
-    set(QT_QTDECLARATIVE_FOUND ${Qt5Declarative_FOUND})
+    if (PHONON_BUILD_DECLARATIVE_PLUGIN)
+      # Declarative/QML1
+        find_package(Qt5Declarative)
+        macro_log_feature(Qt5Declarative_FOUND "Qt5 Declarative (qtquick1)" ${DECL_DESC} "http://www.qt.gitorious.net/qt/qtquick1" FALSE)
+        set(QT_QTDECLARATIVE_FOUND ${Qt5Declarative_FOUND})
+    endif (PHONON_BUILD_DECLARATIVE_PLUGIN)
 
     set(PHONON_PKG_DEPENDENCIES "Qt5Core Qt5Gui Qt5DBus")
 endif (NOT PHONON_BUILD_PHONON4QT5)
