diff -ruN qtbase-opensource-src-5.7.1-orig/src/corelib/io/qstandardpaths_mac.mm qtbase-opensource-src-5.7.1/src/corelib/io/qstandardpaths_mac.mm
--- qtbase-opensource-src-5.7.1-orig/src/corelib/io/qstandardpaths_mac.mm	2016-12-01 02:17:04.000000000 -0600
+++ qtbase-opensource-src-5.7.1/src/corelib/io/qstandardpaths_mac.mm	2021-10-07 20:36:20.000000000 -0500
@@ -224,6 +224,11 @@
                 dirs.append(path);
         }
     }
+    if (type == GenericDataLocation || type == AppDataLocation || type == AppLocalDataLocation) {
+        const QString path = QLatin1String("@QT_FINK_PREFIX@/share");
+        if (!path.isEmpty() && !dirs.contains(path))
+            dirs.append(path);
+    }
 
     if (type == AppDataLocation || type == AppLocalDataLocation) {
         CFBundleRef mainBundle = CFBundleGetMainBundle();
