diff -ruN qtbase-everywhere-src-6.7.0-orig/cmake/QtAutoDetectHelpers.cmake qtbase-everywhere-src-6.7.0/cmake/QtAutoDetectHelpers.cmake
--- qtbase-everywhere-src-6.7.0-orig/cmake/QtAutoDetectHelpers.cmake	2024-03-19 10:46:43
+++ qtbase-everywhere-src-6.7.0/cmake/QtAutoDetectHelpers.cmake	2024-04-24 17:23:04
@@ -284,7 +284,7 @@
         if(NOT CMAKE_OSX_DEPLOYMENT_TARGET)
             if(NOT CMAKE_SYSTEM_NAME)
                 # macOS
-                set(version "11.0")
+                set(version "@MACOSX_DEPLOYMENT_TARGET@")
             elseif(CMAKE_SYSTEM_NAME STREQUAL iOS)
                 set(version "14.0")
             endif()
diff -ruN qtbase-everywhere-src-6.7.0-orig/mkspecs/common/gcc-base-mac.conf qtbase-everywhere-src-6.7.0/mkspecs/common/gcc-base-mac.conf
--- qtbase-everywhere-src-6.7.0-orig/mkspecs/common/gcc-base-mac.conf	2024-03-19 10:46:43
+++ qtbase-everywhere-src-6.7.0/mkspecs/common/gcc-base-mac.conf	2024-04-22 21:01:24
@@ -15,7 +15,7 @@
 QMAKE_COMPILER_DEFINES += __APPLE__ __GNUC__=4 __APPLE_CC__
 
 QMAKE_LFLAGS_SHLIB         += -single_module -dynamiclib
-QMAKE_LFLAGS_PLUGIN        += $$QMAKE_LFLAGS_SHLIB
+QMAKE_LFLAGS_PLUGIN        += -bundle
 QMAKE_LFLAGS_INCREMENTAL   += -undefined suppress -flat_namespace
 QMAKE_LFLAGS_SONAME        += -install_name$${LITERAL_WHITESPACE}
 QMAKE_LFLAGS_HEADERPAD     += -headerpad_max_install_names
diff -ruN qtbase-everywhere-src-6.7.0-orig/mkspecs/common/mac.conf qtbase-everywhere-src-6.7.0/mkspecs/common/mac.conf
--- qtbase-everywhere-src-6.7.0-orig/mkspecs/common/mac.conf	2024-03-19 10:46:43
+++ qtbase-everywhere-src-6.7.0/mkspecs/common/mac.conf	2024-04-22 21:02:11
@@ -14,6 +14,7 @@
 
 QMAKE_RESOURCE          = /Developer/Tools/Rez
 QMAKE_EXTENSION_SHLIB   = dylib
+QMAKE_EXTENSION_PLUGIN  = so
 QMAKE_LIBDIR            =
 
 # sdk.prf will prefix the proper SDK sysroot
diff -ruN qtbase-everywhere-src-6.7.0-orig/mkspecs/common/macx.conf qtbase-everywhere-src-6.7.0/mkspecs/common/macx.conf
--- qtbase-everywhere-src-6.7.0-orig/mkspecs/common/macx.conf	2024-03-19 10:46:43
+++ qtbase-everywhere-src-6.7.0/mkspecs/common/macx.conf	2024-04-23 20:13:13
@@ -5,7 +5,7 @@
 QMAKE_PLATFORM         += macos osx macx
 QMAKE_MAC_SDK           = macosx
 
-QMAKE_MACOSX_DEPLOYMENT_TARGET = 11
+QMAKE_MACOSX_DEPLOYMENT_TARGET = @MACOSX_DEPLOYMENT_TARGET@
 
 QT_MAC_SDK_VERSION_MIN = 11
 
diff -ruN qtbase-everywhere-src-6.7.0-orig/mkspecs/features/create_cmake.prf qtbase-everywhere-src-6.7.0/mkspecs/features/create_cmake.prf
--- qtbase-everywhere-src-6.7.0-orig/mkspecs/features/create_cmake.prf	2024-03-19 10:46:43
+++ qtbase-everywhere-src-6.7.0/mkspecs/features/create_cmake.prf	2024-04-22 21:03:18
@@ -229,7 +229,7 @@
         }
     } else {
         mac {
-            isEmpty(CMAKE_STATIC_TYPE): CMAKE_PLUGIN_EXT = .dylib
+            isEmpty(CMAKE_STATIC_TYPE): CMAKE_PLUGIN_EXT = .so
             else: CMAKE_PLUGIN_EXT = .a
 
             CMAKE_PLUGIN_LOCATION_RELEASE = $$PLUGIN_TYPE/lib$${CMAKE_QT_STEM}$${CMAKE_PLUGIN_EXT}
