diff -Nurd slang-2.3.3.orig/modules/Makefile.in slang-2.3.3/modules/Makefile.in
--- slang-2.3.3.orig/modules/Makefile.in	2022-08-05 06:33:27.000000000 -0400
+++ slang-2.3.3/modules/Makefile.in	2023-04-09 22:33:01.000000000 -0400
@@ -91,7 +91,7 @@
 
 LIBS = $(LDFLAGS) $(RPATH) $(SLANG_LIB) $(DL_LIB) @M_LIB@
 INCS = $(SLANG_SRCINC)
-COMPILE_CMD = $(CC_SHARED) $(INCS)
+COMPILE_CMD = $(CC_SHARED) $(INCS) $(CPPFLAGS)
 # ---------------------------------------------------------------------------
 all: $(OBJDIR)/Makefile
 	cd $(OBJDIR) && $(MAKE) $(MODULES)
diff -Nurd slang-2.3.3.orig/slsh/Makefile.in slang-2.3.3/slsh/Makefile.in
--- slang-2.3.3.orig/slsh/Makefile.in	2022-08-05 06:33:27.000000000 -0400
+++ slang-2.3.3/slsh/Makefile.in	2023-04-09 22:15:38.000000000 -0400
@@ -80,7 +80,7 @@
 INST_LIBS = $(DEST_LIB_DIR) $(RPATH) $(SLANG_INST_LIB) -lslang $(LDFLAGS) $(READLINE_LIB) $(DYNAMIC_LIBS)
 DEFS = -DSLSH_CONF_DIR='"$(SLSH_CONF_DIR)"' -DSLSH_PATH='"$(SLSH_LIB_DIR)"' \
  -DSLSH_CONF_DIR_ENV='$(SLSH_CONF_DIR_ENV)' -DSLSH_LIB_DIR_ENV='$(SLSH_LIB_DIR_ENV)' \
- -DSLSH_PATH_ENV='$(SLSH_PATH_ENV)' $(SLSYSWRAP_DEF)
+ -DSLSH_PATH_ENV='$(SLSH_PATH_ENV)' $(SLSYSWRAP_DEF) $(CPPFLAGS)
 SDEFS = $(DEFS) -DSLSH_STATIC
 SRC_LIBS = $(SLANG_ELFLIB) -lslang $(SLSYSWRAP_LIB) $(LDFLAGS) $(READLINE_LIB) $(DYNAMIC_LIBS)
 STATIC_SRC_LIBS = $(SLANG_OBJLIB) -lslang $(SLSYSWRAP_LIB) $(LDFLAGS) $(READLINE_LIB) $(STATIC_LIBS)
diff -Nurd slang-2.3.3.orig/src/Makefile.in slang-2.3.3/src/Makefile.in
--- slang-2.3.3.orig/src/Makefile.in	2022-08-05 06:33:27.000000000 -0400
+++ slang-2.3.3/src/Makefile.in	2023-04-09 22:07:18.000000000 -0400
@@ -37,7 +37,7 @@
 #---------------------------------------------------------------------------
 # Misc Libraries
 MISC_TERMINFO_DIRS = @MISC_TERMINFO_DIRS@
-OTHER_CFLAGS = @DSLSYSWRAP@
+OTHER_CFLAGS = @DSLSYSWRAP@ $(CPPFLAGS)
 OTHER_LIBS = @LIB_SLSYSWRAP@ @TERMCAP@ @LIBS@ @M_LIB@
 #---------------------------------------------------------------------------
 # Directory where library is going to go when installed
