diff -ruN strigi-0.7.8.orig/libstreamanalyzer/lib/rdfnamespaces.h strigi-0.7.8/libstreamanalyzer/lib/rdfnamespaces.h
--- strigi-0.7.8.orig/libstreamanalyzer/lib/rdfnamespaces.h	2013-02-05 15:34:52.000000000 -0600
+++ strigi-0.7.8/libstreamanalyzer/lib/rdfnamespaces.h	2020-11-16 05:50:41.000000000 -0600
@@ -18,7 +18,7 @@
  * Boston, MA 02110-1301, USA.
  */
 
-#ifndef RDFNAMESAPCES
+#ifndef RDFNAMESPACES
 #define RDFNAMESPACES
 
 // This is a list of the most often encountered rdf namespaces used by analyzers
diff -ruN strigi-0.7.8.orig/libstreamanalyzer/plugins/indexers/cluceneindexer/CMakeLists.txt strigi-0.7.8/libstreamanalyzer/plugins/indexers/cluceneindexer/CMakeLists.txt
--- strigi-0.7.8.orig/libstreamanalyzer/plugins/indexers/cluceneindexer/CMakeLists.txt	2013-02-05 15:34:52.000000000 -0600
+++ strigi-0.7.8/libstreamanalyzer/plugins/indexers/cluceneindexer/CMakeLists.txt	2020-11-16 05:50:20.000000000 -0600
@@ -1,5 +1,7 @@
 include (CheckCXXCompilerFlag)
 
+ADD_DEFINITIONS(-I${CLUCENE_LIBRARY_DIR})
+
 IF(WIN32)
     # why only here and not for the complete project?
     ADD_DEFINITIONS(-D_UNICODE)
diff -ruN strigi-0.7.8.orig/libstreamanalyzer/plugins/indexers/clucenengindexer/cluceneindexreader.cpp strigi-0.7.8/libstreamanalyzer/plugins/indexers/clucenengindexer/cluceneindexreader.cpp
--- strigi-0.7.8.orig/libstreamanalyzer/plugins/indexers/clucenengindexer/cluceneindexreader.cpp	2013-02-05 15:34:52.000000000 -0600
+++ strigi-0.7.8/libstreamanalyzer/plugins/indexers/clucenengindexer/cluceneindexreader.cpp	2020-11-16 05:51:21.000000000 -0600
@@ -321,7 +321,7 @@
         string size = value;
         doc.size = atoi(size.c_str());
     } else {
-        doc.properties.insert(make_pair<const string, string>(
+        doc.properties.insert(make_pair(
             wchartoutf8(name), value));
     }
 }
diff -ruN strigi-0.7.8.orig/strigidaemon/bin/daemon/dbus/dbuscpp/dbusmessagereader.cpp strigi-0.7.8/strigidaemon/bin/daemon/dbus/dbuscpp/dbusmessagereader.cpp
--- strigi-0.7.8.orig/strigidaemon/bin/daemon/dbus/dbuscpp/dbusmessagereader.cpp	2013-01-25 11:49:35.000000000 -0600
+++ strigi-0.7.8/strigidaemon/bin/daemon/dbus/dbuscpp/dbusmessagereader.cpp	2020-11-16 05:50:20.000000000 -0600
@@ -52,7 +52,7 @@
     return *this;
 }
 DBusMessageReader&
-DBusMessageReader::operator>>(dbus_uint64_t& s) {
+DBusMessageReader::operator>>(STRIGI_UINT64_TYPE& s) {
     if (!isOk()) return *this;
     if (DBUS_TYPE_UINT64 != dbus_message_iter_get_arg_type(&it)) {
         close();
@@ -63,7 +63,7 @@
     return *this;
 }
 DBusMessageReader&
-DBusMessageReader::operator>>(dbus_int64_t& s) {
+DBusMessageReader::operator>>(STRIGI_INT64_TYPE& s) {
     if (!isOk()) return *this;
     if (DBUS_TYPE_INT64 != dbus_message_iter_get_arg_type(&it)) {
         close();
diff -ruN strigi-0.7.8.orig/strigidaemon/bin/daemon/dbus/dbuscpp/dbusmessagereader.h strigi-0.7.8/strigidaemon/bin/daemon/dbus/dbuscpp/dbusmessagereader.h
--- strigi-0.7.8.orig/strigidaemon/bin/daemon/dbus/dbuscpp/dbusmessagereader.h	2013-01-25 11:49:35.000000000 -0600
+++ strigi-0.7.8/strigidaemon/bin/daemon/dbus/dbuscpp/dbusmessagereader.h	2020-11-16 05:50:20.000000000 -0600
@@ -32,6 +32,14 @@
 #include <stdint.h>
 #endif
 
+#ifdef __x86_64__
+#define STRIGI_UINT64_TYPE dbus_uint64_t
+#define STRIGI_INT64_TYPE  dbus_int64_t
+#else
+#define STRIGI_UINT64_TYPE uint64_t
+#define STRIGI_INT64_TYPE  int64_t
+#endif
+
 class DBusMessageReader {
 private:
     DBusMessage* msg;
@@ -62,8 +70,8 @@
     DBusMessageReader& operator>>(std::string& s);
     DBusMessageReader& operator>>(dbus_int32_t& s);
     DBusMessageReader& operator>>(dbus_uint32_t& s);
-    DBusMessageReader& operator>>(dbus_uint64_t& s);
-    DBusMessageReader& operator>>(dbus_int64_t& s);
+    DBusMessageReader& operator>>(STRIGI_UINT64_TYPE& s);
+    DBusMessageReader& operator>>(STRIGI_INT64_TYPE& s);
     DBusMessageReader& operator>>(std::multimap<int, std::string>&);
     DBusMessageReader& operator>>(std::vector<std::pair<bool, std::string> >&);
     bool isOk() const { return ok; }
diff -ruN strigi-0.7.8.orig/strigidaemon/bin/daemon/xesam/xesamsearch.h strigi-0.7.8/strigidaemon/bin/daemon/xesam/xesamsearch.h
--- strigi-0.7.8.orig/strigidaemon/bin/daemon/xesam/xesamsearch.h	2013-01-25 11:49:35.000000000 -0600
+++ strigi-0.7.8/strigidaemon/bin/daemon/xesam/xesamsearch.h	2020-11-16 05:50:20.000000000 -0600
@@ -43,7 +43,7 @@
     XesamSearch(Private* p);
     ~XesamSearch();
     void operator=(const XesamSearch& xs);
-    bool operator==(const XesamSearch& xs) { return p == xs.p; }
+    bool operator==(const XesamSearch& xs) const { return p == xs.p; }
     void startSearch();
     void getHitCount(void* msg);
     void getHits(void* msg, uint32_t num);
