diff -ruN taglib-1.13.1-orig/bindings/c/CMakeLists.txt taglib-1.13.1/bindings/c/CMakeLists.txt
--- taglib-1.13.1-orig/bindings/c/CMakeLists.txt	2023-07-01 00:43:27.000000000 -0500
+++ taglib-1.13.1/bindings/c/CMakeLists.txt	2023-08-26 05:27:13.000000000 -0500
@@ -57,6 +57,7 @@
 set_target_properties(tag_c PROPERTIES
   VERSION 0.0.0
   SOVERSION 0
+  LINK_FLAGS "-Wl,-compatibility_version,1 -Wl,-current_version,1"
   DEFINE_SYMBOL MAKE_TAGLIB_C_LIB
   INSTALL_NAME_DIR ${CMAKE_INSTALL_FULL_LIBDIR}
 )
diff -ruN taglib-1.13.1-orig/taglib/CMakeLists.txt taglib-1.13.1/taglib/CMakeLists.txt
--- taglib-1.13.1-orig/taglib/CMakeLists.txt	2023-07-01 00:43:27.000000000 -0500
+++ taglib-1.13.1/taglib/CMakeLists.txt	2023-08-26 05:28:07.000000000 -0500
@@ -344,6 +344,7 @@
 set_target_properties(tag PROPERTIES
   VERSION ${TAGLIB_SOVERSION_MAJOR}.${TAGLIB_SOVERSION_MINOR}.${TAGLIB_SOVERSION_PATCH}
   SOVERSION ${TAGLIB_SOVERSION_MAJOR}
+  LINK_FLAGS "-Wl,-compatibility_version,${TAGLIB_SOVERSION_AGE} -Wl,-current_version,${TAGLIB_SOVERSION_AGE}"
   INSTALL_NAME_DIR ${CMAKE_INSTALL_FULL_LIBDIR}
   DEFINE_SYMBOL MAKE_TAGLIB_LIB
   LINK_INTERFACE_LIBRARIES ""
