diff -Nurd tclreadline-2.2.0.orig/Makefile.am tclreadline-2.2.0/Makefile.am
--- tclreadline-2.2.0.orig/Makefile.am	2016-12-20 18:02:41.000000000 -0500
+++ tclreadline-2.2.0/Makefile.am	2018-02-24 18:04:00.000000000 -0500
@@ -9,18 +9,19 @@
 ## ---
 
 ## AUTOMAKE_OPTIONS = foreign
-
+ACLOCAL_AMFLAGS = -I aux
 lib_LTLIBRARIES = libtclreadline.la
 
 libtclreadline_la_SOURCES = config.h tclreadline.c tclreadline.h
 include_HEADERS = tclreadline.h
 
-INCLUDES = -I$(TCL_INCLUDE_DIR) -I$(READLINE_INCLUDE_DIR)
+AM_CPPFLAGS = -I$(TCL_INCLUDE_DIR) -I$(READLINE_INCLUDE_DIR)
 
+libdir = @TCLRL_DIR@
 ## libtclreadline_la_LIBADD = $(LIBS)
 libtclreadline_la_CFLAGS = -Wall
 libtclreadline_la_LIBADD = $(TCL_LIB_SPEC)
-libtclreadline_la_LDFLAGS = -release $(MAJOR).$(MINOR).$(PATCHLEVEL)
+libtclreadline_la_LDFLAGS = -release $(MAJOR).$(MINOR).$(PATCHLEVEL) -no-undefined
 
 tclrldir = @TCLRL_DIR@
 tclrl_SCRIPTS = pkgIndex.tcl tclreadlineCompleter.tcl tclreadlineInit.tcl tclreadlineSetup.tcl
diff -Nurd tclreadline-2.2.0.orig/configure.ac tclreadline-2.2.0/configure.ac
--- tclreadline-2.2.0.orig/configure.ac	2016-12-20 18:02:41.000000000 -0500
+++ tclreadline-2.2.0/configure.ac	2018-02-25 00:10:21.000000000 -0500
@@ -13,6 +13,7 @@
 AC_PREREQ(2.13)
 AC_REVISION($Revision$)
 AC_CONFIG_AUX_DIR(./aux)
+AC_CONFIG_MACRO_DIRS([aux])
 
 MAJOR=2
 MINOR=2
@@ -42,7 +43,6 @@
 
 AC_C_CONST
 AC_PROG_INSTALL
-AC_PROG_RANLIB
 AC_PROG_LN_S
 
 
@@ -191,7 +191,11 @@
     [ Define the name of the executing macro variable in libreadline. ])
   , AC_MSG_RESULT(no))
 , [AC_INCLUDES_DEFAULT
-#include "$READLINE_INCLUDE_DIR/readline.h"])
+#if defined (READLINE_LIBRARY)
+#include "$READLINE_INCLUDE_DIR/readline.h"
+#else
+#include "$READLINE_INCLUDE_DIR/readline/readline.h"
+#endif])
 
 # check for readline's rl_cleanup_after_signal
 
diff -Nurd tclreadline-2.2.0.orig/tclreadlineInit.tcl.in tclreadline-2.2.0/tclreadlineInit.tcl.in
--- tclreadline-2.2.0.orig/tclreadlineInit.tcl.in	2016-12-20 18:02:41.000000000 -0500
+++ tclreadline-2.2.0/tclreadlineInit.tcl.in	2018-02-24 23:34:49.000000000 -0500
@@ -16,7 +16,7 @@
 proc ::tclreadline::Init {} {
     uplevel #0 {
         if {![info exists tclreadline::library]} {
-            if {[catch {load [file join @TCLRL_LIBDIR@ libtclreadline[info sharedlibextension]]} msg]} {
+            if {[catch {load [file join [file dirname [info script]] libtclreadline[info sharedlibextension]]} msg]} {
                 puts stderr $msg
                 exit 2
             }
