diff -Naur dia-0.96.1.orig/po/Makefile.in.in dia-0.96.1/po/Makefile.in.in
--- dia-0.96.1.orig/po/Makefile.in.in	2007-03-29 02:15:02.000000000 -0700
+++ dia-0.96.1/po/Makefile.in.in	2008-01-05 10:56:30.000000000 -0800
@@ -34,7 +34,7 @@
 datarootdir = @datarootdir@
 libdir = @libdir@
 DATADIRNAME = @DATADIRNAME@
-itlocaledir = $(prefix)/$(DATADIRNAME)/locale
+itlocaledir = $(prefix)/$(DATADIRNAME)/share/locale
 subdir = po
 install_sh = @install_sh@
 # Automake >= 1.8 provides @mkdir_p@.
