diff -Nurd ffmpeg-0.4.20071217.orig/Makefile ffmpeg-0.4.20071217/Makefile
--- ffmpeg-0.4.20071217.orig/Makefile	2008-10-27 23:40:37.000000000 +0100
+++ ffmpeg-0.4.20071217/Makefile	2008-10-28 00:00:21.000000000 +0100
@@ -6,9 +6,10 @@
 
 VPATH=$(SRC_PATH_BARE)
 
-CFLAGS=$(OPTFLAGS) -I$(BUILD_ROOT) -I$(SRC_PATH) -I$(SRC_PATH)/libavutil \
+CFLAGS= -I$(BUILD_ROOT) -I$(SRC_PATH) -I$(SRC_PATH)/libavutil \
        -I$(SRC_PATH)/libavcodec -I$(SRC_PATH)/libavformat -I$(SRC_PATH)/libswscale \
        -I$(SRC_PATH)/libavdevice \
+       $(OPTFLAGS) \
        -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_ISOC9X_SOURCE -DHAVE_AV_CONFIG_H
 LDFLAGS+= -g
 
@@ -61,12 +62,10 @@
 VHOOKCFLAGS += $(VHOOKCFLAGS-yes)
 
 SRCS = $(addsuffix .c, $(PROGS-yes)) cmdutils.c
-LDFLAGS := -L$(BUILD_ROOT)/libavdevice -L$(BUILD_ROOT)/libavformat -L$(BUILD_ROOT)/libavcodec -L$(BUILD_ROOT)/libavutil $(LDFLAGS)
-EXTRALIBS := -lavdevice$(BUILDSUF) -lavformat$(BUILDSUF) -lavcodec$(BUILDSUF) -lavutil$(BUILDSUF) $(EXTRALIBS)
+EXTRALIBS := $(BUILD_ROOT)/libavdevice/libavdevice$(BUILDSUF).dylib $(BUILD_ROOT)/libavformat/libavformat$(BUILDSUF).dylib $(BUILD_ROOT)/libavcodec/libavcodec$(BUILDSUF).dylib $(BUILD_ROOT)/libavutil/libavutil$(BUILDSUF).dylib $(EXTRALIBS)
 
 ifeq ($(CONFIG_SWSCALER),yes)
-LDFLAGS+=-L$(BUILD_ROOT)/libswscale
-EXTRALIBS+=-lswscale$(BUILDSUF)
+EXTRALIBS+=$(BUILD_ROOT)/libswscale/libswscale$(BUILDSUF).dylib
 endif
 
 all: lib $(PROGS) $(ALL_TARGETS-yes)
diff -Nurd ffmpeg-0.4.20071217.orig/libavcodec/Makefile ffmpeg-0.4.20071217/libavcodec/Makefile
--- ffmpeg-0.4.20071217.orig/libavcodec/Makefile	2008-10-27 23:40:35.000000000 +0100
+++ ffmpeg-0.4.20071217/libavcodec/Makefile	2008-10-28 00:03:10.000000000 +0100
@@ -343,9 +343,7 @@
 
 OBJS-$(HAVE_XVMC_ACCEL)                += xvmcvideo.o
 
-ifneq ($(CONFIG_SWSCALER),yes)
 OBJS += imgresample.o
-endif
 
 # processor-specific code
 ifeq ($(HAVE_MMX),yes)
@@ -442,7 +440,7 @@
                                           bfin/idct_bfin.o   \
                                           bfin/vp3_idct_bfin.o   \
 
-EXTRALIBS := -L$(BUILD_ROOT)/libavutil -lavutil$(BUILDSUF) $(EXTRALIBS)
+EXTRALIBS := $(BUILD_ROOT)/libavutil/libavutil$(BUILDSUF).dylib $(EXTRALIBS)
 
 NAME=avcodec
 LIBVERSION=$(LAVCVERSION)
diff -Nurd ffmpeg-0.4.20071217.orig/libavcodec/imgconvert.c ffmpeg-0.4.20071217/libavcodec/imgconvert.c
--- ffmpeg-0.4.20071217.orig/libavcodec/imgconvert.c	2008-10-27 23:40:35.000000000 +0100
+++ ffmpeg-0.4.20071217/libavcodec/imgconvert.c	2008-10-28 00:03:53.000000000 +0100
@@ -2324,7 +2324,6 @@
 }
 #endif
 
-#ifndef CONFIG_SWSCALER
 /* XXX: always use linesize. Return -1 if not supported */
 int img_convert(AVPicture *dst, int dst_pix_fmt,
                 const AVPicture *src, int src_pix_fmt,
@@ -2569,7 +2568,6 @@
     avpicture_free(tmp);
     return ret;
 }
-#endif
 
 /* NOTE: we scan all the pixels to have an exact information */
 static int get_alpha_info_pal8(const AVPicture *src, int width, int height)
diff -Nurd ffmpeg-0.4.20071217.orig/libavdevice/Makefile ffmpeg-0.4.20071217/libavdevice/Makefile
--- ffmpeg-0.4.20071217.orig/libavdevice/Makefile	2008-10-27 23:40:37.000000000 +0100
+++ ffmpeg-0.4.20071217/libavdevice/Makefile	2008-10-28 00:00:21.000000000 +0100
@@ -4,9 +4,9 @@
 LIBVERSION=$(LAVDVERSION)
 LIBMAJOR=$(LAVDMAJOR)
 
-EXTRALIBS := -L$(BUILD_ROOT)/libavutil   -lavutil$(BUILDSUF)  \
-             -L$(BUILD_ROOT)/libavcodec  -lavcodec$(BUILDSUF) \
-             -L$(BUILD_ROOT)/libavformat -lavformat$(BUILDSUF) $(EXTRALIBS)
+EXTRALIBS := $(BUILD_ROOT)/libavutil/libavutil$(BUILDSUF).dylib  \
+             $(BUILD_ROOT)/libavcodec/libavcodec$(BUILDSUF).dylib \
+             $(BUILD_ROOT)/libavformat/libavformat$(BUILDSUF).dylib $(EXTRALIBS)
 
 CFLAGS += -I$(SRC_PATH)/libavcodec -I$(SRC_PATH)/libavformat
 
diff -Nurd ffmpeg-0.4.20071217.orig/libavformat/Makefile ffmpeg-0.4.20071217/libavformat/Makefile
--- ffmpeg-0.4.20071217.orig/libavformat/Makefile	2008-10-27 23:40:29.000000000 +0100
+++ ffmpeg-0.4.20071217/libavformat/Makefile	2008-10-28 00:00:21.000000000 +0100
@@ -156,8 +156,8 @@
 
 OBJS-$(CONFIG_VHOOK)                     += framehook.o
 
-EXTRALIBS := -L$(BUILD_ROOT)/libavutil -lavutil$(BUILDSUF) \
-             -L$(BUILD_ROOT)/libavcodec -lavcodec$(BUILDSUF) $(EXTRALIBS)
+EXTRALIBS := $(BUILD_ROOT)/libavutil/libavutil$(BUILDSUF).dylib \
+             $(BUILD_ROOT)/libavcodec/libavcodec$(BUILDSUF).dylib $(EXTRALIBS)
 
 # protocols I/O
 OBJS+= avio.o aviobuf.o
diff -Nurd ffmpeg-0.4.20071217.orig/libpostproc/Makefile ffmpeg-0.4.20071217/libpostproc/Makefile
--- ffmpeg-0.4.20071217.orig/libpostproc/Makefile	2008-10-27 23:40:37.000000000 +0100
+++ ffmpeg-0.4.20071217/libpostproc/Makefile	2008-10-28 00:00:21.000000000 +0100
@@ -4,7 +4,7 @@
 # Overload incdir, postproc include files go in a different directory.
 INCDIR=$(prefix)/include/postproc
 
-EXTRALIBS := -L$(BUILD_ROOT)/libavutil -lavutil$(BUILDSUF) $(EXTRALIBS)
+EXTRALIBS := $(BUILD_ROOT)/libavutil/libavutil$(BUILDSUF).dylib $(EXTRALIBS)
 
 NAME=postproc
 LIBVERSION=$(SPPVERSION)
diff -Nurd ffmpeg-0.4.20071217.orig/libswscale/Makefile ffmpeg-0.4.20071217/libswscale/Makefile
--- ffmpeg-0.4.20071217.orig/libswscale/Makefile	2008-10-27 23:40:29.000000000 +0100
+++ ffmpeg-0.4.20071217/libswscale/Makefile	2008-10-28 00:00:21.000000000 +0100
@@ -5,7 +5,7 @@
 LIBVERSION=$(SWSVERSION)
 LIBMAJOR=$(SWSMAJOR)
 
-EXTRALIBS := -L$(BUILD_ROOT)/libavutil -lavutil$(BUILDSUF) $(EXTRALIBS)
+EXTRALIBS := $(BUILD_ROOT)/libavutil/libavutil$(BUILDSUF).dylib $(EXTRALIBS)
 
 OBJS = rgb2rgb.o swscale.o
 

 	  	 
