diff -ruN figlet222/Makefile figlet222-patched/Makefile
--- figlet222/Makefile	2005-07-05 05:57:46.000000000 -0400
+++ figlet222-patched/Makefile	2008-08-17 20:46:53.000000000 -0400
@@ -11,19 +11,25 @@
 # Don't change this even if your shell is different. The only reason
 # for changing this is if sh is not in the same place.
 SHELL = /bin/sh
-CC = gcc
+CC = cc
 CFLAGS = -g
 
+prefix = @PREFIX@
+
+INSTALL_PROGRAM = install -m 755
+INSTALL_DATA = install -m 644
+
+
 # Where the executables should be put
-DESTDIR = /usr/local/bin
+BINDIR = $(prefix)/bin
 
 # Where the man page should be put
-MANDIR = /usr/local/man/man6
+MANDIR = $(prefix)/share/man/man6
 
 # Where figlet will search first for fonts (the ".flf" files).
-DEFAULTFONTDIR = /usr/local/share/figlet
+DEFAULTFONTDIR = $(prefix)/share/figlet
 # Use this definition if you can't put things in /usr/local/share/figlet
-DEFAULTFONTDIR = fonts
+# DEFAULTFONTDIR = fonts
 
 # The filename of the font to be used if no other is specified
 #   (standard.flf is recommended, but any other can be used).
@@ -53,10 +59,13 @@
 
 install: figlet chkfont
 	mkdir -p $(DEFAULTFONTDIR)
-	cp figlet $(DESTDIR)
-	cp figlet.6 $(MANDIR)
-	cp chkfont $(DESTDIR)
-	cp figlist $(DESTDIR)
-	cp showfigfonts $(DESTDIR)
-	cp fonts/*.flf $(DEFAULTFONTDIR)
-	cp fonts/*.flc $(DEFAULTFONTDIR)
+	mkdir -p $(BINDIR)
+	mkdir -p $(DEFAULTFONTDIR)
+	mkdir -p $(MANDIR)
+	$(INSTALL_PROGRAM) figlet $(BINDIR)
+	$(INSTALL_PROGRAM) chkfont $(BINDIR)
+	$(INSTALL_PROGRAM) figlist $(BINDIR)
+	$(INSTALL_PROGRAM) showfigfonts $(BINDIR)
+	$(INSTALL_DATA) fonts/*.flf $(DEFAULTFONTDIR)
+	$(INSTALL_DATA) fonts/*.flc $(DEFAULTFONTDIR)
+	$(INSTALL_DATA) figlet.6 $(MANDIR)
