diff -ru fontforge-20070831.orig/Makefile.dynamic.in fontforge-20070831/Makefile.dynamic.in
--- fontforge-20070831.orig/Makefile.dynamic.in	2007-08-17 22:06:04.000000000 -0400
+++ fontforge-20070831/Makefile.dynamic.in	2007-09-01 17:49:07.000000000 -0400
@@ -94,7 +94,7 @@
 	$(INSTALL_DATA) $(srcdir)/fontforge/fontlint.1 $(DESTDIR)$(mandir)/man1
 #explicit use of sh to make sure we don't invoke csh with different "if" syntax
 	sh -c "if test -d $(srcdir)/htdocs >/dev/null ; then $(MAKE) install_docs ; fi"
-	sh < systemspecific
+#	sh < systemspecific
 
 uninstall:
 	-rm $(DESTDIR)$(bindir)/fontforge
diff -ru fontforge-20070831.orig/configure fontforge-20070831/configure
--- fontforge-20070831.orig/configure	2007-08-30 18:13:19.000000000 -0400
+++ fontforge-20070831/configure	2007-09-01 17:49:07.000000000 -0400
@@ -2686,9 +2686,7 @@
  gww_define_caps_for_alt="1"
 
 
- CPPFLAGS="$CPPFLAGS -I/sw/include"
- LDFLAGS="$LDFLAGS -L/sw/lib"
- gww_rpath="-rpath /sw/lib"
+ gww_rpath=""
  LIBS="-Wl,/System/Library/Frameworks/CoreServices.framework/CoreServices $LIBS"
 
  ;;
@@ -19100,7 +19098,7 @@
 echo "$as_me:$LINENO: result: $ac_cv_file__usr_include_libxml2_libxml" >&5
 echo "${ECHO_T}$ac_cv_file__usr_include_libxml2_libxml" >&6
 if test $ac_cv_file__usr_include_libxml2_libxml = yes; then
-  CFLAGS="$CFLAGS -I/usr/include/libxml2/"
+  true
 else
 
  echo "$as_me:$LINENO: checking for /usr/local/include/libxml2/libxml" >&5
@@ -19121,7 +19119,7 @@
 echo "$as_me:$LINENO: result: $ac_cv_file__usr_local_include_libxml2_libxml" >&5
 echo "${ECHO_T}$ac_cv_file__usr_local_include_libxml2_libxml" >&6
 if test $ac_cv_file__usr_local_include_libxml2_libxml = yes; then
-  CFLAGS="$CFLAGS -I/usr/local/include -I/usr/local/include/libxml2/"
+  true
 else
   cat >>confdefs.h <<\_ACEOF
 #define _NO_LIBXML 1
diff -ru fontforge-20070831.orig/fontforge/Makefile.dynamic.in fontforge-20070831/fontforge/Makefile.dynamic.in
--- fontforge-20070831.orig/fontforge/Makefile.dynamic.in	2007-08-27 22:18:49.000000000 -0400
+++ fontforge-20070831/fontforge/Makefile.dynamic.in	2007-09-01 17:49:07.000000000 -0400
@@ -53,9 +53,9 @@
 
 #I used to include -L/usr/lib explicitly after static_libs. Not sure why. Should be implied & breaks 64bit compiles
 LDFLAGS = @LDFLAGS@
-LIBS = -rpath $(libdir) $(X_LIBS) ../libgdraw.la ../libgunicode.la \
-    $(X_PRE_LIBS) $(X_11LIB) $(X_EXTRA_LIBS) @LIBS@ $(STATIC_LIBS) -lm
-DLIBS = -rpath $(libdir) ../libgunicode.la @LIBS@ $(STATIC_LIBS) -lm
+LIBS = -rpath $(libdir) $(STATIC_LIBS) $(X_LIBS) ../libgdraw.la ../libgunicode.la \
+    $(X_PRE_LIBS) $(X_11LIB) $(X_EXTRA_LIBS) @LIBS@ -lm
+DLIBS = -rpath $(libdir) -L$(libdir) -liconv ../libgunicode.la @LIBS@ $(STATIC_LIBS) -lm
 
 all: fontforge sfddiff fontimage fontlint
 
diff -ru fontforge-20070831.orig/fontforge/Makefile.in fontforge-20070831/fontforge/Makefile.in
--- fontforge-20070831.orig/fontforge/Makefile.in	2007-08-27 22:18:49.000000000 -0400
+++ fontforge-20070831/fontforge/Makefile.in	2007-09-01 17:49:07.000000000 -0400
@@ -87,10 +87,10 @@
 # disasterous effects. This file may not be compiled with the optimizer.
 # On the other hand we do need CPPFLAGS to get the right includes...
 splinerefigure.o: $(srcdir)/splinerefigure.c $(srcdir)/splinefont.h
-	$(CC) -g -c -o splinerefigure.o $(_CFLAGS) $(srcdir)/splinerefigure.c
+	$(CC) -g -c -o splinerefigure.o $(CFLAGS) $(srcdir)/splinerefigure.c
 
 splinerefigure.lo: $(srcdir)/splinerefigure.c $(srcdir)/splinefont.h
-	$(LIBTOOL) --mode=compile $(CC) -g -c -o splinerefigure.lo $(_CFLAGS) $(srcdir)/splinerefigure.c
+	$(LIBTOOL) --mode=compile $(CC) -g -c -o splinerefigure.lo $(CFLAGS) $(srcdir)/splinerefigure.c
 
 %.o: %.c
 
diff -ru fontforge-20070831.orig/fontforge/start.c fontforge-20070831/fontforge/start.c
--- fontforge-20070831.orig/fontforge/start.c	2007-08-17 15:27:06.000000000 -0400
+++ fontforge-20070831/fontforge/start.c	2007-09-01 17:49:07.000000000 -0400
@@ -361,7 +361,6 @@
     /*  install image libs. For some reason fink installs in a place */
     /*  the dynamic loader doesn't find */
     /* (And fink's attempts to set the PATH variables generally don't work */
-    setenv("DYLD_LIBRARY_PATH","/sw/lib",0);
 #endif
 }
 
