diff -ru fontforge-20100429.orig/Makefile.dynamic.in fontforge-20100429/Makefile.dynamic.in
--- fontforge-20100429.orig/Makefile.dynamic.in	2010-02-04 20:16:29.000000000 -0500
+++ fontforge-20100429/Makefile.dynamic.in	2010-04-30 17:12:12.000000000 -0400
@@ -127,8 +127,8 @@
 # You may need to be root to do the install
 #explicit use of sh to make sure we don't invoke csh with different "if" syntax
 	sh -c "if test -d $(srcdir)/htdocs >/dev/null ; then $(MAKE) install_docs ; fi"
-	sh -c "if test -d fontforge/FontForge.app >/dev/null ; then rm -rf /Applications/FontForge.app ; mv fontforge/FontForge.app /Applications ; ln -s /usr/local/bin/fontforge /Applications/FontForge.app/Contents/MacOS/FontForge ; fi"
-	sh < systemspecific
+#	sh -c "if test -d fontforge/FontForge.app >/dev/null ; then rm -rf /Applications/FontForge.app ; mv fontforge/FontForge.app /Applications ; ln -s /usr/local/bin/fontforge /Applications/FontForge.app/Contents/MacOS/FontForge ; fi"
+#	sh < systemspecific
 #we don't always create a setup.py
 	-(cd pyhook ; if test -r setup.py ; then python setup.py install --prefix=$(prefix) --root=$(DESTDIR); for dir in * ; do if test -d $$dir -a "$$dir" != "CVS" ; then  rm -rf $$dir ; fi; done ; fi )
 
diff -ru fontforge-20100429.orig/Unicode/Makefile.dynamic.in fontforge-20100429/Unicode/Makefile.dynamic.in
--- fontforge-20100429.orig/Unicode/Makefile.dynamic.in	2010-04-14 16:49:48.000000000 -0400
+++ fontforge-20100429/Unicode/Makefile.dynamic.in	2010-04-30 17:12:12.000000000 -0400
@@ -18,8 +18,8 @@
  cjk.lo memory.lo ucharmap.lo unialt.lo ustring.lo utype.lo \
  usprintf.lo gwwiconv.lo
 
-Incs = -I$(top_srcdir)/inc -I../inc -I/usr/pkg/include -I/usr/pkg/include/giflib
-CFLAGS = @CFLAGS@ @CPPFLAGS@ $(Incs) @WFLAGS@ @DEFS@
+Incs = -I$(top_srcdir)/inc -I../inc
+CFLAGS = $(Incs) @CFLAGS@ @CPPFLAGS@ @WFLAGS@ @DEFS@
 
 ../libgunicode.la: $(libgunicode_OBJECTS)
 	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) -no-undefined @LDFLAGS@ -o ../libgunicode.la $(libgunicode_OBJECTS) \
diff -ru fontforge-20100429.orig/configure fontforge-20100429/configure
--- fontforge-20100429.orig/configure	2010-04-29 14:29:04.000000000 -0400
+++ fontforge-20100429/configure	2010-04-30 17:12:12.000000000 -0400
@@ -4633,7 +4633,7 @@
 oldCPPFLAGS="$CPPFLAGS"
 gww_extraincludes=""
 oldLDFLAGS="$LDFLAGS"
-LDFLAGS="$LDFLAGS -L/usr/local/lib"
+#LDFLAGS="$LDFLAGS -L/usr/local/lib"
 
 case "$host" in
 *-apple-darwin*)
diff -ru fontforge-20100429.orig/fontforge/Makefile.dynamic.in fontforge-20100429/fontforge/Makefile.dynamic.in
--- fontforge-20100429.orig/fontforge/Makefile.dynamic.in	2010-02-01 23:30:20.000000000 -0500
+++ fontforge-20100429/fontforge/Makefile.dynamic.in	2010-04-30 17:12:12.000000000 -0400
@@ -67,12 +67,12 @@
 
 #I used to include -L/usr/lib explicitly after static_libs. Not sure why. Should be implied & breaks 64bit compiles
 LDFLAGS = $(X_LIBS) @LDFLAGS@
-LIBS = -rpath $(libdir) @LINK_WITH_GDRAW@ ../libgutils.la \
-    ../libgunicode.la \
-    $(X_PRE_LIBS) $(X_11LIB) $(X_EXTRA_LIBS) @LIBS@ $(STATIC_LIBS) -lm
-LIBLIBS = -rpath $(libdir) ../libgutils.la ../libgunicode.la \
-    @LIBS@ $(STATIC_LIBS) -lm
-DLIBS = -rpath $(libdir)  ../libfontforge.la ../libgutils.la ../libgunicode.la @LIBS@ $(STATIC_LIBS) -lm
+LIBS = -rpath $(libdir) $(STATIC_LIBS) @LINK_WITH_GDRAW@ ../libgunicode.la \
+    ../libgutils.la \
+    $(X_PRE_LIBS) $(X_11LIB) $(X_EXTRA_LIBS) @LIBS@ -lm
+LIBLIBS = -rpath $(libdir) $(STATIC_LIBS) ../libgunicode.la ../libgutils.la \
+    @LIBS@ -lm
+DLIBS = -rpath $(libdir) $(STATIC_LIBS) ../libgunicode.la ../libgutils.la ../libfontforge.la @LIBS@ -lm
 
 all: @FINAL_TARGET@ @MACAPP@ sfddiff fontimage fontlint
 
diff -ru fontforge-20100429.orig/fontforge/start.c fontforge-20100429/fontforge/start.c
--- fontforge-20100429.orig/fontforge/start.c	2010-02-04 10:33:53.000000000 -0500
+++ fontforge-20100429/fontforge/start.c	2010-04-30 17:12:12.000000000 -0400
@@ -125,7 +125,6 @@
     /*  install image libs. For some reason fink installs in a place */
     /*  the dynamic loader doesn't find */
     /* (And fink's attempts to set the PATH variables generally don't work */
-    setenv("DYLD_LIBRARY_PATH","/sw/lib",0);
 #endif
 }
 
diff -ru fontforge-20100429.orig/gdraw/Makefile.dynamic.in fontforge-20100429/gdraw/Makefile.dynamic.in
--- fontforge-20100429.orig/gdraw/Makefile.dynamic.in	2010-04-15 11:54:05.000000000 -0400
+++ fontforge-20100429/gdraw/Makefile.dynamic.in	2010-04-30 17:12:12.000000000 -0400
@@ -34,12 +34,12 @@
  gtextfield.lo gtextinfo.lo gwidgets.lo gxdraw.lo gxcdraw.lo \
  ghvbox.lo gmatrixedit.lo gdrawable.lo gspacer.lo xkeysyms_unicode.lo
 
-Incs = -I$(top_srcdir)/inc -I../inc -I/usr/pkg/include -I/usr/pkg/include/giflib
-CFLAGS = @CFLAGS@ @CPPFLAGS@ $(Incs) @WFLAGS@ $(X_CFLAGS) -DNOTHREADS @DEFS@ -DLIBDIR='"$(libdir)"'
+Incs = -I$(top_srcdir)/inc -I../inc
+CFLAGS = $(Incs) @CFLAGS@ @CPPFLAGS@ @WFLAGS@ $(X_CFLAGS) -DNOTHREADS @DEFS@ -DLIBDIR='"$(libdir)"'
 
 ../libgdraw.la: $(libgdraw_OBJECTS)
 	$(LIBTOOL) --mode=link $(CC) -no-undefined $(CFLAGS) -o ../libgdraw.la $(libgdraw_OBJECTS) \
-	    ../libgutils.la ../libgunicode.la $(X_LIBS) @LDFLAGS@ @LIBS@ \
+	    ../libgunicode.la ../libgutils.la $(X_LIBS) @LDFLAGS@ @LIBS@ \
 	    $(STATIC_LIBS) $(X_PRE_LIBS) $(X_11LIB) $(X_EXTRA_LIBS)  \
 	    -rpath $(libdir) -version-info $(GD_VERSION):$(GD_REVISION):$(GD_AGE)
 
diff -ru fontforge-20100429.orig/gutils/Makefile.dynamic.in fontforge-20100429/gutils/Makefile.dynamic.in
--- fontforge-20100429.orig/gutils/Makefile.dynamic.in	2009-09-07 21:50:16.000000000 -0400
+++ fontforge-20100429/gutils/Makefile.dynamic.in	2010-04-30 17:12:12.000000000 -0400
@@ -30,21 +30,19 @@
  gio.lo giofile.lo giohosts.lo giomime.lo giothread.lo giotrans.lo
 libgioftp_OBJECTS = gioftp.lo
 
-Incs = -I$(top_srcdir)/inc -I../inc -I/usr/pkg/include -I/usr/pkg/include/giflib
-CFLAGS = @CFLAGS@ @CPPFLAGS@ $(Incs) @WFLAGS@ $(X_CFLAGS) -DNOTHREADS @DEFS@ -DLIBDIR='"$(libdir)"'
+Incs = -I$(top_srcdir)/inc -I../inc
+CFLAGS = $(Incs) @CFLAGS@ @CPPFLAGS@ @WFLAGS@ $(X_CFLAGS) -DNOTHREADS @DEFS@ -DLIBDIR='"$(libdir)"'
 
 both: ../libgutils.la ../libgioftp.la
 
 ../libgutils.la: $(libgutils_OBJECTS)
-	$(LIBTOOL) --mode=link $(CC) -no-undefined $(CFLAGS) @LDFLAGS@ -o ../libgutils.la $(libgutils_OBJECTS) \
+	$(LIBTOOL) --mode=link $(CC) -no-undefined $(CFLAGS) $(STATIC_LIBS) @LDFLAGS@ -o ../libgutils.la $(libgutils_OBJECTS) \
 	    ../libgunicode.la @LIBS@ \
-	    $(STATIC_LIBS)  \
 	    -rpath $(libdir) -version-info $(GU_VERSION):$(GU_REVISION):$(GU_AGE)
 
 ../libgioftp.la: $(libgioftp_OBJECTS)
-	$(LIBTOOL) --mode=link $(CC) -no-undefined $(CFLAGS) @LDFLAGS@ -o ../libgioftp.la $(libgioftp_OBJECTS) \
+	$(LIBTOOL) --mode=link $(CC) -no-undefined $(CFLAGS) $(STATIC_LIBS) @LDFLAGS@ -o ../libgioftp.la $(libgioftp_OBJECTS) \
 	    ../libgunicode.la @LIBS@ \
-	    $(STATIC_LIBS)  \
 	    -rpath $(libdir) -version-info $(GFTP_VERSION):$(GFTP_REVISION):$(GFTP_AGE)
 
 %.lo: %.c
