--- freetype-2.3.1/include/freetype/config/ftoption.h	2007-01-05 11:18:45.000000000 -0500
+++ freetype-2.3.1-new/include/freetype/config/ftoption.h	2007-03-02 11:31:00.000000000 -0500
@@ -468,7 +468,7 @@
   /* FT_PARAM_TAG_UNPATENTED_HINTING; or when the debug hook               */
   /* FT_DEBUG_HOOK_UNPATENTED_HINTING is globally activated.               */
   /*                                                                       */
-#define TT_CONFIG_OPTION_UNPATENTED_HINTING
+/* #define TT_CONFIG_OPTION_UNPATENTED_HINTING */
 
 
   /*************************************************************************/
--- freetype-2.3.1/include/freetype/freetype.h	2007-01-30 18:07:53.000000000 -0500
+++ freetype-2.3.1-new/include/freetype/freetype.h	2007-03-02 11:31:00.000000000 -0500
@@ -17,11 +17,7 @@
 
 
 #ifndef FT_FREETYPE_H
-#error "`ft2build.h' hasn't been included yet!"
-#error "Please always use macros to include FreeType header files."
-#error "Example:"
-#error "  #include <ft2build.h>"
-#error "  #include FT_FREETYPE_H"
+#include <ft2build.h>
 #endif
 
 
--- freetype-2.3.1/modules.cfg	2007-01-31 00:51:06.000000000 -0500
+++ freetype-2.3.1-new/modules.cfg	2007-03-02 11:31:02.000000000 -0500
@@ -124,7 +124,7 @@
 
 # OpenType table validation.  Needs ftotval.c below.
 #
-# AUX_MODULES += otvalid
+AUX_MODULES += otvalid
 
 # Auxiliary PostScript driver component to share common code.
 #
--- freetype-2.3.1/src/base/ftapi.c	2007-01-16 00:10:55.000000000 -0500
+++ freetype-2.3.1-new/src/base/ftapi.c	2007-03-02 11:31:00.000000000 -0500
@@ -117,5 +117,17 @@
     FT_Stream_ExitFrame( stream );
   }
 
+  FT_BASE_DEF( FT_Short )
+  FT_Get_Short( FT_Stream stream )
+  {
+    return FT_GET_SHORT();
+  }
+
+
+  FT_BASE_DEF( FT_Long )
+  FT_Get_Long( FT_Stream stream )
+  {
+    return FT_GET_LONG();
+  }
 
 /* END */
--- freetype-2.3.1/src/base/ftbase.c	2006-10-01 12:57:58.000000000 -0400
+++ freetype-2.3.1-new/src/base/ftbase.c	2007-03-02 11:31:00.000000000 -0500
@@ -30,6 +30,7 @@
 #include "ftstream.c"
 #include "fttrigon.c"
 #include "ftutil.c"
+#include "ftapi.c"
 
 #if defined( __APPLE__ ) && !defined ( DARWIN_NO_CARBON )
 #include "ftmac.c"
--- freetype-2.3.1/src/base/ftrfork.c	2006-05-20 00:57:52.000000000 -0400
+++ freetype-2.3.1-new/src/base/ftrfork.c	2007-03-02 11:31:02.000000000 -0500
@@ -440,7 +440,7 @@
       return error;
 
     FT_MEM_COPY( newpath, base_file_name, base_file_len );
-    FT_MEM_COPY( newpath + base_file_len, "/rsrc", 6 );
+    FT_MEM_COPY( newpath + base_file_len, "/..namedfork/rsrc", 18 );
 
     *result_file_name = newpath;
     *result_offset    = 0;
--- freetype-2.3.1/src/bdf/bdflib.c	2006-08-27 08:20:02.000000000 -0400
+++ freetype-2.3.1-new/src/bdf/bdflib.c	2007-03-02 11:31:02.000000000 -0500
@@ -1570,6 +1570,14 @@
         goto Exit;
       }
 
+      /* Check that the encoding is in the range [0, 65535] because       */
+      /* otherwise p->have (a bitmap with static size) overflows.         */
+      if ( p->glyph_enc >= sizeof(p->have) * 8 )
+      {
+        error = BDF_Err_Invalid_File_Format; /* Not the ideal error code */
+        goto Exit;
+      }
+
       /* Check to see whether this encoding has already been encountered. */
       /* If it has then change it to unencoded so it gets added if        */
       /* indicated.                                                       */
--- freetype-2.3.1/src/sfnt/ttmtx.c	2007-01-21 03:27:28.000000000 -0500
+++ freetype-2.3.1-new/src/sfnt/ttmtx.c	2007-03-02 11:31:02.000000000 -0500
@@ -166,7 +166,7 @@
     }
 
     if ( FT_QNEW_ARRAY( *longs,  num_longs  ) ||
-         FT_QNEW_ARRAY( *shorts, num_shorts ) )
+         (num_shorts > 0 && FT_QNEW_ARRAY( *shorts, num_shorts ) ) )
       goto Fail;
 
     if ( FT_FRAME_ENTER( table_len ) )
