diff -Nurd glew-1.6.0/config/Makefile.darwin glew-1.6.0-new/config/Makefile.darwin
--- glew-1.6.0/config/Makefile.darwin	2011-04-27 12:16:34.000000000 -0400
+++ glew-1.6.0-new/config/Makefile.darwin	2011-08-03 10:09:20.000000000 -0400
@@ -1,7 +1,7 @@
 NAME = $(GLEW_NAME)
 CC = cc
 LD = cc
-CFLAGS.EXTRA = -no-cpp-precomp -dynamic -fno-common
+CFLAGS.EXTRA = -dynamic -fno-common
 PICFLAG = -fPIC
 LDFLAGS.EXTRA =
 ifneq (undefined, $(origin GLEW_APPLE_GLX))
@@ -15,14 +15,14 @@
 WARN = -Wall -W
 POPT = -O2
 BIN.SUFFIX =
-LIB.SONAME    = lib$(NAME).$(SO_MAJOR).dylib
+LIB.SONAME    = lib$(NAME).$(GLEW_MAJOR).dylib
 LIB.DEVLNK    = lib$(NAME).dylib
 LIB.SHARED    = lib$(NAME).$(SO_VERSION).dylib
 LIB.STATIC    = lib$(NAME).a
-LDFLAGS.SO    = -dynamiclib -install_name $(GLEW_DEST)/lib/$(LIB.SHARED)
-LIB.SONAME.MX = lib$(NAME)mx.$(SO_MAJOR).dylib
+LDFLAGS.SO    = -dynamiclib -install_name $(GLEW_DEST)/lib/$(LIB.SONAME) -current_version $(SO_VERSION) -compatibility_version $(SO_MAJOR)
+LIB.SONAME.MX = lib$(NAME)mx.$(GLEW_MAJOR).dylib
 LIB.DEVLNK.MX = lib$(NAME)mx.dylib
 LIB.SHARED.MX = lib$(NAME)mx.$(SO_VERSION).dylib
 LIB.STATIC.MX = lib$(NAME)mx.a
-LDFLAGS.SO.MX = -dynamiclib -install_name $(GLEW_DEST)/lib/$(LIB.SHARED.MX)
+LDFLAGS.SO.MX = -dynamiclib -install_name $(GLEW_DEST)/lib/$(LIB.SONAME.MX) -current_version $(SO_VERSION) -compatibility_version $(SO_MAJOR)
 SHARED_OBJ_EXT = pic_o
