--- graphviz-2.18-orig/configure	2008-03-25 16:13:30.000000000 -0400
+++ graphviz-2.18/configure	2008-03-25 16:16:32.000000000 -0400
@@ -28337,6 +28337,8 @@
 
 if test "x$RUBY" != "x"; then
   RUBY_INCLUDES="-I`$RUBY $srcdir/config/config_ruby.rb INCLUDES`"
+  # hack for powerpc-darwin8 (10.4)
+  RUBY_INCLUDES=`echo $RUBY_INCLUDES | sed 's/powerpc/universal/'`
   RUBY_LIBS="-L`$RUBY $srcdir/config/config_ruby.rb lib` `$RUBY -rrbconfig -e \"puts Config::CONFIG['LIBRUBYARG_SHARED']\"`"
   RUBY_INSTALL_DIR="`$RUBY $srcdir/config/config_ruby.rb archsitelib`"
   save_CPPFLAGS=$CPPFLAGS
--- graphviz-2.18-orig/tclpkg/gv/Makefile.in	2007-12-06 03:54:16.000000000 -0500
+++ graphviz-2.18/tclpkg/gv/Makefile.in	2007-12-06 03:57:16.000000000 -0500
@@ -1934,6 +1934,9 @@
 $(LUA_data): gv_lua.cpp
+# this change could go in Makefile.am
 gv_lua.cpp: gv.i
 	$(SWIG) -c++ -lua -o gv_lua.cpp $(srcdir)/gv.i
+	mv $@ $@.orig
+	sed '/include/s/malloc\.h/cstdlib/' $@.orig > $@
 gv.ml gv.mli: gv_ocaml.cpp
 gv_ocaml.cpp: gv.i
 	$(SWIG) -c++ -ocaml -o gv_ocaml.cpp $(srcdir)/gv.i
