diff -ru jbig2dec-0.9.orig/Makefile.am jbig2dec-0.9/Makefile.am
--- jbig2dec-0.9.orig/Makefile.am	2005-05-23 23:49:47.000000000 -0400
+++ jbig2dec-0.9/Makefile.am	2009-03-15 17:05:33.000000000 -0400
@@ -4,10 +4,10 @@
 # require automake 1.7
 AUTOMAKE_OPTIONS = foreign 1.7 dist-bzip2 dist-zip -Wall
 
-lib_LIBRARIES = libjbig2dec.a
+lib_LTLIBRARIES = libjbig2dec.la
 include_HEADERS = jbig2.h
 
-libjbig2dec_a_SOURCES = jbig2.c \
+libjbig2dec_la_SOURCES = jbig2.c \
 	jbig2_arith.c jbig2_arith_int.c jbig2_arith_iaid.c jbig2_huffman.c \
 	jbig2_segment.c jbig2_page.c \
 	jbig2_symbol_dict.c jbig2_text.c \
@@ -26,7 +26,7 @@
 jbig2dec_SOURCES = jbig2dec.c sha1.c sha1.h \
 	jbig2.h jbig2_image.h getopt.h \
 	os_types.h config_types.h config_win32.h
-jbig2dec_LDADD = libjbig2dec.a @LIBOBJS@
+jbig2dec_LDADD = libjbig2dec.la @LIBOBJS@
 
 EXTRA_DIST = test_jbig2dec.py msvc.mak LICENSE CHANGES
 
@@ -39,9 +39,9 @@
 
 test_arith_SOURCES = jbig2_arith.c
 test_arith_CFLAGS = -DTEST
-test_arith_LDADD = libjbig2dec.a
+test_arith_LDADD = libjbig2dec.la
 
 test_huffman_SOURCES = jbig2_huffman.c
 test_huffman_CFLAGS = -DTEST
-test_huffman_LDADD = libjbig2dec.a
+test_huffman_LDADD = libjbig2dec.la
 
diff -ru jbig2dec-0.9.orig/configure.ac jbig2dec-0.9/configure.ac
--- jbig2dec-0.9.orig/configure.ac	2006-07-26 19:57:39.000000000 -0400
+++ jbig2dec-0.9/configure.ac	2009-03-15 17:07:45.000000000 -0400
@@ -10,8 +10,8 @@
 dnl AC_REVISION([$Rev: 447 $])
 
 # Checks for programs.
-AC_PROG_CC
-AC_PROG_RANLIB
+AM_PROG_CC_C_O
+AC_PROG_LIBTOOL
 
 # platform specific compiler flags
 if test "x$GCC" = xyes; then
