diff -Nurd babl-0.0.20.orig/babl/Makefile.am babl-0.0.20/babl/Makefile.am
--- babl-0.0.20.orig/babl/Makefile.am	2008-01-17 16:47:18.000000000 -0500
+++ babl-0.0.20/babl/Makefile.am	2008-03-02 23:27:10.000000000 -0500
@@ -54,5 +54,5 @@
 	@LTLIBOBJS@
 
 libbabl_@BABL_API_VERSION@_la_LDFLAGS= 		\
-	${no_undefined} ${DL_LIB} $(MATH_LIB)	\
+	${no_undefined} $(MATH_LIB)	\
 	-version-info $(BABL_LIBRARY_VERSION)
diff -Nurd babl-0.0.20.orig/babl/babl-extension.c babl-0.0.20/babl/babl-extension.c
--- babl-0.0.20.orig/babl/babl-extension.c	2008-02-15 16:11:44.000000000 -0500
+++ babl-0.0.20/babl/babl-extension.c	2008-03-02 22:54:05.000000000 -0500
@@ -235,7 +235,7 @@
               stat (path, &st);
 
               if ((extension = strrchr (dentry->d_name, '.')) != NULL &&
-                  !strcmp (extension, SHREXT))
+                  !strcmp (extension, LTDL_SHLIB_EXT))
                 {
                   babl_extension_load (path);
                 }
diff -Nurd babl-0.0.20.orig/babl.pc.in babl-0.0.20/babl.pc.in
--- babl-0.0.20.orig/babl.pc.in	2007-11-10 08:48:08.000000000 -0500
+++ babl-0.0.20/babl.pc.in	2008-03-03 00:04:18.000000000 -0500
@@ -7,5 +7,5 @@
 Description: Dynamic, any to any, pixel format conversion library
 Version: @BABL_REAL_VERSION@
 Cflags: -I${includedir}/@PACKAGE_NAME@-@BABL_API_VERSION@
-Libs: -L${libdir} -l@PACKAGE_NAME@-@BABL_API_VERSION@ @DL_LIB@ @MATH_LIB@
+Libs: -L${libdir} -l@PACKAGE_NAME@-@BABL_API_VERSION@ @MATH_LIB@
 
diff -Nurd babl-0.0.20.orig/configure.ac babl-0.0.20/configure.ac
--- babl-0.0.20.orig/configure.ac	2008-02-27 06:15:22.000000000 -0500
+++ babl-0.0.20/configure.ac	2008-03-02 22:48:43.000000000 -0500
@@ -158,30 +158,9 @@
 ####################################################
 # Check how to generate plug-ins (with gcc at least)
 ####################################################
+# sets loadable-module extension in LTDL_SHLIB_EXT token via AC_DEFINE
+AC_LTDL_SHLIBEXT
 
-AC_MSG_CHECKING([for some Win32 platform])
-case "$target_or_host" in 
-  *-*-darwin*)                 # darwin
-    shrext=.dylib
-    dynamiclib=-dynamiclib
-    ;;
-  *-*-mingw* | *-*-cygwin*)    # windows
-    shrext=.dll
-    dynamiclib=-shared
-    ;;
-  *)                           # linux (and BSD?)
-    dynamiclib=-shared
-    shrext=.so
-    ;;
-esac
-
-SHREXT=$shrext
-AC_SUBST(SHREXT)
-AC_DEFINE_UNQUOTED(SHREXT,     "$shrext",     [File extension for shared libraries])
-
-DYNAMICLIB=$dynamiclib
-AC_SUBST(DYNAMICLIB)
-AC_DEFINE_UNQUOTED(DYNAMICLIB, "$dynamiclib", [Dynamic shared library])
 
 dnl ===========================================================================
 
@@ -226,8 +205,7 @@
 
 dnl ===========================================================================
 
-AC_CHECK_LIB([dl], [dlopen], [DL_LIB="-ldl"])
-AC_SUBST(DL_LIB)
+AC_SEARCH_LIBS([dlopen], [dl])
 
 AC_REPLACE_FUNCS(gettimeofday)
 
@@ -237,7 +215,7 @@
 AC_DEFINE_UNQUOTED(BABL_DIR_SEPARATOR, "$DIR_SEP", [separator between directories in BABL_PATH])
 
 AC_DEFINE_UNQUOTED(BABL_LIBRARY, "$PACKAGE_NAME-$BABL_API_VERSION", [name of BABL library])
-
+AC_SUBST(BABL_LIBRARY)
 
 
 dnl ===========================================================================
diff -Nurd babl-0.0.20.orig/extensions/Makefile.am babl-0.0.20/extensions/Makefile.am
--- babl-0.0.20.orig/extensions/Makefile.am	2008-02-26 21:23:45.000000000 -0500
+++ babl-0.0.20/extensions/Makefile.am	2008-03-03 00:43:46.000000000 -0500
@@ -1,57 +1,49 @@
 # General minimalistic compile file for self contained single
 # file babl_extensions
 
-if OS_WIN32
-no_undefined = -no-undefined
-libbabldlla=$(top_builddir)/babl/.libs/libbabl-$(BABL_API_VERSION).dll.a
-endif
-
-CFILES          = $(wildcard $(srcdir)/*.c)
-SOBJS           = $(subst $(srcdir)/,,$(CFILES:.c=$(SHREXT)))
-INSTALLED_ITEMS = $(subst $(srcdir),$(ext_dir),$(CFILES:.c=$(SHREXT)))
-EXTRA_DIST      = $(wildcard *.[ch])
-
-all-local: $(SOBJS)
-
-CFLAGS += \
-	-I $(top_builddir)		\
-	-I $(top_srcdir)		\
-	-I $(top_srcdir)/babl		\
-	-I $(top_srcdir)/extensions	\
-	-fPIC
+INCLUDES = \
+	-I$(top_builddir)		\
+	-I$(top_srcdir)			\
+	-I$(top_srcdir)/babl		\
+	-I$(top_srcdir)/extensions
 
-LDFLAGS += $(DYNAMICLIB)
+LIBADD = $(top_builddir)/babl/libbabl-$(BABL_API_VERSION).la $(MATH_LIB)
 
+AM_LDFLAGS = -module -avoid-version
 if OS_WIN32
-LDADD    = $(no-undefined) $(libbabldlla) $(MATH_LIB)
-else
-LDADD    = $(top_builddir)/babl/.libs/libbabl-$(BABL_API_VERSION)$(SHREXT) $(MATH_LIB)
+AM_LDFLAGS += -no-undefined
 endif
 
-%$(SHREXT): %.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< $(LDADD)
-
-# if extension needing external libraries are to be # compiled with this make
-# file, each of them can be added according to the this pattern:
-# extra$(SHREXT): extra.c
-#	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $<  [own compile and link flags]
-
-CIE-Lab$(SHREXT): CIE-Lab.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< $(LDADD)
-#lcms$(SHREXT): lcms.c
-#	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $< $(LDADD) `pkg-config lcms --cflags --libs`
-#############################################################################
-
-clean-local:
-	rm -f $(SOBJS)
-
-ext_dir = $(DESTDIR)$(libdir)/babl-@BABL_API_VERSION@
+extensionsdir = $(libdir)/babl-@BABL_API_VERSION@
 
-install-exec-local: all-local
-	$(INSTALL) -d $(ext_dir)
-	$(INSTALL) $(SOBJS) $(ext_dir)
+# each plugin is listed in extensions_LTLIBRARIES with a .la extension
+# (will be replaced by the platform-dependent extension by libtool)
+extensions_LTLIBRARIES =	\
+	CIE-Lab.la	\
+	gegl-fixups.la	\
+	gggl-lies.la	\
+	gggl.la		\
+	gimp-8bit.la	\
+	naive-CMYK.la
 
-uninstall-local:
-	rm -f $(INSTALLED_ITEMS)
+# the source file(s) for each extensions_LTLIBRARIES are listed here
+# (periods and hyphens in an extensions_LTLIBRARIES entry are
+# converted to underscores for the *_SOURCES variable-name)
+CIE_Lab_la_SOURCES = CIE-Lab.c
+CIE_Lab_la_LIBADD = $(LIBADD)
+gegl_fixups_la_SOURCES = gegl-fixups.c
+gegl_fixups_la_LIBADD = $(LIBADD)
+gggl_lies_la_SOURCES = gggl-lies.c
+gggl_lies_la_LIBADD = $(LIBADD)
+gggl_la_SOURCES = gggl.c
+gggl_la_LIBADD = $(LIBADD)
+gimp_8bit_la_SOURCES = gimp-8bit.c
+gimp_8bit_la_LIBADD = $(LIBADD)
+naive_CMYK_la_SOURCES = naive-CMYK.c
+naive_CMYK_la_LIBADD = $(LIBADD)
 
-check-local: all-local
+# INCLUDES is an implicit default for all extensions. If special flags
+# are needed for a certain foo.la extension, it can be over-ridden,
+# and also other libraries can be added as needed:
+# foo_la_CPPFLAGS = -Isomedir $(INCLUDES)
+# foo_la_LIBADD = -lsomelib $(LIBADD)
diff -Nurd babl-0.0.20.orig/tests/Makefile.am babl-0.0.20/tests/Makefile.am
--- babl-0.0.20.orig/tests/Makefile.am	2007-11-10 08:48:07.000000000 -0500
+++ babl-0.0.20/tests/Makefile.am	2008-03-01 15:27:52.000000000 -0500
@@ -23,10 +23,10 @@
 models_SOURCES              = models.c
 
 
-AM_CFLAGS = -I$(top_srcdir) -I$(top_srcdir)/babl
+AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/babl
 
 LDADD = $(top_builddir)/babl/libbabl-@BABL_API_VERSION@.la \
-	$(DL_LIB) $(MATH_LIB)
+	$(MATH_LIB)
 
 noinst_PROGRAMS =		\
 	introspect		\
