diff -ru devil-1.7.8.orig/src-ILU/ilur/ilur.c devil-1.7.8/src-ILU/ilur/ilur.c
--- devil-1.7.8.orig/src-ILU/ilur/ilur.c	2009-03-08 03:10:12.000000000 -0400
+++ devil-1.7.8/src-ILU/ilur/ilur.c	2009-03-16 23:27:15.000000000 -0400
@@ -1,6 +1,6 @@
 #include <string.h>
 #include <stdio.h>
-#include <malloc.h>
+#include <stdlib.h>
 
 #include <IL/il.h>
 #include <IL/ilu.h>
diff -ru devil-1.7.8.orig/test/format_test/format_checks.sh.in devil-1.7.8/test/format_test/format_checks.sh.in
--- devil-1.7.8.orig/test/format_test/format_checks.sh.in	2009-03-08 03:10:07.000000000 -0400
+++ devil-1.7.8/test/format_test/format_checks.sh.in	2009-03-17 07:05:42.000000000 -0400
@@ -25,8 +25,8 @@
 
 for EXTENSION in $EXTENSIONS
 do
-	#echo expr match "' $CAN_SAVE '" '.*\s'$EXTENSION'\s.*'
-	if [ $(expr match "' $CAN_SAVE '" '.*\s'$EXTENSION'\s.*') -eq 0 -o $(expr match "' $PROBLEMATIC '" '.*\s'$EXTENSION'\s.*') -ne 0 ] 
+	#echo expr "' $CAN_SAVE '" : '.*\s'$EXTENSION'\s.*'
+	if [ $(expr "' $CAN_SAVE '" : '.*\s'$EXTENSION'\s.*') -eq 0 -o $(expr "' $PROBLEMATIC '" : '.*\s'$EXTENSION'\s.*') -ne 0 ] 
 	then # This extension is not considered as supported...
 		continue;
 	fi
