diff -ruN libpng-1.2.12/scripts/makefile.darwin libpng-1.2.12-patched/scripts/makefile.darwin
--- libpng-1.2.12/scripts/makefile.darwin	2006-06-27 16:22:40.000000000 -0400
+++ libpng-1.2.12-patched/scripts/makefile.darwin	2006-08-02 21:17:33.000000000 -0400
@@ -38,7 +38,7 @@
 RANLIB=ranlib
 RM_F=/bin/rm -f
 
-CFLAGS=-I$(ZLIBINC) -Wall -O3 -funroll-loops
+CFLAGS=-I$(ZLIBINC) -Wall -O3 -funroll-loops -DPNG_NO_ASSEMBLER_CODE
 LDFLAGS=-L. -L$(ZLIBLIB) -lpng12 -lz
 
 INCPATH=$(prefix)/include
