diff -Nurd libtheora-1.0beta3/doc/Makefile.in libtheora-1.0beta3-new/doc/Makefile.in
--- libtheora-1.0beta3/doc/Makefile.in	2008-04-16 16:26:14.000000000 -0400
+++ libtheora-1.0beta3-new/doc/Makefile.in	2008-08-29 13:37:27.000000000 -0400
@@ -187,7 +187,7 @@
 builddir = @builddir@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/libtheora0
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
@@ -228,7 +228,7 @@
 
 doc_DATA = $(static_docs) doxygen-build.stamp
 EXTRA_DIST = $(static_docs) Doxyfile.in
-dist_docdir = $(distdir)/libtheora
+dist_docdir = $(docdir)
 all: all-recursive
 
 .SUFFIXES:
diff -Nurd libtheora-1.0beta3/lib/enc/x86_32/dct_decode_mmx.c libtheora-1.0beta3-new/lib/enc/x86_32/dct_decode_mmx.c
--- libtheora-1.0beta3/lib/enc/x86_32/dct_decode_mmx.c	2008-04-16 16:25:53.000000000 -0400
+++ libtheora-1.0beta3-new/lib/enc/x86_32/dct_decode_mmx.c	2008-08-29 13:45:12.000000000 -0400
@@ -19,7 +19,7 @@
 
 #include "../codec_internal.h"
 
-#if defined(USE_ASM)
+#if 0
 
 static const __attribute__((aligned(8),used)) ogg_int64_t V3= 0x0003000300030003LL;
 static const __attribute__((aligned(8),used)) ogg_int64_t V804= 0x0804080408040804LL;
@@ -181,3 +181,5 @@
 }
 
 #endif /* USE_ASM */
+
+void dsp_mmx_dct_decode_init(DspFunctions *funcs) { }
diff -Nurd libtheora-1.0beta3/tests/comment.c libtheora-1.0beta3-new/tests/comment.c
--- libtheora-1.0beta3/tests/comment.c	2008-04-16 16:25:49.000000000 -0400
+++ libtheora-1.0beta3-new/tests/comment.c	2008-08-29 13:37:27.000000000 -0400
@@ -15,6 +15,7 @@
 
  ********************************************************************/
 
+#include <sys/types.h>
 #include <theora/theoradec.h>
 
 #include <string.h>
diff -Nurd libtheora-1.0beta3/tests/noop.c libtheora-1.0beta3-new/tests/noop.c
--- libtheora-1.0beta3/tests/noop.c	2008-04-16 16:25:49.000000000 -0400
+++ libtheora-1.0beta3-new/tests/noop.c	2008-08-29 13:37:27.000000000 -0400
@@ -15,6 +15,7 @@
 
  ********************************************************************/
 
+#include <sys/types.h>
 #include <theora/theoraenc.h>
 
 #include "tests.h"
