diff -urN Mesa-6.4.1.orig/bin/mklib Mesa-6.4.1/bin/mklib
--- Mesa-6.4.1.orig/bin/mklib	2005-12-17 15:19:11.000000000 +0000
+++ Mesa-6.4.1/bin/mklib	2005-12-17 15:17:48.000000000 +0000
@@ -505,7 +505,7 @@
         else
             LIBNAME="lib${LIBNAME}.dylib"
             echo "mklib: Making Darwin shared library: " ${LIBNAME}
-            FLAGS="-dynamiclib -multiply_defined suppress -current_version ${MAJOR}.${MINOR}.0 -compatibility_version ${MAJOR}.${MINOR}.0"
+            FLAGS="-dynamiclib -multiply_defined suppress  -install_name @PREFIX@/lib/mesa/${LIBNAME} -current_version ${MAJOR}.${MINOR}.0 -compatibility_version ${MAJOR}.${MINOR}.0"
 	    if [ $CPLUSPLUS = 1 ] ; then
 		LINK="g++"
 	    else
