diff -ur mkvtoolnix-2.1.0/src/output/p_flac.cpp mkvtoolnix-2.1.0.patched/src/output/p_flac.cpp
--- mkvtoolnix-2.1.0/src/output/p_flac.cpp	2006-02-28 02:06:55.000000000 +0900
+++ mkvtoolnix-2.1.0.patched/src/output/p_flac.cpp	2007-10-14 16:13:08.000000000 +0900
@@ -13,6 +13,8 @@
    Written by Moritz Bunkus <moritz@bunkus.org>.
 */
 
+#include "os.h"
+
 #include "config.h"
 
 #if defined(HAVE_FLAC_FORMAT_H)
