diff -ru mpeg2dec-0.4.1.orig/configure mpeg2dec-0.4.1/configure
--- mpeg2dec-0.4.1.orig/configure	2006-02-15 11:34:16.000000000 -0500
+++ mpeg2dec-0.4.1/configure	2006-09-22 19:56:07.000000000 -0400
@@ -4081,6 +4081,7 @@
 _ACEOF
 
 	case "$host" in
+	i?86-apple-darwin*) TRY_CFLAGS="$OPT_CFLAGS";;
 	i386-*) TRY_CFLAGS="$OPT_CFLAGS -mcpu=i386";;
 	i486-*) TRY_CFLAGS="$OPT_CFLAGS -mcpu=i486";;
 	i586-*) TRY_CFLAGS="$OPT_CFLAGS -mcpu=pentium";;
