diff -ruN MPlayer-1.0pre6/configure MPlayer-1.0pre6.new/configure
--- MPlayer-1.0pre6/configure	Thu Dec 23 10:36:00 2004
+++ MPlayer-1.0pre6.new/configure	Tue Dec 28 20:36:49 2004
@@ -272,6 +272,7 @@
 
 Miscellaneous options:
   --enable-runtime-cpudetection    Enable runtime CPU detection [disable]
+  --disable-cpu-optimization       Disable gcc CPU build optimization [enable]
   --cc=COMPILER          use this C compiler to build MPlayer [gcc]
   --host-cc=COMPILER     use this C compiler to build apps needed for the build process [gcc]
   --as=ASSEMBLER         use this assembler to build MPlayer [as]
@@ -365,6 +366,7 @@
 test "$CC" && _cc="$CC"
 _as=auto
 _runtime_cpudetection=no
+_cpu_optimization=yes
 for ac_option do
   case "$ac_option" in
   --target=*)
@@ -406,6 +408,12 @@
   --disable-runtime-cpudetection)
     _runtime_cpudetection=no
     ;;
+  --disable-cpu-optimization)
+    _cpu_optimization=no
+    ;;
+  --enable-cpu-optimization)
+    _cpu_optimization=yes
+    ;;
   --install-path=*)
     _install=`echo $ac_option | cut -d '=' -f 2 | sed 's/\/$//'`"/install"
     ;;
@@ -1029,32 +1037,34 @@
 
     echocheck "GCC & CPU optimization abilities"
 
-    if test -n "$proc"; then
+    if test "$_cpu_optimization" = yes; then
+      if test -n "$proc"; then
         case "$proc" in
-	    601) _march='-mcpu=601' _mcpu='-mtune=601' ;;
-	    603) _march='-mcpu=603' _mcpu='-mtune=603' ;;
-	    603e|603ev) _march='-mcpu=603e' _mcpu='-mtune=603e' ;;
-	    604|604e|604r|604ev) _march='-mcpu=604' _mcpu='-mtune=604' ;;
-	    740|740/750|745/755) _march='-mcpu=740' _mcpu='-mtune=740' ;;
-	    750|750CX) _march='-mcpu=750' _mcpu='-mtune=750' ;;
-    	    *) ;;
+	  601) _march='-mcpu=601' _mcpu='-mtune=601' ;;
+	  603) _march='-mcpu=603' _mcpu='-mtune=603' ;;
+	  603e|603ev) _march='-mcpu=603e' _mcpu='-mtune=603e' ;;
+	  604|604e|604r|604ev) _march='-mcpu=604' _mcpu='-mtune=604' ;;
+	  740|740/750|745/755) _march='-mcpu=740' _mcpu='-mtune=740' ;;
+	  750|750CX) _march='-mcpu=750' _mcpu='-mtune=750' ;;
+    	  *) ;;
         esac
 	# gcc 3.1(.1) and up supports 7400 and 7450
 	if test "$_cc_major" -ge "3" && test "$_cc_minor" -ge "1"; then
-	    case "$proc" in
-		7400*|7410*) _march='-mcpu=7400' _mcpu='-mtune=7400' ;;
-		7450*|7455*) _march='-mcpu=7450' _mcpu='-mtune=7450' ;;
-    		*) ;;
-    	    esac
+	  case "$proc" in
+	    7400*|7410*) _march='-mcpu=7400' _mcpu='-mtune=7400' ;;
+	    7450*|7455*) _march='-mcpu=7450' _mcpu='-mtune=7450' ;;
+    	    *) ;;
+    	  esac
 	fi
 	# gcc 3.2 and up supports 970
 	if test "$_cc_major" -ge "3" && test "$_cc_minor" -ge "3"; then
-	    case "$proc" in
-		970*) _march='-mcpu=970' _mcpu='-mtune=970' ;;
-    		*) ;;
-    	    esac
-	fi
-    fi
+	  case "$proc" in
+	    970*) _march='-mcpu=970' _mcpu='-mtune=970' ;;
+    	    *) ;;
+    	  esac
+	fi #gcc3.3 - 970
+      fi #proc
+    fi #disable_opt
 
     if test -n "$_mcpu"; then
         _optimizing=`echo $_mcpu | cut -c 8-`
@@ -1408,6 +1418,8 @@
   --with-extralibdir=*) ;;
   --enable-runtime-cpudetection) ;;
   --disable-runtime-cpudetection) ;;
+  --enable-cpu-optimization) ;;
+  --disable-cpu-optimization) ;;
   --install-path=*) ;;
   --with-install=*) ;;
 
diff -ruN MPlayer-1.0pre6/etc/example.conf MPlayer-1.0pre6.new/etc/example.conf
--- MPlayer-1.0pre6/etc/example.conf	Tue Jul 20 14:02:22 2004
+++ MPlayer-1.0pre6.new/etc/example.conf	Tue Dec 28 20:39:42 2004
@@ -14,13 +14,13 @@
 ##
 
 # Specify default video driver (see -vo help for a list).
-# vo=xv
+# vo=sdl
 
 # Specify default audio driver (see -ao help for a list).
-# ao=oss
+# ao=sdl
 
 # fullscreen mode
-# fs=yes
+# fs=no
 
 # Change to a different videomode when going fullscreen
 # (dga, x11, sdl video output drivers only).
