diff -Naur netpbm-10.12.old/Makefile.config.fink netpbm-10.12.new/Makefile.config.fink
--- netpbm-10.12.old/Makefile.config.fink	Thu Jan  1 09:00:00 1970
+++ netpbm-10.12.new/Makefile.config.fink	Wed Dec 11 02:34:55 2002
@@ -0,0 +1,19 @@
+#
+# Configuration for fink
+#
+
+DEFAULT_TARGET	= nonmerge
+NETPBMLIBTYPE	= dylib
+NETPBMLIBSUFFIX	= dylib
+STATICLIB_TOO	= n
+NETPBM_RUNTIME_PATH = @PREFIX@/lib
+CC		= cc -no-cpp-precomp
+CFLAGS		= -Os -fstrict-aliasing -funroll-all-loops -isystem @PREFIX@/include
+CFLAGS_SHLIB	= -fno-common
+LADD			= -L@PREFIX@/lib
+LDSHLIB		= -dynamiclib -install_name $(NETPBM_RUNTIME_PATH)/libnetpbm.$(MAJ).dylib -compatibility_version $$(($(MAJ)+1)) -current_version $$(($(MAJ)+1)).$(MIN)
+TIFFLIB		= libtiff.dylib
+JPEGLIB		= libjpeg.dylib
+PNGLIB		= libpng.dylib
+ZLIB		= libz.dylib
+
--- netpbm-10.18.old/buildtools/installnetpbm.pl	Tue Sep  9 12:47:00 2003
+++ netpbm-10.18.new/buildtools/installnetpbm.pl	Tue Oct 14 11:54:06 2003
@@ -105,7 +105,7 @@
     } elsif ($ENV{OSTYPE} && $ENV{OSTYPE} eq "msdosdjgpp") {
         $default = "/dev/env/DJDIR";
     } else {
-        $default = "/usr/local/netpbm";
+        $default = "/sw";
     }
 
     my $response = prompt("install prefix", $default);
@@ -548,7 +548,7 @@
     my $manDir;
 
     while (!$manDir) {
-        my $default = "$prefix/man";
+        my $default = "$prefix/share/man";
 
         my $response = prompt("man page directory", $default);
 
@@ -747,9 +747,9 @@
 
     print("\n");
 
-    removeObsoleteManPage($manDir);
+#    removeObsoleteManPage($manDir);
 
-    makeInManwebPath("$manDir/web");
+#    makeInManwebPath("$manDir/web");
     
 }
 
--- netpbm-10.25/converter/other/pstopnm.c.CAN-2005-2471	2004-06-23 04:22:33.000000000 +0200
+++ netpbm-10.25/converter/other/pstopnm.c	2005-08-09 08:41:42.000000000 +0200
@@ -702,13 +702,13 @@
 
     if (verbose) {
         pm_message("execing '%s' with args '%s' (arg 0), "
-                   "'%s', '%s', '%s', '%s', '%s', '%s', '%s'",
+                   "'%s', '%s', '%s', '%s', '%s', '%s', '%s', '%s'",
                    ghostscriptProg, arg0,
-                   deviceopt, outfileopt, gopt, ropt, "-q", "-dNOPAUSE", "-");
+                   deviceopt, outfileopt, gopt, ropt, "-q", "-dNOPAUSE", "-dPARANOIDSAFER", "-");
     }
 
     execl(ghostscriptProg, arg0, deviceopt, outfileopt, gopt, ropt, "-q",
-          "-dNOPAUSE", "-", NULL);
+          "-dNOPAUSE", "-dPARANOIDSAFER", "-", NULL);
     
     pm_error("execl() of Ghostscript ('%s') failed, errno=%d (%s)",
              ghostscriptProg, errno, strerror(errno));
--- netpbm-10.26.12/converter/other/pnmtopng.c.pnmtopng	2004-08-28 04:53:12.000000000 +0200
+++ netpbm-10.26.12/converter/other/pnmtopng.c	2005-09-16 14:17:47.129390456 +0200
@@ -159,7 +159,7 @@
                       unsigned int * const bestMatchP) {
     
     unsigned int paletteIndex;
-    unsigned int bestIndex;
+    unsigned int bestIndex = 0;
     unsigned int bestMatch;
 
     bestMatch = UINT_MAX;
@@ -1566,7 +1566,7 @@
       /* The color part of the color/alpha palette passed to the PNG
          compressor 
       */
-  unsigned int palette_size;
+  unsigned int palette_size = MAXCOLORS;
 
   gray trans_pnm[MAXCOLORS];
   png_byte  trans[MAXCOLORS];
--- netpbm-10.26.12/converter/other/pnmtopng.c
+++ netpbm-10.26.12/converter/other/pnmtopng.c
@@ -913,9 +913,9 @@
     colorhist_vector chv;
     unsigned int colors;
 
-    gray *alphas_of_color[MAXPALETTEENTRIES];
+    gray *alphas_of_color[MAXPALETTEENTRIES + 1];
     unsigned int alphas_first_index[MAXPALETTEENTRIES];
-    unsigned int alphas_of_color_cnt[MAXPALETTEENTRIES];
+    unsigned int alphas_of_color_cnt[MAXPALETTEENTRIES + 1];
  
     getChv(ifP, imagepos, cols, rows, maxval, format, MAXCOLORS, 
            &chv, &colors);
