diff -Naur netpbm-10.35.75/Makefile.config.fink netpbm-10.35.75-patched//Makefile.config.fink
--- netpbm-10.35.75/Makefile.config.fink	1970-01-01 01:00:00.000000000 +0100
+++ netpbm-10.35.75-patched//Makefile.config.fink	2010-06-01 16:20:14.000000000 +0200
@@ -0,0 +1,18 @@
+#
+# Configuration for fink
+#
+
+DEFAULT_TARGET	= nonmerge
+NETPBMLIBTYPE	= dylib
+NETPBMLIBSUFFIX	= dylib
+STATICLIB_TOO	= n
+NETPBM_RUNTIME_PATH = @PREFIX@/lib
+CC		= cc -no-cpp-precomp
+CFLAGS		= -Os -fstrict-aliasing -funroll-all-loops -isystem @PREFIX@/include
+CFLAGS_SHLIB	= -fno-common
+LADD			= -L@PREFIX@/lib
+LDSHLIB		= -dynamiclib -install_name $(NETPBM_RUNTIME_PATH)/libnetpbm.$(MAJ).dylib -compatibility_version $$(($(MAJ)+1)) -current_version $$(($(MAJ)+1)).$(MIN)
+TIFFLIB		= libtiff.dylib
+JPEGLIB		= libjpeg.dylib
+PNGLIB		= libpng.dylib
+ZLIB		= libz.dylib
diff -Naur netpbm-10.35.75/buildtools/installnetpbm.pl netpbm-10.35.75-patched//buildtools/installnetpbm.pl
--- netpbm-10.35.75/buildtools/installnetpbm.pl	2007-01-15 03:50:02.000000000 +0100
+++ netpbm-10.35.75-patched//buildtools/installnetpbm.pl	2010-06-01 16:17:26.000000000 +0200
@@ -117,7 +117,7 @@
     } elsif ($ENV{OSTYPE} && $ENV{OSTYPE} eq "msdosdjgpp") {
         $default = "/dev/env/DJDIR";
     } else {
-        $default = "/usr/local/netpbm";
+        $default = "@PREFIX@";
     }
 
     my $response = prompt("install prefix", $default);
@@ -572,7 +572,7 @@
     my $manDir;
 
     while (!$manDir) {
-        my $default = "$prefix/man";
+        my $default = "$prefix/share/man";
 
         my $response = prompt("man page directory", $default);
 
@@ -771,9 +771,9 @@
 
     print("\n");
 
-    removeObsoleteManPage($manDir);
+#    removeObsoleteManPage($manDir);
 
-    makeInManwebPath("$manDir/web");
+#    makeInManwebPath("$manDir/web");
     
     $$mandirR = $manDir;
 }
diff -Naur netpbm-10.35.75/converter/other/pnmtopng.c netpbm-10.35.75-patched//converter/other/pnmtopng.c
--- netpbm-10.35.75/converter/other/pnmtopng.c	2009-09-03 17:34:36.000000000 +0200
+++ netpbm-10.35.75-patched//converter/other/pnmtopng.c	2010-06-01 16:14:51.000000000 +0200
@@ -527,7 +527,7 @@
                       unsigned int * const bestMatchP) {
     
     unsigned int paletteIndex;
-    unsigned int bestIndex;
+    unsigned int bestIndex = 0;
     unsigned int bestMatch;
 
     assert(paletteSize > 0);
@@ -1427,9 +1427,9 @@
     colorhist_vector chv;
     unsigned int colors;
 
-    gray * alphas_of_color[MAXPALETTEENTRIES];
+    gray * alphas_of_color[MAXPALETTEENTRIES + 1];
     unsigned int alphas_first_index[MAXPALETTEENTRIES];
-    unsigned int alphas_of_color_cnt[MAXPALETTEENTRIES];
+    unsigned int alphas_of_color_cnt[MAXPALETTEENTRIES + 1];
  
     getChv(ifP, rasterPos, cols, rows, maxval, format, MAXCOLORS, 
            &chv, &colors);
@@ -2348,7 +2348,7 @@
       /* The color part of the color/alpha palette passed to the PNG
          compressor 
       */
-  unsigned int palette_size;
+  unsigned int palette_size = MAXCOLORS;
 
   gray trans_pnm[MAXCOLORS];
   png_byte  trans[MAXCOLORS];
diff -Naur netpbm-10.35.75/pm_config.in.h netpbm-10.35.75-patched//pm_config.in.h
--- netpbm-10.35.75/pm_config.in.h	2006-10-18 20:29:13.000000000 +0200
+++ netpbm-10.35.75-patched//pm_config.in.h	2010-06-01 16:19:18.000000000 +0200
@@ -68,7 +68,7 @@
 #define RGB_DB3 "PBMplus_Dir:RGB.TXT"
 #else
 #define RGB_DB1 "/usr/lib/X11/rgb.txt"
-#define RGB_DB2 "/usr/share/X11/rgb.txt"
+#define RGB_DB2 "/usr/X11/share/X11/rgb.txt"
 #define RGB_DB3 "/usr/X11R6/lib/X11/rgb.txt"
 #endif
 
