--- ParaView-2.6.0/CMake/ParaViewCommon.cmake~	2006-10-24 17:12:48.000000000 +0200
+++ ParaView-2.6.0/CMake/ParaViewCommon.cmake	2007-03-18 17:52:55.000000000 +0100
@@ -66,12 +66,12 @@
 INCLUDE(${VTK_CMAKE_DIR}/vtkSelectSharedLibraries.cmake)
 
 # Disable the install targets if using the rpath feature.
-IF(NOT WIN32)
+IF(NOT WIN32 AND NOT APPLE)
   IF(VTK_USE_RPATH)
     SET(PV_INSTALL_NO_DEVELOPMENT 1)
     SET(PV_INSTALL_NO_RUNTIME 1)
   ENDIF(VTK_USE_RPATH)
-ENDIF(NOT WIN32)
+ENDIF(NOT WIN32 AND NOT APPLE)
 
 # ParaView needs static Tcl/Tk if not using shared libraries.
 IF(NOT BUILD_SHARED_LIBS)
