--- Raster3D_2.7d.orig/Makefile	2007-02-25 12:03:31.000000000 -0500
+++ Raster3D_2.7d/Makefile	2007-02-25 12:04:12.000000000 -0500
@@ -61,9 +61,9 @@
 	@cp Makefile.template Makefile.incl
 	@echo OS = linux                      >> Makefile.incl
 	@echo CC = gcc                        >> Makefile.incl
-	@echo CFLAGS = -g -w -Dgfortran       >> Makefile.incl
+	@echo CFLAGS = -g -O3 -w -Dgfortran       >> Makefile.incl
 	@echo FC = gfortran                   >> Makefile.incl
-	@echo FFLAGS = -g -w -O -ffixed-line-length-132 >> Makefile.incl
+	@echo FFLAGS = -g -w -O3 -ffixed-line-length-132 >> Makefile.incl
 	@echo RM = /bin/rm -f                 >> Makefile.incl
 	@echo OSDEFS =  -DLINUX -DNETWORKBYTEORDER       >> Makefile.incl
 	@echo include Makefile.package        >> Makefile.incl
diff -Nurd -x'*~' Raster3D_2.7d.orig/Makefile.template Raster3D_2.7d/Makefile.template
--- Raster3D_2.7d.orig/Makefile.template	2003-10-20 18:50:11.000000000 -0400
+++ Raster3D_2.7d/Makefile.template	2006-06-26 14:39:18.000000000 -0400
@@ -11,12 +11,12 @@
 prefix  = /usr/local
 bindir  = $(prefix)/bin
 datadir = $(prefix)/share/Raster3D/materials
-mandir  = $(prefix)/man/manl
+mandir  = $(prefix)/share/man/manl
 htmldir = $(prefix)/share/Raster3D/html
 examdir = $(prefix)/share/Raster3D/examples
 
-INCDIRS  =	-I/usr/include -I/usr/local/include
-LIBDIRS  =	-L/usr/local/lib
+INCDIRS  =	-I$(prefix)/include
+LIBDIRS  =	-L$(prefix)/lib
 
 #
 # TIFF support (enabled by default)
--- Raster3D_2.7d.orig/parse.f	2007-02-25 12:43:04.000000000 -0500
+++ Raster3D_2.7d/parse.f	2007-02-25 12:43:36.000000000 -0500
@@ -4,7 +4,8 @@
       COMMON /OPTIONS/ FONTSCALE, GAMMA, ZOOM, NSCHEME, SHADOWFLAG, XBG, 
      &                 NAX, NAY, OTMODE, QUALITY, INVERT, LFLAG
       REAL             FONTSCALE, GAMMA, ZOOM
-      INTEGER          NSCHEME, SHADOWFLAG, XBG
+      INTEGER          NSCHEME, SHADOWFLAG
+      INTEGER*8        XBG
       INTEGER*2        NAX, NAY, OTMODE, QUALITY
       LOGICAL*2        INVERT, LFLAG
 c
--- Raster3D_2.7d.orig/r3dtops.f	2007-02-25 12:43:45.000000000 -0500
+++ Raster3D_2.7d/r3dtops.f	2007-02-25 12:44:23.000000000 -0500
@@ -68,7 +68,8 @@
       COMMON /OPTIONS/ FONTSCALE, GAMMA, ZOOM, NSCHEME, SHADOWFLAG, XBG,
      &                 NAX, NAY, OTMODE, QUALITY, INVERT, LFLAG
       REAL             FONTSCALE, GAMMA, ZOOM
-      INTEGER          NSCHEME, SHADOWFLAG, XBG
+      INTEGER          NSCHEME, SHADOWFLAG
+      INTEGER*8        XBG
       INTEGER*2        NAX, NAY, OTMODE, QUALITY
       LOGICAL*2        INVERT, LFLAG
 *
--- Raster3D_2.7d.orig/render.f	2007-02-25 12:44:33.000000000 -0500
+++ Raster3D_2.7d/render.f	2007-02-25 12:44:58.000000000 -0500
@@ -372,7 +372,8 @@
       COMMON /OPTIONS/ FONTSCALE, GAMMA, ZOOM, NSCHEME, SHADOWFLAG, XBG,
      &                 NAX, NAY, OTMODE, QUALITY, INVERT, LFLAG
       REAL             FONTSCALE, GAMMA, ZOOM
-      INTEGER          NSCHEME, SHADOWFLAG, XBG
+      INTEGER          NSCHEME, SHADOWFLAG
+      INTEGER*8        XBG
       INTEGER*2        NAX, NAY, OTMODE, QUALITY
       LOGICAL*2        INVERT, LFLAG
 *
