diff -Naur sane-backends-1.0.15.orig/frontend/Makefile.in sane-backends-1.0.15/frontend/Makefile.in
--- sane-backends-1.0.15.orig/frontend/Makefile.in	2003-07-26 21:40:06.000000000 +0000
+++ sane-backends-1.0.15/frontend/Makefile.in	2006-03-21 12:27:20.000000000 +0000
@@ -55,7 +55,7 @@
 SBINPROGS = $(SANED)
 PROGRAMS = $(BINPROGS) $(SBINPROGS)
 TESTPROGRAMS = test
-LIBLIB = ../lib/liblib.a
+LIBLIB = ../lib/getopt.o ../lib/getopt1.o ../lib/liblib.a
 LIBSANE = ../backend/libsane.la
 LIBSANEI = ../sanei/libsanei.a
 
diff -Naur sane-backends-1.0.15.orig/lib/Makefile.in sane-backends-1.0.15/lib/Makefile.in
--- sane-backends-1.0.15.orig/lib/Makefile.in	2003-04-02 22:05:26.000000000 +0000
+++ sane-backends-1.0.15/lib/Makefile.in	2006-03-21 12:46:09.000000000 +0000
@@ -48,10 +48,10 @@
 
 @SET_MAKE@
 
-LIBLIB_OBJS      = @LIBOBJS@ @ALLOCA@ getopt.o getopt1.o md5.o
-LIBLIB_LTOBJS    = $(addsuffix .lo,$(basename $(LIBLIB_OBJS)))
+LIBLIB_OBJS      = @LIBOBJS@ @ALLOCA@ md5.o
+LIBLIB_LTOBJS    = $(addsuffix .lo,$(basename $(LIBLIB_OBJS))) getopt.lo getopt1.lo
 
-TARGETS = liblib.a $(LIBLIB_LTOBJS)
+TARGETS = liblib.a getopt.o getopt1.o $(LIBLIB_LTOBJS)
 
 DISTFILES = Makefile.in alloca.c getenv.c getopt.c getopt1.c \
   inet_ntop.c inet_pton.c isfdtype.c md5.c sigprocmask.c snprintf.c strdup.c \
diff -Naur sane-backends-1.0.15.orig/tools/Makefile.in sane-backends-1.0.15/tools/Makefile.in
--- sane-backends-1.0.15.orig/tools/Makefile.in	2004-10-03 17:25:03.000000000 +0000
+++ sane-backends-1.0.15/tools/Makefile.in	2006-03-21 12:26:57.000000000 +0000
@@ -61,7 +61,7 @@
 
 LIBSANEI = ../sanei/libsanei.a
 # Library routines that should be part of libc, but aren't on some systems:
-LIBLIB = ../lib/liblib.a
+LIBLIB = ../lib/getopt.o ../lib/getopt1.o ../lib/liblib.a
 
 DISTFILES = Makefile.in RenSaneDlls.cmd README libtool-get-dll-ext     \
 	mustek600iin-off.c sane-config.in sane-desc.c check-usb-chip.c \
