--- skencil-0.6.17/Filter/Makefile.pre.in	2002-08-06 22:21:15.000000000 +0200
+++ skencil-0.6.17_corr/Filter/Makefile.pre.in	2008-07-22 16:53:12.000000000 +0200
@@ -124,6 +124,7 @@
 MACHDEP=	@MACHDEP@
 SO=		@SO@
 LDSHARED=	@LDSHARED@
+BLDSHARED=      @BLDSHARED@
 CCSHARED=	@CCSHARED@
 LINKFORSHARED=	@LINKFORSHARED@
 #@SET_CCC@
@@ -256,6 +257,7 @@
 	 -e '/^MACHDEP=/s/^MACHDEP=[ 	]*\(.*\)/s%@MACHDEP[@]%\1%/p' \
 	 -e '/^SO=/s/^SO=[ 	]*\(.*\)/s%@SO[@]%\1%/p' \
 	 -e '/^LDSHARED=/s/^LDSHARED=[ 	]*\(.*\)/s%@LDSHARED[@]%\1%/p' \
+         -e '/^BLDSHARED=/s/^BLDSHARED=[        ]*\(.*\)/s%@BLDSHARED[@]%\1%/p' \
 	 -e '/^CCSHARED=/s/^CCSHARED=[ 	]*\(.*\)/s%@CCSHARED[@]%\1%/p' \
 	 -e '/^$L=/s/^$L=[ 	]*\(.*\)/s%@$L[@]%\1%/p' \
 	 -e '/^$P=/s/^$P=\(.*\)/s%^$P=.*%$P=\1%/p' \
diff -ur skencil-0.6.17/Pax/Makefile.pre.in skencil-0.6.17_corr/Pax/Makefile.pre.in
--- skencil-0.6.17/Pax/Makefile.pre.in	2002-08-06 22:20:37.000000000 +0200
+++ skencil-0.6.17_corr/Pax/Makefile.pre.in	2008-07-22 16:55:00.000000000 +0200
@@ -124,6 +124,7 @@
 MACHDEP=	@MACHDEP@
 SO=		@SO@
 LDSHARED=	@LDSHARED@
+BLDSHARED=      @BLDSHARED@
 CCSHARED=	@CCSHARED@
 LINKFORSHARED=	@LINKFORSHARED@
 #@SET_CCC@
@@ -256,6 +257,7 @@
 	 -e '/^MACHDEP=/s/^MACHDEP=[ 	]*\(.*\)/s%@MACHDEP[@]%\1%/p' \
 	 -e '/^SO=/s/^SO=[ 	]*\(.*\)/s%@SO[@]%\1%/p' \
 	 -e '/^LDSHARED=/s/^LDSHARED=[ 	]*\(.*\)/s%@LDSHARED[@]%\1%/p' \
+         -e '/^BLDSHARED=/s/^BLDSHARED=[        ]*\(.*\)/s%@BLDSHARED[@]%\1%/p' \
 	 -e '/^CCSHARED=/s/^CCSHARED=[ 	]*\(.*\)/s%@CCSHARED[@]%\1%/p' \
 	 -e '/^$L=/s/^$L=[ 	]*\(.*\)/s%@$L[@]%\1%/p' \
 	 -e '/^$P=/s/^$P=\(.*\)/s%^$P=.*%$P=\1%/p' \
diff -ur skencil-0.6.17/Sketch/Modules/Makefile.pre.in skencil-0.6.17_corr/Sketch/Modules/Makefile.pre.in
--- skencil-0.6.17/Sketch/Modules/Makefile.pre.in	2002-08-06 22:21:24.000000000 +0200
+++ skencil-0.6.17_corr/Sketch/Modules/Makefile.pre.in	2008-07-22 16:55:53.000000000 +0200
@@ -123,7 +123,8 @@
 RANLIB=		@RANLIB@
 MACHDEP=	@MACHDEP@
 SO=		@SO@
-LDSHARED=	@LDSHARED@
+LDSHARED=       @LDSHARED@
+BLDSHARED=	@BLDSHARED@
 CCSHARED=	@CCSHARED@
 LINKFORSHARED=	@LINKFORSHARED@
 #@SET_CCC@
@@ -256,6 +257,7 @@
 	 -e '/^MACHDEP=/s/^MACHDEP=[ 	]*\(.*\)/s%@MACHDEP[@]%\1%/p' \
 	 -e '/^SO=/s/^SO=[ 	]*\(.*\)/s%@SO[@]%\1%/p' \
 	 -e '/^LDSHARED=/s/^LDSHARED=[ 	]*\(.*\)/s%@LDSHARED[@]%\1%/p' \
+         -e '/^BLDSHARED=/s/^BLDSHARED=[        ]*\(.*\)/s%@BLDSHARED[@]%\1%/p' \
 	 -e '/^CCSHARED=/s/^CCSHARED=[ 	]*\(.*\)/s%@CCSHARED[@]%\1%/p' \
 	 -e '/^$L=/s/^$L=[ 	]*\(.*\)/s%@$L[@]%\1%/p' \
 	 -e '/^$P=/s/^$P=\(.*\)/s%^$P=.*%$P=\1%/p' \
diff -urNad skencil-0.6.17~/Filter/binfile.c skencil-0.6.17/Filter/binfile.c
--- skencil-0.6.17~/Filter/binfile.c	2002-05-01 13:38:49.000000000 -0400
+++ skencil-0.6.17/Filter/binfile.c	2008-03-03 21:22:55.000000000 -0500
@@ -277,7 +277,7 @@
 binfile_dealloc(BinaryInputObject * self)
 {
     Py_DECREF(self->stream);
-    PyMem_DEL(self);
+    PyObject_Del(self);
 }
 
 static PyObject *
diff -urNad skencil-0.6.17~/Filter/filterobj.c skencil-0.6.17/Filter/filterobj.c
--- skencil-0.6.17~/Filter/filterobj.c	2002-05-04 17:58:33.000000000 -0400
+++ skencil-0.6.17/Filter/filterobj.c	2008-03-03 21:22:55.000000000 -0500
@@ -126,7 +126,7 @@
     if (!self->buffer)
     {
     error:
-	PyMem_DEL(self);
+	PyObject_Del(self);
 	PyErr_NoMemory();
 	if (dealloc)
 	    dealloc(client_data);
@@ -136,7 +136,7 @@
     self->filtername = PyString_FromString(name);
     if (!self->filtername)
     {
-	PyMem_DEL(self->buffer);
+	PyObject_Del(self->buffer);
 	goto error;
     }
 
@@ -215,8 +215,8 @@
 	self->dealloc(self->client_data);
     Py_DECREF(self->filtername);
     Py_DECREF(self->stream);
-    PyMem_DEL(self->buffer);
-    PyMem_DEL(self);
+    PyObject_Del(self->buffer);
+    PyObject_Del(self);
 }
 
 static PyObject *
diff -urNad skencil-0.6.17~/Filter/subfilefilter.c skencil-0.6.17/Filter/subfilefilter.c
--- skencil-0.6.17~/Filter/subfilefilter.c	2002-05-01 13:38:49.000000000 -0400
+++ skencil-0.6.17/Filter/subfilefilter.c	2008-03-03 21:22:55.000000000 -0500
@@ -90,7 +90,7 @@
 {
     SubFileDecodeState * state = (SubFileDecodeState*)clientdata;
     Py_DECREF(state->delim_object);
-    PyMem_DEL(state);
+    PyObject_Del(state);
 }
 
 static void
diff -urNad skencil-0.6.17~/Filter/zlibfilter.c skencil-0.6.17/Filter/zlibfilter.c
--- skencil-0.6.17~/Filter/zlibfilter.c	2002-05-01 13:38:49.000000000 -0400
+++ skencil-0.6.17/Filter/zlibfilter.c	2008-03-03 21:22:55.000000000 -0500
@@ -82,7 +82,7 @@
 {
     FlateDecodeState * state = (FlateDecodeState*)clientdata;
     inflateEnd(&state->zstr); /* XXX error handling */
-    PyMem_DEL(state);
+    PyObject_Del(state);
 }
 
 
@@ -102,7 +102,7 @@
     state->buffer = PyMem_NEW(char, INBUFSIZE);
     if (!state->buffer)
     {
-	PyMem_DEL(state);
+	PyObject_Del(state);
 	return PyErr_NoMemory();
     }
 
@@ -132,8 +132,8 @@
 			     "FlateDecode: Zlib Error %i: %.200s",
 			     result, state->zstr.msg);  
 	}
-	PyMem_DEL(state->buffer);
-	PyMem_DEL(state);
+	PyObject_Del(state->buffer);
+	PyObject_Del(state);
 	return NULL;
     }
 
diff -urNad skencil-0.6.17~/Pax/Generate/typedefs.py skencil-0.6.17/Pax/Generate/typedefs.py
--- skencil-0.6.17~/Pax/Generate/typedefs.py	1998-07-25 14:42:55.000000000 -0400
+++ skencil-0.6.17/Pax/Generate/typedefs.py	2008-03-03 21:22:55.000000000 -0500
@@ -24,7 +24,7 @@
 	'intlist': ('O',
 		'PyObject *%(arg)s; int *ints_%(arg)s; int nints_%(arg)s',
 		'checkintlist(%(arg)s, sizeof(int), &ints_%(arg)s, &nints_%(arg)s)',
-		'ints_%(arg)s, nints_%(arg)s', '', 'PyMem_DEL(ints_%(arg)s)'),
+		'ints_%(arg)s, nints_%(arg)s', '', 'PyObject_Del(ints_%(arg)s)'),
 	'long': ('l', 'long %(arg)s', '', '', 'PyInt_FromLong(%(arg)s)'),
 	'unsigned int': ('i', 'unsigned int %(arg)s', '', '',
 		'PyInt_FromLong((long)%(arg)s)'),
@@ -38,7 +38,7 @@
 		'PyObject *%(arg)s; Atom *atoms_%(arg)s; int natoms_%(arg)s',
 		'checkintlist(%(arg)s, sizeof(Atom), &atoms_%(arg)s, &natoms_%(arg)s)',
 		'atoms_%(arg)s, natoms_%(arg)s', '',
-		'PyMem_DEL(atoms_%(arg)s)'),
+		'PyObject_Del(atoms_%(arg)s)'),
 	'Bool': ('i', 'int %(arg)s', '', '(Bool)%(arg)s',
 		'PyInt_FromLong((long)%(arg)s)'),
 	'Boolean': ('i', 'int %(arg)s', '', '(Boolean)%(arg)s',
@@ -90,7 +90,7 @@
 		'PyObject *%(arg)s; XmString *tmp_%(arg)s; int nstrs_%(arg)s',
 		'checkxmstringlist(%(arg)s, &tmp_%(arg)s, &nstrs_%(arg)s)',
 		'tmp_%(arg)s, nstrs_%(arg)s', '',
-		'while (--nstrs_%(arg)s >= 0) XmStringFree(tmp_%(arg)s[nstrs_%(arg)s]); PyMem_DEL(tmp_%(arg)s)'),
+		'while (--nstrs_%(arg)s >= 0) XmStringFree(tmp_%(arg)s[nstrs_%(arg)s]); PyObject_Del(tmp_%(arg)s)'),
 	'XmTextDirection': ('l', 'XmTextDirection %(arg)s', '', '',
 		'PyInt_FromLong((long)%(arg)s)'),
 	'XmTextPosition': ('l', 'XmTextPosition %(arg)s', '', '',
@@ -121,11 +121,11 @@
 	'argdict': ('O',
 		'PyObject *%(arg)s, *dict; ArgList tmp_%(arg)s; int nargs_%(arg)s',
 		'(dict = checkargdict(wClist, sizeof(wClist)/sizeof(wClist[0]), %(w)s, %(arg)s, &tmp_%(arg)s, &nargs_%(arg)s))',
-		'tmp_%(arg)s, nargs_%(arg)s', '', 'PyMem_DEL(tmp_%(arg)s)'),
+		'tmp_%(arg)s, nargs_%(arg)s', '', 'PyObject_Del(tmp_%(arg)s)'),
 	'stringlist': ('O',
 		 'PyObject *%(arg)s; char **tmp_%(arg)s; int nstrs_%(arg)s',
 		'checkstringlist(%(arg)s, &tmp_%(arg)s, &nstrs_%(arg)s)',
-		'tmp_%(arg)s, nstrs_%(arg)s', '', 'PyMem_DEL(tmp_%(arg)s)'),
+		'tmp_%(arg)s, nstrs_%(arg)s', '', 'PyObject_Del(tmp_%(arg)s)'),
 	'cbfunc': ('O', 'PyObject *%(arg)s, *cbfunc', 'cbfunc = %(arg)s;',
 		'univ_callback', ''),
 	'cbarg': ('O', 'PyObject *%(arg)s, *cbarg',
@@ -154,20 +154,20 @@
 	'XPoint[]': ('O',
 		'PyObject *%(arg)s; XPoint *pts_%(arg)s; int npts_%(arg)s',
 		'checkshortlist(2, %(arg)s, (short**)&pts_%(arg)s, &npts_%(arg)s)',
-		'pts_%(arg)s, npts_%(arg)s', '', 'PyMem_DEL(pts_%(arg)s)'),
+		'pts_%(arg)s, npts_%(arg)s', '', 'PyObject_Del(pts_%(arg)s)'),
 	'XSegment[]': ('O',
 		'PyObject *%(arg)s; XSegment *segs_%(arg)s; int nsegs_%(arg)s',
 		'checkshortlist(4, %(arg)s, (short**)&segs_%(arg)s, &nsegs_%(arg)s)',
-		'segs_%(arg)s, nsegs_%(arg)s', '', 'PyMem_DEL(segs_%(arg)s)'),
+		'segs_%(arg)s, nsegs_%(arg)s', '', 'PyObject_Del(segs_%(arg)s)'),
 	'XRectangle[]': ('O',
 		'PyObject *%(arg)s; XRectangle *rects_%(arg)s; int nrects_%(arg)s',
 		'checkshortlist(4, %(arg)s, (short**)&rects_%(arg)s, &nrects_%(arg)s)',
 		'rects_%(arg)s, nrects_%(arg)s', '',
-		'PyMem_DEL(rects_%(arg)s)'),
+		'PyObject_Del(rects_%(arg)s)'),
 	'XArc[]': ('O',
 		'PyObject *%(arg)s; XArc *arcs_%(arg)s; int narcs_%(arg)s',
 		'checkshortlist(6, %(arg)s, (short**)&arcs_%(arg)s, &narcs_%(arg)s)',
-		'arcs_%(arg)s, narcs_%(arg)s', '', 'PyMem_DEL(arcs_%(arg)s)'),
+		'arcs_%(arg)s, narcs_%(arg)s', '', 'PyObject_Del(arcs_%(arg)s)'),
 	'XGCValues#': ('O',
 		'PyObject *%(arg)s; unsigned long mask; XGCValues values',
 		'PaxGC_MakeValues(%(arg)s, &mask, &values)',
diff -urNad skencil-0.6.17~/Pax/borderobject.c skencil-0.6.17/Pax/borderobject.c
--- skencil-0.6.17~/Pax/borderobject.c	2002-05-04 16:49:06.000000000 -0400
+++ skencil-0.6.17/Pax/borderobject.c	2008-03-03 21:22:55.000000000 -0500
@@ -40,7 +40,7 @@
     {
 	Tk_Free3DBorder(self->tkborder);
     }
-    PyMem_DEL(self);
+    PyObject_Del(self);
 }
 
 /* a converter function suitable for PyArg_ParseTuple */
@@ -115,7 +115,7 @@
     
     Tk_Draw3DPolygon(self->tkwin, drawable, self->tkborder, points, npoints,
 		     border_width, left_relief);
-    PyMem_DEL(points);
+    PyObject_Del(points);
 
     Py_INCREF(Py_None);
     return Py_None;
@@ -142,7 +142,7 @@
     
     Tk_Fill3DPolygon(self->tkwin, drawable, self->tkborder, points, npoints,
 		     border_width, left_relief);
-    PyMem_DEL(points);
+    PyObject_Del(points);
 
     Py_INCREF(Py_None);
     return Py_None;
diff -urNad skencil-0.6.17~/Pax/cmapobject.c skencil-0.6.17/Pax/cmapobject.c
--- skencil-0.6.17~/Pax/cmapobject.c	2001-02-11 14:22:12.000000000 -0500
+++ skencil-0.6.17/Pax/cmapobject.c	2008-03-03 21:22:55.000000000 -0500
@@ -30,17 +30,17 @@
     if (plane_masks == NULL || pixels == NULL)
     {
 	if (plane_masks)
-	    PyMem_DEL(plane_masks);
+	    PyObject_Del(plane_masks);
 	if (pixels)
-	    PyMem_DEL(pixels);
+	    PyObject_Del(pixels);
 	return PyErr_NoMemory();
     }
     if (!XAllocColorCells(self->display, self->colormap, contig,
 			  plane_masks, nplanes, pixels, npixels))
     {
 	PyErr_SetString(PyExc_RuntimeError, "XAllocColorCells failed");
-	PyMem_DEL(plane_masks);
-	PyMem_DEL(pixels);
+	PyObject_Del(plane_masks);
+	PyObject_Del(pixels);
 	return NULL;
     }
 
@@ -50,8 +50,8 @@
     res2 = PyList_New(npixels);
     for (i = 0; i < npixels; i++)
 	PyList_SetItem(res2, i, PyInt_FromLong(pixels[i]));
-    PyMem_DEL(plane_masks);
-    PyMem_DEL(pixels);
+    PyObject_Del(plane_masks);
+    PyObject_Del(pixels);
     if (PyErr_Occurred())
     {
 	Py_XDECREF(res1);
@@ -85,7 +85,7 @@
 	item = PyList_GetItem(colorlist, i);
 	if (!PyTuple_Check(item) || PyTuple_Size(item) != 5)
 	{
-	    PyMem_DEL(color);
+	    PyObject_Del(color);
 	    PyErr_BadArgument();
 	    return NULL;
 	}
@@ -96,7 +96,7 @@
 	if (!PyArg_ParseTuple(item, "liiib", &color[i].pixel,
 			      &red, &green, &blue, &color[i].flags))
 	{
-	    PyMem_DEL(color);
+	    PyObject_Del(color);
 	    return NULL;
 	}
 	color[i].red = red;
@@ -106,7 +106,7 @@
 
     XStoreColors(self->display, self->colormap, color, ncolors);
 
-    PyMem_DEL(color);
+    PyObject_Del(color);
     if (PyErr_Occurred())
 	return NULL;
     Py_INCREF(Py_None);
@@ -221,7 +221,7 @@
 	}
     }
 done:
-    PyMem_DEL(defs);
+    PyObject_Del(defs);
     return list;
 }
 
@@ -285,7 +285,7 @@
 	item = PyList_GetItem(pixellist, i);
 	if (!PyInt_Check(item))
 	{
-	    PyMem_DEL(pixels);
+	    PyObject_Del(pixels);
 	    PyErr_BadArgument();
 	    return NULL;
 	}
@@ -294,7 +294,7 @@
 
     XFreeColors(self->display, self->colormap, pixels, npixels, planes);
 
-    PyMem_DEL(pixels);
+    PyObject_Del(pixels);
     if (PyErr_Occurred())
 	return NULL;
     Py_INCREF(Py_None);
@@ -345,7 +345,7 @@
     if (self->owner)
 	XFreeColormap(self->display, self->colormap);
 
-    PyMem_DEL(self);
+    PyObject_Del(self);
 }
 
 PyTypeObject PaxCMapType = {
diff -urNad skencil-0.6.17~/Pax/fontobject.c skencil-0.6.17/Pax/fontobject.c
--- skencil-0.6.17~/Pax/fontobject.c	2001-04-30 10:11:19.000000000 -0400
+++ skencil-0.6.17/Pax/fontobject.c	2008-03-03 21:22:55.000000000 -0500
@@ -69,7 +69,7 @@
     f->font_struct = XLoadQueryFont(display, name);
     if (f->font_struct == NULL)
     {
-	PyMem_DEL(f);
+	PyObject_Del(f);
 	PyErr_SetString(PyExc_RuntimeError, "no such font");
 	return NULL;
     }
@@ -87,7 +87,7 @@
     f->font_struct = XQueryFont(display, fid);
     if (f->font_struct == NULL)
     {
-	PyMem_DEL(f);
+	PyObject_Del(f);
 	PyErr_SetString(PyExc_RuntimeError, "no such font");
 	return NULL;
     }
@@ -269,7 +269,7 @@
 	XFreeFontInfo(NULL, self->font_struct, 1);
     else
 	XFreeFont(self->display, self->font_struct);
-    PyMem_DEL(self);
+    PyObject_Del(self);
 }
 
 PyTypeObject PaxFontType = {
diff -urNad skencil-0.6.17~/Pax/gcmethods.c skencil-0.6.17/Pax/gcmethods.c
--- skencil-0.6.17~/Pax/gcmethods.c	2001-04-30 09:21:10.000000000 -0400
+++ skencil-0.6.17/Pax/gcmethods.c	2008-03-03 21:22:55.000000000 -0500
@@ -45,7 +45,7 @@
 	}
 XDrawArcs(self->display, self->drawable, self->gc,
 			arcs_arg1, narcs_arg1);
-	PyMem_DEL(arcs_arg1);
+	PyObject_Del(arcs_arg1);
 	Py_INCREF(Py_None);
 	return Py_None;
 }
@@ -113,7 +113,7 @@
 XDrawLines(self->display, self->drawable, self->gc,
 			pts_arg1, npts_arg1,
 			arg2);
-	PyMem_DEL(pts_arg1);
+	PyObject_Del(pts_arg1);
 	Py_INCREF(Py_None);
 	return Py_None;
 }
@@ -151,7 +151,7 @@
 XDrawPoints(self->display, self->drawable, self->gc,
 			pts_arg1, npts_arg1,
 			arg2);
-	PyMem_DEL(pts_arg1);
+	PyObject_Del(pts_arg1);
 	Py_INCREF(Py_None);
 	return Py_None;
 }
@@ -192,7 +192,7 @@
 	}
 XDrawRectangles(self->display, self->drawable, self->gc,
 			rects_arg1, nrects_arg1);
-	PyMem_DEL(rects_arg1);
+	PyObject_Del(rects_arg1);
 	Py_INCREF(Py_None);
 	return Py_None;
 }
@@ -211,7 +211,7 @@
 	}
 XDrawSegments(self->display, self->drawable, self->gc,
 			segs_arg1, nsegs_arg1);
-	PyMem_DEL(segs_arg1);
+	PyObject_Del(segs_arg1);
 	Py_INCREF(Py_None);
 	return Py_None;
 }
@@ -282,7 +282,7 @@
 	}
 XFillArcs(self->display, self->drawable, self->gc,
 			arcs_arg1, narcs_arg1);
-	PyMem_DEL(arcs_arg1);
+	PyObject_Del(arcs_arg1);
 	Py_INCREF(Py_None);
 	return Py_None;
 }
@@ -307,7 +307,7 @@
 			pts_arg1, npts_arg1,
 			arg2,
 			arg3);
-	PyMem_DEL(pts_arg1);
+	PyObject_Del(pts_arg1);
 	Py_INCREF(Py_None);
 	return Py_None;
 }
@@ -348,7 +348,7 @@
 	}
 XFillRectangles(self->display, self->drawable, self->gc,
 			rects_arg1, nrects_arg1);
-	PyMem_DEL(rects_arg1);
+	PyObject_Del(rects_arg1);
 	Py_INCREF(Py_None);
 	return Py_None;
 }
@@ -492,7 +492,7 @@
 			arg2,
 			rects_arg3, nrects_arg3,
 			arg4);
-	PyMem_DEL(rects_arg3);
+	PyObject_Del(rects_arg3);
 	Py_INCREF(Py_None);
 	return Py_None;
 }
diff -urNad skencil-0.6.17~/Pax/gcobject.c skencil-0.6.17/Pax/gcobject.c
--- skencil-0.6.17~/Pax/gcobject.c	2001-04-30 10:11:20.000000000 -0400
+++ skencil-0.6.17/Pax/gcobject.c	2008-03-03 21:22:55.000000000 -0500
@@ -152,7 +152,7 @@
 	PyObject *item = PyList_GetItem(list, i);
 	if (!PyInt_Check(item))
 	{
-	    PyMem_DEL(*parray);
+	    PyObject_Del(*parray);
 	    PyErr_SetString(PyExc_TypeError, "list of ints expected");
 	    return 0;
 	}
@@ -241,7 +241,7 @@
 	return NULL;
 
     XSetDashes(self->display, self->gc, dash_offset, dashes, num_dashes);
-    PyMem_DEL(dashes);
+    PyObject_Del(dashes);
 
     Py_INCREF(Py_None);
     return Py_None;
@@ -478,7 +478,7 @@
     else if (self->shared == PAXGC_OWNED)
 	XFreeGC(self->display, self->gc);
     Py_XDECREF(self->drawable_object);
-    PyMem_DEL(self);
+    PyObject_Del(self);
 }
 
 PyTypeObject PaxGCType =
diff -urNad skencil-0.6.17~/Pax/imageobject.c skencil-0.6.17/Pax/imageobject.c
--- skencil-0.6.17~/Pax/imageobject.c	2000-06-15 14:39:18.000000000 -0400
+++ skencil-0.6.17/Pax/imageobject.c	2008-03-03 21:22:55.000000000 -0500
@@ -153,13 +153,13 @@
 	XDestroyImage(self->ximage);
 	shmdt(self->shminfo->shmaddr);
 	shmctl(self->shminfo->shmid, IPC_RMID, 0);
-	PyMem_DEL(self->shminfo);
+	PyObject_Del(self->shminfo);
     }
     else
 #endif
 	XDestroyImage(self->ximage);
     
-    PyMem_DEL(self);
+    PyObject_Del(self);
 }
 
 PyTypeObject PaxImageType = {
@@ -215,7 +215,7 @@
     self = (PaxImageObject*)PaxImage_FromImage(ximage);
     if (!self)
     {
-	PyMem_DEL(shminfo);
+	PyObject_Del(shminfo);
 	return NULL;
     }
 
diff -urNad skencil-0.6.17~/Pax/paxtkinter.c skencil-0.6.17/Pax/paxtkinter.c
--- skencil-0.6.17~/Pax/paxtkinter.c	2004-08-07 15:52:22.000000000 -0400
+++ skencil-0.6.17/Pax/paxtkinter.c	2008-03-03 21:22:55.000000000 -0500
@@ -1201,7 +1201,7 @@
 	ENTER_PYTHON
 	Py_XDECREF(data->self);
 	Py_XDECREF(data->func);
-	PyMem_DEL(data);
+	PyObject_Del(data);
 	LEAVE_PYTHON
 }
 
@@ -1236,7 +1236,7 @@
 	LEAVE_TCL
 	if (err == NULL) {
 		PyErr_SetString(Tkinter_TclError, "can't create Tcl command");
-		PyMem_DEL(data);
+		PyObject_Del(data);
 		return NULL;
 	}
 
@@ -1307,7 +1307,7 @@
 			*pp = p->next;
 			Py_XDECREF(p->func);
 			Py_XDECREF(p->file);
-			PyMem_DEL(p);
+			PyObject_Del(p);
 		}
 		else
 			pp = &p->next;
@@ -1499,7 +1499,7 @@
 
 	Py_XDECREF(func);
 
-	PyMem_DEL(self);
+	PyObject_Del(self);
 }
 
 static PyObject *
@@ -1825,7 +1825,7 @@
 	ENTER_TCL
 	Tcl_DeleteInterp(Tkapp_Interp(self));
 	LEAVE_TCL
-	PyMem_DEL(self);
+	PyObject_Del(self);
 	DisableEventHook();
 }
 
diff -urNad skencil-0.6.17~/Pax/paxutil.c skencil-0.6.17/Pax/paxutil.c
--- skencil-0.6.17~/Pax/paxutil.c	1998-09-25 15:53:34.000000000 -0400
+++ skencil-0.6.17/Pax/paxutil.c	2008-03-03 21:22:55.000000000 -0500
@@ -25,7 +25,7 @@
 	if (!PyTuple_Check(item) || PyTuple_Size(item) != width)
 	{
 	    char buf[100];
-	    PyMem_DEL(*parray);
+	    PyObject_Del(*parray);
 	    sprintf(buf, "list of %d-tuples expected", width);
 	    PyErr_SetString(PyExc_TypeError, buf);
 	    return 0;
@@ -35,7 +35,7 @@
 	    PyObject *elem = PyTuple_GetItem(item, j);
 	    if (!PyInt_Check(elem))
 	    {
-		PyMem_DEL(*parray);
+		PyObject_Del(*parray);
 		PyErr_SetString(PyExc_TypeError,
 				"list of tuples of ints expected");
 		return 0;
diff -urNad skencil-0.6.17~/Pax/pixmapobject.c skencil-0.6.17/Pax/pixmapobject.c
--- skencil-0.6.17~/Pax/pixmapobject.c	2002-02-16 13:54:54.000000000 -0500
+++ skencil-0.6.17/Pax/pixmapobject.c	2008-03-03 21:22:55.000000000 -0500
@@ -381,7 +381,7 @@
 {
     if (self->owner)
 	XFreePixmap(self->display, self->pixmap);
-    PyMem_DEL(self);
+    PyObject_Del(self);
 }
 
 PyTypeObject PaxPixmapType = {
diff -urNad skencil-0.6.17~/Pax/regionobject.c skencil-0.6.17/Pax/regionobject.c
--- skencil-0.6.17~/Pax/regionobject.c	2002-05-04 16:49:49.000000000 -0400
+++ skencil-0.6.17/Pax/regionobject.c	2008-03-03 21:22:55.000000000 -0500
@@ -178,7 +178,7 @@
 region_dealloc(PaxRegionObject  *self)
 {
     XDestroyRegion(self->region);
-    PyMem_DEL(self);
+    PyObject_Del(self);
 }
 
 
diff -urNad skencil-0.6.17~/Pax/tkwinobject.c skencil-0.6.17/Pax/tkwinobject.c
--- skencil-0.6.17~/Pax/tkwinobject.c	2002-05-04 16:50:02.000000000 -0400
+++ skencil-0.6.17/Pax/tkwinobject.c	2008-03-03 21:22:55.000000000 -0500
@@ -47,7 +47,7 @@
 static void
 tkwin_dealloc(TkWinObject * self)
 {
-    PyMem_DEL(self);
+    PyObject_Del(self);
 }
 
 
@@ -365,7 +365,7 @@
     if (ximage == NULL)
     {
 	PyErr_SetString(PyExc_RuntimeError, "XCreateImage failed");
-	PyMem_DEL(newdata);
+	PyObject_Del(newdata);
 	return NULL;
     }
     return PaxImage_FromImage(ximage);
@@ -463,7 +463,7 @@
 	    shmdt(shminfo->shmaddr);
 	if (shminfo->shmid != -1)
 	    shmctl(shminfo->shmid, IPC_RMID, 0);
-	PyMem_DEL(shminfo);
+	PyObject_Del(shminfo);
     }
     return NULL;
 }
@@ -551,7 +551,7 @@
 	XDestroyImage(ximage);
 	shmdt(shminfo->shmaddr);
 	shmctl(shminfo->shmid, IPC_RMID, 0);
-	PyMem_DEL(shminfo);
+	PyObject_Del(shminfo);
 	Py_INCREF(Py_None);
 	return Py_None;
     }
@@ -567,7 +567,7 @@
 	    shmdt(shminfo->shmaddr);
 	if (shminfo->shmid != -1)
 	    shmctl(shminfo->shmid, IPC_RMID, 0);
-	PyMem_DEL(shminfo);
+	PyObject_Del(shminfo);
     }
     return NULL;
 }
@@ -704,7 +704,7 @@
 	return NULL;
     }
     reg = XPolygonRegion(points, npoints, fill_rule);
-    PyMem_DEL(points);
+    PyObject_Del(points);
     if (reg == NULL)
 	return PyErr_NoMemory();
     return PaxRegion_FromRegion(reg);
diff -urNad skencil-0.6.17~/Sketch/Modules/curveobject.c skencil-0.6.17/Sketch/Modules/curveobject.c
--- skencil-0.6.17~/Sketch/Modules/curveobject.c	2003-03-15 17:29:45.000000000 -0500
+++ skencil-0.6.17/Sketch/Modules/curveobject.c	2008-03-03 21:22:55.000000000 -0500
@@ -117,7 +117,7 @@
     self->segments = malloc(length * sizeof(CurveSegment));
     if (!self->segments)
     {
-	PyMem_DEL(self);
+	PyObject_Del(self);
 	return PyErr_NoMemory();
     }
     self->allocated = length;
@@ -218,7 +218,7 @@
 curve_dealloc(SKCurveObject * self)
 {
     free(self->segments);
-    PyMem_DEL(self);
+    PyObject_Del(self);
     paths_allocated--;
 }
 
diff -urNad skencil-0.6.17~/Sketch/Modules/pstokenize.c skencil-0.6.17/Sketch/Modules/pstokenize.c
--- skencil-0.6.17~/Sketch/Modules/pstokenize.c	2001-05-12 15:24:49.000000000 -0400
+++ skencil-0.6.17/Sketch/Modules/pstokenize.c	2008-03-03 21:22:55.000000000 -0500
@@ -691,7 +691,7 @@
 pstokenizer_dealloc(PSTokenizerObject * self)
 {
     Py_DECREF(self->source);
-    PyMem_DEL(self);
+    PyObject_Del(self);
 }
 
 static PyObject *
diff -urNad skencil-0.6.17~/Sketch/Modules/skaux.c skencil-0.6.17/Sketch/Modules/skaux.c
--- skencil-0.6.17~/Sketch/Modules/skaux.c	2002-05-04 16:43:51.000000000 -0400
+++ skencil-0.6.17/Sketch/Modules/skaux.c	2008-03-03 21:22:55.000000000 -0500
@@ -287,7 +287,7 @@
     self->dict = PyDict_New();
     if (!self->dict)
     {
-	PyMem_DEL(self);
+	PyObject_Del(self);
 	return NULL;
     }
 
@@ -298,7 +298,7 @@
 SKCache_dealloc(SKCacheObject * self)
 {
     Py_DECREF(self->dict);
-    PyMem_DEL(self);
+    PyObject_Del(self);
 }
 
 static PyObject *
diff -urNad skencil-0.6.17~/Sketch/Modules/skcolor.c skencil-0.6.17/Sketch/Modules/skcolor.c
--- skencil-0.6.17~/Sketch/Modules/skcolor.c	2002-05-04 16:44:46.000000000 -0400
+++ skencil-0.6.17/Sketch/Modules/skcolor.c	2008-03-03 21:22:55.000000000 -0500
@@ -103,7 +103,7 @@
     self->ob_type = (PyTypeObject*)free_list;
     free_list = self;
 #else
-    PyMem_DEL(self);
+    PyObject_Del(self);
 #endif
 #if SKCOLOR_COUNT_ALLOC
     skcolor_allocated--;
@@ -478,20 +478,20 @@
 	{
 	    for (j = 0; j < 8; j++)
 	    {
-		PyMem_DEL(self->dither_matrix[i][j]);
+		PyObject_Del(self->dither_matrix[i][j]);
 	    }
-	    PyMem_DEL(self->dither_matrix[i]);
+	    PyObject_Del(self->dither_matrix[i]);
 	}
-	PyMem_DEL(self->dither_matrix);
+	PyObject_Del(self->dither_matrix);
     }
     if (self->dither_red)
-	PyMem_DEL(self->dither_red);
+	PyObject_Del(self->dither_red);
     if (self->dither_green)
-	PyMem_DEL(self->dither_green);
+	PyObject_Del(self->dither_green);
     if (self->dither_blue)
-	PyMem_DEL(self->dither_blue);
+	PyObject_Del(self->dither_blue);
     if (self->dither_gray)
-	PyMem_DEL(self->dither_gray);
+	PyObject_Del(self->dither_gray);
 }
 
 static int
@@ -594,7 +594,7 @@
     self->visualinfo = PyMem_NEW(XVisualInfo, 1);
     if (!self->visualinfo)
     {
-	PyMem_DEL(self);
+	PyObject_Del(self);
 	return PyErr_NoMemory();
     }
     memcpy(self->visualinfo, info, sizeof(XVisualInfo));
@@ -629,7 +629,7 @@
     if (self->free_extra)
 	self->free_extra(self);
     free(self->visualinfo);
-    PyMem_DEL(self);
+    PyObject_Del(self);
 }
 
 static PyObject *
diff -urNad skencil-0.6.17~/Sketch/Modules/skfm.c skencil-0.6.17/Sketch/Modules/skfm.c
--- skencil-0.6.17~/Sketch/Modules/skfm.c	2002-05-04 16:45:06.000000000 -0400
+++ skencil-0.6.17/Sketch/Modules/skfm.c	2008-03-03 21:22:55.000000000 -0500
@@ -85,7 +85,7 @@
 static void
 skfm_dealloc(SKFontMetric * self)
 {
-    PyMem_DEL(self);
+    PyObject_Del(self);
 }
 
 
diff -urNad skencil-0.6.17~/Sketch/Modules/skpoint.c skencil-0.6.17/Sketch/Modules/skpoint.c
--- skencil-0.6.17~/Sketch/Modules/skpoint.c	2002-05-04 16:40:50.000000000 -0400
+++ skencil-0.6.17/Sketch/Modules/skpoint.c	2008-03-03 21:22:55.000000000 -0500
@@ -62,7 +62,7 @@
 static void
 skpoint_dealloc(SKPointObject * self)
 {
-    PyMem_DEL(self);
+    PyObject_Del(self);
 #if SKPOINT_COUNT_ALLOC
     allocated -= 1;
 #endif
diff -urNad skencil-0.6.17~/Sketch/Modules/skrect.c skencil-0.6.17/Sketch/Modules/skrect.c
--- skencil-0.6.17~/Sketch/Modules/skrect.c	2002-05-04 16:45:32.000000000 -0400
+++ skencil-0.6.17/Sketch/Modules/skrect.c	2008-03-03 21:22:55.000000000 -0500
@@ -114,7 +114,7 @@
     self->ob_type = (PyTypeObject*)free_list;
     free_list = self;
 #else
-    PyMem_DEL(self);
+    PyObject_Del(self);
 #endif
 #if SKRECT_COUNT_ALLOC
     allocated--;
diff -urNad skencil-0.6.17~/Sketch/Modules/sktrafo.c skencil-0.6.17/Sketch/Modules/sktrafo.c
--- skencil-0.6.17~/Sketch/Modules/sktrafo.c	2002-06-15 14:02:53.000000000 -0400
+++ skencil-0.6.17/Sketch/Modules/sktrafo.c	2008-03-03 21:22:55.000000000 -0500
@@ -82,7 +82,7 @@
 static void
 sktrafo_dealloc(SKTrafoObject * self)
 {
-    PyMem_DEL(self);
+    PyObject_Del(self);
 #if SKTRAFO_COUNT_ALLOC
     allocated--;
 #endif
