diff -Nurd -x'*~' tesseract-2.04.orig/configure tesseract-2.04/configure
--- tesseract-2.04.orig/configure	2009-06-02 20:55:00.000000000 -0400
+++ tesseract-2.04/configure	2010-02-09 05:05:16.000000000 -0500
@@ -1970,17 +1970,6 @@
 fi
 
 # Special cases
-case "$host" in
-  *-darwin* | *-macos10*)
-     if test -d /opt/local ; then
-       CPPFLAGS="$CPPFLAGS -I/opt/local/include"
-       LDFLAGS="$LDFLAGS -L/opt/local/lib"
-     elif test -d /sw ; then
-       CPPFLAGS="$CPPFLAGS -I/sw/include"
-       LDFLAGS="$LDFLAGS -L/sw/lib"
-     fi
-  ;;
-esac
 
 # ----------------------------------------
 # Check Compiler Characteristics and
