diff -Nurd -x'*~' xpaint-2.7.8.1.orig/Imakefile xpaint-2.7.8.1/Imakefile
--- xpaint-2.7.8.1.orig/Imakefile	2005-06-03 08:26:03.000000000 -0400
+++ xpaint-2.7.8.1/Imakefile	2005-09-27 16:59:22.000000000 -0400
@@ -9,6 +9,7 @@
 #include "./Local.xawdefs"
 
 DEFINES = $(ARCH_DEFINES) $(EXTRA_DEFINES) $(XAWLIB_DEFINES) $(XPM_INCLUDE) \
+          -I@PREFIX@/include \
           "-DEDITOR=\"$(EDITOR)\"" \
           "-DSHAREDIR=\"$(SHAREDIR)\"" \
           "-DXAPPLOADDIR=\"$(XAPPLOADDIR)\"" \
@@ -102,7 +103,7 @@
 #endif
 
 DEPLIBS = always xpaint.man XPaint.ad
-LOCAL_LIBRARIES = -Lrw -lrw $(XPM_LIB) $(TIFF_LIB) $(SGI_LIB) $(JPEG_LIB) $(PNG_LIB) -ldl -rdynamic
+LOCAL_LIBRARIES = -Lrw -lrw $(XPM_LIB) $(TIFF_LIB) $(SGI_LIB) $(JPEG_LIB) $(PNG_LIB) -ldl
 
 SUBDIRS = rw
 
@@ -116,6 +117,8 @@
 # Other targets...
 
 start::
+	rm -f xaw_incdir
+	ln -sf @PREFIX@/include/X11/Xaw3d xaw_incdir	
 	@if ! [ -d xaw_incdir ] ; then ./configure ; make ; fi
 
 install::
@@ -151,7 +154,7 @@
 	@echo "'xpaint' compiled with Xaw3d widgets"
 	@echo "Variants are 'make xaw3dg' and 'make nextaw' and 'make xaw95'"
 	@echo ""	
-	
+
 xaw3dg::
 	-$(RM) $(XAWDEPENDS)
 	./configure xaw3dg ; make	
@@ -176,7 +179,7 @@
 	@echo "Another alternative is to use the Xaw3d widgets :"	
 	@echo "Type 'make xaw3d' or 'make xaw3dg'"
 	@echo ""	
-	
+
 clean::
 	-$(RM) preproc substads XPaint.ad XPaint.ad.h DefaultRC.txt.h PGP.*
 	$(RM) -rf share/out xaw_incdir
diff -Nurd -x'*~' xpaint-2.7.8.1.orig/Local.config xpaint-2.7.8.1/Local.config
--- xpaint-2.7.8.1.orig/Local.config	2005-04-30 17:33:14.000000000 -0400
+++ xpaint-2.7.8.1/Local.config	2005-09-27 18:08:27.000000000 -0400
@@ -8,8 +8,8 @@
 
 XCOMM Local customization
 
-DESTDIR = 
-SHAREDIR = /usr/share/xpaint
+DESTDIR = @DESTDIR@
+SHAREDIR = @PREFIX@/share/xpaint
 
 XCOMM Printing system defaults
 PRINT_COMMAND = lpr
@@ -81,7 +81,7 @@
 XCOMM  Which is either 'gcc' or 'acc' on a SparcStation
 XCOMM   most MIPS based machines will do just fine using 'cc'
 XCOMM
-XCOMM CC = gcc
+CC = gcc
 XCOMM CC = acc
 
 XCOMM  If you have the TIFF library, make HaveTIFF defined
@@ -144,10 +144,10 @@
 XCOMM Uncomment these lines and tailor to taste, if you don't like
 XCOMM the default paths and man page extension.
 XCOMM
-XCOMM BINDIR = /usr/local/bin
-XCOMM MANDIR = /usr/local/man
-XCOMM #undef ManSuffix
-XCOMM #define ManSuffix 1
+BINDIR = @PREFIX@/bin
+MANDIR = @PREFIX@/share/man/man1
+#undef ManSuffix
+#define ManSuffix 1
 
 XCOMM Add '-g' here if you want to include debugging information.
 XCOMM Also add any optimizer flags etc. that you want.
@@ -179,3 +179,18 @@
 JPEG_LIB     = -L/usr/local/lib -ljpeg
 XPM_INCLUDE = -I/usr/X11R6/include/X11
 #endif
+
+ARCH_DEFINES = -DHAVE_PARAM_H
+TIFF_INCLUDE = -I@PREFIX@/include
+TIFF_LIB     = -L@PREFIX@/lib -ltiff
+JPEG_INCLUDE = -I@PREFIX@/include
+JPEG_LIB     = -L@PREFIX@/lib -ljpeg
+PNG_LIB      = -lpng -lz
+PNG_INCLUDE  = -I@PREFIX@/include
+XPM_INCLUDE  = -I/usr/X11R6/include/X11
+
+CONFDIR = @PREFIX@/etc
+XAPPLOADDIR = $(CONFDIR)/app-defaults
+DOCDIR = @PREFIX@/share/doc/xpaint
+
+DEPEND = \# gccmakedep
diff -Nurd -x'*~' xpaint-2.7.8.1.orig/bitmaps/Imakefile xpaint-2.7.8.1/bitmaps/Imakefile
--- xpaint-2.7.8.1.orig/bitmaps/Imakefile	2002-11-14 05:09:21.000000000 -0500
+++ xpaint-2.7.8.1/bitmaps/Imakefile	2005-09-27 16:06:24.000000000 -0400
@@ -1,5 +1,5 @@
 MISC = Imakefile
-SRCS = *\/*
+SRCS = 
 
 makelist:
 	sh -c 'for i in '"$(TEXT) $(SRCS) $(MISC) $(HDRS)"' ; do \
diff -Nurd -x'*~' xpaint-2.7.8.1.orig/configure xpaint-2.7.8.1/configure
--- xpaint-2.7.8.1.orig/configure	2005-06-03 08:26:11.000000000 -0400
+++ xpaint-2.7.8.1/configure	2005-09-27 17:36:25.000000000 -0400
@@ -7,7 +7,7 @@
    echo "(looking for Xaw, Xaw3d, Xaw95, neXtaw)"
    echo ""
 else
-   WIDGETS=`echo $1 | tr [a:z] [A:Z]`
+   WIDGETS=`echo $1 | tr [a-z] [A-Z]`
 fi
 
 if test "$WIDGETS" = "" ; then
@@ -66,11 +66,11 @@
 fi
 
 if test "$WIDGETS" = "XAW3D" ; then
-   ln -sf /usr/include/X11/Xaw3d xaw_incdir	
+   ln -sf @PREFIX@/include/X11/Xaw3d xaw_incdir	
    echo "XAWLIB_DEFINES = -DXAW3D"   
    echo "XAWLIB_DEFINES = -DXAW3D" > Local.xawdefs
-   echo "SYS_LIBRARIES = -lXaw3d -L."   
-   echo "SYS_LIBRARIES = -lXaw3d -L." >> Local.xawdefs
+   echo "SYS_LIBRARIES = -lXaw3d -lX11 -lXt -lXext -L."   
+   echo "SYS_LIBRARIES = -lXaw3d -lX11 -lXt -lXext -L." >> Local.xawdefs
    echo ""   
    xmkmf ; echo "cd rw" ; cd rw ; xmkmf
    exit
diff -Nurd -x'*~' xpaint-2.7.8.1.orig/misc.h xpaint-2.7.8.1/misc.h
--- xpaint-2.7.8.1.orig/misc.h	2005-08-15 15:50:05.000000000 -0400
+++ xpaint-2.7.8.1/misc.h	2005-09-27 15:54:26.000000000 -0400
@@ -17,22 +17,6 @@
 #define USE_DRAND
 #endif
 
-#if defined(SVR4) || defined(__osf__)
-#define SHORT_RANGE
-#else
-#if !defined(__GLIBC__) & !defined(random) & !defined(__CYGWIN__)
-long random(void);
-#endif
-
-#if !defined(__VMS) & !defined(linux) & !defined(__EMX__) & !defined(__FreeBSD__) & !defined(__CYGWIN__)
-#if defined(BSD4_4) || defined(HPArchitecture) || defined(SGIArchitecture) || defined(_AIX) || defined(_SCO_DS)
-void srandom(unsigned int);
-#else
-int srandom(unsigned int);
-#endif	/* BSD4_4 */
-#endif	/* linux */
-#endif
-
 #ifdef USE_DRAND
 #ifdef DECLARE_DRAND48
 extern double drand48();
@@ -101,9 +85,7 @@
 /* help.c */
 char *matchGet(char *line, char *pat);
 void HelpDialog(Widget parent, String name);
-#if defined( _STDIO_H ) || defined( __VMS ) 
 void HelpTextOutput(FILE* fd, String name);
-#endif
 void HelpInit(Widget top);
 
 /* main.c */
