--- xvidcore/build/generic/configure.orig	2009-05-28 13:13:31.000000000 -0400
+++ xvidcore/build/generic/configure	2011-02-22 16:38:01.000000000 -0500
@@ -1319,7 +1319,7 @@
 API_MAJOR="4"
 API_MINOR="2"
 
-minimum_yasm_minor_version=8
+minimum_yasm_major_version=1
 minimum_nasm_minor_version=0
 minimum_nasm_major_version=2
 nasm_prog="nasm"
@@ -3977,14 +3977,14 @@
    if test "$ac_yasm" = "yes" ; then
                             echo "$as_me:$LINENO: checking for yasm version" >&5
 echo $ECHO_N "checking for yasm version... $ECHO_C" >&6
-       yasm_minor=`$yasm_prog --version | cut -d '.' -f 2 | cut -d ' ' -f 1`
-       if test -z $yasm_minor ; then
-          yasm_minor=-1
+       yasm_major=`$yasm_prog --version | head -n 1 | cut -d '.' -f 1 | cut -d ' ' -f 2`
+       if test -z $yasm_major ; then
+          yasm_major=-1
        fi
-       echo "$as_me:$LINENO: result: $yasm_minor" >&5
-echo "${ECHO_T}$yasm_minor" >&6
+       echo "$as_me:$LINENO: result: $yasm_major" >&5
+echo "${ECHO_T}$yasm_major" >&6
 
-                     if test "$yasm_minor" -lt "$minimum_yasm_minor_version" ; then
+                     if test "$yasm_major" -lt "$minimum_yasm_major_version" ; then
          { echo "$as_me:$LINENO: WARNING: yasm version is too old" >&5
 echo "$as_me: WARNING: yasm version is too old" >&2;}
        else
