diff -Nurd -x'*~' automake-1.12.6.orig/t/pkg-config-macros.sh automake-1.12.6/t/pkg-config-macros.sh
--- automake-1.12.6.orig/t/pkg-config-macros.sh	2012-12-14 12:21:46.000000000 -0500
+++ automake-1.12.6/t/pkg-config-macros.sh	2013-02-17 17:18:14.000000000 -0500
@@ -48,9 +48,10 @@
 
 # Find the location of the pkg-config executable.
 oIFS=$IFS dir=
+IFS=:
 for d in $PATH; do
   IFS=$oIFS
-  if test -f $dir/pkg-config || test -f $dir/pkg-config.exe; then
+  if test -f $d/pkg-config || test -f $d/pkg-config.exe; then
     dir=$d
     break
   fi
@@ -61,7 +62,7 @@
 # where the corresponding pkg.m4 might be installed.
 if test -n "$dir"; then
   # Only support standard installation layouts.
-  XT_ACLOCAL_PATH=${dir%/bin}/share/alocal:$XT_ACLOCAL_PATH
+  XT_ACLOCAL_PATH=${dir%/bin}/share/aclocal:$XT_ACLOCAL_PATH
 fi
 
 XT_ACLOCAL_PATH=$XT_ACLOCAL_PATH${ACLOCAL_PATH+":$ACLOCAL_PATH"}
