diff --git a/t/python-am-path-iftrue.sh b/t/python-am-path-iftrue.sh
index 3c5201f..d7ba436 100755
--- a/t/python-am-path-iftrue.sh
+++ b/t/python-am-path-iftrue.sh
@@ -45,8 +45,10 @@ cat > mypy.m4 << 'END'
 AM_PATH_PYTHON([0.0], [$PYTHON -c 'print("%u:%u" % (1-1, 2**0))' > py.out])
 END
 
-$ACLOCAL
-$AUTOCONF
+# The "--force" option here is truly needed to avoid potential timestamp
+# races.  See automake bug#12210.
+$ACLOCAL --force
+$AUTOCONF --force
 ./configure
 test x"$(cat py.out)" = x0:1
 
diff --git a/t/python-missing.sh b/t/python-missing.sh
index c88a4f9..b0912d3 100755
--- a/t/python-missing.sh
+++ b/t/python-missing.sh
@@ -58,8 +58,10 @@ rm -rf autom4te*.cache # Will have to re-run aclocal.
 
 # Hopefully the Python team will never release such a version :-)
 echo 'AM_PATH_PYTHON([9999.9], [])' > mypy.m4
-$ACLOCAL
-$AUTOCONF
+# The "--force" options (here and below) are truly needed to avoid
+# potential timestamp races.  See automake bug#12210.
+$ACLOCAL --force
+$AUTOCONF --force
 ./configure >stdout 2>stderr && {
   cat stdout
   cat stderr >&2
--- a/t/distcheck-override-infodir.sh
+++ b/t/distcheck-override-infodir.sh
@@ -39,7 +39,7 @@ installcheck-local:
 END
 
 # Protect with leading " # " to avoid spurious maintainer-check failures.
-sed 's/^ #//' > main.texi << 'END'
+sed 's/^ *# *//' > main.texi << 'END'
  # \input texinfo
  # @setfilename main.info
  # @settitle main
