diff -Naur --exclude='*~' idutils/fink/50idutils.el idutils.fink/fink/50idutils.el
--- idutils/fink/50idutils.el	1969-12-31 18:00:00.000000000 -0600
+++ idutils.fink/fink/50idutils.el	2007-12-26 21:11:37.000000000 -0600
@@ -0,0 +1,1 @@
+(autoload 'gid "idutils")
diff -Naur --exclude='*~' idutils/fink/idutils-install idutils.fink/fink/idutils-install
--- idutils/fink/idutils-install	1969-12-31 18:00:00.000000000 -0600
+++ idutils.fink/fink/idutils-install	2007-12-15 15:04:19.000000000 -0600
@@ -0,0 +1,28 @@
+#!/bin/bash -e
+#
+# install file for the fink idutils emacs package.
+
+set -o posix
+
+FLAVOR=${1}
+
+echo >&2 "install/idutils: Handling install of emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs ]
+then
+    # Nothing to do, say nothing
+    exit 0
+fi
+
+echo >&2 -n "install/idutils: Byte-compiling for ${FLAVOR}..."
+mkdir -p @PREFIX@/share/${FLAVOR}/site-lisp/idutils
+ln -s -f @PREFIX@/share/emacs/site-lisp/idutils.el @PREFIX@/share/${FLAVOR}/site-lisp/idutils.el
+
+cd @PREFIX@/share/${FLAVOR}/site-lisp;
+
+(${FLAVOR} --no-init-file -batch -f batch-byte-compile idutils.el 2>&1) | gzip -9qf > @PREFIX@/share/doc/idutils/CompilationLog-${FLAVOR}.gz
+
+echo >&2 "done."
+echo >&2 "install/idutils: Compilation log saved in @PREFIX@/share/doc/idutils/CompilationLog-${FLAVOR}.gz."
+
+exit 0
diff -Naur --exclude='*~' idutils/fink/idutils-remove idutils.fink/fink/idutils-remove
--- idutils/fink/idutils-remove	1969-12-31 18:00:00.000000000 -0600
+++ idutils.fink/fink/idutils-remove	2007-12-10 18:35:39.000000000 -0600
@@ -0,0 +1,15 @@
+#!/bin/bash -e
+#
+# This is the Fink version of the idutils emacs package.
+set -o posix
+
+FLAVOR=${1}
+
+echo "remove/idutils: Handling removal for emacsen flavor ${FLAVOR}"
+
+echo >&2 -n "remove/idutils: Purging compilation log and byte-compiled files for ${FLAVOR}..."
+rm -f @PREFIX@/share/doc/idutils/CompilationLog-${FLAVOR}.gz
+rm -Rf @PREFIX@/share/${FLAVOR}/site-lisp/idutils
+echo >&2 "done."
+    
+exit 0
--- idutils.orig/lisp/idutils.el
+++ idutils/lisp/idutils.el
@@ -34,7 +34,7 @@
 (require 'compile)
 (provide 'idutils)
 
-(defvar gid-command "gid" "The command run by the gid function.")
+(defvar gid-command "grepid" "The command run by the gid function.")
 
 (defun gid (args)
   "Run gid, with user-specified ARGS, and collect output in a buffer.
