diff -Naur proguard3.2.orig/bin/proguard proguard3.2/bin/proguard
--- proguard3.2.orig/bin/proguard	Thu Jan  1 10:00:00 1970
+++ proguard3.2/bin/proguard	Sun Jun  1 19:33:41 2003
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+exec java -jar @PREFIX@/share/proguard/proguard.jar $*
--- proguard3.2.orig/bin/proguardgui	Thu Jan  1 10:00:00 1970
+++ proguard3.2/bin/proguardgui	Sun Jun  1 19:33:41 2003
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+exec java -Xdock:name=ProGuard -jar @PREFIX@/share/proguard/proguardgui.jar $*
diff -Naur proguard3.2.orig/bin/retrace proguard3.2/bin/retrace
--- proguard3.2.orig/bin/retrace	Thu Jan  1 10:00:00 1970
+++ proguard3.2/bin/retrace	Sun Jun  1 19:33:41 2003
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+exec java -cp @PREFIX@/share/proguard/retrace.jar proguard.retrace.ReTrace $*
diff -Naur proguard3.2.orig/src/proguard/gui/default.pro proguard3.2/src/proguard/gui/default.pro
--- proguard3.2.orig/src/proguard/gui/default.pro    Sun Nov 21 02:39:21 2004
+++ proguard3.2/src/proguard/gui/default.pro       Sat Jan 22 16:39:45 2005
@@ -1,6 +1,6 @@
 # The default configuration when starting up the GUI.
 
--libraryjars <java.home>/lib/rt.jar
+-libraryjars /System/Library/Frameworks/JavaVM.framework/Classes/classes.jar
 
 -verbose
 
diff -Naur proguard3.2.orig/src/proguard/gui/ProGuardGUI.java proguard3.2/src/proguard/gui/ProGuardGUI.java
--- proguard3.2.orig/src/proguard/gui/ProGuardGUI.java       Sun Nov 21 02:08:57 2004
+++ proguard3.2/src/proguard/gui/ProGuardGUI.java  Sat Jan 22 17:17:16 2005
@@ -54,6 +54,9 @@
 
     private static final Border BORDER = BorderFactory.createEtchedBorder(EtchedBorder.RAISED);
 
+    private static final boolean MAC_OS_X =
+        System.getProperty("os.name").toLowerCase().startsWith("mac os x");
+
     static boolean systemOutRedirected;
 
     private JFileChooser configurationChooser = new JFileChooser("");
@@ -203,6 +206,13 @@
         lastBottomButtonConstraints.insets    = bottomButtonConstraints.insets;
         lastBottomButtonConstraints.ipadx     = bottomButtonConstraints.ipadx;
         lastBottomButtonConstraints.ipady     = bottomButtonConstraints.ipady;
+
+        if (MAC_OS_X) {
+            // Leave room for growBox in bottom right of window
+            lastBottomButtonConstraints.insets =
+                (Insets)lastBottomButtonConstraints.insets.clone();
+            lastBottomButtonConstraints.insets.right += 16;
+        }
 
         GridBagLayout layout = new GridBagLayout();
 
