diff -Nurd -x'*~' pychecker-0.8.17.orig/setup.py pychecker-0.8.17/setup.py
--- pychecker-0.8.17.orig/setup.py	2006-02-03 21:29:46.000000000 -0500
+++ pychecker-0.8.17/setup.py	2007-01-05 11:29:50.000000000 -0500
@@ -255,7 +255,7 @@
         'url'              : "http://pychecker.sourceforge.net/",
         'packages'         : [ 'pychecker', ],
         'scripts'          : [ "pychecker" ],   # note: will be replaced by customized action
-        'data_files'       : [ ( "pychecker", DATA_FILES, ) ], 
+#       'data_files'       : [ ( "pychecker", DATA_FILES, ) ], 
         'long_description' : LONG_DESCRIPTION,
         'cmdclass'         : CUSTOMIZED_ACTIONS, 
       }
