diff -ur rapidsvn-0.9.8.orig/configure rapidsvn-0.9.8/configure
--- rapidsvn-0.9.8.orig/configure	2009-03-19 18:22:51.000000000 -0500
+++ rapidsvn-0.9.8/configure	2009-07-06 00:13:11.000000000 -0500
@@ -15057,7 +15057,7 @@
 
 
 abs_srcdir="`cd $srcdir && pwd`"
-CPPFLAGS="$CPPFLAGS -I$abs_srcdir/include"
+CPPFLAGS="-I$abs_srcdir/include $CPPFLAGS"
 
 { echo "$as_me:$LINENO: checking for gcc/g++" >&5
 echo $ECHO_N "checking for gcc/g++... $ECHO_C" >&6; }
diff -ur rapidsvn-0.9.8.orig/packages/osx/make_osx_bundle.sh rapidsvn-0.9.8/packages/osx/make_osx_bundle.sh
--- rapidsvn-0.9.8.orig/packages/osx/make_osx_bundle.sh	2009-03-19 18:08:45.000000000 -0500
+++ rapidsvn-0.9.8/packages/osx/make_osx_bundle.sh	2009-07-07 01:05:15.000000000 -0500
@@ -7,7 +7,6 @@
 DISKIMAGE=RapidSVN-0.9.8
 DISKIMAGE_FILE=$DISKIMAGE.dmg
 test -e RapidSVN.app && rm -rf RapidSVN.app
-rm *.dmg
 
 # Create the bundle
 BUNDLEDIR=RapidSVN.app/Contents
@@ -15,7 +14,6 @@
 mkdir -p $BUNDLEDIR/Resources
 
 cp ../../src/rapidsvn $BUNDLEDIR/MacOS/
-strip $BUNDLEDIR/MacOS/rapidsvn
 cp rapidsvn.icns $BUNDLEDIR/Resources/
 echo -n 'APPL????' > $BUNDLEDIR/PkgInfo
 
@@ -25,12 +23,10 @@
 for DIR in ??; do
   mkdir -p ../../packages/osx/$BUNDLEDIR/Resources/locale/$DIR
   cp $DIR/rapidsvn.po ../../packages/osx/$BUNDLEDIR/Resources/locale/$DIR
-  cp $DIR/rapidsvn.mo ../../packages/osx/$BUNDLEDIR/Resources/locale/$DIR
 done
 for DIR in ??_??; do
   mkdir -p ../../packages/osx/$BUNDLEDIR/Resources/locale/$DIR
   cp $DIR/rapidsvn.po ../../packages/osx/$BUNDLEDIR/Resources/locale/$DIR
-  cp $DIR/rapidsvn.mo ../../packages/osx/$BUNDLEDIR/Resources/locale/$DIR
 done
 popd
 
@@ -68,9 +64,3 @@
 </dict>
 </plist>
 " > $BUNDLEDIR/Info.plist
-
-# Now create the disk image from the bundle
-hdiutil create -quiet -srcfolder RapidSVN.app $DISKIMAGE
-
-test -e $DISKIMAGE_FILE && echo Success!!
-
diff -ur rapidsvn-0.9.8.orig/src/Makefile.in rapidsvn-0.9.8/src/Makefile.in
--- rapidsvn-0.9.8.orig/src/Makefile.in	2009-03-19 18:22:50.000000000 -0500
+++ rapidsvn-0.9.8/src/Makefile.in	2009-07-06 00:14:24.000000000 -0500
@@ -327,8 +327,8 @@
 	verblist.cpp \
 	view_action.cpp 
 
-rapidsvn_LDFLAGS = -Lsvncpp
-rapidsvn_LDADD = -lsvncpp \
+rapidsvn_LDFLAGS =
+rapidsvn_LDADD = svncpp/libsvncpp.la \
 	$(SVN_LIBS) \
 	$(APR_LIBS) \
 	$(WX_LIBS) \
diff -ur rapidsvn-0.9.8.orig/src/svncpp/Makefile.in rapidsvn-0.9.8/src/svncpp/Makefile.in
--- rapidsvn-0.9.8.orig/src/svncpp/Makefile.in	2009-03-19 18:22:50.000000000 -0500
+++ rapidsvn-0.9.8/src/svncpp/Makefile.in	2009-07-06 00:15:03.000000000 -0500
@@ -52,7 +52,7 @@
 am__installdirs = "$(DESTDIR)$(libdir)"
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libsvncpp_la_LIBADD =
+libsvncpp_la_LIBADD = $(SVN_LIBS)
 am_libsvncpp_la_OBJECTS = apr.lo client.lo client_annotate.lo \
 	client_cat.lo client_diff.lo client_ls.lo client_modify.lo \
 	client_property.lo client_status.lo context.lo datetime.lo \
diff -ur rapidsvn-0.9.8.orig/src/tests/svncpp/Makefile.in rapidsvn-0.9.8/src/tests/svncpp/Makefile.in
--- rapidsvn-0.9.8.orig/src/tests/svncpp/Makefile.in	2009-03-19 18:22:50.000000000 -0500
+++ rapidsvn-0.9.8/src/tests/svncpp/Makefile.in	2009-07-06 00:16:04.000000000 -0500
@@ -210,8 +210,8 @@
 	test_path.cpp
 
 svncpptest_CPPFLAGS = -I../../../include @CPPUNIT_CXXFLAGS@
-svncpptest_LDFLAGS = -L../../svncpp @CPPUNIT_LIBS@
-svncpptest_LDADD = -lsvncpp \
+svncpptest_LDFLAGS = @CPPUNIT_LIBS@
+svncpptest_LDADD = ../../svncpp/libsvncpp.la \
 	$(SVN_LIBS) \
 	$(NEON_LIBS) \
 	$(APR_LIBS)	
