--- abiword-1.0.1/abi/src/af/xap/unix/xap_UnixFrame.cpp.orig	Sat Apr 27 06:40:31 2002
+++ abiword-1.0.1/abi/src/af/xap/unix/xap_UnixFrame.cpp	Sat Apr 27 06:41:04 2002
@@ -1000,7 +1000,7 @@
 
 	// ORDER:
 	// Use value of environment variable BROWSER, if valid; otherwise:
-	// 1) konqueror
+	// 1) dillo
 	// 2) mozilla
 	// 3) netscape
 	// 4) kdehelp
@@ -1024,9 +1024,9 @@
   	}
 	if (fmtstring == 0)
 	{
-		if(progExists("konqueror"))
+		if(progExists("dillo"))
 		{
-			fmtstring = "konqueror '%s' &";
+			fmtstring = "dillo '%s' &";
 			execstring = g_strdup_printf(fmtstring, szURL);
 		}
 		else if(progExists("galeon"))
--- abiword-1.0.1/wv/libole2/Makefile.in.orig	Tue Apr 30 15:18:12 2002
+++ abiword-1.0.1/wv/libole2/Makefile.in	Tue Apr 30 15:18:41 2002
@@ -13,7 +13,7 @@
 HAVE_GLIB = @HAVE_GLIB@
 HAVE_GNOMEVFS= @HAVE_GNOMEVFS@
 
-CFLAGS = @CPPFLAGS@ @CFLAGS@ @DEFS@ -I$(srcdir)/ -I$(srcdir)/.. -DTTFDIR=\"$(datadir)/wv/ttf\" -I$(srcdir)/../libole2 @WMF_CFLAGS@ @GLIB_CFLAGS@
+CFLAGS = @CFLAGS@ @DEFS@ -I$(srcdir)/ -I$(srcdir)/.. -DTTFDIR=\"$(datadir)/wv/ttf\" -I$(srcdir)/../libole2 @CPPFLAGS@ @WMF_CFLAGS@ @GLIB_CFLAGS@
 ifneq ($(HAVE_GLIB), 1)
 CFLAGS += -I$(srcdir)/../glib-wv/
 endif
--- abiword-1.0.1/abi/src/pkg/common/unix/scripts/makewrapper.sh.orig	Thu May  2 05:30:41 2002
+++ abiword-1.0.1/abi/src/pkg/common/unix/scripts/makewrapper.sh	Thu May  2 05:32:50 2002
@@ -74,6 +74,12 @@
 # locale-specific dirs could be added to it.
 ABISUITE_FONT_PATH=\$ABISUITE_FONT_HOME
 
+#set LANG to eng_us if unset (otherwise crash results on OS X gtk+).
+if [ -z "$LANG" ]
+then 
+    export LANG=en_US
+fi
+
 #now try to guess locale
 locale=\$LC_ALL	#it's incorrect to set this variable, but someone
 		#might set it incorrectly.
--- abiword-1.0.3/abi/src/wp/main/unix/GNUmakefile.am.orig	Thu Nov  7 14:49:52 2002
+++ abiword-1.0.3/abi/src/wp/main/unix/GNUmakefile.am	Thu Nov  7 14:50:06 2002
@@ -32,7 +32,7 @@
 	$(AF_INCLUDES) $(WP_INCLUDES) $(OTHER_INCLUDES) \
 	$(TEXT_INCLUDES) @GMODULE_CFLAGS@ @GTK_CFLAGS@ 
 
-AbiWord_s_LDFLAGS = -static
+AbiWord_s_LDFLAGS =
 
 AbiWord_d_SOURCES = abi_ver.cpp UnixMain.cpp
 AbiWord_s_SOURCES = abi_ver.cpp UnixMain.cpp
