--- elib-1.0.orig/elib.texi     Mon Dec 11 11:50:55 1995
+++ elib-1.0/elib.texi  Sat Sep 21 20:25:46 2002
@@ -21,6 +21,16 @@
 @comment the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
 @setfilename elib.info
 @settitle Elib - The Emacs Lisp Library
+
+@ifinfo
+@format
+INFO-DIR-SECTION Emacs
+START-INFO-DIR-ENTRY
+* Elib: (elib). The Emacs Lisp Library.
+END-INFO-DIR-ENTRY
+@end format
+@end ifinfo
+
 @setchapternewpage odd
 @syncodeindex fn cp
      
diff -Naur elib-1.0.orig/emacsen-install elib-1.0/emacsen-install
--- elib-1.0.orig/emacsen-install	Thu Jan  1 10:00:00 1970
+++ elib-1.0/emacsen-install	Wed Sep 25 02:24:08 2002
@@ -0,0 +1,24 @@
+#! /bin/sh -e
+
+FLAVOR=$1
+PACKAGE=elib
+
+el_dir=@PREFIX@/share/emacs/site-lisp/elib
+elc_dir=@PREFIX@/share/${FLAVOR}/site-lisp/elib
+el_files="`cd ${el_dir}; echo *.el`"
+
+echo install/elib: Handling install of emacsen flavor ${FLAVOR}
+
+if [ ${FLAVOR} != emacs ]; then
+  echo install/elib: byte-compiling for ${FLAVOR}
+
+    rm -rf ${elc_dir}
+    mkdir -p ${elc_dir}
+    cp ${el_dir}/* ${elc_dir}
+    cd ${elc_dir}
+    make EMACS=${FLAVOR} ELIBDIR=${elc_dir} elcfiles
+
+    rm Makefile *.el
+fi
+
+exit 0
diff -Naur elib-1.0.orig/emacsen-remove elib-1.0/emacsen-remove
--- elib-1.0.orig/emacsen-remove	Thu Jan  1 10:00:00 1970
+++ elib-1.0/emacsen-remove	Wed Sep 25 02:25:09 2002
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+FLAVOR=$1
+
+echo remove/elib: Handling removal of emacsen flavor ${FLAVOR}
+
+if [ ${FLAVOR} != emacs ]
+then
+  echo remove/elib: purging byte-compiled files for ${FLAVOR}
+
+  rm -rf @PREFIX@/share/${FLAVOR}/site-lisp/elib
+
+fi
+
+exit 0
