diff -Naur nxml-mode-20031018/fink/emacsen-install nxml-mode-20031018.fink/fink/emacsen-install
--- nxml-mode-20031018/fink/emacsen-install	Thu Jan  1 10:00:00 1970
+++ nxml-mode-20031018.fink/fink/emacsen-install	Thu Oct 23 17:07:02 2003
@@ -0,0 +1,39 @@
+#! /bin/sh
+# @PREFIX@/lib/emacsen-common/packages/install/nxml-mode
+set -e
+
+FLAVOR=${1}
+PACKAGE=nxml-mode
+
+echo install/${PACKAGE}: Handling install of emacsen flavor ${FLAVOR}
+
+el_files="nxml-mode.el rng-dt.el rng-ns.el rng-valid.el xsd-regexp.el \
+          rng-maint.el rng-pttrn.el rng-xsd.el rng-cmpct.el rng-match.el \
+          rng-nxml.el rng-util.el xmltok.el nxml-enc.el nxml-uchnm.el \
+          nxml-glyph.el rng-auto.el"
+
+el_dir=@PREFIX@/share/emacs/site-lisp/${PACKAGE}/
+elc_dir=@PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}/
+el_path_list=`for i in ${el_files}; do \
+                  echo $i | perl -pe "s|^|${el_dir}|"; done`
+elc_path_list=`for i in ${el_files}; do \
+                  echo $i | perl -pe "s|^|${elc_dir}|"; done`
+
+byte_compile_options="-batch -q --no-init-file --no-site-file"
+
+case $FLAVOR in
+    emacs21)
+	echo install/${PACKAGE}: byte-compiling for ${FLAVOR}
+
+	install -m 755 -d ${elc_dir}
+
+	# Copy the temp .el files to the installation dir
+	cp ${el_path_list} ${elc_dir}
+	
+	;;
+    emacs);;
+    *)
+	echo install/${PACKAGE}: Skipping unsupported flavor ${FLAVOR}
+	;;
+esac
+
diff -Naur nxml-mode-20031018/fink/emacsen-remove nxml-mode-20031018.fink/fink/emacsen-remove
--- nxml-mode-20031018/fink/emacsen-remove	Thu Jan  1 10:00:00 1970
+++ nxml-mode-20031018.fink/fink/emacsen-remove	Thu Oct 23 17:07:02 2003
@@ -0,0 +1,20 @@
+#! /bin/sh
+# @PREFIX@/lib/emacsen-common/packages/remove/nxml-mode
+set -e
+
+FLAVOR=${1}
+PACKAGE=nxml-mode
+
+echo remove/${PACKAGE}: Handling removal of emacsen flavor ${FLAVOR}
+
+case $FLAVOR in
+    emacs21)
+	echo remove/${PACKAGE}: purging byte-compiled files for ${FLAVOR}
+	rm -rf @PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}
+	;;
+    emacs);;
+    *)
+	echo remove/${PACKAGE}: Skipping unsupported flavor ${FLAVOR}
+	;;
+esac
+
diff -Naur nxml-mode-20031018/nxml-mode.info nxml-mode-20031018.fink/nxml-mode.info
--- nxml-mode-20031018/nxml-mode.info	Sat Oct 18 16:43:05 2003
+++ nxml-mode-20031018.fink/nxml-mode.info	Thu Oct 23 17:17:51 2003
@@ -3,7 +3,7 @@
 
 INFO-DIR-SECTION Emacs
 START-INFO-DIR-ENTRY
-* nXML Mode: (nxml-mode.info).
+* nXML Mode: (nxml-mode.info).     Emacs major mode for editing XML
 END-INFO-DIR-ENTRY
 
 
diff -Naur nxml-mode-20031018/rng-auto.el nxml-mode-20031018.fink/rng-auto.el
--- nxml-mode-20031018/rng-auto.el	Sat Oct 18 16:33:14 2003
+++ nxml-mode-20031018.fink/rng-auto.el	Thu Oct 23 17:06:59 2003
@@ -62,28 +62,28 @@
 			(list (car rule)
 			      (abbreviate-file-name
 			       (expand-file-name (cdr rule) dir))))
-		      '((".*\\.xsl\\'" . "schema/xslt.rnc")
-			(".*\\.x?html?\\'" . "schema/xhtml.rnc")
-			(".*\\.rng\\'" . "schema/relaxng.rnc")
-			(".*\\.rdf\\'" . "schema/rdfxml.rnc")))))
+		      '((".*\\.xsl\\'" . "@PREFIX@/share/schema/xslt.rnc")
+			(".*\\.x?html?\\'" . "@PREFIX@/share/schema/xhtml.rnc")
+			(".*\\.rng\\'" . "@PREFIX@/share/schema/relaxng.rnc")
+			(".*\\.rdf\\'" . "@PREFIX@/share/schema/rdfxml.rnc")))))
   (setq rng-auto-element-alist-default
 	(mapcar (lambda (rule)
 		  (cons (car rule)
 			(abbreviate-file-name
 			 (expand-file-name (cdr rule) dir))))
-		'((("http://www.w3.org/1999/XSL/Transform" t t) . "schema/xslt.rnc")
-		  ((t "xsl" "transform") . "schema/xslt.rnc")
-		  ((t t "stylesheet") . "schema/xslt.rnc")
-		  (("http://www.w3.org/1999/xhtml" t t) . "schema/xhtml.rnc")
-		  ((t t "html") . "schema/xhtml.rnc")
-		  (("http://relaxng.org/ns/structure/1.0" t t) . "schema/relaxng.rnc")
-		  ((t t "grammar") . "schema/relaxng.rnc")
-		  ((nil nil "article") . "schema/docbook.rnc")
-		  ((nil nil "book") . "schema/docbook.rnc")
+		'((("http://www.w3.org/1999/XSL/Transform" t t) . "@PREFIX@/share/schema/xslt.rnc")
+		  ((t "xsl" "transform") . "@PREFIX@/share/schema/xslt.rnc")
+		  ((t t "stylesheet") . "@PREFIX@/share/schema/xslt.rnc")
+		  (("http://www.w3.org/1999/xhtml" t t) . "@PREFIX@/share/schema/xhtml.rnc")
+		  ((t t "html") . "@PREFIX@/share/schema/xhtml.rnc")
+		  (("http://relaxng.org/ns/structure/1.0" t t) . "@PREFIX@/share/schema/relaxng.rnc")
+		  ((t t "grammar") . "@PREFIX@/share/schema/relaxng.rnc")
+		  ((nil nil "article") . "@PREFIX@/share/schema/docbook.rnc")
+		  ((nil nil "book") . "@PREFIX@/share/schema/docbook.rnc")
 		  (("http://www.w3.org/1999/02/22-rdf-syntax-ns#" t t)
-		   . "schema/rdfxml.rnc")
-		  ((t t "RDF") . "schema/rdfxml.rnc")
-		  ((t "rdf" t) . "schema/rdfxml.rnc")))))
+		   . "@PREFIX@/share/schema/rdfxml.rnc")
+		  ((t t "RDF") . "@PREFIX@/share/schema/rdfxml.rnc")
+		  ((t "rdf" t) . "@PREFIX@/share/schema/rdfxml.rnc")))))
 
 ;; Users shouldn't edit this.
 ;; Instead they should customize `rng-preferred-prefix-alist'.
