diff -uNr vim62/src/auto/configure vim62-new/src/auto/configure
--- vim62/src/auto/configure	Sun Jun  1 09:42:39 2003
+++ vim62-new/src/auto/configure	Mon Nov  8 16:01:40 2004
@@ -1189,7 +1189,7 @@
 if test -z "$have_local_include" -a -d /usr/local/include; then
   tt=`echo "$CPPFLAGS" | sed -e 's+-I/usr/local/include ++g' -e 's+-I/usr/local/include$++g'`
   if test "$tt" = "$CPPFLAGS"; then
-    CPPFLAGS="$CPPFLAGS -I/usr/local/include"
+    CPPFLAGS="$CPPFLAGS"
   fi
 fi
 
diff -uNr vim62/src/mbyte.c vim62-new/src/mbyte.c
--- vim62/src/mbyte.c	Sat May 31 12:12:56 2003
+++ vim62-new/src/mbyte.c	Mon Nov  8 16:01:41 2004
@@ -541,7 +541,6 @@
 # else
 	    char buf[MB_MAXBYTES];
 # ifdef X_LOCALE
-	    extern int _Xmblen __ARGS((char *, size_t));
 #  ifndef mblen
 #   define mblen _Xmblen
 #  endif
diff -uNr vim62/src/osdef2.h.in vim62-new/src/osdef2.h.in
--- vim62/src/osdef2.h.in	Sat Dec 15 16:24:12 2001
+++ vim62-new/src/osdef2.h.in	Mon Nov  8 16:08:19 2004
@@ -79,3 +79,9 @@
 #else
 extern char	*mktemp __ARGS((char *));
 #endif
+
+#ifdef ISC
+extern int _Xmblen __ARGS((char const *, size_t));
+#else
+extern int _Xmblen __ARGS((char *, int));
+#endif
