diff -Naur sqlite-3.2.2-0002/Makefile.in sqlite-3.2.2-0001/Makefile.in
--- sqlite-3.2.2-0002/Makefile.in	2005-06-06 10:44:41.000000000 -0400
+++ sqlite-3.2.2-0001/Makefile.in	2005-06-16 09:58:54.000000000 -0400
@@ -189,6 +189,10 @@
   $(TOP)/src/vdbe.c \
   $(TOP)/src/md5.c
 
+# In LIBOBJ but not TESTSRC
+COMMONOBJ = $(foreach obj,$(LIBOBJ),\
+	$(if $(findstring $(patsubst %.lo,%.c,$(obj)),$(TESTSRC)),,$(obj)))
+
 # Header files used by all library source files.
 #
 HDR = \
@@ -403,17 +407,15 @@
 	$(LTLINK) -o tclsqlite3 tclsqlite-shell.lo \
 		 libsqlite3.la $(LIBTCL)
 
-testfixture$(TEXE):	$(TOP)/src/tclsqlite.c libtclsqlite3.la libsqlite3.la $(TESTSRC)
-	$(LTLINK) -DTCLSH=1 -DSQLITE_TEST=1\
-                $(THREADSAFE) $(TEMP_STORE)\
-                -o testfixture $(TESTSRC) $(TOP)/src/tclsqlite.c \
-		libtclsqlite3.la $(LIBTCL)
+testfixture$(TEXE):	$(TOP)/src/tclsqlite.c $(TESTSRC) $(COMMONOBJ)
+	$(LTLINK) -DTCLSH=1 -DSQLITE_TEST=1 $(THREADSAFE) $(TEMP_STORE)\
+	-o testfixture \
+	$(TESTSRC) $(TOP)/src/tclsqlite.c $(COMMONOBJ) $(LIBTCL)
 
-crashtest$(TEXE):	$(TOP)/src/tclsqlite.c libsqlite3.la $(TESTSRC) $(TOP)/src/os_test.c
+crashtest$(TEXE): $(TOP)/src/tclsqlite.c $(TOP)/src/os_test.c $(TESTSRC) $(COMMONOBJ)
 	$(LTLINK) -DOS_TEST=1 -DTCLSH=1 -DSQLITE_TEST=1 \
-		-o crashtest \
-		$(TESTSRC) $(TOP)/src/os_test.c $(TOP)/src/tclsqlite.c \
-		libsqlite3.la $(LIBTCL) $(THREADLIB)
+	-o crashtest $(TOP)/src/os_test.c \
+	$(TESTSRC) $(TOP)/src/tclsqlite.c $(COMMONOBJ) $(LIBTCL)
 
 
 
