diff -ru cxboard-0.14-orig/src/callbacks.c cxboard-0.14/src/callbacks.c
--- cxboard-0.14-orig/src/callbacks.c	2005-12-19 12:06:35.000000000 -0800
+++ cxboard-0.14/src/callbacks.c	2005-12-19 12:08:37.000000000 -0800
@@ -15,6 +15,8 @@
 #include "generator.h"
 #include <fcntl.h>
 
+#define clock cxboard_clock
+
 void
 on_new1_activate                       (GtkMenuItem     *menuitem,
                                         gpointer         user_data)
diff -ru cxboard-0.14-orig/src/elements.c cxboard-0.14/src/elements.c
--- cxboard-0.14-orig/src/elements.c	2005-12-19 12:06:35.000000000 -0800
+++ cxboard-0.14/src/elements.c	2005-12-19 12:08:20.000000000 -0800
@@ -14,6 +14,7 @@
 #include <errno.h>
 #include <fcntl.h>
 
+#define clock cxboard_clock
 
 /* used to mark cannons and pawns place */
 void board_pixmap_mark_point(GdkPixmap *pm, GdkGC *gc, int i, int j) {
diff -ru cxboard-0.14-orig/src/elements.h cxboard-0.14/src/elements.h
--- cxboard-0.14-orig/src/elements.h	2005-12-19 12:06:35.000000000 -0800
+++ cxboard-0.14/src/elements.h	2005-12-19 12:07:58.000000000 -0800
@@ -15,7 +15,7 @@
 int engine_level;
 
 int move_time;
-int clock; 
+int cxboard_clock; 
 void show_clock();
 void hide_clock();
 
diff -ru cxboard-0.14-orig/src/network.h cxboard-0.14/src/network.h
--- cxboard-0.14-orig/src/network.h	2005-12-19 12:06:35.000000000 -0800
+++ cxboard-0.14/src/network.h	2005-12-19 12:06:54.000000000 -0800
@@ -1,5 +1,6 @@
 #include <gtk/gtk.h>
 #include <arpa/inet.h>
+#include <stdint.h>
 
 #define DEFAULT_HOST "localhost"
 #define DEFAULT_PORT 22222
