diff -Naur teg-old/README teg-new/README
--- teg-old/README	2001-03-26 07:08:26.000000000 -0500
+++ teg-new/README	2005-06-21 14:50:15.000000000 -0400
@@ -1,5 +1,11 @@
-Tenes Emapandas Graciela (TEG) is a clone of 'Plan Táctico y Estratégico
+Tenes Emapandas Graciela (TEG) is a clone of 'Plan Tï¿½ctico y Estratï¿½gico
 de la Guerra', which is a pseudo-clone of Risk, a multiplayer turn-based
 strategy game. Some rules are different.
 
 http://teg.sourceforge.net
+
+It can be configured to support the GGZ Gaming Zone; see the GGZ
+directory.
+
+http://www.ggzgamingzone.org/games/teg/
+
diff -Naur teg-old/README.GGZ teg-new/README.GGZ
--- teg-old/README.GGZ	2002-04-07 09:50:08.000000000 -0400
+++ teg-new/README.GGZ	1969-12-31 19:00:00.000000000 -0500
@@ -1,46 +0,0 @@
-GGZ:
-----
-
-GGZ is GGZ GAMING ZONE
-http://ggz.sourceforge.net
-
-TEG, since version 0.6.0 (until v0.9.2) has support for GGZ v0.0.4
-TEG version 0.9.3 has support for GGZ v0.0.5
-
-By default, TEG is NOT compiled with ggz support, 
-You need to do this:
-	./configure --enable-ggz=yes
-
-`tegserver' and `tegclient' will be invoked by the ggz client
-and ggz server with `--ggz' command line option.
-
-So, the game will run in standard mode (standalone mode) by
-default. And if you want to run in ggz mode, just run the
-ggz client, that it will enable the ggz mode passing the
---ggz as a command line option.
-
-
-Installing the *.dsc files:
----------------------------
-
-In the ggz/ subdir, are two files:
-	tegclient.dsc	-> teg client dsc file
-	teg.dsc		-> teg server dsc file
-	teg.room	-> teg server room file
-
-You must have ggz installed and the run:
-	ggz-config --install --modfile=./ggz/tegclient.dsc	
-
-	cp ./ggz/teg.dsc `ggz-config -c`/ggzd/games
-	cp ./ggz/teg.room `ggz-config-c`/ggzd/rooms
-
-
-Running:
---------
-	You'll need to have ggz and teg installed.
-
-	run:
-		ggzd	(the ggz server)
-		ggz-gtk	(the ggz-gtk client)
-
-		log in, enter in the TEG room and launch teg.
diff -Naur teg-old/client/ggz_client.c teg-new/client/ggz_client.c
--- teg-old/client/ggz_client.c	2002-10-13 19:29:08.000000000 -0400
+++ teg-new/client/ggz_client.c	2005-06-21 14:55:09.000000000 -0400
@@ -72,7 +72,7 @@
 	while( g_fd == -1 ) {
 		ggzmod_dispatch( g_ggzmod );
 
-		if( ++i > 10 )
+		if( ++i > 10000000 )
 			break;
 	}
 
diff -Naur teg-old/client/gui-gnome/connect.c teg-new/client/gui-gnome/connect.c
--- teg-old/client/gui-gnome/connect.c	2002-10-14 17:04:36.000000000 -0400
+++ teg-new/client/gui-gnome/connect.c	2005-06-21 14:43:08.000000000 -0400
@@ -39,6 +39,7 @@
 #include "connect.h"
 #include "priv.h"
 #include "colors.h"
+#include "ggz_client.h"
 
 static GtkWidget *connect_window = NULL;
 static GtkWidget *con_entry_name=NULL;
@@ -88,7 +89,7 @@
 		}
 #ifdef WITH_GGZ
 		else {
-			gui_private.tag_ggz = gdk_input_add( ggz_client_get_fd(), GDK_INPUT_READ, (GdkInputFunction) ggz_client_handle(), (gpointer) NULL );
+			gui_private.tag_ggz = gdk_input_add( ggz_client_get_fd(), GDK_INPUT_READ, (GdkInputFunction) ggz_client_handle, (gpointer) NULL );
 		}
 #endif /* WITH_GGZ */
 		countries_redraw_all();
diff -Naur teg-old/client/gui-gnome/stock/stock-endturn.xpm teg-new/client/gui-gnome/stock/stock-endturn.xpm
--- teg-old/client/gui-gnome/stock/stock-endturn.xpm	2001-11-01 22:00:46.000000000 -0500
+++ teg-new/client/gui-gnome/stock/stock-endturn.xpm	2005-06-21 14:46:01.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * stock_endturn_xpm[] = {
+static const char * stock_endturn_xpm[] = {
 "24 24 150 2",
 "  	c None",
 ". 	c #010000",
diff -Naur teg-old/client/gui-gnome/stock/stock-getcard.xpm teg-new/client/gui-gnome/stock/stock-getcard.xpm
--- teg-old/client/gui-gnome/stock/stock-getcard.xpm	2000-08-06 20:37:12.000000000 -0400
+++ teg-new/client/gui-gnome/stock/stock-getcard.xpm	2005-06-21 14:46:01.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * stock_getcard_xpm[] = {
+static const char * stock_getcard_xpm[] = {
 "24 24 197 2",
 "  	c None",
 ". 	c #F1EAAB",
diff -Naur teg-old/client/gui-gnome/stock/stock-sendarmies.xpm teg-new/client/gui-gnome/stock/stock-sendarmies.xpm
--- teg-old/client/gui-gnome/stock/stock-sendarmies.xpm	2000-08-06 20:33:53.000000000 -0400
+++ teg-new/client/gui-gnome/stock/stock-sendarmies.xpm	2005-06-21 14:46:01.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * stock_sendarmies_xpm[] = {
+static const char * stock_sendarmies_xpm[] = {
 "24 24 9 1",
 " 	c None",
 ".	c #000000",
diff -Naur teg-old/client/gui-gnome/stock/stock-viewplayers.xpm teg-new/client/gui-gnome/stock/stock-viewplayers.xpm
--- teg-old/client/gui-gnome/stock/stock-viewplayers.xpm	2000-08-13 21:50:28.000000000 -0400
+++ teg-new/client/gui-gnome/stock/stock-viewplayers.xpm	2005-06-21 14:46:01.000000000 -0400
@@ -1,5 +1,5 @@
 /* XPM */
-static char * stock_viewplayers_xpm[] = {
+static const char * stock_viewplayers_xpm[] = {
 "24 24 9 1",
 " 	c None",
 ".	c #000000",
diff -Naur teg-old/client/misc.c teg-new/client/misc.c
--- teg-old/client/misc.c	2002-10-13 19:29:08.000000000 -0400
+++ teg-new/client/misc.c	2005-06-21 14:46:36.000000000 -0400
@@ -28,6 +28,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <dirent.h>
+#include <string.h>
 
 #include <glib.h>
 
diff -Naur teg-old/common/common.c teg-new/common/common.c
--- teg-old/common/common.c	2002-10-13 23:26:36.000000000 -0400
+++ teg-new/common/common.c	2005-06-21 14:46:59.000000000 -0400
@@ -34,6 +34,7 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <assert.h>
+#include <string.h>
 
 char *g_colores[] = {
 	N_("red"),
diff -Naur teg-old/common/stats.c teg-new/common/stats.c
--- teg-old/common/stats.c	2002-09-18 22:32:15.000000000 -0400
+++ teg-new/common/stats.c	2005-06-21 14:47:17.000000000 -0400
@@ -26,6 +26,8 @@
 #include "stats.h"
 #include "common.h"
 
+#include <string.h>
+
 TEG_STATUS stats_init( PPLAYER_STATS s )
 {
 	memset( s, 0, sizeof(*s) );
diff -Naur teg-old/ggz/README teg-new/ggz/README
--- teg-old/ggz/README	2002-04-06 19:26:01.000000000 -0500
+++ teg-new/ggz/README	2005-06-21 14:48:51.000000000 -0400
@@ -1,29 +1,31 @@
 Making TEG work with GGZ
 ------------------------
 
-since TEG v0.9.3 works with GGZ v0.0.5 (the actual version is 0.0.5pre)
-
-
-from TEG v0.6.0 until v0.9.2, TEG works with GGZ v0.0.4
-
-
 Download GGZ from:
-	http://ggz.sourceforge.net
+	http://www.ggzgamingzone.org
 
+Supported features:
+	(x) game spectators
+	(x) statistics
+	(x) bots
 
 Installing TEG with GGZ
 -----------------------
 
-As always:
-
-   ./configure --enable-ggz=yes
-   make
-   su -
-   make install
-
+First, you"ll have to switch on ggz mode.
+This needs ggzmod, ggzdmod, ggz-config and ggzd installed.
+  
+    ./configure --enable-ggz=yes
+
+Then, install as always:
+
+    make
+    su -
+    make install
+  
+History
+-------
 
-   Then:
-    cd ggz
-    ggz-config --install --modfile=tegclient.dsc
-    cp teg.dsc /usr/local/etc/ggzd/games
-    cp teg.room /usr/local/etc/ggzd/rooms
+since TEG 0.11+ works with GGZ v0.0.9 including statistics
+since TEG v0.9.3 works with GGZ v0.0.5 (the actual version is 0.0.5pre)
+from TEG v0.6.0 until v0.9.2, TEG works with GGZ v0.0.4
diff -Naur teg-old/ggz/teg.dsc.in teg-new/ggz/teg.dsc.in
--- teg-old/ggz/teg.dsc.in	2002-10-13 23:25:57.000000000 -0400
+++ teg-new/ggz/teg.dsc.in	2005-06-21 14:55:22.000000000 -0400
@@ -10,10 +10,14 @@
 
 [Protocol]
 Engine = TEG
-Version = 6
+Version = 7
 
 [TableOptions]
 AllowLeave = 1
 BotsAllowed =  1..6
 PlayersAllowed = 2..6
 AllowSpectators = 1
+
+[Statistics]
+Records = 1
+Highscores = 1
diff -Naur teg-old/ggz/tegclient.dsc.in teg-new/ggz/tegclient.dsc.in
--- teg-old/ggz/tegclient.dsc.in	2002-04-08 21:18:26.000000000 -0400
+++ teg-new/ggz/tegclient.dsc.in	2005-06-21 14:49:13.000000000 -0400
@@ -7,5 +7,5 @@
 Homepage = http://teg.sourceforge.net
 Name = teg
 ProtocolEngine = TEG
-ProtocolVersion = 6
-Version = 6
+ProtocolVersion = 7
+Version = @VERSION@
diff -Naur teg-old/server/ggz_server.c teg-new/server/ggz_server.c
--- teg-old/server/ggz_server.c	2002-10-13 23:26:36.000000000 -0400
+++ teg-new/server/ggz_server.c	2005-06-21 14:52:09.000000000 -0400
@@ -34,6 +34,7 @@
 #include <errno.h>
 
 #include "server.h"
+#include "player.h"
 
 #ifdef WITH_GGZ
 
@@ -242,6 +243,39 @@
 	return TEG_STATUS_ERROR;
 }
 
+void ggz_server_gameover( int fd )
+{
+      int max, i;
+      GGZSeat seat;
+      GGZGameResult results[TEG_MAX_PLAYERS];
+      int scores[TEG_MAX_PLAYERS];
+      PLIST_ENTRY pL;
+      PSPLAYER pJ;
+
+      max = ggzdmod_get_num_seats( g_ggzdmod );
+
+      for(i=0;i<max;i++) {
+              seat = ggzdmod_get_seat( g_ggzdmod, i );
+              if( seat.fd == fd ) {
+                      results[i] = GGZ_GAME_WIN;
+              } else {
+                      results[i] = GGZ_GAME_LOSS;
+              }
+
+              scores[i] = 0;
+              pL = g_list_player.Flink;
+              while( !IsListEmpty( &g_list_player ) && (pL != &g_list_player) ) {
+                      pJ = (PSPLAYER) pL;
+                      if( pJ->fd == seat.fd ){
+                              scores[i] = pJ->player_stats.score;
+                      }
+                      pL = LIST_NEXT(pL);
+              }
+      }
+
+      ggzdmod_report_game( g_ggzdmod, NULL, results, scores );
+}
+
 int ggz_server_main_loop()
 {
 	if( teg_ggz_init() == -1 )
diff -Naur teg-old/server/ggz_server.h teg-new/server/ggz_server.h
--- teg-old/server/ggz_server.h	2002-10-13 21:25:04.000000000 -0400
+++ teg-new/server/ggz_server.h	2005-06-21 14:52:34.000000000 -0400
@@ -27,5 +27,6 @@
 
 int ggz_server_main_loop();
 TEG_STATUS ggz_server_find_ggzname( int fd, char *n, int len );
+void ggz_server_gameover( int fd );
 
 #endif /* __TEGS_TEGGGZ_H */
diff -Naur teg-old/server/helper.c teg-new/server/helper.c
--- teg-old/server/helper.c	2002-10-04 18:56:35.000000000 -0400
+++ teg-new/server/helper.c	2005-06-21 14:56:16.000000000 -0400
@@ -554,9 +554,12 @@
 
 	// sockets[0] is the client side of the robot
 	// sockets[1] is the server side
-	int r = socketpair(AF_LOCAL, SOCK_STREAM, 0, sockets );
-	if( r != 0 )
-		return TEG_STATUS_ERROR;
+    sockets[0] = 0;
+	while(sockets[0] < 3) {
+		int r = socketpair(AF_LOCAL, SOCK_STREAM, 0, sockets );
+		if( r != 0 )
+			return TEG_STATUS_ERROR;
+	}
 
 	// launch the server connected
 	if ( (pid = fork()) < 0) {
diff -Naur teg-old/server/main.c teg-new/server/main.c
--- teg-old/server/main.c	2002-10-13 21:25:04.000000000 -0400
+++ teg-new/server/main.c	2005-06-21 14:52:48.000000000 -0400
@@ -32,6 +32,7 @@
 #include <unistd.h>
 #include <netinet/in.h>
 #include <errno.h>
+#include <string.h>
 
 #include "server.h"
 #include "scores.h"
diff -Naur teg-old/server/metaserver.c teg-new/server/metaserver.c
--- teg-old/server/metaserver.c	2002-11-14 06:29:33.000000000 -0500
+++ teg-new/server/metaserver.c	2005-06-21 14:53:36.000000000 -0400
@@ -62,9 +62,12 @@
 #define	NOPTIONS  (sizeof(meta_options)/sizeof(meta_options[0]))
 
 /* threaded funcions ! */
-void * thread_meta_on( char * comment )
+void * thread_meta_on( char * commentptr )
 {
 	int s;
+	char *comment;
+	
+	comment = (char*)commentptr;
 
 	if( ! g_just_once ) {
 		con_text_out_wop(M_INF,_("Adding %s:%d to metaserver %s:%d...\n"),
diff -Naur teg-old/server/missions.c teg-new/server/missions.c
--- teg-old/server/missions.c	2002-10-04 18:56:35.000000000 -0400
+++ teg-new/server/missions.c	2005-06-21 14:53:57.000000000 -0400
@@ -20,6 +20,7 @@
  */
 /* checks that the secret missions are accomplished */
 #include <assert.h>
+#include <string.h>
 #include "server.h"
 
 /* Do player pJ accomplished his secret mission ? */
diff -Naur teg-old/server/play.c teg-new/server/play.c
--- teg-old/server/play.c	2002-10-13 21:25:04.000000000 -0400
+++ teg-new/server/play.c	2005-06-21 14:54:37.000000000 -0400
@@ -693,7 +693,9 @@
 
 		/* Did 'src' player win the game ? */
 		if( mission_chequear( pJ_src ) == TEG_STATUS_GAMEOVER || game_is_finished() ) {
-			con_text_out(M_INF,_("Player %s(%d) is the winner! Game Over\n"),pJ_src->name,pJ_src->numjug);
+#ifdef WITH_GGZ
+                ggz_server_gameover(pJ_src->fd);
+#endif			con_text_out(M_INF,_("Player %s(%d) is the winner! Game Over\n"),pJ_src->name,pJ_src->numjug);
 			pJ_src->estado = PLAYER_STATUS_GAMEOVER;
 			game_end( pJ_src );
 			return TEG_STATUS_SUCCESS;
@@ -820,7 +822,7 @@
 	tarjeta_sacar( &pP->tarjeta, pJ->numjug );
 
 	/*
-	 * Me fijo si el player es dueño del país que dice la tarjeta. Si es así
+	 * Me fijo si el player es dueï¿½o del paï¿½s que dice la tarjeta. Si es asï¿½
 	 * le agrego 2 fichas automaticamente como dice el reglamento.
 	 */
 	if( pP->numjug == pJ->numjug ) {
