diff -ru trackballs-1.1.1-orig/share/Makefile.am trackballs-1.1.1/share/Makefile.am
--- trackballs-1.1.1-orig/share/Makefile.am	2005-12-19 17:42:32.000000000 -0800
+++ trackballs-1.1.1/share/Makefile.am	2005-12-19 17:42:47.000000000 -0800
@@ -4,6 +4,7 @@
 EXTRA_DIST = $(pkgdata_DATA) trackballs.6
 
 install-pkgdataDATA:
-	-install -D -m664 -g games highScores		$(DESTDIR)@with_highscores@
-	-chgrp -R games			$(DESTDIR)$(pkgdatadir)/levels
+	mkdir -p $(DESTDIR)/share/trackballs
+	install -m664 -g games highScores		$(DESTDIR)@with_highscores@
+	chgrp -R games			$(DESTDIR)$(pkgdatadir)/levels
 	chmod -R g+r			$(DESTDIR)$(pkgdatadir)/levels
diff -ru trackballs-1.1.1-orig/src/Makefile.in trackballs-1.1.1/src/Makefile.in
--- trackballs-1.1.1-orig/src/Makefile.in	2005-12-19 17:42:32.000000000 -0800
+++ trackballs-1.1.1/src/Makefile.in	2005-12-19 17:42:47.000000000 -0800
@@ -366,7 +366,7 @@
 	-test -z "$(bin_PROGRAMS)" || rm -f $(bin_PROGRAMS)
 trackballs$(EXEEXT): $(trackballs_OBJECTS) $(trackballs_DEPENDENCIES) 
 	@rm -f trackballs$(EXEEXT)
-	$(CXXLINK) $(trackballs_LDFLAGS) $(trackballs_OBJECTS) $(trackballs_LDADD) $(LIBS)
+	$(CXXLINK) $(trackballs_LDFLAGS) $(trackballs_OBJECTS) $(trackballs_LDADD) $(LIBS) $(LTLIBINTL)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
Only in trackballs-1.1.1/src: Makefile.in.orig
