diff -Naur --exclude='*~' bbdb-2.35/Makefile bbdb-2.35.fink/Makefile
--- bbdb-2.35/Makefile	2005-08-20 05:45:36.000000000 -0500
+++ bbdb-2.35.fink/Makefile	2007-12-26 13:12:59.000000000 -0600
@@ -8,7 +8,9 @@
 LN_S = ln -s
 
 ETAGS = etags
-EMACS_PROG = emacs -no-site-file -no-init-file
+EMACS = emacs 
+EFLAGS = -no-site-file -no-init-file
+EMACS_PROG = ${EMACS} ${EFLAGS}
 
 SUBDIRS = lisp texinfo tex utils
 
diff -Naur --exclude='*~' bbdb-2.35/fink/50bbdb.el bbdb-2.35.fink/fink/50bbdb.el
--- bbdb-2.35/fink/50bbdb.el	1969-12-31 18:00:00.000000000 -0600
+++ bbdb-2.35.fink/fink/50bbdb.el	2007-12-26 13:14:16.000000000 -0600
@@ -0,0 +1 @@
+(require 'bbdb)
diff -Naur --exclude='*~' bbdb-2.35/fink/README.fink bbdb-2.35.fink/fink/README.fink
--- bbdb-2.35/fink/README.fink	1969-12-31 18:00:00.000000000 -0600
+++ bbdb-2.35.fink/fink/README.fink	2007-12-26 13:26:10.000000000 -0600
@@ -0,0 +1,35 @@
+Enable the bbdb by adding the form
+
+  (require 'bbdb)
+
+to your emacs initialization file.  To initialize the bbdb to be used
+with the mail composition, mail and news readers you use (i.e., to
+establish keybindings for various mail and news packages), add a call
+to the function `bbdb-initialize' with various arguments.  For
+example, to initialize the bbdb for use with the message composition
+package (which gnus uses) and gnus, add the form
+
+  (bbdb-initialize 'message 'gnus)
+
+to your .emacs file (or whatever file you use to store your
+customizations).  See the info docs for more information on how to
+call `bbdb-initialize' for your particular needs.
+
+BBDB is compiled only for rmail and gnus, which are standard parts of
+emacs.  In addition to these two mailers, BBDB supports mhe and vm.
+However, as of December 15, 2005, fink lacks these emacs packages.
+However, when fink gets either, compiled support for the BBDB can be
+had by simply invoking the command 
+
+fink reinstall bbdb
+
+Until then, the source code for the mhe and vm support for BBDB can be
+found in @PREFIX@/share/emacs/site-lisp/emacs/bbdb.
+
+The customization group for BBDB is simply `bbdb'.  Thus, to begin
+customization, execute the key sequence
+
+M-x customize-group RET bbdb RET
+
+XEmacs users: bbdb is already available within xemacs itself through
+xemacs's own package management system.
diff -Naur --exclude='*~' bbdb-2.35/fink/bbdb.copyright bbdb-2.35.fink/fink/bbdb.copyright
--- bbdb-2.35/fink/bbdb.copyright	1969-12-31 18:00:00.000000000 -0600
+++ bbdb-2.35.fink/fink/bbdb.copyright	2007-12-26 13:26:23.000000000 -0600
@@ -0,0 +1,23 @@
+This package was finkified by Jesse Alama <alama@stanford.edu>
+
+It was downloaded from http://bbdb.sourceforge.net/
+
+Copyright:
+
+  The Insidious Big Brother Database is free software; you can redistribute
+  it and/or modify it under the terms of the GNU General Public License as
+  published by the Free Software Foundation; either version 2, or (at your
+  option) any later version.
+  
+  BBDB is distributed in the hope that it will be useful, but WITHOUT ANY
+  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+  FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+  details.
+  
+  You should have received a copy of the GNU General Public License
+  along with GNU Emacs; see the file COPYING.  If not, write to
+  the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+  Boston, MA 02111-1307, USA.
+
+The complete text of the GNU General Public License can be found in
+`@PREFIX@/share/doc/fink/COPYING'.
diff -Naur --exclude='*~' bbdb-2.35/fink/emacsen-install bbdb-2.35.fink/fink/emacsen-install
--- bbdb-2.35/fink/emacsen-install	1969-12-31 18:00:00.000000000 -0600
+++ bbdb-2.35.fink/fink/emacsen-install	2007-12-26 13:25:41.000000000 -0600
@@ -0,0 +1,44 @@
+#!/bin/bash -e
+#
+# install file for the fink bbdb emacs package.
+
+set -o posix
+
+FLAVOR=${1}
+
+echo >&2 "install/bbdb: Handling install of emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "install/bbdb: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+if [ ${FLAVOR} == emacs ]
+then
+    # Nothing to do, nothing to print
+    exit 0
+fi
+
+if [ ${FLAVOR} == xemacs ]
+then
+    EFLAGS="-no-site-file -no-init-file -batch -f batch-byte-compile"
+else
+    EFLAGS="--no-site-file --no-init-file -batch -f batch-byte-compile"
+fi
+
+echo >&2 -n "install/bbdb: Byte-compiling for ${FLAVOR}..."
+
+mkdir -p @PREFIX@/share/${FLAVOR}/site-lisp/bbdb;
+for i in bbdb-autoloads.el bbdb-com.el bbdb-ftp.el bbdb-gnus.el bbdb-gui.el bbdb-hooks.el bbdb-merge.el bbdb-mhe.el bbdb-migrate.el bbdb-print.el bbdb-reportmail.el bbdb-rmail.el bbdb-sc.el bbdb-snarf.el bbdb-srv.el bbdb-vm.el bbdb-w3.el bbdb-whois.el bbdb-xemacs.el bbdb.el; do
+    ln -fs @PREFIX@/share/emacs/site-lisp/bbdb/${i} @PREFIX@/share/${FLAVOR}/site-lisp/bbdb/${i};
+done
+
+cd @PREFIX@/share/${FLAVOR}/site-lisp/bbdb;
+
+(${FLAVOR} ${EFLAGS} *.el 2>&1) | gzip -9qf > @PREFIX@/share/doc/bbdb/CompilationLog-${FLAVOR}.gz
+
+echo >&2 "done."
+echo >&2 "install/bbdb: Compilation log saved in @PREFIX@/share/doc/bbdb/CompilationLog-${FLAVOR}.gz."
+
+exit 0
diff -Naur --exclude='*~' bbdb-2.35/fink/emacsen-remove bbdb-2.35.fink/fink/emacsen-remove
--- bbdb-2.35/fink/emacsen-remove	1969-12-31 18:00:00.000000000 -0600
+++ bbdb-2.35.fink/fink/emacsen-remove	2007-12-26 13:25:50.000000000 -0600
@@ -0,0 +1,21 @@
+#!/bin/bash -e
+#
+# This is the Fink version of the bbdb emacs package.
+set -o posix
+
+FLAVOR=${1}
+
+echo "remove/bbdb: Handling removal for emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "remove/bbdb: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+echo >&2 -n "remove/bbdb: Purging compilation log and byte-compiled files for ${FLAVOR}..."
+rm -f @PREFIX@/share/doc/bbdb/CompilationLog-${FLAVOR}.gz
+rm -Rf @PREFIX@/share/${FLAVOR}/site-lisp/bbdb
+echo >&2 "done."
+    
+exit 0
