diff -Naur lha-114i/LICENSE lha-114i.fink/LICENSE
--- lha-114i/LICENSE	Thu Jan  1 09:00:00 1970
+++ lha-114i.fink/LICENSE	Sun Jun 27 21:19:26 2004
@@ -0,0 +1,27 @@
+This license explanation was derived from the copyright notice distributed 
+with the Debian package maintained by Atsushi KAMOSHIDA <kamop@post1.com>.
+
+Upstream Author: Tsugio Okamoto <tsugio@muc.biglobe.ne.jp>
+
+Copyright:
+
+ Original License Statement (in Japanese EUC-JP):
+
+   なお、ネットでの配付は自由ですが、ネットにアクセスできない方（雑誌および、
+   CD-ROM などによる）への配付は、配付前にこちらに Inter-Net の方にE-Mail
+   を お願いします。 どこそこに、掲載するという旨があれば結構ですので。
+   また、それも出来ない際には、後日 E-Mail を必ずお願いします。
+
+ Translated License Statement (translated by GOTO Masanori <gotom@debian.org>):
+
+   It's free to distribute on the network, but if you distribute for
+   the people who cannot access the network (by magazine or CD-ROM),
+   please send E-Mail (Inter-Net address) to the author before the
+   distribution. That's well where this software is appeard.
+   If you cannot do, you must send me the E-Mail later.
+
+ Original Source Code License Statement:
+
+   /*    Copyright (C) MCMLXXXIX Yooichi.Tagawa                      */
+   /*    Modified                Nobutaka Watazaki                   */
+   /*                   Thanks to H.Yoshizaki. (MS-DOS LHarc)        */
\ No newline at end of file
diff -Naur lha-114i/Makefile lha-114i.fink/Makefile
--- lha-114i/Makefile	Fri Dec 15 21:13:16 2000
+++ lha-114i.fink/Makefile	Sun Jun 27 21:24:02 2004
@@ -20,7 +20,7 @@
 
 #MACHINE     = -DSYSTIME_HAS_NO_TM -DFTIME -DEUC
 #MACHINE     = -DSYSTIME_HAS_NO_TM -DEUC -DHAVE_TM_ZONE -DSYSV_SYSTEM_DIR
-MACHINE     = -DSYSTIME_HAS_NO_TM -DEUC -DSYSV_SYSTEM_DIR -DMKTIME
+MACHINE     = -DSYSTIME_HAS_NO_TM -DEUC -DSYSV_SYSTEM_DIR -DMKTIME -DHAVE_NO_LCHOWN
 
 #OPTIMIZE	= -O2 -fstrength-reduce -fomit-frame-pointer
 OPTIMIZE	= -O2 -DSUPPORT_LH7 -DMKSTEMP
diff -Naur lha-114i/README lha-114i.fink/README
--- lha-114i/README	Thu Jan  1 09:00:00 1970
+++ lha-114i.fink/README	Sun Jun 27 21:28:51 2004
@@ -0,0 +1,22 @@
+LHarc    for UNIX  V 1.02  Copyright(C) 1989  Y.Tagawa
+LHx      for MSDOS V C2.01 Copyright(C) 1990  H.Yoshizaki
+LHx(arc) for OSK   V 2.01  Modified     1990  Momozou
+LHa      for UNIX  V 1.00  Copyright(C) 1992  Masaru Oki
+LHa      for UNIX  V 1.14  Modified     1995  Nobutaka Watazaki
+LHa      for UNIX  V 1.14i Modified     2000  Tsugio Okamoto
+usage: lha [-]{axelvudmcp[q[num]][vnfodizg012]}[w=<dir>] archive_file [file...]
+commands:                           options:
+ a   Add(or replace) to archive      q{num} quiet (num:quiet mode)
+ x,e EXtract from archive            v  verbose
+ l,v List / Verbose List             n  not execute
+ u   Update newer files to archive   f  force (over write at extract)
+ d   Delete from archive             t  FILES are TEXT file
+ m   Move to archive (means 'ad')    o[567] compression method (a/u)
+ c   re-Construct new archive        w=<dir> specify extract directory (a/u/m/x/e)
+ p   Print to STDOUT from archive    d  delete FILES after (a/u/c)
+ t   Test file CRC in archive        i  ignore directory path (x/e)
+                                     z  files not compress (a/u)
+                                     g  Generic format (for compatibility)
+                                        or not convert case when extracting
+                                     0/1/2 header level (a/u)
+                                     e  TEXT code convert from/to EUC
