--- pcal-4.11.0/Makefile.orig	2009-04-03 21:20:57.000000000 -0300
+++ pcal-4.11.0/Makefile	2009-04-04 09:01:00.000000000 -0300
@@ -138,19 +138,7 @@
 EXECDIR	= exec
 DOCDIR	= doc
 
-# 
-# Compiling for DOS+DJGPP requires different directories for the installed
-# executable and the 'man'/'cat' pages.  Unix uses the values shown below.
-# 
-ifeq ($(OS),DJGPP)   # DOS+DJGPP
-	BINDIR = $(DJDIR)/bin
-	MANDIR = $(DJDIR)/man/man1
-	CATDIR = $(DJDIR)/man/cat1
-else   # Unix
-	BINDIR = /usr/local/bin
-	MANDIR = /usr/man/man1
-	CATDIR = /usr/man/cat1
-endif
+BINDIR=$(DESTDIR)/bin
 
 OBJECTS = $(OBJDIR)/pcal.o \
 		$(OBJDIR)/encvec.o $(OBJDIR)/exprpars.o \
@@ -342,17 +330,12 @@
 fresh:	clobber $(PCAL)
 
 man:	$(DOCDIR)/pcal.man
-	nroff -man $(DOCDIR)/pcal.man > $(DOCDIR)/pcal.cat
 	groff -man -Tps $(DOCDIR)/pcal.man >$(DOCDIR)/pcal-help.ps
 	groff -man -Thtml $(DOCDIR)/pcal.man >$(DOCDIR)/pcal-help.html
 	groff -man -Tascii $(DOCDIR)/pcal.man >$(DOCDIR)/pcal-help.txt
 
 install:	$(EXECDIR)/$(PCAL) man
-	mkdir -p $(DESTDIR)/$(BINDIR)
-	mkdir -p $(DESTDIR)/$(MANDIR)
-	mkdir -p $(DESTDIR)/$(CATDIR)
-	cp $(EXECDIR)/$(PCAL) $(DESTDIR)/$(BINDIR)
-	cp $(DOCDIR)/pcal.man $(DESTDIR)/$(MANDIR)/pcal.1
-	$(PACK) $(DESTDIR)/$(MANDIR)/pcal.1
-	cp $(DOCDIR)/pcal.cat $(DESTDIR)/$(CATDIR)/pcal.1
-	$(PACK) $(DESTDIR)/$(CATDIR)/pcal.1
+	mkdir -p $(BINDIR)
+	mkdir -p $(INSTALLMAN1DIR)
+	cp $(EXECDIR)/$(PCAL) $(BINDIR)
+	cp $(DOCDIR)/pcal.man $(INSTALLMAN1DIR)/pcal.1
