diff -Naur pkg-order-1.12/Makefile.PL pkg-order-new/Makefile.PL
--- pkg-order-1.12/Makefile.PL	Mon Feb 28 23:58:59 2000
+++ pkg-order-new/Makefile.PL	Tue Apr 26 04:21:16 2005
@@ -19,7 +19,7 @@
 WriteMakefile(
 	      'NAME'         => 'Debian::Package',
 	      'DISTNAME'     => 'pkg-order',
-	      'PMLIBDIRS'    => ['Package/'],
+	      'PMLIBDIRS'    => ['Package'],
 	      'EXE_FILES'    => ['pkg-order', 'pkg-deptree', 'pkg-nodep', 'pkg-revdeps', 'pkg-unused-libs'],
 	      'VERSION_FROM' => "patchlevel",
 	      linkext        => { LINKTYPE => '' },
diff -Naur pkg-order-1.12/pkg-unused-libs pkg-order-new/pkg-unused-libs
--- pkg-order-1.12/pkg-unused-libs	Mon Feb 28 23:58:59 2000
+++ pkg-order-new/pkg-unused-libs	Tue Apr 26 04:12:12 2005
@@ -68,7 +68,7 @@
 other packages that depend on them (optionally also checks Recommends
 and Suggestions).
 
-A package is considered to be a library if it provides a I<PKG}.shlibs file in
+A package is considered to be a library if it provides a I<PKG>.shlibs file in
 I</var/lib/dpkg/info>. Sometimes packages with an shlibs file provide the
 libraries only as a side effect and do somthing really different otherwise
 (e.g., apt and e2fsprogs). Therefore, packages that contain files in
