--- rdiff-backup-1.2.8/setup.py.orig	2009-03-16 15:36:21.000000000 +0100
+++ rdiff-backup-1.2.8/setup.py	2009-10-28 15:58:06.000000000 +0100
@@ -73,9 +73,6 @@
 							   libraries=libname,
 							   extra_link_args=lflags_arg)],
 	  scripts = ['rdiff-backup', 'rdiff-backup-statistics'],
-	  data_files = [('share/man/man1', ['rdiff-backup.1',
-										'rdiff-backup-statistics.1']),
-					('share/doc/rdiff-backup-%s' % (version_string,),
-					 ['CHANGELOG', 'COPYING', 'README', 'FAQ.html'])],
+	  data_files = [('share/man/man1', ['rdiff-backup.1','rdiff-backup-statistics.1'])],
 					**extra_options)
 
