diff -urN shlist-1.0/README shlist-1.0.new/README
--- shlist-1.0/README	Tue May 13 11:47:56 2003
+++ shlist-1.0.new/README	Sun Feb 13 19:04:15 2005
@@ -127,7 +127,7 @@
 
     Example using ls:
     
-    $ ls -l --color='no' / | tail -n+2 | shlist -s -n rootdir -c 7 -
+    $ ls -l --color='no' / | tail -n+2 | shlist.sh -s -n rootdir -c 7 -
     $ shlist -n rootdir -l
     bin     "drwxr-xr-x    2 root     root         2304 2003-04-16 21:11 bin"
     dev     "drwxr-xr-x   28 root     root        94840 2003-05-10 22:48 dev"
@@ -152,7 +152,7 @@
     
     Example using df:
     
-    $ df | tail -n+2 | ./shlist -n df -c5 -s
+    $ df | tail -n+2 | ./shlist.sh -n df -c5 -s
     $ shlist -n df -l
     |       "/dev/hda3              4457860   2999612   1458248  68% /"
     |boot   "/dev/hda5                15522      4624     10097  32% /boot"
diff -urN shlist-1.0/shlist.sh shlist-1.0.new/shlist.sh
--- shlist-1.0/shlist.sh	Sun May 11 22:57:34 2003
+++ shlist-1.0.new/shlist.sh	Sun Feb 13 19:04:37 2005
@@ -1,4 +1,4 @@
-#
+#!/bin/sh
 # Wrapper for shlist to process stdoutput line by line.
 # Paramters are passed to shlist as-is and the input line is 
 # appended.
@@ -10,5 +10,5 @@
     then
         exit
     fi
-    shlist $* "$inline" > ofile
+    shlist $* "$inline"
 done
