diff -Naur lsdb-0.11.orig/fink/lsdb-install lsdb-0.11/fink/lsdb-install
--- lsdb-0.11.orig/fink/lsdb-install	Thu Jan  1 09:00:00 1970
+++ lsdb-0.11/fink/lsdb-install	Sun Apr  3 05:12:20 2005
@@ -0,0 +1,51 @@
+#! /bin/sh -e
+# @PREFIX@/lib/emacsen-common/packages/install/lsdb
+
+# Written by Jim Van Zandt <jrv@vanzandt.mv.com>, borrowing heavily
+# from the install scripts for gettext by Santiago Vila
+# <sanvila@ctv.es> and octave by Dirk Eddelbuettel <edd@debian.org>.
+
+FLAVOR=$1
+PACKAGE=lsdb
+
+if [ ${FLAVOR} = emacs ]; then exit 0; fi
+
+echo install/${PACKAGE}: Handling install for emacsen flavor ${FLAVOR}
+
+FLAVORTEST=`echo $FLAVOR | cut -c-6`
+if [ ${FLAVORTEST} = xemacs ] ; then
+    SITEFLAG="-no-site-file"
+else
+    SITEFLAG="--no-site-file"
+fi
+FLAGS="${SITEFLAG} -q -batch -l path.el -l LSDB-MK -f compile-lsdb"
+
+LOG=`tempfile -pelc_ -s.log -m644`
+ELDIR=@PREFIX@/share/emacs/site-lisp/${PACKAGE}
+ELCDIR=@PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}
+
+# Install-info-altdir does not actually exist. 
+# Maybe somebody will write it.
+# if test -x @PREFIX@/sbin/install-info-altdir; then
+#     echo install/${PACKAGE}: install Info links for ${FLAVOR}
+#     install-info-altdir --quiet --section "" "" --dirname=${FLAVOR} @PREFIX@/info/${PACKAGE}.info.gz
+# fi
+
+install -m 755 -d ${ELCDIR}
+cd ${ELDIR}
+FILES=`echo *.el`
+cp ${FILES} ${ELCDIR}
+FILES=`echo LSDB-*`
+cp ${FILES} ${ELCDIR}
+cd ${ELCDIR}
+
+cat << EOF > path.el
+(setq load-path (cons "@PREFIX@/share/${FLAVOR}/site-lisp/apel" load-path))
+(setq load-path (cons "@PREFIX@/share/${FLAVOR}/site-lisp/flim" load-path))
+(setq load-path (cons "." load-path) byte-compile-warnings nil)
+EOF
+echo ${FLAVOR} ${FLAGS} >> ${LOG}
+"${FLAVOR}" ${FLAGS} >> ${LOG} 2>&1
+rm -f *.el LSDB-*
+
+exit 0
diff -Naur lsdb-0.11.orig/fink/lsdb-remove lsdb-0.11/fink/lsdb-remove
--- lsdb-0.11.orig/fink/lsdb-remove	Thu Jan  1 09:00:00 1970
+++ lsdb-0.11/fink/lsdb-remove	Sun Apr  3 05:12:40 2005
@@ -0,0 +1,15 @@
+#!/bin/sh -e
+# @PREFIX@/lib/emacsen-common/packages/remove/lsdb
+
+FLAVOR=$1
+PACKAGE=lsdb
+
+if [ ${FLAVOR} != emacs ]; then
+    # if test -x @PREFIX@/sbin/install-info-altdir; then
+    #     echo remove/${PACKAGE}: removing Info links for ${FLAVOR}
+    #     install-info-altdir --quiet --remove --dirname=${FLAVOR} @PREFIX@/info/lsdb.info.gz
+    # fi
+
+    echo remove/${PACKAGE}: purging byte-compiled files for ${FLAVOR}
+    rm -rf @PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}
+fi
diff -Naur lsdb-0.11.orig/fink/lsdb-startup lsdb-0.11/fink/lsdb-startup
--- lsdb-0.11.orig/fink/lsdb-startup	Thu Jan  1 09:00:00 1970
+++ lsdb-0.11/fink/lsdb-startup	Sun Apr  3 05:13:20 2005
@@ -0,0 +1,51 @@
+;; -*-emacs-lisp-*-
+;;
+;; Emacs startup file for Fink lsdb package
+;;
+;; Originally contributed by Nils Naumann <naumann@unileoben.ac.at>
+;; Modified by Dirk Eddelbuettel <edd@debian.org>
+;; Adapted for dh-make by Jim Van Zandt <jrv@vanzandt.mv.com>
+
+;; The lsdb package follows the Debian/GNU Linux 'emacsen' policy and
+;; byte-compiles its elisp files for each 'emacs flavor' (emacs19,
+;; xemacs19, emacs20, xemacs20...).  The compiled code is then
+;; installed in a subdirectory of the respective site-lisp directory.
+;; We have to add this to the load-path:
+
+(if (not (file-exists-p "@PREFIX@/share/emacs/site-lisp/lsdb"))
+    (message "Package lsdb removed but not purged.  Skipping setup.")
+  (setq load-path (cons (concat "@PREFIX@/share/"
+				(symbol-name fink-emacs-flavor)
+				"/site-lisp/lsdb") load-path))
+
+  ;; for t-gnus
+  (autoload 'lsdb-gnus-insinuate "lsdb")
+  (autoload 'lsdb-gnus-insinuate-message "lsdb")
+  (add-hook 'gnus-startup-hook 'lsdb-gnus-insinuate)
+  (add-hook 'message-setup-hook
+	    (lambda ()
+	      (define-key message-mode-map "\M-\t" 'lsdb-complete-name)))
+  (add-hook 'gnus-summary-mode-hook
+	    (lambda ()
+	      (define-key gnus-summary-mode-map ":" 'lsdb-toggle-buffer)))
+
+  ;; for wl
+  (autoload 'lsdb-wl-insinuate "lsdb")
+  (add-hook 'wl-init-hook 'lsdb-wl-insinuate)
+  (add-hook 'wl-draft-mode-hook
+	    (lambda ()
+	      (define-key wl-draft-mode-map "\M-\t" 'lsdb-complete-name)))
+  (add-hook 'wl-summary-mode-hook
+	    (lambda ()
+	      (define-key wl-summary-mode-map ":" 'lsdb-toggle-buffer)))
+
+  ;; for mew
+  (autoload 'lsdb-mew-insinuate "lsdb")
+  (add-hook 'mew-init-hook 'lsdb-mew-insinuate)
+  (add-hook 'mew-draft-mode-hook
+	    (lambda ()
+	      (define-key mew-draft-header-map "\M-I" 'lsdb-complete-name)))
+  (add-hook 'mew-summary-mode-hook
+	    (lambda ()
+	      (define-key mew-summary-mode-map "l" 'lsdb-toggle-buffer)))
+  )
