--- freetype-2.1.3/builds/unix/freetype-config.in~	Tue Dec 16 18:58:35 2003
+++ freetype-2.1.3/builds/unix/freetype-config.in	Tue Dec 16 19:00:47 2003
@@ -1,6 +1,6 @@
 #! /bin/sh
 
-prefix=@prefix@
+prefix=@prefix@/lib/freetype2
 exec_prefix=@exec_prefix@
 
 usage()
diff -Naur freetype-2.1.3.old/builds/unix/freetype-config.in freetype-2.1.3.new/builds/unix/freetype-config.in
diff -Naur freetype-2.1.3.old/builds/unix/ltmain.sh freetype-2.1.3.new/builds/unix/ltmain.sh
--- freetype-2.1.3.old/builds/unix/ltmain.sh	Tue Mar  5 10:09:59 2002
+++ freetype-2.1.3.new/builds/unix/ltmain.sh	Fri Mar 28 13:26:18 2003
@@ -2894,6 +2894,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+	  else
+	    tmp_verstring="$verstring"
+	  fi
 	  eval cmds=\"$archive_cmds\"
 	fi
 	save_ifs="$IFS"; IFS='~'
diff -Naur freetype-2.1.3.old/include/freetype/config/ftoption.h freetype-2.1.3.new/include/freetype/config/ftoption.h
--- freetype-2.1.3.old/include/freetype/config/ftoption.h	Wed Nov  6 17:32:53 2002
+++ freetype-2.1.3.new/include/freetype/config/ftoption.h	Fri Mar 28 13:30:36 2003
@@ -378,7 +378,7 @@
   /*   do not #undef this macro here, since the build system might         */
   /*   define for certain configurations                                   */
   /*                                                                       */
-/* #define  TT_CONFIG_OPTION_BYTECODE_INTERPRETER */
+#define  TT_CONFIG_OPTION_BYTECODE_INTERPRETER 
 
 
   /*************************************************************************/
diff -Naur freetype-2.1.3.old/src/base/ftobjs.c freetype-2.1.3.new/src/base/ftobjs.c
--- freetype-2.1.3.old/src/base/ftobjs.c	Thu Oct 31 03:29:32 2002
+++ freetype-2.1.3.new/src/base/ftobjs.c	Fri Mar 28 13:26:18 2003
@@ -798,7 +798,7 @@
   /* there's a Mac-specific extended implementation of FT_New_Face() */
   /* in src/base/ftmac.c                                             */
 
-#ifndef FT_MACINTOSH
+/*#ifndef FT_MACINTOSH*/
 
   /* documentation is in freetype.h */
 
@@ -821,7 +821,7 @@
     return FT_Open_Face( library, &args, face_index, aface );
   }
 
-#endif  /* !FT_MACINTOSH */
+/*#endif  !FT_MACINTOSH */
 
 
   /* documentation is in freetype.h */
