--- ImageMagick-5.5.1/configure.orig	Fri Oct 25 12:50:02 2002
+++ ImageMagick-5.5.1/configure	Fri Oct 25 12:50:26 2002
@@ -31219,7 +31219,7 @@
           freetype_prefix=`freetype-config --prefix`
           freetype_exec_prefix=`freetype-config --exec-prefix`
           LDFLAGS="$LDFLAGS -L${freetype_exec_prefix}/lib"
-          CPPFLAGS="$CPPFLAGS -I${freetype_prefix}/include/freetype2"
+          CPPFLAGS="-I${freetype_prefix}/include/freetype2 $CPPFLAGS"
        fi
     fi
 
diff -ruN ImageMagick-5.5.1/config.cache ImageMagick-5.5.1-patched/config.cache
--- ImageMagick-5.5.1/config.cache	Wed Dec 31 19:00:00 1969
+++ ImageMagick-5.5.1-patched/config.cache	Sat Dec 13 16:56:54 2003
@@ -0,0 +1 @@
+ac_cv_func_poll=${ac_cv_func_poll=no}
