diff -ruN transfig.3.2.4/fig2dev/Imakefile transfig.3.2.4-patched/fig2dev/Imakefile
--- transfig.3.2.4/fig2dev/Imakefile	Thu Dec 19 12:33:44 2002
+++ transfig.3.2.4-patched/fig2dev/Imakefile	Sat Apr  2 15:43:28 2005
@@ -20,13 +20,13 @@
 XCOMM You should point XFIGLIBDIR to the same directory you did when you compiled
 XCOMM and installed xfig.
 
-XFIGLIBDIR =	/usr/local/lib/X11/xfig
+XFIGLIBDIR =	$(PREFIX)/share/xfig
 
 XCOMM ******
 XCOMM If your system has the strerror() function (doesn't have sys_errlist) then
 XCOMM comment out NEED_STRERROR with an XCOMM comment.
 
-NEED_STRERROR = -DNEED_STRERROR
+XCOMM NEED_STRERROR = -DNEED_STRERROR
 
 XCOMM The following probably only applies to Windows 9x/NT:
 XCOMM If your system can open files in text and binary modes and has the
@@ -60,8 +60,8 @@
 XCOMM are in different places
 
 #ifdef USEPNG
-PNGLIBDIR = $(USRLIBDIR)
-PNGINC = -I/usr/include/X11
+PNGLIBDIR = $(PREFIX)/lib
+PNGINC = -I$(PREFIX)/include
 #endif
 
 XCOMM ****************
@@ -73,11 +73,11 @@
 XCOMM Change XPMINC if necessary to point to the include file for xpm (xpm.h)
 XCOMM
 
-XCOMM #define USEXPM
+#define USEXPM
 
 #ifdef USEXPM
-XPMLIBDIR = $(USRLIBDIR)
-XPMINC = -I/usr/include/X11
+XPMLIBDIR = /usr/X11R6/lib
+XPMINC = -I/usr/X11R6/include/X11
 #endif
 
 XCOMM ****************
@@ -102,7 +102,7 @@
 
 DDNFSS = -DNFSS
 DDLATEX2E_GRAPHICS = -DLATEX2E_GRAPHICS
-XCOMM DDEPSFIG = -DEPSFIG
+DDEPSFIG = -DEPSFIG
 
 XCOMM ****************
 XCOMM Comment out the DDDVIPS to NOT use dvips for the DVI-to-PostScript driver
@@ -126,11 +126,11 @@
 
 #ifdef I18N
 I18N_DEFS = -DI18N 
-FIG2DEV_LIBDIR = /usr/local/lib/fig2dev
+FIG2DEV_LIBDIR = $(PREFIX)/lib/fig2dev
 I18N_DEV_DEFS = $(I18N_DEFS) -DFIG2DEV_LIBDIR=\\\"$(FIG2DEV_LIBDIR)\\\"
 
 install::
-	[ -d $(DESTDIR)$(FIG2DEV_LIBDIR) ] || mkdir $(DESTDIR)$(FIG2DEV_LIBDIR)
+	[ -d $(DESTDIR)$(FIG2DEV_LIBDIR) ] || mkdir -p $(DESTDIR)$(FIG2DEV_LIBDIR)
 	$(INSTALL) -c dev/japanese.ps $(DESTDIR)$(FIG2DEV_LIBDIR)
 	( cd $(DESTDIR)$(FIG2DEV_LIBDIR); \
 	  rm -f ja.ps ja_JP.ps ja_JP.ujis.ps ja_JP.eucJP.ps ja_JP.EUC.ps; \
diff -ruN transfig.3.2.4/fig2dev/fig2dev.h transfig.3.2.4-patched/fig2dev/fig2dev.h
--- transfig.3.2.4/fig2dev/fig2dev.h	Wed Dec 18 17:45:28 2002
+++ transfig.3.2.4-patched/fig2dev/fig2dev.h	Sat Apr  2 15:45:20 2005
@@ -22,7 +22,7 @@
 #include <sys/file.h>
 #include <signal.h>
 #include <string.h>
-#include <varargs.h>
+#include <stdarg.h>
 #include <pwd.h>
 #include <errno.h>
 #include <time.h>
diff -ruN transfig.3.2.4/transfig/Imakefile transfig.3.2.4-patched/transfig/Imakefile
--- transfig.3.2.4/transfig/Imakefile	Wed Dec 18 17:44:03 2002
+++ transfig.3.2.4-patched/transfig/Imakefile	Sat Apr  2 15:43:28 2005
@@ -18,7 +18,7 @@
 XCOMM \\usepackage{} command for LaTeX2e. 
 XCOMM The default is to use \\documentstyle{} for LaTeX209.
 
-XCOMM USELATEX2E = -DLATEX2E
+USELATEX2E = -DLATEX2E
 
 XCOMM ******* DON'T CHANGE ANYTHIN BELOW THIS POINT *******
 
diff -ruN transfig.3.2.4.orig/fig2dev/dev/gensvg.c transfig.3.2.4/fig2dev/dev/gensvg.c
--- transfig.3.2.4.orig/fig2dev/dev/gensvg.c	2003-06-24 23:13:43.000000000 +0200
+++ transfig.3.2.4/fig2dev/dev/gensvg.c	2003-06-24 23:18:55.000000000 +0200
@@ -692,16 +692,17 @@
     if (t->angle != 0) {
 	fprintf (tfp, "<g transform=\"translate(%d,%d) rotate(%d)\" >\n",
 		 (int) (t->base_x * mag), (int) (t->base_y * mag), degrees (t->angle));
-	fprintf (tfp, "<text x=\"0\" y=\"0\" fill=\"#%6.6x\"  font-family=\"%s\" 
-		 font-style=\"%s\" font-weight=\"%s\" font-size=\"%d\" text-anchor=\"%s\" >\n",
+	fprintf (tfp, "<text x=\"0\" y=\"0\" fill=\"#%6.6x\" "
+		"font-family=\"%s\" font-style=\"%s\" font-weight=\"%s\" "
+		"font-size=\"%d\" text-anchor=\"%s\" >\n",
 		 rgbColorVal (t->color), family[(int) ceil ((t->font + 1) / 4)],
 		 (t->font % 2 == 0 ? "normal" : "italic"),
 		 (t->font % 4 < 2 ? "normal" : "bold"), (int) (ceil (t->size * 12 * mag)),
 		 anchor[t->type]);
     }
     else
-	fprintf (tfp, "<text x=\"%d\" y=\"%d\" fill=\"#%6.6x\"  font-family=\"%s\" 
-		 font-style=\"%s\" font-weight=\"%s\" font-size=\"%d\" text-anchor=\"%s\" >\n",
+	fprintf (tfp, "<text x=\"%d\" y=\"%d\" fill=\"#%6.6x\"  font-family=\"%s\" "
+	"font-style=\"%s\" font-weight=\"%s\" font-size=\"%d\" text-anchor=\"%s\" >\n",
 		 (int) (t->base_x * mag), (int) (t->base_y * mag), rgbColorVal (t->color),
 		 family[(int) ceil ((t->font + 1) / 4)],
 		 (t->font % 2 == 0 ? "normal" : "italic"),
