diff -uNr coreutils-5.0/src/uname.c coreutils-5.0-patched/src/uname.c
--- coreutils-5.0/src/uname.c	Fri Dec  6 10:40:55 2002
+++ coreutils-5.0-patched/src/uname.c	Sun Nov  2 10:05:35 2003
@@ -247,6 +247,12 @@
 	  element = processor;
       }
 #endif
+#if defined __APPLE__ && defined(__ppc__)
+      element = "powerpc";
+#endif
+#if defined __APPLE__ && defined(__i386__)
+      element = "i386";
+#endif
 #ifdef UNAME_PROCESSOR
       if (element == unknown)
 	{
