diff -ruN jpilot-0.99.6.orig/Expense/Makefile.in jpilot-0.99.6/Expense/Makefile.in
--- jpilot-0.99.6.orig/Expense/Makefile.in	Sat Jul 12 22:34:06 2003
+++ jpilot-0.99.6/Expense/Makefile.in	Wed Jul 23 08:30:36 2003
@@ -194,7 +194,7 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(lib_LTLIBRARIES)
 
-libexpense_la_LDFLAGS =
+libexpense_la_LDFLAGS = -module -avoid-version
 libexpense_la_LIBADD =
 am__libexpense_la_SOURCES_DIST = expense.c
 @MAKE_EXPENSE_TRUE@am_libexpense_la_OBJECTS = expense.lo
diff -ruN jpilot-0.99.6.orig/KeyRing/Makefile.in jpilot-0.99.6/KeyRing/Makefile.in
--- jpilot-0.99.6.orig/KeyRing/Makefile.in	Sat Jul 12 22:34:06 2003
+++ jpilot-0.99.6/KeyRing/Makefile.in	Wed Jul 23 08:31:28 2003
@@ -197,7 +197,7 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(lib_LTLIBRARIES)
 
-libkeyring_la_LDFLAGS =
+libkeyring_la_LDFLAGS = -module -avoid-version
 libkeyring_la_LIBADD =
 am__libkeyring_la_SOURCES_DIST = keyring.c
 @MAKE_KEYRING_TRUE@am_libkeyring_la_OBJECTS = keyring.lo
diff -ruN jpilot-0.99.6.orig/SyncTime/Makefile.in jpilot-0.99.6/SyncTime/Makefile.in
--- jpilot-0.99.6.orig/SyncTime/Makefile.in	Sat Jul 12 22:34:07 2003
+++ jpilot-0.99.6/SyncTime/Makefile.in	Wed Jul 23 08:31:04 2003
@@ -194,7 +194,7 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(lib_LTLIBRARIES)
 
-libsynctime_la_LDFLAGS =
+libsynctime_la_LDFLAGS = -module -avoid-version
 libsynctime_la_LIBADD =
 am__libsynctime_la_SOURCES_DIST = synctime.c
 @MAKE_SYNCTIME_TRUE@am_libsynctime_la_OBJECTS = synctime.lo
diff -ruN jpilot-0.99.6.orig/configure jpilot-0.99.6/configure
--- jpilot-0.99.6.orig/configure	Sat Jul 12 22:34:08 2003
+++ jpilot-0.99.6/configure	Wed Jul 23 08:43:42 2003
@@ -5788,10 +5788,11 @@
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
     #	     `"' quotes if we put them in here... so don't!
-    archive_cmds='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
+    #archive_cmds='$nonopt -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
+    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle $allow_undefined_flag || echo -dynamiclib) -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
-    hardcode_direct=yes
+    hardcode_direct=no
     hardcode_shlibpath_var=no
     whole_archive_flag_spec='-all_load $convenience'
     ;;
diff -ruN jpilot-0.99.6.orig/ltmain.sh jpilot-0.99.6/ltmain.sh
--- jpilot-0.99.6.orig/ltmain.sh	Sat Jul  5 16:59:00 2003
+++ jpilot-0.99.6/ltmain.sh	Tue Jul 22 18:14:16 2003
@@ -1758,7 +1758,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    need_relink=no
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then
diff -ruN jpilot-0.99.6.orig/plugins.c jpilot-0.99.6/plugins.c
--- jpilot-0.99.6.orig/plugins.c	Mon Jun  2 22:45:16 2003
+++ jpilot-0.99.6/plugins.c	Tue Jul 22 18:15:19 2003
@@ -19,6 +19,7 @@
 
 #include "config.h"
 #ifdef  ENABLE_PLUGINS
+#include <sys/types.h>
 #include <dirent.h>
 #include <stdio.h>
 #include <stdlib.h>
