diff -Naur lesspipe-1.40.orig/configure lesspipe-1.40.new/configure
--- lesspipe-1.40.orig/configure	Mon Mar 22 01:08:11 2004
+++ lesspipe-1.40.new/configure	Mon Mar 29 10:52:56 2004
@@ -203,7 +203,7 @@
     }
 # do we have the shell in the PATH
     my $versstr = uc $name.'_VERSION';
-    my $versstr = 'BASH_VERSION' if $name eq 'sh';
+    $versstr = 'BASH_VERSION' if $name eq 'sh';
     my @where = grep { -x $_."/$name" } split ':', $ENV{PATH};
     $where[0] = $path if -x $path.'/'.$name; 
     my $file = $where[0].'/'.$name;
@@ -217,7 +217,7 @@
     chomp $v;
 ###    print "$file $v found in the PATH\n";
     if ( $name eq 'bash' or $name eq 'sh' ) {
-      my $tst = `$file -c \'if [[ \"a\" = \"a\" ]];then /bin/true;fi 2>&1\'`;
+      my $tst = `$file -c \'if [[ \"a\" = \"a\" ]];then true;fi 2>&1\'`;
       if ( $tst ) {
 	print "skipping $shell $v, need at least 2.03\n" if $v < 2.03;
 	print "skipping $shell $v, need at least 2.04\n" if $v == 2.03;
diff -Naur lesspipe-1.40.orig/syntax lesspipe-1.40.new/syntax
--- lesspipe-1.40.orig/syntax	Mon Mar 22 07:32:03 2004
+++ lesspipe-1.40.new/syntax	Mon Mar 29 11:25:21 2004
@@ -20,7 +20,7 @@
   if ( $ENV{LESS} !~ /-\w*r\w*\b/i ) {
 # check if less is called with -r or -R (highly OS dependent)
 # tested only for Linux, Solaris and AIX !!!
-    my $p = `ps -p $PPID -oppid= -oargs=`;
+    my $p = `ps -p $PPID -oppid= -ocommand=`;
     if ( $^O eq 'solaris' ) {
       $PPID = $1 if $p =~ /(\d+)/;
       my $p2 = `ps -p $PPID -oppid= -oargs=`;
