--- rzip-2.0/main.c	Wed Feb 11 19:01:08 2004
+++ rzip-2.0-new/main.c	Thu May 27 21:24:39 2004
@@ -118,7 +118,24 @@
 	fchown(fd_out, st.st_uid, st.st_gid);
 }	
 
-	
+static void*
+strndup (const char *src, size_t n)
+{
+	size_t i;
+	char *dst;
+
+	if (src == NULL)
+		return NULL;
+
+	dst = (char*) malloc (n + 1);
+	if (dst != NULL) {
+		for (i = 0; i < n && src[i]; i++)
+			dst[i] = src[i];
+		dst[i] = '\0';
+	}
+
+	return dst;
+}
 
 /*
   decompress one file from the command line
