--- tcpreplay-1.4.6/configure	2003-12-05 18:36:04.000000000 +0100
+++ tcpreplay-ok/configure	2003-12-07 15:42:03.000000000 +0100
@@ -5248,7 +5248,7 @@
   *)
      echo "$as_me:$LINENO: result: $withval" >&5
 echo "${ECHO_T}$withval" >&6
-     if test -f $withval/include/pcap.h -a -f $withval/lib/libpcap.a; then
+     if test -f $withval/include/pcap.h -a -f $withval/lib/libpcap.dylib; then
         owd=`pwd`
         if cd $withval; then
           withval=`pwd`
@@ -5258,8 +5258,8 @@
 		LPCAPINCDIR="$withval/include"
 		LPCAPLIB="-L$withval/lib -lpcap"
      else
-        { { echo "$as_me:$LINENO: error: pcap.h or libpcap.a not found in $withval" >&5
-echo "$as_me: error: pcap.h or libpcap.a not found in $withval" >&2;}
+        { { echo "$as_me:$LINENO: error: pcap.h or libpcap.dylib not found in $withval" >&5
+echo "$as_me: error: pcap.h or libpcap.dylib not found in $withval" >&2;}
    { (exit 1); exit 1; }; }
      fi
      ;;
