diff -ruN xmltv-0.5.25.orig/Makefile.PL xmltv-0.5.25/Makefile.PL
--- xmltv-0.5.25.orig/Makefile.PL	Sun Dec  7 15:31:53 2003
+++ xmltv-0.5.25/Makefile.PL	Tue Dec 16 19:03:50 2003
@@ -288,17 +288,17 @@
        exes => [ 'grab/dk/tv_grab_dk' ],
        prereqs => { 'HTML::TreeBuilder' => 0 } },
 
-     { name => 'tv_check',
-       blurb => 'Program to report exceptions and changes in a schedule',
-       exes => [ 'choose/tv_check/tv_check' ],
-       docs => [ qw(choose/tv_check/README.tv_check
-		    choose/tv_check/tv_check_doc.html
-		    choose/tv_check/tv_check_doc.jpg
-		   ) ],
-       prereqs => { 'Tk' => 0,
-		    'Tk::TableMatrix' => 0,
-		    'XML::Simple' => 0,
-		  } },
+#     { name => 'tv_check',
+#       blurb => 'Program to report exceptions and changes in a schedule',
+#       exes => [ 'choose/tv_check/tv_check' ],
+#       docs => [ qw(choose/tv_check/README.tv_check
+#		    choose/tv_check/tv_check_doc.html
+#		    choose/tv_check/tv_check_doc.jpg
+#		   ) ],
+#       prereqs => { 'Tk' => 0,
+#		    'Tk::TableMatrix' => 0,
+#		    'XML::Simple' => 0,
+#		  } },
 
      { name => 'tv_pick_cgi',
        blurb => 'CGI program to filter listings (to install manually)',
