diff -Nurd -x'*~' cssed-vte-plugin-0.2.orig/configure cssed-vte-plugin-0.2/configure
--- cssed-vte-plugin-0.2.orig/configure	2005-09-18 06:50:47.000000000 -0400
+++ cssed-vte-plugin-0.2/configure	2011-08-01 22:26:27.000000000 -0400
@@ -5151,7 +5151,7 @@
 i[3456]86-*-linux-gnu*)
 	OSTYPE="LINUX"
 	 ;;
-powerpc-apple-darwin*)
+*-apple-darwin*)
 	OSTYPE="DARWIN"
 	;;
 esac
diff -Nurd -x'*~' cssed-vte-plugin-0.2.orig/src/Makefile.in cssed-vte-plugin-0.2/src/Makefile.in
--- cssed-vte-plugin-0.2.orig/src/Makefile.in	2005-09-18 06:50:43.000000000 -0400
+++ cssed-vte-plugin-0.2/src/Makefile.in	2011-08-01 22:31:16.000000000 -0400
@@ -181,9 +181,9 @@
 @ISDARWIN_FALSE@vte_terminal_so_SOURCES = $(SRCS)
 @ISDARWIN_TRUE@vte_terminal_so_SOURCES = $(SRCS)
 @ISDARWIN_FALSE@vte_terminal_so_LDFLAGS = -shared -fPICS
-@ISDARWIN_TRUE@vte_terminal_so_LDFLAGS = -bundle -flat_namespace -undefined suppress
+@ISDARWIN_TRUE@vte_terminal_so_LDFLAGS = -bundle -undefined dynamic_lookup
 @ISDARWIN_FALSE@vte_terminal_so_LDADD = @PACKAGE_LIBS@
-@ISDARWIN_TRUE@vte_terminal_so_LDADD = @PACKAGE_LIBS@
+@ISDARWIN_TRUE@vte_terminal_so_LDADD = @PACKAGE_LIBS@ @INTLLIBS@
 @ISDARWIN_TRUE@vte_terminal_so_CFLAGS = -fno-common
 INCLUDES = \
 	-DPACKAGE_LOCALE_DIR=\""$(prefix)/share/locale"\" \
diff -Nurd -x'*~' cssed-vte-plugin-0.2.orig/src/vte.c cssed-vte-plugin-0.2/src/vte.c
--- cssed-vte-plugin-0.2.orig/src/vte.c	2005-01-05 08:50:04.000000000 -0500
+++ cssed-vte-plugin-0.2/src/vte.c	2011-08-01 22:37:42.000000000 -0400
@@ -19,6 +19,7 @@
 #  include <config.h>
 #endif
 
+#include <stdlib.h>
 #include <gtk/gtk.h>
 #include <vte/vte.h>
 #include <gmodule.h>
