diff -urN joe-3.5/Makefile.in joe-3.5-p1/Makefile.in
--- joe-3.5/Makefile.in	2006-07-20 19:24:18.000000000 +0000
+++ joe-3.5-p1/Makefile.in	2007-05-28 15:01:58.000000000 +0000
@@ -216,7 +216,7 @@
 sysconf_joedir = $(sysconfdir)/joe
 sysconf_charmapsdir = $(sysconf_joedir)/charmaps
 sysconf_charmaps_DATA = charmaps/klingon
-sysconf_docdir = $(sysconf_joedir)/doc
+sysconf_docdir = $(datadir)/doc/joe
 sysconf_doc_DATA = LIST README HINTS ChangeLog HACKING NEWS
 noinst_HEADERS = b.h blocks.h bw.h cmd.h config.h hash.h help.h kbd.h \
 	macro.h main.h menu.h path.h poshist.h pw.h queue.h qw.h rc.h regex.h \
diff -urN joe-3.5/macro.c joe-3.5-p1/macro.c
--- joe-3.5/macro.c	2006-07-19 00:19:02.000000000 +0000
+++ joe-3.5-p1/macro.c	2007-05-28 15:03:04.000000000 +0000
@@ -113,6 +113,12 @@
 	while (joe_isblank(locale_map,buf[x]))
 		++x;
 
+	/* If the buffer is only whitespace then treat as unknown command */
+	if (!buf[x]) {
+		*sta = -1;
+		return NULL;
+	}
+
 	/* Do we have a string? */
 	if (buf[x] == '\"') {
 		++x;
diff -urN joe-3.5/man/Makefile.in joe-3.5-p1/man/Makefile.in
--- joe-3.5/man/Makefile.in	2006-07-20 19:24:18.000000000 +0000
+++ joe-3.5-p1/man/Makefile.in	2007-05-28 15:01:58.000000000 +0000
@@ -506,7 +506,7 @@
 
 
 joe.1 : joe.1.in
-	sed 's,@JOERC\@,$(sysconf_joedir),; s,@SPELL\@,@SPELL@,; s,@BINDIR\@,$(bindir),' $(srcdir)/joe.1.in > $@
+	sed 's,@JOERC\@,$(DESTDIR)$(sysconf_joedir),; s,@SPELL\@,@SPELL@,; s,@BINDIR\@,$(DESTDIR)$(bindir),; s,@JOEDOC\@,$(DESTDIR)$(sysconf_docdir),' < $(srcdir)/joe.1.in > $@
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -urN joe-3.5/rc/Makefile.in joe-3.5-p1/rc/Makefile.in
--- joe-3.5/rc/Makefile.in	2006-07-20 19:24:20.000000000 +0000
+++ joe-3.5-p1/rc/Makefile.in	2007-05-28 15:01:58.000000000 +0000
@@ -326,22 +326,22 @@
 
 
 joerc : joerc.in
-	sed 's,@JOERC\@,$(sysconf_joedir),; s,@SPELL\@,@SPELL@,' $(srcdir)/joerc.in > $@
+	sed 's,@JOERC\@,$(DESTDIR)$(sysconf_joedir),; s,@SPELL\@,@SPELL@,; s,@JOEDOC\@,$(DESTDIR)$(sysconf_docdir),' < $(srcdir)/joerc.in > $@
 
 jicerc.ru : jicerc.ru.in
-	sed 's,@JOERC\@,$(sysconf_joedir),; s,@SPELL\@,@SPELL@,' $(srcdir)/jicerc.ru.in > $@
+	sed 's,@JOERC\@,$(DESTDIR)$(sysconf_joedir),; s,@SPELL\@,@SPELL@,; s,@JOEDOC\@,$(DESTDIR)$(sysconf_docdir),' < $(srcdir)/jicerc.ru.in > $@
 
 jmacsrc : jmacsrc.in
-	sed 's,@JOERC\@,$(sysconf_joedir),; s,@SPELL\@,@SPELL@,' $(srcdir)/jmacsrc.in > $@
+	sed 's,@JOERC\@,$(DESTDIR)$(sysconf_joedir),; s,@SPELL\@,@SPELL@,; s,@JOEDOC\@,$(DESTDIR)$(sysconf_docdir),' < $(srcdir)/jmacsrc.in > $@
 
 jstarrc : jstarrc.in
-	sed 's,@JOERC\@,$(sysconf_joedir),; s,@SPELL\@,@SPELL@,' $(srcdir)/jstarrc.in > $@
+	sed 's,@JOERC\@,$(DESTDIR)$(sysconf_joedir),; s,@SPELL\@,@SPELL@,; s,@JOEDOC\@,$(DESTDIR)$(sysconf_docdir),' < $(srcdir)/jstarrc.in > $@
 
 jpicorc : jpicorc.in
-	sed 's,@JOERC\@,$(sysconf_joedir),; s,@SPELL\@,@SPELL@,' $(srcdir)/jpicorc.in > $@
+	sed 's,@JOERC\@,$(DESTDIR)$(sysconf_joedir),; s,@SPELL\@,@SPELL@,; s,@JOEDOC\@,$(DESTDIR)$(sysconf_docdir),' < $(srcdir)/jpicorc.in > $@
 
 rjoerc : rjoerc.in
-	sed 's,@JOERC\@,$(sysconf_joedir),; s,@SPELL\@,@SPELL@,' $(srcdir)/rjoerc.in > $@
+	sed 's,@JOERC\@,$(DESTDIR)$(sysconf_joedir),; s,@SPELL\@,@SPELL@,; s,@JOEDOC\@,$(DESTDIR)$(sysconf_docdir),' < $(srcdir)/rjoerc.in > $@
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -urN joe-3.5/rc/jmacsrc.in joe-3.5-p1/rc/jmacsrc.in
--- joe-3.5/rc/jmacsrc.in	2006-06-21 13:15:43.000000000 +0000
+++ joe-3.5-p1/rc/jmacsrc.in	2007-05-28 15:01:58.000000000 +0000
@@ -423,7 +423,7 @@
 \i \i Send bug reports to: http://sourceforge.net/projects/joe-editor \|\i \i
 \i \i \|\i \i
 \i \i  default jmacsrc file is here @JOERC@/jmacsrc \|\i \i
-\i \i  additional documentation can be found here @JOERC@/doc \|\i \i
+\i \i  additional documentation can be found here @JOEDOC@ \|\i \i
 }
 
  FOURTH SECTION: Key bindings:
diff -urN joe-3.5/rc/joerc.in joe-3.5-p1/rc/joerc.in
--- joe-3.5/rc/joerc.in	2006-06-21 13:15:50.000000000 +0000
+++ joe-3.5-p1/rc/joerc.in	2007-05-28 15:01:58.000000000 +0000
@@ -476,7 +476,7 @@
 \i \i Send bug reports to: http://sourceforge.net/projects/joe-editor \|\i \i
 \i \i \|\i \i
 \i \i  default joerc file is here @JOERC@/joerc \|\i \i
-\i \i  additional documentation can be found here @JOERC@/doc \|\i \i
+\i \i  additional documentation can be found here @JOEDOC@ \|\i \i
 }
 
  FOURTH SECTION: Key bindings:
diff -urN joe-3.5/rc/jpicorc.in joe-3.5-p1/rc/jpicorc.in
--- joe-3.5/rc/jpicorc.in	2006-06-21 13:15:56.000000000 +0000
+++ joe-3.5-p1/rc/jpicorc.in	2007-05-28 15:01:58.000000000 +0000
@@ -440,7 +440,7 @@
 \i \i Send bug reports to: http://sourceforge.net/projects/joe-editor \|\i \i
 \i \i \|\i \i
 \i \i  default joerc file is here @JOERC@/jpicorc \|\i \i
-\i \i  additional documentation can be found here @JOERC@/doc \|\i \i
+\i \i  additional documentation can be found here @JOEDOC@ \|\i \i
 }
 
  FOURTH SECTION: Key bindings:
diff -urN joe-3.5/rc/jstarrc.in joe-3.5-p1/rc/jstarrc.in
--- joe-3.5/rc/jstarrc.in	2006-06-21 13:16:13.000000000 +0000
+++ joe-3.5-p1/rc/jstarrc.in	2007-05-28 15:01:58.000000000 +0000
@@ -429,7 +429,7 @@
 \i \i Send bug reports to: http://sourceforge.net/projects/joe-editor \|\i \i
 \i \i \|\i \i
 \i \i  default jstarrc file is here @JOERC@/jstarrc \|\i \i
-\i \i  additional documentation can be found here @JOERC@/doc \|\i \i
+\i \i  additional documentation can be found here @JOEDOC@ \|\i \i
 }
 
  FOURTH SECTION: Key bindings:
diff -urN joe-3.5/rc/rjoerc.in joe-3.5-p1/rc/rjoerc.in
--- joe-3.5/rc/rjoerc.in	2006-06-21 13:16:25.000000000 +0000
+++ joe-3.5-p1/rc/rjoerc.in	2007-05-28 15:01:58.000000000 +0000
@@ -440,7 +440,7 @@
 \i \i Send bug reports to: http://sourceforge.net/projects/joe-editor \|\i \i
 \i \i \|\i \i
 \i \i  default joerc file is here @JOERC@/joerc \|\i \i
-\i \i  additional documentation can be found here @JOERC@/doc \|\i \i
+\i \i  additional documentation can be found here @JOEDOC@ \|\i \i
 }
 
 
