diff -Naur --exclude='*~' mac-key-mode/fink/50mac-key-mode.el mac-key-mode.fink/fink/50mac-key-mode.el
--- mac-key-mode/fink/50mac-key-mode.el	1969-12-31 18:00:00.000000000 -0600
+++ mac-key-mode.fink/fink/50mac-key-mode.el	2007-12-26 21:11:37.000000000 -0600
@@ -0,0 +1 @@
+(require 'mac-key-mode)
diff -Naur --exclude='*~' mac-key-mode/fink/emacsen-install mac-key-mode.fink/fink/emacsen-install
--- mac-key-mode/fink/emacsen-install	1969-12-31 18:00:00.000000000 -0600
+++ mac-key-mode.fink/fink/emacsen-install	2007-12-15 15:04:19.000000000 -0600
@@ -0,0 +1,34 @@
+#!/bin/bash -e
+#
+# install file for the fink mac-key-mode emacs package.
+
+set -o posix
+
+FLAVOR=${1}
+
+echo >&2 "install/mac-key-mode: Handling install of emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "install/mac-key-mode: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+if [ ${FLAVOR} == emacs ]
+then
+    # Nothing to do, say nothing
+    exit 0
+fi
+
+echo >&2 -n "install/mac-key-mode: Byte-compiling for ${FLAVOR}..."
+mkdir -p @PREFIX@/share/${FLAVOR}/site-lisp/mac-key-mode
+ln -s -f @PREFIX@/share/emacs/site-lisp/mac-key-mode/mac-key-mode.el @PREFIX@/share/${FLAVOR}/site-lisp/mac-key-mode
+
+cd @PREFIX@/share/${FLAVOR}/site-lisp/mac-key-mode;
+
+(${FLAVOR} --no-init-file -batch -f batch-byte-compile mac-key-mode.el 2>&1) | gzip -9qf > @PREFIX@/share/doc/mac-key-mode/CompilationLog-${FLAVOR}.gz
+
+echo >&2 "done."
+echo >&2 "install/mac-key-mode: Compilation log saved in @PREFIX@/share/doc/mac-key-mode/CompilationLog-${FLAVOR}.gz."
+
+exit 0
diff -Naur --exclude='*~' mac-key-mode/fink/emacsen-remove mac-key-mode.fink/fink/emacsen-remove
--- mac-key-mode/fink/emacsen-remove	1969-12-31 18:00:00.000000000 -0600
+++ mac-key-mode.fink/fink/emacsen-remove	2007-12-10 18:35:39.000000000 -0600
@@ -0,0 +1,21 @@
+#!/bin/bash -e
+#
+# This is the Fink version of the mac-key-mode emacs package.
+set -o posix
+
+FLAVOR=${1}
+
+echo "remove/mac-key-mode: Handling removal for emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "remove/mac-key-mode: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+echo >&2 -n "remove/mac-key-mode: Purging compilation log and byte-compiled files for ${FLAVOR}..."
+rm -f @PREFIX@/share/doc/mac-key-mode/CompilationLog-${FLAVOR}.gz
+rm -Rf @PREFIX@/share/${FLAVOR}/site-lisp/mac-key-mode
+echo >&2 "done."
+    
+exit 0
