diff -Naur nxml-mode-20040908/fink/emacsen-install nxml-mode-20040908.fink/fink/emacsen-install
--- nxml-mode-20040908/fink/emacsen-install	Thu Jan  1 10:00:00 1970
+++ nxml-mode-20040908.fink/fink/emacsen-install	Wed Sep  8 22:24:18 2004
@@ -0,0 +1,40 @@
+#! /bin/sh
+# @PREFIX@/lib/emacsen-common/packages/install/nxml-mode
+set -e
+
+FLAVOR=$1
+PACKAGE=nxml-mode
+
+if [ ${FLAVOR} == emacs ]; then exit 0; fi
+
+echo install/${PACKAGE}: Handling install for emacsen flavor ${FLAVOR}
+
+FLAGS="-no-site-file -q -batch -l path.el -f batch-byte-compile"
+
+ELDIR=@PREFIX@/share/emacs/site-lisp/${PACKAGE}
+ELCDIR=@PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}
+
+/bin/rm -rf ${ELCDIR}
+
+/usr/bin/install -m 755 -d ${ELCDIR}
+cd ${ELDIR}
+FILES=`echo *.el`
+/bin/cp ${FILES} ${ELCDIR}
+cd ${ELCDIR}
+
+/bin/cat << EOF > path.el
+(setq load-path (cons "." load-path) byte-compile-warnings nil)
+EOF
+
+${FLAVOR} ${FLAGS} ${FILES}
+/bin/rm -f *.el path.el
+
+/usr/bin/install -m 755 -d ${ELCDIR}/char-name/unicode
+ln ${ELDIR}/char-name/unicode/*.el ${ELCDIR}/char-name/unicode
+
+/bin/cat << EOF > @PREFIX@/etc/${FLAVOR}/site-start.d/50${PACKAGE}.el
+(load "@PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}/rng-auto.elc")
+EOF
+
+exit 0
+
diff -Naur nxml-mode-20040908/fink/emacsen-remove nxml-mode-20040908.fink/fink/emacsen-remove
--- nxml-mode-20040908/fink/emacsen-remove	Thu Jan  1 10:00:00 1970
+++ nxml-mode-20040908.fink/fink/emacsen-remove	Wed Sep  8 22:24:18 2004
@@ -0,0 +1,13 @@
+#! /bin/sh
+# @PREFIX@/lib/emacsen-common/packages/remove/nxml-mode
+
+FLAVOR=$1
+PACKAGE=nxml-mode
+if [ ${FLAVOR} != emacs ]; then
+
+    echo remove/${PACKAGE}: purging byte-compiled files for ${FLAVOR}
+    /bin/rm -rf @PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}
+
+    /bin/rm -f @PREFIX@/etc/${FLAVOR}/site-start.d/??${PACKAGE}.el
+fi
+
diff -Naur nxml-mode-20040908/nxml-mode.info nxml-mode-20040908.fink/nxml-mode.info
--- nxml-mode-20040908/nxml-mode.info	Wed Sep  8 19:56:37 2004
+++ nxml-mode-20040908.fink/nxml-mode.info	Wed Sep  8 22:24:17 2004
@@ -3,7 +3,7 @@
 
 INFO-DIR-SECTION Emacs
 START-INFO-DIR-ENTRY
-* nXML Mode: (nxml-mode.info).
+* nXML Mode: (nxml-mode.info).     Emacs major mode for editing XML
 END-INFO-DIR-ENTRY
 
 
diff -Naur nxml-mode-20040908/rng-auto.el nxml-mode-20040908.fink/rng-auto.el
--- nxml-mode-20040908/rng-auto.el	Wed Sep  8 19:51:09 2004
+++ nxml-mode-20040908.fink/rng-auto.el	Wed Sep  8 22:24:17 2004
@@ -55,7 +55,7 @@
 		 (append Info-default-directory-list (list dir)))))))
 
 (let* ((dir (file-name-directory load-file-name))
-       (schema-dir (concat dir "schema/")))
+       (schema-dir "@PREFIX@/share/schema/"))
   (unless (member dir load-path)
     (setq load-path (cons dir load-path)))
   (setq rng-schema-locating-files-default
