--- Jinja-1.2/setup.py.bak        2008-05-30 10:56:29.000000000 -0400
+++ Jinja-1.2/setup.py    2008-05-30 10:57:12.000000000 -0400
@@ -78,10 +78,10 @@
     ],
     keywords=['python.templating.engines'],
     packages=['jinja', 'jinja.translators'],
-    data_files=[
-        ('docs/html', list(list_files('docs/html'))),
-        ('docs/txt', list(list_files('docs/src')))
-    ],
+#    data_files=[
+#        ('docs/html', list(list_files('docs/html'))),
+#        ('docs/txt', list(list_files('docs/src')))
+#    ],
     entry_points='''
     [python.templating.engines]
     jinja = jinja.plugin:BuffetPlugin
