diff -ruN cdrdao-1.2.0/dao/main.cc cdrdao-1.2.0.new/dao/main.cc
--- cdrdao-1.2.0/dao/main.cc	2005-05-09 11:48:40.000000000 -0400
+++ cdrdao-1.2.0.new/dao/main.cc	2005-09-14 10:23:30.000000000 -0400
@@ -122,6 +122,11 @@
 static const char *SCSI_DEVICE = "/dev/cdrecorder";
 static int FIFO_BUFFERS = 32;
 
+#elif defined(__APPLE__) && defined(__GNUC__)
+
+static const char *SCSI_DEVICE = "IOCompactDiscServices";
+static int FIFO_BUFFERS = 32;
+
 #else
 
 static const char *SCSI_DEVICE = "0,0,0";
