diff -Nurd -x'*~' dialog-1.0-20060221.orig/makefile.in dialog-1.0-20060221/makefile.in
--- dialog-1.0-20060221.orig/makefile.in	2006-01-18 17:42:18.000000000 -0500
+++ dialog-1.0-20060221/makefile.in	2008-08-18 12:15:44.000000000 -0400
@@ -27,7 +27,7 @@
 localedir       = $(prefix)/@DATADIRNAME@/locale
 
 CFLAGS		= @CFLAGS@
-CPPFLAGS	= @CPPFLAGS@ @DEFS@ -I. -I$(srcdir) -DLOCALEDIR=\"$(localedir)\"
+CPPFLAGS	= -I. -I$(srcdir) @CPPFLAGS@ @DEFS@ -DLOCALEDIR=\"$(localedir)\"
 EXTRA_CFLAGS	= @EXTRA_CFLAGS@
 CC		= @CC@
 AR		= @AR@
@@ -111,7 +111,7 @@
 	$(RANLIB) $@
 
 dialog$x : $(LIB) dialog$o @INTLDIR_MAKE@ @INTLLIBS@
-	$(LINK) -o $@ dialog$o -L. -ldialog $(LDFLAGS) $(LIBS)
+	$(LINK) -o $@ dialog$o libdialog.a $(LDFLAGS) $(LIBS)
 
 clean	\
 distclean \
