diff -ru fuse-zip-0.2.2.orig/Makefile fuse-zip-0.2.2/Makefile
--- fuse-zip-0.2.2.orig/Makefile	2008-06-15 23:48:38.000000000 -0400
+++ fuse-zip-0.2.2/Makefile	2008-06-16 21:26:22.000000000 -0400
@@ -36,7 +36,7 @@
 
 install: all doc
 	mkdir -p "$(INSTALLPREFIX)/bin"
-	install -m 755 -s -t "$(INSTALLPREFIX)/bin" "$(DEST)"
+	install -m 755 -s "$(DEST)" "$(INSTALLPREFIX)/bin"
 	mkdir -p "$(INSTALLPREFIX)/share/doc/$(DEST)"
 	cp $(DOCFILES) "$(INSTALLPREFIX)/share/doc/$(DEST)"
 	mkdir -p "$(INSTALLPREFIX)/share/man/man1"
