diff -Naur glark-1.6.7/Makefile glark-patched/Makefile
--- glark-1.6.7/Makefile	Mon Feb 16 17:46:12 2004
+++ glark-patched/Makefile	Mon Mar 22 23:31:32 2004
@@ -7,6 +7,8 @@
 DIST_SOURCES  = glark.ps glark.pdf glark.1 glark Makefile ChangeLog
 INSTALL       = /usr/bin/install -c
 PREFIX       ?= /usr
+bindir        = $(PREFIX)/bin
+mandir        = $(PREFIX)/share/man
 
 distdir       = ${PACKAGE}-${VERSION}
 srcdir        = .
@@ -88,14 +90,17 @@
 	tar -zcf $@ $(PACKAGE)-$(VERSION)
 	rm -rf $(PACKAGE)-$(VERSION)
 
-install: glark.1
-	${INSTALL} glark $(PREFIX)/bin
-	${INSTALL} glark.1 $(PREFIX)/share/man/man1
+installdirs:
+	${INSTALL} -d $(bindir)
+	${INSTALL} -d $(mandir)/man1
+
+install: glark.1 installdirs
+	${INSTALL} -m 755 glark $(bindir)
+	${INSTALL} -m 644 glark.1 $(mandir)/man1
 
 uninstall: glark.1
-	rm -f $(PREFIX)/bin/glark
-	-rm -f $(PREFIX)/man/man1/glark.1
-	-rm -f $(PREFIX)/share/man/man1/glark.1
+	-rm -f $(bindir)/glark
+	-rm -f $(mandir)/man1/glark.1
 
 clean:
 	rm -f glark.ps glark.pdf glark.1 glark.dvi glark.tex
