diff -Nurd -x'*~' zebra-0.5.orig/Makefile.in zebra-0.5/Makefile.in
--- zebra-0.5.orig/Makefile.in	2008-07-24 20:11:36.000000000 -0400
+++ zebra-0.5/Makefile.in	2008-08-13 23:41:58.000000000 -0400
@@ -274,7 +274,7 @@
 zebraimg_zebraimg_OBJECTS = $(am_zebraimg_zebraimg_OBJECTS)
 @HAVE_MAGICK_TRUE@zebraimg_zebraimg_DEPENDENCIES =  \
 @HAVE_MAGICK_TRUE@	$(am__DEPENDENCIES_1) zebra/libzebra.la
-DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/include
+DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/_include
 depcomp = $(SHELL) $(top_srcdir)/config/depcomp
 am__depfiles_maybe = depfiles
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
@@ -492,7 +492,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-AM_CPPFLAGS = -I$(srcdir)/include
+AM_CPPFLAGS = 
 AM_CFLAGS = -Wall
 AM_CXXFLAGS = $(AM_CFLAGS)
 ACLOCAL_AMFLAGS = -I config
diff -Nurd -x'*~' zebra-0.5.orig/configure zebra-0.5/configure
--- zebra-0.5.orig/configure	2008-07-24 20:11:34.000000000 -0400
+++ zebra-0.5/configure	2008-08-15 01:52:42.000000000 -0400
@@ -19093,7 +19093,7 @@
 fi
 
    if test "x$PYTHON_LIBS" = "x"; then
-  PYTHON_LIBS=`$PYTHON_CONFIG --libs`
+  PYTHON_LIBS=`$PYTHON_CONFIG --ldflags`
 fi
 
 
diff -Nurd -x'*~' zebra-0.5.orig/pygtk/zebrapygtkmodule.c zebra-0.5/pygtk/zebrapygtkmodule.c
--- zebra-0.5.orig/pygtk/zebrapygtkmodule.c	2008-07-24 20:10:12.000000000 -0400
+++ zebra-0.5/pygtk/zebrapygtkmodule.c	2008-08-15 02:01:12.000000000 -0400
@@ -21,6 +21,7 @@
  *  http://sourceforge.net/projects/zebra
  *------------------------------------------------------------------------*/
 
+#define NO_IMPORT_PYGOBJECT
 #include <pygobject.h>
 
 void zebrapygtk_register_classes(PyObject*);
