diff -Nurd -x'*~' MToolsFM-1.9-3.orig/Makefile.in MToolsFM-1.9-3/Makefile.in
--- MToolsFM-1.9-3.orig/Makefile.in	2001-08-27 02:58:11.000000000 -0400
+++ MToolsFM-1.9-3/Makefile.in	2005-11-02 00:29:13.000000000 -0500
@@ -89,7 +89,7 @@
 l = @l@
 localedir = @localedir@
 
-SUBDIRS = po intl src
+SUBDIRS = po      src
 man_MANS = MToolsFM.1
 EXTRA_DIST = MToolsFM.xpm MToolsFM.mini.xpm $(man_MANS) MToolsFM.1.gz
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
diff -Nurd -x'*~' MToolsFM-1.9-3.orig/po/Makefile.in.in MToolsFM-1.9-3/po/Makefile.in.in
--- MToolsFM-1.9-3.orig/po/Makefile.in.in	2001-01-26 04:50:35.000000000 -0500
+++ MToolsFM-1.9-3/po/Makefile.in.in	2005-11-01 22:34:38.000000000 -0500
@@ -19,7 +19,7 @@
 prefix = @prefix@
 exec_prefix = @exec_prefix@
 datadir = $(prefix)/@DATADIRNAME@
-localedir = $(datadir)/locale
+localedir = $(prefix)/share/locale
 gnulocaledir = $(prefix)/share/locale
 gettextsrcdir = $(prefix)/share/gettext/po
 subdir = po
diff -Nurd -x'*~' MToolsFM-1.9-3.orig/po/fr.po MToolsFM-1.9-3/po/fr.po
--- MToolsFM-1.9-3.orig/po/fr.po	2001-08-21 03:31:41.000000000 -0400
+++ MToolsFM-1.9-3/po/fr.po	2005-11-01 22:35:37.000000000 -0500
@@ -778,8 +778,9 @@
 msgid "/Options/Text mode for copying"
 msgstr "/Options/Utiliser le mode texte pour la copie"
 
-msgid "/Options/Text mode for copying"
-msgstr "/Options/Separator"
+#: src/mtoolsfm.c:2993
+msgid "/Options/Separator"
+msgstr "/Options/Separateur"
 
 #: src/mtoolsfm.c:2817 src/mtoolsfm.c:2833
 msgid "/Options/Configure MToolsFM ..."
diff -Nurd -x'*~' MToolsFM-1.9-3.orig/src/Makefile.in MToolsFM-1.9-3/src/Makefile.in
--- MToolsFM-1.9-3.orig/src/Makefile.in	2001-08-27 02:58:41.000000000 -0400
+++ MToolsFM-1.9-3/src/Makefile.in	2005-11-02 00:40:07.000000000 -0500
@@ -90,7 +90,7 @@
 localedir = @localedir@
 
 DEFS = @DEFS@ -DLOCALEDIR=\"${localedir}\"
-INCLUDES = -I../intl -I$(top_srcdir)/intl @GTK_CFLAGS@ 
+INCLUDES = @GTK_CFLAGS@ 
 
 bin_PROGRAMS = MToolsFM
 MToolsFM_SOURCES = mtoolsfm.c lex.er.c mtoolsfm.h pixmaps.h
