diff -ru ntfs-3g-1.616.orig/src/ntfs-3g.c ntfs-3g-1.616/src/ntfs-3g.c
--- ntfs-3g-1.616.orig/src/ntfs-3g.c	2007-06-16 17:34:13.000000000 -0400
+++ ntfs-3g-1.616/src/ntfs-3g.c	2007-07-06 11:03:41.000000000 -0400
@@ -2157,6 +2157,10 @@
 	fstype = get_fuse_fstype();
 	if (fstype == FSTYPE_NONE || fstype == FSTYPE_UNKNOWN)
 		fstype = load_fuse_module();
+#else
+#if (__FreeBSD__ >= 10)
+	fstype = FSTYPE_FUSE;
+#endif
 #endif	
 	create_dev_fuse();
 	
