diff -ru ntfs-3g-1.0.orig/configure ntfs-3g-1.0/configure
--- ntfs-3g-1.0.orig/configure	2007-02-20 15:15:27.000000000 -0500
+++ ntfs-3g-1.0/configure	2007-02-23 13:36:28.000000000 -0500
@@ -19804,7 +19804,7 @@
 
 
 case "$target_os" in
-linux*)
+darwin*)
 
   succeeded=no
 
diff -ru ntfs-3g-1.0.orig/src/Makefile.in ntfs-3g-1.0/src/Makefile.in
--- ntfs-3g-1.0.orig/src/Makefile.in	2007-02-20 15:15:39.000000000 -0500
+++ ntfs-3g-1.0/src/Makefile.in	2007-02-23 13:36:58.000000000 -0500
@@ -580,8 +580,8 @@
 	(cd ../libntfs-3g && $(MAKE) libs) || exit 1;
 
 install-exec-hook:
-	$(INSTALL) -d $(DESTDIR)/sbin
-	$(LN_S) -f $(bindir)/ntfs-3g $(DESTDIR)/sbin/mount.ntfs-3g
+	$(INSTALL) -d $(DESTDIR)/$(sbindir)
+	$(LN_S) -f $(bindir)/ntfs-3g $(DESTDIR)/$(sbindir)/mount.ntfs-3g
 	$(LDCONFIG)
 
 install-data-hook:
