diff -ru patool-0.12.orig/patoolib/__init__.py patool-0.12/patoolib/__init__.py
--- patool-0.12.orig/patoolib/__init__.py	2010-11-20 08:26:52.000000000 -0500
+++ patool-0.12/patoolib/__init__.py	2010-11-29 12:33:32.000000000 -0500
@@ -157,10 +157,10 @@
         'test': ('7z',),
     },
     'cpio': {
-        'extract': ('cpio', '7z'),
-        'list': ('cpio', '7z'),
+        'extract': ('gcpio', '7z'),
+        'list': ('gcpio', '7z'),
         'test': ('7z',),
-        'create': ('cpio',),
+        'create': ('gcpio',),
     },
     'rpm': {
         'extract': ('rpm2cpio', '7z'),
diff -ru patool-0.12.orig/patoolib/programs/rpm2cpio.py patool-0.12/patoolib/programs/rpm2cpio.py
--- patool-0.12.orig/patoolib/programs/rpm2cpio.py	2010-11-20 08:26:52.000000000 -0500
+++ patool-0.12/patoolib/programs/rpm2cpio.py	2010-11-29 12:34:12.000000000 -0500
@@ -20,7 +20,7 @@
 def extract_rpm (archive, encoding, cmd, **kwargs):
     """Extract a RPM archive."""
     # also check cpio
-    cpio = util.find_program("cpio")
+    cpio = util.find_program("gcpio")
     if not cpio:
         raise util.PatoolError("cpio(1) is required for rpm2cpio extraction; please install it")
     path = util.shell_quote(os.path.abspath(archive))
diff -ru patool-0.12.orig/tests/test_archives.py patool-0.12/tests/test_archives.py
--- patool-0.12.orig/tests/test_archives.py	2010-11-20 08:26:52.000000000 -0500
+++ patool-0.12/tests/test_archives.py	2010-11-29 12:35:11.000000000 -0500
@@ -256,9 +256,9 @@
         self.program = 'ar'
         self.archive_commands('t.a', singlefile=True)
 
-    @needs_program('cpio')
+    @needs_program('gcpio')
     def test_cpio (self):
-        self.program = 'cpio'
+        self.program = 'gcpio'
         self.archive_list('t.cpio')
         self.archive_extract('t.cpio')
         self.archive_create('t.cpio')
@@ -280,7 +280,7 @@
         #self.archive_test('t.rpm')
 
     @needs_program('rpm2cpio')
-    @needs_program('cpio')
+    @needs_program('gcpio')
     def test_rpm_extract (self):
         self.program = 'rpm2cpio'
         self.archive_extract('t.rpm')
diff -ru patool-0.12.orig/tests/test_foo_archives.py patool-0.12/tests/test_foo_archives.py
--- patool-0.12.orig/tests/test_foo_archives.py	2010-11-20 08:26:52.000000000 -0500
+++ patool-0.12/tests/test_foo_archives.py	2010-11-29 12:35:46.000000000 -0500
@@ -287,9 +287,9 @@
         self.archive_commands('t.a.foo', format='ar', singlefile=True)
 
     @needs_program('file')
-    @needs_program('cpio')
+    @needs_program('gcpio')
     def test_cpio (self):
-        self.program = 'cpio'
+        self.program = 'gcpio'
         self.archive_list('t.cpio.foo')
         self.archive_extract('t.cpio.foo')
         self.archive_create('t.cpio.foo', format="cpio")
@@ -314,7 +314,7 @@
 
     @needs_program('file')
     @needs_program('rpm2cpio')
-    @needs_program('cpio')
+    @needs_program('gcpio')
     def test_rpm_extract (self):
         self.program = 'rpm2cpio'
         self.archive_extract('t.rpm.foo')
