--- rkhunter.orig/files/rkhunter	2008-02-27 16:59:22.000000000 +0100
+++ rkhunter.new/files/rkhunter		2008-06-10 12:54:47.000000000 +0200
@@ -3751,7 +3751,7 @@
 		if [ -n "`echo \"${STAT_CMD}\" | grep '\.pl$'`" ]; then
 			SCMD="${STAT_CMD} --modeoct --raw --ino --mode --uid --gid --size --Mtime"
 			INODECMD="${STAT_CMD} --modeoct --raw --ino"
-		elif [ $BSDOS -eq 1 ]; then
+		elif [ $BSDOS -eq 1 -o "${OPERATING_SYSTEM}" = "Darwin" ]; then
 			SCMD="${STAT_CMD} -f '%i %Mp%Lp %u %g %z %m:'"
 			INODECMD="${STAT_CMD} -f '%i'"
 		else
@@ -7008,7 +7008,7 @@
 		if [ -n "`echo \"${STAT_CMD}\" | grep '\.pl$'`" ]; then
 			SCMD="${STAT_CMD} --modeoct --raw --ino --mode --uid --gid --size --Mtime"
 			INODECMD="${STAT_CMD} --modeoct --raw --ino"
-		elif [ $BSDOS -eq 1 ]; then
+		elif [ $BSDOS -eq 1 -o "${OPERATING_SYSTEM}" = "Darwin" ]; then
 			SCMD="${STAT_CMD} -f '%i %Mp%Lp %u %g %z %m:'"
 			INODECMD="${STAT_CMD} -f '%i'"
 		else
