diff -ru tree-1.5.0/Makefile tree-1.5.0-patched/Makefile
--- tree-1.5.0/Makefile	2004-08-15 00:07:56.000000000 +0200
+++ tree-1.5.0-patched/Makefile	2007-10-26 20:50:17.000000000 +0200
@@ -6,10 +6,10 @@
 # warranties, including, without limitation, the implied warranties
 # of merchant-ability and fitness for a particular purpose.
 
-CC=gcc
+#CC=gcc
 #CFLAGS=-ggdb -Wall -DLINUX_BIGFILE
-CFLAGS=-O2 -Wall -fomit-frame-pointer -DLINUX_BIGFILE
-LDFLAGS=-s
+#CFLAGS=-O2 -Wall -fomit-frame-pointer -DLINUX_BIGFILE
+#LDFLAGS=-s
 
 # Uncomment for FreeBSD:
 #CC=gcc
@@ -17,9 +17,9 @@
 #LDFLAGS=-s
 
 # Uncomment for OS X:
-#CC=cc
-#CFLAGS=-O2 -Wall -fomit-frame-pointer -no-cpp-precomp
-#LDFLAGS=
+CC=cc
+#FLAGS=-O2 -Wall -fomit-frame-pointer -no-cpp-precomp
+#DFLAGS=
 
 # Uncomment for HP/UX:
 #CC=cc
@@ -37,7 +37,7 @@
 TREE_DEST=tree
 BINDIR=${prefix}/bin
 MAN=tree.1
-MANDIR=${prefix}/man/man1
+MANDIR=${prefix}/share/man/man1
 
 all:	tree
 
diff -ru tree-1.5.0/tree.c tree-1.5.0-patched/tree.c
--- tree-1.5.0/tree.c	2004-08-16 03:07:21.000000000 +0200
+++ tree-1.5.0-patched/tree.c	2007-10-26 21:05:18.000000000 +0200
@@ -17,13 +17,6 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-#ifdef LINUX_BIGFILE
-#  define _LARGEFILE64_SOURCE
-#else
-#  define stat64 stat
-#  define lstat64 lstat
-#endif
-
 #include <stdlib.h>
 #include <stdio.h>
 #include <time.h>
@@ -61,6 +54,13 @@
 #include <wchar.h>
 #include <wctype.h>
 
+#ifdef LINUX_BIGFILE
+#  define _LARGEFILE64_SOURCE
+#else
+#  define stat64 stat
+#  define lstat64 lstat
+#endif
+
 static char *version ="$Version: $ tree v1.5.0 (c) 1996 - 2004 by Steve Baker, Thomas Moore, Francesc Rocher, Kyosuke Tokoro $";
 static char *hversion="\t\t\t tree v1.5.0 %s 1996 - 2004 by Steve Baker and Thomas Moore <br>\n"
 		      "\t\t\t HTML output hacked and copyleft %s 1998 by Francesc Rocher <br>\n"
@@ -189,8 +189,6 @@
 FILE *outfile;
 int *dirs, maxdirs;
 
-extern size_t MB_CUR_MAX;
-
 int main(int argc, char **argv)
 {
   char **dirname = NULL;
