diff -Naur uim-1.4.2.orig/xinitrc.uim.sh uim-1.4.2/xinitrc.uim.sh
--- uim-1.4.2.orig/xinitrc.uim.sh	1970-01-01 09:00:00.000000000 +0900
+++ uim-1.4.2/xinitrc.uim.sh	2008-03-31 02:51:31.000000000 +0900
@@ -0,0 +1,22 @@
+# If you don't want to launch uim, put xinitrc_uim_enable=NO
+# in your .xinitrc
+: ${xinitrc_uim_enable=YES}
+: ${xinitrc_uim_toolbar_enable=NO}
+
+# 'anthy' and 'skk' is available at this time.
+: ${xinitrc_uim_engine=anthy}
+
+case $xinitrc_uim_enable in
+    [Yy][Ee][Ss])
+    if test -x "$fink_prefix/bin/uim-xim"; then
+        "$fink_prefix/bin/uim-xim" --engine=$xinitrc_uim_engine &
+    fi
+    ;;
+esac
+case $xinitrc_uim_toolbar_enable in
+    [Yy][Ee][Ss])
+    if test -x "$fink_prefix/bin/uim-toolbar-gtk"; then
+        "$fink_prefix/bin/uim-toolbar-gtk" &
+    fi
+    ;;
+esac
