diff -Nurd -x'*~' kcoloredit-2.0.0.orig/src/CMakeLists.txt kcoloredit-2.0.0/src/CMakeLists.txt
--- kcoloredit-2.0.0.orig/src/CMakeLists.txt	2009-03-31 14:57:48.000000000 -0400
+++ kcoloredit-2.0.0/src/CMakeLists.txt	2012-06-03 12:15:03.000000000 -0400
@@ -48,7 +48,7 @@
 target_link_libraries (
   kcoloredit
 
-  ${KDE4_KFILE_LIBS}
+  ${KDE4_KFILE_LIBS} ${X11_LIBRARIES}
 )
 
 install (TARGETS kcoloredit  ${INSTALL_TARGETS_DEFAULT_ARGS} )
