diff -Nurd -x'*~' kftpgrabber-0.8.1.orig/kftpgrabber/src/engine/ssl.cpp kftpgrabber-0.8.1/kftpgrabber/src/engine/ssl.cpp
--- kftpgrabber-0.8.1.orig/kftpgrabber/src/engine/ssl.cpp	2007-04-28 07:35:22.000000000 -0400
+++ kftpgrabber-0.8.1/kftpgrabber/src/engine/ssl.cpp	2012-05-19 23:41:51.000000000 -0400
@@ -41,6 +41,7 @@
 
 #include <openssl/ssl.h>
 #include <openssl/x509.h>
+#include <openssl/opensslv.h>
 
 #include <unistd.h>
 
