diff -Nurd -x'*~' kipi-plugins-0.1.2.orig/configure kipi-plugins-0.1.2/configure
--- kipi-plugins-0.1.2.orig/configure	2006-07-08 19:51:25.000000000 -0400
+++ kipi-plugins-0.1.2/configure	2008-05-18 21:57:40.000000000 -0400
@@ -33859,7 +33859,7 @@
 fi
 
 if test -n "${MAGICK_CONFIG}"; then
-  LIBMAGICK_CFLAGS="`$MAGICK_CONFIG --cxxflags`"
+  LIBMAGICK_CFLAGS="`$MAGICK_CONFIG --cppflags`"
 
   LIBMAGICK_LIBS="`$MAGICK_CONFIG --libs`"
 
diff -Nurd -x'*~' kipi-plugins-0.1.2.orig/kipi-plugins/jpeglossless/Makefile.in kipi-plugins-0.1.2/kipi-plugins/jpeglossless/Makefile.in
--- kipi-plugins-0.1.2.orig/kipi-plugins/jpeglossless/Makefile.in	2006-07-08 19:51:46.000000000 -0400
+++ kipi-plugins-0.1.2/kipi-plugins/jpeglossless/Makefile.in	2008-05-18 21:42:38.000000000 -0400
@@ -401,7 +401,7 @@
 xdg_menudir = @xdg_menudir@
 #>- METASOURCES = AUTO
 KDE_CXXFLAGS = $(USE_EXCEPTIONS)
-INCLUDES = $(LIBKIPI_CFLAGS) $(LIBKEXIF_CFLAGS) $(all_includes)
+INCLUDES = $(LIBMAGICK_CFLAGS) $(LIBKIPI_CFLAGS) $(LIBKEXIF_CFLAGS) $(all_includes)
 
 # --enable-final triggers: http://bugs.kde.org/show_bug.cgi?id=126326
 # digikam: camera download: auto-rotated images loose EXIF info ...
@@ -415,7 +415,7 @@
 		                     actionthread.cpp utils.cpp transupp.c \
 		                     imagerotate.cpp convert2grayscale.cpp imageflip.cpp 
 
-kipiplugin_jpeglossless_la_LIBADD = -ljpeg -lMagick++ $(LIBKEXIF_LIBS) $(LIBKIPI_LIBS) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT)
+kipiplugin_jpeglossless_la_LIBADD = -ljpeg $(LIBMAGICK_LIBS) $(LIBKEXIF_LIBS) $(LIBKIPI_LIBS) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT)
 kipiplugin_jpeglossless_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 kde_services_DATA = kipiplugin_jpeglossless.desktop
 kipijpeglosslessicondir = $(kde_datadir)/kipiplugin_jpeglossless/icons
