diff -Nurd -x'*~' kmplayer-0.10.0.orig/configure kmplayer-0.10.0/configure
--- kmplayer-0.10.0.orig/configure	2007-09-28 18:20:54.000000000 -0400
+++ kmplayer-0.10.0/configure	2012-06-20 12:29:49.000000000 -0400
@@ -1703,7 +1703,7 @@
 			  (and sometimes confusing) to the casual installer
   --enable-koffice-plugin enable koffice plugin
   --enable-expat use expat libs
-  --enable-npplayer    Enables building with npp support
+  --enable-nspr      Enables building with npp support
   --enable-cairo     Enables building with cairo support
 
 Optional Packages:
diff -Nurd -x'*~' kmplayer-0.10.0.orig/src/Makefile.in kmplayer-0.10.0/src/Makefile.in
--- kmplayer-0.10.0.orig/src/Makefile.in	2007-09-28 18:20:57.000000000 -0400
+++ kmplayer-0.10.0/src/Makefile.in	2012-06-20 12:30:49.000000000 -0400
@@ -83,7 +83,7 @@
 kmplayer_la_LINK = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CXXLD) $(AM_CXXFLAGS) \
 	$(CXXFLAGS) $(KDE_CXXFLAGS) $(kmplayer_la_LDFLAGS) $(LDFLAGS) -o $@
-libkmplayerbackend_la_LIBADD =
+libkmplayerbackend_la_LIBADD = $(LIB_QT)
 am_libkmplayerbackend_la_OBJECTS =
 #>- libkmplayerbackend_la_OBJECTS = $(am_libkmplayerbackend_la_OBJECTS)
 #>+ 4
@@ -569,11 +569,11 @@
 #>- libkmplayerbackend_la_SOURCES = kmplayer_backend.skel kmplayer_callback.stub
 #>+ 1
 libkmplayerbackend_la_SOURCES=  kmplayer_backend_skel.cpp kmplayer_callback_stub.cpp
-kxineplayer_LDADD = libkmplayerbackend.la $(LIB_XINE) -lDCOP
+kxineplayer_LDADD = libkmplayerbackend.la $(LIB_XINE) -lDCOP $(LIB_QT)
 kxineplayer_CFLAGS = $(CFLAGS_XINE)
 kxineplayer_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 kxineplayer_SOURCES = xineplayer.cpp
-kxvplayer_LDADD = libkmplayerbackend.la -lDCOP -lXv
+kxvplayer_LDADD = libkmplayerbackend.la -lDCOP $(LIB_QT) -lXv
 kxvplayer_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 kxvplayer_SOURCES = xvplayer.cpp
 kgstplayer_LDADD = libkmplayerbackend.la $(LIB_GST) $(LIB_GST_PLUGINS) -lgstinterfaces-0.10 -lDCOP
@@ -1166,7 +1166,7 @@
 .NOEXPORT:
 
 #>+ 2
-libkdeinit_kmplayer_la_LIBADD = libkmplayercommon.la
+libkdeinit_kmplayer_la_LIBADD = libkmplayercommon.la $(LIB_KIO) $(LIB_KDEUI) $(LIB_KDECORE) -lDCOP $(LIB_QT)
 
 #>+ 2
 libkdeinit_kmplayer_la_LDFLAGS = -no-undefined -avoid-version $(all_libraries)
