diff -Nurd -x'*~' kooldock.orig/src/Makefile.am kooldock/src/Makefile.am
--- kooldock.orig/src/Makefile.am	2007-08-27 05:03:13.000000000 -0400
+++ kooldock/src/Makefile.am	2012-01-29 12:20:19.000000000 -0500
@@ -5,7 +5,7 @@
 
 kooldock_SOURCES = appProp.ui apppropprg.cpp clip.cpp dinfo.cpp item.cpp \
 	kooldock.cpp main.cpp setupdialog.ui setupdialogprg.cpp xosd.cpp
-kooldock_LDADD = $(LIB_KIO) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT) $(LIBSOCKET)
+kooldock_LDADD = $(LIB_KIO) $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_QT) $(LIBSOCKET) -lDCOP -lkdefx
 
 
 EXTRA_DIST = main.cpp kooldock.cpp kooldock.h item.cpp item.h 
