diff -Naur kphotoalbum-3.1.0/src/ImageManager/RawImageDecoder.cpp kphotoalbum-3.1.0.patch/src/ImageManager/RawImageDecoder.cpp
--- kphotoalbum-3.1.0/src/ImageManager/RawImageDecoder.cpp	2007-11-22 01:23:37.000000000 +0100
+++ kphotoalbum-3.1.0.patch/src/ImageManager/RawImageDecoder.cpp	2008-03-15 21:54:45.000000000 +0100
@@ -91,6 +91,7 @@
 void RAWImageDecoder::_initializeExtensionLists()
 {
   static bool extensionListsInitialized = 0;
+  return;
   if (! extensionListsInitialized) {
 #ifdef HASKDCRAW
       _rawExtensions = QStringList::split( ' ', QString::fromAscii(raw_file_extentions) );
