--- kynaptic-0.55+cvs20050115-orig/src/common/rpackagefilter.h	2005-01-15 14:19:23.000000000 -0500
+++ kynaptic-0.55+cvs20050115/src/common/rpackagefilter.h	2012-02-22 12:50:52.000000000 -0500
@@ -130,7 +130,7 @@
    bool and_mode; // patterns are applied in "AND" mode if true, "OR" if false
 
    inline bool filterName(Pattern pat, RPackage *pkg);
-   inline bool RPatternPackageFilter::filterVersion(Pattern pat, RPackage *pkg);
+   inline bool filterVersion(Pattern pat, RPackage *pkg);
    inline bool filterDescription(Pattern pat, RPackage *pkg);
    inline bool filterMaintainer(Pattern pat, RPackage *pkg);
    inline bool filterDepends(Pattern pat, RPackage *pkg, 
--- kynaptic-0.55+cvs20050115-orig/src/common/rpackagelister.cc	2005-01-15 14:19:23.000000000 -0500
+++ kynaptic-0.55+cvs20050115/src/common/rpackagelister.cc	2012-02-22 12:57:39.000000000 -0500
@@ -611,7 +611,7 @@
    bool _ascent;
    RPackageStatus _status;
  public:
-   supportedSortFunc::supportedSortFunc(bool ascent, RPackageStatus &s) 
+   supportedSortFunc(bool ascent, RPackageStatus &s) 
       : _ascent(ascent), _status(s) {};
    bool operator() (RPackage *x, RPackage *y) {
       if(_ascent)
