diff -uNr kdeartwork-3.3.1/admin/ltmain.sh kdeartwork-3.3.1-new/admin/ltmain.sh
--- kdeartwork-3.3.1/admin/ltmain.sh	Tue Jun 15 12:47:06 2004
+++ kdeartwork-3.3.1-new/admin/ltmain.sh	Mon Oct 18 16:19:39 2004
@@ -4397,6 +4397,7 @@
 	# We should set the shlibpath_var
 	rpath=
 	for dir in $temp_rpath; do
+     if test "$dir" != "@FINKPREFIX@/lib"; then
 	  case $dir in
 	  [\\/]* | [A-Za-z]:[\\/]*)
 	    # Absolute path.
@@ -4407,6 +4408,7 @@
 	    rpath="$rpath\$thisdir/$dir:"
 	    ;;
 	  esac
+     fi
 	done
 	temp_rpath="$rpath"
       fi
diff -uNr kdeartwork-3.3.1/configure kdeartwork-3.3.1-new/configure
--- kdeartwork-3.3.1/configure	Sun Oct  3 10:14:44 2004
+++ kdeartwork-3.3.1-new/configure	Mon Oct 18 16:19:39 2004
@@ -7974,6 +7974,19 @@
     lt_cv_sys_max_cmd_len=8192;
     ;;
 
+  netbsd* | freebsd* | openbsd* | darwin* )
+    # This has been around since 386BSD, at least.  Likely further.
+    if test -x /sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/sbin/sysctl -n kern.argmax`
+    elif test -x /usr/sbin/sysctl; then
+      lt_cv_sys_max_cmd_len=`/usr/sbin/sysctl -n kern.argmax`
+    else
+      lt_cv_sys_max_cmd_len=65536 # usable default for *BSD
+    fi
+    # And add a safety zone
+    lt_cv_sys_max_cmd_len=`expr $lt_cv_sys_max_cmd_len \/ 4`
+    ;;
+
  *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -9617,10 +9630,21 @@
       archive_cmds_need_lc=no
       case "$host_os" in
       rhapsody* | darwin1.[012])
-	allow_undefined_flag='-undefined suppress'
+	allow_undefined_flag='-Wl,-undefined -Wl,suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -11470,7 +11494,18 @@
       allow_undefined_flag='-undefined suppress'
       ;;
     *) # Darwin 1.3 on
-      test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
       ;;
     esac
     output_verbose_link_cmd='echo'
@@ -12521,7 +12556,18 @@
       allow_undefined_flag_CXX='-undefined suppress'
       ;;
     *) # Darwin 1.3 on
-      test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_CXX='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_CXX='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_CXX='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_CXX='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
       ;;
     esac
     lt_int_apple_cc_single_mod=no
@@ -16344,7 +16390,18 @@
 	allow_undefined_flag_F77='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_F77='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_F77='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_F77='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_F77='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
@@ -18626,7 +18683,18 @@
 	allow_undefined_flag_GCJ='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	test -z ${LD_TWOLEVEL_NAMESPACE} && allow_undefined_flag_GCJ='-flat_namespace -undefined suppress'
+	if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
+	  allow_undefined_flag_GCJ='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	else
+	  case ${MACOSX_DEPLOYMENT_TARGET} in
+	    10.[012])
+	      allow_undefined_flag_GCJ='-Wl,-flat_namespace -Wl,-undefined -Wl,suppress'
+	      ;;
+	    10.*)
+	      allow_undefined_flag_GCJ='-Wl,-undefined -Wl,dynamic_lookup'
+	      ;;
+	  esac
+	fi
 	;;
       esac
     	lt_int_apple_cc_single_mod=no
diff -uNr kdeartwork-3.3.1/kscreensaver/kdesavers/Makefile.in kdeartwork-3.3.1-new/kscreensaver/kdesavers/Makefile.in
--- kdeartwork-3.3.1/kscreensaver/kdesavers/Makefile.in	Sun Oct  3 10:14:39 2004
+++ kdeartwork-3.3.1-new/kscreensaver/kdesavers/Makefile.in	Mon Oct 18 16:20:33 2004
@@ -21,7 +21,7 @@
 
 
 
-SOURCES = $(kbanner_kss_SOURCES) $(kblob_kss_SOURCES) $(kclock_kss_SOURCES) $(keuphoria_kss_SOURCES) $(kfiresaver_kss_SOURCES) $(kflux_kss_SOURCES) $(kfountain_kss_SOURCES) $(kgravity_kss_SOURCES) $(klines_kss_SOURCES) $(klorenz_kss_SOURCES) $(kpendulum_kss_SOURCES) $(kpolygon_kss_SOURCES) $(krotation_kss_SOURCES) $(kscience_kss_SOURCES) $(kslideshow_kss_SOURCES) $(ksolarwinds_kss_SOURCES) $(kvm_kss_SOURCES) $(kwave_kss_SOURCES)
+SOURCES = $(kbanner_kss_SOURCES) $(kblob_kss_SOURCES) $(kclock_kss_SOURCES) $(keuphoria_kss_SOURCES) $(kfiresaver_kss_SOURCES) $(kflux_kss_SOURCES) $(kfountain_kss_SOURCES) $(kgravity_kss_SOURCES) $(klines_kss_SOURCES) $(klorenz_kss_SOURCES) $(kpendulum_kss_SOURCES) $(kpolygon_kss_SOURCES) $(kscience_kss_SOURCES) $(kslideshow_kss_SOURCES) $(ksolarwinds_kss_SOURCES) $(kvm_kss_SOURCES) $(kwave_kss_SOURCES)
 
 srcdir = @srcdir@
 top_srcdir = @top_srcdir@
@@ -69,7 +69,6 @@
 @COMPILE_GL_KSAVERS_TRUE@	kflux.kss$(EXEEXT) \
 @COMPILE_GL_KSAVERS_TRUE@	keuphoria.kss$(EXEEXT) \
 @COMPILE_GL_KSAVERS_TRUE@	ksolarwinds.kss$(EXEEXT) \
-@COMPILE_GL_KSAVERS_TRUE@	krotation.kss$(EXEEXT) \
 @COMPILE_GL_KSAVERS_TRUE@	kpendulum.kss$(EXEEXT) \
 @COMPILE_GL_KSAVERS_TRUE@	$(am__EXEEXT_2)
 @COMPILE_LIBART_KSAVERS_TRUE@am__EXEEXT_4 = kclock.kss$(EXEEXT)
@@ -236,7 +235,7 @@
 	$(kfountain_kss_SOURCES) $(kgravity_kss_SOURCES) \
 	$(klines_kss_SOURCES) $(klorenz_kss_SOURCES) \
 	$(kpendulum_kss_SOURCES) $(kpolygon_kss_SOURCES) \
-	$(krotation_kss_SOURCES) $(kscience_kss_SOURCES) \
+	$(kscience_kss_SOURCES) \
 	$(kslideshow_kss_SOURCES) $(ksolarwinds_kss_SOURCES) \
 	$(kvm_kss_SOURCES) $(kwave_kss_SOURCES)
 DIST_SOURCES = $(kbanner_kss_SOURCES) $(kblob_kss_SOURCES) \
@@ -245,7 +244,7 @@
 	$(kfountain_kss_SOURCES) $(kgravity_kss_SOURCES) \
 	$(klines_kss_SOURCES) $(klorenz_kss_SOURCES) \
 	$(kpendulum_kss_SOURCES) $(kpolygon_kss_SOURCES) \
-	$(krotation_kss_SOURCES) $(kscience_kss_SOURCES) \
+	$(kscience_kss_SOURCES) \
 	$(kslideshow_kss_SOURCES) $(ksolarwinds_kss_SOURCES) \
 	$(kvm_kss_SOURCES) $(kwave_kss_SOURCES)
 #>- RECURSIVE_TARGETS = all-recursive check-recursive dvi-recursive \
@@ -546,7 +545,7 @@
 LIBART_KSAVERS_LDADD = $(AM_LDADD) $(LIBART_LIBS)
 @COMPILE_ARTS_KSAVERS_TRUE@ARTS_KSAVERS = kfiresaver.kss
 @COMPILE_ARTS_KSAVERS_TRUE@ARTS_DESKTOP_FILES = KFiresaver.desktop
-@COMPILE_GL_KSAVERS_TRUE@GL_KSAVERS = kfountain.kss kwave.kss kgravity.kss kflux.kss keuphoria.kss ksolarwinds.kss krotation.kss kpendulum.kss $(ARTS_KSAVERS)
+@COMPILE_GL_KSAVERS_TRUE@GL_KSAVERS = kfountain.kss kwave.kss kgravity.kss kflux.kss keuphoria.kss ksolarwinds.kss kpendulum.kss $(ARTS_KSAVERS)
 @COMPILE_GL_KSAVERS_TRUE@GL_DESKTOP_FILES = KGravity.desktop KWave.desktop KFountain.desktop KFlux.desktop KEuphoria.desktop KSolarWinds.desktop KRotation.desktop KPendulum.desktop $(ARTS_DESKTOP_FILES)
 GL_KSAVERS_LDADD = $(AM_LDADD) $(GLLIB)
 
@@ -1420,19 +1419,19 @@
 
 #>+ 3
 final:
-	$(MAKE) kfiresaver_kss_OBJECTS="$(kfiresaver_kss_final_OBJECTS)" kfountain_kss_OBJECTS="$(kfountain_kss_final_OBJECTS)" kgravity_kss_OBJECTS="$(kgravity_kss_final_OBJECTS)" kpendulum_kss_OBJECTS="$(kpendulum_kss_final_OBJECTS)" krotation_kss_OBJECTS="$(krotation_kss_final_OBJECTS)" kslideshow_kss_OBJECTS="$(kslideshow_kss_final_OBJECTS)" kvm_kss_OBJECTS="$(kvm_kss_final_OBJECTS)" kwave_kss_OBJECTS="$(kwave_kss_final_OBJECTS)" all-am
+	$(MAKE) kfiresaver_kss_OBJECTS="$(kfiresaver_kss_final_OBJECTS)" kfountain_kss_OBJECTS="$(kfountain_kss_final_OBJECTS)" kgravity_kss_OBJECTS="$(kgravity_kss_final_OBJECTS)" kpendulum_kss_OBJECTS="$(kpendulum_kss_final_OBJECTS)" kslideshow_kss_OBJECTS="$(kslideshow_kss_final_OBJECTS)" kvm_kss_OBJECTS="$(kvm_kss_final_OBJECTS)" kwave_kss_OBJECTS="$(kwave_kss_final_OBJECTS)" all-am
 
 #>+ 3
 final-install:
-	$(MAKE) kfiresaver_kss_OBJECTS="$(kfiresaver_kss_final_OBJECTS)" kfountain_kss_OBJECTS="$(kfountain_kss_final_OBJECTS)" kgravity_kss_OBJECTS="$(kgravity_kss_final_OBJECTS)" kpendulum_kss_OBJECTS="$(kpendulum_kss_final_OBJECTS)" krotation_kss_OBJECTS="$(krotation_kss_final_OBJECTS)" kslideshow_kss_OBJECTS="$(kslideshow_kss_final_OBJECTS)" kvm_kss_OBJECTS="$(kvm_kss_final_OBJECTS)" kwave_kss_OBJECTS="$(kwave_kss_final_OBJECTS)" install-am
+	$(MAKE) kfiresaver_kss_OBJECTS="$(kfiresaver_kss_final_OBJECTS)" kfountain_kss_OBJECTS="$(kfountain_kss_final_OBJECTS)" kgravity_kss_OBJECTS="$(kgravity_kss_final_OBJECTS)" kpendulum_kss_OBJECTS="$(kpendulum_kss_final_OBJECTS)" kslideshow_kss_OBJECTS="$(kslideshow_kss_final_OBJECTS)" kvm_kss_OBJECTS="$(kvm_kss_final_OBJECTS)" kwave_kss_OBJECTS="$(kwave_kss_final_OBJECTS)" install-am
 
 #>+ 3
 no-final:
-	$(MAKE) kfiresaver_kss_OBJECTS="$(kfiresaver_kss_nofinal_OBJECTS)" kfountain_kss_OBJECTS="$(kfountain_kss_nofinal_OBJECTS)" kgravity_kss_OBJECTS="$(kgravity_kss_nofinal_OBJECTS)" kpendulum_kss_OBJECTS="$(kpendulum_kss_nofinal_OBJECTS)" krotation_kss_OBJECTS="$(krotation_kss_nofinal_OBJECTS)" kslideshow_kss_OBJECTS="$(kslideshow_kss_nofinal_OBJECTS)" kvm_kss_OBJECTS="$(kvm_kss_nofinal_OBJECTS)" kwave_kss_OBJECTS="$(kwave_kss_nofinal_OBJECTS)" all-am
+	$(MAKE) kfiresaver_kss_OBJECTS="$(kfiresaver_kss_nofinal_OBJECTS)" kfountain_kss_OBJECTS="$(kfountain_kss_nofinal_OBJECTS)" kgravity_kss_OBJECTS="$(kgravity_kss_nofinal_OBJECTS)" kpendulum_kss_OBJECTS="$(kpendulum_kss_nofinal_OBJECTS)" kslideshow_kss_OBJECTS="$(kslideshow_kss_nofinal_OBJECTS)" kvm_kss_OBJECTS="$(kvm_kss_nofinal_OBJECTS)" kwave_kss_OBJECTS="$(kwave_kss_nofinal_OBJECTS)" all-am
 
 #>+ 3
 no-final-install:
-	$(MAKE) kfiresaver_kss_OBJECTS="$(kfiresaver_kss_nofinal_OBJECTS)" kfountain_kss_OBJECTS="$(kfountain_kss_nofinal_OBJECTS)" kgravity_kss_OBJECTS="$(kgravity_kss_nofinal_OBJECTS)" kpendulum_kss_OBJECTS="$(kpendulum_kss_nofinal_OBJECTS)" krotation_kss_OBJECTS="$(krotation_kss_nofinal_OBJECTS)" kslideshow_kss_OBJECTS="$(kslideshow_kss_nofinal_OBJECTS)" kvm_kss_OBJECTS="$(kvm_kss_nofinal_OBJECTS)" kwave_kss_OBJECTS="$(kwave_kss_nofinal_OBJECTS)" install-am
+	$(MAKE) kfiresaver_kss_OBJECTS="$(kfiresaver_kss_nofinal_OBJECTS)" kfountain_kss_OBJECTS="$(kfountain_kss_nofinal_OBJECTS)" kgravity_kss_OBJECTS="$(kgravity_kss_nofinal_OBJECTS)" kpendulum_kss_OBJECTS="$(kpendulum_kss_nofinal_OBJECTS)" kslideshow_kss_OBJECTS="$(kslideshow_kss_nofinal_OBJECTS)" kvm_kss_OBJECTS="$(kvm_kss_nofinal_OBJECTS)" kwave_kss_OBJECTS="$(kwave_kss_nofinal_OBJECTS)" install-am
 
 #>+ 3
 cvs-clean:
