diff -uNr ../kdelibs-3.2.0/dcop/dcopserver.cpp ./dcop/dcopserver.cpp
--- ../kdelibs-3.2.0/dcop/dcopserver.cpp	Sun Oct 26 05:53:01 2003
+++ ./dcop/dcopserver.cpp	Wed Feb  4 11:10:48 2004
@@ -931,6 +931,7 @@
     return 1;
 }
 
+static int pipeOfDeath[2];
 
 static void sighandler(int sig)
 {
@@ -939,8 +940,7 @@
 	return;
     }
 
-    qApp->quit();
-    //exit(0);
+    write(pipeOfDeath[1], "x", 1);
 }
 
 DCOPServer::DCOPServer(bool _suicide)
@@ -1629,6 +1629,8 @@
 	    return 0; // get rid of controlling terminal
     }
 
+    pipe(pipeOfDeath);
+
     signal(SIGHUP, sighandler);
     signal(SIGTERM, sighandler);
     signal(SIGPIPE, SIG_IGN);
@@ -1636,7 +1638,10 @@
     putenv(strdup("SESSION_MANAGER="));
 
     QApplication a( argc, argv, false );
-
+    
+    QSocketNotifier DEATH(pipeOfDeath[0], QSocketNotifier::Read, 0, 0);
+    a.connect(&DEATH, SIGNAL(activated(int)), SLOT(quit()));
+    
     IceSetIOErrorHandler (IoErrorHandler );
     DCOPServer *server = new DCOPServer(suicide); // this sets the_server
 
diff -uNr ../kdelibs-3.2.0/interfaces/kscript/sample/Makefile.am ./interfaces/kscript/sample/Makefile.am
--- ../kdelibs-3.2.0/interfaces/kscript/sample/Makefile.am	Thu Jan 10 11:56:07 2002
+++ ./interfaces/kscript/sample/Makefile.am	Wed Feb  4 11:10:48 2004
@@ -1,9 +1,9 @@
 INCLUDES = -I$(top_srcdir)/kio -I$(top_srcdir)/interfaces -I$(srcdir) -I$(top_srcdir) -I$(top_srcdir)/interfaces/kscript $(all_includes)
 
-lib_LTLIBRARIES = libshellscript.la
+kde_module_LTLIBRARIES = libshellscript.la
 
 libshellscript_la_SOURCES = shellscript.cpp
-libshellscript_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined
+libshellscript_la_LDFLAGS = $(all_libraries) -module $(KDE_PLUGIN) -no-undefined -avoid-version
 libshellscript_la_LIBADD = ../libkscript.la
 
 
diff -uNr ../kdelibs-3.2.0/kabc/vcardparser/vcardparser.cpp ./kabc/vcardparser/vcardparser.cpp
--- ../kdelibs-3.2.0/kabc/vcardparser/vcardparser.cpp	Sun Jan 25 12:24:31 2004
+++ ./kabc/vcardparser/vcardparser.cpp	Wed Feb  4 11:10:48 2004
@@ -94,7 +94,8 @@
         if ( params.findIndex( "encoding" ) != -1 ) { // have to decode the data
           QByteArray input, output;
           input = value.local8Bit();
-          if ( vCardLine.parameter( "encoding" ).lower() == "b" )
+          if ( vCardLine.parameter( "encoding" ).lower() == "b" ||
+               vCardLine.parameter( "encoding" ).lower() == "base64" )
             KCodecs::base64Decode( input, output );
           else if ( vCardLine.parameter( "encoding" ).lower() == "quoted-printable" )
             KCodecs::quotedPrintableDecode( input, output );
diff -uNr ../kdelibs-3.2.0/kdecore/Makefile.am ./kdecore/Makefile.am
--- ../kdelibs-3.2.0/kdecore/Makefile.am	Sun Nov 30 04:46:42 2003
+++ ./kdecore/Makefile.am	Wed Feb  4 11:10:47 2004
@@ -24,7 +24,7 @@
 
 if include_SVGICONS
 SVGICONS=svgicons
-SVGICON_LIB=svgicons/libkdesvgicons.la
+SVGICON_LIB=svgicons/libkdesvgicons.la $(LIBZ)
 endif
 
 SUBDIRS = malloc $(SVGICONS) . kconfig_compiler tests
diff -uNr ../kdelibs-3.2.0/kdecore/kconfig_compiler/kconfig_compiler.cpp ./kdecore/kconfig_compiler/kconfig_compiler.cpp
--- ../kdelibs-3.2.0/kdecore/kconfig_compiler/kconfig_compiler.cpp	Sat Jan 17 07:53:00 2004
+++ ./kdecore/kconfig_compiler/kconfig_compiler.cpp	Wed Feb  4 11:10:49 2004
@@ -809,6 +809,7 @@
       QDomNode n2;
       for( n2 = e.firstChild(); !n2.isNull(); n2 = n2.nextSibling() ) {
         QDomElement e2 = n2.toElement();
+        if ( e2.tagName() != "entry" ) continue;
         CfgEntry *entry = parseEntry( group, e2 );
         if ( entry ) entries.append( entry );
         else {
diff -uNr ../kdelibs-3.2.0/kdecore/kconfigbackend.cpp ./kdecore/kconfigbackend.cpp
--- ../kdelibs-3.2.0/kdecore/kconfigbackend.cpp	Sun Nov 30 04:46:43 2003
+++ ./kdecore/kconfigbackend.cpp	Wed Feb  4 11:10:48 2004
@@ -32,6 +32,7 @@
 #endif
 #include <fcntl.h>
 #include <signal.h>
+#include <setjmp.h>
 
 #include <qdir.h>
 #include <qfileinfo.h>
@@ -327,8 +328,11 @@
     }
 
     bFileImmutable = false;
-    QStringList list = KGlobal::dirs()->
-      findAllResources(resType, mfileName);
+    QStringList list;
+    if ( mfileName[0] == '/' )
+      list << mfileName;
+    else
+      list = KGlobal::dirs()->findAllResources(resType, mfileName);
 
     QStringList::ConstIterator it;
 
@@ -367,13 +371,14 @@
 
 #ifdef HAVE_MMAP
 #ifdef SIGBUS
-static const char **mmap_pEof;
+static sigjmp_buf mmap_jmpbuf;
+struct sigaction mmap_old_sigact;
 
-static void mmap_sigbus_handler(int)
-{
-   *mmap_pEof = 0;
-   write(2, "SIGBUS\n", 7);
-   signal(SIGBUS, mmap_sigbus_handler);
+extern "C" {
+   static void mmap_sigbus_handler(int)
+   {
+      siglongjmp (mmap_jmpbuf, 1);
+   }
 }
 #endif
 #endif
@@ -382,7 +387,9 @@
 					      KEntryMap *pWriteBackMap,
 					      bool bGlobal, bool bDefault)
 {
-   void (*old_sighandler)(int) = 0;
+   const char *s; // May get clobbered by sigsetjump, but we don't use them afterwards.
+   const char *eof; // May get clobbered by sigsetjump, but we don't use them afterwards.
+   QByteArray data;
 
    if (!rFile.isOpen()) // come back, if you have real work for us ;->
       return;
@@ -394,23 +401,35 @@
 
    QCString aCurrentGroup("<default>");
 
-   const char *s, *eof;
-   QByteArray data;
-
    unsigned int ll = localeString.length();
 
 #ifdef HAVE_MMAP
-   const char *map = ( const char* ) mmap(0, rFile.size(), PROT_READ, MAP_PRIVATE,
+   static volatile const char *map;
+   map = ( const char* ) mmap(0, rFile.size(), PROT_READ, MAP_PRIVATE,
                                           rFile.handle(), 0);
 
    if (map)
    {
-      s = map;
+      s = (const char*) map;
       eof = s + rFile.size();
 
 #ifdef SIGBUS
-      mmap_pEof = &eof;
-      old_sighandler = signal(SIGBUS, mmap_sigbus_handler);
+      struct sigaction act;
+      act.sa_handler = mmap_sigbus_handler;
+      sigemptyset( &act.sa_mask );
+#ifdef SA_ONESHOT
+      act.sa_flags = SA_ONESHOT;
+#else
+      act.sa_flags = SA_RESETHAND;
+#endif      
+      sigaction( SIGBUS, &act, &mmap_old_sigact );
+
+      if (sigsetjmp (mmap_jmpbuf, 1))
+      {
+         munmap(( char* )map, rFile.size());
+         sigaction (SIGBUS, &mmap_old_sigact, 0);
+         return;
+      }
 #endif
    }
    else
@@ -645,7 +664,7 @@
    {
       munmap(( char* )map, rFile.size());
 #ifdef SIGBUS
-      signal(SIGBUS, old_sighandler);
+      sigaction (SIGBUS, &mmap_old_sigact, 0);
 #endif
    }
 #endif
diff -uNr ../kdelibs-3.2.0/kdecore/kglobalsettings.cpp ./kdecore/kglobalsettings.cpp
--- ../kdelibs-3.2.0/kdecore/kglobalsettings.cpp	Mon Oct 27 09:16:00 2003
+++ ./kdecore/kglobalsettings.cpp	Wed Feb  4 11:10:47 2004
@@ -466,7 +466,7 @@
       s_desktopPath->append('/');
 
     // Trash Path
-    *s_trashPath = *s_desktopPath + i18n("Trash") + "/";
+    *s_trashPath = *s_desktopPath + "." + i18n("Trash") + "/";
     *s_trashPath = config->readPathEntry( "Trash" , *s_trashPath);
     if ( !s_trashPath->startsWith("/") )
       s_trashPath->prepend( QDir::homeDirPath() + "/" );
diff -uNr ../kdelibs-3.2.0/kdecore/kidna.cpp ./kdecore/kidna.cpp
--- ../kdelibs-3.2.0/kdecore/kidna.cpp	Tue Jun  3 04:16:25 2003
+++ ./kdecore/kidna.cpp	Wed Feb  4 11:10:48 2004
@@ -31,8 +31,8 @@
 static lt_dlhandle KIDNA_lib; // = 0
 static bool KIDNA_lib_load_failed; // = false
 
-typedef int (*KIDNA_utf8_to_ace_t)(const char *, char **);
-typedef int (*KIDNA_utf8ace_to_utf8_t)(const char *, char **);
+typedef int (*KIDNA_utf8_to_ace_t)(const char *, char **, int);
+typedef int (*KIDNA_utf8ace_to_utf8_t)(const char *, char **, int);
 
 static KIDNA_utf8_to_ace_t KIDNA_utf8_to_ace; // = 0
 static KIDNA_utf8ace_to_utf8_t KIDNA_utf8ace_to_utf8; // = 0
@@ -40,7 +40,7 @@
 static void KIDNA_load_lib()
 {
    KIDNA_lib_load_failed = true; // Unless proven otherwise
-   KIDNA_lib = lt_dlopen("/usr/local/lib/libidn.la");
+   KIDNA_lib = lt_dlopen("@PREFIX@/lib/libidn.la");
    if (!KIDNA_lib) 
    {
       KIDNA_lib = lt_dlopen("/usr/lib/libidn.la");
@@ -49,14 +49,14 @@
    if (!KIDNA_lib) 
       return; // Error
 
-   KIDNA_utf8_to_ace = (KIDNA_utf8_to_ace_t) lt_dlsym(KIDNA_lib, "idna_utf8_to_ace");
+   KIDNA_utf8_to_ace = (KIDNA_utf8_to_ace_t) lt_dlsym(KIDNA_lib, "idna_to_ascii_8z");
    if (!KIDNA_utf8_to_ace)
    {
       kdWarning() << "Symbol idna_utf8_to_ace not found." << endl;   
       return; // Error
    }
          
-   KIDNA_utf8ace_to_utf8 = (KIDNA_utf8ace_to_utf8_t) lt_dlsym(KIDNA_lib, "idna_utf8ace_to_utf8");
+   KIDNA_utf8ace_to_utf8 = (KIDNA_utf8ace_to_utf8_t) lt_dlsym(KIDNA_lib, "idna_to_unicode_8z8z");
    if (!KIDNA_utf8ace_to_utf8)
    {
       kdWarning() << "Symbol idna_utf8ace_to_utf8 not found." << endl;   
@@ -96,7 +96,7 @@
       // doesn't like those
       bool bStartsWithDot = (idna[0] == '.');
       char *pOutput;
-      if ((*KIDNA_utf8_to_ace)(idna.utf8().data()+(bStartsWithDot ? 1: 0), &pOutput) == IDNA_SUCCESS)
+      if ((*KIDNA_utf8_to_ace)(idna.utf8().data()+(bStartsWithDot ? 1: 0), &pOutput, 0) == IDNA_SUCCESS)
       {
          QCString result = pOutput;
          free(pOutput);
@@ -142,7 +142,7 @@
       // doesn't like those
       bool bStartsWithDot = (idna[0] == '.');
       char *pOutput;
-      if ((*KIDNA_utf8_to_ace)(idna.utf8().data()+(bStartsWithDot ? 1: 0), &pOutput) == IDNA_SUCCESS)
+      if ((*KIDNA_utf8_to_ace)(idna.utf8().data()+(bStartsWithDot ? 1: 0), &pOutput, 0) == IDNA_SUCCESS)
       {
          QString result(pOutput);
          free(pOutput);
@@ -171,7 +171,7 @@
    else 
    {
       char *pOutput;
-      if ((*KIDNA_utf8ace_to_utf8)(idna.utf8(), &pOutput) == IDNA_SUCCESS)
+      if ((*KIDNA_utf8ace_to_utf8)(idna.utf8(), &pOutput, 0) == IDNA_SUCCESS)
       {
          QString result = QString::fromUtf8(pOutput);
          free(pOutput);
diff -uNr ../kdelibs-3.2.0/kdecore/kmanagerselection.cpp ./kdecore/kmanagerselection.cpp
--- ../kdelibs-3.2.0/kdecore/kmanagerselection.cpp	Sat Jan 17 07:52:59 2004
+++ ./kdecore/kmanagerselection.cpp	Wed Feb  4 11:10:49 2004
@@ -1,6 +1,6 @@
 /****************************************************************************
 
- $Id: kmanagerselection.cpp,v 1.12 2003/12/16 18:55:48 lunakl Exp $
+ $Id: kmanagerselection.cpp,v 1.12.2.1 2004/01/27 09:29:42 lunakl Exp $
 
  Copyright (C) 2003 Lubos Lunak        <l.lunak@kde.org>
 
@@ -460,15 +460,9 @@
             || ev_P->xclient.data.l[ 1 ] != static_cast< long >( selection ))
             return;
 //        kdDebug() << "handling message" << endl;
-        if( ev_P->xclient.data.l[ 2 ] == static_cast< long >( selection_owner ))
-            {
-//            kdDebug() << "already known" << endl;
-            return; // we know this already
-            }
         if( static_cast< long >( owner()) == ev_P->xclient.data.l[ 2 ] )
             {
-//            kdDebug() << "new owner: " << selection_owner << endl;
-            emit newOwner( selection_owner );
+            // owner() emits newOwner() if needed, no need to do it twice
             }
         return;
         }
diff -uNr ../kdelibs-3.2.0/kdecore/kpty.cpp ./kdecore/kpty.cpp
--- ../kdelibs-3.2.0/kdecore/kpty.cpp	Sun Jan 25 12:24:42 2004
+++ ./kdecore/kpty.cpp	Wed Feb  4 11:10:49 2004
@@ -290,7 +290,7 @@
   unlockpt(d->masterFd);
 #endif
 
-  d->slaveFd = ::open(d->ttyName.data(), O_RDWR);
+  d->slaveFd = ::open(d->ttyName.data(), O_RDWR | O_NOCTTY);
   if (d->slaveFd < 0)
   {
     kdWarning(175) << "Can't open slave pseudo teletype" << endl;
diff -uNr ../kdelibs-3.2.0/kdecore/svgicons/Makefile.am ./kdecore/svgicons/Makefile.am
--- ../kdelibs-3.2.0/kdecore/svgicons/Makefile.am	Tue Sep  2 04:47:43 2003
+++ ./kdecore/svgicons/Makefile.am	Wed Feb  4 11:10:48 2004
@@ -4,3 +4,4 @@
 
 include_HEADERS = ksvgiconengine.h
 libkdesvgicons_la_SOURCES = ksvgiconengine.cpp ksvgiconpainter.cpp
+libkdesvgicons_la_LIBADD = $(LIBZ)
diff -uNr ../kdelibs-3.2.0/kdeprint/cups/cupsdconf2/Makefile.am ./kdeprint/cups/cupsdconf2/Makefile.am
--- ../kdelibs-3.2.0/kdeprint/cups/cupsdconf2/Makefile.am	Mon Jun 30 02:04:37 2003
+++ ./kdeprint/cups/cupsdconf2/Makefile.am	Wed Feb  4 11:10:49 2004
@@ -1,24 +1,21 @@
-#$Id: Makefile.am,v 1.10 2003/06/29 16:13:00 reed Exp $
+#$Id: Makefile.am,v 1.10.2.1 2004/01/26 22:06:46 domi Exp $
 
 INCLUDES= -I$(top_srcdir) -I$(top_srcdir)/kio -I$(top_srcdir)/kfile $(all_includes)
 
-bin_PROGRAMS = cupsdconf
-cupsdconf_SOURCES = main.cpp
-
-cupsdconf_LDADD   =  libcupsdconf.la
-cupsdconf_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+bin_PROGRAMS =
+lib_LTLIBRARIES =
+kdeinit_LTLIBRARIES = cupsdconf.la
 
 # library creation
-lib_LTLIBRARIES = libcupsdconf.la
-libcupsdconf_la_SOURCES = cupsdconf.cpp cupsddialog.cpp cupsdpage.cpp cupsdcomment.cpp cupsdsplash.cpp \
+cupsdconf_la_SOURCES = cupsdconf.cpp cupsddialog.cpp cupsdpage.cpp cupsdcomment.cpp cupsdsplash.cpp \
 			  cupsdserverpage.cpp cupsdlogpage.cpp cupsdjobspage.cpp cupsdfilterpage.cpp \
 			  qdirlineedit.cpp cupsddirpage.cpp portdialog.cpp cupsdnetworkpage.cpp \
 			  editlist.cpp cupsdbrowsingpage.cpp browsedialog.cpp cupsdsecuritypage.cpp \
 			  locationdialog.cpp addressdialog.cpp cups-util.c \
-			  qdirmultilineedit.cpp sizewidget.cpp
-libcupsdconf_la_METASOURCES = AUTO
-libcupsdconf_la_LDFLAGS = $(all_libraries) -avoid-version -no-undefined
-libcupsdconf_la_LIBADD = $(LIB_KIO) $(LIB_CUPS)
+			  qdirmultilineedit.cpp sizewidget.cpp main.cpp
+cupsdconf_la_METASOURCES = AUTO
+cupsdconf_la_LDFLAGS = -avoid-version -module $(all_libraries) $(KDE_RPATH)
+cupsdconf_la_LIBADD = $(LIB_KIO) $(LIB_CUPS)
 
 noinst_HEADERS = cupsdconf.h cupsdpage.h cupsddialog.h cupsdcomment.h cupsdsplash.h \
 		 cupsdserverpage.h cupsdlogpage.h cupsdjobspage.h cupsdfilterpage.h \
diff -uNr ../kdelibs-3.2.0/kdeprint/cups/cupsdconf2/main.cpp ./kdeprint/cups/cupsdconf2/main.cpp
--- ../kdelibs-3.2.0/kdeprint/cups/cupsdconf2/main.cpp	Sat Jan 17 07:53:02 2004
+++ ./kdeprint/cups/cupsdconf2/main.cpp	Wed Feb  4 16:53:21 2004
@@ -30,7 +30,7 @@
 	KCmdLineLastOption
 };
 
-int main(int argc, char *argv[])
+extern "C" int kdemain(int argc, char *argv[])
 {
 	KCmdLineArgs::init(argc,argv,"cupsdconf",
 			   I18N_NOOP("A CUPS configuration tool"),
diff -uNr ../kdelibs-3.2.0/kdeprint/cups/kmcupsmanager.cpp ./kdeprint/cups/kmcupsmanager.cpp
--- ../kdelibs-3.2.0/kdeprint/cups/kmcupsmanager.cpp	Sat Jan 17 07:53:02 2004
+++ ./kdeprint/cups/kmcupsmanager.cpp	Wed Feb  4 11:10:49 2004
@@ -736,16 +736,16 @@
 {
 	if (!m_cupsdconf)
 	{
-		m_cupsdconf = KLibLoader::self()->library("libcupsdconf");
+		m_cupsdconf = KLibLoader::self()->library("cupsdconf");
 		if (!m_cupsdconf)
 		{
-			setErrorMsg(i18n("Library libcupsdconf not found. Check your installation."));
+			setErrorMsg(i18n("Library cupsdconf not found. Check your installation."));
 			return NULL;
 		}
 	}
 	void*	func = m_cupsdconf->symbol(name);
 	if (!func)
-		setErrorMsg(i18n("Symbol %1 not found in libcupsdconf library.").arg(name));
+		setErrorMsg(i18n("Symbol %1 not found in cupsdconf library.").arg(name));
 	return func;
 }
 
diff -uNr ../kdelibs-3.2.0/kdeprint/ppdloader.cpp ./kdeprint/ppdloader.cpp
--- ../kdelibs-3.2.0/kdeprint/ppdloader.cpp	Wed Jun 11 10:12:53 2003
+++ ./kdeprint/ppdloader.cpp	Wed Feb  4 11:10:49 2004
@@ -83,7 +83,8 @@
 		}
 		else
 		{
-			l.append( s.mid( p1 ).toFloat() );
+			// ignore the final quote
+			l.append( s.mid( p1, s.length() - p1 - 1 ).toFloat() );
 			break;
 		}
 	}
diff -uNr ../kdelibs-3.2.0/kdeui/Makefile.am ./kdeui/Makefile.am
--- ../kdelibs-3.2.0/kdeui/Makefile.am	Sat Jan 17 07:53:04 2004
+++ ./kdeui/Makefile.am	Wed Feb  4 11:10:48 2004
@@ -23,11 +23,11 @@
 INCLUDES= -I$(top_srcdir)/kdefx -I$(top_srcdir)/interfaces $(all_includes)
 
 lib_LTLIBRARIES = libkdeui.la libkspell.la
-libkdeui_la_LDFLAGS = $(KDE_MT_LDFLAGS) -no-undefined -version-info 6:0:2
+libkdeui_la_LDFLAGS = $(KDE_MT_LDFLAGS) $(all_libraries) -no-undefined -version-info 6:0:2
 # we should get this to work when libtool 1.3 is released
 libkdeui_la_LIBADD = ../kdecore/libkdecore.la
 
-libkspell_la_LDFLAGS = $(KDE_MT_LDFLAGS) -version-info 6:0:2 -no-undefined
+libkspell_la_LDFLAGS = $(KDE_MT_LDFLAGS) $(all_libraries) -version-info 6:0:2 -no-undefined
 libkspell_la_LIBADD = libkdeui.la
 libkspell_la_SOURCES = dummy.cpp
 
diff -uNr ../kdelibs-3.2.0/kdeui/kconfigdialog.cpp ./kdeui/kconfigdialog.cpp
--- ../kdelibs-3.2.0/kdeui/kconfigdialog.cpp	Sat Jan 17 07:53:04 2004
+++ ./kdeui/kconfigdialog.cpp	Wed Feb  4 11:10:49 2004
@@ -55,7 +55,14 @@
 		  parent, name, modal, i18n("Configure"), dialogButtons, defaultButton ),
     d(new KConfigDialogPrivate(dialogType)) 
 {		  
-  openDialogs.insert(name, this);
+  if ( name ) {
+    openDialogs.insert(name, this);
+  } else {
+    QCString genericName;
+    genericName.sprintf("SettingsDialog-%p", this);
+    openDialogs.insert(genericName, this);
+    setName(genericName);
+  }
 
   d->mgr = new KConfigDialogManager(this, config);
 
diff -uNr ../kdelibs-3.2.0/kdeui/ksystemtray.cpp ./kdeui/ksystemtray.cpp
--- ../kdelibs-3.2.0/kdeui/ksystemtray.cpp	Sun Nov 30 04:46:52 2003
+++ ./kdeui/ksystemtray.cpp	Wed Feb  4 11:10:49 2004
@@ -232,7 +232,7 @@
 #if defined Q_WS_X11 && ! defined K_WS_QTONLY
     KWin::WindowInfo info = KWin::windowInfo( pw->winId() );
     // mapped = visible (but possibly obscured)
-    bool mapped = (info.mappingState() != NET::Withdrawn);
+    bool mapped = (info.mappingState() == NET::Visible) && !info.isMinimized();
 //    - not mapped -> show, raise, focus
 //    - mapped
 //        - obscured -> raise, focus
diff -uNr ../kdelibs-3.2.0/khtml/ecma/kjs_window.cpp ./khtml/ecma/kjs_window.cpp
--- ../kdelibs-3.2.0/khtml/ecma/kjs_window.cpp	Sat Jan 17 07:53:12 2004
+++ ./khtml/ecma/kjs_window.cpp	Wed Feb  4 11:10:49 2004
@@ -115,6 +115,7 @@
 } // namespace KJS
 
 #include "kjs_window.lut.h"
+#include "rendering/render_replaced.h"
 
 ////////////////////// Screen Object ////////////////////////
 
@@ -551,10 +552,12 @@
     case InnerHeight:
       if (!m_part->view())
         return Undefined();
+      khtml::RenderWidget::flushWidgetResizes(); // make sure frames have their final size
       return Number(m_part->view()->visibleHeight());
     case InnerWidth:
       if (!m_part->view())
         return Undefined();
+      khtml::RenderWidget::flushWidgetResizes(); // make sure frames have their final size
       return Number(m_part->view()->visibleWidth());
     case Length:
       return Number(m_part->frames().count());
@@ -1425,7 +1428,10 @@
     		&& args.size() == 2 && widget)
     {
       QWidget * tl = widget->topLevelWidget();
-      window->resizeTo( tl, tl->width() + args[0].toInt32(exec), tl->height() + args[1].toInt32(exec) );
+      QRect geom = tl->frameGeometry();
+      window->resizeTo( tl,
+                        geom.width() + args[0].toInt32(exec),
+                        geom.height() + args[1].toInt32(exec) );
     }
     return Undefined();
   }
diff -uNr ../kdelibs-3.2.0/khtml/html/html_inlineimpl.cpp ./khtml/html/html_inlineimpl.cpp
--- ../kdelibs-3.2.0/khtml/html/html_inlineimpl.cpp	Sat Jan 17 07:53:12 2004
+++ ./khtml/html/html_inlineimpl.cpp	Wed Feb  4 11:10:49 2004
@@ -98,10 +98,17 @@
                 khtml::RenderImage *r = static_cast<khtml::RenderImage *>(img->renderer());
                 if(r && e)
                 {
-                    int absx, absy;
+                    KHTMLView* v = getDocument()->view();
+                    int x = e->clientX();
+                    int y = e->clientY();
+                    int absx = 0;
+                    int absy = 0;
+                    if ( v ) {
+                        x += v->contentsX();
+                        y += v->contentsY();
+                    }
                     r->absolutePosition(absx, absy);
-                    int x(e->clientX() - absx), y(e->clientY() - absy);
-                    url += QString("?%1,%2").arg( x ).arg( y );
+                    url += QString("?%1,%2").arg( x - absx ).arg( y - absy );
                 }
                 else {
                     evt->setDefaultHandled();
@@ -240,8 +247,9 @@
             int size;
             switch (num)
             {
-            case -1:
+            case -2:
             case  1: size = CSS_VAL_X_SMALL;  break;
+            case -1:
             case  2: size = CSS_VAL_SMALL;    break;
             case  0: // treat 0 the same as 3, because people
                      // expect it to be between -1 and +1
diff -uNr ../kdelibs-3.2.0/khtml/html/html_objectimpl.cpp ./khtml/html/html_objectimpl.cpp
--- ../kdelibs-3.2.0/khtml/html/html_objectimpl.cpp	Sat Jan 17 07:53:12 2004
+++ ./khtml/html/html_objectimpl.cpp	Wed Feb  4 11:10:49 2004
@@ -101,8 +101,13 @@
 
 void HTMLObjectBaseElementImpl::renderAlternative()
 {
-    // an unbelievable hack. FIXME!!
+    if ( m_renderAlternative ) return;
+    QTimer::singleShot( 0, this, SLOT( slotRenderAlternative() ) );
+}
 
+void HTMLObjectBaseElementImpl::slotRenderAlternative()
+{
+    // an unbelievable hack. FIXME!!
     if ( m_renderAlternative ) return;
 
     // ### there can be a m_render if this is called from our attach indirectly
diff -uNr ../kdelibs-3.2.0/khtml/html/html_objectimpl.h ./khtml/html/html_objectimpl.h
--- ../kdelibs-3.2.0/khtml/html/html_objectimpl.h	Sun Sep 21 04:39:15 2003
+++ ./khtml/html/html_objectimpl.h	Wed Feb  4 11:10:49 2004
@@ -19,7 +19,7 @@
  * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  *
- * $Id: html_objectimpl.h,v 1.55 2003/09/19 15:18:31 vriezen Exp $
+ * $Id: html_objectimpl.h,v 1.55.2.1 2004/01/26 01:54:36 mueller Exp $
  */
 #ifndef HTML_OBJECTIMPL_H
 #define HTML_OBJECTIMPL_H
@@ -67,6 +67,7 @@
 
 protected slots:
     void liveConnectEvent(const unsigned long, const QString&, const KParts::LiveConnectExtension::ArgList&);
+    void slotRenderAlternative();
 
 private:
     KParts::LiveConnectExtension *liveconnect;
diff -uNr ../kdelibs-3.2.0/khtml/java/kjavaappletviewer.cpp ./khtml/java/kjavaappletviewer.cpp
--- ../kdelibs-3.2.0/khtml/java/kjavaappletviewer.cpp	Sat Jan 17 07:53:13 2004
+++ ./khtml/java/kjavaappletviewer.cpp	Wed Feb  4 11:10:49 2004
@@ -359,7 +359,8 @@
     // delay showApplet if size is unknown and m_view not shown
     if (applet->size().width() > 0 || m_view->isVisible())
         w->showApplet ();
-    emit started (0L);
+    if (!applet->failed ())
+        emit started (0L);
     return url.isValid ();
 }
 
diff -uNr ../kdelibs-3.2.0/khtml/khtmlview.cpp ./khtml/khtmlview.cpp
--- ../kdelibs-3.2.0/khtml/khtmlview.cpp	Sun Jan 25 12:25:17 2004
+++ ./khtml/khtmlview.cpp	Wed Feb  4 11:10:49 2004
@@ -80,6 +80,10 @@
 
 //#define DEBUG_NO_PAINT_BUFFER
 
+//#define DEBUG_FLICKER
+
+//#define DEBUG_PIXEL
+
 #define PAINT_BUFFER_HEIGHT 128
 
 using namespace DOM;
@@ -158,6 +162,11 @@
         vmode = QScrollView::AlwaysOff;
         hmode = QScrollView::AlwaysOff;
 #endif
+#ifdef DEBUG_PIXEL
+        timer.start();
+        pixelbooth = 0;
+        repaintbooth = 0;
+#endif
         scrollBarMoved = false;
         ignoreWheelEvents = false;
 	borderX = 30;
@@ -178,7 +187,7 @@
         firstRelayout = true;
         dirtyLayout = false;
         layoutSchedulingEnabled = true;
-        updateRect = QRect();
+        updateRegion = QRegion();
         m_dialogsAllowed = true;
 #ifndef KHTML_NO_CARET
         if (m_caretViewContext) {
@@ -237,6 +246,12 @@
     }
 #endif // KHTML_NO_CARET
 
+#ifdef DEBUG_PIXEL
+    QTime timer;
+    unsigned int pixelbooth;
+    unsigned int repaintbooth;
+#endif
+
     QPainter *tp;
     QPixmap  *paintBuffer;
     QPixmap  *vertPaintBuffer;
@@ -278,7 +293,7 @@
     bool possibleTripleClick;
     bool dirtyLayout;
     bool m_dialogsAllowed;
-    QRect updateRect;
+    QRegion updateRegion;
     KHTMLToolTip *tooltip;
     QPtrDict<QWidget> visibleWidgets;
 #ifndef KHTML_NO_CARET
@@ -436,6 +451,19 @@
 
 void KHTMLView::drawContents( QPainter *p, int ex, int ey, int ew, int eh )
 {
+#ifdef DEBUG_PIXEL
+
+    if ( d->timer.elapsed() > 5000 ) {
+        qDebug( "drawed %d pixels in %d repaints the last %d milliseconds",
+                d->pixelbooth, d->repaintbooth,  d->timer.elapsed() );
+        d->timer.restart();
+        d->pixelbooth = 0;
+        d->repaintbooth = 0;
+    }
+    d->pixelbooth += ew*eh;
+    d->repaintbooth++;
+#endif
+
 //     kdDebug( 6000 ) << "drawContents this="<< this <<" x=" << ex << ",y=" << ey << ",w=" << ew << ",h=" << eh << endl;
     if(!m_part || !m_part->xmlDocImpl() || !m_part->xmlDocImpl()->renderer()) {
         p->fillRect(ex, ey, ew, eh, palette().active().brush(QColorGroup::Base));
@@ -784,6 +812,9 @@
     QCursor c;
     switch ( style ? style->cursor() : CURSOR_AUTO) {
     case CURSOR_AUTO:
+        if ( r && r->isText() )
+            c = KCursor::ibeamCursor();
+
         if ( mev.url.length() && m_part->settings()->changeCursor() )
             c = m_part->urlCursor();
 
@@ -2192,8 +2223,28 @@
     killTimer(d->repaintTimerId);
     d->repaintTimerId = 0;
 
-    updateContents( d->updateRect );
-    d->updateRect = QRect();
+    QMemArray<QRect> rects = d->updateRegion.rects();
+    QRegion updateRegion;
+    if ( rects.size() )
+        updateRegion = rects[0];
+
+    for ( unsigned i = 1; i < rects.size(); ++i ) {
+        QRect obR = updateRegion.boundingRect();
+        QRegion newRegion = updateRegion.unite(rects[i]);
+        if (2*newRegion.boundingRect().height() > 3*obR.height() )
+        //if ( !obR.intersects( rects[i] ) )
+        {
+            repaintContents( obR, false );
+            updateRegion = rects[i];
+        }
+        else
+            updateRegion = newRegion;
+    }
+
+    if ( !updateRegion.isNull() )
+        repaintContents( updateRegion.boundingRect(), false );
+
+    d->updateRegion = QRegion();
 
     if (d->dirtyLayout && !d->visibleWidgets.isEmpty()) {
         QWidget* w;
@@ -2249,30 +2300,21 @@
 //     kdDebug() << "parsing " << parsing << endl;
 //     kdDebug() << "complete " << d->complete << endl;
 
-    int time;
+    int time = parsing ? 300 : ( !d->complete ? 100 : 20 );
 
-    // if complete...
-    if (d->complete)
-        // ...repaint immediately
-        time = 20;
-    else
-    {
-        if (parsing)
-            // not complete and still parsing
-            time = 300;
-        else
-            // not complete, not parsing, extend the timer if it exists
-            // otherwise, repaint immediately
-            time = d->repaintTimerId ? 400 : 20;
-    }
+#ifdef DEBUG_FLICKER
+    QPainter p;
+    p.begin( viewport() );
+
+    int vx, vy;
+    contentsToViewport( x, y, vx, vy );
+    p.fillRect( vx, vy, w, h, Qt::red );
+    p.end();
+#endif
 
-    if (d->repaintTimerId) {
-        killTimer(d->repaintTimerId);
-        d->updateRect = d->updateRect.unite(QRect(x,y,w,h));
-    } else
-        d->updateRect = QRect(x,y,w,h);
+    d->updateRegion = d->updateRegion.unite(QRect(x,y,w,h));
 
-//	kdDebug(6000) << "scheduled repaint for " << d->updateRect << endl;
+    if ( !d->repaintTimerId )
     d->repaintTimerId = startTimer( time );
 
 //     kdDebug() << "starting timer " << time << endl;
diff -uNr ../kdelibs-3.2.0/khtml/misc/loader.cpp ./khtml/misc/loader.cpp
--- ../kdelibs-3.2.0/khtml/misc/loader.cpp	Sat Jan 17 07:53:13 2004
+++ ./khtml/misc/loader.cpp	Wed Feb  4 11:10:49 2004
@@ -337,7 +337,7 @@
 {
 public:
     ImageSource(QByteArray buf)
-        : buffer( buf ), pos( 0 ), eof( false ), rew(false ), rewable( false )
+        : buffer( buf ), pos( 0 ), eof( false ), rew(false ), rewable( true )
         {}
 
     int readyToSend()
diff -uNr ../kdelibs-3.2.0/khtml/rendering/bidi.cpp ./khtml/rendering/bidi.cpp
--- ../kdelibs-3.2.0/khtml/rendering/bidi.cpp	Sun Jan 25 12:25:23 2004
+++ ./khtml/rendering/bidi.cpp	Wed Feb  4 11:10:49 2004
@@ -18,7 +18,7 @@
  * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  *
- * $Id: bidi.cpp,v 1.177.2.1 2004/01/20 05:12:15 mueller Exp $
+ * $Id: bidi.cpp,v 1.177.2.2 2004/01/26 05:58:24 mueller Exp $
  */
 #include "rendering/bidi.h"
 #include "rendering/break_lines.h"
@@ -1137,6 +1137,9 @@
 	    r->box->setWidth(r->box->width() + spaceAdd);
         else
             x += r->box->width() + spaceAdd;
+        if ( x > m_overflowWidth )
+            m_overflowWidth = x;
+
         r = sruns->next();
     }
 
diff -uNr ../kdelibs-3.2.0/khtml/rendering/render_block.cpp ./khtml/rendering/render_block.cpp
--- ../kdelibs-3.2.0/khtml/rendering/render_block.cpp	Sun Jan 25 12:25:23 2004
+++ ./khtml/rendering/render_block.cpp	Wed Feb  4 11:10:50 2004
@@ -1361,10 +1361,14 @@
                                //kdDebug( 6040 ) << " Object width: " << fwidth << " available width: " << ro - lo << endl;
         if (ro - lo < fwidth)
             fwidth = ro - lo; // Never look for more than what will be available.
-        if (o->style()->floating() == FLEFT)
-        {
+
             if ( o->style()->clear() & CLEFT )
                 y = kMax( leftBottom(), y );
+        if ( o->style()->clear() & CRIGHT )
+            y = kMax( rightBottom(), y );
+
+        if (o->style()->floating() == FLEFT)
+        {
             int heightRemainingLeft = 1;
             int heightRemainingRight = 1;
             int fx = leftRelOffset(y,lo, false, &heightRemainingLeft);
@@ -1380,8 +1384,6 @@
         }
         else
         {
-            if ( o->style()->clear() & CRIGHT )
-                y = kMax( rightBottom(), y );
             int heightRemainingLeft = 1;
             int heightRemainingRight = 1;
             int fx = rightRelOffset(y,ro, false, &heightRemainingRight);
@@ -1786,7 +1788,7 @@
 
 bool RenderBlock::checkClear(RenderObject *child)
 {
-    //kdDebug( 6040 ) << "checkClear oldheight=" << m_height << endl;
+    //kdDebug( 6040 ) << "checkClear on child " << child << " oldheight=" << m_height << endl;
     int bottom = 0;
     switch(child->style()->clear())
     {
diff -uNr ../kdelibs-3.2.0/khtml/rendering/render_canvas.cpp ./khtml/rendering/render_canvas.cpp
--- ../kdelibs-3.2.0/khtml/rendering/render_canvas.cpp	Sat Jan 17 07:53:14 2004
+++ ./khtml/rendering/render_canvas.cpp	Wed Feb  4 11:10:50 2004
@@ -131,6 +131,7 @@
 #endif
     if ( recalcMinMax() )
 	recalcMinMaxWidths();
+
 #ifdef SPEED_DEBUG
     kdDebug() << "RenderCanvas::calcMinMax time used=" << qt.elapsed() << endl;
     qt.start();
@@ -140,19 +141,14 @@
     kdDebug() << "RenderCanvas::layout time used=" << qt.elapsed() << endl;
     qt.start();
 #endif
+    KHTMLView::ScrollBarMode vsmode = m_view->vScrollBarMode();
+    KHTMLView::ScrollBarMode hsmode = m_view->hScrollBarMode();
+
     if (!m_printingMode) {
         QSize s = m_view->viewportSize(m_view->contentsWidth(),
                                        m_view->contentsHeight());
         m_viewportWidth = m_width = s.width();
         m_viewportHeight = m_height = s.height();
-    }
-    else {
-        m_width = m_rootWidth;
-        m_height = m_rootHeight;
-    }
-    
-    KHTMLView::ScrollBarMode vsmode = m_view->vScrollBarMode();
-    KHTMLView::ScrollBarMode hsmode = m_view->hScrollBarMode();
     
     if (m_view->verticalScrollBar()->isVisible())
         m_view->setVScrollBarMode(KHTMLView::AlwaysOn);
@@ -163,7 +159,11 @@
         m_view->setHScrollBarMode(KHTMLView::AlwaysOn);
     else
         m_view->setHScrollBarMode(KHTMLView::AlwaysOff);
-
+    }
+    else {
+        m_width = m_rootWidth;
+        m_height = m_rootHeight;
+    }
 
     RenderBlock::layout();
 
@@ -190,8 +190,10 @@
 
     layer()->resize( kMax( docW,int( m_width ) ), kMax( docH,m_height ) );
 
-    m_view->setVScrollBarMode(vsmode);
+    if ( !m_printingMode ) {
     m_view->setHScrollBarMode(hsmode);
+        m_view->setVScrollBarMode(vsmode);
+    }
 
     setLayouted();
 }
diff -uNr ../kdelibs-3.2.0/khtml/rendering/render_image.cpp ./khtml/rendering/render_image.cpp
--- ../kdelibs-3.2.0/khtml/rendering/render_image.cpp	Sat Jan 17 07:53:14 2004
+++ ./khtml/rendering/render_image.cpp	Wed Feb  4 11:10:50 2004
@@ -20,7 +20,7 @@
  * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  *
- * $Id: render_image.cpp,v 1.128 2004/01/16 13:37:16 mueller Exp $
+ * $Id: render_image.cpp,v 1.128.2.1 2004/01/26 01:53:49 mueller Exp $
  */
 //#define DEBUG_LAYOUT
 
@@ -150,8 +150,10 @@
         // lets see if we need to relayout at all..
         int oldwidth = m_width;
         int oldheight = m_height;
+        if ( parent() ) {
         calcWidth();
         calcHeight();
+        }
 
         if(iwchanged || m_width != oldwidth || m_height != oldheight)
             needlayout = true;
@@ -160,6 +162,9 @@
         m_height = oldheight;
     }
 
+    // we're not fully integrated in the tree yet.. we'll come back.
+    if ( !parent() )
+        return;
 
     if(needlayout)
     {
diff -uNr ../kdelibs-3.2.0/khtml/rendering/render_object.cpp ./khtml/rendering/render_object.cpp
--- ../kdelibs-3.2.0/khtml/rendering/render_object.cpp	Sun Jan 25 12:25:24 2004
+++ ./khtml/rendering/render_object.cpp	Wed Feb  4 11:10:50 2004
@@ -336,8 +336,13 @@
 
 int RenderObject::offsetLeft() const
 {
+    if ( isPositioned() )
+        return xPos();
+
+    if ( isBody() )
+        return 0;
+
     int x = xPos();
-    if (!isPositioned()) {
         if (isRelPositioned()) {
             int y = 0;
             static_cast<const RenderBox*>(this)->relativePositionOffset(x, y);
@@ -348,14 +353,26 @@
              curr && curr != offsetPar;
              curr = curr->parent() )
             x += curr->xPos();
-    }
+
+    if ( offsetPar && offsetPar->isBody() )
+        x += offsetPar->xPos();
+
+    // hacky
+    if ( isInline() && firstChild() && firstChild()->isText() )
+        x += static_cast<RenderText*>(firstChild() )->minXPos();
+
     return x;
 }
 
 int RenderObject::offsetTop() const
 {
+    if ( isPositioned() )
+        return yPos();
+
+    if ( isBody() )
+        return 0;
+
     int y = yPos();
-    if (!isPositioned()) {
         if (isRelPositioned()) {
             int x = 0;
             static_cast<const RenderBox*>(this)->relativePositionOffset(x, y);
@@ -365,7 +382,14 @@
              curr && curr != offsetPar;
              curr = curr->parent() )
             y += curr->yPos();
-    }
+
+    if ( offsetPar && offsetPar->isBody() )
+        y += offsetPar->yPos();
+
+    // hacky
+    if ( isInline() && firstChild() && firstChild()->isText() )
+        y += static_cast<RenderText*>(firstChild() )->yPos();
+
     return y;
 }
 
@@ -373,8 +397,7 @@
 {
     bool skipTables = isPositioned() || isRelPositioned();
     RenderObject* curr = parent();
-    while (curr && !curr->isPositioned() && !curr->isRelPositioned() &&
-           !curr->isBody()) {
+    while (curr && !curr->isPositioned() && !curr->isRelPositioned() && !curr->isBody()) {
         if (!skipTables && (curr->isTableCell() || curr->isTable()))
             break;
         curr = curr->parent();
diff -uNr ../kdelibs-3.2.0/khtml/rendering/render_replaced.cpp ./khtml/rendering/render_replaced.cpp
--- ../kdelibs-3.2.0/khtml/rendering/render_replaced.cpp	Sun Jan 18 04:08:13 2004
+++ ./khtml/rendering/render_replaced.cpp	Wed Feb  4 11:10:50 2004
@@ -19,7 +19,7 @@
  * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  *
- * $Id: render_replaced.cpp,v 1.156.2.1 2004/01/17 21:49:25 waba Exp $
+ * $Id: render_replaced.cpp,v 1.156.2.3 2004/01/29 10:14:22 faure Exp $
  */
 #include "render_replaced.h"
 #include "render_canvas.h"
@@ -116,6 +116,7 @@
     // a widget doesn't support being anonymous
     assert(!isAnonymous());
     m_view = node->getDocument()->view();
+    m_resizePending = false;
 
     // this is no real reference counting, its just there
     // to make sure that we're not deleted while we're recursed
@@ -164,11 +165,12 @@
 void  RenderWidget::resizeWidget( int w, int h )
 {
     // ugly hack to limit the maximum size of the widget ( as X11 has problems if
-         // its bigger )
+    // its bigger )
     h = kMin( h, 3072 );
     w = kMin( w, 2000 );
 
     if (m_widget->width() != w || m_widget->height() != h) {
+        m_resizePending = !strcmp(m_widget->name(), "__khtml");
         ref();
         element()->ref();
         QApplication::postEvent( this, new QWidgetResizeEvent( w, h ) );
@@ -180,8 +182,10 @@
 bool RenderWidget::event( QEvent *e )
 {
     if ( m_widget && (e->type() == (QEvent::Type)QWidgetResizeEvent::Type) ) {
+        m_resizePending = false;
         QWidgetResizeEvent *re = static_cast<QWidgetResizeEvent *>(e);
         m_widget->resize( re->w,  re->h );
+        repaint();
     }
     // eat all events - except if this is a frame (in which case KHTMLView handles it all)
     if ( ::qt_cast<KHTMLView *>( m_widget ) )
@@ -189,6 +193,10 @@
     return true;
 }
 
+void RenderWidget::flushWidgetResizes() //static
+{
+    QApplication::sendPostedEvents( 0, QWidgetResizeEvent::Type );
+}
 
 void RenderWidget::setQWidget(QWidget *widget)
 {
@@ -222,6 +230,8 @@
         }
         m_view->setWidgetVisible(this, false);
         m_view->addChild( m_widget, 0, -500000);
+        if ( m_widget ) m_widget->hide();
+        m_resizePending = false;
     }
 }
 
@@ -336,8 +346,7 @@
         return;
 
     // not visible or not even once layouted
-    if (style()->visibility() != VISIBLE || m_y <=  -500000) {
-        m_widget->hide();
+    if (style()->visibility() != VISIBLE || m_y <= -500000 || m_resizePending ) {
         return;
     }
 
diff -uNr ../kdelibs-3.2.0/khtml/rendering/render_replaced.h ./khtml/rendering/render_replaced.h
--- ../kdelibs-3.2.0/khtml/rendering/render_replaced.h	Sun Nov 30 04:47:04 2003
+++ ./khtml/rendering/render_replaced.h	Wed Feb  4 11:10:50 2004
@@ -18,7 +18,7 @@
  * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  *
- * $Id: render_replaced.h,v 1.67 2003/11/17 07:20:10 coolo Exp $
+ * $Id: render_replaced.h,v 1.67.2.2 2004/01/29 10:14:22 faure Exp $
  */
 #ifndef render_replaced_h
 #define render_replaced_h
@@ -108,7 +108,6 @@
     QWidget *widget() const { return m_widget; }
     KHTMLView* view() const { return m_view; }
 
-    void ref() { _ref++; /*qDebug( "ref(%p): width get count is %d", this, _ref);*/ }
     void deref();
 
     static void paintWidget(QPainter *p, QWidget *widget, int x, int y, int w, int h, int tx, int ty);
@@ -118,6 +117,9 @@
     virtual void dump(QTextStream &stream, const QString &ind) const;
 #endif
 
+    // for ECMA to flush all pending resizes
+    static void flushWidgetResizes();
+
 public slots:
     void slotWidgetDestructed();
 
@@ -131,6 +133,8 @@
 
     QWidget *m_widget;
     KHTMLView* m_view;
+
+    bool m_resizePending;
 
 public:
     class EventPropagator : public QWidget {
diff -uNr ../kdelibs-3.2.0/khtml/test_regression.cpp ./khtml/test_regression.cpp
--- ../kdelibs-3.2.0/khtml/test_regression.cpp	Sat Jan 17 07:53:11 2004
+++ ./khtml/test_regression.cpp	Wed Feb  4 11:10:49 2004
@@ -419,6 +419,8 @@
                                                         args->isSet("genoutput"));
     QObject::connect(part->browserExtension(), SIGNAL(openURLRequest(const KURL &, const KParts::URLArgs &)),
 		     regressionTest, SLOT(slotOpenURL(const KURL&, const KParts::URLArgs &)));
+    QObject::connect(part->browserExtension(), SIGNAL(resizeTopLevelWidget( int, int )),
+		     regressionTest, SLOT(resizeTopLevelWidget( int, int )));
 
     bool result;
     if (!args->getOption("test").isNull())
@@ -911,6 +913,13 @@
     }
     ignoreFile.close();
     return false;
+}
+
+void RegressionTest::resizeTopLevelWidget( int w, int h )
+{
+    qApp->mainWidget()->resize( w, h );
+    // Since we're not visible, this doesn't have an immediate effect, QWidget posts the event
+    QApplication::sendPostedEvents( 0, QEvent::Resize );
 }
 
 #include "test_regression.moc"
diff -uNr ../kdelibs-3.2.0/khtml/test_regression.h ./khtml/test_regression.h
--- ../kdelibs-3.2.0/khtml/test_regression.h	Sat Jan 17 07:53:11 2004
+++ ./khtml/test_regression.h	Wed Feb  4 11:10:49 2004
@@ -18,7 +18,7 @@
  * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  *
- * $Id: test_regression.h,v 1.18 2003/12/10 21:28:51 mueller Exp $
+ * $Id: test_regression.h,v 1.18.2.1 2004/01/29 10:14:22 faure Exp $
  */
 
 #ifndef TEST_REGRESSION_H
@@ -160,6 +160,7 @@
 
 private slots:
     void slotOpenURL(const KURL &url, const KParts::URLArgs &args);
+    void resizeTopLevelWidget( int, int );
 
 };
 
diff -uNr ../kdelibs-3.2.0/kio/kfile/kicondialog.cpp ./kio/kfile/kicondialog.cpp
--- ../kdelibs-3.2.0/kio/kfile/kicondialog.cpp	Sun Nov 30 04:47:06 2003
+++ ./kio/kfile/kicondialog.cpp	Wed Feb  4 11:10:50 2004
@@ -1,6 +1,6 @@
 /* vi: ts=8 sts=4 sw=4
  *
- * $Id: kicondialog.cpp,v 1.32 2003/11/25 08:07:05 aseigo Exp $
+ * $Id: kicondialog.cpp,v 1.32.2.1 2004/02/01 19:10:48 lukas Exp $
  *
  * This file is part of the KDE project, module kfile.
  * Copyright (C) 2000 Geert Jansen <jansen@kde.org>
@@ -261,7 +261,7 @@
     mpCombo->insertItem(i18n("Filesystems"));
     mpCombo->insertItem(i18n("Mimetypes"));
     mpCombo->setFixedSize(mpCombo->sizeHint());
-    mpBrowseBut->setFixedSize(QSize(mpCombo->width(), mpCombo->height()+6));
+    mpBrowseBut->setFixedWidth(mpCombo->width());
 
     // Make the dialog a little taller
     incInitialSize(QSize(0,100));
@@ -450,7 +450,7 @@
             d->custom = file;
 	    if ( mType == 1 )
 	      d->customLocation = QFileInfo( file ).dirPath( true );
-            accept();
+            slotOk();
 	}
 	break;
     }
diff -uNr ../kdelibs-3.2.0/kio/kfile/kpropertiesdialog.cpp ./kio/kfile/kpropertiesdialog.cpp
--- ../kdelibs-3.2.0/kio/kfile/kpropertiesdialog.cpp	Sat Jan 17 07:53:16 2004
+++ ./kio/kfile/kpropertiesdialog.cpp	Wed Feb  4 11:10:50 2004
@@ -783,7 +783,6 @@
     KIconButton *iconButton = new KIconButton( d->m_frame );
     iconButton->setFixedSize(70, 70);
     iconButton->setStrictIconSize(false);
-    iconButton->setIconType(KIcon::Desktop, KIcon::Device);
     // This works for everything except Device icons on unmounted devices
     // So we have to really open .desktop files
     QString iconStr = KMimeType::findByURL( properties->kurl(),
@@ -791,10 +790,15 @@
                                                           isLocal );
     if ( bDesktopFile && isLocal )
     {
-      KSimpleConfig config( properties->kurl().path() );
+      KDesktopFile config( properties->kurl().path(), true );
       config.setDesktopGroup();
       iconStr = config.readEntry( "Icon" );
-    }
+      if ( config.hasDeviceType() )
+	iconButton->setIconType( KIcon::Desktop, KIcon::Device );
+      else
+	iconButton->setIconType( KIcon::Desktop, KIcon::Application );
+    } else
+      iconButton->setIconType( KIcon::Desktop, KIcon::FileSystem );
     iconButton->setIcon(iconStr);
     iconArea = iconButton;
     connect( iconButton, SIGNAL( iconChanged(QString) ),
@@ -2868,7 +2872,7 @@
 
   {
      mw->listView->setRootIsDecorated(true);
-     mw->listView->setSelectionMode(QListView::Multi);
+     mw->listView->setSelectionMode(QListView::Extended);
      mw->listView->setAllColumnsShowFocus(true);
      mw->listView->setFullWidth(true);
      mw->listView->setMinimumSize(500,400);
diff -uNr ../kdelibs-3.2.0/kio/kio/job.cpp ./kio/kio/job.cpp
--- ../kdelibs-3.2.0/kio/kio/job.cpp	Sat Jan 17 07:53:17 2004
+++ ./kio/kio/job.cpp	Wed Feb  4 11:10:50 2004
@@ -3063,10 +3063,12 @@
             }
             if ( err )
             {
+                bool renaming = ( m_currentSrcURL.directory() == dest.directory() );
                 m_currentSrcURL=*m_currentStatSrc;
                 m_currentDestURL=m_dest;
 
-                if ( err == ERR_DIR_ALREADY_EXIST || err == ERR_FILE_ALREADY_EXIST )
+                if ( ( err == ERR_DIR_ALREADY_EXIST || err == ERR_FILE_ALREADY_EXIST ) && 
+		     renaming )
                 {
                     if (m_reportTimer)
                         m_reportTimer->stop();
diff -uNr ../kdelibs-3.2.0/kio/kssl/kssl.cc ./kio/kssl/kssl.cc
--- ../kdelibs-3.2.0/kio/kssl/kssl.cc	Sun Oct 26 05:54:16 2003
+++ ./kio/kssl/kssl.cc	Wed Feb  4 11:10:50 2004
@@ -282,7 +282,12 @@
 		return -1;
 
 	if (d->session) {
-		if (1 == d->kossl->SSL_set_session(d->m_ssl,
+		if (static_cast<SSL_SESSION*>(d->session->_session)->sess_cert == 0)
+		{
+			kdDebug(7029) << "Can't reuse session, no certificate." << endl;
+			delete d->session;
+			d->session = 0;
+		} else if (1 == d->kossl->SSL_set_session(d->m_ssl,
 			static_cast<SSL_SESSION*>(d->session->_session))) {
 			kdDebug(7029) << "Session ID is being reused." << endl;
 		} else {
@@ -363,7 +368,12 @@
 		return -1;
 
 	if (d->session) {
-		if (1 == d->kossl->SSL_set_session(d->m_ssl,
+		if (static_cast<SSL_SESSION*>(d->session->_session)->sess_cert == 0)
+		{
+			kdDebug(7029) << "Can't reuse session, no certificate." << endl;
+			delete d->session;
+			d->session = 0;
+		} else if (1 == d->kossl->SSL_set_session(d->m_ssl,
 			static_cast<SSL_SESSION*>(d->session->_session))) {
 			kdDebug(7029) << "Session ID is being reused." << endl;
 		} else {
diff -uNr ../kdelibs-3.2.0/kio/misc/kpac/Makefile.am ./kio/misc/kpac/Makefile.am
--- ../kdelibs-3.2.0/kio/misc/kpac/Makefile.am	Sat Jan 17 07:53:18 2004
+++ ./kio/misc/kpac/Makefile.am	Wed Feb  4 11:10:48 2004
@@ -10,7 +10,7 @@
 kded_proxyscout_la_SOURCES = proxyscout.skel proxyscout.cpp script.cpp \
                              downloader.cpp discovery.cpp
 kded_proxyscout_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_proxyscout_la_LIBADD = $(LIB_KIO) $(top_builddir)/kjs/libkjs.la
+kded_proxyscout_la_LIBADD = $(LIB_KIO) $(top_builddir)/kjs/libkjs.la $(LIBRESOLV)
 
 kpac_dhcp_helper_SOURCES = kpac_dhcp_helper.c
 kpac_dhcp_helper_LDADD = $(LIBSOCKET)
diff -uNr ../kdelibs-3.2.0/kio/misc/kwalletd/kwalletd.cpp ./kio/misc/kwalletd/kwalletd.cpp
--- ../kdelibs-3.2.0/kio/misc/kwalletd/kwalletd.cpp	Sat Jan 17 07:53:19 2004
+++ ./kio/misc/kwalletd/kwalletd.cpp	Wed Feb  4 11:10:50 2004
@@ -65,7 +65,7 @@
 			client = 0L;
 		}
 
-		enum Type { Unknown, Open, ChangePassword };
+		enum Type { Unknown, Open, ChangePassword, OpenFail };
 		DCOPClient *client;
 		DCOPClientTransaction *transaction;
 		Type tType;
@@ -130,6 +130,10 @@
 			res = doTransactionOpen(xact->appid, xact->wallet, xact->wId);
 			replyType = "int";
 			break;
+		case KWalletTransaction::OpenFail:
+			res = -1;
+			replyType = "int";
+			break;
 		case KWalletTransaction::ChangePassword:
 			doTransactionChangePassword(xact->appid, xact->wallet, xact->wId);
 			// fall through - no return
@@ -151,8 +155,11 @@
 }
 
 void KWalletD::openAsynchronous(const QString& wallet, const QCString& returnObject, uint wId) {
+	DCOPClient *dc = callingDcopClient();
+	if (!dc) return;
+	QCString appid = dc->senderId();
+	
 	int rc = open(wallet, wId);
-	QCString appid = friendlyDCOPPeerName();
 	DCOPRef(appid, returnObject).send("walletOpenResult", rc);
 }
 
@@ -196,13 +203,10 @@
 	_transactions.remove(xact);
 
 	if (rc < 0) {
-		// Kill off multiple requests from the same client on a failure
-		for (KWalletTransaction *x = _transactions.first(); x; /**/) {
-			if (appid == x->appid && x->tType == KWalletTransaction::Open && x->wallet == wallet && x->wId == wId) {
-				KWalletTransaction *tmp = x;
-				x = _transactions.next();
-				_transactions.removeRef(tmp);
-			}
+		// multiple requests from the same client should not produce multiple password dialogs on a failure
+		for (KWalletTransaction *x = _transactions.first(); x; x = _transactions.next()) {
+			if (appid == x->appid && x->tType == KWalletTransaction::Open && x->wallet == wallet && x->wId == wId)
+				x->tType = KWalletTransaction::OpenFail;
 		}
 	}
 
diff -uNr ../kdelibs-3.2.0/kio/misc/uiserver.cpp ./kio/misc/uiserver.cpp
--- ../kdelibs-3.2.0/kio/misc/uiserver.cpp	Sat Jan 17 07:53:18 2004
+++ ./kio/misc/uiserver.cpp	Wed Feb  4 11:10:50 2004
@@ -1054,7 +1054,7 @@
   }
 
   int iTotalFiles = 0;
-  int iTotalSize = 0;
+  KIO::filesize_t iTotalSize = 0;
   int iTotalSpeed = 0;
   QTime totalRemTime;
 
diff -uNr ../kdelibs-3.2.0/kio/tests/kruntest.cpp ./kio/tests/kruntest.cpp
--- ../kdelibs-3.2.0/kio/tests/kruntest.cpp	Mon Apr  7 03:20:36 2003
+++ ./kio/tests/kruntest.cpp	Wed Feb  4 11:10:50 2004
@@ -147,13 +147,16 @@
       for (int te = 0; te < 2; te++)
         for (int ex = 0; ex < 2; ex++) {
           int fd = creat("kruntest.desktop", 0666);
+	  FILE *f;
           if (fd < 0) abort();
-          dprintf(fd, "[Desktop Entry]\n"
+	  f = fdopen(fd, "w");
+          fprintf(f, "[Desktop Entry]\n"
 		"Type=Application\n"
 		"Name=just_a_test\n"
 		"Icon=~/icon.png\n"
 		"%s\n%s\n%s\n",execs[ex],terms[te],sus[su]);
           close(fd);
+	  fclose(f);
           KService s(QDir::currentDirPath() + "/kruntest.desktop");
           unlink("kruntest.desktop");
           checkPDE( s, l0, hs, false, rslts[ex+te*2+su*4+hs*8]);
diff -uNr ../kdelibs-3.2.0/kioslave/file/file.cc ./kioslave/file/file.cc
--- ../kdelibs-3.2.0/kioslave/file/file.cc	Sun Sep 21 04:39:21 2003
+++ ./kioslave/file/file.cc	Wed Feb  4 11:10:50 2004
@@ -20,7 +20,7 @@
    Boston, MA 02111-1307, USA.
 */
 
-// $Id: file.cc,v 1.143 2003/09/15 11:26:06 faure Exp $
+// $Id: file.cc,v 1.143.2.1 2004/01/28 11:35:39 lukas Exp $
 
 #include <config.h>
 
@@ -192,7 +192,7 @@
 
     // Determine the mimetype of the file to be retrieved, and emit it.
     // This is mandatory in all slaves (for KRun/BrowserRun to work).
-    KMimeType::Ptr mt = KMimeType::findByURL( url.path(), buff.st_mode, true /* local URL */ );
+    KMimeType::Ptr mt = KMimeType::findByURL( url, buff.st_mode, true /* local URL */ );
     emit mimeType( mt->name() );
 
     KIO::filesize_t processed_size = 0;
@@ -534,17 +534,19 @@
 
     totalSize( buff_src.st_size );
 
-    off_t processed_size = 0;
+    KIO::filesize_t processed_size = 0;
     char buffer[ MAX_IPC_SIZE ];
     int n;
 #ifdef USE_SENDFILE
-    bool use_sendfile=true;
+    bool use_sendfile=buff_src.st_size < 0x7FFFFFFF;
 #endif
     while( 1 )
     {
 #ifdef USE_SENDFILE
        if (use_sendfile) {
-            n = ::sendfile( dest_fd, src_fd, &processed_size, MAX_IPC_SIZE );
+            off_t sf = processed_size;
+            n = ::sendfile( dest_fd, src_fd, &sf, MAX_IPC_SIZE );
+            processed_size = sf;
             if ( n == -1 && errno == EINVAL ) { //not all filesystems support sendfile()
                 kdDebug(7101) << "sendfile() not supported, falling back " << endl;
                 use_sendfile = false;
diff -uNr ../kdelibs-3.2.0/kioslave/ftp/ftp.cc ./kioslave/ftp/ftp.cc
--- ../kdelibs-3.2.0/kioslave/ftp/ftp.cc	Sun Nov 30 04:47:10 2003
+++ ./kioslave/ftp/ftp.cc	Wed Feb  4 11:10:50 2004
@@ -17,7 +17,7 @@
     Boston, MA 02111-1307, USA.
 */
 
-// $Id: ftp.cc,v 1.192 2003/11/24 08:14:09 faure Exp $
+// $Id: ftp.cc,v 1.192.2.1 2004/01/28 11:29:46 lukas Exp $
 
 #include "ftp.h"
 
@@ -1053,7 +1053,7 @@
 }
 
 bool Ftp::ftpOpenCommand( const char *_command, const QString & _path, char _mode,
-                          int errorcode, unsigned long _offset )
+                          int errorcode, KIO::fileoffset_t _offset )
 {
   QCString buf = "type ";
   buf += _mode;
@@ -1072,7 +1072,7 @@
   if ( _offset > 0 ) {
     // send rest command if offset > 0, this applies to retr and stor commands
     char buf[100];
-    sprintf(buf, "rest %ld", _offset);
+    sprintf(buf, "rest %lld", _offset);
     if ( !ftpSendCmd( buf ) )
        return false;
     if ( rspbuf[0] != '3' ) {
@@ -2006,12 +2006,12 @@
       return;
   }
 
-  unsigned long offset = 0;
+  KIO::fileoffset_t offset = 0;
   QString resumeOffset = metaData(QString::fromLatin1("resume"));
   if ( !resumeOffset.isEmpty() )
   {
-      offset = resumeOffset.toInt();
-      kdDebug(7102) << "Ftp::get got offset from medata : " << offset << endl;
+      offset = resumeOffset.toLongLong();
+      kdDebug(7102) << "Ftp::get got offset from medata : " << (long unsigned int) offset << endl;
   }
 
   if ( !ftpOpenCommand( "retr", url.path(), textMode ? 'A' : 'I', ERR_CANNOT_OPEN_FOR_READING, offset ) ) {
@@ -2029,7 +2029,7 @@
   if ( m_size != UnknownSize )
     bytesLeft = m_size - offset;
 
-  kdDebug(7102) << "Ftp::get starting with offset=" << offset << endl;
+  kdDebug(7102) << "Ftp::get starting with offset=" << (long unsigned int) offset << endl;
   KIO::fileoffset_t processed_size = offset;
 
   char buffer[ 2048 ];
@@ -2245,12 +2245,12 @@
   } else
     dest = dest_orig;
 
-  unsigned long offset = 0;
+  KIO::fileoffset_t offset = 0;
 
   // set the mode according to offset
   if ( resume ) {
     offset = m_size;
-    kdDebug(7102) << "Offset = " << (unsigned int) offset << "d" << endl;
+    kdDebug(7102) << "Offset = " << (long unsigned int) offset << "d" << endl;
   }
 
   if (! ftpOpenCommand( "stor", dest, 'I', ERR_COULD_NOT_WRITE, offset ) )
diff -uNr ../kdelibs-3.2.0/kioslave/ftp/ftp.h ./kioslave/ftp/ftp.h
--- ../kdelibs-3.2.0/kioslave/ftp/ftp.h	Sun Nov 30 04:47:10 2003
+++ ./kioslave/ftp/ftp.h	Wed Feb  4 11:10:50 2004
@@ -17,7 +17,7 @@
     Boston, MA 02111-1307, USA.
 */
 
-// $Id: ftp.h,v 1.49 2003/11/16 00:44:50 faure Exp $
+// $Id: ftp.h,v 1.49.2.1 2004/01/28 11:29:46 lukas Exp $
 
 #ifndef __ftp_h__
 #define __ftp_h__
@@ -164,7 +164,7 @@
    * @return true if the command was accepted by the server.
    */
   bool ftpOpenCommand( const char *command, const QString & path, char mode,
-                       int errorcode, unsigned long offset = 0 );
+                       int errorcode, KIO::fileoffset_t offset = 0 );
 
   /**
    * The counterpart to openCommand.
diff -uNr ../kdelibs-3.2.0/kioslave/http/http.cc ./kioslave/http/http.cc
--- ../kdelibs-3.2.0/kioslave/http/http.cc	Sun Jan 25 12:25:46 2004
+++ ./kioslave/http/http.cc	Wed Feb  4 11:10:50 2004
@@ -4113,6 +4113,8 @@
 
     char buffer[ MAX_IPC_SIZE ];
 
+    m_iContentLeft = NO_SIZE;
+
     // Jippie! It's already in the cache :-)
     while (!feof(m_request.fcache) && !ferror(m_request.fcache))
     {
diff -uNr ../kdelibs-3.2.0/kmdi/kmdidockcontainer.cpp ./kmdi/kmdidockcontainer.cpp
--- ../kdelibs-3.2.0/kmdi/kmdidockcontainer.cpp	Sat Jan 17 07:53:20 2004
+++ ./kmdi/kmdidockcontainer.cpp	Wed Feb  4 11:10:50 2004
@@ -1,4 +1,4 @@
-		/* This file is part of the KDE project
+        /* This file is part of the KDE project
    Copyright (C) 2002 Christoph Cullmann <cullmann@kde.org>
    Copyright (C) 2002,2003 Joseph Wenninger <jowenn@kde.org>
 
@@ -102,16 +102,16 @@
 {
   QMap<KDockWidget*,int>::iterator it;
   while (m_map.count()) {
-	it = m_map.begin();
-	KDockWidget *w=it.key();
-	  if (m_overlapButtons.contains(w)) {
-	    (static_cast<KDockWidgetHeader*>(w->getHeader()->qt_cast("KDockWidgetHeader")))->removeButton(m_overlapButtons[w]);
-	    m_overlapButtons.remove(w);
-	  }
+    it = m_map.begin();
+    KDockWidget *w=it.key();
+      if (m_overlapButtons.contains(w)) {
+        (static_cast<KDockWidgetHeader*>(w->getHeader()->qt_cast("KDockWidgetHeader")))->removeButton(m_overlapButtons[w]);
+        m_overlapButtons.remove(w);
+      }
     m_map.remove(w);
     w->undock();
   }
-	deactivated(this);
+    deactivated(this);
 }
 
 
@@ -203,10 +203,10 @@
 }
 
 void KMdiDockContainer::showWidget(KDockWidget *w) {
-	if (!m_map.contains(w)) return;
-	int id=m_map[w];
-	m_tb->setTab(id,true);
-	tabClicked(id);
+    if (!m_map.contains(w)) return;
+    int id=m_map[w];
+    m_tb->setTab(id,true);
+    tabClicked(id);
 }
 
 void KMdiDockContainer::changeOverlapMode()
@@ -350,40 +350,40 @@
 
 void KMdiDockContainer::save(QDomElement& dockEl)
 {
-	QDomDocument doc=dockEl.ownerDocument();
-	QDomElement el;
-	el=doc.createElement("name");
-	el.appendChild(doc.createTextNode(QString("%1").arg(parent()->name())));
-	dockEl.appendChild(el);
-	el=doc.createElement("overlapMode");
-	el.appendChild(doc.createTextNode(isOverlapMode() ?"true":"false"));
-	dockEl.appendChild(el);
-	QPtrList<KMultiTabBarTab>* tl=m_tb->tabs();
-	QPtrListIterator<KMultiTabBarTab> it(*tl);
-	QStringList::Iterator it2=itemNames.begin();
-	int i=0;
-	for (;it.current()!=0;++it,++it2)
-	{
-		el=doc.createElement("child");
-		el.setAttribute("pos",QString("%1").arg(i));
-		QString s=tabCaptions[*it2];
-		if (!s.isEmpty()) {
-			el.setAttribute("tabCaption",s);
-		}
-		s=tabTooltips[*it2];
-		if (!s.isEmpty()) {
-			el.setAttribute("tabTooltip",s);
-		}
-		el.appendChild(doc.createTextNode(*it2));
-		dockEl.appendChild(el);
-		if (m_tb->isTabRaised(it.current()->id()))
-		{
-			QDomElement el2=doc.createElement("raised");
-			el2.appendChild(doc.createTextNode(m_ws->widget(it.current()->id())->name()));
-			el.appendChild(el2);
-		}
-		++i;
-	}
+    QDomDocument doc=dockEl.ownerDocument();
+    QDomElement el;
+    el=doc.createElement("name");
+    el.appendChild(doc.createTextNode(QString("%1").arg(parent()->name())));
+    dockEl.appendChild(el);
+    el=doc.createElement("overlapMode");
+    el.appendChild(doc.createTextNode(isOverlapMode() ?"true":"false"));
+    dockEl.appendChild(el);
+    QPtrList<KMultiTabBarTab>* tl=m_tb->tabs();
+    QPtrListIterator<KMultiTabBarTab> it(*tl);
+    QStringList::Iterator it2=itemNames.begin();
+    int i=0;
+    for (;it.current()!=0;++it,++it2)
+    {
+        el=doc.createElement("child");
+        el.setAttribute("pos",QString("%1").arg(i));
+        QString s=tabCaptions[*it2];
+        if (!s.isEmpty()) {
+            el.setAttribute("tabCaption",s);
+        }
+        s=tabTooltips[*it2];
+        if (!s.isEmpty()) {
+            el.setAttribute("tabTooltip",s);
+        }
+        el.appendChild(doc.createTextNode(*it2));
+        dockEl.appendChild(el);
+        if (m_tb->isTabRaised(it.current()->id()))
+        {
+            QDomElement el2=doc.createElement("raised");
+            el2.appendChild(doc.createTextNode(m_ws->widget(it.current()->id())->name()));
+            el.appendChild(el2);
+        }
+        ++i;
+    }
 
 
 }
@@ -392,28 +392,28 @@
 {
   QString raise;
 
-	for (QDomNode n=dockEl.firstChild();!n.isNull();n=n.nextSibling()) {
-		QDomElement el=n.toElement();
-		if (el.isNull()) continue;
-		if (el.tagName()=="overlapMode") {
-			if (el.attribute("overlapMode")!="false")
-				activateOverlapMode(m_tb->width());
-			else
-				deactivateOverlapMode();
-		} else if (el.tagName()=="child") {
-			KDockWidget *dw=((KDockWidget*)parent())->dockManager()->getDockWidgetFromName(el.text());
-			if (dw)
-			{
-				if (el.hasAttribute("tabCaption")) {
-					dw->setTabPageLabel(el.attribute("tabCaption"));
-				}
-				if (el.hasAttribute("tabTooltip")) {
-					dw->setToolTipString(el.attribute("tabTooltip"));
-				}
-				dw->manualDock((KDockWidget*)parent(),KDockWidget::DockCenter);
-			}
-		}
-	}
+    for (QDomNode n=dockEl.firstChild();!n.isNull();n=n.nextSibling()) {
+        QDomElement el=n.toElement();
+        if (el.isNull()) continue;
+        if (el.tagName()=="overlapMode") {
+            if (el.attribute("overlapMode")!="false")
+                activateOverlapMode(m_tb->width());
+            else
+                deactivateOverlapMode();
+        } else if (el.tagName()=="child") {
+            KDockWidget *dw=((KDockWidget*)parent())->dockManager()->getDockWidgetFromName(el.text());
+            if (dw)
+            {
+                if (el.hasAttribute("tabCaption")) {
+                    dw->setTabPageLabel(el.attribute("tabCaption"));
+                }
+                if (el.hasAttribute("tabTooltip")) {
+                    dw->setToolTipString(el.attribute("tabTooltip"));
+                }
+                dw->manualDock((KDockWidget*)parent(),KDockWidget::DockCenter);
+            }
+        }
+    }
 
 
   QPtrList<KMultiTabBarTab>* tl=m_tb->tabs();
@@ -607,47 +607,47 @@
 }
 
 void KMdiDockContainer::toggle() {
-	kdDebug(760)<<"DockContainer:activate"<<endl;
-	if (m_tb->isTabRaised(oldtab)) {
-		m_tb->setTab(oldtab,false);
-	    	tabClicked(oldtab);
-		KMdiMainFrm *mainFrm = dynamic_cast<KMdiMainFrm*>(m_mainWin);
-	        if (mainFrm)
+    kdDebug(760)<<"DockContainer:activate"<<endl;
+    if (m_tb->isTabRaised(oldtab)) {
+        m_tb->setTab(oldtab,false);
+            tabClicked(oldtab);
+        KMdiMainFrm *mainFrm = dynamic_cast<KMdiMainFrm*>(m_mainWin);
+            if (mainFrm && mainFrm->activeWindow() )
                     mainFrm->activeWindow()->setFocus();
 
-	} else {
-		kdDebug(760)<<"KMdiDockContainer::toggle(): raising tab"<<endl;
-		if (m_tb->tab(m_previousTab)==0) {
-			if (m_tb->tabs()->count()==0) return;
-			m_previousTab=m_tb->tabs()->getFirst()->id();
-		}
-		m_tb->setTab(m_previousTab,true);
-	    	tabClicked(m_previousTab);
-	}
+    } else {
+        kdDebug(760)<<"KMdiDockContainer::toggle(): raising tab"<<endl;
+        if (m_tb->tab(m_previousTab)==0) {
+            if (m_tb->tabs()->count()==0) return;
+            m_previousTab=m_tb->tabs()->getFirst()->id();
+        }
+        m_tb->setTab(m_previousTab,true);
+            tabClicked(m_previousTab);
+    }
 }
 
 void KMdiDockContainer::prevToolView() {
-	QPtrList<KMultiTabBarTab>* tabs=m_tb->tabs();
-	int pos=tabs->findRef(m_tb->tab(oldtab));
-	if (pos==-1) return;
-	pos--;
-	if (pos<0) pos=tabs->count()-1;
-	KMultiTabBarTab *tab=tabs->at(pos);
-	if (!tab) return; //can never happen here, but who knows
-	m_tb->setTab(tab->id(),true);
-	tabClicked(tab->id());
+    QPtrList<KMultiTabBarTab>* tabs=m_tb->tabs();
+    int pos=tabs->findRef(m_tb->tab(oldtab));
+    if (pos==-1) return;
+    pos--;
+    if (pos<0) pos=tabs->count()-1;
+    KMultiTabBarTab *tab=tabs->at(pos);
+    if (!tab) return; //can never happen here, but who knows
+    m_tb->setTab(tab->id(),true);
+    tabClicked(tab->id());
 }
 
 void KMdiDockContainer::nextToolView() {
-	QPtrList<KMultiTabBarTab>* tabs=m_tb->tabs();
-	int pos=tabs->findRef(m_tb->tab(oldtab));
-	if (pos==-1) return;
-	pos++;
-	if (pos>=(int)tabs->count()) pos=0;
-	KMultiTabBarTab *tab=tabs->at(pos);
-	if (!tab) return; //can never happen here, but who knows
-	m_tb->setTab(tab->id(),true);
-	tabClicked(tab->id());
+    QPtrList<KMultiTabBarTab>* tabs=m_tb->tabs();
+    int pos=tabs->findRef(m_tb->tab(oldtab));
+    if (pos==-1) return;
+    pos++;
+    if (pos>=(int)tabs->count()) pos=0;
+    KMultiTabBarTab *tab=tabs->at(pos);
+    if (!tab) return; //can never happen here, but who knows
+    m_tb->setTab(tab->id(),true);
+    tabClicked(tab->id());
 }
 
 // kate: space-indent on; indent-width 2; replace-tabs on;
diff -uNr ../kdelibs-3.2.0/kmdi/kmdimainfrm.cpp ./kmdi/kmdimainfrm.cpp
--- ../kdelibs-3.2.0/kmdi/kmdimainfrm.cpp	Sat Jan 17 07:53:20 2004
+++ ./kmdi/kmdimainfrm.cpp	Wed Feb  4 11:10:51 2004
@@ -114,12 +114,12 @@
 
 class KMdiMainFrmPrivate {
 public:
-	KMdiMainFrmPrivate(): focusList(0) {
-		for (int i=0;i<4;i++) activeDockPriority[i]=0;
-	}
-	~KMdiMainFrmPrivate(){}
-	KMdiDockContainer* activeDockPriority[4];
-	KMdiFocusList *focusList;
+    KMdiMainFrmPrivate(): focusList(0) {
+        for (int i=0;i<4;i++) activeDockPriority[i]=0;
+    }
+    ~KMdiMainFrmPrivate(){}
+    KMdiDockContainer* activeDockPriority[4];
+    KMdiFocusList *focusList;
     int m_styleIDEAlMode;
 };
 
@@ -212,22 +212,22 @@
    setMenuForSDIModeSysButtons(menuBar());
 
    switch (mdiMode) {
-	case KMdi::IDEAlMode:
-			kdDebug(760)<<"switch(mdiMode): IDEAlMode"<<endl;	
-			switchToIDEAlMode();
-		break;
-	case KMdi::TabPageMode:
-			kdDebug(760)<<"switch(mdiMode): TabPageMode"<<endl;	
-			switchToTabPageMode();
-		break;
-	case KMdi::ToplevelMode:
-			kdDebug(760)<<"switch(mdiMode): TopLevelMode"<<endl;	
-			switchToToplevelMode();
-		break;
-	default:
-		m_mdiMode=KMdi::ChildframeMode;
-		kdDebug(760)<<"switch(mdiMode): default"<<endl;
-		break;
+    case KMdi::IDEAlMode:
+            kdDebug(760)<<"switch(mdiMode): IDEAlMode"<<endl;    
+            switchToIDEAlMode();
+        break;
+    case KMdi::TabPageMode:
+            kdDebug(760)<<"switch(mdiMode): TabPageMode"<<endl;    
+            switchToTabPageMode();
+        break;
+    case KMdi::ToplevelMode:
+            kdDebug(760)<<"switch(mdiMode): TopLevelMode"<<endl;    
+            switchToToplevelMode();
+        break;
+    default:
+        m_mdiMode=KMdi::ChildframeMode;
+        kdDebug(760)<<"switch(mdiMode): default"<<endl;
+        break;
    }   
 
    // drag end timer
@@ -243,14 +243,14 @@
   connect(m_mdiGUIClient,SIGNAL(toggleBottom()),this,SIGNAL(toggleBottom()));
 
   if (m_mdiMode==KMdi::IDEAlMode) {
-	if (m_topContainer) 
-		connect(this,SIGNAL(toggleTop()),m_topContainer->getWidget(),SLOT(toggle()));
-	if (m_leftContainer) 
-		connect(this,SIGNAL(toggleLeft()),m_leftContainer->getWidget(),SLOT(toggle()));
-	if (m_rightContainer) 
-		connect(this,SIGNAL(toggleRight()),m_rightContainer->getWidget(),SLOT(toggle()));
-	if (m_bottomContainer) 
-		connect(this,SIGNAL(toggleBottom()),m_bottomContainer->getWidget(),SLOT(toggle()));
+    if (m_topContainer) 
+        connect(this,SIGNAL(toggleTop()),m_topContainer->getWidget(),SLOT(toggle()));
+    if (m_leftContainer) 
+        connect(this,SIGNAL(toggleLeft()),m_leftContainer->getWidget(),SLOT(toggle()));
+    if (m_rightContainer) 
+        connect(this,SIGNAL(toggleRight()),m_rightContainer->getWidget(),SLOT(toggle()));
+    if (m_bottomContainer) 
+        connect(this,SIGNAL(toggleBottom()),m_bottomContainer->getWidget(),SLOT(toggle()));
   }
 
   mdiModeHasBeenChangedTo(m_mdiMode);
@@ -354,8 +354,8 @@
   Q_ASSERT( view ); // if this assert fails, then some part didn't return a widget. Fix the part ;)
 
   KMdiChildView* pMDICover = new KMdiChildView(name, // caption
-					       0L, // parent
-					       name.latin1()); // object name, necessary later in the dockwidgets
+                           0L, // parent
+                           name.latin1()); // object name, necessary later in the dockwidgets
   QBoxLayout* pLayout = new QHBoxLayout( pMDICover, 0, -1, "layout");
   view->reparent(pMDICover, QPoint(0,0));
   pLayout->addWidget(view);
@@ -506,14 +506,14 @@
 
 
 void KMdiMainFrm::deleteToolWindow( QWidget* pWnd) {
-	if (m_pToolViews->contains(pWnd)) {
-		deleteToolWindow((*m_pToolViews)[pWnd]);
-	}
+    if (m_pToolViews->contains(pWnd)) {
+        deleteToolWindow((*m_pToolViews)[pWnd]);
+    }
 }
 
 void KMdiMainFrm::deleteToolWindow( KMdiToolViewAccessor *accessor) {
-	if (!accessor) return;
-	delete accessor;
+    if (!accessor) return;
+    delete accessor;
 }
 
 //============ addWindow ============//
@@ -767,9 +767,9 @@
    }
 
    if ((m_mdiMode == KMdi::TabPageMode) || (m_mdiMode==KMdi::IDEAlMode)) {
-	if (!m_documentTabWidget) return; //oops
-	if (m_pDocumentViews->count()==0) m_pClose->hide();
-	pWnd->reparent(0L, QPoint(0,0));
+    if (!m_documentTabWidget) return; //oops
+    if (m_pDocumentViews->count()==0) m_pClose->hide();
+    pWnd->reparent(0L, QPoint(0,0));
         if (m_pDocumentViews->count() == 1) {
            m_pDocumentViews->last()->activate(); // all other views are activated by tab switch
         }
@@ -896,7 +896,7 @@
 
 
 void KMdiMainFrm::slotDocCurrentChanged(QWidget* pWnd) {
-	activateView((KMdiChildView*) pWnd);
+    activateView((KMdiChildView*) pWnd);
 }
 void KMdiMainFrm::activateView(KMdiChildView* pWnd)
 {
@@ -1166,7 +1166,7 @@
 void KMdiMainFrm::switchToToplevelMode()
 {
    if (m_mdiMode == KMdi::ToplevelMode) {
-	   mdiModeHasBeenChangedTo(KMdi::ToplevelMode);
+       mdiModeHasBeenChangedTo(KMdi::ToplevelMode);
            return;
    }
 
@@ -1464,7 +1464,7 @@
          QSize sz = pView->size();
          QWidget* pParent = pView->parentWidget();
          QPoint p(pParent->mapToGlobal(pParent->pos())-pParent->pos()+m_undockPositioningOffset);
-	 m_documentTabWidget->removePage(pView);
+     m_documentTabWidget->removePage(pView);
          pView->reparent(0,0,p);
 //         pView->reparent(0,0,p);
          pView->resize(sz);
@@ -1567,7 +1567,7 @@
    setupToolViewsForIDEALMode();
 
    if (pRemActiveWindow)
-	   pRemActiveWindow->setFocus();
+       pRemActiveWindow->setFocus();
 
    m_pTaskBar->switchOn(false);
 
@@ -2021,6 +2021,8 @@
    for (it->first(); !it->isDone(); it->next()) {
       m.insert(it->currentItem()->getTimeStamp(), it->currentItem());
    }
+   
+   if ( !activeWindow() ) return;
 
    QDateTime current = activeWindow()->getTimeStamp();
    QMap<QDateTime,KMdiChildView*>::iterator pos(m.find(current));
@@ -2048,6 +2050,8 @@
       m.insert(it->currentItem()->getTimeStamp(), it->currentItem());
    }
 
+   if ( !activeWindow() ) return;
+   
    QDateTime current = activeWindow()->getTimeStamp();
    QMap<QDateTime,KMdiChildView*>::iterator pos(m.find(current));
    if (pos != m.begin()) {
@@ -2461,49 +2465,49 @@
 }
 
 void KMdiMainFrm::setActiveToolDock(KMdiDockContainer* td) {
-	if (td==d->activeDockPriority[0]) return;
-	if (d->activeDockPriority[0]==0) {
-		d->activeDockPriority[0]=td;
-		d->focusList=new KMdiFocusList(this);
-		if (m_pMdi)  d->focusList->addWidgetTree(m_pMdi);
-		if (m_documentTabWidget) d->focusList->addWidgetTree(m_documentTabWidget);
-		return;
-	}
-	int offset=0;
-	for (int dst=3,src=2;src>=0;dst--,src--) {
-		if (d->activeDockPriority[src]==td) src--;
-		if (src<0) break;
-		d->activeDockPriority[dst]=d->activeDockPriority[src];
-	}
-	d->activeDockPriority[0]=td;
+    if (td==d->activeDockPriority[0]) return;
+    if (d->activeDockPriority[0]==0) {
+        d->activeDockPriority[0]=td;
+        d->focusList=new KMdiFocusList(this);
+        if (m_pMdi)  d->focusList->addWidgetTree(m_pMdi);
+        if (m_documentTabWidget) d->focusList->addWidgetTree(m_documentTabWidget);
+        return;
+    }
+    int offset=0;
+    for (int dst=3,src=2;src>=0;dst--,src--) {
+        if (d->activeDockPriority[src]==td) src--;
+        if (src<0) break;
+        d->activeDockPriority[dst]=d->activeDockPriority[src];
+    }
+    d->activeDockPriority[0]=td;
 }
 
 void KMdiMainFrm::removeFromActiveDockList(KMdiDockContainer* td) {
-	for (int i=0;i<4;i++) {
-		if (d->activeDockPriority[i]==td) {
-			for (int i2=i;i<3;i++)
-				d->activeDockPriority[i]=d->activeDockPriority[i+1];
-			d->activeDockPriority[3]=0;
-			break;
-		}
-	}
-	if (d->activeDockPriority[0]==0) {
-		if (d->focusList) d->focusList->restore();
-		delete d->focusList;
-		d->focusList=0;
-	}
+    for (int i=0;i<4;i++) {
+        if (d->activeDockPriority[i]==td) {
+            for (int i2=i;i<3;i++)
+                d->activeDockPriority[i]=d->activeDockPriority[i+1];
+            d->activeDockPriority[3]=0;
+            break;
+        }
+    }
+    if (d->activeDockPriority[0]==0) {
+        if (d->focusList) d->focusList->restore();
+        delete d->focusList;
+        d->focusList=0;
+    }
 }
 
 void KMdiMainFrm::prevToolViewInDock() {
-	KMdiDockContainer* td=d->activeDockPriority[0];
-	if (!td) return;
-	td->prevToolView();	
+    KMdiDockContainer* td=d->activeDockPriority[0];
+    if (!td) return;
+    td->prevToolView();    
 }
 
 void KMdiMainFrm::nextToolViewInDock() {
-	KMdiDockContainer* td=d->activeDockPriority[0];
-	if (!td) return;
-	td->nextToolView();
+    KMdiDockContainer* td=d->activeDockPriority[0];
+    if (!td) return;
+    td->nextToolView();
 }
 
 #include "kmdimainfrm.moc"
