--- ../kdepim-3.2.0/kontact/src/Makefile.am	Sun Oct 26 05:57:48 2003
+++ ./kontact/src/Makefile.am	Sun Feb  1 23:26:51 2004
@@ -4,7 +4,10 @@
 lib_LTLIBRARIES = libkontact.la
 libkontact_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -version-info 1:0
 libkontact_la_LIBADD = $(LIB_KDECORE)
-libkontact_la_SOURCES = prefs.kcfgc
+libkontact_la_SOURCES = prefs.kcfgc dummy.cpp
+
+dummy.cpp:
+	echo > dummy.cpp
 
 bin_PROGRAMS = kontact
 
--- ../kdepim-3.2.0/korganizer/plugins/Makefile.am	Sun Oct 26 05:57:51 2003
+++ ./korganizer/plugins/Makefile.am	Sun Feb  1 23:26:19 2004
@@ -2,5 +2,5 @@
 exchange_SUBDIR=exchange
 endif
 
-SUBDIRS=birthdays datenums holidays projectview webexport hebrew $(exchange_SUBDIR) 
+SUBDIRS=birthdays datenums projectview webexport hebrew $(exchange_SUBDIR) 
 
--- ../kdepim-3.2.0/kpilot/conduits/malconduit/mal-conduit.cc	Sun Oct 26 05:57:55 2003
+++ ./kpilot/conduits/malconduit/mal-conduit.cc	Sun Feb  1 23:26:19 2004
@@ -34,6 +34,7 @@
 
 #include "options.h"
 
+#include <stdarg.h>
 #include <qregexp.h>
 #include <kconfig.h>
 #include <kdebug.h>
--- ../kdepim-3.2.0/libical/src/libical/Makefile.am	Fri Jul 25 07:47:23 2003
+++ ./libical/src/libical/Makefile.am	Sun Feb  1 23:26:19 2004
@@ -26,7 +26,7 @@
 DESIGNDATA = $(top_srcdir)/libical/design-data
 ICALSCRIPTS = $(top_srcdir)/libical/scripts
 
-noinst_LTLIBRARIES = libical.la
+kde_module_LTLIBRARIES = libical.la
 
 AM_YFLAGS = -d -v -t -pical_yy
 AM_LFLAGS = -Pical_yy -olex.yy.c
--- ../kdepim-3.2.0/libical/src/libicalss/Makefile.am	Tue Sep  2 09:57:08 2003
+++ ./libical/src/libicalss/Makefile.am	Sun Feb  1 23:26:19 2004
@@ -1,5 +1,5 @@
 
-noinst_LTLIBRARIES = libicalss.la
+kde_module_LTLIBRARIES = libicalss.la
 
 AM_YFLAGS =-d -v -p ss
 AM_LFLAGS = -Pss
