--- cdat-3.3/libcdms/cdms.sh.orig	Sun Jun 16 06:43:44 2002
+++ cdat-3.3/libcdms/cdms.sh	Sun Jun 16 07:48:27 2002
@@ -60,5 +60,5 @@
 echo "./configure --enable-drs=${CDMS_INCLUDE_DRS} --enable-hdf=${CDMS_INCLUDE_HDF} --enable-ql=${CDMS_INCLUDE_QL} --cache-file=/dev/null --prefix=${PREFIX} --with-nclib=${PREFIX}/lib --with-ncinc=${PREFIX}/include" 
 ./configure --enable-drs=${CDMS_INCLUDE_DRS} --enable-hdf=${CDMS_INCLUDE_HDF} --enable-ql=${CDMS_INCLUDE_QL} --cache-file=/dev/null --prefix=${PREFIX} --with-nclib=${PREFIX}/lib --with-ncinc=${PREFIX}/include || exit 1
 make ${target} || exit 1
-make bininstall || exit 1
+make bininstall prefix=@INSTALL_PREFIX@ || exit 1
 echo "Done building the CDMS library."
--- cdat-3.3/setup.cfg.orig	Mon Jun 17 06:10:54 2002
+++ cdat-3.3/setup.cfg	Mon Jun 17 06:10:47 2002
@@ -0,0 +1,2 @@
+[install]
+prefix=@INSTALL_PREFIX@
--- cdat-3.3/libcdms/install_script.orig	Mon Jun 17 08:53:17 2002
+++ cdat-3.3/libcdms/install_script	Mon Jun 17 08:53:29 2002
@@ -1,4 +1,4 @@
 #!/bin/sh
 echo 'A message about /dev/null not being a regular file can be ignored here.'
-(./cdms.sh $* >LOG.cdms) || exit 1
+(./cdms.sh $*) || exit 1

--- cdat-3.3/contrib/natgrid/Include/nnchead.h.orig	Mon Jun 17 12:24:39 2002
+++ cdat-3.3/contrib/natgrid/Include/nnchead.h	Mon Jun 17 12:25:12 2002
@@ -1,7 +1,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
+#ifndef __APPLE__
 #include <malloc.h>
+#endif
 
 #define SQ(x)   (x) * (x)
 #define BIGNUM  1E37
--- cdat-3.3/contrib/natgrid/Include/nnghead.h.orig	Mon Jun 17 12:24:47 2002
+++ cdat-3.3/contrib/natgrid/Include/nnghead.h	Mon Jun 17 12:25:33 2002
@@ -1,7 +1,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
+#ifndef __APPLE__
 #include <malloc.h>
+#endif
 #include <time.h>
 
 #define RANSEED        367367
--- cdat-3.3/contrib/natgrid/Include/nnmhead.h.orig	Mon Jun 17 12:24:54 2002
+++ cdat-3.3/contrib/natgrid/Include/nnmhead.h	Mon Jun 17 12:29:27 2002
@@ -1,7 +1,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
+#ifndef __APPLE__
 #include <malloc.h>
+#endif
 
 #define SQ(x)   (x) * (x)
 #define RANSEED        367367
