diff -ru dx-4.2.0/bin/Makefile.am dx-new/bin/Makefile.am
--- dx-4.2.0/bin/Makefile.am	Wed Apr 10 19:32:53 2002
+++ dx-new/bin/Makefile.am	Fri Nov 22 11:50:51 2002
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
 dxbindir = $(prefix)/dx/bin
-bindir = $(prefix)/bin
+bindir = $(exec_prefix)/bin
 
 EXTRA_DIST = dxworker dx mdf2c url_handler.sh
 
diff -ru dx-4.2.0/include/Makefile.am dx-new/include/Makefile.am
--- dx-4.2.0/include/Makefile.am	Fri Jun 29 09:08:15 2001
+++ dx-new/include/Makefile.am	Fri Nov 22 11:50:51 2002
@@ -1,3 +1,3 @@
-dxincdir = $(prefix)/dx/include
+dxincdir = $(exec_prefix)/include
 SUBDIRS = dx
 dxinc_HEADERS = dxl.h dxconfig.h dxstereo.h
diff -ru dx-4.2.0/include/dx/Makefile.am dx-new/include/dx/Makefile.am
--- dx-4.2.0/include/dx/Makefile.am	Tue May 25 19:35:16 1999
+++ dx-new/include/dx/Makefile.am	Fri Nov 22 11:50:51 2002
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-dxincdir = $(prefix)/dx/include/dx
+dxincdir = $(exec_prefix)/include/dx
  
 dxinc_HEADERS = \
 	UserInteractors.h advanced.h arch.h array.h arrayhandles.h \
diff -ru dx-4.2.0/include/dx/arch.h dx-new/include/dx/arch.h
--- dx-4.2.0/include/dx/arch.h	Wed Apr  3 00:28:25 2002
+++ dx-new/include/dx/arch.h	Fri Nov 22 11:50:52 2002
@@ -199,7 +199,7 @@
  */
 #ifdef macos
 
-#define trunc(value) ((float)((int)(value)))
+/*#define trunc(value) ((float)((int)(value))) */
 
 /* default values for gamma correction */
 #undef DXD_GAMMA_8BIT
diff -ru dx-4.2.0/man/catl/Makefile.am dx-new/man/catl/Makefile.am
--- dx-4.2.0/man/catl/Makefile.am	Mon Apr  5 07:56:00 1999
+++ dx-new/man/catl/Makefile.am	Fri Nov 22 11:50:52 2002
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-catdir = $(prefix)/dx/man/catl
+catdir = $(exec_prefix)/share/man/catl
 
 cat_DATA = dx.l
 EXTRA_DIST = $(cat_DATA)
diff -ru dx-4.2.0/man/manl/Makefile.am dx-new/man/manl/Makefile.am
--- dx-4.2.0/man/manl/Makefile.am	Mon May 20 00:49:40 2002
+++ dx-new/man/manl/Makefile.am	Fri Nov 22 11:50:52 2002
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-mandir = $(prefix)/dx/man
+mandir = $(exec_prefix)/share/man
 
 man_MANS = dx.l
 EXTRA_DIST = $(man_MANS)
diff -ru dx-4.2.0/src/exec/mklib dx-new/src/exec/mklib
--- dx-4.2.0/src/exec/mklib	Thu Jan 11 12:08:51 2001
+++ dx-new/src/exec/mklib	Fri Nov 22 11:52:27 2002
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 . ./env
-
+#$INSTALL -d @DESTDIRANDPREFIX@/share/dx/lib_$ARCH
 if test -d tmp ; then
 	rm -rf tmp 
 fi 
@@ -33,7 +33,7 @@
    for i in $FMT_LIBS "" ; do
 	if test "$i" != "" ; then
 		lib=""
-		for j in /lib /usr/lib /usr/local/lib -lnsl -ldl -lm -lSM -lICE -lXm -lGL -lm -lXext -lXt -lX11 -lSM -lICE -lpthread ;  do
+		for j in /lib /usr/lib /usr/local/lib /sw/lib -lnsl -ldl -lm -lSM -lICE -lXm -lGL -lm -lXext -lXt -lX11 -lSM -lICE -lpthread ;  do
 			k=`echo $j | sed -e "s/-L//"` 
 			if test -f $k/$i ; then
 				lib=$k/$i 
@@ -79,7 +79,7 @@
 mv libDX.a .. 
 cd .. 
 $RANLIB libDX.a 
-$INSTALL libDX.a $prefix/dx/lib_$ARCH
+$INSTALL libDX.a @DESTDIRANDPREFIX@/share/dx/lib_$ARCH
 echo making libDXcallm.a 
 mv libDX.a tmp/libDXcallm.a 
 cd tmp
@@ -105,5 +105,5 @@
 mv libDXcallm.a .. 
 cd .. 
 $RANLIB libDXcallm.a 
-$INSTALL libDXcallm.a $prefix/dx/lib_$ARCH 
+$INSTALL libDXcallm.a @DESTDIRANDPREFIX@/share/dx/lib_$ARCH 
 rm libDXcallm.a 
