diff -ruN eden-orig/python/eden.py eden-edited/python/eden.py
--- eden-orig/python/eden.py	2004-06-25 00:57:43.000000000 -0700
+++ eden-edited/python/eden.py	2004-07-17 16:10:44.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/swprefix/bin/python
 
 # File: eden.py
 from random import randint
@@ -187,7 +187,7 @@
         self.root.mainloop()
 
     def browse(self):
-        system("mozilla http://www.edencrystallography.org &")
+        system("open http://www.edencrystallography.org &")
         
     def run_cb(self,k):
         # Here we need to insert code to generate the correct input
diff -ruN eden-orig/source/Makefile eden-edited/source/Makefile
--- eden-orig/source/Makefile	2004-07-02 13:58:10.000000000 -0700
+++ eden-edited/source/Makefile	2004-07-16 16:27:26.000000000 -0700
@@ -54,14 +54,14 @@
 #######################################################################
 # 	Directory where the FFTW include and lib are
 #######################################################################
-FFTW		= /sw
+FFTW		= /swprefix
 LIB  		= $(FFTW)/lib
 INC  		= $(FFTW)/include
 
 #######################################################################
 # 	Directory to install the binary, version
 #######################################################################
-BIN 		= $(HOME)/Programs/bin
+BIN 		= /swinstallprefix/bin
 VERSION		=V5.0
 
 #######################################################################
@@ -72,8 +72,8 @@
 # -O3 on HP, -O2 on SGI or DEC.
 #
 #######################################################################
-CFLAGS		= -O2
-ifneq ($(FFTW),/usr/local)
+CFLAGS		= -O3
+ifneq ($(FFTW),/usr)
 CFLAGS		+= -I$(INC)
 endif
 
@@ -85,7 +85,7 @@
 # Use different libraries and include files for single and
 # double precision versions of EDEN.
 #######################################################################
-DOUBLESWITCH 	= ON
+DOUBLESWITCH = ON
 
 ifeq ($(DOUBLESWITCH),ON)
 LDFLAGS	 = -L$(LIB) -ldfftw -lm  
@@ -118,12 +118,14 @@
 		$(CC) -o $@ $^
 
 install:	all
-		mv fbyteswap $(BIN)
-		mv eden $(BIN)/$(TARGET)
+		cp fbyteswap $(BIN)
+		cp eden $(BIN)/$(TARGET)
 		chmod 751 $(BIN)/$(TARGET)
 		chmod +x mkeden.csh
 		./mkeden.csh $(BIN) $(TARGET)
 
+ 
+
 dist:		clean
 		cd ../..; tar -cvf eden_$(VERSION).tar --exclude CVS --exclude \#\*\# --exclude \*~ --exclude Misc --exclude .\* eden; gzip -f eden_$(VERSION).tar
 
