--- ./freehel98.for	Thu Nov  4 12:42:00 1999
+++ .-patched/freehel98.for	Mon Sep 23 11:35:32 2002
@@ -2349,8 +2349,7 @@
       DATA NMAX/2000/,IBL/' '/,ZER/'0'/,O1/'O1'''/,FOUR/'4'/
       DATA IST/'*'/
       READ(*,'(A80)') INP
-      OPEN(UNIT=L11,FILE=INP(1:(index(INP,' ')-1)),STATUS='OLD',
-     * READONLY)
+      OPEN(UNIT=L11,FILE=INP(1:(index(INP,' ')-1)),STATUS='OLD')
 C     OPEN(UNIT=L11,STATUS='OLD',READONLY)
       IB=3
       NOCH=6

--- ./test.com	Thu Nov  4 12:42:00 1999
+++ .-patched/test.com	Tue Oct 15 21:21:54 2002
@@ -3,8 +3,8 @@
 #*.dat, *.PDB, & *.po4 files have the pdb filename (represented as *) as prefix
 #******* enter the pdb file name on the line after the 'END' keyword *******
 #
-echo    "Executing newhel98 ..."
-freehel97.exe<<eof>test.freehelix
+echo    "Executing freehel98 ..."
+@prefix@/bin/freehel98<<eof>test.freehelix
 TITL TBPJ C-G-T-A-T-A-T-A-T-A-C-G   Juo & Dickerson--Human TBP/DNA
 CELL  1., 1., 1., 90., 90., 90.
 BRKH
@@ -25,7 +25,7 @@
 test.inp
 eof
 #
-echo    "freehel97 finished and output as *.out, *.PDB, *.dat, and *.po4 "
+echo    "freehel98 finished and output as *.out, *.PDB, *.dat, and *.po4 "
 echo    "Executing sel98 ..."
-sel98.exe <test.freehelix >test.sel
+@prefix@/bin/sel98 <test.freehelix >test.sel
 echo    "sel98 finished. "

