--- GMT3.4.3/src/gmt_init.c.orig	Tue May  6 13:09:04 2003
+++ GMT3.4.3/src/gmt_init.c	Tue May  6 13:10:24 2003
@@ -1488,7 +1488,7 @@
 
 	if (get == 0) {	/* Must use GMT system defaults via gmt.conf */
 	
-		sprintf (line, "%s%cshare%cgmt.conf", GMTHOME, DIR_DELIM, DIR_DELIM);
+		sprintf (line, "%s%cshare%cgmt%cgmt.conf", GMTHOME, DIR_DELIM, DIR_DELIM,DIR_DELIM);
 		if ((fp = fopen (line, "r")) == NULL) {
 			fprintf (stderr, "GMT Fatal Error: Cannot open/find GMT configuration file %s\n", line);
 			exit (EXIT_FAILURE);
@@ -1509,7 +1509,7 @@
 		id = get;
 	
 	id--;	/* Get 0 or 1 */
-	sprintf (line, "%s%cshare%c.gmtdefaults_%s", GMTHOME, DIR_DELIM, DIR_DELIM, suffix[id]);
+	sprintf (line, "%s%cshare%cgmt%c.gmtdefaults_%s", GMTHOME, DIR_DELIM, DIR_DELIM, DIR_DELIM,suffix[id]);
 
 	path = (char *) GMT_memory (VNULL, (size_t)(strlen (line) + 1), (size_t)1, GMT_program);
 	
@@ -1643,7 +1643,7 @@
 
 	GMT_set_home ();
 
-	sprintf (line, "%s%cshare%cgmtmedia.d", GMTHOME, DIR_DELIM, DIR_DELIM);
+	sprintf (line, "%s%cshare%cgmt%cgmtmedia.d", GMTHOME, DIR_DELIM, DIR_DELIM,DIR_DELIM);
 	if ((fp = fopen (line, "r")) == NULL) return (0);
 
 	n_alloc = GMT_TINY_CHUNK;
--- GMT3.4.3/src/gmt_support.c.orig	Tue May  6 13:09:32 2003
+++ GMT3.4.3/src/gmt_support.c	Tue May  6 13:10:25 2003
@@ -3487,7 +3487,7 @@
 
 	/* First check the $GMTHOME/share directory */
 	
-	sprintf (path, "%s%cshare%c%s", GMTHOME, DIR_DELIM, DIR_DELIM, name);
+	sprintf (path, "%s%cshare%cgmt%c%s", GMTHOME, DIR_DELIM, DIR_DELIM, DIR_DELIM,name);
 	if (!access (path, R_OK)) return (TRUE);	/* File exists and is readable, return with name */
 	
 	/* File was not readable.  Now check if it exists */
@@ -3501,7 +3501,7 @@
 	 * It is not an error if we cannot find the named file, only if it is found
 	 * but cannot be read due to permission problems */
 	
-	sprintf (dir, "%s%cshare%ccoastline.conf", GMTHOME, DIR_DELIM, DIR_DELIM);
+	sprintf (dir, "%s%cshare%cgmt%ccoastline.conf", GMTHOME, DIR_DELIM, DIR_DELIM,DIR_DELIM);
 	if (!access (dir, F_OK))  { /* File exists... */
 		if (access (dir, R_OK)) {	/* ...but cannot be read */
 			fprintf (stderr, "%s: Error: GMT does not have permission to open %s!\n", GMT_program, dir);
--- GMT3.4.3/src/grd2cpt.c.orig	Tue May  6 13:08:41 2003
+++ GMT3.4.3/src/grd2cpt.c	Tue May  6 13:10:25 2003
@@ -64,7 +64,7 @@
 
 	/* Get list of available color tables in GMT /share */
 
-	sprintf (CPT_lis, "%s%cshare%cGMT_CPT.lis", GMTHOME, DIR_DELIM, DIR_DELIM);
+	sprintf (CPT_lis, "%s%cshare%cgmt%cGMT_CPT.lis", GMTHOME, DIR_DELIM, DIR_DELIM,DIR_DELIM);
 
 	if ((fpc = fopen (CPT_lis, "r")) == NULL) {
 		fprintf (stderr, "%s: ERROR: Cannot open file %s\n", GMT_program, CPT_lis);
@@ -180,9 +180,9 @@
 
 	if (!ok)	{	/* No table in current dir, try /share */
 		if (table)
-			sprintf (CPT_file, "%s%cshare%cGMT_%s.cpt", GMTHOME, DIR_DELIM, DIR_DELIM, table);
+			sprintf (CPT_file, "%s%cshare%cgmt%cGMT_%s.cpt", GMTHOME, DIR_DELIM, DIR_DELIM,DIR_DELIM, table);
 		else	/* Default to rainbow colors */
-			sprintf (CPT_file, "%s%cshare%cGMT_rainbow.cpt", GMTHOME, DIR_DELIM, DIR_DELIM);
+			sprintf (CPT_file, "%s%cshare%cgmt%cGMT_rainbow.cpt", GMTHOME, DIR_DELIM, DIR_DELIM,DIR_DELIM);
 
 		if (access (CPT_file, R_OK)) {
 			fprintf (stderr, "%s: ERROR: Cannot find colortable %s\n", GMT_program, CPT_file);
--- GMT3.4.3/src/grdreformat.c.orig	Tue May  6 13:09:20 2003
+++ GMT3.4.3/src/grdreformat.c	Tue May  6 13:10:26 2003
@@ -86,7 +86,7 @@
 		GMT_explain_option ('V');
 		
 		fprintf (stderr, "	The following formats are supported\n\n");
-		sprintf (line, "%s%cshare%cgmtformats.d", GMTHOME, DIR_DELIM, DIR_DELIM);
+		sprintf (line, "%s%cshare%cgmt%cgmtformats.d", GMTHOME, DIR_DELIM, DIR_DELIM,DIR_DELIM);
 		if ((fp = fopen (line, "r")) == NULL) {
 			fprintf (stderr, "%s: GMT ERROR: Cannot read file %s\n", GMT_program, line);
 			exit (EXIT_FAILURE);
--- GMT3.4.3/src/makecpt.c.orig	Tue May  6 13:08:49 2003
+++ GMT3.4.3/src/makecpt.c	Tue May  6 13:10:26 2003
@@ -48,7 +48,7 @@
 
 	/* Get list of available color tables in GMT /share */
 
-	sprintf (CPT_lis, "%s%cshare%cGMT_CPT.lis", GMTHOME, DIR_DELIM, DIR_DELIM);
+	sprintf (CPT_lis, "%s%cshare%cgmt%cGMT_CPT.lis", GMTHOME, DIR_DELIM, DIR_DELIM,DIR_DELIM);
 
 	if ((fpc = fopen (CPT_lis, "r")) == NULL) {
 		fprintf (stderr, "%s: ERROR: Cannot open file %s\n", GMT_program, CPT_lis);
@@ -132,9 +132,9 @@
 
 	if (!ok)	{	/* No table in current dir, try /share */
 		if (table)
-			sprintf (CPT_file, "%s%cshare%cGMT_%s.cpt", GMTHOME, DIR_DELIM, DIR_DELIM, table);
+			sprintf (CPT_file, "%s%cshare%cgmt%cGMT_%s.cpt", GMTHOME, DIR_DELIM, DIR_DELIM, DIR_DELIM, table);
 		else	/* Default to rainbow colors */
-			sprintf (CPT_file, "%s%cshare%cGMT_rainbow.cpt", GMTHOME, DIR_DELIM, DIR_DELIM);
+			sprintf (CPT_file, "%s%cshare%cgmt%cGMT_rainbow.cpt", GMTHOME, DIR_DELIM, DIR_DELIM, DIR_DELIM);
 
 		ok = !access (CPT_file, R_OK);
 		if (!ok) {
--- GMT3.4.3/src/pslib.c.orig	Tue May  6 13:08:53 2003
+++ GMT3.4.3/src/pslib.c	Tue May  6 13:10:26 2003
@@ -792,7 +792,7 @@
 	if ((image_no >= 0 && image_no < N_PATTERNS) && ps_pattern_status[image_no][invert]) return (image_no);	/* Already done this */
 
 	if ((image_no >= 0 && image_no < N_PATTERNS)) {	/* Premade pattern yet not used */
-		sprintf (file, "%s%cshare%cps_pattern_%2.2d.ras", PSHOME, DIR_DELIM, DIR_DELIM, image_no);
+		sprintf (file, "%s%cshare%cgmt%cps_pattern_%2.2d.ras", PSHOME, DIR_DELIM, DIR_DELIM, DIR_DELIM, image_no);
 		ps_pattern_status[image_no][invert] = 1;
 	}
 	else {	/* User image, check to see if already used */
@@ -810,7 +810,7 @@
 			if (!access (imagefile, R_OK))
 				strcpy (file, imagefile);
 			else
-				sprintf (file, "%s%cshare%c%s", PSHOME, DIR_DELIM, DIR_DELIM, imagefile);
+				sprintf (file, "%s%cshare%cgmt%c%s", PSHOME, DIR_DELIM, DIR_DELIM, DIR_DELIM, imagefile);
 		}
 		ps_user_image[ps_n_userimages].name = (char *) ps_memory (VNULL, (size_t)(strlen (imagefile)+1), sizeof (char));
 		strcpy (ps_user_image[ps_n_userimages].name, imagefile);
@@ -3225,7 +3225,7 @@
 	/* Load PSL_text procedures from file for now */
 
 	if (PSL_first) {
-		sprintf (line, "%s%cshare%cPSL_text.ps", PSHOME, DIR_DELIM, DIR_DELIM);
+		sprintf (line, "%s%cshare%cgmt%cPSL_text.ps", PSHOME, DIR_DELIM, DIR_DELIM, DIR_DELIM);
 
 		if ((fp = fopen (line, "r")) == NULL) {
 			fprintf (stderr, "pslib: ERROR: Cannot open file %s\n", line);
