--- phase-1.1/include/Models/Model.h	2003-03-31 08:32:45.000000000 -0800
+++ phase-1.1-patched/include/Models/Model.h	2003-05-03 23:58:19.000000000 -0700
@@ -14,6 +14,8 @@
 #define MIN(a,b) ((a) < (b) ? (a) : (b))
 #define MAX(a,b) ((a) > (b) ? (a) : (b))
 
+extern "C" int isnan (double);
+extern "C" int isinf (double);
 class Model {
 public:
 
--- phase-1.1/makefile	2003-03-10 14:36:03.000000000 -0800
+++ phase-1.1-patched/makefile	2003-05-03 22:00:43.000000000 -0700
@@ -4,8 +4,9 @@
 LD = g++
 
 #change this flag if you have an optimised version of blas and lapack in your system
-OPTLIBS = false
-LIBS = -llapack -lblas -lm -lg2c
+OPTLIBS = true
+LIBS = -framework vecLib
+#LIBS = -llapack -lblas -lm -lg2c
 
 
 ##### Set the compiler options
