--- phylip/src/Makefile~        2002-07-27 00:17:20.000000000 +0200
+++ phylip/src/Makefile 2003-02-28 15:53:13.000000000 +0100
@@ -62,13 +62,13 @@
 #
 #
 #A minimal one
-CFLAGS =
+#CFLAGS =
 #
 # A basic one for debugging
 #CFLAGS  = -g 
 #
 # An optimized one for gcc
-#CFLAGS  = -O3 -fomit-frame-pointer
+CFLAGS  = -O3 -fomit-frame-pointer
 #
 # For some serious debugging using Gnu gcc
 #
@@ -84,14 +84,14 @@
 # and here are some possible DFLAGS statements:
 #
 # A minimal one
-DFLAGS  = -DX $(CFLAGS)
+#DFLAGS  = -DX $(CFLAGS)
 DLIBS= -lX11 -lXaw -lXt
 #
 # A basic one for debugging
 #DFLAGS  = -g -DX
 #
 # For Gnu C++ for runs
-#DFLAGS  = -O3 -DX -fomit-frame-pointer
+DFLAGS  = -O3 -DX -fomit-frame-pointer -I/usr/X11R6/include
 #
 # For Digital Alpha systems with Compaq Tru64 Unix
 #DFLAGS = -DX -fast 
@@ -115,16 +115,16 @@
 # ordinary compiles and the tree-drawing programs, CC and DC
 #
 # This one specifies the "cc" C compiler
-CC        = cc $(CFLAGS)
+#CC        = cc $(CFLAGS)
 #
 #  To use GCC instead, if it is not the compiler that "cc" invokes
-#CC        = gcc $(CFLAGS)
+CC        = gcc $(CFLAGS)
 #
 # This one specifies the "cc" C compiler for the Draw programs
-DC        = cc $(DFLAGS)
+#DC        = cc $(DFLAGS)
 #
 # To use GCC instead, if it is not the compiler that "cc" invokes
-#DC        = gcc $(DFLAGS)
+DC        = gcc $(DFLAGS)
 #
 # ----------------------------------------------------------------------------
 #  (After this point there should not be any reason to change anything)

--- phylip/src/phylip.h~   2003-02-28 17:50:55.000000000 +0100
+++ phylip/src/phylip.h    2003-02-28 17:51:00.000000000 +0100
@@ -67,7 +67,7 @@
 #define EBCDIC false
 #define INFILE "infile"
 #define OUTFILE "outfile"
-#define FONTFILE "fontfile" /* on unix this might be /usr/local/lib/fontfile */
+#define FONTFILE "@PREFIX@/share/phylip/fontfile" /* on unix this might be /usr/local/lib/fontfile */
 #define PLOTFILE "plotfile"
 #define INTREE "intree"
 #define INTREE2 "intree2"
