diff -u -r1.16 rpymodule.c
--- rpy/src/rpymodule.c	15 Feb 2003 18:34:51 -0000	1.16
+++ rpy/src/rpymodule.c	11 Mar 2003 03:13:23 -0000
@@ -163,16 +163,16 @@
     Py_XDECREF(it);
   }
 
-  UNPROTECT(1);
   if (state == INT_T)
-    return AS_INTEGER(robj);
+    robj = AS_INTEGER(robj);
   if (state == FLOAT_T)
-    return AS_NUMERIC(robj);
+    robj = AS_NUMERIC(robj);
   if (state == COMPLEX_T)
-    return AS_COMPLEX(robj);
+    robj = AS_COMPLEX(robj);
   if (state == STRING_T)
-    return AS_CHARACTER(robj);
+    robj = AS_CHARACTER(robj);
 
+  UNPROTECT(1);
   return robj;
 
  exception:
