--- scilab-2.7/configure.orig	Mon Feb 17 12:27:57 2003
+++ scilab-2.7/configure	Mon Feb 17 13:13:15 2003
@@ -10324,11 +10324,11 @@
      GCC=yes
      CC=cc
      CC_OPTIONS='-no-cpp-precomp -O2 -I/sw/include'
-     CC_LDFLAGS=
+     CC_LDFLAGS=$LDFLAGS
      CC_PICFLAGS='-fPIC'
      # script shell f77 have bugs, get ours
      FC=g77
      FC_OPTIONS='-O2'
      FC_OPTIONS_O0=
-     FC_LDFLAGS='-L/sw/lib -ldl'
+     FC_LDFLAGS='-lcc_dynamic -L/sw/lib -ldl'
      FC_PICFLAGS='-fPIC'
