diff -ru -x *~ SNNSv4.2/kernel/sources/Makefile SNNSv4.2-patched/kernel/sources/Makefile
--- SNNSv4.2/kernel/sources/Makefile	Thu Sep  3 15:38:00 1998
+++ SNNSv4.2-patched/kernel/sources/Makefile	Thu Sep 13 20:21:15 2001
@@ -49,7 +49,8 @@
                      kr_amap.c \
 		     kr_pat_parse.c \
                      kr_newpattern.c\
-                     kr_JordElm.c
+                     kr_JordElm.c \
+                     kr_rand.c
 
 USR_FUNC_SOURCES_C = func_tbl.c \
                      init_f.c \
diff -ru -x *~ SNNSv4.2/kernel/sources/kr_mem.c SNNSv4.2-patched/kernel/sources/kr_mem.c
--- SNNSv4.2/kernel/sources/kr_mem.c	Thu Sep  3 15:38:16 1998
+++ SNNSv4.2-patched/kernel/sources/kr_mem.c	Thu Sep 13 19:32:29 2001
@@ -27,7 +27,9 @@
 #else
 #include <math.h>
 #endif
+#include <limits.h>
 
+#define MAXSHORT 32767
 
 #include "kr_typ.h"	 /*  Kernel Types and Constants  */
 #include "kernel.h"
diff -ru -x *~ SNNSv4.2/kernel/sources/kr_rand.c SNNSv4.2-patched/kernel/sources/kr_rand.c
--- SNNSv4.2/kernel/sources/kr_rand.c	Thu Sep  3 15:38:19 1998
+++ SNNSv4.2-patched/kernel/sources/kr_rand.c	Thu Sep 13 20:22:25 2001
@@ -18,7 +18,6 @@
 
 ******************************************************************************/
 #include <config.h>
-#ifdef  __MSDOS__
 
 #ifndef  MSDOS_RAND_FUNCS
 #define  MSDOS_RAND_FUNCS
@@ -40,5 +39,4 @@
         return ((double) rand() / (double) RAND_MAX);
 }
 
-#endif
 #endif
diff -ru -x *~ SNNSv4.2/kernel/sources/kr_rand.h SNNSv4.2-patched/kernel/sources/kr_rand.h
--- SNNSv4.2/kernel/sources/kr_rand.h	Thu Sep  3 15:38:19 1998
+++ SNNSv4.2-patched/kernel/sources/kr_rand.h	Thu Sep 13 20:22:11 2001
@@ -17,7 +17,6 @@
     Copyright (c) 1996-1998  SNNS Group, WSI, Univ. Tuebingen, FRG
 
 ******************************************************************************/
-#ifdef  __MSDOS__
 
 #ifndef  MSDOS_RAND_FUNCS
 #define  MSDOS_RAND_FUNCS
@@ -26,5 +25,4 @@
 extern long lrand48(void);
 extern double drand48(void);
 
-#endif
 #endif
diff -ru -x *~ SNNSv4.2/kernel/sources/kr_rand.ph SNNSv4.2-patched/kernel/sources/kr_rand.ph
--- SNNSv4.2/kernel/sources/kr_rand.ph	Thu Sep  3 15:38:19 1998
+++ SNNSv4.2-patched/kernel/sources/kr_rand.ph	Thu Sep 13 20:21:59 2001
@@ -17,7 +17,6 @@
     Copyright (c) 1996-1998  SNNS Group, WSI, Univ. Tuebingen, FRG
 
 ******************************************************************************/
-#ifdef  __MSDOS__
 
 #ifndef  MSDOS_RAND_FUNCS
 #define  MSDOS_RAND_FUNCS
@@ -26,5 +25,4 @@
 long lrand48(void);
 double drand48(void);
 
-#endif
 #endif
diff -ru -x *~ SNNSv4.2/kernel/sources/learn_f.c SNNSv4.2-patched/kernel/sources/learn_f.c
--- SNNSv4.2/kernel/sources/learn_f.c	Thu Sep  3 15:38:22 1998
+++ SNNSv4.2-patched/kernel/sources/learn_f.c	Thu Sep 13 19:41:38 2001
@@ -36,6 +36,8 @@
 #endif
 #include <ctype.h>
 
+#define MAXINT 2147483647
+
 #include "kr_typ.h"		/* Kernel Types and Constants  */
 #include "kr_const.h"		/* Constant Declarators for SNNS-Kernel  */
 #include "kr_def.h"		/* Default Values  */
diff -ru -x *~ SNNSv4.2/tools/sources/Makefile SNNSv4.2-patched/tools/sources/Makefile
--- SNNSv4.2/tools/sources/Makefile	Thu Sep  3 15:37:47 1998
+++ SNNSv4.2-patched/tools/sources/Makefile	Thu Sep 13 20:07:38 2001
@@ -26,7 +26,7 @@
 
 # Define final C-preprocessor options for the kernel
 # Additional definitions like -I... and -D... go here
-CPPFLAGS           = -I../.. \
+CPPFLAGS           += -I../.. \
                      -I../../$(KERNELSRCDIR) \
                      $(ENZOFLAGS)
 
@@ -177,43 +177,43 @@
 # Linking of single programs based on kernel parts
 
 snns2c: snns2c.o snns2clib.o $(KERNELLIBS)
-	$(CC) snns2c.o snns2clib.o $(KERNELALLLIBS) -o snns2c
+	$(CC) $(LDFLAGS) snns2c.o snns2clib.o $(KERNELALLLIBS) -o snns2c
 
 pat_sel: pat_sel.o $(KERNELLIBS)
-	$(CC) pat_sel.o $(KERNELALLLIBS) -o pat_sel
+	$(CC) $(LDFLAGS) pat_sel.o $(KERNELALLLIBS) -o pat_sel
 
 netlearn: netlearn.o $(KERNELLIBS)
-	$(CC) netlearn.o $(KERNELALLLIBS) -o netlearn
+	$(CC) $(LDFLAGS) netlearn.o $(KERNELALLLIBS) -o netlearn
 
 td_bignet: td_bignet.o $(KERNELLIBS)
-	$(CC) td_bignet.o $(KERNELALLLIBS) -o td_bignet
+	$(CC) $(LDFLAGS) td_bignet.o $(KERNELALLLIBS) -o td_bignet
 
 ff_bignet: ff_bignet.o $(KERNELLIBS)
-	$(CC) ff_bignet.o $(KERNELALLLIBS) -o ff_bignet
+	$(CC) $(LDFLAGS) ff_bignet.o $(KERNELALLLIBS) -o ff_bignet
 
 netperf: netperf.o $(KERNELLIBS)
-	$(CC) netperf.o $(KERNELALLLIBS) -o netperf
+	$(CC) $(LDFLAGS) netperf.o $(KERNELALLLIBS) -o netperf
 
 snnsbat: snnsbat.o $(KERNELLIBS)
-	$(CC) snnsbat.o $(KERNELALLLIBS) -o snnsbat
+	$(CC) $(LDFLAGS) snnsbat.o $(KERNELALLLIBS) -o snnsbat
 
 feedback-gennet: feedback-gennet.o $(KERNELLIBS)
-	$(CC) feedback-gennet.o $(KERNELALLLIBS) -o feedback-gennet
+	$(CC) $(LDFLAGS) feedback-gennet.o $(KERNELALLLIBS) -o feedback-gennet
 
 convert2snns: convert2snns.o $(KERNELLIBS)
-	$(CC) convert2snns.o $(KERNELALLLIBS) -o convert2snns
+	$(CC) $(LDFLAGS) convert2snns.o $(KERNELALLLIBS) -o convert2snns
 
 isnns: isnns.o $(KERNELLIBS)
-	$(CC) isnns.o $(KERNELALLLIBS) -o isnns
+	$(CC) $(LDFLAGS) isnns.o $(KERNELALLLIBS) -o isnns
 
 linknets: linknets.o snns2clib.o $(KERNELLIBS)
-	$(CC) linknets.o snns2clib.o $(KERNELALLLIBS) -o linknets
+	$(CC) $(LDFLAGS) linknets.o snns2clib.o $(KERNELALLLIBS) -o linknets
 
 
 # Linking of batchman
 
 batchman: $(BATCH_OBJECTS) $(KERNELLIBS)
-	$(CC) $(BATCH_OBJECTS) $(KERNELALLLIBS) -o batchman
+	$(CC) $(LDFLAGS) $(BATCH_OBJECTS) $(KERNELALLLIBS) -o batchman
 
 # Parser generation for batchman
 
diff -ru -x *~ SNNSv4.2/xgui/sources/Makefile SNNSv4.2-patched/xgui/sources/Makefile
--- SNNSv4.2/xgui/sources/Makefile	Thu Sep  3 15:39:03 1998
+++ SNNSv4.2-patched/xgui/sources/Makefile	Thu Sep 13 20:55:15 2001
@@ -26,7 +26,7 @@
 
 # Define final C-preprocessor options for the kernel
 # Additional definitions like -I... and -D... go here
-CPPFLAGS           = -I../.. \
+CPPFLAGS          += -I../.. \
                      -I../../$(KERNELSRCDIR) \
                      -I../../$(XGUIICONSDIR) \
                      $(X_CFLAGS) $(ENZOFLAGS)
@@ -202,7 +202,7 @@
 # Linking of xgui
 
 xgui: $(LIBXGUI1) $(LIBXGUI2) $(BN3DLIB) $(KERNELLIBS)
-	$(CC) $(LIBXGUI1) $(LIBXGUI2) $(BN3DLIB) \
+	$(CC) $(LDFLAGS) $(LIBXGUI1) $(LIBXGUI2) $(BN3DLIB) \
 	$(X_LIBS) -lXaw -lXmu -lXt -lXext $(X_PRE_LIBS) -lX11 $(X_EXTRA_LIBS) \
 	$(KERNELALLLIBS) -o xgui 
 
