diff -ru spim-6.3/Configure spim-6.3-patched/Configure
--- spim-6.3/Configure	Sun Jan 14 21:55:10 2001
+++ spim-6.3-patched/Configure	Fri Dec  7 19:55:13 2001
@@ -78,6 +78,13 @@
   echo Scaning libc
   nm /usr/lib/libc.so > library_contents
   set flag = -q
+else if [ -f /usr/lib/libc.dylib ]; then
+  # Darwin??
+  echo Looks like Darwin OS or Mac OS X ...
+  echo Scaning libc
+  nm /usr/lib/libc.dylib > library_contents
+  set flag = -s
+fi;
 fi;
 fi;
 fi;
diff -ru spim-6.3/Makefile.std spim-6.3-patched/Makefile.std
--- spim-6.3/Makefile.std	Sun Jan 14 21:55:16 2001
+++ spim-6.3-patched/Makefile.std	Fri Dec  7 19:55:13 2001
@@ -54,13 +54,13 @@
 
 
 # Full path for directory that will hold the trap handler file:
-TRAP_DIR = .
+TRAP_DIR = @PREFIX@/share/spim/
 
 # Full path for the directory that will hold the executable files:
-BIN_DIR = /usr/unsup/bin
+BIN_DIR = @PREFIX@/bin
 
 # Full path for the directory that will hold the man files:
-MAN_DIR = /usr/unsup/man
+MAN_DIR = @PREFIX@/share/man
 
 
 # Full path for the trap handler file:
@@ -199,12 +199,10 @@
 	  .spim-made  \
 	  spim.tar.* spim.aux spim.log spim.dvi spim.shar*
 
-install: spim xspim
-	install -c -s  spim $(BIN_DIR)
-	install -c -s  xspim $(BIN_DIR)
-	install -c -m 0444 trap.handler $(TRAP_DIR)
-	install -c -m 0444 spim.man $(MAN_DIR)
-	install -c -m 0444 xspim.man $(MAN_DIR)
+install: spim
+	install -c -s spim $(DESTDIR)$(BIN_DIR)/spim
+	install -c -m 0444 trap.handler $(DESTDIR)$(TRAP_DIR)/trap.handler
+	install -c -m 0444 spim.man $(DESTDIR)$(MAN_DIR)/man1/spim.1
 
 very-clean: clean
 	rm -f y.tab.h y.tab.c lex.yy.c spim.tar* Documentation/spim.ps
