diff -ru starplot-0.93/Makefile starplot-0.93-patched/Makefile
--- starplot-0.93/Makefile	Mon Nov 18 02:00:47 2002
+++ starplot-0.93-patched/Makefile	Mon Nov 18 09:42:56 2002
@@ -7,7 +7,7 @@
 export DATADIR     =  $(DESTDIR)/share/starplot
 export DOCDIR      =  $(DESTDIR)/share/doc/starplot
 INSTALLDIR         =  $(DESTDIR)/bin
-MANDIR             =  $(DESTDIR)/man/man1
+MANDIR             =  $(DESTDIR)/share/man/man1
 
 # locations of programs needed for compiling / installing:
 export CXX	  =  g++
@@ -16,7 +16,7 @@
 export INSTALLPGM =  $(INSTALL) -s
 
 # compiler flags
-export CFLAGS =  -Wall -pedantic -O2
+export CFLAGS =  -O2 -no-cpp-precomp
 LFLAGS        =  `gtk-config --libs`
 
 # targets ---------------------------------------------------------------
