diff -Nurd -x'*~' jwm-2.1.0.orig/configure jwm-2.1.0/configure
--- jwm-2.1.0.orig/configure	2011-09-25 18:28:02.000000000 -0400
+++ jwm-2.1.0/configure	2011-11-27 15:01:12.000000000 -0500
@@ -4574,8 +4574,8 @@
 { echo "$as_me:$LINENO: result: $ac_cv_lib_png_png_read_image" >&5
 echo "${ECHO_T}$ac_cv_lib_png_png_read_image" >&6; }
 if test $ac_cv_lib_png_png_read_image = yes; then
-   LDFLAGS="$LDFLAGS $PNG_LDFLAGS"
-        CFLAGS="$CFLAGS $PNG_CFLAGS"
+   LDFLAGS="$PNG_LDFLAGS $LDFLAGS"
+        CFLAGS="$PNG_CFLAGS $CFLAGS"
         enable_png="yes"
 
 cat >>confdefs.h <<\_ACEOF
@@ -4669,8 +4669,8 @@
 { echo "$as_me:$LINENO: result: $ac_cv_lib_jpeg_jpeg_start_decompress" >&5
 echo "${ECHO_T}$ac_cv_lib_jpeg_jpeg_start_decompress" >&6; }
 if test $ac_cv_lib_jpeg_jpeg_start_decompress = yes; then
-   LDFLAGS="$LDFLAGS $JPEG_LDFLAGS"
-        CFLAGS="$CFLAGS $JPEG_CFLAGS"
+   LDFLAGS="$JPEG_LDFLAGS $LDFLAGS"
+        CFLAGS="$JPEG_CFLAGS $CFLAGS"
         enable_jpeg="yes"
 
 cat >>confdefs.h <<\_ACEOF
@@ -4769,8 +4769,8 @@
 { echo "$as_me:$LINENO: result: $ac_cv_lib_Xft_XftFontOpenName" >&5
 echo "${ECHO_T}$ac_cv_lib_Xft_XftFontOpenName" >&6; }
 if test $ac_cv_lib_Xft_XftFontOpenName = yes; then
-   LDFLAGS="$LDFLAGS $XFT_LDFLAGS"
-        CFLAGS="$CFLAGS $XFT_CFLAGS"
+   LDFLAGS="$XFT_LDFLAGS $LDFLAGS"
+        CFLAGS="$XFT_CFLAGS $CFLAGS"
         enable_xft="yes"
 
 cat >>confdefs.h <<\_ACEOF
@@ -5095,8 +5095,8 @@
 { echo "$as_me:$LINENO: result: $ac_cv_lib_fribidi_fribidi_log2vis" >&5
 echo "${ECHO_T}$ac_cv_lib_fribidi_fribidi_log2vis" >&6; }
 if test $ac_cv_lib_fribidi_fribidi_log2vis = yes; then
-   LDFLAGS="$LDFLAGS $FRIBIDI_LDFLAGS"
-        CFLAGS="$CFLAGS $FRIBIDI_CFLAGS"
+   LDFLAGS="$FRIBIDI_LDFLAGS $LDFLAGS"
+        CFLAGS="$FRIBIDI_CFLAGS $CFLAGS"
         enable_fribidi="yes"
 
 cat >>confdefs.h <<\_ACEOF
diff -Nurd -x'*~' jwm-2.1.0.orig/src/jwm.h jwm-2.1.0/src/jwm.h
--- jwm-2.1.0.orig/src/jwm.h	2011-09-25 18:28:02.000000000 -0400
+++ jwm-2.1.0/src/jwm.h	2011-11-27 14:48:55.000000000 -0500
@@ -97,7 +97,7 @@
 #     include <X11/extensions/Xrender.h>
 #  endif
 #  ifdef USE_FRIBIDI
-#     include <fribidi/fribidi.h>
+#     include <fribidi.h>
 #  endif
 
 #endif /* MAKE_DEPEND */
