diff -ruN libxfce4mcs-4.2.3/configure libxfce4mcs-4.2.3-mod/configure
--- libxfce4mcs-4.2.3/configure	2005-11-05 16:30:38.000000000 -0500
+++ libxfce4mcs-4.2.3-mod/configure	2011-07-30 15:28:21.000000000 -0400
@@ -3927,11 +3927,11 @@
 
 
 if test "`uname | grep \"CYGWIN\"`" != ""; then
-  HAVE_CYGWIN_TRUE=
-  HAVE_CYGWIN_FALSE='#'
-else
   HAVE_CYGWIN_TRUE='#'
   HAVE_CYGWIN_FALSE=
+else
+  HAVE_CYGWIN_TRUE=
+  HAVE_CYGWIN_FALSE='#'
 fi
 
 
diff -ruN libxfce4mcs-4.2.3/libxfce4mcs/Makefile.in libxfce4mcs-4.2.3-mod/libxfce4mcs/Makefile.in
--- libxfce4mcs-4.2.3/libxfce4mcs/Makefile.in	2005-11-06 05:45:15.000000000 -0500
+++ libxfce4mcs-4.2.3-mod/libxfce4mcs/Makefile.in	2011-07-30 15:28:21.000000000 -0400
@@ -38,8 +38,8 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@HAVE_CYGWIN_TRUE@am__append_1 = -no-undefined
-@HAVE_CYGWIN_TRUE@am__append_2 = -no-undefined
+@HAVE_CYGWIN_FALSE@am__append_1 = -no-undefined
+@HAVE_CYGWIN_FALSE@am__append_2 = -no-undefined
 subdir = libxfce4mcs
 DIST_COMMON = $(libxfce4mcs_clientinclude_HEADERS) \
 	$(libxfce4mcs_managerinclude_HEADERS) $(srcdir)/Makefile.am \
diff -ruN libxfce4mcs-4.2.3/libxfce4mcs/libxfce4mcs-client-1.0.pc.in libxfce4mcs-4.2.3-mod/libxfce4mcs/libxfce4mcs-client-1.0.pc.in
--- libxfce4mcs-4.2.3/libxfce4mcs/libxfce4mcs-client-1.0.pc.in	2005-11-05 11:57:42.000000000 -0500
+++ libxfce4mcs-4.2.3-mod/libxfce4mcs/libxfce4mcs-client-1.0.pc.in	2011-07-30 15:29:18.000000000 -0400
@@ -7,5 +7,5 @@
 Description: libxfce4mcs-client
 Requires: libxfce4util-1.0
 Version: @VERSION@
-Libs: @LIBX11_LDFLAGS@ @LIBSM_LDFLAGS@ -L${libdir} -lxfce4mcs-client
-Cflags: @LIBX11_CFLAGS@ @LIBSM_CFLAGS@ -I${includedir}/xfce4
+Libs: -L${libdir} -lxfce4mcs-client @LIBX11_LDFLAGS@ @LIBSM_LDFLAGS@
+Cflags: -I${includedir}/xfce4 @LIBX11_CFLAGS@ @LIBSM_CFLAGS@
diff -ruN libxfce4mcs-4.2.3/libxfce4mcs/libxfce4mcs-manager-1.0.pc.in libxfce4mcs-4.2.3-mod/libxfce4mcs/libxfce4mcs-manager-1.0.pc.in
--- libxfce4mcs-4.2.3/libxfce4mcs/libxfce4mcs-manager-1.0.pc.in	2005-11-05 11:57:42.000000000 -0500
+++ libxfce4mcs-4.2.3-mod/libxfce4mcs/libxfce4mcs-manager-1.0.pc.in	2011-07-30 15:29:26.000000000 -0400
@@ -7,5 +7,5 @@
 Description: libxfce4mcs-manager
 Requires: libxfce4util-1.0
 Version: @VERSION@
-Libs: @LIBX11_LDFLAGS@ @LIBSM_LDFLAGS@ -L${libdir} -lxfce4mcs-manager
-Cflags: @LIBX11_CFLAGS@ @LIBSM_CFLAGS@ -I${includedir}/xfce4
+Libs: -L${libdir} -lxfce4mcs-manager @LIBX11_LDFLAGS@ @LIBSM_LDFLAGS@
+Cflags: -I${includedir}/xfce4 @LIBX11_CFLAGS@ @LIBSM_CFLAGS@
