diff -ruN xplanet-1.2.2.orig/src/Makefile.in xplanet-1.2.2/src/Makefile.in
--- xplanet-1.2.2.orig/src/Makefile.in	2011-02-15 19:57:38.000000000 -0500
+++ xplanet-1.2.2/src/Makefile.in	2012-01-31 10:11:02.000000000 -0500
@@ -263,7 +263,7 @@
 SUBDIRS = libannotate libdisplay libephemeris libimage libmultiple libplanet libprojection libsgp4sdp4
 EXTRA_xplanet_SOURCES = ParseGeom.c ParseGeom.h
 @HAVE_LIBX11_FALSE@parsegeom = ParseGeom.c ParseGeom.h
-AM_CPPFLAGS = -DDATADIR=\"$(datadir)@separator@xplanet\" @X_CFLAGS@ @FREETYPE_CFLAGS@ 
+AM_CPPFLAGS = -DDATADIR=\"$(datadir)@separator@xplanet\" @FREETYPE_CFLAGS@ @X_CFLAGS@ 
 AM_LDFLAGS = @xplanet_LDFLAGS@
 xplanet_SOURCES = \
 	Map.cpp			\
diff -ruN xplanet-1.2.2.orig/src/libdisplay/Makefile.in xplanet-1.2.2/src/libdisplay/Makefile.in
--- xplanet-1.2.2.orig/src/libdisplay/Makefile.in	2011-02-15 19:57:38.000000000 -0500
+++ xplanet-1.2.2/src/libdisplay/Makefile.in	2012-01-31 10:08:30.000000000 -0500
@@ -235,7 +235,7 @@
 @HAVE_LIBX11_TRUE@displayx11 = DisplayX11.cpp DisplayX11.h vroot.h 
 @HAVE_LIBX11_TRUE@timerx11 = TimerX11.h TimerX11.cpp
 EXTRA_libdisplay_a_SOURCES = DisplayMacAqua.cpp DisplayMacAqua.h DisplayMSWin.cpp DisplayMSWin.h TextRendererFT2.cpp TextRendererFT2.h TextRendererPangoFT2.cpp TextRendererPangoFT2.h DisplayX11.cpp DisplayX11.h vroot.h TimerMacAqua.cpp TimerMacAqua.h TimerX11.cpp TimerX11.h
-AM_CPPFLAGS = -I@top_srcdir@/src @X_CFLAGS@ @FREETYPE_CFLAGS@
+AM_CPPFLAGS = -I@top_srcdir@/src @FREETYPE_CFLAGS@ @X_CFLAGS@
 @USE_AR_FALSE@libdisplay_a_AR = $(CXX) @xplanet_ARFLAGS@
 @USE_AR_TRUE@libdisplay_a_AR = $(AR) cru
 @USE_AR_FALSE@libtimer_a_AR = $(CXX) @xplanet_ARFLAGS@
diff -ruN xplanet-1.2.2.orig/src/libimage/Makefile.in xplanet-1.2.2/src/libimage/Makefile.in
--- xplanet-1.2.2.orig/src/libimage/Makefile.in	2011-02-15 19:57:38.000000000 -0500
+++ xplanet-1.2.2/src/libimage/Makefile.in	2012-01-31 10:14:21.000000000 -0500
@@ -218,7 +218,7 @@
 @HAVE_LIBPNM_TRUE@pnm = pnm.c
 @HAVE_LIBTIFF_TRUE@tiff = tiff.c
 EXTRA_libimage_a_SOURCES = bmp.c gif.c jpeg.c png.c pnm.c tiff.c ReadImage.cpp WriteImage.cpp
-AM_CPPFLAGS = -I@top_srcdir@/src
+AM_CPPFLAGS = -I@top_srcdir@/src @FREETYPE_CFLAGS@
 @USE_AR_FALSE@libdisplay_a_AR = $(CXX) @xplanet_ARFLAGS@
 @USE_AR_TRUE@libdisplay_a_AR = $(AR) cru
 @USE_AR_FALSE@libtimer_a_AR = $(CXX) @xplanet_ARFLAGS@
