diff -Nurd -x'*~' xmlstarlet-1.0.1.orig/configure.in xmlstarlet-1.0.1/configure.in
--- xmlstarlet-1.0.1.orig/configure.in	2005-03-13 20:30:42.000000000 -0500
+++ xmlstarlet-1.0.1/configure.in	2008-02-09 18:37:32.000000000 -0500
@@ -21,88 +21,29 @@
       ;;
 esac
 
-dnl
-dnl The following parameters offer
-dnl the ability to specify the location of the libxml
-dnl library during linking and compilation.
-dnl
-LIBXML_PREFIX="/usr"
-AC_ARG_WITH(libxml-prefix,
-        [  --with-libxml-prefix=[PFX]		Specify location of libxml],
-        LIBXML_PREFIX=$withval
-        LIBXML_CFLAGS="-I$withval/include/libxml2 -I$withval/include/libxml"
-        LIBXML_LIBS="$withval/lib/libxml2.a -lz -lm -lpthread -ldl"
-)
-        
-AC_ARG_WITH(libxml-include-prefix,
-        [  --with-libxml-include-prefix=[PFX]	Specify location of libxml headers],
-        LIBXML_CFLAGS="-I$withval/libxml2 -I$withval/libxml -I$withval"
-)
-
-AC_ARG_WITH(libxml-libs-prefix,
-        [  --with-libxml-libs-prefix=[PFX]	Specify location of libxml libs],
-        LIBXML_LIBS="$withval/libxml2.a -lz -lm -lpthread -ldl"
-)
-
-AC_ARG_WITH(libxml-src,
-        [  --with-libxml-src=[DIR]               For libxml thats not installed yet (sets all three above)],
-        LIBXML_SRC="$withval"
-        LIBXML_LIBS="$withval/.libs/libxml2.a -lz -lm -lpthread"
-        LIBXML_CFLAGS="-I$withval/include/libxml2 -I$withval/include/libxml -I$withval/include"
-)
+LIBXML_PREFIX=`xml2-config --prefix`
+LIBXML_CFLAGS=`xml2-config --cflags`
+LIBXML_LIBS=`xml2-config --libs`
 
-AC_DEFUN(VERSION_TO_NUMBER,
+AC_DEFUN([VERSION_TO_NUMBER],
 	 [`$1 | sed -e 's/libxml //' -e 's/libxslt //' | awk 'BEGIN { FS = "."; } { printf "%d", ([$]1 * 1000 + [$]2) * 1000 + [$]3;}'`])
 
 LIBXML_REQUIRED_VERSION=2.6.12
 
-if test "x$LIBXML_SRC" = "x"
-then
 	XMLVERS=`$LIBXML_PREFIX/bin/xml2-config --version`
 	if test VERSION_TO_NUMBER(echo $XMLVERS) -lt VERSION_TO_NUMBER(echo $LIBXML_REQUIRED_VERSION); then
         	AC_MSG_ERROR(xmlstarlet needs at least libxml2 version $LIBXML_REQUIRED_VERSION (http://www.xmlsoft.org/))
 	fi
-fi
-
-dnl
-dnl The following parameters offer
-dnl the ability to specify the location of the libxslt
-dnl library during linking and compilation.
-dnl
-LIBXSLT_PREFIX="/usr"
-AC_ARG_WITH(libxslt-prefix,
-        [  --with-libxslt-prefix=[PFX]		Specify location of libxslt],
-	LIBXSLT_PREFIX=$withval
-        LIBXSLT_CFLAGS="-I$withval/include/libxslt -I$withval/include/libexslt"
-        LIBXSLT_LIBS="$withval/lib/libxslt.a $withval/lib/libexslt.a -lz -lm -lpthread"
-)
-        
-AC_ARG_WITH(libxslt-include-prefix,
-        [  --with-libxslt-include-prefix=[PFX]	Specify location of libxslt headers],
-        LIBXSLT_CFLAGS="-I$withval/libxslt -I$withval/libexslt -I$withval"
-)
-
-AC_ARG_WITH(libxslt-libs-prefix,
-        [  --with-libxslt-libs-prefix=[PFX]	Specify location of libxslt libs],
-        LIBXSLT_LIBS="$withval/libxslt.a $withval/libexslt.a -lz -lm -lpthread"
-)
-
-AC_ARG_WITH(libxslt-src,
-        [  --with-libxslt-src=[DIR]		For libxslt thats not installed yet (sets all three above)],
-        LIBXSLT_SRC="$withval"
-        LIBXSLT_LIBS="$withval/libxslt/.libs/libxslt.a $withval/libexslt/.libs/libexslt.a -lz -lm -lpthread"
-        LIBXSLT_CFLAGS="-I$withval/libxslt -I$withval/libexslt -I$withval"
-)
+LIBXSLT_PREFIX=`xslt-config --prefix`
+LIBXSLT_CFLAGS=`xslt-config --cflags`
+LIBXSLT_LIBS="`xslt-config --libs` -lexslt"
 
 LIBXSLT_REQUIRED_VERSION=1.1.9
 
-if test "x$LIBXSLT_SRC" = "x"
-then
 	XSLTVERS=`$LIBXSLT_PREFIX/bin/xslt-config --version`
 	if test VERSION_TO_NUMBER(echo $XSLTVERS) -lt VERSION_TO_NUMBER(echo $LIBXSLT_REQUIRED_VERSION); then
         	AC_MSG_ERROR(xmlstarlet needs at least libxslt version $LIBXSLT_REQUIRED_VERSION (http://www.xmlsoft.org/))
 	fi
-fi
 
 LIBICONV_PREFIX="/usr"
 AC_ARG_WITH(libiconv-prefix,
@@ -129,7 +70,6 @@
 AC_ISC_POSIX
 AC_PROG_CC
 AC_STDC_HEADERS
-AC_ARG_PROGRAM
 dnl AM_PROG_LIBTOOL
 
 dnl
