diff -Naur ash-20020131-1.old/Makefile.in ash-20020131-1.new/Makefile.in
--- ash-20020131-1.old/Makefile.in	Fri Feb  1 07:26:19 2002
+++ ash-20020131-1.new/Makefile.in	Tue Apr  9 20:00:52 2002
@@ -83,7 +83,7 @@
 
 sh_new_MANS = ash.1
 
-CPPFLAGS = -DSMALL -DSHELL -Dlint -Wall -Wunused
+CPPFLAGS = -DSMALL -DSHELL -Dlint -Wall -Wunused -D__unused=
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_CLEAN_FILES = 
diff -Naur ash-20020131-1.old/cd.c ash-20020131-1.new/cd.c
--- ash-20020131-1.old/cd.c	Sun Apr 15 22:51:45 2001
+++ ash-20020131-1.new/cd.c	Tue Apr  9 19:24:34 2002
@@ -48,6 +48,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <errno.h>
+#include <limits.h>
 
 /*
  * The cd and pwd commands.
@@ -96,9 +97,9 @@
 char *
 getpwd()
 {
-	char buf[MAXPWD];
+	char buf[PATH_MAX];
 
-	if (!getcwd(buf, MAXPWD))
+	if (!getcwd(buf, PATH_MAX))
 		return NULL;
 
 	return savestr(buf);
