diff -ru dnsmasq-2.20/src/config.h dnsmasq-2.20-patched/src/config.h
--- dnsmasq-2.20/src/config.h	2005-01-16 18:25:56.000000000 +0100
+++ dnsmasq-2.20-patched/src/config.h	2005-02-14 17:34:09.000000000 +0100
@@ -31,16 +31,16 @@
 #else
 #  define RESOLVFILE "/etc/resolv.conf"
 #endif
-#define RUNFILE "/var/run/dnsmasq.pid"
+#define RUNFILE "@PREFIX@/var/run/dnsmasq.pid"
 #if defined(__FreeBSD__) || defined (__OpenBSD__)
-#   define LEASEFILE "/var/db/dnsmasq.leases"
+#   define LEASEFILE "@PREFIX@/var/db/dnsmasq.leases"
 #else
-#   define LEASEFILE "/var/lib/misc/dnsmasq.leases"
+#   define LEASEFILE "@PREFIX@/var/lib/misc/dnsmasq.leases"
 #endif
 #if defined(__FreeBSD__)
 #   define CONFFILE "/usr/local/etc/dnsmasq.conf"
 #else
-#   define CONFFILE "/etc/dnsmasq.conf"
+#   define CONFFILE "@PREFIX@/etc/dnsmasq.conf"
 #endif
 #define DEFLEASE 3600 /* default lease time, 1 hour */
 #define CHUSER "nobody"
