diff -urN dsocks-1.6/Makefile dsocks.patched/Makefile
--- dsocks-1.6/Makefile	2006-10-11 21:13:58.000000000 -0700
+++ dsocks.patched/Makefile	2010-08-13 18:45:29.000000000 -0700
@@ -7,13 +7,13 @@
 SRCS		= dsocks.c atomicio.c
 NOPROFILE	= yes
 
-LIBDIR		= /usr/local/lib
+LIBDIR		= @PREFIX@/lib
 
 .if ${unix} == "We run Darwin, not UNIX."
-OSXFLAGS	= -nostdlib -flat_namespace -fno-common -undefined suppress
+OSXFLAGS	= -flat_namespace -fno-common -undefined suppress
 CFLAGS		+= $(OSXFLAGS)
 # XXX - hack around missing LDFLAGS in bsd.lib.mk target
-LDADD		= $(OSXFLAGS)
+LDADD		= $(OSXFLAGS) -install_name $(LIBDIR)/libdsocks.1.dylib
 SHLIB_MAJOR	= 1
 .endif
 
diff -urN dsocks-1.6/dsocks-torify.sh dsocks.patched/dsocks-torify.sh
--- dsocks-1.6/dsocks-torify.sh	2006-10-11 21:13:58.000000000 -0700
+++ dsocks.patched/dsocks-torify.sh	2010-08-13 18:46:42.000000000 -0700
@@ -7,8 +7,9 @@
 # you probably want to run tor-dns-proxy.py also...
 
 # for MacOS X...
-#LIBDSOCKS=/usr/local/lib/libdsocks.dylib
-#DYLD_INSERT_LIBRARIES=$LIBDSOCKS DYLD_FORCE_FLAT_NAMESPACE=1 exec "$@"
+LIBDSOCKS=@PREFIX@/lib/libdsocks.1.dylib
+DYLD_INSERT_LIBRARIES=$LIBDSOCKS DYLD_FORCE_FLAT_NAMESPACE=1 exec "$@"
 
-LIBDSOCKS=/usr/local/lib/libdsocks.so.1.0
-LD_PRELOAD=$LIBDSOCKS exec "$@"
+# for BSD
+#LIBDSOCKS=/usr/local/lib/libdsocks.so.1.0
+#LD_PRELOAD=$LIBDSOCKS exec "$@"
diff -urN dsocks-1.6/dsocks.sh dsocks.patched/dsocks.sh
--- dsocks-1.6/dsocks.sh	2006-10-11 21:13:58.000000000 -0700
+++ dsocks.patched/dsocks.sh	2010-08-13 18:47:03.000000000 -0700
@@ -8,8 +8,9 @@
 #export LOCALDOMAIN="int.example.com"
 
 # for MacOS X...
-#LIBDSOCKS=/usr/local/lib/libdsocks.dylib
-#DYLD_INSERT_LIBRARIES=$LIBDSOCKS DYLD_FORCE_FLAT_NAMESPACE=1 exec "$@"
+LIBDSOCKS==@PREFIX@/lib/libdsocks.1.dylib
+DYLD_INSERT_LIBRARIES=$LIBDSOCKS DYLD_FORCE_FLAT_NAMESPACE=1 exec "$@"
 
-LIBDSOCKS=/usr/local/lib/libdsocks.so.1.0
-LD_PRELOAD=$LIBDSOCKS exec "$@"
+# for BSD
+#LIBDSOCKS=/usr/local/lib/libdsocks.so.1.0
+#LD_PRELOAD=$LIBDSOCKS exec "$@"
