diff -Nurd -x'*~' exact-1.41.orig/Makefile.in exact-1.41/Makefile.in
--- exact-1.41.orig/Makefile.in	2004-03-31 15:40:02.000000000 -0500
+++ exact-1.41/Makefile.in	2012-02-08 08:10:20.000000000 -0500
@@ -65,11 +65,9 @@
 install: all
 	$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(sbindir) $(DESTDIR)/$(sysconfdir) $(DESTDIR)/$(localstatedir)/run/exact \
 		$(DESTDIR)/$(localstatedir)/state/exact $(DESTDIR)/$(mandir)/man8 $(DESTDIR)/$(mandir)/man5
-	chown $(username) $(DESTDIR)/$(localstatedir)/run/exact $(DESTDIR)/$(localstatedir)/state/exact
-	chgrp $(groupname) $(DESTDIR)/$(localstatedir)/run/exact $(DESTDIR)/$(localstatedir)/state/exact
 	$(INSTALL) -m755 exact $(DESTDIR)/$(sbindir)/exact
 	$(INSTALL) -m644 exact.conf $(DESTDIR)/$(sysconfdir)/exact.conf
-	$(INSTALL) -m644 exact-servers.conf $(DEST)/$(sysconfdir)/exact-servers.conf
+	$(INSTALL) -m644 exact-servers.conf $(DESTDIR)/$(sysconfdir)/exact-servers.conf
 	$(INSTALL) -m644 exact.8 $(DESTDIR)/$(mandir)/man8
 	$(INSTALL) -m644 exact.conf.5 $(DESTDIR)/$(mandir)/man5
 
