diff -ruN ircd-hybrid-7.0.3/doc/Makefile.in ircd-hybrid-7.0.3-new/doc/Makefile.in
--- ircd-hybrid-7.0.3/doc/Makefile.in	2002-07-09 20:05:48.000000000 -0400
+++ ircd-hybrid-7.0.3-new/doc/Makefile.in	2012-02-08 13:11:26.000000000 -0500
@@ -14,19 +14,19 @@
 RM		= @RM@
 CP		= @CP@
 
-prefix		= @prefix@
+prefix		= @DSTROOT@@prefix@
 exec_prefix	= @exec_prefix@
 exec_suffix	= @exec_suffix@
 bindir		= @bindir@
 libexecdir	= @libexecdir@
-sysconfdir	= @sysconfdir@
+sysconfdir	= @DSTROOT@@prefix@/etc/ircd
 localstatedir	= @localstatedir@
 # Change this later! -- adrian
-moduledir	= @prefix@/modules
+moduledir	= @DSTROOT@@prefix@/modules
 automoduledir	= @prefix@/modules/autoload
 
 # Local to the etc Makefile
-mandir          = @prefix@/man8
+mandir          = @DSTROOT@@prefix@/share/man/man8
 MANPAGES        = ircd.8
 
 CONFS		= example.conf example.efnet.conf
diff -ruN ircd-hybrid-7.0.3/help/Makefile.in ircd-hybrid-7.0.3-new/help/Makefile.in
--- ircd-hybrid-7.0.3/help/Makefile.in	2002-02-02 11:01:13.000000000 -0500
+++ ircd-hybrid-7.0.3-new/help/Makefile.in	2012-02-08 13:11:26.000000000 -0500
@@ -6,10 +6,10 @@
 INSTALL_DATA=	@INSTALL_DATA@
 RM=		@RM@
 
-prefix=		@prefix@
+prefix=		@DSTROOT@@prefix@
 exec_prefix=	@execprefix@
-uhelpdir=	${prefix}/help/users
-ohelpdir=	${prefix}/help/opers
+uhelpdir=	${prefix}/share/doc/ircd-hybrid/users
+ohelpdir=	${prefix}/share/doc/ircd-hybrid/opers
 
 SYMLINKS=	topic accept cjoin cmode admin names links away whowas \
 		version kick who invite quit join list nick oper part \
@@ -62,14 +62,9 @@
 
 install:
 	@echo installing help files...
-	-@if test ! -d $(prefix)/help; then \
-		echo "mkdir $(prefix)/help"; \
-		mkdir $(prefix)/help; \
-		echo "mkdir $(prefix)/help/users"; \
-		mkdir $(prefix)/help/users; \
-		echo "mkdir $(prefix)/help/opers"; \
-		mkdir $(prefix)/help/opers; \
-	fi
+	mkdir -p $(ohelpdir);
+	mkdir -p $(uhelpdir);
+
 	@for help in opers/*; do \
 	if [ -f $$help ]; then \
 		${INSTALL_DATA} $$help $(ohelpdir); \
diff -ruN ircd-hybrid-7.0.3/include/Makefile.in ircd-hybrid-7.0.3-new/include/Makefile.in
--- ircd-hybrid-7.0.3/include/Makefile.in	2002-04-29 18:30:24.000000000 -0400
+++ ircd-hybrid-7.0.3-new/include/Makefile.in	2012-02-08 13:11:26.000000000 -0500
@@ -6,7 +6,7 @@
 INSTALL_BIN=	@INSTALL_PROGRAM@
 RM=		@RM@
 
-prefix=		@prefix@
+prefix=		@DSTROOT@@prefix@
 exec_prefix=	@exec_prefix@
 bindir=		@bindir@
 sysconfdir=	@sysconfdir@
diff -ruN ircd-hybrid-7.0.3/include/config.h ircd-hybrid-7.0.3-new/include/config.h
--- ircd-hybrid-7.0.3/include/config.h	2003-10-25 22:08:12.000000000 -0400
+++ ircd-hybrid-7.0.3-new/include/config.h	2012-02-08 13:11:26.000000000 -0500
@@ -137,13 +137,13 @@
 /* dirs */
 #define DPATH   IRCD_PREFIX                                                     
 #define BINPATH IRCD_PREFIX "/bin/"
-#define MODPATH IRCD_PREFIX "/modules/"
-#define AUTOMODPATH IRCD_PREFIX "/modules/autoload/"
-#define MSGPATH IRCD_PREFIX "/messages/"
-#define ETCPATH IRCD_PREFIX "/etc"
-#define LOGPATH IRCD_PREFIX "/logs"
-#define UHPATH   IRCD_PREFIX "/help/users"
-#define HPATH  IRCD_PREFIX "/help/opers"
+#define MODPATH IRCD_PREFIX "/lib/ircd-modules/"
+#define AUTOMODPATH IRCD_PREFIX "/lib/ircd-modules/autoload/"
+#define MSGPATH IRCD_PREFIX "/share/locale"
+#define ETCPATH IRCD_PREFIX "/etc/ircd"
+#define LOGPATH IRCD_PREFIX "/var/log"
+#define UHPATH   IRCD_PREFIX "/share/doc/ircd-hybrid/users"
+#define HPATH  IRCD_PREFIX "/share/doc/ircd-hybrid/opers"
 
 /* files */
 #define SPATH   BINPATH "/ircd"                 /* ircd executable */
diff -ruN ircd-hybrid-7.0.3/include/s_conf.h ircd-hybrid-7.0.3-new/include/s_conf.h
--- ircd-hybrid-7.0.3/include/s_conf.h	2004-03-17 22:57:21.000000000 -0500
+++ ircd-hybrid-7.0.3-new/include/s_conf.h	2012-02-08 13:11:26.000000000 -0500
@@ -53,7 +53,6 @@
 
 extern FBFILE* conf_fbfile_in;
 extern char conf_line_in[256];
-extern struct ConfItem* yy_aconf;
 
 struct ConfItem
 {
diff -ruN ircd-hybrid-7.0.3/messages/Makefile.in ircd-hybrid-7.0.3-new/messages/Makefile.in
--- ircd-hybrid-7.0.3/messages/Makefile.in	2001-07-04 16:42:43.000000000 -0400
+++ ircd-hybrid-7.0.3-new/messages/Makefile.in	2012-02-08 13:11:26.000000000 -0500
@@ -9,14 +9,14 @@
 
 OBJS = ${SRCS:.po=.mo}
 
-prefix          = @prefix@
+prefix          = @DSTROOT@@prefix@
 exec_prefix     = @exec_prefix@
 exec_suffix     = @exec_suffix@
 bindir          = @bindir@
 libexecdir      = @libexecdir@
 sysconfdir      = @sysconfdir@
 localstatedir   = @localstatedir@
-messagedir       = @prefix@/messages
+messagedir      = @DSTROOT@@prefix@/share/locale
 
 
 all: ${OBJS}
diff -ruN ircd-hybrid-7.0.3/modules/Makefile.in ircd-hybrid-7.0.3-new/modules/Makefile.in
--- ircd-hybrid-7.0.3/modules/Makefile.in	2004-02-09 23:28:37.000000000 -0500
+++ ircd-hybrid-7.0.3-new/modules/Makefile.in	2012-02-08 13:11:26.000000000 -0500
@@ -27,10 +27,10 @@
 
 IRCDLIBS        = @LIBS@ $(SSL_LIBS)
 
-prefix          = @prefix@
+prefix          = @DSTROOT@@prefix@
 # Change this later! -- adrian
-moduledir       = @prefix@/modules
-automoduledir   = @prefix@/modules/autoload
+moduledir       = @DSTROOT@@prefix@/lib/ircd-modules
+automoduledir   = @DSTROOT@@prefix@/lib/ircd-modules/autoload
 
 INCLUDES	= -I../include -I../adns $(SSL_INCLUDES)
 CPPFLAGS	= ${INCLUDES} @CPPFLAGS@
diff -ruN ircd-hybrid-7.0.3/servlink/Makefile.in ircd-hybrid-7.0.3-new/servlink/Makefile.in
--- ircd-hybrid-7.0.3/servlink/Makefile.in	2002-06-03 08:15:08.000000000 -0400
+++ ircd-hybrid-7.0.3-new/servlink/Makefile.in	2012-02-08 13:11:26.000000000 -0500
@@ -17,7 +17,7 @@
 MKDEP           = @MKDEP@ -DIRCD_PREFIX=\"@prefix@\"
 MV              = @MV@
 RM              = @RM@
-prefix          = @prefix@
+prefix          = @DSTROOT@@prefix@
 exec_prefix     = @exec_prefix@
 bindir          = @bindir@
 libexecdir      = @libexecdir@
diff -ruN ircd-hybrid-7.0.3/src/Makefile.in ircd-hybrid-7.0.3-new/src/Makefile.in
--- ircd-hybrid-7.0.3/src/Makefile.in	2003-10-25 22:08:20.000000000 -0400
+++ ircd-hybrid-7.0.3-new/src/Makefile.in	2012-02-08 13:11:56.000000000 -0500
@@ -17,7 +17,7 @@
 MV		= @MV@
 RM		= @RM@
 YACC		= @YACC@
-prefix		= @prefix@
+prefix		= @DSTROOT@@prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
 libexecdir	= @libexecdir@
@@ -33,7 +33,7 @@
 SSL_LIBS	= @SSL_LIBS@
 SSL_INCLUDES	= @SSL_INCLUDES@
 
-IRCDLIBS	= @MODULES_LIBS@ -L../adns -ladns @LIBS@ $(SSL_LIBS)
+IRCDLIBS	= @MODULES_LIBS@ ../adns/libadns.a @LIBS@ $(SSL_LIBS)
 
 INCLUDES	= -I../include -I../adns $(SSL_INCLUDES)
 CPPFLAGS	= ${INCLUDES} @CPPFLAGS@
@@ -140,10 +140,10 @@
 		echo "mkdir $(sysconfdir)"; \
 		mkdir $(sysconfdir); \
 	fi
-	-@if test ! -d $(localstatedir)/logs; then \
-		echo "mkdir $(localstatedir)/logs"; \
-		mkdir $(localstatedir)/logs; \
-	fi
+#	-@if test ! -d $(localstatedir)/logs; then \
+#		echo "mkdir $(localstatedir)/logs"; \
+#		mkdir $(localstatedir)/logs; \
+#	fi
 
 install: install-mkdirs build
 	@for i in $(PROGS); do \
diff -ruN ircd-hybrid-7.0.3/src/ircd.c ircd-hybrid-7.0.3-new/src/ircd.c
--- ircd-hybrid-7.0.3/src/ircd.c	2003-10-25 22:08:22.000000000 -0400
+++ ircd-hybrid-7.0.3-new/src/ircd.c	2012-02-08 13:11:26.000000000 -0500
@@ -679,8 +679,8 @@
   read_conf_files(YES);         /* cold start init conf files */
 #ifndef STATIC_MODULES
 
-  mod_add_path(IRCD_PREFIX "/modules");
-  mod_add_path(IRCD_PREFIX "/modules/autoload");
+  mod_add_path(IRCD_PREFIX "/lib/ircd-modules");
+  mod_add_path(IRCD_PREFIX "/lib/ircd-modules/autoload");
 #endif
    
   initialize_server_capabs();   /* Set up default_server_capabs */
diff -ruN ircd-hybrid-7.0.3/tools/Makefile.in ircd-hybrid-7.0.3-new/tools/Makefile.in
--- ircd-hybrid-7.0.3/tools/Makefile.in	2002-07-11 19:40:27.000000000 -0400
+++ ircd-hybrid-7.0.3-new/tools/Makefile.in	2012-02-08 13:11:26.000000000 -0500
@@ -22,15 +22,15 @@
 INCLUDES	= -I../include $(SSL_INCLUDES)
 CPPFLAGS	= ${INCLUDES} @CPPFLAGS@
 
-prefix		= @prefix@
+prefix		= @DSTROOT@@prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
 libexecdir	= @libexecdir@
 sysconfdir	= @sysconfdir@
 localstatedir	= @localstatedir@
 # Change this later! -- adrian
-moduledir	= @prefix@/modules
-automoduledir	= @prefix@/modules/autoload
+moduledir	= @DSTROOT@@prefix@/modules
+automoduledir	= @DSTROOT@@prefix@/modules/autoload
 
 PROGS		= viconf mkpasswd convertconf convertilines convertklines encspeed
 
