diff -Nurd -x'*~' libdlna-0.2.4.orig/Makefile libdlna-0.2.4/Makefile
--- libdlna-0.2.4.orig/Makefile	2012-02-26 14:34:00.000000000 -0500
+++ libdlna-0.2.4/Makefile	2012-06-13 10:25:44.000000000 -0400
@@ -18,8 +18,8 @@
 
 SUBDIRS = src \
 
-CFLAGS += -Isrc
-LDFLAGS += -Lsrc -ldlna
+CFLAGS := -Isrc $(CFLAGS)
+LDFLAGS := src/libdlna.$(VERSION).dylib $(LDFLAGS)
 
 ifeq ($(BUILD_STATIC),yes)
   LDFLAGS += $(EXTRALIBS)
diff -Nurd -x'*~' libdlna-0.2.4.orig/configure libdlna-0.2.4/configure
--- libdlna-0.2.4.orig/configure	2012-02-26 14:34:00.000000000 -0500
+++ libdlna-0.2.4/configure	2012-06-13 10:08:37.000000000 -0400
@@ -659,7 +659,7 @@
   return 0;
 }
 EOF
-VERSION=`$TMPE`
+VERSION=@VERSION@
 restore_flags
 
 
@@ -733,7 +733,7 @@
 #################################################
 #   make pkg-config files
 #################################################
-pkgconfig_generate libdlna "DLNA (Digital Living Network Alliance) library" "$VERSION" "-ldlna $extralibs" "libavformat libavcodec"
+pkgconfig_generate libdlna "DLNA (Digital Living Network Alliance) library" "$VERSION" "-ldlna" ""
 
 clean
 exit 0
diff -Nurd -x'*~' libdlna-0.2.4.orig/src/Makefile libdlna-0.2.4/src/Makefile
--- libdlna-0.2.4.orig/src/Makefile	2012-02-26 14:34:00.000000000 -0500
+++ libdlna-0.2.4/src/Makefile	2012-06-13 10:35:02.000000000 -0400
@@ -21,10 +21,10 @@
 endif
 
 LIBNAME = libdlna
-LIBNAME_SHARED = ${LIBNAME}.so
-LIBNAME_STATIC = ${LIBNAME}.a
-LIBNAME_VERSION = $(LIBNAME_SHARED).$(VERSION)
-LIBNAME_MAJOR = $(LIBNAME_SHARED).$(shell echo $(VERSION) | cut -f1 -d.)
+LIBNAME_SHARED = $(LIBNAME).dylib
+LIBNAME_STATIC = $(LIBNAME).a
+LIBNAME_VERSION = $(LIBNAME).$(VERSION).dylib
+LIBNAME_MAJOR = $(LIBNAME).$(shell echo $(VERSION) | cut -f1 -d.).dylib
 
 SRCS =  profiles.c \
 	containers.c \
@@ -97,7 +97,7 @@
 	@echo "#############################################"
 
 lib_shared: lib_shared_info_pre $(LOBJS) lib_shared_info_post
-	$(CC) -shared -Wl,-soname,$(LIBNAME_MAJOR) $(LOBJS) \
+	$(CC) -dynamiclib -compatibility_version 1.0.0 -current_version 1.0.0 -install_name $(PREFIX)/lib/$(LIBNAME_MAJOR) $(LOBJS) \
 	  $(LDFLAGS) $(EXTRALIBS) -o $(LIBNAME_VERSION)
 	$(LN) -sf $(LIBNAME_VERSION) $(LIBNAME_MAJOR)
 	$(LN) -sf $(LIBNAME_MAJOR) $(LIBNAME_SHARED)
@@ -111,7 +111,7 @@
 	 ( find -name '*.[chS]' -print ) | xargs ctags -a;
 
 clean:
-	-$(RM) -f *.o *.lo *.a *.so*
+	-$(RM) -f *.o *.lo *.a *.dylib*
 	-$(RM) -f .depend
 
 install_static: lib_static
diff -Nurd -x'*~' libdlna-0.2.4.orig/src/upnp_dms.c libdlna-0.2.4/src/upnp_dms.c
--- libdlna-0.2.4.orig/src/upnp_dms.c	2012-02-26 14:34:00.000000000 -0500
+++ libdlna-0.2.4/src/upnp_dms.c	2012-06-13 10:07:45.000000000 -0400
@@ -19,6 +19,7 @@
  * Foundation, Inc, 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 
