diff -ru libnet.orig/Makefile.am libnet/Makefile.am
--- libnet.orig/Makefile.am	2004-03-10 15:57:09.000000000 -0500
+++ libnet/Makefile.am	2007-01-01 16:46:45.000000000 -0500
@@ -8,7 +8,7 @@
 
 include $(top_srcdir)/Makefile.am.common
 
-SUBDIRS = include src sample
+SUBDIRS = include src
 
 EXTRA_DIST = Makefile.am.common
 
diff -ru libnet.orig/configure.in libnet/configure.in
--- libnet.orig/configure.in	2004-03-01 15:26:11.000000000 -0500
+++ libnet/configure.in	2007-01-01 17:22:33.000000000 -0500
@@ -21,7 +21,7 @@
 dnl
 AC_PROG_CC
 AC_PROG_INSTALL
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 
 AC_CHECK_HEADERS(sys/sockio.h)
 
@@ -157,7 +157,10 @@
 
 *darwin*)
     AC_DEFINE(HAVE_SOCKADDR_SA_LEN)
-    LIBNET_CONFIG_DEFINES="-DHAVE_SOCKADDR_SA_LEN"
+    AC_DEFINE(LIBNET_BSD_BYTE_SWAP)
+    LIBNET_CONFIG_DEFINES="-DHAVE_SOCKADDR_SA_LEN -DLIBNET_BSD_BYTE_SWAP"
+    LIBNET_CONFIG_CFLAGS="-I${prefix}/include"
+    LIBNET_CONFIG_LIBS="-L${prefix}/libs $LIBNET_CONFIG_LIBS"
     ;;
 
 *solaris*)
@@ -229,6 +232,9 @@
 
 esac
 
+LTLIBOBJS=`echo "$LIBOBJS" | sed 's/\.[^.]* /.lo /g;s/\.[^.]*$/.lo/'`
+AC_SUBST(LTLIBOBJS)
+
 AC_OUTPUT(Makefile src/Makefile include/Makefile \
     include/libnet/Makefile sample/Makefile version.h include/libnet.h \
     libnet-config, chmod +x libnet-config)
diff -ru libnet.orig/src/Makefile.am libnet/src/Makefile.am
--- libnet.orig/src/Makefile.am	2004-03-11 13:50:20.000000000 -0500
+++ libnet/src/Makefile.am	2007-01-01 17:06:26.000000000 -0500
@@ -8,9 +8,9 @@
 
 include $(top_srcdir)/Makefile.am.common
 
-lib_LIBRARIES = libnet.a
+lib_LTLIBRARIES = libnet.la
 
-libnet_a_SOURCES = libnet_asn1.c \
+libnet_la_SOURCES = libnet_asn1.c \
 			libnet_build_802.1q.c \
 			libnet_build_802.1x.c \
 			libnet_build_802.2.c \
@@ -57,15 +57,7 @@
 			libnet_version.c \
 			libnet_write.c
 
-EXTRA_libnet_a_SOURCES = libnet_link_bpf.c \
-			libnet_link_dlpi.c \
-			libnet_link_linux.c \
-			libnet_link_nit.c \
- 			libnet_link_none.c \
-			libnet_link_pf.c \
-			libnet_link_snit.c \
-			libnet_link_snoop.c \
-			libnet_link_win32.c
+libnet_la_LIBADD = @LTLIBOBJS@
 
-libnet_a_LIBADD = @LIBOBJS@
+libnet_la_LDFLAGS = -release 1.1
 
