diff -ruN noip-2.0.12.orig/Makefile noip-2.0.12/Makefile
--- noip-2.0.12.orig/Makefile	Thu May  8 15:13:28 2003
+++ noip-2.0.12/Makefile	Sun Jun  1 23:07:00 2003
@@ -1,13 +1,13 @@
 TGT=noip2
 CC=gcc
 
-PREFIX=/usr/local
+PREFIX=@FINKPREFIX@
 CONFDIR=${PREFIX}/etc
 BINDIR=${PREFIX}/bin
 
 # these defines are for Linux
 LIBS=
-ARCH=linux
+#ARCH=linux
 
 # for BSD systems that have getifaddr(), uncomment the next line
 #ARCH=bsd_with_getifaddrs
@@ -15,6 +15,8 @@
 # for early BSD systems without getifaddrs(), uncomment the next line
 #ARCH=bsd
 
+# for Darwin/OS X
+ARCH=darwin
 
 # for solaris, uncomment the next two lines
 # LIBS=-lsocket -lnsl
diff -ruN noip-2.0.12.orig/noip2.c noip-2.0.12/noip2.c
--- noip-2.0.12.orig/noip2.c	Fri May 23 16:55:14 2003
+++ noip-2.0.12/noip2.c	Sun Jun  1 23:11:39 2003
@@ -109,6 +109,10 @@
 #include <net/if_types.h>
 #endif
 
+#ifdef darwin
+#define bsd
+#endif
+
 #ifdef bsd
 #include <sys/sockio.h>
 #include <net/if_types.h>
@@ -494,6 +498,7 @@
 	if (handle_config_error(parse_config()) != SUCCESS) 
 	    return -1;
 
+#ifndef darwin
         /* drop root privileges after reading config */
         if (geteuid() == 0) {
             if ((nobody = getpwnam("nobody")) != NULL) { // if "nobody" exists
@@ -503,6 +508,7 @@
                 seteuid(nobody->pw_uid);
 	    }
 	}
+#endif /* Darwin's nobody doesn't have shm access */
 
 	if (*IPaddress != 0) {
 	    if (background) {
