diff -urN oinkmaster.orig/oinkmaster.conf oinkmaster/oinkmaster.conf
--- oinkmaster.orig/oinkmaster.conf	2005-04-16 09:29:41.000000000 +0200
+++ oinkmaster/oinkmaster.conf	2005-05-11 13:17:03.000000000 +0200
@@ -92,7 +92,7 @@
 # contains the Cygwin binaries and not the native Win32 binaries or 
 # you will get problems.
 # Assume UNIX style by default:
-path = /bin:/usr/bin:/usr/local/bin
+path = @PREFIX@/bin:/bin:/usr/bin:/usr/local/bin
 
 # Example if running native Win32 or standalone Cygwin:
 # path = c:\oinkmaster;c:\oinkmaster\bin
diff -urN oinkmaster.orig/oinkmaster.pl oinkmaster/oinkmaster.pl
--- oinkmaster.orig/oinkmaster.pl	2005-04-16 09:29:41.000000000 +0200
+++ oinkmaster/oinkmaster.pl	2005-05-11 13:18:27.000000000 +0200
@@ -130,7 +130,7 @@
 
 # Default locations for configuration file.
 my @DEFAULT_CONFIG_FILES = qw(
-    /etc/oinkmaster.conf
+    @PREFIX@/etc/oinkmaster.conf
     /usr/local/etc/oinkmaster.conf
 );
 
