--- ecl-0.9i.orig/src/doc/Makefile.in	2007-07-02 08:00:31.000000000 +0200
+++ ecl-0.9i/src/doc/Makefile.in	2007-07-02 08:44:49.000000000 +0200
@@ -47,9 +47,6 @@
 	$(mkinstalldirs) $(DESTDIR)$(infodir)
 	for k in $(INFO_FILES); do \
 	  $(INSTALL_DATA) $$k $(DESTDIR)$(infodir); \
-	  if [ -x $(INSTALL_INFO) ]; then \
-	    $(INSTALL_INFO) --info-dir=$(DESTDIR)$(infodir) $$k; \
-	  fi; \
 	done
 	$(mkinstalldirs) $(DESTDIR)$(docdir)
 	for i in Copyright LGPL; do \
--- ecl-0.9i.orig/src/compile.lsp.in      2011-08-23 07:45:44.000000000 -0400
+++ ecl-0.9i/src/compile.lsp.in 2006-06-30 08:35:20.000000000 -0400
@@ -107,7 +107,7 @@
        (c::*cc-flags* (concatenate 'string "-DECL_API -I@true_builddir@/c " c::*cc-flags*))
        (extra-args nil))
   #+darwin
-  (push "-Wl,-install_name,@libdir\@/@SHAREDPREFIX@ecl.@SHAREDEXT@"
+  (push "-Wl,-install_name,@libdir\@@SHAREDPREFIX@ecl.@SHAREDEXT@"
 	extra-args)
   #+(or mingw32 cygwin)
   (progn
