diff -duar gpc-3.4.3d6-sources.ori/gpc-3.4.3d6/gcc/configure gpc-3.4.3d6-sources/gpc-3.4.3d6/gcc/configure
--- gpc-3.4.3d6-sources.ori/gpc-3.4.3d6/gcc/configure	Fri Nov  5 13:14:05 2004
+++ gpc-3.4.3d6-sources/gpc-3.4.3d6/gcc/configure	Sat Mar 12 02:12:02 2005
@@ -6368,7 +6368,7 @@
   # sources; use -no-cpp-precomp to get to GNU cpp.
   # Apple's GCC has bugs in designated initializer handling, so disable
   # that too.
-  stage1_cflags="-no-cpp-precomp -DHAVE_DESIGNATED_INITIALIZERS=0"
+  stage1_cflags="-DHAVE_DESIGNATED_INITIALIZERS=0"
   ;;
 esac
 
