diff -urN -x'*~' verilog-0.9.4-orig/libveriuser/getsimtime.c verilog-0.9.4/libveriuser/getsimtime.c
--- verilog-0.9.4-orig/libveriuser/getsimtime.c	2011-03-17 20:26:38.000000000 +0100
+++ verilog-0.9.4/libveriuser/getsimtime.c	2011-09-24 21:09:31.000000000 +0200
@@ -85,7 +85,10 @@
 
 /* Alias for commercial simulators */
 PLI_INT32 tf_getlongsimtime(PLI_INT32 *high) \
-      __attribute__ ((weak, alias ("tf_getlongtime")));
+{
+      return tf_igetlongtime(high, 0);
+}
+//      __attribute ((weak, alias ("tf_getlongtime")));
 
 void tf_scale_longdelay(void*obj, PLI_INT32 low, PLI_INT32 high,
 			PLI_INT32 *alow, PLI_INT32 *ahigh)
