--- mdk-1.2.3.orig/configure.in	2006-09-28 22:16:03.000000000 +0200
+++ mdk-1.2.3/configure.in	2006-09-28 22:21:38.000000000 +0200
@@ -89,15 +89,15 @@
 if test x$wguile = xtrue; then
 dnl  AC_CHECK_HEADER(guile/gh.h,,wguile=false)
   AC_MSG_CHECKING(for Guile)
-  guile-config link > /dev/null || {
+  guile-1.6-config link > /dev/null || {
     AC_MSG_RESULT(no)
     wguile=false
   }
 fi
 
 if test x$wguile = xtrue; then
-  GUILE_CFLAGS="`guile-config compile`"
-  GUILE_LDFLAGS="`guile-config link`"
+  GUILE_CFLAGS="`guile-1.6-config compile`"
+  GUILE_LDFLAGS="`guile-1.6-config link`"
   LIBS="$LIBS $GUILE_LDFLAGS"
   AM_CFLAGS="$AM_CFLAGS $GUILE_CFLAGS"
   AC_SUBST(MAKE_GUILE)
--- mdk-1.2.3.orig/lib/Makefile.am	2005-09-21 17:52:52.000000000 +0200
+++ mdk-1.2.3/lib/Makefile.am	2005-09-21 17:52:52.000000000 +0200
@@ -13,5 +13,5 @@
 EXTRA_DIST = getopt_long.c getopt.h
 
 noinst_LIBRARIES = libreplace.a
-libreplace_a_SOURCES =
+libreplace_a_SOURCES = foo.c
 libreplace_a_LIBADD = @LIBOBJS@
\ No newline at end of file
--- mdk-1.2.3.orig/lib/foo.c	2005-09-21 18:05:48.000000000 +0200
+++ mdk-1.2.3/lib/foo.c	2005-09-21 18:06:00.000000000 +0200
@@ -0,0 +1,5 @@
+int
+foo_for_darwin (void)
+{
+  return 0;
+}
--- mdk-1.2.3.orig/mixgtk/Makefile.am	2005-09-21 18:24:14.000000000 +0200
+++ mdk-1.2.3/mixgtk/Makefile.am	2005-09-21 18:24:22.000000000 +0200
@@ -31,7 +31,6 @@
 LDADD = $(top_builddir)/mixlib/libmix.a $(top_builddir)/lib/libreplace.a\
         $(top_builddir)/mixguile/libmixguile.a $(INTLLIBS)
 
-AM_LDFLAGS = -Wl,--export-dynamic
 bin_PROGRAMS = gmixvm
 gmixvm_SOURCES = gmixvm.c mixgtk.h mixgtk.c \
                  mixgtk_config.h mixgtk_config.c \
--- mdk-1.2.3.orig/mixgtk/mixgtk_device.c	2006-08-24 12:37:25.000000000 +0200
+++ mdk-1.2.3/mixgtk/mixgtk_device.c	2006-08-24 12:43:03.000000000 +0200
@@ -188,7 +188,9 @@
   input_list_ = gtk_list_store_new (1, G_TYPE_STRING);
   gtk_entry_completion_set_model (completion, GTK_TREE_MODEL (input_list_));
   gtk_entry_completion_set_popup_completion (completion, TRUE);
+#if GTK_CHECK_VERSION(2,8,0)
   gtk_entry_completion_set_popup_single_match (completion, TRUE);
+#endif
   gtk_entry_completion_set_inline_completion (completion, FALSE);
   gtk_entry_completion_set_minimum_key_length (completion, 1);
   gtk_entry_completion_set_text_column (completion, 0);
--- mdk-1.2.3.orig/mixgtk/mixgtk_external.c	2006-08-24 12:55:46.000000000 +0200
+++ mdk-1.2.3/mixgtk/mixgtk_external.c	2006-08-24 13:10:46.000000000 +0200
@@ -44,8 +44,8 @@
 
 static const gchar *EDITOR_KEY_ = "Editor";
 static const gchar *MIXASM_KEY_ = "Mixasm";
-static const gchar *DEFAULT_EDITOR_CMD_ = "/usr/bin/xterm -e vi %s";
-static const gchar *DEFAULT_ASM_CMD_ = "/usr/bin/mixasm";
+static const gchar *DEFAULT_EDITOR_CMD_ = "/usr/bin/open -a TextEdit %s";
+static const gchar *DEFAULT_ASM_CMD_ = "@PREFIX@/bin/mixasm %s";
 
 static mix_vm_cmd_dispatcher_t *dispatcher_;
 
@@ -104,6 +104,7 @@
     }
 
   gtk_file_filter_add_mime_type (filter, "application/x-executable");
+  gtk_file_filter_add_mime_type (filter, "application/octet-stream");
   gtk_file_chooser_set_filter
     (GTK_FILE_CHOOSER (ext_wdg_[ext_wdg_asm_chooser]), filter);
   gtk_file_chooser_set_filter
