diff -ru PyQt-x11-gpl-4.9.4.orig/configure.py PyQt-x11-gpl-4.9.4/configure.py
--- PyQt-x11-gpl-4.9.4.orig/configure.py	2012-06-26 09:09:24.000000000 -0400
+++ PyQt-x11-gpl-4.9.4/configure.py	2012-08-12 09:10:03.000000000 -0400
@@ -970,7 +970,7 @@
 
                 config_args = ducfg.get("CONFIG_ARGS", "")
 
-                if sys.platform == "darwin":
+                if sys.platform == "notdarwin":
                     dynamic_pylib = "--enable-framework" in config_args
 
                     # It's probably a Python bug that the library name doesn't
@@ -980,15 +980,7 @@
                     dynamic_pylib = "--enable-shared" in config_args
 
                 if dynamic_pylib:
-                    if glob.glob("%s/lib/libpython%d.%d*" % (ducfg["exec_prefix"], py_major, py_minor)):
-                        lib_dir_flag = quote("-L%s/lib" % ducfg["exec_prefix"])
-                    elif glob.glob("%s/libpython%d.%d*" % (ducfg["LIBDIR"], py_major, py_minor)):
-                        lib_dir_flag = quote("-L%s" % ducfg["LIBDIR"])
-                    else:
-                        sipconfig.inform("Qt Designer plugin disabled because Python library couldn't be found")
-                        lib_dir_flag = ''
-                        opts.designer_plugin = False
-
+                    lib_dir_flag = quote("-L%s" % sipcfg.py_lib_dir)
                     link = "%s -lpython%d.%d%s" % (lib_dir_flag, py_major, py_minor, abi)
                 else:
                     sipconfig.inform("Qt Designer plugin disabled because Python library is static")
@@ -1937,10 +1929,10 @@
         macros['CXXFLAGS'] += ' -mstackrealign'
 
     # Work out how Qt was built on MacOS.
-    if sys.platform == "darwin":
-        if os.access(os.path.join(qt_libdir, "QtCore.framework"), os.F_OK):
-            global qt_framework
-            qt_framework = 1
+#    if sys.platform == "darwin":
+#        if os.access(os.path.join(qt_libdir, "QtCore.framework"), os.F_OK):
+#            global qt_framework
+#            qt_framework = 1
 
     # Get the Makefile generator.
     generator = macros["MAKEFILE_GENERATOR"]
@@ -1991,8 +1983,8 @@
         else:
             exe_file = os.path.join("release", app + ".exe")
             make_target = " release"
-    elif sys.platform == "darwin":
-        exe_file = os.path.join(app + ".app", "Contents", "MacOS", app)
+#    elif sys.platform == "darwin":
+#        exe_file = os.path.join(app + ".app", "Contents", "MacOS", app)
     else:
         exe_file = os.path.join(".", app)
 
