diff -Nurd ruby-1.8.7-p174/config.guess ruby-1.8.7-p174-new/config.guess
--- ruby-1.8.7-p174/config.guess	2007-02-12 18:01:19.000000000 -0500
+++ ruby-1.8.7-p174-new/config.guess	2009-09-08 08:12:43.000000000 -0400
@@ -1200,7 +1200,7 @@
 	    *86) UNAME_PROCESSOR=i686 ;;
 	    powerpc) UNAME_PROCESSOR=powerpc ;;
 	esac
-	echo ${UNAME_PROCESSOR}-apple-darwin${UNAME_RELEASE}
+	echo ${UNAME_PROCESSOR}-apple-darwin
 	exit 0 ;;
     *:procnto*:*:* | *:QNX:[0123456789]*:*)
 	UNAME_PROCESSOR=`uname -p`
diff -Nurd ruby-1.8.7-p174/configure.in ruby-1.8.7-p174-new/configure.in
--- ruby-1.8.7-p174/configure.in	2009-01-19 04:25:38.000000000 -0500
+++ ruby-1.8.7-p174-new/configure.in	2009-09-08 08:12:43.000000000 -0400
@@ -547,12 +547,12 @@
 AC_TYPE_GETGROUPS
 AC_TYPE_SIGNAL
 case "${target_cpu}-${target_os}" in
-powerpc-darwin*)
-  AC_LIBSOURCES(alloca.c)
-  AC_SUBST([ALLOCA], [\${LIBOBJDIR}alloca.${ac_objext}])
-  AC_DEFINE(C_ALLOCA)
-  AC_DEFINE_UNQUOTED(alloca, alloca)
-  ;;
+dnl powerpc-darwin*)
+dnl   AC_LIBSOURCES(alloca.c)
+dnl   AC_SUBST([ALLOCA], [\${LIBOBJDIR}alloca.${ac_objext}])
+dnl   AC_DEFINE(C_ALLOCA)
+dnl   AC_DEFINE_UNQUOTED(alloca, alloca)
+dnl   ;;
 *)
   AC_FUNC_ALLOCA
   ;;
@@ -1213,7 +1213,7 @@
 	rhapsody*)	: ${LDSHARED='cc -dynamic -bundle -undefined suppress'}
 			: ${LDFLAGS=""}
 			rb_cv_dlopen=yes ;;
-	darwin*)	: ${LDSHARED='cc -dynamic -bundle -undefined suppress -flat_namespace'}
+	darwin*)	: ${LDSHARED='cc -dynamic -bundle -Wl,-undefined,dynamic_lookup -Wl,-multiply_defined,suppress'}
 			: ${LDFLAGS=""}
 			: ${LIBPATHENV=DYLD_LIBRARY_PATH}
 			rb_cv_dlopen=yes ;;
@@ -1449,7 +1449,7 @@
     ;;
 esac
 
-RUBY_SO_NAME='$(RUBY_INSTALL_NAME)'
+RUBY_SO_NAME=ruby
 LIBRUBY_LDSHARED=$LDSHARED
 LIBRUBY_DLDFLAGS=$DLDFLAGS
 LIBRUBY_SO='lib$(RUBY_SO_NAME).so.$(MAJOR).$(MINOR).$(TEENY)'
@@ -1526,9 +1526,9 @@
 	;;
     darwin*)
 	LIBRUBY_SO='lib$(RUBY_SO_NAME).$(MAJOR).$(MINOR).$(TEENY).dylib'
-	LIBRUBY_LDSHARED='cc -dynamiclib -undefined suppress -flat_namespace'
-	LIBRUBY_DLDFLAGS='-install_name $(libdir)/lib$(RUBY_SO_NAME).dylib -current_version $(MAJOR).$(MINOR).$(TEENY) -compatibility_version $(MAJOR).$(MINOR)'
-	LIBRUBY_ALIASES='lib$(RUBY_SO_NAME).$(MAJOR).$(MINOR).dylib lib$(RUBY_SO_NAME).dylib'
+	LIBRUBY_LDSHARED='cc -dynamiclib'
+	LIBRUBY_DLDFLAGS='-install_name $(libdir)/lib$(RUBY_SO_NAME).$(MAJOR).$(MINOR).dylib -current_version $(MAJOR).$(MINOR).$(TEENY) -compatibility_version $(MAJOR).$(MINOR)'
+	LIBRUBY_ALIASES='lib$(RUBY_SO_NAME).$(MAJOR).$(MINOR).dylib lib$(RUBY_SO_NAME).dylib libruby.2.dylib'
 	;;
     interix*)
 	LIBRUBYARG_SHARED='-L. -L${libdir} -l$(RUBY_SO_NAME)'
