diff -Nurd -x'*~' tcltk-8.5.10.orig/tcl8.5.10/generic/tclInt.h tcltk-8.5.10/tcl8.5.10/generic/tclInt.h
--- tcltk-8.5.10.orig/tcl8.5.10/generic/tclInt.h	2011-06-23 08:52:57.000000000 -0400
+++ tcltk-8.5.10/tcl8.5.10/generic/tclInt.h	2011-07-17 12:38:39.000000000 -0400
@@ -2873,7 +2873,7 @@
 MODULE_SCOPE int	TclClockOldscanObjCmd(
 			    ClientData clientData, Tcl_Interp *interp,
 			    int objc, Tcl_Obj *const objv[]);
-MODULE_SCOPE int	Tcl_CloseObjCmd(ClientData clientData,
+extern int	Tcl_CloseObjCmd(ClientData clientData,
 			    Tcl_Interp *interp, int objc,
 			    Tcl_Obj *const objv[]);
 MODULE_SCOPE int	Tcl_ConcatObjCmd(ClientData clientData,
@@ -3034,7 +3034,7 @@
 MODULE_SCOPE int	Tcl_RenameObjCmd(ClientData clientData,
 			    Tcl_Interp *interp, int objc,
 			    Tcl_Obj *const objv[]);
-MODULE_SCOPE int	Tcl_ReturnObjCmd(ClientData clientData,
+extern int	Tcl_ReturnObjCmd(ClientData clientData,
 			    Tcl_Interp *interp, int objc,
 			    Tcl_Obj *const objv[]);
 MODULE_SCOPE int	Tcl_ScanObjCmd(ClientData clientData,
diff -Nurd -x'*~' tcltk-8.5.10.orig/tcl8.5.10/generic/tclPort.h tcltk-8.5.10/tcl8.5.10/generic/tclPort.h
--- tcltk-8.5.10.orig/tcl8.5.10/generic/tclPort.h	2011-05-31 11:03:46.000000000 -0400
+++ tcltk-8.5.10/tcl8.5.10/generic/tclPort.h	2011-07-17 12:38:44.000000000 -0400
@@ -22,7 +22,7 @@
 #endif
 #include "tcl.h"
 #if !defined(_WIN32)
-#   include "tclUnixPort.h"
+#   include "../unix/tclUnixPort.h"
 #endif
 
 #if defined(__CYGWIN__)
diff -Nurd -x'*~' tcltk-8.5.10.orig/tcl8.5.10/unix/configure tcltk-8.5.10/tcl8.5.10/unix/configure
--- tcltk-8.5.10.orig/tcl8.5.10/unix/configure	2011-06-23 09:18:36.000000000 -0400
+++ tcltk-8.5.10/tcl8.5.10/unix/configure	2011-07-17 12:38:44.000000000 -0400
@@ -18871,7 +18871,7 @@
     else
         TCL_LIB_FLAG="-ltcl`echo ${TCL_VERSION} | tr -d .`"
     fi
-    TCL_BUILD_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TCL_LIB_FLAG}"
+    TCL_BUILD_LIB_SPEC="`pwd`/${TCL_LIB_FILE}"
     TCL_LIB_SPEC="-L${libdir} ${TCL_LIB_FLAG}"
 fi
 VERSION='${VERSION}'
@@ -18915,7 +18915,7 @@
     TCL_STUB_LIB_FLAG="-ltclstub`echo ${TCL_VERSION} | tr -d .`"
 fi
 
-TCL_BUILD_STUB_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TCL_STUB_LIB_FLAG}"
+TCL_BUILD_STUB_LIB_SPEC="`pwd`/${TCL_STUB_LIB_FILE}"
 TCL_STUB_LIB_SPEC="-L${TCL_STUB_LIB_DIR} ${TCL_STUB_LIB_FLAG}"
 TCL_BUILD_STUB_LIB_PATH="`pwd`/${TCL_STUB_LIB_FILE}"
 TCL_STUB_LIB_PATH="${TCL_STUB_LIB_DIR}/${TCL_STUB_LIB_FILE}"
@@ -19097,7 +19097,7 @@
 
 
 
-CFLAGS="${CFLAGS} ${CPPFLAGS}"; CPPFLAGS=""
+EXTRA_CC_SWITCHES="${EXTRA_CC_SWITCHES} ${CPPFLAGS}"; CPPFLAGS=""
 
 : ${CONFIG_STATUS=./config.status}
 ac_clean_files_save=$ac_clean_files
diff -Nurd -x'*~' tcltk-8.5.10.orig/tk8.5.10/unix/Makefile.in tcltk-8.5.10/tk8.5.10/unix/Makefile.in
--- tcltk-8.5.10.orig/tk8.5.10/unix/Makefile.in	2011-06-23 14:37:39.000000000 -0400
+++ tcltk-8.5.10/tk8.5.10/unix/Makefile.in	2011-11-03 02:24:06.000000000 -0400
@@ -154,7 +154,7 @@
 # Linker switch(es) to use to link with the X11 library archive (the
 # configure script will try to set this value automatically, but you
 # can override it).
-X11_LIB_SWITCHES	= @XLIBSW@ $(XFT_LIBS)
+X11_LIB_SWITCHES	= $(XFT_LIBS) @XLIBSW@
 
 
 # To turn off the security checks that disallow incoming sends when
@@ -324,18 +324,18 @@
 
 CC_SWITCHES_NO_STUBS = ${CFLAGS} ${CFLAGS_WARNING} ${SHLIB_CFLAGS} \
 -I${UNIX_DIR} -I${GENERIC_DIR} -I${BMAP_DIR} -I${TCL_GENERIC_DIR} \
--I${TCL_PLATFORM_DIR} ${@TK_WINDOWINGSYSTEM@_INCLUDES} ${AC_FLAGS} \
+-I${TCL_PLATFORM_DIR} ${AC_FLAGS} \
 ${PROTO_FLAGS} ${SECURITY_FLAGS} ${MEM_DEBUG_FLAGS} ${KEYSYM_FLAGS} \
-${NO_DEPRECATED_FLAGS} @EXTRA_CC_SWITCHES@
+${NO_DEPRECATED_FLAGS} @EXTRA_CC_SWITCHES@ ${@TK_WINDOWINGSYSTEM@_INCLUDES}
 
 CC_SWITCHES = $(CC_SWITCHES_NO_STUBS) ${TCL_STUB_FLAGS} 
 
 APP_CC_SWITCHES = $(CC_SWITCHES_NO_STUBS) @EXTRA_APP_CC_SWITCHES@
 
 DEPEND_SWITCHES = ${CFLAGS} -I${UNIX_DIR} -I${GENERIC_DIR} -I${BMAP_DIR} \
--I${TCL_GENERIC_DIR} -I${TCL_PLATFORM_DIR} ${@TK_WINDOWINGSYSTEM@_INCLUDES} \
+-I${TCL_GENERIC_DIR} -I${TCL_PLATFORM_DIR} \
 ${AC_FLAGS} ${PROTO_FLAGS} ${SECURITY_FLAGS} ${MEM_DEBUG_FLAGS} \
-${KEYSYM_FLAGS} @EXTRA_CC_SWITCHES@
+${KEYSYM_FLAGS} @EXTRA_CC_SWITCHES@ ${@TK_WINDOWINGSYSTEM@_INCLUDES}
 
 WISH_OBJS = tkAppInit.o
 
diff -Nurd -x'*~' tcltk-8.5.10.orig/tk8.5.10/unix/configure tcltk-8.5.10/tk8.5.10/unix/configure
--- tcltk-8.5.10.orig/tk8.5.10/unix/configure	2011-06-23 14:37:39.000000000 -0400
+++ tcltk-8.5.10/tk8.5.10/unix/configure	2011-11-03 02:23:35.000000000 -0400
@@ -10838,19 +10838,14 @@
 	echo "$as_me:$LINENO: result: $enable_xft" >&5
 echo "${ECHO_T}$enable_xft" >&6
     else
-	found_xft="yes"
-			XFT_CFLAGS=`xft-config --cflags 2>/dev/null` || found_xft="no"
-	XFT_LIBS=`xft-config --libs 2>/dev/null` || found_xft="no"
-	if test "$found_xft" = "no" ; then
 	    found_xft=yes
 	    XFT_CFLAGS=`pkg-config --cflags xft 2>/dev/null` || found_xft="no"
 	    XFT_LIBS=`pkg-config --libs xft 2>/dev/null` || found_xft="no"
-	fi
 	echo "$as_me:$LINENO: result: $found_xft" >&5
 echo "${ECHO_T}$found_xft" >&6
 		if test "$found_xft" = "yes" ; then
 	    tk_oldCFlags=$CFLAGS
-	    CFLAGS="$CFLAGS $XINCLUDES $XFT_CFLAGS"
+	    CFLAGS="$CFLAGS $XFT_CFLAGS $XINCLUDES"
 	    tk_oldLibs=$LIBS
 	    LIBS="$tk_oldLIBS $XFT_LIBS $XLIBSW"
 	    echo "$as_me:$LINENO: checking for X11/Xft/Xft.h" >&5
@@ -10915,7 +10910,7 @@
 	fi
 		if test "$found_xft" = "yes" ; then
 	    tk_oldCFlags=$CFLAGS
-	    CFLAGS="$CFLAGS $XINCLUDES $XFT_CFLAGS"
+	    CFLAGS="$CFLAGS $XFT_CFLAGS $XINCLUDES"
 	    tk_oldLibs=$LIBS
 	    LIBS="$tk_oldLIBS $XFT_LIBS $XLIBSW"
 
@@ -11231,7 +11226,7 @@
     else
         TK_LIB_FLAG="-ltk`echo ${TK_VERSION} | tr -d .`"
     fi
-    TK_BUILD_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TK_LIB_FLAG}"
+    TK_BUILD_LIB_SPEC="`pwd`/${TK_LIB_FILE}"
     TK_LIB_SPEC="-L${libdir} ${TK_LIB_FLAG}"
 fi
 
@@ -11265,7 +11260,7 @@
     TK_STUB_LIB_FLAG="-ltkstub`echo ${TK_VERSION} | tr -d .`"
 fi
 
-TK_BUILD_STUB_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TK_STUB_LIB_FLAG}"
+TK_BUILD_STUB_LIB_SPEC="`pwd`/${TK_STUB_LIB_FILE}"
 TK_STUB_LIB_SPEC="-L${TK_STUB_LIB_DIR} ${TK_STUB_LIB_FLAG}"
 TK_BUILD_STUB_LIB_PATH="`pwd`/${TK_STUB_LIB_FILE}"
 TK_STUB_LIB_PATH="${TK_STUB_LIB_DIR}/${TK_STUB_LIB_FILE}"
@@ -11453,7 +11448,7 @@
 LTLIBOBJS=$ac_ltlibobjs
 
 
-CFLAGS="${CFLAGS} ${CPPFLAGS}"; CPPFLAGS=""
+EXTRA_CC_SWITCHES="${EXTRA_CC_SWITCHES} ${CPPFLAGS}"; CPPFLAGS=""
 
 : ${CONFIG_STATUS=./config.status}
 ac_clean_files_save=$ac_clean_files
diff -Nurd -x'*~' tcltk-8.5.10.orig/tk8.5.10/unix/tkConfig.sh.in tcltk-8.5.10/tk8.5.10/unix/tkConfig.sh.in
--- tcltk-8.5.10.orig/tk8.5.10/unix/tkConfig.sh.in	2011-06-23 14:37:39.000000000 -0400
+++ tcltk-8.5.10/tk8.5.10/unix/tkConfig.sh.in	2011-11-03 02:24:40.000000000 -0400
@@ -32,7 +32,7 @@
 TK_LIB_FILE='@TK_LIB_FILE@'
 
 # Additional libraries to use when linking Tk.
-TK_LIBS='@XLIBSW@ @XFT_LIBS@ @LIBS@ @TCL_LIBS@'
+TK_LIBS='@XFT_LIBS@ @XLIBSW@ @LIBS@ @TCL_LIBS@'
 
 # Top-level directory in which Tk's platform-independent files are
 # installed.
