--- F2PY-2.32.225-1419/setup.py	Thu Dec  5 03:55:34 2002
+++ F2PY-2.32.225-1419/setup.py	Mon Dec 16 10:23:05 2002
@@ -37,29 +37,6 @@
     required_scipy_distutils_version = '0.2.0_alpha_2.229'
     _path = os.path.abspath(os.getcwd())
     fl = 0
-    try:
-        os.chdir('src')
-        try:
-            from scipy_distutils.scipy_distutils_version import \
-                 scipy_distutils_version
-            print 'Found scipy_distutils Version',scipy_distutils_version
-            if scipy_distutils_version<required_scipy_distutils_version:
-                print 'This f2py version needs scipy_distutils %s or newer'\
-                      % (required_scipy_distutils_version)
-                while 1:
-                    r = raw_input('-> Let f2py to upgrade scipy_distutils? [y]/n: ')
-                    if not r or r[0] in 'yY':
-                        fl = 1
-                        print 'Adding scipy_distutils to packages'
-                        break
-                    elif r[0] in 'nN':
-                        break
-        except ImportError:
-            print sys.exc_value
-            print 'Adding scipy_distutils to packages'
-            fl = 1
-    finally:
-        os.chdir(_path)
     return fl
 
 def f2py_bat():
