diff -ruN perl/Configure perl-patched/Configure
--- perl/Configure	Wed Apr  9 17:30:28 2003
+++ perl-patched/Configure	Tue May 13 17:39:07 2003
@@ -3380,6 +3380,18 @@
 test "X$gfpthkeep" != Xy && gfpth=""
 EOSC
 
+# gcc 3.1 complains about adding -Idirectories that it already knows about,
+# so we will take those off from locincpth.
+case "$gccversion" in
+3*)
+    echo "main(){}">try.c
+    for incdir in `$cc -v -c try.c 2>&1 | \
+       sed '1,/^#include <\.\.\.>/d;/^End of search list/,$d;s/^ //'` ; do
+       locincpth=`echo $locincpth | sed s!$incdir!!`
+    done
+    $rm -f try try.*
+esac
+
 : What should the include directory be ?
 echo " "
 $echo $n "Hmm...  $c"
diff -ruN perl/hints/darwin.sh perl-patched/hints/darwin.sh
--- perl/hints/darwin.sh	Wed Apr  9 17:30:58 2003
+++ perl-patched/hints/darwin.sh	Tue May 13 17:37:04 2003
@@ -8,17 +8,17 @@
 ##
 
 # BSD paths
-prefix='/usr';
-siteprefix='/usr/local';
-vendorprefix='/usr/local'; usevendorprefix='define';
+# prefix='/usr';
+# siteprefix='/usr/local';
+# vendorprefix='/usr/local'; usevendorprefix='define';
 
 # 4BSD uses /usr/share/man, not /usr/man.
 # Don't put man pages in /usr/lib; that's goofy.
-man1dir='/usr/share/man/man1';
-man3dir='/usr/share/man/man3';
+# man1dir='/usr/share/man/man1';
+# man3dir='/usr/share/man/man3';
 
 # Where to put modules.
-privlib='/System/Library/Perl';
+# privlib='/System/Library/Perl';
 sitelib='/Library/Perl';
 vendorlib='/Network/Library/Perl';
 
