diff -ruN SmallEiffel/loadpath.UNIX SmallEiffel-patched/loadpath.UNIX
--- SmallEiffel/loadpath.UNIX	Thu Jan  1 01:00:00 1970
+++ SmallEiffel-patched/loadpath.UNIX	Tue Jul 30 23:38:55 2002
@@ -0,0 +1,16 @@
+./
+@LIBPREFIX@/kernel/
+@LIBPREFIX@/base/
+@LIBPREFIX@/io/
+@LIBPREFIX@/iterator/
+@LIBPREFIX@/random/
+@LIBPREFIX@/number/
+@LIBPREFIX@/sorting/
+@LIBPREFIX@/time/
+@LIBPREFIX@/tools/
+@LIBPREFIX@/tools/expression/
+@LIBPREFIX@/tools/instruction/
+@LIBPREFIX@/tools/e_type/
+@LIBPREFIX@/tools/feature/
+@LIBPREFIX@/tools/run_feature/
+@LIBPREFIX@/tools/misc/
diff -ruN SmallEiffel/smalleiffel.csh SmallEiffel-patched/smalleiffel.csh
--- SmallEiffel/smalleiffel.csh	Thu Jan  1 01:00:00 1970
+++ SmallEiffel-patched/smalleiffel.csh	Tue Jul 30 23:34:23 2002
@@ -0,0 +1 @@
+setenv SmallEiffel @PREFIX@/sys/system.se
diff -ruN SmallEiffel/smalleiffel.sh SmallEiffel-patched/smalleiffel.sh
--- SmallEiffel/smalleiffel.sh	Thu Jan  1 01:00:00 1970
+++ SmallEiffel-patched/smalleiffel.sh	Tue Jul 30 23:34:50 2002
@@ -0,0 +1,2 @@
+SmallEiffel=@PREFIX@/sys/system.se
+export SmallEiffel
--- SmallEiffel/sys/gc	Tue May  7 17:43:33 2002
+++ SmallEiffel-patched/sys/gc	Wed Jul 31 00:15:12 2002
@@ -1 +1 @@
-NONE
+MacintoshPPC.c
--- SmallEiffel/tools/system_tools.e	Tue May  7 17:43:15 2002
+++ SmallEiffel-patched/tools/system_tools.e	Thu Aug  1 01:02:50 2002
@@ -284,7 +284,7 @@
          if system_name = open_vms_system then
             command.append(once "mcr ");
          end;
-         command.append(bin_directory);
+--         command.append(bin_directory);
          command.append(command_name);
          command.append(x_suffix);
       end;
--- SmallEiffel/misc/GC.SH	Tue May  7 17:41:44 2002
+++ SmallEiffel-patched/misc/GC.SH	Sat Aug  3 16:49:16 2002
@@ -52,6 +52,9 @@
 elif [ $un = QNX ]
  then
   gc=qnx.c
+elif [ $un = Darwin ]
+ then
+  gc=MacintoshPPC.c
 else
   echo GC is not supported for this architecture.
   echo Have a look in the SmallEiffel FAQ to try to fix this.
