--- SWIG-1.3.11/configure.orig	Sun Mar 10 08:25:47 2002
+++ SWIG-1.3.11/configure	Sun Mar 10 08:32:52 2002
@@ -2721,7 +2721,8 @@
 echo "configure:2722: checking for Guile library" >&5
   dirs="$GUILELIB `guile-config info libdir`"
   for i in $dirs ; do
-  	if test -r $i/libguile.so; then
+# 	if test -r $i/libguile.so; then
+  	if test -r $i/libguile.dylib; then
   	    echo "$ac_t""$i" 1>&6
   	    GUILELIB="$i"
   	    break
