diff -ru ucblogo-5.3-orig/emacs/makefile ucblogo-5.3/emacs/makefile
--- ucblogo-5.3-orig/emacs/makefile	Sat Apr 12 00:44:55 2003
+++ ucblogo-5.3/emacs/makefile	Sat Apr 12 00:55:50 2003
@@ -1,7 +1,7 @@
 BUILDIR	      = `pwd`
-INFODIR	      = $(prefix)/info
+INFODIR	      = $(prefix)/share/info
 EMACSDIR      = $(LIBLOC)/emacs
-DOCSDIR	      = $(LIBLOC)/docs
+DOCSDIR	      = $(prefix)/share/doc/ucblogo
 HTMLDIR	      = $(DOCSDIR)/html
 LOADPATH      = $(subst X,$(EMACSDIR),(setq load-path (cons "\"X\"" load-path)))
 LOGOBIN	      = $(subst X,$(BINDIR),(setq logo-binary-name \"X/logo\"))
Only in ucblogo-5.3/emacs: makefile~
diff -ru ucblogo-5.3-orig/makefile.in ucblogo-5.3/makefile.in
--- ucblogo-5.3-orig/makefile.in	Sat Apr 12 00:44:55 2003
+++ ucblogo-5.3/makefile.in	Sat Apr 12 00:46:14 2003
@@ -4,7 +4,7 @@
 LIBS  = @X_PRE_LIBS@ @X_LIBS@ @LIBS@ @X_EXTRA_LIBS@
 prefix = @prefix@
 BINDIR        = $(prefix)/bin
-LIBLOC        = $(prefix)/lib/logo
+LIBLOC        = $(prefix)/lib/ucblogo
 
 # LIBLOC      = `pwd`
 
@@ -54,12 +54,12 @@
 
 clean:
 	rm -f *.o libloc.c
-	cd emacs; gmake clean
+	cd emacs; make clean
 
 ship:
 	rm -f config.h config.cache config.log config.status
 	rm -f makefile makehelp logo *.o libloc.c
-	cd emacs; gmake ship
+	cd emacs; make ship
 
 install: all
 	for d in $(BINDIR) $(LIBLOC) $(LIBLOC)/logolib $(LIBLOC)/helpfiles $(LIBLOC)/csls; do [ -d $$d ] || mkdir -p $$d || exit 1; done
@@ -67,12 +67,12 @@
 	cp -f logolib/* $(LIBLOC)/logolib/.
 	cp -f helpfiles/* $(LIBLOC)/helpfiles/.
 	cp -f csls/* $(LIBLOC)/csls/.
-	(cd emacs; prefix=$(prefix) LIBLOC=$(LIBLOC) BINDIR=$(BINDIR) gmake install)
-#	prefix=$(prefix); LIBLOC=$(LIBLOC); BINDIR=$(BINDIR); export prefix LIBLOC BINDIR; cd emacs; gmake install
+	(cd emacs; prefix=$(prefix) LIBLOC=$(LIBLOC) BINDIR=$(BINDIR) make install)
+#	prefix=$(prefix); LIBLOC=$(LIBLOC); BINDIR=$(BINDIR); export prefix LIBLOC BINDIR; cd emacs; make install
 
 logo-mode: 
-	(cd emacs; prefix=$(prefix) LIBLOC=$(LIBLOC) BINDIR=$(BINDIR) gmake)
-#	@prefix=$(prefix); LIBLOC=$(LIBLOC); BINDIR=$(BINDIR); export prefix LIBLOC BINDIR; cd emacs; gmake
+	(cd emacs; prefix=$(prefix) LIBLOC=$(LIBLOC) BINDIR=$(BINDIR) make)
+#	@prefix=$(prefix); LIBLOC=$(LIBLOC); BINDIR=$(BINDIR); export prefix LIBLOC BINDIR; cd emacs; make
 
 docs:
-	cd emacs; gmake docs
+	cd emacs; make docs
Only in ucblogo-5.3: makefile.in~
