diff -ru Yap-4.3.22.orig/Makefile.in Yap-4.3.22/Makefile.in
--- Yap-4.3.22.orig/Makefile.in	Mon Aug 12 21:54:30 2002
+++ Yap-4.3.22/Makefile.in	Mon Aug 12 22:19:08 2002
@@ -576,16 +576,16 @@
 	$(LD) $(EXP_FILE) --dll -o $(DLLNAME) $(LIB_OBJECTS) $(DLL_LIBS) -e $(ENTRY_FUNCTION)
 
 install_info:
-	make info; \
-	$(INSTALL) yap.info* $(DESTDIR)$(INFODIR); \
+	-mkdir -p $(DESTDIR)$(INFODIR)
+	make info; 
+	$(INSTALL) yap.info* $(DESTDIR)$(INFODIR); 
 	if test -e $(DESTDIR)$(INFODIR)/emacs.bz2; then \
 	  bzip2 --quiet --force $(DESTDIR)$(INFODIR)/yap.info*; \
 	else \
 	  if test -e $(DESTDIR)$(INFODIR)/emacs.gz; then \
 	  gzip --quiet --force $(DESTDIR)$(INFODIR)/yap.info*; \
 	  fi \
-	fi; \
-	$(INSTALL_INFO) --quiet --section "Programming Languages" yap.info $(DESTDIR)$(INFODIR)/dir; \
+	fi; 
 	cd $(srcdir)/LGPL/pillow/doc ; \
 	$(INSTALL) pillow_doc.info $(DESTDIR)$(INFODIR); \
 	if test -e $(DESTDIR)$(INFODIR)/emacs.bz2; then \
@@ -595,7 +595,6 @@
 	  gzip --quiet --force $(DESTDIR)$(INFODIR)/pillow_doc.info; \
 	  fi \
 	fi; \
-	$(INSTALL_INFO) --quiet --section "Programming Languages" --entry="`cat pillow_doc.infoindex`" pillow_doc.info $(INFODIR)/dir
 
 info: yap.info
 
Only in Yap-4.3.22: Makefile.in~
