diff -Naur --exclude='*~' color-theme-6.6.0/fink/50color-theme.el color-theme-6.6.0.fink/fink/50color-theme.el
--- color-theme-6.6.0/fink/50color-theme.el	1969-12-31 18:00:00.000000000 -0600
+++ color-theme-6.6.0.fink/fink/50color-theme.el	2007-12-26 15:50:43.000000000 -0600
@@ -0,0 +1 @@
+(require 'color-theme-autoloads "color-theme-autoloads")
diff -Naur --exclude='*~' color-theme-6.6.0/fink/emacsen-install color-theme-6.6.0.fink/fink/emacsen-install
--- color-theme-6.6.0/fink/emacsen-install	1969-12-31 18:00:00.000000000 -0600
+++ color-theme-6.6.0.fink/fink/emacsen-install	2007-12-26 15:41:25.000000000 -0600
@@ -0,0 +1,40 @@
+#!/bin/bash -e
+#
+# install file for the fink color-theme emacs package.
+
+set -o posix
+
+FLAVOR=${1}
+
+echo >&2 "install/color-theme: Handling install of emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs ]
+then
+    # Nothing to do -- we're done now
+    exit 0
+fi
+
+echo >&2 -n "install/color-theme: Byte-compiling for ${FLAVOR}..."
+mkdir -p @PREFIX@/share/${FLAVOR}/site-lisp/color-theme;
+for i in color-theme-autoloads.el color-theme.el color-theme-autoloads.in Makefile Makefile.defs; do
+    ln -s -f @PREFIX@/share/emacs/site-lisp/color-theme/${i} @PREFIX@/share/${FLAVOR}/site-lisp/color-theme/${i};
+done
+
+mkdir -p @PREFIX@/share/${FLAVOR}/site-lisp/color-theme/themes;
+for i in color-theme-example.el color-theme-library.el; do
+    ln -s -f @PREFIX@/share/emacs/site-lisp/color-theme/themes/${i} @PREFIX@/share/${FLAVOR}/site-lisp/color-theme/themes/${i};
+done
+
+cd @PREFIX@/share/${FLAVOR}/site-lisp/color-theme;
+
+(make PREFIX=@PREFIX@ ELISPDIR=@PREFIX@/share/${FLAVOR}/site-lisp/color-theme EMACS=${FLAVOR} lisp --keep-going 2>&1) | gzip -9qf > @PREFIX@/share/doc/color-theme/CompilationLog-${FLAVOR}.gz
+
+rm -f @PREFIX@/share/${FLAVOR}/site-lisp/color-theme/Makefile;
+rm -f @PREFIX@/share/${FLAVOR}/site-lisp/color-theme/Makefile.defs;
+rm -f @PREFIX@/share/${FLAVOR}/site-lisp/color-theme/color-theme-autoloads.in
+rm -f @PREFIX@/share/${FLAVOR}/site-lisp/color-theme/color-theme-autoloads.el~
+
+echo >&2 "done."
+echo >&2 "install/color-theme: Compilation log saved in @PREFIX@/share/doc/color-theme/CompilationLog-${FLAVOR}.gz."
+
+exit 0
diff -Naur --exclude='*~' color-theme-6.6.0/fink/emacsen-remove color-theme-6.6.0.fink/fink/emacsen-remove
--- color-theme-6.6.0/fink/emacsen-remove	1969-12-31 18:00:00.000000000 -0600
+++ color-theme-6.6.0.fink/fink/emacsen-remove	2007-12-26 15:41:29.000000000 -0600
@@ -0,0 +1,21 @@
+#!/bin/bash -e
+#
+# This is the Fink version of the color-theme emacs package.
+set -o posix
+
+FLAVOR=${1}
+
+echo "remove/color-theme: Handling removal for emacsen flavor ${FLAVOR}"
+
+if [ ${FLAVOR} == emacs20 ]
+then
+    echo "remove/color-theme: Skipping unsupported flavor ${FLAVOR}"
+    exit 0
+fi
+
+echo >&2 -n "remove/color-theme: Purging compilation log and byte-compiled files for ${FLAVOR}..."
+rm -f @PREFIX@/share/doc/color-theme/CompilationLog-${FLAVOR}.gz
+rm -Rf @PREFIX@/share/${FLAVOR}/site-lisp/color-theme
+echo >&2 "done."
+    
+exit 0
