diff -u -Naur crypt++el-2.9.1/crypt++el.install crypt++el-2.9.1.fink/crypt++el.install
--- crypt++el-2.9.1/crypt++el.install	Wed Dec 12 13:46:09 2001
+++ crypt++el-2.9.1.fink/crypt++el.install	Tue Jun 10 15:13:23 2003
@@ -1,5 +1,5 @@
 #!/bin/sh
-# /usr/lib/emacsen-common/packages/install/crypt++el
+# @PREFIX@/lib/emacsen-common/packages/install/crypt++el
 set -e
 
 FLAVOR=$1
@@ -7,31 +7,24 @@
 
 byte_compile_options="-batch -f batch-byte-compile"
 el_files="crypt++.el" 
-el_dir=/usr/share/emacs/site-lisp/
-elc_dir=/usr/share/${FLAVOR}/site-lisp/
-el_path_list=`echo ${el_files} | perl -pe "s|^|${el_dir}|o"`
-elc_path_list=`echo ${el_files} | perl -pe "s|^|${elc_dir}|o"`
-#etc_dir=/etc/${FLAVOR}/site-start.d/
+el_dir=@PREFIX@/share/emacs/site-lisp/
+elc_dir=@PREFIX@/share/${FLAVOR}/site-lisp/
+el_path_list=`for i in ${el_files}; do echo $i | perl -pe "s|^|${el_dir}|"; done`
+elc_path_list=`for i in ${el_files}; do echo $i | perl -pe "s|^|${elc_dir}|"; done`
+#etc_dir=@PREFIX@/etc/${FLAVOR}/site-start.d/
 
 if [ ${FLAVOR} != emacs ]
 then
   echo install/crypt++el: byte-compiling for ${FLAVOR}
 
-#  ln -sf ${el_dir}/crypt++-init.el ${etc_dir}/50crypt++.el
-
   # Copy the temp .el files
   cp ${el_path_list} ${elc_dir}
 
   # Byte compile them
-#  cat <<-EOF >${elc_dir}/load-path-hack.el
-#(setq load-path (cons nil load-path))
-#EOF
   (cd ${elc_dir} && ${FLAVOR} ${byte_compile_options} ${elc_path_list} 2>/dev/null)
 
   # remove the redundant .el files
-  # presumes that any .el files in the <flavor> dir are trash.
-#  rm ${elc_dir}/load-path-hack.el
-  rm ${elc_dir}/*.el
+  rm ${elc_path_list}
 fi
 exit 0;
 
diff -u -Naur crypt++el-2.9.1/crypt++el.remove crypt++el-2.9.1.fink/crypt++el.remove
--- crypt++el-2.9.1/crypt++el.remove	Wed Dec 12 13:46:19 2001
+++ crypt++el-2.9.1.fink/crypt++el.remove	Tue Jun 10 15:11:29 2003
@@ -1,5 +1,5 @@
 #!/bin/sh
-# /usr/lib/emacsen-common/packages/remove/crypt++el
+# @PREFIX@/lib/emacsen-common/packages/remove/crypt++el
 set -e
 
 FLAVOR=$1
@@ -8,6 +8,6 @@
 if [ ${FLAVOR} != emacs ]
 then
   echo emacsen-common: purging byte-compiled files for ${FLAVOR}
-  rm -f /usr/share/${FLAVOR}/site-lisp/crypt++.elc
+  rm -f @PREFIX@/share/${FLAVOR}/site-lisp/crypt++.elc
 fi
 exit 0;
