diff -Nurd -x'*~' cssed-quicksearch-plugin-0.2.orig/configure cssed-quicksearch-plugin-0.2/configure
--- cssed-quicksearch-plugin-0.2.orig/configure	2005-09-18 06:39:29.000000000 -0400
+++ cssed-quicksearch-plugin-0.2/configure	2011-08-01 22:19:51.000000000 -0400
@@ -5151,7 +5151,7 @@
 i[3456]86-*-linux-gnu*)
 	OSTYPE="LINUX"
 	 ;;
-powerpc-apple-darwin*)
+*-apple-darwin*)
 	OSTYPE="DARWIN"
 	;;
 esac
diff -Nurd -x'*~' cssed-quicksearch-plugin-0.2.orig/src/Makefile.in cssed-quicksearch-plugin-0.2/src/Makefile.in
--- cssed-quicksearch-plugin-0.2.orig/src/Makefile.in	2005-09-18 06:39:25.000000000 -0400
+++ cssed-quicksearch-plugin-0.2/src/Makefile.in	2011-08-01 22:23:32.000000000 -0400
@@ -181,9 +181,9 @@
 @ISDARWIN_FALSE@quicksearch_so_SOURCES = $(SRCS)
 @ISDARWIN_TRUE@quicksearch_so_SOURCES = $(SRCS)
 @ISDARWIN_FALSE@quicksearch_so_LDFLAGS = -shared -fPICS
-@ISDARWIN_TRUE@quicksearch_so_LDFLAGS = -bundle -flat_namespace -undefined suppress
+@ISDARWIN_TRUE@quicksearch_so_LDFLAGS = -bundle -undefined dynamic_lookup
 @ISDARWIN_FALSE@quicksearch_so_LDADD = @PACKAGE_LIBS@
-@ISDARWIN_TRUE@quicksearch_so_LDADD = @PACKAGE_LIBS@
+@ISDARWIN_TRUE@quicksearch_so_LDADD = @PACKAGE_LIBS@ @INTLLIBS@
 @ISDARWIN_TRUE@quicksearch_so_CFLAGS = -fno-common
 INCLUDES = \
 	-DPACKAGE_LOCALE_DIR=\""$(prefix)/share/locale"\" \
diff -Nurd -x'*~' cssed-quicksearch-plugin-0.2.orig/src/quicksearch.c cssed-quicksearch-plugin-0.2/src/quicksearch.c
--- cssed-quicksearch-plugin-0.2.orig/src/quicksearch.c	2005-01-05 08:49:52.000000000 -0500
+++ cssed-quicksearch-plugin-0.2/src/quicksearch.c	2011-08-01 22:20:08.000000000 -0400
@@ -19,6 +19,7 @@
 #  include <config.h>
 #endif
 
+#include <string.h>
 #include <gtk/gtk.h>
 #include <gmodule.h>
 #include <plugin.h>
