diff -Nurd -x'*~' ghex-2.24.0.orig/src/Makefile.in ghex-2.24.0/src/Makefile.in
--- ghex-2.24.0.orig/src/Makefile.in	2009-02-19 00:16:58.000000000 -0500
+++ ghex-2.24.0/src/Makefile.in	2010-02-25 03:14:10.000000000 -0500
@@ -66,7 +66,7 @@
 libgtkhex_la_OBJECTS = $(am_libgtkhex_la_OBJECTS)
 libgtkhex_la_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(libgtkhex_la_CFLAGS) \
-	$(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+	$(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@ -no-undefined $(INTLLIBS)
 binPROGRAMS_INSTALL = $(INSTALL_PROGRAM)
 PROGRAMS = $(bin_PROGRAMS)
 am_ghex2_OBJECTS = hex-document-ui.$(OBJEXT) preferences.$(OBJEXT) \
diff -Nurd -x'*~' ghex-2.24.0.orig/src/ghex-window.c ghex-2.24.0/src/ghex-window.c
--- ghex-2.24.0.orig/src/ghex-window.c	2009-02-06 23:05:35.000000000 -0500
+++ ghex-2.24.0/src/ghex-window.c	2010-02-25 02:57:02.000000000 -0500
@@ -20,7 +20,7 @@
 #include <bonobo.h>
 #include <bonobo/bonobo-ui-main.h>
 
-#include <libgnomevfs/gnome-vfs-ops.h>
+#include <gio/gio.h>
 
 #include <math.h>
 #include <ctype.h>
@@ -1183,15 +1183,15 @@
 gboolean
 ghex_window_uri_exists (const gchar* text_uri)
 {
-	GnomeVFSURI *uri;
+	GFile *uri;
 	gboolean res;
 
 	g_return_val_if_fail (text_uri != NULL, FALSE);
-	uri = gnome_vfs_uri_new (text_uri);
+	uri = g_file_new_for_uri (text_uri);
 	g_return_val_if_fail (uri != NULL, FALSE);
-	res = gnome_vfs_uri_exists (uri);
+	res = g_file_query_exists (uri, NULL);
 
-	gnome_vfs_uri_unref (uri);
+	g_object_unref (uri);
 	return res;
 }
 
