diff -u joe-3.7/configure.ac joe-3.7-p1/configure.ac
--- joe-3.7/configure.ac	2009-08-03 14:45:11.000000000 +0000
+++ joe-3.7/configure.ac-p1	2009-08-03 15:24:10.000000000 +0000
@@ -39,7 +39,7 @@
 	curses=$enableval, curses=yes)
 
 if test "$curses" = "yes"; then
-	search_libs="ncurses curses"
+	search_libs="ncursesw ncurses curses"
 fi
 
 AC_ARG_ENABLE(termcap,
@@ -72,7 +72,17 @@
 AC_CHECK_HEADERS([sys/ioctl.h sys/param.h sys/time.h unistd.h utime.h])
 AC_CHECK_HEADERS([sys/dirent.h time.h pwd.h paths.h pty.h libutil.h])
 AC_CHECK_HEADERS([sys/types.h sys/stat.h sys/wait.h limits.h signal.h])
-AC_CHECK_HEADERS([curses.h term.h utmp.h sys/utime.h])
+AC_CHECK_HEADERS([ncursesw/curses.h ncursesw/term.h])
+if test "$ac_cv_header_ncursesw_curses_h" = "yes" -a "$ac_cv_header_ncursesw_term_h" = "yes" ; then
+	echo "Using ncurses wit wchar support!"
+else
+	AC_CHECK_HEADERS([curses.h term.h])
+	if test "$ac_cv_header_curses_h" = "yes" -a "$ac_cv_header_term_h" = "yes" ; then
+		echo "Using ncurses without wchat support!"
+	fi
+fi
+
+AC_CHECK_HEADERS([utmp.h sys/utime.h])
 
 AC_CHECK_HEADERS([selinux/selinux.h selinux/context.h])
 if test "$ac_cv_header_selinux_selinux_h" = "yes" -a "$ac_cv_header_selinux_context_h" = "yes" ; then
