diff -Naur speedbar-0.14beta4.orig/emacsen-install speedbar-0.14beta4/emacsen-install
--- speedbar-0.14beta4.orig/emacsen-install	Thu Jan  1 10:00:00 1970
+++ speedbar-0.14beta4/emacsen-install	Wed Sep 25 02:24:08 2002
@@ -0,0 +1,52 @@
+#! /bin/sh -e
+
+FLAVOR=$1
+PACKAGE=speedbar
+
+echo install/${PACKAGE}: Handling install of emacsen flavor ${FLAVOR}
+
+if [ ${FLAVOR} == emacs ]; then exit 0; fi
+
+FLAVORTEST=`echo $FLAVOR | cut -c-6`
+if [ ${FLAVORTEST} = xemacs ] ; then
+    SITEFLAG="-no-site-file"
+else
+    SITEFLAG="--no-site-file"
+fi
+FLAGS="${SITEFLAG} -q -batch -l path.el -f batch-byte-compile"
+
+ELDIR=@PREFIX@/share/emacs/site-lisp/${PACKAGE}
+ELCDIR=@PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}
+
+install -m 755 -d ${ELCDIR}
+cd ${ELDIR}
+
+FILES=`echo *.el`
+cp ${FILES} ${ELCDIR}
+cd ${ELCDIR}
+
+# emacs-w3 does not install contrib/w3-imenu which is needed
+# for the optional sb-w3, so don't try to compile it.
+rm sb-w3.el
+
+FILES=`echo *.el`
+
+cat << EOF > path.el
+(setq load-path (cons "." load-path) byte-compile-warnings nil)
+EOF
+
+rm -f *.elc
+${FLAVOR} ${FLAGS} ${FILES}
+
+rm -f *.el
+
+ln -sf @PREFIX@/share/pixmaps/speedbar/*.xpm .
+
+#Install a site-start file that ensures that this version of speedbar
+#is found in the load-path before the old one that comes with emacs
+
+cat << EOF > @PREFIX@/etc/${FLAVOR}/site-start.d/50speedbar.el
+(setq load-path (cons "@PREFIX@/share/${FLAVOR}/site-lisp/speedbar" load-path))
+EOF
+
+exit 0
diff -Naur speedbar-0.14beta4.orig/emacsen-remove speedbar-0.14beta4/emacsen-remove
--- speedbar-0.14beta4.orig/emacsen-remove	Thu Jan  1 10:00:00 1970
+++ speedbar-0.14beta4/emacsen-remove	Wed Sep 25 02:25:09 2002
@@ -0,0 +1,17 @@
+#!/bin/sh -e
+
+FLAVOR=$1
+
+echo remove/speedbar: Handling removal of emacsen flavor ${FLAVOR}
+
+if [ ${FLAVOR} != emacs ]
+then
+  echo remove/speedbar: purging byte-compiled files for ${FLAVOR}
+
+  rm -rf @PREFIX@/share/${FLAVOR}/site-lisp/speedbar
+  rm -f  @PREFIX@/etc/${FLAVOR}/site-start.d/??speedbar.el
+
+fi
+
+exit 0;
+

