--- konwert-1.8.orig/Makefile
+++ konwert-1.8/Makefile
@@ -46,9 +46,9 @@
 
 # Dokumentacja:
 
-mydocdir = $(prefix)/doc/konwert-$(version)
+mydocdir = $(sharedir)/doc/konwert
 srcdocdir = doc
-mandir = $(prefix)/man
+mandir = $(sharedir)/man
 srcmandir = man
 
 # Wersje z poprawionymi ∂cieøkami w #!, gotowe do zainstalowania:
@@ -181,6 +181,9 @@
 
 ###############################################################################
 
+$(srcbindir)/filterm: $(srcbindir)/filterm.cc
+	$(CXX) $(CXXFLAGS) $< -o $@
+
 $(installbindir)/konwert: $(srcbindir)/konwert
 	sed 's|/usr/share/konwert/filters|$(filtersdir)|g' $< >$@
 	chmod +x $@
@@ -335,16 +335,16 @@
 	$(INSTALL) -m755 -s $(installbindir)/trs     $(bindir)
 	$(INSTALL) -m755    $(installbindir)/konwert $(bindir)
 	$(INSTALL) -m755 -s $(installbindir)/filterm $(bindir)
-	cp -dRf $(installfiltersdir)/* $(filtersdir)
+	/bin/cp -Rf $(installfiltersdir)/* $(filtersdir)
 	$(perl) $(srcdeveldir)/mkaliases $(srcdeveldir)/aliases \
 		$(installfiltersdir) $(filtersdir)
-	cp -dRf $(installauxdir)/* $(auxdir)
+	/bin/cp -Rf $(installauxdir)/* $(auxdir)
 	$(perl) $(srcdeveldir)/mkaliases $(srcdeveldir)/aliases \
 		$(installauxdir)/charsets $(auxdir)/charsets
-	cp -dRf $(installlibauxdir)/* $(libauxdir)
-	cp -dRf $(installdeveldir)/* $(develdir)
-	cp -dRf $(installdocdir)/* $(mydocdir)
-	cp -dRf $(installmandir)/* $(mandir)
+	/bin/cp -Rf $(installlibauxdir)/* $(libauxdir)
+	/bin/cp -Rf $(installdeveldir)/* $(develdir)
+	/bin/cp -Rf $(installdocdir)/* $(mydocdir)
+	/bin/cp -Rf $(installmandir)/* $(mandir)
 	if [ -z "$(nofixmanconfig)" ]; then $(srcauxdir)/fixmanconfig; fi
 
 ###############################################################################
--- konwert-1.8.orig/doc/en/filters
+++ konwert-1.8/doc/en/filters
@@ -194,6 +194,12 @@
 
 toupper, tolower - Self-explanatory. Currently ASCII only.
 
+prn7pl - Converts polish chars to control sequences for EPSON-compatible
+printer. Using only 7-bit chars, backspacing printer's head and vertical
+positioning chars ,.'` it creates pseudo-polish gryphs. You can specify
+options: /nlq (default) which optimalizes output for better quality
+printers and /draft - useful for ex. for 9-nails printer.
+
 
 	ADDITIONAL FILTERS FOR FILTERM
 
--- konwert-1.8.orig/doc/pl/filtry
+++ konwert-1.8/doc/pl/filtry
@@ -200,6 +200,12 @@
 toupper, tolower - Zamienia tekst na duøe / ma≥e litery. Na razie tylko
 ASCII.
 
+prn7pl - Konwertuje polskie znaki na sekwencje sterujÍce drukarki zgodnej
+z EPSON wykorzystuj±c jedynie podstawowe znaki 7-bitowe, cofaj±c g≥owicÍ
+drukarki i drukuj±c na odpowiedniej wysoko∂ci ,.'` daj±ce w sumie pseudo-
+polskie znaki. Przyjmuje opcje: /nlq (domy∂lne) optymalizowane do drukarek
+o lepszej rozdzielczo∂ci i /draft - np. dla trybu draft 9-ig≥Ûwki.
+
 
 	DODATKOWE FILTRY DLA FILTERMA
 
--- konwert-1.8.orig/aux/charsets/mik
+++ konwert-1.8/aux/charsets/mik
@@ -0,0 +1,129 @@
+	~	~
+	Ä	–ê
+	Å	–ë
+	Ç	–í
+	É	–ì
+	Ñ	–î
+	Ö	–ï
+	Ü	–ñ
+	á	–ó
+	à	–ò
+	â	–ô
+	ä	–ö
+	ã	–õ
+	å	–ú
+	ç	–ù
+	é	–û
+	è	–ü
+	ê	–†
+	ë	–°
+	í	–¢
+	ì	–£
+	î	–§
+	ï	–•
+	ñ	–¶
+	ó	–ß
+	ò	–®
+	ô	–©
+	ö	–™
+	õ	–´
+	ú	–¨
+	ù	–≠
+	û	–Æ
+	ü	–Ø
+	†	–∞
+	°	–±
+	¢	–≤
+	£	–≥
+	§	–¥
+	•	–µ
+	¶	–∂
+	ß	–∑
+	®	–∏
+	©	–π
+	™	–∫
+	´	–ª
+	¨	–º
+	≠	–Ω
+	Æ	–æ
+	Ø	–ø
+	∞	—Ä
+	±	—Å
+	≤	—Ç
+	≥	—É
+	¥	—Ñ
+	µ	—Ö
+	∂	—Ü
+	∑	—á
+	∏	—à
+	π	—â
+	∫	—ä
+	ª	—ã
+	º	—å
+	Ω	—ç
+	æ	—é
+	ø	—è
+	¿	‚îî
+	¡	‚î¥
+	¬	‚î¨
+	√	‚îú
+	ƒ	‚îÄ
+	≈	‚îº
+	∆	‚ï£
+	«	‚ïë
+	»	‚ïö
+	…	‚ïî
+	 	‚ï©
+	À	‚ï¶
+	Ã	‚ï†
+	Õ	‚ïê
+	Œ	‚ï¨
+	œ	‚îê
+	–	‚ñë
+	—	‚ñí
+	“	‚ñì
+	”	‚îÇ
+	‘	‚î§
+	’	‚Ññ
+	÷	¬ß
+	◊	‚ïó
+	ÿ	‚ïù
+	Ÿ	‚îò
+	⁄	‚îå
+	€	‚ñà
+	‹	‚ñÑ
+	›	‚ñå
+	ﬁ	‚ñê
+	ﬂ	‚ñÄ
+	‡	Œ±
+	·	Œ≤
+	‚	Œì
+	„	œÄ
+	‰	Œ£
+	Â	œÉ
+	Ê	Œº
+	Á	œÑ
+	Ë	Œ¶
+	È	Œò
+	Í	Œ©
+	Î	Œ¥
+	Ï	‚àû
+	Ì	‚àÖ
+	Ó	‚àà
+	Ô	‚à©
+		‚â°
+	Ò	¬±
+	Ú	‚â•
+	Û	‚â§
+	Ù	‚å†
+	ı	‚å°
+	ˆ	√∑
+	˜	‚âà
+	¯	¬∞
+	˘	‚àô
+	˙	¬∑
+	˚	‚àö
+	¸	‚Åø
+	˝	¬≤
+	˛	‚ñ†
+	ˇ	¬†
--- konwert-1.8.orig/filters/rtf-8bit
+++ konwert-1.8/filters/rtf-8bit
@@ -1,130 +1,130 @@
 #!/usr/bin/trs -f
 
-	'\\80	\x80
-	'\\81	\x81
-	'\\82	\x82
-	'\\83	\x83
-	'\\84	\x84
-	'\\85	\x85
-	'\\86	\x86
-	'\\87	\x87
-	'\\88	\x88
-	'\\89	\x89
-	'\\8a	\x8A
-	'\\8b	\x8B
-	'\\8c	\x8C
-	'\\8d	\x8D
-	'\\8e	\x8E
-	'\\8f	\x8F
-	'\\90	\x90
-	'\\91	\x91
-	'\\92	\x92
-	'\\93	\x93
-	'\\94	\x94
-	'\\95	\x95
-	'\\96	\x96
-	'\\97	\x97
-	'\\98	\x98
-	'\\99	\x99
-	'\\9a	\x9A
-	'\\9b	\x9B
-	'\\9c	\x9C
-	'\\9d	\x9D
-	'\\9e	\x9E
-	'\\9f	\x9F
-	'\\a0	\xA0
-	'\\a1	\xA1
-	'\\a2	\xA2
-	'\\a3	\xA3
-	'\\a4	\xA4
-	'\\a5	\xA5
-	'\\a6	\xA6
-	'\\a7	\xA7
-	'\\a8	\xA8
-	'\\a9	\xA9
-	'\\aa	\xAA
-	'\\ab	\xAB
-	'\\ac	\xAC
-	'\\ad	\xAD
-	'\\ae	\xAE
-	'\\af	\xAF
-	'\\b0	\xB0
-	'\\b1	\xB1
-	'\\b2	\xB2
-	'\\b3	\xB3
-	'\\b4	\xB4
-	'\\b5	\xB5
-	'\\b6	\xB6
-	'\\b7	\xB7
-	'\\b8	\xB8
-	'\\b9	\xB9
-	'\\ba	\xBA
-	'\\bb	\xBB
-	'\\bc	\xBC
-	'\\bd	\xBD
-	'\\be	\xBE
-	'\\bf	\xBF
-	'\\c0	\xC0
-	'\\c1	\xC1
-	'\\c2	\xC2
-	'\\c3	\xC3
-	'\\c4	\xC4
-	'\\c5	\xC5
-	'\\c6	\xC6
-	'\\c7	\xC7
-	'\\c8	\xC8
-	'\\c9	\xC9
-	'\\ca	\xCA
-	'\\cb	\xCB
-	'\\cc	\xCC
-	'\\cd	\xCD
-	'\\ce	\xCE
-	'\\cf	\xCF
-	'\\d0	\xD0
-	'\\d1	\xD1
-	'\\d2	\xD2
-	'\\d3	\xD3
-	'\\d4	\xD4
-	'\\d5	\xD5
-	'\\d6	\xD6
-	'\\d7	\xD7
-	'\\d8	\xD8
-	'\\d9	\xD9
-	'\\da	\xDA
-	'\\db	\xDB
-	'\\dc	\xDC
-	'\\dd	\xDD
-	'\\de	\xDE
-	'\\df	\xDF
-	'\\e0	\xE0
-	'\\e1	\xE1
-	'\\e2	\xE2
-	'\\e3	\xE3
-	'\\e4	\xE4
-	'\\e5	\xE5
-	'\\e6	\xE6
-	'\\e7	\xE7
-	'\\e8	\xE8
-	'\\e9	\xE9
-	'\\ea	\xEA
-	'\\eb	\xEB
-	'\\ec	\xEC
-	'\\ed	\xED
-	'\\ee	\xEE
-	'\\ef	\xEF
-	'\\f0	\xF0
-	'\\f1	\xF1
-	'\\f2	\xF2
-	'\\f3	\xF3
-	'\\f4	\xF4
-	'\\f5	\xF5
-	'\\f6	\xF6
-	'\\f7	\xF7
-	'\\f8	\xF8
-	'\\f9	\xF9
-	'\\fa	\xFA
-	'\\fb	\xFB
-	'\\fc	\xFC
-	'\\fd	\xFD
-	'\\fe	\xFE
-	'\\ff	\xFF
+	\\'80	\x80
+	\\'81	\x81
+	\\'82	\x82
+	\\'83	\x83
+	\\'84	\x84
+	\\'85	\x85
+	\\'86	\x86
+	\\'87	\x87
+	\\'88	\x88
+	\\'89	\x89
+	\\'8a	\x8A
+	\\'8b	\x8B
+	\\'8c	\x8C
+	\\'8d	\x8D
+	\\'8e	\x8E
+	\\'8f	\x8F
+	\\'90	\x90
+	\\'91	\x91
+	\\'92	\x92
+	\\'93	\x93
+	\\'94	\x94
+	\\'95	\x95
+	\\'96	\x96
+	\\'97	\x97
+	\\'98	\x98
+	\\'99	\x99
+	\\'9a	\x9A
+	\\'9b	\x9B
+	\\'9c	\x9C
+	\\'9d	\x9D
+	\\'9e	\x9E
+	\\'9f	\x9F
+	\\'a0	\xA0
+	\\'a1	\xA1
+	\\'a2	\xA2
+	\\'a3	\xA3
+	\\'a4	\xA4
+	\\'a5	\xA5
+	\\'a6	\xA6
+	\\'a7	\xA7
+	\\'a8	\xA8
+	\\'a9	\xA9
+	\\'aa	\xAA
+	\\'ab	\xAB
+	\\'ac	\xAC
+	\\'ad	\xAD
+	\\'ae	\xAE
+	\\'af	\xAF
+	\\'b0	\xB0
+	\\'b1	\xB1
+	\\'b2	\xB2
+	\\'b3	\xB3
+	\\'b4	\xB4
+	\\'b5	\xB5
+	\\'b6	\xB6
+	\\'b7	\xB7
+	\\'b8	\xB8
+	\\'b9	\xB9
+	\\'ba	\xBA
+	\\'bb	\xBB
+	\\'bc	\xBC
+	\\'bd	\xBD
+	\\'be	\xBE
+	\\'bf	\xBF
+	\\'c0	\xC0
+	\\'c1	\xC1
+	\\'c2	\xC2
+	\\'c3	\xC3
+	\\'c4	\xC4
+	\\'c5	\xC5
+	\\'c6	\xC6
+	\\'c7	\xC7
+	\\'c8	\xC8
+	\\'c9	\xC9
+	\\'ca	\xCA
+	\\'cb	\xCB
+	\\'cc	\xCC
+	\\'cd	\xCD
+	\\'ce	\xCE
+	\\'cf	\xCF
+	\\'d0	\xD0
+	\\'d1	\xD1
+	\\'d2	\xD2
+	\\'d3	\xD3
+	\\'d4	\xD4
+	\\'d5	\xD5
+	\\'d6	\xD6
+	\\'d7	\xD7
+	\\'d8	\xD8
+	\\'d9	\xD9
+	\\'da	\xDA
+	\\'db	\xDB
+	\\'dc	\xDC
+	\\'dd	\xDD
+	\\'de	\xDE
+	\\'df	\xDF
+	\\'e0	\xE0
+	\\'e1	\xE1
+	\\'e2	\xE2
+	\\'e3	\xE3
+	\\'e4	\xE4
+	\\'e5	\xE5
+	\\'e6	\xE6
+	\\'e7	\xE7
+	\\'e8	\xE8
+	\\'e9	\xE9
+	\\'ea	\xEA
+	\\'eb	\xEB
+	\\'ec	\xEC
+	\\'ed	\xED
+	\\'ee	\xEE
+	\\'ef	\xEF
+	\\'f0	\xF0
+	\\'f1	\xF1
+	\\'f2	\xF2
+	\\'f3	\xF3
+	\\'f4	\xF4
+	\\'f5	\xF5
+	\\'f6	\xF6
+	\\'f7	\xF7
+	\\'f8	\xF8
+	\\'f9	\xF9
+	\\'fa	\xFA
+	\\'fb	\xFB
+	\\'fc	\xFC
+	\\'fd	\xFD
+	\\'fe	\xFE
+	\\'ff	\xFF
--- konwert-1.8.orig/filters/UTF8-prn7pl
+++ konwert-1.8/filters/UTF8-prn7pl
@@ -0,0 +1,64 @@
+#!/bin/bash -
+#
+# Wykorzystuje standardowe mozliwosci drukarki zgodnej ze standardem
+# EPSON do emulacji polskich znaczkow. Dziala na zasadzie sekwencji:
+#
+# <literka_podstawowa><cofnij>[w gore/dol]<znaczek ,`.>[w dol/gore]
+#
+# Sekwencje zostaly przetestowane na starej 9-iglowej drukarce
+# Star NL-10 we wszystkich jakosciach/czcionkach na niej dostepnych,
+# a takze w trybie NLQ.
+#
+# Wersja domyslna z parametrem 'nlq' wyglada lepiej w lepszej jakosci
+# czy rozdzielczosci wydruku, natomiast 'draft' w gorszej/nizszej.
+#
+# (c) by Grzegorz B. Prokopski 2003 <gadek@debian.org> licencja GPL2
+#
+
+# rozpoznajemy parametry ARG: domyslny 'nlq' i 'draft'
+P_WAY=nlq
+for A in $ARG; do
+	case $A in
+		nlq) P_WAY=nlq;;
+		draft) P_WAY=draft;;
+	esac
+done
+
+# domyslnie ustawiamy zmienne dla DRAFTu
+P_MOGN=\\b\\e\\112\\004,\\e\\152\\004
+P_MAKC=\\b\\'`'
+P_MKRP=\\b\\e\\152\\021.\\e\\112\\021
+P_DOGN=\\b\\e\\112\\004,\\e\\152\\004
+P_DAKC=\\b\\e\\152\\010\\'`'\\e\\112\\010
+P_DKRP=\\b\\e\\152\\027.\\e\\112\\027
+P_UKN=\\b/
+
+# zmieniamy dla NLQ
+if [ $P_WAY == nlq ]; then
+P_MAKC=\\b\\e\\152\\025,\\e\\112\\025
+P_MKRP=\\b\\e\\152\\023.\\e\\112\\023
+P_DAKC=\\b\\e\\152\\033,\\e\\112\\033
+P_DKRP=\\b\\e\\152\\031.\\e\\112\\031
+fi
+
+trs -e "
+		ƒÖ		a$P_MOGN
+		ƒá		c$P_MAKC
+		ƒô		e$P_MOGN
+		≈Ç		l$P_UKN
+		≈Ñ		n$P_MAKC
+		√≥		o$P_MAKC
+		≈õ		s$P_MAKC
+		≈∫		z$P_MAKC
+		≈º		z$P_MKRP
+		ƒÑ		A$P_DOGN
+		ƒÜ		C$P_DAKC
+		ƒò		E$P_DOGN
+		≈Å		L$P_UKN
+		≈É		N$P_DAKC
+		√ì		O$P_DAKC
+		≈ö		S$P_DAKC
+		≈π		Z$P_DAKC
+		≈ª		Z$P_DKRP
+"
+
--- konwert-1.8.orig/filters/UTF8-mik
+++ konwert-1.8/filters/UTF8-mik
@@ -0,0 +1,3 @@
+APPROX='
+' APPROX1='
+' REPLACE='?' "${0%/*}/../aux/UTF8-charset" -rf "${0%/*}/../aux/charsets/mik"
--- konwert-1.8.orig/bin/trs.cc
+++ konwert-1.8/bin/trs.cc
@@ -10,11 +10,14 @@
 #include <string.h>
 #include <iostream.h>
 #include <fstream.h>
-#include <strstream.h>
+#include <sstream>
 #include <ctype.h>
 #include <unistd.h>
+#include <stdlib.h>
 #define _(String) (String)
 
+using namespace std;
+
 #define WERSJA "1.8"
 #define WLKSLOWA 64*1024	// Maksymalna d≥ugo∂Ê s≥owa do zamiany
 				// albo s≥owa, na ktÛre zamieniamy
@@ -133,7 +136,7 @@
 
 struct zmienna
 {
-	int zmienna, wartosc;
+	int zmienna2, wartosc;
 	zmienna *nast;
 };
 
@@ -315,7 +318,7 @@
 					)
 					{
 						*zmienne = new zmienna;
-						(*zmienne)->zmienna =
+						(*zmienne)->zmienna2 =
 							s1[0] >= 'A' && s1[0] <= 'Z' ?
 								s1[0] - 'A'
 							:
@@ -474,7 +477,7 @@
 
 void regulyzarg (char *s)
 {
-	istrstream f (s);
+	istringstream f (s);
 	czytajreguly (f);
 }
 
@@ -542,7 +545,7 @@
 					zmienna *z = r->warunki;
 					while (z)
 					{
-						if (zmienne[z->zmienna] != z->wartosc)
+						if (zmienne[z->zmienna2] != z->wartosc)
 							goto niepasuje;
 						z = z->nast;
 					}
@@ -572,7 +575,7 @@
 					z = r->zmienne;
 					while (z)
 					{
-						zmienne[z->zmienna] = z->wartosc;
+						zmienne[z->zmienna2] = z->wartosc;
 						z = z->nast;
 					}
 					goto zamienione;
--- konwert-1.8.orig/bin/filterm.cc
+++ konwert-1.8/bin/filterm.cc
@@ -10,16 +10,16 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/wait.h>
+#include <util.h>
 
 #define WERSJA "1.8"
 
 char		*nazwaprogramu;
 char		*shell;
 int		master;
-int		slave;
+int		slave = -1;
 struct termios	tt;
 struct winsize	win;
-char		line[] = "/dev/ptyXX";
 
 void uzycie (int status)
 {
@@ -54,36 +54,31 @@
     exit (0);
 }
 
-void getmaster()
+static char linedata[PATH_MAX];
+char *line = linedata;
+
+int getpty(void) {
+    if (openpty(&master, &slave, line, NULL, NULL)) {
+        return -1;
+    } return master;
+}
+
+
+void getmasterslave()
 {
-    char *pty = &line[strlen ("/dev/ptyp")];
-    for (char *bank = "pqrs"; *bank; bank++)
-    {
-	line[strlen ("/dev/pty")] = *bank;
-	*pty = '0';
-	struct stat stb;
-	if (stat (line, &stb) < 0) break;
-	for (char *cp = "0123456789abcdef"; *cp; cp++)
-	{
-	    *pty = *cp;
-	    if ((master = open (line, O_RDWR)) >= 0)
-	    {
-		char *tp = &line[strlen("/dev/")];
-		*tp = 't';
-		int ok = !access (line, R_OK|W_OK);
-		*tp = 'p';
-		if (ok)
-		{
-		    tcgetattr (0, &tt);
-		    ioctl (0, TIOCGWINSZ, (char *) &win);
-		    return;
-		}
-		close (master);
-	    }
-	}
+    int r;
+    r = getpty();
+    if (r==-1) { 
+        cerr << "Error opening pty\n";
+        exit (1);
     }
-    cerr << "Out of pty's\n";
-    exit (1);
+
+    tcgetattr (0, &tt);
+    ioctl (0, TIOCGWINSZ, (char *) &win);
+    tcsetattr (slave, TCSAFLUSH, &tt);
+    ioctl (slave, TIOCSWINSZ, (char *) &win);
+    setsid();
+    ioctl (slave, TIOCSCTTY, 0);
 }
 
 void terminalsurowy()
@@ -114,24 +109,8 @@
     exit (1);
 }
 
-void getslave()
-{
-    line[strlen ("/dev/")] = 't';
-    slave = open (line, O_RDWR);
-    if (slave < 0)
-    {
-	perror (line);
-	exit (1);
-    }
-    tcsetattr (slave, TCSAFLUSH, &tt);
-    ioctl (slave, TIOCSWINSZ, (char *) &win);
-    setsid();
-    ioctl (slave, TIOCSCTTY, 0);
-}
-
 void komenda (int argc, char *argv[])
 {
-    getslave();
     dup2 (slave, 0);
     dup2 (slave, 1);
     dup2 (slave, 2);
@@ -181,7 +160,7 @@
     }
     if (argc < 3) uzycie (0);
 
-    getmaster();
+    getmasterslave();
     signal (SIGCHLD, koniec);
 
     if (strcmp (argv[2], "-"))
--- konwert-1.8.orig/man/man1/konwert.1
+++ konwert-1.8/man/man1/konwert.1
@@ -533,6 +533,17 @@
 .B tolower
 .PD
 Self-explanatory. Currently ASCII only.
+.TP
+.B prn7pl
+Converts polish chars to control sequences for EPSON-compatible
+printer. Using only 7-bit chars, backspacing printer's head and vertical
+positioning chars ,.'` it creates pseudo-polish gryphs. You can specify
+options:
+.B /nlq
+(default) which optimalizes output for better quality
+printers and
+.B /draft
+- useful for ex. for 9-nails printer.
 .SH FILES
 .TP
 .PD 0
--- konwert-1.8.orig/man/pl/man1/konwert.1
+++ konwert-1.8/man/pl/man1/konwert.1
@@ -52,7 +52,7 @@
 .PP
 .B konwert
 .BI '( "KOMENDA ARG\fR.\|.\|.\|" )'
-wykonuje dane polecenie szela. Jest to przydatne z opcjami
+wykonuje dane polecenie pow≥oki. Jest to przydatne z opcjami
 .B \-o
 albo
 .BR \-O .
@@ -531,6 +531,17 @@
 .B tolower
 .PD
 Zamienia tekst na duøe / ma≥e litery. Na razie tylko ASCII.
+.TP
+.B prn7pl
+Konwertuje polskie znaki na sekwencje sterujÍce drukarki zgodnej
+z EPSON wykorzystuj±c jedynie podstawowe znaki 7-bitowe, cofaj±c g≥owicÍ
+drukarki i drukuj±c na odpowiedniej wysoko∂ci ,.'` daj±ce w sumie pseudo-
+polskie znaki. Przyjmuje opcje:
+.B /nlq
+(domy∂lne) optymalizowane do drukarek
+o lepszej rozdzielczo∂ci i
+.B /draft
+- np. dla trybu draft 9-ig≥Ûwki.
 .SH PLIKI
 .TP
 .PD 0
--- konwert-1.8.orig/debian/README.debian
+++ konwert-1.8/debian/README.debian
@@ -0,0 +1,10 @@
+The Debian Package `konwert'
+----------------------------
+
+There are no shipped manpages in this version, but the 3 user-level
+programs (konvert, filterm, trs) are documented in English and Polish
+in the subdirs of this directory.
+
+These should surely be converted into manpages.
+
+Yann Dirson <dirson@debian.org>, Mon, 22 Jun 1998 09:19:36 +0200
--- konwert-1.8.orig/debian/changelog
+++ konwert-1.8/debian/changelog
@@ -0,0 +1,149 @@
+konwert (1.8-11) unstable; urgency=low
+
+  * Switch from debhelper v1 mode to v5:
+    * bash-completion script now maked as conffile (Closes: #358376).
+    * use ${misc:Depends} for every binary package.
+  * Fixed problem in a polish manpage (Closes: #296246).
+  * Fixed capitalization in README.Debian (lintian).
+  * Bumped Standards-Version to 3.6.2, no change.
+
+ -- Yann Dirson <dirson@debian.org>  Tue, 11 Apr 2006 01:22:10 +0200
+
+konwert (1.8-10) unstable; urgency=low
+
+  * Build with g++-4.0.
+
+ -- Yann Dirson <dirson@debian.org>  Tue,  5 Jul 2005 23:25:19 +0200
+
+konwert (1.8-9) unstable; urgency=low
+
+  * Fixed rtf-8bit to recognize the good escape sequences (Closes: #182439).
+  * Include bash-completion script for konwert(1) from Vitalij K. Lysoff.
+  * Fixed dh_buildinfo call to occur in binary-arch; ship a docdir in
+    konwert package instead of a symlink to allow this; create symlinks to
+    en and pl docdirs to leave docs easily found.  
+  * Reworked debian/rules a bit: added an install target to help
+    accomodating for this change.  
+  * Updated my email in README.Debian and copyright; updated upstream URL
+    in copyright.
+
+ -- Yann Dirson <dirson@debian.org>  Thu, 29 Apr 2004 00:30:26 +0200
+
+konwert (1.8-8) unstable; urgency=low
+
+  * Applied longstanding patch from Radovan Garabik to use UNIX98 ptys
+    (Closes: #108583, #195451).  Link filterm with libutil.
+  * Avoid redifinition of PATH_MAX in filterm.cc.
+  * Use dh-buildinfo.
+
+ -- Yann Dirson <dirson@debian.org>  Fri,  9 Jan 2004 16:31:57 +0100
+
+konwert (1.8-7) unstable; urgency=low
+
+  * Added support for mik from Kaloian Doganov (Closes: #196510).
+  * Added filters/UTF8-prn7pl and associated documentation from Grzegorz
+    B. Prokopski (Closes: #195908).
+  * Bumped Standards-Version from 3.0.1 to 3.5.10 (no change).
+
+ -- Yann Dirson <dirson@debian.org>  Sun,  8 Jun 2003 16:32:09 +0200
+
+konwert (1.8-6) unstable; urgency=low
+
+  * Converted bin/trc.cc to use <sstream> since g++ 3.3 dropped
+    <strstream.h> (Closes: #194942).
+  * Fixed konwert-filters short desc (lintian).
+
+ -- Yann Dirson <dirson@debian.org>  Thu, 29 May 2003 23:23:48 +0200
+
+konwert (1.8-5) unstable; urgency=low
+
+  * Rebuild for g++-3.2.
+  * Removed manually-inserted /usr/doc/-symlinks logic from maintainer
+    scripts.
+
+ -- Yann Dirson <dirson@debian.org>  Mon, 13 Jan 2003 16:03:50 +0100
+
+konwert (1.8-4) unstable; urgency=low
+
+  * Fix typos in package descriptions (Closes: #124848, #146488).  Ran
+    ispell and found a couple more.
+  * Added a lintian override for hardlinks usage.  Nothing forbids
+    hardlinks here, and they don't break anything.
+
+ -- Yann Dirson <dirson@debian.org>  Fri, 24 May 2002 00:55:17 +0200
+
+konwert (1.8-3) unstable; urgency=medium
+
+  * Include stdlib.h in trs.cc (Closes: #105180).
+  * Aknowledge previous NMU (Closes: #105304).
+  * Updated path to GPL in debian/copyright.
+
+ -- Yann Dirson <dirson@debian.org>  Wed,  1 Aug 2001 00:26:07 +0200
+
+konwert (1.8-2.1) unstable; urgency=low
+
+  * NMU
+  * Add build-depends header.  Closes: #105304
+
+ -- LaMont Jones <lamont@smallone.fc.hp.com>  Sat, 14 Jul 2001 22:52:34 -0600
+
+konwert (1.8-2) unstable; urgency=low
+
+  * Moved doc and man to /usr/share, as per FHS.
+  * Manually added handling of /usr/doc symlinks to konwert{,-dev}
+    packages.
+  * Changed the name of zmienna member in struct zmienna in trs.cc to
+    allow compilation.
+  * Bumped Standards-Version to 3.0.1.
+
+ -- Yann Dirson <dirson@debian.org>  Sun, 24 Oct 1999 19:32:48 +0200
+
+konwert (1.8-1) unstable; urgency=low
+
+  * New upstream release.
+  * Closes: Bug#38463 by recompilation.
+  * Make use of dh_perl for perl dependencies.
+
+ -- Yann Dirson <dirson@debian.org>  Tue,  6 Jul 1999 00:52:24 +0200
+
+konwert (1.7-3) frozen unstable; urgency=low
+
+  * Made konwert-filters to Recommend konwert because it is mostly
+    useless without the latter, and it makes an excuse for a lintian
+    override on "E: konwert-filters: trs-script-but-no-konwert-dep"
+    stuff.
+  * Recompiled against libstdc++-2.9 (Fixes: Bug#29044).
+  * Passed lintian 0.9.3 (well, with the restriction stated above...).
+
+ -- Yann Dirson <dirson@debian.org>  Thu, 12 Nov 1998 20:48:33 +0100
+
+konwert (1.7-2) unstable; urgency=low
+
+  * Incorporated patch from upstream to `512bold' filter so that
+    colors aren't reversed after dselect exists.
+
+ -- Yann Dirson <dirson@debian.org>  Fri, 31 Jul 1998 16:28:49 +0200
+
+konwert (1.7-1) unstable; urgency=low
+
+  * New upstream release:
+  * Install `CHANGES' as `changelog'.
+  * Passed lintian 0.6.0.
+
+ -- Yann Dirson <dirson@debian.org>  Fri, 24 Jul 1998 23:28:35 +0200
+
+konwert (1.6-1) unstable; urgency=low
+
+  * New upstream release:
+  ** Provides english manpages.
+  ** Uses /usr/share/ upstream.
+  ** Charset aliases.
+
+ -- Yann Dirson <dirson@debian.org>  Tue,  7 Jul 1998 21:42:02 +0200
+
+konwert (1.5-1) unstable; urgency=low
+
+  * Initial Release.
+  * Passed lintian 0.4.8.
+
+ -- Yann Dirson <dirson@debian.org>  Mon, 22 Jun 1998 09:19:36 +0200
--- konwert-1.8.orig/debian/copyright
+++ konwert-1.8/debian/copyright
@@ -0,0 +1,8 @@
+This package was debianized by Yann Dirson <dirson@debian.org>,
+from sources available from:
+
+http://qrczak.ids.net.pl/programy/linux/konwert/
+
+Upstream author is Marcin 'Qrczak' Kowalczyk <qrczak@knm.org.pl>.
+
+Copyright: GPL (see /usr/share/common-licenses/GPL).

 	  	 
