diff -Nurd lcal-2.1.0.ori/Makefile lcal-2.1.0/Makefile
--- lcal-2.1.0.ori/Makefile	2006-08-04 23:02:48.000000000 -0300
+++ lcal-2.1.0/Makefile	2009-05-24 09:31:46.000000000 -0300
@@ -107,6 +107,7 @@
 	MANDIR = /usr/man/man1
 	CATDIR = /usr/man/cat1
 endif
+BINDIR=$(DESTDIR)/bin
 
 OBJECTS = $(OBJDIR)/lcal.o
 
@@ -186,18 +187,14 @@
 fresh:	clobber $(LCAL)
 
 man:	$(DOCDIR)/lcal.man
-	nroff -man $(DOCDIR)/lcal.man > $(DOCDIR)/lcal.cat
-	groff -man -Tps $(DOCDIR)/lcal.man >$(DOCDIR)/lcal-help.ps
-	groff -man -Thtml $(DOCDIR)/lcal.man >$(DOCDIR)/lcal-help.html
-	groff -man -Tascii $(DOCDIR)/lcal.man >$(DOCDIR)/lcal-help.txt
+#	nroff -man $(DOCDIR)/lcal.man > $(DOCDIR)/lcal.cat
+#	groff -man -Tps $(DOCDIR)/lcal.man >$(DOCDIR)/lcal-help.ps
+#	groff -man -Thtml $(DOCDIR)/lcal.man >$(DOCDIR)/lcal-help.html
+#	groff -man -Tascii $(DOCDIR)/lcal.man >$(DOCDIR)/lcal-help.txt
 
 install:	$(EXECDIR)/$(LCAL) man
+	mkdir -p $(BINDIR)
+	mkdir -p $(INSTALLMAN1DIR)
 	cp $(EXECDIR)/$(LCAL) $(BINDIR)
-	if [ -d $(MANDIR) ]; then \
-		cp $(DOCDIR)/lcal.man $(MANDIR)/lcal.1; \
-		$(PACK) $(MANDIR)/lcal.1; \
-	fi
-	if [ -d $(CATDIR) ]; then \
-		cp $(DOCDIR)/lcal.cat $(CATDIR)/lcal.1; \
-		$(PACK) $(CATDIR)/lcal.1; \
-	fi
+	cp $(DOCDIR)/lcal.man $(INSTALLMAN1DIR)/lcal.1
+
