diff -Nurd strigi-0.7.1/src/CMakeLists.txt strigi-0.7.1-new/src/CMakeLists.txt
--- strigi-0.7.1/src/CMakeLists.txt	2010-01-06 18:09:39.000000000 -0500
+++ strigi-0.7.1-new/src/CMakeLists.txt	2010-01-12 10:49:32.000000000 -0500
@@ -2,7 +2,7 @@
 # Check copied from kdelibs FindX11.cmake.
 CHECK_LIBRARY_EXISTS("socket" "connect" "" CMAKE_LIB_SOCKET_HAS_CONNECT)
 
-set(DIRS streams streamanalyzer archivereader combinedindexer dummyindexer)
+set(DIRS archivereader combinedindexer dummyindexer)
 
 if (BUILD_UTILS)
     add_subdirectory(strigicmd)
@@ -28,7 +28,7 @@
 endif (SQLite_FOUND)
 
 if(NOT WIN32)
-  set(DIRS ${DIRS} htmlgui)
+#  set(DIRS ${DIRS} htmlgui)
 endif(NOT WIN32)
 
 if (QT_QTDBUS_FOUND)
diff -Nurd strigi-0.7.1/src/daemon/dbus/dbuscpp/dbusmessagereader.cpp strigi-0.7.1-new/src/daemon/dbus/dbuscpp/dbusmessagereader.cpp
--- strigi-0.7.1/src/daemon/dbus/dbuscpp/dbusmessagereader.cpp	2010-01-06 18:09:39.000000000 -0500
+++ strigi-0.7.1-new/src/daemon/dbus/dbuscpp/dbusmessagereader.cpp	2010-01-12 10:48:11.000000000 -0500
@@ -51,6 +51,7 @@
     dbus_message_iter_next(&it);
     return *this;
 }
+#ifdef __x86_64__
 DBusMessageReader&
 DBusMessageReader::operator>>(dbus_uint64_t& s) {
     if (!isOk()) return *this;
@@ -62,6 +63,7 @@
     dbus_message_iter_next(&it);
     return *this;
 }
+#endif
 DBusMessageReader&
 DBusMessageReader::operator>>(dbus_int64_t& s) {
     if (!isOk()) return *this;
@@ -74,6 +76,17 @@
     return *this;
 }
 DBusMessageReader&
+DBusMessageReader::operator>>(uint64_t& s) {
+    if (!isOk()) return *this;
+    if (DBUS_TYPE_UINT64 != dbus_message_iter_get_arg_type(&it)) {
+        close();
+        return *this;
+    }
+    dbus_message_iter_get_basic(&it, &s);
+    dbus_message_iter_next(&it);
+    return *this;
+}
+DBusMessageReader&
 DBusMessageReader::operator>>(string& s) {
     if (!isOk()) return *this;
     if (DBUS_TYPE_STRING != dbus_message_iter_get_arg_type(&it)) {
diff -Nurd strigi-0.7.1/src/daemon/dbus/dbuscpp/dbusmessagereader.h strigi-0.7.1-new/src/daemon/dbus/dbuscpp/dbusmessagereader.h
--- strigi-0.7.1/src/daemon/dbus/dbuscpp/dbusmessagereader.h	2010-01-06 18:09:39.000000000 -0500
+++ strigi-0.7.1-new/src/daemon/dbus/dbuscpp/dbusmessagereader.h	2010-01-12 10:48:11.000000000 -0500
@@ -62,8 +62,12 @@
     DBusMessageReader& operator>>(std::string& s);
     DBusMessageReader& operator>>(dbus_int32_t& s);
     DBusMessageReader& operator>>(dbus_uint32_t& s);
+#ifdef __x86_64
     DBusMessageReader& operator>>(dbus_uint64_t& s);
     DBusMessageReader& operator>>(dbus_int64_t& s);
+#endif
+    DBusMessageReader& operator>>(int64_t& s);
+    DBusMessageReader& operator>>(uint64_t& s);
     DBusMessageReader& operator>>(std::multimap<int, std::string>&);
     DBusMessageReader& operator>>(std::vector<std::pair<bool, std::string> >&);
     bool isOk() const { return ok; }
diff -Nurd strigi-0.7.1/src/luceneindexer/CMakeLists.txt strigi-0.7.1-new/src/luceneindexer/CMakeLists.txt
--- strigi-0.7.1/src/luceneindexer/CMakeLists.txt	2010-01-06 18:09:39.000000000 -0500
+++ strigi-0.7.1-new/src/luceneindexer/CMakeLists.txt	2010-01-12 10:48:11.000000000 -0500
@@ -16,6 +16,8 @@
     ADD_DEFINITIONS(-DUNICODE)
 ENDIF(WIN32)
 
+add_definitions(-I${CLUCENE_LIBRARY_DIR})
+
 # We need to compile the clucene index plugin with default visibility, because
 # CLucene does not export the symbol for CLuceneError. If this symbol is not
 # exported, the exception cannot be caught and can cause programs that use
