diff -uNbwr wview-3.4.0/Makefile.in wview-3.4.0-new/Makefile.in
--- wview-3.4.0/Makefile.in	2007-12-16 23:41:55.000000000 -0500
+++ wview-3.4.0-new/Makefile.in	2007-12-19 13:51:58.000000000 -0500
@@ -191,8 +191,8 @@
 SUBDIRS = stations $(am__append_1) $(am__append_2) htmlgenerator \
 	alarms cwop ftp ssh procmon wviewconfig utilities \
 	$(am__append_3)
-WV_CONFIG_DIR = /etc/wview
-WV_RUN_DIR = /var/wview
+WV_CONFIG_DIR = $(sysconfdir)/wview
+WV_RUN_DIR = $(localstatedir)/wview
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
@@ -628,19 +628,17 @@
 
 # this target installs the default environment in the default places
 install-env:	install
-	mkdir -p $(WV_CONFIG_DIR)
-	mkdir -p $(WV_RUN_DIR)
-	mkdir -p $(WV_RUN_DIR)/noaa
-	mkdir -p $(WV_RUN_DIR)/alarms
-	mkdir -p $(WV_RUN_DIR)/archive
-	mkdir -p $(WV_RUN_DIR)/img
-	cp -rf $(WV_RUN_DIR)/img $(WV_RUN_DIR)/img-old
-	cp -rf $(top_srcdir)/bin/img $(WV_RUN_DIR)
-	cp -rf $(WV_CONFIG_DIR) $(WV_CONFIG_DIR)-old
-	cp -rf $(top_srcdir)/examples/html/classic/*.* $(WV_CONFIG_DIR)/html
-	cp -rf $(top_srcdir)/examples/alarms $(WV_CONFIG_DIR)
-	cp -rf $(top_srcdir)/examples/conf/*.* $(WV_CONFIG_DIR)
-	chmod +x $(WV_CONFIG_DIR)/post-generate.sh
+	install -d $(DESTDIR)$(WV_CONFIG_DIR)
+	install -d $(DESTDIR)$(WV_CONFIG_DIR)/html
+	install -c -m 644 $(top_srcdir)/examples/html/classic/* $(DESTDIR)$(WV_CONFIG_DIR)/html/
+	install -c -m 755 $(top_srcdir)/examples/alarms/* $(DESTDIR)$(WV_CONFIG_DIR)/
+	install -c -m 644 $(top_srcdir)/examples/conf/* $(DESTDIR)$(WV_CONFIG_DIR)/
+	install -c -m 755 $(top_srcdir)/examples/conf/*.sh $(DESTDIR)$(WV_CONFIG_DIR)/
+	install -d $(DESTDIR)$(WV_RUN_DIR)/noaa
+	install -d $(DESTDIR)$(WV_RUN_DIR)/alarms
+	install -d $(DESTDIR)$(WV_RUN_DIR)/archive
+	install -d $(DESTDIR)$(WV_RUN_DIR)/img
+	install -c -m 644 $(top_srcdir)/bin/img/* $(DESTDIR)$(WV_RUNDIR)/
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -uNbwr wview-3.4.0/wviewconfig/wviewconfig.sh wview-3.4.0-new/wviewconfig/wviewconfig.sh
--- wview-3.4.0/wviewconfig/wviewconfig.sh	2007-12-12 19:22:28.000000000 -0500
+++ wview-3.4.0-new/wviewconfig/wviewconfig.sh	2007-12-19 14:48:23.000000000 -0500
@@ -23,10 +23,15 @@
 ################################################################################
 #################################  M A C R O S  ################################
 ################################################################################
-WVIEWD_PID=/var/wview/wviewd.pid
+WVIEW_DATA_DIR=/var/wview
+WVIEWD_PID=$WVIEW_DATA_DIR/wviewd.pid
 WVIEW_CONF_DIR=/etc/wview
 STATION_TYPE=VantagePro
-
+if [ -x "/bin/echo" ]; then
+	NONEWLINE_ECHO="/bin/echo -n"
+else
+	NONEWLINE_ECHO="echo -n"
+fi
 
 ################################################################################
 #######################  D E F I N E  F U N C T I O N S  #######################
@@ -139,7 +144,7 @@
         STATION_RAIN_STORM_IDLE_STOP=12
     fi
     if [ "" = "$ARCHIVE_PATH" ]; then
-        ARCHIVE_PATH=/var/wview/archive
+        ARCHIVE_PATH=$WVIEW_DATA_DIR/archive
     fi
     if [ "" = "$POLL_INTERVAL" ]; then
         POLL_INTERVAL=15000
@@ -186,7 +191,7 @@
         STATION_STATE=changeME
     fi
     if [ "" = "$IMAGE_PATH" ]; then
-        IMAGE_PATH=/var/wview/img
+        IMAGE_PATH=$WVIEW_DATA_DIR/img
     fi
     if [ "" = "$HTML_PATH" ]; then
         HTML_PATH=$WVIEW_CONF_DIR/html
@@ -347,7 +352,7 @@
         mv $WVIEW_CONF_DIR/wview.conf $WVIEW_CONF_DIR/wview.conf.old
     fi
     echo ""
-    echo -n "Writing $WVIEW_CONF_DIR/wview.conf: "
+    $NONEWLINE_ECHO "Writing $WVIEW_CONF_DIR/wview.conf: "
     mkdir -p $WVIEW_CONF_DIR
     
     echo "#@" > $WVIEW_CONF_DIR/wview.conf
@@ -453,7 +458,7 @@
         echo "STATION_LONGITUDE=$STATION_LONGITUDE" >> $WVIEW_CONF_DIR/wview.conf
         echo ""  >> $WVIEW_CONF_DIR/wview.conf
     
-        echo "# DO NOT CHANGE this once you have archive records stored in /var/wview/archive!"  >> $WVIEW_CONF_DIR/wview.conf
+        echo "# DO NOT CHANGE this once you have archive records stored in $WVIEW_DATA_DIR/archive!"  >> $WVIEW_CONF_DIR/wview.conf
         echo "# This effects many things (CWOP/Wunderground update interval, etc.) so choose carefully..."  >> $WVIEW_CONF_DIR/wview.conf
         echo "# Weather data archive interval (minutes, one of 5, 10, 15, 30):"  >> $WVIEW_CONF_DIR/wview.conf
         echo "STATION_ARCHIVE_INTERVAL=$STATION_ARCHIVE_INTERVAL" >> $WVIEW_CONF_DIR/wview.conf
@@ -520,11 +525,11 @@
     echo ""
     echo "################################################################################"
     echo ""
-    echo -n "pausing 3 seconds "
+    $NONEWLINE_ECHO "pausing 3 seconds "
     sleep 1
-    echo -n "."
+    $NONEWLINE_ECHO "."
     sleep 1
-    echo -n "."
+    $NONEWLINE_ECHO "."
     sleep 1
     echo "."
     echo ""
@@ -539,7 +544,7 @@
     echo "Generic refers to stations without location or archive interval parameter storage -"
     echo "in this case you must enter location and archive interval now..."
     echo "What weather station do you have (one of VantagePro, Generic):"
-    echo -n "($STATION_TYPE): "
+    $NONEWLINE_ECHO "($STATION_TYPE): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         STATION_TYPE=$INVAL
@@ -562,7 +567,7 @@
     echo "-------------------------------------------------------------" 
     echo "--Note-- USB is considered serial (/dev/ttyUSBx, /dev/tty.SLAB_USBtoUART, etc.)"
     echo "Physical Interface the weather station is connected to (one of serial or ethernet)"
-    echo -n "($STATION_INTERFACE): "
+    $NONEWLINE_ECHO "($STATION_INTERFACE): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         STATION_INTERFACE=$INVAL
@@ -573,7 +578,7 @@
         echo ""
         echo "-------------------------------------------------------------" 
         echo "Serial port (device) the weather station is connected to (/dev/ttyS0, /dev/cuaa0, /dev/ttyUSBx, /dev/tty.SLAB_USBtoUART, etc.)"
-        echo -n "($STATION_DEV): "
+        $NONEWLINE_ECHO "($STATION_DEV): "
         read INVAL
         if [ "" != "$INVAL" ]; then
             STATION_DEV=$INVAL
@@ -582,7 +587,7 @@
         echo ""
         echo "-------------------------------------------------------------" 
         echo "Hostname or IP address the weather station is connected to (i.e. xyplex1 or 10.11.12.13)"
-        echo -n "($STATION_HOST): "
+        $NONEWLINE_ECHO "($STATION_HOST): "
         read INVAL
         if [ "" != "$INVAL" ]; then
             STATION_HOST=$INVAL
@@ -592,7 +597,7 @@
         echo "-------------------------------------------------------------" 
         echo "--Note-- This is dependent upon the terminal server used - consult your terminal server documentation"
         echo "TCP port the weather station is connected to on $STATION_HOST (i.e. 2101)"
-        echo -n "($STATION_PORT): "
+        $NONEWLINE_ECHO "($STATION_PORT): "
         read INVAL
         if [ "" != "$INVAL" ]; then
             STATION_PORT=$INVAL
@@ -605,7 +610,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Station Rain Season Start Month (1 - 12):"
-    echo -n "($STATION_RAIN_SEASON_START): "
+    $NONEWLINE_ECHO "($STATION_RAIN_SEASON_START): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         STATION_RAIN_SEASON_START=$INVAL
@@ -614,7 +619,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Station rain storm start trigger (rainfall rate in inches/hour):"
-    echo -n "($STATION_RAIN_STORM_TRIGGER_START): "
+    $NONEWLINE_ECHO "($STATION_RAIN_STORM_TRIGGER_START): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         STATION_RAIN_STORM_TRIGGER_START=$INVAL
@@ -623,7 +628,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Station rain storm stop time (hours without any rainfall which will end the storm):"
-    echo -n "($STATION_RAIN_STORM_IDLE_STOP): "
+    $NONEWLINE_ECHO "($STATION_RAIN_STORM_IDLE_STOP): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         STATION_RAIN_STORM_IDLE_STOP=$INVAL
@@ -633,7 +638,7 @@
     echo "-------------------------------------------------------------" 
     echo "--Note-- Under normal circumstances, the default selection should be used"
     echo "Where to store archive files (*.wlk) - this must be local to the wview server"
-    echo -n "($ARCHIVE_PATH): "
+    $NONEWLINE_ECHO "($ARCHIVE_PATH): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         ARCHIVE_PATH=$INVAL
@@ -643,7 +648,7 @@
     echo "-------------------------------------------------------------" 
     echo "--Note-- This value must be one of 10000, 15000 or 30000 (milliseconds)"
     echo "How often should wviewd poll the weather station for current conditions (milliseconds)?"
-    echo -n "($POLL_INTERVAL): "
+    $NONEWLINE_ECHO "($POLL_INTERVAL): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         POLL_INTERVAL=$INVAL
@@ -653,7 +658,7 @@
     echo "-------------------------------------------------------------" 
     echo "--Note-- This value must be >= 15000 (milliseconds)"
     echo "How often should wviewd provide current conditions data to clients such as wvalarmd (milliseconds)?"
-    echo -n "($PUSH_INTERVAL): "
+    $NONEWLINE_ECHO "($PUSH_INTERVAL): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         PUSH_INTERVAL=$INVAL
@@ -674,7 +679,7 @@
     echo "01000000     - wvhttpd"
     echo "11111111     - All On"
     echo "Verbose diagnostic log messages per-daemon bitmask (8 characters, 0 or 1 only):"
-    echo -n "($VERBOSE_MSGS): "
+    $NONEWLINE_ECHO "($VERBOSE_MSGS): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         VERBOSE_MSGS=$INVAL
@@ -687,7 +692,7 @@
         echo "--Note--  enabling this can occasionally cause serial IF problems -"
         echo "--Note--  if you are having WAKEUP console errors, try disabling this"
         echo "Should wview generate RX check data (populate rxCheck.png chart) (0 or 1)?"
-        echo -n "($DO_RXCHECK): "
+        $NONEWLINE_ECHO "($DO_RXCHECK): "
         read INVAL
         if [ "" != "$INVAL" ]; then
             DO_RXCHECK=$INVAL
@@ -699,7 +704,7 @@
         echo ""
         echo "-------------------------------------------------------------" 
         echo "Weather station elevation (feet above sea level)?"
-        echo -n "($STATION_ELEVATION): "
+        $NONEWLINE_ECHO "($STATION_ELEVATION): "
         read INVAL
         if [ "" = "$INVAL" ]; then
             if [ "changeME" = "$STATION_ELEVATION" ]; then
@@ -715,7 +720,7 @@
         echo "--Note--  This is given in decimal degrees NORTH (-90.00 to 90.00) -"
         echo "--Note--  Examples: -46.31 (46.31 degrees SOUTH), 33.4 (33.4 degrees NORTH)"
         echo "Weather station latitude?"
-        echo -n "($STATION_LATITUDE): "
+        $NONEWLINE_ECHO "($STATION_LATITUDE): "
         read INVAL
         if [ "" = "$INVAL" ]; then
             if [ "changeME" = "$STATION_LATITUDE" ]; then
@@ -731,7 +736,7 @@
         echo "--Note--  This is given in decimal degrees EAST (-179.99 to 180.00) -"
         echo "--Note--  Examples: -96.9 (96.9 degrees WEST), 112.71 (112.71 degrees EAST)"
         echo "Weather station longitude?"
-        echo -n "($STATION_LONGITUDE): "
+        $NONEWLINE_ECHO "($STATION_LONGITUDE): "
         read INVAL
         if [ "" = "$INVAL" ]; then
             if [ "changeME" = "$STATION_LONGITUDE" ]; then
@@ -744,14 +749,14 @@
     
         echo ""
         echo "-------------------------------------------------------------" 
-        echo "--Note--  DO NOT CHANGE this once you have archive records stored in /var/wview/archive"
+        echo "--Note--  DO NOT CHANGE this once you have archive records stored in $WVIEW_DATA_DIR/archive"
         echo "--Note--      unless you delete all *.wlk files first!"
         echo "--Note--  This effects many things (CWOP/Wunderground update interval, etc.) so"
         echo "--Note--      choose it carefully and be prepared to stay with your choice!"
         echo "--Note--  If unsure what all this affects, CTRL-C to exit this script and do"
         echo "--Note--      a search for 'archive interval' in the wview User Manual..."
         echo "Weather data archive interval (minutes, one of 5, 10, 15, 30)?"
-        echo -n "($STATION_ARCHIVE_INTERVAL): "
+        $NONEWLINE_ECHO "($STATION_ARCHIVE_INTERVAL): "
         read INVAL
         if [ "" != "$INVAL" ]; then
             STATION_ARCHIVE_INTERVAL=$INVAL
@@ -762,7 +767,7 @@
     echo "-------------------------------------------------------------" 
     echo "--Note-- currently only MySQL and PostgreSQL are supported"
     echo "Should wviewd store archive data in a relational database (0 or 1)?"
-    echo -n "($SQLDB_ENABLE): "
+    $NONEWLINE_ECHO "($SQLDB_ENABLE): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         SQLDB_ENABLE=$INVAL
@@ -773,7 +778,7 @@
         echo "-------------------------------------------------------------" 
         echo "--Note-- Do not enable unless you have installed extra sensors or have a VP Plus"
         echo "Should wviewd store extended sensor archive data in the database (0 or 1)?"
-        echo -n "($SQLDB_EXTENDED_INFO): "
+        $NONEWLINE_ECHO "($SQLDB_EXTENDED_INFO): "
         read INVAL
         if [ "" != "$INVAL" ]; then
             SQLDB_EXTENDED_INFO=$INVAL
@@ -782,7 +787,7 @@
         echo ""
         echo "-------------------------------------------------------------" 
         echo "Should wviewd store metric (international) archive data in the database (0 or 1)?"
-        echo -n "($SQLDB_STORE_METRIC): "
+        $NONEWLINE_ECHO "($SQLDB_STORE_METRIC): "
         read INVAL
         if [ "" != "$INVAL" ]; then
             SQLDB_STORE_METRIC=$INVAL
@@ -791,7 +796,7 @@
         echo ""
         echo "-------------------------------------------------------------" 
         echo "Hostname or IP address of the database server:"
-        echo -n "($SQLDB_HOST): "
+        $NONEWLINE_ECHO "($SQLDB_HOST): "
         read INVAL
         if [ "" = "$INVAL" ]; then
             if [ "changeME" = "$SQLDB_HOST" ]; then
@@ -805,7 +810,7 @@
         echo ""
         echo "-------------------------------------------------------------" 
         echo "Username of the database server:"
-        echo -n "($SQLDB_USERNAME): "
+        $NONEWLINE_ECHO "($SQLDB_USERNAME): "
         read INVAL
         if [ "" = "$INVAL" ]; then
             if [ "changeME" = "$SQLDB_USERNAME" ]; then
@@ -819,7 +824,7 @@
         echo ""
         echo "-------------------------------------------------------------" 
         echo "Password for $SQLDB_USERNAME on the database server:"
-        echo -n "($SQLDB_PASSWORD): "
+        $NONEWLINE_ECHO "($SQLDB_PASSWORD): "
         read INVAL
         if [ "" = "$INVAL" ]; then
             if [ "changeME" = "$SQLDB_PASSWORD" ]; then
@@ -833,7 +838,7 @@
         echo ""
         echo "-------------------------------------------------------------" 
         echo "Database name for weather records on the database server:"
-        echo -n "($SQLDB_DB_NAME): "
+        $NONEWLINE_ECHO "($SQLDB_DB_NAME): "
         read INVAL
         if [ "" = "$INVAL" ]; then
             if [ "changeME" = "$SQLDB_DB_NAME" ]; then
@@ -847,7 +852,7 @@
         echo ""
         echo "-------------------------------------------------------------" 
         echo "Table name for weather records on the database server:"
-        echo -n "($SQLDB_TABLE_NAME): "
+        $NONEWLINE_ECHO "($SQLDB_TABLE_NAME): "
         read INVAL
         if [ "" = "$INVAL" ]; then
             if [ "changeME" = "$SQLDB_TABLE_NAME" ]; then
@@ -867,7 +872,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Station Name (no spaces):"
-    echo -n "($STATION_NAME): "
+    $NONEWLINE_ECHO "($STATION_NAME): "
     read INVAL
     if [ "" = "$INVAL" ]; then
         if [ "changeME" = "$STATION_NAME" ]; then
@@ -881,7 +886,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Station City (no spaces):"
-    echo -n "($STATION_CITY): "
+    $NONEWLINE_ECHO "($STATION_CITY): "
     read INVAL
     if [ "" = "$INVAL" ]; then
         if [ "changeME" = "$STATION_CITY" ]; then
@@ -895,7 +900,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Station State (no spaces):"
-    echo -n "($STATION_STATE): "
+    $NONEWLINE_ECHO "($STATION_STATE): "
     read INVAL
     if [ "" = "$INVAL" ]; then
         if [ "changeME" = "$STATION_STATE" ]; then
@@ -913,7 +918,7 @@
     echo "--Note-- your site to a remote web server (such as your ISP), please use the "
     echo "--Note-- default selection."
     echo "Where to store generated web site graphics and HTML files:"
-    echo -n "($IMAGE_PATH): "
+    $NONEWLINE_ECHO "($IMAGE_PATH): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         IMAGE_PATH=$INVAL
@@ -923,7 +928,7 @@
     echo "-------------------------------------------------------------" 
     echo "--Note-- Normal installs should use the default selection"
     echo "Where to find HTML template files for web site generation:"
-    echo -n "($HTML_PATH): "
+    $NONEWLINE_ECHO "($HTML_PATH): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         HTML_PATH=$INVAL
@@ -932,7 +937,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Generation start offset (0-4) in minutes past the 5-minute mark:"
-    echo -n "($START_OFFSET): "
+    $NONEWLINE_ECHO "($START_OFFSET): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         START_OFFSET=$INVAL
@@ -941,7 +946,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Generation interval (refresh interval for your site data) (minutes):"
-    echo -n "($GENERATE_INTERVAL): "
+    $NONEWLINE_ECHO "($GENERATE_INTERVAL): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         GENERATE_INTERVAL=$INVAL
@@ -950,7 +955,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Enable metric conversion/units?"
-    echo -n "($METRIC_UNITS): "
+    $NONEWLINE_ECHO "($METRIC_UNITS): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         METRIC_UNITS=$INVAL
@@ -960,7 +965,7 @@
         echo ""
         echo "-------------------------------------------------------------" 
         echo "Use mm for rain instead of cm?"
-        echo -n "($METRIC_USE_RAIN_MM): "
+        $NONEWLINE_ECHO "($METRIC_USE_RAIN_MM): "
         read INVAL
         if [ "" != "$INVAL" ]; then
             METRIC_USE_RAIN_MM=$INVAL
@@ -970,7 +975,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Store/process VP Plus extended sensor values for site generation?"
-    echo -n "($EXTENDED_DATA): "
+    $NONEWLINE_ECHO "($EXTENDED_DATA): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         EXTENDED_DATA=$INVAL
@@ -981,7 +986,7 @@
     echo "--Note-- Each file contains one day's worth of archive records in ASCII format"
     echo "--Note-- Special Values: -1 disables archive browser file generation and 0 keeps all files"
     echo "How many archive record browser files to keep (days):"
-    echo -n "($ARCHIVE_BROWSER_FILES_TO_KEEP): "
+    $NONEWLINE_ECHO "($ARCHIVE_BROWSER_FILES_TO_KEEP): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         ARCHIVE_BROWSER_FILES_TO_KEEP=$INVAL
@@ -990,7 +995,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Moon Phase increasing text (no spaces):"
-    echo -n "($MPHASE_INCREASE): "
+    $NONEWLINE_ECHO "($MPHASE_INCREASE): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         MPHASE_INCREASE=$INVAL
@@ -999,7 +1004,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Moon Phase decreasing text (no spaces):"
-    echo -n "($MPHASE_DECREASE): "
+    $NONEWLINE_ECHO "($MPHASE_DECREASE): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         MPHASE_DECREASE=$INVAL
@@ -1008,7 +1013,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Moon Phase full text (no spaces):"
-    echo -n "($MPHASE_FULL): "
+    $NONEWLINE_ECHO "($MPHASE_FULL): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         MPHASE_FULL=$INVAL
@@ -1017,7 +1022,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Local Radar URL (no spaces):"
-    echo -n "($LOCAL_RADAR_URL): "
+    $NONEWLINE_ECHO "($LOCAL_RADAR_URL): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         LOCAL_RADAR_URL=$INVAL
@@ -1026,7 +1031,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Local Forecast URL (no spaces):"
-    echo -n "($LOCAL_FORECAST_URL): "
+    $NONEWLINE_ECHO "($LOCAL_FORECAST_URL): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         LOCAL_FORECAST_URL=$INVAL
@@ -1037,7 +1042,7 @@
     echo "--Note-- see 'man strftime' for allowable formats "
     echo "--Note-- underscores will be replaced by spaces in output "
     echo "Date format (no spaces): "
-    echo -n "($DATE_FORMAT): "
+    $NONEWLINE_ECHO "($DATE_FORMAT): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         DATE_FORMAT=$INVAL
@@ -1053,7 +1058,7 @@
         mv $WVIEW_CONF_DIR/htmlgen.conf $WVIEW_CONF_DIR/htmlgen.conf.old
     fi
     echo ""
-    echo -n "Writing $WVIEW_CONF_DIR/htmlgen.conf: "
+    $NONEWLINE_ECHO "Writing $WVIEW_CONF_DIR/htmlgen.conf: "
     
     echo "#@" > $WVIEW_CONF_DIR/htmlgen.conf
     echo "#"  >> $WVIEW_CONF_DIR/htmlgen.conf
@@ -1155,7 +1160,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "APRS Call Sign (6 characters, no spaces):"
-    echo -n "($APRS_CALL_SIGN): "
+    $NONEWLINE_ECHO "($APRS_CALL_SIGN): "
     read INVAL
     if [ "" = "$INVAL" ]; then
         if [ "changeME" = "$APRS_CALL_SIGN" ]; then
@@ -1172,7 +1177,7 @@
     echo "--Note-- for multiple entries, although this is highly discouraged - "
     echo "--Note-- See http://www.wxqa.com/activecwd.html for a list of servers"
     echo "APRS primary server name (hostname or IP address):"
-    echo -n "($APRS_SERVER1): "
+    $NONEWLINE_ECHO "($APRS_SERVER1): "
     read INVAL
     if [ "" = "$INVAL" ]; then
         if [ "changeME" = "$APRS_SERVER1" ]; then
@@ -1186,7 +1191,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "APRS primary port number (usually 23):"
-    echo -n "($APRS_PORTNO1): "
+    $NONEWLINE_ECHO "($APRS_PORTNO1): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         APRS_PORTNO1=$INVAL
@@ -1195,7 +1200,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "APRS secondary server name (hostname or IP address):"
-    echo -n "($APRS_SERVER2): "
+    $NONEWLINE_ECHO "($APRS_SERVER2): "
     read INVAL
     if [ "" = "$INVAL" ]; then
         if [ "changeME" = "$APRS_SERVER2" ]; then
@@ -1209,7 +1214,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "APRS secondary port number (usually 23):"
-    echo -n "($APRS_PORTNO2): "
+    $NONEWLINE_ECHO "($APRS_PORTNO2): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         APRS_PORTNO2=$INVAL
@@ -1218,7 +1223,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "APRS tertiary server name (hostname or IP address):"
-    echo -n "($APRS_SERVER3): "
+    $NONEWLINE_ECHO "($APRS_SERVER3): "
     read INVAL
     if [ "" = "$INVAL" ]; then
         if [ "changeME" = "$APRS_SERVER3" ]; then
@@ -1232,7 +1237,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "APRS tertiary port number (usually 23):"
-    echo -n "($APRS_PORTNO3): "
+    $NONEWLINE_ECHO "($APRS_PORTNO3): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         APRS_PORTNO3=$INVAL
@@ -1252,7 +1257,7 @@
     echo "--Note-- See http://www.topozone.com/viewmaps.asp to get an accurate"
     echo "--Note-- latitude and longitude for your CWOP data"
     echo "Station latitude:"
-    echo -n "($LATITUDE): "
+    $NONEWLINE_ECHO "($LATITUDE): "
     read INVAL
     if [ "" = "$INVAL" ]; then
         if [ "changeME" = "$LATITUDE" ]; then
@@ -1271,7 +1276,7 @@
     echo "--Note--   hh   - hundredths of minutes (00 - 99)"
     echo "--Note--   d    - hemisphere indicator (E or W)"
     echo "Station longitude:"
-    echo -n "($LONGITUDE): "
+    $NONEWLINE_ECHO "($LONGITUDE): "
     read INVAL
     if [ "" = "$INVAL" ]; then
         if [ "changeME" = "$LONGITUDE" ]; then
@@ -1285,7 +1290,7 @@
     echo ""
     echo "-------------------------------------------------------------" 
     echo "Log the APRS WX packet when sending?"
-    echo -n "($LOG_WX_PACKET): "
+    $NONEWLINE_ECHO "($LOG_WX_PACKET): "
     read INVAL
     if [ "" != "$INVAL" ]; then
         LOG_WX_PACKET=$INVAL
@@ -1301,7 +1306,7 @@
         mv $WVIEW_CONF_DIR/wvcwop.conf $WVIEW_CONF_DIR/wvcwop.conf.old
     fi
     echo ""
-    echo -n "Writing $WVIEW_CONF_DIR/wvcwop.conf: "
+    $NONEWLINE_ECHO "Writing $WVIEW_CONF_DIR/wvcwop.conf: "
     
     echo "#@" > $WVIEW_CONF_DIR/wvcwop.conf
     echo "#"  >> $WVIEW_CONF_DIR/wvcwop.conf
@@ -1364,7 +1369,7 @@
     fi
     echo "-------------------------------------------------------------" 
     echo "Enable CWOP (Citizen Weather Observer Program) support?"
-    echo -n "($DEFANS): "
+    $NONEWLINE_ECHO "($DEFANS): "
     read INVAL
     echo ""
     if [ "" != "$INVAL" ]; then
@@ -1375,7 +1380,7 @@
         get_cwop_conf_interactive
         write_cwop_conf
     else
-        echo -n "Disabling the CWOP support by naming the config file $WVIEW_CONF_DIR/wvcwop.conf-no-cwop: "
+        $NONEWLINE_ECHO "Disabling the CWOP support by naming the config file $WVIEW_CONF_DIR/wvcwop.conf-no-cwop: "
         if [ -f $WVIEW_CONF_DIR/wvcwop.conf ]; then
             mv $WVIEW_CONF_DIR/wvcwop.conf $WVIEW_CONF_DIR/wvcwop.conf-no-cwop
             echo "done."
@@ -1398,7 +1403,7 @@
     echo "-------------------------------------------------------------"
     echo "--Note-- You MUST enable HTTP (WUNDERGROUND/WEATHERFORYOU) support with '--enable-http' when ./configure was executed..."
     echo "Enable HTTP (WUNDERGROUND/WEATHERFORYOU) support?"
-    echo -n "($DEFANS): "
+    $NONEWLINE_ECHO "($DEFANS): "
     read INVAL
     echo ""
     if [ "" != "$INVAL" ]; then
@@ -1411,7 +1416,7 @@
         echo ""
         echo "-------------------------------------------------------------" 
         echo "WUNDERGROUND Station ID (obtained when you registered with WUNDERGROUND):"
-        echo -n "($STATIONID): "
+        $NONEWLINE_ECHO "($STATIONID): "
         read INVAL
         if [ "" = "$INVAL" ]; then
             if [ "changeME" = "$STATIONID" ]; then
@@ -1425,7 +1430,7 @@
         echo ""
         echo "-------------------------------------------------------------" 
         echo "WUNDERGROUND Password (obtained when you registered with WUNDERGROUND):"
-        echo -n "($PASSWORD): "
+        $NONEWLINE_ECHO "($PASSWORD): "
         read INVAL
         if [ "" = "$INVAL" ]; then
             if [ "changeME" = "$PASSWORD" ]; then
@@ -1440,7 +1445,7 @@
         echo ""
         echo "-------------------------------------------------------------" 
         echo "WEATHERFORYOU Station ID (obtained when you registered with WEATHERFORYOU):"
-        echo -n "($YOUSTATIONID): "
+        $NONEWLINE_ECHO "($YOUSTATIONID): "
         read INVAL
         if [ "" = "$INVAL" ]; then
             if [ "changeME" = "$YOUSTATIONID" ]; then
@@ -1454,7 +1459,7 @@
         echo ""
         echo "-------------------------------------------------------------" 
         echo "WEATHERFORYOU Password (obtained when you registered with WEATHERFORYOU):"
-        echo -n "($YOUPASSWORD): "
+        $NONEWLINE_ECHO "($YOUPASSWORD): "
         read INVAL
         if [ "" = "$INVAL" ]; then
             if [ "changeME" = "$YOUPASSWORD" ]; then
@@ -1468,7 +1473,7 @@
     
         write_http_conf
     else
-        echo -n "Disabling the HTTP (WUNDERGROUND/WEATHERFORYOU) support by naming the config file $WVIEW_CONF_DIR/http.conf-no-http: "
+        $NONEWLINE_ECHO "Disabling the HTTP (WUNDERGROUND/WEATHERFORYOU) support by naming the config file $WVIEW_CONF_DIR/http.conf-no-http: "
         if [ -f $WVIEW_CONF_DIR/http.conf ]; then
             mv $WVIEW_CONF_DIR/http.conf $WVIEW_CONF_DIR/http.conf-no-http
             echo "done."
@@ -1489,7 +1494,7 @@
         mv $WVIEW_CONF_DIR/http.conf $WVIEW_CONF_DIR/http.conf.old
     fi
     echo ""
-    echo -n "Writing $WVIEW_CONF_DIR/http.conf: "
+    $NONEWLINE_ECHO "Writing $WVIEW_CONF_DIR/http.conf: "
     
     echo "#@" > $WVIEW_CONF_DIR/http.conf
     echo "#"  >> $WVIEW_CONF_DIR/http.conf
@@ -1530,7 +1535,7 @@
         DEFANS=0
     fi
     echo "Run the alarms daemon wvalarmd needed for alarm triggering or data feeds?"
-    echo -n "($DEFANS): "
+    $NONEWLINE_ECHO "($DEFANS): "
     read INVAL
     echo ""
     if [ "" != "$INVAL" ]; then
@@ -1538,7 +1543,7 @@
     fi
     
     if [ "1" = "$DEFANS" ]; then
-        echo -n "Enabling the alarm daemon by naming the config file $WVIEW_CONF_DIR/wvalarm.conf: "
+        $NONEWLINE_ECHO "Enabling the alarm daemon by naming the config file $WVIEW_CONF_DIR/wvalarm.conf: "
         if [ -f $WVIEW_CONF_DIR/wvalarm.conf ]; then
             echo "file already exists - wvalarmd is enabled"
             echo "Remember to edit $WVIEW_CONF_DIR/wvalarm.conf if you want to define alarms..."
@@ -1552,7 +1557,7 @@
             echo "Alarms will be disabled until you copy the config files from the distro and re-run this script..."
         fi
     else
-        echo -n "Disabling the alarm daemon by naming the config file $WVIEW_CONF_DIR/wvalarm.conf-no-alarms: "
+        $NONEWLINE_ECHO "Disabling the alarm daemon by naming the config file $WVIEW_CONF_DIR/wvalarm.conf-no-alarms: "
         if [ -f $WVIEW_CONF_DIR/wvalarm.conf ]; then
             mv $WVIEW_CONF_DIR/wvalarm.conf $WVIEW_CONF_DIR/wvalarm.conf-no-alarms
             echo "done."
@@ -1575,7 +1580,7 @@
     fi
     echo "--Note-- Only needed if you plan to ftp your generated files to a remote web server or ISP account..."
     echo "Run the ftp daemon wviewftpd?"
-    echo -n "($DEFANS): "
+    $NONEWLINE_ECHO "($DEFANS): "
     read INVAL
     echo ""
     if [ "" != "$INVAL" ]; then
@@ -1584,7 +1589,7 @@
     
     if [ "1" = "$DEFANS" ]; then
         USEFTP=1
-        echo -n "Enabling the ftp daemon by naming the config file $WVIEW_CONF_DIR/wviewftp.conf: "
+        $NONEWLINE_ECHO "Enabling the ftp daemon by naming the config file $WVIEW_CONF_DIR/wviewftp.conf: "
         if [ -f $WVIEW_CONF_DIR/wviewftp.conf ]; then
             echo "file already exists - wviewftpd is enabled"
             echo "Remember to edit $WVIEW_CONF_DIR/wviewftp.conf to define files to transfer..."
@@ -1599,7 +1604,7 @@
         fi
     else
         USEFTP=0
-        echo -n "Disabling the ftp daemon by naming the config file $WVIEW_CONF_DIR/wviewftp.conf-no-ftp: "
+        $NONEWLINE_ECHO "Disabling the ftp daemon by naming the config file $WVIEW_CONF_DIR/wviewftp.conf-no-ftp: "
         if [ -f $WVIEW_CONF_DIR/wviewftp.conf ]; then
             mv $WVIEW_CONF_DIR/wviewftp.conf $WVIEW_CONF_DIR/wviewftp.conf-no-ftp
             echo "done."
@@ -1623,7 +1628,7 @@
         fi
         echo "--Note-- Only needed if you plan to synchronize your generated files with a remote web server or ISP account..."
         echo "Run the rsync/ssh daemon wviewsshd?"
-        echo -n "($DEFANS): "
+        $NONEWLINE_ECHO "($DEFANS): "
         read INVAL
         echo ""
         if [ "" != "$INVAL" ]; then
@@ -1631,7 +1636,7 @@
         fi
     
         if [ "1" = "$DEFANS" ]; then
-            echo -n "Enabling the rsync/ssh daemon by naming the config file $WVIEW_CONF_DIR/wviewssh.conf: "
+            $NONEWLINE_ECHO "Enabling the rsync/ssh daemon by naming the config file $WVIEW_CONF_DIR/wviewssh.conf: "
             if [ -f $WVIEW_CONF_DIR/wviewssh.conf ]; then
                 echo "file already exists - wviewsshd is enabled"
                 echo "Remember to edit $WVIEW_CONF_DIR/wviewssh.conf to define sync directories..."
@@ -1645,7 +1650,7 @@
                 echo "rsync/ssh will be disabled until you copy the config files from the distro and re-run this script..."
             fi
         else
-            echo -n "Disabling the rsync/ssh daemon by naming the config file $WVIEW_CONF_DIR/wviewssh.conf-no-ssh: "
+            $NONEWLINE_ECHO "Disabling the rsync/ssh daemon by naming the config file $WVIEW_CONF_DIR/wviewssh.conf-no-ssh: "
             if [ -f $WVIEW_CONF_DIR/wviewssh.conf ]; then
                 mv $WVIEW_CONF_DIR/wviewssh.conf $WVIEW_CONF_DIR/wviewssh.conf-no-ssh
                 echo "done."
@@ -1671,7 +1676,7 @@
         fi
         echo "--Note-- This will require 35 KB or more of memory..."
         echo "Enable the Vantage Pro Forecast icon support and Forecast Rule text support?"
-        echo -n "($DEFANS): "
+        $NONEWLINE_ECHO "($DEFANS): "
         read INVAL
         echo ""
         if [ "" != "$INVAL" ]; then
@@ -1679,7 +1684,7 @@
         fi
         
         if [ "1" = "$DEFANS" ]; then
-            echo -n "Enabling the forecast processing by naming the config file $WVIEW_CONF_DIR/forecast.conf: "
+            $NONEWLINE_ECHO "Enabling the forecast processing by naming the config file $WVIEW_CONF_DIR/forecast.conf: "
             if [ -f $WVIEW_CONF_DIR/forecast.conf ]; then
                 echo "file already exists - forecast processing is enabled"
                 echo "Remember to edit $WVIEW_CONF_DIR/forecast.conf to define icons and text..."
@@ -1693,7 +1698,7 @@
                 echo "forecast processing will be disabled until you copy the config files from the distro and re-run this script..."
             fi
         else
-            echo -n "Disabling the forecast processing by naming the config file $WVIEW_CONF_DIR/forecast.conf-no-forecast: "
+            $NONEWLINE_ECHO "Disabling the forecast processing by naming the config file $WVIEW_CONF_DIR/forecast.conf-no-forecast: "
             if [ -f $WVIEW_CONF_DIR/forecast.conf ]; then
                 mv $WVIEW_CONF_DIR/forecast.conf $WVIEW_CONF_DIR/forecast.conf-no-forecast
                 echo "done."
