diff -ruN gcc-11.2.0-orig/gcc/config/darwin.h gcc-11.2.0/gcc/config/darwin.h
--- gcc-11.2.0-orig/gcc/config/darwin.h	2021-07-28 01:55:07.000000000 -0500
+++ gcc-11.2.0/gcc/config/darwin.h	2021-08-07 12:05:25.000000000 -0500
@@ -1081,9 +1081,8 @@
   darwin_driver_init (&decoded_options_count, &decoded_options)
 #endif
 
-/* The Apple assembler and linker do not support constructor priorities.  */
-#undef SUPPORTS_INIT_PRIORITY
-#define SUPPORTS_INIT_PRIORITY 0
+/* While the Apple assembler and linker do not support constructor priorities,
+   intra-module priority support is available through sort_cdtor_records. */
 
 #undef STACK_CHECK_STATIC_BUILTIN
 #define STACK_CHECK_STATIC_BUILTIN 1
diff -ruN gcc-11.2.0-orig/gcc/testsuite/g++.dg/special/conpr-3.C gcc-11.2.0/gcc/testsuite/g++.dg/special/conpr-3.C
--- gcc-11.2.0-orig/gcc/testsuite/g++.dg/special/conpr-3.C	2021-07-28 01:55:07.000000000 -0500
+++ gcc-11.2.0/gcc/testsuite/g++.dg/special/conpr-3.C	2021-08-07 12:05:25.000000000 -0500
@@ -1,5 +1,6 @@
 /* { dg-do run { target init_priority } } */
 /* { dg-additional-sources "conpr-3a.cc conpr-3b.cc" } */
+/* { dg-shouldfail "No inter-module priority support" { *-apple-darwin* } } */
 
 #include <stdlib.h>
 
