diff -ruN gcc-10.1.0-orig/Makefile.in gcc-10.1.0/Makefile.in
--- gcc-10.1.0-orig/Makefile.in	2020-05-07 05:49:58.000000000 -0500
+++ gcc-10.1.0/Makefile.in	2020-05-09 13:27:34.000000000 -0500
@@ -234,6 +234,7 @@
 @endif gcc-bootstrap
 	$(RPATH_ENVVAR)=`echo "$(HOST_LIB_PATH)$$$(RPATH_ENVVAR)" | sed 's,::*,:,g;s,^:*,,;s,:*$$,,'`; export $(RPATH_ENVVAR);
 
+
 POSTSTAGE1_CXX_EXPORT = \
 	CXX='$(CXX)'; export CXX; \
 	CXX_FOR_BUILD='$(CXX_FOR_BUILD)'; export CXX_FOR_BUILD;
@@ -248,11 +249,13 @@
 	  `if $(LEAN); then echo ' -isystem '; else echo ' -I'; fi`$$r/prev-$(TARGET_SUBDIR)/libstdc++-v3/include \
 	  `if $(LEAN); then echo ' -isystem '; else echo ' -I'; fi`$$s/libstdc++-v3/libsupc++ \
 	  -L$$r/prev-$(TARGET_SUBDIR)/libstdc++-v3/src/.libs \
-	  -L$$r/prev-$(TARGET_SUBDIR)/libstdc++-v3/libsupc++/.libs"; \
-	  export CXX; \
+	  -L$$r/prev-$(TARGET_SUBDIR)/libstdc++-v3/libsupc++/.libs \
+	  $(XGCC_FLAGS_FOR_TARGET) $$TFLAGS "; export CXX; \
 	CXX_FOR_BUILD="$$CXX"; export CXX_FOR_BUILD;
 @endif target-libstdc++-v3-bootstrap
 
+POSTSTAGE1_LDFLAGS += $(SYSROOT_CFLAGS_FOR_TARGET)
+
 # Similar, for later GCC stages.
 POSTSTAGE1_HOST_EXPORTS = \
 	$(HOST_EXPORTS) \
@@ -616,8 +619,6 @@
 CFLAGS_FOR_TARGET = @CFLAGS_FOR_TARGET@
 CXXFLAGS_FOR_TARGET = @CXXFLAGS_FOR_TARGET@
 
-LIBCFLAGS_FOR_TARGET = $(CFLAGS_FOR_TARGET)
-LIBCXXFLAGS_FOR_TARGET = $(CXXFLAGS_FOR_TARGET) -fno-implicit-templates
 LDFLAGS_FOR_TARGET = @LDFLAGS_FOR_TARGET@
 GOCFLAGS_FOR_TARGET = -O2 -g
 GDCFLAGS_FOR_TARGET = -O2 -g
@@ -627,6 +628,12 @@
 DEBUG_PREFIX_CFLAGS_FOR_TARGET = @DEBUG_PREFIX_CFLAGS_FOR_TARGET@
 
 XGCC_FLAGS_FOR_TARGET = $(FLAGS_FOR_TARGET) $(SYSROOT_CFLAGS_FOR_TARGET) $(DEBUG_PREFIX_CFLAGS_FOR_TARGET)
+LDFLAGS_FOR_TARGET += $(SYSROOT_CFLAGS_FOR_TARGET)
+CFLAGS_FOR_TARGET += $(SYSROOT_CFLAGS_FOR_TARGET)
+CXXFLAGS_FOR_TARGET += $(SYSROOT_CFLAGS_FOR_TARGET)
+CPPFLAGS_FOR_TARGET += $(SYSROOT_CFLAGS_FOR_TARGET)
+LIBCFLAGS_FOR_TARGET = $(CFLAGS_FOR_TARGET)
+LIBCXXFLAGS_FOR_TARGET = $(CXXFLAGS_FOR_TARGET) -fno-implicit-templates
 
 # ------------------------------------
 # Miscellaneous targets and flag lists
diff -ruN gcc-10.1.0-orig/gcc/Makefile.in gcc-10.1.0/gcc/Makefile.in
--- gcc-10.1.0-orig/gcc/Makefile.in	2020-05-07 05:49:59.000000000 -0500
+++ gcc-10.1.0/gcc/Makefile.in	2020-05-09 13:27:34.000000000 -0500
@@ -3968,7 +3968,7 @@
 	@echo "set HOSTCXXFLAGS \"$(CXXFLAGS)\"" >> ./site.tmp
 # TEST_ALWAYS_FLAGS are flags that should be passed to every compilation.
 # They are passed first to allow individual tests to override them.
-	@echo "set TEST_ALWAYS_FLAGS \"$(SYSROOT_CFLAGS_FOR_TARGET)\"" >> ./site.tmp
+	@echo "set TEST_ALWAYS_FLAGS \"\"" >> ./site.tmp
 # When running the tests we set GCC_EXEC_PREFIX to the install tree so that
 # files that have already been installed there will be found.  The -B option
 # overrides it, so use of GCC_EXEC_PREFIX will not result in using GCC files
diff -ruN gcc-10.1.0-orig/gcc/ada/gcc-interface/Makefile.in gcc-10.1.0/gcc/ada/gcc-interface/Makefile.in
--- gcc-10.1.0-orig/gcc/ada/gcc-interface/Makefile.in	2020-05-07 05:49:59.000000000 -0500
+++ gcc-10.1.0/gcc/ada/gcc-interface/Makefile.in	2020-05-09 13:27:34.000000000 -0500
@@ -376,6 +376,7 @@
 	"GNATLINK=$(GNATLINK)"	\
 	"GNATBIND=$(GNATBIND)"
 
+GCC_LINK_FLAGS += $(SYSROOT_CFLAGS_FOR_TARGET)
 GCC_LINK=$(CXX) $(GCC_LINK_FLAGS) $(LDFLAGS)
 
 # Build directory for the tools. Let's copy the target-dependent
