diff -uNr gcc-5.4.0.orig/gcc/auto-profile.c gcc-5.4.0/gcc/auto-profile.c
--- gcc-5.4.0.orig/gcc/auto-profile.c	2015-01-17 21:25:42.000000000 -0500
+++ gcc-5.4.0/gcc/auto-profile.c	2017-08-20 10:18:01.000000000 -0400
@@ -19,9 +19,10 @@
 <http://www.gnu.org/licenses/>.  */
 
 #include "config.h"
+#define INCLUDE_MAP
+#define INCLUDE_SET
 #include "system.h"
 
-#include <string.h>
 #include <map>
 #include <set>
 
diff -uNr gcc-5.4.0.orig/gcc/c/c-objc-common.c gcc-5.4.0/gcc/c/c-objc-common.c
--- gcc-5.4.0.orig/gcc/c/c-objc-common.c	2015-01-09 15:18:42.000000000 -0500
+++ gcc-5.4.0/gcc/c/c-objc-common.c	2017-08-20 10:15:22.000000000 -0400
@@ -38,8 +38,6 @@
 #include "langhooks.h"
 #include "c-objc-common.h"
 
-#include <new>                          // For placement new.
-
 static bool c_tree_printer (pretty_printer *, text_info *, const char *,
 			    int, bool, bool, bool);
 
diff -uNr gcc-5.4.0.orig/gcc/config/sh/sh.c gcc-5.4.0/gcc/config/sh/sh.c
--- gcc-5.4.0.orig/gcc/config/sh/sh.c	2015-10-03 11:20:58.000000000 -0400
+++ gcc-5.4.0/gcc/config/sh/sh.c	2017-08-20 10:18:07.000000000 -0400
@@ -20,9 +20,9 @@
 <http://www.gnu.org/licenses/>.  */
 
 #include <sstream>
-#include <vector>
 
 #include "config.h"
+#define INCLUDE_VECTOR
 #include "system.h"
 #include "coretypes.h"
 #include "tm.h"
diff -uNr gcc-5.4.0.orig/gcc/config/sh/sh_treg_combine.cc gcc-5.4.0/gcc/config/sh/sh_treg_combine.cc
--- gcc-5.4.0.orig/gcc/config/sh/sh_treg_combine.cc	2015-01-15 08:28:42.000000000 -0500
+++ gcc-5.4.0/gcc/config/sh/sh_treg_combine.cc	2017-08-20 10:18:07.000000000 -0400
@@ -19,6 +19,9 @@
 <http://www.gnu.org/licenses/>.  */
 
 #include "config.h"
+#define INCLUDE_ALGORITHM
+#define INCLUDE_LIST
+#define INCLUDE_VECTOR
 #include "system.h"
 #include "coretypes.h"
 #include "machmode.h"
@@ -65,10 +68,6 @@
 #include "stmt.h"
 #include "expr.h"
 
-#include <algorithm>
-#include <list>
-#include <vector>
-
 /*
 This pass tries to optimize for example this:
 	mov.l	@(4,r4),r1
diff -uNr gcc-5.4.0.orig/gcc/cp/error.c gcc-5.4.0/gcc/cp/error.c
--- gcc-5.4.0.orig/gcc/cp/error.c	2015-08-12 14:05:49.000000000 -0400
+++ gcc-5.4.0/gcc/cp/error.c	2017-08-20 10:15:22.000000000 -0400
@@ -44,8 +44,6 @@
 #include "ubsan.h"
 #include "internal-fn.h"
 
-#include <new>                    // For placement-new.
-
 #define pp_separate_with_comma(PP) pp_cxx_separate_with (PP, ',')
 #define pp_separate_with_semicolon(PP) pp_cxx_separate_with (PP, ';')
 
diff -uNr gcc-5.4.0.orig/gcc/diagnostic.c gcc-5.4.0/gcc/diagnostic.c
--- gcc-5.4.0.orig/gcc/diagnostic.c	2015-11-22 14:23:39.000000000 -0500
+++ gcc-5.4.0/gcc/diagnostic.c	2017-08-20 10:15:22.000000000 -0400
@@ -41,8 +41,6 @@
 # include <sys/ioctl.h>
 #endif
 
-#include <new>                     // For placement new.
-
 #define pedantic_warning_kind(DC)			\
   ((DC)->pedantic_errors ? DK_ERROR : DK_WARNING)
 #define permissive_error_kind(DC) ((DC)->permissive ? DK_WARNING : DK_ERROR)
diff -uNr gcc-5.4.0.orig/gcc/fortran/error.c gcc-5.4.0/gcc/fortran/error.c
--- gcc-5.4.0.orig/gcc/fortran/error.c	2016-02-24 01:45:41.000000000 -0500
+++ gcc-5.4.0/gcc/fortran/error.c	2017-08-20 10:15:22.000000000 -0400
@@ -34,8 +34,6 @@
 #include "diagnostic-color.h"
 #include "tree-diagnostic.h" /* tree_diagnostics_defaults */
 
-#include <new> /* For placement-new */
-
 static int suppress_errors = 0;
 
 static bool warnings_not_errors = false;
diff -uNr gcc-5.4.0.orig/gcc/fortran/trans-common.c gcc-5.4.0/gcc/fortran/trans-common.c
--- gcc-5.4.0.orig/gcc/fortran/trans-common.c	2015-01-31 19:29:54.000000000 -0500
+++ gcc-5.4.0/gcc/fortran/trans-common.c	2017-08-20 10:19:56.000000000 -0400
@@ -92,8 +92,8 @@
    is examined for still-unused equivalence conditions.  We create a
    block for each merged equivalence list.  */
 
-#include <map>
 #include "config.h"
+#define INCLUDE_MAP
 #include "system.h"
 #include "coretypes.h"
 #include "tm.h"
diff -uNr gcc-5.4.0.orig/gcc/genmatch.c gcc-5.4.0/gcc/genmatch.c
--- gcc-5.4.0.orig/gcc/genmatch.c	2015-07-21 10:04:12.000000000 -0400
+++ gcc-5.4.0/gcc/genmatch.c	2017-08-20 10:15:22.000000000 -0400
@@ -22,7 +22,6 @@
 <http://www.gnu.org/licenses/>.  */
 
 #include "bconfig.h"
-#include <new>
 #include "system.h"
 #include "coretypes.h"
 #include "ggc.h"
diff -uNr gcc-5.4.0.orig/gcc/graphite-isl-ast-to-gimple.c gcc-5.4.0/gcc/graphite-isl-ast-to-gimple.c
--- gcc-5.4.0.orig/gcc/graphite-isl-ast-to-gimple.c	2015-10-12 06:59:21.000000000 -0400
+++ gcc-5.4.0/gcc/graphite-isl-ast-to-gimple.c	2017-08-20 10:20:39.000000000 -0400
@@ -38,6 +38,7 @@
 #endif
 #endif
 
+#define INCLUDE_MAP
 #include "system.h"
 #include "coretypes.h"
 #include "hash-set.h"
@@ -75,7 +76,6 @@
 #include "tree-scalar-evolution.h"
 #include "gimple-ssa.h"
 #include "tree-into-ssa.h"
-#include <map>
 
 #ifdef HAVE_isl
 #include "graphite-poly.h"
diff -uNr gcc-5.4.0.orig/gcc/ipa-icf-gimple.c gcc-5.4.0/gcc/ipa-icf-gimple.c
--- gcc-5.4.0.orig/gcc/ipa-icf-gimple.c	2015-04-15 08:09:56.000000000 -0400
+++ gcc-5.4.0/gcc/ipa-icf-gimple.c	2017-08-20 12:54:37.000000000 -0400
@@ -20,6 +20,7 @@
 <http://www.gnu.org/licenses/>.  */
 
 #include "config.h"
+#define INCLUDE_LIST
 #include "system.h"
 #include "coretypes.h"
 #include "hash-set.h"
@@ -74,7 +75,6 @@
 #include "cgraph.h"
 #include "data-streamer.h"
 #include "ipa-utils.h"
-#include <list>
 #include "tree-ssanames.h"
 #include "tree-eh.h"
 #include "builtins.h"
diff -uNr gcc-5.4.0.orig/gcc/ipa-icf.c gcc-5.4.0/gcc/ipa-icf.c
--- gcc-5.4.0.orig/gcc/ipa-icf.c	2016-03-22 07:30:49.000000000 -0400
+++ gcc-5.4.0/gcc/ipa-icf.c	2017-08-20 10:21:36.000000000 -0400
@@ -52,8 +52,8 @@
 */
 
 #include "config.h"
+#define INCLUDE_LIST
 #include "system.h"
-#include <list>
 #include "coretypes.h"
 #include "hash-set.h"
 #include "machmode.h"
diff -uNr gcc-5.4.0.orig/gcc/pretty-print.c gcc-5.4.0/gcc/pretty-print.c
--- gcc-5.4.0.orig/gcc/pretty-print.c	2015-01-05 07:33:28.000000000 -0500
+++ gcc-5.4.0/gcc/pretty-print.c	2017-08-20 10:15:22.000000000 -0400
@@ -25,8 +25,6 @@
 #include "pretty-print.h"
 #include "diagnostic-color.h"
 
-#include <new>                    // For placement-new.
-
 #if HAVE_ICONV
 #include <iconv.h>
 #endif
diff -uNr gcc-5.4.0.orig/gcc/system.h gcc-5.4.0/gcc/system.h
--- gcc-5.4.0.orig/gcc/system.h	2015-01-05 07:33:28.000000000 -0500
+++ gcc-5.4.0/gcc/system.h	2017-08-20 12:09:00.000000000 -0400
@@ -219,6 +219,18 @@
 # include <cstring>
 # include <utility>
 #endif
+#ifdef INCLUDE_LIST
+# include <list>
+#endif
+#ifdef INCLUDE_MAP
+# include <map>
+#endif
+#ifdef INCLUDE_SET
+# include <set>
+#endif
+#ifdef INCLUDE_VECTOR
+# include <vector>
+#endif
 
 /* Some of glibc's string inlines cause warnings.  Plus we'd rather
    rely on (and therefore test) GCC's string builtins.  */
diff -uNr gcc-5.4.0.orig/gcc/toplev.c gcc-5.4.0/gcc/toplev.c
--- gcc-5.4.0.orig/gcc/toplev.c	2015-07-10 08:33:28.000000000 -0400
+++ gcc-5.4.0/gcc/toplev.c	2017-08-20 10:15:22.000000000 -0400
@@ -135,8 +135,6 @@
 #define HAVE_prologue 0
 #endif
 
-#include <new>
-
 static void general_init (const char *, bool);
 static void do_compile ();
 static void process_options (void);
