diff -Nurd icon-9.5.20i.orig/Makefile icon-9.5.20i/Makefile
--- icon-9.5.20i.orig/Makefile	2020-08-13 12:50:45.000000000 -0400
+++ icon-9.5.20i/Makefile	2021-06-18 09:28:37.000000000 -0400
@@ -85,16 +85,14 @@
 
 D=$(dest)
 Install:
-		mkdir $D
-		mkdir $D/bin $D/lib $D/doc $D/man $D/man/man1
-		cp README $D
+		mkdir -p $D/bin $D/lib/icon $D/share/doc/icon $D/share/man/man1
 		cp bin/[cflpvwx]* $D/bin
 		cp bin/icon[tx]* $D/bin
 		rm -f $D/bin/libI*
 		(cd $D/bin; ln -s icont icon)
-		cp lib/*.* $D/lib
-		cp doc/*.* $D/doc
-		cp man/man1/*.* $D/man/man1
+		cp lib/*.* $D/lib/icon
+		cp doc/*.* $D/share/doc/icon
+		cp man/man1/*.* $D/share/man/man1
 
 
 # Bundle up for binary distribution.
diff -Nurd icon-9.5.20i.orig/config/mac386/Makedefs icon-9.5.20i/config/mac386/Makedefs
--- icon-9.5.20i.orig/config/mac386/Makedefs	2020-08-13 12:50:45.000000000 -0400
+++ icon-9.5.20i/config/mac386/Makedefs	2021-06-18 09:46:41.000000000 -0400
@@ -10,7 +10,7 @@
 #  SFLAGS   flags for stripping iconx
 
 CC = cc -arch i386
-CFLAGS = -I /opt/X11/include
+CFLAGS = -I/opt/X11/include
 CFDYN =
 RLINK = -dynamic
 RLIBS = -lm
diff -Nurd icon-9.5.20i.orig/config/macintosh/Makedefs icon-9.5.20i/config/macintosh/Makedefs
--- icon-9.5.20i.orig/config/macintosh/Makedefs	2020-08-13 12:50:45.000000000 -0400
+++ icon-9.5.20i/config/macintosh/Makedefs	2021-06-18 09:46:38.000000000 -0400
@@ -10,7 +10,7 @@
 #  SFLAGS   flags for stripping iconx
 
 CC = cc
-CFLAGS = -O -I /opt/X11/include
+CFLAGS = -O -I/opt/X11/include
 CFDYN =
 RLINK = -dynamic
 RLIBS = -lm
diff -Nurd icon-9.5.20i.orig/config/setup.sh icon-9.5.20i/config/setup.sh
--- icon-9.5.20i.orig/config/setup.sh	2020-08-13 12:50:45.000000000 -0400
+++ icon-9.5.20i/config/setup.sh	2021-06-18 09:49:47.000000000 -0400
@@ -11,7 +11,7 @@
 
 # check parameters
 case "$GPX" in
-   Graphics)	XL='-L../../bin -lIgpx $(XLIBS)';;
+   Graphics)	XL='../../bin/libIgpx.a $(XLIBS)';;
    NoGraphics)	XL= ;;
    *)		echo "$USAGE" 1>&2; exit 1;;
 esac
diff -Nurd icon-9.5.20i.orig/src/icont/tunix.c icon-9.5.20i/src/icont/tunix.c
--- icon-9.5.20i.orig/src/icont/tunix.c	2020-08-13 12:50:45.000000000 -0400
+++ icon-9.5.20i/src/icont/tunix.c	2021-06-18 09:24:51.000000000 -0400
@@ -300,7 +300,7 @@
    else
       strcpy(buf, ".");
    strcat(buf, ":");
-   strcat(buf, relfile(prog, "/../../lib"));
+   strcat(buf, relfile(prog, "/../../lib/icon"));
    return salloc(buf);
    }
 
