diff -ru icon.v942src-orig/Makefile icon.v942src/Makefile
--- icon.v942src-orig/Makefile	Sun Nov  2 15:47:32 2003
+++ icon.v942src/Makefile	Sun Nov  2 16:09:12 2003
@@ -97,16 +97,13 @@
 
 D=$(dest)
 Install:
-		mkdir $D
-		mkdir $D/bin $D/lib $D/doc $D/man $D/man/man1
-		cp README $D
+		mkdir -p $D/bin $D/lib/icon $D/share/man/man1
 		cp bin/[cflpvwx]* $D/bin
 		cp bin/icon[tx]* $D/bin
 		rm -f $D/bin/libI*
 		(cd $D/bin; ln -s icont icon)
-		cp lib/*.* $D/lib
-		cp doc/*.* $D/doc
-		cp man/man1/*.* $D/man/man1
+		cp lib/*.* $D/lib/icon
+		cp man/man1/*.* $D/share/man/man1
 
 
 # Bundle up for binary distribution.
Only in icon.v942src: Makefile.orig
Only in icon.v942src: Makefile.rej
Only in icon.v942src: Makefile~
diff -ru icon.v942src-orig/src/icont/tunix.c icon.v942src/src/icont/tunix.c
--- icon.v942src-orig/src/icont/tunix.c	Sun Nov  2 15:47:32 2003
+++ icon.v942src/src/icont/tunix.c	Sun Nov  2 15:47:58 2003
@@ -301,7 +301,7 @@
    else
       strcpy(buf, ".");
    strcat(buf, ":");
-   strcat(buf, relfile(prog, "/../../lib"));
+   strcat(buf, relfile(prog, "/../../lib/icon"));
    return salloc(buf);
    }
 
Only in icon.v942src/src/icont: tunix.c.orig
