diff --git a/docs/source/conf.py b/docs/source/conf.py
--- a/docs/source/conf.py       2018-07-29 02:24:17.000000000 +0200
+++ b/docs/source/conf.py       2019-05-31 00:15:00.000000000 +0200
@@ -201,12 +201,11 @@
 # Output file base name for HTML help builder.
 htmlhelp_basename = 'ipythondoc'
 
-intersphinx_mapping = {'python': ('https://docs.python.org/3/', None),
-                       'rpy2': ('https://rpy2.readthedocs.io/en/version_2.8.x/', None),
-                       'traitlets': ('https://traitlets.readthedocs.io/en/latest/', None),
-                       'jupyterclient': ('https://jupyter-client.readthedocs.io/en/latest/', None),
-                       'ipyparallel': ('https://ipyparallel.readthedocs.io/en/latest/', None),
-                       'jupyter': ('https://jupyter.readthedocs.io/en/latest/', None),
+intersphinx_mapping = {'ipywidgets': ('@PREFIX@/share/doc/ipywidgets-py@PY_PKG@/html', None),
+                       'jupyter': ('@PREFIX@/share/doc/jupyter-core-py@PY_PKG@/html', None),
+                       'jupyterclient': ('@PREFIX@/share/doc/jupyter-client-py@PY_PKG@/html', None),
+                       'nbformat': ('@PREFIX@/share/doc/nbformat-py@PY_PKG@/html', None),
+                       'nbconvert': ('@PREFIX@/share/doc/nbconvert-py@PY_PKG@/html', None),
                       }
 
 # Options for LaTeX output
