diff -Nurd mujs-1.1.2.orig/Makefile mujs-1.1.2/Makefile
--- mujs-1.1.2.orig/Makefile	2021-04-23 05:13:50.000000000 -0400
+++ mujs-1.1.2/Makefile	2021-06-01 14:53:19.000000000 -0400
@@ -8,6 +8,7 @@
 bindir ?= $(prefix)/bin
 incdir ?= $(prefix)/include
 libdir ?= $(prefix)/lib
+install_name ?= $(libdir)/libmujs.1.dylib
 
 ifeq "$(wildcard .git)" ".git"
   VERSION := $(shell git describe --tags --always)
@@ -17,15 +18,13 @@
 
 # Compiler flags for various configurations:
 
-CFLAGS := -std=c99 -pedantic -Wall -Wextra -Wno-unused-parameter
+CFLAGS := -std=c99 -pedantic -Wall -Wextra -Wno-unused-parameter -MD
 
 ifeq "$(CC)" "clang"
   CFLAGS += -Wunreachable-code
 endif
 
-ifeq "$(shell uname)" "Linux"
-  HAVE_READLINE := yes
-endif
+HAVE_READLINE := yes
 
 ifeq "$(build)" "debug"
   CFLAGS += -g
@@ -55,7 +54,7 @@
 default: shell
 shell: $(OUT)/mujs $(OUT)/mujs-pp
 static: $(OUT)/libmujs.a
-shared: $(OUT)/libmujs.so
+shared: $(OUT)/libmujs.dylib
 
 astnames.h: jsparse.h
 	grep -E '(AST|EXP|STM)_' jsparse.h | sed 's/^[^A-Z]*\(AST_\)*/"/;s/,.*/",/' | tr A-Z a-z > $@
@@ -80,15 +79,15 @@
 	@ mkdir -p $(dir $@)
 	$(AR) cr $@ $^
 
-$(OUT)/libmujs.so: one.c $(HDRS)
+$(OUT)/libmujs.dylib: one.c $(HDRS)
 	@ mkdir -p $(dir $@)
-	$(CC) $(CFLAGS) $(CPPFLAGS) -fPIC -shared $(LDFLAGS) -o $@ $< -lm
+	$(CC) $(CFLAGS) $(CPPFLAGS) -fPIC -dynamiclib $(LDFLAGS) -o $@ $< -lm -install_name $(install_name) -compatibility_version 2.0.0
 
-$(OUT)/mujs: $(OUT)/libmujs.o $(OUT)/main.o
+$(OUT)/mujs: $(OUT)/libmujs.dylib $(OUT)/main.o
 	@ mkdir -p $(dir $@)
 	$(CC) $(LDFLAGS) -o $@ $^ $(LIBREADLINE) -lm
 
-$(OUT)/mujs-pp: $(OUT)/libmujs.o $(OUT)/pp.o
+$(OUT)/mujs-pp: $(OUT)/libmujs.dylib $(OUT)/pp.o
 	@ mkdir -p $(dir $@)
 	$(CC) $(LDFLAGS) -o $@ $^ -lm
 
@@ -117,8 +116,9 @@
 install-static: install-common $(OUT)/libmujs.a
 	install -m 644 $(OUT)/libmujs.a $(DESTDIR)$(libdir)
 
-install-shared: install-common $(OUT)/libmujs.so
-	install -m 755 $(OUT)/libmujs.so $(DESTDIR)$(libdir)
+install-shared: install-common $(OUT)/libmujs.dylib
+	install -m 755 $(OUT)/libmujs.dylib $(DESTDIR)$(install_name)
+	ln -s $(install_name) $(DESTDIR)/$(libdir)/libmujs.dylib
 
 install: install-static
 
