diff -ruN llvm-project-15.0.7.src-orig/clang/tools/libclang/CMakeLists.txt llvm-project-15.0.7.src/clang/tools/libclang/CMakeLists.txt
--- llvm-project-15.0.7.src-orig/clang/tools/libclang/CMakeLists.txt	2023-01-12 01:12:30.000000000 -0600
+++ llvm-project-15.0.7.src/clang/tools/libclang/CMakeLists.txt	2025-01-17 04:24:15.000000000 -0600
@@ -194,7 +194,7 @@
                  OBJECT_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/libclang.map)
 
     set_target_properties(libclang PROPERTIES
-                          VERSION ${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}${LLVM_VERSION_SUFFIX}
+                          VERSION ${LLVM_VERSION_MAJOR}
                           SOVERSION ${LIBCLANG_SOVERSION})
   endif()
 endif()
