--- cfe-3.5.0.src/lib/Driver/Tools.cpp.orig	2014-10-24 19:10:30.000000000 -0400
+++ cfe-3.5.0.src/lib/Driver/Tools.cpp	2014-10-24 19:16:13.000000000 -0400
@@ -5924,6 +5924,8 @@
 
   if (Args.hasArg(options::OPT_fopenmp)){
     // This is more complicated in gcc...
+    // Help clang find libiomp5.dylib
+    CmdArgs.push_back("-L@FINK_PREFIX@/opt/llvm-3.5/lib");
     CmdArgs.push_back("-liomp5");
 
     if (Args.hasArg(options::OPT_omptargets_EQ))
@@ -7670,6 +7672,8 @@
 
       bool OpenMP = Args.hasArg(options::OPT_fopenmp);
       if (OpenMP) {
+	// Help clang find libiomp5.dylib
+	CmdArgs.push_back("-L@FINK_PREFIX@/opt/llvm-3.5/lib");
         CmdArgs.push_back("-liomp5");
         if (Args.hasArg(options::OPT_omptargets_EQ))
           CmdArgs.push_back("-lomptarget");
