diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index ace15d7..f8a131a 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -64,6 +64,16 @@ add_polly_library(Polly
   ${POLLY_PLUTO_FILES}
   )
 
+if (APPLE)
+# apparently, this is not wanted on linux
+target_link_libraries(Polly
+  LLVMSupport
+  LLVMScalarOpts
+  LLVMTransformUtils
+  LLVMipo
+)
+endif()
+
 # Build a monolithic Polly.a and a thin module LLVMPolly.moduleext that links to
 # that static library.
 add_polly_loadable_module(LLVMPolly
