diff -uNr cfe-6.0.0.src.orig/lib/Driver/ToolChains/Clang.cpp cfe-6.0.0.src/lib/Driver/ToolChains/Clang.cpp
--- cfe-6.0.0.src.orig/lib/Driver/ToolChains/Clang.cpp	2017-08-04 06:46:52.000000000 -0400
+++ cfe-6.0.0.src/lib/Driver/ToolChains/Clang.cpp	2017-08-04 06:58:00.000000000 -0400
@@ -3286,6 +3286,8 @@
     case Driver::OMPRT_OMP:
     case Driver::OMPRT_IOMP5:
       // Clang can generate useful OpenMP code for these two runtime libraries.
+      // Automatically find omp.h from libomp-dev
+      CmdArgs.push_back("-I@FINK_PREFIX@/include/libomp");
       CmdArgs.push_back("-fopenmp");
 
       // If no option regarding the use of TLS in OpenMP codegeneration is
diff -uNr cfe-6.0.0.src.orig/lib/Driver/ToolChains/CommonArgs.cpp cfe-6.0.0.src/lib/Driver/ToolChains/CommonArgs.cpp
--- cfe-6.0.0.src.orig/lib/Driver/ToolChains/CommonArgs.cpp	2017-07-25 18:58:40.000000000 -0400
+++ cfe-6.0.0.src/lib/Driver/ToolChains/CommonArgs.cpp	2017-08-04 06:58:00.000000000 -0400
@@ -460,15 +460,21 @@
 
   switch (TC.getDriver().getOpenMPRuntime(Args)) {
   case Driver::OMPRT_OMP:
+    // Help clang find libomp.dylib
+    CmdArgs.push_back("-L@FINK_PREFIX@/lib/libomp");
     CmdArgs.push_back("-lomp");
     break;
   case Driver::OMPRT_GOMP:
+    // Help clang find libgomp.dylib
+    CmdArgs.push_back("-L@FINK_PREFIX@/lib/libomp");
     CmdArgs.push_back("-lgomp");
 
     if (GompNeedsRT)
       CmdArgs.push_back("-lrt");
     break;
   case Driver::OMPRT_IOMP5:
+    // Help clang find libiomp5.dylib
+    CmdArgs.push_back("-L@FINK_PREFIX@/lib/libomp");
     CmdArgs.push_back("-liomp5");
     break;
   case Driver::OMPRT_Unknown:
