diff -ruN perl-5.34.1-orig/t/porting/libperl.t perl-5.34.1/t/porting/libperl.t
--- perl-5.34.1-orig/t/porting/libperl.t	2022-02-19 06:16:51.000000000 -0600
+++ perl-5.34.1/t/porting/libperl.t	2024-11-04 04:46:45.000000000 -0600
@@ -341,7 +341,7 @@
     $symbols{data}{common} = $symbols{data}{bss};
 }
 
-ok($symbols{data}{common}{PL_hash_seed_w}{'globals.o'}, "has PL_hash_seed_w");
+ok(!$symbols{data}{common}{PL_hash_seed_w}{'globals.o'}, "has PL_hash_seed_w");
 ok($symbols{data}{data}{PL_ppaddr}{'globals.o'}, "has PL_ppaddr");
 
 # See the comments in the beginning for what "undefined symbols"
From b1db17763c817dd84f06c0147a2bc60fe1bfe639 Mon Sep 17 00:00:00 2001
From: Tony Cook <tony@develop-help.com>
Date: Wed, 24 May 2023 13:22:38 +1000
Subject: [PATCH] porting/libperl.t: handle nm output on Mac OS Ventura 13.4

Fixes #21117
---
 t/porting/libperl.t | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/t/porting/libperl.t b/t/porting/libperl.t
index 68da29b7cf67..6d05a2be007e 100644
--- a/t/porting/libperl.t
+++ b/t/porting/libperl.t
@@ -245,7 +245,8 @@ sub nm_parse_gnu {
 sub nm_parse_darwin {
     my $symbols = shift;
     my $line = $_;
-    if (m{^(?:.+)?libperl\.a\((\w+\.o)\):$}) {
+    if (m{^(?:.+)?libperl\.a\((\w+\.o)\):$} ||
+        m{^(\w+\.o):$}) {
         # object file name
         $symbols->{obj}{$1}++;
         $symbols->{o} = $1;
