diff -ruN perl-5.28.2-orig/t/porting/libperl.t perl-5.28.2/t/porting/libperl.t
--- perl-5.28.2-orig/t/porting/libperl.t	2018-11-01 17:07:59.000000000 -0500
+++ perl-5.28.2/t/porting/libperl.t	2020-09-30 05:43:25.000000000 -0500
@@ -437,7 +437,7 @@
         $symbols{data}{common} = $symbols{data}{bss};
     }
 
-    ok($symbols{data}{common}{PL_hash_seed}{'globals.o'}, "has PL_hash_seed");
+    ok(!$symbols{data}{common}{PL_hash_seed}{'globals.o'}, "has no PL_hash_seed");
     ok($symbols{data}{data}{PL_ppaddr}{'globals.o'}, "has PL_ppaddr");
 
     # None of the GLOBAL_STRUCT* business here.
@@ -574,9 +574,11 @@
         while (<$nm_err_fh>) {
             # OS X has weird error where nm warns about
             # "no name list" but then outputs fine.
-            if (/nm: no name list/ && $^O eq 'darwin') {
-                print "# $^O ignoring $nm output: $_";
-                next;
+             if ( $^O eq 'darwin' ) {
+                if (/nm: no name list/ || /^no symbols$/ ) {
+                    print "# $^O ignoring $nm output: $_";
+                    next;
+                }
             }
             warn "$0: Unexpected $nm error: $_";
             $error++;
