diff -ruN perl-5.30.2-orig/dist/Devel-PPPort/parts/inc/misc perl-5.30.2/dist/Devel-PPPort/parts/inc/misc
--- perl-5.30.2-orig/dist/Devel-PPPort/parts/inc/misc	2019-10-24 16:27:55
+++ perl-5.30.2/dist/Devel-PPPort/parts/inc/misc	2024-11-14 18:30:34
@@ -182,17 +182,12 @@
 
 #undef STMT_START
 #undef STMT_END
-#ifdef PERL_USE_GCC_BRACE_GROUPS
-#  define STMT_START    (void)( /* gcc supports ``({ STATEMENTS; })'' */
-#  define STMT_END      )
-#else
-#  if defined(VOIDFLAGS) && (VOIDFLAGS) && (defined(sun) || defined(__sun__)) && !defined(__GNUC__)
+#if defined(VOIDFLAGS) && (VOIDFLAGS) && (defined(sun) || defined(__sun__)) && !defined(__GNUC__)
 #    define STMT_START  if (1)
 #    define STMT_END    else (void)0
-#  else
+#else
 #    define STMT_START  do
 #    define STMT_END    while (0)
-#  endif
 #endif
 
 __UNDEFINED__  boolSV(b)    ((b) ? &PL_sv_yes : &PL_sv_no)
diff -ruN perl-5.30.2-orig/perl.h perl-5.30.2/perl.h
--- perl-5.30.2-orig/perl.h	2019-10-24 16:27:56
+++ perl-5.30.2/perl.h	2024-11-14 18:30:34
@@ -540,13 +540,8 @@
  * Trying to select a version that gives no warnings...
  */
 #if !(defined(STMT_START) && defined(STMT_END))
-# ifdef PERL_USE_GCC_BRACE_GROUPS
-#   define STMT_START	(void)(	/* gcc supports "({ STATEMENTS; })" */
-#   define STMT_END	)
-# else
 #   define STMT_START	do
 #   define STMT_END	while (0)
-# endif
 #endif
 
 #ifndef BYTEORDER  /* Should never happen -- byteorder is in config.h */
