diff -ruN perl-5.28.2-orig/cpan/Time-Local/t/Local.t perl-5.28.2/cpan/Time-Local/t/Local.t
--- perl-5.28.2-orig/cpan/Time-Local/t/Local.t	2018-11-01 17:07:38.000000000 -0500
+++ perl-5.28.2/cpan/Time-Local/t/Local.t	2020-09-29 12:02:58.000000000 -0500
@@ -96,7 +96,7 @@
 
         # Test timelocal()
         {
-            my $year_in = $year < 70 ? $year + 1900 : $year;
+            my $year_in = $year + 1900;
             my $time = timelocal( $sec, $min, $hour, $mday, $mon, $year_in );
 
             my ( $s, $m, $h, $D, $M, $Y ) = localtime($time);
@@ -111,7 +111,7 @@
 
         # Test timegm()
         {
-            my $year_in = $year < 70 ? $year + 1900 : $year;
+            my $year_in = $year + 1900;
             my $time = timegm( $sec, $min, $hour, $mday, $mon, $year_in );
 
             my ( $s, $m, $h, $D, $M, $Y ) = gmtime($time);
@@ -128,7 +128,6 @@
 
 for (@bad_time) {
     my ( $year, $mon, $mday, $hour, $min, $sec ) = @$_;
-    $year -= 1900;
     $mon--;
 
     eval { timegm( $sec, $min, $hour, $mday, $mon, $year ) };
@@ -138,19 +137,19 @@
 
 {
     is(
-        timelocal( 0, 0, 1, 1, 0, 90 ) - timelocal( 0, 0, 0, 1, 0, 90 ), 3600,
+        timelocal( 0, 0, 1, 1, 0, 1990 ) - timelocal( 0, 0, 0, 1, 0, 1990 ), 3600,
         'one hour difference between two calls to timelocal'
     );
 
     is(
-        timelocal( 1, 2, 3, 1, 0, 100 ) - timelocal( 1, 2, 3, 31, 11, 99 ),
+        timelocal( 1, 2, 3, 1, 0, 2000 ) - timelocal( 1, 2, 3, 31, 11, 1999 ),
         24 * 3600,
         'one day difference between two calls to timelocal'
     );
 
     # Diff beween Jan 1, 1980 and Mar 1, 1980 = (31 + 29 = 60 days)
     is(
-        timegm( 0, 0, 0, 1, 2, 80 ) - timegm( 0, 0, 0, 1, 0, 80 ),
+        timegm( 0, 0, 0, 1, 2, 1980 ) - timegm( 0, 0, 0, 1, 0, 1980 ),
         60 * 24 * 3600,
         '60 day difference between two calls to timegm'
     );
