diff -ruN perl-5.34.0-orig/t/porting/libperl.t perl-5.34.0/t/porting/libperl.t
--- perl-5.34.0-orig/t/porting/libperl.t	2021-02-21 05:01:31.000000000 -0600
+++ perl-5.34.0/t/porting/libperl.t	2021-08-28 03:45:46.000000000 -0500
@@ -341,7 +341,7 @@
     $symbols{data}{common} = $symbols{data}{bss};
 }
 
-ok($symbols{data}{common}{PL_hash_seed_w}{'globals.o'}, "has PL_hash_seed_w");
+ok(!$symbols{data}{common}{PL_hash_seed_w}{'globals.o'}, "has PL_hash_seed_w");
 ok($symbols{data}{data}{PL_ppaddr}{'globals.o'}, "has PL_ppaddr");
 
 # See the comments in the beginning for what "undefined symbols"
