diff -ruN Python-3.5.10-orig/configure Python-3.5.10/configure
--- Python-3.5.10-orig/configure	2020-09-05 02:22:07.000000000 -0500
+++ Python-3.5.10/configure	2020-11-14 13:19:46.000000000 -0600
@@ -3492,7 +3492,7 @@
   # has no effect, don't bother defining them
   Darwin/[6789].*)
     define_xopen_source=no;;
-  Darwin/1[0-9].*)
+  Darwin/[12][0-9].*)
     define_xopen_source=no;;
   # On AIX 4 and 5.1, mbstate_t is defined only when _XOPEN_SOURCE == 500 but
   # used in wcsnrtombs() and mbsnrtowcs() even if _XOPEN_SOURCE is not defined
@@ -7222,7 +7222,7 @@
                ARCH_RUN_32BIT=""
                ;;
             64-bit)
-               UNIVERSAL_ARCH_FLAGS="-arch ppc64 -arch x86_64"
+               UNIVERSAL_ARCH_FLAGS="-arch x86_64"
                LIPO_32BIT_FLAGS=""
                ARCH_RUN_32BIT="true"
                ;;
