diff -Nurd ruby-2.2.10.orig/internal.h ruby-2.2.10/internal.h
--- ruby-2.2.10.orig/internal.h	2017-03-25 12:31:54.000000000 -0400
+++ ruby-2.2.10/internal.h	2023-01-14 18:08:14.000000000 -0500
@@ -988,6 +988,7 @@
 #define QUOTE(str) rb_str_quote_unprintable(str)
 #define QUOTE_ID(id) rb_id_quote_unprintable(id)
 void rb_str_fill_terminator(VALUE str, const int termlen);
+void rb_str_change_terminator_length(VALUE str, const int oldtermlen, const int termlen);
 VALUE rb_str_locktmp_ensure(VALUE str, VALUE (*func)(VALUE), VALUE arg);
 #ifdef RUBY_ENCODING_H
 VALUE rb_external_str_with_enc(VALUE str, rb_encoding *eenc);
