diff -ruN ruby-3.2.0-orig/defs/gmake.mk ruby-3.2.0/defs/gmake.mk
--- ruby-3.2.0-orig/defs/gmake.mk	2022-12-25 00:55:25.000000000 -0600
+++ ruby-3.2.0/defs/gmake.mk	2023-01-28 06:10:22.000000000 -0600
@@ -295,9 +295,7 @@
 bundled-gem-srcdir = $(srcdir)/gems/src/$(1)
 bundled-gem-extracted = $(srcdir)/.bundle/gems/$(1)-$(2)
 
-update-gems: | $(patsubst %,$(srcdir)/gems/%.gem,$(bundled-gems))
-update-gems: | $(call foreach-bundled-gems-rev,bundled-gem-gemfile)
-update-gems: | $(call foreach-bundled-gems-rev,bundled-gem-srcdir)
+update-gems:
 
 test-bundler-precheck: | $(srcdir)/.bundle/cache
 
