diff -Nurd ruby-2.0.0-p648.orig/lib/mkmf.rb ruby-2.0.0-p648/lib/mkmf.rb
--- ruby-2.0.0-p648.orig/lib/mkmf.rb	2015-02-25 00:26:38.000000000 -0500
+++ ruby-2.0.0-p648/lib/mkmf.rb	2021-06-01 22:35:21.000000000 -0400
@@ -581,7 +581,7 @@
 
   def with_cppflags(flags)
     cppflags = $CPPFLAGS
-    $CPPFLAGS = flags
+    $CPPFLAGS << " " << flags
     ret = yield
   ensure
     $CPPFLAGS = cppflags unless ret
@@ -593,7 +593,7 @@
 
   def with_cflags(flags)
     cflags = $CFLAGS
-    $CFLAGS = flags
+    $CFLAGS << " " << flags
     ret = yield
   ensure
     $CFLAGS = cflags unless ret
@@ -605,7 +605,7 @@
 
   def with_ldflags(flags)
     ldflags = $LDFLAGS
-    $LDFLAGS = flags
+    $LDFLAGS << " " << flags
     ret = yield
   ensure
     $LDFLAGS = ldflags unless ret
diff -Nurd ruby-2.0.0-p648.orig/thread.c ruby-2.0.0-p648/thread.c
--- ruby-2.0.0-p648.orig/thread.c	2015-02-25 00:20:53.000000000 -0500
+++ ruby-2.0.0-p648/thread.c	2021-06-01 22:35:47.000000000 -0400
@@ -4671,6 +4671,8 @@
 /* variables for recursive traversals */
 static ID recursive_key;
 
+ID rb_frame_last_func(void);
+
 /*
  * Returns the current "recursive list" used to detect recursion.
  * This list is a hash table, unique for the current thread and for
