diff -Nurd ruby-2.6.10.orig/test/fiddle/helper.rb ruby-2.6.10/test/fiddle/helper.rb
--- ruby-2.6.10.orig/test/fiddle/helper.rb	2022-04-12 07:50:11
+++ ruby-2.6.10/test/fiddle/helper.rb	2025-10-17 00:17:00
@@ -95,6 +95,11 @@
 libc_so = nil if !libc_so || (libc_so[0] == ?/ && !File.file?(libc_so))
 libm_so = nil if !libm_so || (libm_so[0] == ?/ && !File.file?(libm_so))
 
+# macOS 11.0+ removed libSystem.B.dylib from /usr/lib. But It works with dlopen.
+if RUBY_PLATFORM =~ /darwin/
+  libc_so = libm_so = "/usr/lib/libSystem.B.dylib"
+end
+
 if !libc_so || !libm_so
   ruby = EnvUtil.rubybin
   ldd = `ldd #{ruby}`
