diff -ru sip-4.15.5.orig/configure.py sip-4.15.5/configure.py
--- sip-4.15.5.orig/configure.py	2014-03-14 10:09:34.000000000 -0400
+++ sip-4.15.5/configure.py	2014-04-04 10:51:23.000000000 -0400
@@ -247,7 +247,7 @@
     else:
         lib_dir = sysconfig.get_python_lib(plat_specific=1, standard_lib=1)
 
-        plat_py_lib_dir = lib_dir + "/config"
+        plat_py_lib_dir = sysconfig.get_config_var('LIBPL')
         plat_bin_dir = sys.exec_prefix + "/bin"
         plat_sip_dir = sys.prefix + "/share/sip"
 
diff -ru sip-4.15.5.orig/siputils.py sip-4.15.5/siputils.py
--- sip-4.15.5.orig/siputils.py	2014-02-21 05:31:35.000000000 -0500
+++ sip-4.15.5/siputils.py	2014-04-04 10:52:12.000000000 -0400
@@ -1316,7 +1316,7 @@
         if self.generator == "UNIX":
             dst = "$(DESTDIR)" + dst
 
-        mfile.write("\t@%s %s " % (self.chkdir, _quote(dst)))
+        mfile.write("\t%s %s " % (self.chkdir, _quote(dst)))
 
         if self.generator == "UNIX":
             mfile.write("|| ")
@@ -1398,14 +1398,7 @@
         mfile.write("\n" + target + ":\n")
 
         for d in self._subdirs:
-            if self.generator == "MINGW":
-                mfile.write("\t@$(MAKE) -C %s%s\n" % (d, tname))
-            elif self.generator == "UNIX":
-                mfile.write("\t@(cd %s; $(MAKE)%s)\n" % (d, tname))
-            else:
-                mfile.write("\tcd %s\n" % d)
-                mfile.write("\t$(MAKE)%s\n" % tname)
-                mfile.write("\t@cd ..\n")
+            mfile.write("\t$(MAKE) -C %s%s\n" % (d, tname))
 
 
 class PythonModuleMakefile(Makefile):
@@ -1551,10 +1544,7 @@
             # can handle extension modules that are bundles or dynamic
             # libraries, but python.org versions need bundles (unless built
             # with DYNLOADFILE=dynload_shlib.o).
-            if sys.platform == "darwin":
-                lflags_plugin = ["-bundle"]
-            else:
-                lflags_plugin = self.optional_list("LFLAGS_PLUGIN")
+            lflags_plugin = self.optional_list("LFLAGS_PLUGIN")
 
             if not lflags_plugin:
                 lflags_plugin = self.optional_list("LFLAGS_SHLIB")
@@ -1563,7 +1553,7 @@
 
         self.LFLAGS.extend(self.optional_list(lflags_console))
 
-        if sys.platform == "darwin":
+        if 0:
             self.LFLAGS.append("-undefined dynamic_lookup")
 
         Makefile.finalise(self)
