diff -uNr tcltk-8.6.10.orig/tcl8.6.10/generic/tclInt.h tcltk-8.6.10/tcl8.6.10/generic/tclInt.h
--- tcltk-8.6.10.orig/tcl8.6.10/generic/tclInt.h	2017-07-18 08:51:45.000000000 -0400
+++ tcltk-8.6.10/tcl8.6.10/generic/tclInt.h	2017-08-17 19:56:13.000000000 -0400
@@ -3277,7 +3277,7 @@
 MODULE_SCOPE int	TclClockOldscanObjCmd(
 			    ClientData clientData, Tcl_Interp *interp,
 			    int objc, Tcl_Obj *const objv[]);
-MODULE_SCOPE int	Tcl_CloseObjCmd(ClientData clientData,
+extern int	Tcl_CloseObjCmd(ClientData clientData,
 			    Tcl_Interp *interp, int objc,
 			    Tcl_Obj *const objv[]);
 MODULE_SCOPE int	Tcl_ConcatObjCmd(ClientData clientData,
@@ -3458,7 +3458,7 @@
 MODULE_SCOPE int	Tcl_RepresentationCmd(ClientData clientData,
 			    Tcl_Interp *interp, int objc,
 			    Tcl_Obj *const objv[]);
-MODULE_SCOPE int	Tcl_ReturnObjCmd(ClientData clientData,
+extern int	Tcl_ReturnObjCmd(ClientData clientData,
 			    Tcl_Interp *interp, int objc,
 			    Tcl_Obj *const objv[]);
 MODULE_SCOPE int	Tcl_ScanObjCmd(ClientData clientData,
diff -uNr tcltk-8.6.10.orig/tcl8.6.10/unix/configure tcltk-8.6.10/tcl8.6.10/unix/configure
--- tcltk-8.6.10.orig/tcl8.6.10/unix/configure	2017-08-09 10:45:58.000000000 -0400
+++ tcltk-8.6.10/tcl8.6.10/unix/configure	2017-08-17 19:56:13.000000000 -0400
@@ -18820,7 +18820,7 @@
     else
         TCL_LIB_FLAG="-ltcl`echo ${TCL_VERSION} | tr -d .`"
     fi
-    TCL_BUILD_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TCL_LIB_FLAG}"
+    TCL_BUILD_LIB_SPEC="`pwd`/${TCL_LIB_FILE}"
     TCL_LIB_SPEC="-L${libdir} ${TCL_LIB_FLAG}"
 fi
 VERSION='${VERSION}'
@@ -18864,7 +18864,7 @@
     TCL_STUB_LIB_FLAG="-ltclstub`echo ${TCL_VERSION} | tr -d .`"
 fi
 
-TCL_BUILD_STUB_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TCL_STUB_LIB_FLAG}"
+TCL_BUILD_STUB_LIB_SPEC="`pwd`/${TCL_STUB_LIB_FILE}"
 TCL_STUB_LIB_SPEC="-L${TCL_STUB_LIB_DIR} ${TCL_STUB_LIB_FLAG}"
 TCL_BUILD_STUB_LIB_PATH="`pwd`/${TCL_STUB_LIB_FILE}"
 TCL_STUB_LIB_PATH="${TCL_STUB_LIB_DIR}/${TCL_STUB_LIB_FILE}"
@@ -19048,7 +19048,7 @@
 
 
 
-CFLAGS="${CFLAGS} ${CPPFLAGS}"; CPPFLAGS=""
+EXTRA_CC_SWITCHES="${EXTRA_CC_SWITCHES} ${CPPFLAGS}"; CPPFLAGS=""
 
 : ${CONFIG_STATUS=./config.status}
 ac_clean_files_save=$ac_clean_files
diff -uNr tcltk-8.6.10.orig/tk8.6.10/unix/Makefile.in tcltk-8.6.10/tk8.6.10/unix/Makefile.in
--- tcltk-8.6.10.orig/tk8.6.10/unix/Makefile.in	2017-08-09 10:46:03.000000000 -0400
+++ tcltk-8.6.10/tk8.6.10/unix/Makefile.in	2017-08-17 19:56:13.000000000 -0400
@@ -149,7 +149,7 @@
 # X11 include files accessible (the configure script will try to
 # set this value, and will cause it to be an empty string if the
 # include files are accessible via /usr/include).
-X11_INCLUDES		= @XINCLUDES@
+X11_INCLUDES		= $(XFT_CFLAGS) @XINCLUDES@
 
 AQUA_INCLUDES		= -I$(MAC_OSX_DIR) -I$(XLIB_DIR)
 
@@ -266,6 +266,9 @@
 LIBS = @LIBS@ $(X11_LIB_SWITCHES) @TCL_LIBS@
 WISH_LIBS = $(TCL_LIB_SPEC) @LIBS@ $(X11_LIB_SWITCHES) @TCL_LIBS@ @EXTRA_WISH_LIBS@
 
+# support for embedded libraries on Darwin / Mac OS X
+DYLIB_INSTALL_DIR       = ${libdir}
+
 # The symbols below provide support for dynamic loading and shared
 # libraries.  See configure.in for a description of what the
 # symbols mean.  The values of the symbols are normally set by the
@@ -283,9 +283,6 @@
 CC_SEARCH_FLAGS	= @CC_SEARCH_FLAGS@
 LD_SEARCH_FLAGS	= @LD_SEARCH_FLAGS@
 
-# support for embedded libraries on Darwin / Mac OS X
-DYLIB_INSTALL_DIR	= ${LIB_RUNTIME_DIR}
-
 # support for building the Aqua resource file
 TK_RSRC_FILE		= @TK_RSRC_FILE@
 WISH_RSRC_FILE		= @WISH_RSRC_FILE@
@@ -330,18 +330,18 @@
 
 CC_SWITCHES_NO_STUBS = ${CFLAGS} ${CFLAGS_WARNING} ${SHLIB_CFLAGS} \
 -I${UNIX_DIR} -I${GENERIC_DIR} -I${BMAP_DIR} -I${TCL_GENERIC_DIR} \
--I${TCL_PLATFORM_DIR} ${@TK_WINDOWINGSYSTEM@_INCLUDES} ${AC_FLAGS} \
-${PROTO_FLAGS} ${SECURITY_FLAGS} ${MEM_DEBUG_FLAGS} ${KEYSYM_FLAGS} \
-${NO_DEPRECATED_FLAGS} @EXTRA_CC_SWITCHES@
+-I${TCL_PLATFORM_DIR} ${AC_FLAGS} ${PROTO_FLAGS} ${SECURITY_FLAGS} \
+${MEM_DEBUG_FLAGS} ${KEYSYM_FLAGS} ${NO_DEPRECATED_FLAGS} \
+@EXTRA_CC_SWITCHES@ ${@TK_WINDOWINGSYSTEM@_INCLUDES}
 
 CC_SWITCHES = $(CC_SWITCHES_NO_STUBS) @TCL_STUB_FLAGS@
 
 APP_CC_SWITCHES = $(CC_SWITCHES_NO_STUBS) @EXTRA_APP_CC_SWITCHES@
 
 DEPEND_SWITCHES = ${CFLAGS} -I${UNIX_DIR} -I${GENERIC_DIR} -I${BMAP_DIR} \
--I${TCL_GENERIC_DIR} -I${TCL_PLATFORM_DIR} ${@TK_WINDOWINGSYSTEM@_INCLUDES} \
-${AC_FLAGS} ${PROTO_FLAGS} ${SECURITY_FLAGS} ${MEM_DEBUG_FLAGS} \
-${KEYSYM_FLAGS} @EXTRA_CC_SWITCHES@
+-I${TCL_GENERIC_DIR} -I${TCL_PLATFORM_DIR} ${AC_FLAGS} ${PROTO_FLAGS} \
+${SECURITY_FLAGS} ${MEM_DEBUG_FLAGS} ${KEYSYM_FLAGS} @EXTRA_CC_SWITCHES@ \
+${@TK_WINDOWINGSYSTEM@_INCLUDES}
 
 WISH_OBJS = tkAppInit.o
 
@@ -1202,7 +1202,7 @@
 
 # NB: tkUnixRFont.o uses nondefault CFLAGS
 tkUnixRFont.o: $(UNIX_DIR)/tkUnixRFont.c
-	$(CC) -c $(CC_SWITCHES) $(XFT_CFLAGS) $(UNIX_DIR)/tkUnixRFont.c
+	$(CC) -c $(CC_SWITCHES) $(UNIX_DIR)/tkUnixRFont.c
 
 tkUnixInit.o: $(UNIX_DIR)/tkUnixInit.c tkConfig.sh
 	$(CC) -c $(CC_SWITCHES) -DTK_LIBRARY=\"${TK_LIBRARY}\" \
diff -uNr tcltk-8.6.10.orig/tcl8.6.10/unix/configure tcltk-8.6.10/tcl8.6.10/unix/configure
--- tcltk-8.6.10.orig/tcl8.6.10/unix/configure	2019-11-20 20:10:12.000000000 +0100
+++ tcltk-8.6.10/tcl8.6.10/unix/configure	2020-04-13 17:19:28.000000000 +0200
@@ -7895,7 +7895,7 @@
 			done
 fi
 
-		    LIBS="$LIBS -framework CoreFoundation"
+		    LIBS="$LIBS -Wl,-framework,CoreFoundation"
 		    cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -7955,7 +7955,7 @@
 echo "${ECHO_T}$tcl_cv_lib_corefoundation" >&6
 		if test $tcl_cv_lib_corefoundation = yes; then
 
-		    LIBS="$LIBS -framework CoreFoundation"
+		    LIBS="$LIBS -Wl,-framework,CoreFoundation"
 
 cat >>confdefs.h <<\_ACEOF
 #define HAVE_COREFOUNDATION 1
diff -uNr tcltk-8.6.10.orig/tk8.6.10/unix/configure tcltk-8.6.10/tk8.6.10/unix/configure
--- tcltk-8.6.10.orig/tk8.6.10/unix/configure	2019-11-20 20:56:52.000000000 +0100
+++ tcltk-8.6.10/tk8.6.10/unix/configure	2020-04-13 17:21:28.000000000 +0200
@@ -5801,7 +5801,7 @@
 			done
 fi
 
-		    LIBS="$LIBS -framework CoreFoundation"
+		    LIBS="$LIBS -Wl,-framework,CoreFoundation"
 		    cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -5861,7 +5861,7 @@
 echo "${ECHO_T}$tcl_cv_lib_corefoundation" >&6
 		if test $tcl_cv_lib_corefoundation = yes; then
 
-		    LIBS="$LIBS -framework CoreFoundation"
+		    LIBS="$LIBS -Wl,-framework,CoreFoundation"
 
 cat >>confdefs.h <<\_ACEOF
 #define HAVE_COREFOUNDATION 1
@@ -10002,21 +10002,16 @@
     XFT_LIBS=""
     if test "$enable_xft" = "no" ; then
 	echo "$as_me:$LINENO: result: $enable_xft" >&5
-echo "${ECHO_T}$enable_xft" >&6
+	echo "${ECHO_T}$enable_xft" >&6
     else
 	found_xft="yes"
-			XFT_CFLAGS=`xft-config --cflags 2>/dev/null` || found_xft="no"
-	XFT_LIBS=`xft-config --libs 2>/dev/null` || found_xft="no"
-	if test "$found_xft" = "no" ; then
-	    found_xft=yes
-	    XFT_CFLAGS=`pkg-config --cflags xft fontconfig 2>/dev/null` || found_xft="no"
-	    XFT_LIBS=`pkg-config --libs xft fontconfig 2>/dev/null` || found_xft="no"
-	fi
+	XFT_CFLAGS=`pkg-config --cflags xft 2>/dev/null` || found_xft="no"
+	XFT_LIBS=`pkg-config --libs xft 2>/dev/null` || found_xft="no"
 	echo "$as_me:$LINENO: result: $found_xft" >&5
-echo "${ECHO_T}$found_xft" >&6
-		if test "$found_xft" = "yes" ; then
+	echo "${ECHO_T}$found_xft" >&6
+	if test "$found_xft" = "yes" ; then
 	    tk_oldCFlags=$CFLAGS
-	    CFLAGS="$CFLAGS $XINCLUDES $XFT_CFLAGS"
+	    CFLAGS="$CFLAGS $XFT_CFLAGS $XINCLUDES"
 	    tk_oldLibs=$LIBS
 	    LIBS="$tk_oldLIBS $XFT_LIBS $XLIBSW"
 	    echo "$as_me:$LINENO: checking for X11/Xft/Xft.h" >&5
@@ -10081,7 +10081,7 @@
 	fi
 		if test "$found_xft" = "yes" ; then
 	    tk_oldCFlags=$CFLAGS
-	    CFLAGS="$CFLAGS $XINCLUDES $XFT_CFLAGS"
+	    CFLAGS="$CFLAGS $XFT_CFLAGS $XINCLUDES"
 	    tk_oldLibs=$LIBS
 	    LIBS="$tk_oldLIBS $XFT_LIBS $XLIBSW"
 
@@ -11059,7 +11059,7 @@
 	else
 	    TK_LIB_FLAG="-ltk`echo ${TK_VERSION} | tr -d .`"
 	fi
-	TK_BUILD_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TK_LIB_FLAG}"
+	TK_BUILD_LIB_SPEC="`pwd`/${TK_LIB_FILE}"
     fi
     TK_LIB_SPEC="-L${libdir} ${TK_LIB_FLAG}"
 fi
@@ -11079,7 +11079,7 @@
     TK_STUB_LIB_FLAG="-ltkstub`echo ${TK_VERSION} | tr -d .`"
 fi
 
-TK_BUILD_STUB_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TK_STUB_LIB_FLAG}"
+TK_BUILD_STUB_LIB_SPEC="`pwd`/${TK_STUB_LIB_FILE}"
 TK_STUB_LIB_SPEC="-L${TK_STUB_LIB_DIR} ${TK_STUB_LIB_FLAG}"
 TK_BUILD_STUB_LIB_PATH="`pwd`/${TK_STUB_LIB_FILE}"
 TK_STUB_LIB_PATH="${TK_STUB_LIB_DIR}/${TK_STUB_LIB_FILE}"
@@ -11268,7 +11268,7 @@
 LTLIBOBJS=$ac_ltlibobjs
 
 
-CFLAGS="${CFLAGS} ${CPPFLAGS}"; CPPFLAGS=""
+EXTRA_CC_SWITCHES="${EXTRA_CC_SWITCHES} ${CPPFLAGS}"; CPPFLAGS=""
 
 : ${CONFIG_STATUS=./config.status}
 ac_clean_files_save=$ac_clean_files
diff -uNr tcltk-8.6.10.orig/tk8.6.10/unix/tkConfig.sh.in tcltk-8.6.10/tk8.6.10/unix/tkConfig.sh.in
--- tcltk-8.6.10.orig/tk8.6.10/unix/tkConfig.sh.in	2017-08-09 10:46:03.000000000 -0400
+++ tcltk-8.6.10/tk8.6.10/unix/tkConfig.sh.in	2017-08-17 19:56:13.000000000 -0400
@@ -32,7 +32,7 @@
 TK_LIB_FILE='@TK_LIB_FILE@'
 
 # Additional libraries to use when linking Tk.
-TK_LIBS='@XLIBSW@ @XFT_LIBS@ @LIBS@ @TCL_LIBS@'
+TK_LIBS='@XFT_LIBS@ @XLIBSW@ @LIBS@ @TCL_LIBS@'
 
 # Top-level directory in which Tk's platform-independent files are
 # installed.
