--- dumpasn1-20101112 2/Makefile	2011-03-19 04:37:06.000000000 -0400
+++ dumpasn1-20101112/Makefile	2011-11-30 11:51:17.000000000 -0500
@@ -1,11 +1,14 @@
 all: dumpasn1
 
 dumpasn1: dumpasn1.c
+	head -46 dumpasn1.c > README.txt
 	gcc -o dumpasn1 -DDEBIAN dumpasn1.c
 
 install:
-	cp dumpasn1 $(DESTDIR)/usr/bin/
-	cp dumpasn1.cfg $(DESTDIR)/etc/dumpasn1/
+	install -d $(DESTDIR)$(prefix)/bin
+	install -m 755 dumpasn1 $(DESTDIR)$(prefix)/bin/
+	install -d $(DESTDIR)$(prefix)/etc/dumpasn1
+	install -m 0644 dumpasn1.cfg $(DESTDIR)$(prefix)/etc/dumpasn1/
 
 clean:
 	rm -f *.o *~
