diff -ru latexmk-408/latexmk.pl latexmk-408-patched/latexmk.pl
--- latexmk-408/latexmk.pl	2009-06-23 14:36:28.000000000 +0200
+++ latexmk-408-patched/latexmk.pl	2009-08-03 22:53:02.000000000 +0200
@@ -759,9 +759,7 @@
     ## /usr/local/lib/latexmk/LatexMk is put in the list for
     ## compatibility with older versions of latexmk.
     @rc_system_files = 
-     ( '/opt/local/share/latexmk/LatexMk', 
-       '/usr/local/share/latexmk/LatexMk',
-       '/usr/local/lib/latexmk/LatexMk' );
+     ( '@PREFIX@/share/latexmk/LatexMk' );
 
     $search_path_separator = ':';  # Separator of elements in search_path
 
@@ -803,8 +801,8 @@
     $ps_update_method = 0;  # gv -watch watches the ps file
     $ps_previewer  = 'start gv %O %S';
     $ps_previewer_landscape  = 'start gv -swap %O %S';
-    $pdf_previewer = 'start acroread %O %S';
-    $pdf_update_method = 1;  # acroread under unix needs manual update
+    $pdf_previewer = 'start open %O %S';
+    $pdf_update_method = 0;
     $lpr = 'lpr %O %S';         # Assume lpr command prints postscript files correctly
     $lpr_dvi =
         'NONE $lpr_dvi variable is not configured to allow printing of dvi files';
@@ -1981,7 +1979,7 @@
     foreach my $rule (keys %rule_list) {
         $source_list{$rule} = [];
         my $PAsources = $source_list{$rule};
-        my ( $cmd_type, $cmd, $source, $dest, $root ) = @{$rule_list{$rule}};
+        my ( $cmd_type, $ext_cmd, $int_cmd, $source, $dest, $base, $test_kind ) = @{$rule_list{$rule}};
         if ($source) {
 	    push @$PAsources, [ $rule, $source, '' ];
 	}
