diff -Nurd -x'*~' transcode-1.1.0.orig/encode/encode_lavc.c transcode-1.1.0/encode/encode_lavc.c
--- transcode-1.1.0.orig/encode/encode_lavc.c	2009-01-02 03:52:19.000000000 -0500
+++ transcode-1.1.0/encode/encode_lavc.c	2012-06-13 23:49:04.000000000 -0400
@@ -955,8 +955,6 @@
     /* 
      * context *transcode* (not libavcodec) defaults
      */
-    pd->ff_vcontext.mb_qmin                 = 2;
-    pd->ff_vcontext.mb_qmax                 = 31;
     pd->ff_vcontext.max_qdiff               = 3;
     pd->ff_vcontext.max_b_frames            = 0;
     pd->ff_vcontext.me_range                = 0;
@@ -1116,8 +1114,6 @@
         //  handled by transcode core
         //  { "vqmax", PCTX(qmax), TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 1, 60 },
         //  handled by transcode core
-        { "mbqmin", PCTX(mb_qmin), TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 1, 60 },
-        { "mbqmax", PCTX(mb_qmax), TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 1, 60 },
         { "lmin", PAUX(lmin), TCCONF_TYPE_FLOAT, TCCONF_FLAG_RANGE, 0.01, 255.0 },
         { "lmax", PAUX(lmax), TCCONF_TYPE_FLOAT, TCCONF_FLAG_RANGE, 0.01, 255.0 },
         { "vqdiff", PCTX(max_qdiff), TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 1, 31 },
diff -Nurd -x'*~' transcode-1.1.0.orig/export/export_ffmpeg.c transcode-1.1.0/export/export_ffmpeg.c
--- transcode-1.1.0.orig/export/export_ffmpeg.c	2008-11-29 10:43:14.000000000 -0500
+++ transcode-1.1.0/export/export_ffmpeg.c	2012-06-13 23:36:32.000000000 -0400
@@ -643,8 +643,6 @@
 
     lavc_venc_context->bit_rate           = vob->divxbitrate * 1000;
     lavc_venc_context->bit_rate_tolerance = lavc_param_vrate_tolerance * 1000;
-    lavc_venc_context->mb_qmin            = lavc_param_mb_qmin;
-    lavc_venc_context->mb_qmax            = lavc_param_mb_qmax;
     lavc_venc_context->lmin= (int)(FF_QP2LAMBDA * lavc_param_lmin + 0.5);
     lavc_venc_context->lmax= (int)(FF_QP2LAMBDA * lavc_param_lmax + 0.5);
     lavc_venc_context->max_qdiff          = lavc_param_vqdiff;
diff -Nurd -x'*~' transcode-1.1.0.orig/export/export_xvid4.c transcode-1.1.0/export/export_xvid4.c
--- transcode-1.1.0.orig/export/export_xvid4.c	2007-08-11 04:15:45.000000000 -0400
+++ transcode-1.1.0/export/export_xvid4.c	2012-06-13 23:33:02.000000000 -0400
@@ -91,6 +91,7 @@
 #endif
 #define XVID_CONFIG_FILE "xvid4.cfg"
 
+
 /*****************************************************************************
  * XviD symbols grouped in a nice struct.
  ****************************************************************************/
@@ -1248,16 +1249,26 @@
                 XVID_SHARED_LIB_SUFX, XVID_API_MAJOR(XVID_API));
 #endif
 #ifdef OS_DARWIN
-    tc_snprintf(soname[1], 4095, "%s.%d.%s", XVID_SHARED_LIB_BASE,
+    tc_snprintf(soname[1], 4095, "%s/../%s.%d.%s", path,XVID_SHARED_LIB_BASE,
                 XVID_API_MAJOR(XVID_API), XVID_SHARED_LIB_SUFX);
 #else
     tc_snprintf(soname[1], 4095, "%s.%s.%d", XVID_SHARED_LIB_BASE,
                 XVID_SHARED_LIB_SUFX, XVID_API_MAJOR(XVID_API));
 #endif
+#ifdef OS_DARWIN
+    tc_snprintf(soname[2], 4095, "%s/%s.%s", path, XVID_SHARED_LIB_BASE,
+                XVID_SHARED_LIB_SUFX);
+#else
     tc_snprintf(soname[2], 4095, "%s/%s.%s", path, XVID_SHARED_LIB_BASE,
                 XVID_SHARED_LIB_SUFX);
+#endif
+#ifdef OS_DRAWIN
     tc_snprintf(soname[3], 4095, "%s.%s", XVID_SHARED_LIB_BASE,
                 XVID_SHARED_LIB_SUFX);
+#else
+   tc_snprintf(soname[3], 4095, "%s/../%s.%s",XVID_SHARED_LIB_BASE,
+                XVID_SHARED_LIB_SUFX);
+#endif
 
     /* Let's try each shared lib until success */
     for(i=0; i<4; i++) {
diff -Nurd -x'*~' transcode-1.1.0.orig/export/ffmpeg_cfg.c transcode-1.1.0/export/ffmpeg_cfg.c
--- transcode-1.1.0.orig/export/ffmpeg_cfg.c	2008-09-21 14:37:09.000000000 -0400
+++ transcode-1.1.0/export/ffmpeg_cfg.c	2012-06-13 23:47:40.000000000 -0400
@@ -34,8 +34,6 @@
 //int lavc_param_vqscale = 0;
 //int lavc_param_vqmin = 2;
 //int lavc_param_vqmax = 31;
-int lavc_param_mb_qmin = 2;
-int lavc_param_mb_qmax = 31;
 int lavc_param_lmin = 2;
 int lavc_param_lmax = 31;
 int lavc_param_vqdiff = 3;
@@ -140,8 +138,6 @@
 //    {"vqscale", &lavc_param_vqscale, TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 1, 31},
 //    {"vqmin", &lavc_param_vqmin, TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 1, 31},
 //    {"vqmax", &lavc_param_vqmax, TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 1, 31},
-    {"mbqmin", &lavc_param_mb_qmin, TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 1, 31},
-    {"mbqmax", &lavc_param_mb_qmax, TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 1, 31},
     {"lmin", &lavc_param_lmin, TCCONF_TYPE_FLOAT, TCCONF_FLAG_RANGE, 0.01, 255.0},
     {"lmax", &lavc_param_lmax, TCCONF_TYPE_FLOAT, TCCONF_FLAG_RANGE, 0.01, 255.0},
     {"vqdiff", &lavc_param_vqdiff, TCCONF_TYPE_INT, TCCONF_FLAG_RANGE, 1, 31},
diff -Nurd -x'*~' transcode-1.1.0.orig/export/ffmpeg_cfg.h transcode-1.1.0/export/ffmpeg_cfg.h
--- transcode-1.1.0.orig/export/ffmpeg_cfg.h	2008-03-25 15:50:40.000000000 -0400
+++ transcode-1.1.0/export/ffmpeg_cfg.h	2012-06-13 23:46:39.000000000 -0400
@@ -13,8 +13,6 @@
 //extern int lavc_param_vqscale;
 //extern int lavc_param_vqmin;
 //extern int lavc_param_vqmax;
-extern int lavc_param_mb_qmin;
-extern int lavc_param_mb_qmax;
 extern int lavc_param_lmin;
 extern int lavc_param_lmax;
 extern int lavc_param_vqdiff;
diff -Nurd -x'*~' transcode-1.1.0.orig/filter/filter_logoaway.c transcode-1.1.0/filter/filter_logoaway.c
--- transcode-1.1.0.orig/filter/filter_logoaway.c	2007-07-30 16:53:37.000000000 -0400
+++ transcode-1.1.0/filter/filter_logoaway.c	2012-06-13 23:33:02.000000000 -0400
@@ -694,6 +694,8 @@
     data[instance]->vcolor   = 128;
     data[instance]->alpha    = 0;
     data[instance]->dump     = 0;
+    data[instance]->image     = (Image *)NULL; 
+    data[instance]->dumpimage = (Image *)NULL;
 
     // filter init ok.
 
@@ -808,7 +810,8 @@
     if (data[instance]->dumpimage != (Image *)NULL) {
       DestroyImage(data[instance]->dumpimage);
       DestroyImageInfo(data[instance]->dumpimage_info);
-      DestroyConstitute();
+      /*DestroyConstitute();*/
+      ConstituteComponentTerminus();
     }
     DestroyExceptionInfo(&data[instance]->exception_info);
     DestroyMagick();
diff -Nurd -x'*~' transcode-1.1.0.orig/filter/subtitler/Makefile.in transcode-1.1.0/filter/subtitler/Makefile.in
--- transcode-1.1.0.orig/filter/subtitler/Makefile.in	2009-01-17 03:30:39.000000000 -0500
+++ transcode-1.1.0/filter/subtitler/Makefile.in	2012-06-13 23:57:43.000000000 -0400
@@ -303,9 +303,9 @@
 	$(PTHREAD_CFLAGS) \
 	-I$(top_srcdir) \
 	-I$(top_srcdir)/src \
-	$(X_CFLAGS) \
 	$(FREETYPE2_CFLAGS) \
-	$(ICONV_CFLAGS)
+	$(ICONV_CFLAGS) \
+	$(X_CFLAGS)
 
 pkgdir = $(MOD_PATH)
 pkg_LTLIBRARIES = filter_subtitler.la
diff -Nurd -x'*~' transcode-1.1.0.orig/import/decode_lavc.c transcode-1.1.0/import/decode_lavc.c
--- transcode-1.1.0.orig/import/decode_lavc.c	2008-09-21 14:37:09.000000000 -0400
+++ transcode-1.1.0/import/decode_lavc.c	2012-06-13 23:42:20.000000000 -0400
@@ -253,8 +253,12 @@
 
       //tc_log_msg(__FILE__, "SIZE: (%d) MP4(%d) blen(%d) BUF(%d) read(%ld)", len, mp4_size, buf_len, READ_BUFFER_SIZE, bytes_read);
       do {
-	  len = avcodec_decode_video(lavc_dec_context, &picture,
-		  &got_picture, buffer+buf_len, mp4_size-buf_len);
+	  AVPacket pkt;
+	  av_init_packet( &pkt );
+	  pkt.data = buffer+buf_len;
+	  pkt.size = mp4_size-buf_len;
+	  len = avcodec_decode_video2(lavc_dec_context, &picture,
+		  &got_picture, &pkt);
 
 	  if (len < 0) {
 	      tc_log_error(__FILE__, "frame decoding failed");
diff -Nurd -x'*~' transcode-1.1.0.orig/import/probe_ffmpeg.c transcode-1.1.0/import/probe_ffmpeg.c
--- transcode-1.1.0.orig/import/probe_ffmpeg.c	2009-01-02 03:52:19.000000000 -0500
+++ transcode-1.1.0/import/probe_ffmpeg.c	2012-06-13 23:45:38.000000000 -0400
@@ -41,7 +41,7 @@
     for (i = 0; i < ctx->nb_streams; i++) {
         st = ctx->streams[i];
 
-        if (st->codec->codec_type == CODEC_TYPE_VIDEO) {
+        if (st->codec->codec_type == AVMEDIA_TYPE_VIDEO) {
             info->bitrate = st->codec->bit_rate / 1000;
             info->width = st->codec->width;
             info->height = st->codec->height;
@@ -59,7 +59,7 @@
     for (i = 0; i < ctx->nb_streams; i++) {
         st = ctx->streams[i];
 
-        if (st->codec->codec_type == CODEC_TYPE_AUDIO
+        if (st->codec->codec_type == AVMEDIA_TYPE_AUDIO
          && j < TC_MAX_AUD_TRACKS) {
             info->track[j].format = 0x1; /* known wrong */
             info->track[j].chan = st->codec->channels;
