diff -Naur dictd-1.7.1/Makefile.in dictd-1.7.1.fink/Makefile.in
--- dictd-1.7.1/Makefile.in	Sat May  4 21:59:56 2002
+++ dictd-1.7.1.fink/Makefile.in	Mon Aug  5 18:21:45 2002
@@ -144,7 +144,7 @@
 	if test ! -d $(bindir); then $(INSTALL) -d 755 $(bindir); fi
 	if test ! -d $(man1_prefix); then $(INSTALL) -d 755 $(man1_prefix); fi
 	$(INSTALL_PROGRAM) dict $(bindir)
-	$(INSTALL_DATA	dict.1 $(man1_prefix)/dict.1
+	$(INSTALL_DATA)	dict.1 $(man1_prefix)/dict.1
 
 install.dictzip: dictzip
 	if test ! -d $(bindir); then $(INSTALL) -d 755 $(bindir); fi
diff -Naur dictd-1.7.1/dict.conf dictd-1.7.1.fink/dict.conf
--- dictd-1.7.1/dict.conf	Wed Dec 31 19:00:00 1969
+++ dictd-1.7.1.fink/dict.conf	Mon Aug  5 18:10:24 2002
@@ -0,0 +1,16 @@
+#  /etc/dict.conf Written by Bob Hilliard <hilliard@debian.org>
+#  1998/03/20.  Last revised Sun, 22 Nov 1998 18:10:04 -0500 This is
+#  the configuration file for /usr/bin/dict.  In most cases only the
+#  server keyword need be specified.
+
+#  This default configuration will try to access a dictd server on the
+#  local host, failing that, it will try the public server.  In many
+#  cases this will be slow, so you should comment out the line for the
+#  server that you don't want to use. To use any other server, enter
+#  its IP address in place of "dict.org".
+
+#  Refer to the dict manpage (man dict) for other options that could
+#  be inserted in here.
+
+server localhost
+server dict.org
diff -Naur dictd-1.7.1/libmaa/config.h.in dictd-1.7.1.fink/libmaa/config.h.in
--- dictd-1.7.1/libmaa/config.h.in	Sat Mar 30 09:52:01 2002
+++ dictd-1.7.1.fink/libmaa/config.h.in	Mon Aug  5 18:10:24 2002
@@ -96,7 +96,7 @@
 #undef HAVE_SYSCONF
 
 /* Define if you have the <getopt.h> header file.  */
-#undef HAVE_GETOPT_H
+#define HAVE_GETOPT_H 1
 
 /* Define if you have the <limits.h> header file.  */
 #undef HAVE_LIMITS_H
diff -Naur dictd-1.7.1/regex/Makefile dictd-1.7.1.fink/regex/Makefile
--- dictd-1.7.1/regex/Makefile	Sun Mar 31 11:41:44 2002
+++ dictd-1.7.1.fink/regex/Makefile	Mon Aug  5 18:10:24 2002
@@ -40,6 +40,7 @@
 libregex.a:	$(OBJPRODN)
 	rm -f libregex.a
 	ar crv libregex.a $(OBJPRODN)
+	ranlib libregex.a
 
 purge:
 	rm -f *.o
