diff -Naur html2text-1.3.1/configure /sw/src/html2text-1.3.1-1/html2text-1.3.1/configure
--- html2text-1.3.1/configure	Sun Apr  7 07:05:48 2002
+++ /sw/src/html2text-1.3.1-1/html2text-1.3.1/configure	Fri Oct 31 11:51:36 2003
@@ -38,7 +38,7 @@
 }
 EOF
 CXX=unknown;
-for i in "CC" "g++"; do
+for i in "c++ -no-cpp-precomp -isystem /sw/include -DAUTO_PTR_BROKEN=1 -Wl,-L/sw/lib -Wl,-lpoll" "CC" "g++"; do
   if $i -c $tmp_file.C 2>/dev/null; then
     CXX="$i";
     break;
diff -Naur html2text-1.3.1/format.C /sw/src/html2text-1.3.1-1/html2text-1.3.1/format.C
--- html2text-1.3.1/format.C	Mon Jul 29 04:12:13 2002
+++ /sw/src/html2text-1.3.1-1/html2text-1.3.1/format.C	Fri Oct 31 12:01:20 2003
@@ -34,7 +34,7 @@
  /***************************************************************************/
 
 
-#include <strstream.h>
+#include <strstream>
 
 #include <stdlib.h>
 #include <ctype.h>
@@ -374,7 +374,7 @@
   case CUSTOM3:   bullet = custom3_bullet; break;
   case ARABIC_NUMBERS:
     {
-      ostrstream oss;
+      std::ostrstream oss;
       oss << number << '.' << std::ends;
       bullet = oss.str();
       oss.rdbuf()->freeze(0);
