diff -ruN -x '*~' pybliographer-1.2.7/setup-check.py pybliographer-1.2.7.patched/setup-check.py
--- pybliographer-1.2.7/setup-check.py	2005-10-13 12:11:19.000000000 -0400
+++ pybliographer-1.2.7.patched/setup-check.py	2005-11-02 23:22:01.000000000 -0500
@@ -86,8 +86,8 @@
 
     error ('unexpected error')
 
-if gtk.pygtk_version < (2,4,0):
-    error ('requested version for PyGtk is %s, but I detected %s' % ('2.4.0', v))
+#if gtk.pygtk_version < (2,4,0):
+#    error ('requested version for PyGtk is %s, but I detected %s' % ('2.4.0', v))
     
 
 try:
