diff -Naur mkhexgrid-0.1.1-old/Makefile mkhexgrid-0.1.1/Makefile
--- mkhexgrid-0.1.1-old/Makefile	2007-01-10 08:37:42.000000000 -0500
+++ mkhexgrid-0.1.1/Makefile	2007-01-10 20:43:09.000000000 -0500
@@ -1,4 +1,4 @@
-DESTDIR=/usr/local
+#DESTDIR=/usr/local
 BINDIR=$(DESTDIR)/bin
 DOCDIR=$(DESTDIR)/share/doc
 MANDIR=$(DESTDIR)/share/man
@@ -9,8 +9,8 @@
 RPMDIR=/home/uckelman/rpmbuild
 
 CC=g++
-CPPFLAGS=-c -g -O2 -W -Wall -DVERSION='"$(VERSION)"'
-LDFLAGS=-lm -lstdc++ -lgd
+CPPFLAGS+=-c -g -O2 -W -Wall -DVERSION='"$(VERSION)"'
+LDFLAGS+=-lm -lstdc++ -lgd
 
 FILES=grid.h \
       grid.cpp \
@@ -60,12 +60,12 @@
 	rm -rf $(DISTDIR)
 
 install:
-	install -m 755 -o 0 -g 0 -d $(BINDIR)
-	install -m 755 -o 0 -g 0 -s mkhexgrid $(BINDIR)
-	install -m 755 -o 0 -g 0 -d $(MANDIR)/man1
-	install -m 644 -o 0 -g 0 doc/mkhexgrid.1 $(MANDIR)/man1
-	install -m 755 -o 0 -g 0 -d $(DOCDIR)/mkhexgrid-$(VERSION) 
-	install -m 644 -o 0 -g 0 $(DOCS) $(DOCDIR)/mkhexgrid-$(VERSION)
+	install -m 755 -d $(BINDIR)
+	install -m 755 -s mkhexgrid $(BINDIR)
+	install -m 755 -d $(MANDIR)/man1
+	install -m 644 doc/mkhexgrid.1 $(MANDIR)/man1
+#	install -m 755 -o 0 -g 0 -d $(DOCDIR)/mkhexgrid-$(VERSION) 
+#	install -m 644 -o 0 -g 0 $(DOCS) $(DOCDIR)/mkhexgrid-$(VERSION)
 
 clean:
 	rm -rf mkhexgrid mkhexgrid-web mkhexgrid.o grid.o png.o ps.o svg.o \

 	  	 
