diff -Nurd -x'*~' xboard-4.5.2a.orig/configure xboard-4.5.2a/configure
--- xboard-4.5.2a.orig/configure	2011-05-01 23:18:58.000000000 -0400
+++ xboard-4.5.2a/configure	2012-08-28 11:09:57.000000000 -0400
@@ -5833,12 +5833,18 @@
   exit 1
 fi
 
+save_CFLAGS="$CFLAGS"
+save_CPPFLAGS="$CPPFLAGS"
+CFLAGS="$CFLAGS $X_CFLAGS"
+CPPFLAGS="$CPPFLAGS $X_CFLAGS"
 ac_fn_c_check_header_mongrel "$LINENO" "X11/Intrinsic.h" "ac_cv_header_X11_Intrinsic_h" "$ac_includes_default"
 if test "x$ac_cv_header_X11_Intrinsic_h" = xyes; then :
   xt="yes"
 else
   xt="no"
 fi
+CFLAGS="$save_CFLAGS"
+CPPFLAGS="$save_CPPFLAGS"
 
 
 
@@ -5907,7 +5913,13 @@
 
 fi
 if test x"$with_Xaw3d" = x"no" ; then
+   save_CFLAGS="$CFLAGS"
+   save_CPPFLAGS="$CPPFLAGS"
+   CFLAGS="$CFLAGS $X_CFLAGS"
+   CPPFLAGS="$CPPFLAGS $X_CFLAGS"
    ac_fn_c_check_header_mongrel "$LINENO" "X11/Xaw/Dialog.h" "ac_cv_header_X11_Xaw_Dialog_h" "$ac_includes_default"
+   CFLAGS="$save_CFLAGS"
+   CPPFLAGS="$save_CPPFLAGS"
 if test "x$ac_cv_header_X11_Xaw_Dialog_h" = xyes; then :
   xaw_headers="yes"
 else
