diff -ru XorCurses-0.1.0.orig/Makefile XorCurses-0.1.0/Makefile
--- XorCurses-0.1.0.orig/Makefile	2009-10-02 08:21:06.000000000 -0400
+++ XorCurses-0.1.0/Makefile	2009-10-02 10:02:39.000000000 -0400
@@ -10,7 +10,7 @@
 #------------------standard options------------------
 # probably fine for most people:
 
-CFLAGS:= -O2 -std=gnu99 -Wall -pedantic -DDATADIR=\"$(SHAREDIR)\"
+CFLAGS:= -O2 -std=gnu99 -Wall -pedantic -DDATADIR=\"$(SHAREDIR)\" -I@PREFIX@/include/ncursesw -I@PREFIX@/include
 
 # -DDATA_DIR is an easy way of giving the src files
 # the default install location. options code relies
@@ -39,7 +39,7 @@
 PROG:=xorcurses
 CC:=gcc
 
-LIBS   :=-lcurses
+LIBS   :=-L@PREFIX@/lib -lncursesw
 SRC    :=$(wildcard *.c)
 OBJS   :=$(patsubst %.c, %.o, $(SRC))
 HEADERS:=$(wildcard *.h)
@@ -56,10 +56,11 @@
 	rm -f $(PROG) $(OBJS)
 
 install: $(PROG)
+	install -d $(BINDIR)
 	install $(PROG) $(BINDIR)
 	install -d $(MAPDIR)
-	install -t $(MAPDIR) maps/[1-9]*
-	install -t $(SHAREDIR) help*.txt
+	install maps/[1-9]* $(MAPDIR)
+	install help*.txt $(SHAREDIR)
 
 uninstall:
 	rm -f $(BINDIR)$(PROG)
