diff -ur gnupg-1.9.19.orig/tools/Makefile.in gnupg-1.9.19/tools/Makefile.in
--- gnupg-1.9.19.orig/tools/Makefile.in	2005-09-12 00:56:51.000000000 -0700
+++ gnupg-1.9.19/tools/Makefile.in	2005-09-17 16:25:26.000000000 -0700
@@ -401,7 +401,7 @@
 gpg_connect_agent_SOURCES = gpg-connect-agent.c no-libgcrypt.c
 gpg_connect_agent_LDADD = ../gl/libgnu.a ../jnlib/libjnlib.a \
                           ../common/libcommon.a \
-                          $(LIBASSUAN_LIBS) $(GPG_ERROR_LIBS) $(LIBINTL)
+                          $(LIBASSUAN_LIBS) $(GPG_ERROR_LIBS) $(LIBINTL) $(PTH_LIBS)
 
 gpgkey2ssh_SOURCES = gpgkey2ssh.c
 gpgkey2ssh_CFLAGS = $(LIBGCRYPT_CFLAGS) $(GPG_ERROR_CFLAGS)
