diff -uNr gtksourceview-sharp-0.2/doc/Makefile.in gtksourceview-sharp-0.2-patched/doc/Makefile.in
--- gtksourceview-sharp-0.2/doc/Makefile.in	Mon May  3 19:46:00 2004
+++ gtksourceview-sharp-0.2-patched/doc/Makefile.in	Sun May 16 16:36:28 2004
@@ -356,9 +356,10 @@
 		$(UPDATER) ../gtksourceview/gtksourceview-sharp.dll -o ./en -f
 
 install-data-local:
-	install -m 644 $(srcdir)/gtksourceview-sharp-docs.source `monodoc --get-sourcesdir`
-	install -m 644 $(srcdir)/gtksourceview-sharp-docs.tree `monodoc --get-sourcesdir`
-	install -m 644 $(srcdir)/gtksourceview-sharp-docs.zip `monodoc --get-sourcesdir`
+	install -d -m 755 $(DESTDIR)`monodoc --get-sourcesdir`
+	install -m 644 $(srcdir)/gtksourceview-sharp-docs.source $(DESTDIR)`monodoc --get-sourcesdir`
+	install -m 644 $(srcdir)/gtksourceview-sharp-docs.tree $(DESTDIR)`monodoc --get-sourcesdir`
+	install -m 644 $(srcdir)/gtksourceview-sharp-docs.zip $(DESTDIR)`monodoc --get-sourcesdir`
 
 dist-hook:
 	mkdir -p $(distdir)/en
