diff -ru opencdk-0.5.4/src/encrypt.c opencdk-0.5.4-patched/src/encrypt.c
--- opencdk-0.5.4/src/encrypt.c	Fri Oct 31 01:11:48 2003
+++ opencdk-0.5.4-patched/src/encrypt.c	Thu Dec 25 03:37:35 2003
@@ -24,7 +24,7 @@
 #endif
 #include <stdio.h>
 #include <stdarg.h>
-#include <malloc.h>
+#include <stdlib.h>
 #include <assert.h>
 
 #include "opencdk.h"
diff -ru opencdk-0.5.4/src/main.c opencdk-0.5.4-patched/src/main.c
--- opencdk-0.5.4/src/main.c	Fri Oct 31 01:19:26 2003
+++ opencdk-0.5.4-patched/src/main.c	Thu Dec 25 03:37:35 2003
@@ -24,7 +24,7 @@
 #endif
 #include <stdio.h>
 #include <errno.h>
-#include <malloc.h>
+#include <stdlib.h>
 #ifdef HAVE_PWD_H
 # include <pwd.h>
 #endif
diff -ru opencdk-0.5.4/src/misc.c opencdk-0.5.4-patched/src/misc.c
--- opencdk-0.5.4/src/misc.c	Mon Nov  3 15:57:37 2003
+++ opencdk-0.5.4-patched/src/misc.c	Thu Dec 25 03:37:35 2003
@@ -27,7 +27,7 @@
 #include <string.h>
 #include <ctype.h>
 #include <time.h>
-#include <malloc.h>
+#include <stdlib.h>
 
 #include "opencdk.h"
 #include "main.h"
