diff -ruN proftpd-1.2.9.orig/contrib/cron.monthly proftpd-1.2.9/contrib/cron.monthly
--- proftpd-1.2.9.orig/contrib/cron.monthly	Wed Dec 31 17:00:00 1969
+++ proftpd-1.2.9/contrib/cron.monthly	Thu Nov  6 09:10:53 2003
@@ -0,0 +1,12 @@
+#!/bin/sh
+#
+# cron script to rotate the proftpd server logfile, based on the
+# wu-ftpd script by Peter Tobias <tobias@et-inf.fho-emden.de>.
+
+[ -x @FINKPREFIX@/sbin/ftpstats ] || exit 0
+
+cd @FINKPREFIX@/var/log
+savelog -u root -g wheel -m 640 -c 12 @FINKPREFIX@/var/log/xferreport >/dev/null 2>/dev/null
+ftpstats -a -r -l 2 -d 2> /dev/null >@FINKPREFIX@/var/log/xferreport
+savelog -u root -g wheel -m 640 -c 7 @FINKPREFIX@/var/log/xferlog >/dev/null 2>/dev/null
+
diff -ruN proftpd-1.2.9.orig/contrib/ftpasswd proftpd-1.2.9/contrib/ftpasswd
--- proftpd-1.2.9.orig/contrib/ftpasswd	Thu Mar 27 18:58:33 2003
+++ proftpd-1.2.9/contrib/ftpasswd	Thu Nov  6 09:10:53 2003
@@ -36,7 +36,7 @@
 my $program = basename($0);
 my $default_passwd_file = "./ftpd.passwd";
 my $default_group_file = "./ftpd.group";
-my $shell_file = "/etc/shells";
+my $shell_file = "@FINKPREFIX@/etc/shells";
 my $default_cracklib_dict = "/usr/lib/cracklib_dict";
 my $cracklib_dict;
 my $output_file;
@@ -747,7 +747,7 @@
                 exist in the specified output-file, an entry will be created
                 for her.  Otherwise, the given fields will be updated.
 
-    --shell     Shell for the user (required).  Recommended: /bin/false
+    --shell     Shell for the user (required).  Recommended: /usr/bin/false
 
     --uid       Numerical user ID (required)
 
diff -ruN proftpd-1.2.9.orig/contrib/ftpstats proftpd-1.2.9/contrib/ftpstats
--- proftpd-1.2.9.orig/contrib/ftpstats	Wed Dec 31 17:00:00 1969
+++ proftpd-1.2.9/contrib/ftpstats	Thu Nov  6 09:10:53 2003
@@ -0,0 +1,389 @@
+#! /usr/bin/perl
+# ---------------------------------------------------------------------------
+#
+# USAGE: xferstats <options>
+#
+# OPTIONS:
+#       -f <filename>   Use <filename> for the log file
+#       -r              include real users 
+#       -a              include anonymous users 
+#       -h		include report on hourly traffic
+#       -d		include report on domain traffic
+#       -t		report on total traffic by section
+#	-i 		report incomming traffic only (uploads)
+#	-o		report outgoing traffic only (download)
+#       -D <domain>     report only on traffic from <domain>
+#				this option leads to problems with the local
+#				domain: e.g. test.com is counted under test
+#				and not recognized under com, -D com will give
+#				you only statistics about com excluding
+#				test.com! use -A com for correct results.
+#       -A <address>    report only on traffic from addresses ends matching
+#				<address> e.g. -A test.domain.com will report
+#				only on addresses ending with test.domain.com
+#       -l <depth>      Depth of path detail for sections
+#       -s <section>    Section to report on, For example: -s /pub will report
+#				only on paths under /pub
+#       -u <user>       Report traffic for specified user.
+#
+# ---------------------------------------------------------------------------
+# 05.09.01: minor change by Jamie Fifield (fifield@chebucto.ns.ca):
+# - added option u which singles out the data for a single user
+#
+# ---------------------------------------------------------------------------
+# 30.09.98: changes by Jan Menzel (jan.menzel@gmx.net):
+#  -documented problems with option -D
+#  -fixed same problems with spaces in the filenames (lines which are not
+#     exactly 16 elements long are skipped in earlier versions)
+#  -added option A which compares the addresses end with a given pattern
+#  -added option i and o which reports either incoming or outgoing traffic only
+
+@mydom = split(/\./, `dnsdomainname`);
+$mydom2 = pop(@mydom); chop($mydom2);
+$mydom1 = pop(@mydom);
+
+# If you want to specify $mydom1 and $mydom2 manually you should edit the
+# next two lines to customize for your domain. This will allow your domain
+# to be separated in the domain listing.
+# $mydom1 = "debian";
+# $mydom2 = "org";
+
+# edit the next line to customize for your default log file
+$usage_file = "@FINKPREFIX@/var/log/xferlog";
+
+# Edit the following lines for default report settings.
+# Entries defined here will be over-ridden by the command line.
+
+$opt_h = 0; 
+$opt_d = 0;
+$opt_t = 1;
+
+require 'getopts.pl';
+&Getopts('f:rahdD:l:s:A:iou:');
+
+$opt_l ||= 3;
+if ($opt_r) { $real = 1;}
+if ($opt_a) { $anon = 1;}
+if ($real == 0 && $anon == 0) { $anon = 1; }
+if ($opt_f) {$usage_file = $opt_f;}
+if ($opt_i) {$opt_i = 1;}
+if ($opt_o) {$opt_o = 1;}
+
+open (LOG,$usage_file) || die "Error opening usage log file: $usage_file\n";
+
+if ($opt_o) {
+   print "Transfer Totals include outgoing traffic only.\n\n";
+}
+
+if ($opt_i) {
+   print "Transfer Totals include incoming traffic only.\n\n";
+}
+
+if ($opt_D) {
+   $opt_D =~ tr/A-Z/a-z/;
+   print "Transfer Totals include the '$opt_D' domain only.\n";
+   print "All other domains are filtered out for this report.\n\n";
+}
+
+if ($opt_A) {
+   $opt_A =~ tr/A-Z/a-z/;
+   print "Transfer Totals include the addresses ending in '$opt_A' only.\n";
+   print "All other addresses are filtered out for this report.\n\n";
+}
+
+if ($opt_s) {
+   print "Transfer Totals include the '$opt_s' section only.\n";
+   print "All other sections are filtered out for this report.\n\n";
+}
+
+if ($opt_u) {
+   print "Transfer Totals are for user '$opt_u' only.\n";
+   print "All other users are filtered out for this report.\n\n";
+}
+
+line: while (<LOG>) {
+
+   @line = split;
+   # is the first entry week day abbreviation?
+   next if (length("$line[0]") != 3);
+
+   # check whether there is a valid 'username'
+   if ($line[-7] eq "a" or $line[-7] eq "b") {
+      # yes, there i
+      # offset points to the first element just behind the filename
+      $offset = $#line - 7;
+   } elsif ($line[-6] eq "a" or $line[-6] eq "b") {
+      $offset = $#line - 6
+   } else {
+      next;
+   }
+   next if (!$opt_u && !$anon && $line[$offset+3] eq "a");
+   next if (!$opt_u && !$real && $line[$offset+3] eq "r");
+   next if ($opt_i && $line[$offset+2] ne "i");
+   next if ($opt_o && $line[$offset+2] ne "o");
+   next if ($opt_u && $line[$offset+4] ne $opt_u);
+
+   $daytime = substr($_, 0, 10) . substr($_, 19, 5);
+   $time = substr($_,11,2); 
+
+   if ($line[8] eq "\.") { $line[8] = "/unreadable/filename";}
+   next if (substr($line[8],0,length("$opt_s")) ne "$opt_s");
+   $line[8] = substr($line[8],length("$opt_s"));
+   @path = split(/\//, $line[8]);
+
+#
+# Why was the original xferstats dropping leading 1 character path
+# segments???
+#
+#  while (length($path[1]) <= 1) {
+#     shift @path;
+#     next line if ($#path == -1);
+#  }
+
+# Things in the top-level directory are assumed to be informational files
+
+   if ($#path == 1)
+      { $pathkey = "Index/Informational Files"; }
+      else {
+	$pathkey = "";
+	for ($i=1; $i <= $#path-1 && $i <= $opt_l;$i++) {
+		$pathkey = $pathkey . "/" . $path[$i];
+		}
+	}
+
+   $line[6] =~ tr/A-Z/a-z/;
+
+   @address = split(/\./, $line[6]);
+
+   $domain = $address[$#address];
+   if ($domain eq "$mydom2" && $address[$#address-1] eq "$mydom1")
+      { $domain = $mydom1 . "." . $mydom2; }
+   if ( int($address[0]) > 0 || $#address < 2 )
+      { $domain = "unresolved"; }
+
+   $count = 1;
+   if ($opt_D and substr($domain,0,length("$opt_D")) ne "$opt_D" ) {
+      $count = 0;
+   }
+   if ($opt_A and substr($line[6],length("$line[6]")-length("$opt_A")) ne "$opt_A" ) {
+      $count = 0;
+   }
+
+
+   if ($count) {
+
+   $systemfiles{$line[6]}++;                    # Systems Accessing the Arc
+   $xferfiles++;                                # total files sent
+   $xfertfiles++;                               # total files sent
+   $xferfiles{$daytime}++;                      # files per day
+   $groupfiles{$pathkey}++;                     # per-group accesses
+   $domainfiles{$domain}++;
+
+   $xfersecs{$daytime}    += $line[5];          # xmit seconds per day
+   $domainsecs{$domain}   += $line[5];		# xmit seconds for domain
+   $xferbytes{$daytime}   += $line[7];          # bytes per day
+   $domainbytes{$domain}  += $line[7];		# xmit bytes to domain
+   $xferbytes             += $line[7];          # total bytes sent
+   $groupbytes{$pathkey}  += $line[7];          # per-group bytes sent
+
+   $xfertfiles{$time}++;                        # files per hour
+   $xfertsecs{$time}      += $line[5];          # xmit seconds per hour
+   $xfertbytes{$time}     += $line[7];          # bytes per hour
+   $xfertbytes            += $line[7];          # total bytes sent
+   }
+}
+close LOG;
+
+@syslist = keys(systemfiles);
+@dates = sort datecompare keys(xferbytes);
+
+if ($xferfiles == 0) {die "There was no data to process.\n";}
+
+
+print "TOTALS FOR SUMMARY PERIOD ", $dates[0], " TO ", $dates[$#dates], "\n\n";
+printf ("Files Transmitted During Summary Period  %12.0f\n", $xferfiles);
+printf ("Bytes Transmitted During Summary Period  %12.0f\n", $xferbytes); 
+printf ("Systems Using Archives                   %12.0f\n\n", $#syslist+1);
+
+printf ("Average Files Transmitted Daily          %12.0f\n",
+   $xferfiles / ($#dates + 1));
+printf ("Average Bytes Transmitted Daily          %12.0f\n",
+   $xferbytes / ($#dates + 1));
+
+format top1 =
+
+Daily Transmission Statistics
+
+                 Number Of    Number of    Average    Percent Of  Percent Of
+     Date        Files Sent  Bytes  Sent  Xmit  Rate  Files Sent  Bytes Sent
+---------------  ----------  -----------  ----------  ----------  ----------
+.
+
+format line1 =
+@<<<<<<<<<<<<<<  @>>>>>>>>>  @>>>>>>>>>>  @>>>>>>>>>  @>>>>>>>    @>>>>>>>  
+$date,           $nfiles,    $nbytes,     $avgrate,   $pctfiles,  $pctbytes
+.
+
+$^ = top1;
+$~ = line1;
+
+# sort daily traffic by bytes sendt
+#foreach $date ( sort datecompare keys(nbytes) ) {
+foreach $date ( sort datecompare keys(xferbytes) ) {
+
+   $nfiles = $xferfiles{$date};
+   $nbytes = $xferbytes{$date};
+   if ($xfersecs{$date}) {
+      $avgrate = sprintf("%5.1f KB/s", $xferbytes{$date}/$xfersecs{$date}/1024);
+   } else {
+      $avgrate = sprintf("undefined");
+   }
+   $pctfiles = sprintf("%8.2f", 100*$xferfiles{$date}/$xferfiles);
+   $pctbytes = sprintf("%8.2f", 100*$xferbytes{$date}/$xferbytes);
+   write;
+}
+
+if ($opt_t) {
+format top2 =
+
+Total Transfers from each Archive Section (By bytes)
+
+                                                   ---- Percent  Of ----
+     Archive Section      Files Sent Bytes Sent    Files Sent Bytes Sent
+------------------------- ---------- ------------- ---------- ----------
+.
+
+format line2 =
+@<<<<<<<<<<<<<<<<<<<<<<<< @>>>>>>>>> @>>>>>>>>>>>> @>>>>>>>   @>>>>>>>
+$section,                 $files,    $bytes,       $pctfiles, $pctbytes
+.
+
+$| = 1;
+$- = 0;
+$^ = top2;
+$~ = line2;
+
+# sort total transfer for each archive by # files transfered
+foreach $section ( sort bytecompare keys(groupfiles) ) {
+
+   $files = $groupfiles{$section};
+   $bytes = $groupbytes{$section};
+   $pctbytes = sprintf("%8.2f", 100 * $groupbytes{$section} / $xferbytes);
+   $pctfiles = sprintf("%8.2f", 100 * $groupfiles{$section} / $xferfiles);
+   write;
+
+}
+
+}
+
+if ($opt_d) {
+format top3 =
+
+Total Transfer Amount By Domain
+
+             Number Of    Number of     Average    Percent Of  Percent Of
+Domain Name  Files Sent   Bytes Sent   Xmit  Rate  Files Sent  Bytes Sent
+-----------  ----------  ------------  ----------  ----------  ----------
+.
+
+format line3 =
+@<<<<<<<<<<  @>>>>>>>>>  @>>>>>>>>>>>  @>>>>>>>>>  @>>>>>>>    @>>>>>>>  
+$domain,     $files,     $bytes,       $avgrate,   $pctfiles,  $pctbytes
+.
+
+$- = 0;
+$^ = top3;
+$~ = line3;
+
+# sort amount per domain by files
+foreach $domain ( sort domnamcompare keys(domainfiles) ) {
+
+   $files = $domainfiles{$domain};
+   $bytes = $domainbytes{$domain};
+   if ($domainsecs{$domain}) {
+      $avgrate = sprintf("%5.1f KB/s", $domainbytes{$domain}/$domainsecs{$domain}/1024);
+   } else {
+      $avgrate = sprintf("undefined");
+   }
+   $pctfiles = sprintf("%8.2f", 100 * $domainfiles{$domain} / $xferfiles);
+   $pctbytes = sprintf("%8.2f", 100 * $domainbytes{$domain} / $xferbytes);
+   write;
+
+}
+
+print "\n";
+
+}
+
+if ($opt_h) {
+
+format top8 =
+
+Hourly Transmission Statistics
+
+                 Number Of    Number of    Average    Percent Of  Percent Of
+     Time        Files Sent  Bytes  Sent  Xmit  Rate  Files Sent  Bytes Sent
+---------------  ----------  -----------  ----------  ----------  ----------
+.
+
+format line8 =
+@<<<<<<<<<<<<<<  @>>>>>>>>>  @>>>>>>>>>>  @>>>>>>>>>  @>>>>>>>    @>>>>>>>  
+$time,           $nfiles,    $nbytes,     $avgrate,   $pctfiles,  $pctbytes
+.
+
+
+$| = 1;
+$- = 0;
+$^ = top8;
+$~ = line8;
+
+# sort hourly transmission by sent bytes
+foreach $time ( sort keys(xfertbytes) ) {
+
+   $nfiles   = $xfertfiles{$time};
+   $nbytes   = $xfertbytes{$time};
+   if ($xfertsecs{$time}) {
+      $avgrate  = sprintf("%5.1f KB/s", $xfertbytes{$time}/$xfertsecs{$time}/1024);
+   } else {
+      $avgrate  = sprintf("undefined");
+   }
+   $pctfiles = sprintf("%8.2f", 100*$xfertfiles{$time} / $xferfiles);
+   $pctbytes = sprintf("%8.2f", 100*$xfertbytes{$time} / $xferbytes);
+   write;
+}
+}
+exit(0);
+
+sub datecompare {
+
+   $date1  = substr($a, 11, 4) * 4800;
+   $date2  = substr($b, 11, 4) * 4800;
+   $date1 += index("JanFebMarAprMayJunJulAugSepOctNovDec",substr($a, 4, 3)) / 3 * 400;
+   $date2 += index("JanFebMarAprMayJunJulAugSepOctNovDec",substr($b, 4, 3)) / 3 * 400;
+   $date1 += substr($a, 8, 2);
+   $date2 += substr($b, 8, 2);
+   $date1 - $date2;
+
+}
+
+sub domnamcompare {
+
+   $sdiff = length($a) - length($b);
+   ($sdiff < 0) ? -1 : ($sdiff > 0) ? 1 : ($a lt $b) ? -1 : ($a gt $b) ? 1 : 0;
+
+}
+
+sub bytecompare {
+
+   $bdiff = $groupbytes{$b} - $groupbytes{$a};
+   ($bdiff < 0) ? -1 : ($bdiff > 0) ? 1 : ($a lt $b) ? -1 : ($a gt $b) ? 1 : 0;
+
+}
+
+sub faccompare {
+
+   $fdiff = $fac{$b} - $fac{$a};
+   ($fdiff < 0) ? -1 : ($fdiff > 0) ? 1 : ($a lt $b) ? -1 : ($a gt $b) ? 1 : 0;
+
+}
+
diff -ruN proftpd-1.2.9.orig/include/default_paths.h proftpd-1.2.9/include/default_paths.h
--- proftpd-1.2.9.orig/include/default_paths.h	Thu Jan  2 10:28:15 2003
+++ proftpd-1.2.9/include/default_paths.h	Thu Nov  6 09:10:53 2003
@@ -36,7 +36,7 @@
  * ftpcount, ftpwho, etc.
  */
 #ifndef RUN_DIR
-#define RUN_DIR			"/var/run/proftpd"
+#define RUN_DIR			"@FINKPREFIX@/var/run/proftpd"
 #endif
 
 /* The location you wish to place any core files produced as a result of
@@ -50,37 +50,37 @@
  * when run in standalone daemon mode.
  */
 #ifndef PID_FILE_PATH
-#define PID_FILE_PATH		"/var/run/proftpd.pid"
+#define PID_FILE_PATH		"@FINKPREFIX@/var/run/proftpd.pid"
 #endif
 
 /* The default location of the proftpd configuration file.  Can be
  * overriden at runtime with the '-c' switch
  */
 #ifndef CONFIG_FILE_PATH
-#define	CONFIG_FILE_PATH	"/etc/proftpd.conf"
+#define	CONFIG_FILE_PATH	"@FINKPREFIX@/etc/proftpd.conf"
 #endif
 
 /* The location of your `shells' file; a newline delimited list of
  * valid shells on your system.
  */
-#define VALID_SHELL_PATH	"/etc/shells"
+#define VALID_SHELL_PATH	"@FINKPREFIX@/etc/shells"
 
 /* Where your log files are kept.  The "wu-ftpd style" xferlog is
  * stored here, as well as "extended" (not yet available) transfer
  * log files.  These can be overridden in the configuration file via
  * "TransferLog" and "ExtendedLog".  (note ExtendedLog does not work yet)
  */
-#define XFERLOG_PATH		"/var/log/xferlog"
-#define EXTLOG_PATH		"/var/log/proftpd-log"
+#define XFERLOG_PATH		"@FINKPREFIX@/var/log/xferlog"
+#define EXTLOG_PATH		"@FINKPREFIX@/var/log/proftpd-log"
 
 /* Location of the file that tells proftpd to discontinue servicing
  * requests.
  */
-#define SHUTMSG_PATH		"/etc/shutmsg"
+#define SHUTMSG_PATH		"@FINKPREFIX@/etc/shutmsg"
 
 /* Location of the file containing users that *cannot* use ftp
  * services (odd, eh?)
  */
-#define FTPUSERS_PATH		"/etc/ftpusers"
+#define FTPUSERS_PATH		"@FINKPREFIX@/etc/ftpusers"
 
 #endif /* PROFTPD_PATHS_H */
diff -ruN proftpd-1.2.9.orig/sample-configurations/basic.conf proftpd-1.2.9/sample-configurations/basic.conf
--- proftpd-1.2.9.orig/sample-configurations/basic.conf	Tue Feb 18 01:16:20 2003
+++ proftpd-1.2.9/sample-configurations/basic.conf	Thu Nov  6 09:32:22 2003
@@ -7,6 +7,11 @@
 ServerType			standalone
 DefaultServer			on
 
+# Fix for pam auth on OSX
+AuthPAMConfig                   ftpd
+PersistentPasswd                off
+RequireValidShell               off
+
 # Port 21 is the standard FTP port.
 Port				21
 
@@ -23,12 +28,12 @@
 MaxInstances			30
 
 # Set the user and group under which the server will run.
-User				nobody
-Group				nogroup
+User				daemon
+Group				daemon
 
 # To cause every FTP user to be "jailed" (chrooted) into their home
 # directory, uncomment this line.
-#DefaultRoot ~
+DefaultRoot ~
 
 # Normally, we want files to be overwriteable.
 AllowOverwrite		on
diff -ruN proftpd-1.2.9.orig/src/dirtree.c proftpd-1.2.9/src/dirtree.c
--- proftpd-1.2.9.orig/src/dirtree.c	Sat Oct 18 16:14:14 2003
+++ proftpd-1.2.9/src/dirtree.c	Thu Nov  6 09:10:53 2003
@@ -754,10 +754,10 @@
 
     /* Perform a fixup on configuration directives so that:
      * -argv[0]--  -argv[1]-- ----argv[2]-----
-     * <Option     /etc/adir  /etc/anotherdir>
+     * <Option     @FINKPREFIX@/etc/adir  @FINKPREFIX@/etc/anotherdir>
      *   .. becomes ..
      * -argv[0]--  -argv[1]-  ----argv[2]----
-     * <Option>    /etc/adir  /etc/anotherdir
+     * <Option>    @FINKPREFIX@/etc/adir  @FINKPREFIX@/etc/anotherdir
      */
 
     if (new_cmd->argc && *(new_cmd->argv[0]) == '<') {
@@ -2543,12 +2543,12 @@
        * Just a partial match on the pathname does not mean that the longer
        * path is the subdirectory of the other -- they might just be sharing
        * the last path component!
-       * /var/www/.1
-       * /var/www/.14
-       *            ^ -- not /, not subdir
-       * /var/www/.1
-       * /var/www/.1/images
-       *            ^ -- /, is subdir
+       * @FINKPREFIX@/var/www/.1
+       * @FINKPREFIX@/var/www/.14
+       *                        ^ -- not /, not subdir
+       * @FINKPREFIX@/var/www/.1
+       * @FINKPREFIX@/var/www/.1/images
+       *                        ^ -- /, is subdir
        */
       if (strlen(path) > len && path[len] != '/')
           continue;
diff -ruN proftpd-1.2.9.orig/src/main.c proftpd-1.2.9/src/main.c
--- proftpd-1.2.9.orig/src/main.c	Wed Oct 22 01:40:38 2003
+++ proftpd-1.2.9/src/main.c	Thu Nov  6 09:10:53 2003
@@ -1424,7 +1424,7 @@
     }
 
     /* If running (a flag signaling whether proftpd is just starting up)
-     * AND shutdownp (a flag signalling the present of /etc/shutmsg) are
+     * AND shutdownp (a flag signalling the present of @FINKPREFIX@/etc/shutmsg) are
      * true, then log an error stating this -- but don't stop the server.
      */
     if (shutdownp && !running) {
@@ -2220,7 +2220,9 @@
   /* setsid() is the preferred way to disassociate from the
    * controlling terminal
    */
+  log_pri(PR_LOG_ERR, "-> Before setsid");
   setsid();
+  log_pri(PR_LOG_ERR, "-> After setsid");
 #else
   /* Open /dev/tty to access our controlling tty (if any) */
   if ((ttyfd = open("/dev/tty", O_RDWR)) != -1) {
diff -ruN proftpd-1.2.9.orig/src/xferlog.5 proftpd-1.2.9/src/xferlog.5
--- proftpd-1.2.9.orig/src/xferlog.5	Fri Oct 31 01:27:24 2003
+++ proftpd-1.2.9/src/xferlog.5	Thu Nov  6 09:10:53 2003
@@ -151,7 +151,7 @@
 .SH FILES
 .PD 0
 .TP 20
-.B /var/log/xferlog
+.B @FINKPREFIX@/var/log/xferlog
 .PP
 .PD
 .SH AUTHORS
diff -ruN proftpd-1.2.9.orig/utils/ftpshut.8 proftpd-1.2.9/utils/ftpshut.8
--- proftpd-1.2.9.orig/utils/ftpshut.8	Fri Oct 31 01:27:24 2003
+++ proftpd-1.2.9/utils/ftpshut.8	Thu Nov  6 09:10:53 2003
@@ -23,7 +23,7 @@
 command provides an automated shutdown procedure that a superuser
 can use to automagically disconnect current proftpd connections,
 and deny new connections.  The command creates a control file as
-.B /etc/shutmsg
+.B @FINKPREFIX@/etc/shutmsg
 to inform proftpd processes about an impending shutdown.
 .SH OPTIONS
 .TP 10
