--- cmine/mine.c	Mon Feb 18 07:45:53 2002
+++ cmine-patched/mine.c	Sun May 12 23:00:19 2002
@@ -94,7 +94,7 @@
 		return NULL;
 	}
 	/* Place the mines */
-	srandomdev();
+	srandom(time(0));
 	for (i = 0; i < rows * cols; i ++)
 		m->map[i] = 0;
 	p = 0;
--- cmine/hiscore.c	Sat Feb 16 05:57:51 2002
+++ cmine-patched/hiscore.c	Sun May 12 23:15:53 2002
@@ -18,7 +18,9 @@
 
 #include "hiscore.h"
 
+#ifndef PATH_SCOREFILE
 #define PATH_SCOREFILE	"/var/games/cmine.scores"
+#endif
 
 struct scorekey {
 	int rows, cols, mines;
