--- funny-manpages-1.3.orig/copyright
+++ funny-manpages-1.3/copyright
@@ -0,0 +1,32 @@
+This package was debianized by Pawel Wiecek coven@pwr.wroc.pl on
+Wed, 10 Dec 1997 01:10:17 +0100.
+
+This set of manpages was collected from all over the net. No specific
+location can be given.
+
+Copyright:
+
+To the best of my knowledge all of these manpages are free to use and
+redistribute.
+
+The authors are:
+
+baby.1fun	- unknown, based on man page by Joe Beck <beck@cs.ualberta.ca>
+celibacy.1fun	- unknown
+condom.1fun	- Ken Maupin <maupin@cs.washington.edu>
+flame.1fun	- unknown
+flog.1fun	- unknown
+gong.1fun	- unknown
+grope.1fun	- unknown
+party.1fun	- unknown
+rescrog.1fun	- unknown
+rtfm.1fun	- unknown
+sex.1fun	- unknown
+tm.1fun		- unknown
+xlart.1fun	- James McPherson
+date.1fun	- Glen Overby <overby@sendit.nodak.edu>
+echo.1fun	- unknown
+rm.1fun		- Matthew Farwell <dylan@ibmpcug.co.uk>
+strfry.3fun	- <chuck@druco.att.com>
+xkill.1fun	- Claudio Calvelli <Claudio@edinburgh.ac.uk>
+uubp.1fun	- unknown
