--- gtkpool-0.5.0/gtkpool/sound.cpp.orig	2003-04-08 15:50:29.000000000 -0700
+++ gtkpool-0.5.0/gtkpool/sound.cpp	2003-04-08 16:29:16.000000000 -0700
@@ -20,7 +20,9 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <unistd.h>
+#if !defined(__APPLE__)
 #include <sys/soundcard.h>
+#endif
 #include <sys/ioctl.h>
 #include <stdio.h>
 #include <errno.h>
@@ -49,6 +51,7 @@
 
 void play_sound(char *sound, int count)
 {
+#if !defined(__APPLE__)
 	//int fragment = 0x20008;
 	if(!sound_enabled)
 	{
@@ -68,14 +71,19 @@
 	}
 	
 	ioctl(audio_fd, SNDCTL_DSP_POST, 0);
+#endif
 }
 
 int load_sound(char *file, char *buffer, int *count)
 {
+#if defined(__APPLE__)
+        return 0;
+#else
 	int fd = open(file, O_RDONLY);
 	*count = read(fd, buffer, *count);
 	close(fd);
 	return *count;
+#endif
 }
 
 void open_sound_device(char *name)
@@ -92,6 +100,9 @@
 
 void configure_sound_device()
 {
+#if defined(__APPLE__)
+	sound_enabled = false;
+#else
 	int fragment = 0x00020008;
 	int format = AFMT_U8;
 	int channels = 2;
@@ -111,4 +122,5 @@
 	}
 	
 	sound_enabled = true;
+#endif
 }
