diff -Nurd automoc4-0.9.88/Automoc4Config.cmake automoc4-0.9.88-new/Automoc4Config.cmake
--- automoc4-0.9.88/Automoc4Config.cmake	2009-01-22 13:50:09.000000000 -0500
+++ automoc4-0.9.88-new/Automoc4Config.cmake	2009-03-03 10:40:11.000000000 -0500
@@ -105,9 +105,9 @@
       if(NOT  _generated  AND NOT  _skip)
          get_filename_component(_suffix "${_current_FILE}" EXT)
          # skip every source file that's not C++
-         if(_suffix STREQUAL ".cpp" OR _suffix STREQUAL ".cc" OR _suffix STREQUAL ".cxx" OR _suffix STREQUAL ".C")
+         if(_suffix STREQUAL ".cpp" OR _suffix STREQUAL ".cc" OR _suffix STREQUAL ".cxx" OR _suffix STREQUAL ".C" OR _suffix STREQUAL ".mm")
              list(APPEND _moc_files ${_abs_current_FILE})
-         endif(_suffix STREQUAL ".cpp" OR _suffix STREQUAL ".cc" OR _suffix STREQUAL ".cxx" OR _suffix STREQUAL ".C")
+         endif(_suffix STREQUAL ".cpp" OR _suffix STREQUAL ".cc" OR _suffix STREQUAL ".cxx" OR _suffix STREQUAL ".C" OR _suffix STREQUAL ".mm")
       endif(NOT  _generated  AND NOT  _skip)
    endforeach (_current_FILE)
 
@@ -156,7 +156,7 @@
       if(NOT  _generated  AND NOT  _skip)
          get_filename_component(_suffix "${_current_FILE}" EXT)
          # skip every source file that's not C++
-         if(_suffix STREQUAL ".cpp" OR _suffix STREQUAL ".cc" OR _suffix STREQUAL ".cxx" OR _suffix STREQUAL ".C")
+         if(_suffix STREQUAL ".cpp" OR _suffix STREQUAL ".cc" OR _suffix STREQUAL ".cxx" OR _suffix STREQUAL ".C" OR _suffix STREQUAL ".mm")
              get_filename_component(_basename "${_current_FILE}" NAME_WE)
              get_filename_component(_abs_path "${_abs_current_FILE}" PATH)
              set(_header "${_abs_path}/${_basename}.h")
@@ -168,7 +168,7 @@
                 list(APPEND _moc_headers ${_pheader})
              endif(EXISTS "${_pheader}")
              list(APPEND _moc_files ${_abs_current_FILE})
-         endif(_suffix STREQUAL ".cpp" OR _suffix STREQUAL ".cc" OR _suffix STREQUAL ".cxx" OR _suffix STREQUAL ".C")
+         endif(_suffix STREQUAL ".cpp" OR _suffix STREQUAL ".cc" OR _suffix STREQUAL ".cxx" OR _suffix STREQUAL ".C" OR _suffix STREQUAL ".mm")
       endif(NOT  _generated  AND NOT  _skip)
    endforeach (_current_FILE)
 
diff -Nurd automoc4-0.9.88/kde4automoc.cpp automoc4-0.9.88-new/kde4automoc.cpp
--- automoc4-0.9.88/kde4automoc.cpp	2009-01-22 13:50:09.000000000 -0500
+++ automoc4-0.9.88-new/kde4automoc.cpp	2009-03-03 10:27:08.000000000 -0500
@@ -326,7 +326,7 @@
     foreach (const QString &absFilename, sourceFiles) {
         //qDebug() << absFilename;
         const QFileInfo sourceFileInfo(absFilename);
-        if (absFilename.endsWith(".cpp") || absFilename.endsWith(".cc") ||
+        if (absFilename.endsWith(".cpp") || absFilename.endsWith(".cc") || absFilename.endsWith(".mm") ||
                 absFilename.endsWith(".cxx") || absFilename.endsWith(".C")) {
             //qDebug() << "check .cpp file";
             QFile sourceFile(absFilename);
