diff -ruN pccts/makefile pccts-patched/makefile
--- pccts/makefile	Wed Oct  3 21:35:34 2001
+++ pccts-patched/makefile	Sat Jan  5 21:25:21 2002
@@ -8,14 +8,15 @@
 PSss=
 #
 # this can be set to /user/local/bin etc...
+# tsstss that's not true, bin is hardcoded in the makefiles
 BINDIR=bin
 # This part added by Thomas Herter, M"unchen, Germany. See also manpages
 # target.
-MANDIR=$(HOME)/man
+MANDIR=@FINK_IPREFIX@/share/man
 MANEXT=1
 MANFILES=dlg/dlg.1 antlr/antlr.1
 
-#CC=cc
+CC=cc
 #CC=gcc
 #COPT=-O2
 
@@ -64,9 +65,9 @@
 
 
 manpages:
-	# mkdir -p $(MANDIR)/man$(MANEXT)
-	if [ ! -d $(MANDIR) ] ; then \
-	  mkdir $(MANDIR) ; fi
-	if [ ! -d $(MANDIR)/man$(MANEXT) ] ; then \
-	  mkdir $(MANDIR)/man$(MANEXT); fi
+	mkdir -p $(MANDIR)/man$(MANEXT)
+	#if [ ! -d $(MANDIR) ] ; then \
+	#  mkdir $(MANDIR) ; fi
+	#if [ ! -d $(MANDIR)/man$(MANEXT) ] ; then \
+	#  mkdir $(MANDIR)/man$(MANEXT); fi
 	cp -p $(MANFILES) $(MANDIR)/man$(MANEXT)
