diff -ruN reportbug/README.developers reportbug.new/README.developers
--- reportbug/README.developers	Fri Oct  1 04:09:29 2004
+++ reportbug.new/README.developers	Tue Jan 18 11:35:44 2005
@@ -10,7 +10,7 @@
 Template Information & Interaction with the user
 ================================================
 
- If /usr/share/bug/$package is executable, then bug executes it and
+ If @FINKPREFIX@/share/bug/$package is executable, then bug executes it and
 takes what comes out from the file descriptor 3 and puts it in the bug
 template.
  The maintainer can then ask questions to the user or run whatever
@@ -22,8 +22,8 @@
 	...
 	system-information-tool >&3
 	
- If /usr/share/bug/$package is a directory, then
-/usr/share/bug/$package/script is executed.
+ If @FINKPREFIX@/share/bug/$package is a directory, then
+@FINKPREFIX@/share/bug/$package/script is executed.
 
  While the script is executed, the following shell functions are
 available:
@@ -34,7 +34,7 @@
 					  in REPLY. The second argument is
 					  the default.
 
- If the file /usr/share/bug/$package/presubj exists, its content is
+ If the file @FINKPREFIX@/share/bug/$package/presubj exists, its content is
 shown to the user before asking him for the bug's subject.
 
  Note: It's your responsibility to check if the information included
@@ -48,7 +48,7 @@
 used to redirect bugs in packages coming from a single source to where the
 maintainer likes to have them.
 
- This is done by having this line in /usr/share/bug/$package/control:
+ This is done by having this line in @FINKPREFIX@/share/bug/$package/control:
 
 	Submit-As: $new-package
 
@@ -57,9 +57,9 @@
 BTS selection
 =============
 
- Packages not distributed by Debian can take advantage of this utility too.
+ Packages not distributed by Fink can take advantage of this utility too.
 They just need to add a "send-to" header to the control file
-/usr/share/bug/$package/control.
+@FINKPREFIX@/share/bug/$package/control.
 
 	Send-To: bugs.myproject.com
 
@@ -76,7 +76,7 @@
 ================
 
  Often programs are distributed across several different packages, for
-example an upstream package 'foo' may be packaged in Debian as foo, libfoo,
+example an upstream package 'foo' may be packaged in Fink as foo, libfoo,
 foo-common and foo-data.  In such cases it can be useful to include related
 package information in bugreports, to minimise the need for 'moreinfo' requests
 to the submitter :) This is done by adding a "report-with" header to the
@@ -90,7 +90,7 @@
 Addendum: Languages other than SH
 =================================
 
-The script in /usr/share/bug/reportbug/script is an example of a bug
+The script in @FINKPREFIX@/share/bug/reportbug/script is an example of a bug
 handling script written in Python.  You can also write bug handlers in
 many other languages that allow direct access to file descriptors,
 including Perl and C/C++.
diff -ruN reportbug/checkbuildd.py reportbug.new/checkbuildd.py
--- reportbug/checkbuildd.py	Wed Feb  4 23:29:10 2004
+++ reportbug.new/checkbuildd.py	Tue Jan 18 11:35:44 2005
@@ -26,7 +26,7 @@
 from urlutils import open_url
 from reportbug_exceptions import *
 
-BUILDD_URL = 'http://buildd.debian.org/build.php?arch=%s&pkg=%s'
+BUILDD_URL = 'http://buildd.finkproject.org/build.php?arch=%s&pkg=%s'
 
 # This is easy; just look for succeeded in an em block...
 
@@ -70,7 +70,8 @@
         arch = archname()
 
     try:
-        page = open_url(BUILDD_URL % (arch, src_package), http_proxy)
+#        page = open_url(BUILDD_URL % (arch, src_package), http_proxy)
+        return {}
     except NoNetwork:
         return {}
     if not page:
diff -ruN reportbug/checkversions.py reportbug.new/checkversions.py
--- reportbug/checkversions.py	Thu Sep 30 05:20:46 2004
+++ reportbug.new/checkversions.py	Tue Jan 18 11:35:44 2005
@@ -125,7 +125,7 @@
 
 def get_versions_available(package, dists=None, http_proxy=None, arch='i386'):
     if not dists:
-        dists = ('stable', 'testing', 'unstable')
+        dists = ('stable', 'unstable')
 
     try:
         page = open_url(PACKAGES_URL % package, http_proxy)
diff -ruN reportbug/debian/copyright reportbug.new/debian/copyright
--- reportbug/debian/copyright	Wed Feb  4 23:29:12 2004
+++ reportbug.new/debian/copyright	Tue Jan 18 11:35:44 2005
@@ -23,4 +23,4 @@
 #  SOFTWARE.
 
 The "handle_bugscript" script is subject to the GNU General Public
-License.  See /usr/share/common-licenses/GPL.
+License.  See @FINKPREFIX@/share/common-licenses/GPL.
diff -ruN reportbug/debian/install-sed reportbug.new/debian/install-sed
--- reportbug/debian/install-sed	Mon Jul 12 14:30:56 2004
+++ reportbug.new/debian/install-sed	Tue Jan 18 11:35:44 2005
@@ -4,5 +4,5 @@
 shift
 
 for filename in $@; do
-    sed -f sed-script < $filename > `pwd`/debian/reportbug/$dir/$filename
+    sed -f sed-script < $filename > $dir/$filename
 done
diff -ruN reportbug/debianbts.py reportbug.new/debianbts.py
--- reportbug/debianbts.py	Fri Oct  1 04:12:02 2004
+++ reportbug.new/debianbts.py	Tue Jan 18 11:35:44 2005
@@ -40,7 +40,7 @@
     'grave' : """makes the package in question unusable by most or all users,
     or causes data loss, or introduces a security hole allowing access 
     to the accounts of users who use the package.""",
-    'serious' : """is a severe violation of Debian policy (that is,
+    'serious' : """is a severe violation of Fink policy (that is,
     the problem is a violation of a 'must' or 'required' directive);
     may or may not affect the usability of the package.  Note that non-severe
     policy violations may be 'normal,' 'minor,' or 'wishlist' bugs.
@@ -131,39 +131,45 @@
     else:
         return severity
 
-# These packages are virtual in Debian; we don't look them up...
-debother = {
+# These packages are virtual in Fink; we don't look them up...
+finkother = {
     'base' : 'General bugs in the base system',
-# Actually a real package, but most people don't have boot-floppies installed for good reason
-    'boot-floppy' : '(Obsolete, please use boot-floppies instead.)',
-    'boot-floppies' : 'Bugs in the woody installation subsystem',
-    'bugs.debian.org' : 'The bug tracking system, @bugs.debian.org',
-    'cdimage.debian.org' : 'CD Image issues',
-    'cdrom' : 'Problems with installation from CD-ROMs',
-# dpkg-iwj -- The dpkg branch maintained by Ian Jackson
-    'debian-policy' : 'Proposed changes in the Debian policy documentation',
-    'ftp.debian.org' : 'Problems with the FTP site',
+    'bugs.finkproject.org' : 'The bug tracking system, @bugs.finkproject.org',
+    'fink-policy' : 'Proposed changes in the Fink policy documentation',
     'general' : 'General problems (e.g., that many manpages are mode 755)',
-    'install' : 'Problems with the sarge installer.',
+    'install' : 'Problems with the fink installer.',
     'installation' : 'General installation problems not covered otherwise.',
-    'kernel' : '(Obsolete, please use "kernel-image" instead.)',
-    'kernel-image' : 'Problems with the Linux kernel, or the kernel shipped with Debian',
     'listarchives' :  'Problems with the WWW mailing list archives',
-    'lists.debian.org' : 'The mailing lists, debian-*@lists.debian.org.',
-    'mirrors' : 'Problems with Debian archive mirrors.',
-    'nonus.debian.org' : 'Problems with the non-US FTP site',
+    'lists.finkproject.org' : 'The mailing lists, fink-*@lists.finkproject.org.',
+    'mirrors' : 'Problems with Fink archive mirrors.',
     'press' : 'Press release issues',
     'project' : 'Problems related to Project administration',
-    'qa.debian.org' : 'Problems related to the quality assurance group',
-#slink-cd -- Slink CD 
-#spam -- Spam (reassign spam to here so we can complain about it)
-    'security.debian.org' : 'Problems with the security updates server',
-    'wnpp' : 'Work-Needing and Prospective Packages list',
-    'www.debian.org' : 'Problems with the WWW site (including other *.debian.org sites)'
+    'fink.sourceforge.net' : 'Problems with the WWW site (including other *.finkproject.org sites)',
+    'broken-gcc' : 'Problems with the broken-gcc virtual package',
+    'cctools' : 'Problems with the cctools virtual package',
+    'cctools-single-module' : 'Problems with the cctools-single-module virtual package',
+    'darwin' : 'Problems with the darwin virtual package',
+    'gcc2' : 'Problems with the gcc2 virtual package',
+    'gcc2.95' : 'Problems with the gcc2.95 virtual package',
+    'gcc3.1' : 'Problems with the gcc3.1 virtual package',
+    'gcc3.3' : 'Problems with the gcc3.3 virtual package',
+    'gimp-print-shlibs' : 'Problems with the gimp-print-shlibs virtual package',
+    'macosx' : 'Problems with the macosx virtual package',
+    'system-java' : 'Problems with the system-java virtual package',
+    'system-java-dev' : 'Problems with the system-java-dev virtual package',
+    'system-java12' : 'Problems with the system-java12 virtual package',
+    'system-java13' : 'Problems with the system-java13 virtual package',
+    'system-java13-dev' : 'Problems with the system-java13-dev virtual package',
+    'system-java14' : 'Problems with the system-java14 virtual package',
+    'system-java14-dev' : 'Problems with the system-java14-dev virtual package',
+    'system-java15' : 'Problems with the system-java15 virtual package',
+    'system-java3d' : 'Problems with the system-java3d virtual package',
+    'system-javaai' : 'Problems with the system-javaai virtual package',
+    'system-perl' : 'Problems with the system-perl virtual package',
     }
 
 progenyother = {
-    'debian-general' : 'Any non-package-specific bug',
+    'fink-general' : 'Any non-package-specific bug',
     }
 
 def handle_wnpp(package, bts, ui, online=True, http_proxy=None):
@@ -184,7 +190,7 @@
         'ITP' :
         "This is an `Intent To Package'. Please submit a package description along with copyright and URL in such a report.",
         'RFP' :
-        "This is a `Request For Package'. You have found an interesting piece of software and would like someone else to maintain it for Debian. Please submit a package description along with copyright and URL in such a report.",
+        "This is a `Request For Package'. You have found an interesting piece of software and would like someone else to maintain it for Fink. Please submit a package description along with copyright and URL in such a report.",
         }, 'Choose the request type: ', empty_ok=True)
     if not tag:
         ui.long_message('To report a bug in a package, use the name of the package, not wnpp.\n')
@@ -223,9 +229,9 @@
             return
 
         if tag == 'ITP':
-            headers.append('X-Debbugs-CC: debian-devel@lists.debian.org')
-            ui.ewrite('Your report will be carbon-copied to debian-devel, '
-                      'per Debian policy.\n')
+            headers.append('X-Debbugs-CC: fink-devel@lists.sourceforge.net')
+            ui.ewrite('Your report will be carbon-copied to fink-devel, '
+                      'per Fink policy.\n')
 
         body = """* Package name    : %s
   Version         : x.y.z
@@ -262,9 +268,9 @@
             severity = 'important'
 
         if tag == 'RFH':
-            headers.append('X-Debbugs-CC: debian-devel@lists.debian.org')
-            ui.ewrite('Your request will be carbon-copied to debian-devel, '
-                      'per Debian policy.\n')
+            headers.append('X-Debbugs-CC: fink-devel@lists.sourceforge.net')
+            ui.ewrite('Your request will be carbon-copied to fink-devel, '
+                      'per Fink policy.\n')
 
         if fulldesc:
             orphstr = 'intend to orphan'
@@ -287,17 +293,14 @@
 # Supported servers
 # Theoretically support for GNATS and Jitterbug could be added here.
 
-SYSTEMS = { 'debian' :
-            { 'name' : 'Debian', 'email': '%s@bugs.debian.org',
-              'btsroot' : 'http://www.debian.org/Bugs/',
+SYSTEMS = { 'fink' :
+            { 'name' : 'Fink', 'email': '%s@bugs.finkproject.org',
+              'btsroot' : 'http://bugs.finkproject.org',
               'query-dpkg' : 1, 'type' : 'debbugs',
-              'otherpkgs' : debother, 'nonvirtual' : ['kernel-image'],
-              'specials' : { 'wnpp': handle_wnpp },
+              'otherpkgs' : finkother, 
+              #'specials' : { 'wnpp': handle_wnpp },
               # Dependency packages
-              'deppkgs' : ('gcc', 'g++', 'cpp', 'gcj', 'gpc', 'gobjc',
-                           'chill', 'gij', 'g77', 'python', 'python-base',
-                           'x-window-system-core', 'x-window-system'),
-              'cgiroot' : 'http://bugs.debian.org/cgi-bin/',
+              'cgiroot' : 'http://bugs.finkproject.org/cgi-bin/',
               'mirrors' : {} },
             'kde' :
             { 'name' : 'KDE Project', 'email': '%s@bugs.kde.org',
@@ -350,7 +353,7 @@
 CLASSLIST = ['sw-bug', 'doc-bug', 'change-request']
 
 CRITICAL_TAGS = {
-    'security' : 'This problem is a security vulnerability in Debian.',
+    'security' : 'This problem is a security vulnerability in Fink.',
 }
 
 TAGS = {
@@ -361,12 +364,12 @@
 ##    'sarge' : 'This bug only applies to the sarge release (Debian 3.1).',
 ##    'sid' : 'This bug only applies to the unstable branch of Debian.',
     'experimental' : 'This bug only applies to a package in the experimental '
-    'branch of Debian.',
+    'branch of Fink.',
     "l10n" : "This bug reports a localization/internationalization issue.",
 ##    'done' : 'No more tags.',
     }
 
-EXTRA_TAGS = ['potato', 'woody', 'sarge', 'security', 'sid', 'upstream']
+EXTRA_TAGS = ['stable', 'unstable', 'upstream']
 
 TAGLIST = ['l10n', 'patch', 'experimental']
 CRITICAL_TAGLIST = ['security']
@@ -432,7 +435,7 @@
     return type, loc
 
 # Dynamically add any additional systems found
-for origin in glob.glob('/etc/dpkg/origins/*'):
+for origin in glob.glob('@FINKPREFIX@/etc/dpkg/origins/*'):
     try:
         fp = file(origin)
         system = os.path.basename(origin)
@@ -613,7 +616,7 @@
 
     return (title, output)
 
-def get_cgi_reports(package, system='debian', http_proxy='', archived=False,
+def get_cgi_reports(package, system='fink', http_proxy='', archived=False,
                     source=False):
     page = open_url(cgi_package_url(system, package, archived, source),
                     http_proxy)
@@ -630,7 +633,7 @@
 
     return parser.bugcount, parser.title, parser.hierarchy
 
-def get_cgi_report(number, system='debian', http_proxy='', archived=False,
+def get_cgi_report(number, system='fink', http_proxy='', archived=False,
                    followups=False):
     number = int(number)
 
@@ -645,7 +648,7 @@
                 return alternates[mirror]
     return SYSTEMS[system]['btsroot']
 
-def get_reports(package, system='debian', mirrors=None,
+def get_reports(package, system='fink', mirrors=None,
                 http_proxy='', archived=False, source=False):
     if isinstance(package, StringTypes):
         if SYSTEMS[system]['cgiroot']:
@@ -685,7 +688,7 @@
 
     return bugcount, title, hierarchy
 
-def get_report(number, system='debian', mirrors=None,
+def get_report(number, system='fink', mirrors=None,
                http_proxy='', archived=False, followups=False):
     number = int(number)
     if SYSTEMS[system]['cgiroot']:
diff -ruN reportbug/querybts reportbug.new/querybts
--- reportbug/querybts	Mon Mar 15 03:59:39 2004
+++ reportbug.new/querybts	Tue Jan 18 11:35:44 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.3
+#!@FINKPREFIX@/bin/python2.6
 # -*- python -*-
 # querybts - Examine the state of a debbugs server
 #   Written by Chris Lawrence <lawrencc@debian.org>
@@ -25,7 +25,7 @@
 # $Id: querybts,v 1.2 2004/03/15 08:59:39 lawrencc Exp $
 
 import sys, os
-sys.path = [os.curdir, '/usr/share/reportbug'] + sys.path
+sys.path = [os.curdir, '@FINKPREFIX@/share/reportbug'] + sys.path
 
 import reportbug, debianbts, commands, getopt, re, mailcap, urlutils
 
@@ -53,7 +53,7 @@
          )
 
 def main():
-    system = 'debian'
+    system = 'fink'
     archived = False
     http_proxy = interface = ''
     use_browser = source = False
diff -ruN reportbug/reportbug reportbug.new/reportbug
--- reportbug/reportbug	Fri Dec 17 23:01:23 2004
+++ reportbug.new/reportbug	Tue Jan 18 11:46:59 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.3 -S
+#!@FINKPREFIX@/bin/python2.6 -S
 # -*- python -*-
 # reportbug - Report a bug in the Debian distribution.
 #   Written by Chris Lawrence <lawrencc@debian.org>
@@ -26,11 +26,11 @@
 
 import site
 
-DEBUGGER = 'lawrencc@debian.org'
-DEFAULT_BTS = 'debian'
+DEBUGGER = 'beren12@bugs.finkproject.org'
+DEFAULT_BTS = 'fink'
 
 import sys, os
-sys.path = [os.curdir, '/usr/share/reportbug'] + sys.path
+sys.path = [os.curdir, '@FINKPREFIX@/share/reportbug'] + sys.path
 
 import optparse, re, os, pwd, time, locale, commands, checkversions
 import rfc822, smtplib, reportbug, cStringIO, socket, debianbts, checkbuildd
@@ -94,7 +94,7 @@
 def spawn_editor(message, filename, default_editor=None):
     if not default_editor:
         editor = (os.environ.get('VISUAL') or os.environ.get('EDITOR') or
-                  '/usr/bin/sensible-editor')
+                  '@FINKPREFIX@/bin/sensible-editor')
     else:
         editor = default_editor
 
@@ -324,7 +324,7 @@
                    "categories:", others, "Enter a package: ", any_ok=True,
                    default='')
 
-def get_package_name(bts='debian', mode=MODE_EXPERT):
+def get_package_name(bts='fink', mode=MODE_EXPERT):
     others = debianbts.SYSTEMS[bts].get('otherpkgs')
     prompt = "Please enter the name of the package in which you have found "\
              "a problem"
@@ -357,14 +357,14 @@
                 return get_package_name(bts, mode)
 
     if mode < MODE_EXPERT:
-        if package in ('bugs.debian.org', 'debbugs'):
+        if package in ('bugs.finkproject.org', 'debbugs'):
             if ui.yes_no('Are you reporting a problem with this program '
                          '(reportbug)', 'Yes, this is actually a bug in '
                          'reportbug.', 'No, this is really a problem in the '
                          'bug tracking system itself.'):
                 package = 'reportbug'
 
-        if package in ('general', 'project', 'debian-general', 'base'):
+        if package in ('general', 'project', 'fink-general', 'base'):
             if not ui.yes_no(
                 "Are you sure this bug doesn't apply to a specific package?",
                 'Yes, this bug is truly general.',
@@ -489,16 +489,16 @@
         print >> fp, '# Disable fallback mode by commenting out the following:'
         print >> fp, 'no-cc'
         print >> fp, 'header "X-Debbugs-CC: %s"' % email_addy
-        print >> fp, 'smtphost master.debian.org'
+        print >> fp, 'smtphost bugs.finkproject.org'
     else:
         print >> fp, '# If nothing else works, remove the # at the beginning'
         print >> fp, '# of the following three lines:'
         print >> fp, '#no-cc'
         print >> fp, '#header "X-Debbugs-CC: %s"' % email_addy
-        print >> fp, '#smtphost master.debian.org'
+        print >> fp, '#smtphost bugs.finkproject.org'
         
     print >> fp, '# You can add other settings after this line.  See'
-    print >> fp, '# /etc/reportbug.conf for a full listing of options.'
+    print >> fp, '# @FINKPREFIX@/etc/reportbug.conf for a full listing of options.'
     fp.close()
     ui.long_message('Default preferences file written.  To reconfigure, '
                     're-run reportbug with the "--configure" option.\n')
@@ -756,7 +756,7 @@
     severity = options.severity
     smtphost = options.smtphost
     subject = options.subject
-    bts = options.bts or 'debian'
+    bts = options.bts or 'fink'
     sysinfo = debianbts.SYSTEMS[bts]
     rtype = options.type or sysinfo['type']
     attachments = options.attachments
@@ -1072,7 +1072,7 @@
          origin, vendor, reportinfo, priority, desc, src_name,
          fulldesc, state) = status
 
-    buginfo = '/usr/share/bug/' + package
+    buginfo = '@FINKPREFIX@/share/bug/' + package
     bugexec = submitas = submitto = presubj = reportwith = None
     if os.path.isfile(buginfo) and os.access(buginfo, os.X_OK):
         bugexec = buginfo
@@ -1085,11 +1085,11 @@
 
         if os.path.isfile(buginfo+'/control'):
             submitas, submitto, reportwith = reportbug.parse_bug_control_file(buginfo+'/control')
-    elif os.path.isfile('/usr/share/bug/default/'+package) \
-         and os.access('/usr/share/bug/default/'+package, os.X_OK):
-        bugexec = '/usr/share/bug/default/'+package
-    elif os.path.isdir('/usr/share/bug/default/'+package):
-        buginfo = '/usr/share/bug/default/'+package
+    elif os.path.isfile('@FINKPREFIX@/share/bug/default/'+package) \
+         and os.access('@FINKPREFIX@/share/bug/default/'+package, os.X_OK):
+        bugexec = '@FINKPREFIX@/share/bug/default/'+package
+    elif os.path.isdir('@FINKPREFIX@/share/bug/default/'+package):
+        buginfo = '@FINKPREFIX@/share/bug/default/'+package
         if os.path.isfile(buginfo+'/script') and os.access(buginfo+'/script',
                                                            os.X_OK):
             bugexec = buginfo+'/script'
@@ -1112,10 +1112,10 @@
         # Remove current package from report-with list
         reportwith = [x for x in reportwith if x != package]
 
-    if (pkgavail and options.verify and os.path.exists('/usr/bin/debsums')
+    if (pkgavail and options.verify and os.path.exists('@FINKPREFIX@/bin/debsums')
         and not (notatty or options.kudos) and state == 'installed'):
         ewrite('Verifying package integrity...\n')
-        rc, output = commands.getstatusoutput('/usr/bin/debsums -s'+
+        rc, output = commands.getstatusoutput('@FINKPREFIX@/bin/debsums -s'+
                                               commands.mkarg(package))
         if rc:
             if not ui.yes_no(
@@ -1136,7 +1136,7 @@
                                        'package this report applies to '
                                        '(blank OK)', force_prompt=True)
     elif (check_available and not (options.kudos or notatty or options.offline)
-          and bts=='debian'):
+          and bts=='fink'):
         ewrite('Checking for newer versions at packages.debian.org...\n')
         arch = reportbug.get_arch()
         (avail, toonew) = checkversions.check_available(
@@ -1144,7 +1144,7 @@
             http_proxy=options.http_proxy, arch=arch)
         if toonew:
             if not ui.yes_no(
-                '\nYour version of %s (%s) is newer than that in Debian!\n'
+                '\nYour version of %s (%s) is newer than that in Fink!\n'
                 'Do you still want to file a report' % (package, pkgversion),
                 'Ignore this problem and continue.  This may be '
                 'appropriate if you know this bug is present in older '
@@ -1163,7 +1163,7 @@
 
             if not ui.yes_no(
                 ('\nYour version (%s) of %s appears to be out of date.\nThe '
-                'following newer release(s) are available in the Debian '
+                'following newer release(s) are available in the Fink '
                 'archive:\n' % (pkgversion, package))+availtext+
                 'Do you still want to file a report',
                 'Ignore this problem and continue.  This may be '
@@ -1268,13 +1268,13 @@
         lmaintainer = maintainer.decode('UTF-8', 'ignore').encode(charset,
                                                                   'replace')
         ewrite("Maintainer for %s is '%s'.\n", package, lmaintainer)
-        if 'qa.debian.org' in maintainer:
+        if 'fink-devel' in maintainer:
             ui.long_message('''\
-This package is currently "orphaned"; if you care a current or prospective
-Debian developer, you might consider adopting it.  Please be aware that your
+This package is currently "orphaned"; if you are a current or prospective
+Fink developer, you might consider adopting it.  Please be aware that your
 report may not be resolved for a while, and that packages that have been
 orphaned for a long period of time are often removed from the archive.\n''')
-            ui.ewrite('\nFor more details, please see: http://www.debian.org/devel/wnpp/\n')
+            ui.ewrite('\nFor more details, please see: http://fink.sourceforge.net/\n')
 
     if options.kudos:
         if maintainer:
@@ -1319,7 +1319,7 @@
                     confinfo[f] = 'changed [not included]'
                 break
             elif x == 'd':
-                PAGER = os.environ.get('PAGER', '/usr/bin/sensible-pager')
+                PAGER = os.environ.get('PAGER', '@FINKPREFIX@/bin/sensible-pager')
                 system(PAGER+' '+' '.join(changed))
             else:
                 break
@@ -1331,14 +1331,14 @@
         for f in files:
             conftext = conftext + '%s %s\n' % (f, confinfo[f])
 
-    if (options.debconf and os.path.exists('/usr/bin/debconf-show') and
+    if (options.debconf and os.path.exists('@FINKPREFIX@/bin/debconf-show') and
         not options.kudos and installed):
         showpkgs = package
         if reportwith:
             showpkgs += ' ' + ' '.join(reportwith)
         (status, output) = commands.getstatusoutput(
             'DEBCONF_SYSTEMRC=1 DEBCONF_NOWARNINGS=yes '
-            '/usr/bin/debconf-show %s' % showpkgs )
+            '@FINKPREFIX@/bin/debconf-show %s' % showpkgs )
         if status:
             conftext += '\n-- debconf-show failed\n'
         elif output:
@@ -1463,7 +1463,7 @@
         if pkgversion and not src_name:
             src_name = package
         if src_name and check_available and not notatty:
-            ewrite('Checking buildd.debian.org for past builds of %s...\n',
+            ewrite('Checking buildd.finkproject.org for past builds of %s...\n',
                    src_name)
             built = checkbuildd.check_built(src_name,
                                             http_proxy=options.http_proxy)
@@ -1484,7 +1484,7 @@
                 if ui.yes_no(
                     'Has this package successfully been built for this '
                     'architecture in the past (you can look this up at '
-                    'buildd.debian.org)',
+                    'buildd.finkproject.org)',
                     'Yes, this is a recently-introduced problem.',
                     'No, it has always been this way.'):
                     severity = 'serious'
@@ -1524,7 +1524,7 @@
         if os.path.exists('handle_bugscript'):
             handler = './handle_bugscript'
         else:
-            handler = '/usr/share/reportbug/handle_bugscript'
+            handler = '@FINKPREFIX@/share/reportbug/handle_bugscript'
 
         fh, filename = TempFile(prefix=tfprefix)
         fh.close()
@@ -1650,7 +1650,7 @@
     return
 
 if __name__ == '__main__':
-    sys.path.append('/usr/share/reportbug')
+    sys.path.append('@FINKPREFIX@/share/reportbug')
     try:
         main()
     except KeyboardInterrupt:
diff -ruN reportbug/reportbug.conf reportbug.new/reportbug.conf
--- reportbug/reportbug.conf	Tue Dec  7 06:29:35 2004
+++ reportbug.new/reportbug.conf	Tue Jan 18 11:35:44 2005
@@ -6,7 +6,7 @@
 # severity normal
 
 # BTS to use
-#bts debian
+bts fink
 # See 'reportbug --bts help' for a current list of supported BTSes
 
 # Submission address: default is 'submit'
@@ -68,7 +68,7 @@
 # Use TLS encryption.
 # smtptls
 
-# Use this to specify the path of your MTA; any SMTP server on Debian
+# Use this to specify the path of your MTA; any SMTP server on OS X
 # should be OK with the default.
 # mta /usr/sbin/sendmail
 
diff -ruN reportbug/reportbug.el reportbug.new/reportbug.el
--- reportbug/reportbug.el	Wed Feb  4 23:29:11 2004
+++ reportbug.new/reportbug.el	Tue Jan 18 11:35:44 2005
@@ -15,7 +15,7 @@
   (require 'gnus)
   (find-file (getenv "REPORTBUG"))
   (let ((subject (message-fetch-field "Subject"))
-        (toaddr (or (message-fetch-field "To") "submit@bugs.debian.org")))
+        (toaddr (or (message-fetch-field "To") "submit@bugs.finkproject.org")))
     (gnus-narrow-to-body)
     (let ((body (or (buffer-string) "")))
       (gnus-summary-mail-other-window)
diff -ruN reportbug/reportbug.py reportbug.new/reportbug.py
--- reportbug/reportbug.py	Wed Oct 20 08:34:05 2004
+++ reportbug.new/reportbug.py	Tue Jan 18 11:35:44 2005
@@ -33,7 +33,7 @@
 from string import ascii_letters, digits
 
 # Paths for dpkg
-DPKGLIB = '/var/lib/dpkg'
+DPKGLIB = '@FINKPREFIX@/var/lib/dpkg'
 AVAILDB = os.path.join(DPKGLIB, 'available')
 STATUSDB = os.path.join(DPKGLIB, 'status')
 
@@ -50,12 +50,12 @@
 MODES = {'novice': 'Offer simple prompts, bypassing technical questions.',
          'standard': 'Offer more extensive prompts, including asking about '
          'things that a moderately sophisticated user would be expected to '
-         'know about Debian.',
+         'know about Fink.',
          'advanced' : 'Like standard, but assumes you know a bit more about '
-         'Debian, including "incoming".',
+         'Fink, including "incoming".',
          'expert': 'Bypass most handholding measures and preliminary triage '
          'routines.  This mode should not be used by people unfamiliar with '
-         'Debian\'s policies and operating procedures.'}
+         'Fink\'s policies and operating procedures.'}
 MODELIST = ['novice', 'standard', 'advanced', 'expert']
 for mode in MODELIST:
     exec 'MODE_%s=%d' % (mode.upper(), MODELIST.index(mode))
@@ -87,8 +87,8 @@
     except:
         return 'Invalid date: '+`timeval`
 
-fhs_directories = ['/', '/usr', '/usr/share', '/var', '/usr/X11R6',
-                   '/usr/man', '/usr/doc', '/usr/bin']
+fhs_directories = ['@FINKPREFIX@', '@FINKPREFIX@/share', '@FINKPREFIX@/var', '/usr/X11R6',
+                   '@FINKPREFIX@/man', '@FINKPREFIX@/share/doc', '@FINKPREFIX@/bin']
 
 def realpath(filename):
     filename = os.path.abspath(filename)
@@ -108,8 +108,7 @@
 
     return filename
 
-pathdirs = ['/usr/sbin', '/usr/bin', '/sbin', '/bin', '/usr/X11R6/bin',
-            '/usr/games']
+pathdirs = ['@FINKPREFIX@/sbin', '@FINKPREFIX@/bin', '@FINKPREFIX@/sbin', '@FINKPREFIX@/bin', '/usr/X11R6/bin' ]
 
 def search_path_for(filename):
     dir, file = os.path.split(filename)
@@ -168,7 +167,7 @@
     return query_dpkg_for(newfilename or filename)
 
 def find_rewritten(username):
-    for filename in ['/etc/email-addresses']:
+    for filename in ['@FINKPREFIX@/etc/email-addresses']:
         if os.path.exists(filename):
             try:
                 fp = file(filename)
@@ -693,7 +692,7 @@
         return token
 
 USERFILE = os.path.expanduser('~/.reportbugrc')
-FILES = ('/etc/reportbug.conf', USERFILE)
+FILES = ('@FINKPREFIX@/etc/reportbug.conf', USERFILE)
 
 CONFIG_ARGS = (
     'sendto', 'severity', 'mua', 'mta', 'email', 'realname', 'bts', 'verify',
@@ -707,7 +706,7 @@
     'mutt' : 'mutt -H',
     'af' : 'af -EH < ',
     'mh' : '/usr/bin/mh/comp -use -file',
-    'gnus' : 'REPORTBUG=%s emacs -l /usr/share/reportbug/reportbug.el -f tfheen-reportbug-insert-template',
+    'gnus' : 'REPORTBUG=%s emacs -l @FINKPREFIX@/share/reportbug/reportbug.el -f tfheen-reportbug-insert-template',
     }
 MUA['nmh'] = MUA['mh']
 
diff -ruN reportbug/reportbug_submit.py reportbug.new/reportbug_submit.py
--- reportbug/reportbug_submit.py	Tue Dec  7 06:32:35 2004
+++ reportbug.new/reportbug_submit.py	Tue Jan 18 11:35:44 2005
@@ -30,7 +30,7 @@
 from reportbug import VERSION, VERSION_NUMBER
 
 import os
-sys.path = [os.curdir, '/usr/share/reportbug'] + sys.path
+sys.path = [os.curdir, '@FINKPREFIX@/share/reportbug'] + sys.path
 
 import re
 import commands
@@ -431,7 +431,7 @@
     'o'
 
 if __name__ == '__main__':
-    sys.path.append('/usr/share/reportbug')
+    sys.path.append('@FINKPREFIX@/share/reportbug')
     try:
         main()
     except KeyboardInterrupt:
diff -ruN reportbug/reportbug_ui_gnome.py reportbug.new/reportbug_ui_gnome.py
--- reportbug/reportbug_ui_gnome.py	Wed Feb  4 23:29:10 2004
+++ reportbug.new/reportbug_ui_gnome.py	Tue Jan 18 11:35:44 2005
@@ -19,7 +19,7 @@
 ##  SOFTWARE.
 
 import sys
-sys.path.append('/usr/lib/python2.1/site-packages')
+sys.path.append('@FINKPREFIX/lib/python2.6/site-packages')
 
 from reportbug_exceptions import *
 import reportbug, gnome
@@ -42,7 +42,7 @@
 
 class ReportBugApp(GnomeApp):
 	def __init__(self):
-		GnomeApp.__init__(self, "Debian Bug Report Druid", "Debian Bug Report Druid")
+		GnomeApp.__init__(self, "Fink Bug Report Druid", "Fink Bug Report Druid")
 		self.connect("destroy", callback_destroy)
 		self.setup()
 
@@ -52,7 +52,7 @@
 		logo_bgcolor = GdkColor(0x02, 0x66, 0x9A)
 
 		self.my_druid = GnomeDruid()
-		self.my_screen = GnomeDruidPageStandard("Debian Bug Report Druid", logo)
+		self.my_screen = GnomeDruidPageStandard("Fink Bug Report Druid", logo)
 		self.my_screen.connect("next", callback_next)
 		self.my_screen.set_bg_color(logo_bgcolor)
 		self.my_screen.set_logo_bg_color(logo_bgcolor)
diff -ruN reportbug/script reportbug.new/script
--- reportbug/script	Fri Dec 10 19:40:25 2004
+++ reportbug.new/script	Tue Jan 18 11:35:44 2005
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.3
+#!@FINKPREFIX@/bin/python2.6
 # -*- python -*-
 
 import os
