diff -Naur slime-2.0.orig/fink/50slime-start.el slime-2.0/fink/50slime-start.el
--- slime-2.0.orig/fink/50slime-start.el	1969-12-31 16:00:00.000000000 -0800
+++ slime-2.0/fink/50slime-start.el	2006-10-13 20:59:21.000000000 -0700
@@ -0,0 +1,3 @@
+(require 'slime)
+(add-hook 'lisp-mode-hook (lambda () (slime-mode t)))
+(add-hook 'inferior-lisp-mode-hook (lambda () (inferior-slime-mode t)))
diff -Naur slime-2.0.orig/fink/emacsen-install slime-2.0/fink/emacsen-install
--- slime-2.0.orig/fink/emacsen-install	1969-12-31 16:00:00.000000000 -0800
+++ slime-2.0/fink/emacsen-install	2006-10-13 21:13:27.000000000 -0700
@@ -0,0 +1,36 @@
+#! /bin/sh
+# @PREFIX@/lib/emacsen-common/packages/install/slime
+set -e
+
+FLAVOR=${1}
+PACKAGE=slime
+
+echo install/${PACKAGE}: Handling install of emacsen flavor ${FLAVOR}
+
+el_files="slime.el hyperspec.el"
+
+el_dir=@PREFIX@/share/emacs/site-lisp/${PACKAGE}/
+elc_dir=@PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}/
+el_path_list=`for i in ${el_files}; do \
+                  echo $i | perl -pe "s|^|${el_dir}|"; done`
+elc_path_list=`for i in ${el_files}; do \
+                  echo $i | perl -pe "s|^|${elc_dir}|"; done`
+
+byte_compile_options="-batch -q --no-init-file --no-site-file"
+
+case $FLAVOR in
+    emacs21 | emacs22 | emacs22-carbon )
+	echo install/${PACKAGE}: byte-compiling for ${FLAVOR}
+
+	install -m 755 -d ${elc_dir}
+
+	cp ${el_path_list} ${elc_dir}
+    cp ${el_dir}/*.lisp ${elc_dir}
+    cp ${el_dir}/ChangeLog ${elc_dir}
+    ;;
+
+    * )
+	echo install/${PACKAGE}: Skipping unsupported flavor ${FLAVOR}
+	;;
+esac
+
diff -Naur slime-2.0.orig/fink/emacsen-remove slime-2.0/fink/emacsen-remove
--- slime-2.0.orig/fink/emacsen-remove	1969-12-31 16:00:00.000000000 -0800
+++ slime-2.0/fink/emacsen-remove	2006-10-13 21:13:30.000000000 -0700
@@ -0,0 +1,18 @@
+#! /bin/sh
+# @PREFIX@/lib/emacsen-common/packages/remove/slime
+set -e
+
+FLAVOR=${1}
+PACKAGE=slime
+
+echo remove/${PACKAGE}: Handling removal of emacsen flavor ${FLAVOR}
+
+case $FLAVOR in
+    emacs21 | emacs22 | emacs22-carbon )
+	echo remove/${PACKAGE}: purging byte-compiled files for ${FLAVOR}
+	rm -rf @PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}
+	;;
+    * )
+	echo remove/${PACKAGE}: Skipping unsupported flavor ${FLAVOR}
+	;;
+esac
