diff -Nurd -x'*~' calc-2.12.4.3.orig/Makefile calc-2.12.4.3/Makefile
--- calc-2.12.4.3.orig/Makefile	2010-09-02 05:48:42.000000000 -0400
+++ calc-2.12.4.3/Makefile	2011-04-18 04:04:10.000000000 -0400
@@ -86,7 +86,7 @@
 #
 # If in doubt, leave TERMCONTROL empty.
 #
-TERMCONTROL=
+TERMCONTROL= -DUSE_TERMIOS
 #TERMCONTROL= -DUSE_TERMIOS
 #TERMCONTROL= -DUSE_TERMIO
 #TERMCONTROL= -DUSE_SGTTY
@@ -585,7 +585,7 @@
 
 #INCDIR= /usr/local/include
 #INCDIR= /dev/env/DJDIR/include
-INCDIR= /usr/include
+INCDIR= @PREFIX@/include
 
 # Where to install calc realted things
 #
@@ -624,15 +624,15 @@
 #
 #BINDIR= /usr/local/bin
 #BINDIR= /dev/env/DJDIR/bin
-BINDIR= /usr/bin
+BINDIR= @PREFIX@/bin
 
 #LIBDIR= /usr/local/lib
 #LIBDIR= /dev/env/DJDIR/lib
-LIBDIR= /usr/lib
+LIBDIR= @PREFIX@/lib
 
 #CALC_SHAREDIR= /usr/local/lib/calc
 #CALC_SHAREDIR= /dev/env/DJDIR/share/calc
-CALC_SHAREDIR= /usr/share/calc
+CALC_SHAREDIR= @PREFIX@/share/calc
 
 # NOTE: Do not set CALC_INCDIR to /usr/include or /usr/local/include!!!
 #	Always be sure that the CALC_INCDIR path ends in /calc to avoid
@@ -667,7 +667,7 @@
 CUSTOMCALDIR= ${CALC_SHAREDIR}/custom
 CUSTOMHELPDIR= ${CALC_SHAREDIR}/custhelp
 CUSTOMINCDIR= ${CALC_INCDIR}/custom
-SCRIPTDIR= ${BINDIR}/cscript
+SCRIPTDIR= ${BINDIR}
 
 # T - top level directory under which calc will be installed
 #
@@ -710,7 +710,7 @@
 #MANDIR=
 #MANDIR= /usr/local/man/man1
 #MANDIR= /usr/man/man1
-MANDIR= /usr/share/man/man1
+MANDIR= @PREFIX@/share/man/man1
 #MANDIR= /dev/env/DJDIR/man/man1
 #MANDIR= /usr/man/u_man/man1
 #MANDIR= /usr/contrib/man/man1
@@ -771,7 +771,7 @@
 # lines below.
 #
 #NROFF= nroff
-NROFF=
+NROFF= nroff
 #NROFF= groff
 NROFF_ARG= -man
 #NROFF_ARG= -mandoc
@@ -837,10 +837,10 @@
 #
 # If in doubt, set USE_READLINE, READLINE_LIB and READLINE_INCLUDE to nothing.
 #
-USE_READLINE=
+USE_READLINE= -DUSE_READLINE
 #USE_READLINE= -DUSE_READLINE
 #
-READLINE_LIB=
+READLINE_LIB= -L@PREFIX@/lib -lreadline -lhistory -lncurses
 READLINE_EXTRAS=
 #
 #READLINE_LIB= -lreadline
@@ -858,7 +858,7 @@
 #READLINE_LIB= -L/sw/lib -lreadline
 #READLINE_EXTRAS= -lhistory -lncurses
 #
-READLINE_INCLUDE=
+READLINE_INCLUDE= -I@PREFIX@/include
 #READLINE_INCLUDE= -I/usr/gnu/include
 #READLINE_INCLUDE= -I/usr/local/include
 
@@ -4485,7 +4485,7 @@
 	${Q}# NOTE: The this makefile installs libcustcalc${LIB_EXT_VERSION}
 	${Q}#       because we only want to perform one ${LDCONFIG} for both
 	${Q}#       libcalc${LIB_EXT_VERSION} and libcustcalc${LIB_EXT_VERSION}.
-	-${Q} if ${CMP} -s libcalc${LIB_EXT_VERSION} \
+	-${Q} if [ ! -e libcalc${LIB_EXT_VERSION} ]; then ${TRUE}; elif ${CMP} -s libcalc${LIB_EXT_VERSION} \
 		     ${T}${LIBDIR}/libcalc${LIB_EXT_VERSION} && \
 	   ${CMP} -s custom/libcustcalc${LIB_EXT_VERSION} \
 		     ${T}${LIBDIR}/libcustcalc${LIB_EXT_VERSION}; then \
