--- xcircuit-3.4.16/Makefile.in	2006-02-23 15:58:51.000000000 -0500
+++ xcircuit-3.4.16.patched/Makefile.in	2006-02-25 21:34:45.000000000 -0500
@@ -239,7 +239,7 @@
 
 # Directories for app-defaults file and manual page
 appdefaultsdir = $(datadir)/$(PACKAGE)-$(VERSION)/app-defaults
-appmandir = $(datadir)/$(PACKAGE)-$(VERSION)/man
+appmandir = $(man1dir)
 PATHNAMES = -DPROG_VERSION=$(VERSION) \
 	-DPROG_REVISION=$(REVISION) \
 	-DCAD_DIR=\"$(libdir)\" \
@@ -791,7 +791,7 @@
 
 lib/tcl/xcircuit${SHDLIB_EXT}: xcwrap.o ${xcircuit_OBJECTS} ${xcircuit_DEPEND}
 	$(RM) lib/xcircuit${SHDLIB_EXT}
-	$(CC) ${CFLAGS} ${SHLIB_CFLAGS} -o $@ ${LDDL_FLAGS} xcwrap.o \
+	$(CC) ${CFLAGS} ${SHLIB_CFLAGS} -o $@ -install_name $(libdir)/xcircuit-3.4/xcircuit${SHDLIB_EXT} ${LDDL_FLAGS} xcwrap.o \
 		${xcircuit_OBJECTS} ${xcircuit_LDADD} ${LDFLAGS} \
 		${X_EXTRA_LIBS} ${EXTRA_LIBS} -lc ${LIBS} ${EXTRA_LIB_SPECS} \
 		${SHLIB_LIB_SPECS}
