diff -Naur linkchecker-1.9.2/lc.cgi linkchecker-1.9.2-patched/lc.cgi
--- linkchecker-1.9.2/lc.cgi	2003-08-11 14:56:01.000000000 +0200
+++ linkchecker-1.9.2-patched/lc.cgi	2003-09-30 12:00:00.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@PREFIX@/bin/python2.3
 # -*- coding: iso-8859-1 -*-
 # Copyright (C) 2000-2002  Bastian Kleineidam
 #
diff -Naur linkchecker-1.9.2/lc.fcgi linkchecker-1.9.2-patched/lc.fcgi
--- linkchecker-1.9.2/lc.fcgi	2003-08-11 14:56:05.000000000 +0200
+++ linkchecker-1.9.2-patched/lc.fcgi	2003-09-30 11:59:57.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@PREFIX@/bin/python2.3
 # -*- coding: iso-8859-1 -*-
 # Copyright (C) 2000-2003  Bastian Kleineidam
 #
diff -Naur linkchecker-1.9.2/lc.sz_fcgi linkchecker-1.9.2-patched/lc.sz_fcgi
--- linkchecker-1.9.2/lc.sz_fcgi	2003-08-11 14:56:10.000000000 +0200
+++ linkchecker-1.9.2-patched/lc.sz_fcgi	2003-09-30 11:59:52.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!@PREFIX@/bin/python2.3
 # -*- coding: iso-8859-1 -*-
 # Copyright (C) 2000-2002  Bastian Kleineidam
 #
