diff -Nurd -x'*~' lynx2-8-4.orig/makefile.in lynx2-8-4/makefile.in
--- lynx2-8-4.orig/makefile.in	2005-01-09 15:24:28.000000000 -0500
+++ lynx2-8-4/makefile.in	2005-01-09 15:25:06.000000000 -0500
@@ -327,7 +327,7 @@
 			       ( cd $(DOCDIR) && chmod -R u+w . && tar xf - )
 	$(SHELL) -c 'for name in COPYING COPYHEADER; do \
 		cd $(HELPDIR) && rm -f $$name ;\
-		cd $(HELPDIR) && ( $(LN_S) $(DOCDIR)/$$name . || cp $(DOCDIR)/$$name . );\
+		cd $(HELPDIR) && cp $(DOCDIR)/$$name . ;\
 		done'
 
 # HPUX 'sed' does not seem to like '=' as pattern delimiter.
