diff -ruN deskmenu-1.4.0/src/xmalloc.c deskmenu-1.4.0-patched/src/xmalloc.c
--- deskmenu-1.4.0/src/xmalloc.c	Fri Jun 14 13:40:52 2002
+++ deskmenu-1.4.0-patched/src/xmalloc.c	Sun Nov 10 19:54:23 2002
@@ -36,6 +36,8 @@
 void free ();
 #endif
 
+# include <stdio.h>
+
 #if ENABLE_NLS
 # include <libintl.h>
 # define _(Text) gettext (Text)
@@ -44,7 +46,7 @@
 # define _(Text) Text
 #endif
 
-#include "error.h"
+
 
 #ifndef EXIT_FAILURE
 # define EXIT_FAILURE 1
@@ -64,11 +66,6 @@
    The caller may set it to some other value.  */
 int xmalloc_exit_failure = EXIT_FAILURE;
 
-#if __STDC__ && (HAVE_VPRINTF || HAVE_DOPRNT)
-void error (int, int, const char *, ...);
-#else
-void error ();
-#endif
 
 static VOID *
 fixup_null_alloc (n)
@@ -80,7 +77,10 @@
   if (n == 0)
     p = malloc ((size_t) 1);
   if (p == 0)
-    error (xmalloc_exit_failure, 0, _("Memory exhausted"));
+    {
+      fprintf (stderr, "%s: %S\%n", PACKAGE, _("Memory exhausted"));
+      exit (xmalloc_exit_failure);
+    }
   return p;
 }
 
