diff -uNbr icemc-0.2.2/Makefile icemc-0.2.2-new/Makefile
--- icemc-0.2.2/Makefile	Tue Dec  4 12:49:24 2001
+++ icemc-0.2.2-new/Makefile	Thu Dec 19 17:03:10 2002
@@ -17,15 +17,15 @@
 
 ####### Compiler, tools and options
 
-CC	=	gcc
-CXX	=	g++
+CC	=	cc
+CXX	=	c++
 CFLAGS	=	-pipe -DNO_DEBUG
 CXXFLAGS=	-pipe -DNO_DEBUG
-INCPATH	=	-I$(QTDIR)/include
-LINK	=	g++
+INCPATH	=	-I$(QTDIR)/include -I$(QTDIR)/include/qt
+LINK	=	c++
 LFLAGS	=	
-#LIBS	=	$(SUBLIBS) -L$(QTDIR)/lib -L/usr/X11R6/lib -lqt -lXext -lX11 -lm
-LIBS    =       $(SUBLIBS) -lqt
+LIBS	=	$(SUBLIBS) -L$(QTDIR)/lib -L/usr/X11R6/lib -lqt-mt -lXext -lX11
+#LIBS    =       $(SUBLIBS) -lqt
 MOC	=	$(QTDIR)/bin/moc
 UIC	=	$(QTDIR)/bin/uic
 
