diff -Nur cicb-5.1.0/configure cicb-5.1.0-patched/configure
--- cicb-5.1.0/configure	Thu Jan  4 04:04:40 2001
+++ cicb-5.1.0-patched/configure	Thu Sep 19 20:25:44 2002
@@ -2086,87 +2086,6 @@
   echo "$ac_t""no" 1>&6
 fi
 
-echo $ac_n "checking for tputs in -ltermcap""... $ac_c" 1>&6
-echo "configure:2091: checking for tputs in -ltermcap" >&5
-ac_lib_var=`echo termcap'_'tputs | sed 'y%./+-%__p_%'`
-if eval "test \"`echo '$''{'ac_cv_lib_$ac_lib_var'+set}'`\" = set"; then
-  echo $ac_n "(cached) $ac_c" 1>&6
-else
-  ac_save_LIBS="$LIBS"
-LIBS="-ltermcap  $LIBS"
-cat > conftest.$ac_ext <<EOF
-#line 2099 "configure"
-#include "confdefs.h"
-/* Override any gcc2 internal prototype to avoid an error.  */
-/* We use char because int might match the return type of a gcc2
-    builtin and then its argument prototype would still apply.  */
-char tputs();
-
-int main() {
-tputs()
-; return 0; }
-EOF
-if { (eval echo configure:2110: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  eval "ac_cv_lib_$ac_lib_var=yes"
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  eval "ac_cv_lib_$ac_lib_var=no"
-fi
-rm -f conftest*
-LIBS="$ac_save_LIBS"
-
-fi
-if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
-  echo "$ac_t""yes" 1>&6
-  LIBS="$LIBS -ltermcap"
-else
-  echo "$ac_t""no" 1>&6
-
-    echo $ac_n "checking for tputs in libtermcap.so""... $ac_c" 1>&6
-echo "configure:2130: checking for tputs in libtermcap.so" >&5
-  for i in /lib \
-           /usr/lib \
-           /usr/local/lib \
-           ; do
-      for j in `find $i -name "*termcap*" -print`; do
-          orig_libs=$LIBS
-	  LIBS="$LIBS $j"
-          cat > conftest.$ac_ext <<EOF
-#line 2139 "configure"
-#include "confdefs.h"
-
-            #include <stdio.h>
-          
-int main() {
-
-            tputs();
-          
-; return 0; }
-EOF
-if { (eval echo configure:2150: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
-  rm -rf conftest*
-  
-             echo "$ac_t""shared library is $j" 1>&6
-          
-else
-  echo "configure: failed program was:" >&5
-  cat conftest.$ac_ext >&5
-  rm -rf conftest*
-  LIBS=${orig_libs}
-fi
-rm -f conftest*
-
-	  if test x"${LIBS}" != x"${orig_libs}"; then
-	     break 2;
-          fi
-      done
-  done
-
-fi
-
 echo $ac_n "checking for sqrt in -lm""... $ac_c" 1>&6
 echo "configure:2172: checking for sqrt in -lm" >&5
 ac_lib_var=`echo m'_'sqrt | sed 'y%./+-%__p_%'`
@@ -2515,11 +2434,11 @@
 
 		# First check to see if --with-readlinelib was specified.
 		if test x"${with_readlinelib}" != x ; then
-		  if test -f "${with_readlinelib}/lib$readlineroot.so" -a \
+		  if test -f "${with_readlinelib}/lib$readlineroot.dylib" -a \
 				-f "${with_readlinelib}/lib$readlineroot.a" ; then
 			icb_ac_cv_readlinelib="-L`(cd ${with_readlinelib}; pwd)` -l$readlineroot"
-		  elif test -f "${with_readlinelib}/lib$readlineroot.so" ; then
-			icb_ac_cv_readlinelib=`(cd ${with_readlinelib}; pwd)`/lib$readlineroot.so
+		  elif test -f "${with_readlinelib}/lib$readlineroot.dylib" ; then
+			icb_ac_cv_readlinelib=`(cd ${with_readlinelib}; pwd)`/lib$readlineroot.dylib
 		  elif test -f "${with_readlinelib}/lib$readlineroot.a" ; then
 			icb_ac_cv_readlinelib=`(cd ${with_readlinelib}; pwd)`/lib$readlineroot.a
 		  else
@@ -2544,12 +2463,12 @@
 					   ${i}/readline \
 					   `ls -dr ${i}/readline/readline[0-9]* 2>/dev/null`  \
 					  ; do
-				if test -f "$j/lib$readlineroot.so" -a \
+				if test -f "$j/lib$readlineroot.dylib" -a \
 					-f "$j/lib$readlineroot.a" ; then
 				  icb_ac_cv_readlinelib="-L`(cd $j; pwd)` -l$readlineroot"
 				  break;
-				elif test -f "$j/lib$readlineroot.so" ; then
-				  icb_ac_cv_readlinelib=`(cd $j; pwd)`/lib$readlineroot.so
+				elif test -f "$j/lib$readlineroot.dylib" ; then
+				  icb_ac_cv_readlinelib=`(cd $j; pwd)`/lib$readlineroot.dylib
 				  break;
 				elif test -f "$j/lib$readlineroot.a" ; then
 				  icb_ac_cv_readlinelib=`(cd $j; pwd)`/lib$readlineroot.a
@@ -2852,27 +2771,27 @@
 		if test x"${with_tcllib}" != x ; then
 
 		  set -x
-		  if test -f "${with_tcllib}/lib$dottedtcllibroot.so" -a \
+		  if test -f "${with_tcllib}/lib$dottedtcllibroot.dylib" -a \
 				-f "${with_tcllib}/lib$dottedtcllibroot.a" ; then
 			icb_ac_cv_tcllib="-L`(cd ${with_tcllib}; pwd)` -l$dottedtcllibroot"
-		  elif test -f "${with_tcllib}/lib$dottedtcllibroot.so" ; then
-			icb_ac_cv_tcllib=`(cd ${with_tcllib}; pwd)`/lib$dottedtcllibroot.so
+		  elif test -f "${with_tcllib}/lib$dottedtcllibroot.dylib" ; then
+			icb_ac_cv_tcllib=`(cd ${with_tcllib}; pwd)`/lib$dottedtcllibroot.dylib
 		  elif test -f "${with_tcllib}/lib$dottedtcllibroot.a" ; then
 			icb_ac_cv_tcllib=`(cd ${with_tcllib}; pwd)`/lib$dottedtcllibroot.a
 
-		  elif test -f "${with_tcllib}/lib$undottedtcllibroot.so" -a \
+		  elif test -f "${with_tcllib}/lib$undottedtcllibroot.dylib" -a \
 				-f "${with_tcllib}/lib$undottedtcllibroot.a" ; then
 			icb_ac_cv_tcllib="-L`(cd ${with_tcllib}; pwd)` -l$undottedtcllibroot"
-		  elif test -f "${with_tcllib}/lib$undottedtcllibroot.so" ; then
-			icb_ac_cv_tcllib=`(cd ${with_tcllib}; pwd)`/lib$undottedtcllibroot.so
+		  elif test -f "${with_tcllib}/lib$undottedtcllibroot.dylib" ; then
+			icb_ac_cv_tcllib=`(cd ${with_tcllib}; pwd)`/lib$undottedtcllibroot.dylib
 		  elif test -f "${with_tcllib}/lib$undottedtcllibroot.a" ; then
 			icb_ac_cv_tcllib=`(cd ${with_tcllib}; pwd)`/lib$undottedtcllibroot.a
 
-		  elif test -f "${with_tcllib}/libtcl.so" -a \
+		  elif test -f "${with_tcllib}/libtcl.dylib" -a \
 				-f "${with_tcllib}/libtcl.a" ; then
 			icb_ac_cv_tcllib="-L`(cd ${with_tcllib}; pwd)` -ltcl"
-		  elif test -f "${with_tcllib}/libtcl.so" ; then
-			icb_ac_cv_tcllib=`(cd ${with_tcllib}; pwd)`/libtcl.so
+		  elif test -f "${with_tcllib}/libtcl.dylib" ; then
+			icb_ac_cv_tcllib=`(cd ${with_tcllib}; pwd)`/libtcl.dylib
 		  elif test -f "${with_tcllib}/libtcl.a" ; then
 			icb_ac_cv_tcllib=`(cd ${with_tcllib}; pwd)`/libtcl.a
 
@@ -2898,23 +2817,23 @@
 					   ${i}/tcl \
 					   `ls -dr ${i}/tcl/tcl[0-9]* 2>/dev/null`  \
 					  ; do
-				if test -f "$j/lib$dottedtcllibroot.so" -a \
+				if test -f "$j/lib$dottedtcllibroot.dylib" -a \
 					-f "$j/lib$dottedtcllibroot.a" ; then
 				  icb_ac_cv_tcllib="-L`(cd $j; pwd)` -l$dottedtcllibroot"
 				  break;
-				elif test -f "$j/lib$dottedtcllibroot.so" ; then
-				  icb_ac_cv_tcllib=`(cd $j; pwd)`/lib$dottedtcllibroot.so
+				elif test -f "$j/lib$dottedtcllibroot.dylib" ; then
+				  icb_ac_cv_tcllib=`(cd $j; pwd)`/lib$dottedtcllibroot.dylib
 				  break;
 				elif test -f "$j/lib$dottedtcllibroot.a" ; then
 				  icb_ac_cv_tcllib=`(cd $j; pwd)`/lib$dottedtcllibroot.a
 				  break;
 
-				elif test -f "$j/lib$undottedtcllibroot.so" -a \
+				elif test -f "$j/lib$undottedtcllibroot.dylib" -a \
 					-f "$j/lib$undottedtcllibroot.a" ; then
 				  icb_ac_cv_tcllib="-L`(cd $j; pwd)` -l$undottedtcllibroot"
 				  break;
-				elif test -f "$j/lib$undottedtcllibroot.so" ; then
-				  icb_ac_cv_tcllib=`(cd $j; pwd)`/lib$undottedtcllibroot.so
+				elif test -f "$j/lib$undottedtcllibroot.dylib" ; then
+				  icb_ac_cv_tcllib=`(cd $j; pwd)`/lib$undottedtcllibroot.dylib
 				  break;
 				elif test -f "$j/lib$undottedtcllibroot.a" ; then
 				  icb_ac_cv_tcllib=`(cd $j; pwd)`/lib$undottedtcllibroot.a
diff -Nur cicb-5.1.0/icb/Makefile.in cicb-5.1.0-patched/icb/Makefile.in
--- cicb-5.1.0/icb/Makefile.in	2000-10-14 23:49:50.000000000 -0400
+++ cicb-5.1.0-patched/icb/Makefile.in	2006-04-04 17:05:39.000000000 -0400
@@ -87,7 +87,7 @@
 default all: icb
 
 icb: $(OBJ) Makefile
-	$(CC) $(CFLAGS) -o icb $(OBJ) $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o icb $(OBJ) $(LIBS)
 
 funcs.h: protos.h
 	sed -e 's/(.*)/()/' protos.h > tmp
@@ -97,12 +97,12 @@
 
 $(DATADIR)/icbserverdb: ../icbserverdb
 	-mkdir $(DATADIR)
-	../support/install-sh -c -o 0 -g 0 -m 0644 ../icbserverdb $(DATADIR)/icbserverdb
+	../support/install-sh -c -m 0644 ../icbserverdb $(DATADIR)/icbserverdb
 
 $(BINDIR)/icb: icb
 	-mkdir $(BINDIR)
 	-rm -f $(BINDIR)/icb
-	../support/install-sh -c -o 0 -g 0 -m 0755 -s icb $(BINDIR)
+	../support/install-sh -c -m 0755 -s icb $(BINDIR)
 
 clean:
 	rm -f *.o icb core
diff -Nur cicb-5.1.0/icb/c_log.c cicb-5.1.0-patched/icb/c_log.c
--- cicb-5.1.0/icb/c_log.c	2000-12-14 23:29:17.000000000 -0500
+++ cicb-5.1.0-patched/icb/c_log.c	2006-04-04 16:49:26.000000000 -0400
@@ -16,6 +16,7 @@
 #ifdef HAVE_ERRNO_H
 #include <errno.h>
 #endif
+#include <string.h>
 
 FILE *logfp = NULL;
 
@@ -30,7 +31,7 @@
 	time_t time();
 	struct tm *t, *localtime();
 	time_t clock;
-	extern int errno, sys_nerr;
+	extern int errno;
 
 	/* determine pathname to use */
 	if (path == NULL || *path == '\0') {
diff -Nur cicb-5.1.0/icb/c_time.c cicb-5.1.0-patched/icb/c_time.c
--- cicb-5.1.0/icb/c_time.c	2000-12-14 22:21:02.000000000 -0500
+++ cicb-5.1.0-patched/icb/c_time.c	2006-04-04 16:49:38.000000000 -0400
@@ -10,6 +10,7 @@
 #ifdef HAVE_TIME_H
 #include <time.h>
 #endif
+#include <string.h>
 
 #include "icb.h"
 #include "externs.h"
diff -Nur cicb-5.1.0/icb/icb.h cicb-5.1.0-patched/icb/icb.h
--- cicb-5.1.0/icb/icb.h	2000-12-14 23:29:18.000000000 -0500
+++ cicb-5.1.0-patched/icb/icb.h	2006-04-04 16:49:08.000000000 -0400
@@ -128,6 +128,13 @@
 # define Tcl_GetStringResult(interp) ((interp)->result)
 #endif
 
+#define COLORSIZE 12	/* how many chars can make an ansi sequence */
+
+
+struct COLORTABLE {
+	char color[COLORSIZE];	/* this color's ansi sequence */
+	int defined;		/* has color been set? */ 
+};
 
 /* include function prototypes if ANSI compiler */
 /* otherwise, include file that shows return type for each function */
@@ -149,13 +156,6 @@
 	int port;
 };
 
-#define COLORSIZE 12	/* how many chars can make an ansi sequence */
-
-
-struct COLORTABLE {
-	char color[COLORSIZE];	/* this color's ansi sequence */
-	int defined;		/* has color been set? */ 
-};
 
 enum { ColSANE, 
        ColNORMAL, 
diff -Nur cicb-5.1.0/icb/main.c cicb-5.1.0-patched/icb/main.c
--- cicb-5.1.0/icb/main.c	Tue Dec 12 07:02:07 2000
+++ cicb-5.1.0-patched/icb/main.c	Wed Sep 18 14:05:26 2002
@@ -287,7 +287,7 @@
 
 	if (myserver)
 	{
-		serverdata = getservbyname(myserver);
+		serverdata = getserverbyname(myserver);
 		if (!serverdata)
 		{
 			fprintf(stderr,"%s: %s: server not found.\n",
diff -Nur cicb-5.1.0/icb/protos.h cicb-5.1.0-patched/icb/protos.h
--- cicb-5.1.0/icb/protos.h	Fri Dec 15 03:21:04 2000
+++ cicb-5.1.0-patched/icb/protos.h	Wed Sep 18 14:05:32 2002
@@ -114,7 +114,7 @@
 void c_packet(char *pkt);
 void c_userchar(void);
 struct server *getserver(void);
-struct server *getservbyname(char *name);
+struct server *getserverbyname(char *name);
 void listservers(void);
 
 /* c_replay.c */
diff -Nur cicb-5.1.0/icb/serverlist.c cicb-5.1.0-patched/icb/serverlist.c
--- cicb-5.1.0/icb/serverlist.c	2000-10-02 14:38:45.000000000 -0400
+++ cicb-5.1.0-patched/icb/serverlist.c	2006-04-04 16:51:36.000000000 -0400
@@ -18,6 +18,7 @@
 #ifdef HAVE_SYS_STAT_H
 #include <sys/stat.h>
 #endif
+#include <string.h>
 
 #include "icb.h"
 
@@ -138,7 +139,7 @@
 	return &serverbuf;
 }
 
-struct server *getservbyname(name)
+struct server *getserverbyname(name)
 char *name;
 {
 	if (!serverfile) {
diff -Nur cicb-5.1.0/icb/strccmp.c cicb-5.1.0-patched/icb/strccmp.c
--- cicb-5.1.0/icb/strccmp.c	2000-12-19 01:46:02.000000000 -0500
+++ cicb-5.1.0-patched/icb/strccmp.c	2006-04-04 16:56:43.000000000 -0400
@@ -237,7 +237,7 @@
 				if (a == '\0')
 					return 0;
 
-shloop:	    }
+shloop:	;   }
 			while (cm[a] != cm[b]);
 
 
diff -Nur cicb-5.1.0/icb/tcl.c cicb-5.1.0-patched/icb/tcl.c
--- cicb-5.1.0/icb/tcl.c	2000-10-02 07:25:55.000000000 -0400
+++ cicb-5.1.0-patched/icb/tcl.c	2006-04-04 16:59:34.000000000 -0400
@@ -15,6 +15,7 @@
 #ifdef HAVE_SYS_PARAM_H
 #include <sys/param.h>
 #endif
+#include <string.h>
 
 #include "icb.h"
 #include "externs.h"
diff -Nur cicb-5.1.0/murgil/Makefile.in cicb-5.1.0-patched/murgil/Makefile.in
--- cicb-5.1.0/murgil/Makefile.in	2000-10-02 08:04:45.000000000 -0400
+++ cicb-5.1.0-patched/murgil/Makefile.in	2006-04-04 16:49:08.000000000 -0400
@@ -7,7 +7,7 @@
 
 CC          = @CC@
 CFLAGS      = @CFLAGS@
-CPPFLAGS    = -I@srcdir@ -I@top_srcdir@
+CPPFLAGS    = @CPPFLAGS@ -I@srcdir@ -I@top_srcdir@
 
 MAKEDEPEND  = @MAKEDEPEND@
 MDFLAGS     = $(CPPFLAGS)
diff -Nur cicb-5.1.0/murgil/globals.c cicb-5.1.0-patched/murgil/globals.c
--- cicb-5.1.0/murgil/globals.c	Mon Oct  2 12:25:59 2000
+++ cicb-5.1.0-patched/murgil/globals.c	Wed Sep 18 14:08:36 2002
@@ -1,5 +1,5 @@
 /* $Id: globals.c,v 1.6 2000/10/02 11:25:59 hoche Exp $ */
 #include "config.h"
 
-int port_fd;
-char *fail;
+int port_fd = -1;
+char *fail = (char *)0;
