--- gnewspost-0.6/src/app.c	Sun Jul 28 21:10:11 2002
+++ gnewspost-0.6/src/app.c	Tue Apr 27 22:02:19 2004
@@ -1013,11 +1013,11 @@
 		return 0;
 
 	if (sb.st_size > 1024 && sb.st_size < 1048576)
-		snprintf(buf, 8, "%dK", sb.st_size / 1024);
+		snprintf(buf, 8, "%dK", (int) sb.st_size / 1024);
 	else if (sb.st_size >= 1048576)
 		snprintf(buf, 8, "%.2fM", (float) sb.st_size / 1024 / 1024);
 	else
-		snprintf(buf, 8, "%dB", sb.st_size);
+		snprintf(buf, 8, "%dB", (int) sb.st_size);
 
 	return buf;
 }
--- gnewspost-0.6/src/app.c	Sun Jul 28 21:10:11 2002
+++ gnewspost-0.6/src/app.c	Tue Apr 27 22:50:47 2004
@@ -737,7 +737,7 @@
 {
 	Prefs *p;
 	GList *sel_list, *node;
-	gchar *titles[6];
+	gchar *titles[3];
 	gchar *dir, *file, *tmp;
 	gint  i;
 
