diff -Naur curl-7.16.4.orig/configure curl-7.16.4/configure
--- curl-7.16.4.orig/configure	2007-06-25 05:18:41.000000000 -0400
+++ curl-7.16.4/configure	2007-09-14 21:50:33.000000000 -0400
@@ -23847,7 +23847,7 @@
 
   if test -z "$GSSAPI_INCS"; then
      if test -f "$GSSAPI_ROOT/bin/krb5-config"; then
-        GSSAPI_INCS=`$GSSAPI_ROOT/bin/krb5-config --cflags gssapi`
+        GSSAPI_INCS=`$GSSAPI_ROOT/bin/krb5-config --cflags gssapi | sed 's,-I/usr/include ,,g'`
      elif test "$GSSAPI_ROOT" != "yes"; then
         GSSAPI_INCS="-I$GSSAPI_ROOT/include"
      fi
@@ -24311,7 +24311,7 @@
     LDFLAGS="$LDFLAGS $GSSAPI_LIB_DIR -lgss"
   elif test -z "$GSSAPI_LIB_DIR"; then
      if test -f "$GSSAPI_ROOT/bin/krb5-config"; then
-        gss_ldflags=`$GSSAPI_ROOT/bin/krb5-config --libs gssapi`
+        gss_ldflags=`$GSSAPI_ROOT/bin/krb5-config --libs gssapi | sed 's,-L/usr/lib ,,g'`
 	LDFLAGS="$LDFLAGS $gss_ldflags"
      elif test "$GSSAPI_ROOT" != "yes"; then
         LDFLAGS="$LDFLAGS -L$GSSAPI_ROOT/lib$libsuff -lgssapi"
diff -Naur curl-7.16.4.orig/curl-config.in curl-7.16.4/curl-config.in
--- curl-7.16.4.orig/curl-config.in	2007-02-18 04:41:27.000000000 -0500
+++ curl-7.16.4/curl-config.in	2007-09-14 21:52:34.000000000 -0400
@@ -187,9 +187,9 @@
 	   CURLLIBDIR=""
 	fi
 	if test "X@REQUIRE_LIB_DEPS@" = "Xyes"; then
-	  echo ${CURLLIBDIR}-lcurl @LDFLAGS@ @LIBCURL_LIBS@ @LIBS@
+	  echo @LDFLAGS@ ${CURLLIBDIR}-lcurl @LIBCURL_LIBS@ @LIBS@
 	else
-	  echo ${CURLLIBDIR}-lcurl @LDFLAGS@ @LIBS@
+	  echo @LDFLAGS@ ${CURLLIBDIR}-lcurl @LIBS@
 	fi
 	;;
 
diff -Naur curl-7.16.4.orig/libcurl.pc.in curl-7.16.4/libcurl.pc.in
--- curl-7.16.4.orig/libcurl.pc.in	2007-02-18 04:41:27.000000000 -0500
+++ curl-7.16.4/libcurl.pc.in	2007-09-14 21:52:34.000000000 -0400
@@ -33,6 +33,6 @@
 URL: http://curl.haxx.se/
 Description: Library to transfer files with ftp, http, etc.
 Version: @VERSION@
-Libs: -L${libdir} -lcurl @LDFLAGS@ @LIBS@
+Libs: @LDFLAGS@ -L${libdir} -lcurl @LIBS@
 Libs.private: @LIBCURL_LIBS@ @LIBS@
 Cflags: -I${includedir}
