diff -Naur packit-0.7/src/globals.h packit-0.7.orig/src/globals.h
--- packit-0.7/src/globals.h	Fri Oct  3 05:52:08 2003
+++ packit-0.7.orig/src/globals.h	Wed Feb 25 17:17:46 2004
@@ -31,7 +31,7 @@
 #include <libnet.h>
 #include <pcap.h>
 #include <sys/types.h>
-#include <net/bpf.h>
+#include <pcap-bpf.h>
 #include <signal.h>
 #include <time.h>
 #include <stdio.h>
diff -Naur packit-0.7/src/main.h packit-0.7.orig/src/main.h
--- packit-0.7/src/main.h	Sat Sep 27 10:22:52 2003
+++ packit-0.7.orig/src/main.h	Wed Feb 25 17:18:19 2004
@@ -26,7 +26,7 @@
 #include <libnet.h>
 #include <pcap.h>
 #include <sys/types.h>
-#include <net/bpf.h>
+#include <pcap-bpf.h>
 #include <signal.h>
 #include <time.h>
 #include <stdio.h>
