diff -urN proxy-2.2.4/README proxy-2.2.4.new/README
--- proxy-2.2.4/README	Mon May 22 17:19:20 2000
+++ proxy-2.2.4.new/README	Wed Oct 27 23:40:14 2004
@@ -37,8 +37,8 @@
 FILTERING 
 Proxy-2.2.4 can also do filtering. In order to take advantage of the
 filtering rules of proxy-2.2, you must make a config file. The file will
-reside in the /etc directory and it will be called "proxy.filters"
-(i.e. /etc/proxy.filters).  The filter file can contain comments, only if
+reside in the @PREFIX@/etc directory and it will be called "proxy.filters"
+(i.e. @PREFIX@/etc/proxy.filters).  The filter file can contain comments, only if
 the first character of the line starts with the pound sign '#'. Example:
 
 Example 1:
diff -urN proxy-2.2.4/proxy.c proxy-2.2.4.new/proxy.c
--- proxy-2.2.4/proxy.c	Mon May 22 17:19:20 2000
+++ proxy-2.2.4.new/proxy.c	Wed Oct 27 23:40:17 2004
@@ -250,7 +250,7 @@
   char *access_str, *deny_str;
   int i;
 
-  access = fopen ("/etc/proxy.filters", "r");
+  access = fopen ("@PREFIX@/etc/proxy.filters", "r");
 
   if (access == NULL)
     return 0;
diff -urN proxy-2.2.4/tcp_listen.c proxy-2.2.4.new/tcp_listen.c
--- proxy-2.2.4/tcp_listen.c	Tue Mar 21 18:42:15 2000
+++ proxy-2.2.4.new/tcp_listen.c	Mon Nov  8 22:54:06 2004
@@ -10,6 +10,7 @@
   struct addrinfo hints, *res, *ressave;
 
   bzero (&hints, sizeof (struct addrinfo));
+  hints.ai_family = AF_INET;
   hints.ai_flags = AI_PASSIVE;
   hints.ai_socktype = SOCK_STREAM;
 
