--- configure	2009-02-26 09:57:32.000000000 +1100
+++ configure	2010-05-12 17:18:00.000000000 +1000
@@ -1703,7 +1703,7 @@
 if test "$with_mysql" = "yes" ; then
 	echo $ac_n "checking for mysql include path""... $ac_c" 1>&6
 echo "configure:1706: checking for mysql include path" >&5
-	mysql_try_includes="/usr/include /usr/include/mysql /usr/local/include /usr/local/include/mysql /usr/local/mysql/include /usr/local/mysql/include/mysql"
+	mysql_try_includes="@FINKPREFIX@/include/mysql"
 	for i in `echo ${mysql_try_includes}`; do
 	    if test -f "${i}/mysql.h" ; then
 		CPPFLAGS="${CPPFLAGS} -I${i}"
@@ -1761,7 +1761,7 @@
 
 	echo $ac_n "checking for mysql lib path""... $ac_c" 1>&6
 echo "configure:1764: checking for mysql lib path" >&5
-	mysql_try_libs="/usr/lib /usr/lib/mysql /usr/local/lib /usr/local/lib/mysql /usr/local/mysql/lib /usr/local/mysql/lib/mysql"
+	mysql_try_libs="@FINKPREFIX@/lib/mysql"
 	found=""
 	for i in `echo ${mysql_try_libs}`; do
 	    if test -f "${i}/libmysqlclient.so" -o -f "${i}/libmysqlclient.a" ; then
@@ -1825,7 +1825,7 @@
 if test "$with_pgsql" = "yes" ; then
 	echo $ac_n "checking for pgsql include path""... $ac_c" 1>&6
 echo "configure:1828: checking for pgsql include path" >&5
-	pgsql_try_includes="/usr/include /usr/include/pgsql /usr/local/include /usr/local/include/pgsql /usr/local/pgsql/include"
+	pgsql_try_includes="@FINKPREFIX@/include/postgresql"
 	for i in `echo ${pgsql_try_includes}`; do
 	    if test -f "${i}/libpq-fe.h" ; then
 		CPPFLAGS="${CPPFLAGS} -I${i}"
@@ -1883,7 +1883,7 @@
 
 	echo $ac_n "checking for pgsql lib path""... $ac_c" 1>&6
 echo "configure:1886: checking for pgsql lib path" >&5
-	pgsql_try_libs="/usr/lib /usr/lib/pgsql /usr/local/lib /usr/local/lib/pgsql /usr/local/pgsql/lib"
+	pgsql_try_libs="@FINKPREFIX@/lib/postgresql-8.0"
 	for i in `echo ${pgsql_try_libs}`; do
 	    if test -f "${i}/libpq.so" -o -f "${i}/libpq.a" ; then
 		save_ldflags="${LDFLAGS}"
