diff -Nurd snownews-1.5.9.orig/configure snownews-1.5.9/configure
--- snownews-1.5.9.orig/configure	2008-02-17 08:43:58.000000000 -0500
+++ snownews-1.5.9/configure	2008-04-16 23:46:52.000000000 -0400
@@ -29,7 +29,7 @@
 	print "Configuring for a GNU based system... ";
 } elsif (lc($os) =~ /bsd|darwin|freemint/) {
 	print "Configuring for a BSD style system... ";
-	$cflags = $cflags.' -DSTATIC_CONST_ICONV -I/usr/local/include';
+	$cflags = $cflags.' -DSTATIC_CONST_ICONV';
 	if ($use_nls == 1) {
 		$ldflags .= ' -lintl ';
 	}
diff -Nurd snownews-1.5.9.orig/doc/man/snownews.1.in snownews-1.5.9/doc/man/snownews.1.in
--- snownews-1.5.9.orig/doc/man/snownews.1.in	2008-02-17 08:43:57.000000000 -0500
+++ snownews-1.5.9/doc/man/snownews.1.in	2008-04-16 23:42:14.000000000 -0400
@@ -67,7 +67,7 @@
 .B 'o'
 will open the link (usually the complete news text) in the browser.
 .P
-The default browser is lynx, but you can change this by pressing
+The default browser depends on 'open', but you can change this by pressing
 .B 'B'
 in the main menu and entering a new default browser. Or you can edit
 the file ~/.snownews/browser. The program replaces
diff -Nurd snownews-1.5.9.orig/main.c snownews-1.5.9/main.c
--- snownews-1.5.9.orig/main.c	2008-02-17 08:43:58.000000000 -0500
+++ snownews-1.5.9/main.c	2008-04-16 23:45:14.000000000 -0400
@@ -90,7 +90,9 @@
 		/* Do this after everything else. If it doesn't work or hang
 		   user can ctrl-c without interfering with the program operation
 		   (like writing caches). */
+		/* Disable the check for later versions...
 		AutoVersionCheck();
+		It is not needed in a packaged distribution. */
 		
 		exit(0);
 	} else {
diff -Nurd snownews-1.5.9.orig/setup.c snownews-1.5.9/setup.c
--- snownews-1.5.9.orig/setup.c	2008-02-17 08:43:58.000000000 -0500
+++ snownews-1.5.9/setup.c	2008-04-16 23:45:33.000000000 -0400
@@ -64,7 +64,7 @@
 		configfile = fopen (file, "w+");
 		if (configfile == NULL)
 			MainQuit (_("Create initial configfile \"config\""), strerror(errno)); /* Still didn't work? */
-		browser = strdup("lynx %s");
+		browser = strdup("open %s");
 		fputs (browser, configfile);
 		fclose (configfile);
 	} else {
