diff -u -r1.43 darwin.cf
--- config/cf/darwin.cf	11 Nov 2003 01:29:46 -0000	1.43
+++ config/cf/darwin.cf	14 Nov 2003 20:48:37 -0000
@@ -155,8 +155,8 @@
 #endif
 
 /* we have a gcc compatible compiler, but its not called gcc */
-#define CcCmd                   /usr/bin/cc
-#define CplusplusCmd            /usr/bin/c++
+#define CcCmd                   gcc-3.3
+#define CplusplusCmd            g++-3.3
 
 /* default flags to pass to cc */
 #ifndef DefaultCCOptions
Index: lib/fontenc/Imakefile
===================================================================
RCS file: /cvs/xc/lib/fontenc/Imakefile,v
retrieving revision 1.8
diff -u -r1.8 Imakefile
--- lib/fontenc/Imakefile	17 Nov 2003 15:30:11 -0000	1.8
+++ lib/fontenc/Imakefile	18 Nov 2003 21:23:28 -0000
@@ -3,6 +3,8 @@
    INCLUDES = -I$(FONTINCSRC) -I$(FONTLIBSRC)/include -I$(FONTLIBSRC)/fontfile \
 	      -I$(INCLUDESRC)
 
+REQUIREDLIBS = -lz
+
 FONTENCDEFS = -DFONTENC_NO_LIBFONT \
 	      -DFONT_ENCODINGS_DIRECTORY=\"$(FONTDIR)/encodings/encodings.dir\"
 
