diff -ru --new-file algae-3.7.0.orig/Makefile.in algae-3.7.0/Makefile.in
--- algae-3.7.0.orig/Makefile.in	Wed Aug 14 02:21:23 2002
+++ algae-3.7.0/Makefile.in	Wed Aug 14 02:22:48 2002
@@ -73,7 +73,7 @@
 libdir = $(exec_prefix)/lib
 
 # Directory in which to install the man page.
-mandir = $(prefix)/man/man$(manext)
+mandir = @mandir@/man$(manext)
 # The manpage extension
 manext = 1
 
diff -ru --new-file algae-3.7.0.orig/configure algae-3.7.0/configure
--- algae-3.7.0.orig/configure	Wed Aug 14 02:21:23 2002
+++ algae-3.7.0/configure	Wed Aug 14 02:22:48 2002
@@ -2935,7 +2935,7 @@
 # checks (except for LAPACK) don't need it and won't find it.
 
 if test "$BLAS" = xblas; then
-  LDFLAGS="$LDFLAGS -L../blas"
+  LDFLAGS="-L../blas $LDFLAGS"
   LOCAL_LIBS="-lblas $LOCAL_LIBS"
 fi
 
@@ -3012,7 +3012,7 @@
 # checks don't need it and won't find it.
 
 if test "$LAPACK" = xlapack; then
-  LDFLAGS="$LDFLAGS -L../lapack"
+  LDFLAGS="-L../lapack $LDFLAGS"
   LOCAL_LIBS="-llapack $LOCAL_LIBS"
 fi
 
diff -ru --new-file algae-3.7.0.orig/doc/Makefile.in algae-3.7.0/doc/Makefile.in
--- algae-3.7.0.orig/doc/Makefile.in	Wed Aug 14 02:21:21 2002
+++ algae-3.7.0/doc/Makefile.in	Wed Aug 14 02:22:48 2002
@@ -174,13 +174,10 @@
 	-if test -f algae.info; then d=.; else d=$(srcdir); fi; \
 	for f in $$d/algae.info $$d/algae.info-*; do \
 	  $(INSTALL_DATA) $$f $(infodir); \
-	done; \
-	if $(SHELL) -c 'install-info --version' > /dev/null 2>&1; then \
-	  install-info --info-dir=$(infodir) $$d/algae.info; else true; fi
+	done;
 	-if test -f algae_toc.html; then d=.; else d=$(srcdir); fi; \
 	for f in $$d/index.html $$d/algae*.html $$d/icon.gif $$d/info.db; do \
 	  $(INSTALL_DATA) $$f $(htmldir); done
-	-rm -f $(datadir)/algae/html; ln -s $(htmldir) $(datadir)/algae/html
 	-$(INSTALL_DATA) algae.1 $(mandir)/algae.$(manext)
 
 uninstall:
diff -ru --new-file algae-3.7.0.orig/fpe/darwin.c algae-3.7.0/fpe/darwin.c
--- algae-3.7.0.orig/fpe/darwin.c	Wed Dec 31 16:00:00 1969
+++ algae-3.7.0/fpe/darwin.c	Wed Aug 14 02:22:48 2002
@@ -0,0 +1,80 @@
+/*
+   fpe/darwin.c -- SIGFPE code
+
+   Copyright (C) 1994-95  K. Scott Hunziker.
+   Copyright (C) 1990-94  The Boeing Company.
+   See the file COPYING for license, warranty, and permission details.
+*/
+
+static char rcsid[] =
+  "$Id: algae-3.5.1-2.patch,v 1.1 2002/02/20 09:56:25 neeri Exp $";
+
+#if HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <stdio.h>
+#include <math.h>
+#include <errno.h>
+#include <setjmp.h>
+#include <signal.h>
+#include <mach/mach.h>
+
+#if !SKIP_LOCAL_INCLUDES
+#include "exception.h"
+#include "message.h"
+#endif
+
+/*
+ * On Mach, we need a mindbogglingly complex setup for floating point errors.
+ * Not the least of the hassles is that we have to do the whole thing from
+ * a different thread.
+ */
+void * fpe_enabler(void * parent)
+{
+	mach_port_t		victim = (mach_port_t)parent;
+	mach_msg_type_number_t	count;
+
+	ppc_thread_state_t	ts;
+	ppc_float_state_t	fs;
+
+	/* First enable the right FP exception conditions */
+	count = PPC_FLOAT_STATE_COUNT;
+	thread_get_state(victim, PPC_FLOAT_STATE, (thread_state_t)&fs, &count);
+	/* Enable VE OE ZE, Disable UE XE */
+	fs.fpscr = (fs.fpscr & ~0x1FFFFF28) | 0x0D0;
+	thread_set_state(victim, PPC_FLOAT_STATE, (thread_state_t)&fs, count);
+
+	/* Now, enable FP exceptions as such */
+	count = PPC_THREAD_STATE_COUNT;
+	thread_get_state(victim, PPC_THREAD_STATE, (thread_state_t)&ts, &count);
+	/* Set FE0 = FE1 = 1 */
+	ts.srr1 |= 0x0900;
+	thread_set_state(victim, PPC_THREAD_STATE, (thread_state_t)&ts, count);
+	
+	return 0;
+}
+
+static void
+catch_sigfpe( sig )
+  int sig;
+{
+    switch( sig ) {
+      default:
+       
+	fail( "Floating point exception.", sig );
+	break;
+    }
+
+    raise_exception();
+}
+
+void
+enable_fpe_traps()
+{
+    pthread_t 	enabler;
+    pthread_create(&enabler, NULL, fpe_enabler, mach_thread_self());
+    pthread_join(enabler, NULL);
+
+    signal( SIGFPE, catch_sigfpe );
+}
diff -ru --new-file algae-3.7.0.orig/src/Makefile.in algae-3.7.0/src/Makefile.in
--- algae-3.7.0.orig/src/Makefile.in	Wed Aug 14 02:21:20 2002
+++ algae-3.7.0/src/Makefile.in	Wed Aug 14 02:22:48 2002
@@ -220,7 +220,7 @@
 install: all
 	$(INSTALL_PROGRAM) algae $(bindir)/algae-$(VERSION_NUMBER)
 	rm -f $(bindir)/algae
-	ln -s $(bindir)/algae-$(VERSION_NUMBER) $(bindir)/algae
+	ln -s algae-$(VERSION_NUMBER) $(bindir)/algae
 .PHONY:	install
 
 uninstall:
diff -ru --new-file algae-3.7.0.orig/src/mem.h algae-3.7.0/src/mem.h
--- algae-3.7.0.orig/src/mem.h	Wed Aug 14 02:21:21 2002
+++ algae-3.7.0/src/mem.h	Wed Aug 14 02:22:48 2002
@@ -24,7 +24,7 @@
 char *PROTO (getenv, (char *));
 #endif
 
-#if !defined(VAXC) && !defined(apollo)
+#if !defined(VAXC) && !defined(apollo) && !defined(__APPLE__)
 #include <malloc.h>
 #endif
 
diff -ru --new-file algae-3.7.0.orig/superlu/sp_coletree.c algae-3.7.0/superlu/sp_coletree.c
--- algae-3.7.0.orig/superlu/sp_coletree.c	Wed Aug 14 02:21:22 2002
+++ algae-3.7.0/superlu/sp_coletree.c	Wed Aug 14 03:00:18 2002
@@ -2,7 +2,6 @@
 /*  Elimination tree computation and layout routines */
 
 #include <stdio.h>
-#include <malloc.h>
 #include <stdlib.h>
 #include "util.h"
 
diff -ru --new-file algae-3.7.0.orig/superlu/supermatrix.h algae-3.7.0/superlu/supermatrix.h
--- algae-3.7.0.orig/superlu/supermatrix.h	Wed Aug 14 02:21:22 2002
+++ algae-3.7.0/superlu/supermatrix.h	Wed Aug 14 02:42:27 2002
@@ -15,6 +15,11 @@
     DN         /* Fortran style column-wise storage for dense matrix */
 } Stype_t;
 
+#undef _S
+#undef _D
+#undef _C
+#undef _Z
+
 typedef enum {
     _S,         /* single */
     _D,         /* double */
diff -ru --new-file algae-3.7.0.orig/superlu/util.h algae-3.7.0/superlu/util.h
--- algae-3.7.0.orig/superlu/util.h	Wed Aug 14 02:21:22 2002
+++ algae-3.7.0/superlu/util.h	Wed Aug 14 02:33:28 2002
@@ -4,7 +4,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <malloc.h>
 #include <assert.h>
 
 #include "../src/algae.h"
