diff -ru tcl8.4.1/unix/configure tcl8.4.1-patched/unix/configure
--- tcl8.4.1/unix/configure	Tue Oct 22 15:18:39 2002
+++ tcl8.4.1-patched/unix/configure	Tue Aug  5 00:31:27 2003
@@ -7172,7 +7172,7 @@
 elif test "$prefix" != "$exec_prefix"; then
     TCL_PACKAGE_PATH="${libdir} ${prefix}/lib"
 else
-    TCL_PACKAGE_PATH="${prefix}/lib"
+    TCL_PACKAGE_PATH="${prefix}/lib /System/Library/Tcl /usr/lib"
 fi
 
 #--------------------------------------------------------------------
