--- gmerlin-avdecoder-1.1.0/lib/aac_frame.c	2009-05-24 08:10:42.000000000 -0700
+++ gmerlin-avdecoder-1.1.0/lib/aac_frame.c	2009-12-28 22:47:14.000000000 -0800
@@ -26,11 +26,11 @@
 #include <aac_frame.h>
 #include <stdlib.h>
 
+#define LOG_DOMAIN "aac_frame"
+
 #ifdef HAVE_NEAACDEC_H
 #include <neaacdec.h>
 
-#define LOG_DOMAIN "aac_frame"
-
 /*
  *  Backwards compatibility names (currently in neaacdec.h,
  *  but might be removed in future versions)
