--- gnu.regexp-1.1.4/src/Makefile	2001-10-21 14:26:42.000000000 -0700
+++ gnu.regexp-1.1.4-patched/src/Makefile	2005-12-08 01:47:01.000000000 -0800
@@ -57,12 +57,12 @@
 # Change JAVAC variable to your java compiler.
 # Change JAVAFLAGS variable to any java flags you want to compile with.
 
-JAVAC = jikes
-JAVAFLAGS = -O
-
-#JAVAC = javac
+#JAVAC = jikes
 #JAVAFLAGS = -O
 
+JAVAC = javac
+JAVAFLAGS = -O
+
 JAVACOMPILE = $(JAVAC) -d "$(LIBPATH)" $(JAVAFLAGSCLASSPATH) $(JAVAFLAGS)
 
 # This is the directory the gnu.regexp JAR file will be installed in when
