diff -Nurd poppler-0.6.4/qt4/src/Makefile.in poppler-0.6.4-new/qt4/src/Makefile.in
--- poppler-0.6.4/qt4/src/Makefile.in	2008-01-25 17:12:28.000000000 -0500
+++ poppler-0.6.4-new/qt4/src/Makefile.in	2009-03-07 11:41:57.000000000 -0500
@@ -38,7 +38,7 @@
 build_triplet = @build@
 host_triplet = @host@
 @BUILD_SPLASH_OUTPUT_TRUE@am__append_1 = \
-@BUILD_SPLASH_OUTPUT_TRUE@	$(top_builddir)/poppler/libpoppler-arthur.la
+@BUILD_SPLASH_OUTPUT_TRUE@	
 
 subdir = qt4/src
 DIST_COMMON = $(poppler_include_HEADERS) $(srcdir)/Makefile.am \
@@ -64,9 +64,9 @@
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
-@BUILD_SPLASH_OUTPUT_TRUE@am__DEPENDENCIES_2 = $(top_builddir)/poppler/libpoppler-arthur.la
+@BUILD_SPLASH_OUTPUT_TRUE@am__DEPENDENCIES_2 = 
 libpoppler_qt4_la_DEPENDENCIES =  \
-	$(top_builddir)/poppler/libpoppler.la $(am__DEPENDENCIES_1) \
+	$(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_2)
 am_libpoppler_qt4_la_OBJECTS = poppler-document.lo poppler-page.lo \
 	poppler-fontinfo.lo poppler-embeddedfile.lo poppler-textbox.lo \
@@ -299,7 +299,7 @@
 	poppler-annotation-private.h		\
 	poppler-private.h
 
-libpoppler_qt4_la_LIBADD = $(top_builddir)/poppler/libpoppler.la \
+libpoppler_qt4_la_LIBADD = -lpoppler \
 	$(FONTCONFIG_LIBS) $(POPPLER_QT4_LIBS) $(am__append_1)
 libpoppler_qt4_la_LDFLAGS = -version-info 2:0:0
 all: all-am
diff -Nurd poppler-0.6.4/qt4/tests/Makefile.in poppler-0.6.4-new/qt4/tests/Makefile.in
--- poppler-0.6.4/qt4/tests/Makefile.in	2008-01-25 17:12:28.000000000 -0500
+++ poppler-0.6.4-new/qt4/tests/Makefile.in	2009-03-07 11:42:09.000000000 -0500
@@ -67,7 +67,7 @@
 @BUILD_POPPLER_QT4_TRUE@	check_attachments.$(OBJEXT)
 check_attachments_OBJECTS = $(am_check_attachments_OBJECTS)
 am__DEPENDENCIES_1 =
-am__DEPENDENCIES_2 = $(top_builddir)/poppler/libpoppler.la \
+am__DEPENDENCIES_2 = \
 	$(top_builddir)/qt4/src/libpoppler-qt4.la \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 @BUILD_POPPLER_QT4_TRUE@check_attachments_DEPENDENCIES =  \
@@ -337,7 +337,7 @@
 	$(POPPLER_QT4_CFLAGS)
 
 LDADDS = \
-	$(top_builddir)/poppler/libpoppler.la	\
+	-lpoppler \
 	$(top_builddir)/qt4/src/libpoppler-qt4.la	\
 	$(FONTCONFIG_LIBS)				\
 	$(POPPLER_QT4_LIBS)
