diff -uNr spice-jndikit-1.2/build.xml spice-jndikit-1.2-new/build.xml
--- spice-jndikit-1.2/build.xml	2006-01-17 00:18:12.000000000 -0500
+++ spice-jndikit-1.2-new/build.xml	2006-06-28 14:17:51.000000000 -0400
@@ -58,7 +58,7 @@
       </fileset>
     </copy>
   </target>
-  <target name="jar" description="o Create the jar" depends="compile,test">
+  <target name="jar" description="o Create the jar" depends="compile">
     <jar jarfile="${defaulttargetdir}/${final.name}.jar" excludes="**/package.html" basedir="${classesdir}">
     </jar>
   </target>
@@ -78,7 +78,7 @@
       </fileset>
     </copy>
   </target>
-  <target name="test" description="o Run the test cases" if="test.failure" depends="internal-test">
+  <target name="test" description="o Run the test cases" if="test.failure" depends="">
     <fail message="There were test failures.">
     </fail>
   </target>
@@ -165,4 +165,4 @@
     <unjar dest="${maven.home}" src="${user.home}/maven-install-latest.jar">
     </unjar>
   </target>
-</project>
\ No newline at end of file
+</project>
diff -uNr spice-jndikit-1.2/src/java/org/codehaus/spice/jndikit/rmi/server/RMINamingProviderImpl.java spice-jndikit-1.2-new/src/java/org/codehaus/spice/jndikit/rmi/server/RMINamingProviderImpl.java
--- spice-jndikit-1.2/src/java/org/codehaus/spice/jndikit/rmi/server/RMINamingProviderImpl.java	2006-01-17 00:18:12.000000000 -0500
+++ spice-jndikit-1.2-new/src/java/org/codehaus/spice/jndikit/rmi/server/RMINamingProviderImpl.java	2006-06-28 14:16:24.000000000 -0400
@@ -76,12 +76,12 @@
     {
         //Remember that the bindings returned by this
         //actually have a nested Binding as an object
-        final NamingEnumeration enum = m_root.listBindings( name );
+        final NamingEnumeration e = m_root.listBindings( name );
         final ArrayList pairs = new ArrayList();
 
-        while( enum.hasMore() )
+        while( e.hasMore() )
         {
-            final Binding binding = (Binding)enum.next();
+            final Binding binding = (Binding)e.next();
             final Object object = binding.getObject();
 
             String className = null;
@@ -114,12 +114,12 @@
     {
         //Remember that the bindings returned by this
         //actually have a nested Binding as an object
-        final NamingEnumeration enum = m_root.listBindings( name );
+        final NamingEnumeration e = m_root.listBindings( name );
         final ArrayList bindings = new ArrayList();
 
-        while( enum.hasMore() )
+        while( e.hasMore() )
         {
-            final Binding binding = (Binding)enum.next();
+            final Binding binding = (Binding)e.next();
             Object object = binding.getObject();
             String className = null;
 
diff -uNr spice-jndikit-1.2/src/test/org/codehaus/spice/jndikit/test/AbstractContextTestCase.java spice-jndikit-1.2-new/src/test/org/codehaus/spice/jndikit/test/AbstractContextTestCase.java
--- spice-jndikit-1.2/src/test/org/codehaus/spice/jndikit/test/AbstractContextTestCase.java	2006-01-17 00:18:12.000000000 -0500
+++ spice-jndikit-1.2-new/src/test/org/codehaus/spice/jndikit/test/AbstractContextTestCase.java	2006-06-28 14:16:24.000000000 -0400
@@ -105,17 +105,17 @@
             m_context.unbind( "O7" );
             m_context.unbind( "O8" );
 
-            final NamingEnumeration enum = m_context.list( "" );
+            final NamingEnumeration e = m_context.list( "" );
 
-            if( enum.hasMoreElements() )
+            if( e.hasMoreElements() )
             {
                 fail( "Failed to unbind all test elements: ie " +
-                      enum.nextElement() );
+                      e.nextElement() );
             }
 
             try
             {
-                enum.nextElement();
+                e.nextElement();
                 fail( "Expected nextElement() to throw NoSuchElementException" );
             }
             catch( final NoSuchElementException nsee )
@@ -175,17 +175,17 @@
             m_context.unbind( "x/O2" );
             m_context.unbind( "x/O3" );
 
-            final Enumeration enum = m_context.list( "x/" );
+            final Enumeration e = m_context.list( "x/" );
 
-            if( enum.hasMoreElements() )
+            if( e.hasMoreElements() )
             {
                 fail( "Failed to unbind all test elements: ie " +
-                      enum.nextElement() );
+                      e.nextElement() );
             }
 
             try
             {
-                enum.nextElement();
+                e.nextElement();
                 fail( "Expected nextElement() to throw NoSuchElementException" );
             }
             catch( final NoSuchElementException nsee )
@@ -298,17 +298,17 @@
             m_context.unbind( "x/O3" );
 
 
-            final Enumeration enum = m_context.list( "x/y" );
+            final Enumeration e = m_context.list( "x/y" );
 
-            if( enum.hasMoreElements() )
+            if( e.hasMoreElements() )
             {
                 fail( "Failed to unbind all test elements: ie " +
-                      enum.nextElement() );
+                      e.nextElement() );
             }
 
             try
             {
-                enum.nextElement();
+                e.nextElement();
                 fail( "Expected nextElement() to throw NoSuchElementException" );
             }
             catch( final NoSuchElementException nsee )
@@ -1068,11 +1068,11 @@
             ExceptionReferenceable trigger = new ExceptionReferenceable();
             m_context.bind( "o4", trigger );
 
-            NamingEnumeration enum = m_context.listBindings( "" );
-            assertTrue( enum.hasMore() );
+            NamingEnumeration e = m_context.listBindings( "" );
+            assertTrue( e.hasMore() );
             try
             {
-                Object result = enum.next();
+                Object result = e.next();
                 fail( "Expected nextElement to throw NamingExceptionn, but returned "
                       + result );
             }
@@ -1100,11 +1100,11 @@
             ExceptionReferenceable trigger = new ExceptionReferenceable();
             m_context.bind( "o5", trigger );
 
-            NamingEnumeration enum = m_context.listBindings( "" );
-            assertTrue( enum.hasMoreElements() );
+            NamingEnumeration e = m_context.listBindings( "" );
+            assertTrue( e.hasMoreElements() );
             try
             {
-                Object result = enum.nextElement();
+                Object result = e.nextElement();
                 fail( "Expected nextElement to throw NoSuchElementException, but returned "
                       + result );
             }
