diff -Naur acme-2.4.0-old/src/Makefile.in acme-2.4.0/src/Makefile.in
--- acme-2.4.0-old/src/Makefile.in	Wed Sep 10 05:58:16 2003
+++ acme-2.4.0/src/Makefile.in	Thu Oct 30 12:11:52 2003
@@ -136,7 +136,8 @@
 acme_LDFLAGS = \
 	$(EXTRA_GNOME_LIBS)				\
 	$(ALSA_LIBS)					\
-	$(INTLLIBS)
+	$(INTLLIBS)					\
+	-lX11
 
 
 acme_properties_SOURCES = \
@@ -156,7 +157,8 @@
 
 acme_properties_LDFLAGS = \
 	$(EXTRA_GNOME_LIBS)			\
-	$(INTLLIBS)
+	$(INTLLIBS)				\
+	-lX11
 
 
 EXTRA_DIST = \
