diff -Naur control-center-2.4.0.old/configure control-center-2.4.0.new/configure
--- control-center-2.4.0.old/configure	Mon Sep  8 23:57:09 2003
+++ control-center-2.4.0.new/configure	Mon Dec  8 16:26:20 2003
@@ -21608,7 +21608,7 @@
 
 fi
 
-x_libs="$X_PRE_LIBS $X_LIBS -lX11 $X_EXTRA_LIBS"
+x_libs="$X_PRE_LIBS $X_LIBS -lX11 -lXext $X_EXTRA_LIBS"
 
 savecppflags=$CPPFLAGS
 CPPFLAGS="$CPPFLAGS $X_CFLAGS"
diff -Naur control-center-2.4.0.old/libwindow-settings/Makefile.in control-center-2.4.0.new/libwindow-settings/Makefile.in
--- control-center-2.4.0.old/libwindow-settings/Makefile.in	Mon Sep  8 23:57:52 2003
+++ control-center-2.4.0.new/libwindow-settings/Makefile.in	Mon Dec  8 16:28:06 2003
@@ -224,7 +224,7 @@
 	wm-properties.h
 
 
-wms_flags = -export_dynamic -avoid-version
+wms_flags = -export_dynamic -avoid-version -module
 wmsdir = $(WM_MODULE_DIR)
 
 wms_LTLIBRARIES = \
diff -Naur control-center-2.4.0.old/typing-break/Makefile.in control-center-2.4.0.new/typing-break/Makefile.in
--- control-center-2.4.0.old/typing-break/Makefile.in	Mon Sep  8 23:58:41 2003
+++ control-center-2.4.0.new/typing-break/Makefile.in	Mon Dec  8 16:26:20 2003
@@ -204,7 +204,7 @@
 	egg-spawn.h
 
 
-gnome_typing_monitor_LDADD = @GNOME_LIBS@ -L/usr/X11R6/lib -lXss
+gnome_typing_monitor_LDADD = @GNOME_LIBS@ -L/usr/X11R6/lib -lXss -lXext
 gnome_typing_monitor_LDFLAGS = -export-dynamic
 
 imagedir = $(GNOMECC_PIXMAPS_DIR)
