diff -Naur gedit-2.4.0-old/gedit/gedit-plugins-engine.c gedit-2.4.0/gedit/gedit-plugins-engine.c
--- gedit-2.4.0-old/gedit/gedit-plugins-engine.c	Thu Jan  2 17:43:31 2003
+++ gedit-2.4.0/gedit/gedit-plugins-engine.c	Fri Oct 17 23:04:41 2003
@@ -31,6 +31,7 @@
 #include <config.h>
 #endif
 
+#include <sys/types.h> 
 #include <dirent.h> 
 #include <string.h>
 
diff -Naur gedit-2.4.0-old/ltmain.sh gedit-2.4.0/ltmain.sh
--- gedit-2.4.0-old/ltmain.sh	Thu Jun 12 05:02:56 2003
+++ gedit-2.4.0/ltmain.sh	Fri Oct 17 23:03:25 2003
@@ -3678,6 +3678,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+	  else
+	    tmp_verstring="$verstring"
+	  fi
 	  eval cmds=\"$archive_cmds\"
 	  fi
 	fi
