diff -Naur gnome-desktop-2.4.0-old/libgnome-desktop/egg-spawn.c gnome-desktop-2.4.0/libgnome-desktop/egg-spawn.c
--- gnome-desktop-2.4.0-old/libgnome-desktop/egg-spawn.c	Tue Dec  3 03:01:21 2002
+++ gnome-desktop-2.4.0/libgnome-desktop/egg-spawn.c	Thu Oct 16 08:45:19 2003
@@ -28,6 +28,10 @@
 #include <glib.h>
 #include <gdk/gdk.h>
 
+#include <sys/time.h>
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+
 extern char **environ;
 
 /**
diff -Naur gnome-desktop-2.4.0-old/libgnome-desktop/gnome-desktop-item.c gnome-desktop-2.4.0/libgnome-desktop/gnome-desktop-item.c
--- gnome-desktop-2.4.0-old/libgnome-desktop/gnome-desktop-item.c	Fri Aug  1 10:51:49 2003
+++ gnome-desktop-2.4.0/libgnome-desktop/gnome-desktop-item.c	Thu Oct 16 08:44:25 2003
@@ -29,6 +29,9 @@
 
 #include "config.h"
 
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+
 #include <limits.h>
 #include <ctype.h>
 #include <stdio.h>
@@ -36,7 +39,7 @@
 #include <sys/types.h>
 #include <dirent.h>
 #include <unistd.h>
-#include <time.h>
+#include <sys/time.h>
 #include <string.h>
 #include <libgnome/gnome-i18n.h>
 #include <libgnome/gnome-util.h>
diff -Naur gnome-desktop-2.4.0-old/ltmain.sh gnome-desktop-2.4.0/ltmain.sh
--- gnome-desktop-2.4.0-old/ltmain.sh	Mon May 19 04:38:38 2003
+++ gnome-desktop-2.4.0/ltmain.sh	Thu Oct 16 08:44:25 2003
@@ -5145,10 +5145,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test "$need_relink" = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
