diff -ru libgnomeprint-2.4.2/configure libgnomeprint-2.4.2-patched/configure
--- libgnomeprint-2.4.2/configure	Wed Nov 26 15:37:56 2003
+++ libgnomeprint-2.4.2-patched/configure	Sat Jan 10 09:00:54 2004
@@ -22560,7 +22560,7 @@
 GNOME_PRINT_DATA_DIR=$GNOME_PRINT_DATA_DIR
 
 
-LIBGNOMEPRINT_CFLAGS="$GP_CFLAGS $FREETYPE_CFLAGS"
+LIBGNOMEPRINT_CFLAGS="$FREETYPE_CFLAGS $GP_CFLAGS"
 LIBGNOMEPRINT_LIBS="$GP_LIBS $FREETYPE_LIBS"
 
 
diff -ru libgnomeprint-2.4.2/libgnomeprint/gnome-fontmap.c libgnomeprint-2.4.2-patched/libgnomeprint/gnome-fontmap.c
--- libgnomeprint-2.4.2/libgnomeprint/gnome-fontmap.c	Thu Feb 13 01:15:47 2003
+++ libgnomeprint-2.4.2-patched/libgnomeprint/gnome-fontmap.c	Sat Jan 10 09:02:55 2004
@@ -194,6 +194,8 @@
 		font_type = GP_FONT_ENTRY_TYPE1;
 	else if (strcasecmp (c, ".ttf") == 0)
 		font_type = GP_FONT_ENTRY_TRUETYPE;
+        else if (strcasecmp (c, ".otf") == 0)
+                font_type = GP_FONT_ENTRY_TRUETYPE;
 	else if (strcasecmp (c, ".font") == 0) /* See #102400 */
 		font_type = GP_FONT_ENTRY_TRUETYPE;
 	else {
diff -ru libgnomeprint-2.4.2/libgnomeprint/gnome-print.c libgnomeprint-2.4.2-patched/libgnomeprint/gnome-print.c
--- libgnomeprint-2.4.2/libgnomeprint/gnome-print.c	Wed Mar 12 14:29:31 2003
+++ libgnomeprint-2.4.2-patched/libgnomeprint/gnome-print.c	Sat Jan 10 09:03:26 2004
@@ -28,6 +28,7 @@
 #include <config.h>
 #include <string.h>
 #include <gmodule.h>
+#include <sys/types.h>
 
 #include <libgnomeprint/gnome-print-i18n.h>
 #include <libgnomeprint/gnome-print-private.h>
diff -ru libgnomeprint-2.4.2/ltmain.sh libgnomeprint-2.4.2-patched/ltmain.sh
--- libgnomeprint-2.4.2/ltmain.sh	Tue Nov 18 12:48:39 2003
+++ libgnomeprint-2.4.2-patched/ltmain.sh	Sat Jan 10 09:09:44 2004
@@ -3830,7 +3830,17 @@
 	  if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	    eval cmds=\"$archive_expsym_cmds\"
 	  else
-	    eval cmds=\"$archive_cmds\"
+            if test "x$verstring" = "x0.0"; then
+              tmp_verstring=
+             else
+               tmp_verstring="$verstring"
+             fi
+             save_deplibs="$deplibs"
+             for conv in $convenience; do
+               deplibs="${deplibs%$conv*} ${deplibs#*$conv}"
+             done
+            eval cmds=\"$archive_cmds\"a
+            deplibs="$save_deplibs"
 	  fi
 
 	  # Append the command to remove the reloadable object files
