diff -Naur libgnomeui-2.4.0.1-old/config.h.in libgnomeui-2.4.0.1/config.h.in
--- libgnomeui-2.4.0.1-old/config.h.in	Sun Aug 17 07:03:20 2003
+++ libgnomeui-2.4.0.1/config.h.in	Wed Oct 15 19:12:37 2003
@@ -20,6 +20,10 @@
 /* Define to 1 if you have the `bind_textdomain_codeset' function. */
 #undef HAVE_BIND_TEXTDOMAIN_CODESET
 
+/* Define if you have the `strtok_r' function. */
+#undef HAVE_STRTOK_R
+
+
 /* Define to 1 if you have the `dcgettext' function. */
 #undef HAVE_DCGETTEXT
 
diff -Naur libgnomeui-2.4.0.1-old/configure libgnomeui-2.4.0.1/configure
--- libgnomeui-2.4.0.1-old/configure	Tue Sep  9 09:24:00 2003
+++ libgnomeui-2.4.0.1/configure	Wed Oct 15 19:04:59 2003
@@ -19457,7 +19457,7 @@
 
       if test "$gt_cv_func_dgettext_libc" = "yes" ; then
 
-for ac_func in bind_textdomain_codeset
+for ac_func in bind_textdomain_codeset strtok_r
 do
 as_ac_var=`echo "ac_cv_func_$ac_func" | $as_tr_sh`
 echo "$as_me:$LINENO: checking for $ac_func" >&5
diff -Naur libgnomeui-2.4.0.1-old/doc/reference/Makefile.in libgnomeui-2.4.0.1/doc/reference/Makefile.in
--- libgnomeui-2.4.0.1-old/doc/reference/Makefile.in	Tue Sep  9 09:31:18 2003
+++ libgnomeui-2.4.0.1/doc/reference/Makefile.in	Wed Oct 15 19:04:59 2003
@@ -173,8 +173,8 @@
 
 # CFLAGS and LDFLAGS for compiling scan program. Only needed if your app/lib
 # contains GtkObjects/GObjects and you want to document signals and properties.
-GTKDOC_CFLAGS = -I$(top_builddir) -I$(top_srcdir) $(LIBGNOMEUI_CFLAGS) $(CFLAGS)
-GTKDOC_LIBS = $(top_builddir)/libgnomeui/libgnomeui-2.la $(LDFLAGS)
+GTKDOC_CFLAGS = -I$(top_builddir) -I$(top_srcdir) $(LIBGNOMEUI_CFLAGS) $(CFLAGS) $(CPPFLAGS)
+GTKDOC_LIBS = $(top_builddir)/libgnomeui/libgnomeui-2.la $(LIBGNOMEUI_LIBS) $(SM_LIBS) $(LDFLAGS)
 
 GTKDOC_CC = $(LIBTOOL) --mode=compile $(CC)
 GTKDOC_LD = $(LIBTOOL) --mode=link $(CC)
diff -Naur libgnomeui-2.4.0.1-old/libgnomeui/Makefile.in libgnomeui-2.4.0.1/libgnomeui/Makefile.in
--- libgnomeui-2.4.0.1-old/libgnomeui/Makefile.in	Tue Sep  9 09:31:07 2003
+++ libgnomeui-2.4.0.1/libgnomeui/Makefile.in	Wed Oct 15 19:04:59 2003
@@ -183,7 +183,7 @@
 libgnomeui_2_la_LIBADD =  	$(LIBGNOMEUI_LIBS)				$(SM_LIBS)
 
 
-LDADD =  	libgnomeui-2.la
+LDADD =  	libgnomeui-2.la $(LIBGNOMEUI_LIBS) $(SM_LIBS)
 
 
 pkgconfigdir = $(libdir)/pkgconfig
diff -Naur libgnomeui-2.4.0.1-old/libgnomeui/gnome-dateedit.c libgnomeui-2.4.0.1/libgnomeui/gnome-dateedit.c
--- libgnomeui-2.4.0.1-old/libgnomeui/gnome-dateedit.c	Thu May  1 12:31:04 2003
+++ libgnomeui-2.4.0.1/libgnomeui/gnome-dateedit.c	Wed Oct 15 19:05:18 2003
@@ -101,6 +101,31 @@
 /* to get around warnings */
 static const char *strftime_date_format = "%x";
 
+#ifndef HAVE_STRTOK_R
+static char *
+strtok_r (char *s, const char *delim, char **save_ptr)
+{
+	char *token;
+
+	if (s == NULL)
+		s = *save_ptr;
+
+	s += strspn (s, delim);
+	if (*s == '\0')
+		return NULL;
+
+	token = s;
+	s = strpbrk (token, delim);
+	if (s == NULL) {
+		*save_ptr = strchr (token, '\0');
+	} else {
+		*s = '\0';
+		*save_ptr = s + 1;
+	}
+	return token;
+}
+#endif
+
 /**
  * gnome_date_edit_get_type:
  *
diff -Naur libgnomeui-2.4.0.1-old/libgnomeui/libgnomeui-access.c libgnomeui-2.4.0.1/libgnomeui/libgnomeui-access.c
--- libgnomeui-2.4.0.1-old/libgnomeui/libgnomeui-access.c	Tue May 21 13:26:36 2002
+++ libgnomeui-2.4.0.1/libgnomeui/libgnomeui-access.c	Wed Oct 15 19:05:18 2003
@@ -30,7 +30,7 @@
 static gint is_gail_loaded (GtkWidget *widget);
 
 /* variable that indicates whether GAIL is loaded or not */
-gint gail_loaded = -1;
+static gint gail_loaded = -1;
 
 /* Accessibility support routines for libgnomeui */
 static gint
diff -Naur libgnomeui-2.4.0.1-old/ltmain.sh libgnomeui-2.4.0.1/ltmain.sh
--- libgnomeui-2.4.0.1-old/ltmain.sh	Sun Aug 17 04:30:36 2003
+++ libgnomeui-2.4.0.1/ltmain.sh	Wed Oct 15 19:10:31 2003
@@ -3678,6 +3678,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+	  else
+	    tmp_verstring="$verstring"
+	  fi
 	  eval cmds=\"$archive_cmds\"
 	  fi
 	fi
@@ -5145,10 +5150,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test "$need_relink" = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
diff -Naur libgnomeui-2.4.0.1-old/test-gnome/Makefile.in libgnomeui-2.4.0.1/test-gnome/Makefile.in
--- libgnomeui-2.4.0.1-old/test-gnome/Makefile.in	Tue Sep  9 09:31:15 2003
+++ libgnomeui-2.4.0.1/test-gnome/Makefile.in	Wed Oct 15 19:05:18 2003
@@ -135,7 +135,7 @@
 INCLUDES =  	-I$(top_builddir)					-I$(top_srcdir)						-I$(top_srcdir)/libgnomeui				$(WARN_CFLAGS)						$(LIBGNOMEUI_CFLAGS)					-DGNOMEUILIBDIR=\""$(libdir)"\"				-DGNOMEUIDATADIR=\""$(datadir)"\"			-DGNOMEUIUIDIR=\""$(datadir)/ui"\"			-DGNOMEUIPIXMAPDIR=\""$(datadir)/pixmaps"\"		-DGNOMEUIBINDIR=\""$(bindir)"\"				-DGNOMEUILOCALSTATEDIR=\""$(localstatedir)"\"		-DGNOMEUISRCDIR=\""$(srcdir)"\"				-DG_LOG_DOMAIN=\"GnomeUI\"
 
 
-LDADD =  	$(top_builddir)/libgnomeui/libgnomeui-2.la 
+LDADD =  	$(top_builddir)/libgnomeui/libgnomeui-2.la $(LIBGNOMEUI_LIBS) $(SM_LIBS)
 
 
 noinst_HEADERS =  	testgnome.h
