diff -Naur libgnomeprintui-2.4.0-old/ltmain.sh libgnomeprintui-2.4.0/ltmain.sh
--- libgnomeprintui-2.4.0-old/ltmain.sh	Wed Sep 10 18:55:56 2003
+++ libgnomeprintui-2.4.0/ltmain.sh	Fri Oct 17 13:49:55 2003
@@ -3794,7 +3794,17 @@
 	  if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	    eval cmds=\"$archive_expsym_cmds\"
 	  else
+	    if test "x$verstring" = "x0.0"; then
+              tmp_verstring=
+            else
+              tmp_verstring="$verstring"
+            fi
+            save_deplibs="$deplibs"
+            for conv in $convenience; do
+              deplibs="${deplibs%$conv*} ${deplibs#*$conv}"
+            done
 	    eval cmds=\"$archive_cmds\"
+	    deplibs="$save_deplibs"
 	  fi
 
 	  # Append the command to remove the reloadable object files
