--- dvipdfmx-20031207/data/config/cid-x.map.org	Wed Mar  3 14:33:16 2004
+++ dvipdfmx-20031207/data/config/cid-x.map	Wed Mar  3 14:33:42 2004
@@ -92,14 +92,14 @@
 rmlv V Ryumin-Light
 gbmv V GothicBBB-Medium
 
-%ryumin-l    H Ryumin-Light
-%gtbbb-m     H GothicBBB-Medium
-%futomin-b   H Ryumin-Light,Bold
-%futogo-b    H GothicBBB-Medium,Bold
-%ryumin-l-v  V Ryumin-Light
-%gtbbb-m-v   V GothicBBB-Medium
-%futomin-b-v V Ryumin-Light,Bold
-%futogo-b-v  V GothicBBB-Medium,Bold
+ryumin-l    H Ryumin-Light
+gtbbb-m     H GothicBBB-Medium
+futomin-b   H Ryumin-Light,Bold
+futogo-b    H GothicBBB-Medium,Bold
+ryumin-l-v  V Ryumin-Light
+gtbbb-m-v   V GothicBBB-Medium
+futomin-b-v V Ryumin-Light,Bold
+futogo-b-v  V GothicBBB-Medium,Bold
 
 %% OpenType version sold by Morisawa:
 %rml  H RyuminPro-Light
--- dvipdfmx-20031207/data/config/dvipdfmx.cfg.org	Tue Mar  9 16:23:05 2004
+++ dvipdfmx-20031207/data/config/dvipdfmx.cfg	Tue Mar  9 16:26:16 2004
@@ -16,7 +16,7 @@
 
 % The following line works with GhostScript
 % UNIX/Linux (borrowed from W32TeX by Akira Kakuto)
-D "gs -q -dNOPAUSE -dBATCH -sPAPERSIZE=a0 -sDEVICE=pdfwrite -dCompatibilityLevel=1.3 -dAutoFilterGrayImages=false -dAutoFilterColorImages=false -dGrayImageFilter=/FlateEncode -dColorImageFilter=/FlateEncode -dUseFlateCompression=true -sOutputFile=%o %i -c quit"
+D "gs -q -dNOPAUSE -dBATCH -sPAPERSIZE=a0 -sDEVICE=pdfwrite -dCompatibilityLevel=1.3 -dAutoFilterGrayImages=false -dAutoFilterColorImages=false -dGrayImageFilter=/FlateEncode -dColorImageFilter=/FlateEncode -dUseFlateCompression=true -sOutputFile=%o -c '.setpdfwrite << /NeverEmbed [/Courier /Courier-Bold /Courier-Oblique /Courier-BoldOblique /Helvetica /Helvetica-Bold /Helvetica-Oblique /Helvetica-BoldOblique /Times-Roman /Times-Bold /Times-Italic /Times-BoldItalic /Symbol /ZapfDingbats /Ryumin-Light /GothicBBB-Medium] >> setdistillerparams' -f %i -c quit"
 % Win32 (borrowed from W32TeX by Akira Kakuto)
 % D "gswin32c -q -dNOPAUSE -dBATCH -sPAPERSIZE=a0 -sDEVICE=pdfwrite -dCompatibilityLevel=1.3 -dAutoFilterGrayImages=false -dAutoFilterColorImages=false -dGrayImageFilter=/FlateEncode -dColorImageFilter=/FlateEncode -dUseFlateCompression=true -sOutputFile=%o %i -c quit"
 % The following line works on compressed or uncompressed postscript with GhostScript
