diff -Nurd -x'*~' xmlto-0.0.18.orig/format/docbook/txt xmlto-0.0.18/format/docbook/txt
--- xmlto-0.0.18.orig/format/docbook/txt	2003-11-30 11:56:45.000000000 -0500
+++ xmlto-0.0.18/format/docbook/txt	2005-11-01 15:57:33.000000000 -0500
@@ -10,6 +10,10 @@
 then
   CONVERT=/usr/bin/links
   ARGS="-dump"
+elif [ -x /usr/bin/elinks ]
+then
+  CONVERT=/usr/bin/elinks
+  ARGS="-force-html -dump -no-numbering"
 else
   echo >&2 "No way to convert HTML to text found."
   exit 1
