diff -Nurd Algorithm-C3-0.08/Makefile.PL Algorithm-C3-0.08-new/Makefile.PL
--- Algorithm-C3-0.08/Makefile.PL	1969-12-31 19:00:00.000000000 -0500
+++ Algorithm-C3-0.08-new/Makefile.PL	2010-06-01 10:35:29.000000000 -0400
@@ -0,0 +1,7 @@
+use ExtUtils::MakeMaker;
+WriteMakefile(
+    NAME         => q[Algorithm::C3],
+    VERSION_FROM => q[lib/Algorithm/C3.pm],
+    INSTALLDIRS  => $] >= 5.007003 ? 'perl' : 'site',
+    PREREQ_PM    => { 'Carp' => '0.01' },
+);
