diff -Nurd -x'*~' Imager-0.96.orig/CountColor/Makefile.PL Imager-0.96/CountColor/Makefile.PL
--- Imager-0.96.orig/CountColor/Makefile.PL	2012-09-28 19:58:03.000000000 -0400
+++ Imager-0.96/CountColor/Makefile.PL	2013-05-30 22:47:51.000000000 -0400
@@ -8,6 +8,7 @@
    VERSION_FROM => 'CountColor.pm',
    OBJECT => 'CountColor.o',
    INC => '-I..',
+   exists $ENV{IM_DFLAGS} ? (DEFINE => $ENV{IM_DFLAGS}) : (),
   );
 my $MM_ver = eval $ExtUtils::MakeMaker::VERSION;
 if ($MM_ver > 6.06) {
diff -Nurd -x'*~' Imager-0.96.orig/DynTest/Makefile.PL Imager-0.96/DynTest/Makefile.PL
--- Imager-0.96.orig/DynTest/Makefile.PL	2012-09-28 19:58:03.000000000 -0400
+++ Imager-0.96/DynTest/Makefile.PL	2013-05-30 22:47:51.000000000 -0400
@@ -7,7 +7,8 @@
    NAME => 'Imager::Filter::DynTest',
    VERSION_FROM => 'DynTest.pm',
    OBJECT => 'DynTest.o linstretch.o',
-   INC => '-I..'
+   INC => '-I..',
+   exists $ENV{IM_DFLAGS} ? (DEFINE => $ENV{IM_DFLAGS}) : (),
   );
 my $MM_ver = eval $ExtUtils::MakeMaker::VERSION;
 if ($MM_ver > 6.06) {
diff -Nurd -x'*~' Imager-0.96.orig/Flines/Makefile.PL Imager-0.96/Flines/Makefile.PL
--- Imager-0.96.orig/Flines/Makefile.PL	2012-09-28 19:58:03.000000000 -0400
+++ Imager-0.96/Flines/Makefile.PL	2013-05-30 22:47:51.000000000 -0400
@@ -7,7 +7,8 @@
    NAME => 'Imager::Filter::Flines',
    VERSION_FROM => 'Flines.pm',
    OBJECT => 'Flines.o',
-   INC => '-I..'
+   INC => '-I..',
+   exists $ENV{IM_DFLAGS} ? (DEFINE => $ENV{IM_DFLAGS}) : (),
   );
 my $MM_ver = eval $ExtUtils::MakeMaker::VERSION;
 if ($MM_ver > 6.06) {
diff -Nurd -x'*~' Imager-0.96.orig/ICO/Makefile.PL Imager-0.96/ICO/Makefile.PL
--- Imager-0.96.orig/ICO/Makefile.PL	2012-09-28 19:58:04.000000000 -0400
+++ Imager-0.96/ICO/Makefile.PL	2013-05-30 22:47:51.000000000 -0400
@@ -9,6 +9,7 @@
    OBJECT => 'ICO.o msicon.o imicon.o',
    INC => '-I..',
    clean => { FILES => 'testout' },
+   exists $ENV{IM_DFLAGS} ? (DEFINE => $ENV{IM_DFLAGS}) : (),
   );
 my $MM_ver = eval $ExtUtils::MakeMaker::VERSION;
 if ($MM_ver > 6.06) {
diff -Nurd -x'*~' Imager-0.96.orig/Makefile.PL Imager-0.96/Makefile.PL
--- Imager-0.96.orig/Makefile.PL	2013-05-06 06:34:45.000000000 -0400
+++ Imager-0.96/Makefile.PL	2013-05-30 22:47:51.000000000 -0400
@@ -294,12 +294,17 @@
   }
 }
 
+my @want_good = (qw/ FT2 GIF JPEG PNG T1 TIFF /);
+my @want_bad = (qw/ Win32 /);
 print "\n";
 print "Libraries found:\n" if @good;
 print "  $_\n" for @good;
+print "(expected: @want_good)\n";
 print "Libraries *not* found:\n" if @bad;
 print "  $_\n" for @bad;
+print "(expected: @want_bad)\n";
 
+exit 1 unless ("@good" eq "@want_good" and "@bad" eq "@want_bad");
 exit;
 
 
@@ -492,7 +497,7 @@
 
 sub init {
 
-  my @definc = qw(/usr/include);
+  my @definc = qw(@PREFIX/include /usr/include);
   @definc{@definc}=(1) x @definc;
   @incs=
     (
@@ -509,12 +514,8 @@
     push @incs, grep -d, split /\Q$Config{path_sep}/, $Config{incpath};
   }
   push @incs, grep -d,
-      qw(/sw/include 
-         /usr/include/freetype2
-         /usr/local/include/freetype2
-         /usr/local/include/freetype1/freetype
-         /usr/include /usr/local/include /usr/include/freetype
-         /usr/local/include/freetype);
+      qw(@PREFIX@/include @PREFIX@/include/freetype2
+         /usr/include);
   if ($Config{ccflags}) {
     my @hidden = map { /^-I(.*)$/ ? ($1) : () } split ' ', $Config{ccflags};
     push @incs, @hidden;
@@ -546,7 +547,7 @@
     # it can't find and it doesn't look for -L in ldflags
     #@deflib{@hidden} = @hidden;
   }
-  push @libs, grep -d, qw(/usr/local/lib);
+  push @libs, grep -d, qw(@PREFIX@/lib);
 
   $formats{'TT-fonts'}=
     {
diff -Nurd -x'*~' Imager-0.96.orig/Mandelbrot/Makefile.PL Imager-0.96/Mandelbrot/Makefile.PL
--- Imager-0.96.orig/Mandelbrot/Makefile.PL	2012-09-28 19:58:04.000000000 -0400
+++ Imager-0.96/Mandelbrot/Makefile.PL	2013-05-30 22:47:51.000000000 -0400
@@ -7,7 +7,8 @@
    NAME => 'Imager::Filter::Mandelbrot',
    VERSION_FROM => 'Mandelbrot.pm',
    OBJECT => 'Mandelbrot.o mandel.o',
-   INC => '-I..'
+   INC => '-I..',
+   exists $ENV{IM_DFLAGS} ? (DEFINE => $ENV{IM_DFLAGS}) : (),
   );
 my $MM_ver = eval $ExtUtils::MakeMaker::VERSION;
 if ($MM_ver > 6.06) {
diff -Nurd -x'*~' Imager-0.96.orig/SGI/Makefile.PL Imager-0.96/SGI/Makefile.PL
--- Imager-0.96.orig/SGI/Makefile.PL	2012-09-28 19:58:04.000000000 -0400
+++ Imager-0.96/SGI/Makefile.PL	2013-05-30 22:47:51.000000000 -0400
@@ -9,6 +9,7 @@
    OBJECT => 'SGI.o imsgi.o',
    INC => '-I..',
    clean => { FILES => 'testout' },
+   exists $ENV{IM_DFLAGS} ? (DEFINE => $ENV{IM_DFLAGS}) : (),
   );
 my $MM_ver = eval $ExtUtils::MakeMaker::VERSION;
 if ($MM_ver > 6.06) {
diff -Nurd -x'*~' Imager-0.96.orig/dynfilt/Makefile.PL Imager-0.96/dynfilt/Makefile.PL
--- Imager-0.96.orig/dynfilt/Makefile.PL	2012-09-28 19:58:04.000000000 -0400
+++ Imager-0.96/dynfilt/Makefile.PL	2013-05-30 22:47:51.000000000 -0400
@@ -17,6 +17,7 @@
     NAME      => 'Imager::plugins',
     SKIP      => [qw(all dynamic static )],
     clean     => {'FILES' => $libstr},
+    exists $ENV{IM_DFLAGS} ? (DEFINE => $ENV{IM_DFLAGS}) : (),
 );
 
 
diff -Nurd -x'*~' Imager-0.96.orig/lib/Imager/Color.pm Imager-0.96/lib/Imager/Color.pm
--- Imager-0.96.orig/lib/Imager/Color.pm	2012-09-28 19:58:05.000000000 -0400
+++ Imager-0.96/lib/Imager/Color.pm	2013-05-30 22:47:51.000000000 -0400
@@ -157,6 +157,9 @@
 
 my @x_search =
   (
+   '/usr/X11/share/X11/rgb.txt',   # OS X modern xorg
+   '/usr/X11R6/share/X11/rgb.txt', # OS X older/compatibility
+
    '/usr/share/X11/rgb.txt', # newer Xorg X11 dists use this
    '/usr/lib/X11/rgb.txt', # seems fairly standard
    '/usr/local/lib/X11/rgb.txt', # seems possible
diff -Nurd -x'*~' Imager-0.96.orig/lib/Imager/Probe.pm Imager-0.96/lib/Imager/Probe.pm
--- Imager-0.96.orig/lib/Imager/Probe.pm	2013-05-18 22:02:32.000000000 -0400
+++ Imager-0.96/lib/Imager/Probe.pm	2013-05-30 22:47:51.000000000 -0400
@@ -64,6 +64,9 @@
 
   $result or return;
 
+  if (exists $ENV{IM_DFLAGS}) {
+      $result->{DEFINE} = $ENV{IM_DFLAGS} . ' ' . $result->{DEFINE};
+  }
   return $result;
 }
 
@@ -396,8 +399,8 @@
      ),
      $^O eq "MSWin32" ? $ENV{LIB} : "",
      $^O eq "cygwin" ? "/usr/lib/w32api" : "",
+     "@PREFIX@/lib",
      "/usr/lib",
-     "/usr/local/lib",
      _gcc_lib_paths(),
     );
 }
@@ -436,8 +439,8 @@
       grep $_,
       @Config{qw/locincpth incpath/}
      ),
+     "@PREFIX@/include",
      "/usr/include",
-     "/usr/local/include",
     );
 
   if ($req->{incsuffix}) {
diff -Nurd -x'*~' Imager-0.96.orig/t/200-file/100-files.t Imager-0.96/t/200-file/100-files.t
--- Imager-0.96.orig/t/200-file/100-files.t	2013-05-06 06:34:46.000000000 -0400
+++ Imager-0.96/t/200-file/100-files.t	2013-05-30 22:49:35.000000000 -0400
@@ -37,7 +37,6 @@
 PERL
   close SCRIPT;
   my $perl = $^X;
-  $perl = qq/"$perl"/ if $perl =~ / /;
   
   print "# script: $test_script\n";
   my $cmd = "$perl -Mblib $test_script t/200-file/100-files.t";
