diff -Nurd -x'*~' SDL-2.511.orig/inc/My/Builder.pm SDL-2.511/inc/My/Builder.pm
--- SDL-2.511.orig/inc/My/Builder.pm	2010-08-06 14:34:42.000000000 -0400
+++ SDL-2.511/inc/My/Builder.pm	2010-08-08 23:44:53.000000000 -0400
@@ -161,6 +161,7 @@
 		${ Alien::SDL::ConfigData->config('build_params') }{'title'} || 'n.a.'
 	);
 	$self->SUPER::ACTION_build;
+	$self->ACTION_bundle;
 }
 
 # both special to MacOS/Darwin, somebody should review whether it is still necessary
