diff -Nurd -x'*~' Sort-Packed-0.06.orig/t/pods.t Sort-Packed-0.06/t/pods.t
--- Sort-Packed-0.06.orig/t/pods.t	2009-11-17 07:48:26.000000000 -0500
+++ Sort-Packed-0.06/t/pods.t	2010-08-14 11:44:06.000000000 -0400
@@ -3,8 +3,7 @@
 use strict;
 use Test::More;
 
-plan skip_all => "Only the author needs to check that POD docs are right"
-    unless eval "no warnings; getlogin eq 'salva'";
+plan skip_all => "Only the author needs to check that POD docs are right";
 
 eval "use Test::Pod 1.00";
 plan skip_all => "Test::Pod 1.00 required for testing POD" if $@;
