diff -uNbr mc-4.5.55/configure mc-4.5.55-new/configure
--- mc-4.5.55/configure	Fri Aug 24 03:05:06 2001
+++ mc-4.5.55-new/configure	Fri Jul 26 10:16:49 2002
@@ -13487,9 +13487,9 @@
 #define USE_INTERNAL_EDIT 1
 EOF
 
-	LIBEDIT_A="libedit.a"
+	LIBEDIT_A="libedit_mc.a"
 	MCEDIT="mcedit"
-	LEDIT="-ledit"
+	LEDIT="-ledit_mc"
 	EDIT_msg="yes"
 	echo "$as_me:13494: result: will call internal editor" >&5
 echo "${ECHO_T}will call internal editor" >&6
@@ -13503,9 +13503,9 @@
 #define USE_INTERNAL_EDIT 1
 EOF
 
-	LIBEDIT_A="libedit.a"
+	LIBEDIT_A="libedit_mc.a"
 	MCEDIT="mcedit"
-	LEDIT="-ledit"
+	LEDIT="-ledit_mc"
 	EDIT_msg="yes"
 	echo "$as_me:13510: result: will call internal editor" >&5
 echo "${ECHO_T}will call internal editor" >&6
diff -uNbr mc-4.5.55/edit/Makefile.in mc-4.5.55-new/edit/Makefile.in
--- mc-4.5.55/edit/Makefile.in	Wed Aug  1 11:17:52 2001
+++ mc-4.5.55-new/edit/Makefile.in	Fri Jul 26 10:16:49 2002
@@ -49,10 +49,10 @@
 		$(LN_S) $(top_srcdir)/gtkedit/$$i . || exit 1; \
 	done
 
-libedit.a: $(EDITOBJS)
-	rm -f libedit.a
-	$(AR) cr libedit.a $(EDITOBJS)
-	-$(RANLIB) libedit.a
+libedit_mc.a: $(EDITOBJS)
+	rm -f libedit_mc.a
+	$(AR) cr libedit_mc.a $(EDITOBJS)
+	-$(RANLIB) libedit_mc.a
 
 mcedit:
 	rm -f $(DESTDIR)$(bindir)/$(binprefix)mcedit
@@ -65,7 +65,7 @@
 	rm -f *.o core a.out
 
 clean: mostlyclean
-	rm -f libedit.a
+	rm -f libedit_mc.a
 
 distclean: clean
 	rm -f $(EDITLINKS)
diff -uNbr mc-4.5.55/edit/Makefile.in~ mc-4.5.55-new/edit/Makefile.in~
--- mc-4.5.55/edit/Makefile.in~	Wed Dec 31 19:00:00 1969
+++ mc-4.5.55-new/edit/Makefile.in~	Wed Aug  1 11:17:52 2001
@@ -0,0 +1,88 @@
+srcdir = @srcdir@
+top_srcdir = @top_srcdir@
+VPATH = @srcdir@
+mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
+top_builddir = ..
+
+@MCF@
+
+CFLAGS = $(XCFLAGS)
+CPPFLAGS = $(XCPPFLAGS) -I$(top_srcdir) -I$(top_srcdir)/edit \
+	-I$(top_srcdir)/gtkedit
+LDFLAGS = $(XLDFLAGS) 
+DEFS = $(XDEFS)
+LIBS = @SHADOWLIB@ $(XLIBS) @TERMNET@ @PAMLIBS@ $(XLIB)
+INSTALL = @INSTALL@
+INSTALL_PROGRAM = @INSTALL_PROGRAM@
+INSTALL_DATA = @INSTALL_DATA@
+AR = @AR@
+
+#
+# Distribution variables
+#
+
+EDITSRC = edit_key_translator.c
+
+EDITLINKS = edit.c editcmd.c editwidget.c editdraw.c editmenu.c wordproc.c \
+	  syntax.c editoptions.c bookmark.c
+
+EDITOBJS = edit.o editcmd.o editwidget.o editdraw.o editmenu.o wordproc.o \
+	  syntax.o editoptions.o bookmark.o
+
+DIST = 	Makefile.in $(EDITSRC)
+
+all: @LIBEDIT_A@
+
+Makefile: $(srcdir)/Makefile.in $(top_builddir)/config.status
+	cd $(top_builddir) && CONFIG_FILES=edit/Makefile CONFIG_HEADERS= $(SHELL) ./config.status
+
+.c.o:
+	$(CC) -c $(CPPFLAGS) $(DEFS) $(CFLAGS) -DMIDNIGHT $<
+
+edit.o: edit_key_translator.c
+
+check:
+	@echo no tests are supplied.
+
+$(EDITLINKS):
+	for i in $(EDITLINKS); do \
+		$(LN_S) $(top_srcdir)/gtkedit/$$i . || exit 1; \
+	done
+
+libedit.a: $(EDITOBJS)
+	rm -f libedit.a
+	$(AR) cr libedit.a $(EDITOBJS)
+	-$(RANLIB) libedit.a
+
+mcedit:
+	rm -f $(DESTDIR)$(bindir)/$(binprefix)mcedit
+	$(LN_S) mc $(DESTDIR)$(bindir)/$(binprefix)mcedit
+
+TAGS: $(EDITSRC)
+	etags $(EDITSRC)
+
+mostlyclean:
+	rm -f *.o core a.out
+
+clean: mostlyclean
+	rm -f libedit.a
+
+distclean: clean
+	rm -f $(EDITLINKS)
+	rm -f TAGS
+	rm -f *~
+	rm -f Makefile
+
+install: all
+
+uninstall:
+
+distdir = $(top_builddir)/$(PACKAGE)-$(VERSION)/$(subdir)
+
+distdir:
+	for I in $(DIST); \
+	  do cp -p $(srcdir)/$$I $(distdir) || exit 1; \
+	done
+
+depend:
+
diff -uNbr mc-4.5.55/gtkedit/edit.h mc-4.5.55-new/gtkedit/edit.h
--- mc-4.5.55/gtkedit/edit.h	Sun Aug 19 12:01:01 2001
+++ mc-4.5.55-new/gtkedit/edit.h	Fri Jul 26 12:01:02 2002
@@ -134,7 +134,6 @@
 #    include "src/wtools.h"		/* for QuickWidgets */
 #    include "src/cmd.h"		/* for menu_edit_cmd */
 #    include "src/win.h"
-#    include "vfs/vfs.h"
 #    include "src/menu.h"
 #    include <regex.h>
 #    define WANT_WIDGETS
@@ -169,7 +168,6 @@
 #      define _(x) x
 #      define N_(x) x
 #   endif
-#include "vfs/vfs.h"
 #include "src/dialog.h"		/* for input_dialog() */
 #include "src/util.h"		/* for is_printable() */
 #    define CDisplay gdk_display
