diff -Naur alpine-2.24.orig/configure alpine-2.24/configure
--- alpine-2.24.orig/configure	2020-10-10 09:40:43.000000000 -0700
+++ alpine-2.24/configure	2021-04-29 10:09:15.000000000 -0700
@@ -17183,12 +17183,12 @@
 	alpine_SSLLIB="$alpine_SSLDIR/lib"
 	alpine_SSLCERTS="/opt/local/etc/openssl/certs"
 	alpine_SSLKEYS="/opt/local/etc/openssl/private"
-      elif test -d /sw/etc/ssl ; then
-	alpine_SSLDIR="/sw/etc/ssl"
-	alpine_SSLINCLUDE="/sw/include"
-	alpine_SSLLIB="/sw/lib"
-	alpine_SSLCERTS="/sw/etc/ssl/certs"
-	alpine_SSLKEYS="/sw/etc/ssl/private"
+      elif test -d /@FINKPREFIX@/etc/ssl ; then
+	alpine_SSLDIR="/@FINKPREFIX@/etc/ssl"
+	alpine_SSLINCLUDE="/@FINKPREFIX@/include"
+	alpine_SSLLIB="/@FINKPREFIX@/lib"
+	alpine_SSLCERTS="/@FINKPREFIX@/etc/ssl/certs"
+	alpine_SSLKEYS="/@FINKPREFIX@/etc/ssl/private"
       elif test -d /usr/local/ssl ; then
 	alpine_SSLDIR="/usr/local/ssl"
       elif test -d /System/Library/OpenSSL ; then
diff -Naur alpine-2.24.orig/configure.ac alpine-2.24/configure.ac
--- alpine-2.24.orig/configure.ac	2020-10-10 09:40:43.000000000 -0700
+++ alpine-2.24/configure.ac	2021-04-29 10:09:27.000000000 -0700
@@ -788,12 +788,12 @@
 	alpine_SSLLIB="$alpine_SSLDIR/lib"
 	alpine_SSLCERTS="/opt/local/etc/openssl/certs"
 	alpine_SSLKEYS="/opt/local/etc/openssl/private"
-      elif test -d /sw/etc/ssl ; then
-	alpine_SSLDIR="/sw/etc/ssl"
-	alpine_SSLINCLUDE="/sw/include"
-	alpine_SSLLIB="/sw/lib"
-	alpine_SSLCERTS="/sw/etc/ssl/certs"
-	alpine_SSLKEYS="/sw/etc/ssl/private"
+      elif test -d /@FINKPREFIX@/etc/ssl ; then
+	alpine_SSLDIR="/@FINKPREFIX@/etc/ssl"
+	alpine_SSLINCLUDE="/@FINKPREFIX@/include"
+	alpine_SSLLIB="/@FINKPREFIX@/lib"
+	alpine_SSLCERTS="/@FINKPREFIX@/etc/ssl/certs"
+	alpine_SSLKEYS="/@FINKPREFIX@/etc/ssl/private"
       elif test -d /usr/local/ssl ; then
 	alpine_SSLDIR="/usr/local/ssl"
       elif test -d /System/Library/OpenSSL ; then
