--- chaosreader/chaosreader0.94	2008-04-03 17:46:47.000000000 -0400
+++ chaosreader/chaosreader0.94	2008-04-03 17:28:05.000000000 -0400
@@ -6233,6 +6233,8 @@
 		$command = "tcpdump -s9999 -w";
 	} elsif ($^O eq "solaris") {
 		$command = "snoop -o";
+	} elsif ($^O eq "darwin") {
+		$command = "tcpdump -i en1 -s9999 -w";
 	} else {
 		die "ERROR54: Can't find the sniffer command for \"$^O\".\n" .
 		 "\t Please use log mode instead.\n";
