diff -ruN claws-mail-3.11.0-orig/src/plugins/python/clawsmailmodule.c claws-mail-3.11.0/src/plugins/python/clawsmailmodule.c
--- claws-mail-3.11.0-orig/src/plugins/python/clawsmailmodule.c	2014-10-20 15:05:31.000000000 -0500
+++ claws-mail-3.11.0/src/plugins/python/clawsmailmodule.c	2014-10-24 05:37:03.000000000 -0500
@@ -33,7 +33,9 @@
 #include "accounttype.h"
 #include "mailboxtype.h"
 
+#define NO_IMPORT_PYGOBJECT
 #include <pygobject.h>
+#define NO_IMPORT_PYGTK
 #include <pygtk/pygtk.h>
 
 #include "main.h"
