diff -Nurd doscan-0.3.3.orig/Makefile.in doscan-0.3.3/Makefile.in
--- doscan-0.3.3.orig/Makefile.in	2014-10-22 13:14:35.000000000 -0400
+++ doscan-0.3.3/Makefile.in	2024-02-19 01:40:37.000000000 -0500
@@ -21,7 +21,7 @@
 PCRE_CFLAGS := $(shell pcre-config --cflags)
 PCRE_LIBS := $(shell pcre-config --libs)
 
-INCLUDES = -I. -I$(srcdir) -I$(srcdir)/lib
+INCLUDES = -I. -I$(srcdir)
 LIBS = $(PCRE_LIBS) @LIBS@
 
 # Use generated inventory file if Git repository data is not present.
@@ -32,8 +32,6 @@
 endif
 
 SRC_C_FILES = $(patsubst src/%,%,$(filter src/%.cc, $(INVENTORY)))
-LIB_C_FILES = $(patsubst lib/%,%,$(filter lib/%.c, $(INVENTORY)))
-LIB_H_FILES = $(patsubst lib/%,%,$(filter lib/%.h, $(INVENTORY)))
 PROTO_C_FILES = $(filter proto_%.cc, $(SRC_C_FILES))
 
 DISTFILES_GENERATED = Make.inventory config.h.in stamp-h.in configure
@@ -85,12 +83,12 @@
 	echo timestamp > stamp-dir
 
 src/%.o : $(srcdir)/src/%.cc
-	$(CXX) $(CFLAGS) $(PCRE_CFLAGS) $(INCLUDES) \
+	$(CXX) $(CFLAGS) $(INCLUDES) $(PCRE_CFLAGS) \
 		-MMD -MF $(patsubst %.o,%.d, $@) \
 		-c -o $@ $<
 
 src/%.o : src/%.cc
-	$(CXX) $(CFLAGS) $(PCRE_CFLAGS) $(INCLUDES) \
+	$(CXX) $(CFLAGS) $(INCLUDES) $(PCRE_CFLAGS) \
 		-MMD -MF $(patsubst %.o,%.d, $@) \
 		-c -o $@ $<
 
@@ -112,7 +110,7 @@
 	@echo "}" >> $@
 
 lib/%.o : $(srcdir)/lib/%.c $(lib_depend)
-	$(CC) $(CFLAGS) $(PCRE_CFLAGS) $(INCLUDES) -c -o $@ $<
+	$(CC) $(CFLAGS) $(INCLUDES) $(PCRE_CFLAGS) -c -o $@ $<
 
 # Check sources for basic conformance rules.
 
