diff -Nurd -x'*~' epic5-1.1.11.orig/configure.in epic5-1.1.11/configure.in
--- epic5-1.1.11.orig/configure.in	2014-12-26 10:26:42.000000000 -0500
+++ epic5-1.1.11/configure.in	2017-01-30 10:35:43.000000000 -0500
@@ -1061,7 +1061,12 @@
 	AC_MSG_CHECKING(whether to support Perl)
 	AC_MSG_RESULT(no)
 else
-	AC_CHECK_PROG(PERLPROG, $perlexe, "yes", "no")
+	dnl Allow a full path to be specified in --with-perl
+	if test -x "$perlexe" ; then
+		PERLPROG=yes
+	else
+		AC_CHECK_PROG(PERLPROG, $perlexe, "yes", "no")
+	fi
 	AC_MSG_CHECKING(whether to support Perl)
 	if test $PERLPROG != "yes" ; then 
 		AC_MSG_RESULT(no)
@@ -1224,7 +1229,12 @@
 	AC_MSG_CHECKING(whether to support Ruby)
 	AC_MSG_RESULT(no)
 else
-	AC_CHECK_PROG(RUBYPROG, $rubyexe, "yes", "no")
+	dnl Support --with-ruby being passed a full path instead of just an executable name.
+	if test -x "$rubyexe" ; then
+		RUBYPROG=yes
+	else
+		AC_CHECK_PROG([RUBYPROG], [$rubyexe], [yes], [no])
+	fi
 	AC_MSG_CHECKING(whether to support Ruby)
 	if test $RUBYPROG != "yes" ; then 
 	    AC_MSG_RESULT(no)
