diff -Nurd ettercap-0.8.3.1.orig/src/CMakeLists.txt ettercap-0.8.3.1/src/CMakeLists.txt
--- ettercap-0.8.3.1.orig/src/CMakeLists.txt	2020-08-01 10:32:06.000000000 -0400
+++ ettercap-0.8.3.1/src/CMakeLists.txt	2024-02-19 04:21:37.000000000 -0500
@@ -231,7 +231,7 @@
   #  LINK_INTERFACE_LIBRARIES ""
   OUTPUT_NAME ettercap
   VERSION ${VERSION}
-  SOVERSION 0
+  SOVERSION ${VERSION}
   )
 
 install(TARGETS lib_ettercap DESTINATION ${INSTALL_LIBDIR})
diff -Nurd ettercap-0.8.3.1.orig/src/interfaces/CMakeLists.txt ettercap-0.8.3.1/src/interfaces/CMakeLists.txt
--- ettercap-0.8.3.1.orig/src/interfaces/CMakeLists.txt	2020-08-01 10:32:06.000000000 -0400
+++ ettercap-0.8.3.1/src/interfaces/CMakeLists.txt	2024-02-19 04:19:58.000000000 -0500
@@ -107,7 +107,7 @@
   #  LINK_INTERFACE_LIBRARIES ""
   OUTPUT_NAME ettercap-ui
   VERSION ${VERSION}
-  SOVERSION 0
+  SOVERSION ${VERSION}
   )
 target_link_libraries(ec_interfaces ${EC_INTERFACES_LIBS} ${EC_LIBS})
 install(TARGETS ec_interfaces DESTINATION ${INSTALL_LIBDIR})
