diff -Nurd -x'*~' jwhois-4.0.orig/src/Makefile.in jwhois-4.0/src/Makefile.in
--- jwhois-4.0.orig/src/Makefile.in	2007-07-01 04:56:14.000000000 -0400
+++ jwhois-4.0/src/Makefile.in	2016-08-27 13:12:38.000000000 -0400
@@ -230,7 +230,7 @@
          -DLOCALSTATEDIR="\"$(localstatedir)\"" \
          -DLOCALEDIR="\"$(localedir)\""
 
-jwhois_LDADD = $(top_srcdir)/gl/libgnu.a
+jwhois_LDADD = $(top_srcdir)/gl/libgnu.a $(INTLLIBS) $(LIBICONV)
 AM_CFLAGS = $(jwhois_CFLAGS)
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/gl
 all: all-am
diff -Nurd -x'*~' jwhois-4.0.orig/src/dns.c jwhois-4.0/src/dns.c
--- jwhois-4.0.orig/src/dns.c	2007-06-26 02:58:38.000000000 -0400
+++ jwhois-4.0/src/dns.c	2016-08-27 12:56:01.000000000 -0400
@@ -113,7 +113,7 @@
 lookup_host_addrinfo(struct addrinfo **res, const char *host, int port)
 {
   struct addrinfo hints;
-  char ascport[10] = "whois";
+  char ascport[10] = "nicname";
   int error;
 
   memset(&hints, 0, sizeof(hints));
diff -ruN jwhois-4.0-orig/configure jwhois-4.0/configure
--- jwhois-4.0-orig/configure	2007-07-01 03:56:12
+++ jwhois-4.0/configure	2023-10-20 06:30:05
@@ -8562,6 +8562,8 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 
+#include <stdlib.h>
+
 #ifdef HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #endif
diff -ruN jwhois-4.0-orig/src/jwhois.c jwhois-4.0/src/jwhois.c
--- jwhois-4.0-orig/src/jwhois.c	2007-06-26 01:59:35
+++ jwhois-4.0/src/jwhois.c	2023-10-20 06:35:39
@@ -71,6 +71,7 @@
 #endif
 
 int jwhois_query(struct s_whois_query *, char **);
+void timeout_init();
 
 int
 main(int argc, char **argv)
