diff -Nurd libproxy-0.4.18.orig/libproxy/cmake/modules/pacrunner_webkit.cmk libproxy-0.4.18/libproxy/cmake/modules/pacrunner_webkit.cmk
--- libproxy-0.4.18.orig/libproxy/cmake/modules/pacrunner_webkit.cmk	2022-06-20 09:54:49
+++ libproxy-0.4.18/libproxy/cmake/modules/pacrunner_webkit.cmk	2025-10-03 08:01:09
@@ -7,10 +7,6 @@
     include_directories("${WEBKIT_INCLUDE_DIR}")
   endif()
 elseif(APPLE)
-  find_library(WEBKIT_LIBRARIES JavaScriptCore)
-  if(WEBKIT_LIBRARIES)
-    set(WEBKIT_FOUND 1)
-  endif()
 else()
   if(WITH_WEBKIT3)
     px_check_modules(WEBKIT "javascriptcoregtk-4.0")
diff -Nurd libproxy-0.4.18.orig/libproxy/cmake/pkgconfig.cmk libproxy-0.4.18/libproxy/cmake/pkgconfig.cmk
--- libproxy-0.4.18.orig/libproxy/cmake/pkgconfig.cmk	2017-01-09 09:23:39
+++ libproxy-0.4.18/libproxy/cmake/pkgconfig.cmk	2025-10-03 08:06:09
@@ -1,4 +1,4 @@
-if(NOT WIN32 AND NOT APPLE)
+if(NOT WIN32)
   find_package(PkgConfig)
 
   # Define our magical px_check_modules function
diff -Nurd libproxy-0.4.18.orig/libproxy/test/CMakeLists.txt libproxy-0.4.18/libproxy/test/CMakeLists.txt
--- libproxy-0.4.18.orig/libproxy/test/CMakeLists.txt	2017-01-09 09:23:39
+++ libproxy-0.4.18/libproxy/test/CMakeLists.txt	2025-10-03 08:03:02
@@ -19,7 +19,7 @@
 add_test(NAME url-encode COMMAND ${CMAKE_CURRENT_BINARY_DIR}/url-encode)
 
 # Get PAC
-if (NOT WIN32)
+if (NOT WIN32 AND NOT APPLE)
 add_executable(get-pac-test ${CMAKE_CURRENT_SOURCE_DIR}/get-pac-test.cpp ${PROJECT_SOURCE_DIR}/url.cpp)
 target_link_libraries(get-pac-test pthread)
 add_test(NAME get-pac-test COMMAND ${CMAKE_CURRENT_BINARY_DIR}/get-pac-test)
