diff -Nurd -x'*~' libsoup-2.32.2.orig/libsoup/soup-gnutls.c libsoup-2.32.2/libsoup/soup-gnutls.c
--- libsoup-2.32.2.orig/libsoup/soup-gnutls.c	2010-11-23 10:45:20.000000000 -0500
+++ libsoup-2.32.2/libsoup/soup-gnutls.c	2014-03-25 00:25:51.000000000 -0400
@@ -56,12 +56,45 @@
 	SoupSSLType type;
 } SoupGNUTLSChannel;
 
+/*
+  taken from gnutls_certificate_verify_peers() from
+  gnutls-2.12.20/lib/gnutls_cert.c
+ */
+int
+_local_gnutls_certificate_verify_peers (gnutls_session_t session);
+
+int
+_local_gnutls_certificate_verify_peers (gnutls_session_t session)
+{
+  unsigned int status;
+  int ret;
+
+  ret = gnutls_certificate_verify_peers2 (session, &status);
+
+  if (ret < 0)
+    {
+	  /*
+	    gnutls_assert() is private in gnutls
+      gnutls_assert ();
+	    so we'll fake something like it
+	  */
+	  g_warning("ASSERT: gnutls_certificate_verify_peers2 assertion at %s:%d", __FILE__, __LINE__);
+	  /* and move on */
+      return ret;
+    }
+
+  return status;
+}
+/*
+  need to keep a local copy because removed from gnutls-3.x
+*/
+
 static gboolean
 verify_certificate (gnutls_session session, const char *hostname, GError **err)
 {
 	int status;
 
-	status = gnutls_certificate_verify_peers (session);
+	status = _local_gnutls_certificate_verify_peers (session);
 
 	if (status == GNUTLS_E_NO_CERTIFICATE_FOUND) {
 		g_set_error (err, SOUP_SSL_ERROR,
diff -Nurd -x'*~' libsoup-2.32.2.orig/libsoup-2.4.pc.in libsoup-2.32.2/libsoup-2.4.pc.in
--- libsoup-2.32.2.orig/libsoup-2.4.pc.in	2010-08-04 14:01:08.000000000 -0400
+++ libsoup-2.32.2/libsoup-2.4.pc.in	2012-03-26 11:20:36.000000000 -0400
@@ -7,6 +7,5 @@
 Description: a glib-based HTTP library
 Version: @VERSION@
 Requires: glib-2.0 gobject-2.0 gio-2.0
-Requires.private: libxml-2.0 @SSL_REQUIREMENT@
 Libs: -L${libdir} -lsoup-2.4
 Cflags: -I${includedir}/libsoup-2.4
