diff -Naur mtr-0.93.orig/Makefile.am mtr-0.93/Makefile.am
--- mtr-0.93.orig/Makefile.am	2020-04-21 15:07:48.000000000 -0700
+++ mtr-0.93/Makefile.am	2020-04-22 10:47:30.000000000 -0700
@@ -40,10 +40,6 @@
 dist_man_MANS = mtr.8 mtr-packet.8
 PATHFILES += man/mtr.8 man/mtr-packet.8
 
-install-exec-hook: 
-	`setcap cap_net_raw+ep $(DESTDIR)$(sbindir)/mtr-packet` \
-	|| chmod u+s $(DESTDIR)$(sbindir)/mtr-packet
-
 mtr_SOURCES = ui/mtr.c ui/mtr.h \
               ui/net.c ui/net.h \
               ui/cmdpipe.c ui/cmdpipe.h \
@@ -84,7 +80,7 @@
 mtr_SOURCES += ui/gtk.c
 endif
 
-mtr_INCLUDES = $(GLIB_CFLAGS) -I$(top_builddir) -I$(top_srcdir)
+mtr_INCLUDES = -I$(top_builddir) -I$(top_srcdir) $(GLIB_CFLAGS)
 mtr_CFLAGS = $(GTK_CFLAGS) $(NCURSES_CFLAGS)
 mtr_LDADD = $(GTK_LIBS) $(NCURSES_LIBS) $(RESOLV_LIBS)
 
diff -Naur mtr-0.93.orig/packet/probe.c mtr-0.93/packet/probe.c
--- mtr-0.93.orig/packet/probe.c	2020-04-21 15:07:48.000000000 -0700
+++ mtr-0.93/packet/probe.c	2020-04-25 13:14:24.000000000 -0700
@@ -323,7 +323,7 @@
        anything to the port.
      */
     *sockaddr_port_offset(&dest_with_port) = htons(1);
-    len = sockaddr_addr_size(&dest_with_port);
+    len = sockaddr_size(&dest_with_port);
 
     sock = socket(destaddr->ss_family, SOCK_DGRAM, IPPROTO_UDP);
     if (sock == -1) {
@@ -364,7 +364,7 @@
        Zero the port, as we may later use this address to finding, and
        we don't want to use the port from the socket we just created.
      */
-    *sockaddr_port_offset(&srcaddr) = 0;
+    *sockaddr_port_offset(srcaddr) = 0;
 
     return 0;
 }
