diff -Naur ncftp-3.2.7-orig/configure ncftp-3.2.7/configure
--- ncftp-3.2.7-orig/configure	2023-12-31 22:42:47
+++ ncftp-3.2.7/configure	2024-12-20 13:29:02
@@ -68,6 +68,7 @@
 subdirs=
 MFLAGS= MAKEFLAGS=
 SHELL=${CONFIG_SHELL-/bin/sh}
+CPPFLAGS=-DHAVE_STDLIB_H=1
 # Maximum number of lines to put in a shell here document.
 ac_max_here_lines=12
 
@@ -1235,7 +1236,7 @@
 	macos*|darwin|rhapsody)
 		OS="macosx"
 		SYS="macosx"
-		os_v=`perl -e '{open(F, "< /System/Library/CoreServices/SystemVersion.plist") or exit(2); my ($doc) = ""; my ($line); while (defined($line = <F>)) { $doc .= $line; } close(F); $doc =~ s/\s+//gs; if ($doc =~ /<key>ProductVersion<\/key><string>([^<]+)<\/string>/) { print , "\n"; exit(0); } exit(1); }' 2>/dev/null`
+		os_v=`perl -e '{open(F, "< /System/Library/CoreServices/SystemVersion.plist") or exit(2); my ($doc) = ""; my ($line); while (defined($line = <F>)) { $doc .= $line; } close(F); $doc =~ s/\s+//gs; if ($doc =~ /<key>ProductVersion<\/key><string>([^<]+)<\/string>/) { print $1, "\n"; exit(0); } exit(1); }' 2>/dev/null`
 		if [ "$os_v" = "" ] && [ -x "$HOME/bin/macosver" ] ; then
 			os_v=`"$HOME/bin/macosver" 2>/dev/null`
 		fi
@@ -6525,7 +6528,7 @@
 #	include <curses.h>
 #endif
 
- 
+int 
 main(int argc, char **argv)
 {
 	/* Note:  don't actually call curses, since it may block;
@@ -10427,7 +10430,8 @@
   else
     return (&dummy > addr) ? 1 : -1;
 }
-main ()
+
+int main ()
 {
   exit (find_stack_direction() < 0);
 }
