diff -Nurd nmap-7.93.orig/configure nmap-7.93/configure
--- nmap-7.93.orig/configure	2022-08-31 14:39:45.000000000 -0400
+++ nmap-7.93/configure	2023-01-23 12:51:07.000000000 -0500
@@ -7783,9 +7783,8 @@
 fi
 
 
-LDFLAGS="$LDFLAGS -L$NBASEDIR"
 CPPFLAGS="$CPPFLAGS -I\$(top_srcdir)/$NBASEDIR"
-LIBNBASE_LIBS="-lnbase"
+LIBNBASE_LIBS="$NBASEDIR/libnbase.a"
 
 
 
@@ -7813,9 +7812,8 @@
 fi
 
 
-LDFLAGS="$LDFLAGS -L$NSOCKDIR/src/"
 CPPFLAGS="$CPPFLAGS -I\$(top_srcdir)/$NSOCKDIR/include"
-LIBNSOCK_LIBS="-lnsock"
+LIBNSOCK_LIBS="$NSOCKDIR/src/libnsock.a"
 
 
 
diff -Nurd nmap-7.93.orig/ncat/Makefile.in nmap-7.93/ncat/Makefile.in
--- nmap-7.93.orig/ncat/Makefile.in	2018-10-16 10:34:39.000000000 -0400
+++ nmap-7.93/ncat/Makefile.in	2023-01-23 14:03:05.000000000 -0500
@@ -47,7 +47,7 @@
 NBASELIB = $(NBASEDIR)/libnbase.a
 
 CC = @CC@
-CPPFLAGS = @CPPFLAGS@
+CPPFLAGS = $(INCLS) @CPPFLAGS@
 CFLAGS = @CFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
@@ -65,7 +65,7 @@
 LUA_CFLAGS = @LUA_CFLAGS@
 
 
-CPPFLAGS += $(DEFS) $(INCLS)
+CPPFLAGS += $(DEFS)
 
 SHTOOL = ../shtool
 
@@ -87,7 +87,8 @@
 LUA_SRCS = ncat_lua.c
 LUA_OBJS = ncat_lua.o
 LUA_LIBS = @LIBLUA_LIBS@ -lm
-LUA_CFLAGS += -DHAVE_LUA=1 -I../liblua
+DEFS += -DHAVE_LUA=1
+INCS += -I../liblua
 else
 LUA_SRCS =
 LUA_OBJS =
diff -Nurd nmap-7.93.orig/nping/Makefile.in nmap-7.93/nping/Makefile.in
--- nmap-7.93.orig/nping/Makefile.in	2020-10-05 19:00:30.000000000 -0400
+++ nmap-7.93/nping/Makefile.in	2023-01-23 13:17:46.000000000 -0500
@@ -43,7 +43,7 @@
 # Should only be enabled during debugging and not in any real release.
 # DEFS += -DMTRACE=1
 CXXFLAGS = @CXXFLAGS@ $(DBGFLAGS) $(CCOPT) $(DEFS)
-CPPFLAGS = @CPPFLAGS@ -I$(NBASEDIR) -I$(NSOCKDIR)/include
+CPPFLAGS = -I$(NBASEDIR) -I$(NSOCKDIR)/include @CPPFLAGS@
 export CFLAGS = $(CXXFLAGS)
 # CFLAGS = $(DEFS) $(INCLS)
 STATIC =
diff -Nurd nmap-7.93.orig/nsock/src/Makefile.in nmap-7.93/nsock/src/Makefile.in
--- nmap-7.93.orig/nsock/src/Makefile.in	2017-08-23 04:52:40.000000000 -0400
+++ nmap-7.93/nsock/src/Makefile.in	2023-01-23 13:19:47.000000000 -0500
@@ -17,7 +17,7 @@
 INCLS = -I../include
 CFLAGS = @CFLAGS@ $(CCOPT)
 # CFLAGS = -g -Wall $(DEFS) $(INCLS)
-CPPFLAGS = @CPPFLAGS@ $(DEFS) $(INCLS)
+CPPFLAGS = $(INCLS) @CPPFLAGS@ $(DEFS)
 STATIC =
 SHTOOL = ./shtool
 INSTALL = $(SHTOOL) install
diff -Nurd nmap-7.93.orig/nsock/src/configure nmap-7.93/nsock/src/configure
--- nmap-7.93.orig/nsock/src/configure	2019-05-20 12:21:41.000000000 -0400
+++ nmap-7.93/nsock/src/configure	2023-01-23 13:38:09.000000000 -0500
@@ -5013,9 +5013,8 @@
 
 NSOCKTESTDIR="../tests"
 
-LDFLAGS="$LDFLAGS -L$NBASEDIR"
 CPPFLAGS="$CPPFLAGS -I$NBASEDIR"
-LIBNBASE_LIBS="$LIBS -lnbase"
+LIBNBASE_LIBS="$NBASEDIR/libnbase.a"
 
 
 
diff -Nurd nmap-7.93.orig/nsock/tests/Makefile.in nmap-7.93/nsock/tests/Makefile.in
--- nmap-7.93.orig/nsock/tests/Makefile.in	2016-07-31 09:57:01.000000000 -0400
+++ nmap-7.93/nsock/tests/Makefile.in	2023-01-23 13:20:03.000000000 -0500
@@ -8,7 +8,7 @@
 NBASELIB=$(NBASEDIR)/libnbase.a
 
 CC 	= @CC@
-CPPFLAGS = @CPPFLAGS@ -I../include
+CPPFLAGS = -I../include @CPPFLAGS@
 CFLAGS	= @CFLAGS@
 LDFLAGS	= @LDFLAGS@
 LIBS	= @OPENSSL_LIBS@ @LIBPCAP_LIBS@ @LIBS@
