diff -Nurd nmap-7.991.orig/configure nmap-7.991/configure
--- nmap-7.991.orig/configure	2026-08-05 17:12:13
+++ nmap-7.991/configure	2026-09-15 07:37:45
@@ -7837,9 +7837,8 @@
 fi
 
 
-LDFLAGS="$LDFLAGS -L\$(top_srcdir)/$NBASEDIR"
-CPPFLAGS="$CPPFLAGS -I\$(top_srcdir)/$NBASEDIR"
-LIBNBASE_LIBS="-lnbase"
+CPPFLAGS="-I\$(top_srcdir)/$NBASEDIR $CPPFLAGS"
+LIBNBASE_LIBS="\$(top_srcdir)/$NBASEDIR/libnbase.a"
 
 
 
@@ -7867,9 +7866,8 @@
 fi
 
 
-LDFLAGS="$LDFLAGS -L\$(top_srcdir)/$NSOCKDIR/src/"
-CPPFLAGS="$CPPFLAGS -I\$(top_srcdir)/$NSOCKDIR/include"
-LIBNSOCK_LIBS="-lnsock"
+CPPFLAGS="-I\$(top_srcdir)/$NSOCKDIR/include $CPPFLAGS"
+LIBNSOCK_LIBS="\$(top_srcdir)/$NSOCKDIR/src/libnsock.a"
 
 
 
diff -Nurd nmap-7.991.orig/ncat/Makefile.in nmap-7.991/ncat/Makefile.in
--- nmap-7.991.orig/ncat/Makefile.in	2026-08-05 17:12:13
+++ nmap-7.991/ncat/Makefile.in	2026-09-15 07:37:45
@@ -47,7 +47,7 @@
 NBASELIB = $(NBASEDIR)/libnbase.a
 
 CC = @CC@
-CPPFLAGS = @CPPFLAGS@
+CPPFLAGS = $(INCLS) @CPPFLAGS@
 CFLAGS = @CFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
@@ -65,7 +65,7 @@
 LUA_CFLAGS = @LUA_CFLAGS@
 
 
-CPPFLAGS += $(DEFS) $(INCLS)
+CPPFLAGS += $(DEFS)
 
 SHTOOL = ../shtool
 
@@ -87,7 +87,8 @@
 LUA_SRCS = ncat_lua.c
 LUA_OBJS = ncat_lua.o
 LUA_LIBS = @LIBLUA_LIBS@ -lm
-LUA_CFLAGS += -DHAVE_LUA=1 -I../liblua
+DEFS += -DHAVE_LUA=1
+INCS += -I../liblua
 else
 LUA_SRCS =
 LUA_OBJS =
diff -Nurd nmap-7.991.orig/ndiff/pyproject.toml nmap-7.991/ndiff/pyproject.toml
--- nmap-7.991.orig/ndiff/pyproject.toml	2026-08-05 17:12:13
+++ nmap-7.991/ndiff/pyproject.toml	2026-09-15 08:16:08
@@ -5,7 +5,7 @@
 [project]
 name = "ndiff"
 license = {text = "Nmap License (https://nmap.org/book/man-legal.html)"}
-dynamic = ["version"]
+version = "@VERSION@"
 authors = [{name = "The Nmap Project"}]
 description = "Nmap scan diff tool"
 readme = {file = "README", content-type = "text/plain"}
@@ -22,9 +22,6 @@
 
 [tool.setuptools]
 py-modules = ["ndiff"]
-
-[tool.setuptools.dynamic]
-version = {attr = "Version.VERSION"}
 
 #[tool.setuptools.packages.find]
 #include = ["ndiff"]
diff -Nurd nmap-7.991.orig/nping/Makefile.in nmap-7.991/nping/Makefile.in
--- nmap-7.991.orig/nping/Makefile.in	2026-08-05 17:12:13
+++ nmap-7.991/nping/Makefile.in	2026-09-15 07:37:45
@@ -43,7 +43,7 @@
 # Should only be enabled during debugging and not in any real release.
 # DEFS += -DMTRACE=1
 CXXFLAGS = @CXXFLAGS@ $(DBGFLAGS) $(CCOPT) $(DEFS)
-CPPFLAGS = @CPPFLAGS@ -I$(NBASEDIR) -I$(NSOCKDIR)/include
+CPPFLAGS = -I$(NBASEDIR) -I$(NSOCKDIR)/include @CPPFLAGS@
 export CFLAGS = $(CXXFLAGS)
 # CFLAGS = $(DEFS) $(INCLS)
 STATIC =
diff -Nurd nmap-7.991.orig/nsock/src/Makefile.in nmap-7.991/nsock/src/Makefile.in
--- nmap-7.991.orig/nsock/src/Makefile.in	2026-08-05 17:12:13
+++ nmap-7.991/nsock/src/Makefile.in	2026-09-15 07:37:45
@@ -17,7 +17,7 @@
 INCLS = -I../include
 CFLAGS = @CFLAGS@ $(CCOPT)
 # CFLAGS = -g -Wall $(DEFS) $(INCLS)
-CPPFLAGS = @CPPFLAGS@ $(DEFS) $(INCLS)
+CPPFLAGS = $(INCLS) @CPPFLAGS@ $(DEFS)
 STATIC =
 SHTOOL = ./shtool
 INSTALL = $(SHTOOL) install
diff -Nurd nmap-7.991.orig/nsock/src/configure nmap-7.991/nsock/src/configure
--- nmap-7.991.orig/nsock/src/configure	2026-08-05 17:12:13
+++ nmap-7.991/nsock/src/configure	2026-09-15 07:37:45
@@ -5289,9 +5289,8 @@
 
 NSOCKTESTDIR="../tests"
 
-LDFLAGS="$LDFLAGS -L$NBASEDIR"
-CPPFLAGS="$CPPFLAGS -I$NBASEDIR"
-LIBNBASE_LIBS="$LIBS -lnbase"
+CPPFLAGS="-I$NBASEDIR $CPPFLAGS"
+LIBNBASE_LIBS="$NBASEDIR/libnbase.a"
 
 
 
diff -Nurd nmap-7.991.orig/nsock/tests/Makefile.in nmap-7.991/nsock/tests/Makefile.in
--- nmap-7.991.orig/nsock/tests/Makefile.in	2026-08-05 17:12:13
+++ nmap-7.991/nsock/tests/Makefile.in	2026-09-15 07:37:45
@@ -8,7 +8,7 @@
 NBASELIB=$(NBASEDIR)/libnbase.a
 
 CC 	= @CC@
-CPPFLAGS = @CPPFLAGS@ -I../include
+CPPFLAGS = -I../include @CPPFLAGS@
 CFLAGS	= @CFLAGS@
 LDFLAGS	= @LDFLAGS@
 LIBS	= @OPENSSL_LIBS@ @LIBPCAP_LIBS@ @LIBS@
