diff -rauN openvpn-2.3.2/src/plugins/auth-pam/auth-pam.c openvpn-2.3.2-p1/src/plugins/auth-pam/auth-pam.c
--- openvpn.orig/src/plugins/auth-pam/auth-pam.c	2016-12-27 18:45:37.638198402 +0100
+++ openvpn/src/plugins/auth-pam/auth-pam.c	2016-12-27 18:45:37.638198402 +0100
@@ -698,7 +698,7 @@
     struct user_pass up;
     int command;
 #ifdef USE_PAM_DLOPEN
-    static const char pam_so[] = "libpam.so";
+    static const char pam_so[] = "libpam.dylib";
 #endif
 
     /*
diff -rauN openvpn-2.3.2/src/openvpn/options.c openvpn-2.3.2-p1/src/openvpn/options.c
--- openvpn.orig/src/openvpn/options.c	2017-06-22 13:17:12.806680520 +0200
+++ openvpn/src/openvpn/options.c	2017-06-22 13:17:12.802680492 +0200
@@ -197,7 +197,7 @@
     "                  is established.  Multiple routes can be specified.\n"
     "                  netmask default: 255.255.255.255\n"
     "                  gateway default: taken from --route-gateway or --ifconfig\n"
-    "                  Specify default by leaving blank or setting to \"nil\".\n"
+    "                  Specify default by leaving blank or setting to \"default\".\n"
     "--route-ipv6 network/bits [gateway] [metric] :\n"
     "                  Add IPv6 route to routing table after connection\n"
     "                  is established.  Multiple routes can be specified.\n"
