diff -Nurd opennms-1.8.13-1/source/opennms-install/src/main/java/org/opennms/install/Installer.java opennms-1.8.13-1-new/source/opennms-install/src/main/java/org/opennms/install/Installer.java
--- opennms-1.8.13-1/source/opennms-install/src/main/java/org/opennms/install/Installer.java	2011-08-10 00:09:32.000000000 -0400
+++ opennms-1.8.13-1-new/source/opennms-install/src/main/java/org/opennms/install/Installer.java	2011-08-24 15:22:24.000000000 -0400
@@ -1029,6 +1029,7 @@
 
         if (!System.getProperty("os.name").contains("Windows")) {
             String[] defaults = {
+                    "@FINKPREFIX@/lib",
                     "/usr/lib/jni",
                     "/usr/lib",
                     "/usr/local/lib",
