diff -Nurd pvpgn-1.8.5.orig/src/Makefile.in pvpgn-1.8.5/src/Makefile.in
--- pvpgn-1.8.5.orig/src/Makefile.in	2009-02-25 17:48:47.000000000 -0500
+++ pvpgn-1.8.5/src/Makefile.in	2021-07-15 03:32:56.000000000 -0400
@@ -775,9 +775,9 @@
 .c.o:
 	@${RMF} $@
 	@echo "Compiling $*.c"
-	@@COMPILEANSI1@
-	@@COMPILEANSI2@
-	@@COMPILEANSI3@
+	@COMPILEANSI1@
+	@COMPILEANSI2@
+	@COMPILEANSI3@
 
 .dep:
 	${MAKE} dep
@@ -786,7 +786,7 @@
 	@${RMF} .dep
 	@for a in ${PARTS}; do \
 	    echo Dependency check: $$a; \
-	    ${COMPILE} -MM `ls $$a/*.c` | sed -e "s@\(.*\)\.o:\(.*\)@$$a\/\1\.o:\2@g" 1>>.dep; \
+	    ${COMPILE} -MD `ls $$a/*.c` | sed -e "s@\(.*\)\.o:\(.*\)@$$a\/\1\.o:\2@g" 1>>.dep; \
 	done
 	
 ${bnetd_SBIN}: SHOWHEADER ${bnetd_OBJECTS}
diff -Nurd pvpgn-1.8.5.orig/src/bnetd/handle_irc.c pvpgn-1.8.5/src/bnetd/handle_irc.c
--- pvpgn-1.8.5.orig/src/bnetd/handle_irc.c	2009-02-25 17:48:47.000000000 -0500
+++ pvpgn-1.8.5/src/bnetd/handle_irc.c	2021-07-15 02:30:56.000000000 -0400
@@ -20,6 +20,7 @@
  
 #define CONNECTION_INTERNAL_ACCESS
 #include "common/setup_before.h"
+#include "common/trans.h"
 #include "common/util.h"
 #ifdef STDC_HEADERS
 # include <stdlib.h>
