diff -Naur rsync-3.1.3/acls.c rsync-3.1.3.new/acls.c
--- rsync-3.1.3/acls.c	2018-01-14 22:55:07.000000000 -0500
+++ rsync-3.1.3.new/acls.c	2020-11-12 22:43:36.000000000 -0500
@@ -278,7 +278,7 @@
 		id_access *ida;
 		if ((rc = sys_acl_get_info(entry, &tag_type, &access, &g_u_id)) != 0) {
 			errfun = "sys_acl_get_info";
-			break;
+			rc = 0;
 		}
 		/* continue == done with entry; break == store in temporary ida list */
 		switch (tag_type) {
diff -Naur rsync-3.1.3/configure.sh rsync-3.1.3.new/configure.sh
--- rsync-3.1.3/configure.sh	2018-01-28 18:51:49.000000000 -0500
+++ rsync-3.1.3.new/configure.sh	2020-11-12 22:44:41.000000000 -0500
@@ -8406,6 +8406,7 @@
 /* end confdefs.h.  */
 #include <sys/time.h>
 #include <unistd.h>
+#include <stdlib.h>
 int
 main ()
 {
--- rsync-3.1.3/patches/crtimes.diff	2018-07-31 16:27:50.000000000 +0200
+++ rsync-3.1.3.new/patches/crtimes.diff	2018-07-31 16:28:51.000000000 +0200
@@ -182,7 +182,7 @@
 +		if (crtimes_ndx) {
 +			if (sxp->crtime == 0)
 +				sxp->crtime = get_create_time(fnamecmp);
-+			if (cmp_time(sxp->crtime, f_crtime(file)) != 0)
++			if (cmp_time(sxp->crtime, 0, f_crtime(file), 0) != 0)
 +				iflags |= ITEM_REPORT_CRTIME;
 +		}
  #if !defined HAVE_LCHMOD && !defined HAVE_SETATTRLIST
@@ -339,7 +339,7 @@
 +		time_t file_crtime = f_crtime(file);
 +		if (sxp->crtime == 0)
 +			sxp->crtime = get_create_time(fname);
-+		if (cmp_time(sxp->crtime, file_crtime) != 0
++		if (cmp_time(sxp->crtime, 0, file_crtime, 0) != 0
 +		 && set_create_time(fname, file_crtime) == 0)
 +			updated = 1;
 +	}
