diff -Nurd -x'*~' slrn-1.0.3.orig/configure slrn-1.0.3/configure
--- slrn-1.0.3.orig/configure	2016-10-23 18:34:16.000000000 -0400
+++ slrn-1.0.3/configure	2021-06-10 06:44:19.000000000 -0400
@@ -7380,7 +7380,7 @@
    elif test "x$use_gnutls" != xno
    then
      SSLINC=
-     SSLLIB="-lgnutls-openssl"
+     SSLLIB="-lgnutls -lgnutls-openssl"
    else
      SSLINC="-I/usr/local/ssl/include"
      SSLLIB="-L/usr/local/ssl/lib -lssl -lcrypto"
@@ -8306,6 +8306,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+	#include <stdlib.h>
 	#include <stdarg.h>
 	void f (int i, ...) {
 	va_list args1, args2;
@@ -8351,6 +8352,7 @@
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+	#include <stdlib.h>
 	#include <stdarg.h>
 	void f (int i, ...) {
 	va_list args1, args2;
