diff -Nurd -x'*~' slrn-pre1.0.3-10.orig/configure slrn-pre1.0.3-10/configure
--- slrn-pre1.0.3-10.orig/configure	2015-12-15 03:55:06.000000000 -0500
+++ slrn-pre1.0.3-10/configure	2016-07-27 23:00:22.000000000 -0400
@@ -7380,7 +7380,7 @@
    elif test "x$use_gnutls" != xno
    then
      SSLINC=
-     SSLLIB="-lgnutls-openssl"
+     SSLLIB="-lgnutls -lgnutls-openssl"
    else
      SSLINC="-I/usr/local/ssl/include"
      SSLLIB="-L/usr/local/ssl/lib -lssl -lcrypto"
diff -Nurd -x'*~' slrn-pre1.0.3-10.orig/src/post.c slrn-pre1.0.3-10/src/post.c
--- slrn-pre1.0.3-10.orig/src/post.c	2015-08-15 03:55:09.000000000 -0400
+++ slrn-pre1.0.3-10/src/post.c	2016-07-28 01:41:53.000000000 -0400
@@ -219,7 +219,7 @@
 		  "Date: %s, %d %s %d %02d:%02d:%02d %+03d%02d",
 		  Weekdays[t->tm_wday], t->tm_mday, Months[t->tm_mon],
 		  t->tm_year + 1900, t->tm_hour, t->tm_min, t->tm_sec,
-		  (int) tz / 60, (int) abs (tz) % 60);
+		  (int) tz / 60, (int) labs (tz) % 60);
 }
 /*}}}*/
 
diff -Nurd -x'*~' slrn-pre1.0.3-10.orig/src/sltcp.c slrn-pre1.0.3-10/src/sltcp.c
--- slrn-pre1.0.3-10.orig/src/sltcp.c	2015-06-02 10:13:36.000000000 -0400
+++ slrn-pre1.0.3-10/src/sltcp.c	2016-07-28 01:07:30.000000000 -0400
@@ -575,7 +575,21 @@
 	  }
 
 	/* SSLv3 is vulnerable to POODLE attacks.  Do not use it. */
+#if SLTCP_HAS_GNUTLS_SUPPORT
+	/* gnutls has never supported SSLv2 */
+#ifdef SSL_OP_NO_SSLv3
+#error "Unexpected gnutls feature: SSL_OP_NO_SSLv3"
+	/* Token not available and no obvious workaround as of
+	   gnutls-3.3.18. Because this feature (if it were available)
+	   would alter the build, we abort here to make sure everyone
+	   gets consistent build results. Eventually we would switch
+	   over to whatever method gnutls has for this feature and set
+	   dependencies to make sure we have that support. */
+	SSL_CTX_set_options (c, SSL_OP_NO_SSLv3);
+#endif
+#else
 	SSL_CTX_set_options (c, SSL_OP_NO_SSLv2|SSL_OP_NO_SSLv3);
+#endif
 	This_SSL_Ctx = c;
 	atexit (deinit_ssl);
 
