diff -Nurd tcpflow-1.6.1.orig/configure tcpflow-1.6.1/configure
--- tcpflow-1.6.1.orig/configure	2021-02-18 21:53:02.000000000 -0500
+++ tcpflow-1.6.1/configure	2021-07-10 09:17:38.000000000 -0400
@@ -5908,37 +5908,6 @@
 
 
 
-if test x"${mingw}" != "xyes" ; then
-
-  case $host in
-    *mingw*)
-       { $as_echo "$as_me:${as_lineno-$LINENO}: Compiling under mingw; will not search other directories." >&5
-$as_echo "$as_me: Compiling under mingw; will not search other directories." >&6;}
-       ;;
-    *)
-       { $as_echo "$as_me:${as_lineno-$LINENO}: Compiling under $host." >&5
-$as_echo "$as_me: Compiling under $host." >&6;}
-       # Bring additional directories where things might be found into our
-       # search path. I don't know why autoconf doesn't do this by default
-       for spfx in /usr/local /opt/local /sw /usr/local/ssl; do
-         { $as_echo "$as_me:${as_lineno-$LINENO}: checking ${spfx}/include" >&5
-$as_echo "$as_me: checking ${spfx}/include" >&6;}
-         if test -d ${spfx}/include; then
-           CPPFLAGS="$CPPFLAGS -I${spfx}/include"
-           LDFLAGS="$LDFLAGS -L${spfx}/lib"
-           { $as_echo "$as_me:${as_lineno-$LINENO}:  *** ADDING ${spfx}/include to CPPFLAGS *** " >&5
-$as_echo "$as_me:  *** ADDING ${spfx}/include to CPPFLAGS *** " >&6;}
-           { $as_echo "$as_me:${as_lineno-$LINENO}:  *** ADDING ${spfx}/lib to LDFLAGS *** " >&5
-$as_echo "$as_me:  *** ADDING ${spfx}/lib to LDFLAGS *** " >&6;}
-         fi
-       done
-       { $as_echo "$as_me:${as_lineno-$LINENO}:  CPPFLAGS = ${CPPFLAGS} " >&5
-$as_echo "$as_me:  CPPFLAGS = ${CPPFLAGS} " >&6;}
-       { $as_echo "$as_me:${as_lineno-$LINENO}:  LDFLAGS = ${LDFLAGS} " >&5
-$as_echo "$as_me:  LDFLAGS = ${LDFLAGS} " >&6;}
-       ;;
-  esac
-fi
 
 
 
@@ -7151,26 +7120,6 @@
      ;;
 esac
 
-if test x"${mingw}" == "xno" ; then
-  # Bring additional directories where things might be found into our
-  # search path. I don't know why autoconf doesn't do this by default
-  for spfx in /usr/local /opt/local /sw /usr/local/ssl /usr/boost/include ; do
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking ${spfx}/include" >&5
-$as_echo "$as_me: checking ${spfx}/include" >&6;}
-    if test -d ${spfx}/include; then
-        CPPFLAGS="-I${spfx}/include $CPPFLAGS"
-        LDFLAGS="-L${spfx}/lib $LDFLAGS"
-        { $as_echo "$as_me:${as_lineno-$LINENO}:  *** ADDING ${spfx}/include to CPPFLAGS *** " >&5
-$as_echo "$as_me:  *** ADDING ${spfx}/include to CPPFLAGS *** " >&6;}
-        { $as_echo "$as_me:${as_lineno-$LINENO}:  *** ADDING ${spfx}/lib to LDFLAGS *** " >&5
-$as_echo "$as_me:  *** ADDING ${spfx}/lib to LDFLAGS *** " >&6;}
-    fi
-  done
-  { $as_echo "$as_me:${as_lineno-$LINENO}:  CPPFLAGS = ${CPPFLAGS} " >&5
-$as_echo "$as_me:  CPPFLAGS = ${CPPFLAGS} " >&6;}
-  { $as_echo "$as_me:${as_lineno-$LINENO}:  LDFLAGS = ${LDFLAGS} " >&5
-$as_echo "$as_me:  LDFLAGS = ${LDFLAGS} " >&6;}
-fi
 
 if test -r /bin/uname.exe ; then
   if test "`uname -o`" == "Msys" ; then
diff -Nurd tcpflow-1.6.1.orig/src/datalink.cpp tcpflow-1.6.1/src/datalink.cpp
--- tcpflow-1.6.1.orig/src/datalink.cpp	2021-02-18 19:46:09.000000000 -0500
+++ tcpflow-1.6.1/src/datalink.cpp	2023-04-05 03:36:45.000000000 -0400
@@ -171,10 +171,8 @@
             break;
         }
     } catch( std::logic_error e){
-        std::string s(std::string("warning: caught std::logic_error ")
-                      + e.what()
-                      + std::string(" in packet"));
-        DEBUG(6)(s.c_str());
+        DEBUG(6)("warning: caught std::logic_error %s in packet",
+                 e.what());
     }
 }
 
