diff -ruN whois-5.5.13-orig/Makefile whois-5.5.13/Makefile
--- whois-5.5.13-orig/Makefile	2022-04-07 19:09:39.000000000 -0400
+++ whois-5.5.13/Makefile	2022-08-04 07:18:57.000000000 -0400
@@ -66,7 +66,7 @@
 DEFS += -DHAVE_CRYPT_H -DHAVE_LINUX_CRYPT_GENSALT -D_OW_SOURCE
 mkpasswd_LDADD += -lcrypt -lowcrypt
 else
-mkpasswd_LDADD += -lcrypt
+mkpasswd_LDADD += 
 endif
 
 CPPFLAGS += $(DEFS) $(INCLUDES)
diff -ruN whois-5.5.13-orig/mkpasswd.c whois-5.5.13/mkpasswd.c
--- whois-5.5.13-orig/mkpasswd.c	2022-04-07 19:09:39.000000000 -0400
+++ whois-5.5.13/mkpasswd.c	2022-08-04 07:18:37.000000000 -0400
@@ -17,7 +17,6 @@
  */
 
 /* for crypt, snprintf and strcasecmp */
-#define _XOPEN_SOURCE 500
 #define _BSD_SOURCE 1
 #define _DEFAULT_SOURCE 1
 #define __EXTENSIONS__ 1
diff -ruN whois-5.5.13-orig/utils.c whois-5.5.13/utils.c
--- whois-5.5.13-orig/utils.c	2022-04-07 19:09:39.000000000 -0400
+++ whois-5.5.13/utils.c	2022-08-04 07:18:30.000000000 -0400
@@ -21,7 +21,7 @@
  */
 
 /* for strdup */
-#define _XOPEN_SOURCE 500
+#define _XOPEN_SOURCE 600
 
 /* System library */
 #include <stdio.h>
