--- chemtool-1.4.1/Makefile	Thu Nov 15 20:53:23 2001
+++ chemtool-1.4.1-patched/Makefile	Thu Nov 15 22:02:30 2001
@@ -1,5 +1,5 @@
-PREFIX=/usr/local
-LOCALEDIR=/usr/share/locale
+PREFIX=@PREFIX1@
+LOCALEDIR=@PREFIX2@/locale
 
 ifdef LOCALEDIR
 CFLAGS=-O2 -Wall `gtk-config --cflags` -DMENU -DLOCALEDIR=\"${LOCALEDIR}\" 
@@ -7,7 +7,7 @@
 CFLAGS=-O2 -Wall `gtk-config --cflags` -DMENU -DDISABLE_NLS
 endif
 
-CC=gcc
+CC=cc
 CP=/bin/cp
 RM=/bin/rm -f
 MAKE=make
@@ -31,11 +31,11 @@
 
 install: chemtool src-cht/cht
 	${CP} chemtool src-cht/cht ${PREFIX}/bin
-	${CP} chemtool.1 ${PREFIX}/man/man1
-	${CP} cht.1	${PREFIX}/man/man1
-	ifdef LOCALEDIR
-	for L in `ls locales`; do ${CP} locales/$$L/chemtool.mo ${LOCALEDIR}/$$L/LC_MESSAGES;done
-	endif 
+	${CP} chemtool.1 ${PREFIX}/share/man/man1
+	${CP} cht.1	${PREFIX}/share/man/man1
+ifdef LOCALEDIR
+	for L in `ls locales`; do mkdir -p  @PREFIX1@/share/locale/$$L/LC_MESSAGES; ${CP} locales/$$L/chemtool.mo @PREFIX1@/share/locale/$$L/LC_MESSAGES;done
+endif 
 clean:
 	${RM} *.o *~ src-cht/*.o src-cht/*~ chemtool src-cht/cht
 
--- chemtool-1.4.1/main.c	Fri Sep 28 20:28:49 2001
+++ chemtool-1.4.1-patched/main.c	Thu Nov 15 20:54:37 2001
@@ -11,6 +11,7 @@
  *      This software comes with ABSOLUTELY NO WARRANTY
  */
 
+#include <ctype.h>
 #include "ct.h"
 #include "bitmap1.h"
 #include "templates.h"
@@ -19,7 +20,7 @@
 #include <stdio.h>
 #include <strings.h>
 #include <string.h>
-#include <ctype.h>
+/*#include <ctype.h>*/
 #include <locale.h>
 #ifdef DISABLE_NLS
 #define _(Text) Text
--- chemtool-1.4.1/src-cht/Makefile	Sat Sep  8 22:56:19 2001
+++ chemtool-1.4.1-patched/src-cht/Makefile	Thu Nov 15 20:54:58 2001
@@ -1,4 +1,4 @@
 default: cht
 
 cht:	cht-1.8.c
-	gcc -O cht-1.8.c -o cht
+	cc -O cht-1.8.c -o cht
