diff -uNr clustalw1.81/clustalw.c ross_clustalw1.81/clustalw.c
--- clustalw1.81/clustalw.c	Wed Jun  7 03:50:05 2000
+++ ross_clustalw1.81/clustalw.c	Tue May  8 10:43:28 2001
@@ -33,7 +33,7 @@
 #ifdef MSDOS
         char *help_file_name = "clustalw.hlp";
 #else
-        char *help_file_name = "clustalw_help";
+        char *help_file_name = LIBDIR "/clustalw_help";
 #endif
 
 sint max_names; /* maximum length of names in current alignment file */
diff -uNr clustalw1.81/makefile ross_clustalw1.81/makefile
--- clustalw1.81/makefile	Wed Jun  7 03:55:06 2000
+++ ross_clustalw1.81/makefile	Tue May  8 10:44:13 2001
@@ -1,17 +1,21 @@
-install: clustalw
-
-clean:
-	rm *.o
+#
+# Where to put things
+#
+prefix = /sw
+bindir = $(prefix)/bin
+libdir = $(prefix)/lib/clustalw
+#
 
 OBJECTS = interface.o sequence.o showpair.o malign.o \
-  	util.o trees.o gcgcheck.o prfalign.o pairalign.o \
-  	calcgapcoeff.o calcprf1.o calcprf2.o calctree.o \
-        readmat.o alnscore.o random.o 
+	util.o trees.o gcgcheck.o prfalign.o pairalign.o \
+	calcgapcoeff.o calcprf1.o calcprf2.o calctree.o \
+	readmat.o alnscore.o random.o 
 
 HEADERS = general.h clustalw.h
 
+INSTALL = /usr/bin/install -o root -g admin
 CC	= cc
-CFLAGS  = -c -O
+CFLAGS  = -c -O -DLIBDIR=\"${libdir}\"
 LFLAGS	= -O -lm 
 
 clustalw : $(OBJECTS) amenu.o clustalw.o
@@ -31,4 +35,17 @@
 
 .c.o :
 	$(CC) $(CFLAGS) $?
+
+install: clustalw
+	$(INSTALL) -d -m 755 $(bindir)
+	$(INSTALL) -d -m 755 $(libdir)
+	$(INSTALL) -c -m 755 clustalw $(bindir)
+	$(INSTALL) -c -m 644 clustalw_help $(libdir)
+
+clean:
+	rm *.o clustalw
+
+
+
+
