--- fftw-3.0/tests/Makefile.in.orig	Mon Apr 21 10:18:05 2003
+++ fftw-3.0/tests/Makefile.in	Mon Apr 21 10:18:37 2003
@@ -174,9 +174,9 @@
 @THREADS_TRUE@LIBFFTWTHREADS = $(top_builddir)/threads/libfftw3@PREC_SUFFIX@_threads.la
 
 bench_SOURCES = bench.c hook.c
-bench_LDADD = $(LIBFFTWTHREADS)				\
+bench_LDADD = $(top_builddir)/libbench2/libbench2.a $(LIBFFTWTHREADS)				\
 $(top_builddir)/libfftw3@PREC_SUFFIX@.la		\
-$(top_builddir)/libbench2/libbench2.a $(THREADLIBS)
+$(THREADLIBS)
 
 subdir = tests
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
--- fftw-3.0/tools/Makefile.in.orig	Mon Apr 21 10:18:44 2003
+++ fftw-3.0/tools/Makefile.in	Mon Apr 21 10:19:05 2003
@@ -177,9 +177,9 @@
 @THREADS_TRUE@LIBFFTWTHREADS = $(top_builddir)/threads/libfftw3@PREC_SUFFIX@_threads.la
 
 fftw@PREC_SUFFIX@_wisdom_SOURCES = fftw-wisdom.c
-fftw@PREC_SUFFIX@_wisdom_LDADD = $(top_builddir)/tests/bench.o	\
+fftw@PREC_SUFFIX@_wisdom_LDADD = $(top_builddir)/tests/bench.o	$(top_builddir)/libbench2/libbench2.a \
 $(LIBFFTWTHREADS) $(top_builddir)/libfftw3@PREC_SUFFIX@.la	\
-$(top_builddir)/libbench2/libbench2.a $(THREADLIBS)
+$(THREADLIBS)
 
 subdir = tools
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
