--- gdis-0.76.1/Makefile	2002-04-26 11:16:55.000000000 +0800
+++ gdis-0.76.1-patched/Makefile	2002-05-16 12:01:29.000000000 +0800
@@ -2,17 +2,18 @@
 
 # --- edit as required
 CC = gcc
-CFLAGS = -O2
-INSTALL = /usr/local/bin
+CFLAGS = -no-cpp-precomp -O2
+INSTALL1 = @PREFIX1@/bin
+INSTALL2 = @PREFIX2@/bin
 USE_OPENGL = YES
 USE_GPROF = NO
 USE_EFENCE = NO
 
 # --- the rest should be ok
 NAME = gdis-0.77.3
-INCS = `gtk-config --cflags`
+INCS = `gtk-config --cflags` -I@PREFIX1@/include
 LIBS = `gtk-config --libs`
-DEFS = -DINSTALL=\""$(INSTALL)"\"
+DEFS = -DINSTALL=\""$(INSTALL1)"\"
 
 ifeq ($(USE_OPENGL), YES)
 LIBS := $(LIBS) -lGLU -lGL -lglut -lgtkgl
@@ -51,8 +52,8 @@
 .PHONY: install
 install:
 	/usr/bin/strip gdis
-	/bin/cp gdis $(INSTALL)
-	/bin/cp gdis.elements $(INSTALL)
+	/bin/cp gdis $(INSTALL2)
+	/bin/cp gdis.elements $(INSTALL2)
 
 .PHONY: tar
 tar:
