--- GMT3.4.1/configure.in.orig	Fri Jun  7 15:52:21 2002
+++ GMT3.4.1/configure.in	Fri Jun  7 15:55:36 2002
@@ -270,6 +270,7 @@
 dnl
 LDFLAGS=$OPTS
 LD_OPT=
+LD=ld
 SL="so"
 if test "$enable_shared" = "yes" ; then
 	AC_MSG_CHECKING(how to make a shared object)
@@ -301,13 +302,15 @@
 		LD_OPT="-shared"
 		LDFLAGS='-Wl,-x'
 	elif test "$os" = "Darwin" ; then	# MacOS X
-		LD_OPT="-shared"
+		LD_OPT="-dynamiclib -flat_namespace -undefined suppress -install_name ${prefix}/lib/\$@"
 		LDFLAGS='-Wl,-x'
+               LD=cc
+               SL=dylib
 	elif test "$os" = "HP-UX" ; then
 		LD_OPT="-b"
 		LDFLAGS=$OPTS'-Wl,+b,$(libdir)'
 	fi
-	AC_MSG_RESULT(ld $LD_OPT)
+	AC_MSG_RESULT($LD $LD_OPT)
 	AC_MSG_CHECKING(the file extension for a shared object)
 	AC_MSG_RESULT($SL)
 fi
@@ -325,7 +328,7 @@
 			CC_OPT=$OPTGO'-ansi -pedantic -traditional-cpp'
 			LDFLAGS=
 		elif test "$os" = "Darwin" ; then	# MacOS X
-			CC_OPT=$OPTGO'-ansi -pedantic -traditional-cpp'
+			CC_OPT=$OPTGO'-fno-common -ansi -pedantic -traditional-cpp'
 			LDFLAGS=
 		else
 			CC_OPT=$OPTGO'-ansi -pedantic'
