--- PGPLOT-2.18/Makefile.PL.orig	Sat Jul 14 06:31:09 2001
+++ PGPLOT-2.18/Makefile.PL	Sat Jul 14 06:32:21 2001
@@ -32,13 +32,13 @@
         'LIBS'	=> ['-L/usr/openwin/lib -L/usr/X11/lib -L/usr/X11R6/lib -lX11 '.
                     ExtUtils::F77->runtime])
        :
-         ('LIBS' => [$ldir.' -L/usr/local/pgplot -L/usr/local/lib -lcpgplot -lpgplot '.
+         ('LIBS' => [$ldir.' -L$PGPLOT_DIR -L<PREFIX>/lib -lcpgplot -lpgplot -lpng '.
                     '-L/usr/openwin/lib -L/usr/X11/lib -L/usr/X11R6/lib -lX11 '.
                     ExtUtils::F77->runtime])
      )),
     'DEFINE'	=>  ExtUtils::F77->trail_ ? "" : "-DNO_TRAILING_USCORE",
     ($^O eq 'VMS' ? ('INC' => 'pgplot_dir:') : (
-    'INC'	=> $idir.' -I/home/aaossm/kgb/soft/pgplot/bld -I/usr/local/pgplot'  # where cpgplot.h should be
+    'INC'	=> $idir.' -I<PREFIX>/pgplot'  # where cpgplot.h should be
     )),
     'dist'      => { COMPRESS=>"gzip", SUFFIX=>"gz" },
     'depend'    => { '$(OBJECT)' => q[pgfun.c arrays.c PGPLOT.c]}
