--- R-1.5.1/config.cache.orig	Fri Jan  4 09:30:21 2002
+++ R-1.5.1/config.cache	Fri Jan  4 09:28:27 2002
@@ -0,0 +1,6 @@
+ac_cv_path_DVIPS=${ac_cv_path_DVIPS=no}
+ac_cv_path_TEX=${ac_cv_path_LATEX=no}
+ac_cv_path_LATEX=${ac_cv_path_LATEX=no}
+ac_cv_path_MAKEINDEX=${ac_cv_path_MAKEINDEX=no}
+ac_cv_path_PDFLATEX=${ac_cv_path_PDFLATEX=no}
+ac_cv_path_PDFTEX=${ac_cv_path_PDFTEX=no}
--- R-1.5.1/src/library/profile/Rprofile.unix.orig	Sun Mar  3 06:13:37 2002
+++ R-1.5.1/src/library/profile/Rprofile.unix	Sun Mar  3 06:13:53 2002
@@ -15,7 +15,7 @@
 ## this must be set for x11 to be used, even non-interactively
 options(X11colortype = "true")
 options(pager = file.path(R.home(), "bin", "pager"))
-options(browser = "netscape")
+options(browser = "open @PREFIX@/lib/R/doc/html/index.html")
 options(mailer = "mailx")
 
 MacRoman <- as.integer(c(0:126, 32, 196, 197, 199, 201, 209, 214, 220,
--- R-1.5.1/configure.orig	Thu Mar  7 17:37:52 2002
+++ R-1.5.1/configure	Thu Mar  7 17:40:59 2002
@@ -11636,7 +11636,7 @@
       darwin1.3)
         shlib_ldflags="-bundle -flat_namespace -undefined suppress" ;;
       *)
-	shlib_ldflags="-bundle -bundle_loader \$(R_HOME)/bin/R.bin" ;;
+	shlib_ldflags="-bundle -flat_namespace -undefined suppress" ;;
     esac
     shlib_cxxldflags="${shlib_ldflags}"
     ;;
@@ -11823,7 +11823,7 @@
 case "${host_os}" in
   darwin*)
     LIBR_EXT=".dylib"
-    LIBR_LDFLAGS="-dynamiclib"
+    LIBR_LDFLAGS="-dynamiclib -flat_namespace -undefined suppress -install_name ${prefix}/lib/R/bin/libR.dylib"
     ;;
 esac
 
--- R-1.5.1/etc/Makeconf.in.orig	Mon Feb  4 11:38:50 2002
+++ R-1.5.1/etc/Makeconf.in	Mon Feb  4 11:39:08 2002
@@ -26,6 +26,7 @@
 LIBS = @LIBS@
 LIBPATHS = @LIBPATHS@
 LIBTOOL = $(SHELL) $(R_HOME)/bin/libtool
+LDFLAGS = @LDFLAGS@
 SHELL = @SHELL@
 SHLIB_CFLAGS = @SHLIB_CFLAGS@
 SHLIB_CXXFLAGS = @SHLIB_CXXFLAGS@
