diff -Naur root07.orig/Makefile root07.new/Makefile
--- root07.orig/Makefile	Sun Jul 27 03:45:57 2003
+++ root07.new/Makefile	Mon Jul 28 19:07:57 2003
@@ -477,6 +477,8 @@
 	      for lib in $(ALLLIBS) $(CINTLIB); do \
 		 rm -f $(DESTDIR)$(LIBDIR)/`basename $$lib` ; \
 		 rm -f $(DESTDIR)$(LIBDIR)/`basename $$lib`.$$vers ; \
+	         rm -f $(DESTDIR)$(LIBDIR)/`basename $$lib $(SOEXT)`$$major.$(SOEXT) ; \
+	         rm -f $(DESTDIR)$(LIBDIR)/`basename $$lib $(SOEXT)`$$major.$$minor.$(SOEXT) ; \
 		 bindll=`echo $$lib | sed 's,lib,bin,'`; \
 		 baselib=`basename $$lib`; \
 		 $(INSTALL) $$bindll $(DESTDIR)$(BINDIR); \
diff -Naur root07.orig/build/unix/compiledata.sh root07.new/build/unix/compiledata.sh
--- root07.orig/build/unix/compiledata.sh	Mon Jun 23 12:51:13 2003
+++ root07.new/build/unix/compiledata.sh	Mon Jul 28 19:10:18 2003
@@ -61,7 +61,7 @@
 echo "#define BUILD_NODE \""`uname -a`"\" " >> __compiledata
 echo "#define COMPILER \""`type $CXX`"\" " >> __compiledata
 if [ "$CUSTOMSHARED" = "" ]; then
-      echo "#define MAKESHAREDLIB  \"cd \$BuildDir ; $CXX -c \$Opt $CXXFLAGS \$IncludePath \$SourceFiles ; $CXX \$ObjectFiles $SOFLAGS $LDFLAGS $EXPLLINKLIBS -o \$SharedLib \`[ -d /sw/lib ] && echo -L/sw/lib\`\"" >> __compiledata
+      echo "#define MAKESHAREDLIB  \"cd \$BuildDir ; $CXX -c \$Opt $CXXFLAGS \$IncludePath \$SourceFiles ; $CXX \$ObjectFiles $SOFLAGS $LDFLAGS $EXPLLINKLIBS -o \$SharedLib\"" >> __compiledata
 else
    echo "#define MAKESHAREDLIB \"$CUSTOMSHARED\"" >> __compiledata
 fi
diff -Naur root07.orig/build/unix/makelib.sh root07.new/build/unix/makelib.sh
--- root07.orig/build/unix/makelib.sh	Sat Apr 26 00:51:51 2003
+++ root07.new/build/unix/makelib.sh	Mon Jul 28 19:04:08 2003
@@ -31,6 +31,26 @@
 
 rm -f $LIB
 
+if [ $PLATFORM = "macosx" ]; then
+   # Look for a fink installation
+   FINKDIR=`which fink 2>&1 | sed -ne "s/\/bin\/fink//p"`
+   # We need beside the shared library (.dylib) a bundle (.so) to load
+   LDFLAGS="`[ -d ${FINKDIR}/lib ] && echo -L${FINKDIR}/lib` -ldl"
+   BUNDLE=`echo $LIB | sed s/.dylib/.so/`
+   if [ "x`echo $SOFLAGS | grep -- '-g'`" != "x" ]; then
+      opt=-g
+   else
+      opt=-O
+   fi
+   echo $LD $opt -bundle -flat_namespace -undefined suppress -o $BUNDLE \
+	$OBJS $LDFLAGS $EXTRA
+   $LD $opt -bundle -flat_namespace -undefined suppress -o $BUNDLE \
+	$OBJS $LDFLAGS $EXTRA
+   if [ "x$MAJOR" != "x" ]; then
+       EXTRA="${EXTRA} -compatibility_version ${MAJOR} -current_version ${MAJOR}.${MINOR}.${REVIS}"
+   fi
+fi
+
 if [ $PLATFORM = "aix" ]; then
    makeshared="/usr/ibmcxx/bin/makeC++SharedLib"
 fi
@@ -74,22 +94,6 @@
    $LD $SOFLAGS $LDFLAGS -o $LIB `lorder $OBJS | tsort -q` $EXTRA $EXPLLNKCORE
    # for elf:  echo $PLATFORM: $LD $SOFLAGS$SONAME $LDFLAGS -o $LIB $OBJS
    # for elf:  $LD $SOFLAGS$SONAME $LDFLAGS -o $LIB `lorder $OBJS | tsort -q`
-elif [ $PLATFORM = "macosx" ]; then
-   # We need two library files: a .dylib to link to and a .so to load
-   BUNDLE=`echo $LIB | sed s/.dylib/.so/`
-   echo $LD $SOFLAGS$SONAME -o $LIB $OBJS \
-	`[ -d /sw/lib ] && echo -L/sw/lib` -ldl $EXTRA
-   $LD $SOFLAGS$SONAME -o $LIB $OBJS \
-	`[ -d /sw/lib ] && echo -L/sw/lib` -ldl $EXTRA
-   if [ "x`echo $SOFLAGS | grep -- '-g'`" != "x" ]; then
-      opt=-g
-   else
-      opt=-O
-   fi
-   echo $LD $opt -bundle -flat_namespace -undefined suppress -o $BUNDLE \
-	$OBJS `[ -d /sw/lib ] && echo -L/sw/lib` -ldl $EXTRA
-   $LD $opt -bundle -flat_namespace -undefined suppress -o $BUNDLE \
-	$OBJS `[ -d /sw/lib ] && echo -L/sw/lib` -ldl $EXTRA
 elif [ $LD = "KCC" ]; then
    echo $LD $LDFLAGS -o $LIB $OBJS $EXTRA $EXPLLNKCORE
    $LD $LDFLAGS -o $LIB $OBJS $EXTRA $EXPLLNKCORE
@@ -107,14 +111,13 @@
    echo $line
    $line
 else
-   if [ "x$MAJOR" = "x" ] ; then
-      echo $LD $SOFLAGS$SONAME $LDFLAGS -o $LIB $OBJS $EXTRA $EXPLLNKCORE
-      $LD $SOFLAGS$SONAME $LDFLAGS -o $LIB $OBJS $EXTRA $EXPLLNKCORE
-   else
-      echo $LD $SOFLAGS$SONAME.$MAJOR.$MINOR $LDFLAGS -o $LIB.$MAJOR.$MINOR $OBJS $EXTRA $EXPLLNKCORE
-      $LD $SOFLAGS$SONAME.$MAJOR.$MINOR $LDFLAGS \
-         -o $LIB.$MAJOR.$MINOR $OBJS $EXTRA $EXPLLNKCORE
+   if [ "x$MAJOR" != "x" ] ; then
+     # add versioning information into shlibs name
+     SONAME=`echo $SONAME | sed "s/.*\./&${MAJOR}./"`
+     LIB=`echo $LIB | sed "s/\/*.*\/.*\./&${MAJOR}.${MINOR}./"`
    fi
+   echo $LD $SOFLAGS$SONAME $LDFLAGS -o $LIB $OBJS $EXTRA
+   $LD $SOFLAGS$SONAME $LDFLAGS -o $LIB $OBJS $EXTRA
 fi
 
 linkstat=$?
@@ -122,9 +125,9 @@
    exit $linkstat
 fi
 
-if [ "x$MAJOR" != "x" ] && [ -f $LIB.$MAJOR.$MINOR ]; then
-   ln -fs $SONAME.$MAJOR.$MINOR $LIB.$MAJOR
-   ln -fs $SONAME.$MAJOR        $LIB
+if [ "x$MAJOR" != "x" ] && [ -f $LIB ]; then
+   ln -fs `echo $SONAME | sed "s/.*\./&${MINOR}./"` `echo $LIB | sed "s/\.${MINOR}//"`
+   ln -fs $SONAME `echo $LIB | sed "s/\.${MAJOR}\.${MINOR}//"`
 fi
 
 if [ $PLATFORM = "hpux" ]; then
diff -Naur root07.orig/config/Makefile.macosx root07.new/config/Makefile.macosx
--- root07.orig/config/Makefile.macosx	Sat Apr 26 00:51:51 2003
+++ root07.new/config/Makefile.macosx	Mon Jul 28 19:04:08 2003
@@ -57,6 +57,7 @@
 INSTALLDATA  := cp -pPR
 
 # Extras
-FINK_CXXFLAGS = $(shell [ -d /sw/include ] && echo -I/sw/include)
-FINK_CFLAGS   = $(shell [ -d /sw/include ] && echo -I/sw/include)
-FINK_LDFLAGS  = $(shell [ -d /sw/lib ] && echo -L/sw/lib)
+FINK_DIR      = $(shell which fink 2>&1 | sed -ne "s/\/bin\/fink//p")
+FINK_CXXFLAGS = $(shell [ -d $(FINK_DIR)/include ] && echo -I$(FINK_DIR)/include)
+FINK_CFLAGS   = $(shell [ -d $(FINK_DIR)/include ] && echo -I$(FINK_DIR)/include)
+FINK_LDFLAGS  = $(shell [ -d $(FINK_DIR)/lib ] && echo -L$(FINK_DIR)/lib)
diff -Naur root07.orig/config/root-config.in root07.new/config/root-config.in
--- root07.orig/config/root-config.in	Tue Jun 17 07:37:07 2003
+++ root07.new/config/root-config.in	Mon Jul 28 19:27:07 2003
@@ -257,9 +257,10 @@
    auxglibs=${auxlibs}
    ;;
 macosx)
-   # MacOS X with cc (GNU cc v2.95.2.)
-   auxcflags=
-   auxlibs="-u _G__cpp_setup_initializerG__Tree -lm `[ -d /sw/lib ] && echo -L/sw/lib` -ldl"
+   # MacOS X with gcc (GNU cc v3.1) and possible fink (fink.sf.net)
+   finkdir=`which fink 2>&1 | sed -ne "s/\/bin\/fink//p"`
+   auxcflags=`[ -d ${finkdir}/include ] && echo -I${finkdir}/include`
+   auxlibs="-u _G__cpp_setup_initializerG__Tree -lm `[ -d ${finkdir}/lib ] && echo -L${finkdir}/lib` -ldl"
    auxglibs=${auxlibs}
    ;;
 hiux)
