diff -Naur spice3f5sfix/conf/fink spice3f5sfix.new/conf/fink
--- spice3f5sfix/conf/fink	Thu Jan  1 09:00:00 1970
+++ spice3f5sfix.new/conf/fink	Mon Dec 17 13:23:53 2001
@@ -0,0 +1,30 @@
+# Fink 
+
+FINK_DIR        = @FINK_PREFIX@
+FINK_INSTALL_DIR= @FINK_INSTALL@
+DEFAULT_EDITOR  = pico
+ASCII_RAWFILE	= 0
+INTERFACE_OPTS	= -DWANT_X11
+MAKE            = /usr/bin/gnumake
+X_DIR           = /usr/X11R6
+INCX            = -I$(X_DIR)/include -I$(X_DIR)/include/X11        \
+                  -I$(X_DIR)/include/X11/Xmu                       \
+                  -I$(X_DIR)/include/X11/Xaw
+LIBX            = -L$(X_DIR)/lib -lXaw -lXt -lXext -lXmu -lSM -lICE -lX11
+
+VERSION         = 3f5sfix
+LDFLAGS	        =  
+CC              = /usr/bin/gcc3
+
+CC_OPT	        = -no-cpp-precomp -Os
+
+BUG_ADDR        = peter@pogma.com
+SPICE_DIR       = $(FINK_INSTALL_DIR)
+SPICE_LIB_DIR   = $(SPICE_DIR)/lib/spice
+SPICE_EXEC_DIR  = $(SPICE_DIR)/bin
+
+S_SPICE_DIR     = $(FINK_DIR)
+S_SPICE_LIB_DIR	= $(S_SPICE_DIR)/lib/spice
+S_SPICE_EXEC_DIR= $(S_SPICE_DIR)/bin
+
+INSTALL_DIRS    = $(SPICE_DIR) $(SPICE_EXEC_DIR) $(SPICE_LIB_DIR)
diff -Naur spice3f5sfix/src/include/misc.h spice3f5sfix.new/src/include/misc.h
--- spice3f5sfix/src/include/misc.h	Thu Dec  1 07:17:02 1994
+++ spice3f5sfix.new/src/include/misc.h	Sun Dec 16 18:20:55 2001
@@ -86,7 +86,9 @@
 extern char *realloc();
 extern char *getenv();
 extern int errno;
+#if 0
 extern char *sys_errlist[];
+#endif
 extern char *getenv();
 extern char *getwd();
 extern int rand();
diff -Naur spice3f5sfix/src/include/port.h spice3f5sfix.new/src/include/port.h
--- spice3f5sfix/src/include/port.h	Thu Dec  1 06:04:36 1994
+++ spice3f5sfix.new/src/include/port.h	Sun Dec 16 18:08:12 2001
@@ -51,6 +51,11 @@
 #  define CONFIGURED
 #endif
 
+#ifdef __APPLE__
+#  include "os_bsd.h"
+#  define CONFIGURED
+#endif
+
 #ifdef NeXT
 #  include "os_bsd.h"
 #  define CONFIGURED
