diff -ru xcircuit-3.1.13/configure xcircuit-3.1.13-patched/configure
--- xcircuit-3.1.13/configure	2003-03-07 17:30:20.000000000 +0100
+++ xcircuit-3.1.13-patched/configure	2003-05-03 19:21:01.000000000 +0200
@@ -2593,7 +2593,7 @@
 
 CPPFLAGS="${CPPFLAGS} ${X_CFLAGS}"
 LDFLAGS="${LDFLAGS} ${X_LIBS}"
-LIBS="${X_PRE_LIBS} ${LIBS}"
+LIBS="${X_PRE_LIBS} -lX11 ${LIBS}"
 echo $ac_n "checking for XtToolkitInitialize in -lXt""... $ac_c" 1>&6
 echo "configure:2599: checking for XtToolkitInitialize in -lXt" >&5
 ac_lib_var=`echo Xt'_'XtToolkitInitialize | sed 'y%./+-%__p_%'`
@@ -3483,7 +3483,7 @@
      *-hpux*)
 	SHDLIB_EXT=".sl"
 	;;
-     *darwin*)
+     *-darwin*)
 	SHDLIB_EXT=".dylib"
 	;;
      *)
@@ -3678,7 +3678,7 @@
       EXTRA_LIB_SPECS="-ldl"
       ;;
 
-    darwin*)
+    *-darwin*)
       SHDLIB_EXT=".dylib"
       LDDL_FLAGS="-dynamiclib -bind_at_load"
       LDFLAGS=${LDFLAGS} ${LIB_SPEC}
diff -ru xcircuit-3.1.13/elements.c xcircuit-3.1.13-patched/elements.c
--- xcircuit-3.1.13/elements.c	2003-02-06 16:49:59.000000000 +0100
+++ xcircuit-3.1.13-patched/elements.c	2003-05-03 18:32:25.000000000 +0200
@@ -10,11 +10,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 
 #include <X11/Intrinsic.h>
 #include <X11/StringDefs.h>
diff -ru xcircuit-3.1.13/events.c xcircuit-3.1.13-patched/events.c
--- xcircuit-3.1.13/events.c	2003-03-06 22:43:16.000000000 +0100
+++ xcircuit-3.1.13-patched/events.c	2003-05-03 18:32:23.000000000 +0200
@@ -10,11 +10,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 #include <math.h>
 #include <ctype.h>
 
diff -ru xcircuit-3.1.13/filelist.c xcircuit-3.1.13-patched/filelist.c
--- xcircuit-3.1.13/filelist.c	2003-02-01 20:59:33.000000000 +0100
+++ xcircuit-3.1.13-patched/filelist.c	2003-05-03 18:32:21.000000000 +0200
@@ -6,11 +6,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 #include <sys/types.h>
 
 #ifdef HAVE_DIRENT_H
diff -ru xcircuit-3.1.13/files.c xcircuit-3.1.13-patched/files.c
--- xcircuit-3.1.13/files.c	2003-04-29 20:32:50.000000000 +0200
+++ xcircuit-3.1.13-patched/files.c	2003-05-03 18:32:18.000000000 +0200
@@ -6,11 +6,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 #include <time.h>
 #include <pwd.h>
 #include <ctype.h>
diff -ru xcircuit-3.1.13/fontfile.c xcircuit-3.1.13-patched/fontfile.c
--- xcircuit-3.1.13/fontfile.c	2003-01-31 04:59:48.000000000 +0100
+++ xcircuit-3.1.13-patched/fontfile.c	2003-05-03 18:32:15.000000000 +0200
@@ -7,11 +7,6 @@
 #include <stdlib.h>
 #include <ctype.h>
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 
 #include <X11/Intrinsic.h>
 #include <X11/StringDefs.h>
diff -ru xcircuit-3.1.13/formats.c xcircuit-3.1.13-patched/formats.c
--- xcircuit-3.1.13/formats.c	2002-05-23 18:00:13.000000000 +0200
+++ xcircuit-3.1.13-patched/formats.c	2003-05-03 18:32:12.000000000 +0200
@@ -6,11 +6,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 #include <ctype.h>
 #include <X11/Intrinsic.h>
 #include <X11/StringDefs.h>
diff -ru xcircuit-3.1.13/functions.c xcircuit-3.1.13-patched/functions.c
--- xcircuit-3.1.13/functions.c	2003-02-28 16:54:19.000000000 +0100
+++ xcircuit-3.1.13-patched/functions.c	2003-05-03 18:32:10.000000000 +0200
@@ -10,11 +10,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 #include <math.h>
 
 #include <X11/Intrinsic.h>
diff -ru xcircuit-3.1.13/keybindings.c xcircuit-3.1.13-patched/keybindings.c
--- xcircuit-3.1.13/keybindings.c	2003-03-05 22:50:20.000000000 +0100
+++ xcircuit-3.1.13-patched/keybindings.c	2003-05-03 18:32:08.000000000 +0200
@@ -11,11 +11,6 @@
 #include <stdlib.h>
 #include <string.h>
 #include <ctype.h>	 /* for tolower(), toupper() */
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 #include <math.h>
 
 #include <X11/Intrinsic.h>
diff -ru xcircuit-3.1.13/libraries.c xcircuit-3.1.13-patched/libraries.c
--- xcircuit-3.1.13/libraries.c	2003-04-29 20:10:54.000000000 +0200
+++ xcircuit-3.1.13-patched/libraries.c	2003-05-03 18:32:06.000000000 +0200
@@ -10,11 +10,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 
 #include <X11/Intrinsic.h>
 #include <X11/StringDefs.h>
diff -ru xcircuit-3.1.13/menucalls.c xcircuit-3.1.13-patched/menucalls.c
--- xcircuit-3.1.13/menucalls.c	2003-03-07 18:30:42.000000000 +0100
+++ xcircuit-3.1.13-patched/menucalls.c	2003-05-03 18:32:04.000000000 +0200
@@ -8,11 +8,6 @@
 #include <stdlib.h>
 #include <string.h>
 #include <ctype.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 #include <sys/types.h>
 #include <errno.h>
 #include <limits.h>
diff -ru xcircuit-3.1.13/netlist.c xcircuit-3.1.13-patched/netlist.c
--- xcircuit-3.1.13/netlist.c	2003-04-08 20:24:58.000000000 +0200
+++ xcircuit-3.1.13-patched/netlist.c	2003-05-03 18:32:01.000000000 +0200
@@ -12,11 +12,6 @@
 #include <stdlib.h>
 #include <string.h>
 #include <ctype.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 
 #ifdef HAVE_PYTHON
 #include <Python.h>
diff -ru xcircuit-3.1.13/parameter.c xcircuit-3.1.13-patched/parameter.c
--- xcircuit-3.1.13/parameter.c	2003-02-28 17:04:07.000000000 +0100
+++ xcircuit-3.1.13-patched/parameter.c	2003-05-03 18:31:59.000000000 +0200
@@ -11,11 +11,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 
 #include <X11/Intrinsic.h>
 #include <X11/StringDefs.h>
diff -ru xcircuit-3.1.13/python.c xcircuit-3.1.13-patched/python.c
--- xcircuit-3.1.13/python.c	2003-01-15 21:06:30.000000000 +0100
+++ xcircuit-3.1.13-patched/python.c	2003-05-03 18:31:56.000000000 +0200
@@ -17,11 +17,6 @@
 #include <stdlib.h>
 #include <unistd.h>	/* for usleep() */
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
    
 #include <Python.h>
 
diff -ru xcircuit-3.1.13/rcfile.c xcircuit-3.1.13-patched/rcfile.c
--- xcircuit-3.1.13/rcfile.c	2003-01-15 21:15:43.000000000 +0100
+++ xcircuit-3.1.13-patched/rcfile.c	2003-05-03 18:31:55.000000000 +0200
@@ -13,11 +13,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 
 #include <X11/Intrinsic.h>
 #include <X11/StringDefs.h>
diff -ru xcircuit-3.1.13/render.c xcircuit-3.1.13-patched/render.c
--- xcircuit-3.1.13/render.c	2003-04-14 21:46:05.000000000 +0200
+++ xcircuit-3.1.13-patched/render.c	2003-05-03 18:31:50.000000000 +0200
@@ -10,11 +10,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 #include <unistd.h>
 #include <time.h>
 #include <sys/wait.h>   /* for waitpid() */
diff -ru xcircuit-3.1.13/schema.c xcircuit-3.1.13-patched/schema.c
--- xcircuit-3.1.13/schema.c	2003-04-29 17:58:33.000000000 +0200
+++ xcircuit-3.1.13-patched/schema.c	2003-05-03 18:31:48.000000000 +0200
@@ -10,11 +10,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 
 #include <X11/Intrinsic.h>
 #include <X11/StringDefs.h>
diff -ru xcircuit-3.1.13/selection.c xcircuit-3.1.13-patched/selection.c
--- xcircuit-3.1.13/selection.c	2003-02-28 14:21:15.000000000 +0100
+++ xcircuit-3.1.13-patched/selection.c	2003-05-03 18:31:45.000000000 +0200
@@ -10,11 +10,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 #include <math.h>
 
 #include <X11/Intrinsic.h>
diff -ru xcircuit-3.1.13/tclxcircuit.c xcircuit-3.1.13-patched/tclxcircuit.c
--- xcircuit-3.1.13/tclxcircuit.c	2003-04-29 20:38:30.000000000 +0200
+++ xcircuit-3.1.13-patched/tclxcircuit.c	2003-05-03 18:31:42.000000000 +0200
@@ -11,11 +11,6 @@
 #include <stdlib.h>
 #include <unistd.h>     /* for usleep() */
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <errno.h>
diff -ru xcircuit-3.1.13/xcircuit.c xcircuit-3.1.13-patched/xcircuit.c
--- xcircuit-3.1.13/xcircuit.c	2003-04-02 18:10:44.000000000 +0200
+++ xcircuit-3.1.13-patched/xcircuit.c	2003-05-03 18:31:40.000000000 +0200
@@ -29,11 +29,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if defined(DARWIN)
-#include <sys/malloc.h>
-#else
-#include <malloc.h>
-#endif
 #include <signal.h>
 #include <sys/types.h>
 #include <sys/stat.h>
