diff -ru xcircuit-3.1.15/configure xcircuit-3.1.15-patched/configure
--- xcircuit-3.1.15/configure	2003-05-19 17:54:41.000000000 +0200
+++ xcircuit-3.1.15-patched/configure	2003-05-19 23:14:43.000000000 +0200
@@ -2593,7 +2593,7 @@
 
 CPPFLAGS="${CPPFLAGS} ${X_CFLAGS}"
 LDFLAGS="${LDFLAGS} ${X_LIBS}"
-LIBS="${X_PRE_LIBS} ${LIBS}"
+LIBS="${X_PRE_LIBS} -lX11 ${LIBS}"
 echo $ac_n "checking for XtToolkitInitialize in -lXt""... $ac_c" 1>&6
 echo "configure:2599: checking for XtToolkitInitialize in -lXt" >&5
 ac_lib_var=`echo Xt'_'XtToolkitInitialize | sed 'y%./+-%__p_%'`
