--- xmakemol-5.06/file.c	Fri May 10 22:42:30 2002
+++ xmakemol-5.06-patched/file.c	Mon Oct 14 10:45:32 2002
@@ -29,6 +29,7 @@
 #include <alloca.h>
 #endif
 #include <ctype.h> /* aro - needed for isalnum() */
+#include <sys/types.h>
 #include <dirent.h>
 #include <locale.h>
 #include <math.h>

--- xmakemol-5.06/xmake_anim.pl	Mon May 27 23:44:30 2002
+++ xmakemol-5.06-patched/xmake_anim.pl	Mon Oct 14 10:52:35 2002
@@ -27,13 +27,13 @@
 
 # Configure: specify where the binaries are
 
-$convert = "/usr/bin/convert";
+$convert = "@prefix@/bin/convert";
 if (! -e $convert)
   {
     die ("Cannot find convert program (part of ImageMagick). Please install it.\n");
   }
 
-$gifsicle = "/usr/bin/gifsicle";
+$gifsicle = "@prefix@/bin/gifsicle";
 if (! -e $gifsicle)
   {
     print ("Cannot find gifsicle program, install it for better performance.\n");

--- xmakemol-5.06/configure	Wed Sep 11 00:47:19 2002
+++ xmakemol-5.06-patched/configure	Thu Nov  7 08:30:02 2002
@@ -3332,9 +3332,9 @@
   
 	XWIDGETLIBS="$X_LIBS $X11_LIBS $XT_LIBS $X_LIBS $X_PRE_LIBS $XM_LIBS"
   	sim_cv_motif_glwidget=UNKNOWN
-  	for lib in GLwM GLw MesaGLwM MesaGLw; do
+  	for lib in GLw; do
     	  if test x"$sim_cv_motif_glwidget" = x"UNKNOWN"; then
-      		LIBS="-l$lib $SAVELIBS $XWIDGETLIBS $GL_LIBS -lm"
+      		LIBS="-l$lib -lXm  -lSM -lICE -lXext -lXt -lXext -lX11 -lgl"
       		cat > conftest.$ac_ext <<EOF
 #line 3340 "configure"
 #include "confdefs.h"
