diff -ruN xmltv-0.5.5.orig/Makefile.PL xmltv-0.5.5/Makefile.PL
--- xmltv-0.5.5.orig/Makefile.PL	2002-12-13 16:11:15.000000000 -0700
+++ xmltv-0.5.5/Makefile.PL	2002-12-17 14:35:04.000000000 -0700
@@ -146,22 +146,7 @@
 # spun off into separate projects.
 #
 my @opt_components
-  = ({ name => 'tv_check',
-       blurb =>
-       'Tv_check is a graphical front-end to listings data.  You pick the
-shows you are interested in, and then tv_check alerts you to
-unexpected extra episodes or schedule changes.',
-       exes => [ 'choose/tv_check/tv_check' ],
-       docs => [ qw(choose/tv_check/README.tv_check
-		    choose/tv_check/tv_check_doc.html
-		    choose/tv_check/tv_check_doc.jpg
-		   ) ],
-       prereqs => { 'Tk' => 0,
-		    'Tk::TableMatrix' => 0,
-		    'XML::Simple' => 0,
-		  } },
-
-     { name => 'tv_pick_cgi',
+  = ({ name => 'tv_pick_cgi',
        blurb =>
        "Tv_pick_cgi is a front-end to choose your TV viewing.  It presents all
 programmes in the listings and asks you about each one.  But you can
@@ -174,7 +159,7 @@
 are present.",
        prereqs => { 'CGI' => 0 },
        type => 'run',
-       default => 0,
+       #default => 0,
      },
 
      { name => 'tv_grab_na',
@@ -263,12 +248,12 @@
 	$msg = "Use $name?";
     }
 
-    if (XMLTV::Ask::askBooleanQuestion($msg, $default)) {
+    #if (XMLTV::Ask::askBooleanQuestion($msg, $default)) {
 	push @exes, @{$info->{exes}}               if $info->{exes};
 	push @docs, @{$info->{docs}}               if $info->{docs};
 	%pm = (%pm, %{$info->{pm}})                if $info->{pm};
 	%prereqs = (%prereqs, %{$info->{prereqs}}) if $info->{prereqs};
-    }
+    #}
     print STDERR "\n";
 }
 
