diff -ruN gramps-1.0.3/configure gramps-1.0.3-patched/configure
--- gramps-1.0.3/configure	Thu Apr 22 07:18:39 2004
+++ gramps-1.0.3-patched/configure	Sun Apr 25 14:15:51 2004
@@ -3615,7 +3615,7 @@
 $PYTHON conftest.py
 has_pygtk=`cat conftest.out`
 rm -f conftest.out conftest.py
-if test YES != "$has_pygtk"
+if 0
 then
    { { echo "$as_me:$LINENO: error:
 
@@ -3753,7 +3753,7 @@
 $PYTHON conftest.py
 has_pygtk=`cat conftest.out`
 rm -f conftest.out conftest.py
-if test YES != "$has_pygtk"
+if 0
 then
    { { echo "$as_me:$LINENO: error:
 
diff -ruN gramps-1.0.3/src/Makefile.in gramps-1.0.3-patched/src/Makefile.in
--- gramps-1.0.3/src/Makefile.in	Thu Apr 22 07:18:37 2004
+++ gramps-1.0.3-patched/src/Makefile.in	Sun Apr 25 14:14:35 2004
@@ -107,7 +107,7 @@
 CCDEPMODE = @CCDEPMODE@
 
 # For intl. support, how do we compile?
-CFLAGS = -fPIC -shared -O @GNOMEINC@ @CFLAGS@ @CPPFLAGS@ -I@includedir@
+CFLAGS = -bundle -undefined dynamic_lookup -O @GNOMEINC@ @CFLAGS@ @CPPFLAGS@ -I@includedir@
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CYGPATH_W = @CYGPATH_W@
diff -ruN gramps-1.0.3/src/po/Makefile.in gramps-1.0.3-patched/src/po/Makefile.in
--- gramps-1.0.3/src/po/Makefile.in	Thu Apr 22 07:18:38 2004
+++ gramps-1.0.3-patched/src/po/Makefile.in	Sun Apr 25 14:15:20 2004
@@ -244,9 +243,7 @@
 	done
 
 .po.mo:
-	$(MSGCONV) --to-code=UTF-8 $< -o temp.po
-	$(MSGFMT) temp.po -o $@
-	rm temp.po
+	$(MSGFMT) $< -o $@
 
 zh_CN.mo: $(top_srcdir)/src/po/zh_CN.po
 	$(ICONV) -f GB2312 -t UTF-8 $< > temp.po
