diff -U 3 MySQL-python-0.9.2/setup.py MySQL-python-0.9.2-p/setup.py
--- MySQL-python-0.9.2/setup.py
+++ MySQL-python-0.9.2-p/setup.py
@@ -11,7 +11,7 @@
 NO = 0
 
 # set this to YES if you have the thread-safe mysqlclient library
-thread_safe_library = YES
+thread_safe_library = NO
 
 # You probably don't have to do anything past this point. If you
 # do, please mail me the configuration for your platform. Don't
