diff -ru pysqlite/setup.py pysqlite-new/setup.py
--- pysqlite/setup.py	Wed Oct  6 04:53:22 2004
+++ pysqlite-new/setup.py	Fri Oct 29 11:21:15 2004
@@ -32,6 +32,12 @@
     libraries = ["sqlite"]
     runtime_library_dirs = []
     extra_objects = []
+elif sys.platform == "darwin":
+    include_dirs = ['@PREFIX@/include/']
+    library_dirs = ['@PREFIX@/lib/']
+    libraries = [sqlite]
+    runtime_library_dirs = []
+    extra_objects = []
 elif os.name == "posix": # most Unixish platforms
     include_dirs = ['/usr/local/include']
     library_dirs = ['/usr/local/lib']
