diff -ru cyrus-sasl-2.1.20.orig/Makefile.in cyrus-sasl-2.1.20/Makefile.in
--- cyrus-sasl-2.1.20.orig/Makefile.in	2004-10-24 16:21:09.000000000 -0400
+++ cyrus-sasl-2.1.20/Makefile.in	2004-12-09 00:25:28.000000000 -0500
@@ -235,7 +235,7 @@
 @JAVA_TRUE@JAV = java
 @JAVA_FALSE@JAV = 
 @MACOSX_FALSE@INSTALLOSX = 
-@MACOSX_TRUE@INSTALLOSX = install-exec-local-osx
+#@MACOSX_TRUE@INSTALLOSX = install-exec-local-osx
 
 SUBDIRS = include sasldb plugins lib utils doc man $(PWC) $(SAM) $(JAV) $(SAD)
 EXTRA_DIST = config cmulocal win32 mac dlcompat-20010505 NTMakefile
@@ -673,13 +673,7 @@
 	  echo "* the library should use."; \
 	  echo "********************************************************"; \
 	fi
-install-exec-local-osx:
-	$(mkinstalldirs) $(framedir)/Versions/A/Headers
-	$(mkinstalldirs) $(framedir)/Versions/A/Resources
-	cd $(framedir)/Versions ; ln -fs A Current
-	cd $(framedir) ; ln -fs Versions/A/Headers .
-	cd $(framedir) ; ln -fs Versions/A/Resources .
-	$(INSTALL_DATA) $(srcdir)/config/Info.plist $(framedir)/Versions/A/Resources
+
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -ru cyrus-sasl-2.1.20.orig/configure cyrus-sasl-2.1.20/configure
--- cyrus-sasl-2.1.20.orig/configure	2004-10-24 16:21:08.000000000 -0400
+++ cyrus-sasl-2.1.20/configure	2004-12-09 09:22:58.000000000 -0500
@@ -5222,7 +5222,7 @@
 
 	CPPFLAGS="${CPPFLAGS} ${BDB_INCADD}"
 	SASL_DB_INC=$BDB_INCADD
-	SASL_DB_LIB="${BDB_LIBADD}"
+	SASL_DB_LIB="-ldb"
 	;;
   gdbm)
 
@@ -6456,7 +6456,7 @@
 	else
 	  	  CPPFLAGS="${CPPFLAGS} ${BDB_INCADD}"
 	  SASL_DB_INC=$BDB_INCADD
-          SASL_DB_LIB="${BDB_LIBADD}"
+          SASL_DB_LIB="-ldb"
 	fi
 	;;
   none)
@@ -6915,8 +6915,8 @@
 if test $ac_cv_lib_pam_pam_start = yes; then
 
 	  if test "${ac_cv_header_security_pam_appl_h+set}" = set; then
-  echo "$as_me:$LINENO: checking for security/pam_appl.h" >&5
-echo $ECHO_N "checking for security/pam_appl.h... $ECHO_C" >&6
+  echo "$as_me:$LINENO: checking for pam/pam_appl.h" >&5
+echo $ECHO_N "checking for pam/pam_appl.h... $ECHO_C" >&6
 if test "${ac_cv_header_security_pam_appl_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 fi
@@ -6924,8 +6924,8 @@
 echo "${ECHO_T}$ac_cv_header_security_pam_appl_h" >&6
 else
   # Is the header compilable?
-echo "$as_me:$LINENO: checking security/pam_appl.h usability" >&5
-echo $ECHO_N "checking security/pam_appl.h usability... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking pam/pam_appl.h usability" >&5
+echo $ECHO_N "checking pam/pam_appl.h usability... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -6934,7 +6934,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <security/pam_appl.h>
+#include <pam/pam_appl.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
@@ -6960,8 +6960,8 @@
 echo "${ECHO_T}$ac_header_compiler" >&6
 
 # Is the header present?
-echo "$as_me:$LINENO: checking security/pam_appl.h presence" >&5
-echo $ECHO_N "checking security/pam_appl.h presence... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking pam/pam_appl.h presence" >&5
+echo $ECHO_N "checking pam/pam_appl.h presence... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -6969,7 +6969,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <security/pam_appl.h>
+#include <pam/pam_appl.h>
 _ACEOF
 if { (eval echo "$as_me:$LINENO: \"$ac_cpp conftest.$ac_ext\"") >&5
   (eval $ac_cpp conftest.$ac_ext) 2>conftest.er1
@@ -7002,10 +7002,10 @@
 # So?  What about this header?
 case $ac_header_compiler:$ac_header_preproc in
   yes:no )
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&5
-echo "$as_me: WARNING: security/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&2;}
     (
       cat <<\_ASBOX
 ## ------------------------------------ ##
@@ -7016,12 +7016,12 @@
       sed "s/^/$as_me: WARNING:     /" >&2
     ;;
   no:yes )
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: present but cannot be compiled" >&5
-echo "$as_me: WARNING: security/pam_appl.h: present but cannot be compiled" >&2;}
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: check for missing prerequisite headers?" >&5
-echo "$as_me: WARNING: security/pam_appl.h: check for missing prerequisite headers?" >&2;}
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: present but cannot be compiled" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: present but cannot be compiled" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: check for missing prerequisite headers?" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: check for missing prerequisite headers?" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&2;}
     (
       cat <<\_ASBOX
 ## ------------------------------------ ##
@@ -7032,8 +7032,8 @@
       sed "s/^/$as_me: WARNING:     /" >&2
     ;;
 esac
-echo "$as_me:$LINENO: checking for security/pam_appl.h" >&5
-echo $ECHO_N "checking for security/pam_appl.h... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking for pam/pam_appl.h" >&5
+echo $ECHO_N "checking for pam/pam_appl.h... $ECHO_C" >&6
 if test "${ac_cv_header_security_pam_appl_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
diff -ru cyrus-sasl-2.1.20.orig/include/Makefile.in cyrus-sasl-2.1.20/include/Makefile.in
--- cyrus-sasl-2.1.20.orig/include/Makefile.in	2004-10-24 16:21:09.000000000 -0400
+++ cyrus-sasl-2.1.20/include/Makefile.in	2004-12-09 00:25:28.000000000 -0500
@@ -238,9 +238,9 @@
 EXTRA_DIST = NTMakefile
 DISTCLEANFILES = md5global.h
 
-@MACOSX_TRUE@framedir = /Library/Frameworks/SASL2.framework
-@MACOSX_TRUE@frameheaderdir = $(framedir)/Versions/A/Headers
-@MACOSX_TRUE@frameheader_DATA = $(saslinclude_HEADERS)
+#@MACOSX_TRUE@framedir = /Library/Frameworks/SASL2.framework
+#@MACOSX_TRUE@frameheaderdir = $(framedir)/Versions/A/Headers
+#@MACOSX_TRUE@frameheader_DATA = $(saslinclude_HEADERS)
 subdir = include
 mkinstalldirs = $(SHELL) $(top_srcdir)/config/mkinstalldirs
 CONFIG_HEADER = $(top_builddir)/config.h
diff -ru cyrus-sasl-2.1.20.orig/lib/Makefile.in cyrus-sasl-2.1.20/lib/Makefile.in
--- cyrus-sasl-2.1.20.orig/lib/Makefile.in	2004-10-24 16:21:09.000000000 -0400
+++ cyrus-sasl-2.1.20/lib/Makefile.in	2004-12-09 00:25:28.000000000 -0500
@@ -245,7 +245,7 @@
 libsasl2_la_DEPENDENCIES = $(LTLIBOBJS)
 libsasl2_la_LIBADD = $(LTLIBOBJS) $(SASL_DL_LIB) $(LIB_SOCKET) $(LIB_DOOR)
 
-@MACOSX_TRUE@framedir = /Library/Frameworks/SASL2.framework
+#@MACOSX_TRUE@framedir = /Library/Frameworks/SASL2.framework
 subdir = lib
 mkinstalldirs = $(SHELL) $(top_srcdir)/config/mkinstalldirs
 CONFIG_HEADER = $(top_builddir)/config.h
@@ -617,10 +617,10 @@
 	uninstall-libLTLIBRARIES
 
 @MACOSX_TRUE@install-exec-hook:
-@MACOSX_TRUE@	$(mkinstalldirs) $(framedir)/Versions/A
-@MACOSX_TRUE@	ln -fs $(libdir)/libsasl2.dylib $(framedir)/Versions/A/SASL2
-@MACOSX_TRUE@	cd $(framedir) ; ln -fs Versions/A/SASL2 .
-@MACOSX_FALSE@install-exec-hook:
+#@MACOSX_TRUE@	$(mkinstalldirs) $(framedir)/Versions/A
+#@MACOSX_TRUE@	ln -fs $(libdir)/libsasl2.dylib $(framedir)/Versions/A/SASL2
+#@MACOSX_TRUE@	cd $(framedir) ; ln -fs Versions/A/SASL2 .
+#@MACOSX_FALSE@install-exec-hook:
 
 libsasl2.a: libsasl2.la $(SASL_STATIC_OBJS)
 	@echo adding static plugins and dependencies
diff -ru cyrus-sasl-2.1.20.orig/saslauthd/auth_pam.c cyrus-sasl-2.1.20/saslauthd/auth_pam.c
--- cyrus-sasl-2.1.20.orig/saslauthd/auth_pam.c	2003-05-31 13:00:24.000000000 -0400
+++ cyrus-sasl-2.1.20/saslauthd/auth_pam.c	2004-12-09 00:25:28.000000000 -0500
@@ -50,7 +50,7 @@
 
 # include <string.h>
 # include <syslog.h>
-# include <security/pam_appl.h>
+# include <pam/pam_appl.h>
 
 # include "auth_pam.h"
 /* END PUBLIC DEPENDENCIES */
diff -ru cyrus-sasl-2.1.20.orig/saslauthd/configure cyrus-sasl-2.1.20/saslauthd/configure
--- cyrus-sasl-2.1.20.orig/saslauthd/configure	2004-10-24 16:21:13.000000000 -0400
+++ cyrus-sasl-2.1.20/saslauthd/configure	2004-12-09 00:25:28.000000000 -0500
@@ -8986,8 +8986,8 @@
 if test $ac_cv_lib_pam_pam_start = yes; then
 
 	  if test "${ac_cv_header_security_pam_appl_h+set}" = set; then
-  echo "$as_me:$LINENO: checking for security/pam_appl.h" >&5
-echo $ECHO_N "checking for security/pam_appl.h... $ECHO_C" >&6
+  echo "$as_me:$LINENO: checking for pam/pam_appl.h" >&5
+echo $ECHO_N "checking for pam/pam_appl.h... $ECHO_C" >&6
 if test "${ac_cv_header_security_pam_appl_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 fi
@@ -8995,8 +8995,8 @@
 echo "${ECHO_T}$ac_cv_header_security_pam_appl_h" >&6
 else
   # Is the header compilable?
-echo "$as_me:$LINENO: checking security/pam_appl.h usability" >&5
-echo $ECHO_N "checking security/pam_appl.h usability... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking pam/pam_appl.h usability" >&5
+echo $ECHO_N "checking pam/pam_appl.h usability... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -9005,7 +9005,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <security/pam_appl.h>
+#include <pam/pam_appl.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
@@ -9031,8 +9031,8 @@
 echo "${ECHO_T}$ac_header_compiler" >&6
 
 # Is the header present?
-echo "$as_me:$LINENO: checking security/pam_appl.h presence" >&5
-echo $ECHO_N "checking security/pam_appl.h presence... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking pam/pam_appl.h presence" >&5
+echo $ECHO_N "checking pam/pam_appl.h presence... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -9040,7 +9040,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <security/pam_appl.h>
+#include <pam/pam_appl.h>
 _ACEOF
 if { (eval echo "$as_me:$LINENO: \"$ac_cpp conftest.$ac_ext\"") >&5
   (eval $ac_cpp conftest.$ac_ext) 2>conftest.er1
@@ -9073,10 +9073,10 @@
 # So?  What about this header?
 case $ac_header_compiler:$ac_header_preproc in
   yes:no )
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&5
-echo "$as_me: WARNING: security/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&2;}
     (
       cat <<\_ASBOX
 ## ------------------------------------ ##
@@ -9087,12 +9087,12 @@
       sed "s/^/$as_me: WARNING:     /" >&2
     ;;
   no:yes )
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: present but cannot be compiled" >&5
-echo "$as_me: WARNING: security/pam_appl.h: present but cannot be compiled" >&2;}
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: check for missing prerequisite headers?" >&5
-echo "$as_me: WARNING: security/pam_appl.h: check for missing prerequisite headers?" >&2;}
-    { echo "$as_me:$LINENO: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: security/pam_appl.h: proceeding with the preprocessor's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: present but cannot be compiled" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: present but cannot be compiled" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: check for missing prerequisite headers?" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: check for missing prerequisite headers?" >&2;}
+    { echo "$as_me:$LINENO: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: pam/pam_appl.h: proceeding with the preprocessor's result" >&2;}
     (
       cat <<\_ASBOX
 ## ------------------------------------ ##
@@ -9103,8 +9103,8 @@
       sed "s/^/$as_me: WARNING:     /" >&2
     ;;
 esac
-echo "$as_me:$LINENO: checking for security/pam_appl.h" >&5
-echo $ECHO_N "checking for security/pam_appl.h... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking for pam/pam_appl.h" >&5
+echo $ECHO_N "checking for pam/pam_appl.h... $ECHO_C" >&6
 if test "${ac_cv_header_security_pam_appl_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
diff -ru cyrus-sasl-2.1.20.orig/utils/Makefile.in cyrus-sasl-2.1.20/utils/Makefile.in
--- cyrus-sasl-2.1.20.orig/utils/Makefile.in	2004-10-24 16:21:10.000000000 -0400
+++ cyrus-sasl-2.1.20/utils/Makefile.in	2004-12-09 00:25:28.000000000 -0500
@@ -237,9 +237,9 @@
 
 @NO_SASL_DB_MANS_TRUE@man_MANS = 
 
-saslpasswd2_LDADD = ../sasldb/libsasldb.la $(all_sasl_libs)
+saslpasswd2_LDADD = $(all_sasl_libs) ../sasldb/libsasldb.la
 saslpasswd2_SOURCES = saslpasswd.c
-sasldblistusers2_LDADD = ../sasldb/libsasldb.la $(all_sasl_libs)
+sasldblistusers2_LDADD = $(all_sasl_libs) ../sasldb/libsasldb.la
 sasldblistusers2_SOURCES = sasldblistusers.c
 dbconverter_2_LDADD = ../sasldb/libsasldb.la $(all_sasl_libs)
 
The only places where global_callbacks is actually used is in a module which
declares it "static".  Remove it from this file to avoid that conflict.
--- cyrus-sasl-2.1.20/lib/saslint.h	2005-03-02 02:08:22.200866937 -0500
+++ cyrus-sasl-2.1.20/lib/saslint.h	2005-03-02 02:08:25.793397780 -0500
@@ -109,7 +109,7 @@
   const char *appname;
 } sasl_global_callbacks_t;
 
-extern sasl_global_callbacks_t global_callbacks;
+/* extern sasl_global_callbacks_t global_callbacks; */
 
 typedef struct _sasl_external_properties 
 {
