diff -ru ethereal-0.10.0a/configure ethereal-0.10.0a-patched/configure
--- ethereal-0.10.0a/configure	2003-12-13 18:12:54.000000000 +0100
+++ ethereal-0.10.0a-patched/configure	2003-12-29 04:53:45.000000000 +0100
@@ -5723,7 +5723,7 @@
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
     #	     `"' quotes if we put them in here... so don't!
-    archive_cmds='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
+    archive_cmds='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $tmp_verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -ru ethereal-0.10.0a/ltmain.sh ethereal-0.10.0a-patched/ltmain.sh
--- ethereal-0.10.0a/ltmain.sh	2003-12-13 18:12:38.000000000 +0100
+++ ethereal-0.10.0a-patched/ltmain.sh	2003-12-29 04:53:45.000000000 +0100
@@ -2961,6 +2961,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+	  else
+	    tmp_verstring="$verstring"
+	  fi
 	  save_deplibs="$deplibs"
 	  for conv in $convenience; do
 	    tmp_deplibs=
