diff -Naur evolution-1.4.6-old/camel/camel-object.c evolution-1.4.6/camel/camel-object.c
--- evolution-1.4.6-old/camel/camel-object.c	2003-05-22 16:17:02.000000000 -0400
+++ evolution-1.4.6/camel/camel-object.c	2004-06-25 18:59:33.000000000 -0400
@@ -1113,7 +1113,7 @@
 	int i;
 
 	sem_getvalue(&bag->reserve_sem, &i);
-	g_assert(i == 1);
+	/*g_assert(i == 1);*/
 
 	g_hash_table_foreach(bag->object_table, (GHFunc)save_object, objects);
 	for (i=0;i<objects->len;i++)
diff -Naur evolution-1.4.6-old/e-util/Makefile.in evolution-1.4.6/e-util/Makefile.in
--- evolution-1.4.6-old/e-util/Makefile.in	2004-03-10 10:22:26.000000000 -0500
+++ evolution-1.4.6/e-util/Makefile.in	2004-06-25 19:01:27.000000000 -0400
@@ -224,7 +224,7 @@
 	$(E_UTIL_CFLAGS)
 
 
-privlib_LTLIBRARIES = libeutil.la libedb3util.la libeconduit.la
+privlib_LTLIBRARIES = libeutil.la libedb3util.la
 noinst_LTLIBRARIES = libeutil-static.la
 
 eutilinclude_HEADERS = \
diff -Naur evolution-1.4.6-old/ltmain.sh evolution-1.4.6/ltmain.sh
--- evolution-1.4.6-old/ltmain.sh	2003-02-08 09:47:44.000000000 -0500
+++ evolution-1.4.6/ltmain.sh	2004-06-25 19:01:27.000000000 -0400
@@ -4050,10 +4050,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test $need_relink = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test $need_relink = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
diff -Naur evolution-1.4.6-old/po/Makefile.in.in evolution-1.4.6/po/Makefile.in.in
--- evolution-1.4.6-old/po/Makefile.in.in	2004-03-04 17:07:23.000000000 -0500
+++ evolution-1.4.6/po/Makefile.in.in	2004-06-25 19:01:27.000000000 -0400
@@ -28,7 +28,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 libdir = @libdir@
-localedir = $(libdir)/locale
+localedir = $(datadir)/locale
 gnulocaledir = $(datadir)/locale
 gettextsrcdir = $(datadir)/glib-2.0/gettext/po
 subdir = po
