diff -Naur gnome-vfs-2.4.2-old/Makefile.in gnome-vfs-2.4.2/Makefile.in
--- gnome-vfs-2.4.2-old/Makefile.in	Tue Jan 27 13:29:31 2004
+++ gnome-vfs-2.4.2/Makefile.in	Tue Jan 27 14:07:56 2004
@@ -191,7 +191,7 @@
 
 SUBDIRS_GNOME = monikers doc schemas
 
-SUBDIRS =  	libgnomevfs					modules						schemas						devel-docs					doc						programs					test						monikers					po
+SUBDIRS =  	libgnomevfs					modules						schemas						devel-docs					doc						programs					monikers					po
 
 
 EXTRA_DIST =  	HACKING						TODO						autogen.sh					gnome-vfs.spec					gnome-vfs-2.0.pc.in				gnome-vfs-module-2.0.pc.in			intltool-extract.in				intltool-merge.in				intltool-update.in
diff -Naur gnome-vfs-2.4.2-old/Makefile.in.orig gnome-vfs-2.4.2/Makefile.in.orig
--- gnome-vfs-2.4.2-old/Makefile.in.orig	Wed Dec 31 19:00:00 1969
+++ gnome-vfs-2.4.2/Makefile.in.orig	Tue Jan 27 13:29:52 2004
@@ -0,0 +1,516 @@
+# Makefile.in generated automatically by automake 1.4-p6 from Makefile.am
+
+# Copyright (C) 1994, 1995-8, 1999, 2001 Free Software Foundation, Inc.
+# This Makefile.in is free software; the Free Software Foundation
+# gives unlimited permission to copy and/or distribute it,
+# with or without modifications, as long as this notice is preserved.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY, to the extent permitted by law; without
+# even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.
+
+
+SHELL = @SHELL@
+
+srcdir = @srcdir@
+top_srcdir = @top_srcdir@
+VPATH = @srcdir@
+prefix = @prefix@
+exec_prefix = @exec_prefix@
+
+bindir = @bindir@
+sbindir = @sbindir@
+libexecdir = @libexecdir@
+datadir = @datadir@
+sysconfdir = @sysconfdir@
+sharedstatedir = @sharedstatedir@
+localstatedir = @localstatedir@
+libdir = @libdir@
+infodir = @infodir@
+mandir = @mandir@
+includedir = @includedir@
+oldincludedir = /usr/include
+
+DESTDIR =
+
+pkgdatadir = $(datadir)/@PACKAGE@
+pkglibdir = $(libdir)/@PACKAGE@
+pkgincludedir = $(includedir)/@PACKAGE@
+
+top_builddir = .
+
+ACLOCAL = @ACLOCAL@
+AUTOCONF = @AUTOCONF@
+AUTOMAKE = @AUTOMAKE@
+AUTOHEADER = @AUTOHEADER@
+
+INSTALL = @INSTALL@
+INSTALL_PROGRAM = @INSTALL_PROGRAM@ $(AM_INSTALL_PROGRAM_FLAGS)
+INSTALL_DATA = @INSTALL_DATA@
+INSTALL_SCRIPT = @INSTALL_SCRIPT@
+transform = @program_transform_name@
+
+NORMAL_INSTALL = :
+PRE_INSTALL = :
+POST_INSTALL = :
+NORMAL_UNINSTALL = :
+PRE_UNINSTALL = :
+POST_UNINSTALL = :
+host_alias = @host_alias@
+host_triplet = @host@
+ACLOCAL_AMFLAGS = @ACLOCAL_AMFLAGS@
+AR = @AR@
+AS = @AS@
+AWK = @AWK@
+BONOBO_ACTIVATION_REQUIRED = @BONOBO_ACTIVATION_REQUIRED@
+BONOBO_IDLDIR = @BONOBO_IDLDIR@
+BONOBO_REQUIRED = @BONOBO_REQUIRED@
+BZ2_LIBS = @BZ2_LIBS@
+CATALOGS = @CATALOGS@
+CATOBJEXT = @CATOBJEXT@
+CC = @CC@
+CFLAGS = @CFLAGS@
+CPP = @CPP@
+CPPFLAGS = @CPPFLAGS@
+CXX = @CXX@
+CXXCPP = @CXXCPP@
+DATADIRNAME = @DATADIRNAME@
+DB2HTML = @DB2HTML@
+DISABLE_DEPRECATED = @DISABLE_DEPRECATED@
+DLLTOOL = @DLLTOOL@
+ECHO = @ECHO@
+EGREP = @EGREP@
+ENABLE_PROFILER = @ENABLE_PROFILER@
+EXEEXT = @EXEEXT@
+F77 = @F77@
+FAM_LIBS = @FAM_LIBS@
+GCJ = @GCJ@
+GCJFLAGS = @GCJFLAGS@
+GCONFTOOL = @GCONFTOOL@
+GCONF_REQUIRED = @GCONF_REQUIRED@
+GCONF_SCHEMA_CONFIG_SOURCE = @GCONF_SCHEMA_CONFIG_SOURCE@
+GCONF_SCHEMA_FILE_DIR = @GCONF_SCHEMA_FILE_DIR@
+GETTEXT_PACKAGE = @GETTEXT_PACKAGE@
+GLIB_REQUIRED = @GLIB_REQUIRED@
+GMOFILES = @GMOFILES@
+GMSGFMT = @GMSGFMT@
+GNOME_ACLOCAL_DIR = @GNOME_ACLOCAL_DIR@
+GNOME_ACLOCAL_FLAGS = @GNOME_ACLOCAL_FLAGS@
+GNOME_VFS_DIR = @GNOME_VFS_DIR@
+GTKDOC = @GTKDOC@
+HAVE_GTK_DOC = @HAVE_GTK_DOC@
+HAVE_LIB = @HAVE_LIB@
+HTML_DIR = @HTML_DIR@
+INSTOBJEXT = @INSTOBJEXT@
+INTLLIBS = @INTLLIBS@
+INTLTOOL_CAVES_RULE = @INTLTOOL_CAVES_RULE@
+INTLTOOL_DESKTOP_RULE = @INTLTOOL_DESKTOP_RULE@
+INTLTOOL_DIRECTORY_RULE = @INTLTOOL_DIRECTORY_RULE@
+INTLTOOL_EXTRACT = @INTLTOOL_EXTRACT@
+INTLTOOL_KBD_RULE = @INTLTOOL_KBD_RULE@
+INTLTOOL_KEYS_RULE = @INTLTOOL_KEYS_RULE@
+INTLTOOL_MERGE = @INTLTOOL_MERGE@
+INTLTOOL_OAF_RULE = @INTLTOOL_OAF_RULE@
+INTLTOOL_PERL = @INTLTOOL_PERL@
+INTLTOOL_PONG_RULE = @INTLTOOL_PONG_RULE@
+INTLTOOL_PROP_RULE = @INTLTOOL_PROP_RULE@
+INTLTOOL_SCHEMAS_RULE = @INTLTOOL_SCHEMAS_RULE@
+INTLTOOL_SERVER_RULE = @INTLTOOL_SERVER_RULE@
+INTLTOOL_SHEET_RULE = @INTLTOOL_SHEET_RULE@
+INTLTOOL_SOUNDLIST_RULE = @INTLTOOL_SOUNDLIST_RULE@
+INTLTOOL_THEME_RULE = @INTLTOOL_THEME_RULE@
+INTLTOOL_UI_RULE = @INTLTOOL_UI_RULE@
+INTLTOOL_UPDATE = @INTLTOOL_UPDATE@
+INTLTOOL_XAM_RULE = @INTLTOOL_XAM_RULE@
+INTLTOOL_XML_RULE = @INTLTOOL_XML_RULE@
+LDFLAGS = @LDFLAGS@
+LIB = @LIB@
+LIBEFS_CFLAGS = @LIBEFS_CFLAGS@
+LIBEFS_LIBS = @LIBEFS_LIBS@
+LIBGNOMEVFS_AGE = @LIBGNOMEVFS_AGE@
+LIBGNOMEVFS_CFLAGS = @LIBGNOMEVFS_CFLAGS@
+LIBGNOMEVFS_CURRENT = @LIBGNOMEVFS_CURRENT@
+LIBGNOMEVFS_LIBS = @LIBGNOMEVFS_LIBS@
+LIBGNOMEVFS_REVISION = @LIBGNOMEVFS_REVISION@
+LIBGNOMEVFS_VERSION = @LIBGNOMEVFS_VERSION@
+LIBGNUTLS_CFLAGS = @LIBGNUTLS_CFLAGS@
+LIBGNUTLS_CONFIG = @LIBGNUTLS_CONFIG@
+LIBGNUTLS_LIBS = @LIBGNUTLS_LIBS@
+LIBIDL_REQUIRED = @LIBIDL_REQUIRED@
+LIBTOOL = @LIBTOOL@
+LN_S = @LN_S@
+LTLIB = @LTLIB@
+MAINT = @MAINT@
+MAKEINFO = @MAKEINFO@
+MKINSTALLDIRS = @MKINSTALLDIRS@
+MODULES_CFLAGS = @MODULES_CFLAGS@
+MODULES_FILE_CFLAGS = @MODULES_FILE_CFLAGS@
+MODULES_FILE_LIBS = @MODULES_FILE_LIBS@
+MODULES_GCONF_CFLAGS = @MODULES_GCONF_CFLAGS@
+MODULES_GCONF_LIBS = @MODULES_GCONF_LIBS@
+MODULES_LIBS = @MODULES_LIBS@
+MODULES_XML_CFLAGS = @MODULES_XML_CFLAGS@
+MODULES_XML_GCONF_CFLAGS = @MODULES_XML_GCONF_CFLAGS@
+MODULES_XML_GCONF_LIBS = @MODULES_XML_GCONF_LIBS@
+MODULES_XML_LIBS = @MODULES_XML_LIBS@
+MONIKERS_CFLAGS = @MONIKERS_CFLAGS@
+MONIKERS_LIBS = @MONIKERS_LIBS@
+OBJDUMP = @OBJDUMP@
+OBJEXT = @OBJEXT@
+ORBIT_IDL = @ORBIT_IDL@
+ORBIT_REQUIRED = @ORBIT_REQUIRED@
+PACKAGE = @PACKAGE@
+PKG_CONFIG = @PKG_CONFIG@
+POFILES = @POFILES@
+POPT_LIBS = @POPT_LIBS@
+POSUB = @POSUB@
+PO_IN_DATADIR_FALSE = @PO_IN_DATADIR_FALSE@
+PO_IN_DATADIR_TRUE = @PO_IN_DATADIR_TRUE@
+RANLIB = @RANLIB@
+RC = @RC@
+STRIP = @STRIP@
+TEST_CFLAGS = @TEST_CFLAGS@
+TEST_LIBS = @TEST_LIBS@
+TOP_BUILDDIR = @TOP_BUILDDIR@
+USE_NLS = @USE_NLS@
+VERSION = @VERSION@
+VFS_CFLAGS = @VFS_CFLAGS@
+VFS_INCLUDEDIR = @VFS_INCLUDEDIR@
+VFS_LIBDIR = @VFS_LIBDIR@
+VFS_LIBS = @VFS_LIBS@
+VFS_OFFSET = @VFS_OFFSET@
+VFS_OFFSET_IS = @VFS_OFFSET_IS@
+VFS_OFFSET_PRINTF = @VFS_OFFSET_PRINTF@
+VFS_SIZE = @VFS_SIZE@
+VFS_SIZE_IS = @VFS_SIZE_IS@
+VFS_SIZE_PRINTF = @VFS_SIZE_PRINTF@
+WARN_CFLAGS = @WARN_CFLAGS@
+WARN_CXXFLAGS = @WARN_CXXFLAGS@
+XML_REQUIRED = @XML_REQUIRED@
+
+SUBDIRS_GNOME = monikers doc schemas
+
+SUBDIRS =  	libgnomevfs					modules						schemas						devel-docs					doc						programs					test						monikers					po
+
+
+EXTRA_DIST =  	HACKING						TODO						autogen.sh					gnome-vfs.spec					gnome-vfs-2.0.pc.in				gnome-vfs-module-2.0.pc.in			intltool-extract.in				intltool-merge.in				intltool-update.in
+
+
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = gnome-vfs-2.0.pc gnome-vfs-module-2.0.pc
+ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
+mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
+CONFIG_HEADER = config.h
+CONFIG_CLEAN_FILES =  gnome-vfs.spec gnome-vfs-2.0.pc \
+gnome-vfs-module-2.0.pc
+DATA =  $(pkgconfig_DATA)
+
+DIST_COMMON =  README ./stamp-h.in AUTHORS COPYING COPYING.LIB ChangeLog \
+INSTALL Makefile.am Makefile.in NEWS TODO acconfig.h acinclude.m4 \
+aclocal.m4 config.guess config.h.in config.sub configure configure.in \
+gnome-vfs-2.0.pc.in gnome-vfs-module-2.0.pc.in gnome-vfs.spec.in \
+install-sh ltmain.sh missing mkinstalldirs
+
+
+DISTFILES = $(DIST_COMMON) $(SOURCES) $(HEADERS) $(TEXINFOS) $(EXTRA_DIST)
+
+TAR = tar
+GZIP_ENV = --best
+all: all-redirect
+.SUFFIXES:
+$(srcdir)/Makefile.in: @MAINTAINER_MODE_TRUE@ Makefile.am $(top_srcdir)/configure.in $(ACLOCAL_M4) 
+	cd $(top_srcdir) && $(AUTOMAKE) --gnu --include-deps Makefile
+
+Makefile: $(srcdir)/Makefile.in  $(top_builddir)/config.status
+	cd $(top_builddir) \
+	  && CONFIG_FILES=$@ CONFIG_HEADERS= $(SHELL) ./config.status
+
+$(ACLOCAL_M4): @MAINTAINER_MODE_TRUE@ configure.in  acinclude.m4
+	cd $(srcdir) && $(ACLOCAL) $(ACLOCAL_AMFLAGS)
+
+config.status: $(srcdir)/configure $(CONFIG_STATUS_DEPENDENCIES)
+	$(SHELL) ./config.status --recheck
+$(srcdir)/configure: @MAINTAINER_MODE_TRUE@$(srcdir)/configure.in $(ACLOCAL_M4) $(CONFIGURE_DEPENDENCIES)
+	cd $(srcdir) && $(AUTOCONF)
+
+config.h: stamp-h
+	@if test ! -f $@; then \
+		rm -f stamp-h; \
+		$(MAKE) stamp-h; \
+	else :; fi
+stamp-h: $(srcdir)/config.h.in $(top_builddir)/config.status
+	cd $(top_builddir) \
+	  && CONFIG_FILES= CONFIG_HEADERS=config.h \
+	     $(SHELL) ./config.status
+	@echo timestamp > stamp-h 2> /dev/null
+$(srcdir)/config.h.in: @MAINTAINER_MODE_TRUE@$(srcdir)/stamp-h.in
+	@if test ! -f $@; then \
+		rm -f $(srcdir)/stamp-h.in; \
+		$(MAKE) $(srcdir)/stamp-h.in; \
+	else :; fi
+$(srcdir)/stamp-h.in: $(top_srcdir)/configure.in $(ACLOCAL_M4) acconfig.h
+	cd $(top_srcdir) && $(AUTOHEADER)
+	@echo timestamp > $(srcdir)/stamp-h.in 2> /dev/null
+
+mostlyclean-hdr:
+
+clean-hdr:
+
+distclean-hdr:
+	-rm -f config.h
+
+maintainer-clean-hdr:
+gnome-vfs.spec: $(top_builddir)/config.status gnome-vfs.spec.in
+	cd $(top_builddir) && CONFIG_FILES=$@ CONFIG_HEADERS= $(SHELL) ./config.status
+gnome-vfs-2.0.pc: $(top_builddir)/config.status gnome-vfs-2.0.pc.in
+	cd $(top_builddir) && CONFIG_FILES=$@ CONFIG_HEADERS= $(SHELL) ./config.status
+gnome-vfs-module-2.0.pc: $(top_builddir)/config.status gnome-vfs-module-2.0.pc.in
+	cd $(top_builddir) && CONFIG_FILES=$@ CONFIG_HEADERS= $(SHELL) ./config.status
+
+install-pkgconfigDATA: $(pkgconfig_DATA)
+	@$(NORMAL_INSTALL)
+	$(mkinstalldirs) $(DESTDIR)$(pkgconfigdir)
+	@list='$(pkgconfig_DATA)'; for p in $$list; do \
+	  if test -f $(srcdir)/$$p; then \
+	    echo " $(INSTALL_DATA) $(srcdir)/$$p $(DESTDIR)$(pkgconfigdir)/$$p"; \
+	    $(INSTALL_DATA) $(srcdir)/$$p $(DESTDIR)$(pkgconfigdir)/$$p; \
+	  else if test -f $$p; then \
+	    echo " $(INSTALL_DATA) $$p $(DESTDIR)$(pkgconfigdir)/$$p"; \
+	    $(INSTALL_DATA) $$p $(DESTDIR)$(pkgconfigdir)/$$p; \
+	  fi; fi; \
+	done
+
+uninstall-pkgconfigDATA:
+	@$(NORMAL_UNINSTALL)
+	list='$(pkgconfig_DATA)'; for p in $$list; do \
+	  rm -f $(DESTDIR)$(pkgconfigdir)/$$p; \
+	done
+
+# This directory's subdirectories are mostly independent; you can cd
+# into them and run `make' without going through this Makefile.
+# To change the values of `make' variables: instead of editing Makefiles,
+# (1) if the variable is set in `config.status', edit `config.status'
+#     (which will cause the Makefiles to be regenerated when you run `make');
+# (2) otherwise, pass the desired values on the `make' command line.
+
+@SET_MAKE@
+
+all-recursive install-data-recursive install-exec-recursive \
+installdirs-recursive install-recursive uninstall-recursive  \
+check-recursive installcheck-recursive info-recursive dvi-recursive:
+	@set fnord $(MAKEFLAGS); amf=$$2; \
+	dot_seen=no; \
+	target=`echo $@ | sed s/-recursive//`; \
+	list='$(SUBDIRS)'; for subdir in $$list; do \
+	  echo "Making $$target in $$subdir"; \
+	  if test "$$subdir" = "."; then \
+	    dot_seen=yes; \
+	    local_target="$$target-am"; \
+	  else \
+	    local_target="$$target"; \
+	  fi; \
+	  (cd $$subdir && $(MAKE) $(AM_MAKEFLAGS) $$local_target) \
+	   || case "$$amf" in *=*) exit 1;; *k*) fail=yes;; *) exit 1;; esac; \
+	done; \
+	if test "$$dot_seen" = "no"; then \
+	  $(MAKE) $(AM_MAKEFLAGS) "$$target-am" || exit 1; \
+	fi; test -z "$$fail"
+
+mostlyclean-recursive clean-recursive distclean-recursive \
+maintainer-clean-recursive:
+	@set fnord $(MAKEFLAGS); amf=$$2; \
+	dot_seen=no; \
+	rev=''; list='$(SUBDIRS)'; for subdir in $$list; do \
+	  rev="$$subdir $$rev"; \
+	  test "$$subdir" != "." || dot_seen=yes; \
+	done; \
+	test "$$dot_seen" = "no" && rev=". $$rev"; \
+	target=`echo $@ | sed s/-recursive//`; \
+	for subdir in $$rev; do \
+	  echo "Making $$target in $$subdir"; \
+	  if test "$$subdir" = "."; then \
+	    local_target="$$target-am"; \
+	  else \
+	    local_target="$$target"; \
+	  fi; \
+	  (cd $$subdir && $(MAKE) $(AM_MAKEFLAGS) $$local_target) \
+	   || case "$$amf" in *=*) exit 1;; *k*) fail=yes;; *) exit 1;; esac; \
+	done && test -z "$$fail"
+tags-recursive:
+	list='$(SUBDIRS)'; for subdir in $$list; do \
+	  test "$$subdir" = . || (cd $$subdir && $(MAKE) $(AM_MAKEFLAGS) tags); \
+	done
+
+tags: TAGS
+
+ID: $(HEADERS) $(SOURCES) $(LISP)
+	list='$(SOURCES) $(HEADERS)'; \
+	unique=`for i in $$list; do echo $$i; done | \
+	  awk '    { files[$$0] = 1; } \
+	       END { for (i in files) print i; }'`; \
+	here=`pwd` && cd $(srcdir) \
+	  && mkid -f$$here/ID $$unique $(LISP)
+
+TAGS: tags-recursive $(HEADERS) $(SOURCES) config.h.in $(TAGS_DEPENDENCIES) $(LISP)
+	tags=; \
+	here=`pwd`; \
+	list='$(SUBDIRS)'; for subdir in $$list; do \
+   if test "$$subdir" = .; then :; else \
+	    test -f $$subdir/TAGS && tags="$$tags -i $$here/$$subdir/TAGS"; \
+   fi; \
+	done; \
+	list='$(SOURCES) $(HEADERS)'; \
+	unique=`for i in $$list; do echo $$i; done | \
+	  awk '    { files[$$0] = 1; } \
+	       END { for (i in files) print i; }'`; \
+	test -z "$(ETAGS_ARGS)config.h.in$$unique$(LISP)$$tags" \
+	  || (cd $(srcdir) && etags -o $$here/TAGS $(ETAGS_ARGS) $$tags config.h.in $$unique $(LISP))
+
+mostlyclean-tags:
+
+clean-tags:
+
+distclean-tags:
+	-rm -f TAGS ID
+
+maintainer-clean-tags:
+
+distdir = $(PACKAGE)-$(VERSION)
+top_distdir = $(distdir)
+
+# This target untars the dist file and tries a VPATH configuration.  Then
+# it guarantees that the distribution is self-contained by making another
+# tarfile.
+distcheck: dist
+	-rm -rf $(distdir)
+	GZIP=$(GZIP_ENV) $(TAR) zxf $(distdir).tar.gz
+	mkdir $(distdir)/=build
+	mkdir $(distdir)/=inst
+	dc_install_base=`cd $(distdir)/=inst && pwd`; \
+	cd $(distdir)/=build \
+	  && ../configure --srcdir=.. --prefix=$$dc_install_base \
+	  && $(MAKE) $(AM_MAKEFLAGS) \
+	  && $(MAKE) $(AM_MAKEFLAGS) dvi \
+	  && $(MAKE) $(AM_MAKEFLAGS) check \
+	  && $(MAKE) $(AM_MAKEFLAGS) install \
+	  && $(MAKE) $(AM_MAKEFLAGS) installcheck \
+	  && $(MAKE) $(AM_MAKEFLAGS) dist
+	-rm -rf $(distdir)
+	@banner="$(distdir).tar.gz is ready for distribution"; \
+	dashes=`echo "$$banner" | sed s/./=/g`; \
+	echo "$$dashes"; \
+	echo "$$banner"; \
+	echo "$$dashes"
+dist: distdir
+	-chmod -R a+r $(distdir)
+	GZIP=$(GZIP_ENV) $(TAR) chozf $(distdir).tar.gz $(distdir)
+	-rm -rf $(distdir)
+dist-all: distdir
+	-chmod -R a+r $(distdir)
+	GZIP=$(GZIP_ENV) $(TAR) chozf $(distdir).tar.gz $(distdir)
+	-rm -rf $(distdir)
+distdir: $(DISTFILES)
+	-rm -rf $(distdir)
+	mkdir $(distdir)
+	-chmod 777 $(distdir)
+	@for file in $(DISTFILES); do \
+	  d=$(srcdir); \
+	  if test -d $$d/$$file; then \
+	    cp -pr $$d/$$file $(distdir)/$$file; \
+	  else \
+	    test -f $(distdir)/$$file \
+	    || ln $$d/$$file $(distdir)/$$file 2> /dev/null \
+	    || cp -p $$d/$$file $(distdir)/$$file || :; \
+	  fi; \
+	done
+	for subdir in $(SUBDIRS); do \
+	  if test "$$subdir" = .; then :; else \
+	    test -d $(distdir)/$$subdir \
+	    || mkdir $(distdir)/$$subdir \
+	    || exit 1; \
+	    chmod 777 $(distdir)/$$subdir; \
+	    (cd $$subdir && $(MAKE) $(AM_MAKEFLAGS) top_distdir=../$(distdir) distdir=../$(distdir)/$$subdir distdir) \
+	      || exit 1; \
+	  fi; \
+	done
+info-am:
+info: info-recursive
+dvi-am:
+dvi: dvi-recursive
+check-am: all-am
+check: check-recursive
+installcheck-am:
+installcheck: installcheck-recursive
+all-recursive-am: config.h
+	$(MAKE) $(AM_MAKEFLAGS) all-recursive
+
+install-exec-am:
+install-exec: install-exec-recursive
+
+install-data-am: install-pkgconfigDATA
+install-data: install-data-recursive
+
+install-am: all-am
+	@$(MAKE) $(AM_MAKEFLAGS) install-exec-am install-data-am
+install: install-recursive
+uninstall-am: uninstall-pkgconfigDATA
+uninstall: uninstall-recursive
+all-am: Makefile $(DATA) config.h
+all-redirect: all-recursive-am
+install-strip:
+	$(MAKE) $(AM_MAKEFLAGS) AM_INSTALL_PROGRAM_FLAGS=-s install
+installdirs: installdirs-recursive
+installdirs-am:
+	$(mkinstalldirs)  $(DESTDIR)$(pkgconfigdir)
+
+
+mostlyclean-generic:
+
+clean-generic:
+
+distclean-generic:
+	-rm -f Makefile $(CONFIG_CLEAN_FILES)
+	-rm -f config.cache config.log stamp-h stamp-h[0-9]*
+
+maintainer-clean-generic:
+mostlyclean-am:  mostlyclean-hdr mostlyclean-tags mostlyclean-generic
+
+mostlyclean: mostlyclean-recursive
+
+clean-am:  clean-hdr clean-tags clean-generic mostlyclean-am
+
+clean: clean-recursive
+
+distclean-am:  distclean-hdr distclean-tags distclean-generic clean-am
+	-rm -f libtool
+
+distclean: distclean-recursive
+	-rm -f config.status
+
+maintainer-clean-am:  maintainer-clean-hdr maintainer-clean-tags \
+		maintainer-clean-generic distclean-am
+	@echo "This command is intended for maintainers to use;"
+	@echo "it deletes files that may require special tools to rebuild."
+
+maintainer-clean: maintainer-clean-recursive
+	-rm -f config.status
+
+.PHONY: mostlyclean-hdr distclean-hdr clean-hdr maintainer-clean-hdr \
+uninstall-pkgconfigDATA install-pkgconfigDATA install-data-recursive \
+uninstall-data-recursive install-exec-recursive \
+uninstall-exec-recursive installdirs-recursive uninstalldirs-recursive \
+all-recursive check-recursive installcheck-recursive info-recursive \
+dvi-recursive mostlyclean-recursive distclean-recursive clean-recursive \
+maintainer-clean-recursive tags tags-recursive mostlyclean-tags \
+distclean-tags clean-tags maintainer-clean-tags distdir info-am info \
+dvi-am dvi check check-am installcheck-am installcheck all-recursive-am \
+install-exec-am install-exec install-data-am install-data install-am \
+install uninstall-am uninstall all-redirect all-am all installdirs-am \
+installdirs mostlyclean-generic distclean-generic clean-generic \
+maintainer-clean-generic clean mostlyclean distclean maintainer-clean
+
+
+# Tell versions [3.59,3.63) of GNU make to not export all variables.
+# Otherwise a system limit (for SysV at least) may be exceeded.
+.NOEXPORT:
diff -Naur gnome-vfs-2.4.2-old/configure gnome-vfs-2.4.2/configure
--- gnome-vfs-2.4.2-old/configure	Tue Jan 27 13:29:31 2004
+++ gnome-vfs-2.4.2/configure	Tue Jan 27 14:05:46 2004
@@ -22448,6 +22448,8 @@
 if eval "test \"\${$as_ac_var+set}\" = set"; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
+  ac_check_lib_save_LIBS=$LIBS
+  LIBS="-lintl  $LIBS"
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
diff -Naur gnome-vfs-2.4.2-old/libgnomevfs/gnome-vfs-configuration.c gnome-vfs-2.4.2/libgnomevfs/gnome-vfs-configuration.c
--- gnome-vfs-2.4.2-old/libgnomevfs/gnome-vfs-configuration.c	Tue Jan 27 13:29:31 2004
+++ gnome-vfs-2.4.2/libgnomevfs/gnome-vfs-configuration.c	Tue Jan 27 14:07:56 2004
@@ -24,6 +24,7 @@
 #include <config.h>
 #include "gnome-vfs-configuration.h"
 
+#include <sys/types.h>
 #include <dirent.h>
 #include <errno.h>
 #include <glib/ghash.h>
diff -Naur gnome-vfs-2.4.2-old/ltmain.sh gnome-vfs-2.4.2/ltmain.sh
--- gnome-vfs-2.4.2-old/ltmain.sh	Tue Jan 27 13:29:31 2004
+++ gnome-vfs-2.4.2/ltmain.sh	Tue Jan 27 14:07:56 2004
@@ -5145,10 +5145,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test "$need_relink" = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test "$need_relink" = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
diff -Naur gnome-vfs-2.4.2-old/modules/file-method.c gnome-vfs-2.4.2/modules/file-method.c
--- gnome-vfs-2.4.2-old/modules/file-method.c	Tue Jan 27 13:29:31 2004
+++ gnome-vfs-2.4.2/modules/file-method.c	Tue Jan 27 14:07:56 2004
@@ -36,6 +36,7 @@
 #include <libgnomevfs/gnome-vfs-utils.h>
 #include <libgnomevfs/gnome-vfs-mime.h>
 #include <libgnomevfs/gnome-vfs-monitor-private.h>
+#include <sys/types.h>
 #include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
@@ -46,7 +47,6 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
-#include <sys/types.h>
 #include <unistd.h>
 #include <utime.h>
 #include <string.h>
diff -Naur gnome-vfs-2.4.2-old/modules/fstype.c gnome-vfs-2.4.2/modules/fstype.c
--- gnome-vfs-2.4.2-old/modules/fstype.c	Tue Jan 27 13:29:31 2004
+++ gnome-vfs-2.4.2/modules/fstype.c	Tue Jan 27 14:07:56 2004
@@ -363,7 +363,7 @@
   char *p;
 
   if (S_ISLNK (statp->st_mode))
-    p = dirname (relpath);
+    p = g_path_get_dirname (relpath);
   else
     p = relpath;
 
diff -Naur gnome-vfs-2.4.2-old/modules/fstype.c.orig gnome-vfs-2.4.2/modules/fstype.c.orig
--- gnome-vfs-2.4.2-old/modules/fstype.c.orig	Wed Dec 31 19:00:00 1969
+++ gnome-vfs-2.4.2/modules/fstype.c.orig	Tue Jan 27 13:29:52 2004
@@ -0,0 +1,427 @@
+/* fstype.c -- determine type of filesystems that files are on
+   Copyright (C) 1990, 91, 92, 93, 94 Free Software Foundation, Inc.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2, or (at your option)
+   any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.  */
+
+/* This code was relicensed by the FSF on May 1 2002
+ 
+   This file is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Library General Public License as
+   published by the Free Software Foundation; either version 2 of the
+   License, or (at your option) any later version.
+
+   this file is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public
+   License along with the Gnome Library; see the file COPYING.LIB.  If not,
+   write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+   Boston, MA 02111-1307, USA.
+ */
+/* Written by David MacKenzie <djm@gnu.ai.mit.edu>. */
+
+#include <config.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <errno.h>
+#ifdef STDC_HEADERS
+#include <stdlib.h>
+#else
+extern int errno;
+#endif
+#include <string.h>
+#include <glib.h>
+#ifdef HAVE_LIBGEN_H
+#include <libgen.h>
+#endif
+
+#if __STDC__
+# define P_(s) s
+#else
+# define P_(s) ()
+#endif
+
+static char *filesystem_type_uncached P_((char *path, char *relpath, struct stat *statp));
+
+void fstype_internal_error (int level, int num, char const *fmt, ...);
+
+#ifdef FSTYPE_MNTENT		/* 4.3BSD etc.  */
+static int xatoi P_((char *cp));
+#endif
+
+#ifdef FSTYPE_MNTENT		/* 4.3BSD, SunOS, HP-UX, Dynix, Irix.  */
+#include <mntent.h>
+#if !defined(MOUNTED)
+# if defined(MNT_MNTTAB)	/* HP-UX.  */
+#  define MOUNTED MNT_MNTTAB
+# endif
+# if defined(MNTTABNAME)	/* Dynix.  */
+#  define MOUNTED MNTTABNAME
+# endif
+#endif
+#endif
+
+#ifdef FSTYPE_GETMNT		/* Ultrix.  */
+#include <sys/param.h>
+#include <sys/mount.h>
+#include <sys/fs_types.h>
+#endif
+
+#ifdef FSTYPE_USG_STATFS	/* SVR3.  */
+#include <sys/statfs.h>
+#include <sys/fstyp.h>
+#endif
+
+#ifdef FSTYPE_STATVFS		/* SVR4.  */
+#include <sys/statvfs.h>
+#include <sys/fstyp.h>
+#endif
+
+#ifdef FSTYPE_STATFS		/* 4.4BSD.  */
+#include <sys/param.h>		/* NetBSD needs this.  */
+#include <sys/mount.h>
+
+#ifndef MFSNAMELEN		/* NetBSD defines this.  */
+static char *
+fstype_to_string (t)
+     short t;
+{
+#ifdef INITMOUNTNAMES		/* Defined in 4.4BSD, not in NET/2.  */
+  static char *mn[] = INITMOUNTNAMES;
+  if (t >= 0 && t <= MOUNT_MAXTYPE)
+    return mn[t];
+  else
+    return "?";
+#else /* !INITMOUNTNAMES */
+  switch (t)
+    {
+    case MOUNT_UFS:
+      return "ufs";
+    case MOUNT_NFS:
+      return "nfs";
+#ifdef MOUNT_PC
+    case MOUNT_PC:
+      return "pc";
+#endif
+#ifdef MOUNT_MFS
+    case MOUNT_MFS:
+      return "mfs";
+#endif
+#ifdef MOUNT_LO
+    case MOUNT_LO:
+      return "lofs";
+#endif
+#ifdef MOUNT_TFS
+    case MOUNT_TFS:
+      return "tfs";
+#endif
+#ifdef MOUNT_TMP
+    case MOUNT_TMP:
+      return "tmp";
+#endif
+#ifdef MOUNT_MSDOS
+    case MOUNT_MSDOS:
+      return "msdos";
+#endif
+#ifdef MOUNT_ISO9660
+    case MOUNT_ISO9660:
+      return "iso9660fs";
+#endif
+    default:
+      return "?";
+    }
+#endif /* !INITMOUNTNAMES */
+}
+#endif /* !MFSNAMELEN */
+#endif /* FSTYPE_STATFS */
+
+#ifdef FSTYPE_AIX_STATFS	/* AIX.  */
+#include <sys/vmount.h>
+#include <sys/statfs.h>
+
+#define FSTYPE_STATFS		/* Otherwise like 4.4BSD.  */
+#define f_type f_vfstype
+
+static char *
+fstype_to_string (t)
+     short t;
+{
+  switch (t)
+    {
+    case MNT_AIX:
+#if 0				/* NFS filesystems are actually MNT_AIX. */
+      return "aix";
+#endif
+    case MNT_NFS:
+      return "nfs";
+    case MNT_JFS:
+      return "jfs";
+    case MNT_CDROM:
+      return "cdrom";
+    default:
+      return "?";
+    }
+}
+#endif /* FSTYPE_AIX_STATFS */
+
+#ifdef AFS
+#include <netinet/in.h>
+#include <afs/venus.h>
+#if __STDC__
+/* On SunOS 4, afs/vice.h defines this to rely on a pre-ANSI cpp.  */
+#undef _VICEIOCTL
+#define _VICEIOCTL(id)  ((unsigned int ) _IOW('V', id, struct ViceIoctl))
+#endif
+#ifndef _IOW
+/* AFS on Solaris 2.3 doesn't get this definition.  */
+#include <sys/ioccom.h>
+#endif
+
+static int
+in_afs (path)
+     char *path;
+{
+  static char space[2048];
+  struct ViceIoctl vi;
+
+  vi.in_size = 0;
+  vi.out_size = sizeof (space);
+  vi.out = space;
+
+  if (pioctl (path, VIOC_FILE_CELL_NAME, &vi, 1)
+      && (errno == EINVAL || errno == ENOENT))
+	return 0;
+  return 1;
+}
+#endif /* AFS */
+
+/* Nonzero if the current filesystem's type is known.  */
+static int fstype_known = 0;
+
+char *filesystem_type (char *path, char *relpath, struct stat *statp);
+/* Return a static string naming the type of filesystem that the file PATH,
+   described by STATP, is on.
+   RELPATH is the file name relative to the current directory.
+   Return "unknown" if its filesystem type is unknown.  */
+
+char *
+filesystem_type (path, relpath, statp)
+     char *path;
+     char *relpath;
+     struct stat *statp;
+{
+  static char *current_fstype = NULL;
+  static dev_t current_dev;
+
+  if (current_fstype != NULL)
+    {
+      if (fstype_known && statp->st_dev == current_dev)
+	return current_fstype;	/* Cached value.  */
+      g_free (current_fstype);
+    }
+  current_dev = statp->st_dev;
+  current_fstype = filesystem_type_uncached (path, relpath, statp);
+  return current_fstype;
+}
+
+void
+fstype_internal_error (int level, int num, char const *fmt, ...)
+{
+}
+
+/* Return a newly allocated string naming the type of filesystem that the
+   file PATH, described by STATP, is on.
+   RELPATH is the file name relative to the current directory.
+   Return "unknown" if its filesystem type is unknown.  */
+
+static char *
+filesystem_type_uncached (path, relpath, statp)
+     char *path;
+     char *relpath;
+     struct stat *statp;
+{
+  char *type = NULL;
+
+#ifdef FSTYPE_MNTENT		/* 4.3BSD, SunOS, HP-UX, Dynix, Irix.  */
+  char *table = MOUNTED;
+  FILE *mfp;
+  struct mntent *mnt;
+
+  mfp = setmntent (table, "r");
+  if (mfp == NULL) {
+    fstype_internal_error (1, errno, "%s", table);
+    goto no_mtab;
+  }
+
+  /* Find the entry with the same device number as STATP, and return
+     that entry's fstype. */
+  while (type == NULL && (mnt = getmntent (mfp)))
+    {
+      char *devopt;
+      dev_t dev;
+      struct stat disk_stats;
+
+#ifdef MNTTYPE_IGNORE
+      if (!strcmp (mnt->mnt_type, MNTTYPE_IGNORE))
+	continue;
+#endif
+
+      /* Newer systems like SunOS 4.1 keep the dev number in the mtab,
+	 in the options string.	 For older systems, we need to stat the
+	 directory that the filesystem is mounted on to get it.
+
+	 Unfortunately, the HPUX 9.x mnttab entries created by automountq
+	 contain a dev= option but the option value does not match the
+	 st_dev value of the file (maybe the lower 16 bits match?).  */
+
+#if !defined(hpux) && !defined(__hpux__)
+      devopt = strstr (mnt->mnt_opts, "dev=");
+      if (devopt)
+	{
+	  if (devopt[4] == '0' && (devopt[5] == 'x' || devopt[5] == 'X'))
+	    dev = xatoi (devopt + 6);
+	  else
+	    dev = xatoi (devopt + 4);
+	}
+      else
+#endif /* not hpux */
+	{
+	  if (stat (mnt->mnt_dir, &disk_stats) == -1) {
+	    if (errno == EACCES)
+	      continue;
+	    else
+	      fstype_internal_error (1, errno, "error in %s: %s", table, mnt->mnt_dir);
+	  }
+	  dev = disk_stats.st_dev;
+	}
+
+      if (dev == statp->st_dev)
+	type = mnt->mnt_type;
+    }
+
+  if (endmntent (mfp) == 0)
+    fstype_internal_error (0, errno, "%s", table);
+ no_mtab:
+#endif
+
+#ifdef FSTYPE_GETMNT		/* Ultrix.  */
+  int offset = 0;
+  struct fs_data fsd;
+
+  while (type == NULL
+	 && getmnt (&offset, &fsd, sizeof (fsd), NOSTAT_MANY, 0) > 0)
+    {
+      if (fsd.fd_req.dev == statp->st_dev)
+	type = gt_names[fsd.fd_req.fstype];
+    }
+#endif
+
+#ifdef FSTYPE_USG_STATFS	/* SVR3.  */
+  struct statfs fss;
+  char typebuf[FSTYPSZ];
+
+  if (statfs (relpath, &fss, sizeof (struct statfs), 0) == -1)
+    {
+      /* Don't die if a file was just removed. */
+      if (errno != ENOENT)
+	fstype_internal_error (1, errno, "%s", path);
+    }
+  else if (!sysfs (GETFSTYP, fss.f_fstyp, typebuf))
+    type = typebuf;
+#endif
+
+#ifdef FSTYPE_STATVFS		/* SVR4.  */
+  struct statvfs fss;
+
+  if (statvfs (relpath, &fss) == -1)
+    {
+      /* Don't die if a file was just removed. */
+      if (errno != ENOENT)
+	fstype_internal_error (1, errno, "%s", path);
+    }
+  else
+    type = fss.f_basetype;
+#endif
+
+#ifdef FSTYPE_STATFS		/* 4.4BSD.  */
+  struct statfs fss;
+  char *p;
+
+  if (S_ISLNK (statp->st_mode))
+    p = dirname (relpath);
+  else
+    p = relpath;
+
+  if (statfs (p, &fss) == -1)
+    {
+      /* Don't die if symlink to nonexisting file, or a file that was
+	 just removed. */
+      if (errno != ENOENT)
+	fstype_internal_error (1, errno, "%s", path);
+    }
+  else
+    {
+#ifdef MFSNAMELEN		/* NetBSD.  */
+      type = fss.f_fstypename;
+#else
+      type = fstype_to_string (fss.f_type);
+#endif
+    }
+  if (p != relpath)
+    free (p);
+#endif
+
+#ifdef AFS
+  if ((!type || !strcmp (type, "xx")) && in_afs (relpath))
+    type = "afs";
+#endif
+
+  /* An unknown value can be caused by an ENOENT error condition.
+     Don't cache those values.  */
+  fstype_known = (type != NULL);
+
+  return g_strdup (type ? type : "unknown");
+}
+
+#ifdef FSTYPE_MNTENT		/* 4.3BSD etc.  */
+/* Return the value of the hexadecimal number represented by CP.
+   No prefix (like '0x') or suffix (like 'h') is expected to be
+   part of CP. */
+
+static int
+xatoi (cp)
+     char *cp;
+{
+  int val;
+  
+  val = 0;
+  while (*cp)
+    {
+      if (*cp >= 'a' && *cp <= 'f')
+	val = val * 16 + *cp - 'a' + 10;
+      else if (*cp >= 'A' && *cp <= 'F')
+	val = val * 16 + *cp - 'A' + 10;
+      else if (*cp >= '0' && *cp <= '9')
+	val = val * 16 + *cp - '0';
+      else
+	break;
+      cp++;
+    }
+  return val;
+}
+#endif
