diff -Naur mozilla-1.5.1.old/.mozconfig mozilla-1.5.1.new/.mozconfig
--- mozilla-1.5.1.old/.mozconfig	Thu Jan  1 09:00:00 1970
+++ mozilla-1.5.1.new/.mozconfig	Mon Dec 15 19:55:03 2003
@@ -0,0 +1,22 @@
+ac_add_options --enable-macos-target=10.3
+ac_add_options --enable-prebinding
+ac_add_options --enable-default-toolkit=gtk2
+ac_add_options --enable-xft
+ac_add_options --with-pthreads
+ac_add_options --mandir=@PREFIX@/share/man
+ac_add_options --with-system-jpeg=@PREFIX@
+ac_add_options --with-system-zlib=@PREFIX@
+ac_add_options --with-system-png=@PREFIX@
+ac_add_options --with-system-mng=@PREFIX@
+ac_add_options --disable-tests
+ac_add_options --disable-debug
+ac_add_options --disable-short-wchar
+ac_add_options --enable-optimize
+ac_add_options --enable-strip
+ac_add_options --enable-crypto
+ac_add_options --enable-mathml
+ac_add_options --enable-extensions=all
+ac_add_options --enable-xinerama
+ac_add_options --with-default-mozilla-five-home=@PREFIX@/lib/mozilla
+mk_add_options MOZ_MAKE_FLAGS=-j4
+
diff -Naur mozilla-1.5.1.old/config/add-chrome.pl mozilla-1.5.1.new/config/add-chrome.pl
--- mozilla-1.5.1.old/config/add-chrome.pl	Tue Mar  4 12:56:02 2003
+++ mozilla-1.5.1.new/config/add-chrome.pl	Mon Dec 15 19:55:03 2003
@@ -14,7 +14,7 @@
 my $jarFileName = $ARGV[4];
 
 my $win32 = ($^O =~ /((MS)?win32)|cygwin|os2/i) ? 1 : 0;
-my $macos = ($^O =~ /MacOS|darwin/i) ? 1 : 0;
+my $macos = 0;
 my $unix  = !($win32 || $macos) ? 1 : 0;
 
 sub foreignPlatformFile
diff -Naur mozilla-1.5.1.old/config/autoconf.mk.in mozilla-1.5.1.new/config/autoconf.mk.in
--- mozilla-1.5.1.old/config/autoconf.mk.in	Sat Aug 30 02:15:03 2003
+++ mozilla-1.5.1.new/config/autoconf.mk.in	Mon Dec 15 19:55:03 2003
@@ -29,14 +29,14 @@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-includedir	= @includedir@/mozilla-$(MOZILLA_VERSION)
+includedir	= @includedir@/mozilla
 libdir		= @libdir@
 datadir		= @datadir@
 mandir		= @mandir@
-idldir		= @datadir@/idl/mozilla-$(MOZILLA_VERSION)
+idldir		= @datadir@/idl/mozilla
 
-mozappdir	= $(libdir)/mozilla-$(MOZILLA_VERSION)
-mredir		= $(libdir)/mre/mre-$(MOZILLA_VERSION)
+mozappdir	= $(libdir)/mozilla
+mredir		= $(libdir)/mre/mre
 mrelibdir	= $(mredir)/lib
 
 DIST		= $(DEPTH)/dist
diff -Naur mozilla-1.5.1.old/config/config.mk mozilla-1.5.1.new/config/config.mk
--- mozilla-1.5.1.old/config/config.mk	Fri Aug 15 06:35:42 2003
+++ mozilla-1.5.1.new/config/config.mk	Mon Dec 15 19:55:03 2003
@@ -661,7 +661,7 @@
 ifeq ($(OS_ARCH),Darwin)
 ifdef USE_PREBINDING
 export LD_PREBIND=1
-export LD_SEG_ADDR_TABLE=$(shell cd $(topsrcdir); pwd)/config/prebind-address-table
+export LD_PREBIND_ALLOW_OVERLAP=1
 endif
 endif
 
diff -Naur mozilla-1.5.1.old/config/rules.mk mozilla-1.5.1.new/config/rules.mk
--- mozilla-1.5.1.old/config/rules.mk	Wed Aug 27 05:28:47 2003
+++ mozilla-1.5.1.new/config/rules.mk	Mon Dec 15 19:55:03 2003
@@ -425,7 +425,7 @@
 ifdef IS_COMPONENT
 EXTRA_DSO_LDOPTS	+= -bundle
 else
-EXTRA_DSO_LDOPTS	+= -dynamiclib -install_name @executable_path/$(SHARED_LIBRARY) -compatibility_version 1 -current_version 1
+EXTRA_DSO_LDOPTS	+= -dynamiclib -install_name \$(mozappdir)/$(SHARED_LIBRARY) -compatibility_version 1 -current_version 1
 endif
 endif
 endif
diff -Naur mozilla-1.5.1.old/configure mozilla-1.5.1.new/configure
--- mozilla-1.5.1.old/configure	Wed Oct  8 02:32:34 2003
+++ mozilla-1.5.1.new/configure	Mon Dec 15 19:55:03 2003
@@ -4707,7 +4707,7 @@
     ;;
 
 *-darwin*)
-    HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX -DXP_MACOSX -DNO_X11"
+    HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX"
     HOST_NSPR_MDCPUCFG='\"md/_darwin.cfg\"'
     HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-O3}"
     ;;
@@ -4915,8 +4915,8 @@
 
 *-darwin*) 
     _PEDANTIC=
-    CFLAGS="$CFLAGS -fpascal-strings -no-cpp-precomp -fno-common"
-    CXXFLAGS="$CXXFLAGS -fpascal-strings -no-cpp-precomp -fno-common"
+    CFLAGS="$CFLAGS -no-cpp-precomp -fno-common"
+    CXXFLAGS="$CXXFLAGS -no-cpp-precomp -fno-common"
     DLL_SUFFIX=".dylib"
     DSO_LDOPTS=''
     STRIP="$STRIP -x -S"
diff -Naur mozilla-1.5.1.old/directory/c-sdk/config/config.mk mozilla-1.5.1.new/directory/c-sdk/config/config.mk
--- mozilla-1.5.1.old/directory/c-sdk/config/config.mk	Mon Feb 24 00:54:18 2003
+++ mozilla-1.5.1.new/directory/c-sdk/config/config.mk	Mon Dec 15 19:55:03 2003
@@ -60,10 +60,10 @@
 
 NFSPWD		= $(MOD_DEPTH)/config/nfspwd
 
-CFLAGS		= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(OS_CFLAGS)\
-		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
-CCCFLAGS	= $(CCC_ONLY_FLAGS) $(OPTIMIZER) $(OS_CFLAGS)\
-		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
+CFLAGS		= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(XP_DEFINE)\
+		  $(DEFINES) $(INCLUDES) $(XCFLAGS) $(OS_CFLAGS)
+CCCFLAGS	= $(CCC_ONLY_FLAGS) $(OPTIMIZER) $(XP_DEFINE)\
+		  $(DEFINES) $(INCLUDES) $(XCFLAGS) $(OS_CFLAGS)
 # For purify
 NOMD_CFLAGS	= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(NOMD_OS_CFLAGS)\
 		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
diff -Naur mozilla-1.5.1.old/directory/c-sdk/configure mozilla-1.5.1.new/directory/c-sdk/configure
--- mozilla-1.5.1.old/directory/c-sdk/configure	Sat Apr 19 10:29:36 2003
+++ mozilla-1.5.1.new/directory/c-sdk/configure	Mon Dec 15 19:55:03 2003
@@ -3296,7 +3296,7 @@
 
         CPU_ARCH=ppc
     fi
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @PREFIX@/lib/mozilla/$@'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS) -o $@'
diff -Naur mozilla-1.5.1.old/fink/Applications/mozilla-editor.desktop mozilla-1.5.1.new/fink/Applications/mozilla-editor.desktop
--- mozilla-1.5.1.old/fink/Applications/mozilla-editor.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.5.1.new/fink/Applications/mozilla-editor.desktop	Mon Dec 15 19:55:03 2003
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Editor
+Comment=HTML Composer
+Exec=mozilla -edit
+Icon=gnome-word.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.5.1.old/fink/Internet/mozilla-address.desktop mozilla-1.5.1.new/fink/Internet/mozilla-address.desktop
--- mozilla-1.5.1.old/fink/Internet/mozilla-address.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.5.1.new/fink/Internet/mozilla-address.desktop	Mon Dec 15 19:55:03 2003
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Addressbook
+Comment=Addressbook from Mozilla Mailnews
+Exec=mozilla -addressbook
+Icon=gnome-calendar.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.5.1.old/fink/Internet/mozilla-chatzilla.desktop mozilla-1.5.1.new/fink/Internet/mozilla-chatzilla.desktop
--- mozilla-1.5.1.old/fink/Internet/mozilla-chatzilla.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.5.1.new/fink/Internet/mozilla-chatzilla.desktop	Mon Dec 15 19:55:03 2003
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Chatzilla
+Comment=IRC client
+Exec=mozilla -chat
+Icon=gnome-irc.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.5.1.old/fink/Internet/mozilla-mail-composer.desktop mozilla-1.5.1.new/fink/Internet/mozilla-mail-composer.desktop
--- mozilla-1.5.1.old/fink/Internet/mozilla-mail-composer.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.5.1.new/fink/Internet/mozilla-mail-composer.desktop	Mon Dec 15 19:55:03 2003
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Mail Composer
+Comment=Mozilla Mail Composer
+Exec=mozilla -compose
+Icon=mozilla-mail-composer.gif
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.5.1.old/fink/Internet/mozilla-mail.desktop mozilla-1.5.1.new/fink/Internet/mozilla-mail.desktop
--- mozilla-1.5.1.old/fink/Internet/mozilla-mail.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.5.1.new/fink/Internet/mozilla-mail.desktop	Mon Dec 15 19:55:03 2003
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla Mail
+Comment=Mozilla Mail
+Exec=mozilla -mail
+Icon=mozilla-mail.gif
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.5.1.old/fink/Internet/mozilla-news.desktop mozilla-1.5.1.new/fink/Internet/mozilla-news.desktop
--- mozilla-1.5.1.old/fink/Internet/mozilla-news.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.5.1.new/fink/Internet/mozilla-news.desktop	Mon Dec 15 19:55:03 2003
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla News
+Comment=Mozilla Mail
+Exec=mozilla -news
+Icon=gnome-news.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.5.1.old/fink/Internet/mozilla.desktop mozilla-1.5.1.new/fink/Internet/mozilla.desktop
--- mozilla-1.5.1.old/fink/Internet/mozilla.desktop	Thu Jan  1 09:00:00 1970
+++ mozilla-1.5.1.new/fink/Internet/mozilla.desktop	Mon Dec 15 19:55:03 2003
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Name=Mozilla
+Comment=Mozilla Web Browser
+Exec=mozilla 
+Icon=mozilla.png
+Terminal=false
+MultipleArgs=false
+Type=Application
diff -Naur mozilla-1.5.1.old/fink/chrome.d/default mozilla-1.5.1.new/fink/chrome.d/default
--- mozilla-1.5.1.old/fink/chrome.d/default	Thu Jan  1 09:00:00 1970
+++ mozilla-1.5.1.new/fink/chrome.d/default	Mon Dec 15 19:55:03 2003
@@ -0,0 +1,2 @@
+skin,install,select,modern/1.0
+locale,install,select,en-US
diff -Naur mozilla-1.5.1.old/fink/mozilla mozilla-1.5.1.new/fink/mozilla
--- mozilla-1.5.1.old/fink/mozilla	Thu Jan  1 09:00:00 1970
+++ mozilla-1.5.1.new/fink/mozilla	Mon Dec 15 19:55:03 2003
@@ -0,0 +1,99 @@
+#!/bin/sh
+#
+# The contents of this file are subject to the Netscape Public
+# License Version 1.1 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of
+# the License at http://www.mozilla.org/NPL/
+#
+# Software distributed under the License is distributed on an "AS
+# IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# rights and limitations under the License.
+#
+# The Original Code is mozilla.org code.
+#
+# The Initial Developer of the Original Code is Netscape
+# Communications Corporation.  Portions created by Netscape are
+# Copyright (C) 1998 Netscape Communications Corporation. All
+# Rights Reserved.
+#
+# Contributor(s): 
+#
+
+## 
+## Usage:
+##
+## $ mozilla
+##
+## This script is meant to run a mozilla program from the mozilla
+## rpm installation.
+##
+## The script will setup all the environment voodoo needed to make
+## mozilla work.
+
+##
+## Variables
+##
+MOZ_PROGRAM="@PREFIX@/lib/mozilla/mozilla-bin"
+
+##
+## Set MOZILLA_FIVE_HOME
+##
+MOZILLA_FIVE_HOME="@PREFIX@/lib/mozilla"
+export MOZILLA_FIVE_HOME
+
+##
+## Set DYLD_LIBRARY_PATH
+##
+DYLD_LIBRARY_PATH=@PREFIX@/lib/mozilla:$DYLD_LIBRARY_PATH
+export DYLD_LIBRARY_PATH
+
+# If there is no command line argument at all then try to open a new
+# window in an already running instance.
+
+if [ -z "$1" ]; then
+  $MOZ_PROGRAM -remote "openurl(about:blank,new-window)" >/dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "2" ]; then
+    $MOZ_PROGRAM ${1+"$@"}
+    RETURN_VAL=$?
+  fi
+  exit $RETURN_VAL
+fi
+
+# If there's a command line argument but it doesn't begin with a -
+# it's probably a url.  Try to send it to a running instance.
+
+USE_EXIST=0
+opt="$1"
+case "$opt" in
+  -*) ;;
+  *) USE_EXIST=1 ;;
+esac
+
+if [ "${USE_EXIST}" -eq "1" ]; then
+  # check to make sure that the command contains at least a :/ in it.
+  echo $opt | grep -e ':/' 2>/dev/null >/dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "1" ]; then
+    # does it begin with a / ?
+    echo $opt | grep -e '^/' >/dev/null
+    RETURN_VAL=$?
+    if [ "$RETURN_VAL" -eq "0" ]; then
+      opt="file:$opt"
+    elif [ -e `pwd`/$opt ]; then
+      opt="`pwd`/$opt"
+    else
+      opt="http://$opt"
+    fi
+  fi
+  $MOZ_PROGRAM -remote "openurl($opt,new-window)" >/dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "2" ]; then
+    $MOZ_PROGRAM ${1+"$@"}
+    RETURN_VAL=$?
+  fi
+  exit $RETURN_VAL
+fi
+
+$MOZ_PROGRAM ${1+"$@"}
diff -Naur mozilla-1.5.1.old/fink/prefs.js mozilla-1.5.1.new/fink/prefs.js
--- mozilla-1.5.1.old/fink/prefs.js	Thu Jan  1 09:00:00 1970
+++ mozilla-1.5.1.new/fink/prefs.js	Mon Dec 15 19:55:03 2003
@@ -0,0 +1,23 @@
+// XIM style
+pref("xim.input_style", "over-the-spot");
+
+// TryeType
+pref("font.FreeType2.enable", true);
+pref("font.freetype2.shared-library", "@PREFIX@/lib/libfreetype.6.dylib");
+pref("font.FreeType2.autohinted", true);
+pref("font.FreeType2.unhinted", true);
+pref("font.antialias.min",        10);
+pref("font.embedded_bitmaps.max", 1000000);
+pref("font.scale.tt_bitmap.dark_text.min", 64);
+pref("font.scale.tt_bitmap.dark_text.gain", "0.0");
+pref("font.directory.truetype.1", "/Library/Fonts");
+pref("font.directory.truetype.2", "/Network/Library/Fonts");
+pref("font.directory.truetype.3", "/System/Library/Fonts");
+pref("font.directory.truetype.4", "/usr/X11R6/lib/X11/fonts/TTF");
+pref("font.directory.truetype.5", "@PREFIX@/share/fonts/truetype");
+
+// AA with Bitmap scaling.
+pref("font.scale.aa_bitmap.enable", true);
+pref("font.scale.aa_bitmap.always", false);
+pref("font.scale.aa_bitmap.min", 6);
+
diff -Naur mozilla-1.5.1.old/fink/update-mozilla-chrome mozilla-1.5.1.new/fink/update-mozilla-chrome
--- mozilla-1.5.1.old/fink/update-mozilla-chrome	Thu Jan  1 09:00:00 1970
+++ mozilla-1.5.1.new/fink/update-mozilla-chrome	Tue Dec 16 01:25:59 2003
@@ -0,0 +1,24 @@
+#!/bin/sh -e
+
+ulimit -s unlimited || true
+
+MOZILLA_FIVE_HOME=@PREFIX@/lib/mozilla
+CHROME_DIR=@PREFIX@/var/lib/mozilla/chrome.d
+
+export DYLD_LIBRARY_PATH=$MOZILLA_FIVE_HOME
+
+TMPFILE=`tempfile`
+cat ${CHROME_DIR}/* >>${TMPFILE}
+mv ${TMPFILE} ${MOZILLA_FIVE_HOME}/chrome/installed-chrome.txt
+chmod 644 ${MOZILLA_FIVE_HOME}/chrome/installed-chrome.txt
+
+rm -rf ${MOZILLA_FIVE_HOME}/chrome/overlayinfo
+rm -f  ${MOZILLA_FIVE_HOME}/chrome/*.rdf
+rm -f  ${MOZILLA_FIVE_HOME}/component.reg
+rm -f  ${MOZILLA_FIVE_HOME}/components/*.dat
+
+sudo -H ${MOZILLA_FIVE_HOME}/regxpcom || true
+sudo -H ${MOZILLA_FIVE_HOME}/regchrome || true
+
+exit 0
+
diff -Naur mozilla-1.5.1.old/modules/libpref/src/nsPrefService.cpp mozilla-1.5.1.new/modules/libpref/src/nsPrefService.cpp
--- mozilla-1.5.1.old/modules/libpref/src/nsPrefService.cpp	Mon Jul 14 16:37:28 2003
+++ mozilla-1.5.1.new/modules/libpref/src/nsPrefService.cpp	Mon Dec 15 19:55:04 2003
@@ -722,6 +722,7 @@
 #elif defined(XP_BEOS)
       "beos.js"
 #endif
+      , "fink.js"
   };
 
   rv = NS_GetSpecialDirectory(NS_APP_PREF_DEFAULTS_50_DIR, getter_AddRefs(defaultPrefDir));
diff -Naur mozilla-1.5.1.old/nsprpub/configure mozilla-1.5.1.new/nsprpub/configure
--- mozilla-1.5.1.old/nsprpub/configure	Sat Aug 30 09:18:12 2003
+++ mozilla-1.5.1.new/nsprpub/configure	Mon Dec 15 19:55:04 2003
@@ -3209,7 +3209,7 @@
 
         CPU_ARCH=ppc
     fi
-    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @executable_path/$@'
+    DSO_LDOPTS='-dynamiclib -compatibility_version 1 -current_version 1 -all_load -install_name @PREFIX@/lib/mozilla/$@'
     # Use the standard preprocessor (cpp)
     CFLAGS="$CFLAGS -no-cpp-precomp"
     MKSHLIB='$(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS) -o $@'
@@ -3224,9 +3224,6 @@
 
     # Add Mac OS X support for loading CFM & CFBundle plugins
     if test -f /System/Library/Frameworks/Carbon.framework/Carbon; then
-        cat >> confdefs.h <<\EOF
-#define XP_MACOSX 1
-EOF
 
         OS_TARGET=MacOSX
 
@@ -4723,8 +4720,6 @@
 
 
 case $target in
-*-darwin*)
-    ;;
 *)
     echo $ac_n "checking for dlopen in -ldl""... $ac_c" 1>&6
 echo "configure:4731: checking for dlopen in -ldl" >&5
diff -Naur mozilla-1.5.1.old/nsprpub/pr/include/md/_darwin.h mozilla-1.5.1.new/nsprpub/pr/include/md/_darwin.h
--- mozilla-1.5.1.old/nsprpub/pr/include/md/_darwin.h	Sun Feb 23 00:08:27 2003
+++ mozilla-1.5.1.new/nsprpub/pr/include/md/_darwin.h	Mon Dec 15 19:55:04 2003
@@ -55,7 +55,8 @@
 
 #undef  HAVE_STACK_GROWING_UP
 #define HAVE_DLL
-#define USE_MACH_DYLD
+/* #define USE_MACH_DYLD */
+#define USE_DLFCN
 #define _PR_HAVE_SOCKADDR_LEN  
 #define _PR_STAT_HAS_ST_ATIMESPEC
 #define _PR_NO_LARGE_FILES
diff -Naur mozilla-1.5.1.old/nsprpub/pr/src/Makefile.in mozilla-1.5.1.new/nsprpub/pr/src/Makefile.in
--- mozilla-1.5.1.old/nsprpub/pr/src/Makefile.in	Thu Apr 10 06:49:39 2003
+++ mozilla-1.5.1.new/nsprpub/pr/src/Makefile.in	Mon Dec 15 19:55:04 2003
@@ -198,9 +198,9 @@
 endif
 endif
 
-ifeq ($(OS_TARGET),MacOSX)
-OS_LIBS		= -framework CoreServices -framework CoreFoundation
-endif
+#ifeq ($(OS_TARGET),MacOSX)
+#OS_LIBS		= -framework CoreServices -framework CoreFoundation
+#endif
 
 ifdef GC_LEAK_DETECTOR
 EXTRA_LIBS	= -L$(dist_libdir) -lboehm
diff -Naur mozilla-1.5.1.old/nsprpub/pr/src/linking/prlink.c mozilla-1.5.1.new/nsprpub/pr/src/linking/prlink.c
--- mozilla-1.5.1.old/nsprpub/pr/src/linking/prlink.c	Wed Mar 26 10:36:09 2003
+++ mozilla-1.5.1.new/nsprpub/pr/src/linking/prlink.c	Mon Dec 15 20:02:07 2003
@@ -164,7 +164,8 @@
 #if defined(SUNOS4) || defined(DARWIN) || defined(NEXTSTEP) \
     || defined(OPENBSD) || defined(WIN16) \
     || (defined(NETBSD) && !defined(__ELF__))
-#define NEED_LEADING_UNDERSCORE
+/* On MacOSX 10.3, dlsym does not accept symbols with '_'.
+/* #define NEED_LEADING_UNDERSCORE */
 #endif
 
 #ifdef XP_PC
diff -Naur mozilla-1.5.1.old/security/coreconf/Darwin.mk mozilla-1.5.1.new/security/coreconf/Darwin.mk
--- mozilla-1.5.1.old/security/coreconf/Darwin.mk	Fri Feb  7 15:57:57 2003
+++ mozilla-1.5.1.new/security/coreconf/Darwin.mk	Mon Dec 15 19:55:04 2003
@@ -59,7 +59,7 @@
 # definitions so that the linker can catch multiply-defined symbols.
 # Also, common symbols are not allowed with Darwin dynamic libraries.
 
-OS_CFLAGS	= $(DSO_CFLAGS) $(OS_REL_CFLAGS) -Wmost -fpascal-strings -traditional-cpp -fno-common -pipe -DDARWIN -DHAVE_STRERROR -DHAVE_BSD_FLOCK
+OS_CFLAGS	= $(DSO_CFLAGS) $(OS_REL_CFLAGS) -Wmost -fpascal-strings -no-cpp-precomp -fno-common -pipe -DDARWIN -DHAVE_STRERROR -DHAVE_BSD_FLOCK
 
 ifdef BUILD_OPT
 OPTIMIZER	= -O2
@@ -68,7 +68,7 @@
 ARCH		= darwin
 
 # May override this with -bundle to create a loadable module.
-DSO_LDOPTS	= -dynamiclib -compatibility_version 1 -current_version 1 -install_name @executable_path/$(notdir $@)
+DSO_LDOPTS	= -dynamiclib -compatibility_version 1 -current_version 1 -install_name @PREFIX@/lib/mozilla/$(notdir $@)
 
 MKSHLIB		= $(CC) -arch $(CPU_ARCH) $(DSO_LDOPTS)
 DLL_SUFFIX	= dylib
diff -Naur mozilla-1.5.1.old/security/nss/cmd/platlibs.mk mozilla-1.5.1.new/security/nss/cmd/platlibs.mk
--- mozilla-1.5.1.old/security/nss/cmd/platlibs.mk	Tue Apr 22 08:01:38 2003
+++ mozilla-1.5.1.new/security/nss/cmd/platlibs.mk	Mon Dec 15 19:55:04 2003
@@ -195,7 +195,7 @@
 endif
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file @PREFIX@/lib/mozilla/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 
diff -Naur mozilla-1.5.1.old/security/nss/lib/smime/config.mk mozilla-1.5.1.new/security/nss/lib/smime/config.mk
--- mozilla-1.5.1.old/security/nss/lib/smime/config.mk	Tue Apr 22 08:02:08 2003
+++ mozilla-1.5.1.new/security/nss/lib/smime/config.mk	Mon Dec 15 19:55:04 2003
@@ -70,7 +70,7 @@
 	$(NULL)
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file @PREFIX@/lib/mozilla/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 endif
diff -Naur mozilla-1.5.1.old/security/nss/lib/ssl/config.mk mozilla-1.5.1.new/security/nss/lib/ssl/config.mk
--- mozilla-1.5.1.old/security/nss/lib/ssl/config.mk	Tue Apr 22 08:02:14 2003
+++ mozilla-1.5.1.new/security/nss/lib/ssl/config.mk	Mon Dec 15 19:55:04 2003
@@ -75,7 +75,7 @@
 endif
 
 ifeq ($(OS_ARCH), Darwin)
-EXTRA_SHARED_LIBS += -dylib_file @executable_path/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
+EXTRA_SHARED_LIBS += -dylib_file @PREFIX@/lib/mozilla/libsoftokn3.dylib:$(DIST)/lib/libsoftokn3.dylib
 endif
 
 endif
diff -Naur mozilla-1.5.1.old/themes/Makefile.in mozilla-1.5.1.new/themes/Makefile.in
--- mozilla-1.5.1.old/themes/Makefile.in	Thu Nov 21 16:10:01 2002
+++ mozilla-1.5.1.new/themes/Makefile.in	Mon Dec 15 19:55:04 2003
@@ -36,7 +36,9 @@
 
 # select classic as the default skin
 libs::
-	echo skin,install,select,classic/1.0 >> $(DIST)/bin/chrome/installed-chrome.txt
+# fix lator
+#	echo skin,install,select,classic/1.0 >> $(DIST)/bin/chrome/installed-chrome.txt
 
 install::
-	echo skin,install,select,classic/1.0 >> $(DESTDIR)$(mozappdir)/chrome/installed-chrome.txt
+# fix lator
+#	echo skin,install,select,classic/1.0 >> $(DESTDIR)$(mozappdir)/chrome/installed-chrome.txt
