--- openldap-2.1.22/build/ltmain.sh	2003-02-11 06:24:24.000000000 +1100
+++ openldap-2.1.22-patched/build/ltmain.sh	2003-07-02 20:56:26.000000000 +1000
@@ -1763,7 +1763,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    need_relink=no
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then

--- openldap-2.1.22/build/top.mk	2003-03-30 02:45:42.000000000 +1100
+++ openldap-2.1.22-patched/build/top.mk	2003-07-02 20:59:07.000000000 +1000
@@ -197,7 +197,7 @@
 DEFS = $(LDAP_INCPATH) $(XINCPATH) $(XDEFS) $(AC_DEFS) $(DEFINES)
 CFLAGS = $(AC_CFLAGS) $(DEFS)
 LDFLAGS = $(LDAP_LIBPATH) $(AC_LDFLAGS) $(XLDFLAGS)
-LIBS = $(XLIBS) $(XXLIBS) $(AC_LIBS) $(XXXLIBS)
+LIBS = $(XLIBS) $(XXLIBS) $(XXXLIBS) $(AC_LIBS)
 
 LT_CFLAGS = $(AC_CFLAGS)
 LT_CPPFLAGS = $(DEFS)

--- openldap-2.1.22/libraries/libldap/Makefile.in	2003-03-31 00:47:09.000000000 +1000
+++ openldap-2.1.22-patched/libraries/libldap/Makefile.in	2003-07-02 21:01:05.000000000 +1000
@@ -35,7 +35,7 @@
 XLIBS = $(LIBRARY) $(LDAP_LIBLBER_LA) $(LDAP_LIBLUTIL_A)
 XXLIBS = $(SECURITY_LIBS) $(LUTIL_LIBS)
 NT_LINK_LIBS = $(LDAP_LIBLBER_LA) $(AC_LIBS) $(SECURITY_LIBS)
-UNIX_LINK_LIBS = $(LDAP_LIBLBER_LA) $(AC_LIBS) $(SECURITY_LIBS)
+UNIX_LINK_LIBS = $(LDAP_LIBLBER_LA) $(SECURITY_LIBS) $(AC_LIBS)
 
 apitest:	$(XLIBS) apitest.o
 	$(LTLINK) -o $@ apitest.o $(LIBS)

--- openldap-2.1.22/libraries/libldap_r/Makefile.in	2003-03-31 00:47:09.000000000 +1000
+++ openldap-2.1.22-patched/libraries/libldap_r/Makefile.in	2003-07-02 21:02:27.000000000 +1000
@@ -45,7 +45,7 @@
 XXLIBS = $(SECURITY_LIBS) $(LUTIL_LIBS)
 XXXLIBS = $(LTHREAD_LIBS)
 NT_LINK_LIBS = $(LDAP_LIBLBER_LA) $(AC_LIBS) $(SECURITY_LIBS)
-UNIX_LINK_LIBS = $(LDAP_LIBLBER_LA) $(AC_LIBS) $(SECURITY_LIBS)
+UNIX_LINK_LIBS = $(LDAP_LIBLBER_LA) $(SECURITY_LIBS) $(AC_LIBS)
 
 .links : Makefile
 	@for i in $(XXSRCS); do \
