diff -Nurd -x'*~' gelemental-1.2.0.orig/configure gelemental-1.2.0/configure
--- gelemental-1.2.0.orig/configure	2007-09-25 00:46:29.000000000 -0400
+++ gelemental-1.2.0/configure	2013-08-19 04:16:29.000000000 -0400
@@ -20424,7 +20424,7 @@
 
 
 
-GETTEXT_PACKAGE=gelemental
+GETTEXT_PACKAGE=gelemental0
 
 
 cat >>confdefs.h <<_ACEOF
diff -Nurd -x'*~' gelemental-1.2.0.orig/data/libelemental.pc.in gelemental-1.2.0/data/libelemental.pc.in
--- gelemental-1.2.0.orig/data/libelemental.pc.in	2007-09-25 00:56:13.000000000 -0400
+++ gelemental-1.2.0/data/libelemental.pc.in	2013-08-19 04:21:00.000000000 -0400
@@ -7,6 +7,6 @@
 Description: A periodic table library with detailed information on elements.
 URL: http://www.kdau.com/projects/gelemental/
 Version: @VERSION@
-Requires: glibmm-2.4 >= 2.6.0 pango
+Requires: glibmm-2.4 >= 2.6.0
 Libs: -L${libdir} -lelemental
 Cflags: -I${includedir}
