diff -Nurd -x'*~' gnome-netstatus-2.28.1.orig/src/netstatus-sysdeps.c gnome-netstatus-2.28.1/src/netstatus-sysdeps.c
--- gnome-netstatus-2.28.1.orig/src/netstatus-sysdeps.c	2009-04-19 14:24:53.000000000 -0400
+++ gnome-netstatus-2.28.1/src/netstatus-sysdeps.c	2010-04-26 03:34:45.000000000 -0400
@@ -35,14 +35,17 @@
 #include <glib.h>
 #include <glib/gi18n.h>
 
-#ifdef __FreeBSD__
+#if defined (__FreeBSD__) || defined (__APPLE__)
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
 #include <net/if.h>
 #include <net/if_var.h>
+#ifdef __FREEBSD__
 #include <dev/an/if_aironet_ieee.h>
 #include <dev/wi/if_wavelan_ieee.h>
+#define HAVE_AIRONET
+#endif
 #endif
 
 static inline gboolean
@@ -78,7 +81,7 @@
   return TRUE;
 }
 
-#if !defined (__FreeBSD__)
+#if !defined (__FreeBSD__) && !defined (__APPLE__)
 
 static inline char *
 parse_iface_name (const char *buf)
@@ -430,13 +433,16 @@
 get_an_data (const char *iface,
 	     int        *signal_strength)
 {
+#ifdef HAVE_AIRONET
 #ifdef AN_RID_RSSI_MAP
   struct an_ltv_rssi_map  an_rssimap;
 #endif
   struct an_req         areq;
   struct an_ltv_status *sts;
   int                   level;
+#endif /* HAVE_AIRONET */
   char                 *error = NULL;
+#ifdef HAVE_AIRONET
   gboolean              rssimap_valid = FALSE;
 
 #ifdef AN_RID_RSSI_MAP
@@ -464,6 +470,7 @@
 
   memcpy (signal_strength, &level, sizeof (signal_strength));
 
+#endif /* HAVE_AIRONET */
   return error;
 }
 
@@ -471,9 +478,12 @@
 get_wi_data (const char *iface,
 	     int        *signal_strength)
 {
+#ifdef HAVE_AIRONET
   struct wi_req  wreq;
   int            level;
+#endif /* HAVE_AIRONET */
   char          *error = NULL;
+#ifdef HAVE_AIRONET
 
   memset (&wreq, 0, sizeof (wreq));
 
@@ -525,6 +535,7 @@
 
   memcpy (signal_strength, &level, sizeof (signal_strength));
 
+#endif /* HAVE_AIRONET */
   return error;
 }
 
@@ -592,7 +603,7 @@
   *out_bytes   = -1;
 
   error = NULL;
-  command_line = g_strdup_printf ("/usr/bin/netstat -n -I %s -b -f inet", iface);
+  command_line = g_strdup_printf ("/usr/sbin/netstat -n -I %s -b -f inet", iface);
   if (!g_shell_parse_argv (command_line, NULL, &argv, &error))
     {
       error_message = g_strdup_printf (_("Could not parse command line '%s': %s"),
@@ -660,7 +671,7 @@
     }
   else
     {
-      error_message = g_strdup_printf ("Error running /usr/bin/netstat for '%s': %s", 
+      error_message = g_strdup_printf ("Error running /usr/sbin/netstat for '%s': %s", 
 				       iface, error->message);
       g_error_free (error);
     }
