diff -Nurd -x'*~' gtkhtml-3.28.3.orig/components/editor/Makefile.in gtkhtml-3.28.3/components/editor/Makefile.in
--- gtkhtml-3.28.3.orig/components/editor/Makefile.in	2010-03-01 04:16:50.000000000 -0500
+++ gtkhtml-3.28.3/components/editor/Makefile.in	2010-03-09 19:11:53.000000000 -0500
@@ -393,7 +393,7 @@
 
 libgtkhtml_editor_la_LIBADD = \
 	@GTKHTML_LIBS@				\
-	$(top_builddir)/gtkhtml/libgtkhtml-@GTKHTML_API_VERSION@.la
+	$(top_builddir)/gtkhtml/libgtkhtml-@GTKHTML_API_VERSION@.la $(INTLLIBS)
 
 libgtkhtml_editor_la_LDFLAGS = -no-undefined
 gtkhtml_editor_test_SOURCES = main.c
diff -Nurd -x'*~' gtkhtml-3.28.3.orig/gtkhtml/Makefile.in gtkhtml-3.28.3/gtkhtml/Makefile.in
--- gtkhtml-3.28.3.orig/gtkhtml/Makefile.in	2010-03-01 04:16:50.000000000 -0500
+++ gtkhtml-3.28.3/gtkhtml/Makefile.in	2010-03-09 19:00:04.000000000 -0500
@@ -609,7 +609,7 @@
 gtest_LDFLAGS = 
 gtest_LDADD = \
 	libgtkhtml-@GTKHTML_API_VERSION@.la			\
-	$(GTKHTML_LIBS)
+	$(GTKHTML_LIBS) $(INTLLIBS)
 
 test_suite_SOURCES = \
 	test-suite.c
