diff -ruN kdoctools-5.4.0-orig/cmake/FindDocBookXML4.cmake kdoctools-5.4.0/cmake/FindDocBookXML4.cmake
--- kdoctools-5.4.0-orig/cmake/FindDocBookXML4.cmake	2014-11-02 06:10:49.000000000 -0600
+++ kdoctools-5.4.0/cmake/FindDocBookXML4.cmake	2014-11-21 06:40:52.000000000 -0600
@@ -16,7 +16,7 @@
 
 
 if (NOT DocBookXML_FIND_VERSION)
-     set(DocBookXML_FIND_VERSION "4.5")
+     set(DocBookXML_FIND_VERSION "4.5.0")
 endif ()
 
 set (DocBookXML4_DTD_VERSION ${DocBookXML_FIND_VERSION}
@@ -30,6 +30,7 @@
 function (locate_version version found_dir)
 
     set (DTD_PATH_LIST
+        share/xml/dtd/docbookx/${version}
         share/xml/docbook/schema/dtd/${version}
         share/xml/docbook/xml-dtd-${version}
         share/sgml/docbook/xml-dtd-${version}
