diff -ruNp kstars-4.13.1-orig/kstars/skycomponents/asteroidscomponent.cpp kstars-4.13.1/kstars/skycomponents/asteroidscomponent.cpp
--- kstars-4.13.1-orig/kstars/skycomponents/asteroidscomponent.cpp	2014-03-16 03:35:18.000000000 -0500
+++ kstars-4.13.1/kstars/skycomponents/asteroidscomponent.cpp	2014-06-05 06:09:38.000000000 -0500
@@ -194,7 +194,7 @@ void AsteroidsComponent::draw( SkyPainte
         // FIXME: God help us!
         KSAsteroid *ast = (KSAsteroid*) so;
 
-        if ( ast->mag() > Options::magLimitAsteroid() || isnan(ast->mag()) != 0)
+        if ( ast->mag() > Options::magLimitAsteroid() || std::isnan(ast->mag()) != 0)
             continue;
 
         bool drawn = skyp->drawPointSource(ast,ast->mag());
diff -ruNp kstars-4.13.1-orig/kstars/skycomponents/cometscomponent.cpp kstars-4.13.1/kstars/skycomponents/cometscomponent.cpp
--- kstars-4.13.1-orig/kstars/skycomponents/cometscomponent.cpp	2014-03-16 03:35:18.000000000 -0500
+++ kstars-4.13.1/kstars/skycomponents/cometscomponent.cpp	2014-06-05 06:05:39.000000000 -0500
@@ -191,7 +191,7 @@ void CometsComponent::draw( SkyPainter *
     foreach ( SkyObject *so, m_ObjectList ) {
         KSComet *com = (KSComet*)so;
         double mag= com->mag();
-        if (isnan(mag) == 0)
+        if (std::isnan(mag) == 0)
         {
             bool drawn = skyp->drawPointSource(com,mag);
             if ( drawn && !(hideLabels || com->rsun() >= rsunLabelLimit) )
diff -ruNp kstars-4.12.2-orig/kstars/skyobjects/ksmoon.cpp kstars-4.12.2/kstars/skyobjects/ksmoon.cpp
--- kstars-4.12.2-orig/kstars/skyobjects/ksmoon.cpp	2014-01-05 06:50:17.000000000 -0500
+++ kstars-4.12.2/kstars/skyobjects/ksmoon.cpp	2014-02-18 18:26:24.000000000 -0500
@@ -227,7 +227,7 @@ void KSMoon::findMagnitude(const KSNumbe
     // relevant data put into ksplanetbase.h) was taken from
     // SkyChart v3 Beta
     double phd = phase().Degrees();
-    if( isnan( phd ) ) // Avoid nanny phases.
+    if( std::isnan( phd ) ) // Avoid nanny phases.
         return;
     int p = floor( phd );
     if( p > 180 )
diff -ruNp kstars-4.12.2-orig/kstars/skyobjects/starobject.cpp kstars-4.12.2/kstars/skyobjects/starobject.cpp
--- kstars-4.12.2-orig/kstars/skyobjects/starobject.cpp	2014-01-05 06:50:17.000000000 -0500
+++ kstars-4.12.2/kstars/skyobjects/starobject.cpp	2014-02-18 18:25:51.000000000 -0500
@@ -282,7 +282,7 @@ void StarObject::getIndexCoords( KSNumbe
 
     pmms = pmMagnitudeSquared();
 
-    if( isnan( pmms ) || pmms * num->julianMillenia() * num->julianMillenia() < 1. ) {
+    if( std::isnan( pmms ) || pmms * num->julianMillenia() * num->julianMillenia() < 1. ) {
         // Ignore corrections
         *ra = ra0().Degrees();
         *dec = dec0().Degrees();
