diff -ruNp analitza-4.13.1-orig/analitzaplot/plotter2d.cpp analitza-4.13.1/analitzaplot/plotter2d.cpp
--- analitza-4.13.1-orig/analitzaplot/plotter2d.cpp	2014-02-15 11:03:35.000000000 -0600
+++ analitza-4.13.1/analitzaplot/plotter2d.cpp	2014-06-09 15:57:57.000000000 -0500
@@ -373,10 +373,10 @@ void Plotter2D::drawCartesianTickLabels(
                 s = (i < 0) ? "-" : "";
                 
                 if (incbig)
-                    s += computeAngleLabelByStep(std::abs(i), bigstep);
+                    s += computeAngleLabelByStep(::abs(i), bigstep);
                 else
                 {
-                    const QPair<unsigned int, unsigned int> frac = simplifyFraction(std::abs(i), step);
+                    const QPair<unsigned int, unsigned int> frac = simplifyFraction(::abs(i), step);
                     
                     s += computeAngleLabelByFrac(frac.first, frac.second);
                 }                
diff -ruNp analitza-4.13.1-orig/analitzaplot/plotter3d.h analitza-4.13.1/analitzaplot/plotter3d.h
--- analitza-4.13.1-orig/analitzaplot/plotter3d.h	2014-02-15 11:03:35.000000000 -0600
+++ analitza-4.13.1/analitzaplot/plotter3d.h	2014-06-09 15:21:22.000000000 -0500
@@ -34,7 +34,7 @@
 #include <GL/glew.h>
 #endif
 
-#include <GL/glu.h>
+#include <OpenGL/glu.h>
 
 class QAbstractItemModel;
 class QPainter;
