diff -ruN kpeople-5.8.0-orig/src/declarative/CMakeLists.txt kpeople-5.8.0/src/declarative/CMakeLists.txt
--- kpeople-5.8.0-orig/src/declarative/CMakeLists.txt	2015-03-13 03:07:30.000000000 -0500
+++ kpeople-5.8.0/src/declarative/CMakeLists.txt	2015-03-19 16:05:54.000000000 -0500
@@ -1,4 +1,4 @@
-add_library(KF5PeopleDeclarative SHARED
+add_library(KF5PeopleDeclarative MODULE
                     declarativepersondata.cpp
                     personactionsmodel.cpp
                     peopleqmlplugin.cpp)
