diff -ruN kdeclarative-5.8.0-orig/CMakeLists.txt kdeclarative-5.8.0/CMakeLists.txt
--- kdeclarative-5.8.0-orig/CMakeLists.txt	2015-03-07 08:40:28.000000000 -0600
+++ kdeclarative-5.8.0/CMakeLists.txt	2015-03-18 07:19:30.000000000 -0500
@@ -60,7 +60,7 @@
     ki18n_install(po)
 endif()
 
-if (NOT WIN32)
+if (NOT APPLE)
 find_package(epoxy)
 set_package_properties(epoxy PROPERTIES DESCRIPTION "libepoxy"
                        URL "http://github.com/anholt/libepoxy"
diff -ruN kdeclarative-5.8.0-orig/src/qmlcontrols/draganddrop/CMakeLists.txt kdeclarative-5.8.0/src/qmlcontrols/draganddrop/CMakeLists.txt
--- kdeclarative-5.8.0-orig/src/qmlcontrols/draganddrop/CMakeLists.txt	2015-03-07 08:40:28.000000000 -0600
+++ kdeclarative-5.8.0/src/qmlcontrols/draganddrop/CMakeLists.txt	2015-03-18 07:19:13.000000000 -0500
@@ -9,7 +9,7 @@
     MimeDataWrapper.cpp
 )
 
-add_library(draganddropplugin SHARED ${declarativedragdrop_SRCS})
+add_library(draganddropplugin MODULE ${declarativedragdrop_SRCS})
 target_link_libraries(draganddropplugin
         Qt5::Core
         Qt5::Quick
diff -ruN kdeclarative-5.8.0-orig/src/qmlcontrols/kcoreaddons/CMakeLists.txt kdeclarative-5.8.0/src/qmlcontrols/kcoreaddons/CMakeLists.txt
--- kdeclarative-5.8.0-orig/src/qmlcontrols/kcoreaddons/CMakeLists.txt	2015-03-07 08:40:28.000000000 -0600
+++ kdeclarative-5.8.0/src/qmlcontrols/kcoreaddons/CMakeLists.txt	2015-03-18 07:19:13.000000000 -0500
@@ -6,7 +6,7 @@
     kuserproxy.cpp
     )
 
-add_library(kcoreaddonsplugin SHARED ${kcoreaddonsplugin_SRCS})
+add_library(kcoreaddonsplugin MODULE ${kcoreaddonsplugin_SRCS})
 target_link_libraries(kcoreaddonsplugin
         Qt5::Core
         Qt5::Quick
diff -ruN kdeclarative-5.8.0-orig/src/qmlcontrols/kioplugin/CMakeLists.txt kdeclarative-5.8.0/src/qmlcontrols/kioplugin/CMakeLists.txt
--- kdeclarative-5.8.0-orig/src/qmlcontrols/kioplugin/CMakeLists.txt	2015-03-07 08:40:28.000000000 -0600
+++ kdeclarative-5.8.0/src/qmlcontrols/kioplugin/CMakeLists.txt	2015-03-18 07:19:13.000000000 -0500
@@ -5,7 +5,7 @@
     krunproxy.cpp
     )
 
-add_library(kio SHARED ${kioplugin_SRCS})
+add_library(kio MODULE ${kioplugin_SRCS})
 target_link_libraries(kio
         Qt5::Core
         Qt5::Qml
diff -ruN kdeclarative-5.8.0-orig/src/qmlcontrols/kquickcontrols/private/CMakeLists.txt kdeclarative-5.8.0/src/qmlcontrols/kquickcontrols/private/CMakeLists.txt
--- kdeclarative-5.8.0-orig/src/qmlcontrols/kquickcontrols/private/CMakeLists.txt	2015-03-07 08:40:28.000000000 -0600
+++ kdeclarative-5.8.0/src/qmlcontrols/kquickcontrols/private/CMakeLists.txt	2015-03-18 07:19:13.000000000 -0500
@@ -6,7 +6,7 @@
     translationcontext.cpp
 )
 
-add_library(kquickcontrolsprivateplugin SHARED ${kquickcontrolsprivate_SRCS})
+add_library(kquickcontrolsprivateplugin MODULE ${kquickcontrolsprivate_SRCS})
 
 target_link_libraries(kquickcontrolsprivateplugin
         Qt5::Core
diff -ruN kdeclarative-5.8.0-orig/src/qmlcontrols/kquickcontrolsaddons/CMakeLists.txt kdeclarative-5.8.0/src/qmlcontrols/kquickcontrolsaddons/CMakeLists.txt
--- kdeclarative-5.8.0-orig/src/qmlcontrols/kquickcontrolsaddons/CMakeLists.txt	2015-03-07 08:40:28.000000000 -0600
+++ kdeclarative-5.8.0/src/qmlcontrols/kquickcontrolsaddons/CMakeLists.txt	2015-03-18 07:19:30.000000000 -0500
@@ -11,13 +11,13 @@
     mimedatabase.cpp
 )
 
-if (NOT WIN32)
+if (NOT APPLE)
     set(kquickcontrolsaddons_SRCS ${kquickcontrolsaddons_SRCS} plotter.cpp)
     set(KQUICKCONTROLSADDONS_EXTRA_LIBS ${epoxy_LIBRARY})
     include_directories(${epoxy_INCLUDE_DIR})
 endif()
 
-add_library(kquickcontrolsaddonsplugin SHARED ${kquickcontrolsaddons_SRCS})
+add_library(kquickcontrolsaddonsplugin MODULE ${kquickcontrolsaddons_SRCS})
 
 target_link_libraries(kquickcontrolsaddonsplugin
         Qt5::Core
diff -ruN kdeclarative-5.8.0-orig/src/qmlcontrols/kquickcontrolsaddons/kquickcontrolsaddonsplugin.cpp kdeclarative-5.8.0/src/qmlcontrols/kquickcontrolsaddons/kquickcontrolsaddonsplugin.cpp
--- kdeclarative-5.8.0-orig/src/qmlcontrols/kquickcontrolsaddons/kquickcontrolsaddonsplugin.cpp	2015-03-07 08:40:28.000000000 -0600
+++ kdeclarative-5.8.0/src/qmlcontrols/kquickcontrolsaddons/kquickcontrolsaddonsplugin.cpp	2015-03-18 07:19:34.000000000 -0500
@@ -32,7 +32,7 @@
 #include "clipboard.h"
 #include "mimedatabase.h"
 
-#ifndef Q_OS_WIN
+#ifndef Q_OS_MAC
 #include "plotter.h"
 #endif
 
@@ -49,7 +49,7 @@
     qmlRegisterType<MimeDatabase>(uri, 2, 0, "MimeDatabase");
 
 //TODO: make this work under Windows
-#ifndef Q_OS_WIN
+#ifndef Q_OS_MAC
     qmlRegisterType<PlotData>(uri, 2, 0, "PlotData");
     qmlRegisterType<Plotter>(uri, 2, 0, "Plotter");
 #endif
diff -ruN kdeclarative-5.8.0-orig/src/qmlcontrols/kwindowsystemplugin/CMakeLists.txt kdeclarative-5.8.0/src/qmlcontrols/kwindowsystemplugin/CMakeLists.txt
--- kdeclarative-5.8.0-orig/src/qmlcontrols/kwindowsystemplugin/CMakeLists.txt	2015-03-07 08:40:28.000000000 -0600
+++ kdeclarative-5.8.0/src/qmlcontrols/kwindowsystemplugin/CMakeLists.txt	2015-03-18 07:20:30.000000000 -0500
@@ -5,7 +5,7 @@
     kwindowsystemproxy.cpp
     )
 
-add_library(kwindowsystem SHARED ${kwindowsystemplugin_SRCS})
+add_library(kwindowsystem MODULE ${kwindowsystemplugin_SRCS})
 target_link_libraries(kwindowsystem
         Qt5::Core
         Qt5::Qml
