Index: admin/Makefile.common
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/Makefile.common,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
Index: admin/acinclude.m4.in
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/acinclude.m4.in,v
retrieving revision 1.1.1.3
diff -u -b -u -r1.1.1.3 acinclude.m4.in
--- admin/acinclude.m4.in	14 Aug 2002 13:18:45 -0000	1.1.1.3
+++ admin/acinclude.m4.in	25 Sep 2002 00:28:53 -0000
@@ -699,13 +699,10 @@
      AC_CHECK_LIB(ipc, shmat, X_EXTRA_LIBS="$X_EXTRA_LIBS -lipc"))
    
    # darwin needs this to initialize the environment
-   AC_CHECK_HEADERS(crt_externs.h)
+   AC_CHECK_HEADERS([crt_externs.h sys/types.h])
    AC_CHECK_FUNC(_NSGetEnviron, [AC_DEFINE(HAVE_NSGETENVIRON, 1, [Define if your system needs _NSGetEnviron to set up the environment])])
  
-   # more headers that need to be explicitly included on darwin
-   AC_CHECK_HEADERS(sys/types.h stdint.h)
-
-   # darwin requires a poll emulation library
+   # darwin requires a poll.h emulation library
    AC_CHECK_LIB(poll, poll, LIB_POLL="-lpoll")
 
    # CoreAudio framework
@@ -714,6 +711,12 @@
      FRAMEWORK_COREAUDIO="-framework CoreAudio"
    ])
 
+   # CoreServices framework
+   AC_CHECK_HEADER(CoreServices/CoreServices.h, [
+     AC_DEFINE(HAVE_CORESERVICES, 1, [Define if you have the CoreServices API])
+     FRAMEWORK_CORESERVICES="-framework CoreServices"
+   ])
+
    # Solaris 2.6 and others need -lresolv for res_init
    AC_CHECK_FUNCS(res_init, , [
      kde_libs_safe="$LIBS"
@@ -736,6 +739,8 @@
    AC_SUBST(FRAMEWORK_COREAUDIO)
    LIBSOCKET="$X_EXTRA_LIBS"
    AC_SUBST(LIBSOCKET)
+   AC_SUBST(FRAMEWORK_COREAUDIO)
+   AC_SUBST(LIB_POLL)
    AC_SUBST(LIBRESOLV)
    AC_SUBST(X_EXTRA_LIBS)
    AC_CHECK_LIB(ucb, killpg, [LIBUCB="-lucb"]) dnl for Solaris2.4
@@ -905,7 +910,7 @@
 kde_ldflags_safe="$LDFLAGS"
 kde_libs_safe="$LIBS"
 
-LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS"
+LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS"
 LIBS="-lXext -lX11 $LIBSOCKET"
 
 AC_TRY_LINK([
@@ -947,9 +952,9 @@
 kde_save_LDFLAGS="$LDFLAGS"
 kde_save_CFLAGS="$CFLAGS"
 kde_save_LIBS="$LIBS"
-LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS"
+LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS"
 CFLAGS="$CFLAGS -I$x_includes"
-LIBS="-lXinerama -lXext"
+LIBS="-lXinerama -lXext -lX11"
 
 if test "x$no_xinerama" = "xno"; then
 
@@ -1846,15 +1851,29 @@
    AC_REQUIRE([K_PATH_X])
 
 if test $kde_qtver = 3; then
+   AC_SUBST(LIB_ARTSKDE, "-lartskde")
+   AC_SUBST(LIB_DCOP, "-lDCOP")
+   AC_SUBST(LIB_KAB, "-lkab")
+   AC_SUBST(LIB_KABC, "-lkabc")
    AC_SUBST(LIB_KDECORE, "-lkdecore")
+   AC_SUBST(LIB_KDED, "@PREFIX@/lib/kded.la")
+   AC_SUBST(LIB_KDEFX, "-lkdefx")
+   AC_SUBST(LIB_KDEPRINT, "-lkdeprint")
+   AC_SUBST(LIB_KDESASL, "-lkdesasl")
+   AC_SUBST(LIB_KDESU, "-lkdesu")
    AC_SUBST(LIB_KDEUI, "-lkdeui")
-   AC_SUBST(LIB_KIO, "-lkio")
-   AC_SUBST(LIB_SMB, "-lsmb")
-   AC_SUBST(LIB_KAB, "-lkab")
    AC_SUBST(LIB_KHTML, "-lkhtml")
-   AC_SUBST(LIB_KSPELL, "-lkspell")
+   AC_SUBST(LIB_KIO, "-lkio")
    AC_SUBST(LIB_KPARTS, "-lkparts")
-   AC_SUBST(LIB_KDEPRINT, "-lkdeprint")
+   AC_SUBST(LIB_KJAVA, "-lkjava")
+   AC_SUBST(LIB_KJS, "-lkjs")
+   AC_SUBST(LIB_KMID, "-lkmid")
+   AC_SUBST(LIB_KSCRIPT, "-lkscript")
+   AC_SUBST(LIB_KSPELL, "-lkspell")
+   AC_SUBST(LIB_KTEXTEDITOR, "-lktexteditor")
+   AC_SUBST(LIB_SHELLSCRIPT, "-lshellscript")
+   AC_SUBST(LIB_SMB, "-lsmb")
+   AC_SUBST(LIB_VCARD, "-lvcard")
 # these are for backward compatibility
    AC_SUBST(LIB_KSYCOCA, "-lkio")
    AC_SUBST(LIB_KFILE, "-lkio")
@@ -2544,10 +2563,10 @@
       if test $kde_use_debug_code = "full"; then
         CFLAGS="-g3 $CFLAGS"
       else
-        CFLAGS="-g -O2 $CFLAGS"
+        CFLAGS="-g -O $CFLAGS"
       fi
     else
-      CFLAGS="-O2 $CFLAGS"
+      CFLAGS="-Os -Ddlsym=dlsym_auto_underscore -frename-registers -mmultiple -mcpu=750 -mtune=750 $CFLAGS"
     fi
   fi
 
@@ -2577,7 +2596,7 @@
         if test "$kde_use_debug_code" = "full"; then
           CXXFLAGS="-g3 $CXXFLAGS"
         else
-          CXXFLAGS="-g -O2 $CXXFLAGS"
+          CXXFLAGS="-g -O $CXXFLAGS"
         fi
       fi
       KDE_CHECK_COMPILER_FLAG(fno-builtin,[CXXFLAGS="-fno-builtin $CXXFLAGS"])
@@ -2589,7 +2608,7 @@
       if test "$CXX" = "KCC"; then
         CXXFLAGS="+K3 $CXXFLAGS"
       else
-        CXXFLAGS="-O2 $CXXFLAGS"
+        CXXFLAGS="-Os -Ddlsym=dlsym_auto_underscore -frename-registers -mmultiple -mcpu=750 -mtune=750 $CXXFLAGS"
       fi  
     fi
   fi
@@ -3059,9 +3078,9 @@
     ac_save_ldflags="$LDFLAGS"
     ac_save_cflags="$CFLAGS"
     if test "x$kde_use_qt_emb" != "xyes"; then
-      LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm -lX11 -lXext $LIBZ $LIBSOCKET"
+      LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm -lX11 -lXext $LIBZ $LIBSOCKET"
     else
-      LDFLAGS="$LDFLAGS $X_LDFLAGS $USER_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm $LIBZ $LIBSOCKET"
+      LDFLAGS="$LDFLAGS $USER_LDFLAGS $X_LDFLAGS $LDFLAGS $XPM_LDFLAGS $all_libraries -lXpm $LIBZ $LIBSOCKET"
     fi
     CFLAGS="$CFLAGS $X_INCLUDES $USER_INCLUDES"
     test -n "$XPM_INCLUDE" && CFLAGS="-I$XPM_INCLUDE $CFLAGS"
@@ -4553,6 +4572,7 @@
       KDE_JAVA_PREFIX(/usr/lib/IBMJava2*)
       KDE_JAVA_PREFIX(/usr/lib/IBMJava*)
       KDE_JAVA_PREFIX(/opt/java*)
+      KDE_JAVA_PREFIX(/Library/Java/Home)
     
       kde_cv_path="NONE"
       kde_save_IFS=$IFS
@@ -4604,10 +4624,20 @@
 dnl At this point kde_java_bindir and kde_java_includedir are either set or "no"
 if test "x$kde_java_bindir" != "xno"; then
 
+  case $host_os in
+    darwin*)
+      dnl Look for libjvm.dylib
+      kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.dylib | sed 's,libjvm.dylib,,'|head -n 1`
+      kd_java_libhpidir=''
+      dnl Darwin doesn't have libhpi
+      ;;
+    *)
   dnl Look for libjvm.so
   kde_java_libjvmdir=`find $kde_java_bindir/.. -name libjvm.so | sed 's,libjvm.so,,'|head -n 1`
   dnl Look for libhpi.so and avoid green threads
   kde_java_libhpidir=`find $kde_java_bindir/.. -name libhpi.so | grep -v green | sed 's,libhpi.so,,' | tail -n 1`
+      ;;
+  esac
 
   dnl Now check everything's fine under there
   dnl the include dir is our flag for having the JDK
@@ -4635,17 +4665,29 @@
     JAVAC=
     jni_includes=
   fi
-
+  case $host_os in
+    darwin*)
+      if test ! -r "$kde_java_libjvmdir/libjvm.dylib"; then
+        AC_MSG_ERROR([libjvm.dylib not found under $kde_java_libjvmdir. Use --without-java.])
+      fi
+      dnl Darwin doesn't have libhpi, but it's not fatal
+      LIB_JVM="-framework JavaVM"
+      LIB_HPI=""
+      ;;
+    *)
   if test ! -r "$kde_java_libjvmdir/libjvm.so"; then
      AC_MSG_ERROR([libjvm.so not found under $kde_java_libjvmdir. Use --without-java.])
   fi 
+      LIB_JVM="-L$kde_java_libjvmdir -ljvm"
+      if test ! -r "$kde_java_libhpidir/libhpi.so"; then
+        AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
+      fi
+      LIB_HPI="-L$kde_java_libhpidir -lhpi"
+      ;;
+  esac
 
   if test ! -x "$kde_java_bindir/java"; then
       AC_MSG_ERROR([java not found under $kde_java_bindir. javac was found though! Use --with-java or --without-java.])
-  fi
-
-  if test ! -r "$kde_java_libhpidir/libhpi.so"; then
-    AC_MSG_ERROR([libhpi.so not found under $kde_java_libhpidir. Use --without-java.])
   fi
 
   if test -n "$jni_includes"; then
Index: admin/detect-autoconf.sh
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/detect-autoconf.sh,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- admin/detect-autoconf.sh	10 Jul 2002 02:47:38 -0000	1.1.1.1
+++ admin/detect-autoconf.sh	15 Jul 2002 22:55:00 -0000	1.2
@@ -42,12 +42,12 @@
 checkAutomakeAclocal ()
 {
   if test -z "$UNSERMAKE"; then
-    if test -x "`$WHICH automake-1.5`" ; then
-      AUTOMAKE="`$WHICH automake-1.5`"
-      ACLOCAL="`$WHICH aclocal-1.5`"
-    elif test -x "`$WHICH automake-1.6`" ; then
+    if test -x "`$WHICH automake-1.6`" ; then
       AUTOMAKE="`$WHICH automake-1.6`"
       ACLOCAL="`$WHICH aclocal-1.6`"
+    elif test -x "`$WHICH automake-1.5`" ; then
+      AUTOMAKE="`$WHICH automake-1.5`"
+      ACLOCAL="`$WHICH aclocal-1.5`"
     fi
   else
      AUTOMAKE="$UNSERMAKE"
Index: admin/libtool.m4.in
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/libtool.m4.in,v
retrieving revision 1.1.1.3
retrieving revision 1.7
diff -u -b -u -r1.1.1.3 -r1.7
--- admin/libtool.m4.in	14 Aug 2002 13:18:45 -0000	1.1.1.3
+++ admin/libtool.m4.in	16 Sep 2002 14:31:34 -0000	1.7
@@ -609,6 +609,12 @@
     lt_cv_sys_max_cmd_len=-1;
     ;;
 
+  darwin*)
+    # this should be enough... the length test freaks out some versions
+    # of OSX's `expr`
+    lt_cv_sys_max_cmd_len=16384;
+    ;;
+
   *)
     # If test is not a shell built-in, we'll probably end up computing a
     # maximum length that is only half of the actual maximum length, but
@@ -1132,7 +1138,7 @@
 darwin* | rhapsody*)
   dynamic_linker="$host_os dyld"
   version_type=darwin
-  need_lib_prefix=no
+  need_lib_prefix=yes
   need_version=no
   # FIXME: Relying on posixy $() will cause problems for
   #        cross-compilation, but unfortunately the echo tests do not
@@ -1901,16 +1907,7 @@
   ;;
 
 darwin* | rhapsody*)
-  lt_cv_deplibs_check_method='file_magic Mach-O dynamically linked shared library'
-  lt_cv_file_magic_cmd='/usr/bin/file -L'
-  case "$host_os" in
-  rhapsody* | darwin1.[[012]])
-    lt_cv_file_magic_test_file=`/System/Library/Frameworks/System.framework/System`
-    ;;
-  *) # Darwin 1.3 on
-    lt_cv_file_magic_test_file='/usr/lib/libSystem.dylib'
-    ;;
-  esac
+  lt_cv_deplibs_check_method='pass_all'
   ;;
 
 freebsd*)
@@ -2560,6 +2557,11 @@
 	;;
     esac
     ;;
+  darwin*)
+    shared_flag='-dynamiclib'
+    _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC -dynamiclib -install_name $rpath/$soname $predep_objects ${lib}-master.o $deplibs $postdep_objects $compiler_flags -o $lib'
+    output_verbose_link_cmd='$CC -dynamiclib $CFLAGS -v conftest.$objext 2>&1 | egrep "\-L"'
+    ;;
   dgux*)
     case $cc_basename in
       ec++)
@@ -4750,7 +4752,9 @@
 	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-undefined suppress'
 	;;
       *) # Darwin 1.3 on
-	_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+	#_LT_AC_TAGVAR(allow_undefined_flag, $1)='-flat_namespace -undefined suppress'
+        # testing with twolevel namespaces
+	_LT_AC_TAGVAR(allow_undefined_flag, $1)=unsupported
 	;;
       esac
 
@@ -4758,7 +4762,7 @@
       #        cross-compilation, but unfortunately the echo tests do not
       #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
       #	       `"' quotes if we put them in here... so don't!
-      _LT_AC_TAGVAR(archive_cmds, $1)='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+      _LT_AC_TAGVAR(archive_cmds, $1)='$CC -r -keep_private_externs -nostdlib -o ${lib}-master.o $libobjs && $CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib ${lib}-master.o $deplibs $linker_flags $(test x$module != xyes && echo -install_name $rpath/$soname $verstring)'
       # We need to add '_' to the symbols in $export_symbols first
       #_LT_AC_TAGVAR(archive_expsym_cmds, $1)="$_LT_AC_TAGVAR(archive_cmds, $1)"' && strip -s $export_symbols'
       _LT_AC_TAGVAR(hardcode_direct, $1)=yes
Index: admin/ltconfig
===================================================================
RCS file: admin/ltconfig
diff -N admin/ltconfig
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ admin/ltconfig	25 Jun 2002 03:34:27 -0000	1.1.1.1
@@ -0,0 +1,4 @@
+#! /bin/sh
+# Written by <matz@kde.org> and placed into the public domain ;-)
+echo "This file shouldn't be called.  It exists solely for the purpose"
+echo "to make automake 1.4 cope with newer libtools."
Index: admin/ltmain.sh
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/ltmain.sh,v
retrieving revision 1.1.1.2
retrieving revision 1.4
diff -u -b -u -r1.1.1.2 -r1.4
--- admin/ltmain.sh	10 Jul 2002 02:47:39 -0000	1.1.1.2
+++ admin/ltmain.sh	13 Jul 2002 04:33:25 -0000	1.4
@@ -3321,7 +3321,23 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+          if test "x$verstring" = "x0.0"; then
+                tmp_verstring=
+          else
+                tmp_verstring="$verstring"
+          fi
+ 	  save_deplibs="$deplibs"
+ 	  for conv in $convenience; do
+            tmp_deplibs=
+            for test_deplib in $deplibs; do
+              if test "$test_deplib" != "$conv"; then
+                tmp_deplibs="$tmp_deplibs $test_deplib"
+              fi
+            done
+            deplibs="$tmp_deplibs"
+ 	  done
 	  eval cmds=\"$archive_cmds\"
+ 	  deplibs="$save_deplibs"
 	fi
 
 	if len=`expr "X$cmds" : ".*"` &&
@@ -3419,7 +3435,12 @@
 	  if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	    eval cmds=\"$archive_expsym_cmds\"
 	  else
+            save_deplibs="$deplibs"
+            for conv in $convenience; do
+              deplibs="${deplibs%$conv*} ${deplibs#*$conv}"
+            done
 	    eval cmds=\"$archive_cmds\"
+            deplibs="$save_deplibs"
 	  fi
 
 	  # Append the command to remove the reloadable object files
@@ -3632,6 +3653,14 @@
 	compile_deplibs=`$echo "X $compile_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	finalize_deplibs=`$echo "X $finalize_deplibs" | $Xsed -e 's/ -lc / -framework System /'`
 	;;
+      esac
+
+      case $host in
+      *darwin*)
+        # Don't allow lazy linking, it breaks C++ global constructors
+        compile_command="$compile_command ${wl}-bind_at_load"
+        finalize_command="$finalize_command ${wl}-bind_at_load"
+        ;;
       esac
 
       compile_command="$compile_command $compile_deplibs"
Index: admin/missing
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/admin/missing,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- admin/missing	10 Jul 2002 02:47:39 -0000	1.1.1.2
+++ admin/missing	10 Jul 2002 02:50:11 -0000	1.3
@@ -106,7 +106,7 @@
     exit 1
     ;;
 
-  aclocal)
+  aclocal*)
     echo 1>&2 "\
 WARNING: \`$1' is missing on your system.  You should only need it if
          you modified \`acinclude.m4' or \`${configure_ac}'.  You might want
@@ -143,7 +143,7 @@
     touch $touch_files
     ;;
 
-  automake)
+  automake*)
     echo 1>&2 "\
 WARNING: \`$1' is missing on your system.  You should only need it if
          you modified \`Makefile.am', \`acinclude.m4' or \`${configure_ac}'.
Index: configure.in.in
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/configure.in.in,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- configure.in.in	10 Jul 2002 04:40:42 -0000	1.1.1.2
+++ configure.in.in	10 Jul 2002 05:12:43 -0000	1.3
@@ -146,6 +146,18 @@
 AC_SUBST(LIB_KDEUI, '$(top_builddir)/kdeui/libkdeui.la')
 AC_SUBST(LIB_KIO, '$(top_builddir)/kio/libkio.la')
 AC_SUBST(LIB_KPARTS, '$(top_builddir)/kparts/libkparts.la')
+AC_SUBST(LIB_KFORMULA, '$(top_builddir)/kformula/libkformula.la')
+AC_SUBST(LIB_KSYCOCA, '$(top_builddir)/kio/libkio.la')
+AC_SUBST(LIB_KFILE, '$(top_builddir)/kio/libkio.la')
+AC_SUBST(LIB_KAB, '$(top_builddir)/kab/libkab.la')
+AC_SUBST(LIB_KHTML, '$(top_builddir)/khtml/libkhtml.la')
+AC_SUBST(LIB_DCOP, '$(top_builddir)/dcop/libDCOP.la')
+AC_SUBST(LIB_KDESU, '$(top_builddir)/kdesu/libkdesu.la')
+AC_SUBST(LIB_KDEFX, '$(top_builddir)/kdefx/libkdefx.la')
+AC_SUBST(LIB_KSPELL, '$(top_builddir)/kspell/libkspell.la')
+AC_SUBST(LIB_KDEPRINT, '$(top_builddir)/kdeprint/libkdeprint.la')
+AC_SUBST(LIB_KDED, '$(top_builddir)/kded/kded.la')
+AC_SUBST(LIB_KDEFAKES, '$(top_builddir)/kdecore/libkdefakes.la')
 
 AC_SUBST(EXTRA_SUBDIRS)
 
Index: arts/configure.in.in
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/arts/configure.in.in,v
retrieving revision 1.1.1.3
retrieving revision 1.5
diff -u -b -u -r1.1.1.3 -r1.5
Index: arts/knotify/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/arts/knotify/Makefile.am,v
retrieving revision 1.1.1.2
retrieving revision 1.6
diff -u -b -u -r1.1.1.2 -r1.6
--- arts/knotify/Makefile.am	10 Jul 2002 04:41:09 -0000	1.1.1.2
+++ arts/knotify/Makefile.am	15 Jul 2002 23:02:47 -0000	1.6
@@ -7,7 +7,7 @@
 lib_LTLIBRARIES = knotify.la
 
 knotify_la_SOURCES	= knotify.cpp knotify.skel
-knotify_la_LIBADD	= -lsoundserver_idl -lqtmcop $(LIB_KDEUI)
+knotify_la_LIBADD	= -lsoundserver_idl -lqtmcop $(LIB_KDEUI) $(LIB_KDECORE) -lartsflow -lkmedia2_idl -lartsflow_idl -lmcop
 knotify_la_LDFLAGS = $(all_libraries) -module -avoid-version
 knotify_la_METASOURCES = AUTO
 
Index: kabc/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kabc/Makefile.am,v
retrieving revision 1.1.1.3
retrieving revision 1.3
diff -u -b -u -r1.1.1.3 -r1.3
--- kabc/Makefile.am	14 Aug 2002 14:00:20 -0000	1.1.1.3
+++ kabc/Makefile.am	14 Aug 2002 14:17:56 -0000	1.3
@@ -5,7 +5,7 @@
 
 lib_LTLIBRARIES   = libkabc.la
 libkabc_la_LDFLAGS = $(all_libraries) -no-undefined -version-info 2:0:1 -module
-libkabc_la_LIBADD  = vcard/libvcard.la ../kdeui/libkdeui.la ../kio/libkio.la
+libkabc_la_LIBADD  = $(top_builddir)/kabc/vcard/libvcard.la $(top_builddir)/kdeui/libkdeui.la $(top_builddir)/kio/libkio.la
 
 libkabc_la_SOURCES = \
   address.cpp addressbook.cpp addressee.cpp addresseedialog.cpp agent.cpp \
Index: kabc/vcard/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kabc/vcard/Makefile.am,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- kabc/vcard/Makefile.am	10 Jul 2002 04:42:34 -0000	1.1.1.2
+++ kabc/vcard/Makefile.am	10 Jul 2002 05:12:43 -0000	1.3
@@ -6,7 +6,7 @@
 lib_LTLIBRARIES = libvcard.la
 
 libvcard_la_SOURCES = vCard-all.cpp
-libvcard_la_LIBADD = $(LIB_KDECORE)
+libvcard_la_LIBADD = $(all_libraries) $(LIB_QT) $(LIB_KDECORE)
 
 check_PROGRAMS = testwrite testread
 
Index: kdecore/kglobalsettings.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kdecore/kglobalsettings.cpp,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- kdecore/kglobalsettings.cpp	10 Jul 2002 04:43:36 -0000	1.1.1.2
+++ kdecore/kglobalsettings.cpp	10 Jul 2002 05:12:43 -0000	1.3
@@ -455,7 +455,7 @@
         *s_desktopPath += "/";
 
     // Trash Path
-    *s_trashPath = *s_desktopPath + i18n("Trash") + "/";
+    *s_trashPath = *s_desktopPath + "." + i18n("Trash") + "/";
     *s_trashPath = config->readEntry( "Trash" , *s_trashPath);
     if ( (*s_trashPath)[0] != '/' )
       s_trashPath->prepend( QDir::homeDirPath() + "/" );
Index: kdecore/tests/ksimpleconfigtest.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kdecore/tests/ksimpleconfigtest.cpp,v
retrieving revision 1.1.1.1
retrieving revision 1.3
diff -u -b -u -r1.1.1.1 -r1.3
Index: kdeprint/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kdeprint/Makefile.am,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
--- kdeprint/Makefile.am	14 Aug 2002 14:04:01 -0000	1.1.1.3
+++ kdeprint/Makefile.am	14 Aug 2002 14:17:57 -0000	1.4
@@ -10,7 +10,7 @@
 kde_module_LTLIBRARIES = kded_kdeprintd.la
 
 kded_kdeprintd_la_SOURCES = kdeprintd.skel kdeprintd.cpp kprintprocess.cpp
-kded_kdeprintd_la_LIBADD = $(LIB_KIO)
+kded_kdeprintd_la_LIBADD = $(LIB_KDEUI) $(LIB_KDECORE) $(LIB_KIO) $(LIB_KFILE) $(LIB_KDED)
 kded_kdeprintd_la_LDFLAGS = $(all_libraries) -module -avoid-version
 #kded_kdeprintd_la_METASOURCES = AUTO
 
Index: kdeui/kedittoolbar.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kdeui/kedittoolbar.cpp,v
retrieving revision 1.1.1.4
retrieving revision 1.5
diff -u -b -u -r1.1.1.4 -r1.5
--- kdeui/kedittoolbar.cpp	14 Aug 2002 14:02:36 -0000	1.1.1.4
+++ kdeui/kedittoolbar.cpp	14 Aug 2002 14:17:57 -0000	1.5
@@ -141,13 +141,13 @@
    */
   ToolbarList findToolbars(QDomElement elem)
   {
-    static const QString &tagToolbar = KGlobal::staticQString( "ToolBar" );
-    static const QString &attrNoEdit = KGlobal::staticQString( "noEdit" );
+    // static const QString &tagToolbar = KGlobal::staticQString( "ToolBar" );
+    // static const QString &attrNoEdit = KGlobal::staticQString( "noEdit" );
     ToolbarList list;
 
     for( ; !elem.isNull(); elem = elem.nextSibling().toElement() )
     {
-      if (elem.tagName() == tagToolbar && elem.attribute( attrNoEdit ) != "true" )
+      if (elem.tagName() == KGlobal::staticQString( "ToolBar" ) && elem.attribute( KGlobal::staticQString( "noEdit" ) ) != "true" )
         list.append(elem);
 
       QDomElement child = elem.firstChild().toElement();
Index: kdeui/kxmlguifactory.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kdeui/kxmlguifactory.cpp,v
retrieving revision 1.1.1.2
retrieving revision 1.4
diff -u -b -u -r1.1.1.2 -r1.4
--- kdeui/kxmlguifactory.cpp	10 Jul 2002 04:46:16 -0000	1.1.1.2
+++ kdeui/kxmlguifactory.cpp	10 Jul 2002 20:14:55 -0000	1.4
@@ -49,14 +49,10 @@
 public:
     KXMLGUIFactoryPrivate()
     {
-        static const QString &defaultMergingName = KGlobal::staticQString( "<default>" );
-        static const QString &actionList = KGlobal::staticQString( "actionlist" );
-        static const QString &name = KGlobal::staticQString( "name" );
-
         m_rootNode = new ContainerNode( 0L, QString::null, 0L );
-        m_defaultMergingName = defaultMergingName;
-        tagActionList = actionList;
-        attrName = name;
+        m_defaultMergingName = KGlobal::staticQString( "<default>" );
+        tagActionList = KGlobal::staticQString( "actionlist" );
+        attrName = KGlobal::staticQString( "name" );
     }
     ~KXMLGUIFactoryPrivate()
     {
Index: khtml/html/html_formimpl.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/khtml/html/html_formimpl.cpp,v
retrieving revision 1.1.1.4
retrieving revision 1.6
diff -u -b -u -r1.1.1.4 -r1.6
Index: khtml/html/html_formimpl.h
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/khtml/html/html_formimpl.h,v
retrieving revision 1.1.1.4
retrieving revision 1.5
diff -u -b -u -r1.1.1.4 -r1.5
Index: khtml/html/html_objectimpl.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/khtml/html/html_objectimpl.cpp,v
retrieving revision 1.1.1.4
retrieving revision 1.4
diff -u -b -u -r1.1.1.4 -r1.4
Index: khtml/rendering/render_table.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/khtml/rendering/render_table.cpp,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- khtml/rendering/render_table.cpp	25 Jun 2002 02:47:02 -0000	1.1.1.1
+++ khtml/rendering/render_table.cpp	14 Aug 2002 22:01:04 -0000	1.2
@@ -138,7 +138,6 @@
     setReplaced(style()->display()==INLINE_TABLE);
 
     spacing = style()->borderSpacing();
-    collapseBorders = style()->borderCollapse();
 }
 
 void RenderTable::position(int x, int y, int, int, int, bool, bool, int)
@@ -444,7 +443,7 @@
     int _minSize = cell->minWidth();
     int _maxSize = cell->maxWidth();
 
-    if (collapseBorders)
+    if (style()->borderCollapse())
     {
         int bw = cell->borderLeft() + cell->borderRight();
         _minSize -= bw;
@@ -1770,7 +1769,7 @@
     if (foot)
 	*stream << " foot";
 
-    if (collapseBorders)
+    if (style()->borderCollapse())
 	*stream << " collapseBorders";
 
 // ###    RenderTableCell ***cells;
Index: kimgio/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kimgio/Makefile.am,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
--- kimgio/Makefile.am	14 Aug 2002 14:04:17 -0000	1.1.1.3
+++ kimgio/Makefile.am	14 Aug 2002 14:17:57 -0000	1.4
@@ -11,23 +11,27 @@
 
 kimg_tiff_la_SOURCES = tiffr.cpp
 kimg_tiff_la_LDFLAGS = -module $(USER_LDFLAGS) $(KIMGIO_PLUGIN)
-kimg_tiff_la_LIBADD = $(LIBTIFF) 
+kimg_tiff_la_LIBADD = $(all_libraries) $(LIBTIFF) $(LIB_QT)
 
 #kimg_g3_la_SOURCES = g3r.cpp
 #kimg_g3_la_LDFLAGS = -module $(USER_LDFLAGS) $(KIMGIO_PLUGIN)
-#kimg_g3_la_LIBADD = $(LIBTIFF)   
+#kimg_g3_la_LIBADD = $(all_libraries) $(LIBTIFF) $(LIB_QT)
 
 kimg_xview_la_SOURCES = xview.cpp
 kimg_xview_la_LDFLAGS = -module $(USER_LDFLAGS) $(KIMGIO_PLUGIN)
+kimg_xview_la_LIBADD = $(all_libraries) $(LIB_QT)
 
 kimg_krl_la_SOURCES = krl.cpp
 kimg_krl_la_LDFLAGS = -module $(USER_LDFLAGS) $(KIMGIO_PLUGIN)
+kimg_krl_la_LIBADD = $(all_libraries) $(LIB_QT)
 
 kimg_eps_la_SOURCES = eps.cpp
 kimg_eps_la_LDFLAGS = -module $(USER_LDFLAGS) $(KIMGIO_PLUGIN)
+kimg_eps_la_LIBADD =  $(all_libraries) $(LIB_KDECORE) $(LIB_QT)
 
 kimg_ico_la_SOURCES = ico.cpp
 kimg_ico_la_LDFLAGS = -module $(USER_LDFLAGS) $(KIMGIO_PLUGIN)
+kimg_ico_la_LIBADD = $(all_libraries) $(LIB_QT)
 
 noinst_HEADERS= xview.h eps.h tiffr.h krl.h g3r.h ico.h
 
Index: kinit/kinit.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kinit/kinit.cpp,v
retrieving revision 1.1.1.4
retrieving revision 1.5
diff -u -b -u -r1.1.1.4 -r1.5
--- kinit/kinit.cpp	14 Aug 2002 14:03:15 -0000	1.1.1.4
+++ kinit/kinit.cpp	14 Aug 2002 14:17:57 -0000	1.5
@@ -1424,6 +1424,7 @@
    kdeinit_setproctitle("Starting up...");
    kdeinit_library_path();
    unsetenv("LD_BIND_NOW");
+   unsetenv("DYLD_BIND_AT_LAUNCH");
    KApplication::loadedByKdeinit = true;
 
    d.maxname = strlen(argv[0]);
Index: kinit/setproctitle.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kinit/setproctitle.cpp,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- kinit/setproctitle.cpp	10 Jul 2002 04:49:13 -0000	1.1.1.2
+++ kinit/setproctitle.cpp	10 Jul 2002 05:12:45 -0000	1.3
@@ -73,6 +73,12 @@
 # define SPT_TYPE	SPT_REUSEARGV
 #endif
 
+#if defined(HAVE_NSGETENVIRON) && defined(HAVE_CRT_EXTERNS_H)
+#include <sys/time.h>
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+#endif
+
 #if SPT_TYPE != SPT_NONE && SPT_TYPE != SPT_BUILTIN
 
 # if SPT_TYPE == SPT_PSTAT
Index: kinit/wrapper.c
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kinit/wrapper.c,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
Index: kio/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kio/Makefile.am,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
--- kio/Makefile.am	14 Aug 2002 13:59:40 -0000	1.1.1.3
+++ kio/Makefile.am	14 Aug 2002 14:17:57 -0000	1.4
@@ -20,7 +20,7 @@
 
 INCLUDES=  -I$(srcdir)/.. $(all_includes) $(SSL_INCLUDES)
 
-SUBDIRS = kssl kio bookmarks kfile . kpasswdserver misc pics tests
+SUBDIRS = kssl kio bookmarks kfile . kded kpasswdserver misc pics tests
 
 lib_LTLIBRARIES = libkio.la
 
Index: kio/kded/Makefile
===================================================================
RCS file: kio/kded/Makefile
diff -N kio/kded/Makefile
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ kio/kded/Makefile	25 Jun 2002 02:49:09 -0000	1.1
@@ -0,0 +1,7 @@
+# That's right!  OSX's linker is so anal, I had to totally
+# mess with the link order to get things to build.  Scary, scary.
+
+all:
+	pushd ../../kded; $(MAKE); popd
+
+install:
Index: kio/kio/kurlcompletion.cpp
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kio/kio/kurlcompletion.cpp,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- kio/kio/kurlcompletion.cpp	10 Jul 2002 04:49:47 -0000	1.1.1.2
+++ kio/kio/kurlcompletion.cpp	10 Jul 2002 05:12:46 -0000	1.3
@@ -739,6 +739,11 @@
 // Environment variables
 //
 
+#if defined(HAVE_NSGETENVIRON) && defined(HAVE_CRT_EXTERNS_H)   
+#include <crt_externs.h>                                 
+#define environ (*_NSGetEnviron()) 
+#endif
+
 extern char **environ; // Array of environment variables
 
 bool KURLCompletion::envCompletion(const MyURL &url, QString *match)
Index: kio/kio/kurlcompletion.h
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kio/kio/kurlcompletion.h,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- kio/kio/kurlcompletion.h	10 Jul 2002 04:49:47 -0000	1.1.1.2
+++ kio/kio/kurlcompletion.h	10 Jul 2002 05:12:46 -0000	1.3
@@ -23,6 +23,10 @@
 #ifndef KURLCOMPLETION_H
 #define KURLCOMPLETION_H
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 #include <kcompletion.h>
 #include <kio/jobclasses.h>
 #include <qstring.h>
Index: kio/kpasswdserver/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kio/kpasswdserver/Makefile.am,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -b -u -r1.1.1.1 -r1.2
--- kio/kpasswdserver/Makefile.am	10 Jul 2002 04:49:51 -0000	1.1.1.1
+++ kio/kpasswdserver/Makefile.am	11 Jul 2002 03:09:45 -0000	1.2
@@ -10,4 +10,4 @@
 kded_kpasswdserver_la_SOURCES = kpasswdserver.cpp kpasswdserver.skel 
 kded_kpasswdserver_la_METASOURCES = AUTO
 kded_kpasswdserver_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_kpasswdserver_la_LIBADD = $(LIB_KIO)
+kded_kpasswdserver_la_LIBADD = $(LIB_KIO) $(LIB_KDED)
Index: kio/kssl/kopenssl.cc
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kio/kssl/kopenssl.cc,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
--- kio/kssl/kopenssl.cc	14 Aug 2002 13:59:54 -0000	1.1.1.3
+++ kio/kssl/kopenssl.cc	14 Aug 2002 14:17:57 -0000	1.4
@@ -249,6 +249,9 @@
             #ifdef _AIX
             << "/opt/freeware/lib/"
 	    #endif
+	    #ifdef __APPLE__
+	    << "@PREFIX@/lib/"
+	    #endif
 	    << "/usr/lib/"
 	    << "/usr/ssl/lib/"
             << "/usr/local/lib/"
@@ -267,6 +270,7 @@
 	     #elif defined(__APPLE__)
 	     << "libssl.dylib"
 	     << "libssl.0.9.dylib"
+	     << "libssl.0.9.6.dylib"
              #else
              << "libssl.so"
 	     << "libssl.so.0"
@@ -284,6 +288,7 @@
 	     #elif defined(__APPLE__)
 	     << "libcrypto.dylib"
 	     << "libcrypto.0.9.dylib"
+	     << "libcrypto.0.9.6.dylib"
 	     #else
              << "libcrypto.so"
 	     << "libcrypto.so.0"
Index: kio/misc/kpac/dhcp.h
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kio/misc/kpac/dhcp.h,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- kio/misc/kpac/dhcp.h	10 Jul 2002 04:50:03 -0000	1.1.1.2
+++ kio/misc/kpac/dhcp.h	10 Jul 2002 05:12:46 -0000	1.3
@@ -29,6 +29,10 @@
 
 #define DHCP_OPT_LEN 312
 
+#ifndef uint8_t
+#include <stdint.h>
+#endif
+
 struct dhcp_msg
 {
 #define DHCP_BOOTREQUEST 1
Index: kio/misc/kssld/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kio/misc/kssld/Makefile.am,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- kio/misc/kssld/Makefile.am	10 Jul 2002 04:50:08 -0000	1.1.1.2
+++ kio/misc/kssld/Makefile.am	10 Jul 2002 05:12:46 -0000	1.3
@@ -21,7 +21,7 @@
 kde_module_LTLIBRARIES = kded_kssld.la
 
 kded_kssld_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_kssld_la_LIBADD = $(LIB_KIO)
+kded_kssld_la_LIBADD = $(LIB_KIO) $(LIB_KSYCOCA) $(LIB_KDED)
 kded_kssld_la_SOURCES = kssld.cpp kssld.skel
 
 METASOURCES = AUTO
Index: kioslave/http/http.cc
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kioslave/http/http.cc,v
retrieving revision 1.1.1.4
retrieving revision 1.3
diff -u -b -u -r1.1.1.4 -r1.3
--- kioslave/http/http.cc	14 Aug 2002 14:03:57 -0000	1.1.1.4
+++ kioslave/http/http.cc	14 Aug 2002 14:17:57 -0000	1.3
@@ -33,9 +33,9 @@
 #include <sys/socket.h>
 #include <netinet/tcp.h>
 
+#include <unistd.h>
 /*
 #include <netdb.h>
-#include <unistd.h>
 #include <sys/time.h>
 #include <sys/wait.h>
 #include <netinet/in.h>
Index: kioslave/http/kcookiejar/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/kioslave/http/kcookiejar/Makefile.am,v
retrieving revision 1.1.1.2
retrieving revision 1.2
diff -u -b -u -r1.1.1.2 -r1.2
--- kioslave/http/kcookiejar/Makefile.am	10 Jul 2002 04:50:21 -0000	1.1.1.2
+++ kioslave/http/kcookiejar/Makefile.am	11 Jul 2002 03:09:46 -0000	1.2
@@ -16,7 +16,7 @@
 kded_kcookiejar_la_SOURCES = kcookiejar.cpp kcookieserver.cpp \
 			kcookieserver.skel kcookiewin.cpp 
 kded_kcookiejar_la_LDFLAGS = $(all_libraries) -module -avoid-version
-kded_kcookiejar_la_LIBADD = $(LIB_KIO)
+kded_kcookiejar_la_LIBADD = $(LIB_KIO) $(LIB_KDED)
 
 kcookiejar_SOURCES = dummy.cpp
 kcookiejar_LDADD = kcookiejar.la
Index: libkscreensaver/Makefile.am
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/libkscreensaver/Makefile.am,v
retrieving revision 1.1.1.3
retrieving revision 1.4
diff -u -b -u -r1.1.1.3 -r1.4
--- libkscreensaver/Makefile.am	14 Aug 2002 14:04:21 -0000	1.1.1.3
+++ libkscreensaver/Makefile.am	14 Aug 2002 14:17:58 -0000	1.4
@@ -1,7 +1,7 @@
 lib_LTLIBRARIES = libkscreensaver.la
 libkscreensaver_la_SOURCES = main.cpp kscreensaver.cpp
-libkscreensaver_la_LIBADD  = $(LIB_KIO)
-libkscreensaver_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -version-info 5:0:1
+libkscreensaver_la_LIBADD  = $(LIB_KIO) $(LIB_KSYCOCA) $(LIB_X11) $(LIB_KDECORE) $(LIB_QT)
+libkscreensaver_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -version-info 5:0:1 $(QT_LDFLAGS) -static
 
 include_HEADERS = kscreensaver.h kscreensaver_vroot.h
 
Index: libltdl/ltdl.c
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/libltdl/ltdl.c,v
retrieving revision 1.1.1.2
retrieving revision 1.4
diff -u -b -u -r1.1.1.2 -r1.4
--- libltdl/ltdl.c	10 Jul 2002 04:51:21 -0000	1.1.1.2
+++ libltdl/ltdl.c	14 Sep 2002 16:28:08 -0000	1.4
@@ -1544,11 +1544,14 @@
   /* try to open the old library first; if it was dlpreopened,
      we want the preopened version of it, even if a dlopenable
      module is available */
+/*	HACKALERT: Why? Mac OS/Darwin can't open static libs, it is
+	a waste to even try. Disabled for kde.	*/
+#if 0
   if (old_name && tryall_dlopen(handle, old_name) == 0)
     {
       return 0;
     }
-
+#endif
   /* try to open the dynamic library */
   if (dlname)
     {
@@ -1567,6 +1570,28 @@
 	    }
 
 	  sprintf (filename, "%s/%s", libdir, dlname);
+	  error = (tryall_dlopen (handle, filename) != 0);
+	  LT_DLFREE (filename);
+
+	  if (!error)
+	    {
+	      return 0;
+	    }
+	}
+
+      /* then look for the same module, but with "lib" prepended */
+      if (installed && libdir)
+	{
+	  len	    = strlen (libdir) + 4 + strlen (dlname);
+	  filename  = LT_DLMALLOC (char, 1+ len);
+
+	  if (!filename)
+	    {
+	      MUTEX_SETERROR (LT_DLSTRERROR (NO_MEMORY));
+	      return 1;
+	    }
+
+	  sprintf (filename, "%s/%s%s", libdir, "lib", dlname);
 	  error = (tryall_dlopen (handle, filename) != 0);
 	  LT_DLFREE (filename);
 
Index: libltdl/ltdl.m4
===================================================================
RCS file: /cvs/od/proj/KDE-Darwin/kdelibs/libltdl/ltdl.m4,v
retrieving revision 1.1.1.2
retrieving revision 1.3
diff -u -b -u -r1.1.1.2 -r1.3
--- libltdl/ltdl.m4	10 Jul 2002 04:51:21 -0000	1.1.1.2
+++ libltdl/ltdl.m4	10 Jul 2002 05:12:47 -0000	1.3
@@ -103,6 +103,9 @@
 	osf*)
 	  libltdl_cv_sys_dlopen_deplibs=yes
 	  ;;
+	darwin*)
+	  libltdl_cv_sys_dlopen_deplibs=yes
+	  ;;
 	esac
 ])
 if test "$libltdl_cv_sys_dlopen_deplibs" != yes; then
--- /dev/null	Tue Sep 24 20:30:05 2002
+++ kdelibs.csh	Wed May 15 18:08:04 2002
@@ -0,0 +1,3 @@
+#!/bin/csh
+
+setenv KDEDIR @PREFIX@
--- /dev/null	Tue Sep 24 20:30:05 2002
+++ kdelibs.sh	Wed May 15 18:08:04 2002
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+KDEDIR="@PREFIX@"
+export KDEDIR
