--- aview-1.3.0/image.c	Wed Apr 25 15:59:01 2001
+++ aview-1.3.0-patched/image.c	Thu Feb 14 23:40:08 2002
@@ -1,6 +1,6 @@
 #include <stdio.h>
 #include <unistd.h>
-#include <malloc.h>
+#include <stdlib.h>
 #include "config.h"
 
 int imgwidth, imgheight;

--- aview-1.3.0/ui.c	Wed Apr 25 16:04:37 2001
+++ aview-1.3.0-patched/ui.c	Thu Feb 14 23:40:29 2002
@@ -1,6 +1,6 @@
 #include <stdio.h>
 #include <ctype.h>
-#include <malloc.h>
+#include <stdlib.h>
 #include <string.h>
 #include <aalib.h>
 #include "shrink.h"
