diff -aur cinepaint-0.18-1/configure cinepaint-0.18-1-new/configure
--- cinepaint-0.18-1/configure	Sun Aug  3 19:32:40 2003
+++ cinepaint-0.18-1-new/configure	Sat Aug  9 10:38:56 2003
@@ -6023,10 +6023,10 @@
   darwin* | rhapsody*)
     case "$host_os" in
     rhapsody* | darwin1.[012])
-      allow_undefined_flag='-undefined suppress'
+      allow_undefined_flag=''
       ;;
     *) # Darwin 1.3 on
-      allow_undefined_flag='-flat_namespace -undefined suppress'
+      allow_undefined_flag=''
       ;;
     esac
     # FIXME: Relying on posixy $() will cause problems for
@@ -10379,9 +10379,9 @@
 case $host_os in
 darwin* | rhapsody*)
 CPPFLAGS="$CPPFLAGS -I/sw/include -no-cpp-precomp -D_REGEX_RE_COMP"
-LDFLAGS="$LDFLAGS -L/sw/lib -flat_namespace -undefined suppress"
+LDFLAGS="$LDFLAGS -L/sw/lib $GTK_LIBS"
 cinepaint_save_CPPFLAGS="$cinepaint_save_CPPFLAGS -I/sw/include -no-cpp-precomp -D_REGEX_RE_COMP"
-cinepaint_save_LDFLAGS="$cinepaint_save_LDFLAGS -L/sw/lib -flat_namespace -undefined suppress"
+cinepaint_save_LDFLAGS="$cinepaint_save_LDFLAGS $LDFLAGS -L/sw/lib $GTK_LIBS"
 ;;
 esac
 
