diff -urd ksudoku-0.4/CMakeLists.txt ksudoku-0.4.fixed/CMakeLists.txt
--- ksudoku-0.4/CMakeLists.txt	2007-02-14 16:45:28.000000000 +0100
+++ ksudoku-0.4.fixed/CMakeLists.txt	2007-07-15 05:03:16.000000000 +0200
@@ -15,7 +15,11 @@
 add_definitions(${QT_DEFINITIONS} ${KDE3_DEFINITIONS})
 
 # tell cmake where to search for libraries:
-link_directories(${KDE3_LIB_DIR})
+link_directories(
+	${KDE3_LIB_DIR}
+	@FINKPREFIX@/lib/qt3/lib
+	/usr/X11R6/lib
+)
 
 IF(DEBUG)
 	ADD_DEFINITIONS(-DDEBUG -ggdb -O)
@@ -24,7 +28,7 @@
 # tell cmake where to search for Qt/KDE headers:
 include_directories(
 	${KDE3_INCLUDE_DIR}
-	${QT_INCLUDE_DIR}
+	${QT_INCLUDE_DIR} /usr/X11R6/include
 	${CMAKE_SOURCE_DIR}/src
 	${CMAKE_SOURCE_DIR}/src/gui
 	${CMAKE_SOURCE_DIR}/src/gui/views
diff -urd ksudoku-0.4/src/CMakeLists.txt ksudoku-0.4.fixed/src/CMakeLists.txt
--- ksudoku-0.4/src/CMakeLists.txt	2007-03-17 14:50:01.000000000 +0100
+++ ksudoku-0.4.fixed/src/CMakeLists.txt	2007-07-15 12:38:11.000000000 +0200
@@ -19,7 +19,8 @@
 
 target_link_libraries(ksudoku
 	ksudoku_gui ksudoku_views ksudoku_export ksudoku_draw ksudoku_logic
-	${QT_AND_KDECORE_LIBRARIES} kdeui kio kdeprint GL GLU knewstuff
+	${QT_AND_KDECORE_LIBRARIES} kdeui kio DCOP kdeprint GL GLU knewstuff qt-mt qui kdecore kdefx
+        iconv
 )
 
 set(BIN_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/bin)
diff -urd ksudoku-0.4/src/gui/CMakeLists.txt ksudoku-0.4.fixed/src/gui/CMakeLists.txt
--- ksudoku-0.4/src/gui/CMakeLists.txt	2007-02-13 00:04:32.000000000 +0100
+++ ksudoku-0.4.fixed/src/gui/CMakeLists.txt	2007-07-15 12:41:56.000000000 +0200
@@ -29,5 +29,7 @@
 
 kde3_add_executable(ksudoku_client ${ksudoku_client_SRCS})
 
-target_link_libraries(ksudoku_client ${QT_AND_KDECORE_LIBRARIES} kdeui)
+target_link_libraries(ksudoku_client ${QT_AND_KDECORE_LIBRARIES} kdeui
+kdefx kdecore qt-mt qui DCOP
+)
 
