--- faad2-2.7.orig/libfaad/common.h	2010-01-11 01:52:42.000000000 +0900
+++ faad2-2.7/libfaad/common.h	2010-01-11 01:53:09.000000000 +0900
@@ -325,6 +325,8 @@
         }
         return i;
     }
+  #elif (defined(__i386__) && defined(__APPLE_CC__))
+    #define HAS_LRINTF
   #elif (defined(__i386__) && defined(__GNUC__) && \
 	!defined(__CYGWIN__) && !defined(__MINGW32__))
     #ifndef HAVE_LRINTF
