diff -ru libxmp-4.1.0.orig/Makefile.in libxmp-4.1.0/Makefile.in
--- libxmp-4.1.0.orig/Makefile.in	2013-04-14 15:18:04.000000000 -0400
+++ libxmp-4.1.0/Makefile.in	2013-04-22 20:41:33.000000000 -0400
@@ -111,7 +111,7 @@
 
 lib/$(DYLIB): $(LOBJS)
 	@mkdir -p lib
-	@CMD='$(CC) -dynamiclib -Wl,-headerpad_max_install_names,-undefined,dynamic_lookup,-compatibility_version,$(COMPAT_VERSION),-current_version,$(VERSION),-install_name,$(prefix)/lib/$(DYLIB) -o $@ $(LOBJS) $(LIBS)'; \
+	@CMD='$(CC) -dynamiclib -Wl,-compatibility_version,$(COMPAT_VERSION),-current_version,$(VERSION),-install_name,$(prefix)/lib/libxmp.$(VERSION_MAJOR).dylib -o $@ $(LOBJS) $(LIBS)'; \
 	if [ "$(V)" -gt 0 ]; then echo $$CMD; else echo LD $@ ; fi; \
 	eval $$CMD
 	ln -sf $(DYLIB) lib/libxmp.$(VERSION_MAJOR).dylib
diff -ru libxmp-4.1.0.orig/configure libxmp-4.1.0/configure
--- libxmp-4.1.0.orig/configure	2013-04-20 21:04:42.000000000 -0400
+++ libxmp-4.1.0/configure	2013-04-22 20:41:33.000000000 -0400
@@ -3182,7 +3182,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_c_flag_w_unused_but_set_variable" >&5
 $as_echo "$ac_cv_c_flag_w_unused_but_set_variable" >&6; }
   if test "x$ac_cv_c_flag_w_unused_but_set_variable" = xyes; then :
-  CFLAGS="${CFLAGS} -Wno-unused-but-set-variable"
+  CFLAGS="${CFLAGS} -Wno-unused-variable"
 fi
 
 
diff -ru libxmp-4.1.0.orig/src/common.h libxmp-4.1.0/src/common.h
--- libxmp-4.1.0.orig/src/common.h	2013-04-17 19:55:06.000000000 -0400
+++ libxmp-4.1.0/src/common.h	2013-04-22 21:18:10.000000000 -0400
@@ -124,14 +124,6 @@
 
 #endif	/* !_MSC_VER */
 
-#ifdef HAVE_STRLCPY
-#define strncpy strlcpy
-#endif
-
-#ifdef HAVE_STRLCAT
-#define strncat strlcat
-#endif
-
 /* Quirks */
 #define QUIRK_S3MLOOP	(1 << 0)	/* S3M loop mode */
 #define QUIRK_ENVFADE	(1 << 1)	/* Fade at end of envelope */
