diff -Naur qsynth-0.3.5-orig/configure qsynth-0.3.5/configure
--- qsynth-0.3.5-orig/configure	2010-05-31 05:25:15.000000000 -0400
+++ qsynth-0.3.5/configure	2010-05-31 05:10:50.000000000 -0400
@@ -1884,7 +1884,7 @@
 
 
 # Standard installation base dirs.
-ac_with_paths="/usr /usr/local"
+ac_with_paths="/usr /usr/local %p"
 
 # Some a-la-debian alternatives...
 for X in /usr/lib /usr/lib64 /usr/share; do

diff -Naur qsynth-0.3.5-orig/src/src.pro qsynth-0.3.5/src/src.pro
--- qsynth-0.3.5-orig/src/src.pro	2010-03-26 10:44:58.000000000 -0400
+++ qsynth-0.3.5/src/src.pro	2010-07-15 10:06:43.000000000 -0400
@@ -4,7 +4,8 @@
 
 TEMPLATE = app
 DEPENDPATH += .
-INCLUDEPATH += .
+INCLUDEPATH += . \
+	@FINKPREFIX@/include
 
 include(src.pri)

