diff -Nurd -x'*~' sphinx3-0.7.orig/configure sphinx3-0.7/configure
--- sphinx3-0.7.orig/configure	2007-08-21 01:58:40.000000000 -0400
+++ sphinx3-0.7/configure	2012-02-14 04:02:54.000000000 -0500
@@ -2483,7 +2483,7 @@
 
 case $host in
      *apple*)
-        LIBS="$LIBS -framework CoreAudio -framework AudioToolbox"
+        LIBS="$LIBS -Wl,-framework,CoreAudio -Wl,-framework,AudioToolbox"
         ;;
      *)
      ;;
