diff -urN xmms_osx_plugin-1.0/src/OSX.c xmms_osx_plugin-1.0_patched/src/OSX.c
--- xmms_osx_plugin-1.0/src/OSX.c	2004-06-14 04:13:23.000000000 +0200
+++ xmms_osx_plugin-1.0_patched/src/OSX.c	2006-03-26 15:07:29.000000000 +0200
@@ -26,7 +26,7 @@
 		NULL, /* Description */
 		osx_init,
 		osx_about,
-		osx_configure,
+		NULL,
 		osx_get_volume,
 		osx_set_volume,
 		osx_open,
diff -urN xmms_osx_plugin-1.0/src/OSX.h xmms_osx_plugin-1.0_patched/src/OSX.h
--- xmms_osx_plugin-1.0/src/OSX.h	2004-06-14 04:13:08.000000000 +0200
+++ xmms_osx_plugin-1.0_patched/src/OSX.h	2006-03-26 15:07:05.000000000 +0200
@@ -85,7 +85,7 @@
 
 void osx_init(void);
 void osx_about(void);
-void osx_configure(void);
+//void osx_configure(void);
 
 void osx_get_volume(int *l, int *r);
 void osx_set_volume(int l, int r);
diff -urN xmms_osx_plugin-1.0/src/audio.c xmms_osx_plugin-1.0_patched/src/audio.c
--- xmms_osx_plugin-1.0/src/audio.c	2006-03-26 13:45:36.000000000 +0200
+++ xmms_osx_plugin-1.0_patched/src/audio.c	2006-03-26 13:46:23.000000000 +0200
@@ -513,6 +513,9 @@
 	base_pitch = 1.0;
 	user_pitch = 1.0;
 
+	// switch this thread to realtime priority if possible
+	xmms_check_realtime_priority();
+	
 	// open audio device
 
 	size = sizeof(device_id);