--- gperiodic-1.3.3/src/Makefile.in	Tue Aug 28 02:04:51 2001
+++ gperiodic-1.3.3-patched/src/Makefile.in	Thu Mar 21 22:09:43 2002
@@ -95,7 +95,7 @@
 bin_PROGRAMS = gperiodic
 
 gperiodic_SOURCES = gperiodic.c cmdline_interface.c gparser.y gplexer.l gpdata.c
-gperiodic_LDADD = -lreadline -ltermcap
+gperiodic_LDADD = -lreadline
 
 INCLUDES = -I$(top_srcdir)/include -DGPERIODIC_LOCALE_DIR=\"$(datadir)/locale\"
 
