--- RooFitModels/GNUmakefile.standalone	Fri Oct  4 16:08:05 2002
+++ RooFitModels.new/GNUmakefile.standalone	Wed Jun 25 23:57:00 2003
@@ -44,28 +44,31 @@
 
 # --- External configuration ---------------------------------
 CC         = g++
-CCFLAGS    = -O2
+CCFLAGS    = -pipe -O
 MFLAGS     = -MM
-INCLUDES   = 
+INCLUDES   = $(shell root-config --cflags)
 WORKDIR    = tmp/
+LDFLAGS    = -O -Xlinker -flat_namespace -bind_at_load -bundle -Wl,-x
+SOFLAGS    = -dynamiclib -flat_namespace -undefined suppress
 LIBDIR     = $(WORKDIR)
 # -------------------------------------------------------------
 
 
 # Internal configuration
 PACKAGE=RooFitModels
-LD_LIBRARY_PATH:=$(ROOTSYS)/lib
+MAJOR  = $(shell echo @VERSION@ | awk -F "." '{printf("%d",$$1)}')
+MINOR  = $(shell echo @VERSION@ | awk -F "." '{printf("%d",$$2)}')
+REVIS  = $(shell echo @VERSION@ | awk -F "." '{printf("%d",$$3)}')
+
 OBJDIR=$(WORKDIR)/objects/
 INCDIR=$(WORKDIR)/$(PACKAGE)/
 VPATH=$(INCDIR) $(OBJDIR)
 
-INCLUDES += -I.. -I$(WORKDIR)/ -I$(ROOTSYS)/include -I$(PWD)/../RooFitCore/$(WORKDIR)
-ROOTSYS  ?= ERROR_RootSysIsNotDefined
+INCLUDES += -I.. -I$(WORKDIR)/
 RDLLIST   = $(filter-out $(PACKAGE)_LinkDef.rdl,$(wildcard *.rdl))
 CINTFILE  = $(PACKAGE)Cint.cc
 CINTOBJ   = $(PACKAGE)Cint.o
-LIBFILE   = $(LIBDIR)/lib$(PACKAGE).a
-SHLIBFILE = $(LIBDIR)/lib$(PACKAGE).so
+SHLIBFILE = $(LIBDIR)/lib$(PACKAGE).dylib
 
 default: shlib
 
@@ -102,22 +105,23 @@
 	@mkdir -p $(INCDIR)
 	@mkdir -p $(OBJDIR)
 	@echo "Running rootcint"
-	@cd $(INCDIR) ; $(ROOTSYS)/bin/rootcint -f $(CINTFILE) -c $(INCLUDES) $(HHLIST)
+	@cd $(INCDIR) ; rootcint -f $(CINTFILE) -c $(INCLUDES) $(HHLIST)
 	@echo "Compiling $(CINTFILE)"
 	@$(CC) $(CCFLAGS) -c $(INCDIR)/$(CINTFILE) -o $(OBJDIR)/$(CINTOBJ) $(INCLUDES)
 
-# Rule to combine objects into a library
-$(LIBFILE): $(OLIST) $(INCDIR)/$(CINTFILE)) $(patsubst %.cc,%.o,$(OBJDIR)/$(CINTFILE))
-	@echo "Making $(LIBFILE)"
-	@rm -f $(LIBFILE)
-	@ar q $(LIBFILE) $(addprefix $(OBJDIR)/,$(OLIST) $(patsubst %.cc,%.o,$(CINTFILE))) 
-	@ranlib $(LIBFILE)
-
 # Rule to combine objects into a shared library
 $(SHLIBFILE): $(OLIST) $(patsubst %.cc,%.o,$(OBJDIR)/$(CINTFILE))
 	@echo "Making $(SHLIBFILE)"
 	@rm -f $(SHLIBFILE)
-	@$(CC) $(addprefix $(OBJDIR)/,$(OLIST) $(patsubst %.cc,%.o,$(CINTFILE))) -shared -o $(SHLIBFILE)
+	@$(CC) $(addprefix $(OBJDIR)/,$(OLIST) $(patsubst \
+	%.cc,%.o,$(CINTFILE))) $(LDFLAGS) `root-config --glibs` -lHtml \
+	-lMinuit -lRooFitCore -o $(subst .dylib,.so,$@) 
+	@$(CC) $(addprefix $(OBJDIR)/,$(OLIST) $(patsubst %.cc,%.o,$(CINTFILE))) \
+	$(SOFLAGS) `root-config --glibs` -lHtml -lMinuit -lRooFitCore \
+	-install_name $(subst $(LIBDIR),@PREFIX@/lib/root,$(subst .dylib,.$(MAJOR).dylib,$@)) \
+	-compatibility_version $(MAJOR) -current_version $(MAJOR).$(MINOR).$(REVIS) \
+	-o $(subst .dylib,.$(MAJOR).$(MINOR).dylib,$@) 
+
 
 # Useful build targets
 lib: $(LIBFILE) 
@@ -125,9 +129,8 @@
 clean:
 	rm -f $(OBJDIR)/*
 	rm -f $(INCDIR)/*
-	rm -f $(LIBFILE)
 	rm -f $(SHLIBFILE)
 
-.PHONY : shlib lib default clean
+.PHONY : shlib default clean
 
 -include $(addprefix $(INCDIR)/,$(DLIST))
