diff -Nurd -x'*~' xbae-4.60.4.orig/Makefile.in xbae-4.60.4/Makefile.in
--- xbae-4.60.4.orig/Makefile.in	2006-05-29 10:02:12.000000000 -0400
+++ xbae-4.60.4/Makefile.in	2007-06-25 22:05:14.000000000 -0400
@@ -230,14 +230,14 @@
 x_includes = @x_includes@
 x_libraries = @x_libraries@
 MAINTAINERCLEANFILES = Makefile.in configure libtool config.guess config.sub ltconfig ltmain.sh aclocal.m4 config.h.in install-sh missing mkinstalldirs
-SUBDIRS = include src examples doc scripts
+SUBDIRS = include src doc scripts
 EXTRA_DIST = configure ac_find_xbae.m4 Imakefile Xbae.tmpl Xbae.rules \
 	CVSMake
 
 aclocaldir = @ACLOCALDIR@
 @Aclocal_TRUE@aclocal_DATA = ac_find_xbae.m4
-docdir = $(datadir)/Xbae
-doc_DATA = NEWS README
+docdir = $(datadir)/xbae
+doc_DATA = AUTHORS ChangeLog COPYING NEWS README
 
 #
 # Some convenient targets for users of Exuberant ctags
diff -Nurd -x'*~' xbae-4.60.4.orig/configure xbae-4.60.4/configure
--- xbae-4.60.4.orig/configure	2006-05-29 10:02:05.000000000 -0400
+++ xbae-4.60.4/configure	2007-06-25 22:06:04.000000000 -0400
@@ -21986,10 +21986,8 @@
 
 fi
 
-if (aclocal --print-ac-dir) </dev/null >/dev/null 2>&1 && test -w `aclocal --print-ac-dir`; then
-    ACLOCALDIR=`aclocal --print-ac-dir`
+    ACLOCALDIR="${prefix}/share/aclocal"
 
-fi
 
 
 if test x$ACLOCALDIR != x ; then
diff -Nurd -x'*~' xbae-4.60.4.orig/doc/Makefile.in xbae-4.60.4/doc/Makefile.in
--- xbae-4.60.4.orig/doc/Makefile.in	2006-05-29 10:02:06.000000000 -0400
+++ xbae-4.60.4/doc/Makefile.in	2007-06-25 22:45:01.000000000 -0400
@@ -224,7 +224,7 @@
 # Lets put the informational stuff in a sub directory to avoid 
 # polluting /usr/local for default installs.
 #
-docdir = $(datadir)/Xbae
+docdir = $(datadir)/xbae
 doc_DATA = \
       contents.html coverplus.html difffonts.html \
       index.html main.html morelines.html \
diff -Nurd -x'*~' xbae-4.60.4.orig/doc/images/Makefile.in xbae-4.60.4/doc/images/Makefile.in
--- xbae-4.60.4.orig/doc/images/Makefile.in	2006-05-29 10:02:06.000000000 -0400
+++ xbae-4.60.4/doc/images/Makefile.in	2007-06-25 22:52:28.000000000 -0400
@@ -213,7 +213,7 @@
 # Lets put the informational stuff in a sub directory to avoid 
 # polluting /usr/local for default installs.
 #
-imagedir = $(datadir)/Xbae/images
+imagedir = $(datadir)/xbae/images
 
 #
 # Cut the distribution size, don't distribute these in a source
