diff -ruN antispam-0.11.orig/Makefile antispam-0.11/Makefile
--- antispam-0.11.orig/Makefile	Wed Feb 28 09:02:19 2001
+++ antispam-0.11/Makefile	Fri Feb 28 12:06:44 2003
@@ -7,29 +7,27 @@
 
 
 # Uncomment this if your system has a daemon function
-#DEFINES= $(DEFINES) -D _HAVE_DAEMON
+# DEFINES= -D_HAVE_DAEMON=1
 
-CFLAGS= -O3 -m486 $(DEFINES) -D VERSION=$(VERSION) -D REVISION=$(REVISION) -D _HAVE_$(DATABASE)
-LDFLAGS= -ldb
+CFLAGS= -g -Os $(DEFINES) -DVERSION=$(VERSION) -DREVISION=$(REVISION) -D_HAVE_$(DATABASE) -I@FINKPREFIX@/include
+LDFLAGS= -L@FINKPREFIX@/lib -ldb
 OBJS_ANTISPAM=antispam.o hostcheck.o util.o database.o config.o
 EXECS_ANTISPAM=antispam dumphash
 all: $(EXECS_ANTISPAM)
 
 install: all
-	install -o root -g root -m 700 antispam /usr/sbin/
-	install -o root -g root -m 700 antispam.cfg-sample /etc/
-	install -o root -g root antispam.8 /usr/man/man8/
-	if [ -d /etc/rc.d/init.d ]; then \
-		install -o root -g root antispam.init /etc/rc.d/init.d/ ; \
-	fi		
-	if [ -f /etc/rc.d/init.d/antispam.init ]; then \
-		ln -sf /etc/rc.d/init.d/antispam.init /etc/rc.d/rc3.d/S99antispam ;\
-		ln -sf /etc/rc.d/init.d/antispam.init /etc/rc.d/rc6.d/K99antispam ;\
-		ln -sf /etc/rc.d/init.d/antispam.init /etc/rc.d/rc0.d/K99antispam ;\
-	fi
-	if [ -d /etc/logrotate.d ]; then \
-		install -o root -g root antispam.conf /etc/logrotate.d/ ; \
+	install -d -o root -g wheel @FINKINSTALL@/sbin/
+	install -o root -g wheel -m 755 antispam @FINKINSTALL@/sbin/
+	install -d -o root -g wheel @FINKINSTALL@/etc/
+	install -o root -g wheel -m 644 antispam.cfg-sample @FINKINSTALL@/etc/antispam.cfg
+	install -d -o root -g wheel @FINKINSTALL@/share/man/man8/
+	install -o root -g wheel -m 644 antispam.8 @FINKINSTALL@/share/man/man8/
+	if [ -d @FINKPREFIX@/etc/logrotate.d ]; then \
+		install -d -o root -g wheel @FINKINSTALL@/etc/logrotate.d/ ; \
+		install -o root -g wheel antispam.conf @FINKINSTALL@/etc/logrotate.d/ ; \
 	fi		
+	install -d -o root -g wheel @FINKINSTALL@/var/log/
+	install -d -o root -g wheel @FINKINSTALL@/var/run/
 
 antispam.o: antispam.c hostcheck.h util.h database.h config.h
 hostcheck.o: hostcheck.c hostcheck.h
diff -ruN antispam-0.11.orig/antispam.8 antispam-0.11/antispam.8
--- antispam-0.11.orig/antispam.8	Mon Feb 26 15:21:40 2001
+++ antispam-0.11/antispam.8	Fri Feb 28 12:04:28 2003
@@ -25,7 +25,7 @@
 .SH CONFIGURATION
 .PP
 The progam has a config file (
-.B /etc/antispam.cfg
+.B @FINKPREFIX@/etc/antispam.cfg
 ) which it will open upon startup. Options on that file are specified on a
 one-per-line basis. Paramenters with more than one word must be double
 quoted (example:  log "my anti spamming log"). Lines beginning with '#' are
@@ -35,7 +35,7 @@
 .TP
 \fBlog <logfile>\fR
 This is the file where \fBantispam\fR will record all it's activity. If not
-specified defaults to /var/log/antispam
+specified defaults to @FINKPREFIX@/var/log/antispam
 .TP
 \fBmaillog <logfile>\fR
 This is the file where \fBantispam\fR will look for all successful POP3
@@ -90,9 +90,7 @@
 authorization.
 .SH INSTALLATION
 .SS Setting up the files
-Create your config file, and put the \fBantispam\fR in /usr/sbin.
-You can also copy the \fBantispam.init\fR script to your /etc/rc.d/init.d/
-directory.
+Create your config file, and put the \fBantispam\fR in @FINKPREFIX@/sbin.
 .SS Configuring the mailer
 This is the tricky part. I use sendmail (no flames please). I changed my
 sendmail.cf by including the following line:
@@ -152,11 +150,11 @@
 R$*<@$+>$*              $#error $@ 5.7.1 $: 551 we do not relay
 .nf
 .SS Getting it up and running
-Just call \fB/usr/sbin/antispam\fR. If everything's ok with the config file
+Just call \fB@FINKPREFIX@/sbin/antispam\fR. If everything's ok with the config file
 it will become a daemon, fork a 'tail -f' to your maillog file and start
 working.
 .PP
-Do a 'tail -f /var/log/antispam' (or whatever log file you named in the
+Do a 'tail -f @FINKPREFIX@/var/log/antispam' (or whatever log file you named in the
 config) and do a POP on the server. If the machine you "popped" from isn't
 on the "ignore list" you should see something like:
 .PP
@@ -182,15 +180,15 @@
 (send it a TERM signal: 'killall -TERM antispam'), rotate the logs and launch
 it again.
 .PP
-I usually create a antispam.conf file on the /etc/logrotate.d/ directory
+I usually create a antispam.conf file on the @FINKPREFIX@/etc/logrotate.d/ directory
 containing:
 .PP
 .nf
 .ta +3i
-/var/log/antispam {
+@FINKPREFIX@/var/log/antispam {
   daily
   postrotate
-    /etc/rc.d/init.d/antispam.init restart > /dev/null
+    daemonic restart antispam > /dev/null
   endscript
 }
 .fi
@@ -202,9 +200,9 @@
 
 .SH "FILES"
 .PP
-/etc/antispam.cfg
-/var/run/antispam.db
-/var/log/antispam
+@FINKPREFIX@/etc/antispam.cfg
+@FINKPREFIX@/var/run/antispam.db
+@FINKPREFIX@/var/log/antispam
 /etc/mail/ip_allow.db
 
 .SH ACKOWLEDGMENTS
diff -ruN antispam-0.11.orig/antispam.c antispam-0.11/antispam.c
--- antispam-0.11.orig/antispam.c	Wed Feb 28 09:05:31 2001
+++ antispam-0.11/antispam.c	Fri Feb 28 11:43:29 2003
@@ -115,7 +115,7 @@
   char *ipend;
   int pipes[2];
   
-  if (setOptions("/etc/antispam.cfg")) /* Read the config file */
+  if (setOptions("@FINKPREFIX@/etc/antispam.cfg")) /* Read the config file */
     exit(1);
 #ifdef _HAVE_DAEMON
   if (daemon(0,0)<0) /* become a daemon */
diff -ruN antispam-0.11.orig/antispam.cfg-sample antispam-0.11/antispam.cfg-sample
--- antispam-0.11.orig/antispam.cfg-sample	Wed Dec 15 09:48:31 1999
+++ antispam-0.11/antispam.cfg-sample	Fri Feb 28 11:40:17 2003
@@ -1,13 +1,13 @@
 # Configuration file for AntiSpam
 
-log /var/log/antispam # were we will log our authorizations
+log @FINKPREFIX@/var/log/antispam # were we will log our authorizations
 
 maillog /var/log/maillog # were we will look for pop requests
 
 ip_allow /etc/mail/ip_allow.db # were will we inform sendmail of the
                                # authorizations
                      
-runtimedb /var/run/antispam.db # were we will keep the temporary IPs we
+runtimedb @FINKPREFIX@/var/run/antispam.db # were we will keep the temporary IPs we
                                # authorized
 
 timeout 15     # expiry time for an authorization
diff -ruN antispam-0.11.orig/antispam.conf antispam-0.11/antispam.conf
--- antispam-0.11.orig/antispam.conf	Wed Dec 15 09:47:28 1999
+++ antispam-0.11/antispam.conf	Fri Feb 28 11:32:52 2003
@@ -1,6 +1,6 @@
-/var/log/antispam {
+@FINKPREFIX@/var/log/antispam {
   daily
   postrotate
-    /etc/rc.d/init.d/antispam.init restart > /dev/null
+    @FINKPREFIX@/bin/daemonic restart antispam
   endscript
 }
diff -ruN antispam-0.11.orig/config.c antispam-0.11/config.c
--- antispam-0.11.orig/config.c	Thu Nov 11 08:28:31 1999
+++ antispam-0.11/config.c	Fri Feb 28 11:44:36 2003
@@ -318,14 +318,14 @@
   }
   if (!ourlog)
   {
-    conferror("'log' not specified. Assuming '/var/log/antispam'",filename,0);
-    ourlog=strdup("/var/log/antispam");    
+    conferror("'log' not specified. Assuming '@FINKPREFIX@/var/log/antispam'",filename,0);
+    ourlog=strdup("@FINKPREFIX@/var/log/antispam");    
   }
 
   if (!ourdb)
   {
-    conferror("'runtimedb' not specified. Assuming '/var/run/antispam.db'",filename,0);
-    ourdb=strdup("/var/run/antispam.db");
+    conferror("'runtimedb' not specified. Assuming '@FINKPREFIX@/var/run/antispam.db'",filename,0);
+    ourdb=strdup("@FINKPREFIX@/var/run/antispam.db");
   }
   
   if (!sendmaildb)
