diff -ruN blootbot.orig/blootbot blootbot/blootbot
--- blootbot.orig/blootbot	Mon Oct 15 07:00:05 2001
+++ blootbot/blootbot	Tue Feb 25 12:54:58 2003
@@ -16,17 +16,11 @@
     }
 
     # set any $bot_*_dir var's that aren't already set
-    $bot_base_dir	||= '.';
-    $bot_config_dir	||= 'files/';
-    $bot_data_dir	||= 'files/';
-    $bot_state_dir	||= 'files/';
-    $bot_run_dir	||= '.';
-    $bot_src_dir	||= "$bot_base_dir/src";
-    $bot_log_dir	||= "$bot_base_dir/log";
-    $bot_misc_dir	||= "$bot_base_dir/files";
-
+    $bot_base_dir	||= '@FINKPREFIX@';
+    $bot_config_dir	||= '@FINKPREFIX@/etc/blootbot/';
     $bot_pid		= $$;
 
+    require "$bot_config_dir"."blootbot.paths";
     require "$bot_src_dir/logger.pl";
     require "$bot_src_dir/core.pl";
     require "$bot_src_dir/interface.pl";
diff -ruN blootbot.orig/files/blootbot.paths blootbot/files/blootbot.paths
--- blootbot.orig/files/blootbot.paths	Wed Dec 31 17:00:00 1969
+++ blootbot/files/blootbot.paths	Tue Feb 25 12:59:48 2003
@@ -0,0 +1,6 @@
+    $bot_data_dir       ||= '@FINKPREFIX@/etc/blootbot/';
+    $bot_state_dir      ||= '@FINKPREFIX@/etc/blootbot/';
+    $bot_run_dir        ||= '@FINKPREFIX@/var/run/blootbot';
+    $bot_src_dir        ||= "@FINKPREFIX@/share/blootbot/src";
+    $bot_log_dir        ||= "@FINKPREFIX@/var/log/blootbot/";
+    $bot_misc_dir       ||= "@FINKPREFIX@/share/blootbot/";
diff -ruN blootbot.orig/setup/setup.pl blootbot/setup/setup.pl
--- blootbot.orig/setup/setup.pl	Mon Feb 18 07:00:07 2002
+++ blootbot/setup/setup.pl	Tue Jan 14 20:31:41 2003
@@ -3,16 +3,16 @@
 # written by the xk.
 ###
 
-require "src/core.pl";
-require "src/logger.pl";
-require "src/modules.pl";
-require "src/Misc.pl";
-require "src/interface.pl";
+require "@FINKPREFIX@/share/blootbot/src/core.pl";
+require "@FINKPREFIX@/share/blootbot/src/logger.pl";
+require "@FINKPREFIX@/share/blootbot/src/modules.pl";
+require "@FINKPREFIX@/share/blootbot/src/Misc.pl";
+require "@FINKPREFIX@/share/blootbot/src/interface.pl";
 
-$bot_src_dir = "./src/";
+$bot_src_dir = "@FINKPREFIX@/share/blootbot/src/";
 
 # read param stuff from blootbot.config.
-&loadConfig("files/blootbot.config");
+&loadConfig("@FINKPREFIX@/etc/blootbot/blootbot.config");
 &loadDBModules();
 my $dbname = $param{'DBName'};
 my $query;
diff -ruN blootbot.orig/src/Modules/Slashdot3.pl blootbot/src/Modules/Slashdot3.pl
--- blootbot.orig/src/Modules/Slashdot3.pl	Sat Feb  3 09:01:29 2001
+++ blootbot/src/Modules/Slashdot3.pl	Tue Feb 25 11:20:37 2003
@@ -33,7 +33,7 @@
 }
 
 sub Slashdot {
-    my @results = &::getURL("http://www.slashdot.org/slashdot.xml");
+    my @results = &::getURL("http://slashdot.org/slashdot.xml");
     my $retval  = "i could not get the headlines.";
 
     if (scalar @results) {
@@ -48,7 +48,7 @@
 sub slashdotAnnounce {
     my $file = "$::param{tempDir}/slashdot.xml";
 
-    my @Cxml = &::getURL("http://www.slashdot.org/slashdot.xml");
+    my @Cxml = &::getURL("http://slashdot.org/slashdot.xml");
     if (!scalar @Cxml) {
 	&::DEBUG("sdA: failure (Cxml == NULL).");
 	return;
diff -ruN blootbot.orig/src/Modules/Units.pl blootbot/src/Modules/Units.pl
--- blootbot.orig/src/Modules/Units.pl	Fri Oct  5 07:00:48 2001
+++ blootbot/src/Modules/Units.pl	Tue Jan 14 20:39:07 2003
@@ -49,7 +49,7 @@
 ################################################################
 
 { my $defs_read = 0;
-  $defs_read += read_defs("$::bot_data_dir/unittab");
+  $defs_read += read_defs("$::bot_data_dir/blootbot.unittab");
 
   unless ($defs_read) {
     &::ERROR("Could not read any of the initialization files UNITTAB");
