diff -ruN blootbot.orig/blootbot blootbot/blootbot
--- blootbot.orig/blootbot	Mon Oct 15 07:00:05 2001
+++ blootbot/blootbot	Tue Jan 14 20:28:09 2003
@@ -16,14 +16,14 @@
     }
 
     # set any $bot_*_dir var's that aren't already set
-    $bot_base_dir	||= '.';
-    $bot_config_dir	||= 'files/';
-    $bot_data_dir	||= 'files/';
-    $bot_state_dir	||= 'files/';
-    $bot_run_dir	||= '.';
-    $bot_src_dir	||= "$bot_base_dir/src";
-    $bot_log_dir	||= "$bot_base_dir/log";
-    $bot_misc_dir	||= "$bot_base_dir/files";
+    $bot_base_dir	||= '@FINKPREFIX@';
+    $bot_config_dir	||= '$bot_base_dir/etc/blootbot/';
+    $bot_data_dir	||= '$bot_base_dir/var/blootbot/';
+    $bot_state_dir	||= '$bot_base_dir/var/blootbot/';
+    $bot_run_dir	||= '$bot_base_dir/var/run/';
+    $bot_src_dir	||= "$bot_base_dir/share/blootbot/src";
+    $bot_log_dir	||= "$bot_base_dir/var/log/blootbot/";
+    $bot_misc_dir	||= "$bot_base_dir/share/blootbot/";
 
     $bot_pid		= $$;
 
diff -ruN blootbot.orig/setup/setup.pl blootbot/setup/setup.pl
--- blootbot.orig/setup/setup.pl	Mon Feb 18 07:00:07 2002
+++ blootbot/setup/setup.pl	Tue Jan 14 20:31:41 2003
@@ -3,16 +3,16 @@
 # written by the xk.
 ###
 
-require "src/core.pl";
-require "src/logger.pl";
-require "src/modules.pl";
-require "src/Misc.pl";
-require "src/interface.pl";
+require "@FINKPREFIX@/share/blootbot/src/core.pl";
+require "@FINKPREFIX@/share/blootbot/src/logger.pl";
+require "@FINKPREFIX@/share/blootbot/src/modules.pl";
+require "@FINKPREFIX@/share/blootbot/src/Misc.pl";
+require "@FINKPREFIX@/share/blootbot/src/interface.pl";
 
-$bot_src_dir = "./src/";
+$bot_src_dir = "@FINKPREFIX@/share/blootbot/src/";
 
 # read param stuff from blootbot.config.
-&loadConfig("files/blootbot.config");
+&loadConfig("@FINKPREFIX@/etc/blootbot/blootbot.config");
 &loadDBModules();
 my $dbname = $param{'DBName'};
 my $query;
diff -ruN blootbot.orig/src/Modules/Units.pl blootbot/src/Modules/Units.pl
--- blootbot.orig/src/Modules/Units.pl	Fri Oct  5 07:00:48 2001
+++ blootbot/src/Modules/Units.pl	Tue Jan 14 20:39:07 2003
@@ -49,7 +49,7 @@
 ################################################################
 
 { my $defs_read = 0;
-  $defs_read += read_defs("$::bot_data_dir/unittab");
+  $defs_read += read_defs("$::bot_data_dir/blootbot.unittab");
 
   unless ($defs_read) {
     &::ERROR("Could not read any of the initialization files UNITTAB");
