diff -ruN ClanLib-0.6.5.orig/Documentation/Examples/Makefile.in ClanLib-0.6.5/Documentation/Examples/Makefile.in
--- ClanLib-0.6.5.orig/Documentation/Examples/Makefile.in	2002-01-24 10:44:44.000000000 -0700
+++ ClanLib-0.6.5/Documentation/Examples/Makefile.in	2002-11-04 12:38:15.000000000 -0700
@@ -1,6 +1,6 @@
-prefix=@prefix@
+prefix=@prefix@/share
 
-EXAMPLE_SRC_PREFIX = $(prefix)/doc/ClanLib-docs-@version_minor@/Examples
+EXAMPLE_SRC_PREFIX = $(prefix)/clanlib/Examples
 
 EXAMPLES= @example_default@ @example_asm@ @example_opengl@ @example_gui@ @example_network@ @example_lua@ @example_mikmod@ @example_jpeg@ @example_vorbis@ @example_ttf@ 
 
diff -ruN ClanLib-0.6.5.orig/Documentation/Makefile.in ClanLib-0.6.5/Documentation/Makefile.in
--- ClanLib-0.6.5.orig/Documentation/Makefile.in	2002-03-28 17:45:31.000000000 -0700
+++ ClanLib-0.6.5/Documentation/Makefile.in	2002-11-04 12:38:15.000000000 -0700
@@ -1,6 +1,4 @@
-prefix=@prefix@/share/
-
-HTML_PREFIX = $(prefix)/doc/ClanLib-docs-@version_minor@/Documentation
+HTML_PREFIX = $(prefix)/share/doc/clanlib/html
 
 all: faq
 	Utilities/webbuilder.pl documentation.theme index.xml
@@ -19,8 +17,8 @@
 	@install -m 0644 *.html $(HTML_PREFIX)
 	
 #	cd Reference; make man_install
-	cd Reference; make html_install
-	cd Overview; make install
+	cd Reference; make html_install prefix=$(prefix)
+	cd Overview; make install prefix=$(prefix)
 #	cd Tutorial; make install
 
 publish : all
diff -ruN ClanLib-0.6.5.orig/Documentation/Overview/Makefile.in ClanLib-0.6.5/Documentation/Overview/Makefile.in
--- ClanLib-0.6.5.orig/Documentation/Overview/Makefile.in	2002-02-28 15:03:20.000000000 -0700
+++ ClanLib-0.6.5/Documentation/Overview/Makefile.in	2002-11-04 12:38:15.000000000 -0700
@@ -1,6 +1,4 @@
-prefix=@prefix@/share
-
-HTML_PREFIX = $(prefix)/doc/ClanLib-docs-@version_minor@/Documentation
+HTML_PREFIX = $(prefix)/share/doc/clanlib/html
 
 all:	index.html
 
diff -ruN ClanLib-0.6.5.orig/Documentation/Reference/Makefile.in ClanLib-0.6.5/Documentation/Reference/Makefile.in
--- ClanLib-0.6.5.orig/Documentation/Reference/Makefile.in	2002-02-28 15:03:20.000000000 -0700
+++ ClanLib-0.6.5/Documentation/Reference/Makefile.in	2002-11-04 12:38:15.000000000 -0700
@@ -1,13 +1,11 @@
-
-prefix=@prefix@/share
-HTML_PREFIX = $(prefix)/doc/ClanLib-docs-@version_minor@/Documentation
+HTML_PREFIX = $(prefix)/share/doc/clanlib/html
 
 all:	html
 
 html:
 	@echo 'Type "make html_install" to install the html reference pages.'
 	if [ ! -d html ]; then mkdir html; fi
-	chmod +x ./pce2
+	chmod 775 ./pce2
 	find ../../Sources/API -name '*.h' | xargs ./pce2
 
 html_install:
diff -ruN ClanLib-0.6.5.orig/Documentation/Tutorial/Makefile.in ClanLib-0.6.5/Documentation/Tutorial/Makefile.in
--- ClanLib-0.6.5.orig/Documentation/Tutorial/Makefile.in	2002-06-29 02:32:45.000000000 -0600
+++ ClanLib-0.6.5/Documentation/Tutorial/Makefile.in	2002-11-04 12:38:15.000000000 -0700
@@ -1,5 +1,6 @@
 prefix=@prefix@/share
-PREFIX = $(prefix)/doc/ClanLib-docs-@version_minor@/Docs/Tutorial
+
+PREFIX = $(prefix)/clanlib/html/Tutorial
 
 all:
 
diff -ruN ClanLib-0.6.5.orig/Makefile.in ClanLib-0.6.5/Makefile.in
--- ClanLib-0.6.5.orig/Makefile.in	2002-03-28 17:45:30.000000000 -0700
+++ ClanLib-0.6.5/Makefile.in	2002-11-04 12:38:15.000000000 -0700
@@ -84,12 +84,12 @@
 	@echo "Removing header files from $(INC_PREFIX)."
 	@rm -rf $(INC_PREFIX)/ClanLib/
 	@echo "Removing libraries from $(LIB_PREFIX)."
-	@rm -f $(LIB_PREFIX)/libclanCore.so
-	@rm -f $(LIB_PREFIX)/libclanCore.so.$(D_VERSION_MAJOR)
-	@rm -f $(LIB_PREFIX)/libclanCore.so.$(D_VERSION_MINOR)
-	-@rm -f $(LIB_PREFIX)/libclanGL.so
-	-@rm -f $(LIB_PREFIX)/libclanGL.so.$(D_VERSION_MAJOR)
-	-@rm -f $(LIB_PREFIX)/libclanGL.so.$(D_VERSION_MINOR)
+	@rm -f $(LIB_PREFIX)/libclanCore.dylib
+	@rm -f $(LIB_PREFIX)/libclanCore.$(D_VERSION_MAJOR).dylib
+	@rm -f $(LIB_PREFIX)/libclanCore.$(D_VERSION_MINOR).dylib
+	-@rm -f $(LIB_PREFIX)/libclanGL.dylib
+	-@rm -f $(LIB_PREFIX)/libclanGL.$(D_VERSION_MAJOR).dylib
+	-@rm -f $(LIB_PREFIX)/libclanGL.$(D_VERSION_MINOR).dylib
 	@echo "Removing display targets from $(TARGET_PREFIX)."
 	@rm -rf $(TARGET_PREFIX)
 	@rm -f $(BIN_PREFIX)/clanlib-config
@@ -97,7 +97,7 @@
 strip:
 	@echo ""
 	@echo "stripping library files..."
-	@strip Libs/*.so*
+	@strip Libs/*.dylib
 
 docs: configure docs_compile
 	@echo ""
@@ -110,7 +110,7 @@
 	cd Documentation; make
 
 docs_install: docs_compile
-	cd Documentation; make install
+	cd Documentation; make install prefix=$(prefix)
 
 utils:  core_libs
 	@cd Utils; make
@@ -159,7 +159,7 @@
 		-path "*[Ww]in32" -prune -o \
 		-path "*MIDI*" -prune -o \
 		-path "*MPEG*" -prune -o \
-	-name *.cpp -print | sed "s/^.\///;" |  xargs g++ -w -M $(INCLUDE_DIRS)| sed "s/\(.*\.o:\)/Libs\/Intermediate\/\1/g;" > Setup/Unix/Makefile.dep 
+	-name *.cpp -print | sed "s/^.\///;" |  xargs c++ -w -M $(INCLUDE_DIRS)| sed "s/\(.*\.o:\)/Libs\/Intermediate\/\1/g;" > Setup/Unix/Makefile.dep 
 
 # Temporary hack:
 dep_beos:
@@ -174,7 +174,7 @@
 		-name '*.cpp' | \
 	grep "\.cpp" | \
 	sed "s/^.\///;" | \
-	xargs g++ $(COMP_OPTIONS) -M | \
+	xargs c++ $(COMP_OPTIONS) -M | \
 	sed "s/\(.*\.o:\)/Libs\/Intermediate\/\1/g;" > Setup/Unix/Makefile.dep
 
 include Setup/Unix/Makefile.dep
diff -ruN ClanLib-0.6.5.orig/Setup/Generic/project.cpp ClanLib-0.6.5/Setup/Generic/project.cpp
--- ClanLib-0.6.5.orig/Setup/Generic/project.cpp	2002-11-02 12:57:05.000000000 -0700
+++ ClanLib-0.6.5/Setup/Generic/project.cpp	2002-11-04 12:38:15.000000000 -0700
@@ -125,12 +125,12 @@
 		#endif
 
 		for (int i=0; exclude_from_build[i] != NULL; i++)
-			if (stricmp(data.cFileName, exclude_from_build[i]) == 0) skip = true;
+			if (strcasecmp(data.cFileName, exclude_from_build[i]) == 0) skip = true;
 
 		std::list<std::string>::const_iterator it;
 		for (it = ignore_list.begin(); it != ignore_list.end(); it++)
 		{
-			if (stricmp(data.cFileName, it->c_str()) == 0)
+			if (strcasecmp(data.cFileName, it->c_str()) == 0)
       	skip = true;
 		}
 
diff -ruN ClanLib-0.6.5.orig/Setup/Tests/lua.cpp ClanLib-0.6.5/Setup/Tests/lua.cpp
--- ClanLib-0.6.5.orig/Setup/Tests/lua.cpp	2001-12-05 05:57:18.000000000 -0700
+++ ClanLib-0.6.5/Setup/Tests/lua.cpp	2002-11-04 12:38:15.000000000 -0700
@@ -17,6 +17,7 @@
 
 #include <lua.h>
 #include <stdio.h>
+#include <string.h>
 #include <tolua.h>
 #undef class
 
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.app.in ClanLib-0.6.5/Setup/Unix/Makefile.app.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.app.in	2001-12-24 09:12:59.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.app.in	2002-11-04 12:38:15.000000000 -0700
@@ -2,26 +2,26 @@
 # Makefile data for clanApp library.
 #
 
-clanApp: Libs/libclanApp.so Libs/libclanApp.a
+clanApp: Libs/libclanApp.dylib Libs/libclanApp.a
 
 installApp: clanApp
 	@for i in `find Sources/API/Application -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/Application -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/application.h $(INC_PREFIX)/ClanLib/application.h
+	@for i in `find Sources/API/Application -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/application.h $(INC_PREFIX)/ClanLib/application.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanApp.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanApp.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanApp.a $(LIB_PREFIX)
-	@ln -s -f libclanApp.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanApp.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanApp.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanApp.so
+	@ln -s -f libclanApp.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanApp.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanApp.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanApp.dylib
 
 
 OBJF_LIB_App = Libs/Intermediate/clanapp.o
 
-Libs/libclanApp.so: Libs/libclanCore.so $(OBJF_LIB_App)
-	@echo "Building Libs/libclanApp.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanApp.so.$(D_VERSION_MAJOR) -o Libs/libclanApp.so.$(D_VERSION_MINOR) $(OBJF_LIB_App) -L Libs -lclanCore
-	@ln -s -f libclanApp.so.$(D_VERSION_MINOR) Libs/libclanApp.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanApp.so.$(D_VERSION_MAJOR) Libs/libclanApp.so
+Libs/libclanApp.dylib: Libs/libclanCore.dylib $(OBJF_LIB_App)
+	@echo "Building Libs/libclanApp.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanApp.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanApp.$(D_VERSION_MINOR).dylib $(OBJF_LIB_App) -LLibs -lclanCore
+	@ln -s -f libclanApp.$(D_VERSION_MINOR).dylib Libs/libclanApp.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanApp.$(D_VERSION_MAJOR).dylib Libs/libclanApp.dylib
 
 Libs/libclanApp.a: $(OBJF_LIB_App)
 	@echo "Building Libs/libclanApp.a"
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.conf.in ClanLib-0.6.5/Setup/Unix/Makefile.conf.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.conf.in	2002-04-06 05:54:58.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.conf.in	2002-11-04 12:38:15.000000000 -0700
@@ -13,7 +13,7 @@
 BIN_PREFIX    = @bindir@
 TARGET_PREFIX = @libdir@/ClanLib
 
-INCLUDE_DIRS  = -I Sources @x_includes@ @extra_includes@
+INCLUDE_DIRS  = -ISources @x_includes@ @extra_includes@
 
 COMP_OPTIONS  = -Wall $(INCLUDE_DIRS) -fPIC -DNOCONTROLS -D__int64=int64_t @DEFS@ @comp_mode@ @directfb_cflags@
 
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.core.in ClanLib-0.6.5/Setup/Unix/Makefile.core.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.core.in	2002-01-08 14:58:35.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.core.in	2002-11-04 12:38:15.000000000 -0700
@@ -2,21 +2,21 @@
 # Makefile data for clanCore library.
 #
 
-clanCore: build_info Libs/libclanCore.so Libs/libclanCore.a
+clanCore: build_info Libs/libclanCore.dylib Libs/libclanCore.a
 
 installCore: clanCore
 	@for i in `find Sources/API/Core -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/Core -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@for i in `find Sources/API/Core -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
 	@for i in `find Sources/API/Sound -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/Sound -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/core.h $(INC_PREFIX)/ClanLib/core.h
-	@install -m 0644 Sources/API/sound.h $(INC_PREFIX)/ClanLib/sound.h
-	@install -m 0644 Sources/API/efence.h $(INC_PREFIX)/ClanLib/efence.h
+	@for i in `find Sources/API/Sound -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/core.h $(INC_PREFIX)/ClanLib/core.h
+	@install -c -m 0644 Sources/API/sound.h $(INC_PREFIX)/ClanLib/sound.h
+	@install -c -m 0644 Sources/API/efence.h $(INC_PREFIX)/ClanLib/efence.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanCore.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanCore.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanCore.a $(LIB_PREFIX)
-	@ln -s -f libclanCore.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanCore.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanCore.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanCore.so
+	@ln -s -f libclanCore.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanCore.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanCore.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanCore.dylib
 	@install -d $(BIN_PREFIX)
 	@install Setup/Unix/clanlib-config $(BIN_PREFIX)
 
@@ -90,11 +90,11 @@
 
 LINK_CORE = @LIBS@
 
-Libs/libclanCore.so: $(OBJF_CORE_ALL)
-	@echo "Building Libs/libclanCore.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanCore.so.$(D_VERSION_MAJOR) -o Libs/libclanCore.so.$(D_VERSION_MINOR) $(OBJF_CORE_ALL) $(LINK_CORE)
-	@ln -s -f libclanCore.so.$(D_VERSION_MINOR) Libs/libclanCore.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanCore.so.$(D_VERSION_MAJOR) Libs/libclanCore.so
+Libs/libclanCore.dylib: $(OBJF_CORE_ALL)
+	@echo "Building Libs/libclanCore.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanCore.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanCore.$(D_VERSION_MINOR).dylib $(OBJF_CORE_ALL) $(LINK_CORE)
+	@ln -s -f libclanCore.$(D_VERSION_MINOR).dylib Libs/libclanCore.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanCore.$(D_VERSION_MAJOR).dylib Libs/libclanCore.dylib
 
 Libs/libclanCore.a: $(OBJF_CORE_ALL)
 	@echo "Building Libs/libclanCore.a"
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.display.in ClanLib-0.6.5/Setup/Unix/Makefile.display.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.display.in	2001-12-24 09:12:59.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.display.in	2002-11-04 12:38:15.000000000 -0700
@@ -1,17 +1,17 @@
 # Makefile data for clanDisplay library.
 #
 
-clanDisplay: Libs/libclanDisplay.so Libs/libclanDisplay.a
+clanDisplay: Libs/libclanDisplay.dylib Libs/libclanDisplay.a
 
 installDisplay: clanDisplay
 	@for i in `find Sources/API/Display -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/Display -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/display.h $(INC_PREFIX)/ClanLib/display.h
+	@for i in `find Sources/API/Display -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/display.h $(INC_PREFIX)/ClanLib/display.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanDisplay.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanDisplay.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanDisplay.a $(LIB_PREFIX)
-	@ln -s -f libclanDisplay.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanDisplay.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanDisplay.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanDisplay.so
+	@ln -s -f libclanDisplay.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanDisplay.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanDisplay.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanDisplay.dylib
 
 
 OBJF_DISP_GENERIC = \
@@ -45,7 +45,6 @@
 	Libs/Intermediate/sprite_subsection_provider.o \
 	Libs/Intermediate/setupdisplay_generic.o \
 	Libs/Intermediate/target_generic.o \
-	Libs/Intermediate/init_linux.o \
 	Libs/Intermediate/input.o \
 	Libs/Intermediate/inputbuffer.o \
 	Libs/Intermediate/generic_surfaceprovider.o \
@@ -114,13 +113,13 @@
 
 # $(OBJF_DISP_X11) $(OBJF_DISP_FBDEV)
 
-LINK_DISP = -L Libs -lclanCore @x_libraries@ -lHermes @directfb_libs@
+LINK_DISP = -LLibs -lclanCore -lclanApp @x_libraries@ -lHermes @directfb_libs@
 
-Libs/libclanDisplay.so: Libs/libclanCore.so $(OBJF_DISP_ALL)
-	@echo "Building Libs/libclanDisplay.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanDisplay.so.$(D_VERSION_MAJOR) -o Libs/libclanDisplay.so.$(D_VERSION_MINOR) $(OBJF_DISP_ALL) $(LINK_DISP)
-	@ln -s -f libclanDisplay.so.$(D_VERSION_MINOR) Libs/libclanDisplay.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanDisplay.so.$(D_VERSION_MAJOR) Libs/libclanDisplay.so
+Libs/libclanDisplay.dylib: Libs/libclanCore.dylib $(OBJF_DISP_ALL)
+	@echo "Building Libs/libclanDisplay.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanDisplay.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanDisplay.$(D_VERSION_MINOR).dylib $(OBJF_DISP_ALL) $(LINK_DISP)
+	@ln -s -f libclanDisplay.$(D_VERSION_MINOR).dylib Libs/libclanDisplay.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanDisplay.$(D_VERSION_MAJOR).dylib Libs/libclanDisplay.dylib
 
 Libs/libclanDisplay.a: $(OBJF_DISP_ALL)
 	@echo "Building Libs/libclanDisplay.a"
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.gl.in ClanLib-0.6.5/Setup/Unix/Makefile.gl.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.gl.in	2001-12-24 09:12:59.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.gl.in	2002-11-04 12:38:15.000000000 -0700
@@ -2,17 +2,17 @@
 # Makefile data for clanGL library.
 #
 
-clanGL: Libs/libclanGL.so Libs/libclanGL.a
+clanGL: Libs/libclanGL.dylib Libs/libclanGL.a
 
 installGL: clanGL
 	@for i in `find Sources/API/GL -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/GL -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/gl.h $(INC_PREFIX)/ClanLib/gl.h
+	@for i in `find Sources/API/GL -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/gl.h $(INC_PREFIX)/ClanLib/gl.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanGL.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanGL.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanGL.a $(LIB_PREFIX)
-	@ln -s -f libclanGL.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanGL.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanGL.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanGL.so
+	@ln -s -f libclanGL.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanGL.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanGL.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanGL.dylib
 
 OBJF_LIB_GL = Libs/Intermediate/texture_gl.o \
 		Libs/Intermediate/opengl.o \
@@ -23,11 +23,11 @@
 		Libs/Intermediate/blit_gl_generic.o \
 		Libs/Intermediate/glx_target.o
 
-Libs/libclanGL.so: Libs/libclanCore.so $(OBJF_LIB_GL)
-	@echo "Building Libs/libclanGL.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanGL.so.$(D_VERSION_MAJOR) -o Libs/libclanGL.so.$(D_VERSION_MINOR) $(OBJF_LIB_GL) -L Libs -lclanDisplay -lGL -lGLU @x_libraries@  
-	@ln -s -f libclanGL.so.$(D_VERSION_MINOR) Libs/libclanGL.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanGL.so.$(D_VERSION_MAJOR) Libs/libclanGL.so
+Libs/libclanGL.dylib: Libs/libclanCore.dylib $(OBJF_LIB_GL)
+	@echo "Building Libs/libclanGL.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanGL.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanGL.$(D_VERSION_MINOR).dylib $(OBJF_LIB_GL) -LLibs -lclanDisplay -lclanCore -lclanApp -lHermes -lGL -lGLU @x_libraries@  
+	@ln -s -f libclanGL.$(D_VERSION_MINOR).dylib Libs/libclanGL.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanGL.$(D_VERSION_MAJOR).dylib Libs/libclanGL.dylib
 
 Libs/libclanGL.a: Libs/libclanCore.a $(OBJF_LIB_GL)
 	@echo "Building Libs/libclanGL.a"
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.gui.in ClanLib-0.6.5/Setup/Unix/Makefile.gui.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.gui.in	2002-03-16 10:28:44.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.gui.in	2002-11-04 12:38:15.000000000 -0700
@@ -2,17 +2,17 @@
 # Makefile data for clanGUI library.
 #
 
-clanGUI: Libs/libclanGUI.so Libs/libclanGUI.a
+clanGUI: Libs/libclanGUI.dylib Libs/libclanGUI.a
 
 installGUI: clanGUI
 	@for i in `find Sources/API/GUI -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/GUI -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/gui.h $(INC_PREFIX)/ClanLib/gui.h
+	@for i in `find Sources/API/GUI -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/gui.h $(INC_PREFIX)/ClanLib/gui.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanGUI.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanGUI.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanGUI.a $(LIB_PREFIX)
-	@ln -s -f libclanGUI.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanGUI.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanGUI.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanGUI.so
+	@ln -s -f libclanGUI.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanGUI.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanGUI.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanGUI.dylib
 
 OBJF_LIB_GUI = \
 	Libs/Intermediate/button.o \
@@ -82,11 +82,11 @@
 	Libs/Intermediate/window_default.o \
 	Libs/Intermediate/window_generic.o
 
-Libs/libclanGUI.so: Libs/libclanCore.so $(OBJF_LIB_GUI)
-	@echo "Building Libs/libclanGUI.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanGUI.so.$(D_VERSION_MAJOR) -o Libs/libclanGUI.so.$(D_VERSION_MINOR) $(OBJF_LIB_GUI) -L Libs -lclanCore
-	@ln -s -f libclanGUI.so.$(D_VERSION_MINOR) Libs/libclanGUI.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanGUI.so.$(D_VERSION_MAJOR) Libs/libclanGUI.so	
+Libs/libclanGUI.dylib: Libs/libclanCore.dylib $(OBJF_LIB_GUI)
+	@echo "Building Libs/libclanGUI.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanGUI.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanGUI.$(D_VERSION_MINOR).dylib $(OBJF_LIB_GUI) -LLibs -lclanCore -lclanDisplay -lclanApp
+	@ln -s -f libclanGUI.$(D_VERSION_MINOR).dylib Libs/libclanGUI.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanGUI.$(D_VERSION_MAJOR).dylib Libs/libclanGUI.dylib
 
 Libs/libclanGUI.a: $(OBJF_LIB_GUI)
 	@echo "Building Libs/libclanGUI.a"
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.jpeg.in ClanLib-0.6.5/Setup/Unix/Makefile.jpeg.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.jpeg.in	2001-12-24 09:12:59.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.jpeg.in	2002-11-04 12:38:15.000000000 -0700
@@ -2,28 +2,28 @@
 # Makefile data for clanJPEG library.
 #
 
-clanJPEG: Libs/libclanJPEG.so Libs/libclanJPEG.a
+clanJPEG: Libs/libclanJPEG.dylib Libs/libclanJPEG.a
 
 installJPEG: clanJPEG
 	@for i in `find Sources/API/JPEG -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/JPEG -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/jpeg.h $(INC_PREFIX)/ClanLib/jpeg.h
+	@for i in `find Sources/API/JPEG -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/jpeg.h $(INC_PREFIX)/ClanLib/jpeg.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanJPEG.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanJPEG.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanJPEG.a $(LIB_PREFIX)
-	@ln -s -f libclanJPEG.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanJPEG.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanJPEG.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanJPEG.so
+	@ln -s -f libclanJPEG.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanJPEG.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanJPEG.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanJPEG.dylib
 
 
 OBJF_LIB_JPEG = \
 	Libs/Intermediate/provider_jpeg.o \
 	Libs/Intermediate/provider_jpeg_generic.o
 
-Libs/libclanJPEG.so: Libs/libclanCore.so $(OBJF_LIB_JPEG)
-	@echo "Building Libs/libclanJPEG.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanJPEG.so.$(D_VERSION_MAJOR) -o Libs/libclanJPEG.so.$(D_VERSION_MINOR) $(OBJF_LIB_JPEG) -L Libs -lclanCore -ljpeg
-	@ln -s -f libclanJPEG.so.$(D_VERSION_MINOR) Libs/libclanJPEG.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanJPEG.so.$(D_VERSION_MAJOR) Libs/libclanJPEG.so
+Libs/libclanJPEG.dylib: Libs/libclanCore.dylib $(OBJF_LIB_JPEG)
+	@echo "Building Libs/libclanJPEG.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanJPEG.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanJPEG.$(D_VERSION_MINOR).dylib $(OBJF_LIB_JPEG) -LLibs -lclanCore -lclanDisplay -lclanApp -ljpeg
+	@ln -s -f libclanJPEG.$(D_VERSION_MINOR).dylib Libs/libclanJPEG.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanJPEG.$(D_VERSION_MAJOR).dylib Libs/libclanJPEG.dylib
 
 Libs/libclanJPEG.a: $(OBJF_LIB_JPEG)
 	@echo "Building Libs/libclanJPEG.a"
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.lua.in ClanLib-0.6.5/Setup/Unix/Makefile.lua.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.lua.in	2001-12-24 09:12:59.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.lua.in	2002-11-04 12:38:15.000000000 -0700
@@ -2,26 +2,26 @@
 # Makefile data for clanLua library
 #
 
-clanLua: Libs/libclanLua.so Libs/libclanLua.a
+clanLua: Libs/libclanLua.dylib Libs/libclanLua.a
 
 installLua: clanLua
 	@for i in `find Sources/API/Lua -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/Lua -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/lua.h $(INC_PREFIX)/ClanLib/lua.h
+	@for i in `find Sources/API/Lua -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/lua.h $(INC_PREFIX)/ClanLib/lua.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanLua.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanLua.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanLua.a $(LIB_PREFIX)
-	@ln -s -f libclanLua.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanLua.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanLua.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanLua.so
+	@ln -s -f libclanLua.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanLua.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanLua.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanLua.dylib
 
 OBJF_LIB_LUA = \
 	Libs/Intermediate/clanbindings.o 
 
-Libs/libclanLua.so: Libs/libclanCore.so $(OBJF_LIB_LUA)
-	@echo "Building Libs/libclanLua.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanLua.so.$(D_VERSION_MAJOR) -o Libs/libclanLua.so.$(D_VERSION_MINOR) $(OBJF_LIB_LUA) -L Libs -lclanCore -llua -llualib -ltolua
-	@ln -s -f libclanLua.so.$(D_VERSION_MINOR) Libs/libclanLua.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanLua.so.$(D_VERSION_MAJOR) Libs/libclanLua.so
+Libs/libclanLua.dylib: Libs/libclanCore.dylib $(OBJF_LIB_LUA)
+	@echo "Building Libs/libclanLua.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanLua.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanLua.$(D_VERSION_MINOR).dylib $(OBJF_LIB_LUA) -LLibs -lclanCore -lclanApp -lclanDisplay -llua -llualib -ltolua
+	@ln -s -f libclanLua.$(D_VERSION_MINOR).dylib Libs/libclanLua.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanLua.$(D_VERSION_MAJOR).dylib Libs/libclanLua.dylib
 
 Libs/libclanLua.a: $(OBJF_LIB_LUA)
 	@echo "Building Libs/libclanLua.a"
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.mikmod.in ClanLib-0.6.5/Setup/Unix/Makefile.mikmod.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.mikmod.in	2001-12-24 09:12:59.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.mikmod.in	2002-11-04 12:38:15.000000000 -0700
@@ -2,17 +2,17 @@
 # Makefile data for clanMikMod library.
 #
 
-clanMikMod: Libs/libclanMikMod.so Libs/libclanMikMod.a
+clanMikMod: Libs/libclanMikMod.dylib Libs/libclanMikMod.a
 
 installMikMod: clanMikMod
 	@for i in `find Sources/API/MikMod -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/MikMod -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/mikmod.h $(INC_PREFIX)/ClanLib/mikmod.h
+	@for i in `find Sources/API/MikMod -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/mikmod.h $(INC_PREFIX)/ClanLib/mikmod.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanMikMod.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanMikMod.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanMikMod.a $(LIB_PREFIX)
-	@ln -s -f libclanMikMod.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanMikMod.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanMikMod.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanMikMod.so
+	@ln -s -f libclanMikMod.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanMikMod.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanMikMod.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanMikMod.dylib
 
 OBJF_LIB_MIKMOD = \
 	Libs/Intermediate/resourcetype_module.o \
@@ -20,11 +20,11 @@
 	Libs/Intermediate/module_reader.o \
 	Libs/Intermediate/streamed_mikmod_sample.o
 
-Libs/libclanMikMod.so: Libs/libclanCore.so $(OBJF_LIB_MIKMOD)
-	@echo "Building Libs/libclanMikMod.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanMikMod.so.$(D_VERSION_MAJOR) -o Libs/libclanMikMod.so.$(D_VERSION_MINOR) $(OBJF_LIB_MIKMOD) -L Libs -lclanCore -lmikmod
-	@ln -s -f libclanMikMod.so.$(D_VERSION_MINOR) Libs/libclanMikMod.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanMikMod.so.$(D_VERSION_MAJOR) Libs/libclanMikMod.so
+Libs/libclanMikMod.dylib: Libs/libclanCore.dylib $(OBJF_LIB_MIKMOD)
+	@echo "Building Libs/libclanMikMod.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanMikMod.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanMikMod.$(D_VERSION_MINOR).dylib $(OBJF_LIB_MIKMOD) -LLibs -lclanCore -lclanSound -lclanDisplay -lclanApp -lmikmod
+	@ln -s -f libclanMikMod.$(D_VERSION_MINOR).dylib Libs/libclanMikMod.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanMikMod.$(D_VERSION_MAJOR).dylib Libs/libclanMikMod.dylib
 
 Libs/libclanMikMod.a: $(OBJF_LIB_MIKMOD)
 	@echo "Building Libs/libclanMikMod.a"
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.mpeg.in ClanLib-0.6.5/Setup/Unix/Makefile.mpeg.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.mpeg.in	2001-12-24 09:12:59.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.mpeg.in	2002-11-04 12:38:15.000000000 -0700
@@ -2,26 +2,26 @@
 # Makefile data for clanMMPEG library.
 #
 
-clanMPEG: Libs/libclanMPEG.so Libs/libclanMPEG.a
+clanMPEG: Libs/libclanMPEG.dylib Libs/libclanMPEG.a
 
 installMPEG: clanMPEG
 	@for i in `find Sources/API/MPEG -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/MPEG -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/mpeg.h $(INC_PREFIX)/ClanLib/mpeg.h
+	@for i in `find Sources/API/MPEG -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/mpeg.h $(INC_PREFIX)/ClanLib/mpeg.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanMPEG.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanMPEG.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanMPEG.a $(LIB_PREFIX)
-	@ln -s -f libclanMPEG.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanMPEG.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanMPEG.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanMPEG.so
+	@ln -s -f libclanMPEG.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanMPEG.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanMPEG.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanMPEG.dylib
 
 OBJF_LIB_MPEG =	\
 	Libs/Intermediate/mpeg_soundprovider.o
 
-Libs/libclanMPEG.so: Libs/libclanCore.so $(OBJF_LIB_MPEG)
-	@echo "Building Libs/libclanMPEG.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanMPEG.so.$(D_VERSION_MAJOR) -o Libs/libclanMPEG.so.$(D_VERSION_MINOR) $(OBJF_LIB_MPEG) -L Libs -lclanCore 
-	@ln -s -f libclanMPEG.so.$(D_VERSION_MINOR) Libs/libclanMPEG.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanMPEG.so.$(D_VERSION_MAJOR) Libs/libclanMPEG.so
+Libs/libclanMPEG.dylib: Libs/libclanCore.dylib $(OBJF_LIB_MPEG)
+	@echo "Building Libs/libclanMPEG.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanMPEG.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanMPEG.$(D_VERSION_MINOR).dylib $(OBJF_LIB_MPEG) -LLibs -lclanCore 
+	@ln -s -f libclanMPEG.$(D_VERSION_MINOR).dylib Libs/libclanMPEG.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanMPEG.$(D_VERSION_MAJOR).dylib Libs/libclanMPEG.dylib
 
 Libs/libclanMPEG.a: $(OBJF_LIB_MPEG)
 	@echo "Building Libs/libclanMPEG.a"
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.network.in ClanLib-0.6.5/Setup/Unix/Makefile.network.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.network.in	2001-12-24 09:12:59.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.network.in	2002-11-04 12:38:15.000000000 -0700
@@ -2,17 +2,17 @@
 # Makefile data for clanNetwork library.
 #
 
-clanNetwork: Libs/libclanNetwork.so Libs/libclanNetwork.a
+clanNetwork: Libs/libclanNetwork.dylib Libs/libclanNetwork.a
 
 installNetwork: clanNetwork
 	@for i in `find Sources/API/Network -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/Network -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/network.h $(INC_PREFIX)/ClanLib/network.h
+	@for i in `find Sources/API/Network -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/network.h $(INC_PREFIX)/ClanLib/network.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanNetwork.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanNetwork.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanNetwork.a $(LIB_PREFIX)
-	@ln -s -f libclanNetwork.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanNetwork.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanNetwork.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanNetwork.so
+	@ln -s -f libclanNetwork.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanNetwork.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanNetwork.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanNetwork.dylib
 
 OBJF_LIB_NETWORK = Libs/Intermediate/netcomputer.o \
 	Libs/Intermediate/browse_client.o \
@@ -50,11 +50,11 @@
 #	Libs/Intermediate/unix_pipe_connection.o 
 #	Libs/Intermediate/network_unix.o 
 
-Libs/libclanNetwork.so: Libs/libclanCore.so $(OBJF_LIB_NETWORK)
-	@echo "Building Libs/libclanNetwork.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanNetwork.so.$(D_VERSION_MAJOR) -o Libs/libclanNetwork.so.$(D_VERSION_MINOR) $(OBJF_LIB_NETWORK) -L Libs -lclanCore
-	@ln -s -f libclanNetwork.so.$(D_VERSION_MINOR) Libs/libclanNetwork.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanNetwork.so.$(D_VERSION_MAJOR) Libs/libclanNetwork.so
+Libs/libclanNetwork.dylib: Libs/libclanCore.dylib $(OBJF_LIB_NETWORK)
+	@echo "Building Libs/libclanNetwork.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanNetwork.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanNetwork.$(D_VERSION_MINOR).dylib $(OBJF_LIB_NETWORK) -LLibs -lclanCore
+	@ln -s -f libclanNetwork.$(D_VERSION_MINOR).dylib Libs/libclanNetwork.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanNetwork.$(D_VERSION_MAJOR).dylib Libs/libclanNetwork.dylib
 
 Libs/libclanNetwork.a: $(OBJF_LIB_NETWORK)
 	@echo "Building Libs/libclanNetwork.a"
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.png.in ClanLib-0.6.5/Setup/Unix/Makefile.png.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.png.in	2002-01-10 13:17:13.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.png.in	2002-11-04 12:38:15.000000000 -0700
@@ -2,27 +2,27 @@
 # Makefile data for clanPNG library.
 #
 
-clanPNG: Libs/libclanPNG.so Libs/libclanPNG.a
+clanPNG: Libs/libclanPNG.dylib Libs/libclanPNG.a
 
 installPNG: clanPNG
 	@for i in `find Sources/API/PNG -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/PNG -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/png.h $(INC_PREFIX)/ClanLib/png.h
+	@for i in `find Sources/API/PNG -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/png.h $(INC_PREFIX)/ClanLib/png.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanPNG.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanPNG.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanPNG.a $(LIB_PREFIX)
-	@ln -s -f libclanPNG.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanPNG.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanPNG.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanPNG.so
+	@ln -s -f libclanPNG.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanPNG.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanPNG.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanPNG.dylib
 
 
 OBJF_LIB_PNG = \
 	Libs/Intermediate/provider_png.o
 
-Libs/libclanPNG.so: Libs/libclanCore.so $(OBJF_LIB_PNG)
-	@echo "Building Libs/libclanPNG.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanPNG.so.$(D_VERSION_MAJOR) -o Libs/libclanPNG.so.$(D_VERSION_MINOR) $(OBJF_LIB_PNG) -L Libs -lclanCore -lpng
-	@ln -s -f libclanPNG.so.$(D_VERSION_MINOR) Libs/libclanPNG.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanPNG.so.$(D_VERSION_MAJOR) Libs/libclanPNG.so
+Libs/libclanPNG.dylib: Libs/libclanCore.dylib $(OBJF_LIB_PNG)
+	@echo "Building Libs/libclanPNG.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanPNG.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanPNG.$(D_VERSION_MINOR).dylib $(OBJF_LIB_PNG) -LLibs -lclanCore -lclanDisplay -lclanApp -lpng
+	@ln -s -f libclanPNG.$(D_VERSION_MINOR).dylib Libs/libclanPNG.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanPNG.$(D_VERSION_MAJOR).dylib Libs/libclanPNG.dylib
 
 Libs/libclanPNG.a: $(OBJF_LIB_PNG)
 	@echo "Building Libs/libclanPNG.a"
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.signals.in ClanLib-0.6.5/Setup/Unix/Makefile.signals.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.signals.in	2001-03-16 16:56:33.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.signals.in	2002-11-04 12:38:15.000000000 -0700
@@ -6,5 +6,5 @@
 
 installSignals: clanSignals
 	@for i in `find Sources/API/Signals -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/Signals -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/signals.h $(INC_PREFIX)/ClanLib/signals.h
+	@for i in `find Sources/API/Signals -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/signals.h $(INC_PREFIX)/ClanLib/signals.h
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.smalljpeg.in ClanLib-0.6.5/Setup/Unix/Makefile.smalljpeg.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.smalljpeg.in	2002-02-13 22:54:15.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.smalljpeg.in	2002-11-04 12:38:15.000000000 -0700
@@ -2,17 +2,17 @@
 # Makefile data for clanSmallJPEG library.
 #
 
-clanSmallJPEG: Libs/libclanSmallJPEG.so Libs/libclanSmallJPEG.a
+clanSmallJPEG: Libs/libclanSmallJPEG.dylib Libs/libclanSmallJPEG.a
 
 installSmallJPEG: clanSmallJPEG
 	@for i in `find Sources/API/SmallJPEG -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/SmallJPEG -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/jpeg.h $(INC_PREFIX)/ClanLib/jpeg.h
+	@for i in `find Sources/API/SmallJPEG -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/jpeg.h $(INC_PREFIX)/ClanLib/jpeg.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanSmallJPEG.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanSmallJPEG.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanSmallJPEG.a $(LIB_PREFIX)
-	@ln -s -f libclanSmallJPEG.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanSmallJPEG.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanSmallJPEG.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanSmallJPEG.so
+	@ln -s -f libclanSmallJPEG.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanSmallJPEG.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanSmallJPEG.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanSmallJPEG.dylib
 
 
 OBJF_LIB_SMALLJPEG = \
@@ -25,11 +25,11 @@
 OBJF_JPEG_INTEL_ASM = \
 	ifast.o
 
-Libs/libclanSmallJPEG.so: Libs/libclanCore.so Libs/libclanDisplay.so $(OBJF_LIB_SMALLJPEG) @small_jpeg_assembly@
-	@echo "Building Libs/libclanSmallJPEG.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanSmallJPEG.so.$(D_VERSION_MAJOR) -o Libs/libclanSmallJPEG.so.$(D_VERSION_MINOR) $(OBJF_LIB_SMALLJPEG) -L Libs -lclanCore -lclanDisplay
-	@ln -s -f libclanSmallJPEG.so.$(D_VERSION_MINOR) Libs/libclanSmallJPEG.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanSmallJPEG.so.$(D_VERSION_MAJOR) Libs/libclanSmallJPEG.so
+Libs/libclanSmallJPEG.dylib: Libs/libclanCore.dylib Libs/libclanDisplay.dylib $(OBJF_LIB_SMALLJPEG) @small_jpeg_assembly@
+	@echo "Building Libs/libclanSmallJPEG.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanSmallJPEG.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanSmallJPEG.$(D_VERSION_MINOR).dylib $(OBJF_LIB_SMALLJPEG) -LLibs -lclanCore -lclanDisplay -lclanApp
+	@ln -s -f libclanSmallJPEG.$(D_VERSION_MINOR).dylib Libs/libclanSmallJPEG.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanSmallJPEG.$(D_VERSION_MAJOR).dylib Libs/libclanSmallJPEG.dylib
 
 Libs/libclanSmallJPEG.a: $(OBJF_LIB_SMALLJPEG)
 	@echo "Building Libs/libclanSmallJPEG.a"
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.sound.in ClanLib-0.6.5/Setup/Unix/Makefile.sound.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.sound.in	2001-12-24 09:12:59.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.sound.in	2002-11-04 12:38:15.000000000 -0700
@@ -1,17 +1,17 @@
 # Makefile data for clanSound library.
 #
 
-clanSound: Libs/libclanSound.so Libs/libclanSound.a
+clanSound: Libs/libclanSound.dylib Libs/libclanSound.a
 
 installSound: clanSound
 	@for i in `find Sources/API/Sound -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/Sound -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/sound.h $(INC_PREFIX)/ClanLib/sound.h
+	@for i in `find Sources/API/Sound -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/sound.h $(INC_PREFIX)/ClanLib/sound.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanSound.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanSound.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanSound.a $(LIB_PREFIX)
-	@ln -s -f libclanSound.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanSound.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanSound.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanSound.so
+	@ln -s -f libclanSound.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanSound.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanSound.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanSound.dylib
 
 OBJF_SOUND_GENERIC = \
 	Libs/Intermediate/sound.o \
@@ -42,18 +42,18 @@
 	Libs/Intermediate/cardplayback_clan.o \
 	Libs/Intermediate/mixer.o \
 	Libs/Intermediate/oss.o \
-	Libs/Intermediate/cdaudio_linux.o
+#	Libs/Intermediate/cdaudio_linux.o
 
 OBJF_SOUND_INTEL_ASM = \
 	Libs/Intermediate/mixer_asm.o
 
 OBJF_SOUND_ALL = $(OBJF_SOUND_GENERIC) $(OBJF_SOUND_CLANSOUND) @sound_assembly@
 
-Libs/libclanSound.so: Libs/libclanCore.so $(OBJF_SOUND_ALL)
-	@echo "Building Libs/libclanSound.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanSound.so.$(D_VERSION_MAJOR) -o Libs/libclanSound.so.$(D_VERSION_MINOR) $(OBJF_SOUND_ALL) -L Libs -lclanCore
-	@ln -s -f libclanSound.so.$(D_VERSION_MINOR) Libs/libclanSound.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanSound.so.$(D_VERSION_MAJOR) Libs/libclanSound.so
+Libs/libclanSound.dylib: Libs/libclanCore.dylib $(OBJF_SOUND_ALL)
+	@echo "Building Libs/libclanSound.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanSound.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanSound.$(D_VERSION_MINOR).dylib $(OBJF_SOUND_ALL) -LLibs -lclanCore -lesd
+	@ln -s -f libclanSound.$(D_VERSION_MINOR).dylib Libs/libclanSound.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanSound.$(D_VERSION_MAJOR).dylib Libs/libclanSound.dylib
 
 Libs/libclanSound.a: $(OBJF_SOUND_ALL)
 	@echo "Building Libs/libclanSound.a"
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.ttf.in ClanLib-0.6.5/Setup/Unix/Makefile.ttf.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.ttf.in	2001-12-24 09:12:59.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.ttf.in	2002-11-04 12:38:15.000000000 -0700
@@ -2,17 +2,17 @@
 # Makefile data for clanTTF library.
 #
 
-clanTTF: Libs/libclanTTF.so Libs/libclanTTF.a
+clanTTF: Libs/libclanTTF.dylib Libs/libclanTTF.a
 
 installTTF: clanTTF
 	@for i in `find Sources/API/TTF -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/TTF -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/ttf.h $(INC_PREFIX)/ClanLib/ttf.h
+	@for i in `find Sources/API/TTF -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/ttf.h $(INC_PREFIX)/ClanLib/ttf.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanTTF.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanTTF.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanTTF.a $(LIB_PREFIX)
-	@ln -s -f libclanTTF.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanTTF.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanTTF.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanTTF.so
+	@ln -s -f libclanTTF.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanTTF.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanTTF.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanTTF.dylib
 
 
 OBJF_LIB_TTF = \
@@ -20,11 +20,11 @@
 	Libs/Intermediate/resourcetype_ttf.o \
 	Libs/Intermediate/setupttf.o
 
-Libs/libclanTTF.so: Libs/libclanCore.so $(OBJF_LIB_TTF)
-	@echo "Building Libs/libclanTTF.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanTTF.so.$(D_VERSION_MAJOR) -o Libs/libclanTTF.so.$(D_VERSION_MINOR) $(OBJF_LIB_TTF) -L Libs -lclanCore -lfreetype
-	@ln -s -f libclanTTF.so.$(D_VERSION_MINOR) Libs/libclanTTF.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanTTF.so.$(D_VERSION_MAJOR) Libs/libclanTTF.so
+Libs/libclanTTF.dylib: Libs/libclanCore.dylib $(OBJF_LIB_TTF)
+	@echo "Building Libs/libclanTTF.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanTTF.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanTTF.$(D_VERSION_MINOR).dylib $(OBJF_LIB_TTF) -LLibs -lclanCore -lclanDisplay -lclanApp -lfreetype
+	@ln -s -f libclanTTF.$(D_VERSION_MINOR).dylib Libs/libclanTTF.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanTTF.$(D_VERSION_MAJOR).dylib Libs/libclanTTF.dylib
 
 Libs/libclanTTF.a: $(OBJF_LIB_TTF)
 	@echo "Building Libs/libclanTTF.a"
diff -ruN ClanLib-0.6.5.orig/Setup/Unix/Makefile.vorbis.in ClanLib-0.6.5/Setup/Unix/Makefile.vorbis.in
--- ClanLib-0.6.5.orig/Setup/Unix/Makefile.vorbis.in	2001-12-24 09:12:59.000000000 -0700
+++ ClanLib-0.6.5/Setup/Unix/Makefile.vorbis.in	2002-11-04 12:38:15.000000000 -0700
@@ -2,27 +2,27 @@
 # Makefile data for clanVorbis library.
 #
 
-clanVorbis: Libs/libclanVorbis.so Libs/libclanVorbis.a
+clanVorbis: Libs/libclanVorbis.dylib Libs/libclanVorbis.a
 
 installVorbis: clanVorbis
 	@for i in `find Sources/API/Vorbis -type d | grep -v CVS | sed "s/Sources\/API\///;"`; do install -d $(INC_PREFIX)/ClanLib/$$i; done
-	@for i in `find Sources/API/Vorbis -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
-	@install -m 0644 Sources/API/vorbis.h $(INC_PREFIX)/ClanLib/vorbis.h
+	@for i in `find Sources/API/Vorbis -type f | grep -v CVS | sed "s/Sources\/API\///;"`; do install -c -m 0644 Sources/API/$$i $(INC_PREFIX)/ClanLib/$$i; done
+	@install -c -m 0644 Sources/API/vorbis.h $(INC_PREFIX)/ClanLib/vorbis.h
 	@install -d $(LIB_PREFIX)
-	@install Libs/libclanVorbis.so.$(D_VERSION_MINOR) $(LIB_PREFIX)
+	@install Libs/libclanVorbis.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)
 	@install Libs/libclanVorbis.a $(LIB_PREFIX)
-	@ln -s -f libclanVorbis.so.$(D_VERSION_MINOR) $(LIB_PREFIX)/libclanVorbis.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanVorbis.so.$(D_VERSION_MAJOR) $(LIB_PREFIX)/libclanVorbis.so
+	@ln -s -f libclanVorbis.$(D_VERSION_MINOR).dylib $(LIB_PREFIX)/libclanVorbis.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanVorbis.$(D_VERSION_MAJOR).dylib $(LIB_PREFIX)/libclanVorbis.dylib
 
 OBJF_LIB_VORBIS = \
 	Libs/Intermediate/vorbis_soundprovider_generic.o \
 	Libs/Intermediate/vorbis_setup.o
 
-Libs/libclanVorbis.so: Libs/libclanCore.so $(OBJF_LIB_VORBIS)
-	@echo "Building Libs/libclanVorbis.so"
-	@$(LINK_COMMAND) -Wl,-soname=libclanVorbis.so.$(D_VERSION_MAJOR) -o Libs/libclanVorbis.so.$(D_VERSION_MINOR) $(OBJF_LIB_VORBIS) -L Libs -lclanCore -lvorbis
-	@ln -s -f libclanVorbis.so.$(D_VERSION_MINOR) Libs/libclanVorbis.so.$(D_VERSION_MAJOR)
-	@ln -s -f libclanVorbis.so.$(D_VERSION_MAJOR) Libs/libclanVorbis.so	
+Libs/libclanVorbis.dylib: Libs/libclanCore.dylib $(OBJF_LIB_VORBIS)
+	@echo "Building Libs/libclanVorbis.dylib"
+	@$(LINK_COMMAND) -dynamiclib -install_name $(LIB_PREFIX)/libclanVorbis.$(D_VERSION_MAJOR).dylib -compatibility_version $(D_VERSION_MAJOR) -current_version $(D_VERSION_MAJOR) -o Libs/libclanVorbis.$(D_VERSION_MINOR).dylib $(OBJF_LIB_VORBIS) -LLibs -lclanCore -lvorbis -logg
+	@ln -s -f libclanVorbis.$(D_VERSION_MINOR).dylib Libs/libclanVorbis.$(D_VERSION_MAJOR).dylib
+	@ln -s -f libclanVorbis.$(D_VERSION_MAJOR).dylib Libs/libclanVorbis.dylib	
 
 Libs/libclanVorbis.a: $(OBJF_LIB_VORBIS)
 	@echo "Building Libs/libclanVorbis.a"
diff -ruN ClanLib-0.6.5.orig/Sources/API/Core/System/clanstring.h ClanLib-0.6.5/Sources/API/Core/System/clanstring.h
--- ClanLib-0.6.5.orig/Sources/API/Core/System/clanstring.h	2002-02-20 01:06:28.000000000 -0700
+++ ClanLib-0.6.5/Sources/API/Core/System/clanstring.h	2002-11-04 12:38:15.000000000 -0700
@@ -39,9 +39,9 @@
 #include <extras.h>
 #endif
 
-#ifdef __MACOS__
+#if defined(__APPLE__) && defined(__GNUC__)
 #include <cctype>
-#define NEED_FCVT
+#define NEED_FCVT 1
 #endif
 
 #ifdef NEED_FCVT
diff -ruN ClanLib-0.6.5.orig/Sources/Core/Math/vector2.cpp ClanLib-0.6.5/Sources/Core/Math/vector2.cpp
--- ClanLib-0.6.5.orig/Sources/Core/Math/vector2.cpp	2001-09-15 08:54:51.000000000 -0600
+++ ClanLib-0.6.5/Sources/Core/Math/vector2.cpp	2002-11-04 12:38:15.000000000 -0700
@@ -254,7 +254,7 @@
 
 float CL_Vector2::length() const
 {
-    return sqrtf(x*x +y*y);
+    return sqrt(x*x +y*y);
 }
 
 CL_Vector2 CL_Vector2::cross() const
diff -ruN ClanLib-0.6.5.orig/Sources/Core/System/Unix/appconf.cpp ClanLib-0.6.5/Sources/Core/System/Unix/appconf.cpp
--- ClanLib-0.6.5.orig/Sources/Core/System/Unix/appconf.cpp	2002-01-23 15:32:40.000000000 -0700
+++ ClanLib-0.6.5/Sources/Core/System/Unix/appconf.cpp	2002-11-04 12:38:15.000000000 -0700
@@ -169,6 +169,8 @@
 # ifdef   __unix__
 //strcasecmp is in string.h
 #   define  StrCmp(s1,s2)  strcasecmp((s1),(s2))
+# elif defined(__APPLE__) && defined(__GNUC__)
+#   define  StrCmp(s1,s2)  strcasecmp((s1),(s2))
 # else
 #   ifdef _MSC_VER
 #     define  StrCmp(s1,s2)  _stricmp((s1),(s2))
diff -ruN ClanLib-0.6.5.orig/Sources/Core/System/Unix/mutex_pthread.cpp ClanLib-0.6.5/Sources/Core/System/Unix/mutex_pthread.cpp
--- ClanLib-0.6.5.orig/Sources/Core/System/Unix/mutex_pthread.cpp	2002-06-22 06:58:04.000000000 -0600
+++ ClanLib-0.6.5/Sources/Core/System/Unix/mutex_pthread.cpp	2002-11-04 12:38:15.000000000 -0700
@@ -19,10 +19,12 @@
 
 // We need to do this because the posix threads library under linux obviously
 // suck:
+#ifdef linux
 extern "C"
 {
 	int pthread_mutexattr_setkind_np(pthread_mutexattr_t *attr, int kind);
 }
+#endif
 
 CL_Mutex *CL_Mutex::create()
 {
@@ -37,7 +39,7 @@
 	pthread_mutexattr_t attr;
 	pthread_mutexattr_init(&attr);
 #ifndef linux
-	pthread_mutexattr_setkind_np(&attr, PTHREAD_MUTEX_RECURSIVE);
+	pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);
 #else
 	pthread_mutexattr_setkind_np(&attr, PTHREAD_MUTEX_RECURSIVE_NP);
 #endif
diff -ruN ClanLib-0.6.5.orig/Sources/Network/Generic/browse_client_generic.cpp ClanLib-0.6.5/Sources/Network/Generic/browse_client_generic.cpp
--- ClanLib-0.6.5.orig/Sources/Network/Generic/browse_client_generic.cpp	2001-10-14 12:05:07.000000000 -0600
+++ ClanLib-0.6.5/Sources/Network/Generic/browse_client_generic.cpp	2002-11-04 12:38:15.000000000 -0700
@@ -12,6 +12,10 @@
 	------------------------------------------------------------------------
 */
 
+#if defined(__APPLE__) && defined(__GNUC__)
+#include <sys/types.h>
+#endif
+
 #ifdef WIN32
 #pragma warning (disable:4786)
 #pragma warning (disable:4355)
diff -ruN ClanLib-0.6.5.orig/Sources/Network/Generic/network_delivery_socket.cpp ClanLib-0.6.5/Sources/Network/Generic/network_delivery_socket.cpp
--- ClanLib-0.6.5.orig/Sources/Network/Generic/network_delivery_socket.cpp	2001-11-01 13:27:57.000000000 -0700
+++ ClanLib-0.6.5/Sources/Network/Generic/network_delivery_socket.cpp	2002-11-04 12:38:15.000000000 -0700
@@ -21,6 +21,10 @@
 #include <Network/Generic/network_delivery_impl.h>
 #include <Network/Generic/network_delivery_socket.h>
 
+#if defined(__APPLE__) && defined(__GNUC__)
+typedef int socklen_t;
+#endif /* Darwin */
+
 #include <string.h>
 #include <errno.h>
 
diff -ruN ClanLib-0.6.5.orig/Sources/Network/Generic/socket.cpp ClanLib-0.6.5/Sources/Network/Generic/socket.cpp
--- ClanLib-0.6.5.orig/Sources/Network/Generic/socket.cpp	2001-11-21 17:43:11.000000000 -0700
+++ ClanLib-0.6.5/Sources/Network/Generic/socket.cpp	2002-11-04 12:38:15.000000000 -0700
@@ -25,6 +25,10 @@
 #define INVALID_SOCKET -1
 #endif
 
+#if defined(__APPLE__) && defined(__GNUC__)
+typedef int socklen_t;
+#endif /* Darwin */
+
 #include "socket_generic.h"
 #include "API/Core/System/error.h"
 
diff -ruN ClanLib-0.6.5.orig/Sources/Sound/Sound/ClanSound/oss.cpp ClanLib-0.6.5/Sources/Sound/Sound/ClanSound/oss.cpp
--- ClanLib-0.6.5.orig/Sources/Sound/Sound/ClanSound/oss.cpp	2002-07-02 14:56:33.000000000 -0600
+++ ClanLib-0.6.5/Sources/Sound/Sound/ClanSound/oss.cpp	2002-11-04 12:38:15.000000000 -0700
@@ -14,6 +14,9 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <sys/ioctl.h>
+#if defined(__APPLE__) && defined(__GNUC__)
+#include <esd.h>
+#else
 #ifndef Solaris
 #include <sys/soundcard.h>
 #define DSP_DEVICE "/dev/dsp"
@@ -21,6 +24,7 @@
 #include "soundcard.h" // we don't have any soundcard.h on Solaris by default
 #define DSP_DEVICE "/dev/audio"
 #endif		       
+#endif
 #include <iostream>
 
 #include <API/Core/System/error.h>
@@ -37,6 +41,24 @@
 
 CL_CSOutput::CL_CSOutput()
 {
+#if defined(__APPLE__) && defined(__GNUC__)
+	esd_format_t format = ESD_STREAM | ESD_PLAY | ESD_BITS16 | ESD_STEREO;
+	int rate = 22050;
+	int esd;
+
+	dev_dsp_fd = -1;
+
+	esd = esd_open_sound(NULL);
+	if (esd < 0) {
+		has_sound = false;
+		return;
+	} else {
+		esd_close(esd);
+	}
+
+	frag_size = 22050/2; // 0.25 sec mixing buffer used.
+	dev_dsp_fd = esd_play_stream(format, rate, NULL, "clanlib");
+#else
 	dev_dsp_fd = open(DSP_DEVICE, O_WRONLY|O_NONBLOCK);
 	if (dev_dsp_fd == -1)
 	{
@@ -95,6 +117,7 @@
 
 	audio_buf_info info;
 	ioctl(dev_dsp_fd, SNDCTL_DSP_GETOSPACE, &info);
+#endif
 }
 
 CL_CSOutput::~CL_CSOutput()
@@ -106,12 +129,18 @@
 // Called when we have no samples to play - and wants to tell the soundcard
 // about this possible event.
 {
+#if defined(__APPLE__) && defined(__GNUC__)
+#else
 	ioctl(dev_dsp_fd, SNDCTL_DSP_POST, 0);
+#endif
 }
 
 bool CL_CSOutput::is_full()
 // Returns true if all fragments are filled with data.
 {
+#if defined(__APPLE__) && defined(__GNUC__)
+	return false;
+#else
 	if (!has_sound) return false;
 	audio_buf_info info;
 	int err = ioctl(dev_dsp_fd, SNDCTL_DSP_GETOSPACE, &info);
@@ -123,6 +152,7 @@
 
 	if (info.fragments == 0) return true;
 	return false;
+#endif
 }
 
 int CL_CSOutput::get_frag_size()
diff -ruN ClanLib-0.6.5.orig/Sources/Sound/System/Unix/implementation_clansound.cpp ClanLib-0.6.5/Sources/Sound/System/Unix/implementation_clansound.cpp
--- ClanLib-0.6.5.orig/Sources/Sound/System/Unix/implementation_clansound.cpp	2001-04-28 06:04:36.000000000 -0600
+++ ClanLib-0.6.5/Sources/Sound/System/Unix/implementation_clansound.cpp	2002-11-04 12:38:15.000000000 -0700
@@ -52,7 +52,10 @@
 	try
 	{
 		CL_Sound::cards.push_back(new CL_SoundCard_ClanSound);
-		CL_CDDrive_Linux::init_cdaudio();
+		#if defined(__APPLE__) && defined(__GNUC__)
+		#else
+			CL_CDDrive_Linux::init_cdaudio();
+		#endif
 	}
 	catch (CL_Error error)
 	{
diff -ruN ClanLib-0.6.5.orig/Sources/Sound/System/Unix/setupsound_unix.cpp ClanLib-0.6.5/Sources/Sound/System/Unix/setupsound_unix.cpp
--- ClanLib-0.6.5.orig/Sources/Sound/System/Unix/setupsound_unix.cpp	2002-01-22 03:14:04.000000000 -0700
+++ ClanLib-0.6.5/Sources/Sound/System/Unix/setupsound_unix.cpp	2002-11-04 12:38:15.000000000 -0700
@@ -52,7 +52,10 @@
 {
 	if (register_resources_only) return;
 
-	CL_CDDrive_Linux::init_cdaudio();
+	#if defined(__APPLE__) && defined(__GNUC__)
+	#else
+		CL_CDDrive_Linux::init_cdaudio();
+	#endif
 }
 
 void CL_SetupCDAudio::deinit()
diff -ruN ClanLib-0.6.5.orig/Sources/TTF/font_ttf.cpp ClanLib-0.6.5/Sources/TTF/font_ttf.cpp
--- ClanLib-0.6.5.orig/Sources/TTF/font_ttf.cpp	2002-01-15 09:18:48.000000000 -0700
+++ ClanLib-0.6.5/Sources/TTF/font_ttf.cpp	2002-11-04 12:38:15.000000000 -0700
@@ -18,7 +18,12 @@
 #include <API/Display/SurfaceProviders/canvas.h>
 #include <API/Display/Display/surface.h>
 #include <API/Display/Font/font.h>
-#include <malloc.h>
+
+#if defined(__APPLE__) && defined(__GNUC__)
+#else
+	#include <malloc.h>
+#endif
+
 #include "Display/Font/font_generic.h"
 
 #include <freetype/freetype.h>
diff -ruN ClanLib-0.6.5.orig/configure ClanLib-0.6.5/configure
--- ClanLib-0.6.5.orig/configure	2002-11-03 14:41:57.000000000 -0700
+++ ClanLib-0.6.5/configure	2002-11-04 12:38:15.000000000 -0700
@@ -2334,7 +2334,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
       # Double quotes because CPP needs to be expanded
-    for CPP in "$CC -E" "$CC -E -traditional-cpp" "/lib/cpp"
+    for CPP in "$CC -E -no-cpp-precomp" "$CC -E -traditional-cpp" "$CC -E" "/lib/cpp"
     do
       ac_preproc_ok=false
 for ac_c_preproc_warn_flag in '' yes
@@ -3064,11 +3064,11 @@
 if test "$no_x" != "yes"; then
 	X11=enabled
 	if test "$x_includes" != ""; then
-		x_includes="-I $x_includes -I $x_includes/X11"
+		x_includes="-I$x_includes -I$x_includes/X11"
 	fi
 
 	if test "$x_libraries" != ""; then
-		x_libraries="-L $x_libraries"
+		x_libraries="-L$x_libraries"
 	fi
 
 	x_libraries="$x_libraries -lX11 -lXext"
@@ -3173,14 +3173,14 @@
     CPPFLAGS="$CPPFLAGS $x_includes"
   fi
   if test "x$x_libraries" != xNONE; then
-    CPPFLAGS="$CPPFLAGS $x_libraries"
+    CPPFLAGS="$CPPFLAGS" # $x_libraries"
   fi
   if test "x$extra_includes" != xNONE; then
 
 	CPPFLAGS="$CPPFLAGS $extra_includes"
   fi
   if test "x$extra_libraries" != xNONE; then
-    CPPFLAGS="$CPPFLAGS $extra_libraries"
+    CPPFLAGS="$CPPFLAGS" # $extra_libraries"
   fi
 
   OLDLIBS="$LIBS"
@@ -3300,14 +3300,14 @@
     CPPFLAGS="$CPPFLAGS $x_includes"
   fi
   if test "x$x_libraries" != xNONE; then
-    CPPFLAGS="$CPPFLAGS $x_libraries"
+    CPPFLAGS="$CPPFLAGS" # $x_libraries"
   fi
   if test "x$extra_includes" != xNONE; then
 
 	CPPFLAGS="$CPPFLAGS $extra_includes"
   fi
   if test "x$extra_libraries" != xNONE; then
-    CPPFLAGS="$CPPFLAGS $extra_libraries"
+    CPPFLAGS="$CPPFLAGS" # $extra_libraries"
   fi
 
   OLDLIBS="$LIBS"
@@ -5316,7 +5316,7 @@
 	link_command='$(CXX) -fPIC -nostart'
 else
 	objf_system='$(OBJF_UNIX)'
-	link_command='$(CXX) -shared -fPIC'
+	link_command='$(CXX) -fPIC'
 fi
 
 
