diff -ruN dash.orig/Makefile dash/Makefile
--- dash.orig/Makefile	2002-10-26 05:34:12.000000000 -0600
+++ dash/Makefile	2002-12-26 21:22:34.000000000 -0700
@@ -1,7 +1,7 @@
 #	$NetBSD: Makefile,v 1.59 2002/09/15 00:19:22 thorpej Exp $
 #	@(#)Makefile	8.4 (Berkeley) 5/5/95
 
-.include <bsd.own.mk>
+#.include <bsd.own.mk>
 
 YHEADER=1
 PROG=	sh
@@ -60,6 +60,6 @@
 mksyntax: mksyntax.c
 	${HOST_LINK.c} ${TARGET_CHARFLAG} -o mksyntax ${.IMPSRC}
 
-.include <bsd.prog.mk>
+#.include <bsd.prog.mk>
 
 ${OBJS}: arith.h builtins.h nodes.h syntax.h token.h
