diff -ruN fakeroot-0.5.9.orig/communicate.c fakeroot-0.5.9/communicate.c
--- fakeroot-0.5.9.orig/communicate.c	Sun Oct 27 22:19:52 2002
+++ fakeroot-0.5.9/communicate.c	Thu Feb  6 13:08:52 2003
@@ -14,7 +14,11 @@
 #include <dlfcn.h>
 #include <stdio.h>
 #include <sys/ipc.h>
-#include <sys/msg.h>
+#if !defined(__APPLE__) && !defined(__GNUC__)
+# include <sys/msg.h>
+#else
+# include <mach/message.h>
+#endif
 #include <sys/sem.h>
 #include <fcntl.h>
 #include <unistd.h>
diff -ruN fakeroot-0.5.9.orig/configure fakeroot-0.5.9/configure
--- fakeroot-0.5.9.orig/configure	Sat Jan 18 07:53:35 2003
+++ fakeroot-0.5.9/configure	Thu Feb  6 12:58:16 2003
@@ -10609,6 +10609,9 @@
 	(*:osf*)
 		libcpath="/shlib/libc.so"
 		;;
+	(*:darwin*)
+		libcpath="/usr/lib/libSystem.dylib"
+		;;
 	(*)
 		{ echo "$as_me:$LINENO: WARNING: don't know where libc is for $target_os on
 			     $target_cpu, setting to /lib/libc.so" >&5
diff -ruN fakeroot-0.5.9.orig/faked.cc fakeroot-0.5.9/faked.cc
--- fakeroot-0.5.9.orig/faked.cc	Sat Jan 18 07:44:31 2003
+++ fakeroot-0.5.9/faked.cc	Thu Feb  6 13:38:59 2003
@@ -71,7 +71,11 @@
 
 #include "communicate.h"
 #include <sys/ipc.h>
-#include <sys/msg.h>
+#if !defined(__APPLE__) && !defined(__GNUC__)
+# include <sys/msg.h>
+#else
+# include <mach/message.h>
+#endif
 #include <sys/wait.h>
 #include <sys/sem.h>
 #include <sys/types.h>
@@ -363,7 +367,11 @@
   if(debug)
     fprintf(stderr,"FAKEROOT: msg=%i, key=%i\n",msg_get,msg_key);
   do{
+#if !defined(__APPLE__) && !defined(__GNUC__)
     r=msgrcv(msg_get,&buf,sizeof(fake_msg),0,0);
+#else
+    r=mach_msg_overwrite(0,0,0,sizeof(fake_msg),msg_get,0,0,0,0);
+#endif
     if(debug)
       fprintf(stderr,"FAKEROOT: r=%i, received message type=%li, message=%i\n",r,buf.mtype,buf.id);
     if(r!=-1)
@@ -388,8 +396,13 @@
   if(debug)
     fprintf(stderr, "fakeroot: clearing up message queues and semaphores,"
 	    " signal=%i\n",  g);
+#if !defined(__APPLE__) && !defined(__GNUC__) 
   msgctl (msg_get, IPC_RMID,NULL);
   msgctl (msg_snd, IPC_RMID,NULL);
+#else
+  mach_msg(0,0,0,0,msg_get,0,IPC_RMID);
+  mach_msg(0,0,0,0,msg_snd,0,IPC_RMID);           
+#endif
   semctl (sem_id,0,IPC_RMID,sem_union);
   if(g!=-1)
     exit(0);
@@ -451,9 +464,14 @@
   }
   if(debug)
     fprintf(stderr,"using %i as msg key\n",msg_key);
-  
+
+#if !defined(__APPLE__) && !defined(__GNUC__) 
   msg_get=msgget(msg_key,IPC_CREAT|0600);
   msg_snd=msgget(msg_key+1,IPC_CREAT|0600);
+#else
+  msg_get=mach_msg(0,IPC_CREAT|0600,0,0,msg_key,0,0);    
+  msg_snd=mach_msg(0,IPC_CREAT|0600,0,0,msg_key+1,0,0);     
+#endif
   sem_id=semget(msg_key+2,1,IPC_CREAT|0600);
   sem_union.val=1;
   semctl (sem_id,0,SETVAL,sem_union);
diff -ruN fakeroot-0.5.9.orig/libfakeroot.c fakeroot-0.5.9/libfakeroot.c
--- fakeroot-0.5.9.orig/libfakeroot.c	Sun Dec  8 09:27:59 2002
+++ fakeroot-0.5.9/libfakeroot.c	Thu Feb  6 13:08:39 2003
@@ -23,7 +23,11 @@
 #include "communicate.h"
 #include <stdlib.h>
 #include <sys/ipc.h>
-#include <sys/msg.h>
+#if !defined(__APPLE__) && !defined(__GNUC__)
+# include <sys/msg.h>
+#else
+# include <mach/message.h>
+#endif
 #include <fcntl.h>
 #include <string.h>
 #include <stdio.h>
@@ -382,7 +386,11 @@
 }
 #endif
 
+#if !defined(__APPLE__) && !defined(__GNUC__)
 int fchown(int fd, uid_t owner, gid_t group){
+#else
+int fchown(int fd, int owner, int group){
+#endif
   struct stat st;
   int r;
 
diff -ruN fakeroot-0.5.9.orig/scripts/fakeroot fakeroot-0.5.9/scripts/fakeroot
--- fakeroot-0.5.9.orig/scripts/fakeroot	Sat Nov 23 17:02:55 2002
+++ fakeroot-0.5.9/scripts/fakeroot	Thu Feb  6 13:55:06 2003
@@ -13,7 +13,7 @@
 BINDIR=`dirname $0`
 PREFIX=`dirname ${BINDIR}`
     
-LIB=libfakeroot.so.0
+LIB=libfakeroot.0.dylib
 PATHS=${PREFIX}/lib/libfakeroot:${PREFIX}/lib64/libfakeroot
 FAKED=${BINDIR}/faked
 
@@ -91,19 +91,19 @@
 fi
 
 # Keep other library paths
-if test -n "$LD_LIBRARY_PATH"; then
-  PATHS="$PATHS:$LD_LIBRARY_PATH"
+if test -n "$DYLD_LIBRARY_PATH"; then
+  PATHS="$PATHS:$DYLD_LIBRARY_PATH"
 fi
 # ...and preloaded libs
-if test -n "$LD_PRELOAD"; then
-  LIB="$LIB $LD_PRELOAD"
+if test -n "$DYLD_INSERT_LIBRARIES"; then
+  LIB="$LIB $DYLD_INSERT_LIBRARIES"
 fi
 
 if test -z "$*"; then
-  FAKEROOTKEY=$FAKEROOTKEY LD_LIBRARY_PATH="$PATHS" LD_PRELOAD="$LIB" ${SHELL:-/bin/sh}
+  FAKEROOTKEY=$FAKEROOTKEY DYLD_LIBRARY_PATH="$PATHS" DYLD_INSERT_LIBRARIES="$LIB" ${SHELL:-/bin/sh}
   RESULT=$?
 else
-  FAKEROOTKEY=$FAKEROOTKEY LD_LIBRARY_PATH="$PATHS" LD_PRELOAD="$LIB" "$@"
+  FAKEROOTKEY=$FAKEROOTKEY DYLD_LIBRARY_PATH="$PATHS" DYLD_INSERT_LIBRARIES="$LIB" "$@"
   RESULT=$?
 fi
 
