diff -ruN gmyclient-0.3.orig/plugins/exporter/csv/Makefile.in gmyclient-0.3/plugins/exporter/csv/Makefile.in
--- gmyclient-0.3.orig/plugins/exporter/csv/Makefile.in	Wed Apr 17 08:34:52 2002
+++ gmyclient-0.3/plugins/exporter/csv/Makefile.in	Tue Apr 15 15:40:59 2003
@@ -144,7 +144,7 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-libcsv_la_LDFLAGS = 
+libcsv_la_LDFLAGS =  -bundle -avoid-version
 libcsv_la_LIBADD = 
 libcsv_la_OBJECTS =  csv.lo
 CFLAGS = @CFLAGS@
diff -ruN gmyclient-0.3.orig/plugins/exporter/html/Makefile.in gmyclient-0.3/plugins/exporter/html/Makefile.in
--- gmyclient-0.3.orig/plugins/exporter/html/Makefile.in	Wed Apr 17 08:34:53 2002
+++ gmyclient-0.3/plugins/exporter/html/Makefile.in	Tue Apr 15 15:41:14 2003
@@ -144,7 +144,7 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-libhtml_la_LDFLAGS = 
+libhtml_la_LDFLAGS = -bundle -avoid-version
 libhtml_la_LIBADD = 
 libhtml_la_OBJECTS =  html.lo
 CFLAGS = @CFLAGS@
diff -ruN gmyclient-0.3.orig/plugins/exporter/sample/Makefile.in gmyclient-0.3/plugins/exporter/sample/Makefile.in
--- gmyclient-0.3.orig/plugins/exporter/sample/Makefile.in	Wed Apr 17 08:34:54 2002
+++ gmyclient-0.3/plugins/exporter/sample/Makefile.in	Tue Apr 15 15:41:35 2003
@@ -138,7 +138,7 @@
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
-libsample_la_LDFLAGS = 
+libsample_la_LDFLAGS = -bundle -avoid-version
 libsample_la_LIBADD = 
 libsample_la_OBJECTS =  sample.lo
 CFLAGS = @CFLAGS@
diff -ruN gmyclient-0.3.orig/src/gm-qbuffer.h gmyclient-0.3/src/gm-qbuffer.h
--- gmyclient-0.3.orig/src/gm-qbuffer.h	Tue Apr 16 08:28:52 2002
+++ gmyclient-0.3/src/gm-qbuffer.h	Tue Apr 15 15:39:16 2003
@@ -17,8 +17,11 @@
 typedef struct _GmQueryBufferClass  GmQueryBufferClass;
 
 typedef struct _Regex {
+#ifndef __APPLE__
     struct re_pattern_buffer buf;
     struct re_registers reg;
+#else
+#endif /*darwin*/
 } Regex;
 
 
