Index: kdecore/Makefile.am
===================================================================
RCS file: /home/kde/kdelibs/kdecore/Makefile.am,v
retrieving revision 1.343
diff -u -u -r1.343 Makefile.am
--- kdecore/Makefile.am	2 Oct 2003 14:42:21 -0000	1.343
+++ kdecore/Makefile.am	20 Nov 2003 00:43:48 -0000
@@ -24,7 +24,7 @@
 
 if include_SVGICONS
 SVGICONS=svgicons
-SVGICON_LIB=svgicons/libkdesvgicons.la
+SVGICON_LIB=svgicons/libkdesvgicons.la $(LIBZ)
 endif
 
 SUBDIRS = malloc $(SVGICONS) . kconfig_compiler tests
Index: kdecore/kglobalsettings.cpp
===================================================================
RCS file: /home/kde/kdelibs/kdecore/kglobalsettings.cpp,v
retrieving revision 1.76
diff -u -u -r1.76 kglobalsettings.cpp
--- kdecore/kglobalsettings.cpp	26 Oct 2003 20:42:15 -0000	1.76
+++ kdecore/kglobalsettings.cpp	20 Nov 2003 00:43:49 -0000
@@ -466,7 +466,7 @@
       s_desktopPath->append('/');
 
     // Trash Path
-    *s_trashPath = *s_desktopPath + i18n("Trash") + "/";
+    *s_trashPath = *s_desktopPath + "." + i18n("Trash") + "/";
     *s_trashPath = config->readPathEntry( "Trash" , *s_trashPath);
     if ( !s_trashPath->startsWith("/") )
       s_trashPath->prepend( QDir::homeDirPath() + "/" );
Index: kdecore/kpty.cpp
===================================================================
RCS file: /home/kde/kdelibs/kdecore/kpty.cpp,v
retrieving revision 1.9
diff -u -u -r1.9 kpty.cpp
--- kdecore/kpty.cpp	29 Sep 2003 08:52:52 -0000	1.9
+++ kdecore/kpty.cpp	20 Nov 2003 00:43:49 -0000
@@ -225,6 +225,7 @@
       struct group* p = getgrnam(TTY_GROUP);    /* posix */
       gid_t gid = p ? p->gr_gid : getgid ();    /* posix */
 
+#if !defined(__APPLE__)
       if (fchown(d->slaveFd, (uid_t) -1, gid) < 0)
       {
          int e = errno;
@@ -239,6 +240,7 @@
          kdWarning(175) << "Cannot chmod " << d->ttyName << endl
                         << "Reason " << strerror(e) << endl;
       }
+#endif
       goto gotpty;
     }
 #endif
Index: kdecore/svgicons/Makefile.am
===================================================================
RCS file: /home/kde/kdelibs/kdecore/svgicons/Makefile.am,v
retrieving revision 1.7
diff -u -u -r1.7 Makefile.am
--- kdecore/svgicons/Makefile.am	24 Aug 2003 09:58:48 -0000	1.7
+++ kdecore/svgicons/Makefile.am	20 Nov 2003 00:43:49 -0000
@@ -4,3 +4,4 @@
 
 include_HEADERS = ksvgiconengine.h
 libkdesvgicons_la_SOURCES = ksvgiconengine.cpp ksvgiconpainter.cpp
+libkdesvgicons_la_LIBADD = $(LIBZ)
Index: kparts/Makefile.am
===================================================================
RCS file: /home/kde/kdelibs/kparts/Makefile.am,v
retrieving revision 1.49
diff -u -r1.49 Makefile.am
--- kparts/Makefile.am	20 Aug 2003 09:19:41 -0000	1.49
+++ kparts/Makefile.am	22 Nov 2003 14:08:29 -0000
@@ -1,5 +1,7 @@
 ####### General stuff
 
+KDE_CXXFLAGS = $(USE_EXCEPTIONS)
+
 INCLUDES= -I$(srcdir)/../ -I$(top_srcdir)/kio/ -I$(top_srcdir)/libltdl \
 	-I$(top_srcdir)/kfile $(all_includes)
 libkparts_la_LDFLAGS = $(KDE_MT_LDFLAGS) -version-info 3:0:1 -no-undefined
Index: kio/kio/Makefile.am
===================================================================
RCS file: /home/kde/kdelibs/kio/kio/Makefile.am,v
retrieving revision 1.208
diff -u -r1.208 Makefile.am
--- kio/kio/Makefile.am	18 Oct 2003 19:45:40 -0000	1.208
+++ kio/kio/Makefile.am	22 Nov 2003 14:08:29 -0000
@@ -17,6 +17,7 @@
 #    Boston, MA 02111-1307, USA.
 
 AM_CPPFLAGS = -D_LARGEFILE64_SOURCE
+KDE_CXXFLAGS = $(USE_EXCEPTIONS)
 
 INCLUDES=  -I$(top_srcdir) -I$(srcdir)/.. -I$(srcdir)/../kssl -I../kssl -I$(srcdir)/../../interfaces  $(all_includes) $(SSL_INCLUDES)
 
