diff -ruN kmysql-1.2.1.orig/configure kmysql-1.2.1/configure
--- kmysql-1.2.1.orig/configure	Wed Dec  8 21:07:07 1999
+++ kmysql-1.2.1/configure	Tue Apr 15 12:57:33 2003
@@ -1967,7 +1967,7 @@
 
 CXXFLAGS="$CXXFLAGS -I$qt_incdir"
 LDFLAGS="-L$qt_libdir $X_LDFLAGS"
-LIBS="$LIBS -lqt -lXext -lX11 $LIBSOCKET"
+LIBS="$LIBS -lqt-mt -lXext -lX11 $LIBSOCKET"
 
 
 ac_ext=C
@@ -2054,7 +2054,7 @@
 
 CXXFLAGS="$CXXFLAGS -I$qt_includes"
 LDFLAGS="$X_LDFLAGS"
-LIBS="-lqt -lXext -lX11 $LIBSOCKET"
+LIBS="-lqt-mt -lXext -lX11 $LIBSOCKET"
 LD_LIBRARY_PATH=
 export LD_LIBRARY_PATH
 LIBRARY_PATH=
@@ -3592,19 +3592,6 @@
 #include <kapp.h>
 
 int main() {
-printf("kde_htmldir=\\"%s\\"\n", KApplication::kde_htmldir().data());
-printf("kde_appsdir=\\"%s\\"\n", KApplication::kde_appsdir().data());
-printf("kde_icondir=\\"%s\\"\n", KApplication::kde_icondir().data());
-printf("kde_sounddir=\\"%s\\"\n", KApplication::kde_sounddir().data());
-printf("kde_datadir=\\"%s\\"\n", KApplication::kde_datadir().data());
-printf("kde_locale=\\"%s\\"\n", KApplication::kde_localedir().data());
-printf("kde_cgidir=\\"%s\\"\n", KApplication::kde_cgidir().data());
-printf("kde_confdir=\\"%s\\"\n", KApplication::kde_configdir().data());
-printf("kde_mimedir=\\"%s\\"\n", KApplication::kde_mimedir().data());
-printf("kde_toolbardir=\\"%s\\"\n", KApplication::kde_toolbardir().data());
-printf("kde_wallpaperdir=\\"%s\\"\n", KApplication::kde_wallpaperdir().data());
-printf("kde_bindir=\\"%s\\"\n", KApplication::kde_bindir().data());
-printf("kde_partsdir=\\"%s\\"\n", KApplication::kde_partsdir().data());
 return 0;
 }
 EOF
@@ -3633,7 +3620,7 @@
 
 echo $ac_n "checking for kde libraries installed""... $ac_c" 1>&6
 echo "configure:3636: checking for kde libraries installed" >&5
-ac_link='/bin/sh ./libtool --mode=link ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS $all_libraries conftest.$ac_ext $LIBS -lkdecore -lqt -lXext -lX11 $LIBSOCKET $KDE_TEST_RPATH 1>&5'
+ac_link='/bin/sh ./libtool --mode=link ${CXX-g++} -o conftest $CXXFLAGS $all_includes $CPPFLAGS $LDFLAGS $all_libraries conftest.$ac_ext $LIBS -lkdecore -lqt-mt -lXext -lX11 $LIBSOCKET $KDE_TEST_RPATH 1>&5'
 
 if { (eval echo configure:3639: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest; then 
   echo "$ac_t""yes" 1>&6
@@ -3756,7 +3743,7 @@
 
    LIB_X11='-lX11 $(LIBSOCKET)'
    
-   LIB_QT='-lqt $(LIB_X11)'
+   LIB_QT='-lqt-mt $(LIB_X11)'
    
    LIB_KDECORE='-lkdecore -lXext $(LIB_QT)'
    
