diff -ruN xine-lib-1-beta2.orig/configure xine-lib-1-beta2/configure
--- xine-lib-1-beta2.orig/configure	2003-01-01 17:57:33.000000000 -0700
+++ xine-lib-1-beta2/configure	2003-01-04 01:21:57.000000000 -0700
@@ -6519,7 +6519,7 @@
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
     #	     `"' quotes if we put them in here... so don't!
-    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
@@ -12086,7 +12086,7 @@
 
   # It would also be nice to do this for all -L options, not just this one.
   if test -n "$x_libraries"; then
-    X_LIBS="$X_LIBS -L$x_libraries"
+    X_LIBS="$X_LIBS -L$x_libraries -lX11"
     # For Solaris; some versions of Sun CC require a space after -R and
     # others require no space.  Words are not sufficient . . . .
     case `(uname -sr) 2>/dev/null` in
diff -ruN xine-lib-1-beta2.orig/src/input/libdvdnav/vm.c xine-lib-1-beta2/src/input/libdvdnav/vm.c
--- xine-lib-1-beta2.orig/src/input/libdvdnav/vm.c	2002-11-24 14:08:51.000000000 -0700
+++ xine-lib-1-beta2/src/input/libdvdnav/vm.c	2003-01-04 13:59:47.000000000 -0700
@@ -178,7 +178,11 @@
 
 void dvd_read_name( vm_t *this, char *devname) {
     int fd, i;
+#if defined(__APPLE__) && defined(__GNUC__)
+    off_t	off;
+#else
     off64_t off;
+#endif
     uint8_t data[DVD_VIDEO_LB_LEN];
 
     /* Read DVD name */
diff -ruN xine-lib-1-beta2.orig/src/input/libreal/rmff.h xine-lib-1-beta2/src/input/libreal/rmff.h
--- xine-lib-1-beta2.orig/src/input/libreal/rmff.h	2002-12-22 09:46:27.000000000 -0700
+++ xine-lib-1-beta2/src/input/libreal/rmff.h	2003-01-04 12:12:38.000000000 -0700
@@ -23,6 +23,8 @@
  * adopted from joschkas real tools
  */
 
+#include <stdint.h>
+#include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <netdb.h>
@@ -32,7 +34,6 @@
 #include <stdlib.h>
 #include <string.h>
 
-
 #ifndef HAVE_RMFF_H
 #define HAVE_RMFF_H
 
diff -ruN xine-lib-1-beta2.orig/src/input/librtsp/rtsp.c xine-lib-1-beta2/src/input/librtsp/rtsp.c
--- xine-lib-1-beta2.orig/src/input/librtsp/rtsp.c	2002-12-21 17:35:05.000000000 -0700
+++ xine-lib-1-beta2/src/input/librtsp/rtsp.c	2003-01-04 12:25:45.000000000 -0700
@@ -26,6 +26,8 @@
 #include <unistd.h>
 #include <stdio.h>
 #include <assert.h>
+#include <stdint.h>
+#include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <netdb.h>
diff -ruN xine-lib-1-beta2.orig/src/input/librtsp/rtsp_session.c xine-lib-1-beta2/src/input/librtsp/rtsp_session.c
--- xine-lib-1-beta2.orig/src/input/librtsp/rtsp_session.c	2002-12-24 02:31:42.000000000 -0700
+++ xine-lib-1-beta2/src/input/librtsp/rtsp_session.c	2003-01-04 12:51:23.000000000 -0700
@@ -22,6 +22,8 @@
  * high level interface to rtsp servers.
  */
 
+#include <stdint.h>
+#include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <netdb.h>
diff -ruN xine-lib-1-beta2.orig/src/input/ost/frontend.h xine-lib-1-beta2/src/input/ost/frontend.h
--- xine-lib-1-beta2.orig/src/input/ost/frontend.h	2003-01-01 14:38:43.000000000 -0700
+++ xine-lib-1-beta2/src/input/ost/frontend.h	2003-01-05 01:55:48.000000000 -0700
@@ -27,8 +27,10 @@
 #if defined(__sun)
 #include <sys/ioccom.h>
 #else
+#if !defined(__APPLE__) && !defined(__GNUC__)
 #include <asm/types.h>
 #endif
+#endif
 
 #define ENOSIGNAL 768
 #ifndef EBUFFEROVERFLOW
diff -ruN xine-lib-1-beta2.orig/src/libffmpeg/libavcodec/error_resilience.c xine-lib-1-beta2/src/libffmpeg/libavcodec/error_resilience.c
--- xine-lib-1-beta2.orig/src/libffmpeg/libavcodec/error_resilience.c	2002-12-22 13:48:26.000000000 -0700
+++ xine-lib-1-beta2/src/libffmpeg/libavcodec/error_resilience.c	2003-01-05 02:35:26.000000000 -0700
@@ -18,7 +18,9 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#if !defined(__APPLE__) && !defined(__GNUC__)
 #include <alloca.h>
+#endif
 
 #include "avcodec.h"
 #include "dsputil.h"
diff -ruN xine-lib-1-beta2.orig/src/video_out/video_out_syncfb.c xine-lib-1-beta2/src/video_out/video_out_syncfb.c
--- xine-lib-1-beta2.orig/src/video_out/video_out_syncfb.c	2002-12-22 13:48:28.000000000 -0700
+++ xine-lib-1-beta2/src/video_out/video_out_syncfb.c	2003-01-04 01:46:41.000000000 -0700
@@ -41,7 +41,7 @@
 #endif
 
 #include <sys/ioctl.h>
-#if defined (__FreeBSD__)
+#if defined (__FreeBSD__) || (defined(__APPLE__) && defined(__GNUC__))
 #include <sys/types.h>
 #endif
 #include <sys/mman.h>
diff -ruN xine-lib-1-beta2.orig/src/xine-engine/Makefile.in xine-lib-1-beta2/src/xine-engine/Makefile.in
--- xine-lib-1-beta2.orig/src/xine-engine/Makefile.in	2003-01-01 17:57:27.000000000 -0700
+++ xine-lib-1-beta2/src/xine-engine/Makefile.in	2003-01-04 01:14:17.000000000 -0700
@@ -216,7 +216,7 @@
 	video_overlay.c osd.c scratch.c locale.c demux.c vo_scale.c \
 	xine_interface.c post.c
 
-libxine_la_DEPENDENCIES = @INTLLIBS@ $(XINEUTILS_LIB)
+libxine_la_DEPENDENCIES = $(XINEUTILS_LIB)
 
 libxine_la_LIBADD = $(THREAD_LIBS) $(DYNAMIC_LD_LIBS) @INTLLIBS@ $(ZLIB_LIBS)\
 	             -lm $(XINEUTILS_LIB)
diff -ruN xine-lib-1-beta2.orig/src/xine-utils/ppcasm_string.S xine-lib-1-beta2/src/xine-utils/ppcasm_string.S
--- xine-lib-1-beta2.orig/src/xine-utils/ppcasm_string.S	2002-10-24 12:49:04.000000000 -0600
+++ xine-lib-1-beta2/src/xine-utils/ppcasm_string.S	2003-01-05 00:04:58.000000000 -0700
@@ -21,11 +21,15 @@
 #define __KERNEL__
 #define __ASSEMBLY__
 
-#include "ppc_asm.tmpl"
-#include <linux/config.h>
-#include <asm/processor.h>
-#include <asm/cache.h>
-#include <asm/errno.h>
+#if defined(__APPLE__) && defined(__GNUC__)
+# define ASSEMBLER
+#else
+# include "ppc_asm.tmpl"
+# include <linux/config.h>
+# include <asm/processor.h>
+# include <asm/cache.h>
+# include <asm/errno.h>
+#endif
 
 #define COPY_16_BYTES		\
 	lwz	r7,4(r4);	\
@@ -80,7 +84,7 @@
 LG_CACHELINE_BYTES = 5
 CACHELINE_MASK = (32 -1)
 
-	.global	ppcasm_cacheable_memcpy
+	.globl ppcasm_cacheable_memcpy
 ppcasm_cacheable_memcpy:
 #if 0 /* this part causes "error loading shared library: unexpected reloc type
          0x0b (???) */
@@ -95,32 +99,32 @@
 	addi	r6,r3,-4
 	neg	r0,r3
 	andi.	r0,r0,CACHELINE_MASK	/* # bytes to start of cache line */
-	beq	58f
+	beq	L58
 
 	cmplw	0,r5,r0			/* is this more than total to do? */
-	blt	63f			/* if not much to do */
+	blt	L63			/* if not much to do */
 	andi.	r8,r0,3			/* get it word-aligned first */
 	subf	r5,r0,r5
 	mtctr	r8
-	beq+	61f
-70:	lbz	r9,4(r4)		/* do some bytes */
+	beq+	L61
+L70:	lbz	r9,4(r4)		/* do some bytes */
 	stb	r9,4(r6)
 	addi	r4,r4,1
 	addi	r6,r6,1
-	bdnz	70b
-61:	srwi.	r0,r0,2
+	bdnz	L70
+L61:	srwi.	r0,r0,2
 	mtctr	r0
-	beq	58f
-72:	lwzu	r9,4(r4)		/* do some words */
+	beq	L58
+L72:	lwzu	r9,4(r4)		/* do some words */
 	stwu	r9,4(r6)
-	bdnz	72b
+	bdnz	L72
 
-58:	srwi.	r0,r5,LG_CACHELINE_BYTES /* # complete cachelines */
+L58:	srwi.	r0,r5,LG_CACHELINE_BYTES /* # complete cachelines */
 	clrlwi	r5,r5,32-LG_CACHELINE_BYTES
 	li	r11,4
 	mtctr	r0
-	beq	63f
-53:
+	beq	L63
+L53:
 #if !defined(CONFIG_8xx)
 	dcbz	r11,r6
 #endif
@@ -138,65 +142,65 @@
 #endif
 #endif
 #endif
-	bdnz	53b
+	bdnz	L53
 
-63:	srwi.	r0,r5,2
+L63:	srwi.	r0,r5,2
 	mtctr	r0
-	beq	64f
-30:	lwzu	r0,4(r4)
+	beq	L64
+L30:	lwzu	r0,4(r4)
 	stwu	r0,4(r6)
-	bdnz	30b
+	bdnz	L30
 
-64:	andi.	r0,r5,3
+L64:	andi.	r0,r5,3
 	mtctr	r0
-	beq+	65f
-40:	lbz	r0,4(r4)
+	beq+	L65
+L40:	lbz	r0,4(r4)
 	stb	r0,4(r6)
 	addi	r4,r4,1
 	addi	r6,r6,1
-	bdnz	40b
-65:	blr
+	bdnz	L40
+L65:	blr
 
 	.globl	ppcasm_memcpy
 ppcasm_memcpy:
 	srwi.	r7,r5,3
 	addi	r6,r3,-4
 	addi	r4,r4,-4
-	beq	2f			/* if less than 8 bytes to do */
+	beq	L2			/* if less than 8 bytes to do */
 	andi.	r0,r6,3			/* get dest word aligned */
 	mtctr	r7
-	bne	5f
-1:	lwz	r7,4(r4)
+	bne	L5
+L1:	lwz	r7,4(r4)
 	lwzu	r8,8(r4)
 	stw	r7,4(r6)
 	stwu	r8,8(r6)
-	bdnz	1b
+	bdnz	L1
 	andi.	r5,r5,7
-2:	cmplwi	0,r5,4
-	blt	3f
+L2:	cmplwi	0,r5,4
+	blt	L3
 	lwzu	r0,4(r4)
 	addi	r5,r5,-4
 	stwu	r0,4(r6)
-3:	cmpwi	0,r5,0
+L3:	cmpwi	0,r5,0
 	beqlr
 	mtctr	r5
 	addi	r4,r4,3
 	addi	r6,r6,3
-4:	lbzu	r0,1(r4)
+L4:	lbzu	r0,1(r4)
 	stbu	r0,1(r6)
-	bdnz	4b
+	bdnz	L4
 	blr
-5:	subfic	r0,r0,4
+L5:	subfic	r0,r0,4
 	mtctr	r0
-6:	lbz	r7,4(r4)
+L6:	lbz	r7,4(r4)
 	addi	r4,r4,1
 	stb	r7,4(r6)
 	addi	r6,r6,1
-	bdnz	6b
+	bdnz	L6
 	subf	r5,r0,r5
 	rlwinm.	r7,r5,32-3,3,31
-	beq	2b
+	beq	L2
 	mtctr	r7
-	b	1b
+	b	L1
 
 #endif /* ARCH_PPC */
