diff -ruN xine-lib-1-beta2.orig/config.h.in xine-lib-1-beta2/config.h.in
--- xine-lib-1-beta2.orig/config.h.in	2003-01-01 18:01:19.000000000 -0700
+++ xine-lib-1-beta2/config.h.in	2003-01-06 11:05:50.000000000 -0700
@@ -6,6 +6,9 @@
 /* Define this if you're running x86 architecture */
 #undef ARCH_X86
 
+/* Define this if you're running Darwin OS or Mac OS X*/
+#undef DARWIN
+
 /* maximum supported data alignment */
 #undef ATTRIBUTE_ALIGNED_MAX
 
diff -ruN xine-lib-1-beta2.orig/configure xine-lib-1-beta2/configure
--- xine-lib-1-beta2.orig/configure	2003-01-01 17:57:33.000000000 -0700
+++ xine-lib-1-beta2/configure	2003-01-06 16:42:58.000000000 -0700
@@ -6519,7 +6519,7 @@
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
     #	     `"' quotes if we put them in here... so don't!
-    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
@@ -11727,7 +11727,7 @@
 LIBMPEG2_CFLAGS="" LIBA52_CFLAGS=""
 LIBFFMPEG_CFLAGS="-DSIMPLE_IDCT -DHAVE_AV_CONFIG_H -DRUNTIME_CPUDETECT"
 
-# Check whether --enable-altivec or --disable-altivec was given.
+# Check whether --enable-altivec or --disable-altivec was given.   
 if test "${enable_altivec+set}" = set; then
   enableval="$enable_altivec"
   enable_altivec=no
@@ -12086,7 +12086,7 @@
 
   # It would also be nice to do this for all -L options, not just this one.
   if test -n "$x_libraries"; then
-    X_LIBS="$X_LIBS -L$x_libraries"
+    X_LIBS="$X_LIBS -L$x_libraries -lX11"
     # For Solaris; some versions of Sun CC require a space after -R and
     # others require no space.  Words are not sufficient . . . .
     case `(uname -sr) 2>/dev/null` in
@@ -12197,7 +12197,7 @@
     # Martyn Johnson says this is needed for Ultrix, if the X
     # libraries were built with DECnet support.  And Karl Berry says
     # the Alpha needs dnet_stub (dnet does not exist).
-    ac_xsave_LIBS="$LIBS"; LIBS="$LIBS $X_LIBS -lX11"
+    ac_xsave_LIBS="$LIBS"; LIBS="$LIBS $X_LIBS"
     cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 #include "confdefs.h"
@@ -13710,6 +13710,12 @@
 
   ppc-*-linux* | powerpc-*)
         ppc_arch="yes"
+
+        case "$host_or_hostalias" in
+          *-darwin*)
+            darwin="yes"
+            ;;
+        esac
         ;;
 
   *)
@@ -13737,6 +13743,14 @@
   PPC_ARCH_FALSE=
 fi
 
+if test x$darwin = "xyes"; then
+  DARWIN_TRUE=
+  DARWIN_FALSE='#'
+else
+  DARWIN_TRUE='#'
+  DARWIN_FALSE=
+fi
+
 
 
 
@@ -19779,14 +19793,33 @@
 
 
     if test x$enable_altivec = xyes; then
+      echo "ALTIVEC support will be used"
 
 cat >>confdefs.h <<_ACEOF
 #define ENABLE_ALTIVEC
 _ACEOF
 
+      case "$host_or_hostalias" in
+      *-darwin*)
+        CFLAGS="$CFLAGS -faltivec"  
+        DEBUG_CFLAGS="$DEBUG_CFLAGS -faltivec"
+        ;;
+      *)
         CFLAGS="$CFLAGS -Wa,-m7400"
 	DEBUG_CFLAGS="$DEBUG_CFLAGS -Wa,-m7400"
+        ;;
+      esac
+    else
+      echo "ALTIVEC support will *NOT* be used"
     fi
+
+    case "$host_or_hostalias" in
+    *-darwin*)
+cat >>confdefs.h <<_ACEOF 
+#define DARWIN
+_ACEOF
+      ;;
+    esac
     ;;
 
   sparc*-*-linux*)
@@ -20296,6 +20329,13 @@
 Usually this means the macro was only invoked conditionally." >&2;}
    { (exit 1); exit 1; }; }
 fi
+if test -z "${DARWIN_TRUE}" && test -z "${DARWIN_FALSE}"; then
+  { { echo "$as_me:$LINENO: error: conditional \"DARWIN\" was never defined.
+Usually this means the macro was only invoked conditionally." >&5
+echo "$as_me: error: conditional \"DARWIN\" was never defined.
+Usually this means the macro was only invoked conditionally." >&2;}
+   { (exit 1); exit 1; }; }
+fi
 if test -z "${HAVE_XV_TRUE}" && test -z "${HAVE_XV_FALSE}"; then
   { { echo "$as_me:$LINENO: error: conditional \"HAVE_XV\" was never defined.
 Usually this means the macro was only invoked conditionally." >&5
@@ -21206,6 +21246,8 @@
 s,@HAVE_GICONV_FALSE@,$HAVE_GICONV_FALSE,;t t
 s,@PPC_ARCH_TRUE@,$PPC_ARCH_TRUE,;t t
 s,@PPC_ARCH_FALSE@,$PPC_ARCH_FALSE,;t t
+s,@DARWIN_TRUE@,$DARWIN_TRUE,;t t
+s,@DARWIN_FALSE@,$DARWIN_FALSE,;t t
 s,@HAVE_XV_TRUE@,$HAVE_XV_TRUE,;t t
 s,@HAVE_XV_FALSE@,$HAVE_XV_FALSE,;t t
 s,@XV_LIB@,$XV_LIB,;t t
diff -ruN xine-lib-1-beta2.orig/src/input/libdvdnav/vm.c xine-lib-1-beta2/src/input/libdvdnav/vm.c
--- xine-lib-1-beta2.orig/src/input/libdvdnav/vm.c	2002-11-24 14:08:51.000000000 -0700
+++ xine-lib-1-beta2/src/input/libdvdnav/vm.c	2003-01-06 08:33:20.000000000 -0700
@@ -178,7 +178,11 @@
 
 void dvd_read_name( vm_t *this, char *devname) {
     int fd, i;
+#ifdef DARWIN
+    off_t	off;
+#else
     off64_t off;
+#endif
     uint8_t data[DVD_VIDEO_LB_LEN];
 
     /* Read DVD name */
diff -ruN xine-lib-1-beta2.orig/src/input/libreal/rmff.h xine-lib-1-beta2/src/input/libreal/rmff.h
--- xine-lib-1-beta2.orig/src/input/libreal/rmff.h	2002-12-22 09:46:27.000000000 -0700
+++ xine-lib-1-beta2/src/input/libreal/rmff.h	2003-01-06 11:37:14.000000000 -0700
@@ -23,6 +23,8 @@
  * adopted from joschkas real tools
  */
 
+#include <stdint.h>
+#include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <netdb.h>
diff -ruN xine-lib-1-beta2.orig/src/input/librtsp/rtsp.c xine-lib-1-beta2/src/input/librtsp/rtsp.c
--- xine-lib-1-beta2.orig/src/input/librtsp/rtsp.c	2002-12-21 17:35:05.000000000 -0700
+++ xine-lib-1-beta2/src/input/librtsp/rtsp.c	2003-01-06 11:57:17.000000000 -0700
@@ -26,6 +26,8 @@
 #include <unistd.h>
 #include <stdio.h>
 #include <assert.h>
+#include <stdint.h>
+#include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <netdb.h>
diff -ruN xine-lib-1-beta2.orig/src/input/librtsp/rtsp_session.c xine-lib-1-beta2/src/input/librtsp/rtsp_session.c
--- xine-lib-1-beta2.orig/src/input/librtsp/rtsp_session.c	2002-12-24 02:31:42.000000000 -0700
+++ xine-lib-1-beta2/src/input/librtsp/rtsp_session.c	2003-01-06 11:57:32.000000000 -0700
@@ -22,6 +22,8 @@
  * high level interface to rtsp servers.
  */
 
+#include <stdint.h>
+#include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <netdb.h>
diff -ruN xine-lib-1-beta2.orig/src/input/ost/frontend.h xine-lib-1-beta2/src/input/ost/frontend.h
--- xine-lib-1-beta2.orig/src/input/ost/frontend.h	2003-01-01 14:38:43.000000000 -0700
+++ xine-lib-1-beta2/src/input/ost/frontend.h	2003-01-06 08:35:46.000000000 -0700
@@ -27,8 +27,10 @@
 #if defined(__sun)
 #include <sys/ioccom.h>
 #else
+#ifndef DARWIN
 #include <asm/types.h>
 #endif
+#endif
 
 #define ENOSIGNAL 768
 #ifndef EBUFFEROVERFLOW
diff -ruN xine-lib-1-beta2.orig/src/libffmpeg/libavcodec/error_resilience.c xine-lib-1-beta2/src/libffmpeg/libavcodec/error_resilience.c
--- xine-lib-1-beta2.orig/src/libffmpeg/libavcodec/error_resilience.c	2002-12-22 13:48:26.000000000 -0700
+++ xine-lib-1-beta2/src/libffmpeg/libavcodec/error_resilience.c	2003-01-06 11:11:40.000000000 -0700
@@ -18,7 +18,9 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
+#ifdef HAVE_ALLOCA_H
 #include <alloca.h>
+#endif
 
 #include "avcodec.h"
 #include "dsputil.h"
diff -ruN xine-lib-1-beta2.orig/src/libffmpeg/libavcodec/ratecontrol.c xine-lib-1-beta2/src/libffmpeg/libavcodec/ratecontrol.c
--- xine-lib-1-beta2.orig/src/libffmpeg/libavcodec/ratecontrol.c	2002-12-22 13:48:26.000000000 -0700
+++ xine-lib-1-beta2/src/libffmpeg/libavcodec/ratecontrol.c	2003-01-06 11:12:03.000000000 -0700
@@ -18,7 +18,9 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 #include <math.h>
+#ifdef HAVE_ALLOCA_H
 #include <alloca.h>
+#endif
 #include "common.h"
 #include "avcodec.h"
 #include "dsputil.h"
diff -ruN xine-lib-1-beta2.orig/src/libffmpeg/libavcodec/svq1.c xine-lib-1-beta2/src/libffmpeg/libavcodec/svq1.c
--- xine-lib-1-beta2.orig/src/libffmpeg/libavcodec/svq1.c	2002-12-22 13:48:26.000000000 -0700
+++ xine-lib-1-beta2/src/libffmpeg/libavcodec/svq1.c	2003-01-06 11:12:27.000000000 -0700
@@ -26,7 +26,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#ifdef HAVE_ALLOCA_H
 #include <alloca.h>
+#endif
 
 #include "common.h"
 #include "avcodec.h"
diff -ruN xine-lib-1-beta2.orig/src/video_out/video_out_syncfb.c xine-lib-1-beta2/src/video_out/video_out_syncfb.c
--- xine-lib-1-beta2.orig/src/video_out/video_out_syncfb.c	2002-12-22 13:48:28.000000000 -0700
+++ xine-lib-1-beta2/src/video_out/video_out_syncfb.c	2003-01-06 08:40:07.000000000 -0700
@@ -41,7 +41,7 @@
 #endif
 
 #include <sys/ioctl.h>
-#if defined (__FreeBSD__)
+#if defined (__FreeBSD__) || defined(DARWIN)
 #include <sys/types.h>
 #endif
 #include <sys/mman.h>
diff -ruN xine-lib-1-beta2.orig/src/xine-engine/Makefile.in xine-lib-1-beta2/src/xine-engine/Makefile.in
--- xine-lib-1-beta2.orig/src/xine-engine/Makefile.in	2003-01-01 17:57:27.000000000 -0700
+++ xine-lib-1-beta2/src/xine-engine/Makefile.in	2003-01-06 08:41:00.000000000 -0700
@@ -216,7 +216,8 @@
 	video_overlay.c osd.c scratch.c locale.c demux.c vo_scale.c \
 	xine_interface.c post.c
 
-libxine_la_DEPENDENCIES = @INTLLIBS@ $(XINEUTILS_LIB)
+libxine_la_DEPENDENCIES = $(XINEUTILS_LIB) \
+@DARWIN_FALSE@		@INTLLIBS@
 
 libxine_la_LIBADD = $(THREAD_LIBS) $(DYNAMIC_LD_LIBS) @INTLLIBS@ $(ZLIB_LIBS)\
 	             -lm $(XINEUTILS_LIB)
diff -ruN xine-lib-1-beta2.orig/src/xine-utils/ppcasm_string.S xine-lib-1-beta2/src/xine-utils/ppcasm_string.S
--- xine-lib-1-beta2.orig/src/xine-utils/ppcasm_string.S	2002-10-24 12:49:04.000000000 -0600
+++ xine-lib-1-beta2/src/xine-utils/ppcasm_string.S	2003-01-06 11:21:33.000000000 -0700
@@ -16,16 +16,20 @@
 
 #ifdef ARCH_PPC
 
-#warning Be forewarned - using PowerPC assembly
-
 #define __KERNEL__
 #define __ASSEMBLY__
 
+#ifdef DARWIN
+#warning Be forewarned - using PowerPC assembly (Darwin)
+#define ASSEMBLER
+#else
+#warning Be forewarned - using PowerPC assembly
 #include "ppc_asm.tmpl"
 #include <linux/config.h>
 #include <asm/processor.h>
 #include <asm/cache.h>
 #include <asm/errno.h>
+#endif
 
 #define COPY_16_BYTES		\
 	lwz	r7,4(r4);	\
@@ -80,7 +84,7 @@
 LG_CACHELINE_BYTES = 5
 CACHELINE_MASK = (32 -1)
 
-	.global	ppcasm_cacheable_memcpy
+	.globl ppcasm_cacheable_memcpy
 ppcasm_cacheable_memcpy:
 #if 0 /* this part causes "error loading shared library: unexpected reloc type
          0x0b (???) */
@@ -95,32 +99,66 @@
 	addi	r6,r3,-4
 	neg	r0,r3
 	andi.	r0,r0,CACHELINE_MASK	/* # bytes to start of cache line */
+#ifdef DARWIN
+	beq	L58
+#else
 	beq	58f
+#endif
 
 	cmplw	0,r5,r0			/* is this more than total to do? */
+#ifdef DARWIN
+	blt	L63			/* if not much to do */
+#else
 	blt	63f			/* if not much to do */
+#endif
 	andi.	r8,r0,3			/* get it word-aligned first */
 	subf	r5,r0,r5
 	mtctr	r8
+#ifdef DARWIN
+	beq+	L61
+L70:	lbz	r9,4(r4)		/* do some bytes */
+#else
 	beq+	61f
 70:	lbz	r9,4(r4)		/* do some bytes */
+#endif
 	stb	r9,4(r6)
 	addi	r4,r4,1
 	addi	r6,r6,1
+#ifdef DARWIN
+	bdnz	L70
+L61:	srwi.	r0,r0,2
+#else
 	bdnz	70b
 61:	srwi.	r0,r0,2
+#endif
 	mtctr	r0
+#ifdef DARWIN 
+	beq	L58
+L72:	lwzu	r9,4(r4)		/* do some words */
+#else
 	beq	58f
 72:	lwzu	r9,4(r4)		/* do some words */
+#endif
 	stwu	r9,4(r6)
+#ifdef DARWIN
+	bdnz	L72
+
+L58:	srwi.	r0,r5,LG_CACHELINE_BYTES /* # complete cachelines */
+#else
 	bdnz	72b
 
 58:	srwi.	r0,r5,LG_CACHELINE_BYTES /* # complete cachelines */
+#endif
 	clrlwi	r5,r5,32-LG_CACHELINE_BYTES
 	li	r11,4
 	mtctr	r0
+#ifdef DARWIN
+	beq	L63
+L53:
+#else
 	beq	63f
 53:
+#endif
 #if !defined(CONFIG_8xx)
 	dcbz	r11,r6
 #endif
@@ -138,65 +176,138 @@
 #endif
 #endif
 #endif
+#ifdef DARWIN
+	bdnz	L53
+
+L63:	srwi.	r0,r5,2
+#else
 	bdnz	53b
 
 63:	srwi.	r0,r5,2
+#endif
 	mtctr	r0
+#ifdef DARWIN
+	beq	L64
+L30:	lwzu	r0,4(r4)
+#else
 	beq	64f
 30:	lwzu	r0,4(r4)
+#endif
 	stwu	r0,4(r6)
+#ifdef DARWIN
+	bdnz	L30
+
+L64:	andi.	r0,r5,3
+#else
 	bdnz	30b
 
 64:	andi.	r0,r5,3
+#endif
 	mtctr	r0
+#ifdef DARWIN
+	beq+	L65
+L40:	lbz	r0,4(r4)
+#else
 	beq+	65f
 40:	lbz	r0,4(r4)
+#endif
 	stb	r0,4(r6)
 	addi	r4,r4,1
 	addi	r6,r6,1
+#ifdef DARWIN 
+	bdnz	L40
+L65:	blr
+#else
 	bdnz	40b
 65:	blr
+#endif
 
 	.globl	ppcasm_memcpy
 ppcasm_memcpy:
 	srwi.	r7,r5,3
 	addi	r6,r3,-4
 	addi	r4,r4,-4
-	beq	2f			/* if less than 8 bytes to do */
+	beq	L2			/* if less than 8 bytes to do */
 	andi.	r0,r6,3			/* get dest word aligned */
 	mtctr	r7
+#ifdef DARWIN
+	bne	L5
+L1:	lwz	r7,4(r4)
+#else
 	bne	5f
 1:	lwz	r7,4(r4)
+#endif
 	lwzu	r8,8(r4)
 	stw	r7,4(r6)
 	stwu	r8,8(r6)
+#ifdef DARWIN
+	bdnz	L1
+#else
 	bdnz	1b
+#endif
 	andi.	r5,r5,7
-2:	cmplwi	0,r5,4
+#ifdef DARWIN
+L2:	cmplwi	0,r5,4
+	blt	L3
+#else
+2:	complwi	0,r5,4
 	blt	3f
+#endif
 	lwzu	r0,4(r4)
 	addi	r5,r5,-4
 	stwu	r0,4(r6)
+#ifdef DARWIN
+L3:	cmpwi	0,r5,0
+#else
 3:	cmpwi	0,r5,0
+#endif
 	beqlr
 	mtctr	r5
 	addi	r4,r4,3
 	addi	r6,r6,3
+#ifdef DARWIN
+L4:	lbzu	r0,1(r4)
+#else
 4:	lbzu	r0,1(r4)
+#endif
 	stbu	r0,1(r6)
+#ifdef DARWIN
+	bdnz	L4
+#else
 	bdnz	4b
+#endif
 	blr
+#ifdef DARWIN
+L5:	subfic	r0,r0,4
+#else
 5:	subfic	r0,r0,4
+#endif
 	mtctr	r0
+#ifdef DARWIN
+L6:	lbz	r7,4(r4)
+#else
 6:	lbz	r7,4(r4)
+#endif
 	addi	r4,r4,1
 	stb	r7,4(r6)
 	addi	r6,r6,1
+#ifdef DARWIN
+	bdnz	L6
+#else
 	bdnz	6b
+#endif
 	subf	r5,r0,r5
 	rlwinm.	r7,r5,32-3,3,31
+#ifdef DARWIN
+	beq	L2
+#else
 	beq	2b
+#endif
 	mtctr	r7
+#ifdef DARWIN
+	b	L1
+#else
 	b	1b
+#endif
 
 #endif /* ARCH_PPC */
