diff -ruN xine-lib-1-beta2.orig/configure xine-lib-1-beta2/configure
--- xine-lib-1-beta2.orig/configure	2003-01-01 17:57:33.000000000 -0700
+++ xine-lib-1-beta2/configure	2003-01-04 01:21:57.000000000 -0700
@@ -6519,7 +6519,7 @@
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.  Also zsh mangles
     #	     `"' quotes if we put them in here... so don't!
-    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
@@ -12086,7 +12086,7 @@
 
   # It would also be nice to do this for all -L options, not just this one.
   if test -n "$x_libraries"; then
-    X_LIBS="$X_LIBS -L$x_libraries"
+    X_LIBS="$X_LIBS -L$x_libraries -lX11"
     # For Solaris; some versions of Sun CC require a space after -R and
     # others require no space.  Words are not sufficient . . . .
     case `(uname -sr) 2>/dev/null` in
diff -ruN xine-lib-1-beta2.orig/src/input/libdvdnav/vm.c xine-lib-1-beta2/src/input/libdvdnav/vm.c
--- xine-lib-1-beta2.orig/src/input/libdvdnav/vm.c	2002-11-24 14:08:51.000000000 -0700
+++ xine-lib-1-beta2/src/input/libdvdnav/vm.c	2003-01-04 13:59:47.000000000 -0700
@@ -178,7 +178,11 @@
 
 void dvd_read_name( vm_t *this, char *devname) {
     int fd, i;
+#if defined(__APPLE__) && defined(__GNUC__)
+    off_t	off;
+#else
     off64_t off;
+#endif
     uint8_t data[DVD_VIDEO_LB_LEN];
 
     /* Read DVD name */
diff -ruN xine-lib-1-beta2.orig/src/input/libreal/rmff.h xine-lib-1-beta2/src/input/libreal/rmff.h
--- xine-lib-1-beta2.orig/src/input/libreal/rmff.h	2002-12-22 09:46:27.000000000 -0700
+++ xine-lib-1-beta2/src/input/libreal/rmff.h	2003-01-04 12:12:38.000000000 -0700
@@ -23,6 +23,8 @@
  * adopted from joschkas real tools
  */
 
+#include <stdint.h>
+#include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <netdb.h>
@@ -32,7 +34,6 @@
 #include <stdlib.h>
 #include <string.h>
 
-
 #ifndef HAVE_RMFF_H
 #define HAVE_RMFF_H
 
diff -ruN xine-lib-1-beta2.orig/src/input/librtsp/rtsp.c xine-lib-1-beta2/src/input/librtsp/rtsp.c
--- xine-lib-1-beta2.orig/src/input/librtsp/rtsp.c	2002-12-21 17:35:05.000000000 -0700
+++ xine-lib-1-beta2/src/input/librtsp/rtsp.c	2003-01-04 12:25:45.000000000 -0700
@@ -26,6 +26,8 @@
 #include <unistd.h>
 #include <stdio.h>
 #include <assert.h>
+#include <stdint.h>
+#include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <netdb.h>
diff -ruN xine-lib-1-beta2.orig/src/input/librtsp/rtsp_session.c xine-lib-1-beta2/src/input/librtsp/rtsp_session.c
--- xine-lib-1-beta2.orig/src/input/librtsp/rtsp_session.c	2002-12-24 02:31:42.000000000 -0700
+++ xine-lib-1-beta2/src/input/librtsp/rtsp_session.c	2003-01-04 12:51:23.000000000 -0700
@@ -22,6 +22,8 @@
  * high level interface to rtsp servers.
  */
 
+#include <stdint.h>
+#include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <netdb.h>
diff -ruN xine-lib-1-beta2.orig/src/video_out/video_out_syncfb.c xine-lib-1-beta2/src/video_out/video_out_syncfb.c
--- xine-lib-1-beta2.orig/src/video_out/video_out_syncfb.c	2002-12-22 13:48:28.000000000 -0700
+++ xine-lib-1-beta2/src/video_out/video_out_syncfb.c	2003-01-04 01:46:41.000000000 -0700
@@ -41,7 +41,7 @@
 #endif
 
 #include <sys/ioctl.h>
-#if defined (__FreeBSD__)
+#if defined (__FreeBSD__) || (defined(__APPLE__) && defined(__GNUC__))
 #include <sys/types.h>
 #endif
 #include <sys/mman.h>
diff -ruN xine-lib-1-beta2.orig/src/xine-engine/Makefile.in xine-lib-1-beta2/src/xine-engine/Makefile.in
--- xine-lib-1-beta2.orig/src/xine-engine/Makefile.in	2003-01-01 17:57:27.000000000 -0700
+++ xine-lib-1-beta2/src/xine-engine/Makefile.in	2003-01-04 01:14:17.000000000 -0700
@@ -216,7 +216,7 @@
 	video_overlay.c osd.c scratch.c locale.c demux.c vo_scale.c \
 	xine_interface.c post.c
 
-libxine_la_DEPENDENCIES = @INTLLIBS@ $(XINEUTILS_LIB)
+libxine_la_DEPENDENCIES = $(XINEUTILS_LIB)
 
 libxine_la_LIBADD = $(THREAD_LIBS) $(DYNAMIC_LD_LIBS) @INTLLIBS@ $(ZLIB_LIBS)\
 	             -lm $(XINEUTILS_LIB)
diff -ruN xine-lib-1-beta2.orig/src/xine-utils/Makefile.in xine-lib-1-beta2/src/xine-utils/Makefile.in
--- xine-lib-1-beta2.orig/src/xine-utils/Makefile.in	2003-01-01 17:57:28.000000000 -0700
+++ xine-lib-1-beta2/src/xine-utils/Makefile.in	2003-01-04 01:03:00.000000000 -0700
@@ -207,7 +207,7 @@
 
 EXTRA_DIST = ppcasm_string.S ppc_asm.tmpl
 
-@PPC_ARCH_TRUE@pppc_files = ppcasm_string.S 
+@PPC_ARCH_FALSE@pppc_files = ppcasm_string.S 
 
 AM_CFLAGS = $(THREAD_CFLAGS) @ANSI_FLAGS@
 
@@ -244,7 +244,7 @@
 
 libxineutils_la_LDFLAGS =
 libxineutils_la_DEPENDENCIES =
-@PPC_ARCH_TRUE@am__objects_1 = ppcasm_string.lo
+@PPC_ARCH_FALSE@am__objects_1 = ppcasm_string.lo
 am_libxineutils_la_OBJECTS = $(am__objects_1) cpu_accel.lo color.lo \
 	list.lo memcpy.lo monitor.lo utils.lo xine_check.lo \
 	xine_mutex.lo xmllexer.lo xmlparser.lo xine_buffer.lo
