diff -ruN mpg123.orig/Makefile mpg123/Makefile
--- mpg123.orig/Makefile	2001-01-23 11:00:46.000000000 -0700
+++ mpg123/Makefile	2003-01-06 16:33:58.000000000 -0700
@@ -4,9 +4,9 @@
 
 # Where to install binary and manpage on "make install":
 
-PREFIX=/usr/local
+PREFIX=@FINKPREFIX@
 BINDIR=$(PREFIX)/bin
-MANDIR=$(PREFIX)/man
+MANDIR=$(PREFIX)/share/man
 SECTION=1
 
 ########################################################
@@ -54,6 +54,8 @@
 	@echo "make netbsd         NetBSD"
 	@echo "make openbsd        OpenBSD"
 	@echo "make mint           MiNT on Atari"
+	@echo "make darwin-esd     Darwin (OSX) And ESound
+	@echo "make darwin-oss     Darwin (OSX) And OSS
 	@echo "make generic        try this one if your system isn't listed above"
 	@echo ""
 	@echo "Please read the file INSTALL for additional information."
@@ -738,6 +740,20 @@
 		AUDIO_LIB='-lsocket' \
 		mpg123-make
 
+darwin-esd:
+	$(MAKE) CC=cc LDFLAGS='$(LDFLAGS)' AUDIO_LIB='-lesd -laudiofile' \
+		OBJECTS='decode.o dct64.o audio_esd.o' \
+		CFLAGS='$(CFLAGS) -O -DDARWIN -Wall -O2 -DPPC_ENDIAN -DUSE_ESD' \
+		mpg123-make
+
+darwin-oss:
+	$(MAKE) CC=gcc LDFLAGS='$(LDFLAGS)' \
+		AUDIO_LIB='-L/sw/lib -loss -lesd -laudiofile' \
+		OBJECTS='decode.o dct64.o audio_oss.o' \
+		CFLAGS='$(CFLAGS) -O -Wall -O2 -DPPC_ENDIAN \
+			-DOSS -I/sw/include' \
+		mpg123-make
+
 # maybe you need the additonal options LDFLAGS='-lnsl -lsocket' when linking (see solaris:)
 generic:
 	$(MAKE) LDFLAGS= OBJECTS='decode.o dct64.o audio_dummy.o' \
diff -ruN mpg123.orig/audio.c mpg123/audio.c
--- mpg123.orig/audio.c	2001-01-18 07:00:33.000000000 -0700
+++ mpg123/audio.c	2003-01-06 16:13:33.000000000 -0700
@@ -1,4 +1,5 @@
 
+#include <stdlib.h>
 #include "mpg123.h"
 
 void audio_info_struct_init(struct audio_info_struct *ai)
diff -ruN mpg123.orig/audio_oss.c mpg123/audio_oss.c
--- mpg123.orig/audio_oss.c	2000-10-25 05:05:21.000000000 -0600
+++ mpg123/audio_oss.c	2003-01-06 16:27:40.000000000 -0700
@@ -30,6 +30,8 @@
 #include <linux/soundcard.h>
 #elif defined(__bsdi__)
 #include <sys/soundcard.h>
+#elif defined(__APPLE__) && defined(__GNUC__)
+#include <liboss/soundcard.h>
 #else
 #include <machine/soundcard.h>
 #endif
@@ -39,8 +41,7 @@
   if(!ai)
     return -1;
 
-  if(!ai->device)
-    ai->device = "/dev/dsp";
+  ai->device = "/dev/dsp";
 
   ai->fn = open(ai->device,O_WRONLY);  
 
