diff -ruN mpg123.orig/Makefile mpg123/Makefile
--- mpg123.orig/Makefile	Tue Jan 23 11:00:46 2001
+++ mpg123/Makefile	Fri Feb 21 23:44:26 2003
@@ -4,9 +4,9 @@
 
 # Where to install binary and manpage on "make install":
 
-PREFIX=/usr/local
+PREFIX=@FINKINSTALL@
 BINDIR=$(PREFIX)/bin
-MANDIR=$(PREFIX)/man
+MANDIR=$(PREFIX)/share/man
 SECTION=1
 
 ########################################################
@@ -54,6 +54,8 @@
 	@echo "make netbsd         NetBSD"
 	@echo "make openbsd        OpenBSD"
 	@echo "make mint           MiNT on Atari"
+	@echo "make darwin-esd     Darwin (OSX) And ESound
+	@echo "make darwin-oss     Darwin (OSX) And OSS
 	@echo "make generic        try this one if your system isn't listed above"
 	@echo ""
 	@echo "Please read the file INSTALL for additional information."
@@ -736,6 +738,20 @@
 		-finline-functions -ffast-math \
 		-DREAL_IS_FLOAT -DMINT -DNOXFERMEM' \
 		AUDIO_LIB='-lsocket' \
+		mpg123-make
+
+darwin-esd:
+	$(MAKE) CC=cc LDFLAGS='$(LDFLAGS)' AUDIO_LIB='-lesd -laudiofile' \
+		OBJECTS='decode.o dct64.o audio_esd.o' \
+		CFLAGS='$(CFLAGS) -O -DDARWIN -Wall -O2 -DPPC_ENDIAN -DUSE_ESD' \
+		mpg123-make
+
+darwin-oss:
+	$(MAKE) CC=gcc LDFLAGS='$(LDFLAGS)' \
+		AUDIO_LIB='-L@FINKREFIX@/lib -loss -lesd -laudiofile' \
+		OBJECTS='decode.o dct64.o audio_oss.o' \
+		CFLAGS='$(CFLAGS) -O -Wall -O2 \
+			-DOSS -DOSS_BIG_ENDIAN -I@FINKPREFIX@/include' \
 		mpg123-make
 
 # maybe you need the additonal options LDFLAGS='-lnsl -lsocket' when linking (see solaris:)
diff -ruN mpg123.orig/audio.c mpg123/audio.c
--- mpg123.orig/audio.c	Thu Jan 18 07:00:33 2001
+++ mpg123/audio.c	Fri Feb 21 23:41:08 2003
@@ -1,4 +1,5 @@
 
+#include <stdlib.h>
 #include "mpg123.h"
 
 void audio_info_struct_init(struct audio_info_struct *ai)
diff -ruN mpg123.orig/audio_oss.c mpg123/audio_oss.c
--- mpg123.orig/audio_oss.c	Wed Oct 25 05:05:21 2000
+++ mpg123/audio_oss.c	Fri Feb 21 23:41:08 2003
@@ -30,6 +30,8 @@
 #include <linux/soundcard.h>
 #elif defined(__bsdi__)
 #include <sys/soundcard.h>
+#elif defined(__APPLE__) && defined(__GNUC__)
+#include <liboss/soundcard.h>
 #else
 #include <machine/soundcard.h>
 #endif
@@ -39,8 +41,7 @@
   if(!ai)
     return -1;
 
-  if(!ai->device)
-    ai->device = "/dev/dsp";
+  ai->device = "/dev/dsp";
 
   ai->fn = open(ai->device,O_WRONLY);  
 
