diff -ruN oidentd-2.0.4.orig/fink/default oidentd-2.0.4/fink/default
--- oidentd-2.0.4.orig/fink/default	1969-12-31 17:00:00.000000000 -0700
+++ oidentd-2.0.4/fink/default	2002-10-15 17:16:16.000000000 -0600
@@ -0,0 +1,10 @@
+# options to use when starting oidentd as daemon:
+# -m	lookup masquaraded connections in /etc/oidentd.users
+# -f	forward requests for masquaraded connections to real host
+# -q	don't log connections to oidentd
+# see oidentd(8) for detailed list
+OIDENT_OPTIONS="-m"
+
+# user / group
+OIDENT_USER=nobody
+OIDENT_GROUP=nogroup
diff -ruN oidentd-2.0.4.orig/fink/init oidentd-2.0.4/fink/init
--- oidentd-2.0.4.orig/fink/init	1969-12-31 17:00:00.000000000 -0700
+++ oidentd-2.0.4/fink/init	2002-10-15 21:16:02.000000000 -0600
@@ -0,0 +1,52 @@
+#!/bin/sh
+#
+# Start/stops the oident daemon.
+#
+
+PATH=@FINKPREFIX@/sbin:@FINKPREFIX@/bin:/sbin:/bin:/usr/sbin:/usr/bin
+OIDENTD=@FINKPREFIX@/sbin/oidentd
+
+# See if the daemons are there
+test -f ${OIDENTD} || exit 0
+
+# oidentd configuration
+OIDENT_OPTIONS=""
+OIDENT_USER="nobody"
+OIDENT_GROUP="nogroup"
+test -f @FINKPREFIX@/etc/default/oidentd && . @FINKPREFIX@/etc/default/oidentd
+# remove obsolete options, these have to be moved to oidentd.conf
+# run 'oidentdconfig --show' to see how to do that
+OIDENT_OPTIONS=`echo "$OIDENT_OPTIONS"|
+	sed -e 's/-[isSnNrwW]//g' -e 's/-F/-f/' -e 's/-x/-r/'`
+
+OPTIONS="${OIDENT_OPTIONS} -u ${OIDENT_USER} -g ${OIDENT_GROUP}"
+
+case "$1" in
+	start)
+		echo -n "Starting ident daemon:"
+		echo -n " oidentd"
+		start-stop-daemon --start --quiet --exec ${OIDENTD} -- ${OPTIONS}
+		echo "."
+		;;
+	stop)
+		echo -n "Stopping ident daemon:"
+		echo -n " oidentd"
+		start-stop-daemon --stop --quiet --exec ${OIDENTD} -- ${OPTIONS}
+		echo "."
+		;;
+	reload|restart|force-reload)
+		echo -n "Restarting ident daemon:"
+		echo -n " oidentd"
+		start-stop-daemon --stop --quiet --exec ${OIDENTD} -- ${OPTIONS}
+		sleep 2
+		start-stop-daemon --start --quiet --exec ${OIDENTD} -- ${OPTIONS}
+		echo "."
+		;;
+	*)
+		echo "Usage: $0 {start|stop|restart|reload|force-reload}"
+		exit 1
+		;;
+esac
+
+exit 0
+
diff -ruN oidentd-2.0.4.orig/fink/oidentd.conf oidentd-2.0.4/fink/oidentd.conf
--- oidentd-2.0.4.orig/fink/oidentd.conf	1969-12-31 17:00:00.000000000 -0700
+++ oidentd-2.0.4/fink/oidentd.conf	2002-10-15 17:13:37.000000000 -0600
@@ -0,0 +1,22 @@
+# Configuration for oidentd
+# see oidentd.conf(5)
+# 
+default {
+	default {
+		deny spoof
+		deny spoof_all
+		deny spoof_privport
+		allow random
+		allow random_numeric
+		allow numeric
+		deny hide
+	}
+}
+
+# you may want to hide root connections
+#user "root" {
+#	default {
+#		force reply "UNKNOWN"
+#	}
+#}
+
diff -ruN oidentd-2.0.4.orig/fink/oidentdconfig oidentd-2.0.4/fink/oidentdconfig
--- oidentd-2.0.4.orig/fink/oidentdconfig	1969-12-31 17:00:00.000000000 -0700
+++ oidentd-2.0.4/fink/oidentdconfig	2002-10-15 21:14:35.000000000 -0600
@@ -0,0 +1,181 @@
+#!/bin/sh -e
+
+ACTION=probe
+case "$1" in
+	--help|-h)
+		echo "$0 [--show] [--exec]"
+		exit 0
+		;;
+	--show)
+		ACTION=show
+		;;
+	--exec)
+		ACTION=exec
+		;;
+esac
+
+# read old config
+. @FINKPREFIX@/etc/default/oidentd
+
+CLEAN=yes
+SPOOF=no
+SPOOFPRIV=no
+HIDE=no
+MASQ=no
+NEW_OPTIONS="$OIDENT_OPTIONS"
+case "$OIDENT_OPTIONS" in
+	*-A*)
+		SPOOFPRIV=yes
+		NEW_OPTIONS="`echo $NEW_OPTIONS|sed -e 's/-A//g'`"
+		;;
+esac
+case "$OIDENT_OPTIONS" in
+	*-s*)
+		test -f @FINKPREFIX@/etc/identd.spoof &&
+		grep -v '^#' @FINKPREFIX@/etc/identd.spoof | grep -qv '^$' &&
+		SPOOF=normal &&
+		CLEAN=no &&
+		NEW_OPTIONS="`echo $NEW_OPTIONS|sed -e 's/-s//g'`"
+		;;
+esac
+case "$OIDENT_OPTIONS" in
+	*-S*)
+		# all users but those in identd.spoof are allowed to spoof
+		SPOOF=reverse
+		NEW_OPTIONS="`echo $NEW_OPTIONS|sed -e 's/-S//g'`"
+		;;
+esac
+case "$OIDENT_OPTIONS" in
+	*-N*)
+		# users are allowed to hide them
+		CLEAN=no
+		HIDE=yes
+		NEW_OPTIONS="`echo $NEW_OPTIONS|sed -e 's/-N//g'`"
+		;;
+esac
+case "$OIDENT_OPTIONS" in
+	*-m*)
+		# handle new location of masquarading config file
+		test -f @FINKPREFIX@/etc/oidentd.users &&
+		grep -v '^#' @FINKPREFIX@/etc/oidentd.users | grep -qv '^$' &&
+		case "$ACTION" in
+		show)
+			echo "### Please move @FINKPREFIX@/etc/oidentd.users to @FINKPREFIX@/etc/oidentd_masq.conf"
+			echo
+			;;
+		exec)
+			mv @FINKPREFIX@/etc/oidentd.users @FINKPREFIX@/etc/oidentd_masq.conf
+			;;
+		esac &&
+		CLEAN=no
+		;;
+esac
+case "$OIDENT_OPTIONS" in
+	*-F*)
+		CLEAN=no
+		NEW_OPTIONS="`echo $NEW_OPTIONS|sed -e 's/-F/-f/g'`"
+		;;
+esac
+case "$OIDENT_OPTIONS" in
+	*-x*)
+		CLEAN=no
+		NEW_OPTIONS="`echo $NEW_OPTIONS|sed -e 's/-x/-r/g'`"
+		;;
+esac
+
+if [ "$SPOOF" = reverse -o "$HIDE" = yes ]; then
+	{
+		[ "$SPOOF" = reverse ] &&
+			echo "		allow spoof"
+		[ "$SPOOFPRIV" = yes ] &&
+			echo "		allow spoof_privport"
+		[ "$HIDE" = yes ] &&
+			echo "		allow hide"
+		echo
+	} | case "$ACTION" in
+	show)
+		echo "### Please add these lines to the default section of your @FINKPREFIX@/etc/oidentd.conf:"
+		cat
+		;;
+	exec)
+		ln @FINKPREFIX@/etc/oidentd.conf @FINKPREFIX@/etc/oidentd.conf.old &&
+		perl -pe 'if(/^\s+default/){print;system("cat");$_=""};' @FINKPREFIX@/etc/oidentd.conf > @FINKPREFIX@/etc/oidentd.conf.new &&
+		mv @FINKPREFIX@/etc/oidentd.conf.new @FINKPREFIX@/etc/oidentd.conf
+		;;
+	esac
+fi
+if [ "$SPOOF" != no ]; then
+	USER=undef
+	grep -v '^#' @FINKPREFIX@/etc/identd.spoof | grep -v '^$' | {
+	# convert old to new format
+	  IFS=:
+	  read USER REPLY
+	  while [ -n "$USER" ]; do
+		echo "user \"$USER\" {"
+		echo "	default {"
+		if [ -z "$REPLY" ]; then
+			if [ "$SPOOF" = reverse ]; then
+				echo "		deny spoof"
+			else
+				echo "		allow spoof"
+				[ "$SPOOFPRIV" = yes ] && echo "		allow spoof_privport"
+			fi
+		else
+			echo "		force reply \"$REPLY\""
+		fi
+		echo "	}"
+		echo "}"
+		echo
+		read USER REPLY
+	done } | case "$ACTION" in
+	show)
+		echo "### Please add these lines to your @FINKPREFIX@/etc/oidentd.conf:"
+		cat
+		echo "### afterwards, you may delete @FINKPREFIX@/etc/identd.spoof."
+		echo
+		;;
+	exec)
+		cat >> @FINKPREFIX@/etc/oidentd.conf
+		rm @FINKPREFIX@/etc/identd.spoof
+		;;
+	esac
+fi
+
+if [ "$OIDENT_OPTIONS" != "$NEW_OPTIONS" ]; then
+	case "$ACTION" in
+	show)
+		echo "### Edit @FINKPREFIX@/etc/default/oidentd to contain:"
+		echo "OIDENT_OPTIONS=\"$NEW_OPTIONS\""
+		echo
+		;;
+	exec)
+		cat @FINKPREFIX@/etc/default/oidentd |
+		sed -e 's/^OIDENT_OPTIONS=.*$/OIDENT_OPTIONS="'$NEW_OPTIONS'"/' > @FINKPREFIX@/etc/default/oidentd.new
+		mv @FINKPREFIX@/etc/default/oidentd.new @FINKPREFIX@/etc/default/oidentd
+		;;
+	esac
+fi
+
+if [ "$CLEAN" = yes ]; then
+	if [ "$ACTION" = show ]; then
+		echo "You already have a new oidentd config."
+	fi
+	exit 0
+fi
+
+case "$ACTION" in
+probe)
+	echo "Your new oidentd won't work with the old configuration files."
+	echo "please see '@FINKPREFIX@/sbin/oidentdconfig --show' to fix it."
+	exit 1
+	;;
+show)
+	echo "### you may run '@FINKPREFIX@/sbin/oidentdconfig --exec' to do all this automatically"
+	exit 1
+	;;
+exec)
+	echo "your oidentd config should work with the new oidentd now,"
+	echo "but please have a look at it for yourself before starting the daemon."
+	exit 0
+	;;
+esac
diff -ruN oidentd-2.0.4.orig/src/kernel/darwin.c oidentd-2.0.4/src/kernel/darwin.c
--- oidentd-2.0.4.orig/src/kernel/darwin.c	2002-08-08 17:23:54.000000000 -0600
+++ oidentd-2.0.4/src/kernel/darwin.c	2002-10-15 14:53:05.000000000 -0600
@@ -39,7 +39,7 @@
 #include <netinet/in_pcb.h>
 #include <arpa/inet.h>
 #include <netinet/tcp.h>
-#include <netinet/ip_compat.h>
+#include <netinet/ip.h>
 
 #define KERNEL
 #include <sys/file.h>
