diff -ruN php-4.3.0.orig/Makefile.global php-4.3.0/Makefile.global
--- php-4.3.0.orig/Makefile.global	Tue Nov 12 10:40:38 2002
+++ php-4.3.0/Makefile.global	Fri Jan 31 19:40:08 2003
@@ -6,7 +6,7 @@
 COMMON_FLAGS = $(DEFS) $(INCLUDES) $(EXTRA_INCLUDES) $(CPPFLAGS) $(PHP_FRAMEWORKPATH)
 
 
-all: $(all_targets) 
+all: $(all_targets) libphp4.la build-modules
 
 build-modules: $(PHP_MODULES)
 
@@ -14,23 +14,24 @@
 	$(LIBTOOL) --mode=link $(CC) $(CFLAGS) $(EXTRA_CFLAGS) -rpath $(phptempdir) $(EXTRA_LDFLAGS) $(LDFLAGS) $(PHP_RPATHS) $(PHP_GLOBAL_OBJS) $(PHP_SAPI_OBJS) $(EXTRA_LIBS) $(ZEND_EXTRA_LIBS) -o $@
 
 libs/libphp4.bundle: $(PHP_GLOBAL_OBJS) $(PHP_SAPI_OBJS)
-	$(CC) $(MH_BUNDLE_FLAGS) $(CFLAGS_CLEAN) $(EXTRA_CFLAGS) $(LDFLAGS) $(EXTRA_LDFLAGS) $(PHP_GLOBAL_OBJS:.lo=.o) $(PHP_SAPI_OBJS:.lo=.o) $(PHP_FRAMEWORKS) $(EXTRA_LIBS) $(ZEND_EXTRA_LIBS) -o $@ && cp $@ libs/libphp4.so
+	$(CXX) $(MH_BUNDLE_FLAGS) $(CFLAGS_CLEAN) $(EXTRA_CFLAGS) $(LDFLAGS) $(EXTRA_LDFLAGS) $(PHP_GLOBAL_OBJS:.lo=.o) $(PHP_SAPI_OBJS:.lo=.o) $(PHP_FRAMEWORKS) $(EXTRA_LIBS) $(ZEND_EXTRA_LIBS) -o $@ && cp $@ libs/libphp4.so
 
 install: $(install_targets)
 
 install-sapi: libphp4.la
-	@echo "Installing PHP SAPI module"
-	-@$(LIBTOOL) --silent --mode=install cp libphp4.la $(phptempdir)/libphp4.la >/dev/null 2>&1
-	-@$(mkinstalldirs) $(INSTALL_ROOT)$(bindir)
-	-@if test ! -r $(phptempdir)/libphp4.$(SHLIB_SUFFIX_NAME); then \
-		for i in 0.0.0 0.0 0; do \
-			if test -r $(phptempdir)/libphp4.$(SHLIB_SUFFIX_NAME).$$i; then \
-				$(LN_S) $(phptempdir)/libphp4.$(SHLIB_SUFFIX_NAME).$$i $(phptempdir)/libphp4.$(SHLIB_SUFFIX_NAME); \
-				break; \
-			fi; \
-		done; \
-	fi
-	@$(INSTALL_IT)
+#	@test -f libphp4.la && \
+#	$(mkinstalldirs) $(INSTALL_ROOT)$($libdir) && \
+#	echo "Installing PHP SAPI module"
+#	-@$(LIBTOOL) --silent --mode=install cp libphp4.la $(INSTALL_ROOT)$(libdir)/libphp4.la >/dev/null 2>&1
+#	-@if test ! -r $(INSTALL_ROOT)$(libdir)/libphp4.$(SHLIB_SUFFIX_NAME); then \
+#		for i in 0.0.0 0.0 0; do \
+#			if test -r $(INSTALL_ROOT)$(libdir)/libphp4.$(SHLIB_SUFFIX_NAME).$$i; then \
+#				$(LN_S) $(INSTALL_ROOT)$(libdir)/libphp4.$(SHLIB_SUFFIX_NAME).$$i $(INSTALL_ROOT)$(libdir)/libphp4.$(SHLIB_SUFFIX_NAME); \
+#				break; \
+#			fi; \
+#		done; \
+#	fi
+#	@$(INSTALL_IT)
 
 install-modules: build-modules
 	@test -d modules && \
diff -ruN php-4.3.0.orig/configure php-4.3.0/configure
--- php-4.3.0.orig/configure	Thu Dec 26 21:50:56 2002
+++ php-4.3.0/configure	Fri Jan 31 19:14:32 2003
@@ -4569,9 +4569,9 @@
 
 fi
 
-if test -n "$APACHE_INSTALL"; then
-  INSTALL_IT=$APACHE_INSTALL
-fi
+#if test -n "$APACHE_INSTALL"; then
+  # INSTALL_IT=$APACHE_INSTALL
+#fi
 
 
 
@@ -5674,7 +5674,7 @@
 
   case $host_alias in
   *darwin*)
-    BUILD_CLI="\$(CC) \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS) \$(LDFLAGS) \$(NATIVE_RPATHS) \$(PHP_GLOBAL_OBJS:.lo=.o) \$(PHP_CLI_OBJS:.lo=.o) \$(PHP_FRAMEWORKS) \$(EXTRA_LIBS) \$(ZEND_EXTRA_LIBS) -o \$(SAPI_CLI_PATH)"
+    BUILD_CLI="\$(CXX) \$(CFLAGS_CLEAN) \$(EXTRA_CFLAGS) \$(EXTRA_LDFLAGS) \$(LDFLAGS) \$(NATIVE_RPATHS) \$(PHP_GLOBAL_OBJS:.lo=.o) \$(PHP_CLI_OBJS:.lo=.o) \$(PHP_FRAMEWORKS) \$(EXTRA_LIBS) \$(ZEND_EXTRA_LIBS) -o \$(SAPI_CLI_PATH)"
     ;;
   *cygwin*)
     SAPI_CLI_PATH=sapi/cli/php.exe
@@ -62061,7 +62061,7 @@
 	
 
   if test "$ext_shared" = "yes"; then
-    PSPELL_SHARED_LIBADD="-lpspell $PSPELL_SHARED_LIBADD"
+    PSPELL_SHARED_LIBADD="-lpspell -lpspell-modules $PSPELL_SHARED_LIBADD"
     if test -n "$PSPELL_LIBDIR"; then
       
   if test "$PSPELL_LIBDIR" != "/usr/lib"; then
@@ -62144,7 +62144,7 @@
   case pspell in
   c|c_r|pthread*) ;;
   *) 
-      LIBS="-lpspell $LIBS" 
+      LIBS="-lpspell -lpspell-modules $LIBS" 
    ;;
   esac
 
@@ -62240,6 +62240,9 @@
       if test -f $QTDIR/include/qdom.h; then
         QTDOM_LIBDIR=$QTDIR/lib
         QTDOM_INCDIR=$QTDIR/include
+      elif test -f $QTDIR/include/qt/qdom.h; then
+        QTDOM_LIBDIR=$QTDIR/lib
+        QTDOM_INCDIR=$QTDIR/include/qt
       elif test -f /usr/lib/qt2/include/qdom.h; then
         QTDOM_LIBDIR=/usr/lib
         QTDOM_INCDIR=/usr/lib/qt2/include
@@ -62260,7 +62263,7 @@
   
 
   if test "$ext_shared" = "yes"; then
-    QTDOM_SHARED_LIBADD="-lqt $QTDOM_SHARED_LIBADD"
+    QTDOM_SHARED_LIBADD="-lqt-mt $QTDOM_SHARED_LIBADD"
     if test -n "$QTDOM_LIBDIR"; then
       
   if test "$QTDOM_LIBDIR" != "/usr/lib"; then
@@ -62343,7 +62346,7 @@
   case qt in
   c|c_r|pthread*) ;;
   *) 
-      LIBS="-lqt $LIBS" 
+      LIBS="-lqt-mt $LIBS" 
    ;;
   esac
 
