diff -ruN pmake-1.45.orig/Makefile.darwin pmake-1.45/Makefile.darwin
--- pmake-1.45.orig/Makefile.darwin	1969-12-31 17:00:00.000000000 -0700
+++ pmake-1.45/Makefile.darwin	2002-12-27 09:22:38.000000000 -0700
@@ -0,0 +1,41 @@
+#	$NetBSD: Makefile.boot,v 1.9 1998/05/21 17:19:46 tv Exp $
+#
+# a very simple makefile...
+#
+# You only want to use this if you aren't running NetBSD.
+#
+# modify MACHINE and MACHINE_ARCH as appropriate for your target architecture
+#
+
+.c.o:
+	${CC} ${CFLAGS} -c $< -o $@
+
+MACHINE=$(shell uname -s)
+MACHINE_ARCH=$(shell uname -p)
+CFLAGS= -O2  -g -Wall -I. -DMACHINE=\"${MACHINE}\" -DMACHINE_ARCH=\"${MACHINE_ARCH}\" -D_GNU_SOURCE 
+#-DMAKE_BOOTSTRAP
+
+LIBS=
+
+OBJ=arch.o buf.o compat.o cond.o dir.o for.o hash.o job.o main.o make.o \
+    parse.o str.o suff.o targ.o var.o util.o
+
+LIBOBJ= lst.lib/lstAppend.o lst.lib/lstAtEnd.o lst.lib/lstAtFront.o \
+	lst.lib/lstClose.o lst.lib/lstConcat.o lst.lib/lstDatum.o \
+	lst.lib/lstDeQueue.o lst.lib/lstDestroy.o lst.lib/lstDupl.o \
+	lst.lib/lstEnQueue.o lst.lib/lstFind.o lst.lib/lstFindFrom.o \
+	lst.lib/lstFirst.o lst.lib/lstForEach.o lst.lib/lstForEachFrom.o \
+	lst.lib/lstInit.o lst.lib/lstInsert.o lst.lib/lstIsAtEnd.o \
+	lst.lib/lstIsEmpty.o lst.lib/lstLast.o lst.lib/lstMember.o \
+	lst.lib/lstNext.o lst.lib/lstOpen.o lst.lib/lstRemove.o \
+	lst.lib/lstReplace.o lst.lib/lstSucc.o
+
+bmake: ${OBJ} ${LIBOBJ}
+#	@echo 'make of make and make.0 started.'
+	${CC} ${CFLAGS} ${OBJ} ${LIBOBJ} -o bmake ${LIBS}
+	@ls -l $@
+#	nroff -h -man make.1 > make.0
+#	@echo 'make of make and make.0 completed.'
+
+clean:
+	rm -f ${OBJ} ${LIBOBJ} ${PORTOBJ} bmake
diff -ruN pmake-1.45.orig/main.c pmake-1.45/main.c
--- pmake-1.45.orig/main.c	2001-09-08 08:30:01.000000000 -0600
+++ pmake-1.45/main.c	2002-12-27 09:07:50.000000000 -0700
@@ -109,6 +109,15 @@
 #include "job.h"
 #include "pathnames.h"
 
+#ifdef TARGET_MACHINE
+#undef MACHINE
+#define MACHINE TARGET_MACHINE
+#endif
+#ifdef TARGET_MACHINE_ARCH
+#undef MACHINE_ARCH
+#define MACHINE_ARCH TARGET_MACHINE_ARCH
+#endif
+
 #ifndef	DEFMAXLOCAL
 #define	DEFMAXLOCAL DEFMAXJOBS
 #endif	/* DEFMAXLOCAL */
diff -ruN pmake-1.45.orig/make.1 pmake-1.45/make.1
--- pmake-1.45.orig/make.1	2002-03-12 15:12:36.000000000 -0700
+++ pmake-1.45/make.1	2002-12-27 09:36:48.000000000 -0700
@@ -167,7 +167,7 @@
 .It Fl m Ar directory
 Specify a directory in which to search for sys.mk and makefiles included
 via the <...> style.  Multiple directories can be added to form a search path.
-This path will override the default system include path: /usr/share/mk.
+This path will override the default system include path: @FINKPREFIX@/share/mk.
 Furthermore the system include path will be appended to the search path used
 for "..."-style inclusions (see the
 .Fl I
@@ -1021,7 +1021,7 @@
 and
 .Ev PWD .
 .Sh FILES
-.Bl -tag -width /usr/share/mk -compact
+.Bl -tag -width @FINKPREFIX@/share/mk -compact
 .It .depend
 list of dependencies
 .It Makefile
@@ -1030,7 +1030,7 @@
 list of dependencies
 .It sys.mk
 system makefile
-.It /usr/share/mk
+.It @FINKPREFIX@/share/mk
 system makefile directory
 .El
 .Sh SEE ALSO
diff -ruN pmake-1.45.orig/mk/Makefile pmake-1.45/mk/Makefile
--- pmake-1.45.orig/mk/Makefile	1999-02-13 05:31:19.000000000 -0700
+++ pmake-1.45/mk/Makefile	2002-12-27 09:46:57.000000000 -0700
@@ -7,7 +7,7 @@
 FILES=	bsd.README bsd.doc.mk bsd.dep.mk bsd.files.mk bsd.inc.mk bsd.info.mk \
 	bsd.kinc.mk bsd.kmod.mk bsd.lib.mk bsd.links.mk bsd.man.mk bsd.nls.mk \
 	bsd.obj.mk bsd.own.mk bsd.prog.mk bsd.subdir.mk bsd.sys.mk sys.mk
-FILESDIR=/usr/share/mk
+FILESDIR=@FINKPREFIX@/share/mk
 .endif
 MKOBJ=	no
 
diff -ruN pmake-1.45.orig/mk/bsd.own.mk pmake-1.45/mk/bsd.own.mk
--- pmake-1.45.orig/mk/bsd.own.mk	2002-06-30 16:54:36.000000000 -0600
+++ pmake-1.45/mk/bsd.own.mk	2002-12-28 00:20:29.000000000 -0700
@@ -5,23 +5,23 @@
 
 .if defined(MAKECONF) && exists(${MAKECONF})
 .include "${MAKECONF}"
-.elif exists(/etc/mk.conf)
-.include "/etc/mk.conf"
+.elif exists(@FINKPREFIX@/etc/mk.conf)
+.include "@FINKPREFIX@/etc/mk.conf"
 .endif
 
 # Defining `SKEY' causes support for S/key authentication to be compiled in.
-SKEY=		yes
+#SKEY=		yes
 # Defining `KERBEROS' causes support for Kerberos authentication to be
 # compiled in.
 #KERBEROS=	yes
 # Defining 'KERBEROS5' causes support for Kerberos5 authentication to be
 # compiled in.
-#KERBEROS5=	yes
+KERBEROS5=	yes
 
 # where the system object and source trees are kept; can be configurable
 # by the user in case they want them in ~/foosrc and ~/fooobj, for example
-BSDSRCDIR?=	/usr/src
-BSDOBJDIR?=	/usr/obj
+BSDSRCDIR?=	@FINKPREFIX@/src
+BSDOBJDIR?=	@FINKPREFIX@/share/obj
 
 BINGRP?=	root
 BINOWN?=	root
@@ -31,34 +31,34 @@
 # Define MANZ to have the man pages compressed (gzip)
 MANZ=		1
 
-MANDIR?=	/usr/share/man
+MANDIR?=	@FINKPREFIX@/share/man
 MANGRP?=	root
 MANOWN?=	root
 MANMODE?=	${NONBINMODE}
 MANINSTALL?=	maninstall catinstall
 
-INFODIR?=	/usr/share/info
+INFODIR?=	@FINKPREFIX@/share/info
 INFOGRP?=	root
 INFOOWN?=	root
 INFOMODE?=	${NONBINMODE}
 
-LIBDIR?=	/usr/lib
-LINTLIBDIR?=	/usr/libdata/lint
+LIBDIR?=	@FINKPREFIX@/lib
+LINTLIBDIR?=	@FINKPREFIX@/lib/lint
 LIBGRP?=	${BINGRP}
 LIBOWN?=	${BINOWN}
 LIBMODE?=	${NONBINMODE}
 
-DOCDIR?=        /usr/share/doc
+DOCDIR?=        @FINKPREFIX@/share/doc
 DOCGRP?=	root
 DOCOWN?=	root
 DOCMODE?=       ${NONBINMODE}
 
-NLSDIR?=	/usr/share/nls
+NLSDIR?=	@FINKPREFIX@/share/locale
 NLSGRP?=	root
 NLSOWN?=	root
 NLSMODE?=	${NONBINMODE}
 
-KMODDIR?=	/usr/lkm
+KMODDIR?=	@FINKPREFIX@/share/lkm
 KMODGRP?=	root
 KMODOWN?=	root
 KMODMODE?=	${NONBINMODE}
@@ -88,6 +88,7 @@
 .if ${MACHINE} == "alpha" || \
     ${MACHINE} == "i386" || \
     ${MACHINE} == "pc532" || \
+    ${MACHINE} == "Darwin" || \
     ${MACHINE} == "vax"
 PMAP_NEW?=	yes
 .endif
@@ -101,7 +102,7 @@
 
 # The PowerPC port is incomplete.
 .if ${MACHINE_ARCH} == "powerpc"
-NOPROFILE=
+#NOPROFILE=
 .endif
 
 # Data-driven table using make variables to control how 
diff -ruN pmake-1.45.orig/mk/bsd.prog.mk pmake-1.45/mk/bsd.prog.mk
--- pmake-1.45.orig/mk/bsd.prog.mk	2000-03-05 13:17:04.000000000 -0700
+++ pmake-1.45/mk/bsd.prog.mk	2002-12-27 23:51:57.000000000 -0700
@@ -29,37 +29,37 @@
 LIBCRT0?=	
 
 LIBBZ2?=	${DESTDIR}/usr/lib/libbz2.a
-LIBC?=		${DESTDIR}/usr/lib/libc.a
-LIBC_PIC?=	${DESTDIR}/usr/lib/libc_pic.a
-LIBCOMPAT?=	${DESTDIR}/usr/lib/libcompat.a
-LIBCRYPT?=	${DESTDIR}/usr/lib/libcrypt.a
-LIBCURSES?=	${DESTDIR}/usr/lib/libcurses.a
-LIBDBM?=	${DESTDIR}/usr/lib/libdbm.a
-LIBDES?=	${DESTDIR}/usr/lib/libdes.a
-LIBEDIT?=	${DESTDIR}/usr/lib/libedit.a
+LIBC?=		${DESTDIR}/usr/lib/libc.dylib
+#LIBC_PIC?=	${DESTDIR}/usr/lib/libc_pic.a
+#LIBCOMPAT?=	${DESTDIR}/usr/lib/libcompat.a
+LIBCRYPT?=	${DESTDIR}/usr/lib/libSystem.dylib
+LIBCURSES?=	${DESTDIR}/usr/lib/libcurses.dylib
+LIBDBM?=	${DESTDIR}/usr/lib/libSystem.dylib
+LIBDES?=	${DESTDIR}/usr/lib/libdes425.dylib
+LIBEDIT?=	${DESTDIR}/usr/lib/libedit.dylib
 LIBGCC?=	${DESTDIR}/usr/lib/libgcc.a
-LIBGNUMALLOC?=	${DESTDIR}/usr/lib/libgnumalloc.a
-LIBKDB?=	${DESTDIR}/usr/lib/libkdb.a
-LIBKRB?=	${DESTDIR}/usr/lib/libkrb.a
-LIBKVM?=	${DESTDIR}/usr/lib/libkvm.a
+LIBGNUMALLOC?=	${DESTDIR}/usr/lib/libSystem.dylib
+#LIBKDB?=	${DESTDIR}/usr/lib/libkdb.a
+#LIBKRB?=	${DESTDIR}/usr/lib/libkrb.a
+LIBKVM?=	${DESTDIR}/usr/lib/libkvm.dylib
 LIBL?=		${DESTDIR}/usr/lib/libl.a
-LIBM?=		${DESTDIR}/usr/lib/libm.a
-LIBMP?=		${DESTDIR}/usr/lib/libmp.a
-LIBNTP?=	${DESTDIR}/usr/lib/libntp.a
-LIBOBJC?=	${DESTDIR}/usr/lib/libobjc.a
-LIBPC?=		${DESTDIR}/usr/lib/libpc.a
+LIBM?=		${DESTDIR}/usr/lib/libm.dylib
+#LIBMP?=		${DESTDIR}/usr/lib/libmp.a
+#LIBNTP?=	${DESTDIR}/usr/lib/libntp.a
+LIBOBJC?=	${DESTDIR}/usr/lib/libobjc.dylib
+#LIBPC?=		${DESTDIR}/usr/lib/libpc.a
 LIBPCAP?=	${DESTDIR}/usr/lib/libpcap.a
-LIBPLOT?=	${DESTDIR}/usr/lib/libplot.a
-LIBPOSIX?=	${DESTDIR}/usr/lib/libposix.a
-LIBRESOLV?=	${DESTDIR}/usr/lib/libresolv.a
-LIBRPCSVC?=	${DESTDIR}/usr/lib/librpcsvc.a
-LIBSKEY?=	${DESTDIR}/usr/lib/libskey.a
-LIBTERMCAP?=	${DESTDIR}/usr/lib/libtermcap.a
-LIBTELNET?=	${DESTDIR}/usr/lib/libtelnet.a
-LIBUTIL?=	${DESTDIR}/usr/lib/libutil.a
+#LIBPLOT?=	${DESTDIR}/usr/lib/libplot.a
+#LIBPOSIX?=	${DESTDIR}/usr/lib/libposix.a
+#LIBRESOLV?=	${DESTDIR}/usr/lib/libresolv.a
+#LIBRPCSVC?=	${DESTDIR}/usr/lib/librpcsvc.a
+#LIBSKEY?=	${DESTDIR}/usr/lib/libskey.a
+LIBTERMCAP?=	${DESTDIR}/usr/lib/libtermcap.dylib
+#LIBTELNET?=	${DESTDIR}/usr/lib/libtelnet.a
+#LIBUTIL?=	${DESTDIR}/usr/lib/libutil.a
 LIBWRAP?=	${DESTDIR}/usr/lib/libwrap.a
 LIBY?=		${DESTDIR}/usr/lib/liby.a
-LIBZ?=		${DESTDIR}/usr/lib/libz.a
+LIBZ?=		${DESTDIR}/usr/lib/libz.dylib
 
 .if defined(SHAREDSTRINGS)
 CLEANFILES+=strings
diff -ruN pmake-1.45.orig/mk/sys.mk pmake-1.45/mk/sys.mk
--- pmake-1.45.orig/mk/sys.mk	2000-03-05 13:17:04.000000000 -0700
+++ pmake-1.45/mk/sys.mk	2002-12-28 00:29:43.000000000 -0700
@@ -8,14 +8,14 @@
 .LIBS:		.a
 
 AR?=		ar
-ARFLAGS?=	rl
+ARFLAGS?=	cur
 RANLIB?=	ranlib
 
 AS?=		as
 AFLAGS?=
 COMPILE.s?=	${CC} ${AFLAGS} -c
 LINK.s?=	${CC} ${AFLAGS} ${LDFLAGS}
-COMPILE.S?=	${CC} ${AFLAGS} ${CPPFLAGS} -c -traditional-cpp
+COMPILE.S?=	${CC} ${AFLAGS} ${CPPFLAGS} -c -no-cpp-precomp
 LINK.S?=	${CC} ${AFLAGS} ${CPPFLAGS} ${LDFLAGS}
 
 CC?=		cc
diff -ruN pmake-1.45.orig/pathnames.h pmake-1.45/pathnames.h
--- pmake-1.45.orig/pathnames.h	1998-12-20 05:17:52.000000000 -0700
+++ pmake-1.45/pathnames.h	2002-12-28 00:31:05.000000000 -0700
@@ -36,9 +36,9 @@
  */
 
 #define	_PATH_OBJDIR		"obj"
-#define	_PATH_OBJDIRPREFIX	"/usr/obj"
+#define	_PATH_OBJDIRPREFIX	"@FINKPREFIX@/share/obj"
 #define	_PATH_DEFSHELLDIR	"/bin"
-#define	_PATH_DEFSYSMK		"sys.mk"
+#define	_PATH_DEFSYSMK		"bsd.sys.mk"
 #ifndef _PATH_DEFSYSPATH
-#define	_PATH_DEFSYSPATH	"/usr/share/mk"
+#define	_PATH_DEFSYSPATH	"@FINKPREFIX@/share/mk"
 #endif
