diff -uNr fontconfig-2.2.3/fontconfig/fontconfig.h fontconfig-2.2.3-new/fontconfig/fontconfig.h
--- fontconfig-2.2.3/fontconfig/fontconfig.h	Wed Jun 30 14:45:31 2004
+++ fontconfig-2.2.3-new/fontconfig/fontconfig.h	Tue Jan 18 11:30:25 2005
@@ -70,6 +70,7 @@
 #define FC_FOUNDRY	    "foundry"		/* String */
 #define FC_ANTIALIAS	    "antialias"		/* Bool (depends) */
 #define FC_HINTING	    "hinting"		/* Bool (true) */
+#define FC_HINT_STYLE	    "hintstyle"		/* Int */
 #define FC_VERTICAL_LAYOUT  "verticallayout"	/* Bool (false) */
 #define FC_AUTOHINT	    "autohint"		/* Bool (false) */
 #define FC_GLOBAL_ADVANCE   "globaladvance"	/* Bool (true) */
@@ -138,6 +139,12 @@
 #define FC_RGBA_VBGR	    4
 #define FC_RGBA_NONE	    5
 
+/* hinting style */
+#define FC_HINT_NONE        0
+#define FC_HINT_SLIGHT      1
+#define FC_HINT_MEDIUM      2
+#define FC_HINT_FULL        3
+ 
 typedef enum _FcType {
     FcTypeVoid, 
     FcTypeInteger, 
diff -uNr fontconfig-2.2.3/src/fcdefault.c fontconfig-2.2.3-new/src/fcdefault.c
--- fontconfig-2.2.3/src/fcdefault.c	Wed Mar 12 17:16:25 2003
+++ fontconfig-2.2.3-new/src/fcdefault.c	Tue Jan 18 11:30:25 2005
@@ -144,4 +144,9 @@
     {
 	FcPatternAddInteger (pattern, FC_FONTVERSION, 0x7fffffff);
     }
+
+    if (FcPatternGet (pattern, FC_HINT_STYLE, 0, &v) == FcResultNoMatch)
+    {
+	FcPatternAddInteger (pattern, FC_HINT_STYLE, FC_HINT_FULL);
+    }
 }
diff -uNr fontconfig-2.2.3/src/fcname.c fontconfig-2.2.3-new/src/fcname.c
--- fontconfig-2.2.3/src/fcname.c	Wed Mar 12 17:16:25 2003
+++ fontconfig-2.2.3-new/src/fcname.c	Tue Jan 18 11:30:25 2005
@@ -41,6 +41,7 @@
     { FC_FOUNDRY,	FcTypeString, },
 /*    { FC_CORE,		FcTypeBool, }, */
     { FC_ANTIALIAS,	FcTypeBool, },
+    { FC_HINT_STYLE,    FcTypeInteger, },
     { FC_HINTING,	FcTypeBool, },
     { FC_VERTICAL_LAYOUT,   FcTypeBool, },
     { FC_AUTOHINT,	FcTypeBool, },
@@ -175,6 +176,11 @@
     { (FcChar8 *) "vrgb",	    "rgba",	    FC_RGBA_VRGB },
     { (FcChar8 *) "vbgr",	    "rgba",	    FC_RGBA_VBGR },
     { (FcChar8 *) "none",	    "rgba",	    FC_RGBA_NONE },
+
+    { (FcChar8 *) "hintnone",	    "hintstyle",   FC_HINT_NONE },
+    { (FcChar8 *) "hintslight",	    "hintstyle",   FC_HINT_SLIGHT },
+    { (FcChar8 *) "hintmedium",	    "hintstyle",   FC_HINT_MEDIUM },
+    { (FcChar8 *) "hintfull",	    "hintstyle",   FC_HINT_FULL },
 };
 
 #define NUM_FC_CONSTANTS   (sizeof _FcBaseConstants/sizeof _FcBaseConstants[0])
