diff -Nru mlterm-2.9.2.orig/tool/mlconfig/po/Makefile.in.in mlterm-2.9.2/tool/mlconfig/po/Makefile.in.in
--- mlterm-2.9.2.orig/tool/mlconfig/po/Makefile.in.in	Sun Jan 12 19:08:51 2003
+++ mlterm-2.9.2/tool/mlconfig/po/Makefile.in.in	Sun Apr  3 02:29:55 2005
@@ -26,8 +26,7 @@
 
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
-MKINSTALLDIRS = @MKINSTALLDIRS@
-mkinstalldirs = $(SHELL) `case "$(MKINSTALLDIRS)" in /*) echo "$(MKINSTALLDIRS)" ;; *) echo "$(top_builddir)/$(MKINSTALLDIRS)" ;; esac`
+mkinstalldirs = @MKINSTALLDIRS@
 
 GMSGFMT = @GMSGFMT@
 MSGFMT = @MSGFMT@
