diff -ruN qemu-0.6.1.orig/Makefile.target qemu-0.6.1/Makefile.target
--- qemu-0.6.1.orig/Makefile.target	Sun Nov 14 21:51:33 2004
+++ qemu-0.6.1/Makefile.target	Tue Nov 16 00:15:15 2004
@@ -7,8 +7,8 @@
 VPATH+=:$(SRC_PATH)/linux-user
 DEFINES+=-I$(SRC_PATH)/linux-user -I$(SRC_PATH)/linux-user/$(TARGET_ARCH)
 endif
-CFLAGS=-Wall -O2 -g -fno-strict-aliasing
-LDFLAGS=-g
+CFLAGS=-Wall -O2 -g -fno-strict-aliasing -I@PREFIX@/include 
+LDFLAGS=-g -L@PREFIX@/lib 
 LIBS=
 HELPER_CFLAGS=$(CFLAGS)
 DYNGEN=../dyngen$(EXESUF)
@@ -180,7 +180,7 @@
 #########################################################
 
 DEFINES+=-D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
-LIBS+=-lm
+LIBS+=-lm -lmx
 ifndef CONFIG_USER_ONLY
 LIBS+=-lz
 endif
