diff -Naur rxvt-unicode-7.7.orig/doc/Makefile.in rxvt-unicode-7.7/doc/Makefile.in
--- rxvt-unicode-7.7.orig/doc/Makefile.in	2006-02-01 05:55:51.000000000 +0900
+++ rxvt-unicode-7.7/doc/Makefile.in	2006-03-26 11:20:44.000000000 +0900
@@ -107,7 +107,7 @@
 @IF_PERL@	$(INSTALL) -d $(DESTDIR)$(man3dir)
 @IF_PERL@	$(SED) $(SEDREPLACE) < rxvtperl.3.man.in >rxvtperl.3.man
 @IF_PERL@	$(INSTALL_DATA) rxvtperl.3.man $(DESTDIR)$(man3dir)/$(RXVTNAME)perl.$(man3ext)
-	if test -x "@TIC@"; then if test -n "@TIC@"; then @TIC@ etc/rxvt-unicode.terminfo; fi; fi
+	if test -x "@TIC@"; then if test -n "@TIC@"; then @TIC@ -o $(DESTDIR)$(prefix)/share/terminfo etc/rxvt-unicode.terminfo; fi; fi
 
 uninstall:
 	-(cd $(mandir); rm -f $(RXVTNAME).$(manext) )
diff -Naur rxvt-unicode-7.7.orig/src/rxvtc.C rxvt-unicode-7.7/src/rxvtc.C
--- rxvt-unicode-7.7.orig/src/rxvtc.C	2006-02-21 07:41:16.000000000 +0900
+++ rxvt-unicode-7.7/src/rxvtc.C	2006-03-26 11:17:24.000000000 +0900
@@ -29,6 +29,7 @@
 #include <unistd.h>
 #include <sys/socket.h>
 #include <sys/un.h>
+#include <limits.h>
 
 #include "rxvtdaemon.h"
 #include "libptytty.h"
