diff -Nurd -x'*~' desktop-file-utils-0.15.orig/src/Makefile.in desktop-file-utils-0.15/src/Makefile.in
--- desktop-file-utils-0.15.orig/src/Makefile.in	2008-02-11 14:17:06.000000000 -0500
+++ desktop-file-utils-0.15/src/Makefile.in	2010-01-14 10:01:47.000000000 -0500
@@ -176,9 +176,9 @@
 	update-desktop-database.c
 
 
-desktop_file_validate_LDADD = $(DESKTOP_FILE_UTILS_LIBS)
-desktop_file_install_LDADD = $(DESKTOP_FILE_UTILS_LIBS)
-update_desktop_database_LDADD = $(DESKTOP_FILE_UTILS_LIBS)
+desktop_file_validate_LDADD = $(DESKTOP_FILE_UTILS_LIBS) -lintl
+desktop_file_install_LDADD = $(DESKTOP_FILE_UTILS_LIBS) -lintl
+update_desktop_database_LDADD = $(DESKTOP_FILE_UTILS_LIBS) -lintl
 subdir = src
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
