diff -Nurd -x'*~' eog-2.26.2.orig/bindings/python/Makefile.am eog-2.26.2/bindings/python/Makefile.am
--- eog-2.26.2.orig/bindings/python/Makefile.am	2009-05-18 15:37:55.000000000 -0400
+++ eog-2.26.2/bindings/python/Makefile.am	2009-06-10 12:06:02.000000000 -0400
@@ -13,7 +13,7 @@
 	$(PYTHON_EXTRA_LIBS)	\
 	$(PYGTK_LIBS)
 
-eog_la_CFLAGS = 					\
+eog_la_CPPFLAGS = 					\
 	-I$(top_srcdir)					\
 	-I$(top_builddir)				\
 	-I$(top_srcdir)/src				\
diff -Nurd -x'*~' eog-2.26.2.orig/configure.ac eog-2.26.2/configure.ac
--- eog-2.26.2.orig/configure.ac	2009-05-18 15:37:55.000000000 -0400
+++ eog-2.26.2/configure.ac	2009-06-10 12:06:02.000000000 -0400
@@ -32,8 +32,7 @@
 AC_PROG_LIBTOOL
 
 AC_ISC_POSIX
-AC_PROG_CC
-AM_PROG_CC_STDC
+AM_PROG_CC_C_O
 AC_HEADER_STDC
 AC_PATH_PROG([GLIB_GENMARSHAL], [glib-genmarshal])
 AC_PATH_PROG([GLIB_MKENUMS],[glib-mkenums])
diff -Nurd -x'*~' eog-2.26.2.orig/cut-n-paste/toolbar-editor/Makefile.am eog-2.26.2/cut-n-paste/toolbar-editor/Makefile.am
--- eog-2.26.2.orig/cut-n-paste/toolbar-editor/Makefile.am	2009-05-18 15:37:55.000000000 -0400
+++ eog-2.26.2/cut-n-paste/toolbar-editor/Makefile.am	2009-06-10 12:08:41.000000000 -0400
@@ -21,15 +21,13 @@
 
 libtoolbareditor_la_CPPFLAGS = \
 	-I$(top_builddir)/lib/egg		\
-	$(AM_CPPFLAGS)
-
-libtoolbareditor_la_LIBADD = $(LIBXML2_LIBS)
-
-libtoolbareditor_la_CFLAGS = \
 	$(EOG_CFLAGS)				\
 	$(WARN_CFLAGS)				\
 	$(LIBXML2_CFLAGS)			\
 	-DCURSOR_DIR=\"$(pkgdatadir)\"
+	$(AM_CPPFLAGS)
+
+libtoolbareditor_la_LIBADD = $(LIBXML2_LIBS)
 
 BUILT_SOURCES = \
 	eggmarshalers.c \
diff -Nurd -x'*~' eog-2.26.2.orig/cut-n-paste/totem-screensaver/Makefile.am eog-2.26.2/cut-n-paste/totem-screensaver/Makefile.am
--- eog-2.26.2.orig/cut-n-paste/totem-screensaver/Makefile.am	2009-05-18 15:37:55.000000000 -0400
+++ eog-2.26.2/cut-n-paste/totem-screensaver/Makefile.am	2009-06-10 12:06:02.000000000 -0400
@@ -4,7 +4,7 @@
 	totem-scrsaver.h \
 	totem-scrsaver.c
 
-libtotemscrsaver_la_CFLAGS = \
+libtotemscrsaver_la_CPPFLAGS = \
 	-I$(top_srcdir)/src	\
 	$(EOG_CFLAGS)		\
 	$(WARN_CFLAGS)
diff -Nurd -x'*~' eog-2.26.2.orig/src/Makefile.am eog-2.26.2/src/Makefile.am
--- eog-2.26.2.orig/src/Makefile.am	2009-05-18 15:37:55.000000000 -0400
+++ eog-2.26.2/src/Makefile.am	2009-06-10 12:09:53.000000000 -0400
@@ -139,7 +139,7 @@
 	eog-exif-details.c
 endif
 
-libeog_la_CFLAGS =						\
+libeog_la_CPPFLAGS =						\
 	-I$(top_srcdir)/jpegutils				\
 	-I$(top_srcdir)/cut-n-paste/toolbar-editor		\
 	-I$(top_srcdir)/cut-n-paste/totem-screensaver		\
@@ -156,7 +156,7 @@
 	$(EOG_LIBS)
 
 if ENABLE_PYTHON
-libeog_la_CFLAGS += 			\
+libeog_la_CPPFLAGS += 			\
 	$(NO_STRICT_ALIASING_CFLAGS)	\
 	$(PYGTK_CFLAGS)			\
 	$(PYTHON_CFLAGS)		\
@@ -170,7 +170,7 @@
 
 eog_SOURCES = main.c
 
-eog_CFLAGS = 							\
+eog_CPPFLAGS = 							\
 	-I$(top_srcdir)/cut-n-paste/toolbar-editor		\
 	-I$(top_srcdir)/cut-n-paste/totem-screensaver		\
 	$(EOG_CFLAGS)						\
