diff -Nurd -x'*~' evince-2.22.2.orig/backend/pdf/ev-poppler.cc evince-2.22.2/backend/pdf/ev-poppler.cc
--- evince-2.22.2.orig/backend/pdf/ev-poppler.cc	2008-05-28 17:13:58.000000000 -0400
+++ evince-2.22.2/backend/pdf/ev-poppler.cc	2013-03-08 17:19:41.000000000 -0500
@@ -960,6 +960,8 @@
 	const char   *unimplemented_action = NULL;
 
 	switch (action->type) {
+	        case POPPLER_ACTION_NONE:
+			break;
 	        case POPPLER_ACTION_GOTO_DEST: {
 			EvLinkDest *dest;
 			
@@ -989,6 +991,15 @@
 	        case POPPLER_ACTION_MOVIE:
 			unimplemented_action = "POPPLER_ACTION_MOVIE";
 			break;
+	        case POPPLER_ACTION_RENDITION:
+			unimplemented_action = "POPPLER_ACTION_RENDITION";
+			break;
+	        case POPPLER_ACTION_OCG_STATE:
+			unimplemented_action = "POPPLER_ACTION_OGG_STATE";
+			break;
+	        case POPPLER_ACTION_JAVASCRIPT:
+			unimplemented_action = "POPPLER_ACTION_JAVASCRIPT";
+			break;
 	        case POPPLER_ACTION_UNKNOWN:
 			unimplemented_action = "POPPLER_ACTION_UNKNOWN";
 	}
@@ -1979,7 +1990,7 @@
 	r.x2 = points->x2;
 	r.y2 = height - points->y1;
 
-	retval = poppler_page_get_text (poppler_page,
+	retval = poppler_page_get_selected_text (poppler_page,
 					(PopplerSelectionStyle)style,
 					&r);
 
