diff -Nurd -x'*~' evince-2.20.2.orig/configure.ac evince-2.20.2/configure.ac
--- evince-2.20.2.orig/configure.ac	2007-11-26 16:52:17.000000000 -0500
+++ evince-2.20.2/configure.ac	2008-01-26 15:22:52.000000000 -0500
@@ -84,48 +84,41 @@
 	    PRINT=$withval,
             PRINT="auto")
 
-PKG_CHECK_MODULES(GNOME_PRINT, 
-		  libgnomeprintui-2.2 >= $LIBGNOMEPRINTUI_REQUIRED,
-		  enable_gnome_print=yes, enable_gnome_print=no)
-PKG_CHECK_MODULES(GTK_PRINT,
-		  gtk+-unix-print-2.0 >= $GTK_PRINT_REQUIRED,
-		  enable_gtk_print=yes, enable_gtk_print=no)
-enable_print=no
-if test x$PRINT = xgnome -a x$enable_gnome_print = xyes; then
-   enable_print=gnome
-else
-   if test x$PRINT = xgtk -a x$enable_gtk_print = xyes; then
-      enable_print=gtk
-   else
-      if test x$PRINT = xauto; then
-	 if test x$enable_gtk_print = xyes; then
-	    enable_print=gtk
-	 else
-	    if test x$enable_gnome_print = xyes; then
-	       enable_print=gnome
-	    fi
-	 fi
-      fi
-   fi
+if test x$PRINT = xauto -o x$PRINT = xgtk; then
+	PKG_CHECK_MODULES(GTK_PRINT,
+			  gtk+-unix-print-2.0 >= $GTK_PRINT_REQUIRED,
+			  [PRINT=gtk
+			   AC_DEFINE([WITH_GTK_PRINT],[1],[Enable GTK Print Support.])
+			   AC_DEFINE([WITH_PRINT],[1],[Enable Print Support.])
+			   EVINCE_PRINT_CFLAGS=$GTK_PRINT_CFLAGS
+			   EVINCE_PRINT_LIBS=$GTK_PRINT_LIBS
+			  ],
+			  [if test x$PRINT = xgtk; then
+        		  	AC_MSG_ERROR([Support for requested $PRINT printing not available])
+			  fi])
 fi
-
-if test x$enable_print = xgnome; then
-   AC_SUBST(GNOME_PRINT_CFLAGS)
-   AC_SUBST(GNOME_PRINT_LIBS)
-   AC_DEFINE([WITH_GNOME_PRINT],[1],[Enable GNOME Print Support.])
-   AC_DEFINE([WITH_PRINT],[1],[Enable Print Support.])
+if test x$PRINT = xauto -o x$PRINT = xgnome; then
+	PKG_CHECK_MODULES(GNOME_PRINT,
+			  libgnomeprintui-2.2 >= $LIBGNOMEPRINTUI_REQUIRED,
+			  [PRINT=gnome
+			   AC_DEFINE([WITH_GNOME_PRINT],[1],[Enable GNOME Print Support.])
+			   AC_DEFINE([WITH_PRINT],[1],[Enable Print Support.])
+			   EVINCE_PRINT_CFLAGS=$GNOME_PRINT_CFLAGS
+			   EVINCE_PRINT_LIBS=$GNOME_PRINT_LIBS
+			  ],
+			  [if test x$PRINT = xgnome; then
+        		  	AC_MSG_ERROR([Support for requested $PRINT printing not available])
+			  fi])
 fi
-
-if test x$enable_print = xgtk; then   
-   AC_SUBST(GTK_PRINT_CFLAGS)
-   AC_SUBST(GTK_PRINT_LIBS)
-   AC_DEFINE([WITH_GTK_PRINT],[1],[Enable GTK Print Support.])
-   AC_DEFINE([WITH_PRINT],[1],[Enable Print Support.])
+if test x$PRINT = xauto; then
+	PRINT=no
 fi
-   
-AM_CONDITIONAL(WITH_GNOME_PRINT, test x$enable_print = xgnome)
-AM_CONDITIONAL(WITH_GTK_PRINT, test x$enable_print = xgtk)
 
+AC_SUBST(EVINCE_PRINT_CFLAGS)
+AC_SUBST(EVINCE_PRINT_LIBS)
+   
+AM_CONDITIONAL(WITH_GNOME_PRINT, test x$PRINT = xgnome)
+AM_CONDITIONAL(WITH_GTK_PRINT, test x$PRINT = xgtk)
 
 dnl ========= Check for DBUS
 PKG_CHECK_MODULES([DBUS], [dbus-glib-1 >= $DBUS_GLIB_REQUIRED],
@@ -499,7 +492,7 @@
 echo "
 Configure summary:
 	GNOME Support......:  $WITH_GNOME
-	Print Support......:  $enable_print
+	Print Support......:  $PRINT
 	DBUS Support.......:  $enable_dbus
 	Nautilus Plugin....:  $HAVE_NAUTILUS
 	Thumbnailer........:  $ENABLE_THUMBNAILER
diff -Nurd -x'*~' evince-2.20.2.orig/help/reference/Makefile.am evince-2.20.2/help/reference/Makefile.am
--- evince-2.20.2.orig/help/reference/Makefile.am	2007-11-26 15:55:18.000000000 -0500
+++ evince-2.20.2/help/reference/Makefile.am	2008-01-26 15:22:51.000000000 -0500
@@ -87,8 +87,7 @@
 	$(SHELL_CFLAGS)					\
 	$(WARN_CFLAGS)					\
 	$(DISABLE_DEPRECATED)				\
-	$(GNOME_PRINT_CFLAGS)				\
-	$(GTK_PRINT_CFLAGS)
+	$(EVINCE_PRINT_CFLAGS)
 
 GTKDOC_LIBS= \
 	$(top_builddir)/cut-n-paste/zoom-control/libephyzoom.la		\
@@ -97,7 +96,7 @@
 	$(top_builddir)/properties/libevproperties.la			\
 	$(top_builddir)/libdocument/libevbackend.la			\
 	$(SHELL_LIBS)							\
-	$(GNOME_PRINT_LIBS)						\
+	$(EVINCE_PRINT_LIBS)						\
 	$(filter-out ../../shell/main.o,$(wildcard ../../shell/*.o))
 
 if HAVE_GTK_RECENT
diff -Nurd -x'*~' evince-2.20.2.orig/libdocument/Makefile.am evince-2.20.2/libdocument/Makefile.am
--- evince-2.20.2.orig/libdocument/Makefile.am	2007-11-26 15:55:20.000000000 -0500
+++ evince-2.20.2/libdocument/Makefile.am	2008-10-27 01:41:45.000000000 -0400
@@ -72,7 +72,7 @@
 
 if ENABLE_PS
 libevbackend_la_LIBADD += 			\
-	$(top_builddir)/backend/ps/libpsdocument.la
+	$(top_builddir)/backend/ps/libpsdocument.la -lX11
 endif
 
 if ENABLE_PIXBUF
diff -Nurd -x'*~' evince-2.20.2.orig/shell/Makefile.am evince-2.20.2/shell/Makefile.am
--- evince-2.20.2.orig/shell/Makefile.am	2007-11-26 15:55:23.000000000 -0500
+++ evince-2.20.2/shell/Makefile.am	2008-01-26 15:22:57.000000000 -0500
@@ -11,8 +11,7 @@
 	$(SHELL_CFLAGS)					\
 	$(WARN_CFLAGS)					\
 	$(DISABLE_DEPRECATED)				\
-	$(GNOME_PRINT_CFLAGS)				\
-	$(GTK_PRINT_CFLAGS)
+	$(EVINCE_PRINT_CFLAGS)
 
 if HAVE_GTK_RECENT
 INCLUDES +=
@@ -100,7 +99,7 @@
 	$(top_builddir)/properties/libevproperties.la			\
 	$(top_builddir)/libdocument/libevbackend.la			\
 	$(SHELL_LIBS)							\
-	$(GNOME_PRINT_LIBS)
+	$(EVINCE_PRINT_LIBS)
 
 if HAVE_GTK_RECENT
 evince_LDADD +=
