diff -Naur gcalctool-4.3.16.old/gcalctool/calctool.c gcalctool-4.3.16.new/gcalctool/calctool.c
--- gcalctool-4.3.16.old/gcalctool/calctool.c	Tue Oct 14 00:35:37 2003
+++ gcalctool-4.3.16.new/gcalctool/calctool.c	Mon Dec  8 16:00:45 2003
@@ -21,7 +21,7 @@
 
 #include <stdio.h>
 #include <string.h>
-#include <malloc.h>
+#include <sys/malloc.h>
 #include <sys/types.h>
 #include "calctool.h"
 #include <gdk/gdkkeysyms.h>
diff -Naur gcalctool-4.3.16.old/po/Makefile.in.in gcalctool-4.3.16.new/po/Makefile.in.in
--- gcalctool-4.3.16.old/po/Makefile.in.in	Tue Oct 14 03:05:32 2003
+++ gcalctool-4.3.16.new/po/Makefile.in.in	Mon Dec  8 16:00:58 2003
@@ -28,7 +28,7 @@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
 libdir = @libdir@
-localedir = $(libdir)/locale
+localedir = $(datadir)/locale
 gnulocaledir = $(datadir)/locale
 gettextsrcdir = $(datadir)/glib-2.0/gettext/po
 subdir = po
