diff -Nurd -x'*~' glib-2.22.4.orig/configure glib-2.22.4/configure
--- glib-2.22.4.orig/configure	2010-01-06 19:48:41.000000000 -0500
+++ glib-2.22.4/configure	2010-01-13 08:56:35.000000000 -0500
@@ -4801,7 +4801,7 @@
 #define HAVE_CARBON 1
 _ACEOF
 
-  LDFLAGS="$LDFLAGS -framework Carbon"
+  LDFLAGS="$LDFLAGS -Wl,-framework,Carbon"
 fi
 
 
@@ -8323,13 +8323,13 @@
 
         if test -n "$PYTHON"; then
       # If the user set $PYTHON, use it and don't search something else.
-      { $as_echo "$as_me:$LINENO: checking whether $PYTHON version >= 2.4" >&5
-$as_echo_n "checking whether $PYTHON version >= 2.4... " >&6; }
+      { $as_echo "$as_me:$LINENO: checking whether $PYTHON version >= 2.3" >&5
+$as_echo_n "checking whether $PYTHON version >= 2.3... " >&6; }
       prog="import sys
 # split strings by '.' and convert to numeric.  Append some zeros
 # because we need at least 4 digits for the hex conversion.
 # map returns an iterator in Python 3.0 and a list in 2.x
-minver = list(map(int, '2.4'.split('.'))) + [0, 0, 0]
+minver = list(map(int, '2.3'.split('.'))) + [0, 0, 0]
 minverhex = 0
 # xrange is not present in Python 3.0 and range returns an iterator
 for i in list(range(0, 4)): minverhex = (minverhex << 8) + minver[i]
@@ -40056,6 +40056,9 @@
 			G_THREAD_CFLAGS="-D_THREAD_SAFE"
 		fi
 		;;
+	*-darwin*)
+		# Nothing needed.
+		;;
 	*-dg-dgux*)  # DG/UX
 		G_THREAD_CFLAGS="-D_REENTRANT -D_POSIX4A_DRAFT10_SOURCE"
 		;;
diff -Nurd -x'*~' glib-2.22.4.orig/gio/Makefile.in glib-2.22.4/gio/Makefile.in
--- glib-2.22.4.orig/gio/Makefile.in	2010-01-06 19:48:36.000000000 -0500
+++ glib-2.22.4/gio/Makefile.in	2010-01-13 09:57:06.000000000 -0500
@@ -677,6 +677,7 @@
 	$(SELINUX_LIBS) 				\
 	$(GLIB_LIBS) 					\
 	$(XATTR_LIBS) 					\
+	-lresolv					\
 	$(NULL)
 
 @PLATFORM_WIN32_TRUE@no_undefined = -no-undefined
diff -Nurd -x'*~' glib-2.22.4.orig/gio/gdesktopappinfo.c glib-2.22.4/gio/gdesktopappinfo.c
--- glib-2.22.4.orig/gio/gdesktopappinfo.c	2010-01-06 19:24:28.000000000 -0500
+++ glib-2.22.4/gio/gdesktopappinfo.c	2010-01-13 08:35:24.000000000 -0500
@@ -1396,7 +1396,7 @@
       g_file_set_contents (filename, contents, -1, NULL);
       g_free (contents);
 
-      run_update_command ("update-mime-database", "mime");
+      run_update_command ("@PREFIX@/bin/update-mime-database", "mime");
     }
   g_free (filename);
   
@@ -1527,7 +1527,7 @@
   info->filename = filename;
   info->desktop_id = desktop_id;
   
-  run_update_command ("update-desktop-database", "applications");
+  run_update_command ("@PREFIX@/bin/update-desktop-database", "applications");
   
   return TRUE;
 }
diff -Nurd -x'*~' glib-2.22.4.orig/gio/gdrive.h glib-2.22.4/gio/gdrive.h
--- glib-2.22.4.orig/gio/gdrive.h	2009-08-28 22:52:22.000000000 -0400
+++ glib-2.22.4/gio/gdrive.h	2010-01-13 09:13:20.000000000 -0500
@@ -164,7 +164,7 @@
 gboolean g_drive_is_media_check_automatic (GDrive               *drive);
 gboolean g_drive_can_poll_for_media       (GDrive               *drive);
 gboolean g_drive_can_eject                (GDrive               *drive);
-#ifndef G_DISABLE_DEPRECATED
+#if 1
 void     g_drive_eject                    (GDrive               *drive,
 					   GMountUnmountFlags    flags,
                                            GCancellable         *cancellable,
diff -Nurd -x'*~' glib-2.22.4.orig/gio/gfile.h glib-2.22.4/gio/gfile.h
--- glib-2.22.4.orig/gio/gfile.h	2010-01-06 19:19:12.000000000 -0500
+++ glib-2.22.4/gio/gfile.h	2010-01-13 09:13:13.000000000 -0500
@@ -867,7 +867,7 @@
 GFile *                 g_file_mount_mountable_finish     (GFile                      *file,
 							   GAsyncResult               *result,
 							   GError                    **error);
-#ifndef G_DISABLE_DEPRECATED
+#if 1
 void                    g_file_unmount_mountable          (GFile                      *file,
 							   GMountUnmountFlags          flags,
 							   GCancellable               *cancellable,
@@ -886,7 +886,7 @@
 gboolean                g_file_unmount_mountable_with_operation_finish (GFile         *file,
 							   GAsyncResult               *result,
 							   GError                    **error);
-#ifndef G_DISABLE_DEPRECATED
+#if 1
 void                    g_file_eject_mountable            (GFile                      *file,
 							   GMountUnmountFlags          flags,
 							   GCancellable               *cancellable,
diff -Nurd -x'*~' glib-2.22.4.orig/gio/gmount.h glib-2.22.4/gio/gmount.h
--- glib-2.22.4.orig/gio/gmount.h	2010-01-06 19:19:12.000000000 -0500
+++ glib-2.22.4/gio/gmount.h	2010-01-13 09:12:58.000000000 -0500
@@ -167,7 +167,7 @@
 gboolean    g_mount_can_unmount               (GMount              *mount);
 gboolean    g_mount_can_eject                 (GMount              *mount);
 
-#ifndef G_DISABLE_DEPRECATED
+#if 1
 void        g_mount_unmount                   (GMount              *mount,
                                                GMountUnmountFlags   flags,
                                                GCancellable        *cancellable,
diff -Nurd -x'*~' glib-2.22.4.orig/gio/gunixmounts.c glib-2.22.4/gio/gunixmounts.c
--- glib-2.22.4.orig/gio/gunixmounts.c	2009-03-31 19:04:20.000000000 -0400
+++ glib-2.22.4/gio/gunixmounts.c	2010-01-13 08:39:56.000000000 -0500
@@ -243,6 +243,7 @@
     "/proc",
     "/sbin",
     "/net",
+    "@PERFIX@",
     NULL
   };
 
diff -Nurd -x'*~' glib-2.22.4.orig/gio/gvolume.h glib-2.22.4/gio/gvolume.h
--- glib-2.22.4.orig/gio/gvolume.h	2009-08-28 22:52:23.000000000 -0400
+++ glib-2.22.4/gio/gvolume.h	2010-01-13 09:12:51.000000000 -0500
@@ -180,7 +180,7 @@
 gboolean g_volume_mount_finish          (GVolume              *volume,
 					 GAsyncResult         *result,
 					 GError              **error);
-#ifndef G_DISABLE_DEPRECATED
+#if 1
 void     g_volume_eject                 (GVolume              *volume,
 					 GMountUnmountFlags    flags,
 					 GCancellable         *cancellable,
diff -Nurd -x'*~' glib-2.22.4.orig/gio/gvolumemonitor.h glib-2.22.4/gio/gvolumemonitor.h
--- glib-2.22.4.orig/gio/gvolumemonitor.h	2009-08-28 22:52:23.000000000 -0400
+++ glib-2.22.4/gio/gvolumemonitor.h	2010-01-13 09:11:16.000000000 -0500
@@ -142,7 +142,7 @@
 GMount *        g_volume_monitor_get_mount_for_uuid   (GVolumeMonitor *volume_monitor,
                                                        const char     *uuid);
 
-#ifndef G_DISABLE_DEPRECATED
+#if 1
 GVolume *       g_volume_monitor_adopt_orphan_mount   (GMount         *mount);
 #endif
 
diff -Nurd -x'*~' glib-2.22.4.orig/gio/xdgmime/xdgmime.c glib-2.22.4/gio/xdgmime/xdgmime.c
--- glib-2.22.4.orig/gio/xdgmime/xdgmime.c	2009-10-06 16:07:59.000000000 -0400
+++ glib-2.22.4/gio/xdgmime/xdgmime.c	2010-01-13 08:35:09.000000000 -0500
@@ -257,7 +257,7 @@
 
   xdg_data_dirs = getenv ("XDG_DATA_DIRS");
   if (xdg_data_dirs == NULL)
-    xdg_data_dirs = "/usr/local/share/:/usr/share/";
+    xdg_data_dirs = "@PREFIX@/share:/usr/local/share/:/usr/share/";
 
   ptr = xdg_data_dirs;
 
diff -Nurd -x'*~' glib-2.22.4.orig/glib/gmacros.h glib-2.22.4/glib/gmacros.h
--- glib-2.22.4.orig/glib/gmacros.h	2009-03-31 19:04:20.000000000 -0400
+++ glib-2.22.4/glib/gmacros.h	2010-01-13 09:11:26.000000000 -0500
@@ -120,7 +120,8 @@
 #define G_GNUC_WARN_UNUSED_RESULT
 #endif /* __GNUC__ */
 
-#ifndef G_DISABLE_DEPRECATED
+#if 1
+/* don't supress (if G_DISABLE_DEPRECATED) for fink yet */
 /* Wrap the gcc __PRETTY_FUNCTION__ and __FUNCTION__ variables with
  * macros, so we can refer to them as strings unconditionally.
  * usage not-recommended since gcc-3.0
diff -Nurd -x'*~' glib-2.22.4.orig/glib/gmessages.h glib-2.22.4/glib/gmessages.h
--- glib-2.22.4.orig/glib/gmessages.h	2009-08-28 22:52:23.000000000 -0400
+++ glib-2.22.4/glib/gmessages.h	2010-01-13 09:11:36.000000000 -0500
@@ -121,7 +121,8 @@
                                int             line,
                                const char     *func,
                                const char     *warnexpr);
-#ifndef G_DISABLE_DEPRECATED
+#if 1
+/* don't supress (if G_DISABLE_DEPRECATED) for fink yet */
 void g_assert_warning         (const char *log_domain,
 			       const char *file,
 			       const int   line,
diff -Nurd -x'*~' glib-2.22.4.orig/glib/gslice.c glib-2.22.4/glib/gslice.c
--- glib-2.22.4.orig/glib/gslice.c	2009-03-31 19:04:20.000000000 -0400
+++ glib-2.22.4/glib/gslice.c	2010-01-13 09:06:27.000000000 -0500
@@ -1131,19 +1131,18 @@
                     gsize memsize)
 {
   gpointer aligned_memory = NULL;
-  gint err = ENOMEM;
 #if     HAVE_COMPLIANT_POSIX_MEMALIGN
+  gint err;
   err = posix_memalign (&aligned_memory, alignment, memsize);
+  if (!aligned_memory)
+    errno = err;
 #elif   HAVE_MEMALIGN
-  errno = 0;
   aligned_memory = memalign (alignment, memsize);
-  err = errno;
 #elif   HAVE_VALLOC
-  errno = 0;
   aligned_memory = valloc (memsize);
-  err = errno;
 #else
   /* simplistic non-freeing page allocator */
+  gint err = ENOMEM;
   mem_assert (alignment == sys_page_size);
   mem_assert (memsize <= sys_page_size);
   if (!compat_valloc_trash)
@@ -1162,9 +1161,9 @@
         }
     }
   aligned_memory = g_trash_stack_pop (&compat_valloc_trash);
-#endif
   if (!aligned_memory)
     errno = err;
+#endif
   return aligned_memory;
 }
 
diff -Nurd -x'*~' glib-2.22.4.orig/glib/gutils.c glib-2.22.4/glib/gutils.c
--- glib-2.22.4.orig/glib/gutils.c	2009-12-21 09:20:17.000000000 -0500
+++ glib-2.22.4/glib/gutils.c	2010-01-13 08:34:24.000000000 -0500
@@ -2779,7 +2779,7 @@
       gchar *data_dirs = (gchar *) g_getenv ("XDG_DATA_DIRS");
 
       if (!data_dirs || !data_dirs[0])
-          data_dirs = "/usr/local/share/:/usr/share/";
+          data_dirs = "@PREFIX@/share:/usr/local/share/:/usr/share/";
 
       data_dir_vector = g_strsplit (data_dirs, G_SEARCHPATH_SEPARATOR_S, 0);
 #endif
@@ -2833,7 +2833,7 @@
       conf_dirs = (gchar *) g_getenv ("XDG_CONFIG_DIRS");
 
       if (!conf_dirs || !conf_dirs[0])
-          conf_dirs = "/etc/xdg";
+          conf_dirs = "@PREFIX@/etc/xdg";
 
       conf_dir_vector = g_strsplit (conf_dirs, G_SEARCHPATH_SEPARATOR_S, 0);
 #endif
diff -Nurd -x'*~' glib-2.22.4.orig/glib/gutils.h glib-2.22.4/glib/gutils.h
--- glib-2.22.4.orig/glib/gutils.h	2010-01-06 19:19:20.000000000 -0500
+++ glib-2.22.4/glib/gutils.h	2010-01-13 08:33:50.000000000 -0500
@@ -101,7 +101,11 @@
 #  define G_INLINE_FUNC
 #  undef  G_CAN_INLINE
 #elif defined (__GNUC__) 
+#if defined(__APPLE_CC__) && (__APPLE_CC__ >= 5400)
+#  define G_INLINE_FUNC static inline
+#else
 #  define G_INLINE_FUNC static __inline __attribute__ ((unused))
+#endif
 #elif defined (G_CAN_INLINE) 
 #  define G_INLINE_FUNC static inline
 #else /* can't inline */
diff -Nurd -x'*~' glib-2.22.4.orig/glib/libcharset/Makefile.in glib-2.22.4/glib/libcharset/Makefile.in
--- glib-2.22.4.orig/glib/libcharset/Makefile.in	2010-01-06 19:48:38.000000000 -0500
+++ glib-2.22.4/glib/libcharset/Makefile.in	2010-01-13 08:33:32.000000000 -0500
@@ -310,7 +310,7 @@
 	codeset.m4 update.sh make-patch.sh libcharset-glib.patch
 TEST_PROGS = 
 INCLUDES = \
-	-DLIBDIR=\"$(libdir)\" -I$(top_srcdir)
+	-DLIBDIR=\"$(sysconfdir)/glib-2.0\" -I$(top_srcdir)
 
 noinst_LTLIBRARIES = libcharset.la
 libcharset_la_SOURCES = \
@@ -318,8 +318,8 @@
 	localcharset.h		\
 	localcharset.c
 
-charset_alias = $(DESTDIR)$(libdir)/charset.alias
-charset_tmp = $(DESTDIR)$(libdir)/charset.tmp
+charset_alias = $(DESTDIR)$(sysconfdir)/glib-2.0/charset.alias
+charset_tmp = $(DESTDIR)$(sysconfdir)/glib-2.0/charset.tmp
 SUFFIXES = .sed .sin
 CLEANFILES = charset.alias ref-add.sed ref-del.sed
 all: all-am
@@ -658,7 +658,7 @@
 # run make test as part of make check
 check-local: test
 install-exec-local: all-local
-	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(mkinstalldirs) $(DESTDIR)$(sysconfdir)/glib-2.0
 	if test -f $(charset_alias); then \
 	  sed -f ref-add.sed $(charset_alias) > $(charset_tmp) ; \
 	  $(INSTALL_DATA) $(charset_tmp) $(charset_alias) ; \
diff -Nurd -x'*~' glib-2.22.4.orig/glib/libcharset/charset.alias glib-2.22.4/glib/libcharset/charset.alias
--- glib-2.22.4.orig/glib/libcharset/charset.alias	1969-12-31 19:00:00.000000000 -0500
+++ glib-2.22.4/glib/libcharset/charset.alias	2010-01-13 08:32:49.000000000 -0500
@@ -0,0 +1,111 @@
+# This file contains a table of character encoding aliases,
+# suitable for operating system 'darwin'.
+C			ASCII
+bg_BG			UTF-8
+bg_BG.CP1251		CP1251
+cs_CZ			UTF-8
+cs_CZ.ISO8859-2		ISO-8859-2
+da_DK			UTF-8
+da_DK.ISO8859-1		ISO-8859-1
+da_DK.ISO8859-15	ISO-8859-15
+de_AT			UTF-8
+de_AT.ISO8859-1		ISO-8859-1
+de_AT.ISO8859-15	ISO-8859-15
+de_CH			UTF-8
+de_CH.ISO8859-1		ISO-8859-1
+de_CH.ISO8859-15	ISO-8859-15
+de_DE			UTF-8
+de_DE.ISO8859-1		ISO-8859-1
+de_DE.ISO8859-15	ISO-8859-15
+en_AU			UTF-8
+en_AU.ISO8859-1		ISO-8859-1
+en_AU.ISO8859-15	ISO-8859-15
+en_AU.US-ASCII		ASCII
+en_CA			UTF-8
+en_CA.ISO8859-1		ISO-8859-1
+en_CA.ISO8859-15	ISO-8859-15
+en_CA.US-ASCII		ASCII
+en_GB			UTF-8
+en_GB.ISO8859-1		ISO-8859-1
+en_GB.ISO8859-15	ISO-8859-15
+en_GB.US-ASCII		ASCII
+en_US			UTF-8
+en_US.ISO8859-1		ISO-8859-1
+en_US.ISO8859-15	ISO-8859-15
+en_US.US-ASCII		ASCII
+es_ES			UTF-8
+es_ES.ISO8859-1		ISO-8859-1
+es_ES.ISO8859-15	ISO-8859-15
+fi_FI			UTF-8
+fi_FI.ISO8859-1		ISO-8859-1
+fi_FI.ISO8859-15	ISO-8859-15
+fr_BE			UTF-8
+fr_BE.ISO8859-1		ISO-8859-1
+fr_BE.ISO8859-15	ISO-8859-15
+fr_CA			UTF-8
+fr_CA.ISO8859-1		ISO-8859-1
+fr_CA.ISO8859-15	ISO-8859-15
+fr_CH			UTF-8
+fr_CH.ISO8859-1		ISO-8859-1
+fr_CH.ISO8859-15	ISO-8859-15
+fr_FR			UTF-8
+fr_FR.ISO8859-1		ISO-8859-1
+fr_FR.ISO8859-15	ISO-8859-15
+hr_HR			UTF-8
+hr_HR.ISO8859-2		ISO-8859-2
+hu_HU			UTF-8
+hu_HU.ISO8859-2		ISO-8859-2
+is_IS			UTF-8
+is_IS.ISO8859-1		ISO-8859-1
+is_IS.ISO8859-15	ISO-8859-15
+it_CH			UTF-8
+it_CH.ISO8859-1		ISO-8859-1
+it_CH.ISO8859-15	ISO-8859-15
+it_IT			UTF-8
+it_IT.ISO8859-1		ISO-8859-1
+it_IT.ISO8859-15	ISO-8859-15
+ja_JP			UTF-8
+ja_JP.EUC		EUC-JP
+ja_JP.SJIS		SHIFT_JIS
+ko_KR			UTF-8
+ko_KR.EUC		EUC-KR
+la_LN			UTF-8
+la_LN.ISO8859-1		ISO-8859-1
+la_LN.ISO8859-2		ISO-8859-2
+la_LN.ISO8859-4		ISO-8859-4
+la_LN.ISO8859-15	ISO-8859-15
+la_LN.US-ASCII		ASCII
+lt_LT			UTF-8
+lt_LT.ISO8859-4		ISO-8859-4
+nl_BE			UTF-8
+nl_BE.ISO8869-1		ISO-8859-1
+nl_BE.ISO8869-15	ISO-8859-15
+nl_NL			UTF-8
+nl_NL.ISO8869-1		ISO-8859-1
+nl_NL.ISO8869-15	ISO-8859-15
+no_NO			UTF-8
+no_NO.ISO8869-1		ISO-8859-1
+no_NO.ISO8869-15	ISO-8859-15
+pl_PL			UTF-8
+pl_PL.ISO8859-2		ISO-8859-2
+pt_PT			UTF-8
+pt_PT.ISO8859-1		ISO-8859-1
+pt_PT.ISO8859-15	ISO-8859-15
+ru_RU			UTF-8
+ru_RU.CP866		CP866
+ru_RU.ISO8859-5		ISO-8859-5
+ru_RU.KOI8-R		KOI8-R
+ru_RU.cp1251		CP1251
+ru_RU.koi8r		KOI8-R
+sl_SI			UTF-8
+sl_SI.ISO8859-2		ISO-8859-2
+sv_SE			UTF-8
+sv_SE.ISO8859-1		ISO-8859-1
+sv_SE.ISO8859-15	ISO-8859-15
+uk_UA			UTF-8
+uk_UA.KOI8-U		KOI8-U
+uk_UA.cp1251		CP1251
+zh_CN			UTF-8
+zh_CN.EUC		GB2312
+zh_TW			UTF-8
+zh_TW.Big5		BIG5
diff -Nurd -x'*~' glib-2.22.4.orig/glib-gettextize.in glib-2.22.4/glib-gettextize.in
--- glib-2.22.4.orig/glib-gettextize.in	2009-03-31 19:04:20.000000000 -0400
+++ glib-2.22.4/glib-gettextize.in	2010-01-13 08:31:30.000000000 -0500
@@ -49,8 +49,8 @@
     ;;
 esac
 
-datadir=@datadir@
 datarootdir=@datarootdir@
+datadir=@datadir@
 
 gettext_dir=$prefix/share/glib-2.0/gettext
 
diff -Nurd -x'*~' glib-2.22.4.orig/tests/child-test.c glib-2.22.4/tests/child-test.c
--- glib-2.22.4.orig/tests/child-test.c	2009-03-31 19:04:20.000000000 -0400
+++ glib-2.22.4/tests/child-test.c	2010-01-13 08:30:21.000000000 -0500
@@ -175,7 +175,7 @@
 #ifdef G_OS_WIN32
   system ("ipconfig /all");
 #else
-  system ("/bin/true");
+  system ("/usr/bin/true");
 #endif
 
   alive = 2;
