diff -Nurd -x'*~' gmetadom-0.2.6.orig/src/gdome_cpp_smart/Makefile.in gmetadom-0.2.6/src/gdome_cpp_smart/Makefile.in
--- gmetadom-0.2.6.orig/src/gdome_cpp_smart/Makefile.in	2007-10-15 05:42:42.000000000 -0400
+++ gmetadom-0.2.6/src/gdome_cpp_smart/Makefile.in	2008-09-28 03:16:16.000000000 -0400
@@ -229,17 +229,16 @@
   $(GDOME_LIBS) \
   $(top_builddir)/src/gdome_cpp_smart/events/libevents.la \
   $(top_builddir)/src/gdome_cpp_smart/core/libcore.la \
-  $(top_builddir)/src/gdome_cpp_smart/basic/libbasic.la \
-  -lstdc++
+  $(top_builddir)/src/gdome_cpp_smart/basic/libbasic.la
 
 libgmetadom_gdome_cpp_smart_la_LDFLAGS = -version-info @GMETADOM_VERSION_INFO@
 libgmetadom_gdome_cpp_smart_la_SOURCES = \
   GdomeSmartDOMDOMImplementation.cc
 
 INCLUDES = \
-  $(GDOME_CFLAGS) \
   -I$(srcdir)/include \
-  -I$(top_builddir)/src/gdome_cpp_smart/include
+  -I$(top_builddir)/src/gdome_cpp_smart/include \
+  $(GDOME_CFLAGS)
 
 all: all-recursive
 
diff -Nurd -x'*~' gmetadom-0.2.6.orig/src/gdome_cpp_smart/basic/Makefile.in gmetadom-0.2.6/src/gdome_cpp_smart/basic/Makefile.in
--- gmetadom-0.2.6.orig/src/gdome_cpp_smart/basic/Makefile.in	2007-10-15 05:42:42.000000000 -0400
+++ gmetadom-0.2.6/src/gdome_cpp_smart/basic/Makefile.in	2008-09-28 03:16:16.000000000 -0400
@@ -213,9 +213,9 @@
 noinst_LTLIBRARIES = libbasic.la
 libbasic_la_SOURCES = $(AUTO_CC)
 INCLUDES = \
-  $(GDOME_CFLAGS) \
   -I$(srcdir)/../include \
-  -I$(top_builddir)/src/gdome_cpp_smart/include
+  -I$(top_builddir)/src/gdome_cpp_smart/include \
+  $(GDOME_CFLAGS)
 
 all: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) all-am
diff -Nurd -x'*~' gmetadom-0.2.6.orig/src/gdome_cpp_smart/core/Makefile.in gmetadom-0.2.6/src/gdome_cpp_smart/core/Makefile.in
--- gmetadom-0.2.6.orig/src/gdome_cpp_smart/core/Makefile.in	2007-10-15 05:42:43.000000000 -0400
+++ gmetadom-0.2.6/src/gdome_cpp_smart/core/Makefile.in	2008-09-28 03:16:16.000000000 -0400
@@ -238,9 +238,9 @@
   $(AUTO_CC)
 
 INCLUDES = \
-  $(GDOME_CFLAGS) \
   -I$(srcdir)/../include \
-  -I$(top_builddir)/src/gdome_cpp_smart/include
+  -I$(top_builddir)/src/gdome_cpp_smart/include \
+  $(GDOME_CFLAGS)
 
 all: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) all-am
diff -Nurd -x'*~' gmetadom-0.2.6.orig/src/gdome_cpp_smart/events/Makefile.in gmetadom-0.2.6/src/gdome_cpp_smart/events/Makefile.in
--- gmetadom-0.2.6.orig/src/gdome_cpp_smart/events/Makefile.in	2007-10-15 05:42:43.000000000 -0400
+++ gmetadom-0.2.6/src/gdome_cpp_smart/events/Makefile.in	2008-09-28 03:16:16.000000000 -0400
@@ -218,9 +218,9 @@
   $(AUTO_CC)
 
 INCLUDES = \
-  $(GDOME_CFLAGS) \
   -I$(srcdir)/../include \
-  -I$(top_builddir)/src/gdome_cpp_smart/include
+  -I$(top_builddir)/src/gdome_cpp_smart/include \
+  $(GDOME_CFLAGS)
 
 all: all-am
 
diff -Nurd -x'*~' gmetadom-0.2.6.orig/src/gdome_cpp_smart/test/Makefile.in gmetadom-0.2.6/src/gdome_cpp_smart/test/Makefile.in
--- gmetadom-0.2.6.orig/src/gdome_cpp_smart/test/Makefile.in	2007-10-15 05:42:44.000000000 -0400
+++ gmetadom-0.2.6/src/gdome_cpp_smart/test/Makefile.in	2008-09-28 03:16:16.000000000 -0400
@@ -219,16 +219,15 @@
 LDADDS = \
   $(GDOME_LIBS) \
   $(top_builddir)/src/gdome_cpp_smart/.libs/libgmetadom_gdome_cpp_smart.a \
-  $(top_builddir)/src/gdome_cpp_smart/libgmetadom_gdome_cpp_smart.la \
-  -lstdc++
+  $(top_builddir)/src/gdome_cpp_smart/libgmetadom_gdome_cpp_smart.la
 
 test_LDADD = $(LDADDS)
 basic_LDADD = $(LDADDS)
 deep_LDADD = $(LDADDS)
 INCLUDES = \
-  $(GDOME_CFLAGS) \
   -I$(srcdir)/../include \
-  -I$(top_builddir)/src/gdome_cpp_smart/include
+  -I$(top_builddir)/src/gdome_cpp_smart/include \
+  $(GDOME_CFLAGS)
 
 all: all-am
 
