diff -Nurd -x'*~' gnome-utils-2.24.1.orig/baobab/src/baobab.c gnome-utils-2.24.1/baobab/src/baobab.c
--- gnome-utils-2.24.1.orig/baobab/src/baobab.c	2008-08-09 11:18:50.000000000 -0400
+++ gnome-utils-2.24.1/baobab/src/baobab.c	2009-03-20 04:01:47.000000000 -0400
@@ -20,6 +20,8 @@
  * Boston, MA  02110-1301  USA
  */
  
+#define NO_EXTERN_BAOBAB 1
+
 #include <config.h>
 
 #include <gtk/gtk.h>
diff -Nurd -x'*~' gnome-utils-2.24.1.orig/baobab/src/baobab.h gnome-utils-2.24.1/baobab/src/baobab.h
--- gnome-utils-2.24.1.orig/baobab/src/baobab.h	2008-08-09 11:18:50.000000000 -0400
+++ gnome-utils-2.24.1/baobab/src/baobab.h	2009-03-20 04:01:21.000000000 -0400
@@ -103,9 +103,15 @@
 };
 
 /* globals */
+#ifdef NO_EXTERN_BAOBAB
 gboolean g_noactivescans;
 baobab_application baobab;
 baobab_fs g_fs;
+#else
+extern gboolean g_noactivescans;
+extern baobab_application baobab;
+extern baobab_fs g_fs;
+#endif
 
 void baobab_set_busy (gboolean busy);
 void baobab_scan_location (GFile *);
diff -Nurd -x'*~' gnome-utils-2.24.1.orig/gnome-dictionary/src/Makefile.in gnome-utils-2.24.1/gnome-dictionary/src/Makefile.in
--- gnome-utils-2.24.1.orig/gnome-dictionary/src/Makefile.in	2008-10-20 15:05:40.000000000 -0400
+++ gnome-utils-2.24.1/gnome-dictionary/src/Makefile.in	2009-03-20 05:15:56.000000000 -0400
@@ -325,6 +325,11 @@
 	-DUIDATADIR=\""$(datadir)/gnome-2.0/ui"\" 	\
 	-DPKGDATADIR=\""$(datadir)/gnome-dictionary"\" 	\
 	-DGNOMELOCALEDIR=\""$(prefix)/$(DATADIRNAME)/locale"\" \
+	-I$(top_builddir)/gnome-dictionary \
+	-I$(top_srcdir)/gnome-dictionary \
+	$(GNOME_UTILS_CFLAGS) \
+	$(LIBGLADE_CFLAGS) \
+	$(APPLET_CFLAGS) \
 	$(NULL)
 
 gnome_dictionary_SOURCES = \
@@ -348,10 +353,6 @@
 	$(NULL)
 
 gnome_dictionary_CFLAGS = \
-	-I$(top_builddir)/gnome-dictionary \
-	-I$(top_srcdir)/gnome-dictionary \
-	$(GNOME_UTILS_CFLAGS) \
-	$(LIBGLADE_CFLAGS) \
 	$(NULL)
 
 gnome_dictionary_LDADD = \
@@ -380,11 +381,6 @@
 	$(NULL)
 
 gnome_dictionary_applet_CFLAGS = \
-	-I$(top_builddir)/gnome-dictionary \
-	-I$(top_srcdir)/gnome-dictionary \
-	$(GNOME_UTILS_CFLAGS) \
-	$(LIBGLADE_CFLAGS) \
-	$(APPLET_CFLAGS) \
 	$(NULL)
 
 gnome_dictionary_applet_LDADD = \
