diff -Naur gnome-applets-1.4.0.5.old/charpick/Makefile.in gnome-applets-1.4.0.5.new/charpick/Makefile.in
--- gnome-applets-1.4.0.5.old/charpick/Makefile.in	Sun Feb  3 09:39:24 2002
+++ gnome-applets-1.4.0.5.new/charpick/Makefile.in	Tue Nov  5 22:41:49 2002
@@ -180,7 +180,7 @@
 	-I$(includedir) $(GNOME_INCLUDEDIR)
 
 
-CFLAGS = -ansi -Werror -Wmissing-prototypes -Wmissing-declarations
+CFLAGS = -ansi -Wmissing-prototypes -Wmissing-declarations
 
 bin_PROGRAMS = charpick_applet
 
diff -Naur gnome-applets-1.4.0.5.old/gkb-new/preset.c gnome-applets-1.4.0.5.new/gkb-new/preset.c
--- gnome-applets-1.4.0.5.old/gkb-new/preset.c	Fri Oct 26 18:12:16 2001
+++ gnome-applets-1.4.0.5.new/gkb-new/preset.c	Tue Nov  5 22:40:25 2002
@@ -29,6 +29,7 @@
  * USA
  */
 
+#include <sys/types.h>
 #include <dirent.h>
 #include "gkb.h"
 
diff -Naur gnome-applets-1.4.0.5.old/jbc/jbc-applet.c gnome-applets-1.4.0.5.new/jbc/jbc-applet.c
--- gnome-applets-1.4.0.5.old/jbc/jbc-applet.c	Fri Oct 26 18:11:16 2001
+++ gnome-applets-1.4.0.5.new/jbc/jbc-applet.c	Tue Nov  5 22:40:25 2002
@@ -74,7 +74,11 @@
 do_flicker ()
 {
   thetime = time (NULL);
+#ifdef HAVE_LOCALTIME_R
   localtime_r (&thetime, &atime);
+#else
+  memcpy (&atime, localtime (&thetime), sizeof (struct tm));
+#endif
 
   d[0] = atime.tm_hour / 10;
   d[1] = atime.tm_hour % 10;
diff -Naur gnome-applets-1.4.0.5.old/omf-install/Makefile.in gnome-applets-1.4.0.5.new/omf-install/Makefile.in
--- gnome-applets-1.4.0.5.old/omf-install/Makefile.in	Sun Feb  3 09:50:53 2002
+++ gnome-applets-1.4.0.5.new/omf-install/Makefile.in	Tue Nov  5 22:40:25 2002
@@ -296,7 +296,7 @@
 	-for file in $(srcdir)/*.omf; do \
 		$(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(omf_dest_dir); \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) 
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) 
 
 uninstall-local:
 	-for file in $(srcdir)/*.omf; do \
