diff -Nurd -x'*~' gnome-media-2.24.0.1.orig/cddb-slave2/Makefile.in gnome-media-2.24.0.1/cddb-slave2/Makefile.in
--- gnome-media-2.24.0.1.orig/cddb-slave2/Makefile.in	2008-09-23 18:16:50.000000000 -0400
+++ gnome-media-2.24.0.1/cddb-slave2/Makefile.in	2009-01-21 13:44:50.000000000 -0500
@@ -959,8 +959,8 @@
 
 
 $(IDL_GENERATED): $(IDLS)
-	$(ORBIT_IDL) -I $(BONOBO_IDLDIR) -I $(BONOBO_ACT_IDLDIR)	\
-	-I $(srcdir) -I $(datadir)/idl 	\
+	$(ORBIT_IDL) -I $(srcdir) -I $(datadir)/idl 	\
+	-I $(BONOBO_IDLDIR) -I $(BONOBO_ACT_IDLDIR)	\
 		$(srcdir)/GNOME_Media_CDDBSlave2.idl
 
 # since automake builds dependencies AFTER objects, and the object
diff -Nurd -x'*~' gnome-media-2.24.0.1.orig/gnome-cd/access/Makefile.in gnome-media-2.24.0.1/gnome-cd/access/Makefile.in
--- gnome-media-2.24.0.1.orig/gnome-cd/access/Makefile.in	2008-09-23 18:16:51.000000000 -0400
+++ gnome-media-2.24.0.1/gnome-cd/access/Makefile.in	2009-01-21 13:44:18.000000000 -0500
@@ -275,8 +275,8 @@
 top_srcdir = @top_srcdir@
 AM_CPPFLAGS = \
         -I$(top_srcdir) \
-	$(MEDIA_CFLAGS)	\
 	-I$(top_srcdir)/cddb-slave2	\
+	$(MEDIA_CFLAGS)	\
 	$(GAILUTIL_CFLAGS) \
 	$(LIBXML2_CFLAGS)
 
diff -Nurd -x'*~' gnome-media-2.24.0.1.orig/profiles/Makefile.in gnome-media-2.24.0.1/profiles/Makefile.in
--- gnome-media-2.24.0.1.orig/profiles/Makefile.in	2008-09-23 18:16:54.000000000 -0400
+++ gnome-media-2.24.0.1/profiles/Makefile.in	2009-01-21 13:42:52.000000000 -0500
@@ -355,13 +355,13 @@
 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"
 
 @HAVE_GSTREAMER_TRUE@gnome_audio_profiles_test_SOURCES = gnome-audio-profiles-test.c
-@HAVE_GSTREAMER_TRUE@gnome_audio_profiles_test_CFLAGS = $(GMP_CFLAGS) $(GSR_CFLAGS) $(defines) -I$(top_srcdir)
+@HAVE_GSTREAMER_TRUE@gnome_audio_profiles_test_CFLAGS = -I$(top_srcdir) $(GMP_CFLAGS) $(GSR_CFLAGS) $(defines)
 @HAVE_GSTREAMER_TRUE@gnome_audio_profiles_test_LDADD = libgnome-media-profiles.la $(GMP_LIBS) $(GSR_LIBS)
 lib_LTLIBRARIES = libgnome-media-profiles.la
 gladedir = $(pkgdatadir)/glade
 glade_DATA = gnome-audio-profiles.glade2
 gnome_audio_profiles_properties_SOURCES = gnome-audio-profiles-properties.c
-gnome_audio_profiles_properties_CFLAGS = $(GMP_CFLAGS) $(defines) -I$(top_srcdir)
+gnome_audio_profiles_properties_CFLAGS = -I$(top_srcdir) $(GMP_CFLAGS) $(defines)
 gnome_audio_profiles_properties_LDADD = libgnome-media-profiles.la $(GMP_LIBS)
 libgnome_media_profiles_la_SOURCES = \
 	audio-profile.c \
diff -Nurd -x'*~' gnome-media-2.24.0.1.orig/profiles/glade/Makefile.in gnome-media-2.24.0.1/profiles/glade/Makefile.in
--- gnome-media-2.24.0.1.orig/profiles/glade/Makefile.in	2008-09-23 18:16:55.000000000 -0400
+++ gnome-media-2.24.0.1/profiles/glade/Makefile.in	2009-01-21 13:43:03.000000000 -0500
@@ -292,7 +292,7 @@
 	-DGNOMELOCALEDIR=\""$(prefix)/${DATADIRNAME}/locale"\"
 
 libgnome_media_profiles_la_SOURCES = gnome-media-profiles-glade.c
-libgnome_media_profiles_la_CFLAGS = $(GMP_CFLAGS) $(defines) -I$(top_srcdir)
+libgnome_media_profiles_la_CFLAGS = -I$(top_srcdir) $(GMP_CFLAGS) $(defines)
 libgnome_media_profiles_la_LIBADD = $(GMP_LIBS) $(top_builddir)/profiles/libgnome-media-profiles.la
 libgnome_media_profiles_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
 all: all-am
