diff -Nurd -x'*~' gnome-panel-2.26.2.orig/doc/reference/panel-applet/Makefile.in gnome-panel-2.26.2/doc/reference/panel-applet/Makefile.in
--- gnome-panel-2.26.2.orig/doc/reference/panel-applet/Makefile.in	2009-05-20 09:22:49.000000000 -0400
+++ gnome-panel-2.26.2/doc/reference/panel-applet/Makefile.in	2009-06-10 00:54:24.000000000 -0400
@@ -318,11 +318,11 @@
 # CFLAGS and LDFLAGS for compiling scan program. Only needed if your app/lib
 # contains GtkObjects/GObjects and you want to document signals and properties.
 GTKDOC_CFLAGS = \
+	-I$(srcdir)/../../../libpanel-applet    \
+	-I$(top_builddir)/libpanel-applet	\
 	$(CFLAGS)				\
 	$(CPPFLAGS)				\
-	$(PANEL_CFLAGS)				\
-	-I$(srcdir)/../../../libpanel-applet    \
-	-I$(top_builddir)/libpanel-applet
+	$(PANEL_CFLAGS)
 
 GTKDOC_LIBS = \
 	$(LDFLAGS)					\
