diff -Nurd -x'*~' gnome-media-2.26.0.orig/cddb-slave2/Makefile.am gnome-media-2.26.0/cddb-slave2/Makefile.am
--- gnome-media-2.26.0.orig/cddb-slave2/Makefile.am	2008-09-07 14:54:07.000000000 -0400
+++ gnome-media-2.26.0/cddb-slave2/Makefile.am	2009-06-10 16:21:09.000000000 -0400
@@ -17,8 +17,9 @@
 	GNOME_Media_CDDBSlave2-stubs.c
 
 $(IDL_GENERATED): $(IDLS)
-	$(ORBIT_IDL) -I $(BONOBO_IDLDIR) -I $(BONOBO_ACT_IDLDIR)	\
+	$(ORBIT_IDL) \
 	-I $(srcdir) -I $(datadir)/idl 	\
+	-I $(BONOBO_IDLDIR) -I $(BONOBO_ACT_IDLDIR)	\
 		$(srcdir)/GNOME_Media_CDDBSlave2.idl
 
 idldir = $(datadir)/idl
diff -Nurd -x'*~' gnome-media-2.26.0.orig/configure.ac gnome-media-2.26.0/configure.ac
--- gnome-media-2.26.0.orig/configure.ac	2009-03-04 19:55:59.000000000 -0500
+++ gnome-media-2.26.0/configure.ac	2009-06-10 15:50:28.000000000 -0400
@@ -111,18 +111,6 @@
 dnl The new gnome-volume-control needs a newer GTK+
 GTK_REQUIRED_VERSION=2.15.1
 
-PKG_CHECK_MODULES(VOLUME_CONTROL,
-        gobject-2.0 >= $GLIB_REQUIRED_VERSION
-        gtk+-2.0 >= $GTK_REQUIRED_VERSION
-        gio-2.0
-        gconf-2.0 >= $GCONF_REQUIRED_VERSION
-        libcanberra-gtk >= $CANBERRA_REQUIRED_VERSION
-        unique-1.0
-        libxml-2.0
-)
-AC_SUBST(VOLUME_CONTROL_CFLAGS)
-AC_SUBST(VOLUME_CONTROL_LIBS)
-
 PKG_CHECK_MODULES(PULSEAUDIO,
         libpulse >= $PA_REQUIRED_VERSION libpulse-mainloop-glib >= $PA_REQUIRED_VERSION,
         have_pulseaudio=yes,
@@ -139,6 +127,17 @@
   fi
 
   if test "x$have_pulseaudio" = "xyes" ; then
+    PKG_CHECK_MODULES(VOLUME_CONTROL,
+            gobject-2.0 >= $GLIB_REQUIRED_VERSION
+            gtk+-2.0 >= $GTK_REQUIRED_VERSION
+            gio-2.0
+            gconf-2.0 >= $GCONF_REQUIRED_VERSION
+            libcanberra-gtk >= $CANBERRA_REQUIRED_VERSION
+            unique-1.0
+            libxml-2.0
+    )
+    AC_SUBST(VOLUME_CONTROL_CFLAGS)
+    AC_SUBST(VOLUME_CONTROL_LIBS)
     AC_DEFINE(HAVE_PULSEAUDIO, [], [Define if we have pulseaudio])
   fi
 else
diff -Nurd -x'*~' gnome-media-2.26.0.orig/gnome-cd/access/Makefile.in gnome-media-2.26.0/gnome-cd/access/Makefile.in
--- gnome-media-2.26.0.orig/gnome-cd/access/Makefile.in	2009-03-16 15:33:37.000000000 -0400
+++ gnome-media-2.26.0/gnome-cd/access/Makefile.in	2009-06-10 14:12:25.000000000 -0400
@@ -291,8 +291,8 @@
 top_srcdir = @top_srcdir@
 AM_CPPFLAGS = \
         -I$(top_srcdir) \
-	$(MEDIA_CFLAGS)	\
 	-I$(top_srcdir)/cddb-slave2	\
+	$(MEDIA_CFLAGS)	\
 	$(GAILUTIL_CFLAGS) \
 	$(LIBXML2_CFLAGS)
 
diff -Nurd -x'*~' gnome-media-2.26.0.orig/profiles/Makefile.am gnome-media-2.26.0/profiles/Makefile.am
--- gnome-media-2.26.0.orig/profiles/Makefile.am	2009-01-20 17:26:11.000000000 -0500
+++ gnome-media-2.26.0/profiles/Makefile.am	2009-06-10 16:21:47.000000000 -0400
@@ -11,7 +11,7 @@
 if HAVE_GSTREAMER
 noinst_PROGRAMS += gnome-audio-profiles-test
 gnome_audio_profiles_test_SOURCES = gnome-audio-profiles-test.c
-gnome_audio_profiles_test_CFLAGS = $(GMP_CFLAGS) $(GSR_CFLAGS) $(defines) -I$(top_srcdir)
+gnome_audio_profiles_test_CFLAGS = -I$(top_srcdir) $(GMP_CFLAGS) $(GSR_CFLAGS) $(defines)
 gnome_audio_profiles_test_LDADD = libgnome-media-profiles.la $(GMP_LIBS) $(GSR_LIBS)
 endif
 
@@ -24,7 +24,7 @@
 glade_DATA = gnome-audio-profiles.glade2
 
 gnome_audio_profiles_properties_SOURCES = gnome-audio-profiles-properties.c
-gnome_audio_profiles_properties_CFLAGS = $(GMP_CFLAGS) $(defines) -I$(top_srcdir)
+gnome_audio_profiles_properties_CFLAGS = -I$(top_srcdir) $(GMP_CFLAGS) $(defines)
 gnome_audio_profiles_properties_LDADD = libgnome-media-profiles.la $(GMP_LIBS)
 
 libgnome_media_profiles_la_SOURCES = \
diff -Nurd -x'*~' gnome-media-2.26.0.orig/profiles/glade/Makefile.am gnome-media-2.26.0/profiles/glade/Makefile.am
--- gnome-media-2.26.0.orig/profiles/glade/Makefile.am	2009-01-20 17:26:11.000000000 -0500
+++ gnome-media-2.26.0/profiles/glade/Makefile.am	2009-06-10 16:22:13.000000000 -0400
@@ -6,7 +6,7 @@
 	-DGNOMELOCALEDIR=\""$(prefix)/${DATADIRNAME}/locale"\"
 
 libgnome_media_profiles_la_SOURCES = gnome-media-profiles-glade.c
-libgnome_media_profiles_la_CFLAGS = $(GMP_CFLAGS) $(defines) -I$(top_srcdir)
+libgnome_media_profiles_la_CFLAGS = -I$(top_srcdir) $(GMP_CFLAGS) $(defines)
 libgnome_media_profiles_la_LIBADD = $(GMP_LIBS) $(top_builddir)/profiles/libgnome-media-profiles.la
 libgnome_media_profiles_la_LDFLAGS = -export-dynamic -module -avoid-version -no-undefined
 
