diff -Nurd -x'*~' gnome-menus-2.14.3.orig/ltmain.sh gnome-menus-2.14.3/ltmain.sh
--- gnome-menus-2.14.3.orig/ltmain.sh	2006-04-29 04:39:27.000000000 -0400
+++ gnome-menus-2.14.3/ltmain.sh	2006-08-11 02:26:59.000000000 -0400
@@ -4015,6 +4015,11 @@
 	  eval test_cmds=\"$archive_expsym_cmds\"
 	  cmds=$archive_expsym_cmds
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+          else
+            tmp_verstring="$verstring"
+          fi
 	  eval test_cmds=\"$archive_cmds\"
 	  cmds=$archive_cmds
 	  fi
