diff -Nurd -x'*~' gnome-doc-utils-0.16.1.orig/Makefile.am gnome-doc-utils-0.16.1/Makefile.am
--- gnome-doc-utils-0.16.1.orig/Makefile.am	2008-09-22 16:06:12.000000000 -0400
+++ gnome-doc-utils-0.16.1/Makefile.am	2009-05-13 19:16:06.000000000 -0400
@@ -12,7 +12,7 @@
 # doc must go last
 SUBDIRS = data po xslt tools $(build_SUBDIRS) doc
 
-ACLOCAL_AMFLAGS = -I m4
+ACLOCAL_AMFLAGS = -I m4 -I tools
 
 EXTRA_DIST = 						\
 	$(pkgdata_DATA)					\
diff -Nurd -x'*~' gnome-doc-utils-0.16.1.orig/configure.in gnome-doc-utils-0.16.1/configure.in
--- gnome-doc-utils-0.16.1.orig/configure.in	2009-04-14 15:00:47.000000000 -0400
+++ gnome-doc-utils-0.16.1/configure.in	2009-05-13 23:26:57.000000000 -0400
@@ -58,8 +58,6 @@
 doc/xslt/Makefile
 po/Makefile.in
 tools/Makefile
-tools/gnome-doc-tool
-tools/gnome-doc-prepare
 tools/gnome-doc-utils.pc
 xslt/Makefile
 xml2po/Makefile
@@ -73,9 +71,9 @@
 xslt/docbook/omf/Makefile
 xslt/docbook/utils/Makefile
 xslt/gettext/Makefile
-],[
-chmod +x tools/gnome-doc-tool
-chmod +x tools/gnome-doc-prepare
 ])
 
+AC_CONFIG_FILES([tools/gnome-doc-tool],[chmod +x tools/gnome-doc-tool])
+AC_CONFIG_FILES([tools/gnome-doc-prepare],[chmod +x tools/gnome-doc-prepare])
+
 AC_OUTPUT
