--- gst-plugins-base-0.10.2/ext/cdparanoia/gstcdparanoiasrc.c	2006-01-10 13:06:56.000000000 -0500
+++ gst-plugins-base-0.10.2-new/ext/cdparanoia/gstcdparanoiasrc.c	2006-01-18 23:06:40.000000000 -0500
@@ -24,6 +24,9 @@
 #include "config.h"
 #endif
 
+#define GSTVERSION "@GSTVERSION@"
+#undef VERSION
+
 #include <string.h>
 #include <errno.h>
 
@@ -401,4 +404,4 @@
     GST_VERSION_MINOR,
     "cdparanoia",
     "Read audio from CD in paranoid mode",
-    plugin_init, VERSION, "GPL", GST_PACKAGE_NAME, GST_PACKAGE_ORIGIN)
+    plugin_init, GSTVERSION, "GPL", GST_PACKAGE_NAME, GST_PACKAGE_ORIGIN)
