diff -Nurd gst-python-0.10.21/configure gst-python-0.10.21-new/configure
--- gst-python-0.10.21/configure	2011-01-20 15:13:22.000000000 -0500
+++ gst-python-0.10.21-new/configure	2011-05-18 19:06:23.000000000 -0400
@@ -15070,8 +15070,8 @@
     fi
   fi
 else
-  PYTHON_LIBS="-L${py_prefix}/lib -lpython${PYTHON_VERSION}"
-  PYTHON_LIB_LOC="${py_prefix}/lib"
+  PYTHON_LIBS="-L${py_prefix}/lib -L${py_prefix}/lib/python${PYTHON_VERSION}/config -lpython${PYTHON_VERSION}"
+  PYTHON_LIB_LOC="${py_prefix}/lib/python${PYTHON_VERSION}/config"
 fi
 
 
