diff -Nurd gst-plugins-bad-0.10.20/ext/mpeg2enc/gstmpeg2encoptions.cc gst-plugins-bad-0.10.20-new/ext/mpeg2enc/gstmpeg2encoptions.cc
--- gst-plugins-bad-0.10.20/ext/mpeg2enc/gstmpeg2encoptions.cc	2010-05-13 20:48:27.000000000 -0400
+++ gst-plugins-bad-0.10.20-new/ext/mpeg2enc/gstmpeg2encoptions.cc	2010-10-02 18:58:02.000000000 -0400
@@ -30,6 +30,8 @@
 
 #include <gst/gst.h>
 
+int force_cbr = 0;
+
 /*
  * Property enumeration.
  */
diff -Nurd gst-plugins-bad-0.10.20/tests/examples/camerabin/Makefile.am gst-plugins-bad-0.10.20-new/tests/examples/camerabin/Makefile.am
--- gst-plugins-bad-0.10.20/tests/examples/camerabin/Makefile.am	2010-09-02 17:30:31.000000000 -0400
+++ gst-plugins-bad-0.10.20-new/tests/examples/camerabin/Makefile.am	2010-10-02 18:58:43.000000000 -0400
@@ -17,7 +17,7 @@
         -lgstinterfaces-@GST_MAJORMINOR@ \
         $(GST_LIBS) \
 	$(GTK_LIBS) \
-	$(GMODULE_EXPORT_LIBS)
+	$(GMODULE_EXPORT_LIBS) $(X_LIBS)
 
 noinst_DATA = $(GST_CAMERABIN_UI_FILES)
 
@@ -29,7 +29,7 @@
 
 gst_camera_perf_SOURCES = gst-camera-perf.c
 gst_camera_perf_CFLAGS  = $(GST_CFLAGS)
-gst_camera_perf_LDADD = $(GST_LIBS) 
+gst_camera_perf_LDADD = $(GST_LIBS) $(X_LIBS)
 
 if HAVE_X11
 
diff -Nurd gst-plugins-bad-0.10.20/tests/examples/camerabin/Makefile.in gst-plugins-bad-0.10.20-new/tests/examples/camerabin/Makefile.in
--- gst-plugins-bad-0.10.20/tests/examples/camerabin/Makefile.in	2010-09-02 17:31:24.000000000 -0400
+++ gst-plugins-bad-0.10.20-new/tests/examples/camerabin/Makefile.in	2010-10-02 18:59:22.000000000 -0400
@@ -542,13 +542,13 @@
 @HAVE_GTK_TRUE@        -lgstinterfaces-@GST_MAJORMINOR@ \
 @HAVE_GTK_TRUE@        $(GST_LIBS) \
 @HAVE_GTK_TRUE@	$(GTK_LIBS) \
-@HAVE_GTK_TRUE@	$(GMODULE_EXPORT_LIBS)
+@HAVE_GTK_TRUE@	$(GMODULE_EXPORT_LIBS) $(X_LIBS)
 
 @HAVE_GTK_TRUE@noinst_DATA = $(GST_CAMERABIN_UI_FILES)
 @HAVE_GTK_TRUE@INCLUDES = -DCAMERA_APPS_UIDIR=\""$(uidir)"\"
 gst_camera_perf_SOURCES = gst-camera-perf.c
 gst_camera_perf_CFLAGS = $(GST_CFLAGS)
-gst_camera_perf_LDADD = $(GST_LIBS) 
+gst_camera_perf_LDADD = $(GST_LIBS) $(X_LIBS)
 @HAVE_X11_FALSE@GST_CAMERABIN_X11_EXAMPLES = 
 @HAVE_X11_TRUE@GST_CAMERABIN_X11_EXAMPLES = gst-camerabin-test
 @HAVE_X11_TRUE@gst_camerabin_test_SOURCES = gst-camerabin-test.c
