diff -Nurd -x'*~' gtk-engines-2.16.1.orig/engines/clearlooks/Makefile.am gtk-engines-2.16.1/engines/clearlooks/Makefile.am
--- gtk-engines-2.16.1.orig/engines/clearlooks/Makefile.am	2008-10-20 11:54:26.000000000 -0400
+++ gtk-engines-2.16.1/engines/clearlooks/Makefile.am	2008-11-09 12:15:11.000000000 -0500
@@ -1,5 +1,5 @@
-INCLUDES = $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS) \
-           -I$(top_srcdir)/engines/support
+INCLUDES = -I$(top_srcdir)/engines/support \
+           $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS)
 
 enginedir = $(libdir)/gtk-2.0/$(GTK_VERSION)/engines
 engine_LTLIBRARIES = libclearlooks.la
diff -Nurd -x'*~' gtk-engines-2.16.1.orig/engines/crux/Makefile.am gtk-engines-2.16.1/engines/crux/Makefile.am
--- gtk-engines-2.16.1.orig/engines/crux/Makefile.am	2008-10-20 11:54:24.000000000 -0400
+++ gtk-engines-2.16.1/engines/crux/Makefile.am	2008-11-09 12:15:02.000000000 -0500
@@ -1,5 +1,5 @@
-INCLUDES = $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS) \
-           -I$(top_srcdir)/engines/support
+INCLUDES = -I$(top_srcdir)/engines/support \
+           $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS)
 
 enginedir = $(libdir)/gtk-2.0/$(GTK_VERSION)/engines
 
diff -Nurd -x'*~' gtk-engines-2.16.1.orig/engines/glide/Makefile.am gtk-engines-2.16.1/engines/glide/Makefile.am
--- gtk-engines-2.16.1.orig/engines/glide/Makefile.am	2008-10-20 11:54:26.000000000 -0400
+++ gtk-engines-2.16.1/engines/glide/Makefile.am	2008-11-09 12:14:55.000000000 -0500
@@ -1,5 +1,5 @@
-INCLUDES = $(GTK_CFLAGS)  $(DEVELOPMENT_CFLAGS) \
-           -I$(top_srcdir)/engines/support
+INCLUDES = -I$(top_srcdir)/engines/support \
+           $(GTK_CFLAGS)  $(DEVELOPMENT_CFLAGS)
 
 enginedir = $(libdir)/gtk-2.0/$(GTK_VERSION)/engines
 
diff -Nurd -x'*~' gtk-engines-2.16.1.orig/engines/hc/Makefile.am gtk-engines-2.16.1/engines/hc/Makefile.am
--- gtk-engines-2.16.1.orig/engines/hc/Makefile.am	2008-10-20 11:54:26.000000000 -0400
+++ gtk-engines-2.16.1/engines/hc/Makefile.am	2008-11-09 12:23:49.000000000 -0500
@@ -1,5 +1,5 @@
-INCLUDES = $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS) \
-           -I$(top_srcdir)/engines/support
+INCLUDES = -I$(top_srcdir)/engines/support \
+           $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS)
 
 enginedir = $(libdir)/gtk-2.0/$(GTK_VERSION)/engines
 
diff -Nurd -x'*~' gtk-engines-2.16.1.orig/engines/industrial/Makefile.am gtk-engines-2.16.1/engines/industrial/Makefile.am
--- gtk-engines-2.16.1.orig/engines/industrial/Makefile.am	2008-10-20 11:54:27.000000000 -0400
+++ gtk-engines-2.16.1/engines/industrial/Makefile.am	2008-11-09 12:16:55.000000000 -0500
@@ -1,5 +1,5 @@
-INCLUDES = $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS) \
-	-I$(top_srcdir)/engines/support
+INCLUDES = -I$(top_srcdir)/engines/support \
+           $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS)
 
 enginedir = $(libdir)/gtk-2.0/$(GTK_VERSION)/engines
 
diff -Nurd -x'*~' gtk-engines-2.16.1.orig/engines/lua/Makefile.am gtk-engines-2.16.1/engines/lua/Makefile.am
--- gtk-engines-2.16.1.orig/engines/lua/Makefile.am	2008-10-20 11:54:25.000000000 -0400
+++ gtk-engines-2.16.1/engines/lua/Makefile.am	2008-11-09 12:14:25.000000000 -0500
@@ -1,9 +1,9 @@
 NULL =
 
 INCLUDES = \
-	$(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS) \
 	-I$(top_srcdir)/engines/support \
 	-I$(top_srcdir)/engines/lua/src/liblua \
+	$(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS) \
 	$(NULL)
 
 enginedir = $(libdir)/gtk-2.0/$(GTK_VERSION)/engines
diff -Nurd -x'*~' gtk-engines-2.16.1.orig/engines/mist/Makefile.am gtk-engines-2.16.1/engines/mist/Makefile.am
--- gtk-engines-2.16.1.orig/engines/mist/Makefile.am	2008-10-20 11:54:26.000000000 -0400
+++ gtk-engines-2.16.1/engines/mist/Makefile.am	2008-11-09 12:14:18.000000000 -0500
@@ -1,5 +1,5 @@
-INCLUDES = $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS) \
-           -I$(top_srcdir)/engines/support
+INCLUDES = -I$(top_srcdir)/engines/support \
+           $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS)
 
 enginedir = $(libdir)/gtk-2.0/$(GTK_VERSION)/engines
 
diff -Nurd -x'*~' gtk-engines-2.16.1.orig/engines/redmond/Makefile.am gtk-engines-2.16.1/engines/redmond/Makefile.am
--- gtk-engines-2.16.1.orig/engines/redmond/Makefile.am	2008-10-20 11:54:26.000000000 -0400
+++ gtk-engines-2.16.1/engines/redmond/Makefile.am	2008-11-09 12:14:09.000000000 -0500
@@ -1,5 +1,6 @@
-INCLUDES = $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS) \
-           -I$(top_srcdir)/engines/support
+INCLUDES = -I$(top_srcdir)/engines/support \
+           $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS)
+
 
 enginedir = $(libdir)/gtk-2.0/$(GTK_VERSION)/engines
 
diff -Nurd -x'*~' gtk-engines-2.16.1.orig/engines/thinice/Makefile.am gtk-engines-2.16.1/engines/thinice/Makefile.am
--- gtk-engines-2.16.1.orig/engines/thinice/Makefile.am	2008-10-20 11:54:25.000000000 -0400
+++ gtk-engines-2.16.1/engines/thinice/Makefile.am	2008-11-09 12:13:56.000000000 -0500
@@ -1,5 +1,5 @@
-INCLUDES = $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS) \
-           -I$(top_srcdir)/engines/support
+INCLUDES = -I$(top_srcdir)/engines/support \
+           $(GTK_CFLAGS) $(DEVELOPMENT_CFLAGS)
 
 enginedir = $(libdir)/gtk-2.0/$(GTK_VERSION)/engines
 
