diff -Naur gtkhtml-1.0.2.old/components/html-editor/Makefile.in gtkhtml-1.0.2.new/components/html-editor/Makefile.in
--- gtkhtml-1.0.2.old/components/html-editor/Makefile.in	Wed Mar 20 02:29:02 2002
+++ gtkhtml-1.0.2.new/components/html-editor/Makefile.in	Wed Jun 12 21:41:49 2002
@@ -153,13 +153,13 @@
 IDLS =  	Editor.idl
 
 
-SPELL_IDLS =  	Spell.idl
+SPELL_IDLS =  	SpellI.idl
 
 
 IDL_GENERATED =  	Editor.h			Editor-common.c		Editor-skels.c		Editor-stubs.c
 
 
-SPELL_IDL_GENERATED =  	Spell.h				Spell-common.c			Spell-skels.c			Spell-stubs.c
+SPELL_IDL_GENERATED =  	SpellI.h				SpellI-common.c			SpellI-skels.c			SpellI-stubs.c
 
 
 LDFLAGS = @LDFLAGS@ -dynamic
@@ -223,7 +223,7 @@
 libgnome_gtkhtml_editor_la_LIBADD = 
 @EDITOR_SHLIB_TRUE@libgnome_gtkhtml_editor_la_OBJECTS =  \
 @EDITOR_SHLIB_TRUE@Editor-common.lo Editor-skels.lo Editor-stubs.lo \
-@EDITOR_SHLIB_TRUE@Spell-common.lo Spell-skels.lo Spell-stubs.lo \
+@EDITOR_SHLIB_TRUE@SpellI-common.lo SpellI-skels.lo SpellI-stubs.lo \
 @EDITOR_SHLIB_TRUE@body.lo cell.lo control-data.lo dialog.lo \
 @EDITOR_SHLIB_TRUE@editor-control-factory.lo engine.lo image.lo link.lo \
 @EDITOR_SHLIB_TRUE@menubar.lo paragraph.lo persist-file-impl.lo \
@@ -239,8 +239,8 @@
 
 @EDITOR_SHLIB_FALSE@gnome_gtkhtml_editor_OBJECTS =  \
 @EDITOR_SHLIB_FALSE@Editor-common.$(OBJEXT) Editor-skels.$(OBJEXT) \
-@EDITOR_SHLIB_FALSE@Editor-stubs.$(OBJEXT) Spell-common.$(OBJEXT) \
-@EDITOR_SHLIB_FALSE@Spell-skels.$(OBJEXT) Spell-stubs.$(OBJEXT) \
+@EDITOR_SHLIB_FALSE@Editor-stubs.$(OBJEXT) SpellI-common.$(OBJEXT) \
+@EDITOR_SHLIB_FALSE@SpellI-skels.$(OBJEXT) SpellI-stubs.$(OBJEXT) \
 @EDITOR_SHLIB_FALSE@body.$(OBJEXT) cell.$(OBJEXT) \
 @EDITOR_SHLIB_FALSE@control-data.$(OBJEXT) dialog.$(OBJEXT) \
 @EDITOR_SHLIB_FALSE@editor-control-factory.$(OBJEXT) engine.$(OBJEXT) \
@@ -621,7 +621,7 @@
 
 
 Editor-common.o: Editor.h
-Spell-common.o: Spell.h
+SpellI-common.o: SpellI.h
 
 $(IDL_GENERATED): $(IDLS)
 	$(ORBIT_IDL) -I$(datadir)/idl `$(GNOME_CONFIG) --cflags idl`	\
@@ -629,7 +629,7 @@
 
 $(SPELL_IDL_GENERATED): $(SPELL_IDLS)
 	$(ORBIT_IDL) -I$(datadir)/idl `$(GNOME_CONFIG) --cflags idl`	\
-		-I$(srcdir) $(srcdir)/Spell.idl
+		-I$(srcdir) $(srcdir)/SpellI.idl
 
 @XML_I18N_MERGE_OAF_RULE@
 
diff -Naur gtkhtml-1.0.2.old/components/html-editor/Spell-common.c gtkhtml-1.0.2.new/components/html-editor/Spell-common.c
--- gtkhtml-1.0.2.old/components/html-editor/Spell-common.c	Tue Mar 19 02:02:52 2002
+++ gtkhtml-1.0.2.new/components/html-editor/Spell-common.c	Wed Jun 12 21:41:49 2002
@@ -3,7 +3,7 @@
  */
 
 #include <string.h>
-#include "Spell.h"
+#include "SpellI.h"
 
 #if ( (TC_IMPL_TC_CORBA_sequence_CORBA_string_0 == 'S') \
 && (TC_IMPL_TC_CORBA_sequence_CORBA_string_1 == 'p') \
diff -Naur gtkhtml-1.0.2.old/components/html-editor/Spell-skels.c gtkhtml-1.0.2.new/components/html-editor/Spell-skels.c
--- gtkhtml-1.0.2.old/components/html-editor/Spell-skels.c	Tue Mar 19 02:02:52 2002
+++ gtkhtml-1.0.2.new/components/html-editor/Spell-skels.c	Wed Jun 12 21:41:49 2002
@@ -3,7 +3,7 @@
  */
 
 #include <string.h>
-#include "Spell.h"
+#include "SpellI.h"
 
 void
 _ORBIT_GNOME_Spell_Dictionary_Error_marshal(GIOPSendBuffer *
diff -Naur gtkhtml-1.0.2.old/components/html-editor/Spell-stubs.c gtkhtml-1.0.2.new/components/html-editor/Spell-stubs.c
--- gtkhtml-1.0.2.old/components/html-editor/Spell-stubs.c	Tue Mar 19 02:02:52 2002
+++ gtkhtml-1.0.2.new/components/html-editor/Spell-stubs.c	Wed Jun 12 21:41:49 2002
@@ -3,7 +3,7 @@
  */
 
 #include <string.h>
-#include "Spell.h"
+#include "SpellI.h"
 
 void
 _ORBIT_GNOME_Spell_Dictionary_Error_demarshal(GIOPRecvBuffer *
diff -Naur gtkhtml-1.0.2.old/components/html-editor/control-data.h gtkhtml-1.0.2.new/components/html-editor/control-data.h
--- gtkhtml-1.0.2.old/components/html-editor/control-data.h	Tue Mar 19 01:56:08 2002
+++ gtkhtml-1.0.2.new/components/html-editor/control-data.h	Wed Jun 12 21:41:49 2002
@@ -37,7 +37,7 @@
 #include "link.h"
 #include "rule.h"
 #include "engine.h"
-#include "Spell.h"
+#include "SpellI.h"
 
 struct _GtkHTMLControlData {
 	GtkHTML    *html;
diff -Naur gtkhtml-1.0.2.old/components/http-storage-module/Makefile.in gtkhtml-1.0.2.new/components/http-storage-module/Makefile.in
--- gtkhtml-1.0.2.old/components/http-storage-module/Makefile.in	Wed Mar 20 02:29:04 2002
+++ gtkhtml-1.0.2.new/components/http-storage-module/Makefile.in	Wed Jun 12 21:41:49 2002
@@ -145,7 +145,7 @@
 INCLUDES =          -DGNOMELOCALEDIR=\""$(datadir)/locale"\"        	-I$(srcdir) -I$(top_srcdir) 				-I$(top_builddir)					-DPLUGIN_DIR=\""$(PLUGIN_DIR)"\"		        -I$(includedir)                                 	$(EXTRA_GNOME_CFLAGS)
 
 
-common_ldflags = -avoid-version
+common_ldflags = -module -avoid-version
 
 bonobo_plugindir = $(PLUGIN_DIR)
 bonobo_plugin_LTLIBRARIES = libstorage_http.la
diff -Naur gtkhtml-1.0.2.old/configure gtkhtml-1.0.2.new/configure
--- gtkhtml-1.0.2.old/configure	Wed Mar 20 02:28:29 2002
+++ gtkhtml-1.0.2.new/configure	Wed Jun 12 21:42:10 2002
@@ -5042,7 +5042,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test x$module = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
@@ -9483,7 +9483,7 @@
 echo "${ECHO_T}found" >&6
 
 		GLIBWWW_CFLAGS="`gnome-config --cflags glibwww`"
-		GLIBWWW_LIBS="`gnome-config --libs glibwww` -lwwwhtml -lmd5"
+		GLIBWWW_LIBS="`gnome-config --libs glibwww` `libwww-config --libs`"
 
 		have_glibwww=true
 	else
