diff -Naur Guppi-0.40.3.old/configure Guppi-0.40.3/configure
--- Guppi-0.40.3.old/configure	Tue Jan 22 13:15:34 2002
+++ Guppi-0.40.3/configure	Wed Jul 10 22:21:50 2002
@@ -4784,11 +4786,11 @@
     ;;
 
   darwin* | rhapsody*)
-    allow_undefined_flag='-undefined suppress'
+    allow_undefined_flag='-undefined dynamic_lookup'
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linkopts -install_name $rpath/$soname $(test -n "$verstring" -a x$verstring != x0.0 && echo $verstring)'
+    archive_cmds='$CC $(test .$module = .yes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linkopts $(test .$module != .yes && echo -install_name $rpath/$soname $tmp_verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -Naur Guppi-0.40.3.old/ltmain.sh Guppi-0.40.3/ltmain.sh
--- Guppi-0.40.3.old/ltmain.sh	Tue Jan 22 13:15:19 2002
+++ Guppi-0.40.3/ltmain.sh	Wed Jul 10 22:16:10 2002
@@ -2889,8 +2889,14 @@
 
 	# Do each of the archive commands.
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
-	  eval cmds=\"$archive_expsym_cmds\"
+	  
+eval cmds=\"$archive_expsym_cmds\"
 	else
+         if test "x$verstring" = "x0.0"; then
+           tmp_verstring=
+         else
+           tmp_verstring="$verstring"
+         fi
 	  eval cmds=\"$archive_cmds\"
 	fi
 	IFS="${IFS= 	}"; save_ifs="$IFS"; IFS='~'
diff -Naur Guppi-0.40.3.old/src/libguppispecfns/const.c Guppi-0.40.3/src/libguppispecfns/const.c
--- Guppi-0.40.3.old/src/libguppispecfns/const.c	Sun May  6 03:16:23 2001
+++ Guppi-0.40.3/src/libguppispecfns/const.c	Wed Jul 10 22:17:10 2002
@@ -62,6 +62,10 @@
 #include "mconf.h"
 #include <gnan.h>
 
+#ifdef INFINITY
+#undef INFINITY
+#endif
+
 #ifdef UNK
 #if 1
 double MACHEP = 1.11022302462515654042E-16;	/* 2**-53 */
