diff -Nurd -x'*~' libgksu-2.0.9.orig/libgksu/Makefile.am libgksu-2.0.9/libgksu/Makefile.am
--- libgksu-2.0.9.orig/libgksu/Makefile.am	2008-05-09 13:39:55.000000000 -0400
+++ libgksu-2.0.9/libgksu/Makefile.am	2009-06-25 10:32:59.000000000 -0400
@@ -9,7 +9,7 @@
 # minor -> keeps compatibility (additions to the API)
 # micro -> no change to the API/ABI
 libgksu2_la_LIBADD = ../libgksuui/libgksuui1.0.la
-libgksu2_la_LDFLAGS = -version-info 0:2:0 -Wl,-O1 -lutil ${LIBGKSU_LIBS}
+libgksu2_la_LDFLAGS = -version-info 0:2:0 ${LIBGKSU_LIBS}
 if USE_VERSION_SCRIPT
 libgksu2_la_LDFLAGS += -Wl,--version-script=libgksu.ver
 endif
diff -Nurd -x'*~' libgksu-2.0.9.orig/libgksu/Makefile.in libgksu-2.0.9/libgksu/Makefile.in
--- libgksu-2.0.9.orig/libgksu/Makefile.in	2009-03-04 13:56:14.000000000 -0500
+++ libgksu-2.0.9/libgksu/Makefile.in	2009-06-25 04:49:13.000000000 -0400
@@ -276,7 +276,7 @@
 # minor -> keeps compatibility (additions to the API)
 # micro -> no change to the API/ABI
 libgksu2_la_LIBADD = ../libgksuui/libgksuui1.0.la
-libgksu2_la_LDFLAGS = -version-info 0:2:0 -Wl,-O1 -lutil \
+libgksu2_la_LDFLAGS = -version-info 0:2:0 \
 	${LIBGKSU_LIBS} $(am__append_1)
 noinst_HEADERS = defines.h
 include_HEADERS = gksu.h libgksu.h
diff -Nurd -x'*~' libgksu-2.0.9.orig/libgksu/libgksu.c libgksu-2.0.9/libgksu/libgksu.c
--- libgksu-2.0.9.orig/libgksu/libgksu.c	2009-03-04 13:53:02.000000000 -0500
+++ libgksu-2.0.9/libgksu/libgksu.c	2009-06-25 11:33:30.000000000 -0400
@@ -23,7 +23,7 @@
 #include <unistd.h>
 #include <string.h>
 #include <fcntl.h>
-#include <pty.h>
+#include <util.h>
 #include <pwd.h>
 #include <sys/types.h>
 #include <sys/wait.h>
@@ -1932,7 +1932,7 @@
   GQuark gksu_quark;
   int i = 0;
 
-  gchar auxcommand[] = PREFIX "/lib/" PACKAGE "/gksu-run-helper";
+  gchar auxcommand[] = PREFIX "/lib/" PACKAGE "/gksu-run-helper-2";
 
   int fdpty;
   pid_t pid;
@@ -1976,7 +1976,7 @@
       gchar **cmd = g_malloc (sizeof(gchar*)*7);
 
       setsid();   // make us session leader
-      cmd[i] = g_strdup ("/bin/su"); i++;
+      cmd[i] = g_strdup ("/usr/bin/su"); i++;
       if (context->login_shell)
 	{
 	  cmd[i] = g_strdup ("-"); i++;
@@ -1998,7 +1998,7 @@
       if (execv (cmd[0], cmd) == -1)
 	{
 	  fprintf (stderr,
-		   "Unable to run /bin/su: %s",
+		   "Unable to run /usr/bin/su: %s",
 		   strerror(errno));
 	}
 
diff -Nurd -x'*~' libgksu-2.0.9.orig/libgksuui/Makefile.am libgksu-2.0.9/libgksuui/Makefile.am
--- libgksu-2.0.9.orig/libgksuui/Makefile.am	2008-05-09 13:39:54.000000000 -0400
+++ libgksu-2.0.9/libgksuui/Makefile.am	2009-06-25 10:33:15.000000000 -0400
@@ -4,7 +4,7 @@
 
 noinst_LTLIBRARIES = libgksuui1.0.la
 libgksuui1_0_la_SOURCES = gksuui-dialog.c
-libgksuui1_0_la_LDFLAGS = -Wl,-O1 `pkg-config --libs gtk+-2.0 gconf-2.0`
+libgksuui1_0_la_LDFLAGS = `pkg-config --libs gtk+-2.0 gconf-2.0`
 
 noinst_HEADERS = defines.h gksuui.h gksuui-dialog.h
 includedir = ${prefix}/include/$(PACKAGE)
diff -Nurd -x'*~' libgksu-2.0.9.orig/libgksuui/Makefile.in libgksu-2.0.9/libgksuui/Makefile.in
--- libgksu-2.0.9.orig/libgksuui/Makefile.in	2009-03-04 13:56:14.000000000 -0500
+++ libgksu-2.0.9/libgksuui/Makefile.in	2009-06-25 04:49:14.000000000 -0400
@@ -246,7 +246,7 @@
 AM_CPPFLAGS = -DLOCALEDIR=\"$(datadir)/locale\" -DDATA_DIR=\"$(datadir)\" -DPREFIX=\"$(prefix)\"
 noinst_LTLIBRARIES = libgksuui1.0.la
 libgksuui1_0_la_SOURCES = gksuui-dialog.c
-libgksuui1_0_la_LDFLAGS = -Wl,-O1 `pkg-config --libs gtk+-2.0 gconf-2.0`
+libgksuui1_0_la_LDFLAGS = `pkg-config --libs gtk+-2.0 gconf-2.0`
 noinst_HEADERS = defines.h gksuui.h gksuui-dialog.h
 test_gksuui_SOURCES = test-gksuui.c
 test_gksuui_LDADD = libgksuui1.0.la
