diff -Nurd -x'*~' libgnomeprintui-2.18.1.orig/tests/Makefile.in libgnomeprintui-2.18.1/tests/Makefile.in
--- libgnomeprintui-2.18.1.orig/tests/Makefile.in	2007-09-17 09:35:44.000000000 -0400
+++ libgnomeprintui-2.18.1/tests/Makefile.in	2007-11-18 17:26:26.000000000 -0500
@@ -270,8 +270,7 @@
 printui_libs = \
 	$(top_builddir)/libgnomeprintui/libgnomeprintui-2-2.la \
 	$(libgnomeprint_local) \
-	$(LIBGNOMEPRINTUI_LIBS) \
-	-lpopt
+	$(LIBGNOMEPRINTUI_LIBS)
 
 test_dialog_SOURCES = test-dialog.c test-common.c test-common.h
 test_dialog_LDFLAGS = 
