diff -Nurd -x'*~' libglade-2.5.1.orig/configure.in libglade-2.5.1/configure.in
--- libglade-2.5.1.orig/configure.in	2005-02-11 08:10:05.000000000 -0500
+++ libglade-2.5.1/configure.in	2005-06-08 04:00:23.000000000 -0400
@@ -34,10 +34,6 @@
 AC_ISC_POSIX
 AC_HEADER_STDC
 
-dnl XXXX hack to kill off all the libtool tags ...
-dnl it isn't like we are using C++ or Fortran.
-m4_define([_LT_AC_TAGCONFIG],[])
-
 AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 
diff -ruN libglade-2.3.6/glade/glade-gtk.c libglade-2.3.6-patched/glade/glade-gtk.c
--- libglade-2.3.6/glade/glade-gtk.c	2004-03-10 00:13:29.000000000 -0800
+++ libglade-2.3.6-patched/glade/glade-gtk.c	2007-07-08 13:43:59.000000000 -0700
@@ -26,6 +26,7 @@
 #include <string.h>
 #include <stdlib.h>
 
+#define G_IMPLEMENT_INLINES
 /* for GtkText et all */
 #ifndef GTK_ENABLE_BROKEN
 #define GTK_ENABLE_BROKEN
diff -ruN libglade-2.3.6/glade/glade-parser.c libglade-2.3.6-patched/glade/glade-parser.c
--- libglade-2.3.6/glade/glade-parser.c	2004-02-03 01:40:37.000000000 -0800
+++ libglade-2.3.6-patched/glade/glade-parser.c	2007-07-08 13:43:32.000000000 -0700
@@ -36,6 +36,7 @@
 
 #include <libxml/parser.h>
 
+#define G_IMPLEMENT_INLINES
 #include "glade-parser.h"
 #include "glade-private.h"
 
diff -ruN libglade-2.3.6/glade/glade-xml.c libglade-2.3.6-patched/glade/glade-xml.c
--- libglade-2.3.6/glade/glade-xml.c	2004-02-03 01:40:38.000000000 -0800
+++ libglade-2.3.6-patched/glade/glade-xml.c	2007-07-08 13:43:32.000000000 -0700
@@ -25,6 +25,7 @@
 
 #include <stdlib.h>
 #include <string.h>
+#define G_IMPLEMENT_INLINES
 #include <glade/glade-xml.h>
 #include <glade/glade-init.h>
 #include <glade/glade-build.h>
