diff -Naur libIDL-0.8.2-old/lexer.l libIDL-0.8.2/lexer.l
--- libIDL-0.8.2-old/lexer.l	Fri May  9 11:19:06 2003
+++ libIDL-0.8.2/lexer.l	Tue Oct 14 08:46:53 2003
@@ -225,18 +225,30 @@
 <*>{cpp_other}						;
 <*>{whitespace}						;
 {b8_int}						{
+#if defined(__APPLE__)
+	yylval.integer = strtoq (yytext, NULL, 8);
+#else
 	yylval.integer = 0;
 	sscanf (yytext, "%" IDL_LL "o", &yylval.integer);
+#endif
 	tokreturn (TOK_INTEGER);
 }
 {b10_uint}						{
+#if defined(__APPLE__)
+	yylval.integer = strtoq (yytext, NULL, 10);
+#else
 	yylval.integer = 0;
 	sscanf (yytext, "%" IDL_LL "u", &yylval.integer);
+#endif
 	tokreturn (TOK_INTEGER);
 }
 {b16_int}						{
+#if defined(__APPLE__)
+	yylval.integer = strtoq (yytext + 2, NULL, 16);
+#else
 	yylval.integer = 0;
 	sscanf (yytext + 2, "%" IDL_LL "x", &yylval.integer);
+#endif
 	tokreturn (TOK_INTEGER);
 }
 {fixed_lit}						{
diff -Naur libIDL-0.8.2-old/ltmain.sh libIDL-0.8.2/ltmain.sh
--- libIDL-0.8.2-old/ltmain.sh	Mon May 26 13:12:13 2003
+++ libIDL-0.8.2/ltmain.sh	Tue Oct 14 08:55:44 2003
@@ -2982,6 +2982,11 @@
 	    done
 	    deplibs="$tmp_deplibs"
 	  done
+	  if test "x$verstring" = "x0.0"; then
+            tmp_verstring=
+          else
+            tmp_verstring="$verstring"
+          fi
 	  eval cmds=\"$archive_cmds\"
 	  deplibs="$save_deplibs"
 	fi
