diff -Nurd -x'*~' librsvg-2.22.3.orig/Makefile.am librsvg-2.22.3/Makefile.am
--- librsvg-2.22.3.orig/Makefile.am	2008-09-22 20:37:57.000000000 -0400
+++ librsvg-2.22.3/Makefile.am	2008-10-26 22:52:52.000000000 -0400
@@ -70,7 +70,7 @@
 	rsvg-file-util.c
 
 librsvg_2_la_LDFLAGS = -version-info @VERSION_INFO@ -export-dynamic -no-undefined -export-symbols $(srcdir)/librsvg.def
-librsvg_2_la_LIBADD = $(LIBGSF_LIBS) $(LIBCROCO_LIBS) $(LIBRSVG_LIBS) $(FREETYPE_LIBS)
+librsvg_2_la_LIBADD = $(LIBGSF_LIBS) $(LIBCROCO_LIBS) $(LIBRSVG_LIBS)
 
 librsvgincdir = $(includedir)/librsvg-2/librsvg
 librsvginc_HEADERS = 	\
@@ -79,10 +79,10 @@
 	librsvg-enum-types.h
 
 INCLUDES =				\
-	$(LIBGSF_CFLAGS)		\
-	$(LIBCROCO_CFLAGS)		\
 	-I$(top_srcdir) 		\
 	-I$(top_builddir) 		\
+	$(LIBGSF_CFLAGS)		\
+	$(LIBCROCO_CFLAGS)		\
 	$(LIBRSVG_CFLAGS) 		\
 	$(GTK_CFLAGS)			\
 	$(GDK_X11_CFLAGS)		\
diff -Nurd -x'*~' librsvg-2.22.3.orig/gdk-pixbuf-loader/Makefile.am librsvg-2.22.3/gdk-pixbuf-loader/Makefile.am
--- librsvg-2.22.3.orig/gdk-pixbuf-loader/Makefile.am	2008-09-22 20:37:51.000000000 -0400
+++ librsvg-2.22.3/gdk-pixbuf-loader/Makefile.am	2008-10-26 22:59:53.000000000 -0400
@@ -6,11 +6,11 @@
 endif
 
 INCLUDES = \
+	-I$(top_srcdir) \
 	$(LIBRSVG_CFLAGS) \
 	$(LIBGSF_CFLAGS) \
 	$(LIBCROCO_CFLAGS)\
-	-DGDK_PIXBUF_ENABLE_BACKEND \
-	-I$(top_srcdir)
+	-DGDK_PIXBUF_ENABLE_BACKEND
 
 if ENABLE_PIXBUF_LOADER
 target_pixbuf_loader = svg_loader.la 
@@ -23,7 +23,6 @@
 svg_loader_la_LIBADD = \
 	$(top_builddir)/librsvg-2.la	\
 	$(LIBRSVG_LIBS) 	\
-	$(LIBGNOME_VFS_LIBS)		\
 	$(LIBGSF_LIBS) $(LIBCROCO_LIBS)
 
 loader_LTLIBRARIES = $(target_pixbuf_loader)
diff -Nurd -x'*~' librsvg-2.22.3.orig/gtk-engine/Makefile.am librsvg-2.22.3/gtk-engine/Makefile.am
--- librsvg-2.22.3.orig/gtk-engine/Makefile.am	2008-09-22 20:37:57.000000000 -0400
+++ librsvg-2.22.3/gtk-engine/Makefile.am	2008-10-26 22:59:50.000000000 -0400
@@ -4,7 +4,7 @@
 no_undefined = -no-undefined
 endif
 
-INCLUDES = $(GTK_CFLAGS) $(LIBGSF_CFLAGS) $(LIBCROCO_CFLAGS) $(LIBRSVG_CFLAGS) -I$(top_srcdir)
+INCLUDES = -I$(top_srcdir) $(GTK_CFLAGS) $(LIBGSF_CFLAGS) $(LIBCROCO_CFLAGS) $(LIBRSVG_CFLAGS)
 
 enginedir = $(libdir)/gtk-2.0/$(GTK_VERSION)/engines
 themedir = $(datadir)/themes
@@ -29,6 +29,6 @@
 	svg.h
 
 libsvg_la_LDFLAGS =  -avoid-version -module $(no_undefined)
-libsvg_la_LIBADD = $(top_builddir)/librsvg-2.la $(GTK_LIBS) $(LIBRSVG_LIBS) $(LIBGNOME_VFS_LIBS) $(LIBCROCO_LIBS)
+libsvg_la_LIBADD = $(top_builddir)/librsvg-2.la $(GTK_LIBS) $(LIBRSVG_LIBS) $(LIBCROCO_LIBS)
 
 EXTRA_DIST = $(libsvg_la_SOURCES)
diff -Nurd -x'*~' librsvg-2.22.3.orig/rsvg-base.c librsvg-2.22.3/rsvg-base.c
--- librsvg-2.22.3.orig/rsvg-base.c	2008-09-22 20:37:57.000000000 -0400
+++ librsvg-2.22.3/rsvg-base.c	2008-10-26 22:59:28.000000000 -0400
@@ -1605,10 +1605,6 @@
     return rsvg_handle_close_impl (handle, error);
 }
 
-#ifdef HAVE_GNOME_VFS
-#include <libgnomevfs/gnome-vfs.h>
-#endif
-
 /**
  * rsvg_init:
  *
