diff -Naur linc-1.0.3-old/include/linc/linc-protocol.h linc-1.0.3/include/linc/linc-protocol.h
--- linc-1.0.3-old/include/linc/linc-protocol.h	Fri Jun 20 07:31:21 2003
+++ linc-1.0.3/include/linc/linc-protocol.h	Tue Oct 14 10:20:14 2003
@@ -14,11 +14,11 @@
 
 #include <glib/gmacros.h>
 
+#include <sys/socket.h>
 G_BEGIN_DECLS
 
 #include <linc/linc-types.h>
 #include <sys/types.h>
-#include <sys/socket.h>
 #include <netdb.h>
 
 /* socklen_t seems rather un-portable */
diff -Nurd -x'*~' linc-1.0.3.orig/linc.m4 linc-1.0.3/linc.m4
--- linc-1.0.3.orig/linc.m4	2001-03-14 00:12:41.000000000 -0500
+++ linc-1.0.3/linc.m4	2005-11-11 01:12:04.000000000 -0500
@@ -1,4 +1,4 @@
-AC_DEFUN(AM_PATH_LINC,
+AC_DEFUN([AM_PATH_LINC],
 [dnl
 dnl Get the cflags and libs from the linc-config script
 dnl
diff -Naur linc-1.0.3-old/src/linc-protocols.c linc-1.0.3/src/linc-protocols.c
--- linc-1.0.3-old/src/linc-protocols.c	Fri Jun 20 07:31:13 2003
+++ linc-1.0.3/src/linc-protocols.c	Tue Oct 14 10:20:14 2003
@@ -11,6 +11,7 @@
  */
 #include <config.h>
 #include "linc-compat.h"
+#include <sys/types.h>
 #include <dirent.h>
 #include <linc/linc-protocol.h>
 #include <linc/linc-connection.h>
@@ -609,7 +610,7 @@
 /* FIXME: is IN6ADDR_ANY_INIT exported on Mac OS X ? */
 /* on Mac OS X 10.1 inaddr6_any isn't exported by libc */
 #ifndef in6addr_any
-	static const struct in6_addr in6addr_any = { { { 0 } } };
+	const struct in6_addr in6addr_any = { { { 0 } } };
 #endif
 
 static gboolean
