diff -Naur linc-1.0.3-old/Makefile.in linc-1.0.3/Makefile.in
--- linc-1.0.3-old/Makefile.in	Fri Jun 20 08:27:27 2003
+++ linc-1.0.3/Makefile.in	Tue Oct 14 10:22:38 2003
@@ -98,7 +98,7 @@
 VERSION = @VERSION@
 WARN_CFLAGS = @WARN_CFLAGS@
 
-SUBDIRS = include src test docs
+SUBDIRS = include src docs
 
 m4datadir = $(datadir)/aclocal
 m4data_DATA = linc.m4
diff -Naur linc-1.0.3-old/config.h.in linc-1.0.3/config.h.in
--- linc-1.0.3-old/config.h.in	Fri Jun 20 08:24:38 2003
+++ linc-1.0.3/config.h.in	Tue Oct 14 10:20:14 2003
@@ -10,7 +10,9 @@
  * Copyright 2001, Red Hat, Inc., Ximian, Inc.,
  *                 Sun Microsystems, Inc.
  */
+#if !defined (__APPLE__)
 #undef socklen_t
+#endif
 #undef HAVE_SOCKADDR_SA_LEN
 #undef HAVE_INET_PTON
 #undef HAVE_GETADDRINFO
diff -Naur linc-1.0.3-old/include/linc/linc-protocol.h linc-1.0.3/include/linc/linc-protocol.h
--- linc-1.0.3-old/include/linc/linc-protocol.h	Fri Jun 20 07:31:21 2003
+++ linc-1.0.3/include/linc/linc-protocol.h	Tue Oct 14 10:20:14 2003
@@ -14,6 +14,10 @@
 
 #include <glib/gmacros.h>
 
+#if defined(__APPLE__) && !defined(socklen_t)
+#  define socklen_t int
+#endif
+
 G_BEGIN_DECLS
 
 #include <linc/linc-types.h>
diff -Naur linc-1.0.3-old/src/linc-protocols.c linc-1.0.3/src/linc-protocols.c
--- linc-1.0.3-old/src/linc-protocols.c	Fri Jun 20 07:31:13 2003
+++ linc-1.0.3/src/linc-protocols.c	Tue Oct 14 10:20:14 2003
@@ -11,12 +11,17 @@
  */
 #include <config.h>
 #include "linc-compat.h"
+#include <sys/types.h>
 #include <dirent.h>
 #include <linc/linc-protocol.h>
 #include <linc/linc-connection.h>
 
 #include "linc-private.h"
 #include "linc-debug.h"
+
+#if defined(__APPLE__) && !defined(RES_USE_INET6)
+#  undef AF_INET6
+#endif
 
 #undef LOCAL_DEBUG
 
