diff -Nurd -x'*~' pygtksourceview-2.6.0.orig/configure.ac pygtksourceview-2.6.0/configure.ac
--- pygtksourceview-2.6.0.orig/configure.ac	2009-03-21 16:06:40.000000000 -0400
+++ pygtksourceview-2.6.0/configure.ac	2009-06-08 21:23:55.000000000 -0400
@@ -54,6 +54,8 @@
 AC_DISABLE_STATIC
 AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
+dnl when using libtool 2.x create libtool early, because it's used in configure
+m4_ifdef([LT_OUTPUT], [LT_OUTPUT])
 
 if test "x$os_win32" = xyes -a "x$cross_compiling" = xyes; then
 MOO_AM_PYTHON_DEVEL_CROSS_MINGW(,[AC_MSG_ERROR(could not find Python headers)])
