diff -Nurd '-x*~' rhythmbox-0.11.2.orig/data/Makefile.in rhythmbox-0.11.2/data/Makefile.in
--- rhythmbox-0.11.2.orig/data/Makefile.in	2007-08-15 08:20:34.000000000 -0400
+++ rhythmbox-0.11.2/data/Makefile.in	2012-02-19 23:13:17.000000000 -0500
@@ -660,7 +660,7 @@
 install-data-am: install-MultimediaDATA install-data-local \
 	install-playlistsDATA install-schemaDATA install-serviceDATA
 	@$(NORMAL_INSTALL)
-	$(MAKE) $(AM_MAKEFLAGS) install-data-hook
+	$(MAKE) $(AM_MAKEFLAGS)
 
 install-dvi: install-dvi-recursive
 
diff -Nurd '-x*~' rhythmbox-0.11.2.orig/plugins/cd-recorder/rb-recorder-gst.c rhythmbox-0.11.2/plugins/cd-recorder/rb-recorder-gst.c
--- rhythmbox-0.11.2.orig/plugins/cd-recorder/rb-recorder-gst.c	2007-06-02 21:55:38.000000000 -0400
+++ rhythmbox-0.11.2/plugins/cd-recorder/rb-recorder-gst.c	2012-02-19 23:13:17.000000000 -0500
@@ -26,7 +26,9 @@
 #include <string.h>
 #include <math.h>
 #include <time.h>
+#ifndef __APPLE__
 #include <sys/vfs.h>
+#endif
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
diff -Nurd '-x*~' rhythmbox-0.11.2.orig/plugins/daap/rb-daap-src.c rhythmbox-0.11.2/plugins/daap/rb-daap-src.c
--- rhythmbox-0.11.2.orig/plugins/daap/rb-daap-src.c	2007-06-17 07:56:21.000000000 -0400
+++ rhythmbox-0.11.2/plugins/daap/rb-daap-src.c	2012-02-19 23:13:17.000000000 -0500
@@ -53,6 +53,10 @@
 #endif
 
 
+#ifdef __APPLE__
+#define MSG_NOSIGNAL SO_NOSIGPIPE
+#endif
+
 #define RB_TYPE_DAAP_SRC (rb_daap_src_get_type())
 #define RB_DAAP_SRC(obj) (G_TYPE_CHECK_INSTANCE_CAST((obj),RB_TYPE_DAAP_SRC,RBDAAPSrc))
 #define RB_DAAP_SRC_CLASS(klass) (G_TYPE_CHECK_CLASS_CAST((klass),RB_TYPE_DAAP_SRC,RBDAAPSrcClass))
diff -Nurd '-x*~' rhythmbox-0.11.2.orig/shell/Makefile.in rhythmbox-0.11.2/shell/Makefile.in
--- rhythmbox-0.11.2.orig/shell/Makefile.in	2007-08-15 08:20:41.000000000 -0400
+++ rhythmbox-0.11.2/shell/Makefile.in	2012-02-19 23:14:38.000000000 -0500
@@ -54,7 +54,7 @@
 CONFIG_HEADER = $(top_builddir)/config.h
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(noinst_LTLIBRARIES)
-librbshell_la_LIBADD =
+librbshell_la_LIBADD = -L/usr/X11R6/lib -lX11
 am_librbshell_la_OBJECTS = rb-shell.lo rb-shell-player.lo \
 	rb-source-header.lo rb-statusbar.lo rb-shell-preferences.lo \
 	rb-shell-clipboard.lo rb-playlist-manager.lo \
