diff -Nurd '-x*~' viking-0.9.9.orig/src/Makefile.in viking-0.9.9/src/Makefile.in
--- viking-0.9.9.orig/src/Makefile.in	2009-09-09 16:36:14.000000000 -0400
+++ viking-0.9.9/src/Makefile.in	2012-02-19 15:52:08.000000000 -0500
@@ -337,7 +337,7 @@
 	$(am__append_1) $(am__append_2) $(am__append_3) \
 	$(am__append_4) $(am__append_5)
 viking_SOURCES = main.c
-LDADD = libviking.a $(PACKAGE_LIBS) @EXPAT_LIBS@ @LIBCURL@ \
+LDADD = libviking.a -lz $(PACKAGE_LIBS) @EXPAT_LIBS@ @LIBCURL@ \
 	icons/libicons.a $(am__append_6)
 AM_CFLAGS = -Wall -g -D_GNU_SOURCE \
 	$(PACKAGE_CFLAGS) \
