diff -Naur zenity-2.6.2-old/src/Makefile.in zenity-2.6.2/src/Makefile.in
--- zenity-2.6.2-old/src/Makefile.in	2004-04-28 01:39:13.000000000 -0400
+++ zenity-2.6.2/src/Makefile.in	2004-06-18 14:05:23.000000000 -0400
@@ -174,7 +174,7 @@
 
 
 zenity_LDADD = \
-	$(ZENITY_LIBS)
+	-lX11 $(ZENITY_LIBS)
 
 
 gladedir = $(datadir)/zenity
@@ -205,7 +205,7 @@
 	util.$(OBJEXT)
 zenity_OBJECTS = $(am_zenity_OBJECTS)
 zenity_DEPENDENCIES =
-zenity_LDFLAGS =
+zenity_LDFLAGS = -L/usr/X11R6/lib
 SCRIPTS = $(bin_SCRIPTS)
 
 
