diff -ruN mozilla-release-orig/.mozconfig mozilla-release/.mozconfig
--- mozilla-release-orig/.mozconfig	1969-12-31 19:00:00.000000000 -0500
+++ mozilla-release/.mozconfig	2011-08-16 10:32:15.000000000 -0400
@@ -0,0 +1,43 @@
+# sh
+# Build configuration script
+#
+# See https://developer.mozilla.org/en/Configuring_Build_Options for build options.
+#
+
+. $topsrcdir/browser/config/mozconfig
+
+# Options for client.mk.
+# environment variables to set
+
+mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/obj-@FINK_ARCH_TARGET@-apple-darwin`uname -r`
+mk_add_options AUTOCONF=@FINKPREFIX@/lib/autoconf2.13/bin/autoconf
+ac_add_options --target=@FINK_ARCH_TARGET@-apple-darwin`uname -r`
+
+# Options for 'configure' (same as command-line options).
+ac_add_options --prefix=@FINKPREFIX@
+ac_add_options --enable-official-branding
+ac_add_options --with-distribution-id=org.fink
+ac_add_options --enable-default-toolkit=cairo-gtk2
+ac_add_options --x-libraries=/usr/X11R6/lib
+ac_add_options --x-includes=/usr/X11R6/include
+ac_add_options --disable-gnomevfs
+ac_add_options --disable-gnomeui
+ac_add_options --enable-extensions=default,-gnomevfs
+
+ac_add_options --disable-dependency-tracking
+ac_add_options --disable-crashreporter
+#ac_add_options --disable-ipc
+#ac_add_options --enable-necko-wifi
+ac_add_options --disable-libnotify
+
+ac_add_options --disable-debug
+ac_add_options --disable-tests
+ac_add_options --enable-strip
+#ac_add_options --with-default-mozilla-five-home=@FINKPREFIX@/lib/firefox@FIREFOX_MAJOR_VERSION@
+ac_add_options --with-system-jpeg=@FINKPREFIX@/lib/libjpeg8-turbo
+ac_add_options --disable-system-cairo
+ac_add_options --enable-system-sqlite
+#ac_add_options --with-system-nspr
+#ac_add_options --with-nspr-prefix=@FINKPREFIX@
+#ac_add_options --with-system-nss
+#ac_add_options --with-nss-prefix=@FINKPREFIX@
diff -ruN mozilla-release-orig/browser/locales/generic/profile/bookmarks.html.in mozilla-release/browser/locales/generic/profile/bookmarks.html.in
--- mozilla-release-orig/browser/locales/generic/profile/bookmarks.html.in	2009-04-23 21:46:37.000000000 -0400
+++ mozilla-release/browser/locales/generic/profile/bookmarks.html.in	2009-04-27 16:56:30.000000000 -0400
@@ -13,6 +13,7 @@
     <DL><p>
         <DT><A HREF="http://www.mozilla.com/@AB_CD@/firefox/central/" ID="rdf:#$GvPhC3">@getting_started@</A>
         <DT><A HREF="http://fxfeeds.mozilla.com/@AB_CD@/firefox/livebookmarks/" FEEDURL="http://fxfeeds.mozilla.com/@AB_CD@/firefox/headlines.xml" ID="rdf:#$HvPhC3">@latest_headlines@</A>
+        <DT><A HREF="http://www.finkproject.org/" ICON_URI="http://www.finkproject.org/favicon.ico" ICON="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAA+0lEQVQ4jWNgGDLg/6sTqv9OVe75uzv43J+TpUv/MzAwEqfxfEfU3y3O///u9P//d5v7/7/HCir/vz8n//9ir9+fffF+eDX/O5yxCqLR4///28u0GRgYGP6uN/sPN/zNBWlMG9+e4Pt7tq0bbutO//9/t3v9///qFQ9e2/7uCZ3/b3fwvX9HMg//PZpz7e9O//9/9ye8+Lc34ubf7V7/f1+e4Irdj3dWVf//f5UNzn91VeLvzoD/f0+U1OK1EadLtrr8/3+uoYcszf+PZHX/OxB/jizNDAwMDP92BX0hW/P/o1k7ydbMwMDA8P98D/ZQJkrz+6sJFNk+oAAAPwyICxl3w8YAAAAASUVORK5CYII=">Fink - Home</A>
     </DL><p>
     <DT><H3 ID="rdf:#$ZvPhC3">@firefox_heading@</H3>
     <DL><p>
diff -ruN mozilla-release-orig/config/autoconf.mk.in mozilla-release/config/autoconf.mk.in
--- mozilla-release-orig/config/autoconf.mk.in	2011-08-11 17:40:51.000000000 -0400
+++ mozilla-release/config/autoconf.mk.in	2011-08-16 09:15:05.000000000 -0400
@@ -58,24 +58,24 @@
 MOZ_UA_FIREFOX_VERSION = @FIREFOX_VERSION@
 MOZ_UA_BUILDID = @MOZ_UA_BUILDID@
 
 MOZ_PKG_SPECIAL = @MOZ_PKG_SPECIAL@
 
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-includedir	= @includedir@/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+includedir	= @includedir@/$(MOZ_APP_NAME)@FIREFOX_MAJOR_VERSION@
 libdir		= @libdir@
 datadir		= @datadir@
 mandir		= @mandir@
-idldir		= $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+idldir		= $(datadir)/idl/$(MOZ_APP_NAME)@FIREFOX_MAJOR_VERSION@
 
-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/$(MOZ_APP_NAME)@FIREFOX_MAJOR_VERSION@
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)@FIREFOX_MAJOR_VERSION@-devel
 
 DIST		= $(DEPTH)/dist
 LIBXUL_SDK      = @LIBXUL_SDK@
 
 MOZ_FS_LAYOUT = @MOZ_FS_LAYOUT@
 
 L10NBASEDIR     = @L10NBASEDIR@
 
diff -ruN mozilla-release-orig/configure.in mozilla-release/configure.in
--- mozilla-release-orig/configure.in	2011-09-02 16:15:20.000000000 -0400
+++ mozilla-release/configure.in	2011-11-01 17:06:31.000000000 -0400
@@ -1876,17 +1876,18 @@
     case "${host_cpu}" in
     x86_64)
         HOST_CFLAGS="$HOST_CFLAGS -D_AMD64_"
         ;;
     esac
     ;;
 
 *-darwin*)
-    HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX -DXP_MACOSX -DNO_X11"
+    HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX -DXP_MACOSX"
+    HOST_CXXFLAGS="$HOST_CFLAGS -DXP_UNIX -DXP_MACOSX"
     HOST_NSPR_MDCPUCFG='\"md/_darwin.cfg\"'
     HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-O3}"
     ;;
 
 *-linux*|*-kfreebsd*-gnu|*-gnu*)
     HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX"
     HOST_NSPR_MDCPUCFG='\"md/_linux.cfg\"'
     HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-O3}"
diff -ruN mozilla-release-orig/fink/applications/firefox.desktop mozilla-release/fink/applications/firefox.desktop
--- mozilla-release-orig/fink/applications/firefox.desktop	1969-12-31 19:00:00.000000000 -0500
+++ mozilla-release/fink/applications/firefox.desktop	2009-04-27 16:56:30.000000000 -0400
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Firefox Web Browser
+Comment=Browse the World Wide Web
+Exec=firefox
+Icon=firefox.png
+Terminal=false
+Type=Application
+Categories=Application;Network
+StartupNotify=true
diff -ruN -U 8 mozilla-release-orig/netwerk/wifi/nsWifiMonitor.h mozilla-release/netwerk/wifi/nsWifiMonitor.h
--- mozilla-release-orig/netwerk/wifi/nsWifiMonitor.h	2011-08-11 17:41:20.000000000 -0400
+++ mozilla-release/netwerk/wifi/nsWifiMonitor.h	2011-08-16 09:15:05.000000000 -0400
@@ -82,17 +82,17 @@
 
   nsWifiMonitor();
 
  private:
   ~nsWifiMonitor();
 
   nsresult DoScan();
 
-#if defined(XP_MACOSX)
+#if defined(__APPLE__)
   nsresult DoScanWithCoreWLAN();
   nsresult DoScanOld();
 #endif
 
   PRBool mKeepGoing;
   nsCOMPtr<nsIThread> mThread;
 
   nsTArray<nsWifiListener> mListeners;
diff -ruN -U 8 mozilla-release-orig/xpcom/io/nsAppFileLocationProvider.cpp mozilla-release/xpcom/io/nsAppFileLocationProvider.cpp
--- mozilla-release-orig/xpcom/io/nsAppFileLocationProvider.cpp	2011-08-11 17:41:33.000000000 -0400
+++ mozilla-release/xpcom/io/nsAppFileLocationProvider.cpp	2011-08-16 09:15:05.000000000 -0400
@@ -231,19 +231,19 @@
         rv = GetProductDirectory(getter_AddRefs(localFile));
         if (NS_SUCCEEDED(rv))
             rv = localFile->AppendRelativeNativePath(PLUGINS_DIR_NAME);
     }
 #ifdef XP_UNIX
     else if (nsCRT::strcmp(prop, NS_SYSTEM_PLUGINS_DIR) == 0) {
         static const char *const sysLPlgDir = 
 #if defined(HAVE_USR_LIB64_DIR) && defined(__LP64__)
-          "/usr/lib64/mozilla/plugins";
+          "@FINKPREFIX@/lib64/mozilla/plugins";
 #else
-          "/usr/lib/mozilla/plugins";
+          "@FINKPREFIX@/lib/mozilla/plugins";
 #endif
         rv = NS_NewNativeLocalFile(nsDependentCString(sysLPlgDir),
                                    PR_FALSE, getter_AddRefs(localFile));
     }
 #endif
 #endif
     else if (nsCRT::strcmp(prop, NS_APP_SEARCH_DIR) == 0)
     {
diff -ruN -U 8 mozilla-release-orig/security/nss/lib/softoken/config.mk mozilla-release/security/nss/lib/softoken/config.mk
--- mozilla-release-orig/security/nss/lib/softoken/config.mk	2011-08-11 17:41:33.000000000 -0400
+++ mozilla-release/security/nss/lib/softoken/config.mk	2011-08-16 09:15:05.000000000 -0400
@@ -75,16 +75,17 @@
 endif # NS_USE_GCC
 
 else
 
 # $(PROGRAM) has NO explicit dependencies on $(EXTRA_SHARED_LIBS)
 # $(EXTRA_SHARED_LIBS) come before $(OS_LIBS), except on AIX.
 EXTRA_SHARED_LIBS += \
 	-L$(DIST)/lib \
+	-L@FINKPREFIX@/lib \
 	-l$(SQLITE_LIB_NAME) \
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
 	-L$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
 	$(NULL)
