diff -ru monotone-1.0/extra/shell/monotone.bash_completion.in monotone-1.0-patched/extra/shell/monotone.bash_completion.in
--- monotone-1.0/extra/shell/monotone.bash_completion.in	2011-03-26 01:40:57.000000000 -0500
+++ monotone-1.0-patched/extra/shell/monotone.bash_completion.in	2011-05-27 20:56:32.000000000 -0500
@@ -26,6 +26,9 @@
 # If you do it, be careful to set your locale to "C" and be sure the right
 # mtn is used (you can use the environment variable MTN to point at the right
 # one).
+
+if [ ${BASH_VERSINFO[0]} -ge 4 ]; then
+
 # BEGIN ----------------------------------------------------------------------
 @generated-command-list@
 # END ------------------------------------------------------------------------
@@ -396,3 +399,5 @@
 }
 
 complete -F _monotone mtn
+
+fi
