diff -ruN mongodb-src-r3.4.11-orig/SConstruct mongodb-src-r3.4.11/SConstruct
--- mongodb-src-r3.4.11-orig/SConstruct	2018-01-22 13:21:09.000000000 -0600
+++ mongodb-src-r3.4.11/SConstruct	2018-06-17 09:31:33.000000000 -0500
@@ -1562,6 +1562,8 @@
                          "-Wall",
                          "-Wsign-compare",
                          "-Wno-unknown-pragmas",
+                         "-Wno-unused-private-field",
+                         "-Wno-unused-local-typedef",
                          "-Winvalid-pch"] )
     # env.Append( " -Wconversion" ) TODO: this doesn't really work yet
     if env.TargetOSIs('linux', 'osx', 'solaris'):
diff -ruN mongodb-src-r3.4.11-orig/debian/mongod.conf mongodb-src-r3.4.11/debian/mongod.conf
--- mongodb-src-r3.4.11-orig/debian/mongod.conf	2018-01-22 13:21:09.000000000 -0600
+++ mongodb-src-r3.4.11/debian/mongod.conf	2018-06-17 09:31:23.000000000 -0500
@@ -5,7 +5,7 @@
 
 # Where and how to store data.
 storage:
-  dbPath: /var/lib/mongodb
+  dbPath: @FINK_PREFIX@/var/db/mongodb
   journal:
     enabled: true
 #  engine:
@@ -16,7 +16,7 @@
 systemLog:
   destination: file
   logAppend: true
-  path: /var/log/mongodb/mongod.log
+  path: @FINK_PREFIX@/var/log/mongodb/mongod.log
 
 # network interfaces
 net:
diff -ruN mongodb-src-r3.4.11-orig/src/mongo/shell/linenoise_utf8.h mongodb-src-r3.4.11/src/mongo/shell/linenoise_utf8.h
--- mongodb-src-r3.4.11-orig/src/mongo/shell/linenoise_utf8.h	2018-01-22 13:21:09.000000000 -0600
+++ mongodb-src-r3.4.11/src/mongo/shell/linenoise_utf8.h	2018-06-17 09:31:33.000000000 -0500
@@ -30,6 +30,7 @@
 #include <algorithm>
 #include <memory>
 #include <string.h>
+#include <algorithm>
 
 namespace linenoise_utf8 {
 
