diff -Nurd mongodb-src-r2.4.8-orig/src/mongo/db/fts/stemmer.cpp mongodb-src-r2.4.8/src/mongo/db/fts/stemmer.cpp
--- mongodb-src-r2.4.8-orig/src/mongo/db/fts/stemmer.cpp	2013-11-07 13:48:29.000000000 +0100
+++ mongodb-src-r2.4.8/src/mongo/db/fts/stemmer.cpp	2013-11-07 15:23:25.000000000 +0100
@@ -17,6 +17,7 @@
 */
 
 #include <string>
+#include <cstdlib>
 
 #include "mongo/db/fts/stemmer.h"
 
diff -Nurd mongodb-src-r2.4.8-orig/src/mongo/dbtests/documentsourcetests.cpp mongodb-src-r2.4.8/src/mongo/dbtests/documentsourcetests.cpp
--- mongodb-src-r2.4.8-orig/src/mongo/dbtests/documentsourcetests.cpp	2013-11-07 13:48:29.000000000 +0100
+++ mongodb-src-r2.4.8/src/mongo/dbtests/documentsourcetests.cpp	2013-11-07 16:20:02.000000000 +0100
@@ -583,7 +583,7 @@
         };
 
         struct ValueCmp {
-            bool operator()(const Value& a, const Value& b) {
+            bool operator()(const Value& a, const Value& b) const {
                 return Value::compare( a, b ) < 0;
             }
         };
diff -Nurd mongodb-src-r2.4.8-orig/src/mongo/platform/unordered_map.h mongodb-src-r2.4.8/src/mongo/platform/unordered_map.h
--- mongodb-src-r2.4.8-orig/src/mongo/platform/unordered_map.h	2013-11-07 13:48:29.000000000 +0100
+++ mongodb-src-r2.4.8/src/mongo/platform/unordered_map.h	2013-11-07 13:52:00.000000000 +0100
@@ -18,7 +18,8 @@
 // NOTE(acm): Before gcc-4.7, __cplusplus is always defined to be 1,
 // so we can't reliably detect C++11 support by exclusively checking
 // the value of __cplusplus.
-#if __cplusplus >= 201103L || defined(__GXX_EXPERIMENTAL_CXX0X__)
+//#if __cplusplus >= 201103L || defined(__GXX_EXPERIMENTAL_CXX0X__)
+#if 1
 
 #include <unordered_map>
 
diff -Nurd mongodb-src-r2.4.8-orig/src/mongo/platform/unordered_set.h mongodb-src-r2.4.8/src/mongo/platform/unordered_set.h
--- mongodb-src-r2.4.8-orig/src/mongo/platform/unordered_set.h	2013-11-07 13:48:29.000000000 +0100
+++ mongodb-src-r2.4.8/src/mongo/platform/unordered_set.h	2013-11-07 14:24:38.000000000 +0100
@@ -18,7 +18,8 @@
 // NOTE(acm): Before gcc-4.7, __cplusplus is always defined to be 1,
 // so we can't reliably detect C++11 support by exclusively checking
 // the value of __cplusplus.
-#if __cplusplus >= 201103L || defined(__GXX_EXPERIMENTAL_CXX0X__)
+//#if __cplusplus >= 201103L || defined(__GXX_EXPERIMENTAL_CXX0X__)
+#if 1
 
 #include <unordered_set>
 
diff -Nurd mongodb-src-r2.4.8-orig/src/third_party/s2/hash.h mongodb-src-r2.4.8/src/third_party/s2/hash.h
--- mongodb-src-r2.4.8-orig/src/third_party/s2/hash.h	2013-11-07 13:48:30.000000000 +0100
+++ mongodb-src-r2.4.8/src/third_party/s2/hash.h	2013-11-07 15:42:44.000000000 +0100
@@ -7,13 +7,8 @@
 #include "mongo/platform/unordered_set.h"
 #define hash_set mongo::unordered_set
 
-#if defined OS_LINUX || defined OS_MACOSX || defined __sunos__ || defined __freebsd__
-#define HASH_NAMESPACE_START namespace std { namespace tr1 {
-#define HASH_NAMESPACE_END }}
-#elif defined OS_WINDOWS
 #define HASH_NAMESPACE_START namespace std {
 #define HASH_NAMESPACE_END }
-#endif
 
 // Places that hash-related functions are defined:
 // end of s2cellid.h for hashing on S2CellId
diff -Nurd mongodb-src-r2.4.14-orig/SConstruct mongodb-src-r2.4.14/SConstruct
--- mongodb-src-r2.4.14-orig/SConstruct	2015-04-27 11:18:57.000000000 -0500
+++ mongodb-src-r2.4.14/SConstruct	2015-09-12 15:37:59.000000000 -0500
@@ -512,20 +512,8 @@
     darwin = True
     platform = "osx" # prettier than darwin
 
-    if env["CXX"] is None:
-        if os.path.exists( "/usr/bin/g++-4.2" ):
-            env["CXX"] = "g++-4.2"
-
     nix = True
-
-    if force64:
-       env.Append( EXTRACPPPATH=["/usr/64/include"] )
-       env.Append( EXTRALIBPATH=["/usr/64/lib"] )
-       if installDir == DEFAULT_INSTALL_DIR:
-           installDir = "/usr/64/"
-    else:
-       env.Append( EXTRACPPPATH=filterExists(["/sw/include" , "/opt/local/include"]) )
-       env.Append( EXTRALIBPATH=filterExists(["/sw/lib/", "/opt/local/lib"]) )
+    env.MergeFlags( [ os.getenv("CPPFLAGS"), os.getenv("LDFLAGS") ] )
 
 elif os.sys.platform.startswith("linux"):
     linux = True
diff -Nurd mongodb-src-r2.4.14-orig/distsrc/client/SConstruct mongodb-src-r2.4.14/distsrc/client/SConstruct
--- mongodb-src-r2.4.14-orig/distsrc/client/SConstruct	2015-04-27 11:18:57.000000000 -0500
+++ mongodb-src-r2.4.14/distsrc/client/SConstruct	2015-09-12 15:33:42.000000000 -0500
@@ -62,7 +62,7 @@
 darwin = False
 
 if "darwin" == sys.platform:
-    addExtraLibs( "/opt/local/" )
+    addExtraLibs( "@FINK_PREFIX@/" )
     nix = True
     darwin = True
 elif sys.platform in ("linux2", "linux3"):
diff -Nurd mongodb-src-r2.4.14-orig/src/mongo/util/compress.cpp mongodb-src-r2.4.14/src/mongo/util/compress.cpp
--- mongodb-src-r2.4.14-orig/src/mongo/util/compress.cpp	2015-04-27 11:18:57.000000000 -0500
+++ mongodb-src-r2.4.14/src/mongo/util/compress.cpp	2015-09-12 15:39:05.000000000 -0500
@@ -18,7 +18,7 @@
 
 #include "mongo/util/compress.h"
 
-#include "third_party/snappy/snappy.h"
+#include <snappy.h>
 
 namespace mongo {
 
diff -Nurd mongodb-src-r2.4.14-orig/src/third_party/v8/SConscript mongodb-src-r2.4.14/src/third_party/v8/SConscript
--- mongodb-src-r2.4.14-orig/src/third_party/v8/SConscript	2015-04-27 11:18:57.000000000 -0500
+++ mongodb-src-r2.4.14/src/third_party/v8/SConscript	2015-09-12 15:40:46.000000000 -0500
@@ -47,9 +47,9 @@
   'gcc': {
     'all': {
       'CCFLAGS':      ['-Wall',
-                       '-Werror',
                        '-W',
                        '-Wno-unused-parameter',
+                       '-Wno-unused-private-field',
                        '-Woverloaded-virtual',
                        '-Wnon-virtual-dtor']
     },
