diff -ruN mysql-5.0.96-orig/config/ac-macros/misc.m4 mysql-5.0.96/config/ac-macros/misc.m4
--- mysql-5.0.96-orig/config/ac-macros/misc.m4	2012-03-02 09:04:12
+++ mysql-5.0.96/config/ac-macros/misc.m4	2026-06-19 11:20:37
@@ -160,6 +160,7 @@
 [AC_MSG_CHECKING(for type ulong)
 AC_CACHE_VAL(ac_cv_ulong,
 [AC_TRY_RUN([#include <stdio.h>
+#include <stdlib.h>
 #include <sys/types.h>
 main()
 {
@@ -178,6 +179,7 @@
 [AC_MSG_CHECKING(for type uchar)
 AC_CACHE_VAL(ac_cv_uchar,
 [AC_TRY_RUN([#include <stdio.h>
+#include <stdlib.h>
 #include <sys/types.h>
 main()
 {
@@ -196,6 +198,7 @@
 [AC_MSG_CHECKING(for type uint)
 AC_CACHE_VAL(ac_cv_uint,
 [AC_TRY_RUN([#include <stdio.h>
+#include <stdlib.h>
 #include <sys/types.h>
 main()
 {
@@ -215,6 +218,7 @@
 [AC_MSG_CHECKING(for type in_addr_t)
 AC_CACHE_VAL(ac_cv_in_addr_t,
 [AC_TRY_RUN([#include <stdio.h>
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
@@ -276,6 +280,7 @@
 [AC_MSG_CHECKING(for type fp_except)
 AC_CACHE_VAL(ac_cv_fp_except,
 [AC_TRY_RUN([#include <stdio.h>
+#include <stdlib.h>
 #include <sys/types.h>
 #include <ieeefp.h>
 main()
@@ -612,6 +617,7 @@
 AC_MSG_CHECKING(if conversion of longlong to float works)
 AC_CACHE_VAL(ac_cv_conv_longlong_to_float,
 [AC_TRY_RUN([#include <stdio.h>
+#include <stdlib.h>
 typedef long long longlong;
 main()
 {
diff -ruN mysql-5.0.96-orig/configure.in mysql-5.0.96/configure.in
--- mysql-5.0.96-orig/configure.in	2012-03-02 09:04:07
+++ mysql-5.0.96/configure.in	2026-06-19 11:21:36
@@ -293,6 +293,8 @@
     ;;
   *)
 AC_TRY_RUN([
+  #include <stdio.h>
+  #include <string.h>
   int main()
     {
       char* s = "hello";
@@ -2435,6 +2437,7 @@
 AC_MSG_CHECKING("can netinet files be included")
 AC_TRY_COMPILE(
 [#include <sys/types.h>
+#include <stdio.h>
 #include <sys/socket.h>
 #include <netinet/in_systm.h>
 #include <netinet/in.h>
diff -ruN mysql-5.0.96-orig/ndb/src/common/transporter/TCP_Transporter.cpp mysql-5.0.96/ndb/src/common/transporter/TCP_Transporter.cpp
--- mysql-5.0.96-orig/ndb/src/common/transporter/TCP_Transporter.cpp	2012-03-02 08:04:13.000000000 -0600
+++ mysql-5.0.96/ndb/src/common/transporter/TCP_Transporter.cpp	2020-11-29 15:17:54.000000000 -0600
@@ -19,6 +19,7 @@
 #include "TCP_Transporter.hpp"
 #include <NdbOut.hpp>
 #include <NdbSleep.h>
+#include <netinet/tcp.h>
 // End of stuff to be moved
 
 #if defined NDB_OSE || defined NDB_SOFTOSE
diff -ruN mysql-5.0.96-orig/vio/viosocket.c mysql-5.0.96/vio/viosocket.c
--- mysql-5.0.96-orig/vio/viosocket.c	2012-03-02 08:04:08.000000000 -0600
+++ mysql-5.0.96/vio/viosocket.c	2020-11-29 15:35:17.000000000 -0600
@@ -21,6 +21,7 @@
   the file descriptior.
 */
 
+#include <netinet/tcp.h>
 #include "vio_priv.h"
 
 int vio_errno(Vio *vio __attribute__((unused)))
