diff -ruN postgresql-9.4.26-orig/configure postgresql-9.4.26/configure
--- postgresql-9.4.26-orig/configure	2020-02-10 16:25:31.000000000 -0600
+++ postgresql-9.4.26/configure	2021-01-22 19:50:25.000000000 -0600
@@ -11540,6 +11540,7 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <xlocale.h>
 #include <time.h>
 #ifndef tzname /* For SGI.  */
 extern char *tzname[]; /* RS6000 and others reject char **tzname.  */
@@ -12607,7 +12608,6 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
 int
 main ()
 {
@@ -13713,7 +13713,7 @@
 # trust that the arithmetic works.
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
+#include <stdlib.h>
 int
 main ()
 {
@@ -13734,6 +13734,7 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdlib.h>
 typedef long int ac_int64;
 
 /*
@@ -13795,7 +13796,7 @@
 # trust that the arithmetic works.
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
+#include <stdlib.h>
 int
 main ()
 {
@@ -13816,6 +13817,7 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdlib.h>
 typedef long long int ac_int64;
 
 /*
