diff -ruN postgresql-9.5.24-orig/configure postgresql-9.5.24/configure
--- postgresql-9.5.24-orig/configure	2020-11-09 16:34:09.000000000 -0600
+++ postgresql-9.5.24/configure	2021-01-23 05:45:01.000000000 -0600
@@ -11642,6 +11642,7 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdlib.h>
 #include <time.h>
 #ifndef tzname /* For SGI.  */
 extern char *tzname[]; /* RS6000 and others reject char **tzname.  */
@@ -13761,7 +13762,7 @@
 # trust that the arithmetic works.
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-
+#include <stdlib.h>
 int
 main ()
 {
@@ -13782,6 +13783,7 @@
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdlib.h>
 typedef long int ac_int64;
 
 /*
@@ -13845,6 +13847,7 @@
 # trust that the arithmetic works.
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdlib.h>
 
 int
 main ()
