diff -x '*~' -urN redis-5.0.5_org/redis.conf redis-5.0.5/redis.conf
--- redis-5.0.5_org/redis.conf	2019-05-15 18:07:37.000000000 +0200
+++ redis-5.0.5/redis.conf	2019-08-31 21:18:57.000000000 +0200
@@ -132,8 +132,8 @@
 ################################# GENERAL #####################################
 
 # By default Redis does not run as a daemon. Use 'yes' if you need it.
-# Note that Redis will write a pid file in /var/run/redis.pid when daemonized.
-daemonize no
+# Note that Redis will write a pid file in @PREFIX@/var/run/redis/redis.pid when daemonized.
+daemonize yes
 
 # If you run Redis from upstart or systemd, Redis can interact with your
 # supervision tree. Options:
@@ -151,11 +151,11 @@
 #
 # When the server runs non daemonized, no pid file is created if none is
 # specified in the configuration. When the server is daemonized, the pid file
-# is used even if not specified, defaulting to "/var/run/redis.pid".
+# is used even if not specified, defaulting to "@PREFIX@/var/run/redis/redis.pid".
 #
 # Creating a pid file is best effort: if Redis is not able to create it
 # nothing bad happens, the server will start and run normally.
-pidfile /var/run/redis_6379.pid
+pidfile @PREFIX@/var/run/redis/redis.pid
 
 # Specify the server verbosity level.
 # This can be one of:
@@ -168,7 +168,7 @@
 # Specify the log file name. Also the empty string can be used to force
 # Redis to log on the standard output. Note that if you use standard
 # output for logging but daemonize, logs will be sent to /dev/null
-logfile ""
+logfile @PREFIX@/var/log/redis/redis.log
 
 # To enable logging to the system logger, just set 'syslog-enabled' to yes,
 # and optionally update the other syslog parameters to suit your needs.
@@ -214,10 +214,10 @@
 #   like in the following example:
 #
 #   save ""
-
-save 900 1
-save 300 10
-save 60 10000
+#
+# save 900 1
+# save 300 10
+# save 60 10000
 
 # By default Redis will stop accepting writes if RDB snapshots are enabled
 # (at least one save point) and the latest background save failed.
@@ -260,7 +260,7 @@
 # The Append Only File will also be created inside this directory.
 #
 # Note that you must specify a directory here, not a file name.
-dir ./
+dir @PREFIX@/var/db/redis/
 
 ################################# REPLICATION #################################
 
diff -x '*~' -urN redis-5.0.5_org/redis.sh redis-5.0.5/redis.sh
--- redis-5.0.5_org/redis.sh	1970-01-01 01:00:00.000000000 +0100
+++ redis-5.0.5/redis.sh	2019-08-31 21:18:57.000000000 +0200
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+export PREFIX="@PREFIX@"
+export PATH="$PREFIX/bin:$PATH"
+export CONFIG="$PREFIX/etc/redis.conf"
+
+case "$1" in
+    start)
+        sudo -u redis redis-server "$CONFIG"
+        ;;
+esac
diff -x '*~' -urN redis-5.0.5_org/sentinel.conf redis-5.0.5/sentinel.conf
--- redis-5.0.5_org/sentinel.conf	2019-05-15 18:07:37.000000000 +0200
+++ redis-5.0.5/sentinel.conf	2019-08-31 21:18:57.000000000 +0200
@@ -182,7 +182,7 @@
 #
 # Example:
 #
-# sentinel notification-script mymaster /var/redis/notify.sh
+# sentinel notification-script mymaster @PREFIX@/var/redis/notify.sh
 
 # CLIENTS RECONFIGURATION SCRIPT
 #
@@ -207,7 +207,7 @@
 #
 # Example:
 #
-# sentinel client-reconfig-script mymaster /var/redis/reconfig.sh
+# sentinel client-reconfig-script mymaster @PREFIX@/var/redis/reconfig.sh
 
 # SECURITY
 #
diff -x '*~' -urN redis-5.0.5_org/src/server.h redis-5.0.5/src/server.h
--- redis-5.0.5_org/src/server.h	2019-05-15 18:07:37.000000000 +0200
+++ redis-5.0.5/src/server.h	2019-08-31 21:18:57.000000000 +0200
@@ -111,7 +111,7 @@
 #define CONFIG_DEFAULT_REPL_BACKLOG_TIME_LIMIT (60*60)  /* 1 hour */
 #define CONFIG_REPL_BACKLOG_MIN_SIZE (1024*16)          /* 16k */
 #define CONFIG_BGSAVE_RETRY_DELAY 5 /* Wait a few secs before trying again. */
-#define CONFIG_DEFAULT_PID_FILE "/var/run/redis.pid"
+#define CONFIG_DEFAULT_PID_FILE "@PREFIX@/var/run/redis/redis.pid"
 #define CONFIG_DEFAULT_SYSLOG_IDENT "redis"
 #define CONFIG_DEFAULT_CLUSTER_CONFIG_FILE "nodes.conf"
 #define CONFIG_DEFAULT_CLUSTER_ANNOUNCE_IP NULL         /* Auto detect. */
diff -x '*~' -urN redis-5.0.5_org/tests/support/server.tcl redis-5.0.5/tests/support/server.tcl
--- redis-5.0.5_org/tests/support/server.tcl	2019-05-15 18:07:37.000000000 +0200
+++ redis-5.0.5/tests/support/server.tcl	2019-08-31 21:18:57.000000000 +0200
@@ -32,25 +32,29 @@
     if {![is_alive $config]} { return }
     set pid [dict get $config pid]
 
-    # check for leaks
-    if {![dict exists $config "skipleaks"]} {
-        catch {
-            if {[string match {*Darwin*} [exec uname -a]]} {
-                tags {"leaks"} {
-                    test "Check for memory leaks (pid $pid)" {
-                        set output {0 leaks}
-                        catch {exec leaks $pid} output
-                        if {[string match {*process does not exist*} $output] ||
-                            [string match {*cannot examine*} $output]} {
-                            # In a few tests we kill the server process.
-                            set output "0 leaks"
-                        }
-                        set output
-                    } {*0 leaks*}
-                }
-            }
-        }
-    }
+# finkproject patch:  
+#     removed memory leak check to prevent
+#       "leaks[...]: [fatal] unable to ask for permission to examine process;..."
+#
+#    # check for leaks
+#    if {![dict exists $config "skipleaks"]} {
+#        catch {
+#            if {[string match {*Darwin*} [exec uname -a]]} {
+#                tags {"leaks"} {
+#                    test "Check for memory leaks (pid $pid)" {
+#                        set output {0 leaks}
+#                        catch {exec leaks $pid} output
+#                        if {[string match {*process does not exist*} $output] ||
+#                            [string match {*cannot examine*} $output]} {
+#                            # In a few tests we kill the server process.
+#                            set output "0 leaks"
+#                        }
+#                        set output
+#                    } {*0 leaks*}
+#                }
+#            }
+#        }
+#    }
 
     # kill server and wait for the process to be totally exited
     catch {exec kill $pid}
