diff -Nurd -x'*~' spatialite_gui-1.7.1.orig/configure spatialite_gui-1.7.1/configure
--- spatialite_gui-1.7.1.orig/configure	2013-06-29 04:29:11.000000000 -0400
+++ spatialite_gui-1.7.1/configure	2020-12-20 13:57:49.000000000 -0500
@@ -15543,7 +15543,7 @@
 fi
 CXXFLAGS="$(wx-config --cxxflags)"
 AM_CXXFLAGS="$(wx-config --cxxflags)"
-WX_LIBS="$(wx-config --libs)"
+WX_LIBS="$(wx-config --libs std,aui)"
 
 
 # Checks for header files.
@@ -16271,6 +16271,7 @@
 fi
 done
 
+LIBS="-lsqlite3 $LIBS"
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for pj_init_plus in -lproj" >&5
 $as_echo_n "checking for pj_init_plus in -lproj... " >&6; }
