diff -Nurd -x'*~' unixODBC-2.2.14.orig/Drivers/MiniSQL/Makefile.in unixODBC-2.2.14/Drivers/MiniSQL/Makefile.in
--- unixODBC-2.2.14.orig/Drivers/MiniSQL/Makefile.in	2008-11-19 10:11:05.000000000 -0600
+++ unixODBC-2.2.14/Drivers/MiniSQL/Makefile.in	2016-03-02 21:59:47.000000000 -0600
@@ -307,7 +307,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 @MSQL_TRUE@lib_LTLIBRARIES = libodbcmini.la
-@MSQL_TRUE@INCLUDES = -I@top_srcdir@/include -I. -I@msql_headers@ -I${LTDLINCL}
+@MSQL_TRUE@INCLUDES = -I@top_srcdir@/include -I. -I${LTDLINCL} -I@msql_headers@
 
 # PAH - Redundant?
 # DEFS = -DUNIXODBC @DEFS@ -DDRIVER_NAME=\"odbcminisql\"
diff -Nurd -x'*~' unixODBC-2.2.14.orig/ODBCConfig/Makefile.in unixODBC-2.2.14/ODBCConfig/Makefile.in
--- unixODBC-2.2.14.orig/ODBCConfig/Makefile.in	2008-11-19 10:11:13.000000000 -0600
+++ unixODBC-2.2.14/ODBCConfig/Makefile.in	2016-03-02 22:00:49.000000000 -0600
@@ -241,8 +241,8 @@
 subdirs = @subdirs@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-@QT_TRUE@INCLUDES = -I@top_srcdir@/include $(QT_CXXFLAGS) \
-@QT_TRUE@    -DSYSTEM_FILE_PATH=\"@sysconfdir@\" -DDEFLIB_PATH=\"@libdir@\" $(LTDLINCL)
+@QT_TRUE@INCLUDES = -I@top_srcdir@/include $(LTDLINCL) $(QT_CXXFLAGS) \
+@QT_TRUE@    -DSYSTEM_FILE_PATH=\"@sysconfdir@\" -DDEFLIB_PATH=\"@libdir@\"
 
 @QT_TRUE@ODBCConfig_LDADD = \
 @QT_TRUE@	$(QT_LDFLAGS) \
diff -Nurd -x'*~' unixODBC-2.2.14.orig/odbcinstQ/Makefile.in unixODBC-2.2.14/odbcinstQ/Makefile.in
--- unixODBC-2.2.14.orig/odbcinstQ/Makefile.in	2008-11-19 10:11:32.000000000 -0600
+++ unixODBC-2.2.14/odbcinstQ/Makefile.in	2016-03-02 22:02:53.000000000 -0600
@@ -280,8 +280,8 @@
 	    -module
 
 @QTWIZARD_FALSE@@QT_TRUE@lib_LTLIBRARIES = libodbcinstQ.la 
-@QTWIZARD_FALSE@@QT_TRUE@INCLUDES = -I@top_srcdir@/include $(QT_CXXFLAGS) \
-@QTWIZARD_FALSE@@QT_TRUE@    -DSYSTEM_FILE_PATH=\"@sysconfdir@\" -DDEFLIB_PATH=\"@libdir@\" $(LTDLINCL)
+@QTWIZARD_FALSE@@QT_TRUE@INCLUDES = -I@top_srcdir@/include $(LTDLINCL) $(QT_CXXFLAGS) \
+@QTWIZARD_FALSE@@QT_TRUE@   -DSYSTEM_FILE_PATH=\"@sysconfdir@\" -DDEFLIB_PATH=\"@libdir@\" 
 
 @QTWIZARD_FALSE@@QT_TRUE@libodbcinstQ_la_LIBADD = \
 @QTWIZARD_FALSE@@QT_TRUE@	$(QT_LDFLAGS) \
diff -Nurd -x'*~' unixODBC-2.2.14.orig/odbctest/Makefile.in unixODBC-2.2.14/odbctest/Makefile.in
--- unixODBC-2.2.14.orig/odbctest/Makefile.in	2008-11-19 10:11:35.000000000 -0600
+++ unixODBC-2.2.14/odbctest/Makefile.in	2016-03-02 22:04:30.000000000 -0600
@@ -257,7 +257,7 @@
 subdirs = @subdirs@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-@QT4_FALSE@@QT_TRUE@INCLUDES = -I@top_srcdir@/include $(QT_CXXFLAGS) -DSYSTEM_FILE_PATH=\"@sysconfdir@\" $(LTDLINCL)
+@QT4_FALSE@@QT_TRUE@INCLUDES = -I@top_srcdir@/include $(LTDLINCL) $(QT_CXXFLAGS) -DSYSTEM_FILE_PATH=\"@sysconfdir@\"
 @QT4_FALSE@@QT_TRUE@odbctest_LDADD = \
 @QT4_FALSE@@QT_TRUE@	$(QT_LDFLAGS) \
 @QT4_FALSE@@QT_TRUE@	$(QT_LIBS) \
