diff -ruN unixODBC-2.3.9-orig/Drivers/Postgre7.1/connection.c unixODBC-2.3.9/Drivers/Postgre7.1/connection.c
--- unixODBC-2.3.9-orig/Drivers/Postgre7.1/connection.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/Postgre7.1/connection.c	2020-12-09 05:51:41.000000000 -0600
@@ -45,6 +45,12 @@
 
 extern GLOBAL_VALUES globals;
 
+RETCODE SQL_API PG_SQLAllocStmt(HDBC      hdbc,
+			     HSTMT FAR *phstmt);
+
+/*      Perform a Prepare on the SQL statement */
+SQLRETURN   PG_SQLPrepare(SQLHSTMT hstmt,
+           SQLCHAR *szSqlStr, SQLINTEGER cbSqlStr);
 
 RETCODE SQL_API SQLAllocConnect(HENV     henv,
                                 HDBC FAR *phdbc)
@@ -80,6 +86,28 @@
     return SQL_SUCCESS;
 }
 
+/*      Execute a prepared SQL statement */
+RETCODE SQL_API PG_SQLExecute(
+        HSTMT   hstmt);
+
+RETCODE SQL_API PG_SQLExecDirect(
+        HSTMT     hstmt,
+        UCHAR FAR *szSqlStr,
+        SDWORD    cbSqlStr);
+
+RETCODE SQL_API PG_SQLFreeStmt(HSTMT     hstmt,
+			    UWORD     fOption);
+
+RETCODE SQL_API PG_SQLFetch(
+        HSTMT   hstmt);
+
+RETCODE SQL_API PG_SQLGetData(
+        HSTMT      hstmt,
+        UWORD      icol,
+        SWORD      fCType,
+        PTR        rgbValue,
+        SDWORD     cbValueMax,
+        SDWORD FAR *pcbValue);
 
 /*      -       -       -       -       -       -       -       -       - */
 
diff -ruN unixODBC-2.3.9-orig/Drivers/Postgre7.1/execute.c unixODBC-2.3.9/Drivers/Postgre7.1/execute.c
--- unixODBC-2.3.9-orig/Drivers/Postgre7.1/execute.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/Postgre7.1/execute.c	2020-12-09 05:52:30.000000000 -0600
@@ -325,6 +325,9 @@
 
 }
 
+RETCODE SQL_API PG_SQLFreeStmt(HSTMT     hstmt,
+			    UWORD     fOption);
+
 RETCODE SQL_API SQLExecute(
         HSTMT   hstmt)
 {
diff -ruN unixODBC-2.3.9-orig/Drivers/Postgre7.1/info.c unixODBC-2.3.9/Drivers/Postgre7.1/info.c
--- unixODBC-2.3.9-orig/Drivers/Postgre7.1/info.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/Postgre7.1/info.c	2020-12-09 05:54:16.000000000 -0600
@@ -45,7 +45,6 @@
 #include "misc.h"
 #include "pgtypes.h"
 
-
 /*	Trigger related stuff for SQLForeign Keys */
 #define TRIGGER_SHIFT 3
 #define TRIGGER_MASK   0x03
@@ -682,6 +681,36 @@
 	return result;
 }
 
+RETCODE SQL_API PG_SQLAllocStmt(HDBC      hdbc,
+			     HSTMT FAR *phstmt);
+
+SQLRETURN   PG_SQLPrepare(SQLHSTMT hstmt,
+           SQLCHAR *szSqlStr, SQLINTEGER cbSqlStr);
+
+/*      Execute a prepared SQL statement */
+RETCODE SQL_API PG_SQLExecute(
+        HSTMT   hstmt);
+
+RETCODE SQL_API PG_SQLExecDirect(
+        HSTMT     hstmt,
+        UCHAR FAR *szSqlStr,
+        SDWORD    cbSqlStr);
+
+RETCODE SQL_API PG_SQLFreeStmt(HSTMT     hstmt,
+			    UWORD     fOption);
+
+RETCODE SQL_API PG_SQLFetch(
+        HSTMT   hstmt);
+
+/*      Associate a user-supplied buffer with a database column. */
+RETCODE SQL_API PG_SQLBindCol(
+        HSTMT      hstmt,
+        UWORD      icol,
+        SWORD      fCType,
+        PTR        rgbValue,
+        SQLLEN     cbValueMax,
+        SQLLEN    *pcbValue);
+
 /*      -       -       -       -       -       -       -       -       - */
 
 
diff -ruN unixODBC-2.3.9-orig/Drivers/Postgre7.1/parse.c unixODBC-2.3.9/Drivers/Postgre7.1/parse.c
--- unixODBC-2.3.9-orig/Drivers/Postgre7.1/parse.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/Postgre7.1/parse.c	2020-12-09 05:55:59.000000000 -0600
@@ -41,6 +41,50 @@
 #define TAB_INCR	8
 #define COL_INCR	16
 
+extern GLOBAL_VALUES globals;
+
+RETCODE SQL_API PG_SQLAllocStmt(HDBC      hdbc,
+			     HSTMT FAR *phstmt);
+
+/*      Perform a Prepare on the SQL statement */
+SQLRETURN   PG_SQLPrepare(SQLHSTMT hstmt,
+           SQLCHAR *szSqlStr, SQLINTEGER cbSqlStr);
+
+/*      Execute a prepared SQL statement */
+RETCODE SQL_API PG_SQLExecute(
+        HSTMT   hstmt);
+
+RETCODE SQL_API PG_SQLExecDirect(
+        HSTMT     hstmt,
+        UCHAR FAR *szSqlStr,
+        SDWORD    cbSqlStr);
+
+RETCODE SQL_API PG_SQLFreeStmt(HSTMT     hstmt,
+			    UWORD     fOption);
+
+RETCODE SQL_API PG_SQLFetch(
+        HSTMT   hstmt);
+
+/*      Associate a user-supplied buffer with a database column. */
+RETCODE SQL_API PG_SQLBindCol(
+        HSTMT      hstmt,
+        UWORD      icol,
+        SWORD      fCType,
+        PTR        rgbValue,
+        SQLLEN     cbValueMax,
+        SQLLEN    *pcbValue);
+
+RETCODE SQL_API PG_SQLColumns(
+                           HSTMT        hstmt,
+                           UCHAR FAR *  szTableQualifier,
+                           SWORD        cbTableQualifier,
+                           UCHAR FAR *  szTableOwner,
+                           SWORD        cbTableOwner,
+                           UCHAR FAR *  szTableName,
+                           SWORD        cbTableName,
+                           UCHAR FAR *  szColumnName,
+                           SWORD        cbColumnName);
+
 char *getNextToken(char *s, char *token, int smax, char *delim, char *quote, char *dquote, char *numeric);
 void getColInfo(COL_INFO *col_info, FIELD_INFO *fi, int k);
 char searchColInfo(COL_INFO *col_info, FIELD_INFO *fi);
diff -ruN unixODBC-2.3.9-orig/Drivers/Postgre7.1/statement.c unixODBC-2.3.9/Drivers/Postgre7.1/statement.c
--- unixODBC-2.3.9-orig/Drivers/Postgre7.1/statement.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/Postgre7.1/statement.c	2020-12-09 05:56:50.000000000 -0600
@@ -113,6 +113,12 @@
     return PG_SQLAllocStmt( hdbc, phstmt );
 }
 
+SQLRETURN   PG_SQLPrepare(SQLHSTMT hstmt,
+           SQLCHAR *szSqlStr, SQLINTEGER cbSqlStr);
+
+/*      Execute a prepared SQL statement */
+RETCODE SQL_API PG_SQLExecute(
+        HSTMT   hstmt);
 
 RETCODE SQL_API PG_SQLFreeStmt(HSTMT     hstmt,
 			    UWORD     fOption)
diff -ruN unixODBC-2.3.9-orig/Drivers/nn/SQLDriverConnect.c unixODBC-2.3.9/Drivers/nn/SQLDriverConnect.c
--- unixODBC-2.3.9-orig/Drivers/nn/SQLDriverConnect.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/nn/SQLDriverConnect.c	2020-12-09 19:44:23.000000000 -0600
@@ -15,6 +15,8 @@
 #include <config.h>
 #include "driver.h"
 
+int	nnodbc_conndialog();
+
 RETCODE SQL_API SQLDriverConnect(
 										  HDBC  hdbc,
 										  HWND  hwnd,
diff -ruN unixODBC-2.3.9-orig/Drivers/nn/SQLExecDirect.c unixODBC-2.3.9/Drivers/nn/SQLExecDirect.c
--- unixODBC-2.3.9-orig/Drivers/nn/SQLExecDirect.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/nn/SQLExecDirect.c	2020-12-09 19:44:20.000000000 -0600
@@ -15,6 +15,9 @@
 #include <config.h>
 #include "driver.h"
 
+int	sqlexecute (
+	stmt_t*   pstmt);
+
 RETCODE SQL_API SQLExecDirect(
 									  HSTMT hstmt,
 									  UCHAR*   szSqlStr,
diff -ruN unixODBC-2.3.9-orig/Drivers/nn/SQLExecute.c unixODBC-2.3.9/Drivers/nn/SQLExecute.c
--- unixODBC-2.3.9-orig/Drivers/nn/SQLExecute.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/nn/SQLExecute.c	2020-12-09 19:44:17.000000000 -0600
@@ -15,6 +15,9 @@
 #include <config.h>
 #include "driver.h"
 
+int	sqlexecute (
+	stmt_t*   pstmt);
+
 RETCODE SQL_API SQLExecute (
 									HSTMT hstmt )
 {
diff -ruN unixODBC-2.3.9-orig/Drivers/nn/SQLParamData.c unixODBC-2.3.9/Drivers/nn/SQLParamData.c
--- unixODBC-2.3.9-orig/Drivers/nn/SQLParamData.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/nn/SQLParamData.c	2020-12-09 19:46:39.000000000 -0600
@@ -15,6 +15,11 @@
 #include <config.h>
 #include "driver.h"
 
+int	sqlputdata (
+		stmt_t* 	pstmt,
+		int		ipar,
+		char*		data );
+
 RETCODE SQL_API SQLParamData(
 									 HSTMT hstmt,
 									 PTR*  prgbValue)
diff -ruN unixODBC-2.3.9-orig/Drivers/nn/SQLPutData.c unixODBC-2.3.9/Drivers/nn/SQLPutData.c
--- unixODBC-2.3.9-orig/Drivers/nn/SQLPutData.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/nn/SQLPutData.c	2020-12-09 19:47:08.000000000 -0600
@@ -15,6 +15,11 @@
 #include <config.h>
 #include "driver.h"
 
+int	sqlputdata (
+		stmt_t* 	pstmt,
+		int		ipar,
+		char*		data );
+
 RETCODE SQL_API SQLPutData(
 								  HSTMT hstmt,
 								  PTR   rgbValue,
diff -ruN unixODBC-2.3.9-orig/Drivers/nn/SQLSetParam.c unixODBC-2.3.9/Drivers/nn/SQLSetParam.c
--- unixODBC-2.3.9-orig/Drivers/nn/SQLSetParam.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/nn/SQLSetParam.c	2020-12-09 19:48:09.000000000 -0600
@@ -15,6 +15,18 @@
 #include <config.h>
 #include "driver.h"
 
+RETCODE SQL_API SQLBindParameter(
+										  HSTMT hstmt,
+										  UWORD ipar,
+										  SWORD fParamType,
+										  SWORD fCType,
+										  SWORD fSqlType,
+										  UDWORD   cbColDef,
+										  SWORD ibScale,
+										  PTR   rgbValue,
+										  SDWORD   cbValueMax,
+										  SDWORD* pcbValue);
+
 RETCODE SQL_API SQLSetParam (
 									 HSTMT     hstmt,
 									 UWORD     ipar,
diff -ruN unixODBC-2.3.9-orig/Drivers/nn/connect.c unixODBC-2.3.9/Drivers/nn/connect.c
--- unixODBC-2.3.9-orig/Drivers/nn/connect.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/nn/connect.c	2020-12-09 19:50:07.000000000 -0600
@@ -14,6 +14,12 @@
 **/
 #include <config.h>
 #include "driver.h"
+#include <unistd.h>
+
+int	upper_strneq(
+	char*	s1,
+	char*	s2,
+	int	n );
 
 void*	nnodbc_getenverrstack(void* henv)
 {
diff -ruN unixODBC-2.3.9-orig/Drivers/nn/driver.h unixODBC-2.3.9/Drivers/nn/driver.h
--- unixODBC-2.3.9-orig/Drivers/nn/driver.h	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/nn/driver.h	2020-12-09 19:42:50.000000000 -0600
@@ -27,6 +27,8 @@
 #include "hstmt.h"
 #include "herr.h"
 
+void	nnodbc_errstkunset(void* stack);
+
 typedef struct GSTMT {
 	void*	hdbc;
 	void*	hstmt;
diff -ruN unixODBC-2.3.9-orig/Drivers/nn/nncol.c unixODBC-2.3.9/Drivers/nn/nncol.c
--- unixODBC-2.3.9-orig/Drivers/nn/nncol.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/nn/nncol.c	2020-12-09 20:03:27.000000000 -0600
@@ -19,6 +19,11 @@
 #include	<nncol.h>
 #include	"nncol.ci"
 
+int	upper_strneq(
+	char*	s1,
+	char*	s2,
+	int	n );
+
 int	nnsql_getcolidxbyname( char* col_name )
 {
 	int	i;
diff -ruN unixODBC-2.3.9-orig/Drivers/nn/nndate.c unixODBC-2.3.9/Drivers/nn/nndate.c
--- unixODBC-2.3.9-orig/Drivers/nn/nndate.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/nn/nndate.c	2020-12-09 20:04:50.000000000 -0600
@@ -17,6 +17,11 @@
 
 #include	<nndate.h>
 
+int	upper_strneq(
+	char*	s1,
+	char*	s2,
+	int	n );
+
 static char* month_name[] =
 { "Jan", "Feb", "Mar", "Apr", "May", "Jun",
   "Jul", "Aug", "Sep", "Oct", "Nov", "Dec" };
diff -ruN unixODBC-2.3.9-orig/Drivers/nn/nntp.c unixODBC-2.3.9/Drivers/nn/nntp.c
--- unixODBC-2.3.9-orig/Drivers/nn/nntp.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/nn/nntp.c	2020-12-09 20:06:15.000000000 -0600
@@ -19,6 +19,8 @@
 #include    <string.h>
 #include	<errno.h>
 #include	<nntp.h>
+#include <unistd.h>
+#include <arpa/inet.h>
 
 # include	"nntp.ci"
 
@@ -37,6 +39,11 @@
 
 #include	<nntp.h>
 
+int	upper_strneq(
+	char*	s1,
+	char*	s2,
+	int	n );
+
 typedef struct {
 	long	article_num;
 	long	data;
diff -ruN unixODBC-2.3.9-orig/Drivers/nn/yylex.c unixODBC-2.3.9/Drivers/nn/yylex.c
--- unixODBC-2.3.9-orig/Drivers/nn/yylex.c	2018-02-26 08:43:27.000000000 -0600
+++ unixODBC-2.3.9/Drivers/nn/yylex.c	2020-12-09 19:53:04.000000000 -0600
@@ -27,6 +27,11 @@
 #include	<stdio.h>
 #include	<ctype.h>
 
+int	upper_strneq(
+	char*	s1,
+	char*	s2,
+	int	n );
+
 static int	getcmpopidxbyname(char* name)
 {
 	int	i, size;
diff -ruN unixODBC-2.3.9-orig/Drivers/nn/yyparse.c unixODBC-2.3.9/Drivers/nn/yyparse.c
--- unixODBC-2.3.9-orig/Drivers/nn/yyparse.c	2018-05-03 12:48:49.000000000 -0500
+++ unixODBC-2.3.9/Drivers/nn/yyparse.c	2020-12-09 19:51:51.000000000 -0600
@@ -300,6 +300,7 @@
 
 
 int yyparse (void);
+int	nnsql_yylex(YYSTYPE* pyylval, yyenv_t* penv);
 
 
 
diff -ruN unixODBC-2.3.9-orig/Drivers/nn/yystmt.c unixODBC-2.3.9/Drivers/nn/yystmt.c
--- unixODBC-2.3.9-orig/Drivers/nn/yystmt.c	2014-01-06 03:37:01.000000000 -0600
+++ unixODBC-2.3.9/Drivers/nn/yystmt.c	2020-12-09 20:02:28.000000000 -0600
@@ -14,11 +14,19 @@
 **/
 #include <config.h>
 #include "driver.h"
+#include <unistd.h>
 
 static char	sccsid[]
 	= "@(#)NNSQL(NetNews SQL) v0.5, Copyright(c) 1995, 1996 by Ke Jin";
 
 static int	yyunbindpar(yystmt_t* yystmt, int ipar);
+void	nnsql_getrange(void* hstmt, long* pmin, long* pmax);
+int	nnsql_srchtree_evl(void* hstmt);
+int	nnsql_strlike(char* str, char* pattern, char esc, int flag);
+int	nntp_postok( void* hcndes );
+void	nnsql_yyinit(yyenv_t* penv, yystmt_t* yystmt);
+int	nnsql_srchtree_tchk(void* hstmt);
+int nnsql_yyparse 	( 	yyenv_t *  	pyyenv	);
 
 void*	nnsql_allocyystmt(void* hcndes)
 {
diff -ruN unixODBC-2.3.9-orig/Drivers/template/SQLAllocHandle.c unixODBC-2.3.9/Drivers/template/SQLAllocHandle.c
--- unixODBC-2.3.9-orig/Drivers/template/SQLAllocHandle.c	2014-01-06 03:37:02.000000000 -0600
+++ unixODBC-2.3.9/Drivers/template/SQLAllocHandle.c	2020-12-09 19:26:08.000000000 -0600
@@ -13,6 +13,14 @@
 #include <config.h>
 #include "driver.h"
 
+SQLRETURN _AllocConnect(	SQLHENV    hDrvEnv,
+							SQLHDBC    *phDrvDbc );
+
+SQLRETURN _AllocEnv( SQLHENV *phDrvEnv );
+
+SQLRETURN _AllocStmt(   SQLHDBC     hDrvDbc,
+                          SQLHSTMT    *phDrvStmt );
+
 SQLRETURN  SQLAllocHandle(	SQLSMALLINT	nHandleType,
 							SQLHANDLE	nInputHandle,
 							SQLHANDLE	*pnOutputHandle )
diff -ruN unixODBC-2.3.9-orig/Drivers/template/SQLColumns.c unixODBC-2.3.9/Drivers/template/SQLColumns.c
--- unixODBC-2.3.9-orig/Drivers/template/SQLColumns.c	2014-01-06 03:37:02.000000000 -0600
+++ unixODBC-2.3.9/Drivers/template/SQLColumns.c	2020-12-09 19:27:35.000000000 -0600
@@ -22,6 +22,8 @@
 #include <config.h>
 #include "driver.h"
 
+SQLRETURN _FreeResults( HSTMTEXTRAS hStmt );
+
 enum nSQLColumns
 {
 	TABLE_CAT		= 1,
diff -ruN unixODBC-2.3.9-orig/Drivers/template/SQLExecute.c unixODBC-2.3.9/Drivers/template/SQLExecute.c
--- unixODBC-2.3.9-orig/Drivers/template/SQLExecute.c	2014-01-06 03:37:02.000000000 -0600
+++ unixODBC-2.3.9/Drivers/template/SQLExecute.c	2020-12-09 19:27:54.000000000 -0600
@@ -13,6 +13,8 @@
 #include <config.h>
 #include "driver.h"
 
+SQLRETURN _FreeResults( HSTMTEXTRAS hStmt );
+
 SQLRETURN SQLExecute( SQLHSTMT  hDrvStmt )
 {
     HDRVSTMT 	hStmt	= (HDRVSTMT)hDrvStmt;
diff -ruN unixODBC-2.3.9-orig/Drivers/template/SQLFreeConnect.c unixODBC-2.3.9/Drivers/template/SQLFreeConnect.c
--- unixODBC-2.3.9-orig/Drivers/template/SQLFreeConnect.c	2015-01-23 06:18:36.000000000 -0600
+++ unixODBC-2.3.9/Drivers/template/SQLFreeConnect.c	2020-12-09 19:28:40.000000000 -0600
@@ -16,6 +16,8 @@
 #include <config.h>
 #include "driver.h"
 
+SQLRETURN _FreeDbc( SQLHDBC hDrvDbc );
+
 SQLRETURN _FreeConnect( SQLHDBC hDrvDbc )
 {
 	HDRVDBC	hDbc	= (HDRVDBC)hDrvDbc;
diff -ruN unixODBC-2.3.9-orig/Drivers/template/SQLFreeHandle.c unixODBC-2.3.9/Drivers/template/SQLFreeHandle.c
--- unixODBC-2.3.9-orig/Drivers/template/SQLFreeHandle.c	2014-01-06 03:37:02.000000000 -0600
+++ unixODBC-2.3.9/Drivers/template/SQLFreeHandle.c	2020-12-09 19:30:24.000000000 -0600
@@ -13,6 +13,12 @@
 #include <config.h>
 #include "driver.h"
 
+SQLRETURN _FreeEnv( SQLHENV hDrvEnv );
+
+SQLRETURN _FreeConnect( SQLHDBC hDrvDbc );
+
+SQLRETURN _FreeStmt( SQLHSTMT hDrvStmt );
+
 SQLRETURN  SQLFreeHandle(    SQLSMALLINT    nHandleType,
                              SQLHANDLE      nHandle
                         )
diff -ruN unixODBC-2.3.9-orig/Drivers/template/SQLFreeStmt.c unixODBC-2.3.9/Drivers/template/SQLFreeStmt.c
--- unixODBC-2.3.9-orig/Drivers/template/SQLFreeStmt.c	2014-01-06 03:37:02.000000000 -0600
+++ unixODBC-2.3.9/Drivers/template/SQLFreeStmt.c	2020-12-09 19:30:55.000000000 -0600
@@ -13,6 +13,8 @@
 #include <config.h>
 #include "driver.h"
 
+SQLRETURN _FreeStmt( SQLHSTMT hDrvStmt );
+
 SQLRETURN SQLFreeStmt(	SQLHSTMT        hDrvStmt,
 						SQLUSMALLINT    nOption )
 {
diff -ruN unixODBC-2.3.9-orig/Drivers/template/SQLStatistics.c unixODBC-2.3.9/Drivers/template/SQLStatistics.c
--- unixODBC-2.3.9-orig/Drivers/template/SQLStatistics.c	2014-01-06 03:37:02.000000000 -0600
+++ unixODBC-2.3.9/Drivers/template/SQLStatistics.c	2020-12-09 19:32:11.000000000 -0600
@@ -13,6 +13,8 @@
 #include <config.h>
 #include "driver.h"
 
+SQLRETURN _FreeResults( HSTMTEXTRAS hStmt );
+
 enum nSQLStatistics
 {
 	TABLE_CAT		= 1,
diff -ruN unixODBC-2.3.9-orig/Drivers/template/SQLTables.c unixODBC-2.3.9/Drivers/template/SQLTables.c
--- unixODBC-2.3.9-orig/Drivers/template/SQLTables.c	2014-01-06 03:37:02.000000000 -0600
+++ unixODBC-2.3.9/Drivers/template/SQLTables.c	2020-12-09 19:32:28.000000000 -0600
@@ -13,6 +13,8 @@
 #include <config.h>
 #include "driver.h"
 
+SQLRETURN _FreeResults( HSTMTEXTRAS hStmt );
+
 /****************************
  * STANDARD COLUMNS RETURNED BY SQLTables
  ***************************/
diff -ruN unixODBC-2.3.9-orig/Drivers/template/_FreeDbc.c unixODBC-2.3.9/Drivers/template/_FreeDbc.c
--- unixODBC-2.3.9-orig/Drivers/template/_FreeDbc.c	2014-01-06 03:37:02.000000000 -0600
+++ unixODBC-2.3.9/Drivers/template/_FreeDbc.c	2020-12-09 19:33:18.000000000 -0600
@@ -6,6 +6,8 @@
 #include <config.h>
 #include "driver.h"
 
+SQLRETURN _FreeStmtList( SQLHDBC hDrvDbc );
+
 SQLRETURN _FreeDbc( SQLHDBC hDrvDbc )
 {
 	HDRVDBC		hDbc	= (HDRVDBC)hDrvDbc;
diff -ruN unixODBC-2.3.9-orig/Drivers/template/_FreeDbcList.c unixODBC-2.3.9/Drivers/template/_FreeDbcList.c
--- unixODBC-2.3.9-orig/Drivers/template/_FreeDbcList.c	2014-01-06 03:37:02.000000000 -0600
+++ unixODBC-2.3.9/Drivers/template/_FreeDbcList.c	2020-12-09 19:34:23.000000000 -0600
@@ -13,6 +13,8 @@
 #include <config.h>
 #include "driver.h"
 
+SQLRETURN _FreeDbc( SQLHDBC hDrvDbc );
+
 SQLRETURN _FreeDbcList( SQLHENV hDrvEnv )
 {
 	HDRVENV hEnv	= (HDRVENV)hDrvEnv;
diff -ruN unixODBC-2.3.9-orig/Drivers/template/_FreeStmt.c unixODBC-2.3.9/Drivers/template/_FreeStmt.c
--- unixODBC-2.3.9-orig/Drivers/template/_FreeStmt.c	2014-01-06 03:37:02.000000000 -0600
+++ unixODBC-2.3.9/Drivers/template/_FreeStmt.c	2020-12-09 19:33:56.000000000 -0600
@@ -6,6 +6,8 @@
 #include <config.h>
 #include "driver.h"
 
+SQLRETURN _FreeResults( HSTMTEXTRAS hStmt );
+
 SQLRETURN _FreeStmt( SQLHSTMT hDrvStmt )
 {
     HDRVSTMT hStmt	= (HDRVSTMT)hDrvStmt;
diff -ruN unixODBC-2.3.9-orig/Drivers/template/_FreeStmtList.c unixODBC-2.3.9/Drivers/template/_FreeStmtList.c
--- unixODBC-2.3.9-orig/Drivers/template/_FreeStmtList.c	2014-01-06 03:37:02.000000000 -0600
+++ unixODBC-2.3.9/Drivers/template/_FreeStmtList.c	2020-12-09 19:34:53.000000000 -0600
@@ -13,6 +13,8 @@
 #include <config.h>
 #include "driver.h"
 
+SQLRETURN _FreeStmt( SQLHSTMT hDrvStmt );
+
 SQLRETURN _FreeStmtList( SQLHDBC hDrvDbc )
 {
 	HDRVDBC hDbc	= (HDRVDBC)hDrvDbc;
