diff -ruN Coin-2.3.0/configure Coin-2.3.0-patched/configure
--- Coin-2.3.0/configure	2004-06-22 08:31:22.000000000 -0400
+++ Coin-2.3.0-patched/configure	2006-02-16 02:01:52.000000000 -0500
@@ -3633,6 +3633,11 @@
   sim_ac_macosx_10_2ff=true
   sim_ac_macosx_name="Panther"
   ;;
+darwin8.* )
+  sim_ac_macosx=true
+  sim_ac_macosx_10_2ff=true
+  sim_ac_macosx_name="Tiger"
+  ;;
 darwin* )
   sim_ac_macosx=true
   ;;
