diff -uNr fltk-1.1.4/Makefile fltk-1.1.4-new/Makefile
--- fltk-1.1.4/Makefile	2003-06-15 01:39:12.000000000 -0400
+++ fltk-1.1.4-new/Makefile	2005-01-31 12:46:58.000000000 -0500
@@ -25,7 +25,7 @@
 
 include makeinclude
 
-DIRS	=	src fluid test documentation
+DIRS	=	src fluid documentation
 
 all: makeinclude
 	for dir in $(DIRS); do\
diff -uNr fltk-1.1.4/configure fltk-1.1.4-new/configure
--- fltk-1.1.4/configure	2003-09-08 15:18:58.000000000 -0400
+++ fltk-1.1.4-new/configure	2005-01-31 12:58:40.000000000 -0500
@@ -1305,7 +1305,7 @@
             FLDSONAME="libfltk_forms.$FL_API_VERSION.dylib"
             GLDSONAME="libfltk_gl.$FL_API_VERSION.dylib"
             IMGDSONAME="libfltk_images.$FL_API_VERSION.dylib"
-	    DSOCOMMAND="\$(CC) $DSOFLAGS -dynamiclib -lc -o"
+	    DSOCOMMAND="\$(CXX) $DSOFLAGS -single_module -dynamiclib -lc -o"
 	    ;;
 
 	SunOS* | UNIX_S*)
@@ -6570,7 +6570,17 @@
     fi
 fi
 
-HLINKS=
+# Don't make symlinks if filesystem is not case sensitive...
+case $uname in
+    CYGWIN* | MINGW* | Darwin*)
+	HLINKS="#"
+    ;;
+    *)
+	HLINKS=
+    ;;
+esac
+
+
 POSTBUILD=:
 THREADS=
 
@@ -6818,44 +6828,6 @@
 	    THREADS="threads.exe"
 	fi
 
-	# Don't make symlinks since Windows is not case sensitive.
-	HLINKS="#"
-	;;
-
-    Darwin*)
-        # MacOS X uses Carbon for graphics...
-        LIBS="$LIBS -framework Carbon -framework ApplicationServices"
-
-	if test x$have_pthread = xyes; then
-	    cat >>confdefs.h <<\_ACEOF
-#define HAVE_PTHREAD 1
-_ACEOF
-
-	    THREADS="threads"
-	fi
-
-	if test x$enable_gl != xno; then
-            cat >>confdefs.h <<\_ACEOF
-#define HAVE_GL 1
-_ACEOF
-
-            cat >>confdefs.h <<\_ACEOF
-#define HAVE_GL_GLU_H 1
-_ACEOF
-
-            GLLIB="-framework AGL -framework OpenGL"
-        else
-	    LINKFLTKGL=""
-	    GLLIBNAME=""
-	    GLDSONAME=""
-	    GLDEMOS=""
-        fi
-
-	# Don't make symlinks because HFS+ is not case sensitive...
-	HLINKS="#"
-
-	# Add a postbuild step after linking applications
-	POSTBUILD="/Developer/Tools/Rez -t APPL -o"
 	;;
 
     *)
diff -uNr fltk-1.1.4/fluid/fluid.cxx fltk-1.1.4-new/fluid/fluid.cxx
--- fltk-1.1.4/fluid/fluid.cxx	2003-08-02 17:17:30.000000000 -0400
+++ fltk-1.1.4-new/fluid/fluid.cxx	2005-01-31 12:46:58.000000000 -0500
@@ -617,11 +617,7 @@
   fl_filename_absolute(absolute, sizeof(absolute), flname);
 
   for (i = 0; i < 10; i ++)
-#if defined(WIN32) || defined(__APPLE__)
     if (!strcasecmp(absolute, absolute_history[i])) break;
-#else
-    if (!strcmp(absolute, absolute_history[i])) break;
-#endif // WIN32 || __APPLE__
 
   if (i == 0) return;
 
diff -uNr fltk-1.1.4/src/Makefile fltk-1.1.4-new/src/Makefile
--- fltk-1.1.4/src/Makefile	2003-08-24 09:09:06.000000000 -0400
+++ fltk-1.1.4-new/src/Makefile	2005-01-31 12:46:58.000000000 -0500
@@ -216,7 +216,7 @@
 		-install_name $(libdir)/$@ \
 		-current_version 1.1.4 \
 		-compatibility_version 1.1.0 \
-		$(OBJECTS) $(LDLIBS)
+		$(OBJECTS) $(LDLIBS) $(FONTCONFIGLIB)
 	-$(RM) libfltk.dylib
 	$(LN) libfltk.1.1.dylib libfltk.dylib
 
diff -uNr fltk-1.1.4/src/filename_absolute.cxx fltk-1.1.4-new/src/filename_absolute.cxx
--- fltk-1.1.4/src/filename_absolute.cxx	2003-01-30 16:43:20.000000000 -0500
+++ fltk-1.1.4-new/src/filename_absolute.cxx	2005-01-31 12:46:58.000000000 -0500
@@ -144,11 +144,7 @@
        *slash != '\0' && *newslash != '\0';
        slash ++, newslash ++)
     if (isdirsep(*slash) && isdirsep(*newslash)) continue;
-#if defined(WIN32) || defined(__EMX__) || defined(__APPLE__)
     else if (tolower(*slash) != tolower(*newslash)) break;
-#else
-    else if (*slash != *newslash) break;
-#endif // WIN32 || __EMX__ || __APPLE__
 
   if (*newslash == '\0' && *slash != '\0' && !isdirsep(*slash))
     newslash--;
diff -uNr fltk-1.1.4/test/colbrowser.cxx fltk-1.1.4-new/test/colbrowser.cxx
--- fltk-1.1.4/test/colbrowser.cxx	2003-01-30 16:45:08.000000000 -0500
+++ fltk-1.1.4-new/test/colbrowser.cxx	2005-01-31 12:46:58.000000000 -0500
@@ -42,21 +42,7 @@
 static int load_browser(char *);
 
 /* the RGB data file does not have a standard location on unix. */
-
-#ifdef __VMS
-  static const char *rgbfile = "SYS$MANAGER:DECW$RGB.DAT";
-#else
-#ifdef __EMX__   /* OS2 */
-#include <X11/XlibInt.h>
-  static const char *rgbfile = "/XFree86/lib/X11/rgb.txt";
-#else
-#ifdef __FreeBSD__
-  static const char *rgbfile = "/usr/X11R6/lib/X11/rgb.txt";
-#else
-   static const char *rgbfile = "/usr/lib/X11/rgb.txt";
-#endif
-#endif
-#endif
+static const char *rgbfile = "/usr/X11R6/lib/X11/rgb.txt";
 
 typedef struct { int r, g, b; } RGBdb;
 
