diff -Naur ggz-client-libs-0.0.10/ggz-config-fake client-new/ggz-config-fake
--- ggz-client-libs-0.0.10/ggz-config-fake	Wed Dec 31 19:00:00 1969
+++ client-new/ggz-config-fake	Wed Feb 23 09:02:01 2005
@@ -0,0 +1,43 @@
+#!/usr/bin/perl
+
+use Getopt::Long;
+use File::Temp;
+use File::Copy;
+
+my $prefix = shift;
+my $pkg = shift;
+
+unless ( grep { /--install/ } @ARGV ) {
+	print "Passing through to real ggz-config...\n";
+	exec("$prefix/bin/ggz-config", @ARGV);
+}
+
+my $destdir = $ENV{DESTDIR} or die "No DESTDIR specified!";
+
+my $moddir = "$destdir/$prefix/share/ggz/modules/$pkg";
+
+my $modfile;
+Getopt::Long::Configure qw(bundling pass_through);
+GetOptions("modfile=s" => \$modfile);
+defined $modfile || die "No module file specified!";
+
+# Get a name
+my $destfile;
+open MOD, $modfile;
+while (<MOD>) {
+	if (/^\s*name\s*=\s*(\S+)\s*$/i) {
+		$destfile = "$moddir/\L$1\E.dsc";
+		last;
+	}
+}
+close MOD;
+
+if (!defined $destfile || -f $destfile) {
+	# Choose a better name
+	(undef, $destfile) = tempfile("moduleXXXXXX", DIR => $moddir,
+		SUFFIX => ".dsc", UNLINK => 0);
+}
+
+print "Installing module file to $destfile\n";
+system("mkdir -p \Q$moddir\E") == 0 or die "Can't create module directory";
+copy($modfile, $destfile) or die "Can't copy module file";
