diff -ruN libdnet-1.5.orig/Makefile.in libdnet-1.5/Makefile.in
--- libdnet-1.5.orig/Makefile.in	2002-08-19 23:36:08.000000000 -0600
+++ libdnet-1.5/Makefile.in	2002-12-01 15:12:42.000000000 -0700
@@ -109,7 +109,7 @@
 config.status: $(srcdir)/configure $(CONFIG_STATUS_DEPENDENCIES)
 	$(SHELL) ./config.status --recheck
 $(srcdir)/configure: $(srcdir)/configure.in $(ACLOCAL_M4) $(CONFIGURE_DEPENDENCIES)
-	cd $(srcdir) && $(AUTOCONF)
+#	cd $(srcdir) && $(AUTOCONF)
 dnet-config: $(top_builddir)/config.status dnet-config.in
 	cd $(top_builddir) && CONFIG_FILES=$@ CONFIG_HEADERS= $(SHELL) ./config.status
 
diff -ruN libdnet-1.5.orig/src/Makefile.in libdnet-1.5/src/Makefile.in
--- libdnet-1.5.orig/src/Makefile.in	2002-08-19 23:35:23.000000000 -0600
+++ libdnet-1.5/src/Makefile.in	2002-12-01 15:13:30.000000000 -0700
@@ -90,7 +90,7 @@
 libdnet_a_DEPENDENCIES =  @LIBOBJS@
 libdnet_a_OBJECTS =  addr.o blob.o ip-util.o rand.o
 AR = ar
-CFLAGS = @CFLAGS@
+CFLAGS = -fno-common @CFLAGS@
 COMPILE = $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
 LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(LDFLAGS) -o $@
@@ -138,6 +138,9 @@
 	    $(INSTALL_DATA) $$p $(DESTDIR)$(libdir)/$$p; \
 	  else :; fi; \
 	done
+	  $(INSTALL_DATA) libdnet.1.7.0.dylib $(DESTDIR)$(libdir)/libdnet.1.7.0.dylib
+	  (cd $(DESTDIR)$(libdir) && rm -f libdnet.1.dylib && ln -s libdnet.1.7.0.dylib libdnet.1.dylib) 
+	  (cd $(DESTDIR)$(libdir) && rm -f libdnet.dylib && ln -s libdnet.1.7.0.dylib libdnet.dylib) 
 	@$(POST_INSTALL)
 	@list='$(lib_LIBRARIES)'; for p in $$list; do \
 	  if test -f $$p; then \
@@ -175,6 +178,11 @@
 	-rm -f libdnet.a
 	$(AR) cru libdnet.a $(libdnet_a_OBJECTS) $(libdnet_a_LIBADD)
 	$(RANLIB) libdnet.a
+	$(CC) -dynamiclib -install_name $(prefix)/lib/libdnet.1.dylib \
+	-compatibility_version 1.0.0 -current_version 1.7.0 \
+	-o libdnet.1.7.0.dylib $(libdnet_a_OBJECTS) $(libdnet_a_LIBADD)
+	rm -f libdnet.1.dylib && ln -sf $(prefix)/lib/libdnet.1.7.0.dylib libdnet.1.dylib
+	rm -f libdnet.dylib && ln -sf $(prefix)/lib/libdnet.1.7.0.dylib libdnet.dylib
 
 tags: TAGS
 
