diff -uNr libgdiplus-1.0/src/gdip_win32.h libgdiplus-1.0-new/src/gdip_win32.h
--- libgdiplus-1.0/src/gdip_win32.h	Thu May 20 16:57:24 2004
+++ libgdiplus-1.0-new/src/gdip_win32.h	Wed Jun 30 15:29:02 2004
@@ -137,7 +137,7 @@
 
 #define BFT_BITMAP 0x4d42
 
-#ifdef __GNUC__
+#if defined(__GNUC__) && !defined(__APPLE__)
 #  define __stdcall __attribute__((__stdcall__))
 #else
 #  define __stdcall 
diff -uNr libgdiplus-1.0/src/tiffcodec.c libgdiplus-1.0-new/src/tiffcodec.c
--- libgdiplus-1.0/src/tiffcodec.c	Tue Jun 29 11:12:44 2004
+++ libgdiplus-1.0-new/src/tiffcodec.c	Wed Jun 30 15:29:02 2004
@@ -35,7 +35,7 @@
 #include <stdio.h>
 
 #ifdef HAVE_BYTESWAP_H
-#include <byteswap.h>
+/* #include <byteswap.h> */
 #endif
 
 #include "tiffcodec.h"
