diff -uNr libgdiplus-1.1.7/cairo/configure libgdiplus-1.1.7-new/cairo/configure
--- libgdiplus-1.1.7/cairo/configure	Tue May  3 14:44:58 2005
+++ libgdiplus-1.1.7-new/cairo/configure	Mon Jun  6 10:53:19 2005
@@ -22369,7 +22369,7 @@
   enableval="$enable_glitz"
   use_glitz=$enableval
 else
-  use_glitz=yes
+  use_glitz=no
 fi;
 
 if test "x$use_glitz" = "xyes"; then
diff -uNr libgdiplus-1.1.7/configure libgdiplus-1.1.7-new/configure
--- libgdiplus-1.1.7/configure	Tue May  3 14:44:36 2005
+++ libgdiplus-1.1.7-new/configure	Mon Jun  6 11:08:08 2005
@@ -19633,7 +19633,7 @@
 _ACEOF
 
 else
-	if pkg-config --exact-version 0.3.0 cairo; then
+	if pkg-config --atleast-version 0.3.0 cairo; then
 		echo Cairo installation OK
 	else
 		{ { echo "$as_me:$LINENO: error: \"Did not find Cairo == 0.3.0\"" >&5
@@ -20004,7 +20004,7 @@
 #define _REENTRANT 1
 _ACEOF
 
-		CFLAGS="-pthreads $CFLAGS"
+		CFLAGS="$CFLAGS"
 		;;
 
 	*-*-darwin*)
@@ -20014,7 +20014,7 @@
 _ACEOF
 
 		CFLAGS="-no-cpp-precomp $CFLAGS"
-		LDFLAGS="-flat_namespace -undefined suppress $LDFLAGS"
+		LDFLAGS="-Xlinker -framework -Xlinker ApplicationServices $LDFLAGS"
 		;;
 
 	*)
@@ -20037,7 +20037,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_save_CFLAGS=$CFLAGS
-   CFLAGS="$CFLAGS -pthread"
+   CFLAGS="$CFLAGS"
 
 
    ac_ext=c
@@ -20106,7 +20106,7 @@
 if test $myapp_cv_gcc_pthread = yes
 then
    myapp_threads=""
-   CFLAGS="$CFLAGS -pthread"
+   CFLAGS="$CFLAGS"
 else
    echo "$as_me:$LINENO: checking for pthread_create in -lpthread" >&5
 echo $ECHO_N "checking for pthread_create in -lpthread... $ECHO_C" >&6
@@ -21197,13 +21197,13 @@
 fi
 
   if test x$with_libungif != xno && test -z "$LIBUNGIF"; then
-    echo "$as_me:$LINENO: checking for DGifOpenFileName in -lungif" >&5
-echo $ECHO_N "checking for DGifOpenFileName in -lungif... $ECHO_C" >&6
+    echo "$as_me:$LINENO: checking for DGifOpenFileName in -lgif" >&5
+echo $ECHO_N "checking for DGifOpenFileName in -lgif... $ECHO_C" >&6
 if test "${ac_cv_lib_ungif_DGifOpenFileName+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lungif  $LIBS"
+LIBS="-lgif  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -21400,7 +21400,7 @@
 
 fi
 if test $ac_cv_header_gif_lib_h = yes; then
-  UNGIF='ungif'; LIBUNGIF='-lungif'
+  UNGIF='ungif'; LIBUNGIF='-lgif'
 else
   { echo "$as_me:$LINENO: WARNING: *** GIF loader will not be built (ungiflibrary not found) ***" >&5
 echo "$as_me: WARNING: *** GIF loader will not be built (ungiflibrary not found) ***" >&2;}
