diff -uNr libgdiplus-0.5/src/gdip_win32.h libgdiplus-0.5-patched/src/gdip_win32.h
--- libgdiplus-0.5/src/gdip_win32.h	Fri Apr 30 15:46:49 2004
+++ libgdiplus-0.5-patched/src/gdip_win32.h	Fri May 14 13:30:27 2004
@@ -136,7 +136,7 @@
 
 #define BFT_BITMAP 0x4d42
 
-#ifdef __GNUC__
+#if defined(__GNUC__) && !defined(__APPLE__)
 #  define __stdcall __attribute__((__stdcall__))
 #else
 #  define __stdcall 
diff -uNr libgdiplus-0.5/src/tiffcodec.c libgdiplus-0.5-patched/src/tiffcodec.c
--- libgdiplus-0.5/src/tiffcodec.c	Wed Apr 28 10:39:08 2004
+++ libgdiplus-0.5-patched/src/tiffcodec.c	Fri May 14 13:44:13 2004
@@ -14,7 +14,7 @@
 #endif
 
 #include <stdio.h>
-#include <byteswap.h>
+/* #include <byteswap.h> */
 #include "tiffcodec.h"
 
 #ifdef HAVE_LIBTIFF
