diff -uNr libidn-0.5.1/java/Makefile.in libidn-0.5.1-new/java/Makefile.in
--- libidn-0.5.1/java/Makefile.in	2004-07-09 09:13:05.000000000 -0400
+++ libidn-0.5.1-new/java/Makefile.in	2004-07-13 22:53:43.000000000 -0400
@@ -553,7 +553,8 @@
 @JAVA_TRUE@		gnu/inet/encoding/Stringprep.class		\
 @JAVA_TRUE@		gnu/inet/encoding/StringprepException.class
 @JAVA_TRUE@install-jar:
-@JAVA_TRUE@	ln -sf libidn-@VERSION@.jar $(libidn_jardir)/libidn.jar
+@JAVA_TRUE@	install -d -m 755 $(DESTDIR)$(libidn_jardir)
+@JAVA_TRUE@	ln -sf libidn-@VERSION@.jar $(DESTDIR)$(libidn_jardir)/libidn.jar
 
 @JAVA_TRUE@clean-local:
 @JAVA_TRUE@	rm -f $(libidn_jar_DATA)
