diff -uNr libmal-0.40/configure libmal-0.40-new/configure
--- libmal-0.40/configure	Thu Jan 29 22:00:34 2004
+++ libmal-0.40-new/configure	Tue Nov 16 11:45:59 2004
@@ -8401,7 +8401,7 @@
 
 for pilot_libs in $pilot_prefix/lib /usr/lib /usr/local/lib/ \
   /usr/extra/pilot/lib ; do
-   if test -r "$pilot_libs/libpisock.so" ; then
+   if test -r "$pilot_libs/libpisock.dylib" ; then
       pilotlibs=yes
       PILOT_LIBS="-L$pilot_libs $PILOT_LIBS"
       break
diff -uNr libmal-0.40/src/Makefile.in libmal-0.40-new/src/Makefile.in
--- libmal-0.40/src/Makefile.in	Thu Jan 29 22:00:33 2004
+++ libmal-0.40-new/src/Makefile.in	Tue Nov 16 11:49:50 2004
@@ -198,7 +198,7 @@
 
 libmal_la_SOURCES = libmal.c libmal.h $(MAL_SRC) $(MAL_HDRS)
 
-libmal_la_LDFLAGS = $(PILOT_LIBS) -module -version-info 1:0:0
+libmal_la_LDFLAGS = $(PILOT_LIBS) -ldl -version-info 1:0:0
 
 c_header_DATA = libmal.h $(MAL_HDRS)
 c_headerdir = $(includedir)/libmal
