diff -Nurd -x'*~' libole2-0.2.4.orig/doc/Makefile.in libole2-0.2.4/doc/Makefile.in
--- libole2-0.2.4.orig/doc/Makefile.in	2001-11-04 01:05:36.000000000 -0500
+++ libole2-0.2.4/doc/Makefile.in	2005-12-29 02:59:28.000000000 -0500
@@ -276,10 +276,8 @@
 	    echo '-- Installing '$$i ; \
 	    $(INSTALL_DATA) $$i $(DESTDIR)$(TARGET_DIR); \
 	  done; \
-	  echo '-- Installing $(srcdir)/html/index.sgml' ; \
-	  $(INSTALL_DATA) $(srcdir)/html/index.sgml $(DESTDIR)$(TARGET_DIR); \
 	  echo '-- Fixing Crossreferences' ; \
-	  gtkdoc-fixxref --module=$(DOC_MODULE) --html-dir=$(HTML_DIR)|| true; \
+	  gtkdoc-fixxref --module=$(DOC_MODULE) --html-dir=$(HTML_DIR) --module-dir=$(DESTDIR)$(HTML_DIR)/$(DOC_MODULE) || true; \
 	fi)
 
 dist-hook:
diff -uNr libole2-0.2.4/libole2/ms-ole.h libole2-0.2.4-new/libole2/ms-ole.h
--- libole2-0.2.4/libole2/ms-ole.h	2001-01-22 20:50:24.000000000 -0500
+++ libole2-0.2.4-new/libole2/ms-ole.h	2005-03-12 18:05:48.000000000 -0500
@@ -13,6 +13,7 @@
 
 /* This should be done in glib */
 #ifndef _WIN32
+#	include <sys/types.h>
 #	include <fcntl.h>	/* for mode_t */
 #else
 	typedef unsigned long mode_t;
