diff -Naur libxml2-2.6.4-old/catalog.c libxml2-2.6.4/catalog.c
--- libxml2-2.6.4-old/catalog.c	Sat Nov 15 09:31:22 2003
+++ libxml2-2.6.4/catalog.c	Thu Feb  5 09:04:10 2004
@@ -62,10 +62,10 @@
 #define XML_URN_PUBID "urn:publicid:"
 #define XML_CATAL_BREAK ((xmlChar *) -1)
 #ifndef XML_XML_DEFAULT_CATALOG
-#define XML_XML_DEFAULT_CATALOG "file:///etc/xml/catalog"
+#define XML_XML_DEFAULT_CATALOG "file://@PREFIX@/etc/xml/catalog"
 #endif
 #ifndef XML_SGML_DEFAULT_CATALOG
-#define XML_SGML_DEFAULT_CATALOG "file:///etc/sgml/catalog"
+#define XML_SGML_DEFAULT_CATALOG "file://@PREFIX@/etc/sgml/catalog"
 #endif
 
 static int xmlExpandCatalog(xmlCatalogPtr catal, const char *filename);
diff -Naur libxml2-2.6.4-old/doc/xmllint.1 libxml2-2.6.4/doc/xmllint.1
--- libxml2-2.6.4-old/doc/xmllint.1	Fri Sep 26 09:18:41 2003
+++ libxml2-2.6.4/doc/xmllint.1	Thu Feb  5 09:04:10 2004
@@ -160,7 +160,7 @@
 
 .TP
 \fB--catalogs\fR
-Use the catalogs from $SGML_CATALOG_FILES. Otherwise /etc/xml/catalog is used by default.
+Use the catalogs from $SGML_CATALOG_FILES. Otherwise @PREFIX@/etc/xml/catalog is used by default.
 
 .TP
 \fB--nocatalogs\fR
diff -Naur libxml2-2.6.4-old/ltmain.sh libxml2-2.6.4/ltmain.sh
--- libxml2-2.6.4-old/ltmain.sh	Wed Dec 24 06:53:01 2003
+++ libxml2-2.6.4/ltmain.sh	Thu Feb  5 09:04:10 2004
@@ -2211,7 +2211,7 @@
 	   { test "$prefer_static_libs" = no || test -z "$old_library"; }; then
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    need_relink=no
 	  fi
 	  # This is a shared library
 	
diff -Naur libxml2-2.6.4-old/update-sgml-catalog libxml2-2.6.4/update-sgml-catalog
--- libxml2-2.6.4-old/update-sgml-catalog	Wed Dec 31 19:00:00 1969
+++ libxml2-2.6.4/update-sgml-catalog	Thu Feb  5 09:04:10 2004
@@ -0,0 +1,28 @@
+#!/bin/sh -e
+#
+# update-sgml-catalog
+# see DescPackaging field of libxml2 .info file for usage.
+#
+
+CATALOG=@PREFIX@/etc/sgml/catalog
+CATALOG_D=@PREFIX@/etc/sgml/catalog.d
+
+mkdir -p ${CATALOG_D}
+
+TMPFILE=`tempfile`
+
+if [ "`echo ${CATALOG_D}/*`" != "${CATALOG_D}/*" ]; then
+  cat ${CATALOG_D}/* | sed 's/#.*$//g' | {
+    while read cat; do
+      if [ -z "${cat}" ]; then
+        continue
+      fi
+      echo CATALOG ${cat} >>${TMPFILE}
+    done
+  }
+fi
+
+mv -f ${TMPFILE} ${CATALOG}
+chmod 0644 ${CATALOG}
+
+exit 0
diff -Naur libxml2-2.6.4-old/update-xml-catalog libxml2-2.6.4/update-xml-catalog
--- libxml2-2.6.4-old/update-xml-catalog	Wed Dec 31 19:00:00 1969
+++ libxml2-2.6.4/update-xml-catalog	Thu Feb  5 09:04:10 2004
@@ -0,0 +1,29 @@
+#!/bin/sh -e
+#
+# update-xml-catalog
+# see DescPackaging field of libxml2 .info file for usage.
+#
+
+CATALOG=@PREFIX@/etc/xml/catalog
+CATALOG_D=@PREFIX@/etc/xml/catalog.d
+
+mkdir -p ${CATALOG_D}
+
+TMPFILE=`tempfile`
+xmlcatalog --create >>${TMPFILE}
+
+if [ "`echo ${CATALOG_D}/*`" != "${CATALOG_D}/*" ]; then
+  cat ${CATALOG_D}/* | sed 's/#.*$//g' | {
+    while read type orig replace; do
+      if [ -z "${type}" -o -z "${orig}" -o -z "${replace}" ]; then
+        continue
+      fi
+      eval xmlcatalog --noout --add ${type} ${orig} ${replace} ${TMPFILE}
+    done
+  }
+fi
+
+mv -f ${TMPFILE} ${CATALOG}
+chmod 0644 ${CATALOG}
+
+exit 0
diff -Naur libxml2-2.6.4-old/xmlcatalog.c libxml2-2.6.4/xmlcatalog.c
--- libxml2-2.6.4-old/xmlcatalog.c	Fri Dec 12 09:54:53 2003
+++ libxml2-2.6.4/xmlcatalog.c	Thu Feb  5 09:04:10 2004
@@ -42,7 +42,7 @@
 
 
 #ifndef XML_SGML_DEFAULT_CATALOG
-#define XML_SGML_DEFAULT_CATALOG "/etc/sgml/catalog"
+#define XML_SGML_DEFAULT_CATALOG "@PREFIX@/etc/sgml/catalog"
 #endif
 
 /************************************************************************
diff -Naur libxml2-2.6.4-old/xmllint.c libxml2-2.6.4/xmllint.c
--- libxml2-2.6.4-old/xmllint.c	Wed Dec 10 05:16:04 2003
+++ libxml2-2.6.4/xmllint.c	Thu Feb  5 09:04:10 2004
@@ -94,7 +94,7 @@
 #endif
 
 #ifndef XML_XML_DEFAULT_CATALOG
-#define XML_XML_DEFAULT_CATALOG "file:///etc/xml/catalog"
+#define XML_XML_DEFAULT_CATALOG "file://@PREFIX@/etc/xml/catalog"
 #endif
 
 #ifdef LIBXML_DEBUG_ENABLED
