--- ode-0.7.orig/configure	2006-09-08 18:34:06.000000000 -0700
+++ ode-0.7/configure	2006-09-28 16:22:07.000000000 -0700
@@ -9375,9 +9375,9 @@
     drawstuff="Win32" # if in a Windows enviroment
     ;;
   *apple* | *darwin*) # For Mac OS X
-    so_ext=".dylib"
+    so_ext=".0.dylib"
     DLLDEFINE=""
-    SHARED_LDFLAGS="-dynamiclib"
+    SHARED_LDFLAGS="-dynamiclib -install_name $libdir/libode.0.dylib"
     drawstuff="OSX"
             CC="$CXX"
     LINK="$CXXLINK"
