diff -Nurd -x'*~' freetype-2.4.4.orig/builds/unix/freetype-config.in freetype-2.4.4/builds/unix/freetype-config.in
--- freetype-2.4.4.orig/builds/unix/freetype-config.in	2009-12-21 18:23:42.000000000 -0500
+++ freetype-2.4.4/builds/unix/freetype-config.in	2011-04-23 10:59:37.000000000 -0400
@@ -144,7 +144,7 @@
   if test "$enable_shared" = "yes" ; then
     eval "rpath=\"$hardcode_libdir_flag_spec\""
   fi
-  libs="-lfreetype @LIBZ@ @FT2_EXTRA_LIBS@"
+  libs="-lfreetype"
   if test "${SYSROOT}$libdir" != "/usr/lib" && test "${SYSROOT}$libdir" != "/usr/lib64"; then
     echo -L${SYSROOT}$libdir $libs
   else
diff -Nurd -x'*~' freetype-2.4.4.orig/builds/unix/freetype2.in freetype-2.4.4/builds/unix/freetype2.in
--- freetype-2.4.4.orig/builds/unix/freetype2.in	2009-07-03 09:28:24.000000000 -0400
+++ freetype-2.4.4/builds/unix/freetype2.in	2011-04-23 10:58:18.000000000 -0400
@@ -8,5 +8,4 @@
 Version: @ft_version@
 Requires:
 Libs: -L${libdir} -lfreetype
-Libs.private: @LIBZ@ @FT2_EXTRA_LIBS@
 Cflags: -I${includedir}/freetype2 -I${includedir}
diff -Nurd -x'*~' freetype-2.4.4.orig/include/freetype/t1tables.h freetype-2.4.4/include/freetype/t1tables.h
--- freetype-2.4.4.orig/include/freetype/t1tables.h	2009-07-03 09:28:24.000000000 -0400
+++ freetype-2.4.4/include/freetype/t1tables.h	2011-04-23 10:55:23.000000000 -0400
@@ -78,6 +78,15 @@
     FT_Short    underline_position;
     FT_UShort   underline_thickness;
 
+    /*
+      2.3.8 transiently had added:
+    FT_UShort   fs_type;
+      then removed it again in 2.3.9. Field was private(?) but size of
+      struct was public, so this was backard-incompatible ABI break.
+      Fink didn't notice and pushed 2.3.8 widely. So now we add a
+      dummy member to remain at 2.3.8 size and avoid further breakage.
+    */
+    FT_UShort   fink_ABI_shim;
   } PS_FontInfoRec;
 
 
