diff -Nurd -x'*~' freetype-2.5.5.orig/builds/unix/install.mk freetype-2.5.5/builds/unix/install.mk
--- freetype-2.5.5.orig/builds/unix/install.mk	2014-11-19 16:36:25.000000000 -0500
+++ freetype-2.5.5/builds/unix/install.mk	2015-01-17 22:48:05.000000000 -0500
@@ -24,9 +24,6 @@
 
 # Unix installation and deinstallation targets.
 #
-# Note that we remove any data in the `freetype' subdirectory found in
-# `$(includedir)/freetype2', which was the previous location of the header
-# files up to version 2.5.0.
 #
 install: $(PROJECT_LIBRARY)
 	$(MKINSTALLDIRS) $(DESTDIR)$(libdir)                      \
@@ -45,10 +42,6 @@
           $(INSTALL_DATA)                                  \
             $$P $(DESTDIR)$(includedir)/freetype2/config ; \
         done
-	-$(DELETE) $(DESTDIR)$(includedir)/freetype2/freetype/config/*
-	-$(DELDIR) $(DESTDIR)$(includedir)/freetype2/freetype/config
-	-$(DELETE) $(DESTDIR)$(includedir)/freetype2/freetype/*
-	-$(DELDIR) $(DESTDIR)$(includedir)/freetype2/freetype
 	$(INSTALL_DATA) $(OBJ_BUILD)/ftconfig.h               \
           $(DESTDIR)$(includedir)/freetype2/config/ftconfig.h
 	$(INSTALL_DATA) $(OBJ_DIR)/ftmodule.h                 \
diff -Nurd -x'*~' freetype-2.5.5.orig/include/t1tables.h freetype-2.5.5/include/t1tables.h
--- freetype-2.5.5.orig/include/t1tables.h	2014-12-02 07:14:45.000000000 -0500
+++ freetype-2.5.5/include/t1tables.h	2015-01-17 15:09:49.000000000 -0500
@@ -98,6 +98,15 @@
     FT_Short    underline_position;
     FT_UShort   underline_thickness;
 
+    /*
+      2.3.8 transiently had added:
+    FT_UShort   fs_type;
+      then removed it again in 2.3.9. Field was private(?) but size of
+      struct was public, so this was backard-incompatible ABI break.
+      Fink didn't notice and pushed 2.3.8 widely. So now we add a
+      dummy member to remain at 2.3.8 size and avoid further breakage.
+    */
+    FT_UShort   fink_ABI_shim;
   } PS_FontInfoRec;
 
 
