diff -Nurd -x'*~' freetype-2.4.12.orig/builds/unix/freetype-config.in freetype-2.4.12/builds/unix/freetype-config.in
--- freetype-2.4.12.orig/builds/unix/freetype-config.in	2013-02-08 06:34:57.000000000 -0500
+++ freetype-2.4.12/builds/unix/freetype-config.in	2014-07-10 14:46:21.000000000 -0400
@@ -141,7 +141,7 @@
 fi
 
 if test "$echo_libs" = "yes" ; then
-  libs="-lfreetype %LIBZ% %LIBBZ2% %FT2_EXTRA_LIBS%"
+  libs="-lfreetype"
   if test "${SYSROOT}$libdir" != "/usr/lib"  &&
      test "${SYSROOT}$libdir" != "/usr/lib64"; then
     echo -L${SYSROOT}$libdir $libs
diff -Nurd -x'*~' freetype-2.4.12.orig/builds/unix/freetype2.in freetype-2.4.12/builds/unix/freetype2.in
--- freetype-2.4.12.orig/builds/unix/freetype2.in	2013-02-08 06:34:20.000000000 -0500
+++ freetype-2.4.12/builds/unix/freetype2.in	2014-07-10 14:47:49.000000000 -0400
@@ -8,5 +8,4 @@
 Version: %ft_version%
 Requires:
 Libs: -L${libdir} -lfreetype
-Libs.private: %LIBZ% %LIBBZ2% %FT2_EXTRA_LIBS%
 Cflags: -I${includedir}/freetype2 -I${includedir}
diff -Nurd -x'*~' freetype-2.4.12.orig/include/freetype/t1tables.h freetype-2.4.12/include/freetype/t1tables.h
--- freetype-2.4.12.orig/include/freetype/t1tables.h	2011-11-15 02:17:55.000000000 -0500
+++ freetype-2.4.12/include/freetype/t1tables.h	2014-07-10 14:45:48.000000000 -0400
@@ -78,6 +78,15 @@
     FT_Short    underline_position;
     FT_UShort   underline_thickness;
 
+    /*
+      2.3.8 transiently had added:
+    FT_UShort   fs_type;
+      then removed it again in 2.3.9. Field was private(?) but size of
+      struct was public, so this was backard-incompatible ABI break.
+      Fink didn't notice and pushed 2.3.8 widely. So now we add a
+      dummy member to remain at 2.3.8 size and avoid further breakage.
+    */
+    FT_UShort   fink_ABI_shim;
   } PS_FontInfoRec;
 
 
