--- a/zart/zart.pro.orig	2012-09-16 15:52:03.000000000 -0400
+++ b/zart/zart.pro	2012-09-16 15:52:18.000000000 -0400
@@ -3,7 +3,7 @@
 CONFIG	+= qt
 #CONFIG	+= warn_on debug
 !macx {
-	INCLUDEPATH	+= .. ./include /usr/include/opencv ../src/
+	INCLUDEPATH	+= .. ./include @FINKPREFIX@/include/opencv @FINKPREFIX@/include ../src/
 } else {
 	INCLUDEPATH	+= .. ./include /opt/local/include /opt/local/include/opencv ../src/
 }
@@ -34,7 +34,7 @@
 FORMS = ui/MainWindow.ui ui/DialogAbout.ui ui/DialogLicence.ui
 
 !macx {
-	LIBS += -lX11 ../src/libgmic.a -lcxcore -lcv -lml -lhighgui -lml -lfftw3
+	LIBS += -lX11 ../src/libgmic.a `pkg-config opencv --libs` -lfftw3
 } else {
 	LIBS += -lX11 ../src/libgmic.a `pkg-config opencv --libs` -lfftw3
 }
--- a/src/Makefile.orig	2012-09-16 15:53:48.000000000 -0400
+++ b/src/Makefile	2012-09-16 15:56:32.000000000 -0400
@@ -47,16 +47,13 @@
 # Set correct variables and paths
 #---------------------------------
 OS := $(shell uname)
-CC = g++
+CC = @FINKPREFIX@/opt/llvm-3.1/bin/clang++ -MD
 ifeq ($(OS),GNU/kFreeBSD)
 OS = Linux
 endif
 ifeq ($(OS),Darwin)
-PLUGINDIR = ~/Library/Application\ Support/Gimp/plug-ins
-USR = /opt/local
-else
 PLUGINDIR = `gimptool-2.0 --gimpplugindir`/plug-ins
-USR = /usr
+USR = @FINKPREFIX@
 endif
 ifeq ($(OS),MINGW32_NT-5.1)
 EXE = .exe
@@ -401,7 +404,7 @@
 	cp -f ../man/gmic.1.gz $(DESTDIR)$(USR)/share/man/man1/gmic.1.gz
 	cp -f ../man/gmic.1.gz $(DESTDIR)$(USR)/share/man/fr/man1/gmic.1.gz
 	if test -d /etc/bash_completion.d/; then mkdir -p $(DESTDIR)/etc/bash_completion.d/; cp -f gmic_bashcompletion.sh $(DESTDIR)/etc/bash_completion.d/gmic; fi
-	if test -d /opt/local/etc/bash_completion.d/; then mkdir -p $(DESTDIR)/opt/local/etc/bash_completion.d/; cp -f gmic_bashcompletion.sh $(DESTDIR)/opt/local/etc/bash_completion.d/gmic; fi
+	if test -d $(USR)/etc/bash_completion.d/; then mkdir -p $(DESTDIR)$(USR)/etc/bash_completion.d/; cp -f gmic_bashcompletion.sh $(DESTDIR)$(USR)/etc/bash_completion.d/gmic; fi
 
 uninstall:
 	rm -f $(DESTDIR)$(USR)/bin/gmic
