--- a/src/lib/Makefile.in.orig	2011-08-15 15:31:19.000000000 -0400
+++ b/src/lib/Makefile.in	2012-07-30 12:13:12.000000000 -0400
@@ -354,7 +354,7 @@
 @BUILD_MMX_TRUE@libImlib2_la_DEPENDENCIES = $(top_builddir)/config.h $(MMX_OBJS)
 @BUILD_AMD64_TRUE@@BUILD_MMX_FALSE@EXTRA_libImlib2_la_SOURCES = $(AMD64_SRCS)
 @BUILD_MMX_TRUE@EXTRA_libImlib2_la_SOURCES = $(MMX_SRCS) 
-libImlib2_la_LDFLAGS = -version-info @lt_version@
+libImlib2_la_LDFLAGS = -version-info @lt_version@ -no-undefined
 all: all-am
 
 .SUFFIXES:
diff -ruN imlib2-1.4.5-orig/src/modules/filters/Makefile.in imlib2-1.4.5/src/modules/filters/Makefile.in
--- imlib2-1.4.5-orig/src/modules/filters/Makefile.in	2011-08-15 15:31:19.000000000 -0400
+++ imlib2-1.4.5/src/modules/filters/Makefile.in	2012-07-30 12:24:25.000000000 -0400
@@ -269,15 +269,15 @@
 pkg_LTLIBRARIES = testfilter.la bumpmap.la colormod.la
 EXTRA_DIST = filter_common.h
 testfilter_la_SOURCES = filter_test.c
-testfilter_la_LDFLAGS = -module -avoid-version
+testfilter_la_LDFLAGS = -module -no-undefined -avoid-version
 testfilter_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 testfilter_la_LIBTOOLFLAGS = --tag=disable-static
 bumpmap_la_SOURCES = filter_bumpmap.c
-bumpmap_la_LDFLAGS = -module -avoid-version
+bumpmap_la_LDFLAGS = -module -no-undefined -avoid-version
 bumpmap_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la -lm
 bumpmap_la_LIBTOOLFLAGS = --tag=disable-static
 colormod_la_SOURCES = filter_colormod.c
-colormod_la_LDFLAGS = -module -avoid-version
+colormod_la_LDFLAGS = -module -no-undefined -avoid-version
 colormod_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 colormod_la_LIBTOOLFLAGS = --tag=disable-static
 all: all-am
diff -ruN imlib2-1.4.5-orig/src/modules/loaders/Makefile.in imlib2-1.4.5/src/modules/loaders/Makefile.in
--- imlib2-1.4.5-orig/src/modules/loaders/Makefile.in	2011-08-15 15:31:19.000000000 -0400
+++ imlib2-1.4.5/src/modules/loaders/Makefile.in	2012-07-30 12:24:25.000000000 -0400
@@ -361,56 +361,56 @@
 
 EXTRA_DIST = loader_common.h
 jpeg_la_SOURCES = loader_jpeg.c
-jpeg_la_LDFLAGS = -module -avoid-version
+jpeg_la_LDFLAGS = -module -no-undefined -avoid-version
 jpeg_la_LIBADD = @JPEGLIBS@ $(top_builddir)/src/lib/libImlib2.la
 jpeg_la_LIBTOOLFLAGS = --tag=disable-static
 png_la_SOURCES = loader_png.c
 png_la_CPPFLAGS = @PNG_CFLAGS@
-png_la_LDFLAGS = -module -avoid-version
+png_la_LDFLAGS = -module -no-undefined -avoid-version
 png_la_LIBADD = @PNG_LIBS@ $(top_builddir)/src/lib/libImlib2.la
 png_la_LIBTOOLFLAGS = --tag=disable-static
 tiff_la_SOURCES = loader_tiff.c
-tiff_la_LDFLAGS = -module -avoid-version
+tiff_la_LDFLAGS = -module -no-undefined -avoid-version
 tiff_la_LIBADD = @TIFFLIBS@ $(top_builddir)/src/lib/libImlib2.la
 tiff_la_LIBTOOLFLAGS = --tag=disable-static
 gif_la_SOURCES = loader_gif.c
-gif_la_LDFLAGS = -module -avoid-version
+gif_la_LDFLAGS = -module -no-undefined -avoid-version
 gif_la_LIBADD = @GIFLIBS@ $(top_builddir)/src/lib/libImlib2.la
 gif_la_LIBTOOLFLAGS = --tag=disable-static
 zlib_la_SOURCES = loader_zlib.c
-zlib_la_LDFLAGS = -module -avoid-version
+zlib_la_LDFLAGS = -module -no-undefined -avoid-version
 zlib_la_LIBADD = @ZLIBLIBS@ $(top_builddir)/src/lib/libImlib2.la
 zlib_la_LIBTOOLFLAGS = --tag=disable-static
 bz2_la_SOURCES = loader_bz2.c
-bz2_la_LDFLAGS = -module -avoid-version
+bz2_la_LDFLAGS = -module -no-undefined -avoid-version
 bz2_la_LIBADD = @BZ2LIBS@ $(top_builddir)/src/lib/libImlib2.la
 bz2_la_LIBTOOLFLAGS = --tag=disable-static
 id3_la_SOURCES = loader_id3.c
-id3_la_LDFLAGS = -module -avoid-version
+id3_la_LDFLAGS = -module -no-undefined -avoid-version
 id3_la_LIBADD = @ID3LIBS@ $(top_builddir)/src/lib/libImlib2.la
 id3_la_LIBTOOLFLAGS = --tag=disable-static
 pnm_la_SOURCES = loader_pnm.c
-pnm_la_LDFLAGS = -module -avoid-version
+pnm_la_LDFLAGS = -module -no-undefined -avoid-version
 pnm_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 pnm_la_LIBTOOLFLAGS = --tag=disable-static
 argb_la_SOURCES = loader_argb.c
-argb_la_LDFLAGS = -module -avoid-version
+argb_la_LDFLAGS = -module -no-undefined -avoid-version
 argb_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 argb_la_LIBTOOLFLAGS = --tag=disable-static
 bmp_la_SOURCES = loader_bmp.c
-bmp_la_LDFLAGS = -module -avoid-version
+bmp_la_LDFLAGS = -module -no-undefined -avoid-version
 bmp_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 bmp_la_LIBTOOLFLAGS = --tag=disable-static
 xpm_la_SOURCES = loader_xpm.c
-xpm_la_LDFLAGS = -module -avoid-version
+xpm_la_LDFLAGS = -module -no-undefined -avoid-version
 xpm_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 xpm_la_LIBTOOLFLAGS = --tag=disable-static
 tga_la_SOURCES = loader_tga.c
-tga_la_LDFLAGS = -module -avoid-version
+tga_la_LDFLAGS = -module -no-undefined -avoid-version
 tga_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 tga_la_LIBTOOLFLAGS = --tag=disable-static
 lbm_la_SOURCES = loader_lbm.c
-lbm_la_LDFLAGS = -module -avoid-version
+lbm_la_LDFLAGS = -module -no-undefined -avoid-version
 lbm_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 lbm_la_LIBTOOLFLAGS = --tag=disable-static
 all: all-am
