diff -ruN imlib2-1.4.6-orig/src/lib/Makefile.in imlib2-1.4.6/src/lib/Makefile.in
--- imlib2-1.4.6-orig/src/lib/Makefile.in	2013-12-21 07:42:42.000000000 -0500
+++ imlib2-1.4.6/src/lib/Makefile.in	2014-01-23 15:57:17.000000000 -0500
@@ -362,8 +362,8 @@
 AM_CFLAGS = $(CWARNFLAGS)
 AM_CPPFLAGS = -DPACKAGE_LIB_DIR=\"$(libdir)\" \
                        -I$(top_builddir) \
-                       $(X_CFLAGS) \
-                       $(FREETYPE_CFLAGS)
+                       $(FREETYPE_CFLAGS) \
+                       $(X_CFLAGS)
 
 
 # Older automakes (<1.10) do not use INCLUDES when compiling .S files
@@ -443,7 +443,7 @@
 MMX_OBJS = $(MMX_SRCS:.S=.lo)
 AMD64_OBJS = $(AMD64_SRCS:.S=.lo)
 EXTRA_DIST = $(MMX_SRCS) $(AMD64_SRCS) asm_loadimmq.S
-MY_LIBS = -lXext -lX11 $(FREETYPE_LIBS) $(DLOPEN_LIBS) -lm
+MY_LIBS = $(FREETYPE_LIBS) $(X_LIBS) -lXext -lX11 $(DLOPEN_LIBS) -lm
 @BUILD_AMD64_FALSE@@BUILD_MMX_FALSE@libImlib2_la_LIBADD = $(MY_LIBS)
 @BUILD_AMD64_TRUE@@BUILD_MMX_FALSE@libImlib2_la_LIBADD = $(AMD64_OBJS) $(MY_LIBS)
 @BUILD_MMX_TRUE@libImlib2_la_LIBADD = $(MMX_OBJS) $(MY_LIBS)
@@ -452,7 +452,7 @@
 @BUILD_MMX_TRUE@libImlib2_la_DEPENDENCIES = $(top_builddir)/config.h $(MMX_OBJS)
 @BUILD_AMD64_TRUE@@BUILD_MMX_FALSE@EXTRA_libImlib2_la_SOURCES = $(AMD64_SRCS)
 @BUILD_MMX_TRUE@EXTRA_libImlib2_la_SOURCES = $(MMX_SRCS) 
-libImlib2_la_LDFLAGS = -version-info @lt_version@
+libImlib2_la_LDFLAGS = -version-info @lt_version@ -no-undefined
 all: all-am
 
 .SUFFIXES:
diff -ruN imlib2-1.4.6-orig/src/modules/filters/Makefile.in imlib2-1.4.6/src/modules/filters/Makefile.in
--- imlib2-1.4.6-orig/src/modules/filters/Makefile.in	2013-12-21 07:42:42.000000000 -0500
+++ imlib2-1.4.6/src/modules/filters/Makefile.in	2014-01-23 15:33:11.000000000 -0500
@@ -224,7 +224,7 @@
 CCDEPMODE = @CCDEPMODE@
 CFLAGS = @CFLAGS@
 CPP = @CPP@
-CPPFLAGS = @CPPFLAGS@
+CPPFLAGS = @CPPFLAGS@ $(X_CFLAGS)
 CWARNFLAGS = @CWARNFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
@@ -353,15 +353,15 @@
 pkg_LTLIBRARIES = testfilter.la bumpmap.la colormod.la
 EXTRA_DIST = filter_common.h
 testfilter_la_SOURCES = filter_test.c
-testfilter_la_LDFLAGS = -module -avoid-version
+testfilter_la_LDFLAGS = -module -no-undefined -avoid-version
 testfilter_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 testfilter_la_LIBTOOLFLAGS = --tag=disable-static
 bumpmap_la_SOURCES = filter_bumpmap.c
-bumpmap_la_LDFLAGS = -module -avoid-version
+bumpmap_la_LDFLAGS = -module -no-undefined -avoid-version
 bumpmap_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la -lm
 bumpmap_la_LIBTOOLFLAGS = --tag=disable-static
 colormod_la_SOURCES = filter_colormod.c
-colormod_la_LDFLAGS = -module -avoid-version
+colormod_la_LDFLAGS = -module -no-undefined -avoid-version
 colormod_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 colormod_la_LIBTOOLFLAGS = --tag=disable-static
 all: all-am
diff -ruN imlib2-1.4.6-orig/src/modules/loaders/Makefile.in imlib2-1.4.6/src/modules/loaders/Makefile.in
--- imlib2-1.4.6-orig/src/modules/loaders/Makefile.in	2013-12-21 07:42:42.000000000 -0500
+++ imlib2-1.4.6/src/modules/loaders/Makefile.in	2014-01-23 15:34:40.000000000 -0500
@@ -295,7 +295,7 @@
 CCDEPMODE = @CCDEPMODE@
 CFLAGS = @CFLAGS@
 CPP = @CPP@
-CPPFLAGS = @CPPFLAGS@
+CPPFLAGS = @CPPFLAGS@ $(X_CFLAGS)
 CWARNFLAGS = @CWARNFLAGS@
 CYGPATH_W = @CYGPATH_W@
 DEFS = @DEFS@
@@ -445,56 +445,56 @@
 
 EXTRA_DIST = loader_common.h
 jpeg_la_SOURCES = loader_jpeg.c
-jpeg_la_LDFLAGS = -module -avoid-version
+jpeg_la_LDFLAGS = -module -no-undefined -avoid-version
 jpeg_la_LIBADD = @JPEGLIBS@ $(top_builddir)/src/lib/libImlib2.la
 jpeg_la_LIBTOOLFLAGS = --tag=disable-static
 png_la_SOURCES = loader_png.c
 png_la_CPPFLAGS = @PNG_CFLAGS@ $(AM_CPPFLAGS)
-png_la_LDFLAGS = -module -avoid-version
+png_la_LDFLAGS = -module -no-undefined -avoid-version
 png_la_LIBADD = @PNG_LIBS@ $(top_builddir)/src/lib/libImlib2.la
 png_la_LIBTOOLFLAGS = --tag=disable-static
 tiff_la_SOURCES = loader_tiff.c
-tiff_la_LDFLAGS = -module -avoid-version
+tiff_la_LDFLAGS = -module -no-undefined -avoid-version
 tiff_la_LIBADD = @TIFFLIBS@ $(top_builddir)/src/lib/libImlib2.la
 tiff_la_LIBTOOLFLAGS = --tag=disable-static
 gif_la_SOURCES = loader_gif.c
-gif_la_LDFLAGS = -module -avoid-version
+gif_la_LDFLAGS = -module -no-undefined -avoid-version
 gif_la_LIBADD = @GIFLIBS@ $(top_builddir)/src/lib/libImlib2.la
 gif_la_LIBTOOLFLAGS = --tag=disable-static
 zlib_la_SOURCES = loader_zlib.c
-zlib_la_LDFLAGS = -module -avoid-version
+zlib_la_LDFLAGS = -module -no-undefined -avoid-version
 zlib_la_LIBADD = @ZLIBLIBS@ $(top_builddir)/src/lib/libImlib2.la
 zlib_la_LIBTOOLFLAGS = --tag=disable-static
 bz2_la_SOURCES = loader_bz2.c
-bz2_la_LDFLAGS = -module -avoid-version
+bz2_la_LDFLAGS = -module -no-undefined -avoid-version
 bz2_la_LIBADD = @BZ2LIBS@ $(top_builddir)/src/lib/libImlib2.la
 bz2_la_LIBTOOLFLAGS = --tag=disable-static
 id3_la_SOURCES = loader_id3.c
-id3_la_LDFLAGS = -module -avoid-version
+id3_la_LDFLAGS = -module -no-undefined -avoid-version
 id3_la_LIBADD = @ID3LIBS@ $(top_builddir)/src/lib/libImlib2.la
 id3_la_LIBTOOLFLAGS = --tag=disable-static
 pnm_la_SOURCES = loader_pnm.c
-pnm_la_LDFLAGS = -module -avoid-version
+pnm_la_LDFLAGS = -module -no-undefined -avoid-version
 pnm_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 pnm_la_LIBTOOLFLAGS = --tag=disable-static
 argb_la_SOURCES = loader_argb.c
-argb_la_LDFLAGS = -module -avoid-version
+argb_la_LDFLAGS = -module -no-undefined -avoid-version
 argb_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 argb_la_LIBTOOLFLAGS = --tag=disable-static
 bmp_la_SOURCES = loader_bmp.c
-bmp_la_LDFLAGS = -module -avoid-version
+bmp_la_LDFLAGS = -module -no-undefined -avoid-version
 bmp_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 bmp_la_LIBTOOLFLAGS = --tag=disable-static
 xpm_la_SOURCES = loader_xpm.c
-xpm_la_LDFLAGS = -module -avoid-version
+xpm_la_LDFLAGS = -module -no-undefined -avoid-version
 xpm_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 xpm_la_LIBTOOLFLAGS = --tag=disable-static
 tga_la_SOURCES = loader_tga.c
-tga_la_LDFLAGS = -module -avoid-version
+tga_la_LDFLAGS = -module -no-undefined -avoid-version
 tga_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 tga_la_LIBTOOLFLAGS = --tag=disable-static
 lbm_la_SOURCES = loader_lbm.c
-lbm_la_LDFLAGS = -module -avoid-version
+lbm_la_LDFLAGS = -module -no-undefined -avoid-version
 lbm_la_LIBADD = $(top_builddir)/src/lib/libImlib2.la
 lbm_la_LIBTOOLFLAGS = --tag=disable-static
 all: all-am
diff -ruN imlib2-1.4.6-orig/src/modules/loaders/loader_xpm.c imlib2-1.4.6/src/modules/loaders/loader_xpm.c
--- imlib2-1.4.6-orig/src/modules/loaders/loader_xpm.c	2013-12-21 05:16:10.000000000 -0500
+++ imlib2-1.4.6/src/modules/loaders/loader_xpm.c	2014-01-23 15:49:14.000000000 -0500
@@ -52,7 +52,7 @@
    /* look in rgb txt database */
    if (!rgb_txt)
 #ifndef __EMX__
-      rgb_txt = fopen("/usr/share/X11/rgb.txt", "r");
+      rgb_txt = fopen("/usr/X11R6/share/X11/rgb.txt", "r");
    if (!rgb_txt)
       rgb_txt = fopen("/usr/X11R6/lib/X11/rgb.txt", "r");
    if (!rgb_txt)
diff -ruN imlib2-1.4.6-orig/src/modules/loaders/loader_gif.c imlib2-1.4.6/src/modules/loaders/loader_gif.c
--- imlib2-1.4.6-orig/src/modules/loaders/loader_gif.c	2013-12-21 04:16:10.000000000 -0600
+++ imlib2-1.4.6-orig/src/modules/loaders/loader_gif.c	2015-01-22 21:22:40.000000000 -0600
@@ -29,6 +29,7 @@
    /* already data in this image - dont load it again */
    if (im->data)
       return 0;
+
 #ifndef __EMX__
    fd = open(im->real_file, O_RDONLY);
 #else
@@ -36,12 +37,17 @@
 #endif
    if (fd < 0)
       return 0;
+#if GIFLIB_MAJOR >= 5
+   gif = DGifOpenFileHandle(fd, NULL);
+#else
    gif = DGifOpenFileHandle(fd);
+#endif
    if (!gif)
      {
         close(fd);
         return 0;
      }
+
    do
      {
         if (DGifGetRecordType(gif, &rec) == GIF_ERROR)
@@ -60,13 +66,13 @@
              h = gif->Image.Height;
              if (!IMAGE_DIMENSIONS_OK(w, h))
                {
-                  DGifCloseFile(gif);
+                  DGifCloseFile(gif, NULL);
                   return 0;
                }
              rows = malloc(h * sizeof(GifRowType *));
              if (!rows)
                {
-                  DGifCloseFile(gif);
+                  DGifCloseFile(gif, NULL);
                   return 0;
                }
              for (i = 0; i < h; i++)
@@ -78,7 +84,7 @@
                   rows[i] = malloc(w * sizeof(GifPixelType));
                   if (!rows[i])
                     {
-                       DGifCloseFile(gif);
+                       DGifCloseFile(gif, NULL);
                        for (i = 0; i < h; i++)
                          {
                             if (rows[i])
@@ -150,7 +156,7 @@
         im->data = (DATA32 *) malloc(sizeof(DATA32) * w * h);
         if (!im->data)
           {
-             DGifCloseFile(gif);
+             DGifCloseFile(gif, NULL);
              free(rows);
              return 0;
           }
@@ -181,7 +187,7 @@
                        last_per = (int)per;
                        if (!(progress(im, (int)per, 0, last_y, w, i)))
                          {
-                            DGifCloseFile(gif);
+                            DGifCloseFile(gif, NULL);
                             for (i = 0; i < h; i++)
                               {
                                  free(rows[i]);
@@ -198,7 +204,7 @@
      {
         progress(im, 100, 0, last_y, w, h);
      }
-   DGifCloseFile(gif);
+   DGifCloseFile(gif, NULL);
    for (i = 0; i < h; i++)
      {
         free(rows[i]);
