=== modified file 'configure.ac'
--- a/configure.ac	2011-07-02 20:43:05 +0000
+++ b/configure.ac	2011-07-10 17:54:41 +0000
@@ -323,7 +323,6 @@
   platform_osx=no
 fi
 AC_MSG_RESULT([$platform_osx])
-AM_CONDITIONAL(PLATFORM_OSX, test "$platform_osx" = "yes")
 	
 AC_MSG_CHECKING([for Solaris platform])
 case "$host" in

=== modified file 'src/Makefile.am'
--- a/src/Makefile.am	2011-07-05 20:22:31 +0000
+++ b/src/Makefile.am	2011-07-10 17:54:41 +0000
@@ -205,11 +205,7 @@
 
 inkscape_SOURCES += main.cpp $(win32_sources)
 inkscape_LDADD = $(all_libs)
-if PLATFORM_OSX
-inkscape_LDFLAGS = --export-dynamic $(kdeldflags) $(mwindows)
-else
-inkscape_LDFLAGS = -Wl,--export-dynamic $(kdeldflags) $(mwindows)
-endif
+inkscape_LDFLAGS = $(kdeldflags) $(mwindows)
 
 inkview_SOURCES += inkview.cpp $(win32_sources)
 inkview_LDADD = $(all_libs)

