diff -Nurd -x'*~' ffmpeg-0.8.15.orig/configure ffmpeg-0.8.15/configure
--- ffmpeg-0.8.15.orig/configure	2013-10-06 12:18:07.000000000 -0400
+++ ffmpeg-0.8.15/configure	2015-02-13 17:46:19.000000000 -0500
@@ -856,6 +856,26 @@
     add_extralibs $(get_safe ${pkg}_libs)
 }
 
+require_libfreetype(){
+    log require_libfreetype "$@"
+    pkg="freetype2"
+    check_cmd $pkg_config --exists --print-errors $pkg \
+      || die "ERROR: $pkg not found"
+    pkg_cflags=$($pkg_config --cflags $pkg)
+    pkg_libs=$($pkg_config --libs $pkg)
+    {
+        echo "#include <ft2build.h>"
+        echo "#include FT_FREETYPE_H"
+        echo "long check_func(void) { return (long) FT_Init_FreeType; }"
+        echo "int main(void) { return 0; }"
+    } | check_ld $pkg_cflags $pkg_libs \
+      && set_safe ${pkg}_cflags $pkg_cflags \
+      && set_safe ${pkg}_libs   $pkg_libs \
+      || die "ERROR: $pkg not found"
+    add_cflags    $(get_safe ${pkg}_cflags)
+    add_extralibs $(get_safe ${pkg}_libs)
+}
+
 check_host_cc(){
     log check_host_cc "$@"
     cat > $TMPC
@@ -2932,7 +2952,7 @@
     "libdirac_decoder/dirac_parser.h libdirac_encoder/dirac_encoder.h"  \
     "dirac_decoder_init dirac_encoder_init"
 enabled libfaac    && require2 libfaac "stdint.h faac.h" faacEncGetVersion -lfaac
-enabled libfreetype && require_pkg_config freetype2 "ft2build.h freetype/freetype.h" FT_Init_FreeType
+enabled libfreetype && require_libfreetype
 enabled libgsm     && require  libgsm gsm/gsm.h gsm_create -lgsm
 enabled libmp3lame && require  "libmp3lame >= 3.98.3" lame/lame.h lame_set_VBR_quality -lmp3lame
 enabled libnut     && require  libnut libnut.h nut_demuxer_init -lnut
diff -Nurd -x'*~' ffmpeg-0.8.15.orig/libavfilter/vf_drawtext.c ffmpeg-0.8.15/libavfilter/vf_drawtext.c
--- ffmpeg-0.8.15.orig/libavfilter/vf_drawtext.c	2013-10-06 12:18:09.000000000 -0400
+++ ffmpeg-0.8.15/libavfilter/vf_drawtext.c	2015-02-13 17:45:31.000000000 -0500
@@ -41,7 +41,6 @@
 #undef time
 
 #include <ft2build.h>
-#include <freetype/config/ftheader.h>
 #include FT_FREETYPE_H
 #include FT_GLYPH_H
 
--- ffmpeg-0.8.15.orig/doc/protocols.texi	2013-10-06 11:18:07.000000000 -0500
+++ ffmpeg-0.8.15/doc/protocols.texi	2015-02-21 05:17:09.000000000 -0600
@@ -155,8 +155,8 @@
 
 Real-Time Messaging Protocol.
 
-The Real-Time Messaging Protocol (RTMP) is used for streaming multime‐
-dia content across a TCP/IP network.
+The Real-Time Messaging Protocol (RTMP) is used for streaming multimedia
+content across a TCP/IP network.
 
 The required syntax is:
 @example
