diff -Nurd -x'*~' ffmpeg-1.0.9.orig/configure ffmpeg-1.0.9/configure
--- ffmpeg-1.0.9.orig/configure	2014-03-10 09:12:52.000000000 -0400
+++ ffmpeg-1.0.9/configure	2015-02-14 14:36:31.000000000 -0500
@@ -1029,6 +1029,26 @@
     eval printf '%s\\n' $HOSTCC_O
 }
 
+require_libfreetype(){
+    log require_libfreetype "$@"
+    pkg="freetype2"
+    check_cmd $pkg_config --exists --print-errors $pkg \
+      || die "ERROR: $pkg not found"
+    pkg_cflags=$($pkg_config --cflags $pkg)
+    pkg_libs=$($pkg_config --libs $pkg)
+    {
+        echo "#include <ft2build.h>"
+        echo "#include FT_FREETYPE_H"
+        echo "long check_func(void) { return (long) FT_Init_FreeType; }"
+        echo "int main(void) { return 0; }"
+    } | check_ld "cc" $pkg_cflags $pkg_libs \
+      && set_safe ${pkg}_cflags $pkg_cflags \
+      && set_safe ${pkg}_libs   $pkg_libs \
+      || die "ERROR: $pkg not found"
+    add_cflags    $(get_safe ${pkg}_cflags)
+    add_extralibs $(get_safe ${pkg}_libs)
+}
+
 check_host_cc(){
     log check_host_cc "$@"
     cat > $TMPC
@@ -3648,7 +3668,7 @@
 enabled libfdk_aac && require  libfdk_aac fdk-aac/aacenc_lib.h aacEncOpen -lfdk-aac
 flite_libs="-lflite_cmu_time_awb -lflite_cmu_us_awb -lflite_cmu_us_kal -lflite_cmu_us_kal16 -lflite_cmu_us_rms -lflite_cmu_us_slt -lflite_usenglish -lflite_cmulex -lflite"
 enabled libflite   && require2 libflite "flite/flite.h" flite_init $flite_libs
-enabled libfreetype && require_pkg_config freetype2 "ft2build.h freetype/freetype.h" FT_Init_FreeType
+enabled libfreetype && require_libfreetype
 enabled libgsm     && require  libgsm gsm/gsm.h gsm_create -lgsm
 enabled libilbc    && require  libilbc ilbc.h WebRtcIlbcfix_InitDecode -lilbc
 enabled libmodplug && require  libmodplug libmodplug/modplug.h ModPlug_Load -lmodplug
diff -Nurd -x'*~' ffmpeg-1.0.9.orig/libavfilter/vf_drawtext.c ffmpeg-1.0.9/libavfilter/vf_drawtext.c
--- ffmpeg-1.0.9.orig/libavfilter/vf_drawtext.c	2014-03-10 09:12:42.000000000 -0400
+++ ffmpeg-1.0.9/libavfilter/vf_drawtext.c	2015-02-14 14:35:38.000000000 -0500
@@ -49,7 +49,6 @@
 #undef time
 
 #include <ft2build.h>
-#include <freetype/config/ftheader.h>
 #include FT_FREETYPE_H
 #include FT_GLYPH_H
 #if CONFIG_FONTCONFIG
