--- a/configure.orig	2011-11-15 16:14:42.000000000 -0500
+++ b/configure	2011-11-15 14:02:14.000000000 -0500
@@ -6240,17 +6240,17 @@
   def_speex='#undef CONFIG_SPEEX'
   nocodecmodules="speex $nocodecmodules"
 fi
 echores "$_speex"
 
 echocheck "libgsm"
 if test "$_libgsm" = auto ; then
   _libgsm=no
-  statement_check gsm/gsm.h 'gsm_create()' -lgsm && _libgsm=yes
+  statement_check gsm.h 'gsm_create()' -lgsm && _libgsm=yes
 fi
 if test "$_libgsm" = yes ; then
   def_libgsm='#define CONFIG_LIBGSM 1'
   extra_ldflags="$extra_ldflags -lgsm"
   libavencoders="$libavencoders LIBGSM_ENCODER LIBGSM_MS_ENCODER"
   libavdecoders="$libavdecoders LIBGSM_DECODER LIBGSM_MS_DECODER"
   codecmodules="libgsm $codecmodules"
 else
--- a/ffmpeg/libavcodec/libgsm.c.orig	2011-11-15 13:41:47.000000000 -0500
+++ b/ffmpeg/libavcodec/libgsm.c	2011-11-15 16:18:44.000000000 -0500
@@ -23,17 +23,17 @@
 
 /**
  * @file
  * Interface to libgsm for gsm encoding/decoding
  */
 
 // The idiosyncrasies of GSM-in-WAV are explained at http://kbs.cs.tu-berlin.de/~jutta/toast.html
 
-#include <gsm/gsm.h>
+#include <gsm.h>
 
 #include "avcodec.h"
 #include "internal.h"
 #include "gsm.h"
 
 static av_cold int libgsm_encode_close(AVCodecContext *avctx) {
 #if FF_API_OLD_ENCODE_AUDIO
     av_freep(&avctx->coded_frame);
