diff -Naur netpbm-10.47.44-orig/Makefile.config.fink netpbm-10.47.44/Makefile.config.fink
--- netpbm-10.47.44-orig/Makefile.config.fink	1969-12-31 19:00:00.000000000 -0500
+++ netpbm-10.47.44/Makefile.config.fink	2013-06-03 10:45:10.000000000 -0400
@@ -0,0 +1,22 @@
+#
+# Configuration for fink
+# Copied from a manual run of ./configure (plus some additions)
+
+DEFAULT_TARGET	= nonmerge
+NETPBMLIBTYPE	= dylib
+NETPBMLIBSUFFIX	= dylib
+STATICLIB_TOO	= n
+NETPBMLIB_RUNTIME_PATH = @PREFIX@/lib
+CC		= cc -no-cpp-precomp
+CFLAGS = -O3 -ffast-math  -pedantic -fno-common -Wall -Wno-uninitialized -Wmissing-declarations -Wimplicit -Wwrite-strings -Wmissing-prototypes -Wundef -MD
+CFLAGS_SHLIB	= -fno-common
+LDSHLIB		= -dynamiclib -install_name $(NETPBMLIB_RUNTIME_PATH)/libnetpbm.$(MAJ).dylib -compatibility_version $$(($(MAJ)+1)) -current_version $$(($(MAJ)+1)).$(MIN)
+LINKERISCOMPILER	= Y
+CFLAGS_SHLIB	+= -fPIC
+TIFFLIB		= libtiff.dylib
+JPEGLIB		= libjpeg.dylib
+PNGLIB		= libpng.dylib
+ZLIB		= libz.dylib
+JBIGLIB		= libjbig.dylib
+JASPERLIB       = libjasper.dylib
+
diff -Naur netpbm-10.47.44-orig/buildtools/installnetpbm.pl netpbm-10.47.44/buildtools/installnetpbm.pl
--- netpbm-10.47.44-orig/buildtools/installnetpbm.pl	2013-06-03 10:18:28.000000000 -0400
+++ netpbm-10.47.44/buildtools/installnetpbm.pl	2013-06-03 10:48:01.000000000 -0400
@@ -117,7 +117,7 @@
     } elsif ($ENV{OSTYPE} && $ENV{OSTYPE} eq "msdosdjgpp") {
         $default = "/dev/env/DJDIR";
     } else {
-        $default = "/usr/local/netpbm";
+        $default = "@PREFIX@";
     }
 
     my $response = prompt("install prefix", $default);
@@ -584,7 +584,7 @@
     my $manDir;
 
     while (!$manDir) {
-        my $default = "$prefix/man";
+        my $default = "$prefix/share/man";
 
         my $response = prompt("man page directory", $default);
 
@@ -783,9 +783,9 @@
 
     print("\n");
 
-    removeObsoleteManPage($manDir);
+#    removeObsoleteManPage($manDir);
 
-    makeInManwebPath("$manDir/web");
+#    makeInManwebPath("$manDir/web");
     
     $$mandirR = $manDir;
 }
diff -Naur netpbm-10.47.44-orig/converter/other/pnmtopng.c netpbm-10.47.44/converter/other/pnmtopng.c
--- netpbm-10.47.44-orig/converter/other/pnmtopng.c	2013-06-03 10:18:31.000000000 -0400
+++ netpbm-10.47.44/converter/other/pnmtopng.c	2013-06-03 10:49:18.000000000 -0400
@@ -529,7 +529,7 @@
                       unsigned int * const bestMatchP) {
     
     unsigned int paletteIndex;
-    unsigned int bestIndex;
+    unsigned int bestIndex = 0;
     unsigned int bestMatch;
 
     assert(paletteSize > 0);
@@ -1464,9 +1464,9 @@
     colorhist_vector chv;
     unsigned int colors;
 
-    gray * alphas_of_color[MAXPALETTEENTRIES];
+    gray * alphas_of_color[MAXPALETTEENTRIES + 1];
     unsigned int alphas_first_index[MAXPALETTEENTRIES];
-    unsigned int alphas_of_color_cnt[MAXPALETTEENTRIES];
+    unsigned int alphas_of_color_cnt[MAXPALETTEENTRIES + 1];
  
     getChv(ifP, rasterPos, cols, rows, maxval, format, MAXCOLORS, 
            &chv, &colors);
@@ -2385,7 +2385,7 @@
       /* The color part of the color/alpha palette passed to the PNG
          compressor 
       */
-  unsigned int palette_size;
+  unsigned int palette_size = MAXCOLORS;
 
   gray trans_pnm[MAXCOLORS];
   png_byte  trans[MAXCOLORS];
diff -Naur netpbm-10.47.44-orig/pm_config.in.h netpbm-10.47.44/pm_config.in.h
--- netpbm-10.47.44-orig/pm_config.in.h	2013-06-03 10:18:34.000000000 -0400
+++ netpbm-10.47.44/pm_config.in.h	2013-06-03 10:51:28.000000000 -0400
@@ -61,8 +61,8 @@
 
 #define RGB_DB_PATH \
-"/usr/share/netpbm/rgb.txt:" \
+"@PREFIX@/share/netpbm/rgb.txt:" \
 "/usr/lib/X11/rgb.txt:" \
-"/usr/share/X11/rgb.txt:" \
+"/usr/X11/share/X11/rgb.txt:" \
 "/usr/X11R6/lib/X11/rgb.txt"
 
 /* CONFIGURE: This is the name of an environment variable that tells
