diff -Nurd -x'*~' py2cairo-1.10.0.orig/configure.ac py2cairo-1.10.0/configure.ac
--- py2cairo-1.10.0.orig/configure.ac	2011-05-01 06:43:49.000000000 -0400
+++ py2cairo-1.10.0/configure.ac	2013-01-07 17:04:01.000000000 -0500
@@ -43,13 +43,23 @@
 export_dynamic=`(./libtool --config; echo eval echo \\$export_dynamic_flag_spec) | sh`
 
 # cairo
-PKG_CHECK_MODULES(CAIRO, cairo >= cairo_required_version)
+cairo_yes=""
+cairo_no=""
+PKG_CHECK_MODULES(CAIRO, cairo >= cairo_required_version, cairo_yes="$cairo_yes cairo", cairo_no="$cairo_no cairo")
+PKG_CHECK_MODULES(CAIRO_PDF, cairo-pdf >= cairo_required_version, cairo_yes="$cairo_yes cairo-pdf", cairo_no="$cairo_no cairo-pdf")
+PKG_CHECK_MODULES(CAIRO_PS, cairo-ps >= cairo_required_version, cairo_yes="$cairo_yes cairo-ps", cairo_no="$cairo_no cairo-ps")
+PKG_CHECK_MODULES(CAIRO_SVG, cairo-svg >= cairo_required_version, cairo_yes="$cairo_yes cairo-svg", cairo_no="$cairo_no cairo-svg")
+PKG_CHECK_MODULES(CAIRO_WIN32, cairo-win32 >= cairo_required_version, cairo_yes="$cairo_yes cairo-win32", cairo_no="$cairo_no cairo-win32")
+PKG_CHECK_MODULES(CAIRO_XCB, cairo-xcb >= cairo_required_version, cairo_yes="$cairo_yes cairo-xcb", cairo_no="$cairo_no cairo-xcb")
+PKG_CHECK_MODULES(CAIRO_XLIB, cairo-xlib >= cairo_required_version, cairo_yes="$cairo_yes cairo-xlib", cairo_no="$cairo_no cairo-xlib")
+CAIRO_LIBS="$CAIRO_LIBS $CAIRO_PDF_LIBS $CAIRO_PS_LIBS $CAIRO_SVG_LIBS $CAIRO_WIN32_LIBS $CAIRO_XCB_LIBS $CAIRO_XLIB_LIBS"
+CAIRO_CFLAGS="$CAIRO_CFLAGS $CAIRO_PDF_CFLAGS $CAIRO_PS_CFLAGS $CAIRO_SVG_CFLAGS $CAIRO_WIN32_CFLAGS $CAIRO_XCB_CFLAGS $CAIRO_XLIB_CFLAGS"
 if test -n "$export_dynamic"; then
   CAIRO_LIBS=`echo $CAIRO_LIBS | sed -e "s/$export_dynamic//"`
 fi
 
 # xpyb
-PKG_CHECK_MODULES(XPYB, xpyb >= xpyb_required_version,[have_xpyb=yes],[have_xpyb=no])
+PKG_CHECK_MODULES(XPYB, xpyb >= xpyb_required_version,[have_xpyb=yes;cairo_yes="$cairo_yes xpyb"],[have_xpyb=no;cairo_no="$cairo_no xpyb"])
 AS_IF([test "x$have_xpyb" = "xyes"],
       [AC_DEFINE([HAVE_XPYB], [1], [Define to 1 if xpyb is available])])
 
@@ -111,7 +121,16 @@
 
 AC_OUTPUT
 
+cairo_want=" cairo cairo-pdf cairo-ps cairo-svg cairo-xlib"
 echo "
 Configuration:
 Installation prefix             ${prefix}
+Cairo intended                  '${cairo_want}'
+Cairo includes                  '${cairo_yes}'
+Cairo excludes                  '${cairo_no}'
 "
+
+if test "x$cairo_want" != "x$cairo_yes"; then
+   echo "cairo bits mismatch"
+   exit 1
+fi
