diff -ruN vigra-1.9.0-orig/vigranumpy/src/core/CMakeLists.txt vigra-1.9.0/vigranumpy/src/core/CMakeLists.txt
--- vigra-1.9.0-orig/config/VIGRA_ADD_NUMPY_MODULE.cmake	2012-11-06 07:34:42.000000000 -0500
+++ vigra-1.9.0/config/VIGRA_ADD_NUMPY_MODULE.cmake	2013-02-09 17:08:34.000000000 -0500
@@ -49,7 +49,7 @@
         set(LIBRARY_NAME ${target})
     ENDIF()
 
-    ADD_LIBRARY(${TARGET_NAME} SHARED ${SOURCES})    
+    ADD_LIBRARY(${TARGET_NAME} MODULE ${SOURCES})    
     
     IF(PART_OF_VIGRANUMPY)
         ADD_DEPENDENCIES(vigranumpy ${TARGET_NAME})
