diff -Nurd -x'*~' docbook-utils-0.6.14.orig/backends/txt docbook-utils-0.6.14/backends/txt
--- docbook-utils-0.6.14.orig/backends/txt	2004-02-11 08:58:03.000000000 -0500
+++ docbook-utils-0.6.14/backends/txt	2005-11-08 13:35:42.000000000 -0500
@@ -14,6 +14,10 @@
 then
   CONVERT=/usr/bin/w3m
   ARGS="-dump"
+elif [ -x /usr/bin/elinks ]
+then
+  CONVERT=/usr/bin/elinks
+  ARGS="-force-html -dump -no-numbering"
 else
   echo >&2 "No way to convert HTML to text found."
   exit 1
diff -Nurd docbook-utils-0.6.14.orig/configure.in docbook-utils-0.6.14/configure.in
--- docbook-utils-0.6.14.orig/configure.in	2004-02-11 09:14:15.000000000 -0500
+++ docbook-utils-0.6.14/configure.in	2015-11-13 17:42:55.000000000 -0500
@@ -16,8 +16,6 @@
 
 AM_INIT_AUTOMAKE(docbook-utils, 0.6.14)
 
-docdir='$(prefix)/doc'
-AC_SUBST(docdir)
 
 dnl Checks for programs.
 jade_bindirs="/usr/bin /usr/local/bin"
diff -Nurd docbook-utils-0.6.14.orig/doc/Makefile.am docbook-utils-0.6.14/doc/Makefile.am
--- docbook-utils-0.6.14.orig/doc/Makefile.am	2001-05-23 11:12:49.000000000 -0400
+++ docbook-utils-0.6.14/doc/Makefile.am	2015-11-13 17:53:29.000000000 -0500
@@ -3,7 +3,7 @@
 EXTRA_DIST = docbook-utils.sgml
 
 install-data-local:
-	cd $(DESTDIR)/$(mandir)/man1 && ( \
+	cd $(DESTDIR)$(mandir)/man1 && ( \
 	ln -sf jw.1 docbook2dvi.1; \
 	ln -sf jw.1 docbook2html.1; \
 	ln -sf jw.1 docbook2man.1; \
