diff -Nurd -x'*~' epstool-3.09.orig/makefile epstool-3.09/makefile
--- epstool-3.09.orig/makefile	2015-03-15 11:25:28.000000000 +0100
+++ epstool-3.09/makefile	2019-03-28 15:19:47.000000000 +0100
@@ -40,10 +40,10 @@
 
 include $(SRCDIR)/common.mak
 
-EPSTOOL_ROOT=/usr/local
+EPSTOOL_ROOT=@PREFIX@
 EPSTOOL_BASE=$(prefix)$(EPSTOOL_ROOT)
-EPSTOOL_DOCDIR=$(EPSTOOL_BASE)/share/doc/epstool-$(EPSTOOL_VERSION)
-EPSTOOL_MANDIR=$(EPSTOOL_BASE)/man
+EPSTOOL_DOCDIR=$(EPSTOOL_BASE)/share/doc/epstool
+EPSTOOL_MANDIR=$(EPSTOOL_BASE)/share/man
 EPSTOOL_BINDIR=$(EPSTOOL_BASE)/bin
 
 epstool: $(BD)epstool$(EXE)
