diff -Nurd ghostscript-9.53.1.orig/base/sjpx_openjpeg.c ghostscript-9.53.1/base/sjpx_openjpeg.c
--- ghostscript-9.53.1.orig/base/sjpx_openjpeg.c	2020-09-14 14:40:08.000000000 +0100
+++ ghostscript-9.53.1/base/sjpx_openjpeg.c	2020-09-19 20:36:13.000000000 +0100
@@ -25,7 +25,6 @@
 #include "gxsync.h"
 #include "assert_.h"
 #if !defined(SHARE_JPX) || (SHARE_JPX == 0)
-#include "opj_malloc.h"
 #endif
 /* Some locking to get around the criminal lack of context
  * in the openjpeg library. */
diff -Nurd ghostscript-9.53.1.orig/base/stdpre.h ghostscript-9.53.1/base/stdpre.h
--- ghostscript-9.53.1.orig/base/stdpre.h	2020-09-14 14:40:08.000000000 +0100
+++ ghostscript-9.53.1/base/stdpre.h	2020-09-19 20:36:13.000000000 +0100
@@ -316,7 +316,9 @@
  * header file that includes sys/types.h.
  *
  */
-#define bool bool_		/* (maybe not needed) */
+#if !defined(bool)
+#  define bool bool_		/* (maybe not needed) */
+#endif
 #define uchar uchar_
 #define uint uint_
 #define ushort ushort_
diff -Nurd ghostscript-9.53.1.orig/base/unistd_.h ghostscript-9.53.1/base/unistd_.h
--- ghostscript-9.53.1.orig/base/unistd_.h	2020-09-14 14:40:08.000000000 +0100
+++ ghostscript-9.53.1/base/unistd_.h	2020-09-19 20:36:13.000000000 +0100
@@ -52,7 +52,9 @@
    /* _XOPEN_SOURCE 500 define is needed to get
     * access to pread and pwrite */
 #  define _XOPEN_SOURCE 500
-#  define __USE_UNIX98
+#  if !defined(__USE_UNIX98)
+#    define __USE_UNIX98
+#  endif
 #  include <unistd.h>
 #endif
 
diff -Nurd ghostscript-9.53.1.orig/configure ghostscript-9.53.1/configure
--- ghostscript-9.53.1.orig/configure	2020-09-14 14:40:41.000000000 +0100
+++ ghostscript-9.53.1/configure	2020-09-19 20:36:13.000000000 +0100
@@ -8238,7 +8238,7 @@
 $as_echo_n "checking for local zlib source... " >&6; }
 # we must define ZLIBDIR regardless because png.mak does a -I$(ZLIBDIR)
 # this seems a harmless default
-ZLIBDIR=src
+ZLIBDIR=$oldincludedir
 AUX_SHARED_ZLIB=
 ZLIBCFLAGS=""
 
diff -Nurd ghostscript-9.53.1.orig/configure.ac ghostscript-9.53.1/configure.ac
--- ghostscript-9.53.1.orig/configure.ac	2020-09-14 14:40:08.000000000 +0100
+++ ghostscript-9.53.1/configure.ac	2020-09-19 20:36:13.000000000 +0100
@@ -1363,7 +1363,7 @@
 dnl zlib is needed for language level 3, and libpng
 # we must define ZLIBDIR regardless because png.mak does a -I$(ZLIBDIR)
 # this seems a harmless default
-ZLIBDIR=src
+ZLIBDIR=$oldincludedir
 AUX_SHARED_ZLIB=
 ZLIBCFLAGS=""
 
diff -Nurd ghostscript-9.53.1.orig/devices/gdevpng.c ghostscript-9.53.1/devices/gdevpng.c
--- ghostscript-9.53.1.orig/devices/gdevpng.c	2020-09-14 14:40:08.000000000 +0100
+++ ghostscript-9.53.1/devices/gdevpng.c	2020-09-19 20:36:13.000000000 +0100
@@ -760,6 +760,7 @@
 
 #if PNG_LIBPNG_VER_MINOR < 5
 
+#if 0 /* not required in 1.5 */
 /*
  * Patch around a static reference to a never-used procedure.
  * This could be avoided if we were willing to edit pngconf.h to
@@ -780,6 +781,7 @@
 }
 #endif
 #endif
+#endif
 
 static int
 pngalpha_open(gx_device * pdev)
