diff -uNr ghostscript-9.20.orig/base/mkromfs.c ghostscript-9.20/base/mkromfs.c
--- ghostscript-9.20.orig/base/mkromfs.c	2016-09-26 06:41:28.000000000 -0400
+++ ghostscript-9.20/base/mkromfs.c	2016-10-03 14:32:24.000000000 -0400
@@ -2336,7 +2336,7 @@
 #endif
     fprintf(out,"\n#include \"stdint_.h\"\n");
     fprintf(out,"\n#include \"time_.h\"\n\n");
-    fprintf(out,"    time_t gs_romfs_buildtime = %ld;\n\n", time(NULL));
+    fprintf(out,"    time_t gs_romfs_buildtime = %lld;\n\n", time(NULL));
 
     /* process the remaining arguments (options interspersed with paths) */
     for (; atarg < argc; atarg++) {
diff -uNr ghostscript-9.20.orig/base/stdpre.h ghostscript-9.20/base/stdpre.h
--- ghostscript-9.20.orig/base/stdpre.h	2016-09-26 06:41:28.000000000 -0400
+++ ghostscript-9.20/base/stdpre.h	2016-10-03 14:32:24.000000000 -0400
@@ -271,7 +271,9 @@
  * header file that includes sys/types.h.
  *
  */
-#define bool bool_		/* (maybe not needed) */
+#if !defined(bool)
+#    define bool bool_		/* (maybe not needed) */
+#endif
 #define uchar uchar_
 #define uint uint_
 #define ushort ushort_
diff -uNr ghostscript-9.20.orig/base/unistd_.h ghostscript-9.20/base/unistd_.h
--- ghostscript-9.20.orig/base/unistd_.h	2016-09-26 06:41:28.000000000 -0400
+++ ghostscript-9.20/base/unistd_.h	2016-10-03 14:32:24.000000000 -0400
@@ -53,7 +53,9 @@
    /* _XOPEN_SOURCE 500 define is needed to get
     * access to pread and pwrite */
 #  define _XOPEN_SOURCE 500
-#  define __USE_UNIX98
+#if !defined(__USE_UNIX98)
+#    define __USE_UNIX98
+# endif
 #  include <unistd.h>
 #endif
 
diff -uNr ghostscript-9.20.orig/configure ghostscript-9.20/configure
--- ghostscript-9.20.orig/configure	2016-09-26 06:41:58.000000000 -0400
+++ ghostscript-9.20/configure	2016-10-03 14:32:24.000000000 -0400
@@ -6820,7 +6820,7 @@
 $as_echo_n "checking for local zlib source... " >&6; }
 # we must define ZLIBDIR regardless because png.mak does a -I$(ZLIBDIR)
 # this seems a harmless default
-ZLIBDIR=src
+ZLIBDIR=$oldincludedir
 AUX_SHARED_ZLIB=
 
 if test -d $srcdir/zlib; then
diff -uNr ghostscript-9.20.orig/configure.ac ghostscript-9.20/configure.ac
--- ghostscript-9.20.orig/configure.ac	2016-09-26 06:41:28.000000000 -0400
+++ ghostscript-9.20/configure.ac	2016-10-03 14:32:24.000000000 -0400
@@ -914,7 +914,7 @@
 dnl zlib is needed for language level 3, and libpng
 # we must define ZLIBDIR regardless because png.mak does a -I$(ZLIBDIR)
 # this seems a harmless default
-ZLIBDIR=src
+ZLIBDIR=$oldincludedir
 AUX_SHARED_ZLIB=
 
 if test -d $srcdir/zlib; then
diff -uNr ghostscript-9.20.orig/contrib/gdevhl12.c ghostscript-9.20/contrib/gdevhl12.c
--- ghostscript-9.20.orig/contrib/gdevhl12.c	2016-09-26 06:41:28.000000000 -0400
+++ ghostscript-9.20/contrib/gdevhl12.c	2016-10-03 14:32:24.000000000 -0400
@@ -481,7 +481,7 @@
         break;
     }
     if (s->out_count) {
-        fprintf(prn_stream, "\033*b%uW", s->out_count * sizeof(u16) + 9);
+        fprintf(prn_stream, "\033*b%luW", s->out_count * sizeof(u16) + 9);
         put_be16(prn_stream, s->out_count * sizeof(u16) + 7);
         put_be16(prn_stream, s->xl * 16);
         put_be16(prn_stream, band + ytop);
diff -uNr ghostscript-9.20.orig/contrib/japanese/gdevp201.c ghostscript-9.20/contrib/japanese/gdevp201.c
--- ghostscript-9.20.orig/contrib/japanese/gdevp201.c	2016-09-26 06:41:28.000000000 -0400
+++ ghostscript-9.20/contrib/japanese/gdevp201.c	2016-10-03 14:32:24.000000000 -0400
@@ -242,7 +242,7 @@
                 out_beg -= (out_beg - out) % bytes_per_column;
 
                 /* Dot addressing */
-                fprintf(pdev->file, "\033F%04d",
+                fprintf(pdev->file, "\033F%04ld",
                         (out_beg - out) / bytes_per_column);
 
                 /* Dot graphics */
diff -uNr ghostscript-9.20.orig/devices/gdevlp8k.c ghostscript-9.20/devices/gdevlp8k.c
--- ghostscript-9.20.orig/devices/gdevlp8k.c	2016-09-26 06:41:29.000000000 -0400
+++ ghostscript-9.20/devices/gdevlp8k.c	2016-10-03 14:35:11.000000000 -0400
@@ -360,9 +360,9 @@
         fprintf(prn_stream,"%d",lnum-60);
         fwrite("Y\035",1,2,prn_stream);
         report_size = outp - out;
-        fprintf(prn_stream,"%d;",report_size);
+        fprintf(prn_stream,"%ld;",report_size);
         report_size = (in_end - inp) << 3;
-        fprintf(prn_stream,"%d;",report_size);
+        fprintf(prn_stream,"%ld;",report_size);
         fwrite("1;0bi{I",1,7,prn_stream);
         fwrite(out,1,(outp - out),prn_stream);
 
diff -uNr ghostscript-9.20.orig/devices/gdevpng.c ghostscript-9.20/devices/gdevpng.c
--- ghostscript-9.20.orig/devices/gdevpng.c	2016-09-26 06:41:29.000000000 -0400
+++ ghostscript-9.20/devices/gdevpng.c	2016-10-03 14:32:24.000000000 -0400
@@ -711,6 +711,7 @@
 
 #if PNG_LIBPNG_VER_MINOR < 5
 
+#if 0 /* not required in 1.5 */
 /*
  * Patch around a static reference to a never-used procedure.
  * This could be avoided if we were willing to edit pngconf.h to
@@ -731,6 +732,7 @@
 }
 #endif
 #endif
+#endif
 
 static int
 pngalpha_open(gx_device * pdev)
