diff -uNr ghostscript-9.14/base/mkromfs.c ghostscript-9.14.fink/base/mkromfs.c
--- ghostscript-9.14/base/mkromfs.c	2014-03-26 08:53:47.000000000 -0400
+++ ghostscript-9.14.fink/base/mkromfs.c	2014-04-17 19:14:54.000000000 -0400
@@ -2302,7 +2302,7 @@
 #endif
     fprintf(out,"\n#include \"stdint_.h\"\n");
     fprintf(out,"\n#include \"time_.h\"\n\n");
-    fprintf(out,"    time_t gs_romfs_buildtime = %ld;\n\n", time(NULL));
+    fprintf(out,"    time_t gs_romfs_buildtime = %lld;\n\n", (long long)time(NULL));
 
     /* process the remaining arguments (options interspersed with paths) */
     for (; atarg < argc; atarg++) {
diff -uNr ghostscript-9.14/configure ghostscript-9.14.fink/configure
--- ghostscript-9.14/configure	2014-03-26 08:53:47.000000000 -0400
+++ ghostscript-9.14.fink/configure	2014-04-17 19:15:50.000000000 -0400
@@ -6254,7 +6255,7 @@
 $as_echo_n "checking for local zlib source... " >&6; }
 # we must define ZLIBDIR regardless because png.mak does a -I$(ZLIBDIR)
 # this seems a harmless default
-ZLIBDIR=src
+ZLIBDIR=$oldincludedir
 AUX_SHARED_ZLIB=
 
 if test -d $srcdir/zlib; then
diff -uNr ghostscript-9.14/psi/imainarg.c ghostscript-9.14.fink/psi/imainarg.c
--- ghostscript-9.14/psi/imainarg.c	2014-03-26 08:53:47.000000000 -0400
+++ ghostscript-9.14.fink/psi/imainarg.c	2014-04-17 19:14:54.000000000 -0400
@@ -217,6 +217,11 @@
 #endif
         }
     }
+
+	code = gs_main_init1(minst);
+	if (code < 0)
+		return code;
+
     while ((arg = arg_next(&args, &code, minst->heap)) != 0) {
         switch (*arg) {
             case '-':
diff -urN -x'*~' ghostscript-8.71/Resource/Init/cidfmap.mac ghostscript-8.71-1/Resource/Init/cidfmap.mac
--- ghostscript-8.71/Resource/Init/cidfmap.mac	2009-10-06 00:33:33.000000000 -0700
+++ ghostscript-8.71-1/Resource/Init/cidfmap.mac	2010-08-23 21:24:00.000000000 -0700
@@ -0,0 +1,92 @@
+%!
+%%% aliases
+/Ryumin-Light                 /HiraMinProN-W3              ;
+/Ryumin-Medium                /HiraMinProN-W3              ;
+/FutoMinA101-Bold             /HiraMinProN-W6              ;
+/MidashiMin-MA31              /HiraMinProN-W6              ;
+/GothicBBB-Medium             /HiraKakuProN-W3             ;
+/FutoGoB101-Bold              /HiraKakuProN-W6             ;
+/MidashiGo-MB31               /HiraKakuStdN-W8             ;
+/Jun101-Light                 /HiraMaruProN-W4             ;
+/HeiseiMin-W3                 /Ryumin-Light                ;
+/HeiseiKakuGo-W5              /GothicBBB-Medium            ;
+/HiraMinStdN-W2               /MS-Mincho                   ;
+/KozMinPr6N-Regular           /Ryumin-Light                ;
+/KozMinPro-Regular            /KozMinPr6N-Regular          ;
+/KozMinPro-Regular-Acro       /KozMinPro-Regular           ;
+/HeiseiMin-W3-Acro            /KozMinPro-Regular-Acro      ;
+/KozGoPr6N-Medium             /GothicBBB-Medium            ;
+/KozGoPro-Medium              /KozGoPr6N-Medium            ;
+/KozGoPro-Medium-Acro         /KozGoPro-Medium             ;
+/HeiseiKakuGo-W5-Acro         /KozGoPro-Medium-Acro        ;
+/STSong-Light                 /STSong                      ;
+/STSong-Regular               /STSong                      ;
+/STHeiti-Regular              /STHeiti                     ;
+/STHeiti-Light                /STXihei                     ;
+/STKaiti-Regular              /STKaiti                     ;
+/STFangsong-Regular           /STFangsong                  ;
+/STFangsong-Light             /STFangsong                  ;
+/AdobeSongStd-Light           /STSong-Light                ;
+/AdobeSongStd-Light-Acro      /AdobeSongStd-Light          ;
+/STSongStd-Light-Acro         /AdobeSongStd-Light-Acro     ;
+/STSongStd-Light              /STSongStd-Light-Acro        ;
+/STSong-Light-Acro            /STSongStd-Light-Acro        ;
+/AdobeHeitiStd-Regular        /STHeiti-Regular             ;
+/MSung-Light                  /LiSongPro                   ;
+/MSung-Medium                 /LiSongPro                   ;
+/MHei-Medium                  /LiHeiPro                    ;
+/MKai-Medium                  /STKaiti-Adobe-CNS1          ;
+/AdobeMingStd-Light           /MSung-Light                 ;
+/AdobeMingStd-Light-Acro      /AdobeMingStd-Light          ;
+/MSungStd-Light-Acro          /AdobeMingStd-Light-Acro     ;
+/MSungStd-Light               /MSungStd-Light-Acro         ;
+/MSung-Light-Acro             /MSungStd-Light-Acro         ;
+/MHei-Medium-Acro             /MHei-Medium                 ;
+/HYSMyeongJo-Medium           /Batang                      ;
+/HYGoThic-Medium              /Baekmuk-Dotum               ;
+/HYKHeadLine-Bold             /Baekmuk-Headline            ;
+/HYKHeadLine-Medium           /Baekmuk-Headline            ;
+/HYRGoThic-Medium             /Gulim                       ;
+/HYGungSo-Bold                /UnGungseo                   ;
+/AdobeMyungjoStd-Medium       /HYSMyeongJo-Medium          ;
+/AdobeMyungjoStd-Medium-Acro  /AdobeMyungjoStd-Medium      ;
+/HYSMyeongJoStd-Medium-Acro   /AdobeMyungjoStd-Medium-Acro ;
+/HYSMyeongJoStd-Medium        /HYSMyeongJoStd-Medium-Acro  ;
+/HYSMyeongJo-Medium-Acro      /HYSMyeongJoStd-Medium-Acro  ;
+/HYGoThic-Medium-Acro         /HYGoThic-Medium             ;
+
+%%% TrueType outline fonts in OS X
+%%% See /System/Library/Fonts/
+%%%     /Library/Fonts/
+/STSong                       << /FileType /TrueType  /CSI [(GB1) 5]     /Path (@XFONTPATH@/applettf/STSong.ttf)     >> ;
+/STHeiti                      << /FileType /TrueType  /CSI [(GB1) 5]     /Path (@XFONTPATH@/applettf/STHeiti.ttf)    >> ;
+/STXihei                      << /FileType /TrueType  /CSI [(GB1) 5]     /Path (@XFONTPATH@/applettf/STXihei.ttf)    >> ;
+/STKaiti                      << /FileType /TrueType  /CSI [(GB1) 5]     /Path (@XFONTPATH@/applettf/STKaiti.ttf)    >> ;
+/STKaiti-Adobe-CNS1           << /FileType /TrueType  /CSI [(CNS1) 5]    /Path (@XFONTPATH@/applettf/STKaiti.ttf)    >> ;
+/STFangsong                   << /FileType /TrueType  /CSI [(GB1) 5]     /Path (@XFONTPATH@/applettf/STFangsong.ttf) >> ;
+/LiSongPro                    << /FileType /TrueType  /CSI [(CNS1) 5]    /Path (@XFONTPATH@/applettf/LiSongPro.ttf)  >> ;
+/LiHeiPro                     << /FileType /TrueType  /CSI [(CNS1) 5]    /Path (@XFONTPATH@/applettf/LiHeiPro.ttf)   >> ;
+
+%%% TrueType outline fonts in Microsoft Office for Mac 2011
+%%% See http://www.microsoft.com/japan/mac/trial
+%%%     /Library/Fonts/Microsoft/
+/MS-Mincho                    << /FileType /TrueType  /CSI [(Japan1) 6]  /Path (/Library/Fonts/Microsoft/MS Mincho.ttf)  >> ;
+/MS-PMincho                   << /FileType /TrueType  /CSI [(Japan1) 6]  /Path (/Library/Fonts/Microsoft/MS PMincho.ttf) >> ;
+/MS-Gothic                    << /FileType /TrueType  /CSI [(Japan1) 6]  /Path (/Library/Fonts/Microsoft/MS Gothic.ttf)  >> ;
+/MS-PGothic                   << /FileType /TrueType  /CSI [(Japan1) 6]  /Path (/Library/Fonts/Microsoft/MS PGothic.ttf) >> ;
+/SimSun                       << /FileType /TrueType  /CSI [(GB1) 5]     /Path (/Library/Fonts/Microsoft/SimSun.ttf)     >> ;
+/PMingLiU                     << /FileType /TrueType  /CSI [(CNS1) 5]    /Path (/Library/Fonts/Microsoft/PMingLiU.ttf)   >> ;
+/Batang                       << /FileType /TrueType  /CSI [(Korea1) 2]  /Path (/Library/Fonts/Microsoft/Batang.ttf)     >> ;
+/Gulim                        << /FileType /TrueType  /CSI [(Korea1) 2]  /Path (/Library/Fonts/Microsoft/Gulim.ttf)      >> ;
+
+%%% OpenType fonts with PostScript outlines in Adobe Reader/Acrobat XI
+%%% See http://www.adobe.com/support/downloads/product.jsp?product=10&platform=Mac
+%%%     ftp://ftp.adobe.com/pub/adobe/reader/mac/11.x/11.0.00/misc/
+%%%     /Library/"Application Support"/Adobe/Fonts/
+%/Ryumin-Light                 /KozMinPr6N-Regular     ;
+%/GothicBBB-Medium             /KozGoPr6N-Medium       ;
+%/GothicBBB-Medium             /KozGoPro-Medium        ;
+%/STSong-Light                 /AdobeSongStd-Light     ;
+%/STHeiti-Regular              /AdobeHeitiStd-Regular  ;
+%/MSung-Light                  /AdobeMingStd-Light     ;
+%/HYSMyeongJo-Medium           /AdobeMyungjoStd-Medium ;
+++ espgs-7.07.1/finkcups	2007-10-02 09:51:41.000000000 +0900
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+show="/bin/echo"
+run=
+function show_and_run () {
+
+	${show} "$@"
+	${run}  "$@"
+}
+
+function get_cups_location () {
+
+	if type cups-config >/dev/null 2>&1
+	then
+		CUPSSERVER=`cups-config --serverbin`
+		CUPSCONFIG=`cups-config --serverroot`
+		CUPSDATA=`cups-config --datadir`
+	else
+		${show} " * ${0##*/}: Cannot find the command \"cups-config\" in your PATH!"
+		exit 1
+	fi
+}
+
+function backup () {
+
+	if [ -f ${1} ]
+	then
+		show_and_run /bin/mv ${1} ${1}.finksave
+	fi
+}
+
+function restore () {
+
+	if [ -f ${1}.finksave ]
+	then
+		show_and_run /bin/mv ${1}.finksave ${1}
+	fi
+}
+
+function unlink () {
+
+	if [ -L ${1} ]
+	then
+		show_and_run /bin/rm ${1}
+	fi
+}
+
+function make_links () {
+
+	get_cups_location
+
+	/bin/mkdir -p ${CUPSSERVER}/filter
+	for file in <PREFIX>/lib/cups/filter/*
+	do
+		backup ${CUPSSERVER}/filter/${file##*/}
+		show_and_run /bin/ln -s ${file} ${CUPSSERVER}/filter
+	done
+
+	/bin/mkdir -p ${CUPSCONFIG}
+	for file in <PREFIX>/etc/cups/*
+	do
+		backup ${CUPSCONFIG}/${file##*/}
+		show_and_run /bin/ln -s ${file} ${CUPSCONFIG}
+	done
+
+	/bin/mkdir -p ${CUPSDATA}/model
+	/bin/mkdir -p /Library/Printers/PPDs/Contents/Resources
+	for file in <PREFIX>/share/cups/model/*
+	do
+		backup ${CUPSDATA}/model/${file##*/}
+		show_and_run /bin/ln -s ${file} ${CUPSDATA}/model
+		
+		backup /Library/Printers/PPDs/Contents/Resources/${file##*/}
+		show_and_run /bin/ln -s ${file} /Library/Printers/PPDs/Contents/Resources
+	done
+}
+
+function remove_links () {
+
+	get_cups_location
+
+	for file in <PREFIX>/lib/cups/filter/*
+	do
+		unlink  ${CUPSSERVER}/filter/${file##*/}
+		restore ${CUPSSERVER}/filter/${file##*/}
+	done
+
+	for file in <PREFIX>/etc/cups/*
+	do
+		unlink  ${CUPSCONFIG}/${file##*/}
+		restore ${CUPSCONFIG}/${file##*/}
+	done
+
+	for file in <PREFIX>/share/cups/model/*
+	do
+		unlink  ${CUPSDATA}/model/${file##*/}
+		restore ${CUPSDATA}/model/${file##*/}
+		
+		unlink  /Library/Printers/PPDs/Contents/Resources/${file##*/}
+		restore /Library/Printers/PPDs/Contents/Resources/${file##*/}
+	done
+}
+
+case "$1" in
+	add)
+		make_links ;;
+	remove)
+		remove_links ;;
+	*)
+		${show} "usage: ${0##*/} {add|remove}" ;;
+esac
