diff -ruN hevea-2.29-orig/Makefile hevea-2.29/Makefile
--- hevea-2.29-orig/Makefile	2014-09-23 02:03:10.000000000 -0500
+++ hevea-2.29/Makefile	2017-06-27 23:36:54.000000000 -0500
@@ -2,13 +2,14 @@
 # Compile using ocamlopt, to use ocamlc set TARGET=byte
 TARGET=opt
 # Install prefix
-PREFIX?=/usr/local
+PREFIX=@FINK_PREFIX@
 # Library directory of hevea
-LIBDIR=$(PREFIX)/lib/hevea
+LIBDIR=$(PREFIX)/share/hevea
 # Where to install programms
 BINDIR=$(PREFIX)/bin
 #Where to install hevea.sty
-LATEXLIBDIR=$(PREFIX)/lib/hevea
+LATEXLIBDIR=$(PREFIX)/share/hevea
+DESTDIR=@FINK_DESTDIR@
 ##### Advanced configuration parameters
 SUF=
 DIR=
@@ -45,6 +46,7 @@
 	echo ALLLIB=\"$(ALLLIB)\" && \
 	echo HTMLLIB=\"$(HTMLLIB)\" && \
 	echo TEXTLIB=\"$(TEXTLIB)\" && \
+	echo DESTDIR=\"$(DESTDIR)\" && \
 	echo INFOLIB=\"$(INFOLIB)\") > $@
 
 clean:: config.sh
