--- a/configure	2020-11-10 04:36:42.000000000 -0600
+++ b/configure	2020-11-10 04:40:14.000000000 -0600
@@ -13460,16 +13460,17 @@
 if ${kb_cv_func_getcwd_forks+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   if test "$cross_compiling" = yes; then :
   kb_cv_func_getcwd_forks=no
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdlib.h>
 int fork() { exit(1); }
                                                  int vfork() { exit(1); }
                                                  extern char *getcwd();
                                                  char path[100];
 int
 main ()
 {
 getcwd(path,100);
@@ -13579,16 +13580,17 @@
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $snfunc and $vsnfunc" >&5
 $as_echo_n "checking for $snfunc and $vsnfunc... " >&6; }
 if ${kpse_cv_have_snfuncs+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdio.h>
 #include <stdarg.h>
                                                   char buf[16];
                                                   va_list ap;
 int
 main ()
 {
 $snfunc (buf, 16, "%s", "abc");
                                                   $vsnfunc (buf, 16, "%s", ap);
@@ -13609,16 +13611,17 @@
 if test "x$kpse_cv_have_snfuncs" != xyes; then
   as_fn_error $? "Sorry, you need $snfunc and $vsnfunc." "$LINENO" 5
 fi
 if test "$cross_compiling" = yes; then :
   :
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdio.h>
 char buf[4] = "abc";
 int
 main ()
 {
 if ($snfunc (buf, 1, "%s", "x") == 0 || buf[1] != 'b')
                                    return 1;
   ;
   return 0;
