--- a/configure	2017-03-18 17:51:07.000000000 -0500
+++ b/configure	2021-07-29 20:06:46.000000000 -0500
@@ -13918,16 +13918,17 @@
 if ${kb_cv_func_getcwd_forks+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   if test "$cross_compiling" = yes; then :
   kb_cv_func_getcwd_forks=no
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdlib.h>
 int fork() { exit(1); }
                                                  int vfork() { exit(1); }
                                                  extern char *getcwd();
                                                  char path[100];
 int
 main ()
 {
 getcwd(path,100);
@@ -14037,16 +14038,17 @@
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $snfunc and $vsnfunc" >&5
 $as_echo_n "checking for $snfunc and $vsnfunc... " >&6; }
 if ${kpse_cv_have_snfuncs+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdio.h>
 #include <stdarg.h>
                                                   char buf[16];
                                                   va_list ap;
 int
 main ()
 {
 $snfunc (buf, 16, "%s", "abc");
                                                   $vsnfunc (buf, 16, "%s", ap);
@@ -14067,16 +14069,17 @@
 if test "x$kpse_cv_have_snfuncs" != xyes; then
   as_fn_error $? "Sorry, you need $snfunc and $vsnfunc." "$LINENO" 5
 fi
 if test "$cross_compiling" = yes; then :
   :
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdio.h>
 char buf[4] = "abc";
 int
 main ()
 {
 if ($snfunc (buf, 1, "%s", "x") == 0 || buf[1] != 'b')
                                    return 1;
   ;
   return 0;
@@ -14096,16 +14099,17 @@
 if ${kpse_cv_wrap_snprintf+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   if test "$cross_compiling" = yes; then :
   kpse_cv_wrap_snprintf=no
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdio.h>
 char buf[4] = "abc";
 int
 main ()
 {
 $snfunc (buf, 3, "xyz");
                                                  if (buf[2] != 0) return 1;
                                                  $snfunc (buf, 2, "xyz");
                                                  if (buf[1] != 0) return 1;
