diff -Nurd discount-2.2.7.orig/configure.inc discount-2.2.7/configure.inc
--- discount-2.2.7.orig/configure.inc	2020-08-24 01:45:49.000000000 -0400
+++ discount-2.2.7/configure.inc	2021-06-26 10:49:37.000000000 -0400
@@ -1425,11 +1425,11 @@
 shift
 
 LIBNAME=\$LIBRARY.dylib
-FULLNAME=\$LIBNAME
+FULLNAME=\$LIBRARY.1.dylib
 
 case "\$ACTION" in
 make)   FLAGS="$AC_CFLAGS -dynamiclib"
-	VFLAGS="-current_version \$VERSION -compatibility_version \$MAJOR"
+	VFLAGS="-current_version \$VERSION -compatibility_version \$MAJOR -install_name \$LIBDIR/\$FULLNAME"
 
 	rm -f \$LIBRARY
 	if $AC_CC \$FLAGS \$VFLAGS -o \$FULLNAME "\$@"; then
diff -Nurd discount-2.2.7.orig/tests/functions.sh discount-2.2.7/tests/functions.sh
--- discount-2.2.7.orig/tests/functions.sh	2020-08-24 01:45:49.000000000 -0400
+++ discount-2.2.7/tests/functions.sh	2021-06-26 11:28:13.000000000 -0400
@@ -49,8 +49,8 @@
     try_header $1
 
     case "$2" in
-    -t*) Q=`./markdown $FLAGS "$2"` ;;
-    *)   Q=`./echo "$2" | ./markdown $FLAGS` ;;
+    -t*) Q=`DYLD_LIBRARY_PATH=$PWD ./markdown $FLAGS "$2"` ;;
+    *)   Q=`./echo "$2" | DYLD_LIBRARY_PATH=$PWD ./markdown $FLAGS` ;;
     esac
 
     if [ "$3" = "$Q" ]; then
@@ -78,7 +78,7 @@
 
     test $VERBOSE && ./echo -n "$testcase"
 
-    if ./echo "$2" | ./markdown | grep "$3" >/dev/null; then
+    if ./echo "$2" | DYLD_LIBRARY_PATH=$PWD ./markdown | grep "$3" >/dev/null; then
 	test $VERBOSE && ./echo " ok"
     else
 	if [ -z "$VERBOSE" ]; then
diff -Nurd discount-2.2.7.orig/tests/linkylinky.t discount-2.2.7/tests/linkylinky.t
--- discount-2.2.7.orig/tests/linkylinky.t	2020-08-24 01:45:49.000000000 -0400
+++ discount-2.2.7/tests/linkylinky.t	2021-06-26 11:25:46.000000000 -0400
@@ -12,7 +12,7 @@
 try 'url contains whitespace' '[ha](r u)' '<p><a href="r%20u">ha</a></p>'
 
 # latex collides with this test
-if ./markdown -V | grep LATEX >/dev/null; then
+if DYLD_LIBRARY_PATH=$PWD ./markdown -V | grep LATEX >/dev/null; then
     RESULT='<p><a href="d">a\[b\]c</a></p>'
 else
     RESULT='<p><a href="d">a[b]c</a></p>'
diff -Nurd discount-2.2.7.orig/tests/pandoc.t discount-2.2.7/tests/pandoc.t
--- discount-2.2.7.orig/tests/pandoc.t	2020-08-24 01:45:49.000000000 -0400
+++ discount-2.2.7/tests/pandoc.t	2021-06-26 11:31:37.000000000 -0400
@@ -45,7 +45,7 @@
 
 check_null() {
     try_header "$3"
-    res=`echo "$2" | ./pandoc_headers $1`
+    res=`echo "$2" | DYLD_LIBRARY_PATH=$PWD ./pandoc_headers $1`
     if [ "$res" ]; then
 	./echo
 	test $VERBOSE || ./echo "$3"
diff -Nurd discount-2.2.7.orig/tests/tabstop.t discount-2.2.7/tests/tabstop.t
--- discount-2.2.7.orig/tests/tabstop.t	2020-08-24 01:45:49.000000000 -0400
+++ discount-2.2.7/tests/tabstop.t	2021-06-26 11:25:47.000000000 -0400
@@ -4,7 +4,7 @@
 unset MARKDOWN_FLAGS
 unset MKD_TABSTOP
 
-eval `./markdown -V | tr ' ' '\n' | grep TAB`
+eval `DYLD_LIBRARY_PATH=$PWD ./markdown -V | tr ' ' '\n' | grep TAB`
 
 if [ "${TAB:-4}" -eq 8 ]; then
     title "dealing with tabstop derangement"
