diff -Nurd lxml-4.6.3.orig/src/lxml/tests/test_etree.py lxml-4.6.3/src/lxml/tests/test_etree.py
--- lxml-4.6.3.orig/src/lxml/tests/test_etree.py	2021-06-06 08:04:35.000000000 -0400
+++ lxml-4.6.3/src/lxml/tests/test_etree.py	2021-06-06 08:05:13.000000000 -0400
@@ -3036,7 +3036,10 @@
     def test_html_prefix_nsmap(self):
         etree = self.etree
         el = etree.HTML('<hha:page-description>aa</hha:page-description>').find('.//page-description')
-        self.assertEqual({'hha': None}, el.nsmap)
+        if etree.LIBXML_VERSION < (2, 9, 11):
+            self.assertEqual({'hha': None}, el.nsmap)
+        else:
+            self.assertEqual({}, el.nsmap)
 
     def test_getchildren(self):
         Element = self.etree.Element
