diff -ruN miktex-2.9.7440-orig/CMakeLists.txt miktex-2.9.7440/CMakeLists.txt
--- miktex-2.9.7440-orig/CMakeLists.txt	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/CMakeLists.txt	2023-03-14 05:33:53.000000000 -0500
@@ -137,8 +137,12 @@
   set(MIKTEX_HOMEBREW TRUE)
 endif()
 
-if(APPLE AND NOT MIKTEX_HOMEBREW)
-  set(MIKTEX_MACOS_BUNDLE TRUE)
+if(${CMAKE_INSTALL_PREFIX} STREQUAL @FINK_PREFIX@)
+  set(MIKTEX_FINK TRUE)
+endif()
+
+if(APPLE AND MIKTEX_FINK)
+  set(MIKTEX_MACOS_BUNDLE FALSE)
 endif()
 
 ###############################################################################
@@ -146,7 +150,9 @@
 
 if(CMAKE_INSTALL_PREFIX STREQUAL "/usr")
   set(MIKTEX_SELF_CONTAINED_DEFAULT FALSE)
-elseif(CMAKE_INSTALL_PREFIX STREQUAL "/usr/local")
+elseif(CMAKE_INSTALL_PREFIX STREQUAL "@FINK_PREFIX@")
+  set(MIKTEX_SELF_CONTAINED_DEFAULT FALSE)
+elseif(MIKTEX_FINK)
   set(MIKTEX_SELF_CONTAINED_DEFAULT FALSE)
 elseif(MIKTEX_HOMEBREW)
   set(MIKTEX_SELF_CONTAINED_DEFAULT FALSE)
diff -ruN miktex-2.9.7440-orig/Documentation/CMakeLists.txt miktex-2.9.7440/Documentation/CMakeLists.txt
--- miktex-2.9.7440-orig/Documentation/CMakeLists.txt	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Documentation/CMakeLists.txt	2023-03-14 04:26:24.000000000 -0500
@@ -26,7 +26,7 @@
 set(PROMPTWIN "C:\&gt; ")
 
 if(MIKTEX_UNIX_ALIKE)
-  set(COMMONINSTALL "/usr/local/share/miktex-texmf")
+  set(COMMONINSTALL "@FINK_PREFIX@/share/miktex-texmf")
   set(USERDATA "~/.miktex/texmfs/data")
   set(USERINSTALL "~/.miktex/texmfs/install")
   set(EXESUFFIX "")
diff -ruN miktex-2.9.7440-orig/Libraries/3rd/log4cxx/CMakeLists.txt miktex-2.9.7440/Libraries/3rd/log4cxx/CMakeLists.txt
--- miktex-2.9.7440-orig/Libraries/3rd/log4cxx/CMakeLists.txt	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Libraries/3rd/log4cxx/CMakeLists.txt	2023-03-14 04:27:51.000000000 -0500
@@ -30,6 +30,7 @@
   ${CMAKE_CURRENT_BINARY_DIR}/privateinclude
   ${CMAKE_CURRENT_SOURCE_DIR}
   ${CMAKE_CURRENT_SOURCE_DIR}/source/src/main/include
+  ${APR_INCLUDE_DIR}
 )
 
 set(public_include_directories
diff -ruN miktex-2.9.7440-orig/Libraries/3rd/log4cxx/source/src/main/cpp/stringhelper.cpp miktex-2.9.7440/Libraries/3rd/log4cxx/source/src/main/cpp/stringhelper.cpp
--- miktex-2.9.7440-orig/Libraries/3rd/log4cxx/source/src/main/cpp/stringhelper.cpp	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Libraries/3rd/log4cxx/source/src/main/cpp/stringhelper.cpp	2023-03-14 04:29:39.000000000 -0500
@@ -28,6 +28,7 @@
 #endif
 #include <log4cxx/private/log4cxx_private.h>
 #include <cctype>
+#include <cstdlib>
 #include <apr.h>
 
 #if defined(MIKTEX)
diff -ruN miktex-2.9.7440-orig/Libraries/3rd/log4cxx/source/src/main/include/log4cxx/helpers/simpledateformat.h miktex-2.9.7440/Libraries/3rd/log4cxx/source/src/main/include/log4cxx/helpers/simpledateformat.h
--- miktex-2.9.7440-orig/Libraries/3rd/log4cxx/source/src/main/include/log4cxx/helpers/simpledateformat.h	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Libraries/3rd/log4cxx/source/src/main/include/log4cxx/helpers/simpledateformat.h	2023-03-14 04:30:02.000000000 -0500
@@ -27,10 +27,9 @@
 
 #include <log4cxx/helpers/dateformat.h>
 #include <vector>
+#include <locale>
 #include <time.h>
 
-namespace std { class locale; }
-
 namespace log4cxx
 {
         namespace helpers
diff -ruN miktex-2.9.7440-orig/Libraries/3rd/lua53/source/src/luaconf.h miktex-2.9.7440/Libraries/3rd/lua53/source/src/luaconf.h
--- miktex-2.9.7440-orig/Libraries/3rd/lua53/source/src/luaconf.h	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Libraries/3rd/lua53/source/src/luaconf.h	2023-03-14 04:46:00.000000000 -0500
@@ -203,7 +203,7 @@
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"@FINK_PREFIX@"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 #define LUA_PATH_DEFAULT  \
diff -ruN miktex-2.9.7440-orig/Libraries/3rd/luajit/source/src/luaconf.h miktex-2.9.7440/Libraries/3rd/luajit/source/src/luaconf.h
--- miktex-2.9.7440-orig/Libraries/3rd/luajit/source/src/luaconf.h	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Libraries/3rd/luajit/source/src/luaconf.h	2023-03-14 04:45:51.000000000 -0500
@@ -35,7 +35,7 @@
 #ifndef LUA_LMULTILIB
 #define LUA_LMULTILIB	"lib"
 #endif
-#define LUA_LROOT	"/usr/local"
+#define LUA_LROOT	"@FINK_PREFIX@"
 #define LUA_LUADIR	"/lua/5.1/"
 #define LUA_LJDIR	"/luajit-2.1.0-beta1/"
 
diff -ruN miktex-2.9.7440-orig/Libraries/3rd/poppler/source/poppler/GlobalParams.cc miktex-2.9.7440/Libraries/3rd/poppler/source/poppler/GlobalParams.cc
--- miktex-2.9.7440-orig/Libraries/3rd/poppler/source/poppler/GlobalParams.cc	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Libraries/3rd/poppler/source/poppler/GlobalParams.cc	2023-03-14 04:45:10.000000000 -0500
@@ -1300,11 +1300,11 @@
 };
 
 static const char *displayFontDirs[] = {
-  "/usr/share/ghostscript/fonts",
-  "/usr/local/share/ghostscript/fonts",
-  "/usr/share/fonts/default/Type1",
-  "/usr/share/fonts/default/ghostscript",
-  "/usr/share/fonts/type1/gsfonts",
+  "@FINK_PREFIX@/share/ghostscript/fonts",
+  "@FINK_PREFIX@/share/ghostscript/fonts",
+  "@FINK_PREFIX@/lib/X11/fonts/ghostscript/Type1",
+  "@FINK_PREFIX@/share/fonts/default/ghostscript",
+  "@FINK_PREFIX@/share/fonts/type1/gsfonts",
   NULL
 };
 
diff -ruN miktex-2.9.7440-orig/Libraries/MiKTeX/Core/Session/unx/unxSession.cpp miktex-2.9.7440/Libraries/MiKTeX/Core/Session/unx/unxSession.cpp
--- miktex-2.9.7440-orig/Libraries/MiKTeX/Core/Session/unx/unxSession.cpp	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Libraries/MiKTeX/Core/Session/unx/unxSession.cpp	2023-03-14 04:43:01.000000000 -0500
@@ -94,12 +94,12 @@
  * UserConfig:    $HOME/.miktex/texmfs/config
  * UserData:      $HOME/.miktex/texmfs/data
  * UserInstall:   $HOME/.miktex/texmfs/install
- * CommonConfig:  /var/lib/miktex-texmf           (DEB,RPM)
- *             or /var/local/lib/miktex-texmf     (TGZ)
+ * CommonConfig:  @FINK_PREFIX@/var/lib/miktex-texmf           (DEB,RPM)
+ *             or @FINK_PREFIX@/var/local/lib/miktex-texmf     (TGZ)
  *             or /opt/miktex/texmfs/config       (self-contained)
- * CommonData:    /var/cache/miktex-texmf         (DEB,RPM,TGZ)
+ * CommonData:    @FINK_PREFIX@/var/cache/miktex-texmf         (DEB,RPM,TGZ)
  *             or /opt/miktex/texmfs/data         (self-contained)
- * CommonInstall: /usr/local/share/miktex-texmf   (DEB,RPM,TGZ)
+ * CommonInstall: @FINK_PREFIX@/share/miktex-texmf   (DEB,RPM,TGZ)
  *             or /opt/miktex/texmfs/install      (self-contained)
  */
 StartupConfig SessionImpl::DefaultConfig(MiKTeXConfiguration config, const PathName& commonPrefixArg, const PathName& userPrefixArg)
diff -ruN miktex-2.9.7440-orig/Libraries/MiKTeX/Setup/unx/unxSetupService.cpp miktex-2.9.7440/Libraries/MiKTeX/Setup/unx/unxSetupService.cpp
--- miktex-2.9.7440-orig/Libraries/MiKTeX/Setup/unx/unxSetupService.cpp	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Libraries/MiKTeX/Setup/unx/unxSetupService.cpp	2023-03-14 04:33:05.000000000 -0500
@@ -25,7 +25,7 @@
 PathName SetupService::GetDefaultCommonInstallDir()
 {
   // TODO
-  return PathName("/usr/local/miktex");
+  return PathName("@FINK_PREFIX@/lib/miktex");
 }
 
 PathName SetupService::GetDefaultUserInstallDir()
diff -ruN miktex-2.9.7440-orig/Programs/Bibliography/bibtex-x/source/bibtex.c miktex-2.9.7440/Programs/Bibliography/bibtex-x/source/bibtex.c
--- miktex-2.9.7440-orig/Programs/Bibliography/bibtex-x/source/bibtex.c	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Programs/Bibliography/bibtex-x/source/bibtex.c	2023-03-14 04:33:45.000000000 -0500
@@ -229,7 +229,7 @@
 #if defined(MIKTEX)
 #  define main MIKTEXCEECALL Main
 #endif
-int                     main (int argc, char **argv)
+int main (int argc, char **argv)
 BEGIN
     extern Integer8_T       history;
     int			    exit_status;
diff -ruN miktex-2.9.7440-orig/Programs/DviWare/dvipdfm-x/source/dpxfile.c miktex-2.9.7440/Programs/DviWare/dvipdfm-x/source/dpxfile.c
--- miktex-2.9.7440-orig/Programs/DviWare/dvipdfm-x/source/dpxfile.c	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Programs/DviWare/dvipdfm-x/source/dpxfile.c	2023-03-14 04:34:27.000000000 -0500
@@ -82,7 +82,7 @@
 static int
 miktex_get_acrobat_font_dir (char *buf)
 {
-  strcpy(buf, "/usr/share/ghostscript/Resource/Font/");
+  strcpy(buf, "@FINK_PREFIX@/share/ghostscript/fonts/");
   return  1;
 }
 
diff -ruN miktex-2.9.7440-orig/Programs/Editors/TeXworks/miktex/DefaultBinaryPaths.h miktex-2.9.7440/Programs/Editors/TeXworks/miktex/DefaultBinaryPaths.h
--- miktex-2.9.7440-orig/Programs/Editors/TeXworks/miktex/DefaultBinaryPaths.h	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Programs/Editors/TeXworks/miktex/DefaultBinaryPaths.h	2023-03-14 04:39:59.000000000 -0500
@@ -1 +1 @@
-#define DEFAULT_BIN_PATHS "/usr/local/bin"
+#define DEFAULT_BIN_PATHS "@FINK_PREFIX@/bin"
diff -ruN miktex-2.9.7440-orig/Programs/Editors/TeXworks/source/src/TWUtils.cpp miktex-2.9.7440/Programs/Editors/TeXworks/source/src/TWUtils.cpp
--- miktex-2.9.7440-orig/Programs/Editors/TeXworks/source/src/TWUtils.cpp	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Programs/Editors/TeXworks/source/src/TWUtils.cpp	2023-03-14 04:42:21.000000000 -0500
@@ -54,10 +54,10 @@
 #if defined(Q_OS_UNIX) && !defined(Q_OS_DARWIN)
 // compile-time default paths - customize by defining in the .pro file
 #ifndef TW_DICPATH
-#define TW_DICPATH "/usr/share/hunspell" PATH_LIST_SEP "/usr/share/myspell/dicts"
+#define TW_DICPATH "@FINK_PREFIX@/share/hunspell" PATH_LIST_SEP "@FINK_PREFIX@/share/myspell/dicts"
 #endif
 #ifndef TW_HELPPATH
-#define TW_HELPPATH "/usr/local/share/texworks-help"
+#define TW_HELPPATH "@FINK_PREFIX@/share/texworks-help"
 #endif
 #endif
 
diff -ruN miktex-2.9.7440-orig/Programs/MiKTeX/initexmf/initexmf.cpp miktex-2.9.7440/Programs/MiKTeX/initexmf/initexmf.cpp
--- miktex-2.9.7440-orig/Programs/MiKTeX/initexmf/initexmf.cpp	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Programs/MiKTeX/initexmf/initexmf.cpp	2023-03-14 04:35:56.000000000 -0500
@@ -1402,7 +1402,7 @@
 #if defined(MIKTEX_WINDOWS)
   { "arctrl" MIKTEX_EXE_FILE_SUFFIX, { "pdfclose", "pdfdde", "pdfopen" } },
 #endif
-  { "cjklatex" MIKTEX_EXE_FILE_SUFFIX, { "bg5pluslatex", "bg5pluspdflatex", "bg5latex", "bg5pdflatex", "bg5platex", "bg5ppdflatex", "cef5latex", "cef5pdflatex", "ceflatex", "cefpdflatex", "cefslatex", "cefspdflatex", "gbklatex", "gbkpdflatex", "sjislatex", "sjispdflatex" } },
+  { "cjklatex" MIKTEX_EXE_FILE_SUFFIX, { "bg5+latex", "bg5+pdflatex", "bg5latex", "bg5pdflatex", "bg5+latex", "bg5+pdflatex", "cef5latex", "cef5pdflatex", "ceflatex", "cefpdflatex", "cefslatex", "cefspdflatex", "gbklatex", "gbkpdflatex", "sjislatex", "sjispdflatex" } },
 
   { MIKTEX_AFM2TFM_EXE, { "afm2tfm" } },
 #if defined(MIKTEX_WINDOWS)
diff -ruN miktex-2.9.7440-orig/Programs/TeXAndFriends/xetex/xetex-miktex.cpp miktex-2.9.7440/Programs/TeXAndFriends/xetex/xetex-miktex.cpp
--- miktex-2.9.7440-orig/Programs/TeXAndFriends/xetex/xetex-miktex.cpp	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Programs/TeXAndFriends/xetex/xetex-miktex.cpp	2023-03-14 04:36:21.000000000 -0500
@@ -18,6 +18,7 @@
    USA. */
 
 #include "miktex-first.h"
+#include "xetex-miktex.h"
 
 #include <poppler-config.h>
 #include <hb-icu.h>
@@ -27,8 +28,6 @@
 #include <TECkit_Engine.h>
 #include <zlib.h>
 
-#include "xetex-miktex.h"
-
 using namespace MiKTeX::Core;
 
 XETEXPROGCLASS::unicodescalar*& buffer = XETEXPROG.buffer;
diff -ruN miktex-2.9.7440-orig/Programs/Validation/chktex/source/OpSys.c miktex-2.9.7440/Programs/Validation/chktex/source/OpSys.c
--- miktex-2.9.7440-orig/Programs/Validation/chktex/source/OpSys.c	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/Programs/Validation/chktex/source/OpSys.c	2023-03-14 04:36:53.000000000 -0500
@@ -98,7 +98,7 @@
 
 #ifndef SYSCONFDIR
 #  if defined(__unix__)
-#    define SYSCONFDIR "/usr/local/lib/"
+#    define SYSCONFDIR "@FINK_PREFIX@/lib/"
 #  elif defined(__MSDOS__)
 #    define SYSCONFDIR "\\emtex\\data\\"
 #  else
diff -ruN miktex-2.9.7440-orig/cmake/modules/RuntimePaths.cmake miktex-2.9.7440/cmake/modules/RuntimePaths.cmake
--- miktex-2.9.7440-orig/cmake/modules/RuntimePaths.cmake	2020-05-17 14:37:59.000000000 -0500
+++ miktex-2.9.7440/cmake/modules/RuntimePaths.cmake	2023-03-14 04:39:38.000000000 -0500
@@ -28,10 +28,10 @@
 endif()
 
 if(MIKTEX_UNIX_ALIKE)
-  set(MIKTEX_SYSTEM_ETC_FONTS_CONFD_DIR "/etc/fonts/conf.d" CACHE PATH "System-wide directory for font configuration data.")
-  set(MIKTEX_SYSTEM_LINK_TARGET_DIR "/usr/local/bin" CACHE PATH "System-wide directory in which to create symbolic links to MiKTeX binaries.")
-  set(MIKTEX_SYSTEM_VAR_CACHE_DIR "/var/cache" CACHE PATH "Directory for application cache data.")
-  set(MIKTEX_SYSTEM_VAR_LIB_DIR "/var/lib" CACHE PATH "Directory for state information.")
-  set(MIKTEX_SYSTEM_VAR_LOG_DIR "/var/log" CACHE PATH "Directory for log files.")
+  set(MIKTEX_SYSTEM_ETC_FONTS_CONFD_DIR "@FINK_PREFIX@/etc/fonts/conf.d" CACHE PATH "System-wide directory for font configuration data.")
+  set(MIKTEX_SYSTEM_LINK_TARGET_DIR "@FINK_PREFIX@/bin" CACHE PATH "System-wide directory in which to create symbolic links to MiKTeX binaries.")
+  set(MIKTEX_SYSTEM_VAR_CACHE_DIR "@FINK_PREFIX@/var/cache" CACHE PATH "Directory for application cache data.")
+  set(MIKTEX_SYSTEM_VAR_LIB_DIR "@FINK_PREFIX@/var/lib" CACHE PATH "Directory for state information.")
+  set(MIKTEX_SYSTEM_VAR_LOG_DIR "@FINK_PREFIX@/var/log" CACHE PATH "Directory for log files.")
   set(MIKTEX_USER_LINK_TARGET_DIR "~/bin" CACHE STRING "Per-user directory in which to create symbolic links to MiKTeX binaries.")
 endif()
