diff -Nurd mupdf-1.25.2-source.orig/Makefile mupdf-1.25.2-source/Makefile
--- mupdf-1.25.2-source.orig/Makefile	2024-12-10 08:08:50.000000000 -0500
+++ mupdf-1.25.2-source/Makefile	2024-12-15 21:10:25.000000000 -0500
@@ -323,7 +323,11 @@
 else
 MUPDF_LIB = $(OUT)/libmupdf.a
 LIBS_TO_INSTALL_IN_LIB = $(MUPDF_LIB) $(THIRD_LIB)
+ifneq ($(strip $(THIRD_OBJ)),)
+# only declare this lib to exist if it will have any .o in it
+# (OS X 'ar' cannot create empty archives)
 THIRD_LIB = $(OUT)/libmupdf-third.a
+endif
 ifneq ($(USE_SYSTEM_GLUT),yes)
 THIRD_GLUT_LIB = $(OUT)/libmupdf-glut.a
 endif
diff -Nurd mupdf-1.25.2-source.orig/Makerules mupdf-1.25.2-source/Makerules
--- mupdf-1.25.2-source.orig/Makerules	2024-12-10 08:08:50.000000000 -0500
+++ mupdf-1.25.2-source/Makerules	2024-12-15 21:37:04.000000000 -0500
@@ -164,7 +164,7 @@
 SYS_HARFBUZZ_LIBS := -lharfbuzz
 SYS_JBIG2DEC_LIBS := -ljbig2dec
 SYS_JPEGXR_LIBS := -ljpegxr
-SYS_LCMS2_LIBS := -llcms2
+SYS_LCMS2_LIBS := -llcms2mt
 SYS_LIBJPEG_LIBS := -ljpeg
 SYS_MUJS_LIBS := -lmujs
 SYS_OPENJPEG_LIBS := -lopenjp2
@@ -192,7 +192,7 @@
   HAVE_GLUT := yes
   SYS_GLUT_CFLAGS := -Wno-deprecated-declarations
   SYS_GLUT_LIBS := -framework GLUT -framework OpenGL
-  CC = xcrun cc
+  CC = flag-sort cc
   AR = xcrun ar
   LD = xcrun ld
   RANLIB = xcrun ranlib
@@ -210,7 +210,6 @@
     endif
   endif
 
-else ifeq ($(LINUX_OR_OPENBSD),yes)
 
   ifeq ($(OS),Linux)
     HAVE_OBJCOPY := yes
@@ -228,9 +227,9 @@
 	SYS_HARFBUZZ_CFLAGS := $(shell pkg-config --cflags harfbuzz)
 	SYS_HARFBUZZ_LIBS := $(shell pkg-config --libs harfbuzz)
   endif
-  ifeq ($(shell pkg-config --exists lcms2 && echo yes),yes)
-	SYS_LCMS2_CFLAGS := $(shell pkg-config --cflags lcms2)
-	SYS_LCMS2_LIBS := $(shell pkg-config --libs lcms2)
+  ifeq ($(shell pkg-config --exists lcms2mt && echo yes),yes)
+	SYS_LCMS2_CFLAGS := $(shell pkg-config --cflags lcms2mt)
+	SYS_LCMS2_LIBS := $(shell pkg-config --libs lcms2mt)
   endif
   ifeq ($(shell pkg-config --exists libjpeg && echo yes),yes)
 	SYS_LIBJPEG_CFLAGS := $(shell pkg-config --cflags libjpeg)
@@ -269,16 +268,6 @@
 	SYS_CURL_LIBS := $(shell pkg-config --libs libcurl)
   endif
 
-  HAVE_GLUT := yes
-  ifeq ($(HAVE_GLUT),yes)
-    ifeq ($(OS),OpenBSD)
-      SYS_GLUT_CFLAGS := $(shell pkg-config --cflags glut gl)
-      SYS_GLUT_LIBS := $(shell pkg-config --libs glut gl)
-    else
-      SYS_GLUT_CFLAGS :=
-      SYS_GLUT_LIBS := -lglut -lGL
-    endif
-  endif
 
   HAVE_X11 := $(shell pkg-config --exists x11 xext && echo yes)
   ifeq ($(HAVE_X11),yes)
diff -Nurd mupdf-1.25.2-source.orig/Makethird mupdf-1.25.2-source/Makethird
--- mupdf-1.25.2-source.orig/Makethird	2024-12-10 08:08:50.000000000 -0500
+++ mupdf-1.25.2-source/Makethird	2024-12-15 21:10:25.000000000 -0500
@@ -6,9 +6,9 @@
   USE_SYSTEM_HARFBUZZ := yes
   USE_SYSTEM_JBIG2DEC := yes
   USE_SYSTEM_JPEGXR := no # not available
-  USE_SYSTEM_LCMS2 := no # lcms2mt is strongly preferred
+  USE_SYSTEM_LCMS2 := yes
   USE_SYSTEM_LIBJPEG := yes
-  USE_SYSTEM_MUJS := no # not available
+  USE_SYSTEM_MUJS := yes
   USE_SYSTEM_OPENJPEG := yes
   USE_SYSTEM_ZLIB := yes
   USE_SYSTEM_GLUT := yes
@@ -133,7 +133,7 @@
 # --- LCMS2 ---
 
 ifeq ($(USE_SYSTEM_LCMS2),yes)
-  THIRD_CFLAGS += $(SYS_LCMS2_CFLAGS)
+  THIRD_CFLAGS += $(SYS_LCMS2_CFLAGS) -DHAVE_LCMS2MT
   THIRD_LIBS += $(SYS_LCMS2_LIBS)
 else
   THIRD_CFLAGS += $(LCMS2_CFLAGS)
