diff -Nurd -x'*~' mupdf-1.18.0-source.orig/Makefile mupdf-1.18.0-source/Makefile
--- mupdf-1.18.0-source.orig/Makefile	2020-10-07 06:35:03.000000000 -0400
+++ mupdf-1.18.0-source/Makefile	2021-01-04 05:15:29.000000000 -0500
@@ -215,7 +215,11 @@
 $(MUPDF_LIB) : $(MUPDF_OBJ) $(THIRD_OBJ) $(THREAD_OBJ) $(PKCS7_OBJ)
 else
 MUPDF_LIB = $(OUT)/libmupdf.a
+ifneq ($(strip $(THIRD_OBJ)),)
+# only declare this lib to exist if it will have any .o in it
+# (OS X 'ar' cannot create empty archives)
 THIRD_LIB = $(OUT)/libmupdf-third.a
+endif
 THREAD_LIB = $(OUT)/libmupdf-threads.a
 PKCS7_LIB = $(OUT)/libmupdf-pkcs7.a
 
diff -Nurd -x'*~' mupdf-1.18.0-source.orig/Makerules mupdf-1.18.0-source/Makerules
--- mupdf-1.18.0-source.orig/Makerules	2020-10-07 06:35:03.000000000 -0400
+++ mupdf-1.18.0-source/Makerules	2021-01-04 04:43:23.000000000 -0500
@@ -90,7 +90,7 @@
 SYS_HARFBUZZ_LIBS := -lharfbuzz
 SYS_JBIG2DEC_LIBS := -ljbig2dec
 SYS_JPEGXR_LIBS := -ljpegxr
-SYS_LCMS2_LIBS := -llcms2
+SYS_LCMS2_LIBS := -llcms2mt
 SYS_LIBJPEG_LIBS := -ljpeg
 SYS_MUJS_LIBS := -lmujs
 SYS_OPENJPEG_LIBS := -lopenjp2
@@ -110,13 +110,11 @@
   HAVE_GLUT := yes
   SYS_GLUT_CFLAGS := -Wno-deprecated-declarations
   SYS_GLUT_LIBS := -framework GLUT -framework OpenGL
-  CC = xcrun cc
+  CC = flag-sort cc
   AR = xcrun ar
   LD = xcrun ld
   RANLIB = xcrun ranlib
 
-else ifeq ($(OS),Linux)
-  HAVE_OBJCOPY := yes
 
   ifeq ($(shell pkg-config --exists freetype2 && echo yes),yes)
 	SYS_FREETYPE_CFLAGS := $(shell pkg-config --cflags freetype2)
@@ -130,9 +128,9 @@
 	SYS_HARFBUZZ_CFLAGS := $(shell pkg-config --cflags harfbuzz)
 	SYS_HARFBUZZ_LIBS := $(shell pkg-config --libs harfbuzz)
   endif
-  ifeq ($(shell pkg-config --exists lcms2 && echo yes),yes)
-	SYS_LCMS2_CFLAGS := $(shell pkg-config --cflags lcms2)
-	SYS_LCMS2_LIBS := $(shell pkg-config --libs lcms2)
+  ifeq ($(shell pkg-config --exists lcms2mt && echo yes),yes)
+	SYS_LCMS2_CFLAGS := $(shell pkg-config --cflags lcms2mt)
+	SYS_LCMS2_LIBS := $(shell pkg-config --libs lcms2mt)
   endif
   ifeq ($(shell pkg-config --exists libjpeg && echo yes),yes)
 	SYS_LIBJPEG_CFLAGS := $(shell pkg-config --cflags libjpeg)
@@ -165,11 +163,6 @@
 	SYS_CURL_LIBS := $(shell pkg-config --libs libcurl)
   endif
 
-  HAVE_GLUT := yes
-  ifeq ($(HAVE_GLUT),yes)
-	SYS_GLUT_CFLAGS :=
-	SYS_GLUT_LIBS := -lglut -lGL
-  endif
 
   HAVE_X11 := $(shell pkg-config --exists x11 xext && echo yes)
   ifeq ($(HAVE_X11),yes)
diff -Nurd -x'*~' mupdf-1.18.0-source.orig/Makethird mupdf-1.18.0-source/Makethird
--- mupdf-1.18.0-source.orig/Makethird	2020-10-07 06:35:03.000000000 -0400
+++ mupdf-1.18.0-source/Makethird	2021-01-04 04:34:51.000000000 -0500
@@ -6,9 +6,9 @@
   USE_SYSTEM_HARFBUZZ := yes
   USE_SYSTEM_JBIG2DEC := yes
   USE_SYSTEM_JPEGXR := no # not available
-  USE_SYSTEM_LCMS2 := no # lcms2mt is strongly preferred
+  USE_SYSTEM_LCMS2 := yes
   USE_SYSTEM_LIBJPEG := yes
-  USE_SYSTEM_MUJS := no # not available
+  USE_SYSTEM_MUJS := yes
   USE_SYSTEM_OPENJPEG := yes
   USE_SYSTEM_ZLIB := yes
   USE_SYSTEM_GLUT := yes
@@ -278,7 +278,7 @@
 # --- LCMS2 ---
 
 ifeq ($(USE_SYSTEM_LCMS2),yes)
-  THIRD_CFLAGS += $(SYS_LCMS2_CFLAGS)
+  THIRD_CFLAGS += $(SYS_LCMS2_CFLAGS) -DHAVE_LCMS2MT
   THIRD_LIBS += $(SYS_LCMS2_LIBS)
 else
 
Binary files mupdf-1.18.0-source.orig/thirdparty/lcms2/Projects/mac/LittleCMS/._Info.plist and mupdf-1.18.0-source/thirdparty/lcms2/Projects/mac/LittleCMS/._Info.plist differ
Binary files mupdf-1.18.0-source.orig/thirdparty/lcms2/Projects/mac/LittleCMS/._LittleCMS.xcodeproj and mupdf-1.18.0-source/thirdparty/lcms2/Projects/mac/LittleCMS/._LittleCMS.xcodeproj differ
