diff -ruN psutils-1.17.orig/Makefile.unix psutils-1.17/Makefile.unix
--- psutils-1.17.orig/Makefile.unix	1997-03-11 16:52:59.000000000 -0600
+++ psutils-1.17/Makefile.unix	2024-10-19 19:32:07.000000000 -0500
@@ -19,16 +19,17 @@
 # psnup puts multiple logical pages on one physical page
 # psresize scales and moves pages to fit on different paper sizes
 
-PAPER=a4
+PAPER=@FINK_TYPE_PKG_PAPER@
 
 # Makefile for PSUtils under Unix
 
 OS = UNIX
 
-BINDIR = /usr/local/bin
+prefix = PROVIDED_ON_COMMAND_LINE
+BINDIR = $(prefix)/bin
 SCRIPTDIR = $(BINDIR)
-INCLUDEDIR = /usr/local/share/psutils
-PERL = /usr/local/bin/perl
+INCLUDEDIR = $(prefix)/share/psutils
+PERL = /usr/bin/perl
 
 BINMODE = 0755
 MANMODE = 0644
@@ -36,7 +37,7 @@
 INSTALL = install -c -m $(BINMODE)
 INSTALLMAN = install -c -m $(MANMODE)
 MANEXT = 1
-MANDIR = /usr/local/share/man/man$(MANEXT)
+MANDIR = $(prefix)/share/man/man$(MANEXT)
 
 CC = gcc
 CFLAGS = -DPAPER=\"$(PAPER)\" -DUNIX -O -Wall
@@ -219,28 +220,28 @@
 install: install.bin install.script install.man install.include
 
 install.bin: $(BIN)
-	-mkdir $(BINDIR)
+	-mkdir -p $(BINDIR)
 	@for i in $(BIN); do \
 		echo Installing $$i; \
 		$(INSTALL) $$i $(BINDIR); \
 	done
 
 install.script: $(PERLSCRIPTS) $(SHELLSCRIPTS)
-	-mkdir $(SCRIPTDIR)
+	-mkdir -p $(SCRIPTDIR)
 	@for i in $(PERLSCRIPTS) $(SHELLSCRIPTS); do \
 		echo Installing $$i; \
 		$(INSTALL) $$i $(SCRIPTDIR); \
 	done
 
 install.include: $(INCLUDES)
-	-mkdir $(INCLUDEDIR)
+	-mkdir -p $(INCLUDEDIR)
 	@for i in $(INCLUDES); do \
 		echo Installing $$i; \
 		$(INSTALLMAN) $$i $(INCLUDEDIR); \
 	done
 
 install.man: $(MANPAGES)
-	-mkdir $(MANDIR)
+	-mkdir -p $(MANDIR)
 	@for i in $(MANPAGES); do \
 		echo Installing manual page for $$i; \
 		$(INSTALLMAN) $$i $(MANDIR)/$$i; \
diff -ruN psutils-1.17.orig/epsffit.c psutils-1.17/epsffit.c
--- psutils-1.17.orig/epsffit.c	1997-03-11 16:52:52.000000000 -0600
+++ psutils-1.17/epsffit.c	2024-10-19 18:36:24.000000000 -0500
@@ -39,7 +39,7 @@
    exit(1);
 }
 
-void main(int argc, char **argv)
+int main(int argc, char **argv)
 {
    int bbfound = 0;              /* %%BoundingBox: found */
    int urx, ury, llx, lly;
diff -ruN psutils-1.17.orig/psbook.c psutils-1.17/psbook.c
--- psutils-1.17.orig/psbook.c	1997-03-11 16:53:01.000000000 -0600
+++ psutils-1.17/psbook.c	2024-10-19 18:36:24.000000000 -0500
@@ -32,7 +32,7 @@
 }
 
 
-void main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
    int signature = 0;
    int currentpg, maxpage;
diff -ruN psutils-1.17.orig/psnup.c psutils-1.17/psnup.c
--- psutils-1.17.orig/psnup.c	1997-03-11 16:53:02.000000000 -0600
+++ psutils-1.17/psnup.c	2024-10-19 18:36:24.000000000 -0500
@@ -66,7 +66,7 @@
    return (0);
 }
 
-void main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
    int horiz, vert, rotate, column, flip, leftright, topbottom;
    int nup = 1;
diff -ruN psutils-1.17.orig/psresize.c psutils-1.17/psresize.c
--- psutils-1.17.orig/psresize.c	1997-03-11 16:53:03.000000000 -0600
+++ psutils-1.17/psresize.c	2024-10-19 18:36:24.000000000 -0500
@@ -46,7 +46,7 @@
 #define MIN(x,y) ((x) > (y) ? (y) : (x))
 #define MAX(x,y) ((x) > (y) ? (x) : (y))
 
-void main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
    double scale, rscale;			/* page scale */
    double waste, rwaste;			/* amount wasted */
diff -ruN psutils-1.17.orig/psselect.c psutils-1.17/psselect.c
--- psutils-1.17.orig/psselect.c	1997-03-11 16:53:03.000000000 -0600
+++ psutils-1.17/psselect.c	2024-10-19 18:36:24.000000000 -0500
@@ -91,7 +91,7 @@
 }
 
 
-void main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
    int currentpg, maxpage = 0;
    int even = 0, odd = 0, reverse = 0;
diff -ruN psutils-1.17.orig/pstops.c psutils-1.17/pstops.c
--- psutils-1.17.orig/pstops.c	1997-03-11 16:53:04.000000000 -0600
+++ psutils-1.17/pstops.c	2024-10-19 18:36:24.000000000 -0500
@@ -112,7 +112,7 @@
    return (head);
 }
 
-void main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
    PageSpec *specs = NULL;
    int nobinding = 0;
