diff -ruN teckit-2.5.10-orig/bin/Makefile.in teckit-2.5.10/bin/Makefile.in
--- teckit-2.5.10-orig/bin/Makefile.in	2020-05-05 09:10:40.000000000 -0500
+++ teckit-2.5.10/bin/Makefile.in	2021-05-30 21:41:35.000000000 -0500
@@ -90,8 +90,7 @@
 build_triplet = @build@
 host_triplet = @host@
 target_triplet = @target@
-bin_PROGRAMS = teckit_compile$(EXEEXT) txtconv$(EXEEXT) \
-	sfconv$(EXEEXT)
+bin_PROGRAMS = teckit_compile$(EXEEXT) txtconv$(EXEEXT)
 @OLD_LIB_NAMES_FALSE@teckit_compile_DEPENDENCIES = $(top_builddir)/lib/libTECkit_Compiler.la
 @OLD_LIB_NAMES_FALSE@txtconv_DEPENDENCIES =  \
 @OLD_LIB_NAMES_FALSE@	$(top_builddir)/lib/libTECkit.la
