diff -Nurd texinfo-7.2.orig/tp/Texinfo/XS/configure texinfo-7.2/tp/Texinfo/XS/configure
--- texinfo-7.2.orig/tp/Texinfo/XS/configure	2024-12-23 04:02:27.000000000 -0500
+++ texinfo-7.2/tp/Texinfo/XS/configure	2025-02-01 15:45:20.000000000 -0500
@@ -5305,6 +5305,7 @@
   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking Perl configuration value ccflags" >&5
 printf %s "checking Perl configuration value ccflags... " >&6; }
           fetch_conf ccflags
+	   conf_value=`echo $conf_value | sed 's/-arch\s\+\S\+//g'`
           { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $conf_value" >&5
 printf "%s\n" "$conf_value" >&6; }
           PERL_CONF_ccflags=$conf_value
