diff -Nurd texlive-20170524-source.orig/texk/dvisvgm/dvisvgm-src/libs/Makefile.in texlive-20170524-source/texk/dvisvgm/dvisvgm-src/libs/Makefile.in
--- texlive-20170524-source.orig/texk/dvisvgm/dvisvgm-src/libs/Makefile.in	2017-03-17 18:45:29.000000000 -0400
+++ texlive-20170524-source/texk/dvisvgm/dvisvgm-src/libs/Makefile.in	2022-01-09 10:11:12.000000000 -0500
@@ -338,7 +338,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = clipper xxHash $(am__append_1)
+SUBDIRS = clipper $(am__append_1)
 all: all-recursive
 
 .SUFFIXES:
diff -Nurd texlive-20170524-source.orig/texk/dvisvgm/dvisvgm-src/src/Makefile.in texlive-20170524-source/texk/dvisvgm/dvisvgm-src/src/Makefile.in
--- texlive-20170524-source.orig/texk/dvisvgm/dvisvgm-src/src/Makefile.in	2017-03-17 18:45:30.000000000 -0400
+++ texlive-20170524-source/texk/dvisvgm/dvisvgm-src/src/Makefile.in	2022-01-09 10:09:11.000000000 -0500
@@ -472,7 +472,7 @@
 noinst_LIBRARIES = libdvisvgm.a
 dvisvgm_SOURCES = dvisvgm.cpp
 dvisvgm_LDADD = $(noinst_LIBRARIES) ../libs/clipper/libclipper.a \
-	../libs/xxHash/libxxhash.a $(am__append_1) $(KPATHSEA_LIBS) \
+	-lxxhash $(am__append_1) $(KPATHSEA_LIBS) \
 	$(POTRACE_LIBS) $(FREETYPE2_LIBS) $(FONTFORGE_LIBS) \
 	$(ZLIB_LIBS) $(LIBGS_LIBS)
 dvisvgm_DEPENDENCIES = $(noinst_LIBRARIES) $(KPATHSEA_DEPEND) \
@@ -542,7 +542,7 @@
 	$(am__append_3)
 AM_CXXFLAGS = $(WARNING_CFLAGS) -Wnon-virtual-dtor \
 	-I$(dvisvgm_srcdir)/libs/clipper \
-	-I$(dvisvgm_srcdir)/libs/xxHash $(am__append_4) \
+	$(am__append_4) \
 	$(KPATHSEA_INCLUDES) $(POTRACE_INCLUDES) $(FREETYPE2_INCLUDES) \
 	$(ZLIB_INCLUDES) $(LIBGS_INCLUDES) $(CODE_COVERAGE_CFLAGS) \
 	$(am__append_5)
