diff -ruN texlive-20240312-source.orig/texk/dvisvgm/configure texlive-20240312-source/texk/dvisvgm/configure
--- texlive-20240312-source.orig/texk/dvisvgm/configure	2024-03-07 16:55:09.000000000 -0600
+++ texlive-20240312-source/texk/dvisvgm/configure	2024-04-09 11:30:01.000000000 -0500
@@ -24490,7 +24490,7 @@
 
 
 # TL: these conditionals for system libraries are in the Makefile.am's.
- if false; then
+ if true; then
   HAVE_BROTLI_TRUE=
   HAVE_BROTLI_FALSE='#'
 else
@@ -24498,7 +24498,7 @@
   HAVE_BROTLI_FALSE=
 fi
 
- if false; then
+ if true; then
   HAVE_WOFF2_TRUE=
   HAVE_WOFF2_FALSE='#'
 else
@@ -24506,7 +24506,7 @@
   HAVE_WOFF2_FALSE=
 fi
 
- if false; then
+ if true; then
   HAVE_XXHASH_TRUE=
   HAVE_XXHASH_FALSE='#'
 else
diff -ruN texlive-20240312-source.orig/texk/dvisvgm/dvisvgm-src/libs/woff2/Makefile.in texlive-20240312-source/texk/dvisvgm/dvisvgm-src/libs/woff2/Makefile.in
--- texlive-20240312-source.orig/texk/dvisvgm/dvisvgm-src/libs/woff2/Makefile.in	2023-12-23 10:57:39.000000000 -0600
+++ texlive-20240312-source/texk/dvisvgm/dvisvgm-src/libs/woff2/Makefile.in	2024-04-09 11:30:01.000000000 -0500
@@ -410,7 +410,8 @@
 @HAVE_WOFF2_FALSE@	src/woff2_enc.cc \
 @HAVE_WOFF2_FALSE@	src/woff2_out.cc
 
-@HAVE_WOFF2_FALSE@AM_CXXFLAGS = -Wall -I$(srcdir)/include -I$(srcdir)/../brotli/include
+@HAVE_WOFF2_FALSE@@HAVE_BROTLI_FALSE@AM_CXXFLAGS = -Wall -I$(srcdir)/include -I$(srcdir)/../brotli/include
+@HAVE_WOFF2_FALSE@@HAVE_BROTLI_TRUE@AM_CXXFLAGS = -Wall -I$(srcdir)/include
 EXTRA_DIST = LICENSE
 CLEANFILES = *.gcda *.gcno
 all: all-am
diff -ruN texlive-20240312-source.orig/texk/dvisvgm/dvisvgm-src/src/Makefile.in texlive-20240312-source/texk/dvisvgm/dvisvgm-src/src/Makefile.in
--- texlive-20240312-source.orig/texk/dvisvgm/dvisvgm-src/src/Makefile.in	2024-03-07 16:55:09.000000000 -0600
+++ texlive-20240312-source/texk/dvisvgm/dvisvgm-src/src/Makefile.in	2024-04-09 11:30:01.000000000 -0500
@@ -92,8 +92,10 @@
 bin_PROGRAMS = dvisvgm$(EXEEXT)
 @HAVE_BROTLI_FALSE@am__append_1 = -I$(dvisvgm_srcdir)/libs/brotli/include
 @HAVE_BROTLI_FALSE@am__append_2 = ../libs/brotli/libbrotli.a
+@HAVE_BROTLI_TRUE@am__append_2 = -lbrotlienc
 @HAVE_WOFF2_FALSE@am__append_3 = -I$(dvisvgm_srcdir)/libs/woff2/include
 @HAVE_WOFF2_FALSE@am__append_4 = ../libs/woff2/libwoff2.a
+@HAVE_WOFF2_TRUE@am__append_4 = -lwoff2enc
 @ENABLE_WOFF_TRUE@am__append_5 = ttf
 @ENABLE_WOFF_TRUE@am__append_6 = \
 @ENABLE_WOFF_TRUE@	$(WOFF2_LIBS) \
@@ -552,6 +554,7 @@
 @TEXLIVE_BUILD_TRUE@POTRACE_CFLAGS = $(POTRACE_INCLUDES)
 @HAVE_XXHASH_FALSE@XXHASH_CFLAGS = -I$(dvisvgm_srcdir)/libs/xxHash
 @HAVE_XXHASH_FALSE@XXHASH_LIBS = ../libs/xxHash/libxxhash.a
+@HAVE_XXHASH_TRUE@XXHASH_LIBS = -lxxhash
 dvisvgm_LDADD = $(noinst_LTLIBRARIES) ../libs/clipper/libclipper.a \
 	../libs/md5/libmd5.a $(POTRACE_LIBS) $(XXHASH_LIBS) \
 	$(am__append_6) $(am__append_7) $(am__append_8)
diff -ruN texlive-20240312-source.orig/texk/dvisvgm/dvisvgm-src/tests/Makefile.in texlive-20240312-source/texk/dvisvgm/dvisvgm-src/tests/Makefile.in
--- texlive-20240312-source.orig/texk/dvisvgm/dvisvgm-src/tests/Makefile.in	2024-03-07 16:55:09.000000000 -0600
+++ texlive-20240312-source/texk/dvisvgm/dvisvgm-src/tests/Makefile.in	2024-04-09 11:30:01.000000000 -0500
@@ -89,6 +89,7 @@
 host_triplet = @host@
 @HAVE_BROTLI_FALSE@am__append_1 = -I$(dvisvgm_srcdir)/libs/brotli/include
 @HAVE_BROTLI_FALSE@am__append_2 = ../libs/brotli/libbrotli.a
+@HAVE_BROTLI_TRUE@am__append_2 = -lbrotlienc
 @HAVE_WOFF2_FALSE@am__append_3 = -I$(dvisvgm_srcdir)/libs/woff2/include
 @HAVE_WOFF2_FALSE@am__append_4 = ../libs/woff2/libwoff2.a
 @ENABLE_WOFF_TRUE@am__append_5 = $(WOFF2_LIBS) $(BROTLI_LIBS)
@@ -1025,8 +1026,10 @@
 @HAVE_POTRACE_FALSE@@TEXLIVE_BUILD_FALSE@POTRACE_CFLAGS = -I$(dvisvgm_srcdir)/libs/potrace
 @TEXLIVE_BUILD_TRUE@POTRACE_CFLAGS = $(POTRACE_INCLUDES)
 @HAVE_POTRACE_FALSE@@TEXLIVE_BUILD_FALSE@POTRACE_LIBS = ../libs/potrace/libpotrace.a
+@HAVE_POTRACE_TRUE@POTRACE_LIBS = -lpotrace
 @HAVE_XXHASH_FALSE@XXHASH_CFLAGS = -I$(dvisvgm_srcdir)/libs/xxHash
 @HAVE_XXHASH_FALSE@XXHASH_LIBS = ../libs/xxHash/libxxhash.a
+@HAVE_XXHASH_TRUE@XXHASH_LIBS = -lxxhash
 LIBS_CFLAGS = $(XXHASH_CFLAGS) -I$(dvisvgm_srcdir)/libs/md5
 LIBS_LIBS = $(POTRACE_LIBS) $(XXHASH_LIBS) ../libs/md5/libmd5.a \
 	$(am__append_5)
