--- texlive/texk/web2c/tests/enc-p.bbl.orig	2022-06-12 03:18:23.000000000 -0500
+++ texlive/texk/web2c/tests/enc-p.bbl	2023-11-22 18:08:04.000000000 -0600
@@ -7,8 +7,8 @@
 \item ABC xyz
 \item ABC xyz
 \item ABC xyz
-\item C2Ax: ￠￡￥§¨￢
-\item C2Ax: ￠￡￥§¨￢
+\item C2Ax: ¢£￥§¨¬
+\item C2Ax: ¢£￥§¨¬
 \item C2Bx: °±¶÷
 \item CEAx: ΡΣΤΥΦΧΨΩ
 \item CEBx: αβγδεζηθικλμνξο
--- texlive/texk/web2c/tests/enc-up.bbl.orig	2022-09-03 00:43:53.000000000 -0500
+++ texlive/texk/web2c/tests/enc-up.bbl	2023-11-22 04:20:35.000000000 -0600
@@ -5,8 +5,8 @@
 \item ABC xyz
 \item ABC xyz
 \item ABC xyz
-\item C2Ax: ￠￡￥§¨￢
-\item C2Ax: ￠￡￥§¨￢
+\item C2Ax: ¢£￥§¨¬
+\item C2Ax: ¢£￥§¨¬
 \item C2Bx: °±¶÷
 \item CEAx: ΡΣΤΥΦΧΨΩ
 \item CEBx: αβγδεζηθικλμνξο
--- texlive/texk/web2c/ptexdir/tests/nissya_bib.bbl.orig	2022-11-27 00:52:11.000000000 -0600
+++ texlive/texk/web2c/ptexdir/tests/nissya_bib.bbl	2023-11-22 04:22:56.000000000 -0600
@@ -43,7 +43,7 @@
 
 \nissyaitem[川端編]{川端亮}{川端亮編}{川端亮編}{1999}{kwbt99}
 川端亮編著, ~1999,
-  『非定型データのコーディング・システムとその利用』平成8年度～10年度科学研究費補助金(基盤研究(A)(1))(課題番号 08551003)研究成果報告書,\hspace{.6em}大阪大学．
+  『非定型データのコーディング・システムとその利用』平成8年度〜10年度科学研究費補助金(基盤研究(A)(1))(課題番号 08551003)研究成果報告書,\hspace{.6em}大阪大学．
 
 \nissyaitem[川端・樋口]{川端亮・樋口耕一}{川端亮・樋口耕一}{川端亮・樋口耕一}{2003}{kh2003}
 川端亮・樋口耕一, ~2003,
--- texlive/texk/dviout-util/tests/jisx0208.txt.orig	2021-06-21 12:44:32.000000000 -0500
+++ texlive/texk/dviout-util/tests/jisx0208.txt	2023-11-22 04:26:04.000000000 -0600
@@ -42,8 +42,8 @@
 set2 0x213e "‐"
 set2 0x213f "／"
 set2 0x2140 "＼"
-set2 0x2141 "～"
-set2 0x2142 "∥"
+set2 0x2141 "〜"
+set2 0x2142 "‖"
 set2 0x2143 "｜"
 set2 0x2144 "…"
 set2 0x2145 "‥"
@@ -73,7 +73,7 @@
 set2 0x215a "【"
 set2 0x215b "】"
 set2 0x215c "＋"
-set2 0x215d "－"
+set2 0x215d "−"
 set2 0x215e "±"
 set2 0x215f "×"
 set2 0x2160 "÷"
@@ -93,8 +93,8 @@
 set2 0x216e "℃"
 set2 0x216f "￥"
 set2 0x2170 "＄"
-set2 0x2171 "￠"
-set2 0x2172 "￡"
+set2 0x2171 "¢"
+set2 0x2172 "£"
 set2 0x2173 "％"
 set2 0x2174 "＃"
 set2 0x2175 "＆"
@@ -134,7 +134,7 @@
 set2 0x2241 "∩"
 set2 0x224a "∧"
 set2 0x224b "∨"
-set2 0x224c "￢"
+set2 0x224c "¬"
 set2 0x224d "⇒"
 set2 0x224e "⇔"
 set2 0x224f "∀"
