diff -ruN xdvi-22.86-orig/configure xdvi-22.86/configure
--- xdvi-22.86-orig/configure	2013-05-07 14:14:29
+++ xdvi-22.86/configure	2023-08-29 19:20:45
@@ -2080,6 +2080,7 @@
 #line 2081 "configure"
 #include "confdefs.h"
 #include <ctype.h>
+#include <stdlib.h>
 #define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 #define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
 #define XOR(e, f) (((e) && !(f)) || (!(e) && (f)))
@@ -2460,6 +2461,7 @@
 #line 2461 "configure"
 #include "confdefs.h"
 #include <stdio.h>
+#include <stdlib.h>
 main()
 {
   FILE *f=fopen("conftestval", "w");
@@ -2955,6 +2957,7 @@
 #include "confdefs.h"
 /* Thanks to Paul Eggert for this test.  */
 #include <stdio.h>
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #ifdef HAVE_UNISTD_H
diff -ruN xdvi-22.86-orig/events.c xdvi-22.86/events.c
--- xdvi-22.86-orig/events.c	2012-05-23 14:56:08
+++ xdvi-22.86/events.c	2023-08-29 20:06:22
@@ -44,12 +44,17 @@
 #include <X11/Shell.h>	/* needed for def. of XtNiconX */
 
 #if	XtSpecificationRelease >= 4
+#include <X11/IntrinsicP.h>
+#include <X11/ConvertI.h>
+#include <X11/TranslateI.h> /* needed for _XtComputeLateBindings */
 
 #if XAW
 # if XAW3D
 #  include <X11/Xaw3d/Viewport.h>
+#  include <X11/Xaw3d/XawImP.h>
 # else
 #  include <X11/Xaw/Viewport.h>
+#  include <X11/Xaw/XawImP.h>
 # endif
 # include <X11/cursorfont.h>
 #else /* MOTIF */
diff -ruN xdvi-22.86-orig/ft.c xdvi-22.86/ft.c
--- xdvi-22.86-orig/ft.c	2013-05-07 14:14:29
+++ xdvi-22.86/ft.c	2023-08-29 20:11:27
@@ -29,6 +29,7 @@
 #include "xdvi.h"
 
 #include <math.h>
+#include <ctype.h>
 
 #include <ft2build.h>
 #include FT_FREETYPE_H
