diff -Nurd -x'*~' xmlstarlet-1.6.1.orig/src/xml_ls.c xmlstarlet-1.6.1/src/xml_ls.c
--- xmlstarlet-1.6.1.orig/src/xml_ls.c	2012-08-12 11:18:59.000000000 -0400
+++ xmlstarlet-1.6.1/src/xml_ls.c	2017-10-14 17:54:04.000000000 -0400
@@ -146,7 +146,7 @@
       char mtime[20];
       int size_len;
 
-      if ((d->d_name == NULL) || !strcmp(d->d_name, ".") || !strcmp(d->d_name, ".."))
+      if ((d->d_namlen == 0) || !strcmp(d->d_name, ".") || !strcmp(d->d_name, ".."))
           continue;
 
       if(lstat(d->d_name, &stats) != 0)
