diff -Nurd -x'*~' xmlto-0.0.28.orig/xmlif/xmlif.c xmlto-0.0.28/xmlif/xmlif.c
--- xmlto-0.0.28.orig/xmlif/xmlif.c	2015-11-10 07:05:27.000000000 -0500
+++ xmlto-0.0.28/xmlif/xmlif.c	2016-07-06 22:49:12.000000000 -0400
@@ -1151,7 +1151,7 @@
 
 static char **selections;	/* selection tokens */
 static int nselections;		/* number of selections */
-static ifsense;			/* sense of last `if' or unless seen */
+static int ifsense;		/* sense of last `if' or unless seen */
 static char *attribute;		/* last attribute scanned */
 
 struct stack_t {
@@ -2632,7 +2632,7 @@
 
 int yywrap() {exit(0);};
 
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
     int i;
 
