diff -Nurd -x'*~' xmlto-0.0.25.orig/xmlif/xmlif.c xmlto-0.0.25/xmlif/xmlif.c
--- xmlto-0.0.25.orig/xmlif/xmlif.c	2011-03-03 10:19:13.000000000 -0500
+++ xmlto-0.0.25/xmlif/xmlif.c	2013-06-10 13:01:00.000000000 -0400
@@ -1081,7 +1081,7 @@
 
 static char **selections;	/* selection tokens */
 static int nselections;		/* number of selections */
-static ifsense;			/* sense of last `if' or unless seen */
+static int ifsense;		/* sense of last `if' or unless seen */
 static char *attribute;		/* last attribute scanned */
 
 struct stack_t {
@@ -2408,7 +2408,7 @@
 
 int yywrap() {exit(0);};
 
-main(int argc, char *argv[])
+int main(int argc, char *argv[])
 {
     int i;
 
