diff -Nurd -x'*~' autoconf-2.63.orig/lib/autoconf/libs.m4 autoconf-2.63/lib/autoconf/libs.m4
--- autoconf-2.63.orig/lib/autoconf/libs.m4	2008-08-22 08:05:31.000000000 -0400
+++ autoconf-2.63/lib/autoconf/libs.m4	2015-09-22 01:01:30.000000000 -0400
@@ -265,6 +265,7 @@
 /usr/athena/include
 /usr/local/x11r5/include
 /usr/lpp/Xamples/include
+/opt/X11/include
 
 /usr/openwin/include
 /usr/openwin/share/include'
