diff -rupN old/bfd/archures.c new/bfd/archures.c
--- old/bfd/archures.c	2013-11-08 10:02:26.000000000 +0000
+++ new/bfd/archures.c	2014-05-17 20:49:45.000000000 +0100
@@ -405,6 +405,7 @@ DESCRIPTION
 .#define bfd_mach_avrxmega5 105
 .#define bfd_mach_avrxmega6 106
 .#define bfd_mach_avrxmega7 107
+.#define bfd_mach_avrtiny10 201
 .  bfd_arch_bfin,        {* ADI Blackfin *}
 .#define bfd_mach_bfin          1
 .  bfd_arch_cr16,       {* National Semiconductor CompactRISC (ie CR16). *}
diff -rupN old/bfd/bfd-in2.h new/bfd/bfd-in2.h
--- old/bfd/bfd-in2.h	2013-11-18 08:40:15.000000000 +0000
+++ new/bfd/bfd-in2.h	2014-05-17 20:49:45.000000000 +0100
@@ -2160,6 +2160,7 @@ enum bfd_architecture
 #define bfd_mach_avrxmega5 105
 #define bfd_mach_avrxmega6 106
 #define bfd_mach_avrxmega7 107
+#define bfd_mach_avrtiny10 201
   bfd_arch_bfin,        /* ADI Blackfin */
 #define bfd_mach_bfin          1
   bfd_arch_cr16,       /* National Semiconductor CompactRISC (ie CR16). */
diff -rupN old/bfd/cpu-avr.c new/bfd/cpu-avr.c
--- old/bfd/cpu-avr.c	2013-11-04 15:33:37.000000000 +0000
+++ new/bfd/cpu-avr.c	2014-05-17 20:51:40.000000000 +0100
@@ -155,7 +155,10 @@ static const bfd_arch_info_type arch_inf
   N (24, bfd_mach_avrxmega6, "avr:106", FALSE, & arch_info_struct[16]),
 
   /* Xmega 7 */
-  N (24, bfd_mach_avrxmega7, "avr:107", FALSE, NULL)
+  N (24, bfd_mach_avrxmega7, "avr:107", FALSE, & arch_info_struct[17]),
+  
+  /* attiny 10 */
+  N (16, bfd_mach_avrtiny10, "avr:201", FALSE, NULL)
 
 };
 
diff -rupN old/bfd/elf32-avr.c new/bfd/elf32-avr.c
--- old/bfd/elf32-avr.c	2013-11-08 10:13:48.000000000 +0000
+++ new/bfd/elf32-avr.c	2014-05-17 20:49:45.000000000 +0100
@@ -1361,6 +1361,10 @@ bfd_elf_avr_final_write_processing (bfd 
     case bfd_mach_avrxmega7:
       val = E_AVR_MACH_XMEGA7;
       break;
+
+    case bfd_mach_avrtiny10:
+      val = E_AVR_MACH_AVRTINY10;
+      break;
     }
 
   elf_elfheader (abfd)->e_machine = EM_AVR;
@@ -1451,6 +1455,10 @@ elf32_avr_object_p (bfd *abfd)
 	case E_AVR_MACH_XMEGA7:
 	  e_set = bfd_mach_avrxmega7;
 	  break;
+
+	case E_AVR_MACH_AVRTINY10:
+	  e_set = bfd_mach_avrtiny10;
+	  break;
 	}
     }
   return bfd_default_set_arch_mach (abfd, bfd_arch_avr,
Binary files old/gas/config/.DS_Store and new/gas/config/.DS_Store differ
diff -rupN old/gas/config/tc-avr.c new/gas/config/tc-avr.c
--- old/gas/config/tc-avr.c	2013-11-04 15:33:37.000000000 +0000
+++ new/gas/config/tc-avr.c	2014-05-17 20:54:45.000000000 +0100
@@ -89,6 +89,7 @@ static struct mcu_type_s mcu_types[] =
   {"avrxmega5",  AVR_ISA_XMEGA,   bfd_mach_avrxmega5},
   {"avrxmega6",  AVR_ISA_XMEGA,   bfd_mach_avrxmega6},
   {"avrxmega7",  AVR_ISA_XMEGA,   bfd_mach_avrxmega7},
+  {"avrtiny10",  AVR_ISA_AVRTINY10, bfd_mach_avrtiny10},
   {"at90s1200",  AVR_ISA_1200,    bfd_mach_avr1},
   {"attiny11",   AVR_ISA_AVR1,    bfd_mach_avr1},
   {"attiny12",   AVR_ISA_AVR1,    bfd_mach_avr1},
@@ -280,6 +281,12 @@ static struct mcu_type_s mcu_types[] =
   {"atxmega256d3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega128a1", AVR_ISA_XMEGA, bfd_mach_avrxmega7},
   {"atxmega128a1u", AVR_ISA_XMEGAU, bfd_mach_avrxmega7},
+  {"attiny4",      AVR_ISA_AVRTINY10, bfd_mach_avrtiny10},
+  {"attiny5",      AVR_ISA_AVRTINY10, bfd_mach_avrtiny10},
+  {"attiny9",      AVR_ISA_AVRTINY10, bfd_mach_avrtiny10},
+  {"attiny10",     AVR_ISA_AVRTINY10, bfd_mach_avrtiny10},
+  {"attiny20",     AVR_ISA_AVRTINY10, bfd_mach_avrtiny10},
+  {"attiny40",     AVR_ISA_AVRTINY10, bfd_mach_avrtiny10},
   {NULL, 0, 0}
 };
 
@@ -462,6 +469,7 @@ md_show_usage (FILE *stream)
 	"                   avrxmega5 - XMEGA, > 64K, <= 128K FLASH, > 64K RAM\n"
 	"                   avrxmega6 - XMEGA, > 128K, <= 256K FLASH, <= 64K RAM\n"
 	"                   avrxmega7 - XMEGA, > 128K, <= 256K FLASH, > 64K RAM\n"
+	"                   avrtiny10 - tiny devices with 16 gp registers\n"
 	"                   or immediate microcontroller name.\n"));
   fprintf (stream,
       _("  -mall-opcodes    accept all AVR opcodes, even if not supported by MCU\n"
@@ -809,6 +817,17 @@ avr_operand (struct avr_opcodes_s *opcod
 	  op_mask = avr_get_constant (str, 31);
 	  str = input_line_pointer;
 	}
+	  if (strcmp(avr_mcu->name, "avrtiny10") == 0 
+	     || strcmp(avr_mcu->name, "attiny10") == 0
+	     || strcmp(avr_mcu->name, "attiny4")  == 0
+	     || strcmp(avr_mcu->name, "attiny5")  == 0
+	     || strcmp(avr_mcu->name, "attiny9")  == 0
+	     || strcmp(avr_mcu->name, "attiny20") == 0
+	     || strcmp(avr_mcu->name, "attiny40") == 0)
+	{
+		  if(op_mask < 16)
+		  as_bad (_("register number above 15 required"));
+	}
 
       if (op_mask <= 31)
 	{
Binary files old/include/elf/.DS_Store and new/include/elf/.DS_Store differ
diff -rupN old/include/elf/avr.h new/include/elf/avr.h
--- old/include/elf/avr.h	2013-11-04 15:33:39.000000000 +0000
+++ new/include/elf/avr.h	2014-05-17 20:49:45.000000000 +0100
@@ -48,6 +48,7 @@
 #define E_AVR_MACH_XMEGA5 105
 #define E_AVR_MACH_XMEGA6 106
 #define E_AVR_MACH_XMEGA7 107
+#define E_AVR_MACH_AVRTINY10 201
 
 /* Relocations.  */
 START_RELOC_NUMBERS (elf_avr_reloc_type)
Binary files old/include/opcode/.DS_Store and new/include/opcode/.DS_Store differ
diff -rupN old/include/opcode/avr.h new/include/opcode/avr.h
--- old/include/opcode/avr.h	2013-11-04 15:33:39.000000000 +0000
+++ new/include/opcode/avr.h	2014-05-17 20:57:55.000000000 +0100
@@ -71,6 +71,7 @@
                         AVR_ISA_SRAM | AVR_ISA_MEGA | AVR_ISA_MUL | \
                         AVR_ISA_ELPM | AVR_ISA_ELPMX | AVR_ISA_SPM | \
                         AVR_ISA_BRK | AVR_ISA_EIND | AVR_ISA_MOVW)
+#define AVR_ISA_AVRTINY10 (AVR_ISA_1200 | AVR_ISA_BRK | AVR_ISA_SRAM)
 
 #define REGISTER_P(x) ((x) == 'r'		\
 		       || (x) == 'd'		\
@@ -160,8 +161,8 @@ AVR_INSN (sez,  "",    "1001010000011000
 AVR_INSN (bclr, "S",   "100101001SSS1000", 1, AVR_ISA_1200, 0x9488)
 AVR_INSN (bset, "S",   "100101000SSS1000", 1, AVR_ISA_1200, 0x9408)
 
-AVR_INSN (icall,"",    "1001010100001001", 1, AVR_ISA_2xxx, 0x9509)
-AVR_INSN (ijmp, "",    "1001010000001001", 1, AVR_ISA_2xxx, 0x9409)
+AVR_INSN (icall,"",    "1001010100001001", 1, AVR_ISA_SRAM, 0x9509)
+AVR_INSN (ijmp, "",    "1001010000001001", 1, AVR_ISA_SRAM, 0x9409)
 
 AVR_INSN (lpm,  "?",   "1001010111001000", 1, AVR_ISA_TINY1,0x95c8)
 AVR_INSN (lpm,  "r,z", "1001000ddddd010+", 1, AVR_ISA_LPMX, 0x9004)
@@ -261,8 +262,8 @@ AVR_INSN (dec,  "r",   "1001010rrrrr1010
 AVR_INSN (inc,  "r",   "1001010rrrrr0011", 1, AVR_ISA_1200, 0x9403)
 AVR_INSN (lsr,  "r",   "1001010rrrrr0110", 1, AVR_ISA_1200, 0x9406)
 AVR_INSN (neg,  "r",   "1001010rrrrr0001", 1, AVR_ISA_1200, 0x9401)
-AVR_INSN (pop,  "r",   "1001000rrrrr1111", 1, AVR_ISA_2xxx, 0x900f)
-AVR_INSN (push, "r",   "1001001rrrrr1111", 1, AVR_ISA_2xxx, 0x920f)
+AVR_INSN (pop,  "r",   "1001000rrrrr1111", 1, AVR_ISA_SRAM, 0x900f)
+AVR_INSN (push, "r",   "1001001rrrrr1111", 1, AVR_ISA_SRAM, 0x920f)
 AVR_INSN (ror,  "r",   "1001010rrrrr0111", 1, AVR_ISA_1200, 0x9407)
 AVR_INSN (swap, "r",   "1001010rrrrr0010", 1, AVR_ISA_1200, 0x9402)
 
@@ -280,8 +281,8 @@ AVR_INSN (fmul, "a,a", "000000110ddd1rrr
 AVR_INSN (fmuls,"a,a", "000000111ddd0rrr", 1, AVR_ISA_MUL,  0x0380)
 AVR_INSN (fmulsu,"a,a","000000111ddd1rrr", 1, AVR_ISA_MUL,  0x0388)
 
-AVR_INSN (sts,  "i,r", "1001001ddddd0000", 2, AVR_ISA_2xxx, 0x9200)
-AVR_INSN (lds,  "r,i", "1001000ddddd0000", 2, AVR_ISA_2xxx, 0x9000)
+AVR_INSN (sts,  "i,r", "1001001ddddd0000", 2, AVR_ISA_SRAM, 0x9200)
+AVR_INSN (lds,  "r,i", "1001000ddddd0000", 2, AVR_ISA_SRAM, 0x9000)
 
    /* Special case for b+0, `e' must be next entry after `b',
       b={Y=1,Z=0}, ee={X=11,Y=10,Z=00}, !=1 if -e or e+ or X.  */
Binary files old/ld/.DS_Store and new/ld/.DS_Store differ
diff -rupN old/ld/Makefile.am new/ld/Makefile.am
--- old/ld/Makefile.am	2013-11-26 11:37:33.000000000 +0000
+++ new/ld/Makefile.am	2014-05-17 21:00:22.000000000 +0100
@@ -190,6 +190,7 @@ ALL_EMULATION_SOURCES = \
 	eavrxmega5.c \
 	eavrxmega6.c \
 	eavrxmega7.c \
+	eavrtiny10.c \
 	ecoff_i860.c \
 	ecoff_sparc.c \
 	ecrisaout.c \
@@ -873,6 +874,10 @@ eavrxmega7.c: $(srcdir)/emulparams/avrxm
   $(srcdir)/emultempl/avrelf.em $(ELF_DEPS) $(srcdir)/scripttempl/avr.sc \
   ${GEN_DEPENDS}
 	${GENSCRIPTS} avrxmega7 "$(tdir_avr2)"
+eavrtiny10.c: $(srcdir)/emulparams/avrtiny10.sh \
+  $(srcdir)/emultempl/avrelf.em $(ELF_DEPS) $(srcdir)/scripttempl/avr.sc \
+  ${GEN_DEPENDS}
+	${GENSCRIPTS} avrtiny10 "$(tdir_avr2)"
 ecoff_i860.c: $(srcdir)/emulparams/coff_i860.sh \
   $(srcdir)/emultempl/generic.em $(srcdir)/scripttempl/i860coff.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} coff_i860 "$(tdir_coff_i860)"
diff -rupN old/ld/Makefile.in new/ld/Makefile.in
--- old/ld/Makefile.in	2013-11-26 11:37:33.000000000 +0000
+++ new/ld/Makefile.in	2014-05-17 22:03:14.000000000 +0100
@@ -498,6 +498,7 @@ ALL_EMULATION_SOURCES = \
 	eavrxmega5.c \
 	eavrxmega6.c \
 	eavrxmega7.c \
+	eavrtiny10.c \
 	ecoff_i860.c \
 	ecoff_sparc.c \
 	ecrisaout.c \
@@ -1134,6 +1135,7 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eavrxmega5.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eavrxmega6.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eavrxmega7.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eavrtiny10.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/ecoff_i860.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/ecoff_sparc.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/ecrisaout.Po@am__quote@
@@ -2357,6 +2359,10 @@ eavrxmega7.c: $(srcdir)/emulparams/avrxm
   $(srcdir)/emultempl/avrelf.em $(ELF_DEPS) $(srcdir)/scripttempl/avr.sc \
   ${GEN_DEPENDS}
 	${GENSCRIPTS} avrxmega7 "$(tdir_avr2)"
+eavrtiny10.c: $(srcdir)/emulparams/avrtiny10.sh \
+  $(srcdir)/emultempl/avrelf.em $(ELF_DEPS) $(srcdir)/scripttempl/avr.sc \
+  ${GEN_DEPENDS}
+	${GENSCRIPTS} avrtiny10 "$(tdir_avr2)"
 ecoff_i860.c: $(srcdir)/emulparams/coff_i860.sh \
   $(srcdir)/emultempl/generic.em $(srcdir)/scripttempl/i860coff.sc ${GEN_DEPENDS}
 	${GENSCRIPTS} coff_i860 "$(tdir_coff_i860)"
diff -rupN old/ld/configure.tgt new/ld/configure.tgt
--- old/ld/configure.tgt	2013-11-26 11:37:33.000000000 +0000
+++ new/ld/configure.tgt	2014-05-17 20:49:45.000000000 +0100
@@ -138,7 +138,7 @@ arm*-*-uclinux*)	targ_emul=armelf_linux
 arm-*-vxworks)		targ_emul=armelf_vxworks ;;
 arm*-*-conix*)		targ_emul=armelf ;;
 avr-*-*)		targ_emul=avr2
-			targ_extra_emuls="avr1 avr25 avr3 avr31 avr35 avr4 avr5 avr51 avr6 avrxmega1 avrxmega2 avrxmega3 avrxmega4 avrxmega5 avrxmega6 avrxmega7"
+			targ_extra_emuls="avr1 avr25 avr3 avr31 avr35 avr4 avr5 avr51 avr6 avrxmega1 avrxmega2 avrxmega3 avrxmega4 avrxmega5 avrxmega6 avrxmega7 avrtiny10"
 			;;
 bfin-*-elf)		targ_emul=elf32bfin;
 			targ_extra_emuls="elf32bfinfd"
diff -rupN old/ld/emulparams/avrtiny10.sh new/ld/emulparams/avrtiny10.sh
--- old/ld/emulparams/avrtiny10.sh	1970-01-01 01:00:00.000000000 +0100
+++ new/ld/emulparams/avrtiny10.sh	2014-05-17 20:49:45.000000000 +0100
@@ -0,0 +1,12 @@
+ARCH=avr:201
+MACHINE=
+SCRIPT_NAME=avr
+OUTPUT_FORMAT="elf32-avr"
+MAXPAGESIZE=1
+EMBEDDED=yes
+TEMPLATE_NAME=elf32
+
+TEXT_LENGTH=4K
+DATA_ORIGIN=0x800040
+DATA_LENGTH=0x140
+EXTRA_EM_FILE=avrelf
