diff -uwNr --exclude=.svn --exclude=Makefile --exclude=Makefile.fpc --exclude=Makefile.compiled --exclude='*.rsj' --exclude='*.bak' --exclude='*.po' lazarus/lcl/interfaces/carbon/carbonwinapi.inc lazarus.w/lcl/interfaces/carbon/carbonwinapi.inc
--- lazarus/lcl/interfaces/carbon/carbonwinapi.inc	2016-04-30 11:12:40.000000000 +0100
+++ lazarus.w/lcl/interfaces/carbon/carbonwinapi.inc	2016-09-08 22:17:24.000000000 +0100
@@ -1811,8 +1811,11 @@
       DebugLn('TCarbonWidgetSet.GetParent Widget: ' + DbgS(TCarbonControl(Handle).Widget));
     {$ENDIF}
     Result := HWnd(GetCarbonWidget(HIViewGetSuperview(TCarbonControl(Handle).Widget)));
-    if Result = 0 then // no parent control => then parent is a window?
+    if Result = 0 then begin // no parent control => then parent is a window?
       Result := HWnd(GetCarbonWidget(HIViewGetWindow(TCarbonControl(Handle).Widget)));
+      if Result = Handle then
+        Result := 0; // Sanity check - otherwise FindLCLWindow can be stuck in a loop
+    end;
   end;
   // Carbon windows has no parent
   
