diff -ruN make-3.81/Makefile.in make-3.81-patched/Makefile.in
--- make-3.81/Makefile.in	2006-04-01 08:40:04.000000000 +0200
+++ make-3.81-patched/Makefile.in	2007-07-24 22:28:38.000000000 +0200
@@ -93,7 +93,7 @@
 	expand.c file.c function.c getopt.c getopt1.c implicit.c job.c \
 	main.c misc.c read.c remake.c remote-stub.c remote-cstms.c \
 	rule.c signame.c strcache.c variable.c version.c vpath.c \
-	hash.c
+	hash.c next.c
 @USE_CUSTOMS_FALSE@am__objects_1 = remote-stub$U.$(OBJEXT)
 @USE_CUSTOMS_TRUE@am__objects_1 = remote-cstms$U.$(OBJEXT)
 am_make_OBJECTS = ar$U.$(OBJEXT) arscan$U.$(OBJEXT) \
@@ -104,7 +104,7 @@
 	read$U.$(OBJEXT) remake$U.$(OBJEXT) $(am__objects_1) \
 	rule$U.$(OBJEXT) signame$U.$(OBJEXT) strcache$U.$(OBJEXT) \
 	variable$U.$(OBJEXT) version$U.$(OBJEXT) vpath$U.$(OBJEXT) \
-	hash$U.$(OBJEXT)
+	hash$U.$(OBJEXT) next$U.$(OBJEXT)
 make_OBJECTS = $(am_make_OBJECTS)
 am__DEPENDENCIES_1 =
 @WINDOWSENV_TRUE@am__DEPENDENCIES_2 = $(am__DEPENDENCIES_1)
@@ -267,7 +267,7 @@
 make_SOURCES = ar.c arscan.c commands.c default.c dir.c expand.c file.c \
 		function.c getopt.c getopt1.c implicit.c job.c main.c \
 		misc.c read.c remake.c $(remote) rule.c signame.c \
-		strcache.c variable.c version.c vpath.c hash.c
+		strcache.c variable.c version.c vpath.c hash.c next.c
 
 EXTRA_make_SOURCES = vmsjobs.c remote-stub.c remote-cstms.c
 noinst_HEADERS = commands.h dep.h filedef.h job.h make.h rule.h variable.h \
@@ -454,6 +454,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/version$U.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/vmsjobs$U.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/vpath$U.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/next$U.Po@am__quote@
 
 .c.o:
 @am__fastdepCC_TRUE@	if $(COMPILE) -MT $@ -MD -MP -MF "$(DEPDIR)/$*.Tpo" -c -o $@ $<; \
@@ -545,7 +546,7 @@
 main_.$(OBJEXT) misc_.$(OBJEXT) read_.$(OBJEXT) remake_.$(OBJEXT) \
 remote-cstms_.$(OBJEXT) remote-stub_.$(OBJEXT) rule_.$(OBJEXT) \
 signame_.$(OBJEXT) strcache_.$(OBJEXT) variable_.$(OBJEXT) \
-version_.$(OBJEXT) vmsjobs_.$(OBJEXT) vpath_.$(OBJEXT) : $(ANSI2KNR)
+version_.$(OBJEXT) vmsjobs_.$(OBJEXT) vpath_.$(OBJEXT) next_.$(OBJEXT) : $(ANSI2KNR)
 uninstall-info-am:
 install-man1: $(man1_MANS) $(man_MANS)
 	@$(NORMAL_INSTALL)
diff -ruN make-3.81/default.c make-3.81-patched/default.c
--- make-3.81/default.c	2006-02-11 23:16:04.000000000 +0100
+++ make-3.81-patched/default.c	2007-07-24 22:28:38.000000000 +0200
@@ -45,7 +45,7 @@
 .mod .sym .def .h .info .dvi .tex .texinfo .texi .txinfo \
 .w .ch .web .sh .elc .el .obj .exe .dll .lib";
 #else
-  = ".out .a .ln .o .c .cc .C .cpp .p .f .F .r .y .l .s .S \
+  = ".out .a .ln .o .c .cc .C .cpp .p .f .F .m .r .y .l .ym .lm .s .S \
 .mod .sym .def .h .info .dvi .tex .texinfo .texi .txinfo \
 .w .ch .web .sh .elc .el";
 #endif
@@ -192,6 +192,8 @@
     "$(LINK.cpp) $^ $(LOADLIBES) $(LDLIBS) -o $@",
     ".f",
     "$(LINK.f) $^ $(LOADLIBES) $(LDLIBS) -o $@",
+    ".m",
+    "$(LINK.m) $^ $(LOADLIBES) $(LDLIBS) -o $@",
     ".p",
     "$(LINK.p) $^ $(LOADLIBES) $(LDLIBS) -o $@",
     ".F",
@@ -221,6 +223,8 @@
     "$(COMPILE.cpp) $(OUTPUT_OPTION) $<",
     ".f.o",
     "$(COMPILE.f) $(OUTPUT_OPTION) $<",
+    ".m.o",
+    "$(COMPILE.m) $(OUTPUT_OPTION) $<",
     ".p.o",
     "$(COMPILE.p) $(OUTPUT_OPTION) $<",
     ".F.o",
@@ -250,6 +254,11 @@
     ".l.c",
     "@$(RM) $@ \n $(LEX.l) $< > $@",
 
+    ".ym.m",
+    "$(YACC.m) $< \n mv -f y.tab.c $@",
+    ".lm.m",
+    "@$(RM) $@ \n $(LEX.m) $< > $@",
+
     ".F.f",
     "$(PREPROCESS.F) $(OUTPUT_OPTION) $<",
     ".r.f",
@@ -304,6 +313,10 @@
 
 static char *default_variables[] =
   {
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+    "GNUMAKE", "YES",	/* I'm not sure who uses this.  Dave Payne 8/10/99 */
+    "MAKEFILEPATH", "$(NEXT_ROOT)/Developer/Makefiles",
+#endif /* __APPLE__ || NeXT || NeXT_PDO */
 #ifdef VMS
 #ifdef __ALPHA
     "ARCH", "ALPHA",
@@ -468,6 +481,8 @@
     "LINK.o", "$(CC) $(LDFLAGS) $(TARGET_ARCH)",
     "COMPILE.c", "$(CC) $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -c",
     "LINK.c", "$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $(TARGET_ARCH)",
+    "COMPILE.m", "$(COMPILE.c)",
+    "LINK.m", "$(LINK.c)",
     "COMPILE.cc", "$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -c",
     "COMPILE.C", "$(COMPILE.cc)",
     "COMPILE.cpp", "$(COMPILE.cc)",
@@ -476,6 +491,8 @@
     "LINK.cpp", "$(LINK.cc)",
     "YACC.y", "$(YACC) $(YFLAGS)",
     "LEX.l", "$(LEX) $(LFLAGS) -t",
+    "YACC.m", "$(YACC) $(YFLAGS)",
+    "LEX.m", "$(LEX) $(LFLAGS) -t",
     "COMPILE.f", "$(FC) $(FFLAGS) $(TARGET_ARCH) -c",
     "LINK.f", "$(FC) $(FFLAGS) $(LDFLAGS) $(TARGET_ARCH)",
     "COMPILE.F", "$(FC) $(FFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -c",
diff -ruN make-3.81/expand.c make-3.81-patched/expand.c
--- make-3.81/expand.c	2006-03-15 04:31:30.000000000 +0100
+++ make-3.81-patched/expand.c	2007-07-24 22:28:38.000000000 +0200
@@ -76,7 +76,11 @@
 
 /* Return a pointer to the beginning of the variable buffer.  */
 
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+char *
+#else
 static char *
+#endif
 initialize_variable_output (void)
 {
   /* If we don't have a variable output buffer yet, get one.  */
@@ -90,6 +94,35 @@
 
   return variable_buffer;
 }
+
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+char *
+save_variable_output(savelenp)
+    int *savelenp;
+{
+  char *save;
+
+  save = variable_buffer;
+  *savelenp = variable_buffer_length;
+  
+  variable_buffer = 0;
+  variable_buffer_length = 0;
+
+  return (save);
+}
+
+void
+restore_variable_output (save, savelen)
+    char *save;
+    int savelen;
+{
+  if (variable_buffer != 0)
+    free (variable_buffer);
+
+  variable_buffer = save;
+  variable_buffer_length = savelen;
+}
+#endif /* __APPLE__ || NeXT || NeXT_PDO */
 
 /* Recursively expand V.  The returned string is malloc'd.  */
 
diff -ruN make-3.81/file.c make-3.81-patched/file.c
--- make-3.81/file.c	2006-03-17 15:24:20.000000000 +0100
+++ make-3.81-patched/file.c	2007-07-24 22:28:38.000000000 +0200
@@ -207,6 +207,9 @@
   rehash_file (from_file, to_hname);
   while (from_file)
     {
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+      from_file->old_name = from_file->name;
+#endif  /* __APPLE__ || NeXT || NeXT_PDO */
       from_file->name = from_file->hname;
       from_file = from_file->prev;
     }
diff -ruN make-3.81/filedef.h make-3.81-patched/filedef.h
--- make-3.81/filedef.h	2006-02-11 23:16:04.000000000 +0100
+++ make-3.81-patched/filedef.h	2007-07-24 22:28:38.000000000 +0200
@@ -42,6 +42,10 @@
 				   entries for the same file.  */
     struct file *last;          /* Last entry for the same file name.  */
 
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+    char *old_name;
+#endif
+
     /* File that this file was renamed to.  After any time that a
        file could be renamed, call `check_renamed' (below).  */
     struct file *renamed;
diff -ruN make-3.81/implicit.c make-3.81-patched/implicit.c
--- make-3.81/implicit.c	2006-04-01 08:36:40.000000000 +0200
+++ make-3.81-patched/implicit.c	2007-07-24 22:28:38.000000000 +0200
@@ -786,8 +786,24 @@
 
   /* RULE is nil if the loop went all the way
      through the list and everything failed.  */
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO) /* for NEXT_VPATH_FLAG support */
+  if (rule == 0) {
+      if ((next_flag & NEXT_VPATH_FLAG) && file->old_name != 0) {
+	  int v;
+	  char *save_name = file->name;
+	  file->name = file->old_name;
+	  file->old_name = 0;
+	  v = pattern_search(file, archive, depth, recursions);
+	  file->old_name = file->name;
+	  file->name = save_name;
+	  return v;
+      }
+      return 0;
+  }
+#else
   if (rule == 0)
     goto done;
+#endif	/* __APPLE__ || NeXT || NeXT_PDO */
 
   foundrule = i;
 
diff -ruN make-3.81/job.c make-3.81-patched/job.c
--- make-3.81/job.c	2006-03-20 04:03:04.000000000 +0100
+++ make-3.81-patched/job.c	2007-07-24 22:28:46.000000000 +0200
@@ -1107,8 +1107,16 @@
 #else
       (argv[0] && !strcmp (argv[0], "/bin/sh"))
 #endif
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+      /* allow either -ec or -c */
+      && ((argv[1]
+	   && argv[1][0] == '-' && argv[1][1] == 'c' && argv[1][2] == '\0') ||
+          (argv[1]
+	   && argv[1][0] == '-' && argv[1][1] == 'e' && argv[1][2] == 'c' && argv[1][3] == '\0'))
+#else
       && (argv[1]
           && argv[1][0] == '-' && argv[1][1] == 'c' && argv[1][2] == '\0')
+#endif __APPLE__ || NeXT || NeXT_PDO
       && (argv[2] && argv[2][0] == ':' && argv[2][1] == '\0')
       && argv[3] == NULL)
     {
@@ -1601,6 +1609,19 @@
 						     file);
     }
 
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO) /* for NEXT_VPATH_FLAG support */
+  if (next_flag & NEXT_VPATH_FLAG) {
+      for (i = 0; i < cmds->ncommand_lines; ++i) {
+	  char *line;
+	  if (lines[i] != 0) {
+	      line = allocated_vpath_expand_for_file (lines[i], file);
+	      free (lines[i]);
+	      lines[i] = line;
+	  }
+      }
+  }
+#endif	/* __APPLE__ || NeXT || NeXT_PDO */
+
   /* Start the command sequence, record it in a new
      `struct child', and add that to the chain.  */
 
@@ -2423,23 +2444,7 @@
 	    }
 	  else if (*p == '\\' && p[1] == '\n')
             {
-              /* Backslash-newline is handled differently depending on what
-                 kind of string we're in: inside single-quoted strings you
-                 keep them; in double-quoted strings they disappear.
-	         For DOS/Windows/OS2, if we don't have a POSIX shell,
-		 we keep the pre-POSIX behavior of removing the
-		 backslash-newline.  */
-              if (instring == '"'
-#if defined (__MSDOS__) || defined (__EMX__) || defined (WINDOWS32)
-		  || !unixy_shell
-#endif
-		  )
                 ++p;
-              else
-                {
-                  *(ap++) = *(p++);
-                  *(ap++) = *p;
-                }
               /* If there's a TAB here, skip it.  */
               if (p[1] == '\t')
                 ++p;
@@ -2690,16 +2695,33 @@
        argument list.  */
 
     unsigned int shell_len = strlen (shell);
+    unsigned int line_len = strlen (line);
+    char *new_line;
+    char *command_ptr = NULL; /* used for batch_mode_shell mode */
+
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+    char *minus_c;
+    int minus_c_len;
+
+    if (next_flag & NEXT_ERREXIT_FLAG) {
+      minus_c = " -ec ";
+      minus_c_len = 5;
+    } else {
+      minus_c = " -c ";
+      minus_c_len = 4;
+    }
+#else
 #ifndef VMS
     static char minus_c[] = " -c ";
+    int minus_c_len = 4;
 #else
     static char minus_c[] = "";
+    int minus_c_len = 0;
 #endif
-    unsigned int line_len = strlen (line);
+#endif /* __APPLE__ || NeXT || NeXT_PDO */
 
-    char *new_line = (char *) alloca (shell_len + (sizeof (minus_c) - 1)
-				      + (line_len * 2) + 1);
-    char *command_ptr = NULL; /* used for batch_mode_shell mode */
+    new_line = (char *) alloca (shell_len + minus_c_len
+				+ (line_len * 2) + 1);
 
 # ifdef __EMX__ /* is this necessary? */
     if (!unixy_shell)
@@ -2709,8 +2731,8 @@
     ap = new_line;
     bcopy (shell, ap, shell_len);
     ap += shell_len;
-    bcopy (minus_c, ap, sizeof (minus_c) - 1);
-    ap += sizeof (minus_c) - 1;
+    bcopy (minus_c, ap, minus_c_len);
+    ap += minus_c_len;
     command_ptr = ap;
     for (p = line; *p != '\0'; ++p)
       {
@@ -2721,22 +2743,6 @@
 	  }
 	else if (*p == '\\' && p[1] == '\n')
 	  {
-	    /* POSIX says we keep the backslash-newline, but throw out
-               the next char if it's a TAB.  If we don't have a POSIX
-               shell on DOS/Windows/OS2, mimic the pre-POSIX behavior
-               and remove the backslash/newline.  */
-#if defined (__MSDOS__) || defined (__EMX__) || defined (WINDOWS32)
-# define PRESERVE_BSNL  unixy_shell
-#else
-# define PRESERVE_BSNL  1
-#endif
-	    if (PRESERVE_BSNL)
-	      {
-		*(ap++) = '\\';
-		*(ap++) = '\\';
-		*(ap++) = '\n';
-	      }
-
 	    ++p;
 	    if (p[1] == '\t')
 	      ++p;
@@ -2761,7 +2767,7 @@
 #endif
 	*ap++ = *p;
       }
-    if (ap == new_line + shell_len + sizeof (minus_c) - 1)
+    if (ap == new_line + shell_len + minus_c_len)
       /* Line was empty.  */
       return 0;
     *ap = '\0';
@@ -2891,10 +2897,10 @@
            instead of recursively calling ourselves, because we
            cannot backslash-escape the special characters (see above).  */
         new_argv = (char **) xmalloc (sizeof (char *));
-        line_len = strlen (new_line) - shell_len - sizeof (minus_c) + 1;
+        line_len = strlen (new_line) - shell_len - minus_c_len;
         new_argv[0] = xmalloc (line_len + 1);
         strncpy (new_argv[0],
-                 new_line + shell_len + sizeof (minus_c) - 1, line_len);
+                 new_line + shell_len + minus_c_len, line_len);
         new_argv[0][line_len] = '\0';
       }
 #else
diff -ruN make-3.81/main.c make-3.81-patched/main.c
--- make-3.81/main.c	2006-03-20 03:36:37.000000000 +0100
+++ make-3.81-patched/main.c	2007-07-24 22:28:38.000000000 +0200
@@ -90,6 +90,27 @@
 static char *quote_for_env PARAMS ((char *out, char *in));
 static void initialize_global_hash_tables PARAMS ((void));
 
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+/*
+ * The various Apple, NeXT, and NeXT PDO hacks are no longer enabled by
+ * default, so that default behaves matches GNU make.
+ *
+ * The hacks can be enabled selectively enabling the following options.
+ * They can be enabled by setting the variable USE_APPLE_PB_SUPPORT to one
+ * or more of the options, or by specify "-N <option>" on the command line.
+ *
+ *	all		Turn on all NeXT features.
+ *
+ *	makefiles	DON'T remake Makefiles
+ *
+ *	quiet		Be quiet: warn about using vpath compatibility
+ *			mode or missing targets or overriding implicit rules.
+ *
+ *	vpath		Use the System V vpath compatibility mode.
+ *
+ *	errexit		Use "sh -ec" (instead of "sh -c") to execute rules.
+ */
+#endif
 
 /* The structure that describes an accepted command switch.  */
 
@@ -213,6 +234,13 @@
 
 static struct stringlist *makefiles = 0;
 
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+/* Apple's hacks are OFF by default, and are enabled by putting the
+   "use-apple-pbhacks-*" directives in the pb_makefiles. */
+unsigned int next_flag = 0;
+static struct stringlist *next_flag_list = 0;
+#endif
+
 /* Number of job slots (commands that can be run at once).  */
 
 unsigned int job_slots = 1;
@@ -354,6 +382,11 @@
                               Consider FILE to be infinitely new.\n"),
     N_("\
   --warn-undefined-variables  Warn when an undefined variable is referenced.\n"),
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+    N_("\
+  -N OPTION, --NeXT-option=OPTION\n\
+                              Turn on value of NeXT OPTION.\n"),
+#endif /* __APPLE__ || NeXT || NeXT_PDO */
     NULL
   };
 
@@ -416,6 +449,9 @@
     { 'W', string, (char *) &new_files, 0, 0, 0, 0, 0, "what-if" },
     { CHAR_MAX+4, flag, (char *) &warn_undefined_variables_flag, 1, 1, 0, 0, 0,
       "warn-undefined-variables" },
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+    { 'N', string, (char *) &next_flag_list, 0, 0, 0, 0, 0, "NeXT-option" },
+#endif /* __APPLE__ || NeXT || NeXT_PDO */
     { 0, 0, 0, 0, 0, 0, 0, 0, 0 }
   };
 
@@ -1230,6 +1266,27 @@
   decode_env_switches (STRING_SIZE_TUPLE ("MFLAGS"));
 #endif
   decode_switches (argc, argv, 0);
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+  if (next_flag_list != 0) {
+        char **p;
+      next_flag = 0;
+      for (p = next_flag_list->list; *p != 0; p++) {
+	  if (strcmp(*p, "vpath") == 0) {
+	      next_flag |= NEXT_VPATH_FLAG;
+	  } else if (strcmp(*p, "quiet") == 0) {
+	      next_flag |= NEXT_QUIET_FLAG;
+	  } else if (strcmp(*p, "makefiles") == 0) {
+	      next_flag |= NEXT_MAKEFILES_FLAG;
+	  } else if (strcmp(*p, "errexit") == 0) {
+	      next_flag |= NEXT_ERREXIT_FLAG;
+	  } else if (strcmp(*p, "all") == 0) {
+	      next_flag = NEXT_ALL_FLAGS;
+	  } else {
+	      error ("Unrecognized flag `%s'.", *p);
+	  }
+      }
+  }
+#endif	/* __APPLE__ || NeXT || NeXT_PDO */
 #ifdef WINDOWS32
   if (suspend_flag) {
         fprintf(stderr, "%s (pid = %ld)\n", argv[0], GetCurrentProcessId());
@@ -1406,9 +1463,11 @@
       makelevel = 0;
   }
 
+#if !(defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO))
   /* Except under -s, always do -w in sub-makes and under -C.  */
   if (!silent_flag && (directories != 0 || makelevel > 0))
     print_directory_flag = 1;
+#endif
 
   /* Let the user disable that with --no-print-directory.  */
   if (inhibit_print_directory_flag)
@@ -1804,6 +1863,9 @@
   remote_setup ();
 
   if (read_makefiles != 0)
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+      if (!(next_flag & NEXT_MAKEFILES_FLAG))
+#endif
     {
       /* Update any makefiles if necessary.  */
 
diff -ruN make-3.81/make.h make-3.81-patched/make.h
--- make-3.81/make.h	2006-02-16 00:54:43.000000000 +0100
+++ make-3.81-patched/make.h	2007-07-24 22:28:38.000000000 +0200
@@ -508,6 +508,21 @@
 extern int warn_undefined_variables_flag, posix_pedantic, not_parallel;
 extern int second_expansion, clock_skew_detected, rebuilding_makefiles;
 
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+extern unsigned int next_flag;
+#define NEXT_VPATH_FLAG			(1 << 0)
+#define NEXT_QUIET_FLAG			(1 << 1)
+#define NEXT_MAKEFILES_FLAG		(1 << 2)
+#define NEXT_ERREXIT_FLAG		(1 << 3)
+#define NEXT_ALL_FLAGS			(NEXT_VPATH_FLAG | 	\
+					 NEXT_QUIET_FLAG | 	\
+					 NEXT_MAKEFILES_FLAG |	\
+					 NEXT_ERREXIT_FLAG)
+
+extern int general_vpath_search();
+extern char *allocated_vpath_expand_for_file();
+#endif	/* __APPLE__ || NeXT || NeXT_PDO */
+
 /* can we run commands via 'sh -c xxx' or must we use batch files? */
 extern int batch_mode_shell;
 
diff -ruN make-3.81/next.c make-3.81-patched/next.c
--- make-3.81/next.c	1970-01-01 01:00:00.000000000 +0100
+++ make-3.81-patched/next.c	2007-07-24 22:28:38.000000000 +0200
@@ -0,0 +1,226 @@
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO) /* for NEXT_VPATH_FLAG support */
+#include "make.h"
+#include "job.h"
+#include "commands.h"
+#include "filedef.h"
+#include "variable.h"
+#include "dep.h"
+#include <sys/types.h>
+#include <sys/stat.h>
+
+#if !__STDC__
+#define const
+#endif
+
+/* Search through a pathlist for a file.  `search_path' will parse `path',
+ * a list of pathnames separated by colons, prepending each pathname to
+ * `filename'.  The resulting filename will be checked for existence via
+ * stat().
+ */
+static int
+search_path(path, file)
+    const char *path;
+    char **file;
+{
+    int n, length;
+    char *filename, *name;
+    const char *nextchar, *lastchar;
+
+    filename = *file;
+    n = strlen(filename);
+    length = strlen(path) + n + 10;
+    name = alloca(length) + length - 1;
+    *name = '\0';
+
+    filename += n;
+    while (--n >= 0)
+	*--name = *--filename;
+
+    if (*name == '/' || path == 0)
+	path = "";
+
+    /* Strip off leading './'s, if any. */
+    while (*name == '.' && *(name + 1) == '/')
+	name += 2;
+
+    do {
+	/* Advance to the end of the next path in our path list. */
+	nextchar = path;
+#if defined (__MSDOS__) || defined (WIN32)
+	while ((*nextchar != '\0' && *nextchar != ':' && !isspace (*nextchar))
+	|| (*nextchar == ':' && nextchar - path == 1
+	  && (nextchar[1] == '/' || nextchar[1] == '\\')))
+	    nextchar++;
+#else
+	while (*nextchar != '\0' && *nextchar != ':' && !isspace (*nextchar))
+	    nextchar++;
+#endif
+
+	lastchar = nextchar;
+	filename = name;
+
+	/* If we actually have a path, prepend the file name with a '/'. */
+	if (nextchar != path)
+	    *--filename = '/';
+
+	/* Prepend the file name with the path. */
+	while (nextchar != path)
+	    *--filename = *--nextchar;
+
+	path = (*lastchar) ? lastchar + 1 : lastchar;
+
+	{
+	    struct stat s;
+	    if (stat(filename, &s) >= 0) {
+		/* We have found a file.
+		 * Store the name we found into *FILE for the caller.  */
+		*file = savestring(filename, strlen(filename));
+		return (1);
+	    }
+	}
+    } while (*path != 0);
+    return (0);
+}
+
+int
+general_vpath_search(file)
+    char **file;
+{
+    int s;
+    int savelen;
+    char *vpath, *save;
+    
+    save = save_variable_output(&savelen);
+    vpath = variable_expand ("$(VPATH)");
+    if (**file == '/' || *vpath == '\0') {
+	restore_variable_output(save, savelen);
+	return 0;
+    }
+    s = search_path(vpath, file);
+    restore_variable_output(save, savelen);
+    return s;
+}
+
+static int
+match_dep(filename, file)
+    char **filename;
+    struct file *file;
+{
+    struct dep *d;
+
+    /* don't substitute for . or .. */
+    if (!strcmp (*filename, ".") || !strcmp (*filename, ".."))
+        return 0;
+    
+    for (d = file->deps; d != 0; d = d->next) {
+	if (d->file->old_name != 0) {
+	    if (strcmp(*filename, d->file->old_name) == 0) {
+		*filename = dep_name(d);
+		return 1;
+	    }
+	}
+	if (strcmp(*filename, dep_name(d)) == 0) {
+	    if (general_vpath_search(filename))
+		return 1;
+	}
+    }
+    return 0;
+}
+
+/* Scan LINE for vpath references. */
+
+static char *
+vpath_expand(line, file)
+    char *line;
+    struct file *file;
+{
+    struct variable *v;
+    char *p, *p1, *o;
+    static char *meta = 0;
+
+    if (meta == 0) {
+	static char buffer[256] = {0};
+	meta = buffer;
+	meta['\0'] = 1;
+	for (p = "=|^();&<>*?[]:$`'\"\\\n"; *p != 0; p++)
+	    meta[*p] = 1;
+    }
+
+    p = line;
+    o = initialize_variable_output ();
+
+    while (1) {
+	/* Copy all following uninteresting chars all at once to the
+	   variable output buffer, and skip them.  Uninteresting chars end
+	   at the next space or semicolon. */
+
+	for (p1 = p; *p1 != 0 && (isspace(*p1) || meta[*p1]); p1++)
+	    ;
+	o = variable_buffer_output (o, p, p1 - p);
+	if (*p1 == 0)
+	    break;
+	p = p1;
+	while (*p1 != 0 && !(isspace(*p1) || meta[*p1]))
+	    p1++;
+	{
+	    unsigned int n = p1 - p;
+	    char *buffer = malloc(n + 1);
+	    char *filename = buffer;
+	    struct dep *dep;
+
+	    strncpy(filename, p, n);
+	    filename[n] = 0;
+
+	    if (match_dep(&filename, file)) {
+		static struct file *last_file = 0;
+		if (last_file != file) {
+		    last_file = file;
+		    if (!(next_flag & NEXT_QUIET_FLAG)) {
+		      error(&file->cmds->fileinfo,
+			    "Using old-style VPATH substitution.");
+		      error(&file->cmds->fileinfo,
+			    "Consider using automatic variable substitution instead.");
+		    }
+		}
+		o = variable_buffer_output (o, filename, strlen(filename));
+	    } else {
+		o = variable_buffer_output (o, filename, n);
+	    }
+	    p = p1;
+	    free(buffer);
+	}
+	if (*p == '\0')
+	    break;
+    }
+    (void) variable_buffer_output (o, "", 1);
+    return initialize_variable_output ();
+}
+
+char *
+allocated_vpath_expand_for_file(line, file)
+    char *line;
+    struct file *file;
+{
+    char *save, *value;
+    struct variable_set_list *save_set_list;
+    int savelen;
+
+    if (file == 0)
+	fatal(NILF, "Can't do VPATH expansion on a null file.\n");
+    
+    save = save_variable_output (&savelen);
+    
+    save_set_list = current_variable_set_list;
+    current_variable_set_list = file->variables;
+    reading_file = &file->cmds->fileinfo;
+    value = vpath_expand (line, file);
+    current_variable_set_list = save_set_list;
+    reading_file = 0;
+
+    value = savestring (value, strlen (value));
+    
+    restore_variable_output (save, savelen);
+    
+    return value;
+}
+#endif	/* __APPLE__ || NeXT || NeXT_PDO */
diff -ruN make-3.81/read.c make-3.81-patched/read.c
--- make-3.81/read.c	2006-03-17 15:24:20.000000000 +0100
+++ make-3.81-patched/read.c	2007-07-24 22:28:38.000000000 +0200
@@ -1972,6 +1972,9 @@
 	    fatal (flocp,
                    _("target file `%s' has both : and :: entries"), f->name);
 
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+          if (!(next_flag & NEXT_QUIET_FLAG)) {
+#endif
 	  /* If CMDS == F->CMDS, this target was listed in this rule
 	     more than once.  Just give a warning since this is harmless.  */
 	  if (cmds != 0 && cmds == f->cmds)
@@ -1991,6 +1994,9 @@
                      _("warning: ignoring old commands for target `%s'"),
                      f->name);
 	    }
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+	      }
+#endif
 
 	  f->is_target = 1;
 
diff -ruN make-3.81/remake.c make-3.81-patched/remake.c
--- make-3.81/remake.c	2006-03-20 03:36:37.000000000 +0100
+++ make-3.81-patched/remake.c	2007-07-24 22:28:38.000000000 +0200
@@ -223,6 +223,9 @@
 		     or not at all.  G->changed will have been set above if
 		     any commands were actually started for this goal.  */
 		  && file->update_status == 0 && !g->changed
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+		  && !(next_flag & NEXT_QUIET_FLAG)
+#endif
 		  /* Never give a message under -s or -q.  */
 		  && !silent_flag && !question_flag)
 		message (1, ((file->phony || file->cmds == 0)
@@ -481,6 +484,9 @@
 
       if (is_updating (d->file))
 	{
+#if __APPLE__ || NeXT || NeXT_PDO
+         if (!(next_flag & NEXT_QUIET_FLAG))
+#endif
 	  error (NILF, _("Circular %s <- %s dependency dropped."),
 		 file->name, d->file->name);
 	  /* We cannot free D here because our the caller will still have
@@ -740,6 +746,9 @@
 
       while (file)
         {
+#if __APPLE__ || NeXT || NeXT_PDO
+          file->old_name = file->name;
+#endif	/* __APPLE__ || NeXT || NeXT_PDO */
           file->name = file->hname;
           file = file->prev;
         }
@@ -993,6 +1002,9 @@
 
 	      if (is_updating (d->file))
 		{
+#if __APPLE__ || NeXT || NeXT_PDO
+		  if (!(next_flag & NEXT_QUIET_FLAG))
+#endif
 		  error (NILF, _("Circular %s <- %s dependency dropped."),
 			 file->name, d->file->name);
 		  if (lastd == 0)
@@ -1106,12 +1118,23 @@
 	   Pretend it was successfully remade.  */
 	file->update_status = 0;
       else
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+      {
+	char *name = file->name;
+	if ((next_flag & NEXT_VPATH_FLAG) && general_vpath_search(&name)) {
+	  free(name);
+	  file->update_status = 0;
+	} else
+#endif /* defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO) */
         {
           /* This is a dependency file we cannot remake.  Fail.  */
           if (!rebuilding_makefiles || !file->dontcare)
             complain (file);
           file->update_status = 2;
         }
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+      }
+#endif
     }
   else
     {
diff -ruN make-3.81/variable.c make-3.81-patched/variable.c
--- make-3.81/variable.c	2006-03-08 21:15:08.000000000 +0100
+++ make-3.81-patched/variable.c	2007-07-24 22:28:38.000000000 +0200
@@ -144,6 +144,40 @@
 
 /* Implement variables.  */
 
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+static void check_apple_pb_support (name, length, value)
+     char *name;
+     unsigned int length;
+     char *value;
+{
+  char *p;
+
+  if (length == 20 && !strncmp (name, "USE_APPLE_PB_SUPPORT", length)) {
+    for (p = value; *p != '\0'; p++) {
+      if (isspace (*p)) {
+	continue;
+      }
+      if (!strncmp (p, "all", 3)) {
+	p += 3;
+	next_flag |= NEXT_ALL_FLAGS;
+      } else if (!strncmp (p, "vpath", 5)) {
+	p += 5;
+	next_flag |= NEXT_VPATH_FLAG;
+      } else if (!strncmp (p, "quiet", 5)) {
+	p += 5;
+	next_flag |= NEXT_QUIET_FLAG;
+      } else if (!strncmp (p, "makefiles", 9)) {
+	p += 9;
+	next_flag |= NEXT_MAKEFILES_FLAG;
+      } else if (!strncmp (p, "errexit", 7)) {
+	p += 7;
+	next_flag |= NEXT_ERREXIT_FLAG;
+      }
+    }
+  }
+}
+#endif /* __APPLE__ || NeXT || NeXT_PDO */
+
 void
 init_hash_global_variable_set (void)
 {
@@ -168,6 +202,10 @@
   struct variable **var_slot;
   struct variable var_key;
 
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+  check_apple_pb_support (name, length, value);
+#endif /* __APPLE__ || NeXT || NeXT_PDO */
+
   if (set == NULL)
     set = &global_variable_set;
 
diff -ruN make-3.81/variable.h make-3.81-patched/variable.h
--- make-3.81/variable.h	2006-02-14 16:42:18.000000000 +0100
+++ make-3.81-patched/variable.h	2007-07-24 22:28:38.000000000 +0200
@@ -110,6 +110,13 @@
 
 /* expand.c */
 extern char *variable_buffer_output PARAMS ((char *ptr, char *string, unsigned int length));
+
+#if defined(__APPLE__) || defined(NeXT) || defined(NeXT_PDO)
+extern char *initialize_variable_output PARAMS ((void));
+extern char *save_variable_output PARAMS ((int *savelenp));
+extern void restore_variable_output PARAMS ((char *save, int savelen));
+#endif /* __APPLE__ || NeXT || NeXT_PDO */
+
 extern char *variable_expand PARAMS ((char *line));
 extern char *variable_expand_for_file PARAMS ((char *line, struct file *file));
 extern char *allocated_variable_expand_for_file PARAMS ((char *line, struct file *file));
