--- ppl-0.11.2-orig/src/mp_std_bits.defs.hh	2012-12-23 12:04:41.000000000 -0800
+++ ppl-0.11.2/src/mp_std_bits.defs.hh	2012-12-23 12:06:11.000000000 -0800
@@ -3683,6 +3683,7 @@
 #ifdef PPL_DOXYGEN_INCLUDE_IMPLEMENTATION_DETAILS
 //! Specialization of std::numeric_limits.
 #endif // defined(PPL_DOXYGEN_INCLUDE_IMPLEMENTATION_DETAILS)
+#if (__GNU_MP_VERSION < 5) || (__GNU_MP_VERSION == 5 && __GNU_MP_VERSION_MINOR < 1)
 template <>
 class numeric_limits<mpz_class> {
 private:
@@ -3802,6 +3803,7 @@
   }
 };
 
+#endif
 } // namespace std
 
 #include "mp_std_bits.inlines.hh"
--- ppl-0.11.2-orig/src/ppl.hh.dist	2012-12-23 12:04:41.000000000 -0800
+++ ppl-0.11.2/src/ppl.hh.dist	2012-12-23 12:06:11.000000000 -0800
@@ -3683,6 +3683,7 @@
 #ifdef PPL_DOXYGEN_INCLUDE_IMPLEMENTATION_DETAILS
 //! Specialization of std::numeric_limits.
 #endif // defined(PPL_DOXYGEN_INCLUDE_IMPLEMENTATION_DETAILS)
+#if (__GNU_MP_VERSION < 5) || (__GNU_MP_VERSION == 5 && __GNU_MP_VERSION_MINOR < 1)
 template <>
 class numeric_limits<mpz_class> {
 private:
@@ -3802,6 +3803,7 @@
   }
 };
 
+#endif
 } // namespace std
 
 /* Automatically generated from PPL source file ../src/mp_std_bits.inlines.hh line 1. */
--- ppl-0.11.2-orig/src/mp_std_bits.cc	2012-12-23 12:59:55.000000000 -0800
+++ ppl-0.11.2/src/mp_std_bits.cc	2012-12-23 13:00:13.000000000 -0800
@@ -46,7 +46,9 @@
 const bool std::numeric_limits<mpz_class>::is_bounded;
 const bool std::numeric_limits<mpz_class>::is_modulo;
 const bool std::numeric_limits<mpz_class>::traps;
+#if (__GNU_MP_VERSION < 5) || (__GNU_MP_VERSION == 5 && __GNU_MP_VERSION_MINOR < 1)
 const bool std::numeric_limits<mpz_class>::tininess_before;
+#endif
 const std::float_round_style std::numeric_limits<mpz_class>::round_style;
 
 const bool std::numeric_limits<mpq_class>::is_specialized;
@@ -69,5 +71,7 @@
 const bool std::numeric_limits<mpq_class>::is_bounded;
 const bool std::numeric_limits<mpq_class>::is_modulo;
 const bool std::numeric_limits<mpq_class>::traps;
+#if (__GNU_MP_VERSION < 5) || (__GNU_MP_VERSION == 5 && __GNU_MP_VERSION_MINOR < 1)
 const bool std::numeric_limits<mpq_class>::tininess_before;
+#endif
 const std::float_round_style std::numeric_limits<mpq_class>::round_style;
