diff -ru smake-1.2.1.orig/include/schily/mconfig.h smake-1.2.1/include/schily/mconfig.h
--- smake-1.2.1.orig/include/schily/mconfig.h	2009-10-17 14:52:18.000000000 -0400
+++ smake-1.2.1/include/schily/mconfig.h	2011-08-22 19:37:56.000000000 -0400
@@ -150,6 +150,11 @@
 #endif
 #endif
 
+/* Build hangs when using LLVM unless we disable SCANSTACK */
+#ifdef __llvm__
+#	undef	HAVE_SCANSTACK
+#endif
+
 /*
  * Allow to overwrite the defines in the makefiles by calling
  *
