--- valgrind-3.7.0/coregrind/m_main.c.orig	2012-03-05 17:47:39.000000000 -0500
+++ valgrind-3.7.0/coregrind/m_main.c	2012-03-05 17:53:43.000000000 -0500
@@ -2830,6 +2830,7 @@
     "\tandl  $~15, %eax\n"
     /* install it, and collect the original one */
     "\txchgl %eax, %esp\n"
+    "\tsubl  $12, %esp\n"
     /* call _start_in_C_darwin, passing it the startup %esp */
     "\tpushl %eax\n"
     "\tcall  __start_in_C_darwin\n"
