diff -Naur eog-0.6.old/Makefile.in eog-0.6.new/Makefile.in
--- eog-0.6.old/Makefile.in	Thu Feb 15 11:08:40 2001
+++ eog-0.6.new/Makefile.in	Thu Aug  2 03:28:20 2001
@@ -500,7 +500,7 @@
 @XML_I18N_MERGE_DESKTOP_RULE@
 
 install-data-local:
-	GCONF_CONFIG_SOURCE=$(GCONF_CONFIG_SOURCE) gconftool --makefile-install-rule $(top_srcdir)/$(schema_DATA)
+	GCONF_CONFIG_SOURCE=$(GCONF_CONFIG_SOURCE) gconftool $(schemadir)/$(schema_DATA)
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.

