diff -Naur GConf-1.2.1.old/configure GConf-1.2.1.new/configure
--- GConf-1.2.1.old/configure	Mon Aug  5 03:18:52 2002
+++ GConf-1.2.1.new/configure	Mon Sep 16 20:09:51 2002
@@ -5567,7 +5567,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test x$module = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $tmp_verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -Naur GConf-1.2.1.old/doc/gconf/Makefile.in GConf-1.2.1.new/doc/gconf/Makefile.in
--- GConf-1.2.1.old/doc/gconf/Makefile.in	Mon Aug  5 03:20:47 2002
+++ GConf-1.2.1.new/doc/gconf/Makefile.in	Mon Sep 16 20:09:51 2002
@@ -170,7 +170,7 @@
 
 # CFLAGS and LDFLAGS for compiling scan program. Only needed
 # if $(DOC_MODULE).types is non-empty.
-GTKDOC_CFLAGS = -I$(top_srcdir) $(CFLAGS) $(DEPENDENT_CFLAGS)
+GTKDOC_CFLAGS = -I$(top_srcdir) $(CFLAGS) $(CPPFLAGS) $(DEPENDENT_CFLAGS)
 
 GTKDOC_LIBS = $(LDFLAGS) $(DEPENDENT_LIBS) $(top_builddir)/gconf/libgconf-$(MAJOR_VERSION).la
 
diff -Naur GConf-1.2.1.old/gconf/Makefile.in GConf-1.2.1.new/gconf/Makefile.in
--- GConf-1.2.1.old/gconf/Makefile.in	Mon Aug  5 03:20:33 2002
+++ GConf-1.2.1.new/gconf/Makefile.in	Mon Sep 16 20:15:43 2002
@@ -162,7 +162,7 @@
 bin_PROGRAMS = gconftool-2 
 libexec_PROGRAMS = gconfd-2 $(SANITY_CHECK)
 
-CORBA_SOURCECODE = GConf-common.c GConf-skels.c GConf-stubs.c GConf.h
+CORBA_SOURCECODE = GConfI-common.c GConfI-skels.c GConfI-stubs.c GConfI.h
 
 built_sourcecode = \
 	$(CORBA_SOURCECODE)		\
@@ -203,24 +203,24 @@
 	gconfd.c
 
 
-gconfd_2_LDADD = $(EFENCE) $(INTLLIBS) $(DEPENDENT_LIBS) libgconf-$(MAJOR_VERSION).la
+gconfd_2_LDADD = libgconf-$(MAJOR_VERSION).la $(EFENCE) $(INTLLIBS) $(DEPENDENT_LIBS)
 
 # gconf_testclient_SOURCES = \
 # 	testclient.c
 
-# gconf_testclient_LDADD = $(GLIB_LIBS) $(OAF_LIBS) libgconf-client.la
+# gconf_testclient_LDADD = libgconf-client.la $(GLIB_LIBS) $(OAF_LIBS)
 
 gconftool_2_SOURCES = \
 	gconftool.c
 
 
-gconftool_2_LDADD = $(EFENCE) $(INTLLIBS) $(DEPENDENT_WITH_XML_LIBS) $(POPT_LIBS) libgconf-$(MAJOR_VERSION).la
+gconftool_2_LDADD = libgconf-$(MAJOR_VERSION).la $(EFENCE) $(INTLLIBS) $(DEPENDENT_WITH_XML_LIBS) $(POPT_LIBS)
 
 gconf_sanity_check_2_SOURCES = \
 	gconf-sanity-check.c
 
 
-gconf_sanity_check_2_LDADD = $(EFENCE) $(INTLLIBS) $(DEPENDENT_WITH_XML_AND_GTK_LIBS) $(POPT_LIBS) libgconf-$(MAJOR_VERSION).la
+gconf_sanity_check_2_LDADD = libgconf-$(MAJOR_VERSION).la $(EFENCE) $(INTLLIBS) $(DEPENDENT_WITH_XML_AND_GTK_LIBS) $(POPT_LIBS)
 
 libgconf_2_la_SOURCES = \
 	gconf-internals.c	\
@@ -245,7 +245,7 @@
 
 libgconf_2_la_LIBADD = $(INTLLIBS) $(DEPENDENT_LIBS)
 
-EXTRA_DIST = GConf.idl default.path.in gconfmarshal.list regenerate-enum-header.sh regenerate-enum-footer.sh
+EXTRA_DIST = GConfI.idl default.path.in gconfmarshal.list regenerate-enum-header.sh regenerate-enum-footer.sh
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../config.h
 CONFIG_CLEAN_FILES =  default.path
@@ -258,8 +258,8 @@
 libgconf_2_la_OBJECTS =  gconf-internals.lo gconf-backend.lo \
 gconf-changeset.lo gconf-error.lo gconf-listeners.lo gconf-locale.lo \
 gconf-schema.lo gconf-sources.lo gconf-value.lo gconf.lo \
-gconf-client.lo gconf-enum-types.lo GConf-common.lo GConf-skels.lo \
-GConf-stubs.lo
+gconf-client.lo gconf-enum-types.lo GConfI-common.lo GConfI-skels.lo \
+GConfI-stubs.lo
 bin_PROGRAMS =  gconftool-2$(EXEEXT)
 @GTK_FALSE@libexec_PROGRAMS =  gconfd-2$(EXEEXT)
 @GTK_TRUE@libexec_PROGRAMS =  gconfd-2$(EXEEXT) \
@@ -595,8 +595,8 @@
 dist-hook:
 	cd $(distdir) ; rm -f $(CLEANFILES)
 
-$(CORBA_SOURCECODE): $(srcdir)/GConf.idl $(ORBIT_IDL)
-	$(ORBIT_IDL) $(srcdir)/GConf.idl
+$(CORBA_SOURCECODE): $(srcdir)/GConfI.idl $(ORBIT_IDL)
+	$(ORBIT_IDL) $(srcdir)/GConfI.idl
 
 install-data-local:
 	-mkdir -p $(DESTDIR)$(sysconfdir)/gconf/$(MAJOR_VERSION)
diff -Naur GConf-1.2.1.old/gconf/gconf-database.h GConf-1.2.1.new/gconf/gconf-database.h
--- GConf-1.2.1.old/gconf/gconf-database.h	Sat Feb  2 05:22:48 2002
+++ GConf-1.2.1.new/gconf/gconf-database.h	Mon Sep 16 20:09:51 2002
@@ -27,7 +27,7 @@
 #endif /* __cplusplus */
 
 #include "gconf-error.h"
-#include "GConf.h"
+#include "GConfI.h"
 #include "gconf-listeners.h"
 #include "gconf-sources.h"
 #include "gconf-internals.h"
diff -Naur GConf-1.2.1.old/gconf/gconf-internals.c GConf-1.2.1.new/gconf/gconf-internals.c
--- GConf-1.2.1.old/gconf/gconf-internals.c	Wed Jul 24 02:14:07 2002
+++ GConf-1.2.1.new/gconf/gconf-internals.c	Mon Sep 16 20:09:51 2002
@@ -2573,6 +2573,9 @@
         }
     }
 
+  /* needed on Darwin */
+  fsync (lock->lock_fd);
+
   return lock;
 }
 
diff -Naur GConf-1.2.1.old/gconf/gconf-internals.h GConf-1.2.1.new/gconf/gconf-internals.h
--- GConf-1.2.1.old/gconf/gconf-internals.h	Thu Jun 13 05:32:21 2002
+++ GConf-1.2.1.new/gconf/gconf-internals.h	Mon Sep 16 20:09:51 2002
@@ -37,7 +37,7 @@
 #include "gconf-value.h"
 #include "gconf-engine.h"
 #include "gconf-sources.h"
-#include "GConf.h"
+#include "GConfI.h"
 
 gchar*       gconf_key_directory  (const gchar* key);
 const gchar* gconf_key_key        (const gchar* key);
diff -Naur GConf-1.2.1.old/gconf/gconf.c GConf-1.2.1.new/gconf/gconf.c
--- GConf-1.2.1.old/gconf/gconf.c	Thu Jun 13 05:32:23 2002
+++ GConf-1.2.1.new/gconf/gconf.c	Mon Sep 16 20:09:51 2002
@@ -18,7 +18,7 @@
  */
 
 #include <popt.h>
-#include "GConf.h"
+#include "GConfI.h"
 #include "gconf.h"
 #include "gconf-internals.h"
 #include "gconf-sources.h"
diff -Naur GConf-1.2.1.old/gconf/gconfd.c GConf-1.2.1.new/gconf/gconfd.c
--- GConf-1.2.1.old/gconf/gconfd.c	Wed Jul 24 02:14:08 2002
+++ GConf-1.2.1.new/gconf/gconfd.c	Mon Sep 16 20:09:51 2002
@@ -37,7 +37,7 @@
 #include "gconf-database.h"
 #include <orbit/orbit.h>
 
-#include "GConf.h"
+#include "GConfI.h"
 
 #include <sys/types.h>
 #include <stdio.h>
diff -Naur GConf-1.2.1.old/gconf/gconfd.h GConf-1.2.1.new/gconf/gconfd.h
--- GConf-1.2.1.old/gconf/gconfd.h	Sat Feb  2 05:22:49 2002
+++ GConf-1.2.1.new/gconf/gconfd.h	Mon Sep 16 20:09:51 2002
@@ -27,7 +27,7 @@
 #endif /* __cplusplus */
 
 #include "gconf-error.h"
-#include "GConf.h"
+#include "GConfI.h"
 #include "gconf-database.h"
 
 PortableServer_POA gconf_get_poa ();
diff -Naur GConf-1.2.1.old/ltmain.sh GConf-1.2.1.new/ltmain.sh
--- GConf-1.2.1.old/ltmain.sh	Mon Aug  5 03:18:41 2002
+++ GConf-1.2.1.new/ltmain.sh	Mon Sep 16 20:09:51 2002
@@ -2952,6 +2952,11 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+	  else
+	    tmp_verstring="$verstring"
+	  fi
 	  eval cmds=\"$archive_cmds\"
 	fi
 	save_ifs="$IFS"; IFS='~'
