diff -Naur gnome-applets-2.0.4.old/configure gnome-applets-2.0.4.new/configure
--- gnome-applets-2.0.4.old/configure	Tue Nov 19 08:04:37 2002
+++ gnome-applets-2.0.4.new/configure	Thu Jan 23 18:06:36 2003
@@ -8889,6 +8889,9 @@
   *solaris*)
 	build_battstat_applet=no
 	;;
+  *darwin*)
+	build_battstat_applet=no
+	;;
 esac
 
 
diff -Naur gnome-applets-2.0.4.old/gkb-new/preset.c gnome-applets-2.0.4.new/gkb-new/preset.c
--- gnome-applets-2.0.4.old/gkb-new/preset.c	Tue May 14 20:13:47 2002
+++ gnome-applets-2.0.4.new/gkb-new/preset.c	Thu Jan 23 18:06:36 2003
@@ -29,6 +29,7 @@
  * USA
  */
 
+#include <sys/types.h>
 #include <dirent.h>
 #include "gkb.h"
 
diff -Naur gnome-applets-2.0.4.old/mini-commander/src/mini-commander_applet.c gnome-applets-2.0.4.new/mini-commander/src/mini-commander_applet.c
--- gnome-applets-2.0.4.old/mini-commander/src/mini-commander_applet.c	Tue Jul 23 11:19:23 2002
+++ gnome-applets-2.0.4.new/mini-commander/src/mini-commander_applet.c	Thu Jan 23 18:06:49 2003
@@ -407,13 +407,12 @@
 {
     MCData *mcdata;
 
-    mcdata->history_position = LENGTH_HISTORY_LIST;
-            
     gnome_window_icon_set_default_from_file (GNOME_ICONDIR"/gnome-mini-commander.png");
     
     panel_applet_add_preferences (applet, "/schemas/apps/mini-commander/prefs", NULL);
     
     mcdata = g_new0 (MCData, 1);
+    mcdata->history_position = LENGTH_HISTORY_LIST;
     mcdata->applet = applet;
   
     g_signal_connect(GTK_OBJECT(applet),
diff -Naur gnome-applets-2.0.4.old/multiload/Makefile.in gnome-applets-2.0.4.new/multiload/Makefile.in
--- gnome-applets-2.0.4.old/multiload/Makefile.in	Tue Nov 19 08:34:37 2002
+++ gnome-applets-2.0.4.new/multiload/Makefile.in	Thu Jan 23 18:06:36 2003
@@ -146,7 +146,7 @@
 multiload_applet_2_SOURCES =  			global.h				linux-proc.h				load-graph.h				cpuload.c				linux-proc.c				load-graph.c				loadavg.c				main.c				memload.c				netload.c				properties.c				swapload.c
 
 
-multiload_applet_2_LDADD =  				$(GNOME_APPLETS_LIBS) 					$(GTOP_APPLETS_LIBS)
+multiload_applet_2_LDADD =  				$(GNOME_APPLETS_LIBS) 					$(GTOP_APPLETS_LIBS)	$(X_LIBS)
 
 
 pixmaps = gnome-cpu.png
diff -Naur gnome-applets-2.0.4.old/omf-install/Makefile.in gnome-applets-2.0.4.new/omf-install/Makefile.in
--- gnome-applets-2.0.4.old/omf-install/Makefile.in	Tue Nov 19 08:34:57 2002
+++ gnome-applets-2.0.4.new/omf-install/Makefile.in	Thu Jan 23 18:06:36 2003
@@ -261,7 +261,7 @@
 	-for file in $(srcdir)/*.omf; do \
 		$(INSTALL_DATA) $(srcdir)/$$file $(DESTDIR)$(omf_dest_dir); \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) 
+#	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) 
 
 uninstall-local:
 	-for file in $(srcdir)/*.omf; do \
