diff -Naur gnome-utils-1.4.1.2.old/Makefile.in gnome-utils-1.4.1.2.new/Makefile.in
--- gnome-utils-1.4.1.2.old/Makefile.in	Fri Feb  1 13:43:50 2002
+++ gnome-utils-1.4.1.2.new/Makefile.in	Sun Mar  3 00:05:21 2002
@@ -206,7 +206,7 @@
 
 @BUILD_GFLOPPY_TRUE@gfloppy = gfloppy
 
-SUBDIRS = po intl macros idl $(gfloppy) logview gcharmap 	gcalc gsearchtool $(gdiskfree) $(stripchart) gdictsrc 	$(guile_DIRS) mini-utils splash gdialog 	meat-grinder exe-handler omf-install
+SUBDIRS = po intl macros idl $(gfloppy) logview gcharmap 	gcalc gsearchtool $(gdiskfree) $(stripchart) gdictsrc 	$(guile_DIRS) mini-utils splash meat-grinder exe-handler omf-install
 
 
 DIST_SUBDIRS = po intl macros idl gfloppy logview gcharmap 	gcalc gsearchtool gdiskfree stripchart gdictsrc 	gtt cromagnon mini-utils splash gdialog 	meat-grinder exe-handler omf-install
diff -Naur gnome-utils-1.4.1.2.old/gtt/util.c gnome-utils-1.4.1.2.new/gtt/util.c
--- gnome-utils-1.4.1.2.old/gtt/util.c	Sat Oct  6 03:26:13 2001
+++ gnome-utils-1.4.1.2.new/gtt/util.c	Sun Mar  3 00:05:21 2002
@@ -32,7 +32,9 @@
 #include <gnome.h>
 
 /* hack alert --xxx fixme -- we need to configure.in for have_langinfo */
+#if !defined(__APPLE__)
 #define HAVE_LANGINFO_D_FMT
+#endif
 #ifdef HAVE_LANGINFO_D_FMT
 #include <langinfo.h>
 #endif
diff -Naur gnome-utils-1.4.1.2.old/logview/log_repaint.c gnome-utils-1.4.1.2.new/logview/log_repaint.c
--- gnome-utils-1.4.1.2.old/logview/log_repaint.c	Fri Aug 24 14:19:38 2001
+++ gnome-utils-1.4.1.2.new/logview/log_repaint.c	Sun Mar  3 00:05:21 2002
@@ -24,7 +24,9 @@
 #include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>
+#endif
 #include <string.h>
 #include <gnome.h>
 #include "logview.h"
diff -Naur gnome-utils-1.4.1.2.old/logview/logrtns.c gnome-utils-1.4.1.2.new/logview/logrtns.c
--- gnome-utils-1.4.1.2.old/logview/logrtns.c	Fri Aug 24 14:19:38 2001
+++ gnome-utils-1.4.1.2.new/logview/logrtns.c	Sun Mar  3 00:05:21 2002
@@ -821,7 +821,7 @@
       if (IsLeapYear (curmark->year - lastyear + thisyear))
 	 curmark->time += 24 * 60 * 60;		/*  Add one day */
 
-#if defined(__NetBSD__) || defined(__FreeBSD__)
+#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__APPLE__)
       curmark->time += correction - tmptm->tm_gmtoff;
 #else
       curmark->time += correction - timezone;
@@ -847,7 +847,7 @@
    if (IsLeapYear (thisyear))
       log->lstats.enddate += 24 * 60 * 60;	/*  Add one day */
 
-#if defined(__NetBSD__) || defined(__FreeBSD__)
+#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__APPLE__)
    log->lstats.enddate += correction - tmptm->tm_gmtoff;
 #else
    log->lstats.enddate += correction - timezone;
@@ -858,7 +858,7 @@
    if (IsLeapYear (thisyear - lastyear))
       log->lstats.startdate += 24 * 60 * 60;	/*  Add one day */
 
-#if defined(__NetBSD__) || defined(__FreeBSD__)
+#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__APPLE__)
    log->lstats.startdate += correction - tmptm->tm_gmtoff;
 #else
    log->lstats.startdate += correction - timezone;
diff -Naur gnome-utils-1.4.1.2.old/logview/logrtns.h gnome-utils-1.4.1.2.new/logview/logrtns.h
--- gnome-utils-1.4.1.2.old/logview/logrtns.h	Mon Jun 25 07:44:47 2001
+++ gnome-utils-1.4.1.2.new/logview/logrtns.h	Sun Mar  3 00:05:21 2002
@@ -24,7 +24,9 @@
 #include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>
+#endif
 #include <string.h>
 #include "logview.h"
 
diff -Naur gnome-utils-1.4.1.2.old/po/Makefile.in.in gnome-utils-1.4.1.2.new/po/Makefile.in.in
--- gnome-utils-1.4.1.2.old/po/Makefile.in.in	Fri Feb  1 09:55:27 2002
+++ gnome-utils-1.4.1.2.new/po/Makefile.in.in	Sun Mar  3 00:05:21 2002
@@ -58,6 +58,7 @@
 POTFILES = \
 
 CATALOGS = @CATALOGS@
+CATOBJEXT = @CATOBJEXT@
 
 .SUFFIXES:
 .SUFFIXES: .c .o .po .pox .gmo .mo
@@ -106,7 +107,7 @@
 	@catalogs='$(CATALOGS)'; \
 	for cat in $$catalogs; do \
 	  cat=`basename $$cat`; \
-	  lang=`echo $$cat | sed 's/\.gmo$$//'`; \
+	  lang=`echo $$cat | sed 's/\$(CATOBJEXT)$$//'`; \
 	  dir=$(localedir)/$$lang/LC_MESSAGES; \
 	  $(mkinstalldirs) $(DESTDIR)$$dir; \
 	  if test -r $$cat; then \
