diff -u -d -b -w -u -r1.7 ltmain.sh
--- foo/ltmain.sh	29 Aug 2002 04:12:30 -0000	1.7
+++ foo/ltmain.sh	13 May 2003 12:33:35 -0000
@@ -2370,7 +2370,11 @@
 		  ;;
 		esac
 		if grep "^installed=no" $deplib > /dev/null; then
-		  path="-L$absdir/$objdir"
+		  if test -d "$absdir/$objdir"; then
+			path="$absdir/$objdir"
+		  else
+			path="$absdir"
+		  fi
 		else
 		  eval libdir=`sed -n -e 's/^libdir=\(.*\)$/\1/p' $deplib`
 		  if test -z "$libdir"; then
@@ -2380,11 +2384,56 @@
 		  if test "$absdir" != "$libdir"; then
 		    $echo "$modename: warning: \`$deplib' seems to be moved" 1>&2
 		  fi
-		  path="-L$absdir"
+		  path="$absdir"
+		fi
+		depdepl=
+		case $host in
+		*-*-darwin*)
+		  # we do not want to link against static libs, but need to link against shared
+		  eval deplibrary_names=`sed -n -e 's/^library_names=\(.*\)$/\1/p' $deplib`
+		  if test -n "$deplibrary_names" ; then
+		    for tmp in $deplibrary_names ; do
+		      depdepl=$tmp
+		    done
+		    if test -f "$path/$depdepl" ; then
+		      depdepl="$path/$depdepl"
+		   fi
+
+			case " $newlib_search_path " in
+			*" $path "*) ;;
+		    *) newlib_search_path="$newlib_search_path $path" ;;
+			esac
+		    path=""
 		fi
 		;;
+		*)
+		path="-L$path"
+		;;
+		esac 
+		
+		;;
+		  -l*)
+		case $host in
+		*-*-darwin*)
+		 # Again, we only want to link against shared libraries
+		 eval tmp_libs=`$echo "X$deplib" | $Xsed -e "s,^\-l,,"`
+		 for tmp in $newlib_search_path ; do
+		     if test -f "$tmp/lib$tmp_libs.dylib" ; then
+		       eval depdepl="$tmp/lib$tmp_libs.dylib"
+		       break
+		     fi  
+         done
+         path=""
+		  ;;
 	      *) continue ;;
 	      esac
+		;;
+	      *) continue ;;
+	      esac
+	      case " $deplibs " in
+	      *" $depdepl "*) ;;
+	      *) deplibs="$deplibs $depdepl" ;;
+	      esac	      
 	      case " $deplibs " in
 	      *" $path "*) ;;
 	      *) deplibs="$deplibs $path" ;;
--- gnucash-1.8.3/src/scm/main.scm.old	Thu May 15 00:28:39 2003
+++ gnucash-1.8.3/src/scm/main.scm	Thu May 15 00:48:28 2003
@@ -316,8 +316,8 @@
   (define (locale-prefixes)
     ;; Mac OS X. 10.1 and earlier don't have LC_MESSAGES. Fall back to
     ;; LC_ALL for those systems.
-    (let* ((locale (or (false-if-exception (setlocale LC_MESSAGES))
-		       (setlocale LC_ALL)))
+    (let* ((locale (or (false-if-exception (getenv "LANG"))
+		       (getenv "LANG")))
            (strings (cond ((not (string? locale)) '())
                           ((equal? locale "C") '())
                           ((<= (string-length locale) 4) (list locale))
