diff -Naur gtk+-2.0.9.old/configure gtk+-2.0.9.new/configure
--- gtk+-2.0.9.old/configure	Thu Nov 21 05:15:08 2002
+++ gtk+-2.0.9.new/configure	Sat Jan 18 03:54:38 2003
@@ -14241,7 +14241,7 @@
 
     if $have_freetype ; then
       gtk_save_cppflags="$CPPFLAGS"
-      CPPFLAGS="$CPPFLAGS $FREETYPE_CFLAGS"
+      CPPFLAGS="$FREETYPE_CFLAGS $CPPFLAGS"
 
       echo "$as_me:$LINENO: checking For sufficiently new FreeType (at least 2.0.1)" >&5
 echo $ECHO_N "checking For sufficiently new FreeType (at least 2.0.1)... $ECHO_C" >&6
diff -Naur gtk+-2.0.9.old/demos/gtk-demo/main.c gtk+-2.0.9.new/demos/gtk-demo/main.c
--- gtk+-2.0.9.old/demos/gtk-demo/main.c	Fri May 17 04:11:57 2002
+++ gtk+-2.0.9.new/demos/gtk-demo/main.c	Sat Jan 18 03:54:38 2003
@@ -96,7 +96,7 @@
     {
       int c;
       
-#ifndef G_OS_WIN32
+#ifdef HAVE_FLOCKFILE
       c = getc_unlocked (stream);
 #else
       c = getc (stream);
diff -Naur gtk+-2.0.9.old/gdk/x11/gdkdrawable-x11.c gtk+-2.0.9.new/gdk/x11/gdkdrawable-x11.c
--- gtk+-2.0.9.old/gdk/x11/gdkdrawable-x11.c	Fri Nov  8 07:39:56 2002
+++ gtk+-2.0.9.new/gdk/x11/gdkdrawable-x11.c	Sat Jan 18 03:54:38 2003
@@ -853,6 +853,7 @@
     }
   else
     {
+      /* RGBA */
       pf.direct.alpha = 0;
     }
   
@@ -869,10 +870,18 @@
    * RENDER's preferred order
    */
   pf.direct.alphaMask = 0;
-  /* ARGB */
-  pf.direct.red = 16;
-  pf.direct.green = 8;
-  pf.direct.blue = 0;
+  if (ImageByteOrder (xdisplay) == LSBFirst)
+    {
+      /* ARGB */
+      pf.direct.red = 16;
+      pf.direct.green = 8;
+      pf.direct.blue = 0;
+    } else {
+      /* BGRA */
+      pf.direct.red = 8;
+      pf.direct.green = 16;
+      pf.direct.blue = 24;
+    }
   
   *format = XRenderFindFormat (xdisplay,
 			       (PictFormatType | PictFormatDepth |
@@ -884,7 +893,14 @@
 			       0);
 
   pf.direct.alphaMask = 0xff;
-  pf.direct.alpha = 24;
+  if (ImageByteOrder (xdisplay) == LSBFirst)
+    {
+      /* ARGB */
+      pf.direct.alpha = 24;
+    } else {
+      /* BGRA */
+      pf.direct.alpha = 0;
+    }
   
   *mask = XRenderFindFormat (xdisplay,
 			     (PictFormatType | PictFormatDepth |
@@ -901,11 +917,19 @@
 
   pf.type = PictTypeDirect;
   pf.depth = 32;
-  pf.direct.red = 16;
-  pf.direct.green = 8;
-  pf.direct.blue = 0;
+  if (ImageByteOrder (xdisplay) == LSBFirst)
+    {
+      pf.direct.red = 16;
+      pf.direct.green = 8;
+      pf.direct.blue = 0;
+      pf.direct.alpha = 24;
+    } else {
+      pf.direct.red = 8;
+      pf.direct.green = 16;
+      pf.direct.blue = 24;
+      pf.direct.alpha = 0;
+    }
   pf.direct.alphaMask = 0xff;
-  pf.direct.alpha = 24;
 
   *format = XRenderFindFormat (xdisplay,
 			       (PictFormatType | PictFormatDepth |
diff -Naur gtk+-2.0.9.old/ltmain.sh gtk+-2.0.9.new/ltmain.sh
--- gtk+-2.0.9.old/ltmain.sh	Sat Nov  9 05:40:11 2002
+++ gtk+-2.0.9.new/ltmain.sh	Sun Jan 19 02:40:52 2003
@@ -4047,10 +4047,10 @@
 
 # Directory that this library needs to be installed in:
 libdir='$install_libdir'"
-	  if test "$installed" = no && test $need_relink = yes; then
-	    $echo >> $output "\
-relink_command=\"$relink_command\""
-	  fi
+#	  if test "$installed" = no && test $need_relink = yes; then
+#	    $echo >> $output "\
+#relink_command=\"$relink_command\""
+#	  fi
 	done
       fi
 
diff -Naur gtk+-2.0.9.old/modules/input/Makefile.in gtk+-2.0.9.new/modules/input/Makefile.in
--- gtk+-2.0.9.old/modules/input/Makefile.in	Thu Nov 21 05:43:02 2002
+++ gtk+-2.0.9.new/modules/input/Makefile.in	Sat Jan 18 04:02:23 2003
@@ -782,19 +782,12 @@
 # Running this if cross compiling or if DESTDIR is set is going to
 # not work at all, so skip it.
 install-data-local:
-	@if $(RUN_QUERY_IMMODULES_TEST) ; then \
-	  echo $(mkinstalldirs) $(DESTDIR)$(sysconfdir)/gtk-2.0 ; \
-	  $(mkinstalldirs) $(DESTDIR)$(sysconfdir)/gtk-2.0 ; \
-	  echo "$(top_builddir)/gtk/gtk-query-immodules-2.0 > $(DESTDIR)$(sysconfdir)/gtk-2.0/gtk.immodules" ; \
-	  $(top_builddir)/gtk/gtk-query-immodules-2.0 > $(DESTDIR)$(sysconfdir)/gtk-2.0/gtk.immodules ; \
-	else \
-	  echo "***" ; \
-	  echo "*** Warning: gtk.immodules not built" ; \
-	  echo "***" ; \
-	  echo "*** Generate this file manually on host" ; \
-	  echo "*** system using gtk-query-immodules-2.0" ; \
-	  echo "***" ; \
-        fi
+	$(mkinstalldirs) $(DESTDIR)$(sysconfdir)/gtk-2.0 ;	\
+	$(top_builddir)/gtk/gtk-query-immodules-2.0		\
+	  $(DESTDIR)$(moduledir)/*				\
+	  | sed -e 's|$(DESTDIR)||g'				\
+	  > $(DESTDIR)$(sysconfdir)/gtk-2.0/gtk.immodules	\
+	  2>&1
 
 gtk.immodules: Makefile.am $(module_LTLIBRARIES)
 	$(top_builddir)/gtk/gtk-query-immodules-2.0 *.la > gtk.immodules
