diff -Naur libgnomeprintui-1.116.0.old/configure libgnomeprintui-1.116.0.new/configure
--- libgnomeprintui-1.116.0.old/configure	Mon Jul 29 06:01:05 2002
+++ libgnomeprintui-1.116.0.new/configure	Mon Sep 16 23:13:59 2002
@@ -5146,7 +5146,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test x$module = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $tmp_verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
diff -Naur libgnomeprintui-1.116.0.old/libgnomeprintui/gpaui/Makefile.in libgnomeprintui-1.116.0.new/libgnomeprintui/gpaui/Makefile.in
--- libgnomeprintui-1.116.0.old/libgnomeprintui/gpaui/Makefile.in	Mon Jul 29 06:02:36 2002
+++ libgnomeprintui-1.116.0.new/libgnomeprintui/gpaui/Makefile.in	Mon Sep 16 23:13:59 2002
@@ -104,7 +104,7 @@
 INCLUDES =  	-I$(top_srcdir) 	-I$(top_srcdir)/intl 	$(LIBGNOMEPRINTUI_CFLAGS)
 
 
-CPPFLAGS =  	-DWE_ARE_LIBGNOMEPRINT_UI 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\" 	-DBINDIR=\""$(bindir)"\" 	-DDATADIR=\""$(datadir)"\" 
+CPPFLAGS =  	-DWE_ARE_LIBGNOMEPRINT_UI 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\" 	-DBINDIR=\""$(bindir)"\" 	-DDATADIR=\""$(datadir)"\" @CPPFLAGS@
 
 
 libgpaui_la_SOURCES =  	gpa-widget.c gpa-widget.h 	printer-selector.c printer-selector.h 	settings-selector.c settings-selector.h 	transport-selector.c transport-selector.h
diff -Naur libgnomeprintui-1.116.0.old/ltmain.sh libgnomeprintui-1.116.0.new/ltmain.sh
--- libgnomeprintui-1.116.0.old/ltmain.sh	Mon Jul 29 06:00:51 2002
+++ libgnomeprintui-1.116.0.new/ltmain.sh	Mon Sep 16 23:13:59 2002
@@ -2894,7 +2894,17 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+	  else
+	    tmp_verstring="$verstring"
+	  fi
+	  save_deplibs="$deplibs"
+	  for conv in $convenience; do
+	    deplibs="${deplibs%$conv*} ${deplibs#*$conv}"
+	  done     
 	  eval cmds=\"$archive_cmds\"
+	  deplibs="$save_deplibs"
 	fi
 	save_ifs="$IFS"; IFS='~'
 	for cmd in $cmds; do
