diff -Naur libgnomeprint-1.116.0.old/configure libgnomeprint-1.116.0.new/configure
--- libgnomeprint-1.116.0.old/configure	Mon Jul  8 15:12:23 2002
+++ libgnomeprint-1.116.0.new/configure	Wed Apr 16 03:19:26 2003
@@ -5454,7 +5454,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test x$module = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test .$module != .yes && echo -install_name $rpath/$soname $tmp_verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
@@ -9773,7 +9773,7 @@
   ENABLE_GTK_DOC_FALSE=
 fi
 
-LIBGNOMEPRINT_CFLAGS="$GP_CFLAGS $FREETYPE_CFLAGS"
+LIBGNOMEPRINT_CFLAGS="$FREETYPE_CFLAGS $GP_CFLAGS"
 LIBGNOMEPRINT_LIBS="$GP_LIBS $FREETYPE_LIBS"
 
 if test x"${CC}" = "xgcc"; then
diff -Naur libgnomeprint-1.116.0.old/installer/Makefile.in libgnomeprint-1.116.0.new/installer/Makefile.in
--- libgnomeprint-1.116.0.old/installer/Makefile.in	Mon Jul 29 05:03:31 2002
+++ libgnomeprint-1.116.0.new/installer/Makefile.in	Wed Apr 16 03:18:53 2003
@@ -118,7 +118,7 @@
 
 EXTRA_DIST = README.installer
 
-readmedir = $(datadir)/fonts
+readmedir = $(FONTMAPDIR_STATIC)
 readme_DATA = README.installer
 @INSTALL_FONTS_TRUE@INSTALL_COMMAND = ./libgnomeprint-2.0-font-install 		--debug 		--smart 		--refresh 		--static 		--aliases=$(FONTMAPDIR_STATIC)/adobe-urw.font 		--aliases=$(top_srcdir)/fonts/adobe-urw.font 		--target=$(FONTMAPDIR_STATIC)/gnome-print.fontmap
 @INSTALL_FONTS_FALSE@INSTALL_COMMAND = 
diff -Naur libgnomeprint-1.116.0.old/libgnomeprint/drivers/Makefile.in libgnomeprint-1.116.0.new/libgnomeprint/drivers/Makefile.in
--- libgnomeprint-1.116.0.old/libgnomeprint/drivers/Makefile.in	Mon Jul 29 05:03:27 2002
+++ libgnomeprint-1.116.0.new/libgnomeprint/drivers/Makefile.in	Wed Apr 16 03:18:53 2003
@@ -129,7 +129,7 @@
 
 noinst_LTLIBRARIES = 
 
-CPPFLAGS =  	-DWE_ARE_LIBGNOMEPRINT_UI 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\" 		-DBINDIR=\""$(bindir)"\"		-DDATADIR=\""$(datadir)"\" 
+CPPFLAGS =  	-DWE_ARE_LIBGNOMEPRINT_UI 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\" 		-DBINDIR=\""$(bindir)"\"		-DDATADIR=\""$(datadir)"\" @CPPFLAGS@
 
 
 INCLUDES =  	-I$(top_srcdir) 	$(OMNI_CFLAGS) 	$(LIBGNOMEPRINT_CFLAGS)
diff -Naur libgnomeprint-1.116.0.old/libgnomeprint/gpa/Makefile.in libgnomeprint-1.116.0.new/libgnomeprint/gpa/Makefile.in
--- libgnomeprint-1.116.0.old/libgnomeprint/gpa/Makefile.in	Mon Jul 29 05:03:26 2002
+++ libgnomeprint-1.116.0.new/libgnomeprint/gpa/Makefile.in	Wed Apr 16 03:18:53 2003
@@ -107,7 +107,7 @@
 
 noinst_LTLIBRARIES = libgpa.la
 
-CPPFLAGS =  	-DWE_ARE_LIBGNOMEPRINT_UI 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\" 	-DBINDIR=\""$(bindir)"\" 	-DDATADIR=\""$(datadir)"\" 
+CPPFLAGS =  	-DWE_ARE_LIBGNOMEPRINT_UI 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\" 	-DBINDIR=\""$(bindir)"\" 	-DDATADIR=\""$(datadir)"\" @CPPFLAGS@
 
 
 INCLUDES =  	-I$(top_srcdir) 	$(LIBGNOMEPRINT_CFLAGS)
diff -Naur libgnomeprint-1.116.0.old/libgnomeprint/transports/Makefile.in libgnomeprint-1.116.0.new/libgnomeprint/transports/Makefile.in
--- libgnomeprint-1.116.0.old/libgnomeprint/transports/Makefile.in	Mon Jul 29 05:03:28 2002
+++ libgnomeprint-1.116.0.new/libgnomeprint/transports/Makefile.in	Wed Apr 16 03:18:53 2003
@@ -108,7 +108,7 @@
 libexec_LTLIBRARIES =  	$(FILE_LIB) 	$(LPR_LIB) 	$(CUSTOM_LIB)
 
 
-CPPFLAGS =  	-DWE_ARE_LIBGNOMEPRINT_UI 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\" 		-DBINDIR=\""$(bindir)"\"		-DDATADIR=\""$(datadir)"\" 
+CPPFLAGS =  	-DWE_ARE_LIBGNOMEPRINT_UI 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\" 		-DBINDIR=\""$(bindir)"\"		-DDATADIR=\""$(datadir)"\" @CPPFLAGS@
 
 
 INCLUDES =  	-I$(top_srcdir) 	$(LIBGNOMEPRINT_CFLAGS)
diff -Naur libgnomeprint-1.116.0.old/ltmain.sh libgnomeprint-1.116.0.new/ltmain.sh
--- libgnomeprint-1.116.0.old/ltmain.sh	Mon Jul  8 15:12:06 2002
+++ libgnomeprint-1.116.0.new/ltmain.sh	Wed Apr 16 03:18:53 2003
@@ -2894,7 +2894,17 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
+	  if test "x$verstring" = "x0.0"; then
+	    tmp_verstring=
+	  else
+	    tmp_verstring="$verstring"
+	  fi
+	  save_deplibs="$deplibs"
+	  for conv in $convenience; do
+	    deplibs="${deplibs%$conv*} ${deplibs#*$conv}"
+	  done     
 	  eval cmds=\"$archive_cmds\"
+	  deplibs="$save_deplibs"
 	fi
 	save_ifs="$IFS"; IFS='~'
 	for cmd in $cmds; do
diff -Naur libgnomeprint-1.116.0.old/tests/Makefile.in libgnomeprint-1.116.0.new/tests/Makefile.in
--- libgnomeprint-1.116.0.old/tests/Makefile.in	Mon Jul 29 05:03:32 2002
+++ libgnomeprint-1.116.0.new/tests/Makefile.in	Wed Apr 16 03:18:53 2003
@@ -104,7 +104,7 @@
 VERSION = @VERSION@
 WARN_CFLAGS = @WARN_CFLAGS@
 cxxflags_set = @cxxflags_set@
-@USING_GCC_TRUE@CFLAGS = $(WARN_CFLAGS) -Werror
+@USING_GCC_TRUE@CFLAGS = $(WARN_CFLAGS)
 @USING_GCC_FALSE@CFLAGS = $(WARN_CFLAGS)
 
 noinst_PROGRAMS = generate
