diff -Naur librsvg-2.0.1.old/configure librsvg-2.0.1.new/configure
--- librsvg-2.0.1.old/configure	Tue Jul 23 03:20:16 2002
+++ librsvg-2.0.1.new/configure	Wed Apr 16 02:31:52 2003
@@ -7361,11 +7361,11 @@
             echo "$as_me:7361: checking LIBRSVG_CFLAGS" >&5
 echo $ECHO_N "checking LIBRSVG_CFLAGS... $ECHO_C" >&6
             LIBRSVG_CFLAGS=`$PKG_CONFIG --cflags "\
+	pangoft2 >= $PANGOFT2_REQUIRED \
 	gdk-pixbuf-2.0 >= $GDK_PIXBUF_REQUIRED \
 	glib-2.0 >= $GLIB_REQUIRED \
 	libart-2.0 >= $LIBART_REQUIRED \
-	libxml-2.0 >= $LIBXML_REQUIRED \
-	pangoft2 >= $PANGOFT2_REQUIRED"`
+	libxml-2.0 >= $LIBXML_REQUIRED"`
             echo "$as_me:7369: result: $LIBRSVG_CFLAGS" >&5
 echo "${ECHO_T}$LIBRSVG_CFLAGS" >&6
 
