diff -Naur nautilus-2.0.8.old/Makefile.in nautilus-2.0.8.new/Makefile.in
--- nautilus-2.0.8.old/Makefile.in	Tue Nov 19 07:58:14 2002
+++ nautilus-2.0.8.new/Makefile.in	Tue Jan 21 01:12:57 2003
@@ -150,7 +150,6 @@
 	libnautilus-private 	\
 	libnautilus-adapter	\
 	src 			\
-	test 			\
 	components		\
 	po 			\
 	data 			\
diff -Naur nautilus-2.0.8.old/components/Makefile.in nautilus-2.0.8.new/components/Makefile.in
--- nautilus-2.0.8.old/components/Makefile.in	Tue Nov 19 07:58:26 2002
+++ nautilus-2.0.8.new/components/Makefile.in	Tue Jan 21 01:12:57 2003
@@ -146,7 +146,6 @@
 	loser 					\
 	music					\
 	notes 					\
-	sample 					\
 	text 					\
 	throbber 				\
 	tree 					\
diff -Naur nautilus-2.0.8.old/configure nautilus-2.0.8.new/configure
--- nautilus-2.0.8.old/configure	Tue Nov 19 07:57:23 2002
+++ nautilus-2.0.8.new/configure	Tue Jan 21 02:29:35 2003
@@ -6281,7 +6281,7 @@
     # FIXME: Relying on posixy $() will cause problems for
     #        cross-compilation, but unfortunately the echo tests do not
     #        yet detect zsh echo's removal of \ escapes.
-    archive_cmds='$nonopt $(test "x$module" = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags -install_name $rpath/$soname $verstring'
+    archive_cmds='$nonopt $(test x$module = xyes && echo -bundle || echo -dynamiclib) $allow_undefined_flag -o $lib $libobjs $deplibs$linker_flags $(test x$module = xyes || echo -install_name $rpath/$soname $verstring)'
     # We need to add '_' to the symbols in $export_symbols first
     #archive_expsym_cmds="$archive_cmds"' && strip -s $export_symbols'
     hardcode_direct=yes
@@ -10964,7 +10964,8 @@
      INTLTOOL_PROP_RULE='%.prop:      %.prop.in      $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po) ; $(INTLTOOL_MERGE) $(top_srcdir)/po $< $@ -d -u -c $(top_builddir)/po/.intltool-merge-cache'
       INTLTOOL_OAF_RULE='%.oaf:       %.oaf.in       $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po) ; $(INTLTOOL_MERGE) $(top_srcdir)/po $< $@ -o -p'
      INTLTOOL_PONG_RULE='%.pong:      %.pong.in      $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po) ; $(INTLTOOL_MERGE) $(top_srcdir)/po $< $@ -x -u -c $(top_builddir)/po/.intltool-merge-cache'
-   INTLTOOL_SERVER_RULE='%.server:    %.server.in    $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po) ; $(INTLTOOL_MERGE) $(top_srcdir)/po $< $@ -o -u -c $(top_builddir)/po/.intltool-merge-cache'
+   INTLTOOL_SERVER_RULE='%.server:    %.server.in    $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po) ; $(INTLTOOL_MERGE) $(top_srcdir)/po $< $@ -o -u -c $(top_builddir)/po/.intltool-merge-cache\
+%.server.in: Makefile ;	sed -e "s|\@LIBDIR\@|$(libdir)|" $@ > $@.tmp; mv -f $@.tmp $@ '
     INTLTOOL_SHEET_RULE='%.sheet:     %.sheet.in     $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po) ; $(INTLTOOL_MERGE) $(top_srcdir)/po $< $@ -x -u -c $(top_builddir)/po/.intltool-merge-cache'
 INTLTOOL_SOUNDLIST_RULE='%.soundlist: %.soundlist.in $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po) ; $(INTLTOOL_MERGE) $(top_srcdir)/po $< $@ -d -u -c $(top_builddir)/po/.intltool-merge-cache'
        INTLTOOL_UI_RULE='%.ui:        %.ui.in        $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po) ; $(INTLTOOL_MERGE) $(top_srcdir)/po $< $@ -x -u -c $(top_builddir)/po/.intltool-merge-cache'
@@ -13384,7 +13385,7 @@
 
 LIBNAUTILUS_LIBS="`$PKG_CONFIG --libs $LIBNAUTILUS_MODULES`"
 
-LIBNAUTILUS_IDL_INCLUDES="`$PKG_CONFIG --variable=idldir $LIBNAUTILUS_MODULES | $srcdir/add-include-prefix`"
+LIBNAUTILUS_IDL_INCLUDES="`$PKG_CONFIG --variable=idldir $LIBNAUTILUS_MODULES | xargs -n1 echo -n ' -I '`"
 
 
 CORE_MODULES="eel-2.0 librsvg-2.0 bonobo-activation-2.0 libbonobo-2.0 libbonoboui-2.0 esound gnome-desktop-2.0 $EXTRA_CORE_MODULES"
@@ -13392,7 +13393,7 @@
 
 CORE_LIBS="`$PKG_CONFIG --libs $CORE_MODULES` $CDDA_LIBS $LIBJPEG $x_libs"
 
-CORE_IDL_INCLUDES="`$PKG_CONFIG --variable=idldir $CORE_MODULES | $srcdir/add-include-prefix`"
+CORE_IDL_INCLUDES="`$PKG_CONFIG --variable=idldir $CORE_MODULES | xargs -n1 echo -n ' -I '`"
 
 
 COMPONENT_MODULES="eel-2.0 libbonoboui-2.0 gnome-vfs-module-2.0 gnome-desktop-2.0"
diff -Naur nautilus-2.0.8.old/libnautilus-private/nautilus-medusa-support.c nautilus-2.0.8.new/libnautilus-private/nautilus-medusa-support.c
--- nautilus-2.0.8.old/libnautilus-private/nautilus-medusa-support.c	Mon Oct 29 05:21:46 2001
+++ nautilus-2.0.8.new/libnautilus-private/nautilus-medusa-support.c	Tue Jan 21 01:12:57 2003
@@ -30,6 +30,7 @@
 
 #include <eel/eel-glib-extensions.h>
 #include <eel/eel-string.h>
+#include <stdint.h>
 #include <dirent.h>
 #include <stdio.h>
 #include <sys/types.h>
diff -Naur nautilus-2.0.8.old/libnautilus-private/nautilus-volume-monitor.c nautilus-2.0.8.new/libnautilus-private/nautilus-volume-monitor.c
--- nautilus-2.0.8.old/libnautilus-private/nautilus-volume-monitor.c	Thu Aug 29 19:56:02 2002
+++ nautilus-2.0.8.new/libnautilus-private/nautilus-volume-monitor.c	Tue Jan 21 01:12:57 2003
@@ -95,6 +95,10 @@
 #include <sys/ucred.h>
 #endif
 
+#ifdef MACOSX
+#define HAVE_SYS_MOUNT_H
+#endif
+
 #ifdef HAVE_SYS_MOUNT_H
 #include <sys/mount.h>
 #endif
@@ -1019,6 +1023,30 @@
 static GList *
 get_mount_list (NautilusVolumeMonitor *monitor) 
 {
+#ifdef MACOSX
+	GList *volumes = NULL;
+	struct statfs *fsinfos = NULL;
+	int fsnum, fsidx;
+	
+	fsnum = getmntinfo(&fsinfos, MNT_WAIT);
+	if (fsnum == -1) {
+		g_warning ("Unable to stat filesystems.");
+		return NULL;
+	}
+	
+	for (fsidx = 0; fsidx < fsnum; ++fsidx) {
+		struct statfs afs = fsinfos[fsidx];
+		NautilusVolume *volume = NULL;
+		
+		volume = create_volume (afs.f_mntfromname, afs.f_mntonname);
+		if (afs.f_flags & MNT_RDONLY)
+			volume->is_read_only = TRUE;
+		volumes = finish_creating_volume_and_prepend
+			(monitor, volume, afs.f_fstypename, volumes);
+  	}
+  	
+  	return volumes;
+#else /* !MACOSX */
         GList *volumes;
         NautilusVolume *volume;
 	static time_t last_mtime = 0;
@@ -1117,6 +1145,7 @@
         saved_list = volumes;
 	
         return copy_mount_list (volumes);
+#endif /* !MACOSX */
 }
 
 #endif /* !SOLARIS_MNT */
diff -Naur nautilus-2.0.8.old/ltmain.sh nautilus-2.0.8.new/ltmain.sh
--- nautilus-2.0.8.old/ltmain.sh	Tue Nov 19 07:57:07 2002
+++ nautilus-2.0.8.new/ltmain.sh	Tue Jan 21 01:12:58 2003
@@ -1739,7 +1739,7 @@
 
 	  if test "$installed" = no; then
 	    notinst_deplibs="$notinst_deplibs $lib"
-	    need_relink=yes
+	    # need_relink=yes
 	  fi
 
 	  if test -n "$old_archive_from_expsyms_cmds"; then
@@ -2894,8 +2894,19 @@
 	if test -n "$export_symbols" && test -n "$archive_expsym_cmds"; then
 	  eval cmds=\"$archive_expsym_cmds\"
 	else
-	  eval cmds=\"$archive_cmds\"
-	fi
+      save_deplibs="$deplibs"
+      for conv in $convenience; do
+    tmp_deplibs=
+    for test_deplib in $deplibs; do
+      if test "$test_deplib" != "$conv"; then
+        tmp_deplibs="$tmp_deplibs $test_deplib"
+      fi
+    done
+    deplibs="$tmp_deplibs"
+      done
+      eval cmds=\"$archive_cmds\"
+      deplibs="$save_deplibs"
+    fi
 	save_ifs="$IFS"; IFS='~'
 	for cmd in $cmds; do
 	  IFS="$save_ifs"
