--- shermans_aquarium-2.1.1/src/Makefile.in	Wed Jul 10 06:14:11 2002
+++ shermans_aquarium-2.1.1-patched/src/Makefile.in	Fri Jul 26 13:17:00 2002
@@ -67,30 +67,30 @@
 install: @INSTALL_CHOISE@
 
 install-gnome1:
-	mkdir -p @GNOME_INSTALL_DIR@/bin
-	cp sherman @GNOME_INSTALL_DIR@/bin
-	mkdir -p $(DATADIR)/aquarium
-	cp -R ../aquarium/* $(DATADIR)/aquarium
-	cp ../etc/gnome1/shermans_aquarium.gnorba $(GNOMEDIR)/etc/CORBA/servers
-	cp ../etc/gnome1/shermans_aquarium.desktop $(GNOMEDIR)/share/applets/Amusements
-	cp ../aquarium/sherman_icon.png $(GNOMEDIR)/share/pixmaps
+	mkdir -p $(DESTDIR)@GNOME_INSTALL_DIR@/bin
+	cp sherman $(DESTDIR)@GNOME_INSTALL_DIR@/bin
+	mkdir -p $(DESTDIR)$(DATADIR)/aquarium
+	cp -R ../aquarium/* $(DESTDIR)$(DATADIR)/aquarium
+	cp ../etc/gnome1/shermans_aquarium.gnorba $(DESTDIR)$(GNOMEDIR)/etc/CORBA/servers
+	cp ../etc/gnome1/shermans_aquarium.desktop $(DESTDIR)$(GNOMEDIR)/share/applets/Amusements
+	cp ../aquarium/sherman_icon.png $(DESTDIR)$(GNOMEDIR)/share/pixmaps
 
 install-gnome2:
-	mkdir -p @GNOME_INSTALL_DIR@/bin
-	cp sherman @GNOME_INSTALL_DIR@/bin
+	mkdir -p $(DESTDIR)@GNOME_INSTALL_DIR@/bin
+	cp sherman $(DESTDIR)@GNOME_INSTALL_DIR@/bin
 	mkdir -p $(DATADIR)/aquarium
 	cp -R ../aquarium/* $(DATADIR)/aquarium
-	cp ../etc/gnome2/GNOME_ShermansaquariumApplet.server $(GNOMEDIR)/lib/bonobo/servers
-	cp ../etc/gnome2/GNOME_ShermansaquariumApplet.xml $(GNOMEDIR)/share/gnome-2.0/ui
-	cp ../aquarium/sherman_icon.png $(GNOMEDIR)/share/pixmaps
+	cp ../etc/gnome2/GNOME_ShermansaquariumApplet.server $(DESTDIR)$(GNOMEDIR)/lib/bonobo/servers
+	cp ../etc/gnome2/GNOME_ShermansaquariumApplet.xml $(DESTDIR)$(GNOMEDIR)/share/gnome-2.0/ui
+	cp ../aquarium/sherman_icon.png $(DESTDIR)$(GNOMEDIR)/share/pixmaps
 	
 	
 	
 install-wmonly:
-	mkdir -p $(PREFIX)/bin
-	cp sherman $(PREFIX)/bin
-	mkdir -p $(DATADIR)/aquarium
-	cp -R ../aquarium/* $(DATADIR)/aquarium
+	mkdir -p $(DESTDIR)$(PREFIX)/bin
+	cp sherman $(DESTDIR)$(PREFIX)/bin
+	mkdir -p $(DESTDIR)$(DATADIR)/aquarium
+	cp -R ../aquarium/* $(DESTDIR)$(DATADIR)/aquarium
 
 clean:
-	rm -rf *.o .xvpics *~ sherman
\ No newline at end of file
+	rm -rf *.o .xvpics *~ sherman
