diff -ru arj-3.10.22/environ.c arj-3.10.22-patched/environ.c
--- arj-3.10.22/environ.c	2004-06-18 18:19:36.000000000 +0200
+++ arj-3.10.22-patched/environ.c	2010-09-28 12:51:48.000000000 +0200
@@ -58,7 +58,7 @@
   #include <sys/ioctl.h>
   #include <sys/statfs.h>
   #include <sys/statvfs.h>
- #elif defined(__FreeBSD__)||defined(__NetBSD__)
+ #elif defined(__APPLE__)||defined(__NetBSD__)
   #include <sys/param.h>
   #include <sys/mount.h>
  #elif defined(__QNXNTO__)
diff -ru arj-3.10.22/gnu/configure.in arj-3.10.22-patched/gnu/configure.in
--- arj-3.10.22/gnu/configure.in	2005-06-21 20:27:20.000000000 +0200
+++ arj-3.10.22-patched/gnu/configure.in	2010-09-28 12:51:48.000000000 +0200
@@ -94,7 +94,7 @@
 
 case $host_cpu in
 alpha*)
-	AC_DEFINE(ALIGN_POINTERS)
+	AC_DEFINE(ALIGN_POINTERS, 1, [Used by encode.c])
 	;;
 sparc*)
 	AC_DEFINE(ALIGN_POINTERS)
@@ -176,7 +176,7 @@
         [color_output=yes],
         [color_output=no])
 AC_MSG_RESULT($color_output)
-test "$color_output" = "yes" && AC_DEFINE(USE_COLORS)
+AC_DEFINE(USE_COLORS, test "$color_output" = "yes", [Enable color output])
 
 AC_SUBST(HAVE_MKDTEMP)
 AC_SUBST(HAVE_FCLOSEALL)
Nur in arj-3.10.22-patched/gnu: configure.in.orig.
diff -ru arj-3.10.22/integr.c arj-3.10.22-patched/integr.c
--- arj-3.10.22/integr.c	2005-06-21 20:27:18.000000000 +0200
+++ arj-3.10.22-patched/integr.c	2010-09-28 12:54:23.000000000 +0200
@@ -5,7 +5,7 @@
  *
  */
 
-static const char intergrity_identifier[] = {
+const char intergrity_identifier[] = {
 	0xB0, 0x03, 0xB0, 0x02, 0xB0, 0x03,
 	0xB0, 0x04, 0xB0, 0x05,
 	0x90, 0x90, 0x90, 0x90,
