diff -Nurd boo/default.build boo-new/default.build
--- boo/default.build	2009-12-23 10:57:14.000000000 -0500
+++ boo-new/default.build	2010-02-19 14:56:11.000000000 -0500
@@ -71,7 +71,7 @@
 		<property name="nosign" value="true" overwrite="true" />
 	</target>
 
-	<target name="all" depends="booc, booi, booish, boo-pkgconfig, Boo.Lang.CodeDom, Boo.Lang.Useful, boo, Boo.Microsoft.Build.Tasks, update-vs2005-env, Boo.Lang.PatternMatching">
+	<target name="all" depends="booc, booi, booish, boo-pkgconfig, boo">
 		<call target="gendarme" if="${gendarme.dir != ''}" />
 	</target>
 
@@ -119,7 +119,7 @@
 		</exec>
 	</target>
 	
-	<target name="boo" depends="Boo.Lang.Useful, Boo.Lang.Interpreter">
+	<target name="boo" depends="init">
 		<booc
 			target="exe"
 			output="${build.dir}/boo.exe"
@@ -138,7 +138,7 @@
 		</booc>
 	</target>
 	
-	<target name="booish" depends="Boo.Lang.Interpreter">
+	<target name="booish" depends="init">
 		<booc
 			target="exe"
 			output="${build.dir}/booish.exe"
@@ -174,7 +174,7 @@
 		</if>
 	</target>
 
-	<target name="booi" depends="core, Boo.Lang.Useful, Boo.Lang.Parser">
+	<target name="booi" depends="init">
 
 		<booc
 			target="exe"
@@ -218,7 +218,7 @@
 
 	</target>
 	
-	<target name="dump-properties" depends="core">
+	<target name="dump-properties" depends="init">
 		<boo>
 		for p as System.Collections.DictionaryEntry in Project.Properties:
 			print("${p.Key}: ${p.Value}")		
@@ -289,7 +289,7 @@
 		<loadtasks assembly="${build.dir}/Boo.NAnt.Tasks.dll" />
 	</target>
 	
-	<target name="booc" depends="corecs">
+	<target name="booc" depends="init">
 		<csc target="exe" 
 				output="${build.dir}/booc.exe" 
 				debug="${debug}" 
@@ -1149,6 +1149,7 @@
 		<!-- This property define is a way to get around the 0.85 NAnt bug regarding signing satelite resource assemblies. -->
 		<property name="csc.define" value="${csc.define};IGNOREKEYFILE" unless="${nant-csc-keyfile-bug and not nosign}" />
 		<mkdir dir="${build.dir}" />
+		<loadtasks assembly="${build.dir}/Boo.NAnt.Tasks.dll" />
 	</target>
 	
 	<target name="makedeb" depends="rebuild">
