diff -Nurd -x'*~' ekiga-3.1.2.orig/lib/engine/Makefile.in ekiga-3.1.2/lib/engine/Makefile.in
--- ekiga-3.1.2.orig/lib/engine/Makefile.in	2009-03-02 16:02:55.000000000 -0500
+++ ekiga-3.1.2/lib/engine/Makefile.in	2009-03-08 21:52:09.000000000 -0400
@@ -96,7 +96,6 @@
 am__DEPENDENCIES_1 =
 am__DEPENDENCIES_2 = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 libekiga_engine_la_DEPENDENCIES =  \
-	$(top_builddir)/lib/engine/notification/libnotification.la \
 	$(top_builddir)/lib/engine/gui/gtk-core/libgmgtk-core.la \
 	$(top_builddir)/lib/engine/gui/gtk-frontend/libgmgtk-frontend.la \
 	$(top_builddir)/lib/engine/chat/libgmchat.la \
@@ -373,7 +372,7 @@
 target_vendor = @target_vendor@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = framework  plugin notification account addressbook presence chat gui protocol videooutput videoinput audioinput audiooutput hal components
+SUBDIRS = framework  plugin account addressbook presence chat gui protocol videooutput videoinput audioinput audiooutput hal components
 noinst_LTLIBRARIES = libekiga_engine.la
 
 # Default Input Devices
@@ -433,7 +432,6 @@
 
 # PTLIB Input Devices
 libekiga_engine_la_LIBADD =  \
-	$(top_builddir)/lib/engine/notification/libnotification.la \
 	$(top_builddir)/lib/engine/gui/gtk-core/libgmgtk-core.la \
 	$(top_builddir)/lib/engine/gui/gtk-frontend/libgmgtk-frontend.la \
 	$(top_builddir)/lib/engine/chat/libgmchat.la \
diff -Nurd -x'*~' ekiga-3.1.2.orig/lib/engine/components/libnotify/Makefile.in ekiga-3.1.2/lib/engine/components/libnotify/Makefile.in
--- ekiga-3.1.2.orig/lib/engine/components/libnotify/Makefile.in	2009-03-02 16:02:57.000000000 -0500
+++ ekiga-3.1.2/lib/engine/components/libnotify/Makefile.in	2009-03-08 21:51:42.000000000 -0400
@@ -44,8 +44,7 @@
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(noinst_LTLIBRARIES)
 libgmlibnotify_la_DEPENDENCIES =  \
-	$(top_builddir)/lib/engine/framework/libgmframework.la \
-	$(top_builddir)/lib/engine/notification/libnotification.la
+	$(top_builddir)/lib/engine/framework/libgmframework.la
 am_libgmlibnotify_la_OBJECTS = libnotify-main.lo
 libgmlibnotify_la_OBJECTS = $(am_libgmlibnotify_la_OBJECTS)
 libgmlibnotify_la_LINK = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) \
@@ -297,8 +296,7 @@
 	$(libnotify_dir)/libnotify-main.cpp
 
 libgmlibnotify_la_LIBADD = \
-	$(top_builddir)/lib/engine/framework/libgmframework.la \
-	$(top_builddir)/lib/engine/notification/libnotification.la
+	$(top_builddir)/lib/engine/framework/libgmframework.la
 
 libgmlibnotify_la_LDFLAGS = -export-dynamic -no-undefined $(SIGC_LIBS) $(LIBNOTIFY_LIBS)
 all: all-am
