diff -ruN nessus-libraries.orig/libhosts_gatherer/Makefile nessus-libraries/libhosts_gatherer/Makefile
--- nessus-libraries.orig/libhosts_gatherer/Makefile	2001-11-15 07:13:22.000000000 -0700
+++ nessus-libraries/libhosts_gatherer/Makefile	2002-12-23 09:01:57.000000000 -0700
@@ -23,7 +23,7 @@
 	
 libhosts_gatherer.la : $(OBJS)
 	$(LIBTOOL) $(CC) -o libhosts_gatherer.la $(LO_OBJS) -rpath ${libdir} \
-	-version-info ${PACKAGE_VERSION}
+	-version-info ${PACKAGE_VERSION} $(LIBS)
 	        
 hg_utils.o : hg_utils.h hosts_gatherer.h hg_utils.c
 	$(LIBTOOL) --mode=compile  $(COMPILE) -c hg_utils.c
diff -ruN nessus-libraries.orig/libnessus/Makefile nessus-libraries/libnessus/Makefile
--- nessus-libraries.orig/libnessus/Makefile	2002-09-19 07:30:02.000000000 -0600
+++ nessus-libraries/libnessus/Makefile	2002-12-23 09:01:21.000000000 -0700
@@ -74,7 +74,7 @@
 
 libnessus.la: $(OBJS)
 	$(LIBTOOL) $(CC) -o libnessus.la $(LO_OBJS) -rpath ${libdir} \
-	-version-info ${PACKAGE_VERSION}
+	-version-info ${PACKAGE_VERSION} $(LIBS)
 
 install : libnessus.la
 	test -d ${libdir} || $(INSTALL_DIR) -m 755 ${libdir}
