diff -Nurd -x'*~' librsvg-2.22.3.orig/Makefile.in librsvg-2.22.3/Makefile.in
--- librsvg-2.22.3.orig/Makefile.in	2008-09-22 20:54:52.000000000 -0400
+++ librsvg-2.22.3/Makefile.in	2009-03-18 02:44:54.000000000 -0400
@@ -306,7 +306,7 @@
 top_srcdir = @top_srcdir@
 @HAVE_GTK_FALSE@target_rsvg_view = 
 @HAVE_GTK_TRUE@target_rsvg_view = rsvg-view
-SUBDIRS = . moz-plugin gdk-pixbuf-loader gtk-engine data tests doc
+SUBDIRS = moz-plugin
 lib_LTLIBRARIES = librsvg-2.la
 man_MANS = rsvg.1
 @OS_WIN32_FALSE@libm = -lm
@@ -745,7 +745,7 @@
 	    *k*) failcom='fail=yes';; \
 	  esac; \
 	done; \
-	dot_seen=no; \
+	dot_seen=yes; \
 	target=`echo $@ | sed s/-recursive//`; \
 	list='$(SUBDIRS)'; for subdir in $$list; do \
 	  echo "Making $$target in $$subdir"; \
