diff -Nurd -x'*~' xmlsec1-1.2.9.orig/configure xmlsec1-1.2.9/configure
--- xmlsec1-1.2.9.orig/configure	2005-07-12 22:18:37.000000000 -0400
+++ xmlsec1-1.2.9/configure	2005-12-05 16:03:55.000000000 -0500
@@ -25070,7 +25070,7 @@
 echo "${ECHO_T}no" >&6
 elif test "z$with_openssl" != "z" ; then
     OPENSSL_CFLAGS="$OPENSSL_CFLAGS -I$with_openssl/include"
-    if test -f "$with_openssl/lib/libcrypto$shrext" ; then
+    if test -f "$with_openssl/lib/libcrypto.dylib" ; then
 	OPENSSL_LIBS="-L$with_openssl/lib -lcrypto -ldl"
     else
 	OPENSSL_LIBS="$with_openssl/lib/libcrypto.a -ldl"
