diff -ur compface-1.4/Makefile.in compface-1.4-patched/Makefile.in
--- compface-1.4/Makefile.in	Wed Nov 22 08:38:39 2000
+++ compface-1.4-patched/Makefile.in	Sun May 19 14:46:43 2002
@@ -17,11 +17,14 @@
 SHELL           = /bin/sh
 
 NAME		= compface
+VERSION		= 1.4
 UNNAME		= uncompface
 EXECUTABLE	= $(BINDIR)/$(NAME)
 UNEXECUTABLE	= $(BINDIR)/$(UNNAME)
 LIBNAME		= lib$(NAME).a
 LIBRARY		= $(LIBDIR)/$(LIBNAME)
+DYLIBNAME	= lib$(NAME).0.dylib
+DYLIBLONGNAME	= lib$(NAME).0.$(VERSION).dylib
 MAN1DIR		= $(MANDIR)/man1
 MAN3DIR		= $(MANDIR)/man3
 OBJECTS		= arith.o file.o compress.o gen.o uncompface.o
@@ -52,6 +55,9 @@
 		ar rc $(LIBNAME) $(OBJECTS)
 		-$(RANLIB) $(LIBNAME)
 
+$(DYLIBLONGNAME) :	$(OBJECTS)
+		$(CC) -dynamiclib $(CFLAGS) $(LDFLAGS) -install_name $(LIBDIR)/$(DYLIBNAME) -current_version 0.$(VERSION) -compatibility_version 0.$(VERSION) -o $(DYLIBLONGNAME) $(OBJECTS)
+
 lint :
 		lint -abchx $(SOURCES)
 
@@ -71,18 +77,18 @@
 		$(RM) -rf `cat .fname` .fname
 
 install: 	$(NAME) $(UNNAME) $(LIBNAME)
-		$(INSTALL_PROGRAM) $(srcdir)/$(NAME) $(EXECUTABLE) 
-		-chmod 0755 $(EXECUTABLE)
-		-strip $(EXECUTABLE)
-		$(INSTALL_PROGRAM) $(srcdir)/$(UNNAME) $(UNEXECUTABLE)
-		-chmod 0755 $(UNEXECUTABLE)
-		-strip $(UNEXECUTABLE)
-		$(INSTALL_DATA) $(srcdir)/$(NAME).1 $(MAN1DIR)/$(NAME).1
-		cd $(MAN1DIR) && $(RM) ./$(UNNAME).1 && $(LN_S) $(NAME).1 $(UNNAME).1
-		$(INSTALL_DATA) $(srcdir)/$(NAME).3 $(MAN3DIR)/$(NAME).3
-		cd $(MAN3DIR) && $(RM) ./$(UNNAME).3 && $(LN_S) $(NAME).3 $(UNNAME).3
-		for hdr in $(INSTALLHEADERS); do $(INSTALL_DATA) $(srcdir)/$${hdr} $(INCLUDEDIR)/$${hdr}; done
-		$(INSTALL_DATA) $(srcdir)/$(LIBNAME) $(LIBRARY)
+		$(INSTALL_PROGRAM) $(srcdir)/$(NAME) $(DESTDIR)$(EXECUTABLE) 
+		-chmod 0755 $(DESTDIR)$(EXECUTABLE)
+		-strip $(DESTDIR)$(EXECUTABLE)
+		$(INSTALL_PROGRAM) $(srcdir)/$(UNNAME) $(DESTDIR)$(UNEXECUTABLE)
+		-chmod 0755 $(DESTDIR)$(UNEXECUTABLE)
+		-strip $(DESTDIR)$(UNEXECUTABLE)
+		$(INSTALL_DATA) $(srcdir)/$(NAME).1 $(DESTDIR)$(MAN1DIR)/$(NAME).1
+		cd $(DESTDIR)$(MAN1DIR) && $(RM) ./$(UNNAME).1 && $(LN_S) $(NAME).1 $(UNNAME).1
+		$(INSTALL_DATA) $(srcdir)/$(NAME).3 $(DESTDIR)$(MAN3DIR)/$(NAME).3
+		cd $(DESTDIR)$(MAN3DIR) && $(RM) ./$(UNNAME).3 && $(LN_S) $(NAME).3 $(UNNAME).3
+		for hdr in $(INSTALLHEADERS); do $(INSTALL_DATA) $(srcdir)/$${hdr} $(DESTDIR)$(INCLUDEDIR)/$${hdr}; done
+		$(INSTALL_DATA) $(srcdir)/$(LIBNAME) $(DESTDIR)$(LIBRARY)
 
 shar :
 		shar.script $(OTHERS) $(HDRS) $(SOURCES) > $(NAME).sh
