diff -ur lcms-1.09/include/lcms.h lcms-1.09.corr/include/lcms.h
--- lcms-1.09/include/lcms.h	Wed Aug  7 21:34:37 2002
+++ lcms-1.09.corr/include/lcms.h	Sat Feb  1 08:15:43 2003
@@ -112,7 +112,7 @@
 #  define USE_C               1
 #endif
 
-#if defined(__sgi__) || defined(__sgi) || defined(__powerpc__)
+#if defined(__sgi__) || defined(__sgi) || defined(__powerpc__) || defined(__ppc__)
 #   define USE_BIG_ENDIAN      1
 #endif
 
@@ -128,7 +128,7 @@
 #   define USE_BIG_ENDIAN      1
 #endif
 
-#if defined(__OpenBSD__) || defined(__NetBSD__) || defined(__FreeBSD__)
+#if defined(__OpenBSD__) || defined(__NetBSD__) || defined(__FreeBSD__) || defined(__Apple__)
 #  include <sys/types.h>
 #  define USE_INT64           1
 #  undef  LONGLONG
diff -ur lcms-1.09/samples/Makefile lcms-1.09.corr/samples/Makefile
--- lcms-1.09/samples/Makefile	Fri Jun 14 17:07:01 2002
+++ lcms-1.09.corr/samples/Makefile	Sat Feb  1 08:37:04 2003
@@ -1,7 +1,7 @@
 SHELL = /bin/sh
 
 CFLAGS = -g -O4 -Wall
-BASEDIR = $(DESTDIR)/usr
+BASEDIR = $(DESTDIR)/$(PREFIX)
 BINDIR  = $(BASEDIR)/bin
 
 
diff -ur lcms-1.09/samples/icctrans.c lcms-1.09.corr/samples/icctrans.c
--- lcms-1.09/samples/icctrans.c	Fri Jun 14 17:17:36 2002
+++ lcms-1.09.corr/samples/icctrans.c	Sat Feb  1 08:15:43 2003
@@ -31,6 +31,9 @@
 #include "lcms.h"
 #include <stdarg.h>
 #include <ctype.h>
+#ifdef __APPLE__
+#include <unistd.h>
+#endif
 
 // xgetopt() interface -----------------------------------------------------
 
diff -ur lcms-1.09/src/cmsio1.c lcms-1.09.corr/src/cmsio1.c
--- lcms-1.09/src/cmsio1.c	Wed Aug  7 20:45:57 2002
+++ lcms-1.09.corr/src/cmsio1.c	Sat Feb  1 08:15:43 2003
@@ -38,6 +38,9 @@
 
 #include <stdlib.h>
 #include <time.h>
+#ifdef __APPLE__
+#include <unistd.h>
+#endif
 
 cmsHPROFILE   LCMSEXPORT cmsOpenProfileFromFile(const char *ICCProfile, const char *sAccess);
 cmsHPROFILE   LCMSEXPORT cmsOpenProfileFromMem(LPVOID MemPtr, DWORD dwSize);
diff -ur lcms-1.09/src/makefile lcms-1.09.corr/src/makefile
--- lcms-1.09/src/makefile	Mon Jun 17 16:29:45 2002
+++ lcms-1.09.corr/src/makefile	Sat Feb  1 08:15:43 2003
@@ -2,7 +2,7 @@
 
 CFLAGS  = -g -O4 -I../include -fomit-frame-pointer -Wall
 LDFLAGS = -lm
-BASEDIR = $(DESTDIR)/usr
+BASEDIR = $(DESTDIR)/$(PREFIX)
 BINDIR  = $(BASEDIR)/bin
 LIBDIR  = $(BASEDIR)/lib
 INCDIR  = $(BASEDIR)/include
@@ -21,7 +21,7 @@
 
 
 
-all:	liblcms.a liblcms.so.${VERSION} 
+all:	liblcms.a liblcms.${VERSION}.dylib 
 
 
 #
@@ -32,19 +32,22 @@
 	$(AR) liblcms.a $(OBJS)
 	$(RANLIB) liblcms.a
 
-liblcms.so.${VERSION}: $(OBJS) 
-	-$(CC) $(CFLAGS) $(LDFLAGS) -shared -o liblcms.so.${VERSION} $(OBJS) 
-	-ln -s liblcms.so.${VERSION} liblcms.so.1
-	-ln -s liblcms.so.1 liblcms.so
+liblcms.${VERSION}.dylib: $(OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -prebind -dynamiclib \
+	 -install_name $(PREFIX)/lib/liblcms.1.dylib \
+	 -compatibility_version $(VERSION) -current_version $(VERSION) \
+	 -o liblcms.${VERSION}.dylib $(OBJS)
+	ln -s liblcms.${VERSION}.dylib liblcms.1.dylib
+	ln -s liblcms.1.dylib liblcms.dylib
 
 #
 # Installation
 #
 
-install: liblcms.a  liblcms.so.${VERSION}
+install: liblcms.a  liblcms.${VERSION}.dylib
 	mkdir -p $(BINDIR) $(LIBDIR) $(INCDIR)
-	-cp -a liblcms.so* $(LIBDIR)
-	cp -a liblcms.a $(LIBDIR)
+	-cp -R liblcms*.dylib $(LIBDIR)
+	cp -R liblcms.a $(LIBDIR)
 	cp ../include/lcms.h ../include/icc34.h $(INCDIR)
 
 #
@@ -61,4 +64,4 @@
 #
 
 clean:
-	-rm $(OBJS)  liblcms.a liblcms.so.${VERSION} liblcms.so.1 liblcms.so
+	-rm $(OBJS)  liblcms.a liblcms.${VERSION}.dylib liblcms.1.dylib liblcms.dylib
