diff -ruN MPlayer-0.90rc4/configure MPlayer-0.90rc4.new/configure
--- MPlayer-0.90rc4/configure	2003-02-08 17:29:05.000000000 -0500
+++ MPlayer-0.90rc4.new/configure	2003-03-01 20:39:53.000000000 -0500
@@ -29,9 +29,9 @@
   echo >> "$TMPLOG"
   cat "$TMPC" >> "$TMPLOG"
   echo >> "$TMPLOG"
-  echo "$_cc $_inc_extra $_ld_static $_ld_extra $TMPC -o $TMPO $@" >> "$TMPLOG"
+  echo "$_cc $CFLAGS $_inc_extra $_ld_static $_ld_extra $TMPC -o $TMPO $@" >> "$TMPLOG"
   rm -f "$TMPO"
-  ( $_cc $_inc_extra $_ld_static $_ld_extra "$TMPC" -o "$TMPO" "$@" ) >> "$TMPLOG" 2>&1
+  ( $_cc $CFLAGS $_inc_extra $_ld_static $_ld_extra "$TMPC" -o "$TMPO" "$@" ) >> "$TMPLOG" 2>&1
   TMP="$?"
   echo >> "$TMPLOG"
   echo "ldd $TMPO" >> "$TMPLOG"
@@ -1723,6 +1723,10 @@
 cc_check -lsocket -lnsl && _ld_sock="-lsocket -lnsl"
 if test "$_ld_sock" ; then
   echores "yes (using $_ld_sock)"
+elif cc_check $_ld_sock -ldnet ; then
+  # NOTE: needed for Darwin at least
+  _ld_sock="$_ld_sock -ldnet"
+  echores "yes (using $_ld_sock)"
 else
   echores "no"
 fi
@@ -2183,6 +2187,16 @@
 else
   _def_shm='#undef HAVE_SHM'
 fi
+# Darwin Xfree86 has x-shm disabled thus shm has to be disabled
+if darwin ; then
+  if test `uname -r` > 5.9 ; then
+    _def_shm='#define HAVE_SHM 1'
+    _shm='yes (Darwin 6+/OSX 10.2+ supports SHM)'
+  else
+    _def_shm='#undef HAVE_SHM'
+    _shm='no (not supported by Darwin)'
+  fi
+fi
 echores "$_shm"
 
 
diff -ruN MPlayer-0.90rc4/configure.rej MPlayer-0.90rc4.new/configure.rej
--- MPlayer-0.90rc4/configure.rej	1969-12-31 19:00:00.000000000 -0500
+++ MPlayer-0.90rc4.new/configure.rej	2003-03-01 20:38:28.000000000 -0500
@@ -0,0 +1,29 @@
+***************
+*** 29,41 ****
+    echo >> "$TMPLOG"
+    cat "$TMPC" >> "$TMPLOG"
+    echo >> "$TMPLOG"
+-   echo "$_cc $_inc_extra $_ld_static $_ld_extra $TMPC -o $TMPO $@" >> "$TMPLOG"
+    rm -f "$TMPO"
+-   ( "$_cc" $_inc_extra $_ld_static $_ld_extra "$TMPC" -o "$TMPO" "$@" ) >> "$TMPLOG" 2>&1
+    TMP="$?"
+    echo >> "$TMPLOG"
+-   echo "ldd $TMPO" >> "$TMPLOG"
+-   ( ldd "$TMPO" ) >> "$TMPLOG" 2>&1
+    echo >> "$TMPLOG"
+    return "$TMP"
+  }
+--- 29,41 ----
+    echo >> "$TMPLOG"
+    cat "$TMPC" >> "$TMPLOG"
+    echo >> "$TMPLOG"
++   echo "$_cc $CFLAGS $_inc_extra $_ld_static $_ld_extra $TMPC -o $TMPO $@" >> "$TMPLOG"
+    rm -f "$TMPO"
++   ( "$_cc" $CFLAGS $_inc_extra $_ld_static $_ld_extra "$TMPC" -o "$TMPO" "$@" ) >> "$TMPLOG" 2>&1
+    TMP="$?"
+    echo >> "$TMPLOG"
++   echo "nm $TMPO" >> "$TMPLOG"
++   ( nm "$TMPO" ) >> "$TMPLOG" 2>&1
+    echo >> "$TMPLOG"
+    return "$TMP"
+  }
diff -ruN MPlayer-0.90rc4/etc/example.conf MPlayer-0.90rc4.new/etc/example.conf
--- MPlayer-0.90rc4/etc/example.conf	2002-12-07 15:31:37.000000000 -0500
+++ MPlayer-0.90rc4.new/etc/example.conf	2003-03-01 20:38:28.000000000 -0500
@@ -8,13 +8,13 @@
 ## more flexibly here. See below.
 ##
 
-# vo=xv			# To specify default video driver (see -vo help for
+vo=sdl			# To specify default video driver (see -vo help for
 			# list)
 
-# ao=oss		# To specify default audio driver (see -ao help for
+ao=sdl			# To specify default audio driver (see -ao help for
 			# list)
 
-fs=yes			# Enlarges movie window to your desktop's size.
+fs=no			# Enlarges movie window to your desktop's size.
 			# Used by drivers: all
 
 # vm=no			# Tries to change to a different videomode
@@ -38,7 +38,7 @@
 ## Use GUI mode by default
 ##
 
-# gui = yes
+gui = yes
 
 ##
 ## Specify your preferred default skin here
@@ -46,7 +46,7 @@
 ##  and ~/.mplayer/Skin/yourskin)
 ##
 
-# skin = default
+skin = default
 
 ##
 ## Multiple languages are available :)
@@ -133,4 +133,4 @@
 ## Delete this default :)
 ##
 
-include = /home/gabucino/.mplayer/i_did_not_RTFM_carefully_enough...
+#include = /home/gabucino/.mplayer/i_did_not_RTFM_carefully_enough...
diff -ruN MPlayer-0.90rc4/libmpcodecs/vf_fame.c MPlayer-0.90rc4.new/libmpcodecs/vf_fame.c
--- MPlayer-0.90rc4/libmpcodecs/vf_fame.c	2002-09-29 15:19:31.000000000 -0400
+++ MPlayer-0.90rc4.new/libmpcodecs/vf_fame.c	2003-03-01 20:38:28.000000000 -0500
@@ -16,7 +16,7 @@
 #include "vf.h"
 
 //#include "../libvo/fastmemcpy.h"
-#include "../libfame/fame.h"
+#include <fame.h>
 
 struct vf_priv_s {
     unsigned char* outbuf;
diff -ruN MPlayer-0.90rc4/libvo/vo_sdl.c MPlayer-0.90rc4.new/libvo/vo_sdl.c
--- MPlayer-0.90rc4/libvo/vo_sdl.c	2003-01-12 20:20:21.000000000 -0500
+++ MPlayer-0.90rc4.new/libvo/vo_sdl.c	2003-03-01 20:38:28.000000000 -0500
@@ -893,7 +893,7 @@
 		||(strcmp(priv->driver, "directx") == 0)
 		||(strcmp(priv->driver, "Quartz") == 0)
 		||((strcmp(priv->driver, "aalib") == 0)
-		&& priv->X)) {
+		&& priv->X) || (strcmp(priv->driver, "Quartz") == 0)) {
 			if(verbose) printf("SDL: setting windowed mode\n");
             set_video_mode(priv->dstwidth, priv->dstheight, priv->bpp, priv->sdlflags);
 		}
