--- Raster3D_2.6e/Makefile	2002-06-04 04:41:32.000000000 +1000
+++ Raster3D_2.6e-patched/Makefile	2002-06-18 16:15:48.000000000 +1000
@@ -49,10 +49,10 @@
 linux:	strip-for-g77
 	@cp Makefile.template Makefile.incl
 	@echo OS = linux                      >> Makefile.incl
-	@echo CC = gcc                        >> Makefile.incl
-	@echo CFLAGS = -g -m486 -w            >> Makefile.incl
+	@echo CC = cc                         >> Makefile.incl
+	@echo CFLAGS = -g -dynamic            >> Makefile.incl
 	@echo FC = g77                        >> Makefile.incl
-	@echo FFLAGS = -g -O -w -malign-double>> Makefile.incl
+	@echo FFLAGS = -g -O -w               >> Makefile.incl
 	@echo RM = /bin/rm -f                 >> Makefile.incl
 	@echo OSDEFS =  -DLINUX -DNETWORKBYTEORDER       >> Makefile.incl
 	@echo include Makefile.package        >> Makefile.incl

--- Raster3D_2.6e/Makefile.template	2002-06-04 04:41:32.000000000 +1000
+++ Raster3D_2.6e-patched/Makefile.template	2002-06-18 16:21:45.000000000 +1000
@@ -8,15 +8,15 @@
 #
 # Where to install the program, data files, and manuals
 #
-prefix  = /usr/local
+prefix  = $(prefix)
 bindir  = $(prefix)/bin
 datadir = $(prefix)/share/Raster3D/materials
-mandir  = $(prefix)/man/manl
+mandir  = $(prefix)/share/man/manl
 htmldir = $(prefix)/share/Raster3D/html
 examdir = $(prefix)/share/Raster3D/examples
 
-INCDIRS  =	-I/usr/include -I/usr/local/include
-LIBDIRS  =	-L/usr/local/lib
+INCDIRS  =	-I/usr/include -I@prefix@/include
+LIBDIRS  =	-L@prefix@/lib
 
 #
 # TIFF support (enabled by default)

--- Raster3D_2.6e/label3d	Tue Jun  4 04:41:32 2002
+++ Raster3D_2.6e-patched/label3d	Mon Sep  9 13:07:45 2002
@@ -38,7 +38,7 @@
 if [ "$TMPDIR" ]; then
     tmp=$TMPDIR/$$
 else
-    tmp=/usr/tmp/$$
+    tmp=/tmp/$$
 fi
 #echo "DEBUG scratch files to $tmp"  1>&2
 
--- Raster3D_2.6e/stereo3d	Tue Jun  4 04:41:33 2002
+++ Raster3D_2.6e-patched/stereo3d	Mon Sep  9 13:38:31 2002
@@ -42,7 +42,7 @@
 if [ "$TMPDIR" ]; then
     tmp=$TMPDIR/$$
 else
-    tmp=/usr/tmp/$$
+    tmp=/tmp/$$
 fi
 # echo "DEBUG scratch files to $tmp"  1>&2
 
@@ -131,9 +131,9 @@
 #
 echo "stereo3d: joining left and right images" 1>&2
 if [ "$STEREOBORDER" ]; then
-   montage +frame +label -background black -geometry $IMAGESIZE+2+0! -scene 0 ${tmp}_left.${img} ${tmp}_right.${img} ${tmp}_stereo3d.${img}
+   montage +frame +label -background black -geometry $IMAGESIZE+2+0! ${tmp}_left.${img} ${tmp}_right.${img} ${tmp}_stereo3d.${img}
 else
-   montage +frame +label -background white -geometry $IMAGESIZE+0+0! -scene 0 ${tmp}_left.${img} ${tmp}_right.${img} ${tmp}_stereo3d.${img}
+   montage +frame +label -background white -geometry $IMAGESIZE+0+0! ${tmp}_left.${img} ${tmp}_right.${img} ${tmp}_stereo3d.${img}
 fi
 #
 # This next bit is only necessary because ImageMagick _always_ writes a label

