diff -ruN xaos-3.0/src/engine/blur.c xaos-3.0-patched/src/engine/blur.c
--- xaos-3.0/src/engine/blur.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/blur.c	Thu May 30 14:04:40 2002
@@ -1,5 +1,7 @@
 #ifndef _plan9_
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <string.h>
 #include <math.h>
 #include <config.h>
diff -ruN xaos-3.0/src/engine/btrace.c xaos-3.0-patched/src/engine/btrace.c
--- xaos-3.0/src/engine/btrace.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/btrace.c	Thu May 30 14:04:40 2002
@@ -1,5 +1,7 @@
 #ifndef _plan9_
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <math.h>
 #include <stdio.h>
 #include <string.h>
diff -ruN xaos-3.0/src/engine/dither.c xaos-3.0-patched/src/engine/dither.c
--- xaos-3.0/src/engine/dither.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/dither.c	Thu May 30 14:04:40 2002
@@ -14,7 +14,9 @@
  * fixedcolor displays.
  */
 #ifndef _plan9_
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <config.h>
 #include <stdio.h>
 #include <string.h>
diff -ruN xaos-3.0/src/engine/edge.c xaos-3.0-patched/src/engine/edge.c
--- xaos-3.0/src/engine/edge.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/edge.c	Thu May 30 14:04:40 2002
@@ -3,7 +3,9 @@
  * does an simple edge detection algo on it.
  */
 #ifndef _plan9_
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <stdio.h>		/*for NULL */
 #else
 #include <u.h>
diff -ruN xaos-3.0/src/engine/edge2.c xaos-3.0-patched/src/engine/edge2.c
--- xaos-3.0/src/engine/edge2.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/edge2.c	Thu May 30 14:04:40 2002
@@ -3,7 +3,9 @@
  * does an simple edge detection algo on it.
  */
 #ifndef _plan9_
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <stdio.h>		/*for NULL */
 #else
 #include <u.h>
diff -ruN xaos-3.0/src/engine/emboss.c xaos-3.0-patched/src/engine/emboss.c
--- xaos-3.0/src/engine/emboss.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/emboss.c	Thu May 30 14:26:19 2002
@@ -1,5 +1,9 @@
 #ifndef _plan9_
+#ifdef NO_MALLOC_H
+#include <stdlib.h>
+#else
 #include <malloc.h>
+#endif
 #include <stdio.h>
 #else
 #include <u.h>
@@ -100,7 +104,7 @@
 getinstance (struct filteraction *a)
 {
   struct filter *f = createfilter (a);
-  struct embossdata *i = calloc (sizeof (*i), 1);
+  struct embossdata *i = calloc (1, sizeof (*i));
   i->savedpalette = NULL;
   i->palette = createpalette (0, 256, GRAYSCALE, 0, 256, NULL, NULL, NULL, NULL);
   f->childimage = NULL;
diff -ruN xaos-3.0/src/engine/fractal.c xaos-3.0-patched/src/engine/fractal.c
--- xaos-3.0/src/engine/fractal.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/fractal.c	Thu May 30 14:04:40 2002
@@ -27,7 +27,9 @@
 #else
 #include <stdio.h>
 #ifndef _MAC
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #endif
 #include <aconfig.h>
 #include <string.h>
diff -ruN xaos-3.0/src/engine/interlace.c xaos-3.0-patched/src/engine/interlace.c
--- xaos-3.0/src/engine/interlace.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/interlace.c	Thu May 30 14:24:58 2002
@@ -1,5 +1,9 @@
 #ifndef _plan9_
+#ifdef NO_MALLOC_H
+#include <stdlib.h>
+#else
 #include <malloc.h>
+#endif
 #include <stdio.h>		/*for NULL */
 #include <string.h>		/*for memcpy */
 #else
@@ -59,7 +63,7 @@
 getinstance (struct filteraction *a)
 {
   struct filter *f = createfilter (a);
-  struct intdata *i = calloc (sizeof (*i), 1);
+  struct intdata *i = calloc (1, sizeof (*i));
   f->data = i;
   f->name = "Interlace filter";
   return (f);
diff -ruN xaos-3.0/src/engine/itersmall.c xaos-3.0-patched/src/engine/itersmall.c
--- xaos-3.0/src/engine/itersmall.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/itersmall.c	Thu May 30 14:04:40 2002
@@ -1,5 +1,7 @@
 #ifndef _plan9_
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <config.h>
 #ifdef HAVE_ALLOCA_H
 #include <alloca.h>
diff -ruN xaos-3.0/src/engine/julia.c xaos-3.0-patched/src/engine/julia.c
--- xaos-3.0/src/engine/julia.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/julia.c	Thu May 30 14:04:40 2002
@@ -5,7 +5,9 @@
 #else
 #include <math.h>
 #include <string.h>
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <stdio.h>
 #include <stdlib.h>
 #endif
diff -ruN xaos-3.0/src/engine/palettef.c xaos-3.0-patched/src/engine/palettef.c
--- xaos-3.0/src/engine/palettef.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/palettef.c	Thu May 30 14:24:38 2002
@@ -1,5 +1,9 @@
 #ifndef _plan9_
+#ifdef NO_MALLOC_H
+#include <stdlib.h>
+#else
 #include <malloc.h>
+#endif
 #include <stdio.h>
 #else
 #include <u.h>
@@ -90,7 +94,7 @@
 getinstance (struct filteraction *a)
 {
   struct filter *f = createfilter (a);
-  struct palettedata *i = calloc (sizeof (*i), 1);
+  struct palettedata *i = calloc (1, sizeof (*i));
   i->active = 0;
   i->palette = createpalette (0, 256, C256, 0, 256, NULL, mysetcolor, NULL, NULL);
   f->childimage = NULL;
diff -ruN xaos-3.0/src/engine/rotate.c xaos-3.0-patched/src/engine/rotate.c
--- xaos-3.0/src/engine/rotate.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/rotate.c	Thu May 30 14:24:49 2002
@@ -9,7 +9,11 @@
 #include <limits.h>
 #include <archaccel.h>
 #include <math.h>
+#ifdef NO_MALLOC_H
+#include <stdlib.h>
+#else
 #include <malloc.h>
+#endif
 #else
 #include <u.h>
 #include <libc.h>
@@ -74,7 +78,7 @@
 getinstance (struct filteraction *a)
 {
   struct filter *f = createfilter (a);
-  struct rotatedata *i = calloc (sizeof (*i), 1);
+  struct rotatedata *i = calloc (1, sizeof (*i));
   f->name = "Rotation filter";
   f->data = i;
   return (f);
diff -ruN xaos-3.0/src/engine/star.c xaos-3.0-patched/src/engine/star.c
--- xaos-3.0/src/engine/star.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/star.c	Thu May 30 14:04:40 2002
@@ -1,5 +1,7 @@
 #ifndef _plan9_
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <config.h>
 #include <limits.h>
 #ifdef HAVE_ALLOCA_H
diff -ruN xaos-3.0/src/engine/stereogram.c xaos-3.0-patched/src/engine/stereogram.c
--- xaos-3.0/src/engine/stereogram.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/stereogram.c	Thu May 30 14:04:41 2002
@@ -1,5 +1,7 @@
 #ifndef _plan9_
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <config.h>
 #ifdef HAVE_ALLOCA_H
 #include <alloca.h>
diff -ruN xaos-3.0/src/engine/subwindow.c xaos-3.0-patched/src/engine/subwindow.c
--- xaos-3.0/src/engine/subwindow.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/subwindow.c	Thu May 30 14:25:13 2002
@@ -1,5 +1,9 @@
 #ifndef _plan9_
+#ifdef NO_MALLOC_H
+#include <stdlib.h>
+#else
 #include <malloc.h>
+#endif
 #include <stdio.h>		/*for NULL */
 #include <string.h>		/*for memcpy */
 #else
@@ -111,7 +115,7 @@
 getinstance (struct filteraction *a)
 {
   struct filter *f = createfilter (a);
-  struct subdata *s = calloc (sizeof (*s), 1);
+  struct subdata *s = calloc (1, sizeof (*s));
   f->name = "Subwindow";
   f->data = s;
   s->second = NULL;
diff -ruN xaos-3.0/src/engine/zoom.c xaos-3.0-patched/src/engine/zoom.c
--- xaos-3.0/src/engine/zoom.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/engine/zoom.c	Thu May 30 14:04:41 2002
@@ -28,7 +28,9 @@
 #include <stdlib.h>
 #include <stdio.h>
 #ifndef _MAC
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #endif
 #ifdef __EMX__
 #include <float.h>
diff -ruN xaos-3.0/src/filter/filter.c xaos-3.0-patched/src/filter/filter.c
--- xaos-3.0/src/filter/filter.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/filter/filter.c	Thu May 30 14:23:14 2002
@@ -1,5 +1,9 @@
 #ifndef _plan9_
+#ifdef NO_MALLOC_H
+#include <stdlib.h>
+#else
 #include <malloc.h>
+#endif
 #else
 #include <u.h>
 #include <libc.h>
@@ -10,7 +14,7 @@
 struct queue *
 create_queue (struct filter *f)
 {
-  struct queue *q = calloc (sizeof (struct queue), 1);
+  struct queue *q = calloc (1, sizeof (struct queue));
   q->first = q->last = f;
   f->queue = q;
   f->next = f->previous = NULL;
diff -ruN xaos-3.0/src/filter/image.c xaos-3.0-patched/src/filter/image.c
--- xaos-3.0/src/filter/image.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/filter/image.c	Thu May 30 14:25:43 2002
@@ -1,8 +1,12 @@
 #ifndef _plan9_
-#include <malloc.h>
 #include <string.h>
 #include <fconfig.h>
 #include <assert.h>
+#ifdef NO_MALLOC_H
+#include <stdlib.h>
+#else
+#include <malloc.h>
+#endif
 #else
 #include <u.h>
 #include <libc.h>
@@ -55,7 +59,7 @@
 {
   int i;
   static int version = 1;
-  struct image *img = calloc (sizeof (*img), 1);
+  struct image *img = calloc (1, sizeof (*img));
   if (img == NULL)
     return NULL;
   if (flip == NULL)
diff -ruN xaos-3.0/src/filter/palette.c xaos-3.0-patched/src/filter/palette.c
--- xaos-3.0/src/filter/palette.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/filter/palette.c	Thu May 30 14:04:41 2002
@@ -10,7 +10,9 @@
 #include <config.h>
 #include <assert.h>
 #include <stdio.h>
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <limits.h>
 #endif
 #include <fconfig.h>
diff -ruN xaos-3.0/src/ui/play.c xaos-3.0-patched/src/ui/play.c
--- xaos-3.0/src/ui/play.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/ui/play.c	Thu May 30 14:26:27 2002
@@ -1,6 +1,10 @@
 #ifndef _plan9_
 #include <limits.h>
+#ifdef NO_MALLOC_H
+#include <stdlib.h>
+#else
 #include <malloc.h>
+#endif
 #include <string.h>
 #else
 #include <u.h>
diff -ruN xaos-3.0/src/ui/save.c xaos-3.0-patched/src/ui/save.c
--- xaos-3.0/src/ui/save.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/ui/save.c	Thu May 30 14:04:42 2002
@@ -2,7 +2,9 @@
 #include <limits.h>
 #include <string.h>
 #include <errno.h>
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #else
 #include <u.h>
 #include <libc.h>
diff -ruN xaos-3.0/src/ui/ui-drv/curses/mktable.c xaos-3.0-patched/src/ui/ui-drv/curses/mktable.c
--- xaos-3.0/src/ui/ui-drv/curses/mktable.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/ui/ui-drv/curses/mktable.c	Thu May 30 14:04:43 2002
@@ -23,7 +23,9 @@
 #include <stdio.h>
 #include <ctype.h>
 #include <stdlib.h>
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include "mktable.h"
 #include <config.h>
 #ifdef CURSES_DRIVER
diff -ruN xaos-3.0/src/ui/ui-drv/curses/ui_curses.c xaos-3.0-patched/src/ui/ui-drv/curses/ui_curses.c
--- xaos-3.0/src/ui/ui-drv/curses/ui_curses.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/ui/ui-drv/curses/ui_curses.c	Thu May 30 14:04:43 2002
@@ -23,7 +23,9 @@
 #ifdef CURSES_DRIVER
 #include <stdio.h>
 #include <stdlib.h>
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <signal.h>
 #include <sys/stat.h>
 #include <unistd.h>
diff -ruN xaos-3.0/src/ui/ui-drv/x11/xlib.c xaos-3.0-patched/src/ui/ui-drv/x11/xlib.c
--- xaos-3.0/src/ui/ui-drv/x11/xlib.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/ui/ui-drv/x11/xlib.c	Thu May 30 14:04:44 2002
@@ -26,7 +26,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include "xlib.h"
 #ifdef AMIGA
 #define XFlush(x) while(0)
diff -ruN xaos-3.0/src/ui/ui_helper.c xaos-3.0-patched/src/ui/ui_helper.c
--- xaos-3.0/src/ui/ui_helper.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/ui/ui_helper.c	Thu May 30 14:04:44 2002
@@ -1,5 +1,7 @@
 #ifndef _plan9_
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <ctype.h>
 #include <stdlib.h>
 #include <limits.h>
diff -ruN xaos-3.0/src/util/catalog.c xaos-3.0-patched/src/util/catalog.c
--- xaos-3.0/src/util/catalog.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/util/catalog.c	Thu May 30 14:04:44 2002
@@ -1,6 +1,8 @@
 #ifndef _plan9_
 #include <string.h>
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <string.h>
 #else
 #include <u.h>
diff -ruN xaos-3.0/src/util/png.c xaos-3.0-patched/src/util/png.c
--- xaos-3.0/src/util/png.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/util/png.c	Thu May 30 14:04:44 2002
@@ -5,7 +5,9 @@
 #endif
 #include <stdlib.h>
 #include <stdio.h>
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <errno.h>
 #else
 #include <u.h>
diff -ruN xaos-3.0/src/util/timers.c xaos-3.0-patched/src/util/timers.c
--- xaos-3.0/src/util/timers.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/util/timers.c	Thu May 30 14:04:44 2002
@@ -27,11 +27,15 @@
 #include <libc.h>
 #else
 #include <aconfig.h>
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #include <stdio.h>
 #include <unistd.h>
 #ifndef _MAC
+#ifndef NO_MALLOC_H
 #include <malloc.h>
+#endif
 #endif
 #ifdef __BEOS__
 #include <OS.h>
diff -ruN xaos-3.0/src/util/xstdio.c xaos-3.0-patched/src/util/xstdio.c
--- xaos-3.0/src/util/xstdio.c	Wed Mar  4 13:49:12 1998
+++ xaos-3.0-patched/src/util/xstdio.c	Thu May 30 13:55:42 2002
@@ -1,6 +1,6 @@
 #ifndef _plan9_
 #include <string.h>
-#ifdef __EMX__
+#if defined(__EMX__) || defined(__APPLE__)
 #include <sys/types.h>
 #endif
 #include <dirent.h>
--- xaos-3.0/Makefile.in.orig	2003-01-28 16:11:43.000000000 -0800
+++ xaos-3.0/Makefile.in	2003-01-28 16:12:19.000000000 -0800
@@ -28,20 +28,20 @@
 	rm -vf `find . -size 0`
 
 install: 
-	@INSTALL@ -d $(datadir)/XaoS
-	@INSTALL@ -d $(datadir)/XaoS/tutorial
-	@INSTALL@ -d $(datadir)/XaoS/examples
-	@INSTALL@ -d $(datadir)/XaoS/catalogs
-	@INSTALL@ -d $(datadir)/XaoS/doc
-	@INSTALL@ -d $(bindir)
-	@INSTALL@ -d $(mandir)/man6
-	@INSTALL@ -s bin/xaos $(bindir)
-	@INSTALL@ -m 444 tutorial/*.x[ah]f $(datadir)/XaoS/tutorial
-	@INSTALL@ -m 444 examples/* $(datadir)/XaoS/examples
-	@INSTALL@ -m 444 catalogs/* $(datadir)/XaoS/catalogs
-	@INSTALL@ -m 444 doc/README doc/README.bugs doc/compilers.txt doc/ANNOUNCE doc/PROBLEMS doc/tutorial.txt $(datadir)/XaoS/doc
-	@INSTALL@ -m 444 doc/xaos.6 $(mandir)/man6
+	@INSTALL@ -d $(DESTDIR)$(datadir)/XaoS
+	@INSTALL@ -d $(DESTDIR)$(datadir)/XaoS/tutorial
+	@INSTALL@ -d $(DESTDIR)$(datadir)/XaoS/examples
+	@INSTALL@ -d $(DESTDIR)$(datadir)/XaoS/catalogs
+	@INSTALL@ -d $(DESTDIR)$(datadir)/XaoS/doc
+	@INSTALL@ -d $(DESTDIR)$(bindir)
+	@INSTALL@ -d $(DESTDIR)$(mandir)/man6
+	@INSTALL@ -s bin/xaos $(DESTDIR)$(bindir)
+	@INSTALL@ -m 444 tutorial/*.x[ah]f $(DESTDIR)$(datadir)/XaoS/tutorial
+	@INSTALL@ -m 444 examples/* $(DESTDIR)$(datadir)/XaoS/examples
+	@INSTALL@ -m 444 catalogs/* $(DESTDIR)$(datadir)/XaoS/catalogs
+	@INSTALL@ -m 444 doc/README doc/README.bugs doc/compilers.txt doc/ANNOUNCE doc/PROBLEMS doc/tutorial.txt $(DESTDIR)$(datadir)/XaoS/doc
+	@INSTALL@ -m 444 doc/xaos.6 $(DESTDIR)$(mandir)/man6
 	@STICKY@
-	@INSTALL@ -m 444 doc/xaos.info $(infodir)
+	@INSTALL@ -m 444 doc/xaos.info $(DESTDIR)$(infodir)
 	#install-info doc/xaos.info 
 
