--- xvidcore-0.9.1/build/generic/configure.orig	2003-03-26 23:10:13.000000000 -0800
+++ xvidcore-0.9.1/build/generic/configure	2003-03-26 23:10:59.000000000 -0800
@@ -3649,10 +3649,10 @@
 	SPECIFIC_LDFLAGS="-shared -Wl,--dll,--out-implib,\$@.a libxvidcore.def"
 	SPECIFIC_CFLAGS="-mno-cygwin"
 	;;
-     darwin*|raphsody*)
-	echo "$as_me:$LINENO: result: -dynamiclib -flat_namespace" >&5
-echo "${ECHO_T}-dynamiclib -flat_namespace" >&6
-	SPECIFIC_LDFLAGS="-dynamiclib -flat_namespace"
+     darwin*)
+	echo "$as_me:$LINENO: result:  -compatibility_version 1 -current_version 1 -install_name ${prefix}/lib/libxvidcore.1.dylib" >&5
+echo "${ECHO_T} -compatibility_version 1 -current_version 1 -install_name ${prefix}/lib/libxvidcore.1.dylib" >&6
+	SPECIFIC_LDFLAGS="-dynamiclib -flat_namespace -compatibility_version 1 -current_version 1 -install_name ${prefix}/lib/libxvidcore.1.dylib"
 	SPECIFIC_CFLAGS="-fPIC -fno-common"
 	;;
      beos)
