diff -Nurd kdeadmin-4.4.1/CMakeLists.txt kdeadmin-4.4.1-new/CMakeLists.txt
--- kdeadmin-4.4.1/CMakeLists.txt	2010-02-26 13:30:57.000000000 -0500
+++ kdeadmin-4.4.1-new/CMakeLists.txt	2010-03-06 09:11:55.000000000 -0500
@@ -14,14 +14,11 @@
 if ( UNIX )
   macro_optional_add_subdirectory(kuser)
   macro_optional_add_subdirectory(kcron)
+  macro_optional_add_subdirectory(knetworkconf)
   macro_optional_add_subdirectory(ksystemlog)
   macro_optional_add_subdirectory(system-config-printer-kde)
 endif( UNIX )
 
-if(Q_WS_X11)
-   macro_optional_add_subdirectory(knetworkconf)
-endif(Q_WS_X11)
-
 macro_optional_add_subdirectory(doc)
 
 macro_display_feature_log()
diff -Nurd kdeadmin-4.4.1/doc/CMakeLists.txt kdeadmin-4.4.1-new/doc/CMakeLists.txt
--- kdeadmin-4.4.1/doc/CMakeLists.txt	2010-02-26 13:30:57.000000000 -0500
+++ kdeadmin-4.4.1-new/doc/CMakeLists.txt	2010-03-06 09:12:14.000000000 -0500
@@ -9,9 +9,6 @@
   if(BUILD_ksystemlog)
     add_subdirectory(ksystemlog)
   endif()
+  add_subdirectory(kcontrol)
 endif( UNIX )
 
-if(Q_WS_X11)
-   add_subdirectory(kcontrol)
-endif(Q_WS_X11)
-
