diff -Nurd kdeadmin-4.2.1/CMakeLists.txt kdeadmin-4.2.1-new/CMakeLists.txt
--- kdeadmin-4.2.1/CMakeLists.txt	2008-12-04 03:35:51.000000000 -0500
+++ kdeadmin-4.2.1-new/CMakeLists.txt	2009-03-10 09:00:36.000000000 -0400
@@ -17,14 +17,12 @@
   macro_optional_add_subdirectory(kcron)
   macro_optional_add_subdirectory(ksystemlog)
   macro_optional_add_subdirectory(system-config-printer-kde)
+  macro_optional_add_subdirectory(kpackage)
+  macro_optional_add_subdirectory(lilo-config)
+  macro_optional_add_subdirectory(knetworkconf)
 endif( UNIX )
 
 #macro_optional_add_subdirectory(ksysv)
-if(Q_WS_X11)
-   macro_optional_add_subdirectory(kpackage)
-   macro_optional_add_subdirectory(lilo-config)
-   macro_optional_add_subdirectory(knetworkconf)
-endif(Q_WS_X11)
 MESSAGE(STATUS "readd kdat-ksysv when porting")
 
 macro_optional_add_subdirectory(doc)
diff -Nurd kdeadmin-4.2.1/doc/CMakeLists.txt kdeadmin-4.2.1-new/doc/CMakeLists.txt
--- kdeadmin-4.2.1/doc/CMakeLists.txt	2008-09-13 04:16:28.000000000 -0400
+++ kdeadmin-4.2.1-new/doc/CMakeLists.txt	2009-03-10 09:49:59.000000000 -0400
@@ -8,13 +8,13 @@
   add_subdirectory(kuser)
   add_subdirectory(kcron)
   add_subdirectory(ksystemlog)
+  add_subdirectory(kpackage)
+  add_subdirectory(kcontrol)
 endif( UNIX )
 
 if(Q_WS_X11)
-   add_subdirectory(kpackage)
    if(LILO_EXECUTABLE)
       add_subdirectory(lilo-config)
    endif(LILO_EXECUTABLE)
-   add_subdirectory(kcontrol)
 endif(Q_WS_X11)
 
