diff -Nurd kdebase-runtime-4.2.2/kcontrol/componentchooser/CMakeLists.txt kdebase-runtime-4.2.2-new/kcontrol/componentchooser/CMakeLists.txt
--- kdebase-runtime-4.2.2/kcontrol/componentchooser/CMakeLists.txt	2008-09-17 16:54:29.000000000 -0400
+++ kdebase-runtime-4.2.2-new/kcontrol/componentchooser/CMakeLists.txt	2009-03-27 11:34:48.000000000 -0400
@@ -19,10 +19,10 @@
     componentchooserterminal.cpp ktimerdialog.cpp)
 endif(NOT WIN32)
 
-if(NOT APPLE AND NOT WIN32)
+if(Q_WS_X11)
   set(kcm_componentchooser_SRCS ${kcm_componentchooser_SRCS}
      componentchooserwm.cpp )
-endif(NOT APPLE AND NOT WIN32)
+endif(Q_WS_X11)
 
 
 kde4_add_ui_files(kcm_componentchooser_SRCS 
diff -Nurd kdebase-runtime-4.2.2/kioslave/thumbnail/CMakeLists.txt kdebase-runtime-4.2.2-new/kioslave/thumbnail/CMakeLists.txt
--- kdebase-runtime-4.2.2/kioslave/thumbnail/CMakeLists.txt	2009-01-17 05:59:27.000000000 -0500
+++ kdebase-runtime-4.2.2-new/kioslave/thumbnail/CMakeLists.txt	2009-03-27 11:34:48.000000000 -0400
@@ -86,7 +86,7 @@
 
    kde4_add_plugin(exrthumbnail ${exrthumbnail_PART_SRCS})
 
-   target_link_libraries(exrthumbnail ${KDE4_KIO_LIBS} ${OPENEXR_LIBRARIES})
+   target_link_libraries(exrthumbnail ${KDE4_KIO_LIBS} ${OPENEXR_LIBRARIES} z)
 
    install(TARGETS exrthumbnail DESTINATION ${PLUGIN_INSTALL_DIR})
    install( FILES exrthumbnail.desktop DESTINATION ${SERVICES_INSTALL_DIR})
diff -Nurd kdebase-runtime-4.2.2/nepomuk/kioslaves/search/querymetatype.h kdebase-runtime-4.2.2-new/nepomuk/kioslaves/search/querymetatype.h
--- kdebase-runtime-4.2.2/nepomuk/kioslaves/search/querymetatype.h	2008-09-26 10:55:58.000000000 -0400
+++ kdebase-runtime-4.2.2-new/nepomuk/kioslaves/search/querymetatype.h	2009-03-27 11:38:24.000000000 -0400
@@ -5,4 +5,7 @@
 
 #include "query.h"
 
+#ifndef _KDE_FINAL_Nepomuk_Search_Query
+#define _KDE_FINAL_Nepomuk_Search_Query
 Q_DECLARE_METATYPE(Nepomuk::Search::Query)
+#endif
diff -Nurd kdebase-runtime-4.2.2/nepomuk/kioslaves/search/searchfolder.cpp kdebase-runtime-4.2.2-new/nepomuk/kioslaves/search/searchfolder.cpp
--- kdebase-runtime-4.2.2/nepomuk/kioslaves/search/searchfolder.cpp	2008-11-28 10:33:43.000000000 -0500
+++ kdebase-runtime-4.2.2-new/nepomuk/kioslaves/search/searchfolder.cpp	2009-03-27 11:36:15.000000000 -0400
@@ -33,12 +33,7 @@
 #include <KIO/NetAccess>
 #include <KUser>
 #include <kdirnotify.h>
-
-uint qHash( const QUrl& url )
-{
-    return qHash( url.toString() );
-}
-
+#include <qurlhash.h>
 
 namespace {
     QString addCounterToFileName( const QString& name, int i )
diff -Nurd kdebase-runtime-4.2.2/nepomuk/libnepomukquery/dbusoperators.cpp kdebase-runtime-4.2.2-new/nepomuk/libnepomukquery/dbusoperators.cpp
--- kdebase-runtime-4.2.2/nepomuk/libnepomukquery/dbusoperators.cpp	2008-11-28 10:33:44.000000000 -0500
+++ kdebase-runtime-4.2.2-new/nepomuk/libnepomukquery/dbusoperators.cpp	2009-03-27 11:47:03.000000000 -0400
@@ -23,9 +23,14 @@
 
 Q_DECLARE_METATYPE(Nepomuk::Search::Result)
 Q_DECLARE_METATYPE(Nepomuk::Search::Term)
+#ifndef _KDE_FINAL_Nepomuk_Search_Query
+#define _KDE_FINAL_Nepomuk_Search_Query
 Q_DECLARE_METATYPE(Nepomuk::Search::Query)
+#endif
 Q_DECLARE_METATYPE(Soprano::Node)
+#ifndef _NEPOMUK_VARIANT_H_
 Q_DECLARE_METATYPE(QList<int>)
+#endif
 Q_DECLARE_METATYPE(QList<Nepomuk::Search::Result>)
 
 
diff -Nurd kdebase-runtime-4.2.2/phonon/platform_kde/devicelisting.cpp kdebase-runtime-4.2.2-new/phonon/platform_kde/devicelisting.cpp
--- kdebase-runtime-4.2.2/phonon/platform_kde/devicelisting.cpp	2008-10-17 07:00:15.000000000 -0400
+++ kdebase-runtime-4.2.2-new/phonon/platform_kde/devicelisting.cpp	2009-03-27 12:02:31.000000000 -0400
@@ -35,7 +35,10 @@
 #endif // HAVE_ALSA_ASOUNDLIB_H
 
 typedef QList<QPair<QByteArray, QString> > PhononDeviceAccessList;
+#ifndef _KDE_FINAL_PhononDeviceAccessList
+#define _KDE_FINAL_PhononDeviceAccessList
 Q_DECLARE_METATYPE(PhononDeviceAccessList)
+#endif
 
 static void installAlsaPhononDeviceHandle()
 {
diff -Nurd kdebase-runtime-4.2.2/phonon/platform_kde/kdeplatformplugin.cpp kdebase-runtime-4.2.2-new/phonon/platform_kde/kdeplatformplugin.cpp
--- kdebase-runtime-4.2.2/phonon/platform_kde/kdeplatformplugin.cpp	2008-11-28 10:33:42.000000000 -0500
+++ kdebase-runtime-4.2.2-new/phonon/platform_kde/kdeplatformplugin.cpp	2009-03-27 12:02:31.000000000 -0400
@@ -43,7 +43,10 @@
 
 typedef QPair<QByteArray, QString> PhononDeviceAccess;
 typedef QList<PhononDeviceAccess> PhononDeviceAccessList;
+#ifndef _KDE_FINAL_PhononDeviceAccessList
+#define _KDE_FINAL_PhononDeviceAccessList
 Q_DECLARE_METATYPE(PhononDeviceAccessList)
+#endif
 
 namespace Phonon
 {
