diff -Nurd kdebase-runtime-4.2.4/kcontrol/componentchooser/CMakeLists.txt kdebase-runtime-4.2.4-new/kcontrol/componentchooser/CMakeLists.txt
--- kdebase-runtime-4.2.4/kcontrol/componentchooser/CMakeLists.txt	2008-09-17 16:54:29.000000000 -0400
+++ kdebase-runtime-4.2.4-new/kcontrol/componentchooser/CMakeLists.txt	2009-06-09 17:01:51.000000000 -0400
@@ -19,10 +19,10 @@
     componentchooserterminal.cpp ktimerdialog.cpp)
 endif(NOT WIN32)
 
-if(NOT APPLE AND NOT WIN32)
+if(Q_WS_X11)
   set(kcm_componentchooser_SRCS ${kcm_componentchooser_SRCS}
      componentchooserwm.cpp )
-endif(NOT APPLE AND NOT WIN32)
+endif(Q_WS_X11)
 
 
 kde4_add_ui_files(kcm_componentchooser_SRCS 
diff -Nurd kdebase-runtime-4.2.4/kioslave/thumbnail/CMakeLists.txt kdebase-runtime-4.2.4-new/kioslave/thumbnail/CMakeLists.txt
--- kdebase-runtime-4.2.4/kioslave/thumbnail/CMakeLists.txt	2009-01-17 05:59:27.000000000 -0500
+++ kdebase-runtime-4.2.4-new/kioslave/thumbnail/CMakeLists.txt	2009-06-09 17:01:51.000000000 -0400
@@ -86,7 +86,7 @@
 
    kde4_add_plugin(exrthumbnail ${exrthumbnail_PART_SRCS})
 
-   target_link_libraries(exrthumbnail ${KDE4_KIO_LIBS} ${OPENEXR_LIBRARIES})
+   target_link_libraries(exrthumbnail ${KDE4_KIO_LIBS} ${OPENEXR_LIBRARIES} z)
 
    install(TARGETS exrthumbnail DESTINATION ${PLUGIN_INSTALL_DIR})
    install( FILES exrthumbnail.desktop DESTINATION ${SERVICES_INSTALL_DIR})
diff -Nurd kdebase-runtime-4.2.4/knotify/CMakeLists.txt kdebase-runtime-4.2.4-new/knotify/CMakeLists.txt
--- kdebase-runtime-4.2.4/knotify/CMakeLists.txt	2008-05-29 07:18:43.000000000 -0400
+++ kdebase-runtime-4.2.4-new/knotify/CMakeLists.txt	2009-06-09 17:02:50.000000000 -0400
@@ -17,6 +17,12 @@
 
 kde4_add_executable( knotify ${knotify_SRCS})
 
+if (Q_WS_MAC)
+    set_target_properties(knotify PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist.template)
+    set_target_properties(knotify PROPERTIES MACOSX_BUNDLE_GUI_IDENTIFIER "org.kde.knotify4")
+    set_target_properties(knotify PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "KDE Notify")
+endif (Q_WS_MAC)
+
 target_link_libraries( knotify ${KDE4_KDEUI_LIBS} ${KDE4_PHONON_LIBS})
 
 set_target_properties(knotify PROPERTIES OUTPUT_NAME knotify4 )
diff -Nurd kdebase-runtime-4.2.4/knotify/Info.plist.template kdebase-runtime-4.2.4-new/knotify/Info.plist.template
--- kdebase-runtime-4.2.4/knotify/Info.plist.template	1969-12-31 19:00:00.000000000 -0500
+++ kdebase-runtime-4.2.4-new/knotify/Info.plist.template	2009-06-09 17:02:04.000000000 -0400
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+    <key>CFBundleDevelopmentRegion</key>
+    <string>English</string>
+    <key>CFBundleExecutable</key>
+    <string>${MACOSX_BUNDLE_EXECUTABLE_NAME}</string>
+    <key>CFBundleGetInfoString</key>
+    <string>${MACOSX_BUNDLE_INFO_STRING}</string>
+    <key>CFBundleIconFile</key>
+    <string>${MACOSX_BUNDLE_ICON_FILE}</string>
+    <key>CFBundleIdentifier</key>
+    <string>${MACOSX_BUNDLE_GUI_IDENTIFIER}</string>
+    <key>CFBundleInfoDictionaryVersion</key>
+    <string>6.0</string>
+    <key>CFBundleLongVersionString</key>
+    <string>${MACOSX_BUNDLE_LONG_VERSION_STRING}</string>
+    <key>CFBundleName</key>
+    <string>${MACOSX_BUNDLE_BUNDLE_NAME}</string>
+    <key>CFBundlePackageType</key>
+    <string>APPL</string>
+    <key>CFBundleShortVersionString</key>
+    <string>${MACOSX_BUNDLE_SHORT_VERSION_STRING}</string>
+    <key>CFBundleVersion</key>
+    <string>${MACOSX_BUNDLE_BUNDLE_VERSION}</string>
+    <key>CSResourcesFileMapped</key>
+    <true/>
+    <key>LSRequiresCarbon</key>
+    <true/>
+    <key>LSUIElement</key>
+    <string>1</string>
+    <key>NSHumanReadableCopyright</key>
+    <string>${MACOSX_BUNDLE_COPYRIGHT}</string>
+</dict>
+</plist>
diff -Nurd kdebase-runtime-4.2.4/nepomuk/kioslaves/search/CMakeLists.txt kdebase-runtime-4.2.4-new/nepomuk/kioslaves/search/CMakeLists.txt
--- kdebase-runtime-4.2.4/nepomuk/kioslaves/search/CMakeLists.txt	2008-09-26 10:55:58.000000000 -0400
+++ kdebase-runtime-4.2.4-new/nepomuk/kioslaves/search/CMakeLists.txt	2009-06-09 17:01:51.000000000 -0400
@@ -4,6 +4,7 @@
   ${QT_INCLUDES}
   ${KDE4_INCLUDES}
   ${KDE4_KIO_INCLUDES}
+  ${SOPRANO_INCLUDE_DIR}
   ${NEPOMUK_INCLUDE_DIR}
   ${nepomuk_SOURCE_DIR}/libnepomukquery
 )
diff -Nurd kdebase-runtime-4.2.4/nepomuk/kioslaves/search/querymetatype.h kdebase-runtime-4.2.4-new/nepomuk/kioslaves/search/querymetatype.h
--- kdebase-runtime-4.2.4/nepomuk/kioslaves/search/querymetatype.h	2008-09-26 10:55:58.000000000 -0400
+++ kdebase-runtime-4.2.4-new/nepomuk/kioslaves/search/querymetatype.h	2009-06-09 17:01:51.000000000 -0400
@@ -5,4 +5,7 @@
 
 #include "query.h"
 
+#ifndef _KDE_FINAL_Nepomuk_Search_Query
+#define _KDE_FINAL_Nepomuk_Search_Query
 Q_DECLARE_METATYPE(Nepomuk::Search::Query)
+#endif
diff -Nurd kdebase-runtime-4.2.4/nepomuk/kioslaves/search/searchfolder.cpp kdebase-runtime-4.2.4-new/nepomuk/kioslaves/search/searchfolder.cpp
--- kdebase-runtime-4.2.4/nepomuk/kioslaves/search/searchfolder.cpp	2008-11-28 10:33:43.000000000 -0500
+++ kdebase-runtime-4.2.4-new/nepomuk/kioslaves/search/searchfolder.cpp	2009-06-09 17:01:51.000000000 -0400
@@ -33,12 +33,7 @@
 #include <KIO/NetAccess>
 #include <KUser>
 #include <kdirnotify.h>
-
-uint qHash( const QUrl& url )
-{
-    return qHash( url.toString() );
-}
-
+#include <qurlhash.h>
 
 namespace {
     QString addCounterToFileName( const QString& name, int i )
diff -Nurd kdebase-runtime-4.2.4/nepomuk/libnepomukquery/dbusoperators.cpp kdebase-runtime-4.2.4-new/nepomuk/libnepomukquery/dbusoperators.cpp
--- kdebase-runtime-4.2.4/nepomuk/libnepomukquery/dbusoperators.cpp	2008-11-28 10:33:44.000000000 -0500
+++ kdebase-runtime-4.2.4-new/nepomuk/libnepomukquery/dbusoperators.cpp	2009-06-09 17:01:51.000000000 -0400
@@ -23,9 +23,14 @@
 
 Q_DECLARE_METATYPE(Nepomuk::Search::Result)
 Q_DECLARE_METATYPE(Nepomuk::Search::Term)
+#ifndef _KDE_FINAL_Nepomuk_Search_Query
+#define _KDE_FINAL_Nepomuk_Search_Query
 Q_DECLARE_METATYPE(Nepomuk::Search::Query)
+#endif
 Q_DECLARE_METATYPE(Soprano::Node)
+#ifndef _NEPOMUK_VARIANT_H_
 Q_DECLARE_METATYPE(QList<int>)
+#endif
 Q_DECLARE_METATYPE(QList<Nepomuk::Search::Result>)
 
 
diff -Nurd kdebase-runtime-4.2.4/nepomuk/services/queryservice/CMakeLists.txt kdebase-runtime-4.2.4-new/nepomuk/services/queryservice/CMakeLists.txt
--- kdebase-runtime-4.2.4/nepomuk/services/queryservice/CMakeLists.txt	2008-09-26 10:55:58.000000000 -0400
+++ kdebase-runtime-4.2.4-new/nepomuk/services/queryservice/CMakeLists.txt	2009-06-09 17:01:51.000000000 -0400
@@ -3,6 +3,7 @@
 include_directories(
   ${QT_INCLUDES}
   ${KDE4_INCLUDES}
+  ${SOPRANO_INCLUDE_DIR}
   ${NEPOMUK_INCLUDES}
   ${nepomuk_SOURCE_DIR}/libnepomukquery
 )
diff -Nurd kdebase-runtime-4.2.4/phonon/platform_kde/devicelisting.cpp kdebase-runtime-4.2.4-new/phonon/platform_kde/devicelisting.cpp
--- kdebase-runtime-4.2.4/phonon/platform_kde/devicelisting.cpp	2008-10-17 07:00:15.000000000 -0400
+++ kdebase-runtime-4.2.4-new/phonon/platform_kde/devicelisting.cpp	2009-06-09 17:01:51.000000000 -0400
@@ -35,7 +35,10 @@
 #endif // HAVE_ALSA_ASOUNDLIB_H
 
 typedef QList<QPair<QByteArray, QString> > PhononDeviceAccessList;
+#ifndef _KDE_FINAL_PhononDeviceAccessList
+#define _KDE_FINAL_PhononDeviceAccessList
 Q_DECLARE_METATYPE(PhononDeviceAccessList)
+#endif
 
 static void installAlsaPhononDeviceHandle()
 {
diff -Nurd kdebase-runtime-4.2.4/phonon/platform_kde/kdeplatformplugin.cpp kdebase-runtime-4.2.4-new/phonon/platform_kde/kdeplatformplugin.cpp
--- kdebase-runtime-4.2.4/phonon/platform_kde/kdeplatformplugin.cpp	2008-11-28 10:33:42.000000000 -0500
+++ kdebase-runtime-4.2.4-new/phonon/platform_kde/kdeplatformplugin.cpp	2009-06-09 17:01:51.000000000 -0400
@@ -43,7 +43,10 @@
 
 typedef QPair<QByteArray, QString> PhononDeviceAccess;
 typedef QList<PhononDeviceAccess> PhononDeviceAccessList;
+#ifndef _KDE_FINAL_PhononDeviceAccessList
+#define _KDE_FINAL_PhononDeviceAccessList
 Q_DECLARE_METATYPE(PhononDeviceAccessList)
+#endif
 
 namespace Phonon
 {
