diff -Nurd kdepimlibs-4.2.2/CMakeLists.txt kdepimlibs-4.2.2-new/CMakeLists.txt
--- kdepimlibs-4.2.2/CMakeLists.txt	2009-03-26 10:42:06.000000000 -0400
+++ kdepimlibs-4.2.2-new/CMakeLists.txt	2009-03-27 09:16:54.000000000 -0400
@@ -25,23 +25,23 @@
 endif (NOT ONLY_KLEO)
 
 # gpgme is a hard dependency
-find_package(Gpgme REQUIRED)
+#find_package(Gpgme REQUIRED)
 
-include (gpgme++/ConfigureChecks.cmake)
+#include (gpgme++/ConfigureChecks.cmake)
 include (ConfigureChecks.cmake)
 
 add_definitions (${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
 include_directories (${CMAKE_SOURCE_DIR} ${CMAKE_BINARY_DIR} ${KDE4_INCLUDES})
 
 # TODO(move to kdesupport scripts) temporary way to support gpgme-qt under win32
-if (EXISTS ${CMAKE_SOURCE_DIR}/gpgme-qt)
-  message(STATUS "gpgme-qt for Windows will be compiled: ${CMAKE_SOURCE_DIR}/gpgme-qt")
-  add_subdirectory(gpgme-qt)
-endif (EXISTS ${CMAKE_SOURCE_DIR}/gpgme-qt)
+#if (EXISTS ${CMAKE_SOURCE_DIR}/gpgme-qt)
+#  message(STATUS "gpgme-qt for Windows will be compiled: ${CMAKE_SOURCE_DIR}/gpgme-qt")
+#  add_subdirectory(gpgme-qt)
+#endif (EXISTS ${CMAKE_SOURCE_DIR}/gpgme-qt)
 
 macro_optional_add_subdirectory(doc)
-add_subdirectory(gpgme++)
-add_subdirectory(qgpgme)
+#add_subdirectory(gpgme++)
+#add_subdirectory(qgpgme)
 add_subdirectory(kmime)
 
 if (NOT ONLY_KLEO)
