diff -Nurd kdeutils-4.2.2/CMakeLists.txt kdeutils-4.2.2-new/CMakeLists.txt
--- kdeutils-4.2.2/CMakeLists.txt	2008-12-21 15:18:58.000000000 -0500
+++ kdeutils-4.2.2-new/CMakeLists.txt	2009-03-30 15:53:50.000000000 -0400
@@ -22,10 +22,8 @@
 add_subdirectory( cmake )
 
 # --- check for dependencies ---
-if(X11_FOUND)
-  macro_optional_find_package( QImageBlitz )
-  macro_log_feature( QIMAGEBLITZ_FOUND "QImageBlitz" "An image effects library" "kdesupport" FALSE "" "Required for building SuperKaramba.")
-endif(X11_FOUND)
+macro_optional_find_package( QImageBlitz )
+macro_log_feature( QIMAGEBLITZ_FOUND "QImageBlitz" "An image effects library" "kdesupport" FALSE "" "Required for building SuperKaramba.")
 macro_optional_find_package( GMP )
 macro_log_feature( GMP_FOUND "GMP" "The GNU Multiple Precision Arithmetic Library" "http://www.swox.com/gmp" FALSE "" "Required for building KCalc.")
 
@@ -53,7 +51,7 @@
 macro_optional_add_subdirectory( okteta )
 macro_optional_add_subdirectory( sweeper )
 
-if( X11_FOUND AND QIMAGEBLITZ_FOUND)
+if( Q_WS_X11 AND QIMAGEBLITZ_FOUND )
     #find_file( BACKGROUND_INTERFACE_FILE_FOUND org.kde.kdesktop.Background.xml PATHS ${DBUS_INTERFACES_INSTALL_DIR} )
 
     #if( BACKGROUND_INTERFACE_FILE_FOUND )
@@ -61,7 +59,7 @@
     #else( BACKGROUND_INTERFACE_FILE_FOUND )
         #MESSAGE( STATUS "org.kde.kdesktop.Background.xml was not found install kdebase before please" )
     #endif( BACKGROUND_INTERFACE_FILE_FOUND )
-endif( X11_FOUND AND QIMAGEBLITZ_FOUND)
+endif( Q_WS_X11 AND QIMAGEBLITZ_FOUND )
 
 # --- show log ---
 macro_display_feature_log()
