diff -Naur kdirstat-2.4.3-0001/kdirstat/Makefile.am kdirstat-2.4.3/kdirstat/Makefile.am
--- kdirstat-2.4.3-0001/kdirstat/Makefile.am	2004-11-24 06:49:10.000000000 -0500
+++ kdirstat-2.4.3/kdirstat/Makefile.am	2010-09-01 09:58:01.000000000 -0400
@@ -58,7 +58,7 @@
 update_SCRIPTS	= fix_move_to_trash_bin.pl
 
 
-kdirstat_LDADD = $(LIB_KFILE)
+kdirstat_LDADD = $(LIB_KFILE) $(LIB_QT)
 
 KDE_ICON = kdirstat
 
diff -Naur kdirstat-2.4.3-0001/kdirstat/kdirtree.cpp kdirstat-2.4.3/kdirstat/kdirtree.cpp
--- kdirstat-2.4.3-0001/kdirstat/kdirtree.cpp	2004-11-23 07:29:02.000000000 -0500
+++ kdirstat-2.4.3/kdirstat/kdirtree.cpp	2005-06-01 08:48:17.000000000 -0400
@@ -71,8 +71,8 @@
     }
     else
     {
-	_size	 = statInfo->st_size;
 	_blocks	 = statInfo->st_blocks;
+	_size	 = _blocks * blockSize();
     }
     
 #if 0
diff -Naur kdirstat-2.4.3-0001/kdirstat/kdirtree.h kdirstat-2.4.3/kdirstat/kdirtree.h
--- kdirstat-2.4.3-0001/kdirstat/kdirtree.h	2003-08-26 12:35:12.000000000 -0400
+++ kdirstat-2.4.3/kdirstat/kdirtree.h	2005-06-01 08:48:43.000000000 -0400
@@ -213,7 +213,7 @@
 	 * The size of one single block that @ref blocks() returns.
 	 * Notice: This is _not_ the blocksize that lstat() returns!
 	 **/
-	KFileSize		blockSize()	const { return 512L;	}
+	inline KFileSize		blockSize()	const { return 512L;	}
 
 	/**
 	 * The modification time of the file (not the inode).
diff -Naur kdirstat-2.4.3-0001/kdirstat/ktreemapview.cpp kdirstat-2.4.3/kdirstat/ktreemapview.cpp
--- kdirstat-2.4.3-0001/kdirstat/ktreemapview.cpp	2003-10-20 10:17:31.000000000 -0400
+++ kdirstat-2.4.3/kdirstat/ktreemapview.cpp	2005-06-01 08:29:51.000000000 -0400
@@ -625,6 +625,16 @@
 		if ( ext == "hpp"	)	return Qt::blue;
 		if ( ext == "el"	)	return Qt::blue;
 
+		if ( lowerExt == "m"    )	return Qt::blue;
+		if ( lowerExt == "mm"   )	return Qt::blue;
+		if ( lowerExt == "nib"  )	return Qt::blue;
+		if ( lowerExt == "strings")	return Qt::blue;
+		if ( lowerExt == "py"   )	return Qt::blue;
+		if ( lowerExt == "pl"   )	return Qt::blue;
+		if ( lowerExt == "pm"   )	return Qt::blue;
+		if ( lowerExt == "java" )	return Qt::blue;
+		if ( lowerExt == "scpt" )	return Qt::blue;
+
 		if ( ext == "o"		)	return QColor( 0xff, 0xa0, 0x00 );
 		if ( ext == "lo"	)	return QColor( 0xff, 0xa0, 0x00 );
 		if ( ext == "Po"	)	return QColor( 0xff, 0xa0, 0x00 );
@@ -636,18 +646,42 @@
 		if ( ext == "a"		)	return QColor( 0xff, 0xa0, 0x00 );
 		if ( ext == "rpm"	)	return QColor( 0xff, 0xa0, 0x00 );
 
+		if ( lowerExt == "dylib")	return QColor( 0xff, 0xa0, 0x00 );
+		if ( lowerExt == "rsrc" )	return QColor( 0xff, 0xa0, 0x00 );
+		if ( lowerExt == "jar"  )	return QColor( 0xff, 0xa0, 0x00 );
+
 		if ( lowerExt == "tar.bz2" )	return Qt::green;
 		if ( lowerExt == "tar.gz"  )	return Qt::green;
 		if ( lowerExt == "tgz"	)	return Qt::green;
 		if ( lowerExt == "bz2"	)	return Qt::green;
 		if ( lowerExt == "bz"	)	return Qt::green;
 		if ( lowerExt == "gz"	)	return Qt::green;
+        
+        if ( lowerExt == "tbz2"	)	return Qt::green;
+        if ( lowerExt == "tzo"	)	return Qt::green;
+        if ( lowerExt == "lzo"	)	return Qt::green;
+        if ( lowerExt == "sit"	)	return Qt::green;
+        if ( lowerExt == "sitx"	)	return Qt::green;
+        if ( lowerExt == "deb"	)	return Qt::green;
+        if ( lowerExt == "rpm"	)	return Qt::green;
+        if ( lowerExt == "rar"	)	return Qt::green;
+		if ( lowerExt == "dmg"	)	return Qt::green;
+		if ( lowerExt == "img"	)	return Qt::green;
+		if ( lowerExt == "toast")	return Qt::green;
+		if ( lowerExt == "smi"  )	return Qt::green;
+		if ( lowerExt == "tar"  )	return Qt::green;
 
 		if ( lowerExt == "html"	)	return Qt::blue;
 		if ( lowerExt == "htm"	)	return Qt::blue;
 		if ( lowerExt == "txt"	)	return Qt::blue;
 		if ( lowerExt == "doc"	)	return Qt::blue;
 
+		if ( lowerExt == "rtf"	)	return Qt::blue;
+		if ( lowerExt == "plist")	return Qt::blue;
+		if ( lowerExt == "xml"  )	return Qt::blue;
+		if ( lowerExt == "info" )	return Qt::blue;
+		if ( lowerExt == "patch")	return Qt::blue;
+
 		if ( lowerExt == "png"	)	return Qt::cyan;
 		if ( lowerExt == "jpg"	)	return Qt::cyan;
 		if ( lowerExt == "jpeg"	)	return Qt::cyan;
@@ -658,14 +692,28 @@
 		if ( lowerExt == "xpm"	)	return Qt::cyan;
 		if ( lowerExt == "tga"	)	return Qt::cyan;
 
+		if ( lowerExt == "icns"	)	return Qt::cyan;
+		if ( lowerExt == "psd"	)	return Qt::cyan;
+
 		if ( lowerExt == "wav"	)	return Qt::yellow;
 		if ( lowerExt == "mp3"	)	return Qt::yellow;
 
+		if ( lowerExt == "ogg"	)	return Qt::yellow;
+		if ( lowerExt == "m4a"	)	return Qt::yellow;
+		if ( lowerExt == "m4b"	)	return Qt::yellow;
+		if ( lowerExt == "m4p"	)	return Qt::yellow;
+		if ( lowerExt == "aiff"	)	return Qt::yellow;
+		if ( lowerExt == "aif"	)	return Qt::yellow;
+		if ( lowerExt == "wma"	)	return Qt::yellow;
+
 		if ( lowerExt == "avi"	)	return QColor( 0xa0, 0xff, 0x00 );
 		if ( lowerExt == "mov"	)	return QColor( 0xa0, 0xff, 0x00 );
 		if ( lowerExt == "mpg"	)	return QColor( 0xa0, 0xff, 0x00 );
 		if ( lowerExt == "mpeg"	)	return QColor( 0xa0, 0xff, 0x00 );
 
+		if ( lowerExt == "mp4"	)	return QColor( 0xa0, 0xff, 0x00 );
+		if ( lowerExt == "wmv"	)	return QColor( 0xa0, 0xff, 0x00 );
+
 		if ( lowerExt == "pdf"	)	return Qt::blue;
 		if ( lowerExt == "ps"	)	return Qt::cyan;
 
