diff -Nurd kompose-0.5.4/src/komposetaskprefswidget.cpp kompose-0.5.4-new/src/komposetaskprefswidget.cpp
--- kompose-0.5.4/src/komposetaskprefswidget.cpp	2005-07-05 18:32:41.000000000 -0400
+++ kompose-0.5.4-new/src/komposetaskprefswidget.cpp	2008-06-12 14:33:56.000000000 -0400
@@ -28,7 +28,7 @@
 KomposeTaskPrefsWidget::KomposeTaskPrefsWidget(KomposeTaskWidget *parent, const char *name)
     : QDockArea(Qt::Horizontal , QDockArea::Normal, parent, name)
 {
-  KToolBar* pToolBar = new KToolBar( this, i18n("Task Actions") );
+  KToolBar* pToolBar = new KToolBar( this, i18n("Task Actions").latin1() );
   pToolBar->setLabel(i18n("Task"));
   pToolBar->setResizeEnabled ( false );
   pToolBar->setMovingEnabled ( false );
