diff -Nurd QtCurve-KDE3-0.59.5/config/CMakeLists.txt QtCurve-KDE3-0.59.5-new/config/CMakeLists.txt
--- QtCurve-KDE3-0.59.5/config/CMakeLists.txt	2008-07-30 15:18:18.000000000 -0400
+++ QtCurve-KDE3-0.59.5-new/config/CMakeLists.txt	2008-10-24 10:19:17.000000000 -0400
@@ -3,5 +3,5 @@
 kde3_add_ui_files(kstyle_qtcurve_config_PART_SRCS qtcurveconfigbase.ui)
 kde3_add_kpart(kstyle_qtcurve_config ${kstyle_qtcurve_config_PART_SRCS})
 kde3_automoc(${kstyle_qtcurve_config_PART_SRCS})
-target_link_libraries(kstyle_qtcurve_config ${KDE3_KDEUI_LIBS} ${KDE3_KIO_LIBS} ${KDE3_KDECORE_LIBS} ${QT_QTGUI_LIBRARY})
+target_link_libraries(kstyle_qtcurve_config kdeui kio ${KDE3_KDECORE_LIBRARY} ${QT_QT_LIBRARY})
 install(TARGETS kstyle_qtcurve_config DESTINATION ${KDE3_LIB_DIR}/kde3)
diff -Nurd QtCurve-KDE3-0.59.5/style/CMakeLists.txt QtCurve-KDE3-0.59.5-new/style/CMakeLists.txt
--- QtCurve-KDE3-0.59.5/style/CMakeLists.txt	2008-07-30 15:18:18.000000000 -0400
+++ QtCurve-KDE3-0.59.5-new/style/CMakeLists.txt	2008-10-24 10:18:34.000000000 -0400
@@ -5,6 +5,6 @@
 kde3_automoc(${qtcurve_SRCS})
 add_library(qtcurve SHARED ${qtcurve_SRCS})
 set_target_properties(qtcurve PROPERTIES PREFIX "")
-target_link_libraries(qtcurve ${QT_QTGUI_LIBRARY} kdefx)
+target_link_libraries(qtcurve ${QT_QT_LIBRARY} ${X11_X11_LIB} kdefx)
 install(TARGETS qtcurve LIBRARY DESTINATION ${KDE3_LIB_DIR}/kde3/plugins/styles)
 install(FILES qtcurve.themerc DESTINATION ${KDE3PREFIX}/share/apps/kstyle/themes)
