diff -Naur gpc-3.4.3d6-sources/gpc-3.4.4d2/gcc/configure gpc-3.4.3d6-sources.fink/gpc-3.4.4d2/gcc/configure
--- gpc-3.4.3d6-sources/gpc-3.4.4d2/gcc/configure	2005-01-08 10:20:19.000000000 +0900
+++ gpc-3.4.3d6-sources.fink/gpc-3.4.4d2/gcc/configure	2005-08-01 16:52:14.000000000 +0900
@@ -6368,7 +6368,7 @@
   # sources; use -no-cpp-precomp to get to GNU cpp.
   # Apple's GCC has bugs in designated initializer handling, so disable
   # that too.
-  stage1_cflags="-no-cpp-precomp -DHAVE_DESIGNATED_INITIALIZERS=0"
+  stage1_cflags="-DHAVE_DESIGNATED_INITIALIZERS=0"
   ;;
 esac
 
