diff -uNr ikvm/native/native.build ikvm-patched/native/native.build
--- ikvm/native/native.build	2004-08-30 17:56:26.000000000 -0400
+++ ikvm-patched/native/native.build	2005-06-05 21:18:15.000000000 -0400
@@ -9,7 +9,7 @@
         </if>
     </target>
     <target name="ikvm-native-unix">
-        <exec program="/bin/bash" commandline="-c 'gcc -o libikvm-native.so --shared -fPIC `pkg-config --cflags gmodule-2.0` jni.c os.c'"/>
+        <exec program="/bin/bash" commandline="-c 'gcc -o libikvm-native.so -bundle -fPIC `pkg-config --cflags gmodule-2.0; pkg-config --libs gmodule-2.0` jni.c os.c'"/>
         <copy file="libikvm-native.so" todir="../bin"/>
     </target>
     <target name="ikvm-native-win32">
diff -uNr ikvm/runtime/JniInterface.cs ikvm-patched/runtime/JniInterface.cs
--- ikvm/runtime/JniInterface.cs	2005-05-03 16:26:00.000000000 -0400
+++ ikvm-patched/runtime/JniInterface.cs	2005-06-05 21:31:53.000000000 -0400
@@ -903,12 +903,7 @@
 		{
 			JNI.jvmCreated = true;
 			pJavaVM = (JavaVM*)(void*)JniMem.Alloc(IntPtr.Size * (1 + vtableDelegates.Length));
-#if __MonoCS__ 
-		// MONOBUG mcs requires this bogus fixed construct (and Microsoft doesn't allow it)
-		fixed(void** p = &pJavaVM->firstVtableEntry) { pJavaVM->vtable = p; }
-#else
 			pJavaVM->vtable = &pJavaVM->firstVtableEntry;
-#endif
 			for(int i = 0; i < vtableDelegates.Length; i++)
 			{
 				pJavaVM->vtable[i] = JniHelper.ikvm_MarshalDelegate(vtableDelegates[i]);
