diff -ur verilog-0.8.bak/LineInfo.h verilog-0.8/LineInfo.h
--- verilog-0.8.bak/LineInfo.h	2003-01-16 21:49:03.000000000 -0800
+++ verilog-0.8/LineInfo.h	2005-06-13 12:10:18.000000000 -0700
@@ -24,6 +24,8 @@
 
 # include  <string>
 
+using namespace std;
+
 /*
  * This class holds line information for an internal object.
  *
diff -ur verilog-0.8.bak/StringHeap.h verilog-0.8/StringHeap.h
--- verilog-0.8.bak/StringHeap.h	2004-02-18 09:11:54.000000000 -0800
+++ verilog-0.8/StringHeap.h	2005-06-13 12:03:27.000000000 -0700
@@ -25,6 +25,8 @@
 # include  "config.h"
 # include  <string>
 
+using namespace std;
+
 class perm_string {
 
     public:
diff -ur verilog-0.8.bak/svector.h verilog-0.8/svector.h
--- verilog-0.8.bak/svector.h	2003-07-22 19:35:44.000000000 -0700
+++ verilog-0.8/svector.h	2005-06-13 12:02:02.000000000 -0700
@@ -101,8 +101,8 @@
  * Override the implementation of the above template for the string
  * type parameter. The initialization to nil works different here.
  */
-inline svector<string>::svector<string>(unsigned size)
-: nitems_(size), items_(new string[size])
+template <> inline svector<std::string>::svector(unsigned size)
+: nitems_(size), items_(new std::string[size])
 {
 }
 
diff -ur verilog-0.8.bak/verinum.h verilog-0.8/verinum.h
--- verilog-0.8.bak/verinum.h	2004-02-16 22:52:55.000000000 -0800
+++ verilog-0.8/verinum.h	2005-06-13 11:57:39.000000000 -0700
@@ -31,6 +31,8 @@
 class ostream;
 #endif
 
+using namespace std;
+
 /*
  * Numbers in Verilog are multibit strings, where each bit has 4
  * possible values: 0, 1, x or z. The verinum number is store in
diff -ur verilog-0.8.bak/verireal.h verilog-0.8/verireal.h
--- verilog-0.8.bak/verireal.h	2004-06-04 16:33:51.000000000 -0700
+++ verilog-0.8/verireal.h	2005-06-13 12:06:29.000000000 -0700
@@ -28,6 +28,8 @@
 class ostream;
 #endif
 
+using namespace std;
+
 class verinum;
 
 /*
