diff -uNr mono-1.1.5/data/config.in mono-1.1.5-new/data/config.in
--- mono-1.1.5/data/config.in	Tue Mar 15 19:38:18 2005
+++ mono-1.1.5-new/data/config.in	Thu Mar 24 11:28:58 2005
@@ -9,4 +9,8 @@
 	<dllmap dll="db2cli" target="libdb2_36@libsuffix@"/>
 	<dllmap dll="msvcrt" target="@LIBC@"/>
 	<dllmap dll="MonoPosixHelper" target="libMonoPosixHelper@libsuffix@"/>
+	<dllmap dll="libgtk-win32-2.0-0.dll" target="libgtk-x11-2.0@libsuffix@"/>
+	<dllmap dll="glib-2.0" target="libglib-2.0@libsuffix@"/>
+	<dllmap dll="gnomevfs-2" target="libgnomevfs-2@libsuffix@"/>
+	<dllmap dll="gtksourceview-1.0" target="libgtksourceview-1.0@libsuffix@"/>
 </configuration>
diff -uNr mono-1.1.5/mono/metadata/Makefile.in mono-1.1.5-new/mono/metadata/Makefile.in
--- mono-1.1.5/mono/metadata/Makefile.in	Fri Mar 18 10:08:49 2005
+++ mono-1.1.5-new/mono/metadata/Makefile.in	Thu Mar 24 11:28:46 2005
@@ -68,7 +68,7 @@
 am_libmonoruntime_static_la_OBJECTS = $(am__objects_1)
 libmonoruntime_static_la_OBJECTS =  \
 	$(am_libmonoruntime_static_la_OBJECTS)
-libmonoruntime_la_LIBADD =
+libmonoruntime_la_LIBADD = $(ICU_LIBS)
 am_libmonoruntime_la_OBJECTS = reflection.lo object.lo icall.lo \
 	decimal.lo boehm-gc.lo null-gc.lo gc.lo marshal.lo monitor.lo \
 	threads.lo threadpool.lo file-io.lo socket-io.lo exception.lo \
diff -uNr mono-1.1.5/mono/metadata/loader.c mono-1.1.5-new/mono/metadata/loader.c
--- mono-1.1.5/mono/metadata/loader.c	Thu Feb 24 22:52:24 2005
+++ mono-1.1.5-new/mono/metadata/loader.c	Thu Mar 24 11:28:58 2005
@@ -704,6 +704,19 @@
 		}
 
 		if (!gmodule) {
+			full_name = g_module_build_path ("@PREFIX@/lib", file_name);
+			mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_DLLIMPORT,
+					"DllImport loading library: '@PREFIX@/lib/%s'.", full_name);
+			gmodule = g_module_open (full_name, G_MODULE_BIND_LAZY);
+			if (!gmodule) {
+				mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_DLLIMPORT,
+						"DllImport error loading library '%s'.",
+						g_module_error ());
+			}
+			g_free (full_name);
+		}
+
+		if (!gmodule) {
 			full_name = g_module_build_path (".", file_name);
 			mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_DLLIMPORT,
 					"DllImport loading library: '%s'.", full_name);
