### Patch 000-main-ruby18.patch
diff -ur ruby-1.8.3.org/Makefile.in ruby-1.8.3/Makefile.in
--- ruby-1.8.3.org/Makefile.in	Wed Sep  7 01:22:56 2005
+++ ruby-1.8.3/Makefile.in	Sun Oct 16 23:51:07 2005
@@ -96,7 +96,7 @@
 
 $(PROGRAM):
 		@$(RM) $@
-		$(PURIFY) $(CC) $(LDFLAGS) $(XLDFLAGS) $(MAINLIBS) $(MAINOBJ) $(EXTOBJS) $(LIBRUBYARG) $(LIBS) $(OUTFLAG)$@
+		$(PURIFY) $(CC) $(XLDFLAGS) $(LDFLAGS) $(MAINLIBS) $(MAINOBJ) $(EXTOBJS) $(LIBRUBYARG) $(LIBS) $(OUTFLAG)$@
 
 # We must `rm' the library each time this rule is invoked because "updating" a
 # MAB library on Apple/NeXT (see --enable-fat-binary in configure) is not
diff -ur ruby-1.8.3.org/config.guess ruby-1.8.3/config.guess
--- ruby-1.8.3.org/config.guess	Sat Jul 17 04:23:52 2004
+++ ruby-1.8.3/config.guess	Sun Oct 16 23:44:11 2005
@@ -1200,7 +1200,7 @@
 	    *86) UNAME_PROCESSOR=i686 ;;
 	    powerpc) UNAME_PROCESSOR=powerpc ;;
 	esac
-	echo ${UNAME_PROCESSOR}-apple-darwin${UNAME_RELEASE}
+	echo ${UNAME_PROCESSOR}-apple-darwin
 	exit 0 ;;
     *:procnto*:*:* | *:QNX:[0123456789]*:*)
 	UNAME_PROCESSOR=`uname -p`
diff -ur ruby-1.8.3.org/configure ruby-1.8.3/configure
--- ruby-1.8.3.org/configure	Wed Sep 21 02:10:31 2005
+++ ruby-1.8.3/configure	Sun Oct 16 23:44:11 2005
@@ -14331,7 +14331,7 @@
 	rhapsody*)	: ${LDSHARED='cc -dynamic -bundle -undefined suppress'}
 			: ${LDFLAGS=""}
 			rb_cv_dlopen=yes ;;
-	darwin*)	: ${LDSHARED='cc -dynamic -bundle -undefined suppress -flat_namespace'}
+	darwin*)	: ${LDSHARED='cc -dynamic -bundle -flat_namespace'}
 			: ${LDFLAGS=""}
 			: ${LIBPATHENV=DYLD_LIBRARY_PATH}
 			rb_cv_dlopen=yes ;;
@@ -14997,7 +14997,7 @@
     ;;
 esac
 
-RUBY_SO_NAME='$(RUBY_INSTALL_NAME)'
+RUBY_SO_NAME=ruby
 LIBRUBY_LDSHARED=$LDSHARED
 LIBRUBY_DLDFLAGS=$DLDFLAGS
 LIBRUBY_SO='lib$(RUBY_SO_NAME).so.$(MAJOR).$(MINOR).$(TEENY)'
@@ -15084,9 +15084,9 @@
 	;;
     darwin*)
 	LIBRUBY_SO='lib$(RUBY_SO_NAME).$(MAJOR).$(MINOR).$(TEENY).dylib'
-	LIBRUBY_LDSHARED='cc -dynamiclib -undefined suppress -flat_namespace'
-	LIBRUBY_DLDFLAGS='-install_name $(libdir)/lib$(RUBY_SO_NAME).dylib -current_version $(MAJOR).$(MINOR).$(TEENY) -compatibility_version $(MAJOR).$(MINOR)'
-	LIBRUBY_ALIASES='lib$(RUBY_SO_NAME).$(MAJOR).$(MINOR).dylib lib$(RUBY_SO_NAME).dylib'
+	LIBRUBY_LDSHARED='cc -dynamiclib'
+	LIBRUBY_DLDFLAGS='-install_name $(libdir)/lib$(RUBY_SO_NAME).$(MAJOR).$(MINOR).dylib -current_version $(MAJOR).$(MINOR).$(TEENY) -compatibility_version $(MAJOR).$(MINOR)'
+	LIBRUBY_ALIASES='lib$(RUBY_SO_NAME).$(MAJOR).$(MINOR).dylib lib$(RUBY_SO_NAME).dylib libruby.2.dylib'
 	;;
     interix*)
 	LIBRUBYARG_SHARED='-L${libdir} -L. -l$(RUBY_SO_NAME)'
diff -ur ruby-1.8.3.org/ext/curses/extconf.rb ruby-1.8.3/ext/curses/extconf.rb
--- ruby-1.8.3.org/ext/curses/extconf.rb	Wed Feb  9 16:55:35 2005
+++ ruby-1.8.3/ext/curses/extconf.rb	Sun Oct 16 23:44:11 2005
@@ -7,9 +7,9 @@
 make=false
 have_library("mytinfo", "tgetent") if /bow/ =~ RUBY_PLATFORM
 have_library("tinfo", "tgetent") or have_library("termcap", "tgetent")
-if have_header(*curses=%w"ncurses.h") and have_library("ncurses", "initscr")
+if have_header(*curses=%w"ncurses/curses.h") and have_library("ncurses", "initscr")
   make=true
-elsif have_header(*curses=%w"ncurses/curses.h") and have_library("ncurses", "initscr")
+elsif have_header(*curses=%w"ncurses.h") and have_library("ncurses", "initscr")
   make=true
 elsif have_header(*curses=%w"curses_colr/curses.h") and have_library("cur_colr", "initscr")
   curses.unshift("varargs.h")
diff -ur ruby-1.8.3.org/lib/mkmf.rb ruby-1.8.3/lib/mkmf.rb
--- ruby-1.8.3.org/lib/mkmf.rb	Wed Sep  7 01:35:31 2005
+++ ruby-1.8.3/lib/mkmf.rb	Sun Oct 16 23:49:20 2005
@@ -1240,7 +1240,7 @@
   $CFLAGS = with_config("cflags", arg_config("CFLAGS", config["CFLAGS"])).dup
   $ARCH_FLAG = with_config("arch_flag", arg_config("ARCH_FLAG", config["ARCH_FLAG"])).dup
   $CPPFLAGS = with_config("cppflags", arg_config("CPPFLAGS", config["CPPFLAGS"])).dup
-  $LDFLAGS = (with_config("ldflags") || "").dup
+  $LDFLAGS = with_config("ldflags", arg_config("LDFLAGS", config["LDFLAGS"])).dup
   $INCFLAGS = "-I$(topdir)"
   $DLDFLAGS = with_config("dldflags", arg_config("DLDFLAGS", config["DLDFLAGS"])).dup
   $LIBEXT = config['LIBEXT'].dup
@@ -1341,7 +1341,7 @@
   if CONFIG["DLEXT"] == $OBJEXT
     "ld $(DLDFLAGS) -r -o $@ $(OBJS)\n"
   else
-    "$(LDSHARED) $(DLDFLAGS) $(LIBPATH) #{OUTFLAG}$@ " \
+    "$(LDSHARED) $(LIBPATH) $(DLDFLAGS) #{OUTFLAG}$@ " \
     "$(OBJS) $(LOCAL_LIBS) $(LIBS)"
   end
 LIBPATHFLAG = config_string('LIBPATHFLAG') || ' -L"%s"'
diff -ur ruby-1.8.3.org/ruby.c ruby-1.8.3/ruby.c
--- ruby-1.8.3.org/ruby.c	Tue Jun 28 15:09:58 2005
+++ ruby-1.8.3/ruby.c	Sun Oct 16 23:42:31 2005
@@ -975,7 +975,13 @@
 static struct {
     char *begin, *end;
 } envspace;
+
+#ifdef __APPLE__
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+#else
 extern char **environ;
+#endif
 
 static void
 set_arg0space()


### Patch 001-tcltk-paths.patch
--- ruby-1.8.3/ext/tcltklib/extconf.rb.orig	2005-05-23 00:27:37.000000000 -0700
+++ ruby-1.8.3/ext/tcltklib/extconf.rb	2005-10-26 22:55:16.000000000 -0700
@@ -41,14 +41,14 @@
 
 dir_config("tk")
 dir_config("tcl")
-dir_config("X11")
+dir_config("X11", "/usr/X11R6/include", "/usr/X11R6/lib")
 
 tklib = with_config("tklib")
 tcllib = with_config("tcllib")
 stubs = enable_config("tcltk_stubs") || with_config("tcltk_stubs")
 
 def find_tcl(tcllib, stubs)
-  paths = ["/usr/local/lib", "/usr/pkg/lib", "/usr/lib"]
+  paths = ["@PREFIX@/lib", "/usr/local/lib", "/usr/pkg/lib", "/usr/lib"]
   if stubs
     func = "Tcl_InitStubs"
     lib = "tclstub"
@@ -71,7 +71,7 @@
 end
 
 def find_tk(tklib, stubs)
-  paths = ["/usr/local/lib", "/usr/pkg/lib", "/usr/lib"]
+  paths = ["@PREFIX@/lib", "/usr/local/lib", "/usr/pkg/lib", "/usr/lib"]
   if stubs
     func = "Tk_InitStubs"
     lib = "tkstub"


### Patch 002-prevent-core-dumps-from-older-installed-rubies.patch
diff -ru ruby-1.8.3.orig/Makefile.in ruby-1.8.3/Makefile.in
--- ruby-1.8.3.orig/Makefile.in	2005-09-06 16:22:56.000000000 -0700
+++ ruby-1.8.3/Makefile.in	2005-10-26 22:35:36.000000000 -0700
@@ -45,12 +45,15 @@
 MAINLIBS = @MAINLIBS@
 MINIOBJS = @MINIOBJS@
 
+extout = $(srcdir)/$(EXTOUT)
+RUBYARCHDIR = $(extout)/$(arch)$(target_prefix)
+
 RUBY_INSTALL_NAME=@RUBY_INSTALL_NAME@
 RUBY_SO_NAME=@RUBY_SO_NAME@
 EXEEXT = @EXEEXT@
 PROGRAM=$(RUBY_INSTALL_NAME)$(EXEEXT)
 RUBY = $(RUBY_INSTALL_NAME)
-MINIRUBY = @MINIRUBY@
+MINIRUBY = DYLD_LIBRARY_PATH=$(srcdir) @MINIRUBY@ -I$(RUBYARCHDIR)
 RUNRUBY = @RUNRUBY@
 
 #### End of system configuration section. ####
diff -ru ruby-1.8.3.orig/common.mk ruby-1.8.3/common.mk
--- ruby-1.8.3.orig/common.mk	2005-08-03 08:27:06.000000000 -0700
+++ ruby-1.8.3/common.mk	2005-10-26 21:48:28.000000000 -0700
@@ -54,10 +54,12 @@
 		--make="$(MAKE)" \
 		--mflags="$(MFLAGS)" \
 		--make-flags="$(MAKEFLAGS)"
+EXTMK_ARGS_ETC=	$(SCRIPT_ARGS) --extout="$(EXTOUT)" --extension etc --extstatic $(EXTSTATIC) --
 EXTMK_ARGS    =	$(SCRIPT_ARGS) --extout="$(EXTOUT)" --extension $(EXTS) --extstatic $(EXTSTATIC) --
 
 all: $(MKFILES) $(PREP) $(RBCONFIG) $(LIBRUBY)
-	@$(MINIRUBY) $(srcdir)/ext/extmk.rb $(EXTMK_ARGS)
+	$(MINIRUBY) $(srcdir)/ext/extmk.rb $(EXTMK_ARGS_ETC)
+	$(MINIRUBY) $(srcdir)/ext/extmk.rb $(EXTMK_ARGS)
 prog: $(PROGRAM) $(WPROGRAM)
 
 miniruby$(EXEEXT): config.status $(LIBRUBY_A) $(MAINOBJ) $(MINIOBJS) $(OBJS) $(DMYEXT)
diff -ru ruby-1.8.3.orig/ext/dl/depend ruby-1.8.3/ext/dl/depend
--- ruby-1.8.3.orig/ext/dl/depend	2003-10-23 01:59:42.000000000 -0700
+++ ruby-1.8.3/ext/dl/depend	2005-10-26 21:49:24.000000000 -0700
@@ -32,15 +32,15 @@
 
 call.func: $(srcdir)/mkcall.rb ./dlconfig.rb
 	@echo "Generating call.func"
-	@$(RUBY) $(srcdir)/mkcall.rb > $@
+	DYLD_LIBRARY_PATH=../.. $(RUBY) -I../../.ext/powerpc-darwin $(srcdir)/mkcall.rb > $@
 
 callback.func: $(srcdir)/mkcallback.rb ./dlconfig.rb
 	@echo "Generating callback.func"
-	@$(RUBY) $(srcdir)/mkcallback.rb > $@
+	DYLD_LIBRARY_PATH=../.. $(RUBY) -I../../.ext/powerpc-darwin $(srcdir)/mkcallback.rb > $@
 
 cbtable.func: $(srcdir)/mkcbtable.rb ./dlconfig.rb
 	@echo "Generating cbtable.func"
-	@$(RUBY) $(srcdir)/mkcbtable.rb > $@
+	DYLD_LIBRARY_PATH=../.. $(RUBY) -I../../.ext/powerpc-darwin $(srcdir)/mkcbtable.rb > $@
 
 debug:
 	$(MAKE) CPPFLAGS="$(CPPFLAGS) -DDEBUG"


### Patch 101_remove_unnecessary_file.patch
[ruby-dev:27270] removed unnecessary file.

Index: ext/tk/sample/demos-en/tcolor.bak
===================================================================
RCS file: ext/tk/sample/demos-en/tcolor.bak
diff -N ext/tk/sample/demos-en/tcolor.bak
--- ruby-1.8.3/ext/tk/sample/demos-en/tcolor.bak	31 Jul 2003 20:52:37 -0000	1.1
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,513 +0,0 @@
-#!/usr/local/bin/ruby
-#
-# tcolor --
-# このスクリプトはRGB,HSB,CYM形式をサポートする
-# 簡易カラーエディタです。
-#
-# Copyright (C) 1998 Takaaki Tateishi(ttate@jaist.ac.jp)
-# last update: Thu Jun 18 06:32:35 JST 1998
-#
-
-# まずはtk.rbを読み込む。
-
-require "tk"
-
-
-# Tkによって変更される変数はTkVariableのインスタンスを使う。
-
-$colorSpace = TkVariable.new(:rgb)
-$red = 65535
-$green = 0
-$blue = 0
-$color = "#ffff00000000"
-$updating = TkVariable.new(0)
-$autoUpdate = TkVariable.new(1)
-$name = TkVariable.new("")
-# $command = TkVariable.new("print(%%,\"\n\")")
-$command = TkVariable.new("")
-$label1 = TkVariable.new("label1")
-$label2 = TkVariable.new("label2")
-$label3 = TkVariable.new("label3")
-
-
-# 各イベント用のメソッド
-
-def rgbToHsv(red,green,blue)
-
-  if ( red > green )
-    max = red
-    min = green
-  else
-    max = green
-    min = red
-  end
-
-  if ( blue > max )
-    max = blue
-  else
-    if ( blue < min )
-      min = blue
-    end
-  end
-
-  range = max - min
-
-  if ( max == 0 )
-    sat = 0.0
-  else
-    sat = (max-min)/max
-  end
-
-  if ( sat == 0 )
-    hue = 0.0
-  else
-    rc = (max-red)/range
-    gc = (max-green)/range
-    bc = (max-blue)/range
-    if ( red == max )
-      hue = 0.166667 * (bc - gc)
-    else
-      if ( green == max )
-	hue = 0.166667 * (2.0 + rc - bc)
-      else
-	hue = 0.166667 * (4.0 + gc - rc)
-      end
-    end
-    if ( hue < 0.0 )
-      hue = hue + 1.0
-    end
-  end
-
-  [hue,sat,max/65535]
-end
-
-
-def hsbToRgb(hue,sat,value)
-  v = 65535.0 * value
-  if( sat == 0 )
-    ans = [v,v,v]
-  else
-    hue = hue*6.0
-    if ( hue >= 6 )
-      hue = 0.0
-    end
-    i = hue.to_i
-    f = hue - i
-    p = 65535.0 * value * (1.0 - sat)
-    q = 65535.0 * value * (1.0 - (sat * f))
-    t = 65535.0 * value * (1.0 - (sat * (1.0 - f)))
-    case i
-    when 0
-      ans = [v,t,p]
-    when 1
-      ans = [q,v,p]
-    when 2
-      ans = [p,v,t]
-    when 3
-      ans = [p,q,v]
-    when 4
-      ans = [t,p,v]
-    when 5
-      ans = [v,p,q]
-    else
-      raise(eException,"i value #{i} is out of range")
-    end
-  end
-  return ans
-end
-
-
-def doUpdate 
-  newCmd = $command.to_s.gsub("%%","\"#{$color}\"")
-  eval(newCmd)
-end
-
-
-def tc_scaleChanged
-  if( $updating.to_i == 1 )
-    return 
-  end
-
-  scale1 = $root.middle.middle.scale1
-  scale2 = $root.middle.middle.scale2
-  scale3 = $root.middle.middle.scale3
-
-  case $colorSpace.to_i
-  when :rgb
-    $red = (scale1.get * 65.535).to_i
-    $green = (scale2.get * 65.535).to_i
-    $blue = (scale3.get * 65.535).to_i
-  when :cmy
-    $red = (65535 - scale1.get * 65.535).to_i
-    $green = (65535 - scale2.get * 65.535).to_i
-    $blue = (65535 - scale3.get * 65.535).to_i        
-  when :hsb
-    list = hsbToRgb(scale1.get / 1000.0,
-		    scale2.get / 1000.0,
-		    scale3.get / 1000.0)
-    $red = list[0]
-    $green = list[1]
-    $blue = list[2]
-  else
-    raise(Exception,"unknown colorSpace")
-  end
-  $color = format("#%04x%04x%04x",$red.to_i,$green.to_i,$blue.to_i)
-  $root.middle.right.set_color($color)
-  if( $autoUpdate.to_i == 1 )
-    doUpdate
-  end
-  Tk.update(TRUE)
-end
-
-
-def tc_setScales
-  $updating.value = 1
-
-  scale1 = $root.middle.middle.scale1
-  scale2 = $root.middle.middle.scale2
-  scale3 = $root.middle.middle.scale3
-  
-  case $colorSpace.to_i
-  when :rgb
-    scale1.set($red / 65.535)
-    scale2.set($green / 65.535)
-    scale3.set($blue / 65.535)
-  when :cmy
-    scale1.set((65535 - $red) / 65.535)
-    scale2.set((65535 - $green) / 65.535)
-    scale3.set((65535 - $blue) / 65.535)
-  when :hsb
-    list = rgbToHsv($red,$green,$blue)
-    scale1.set( list[0] * 1000.0 )
-    scale2.set( list[1] * 1000.0 )
-    scale3.set( list[2] * 1000.0 )
-  else
-    raise(Exception,"unknown colorSpace")
-  end
-
-  $updating.value = 0
-end
-
-
-def tc_loadNamedColor(name)
-  if name[0,1] != "#" 
-    list = TkWinfo.rgb($root.middle.right.swatch,name)
-    $red = list[0]
-    $green = list[1]
-    $blue = list[2]
-  else
-    case name.length
-    when 4
-      format = /#(.{1})(.{1})(.{1})/
-      shift = 12
-    when 7
-      format = /#(.{2})(.{2})(.{2})/
-      shift = 8
-    when 10
-      format = /#(.{3})(.{3})(.{3})/
-      shift = 4
-    when 13
-      format = /#(.{4})(.{4})(.{4})/
-      shift = 0
-    else
-      raise(eException,"syntax error in color name \"#{name}\"")
-    end
-    name.scan(format){|strlist|
-      if strlist.length != 3
-	raise(eException,"syntax error in color name \"#{name}\"")
-      end
-      $red = strlist[0].to_i
-      $green = strlist[1].to_i
-      $blue = strlist[2].to_i
-    }
-    $red = $red << shift
-    $green = $green << shift
-    $blue = $blue << shift
-  end
-  
-  tc_setScales
-  $color = format("#%04x%04x%04x",$red,$green,$blue)
-  $root.middle.right.set_color($color)
-  if $autoUpdate.to_i == 1
-    doUpdate
-  end
-end
-
-
-def changeColorSpace(space)
-  case space
-  when :rgb
-    $label1.value = "Red"
-    $label2.value = "Green"
-    $label3.value = "Blue"
-  when :cmy
-    $label1.value = "Cyan"
-    $label2.value = "Magenta"
-    $label3.value = "Yellow"
-  when :hsb
-    $label1.value = "Hue"
-    $label2.value = "Saturation"
-    $label3.value = "Brightness"
-  end
-  tc_setScales
-end
-
-
-
-
-
-# tcolor用のメニュー
-
-class TkColorMenuFrame<TkFrame
-  def initialize(parent)
-    super(parent,
-	  "relief"=>"raised",
-	  "borderwidth"=>"2")
-
-    # Fileメニューボタンの生成
-    @file = TkMenubutton.new(self){|button|
-
-      # Fileメニューの作成
-      @file_menu = TkMenu.new(button){
-	add "radio",
-	  "label" => "RGB color space",
-	  "variable" => $colorSpace,
-	  "value" => :rgb,
-	  "underline" => "0",
-	  "command" => proc{changeColorSpace(:rgb)}
-	add "radio",
-	  "label" => "CMY color space",
-	  "variable" => $colorSpace,
-	  "value" => :cmy,
-	  "underline" => "0",
-	  "command" => proc{changeColorSpace(:cmy)}
-	add "radio",
-	  "label" => "HSB color space",
-	  "variable" => $colorSpace,
-	  "value" => :hsb,
-	  "underline" => "0",
-	  "command" => proc{changeColorSpace(:hsb)}
-	add "separator"
-	add "radio",
-	  "label" => "Qutomatic updates",
-	  "variable" => $autoUpdate,
-	  "value" => "1",
-	  "underline" => "0"
-	add "radio",
-	  "label" => "Manual updates",
-	  "variable" => $autoUpdate,
-	  "value" => "0",
-	  "underline" => "0"
-	add "separator"
-	add "command",
-	  "label" => "Exit program",
-	  "underline" => "0",
-	  "command" => proc{exit}
-      }
-      
-      # FileメニューとFileボタンを関連付ける
-      menu @file_menu
-
-      text "File"
-      underline "0"
-    }.pack("side"=>"left")
-
-    self
-  end
-end
-
-
-# 下部のフレームのためのクラス
-class TkColorBotFrame<TkFrame
-  def initialize(parent)
-    super(parent,
-	  "relief"=> "raised",
-	  "borderwidth"=> 2)
-
-    @commandLabel = TkLabel.new(self,
-				"text"=> "Command:")
-    @command = TkEntry.new(self,
-			   "relief"=> "sunken",
-			   "borderwidth"=> "2",
-			   "textvariable"=> $command,
-			   "font"=> "-Adobe-Courier-Medium-R-Normal--*-120-*-*-*-*-*-*")
-    @update = TkButton.new(self,
-			   "text"=> "Update",
-			   "command"=> proc{doUpdate})
-    @commandLabel.pack("side"=>"left")
-    @update.pack("side"=>"right","pady"=>".1c","padx"=>".25c")
-    @command.pack("expand"=>"yes","fill"=>"x","ipadx"=>".25c")
-
-    self
-  end
-end    
-
-
-# 中段左のフレーム
-class TkColorMiddleLeftFrame<TkFrame
-  def initialize(parent)
-    super(parent)
-
-    for i in ["/usr/local/lib/X11rgb.txt","/usr/lib/X11/rgb.txt",
-	"/X11/R5/lib/X11/rgb.txt","/X11/R4/lib/rgb/rgb.txt",
-	"/usr/openwin/lib/X11/rgb.txt"]
-      if !File.readable?(i)
-	next
-      end
-      f = File.open(i)
-      @scroll = TkScrollbar.new(self,
-				"orient"=>"vertical",
-				"relief"=>"sunken",
-				"borderwidth"=>"2")
-      @scroll.pack("side"=>"right","fill"=>"y")
-      @names = TkListbox.new(self,
-			     "width"=>"20",
-			     "height"=>"12",
-			     "yscrollcommand"=> proc{|first,last| @scroll.set first,last},
-			     "relief"=>"sunken",
-			     "borderwidth"=>"2",
-			     "exportselection"=>"false")
-      @scroll.command(proc{|*args| @names.yview *args})
-      @names.bind("Double-1",proc{
-		    tc_loadNamedColor(@names.get(@names.curselection))})
-      @names.pack("side"=>"left")
-      while (line = f.gets)
-	line.chop!
-	linelist = line.split(/[ \t]+/)
-	if linelist.length == 4
-	  @names.insert("end",linelist[3])
-	end
-      end
-      f.close
-      break
-    end
-
-    self
-  end
-end
-
-
-# 中段中央のフレーム
-class TkColorMiddleMiddleFrame<TkFrame
-  # @scale1,@scale2,@scale3を外部から参照のみ許可する。(変更不可)
-  attr_reader :scale1, :scale2, :scale3
-
-  def initialize(parent)
-    super(parent)
-
-    @f1 = TkFrame.new(self)
-    @f2 = TkFrame.new(self)
-    @f3 = TkFrame.new(self)
-    @f4 = TkFrame.new(self)
-
-    for f in [@f1,@f2,@f3]
-      f.pack("side"=>"top","expand"=>"yes")
-    end
-    @f4.pack("side"=>"top","expand"=>"yes","fill"=>"x")
-
-    @label1 = TkLabel.new(self,"textvariable"=>$label1)
-    @scale1 = TkScale.new(self,"from"=>"0","to"=>"1000","length"=>"6c",
-			  "orient"=>"horizontal",
-			  "command"=>proc{tc_scaleChanged})
-    @scale1.pack("side"=>"top","anchor"=>"w")
-    @label1.pack("side"=>"top","anchor"=>"w")
-
-    @label2 = TkLabel.new(self,"textvariable"=>$label2)
-    @scale2 = TkScale.new(self,"from"=>"0","to"=>"1000","length"=>"6c",
-			  "orient"=>"horizontal",
-			  "command"=>proc{tc_scaleChanged})
-    @scale2.pack("side"=>"top","anchor"=>"w")
-    @label2.pack("side"=>"top","anchor"=>"w")
-
-    @label3 = TkLabel.new(self,"textvariable"=>$label3)
-    @scale3 = TkScale.new(self,"from"=>"0","to"=>"1000","length"=>"6c",
-			  "orient"=>"horizontal",
-			  "command"=>proc{tc_scaleChanged})
-    @scale3.pack("side"=>"top","anchor"=>"w")
-    @label3.pack("side"=>"top","anchor"=>"w")
-
-    @nameLabel = TkLabel.new(self,"text"=>"Name:")
-    @name = TkEntry.new(self,"relief"=>"sunken","borderwidth"=>"2",
-			"textvariable"=>$name,"width"=>"10",
-			"font"=>"-Adobe-Courier-Medium-R-Normal--*-120-*-*-*-*-*-*")
-    @nameLabel.pack("side"=>"left")
-    @name.pack("side"=>"right", "expand"=>"1", "fill"=>"x")
-    @name.bind("Return",proc{tc_loadNamedColor $name.to_s})
-
-    self
-  end
-end
-
-
-class TkColorMiddleRightFrame<TkFrame
-  attr_reader :swatch
-
-  def initialize(parent)
-    super(parent)
-    @swatch = TkFrame.new(self, "width"=>"2c", "height"=>"5c",
-			  "background"=>$color)
-    @value = TkLabel.new(self, 
-			 "text"=>$color,
-			 "width"=>"13",
-			 "font"=>"-Adobe-Courier-Medium-R-Normal--*-120-*-*-*-*-*-*")
-    @swatch.pack("side"=>"top","expand"=>"yes","fill"=>"both")
-    @value.pack("side"=>"bottom","pady"=>".25c")
-
-    self
-  end
-
-  def set_color(color)
-    @swatch["background"] = color
-    @value["text"] = color
-  end
-end
-
-
-
-# 中段のフレーム
-class TkColorMiddleFrame<TkFrame
-  attr_reader :left, :middle, :right
-
-  def initialize(parent)
-    super(parent,
-	  "relief"=> "raised",
-	  "borderwidth"=> "2")
-
-    @left = TkColorMiddleLeftFrame.new(self)
-    @left.pack("side"=>"left","padx"=>".25c","pady"=>".25c")
-
-    @middle = TkColorMiddleMiddleFrame.new(self)
-    @middle.pack("side"=>"left","expand"=>"yes","fill"=>"y")
-
-    @right = TkColorMiddleRightFrame.new(self)
-    @right.pack("side"=>"left","padx"=>".25c","pady"=>".25c","anchor"=>"s")
-
-    self
-  end
-end
-
-
-class TkColor<TkRoot
-  attr_reader :menu, :bottom, :middle
-
-  def initialize
-    super
-    @menu = TkColorMenuFrame.new(self)
-    @menu.pack("side"=>"top", "fill"=>"x")
-
-    @bottom = TkColorBotFrame.new(self)
-    @bottom.pack("side"=>"bottom","fill"=>"x")
-
-    @middle = TkColorMiddleFrame.new(self)
-    @middle.pack("side"=>"top","fill"=>"both")
-
-    self
-  end
-end
-
-
-$root = TkColor.new
-
-# イベントを待つ為にループに入る。
-changeColorSpace :rgb
-Tk.mainloop


### Patch 102_mkmf.rb_default_path.patch
[ruby-dev:27281] default path if environment is not set. 

Index: lib/mkmf.rb
===================================================================
RCS file: /var/cvs/src/ruby/lib/mkmf.rb,v
retrieving revision 1.162.2.41
retrieving revision 1.162.2.42
diff -u -r1.162.2.41 -r1.162.2.42
--- ruby-1.8.3/lib/mkmf.rb	21 Sep 2005 14:31:20 -0000	1.162.2.41
+++ ruby-1.8.3/lib/mkmf.rb	22 Sep 2005 14:37:02 -0000	1.162.2.42
@@ -762,8 +762,17 @@
 end
 
 def find_executable0(bin, path = nil)
-  path = (path || ENV['PATH']).split(File::PATH_SEPARATOR)
   ext = config_string('EXEEXT')
+  if File.expand_path(bin) == bin
+    return bin if File.executable?(bin)
+    return file if ext and File.executable?(file = bin + ext)
+    return nil
+  end
+  if path ||= ENV['PATH']
+    path = path.split(File::PATH_SEPARATOR)
+  else
+    path = %w[/usr/local/bin /usr/ucb /usr/bin /bin]
+  end
   file = nil
   path.each do |dir|
     return file if File.executable?(file = File.join(dir, bin))


### Patch 103_fileutils_visibility.patch
[ruby-core:05954] fix visibility of FileUtils::NoWrite, Verbose, DryRun (backported from trunk, rev 1.66).
[ruby-core:05858] [ruby-Bugs:2494] no longer accept :noop option, related code is useless (backported from trunk, rev 1.67).

Index: lib/fileutils.rb
===================================================================
RCS file: /var/cvs/src/ruby/lib/fileutils.rb,v
retrieving revision 1.36.2.11
retrieving revision 1.36.2.13
diff -u -r1.36.2.11 -r1.36.2.13
--- ruby-1.8.3/lib/fileutils.rb	19 Sep 2005 01:37:27 -0000	1.36.2.11
+++ ruby-1.8.3/lib/fileutils.rb	23 Sep 2005 23:56:42 -0000	1.36.2.13
@@ -118,7 +118,7 @@
   def cd(dir, options = {}, &block) # :yield: dir
     fu_check_options options, :verbose
     fu_output_message "cd #{dir}" if options[:verbose]
-    Dir.chdir(dir, &block) unless options[:noop]
+    Dir.chdir(dir, &block)
     fu_output_message 'cd -' if options[:verbose] and block
   end
   module_function :cd
@@ -1443,6 +1443,8 @@
   end
   private_module_function :fu_output_message
 
+  METHODS = singleton_methods() - ['private_module_function']
+
   #
   # Returns an Array of method names which have any options.
   #
@@ -1505,9 +1507,15 @@
         def #{name}(*args)
           super(*fu_update_option(args, :verbose => true))
         end
+        private :#{name}
       EOS
     end
     extend self
+    class << self
+      ::FileUtils::METHODS.each do |m|
+        public m
+      end
+    end
   end
 
   # 
@@ -1524,9 +1532,15 @@
         def #{name}(*args)
           super(*fu_update_option(args, :noop => true))
         end
+        private :#{name}
       EOS
     end
     extend self
+    class << self
+      ::FileUtils::METHODS.each do |m|
+        public m
+      end
+    end
   end
 
   # 
@@ -1544,9 +1558,15 @@
         def #{name}(*args)
           super(*fu_update_option(args, :noop => true, :verbose => true))
         end
+        private :#{name}
       EOS
     end
     extend self
+    class << self
+      ::FileUtils::METHODS.each do |m|
+        public m
+      end
+    end
   end
 
 end


### Patch 104_ruby-mode.el_indent.patch
[ruby-list:41168] deal with heredoc separately.
[] arrange deep-indent closing parenthesis at same column as the opening.

Index: misc/ruby-mode.el
===================================================================
RCS file: /var/cvs/src/ruby/misc/ruby-mode.el,v
retrieving revision 1.74.2.11
retrieving revision 1.74.2.13
diff -u -r1.74.2.11 -r1.74.2.13
--- ruby-1.8.3/misc/ruby-mode.el	12 Jun 2005 16:58:40 -0000	1.74.2.11
+++ ruby-1.8.3/misc/ruby-mode.el	24 Sep 2005 16:47:20 -0000	1.74.2.13
@@ -1,12 +1,12 @@
 ;;;
 ;;;  ruby-mode.el -
 ;;;
-;;;  $Author: matz $
-;;;  $Date: 2005/06/12 16:58:40 $
+;;;  $Author: nobu $
+;;;  $Date: 2005/09/24 16:47:20 $
 ;;;  created at: Fri Feb  4 14:49:13 JST 1994
 ;;;
 
-(defconst ruby-mode-revision "$Revision: 1.74.2.11 $")
+(defconst ruby-mode-revision "$Revision: 1.74.2.13 $")
 
 (defconst ruby-mode-version
   (progn
@@ -152,7 +152,7 @@
 Also ignores spaces after parenthesis when 'space."
   :group 'ruby)
 
-(defcustom ruby-deep-indent-paren '(?\( t)
+(defcustom ruby-deep-indent-paren '(?\( ?\[ ?\] t)
   "*Deep indent lists in parenthesis when non-nil. t means continuous line.
 Also ignores spaces after parenthesis when 'space."
   :group 'ruby)
@@ -312,11 +312,10 @@
 	       (or (eq (char-syntax (char-before (point))) ?w)
 		   (ruby-special-char-p))))
 	nil)
-       ((or (save-excursion (goto-char start) (looking-at ruby-operator-re))
+       ((and (eq option 'heredoc) (< space 0)) t)
+       ((or (looking-at ruby-operator-re)
 	    (looking-at "[\\[({,;]")
-	    (and (or (not (eq option 'heredoc))
-		     (< space 0))
-		 (looking-at "[!?]")
+	    (and (looking-at "[!?]")
 		 (or (not (eq option 'modifier))
 		     (bolp)
 		     (save-excursion (forward-char -1) (looking-at "\\Sw"))))
@@ -327,7 +326,7 @@
 					   "|" ruby-block-op-re
 					   "|" ruby-block-mid-re "\\)\\>")))
 		   (goto-char (match-end 0))
-                  (not (looking-at "\\s_")))
+		   (not (looking-at "\\s_")))
 		  ((eq option 'expr-qstr)
 		   (looking-at "[a-zA-Z][a-zA-z0-9_]* +%[^ \t]"))
 		  ((eq option 'expr-re)
@@ -618,7 +617,11 @@
 			   (t (setq indent (ruby-indent-size (1- indent) 1))))))
 	    (if (nth 3 state) (goto-char (nth 3 state))
 	      (goto-char parse-start) (back-to-indentation))
-	    (setq indent (ruby-indent-size (current-column) (nth 2 state))))))
+	    (setq indent (ruby-indent-size (current-column) (nth 2 state))))
+	  (and (eq (car (nth 1 state)) paren)
+	       (ruby-deep-indent-paren-p (matching-paren paren))
+	       (search-backward (char-to-string paren))
+	       (setq indent (current-column)))))
        ((and (nth 2 state) (> (nth 2 state) 0)) ; in nest
 	(if (null (cdr (nth 1 state)))
 	    (error "invalid nest"))
@@ -727,6 +730,7 @@
 		     (goto-char (or begin parse-start))
 		     (skip-syntax-forward " ")
 		     (current-column)))
+		  ((car (nth 1 state)) indent)
 		  (t
 		   (+ indent ruby-indent-level))))))))
       indent)))


### Patch 105_xmlrpc-server.patch
[] delete wrong call of "join".
[] should mount the servlet on "/".

Index: lib/xmlrpc/server.rb
===================================================================
RCS file: /var/cvs/src/ruby/lib/xmlrpc/server.rb,v
retrieving revision 1.2.2.4
retrieving revision 1.2.2.6
diff -u -r1.2.2.4 -r1.2.2.6
--- ruby-1.8.3/lib/xmlrpc/server.rb	20 Sep 2005 08:51:02 -0000	1.2.2.4
+++ ruby-1.8.3/lib/xmlrpc/server.rb	4 Oct 2005 19:46:35 -0000	1.2.2.6
@@ -634,7 +634,7 @@
     require 'webrick'
     @server = WEBrick::HTTPServer.new(:Port => port, :BindAddress => host, :MaxClients => maxConnections, 
                                       :Logger => WEBrick::Log.new(stdlog))
-    @server.mount("/RPC2", self)
+    @server.mount("/", self)
   end
   
   def serve
@@ -645,7 +645,7 @@
     end
     trap(signal) { @server.shutdown }
 
-    @server.start.join
+    @server.start
   end
   
   def shutdown


### Patch 106_optparse_unmatched_args.patch
[ruby-dev:27316] not consume unmatched argument.

Index: lib/optparse.rb
===================================================================
RCS file: /var/cvs/src/ruby/lib/optparse.rb,v
retrieving revision 1.40.2.5
retrieving revision 1.40.2.7
diff -u -r1.40.2.5 -r1.40.2.7
--- ruby-1.8.3/lib/optparse.rb	20 Sep 2005 22:45:06 -0000	1.40.2.5
+++ ruby-1.8.3/lib/optparse.rb	26 Sep 2005 12:44:47 -0000	1.40.2.7
@@ -192,7 +192,7 @@
 #
 class OptionParser
   # :stopdoc:
-  RCSID = %w$Id: optparse.rb,v 1.40.2.5 2005/09/20 22:45:06 nobu Exp $[1..-1].each {|s| s.freeze}.freeze
+  RCSID = %w$Id: optparse.rb,v 1.40.2.7 2005/09/26 12:44:47 nobu Exp $[1..-1].each {|s| s.freeze}.freeze
   Version = (RCSID[1].split('.').collect {|s| s.to_i}.extend(Comparable).freeze if RCSID[1])
   LastModified = (Time.gm(*RCSID[2, 2].join('-').scan(/\d+/).collect {|s| s.to_i}) if RCSID[2])
   Release = RCSID[2]
@@ -441,12 +441,12 @@
       #
       # Raises an exception if argument is not present.
       #
-      def parse(arg, argv, &error)
+      def parse(arg, argv)
         unless arg
           raise MissingArgument if argv.empty?
           arg = argv.shift
         end
-        conv_arg(*parse_arg(arg, &error))
+        conv_arg(*parse_arg(arg) {|*exc| raise(*exc)})
       end
     end
 
@@ -467,11 +467,11 @@
     end
 
     #
-    # ?
+    # Switch that takes an argument, which does not begin with '-'.
     #
     class PlacedArgument < self
       #
-      # ?
+      # Returns nil if argument is not present or begins with '-'.
       #
       def parse(arg, argv, &error)
         if !(val = arg) and (argv.empty? or /\A-/ =~ (val = argv[0]))


### Patch 107_webrick-cgi_query_string.patch
[ruby-list:41186] req.query_string should refer the value of QUERY_STRING.

Index: lib/webrick/cgi.rb
===================================================================
RCS file: /var/cvs/src/ruby/lib/webrick/cgi.rb,v
retrieving revision 1.4.2.8
retrieving revision 1.4.2.9
diff -u -r1.4.2.8 -r1.4.2.9
--- ruby-1.8.3/lib/webrick/cgi.rb	15 Sep 2005 15:07:05 -0000	1.4.2.8
+++ ruby-1.8.3/lib/webrick/cgi.rb	28 Sep 2005 06:16:59 -0000	1.4.2.9
@@ -66,6 +66,7 @@
         req.parse(sock)
         req.script_name = (env["SCRIPT_NAME"] || File.expand_path($0)).dup
         req.path_info = (env["PATH_INFO"] || "").dup
+        req.query_string = env["QUERY_STRING"]
         req.user = env["REMOTE_USER"]
         res.request_method = req.request_method
         res.request_uri = req.request_uri
Index: lib/webrick/httprequest.rb
===================================================================
RCS file: /var/cvs/src/ruby/lib/webrick/httprequest.rb,v
retrieving revision 1.5.2.3
retrieving revision 1.5.2.4
diff -u -r1.5.2.3 -r1.5.2.4
--- ruby-1.8.3/lib/webrick/httprequest.rb	14 Jul 2005 23:00:22 -0000	1.5.2.3
+++ ruby-1.8.3/lib/webrick/httprequest.rb	28 Sep 2005 06:16:59 -0000	1.5.2.4
@@ -27,8 +27,8 @@
     attr_reader :request_method, :unparsed_uri, :http_version
 
     # Request-URI
-    attr_reader :request_uri, :host, :port, :path, :query_string
-    attr_accessor :script_name, :path_info
+    attr_reader :request_uri, :host, :port, :path
+    attr_accessor :script_name, :path_info, :query_string
 
     # Header and entity body
     attr_reader :raw_header, :header, :cookies


### Patch 108_webrick_default_userdir.patch
[] :UserDir should be nil.  It is harmful to permit the access to ~/public_html by default.

Index: lib/webrick/config.rb
===================================================================
RCS file: /var/cvs/src/ruby/lib/webrick/config.rb,v
retrieving revision 1.2.2.3
retrieving revision 1.2.2.4
diff -u -r1.2.2.3 -r1.2.2.4
--- ruby-1.8.3/lib/webrick/config.rb	16 Dec 2004 09:45:59 -0000	1.2.2.3
+++ ruby-1.8.3/lib/webrick/config.rb	14 Oct 2005 07:58:39 -0000	1.2.2.4
@@ -71,7 +71,7 @@
       :HandlerCallback   => nil,
       :DirectoryCallback => nil,
       :FileCallback      => nil,
-      :UserDir           => "public_html",
+      :UserDir           => nil,  # e.g. "public_html"
       :AcceptableLanguages => []  # ["en", "ja", ... ]
     }
 


### Patch 109_suppress_warnings.patch
[ruby-dev:27237][ruby-core:05854] fix typo and suppress warnings.
[ruby-core:06247] get rid of warnings.
[ruby-dev:27383] fixed typo.

Index: lib/yaml/basenode.rb
===================================================================
RCS file: /var/cvs/src/ruby/lib/yaml/basenode.rb,v
retrieving revision 1.2.2.2
retrieving revision 1.2.2.3
diff -u -r1.2.2.2 -r1.2.2.3
--- ruby-1.8.3/lib/yaml/basenode.rb	20 Sep 2005 06:46:45 -0000	1.2.2.2
+++ ruby-1.8.3/lib/yaml/basenode.rb	27 Sep 2005 22:57:53 -0000	1.2.2.3
@@ -148,7 +148,7 @@
             if pred
                 case pred
                 when /^\.=/
-                    pred = $'
+                    pred = $'   # '
                     match_nodes.reject! { |n|
                         n.last.value != pred
                     }
@@ -187,7 +187,7 @@
                 v = @value.detect { |k,v| k.transform == key.first }
                 v[1] if v
             elsif Array === @value
-                @value.[]( *k )
+                @value.[]( *key )
             end
         end
 
Index: lib/yaml/tag.rb
===================================================================
RCS file: /var/cvs/src/ruby/lib/yaml/tag.rb,v
retrieving revision 1.1.2.1
retrieving revision 1.1.2.2
diff -u -r1.1.2.1 -r1.1.2.2
--- ruby-1.8.3/lib/yaml/tag.rb	13 Sep 2005 03:58:33 -0000	1.1.2.1
+++ ruby-1.8.3/lib/yaml/tag.rb	27 Sep 2005 22:57:53 -0000	1.1.2.2
@@ -1,5 +1,5 @@
 # -*- mode: ruby; ruby-indent-level: 4; tab-width: 4 -*- vim: sw=4 ts=4
-# $Id: tag.rb,v 1.1.2.1 2005/09/13 03:58:33 why Exp $
+# $Id: tag.rb,v 1.1.2.2 2005/09/27 22:57:53 nobu Exp $
 #
 # = yaml/tag.rb: methods for associating a taguri to a class.
 #
@@ -55,13 +55,14 @@
     # in YAML.  See YAML::tag_class for detailed information on typing and
     # taguris.
     def yaml_as( tag, sc = true )
-        class_eval <<-"end;"
-            attr_accessor :taguri
+        verbose, $VERBOSE = $VERBOSE, nil
+        class_eval <<-"end;", __FILE__, __LINE__+1
+            attr_writer :taguri
             def taguri
                 if respond_to? :to_yaml_type
                     YAML::tagurize( to_yaml_type[1..-1] )
                 else
-                    return @taguri if @taguri
+                    return @taguri if defined?(@taguri) and @taguri
                     tag = #{ tag.dump }
                     if self.class.yaml_tag_subclasses? and self.class != YAML::tagged_classes[tag]
                         tag = "\#{ tag }:\#{ self.class.yaml_tag_class_name }"
@@ -72,6 +73,8 @@
             def self.yaml_tag_subclasses?; #{ sc ? 'true' : 'false' }; end
         end;
         YAML::tag_class tag, self
+    ensure
+        $VERBOSE = verbose
     end
     # Transforms the subclass name into a name suitable for display
     # in a subclassed tag.
Index: lib/yaml/types.rb
===================================================================
RCS file: /var/cvs/src/ruby/lib/yaml/types.rb,v
retrieving revision 1.4.2.1
retrieving revision 1.4.2.2
diff -u -r1.4.2.1 -r1.4.2.2
--- ruby-1.8.3/lib/yaml/types.rb	13 Sep 2005 03:58:33 -0000	1.4.2.1
+++ ruby-1.8.3/lib/yaml/types.rb	27 Sep 2005 22:57:53 -0000	1.4.2.2
@@ -1,45 +1,52 @@
+# -*- mode: ruby; ruby-indent-level: 4 -*- vim: sw=4
 #
 # Classes required by the full core typeset
 #
 
 module YAML
 
-	#
-	# Default private type
-	#
-	class PrivateType
+    #
+    # Default private type
+    #
+    class PrivateType
         def self.tag_subclasses?; false; end
-		attr_accessor :type_id, :value
-		def initialize( type, val )
-			@type_id = type; @value = val
+        attr_accessor :type_id, :value
+        verbose, $VERBOSE = $VERBOSE, nil
+        def initialize( type, val )
+            @type_id = type; @value = val
             @value.taguri = "x-private:#{ @type_id }"
-		end
-		def to_yaml( opts = {} )
+        end
+        def to_yaml( opts = {} )
             @value.to_yaml( opts )
-		end
-	end
+        end
+    ensure
+        $VERBOSE = verbose
+    end
 
     #
     # Default domain type
     #
     class DomainType
         def self.tag_subclasses?; false; end
-		attr_accessor :domain, :type_id, :value
-		def initialize( domain, type, val )
-			@domain = domain; @type_id = type; @value = val
+        attr_accessor :domain, :type_id, :value
+        verbose, $VERBOSE = $VERBOSE, nil
+        def initialize( domain, type, val )
+            @domain = domain; @type_id = type; @value = val
             @value.taguri = "tag:#{ @domain }:#{ @type_id }"
-		end
-		def to_yaml( opts = {} )
+        end
+        def to_yaml( opts = {} )
             @value.to_yaml( opts )
-		end
-	end
+        end
+    ensure
+        $VERBOSE = verbose
+    end
 
     #
     # Unresolved objects
     #
     class Object
         def self.tag_subclasses?; false; end
-		def to_yaml( opts = {} )
+        def to_yaml( opts = {} )
             YAML::quick_emit( object_id, opts ) do |out|
                 out.map( "tag:ruby.yaml.org,2002:object:#{ @class }", to_yaml_style ) do |map|
                     @ivars.each do |k,v|
@@ -47,31 +54,31 @@
                     end
                 end
             end
-		end
-	end
+        end
+    end
 
-	#
-	# YAML Hash class to support comments and defaults
-	#
-	class SpecialHash < ::Hash 
-		attr_accessor :default
+    #
+    # YAML Hash class to support comments and defaults
+    #
+    class SpecialHash < ::Hash 
+        attr_accessor :default
         def inspect
             self.default.to_s
         end
-		def to_s
-			self.default.to_s
-		end
-		def update( h )
-			if YAML::SpecialHash === h
-				@default = h.default if h.default
-			end
-			super( h )
-		end
+        def to_s
+            self.default.to_s
+        end
+        def update( h )
+            if YAML::SpecialHash === h
+                @default = h.default if h.default
+            end
+            super( h )
+        end
         def to_yaml( opts = {} )
             opts[:DefaultKey] = self.default
             super( opts )
         end
-	end
+    end
 
     #
     # Builtin collection: !omap
Index: ext/syck/syck.h
===================================================================
RCS file: /var/cvs/src/ruby/ext/syck/syck.h,v
retrieving revision 1.21.2.7
retrieving revision 1.21.2.8
diff -u -r1.21.2.7 -r1.21.2.8
--- ruby-1.8.3/ext/syck/syck.h	20 Sep 2005 06:46:44 -0000	1.21.2.7
+++ ruby-1.8.3/ext/syck/syck.h	13 Oct 2005 14:30:53 -0000	1.21.2.8
@@ -445,6 +445,9 @@
  * Lexer prototypes
  */
 void syckerror( char * );
+int syckparse( void * );
+union YYSTYPE;
+int sycklex( union YYSTYPE *, SyckParser * );
 
 #if defined(__cplusplus)
 }  /* extern "C" { */
Index: io.c
===================================================================
RCS file: /var/cvs/src/ruby/io.c,v
retrieving revision 1.246.2.89
retrieving revision 1.246.2.90
diff -u -r1.246.2.89 -r1.246.2.90
--- ruby-1.8.3/io.c	27 Sep 2005 23:12:43 -0000	1.246.2.89
+++ ruby-1.8.3/io.c	13 Oct 2005 14:30:48 -0000	1.246.2.90
@@ -72,6 +72,10 @@
 #endif
 
 #ifdef HAVE_UNISTD_H
+#ifdef HAVE_SYSCALL_H
+#include <syscall.h>
+#endif
+
 #include <unistd.h>
 #endif
 
Index: pack.c
===================================================================
RCS file: /var/cvs/src/ruby/pack.c,v
retrieving revision 1.62.2.11
retrieving revision 1.62.2.12
diff -u -r1.62.2.11 -r1.62.2.12
--- ruby-1.8.3/pack.c	7 Oct 2005 01:01:22 -0000	1.62.2.11
+++ ruby-1.8.3/pack.c	13 Oct 2005 14:30:49 -0000	1.62.2.12
@@ -2036,7 +2036,7 @@
     rb_raise(rb_eRangeError, "pack(U): value out of range");
 }
 
-static const long utf8_limits[] = {
+static const unsigned long utf8_limits[] = {
     0x0,			/* 1 */
     0x80,			/* 2 */
     0x800,			/* 3 */
Index: missing/isinf.c
===================================================================
RCS file: /var/cvs/src/ruby/missing/isinf.c,v
retrieving revision 1.5
retrieving revision 1.5.2.1
diff -u -r1.5 -r1.5.2.1
--- ruby-1.8.3/missing/isinf.c	21 Dec 2003 10:30:24 -0000	1.5
+++ ruby-1.8.3/missing/isinf.c	13 Oct 2005 14:30:54 -0000	1.5.2.1
@@ -23,6 +23,7 @@
 
 #if defined(HAVE_FINITE) && defined(HAVE_ISNAN)
 
+#include <math.h>
 #ifdef HAVE_IEEEFP_H
 #include <ieeefp.h>
 #endif
Index: ext/syck/rubyext.c
===================================================================
RCS file: /var/cvs/src/ruby/ext/syck/rubyext.c,v
retrieving revision 1.30.2.14
retrieving revision 1.30.2.16
diff -u -r1.30.2.14 -r1.30.2.16
--- ruby-1.8.3/ext/syck/rubyext.c	17 Sep 2005 17:17:06 -0000	1.30.2.14
+++ ruby-1.8.3/ext/syck/rubyext.c	13 Oct 2005 14:30:53 -0000	1.30.2.16
@@ -76,6 +76,9 @@
 void rb_syck_output_handler _((SyckEmitter *, char *, long));
 void rb_syck_emitter_handler _((SyckEmitter *, st_data_t));
 int syck_parser_assign_io _((SyckParser *, VALUE));
+VALUE syck_scalar_alloc _((VALUE class));
+VALUE syck_seq_alloc _((VALUE class));
+VALUE syck_map_alloc _((VALUE class));
 
 struct parser_xtra {
     VALUE data;  /* Borrowed this idea from marshal.c to fix [ruby-core:8067] problem */
@@ -1336,7 +1339,7 @@
  */
 VALUE
 syck_domaintype_initialize( self, domain, type_id, val )
-    VALUE self, type_id, val;
+    VALUE self, domain, type_id, val;
 {
     rb_iv_set( self, "@domain", domain );
     rb_iv_set( self, "@type_id", type_id );
Index: pack.c
===================================================================
RCS file: /var/cvs/src/ruby/pack.c,v
retrieving revision 1.62.2.10
retrieving revision 1.62.2.11
diff -u -r1.62.2.10 -r1.62.2.11
--- ruby-1.8.3/pack.c	28 Feb 2005 02:45:19 -0000	1.62.2.10
+++ ruby-1.8.3/pack.c	7 Oct 2005 01:01:22 -0000	1.62.2.11
@@ -351,12 +351,12 @@
 # define EXTEND32(x) 
 #else
 /* invariant in modulo 1<<31 */
-# define EXTEND32(x) do {if (!natint) {(x) = (I32)(((1<<31)-1-(x))^~(~0<<31))}} while(0)
+# define EXTEND32(x) do {if (!natint) {(x) = (I32)(((1<<31)-1-(x))^~(~0<<31));}} while(0)
 #endif
 #if SIZEOF_SHORT == SIZE16
 # define EXTEND16(x) 
 #else
-# define EXTEND16(x) do { if (!natint) {(x) = (short)(((1<<15)-1-(x))^~(~0<<15))}} while(0)
+# define EXTEND16(x) do { if (!natint) {(x) = (short)(((1<<15)-1-(x))^~(~0<<15));}} while(0)
 #endif
 
 #ifdef HAVE_LONG_LONG


### Patch 110_document_update.patch
[ruby-core:05942][ruby-core:06027][ruby-core:06053][] document update.

Index: lib/delegate.rb
===================================================================
RCS file: /var/cvs/src/ruby/lib/delegate.rb,v
retrieving revision 1.14.2.10
retrieving revision 1.14.2.12
diff -u -r1.14.2.10 -r1.14.2.12
--- ruby-1.8.3/lib/delegate.rb	30 Jun 2005 15:22:00 -0000	1.14.2.10
+++ ruby-1.8.3/lib/delegate.rb	28 Sep 2005 14:23:24 -0000	1.14.2.12
@@ -1,23 +1,125 @@
-#  Delegation class that delegates even methods defined in super class,
-# which can not be covered with normal method_missing hack.
-#  
-#  Delegator is the abstract delegation class. Need to redefine
-# `__getobj__' method in the subclass.  SimpleDelegator is the 
-# concrete subclass for simple delegation.
-#
-# Usage:
-#   foo = Object.new
-#   foo2 = SimpleDelegator.new(foo)
-#   foo.hash == foo2.hash # => false
-#
-#   Foo = DelegateClass(Array)
-#
-#  class ExtArray<DelegateClass(Array)
-#    ...
-#  end
+# = delegate -- Support for the Delegation Pattern
+#
+# Documentation by James Edward Gray II and Gavin Sinclair
+#
+# == Introduction
+#
+# This library provides three different ways to delegate method calls to an
+# object.  The easiest to use is SimpleDelegator.  Pass an object to the
+# constructor and all methods supported by the object will be delegated.  This
+# object can be changed later.
+#
+# Going a step further, the top level DelegateClass method allows you to easily
+# setup delegation through class inheritance.  This is considerably more
+# flexible and thus probably the most common use for this library.
+#
+# Finally, if you need full control over the delegation scheme, you can inherit
+# from the abstract class Delegator and customize as needed.  (If you find
+# yourself needing this control, have a look at _forwardable_, also in the
+# standard library.  It may suit your needs better.)
+#
+# == Notes
+#
+# Be advised, RDoc will not detect delegated methods.
+#
+# <b>delegate.rb provides full-class delegation via the
+# DelegateClass() method.  For single-method delegation via
+# def_delegator(), see forwardable.rb.</b>
+#
+# == Examples
+#
+# === SimpleDelegator
+#
+# Here's a simple example that takes advantage of the fact that
+# SimpleDelegator's delegation object can be changed at any time.
+#
+#   class Stats
+#     def initialize
+#       @source = SimpleDelegator.new([])
+#     end
+#     
+#     def stats( records )
+#       @source.__setobj__(records)
+#       	
+#       "Elements:  #{@source.size}\n" +
+#       " Non-Nil:  #{@source.compact.size}\n" +
+#       "  Unique:  #{@source.uniq.size}\n"
+#     end
+#   end
+#   
+#   s = Stats.new
+#   puts s.stats(%w{James Edward Gray II})
+#   puts
+#   puts s.stats([1, 2, 3, nil, 4, 5, 1, 2])
+#
+# <i>Prints:</i>
+#
+#   Elements:  4
+#    Non-Nil:  4
+#     Unique:  4
+# 
+#   Elements:  8
+#    Non-Nil:  7
+#     Unique:  6
+#
+# === DelegateClass()
+#
+# Here's a sample of use from <i>tempfile.rb</i>.
+#
+# A _Tempfile_ object is really just a _File_ object with a few special rules
+# about storage location and/or when the File should be deleted.  That makes for
+# an almost textbook perfect example of how to use delegation.
+#
+#   class Tempfile < DelegateClass(File)
+#     # constant and class member data initialization...
+#   
+#     def initialize(basename, tmpdir=Dir::tmpdir)
+#       # build up file path/name in var tmpname...
+#     
+#       @tmpfile = File.open(tmpname, File::RDWR|File::CREAT|File::EXCL, 0600)
+#     
+#       # ...
+#     
+#       super(@tmpfile)
+#     
+#       # below this point, all methods of File are supported...
+#     end
+#   
+#     # ...
+#   end
+#
+# === Delegator
+#
+# SimpleDelegator's implementation serves as a nice example here.
+#
+#    class SimpleDelegator < Delegator
+#      def initialize(obj)
+#        super             # pass obj to Delegator constructor, required
+#        @_sd_obj = obj    # store obj for future use
+#      end
+# 
+#      def __getobj__
+#        @_sd_obj          # return object we are delegating to, required
+#      end
+# 
+#      def __setobj__(obj)
+#        @_sd_obj = obj    # change delegation object, a feature we're providing
+#      end
+# 
+#      # ...
+#    end
 
+#
+# Delegator is an abstract class used to build delegator pattern objects from
+# subclasses.  Subclasses should redefine \_\_getobj\_\_.  For a concrete
+# implementation, see SimpleDelegator.
+#
 class Delegator
 
+  #
+  # Pass in the _obj_ to delegate method calls to.  All methods supported by
+  # _obj_ will be delegated to.
+  #
   def initialize(obj)
     preserved = ::Kernel.public_instance_methods(false)
     preserved -= ["to_s","to_a","inspect","==","=~","==="]
@@ -49,6 +151,7 @@
   end
   alias initialize_methods initialize
 
+  # Handles the magic of delegation through \_\_getobj\_\_.
   def method_missing(m, *args)
     target = self.__getobj__
     unless target.respond_to?(m)
@@ -57,85 +160,130 @@
     target.__send__(m, *args)
   end
 
+  # 
+  # Checks for a method provided by this the delegate object by fowarding the 
+  # call through \_\_getobj\_\_.
+  # 
   def respond_to?(m)
     return true if super
     return self.__getobj__.respond_to?(m)
   end
 
+  #
+  # This method must be overridden by subclasses and should return the object
+  # method calls are being delegated to.
+  #
   def __getobj__
     raise NotImplementedError, "need to define `__getobj__'"
   end
 
+  # Serialization support for the object returned by \_\_getobj\_\_.
   def marshal_dump
     __getobj__
   end
+  # Reinitializes delegation from a serialized object.
   def marshal_load(obj)
     initialize_methods(obj)
   end
 end
 
+#
+# A concrete implementation of Delegator, this class provides the means to
+# delegate all supported method calls to the object passed into the constructor
+# and even to change the object being delegated to at a later time with
+# \_\_setobj\_\_ .
+#
 class SimpleDelegator<Delegator
 
+  # Pass in the _obj_ you would like to delegate method calls to.
   def initialize(obj)
     super
     @_sd_obj = obj
   end
 
+  # Returns the current object method calls are being delegated to.
   def __getobj__
     @_sd_obj
   end
 
+  #
+  # Changes the delegate object to _obj_.
+  #
+  # It's important to note that this does *not* cause SimpleDelegator's methods
+  # to change.  Because of this, you probably only want to change delegation
+  # to objects of the same type as the original delegate.
+  #
+  # Here's an example of changing the delegation object.
+  #
+  #   names = SimpleDelegator.new(%w{James Edward Gray II})
+  #   puts names[1]    # => Edward
+  #   names.__setobj__(%w{Gavin Sinclair})
+  #   puts names[1]    # => Sinclair
+  #
   def __setobj__(obj)
     raise ArgumentError, "cannot delegate to self" if self.equal?(obj)
     @_sd_obj = obj
   end
 
+  # Clone support for the object returned by \_\_getobj\_\_.
   def clone
     super
     __setobj__(__getobj__.clone)
   end
+  # Duplication support for the object returned by \_\_getobj\_\_.
   def dup(obj)
     super
     __setobj__(__getobj__.dup)
   end
 end
 
+# :stopdoc:
 # backward compatibility ^_^;;;
 Delegater = Delegator
 SimpleDelegater = SimpleDelegator
+# :startdoc:
 
 #
+# The primary interface to this library.  Use to setup delegation when defining
+# your class.
+#
+#   class MyClass < DelegateClass( ClassToDelegateTo )    # Step 1
+#     def initiaize
+#       super(obj_of_ClassToDelegateTo)                   # Step 2
+#     end
+#   end
+#
 def DelegateClass(superclass)
   klass = Class.new
   methods = superclass.public_instance_methods(true)
   methods -= ::Kernel.public_instance_methods(false)
   methods |= ["to_s","to_a","inspect","==","=~","==="]
   klass.module_eval {
-    def initialize(obj)
+    def initialize(obj)  # :nodoc:
       @_dc_obj = obj
     end
-    def method_missing(m, *args)
+    def method_missing(m, *args)  # :nodoc:
       unless @_dc_obj.respond_to?(m)
         super(m, *args)
       end
       @_dc_obj.__send__(m, *args)
     end
-    def respond_to?(m)
+    def respond_to?(m)  # :nodoc:
       return true if super
       return @_dc_obj.respond_to?(m)
     end
-    def __getobj__
+    def __getobj__  # :nodoc:
       @_dc_obj
     end
-    def __setobj__(obj)
+    def __setobj__(obj)  # :nodoc:
       raise ArgumentError, "cannot delegate to self" if self.equal?(obj)
       @_dc_obj = obj
     end
-    def clone
+    def clone  # :nodoc:
       super
       __setobj__(__getobj__.clone)
     end
-    def dup
+    def dup  # :nodoc:
       super
       __setobj__(__getobj__.dup)
     end
@@ -159,6 +307,8 @@
   return klass
 end
 
+# :enddoc:
+
 if __FILE__ == $0
   class ExtArray<DelegateClass(Array)
     def initialize()
Index: lib/forwardable.rb
===================================================================
RCS file: /var/cvs/src/ruby/lib/forwardable.rb,v
retrieving revision 1.2
retrieving revision 1.2.2.1
diff -u -r1.2 -r1.2.2.1
--- ruby-1.8.3/lib/forwardable.rb	3 Nov 2001 13:41:57 -0000	1.2
+++ ruby-1.8.3/lib/forwardable.rb	26 Sep 2005 13:59:45 -0000	1.2.2.1
@@ -1,45 +1,145 @@
+# = forwardable - Support for the Delegation Pattern
 #
-#   forwardable.rb - 
-#   	$Release Version: 1.1$
-#   	$Revision: 1.2 $
-#   	$Date: 2001/11/03 13:41:57 $
-#   	by Keiju ISHITSUKA(keiju@ishitsuka.com)
-#	original definition by delegator.rb
-# --
-# Usage:
+#    $Release Version: 1.1$
+#    $Revision: 1.2.2.1 $
+#    $Date: 2005/09/26 13:59:45 $
+#    by Keiju ISHITSUKA(keiju@ishitsuka.com)
 #
-#   class Foo
+#    Documentation by James Edward Gray II and Gavin Sinclair
+#
+# == Introduction
+#
+# This library allows you delegate method calls to an object, on a method by
+# method basis.  You can use Forwardable to setup this delegation at the class
+# level, or SingleForwardable to handle it at the object level.
+#
+# == Notes
+#
+# Be advised, RDoc will not detect delegated methods.
+#
+# <b>forwardable.rb provides single-method delegation via the
+# def_delegator() and def_delegators() methods.  For full-class
+# delegation via DelegateClass(), see delegate.rb.</b>
+#
+# == Examples
+#
+# === Forwardable
+#
+# Forwardable makes building a new class based on existing work, with a proper
+# interface, almost trivial.  We want to rely on what has come before obviously,
+# but with delegation we can take just the methods we need and even rename them
+# as appropriate.  In many cases this is preferable to inheritance, which gives
+# us the entire old interface, even if much of it isn't needed.
+#
+#   class Queue
 #     extend Forwardable
+#     
+#     def initialize
+#       @q = [ ]    # prepare delegate object
+#     end
+#     
+#     # setup prefered interface, enq() and deq()...
+#     def_delegator :@q, :push, :enq
+#     def_delegator :@q, :shift, :deq
+#     
+#     # support some general Array methods that fit Queues well
+#     def_delegators :@q, :clear, :first, :push, :shift, :size
+#   end
+# 
+#   q = Queue.new
+#   q.enq 1, 2, 3, 4, 5
+#   q.push 6
+# 
+#   q.shift    # => 1
+#   while q.size > 0
+#     puts q.deq
+#   end
+# 
+#   q.enq "Ruby", "Perl", "Python"
+#   puts q.first
+#   q.clear
+#   puts q.first
+#
+# <i>Prints:</i>
+#
+#   2
+#   3
+#   4
+#   5
+#   6
+#   Ruby
+#   nil
+#
+# === SingleForwardable
+#
+#    printer = String.new
+#    printer.extend SingleForwardable        # prepare object for delegation
+#    printer.def_delegator "STDOUT", "puts"  # add delegation for STDOUT.puts()
+#    printer.puts "Howdy!"
+#
+# <i>Prints:</i>
+#
+#    Howdy!
+
+#
+# The Forwardable module provides delegation of specified
+# methods to a designated object, using the methods #def_delegator
+# and #def_delegators.
+#
+# For example, say you have a class RecordCollection which
+# contains an array <tt>@records</tt>.  You could provide the lookup method
+# #record_number(), which simply calls #[] on the <tt>@records</tt>
+# array, like this:
+#
+#   class RecordCollection
+#     extends Forwardable
+#     def_delegator :@records, :[], :record_number
+#   end
+#
+# Further, if you wish to provide the methods #size, #<<, and #map,
+# all of which delegate to @records, this is how you can do it:
 #
-#     def_delegators("@out", "printf", "print")
-#     def_delegators(:@in, :gets)
-#     def_delegator(:@contents, :[], "content_at")
+#   class RecordCollection
+#     # extends Forwardable, but we did that above
+#     def_delegators :@records, :size, :<<, :map
 #   end
-#   f = Foo.new
-#   f.printf ...
-#   f.gets
-#   f.content_at(1)
-#
-#   g = Goo.new
-#   g.extend SingleForwardable
-#   g.def_delegator("@out", :puts)
-#   g.puts ...
 #
+# Also see the example at forwardable.rb.
 #
-
 module Forwardable
 
   @debug = nil
   class<<self
+    # force Forwardable to show up in stack backtraces of delegated methods
     attr_accessor :debug
   end
 
+  #
+  # Shortcut for defining multiple delegator methods, but with no
+  # provision for using a different name.  The following two code
+  # samples have the same effect:
+  #
+  #   def_delegators :@records, :size, :<<, :map
+  #
+  #   def_delegator :@records, :size
+  #   def_delegator :@records, :<<
+  #   def_delegator :@records, :map
+  #
+  # See the examples at Forwardable and forwardable.rb.
+  #
   def def_instance_delegators(accessor, *methods)
     for method in methods
       def_instance_delegator(accessor, method)
     end
   end
 
+  #
+  # Defines a method _method_ which delegates to _obj_ (i.e. it calls
+  # the method of the same name in _obj_).  If _new_name_ is
+  # provided, it is used as the name for the delegate method.
+  #
+  # See the examples at Forwardable and forwardable.rb.
+  #
   def def_instance_delegator(accessor, method, ali = method)
     accessor = accessor.id2name if accessor.kind_of?(Integer)
     method = method.id2name if method.kind_of?(Integer)
@@ -61,13 +161,41 @@
   alias def_delegator def_instance_delegator
 end
 
+#
+# The SingleForwardable module provides delegation of specified
+# methods to a designated object, using the methods #def_delegator
+# and #def_delegators.  This module is similar to Forwardable, but it works on
+# objects themselves, instead of their defining classes.
+#
+# Also see the example at forwardable.rb.
+#
 module SingleForwardable
+  #
+  # Shortcut for defining multiple delegator methods, but with no
+  # provision for using a different name.  The following two code
+  # samples have the same effect:
+  #
+  #   single_forwardable.def_delegators :@records, :size, :<<, :map
+  #
+  #   single_forwardable.def_delegator :@records, :size
+  #   single_forwardable.def_delegator :@records, :<<
+  #   single_forwardable.def_delegator :@records, :map
+  #
+  # See the example at forwardable.rb.
+  #
   def def_singleton_delegators(accessor, *methods)
     for method in methods
       def_singleton_delegator(accessor, method)
     end
   end
 
+  #
+  # Defines a method _method_ which delegates to _obj_ (i.e. it calls
+  # the method of the same name in _obj_).  If _new_name_ is
+  # provided, it is used as the name for the delegate method.
+  #
+  # See the example at forwardable.rb.
+  #
   def def_singleton_delegator(accessor, method, ali = method)
     accessor = accessor.id2name if accessor.kind_of?(Integer)
     method = method.id2name if method.kind_of?(Integer)
@@ -88,7 +216,3 @@
   alias def_delegators def_singleton_delegators
   alias def_delegator def_singleton_delegator
 end
-
-
-
-
Index: eval.c
===================================================================
RCS file: /var/cvs/src/ruby/eval.c,v
retrieving revision 1.616.2.128
retrieving revision 1.616.2.129
diff -u -r1.616.2.128 -r1.616.2.129
--- ruby-1.8.3/eval.c	28 Sep 2005 15:58:30 -0000	1.616.2.128
+++ ruby-1.8.3/eval.c	28 Sep 2005 22:22:44 -0000	1.616.2.129
@@ -5930,7 +5930,7 @@
  *     obj.__send__(symbol [, args...])    => obj
  *  
  *  Invokes the method identified by _symbol_, passing it any
- *  arguments specified. You can use <code>__send__</code> if the name
+ *  arguments specified. You can use <code>\_\_send__</code> if the name
  *  +send+ clashes with an existing method in _obj_.
  *     
  *     class Klass
Index: class.c
===================================================================
RCS file: /var/cvs/src/ruby/class.c,v
retrieving revision 1.80.2.5
retrieving revision 1.80.2.6
diff -u -r1.80.2.5 -r1.80.2.6
--- ruby-1.8.3/class.c	16 May 2005 13:28:59 -0000	1.80.2.5
+++ ruby-1.8.3/class.c	28 Sep 2005 14:42:46 -0000	1.80.2.6
@@ -58,6 +58,7 @@
     return ST_CONTINUE;
 }
 
+/* :nodoc: */
 VALUE
 rb_mod_init_copy(clone, orig)
     VALUE clone, orig;
@@ -85,6 +86,7 @@
     return clone;
 }
 
+/* :nodoc: */
 VALUE
 rb_class_init_copy(clone, orig)
     VALUE clone, orig;
Index: numeric.c
===================================================================
RCS file: /var/cvs/src/ruby/numeric.c,v
retrieving revision 1.101.2.18
retrieving revision 1.101.2.19
diff -u -r1.101.2.18 -r1.101.2.19
--- ruby-1.8.3/numeric.c	22 Jul 2005 13:04:24 -0000	1.101.2.18
+++ ruby-1.8.3/numeric.c	28 Sep 2005 14:42:46 -0000	1.101.2.19
@@ -198,6 +198,7 @@
     return Qnil;		/* not reached */
 }
 
+/* :nodoc: */
 static VALUE
 num_init_copy(x, y)
     VALUE x, y;
Index: object.c
===================================================================
RCS file: /var/cvs/src/ruby/object.c,v
retrieving revision 1.134.2.35
retrieving revision 1.134.2.36
diff -u -r1.134.2.35 -r1.134.2.36
--- ruby-1.8.3/object.c	8 Sep 2005 05:59:33 -0000	1.134.2.35
+++ ruby-1.8.3/object.c	28 Sep 2005 14:42:46 -0000	1.134.2.36
@@ -303,6 +303,7 @@
     return dup;
 }
 
+/* :nodoc: */
 VALUE
 rb_obj_init_copy(obj, orig)
     VALUE obj, orig;
@@ -2701,7 +2702,7 @@
     rb_define_method(rb_cModule, "<=", rb_class_inherited_p, 1);
     rb_define_method(rb_cModule, ">",  rb_mod_gt, 1);
     rb_define_method(rb_cModule, ">=", rb_mod_ge, 1);
-    rb_define_method(rb_cModule, "initialize_copy", rb_mod_init_copy, 1);
+    rb_define_method(rb_cModule, "initialize_copy", rb_mod_init_copy, 1); /* in class.c */
     rb_define_method(rb_cModule, "to_s", rb_mod_to_s, 0);
     rb_define_method(rb_cModule, "included_modules", 
 		     rb_mod_included_modules, 0); /* in class.c */
@@ -2743,7 +2744,7 @@
     rb_define_method(rb_cClass, "allocate", rb_obj_alloc, 0);
     rb_define_method(rb_cClass, "new", rb_class_new_instance, -1);
     rb_define_method(rb_cClass, "initialize", rb_class_initialize, -1);
-    rb_define_method(rb_cClass, "initialize_copy", rb_class_init_copy, 1);
+    rb_define_method(rb_cClass, "initialize_copy", rb_class_init_copy, 1); /* in class.c */
     rb_define_method(rb_cClass, "superclass", rb_class_superclass, 0);
     rb_define_alloc_func(rb_cClass, rb_class_s_alloc);
     rb_undef_method(rb_cClass, "extend_object");
Index: re.c
===================================================================
RCS file: /var/cvs/src/ruby/re.c,v
retrieving revision 1.114.2.14
retrieving revision 1.114.2.15
diff -u -r1.114.2.14 -r1.114.2.15
--- ruby-1.8.3/re.c	23 May 2005 03:24:28 -0000	1.114.2.14
+++ ruby-1.8.3/re.c	28 Sep 2005 14:42:46 -0000	1.114.2.15
@@ -680,6 +680,7 @@
     return (VALUE)match;
 }
 
+/* :nodoc: */
 static VALUE
 match_init_copy(obj, orig)
     VALUE obj, orig;
@@ -1974,6 +1975,7 @@
     }
 }
 
+/* :nodoc: */
 static VALUE
 rb_reg_init_copy(copy, re)
     VALUE copy, re;
Index: time.c
===================================================================
RCS file: /var/cvs/src/ruby/time.c,v
retrieving revision 1.90.2.10
retrieving revision 1.90.2.11
diff -u -r1.90.2.10 -r1.90.2.11
--- ruby-1.8.3/time.c	20 Jan 2005 09:34:36 -0000	1.90.2.10
+++ ruby-1.8.3/time.c	28 Sep 2005 14:42:46 -0000	1.90.2.11
@@ -1031,6 +1031,7 @@
     return LONG2FIX(hash);
 }
 
+/* :nodoc: */
 static VALUE
 time_init_copy(copy, time)
     VALUE copy, time;
Index: file.c
===================================================================
RCS file: /var/cvs/src/ruby/file.c,v
retrieving revision 1.169.2.22
retrieving revision 1.169.2.25
diff -u -r1.169.2.22 -r1.169.2.25
--- ruby-1.8.3/file.c	28 Sep 2005 14:41:58 -0000	1.169.2.22
+++ ruby-1.8.3/file.c	12 Oct 2005 02:26:52 -0000	1.169.2.25
@@ -3072,14 +3072,14 @@
  *     ?d  | boolean | True if file1 exists and is a directory
  *     ?e  | boolean | True if file1 exists
  *     ?f  | boolean | True if file1 exists and is a regular file
- *     ?g  | boolean | True if files has the \CF{setgid} bit
+ *     ?g  | boolean | True if file1 has the \CF{setgid} bit
  *         |         | set (false under NT)
  *     ?G  | boolean | True if file1 exists and has a group
  *         |         | ownership equal to the caller's group
  *     ?k  | boolean | True if file1 exists and has the sticky bit set
- *     ?l  | boolean | True if files exists and is a symbolic link
+ *     ?l  | boolean | True if file1 exists and is a symbolic link
  *     ?M  | Time    | Last modification time for file1
- *     ?o  | boolean | True if files exists and is owned by 
+ *     ?o  | boolean | True if file1 exists and is owned by 
  *         |         | the caller's effective uid
  *     ?O  | boolean | True if file1 exists and is owned by
  *         |         | the caller's real uid
@@ -3088,7 +3088,7 @@
  *         |         | uid/gid of the caller
  *     ?R  | boolean | True if file is readable by the real
  *         |         | uid/gid of the caller
- *     ?s  | int/nil | If files has nonzero size, return the size,
+ *     ?s  | int/nil | If file1 has nonzero size, return the size,
  *         |         | otherwise return nil
  *     ?S  | boolean | True if file1 exists and is a socket
  *     ?u  | boolean | True if file1 has the setuid bit set
@@ -3304,6 +3304,7 @@
     return Qnil;
 }
 
+/* :nodoc: */
 static VALUE
 rb_stat_init_copy(copy, orig)
     VALUE copy, orig;
Index: bin/erb
===================================================================
RCS file: /var/cvs/src/ruby/bin/erb,v
retrieving revision 1.3.2.1
retrieving revision 1.3.2.3
diff -u -r1.3.2.1 -r1.3.2.3
--- ruby-1.8.3/bin/erb	11 Sep 2005 13:14:12 -0000	1.3.2.1
+++ ruby-1.8.3/bin/erb	15 Oct 2005 10:57:50 -0000	1.3.2.3
@@ -102,12 +102,12 @@
   -x               print ruby script
   -n               print ruby script with line number
   -v               enable verbose mode
-  -d               set $DBEUG to true
+  -d               set $DEBUG to true
   -r [library]     load a library
   -K [kcode]       specify KANJI code-set
   -S [safe_level]  set $SAFE (0..4)
   -T [trim_mode]   specify trim_mode (0..2, -)
-  -P               disregard the lin which starts in "%" 
+  -P               ignore lines which start with "%"
 EOU
         exit 1
       end


### Patch 111_moreinfo_unknown_node_type.patch
[ruby-dev:26196] show more information of "unknown node type".

Index: eval.c
===================================================================
RCS file: /var/cvs/src/ruby/eval.c,v
retrieving revision 1.616.2.125
retrieving revision 1.616.2.126
diff -u -r1.616.2.125 -r1.616.2.126
--- ruby-1.8.3/eval.c	20 Sep 2005 09:26:35 -0000	1.616.2.125
+++ ruby-1.8.3/eval.c	24 Sep 2005 15:20:06 -0000	1.616.2.126
@@ -2814,7 +2814,15 @@
     NODE *volatile node;
 {
     ruby_current_node = 0;
-    rb_bug("unknown node type %d", nd_type(node));
+    if (node->flags == 0) {
+        rb_bug("terminated node (0x%lx)", node);
+    }
+    else if (BUILTIN_TYPE(node) != T_NODE) {
+        rb_bug("not a node 0x%02lx (0x%lx)", BUILTIN_TYPE(node), node);
+    }
+    else {
+        rb_bug("unknown node type %d (0x%lx)", nd_type(node), node);
+    }
 }
 
 static VALUE


### Patch 112_prevent_add_trace-func.patch
[] add rb_secure(4) to prevent adding tracing function.

Index: eval.c
===================================================================
RCS file: /var/cvs/src/ruby/eval.c,v
retrieving revision 1.616.2.126
retrieving revision 1.616.2.127
diff -u -r1.616.2.126 -r1.616.2.127
--- ruby-1.8.3/eval.c	24 Sep 2005 15:20:06 -0000	1.616.2.126
+++ ruby-1.8.3/eval.c	26 Sep 2005 13:59:45 -0000	1.616.2.127
@@ -2545,6 +2545,7 @@
 {
     rb_event_hook_t *hook;
 
+    rb_secure(4);
     if (NIL_P(trace)) {
 	trace_func = 0;
 	rb_remove_event_hook(call_trace_func);


### Patch 113_autoload.patch
[ruby-dev:27331] now return true if autoload succeeded.

Index: eval.c
===================================================================
RCS file: /var/cvs/src/ruby/eval.c,v
retrieving revision 1.616.2.127
retrieving revision 1.616.2.128
diff -u -r1.616.2.127 -r1.616.2.128
--- ruby-1.8.3/eval.c	26 Sep 2005 13:59:45 -0000	1.616.2.127
+++ ruby-1.8.3/eval.c	28 Sep 2005 15:58:30 -0000	1.616.2.128
@@ -1837,7 +1837,7 @@
 	if (NIL_P(klass)) return rb_const_get(CLASS_OF(self), id);
 	while (RCLASS(klass)->iv_tbl && st_lookup(RCLASS(klass)->iv_tbl, id, &result)) {
 	    if (result == Qundef) {
-		rb_autoload_load(klass, id);
+		if (!RTEST(rb_autoload_load(klass, id))) break;
 		continue;
 	    }
 	    return result;
Index: intern.h
===================================================================
RCS file: /var/cvs/src/ruby/intern.h,v
retrieving revision 1.139.2.13
retrieving revision 1.139.2.14
diff -u -r1.139.2.13 -r1.139.2.14
--- ruby-1.8.3/intern.h	12 Jun 2005 16:58:41 -0000	1.139.2.13
+++ ruby-1.8.3/intern.h	28 Sep 2005 15:58:32 -0000	1.139.2.14
@@ -449,7 +449,7 @@
 void rb_name_class _((VALUE, ID));
 VALUE rb_class_name _((VALUE));
 void rb_autoload _((VALUE, ID, const char*));
-void rb_autoload_load _((VALUE, ID));
+VALUE rb_autoload_load _((VALUE, ID));
 VALUE rb_autoload_p _((VALUE, ID));
 void rb_gc_mark_global_tbl _((void));
 VALUE rb_f_trace_var _((int, VALUE*));
Index: variable.c
===================================================================
RCS file: /var/cvs/src/ruby/variable.c,v
retrieving revision 1.108.2.12
retrieving revision 1.108.2.13
diff -u -r1.108.2.12 -r1.108.2.13
--- ruby-1.8.3/variable.c	27 Jul 2005 03:07:01 -0000	1.108.2.12
+++ ruby-1.8.3/variable.c	28 Sep 2005 15:58:32 -0000	1.108.2.13
@@ -1329,7 +1329,7 @@
     return (NODE *)load;
 }
 
-void
+VALUE
 rb_autoload_load(klass, id)
     VALUE klass;
     ID id;
@@ -1338,9 +1338,9 @@
     NODE *load = autoload_delete(klass, id);
 
     if (!load || !(file = load->nd_lit) || rb_provided(RSTRING(file)->ptr)) {
-	const_missing(klass, id);
+	return Qfalse;
     }
-    rb_require_safe(file, load->nd_nth);
+    return rb_require_safe(file, load->nd_nth);
 }
 
 static VALUE
@@ -1406,7 +1406,7 @@
     while (tmp) {
 	while (RCLASS(tmp)->iv_tbl && st_lookup(RCLASS(tmp)->iv_tbl,id,&value)) {
 	    if (value == Qundef) {
-		rb_autoload_load(tmp, id);
+		if (!RTEST(rb_autoload_load(tmp, id))) break;
 		continue;
 	    }
 	    if (exclude && tmp == rb_cObject && klass != rb_cObject) {


### Patch 114_binding_dup.patch
[yarv-dev:666] add Binding#dup method.

Index: eval.c
===================================================================
RCS file: /var/cvs/src/ruby/eval.c,v
retrieving revision 1.616.2.129
retrieving revision 1.616.2.130
diff -u -r1.616.2.129 -r1.616.2.130
--- ruby-1.8.3/eval.c	28 Sep 2005 22:22:44 -0000	1.616.2.129
+++ ruby-1.8.3/eval.c	8 Oct 2005 11:05:08 -0000	1.616.2.130
@@ -9392,6 +9392,7 @@
 	noex = NOEX_PUBLIC;
     }
     rb_add_method(mod, id, node, noex);
+    rb_define_method(rb_cBinding, "dup", proc_dup, 0);
     return body;
 }
 


### Patch 115_respond_to.patch
[ruby-dev:27408] check if obj responds to the given method with the given visibility.
[ruby-dev:27411] conform to Object#respond_to?.

Index: eval.c
===================================================================
RCS file: /var/cvs/src/ruby/eval.c,v
retrieving revision 1.616.2.130
retrieving revision 1.616.2.131
diff -u -r1.616.2.130 -r1.616.2.131
--- ruby-1.8.3/eval.c	8 Oct 2005 11:05:08 -0000	1.616.2.130
+++ ruby-1.8.3/eval.c	11 Oct 2005 12:42:49 -0000	1.616.2.131
@@ -4100,21 +4100,32 @@
 static NODE *basic_respond_to = 0;
 
 int
-rb_respond_to(obj, id)
+rb_obj_respond_to(obj, id, priv)
     VALUE obj;
     ID id;
+    int priv;
 {
     VALUE klass = CLASS_OF(obj);
-    if (rb_method_node(klass, respond_to) == basic_respond_to &&
-	rb_method_boundp(klass, id, 0)) {
-	return Qtrue;
+
+    if (rb_method_node(klass, respond_to) == basic_respond_to) {
+	return rb_method_boundp(klass, id, !priv);
     }
-    else{
-	return rb_funcall(obj, respond_to, 1, ID2SYM(id));
+    else {
+	VALUE args[2];
+	int n = 0;
+	args[n++] = ID2SYM(id);
+	if (priv) args[n++] = Qtrue;
+	return rb_funcall2(obj, respond_to, n, args);
     }
-    return Qfalse;
 }
 
+int
+rb_respond_to(obj, id)
+    VALUE obj;
+    ID id;
+{
+    return rb_obj_respond_to(obj, id, Qfalse);
+}
 
 /*
  *  call-seq:
@@ -4126,7 +4137,7 @@
  */
 
 static VALUE
-rb_obj_respond_to(argc, argv, obj)
+obj_respond_to(argc, argv, obj)
     int argc;
     VALUE *argv;
     VALUE obj;
@@ -7760,7 +7771,7 @@
     rb_define_global_function("method_missing", rb_method_missing, -1);
     rb_define_global_function("loop", rb_f_loop, 0);
 
-    rb_define_method(rb_mKernel, "respond_to?", rb_obj_respond_to, -1);
+    rb_define_method(rb_mKernel, "respond_to?", obj_respond_to, -1);
     respond_to   = rb_intern("respond_to?");
     basic_respond_to = rb_method_node(rb_cObject, respond_to);
     rb_global_variable((VALUE*)&basic_respond_to);
Index: intern.h
===================================================================
RCS file: /var/cvs/src/ruby/intern.h,v
retrieving revision 1.139.2.14
retrieving revision 1.139.2.15
diff -u -r1.139.2.14 -r1.139.2.15
--- ruby-1.8.3/intern.h	28 Sep 2005 15:58:32 -0000	1.139.2.14
+++ ruby-1.8.3/intern.h	11 Oct 2005 12:42:50 -0000	1.139.2.15
@@ -164,6 +164,7 @@
 void rb_dvar_push _((ID, VALUE));
 VALUE *rb_svar _((int));
 VALUE rb_eval_cmd _((VALUE, VALUE, int));
+int rb_obj_respond_to _((VALUE, ID, int));
 int rb_respond_to _((VALUE, ID));
 void rb_interrupt _((void));
 VALUE rb_apply _((VALUE, ID, VALUE));


### Patch 116_pthread_setitimer.patch
[] fixed build problem with --enable-pthread on platforms which don't have setitimer().

Index: rubysig.h
===================================================================
RCS file: /var/cvs/src/ruby/rubysig.h,v
retrieving revision 1.18.2.2
retrieving revision 1.18.2.3
diff -u -r1.18.2.2 -r1.18.2.3
--- ruby-1.8.3/rubysig.h	6 Dec 2004 08:19:20 -0000	1.18.2.2
+++ ruby-1.8.3/rubysig.h	27 Sep 2005 18:24:11 -0000	1.18.2.3
@@ -79,7 +79,7 @@
 
 RUBY_EXTERN int rb_thread_critical;
 void rb_thread_schedule _((void));
-#if defined(HAVE_SETITIMER)
+#if defined(HAVE_SETITIMER) || defined(_THREAD_SAFE)
 RUBY_EXTERN int rb_thread_pending;
 # define CHECK_INTS do {\
     if (!rb_prohibit_interrupt) {\


### Patch 117_check_eof.patch
[ruby-dev:27334] check if reached EOF.

Index: io.c
===================================================================
RCS file: /var/cvs/src/ruby/io.c,v
retrieving revision 1.246.2.88
retrieving revision 1.246.2.89
diff -u -r1.246.2.88 -r1.246.2.89
--- ruby-1.8.3/io.c	27 Sep 2005 07:08:37 -0000	1.246.2.88
+++ ruby-1.8.3/io.c	27 Sep 2005 23:12:43 -0000	1.246.2.89
@@ -1020,8 +1020,10 @@
     if (n > len) n = len;
     return fread(ptr, 1, n, f);
 #else
-    for (n = 0; n < len && READ_DATA_PENDING(f); ++n) {
-        *ptr++ = getc(f);
+    int c;
+
+    for (n = 0; n < len && READ_DATA_PENDING(f) && (c = getc(f)) != EOF; ++n) {
+	*ptr++ = c;
     }
     return n;
 #endif


### Patch 118_prototype_and_const.patch
[] add prototypes and constified.

Index: file.c
===================================================================
RCS file: /var/cvs/src/ruby/file.c,v
retrieving revision 1.169.2.21
retrieving revision 1.169.2.22
diff -u -r1.169.2.21 -r1.169.2.22
--- ruby-1.8.3/file.c	20 Sep 2005 17:51:20 -0000	1.169.2.21
+++ ruby-1.8.3/file.c	28 Sep 2005 14:41:58 -0000	1.169.2.22
@@ -2,8 +2,8 @@
 
   file.c -
 
-  $Author: matz $
-  $Date: 2005/09/20 17:51:20 $
+  $Author: nobu $
+  $Date: 2005/09/28 14:41:58 $
   created at: Mon Nov 15 12:24:34 JST 1993
 
   Copyright (C) 1993-2003 Yukihiro Matsumoto
@@ -75,9 +75,10 @@
 VALUE rb_mFileTest;
 static VALUE rb_cStat;
 
+static long apply2files _((void (*)(const char *, void *), VALUE, void *));
 static long
 apply2files(func, vargs, arg)
-    void (*func)();
+    void (*func)_((const char *, void *));
     VALUE vargs;
     void *arg;
 {
@@ -179,6 +180,20 @@
     return Qnil;
 }
 
+static VALUE rb_stat_dev _((VALUE));
+static VALUE rb_stat_ino _((VALUE));
+static VALUE rb_stat_mode _((VALUE));
+static VALUE rb_stat_nlink _((VALUE));
+static VALUE rb_stat_uid _((VALUE));
+static VALUE rb_stat_gid _((VALUE));
+static VALUE rb_stat_rdev _((VALUE));
+static VALUE rb_stat_size _((VALUE));
+static VALUE rb_stat_blksize _((VALUE));
+static VALUE rb_stat_blocks _((VALUE));
+static VALUE rb_stat_atime _((VALUE));
+static VALUE rb_stat_mtime _((VALUE));
+static VALUE rb_stat_ctime _((VALUE));
+
 /*
  *  call-seq:
  *     stat.dev    => fixnum
@@ -547,23 +562,23 @@
 {
     VALUE str;
     int i;
-    static struct {
-        char *name;
-        VALUE (*func)();
+    static const struct {
+	const char *name;
+	VALUE (*func)_((VALUE));
     } member[] = {
-        {"dev",     rb_stat_dev},
-        {"ino",     rb_stat_ino},
-        {"mode",    rb_stat_mode},
-        {"nlink",   rb_stat_nlink},
-        {"uid",     rb_stat_uid},
-        {"gid",     rb_stat_gid},
-        {"rdev",    rb_stat_rdev},
-        {"size",    rb_stat_size},
-        {"blksize", rb_stat_blksize},
-        {"blocks",  rb_stat_blocks},
-        {"atime",   rb_stat_atime},
-        {"mtime",   rb_stat_mtime},
-        {"ctime",   rb_stat_ctime},
+	{"dev",	    rb_stat_dev},
+	{"ino",	    rb_stat_ino},
+	{"mode",    rb_stat_mode},
+	{"nlink",   rb_stat_nlink},
+	{"uid",	    rb_stat_uid},
+	{"gid",	    rb_stat_gid},
+	{"rdev",    rb_stat_rdev},
+	{"size",    rb_stat_size},
+	{"blksize", rb_stat_blksize},
+	{"blocks",  rb_stat_blocks},
+	{"atime",   rb_stat_atime},
+	{"mtime",   rb_stat_mtime},
+	{"ctime",   rb_stat_ctime},
     };
 
     str = rb_str_buf_new2("#<");
@@ -1545,12 +1560,13 @@
     return rb_time_new(st.st_ctime, 0);
 }
 
+static void chmod_internal _((const char *, void *));
 static void
 chmod_internal(path, mode)
     const char *path;
-    int mode;
+    void *mode;
 {
-    if (chmod(path, mode) < 0)
+    if (chmod(path, (int)mode) < 0)
 	rb_sys_fail(path);
 }
 
@@ -1622,12 +1638,13 @@
 }
 
 #if defined(HAVE_LCHMOD)
+static void lchmod_internal _((const char *, void *));
 static void
 lchmod_internal(path, mode)
     const char *path;
-    int mode;
+    void *mode;
 {
-    if (lchmod(path, mode) < 0)
+    if (lchmod(path, (int)mode) < 0)
 	rb_sys_fail(path);
 }
 
@@ -1823,13 +1840,16 @@
 
 struct timeval rb_time_timeval();
 
+static void utime_internal _((const char *, void *));
+
 #if defined(HAVE_UTIMES) && !defined(__CHECKER__)
 
 static void
-utime_internal(path, tvp)
-    char *path;
-    struct timeval tvp[];
+utime_internal(path, arg)
+    const char *path;
+    void *arg;
 {
+    struct timeval *tvp = arg;
     if (utimes(path, tvp) < 0)
 	rb_sys_fail(path);
 }
@@ -1871,10 +1891,11 @@
 #endif
 
 static void
-utime_internal(path, utp)
+utime_internal(path, arg)
     const char *path;
-    struct utimbuf *utp;
+    void *arg;
 {
+    struct utimbuf *utp = arg;
     if (utime(path, utp) < 0)
 	rb_sys_fail(path);
 }
@@ -2012,9 +2033,11 @@
 #endif
 }
 
+static void unlink_internal _((const char *, void *));
 static void
-unlink_internal(path)
+unlink_internal(path, arg)
     const char *path;
+    void *arg;
 {
     if (unlink(path) < 0)
 	rb_sys_fail(path);
@@ -2625,8 +2648,8 @@
     name = StringValueCStr(fname);
     root = skiproot(name);
 #ifdef DOSISH_UNC
-    if (root > name + 2 && isdirsep(*name))
-	name = root - 2;
+    if (root > name + 1 && isdirsep(*name))
+	root = skipprefix(name = root - 2);
 #else
     if (root > name + 1)
 	name = root - 1;
@@ -2915,23 +2938,23 @@
     OpenFile *fptr;
 {
     if (rb_thread_alone() || (op & LOCK_NB)) {
-        int ret;
-        TRAP_BEG;
+	int ret;
+	TRAP_BEG;
 	ret = flock(fd, op);
-        TRAP_END;
+	TRAP_END;
 	return ret;
     }
     op |= LOCK_NB;
     while (flock(fd, op) < 0) {
 	switch (errno) {
-          case EAGAIN:
-          case EACCES:
+	  case EAGAIN:
+	  case EACCES:
 #if defined(EWOULDBLOCK) && EWOULDBLOCK != EAGAIN
 	  case EWOULDBLOCK:
 #endif
 	    rb_thread_polling();	/* busy wait */
 	    rb_io_check_closed(fptr);
-            continue;
+	    continue;
 	  default:
 	    return -1;
 	}
@@ -2988,19 +3011,19 @@
     }
   retry:
     if (flock(fileno(fptr->f), op) < 0) {
-        switch (errno) {
-          case EAGAIN:
-          case EACCES:
+	switch (errno) {
+	  case EAGAIN:
+	  case EACCES:
 #if defined(EWOULDBLOCK) && EWOULDBLOCK != EAGAIN
-          case EWOULDBLOCK:
+	  case EWOULDBLOCK:
 #endif
-              return Qfalse;
+	      return Qfalse;
 	  case EINTR:
 #if defined(ERESTART)
 	  case ERESTART:
 #endif
 	    goto retry;
-        }
+	}
 	rb_sys_fail(fptr->path);
     }
 #endif
@@ -3206,7 +3229,7 @@
 	  case '-':
 	    if (st1.st_dev == st2.st_dev && st1.st_ino == st2.st_ino)
 		return Qtrue;
-            return Qfalse;
+	    return Qfalse;
 
 	  case '=':
 	    if (st1.st_mtime == st2.st_mtime) return Qtrue;
@@ -3219,7 +3242,7 @@
 	  case '<':
 	    if (st1.st_mtime < st2.st_mtime) return Qtrue;
 	    return Qfalse;
-        }
+	}
     }
     /* unknown command */
     rb_raise(rb_eArgError, "unknown command ?%c", cmd);


### Patch 119_abolish_sizeof_file.patch
[ruby-dev:27317] abolish sizeof(FILE).

Index: ext/dl/dl.c
===================================================================
RCS file: /var/cvs/src/ruby/ext/dl/dl.c,v
retrieving revision 1.20.2.3
retrieving revision 1.20.2.4
diff -u -r1.20.2.3 -r1.20.2.4
--- ruby-1.8.3/ext/dl/dl.c	19 Sep 2005 09:38:39 -0000	1.20.2.3
+++ ruby-1.8.3/ext/dl/dl.c	29 Sep 2005 01:29:24 -0000	1.20.2.4
@@ -557,11 +557,7 @@
   GetOpenFile(self, fptr);
   fp = fptr->f;
 
-#if defined(__DragonFly__)
   return fp ? rb_dlptr_new(fp, 0, 0) : Qnil;
-#else
-  return fp ? rb_dlptr_new(fp, sizeof(FILE), 0) : Qnil;
-#endif
 }
 
 VALUE


### Patch 120_tktable_call_border.patch
[] border_* instance methods don't call 'border' subcommands.

Index: ext/tk/lib/tkextlib/tktable/tktable.rb
===================================================================
RCS file: /var/cvs/src/ruby/ext/tk/lib/tkextlib/tktable/tktable.rb,v
retrieving revision 1.1.2.7
retrieving revision 1.1.2.8
diff -u -r1.1.2.7 -r1.1.2.8
--- ruby-1.8.3/ext/tk/lib/tkextlib/tktable/tktable.rb	9 Aug 2005 06:16:04 -0000	1.1.2.7
+++ ruby-1.8.3/ext/tk/lib/tkextlib/tktable/tktable.rb	3 Oct 2005 15:34:26 -0000	1.1.2.8
@@ -360,16 +360,16 @@
   end
 
   def border_mark(x, y)
-    simplelist(tk_send('scan', 'mark', x, y))
+    simplelist(tk_send('border', 'mark', x, y))
   end
   def border_mark_row(x, y)
-    tk_send('scan', 'mark', x, y, 'row')
+    tk_send('border', 'mark', x, y, 'row')
   end
   def border_mark_col(x, y)
-    tk_send('scan', 'mark', x, y, 'col')
+    tk_send('border', 'mark', x, y, 'col')
   end
   def border_dragto(x, y)
-    tk_send('scan', 'dragto', x, y)
+    tk_send('border', 'dragto', x, y)
   end
 
   def clear_cache(first=None, last=None)


### Patch 121_strscan.c_remove_useless_code.patch
[ruby-dev:26368][ruby-dev:27389] remove useless code.

Index: ext/strscan/strscan.c
===================================================================
RCS file: /var/cvs/src/ruby/ext/strscan/strscan.c,v
retrieving revision 1.7.2.6
retrieving revision 1.7.2.7
diff -u -r1.7.2.6 -r1.7.2.7
--- ruby-1.8.3/ext/strscan/strscan.c	4 Mar 2005 10:39:45 -0000	1.7.2.6
+++ ruby-1.8.3/ext/strscan/strscan.c	6 Oct 2005 11:13:36 -0000	1.7.2.7
@@ -8,7 +8,7 @@
     You can distribute/modify this program under the terms of
     the Ruby License. For details, see the file COPYING.
 
-    $Id: strscan.c,v 1.7.2.6 2005/03/04 10:39:45 nobu Exp $
+    $Id: strscan.c,v 1.7.2.7 2005/10/06 11:13:36 aamine Exp $
 
 */
 
@@ -176,7 +176,6 @@
     struct strscanner *p;
 {
     re_free_registers(&(p->regs));
-    memset(p, sizeof(struct strscanner), 0);
     free(p);
 }
 
@@ -1297,7 +1296,7 @@
     tmp = rb_str_new2(STRSCAN_VERSION);
     rb_obj_freeze(tmp);
     rb_const_set(StringScanner, rb_intern("Version"), tmp);
-    tmp = rb_str_new2("$Id: strscan.c,v 1.7.2.6 2005/03/04 10:39:45 nobu Exp $");
+    tmp = rb_str_new2("$Id: strscan.c,v 1.7.2.7 2005/10/06 11:13:36 aamine Exp $");
     rb_obj_freeze(tmp);
     rb_const_set(StringScanner, rb_intern("Id"), tmp);
     


### Patch 122_parse.y_parser_stack.patch
[ruby-list:41199] manage parser stack on heap.
[ruby-core:06261] bison allocates indivisible size.
[ruby-dev:27428] byacc never free parser stack.

Index: parse.y
===================================================================
RCS file: /var/cvs/src/ruby/parse.y,v
retrieving revision 1.307.2.15
retrieving revision 1.307.2.18
diff -u -r1.307.2.15 -r1.307.2.18
--- ruby-1.8.3/parse.y	11 May 2005 16:21:33 -0000	1.307.2.15
+++ ruby-1.8.3/parse.y	17 Oct 2005 12:19:28 -0000	1.307.2.18
@@ -13,6 +13,10 @@
 %{
 
 #define YYDEBUG 1
+#define YYERROR_VERBOSE 1
+#ifndef YYSTACK_USE_ALLOCA
+#define YYSTACK_USE_ALLOCA 0
+#endif
 
 #include "ruby.h"
 #include "env.h"
@@ -23,6 +27,19 @@
 #include <errno.h>
 #include <ctype.h>
 
+#define YYMALLOC	rb_parser_malloc
+#define YYREALLOC	rb_parser_realloc
+#define YYCALLOC	rb_parser_calloc
+#define YYFREE 	rb_parser_free
+#define malloc	YYMALLOC
+#define realloc	YYREALLOC
+#define calloc	YYCALLOC
+#define free	YYFREE
+static void *rb_parser_malloc _((size_t));
+static void *rb_parser_realloc _((void *, size_t));
+static void *rb_parser_calloc _((size_t, size_t));
+static void rb_parser_free _((void *));
+
 #define yyparse ruby_yyparse
 #define yylex ruby_yylex
 #define yyerror ruby_yyerror
@@ -2528,6 +2545,9 @@
 int ruby__end__seen;
 
 static VALUE ruby_debug_lines;
+#ifdef YYMALLOC
+static NODE *parser_heap;
+#endif
 
 static NODE*
 yycompile(f, line)
@@ -2992,7 +3012,7 @@
 dispose_string(str)
     VALUE str;
 {
-    free(RSTRING(str)->ptr);
+    xfree(RSTRING(str)->ptr);
     rb_gc_force_recycle(str);
 }
 
@@ -5563,11 +5583,11 @@
     struct local_vars *local = lvtbl->prev;
 
     if (lvtbl->tbl) {
-	if (!lvtbl->nofree) free(lvtbl->tbl);
+	if (!lvtbl->nofree) xfree(lvtbl->tbl);
 	else lvtbl->tbl[0] = lvtbl->cnt;
     }
     ruby_dyna_vars = lvtbl->dyna_vars;
-    free(lvtbl);
+    xfree(lvtbl);
     lvtbl = local;
 }
 
@@ -5675,7 +5695,7 @@
 		rb_mem_clear(ruby_scope->local_vars+i, len-i);
 	    }
 	    if (ruby_scope->local_tbl && ruby_scope->local_vars[-1] == 0) {
-		free(ruby_scope->local_tbl);
+		xfree(ruby_scope->local_tbl);
 	    }
 	    ruby_scope->local_vars[-1] = 0;
 	    ruby_scope->local_tbl = local_tbl();
@@ -5726,7 +5746,7 @@
 int
 ruby_parser_stack_on_heap()
 {
-#if defined(YYBISON) && !defined(C_ALLOCA)
+#if defined(YYMALLOC)
     return Qfalse;
 #else
     return Qtrue;
@@ -5736,6 +5756,7 @@
 void
 rb_gc_mark_parser()
 {
+    rb_gc_mark((VALUE)parser_heap);
     if (!ruby_in_compile) return;
 
     rb_gc_mark_maybe((VALUE)yylval.node);
@@ -6085,3 +6106,65 @@
 	special_local_set('_', val);
     }
 }
+
+#ifdef YYMALLOC
+#define HEAPCNT(n, size) ((n) * (size) / sizeof(YYSTYPE))
+#define NEWHEAP(cnt) rb_node_newnode(NODE_ALLOCA, 0, (VALUE)parser_heap, cnt)
+#define ADD2HEAP(n, ptr) ((parser_heap = (n))->u1.node = (ptr))
+
+static void *
+rb_parser_malloc(size)
+    size_t size;
+{
+    NODE *n = NEWHEAP(HEAPCNT(1, size));
+
+    return ADD2HEAP(n, xmalloc(size));
+}
+
+static void *
+rb_parser_calloc(nelem, size)
+    size_t nelem, size;
+{
+    NODE *n = NEWHEAP(HEAPCNT(nelem, size));
+
+    return ADD2HEAP(n, xcalloc(nelem, size));
+}
+
+static void *
+rb_parser_realloc(ptr, size)
+    void *ptr;
+    size_t size;
+{
+    NODE *n;
+    size_t cnt = HEAPCNT(1, size);
+
+    if (ptr && (n = parser_heap) != NULL) {
+	do {
+	    if (n->u1.node == ptr) {
+		n->u1.node = ptr = xrealloc(ptr, size);
+		if (n->u3.cnt) n->u3.cnt = cnt;
+		return ptr;
+	    }
+	} while ((n = n->u2.node) != NULL);
+    }
+    n = NEWHEAP(cnt);
+    return ADD2HEAP(n, xrealloc(ptr, size));
+}
+
+static void
+rb_parser_free(ptr)
+    void *ptr;
+{
+    NODE **prev = &parser_heap, *n;
+
+    while (n = *prev) {
+	if (n->u1.node == ptr) {
+	    *prev = n->u2.node;
+	    rb_gc_force_recycle((VALUE)n);
+	    break;
+	}
+	prev = &n->u2.node;
+    }
+    xfree(ptr);
+}
+#endif
Index: gc.c
===================================================================
RCS file: /var/cvs/src/ruby/gc.c,v
retrieving revision 1.168.2.25
retrieving revision 1.168.2.26
diff -u -r1.168.2.25 -r1.168.2.26
--- ruby-1.8.3/gc.c	27 Jul 2005 14:29:11 -0000	1.168.2.25
+++ ruby-1.8.3/gc.c	8 Oct 2005 09:57:12 -0000	1.168.2.26
@@ -880,13 +880,11 @@
 	  case NODE_BLOCK_ARG:
 	  case NODE_POSTEXE:
 	    break;
-#ifdef C_ALLOCA
 	  case NODE_ALLOCA:
 	    mark_locations_array((VALUE*)obj->as.node.u1.value,
 				 obj->as.node.u3.cnt);
 	    ptr = (VALUE)obj->as.node.u2.node;
 	    goto again;
-#endif
 
 	  default:		/* unlisted NODE */
 	    if (is_pointer_to_heap(obj->as.node.u1.node)) {
@@ -1230,11 +1228,9 @@
 		RUBY_CRITICAL(free(RANY(obj)->as.node.u1.tbl));
 	    }
 	    break;
-#ifdef C_ALLOCA
 	  case NODE_ALLOCA:
 	    RUBY_CRITICAL(free(RANY(obj)->as.node.u1.node));
 	    break;
-#endif
 	}
 	return;			/* no need to free iv_tbl */
 
Index: node.h
===================================================================
RCS file: /var/cvs/src/ruby/node.h,v
retrieving revision 1.50.2.3
retrieving revision 1.50.2.4
diff -u -r1.50.2.3 -r1.50.2.4
--- ruby-1.8.3/node.h	22 Mar 2005 08:35:23 -0000	1.50.2.3
+++ ruby-1.8.3/node.h	8 Oct 2005 09:57:12 -0000	1.50.2.4
@@ -115,9 +115,7 @@
     NODE_DEFINED,
     NODE_NEWLINE,
     NODE_POSTEXE,
-#ifdef C_ALLOCA
     NODE_ALLOCA,
-#endif
     NODE_DMETHOD,
     NODE_BMETHOD,
     NODE_MEMO,


### Patch 123_openssl_add_ssl_algorithms.patch
[] should call OpenSSL_add_ssl_algorithms().

Index: ext/openssl/ossl.c
===================================================================
RCS file: /var/cvs/src/ruby/ext/openssl/ossl.c,v
retrieving revision 1.11.2.4
retrieving revision 1.11.2.5
diff -u -r1.11.2.4 -r1.11.2.5
--- ruby-1.8.3/ext/openssl/ossl.c	10 Sep 2005 00:54:29 -0000	1.11.2.4
+++ ruby-1.8.3/ext/openssl/ossl.c	12 Oct 2005 03:56:04 -0000	1.11.2.5
@@ -382,6 +382,7 @@
      */
     /* CRYPTO_malloc_init(); */
     /* ENGINE_load_builtin_engines(); */
+    OpenSSL_add_ssl_algorithms();
     OpenSSL_add_all_algorithms();
     ERR_load_crypto_strings();
     SSL_load_error_strings();


### Patch 124_rinda_check_remote_hash.patch
[] check remote hash tuple

Index: lib/rinda/rinda.rb
===================================================================
RCS file: /var/cvs/src/ruby/lib/rinda/rinda.rb,v
retrieving revision 1.2.2.6
retrieving revision 1.2.2.7
diff -u -r1.2.2.6 -r1.2.2.7
--- ruby-1.8.3/lib/rinda/rinda.rb	21 Apr 2004 14:00:34 -0000	1.2.2.6
+++ ruby-1.8.3/lib/rinda/rinda.rb	16 Oct 2005 05:36:26 -0000	1.2.2.7
@@ -31,7 +31,7 @@
   class Tuple
     # Initialize a tuple with an Array or a Hash.
     def initialize(ary_or_hash)
-      if Hash === ary_or_hash
+      if hash?(ary_or_hash)
 	init_with_hash(ary_or_hash)
       else
 	init_with_ary(ary_or_hash)
@@ -68,6 +68,10 @@
     end
 
     private
+    def hash?(ary_or_hash)
+      ary_or_hash.respond_to?(:keys)
+    end
+    
     def init_with_ary(ary)
       @tuple = Array.new(ary.size)
       @tuple.size.times do |i|


### Patch 125_file.c_fix_type.patch
[] fixed type of 2nd argument of chmod_internal and lchmod_internal

Index: file.c
===================================================================
RCS file: /var/cvs/src/ruby/file.c,v
retrieving revision 1.169.2.25
retrieving revision 1.169.2.26
diff -u -r1.169.2.25 -r1.169.2.26
--- ruby-1.8.3/file.c	12 Oct 2005 02:26:52 -0000	1.169.2.25
+++ ruby-1.8.3/file.c	17 Oct 2005 07:06:53 -0000	1.169.2.26
@@ -1689,11 +1689,13 @@
     int owner, group;
 };
 
+static void chown_internal _((const char *, void *));
 static void
-chown_internal(path, args)
+chown_internal(path, argp)
     const char *path;
-    struct chown_args *args;
+    void *argp;
 {
+    struct chown_args *args = (struct chown_args *)argp;
     if (chown(path, args->owner, args->group) < 0)
 	rb_sys_fail(path);
 }
@@ -1780,11 +1782,13 @@
 }
 
 #if defined(HAVE_LCHOWN) && !defined(__CHECKER__)
+static void lchown_internal _((const char *, void *));
 static void
-lchown_internal(path, args)
+lchown_internal(path, argp)
     const char *path;
-    struct chown_args *args;
+    void *argp;
 {
+    struct chown_args *args = (struct chown_args *)argp;
     if (lchown(path, args->owner, args->group) < 0)
 	rb_sys_fail(path);
 }


### Patch 126_file_join.patch
[ruby-core:06326] segv during File.join.

Index: file.c
===================================================================
RCS file: /var/cvs/src/ruby/file.c,v
retrieving revision 1.169.2.26
retrieving revision 1.169.2.27
diff -u -r1.169.2.26 -r1.169.2.27
--- ruby-1.8.3/file.c	17 Oct 2005 07:06:53 -0000	1.169.2.26
+++ ruby-1.8.3/file.c	18 Oct 2005 23:29:24 -0000	1.169.2.27
@@ -2789,7 +2789,7 @@
 	name = StringValueCStr(result);
 	if (i > 0 && !NIL_P(sep)) {
 	    tail = chompdirsep(name);
-	    if (isdirsep(RSTRING(tmp)->ptr[0])) {
+	    if (RSTRING(tmp)->ptr && isdirsep(RSTRING(tmp)->ptr[0])) {
 		RSTRING(result)->len = tail - name;
 	    }
 	    else if (!*tail) {


### Patch 127_regex_numeric_literal.patch
[ruby-list:41328] numeric literal inside character class disabled succeeding backtrack.
     
Index: regex.c
===================================================================
RCS file: /var/cvs/src/ruby/Attic/regex.c,v
retrieving revision 1.96.2.6
retrieving revision 1.96.2.7
diff -u -r1.96.2.6 -r1.96.2.7
--- ruby-1.8.3/regex.c	10 Jul 2005 23:32:50 -0000	1.96.2.6
+++ ruby-1.8.3/regex.c	18 Oct 2005 16:27:53 -0000	1.96.2.7
@@ -1725,6 +1725,7 @@
 	memmove(&b[(unsigned char)b[-1]], &b[(1 << BYTEWIDTH) / BYTEWIDTH],
 		2 + EXTRACT_UNSIGNED(&b[(1 << BYTEWIDTH) / BYTEWIDTH])*8);
       b += b[-1] + 2 + EXTRACT_UNSIGNED(&b[(unsigned char)b[-1]])*8;
+      had_num_literal = 0;
       break;
 
     case '(':


### Patch 800_delegate.rb.patch
Index: ./lib/delegate.rb
===================================================================
RCS file: /src/ruby/lib/delegate.rb,v
retrieving revision 1.14.2.3
diff -u -r1.14.2.3 delegate.rb
--- ruby-1.8.3/lib/delegate.rb	17 May 2004 07:18:29 -0000	1.14.2.3
+++ ruby-1.8.3/lib/delegate.rb	25 May 2004 03:33:51 -0000
@@ -38,7 +38,7 @@
 	    rescue Exception
 	      $@.delete_if{|s| /:in `__getobj__'$/ =~ s} #`
 	      $@.delete_if{|s| /^\\(eval\\):/ =~ s}
-	      Kernel::raise
+	      ::Kernel::raise
 	    end
 	  end
 	EOS
@@ -107,7 +107,7 @@
   }
   for method in methods
     begin
-      klass.module_eval <<-EOS
+      klass.module_eval <<-EOS, __FILE__, __LINE__+1
         def #{method}(*args, &block)
 	  begin
 	    @_dc_obj.__send__(:#{method}, *args, &block)


### Patch 801_yaml_bignum.patch
[ruby-core:6159] YAML.dump/load cannot handle Bignum
[DebianBug#331050] patch: YAML.dump(1234567890) #=> "1234567890\n"

Index: ext/syck/rubyext.c
===================================================================
RCS file: /var/cvs/src/ruby/ext/syck/rubyext.c,v
retrieving revision 1.30.2.16
diff -u -r1.30.2.16 rubyext.c
--- ruby-1.8.3/ext/syck/rubyext.c	13 Oct 2005 14:30:53 -0000	1.30.2.16
+++ ruby-1.8.3/ext/syck/rubyext.c	19 Oct 2005 08:03:32 -0000
@@ -1142,6 +1142,9 @@
             }
             else if ( !NIL_P( target_class ) )
             {
+                if (subclass == rb_cBignum)
+                    obj = rb_str2inum(val, 10);
+                else
                 obj = rb_obj_alloc( subclass );
                 if ( rb_respond_to( obj, s_yaml_initialize ) )
                 {
Index: lib/yaml/rubytypes.rb
===================================================================
RCS file: /var/cvs/src/ruby/lib/yaml/rubytypes.rb,v
retrieving revision 1.16.2.9
diff -u -r1.16.2.9 rubytypes.rb
--- ruby-1.8.3/lib/yaml/rubytypes.rb	20 Sep 2005 06:46:45 -0000	1.16.2.9
+++ ruby-1.8.3/lib/yaml/rubytypes.rb	19 Oct 2005 08:38:55 -0000
@@ -372,6 +372,10 @@
     yaml_as "tag:yaml.org,2002:int"
 end
 
+class Bignum
+    yaml_as "tag:yaml.org,2002:int#yes"
+end
+
 class Float
     yaml_as "tag:yaml.org,2002:float"
 end


### Patch 900_ri_pager.patch
--- ruby-1.8.3/lib/rdoc/ri/ri_display.rb	12 Jan 2004 03:11:25 -0000	1.2
+++ ruby-1.8.3/lib/rdoc/ri/ri_display.rb	17 Jan 2004 17:52:40 -0000
@@ -211,7 +211,7 @@
       STDOUT.reopen(@save_stdout)
       @save_stdout = nil
       paged = false
-      for pager in [ ENV['PAGER'], "less", "more <", 'pager' ].compact.uniq
+      for pager in [ ENV['PAGER'], 'pager', "less", "more <" ].compact.uniq
         if system("#{pager} #{path}")
           paged = true
           break


### Patch 901_.document.patch
Index: lib/.document
===================================================================
RCS file: /var/cvs/src/ruby/lib/.document,v
retrieving revision 1.1.2.4
diff -u -r1.1.2.4 .document
--- ruby-1.8.3/lib/.document	16 Sep 2005 03:04:58 -0000	1.1.2.4
+++ ruby-1.8.3/lib/.document	21 Sep 2005 03:37:47 -0000
@@ -13,22 +13,40 @@
 cgi
 complex.rb
 date.rb
+date
+delegate.rb
+drb.rb
+drb
 English.rb
+erb.rb
 fileutils.rb
 find.rb
 generator.rb
+gserver.rb
+ipaddr.rb
 logger.rb
 matrix.rb
+monitor.rb
+net
 observer.rb
-optionparser.rb
+open-uri.rb
+optparse
+optparse.rb
+ostruct.rb
 pathname.rb
+pp.rb
+rexml
+rinda
 set.rb
 shellwords.rb
 singleton.rb
 tempfile.rb
-test/unit.rb
+test
 thread.rb
 thwait.rb
 time.rb
+timeout.rb
+tsort.rb
+uri.rb
 uri
 yaml.rb
Index: lib/cgi/.document
===================================================================
RCS file: /var/cvs/src/ruby/lib/cgi/.document,v
retrieving revision 1.1.2.1
diff -u -r1.1.2.1 .document
--- ruby-1.8.3/lib/cgi/.document	26 Feb 2004 13:50:40 -0000	1.1.2.1
+++ ruby-1.8.3/lib/cgi/.document	21 Sep 2005 03:37:47 -0000
@@ -1,2 +1,2 @@
+session
 session.rb
-


