diff -Nurd -x'*~' icewm-1.2.26.orig/Makefile.in icewm-1.2.26/Makefile.in
--- icewm-1.2.26.orig/Makefile.in	2006-04-16 12:12:31.000000000 -0400
+++ icewm-1.2.26/Makefile.in	2006-04-28 01:34:00.000000000 -0400
@@ -82,19 +82,19 @@
 install-base: base
 	@echo ------------------------------------------
 	@echo "Installing binaries in $(DESTDIR)$(BINDIR)"
-	@$(INSTALLDIR) "$(DESTDIR)$(BINDIR)"
-	@for bin in $(BINFILES); do \
+	$(INSTALLDIR) "$(DESTDIR)$(BINDIR)"
+	for bin in $(BINFILES); do \
              $(INSTALLBIN) "$${bin}" "$(DESTDIR)$(BINDIR)"; \
          done
 	
 	@echo "Installing presets and icons in $(DESTDIR)$(LIBDIR)"
-	@$(INSTALLDIR) "$(DESTDIR)$(LIBDIR)"
+	$(INSTALLDIR) "$(DESTDIR)$(LIBDIR)"
 	#-@$(INSTALLDIR) "$(DESTDIR)$(CFGDIR)"
-	@for lib in $(LIBFILES); do \
+	for lib in $(LIBFILES); do \
              $(INSTALLLIB) "$${lib}" "$(DESTDIR)$(LIBDIR)"; \
          done
 
-	@for xpmdir in $(XPMDIRS); do \
+	for xpmdir in $(XPMDIRS); do \
 	     if test -d "lib/$${xpmdir}"; then \
 		$(INSTALLDIR) "$(DESTDIR)$(LIBDIR)/$${xpmdir}"; \
 		for pixmap in "lib/$${xpmdir}/"*.xpm; do \
@@ -104,7 +104,7 @@
         done
 	
 	@echo ------------------------------------------
-	@for theme in $(THEMES); do \
+	for theme in $(THEMES); do \
 	     SRCDIR="$(top_srcdir)" \
 	     DESTDIR="$(DESTDIR)" \
 	     LIBDIR="$(LIBDIR)" \
@@ -119,27 +119,28 @@
 
 install-docs: docs
 	@echo ------------------------------------------
-	@rm -fr "$(DESTDIR)$(DOCDIR)/icewm-$(VERSION)"
-	@$(INSTALLDIR) "$(DESTDIR)$(DOCDIR)/icewm-$(VERSION)"
+	rm -fr "$(DESTDIR)$(DOCDIR)/icewm-$(VERSION)"
+	$(INSTALLDIR) "$(DESTDIR)$(DOCDIR)/icewm-$(VERSION)"
 	@echo "Installing documentation in $(DESTDIR)$(DOCDIR)"
-	@$(INSTALLLIB) $(DOCFILES) "$(DESTDIR)$(DOCDIR)/icewm-$(VERSION)"
-	@$(INSTALLLIB) "$(top_srcdir)/doc/"*.sgml "$(DESTDIR)$(DOCDIR)/icewm-$(VERSION)"
-	@$(INSTALLLIB) "$(top_srcdir)/doc/"*.html "$(DESTDIR)$(DOCDIR)/icewm-$(VERSION)"
+	$(INSTALLLIB) $(DOCFILES) "$(DESTDIR)$(DOCDIR)/icewm-$(VERSION)"
+	$(INSTALLLIB) "$(top_srcdir)/doc/"*.sgml "$(DESTDIR)$(DOCDIR)/icewm-$(VERSION)"
+	$(INSTALLLIB) "$(top_srcdir)/doc/"*.html "$(DESTDIR)$(DOCDIR)/icewm-$(VERSION)"
 	@echo ------------------------------------------
 
 install-nls: nls
 	@echo ------------------------------------------
-	@cd po; $(MAKE) install
+	cd po; $(MAKE) install
 	@echo ------------------------------------------
 
 install-man:
-	@for man in $(MANFILES); do \
+	$(INSTALLDIR) "$(DESTDIR)$(MANDIR)/man1"
+	for man in $(MANFILES); do \
 		$(INSTALLMAN) doc/$$man.man $(DESTDIR)$(MANDIR)/man1/$$man; \
 	done
 
 install-gnome: gnome
 	@echo ------------------------------------------
 	@echo Installing icewm-gnome in $(DESTDIR)$(GWMDIR)
-	@$(INSTALLDIR) "$(DESTDIR)$(GWMDIR)"
-	@$(INSTALLLIB) "$(top_srcdir)/lib/IceWM.desktop" "$(DESTDIR)$(GWMDIR)"
+	$(INSTALLDIR) "$(DESTDIR)$(GWMDIR)"
+	$(INSTALLLIB) "$(top_srcdir)/lib/IceWM.desktop" "$(DESTDIR)$(GWMDIR)"
 	@echo ------------------------------------------
diff -Nurd -x'*~' icewm-1.2.26.orig/configure icewm-1.2.26/configure
--- icewm-1.2.26.orig/configure	2006-04-16 12:12:38.000000000 -0400
+++ icewm-1.2.26/configure	2006-04-28 01:38:39.000000000 -0400
@@ -11934,7 +11934,7 @@
 
 VERSION=`sed -ne 's/VERSION=//p' VERSION`
 
-HOSTOS=`uname -sr || echo unknown`
+HOSTOS=`uname -s || echo unknown`
 
 HOSTCPU=`uname -m || echo unknown`
 
diff -Nurd -x'*~' icewm-1.2.26.orig/po/Makefile.in icewm-1.2.26/po/Makefile.in
--- icewm-1.2.26.orig/po/Makefile.in	2006-04-16 12:12:31.000000000 -0400
+++ icewm-1.2.26/po/Makefile.in	2006-04-28 01:31:55.000000000 -0400
@@ -28,7 +28,7 @@
 
 install: all
 	@echo "Installing message catalogues in $(DESTDIR)$(LOCDIR)"
-	@for catalog in $(MOFILES); do \
+	for catalog in $(MOFILES); do \
 	     lang=`echo $${catalog} | sed -e 's/\.mo//'` ; \
 	     msgdir="$(DESTDIR)$(LOCDIR)/$${lang}/LC_MESSAGES"; \
 	     echo "Installing language: $${lang}" ; \
diff -Nurd -x'*~' icewm-1.2.26.orig/src/Makefile.in icewm-1.2.26/src/Makefile.in
--- icewm-1.2.26.orig/src/Makefile.in	2006-04-16 12:12:31.000000000 -0400
+++ icewm-1.2.26/src/Makefile.in	2006-04-28 01:10:24.000000000 -0400
@@ -204,22 +204,18 @@
 ################################################################################
 
 %.o: %.cc
-	@echo "  CXX   " $@
-	@$(CXX) $(CXXFLAGS) $(GCCDEP) -c $<
+	$(CXX) $(CXXFLAGS) $(GCCDEP) -c $<
 	
 genpref $(BINARIES):
-	@echo "  LD    " $@
-	@$(LD) -o $@$(EXEEXT) $($@_OBJS) $(LFLAGS) $($@_LFLAGS) $(LIBS) $($@_LIBS)
+	$(LD) -o $@$(EXEEXT) $($@_OBJS) $(LFLAGS) $($@_LFLAGS) $(LIBS) $($@_LIBS)
 
 ################################################################################
 
 gnome.o: gnome.cc
-	@echo "  CXX   " $@
-	@$(CXX) $(CXXFLAGS)  @GNOME1_CFLAGS@ $(GCCDEP) -c $<
+	$(CXX) $(CXXFLAGS)  @GNOME1_CFLAGS@ $(GCCDEP) -c $<
 
 gnome2.o: gnome2.cc
-	@echo "  CXX   " $@
-	@$(CXX) $(CXXFLAGS)  @GNOME2_CFLAGS@ $(GCCDEP) -c $<
+	$(CXX) $(CXXFLAGS)  @GNOME2_CFLAGS@ $(GCCDEP) -c $<
 
 ################################################################################
 
