--- pwm-20010309/system.mk	Thu Dec 28 15:08:05 2000
+++ pwm-20010309/system.mk	Sun Nov  4 18:14:35 2001
@@ -12,8 +12,8 @@
 # No need to modify these usually
 BINDIR=$(PREFIX)/bin
 ETCDIR=$(PREFIX)/etc
-MANDIR=$(PREFIX)/man
-DOCDIR=$(PREFIX)/doc
+MANDIR=$(PREFIX)/share/man
+DOCDIR=$(PREFIX)/share/doc
 # Not used
 INCDIR=$(PREFIX)/include
 LIBDIR=$(PREFIX)/lib
@@ -40,7 +40,7 @@
 ## C compiler
 ##
 
-CC=gcc
+CC=cc
 
 # The POSIX_SOURCE, XOPEN_SOURCE and WARN options should not be necessary,
 # they're mainly for development use. So, if they cause trouble (not
@@ -62,12 +62,12 @@
 
 # Same as '-Wall -pedantic-errors' without '-Wunused' as callbacks often
 # have unused variables.
-WARN=	-W -Wimplicit -Wreturn-type -Wswitch -Wcomment \
+#WARN=	-W -Wimplicit -Wreturn-type -Wswitch -Wcomment \
 	-Wtrigraphs -Wformat -Wchar-subscripts \
 	-Wparentheses -pedantic-errors -Wuninitialized
 
 
-CFLAGS=-g -O2 $(WARN) $(DEFINES) $(INCLUDES) $(EXTRA_INCLUDES)
+CFLAGS=-g -O2 -no-cpp-precomp $(WARN) $(DEFINES) $(INCLUDES) $(EXTRA_INCLUDES)
 LDFLAGS=-g $(LIBS) $(EXTRA_LIBS)
 
 
@@ -93,9 +93,9 @@
 ##
 
 # Should work almost everywhere
-INSTALL=install
+#INSTALL=install
 # On a system with pure BSD install, -c might be preferred
-#INSTALL=install -c
+INSTALL=install -c
 
 INSTALLDIR=mkdir -p
 
