diff -uNr xfce-3.8.18/xfce/taskbar.c xfce-3.8.18-new/xfce/taskbar.c
--- xfce-3.8.18/xfce/taskbar.c	2002-09-26 05:22:22.000000000 -0400
+++ xfce-3.8.18-new/xfce/taskbar.c	2007-07-19 16:02:54.000000000 -0400
@@ -1696,7 +1696,6 @@
   case TASKBAR_SORT_UNSORTED:
     fprintf(f,"\tSortByUnsorted\n");
     break;
-  default:
   } 
 }
 
diff -uNr xfce-3.8.18/xfgnome/xfgnome.c xfce-3.8.18-new/xfgnome/xfgnome.c
--- xfce-3.8.18/xfgnome/xfgnome.c	2001-08-10 09:11:08.000000000 -0400
+++ xfce-3.8.18-new/xfgnome/xfgnome.c	2007-07-19 16:07:36.000000000 -0400
@@ -20,6 +20,8 @@
 #  include <config.h>
 #endif
 
+#define XFGNOME
+
 #include <X11/Intrinsic.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
diff -uNr xfce-3.8.18/xfsamba/xfsamba.c xfce-3.8.18-new/xfsamba/xfsamba.c
--- xfce-3.8.18/xfsamba/xfsamba.c	2002-05-10 20:42:33.000000000 -0400
+++ xfce-3.8.18-new/xfsamba/xfsamba.c	2007-07-19 16:13:09.000000000 -0400
@@ -26,6 +26,8 @@
 #  include <config.h>
 #endif
 
+#define _XFSAMBA_NOHEADH_
+
 #include <unistd.h>
 #include <stdarg.h>
 #include <signal.h>
diff -uNr xfce-3.8.18/xfsamba/xfsamba.h xfce-3.8.18-new/xfsamba/xfsamba.h
--- xfce-3.8.18/xfsamba/xfsamba.h	2002-11-10 08:27:33.000000000 -0500
+++ xfce-3.8.18-new/xfsamba/xfsamba.h	2007-07-19 16:13:09.000000000 -0400
@@ -240,7 +240,10 @@
 
 EXTERN selected_struc selected;
 EXTERN nmb_list *thisN, *headN;
-EXTERN nmb_history *thisH, *headH;
+EXTERN nmb_history *thisH;
+#ifndef _XFSAMBA_NOHEADH_
+EXTERN nmb_history *headH;
+#endif
 EXTERN void *fork_obj;
 EXTERN int SMBResult;
 EXTERN int SMBabortdrop;
diff -uNr xfce-3.8.18/xfsamba/xfsamba_mem.c xfce-3.8.18-new/xfsamba/xfsamba_mem.c
--- xfce-3.8.18/xfsamba/xfsamba_mem.c	2002-11-10 08:27:33.000000000 -0500
+++ xfce-3.8.18-new/xfsamba/xfsamba_mem.c	2007-07-19 16:13:09.000000000 -0400
@@ -26,6 +26,8 @@
 #  include <config.h>
 #endif
 
+#define _XFSAMBA_NOHEADH_
+
 #ifndef HAVE_SNPRINTF
 #include "snprintf.h"
 #endif
diff -uNr xfce-3.8.18/xfwm/xfwm.h xfce-3.8.18-new/xfwm/xfwm.h
--- xfce-3.8.18/xfwm/xfwm.h	2002-05-22 17:24:36.000000000 -0400
+++ xfce-3.8.18-new/xfwm/xfwm.h	2007-07-19 16:07:44.000000000 -0400
@@ -352,10 +352,14 @@
 extern Atom _XA_OL_DECOR_RESIZE;
 extern Atom _XA_OL_DECOR_HEADER;
 extern Atom _XA_OL_DECOR_ICON_NAME;
+#ifndef XFGNOME
 extern Atom _XA_WIN_WORKSPACE_COUNT;
 extern Atom _XA_WIN_WORKSPACE;
+#endif
 extern Atom _XA_WIN_LAYER;
+#ifndef XFGNOME
 extern Atom _XA_WIN_STATE;
+#endif
 
 extern Atom _XA_WM_WINDOW_ROLE;
 extern Atom _XA_WM_CLIENT_LEADER;
