diff -ruN xfce4-session-4.2.3/configure xfce4-session-4.2.3-mod/configure
--- xfce4-session-4.2.3/configure	2005-11-05 12:03:54.000000000 -0600
+++ xfce4-session-4.2.3-mod/configure	2005-11-08 22:40:35.000000000 -0600
@@ -3930,11 +3930,11 @@
 
 
 if test x"`uname | grep \"CYGWIN\"`" != x""; then
-  HAVE_OS_CYGWIN_TRUE=
-  HAVE_OS_CYGWIN_FALSE='#'
-else
   HAVE_OS_CYGWIN_TRUE='#'
   HAVE_OS_CYGWIN_FALSE=
+else
+  HAVE_OS_CYGWIN_TRUE=
+  HAVE_OS_CYGWIN_FALSE='#'
 fi
 
 if test x"`uname | grep \"CYGWIN\"`" != x""; then
diff -ruN xfce4-session-4.2.3/engines/balou/Makefile.in xfce4-session-4.2.3-mod/engines/balou/Makefile.in
--- xfce4-session-4.2.3/engines/balou/Makefile.in	2005-11-06 04:35:51.000000000 -0600
+++ xfce4-session-4.2.3-mod/engines/balou/Makefile.in	2005-11-08 22:40:35.000000000 -0600
@@ -264,7 +264,7 @@
 
 libbalou_la_LDFLAGS = \
 	-avoid-version							\
-	-no-undefined							\
+	-undefined dynamic_lookup					\
 	-module
 
 libbalou_la_LIBADD = $(top_builddir)/libxfsm/libxfsm-4.2.la \
diff -ruN xfce4-session-4.2.3/libxfsm/Makefile.in xfce4-session-4.2.3-mod/libxfsm/Makefile.in
--- xfce4-session-4.2.3/libxfsm/Makefile.in	2005-11-06 04:35:54.000000000 -0600
+++ xfce4-session-4.2.3-mod/libxfsm/Makefile.in	2005-11-08 22:40:35.000000000 -0600
@@ -39,7 +39,7 @@
 build_triplet = @build@
 host_triplet = @host@
 @HAVE_OS_CYGWIN_TRUE@am__append_1 = \
-@HAVE_OS_CYGWIN_TRUE@	-no-undefined							\
+@HAVE_OS_CYGWIN_FALSE@	-no-undefined							\
 @HAVE_OS_CYGWIN_TRUE@	@LIBX11_LDFLAGS@
 
 subdir = libxfsm
diff -ruN xfce4-session-4.2.3/settings/session/Makefile.in xfce4-session-4.2.3-mod/settings/session/Makefile.in
--- xfce4-session-4.2.3/settings/session/Makefile.in	2005-11-06 04:35:55.000000000 -0600
+++ xfce4-session-4.2.3-mod/settings/session/Makefile.in	2005-11-08 22:40:35.000000000 -0600
@@ -37,8 +37,8 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@HAVE_OS_CYGWIN_TRUE@am__append_1 = \
-@HAVE_OS_CYGWIN_TRUE@	-export-symbols $(datadir)/xfce4/devel/mcs-manager.def
+@HAVE_OS_CYGWIN_FALSE@am__append_1 = \
+@HAVE_OS_CYGWIN_FALSE@	-export-symbols $(datadir)/xfce4/devel/mcs-manager.def
 
 subdir = settings/session
 DIST_COMMON = $(srcdir)/Makefile.am $(srcdir)/Makefile.in
@@ -238,7 +238,7 @@
 session_LTLIBRARIES = \
 	session_settings.la
 
-session_settings_la_LDFLAGS = -avoid-version -no-undefined -module \
+session_settings_la_LDFLAGS = -avoid-version -undefined dynamic_lookup -module \
 	$(am__append_1)
 session_settings_la_SOURCES = \
 	session.c
diff -ruN xfce4-session-4.2.3/settings/splash/Makefile.in xfce4-session-4.2.3-mod/settings/splash/Makefile.in
--- xfce4-session-4.2.3/settings/splash/Makefile.in	2005-11-06 04:35:55.000000000 -0600
+++ xfce4-session-4.2.3-mod/settings/splash/Makefile.in	2005-11-08 22:40:35.000000000 -0600
@@ -37,8 +37,8 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@HAVE_OS_CYGWIN_TRUE@am__append_1 = \
-@HAVE_OS_CYGWIN_TRUE@	-export-symbols $(datadir)/xfce4/devel/mcs-manager.def
+@HAVE_OS_CYGWIN_FALSE@am__append_1 = \
+@HAVE_OS_CYGWIN_FALSE@	-export-symbols $(datadir)/xfce4/devel/mcs-manager.def
 
 @HAVE_OS_CYGWIN_TRUE@am__append_2 = \
 @HAVE_OS_CYGWIN_TRUE@	@XFCE_MCS_MANAGER_LIBS@						\
@@ -243,7 +243,7 @@
 splash_LTLIBRARIES = \
 	splash_settings.la
 
-splash_settings_la_LDFLAGS = -avoid-version -module -no-undefined \
+splash_settings_la_LDFLAGS = -avoid-version -module -undefined dynamic_lookup \
 	$(am__append_1)
 splash_settings_la_SOURCES = \
 	module.c							\
diff -ruN xfce4-session-4.2.3/xfce4-session/xfsm-global.h xfce4-session-4.2.3-mod/xfce4-session/xfsm-global.h
--- xfce4-session-4.2.3/xfce4-session/xfsm-global.h	2005-11-05 11:08:48.000000000 -0600
+++ xfce4-session-4.2.3-mod/xfce4-session/xfsm-global.h	2005-11-08 22:50:18.000000000 -0600
@@ -64,7 +64,7 @@
 #define xfsm_verbose(format, ...)     \
 G_STMT_START{                         \
   if (G_UNLIKELY (verbose))           \
-    xfsm_verbose_real ( ## format);   \
+    xfsm_verbose_real ( format);   \
 }G_STMT_END
   
 #endif
