diff -ruN xfce4-toys-4.2.0/configure xfce4-toys-4.2.0-mod/configure
--- xfce4-toys-4.2.0/configure	Wed Jan  5 16:50:30 2005
+++ xfce4-toys-4.2.0-mod/configure	Thu Jan 20 13:06:50 2005
@@ -3890,11 +3890,11 @@
 
 
 if test "`uname | grep \"CYGWIN\"`" != ""; then
-  HAVE_CYGWIN_TRUE=
-  HAVE_CYGWIN_FALSE='#'
-else
   HAVE_CYGWIN_TRUE='#'
   HAVE_CYGWIN_FALSE=
+else
+  HAVE_CYGWIN_TRUE=
+  HAVE_CYGWIN_FALSE='#'
 fi
 
 
diff -ruN xfce4-toys-4.2.0/xfce4-eyes/panel-plugin/Makefile.in xfce4-toys-4.2.0-mod/xfce4-eyes/panel-plugin/Makefile.in
--- xfce4-toys-4.2.0/xfce4-eyes/panel-plugin/Makefile.in	Wed Dec 22 15:06:29 2004
+++ xfce4-toys-4.2.0-mod/xfce4-eyes/panel-plugin/Makefile.in	Thu Jan 20 13:07:23 2005
@@ -38,9 +38,9 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@HAVE_CYGWIN_TRUE@am__append_1 = \
-@HAVE_CYGWIN_TRUE@	-no-undefined							\
-@HAVE_CYGWIN_TRUE@	-export-symbols $(datadir)/xfce4/devel/panel.def
+@HAVE_CYGWIN_FALSE@am__append_1 = \
+@HAVE_CYGWIN_FALSE@	-no-undefined							\
+@HAVE_CYGWIN_FALSE@	-export-symbols $(datadir)/xfce4/devel/panel.def
 
 subdir = xfce4-eyes/panel-plugin
 DIST_COMMON = $(srcdir)/Makefile.am $(srcdir)/Makefile.in
