diff -ruN fltk-1.1.2.orig/FL/Makefile.in fltk-1.1.2/FL/Makefile.in
--- fltk-1.1.2.orig/FL/Makefile.in	Sun Jul 14 11:03:30 2002
+++ fltk-1.1.2/FL/Makefile.in	Sun Apr 13 19:34:07 2003
@@ -35,17 +35,17 @@
 	$(CHMOD) 755 $(includedir)/FL
 	$(CP) ../FL/*.[hHr] $(includedir)/FL
 	$(CHMOD) 644 $(includedir)/FL/*.[hHr]
-@HLINKS@	cd $(includedir)/FL;\
-@HLINKS@	for file in *.H; do\
-@HLINKS@		$(RM) "`basename $$file H`h";\
-@HLINKS@		$(LN) $$file "`basename $$file H`h";\
-@HLINKS@	done
-@HLINKS@	$(RM) $(includedir)/FL/fl_file_chooser.H
-@HLINKS@	$(LN) Fl_File_Chooser.H $(includedir)/FL/fl_file_chooser.H
-@HLINKS@	$(RM) $(includedir)/FL/fl_file_chooser.h
-@HLINKS@	$(LN) Fl_File_Chooser.H $(includedir)/FL/fl_file_chooser.h
-@HLINKS@	$(RM) $(includedir)/Fl
-@HLINKS@	$(LN) FL $(includedir)/Fl
+#@HLINKS@	cd $(includedir)/FL;\
+#@HLINKS@	for file in *.H; do\
+#@HLINKS@		$(RM) "`basename $$file H`h";\
+#@HLINKS@		$(LN) $$file "`basename $$file H`h";\
+#@HLINKS@	done
+#@HLINKS@	$(RM) $(includedir)/FL/fl_file_chooser.H
+#@HLINKS@	$(LN) Fl_File_Chooser.H $(includedir)/FL/fl_file_chooser.H
+#@HLINKS@	$(RM) $(includedir)/FL/fl_file_chooser.h
+#@HLINKS@	$(LN) Fl_File_Chooser.H $(includedir)/FL/fl_file_chooser.h
+#@HLINKS@	$(RMDIR) $(includedir)/Fl
+#@HLINKS@	$(LN) FL $(includedir)/Fl
 
 uninstall:
 	echo "Uninstalling include files..."
diff -ruN fltk-1.1.2.orig/Makefile fltk-1.1.2/Makefile
--- fltk-1.1.2.orig/Makefile	Wed Sep 25 19:15:33 2002
+++ fltk-1.1.2/Makefile	Sun Apr 13 13:06:03 2003
@@ -25,7 +25,7 @@
 
 include makeinclude
 
-DIRS	=	src fluid test documentation
+DIRS	=	src fluid documentation
 
 all: makeinclude
 	for dir in $(DIRS); do\
diff -ruN fltk-1.1.2.orig/configure fltk-1.1.2/configure
--- fltk-1.1.2.orig/configure	Mon Nov 25 10:38:38 2002
+++ fltk-1.1.2/configure	Sun Apr 13 13:05:39 2003
@@ -6826,7 +6826,7 @@
 	HLINKS="#"
 	;;
 
-    Darwin*)
+    MacOSX*)
         # MacOS X uses Carbon for graphics...
         LIBS="$LIBS -framework Carbon -framework ApplicationServices"
 
diff -ruN fltk-1.1.2.orig/fltk-config.in fltk-1.1.2/fltk-config.in
--- fltk-1.1.2.orig/fltk-config.in	Fri Oct 18 11:18:33 2002
+++ fltk-1.1.2/fltk-config.in	Sun Apr 13 13:06:22 2003
@@ -116,7 +116,7 @@
 
     case $1 in
 	--prefix=*)
-	    prefix=$optarg
+	    prefix=$optarg ; includedir=${prefix}/include
 	    if test $exec_prefix_set = no ; then
               exec_prefix=$optarg
 	    fi
diff -ruN fltk-1.1.2.orig/fluid/Makefile fltk-1.1.2/fluid/Makefile
--- fltk-1.1.2.orig/fluid/Makefile	Tue Jul 30 08:11:21 2002
+++ fltk-1.1.2/fluid/Makefile	Sun Apr 13 13:06:42 2003
@@ -77,7 +77,7 @@
 	-$(MKDIR) $(bindir)
 	$(CP) $(FLUID) $(bindir)/fluid$(EXEEXT)
 	$(STRIP) $(bindir)/fluid$(EXEEXT)
-	$(bindir)/fltk-config --post $(bindir)/fluid$(EXEEXT)
+	$(bindir)/fltk-config --prefix=$(prefix) --post $(bindir)/fluid$(EXEEXT)
 	$(CHMOD) 755 $(bindir)/fluid$(EXEEXT)
 
 uninstall:
