diff -ruN SDL-1.2.5.orig/src/video/SDL_stretch.c SDL-1.2.5/src/video/SDL_stretch.c
--- SDL-1.2.5.orig/src/video/SDL_stretch.c	Wed Mar  6 04:23:03 2002
+++ SDL-1.2.5/src/video/SDL_stretch.c	Fri Aug 15 11:07:13 2003
@@ -261,12 +261,7 @@
 			break;
 		    default:
 #ifdef __GNUC__
-			__asm__ __volatile__ ("
-				call _copy_row
-			"
-			: "=&D" (u1), "=&S" (u2)
-			: "0" (dstp), "1" (srcp)
-			: "memory" );
+			__asm__ __volatile__ ("call _copy_row" : "=&D" (u1), "=&S" (u2) : "0" (dstp), "1" (srcp) : "memory" );
 #else
 #ifdef WIN32
 		{ void *code = &copy_row;
