diff -Nurd -x'*~' gconf-editor-2.28.0.orig/src/Makefile.in gconf-editor-2.28.0/src/Makefile.in
--- gconf-editor-2.28.0.orig/src/Makefile.in	2009-09-21 08:05:18.000000000 -0400
+++ gconf-editor-2.28.0/src/Makefile.in	2010-02-17 01:35:45.000000000 -0500
@@ -277,7 +277,7 @@
 
 gconf_editor_LDADD = \
 	$(GCONF_EDITOR_LIBS) \
-	$(GCONF_EDITOR_POLKIT_LIBS)
+	$(GCONF_EDITOR_POLKIT_LIBS) $(INTLLIBS)
 
 EXTRA_DIST = \
 	gconf-marshal.list
