diff -ruNp ghemical-3.0.0-orig/src/project.h ghemical-3.0.0/src/project.h
--- ghemical-3.0.0-orig/src/project.h	2008-11-26 05:09:46.000000000 -0600
+++ ghemical-3.0.0/src/project.h	2014-09-03 07:10:51.000000000 -0500
@@ -63,6 +63,8 @@ using namespace std;
 
 /*################################################################################################*/
 
+bool ReadGPR(project &, istream &, bool, bool = false);
+
 class jobinfo_GeomOpt
 {
 	public:
@@ -207,11 +209,11 @@ class project :
 
 // methods for file I/O : ReadGPR and WriteGPR are friend functions so that it would be easier to "borrow" them elsewhere...
 // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-	friend bool ReadGPR_OLD(project &, istream &, bool, bool = false);		///< this is for the very old version.
-	friend bool ReadGPR_v100(project &, istream &, bool, bool = false);		///< this is for the version 1.00.
-	friend bool ReadGPR_v110(project &, istream &, bool, bool = false);		///< this is for the version 1.10.
+	friend bool ReadGPR_OLD(project &, istream &, bool, bool);		///< this is for the very old version.
+	friend bool ReadGPR_v100(project &, istream &, bool, bool);		///< this is for the version 1.00.
+	friend bool ReadGPR_v110(project &, istream &, bool, bool);		///< this is for the version 1.10.
 /// This is an input function for the v1.11 ghemical file format.
-	friend bool ReadGPR(project &, istream &, bool, bool = false);
+	friend bool ReadGPR(project &, istream &, bool, bool);
 	
 	friend void WriteGPR_v100(project &, ostream &);	///< this is for the version 1.00.
 /// This is an output function for the v1.11 ghemical file format.
