diff -Nurd -x'*~' gnome-menus-2.28.0.1.orig/Makefile.in gnome-menus-2.28.0.1/Makefile.in
--- gnome-menus-2.28.0.1.orig/Makefile.in	2009-10-01 07:24:51.000000000 -0400
+++ gnome-menus-2.28.0.1/Makefile.in	2010-04-13 13:25:49.000000000 -0400
@@ -294,11 +294,6 @@
 NULL = 
 @HAVE_PYTHON_TRUE@PYTHON_SUBDIRS = python simple-editor
 ALL_SUBDIRS = \
-	po 			\
-	libmenu			\
-	layout			\
-	desktop-directories	\
-	util			\
 	$(NULL)
 
 SUBDIRS = $(ALL_SUBDIRS) $(PYTHON_SUBDIRS)
diff -Nurd -x'*~' gnome-menus-2.28.0.1.orig/python/Makefile.in gnome-menus-2.28.0.1/python/Makefile.in
--- gnome-menus-2.28.0.1.orig/python/Makefile.in	2009-10-01 07:24:51.000000000 -0400
+++ gnome-menus-2.28.0.1/python/Makefile.in	2010-04-13 13:28:09.000000000 -0400
@@ -72,8 +72,7 @@
 am__installdirs = "$(DESTDIR)$(gmenudir)"
 LTLIBRARIES = $(gmenu_LTLIBRARIES)
 am__DEPENDENCIES_1 =
-gmenu_la_DEPENDENCIES = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
-	$(top_builddir)/libmenu/libgnome-menu.la
+gmenu_la_DEPENDENCIES = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 am_gmenu_la_OBJECTS = gmenu.lo
 gmenu_la_OBJECTS = $(am_gmenu_la_OBJECTS)
 AM_V_lt = $(am__v_lt_$(V))
@@ -296,7 +295,6 @@
 NULL = 
 INCLUDES = \
 	-DGMENU_I_KNOW_THIS_IS_UNSTABLE	\
-	-I$(srcdir)/../libmenu		\
 	$(WARN_CFLAGS)			\
 	$(NO_STRICT_ALIASING_CFLAGS)	\
 	$(GLIB_CFLAGS)			\
@@ -308,7 +306,7 @@
 gmenudir = $(pyexecdir)
 gmenu_LTLIBRARIES = gmenu.la
 gmenu_la_LDFLAGS = -module -avoid-version -fPIC -export-symbols-regex initgmenu
-gmenu_la_LIBADD = $(GLIB_LIBS) $(PYTHON_LIBS) $(top_builddir)/libmenu/libgnome-menu.la
+gmenu_la_LIBADD = $(GLIB_LIBS) $(PYTHON_LIBS)
 gmenu_la_SOURCES = gmenu.c
 all: all-am
 
