diff -Nurd -x'*~' gnome-system-monitor-2.20.2.orig/src/Makefile.in gnome-system-monitor-2.20.2/src/Makefile.in
--- gnome-system-monitor-2.20.2.orig/src/Makefile.in	2008-01-07 14:34:37.000000000 -0500
+++ gnome-system-monitor-2.20.2/src/Makefile.in	2010-04-28 01:58:24.000000000 -0400
@@ -293,7 +293,7 @@
 	iconthemewrapper.cpp iconthemewrapper.h \
 	e_date.c e_date.h
 
-gnome_system_monitor_LDADD = @PROCMAN_LIBS@ @PCRECPP_LIBS@ libbacon.la
+gnome_system_monitor_LDADD = @PROCMAN_LIBS@ @PCRECPP_LIBS@ libbacon.la $(INTLLIBS)
 noinst_LTLIBRARIES = libbacon.la
 libbacon_la_SOURCES = \
 	bacon-message-connection.c \
