diff -Nurd -x'*~' goocanvasmm-0.15.2.orig/goocanvas/goocanvasmm/enums.cc goocanvasmm-0.15.2/goocanvas/goocanvasmm/enums.cc
--- goocanvasmm-0.15.2.orig/goocanvas/goocanvasmm/enums.cc	2010-04-16 08:17:31.000000000 -0400
+++ goocanvasmm-0.15.2/goocanvas/goocanvasmm/enums.cc	2016-03-08 04:04:54.000000000 -0500
@@ -31,7 +31,6 @@
 
 // We can't simply derive from Glib::ValueBoxed<> because these have no
 // gobj() function to return the underlying boxed C object.
-GType Value<Cairo::Pattern>::value_type() { return goo_cairo_pattern_get_type(); }
 GType Value<Cairo::Matrix*>::value_type() { return goo_cairo_matrix_get_type(); }
 
 // Normally, _WRAP_ENUM does generate this code for us, but we can't use
diff -Nurd -x'*~' goocanvasmm-0.15.2.orig/goocanvas/goocanvasmm/enums.h goocanvasmm-0.15.2/goocanvas/goocanvasmm/enums.h
--- goocanvasmm-0.15.2.orig/goocanvas/goocanvasmm/enums.h	2010-04-16 08:17:31.000000000 -0400
+++ goocanvasmm-0.15.2/goocanvas/goocanvasmm/enums.h	2016-03-08 04:04:27.000000000 -0500
@@ -193,16 +193,6 @@
 // GTypes goocanvas defines for them in goocanvasutil.h
 
 template<>
-class Value<Cairo::Pattern> : public Glib::ValueBase_Boxed
-{
-public:
-  static GType value_type() G_GNUC_CONST;
-
-  void set(const Cairo::Pattern& data) { set_boxed(data.cobj()); }
-  Cairo::Pattern get() const { return Cairo::Pattern(static_cast<cairo_pattern_t*>(get_boxed())); }
-};
-
-template<>
 class Value<Cairo::Matrix*> : public Glib::ValueBase_Boxed
 {
 public:
