--- a/tests/check/elements/udpsink.c	2011-06-13 19:24:13.000000000 -0400
+++ a/tests/check/elements/udpsink.c	2013-02-15 09:09:26.000000000 -0500
@@ -22,6 +22,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 
+#if 0
 static GstStaticPadTemplate srctemplate = GST_STATIC_PAD_TEMPLATE ("src",
     GST_PAD_SRC,
     GST_PAD_ALWAYS,
@@ -182,6 +183,7 @@
 }
 
 GST_END_TEST;
+#endif
 
 /*
  * Creates the test suite.
@@ -199,8 +201,10 @@
   tcase_set_timeout (tc_chain, 60);
 
   suite_add_tcase (s, tc_chain);
+#if 0
   tcase_add_test (tc_chain, test_udpsink);
   tcase_add_test (tc_chain, test_udpsink_bufferlist);
+#endif
   return s;
 }
 
diff --git a/gst/audiofx/audiocheblimit.c b/gst/audiofx/audiocheblimit.c
index 549a9d2..7568d6a 100644
--- a/gst/audiofx/audiocheblimit.c
+++ b/gst/audiofx/audiocheblimit.c
@@ -46,12 +46,12 @@
  * be at most this value. A lower ripple value will allow a faster rolloff.
  *
  * As a special case, a Chebyshev type 1 filter with no ripple is a Butterworth filter.
- * </para>
+ * <para>
  * <note><para>
  * Be warned that a too large number of poles can produce noise. The most poles are possible with
  * a cutoff frequency at a quarter of the sampling rate.
  * </para></note>
- * <para>
+ * </para>
  * <refsect2>
  * <title>Example launch line</title>
  * |[
diff --git a/gst/udp/gstudpsrc.c b/gst/udp/gstudpsrc.c
index c2cbd92..7dffef8 100644
--- a/gst/udp/gstudpsrc.c
+++ b/gst/udp/gstudpsrc.c
@@ -78,14 +78,14 @@
  * </itemizedlist>
  * The message is typically used to detect that no UDP arrives in the receiver
  * because it is blocked by a firewall.
- * </para>
+ *
  * <para>
  * A custom file descriptor can be configured with the
  * #GstUDPSrc:sockfd property. The socket will be closed when setting the
  * element to READY by default. This behaviour can be
  * overriden with the #GstUDPSrc:closefd property, in which case the application
  * is responsible for closing the file descriptor.
- *
+ * </para>
  * <refsect2>
  * <title>Examples</title>
  * |[
