diff -Nurd -x'*~' shared-mime-info-1.4.orig/update-mime-database.c shared-mime-info-1.4/update-mime-database.c
--- shared-mime-info-1.4.orig/update-mime-database.c	2015-02-05 09:16:52.000000000 -0500
+++ shared-mime-info-1.4/update-mime-database.c	2015-03-05 16:31:22.000000000 -0500
@@ -11,6 +11,9 @@
 #include <glib.h>
 #include <glib/gprintf.h>
 #include <glib/gstdio.h>
+#if !GLIB_CHECK_VERSION(2,26,0)
+typedef struct stat GStatBuf;
+#endif
 #include <errno.h>
 #include <dirent.h>
 #include <libxml/parser.h>
