diff -ruN ipe-6.0pre10/src/include/ipebase.h ipe-6.0pre10.patched/src/include/ipebase.h
--- ipe-6.0pre10/src/include/ipebase.h	2003-04-14 05:13:20.000000000 -0400
+++ ipe-6.0pre10.patched/src/include/ipebase.h	2009-04-29 14:00:43.000000000 -0400
@@ -32,7 +32,7 @@
 //#pragma warning(disable: 4251)
 //#pragma warning(disable: 4275)
 #endif
-
+#include <sys/types.h>
 #include <cassert>
 #include <cstdio>
 #include <vector>
diff -ruN ipe-6.0pre10/src/ipelets/qvoronoi/qvoronoi.cpp ipe-6.0pre10.patched/src/ipelets/qvoronoi/qvoronoi.cpp
--- ipe-6.0pre10/src/ipelets/qvoronoi/qvoronoi.cpp	2003-04-14 05:13:22.000000000 -0400
+++ ipe-6.0pre10.patched/src/ipelets/qvoronoi/qvoronoi.cpp	2009-04-29 13:56:20.000000000 -0400
@@ -396,10 +396,9 @@
       std::sort(iDelaunay.begin(), iDelaunay.end());
       for (int j = 0; j < iDelaunay.size(); j++) {
 	if (!j || iDelaunay[j] != iDelaunay[j-1]) {
-	  IpeSegment seg(IpeVector(points[3*iDelaunay[j].a],
-				   points[3*iDelaunay[j].a + 1]),
-			 IpeVector(points[3*iDelaunay[j].b],
-				   points[3*iDelaunay[j].b + 1]));
+	  IpeVector a(points[3*iDelaunay[j].a], points[3*iDelaunay[j].a + 1]);
+  	  IpeVector b(points[3*iDelaunay[j].b], points[3*iDelaunay[j].b + 1]);
+  	  IpeSegment seg(a,b);
 	  group->push_back(new IpePath(attr, seg));
 	}
       }
