diff -Nurd -x'*~' Imaging-1.1.6.orig/setup.py Imaging-1.1.6/setup.py
--- Imaging-1.1.6.orig/setup.py	2006-12-03 06:37:29.000000000 -0500
+++ Imaging-1.1.6/setup.py	2008-08-12 15:01:14.000000000 -0400
@@ -136,17 +136,9 @@
             # fink installation directories
-            add_directory(library_dirs, "/sw/lib")
-            add_directory(include_dirs, "/sw/include")
-            # darwin ports installation directories
-            add_directory(library_dirs, "/opt/local/lib")
-            add_directory(include_dirs, "/opt/local/include")
+            add_directory(library_dirs, "@PREFIX@/lib")
+            add_directory(include_dirs, "@PREFIX@/include")
 
-        add_directory(library_dirs, "/usr/local/lib")
         # FIXME: check /opt/stuff directories here?
 
-        prefix = sysconfig.get_config_var("prefix")
-        if prefix:
-            add_directory(library_dirs, os.path.join(prefix, "lib"))
-            add_directory(include_dirs, os.path.join(prefix, "include"))
 
         #
         # locate tkinter libraries
@@ -193,8 +185,10 @@
         #
         # add standard directories
 
-        add_directory(library_dirs, "/usr/local/lib")
-        add_directory(include_dirs, "/usr/local/include")
+        prefix = sysconfig.get_config_var("prefix")
+        if prefix:
+            add_directory(library_dirs, os.path.join(prefix, "lib"))
+            add_directory(include_dirs, os.path.join(prefix, "include"))
 
         add_directory(library_dirs, "/usr/lib")
         add_directory(include_dirs, "/usr/include")
--- Imaging-1.1.6/Scripts/pildriver.py~	2006-12-03 12:37:18.000000000 +0100
+++ Imaging-1.1.6/Scripts/pildriver.py	2008-11-08 21:41:12.000000000 +0100
@@ -48,7 +48,8 @@
 # 3. Add support for composing and decomposing multiple-image files.
 #
 
-import Image, string
+from PIL import Image
+import string
 
 class PILDriver:
 
@@ -204,7 +205,7 @@
 
         Process the top image with the given filter.
         """
-        import ImageFilter
+        from PIL import ImageFilter
         filter = eval("ImageFilter." + string.upper(self.do_pop()))
         image = self.do_pop()
         self.push(image.filter(filter))
@@ -349,7 +350,7 @@
 
         Invert the top image.
         """
-        import ImageChops
+        from PIL import ImageChops
         self.push(ImageChops.invert(self.do_pop()))
 
     def do_lighter(self):
@@ -357,7 +358,7 @@
 
         Pop the two top images, push an image of the lighter pixels of both.
         """
-        import ImageChops
+        from PIL import ImageChops
         image1 = self.do_pop()
         image2 = self.do_pop()
         self.push(ImageChops.lighter(image1, image2))
@@ -367,7 +368,7 @@
 
         Pop the two top images, push an image of the darker pixels of both.
         """
-        import ImageChops
+        from PIL import ImageChops
         image1 = self.do_pop()
         image2 = self.do_pop()
         self.push(ImageChops.darker(image1, image2))
@@ -377,7 +378,7 @@
 
         Pop the two top images, push the difference image
         """
-        import ImageChops
+        from PIL import ImageChops
         image1 = self.do_pop()
         image2 = self.do_pop()
         self.push(ImageChops.difference(image1, image2))
@@ -387,7 +388,7 @@
 
         Pop the two top images, push the multiplication image.
         """
-        import ImageChops
+        from PIL import ImageChops
         image1 = self.do_pop()
         image2 = self.do_pop()
         self.push(ImageChops.multiply(image1, image2))
@@ -397,7 +398,7 @@
 
         Pop the two top images, superimpose their inverted versions.
         """
-        import ImageChops
+        from PIL import ImageChops
         image2 = self.do_pop()
         image1 = self.do_pop()
         self.push(ImageChops.screen(image1, image2))
@@ -407,7 +408,7 @@
 
         Pop the two top images, produce the scaled sum with offset.
         """
-        import ImageChops
+        from PIL import ImageChops
         image1 = self.do_pop()
         image2 = self.do_pop()
         scale = float(self.do_pop())
@@ -419,8 +420,8 @@
 
         Pop the two top images, produce the scaled difference with offset.
         """
-        import ImageChops
-        image1 = self.do_pop()
+        from PIL import ImageChops
+	image1 = self.do_pop()
         image2 = self.do_pop()
         scale = float(self.do_pop())
         offset = int(self.do_pop())
@@ -433,7 +434,7 @@
 
         Enhance color in the top image.
         """
-        import ImageEnhance
+        from PIL import ImageEnhance
         factor = float(self.do_pop())
         image = self.do_pop()
         enhancer = ImageEnhance.Color(image)
@@ -444,7 +445,7 @@
 
         Enhance contrast in the top image.
         """
-        import ImageEnhance
+        from PIL import ImageEnhance
         factor = float(self.do_pop())
         image = self.do_pop()
         enhancer = ImageEnhance.Color(image)
@@ -455,7 +456,7 @@
 
         Enhance brightness in the top image.
         """
-        import ImageEnhance
+        from PIL import ImageEnhance
         factor = float(self.do_pop())
         image = self.do_pop()
         enhancer = ImageEnhance.Color(image)
@@ -466,7 +467,7 @@
 
         Enhance sharpness in the top image.
         """
-        import ImageEnhance
+        from PIL import ImageEnhance
         factor = float(self.do_pop())
         image = self.do_pop()
         enhancer = ImageEnhance.Color(image)
--- Imaging-1.1.6/Scripts/pilprint.py~	2006-12-03 12:37:18.000000000 +0100
+++ Imaging-1.1.6/Scripts/pilprint.py	2008-11-08 21:43:00.000000000 +0100
@@ -13,8 +13,8 @@
 
 VERSION = "pilprint 0.3/2003-05-05"
 
-import Image
-import PSDraw
+from PIL import Image
+from PIL import PSDraw
 
 letter = ( 1.0*72, 1.0*72, 7.5*72, 10.0*72 )
 
