diff -Nurd -x'*~' libcaca-0.99.beta18.orig/configure libcaca-0.99.beta18/configure
--- libcaca-0.99.beta18.orig/configure	2012-04-06 16:48:32.000000000 -0400
+++ libcaca-0.99.beta18/configure	2013-03-21 22:04:13.000000000 -0400
@@ -17393,7 +17393,7 @@
 
 $as_echo "#define USE_GL 1" >>confdefs.h
 
-    GL_LIBS="${GL_LIBS} -lGL -lGLU -lglut"
+    GL_LIBS="${GL_LIBS} -lGL -lGLU -lglut -L/usr/X11R6/lib"
     CACA_DRIVERS="${CACA_DRIVERS} gl"
   elif test "${enable_gl}" = "yes"; then
     as_fn_error $? "cannot find OpenGL+FreeGLUT development files" "$LINENO" 5
