
--- dbus-0.60/configure	2005-11-30 16:02:26.000000000 -0500
+++ dbus-0.60-new/configure	2006-03-20 14:25:16.000000000 -0500
@@ -28789,6 +28789,7 @@
 if test -n "$QTDIR" -a -f "$QTDIR/include/qglobal.h"; then
     have_qt3=yes
     DBUS_QT3_CXXFLAGS="-I$QTDIR/include"
+    DBUS_QT3_LIBS="-L$QTDIR/lib -lqt-mt"
 else
     for dir in "${prefix}/include/qt" "/usr/include/qt-3.1" "/usr/include/qt3" "/usr/include/qt" "/usr/lib/qt/include" "/usr/lib/qt-3.1/include"; do
         if test -f "$dir/qglobal.h"; then
--- dbus-0.60/qt3/Makefile.in	2005-11-30 16:02:22.000000000 -0500
+++ dbus-0.60-new/qt3/Makefile.in	2006-03-20 14:24:42.000000000 -0500
@@ -60,7 +60,7 @@
 LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
 @HAVE_QT3_TRUE@libdbus_qt_1_la_DEPENDENCIES = $(am__DEPENDENCIES_1) \
-@HAVE_QT3_TRUE@	$(top_builddir)/dbus/libdbus-1.la
+@HAVE_QT3_TRUE@	
 am__libdbus_qt_1_la_SOURCES_DIST = $(top_srcdir)/qt3/dbus-qthread.cpp \
 	$(top_srcdir)/qt3/message.cpp $(top_srcdir)/qt3/connection.cpp \
 	$(top_srcdir)/qt3/integrator.cpp $(top_srcdir)/qt3/server.cpp \
@@ -368,7 +368,7 @@
 @HAVE_QT3_TRUE@	$(top_srcdir)/qt3/server.h
 
 @HAVE_QT3_TRUE@CLEANFILES = connection.moc integrator.moc server.moc
-@HAVE_QT3_TRUE@libdbus_qt_1_la_LIBADD = $(DBUS_QT3_LIBS) $(top_builddir)/dbus/libdbus-1.la
+@HAVE_QT3_TRUE@libdbus_qt_1_la_LIBADD = $(DBUS_QT3_LIBS) -ldbus-1
 @HAVE_QT3_TRUE@libdbus_qt_1_la_LDFLAGS = -version-info 1:0 -no-undefined
 all: all-am
 
