diff -Naur dclib-0.3.19.orig/dclib/core/cdir.cpp dclib-0.3.19/dclib/core/cdir.cpp
--- dclib-0.3.19.orig/dclib/core/cdir.cpp	2008-08-16 13:13:31.000000000 +0100
+++ dclib-0.3.19/dclib/core/cdir.cpp	2008-08-16 13:14:05.000000000 +0100
@@ -20,22 +20,10 @@
 #endif
 
 #ifndef WIN32
-#include <unistd.h>
-#ifdef HAVE_SYS_PARAM_H
-#include <sys/param.h>
-#endif
-#include <dirent.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#ifdef HAVE_SYS_VFS_H
-#include <sys/vfs.h>
-#endif
-#ifdef HAVE_SYS_MOUNT_H
-#include <sys/mount.h>
-#endif
-#ifdef HAVE_SYS_PARAM_H
-#include <sys/param.h>
-#endif
+#include <unistd.h>
+#include <dirent.h>
 #endif
 
 #include <string.h>
