diff -ru libdvdread-0.9.4.orig/dvdread/bswap.h libdvdread-0.9.4/dvdread/bswap.h
--- libdvdread-0.9.4.orig/dvdread/bswap.h	2002-12-14 19:09:12.000000000 -0500
+++ libdvdread-0.9.4/dvdread/bswap.h	2006-03-02 20:27:12.000000000 -0500
@@ -84,6 +84,12 @@
       (((x) & 0x000000000000ff00) << 40) | \
       (((x) & 0x00000000000000ff) << 56))
 
+#elif defined(__APPLE__)
+#include <libkern/OSByteOrder.h>
+#define B2N_16(x) x = OSSwapInt16(x)
+#define B2N_32(x) x = OSSwapInt32(x)
+#define B2N_64(x) x = OSSwapInt64(x)
+
 #else
 
 /* If there isn't a header provided with your system with this functionality
