diff -Nurd libmal-0.44.1/configure libmal-0.44.1-new/configure
--- libmal-0.44.1/configure	2009-02-16 11:30:38.000000000 -0500
+++ libmal-0.44.1-new/configure	2009-03-19 09:30:27.000000000 -0400
@@ -20069,7 +20069,7 @@
 if test $pilotlibs = no ; then
     for pilot_libs in $pilot_prefix/lib /usr/lib /usr/local/lib/ \
         /usr/extra/pilot/lib ; do
-        if test -r "$pilot_libs/libpisock.so" ; then
+        if test -r "$pilot_libs/libpisock.dylib" ; then
             pilotlibs=yes
             PILOT_LIBS="-L$pilot_libs $PILOT_LIBS"
             break
diff -Nurd libmal-0.44.1/malsync/mal/common/AGNet.h libmal-0.44.1-new/malsync/mal/common/AGNet.h
--- libmal-0.44.1/malsync/mal/common/AGNet.h	2009-02-16 11:07:04.000000000 -0500
+++ libmal-0.44.1-new/malsync/mal/common/AGNet.h	2009-03-19 09:31:18.000000000 -0400
@@ -62,7 +62,7 @@
 #               include <sys/filio.h>
 #               include <arpa/inet.h>
 #           else
-#               if (defined(BSD) && BSD >= 199306) ||  defined(__OpenBSD__) || defined(_HPUX_SOURCE)
+#               if (defined(BSD) && BSD >= 199306) ||  defined(__OpenBSD__) || defined(_HPUX_SOURCE) || defined(__APPLE__)
 #                   include <sys/ioctl.h>
 #                   include <arpa/inet.h>
 #               elif defined(__GLIBC__)
