diff -Nurd -x'*~' srtp.orig/Makefile.in srtp/Makefile.in
--- srtp.orig/Makefile.in	2006-03-17 16:00:46.000000000 -0500
+++ srtp/Makefile.in	2009-03-08 04:16:24.000000000 -0400
@@ -36,7 +36,7 @@
 CPPFLAGS= @CPPFLAGS@
 CFLAGS	= @CFLAGS@
 LIBS	= @LIBS@
-LDFLAGS	= @LDFLAGS@ -L.
+LDFLAGS	= -L. @LDFLAGS@
 COMPILE = $(CC) $(DEFS) $(INCDIR) $(CPPFLAGS) $(CFLAGS)
 SRTPLIB	= -lsrtp
 
diff -Nurd -x'*~' srtp.orig/crypto/Makefile.in srtp/crypto/Makefile.in
--- srtp.orig/crypto/Makefile.in	2005-10-03 11:16:37.000000000 -0400
+++ srtp/crypto/Makefile.in	2009-03-08 04:16:11.000000000 -0400
@@ -14,7 +14,7 @@
 CPPFLAGS= @CPPFLAGS@
 CFLAGS	= @CFLAGS@
 LIBS	= @LIBS@
-LDFLAGS	= @LDFLAGS@ -L.
+LDFLAGS	= -L. @LDFLAGS@
 COMPILE = $(CC) $(DEFS) $(INCDIR) $(CPPFLAGS) $(CFLAGS)
 CRYPTOLIB = -lcryptomodule
 
