diff -urN log4c-1.0.10.orig/doc/Makefile.in log4c-1.0.10/doc/Makefile.in
--- log4c-1.0.10.orig/doc/Makefile.in	2002-11-22 15:44:31.000000000 +0100
+++ log4c-1.0.10/doc/Makefile.in	2003-04-07 23:58:49.000000000 +0200
@@ -87,14 +87,13 @@
 install_sh = @install_sh@
 
 man3dir = $(mandir)/man3
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 docfiles = \
 	$(top_srcdir)/README \
 	$(top_srcdir)/AUTHORS \
 	$(top_srcdir)/NEWS \
 	$(top_srcdir)/COPYING \
-	$(top_srcdir)/ChangeLog \
-	$(PACKAGE).pdf
+	$(top_srcdir)/ChangeLog
 
 subdir = doc
 mkinstalldirs = $(SHELL) $(top_srcdir)/config/mkinstalldirs
@@ -231,8 +230,6 @@
 
 $(PACKAGE).pdf: Doxyfile main.doc
 	@DOXYGEN@
-	$(MAKE) -C ./latex pdf
-	cp -f ./latex/refman.pdf $(PACKAGE).pdf
 
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(man3dir)
