diff -ru PDCurses-2.6/pdcurses/Makefile.in PDCurses-2.6_patched/pdcurses/Makefile.in
--- PDCurses-2.6/pdcurses/Makefile.in	Thu Jul 11 12:20:44 2002
+++ PDCurses-2.6_patched/pdcurses/Makefile.in	Mon Apr 21 00:27:14 2003
@@ -33,7 +33,7 @@
 DYN_COMP =         @DYN_COMP@
 SHLPRE =           @SHLPRE@
 SHLPST =           @SHLPST@
-LD_RXLIB1  =       @LD_RXLIB1@
+LD_RXLIB1  =       @LD_RXLIB1@ -flat_namespace -undefined suppress
 LD_RXLIB2  =       # @MH_XLIBS@ @MH_EXTRA_LIBS@ -lc
 O2SAVE  =          @O2SAVE@
 O2SHO  =           @O2SHO@
 diff -ru PDCurses-2.6/pdcurses/initscr.c PDCurses-2.6_patched/pdcurses/initscr.c
--- PDCurses-2.6/pdcurses/initscr.c	Mon Dec 16 07:58:09 2002
+++ PDCurses-2.6_patched/pdcurses/initscr.c	Mon Apr 21 00:27:14 2003
@@ -73,7 +73,7 @@
 WINDOW*  stdscr=NULL;         /* the default screen window  */
 int   LINES=0;       /* current terminal height */
 int   COLS=0;        /* current terminal width  */
-MOUSE_STATUS Mouse_status;
+//MOUSE_STATUS Mouse_status;
 int    use_emalloc = FALSE;
 
 #if defined DOS
diff -ru PDCurses-2.6/x11/pdcx11.c PDCurses-2.6_patched/x11/pdcx11.c
--- PDCurses-2.6/x11/pdcx11.c	Tue Sep  3 10:03:50 2002
+++ PDCurses-2.6_patched/x11/pdcx11.c	Mon Apr 21 00:27:14 2003
@@ -23,7 +23,7 @@
 
 #if defined(XCURSES)
 #include "pdcx11.h"
-
+AppData app_data;
 #if NeedWidePrototypes
 # define PDC_SCROLLBAR_TYPE double
 #else
diff -ru PDCurses-2.6/x11/pdcx11.h PDCurses-2.6_patched/x11/pdcx11.h
--- PDCurses-2.6/x11/pdcx11.h	Tue Sep  3 10:05:24 2002
+++ PDCurses-2.6_patched/x11/pdcx11.h	Mon Apr 21 00:27:14 2003
@@ -232,7 +232,7 @@
  XtWidgetGeometry geometry;
 #endif
 } AppData;
-AppData app_data;
+//AppData app_data;
 
 #define XCURSESNORMALFONTINFO   app_data.normalfont
 #define XCURSESITALICFONTINFO   app_data.italicfont
diff -ru PDCurses-2.6/x11/process/x11.c PDCurses-2.6_patched/x11/process/x11.c
--- PDCurses-2.6/x11/process/x11.c	Wed Nov 27 12:21:27 2002
+++ PDCurses-2.6_patched/x11/process/x11.c	Mon Apr 21 00:27:14 2003
@@ -22,7 +22,7 @@
 */
 
 #include "pdcx11.h"
-
+extern AppData app_data;
 int visible_cursor=0;
 int windowEntered = 1;
 static char *XCursesProgramName;
diff -ru PDCurses-2.6/x11/process/x11curses.c PDCurses-2.6_patched/x11/process/x11curses.c
--- PDCurses-2.6/x11/process/x11curses.c	Wed Nov 27 12:21:27 2002
+++ PDCurses-2.6_patched/x11/process/x11curses.c	Mon Apr 21 00:27:14 2003
@@ -27,7 +27,7 @@
  */
 
 #include "pdcx11.h"
-
+extern AppData app_data;
 #ifdef PROTO
 static void XCursesExitCursesProcess(int, char *);
 #else
--- PDCurses-2.6/configure.orig	Sun Oct 26 13:44:28 2003
+++ PDCurses-2.6/configure	Sun Oct 26 13:45:13 2003
@@ -2730,7 +2730,7 @@
 		;;
 	*darwin*)
 		DYN_COMP="-fno-common"
-		LD_RXLIB1="${CC} -dynamiclib -install_name=\$(@)"
+		LD_RXLIB1="${CC} -dynamiclib -install_name $libdir/\$(@)"
 		LD_RXTRANSLIB1="$LD_RXLIB1"
 		;;
 	*)
diff -Naur PDCurses-2.6.orig/configure PDCurses-2.6/configure
--- PDCurses-2.6.orig/configure	2008-02-10 20:08:27.000000000 -0500
+++ PDCurses-2.6/configure	2008-02-10 20:14:45.000000000 -0500
@@ -2267,7 +2267,7 @@
 for mh_lib in $mh_libs; do
   mh_lib_found=no
   for ac_dir in $mh_lib_dirs ; do
-    for mh_ext in a so sl; do
+    for mh_ext in a dylib so sl; do
       if test -r $ac_dir/lib$mh_lib.$mh_ext; then
         if test "x$mh_prev_dir" != "x$ac_dir" ; then
           if test "x$mh_prev_dir" = "x" ; then
