--- publib-0.39.orig/debian/changelog
+++ publib-0.39/debian/changelog
@@ -0,0 +1,225 @@
+publib (0.39-3.1) unstable; urgency=low
+
+  * NMU on maintainer's request.
+  * Re-add man pages (closes: #441963).
+
+ -- Robert Luberda <robert@debian.org>  Wed, 19 Mar 2008 19:28:22 +0100
+
+publib (0.39-3) unstable; urgency=low
+
+  * New maintainer 
+    Closes: #390744 (ITA: publib -- library of miscellaneous C functions)
+  * Note: liw released 0.39-2 on 2 Oct 2006, despite the changelog tagline date
+  * debian/control [Standards-Version]: 3.7.2, no changes required.
+
+ -- Antti-Juhani Kaijanaho <ajk@debian.org>  Tue,  3 Oct 2006 23:01:04 +0300
+
+publib (0.39-2) unstable; urgency=low
+
+  * Orphaning.
+
+ -- Lars Wirzenius <liw@iki.fi>  Sun, 12 Mar 2006 12:08:21 +0200
+
+publib (0.39-1) unstable; urgency=low
+
+  * New upstream version.
+      - strsplit is fixed, thanks to suggestion from Ron Fox.
+        Closes: #354750.
+      - cfgtool routines are removed, but they haven't been used by
+        anything for a decade, so no-one should have to worry
+  * debian/control: Bumped Standards-Version to 3.6.2, no other changes
+    necessary.
+
+ -- Lars Wirzenius <liw@iki.fi>  Sun, 12 Mar 2006 12:08:21 +0200
+
+publib (0.38-2) unstable; urgency=low
+
+  * Updated the package description. Closes: #268540, #277236.
+  * Added patch for cross building, by NIIBE Yutaka. Closes: #285428.
+  * Removed build depdency on autoconf.
+
+ -- Lars Wirzenius <liw@iki.fi>  Wed, 19 Jan 2005 18:36:00 +0300
+
+publib (0.38-1) unstable; urgency=low
+
+  * New upstream version: Changed a Bash shell function to a Posix shell 
+    function ("function foo {" to "foo() {" (closes: #264048).
+
+ -- Lars Wirzenius <liw@iki.fi>  Sat,  7 Aug 2004 01:20:00 +0300
+
+publib (0.37-1) unstable; urgency=low
+
+  * New upstream release to remove modules that won't compile 
+    (closes: #258992).
+
+ -- Lars Wirzenius <liw@iki.fi>  Thu,  5 Aug 2004 11:20:00 +0300
+
+publib (0.36-1) unstable; urgency=low
+
+  * New upstream release to fix struncstr's handling of three
+    digit octal escapes (closes: #246816).
+
+ -- Lars Wirzenius <liw@iki.fi>  Sun, 11 Jul 2004 13:23:00 +0300
+
+publib (0.35-3) unstable; urgency=low
+
+  * debian/control: Set section to be libdevel, instead of devel.
+    Forgot to do this in -2. I'm a klutz and a nincompoop.
+
+ -- Lars Wirzenius <liw@iki.fi>  Sun, 16 Nov 2003 22:32:04 +0200
+
+publib (0.35-2) unstable; urgency=low
+
+  * debian/control: Build dependency on autoconf was missing, added.
+    (closes: #220967)
+  * liw/cfg/configure, liw/lsm/configure: Generated. They're missing
+    in the upstream sources as well, but I don't want to make an
+    upstream release just for this, so I fix it in the Debian version.
+
+ -- Lars Wirzenius <liw@iki.fi>  Sun, 16 Nov 2003 22:23:02 +0200
+
+publib (0.35-1) unstable; urgency=low
+
+  * New upstream release. Fixes:
+    - base64 broken on little-endian machines (closes: #197039)
+
+ -- Lars Wirzenius <liw@iki.fi>  Sat, 15 Nov 2003 20:28:19 +0200
+
+publib (0.34-2) unstable; urgency=low
+
+  * Taking over package maintainership again, with permission from
+    Antti-Juhani.
+  * debian/control: Standards version updated to 3.5.8.0. No other changes
+    were necessary, as far as I could see.
+
+ -- Lars Wirzenius <liw@iki.fi>  Sun, 26 Jan 2003 20:58:45 +0200
+
+publib (0.34-1) unstable; urgency=low
+
+  * New upstream version.  Excerpts from upstream changelog:
+    + liw/sbuf/sbuf_srch.c: Added missing #include for <sys/types.h>
+      (<regex.h> needs it). Thanks to Dmitry Astapov <adept@umc.com.ua>.
+      Closes: #144483.
+    + framework/Generic.mk, framework/link-alternative-names,
+      liw/publibdocs/Makefile: Manual pages are installed compressed,
+      and links to alternative names are created
+      Closes: #99605.
+  * debian/control [Standards-Version]: Upgraded to 3.5.5.
+
+ -- Antti-Juhani Kaijanaho <ajk@debian.org>  Mon, 22 Jul 2002 21:50:30 +0300
+
+publib (0.32-1) unstable; urgency=low
+
+  * debian/control: New maintainer.
+  * Converted from native to non-native packaging, as upstream is no
+    longer the Debian maintainer.
+  * debian/control [Standards-Version]: Upgraded to 3.5.4; due to
+    bug#99605 we are not yet compliant to 3.5.5 or later.
+  * debian/control [Build-Depends]: Added.
+
+ -- Antti-Juhani Kaijanaho <ajk@debian.org>  Sun, 23 Sep 2001 14:07:47 +0300
+
+publib (0.32) unstable; urgency=low
+
+  * Moved files from /usr/doc to /usr/share/doc, as per current policy.
+    (Closes: Bug#91623)
+  * Moved files from /usr/man to /usr/share/man.
+  * Converted to use debhelper.
+
+ -- Lars Wirzenius <liw@iki.fi>  Sat, 31 Mar 2001 12:56:42 +0300
+
+publib (0.31) unstable; urgency=low
+
+  * liw/errormsg/errormsg.c: Treat "" same as NULL for the program name.
+    Suggestion sent by Gary Kline <kline@tao.thought.org>.
+  * This recompile should also fix dwww related problems (closes:
+    Bug #45092).
+
+ -- Lars Wirzenius <liw@iki.fi>  Sun, 19 Sep 1999 19:14:04 +0300
+
+publib (0.30) unstable; urgency=low
+
+  * framework/Makefile.in: Add C++ wrappers in the generated header;
+    if the module headers are to be included directly, they need to add
+    the wrappers themselves.
+
+ -- Lars Wirzenius <liw@iki.fi>  Mon,  8 Feb 1999 17:55:41 +0200
+
+publib (0.29) unstable; urgency=low
+
+  * Added tbuf module. Not working yet, however.
+  * Changed name of changelog in /usr/log.
+  * liw/alloc/alloc.h: Removed prototype for strdup. It isn't included 
+    in publib, anyway. Fixes bug #31977, although differently than the
+    suggested patch in the bug report. Also #29887 and #22561.
+  * liw/sbuf: Added code to log the size and position of changes, triggered
+    by macro SBUF_LOG. This is needed for optimizing, and is not turned on
+    in the release versions.
+
+ -- Lars Wirzenius <liw@iki.fi>  Mon, 25 Jan 1999 15:35:56 +0200
+
+publib (0.28) unstable; urgency=low
+
+  * No longer compresses /usr/doc/publib-dev/copyright. (Fixes bug #14504.)
+  * Moved upstream changelog to debian/changelog (this file).
+  * Compresses manual pages.
+  * Changes section manual pages on the .TH line to 3pub.  This fixes
+    bug #8415.
+  * Updated to conform to Debian Policy Manual 2.4.0.0.
+
+ -- Lars Wirzenius <liw@iki.fi>  Fri, 13 Mar 1998 15:55:54 +0200
+
+publib (0.27) unstable; urgency=low
+
+  * Re-compiled for libc6.
+  * Changes to liw/sbuf, liw/log, and liw/cfg in the upstream version.
+  * Removed Debian version from version number (but didn't combine upstream
+    changelog with Debian changelog).
+
+ -- Lars Wirzenius <liw@iki.fi>  Tue, 11 Nov 1997 04:37:02 +0200
+
+publib (0.26-1) unstable; urgency=low
+
+  * New upstream release that only fixes bugs.
+  * Fixes bug #8228: adds one missing #include, and removes an extra one.
+
+ -- Lars Wirzenius <liw@iki.fi>  Sun, 27 Apr 1997 03:02:43 +0300
+
+publib (0.25-1) unstable; urgency=low
+
+  * New upstream release; mostly just bug fixes.
+
+ -- Lars Wirzenius <liw@iki.fi>  Sat, 28 Dec 1996 00:50:15 +0200
+
+publib (0.24-1) unstable; urgency=low
+
+  * New upstream release with important changes in sbuf and cfgtool.
+
+ -- Lars Wirzenius <liw@iki.fi>  Thu, 5 Dec 1996 11:27:41 +0200
+
+publib (0.23-1) unstable; urgency=low
+
+  * New upstream release (read Changes).
+  * Minor tweaking to debian/rules.
+
+ -- Lars Wirzenius <liw@iki.fi>  Wed, 6 Nov 1996 20:04:02 +0200
+
+publib (0.22-1) unstable; urgency=low
+
+  * New upstream release (read Changes).
+  * Added debian/* to upstream sources.
+
+ -- Lars Wirzenius <liw@iki.fi>  Tue, 5 Nov 1996 23:55:00 +0200
+
+publib (0.21-1) unstable; urgency=low
+
+  * New upstream release (mostly just changes in the sbuf module).
+
+ -- Lars Wirzenius <liw@iki.fi>  Mon, 4 Nov 1996 19:31:04 +0200
+
+publib (0.20-1) unstable; urgency=low
+
+  * First release.
+
+ -- Lars Wirzenius <liw@iki.fi>  Sun, 22 Sep 1996 02:47:04 +0300
+
--- publib-0.39.orig/debian/control
+++ publib-0.39/debian/control
@@ -0,0 +1,14 @@
+Source: publib
+Maintainer: Antti-Juhani Kaijanaho <ajk@debian.org>
+Section: libdevel
+Priority: optional
+Standards-Version: 3.7.2
+Build-Depends: debhelper (>= 5.0.0)
+
+Package: publib-dev
+Architecture: any
+Description: library of miscellaneous C functions
+ This library contains functions for memory allocation, bit arrays,
+ comparing standard C types for qsort and bsearch, error messages,
+ filenames, hash tables, integer sets, log files, NNTP, priority queues,
+ normal queues, editor buffers, stacks, and strings.
--- publib-0.39.orig/debian/rules
+++ publib-0.39/debian/rules
@@ -0,0 +1,61 @@
+#!/usr/bin/make -f
+
+DEB_BUILD_GNU_TYPE = $(shell dpkg-architecture -qDEB_BUILD_GNU_TYPE)
+DEB_HOST_GNU_TYPE  = $(shell dpkg-architecture -qDEB_HOST_GNU_TYPE)
+ifneq ($(DEB_BUILD_GNU_TYPE),$(DEB_HOST_GNU_TYPE))
+CONFARGS= --host=$(DEB_HOST_GNU_TYPE)
+endif
+
+DESTDIR=$(CURDIR)/debian/$(shell dh_listpackages)
+
+build: debian/objs debian/objs/Makefile
+	cd debian/objs && $(MAKE)
+
+binary: binary-arch binary-indep
+
+binary-arch: build
+	dh_testdir
+	dh_testroot
+	dh_clean -k
+	dh_installdirs
+
+	install -d $(DESTDIR)/usr/share/doc/publib-dev
+	install -d $(DESTDIR)/usr/include
+	install -d $(DESTDIR)/usr/lib
+	install -d $(DESTDIR)/usr/share/man 
+	install -d $(DESTDIR)/usr/share/man/man3
+	cd debian/objs && \
+	    $(MAKE) --no-print-dir \
+	    	mandir=$(DESTDIR)/usr/share/man man3ext=3pub install
+	find $(DESTDIR)/usr/share/man -type f -name '*.gz' | xargs gunzip
+	set -e; find $(DESTDIR)/usr/share/man -type f | \
+		while read file; do \
+		   sed 's/^\.TH [_A-Z0-9]* [0-9]/&pub/' $$file > $$file.new; \
+		    mv $$file.new $$file; done
+	find $(DESTDIR)/usr/share/man -type f | xargs gzip --best
+
+	dh_installdocs
+	dh_installchangelogs
+	dh_link
+	dh_compress
+	dh_fixperms
+	dh_installdeb
+	dh_strip
+	dh_shlibdeps
+	dh_gencontrol -- -isp
+	dh_md5sums
+	dh_builddeb
+
+binary-indep:
+
+clean:
+	dh_clean
+	rm -rf debian/objs
+
+debian/objs:
+	mkdir debian/objs
+
+debian/objs/Makefile:
+	cd debian/objs && \
+		cp ../Default.config . && \
+		../../framework/configure $(CONFARGS)
--- publib-0.39.orig/debian/compat
+++ publib-0.39/debian/compat
@@ -0,0 +1 @@
+5
--- publib-0.39.orig/debian/docs
+++ publib-0.39/debian/docs
@@ -0,0 +1,2 @@
+README
+framework/Blurb
--- publib-0.39.orig/debian/Default.config
+++ publib-0.39/debian/Default.config
@@ -0,0 +1,7 @@
+library=publib
+libshort=pub
+modules=../../liw
+CFLAGS='-Wall -O'
+verbmake=no
+prefix=../publib-dev/usr
+mandir=../publib-dev/usr/share/man
--- publib-0.39.orig/debian/copyright
+++ publib-0.39/debian/copyright
@@ -0,0 +1,34 @@
+This is a Debian pre-packaged version of Publib. Both the original
+library and the Debian packaging by Lars Wirzenius (liw@iki.fi).  The
+Debian package was by Antti-Juhani Kaijanaho <ajk@debian.org> but is
+now again maintained by Lars Wirzenius.
+
+Publib consists of the library functions and the build framework
+that configures, compiles, and installs the library. The framework
+is in the public domain, the library functions have the following
+copyright:
+
+Copyright (c) 1996 Lars Wirzenius.  All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+   notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+   notice, this list of conditions and the following disclaimer in the
+   documentation and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
