diff --exclude='*~' -ruN emacs-23.1/configure emacs-23.1.patched/configure
--- emacs-23.1/configure	2009-07-29 17:12:00.000000000 +0200
+++ emacs-23.1.patched/configure	2009-08-26 16:55:27.000000000 +0200
@@ -1854,7 +1854,7 @@
 etcdir='${datadir}/emacs/${version}/etc'
 archlibdir='${libexecdir}/emacs/${version}/${configuration}'
 docdir='${datadir}/emacs/${version}/etc'
-gamedir='${localstatedir}/games/emacs'
+gamedir='${localstatedir}/games/emacs23'
 
 gameuser=games
 
@@ -14610,9 +14610,9 @@
 fi
 { echo "$as_me:$LINENO: result: $ac_cv_lib_ungif_EGifPutExtensionLast" >&5
 echo "${ECHO_T}$ac_cv_lib_ungif_EGifPutExtensionLast" >&6; }
-if test $ac_cv_lib_ungif_EGifPutExtensionLast = yes; then
+#if test $ac_cv_lib_ungif_EGifPutExtensionLast = yes; then
   HAVE_GIF=yes
-fi
+#fi
 
 
     if test "$HAVE_GIF" = yes; then
diff --exclude='*~' -ruN emacs-23.1/fix-info emacs-23.1.patched/fix-info
--- emacs-23.1/fix-info	1970-01-01 01:00:00.000000000 +0100
+++ emacs-23.1.patched/fix-info	2009-08-26 17:44:01.000000000 +0200
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+THE_INFO_PREFIX=$1
+
+for file in ada-mode auth autotype calc ccmode cl dbus dired-x \
+           ebrowse ediff efaq eintr elisp emacs emacs-mime epa erc \
+           eshell eudc flymake forms gnus idlwave info mairix-el \
+           message mh-e newsticker nxml-mode org pcl-cvs pgg rcirc \
+           reftex remember sasl sc ses sieve smtpmail speedbar tramp \
+           url vip viper widget woman
+do
+    ./mangle-info ${THE_INFO_PREFIX}/${file}
+done
\ No newline at end of file
diff --exclude='*~' -ruN emacs-23.1/lisp/startup.el emacs-23.1.patched/lisp/startup.el
--- emacs-23.1/lisp/startup.el	2009-06-21 06:37:46.000000000 +0200
+++ emacs-23.1.patched/lisp/startup.el	2009-08-26 16:57:34.000000000 +0200
@@ -375,6 +375,10 @@
 
 (defvar normal-top-level-add-subdirs-inode-list nil)
 
+(defconst fink-emacs-flavor 'emacs23
+  "A symbol representing the particular fink flavor of emacs running.
+Something like 'emacs20, 'xemacs20, etc.")
+
 (defvar no-blinking-cursor nil)
 
 (defvar default-frame-background-mode)
@@ -977,9 +981,19 @@
     ;; be loaded from site-run-file and wants to test if -q was given
     ;; should check init-file-user instead, since that is already set.
     ;; See cus-edit.el for an example.
-    (if site-run-file
-	(load site-run-file t t))
-
+    ;; (if site-run-file                                                       
+    ;;     (load site-run-file t t))                                           
+                                                                               
+    ;; Fink version of site-start.                                           
+    (if site-run-file                                                          
+ 	(progn                                                                 
+ 	  ;; Load all the fink package snippets.                             
+ 	  ;; It's in here because we want -q to kill it too.                   
+ 	  (if (load "fink-startup" t t nil)                                  
+ 	      (fink-startup fink-emacs-flavor))                            
+ 	  ;; Now the normal site file...                                       
+ 	  (load site-run-file t t nil)))
+    
     ;; Sites should not disable this.  Only individuals should disable
     ;; the startup screen.
     (setq inhibit-startup-screen nil)
diff --exclude='*~' -ruN emacs-23.1/mangle-info emacs-23.1.patched/mangle-info
--- emacs-23.1/mangle-info	1970-01-01 01:00:00.000000000 +0100
+++ emacs-23.1.patched/mangle-info	2009-08-26 16:56:06.000000000 +0200
@@ -0,0 +1,18 @@
+#!/usr/bin/perl -w -i
+
+use English;
+use strict;
+
+$RS = undef;
+
+my $prefix = "emacs-23";
+
+my $x = <>;
+if(!($x =~ m/^(\s*START-INFO-DIR-ENTRY\s+\*\s*[^:]+:\s*)\(([^\)]+)\)/mo)) {
+  die "Couldn't find START-INFO-DIR-ENTRY.";
+}
+   
+print ${PREMATCH};
+print "$1($prefix/$2)";
+print ${POSTMATCH};
+
