diff -Nurd -x'*~' evince-2.26.2.orig/cut-n-paste/totem-screensaver/Makefile.in evince-2.26.2/cut-n-paste/totem-screensaver/Makefile.in
--- evince-2.26.2.orig/cut-n-paste/totem-screensaver/Makefile.in	2009-05-18 16:53:49.000000000 -0400
+++ evince-2.26.2/cut-n-paste/totem-screensaver/Makefile.in	2011-01-06 02:05:39.000000000 -0500
@@ -42,7 +42,7 @@
 CONFIG_HEADER = $(top_builddir)/config.h
 CONFIG_CLEAN_FILES =
 LTLIBRARIES = $(noinst_LTLIBRARIES)
-libtotemscrsaver_la_LIBADD =
+libtotemscrsaver_la_LIBADD = -lX11 -L/usr/X11/lib -L/usr/X11R6/lib
 am_libtotemscrsaver_la_OBJECTS =  \
 	libtotemscrsaver_la-totem-scrsaver.lo
 libtotemscrsaver_la_OBJECTS = $(am_libtotemscrsaver_la_OBJECTS)
diff -Nurd -x'*~' evince-2.26.2.orig/libdocument/Makefile.in evince-2.26.2/libdocument/Makefile.in
--- evince-2.26.2.orig/libdocument/Makefile.in	2009-05-18 16:53:53.000000000 -0400
+++ evince-2.26.2/libdocument/Makefile.in	2011-01-06 01:57:56.000000000 -0500
@@ -431,7 +431,7 @@
 	-export-symbols-regex "^ev_*"	\
 	$(AM_LDFLAGS)
 
-libevdocument_la_LIBADD = $(LIBDOCUMENT_LIBS)
+libevdocument_la_LIBADD = $(LIBDOCUMENT_LIBS) $(INTLLIBS)
 BUILT_SOURCES = \
 	ev-document-type-builtins.c	\
 	ev-document-type-builtins.h
