diff -Nurd -x'*~' evince-2.24.2.orig/cut-n-paste/evmountoperation/Makefile.am evince-2.24.2/cut-n-paste/evmountoperation/Makefile.am
--- evince-2.24.2.orig/cut-n-paste/evmountoperation/Makefile.am	2008-11-24 14:09:31.000000000 -0500
+++ evince-2.24.2/cut-n-paste/evmountoperation/Makefile.am	2008-11-28 21:53:39.000000000 -0500
@@ -4,7 +4,7 @@
 	ev-mount-operation.h		\
 	ev-mount-operation.c
 
-libevmountoperation_la_CFLAGS =		\
+libevmountoperation_la_CPPFLAGS =		\
 	-I$(top_srcdir)/lib		\
 	$(LIB_CFLAGS)			\
 	$(WARNING_CFLAGS)		\
diff -Nurd -x'*~' evince-2.24.2.orig/cut-n-paste/gedit-message-area/Makefile.am evince-2.24.2/cut-n-paste/gedit-message-area/Makefile.am
--- evince-2.24.2.orig/cut-n-paste/gedit-message-area/Makefile.am	2008-11-24 14:09:31.000000000 -0500
+++ evince-2.24.2/cut-n-paste/gedit-message-area/Makefile.am	2008-11-28 21:53:39.000000000 -0500
@@ -3,7 +3,7 @@
 	gedit-message-area.c	\
 	gedit-message-area.h
 
-libgeditmsgarea_la_CFLAGS =	\
+libgeditmsgarea_la_CPPFLAGS =	\
 	-I$(top_srcdir)/lib	\
 	$(LIB_CFLAGS)		\
 	$(WARNING_CFLAGS)	\
diff -Nurd -x'*~' evince-2.24.2.orig/cut-n-paste/smclient/Makefile.am evince-2.24.2/cut-n-paste/smclient/Makefile.am
--- evince-2.24.2.orig/cut-n-paste/smclient/Makefile.am	2008-11-24 14:09:31.000000000 -0500
+++ evince-2.24.2/cut-n-paste/smclient/Makefile.am	2008-11-28 21:53:39.000000000 -0500
@@ -8,7 +8,7 @@
 	eggsmclient-private.h	\
 	eggsmclient-xsmp.c
 
-libsmclient_la_CFLAGS =				\
+libsmclient_la_CPPFLAGS =				\
 	-I$(top_srcdir)/lib			\
 	-DG_LOG_DOMAIN=\""EggSMClient"\"	\	
 	-DEGG_SM_CLIENT_BACKEND_XSMP		\
diff -Nurd -x'*~' evince-2.24.2.orig/cut-n-paste/totem-screensaver/Makefile.am evince-2.24.2/cut-n-paste/totem-screensaver/Makefile.am
--- evince-2.24.2.orig/cut-n-paste/totem-screensaver/Makefile.am	2008-11-24 14:09:31.000000000 -0500
+++ evince-2.24.2/cut-n-paste/totem-screensaver/Makefile.am	2008-11-28 21:53:39.000000000 -0500
@@ -3,7 +3,7 @@
 	totem-scrsaver.h \
 	totem-scrsaver.c
 
-libtotemscrsaver_la_CFLAGS = \
+libtotemscrsaver_la_CPPFLAGS = \
 	-I$(top_srcdir)/lib	\
 	$(LIB_CFLAGS)		\
 	$(DBUS_CFLAGS)		\
diff -Nurd -x'*~' evince-2.24.2.orig/cut-n-paste/zoom-control/Makefile.am evince-2.24.2/cut-n-paste/zoom-control/Makefile.am
--- evince-2.24.2.orig/cut-n-paste/zoom-control/Makefile.am	2008-11-24 14:09:31.000000000 -0500
+++ evince-2.24.2/cut-n-paste/zoom-control/Makefile.am	2008-11-28 21:53:39.000000000 -0500
@@ -8,7 +8,7 @@
 	ephy-zoom.c				\
 	ephy-zoom.h
 
-libephyzoom_la_CFLAGS = \
+libephyzoom_la_CPPFLAGS = \
 	-I$(top_srcdir)/lib	\
 	$(LIB_CFLAGS)		\
 	$(WARNING_CFLAGS)	\
diff -Nurd -x'*~' evince-2.24.2.orig/properties/Makefile.am evince-2.24.2/properties/Makefile.am
--- evince-2.24.2.orig/properties/Makefile.am	2008-11-24 14:09:31.000000000 -0500
+++ evince-2.24.2/properties/Makefile.am	2008-11-28 21:53:39.000000000 -0500
@@ -22,7 +22,7 @@
 nautilus_LTLIBRARIES = $(EVINCE_PROPERTIES_PAGE)
 
 nautilusdir=$(NAUTILUS_EXTENSION_DIR)
-libevince_properties_page_la_CFLAGS = -I$(top_srcdir)
+libevince_properties_page_la_CPPFLAGS = -I$(top_srcdir)
 libevince_properties_page_la_SOURCES = 	\
 	ev-properties-main.c
 
