diff -Nurd -x'*~' mozilla-release.orig/ipc/chromium/src/base/file_util_linux.cc mozilla-release/ipc/chromium/src/base/file_util_linux.cc
--- mozilla-release.orig/ipc/chromium/src/base/file_util_linux.cc	2011-11-20 14:44:33.000000000 -0500
+++ mozilla-release/ipc/chromium/src/base/file_util_linux.cc	2015-09-24 03:54:43.000000000 -0400
@@ -5,6 +5,9 @@
 #include "base/file_util.h"
 
 #include <fcntl.h>
+#if defined(ANDROID) || defined(OS_POSIX)
+#include <unistd.h>
+#endif
 
 #include <string>
 #include <vector>
diff -Nurd -x'*~' mozilla-release.orig/ipc/chromium/src/base/message_pump_libevent.cc mozilla-release/ipc/chromium/src/base/message_pump_libevent.cc
--- mozilla-release.orig/ipc/chromium/src/base/message_pump_libevent.cc	2011-11-20 14:44:33.000000000 -0500
+++ mozilla-release/ipc/chromium/src/base/message_pump_libevent.cc	2015-09-24 03:54:43.000000000 -0400
@@ -6,6 +6,9 @@
 
 #include <errno.h>
 #include <fcntl.h>
+#if defined(ANDROID) || defined(OS_POSIX)
+#include <unistd.h>
+#endif
 
 #include "eintr_wrapper.h"
 #include "base/logging.h"
diff -Nurd -x'*~' mozilla-release.orig/ipc/chromium/src/base/time_posix.cc mozilla-release/ipc/chromium/src/base/time_posix.cc
--- mozilla-release.orig/ipc/chromium/src/base/time_posix.cc	2011-11-20 14:44:34.000000000 -0500
+++ mozilla-release/ipc/chromium/src/base/time_posix.cc	2015-09-24 03:54:44.000000000 -0400
@@ -13,6 +13,9 @@
 #else
 #include <time.h>
 #endif
+#if defined(ANDROID) || defined(OS_POSIX)
+#include <unistd.h>
+#endif
 
 #include <limits>
 
diff -Nurd -x'*~' mozilla-release.orig/ipc/chromium/src/chrome/common/ipc_channel_posix.cc mozilla-release/ipc/chromium/src/chrome/common/ipc_channel_posix.cc
--- mozilla-release.orig/ipc/chromium/src/chrome/common/ipc_channel_posix.cc	2011-11-20 14:44:34.000000000 -0500
+++ mozilla-release/ipc/chromium/src/chrome/common/ipc_channel_posix.cc	2015-09-24 04:16:39.000000000 -0400
@@ -11,6 +11,9 @@
 #include <sys/socket.h>
 #include <sys/stat.h>
 #include <sys/un.h>
+#if defined(ANDROID) || defined(OS_POSIX)
+#include <unistd.h>
+#endif
 
 #include <string>
 #include <map>
