diff -Nurd -x'*~' fltk-1.1.5.orig/configure fltk-1.1.5/configure
--- fltk-1.1.5.orig/configure	2004-10-19 16:01:57.000000000 -0400
+++ fltk-1.1.5/configure	2004-10-29 06:24:15.000000000 -0400
@@ -6674,19 +6674,45 @@
     fi
 fi
 
-HLINKS=
+# Don't make symlinks if filesystem is not case sensitive...
+# (can override with --with-links=yes and --with-links=no)
+if   test "x$with_links" = xyes; then
+    HLINKS=
+elif test "x$with_links" = xno;  then
+    HLINKS="#"
+else
+    case $uname in
+	CYGWIN* | MINGW* | Darwin*)
+	    HLINKS="#"
+	    ;;
+	*)
+	    HLINKS=
+	    ;;
+    esac
+fi
+
 POSTBUILD=:
 THREADS=
 
-
-# Check whether --with-links or --without-links was given.
-if test "${with_links+set}" = set; then
-  withval="$with_links"
-
-fi;
-
 case $uname in
     CYGWIN* | MINGW*)
+	graphics_model=Win32
+	;;
+    Darwin*)
+	# can use Carbon (default) or x11 (by request) graphics
+	if test "x$with_x" = xyes; then
+	    graphics_model=x11
+	else
+	    graphics_model=Carbon
+	fi
+	;;
+    *)
+	graphics_model=x11
+	;;
+esac
+
+case $graphics_model in
+    Win32)
 		CFLAGS="-mwindows -DWIN32 $CFLAGS"
 	CXXFLAGS="-mwindows -DWIN32 $CXXFLAGS"
 	LDFLAGS="-mwindows $LDFLAGS"
@@ -6984,14 +7010,9 @@
 
 	    THREADS="threads.exe"
 	fi
-
-	# Don't make symlinks since Windows is not case sensitive.
-	if test "x$with_links" != xyes; then
-		HLINKS="#"
-	fi
 	;;
 
-    Darwin*)
+    Carbon)
         # MacOS X uses Carbon for graphics...
         LIBS="$LIBS -framework Carbon -framework ApplicationServices"
 
@@ -7020,16 +7041,11 @@
 	    GLDEMOS=""
         fi
 
-	# Don't make symlinks because HFS+ is not case sensitive...
-	if test "x$with_links" != xyes; then
-		HLINKS="#"
-	fi
-
 	# Add a postbuild step after linking applications
 	POSTBUILD="/Developer/Tools/Rez -t APPL -o"
 	;;
 
-    *)
+    x11)
 	# All others are UNIX/X11...
 	if test x$have_pthread = xyes; then
 	    cat >>confdefs.h <<\_ACEOF
@@ -9108,11 +9124,6 @@
 _ACEOF
 
 	fi
-
-	# Make symlinks since UNIX/Linux is case sensitive.
-	if test "x$with_links" = xno; then
-		HLINKS="#"
-	fi
 	;;
 esac
 
diff -Nurd -x'*~' fltk-1.1.5.orig/fluid/fluid.cxx fltk-1.1.5/fluid/fluid.cxx
--- fltk-1.1.5.orig/fluid/fluid.cxx	2004-09-08 11:42:47.000000000 -0400
+++ fltk-1.1.5/fluid/fluid.cxx	2004-10-29 05:56:35.000000000 -0400
@@ -617,11 +617,7 @@
   fl_filename_absolute(absolute, sizeof(absolute), flname);
 
   for (i = 0; i < 10; i ++)
-#if defined(WIN32) || defined(__APPLE__)
     if (!strcasecmp(absolute, absolute_history[i])) break;
-#else
-    if (!strcmp(absolute, absolute_history[i])) break;
-#endif // WIN32 || __APPLE__
 
   if (i == 0) return;
 
diff -Nurd -x'*~' fltk-1.1.5.orig/makeinclude.in fltk-1.1.5/makeinclude.in
--- fltk-1.1.5.orig/makeinclude.in	2004-10-18 16:22:21.000000000 -0400
+++ fltk-1.1.5/makeinclude.in	2004-10-29 06:08:58.000000000 -0400
@@ -102,7 +102,7 @@
 CAT3EXT		= @CAT3EXT@
 
 # Be quiet when building...
-.SILENT:
+#.SILENT:
 
 # Build commands and filename extensions...
 .SUFFIXES:	.0 .1 .3 .c .cxx .h .fl .man .o .z $(EXEEXT)
@@ -113,11 +113,9 @@
 	$(POSTBUILD) $@ ../FL/mac.r
 
 .c.o:
-	echo Compiling $<...
 	$(CC) -I.. @PNGINC@ @JPEGINC@ @ZLIBINC@ $(CFLAGS) -c $<
 
 .cxx.o:
-	echo Compiling $<...
 	$(CXX) -I.. @PNGINC@ @JPEGINC@ @ZLIBINC@ $(CXXFLAGS) -c $<
 
 .man.0 .man.1 .man.3:
diff -Nurd -x'*~' fltk-1.1.5.orig/src/Makefile fltk-1.1.5/src/Makefile
--- fltk-1.1.5.orig/src/Makefile	2004-10-18 16:22:25.000000000 -0400
+++ fltk-1.1.5/src/Makefile	2004-10-29 06:09:30.000000000 -0400
@@ -194,7 +194,6 @@
 	$(IMGLIBNAME) $(IMGDSONAME)
 
 $(LIBNAME): $(OBJECTS)
-	echo $(LIBCOMMAND) $@ ...
 	$(RM) $@
 	$(LIBCOMMAND) $@ $(OBJECTS)
 	$(RANLIB) $@
@@ -212,12 +211,11 @@
 	$(LN) libfltk.sl.1.1 libfltk.sl
 
 libfltk.1.1.dylib: $(OBJECTS)
-	echo $(DSOCOMMAND) $@ ...
 	$(DSOCOMMAND) $@ \
 		-install_name $(libdir)/$@ \
 		-current_version 1.1.4 \
 		-compatibility_version 1.1.0 \
-		$(OBJECTS) $(LDLIBS)
+		$(OBJECTS) $(LDLIBS) $(FONTCONFIGLIB)
 	$(RM) libfltk.dylib
 	$(LN) libfltk.1.1.dylib libfltk.dylib
 
@@ -248,12 +246,11 @@
 	$(LN) libfltk_forms.sl.1.1 libfltk_forms.sl
 
 libfltk_forms.1.1.dylib: $(FLOBJECTS) libfltk.1.1.dylib
-	echo $(DSOCOMMAND) $@ ...
 	$(DSOCOMMAND) $@ \
 		-install_name $(libdir)/$@ \
 		-current_version 1.1.4 \
 		-compatibility_version 1.1.0 \
-		$(FLOBJECTS) $(LDLIBS) -L. -lfltk
+		$(FLOBJECTS) -L. $(LDLIBS) -lfltk
 	$(RM) libfltk_forms.dylib
 	$(LN) libfltk_forms.1.1.dylib libfltk_forms.dylib
 
@@ -284,12 +281,11 @@
 	$(LN) libfltk_gl.sl.1.1 libfltk_gl.sl
 
 libfltk_gl.1.1.dylib: $(GLOBJECTS) libfltk.1.1.dylib
-	echo $(DSOCOMMAND) $@ ...
 	$(DSOCOMMAND) $@ \
 		-install_name $(libdir)/$@ \
 		-current_version 1.1.4 \
 		-compatibility_version 1.1.0 \
-		$(GLOBJECTS) $(GLDLIBS) -L. -lfltk
+		$(GLOBJECTS) -L. $(GLDLIBS) -lfltk
 	$(RM) libfltk_gl.dylib
 	$(LN) libfltk_gl.1.1.dylib libfltk_gl.dylib
 
@@ -320,12 +316,11 @@
 	$(LN) libfltk_images.sl.1.1 libfltk_images.sl
 
 libfltk_images.1.1.dylib: $(IMGOBJECTS) libfltk.1.1.dylib
-	echo $(DSOCOMMAND) $@ ...
 	$(DSOCOMMAND) $@ \
 		-install_name $(libdir)/$@ \
 		-current_version 1.1.4 \
 		-compatibility_version 1.1.0 \
-		$(IMGOBJECTS) $(LDLIBS) $(IMAGELIBS) -L. -lfltk
+		$(IMGOBJECTS) -L. $(LDLIBS) $(IMAGELIBS) -lfltk
 	$(RM) libfltk_images.dylib
 	$(LN) libfltk_images.1.1.dylib libfltk_images.dylib
 
diff -Nurd -x'*~' fltk-1.1.5.orig/src/filename_absolute.cxx fltk-1.1.5/src/filename_absolute.cxx
--- fltk-1.1.5.orig/src/filename_absolute.cxx	2004-04-11 00:38:59.000000000 -0400
+++ fltk-1.1.5/src/filename_absolute.cxx	2004-10-29 05:56:35.000000000 -0400
@@ -154,11 +154,7 @@
        *slash != '\0' && *newslash != '\0';
        slash ++, newslash ++)
     if (isdirsep(*slash) && isdirsep(*newslash)) continue;
-#if defined(WIN32) || defined(__EMX__) || defined(__APPLE__)
     else if (tolower(*slash) != tolower(*newslash)) break;
-#else
-    else if (*slash != *newslash) break;
-#endif // WIN32 || __EMX__ || __APPLE__
 
   if (*newslash == '\0' && *slash != '\0' && !isdirsep(*slash))
     newslash--;
diff -Nurd -x'*~' fltk-1.1.5.orig/test/colbrowser.cxx fltk-1.1.5/test/colbrowser.cxx
--- fltk-1.1.5.orig/test/colbrowser.cxx	2004-04-11 00:39:01.000000000 -0400
+++ fltk-1.1.5/test/colbrowser.cxx	2004-10-29 05:56:35.000000000 -0400
@@ -42,21 +42,7 @@
 static int load_browser(char *);
 
 /* the RGB data file does not have a standard location on unix. */
-
-#ifdef __VMS
-  static const char *rgbfile = "SYS$MANAGER:DECW$RGB.DAT";
-#else
-#ifdef __EMX__   /* OS2 */
-#include <X11/XlibInt.h>
-  static const char *rgbfile = "/XFree86/lib/X11/rgb.txt";
-#else
-#ifdef __FreeBSD__
-  static const char *rgbfile = "/usr/X11R6/lib/X11/rgb.txt";
-#else
-   static const char *rgbfile = "/usr/lib/X11/rgb.txt";
-#endif
-#endif
-#endif
+static const char *rgbfile = "/usr/X11R6/lib/X11/rgb.txt";
 
 typedef struct { int r, g, b; } RGBdb;
 
