diff -Nurd -x'*~' gamin-0.1.10.orig/configure gamin-0.1.10/configure
--- gamin-0.1.10.orig/configure	2008-11-24 10:23:27.000000000 -0500
+++ gamin-0.1.10/configure	2017-05-22 02:22:00.000000000 -0400
@@ -2177,7 +2177,7 @@
 
 
 
-FAM_VERSION_INFO="0:0:0"
+FAM_VERSION_INFO="1:0:1"
 
 
 
diff -Nurd -x'*~' gamin-0.1.10.orig/libgamin/Makefile.in gamin-0.1.10/libgamin/Makefile.in
--- gamin-0.1.10.orig/libgamin/Makefile.in	2008-11-24 10:25:07.000000000 -0500
+++ gamin-0.1.10/libgamin/Makefile.in	2017-05-22 01:43:02.000000000 -0400
@@ -255,12 +255,12 @@
 
 CLEANFILES = gam_error.c gam_event.c
 libgamin_1_la_LIBADD = 
-libgamin_1_la_LDFLAGS = -Wl,--version-script=$(srcdir)/gamin_sym.version \
+libgamin_1_la_LDFLAGS = \
                         -version-info @GAMIN_VERSION_INFO@ @THREAD_LIBS@
 
 libfam_la_SOURCES = $(libgamin_1_la_SOURCES)
 libfam_la_LIBADD = $(libgamin_1_la_LIBADD)
-libfam_la_LDFLAGS = -Wl,--version-script=$(srcdir)/gamin_sym.version	\
+libfam_la_LDFLAGS = \
                     -version-info @FAM_VERSION_INFO@ @THREAD_LIBS@
 
 gamin_SOURCES = gamin.c
diff -Nurd -x'*~' gamin-0.1.10.orig/libgamin/gam_api.c gamin-0.1.10/libgamin/gam_api.c
--- gamin-0.1.10.orig/libgamin/gam_api.c	2007-08-27 06:21:03.000000000 -0400
+++ gamin-0.1.10/libgamin/gam_api.c	2017-05-22 01:43:02.000000000 -0400
@@ -14,6 +14,8 @@
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <sys/uio.h>
+#include <string.h>
+#include <sys/select.h>
 #include "fam.h"
 #include "gam_protocol.h"
 #include "gam_data.h"
@@ -358,7 +360,7 @@
     int retries = 0;
 
   retry_start:
-    fd = socket(PF_UNIX, SOCK_STREAM, 0);
+    fd = socket(AF_UNIX, SOCK_STREAM, 0);
     if (fd < 0) {
         gam_error(DEBUG_INFO, "Failed to create unix socket\n");
         return (-1);
diff -Nurd -x'*~' gamin-0.1.10.orig/python/Makefile.in gamin-0.1.10/python/Makefile.in
--- gamin-0.1.10.orig/python/Makefile.in	2008-11-24 10:25:07.000000000 -0500
+++ gamin-0.1.10/python/Makefile.in	2017-05-22 01:45:43.000000000 -0400
@@ -222,9 +222,9 @@
 top_srcdir = @top_srcdir@
 SUBDIRS = tests
 INCLUDES = \
-	-I$(PYTHON_INCLUDES)	\
 	-I$(top_srcdir)/libgamin\
-	-I$(top_srcdir)/lib
+	-I$(top_srcdir)/lib	\
+	-I$(PYTHON_INCLUDES)
 
 python_LTLIBRARIES = _gamin.la
 _gamin_la_SOURCES = gamin.c
diff -Nurd -x'*~' gamin-0.1.10.orig/server/gam_channel.c gamin-0.1.10/server/gam_channel.c
--- gamin-0.1.10.orig/server/gam_channel.c	2007-07-04 09:36:49.000000000 -0400
+++ gamin-0.1.10/server/gam_channel.c	2017-05-22 01:43:02.000000000 -0400
@@ -7,6 +7,7 @@
 #include <sys/stat.h>
 #include <sys/un.h>
 #include <sys/uio.h>
+#include <string.h>
 #include "gam_error.h"
 #include "gam_connection.h"
 #include "gam_channel.h"
@@ -621,7 +622,7 @@
     int fd;
     struct sockaddr_un addr;
 
-    fd = socket(PF_UNIX, SOCK_STREAM, 0);
+    fd = socket(AF_UNIX, SOCK_STREAM, 0);
     if (fd < 0) {
         GAM_DEBUG(DEBUG_INFO, "Failed to create unix socket");
         return (-1);
diff -Nurd -x'*~' gamin-0.1.10.orig/server/gam_conf.c gamin-0.1.10/server/gam_conf.c
--- gamin-0.1.10.orig/server/gam_conf.c	2007-07-04 09:36:49.000000000 -0400
+++ gamin-0.1.10/server/gam_conf.c	2017-05-22 01:43:02.000000000 -0400
@@ -7,6 +7,7 @@
 #include <sys/stat.h>
 #include <unistd.h>
 #include <string.h>
+#include <strings.h>
 #include <errno.h>
 #include <glib.h>
 
diff -Nurd -x'*~' gamin-0.1.10.orig/server/gam_eq.c gamin-0.1.10/server/gam_eq.c
--- gamin-0.1.10.orig/server/gam_eq.c	2007-07-04 09:36:49.000000000 -0400
+++ gamin-0.1.10/server/gam_eq.c	2017-05-22 01:55:23.000000000 -0400
@@ -124,7 +124,7 @@
 {
 	gboolean done_work = FALSE;
 	if (!eq)
-		return;
+		return done_work;
 
 #ifdef GAM_EQ_VERBOSE
 	GAM_DEBUG(DEBUG_INFO, "gam_eq: Flushing event queue for %s\n", gam_connection_get_pidname (conn));
diff -Nurd -x'*~' gamin-0.1.10.orig/server/gam_poll_basic.c gamin-0.1.10/server/gam_poll_basic.c
--- gamin-0.1.10.orig/server/gam_poll_basic.c	2007-07-04 09:46:29.000000000 -0400
+++ gamin-0.1.10/server/gam_poll_basic.c	2017-05-22 01:43:02.000000000 -0400
@@ -38,7 +38,6 @@
 #include "gam_event.h"
 #include "gam_excludes.h"
 
-#define VERBOSE_POLL
 
 static gboolean gam_poll_basic_add_subscription(GamSubscription * sub);
 static gboolean gam_poll_basic_remove_subscription(GamSubscription * sub);
diff -Nurd -x'*~' gamin-0.1.10.orig/tests/Makefile.in gamin-0.1.10/tests/Makefile.in
--- gamin-0.1.10.orig/tests/Makefile.in	2008-11-24 10:25:07.000000000 -0500
+++ gamin-0.1.10/tests/Makefile.in	2017-05-22 01:43:02.000000000 -0400
@@ -208,7 +208,7 @@
 testgam_SOURCES = testing.c
 testgam_LDFLAGS = 
 testgam_DEPENDENCIES = $(DEPS)
-testgam_LDADD = $(LDADDS) -L$(top_builddir)/libgamin -lgamin-1
+testgam_LDADD = $(LDADDS) $(top_builddir)/libgamin/libgamin-1.la
 all: all-am
 
 .SUFFIXES:
