diff -Nurd -x'*~' gOpenMol-2.30.orig/src/configure gOpenMol-2.30/src/configure
--- gOpenMol-2.30.orig/src/configure	2003-09-19 08:32:03.000000000 -0400
+++ gOpenMol-2.30/src/configure	2003-09-25 05:57:38.000000000 -0400
@@ -18449,7 +18449,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   gom4_cv_tclsh_LIBRARY="${TCL_LIBRARY-`\
-	    echo 'puts [info library]' | $TCLSH`}"
+	    echo 'fconfigure stdout -translation lf; puts [info library]' | $TCLSH`}"
 fi
 echo "$as_me:$LINENO: result: $gom4_cv_tclsh_LIBRARY" >&5
 echo "${ECHO_T}$gom4_cv_tclsh_LIBRARY" >&6
@@ -18794,7 +18794,7 @@
     # }}}
 
 # Search for Tcl library.
-gom_tcl_version=`echo 'puts [info tclversion]' | $TCLSH`
+gom_tcl_version=`echo 'fconfigure stdout -translation lf; puts [info tclversion]' | $TCLSH`
 # Check for the library.
 gom4_GOM_CHECK_LIB_found=yes
 for gom4_GOM_CHECK_LIB_header in tcl.h
diff -Nurd -x'*~' gOpenMol-2.30.orig/src/general/trajectory.c gOpenMol-2.30/src/general/trajectory.c
--- gOpenMol-2.30.orig/src/general/trajectory.c	2003-09-09 08:49:05.000000000 -0400
+++ gOpenMol-2.30/src/general/trajectory.c	2003-09-24 05:01:12.000000000 -0400
@@ -399,7 +399,7 @@
 /*************************************************************************/
 IMPLEMENT_GET_MODIFIABLE_POINTER(
     int,FreeAtomListPointer,(int Set),
-    FreeAtomList.free_atom_list,)
+    FreeAtomList.free_atom_list,;)
 /*************************************************************************/
 int gomp_SetFreeAtomListPointer(int Set , int NFreat)
 /*************************************************************************/
@@ -748,7 +748,7 @@
 IMPLEMENT_GET_MODIFIABLE_POINTER(
     float,VelocityXComponentPointer,(void),
     TrajectoryInfo.retrieve_velocity_ready ?
-    TrajectoryInfo.Xvelocities : NULL,)
+    TrajectoryInfo.Xvelocities : NULL,;)
 /*************************************************************************/
 /* gomp_GetVelocityYComponentPointer,                                    */
 /* gomp_GetModifiableVelocityYComponentPointer                           */
@@ -756,7 +756,7 @@
 IMPLEMENT_GET_MODIFIABLE_POINTER(
     float,VelocityYComponentPointer,(void),
     TrajectoryInfo.retrieve_velocity_ready ?
-    TrajectoryInfo.Yvelocities : NULL,)
+    TrajectoryInfo.Yvelocities : NULL,;)
 /*************************************************************************/
 /* gomp_GetVelocityXComponentPointer,                                    */
 /* gomp_GetModifiableVelocityXComponentPointer                           */
@@ -764,7 +764,7 @@
 IMPLEMENT_GET_MODIFIABLE_POINTER(
     float,VelocityZComponentPointer,(void),
     TrajectoryInfo.retrieve_velocity_ready ?
-    TrajectoryInfo.Zvelocities : NULL,)
+    TrajectoryInfo.Zvelocities : NULL,;)
 /*************************************************************************/
 int    gomp_GetForceSpace(int Atoms)
 /*************************************************************************/
@@ -800,21 +800,21 @@
 /*************************************************************************/
 IMPLEMENT_GET_MODIFIABLE_POINTER(
     float,ForceXComponentPointer,(void),
-    TrajectoryInfo.retrieve_force_ready ? TrajectoryInfo.Xforces : NULL,)
+    TrajectoryInfo.retrieve_force_ready ? TrajectoryInfo.Xforces : NULL,;)
 /*************************************************************************/
 /* gomp_GetForceYComponenPointer,                                        */
 /* gomp_GetModifiableForceYComponentPointer                              */
 /*************************************************************************/
 IMPLEMENT_GET_MODIFIABLE_POINTER(
     float,ForceYComponentPointer,(void),
-    TrajectoryInfo.retrieve_force_ready ? TrajectoryInfo.Yforces : NULL,)
+    TrajectoryInfo.retrieve_force_ready ? TrajectoryInfo.Yforces : NULL,;)
 /*************************************************************************/
 /* gomp_GetForceZComponenPointer,                                        */
 /* gomp_GetModifiableForceZComponentPointer                              */
 /*************************************************************************/
 IMPLEMENT_GET_MODIFIABLE_POINTER(
     float,ForceZComponentPointer,(void),
-    TrajectoryInfo.retrieve_force_ready ? TrajectoryInfo.Zforces : NULL,)
+    TrajectoryInfo.retrieve_force_ready ? TrajectoryInfo.Zforces : NULL,;)
 /*************************************************************************/
 int    gomp_SetVelocityMinMax(float Vmin , float Vmax)
 /*************************************************************************/
