diff -Nurd -x'*~' gpsdrive-2.0pre9.orig/configure.ac gpsdrive-2.0pre9/configure.ac
--- gpsdrive-2.0pre9.orig/configure.ac	2003-06-27 02:57:02.000000000 -0400
+++ gpsdrive-2.0pre9/configure.ac	2003-06-27 12:19:40.000000000 -0400
@@ -6,15 +6,15 @@
 
 AC_CONFIG_SRCDIR(src/gpsdrive.c)
 AC_CONFIG_HEADER(config.h)
-dnl AM_INIT_AUTOMAKE(gpsdrive,2.0pre1)
+dnl AM_INIT_AUTOMAKE(gpsdrive,2.0pre9)
 dnl PACKAGE=gpsdrive
-dnl VERSION=2.0pre1
+dnl VERSION=2.0pre9
 dnl AC_SUBST(PACKAGE)
 dnl AC_SUBST(VERSION)
 dnl AC_DEFINE_UNQUOTED(PACKAGE, "$PACKAGE")
 dnl AC_DEFINE_UNQUOTED(VERSION, "$VERSION")
 
-
+#CFLAGS="$CFLAGS -Ddlsym=dlsym_prepend_underscore"
 
 ALL_LINGUAS="de es fr it da nl tr de_AT hu sk sv no pt_BR gr ja"
 
@@ -152,7 +152,7 @@
 dnl ************************
 dnl Checks for header files.
 dnl ************************
-AC_CHECK_HEADERS(sys/types.h sys/termios.h termio.h termios.h sys/stat.h fcntl.h stdio.h unistd.h  netdb.h X11/X.h)
+AC_CHECK_HEADERS(sys/types.h sys/termios.h termio.h termios.h sys/stat.h fcntl.h stdio.h unistd.h  netdb.h X11/X.h linux/inet.h sys/ioctl.h)
 
 
 dnl Checks select argument types
@@ -225,6 +225,24 @@
         echo "Using $CC $CFLAGS"
 ])
 
+dnl ************************
+dnl crypt.h and libcrypt
+dnl ************************
+AC_CHECK_HEADERS(crypt.h)
+AC_CHECK_LIB(crypt,crypt)
+
+dnl ************************
+dnl check dynamic loading flags
+dnl ************************
+AC_LTDL_DLLIB
+AC_LTDL_DLSYM_USCORE
+# putting this here gets the option enabled, but program segfaults
+# putting a non-conditional CFLAGS="$CFLAGS -Ddlsym..." early in
+# configure.ac works correctly, but putting the two AC_ and the
+# conditional CFLAGS= does not.
+#if test x"$libltdl_cv_need_uscore" = xyes; then
+#    AC_SUBST(DLSYM_CFLAGS,'-Ddlsym=dlsym_prepend_underscore') 
+#fi
 
 CFLAGS="$CFLAGS -Imysql"
 
diff -Nurd -x'*~' gpsdrive-2.0pre9.orig/src/Makefile.am gpsdrive-2.0pre9/src/Makefile.am
--- gpsdrive-2.0pre9.orig/src/Makefile.am	2003-06-27 02:57:03.000000000 -0400
+++ gpsdrive-2.0pre9/src/Makefile.am	2003-06-27 12:24:23.000000000 -0400
@@ -12,17 +12,20 @@
 LIBS=
 
 # gpsdrive_LDADD= $(GLIB_LIBS) $(GTK_LIBS) $(PIXLIBS)
-gpsdrive_LDADD=@LIBS@ -lcrypt
+gpsdrive_LDADD=@LIBS@ $(LIBADD_DL)
 #gpsdrive_LDADD= @XLIBS@
 # gpsdrive_CFLAGS=@CFLAGS@
 # libgpsd_a_CFLAGS=@CFLAGS@
 # CFLAGS=
 noinst_LIBRARIES = libgpsd.a
+if DISABLEPLUGINS
+else
 lib_LTLIBRARIES =libfly.la libnautic.la
 libfly_la_SOURCES=fly.c
 libfly_la_LDFLAGS = -version-info 2:0:0
 libnautic_la_SOURCES=nautic.c
 libnautic_la_LDFLAGS = -version-info 2:0:0
+endif
 libgpsd_a_SOURCES=netlib.c nmea_parse.c serial.c  tm.c em.c  \
 	gps.h version.h 
 gpsd_SOURCES= gpsd.c  gpsd.h nmea.h version.h
@@ -52,7 +55,7 @@
 gpsdrive_DATA = gpsdrivesplash.png friendsicon.png map_GPSWORLD.jpg
 
 server_SOURCES=server.c
-server_LDADD = $(LDADD) $(LIBINTL) -lcrypt
+server_LDADD = $(LDADD) $(LIBINTL)
 
 client_SOURCES=client.c
-client_LDADD = $(LDADD) $(LIBINTL) -lcrypt
+client_LDADD = $(LDADD) $(LIBINTL)
diff -Nurd -x'*~' gpsdrive-2.0pre9.orig/src/client.c gpsdrive-2.0pre9/src/client.c
--- gpsdrive-2.0pre9.orig/src/client.c	2003-06-27 02:57:03.000000000 -0400
+++ gpsdrive-2.0pre9/src/client.c	2003-06-27 02:57:00.000000000 -0400
@@ -50,10 +50,14 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#ifdef HAVE_CRYPT_H
 #include <crypt.h>
+#endif
 #include <sys/stat.h>
 #include <fcntl.h>
+#ifdef HAVE_LINUX_INET_H
 #include "linux/inet.h"
+#endif
 #include <unistd.h>
 #include <time.h>
 
diff -Nurd -x'*~' gpsdrive-2.0pre9.orig/src/friends.c gpsdrive-2.0pre9/src/friends.c
--- gpsdrive-2.0pre9.orig/src/friends.c	2003-06-27 02:57:03.000000000 -0400
+++ gpsdrive-2.0pre9/src/friends.c	2003-06-27 02:57:00.000000000 -0400
@@ -57,10 +57,14 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#ifdef HAVE_CRYPT_H
 #include <crypt.h>
+#endif
 #include <sys/stat.h>
 #include <fcntl.h>
+#ifdef HAVE_LINUX_INET_H
 #include "linux/inet.h"
+#endif
 #include <unistd.h>
 #include <time.h>
 #include <gpsdrive.h>
diff -Nurd -x'*~' gpsdrive-2.0pre9.orig/src/friendsd.c gpsdrive-2.0pre9/src/friendsd.c
--- gpsdrive-2.0pre9.orig/src/friendsd.c	2003-06-27 02:57:03.000000000 -0400
+++ gpsdrive-2.0pre9/src/friendsd.c	2003-06-27 02:57:00.000000000 -0400
@@ -60,7 +60,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <signal.h>
+#ifdef HAVE_LINUX_INET_H
 #include "linux/inet.h"
+#endif
 #include <fcntl.h>
 #include <netdb.h>
 #include <gpsdrive.h>
diff -Nurd -x'*~' gpsdrive-2.0pre9.orig/src/gpsdrive.c gpsdrive-2.0pre9/src/gpsdrive.c
--- gpsdrive-2.0pre9.orig/src/gpsdrive.c	2003-06-27 02:57:03.000000000 -0400
+++ gpsdrive-2.0pre9/src/gpsdrive.c	2003-06-27 12:09:51.000000000 -0400
@@ -1922,7 +1922,7 @@
     }
   if ((j != 11) && (j != 12))
     {
-      g_print ("\nGPRMC: wrong number of fields (%d)", j);
+      g_print ("\nGPRMC: wrong number of fields (%ld)", j);
       return;
     }
 
@@ -1933,7 +1933,7 @@
     {
       g_print ("\nRMC Fields:\n");
       for (i = 0; i < j; i++)
-	g_print ("\n%d:%s", i, field[i]);
+	g_print ("\n%ld:%s", i, field[i]);
     }
   sprintf (b, "%c%c:%c%c.%c%c ", field[1][0], field[1][1], field[1][2],
 	   field[1][3], field[1][4], field[1][5]);
@@ -8734,9 +8734,11 @@
   /*** Mod by Arms */
   GtkRequisition requ, *reqptr;
   GtkWidget *mainnotebook;
+#ifndef NOPLUGINS
   GModule *mod1;
   void (*modulefunction) ();
   gchar *modpath;
+#endif
   time_t now = time (NULL);
   struct tm *st = localtime (&now);
   void *handle;
@@ -8829,7 +8831,9 @@
 
   if (usesql)
     {
-      handle = dlopen ("libmysqlclient.so", RTLD_LAZY);
+      handle = dlopen ("@PREFIX@/lib/mysql/libmysqlclient.10.dylib", RTLD_LAZY);
+      if (!handle)
+	handle = dlopen ("libmysqlclient.so", RTLD_LAZY);
       if (!handle)
 	handle = dlopen ("libmysqlclient.so.10", RTLD_LAZY);
       if (!handle)
@@ -9304,6 +9308,7 @@
   setupfunction[mod_setupcounter] = &(friendssetup);
   friendsplace = mod_setupcounter;
 
+#ifndef NOPLUGINS
   if (useplugins)
     {
 /*  fly module */
@@ -9343,6 +9348,7 @@
 	  setupfunction[mod_setupcounter] = modulefunction;
 	}
     }
+#endif
 
 
   mainwindow = gtk_window_new (GTK_WINDOW_TOPLEVEL);
diff -Nurd -x'*~' gpsdrive-2.0pre9.orig/src/serial.c gpsdrive-2.0pre9/src/serial.c
--- gpsdrive-2.0pre9.orig/src/serial.c	2003-06-27 02:57:03.000000000 -0400
+++ gpsdrive-2.0pre9/src/serial.c	2003-06-27 02:57:00.000000000 -0400
@@ -29,6 +29,10 @@
 #define tcflag_t	ushort
 #endif
 
+#ifdef HAVE_SYS_IOCTL_H
+#include <sys/ioctl.h>
+#endif
+
 #include "gpsd.h"
 
 #define DEFAULTPORT "2947"
diff -Nurd -x'*~' gpsdrive-2.0pre9.orig/src/server.c gpsdrive-2.0pre9/src/server.c
--- gpsdrive-2.0pre9.orig/src/server.c	2003-06-27 02:57:03.000000000 -0400
+++ gpsdrive-2.0pre9/src/server.c	2003-06-27 02:57:00.000000000 -0400
@@ -53,7 +53,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <signal.h>
+#ifdef HAVE_LINUX_INET_H
 #include "linux/inet.h"
+#endif
 #include <fcntl.h>
 #include <netdb.h>
 #include <gpsdrive.h>
