diff -Nurd -x'*~' granule-1.2.3.orig/src/DeckPlayer.cpp granule-1.2.3/src/DeckPlayer.cpp
--- granule-1.2.3.orig/src/DeckPlayer.cpp	2006-10-07 22:46:30.000000000 -0400
+++ granule-1.2.3/src/DeckPlayer.cpp	2007-01-03 02:34:03.000000000 -0500
@@ -240,7 +240,7 @@
 		Glib::ustring lhs (text1);
 		Glib::ustring rhs (text2);
 
-		for (int idx = 0; idx < lhs.size () && idx < rhs.size (); idx++) 
+		for (unsigned int idx = 0; idx < lhs.size () && idx < rhs.size (); idx++) 
 		{
 			if (lhs [idx] != rhs [idx]) 
 			{
diff -Nurd -x'*~' granule-1.2.3.orig/src/DeckPlayer.h granule-1.2.3/src/DeckPlayer.h
--- granule-1.2.3.orig/src/DeckPlayer.h	2006-10-05 22:04:47.000000000 -0400
+++ granule-1.2.3/src/DeckPlayer.h	2007-01-03 02:26:02.000000000 -0500
@@ -99,7 +99,7 @@
 	Gdk::Color       m_black;
 
 	Gtk::SpinButton* m_test_line;         // Which line to compare for answer.
-	int              m_line_num;          // Number of line to test
+	unsigned int     m_line_num;          // Number of line to test
 };
 
 /**=============================================================================
diff -Nurd -x'*~' granule-1.2.3.orig/src/DeckView.h granule-1.2.3/src/DeckView.h
--- granule-1.2.3.orig/src/DeckView.h	2006-09-20 21:25:20.000000000 -0400
+++ granule-1.2.3/src/DeckView.h	2007-01-03 02:24:22.000000000 -0500
@@ -165,7 +165,7 @@
 	Gtk::Label*  m_label_count;
 
 	VDeck&       m_deck;
-	int          m_row_idx;		// The row selected in the model
+	unsigned int m_row_idx;		// The row selected in the model
 	bool         m_modified;	// Has Deck been modified to invalidate
 
 #ifdef IS_HILDON
diff -Nurd -x'*~' granule-1.2.3.orig/src/Granule-main.cpp granule-1.2.3/src/Granule-main.cpp
--- granule-1.2.3.orig/src/Granule-main.cpp	2006-09-30 16:42:27.000000000 -0400
+++ granule-1.2.3/src/Granule-main.cpp	2007-01-03 01:01:44.000000000 -0500
@@ -68,6 +68,7 @@
 "                                                                           \n";
 //------------------------------------------------------------------------------
 
+#include "libintl.h"
 #include "Granule-main.h"
 #include "Granule.h"
 #include "Intern.h"
diff -Nurd -x'*~' granule-1.2.3.orig/src/GrappConf.cpp granule-1.2.3/src/GrappConf.cpp
--- granule-1.2.3.orig/src/GrappConf.cpp	2006-09-30 16:42:27.000000000 -0400
+++ granule-1.2.3/src/GrappConf.cpp	2007-01-03 02:36:42.000000000 -0500
@@ -249,7 +249,7 @@
     trace_with_mask("GrappConf::add_document_history",GUITRACE);
 
     string fullpath;
-	int hsz = 0;
+	unsigned int hsz = 0;
 
     if (m_path_name.size () == 0) {
 		fullpath = m_proj_name;
@@ -260,7 +260,7 @@
 
 	if (fullpath != UNKNOWN_FNAME) {
 		if ((hsz = m_history.size ()) > 0) {
-			for (int idx = 0; idx < hsz; idx++) {
+			for (unsigned int idx = 0; idx < hsz; idx++) {
 				if (m_history [idx] == fullpath) {
 					m_history.erase (m_history.begin () + idx);
 					break;
@@ -292,7 +292,7 @@
     if ((hsz = m_history.size ()) == 0) {
 		return;
 	}
-	for (int idx = 0; idx < hsz; idx++) {
+	for (unsigned int idx = 0; idx < hsz; idx++) {
 		if (m_history [idx] == fullpath_) {
 			m_history.erase (m_history.begin () + idx);
 			break;
diff -Nurd -x'*~' granule-1.2.3.orig/src/GrappConf.h granule-1.2.3/src/GrappConf.h
--- granule-1.2.3.orig/src/GrappConf.h	2006-09-06 22:34:04.000000000 -0400
+++ granule-1.2.3/src/GrappConf.h	2007-01-03 02:27:46.000000000 -0500
@@ -234,7 +234,7 @@
     string m_user_name;			// Real user name from /etc/passwd
 
     DHList         m_history;	        // CardFile history list.
-    int            m_history_size;      // Maximum size of the history list.
+    unsigned int   m_history_size;      // Maximum size of the history list.
     ASSA::IniFile* m_config;
     flipside_t     m_flipside_history;	// Which side to show for each Deck.
     bool           m_remove_dups;       // Purge CardDeck of duplicate VCards
