diff -Nurd -x'*~' libgadu-1.11.2.orig/configure.ac libgadu-1.11.2/configure.ac
--- libgadu-1.11.2.orig/configure.ac	2012-06-18 17:09:32.000000000 -0400
+++ libgadu-1.11.2/configure.ac	2012-08-26 15:54:54.000000000 -0400
@@ -17,8 +17,8 @@
 AC_PROG_LIBTOOL
 AC_SUBST(LIBTOOL_DEPS)
 
-AC_SUBST(INCLUDES)
 dnl Zależności dla pkgconfig
+AC_SUBST(INCLUDES_PUBLIC)
 AC_SUBST(REQUIRES_PRIVATE)
 AC_SUBST(REQUIRES_PUBLIC)
 AC_SUBST(LIBS_PRIVATE)
@@ -256,8 +256,8 @@
 	# Note: not adding to LIBS_PUBLIC, because "REQUIRES_PUBLIC" takes care
 	# of that for pkg-config.
 	LIBS="$LIBS $OPENSSL_LIBS"
-	CFLAGS="$CFLAGS $OPENSSL_INCLUDES"
-	INCLUDES="$INCLUDES $OPENSSL_INCLUDES"
+	CPPFLAGS="$CPPFLAGS $OPENSSL_INCLUDES"
+	INCLUDES_PUBLIC="$INCLUDES_PUBLIC $OPENSSL_INCLUDES"
 	REQUIRES_PUBLIC="$REQUIRES_PUBLIC openssl"
 	AC_DEFINE([GG_CONFIG_HAVE_OPENSSL], [], [Defined if libgadu was compiled and linked with OpenSSL support.])
 fi
@@ -271,12 +271,14 @@
 dnl
 
 AC_SUBST([SUBDIRS_TEST])
-AC_SUBST([OPTIONAL_TESTS_GLIBC])
+AC_SUBST([OPTIONAL_TESTS_GLIBC_AUTOMATIC])
+AC_SUBST([OPTIONAL_TESTS_GLIBC_MANUAL])
 AC_SUBST([OPTIONAL_TESTS_PERL])
 AC_SUBST([OPTIONAL_TESTS_VOICE7])
 AC_SUBST([OPTIONAL_TESTS_SEARCH])
 SUBDIRS_TEST=""
-OPTIONAL_TESTS_GLIBC=""
+OPTIONAL_TESTS_GLIBC_AUTOMATIC=""
+OPTIONAL_TESTS_GLIBC_MANUAL=""
 OPTIONAL_TESTS_PERL=""
 OPTIONAL_TESTS_VOICE7=""
 OPTIONAL_TESTS_SEARCH=""
@@ -291,7 +293,7 @@
 	dnl  Sprawdź, czy można korzystać z __funkcji w libc.
 	dnl 
 
-	AC_CHECK_LIB(c, __connect, [OPTIONAL_TESTS_GLIBC="connect resolver packet"])
+	AC_CHECK_LIB(c, __connect, [OPTIONAL_TESTS_GLIBC_AUTOMATIC="connect resolver packet"; OPTIONAL_TESTS_GLIBC_MANUAL="dcc7"])
 
 	dnl
 	dnl  libgsm i libspeex dla voice7
@@ -336,7 +338,6 @@
 	dnl  Sprawdź, czy mamy libxml2 do testów automatycznych
 	dnl
 
-	export PKG_CONFIG_PATH="$PKG_CONFIG_PATH:/usr/local/lib/pkgconfig"
 
 	PKG_PROG_PKG_CONFIG
 
diff -Nurd -x'*~' libgadu-1.11.2.orig/examples/Makefile.am libgadu-1.11.2/examples/Makefile.am
--- libgadu-1.11.2.orig/examples/Makefile.am	2011-05-29 07:41:30.000000000 -0400
+++ libgadu-1.11.2/examples/Makefile.am	2012-08-26 15:17:12.000000000 -0400
@@ -1,20 +1,20 @@
 noinst_PROGRAMS = send httphash conn-async register-sync register-async remind-sync remind-async status token-sync token-async
 LDADD = ../src/libgadu.la
 
-httphash_CFLAGS = -DGG_IGNORE_DEPRECATED
+httphash_CPPFLAGS = -DGG_IGNORE_DEPRECATED
 
 register_sync_SOURCES = register.c
 
 register_async_SOURCES = register.c
-register_async_CFLAGS = -DASYNC
+register_async_CPPFLAGS = -DASYNC
 
 remind_sync_SOURCES = remind.c
 
 remind_async_SOURCES = remind.c
-remind_async_CFLAGS = -DASYNC
+remind_async_CPPFLAGS = -DASYNC
 
 token_sync_SOURCES = token.c
 
 token_async_SOURCES = token.c
-token_async_CFLAGS = -DASYNC
+token_async_CPPFLAGS = -DASYNC
 
diff -Nurd -x'*~' libgadu-1.11.2.orig/pkgconfig/libgadu.pc.in libgadu-1.11.2/pkgconfig/libgadu.pc.in
--- libgadu-1.11.2.orig/pkgconfig/libgadu.pc.in	2011-11-04 16:54:35.000000000 -0400
+++ libgadu-1.11.2/pkgconfig/libgadu.pc.in	2012-08-26 15:54:12.000000000 -0400
@@ -10,5 +10,5 @@
 Requires.private: @REQUIRES_PRIVATE@
 Libs: -L${libdir} -lgadu @LIBS_PUBLIC@
 Libs.private: -L${libdir} -lgadu @LIBS_PRIVATE@
-Cflags: -I${includedir} @INCLUDES@
+Cflags: -I${includedir} @INCLUDES_PUBLIC@
 
diff -Nurd -x'*~' libgadu-1.11.2.orig/src/Makefile.am libgadu-1.11.2/src/Makefile.am
--- libgadu-1.11.2.orig/src/Makefile.am	2011-11-04 18:46:19.000000000 -0400
+++ libgadu-1.11.2/src/Makefile.am	2012-08-26 15:17:01.000000000 -0400
@@ -1,5 +1,5 @@
 lib_LTLIBRARIES = libgadu.la
 libgadu_la_SOURCES = common.c dcc.c dcc7.c debug.c deflate.c encoding.c events.c handlers.c http.c libgadu.c message.c obsolete.c pubdir.c pubdir50.c resolver.c sha1.c
-libgadu_la_CFLAGS = -I$(top_srcdir) -I$(top_srcdir)/include -DGG_IGNORE_DEPRECATED
+libgadu_la_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/include -DGG_IGNORE_DEPRECATED
 libgadu_la_LDFLAGS = -version-number 3:13 -export-symbols $(srcdir)/libgadu.sym
 EXTRA_DIST = libgadu.sym
diff -Nurd -x'*~' libgadu-1.11.2.orig/test/automatic/Makefile.am libgadu-1.11.2/test/automatic/Makefile.am
--- libgadu-1.11.2.orig/test/automatic/Makefile.am	2011-11-04 16:54:35.000000000 -0400
+++ libgadu-1.11.2/test/automatic/Makefile.am	2012-08-26 15:38:46.000000000 -0400
@@ -1,12 +1,12 @@
 TESTS = convert message2 message1 $(OPTIONAL_TESTS_GLIBC) $(OPTIONAL_TESTS_PERL)
-check_PROGRAMS = convert message2 message1 $(OPTIONAL_TESTS_GLIBC) $(OPTIONAL_TESTS_PERL)
+check_PROGRAMS = convert message2 message1 $(OPTIONAL_TESTS_GLIBC_AUTOMATIC) $(OPTIONAL_TESTS_PERL)
 EXTRA_PROGRAMS = convert message2 message1 connect packet resolver protocol
 
-CFLAGS = -Wall -ggdb -DGG_IGNORE_DEPRECATED
+AM_CFLAGS = -ggdb -DGG_IGNORE_DEPRECATED
 
 protocol_SOURCES = protocol.c
 nodist_protocol_SOURCES = script/script.c
-protocol_CFLAGS = -Iscript
+protocol_CPPFLAGS = -Iscript
 protocol_LDADD = $(top_builddir)/src/libgadu.la
 
 convert_SOURCES = convert.c $(top_builddir)/src/encoding.c
@@ -15,7 +15,7 @@
 
 message2_SOURCES = message2.c $(top_builddir)/src/message.c
 message2_LDADD = @LIBXML2_LIBS@
-message2_CFLAGS = @LIBXML2_CFLAGS@
+message2_CPPFLAGS = @LIBXML2_CFLAGS@
 
 connect_LDADD = $(top_builddir)/src/libgadu.la
 
diff -Nurd -x'*~' libgadu-1.11.2.orig/test/automatic/protocol.c libgadu-1.11.2/test/automatic/protocol.c
--- libgadu-1.11.2.orig/test/automatic/protocol.c	2011-05-29 07:41:28.000000000 -0400
+++ libgadu-1.11.2/test/automatic/protocol.c	2012-08-26 16:20:32.000000000 -0400
@@ -6,19 +6,19 @@
 #include <time.h>
 #include <string.h>
 #include <errno.h>
-#include <signal.h>
 #include <ctype.h>
 #include <arpa/inet.h>
 #include <sys/select.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
+#include <netdb.h>
 
 #include <libgadu.h>
 
 #include "script.h"
 
-#define LOCALHOST "127.0.67.67"
-#define LOCALPORT 17219
+#define LOCALHOST_NAME "localhost"
+#define LOCALHOST_ADDR "127.0.0.1"
 
 #define debug(msg...) \
 	do { \
@@ -44,12 +44,6 @@
 static int outbuflen = 0;
 static int fd = -1;	/* connected socket */
 
-int buffered_write(char *data, int len)
-{
-
-	return 0;
-}
-
 int main(int argc, char **argv)
 {
 	struct gg_login_params glp;
@@ -57,12 +51,15 @@
 	int lfd;	/* listening socket */
 	int value = 1;
 	struct sockaddr_in sin;
+	socklen_t sin_len;
 	char inbuf[4096];
 	int inbuflen = 0;
 	int state = 0;
 	time_t last = 0;
+	struct hostent *he;
+	uint32_t server_addr;
+	uint16_t server_port;
 
-	signal(SIGPIPE, SIG_IGN);
 	gg_debug_file = stdout;
 	gg_debug_level = ~0;
 
@@ -75,15 +72,30 @@
 
 	memset(&sin, 0, sizeof(sin));
 	sin.sin_family = AF_INET;
-	sin.sin_port = htons(LOCALPORT);
-	sin.sin_addr.s_addr = inet_addr(LOCALHOST);
 
-	if (bind(lfd, (struct sockaddr*) &sin, sizeof(sin))) {
+	he = gethostbyname(LOCALHOST_NAME);
+
+	if (he != NULL)
+	  memcpy(&sin.sin_addr.s_addr, he->h_addr, sizeof(sin.sin_addr.s_addr));
+	else
+	  sin.sin_addr.s_addr = inet_addr(LOCALHOST_ADDR);
+
+	if (bind(lfd, (struct sockaddr*) &sin, sizeof(sin)) == -1) {
 		perror("bind");
 		exit(1);
 	}
 
-	if (listen(lfd, 5)) {
+	sin_len = sizeof(sin);
+
+	if (getsockname(lfd, (struct sockaddr*) &sin, &sin_len) == -1) {
+	  perror("getsockname");
+	  exit(1);
+	}
+
+	server_addr = sin.sin_addr.s_addr;
+	server_port = ntohs(sin.sin_port);
+
+	if (listen(lfd, 5) == -1) {
 		perror("listen");
 		exit(1);
 	}
@@ -98,7 +110,7 @@
 
 		if (script[state].type == ACTION_END) {
 			debug("state %d: ending\n", state);
-			exit(0);
+			break;
 		}
 
 		if (script[state].type == ACTION_LOGIN) {
@@ -108,8 +120,8 @@
 				gg_free_session(gs);
 
 			memcpy(&glp, script[state].glp, sizeof(glp));
-			glp.server_addr = inet_addr(LOCALHOST);
-			glp.server_port = LOCALPORT;
+			glp.server_addr = server_addr;
+			glp.server_port = server_port;
 			glp.async = 1;
 			glp.resolver = GG_RESOLVER_PTHREAD;
 
@@ -263,9 +275,9 @@
 		}
 
 		if (FD_ISSET(lfd, &rds)) {
-			socklen_t sinlen = sizeof(sin);
+			sin_len = sizeof(sin);
 
-			res = accept(lfd, (struct sockaddr*) &sin, &sinlen);
+			res = accept(lfd, (struct sockaddr*) &sin, &sin_len);
 
 			if (res == -1) {
 				perror("accept");
@@ -368,5 +380,7 @@
 		}
 	}
 
+	close(lfd);
+
 	return 0;
 }
diff -Nurd -x'*~' libgadu-1.11.2.orig/test/manual/Makefile.am libgadu-1.11.2/test/manual/Makefile.am
--- libgadu-1.11.2.orig/test/manual/Makefile.am	2011-05-29 07:41:29.000000000 -0400
+++ libgadu-1.11.2/test/manual/Makefile.am	2012-08-26 15:38:43.000000000 -0400
@@ -1,9 +1,8 @@
-check_PROGRAMS = client dcc7 userlist $(OPTIONAL_TESTS_SEARCH) $(OPTIONAL_TESTS_VOICE7)
-EXTRA_PROGRAMS = client dcc7 userlist search voice7
+check_PROGRAMS = client userlist $(OPTIONAL_TESTS_SEARCH) $(OPTIONAL_TESTS_VOICE7) $(OPTIONAL_TESTS_GLIBC_MANUAL)
+EXTRA_PROGRAMS = client userlist search voice7 dcc7
 
-CFLAGS = -DGG_IGNORE_DEPRECATED
+AM_CPPFLAGS = -DGG_IGNORE_DEPRECATED
 
-dcc7_CFLAGS = -Wall
 dcc7_LDADD = $(top_builddir)/src/libgadu.la
 dcc7_SOURCES = dcc7.c userconfig.c userconfig.h
 
@@ -14,10 +13,10 @@
 userlist_SOURCES = userlist.c userconfig.c userconfig.h
 
 search_SOURCES = search.c lib/base64.c lib/base64.h ../../config.h lib/hmac.c lib/hmac.h lib/http.c lib/http.h lib/oauth.c lib/oauth.h lib/oauth_parameter.c lib/oauth_parameter.h lib/sha1.c lib/sha1.h lib/urlencode.c lib/urlencode.h lib/xml.c lib/xml.h
-search_CFLAGS = -Wall -DHAVE_OPENSSL
+search_CPPFLAGS = -DHAVE_OPENSSL
 search_LDFLAGS = -lcurl -lexpat -lssl
 
-voice7_CFLAGS = -Wall -DHAVE_SPEEX -DHAVE_GSM
+voice7_CPPFLAGS = -DHAVE_SPEEX -DHAVE_GSM
 voice7_LDADD = $(top_builddir)/src/libgadu.la
 voice7_LDFLAGS = -lgsm -lspeex
 
