diff -Nurd -x'*~' libXt-1.1.5.orig/configure libXt-1.1.5/configure
--- libXt-1.1.5.orig/configure	2015-05-01 01:36:38.000000000 -0400
+++ libXt-1.1.5/configure	2016-10-22 22:39:13.000000000 -0400
@@ -19303,7 +19303,7 @@
 
 
 # Replaces XFileSearchPathDefault from Imake configs
-XFILESEARCHPATHDEFAULT='$(sysconfdir)/X11/%L/%T/%N%C%S:$(sysconfdir)/X11/%l/%T/%N%C%S:$(sysconfdir)/X11/%T/%N%C%S:$(sysconfdir)/X11/%L/%T/%N%S:$(sysconfdir)/X11/%l/%T/%N%S:$(sysconfdir)/X11/%T/%N%S:$(datadir)/X11/%L/%T/%N%C%S:$(datadir)/X11/%l/%T/%N%C%S:$(datadir)/X11/%T/%N%C%S:$(datadir)/X11/%L/%T/%N%S:$(datadir)/X11/%l/%T/%N%S:$(datadir)/X11/%T/%N%S:$(libdir)/X11/%L/%T/%N%C%S:$(libdir)/X11/%l/%T/%N%C%S:$(libdir)/X11/%T/%N%C%S:$(libdir)/X11/%L/%T/%N%S:$(libdir)/X11/%l/%T/%N%S:$(libdir)/X11/%T/%N%S'
+XFILESEARCHPATHDEFAULT='$(sysconfdir)/X11/%L/%T/%N%C%S:$(sysconfdir)/X11/%l/%T/%N%C%S:$(sysconfdir)/X11/%T/%N%C%S:$(sysconfdir)/X11/%L/%T/%N%S:$(sysconfdir)/X11/%l/%T/%N%S:$(sysconfdir)/X11/%T/%N%S:$(datadir)/X11/%L/%T/%N%C%S:$(datadir)/X11/%l/%T/%N%C%S:$(datadir)/X11/%T/%N%C%S:$(datadir)/X11/%L/%T/%N%S:$(datadir)/X11/%l/%T/%N%S:$(datadir)/X11/%T/%N%S:$(prefix)/lib/X11/%L/%T/%N%C%S:$(prefix)/lib/X11/%l/%T/%N%C%S:$(prefix)/lib/X11/%T/%N%C%S:$(prefix)/lib/X11/%L/%T/%N%S:$(prefix)/lib/X11/%l/%T/%N%S:$(prefix)/lib/X11/%T/%N%S'
 
 
 # Check whether --with-xfile-search-path was given.
@@ -19360,7 +19360,7 @@
 
 case $host_os in
     darwin*)
-        OS_CFLAGS="-Wl,-flat_namespace"
+        OS_CFLAGS=
 	;;
     *)
         OS_CFLAGS=
diff -Nurd -x'*~' libXt-1.1.5.orig/test/Makefile.in libXt-1.1.5/test/Makefile.in
--- libXt-1.1.5.orig/test/Makefile.in	2015-05-01 01:36:41.000000000 -0400
+++ libXt-1.1.5/test/Makefile.in	2016-10-22 22:34:52.000000000 -0400
@@ -910,7 +910,7 @@
 	$(MAKE) $(AM_MAKEFLAGS) $(check_PROGRAMS)
 	$(MAKE) $(AM_MAKEFLAGS) check-TESTS
 check: check-am
-all-am: Makefile
+all-am: Makefile $(check_PROGRAMS)
 installdirs:
 install: install-am
 install-exec: install-exec-am
