diff -Nurd -x'*~' 2.9.orig/BuildUtilities/c4p/gram.y 2.9/BuildUtilities/c4p/gram.y
--- 2.9.orig/BuildUtilities/c4p/gram.y	2009-01-20 11:41:55.000000000 -0500
+++ 2.9/BuildUtilities/c4p/gram.y	2015-04-12 23:21:07.000000000 -0400
@@ -23,7 +23,6 @@
 #include <cstring>
 #include <cstdlib>
 #include <cctype>
-#include <malloc.h>
 
 #include "common.h"
 
diff -Nurd -x'*~' 2.9.orig/CMakeLists.txt 2.9/CMakeLists.txt
--- 2.9.orig/CMakeLists.txt	2010-07-29 03:01:53.000000000 -0400
+++ 2.9/CMakeLists.txt	2015-04-12 23:38:24.000000000 -0400
@@ -297,14 +297,15 @@
   set(HAVE_MPM_QT_SOURCES 1)
 endif(EXISTS ${CMAKE_SOURCE_DIR}/${rel_mpm_qt_dir}/)
 
+# Fink on Mac OS X might have flex not at appropriate version
+SET( FLEX_EXE /usr/bin/flex )
+
 set(required_programs
   bison
   cat
   cp
   date
   diff
-  flex
-  lynx
   perl
   sed
   tr
@@ -580,6 +581,9 @@
   #)
 else(NATIVE_WINDOWS)
   add_definitions(-DMIKTEX_UNIX)
+
+  link_libraries(z)
+  link_libraries(png)
 endif(NATIVE_WINDOWS)
 
 ###############################################################################
diff -Nurd -x'*~' 2.9.orig/Libraries/3rd/obsdcompat/CMakeLIsts.txt 2.9/Libraries/3rd/obsdcompat/CMakeLIsts.txt
--- 2.9.orig/Libraries/3rd/obsdcompat/CMakeLIsts.txt	2010-07-29 03:01:53.000000000 -0400
+++ 2.9/Libraries/3rd/obsdcompat/CMakeLIsts.txt	2015-04-13 00:59:53.000000000 -0400
@@ -25,6 +25,8 @@
 set(HAVE_VSNPRINTF 1)
 set(HAVE_SIZE_T 1)
 set(HAVE_STDINT_T 1)
+set(HAVE_SSIZE_T 1)
+set(HAVE_VA_COPY 1)
 
 configure_file(
     ${CMAKE_CURRENT_SOURCE_DIR}/obsdcompat/config.h.cmake
diff -Nurd -x'*~' 2.9.orig/Libraries/MiKTeX/Core/filetypes.cpp 2.9/Libraries/MiKTeX/Core/filetypes.cpp
--- 2.9.orig/Libraries/MiKTeX/Core/filetypes.cpp	2011-02-28 06:38:55.000000000 -0500
+++ 2.9/Libraries/MiKTeX/Core/filetypes.cpp	2015-04-13 01:29:35.000000000 -0400
@@ -44,6 +44,7 @@
 #define P7_(a, b, c, d, e, f, g) a DELIM b DELIM c DELIM d DELIM e DELIM f DELIM g
 #define P8_(a, b, c, d, e, f, g, h) a DELIM b DELIM c DELIM d DELIM e DELIM f DELIM g DELIM h
 #define P9_(a, b, c, d, e, f, g, h, i) a DELIM b DELIM c DELIM d DELIM e DELIM f DELIM g DELIM h DELIM i
+#define P10_(a, b, c, d, e, f, g, h, i, j) a DELIM b DELIM c DELIM d DELIM e DELIM f DELIM g DELIM h DELIM i DELIM j
 
 #define P_ ListBuilder
 
@@ -782,7 +783,7 @@
 	"PostScript header",
 	0,
 	P2_(".pro", ".enc"),
-	P9_(CURRENT_DIRECTORY,
+	P10_(CURRENT_DIRECTORY,
 	    R_(MIKTEX_PATH_MIKTEX_CONFIG_DIR),
 	    R_(MIKTEX_PATH_DVIPS_DIR),
 	    R_(MIKTEX_PATH_PDFTEX_DIR),
diff -Nurd -x'*~' 2.9.orig/Libraries/MiKTeX/Core/fnamedb.cpp 2.9/Libraries/MiKTeX/Core/fnamedb.cpp
--- 2.9.orig/Libraries/MiKTeX/Core/fnamedb.cpp	2010-10-23 03:57:10.000000000 -0400
+++ 2.9/Libraries/MiKTeX/Core/fnamedb.cpp	2015-04-13 02:53:21.000000000 -0400
@@ -1332,7 +1332,7 @@
   RemoveFileName (pDir, pathFile.Get());
 
   // also from the hash table
-  pair<FileNameHashTable::const_iterator, FileNameHashTable::const_iterator> range =
+  pair<FileNameHashTable::iterator, FileNameHashTable::iterator> range =
     fileNames.equal_range(pathFile.Get());
   if (range.first == range.second)
   {
@@ -1341,7 +1341,7 @@
 			  lpszPath);
   }
   bool removed = false;
-  for (FileNameHashTable::const_iterator it = range.first; it != range.second; ++ it)
+  for (FileNameHashTable::iterator it = range.first; it != range.second; ++ it)
   {
     if (it->second == pDir)
     {
@@ -1388,7 +1388,10 @@
 FileNameDatabase::ReadFileNames ()
 {
   fileNames.clear ();
+#if defined(HAVE_UNORDERED_MAP)
+  // doesn't seem valid for multimap class
   fileNames.rehash (pHeader->numFiles);
+#endif
   AutoTraceTime att ("fndb read files", rootDirectory.Get());
   ReadFileNames (GetTopDirectory());
 }
@@ -1457,8 +1460,10 @@
 private:
   string pattern;
 
+#if defined(MIKTEX_WINDOWS)
 private:
   HANDLE handle;
+#endif
 
 private:
   friend class DirectoryLister;
diff -Nurd -x'*~' 2.9.orig/Libraries/MiKTeX/Core/fnamedb.h 2.9/Libraries/MiKTeX/Core/fnamedb.h
--- 2.9.orig/Libraries/MiKTeX/Core/fnamedb.h	2010-07-29 03:01:53.000000000 -0400
+++ 2.9/Libraries/MiKTeX/Core/fnamedb.h	2015-04-13 02:54:00.000000000 -0400
@@ -296,7 +296,6 @@
 #else
   typedef tr1::unordered_multimap<string, const FileNameDatabaseDirectory *> FileNameHashTable;
 #endif
-  FileNameHashTable fileNames;
 #else
 #if defined(MIKTEX_WINDOWS)
   typedef multimap<string, const FileNameDatabaseDirectory *, less_icase> FileNameHashTable;
@@ -304,6 +303,7 @@
   typedef multimap<string, const FileNameDatabaseDirectory *> FileNameHashTable;
 #endif
 #endif
+  FileNameHashTable fileNames;
 
 private:
   auto_ptr<TraceStream> traceStream;
diff -Nurd -x'*~' 2.9.orig/Libraries/MiKTeX/Core/internal.h 2.9/Libraries/MiKTeX/Core/internal.h
--- 2.9.orig/Libraries/MiKTeX/Core/internal.h	2011-03-24 09:09:21.000000000 -0400
+++ 2.9/Libraries/MiKTeX/Core/internal.h	2015-04-13 01:24:02.000000000 -0400
@@ -785,6 +785,7 @@
 
 public:
   bool operator < (/*[in]*/ const LanguageInfo_ & rhs)
+    const
   {
     if (this->key == "english")
     {
diff -Nurd -x'*~' 2.9.orig/Libraries/MiKTeX/KPathSeaEmulation/StdAfx.h 2.9/Libraries/MiKTeX/KPathSeaEmulation/StdAfx.h
--- 2.9.orig/Libraries/MiKTeX/KPathSeaEmulation/StdAfx.h	2010-10-02 04:22:21.000000000 -0400
+++ 2.9/Libraries/MiKTeX/KPathSeaEmulation/StdAfx.h	2015-04-12 23:39:17.000000000 -0400
@@ -23,7 +23,6 @@
 
 #include <cstdarg>
 #include <cstdlib>
-#include <malloc.h>
 #if defined(MIKTEX_UNIX)
 #include <sys/time.h>
 #endif
diff -Nurd -x'*~' 2.9.orig/Libraries/MiKTeX/KPathSeaEmulation/kpsemu.cpp 2.9/Libraries/MiKTeX/KPathSeaEmulation/kpsemu.cpp
--- 2.9.orig/Libraries/MiKTeX/KPathSeaEmulation/kpsemu.cpp	2010-11-22 05:10:29.000000000 -0500
+++ 2.9/Libraries/MiKTeX/KPathSeaEmulation/kpsemu.cpp	2015-04-12 23:42:21.000000000 -0400
@@ -671,6 +671,13 @@
       FATAL_CRT_ERROR ("fseeko64", lpszFileName);
     }
   return (0);
+#elif defined(__APPLE__)
+  // Mac OS X fseeko is already 64-bits
+  if (fseeko(pfile, offset, where) != 0)
+    {
+      FATAL_CRT_ERROR ("fseeko", lpszFileName);
+    }
+  return (0);
 #else
 #  error Unimplemented: miktex_xfseeko64()
 #endif
@@ -734,6 +741,14 @@
     }
   return (pos);
   
+#elif defined(__APPLE__)
+  // Mac OS X ftello is already 64-bits
+  MIKTEX_INT64 pos = ftello(pfile);
+  if (pos < 0)
+    {
+      FATAL_CRT_ERROR ("ftello", lpszFileName);
+    }
+  return (pos);
 #else
 #  error Unimplemented: miktex_xftello64
 #endif
diff -Nurd -x'*~' 2.9.orig/Programs/MiKTeX/makex/internal.h 2.9/Programs/MiKTeX/makex/internal.h
--- 2.9.orig/Programs/MiKTeX/makex/internal.h	2009-09-02 08:35:29.000000000 -0400
+++ 2.9/Programs/MiKTeX/makex/internal.h	2015-04-12 23:20:33.000000000 -0400
@@ -37,7 +37,6 @@
 #include <cstdlib>
 #include <string>
 
-#include <malloc.h>
 
 #include <getopt.h>
 #include <miktex/App/Application>
