diff -Nurd -x'*~' evince-2.24.2.orig/properties/Makefile.am evince-2.24.2/properties/Makefile.am
--- evince-2.24.2.orig/properties/Makefile.am	2008-11-24 14:09:31.000000000 -0500
+++ evince-2.24.2/properties/Makefile.am	2008-11-30 19:00:27.000000000 -0500
@@ -22,13 +22,13 @@
 nautilus_LTLIBRARIES = $(EVINCE_PROPERTIES_PAGE)
 
 nautilusdir=$(NAUTILUS_EXTENSION_DIR)
-libevince_properties_page_la_CFLAGS = -I$(top_srcdir)
+libevince_properties_page_la_CPPFLAGS = -I$(top_srcdir)
 libevince_properties_page_la_SOURCES = 	\
 	ev-properties-main.c
 
 libevince_properties_page_la_LIBADD = 		\
 	libevproperties.la			\
-	$(top_builddir)/libdocument/libevbackend.la	\
+	-levbackend				\
 	$(FRONTEND_LIBS)			\
 	$(NAUTILUS_LIBS)
 
