diff -Nurd Net-Daemon/ChangeLog Net-Daemon-new/ChangeLog
--- Net-Daemon/ChangeLog	2007-06-17 14:42:36.000000000 -0400
+++ Net-Daemon-new/ChangeLog	2010-08-16 02:13:05.000000000 -0400
@@ -1,4 +1,6 @@
-
+2009-03-12  Joe McMahon     <mcmahon@blekko.com> (0.44)
+	* Added necessary thread sharing to work with 5.10
+	  threads model: regexp-threads, 
 2007-06-17  Malcolm Nooning  <m.nooning@comcast.net> (0.43)
         * lib/Net/Daemon.pm  Needed to up the VERSION number
 2007-06-16  Malcolm Nooning  <m.nooning@comcast.net> (0.42)
diff -Nurd Net-Daemon/lib/Net/Daemon.pm Net-Daemon-new/lib/Net/Daemon.pm
--- Net-Daemon/lib/Net/Daemon.pm	2007-06-17 14:39:46.000000000 -0400
+++ Net-Daemon-new/lib/Net/Daemon.pm	2010-08-16 02:16:42.000000000 -0400
@@ -34,6 +34,18 @@
 package Net::Daemon;
 
 $Net::Daemon::VERSION = '0.43';
+# Dummy share() in case we're not 5.10. If we are, require/import of 
+# threads::shared will replace it appropriately.
+my $this_is_510 = $^V ge v5.10.0;
+if ($this_is_510) {
+ no warnings 'redefine';
+ eval { require threads; };
+ eval { require threads::shared; };
+}
+else {
+  eval { require Threads; };
+}
+
 @Net::Daemon::ISA = qw(Net::Daemon::Log);
 
 #
@@ -41,6 +53,7 @@
 #   regexp-threads.)
 #
 $Net::Daemon::RegExpLock = 1;
+threads::shared::share(\$Net::Daemon::RegExpLock) if $this_is_510;
 
 use vars qw($exit);
 
diff -Nurd Net-Daemon/regexp-threads Net-Daemon-new/regexp-threads
--- Net-Daemon/regexp-threads	2007-05-23 09:56:02.000000000 -0400
+++ Net-Daemon-new/regexp-threads	2010-08-16 02:13:09.000000000 -0400
@@ -13,9 +13,18 @@
 #
 #
 
+my $this_is_510 = $^V ge v5.10.0;
+
 use Thread ();
+if ($this_is_510) {
+  eval {require threads::shared};
+}
 
 my $numChilds;
+if ($this_is_510) {
+  eval { threads::shared::share($numChilds) };
+}
+
 my $regExpLock = @ARGV ? 1 : 0;
 
 # Repeat generating a random number and check if it contains the
