diff -Nurd -x'*~' scintilla.orig/gtk/makefile scintilla/gtk/makefile
--- scintilla.orig/gtk/makefile	2011-09-16 04:21:43.000000000 -0400
+++ scintilla/gtk/makefile	2011-09-19 10:47:43.000000000 -0400
@@ -66,9 +66,9 @@
 MARSHALLER=scintilla-marshal.o
 
 .cxx.o:
-	$(CC) $(CONFIGFLAGS) $(CXXFLAGS) -c $<
+	$(CC) $(CXXFLAGS) $(CONFIGFLAGS) -c $<
 .c.o:
-	$(CCOMP) $(CONFIGFLAGS) $(CFLAGS) -w -c $<
+	$(CCOMP) $(CFLAGS) $(CONFIGFLAGS) -w -c $<
 
 LEXOBJS:=$(addsuffix .o,$(basename $(notdir $(wildcard ../lexers/Lex*.cxx))))
 
@@ -78,7 +78,7 @@
 	$(DEL) *.o $(COMPLIB)
 
 deps:
-	$(CC) -MM $(CONFIGFLAGS) $(CXXFLAGS) *.cxx ../src/*.cxx | sed -e 's/\/usr.* //' | grep [a-zA-Z] >deps.mak
+	$(CC) -MM $(CXXFLAGS) $(CONFIGFLAGS) *.cxx ../src/*.cxx | sed -e 's/\/usr.* //' | grep [a-zA-Z] >deps.mak
 
 $(COMPLIB): Accessor.o CharacterSet.o LexerBase.o LexerModule.o LexerSimple.o StyleContext.o WordList.o \
 	CharClassify.o Decoration.o Document.o PerLine.o Catalogue.o CallTip.o \
diff -Nurd -x'*~' scite.orig/gtk/makefile scite/gtk/makefile
--- scite.orig/gtk/makefile	2011-09-16 04:21:45.000000000 -0400
+++ scite/gtk/makefile	2011-09-19 10:42:32.000000000 -0400
@@ -73,13 +73,13 @@
 INCLUDEDIRS=-I ../../scintilla/include -I ../src -I../lua/include
 
 .c.o:
-	$(CCOMP) $(CFLAGS) $(INCLUDEDIRS) -DLUA_USE_POSIX $(CXXTFLAGS) -c $< -o $@
+	$(CCOMP) $(INCLUDEDIRS) -DLUA_USE_POSIX $(CXXTFLAGS) $(CFLAGS) -c $< -o $@
 
 else
 CXXFLAGS+=-DNO_LUA
 endif
 
-CXXFLAGS:=$(CXXFLAGS) $(CXXTFLAGS) -Wno-long-long
+CXXFLAGS:= $(CXXTFLAGS)$(CXXFLAGS) -Wno-long-long
 
 UNAME:=$(shell uname -s)
 ifeq ($(UNAME), Linux)
@@ -90,13 +90,13 @@
 endif
 
 .cxx.o:
-	$(CC) $(CONFIGFLAGS) $(CXXFLAGS) -c $< -o $@
+	$(CC) $(CXXFLAGS) $(CONFIGFLAGS) -c $< -o $@
 
 clean:
 	rm -f *.o $(PROG)
 
 deps:
-	$(CC) -MM $(CONFIGFLAGS) $(CXXFLAGS) *.cxx ../src/*.cxx | sed -e 's/\/usr.* //' | grep [a-zA-Z] >deps.mak
+	$(CC) -MM $(CXXFLAGS) $(CONFIGFLAGS) *.cxx ../src/*.cxx | sed -e 's/\/usr.* //' | grep [a-zA-Z] >deps.mak
 
 # make should be run in ../../scintilla/gtk to compile all the lexers.
 COMPLIB=../../scintilla/bin/scintilla.a
