diff -Nurd -x'*~' slrnconf-0.8.2.orig/Makefile slrnconf-0.8.2/Makefile
--- slrnconf-0.8.2.orig/Makefile	2003-07-22 11:21:35.000000000 -0400
+++ slrnconf-0.8.2/Makefile	2003-07-22 22:40:46.000000000 -0400
@@ -1,11 +1,14 @@
 PREFIX=/usr/local
 BINDIR=$(PREFIX)/bin
 DATADIR=$(PREFIX)/share/slrnconf
+PERL=/usr/bin/perl
 
-all:
-	sed s,@DATADIR@,$(DATADIR),g slrnconf.in > slrnconf
+all: slrnconf
 
-install: all
+slrnconf: slrnconf.in
+	sed -e "s,@DATADIR@,$(DATADIR),g" -e "s,@PERL@,$(PERL),g" slrnconf.in > slrnconf
+
+install: slrnconf
 	mkdir -p $(PREFIX)
 	mkdir -p $(BINDIR)
 	mkdir -p $(DATADIR)/help
@@ -14,7 +17,7 @@
 	install -m 644 help/* $(DATADIR)/help
 	install -m 644 data/* $(DATADIR)/data
 
-	install -m 644 ABOUT COPYING README *.pm $(DATADIR)
+	install -m 644 *.pm $(DATADIR)
 
 	install -m 755 slrnconf $(BINDIR)
 
diff -Nurd -x'*~' slrnconf-0.8.2.orig/slrnconf.in slrnconf-0.8.2/slrnconf.in
--- slrnconf-0.8.2.orig/slrnconf.in	2003-07-22 11:21:35.000000000 -0400
+++ slrnconf-0.8.2/slrnconf.in	2003-07-22 12:52:43.000000000 -0400
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -w
+#!@PERL@ -w
 
 ###############################################################################
 #                                                                             #
