diff -ruN spatialite-gis-1.0-ALPHA-1.orig/Makefile-OsX spatialite-gis-1.0-ALPHA-1/Makefile-OsX
--- spatialite-gis-1.0-ALPHA-1.orig/Makefile-OsX	2009-07-01 18:12:24.000000000 +0900
+++ spatialite-gis-1.0-ALPHA-1/Makefile-OsX	2009-12-15 08:25:21.000000000 +0900
@@ -11,14 +11,14 @@
 INCLUDE = Classdef.h 
 
 # Define default flags:
-CXXFLAGS = $(shell wx-config --cxxflags) 
-LIB = $(shell wx-config --libs)
+CXXFLAGS = $(shell @PREFIX@/bin/wx-config --cxxflags) 
+LIB = $(shell @PREFIX@/bin/wx-config --libs)
 EXTRAFLAGS = -Wall -Wextra -Wno-ctor-dtor-privacy \
-	-fno-strict-aliasing -I/usr/local/include 
-EXTRALIBS = /usr/local/lib/libspatialite.a /usr/local/lib/libgeos_c.a \
-	/usr/local/lib/libgeos.a /usr/local/lib/libproj.a \
-	/usr/local/lib/librasterlite.a /usr/local/lib/libgeotiff.a
-EXTRALIB2 = -liconv
+	-fno-strict-aliasing -I@PREFIX@/include 
+EXTRALIBS = @PREFIX@/lib/libspatialite.dylib @PREFIX@/lib/libgeos3/libgeos_c.dylib \
+	@PREFIX@/lib/libgeos3/libgeos.dylib @PREFIX@/lib/libproj.dylib \
+	@PREFIX@/lib/librasterlite.dylib @PREFIX@/lib/libgeotiff.dylib
+EXTRALIB2 = -lintl -liconv -lz
 
 all: $(EXE)
 
