diff -Nurd -x '*~' xmysql-1.9.orig/XmysqlDB.c xmysql-1.9/XmysqlDB.c
--- xmysql-1.9.orig/XmysqlDB.c	2003-03-04 07:20:27.000000000 -0500
+++ xmysql-1.9/XmysqlDB.c	2003-03-04 07:21:46.000000000 -0500
@@ -26,7 +26,7 @@
   unsigned long rows_affected = 0;
   char *semicolon = 0;
   char *pos = 0, *r, *r_end;
-  ulong *lengths = 0;
+  unsigned long *lengths = 0;
   int total_lines = 0;
   int maxSize = 0;
   int maxLen = 0;
diff -Nurd -x '*~' xmysql-1.9.orig/Xmysql_cb.c xmysql-1.9/Xmysql_cb.c
--- xmysql-1.9.orig/Xmysql_cb.c	2003-03-04 07:20:27.000000000 -0500
+++ xmysql-1.9/Xmysql_cb.c	2003-03-04 07:22:07.000000000 -0500
@@ -1,5 +1,4 @@
 #include <string.h>
-#include <malloc.h>
 #include "forms.h"
 #include "Xmysql.h"
 #include "XmysqlDB.h"
