diff -Nurd -x '*~' xscreensaver-4.08.orig/configure xscreensaver-4.08/configure
--- xscreensaver-4.08.orig/configure	2003-02-24 01:12:13.000000000 -0500
+++ xscreensaver-4.08/configure	2003-02-24 01:36:05.000000000 -0500
@@ -5860,7 +5860,7 @@
     # Make sure to not put "make" in the Imakefile rules, since we grep it out.
     cat > Imakefile <<'EOF'
 acfindx:
-	@echo 'ac_x_app_defaults="${XAPPLOADDIR}"'
+	@echo 'ac_x_app_defaults="${XAPPLRESDIR}"'
 EOF
     if (xmkmf) >/dev/null 2>&1 && test -f Makefile; then
       # GNU make sometimes prints "make[1]: Entering...", which'd confuse us.
