--- coq/README.fink~	2010-05-28 13:58:15.000000000 +0200
+++ coq/README.fink	2010-05-28 14:40:12.000000000 +0200
@@ -0,0 +1,4 @@
+This package was originally finkified by Roland Zumkeller
+(roland.zumkeller@polytechnique.fr); it was modified and submitted to the
+fink package submission tracker by Jesse Alama (alama@stanford.edu). It was
+continued from there by Bruno De Fraine (bruno@defraine.net).
--- coq/Makefile.build~	2010-06-04 13:32:43.000000000 +0200
+++ coq/Makefile.build	2010-06-04 13:42:01.000000000 +0200
@@ -688,7 +688,7 @@
 
 install-opt::
 	$(MKDIR) $(FULLBINDIR)
-	$(INSTALLBIN) $(COQMKTOP) $(COQC) $(COQTOPBYTE) $(COQTOPOPT) $(CHICKEN) $(CHICKENOPT) $(FULLBINDIR)
+	$(INSTALLBIN) $(COQMKTOP) $(COQC) $(COQTOPOPT) $(CHICKEN) $(CHICKENOPT) $(FULLBINDIR)
 	cd $(FULLBINDIR); ln -sf coqtop.opt$(EXE) coqtop$(EXE); ln -sf coqchk.opt$(EXE) coqchk$(EXE)
 
 install-tools::
--- coq/Makefile.doc~	2009-07-20 14:30:02.000000000 +0200
+++ coq/Makefile.doc	2009-07-20 15:17:26.000000000 +0200
@@ -190,7 +190,7 @@
 endif
 	- rm -rf doc/stdlib/html
 	$(MKDIR) doc/stdlib/html
-	$(COQDOC) -q -d doc/stdlib/html --with-header doc/common/styles/html/$(HTMLSTYLE)/header.html --with-footer doc/common/styles/html/$(HTMLSTYLE)/footer.html --multi-index --html -g \
+	$(COQDOC) -q -d doc/stdlib/html --with-header doc/common/styles/html/$(HTMLSTYLE)/header.html --with-footer doc/common/styles/html/$(HTMLSTYLE)/footer.html --multi-index --html \
 	  -R theories Coq $(THEORIESVO:.vo=.v)
 	mv doc/stdlib/html/index.html doc/stdlib/html/genindex.html
 
--- coq/doc/common/styles/html/simple/header.html~	2009-07-20 15:17:44.000000000 +0200
+++ coq/doc/common/styles/html/simple/header.html	2009-07-20 15:18:51.000000000 +0200
@@ -11,3 +11,4 @@
 
 <body>
 
+<div id="main">
--- coq/doc/common/styles/html/simple/footer.html~	2009-07-20 15:17:51.000000000 +0200
+++ coq/doc/common/styles/html/simple/footer.html	2009-07-20 15:19:01.000000000 +0200
@@ -1,2 +1,3 @@
+</div>
 </body>
 </html>
