diff -Nurd -x'*~' mapm3-source.orig/Makefile mapm3-source/Makefile
--- mapm3-source.orig/Makefile	2003-04-19 07:53:22.000000000 -0400
+++ mapm3-source/Makefile	2003-04-29 01:23:22.000000000 -0400
@@ -24,7 +24,7 @@
 #### datasets (roughly 5000 loci vs 1000 loci). For even bigger datasets, 
 #### modify mapm/mapm.h.
 
-SYS= -D_SYS_SUNOS
+SYS= -D_BIG_DATASETS
 
 #### Next define the system libraries to link with Mapmaker. On SunOS, Ultrix
 #### and other vanilla-ish BSD systems, the correct setting is:
@@ -35,7 +35,7 @@
 #### is probably best (for ex: on A/UX or HP/UX, respectively). 
 #### On A/UX -lmalloc is likely a good idea too.
 
-SYS_LIBS= -lm -ltermcap
+SYS_LIBS= -ltermcap
 
 #### Below we define the directories for these programs. Set DIR to the place
 #### in which the executables and help files should permanently be installed
@@ -70,9 +70,9 @@
 #### (see the end of readline/readline.c). Readline will not compile on
 #### A/UX without using GCC instead of A/UX's cc (because it needs alloca).
 
-OUR_LIB= gnu.o
-GNU_OPT= -D_GNU_READLINE -I.
-GNU_LIBS= -L$(RDLN) -lreadline
+OUR_LIB= lib.o
+GNU_OPT= 
+GNU_LIBS= 
 
 #### Now we specify the commands used to compile MAPMAKER.  The stuff
 #### below works on Sun SPARCStations running SunOS 4.1.x (aka Solaris 1.x)
@@ -82,8 +82,8 @@
 #### person at your site should be able to figure out what to do almost
 #### trivially. See the file INSTALL.ME for details. 
 
-COMPILE= cc
-LINKALL= cc
+COMPILE= cc -fwritable-strings
+LINKALL= cc -Wl,-stack_size -Wl,4048000
 LINKLIB= ld -r
 DELETE=  rm -f
 INSTALL= cp
@@ -249,7 +249,7 @@
 	$(LINTALL) $(MAPM_LN)   $(LIB_LN) $(SYS_LIBS)   >Temp.lint
 	echo "=======================================" >>Temp.lint
 	fgrep -v -f Lint.grep Temp.lint $(MAPM_LINT) $(LIB_LINT) >mapm.lint
-		
+
 .c.ln:
 	$(LINT) -I$(LIB) $(SYS) -c $< -o $*.ln >$*.lint
 
diff -Nurd -x'*~' mapm3-source.orig/lib/system.h mapm3-source/lib/system.h
--- mapm3-source.orig/lib/system.h	2003-04-19 07:53:22.000000000 -0400
+++ mapm3-source/lib/system.h	2003-04-27 23:18:40.000000000 -0400
@@ -94,13 +94,23 @@
 #endif
 
 
+#define _SYS_UNIX
+#define SIGHANDLE void
+#define USE_DRAND48
+#define CALLOC_PTR_TO void
+#define CALLOC_NUM_TYPE size_t
+#define SIZEOF_TYPE size_t
+#define QSORT_LENGTH size_t
+#define QSORT_DATA_PTR_TO void
+#define INC_LIB
 
 /************************ File name syntax ****************************
 Setup the following for the OS's path types (these are used by the 
 make_filename() procedure in iolib.c). Note that the lengths of various 
 elements of a path are not checked, only its total length.
 *************************************************************************/
-#define PATH_LENGTH		200
+#include <sys/param.h>
+#define PATH_LENGTH		MAXPATHLEN
 
 #ifndef _SYS_DOS /* e.g. _SYS_UNIX or some POSIX like thing */
 #define HELP_EXT                ".help"
@@ -583,7 +593,6 @@
 #include <signal.h> 
 #include <errno.h> 	
 #include <time.h>       /* for ctime() def - Who does not have this file? */
-#include <malloc.h>
 #include <sys/types.h>
 
 #ifdef TRY_WINSIZE
@@ -598,8 +607,6 @@
 
 /* The HPUX (and System V?) random number functions... ditto */
 #ifdef USE_DRAND48
-double drand48(); 
-void srand48(); 
 #endif
 
 /* Library declarations only to be used by the helpers library code itself */
diff -Nurd -x'*~' mapm3-source.orig/mapm/lowlevel.h mapm3-source/mapm/lowlevel.h
--- mapm3-source.orig/mapm/lowlevel.h	1993-02-09 11:35:06.000000000 -0500
+++ mapm3-source/mapm/lowlevel.h	2006-06-12 16:00:55.000000000 -0400
@@ -44,7 +44,11 @@
     } data;
 } RAW_DATA;
 
+#ifdef ACTUAL_RAW
+static RAW_DATA raw;
+#else
 extern RAW_DATA raw;
+#endif
 
 /****************************************************************************/
 
diff -Nurd -x'*~' mapm3-source.orig/mapm/reader.c mapm3-source/mapm/reader.c
--- mapm3-source.orig/mapm/reader.c	1993-02-09 11:35:05.000000000 -0500
+++ mapm3-source/mapm/reader.c	2006-06-12 16:01:10.000000000 -0400
@@ -14,9 +14,9 @@
 #define INC_LIB
 #define INC_SHELL
 #define INC_EQN
+#define ACTUAL_RAW
 #include "mapm.h"
 
-RAW_DATA raw;
 int original_markers;
 
 #define end_of_text(fp) feof(fp)
diff -Nurd -x'*~' mapm3-source.orig/sun/xmapmaker mapm3-source/sun/xmapmaker
--- mapm3-source.orig/sun/xmapmaker	2003-04-19 07:53:23.000000000 -0400
+++ mapm3-source/sun/xmapmaker	2003-04-19 09:15:38.000000000 -0400
@@ -10,5 +10,5 @@
 else if (-e /usr/openwin/demo/xterm) then
 	exec /usr/openwin/demo/xterm $xoptions $fontsize -e $program $*
 else
-	exec xterm $options $fontsize -e $program
+	exec xterm $xoptions $fontsize -e $program
 endif
diff -Nurd -x'*~' mapm3-source.orig/sun/xqtl mapm3-source/sun/xqtl
--- mapm3-source.orig/sun/xqtl	2003-04-19 07:53:23.000000000 -0400
+++ mapm3-source/sun/xqtl	2003-04-19 09:15:44.000000000 -0400
@@ -10,5 +10,5 @@
 else if (-e /usr/openwin/demo/xterm) then
 	exec /usr/openwin/demo/xterm $xoptions $fontsize -e $program $*
 else
-	exec xterm $options $fontsize -e $program
+	exec xterm $xoptions $fontsize -e $program
 endif
