diff -Nurd qhull-2011.1/Makefile qhull-2011.1.patched/Makefile
--- qhull-2011.1/Makefile	2011-05-14 14:00:48.000000000 -0400
+++ qhull-2011.1.patched/Makefile	2011-08-25 12:07:26.000000000 -0400
@@ -64,12 +64,12 @@
 # You may build the qhull programs without using a library
 # make qhullx
 
-DESTDIR = /usr/local
+DESTDIR = @INSTPREFIX@
 BINDIR	= $(DESTDIR)/bin
-DOCDIR	= $(DESTDIR)/share/doc/packages/qhull
+DOCDIR	= $(DESTDIR)/share/doc/qhull
 INCDIR	= $(DESTDIR)/include
 LIBDIR	= $(DESTDIR)/lib
-MANDIR	= $(DESTDIR)/man/man1
+MANDIR	= $(DESTDIR)/share/man/man1
 
 # if you do not have enscript, try a2ps or just use lpr.  The files are text.
 PRINTMAN = enscript -2rl
@@ -80,16 +80,16 @@
 #for Gnu's gcc compiler, -O2 for optimization, -g for debugging
 #   -fno-strict-aliasing not needed for gcc 4.4+ (segfault in qset.c)
 CC        = gcc
-CC_OPTS1  = -O2 -fPIC -ansi -fno-strict-aliasing -Isrc/libqhull $(CC_WARNINGS)
+CC_OPTS1  = -O2 -fPIC -ansi -fno-common -fno-strict-aliasing -Isrc/libqhull $(CC_WARNINGS)
 CXX       = g++
 # libqhullcpp must be before libqhull
 CXX_OPTS1 = -O2 -Dqh_QHpointer -Isrc/ -Isrc/libqhullcpp -Isrc/libqhull $(CXX_WARNINGS)
 
 # for shared library link
-CC_OPTS3  =
+CC_OPTS3  = -dynamiclib -install_name @FINKPREFIX@/lib/libqhull6.2.0.dylib
 # Define qhull_VERSION in CMakeLists.txt, Makefile, qhull-exports.def, and qhull-warn.pri
 qhull_VERSION_MAJOR=6
-SO  = so.6.2.0
+SO  = 2.0.dylib
 
 # On MinGW, 
 #   make SO=dll
@@ -432,7 +432,7 @@
 
 lib/libqhull$(qhull_VERSION_MAJOR).$(SO): $(LIBQHULLSP_OBJS)
 	$(CC) -shared -o $@ $(CC_OPTS3) $^
-	cd lib && ln -f -s libqhull$(qhull_VERSION_MAJOR).$(SO) libqhull$(qhull_VERSION_MAJOR).so
+	cd lib && ln -f -s libqhull$(qhull_VERSION_MAJOR).$(SO) libqhull$(qhull_VERSION_MAJOR).dylib
 
 # don't use ../qconvex.	 Does not work on Red Hat Linux
 bin/qconvex: src/qconvex/qconvex.o lib/libqhullstatic.a
@@ -457,7 +457,7 @@
 bin/user_eg: src/user_eg/user_eg.c lib/libqhull$(qhull_VERSION_MAJOR).$(SO)
 	echo -e '\n== If user_eg fails to link, switch to -lqhullstatic_p.\n== On MinGW/Cygwin, use "make SO=dll"'
 	echo -e "\\n==  and copy lib/libqhull$(qhull_VERSION_MAJOR).$(SO) to bin\\n\\n"
-	$(CC) -o $@ $< -Dqh_QHpointer  $(CC_OPTS1) $(CC_OPTS3) -Llib -lqhull$(qhull_VERSION_MAJOR) -lm
+	$(CC) -o $@ $< -Dqh_QHpointer  $(CC_OPTS1) -Llib -lqhull$(qhull_VERSION_MAJOR) -lm
 
 bin/user_eg2: src/user_eg2/user_eg2.o lib/libqhullstatic.a 
 	$(CC) -o $@ $< $(CC_OPTS2) -Llib -lqhullstatic -lm
