diff -Nurd rpy-1.0.3.orig/rpy_tools.py rpy-1.0.3/rpy_tools.py
--- rpy-1.0.3.orig/rpy_tools.py	2008-01-25 15:13:23.000000000 -0500
+++ rpy-1.0.3/rpy_tools.py	2010-09-25 02:00:52.000000000 -0400
@@ -98,7 +98,7 @@
       raise RuntimeError("Couldn't execute the R interpreter" +
                          " `%s'.\n" % rexec )
     # edd 05 Apr 2006  version = re.search("R +([0-9]\.[0-9]\.[0-9])", output)
-    version = re.search(" +([0-9]\.[0-9]\.[0-9])", output)
+    version = re.search(" +([0-9]+\.[0-9]+\.[0-9]+)", output)
     if not version:
       raise RuntimeError("Couldn't obtain version number from output\n"
                                "of `R --version'.\n")
diff -Nurd rpy-1.0.3.orig/setup.py rpy-1.0.3/setup.py
--- rpy-1.0.3.orig/setup.py	2009-03-15 10:31:06.000000000 -0300
+++ rpy-1.0.3/setup.py	2009-03-15 10:33:02.000000000 -0300
@@ -73,7 +73,7 @@
 # for that version of R to execute.  So, we need to store the original
 # symlink here so we can restore it when we are done
 if sys.platform=="darwin":
-  darwin_original_version = os.readlink("/Library/Frameworks/R.framework/Versions/Current")
+  darwin_original_version = os.readlink("@PREFIX@/Library/Frameworks/R.framework/Versions/Current")
 
 modules = []
 for RHOME in RHOMES:
diff -Nurd rpy-1.0.3.orig/src/RPy.h rpy-1.0.3/src/RPy.h
--- rpy-1.0.3.orig/src/RPy.h	2007-11-29 19:40:14.000000000 -0500
+++ rpy-1.0.3/src/RPy.h	2010-09-25 02:18:46.000000000 -0400
@@ -74,7 +74,6 @@
 #  endif
 #endif  /* _WIN32 */
 
-#include <Rdevices.h> /* must follow Graphics.h */
 
 
 /* Missing definitions from Rinterface.h or RStartup.h */
