--- visual-3.2.1/include/vector.h~	2005-02-08 04:32:28.000000000 +0100
+++ visual-3.2.1/include/vector.h	2005-07-14 17:28:09.000000000 +0200
@@ -348,6 +348,7 @@
 class shared_vector : public vector 
 {
  private:
+	 bool dummy___;
 	typedef Cache::write_lock write_lock;
 	mutex* owner; //< The owner of this mutex must be a Cache
 	// object.  Since return_internal_reference<>() binds the lifetime of this
--- visual-3.2.1/bin/vpython.in~	2005-02-25 21:39:38.000000000 +0100
+++ visual-3.2.1/bin/vpython.in	2005-07-14 18:24:19.000000000 +0200
@@ -1,8 +1,14 @@
 #! /bin/sh
 # Variables determined by configure.
 PYTHON=@PYTHON@
+
+prefix=@prefix@
+[ -d ~/.idlerc ] || mkdir ~/.idlerc
+[ -r ~/.idlerc/config-main.cfg ] || cp @prefix@/share/doc/@N@/config-main.cfg ~/.idlerc/
+export BROWSER=${BROWSER:-open}
+
 visualexampledir=@visualexampledir@
-IDLE=idle
+IDLE=@prefix@/bin/idle@XY@
 PYTHONPATH=@PYTHONPATH@
 export PYTHONPATH
 
--- visual-3.2.9/src/Makefile.in~	2006-02-14 03:20:22.000000000 +0100
+++ visual-3.2.9/src/Makefile.in	2006-07-21 23:50:35.000000000 +0200
@@ -195,7 +195,7 @@
   INSTALL_RULE = $(GENERIC_INSTALLRULE)
 else 
   # These options are common to both Linux and OSX.
-  CVISUAL_LIBS += $(GTK_LIBS) $(GTKGL_LIBS) $(GTHREAD_LIBS) -lboost_python
+  CVISUAL_LIBS += $(GTK_LIBS) $(GTKGL_LIBS) $(GTHREAD_LIBS) @prefix@/lib/libboost_python.a 
   CVISUAL_CPPFLAGS += $(GTK_CFLAGS) $(GTKGL_CFLAGS) $(GTHREAD_CFLAGS)
   CVISUAL_CXXFLAGS += -fpic -DPIC
 
