diff -Nur gforth-0.5.0/arch/power/_sync_cache_range.c gforth-0.5.0-patched/arch/power/_sync_cache_range.c
--- gforth-0.5.0/arch/power/_sync_cache_range.c	Sat Sep 23 20:22:05 2000
+++ gforth-0.5.0-patched/arch/power/_sync_cache_range.c	Wed Feb  6 01:12:15 2002
@@ -34,7 +34,7 @@
      a system call, because the architecture is pretty shoddy in this
      area */
   for (; p < (addr+size); p+=cache_block_size)
-    asm("dcbst 0,%0; sync; icbi 0,%0"::"r"(p));
-  asm("sync; isync"); /* PPC 604e needs the additional sync
+    asm("dcbst 0,%0 \n sync \n icbi 0,%0"::"r"(p));
+  asm("sync \n isync"); /* PPC 604e needs the additional sync
 			 according to Tim Olson */
 }

