diff -ru gwydion-dylan-2.3.11-orig/Defaults.in gwydion-dylan-2.3.11/Defaults.in
--- gwydion-dylan-2.3.11-orig/Defaults.in	Sat Oct  4 13:21:01 2003
+++ gwydion-dylan-2.3.11/Defaults.in	Sat Oct  4 13:23:16 2003
@@ -96,10 +96,10 @@
 };
 
 if ($enable_bootstrap && $bootstrap_compiler eq "d2c") {
-    $CPPFLAGS = '-I@LOCAL_RUNTIME@';
+    $CPPFLAGS = '-I@LOCAL_RUNTIME@ @CPPFLAGS@';
     $d2c_runtime = '-L@LOCAL_RUNTIME@ ';
 } else {
-    $CPPFLAGS = '-I'.$srcroot.'/d2c/runtime';
+    $CPPFLAGS = '-I'.$srcroot.'/d2c/runtime @CPPFLAGS@';
     @runtime_directories = ('c-code', 'dylan', 'melange');
     foreach (@runtime_directories) {
 	$d2c_runtime .= "-L$buildroot/d2c/runtime/$_ ";
diff -ru gwydion-dylan-2.3.11-orig/d2c/compiler/main/main.dylan gwydion-dylan-2.3.11/d2c/compiler/main/main.dylan
--- gwydion-dylan-2.3.11-orig/d2c/compiler/main/main.dylan	Sat Oct  4 13:20:51 2003
+++ gwydion-dylan-2.3.11/d2c/compiler/main/main.dylan	Sat Oct  4 13:21:30 2003
@@ -215,7 +215,7 @@
 
 // Location of runtime.h
 define constant $runtime-include-dir
-  = concatenate($dylan-dir, "/include");
+  = concatenate($dylan-dir, "/include/dylan");
 
 #endif
 
diff -ru gwydion-dylan-2.3.11-orig/ltmain.sh gwydion-dylan-2.3.11/ltmain.sh
--- gwydion-dylan-2.3.11-orig/ltmain.sh	Sat Oct  4 13:21:01 2003
+++ gwydion-dylan-2.3.11/ltmain.sh	Sat Oct  4 13:21:30 2003
@@ -1229,7 +1229,7 @@
 	fi
 	;;
 
-      *.$libext)
+      *.$libext | /System/Library/*Frameworks/*)
 	# An archive.
 	deplibs="$deplibs $arg"
 	old_deplibs="$old_deplibs $arg"
@@ -1456,7 +1456,7 @@
 	  continue
 	  ;;
 	*.la) lib="$deplib" ;;
-	*.$libext)
+	*.$libext | /System/Library/*Frameworks/*)
 	  if test $pass = conv; then
 	    deplibs="$deplib $deplibs"
 	    continue
diff -ru gwydion-dylan-2.3.11-orig/mindy/comp/Makefile.in gwydion-dylan-2.3.11/mindy/comp/Makefile.in
--- gwydion-dylan-2.3.11-orig/mindy/comp/Makefile.in	Sat Oct  4 13:20:58 2003
+++ gwydion-dylan-2.3.11/mindy/comp/Makefile.in	Sat Oct  4 13:21:30 2003
@@ -49,8 +49,9 @@
 
 CC	=		@CC@
 CCOPTS	=		@CCOPTS@
-CFLAGS	=		$(CCOPTS) -I${SRCDIR} -I${CONFIGDIR} @DEFS@ -DBINDIR=\"${BINDIR}\" -DLIBDIR=\"${LIBDIR}\"
+CFLAGS	=		@CPPFLAGS@ $(CCOPTS) -I${SRCDIR} -I${CONFIGDIR} @DEFS@ -DBINDIR=\"${BINDIR}\" -DLIBDIR=\"${LIBDIR}\"
 CFLAGS_NO_WERROR =      `echo $(CFLAGS) | sed 's/-Werror//g'`
+LDFLAGS =		@LDFLAGS@
 YACC	=		@YACC@
 YFLAGS	=		-d
 LEX	=		@LEX@
@@ -76,7 +77,7 @@
 	dump.c free.c dup.c feature.c
 
 mindycomp$(exe_ext): ${OBJS}
-	${CC} ${OBJS} -DVERSION=\""`/bin/date`"\" ${SRCDIR}/version.c ${LIBS} -o ,$@
+	${CC} ${OBJS} ${LDFLAGS} -DVERSION=\""`/bin/date`"\" ${SRCDIR}/version.c ${LIBS} -o ,$@
 	mv ,$@ $@
 
 ${SRCDIR}/parser-tab.c: ${SRCDIR}/parser.y
diff -ru gwydion-dylan-2.3.11-orig/mindy/compat/Makefile.in gwydion-dylan-2.3.11/mindy/compat/Makefile.in
--- gwydion-dylan-2.3.11-orig/mindy/compat/Makefile.in	Sat Oct  4 13:20:58 2003
+++ gwydion-dylan-2.3.11/mindy/compat/Makefile.in	Sat Oct  4 13:21:30 2003
@@ -49,7 +49,7 @@
 AR      =               ar
 CC	=		@CC@
 CCOPTS	=		@CCOPTS@
-CFLAGS	=		$(CCOPTS) -I${SRCDIR} -I${CONFIGDIR} @DEFS@ -DBINDIR=\"${BINDIR}\" -DLIBDIR=\"${LIBDIR}\" -DTARGET=\"@d2c_target@\"
+CFLAGS	=		@CPPFLAGS@ $(CCOPTS) -I${SRCDIR} -I${CONFIGDIR} @DEFS@ -DBINDIR=\"${BINDIR}\" -DLIBDIR=\"${LIBDIR}\" -DTARGET=\"@d2c_target@\"
 YACC	=		@YACC@
 YFLAGS	=		-d
 LEX	=		@LEX@
diff -ru gwydion-dylan-2.3.11-orig/mindy/etc/Makefile.in gwydion-dylan-2.3.11/mindy/etc/Makefile.in
--- gwydion-dylan-2.3.11-orig/mindy/etc/Makefile.in	Sat Oct  4 13:21:00 2003
+++ gwydion-dylan-2.3.11/mindy/etc/Makefile.in	Sat Oct  4 13:21:30 2003
@@ -50,7 +50,8 @@
 
 CC	=		@CC@
 CCOPTS	=		@CCOPTS@
-CFLAGS	=		$(CCOPTS) -I${SRCDIR} -I${CONFIGDIR} @DEFS@ -DBINDIR=\"${BINDIR}\" -DLIBDIR=\"${LIBDIR}\"
+CFLAGS	=		@CPPFLAGS@ $(CCOPTS) -I${SRCDIR} -I${CONFIGDIR} @DEFS@ -DBINDIR=\"${BINDIR}\" -DLIBDIR=\"${LIBDIR}\"
+LDFLAGS	=		@LDFLAGS@
 YACC	=		@YACC@
 YFLAGS	=		-d
 LEX	=		@LEX@
@@ -74,7 +75,7 @@
 all:	mindyexec$(exe_ext)
 
 mindyexec$(exe_ext): mindyexec.o
-	${CC} mindyexec.o ${LIBS} -o ,$@
+	${CC} mindyexec.o ${LDFLAGS} ${LIBS} -o ,$@
 	mv ,$@ $@
 
 mindyexec.o: ${SRCDIR}/mindyexec.c
diff -ru gwydion-dylan-2.3.11-orig/mindy/interp/Makefile.in gwydion-dylan-2.3.11/mindy/interp/Makefile.in
--- gwydion-dylan-2.3.11-orig/mindy/interp/Makefile.in	Sat Oct  4 13:21:00 2003
+++ gwydion-dylan-2.3.11/mindy/interp/Makefile.in	Sat Oct  4 13:21:30 2003
@@ -52,8 +52,9 @@
 
 CC	=		@CC@
 CCOPTS	=		@CCOPTS@
-CFLAGS	=		$(CCOPTS) -I${SRCDIR} -I${CONFIGDIR} -I. @DEFS@ -DBINDIR=\"${BINDIR}\" -DLIBDIR=\"${LIBDIR}\" -DVERSION=\"@version@\" -DTARGET=\"@d2c_target@\"
+CFLAGS	=		@CPPFLAGS@ $(CCOPTS) -I${SRCDIR} -I${CONFIGDIR} -I. @DEFS@ -DBINDIR=\"${BINDIR}\" -DLIBDIR=\"${LIBDIR}\" -DVERSION=\"@version@\" -DTARGET=\"@d2c_target@\"
 CFLAGS_NO_WERROR =      `echo $(CFLAGS) | sed 's/-Werror//g'`
+LDFLAGS	=		@LDFLAGS@ 
 YACC	=		@YACC@
 YFLAGS	=		-d
 LEX	=		@LEX@
@@ -89,7 +90,7 @@
 EXTERN-INCLUDES = ${SRCDIR}/malloc.inc ${TIME}/time.inc
 
 mindy$(exe_ext): ${OBJS} ../compat/libcompat.a
-	${CC} ${OBJS} ${LIBS} -o ,$@
+	${CC} ${OBJS} ${LDFLAGS} ${LIBS} -o ,$@
 	mv -f ,$@ $@
 
 ${SRCDIR}/parser-tab.c: ${SRCDIR}/parser.y
diff -ru gwydion-dylan-2.3.11-orig/platform/Makegen gwydion-dylan-2.3.11/platform/Makegen
--- gwydion-dylan-2.3.11-orig/platform/Makegen	Sat Oct  4 13:20:53 2003
+++ gwydion-dylan-2.3.11/platform/Makegen	Sat Oct  4 13:21:30 2003
@@ -5,6 +5,9 @@
 if ($features{'compiled_for_darwin'}) {
     &compile_subdirs('carbon');
 }
-if ($have_gtk eq 'yes') {
-    &compile_subdirs('gtk+');
-}
+##
+## I've run into too many parser problems with this. Sorry
+##
+## if ($have_gtk eq 'yes') {
+##    &compile_subdirs('gtk+');
+## }
diff -ru gwydion-dylan-2.3.11-orig/platform/carbon/carbon/carbon-events.dylan gwydion-dylan-2.3.11/platform/carbon/carbon/carbon-events.dylan
--- gwydion-dylan-2.3.11-orig/platform/carbon/carbon/carbon-events.dylan	Sat Oct  4 13:20:52 2003
+++ gwydion-dylan-2.3.11/platform/carbon/carbon/carbon-events.dylan	Sun Oct  5 11:15:08 2003
@@ -435,8 +435,8 @@
 define constant $kEventControlInitialize :: <integer> = c-expr(int: "kEventControlInitialize");
 define constant $kEventControlDispose :: <integer> = c-expr(int: "kEventControlDispose");
 define constant $kEventControlGetOptimalBounds :: <integer> = c-expr(int: "kEventControlGetOptimalBounds");
-define constant $kEventControlDefInitialize :: <integer> = c-expr(int: "kEventControlDefInitialize");
-define constant $kEventControlDefDispose :: <integer> = c-expr(int: "kEventControlDefDispose");
+define constant $kEventControlDefInitialize :: <integer> = c-expr(int: "kEventControlInitialize");
+define constant $kEventControlDefDispose :: <integer> = c-expr(int: "kEventControlDispose");
 define constant $kEventControlHit :: <integer> = c-expr(int: "kEventControlHit");
 define constant $kEventControlSimulateHit :: <integer> = c-expr(int: "kEventControlSimulateHit");
 define constant $kEventControlHitTest :: <integer> = c-expr(int: "kEventControlHitTest");
Only in gwydion-dylan-2.3.11/platform/carbon/carbon: carbon-events.dylan~
diff -ru gwydion-dylan-2.3.11-orig/platform/carbon/carbon-event-handling/carbon-event-handling.lid gwydion-dylan-2.3.11/platform/carbon/carbon-event-handling/carbon-event-handling.lid
--- gwydion-dylan-2.3.11-orig/platform/carbon/carbon-event-handling/carbon-event-handling.lid	Sat Oct  4 13:20:52 2003
+++ gwydion-dylan-2.3.11/platform/carbon/carbon-event-handling/carbon-event-handling.lid	Sat Oct  4 13:21:30 2003
@@ -1,5 +1,5 @@
 library: carbon-event-handling
 executable: carbon-event-handling
-linker-options: -framework Carbon
+linker-options: /System/Library/Frameworks/Carbon.framework/Carbon
 files: carbon-event-handling-exports
   carbon-event-handling
diff -ru gwydion-dylan-2.3.11-orig/platform/carbon/sillyballs/sillyballs.lid gwydion-dylan-2.3.11/platform/carbon/sillyballs/sillyballs.lid
--- gwydion-dylan-2.3.11-orig/platform/carbon/sillyballs/sillyballs.lid	Sat Oct  4 13:20:52 2003
+++ gwydion-dylan-2.3.11/platform/carbon/sillyballs/sillyballs.lid	Sat Oct  4 13:21:30 2003
@@ -1,7 +1,7 @@
 library: sillyballs
 executable: sillyballs
 entry-point: sillyballs:%main
-linker-options: -framework Carbon
+linker-options: /System/Library/Frameworks/Carbon.framework/Carbon
 
 sillyballs-exports.dylan
 sillyballs.dylan
diff -ru gwydion-dylan-2.3.11-orig/platform/carbon/simple-starter/my-simple.lid gwydion-dylan-2.3.11/platform/carbon/simple-starter/my-simple.lid
--- gwydion-dylan-2.3.11-orig/platform/carbon/simple-starter/my-simple.lid	Sat Oct  4 13:20:52 2003
+++ gwydion-dylan-2.3.11/platform/carbon/simple-starter/my-simple.lid	Sat Oct  4 13:21:30 2003
@@ -1,7 +1,7 @@
 library: my-simple
 executable: my-simple
 entry-point: my-simple:%main
-linker-options: -framework Carbon
+linker-options: /System/Library/Frameworks/Carbon.framework/Carbon
 
 my-exports.dylan
 my-window.dylan
diff -ru gwydion-dylan-2.3.11-orig/tools/pidgin/src/c-parser-exports.dylan gwydion-dylan-2.3.11/tools/pidgin/src/c-parser-exports.dylan
--- gwydion-dylan-2.3.11-orig/tools/pidgin/src/c-parser-exports.dylan	Sat Oct  4 13:20:57 2003
+++ gwydion-dylan-2.3.11/tools/pidgin/src/c-parser-exports.dylan	Sat Oct  4 13:21:30 2003
@@ -116,5 +116,6 @@
     $i386-linux-platform,
     $ppc-linux-platform,
     $i386-freebsd-platform,
-    $sparc-solaris-platform;
+    $sparc-solaris-platform,
+    $ppc-darwin-platform;
 end module c-parser;
diff -ru gwydion-dylan-2.3.11-orig/tools/pidgin/src/platforms.dylan gwydion-dylan-2.3.11/tools/pidgin/src/platforms.dylan
--- gwydion-dylan-2.3.11-orig/tools/pidgin/src/platforms.dylan	Sat Oct  4 13:20:57 2003
+++ gwydion-dylan-2.3.11/tools/pidgin/src/platforms.dylan	Sat Oct  4 13:21:30 2003
@@ -133,6 +133,42 @@
        default-include-path:
 	 #["/usr/include"]);
 
+define constant $ppc-darwin-platform =
+  make(<c-platform>,
+       default-defines:
+	 #["const", "",
+	   "volatile", "",
+	   "__STDC__", "",
+	   
+	   // Parameterized macros which remove various GCC extensions from our
+	   // source code. The last item in the list is the right-hand side of
+	   // the define; all the items preceding it are named parameters.
+	   "__attribute__", #(#("x"), ""), 
+	   "__signed__", "", 
+	   "__signed", "", 
+	   "__inline__", "",
+	   "inline", "",
+	   "__inline", "",
+	   
+	   // These are the standard platform defines. Some day, configure should
+	   // figure them out by running gcc -E -o - -dM on an empty file.
+	   "__GNUC__", "2",
+	   "__GNUC_MINOR__", "95",
+	   "__ppc__", "",
+	   "__NATURAL_ALIGNMENT__", "",
+	   "__MACH__", "",
+	   "__BIG_ENDIAN__", "",
+	   "__APPLE__", "",
+      	   "__STDC__", "",
+	   "__APPLE_CC__", "934",
+	   "__DYNAMIC__", ""], 
+       default-include-path:
+	 #["/usr/local/include",
+	   // Find out the correct value for this path by calling
+	   // gcc -print-file-name=include
+	   "/usr/include/gcc/darwin/2.95.2/g++/..",
+           "/usr/include"]);
+
        
 define constant $sparc-solaris-platform =
   make(<c-platform>,
