diff -urP cml/LICENSE new/LICENSE
--- cml/LICENSE	Wed Dec 31 19:00:00 1969
+++ new/LICENSE	Mon Apr  1 11:02:13 2002
@@ -0,0 +1,23 @@
+STANDARD ML OF NEW JERSEY COPYRIGHT NOTICE, LICENSE AND DISCLAIMER.
+http://cm.bell-labs.com/cm/cs/what/smlnj/license.html
+
+Copyright (c) Lucent Technologies
+
+Permission to use, copy, modify, and distribute this software and
+its documentation for any purpose and without fee is hereby
+granted, provided that the above copyright notice appears in all
+copies and that both the copyright notice and this permission
+notice and warranty disclaimer appear in supporting documentation,
+and that the name of Lucent Technologies, Bell Labs or any Lucent
+entity not be used in advertising or publicity pertaining to
+distribution of the software without specific, written prior
+permission.
+
+Lucent disclaims all warranties with regard to this software,
+including all implied warranties of merchantability and
+fitness. In no event shall Lucent be liable for any special,
+indirect or consequential damages or any damages whatsoever
+resulting from loss of use, data or profits, whether in an action
+of contract, negligence or other tortious action, arising out of
+or in connection with the use or performance of this software.
+
diff -urP cml/make.sml new/make.sml
--- cml/make.sml	Wed Dec 31 19:00:00 1969
+++ new/make.sml	Mon Apr  1 11:01:48 2002
@@ -0,0 +1,6 @@
+#set (CM.Anchor.anchor "cml") (SOME "src");
+#set (CM.Anchor.anchor "cml-lib") (SOME "cml-lib/cm");
+app (CM.stabilize false) 
+    ["src/core-cml.cm", "src/cml-internal.cm", "src/cml.cm", "src/basis.cm"];
+app (CM.stabilize false) 
+    ["cml-lib/cm/trace-cml.cm", "cml-lib/cm/smlnj-lib.cm"];
