diff -ruN nokogiri-1.14.0-orig/ext/nokogiri/extconf.rb nokogiri-1.14.0/ext/nokogiri/extconf.rb
--- nokogiri-1.14.0-orig/ext/nokogiri/extconf.rb	2023-01-29 06:33:27.000000000 -0600
+++ nokogiri-1.14.0/ext/nokogiri/extconf.rb	2023-01-29 06:36:41.000000000 -0600
@@ -684,7 +684,7 @@
 # append_cflags(["-Wcast-qual", "-Wwrite-strings"])
 
 # Add SDK-specific include path for macOS and brew versions before v2.2.12 (2020-04-08) [#1851, #1801]
-macos_mojave_sdk_include_path = "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/libxml2"
+macos_mojave_sdk_include_path = "@FINK_PREFIX@/include/libxml2"
 if config_system_libraries? && darwin? && Dir.exist?(macos_mojave_sdk_include_path)
   append_cppflags("-I#{macos_mojave_sdk_include_path}")
 end
