diff -Nurd -x'*~' glib-2.20.2.orig/configure glib-2.20.2/configure
--- glib-2.20.2.orig/configure	2009-05-16 22:54:00.000000000 -0400
+++ glib-2.20.2/configure	2009-05-17 22:45:01.000000000 -0400
@@ -4708,7 +4708,7 @@
 #define HAVE_CARBON 1
 _ACEOF
 
-  LDFLAGS="$LDFLAGS -framework Carbon"
+  LDFLAGS="$LDFLAGS -Wl,-framework,Carbon"
 fi
 
 
@@ -8095,12 +8095,12 @@
 
         if test -n "$PYTHON"; then
       # If the user set $PYTHON, use it and don't search something else.
-      { $as_echo "$as_me:$LINENO: checking whether $PYTHON version >= 2.4" >&5
-$as_echo_n "checking whether $PYTHON version >= 2.4... " >&6; }
+      { $as_echo "$as_me:$LINENO: checking whether $PYTHON version >= 2.3" >&5
+$as_echo_n "checking whether $PYTHON version >= 2.3... " >&6; }
       prog="import sys, string
 # split strings by '.' and convert to numeric.  Append some zeros
 # because we need at least 4 digits for the hex conversion.
-minver = map(int, string.split('2.4', '.')) + [0, 0, 0]
+minver = map(int, string.split('2.3', '.')) + [0, 0, 0]
 minverhex = 0
 for i in xrange(0, 4): minverhex = (minverhex << 8) + minver[i]
 sys.exit(sys.hexversion < minverhex)"
@@ -37179,6 +37179,9 @@
 			G_THREAD_CFLAGS="-D_THREAD_SAFE"
 		fi
 		;;
+	*-darwin*)
+		# Nothing needed.
+		;;
 	*-dg-dgux*)  # DG/UX
 		G_THREAD_CFLAGS="-D_REENTRANT -D_POSIX4A_DRAFT10_SOURCE"
 		;;
diff -Nurd -x'*~' glib-2.20.2.orig/gio/gdesktopappinfo.c glib-2.20.2/gio/gdesktopappinfo.c
--- glib-2.20.2.orig/gio/gdesktopappinfo.c	2009-05-16 20:59:14.000000000 -0400
+++ glib-2.20.2/gio/gdesktopappinfo.c	2009-05-17 22:47:03.000000000 -0400
@@ -1532,7 +1532,7 @@
       g_file_set_contents (filename, contents, -1, NULL);
       g_free (contents);
 
-      run_update_command ("update-mime-database", "mime");
+      run_update_command ("@PREFIX@/bin/update-mime-database", "mime");
     }
   g_free (filename);
   
@@ -1663,7 +1663,7 @@
   info->filename = filename;
   info->desktop_id = desktop_id;
   
-  run_update_command ("update-desktop-database", "applications");
+  run_update_command ("@PREFIX@/bin/update-desktop-database", "applications");
   
   return TRUE;
 }
diff -Nurd -x'*~' glib-2.20.2.orig/gio/gvolumemonitor.h glib-2.20.2/gio/gvolumemonitor.h
--- glib-2.20.2.orig/gio/gvolumemonitor.h	2009-03-31 19:04:20.000000000 -0400
+++ glib-2.20.2/gio/gvolumemonitor.h	2009-05-18 01:57:05.000000000 -0400
@@ -139,7 +139,7 @@
 GMount *        g_volume_monitor_get_mount_for_uuid   (GVolumeMonitor *volume_monitor,
                                                        const char     *uuid);
 
-#ifndef G_DISABLE_DEPRECATED
+#if 1
 GVolume *       g_volume_monitor_adopt_orphan_mount   (GMount         *mount);
 #endif
 
diff -Nurd -x'*~' glib-2.20.2.orig/gio/xdgmime/xdgmime.c glib-2.20.2/gio/xdgmime/xdgmime.c
--- glib-2.20.2.orig/gio/xdgmime/xdgmime.c	2009-03-31 19:04:20.000000000 -0400
+++ glib-2.20.2/gio/xdgmime/xdgmime.c	2009-05-17 22:47:28.000000000 -0400
@@ -257,7 +257,7 @@
 
   xdg_data_dirs = getenv ("XDG_DATA_DIRS");
   if (xdg_data_dirs == NULL)
-    xdg_data_dirs = "/usr/local/share/:/usr/share/";
+    xdg_data_dirs = "@PREFIX@/share:/usr/local/share/:/usr/share/";
 
   ptr = xdg_data_dirs;
 
diff -Nurd -x'*~' glib-2.20.2.orig/glib/gmacros.h glib-2.20.2/glib/gmacros.h
--- glib-2.20.2.orig/glib/gmacros.h	2009-03-31 19:04:20.000000000 -0400
+++ glib-2.20.2/glib/gmacros.h	2009-05-17 23:11:48.000000000 -0400
@@ -120,7 +120,8 @@
 #define G_GNUC_WARN_UNUSED_RESULT
 #endif /* __GNUC__ */
 
-#ifndef G_DISABLE_DEPRECATED
+#if 1
+/* don't supress (if G_DISABLE_DEPRECATED) for fink yet */
 /* Wrap the gcc __PRETTY_FUNCTION__ and __FUNCTION__ variables with
  * macros, so we can refer to them as strings unconditionally.
  * usage not-recommended since gcc-3.0
diff -Nurd -x'*~' glib-2.20.2.orig/glib/gmessages.h glib-2.20.2/glib/gmessages.h
--- glib-2.20.2.orig/glib/gmessages.h	2009-03-31 19:04:20.000000000 -0400
+++ glib-2.20.2/glib/gmessages.h	2009-05-17 23:11:59.000000000 -0400
@@ -121,7 +121,8 @@
                                int             line,
                                const char     *func,
                                const char     *warnexpr);
-#ifndef G_DISABLE_DEPRECATED
+#if 1
+/* don't supress (if G_DISABLE_DEPRECATED) for fink yet */
 void g_assert_warning         (const char *log_domain,
 			       const char *file,
 			       const int   line,
diff -Nurd -x'*~' glib-2.20.2.orig/glib/gslice.c glib-2.20.2/glib/gslice.c
--- glib-2.20.2.orig/glib/gslice.c	2009-03-31 19:04:20.000000000 -0400
+++ glib-2.20.2/glib/gslice.c	2009-05-17 23:12:19.000000000 -0400
@@ -1131,19 +1131,18 @@
                     gsize memsize)
 {
   gpointer aligned_memory = NULL;
-  gint err = ENOMEM;
 #if     HAVE_COMPLIANT_POSIX_MEMALIGN
+  gint err;
   err = posix_memalign (&aligned_memory, alignment, memsize);
+  if (!aligned_memory)
+    errno = err;
 #elif   HAVE_MEMALIGN
-  errno = 0;
   aligned_memory = memalign (alignment, memsize);
-  err = errno;
 #elif   HAVE_VALLOC
-  errno = 0;
   aligned_memory = valloc (memsize);
-  err = errno;
 #else
   /* simplistic non-freeing page allocator */
+  gint err = ENOMEM;
   mem_assert (alignment == sys_page_size);
   mem_assert (memsize <= sys_page_size);
   if (!compat_valloc_trash)
@@ -1162,9 +1161,9 @@
         }
     }
   aligned_memory = g_trash_stack_pop (&compat_valloc_trash);
-#endif
   if (!aligned_memory)
     errno = err;
+#endif
   return aligned_memory;
 }
 
diff -Nurd -x'*~' glib-2.20.2.orig/glib/gutils.c glib-2.20.2/glib/gutils.c
--- glib-2.20.2.orig/glib/gutils.c	2009-05-16 20:56:51.000000000 -0400
+++ glib-2.20.2/glib/gutils.c	2009-05-18 01:37:03.000000000 -0400
@@ -2726,7 +2726,7 @@
       gchar *data_dirs = (gchar *) g_getenv ("XDG_DATA_DIRS");
 
       if (!data_dirs || !data_dirs[0])
-          data_dirs = "/usr/local/share/:/usr/share/";
+          data_dirs = "@PREFIX@/share:/usr/local/share/:/usr/share/";
 
       data_dir_vector = g_strsplit (data_dirs, G_SEARCHPATH_SEPARATOR_S, 0);
 #endif
@@ -2780,7 +2780,7 @@
       conf_dirs = (gchar *) g_getenv ("XDG_CONFIG_DIRS");
 
       if (!conf_dirs || !conf_dirs[0])
-          conf_dirs = "/etc/xdg";
+          conf_dirs = "@PREFIX@/etc/xdg";
 
       conf_dir_vector = g_strsplit (conf_dirs, G_SEARCHPATH_SEPARATOR_S, 0);
 #endif
diff -Nurd -x'*~' glib-2.20.2.orig/glib/gutils.h glib-2.20.2/glib/gutils.h
--- glib-2.20.2.orig/glib/gutils.h	2009-03-31 19:04:20.000000000 -0400
+++ glib-2.20.2/glib/gutils.h	2009-05-17 23:18:24.000000000 -0400
@@ -101,7 +101,11 @@
 #  define G_INLINE_FUNC
 #  undef  G_CAN_INLINE
 #elif defined (__GNUC__) 
+#if defined(__APPLE_CC__) && (__APPLE_CC__ >= 5400)
+#  define G_INLINE_FUNC static inline
+#else
 #  define G_INLINE_FUNC static __inline __attribute__ ((unused))
+#endif
 #elif defined (G_CAN_INLINE) 
 #  define G_INLINE_FUNC static inline
 #else /* can't inline */
diff -Nurd -x'*~' glib-2.20.2.orig/glib/libcharset/Makefile.in glib-2.20.2/glib/libcharset/Makefile.in
--- glib-2.20.2.orig/glib/libcharset/Makefile.in	2009-05-16 22:53:57.000000000 -0400
+++ glib-2.20.2/glib/libcharset/Makefile.in	2009-05-17 23:16:51.000000000 -0400
@@ -284,7 +284,7 @@
 	codeset.m4 update.sh make-patch.sh libcharset-glib.patch
 TEST_PROGS = 
 INCLUDES = \
-	-DLIBDIR=\"$(libdir)\" -I$(top_srcdir)
+	-DLIBDIR=\"$(sysconfdir)/glib-2.0\" -I$(top_srcdir)
 
 noinst_LTLIBRARIES = libcharset.la
 libcharset_la_SOURCES = \
@@ -292,8 +292,8 @@
 	localcharset.h		\
 	localcharset.c
 
-charset_alias = $(DESTDIR)$(libdir)/charset.alias
-charset_tmp = $(DESTDIR)$(libdir)/charset.tmp
+charset_alias = $(DESTDIR)$(sysconfdir)/glib-2.0/charset.alias
+charset_tmp = $(DESTDIR)$(sysconfdir)/glib-2.0/charset.tmp
 SUFFIXES = .sed .sin
 CLEANFILES = charset.alias ref-add.sed ref-del.sed
 all: all-am
@@ -606,7 +606,7 @@
 # run make test as part of make check
 check-local: test
 install-exec-local: all-local
-	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(mkinstalldirs) $(DESTDIR)$(sysconfdir)/glib-2.0
 	if test -f $(charset_alias); then \
 	  sed -f ref-add.sed $(charset_alias) > $(charset_tmp) ; \
 	  $(INSTALL_DATA) $(charset_tmp) $(charset_alias) ; \
diff -Nurd -x'*~' glib-2.20.2.orig/glib/libcharset/charset.alias glib-2.20.2/glib/libcharset/charset.alias
--- glib-2.20.2.orig/glib/libcharset/charset.alias	1969-12-31 19:00:00.000000000 -0500
+++ glib-2.20.2/glib/libcharset/charset.alias	2009-05-17 23:16:27.000000000 -0400
@@ -0,0 +1,111 @@
+# This file contains a table of character encoding aliases,
+# suitable for operating system 'darwin'.
+C			ASCII
+bg_BG			UTF-8
+bg_BG.CP1251		CP1251
+cs_CZ			UTF-8
+cs_CZ.ISO8859-2		ISO-8859-2
+da_DK			UTF-8
+da_DK.ISO8859-1		ISO-8859-1
+da_DK.ISO8859-15	ISO-8859-15
+de_AT			UTF-8
+de_AT.ISO8859-1		ISO-8859-1
+de_AT.ISO8859-15	ISO-8859-15
+de_CH			UTF-8
+de_CH.ISO8859-1		ISO-8859-1
+de_CH.ISO8859-15	ISO-8859-15
+de_DE			UTF-8
+de_DE.ISO8859-1		ISO-8859-1
+de_DE.ISO8859-15	ISO-8859-15
+en_AU			UTF-8
+en_AU.ISO8859-1		ISO-8859-1
+en_AU.ISO8859-15	ISO-8859-15
+en_AU.US-ASCII		ASCII
+en_CA			UTF-8
+en_CA.ISO8859-1		ISO-8859-1
+en_CA.ISO8859-15	ISO-8859-15
+en_CA.US-ASCII		ASCII
+en_GB			UTF-8
+en_GB.ISO8859-1		ISO-8859-1
+en_GB.ISO8859-15	ISO-8859-15
+en_GB.US-ASCII		ASCII
+en_US			UTF-8
+en_US.ISO8859-1		ISO-8859-1
+en_US.ISO8859-15	ISO-8859-15
+en_US.US-ASCII		ASCII
+es_ES			UTF-8
+es_ES.ISO8859-1		ISO-8859-1
+es_ES.ISO8859-15	ISO-8859-15
+fi_FI			UTF-8
+fi_FI.ISO8859-1		ISO-8859-1
+fi_FI.ISO8859-15	ISO-8859-15
+fr_BE			UTF-8
+fr_BE.ISO8859-1		ISO-8859-1
+fr_BE.ISO8859-15	ISO-8859-15
+fr_CA			UTF-8
+fr_CA.ISO8859-1		ISO-8859-1
+fr_CA.ISO8859-15	ISO-8859-15
+fr_CH			UTF-8
+fr_CH.ISO8859-1		ISO-8859-1
+fr_CH.ISO8859-15	ISO-8859-15
+fr_FR			UTF-8
+fr_FR.ISO8859-1		ISO-8859-1
+fr_FR.ISO8859-15	ISO-8859-15
+hr_HR			UTF-8
+hr_HR.ISO8859-2		ISO-8859-2
+hu_HU			UTF-8
+hu_HU.ISO8859-2		ISO-8859-2
+is_IS			UTF-8
+is_IS.ISO8859-1		ISO-8859-1
+is_IS.ISO8859-15	ISO-8859-15
+it_CH			UTF-8
+it_CH.ISO8859-1		ISO-8859-1
+it_CH.ISO8859-15	ISO-8859-15
+it_IT			UTF-8
+it_IT.ISO8859-1		ISO-8859-1
+it_IT.ISO8859-15	ISO-8859-15
+ja_JP			UTF-8
+ja_JP.EUC		EUC-JP
+ja_JP.SJIS		SHIFT_JIS
+ko_KR			UTF-8
+ko_KR.EUC		EUC-KR
+la_LN			UTF-8
+la_LN.ISO8859-1		ISO-8859-1
+la_LN.ISO8859-2		ISO-8859-2
+la_LN.ISO8859-4		ISO-8859-4
+la_LN.ISO8859-15	ISO-8859-15
+la_LN.US-ASCII		ASCII
+lt_LT			UTF-8
+lt_LT.ISO8859-4		ISO-8859-4
+nl_BE			UTF-8
+nl_BE.ISO8869-1		ISO-8859-1
+nl_BE.ISO8869-15	ISO-8859-15
+nl_NL			UTF-8
+nl_NL.ISO8869-1		ISO-8859-1
+nl_NL.ISO8869-15	ISO-8859-15
+no_NO			UTF-8
+no_NO.ISO8869-1		ISO-8859-1
+no_NO.ISO8869-15	ISO-8859-15
+pl_PL			UTF-8
+pl_PL.ISO8859-2		ISO-8859-2
+pt_PT			UTF-8
+pt_PT.ISO8859-1		ISO-8859-1
+pt_PT.ISO8859-15	ISO-8859-15
+ru_RU			UTF-8
+ru_RU.CP866		CP866
+ru_RU.ISO8859-5		ISO-8859-5
+ru_RU.KOI8-R		KOI8-R
+ru_RU.cp1251		CP1251
+ru_RU.koi8r		KOI8-R
+sl_SI			UTF-8
+sl_SI.ISO8859-2		ISO-8859-2
+sv_SE			UTF-8
+sv_SE.ISO8859-1		ISO-8859-1
+sv_SE.ISO8859-15	ISO-8859-15
+uk_UA			UTF-8
+uk_UA.KOI8-U		KOI8-U
+uk_UA.cp1251		CP1251
+zh_CN			UTF-8
+zh_CN.EUC		GB2312
+zh_TW			UTF-8
+zh_TW.Big5		BIG5
diff -Nurd -x'*~' glib-2.20.2.orig/glib-gettextize.in glib-2.20.2/glib-gettextize.in
--- glib-2.20.2.orig/glib-gettextize.in	2009-03-31 19:04:20.000000000 -0400
+++ glib-2.20.2/glib-gettextize.in	2009-05-17 23:15:52.000000000 -0400
@@ -49,8 +49,8 @@
     ;;
 esac
 
-datadir=@datadir@
 datarootdir=@datarootdir@
+datadir=@datadir@
 
 gettext_dir=$prefix/share/glib-2.0/gettext
 
diff -Nurd -x'*~' glib-2.20.2.orig/tests/Makefile.in glib-2.20.2/tests/Makefile.in
--- glib-2.20.2.orig/tests/Makefile.in	2009-05-16 22:53:58.000000000 -0400
+++ glib-2.20.2/tests/Makefile.in	2009-05-17 23:16:06.000000000 -0400
@@ -674,7 +674,7 @@
 	uri-test				\
 	regex-test
 
-test_scripts = run-markup-tests.sh run-collate-tests.sh run-bookmark-test.sh
+test_scripts = run-markup-tests.sh run-bookmark-test.sh
 test_script_support_programs = markup-test unicode-collate bookmarkfile-test
 TESTS_ENVIRONMENT = srcdir=$(srcdir) \
 	LIBCHARSET_ALIAS_DIR=$(top_builddir)/glib/libcharset \
diff -Nurd -x'*~' glib-2.20.2.orig/tests/child-test.c glib-2.20.2/tests/child-test.c
--- glib-2.20.2.orig/tests/child-test.c	2009-03-31 19:04:20.000000000 -0400
+++ glib-2.20.2/tests/child-test.c	2009-05-17 23:15:29.000000000 -0400
@@ -175,7 +175,7 @@
 #ifdef G_OS_WIN32
   system ("ipconfig /all");
 #else
-  system ("/bin/true");
+  system ("/usr/bin/true");
 #endif
 
   alive = 2;
