diff -Nurd -x'*~' libcanberra-0.12.orig/Makefile.am libcanberra-0.12/Makefile.am
--- libcanberra-0.12.orig/Makefile.am	2009-01-31 18:48:59.000000000 -0500
+++ libcanberra-0.12/Makefile.am	2009-06-01 22:07:44.000000000 -0400
@@ -19,7 +19,7 @@
 dist_doc_DATA = README
 
 EXTRA_DIST = bootstrap.sh autogen.sh LGPL libcanberra.schemas
-SUBDIRS = src gtkdoc doc
+SUBDIRS = src doc
 
 MAINTAINERCLEANFILES = README
 noinst_DATA = README
@@ -29,7 +29,6 @@
 
 if HAVE_GTK
 pkgconfig_DATA += libcanberra-gtk.pc
-endif
 
 schemasdir       = $(GCONF_SCHEMA_FILE_DIR)
 schemas_DATA     = libcanberra.schemas
@@ -44,6 +43,9 @@
 else
 install-data-local:
 endif
+else
+install-data-local:
+endif
 
 DISTCHECK_CONFIGURE_FLAGS=--enable-gtk-doc
 ACLOCAL_AMFLAGS = -I m4
diff -Nurd -x'*~' libcanberra-0.12.orig/configure.ac libcanberra-0.12/configure.ac
--- libcanberra-0.12.orig/configure.ac	2009-04-13 20:31:54.000000000 -0400
+++ libcanberra-0.12/configure.ac	2009-06-01 22:08:11.000000000 -0400
@@ -170,10 +170,6 @@
 
 AC_SYS_LARGEFILE
 
-#### [lib]iconv ####
-
-AM_ICONV
-
 ###################################
 #      External libraries         #
 ###################################
@@ -350,6 +346,7 @@
         ],
         [gtk=auto])
 
+AM_CONDITIONAL([GCONF_SCHEMAS_INSTALL], [false])
 if test "x${gtk}" != xno ; then
     PKG_CHECK_MODULES(GTK, [ gtk+-2.0 >= 2.13.4 gthread-2.0 ],
         [
@@ -357,6 +354,7 @@
             AC_DEFINE([HAVE_GTK], 1, [Have GTK?])
 
             AC_PATH_PROG(GCONFTOOL, gconftool-2)
+            AM_GCONF_SOURCE_2
 
             GTK_VERSION=`$PKG_CONFIG --variable=gtk_binary_version gtk+-2.0`
             GTK_MODULES_DIR="${libdir}/gtk-2.0/modules"
@@ -370,8 +368,7 @@
 else
     HAVE_GTK=0
 fi
-
-AM_GCONF_SOURCE_2
+PKG_CHECK_MODULES(LIBCANBERRA, [ libcanberra >= 0.12 ])
 
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
diff -Nurd -x'*~' libcanberra-0.12.orig/src/Makefile.am libcanberra-0.12/src/Makefile.am
--- libcanberra-0.12.orig/src/Makefile.am	2009-03-23 16:17:59.000000000 -0400
+++ libcanberra-0.12/src/Makefile.am	2009-06-01 22:13:22.000000000 -0400
@@ -31,14 +31,10 @@
 EXTRA_DIST = \
 	map-file
 
-lib_LTLIBRARIES = \
-        libcanberra.la
+lib_LTLIBRARIES =
 
-include_HEADERS = \
-	canberra.h
+include_HEADERS =
 
-noinst_PROGRAMS = \
-	test-canberra
 
 libcanberra_la_SOURCES = \
 	canberra.h \
@@ -287,20 +283,22 @@
 	canberra-gtk.h \
 	canberra-gtk.c
 libcanberra_gtk_la_CFLAGS = \
-	$(GTK_CFLAGS)
+	$(GTK_CFLAGS) \
+	$(LIBCANBERRA_CFLAGS)
 libcanberra_gtk_la_LIBADD = \
 	$(GTK_LIBS) \
-	libcanberra.la
+	$(LIBCANBERRA_LIBS)
 libcanberra_gtk_la_LDFLAGS = \
 	-export-dynamic -version-info $(LIBCANBERRA_GTK_VERSION_INFO)
 
 libcanberra_gtk_module_la_SOURCES = \
 	canberra-gtk-module.c
 libcanberra_gtk_module_la_CFLAGS = \
-	$(GTK_CFLAGS)
+	$(GTK_CFLAGS) \
+	$(LIBCANBERRA_CFLAGS)
 libcanberra_gtk_module_la_LIBADD = \
 	$(GTK_LIBS) \
-	libcanberra.la \
+	$(LIBCANBERRA_LIBS) \
 	libcanberra-gtk.la
 libcanberra_gtk_module_la_LDFLAGS = \
 	-avoid-version -module -export-dynamic
@@ -309,10 +307,11 @@
 	canberra-gtk-play.c
 canberra_gtk_play_LDADD = \
 	$(GTK_LIBS) \
-	libcanberra.la \
+	$(LIBCANBERRA_LIBS) \
 	libcanberra-gtk.la
 canberra_gtk_play_CFLAGS = \
-	$(GTK_CFLAGS)
+	$(GTK_CFLAGS) \
+	$(LIBCANBERRA_LIBS)
 
 EXTRA_DIST += \
 	libcanberra-login-sound.desktop.in \
