diff -Nurd -x'*~' libgnomeui-2.24.0.orig/doc/reference/Makefile.in libgnomeui-2.24.0/doc/reference/Makefile.in
--- libgnomeui-2.24.0.orig/doc/reference/Makefile.in	2008-09-22 08:26:11.000000000 -0400
+++ libgnomeui-2.24.0/doc/reference/Makefile.in	2008-10-16 12:58:36.000000000 -0400
@@ -268,8 +268,8 @@
 
 # CFLAGS and LDFLAGS for compiling scan program. Only needed if your app/lib
 # contains GtkObjects/GObjects and you want to document signals and properties.
-INCLUDES = -I$(top_builddir) -I$(top_srcdir) $(LIBGNOMEUI_CFLAGS)
-GTKDOC_LIBS = $(top_builddir)/libgnomeui/libgnomeui-2.la
+INCLUDES = -I$(top_builddir) -I$(top_srcdir) $(LIBGNOMEUI_CFLAGS) $(CFLAGS) $(CPPFLAGS)
+GTKDOC_LIBS = $(top_builddir)/libgnomeui/libgnomeui-2.la $(LIBGNOMEUI_LIBS) $(SM_LIBS) $(LDFLAGS)
 
 # Extra options to supply to gtkdoc-mkdb.
 MKDB_OPTIONS = --output-format=xml
