diff -Nurd -x'*~' libgail-gnome-1.20.1.orig/Makefile.in libgail-gnome-1.20.1/Makefile.in
--- libgail-gnome-1.20.1.orig/Makefile.in	2008-09-23 07:49:04.000000000 -0400
+++ libgail-gnome-1.20.1/Makefile.in	2009-02-19 01:04:05.000000000 -0500
@@ -215,7 +215,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = gail-gnome test
+SUBDIRS = gail-gnome
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libgail-gnome.pc
 EXTRA_DIST = libgail-gnome.pc.in
