diff -Nurd -x'*~' pango-1.24.2.orig/Makefile.in pango-1.24.2/Makefile.in
--- pango-1.24.2.orig/Makefile.in	2009-05-04 16:05:21.000000000 -0400
+++ pango-1.24.2/Makefile.in	2009-06-05 01:01:54.000000000 -0400
@@ -272,7 +272,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = pango modules pango-view examples docs tools tests
+SUBDIRS = pango modules pango-view          docs tools tests
 EXTRA_DIST = \
 	autogen.sh		\
 	pango.pc.in		\
diff -Nurd -x'*~' pango-1.24.2.orig/docs/Makefile.in pango-1.24.2/docs/Makefile.in
--- pango-1.24.2.orig/docs/Makefile.in	2009-05-04 16:05:17.000000000 -0400
+++ pango-1.24.2/docs/Makefile.in	2009-06-05 01:07:04.000000000 -0400
@@ -285,9 +285,9 @@
 	-DPANGO_ENABLE_ENGINE	\
 	-I$(top_srcdir)		\
 	$(GLIB_CFLAGS)		\
-	$(XFT_CFLAGS)		\
-	$(CAIRO_CFLAGS)		\
 	$(FREETYPE_CFLAGS)	\
+	$(CAIRO_CFLAGS)		\
+	$(XFT_CFLAGS)		\
 	$(X_CFLAGS)
 
 
diff -Nurd -x'*~' pango-1.24.2.orig/docs/html/pango-Miscellaneous-Utilities.html pango-1.24.2/docs/html/pango-Miscellaneous-Utilities.html
--- pango-1.24.2.orig/docs/html/pango-Miscellaneous-Utilities.html	2009-05-04 16:14:38.000000000 -0400
+++ pango-1.24.2/docs/html/pango-Miscellaneous-Utilities.html	2009-06-05 01:02:35.000000000 -0400
@@ -408,7 +408,7 @@
 <pre class="programlisting">char *              pango_config_key_get                (const char *key);</pre>
 <p>
 Looks up a key in the Pango config database
-(pseudo-win.ini style, read from $sysconfdir/pango/pangorc,
+(pseudo-win.ini style, read from @SYSCONFDIR@/pango/pangorc,
  ~/.pangorc, and getenv (PANGO_RC_FILE).)</p>
 <p>
 </p>
diff -Nurd -x'*~' pango-1.24.2.orig/docs/pango-querymodules.1 pango-1.24.2/docs/pango-querymodules.1
--- pango-1.24.2.orig/docs/pango-querymodules.1	2009-04-17 21:19:00.000000000 -0400
+++ pango-1.24.2/docs/pango-querymodules.1	2009-06-05 01:03:05.000000000 -0400
@@ -38,7 +38,7 @@
 The Pango module path is specified by the key
 Pango/ModulesPath
 in the Pango config database, which is read from
-\fIsysconfdir\fR/pango/pangorc,
+\fI@SYSCONFDIR@\fR/pango/pangorc,
 ~/\&.pangorc
 and the file specified in the environment variable
 \fBPANGO_RC_FILE\fR\&.
diff -Nurd -x'*~' pango-1.24.2.orig/pango/Makefile.in pango-1.24.2/pango/Makefile.in
--- pango-1.24.2.orig/pango/Makefile.in	2009-05-04 16:05:20.000000000 -0400
+++ pango-1.24.2/pango/Makefile.in	2009-06-05 01:03:38.000000000 -0400
@@ -455,9 +455,9 @@
 	-I$(top_srcdir)					\
 	$(PANGO_DEBUG_FLAGS)				\
 	$(GLIB_CFLAGS)					\
-	$(XFT_CFLAGS)					\
-	$(CAIRO_CFLAGS)					\
 	$(FREETYPE_CFLAGS)				\
+	$(CAIRO_CFLAGS)					\
+	$(XFT_CFLAGS)					\
 	$(X_CFLAGS)
 
 BUILT_SOURCES = pango-enum-types.h pango-enum-types.c pango-features.h
@@ -609,9 +609,9 @@
 	libpangoft2-$(PANGO_API_VERSION).la	\
 	libpango-$(PANGO_API_VERSION).la	\
 	$(GLIB_LIBS)				\
-	$(X_LIBS)				\
-	$(XFT_LIBS)				\
 	$(FREETYPE_LIBS)			\
+	$(XFT_LIBS)				\
+	$(X_LIBS)				\
 	$(libm)
 
 libpangoxft_1_0_la_DEPENDENCIES = \
diff -Nurd -x'*~' pango-1.24.2.orig/pango/pangofc-fontmap.c pango-1.24.2/pango/pangofc-fontmap.c
--- pango-1.24.2.orig/pango/pangofc-fontmap.c	2009-04-09 13:28:05.000000000 -0400
+++ pango-1.24.2/pango/pangofc-fontmap.c	2009-06-05 01:41:42.000000000 -0400
@@ -1359,10 +1359,16 @@
     return FC_WEIGHT_BOLD;
   else if (pango_weight <= (PANGO_WEIGHT_ULTRABOLD + PANGO_WEIGHT_HEAVY) / 2)
     return FC_WEIGHT_ULTRABOLD;
+#if 0
+  /* only have FC_WEIGHT_EXTRABLACK on fontconfig >= 2.5 */
   else if (pango_weight <= (PANGO_WEIGHT_HEAVY + PANGO_WEIGHT_ULTRAHEAVY) / 2)
     return FC_WEIGHT_BLACK;
   else
     return FC_WEIGHT_EXTRABLACK;
+#else
+  else
+    return FC_WEIGHT_BLACK;
+#endif
 }
 
 static int
@@ -2049,10 +2056,16 @@
     return PANGO_WEIGHT_BOLD;
   else if (fc_weight <= (FC_WEIGHT_EXTRABOLD + FC_WEIGHT_BLACK) / 2)
     return PANGO_WEIGHT_ULTRABOLD;
+#if 0
+  /* only have FC_WEIGHT_EXTRABLACK on fontconfig >= 2.5 */
   else if (fc_weight <= (FC_WEIGHT_BLACK + FC_WEIGHT_EXTRABLACK) / 2)
     return PANGO_WEIGHT_HEAVY;
   else
     return PANGO_WEIGHT_ULTRAHEAVY;
+#else
+  else
+    return PANGO_WEIGHT_HEAVY;
+#endif
 }
 
 static PangoStyle
diff -Nurd -x'*~' pango-1.24.2.orig/pangoxft.pc.in pango-1.24.2/pangoxft.pc.in
--- pango-1.24.2.orig/pangoxft.pc.in	2009-02-06 18:49:34.000000000 -0500
+++ pango-1.24.2/pangoxft.pc.in	2009-06-05 01:03:57.000000000 -0400
@@ -6,6 +6,6 @@
 Name: Pango Xft
 Description: Xft font support for Pango
 Version: @VERSION@
-Requires: pango pangoft2 xft
+Requires: pango pangoft2 xft pangox
 Libs: -L${libdir} -lpangoxft-@PANGO_API_VERSION@
 Cflags: -I${includedir}/pango-1.0
diff -Nurd -x'*~' pango-1.24.2.orig/tests/Makefile.in pango-1.24.2/tests/Makefile.in
--- pango-1.24.2.orig/tests/Makefile.in	2009-05-04 16:05:21.000000000 -0400
+++ pango-1.24.2/tests/Makefile.in	2009-06-05 01:04:57.000000000 -0400
@@ -303,7 +303,7 @@
 CLEANFILES = pangorc
 DISTCLEANFILES = all-unicode.txt runtests.log
 INCLUDES = -I$(top_srcdir) $(PANGO_DEBUG_FLAGS) $(GLIB_CFLAGS) \
-	$(XFT_CFLAGS) $(CAIRO_CFLAGS) $(FREETYPE_CFLAGS) $(X_CFLAGS) \
+	$(FREETYPE_CFLAGS) $(XFT_CFLAGS) $(CAIRO_CFLAGS) $(X_CFLAGS) \
 	$(am__append_1) $(am__append_2) $(am__append_3)
 TESTS_ENVIRONMENT = \
    srcdir=$(srcdir)                    \
diff -Nurd -x'*~' pango-1.24.2.orig/update-pango-modules pango-1.24.2/update-pango-modules
--- pango-1.24.2.orig/update-pango-modules	1969-12-31 19:00:00.000000000 -0500
+++ pango-1.24.2/update-pango-modules	2009-06-05 01:05:06.000000000 -0400
@@ -0,0 +1,20 @@
+#!/bin/sh -e
+#
+# update-pango-modules
+#
+#   Packages which install pango module(s) should call update-pango-modules
+#   in their PostInstScript and PostRmScript to update the pango.module file.
+#
+
+CONFDIR=@SYSCONFDIR@/pango
+CONFFILE=$CONFDIR/pango.modules
+
+mkdir -p $CONFDIR
+
+TMPFILE=`tempfile`
+pango-querymodules >$TMPFILE
+
+mv -f $TMPFILE $CONFFILE
+chmod 0644 $CONFFILE
+
+exit 0
