--- kvim-6.1.141/src/auto/configure	Tue Oct  1 10:54:21 2002
+++ kvim-6.1.141-new/src/auto/configure	Wed Oct  9 17:52:59 2002
@@ -1203,7 +1203,7 @@
 if test -z "$have_local_include" -a -d /usr/local/include; then
   tt=`echo "$CPPFLAGS" | sed -e 's+-I/usr/local/include ++g' -e 's+-I/usr/local/include$++g'`
   if test "$tt" = "$CPPFLAGS"; then
-    CPPFLAGS="$CPPFLAGS -I/usr/local/include"
+    CPPFLAGS="$CPPFLAGS"
   fi
 fi
 
@@ -1895,7 +1895,7 @@
 	  echo $ac_n "checking for Tcl library by myself""... $ac_c" 1>&6
 echo "configure:1897: checking for Tcl library by myself" >&5
 	  tcllib=`echo $tclinc | sed s/include/lib/g`
-	  for ext in .so .a ; do
+	  for ext in .dylib .so .a ; do
 	    for ver in "" $tclver ; do
 	      for try in $tcllib ; do
 		trylib=tcl$ver$ext
@@ -3674,9 +3674,15 @@
 	  fi
 	if test "x$no_kde" = x ; then
 	  echo "$ac_t""found KDE $kde_major_version.$kde_minor_version" 1>&6
-	  GUI_LIB_LOC="-L$KDE_LIBS $KDE_LIBS/libkdeui.so $KDE_LIBS/libkdecore.so $KDE_LIBS/libDCOP.so" 
+	  for ext in so dylib; do
+	    if test -f "$KDE_LIBS/libkdeui.${ext}" \
+	      && test -f "$KDE_LIBS/libkdecore.${ext}" \
+	      && test -f "$KDE_LIBS/libDCOP.${ext}"; then
+	      GUI_LIB_LOC="-L$KDE_LIBS $KDE_LIBS/libkdeui.${ext} $KDE_LIBS/libkdecore.${ext} $KDE_LIBS/libDCOP.${ext}" 
 								 GUI_INC_LOC="-I$KDE_INCLUDES"
 								 KDEDIR="$KDE_PREFIX"
+	    fi
+	  done
 	else
 	  echo "$ac_t""no" 1>&6
 	  KDE_LIBS=""
@@ -3705,13 +3711,19 @@
 					GUI_LIB_LOC="$GUI_LIB_LOC -pthread"
 					LIBS="$LIBS -pthread"
 				fi
+				for ext in so dylib; do
 								if test $qt_major_version = 2; then
-					GUI_LIB_LOC="$KDE_LIBS/libkfile.so -L$QT_LIBS -lqt $GUI_LIB_LOC -Wl,--rpath \
--Wl,$KDE_LIBS -Wl,--rpath -Wl,$QT_LIBS"
+						if test -f "$KDE_LIBS/libkfile.${ext}"; then
+							GUI_LIB_LOC="$KDE_LIBS/libkfile.${ext} \
+							-L$QT_LIBS -lqt $GUI_LIB_LOC -lstdc++"
+						fi
 				else
-					GUI_LIB_LOC="$KDE_LIBS/libkio.so -L$QT_LIBS -lqt-mt $GUI_LIB_LOC -Wl,--rpath \
--Wl,$KDE_LIBS -Wl,--rpath -Wl,$QT_LIBS"
+						if test -f "$KDE_LIBS/libkio.${ext}"; then
+							GUI_LIB_LOC="$KDE_LIBS/libkio.${ext} \
+							-L$QT_LIBS -lqt-mt $GUI_LIB_LOC -lstdc++"
 				fi
+				fi
+				done
 				echo "$ac_t""found $qt_major_version.$qt_minor_version in $ROOTQT" 1>&6
 				SKIP_GTK=YES
 				SKIP_OPIE=YES
@@ -4259,7 +4271,7 @@
     gui_libs="`echo $x_libraries|sed 's%/^/^/*$%%'` `echo "$gui_XXX" | sed s/XXX/lib/g` `echo "$GUI_INC_LOC" | sed s/include/lib/` $GUI_LIB_LOC"
     GUI_LIB_LOC=
     for try in $gui_libs; do
-      for libtry in "$try"/libXm.a "$try"/libXm.so* "$try"/libXm.sl; do
+      for libtry in "$try"/libXm.a "$try"/libXm.so* "$try"/libXm.dylib "$try"/libXm.sl; do
 	if test -f "$libtry"; then
 	  GUI_LIB_LOC=$try
 	fi
--- kvim-6.1.141/src/configure	Tue Oct  1 10:54:48 2002
+++ kvim-6.1.141-new/src/configure	Wed Oct  9 17:53:24 2002
@@ -1203,7 +1203,7 @@
 if test -z "$have_local_include" -a -d /usr/local/include; then
   tt=`echo "$CPPFLAGS" | sed -e 's+-I/usr/local/include ++g' -e 's+-I/usr/local/include$++g'`
   if test "$tt" = "$CPPFLAGS"; then
-    CPPFLAGS="$CPPFLAGS -I/usr/local/include"
+    CPPFLAGS="$CPPFLAGS"
   fi
 fi
 
@@ -1895,7 +1895,7 @@
 	  echo $ac_n "checking for Tcl library by myself""... $ac_c" 1>&6
 echo "configure:1897: checking for Tcl library by myself" >&5
 	  tcllib=`echo $tclinc | sed s/include/lib/g`
-	  for ext in .so .a ; do
+	  for ext in .dylib .so .a ; do
 	    for ver in "" $tclver ; do
 	      for try in $tcllib ; do
 		trylib=tcl$ver$ext
@@ -3674,9 +3674,15 @@
 	  fi
 	if test "x$no_kde" = x ; then
 	  echo "$ac_t""found KDE $kde_major_version.$kde_minor_version" 1>&6
-	  GUI_LIB_LOC="-L$KDE_LIBS $KDE_LIBS/libkdeui.so $KDE_LIBS/libkdecore.so $KDE_LIBS/libDCOP.so" 
+	  for ext in so dylib; do
+	    if test -f "$KDE_LIBS/libkdeui.${ext}" \
+	      && test -f "$KDE_LIBS/libkdecore.${ext}" \
+	      && test -f "$KDE_LIBS/libDCOP.${ext}"; then
+	      GUI_LIB_LOC="-L$KDE_LIBS $KDE_LIBS/libkdeui.${ext} $KDE_LIBS/libkdecore.${ext} $KDE_LIBS/libDCOP.${ext}" 
 								 GUI_INC_LOC="-I$KDE_INCLUDES"
 								 KDEDIR="$KDE_PREFIX"
+	    fi
+	  done
 	else
 	  echo "$ac_t""no" 1>&6
 	  KDE_LIBS=""
@@ -3707,13 +3713,19 @@
 				else
 					GUI_LIB_LOC="$GUI_LIB_LOC"
 				fi
+				for ext in so dylib; do
 								if test $qt_major_version = 2; then
-					GUI_LIB_LOC="$KDE_LIBS/libkfile.so -L$QT_LIBS -lqt $GUI_LIB_LOC -Wl,--rpath \
--Wl,$KDE_LIBS -Wl,--rpath -Wl,$QT_LIBS"
+						if test -f "$KDE_LIBS/libkfile.${ext}"; then
+							GUI_LIB_LOC="$KDE_LIBS/libkfile.${ext} \
+							-L$QT_LIBS -lqt $GUI_LIB_LOC -lstdc++"
+						fi
 				else
-					GUI_LIB_LOC="$KDE_LIBS/libkio.so -L$QT_LIBS -lqt-mt $GUI_LIB_LOC -Wl,--rpath \
--Wl,$KDE_LIBS -Wl,--rpath -Wl,$QT_LIBS"
+						if test -f "$KDE_LIBS/libkio.${ext}"; then
+							GUI_LIB_LOC="$KDE_LIBS/libkio.${ext} \
+							-L$QT_LIBS -lqt-mt $GUI_LIB_LOC -lstdc++"
 				fi
+				fi
+				done
 				echo "$ac_t""found $qt_major_version.$qt_minor_version in $ROOTQT" 1>&6
 				SKIP_GTK=YES
 				SKIP_OPIE=YES
@@ -4261,7 +4273,7 @@
     gui_libs="`echo $x_libraries|sed 's%/^/^/*$%%'` `echo "$gui_XXX" | sed s/XXX/lib/g` `echo "$GUI_INC_LOC" | sed s/include/lib/` $GUI_LIB_LOC"
     GUI_LIB_LOC=
     for try in $gui_libs; do
-      for libtry in "$try"/libXm.a "$try"/libXm.so* "$try"/libXm.sl; do
+      for libtry in "$try"/libXm.a "$try"/libXm.so* "$try"/libXm.dylib "$try"/libXm.sl; do
 	if test -f "$libtry"; then
 	  GUI_LIB_LOC=$try
 	fi
