diff -uNbwr rubygems-1.0.1/lib/rubygems/defaults.rb rubygems-1.0.1-new/lib/rubygems/defaults.rb
--- rubygems-1.0.1/lib/rubygems/defaults.rb	2007-12-17 15:09:42.000000000 -0500
+++ rubygems-1.0.1-new/lib/rubygems/defaults.rb	2007-12-27 10:58:55.000000000 -0500
@@ -12,7 +12,8 @@
       File.join File.dirname(ConfigMap[:sitedir]), 'Gems',
                 ConfigMap[:ruby_version]
     else
-      File.join ConfigMap[:libdir], 'ruby', 'gems', ConfigMap[:ruby_version]
+      # File.join ConfigMap[:libdir], 'ruby', 'gems', ConfigMap[:ruby_version]
+      File.join('@PREFIX@', 'var', 'lib', 'gems', Config::CONFIG['ruby_version'])
     end
   end
 
diff -uNbwr rubygems-1.0.1/lib/rubygems/source_index.rb rubygems-1.0.1-new/lib/rubygems/source_index.rb
--- rubygems-1.0.1/lib/rubygems/source_index.rb	2007-12-17 22:36:44.000000000 -0500
+++ rubygems-1.0.1-new/lib/rubygems/source_index.rb	2007-12-27 10:46:55.000000000 -0500
@@ -217,7 +217,7 @@
                       end
       else
         version_requirement = platform_only || Gem::Requirement.default
-        gem_pattern = /#{gem_pattern}/i
+        gem_pattern = /^#{gem_pattern}/i
       end
 
       unless Gem::Requirement === version_requirement then
