diff -Nurd RubyCocoa-0.13.2/framework/src/ruby/osx/objc/oc_import.rb RubyCocoa-0.13.2-new/framework/src/ruby/osx/objc/oc_import.rb
--- RubyCocoa-0.13.2/framework/src/ruby/osx/objc/oc_import.rb	2008-02-14 03:38:14.000000000 -0500
+++ RubyCocoa-0.13.2-new/framework/src/ruby/osx/objc/oc_import.rb	2008-05-21 16:32:19.000000000 -0400
@@ -10,11 +10,13 @@
 module OSX
 
   FRAMEWORK_PATHS = [
+    '@FINKPREFIX@/lib/Frameworks',
     '/System/Library/Frameworks',
     '/Library/Frameworks'
   ]
 
   SIGN_PATHS = [
+    '@FINKPREFIX@/lib/BridgeSupport',
     '/System/Library/BridgeSupport', 
     '/Library/BridgeSupport'
   ]
diff -Nurd RubyCocoa-0.13.2/misc/libffi/Makefile.rubycocoa RubyCocoa-0.13.2-new/misc/libffi/Makefile.rubycocoa
--- RubyCocoa-0.13.2/misc/libffi/Makefile.rubycocoa	2008-02-14 03:38:13.000000000 -0500
+++ RubyCocoa-0.13.2-new/misc/libffi/Makefile.rubycocoa	2008-05-21 16:32:42.000000000 -0400
@@ -3,7 +3,7 @@
 LIB = libffi.a
 
 SYSROOT = 
-CFLAGS = -O3 -I./include -DMACOSX -arch ppc -arch i386
+CFLAGS = -O3 -I./include -DMACOSX
 
 all: $(LIB)
 
diff -Nurd RubyCocoa-0.13.2/misc/libffi/ltcf-c.sh RubyCocoa-0.13.2-new/misc/libffi/ltcf-c.sh
--- RubyCocoa-0.13.2/misc/libffi/ltcf-c.sh	2008-02-14 03:38:13.000000000 -0500
+++ RubyCocoa-0.13.2-new/misc/libffi/ltcf-c.sh	2008-05-22 12:29:47.000000000 -0400
@@ -389,7 +389,7 @@
 		allow_undefined_flag='-flat_namespace -undefined suppress'
 		;;
 	    10.*)
-		allow_undefined_flag='-undefined dynamic_lookup'
+		allow_undefined_flag='-flat_namespace -undefined suppress'
 		;;
 	esac
     fi
