--- bluefish-0.9/configure	Tue Feb 18 09:48:52 2003
+++ bluefish-0.9-patched/configure	Sun Feb 23 18:12:45 2003
@@ -3520,7 +3520,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-laspell  $LIBS"
+LIBS="-laspell -laspell-common -lstdc++ $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 #line $LINENO "configure"
 /* confdefs.h.  */
@@ -3569,7 +3569,7 @@
 echo "$as_me:$LINENO: result: $ac_cv_lib_aspell_new_aspell_config" >&5
 echo "${ECHO_T}$ac_cv_lib_aspell_new_aspell_config" >&6
 if test $ac_cv_lib_aspell_new_aspell_config = yes; then
-  LIBS="$LIBS -laspell"
+  LIBS="$LIBS -laspell -laspell-common -lstdc++"
 		cat >>confdefs.h <<\_ACEOF
 #define HAVE_LIBASPELL 1
 _ACEOF
