--- TeXmacs-1.0.1.13-src/src/Window/X/x_drawable.cc~	Tue May  6 16:37:36 2003
+++ TeXmacs-1.0.1.13-src/src/Window/X/x_drawable.cc	Thu May  8 22:00:38 2003
@@ -413,7 +413,6 @@
     url temp_name= url_temp ();
     save_string (temp_name, nice_ps, true);
     string cmd= "tm_gs";
-    if (gs_type == 2) cmd= "timeout -s 9 2 " * cmd;
     system (cmd, temp_name);
     remove (temp_name);
     if (N(ps_figs_nr) == 0) ps_figs_last_gc= texmacs_time ();
--- TeXmacs-1.0.1.13-src/misc/scripts/tm_gs~ Wed May  7 12:32:27 2003
+++ TeXmacs-1.0.1.13-src/misc/scripts/tm_gs  Sun May 11 13:49:04 2003
@@ -1,2 +1,2 @@
 #!/bin/sh
-exec gs -q -sDEVICE=x11 -dNOPAUSE -dSAFER "$@" -c quit
+exec gs -q -sDEVICE=x11 -dNOPAUSE -dSAFER -dNOEPS "$@" -c quit

