
--- lib/qextmdi/src/Makefile.am	13 May 2003 14:57:23 -0000	1.1.1.1
+++ lib/qextmdi/src/Makefile.am	26 May 2003 05:22:23 -0000
@@ -3,7 +3,7 @@
 INCLUDES = -I$(top_srcdir)/lib/qextmdi/include -I$(top_srcdir)/lib/qextmdi/res -I$(top_srcdir)/lib/structure $(all_includes)
 
 lib_LTLIBRARIES = libkdevelopqextmdi.la
-libkdevelopqextmdi_la_LDFLAGS = $(all_libraries) -version-info 3:0
+libkdevelopqextmdi_la_LDFLAGS = $(all_libraries) -version-info 3:0 -no-undefined
 libkdevelopqextmdi_la_LIBADD  = $(LIB_KPARTS)
 
 libkdevelopqextmdi_la_SOURCES = qextmdichildarea.cpp qextmdichildfrm.cpp qextmdichildfrmcaption.cpp qextmdichildview.cpp qextmdimainfrm.cpp qextmditaskbar.cpp
--- parts/cppsupport/Makefile.am	13 May 2003 14:57:51 -0000	1.1.1.1
+++ parts/cppsupport/Makefile.am	26 May 2003 05:22:23 -0000
@@ -5,10 +5,16 @@
 INCLUDES = -I$(top_srcdir)/lib/interfaces -I$(top_srcdir)/lib/sourceinfo -I$(top_srcdir)/lib/util $(all_includes)
 
 kde_module_LTLIBRARIES = libkdevcppsupport.la
-libkdevcppsupport_la_LDFLAGS = $(LEXLIB) $(all_libraries) $(KDE_PLUGIN)
-libkdevcppsupport_la_LIBADD = $(top_builddir)/lib/libkdevelop.la
+noinst_LTLIBRARIES = libkdevcppsupport_shared.la
 
-libkdevcppsupport_la_SOURCES = cppsupportpart.cpp ccconfigwidget.cpp ccconfigwidgetbase.ui cppnewclassdlgbase.ui cppnewclassdlg.cpp cppcodecompletion.cpp simpleparser.cpp kdevregexp.cpp problemreporter.cpp driver.cpp errors.cpp lexer.cpp parser.cpp ast.cpp configproblemreporter.ui backgroundparser.cpp implementmethodsdialog.ui implementmethodsdialog.ui.h cppsupport_events.h ast_utils.h ast_utils.cpp tree_parser.h tree_parser.cpp store_walker.h store_walker.cpp parser.h realtime_classbrowser.h realtime_classbrowser.cpp subclassingdlg.cpp subclassingdlgbase.ui lookup.h lookup.cpp keywords.lut.h addmethoddialogbase.ui addmethoddialog.cpp addattributedialogbase.ui addattributedialog.cpp addattributedialog.h KDevCppSupportIface.h KDevCppSupportIface.cpp KDevCppSupportIface.skel classgeneratorconfigbase.ui classgeneratorconfig.cpp classgeneratorconfig.h cppsupportfactory.cpp cppsupportfactory.h
+libkdevcppsupport_shared_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined
+libkdevcppsupport_shared_la_LIBADD  = $(top_builddir)/lib/libkdevelop.la
+libkdevcppsupport_shared_la_SOURCES = cppsupportpart.cpp ccconfigwidget.cpp ccconfigwidgetbase.ui cppnewclassdlgbase.ui cppnewclassdlg.cpp cppcodecompletion.cpp simpleparser.cpp kdevregexp.cpp problemreporter.cpp driver.cpp errors.cpp lexer.cpp parser.cpp ast.cpp configproblemreporter.ui backgroundparser.cpp implementmethodsdialog.ui implementmethodsdialog.ui.h cppsupport_events.h ast_utils.h ast_utils.cpp tree_parser.h tree_parser.cpp store_walker.h store_walker.cpp parser.h realtime_classbrowser.h realtime_classbrowser.cpp subclassingdlg.cpp subclassingdlgbase.ui lookup.h lookup.cpp keywords.lut.h addmethoddialogbase.ui addmethoddialog.cpp addattributedialogbase.ui addattributedialog.cpp addattributedialog.h KDevCppSupportIface.h KDevCppSupportIface.cpp KDevCppSupportIface.skel classgeneratorconfigbase.ui classgeneratorconfig.cpp classgeneratorconfig.h cppsupportfactory.cpp cppsupportfactory.h
+
+libkdevcppsupport_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
+libkdevcppsupport_la_LIBADD  = libkdevcppsupport_shared.la
+libkdevcppsupport_la_SOURCES = cppsupport_init.cpp
+libkdevcppsupport_la_DEPENDENCIES = libkdevcppsupport_shared.la
 
 EXTRA_DIST =
 DISTCLEANFILES =
@@ -29,5 +35,5 @@
 bin_PROGRAMS =  r++
 r___LDFLAGS = $(all_libraries)
 r___SOURCES = main.cpp cppsupportfactory.h
-r___DEPENDENCIES = libkdevcppsupport.la
-r___LDADD = libkdevcppsupport.la
+r___DEPENDENCIES = libkdevcppsupport_shared.la
+r___LDADD = libkdevcppsupport_shared.la
RCS file: parts/cppsupport/cppsupport_init.cpp
diff -N parts/cppsupport/cppsupport_init.cpp
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ parts/cppsupport/cppsupport_init.cpp	13 May 2003 15:17:46 -0000	1.1
@@ -0,0 +1,6 @@
+extern void showMemUsage();
+
+void dummy_memusage () {
+	showMemUsage();
+}
+
--- parts/javasupport/Makefile.am	13 May 2003 14:58:02 -0000	1.1.1.1
+++ parts/javasupport/Makefile.am	14 May 2003 00:15:17 -0000	1.2
@@ -7,7 +7,7 @@
 		$(all_includes)
 
 kde_module_LTLIBRARIES = libkdevjavasupport.la
-libkdevjavasupport_la_LDFLAGS = $(LEXLIB) $(all_libraries) $(KDE_PLUGIN) 
+libkdevjavasupport_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) 
 libkdevjavasupport_la_LIBADD = 	$(top_builddir)/lib/libkdevelop.la \
 				$(top_builddir)/lib/antlr/src/libantlr.la
 
--- parts/nedit/Makefile.am	13 May 2003 14:58:02 -0000	1.1.1.1
+++ parts/nedit/Makefile.am	26 May 2003 05:22:23 -0000
@@ -8,6 +8,6 @@
 
 libkneditpart_la_SOURCES = document.cpp view.cpp kneditfactory.cpp
 libkneditpart_la_LIBADD   = -lktexteditor
-libkneditpart_la_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+libkneditpart_la_LDFLAGS = $(all_libraries) $(KDE_RPATH) -no-undefined
 
 INCLUDES= -I$(top_srcdir)/interfaces -I$(top_srcdir)/kparts -I$(top_srcdir) $(all_includes)
--- parts/qeditor/qeditor.cpp	13 May 2003 14:58:05 -0000	1.1.1.1
+++ parts/qeditor/qeditor.cpp	13 May 2003 15:17:46 -0000	1.2
@@ -203,12 +203,12 @@
 	    insert( "\t" );
 	e->accept();
     } else if( m_electricKeys.contains( e->ascii() ) ){
-	insert( e->text(), FALSE );
+	insert( e->text(), (bool)FALSE );
         indent();
 	e->accept();
     } else if( e->ascii() == '{' || e->ascii() == '}' ||
                e->ascii() == ':' || e->ascii() == '#' ){
-	insert( e->text(), FALSE );
+	insert( e->text(), (bool)FALSE );
 	e->accept();
     } else if (e->state() == Qt::ControlButton) {
 	bool bRemove = false;
--- parts/scripting/scriptingpart.cpp	13 May 2003 14:58:06 -0000	1.1.1.1
+++ parts/scripting/scriptingpart.cpp	26 May 2003 05:22:24 -0000
@@ -9,6 +9,8 @@
  *                                                                         *
  ***************************************************************************/
 
+#include "config.h"
+
 #include "scriptingpart.h"
 #include <Python.h>
 #include <stdlib.h>
