diff -Naur mysql-5.0.13-rc/mysys/default.c mysql-5.0.13-rc.fink/mysys/default.c
--- mysql-5.0.13-rc/mysys/default.c	2005-09-23 04:39:57.000000000 +0900
+++ mysql-5.0.13-rc.fink/mysys/default.c	2005-10-16 02:13:52.000000000 +0900
@@ -980,6 +980,8 @@
 {
   const char *env, **ptr= default_directories;
 
+  *ptr++= "@PREFIX@/etc/";
+
 #ifdef __WIN__
   *ptr++= "C:/";
 
diff -Naur mysql-5.0.13-rc/scripts/mysqld_safe.sh mysql-5.0.13-rc.fink/scripts/mysqld_safe.sh
--- mysql-5.0.13-rc/scripts/mysqld_safe.sh	2005-09-23 04:40:24.000000000 +0900
+++ mysql-5.0.13-rc.fink/scripts/mysqld_safe.sh	2005-10-16 02:18:56.000000000 +0900
@@ -132,22 +132,24 @@
 # Second, try to find the data directory
 #
 
-# Try where the binary installs put it
-if test -d $MY_BASEDIR_VERSION/data/mysql
-then
-  DATADIR=$MY_BASEDIR_VERSION/data
-  if test -z "$defaults" -a -r "$DATADIR/my.cnf"
-  then
-    defaults="--defaults-extra-file=$DATADIR/my.cnf"
-  fi
-# Next try where the source installs put it
-elif test -d $MY_BASEDIR_VERSION/var/mysql
-then
-  DATADIR=$MY_BASEDIR_VERSION/var
-# Or just give up and use our compiled-in default
-else
+# Fink's MySQL uses @PREFIX@/var/lib by default..
+
+# # Try where the binary installs put it
+# if test -d $MY_BASEDIR_VERSION/data/mysql
+# then
+#   DATADIR=$MY_BASEDIR_VERSION/data
+#   if test -z "$defaults" -a -r "$DATADIR/my.cnf"
+#   then
+#     defaults="--defaults-extra-file=$DATADIR/my.cnf"
+#   fi
+# # Next try where the source installs put it
+# elif test -d $MY_BASEDIR_VERSION/var/mysql
+# then
+#   DATADIR=$MY_BASEDIR_VERSION/var
+# # Or just give up and use our compiled-in default
+# else
   DATADIR=@localstatedir@
-fi
+# fi
 
 if test -z "$MYSQL_HOME"
 then 
From: msvensson at mysql dot com
Date: December 22 2006 12:38am
Subject: bk commit into 5.0 tree (msvensson:1.2353) BUG#23125

Below is the list of changes that have just been committed into a local
5.0 repository of msvensson. When msvensson does a push these changes will
be propagated to the main repository and, within 24 hours after the
push, to the public repository.
For information on how to access the public repository
see http://dev.mysql.com/doc/mysql/en/installing-source-tree.html

ChangeSet@stripped, 2006-12-22 00:38:34+01:00, msvensson@stripped +5 -0
  Bug #23125 [patch] trigger test fails when run as root
   - Skip tests using chmod when running as root

  mysql-test/t/information_schema_chmod.test@stripped, 2006-12-22 00:38:30+01:00, msvensson@stripped +2 -2
    Use mysqltest builtin "chmode"

  mysql-test/t/mysqltest.test@stripped, 2006-12-22 00:38:30+01:00, msvensson@stripped +3 -0
    Dont run as root - test chmod

  mysql-test/t/rpl_rotate_logs.test@stripped, 2006-12-22 00:38:30+01:00, msvensson@stripped +5 -3
    Use mysqltest builtin "chmode"
    Use 'write_file' and 'chmod' instead of system

  mysql-test/t/trigger.test@stripped, 2006-12-22 00:38:30+01:00, msvensson@stripped +10 -4
    Don't run as root
    Use 'write_file' and 'chmod'

  mysql-test/t/varbinary.test@stripped, 2006-12-22 00:38:31+01:00, msvensson@stripped +4 -0
    Don't run as root

# This is a BitKeeper patch.  What follows are the unified diffs for the
# set of deltas contained in the patch.  The rest of the patch, the part
# that BitKeeper cares about, is below these diffs.
# User:	msvensson
# Host:	pilot.mysql.com
# Root:	/home/msvensson/mysql/mysql-5.0-maint

--- 1.56/mysql-test/t/trigger.test	2006-11-13 23:38:34 +01:00
+++ 1.57/mysql-test/t/trigger.test	2006-12-22 00:38:30 +01:00
@@ -1,3 +1,7 @@
+# This test uses chmod, can't be run with root permissions
+-- source include/not_as_root.inc
+
+
 #
 # Basic triggers test
 #
@@ -1138,8 +1142,10 @@
        event_object_table, action_statement from information_schema.triggers
        where event_object_schema = 'test';
 # Trick which makes update of second .TRN file impossible
-system echo dummy >$MYSQLTEST_VARDIR/master-data/test/t1_ai.TRN~;
-system chmod 000 $MYSQLTEST_VARDIR/master-data/test/t1_ai.TRN~;
+write_file $MYSQLTEST_VARDIR/master-data/test/t1_ai.TRN~;
+dummy
+EOF
+chmod 0000 $MYSQLTEST_VARDIR/master-data/test/t1_ai.TRN~;
 --error 1
 rename table t1 to t2;
 # 't1' should be still there and triggers should work correctly
@@ -1148,8 +1154,8 @@
 select trigger_schema, trigger_name, event_object_schema,
        event_object_table, action_statement from information_schema.triggers
        where event_object_schema = 'test';
-system chmod 600 $MYSQLTEST_VARDIR/master-data/test/t1_ai.TRN~;
-system rm $MYSQLTEST_VARDIR/master-data/test/t1_ai.TRN~;
+chmod 0600 $MYSQLTEST_VARDIR/master-data/test/t1_ai.TRN~;
+remove_file $MYSQLTEST_VARDIR/master-data/test/t1_ai.TRN~;
 # Let us check that updates to .TRN files were rolled back too
 drop trigger t1_bi;
 drop trigger t1_ai;

--- 1.49/mysql-test/t/mysqltest.test	2006-11-30 10:56:30 +01:00
+++ 1.50/mysql-test/t/mysqltest.test	2006-12-22 00:38:30 +01:00
@@ -1,6 +1,9 @@
 # This test should work in embedded server after mysqltest is fixed
 -- source include/not_embedded.inc
 
+# This test uses chmod, can't be run with root permissions
+-- source include/not_as_root.inc
+
 # ============================================================================
 #
 # Test of mysqltest itself

--- 1.11/mysql-test/t/varbinary.test	2006-11-29 23:03:43 +01:00
+++ 1.12/mysql-test/t/varbinary.test	2006-12-22 00:38:31 +01:00
@@ -1,3 +1,7 @@
+# This test uses chmod, can't be run with root permissions
+-- source include/not_as_root.inc
+
+
 # Initialise
 --disable_warnings
 drop table if exists t1;

--- 1.2/mysql-test/t/information_schema_chmod.test	2006-05-23 10:52:46 +02:00
+++ 1.3/mysql-test/t/information_schema_chmod.test	2006-12-22 00:38:30 +01:00
@@ -17,7 +17,7 @@
 #
 create database mysqltest;
 create table mysqltest.t1(a int);
---exec chmod -r $MYSQLTEST_VARDIR/master-data/mysqltest
+chmod 0000 $MYSQLTEST_VARDIR/master-data/mysqltest;
 select table_schema from information_schema.tables where table_schema='mysqltest';
---exec chmod +r $MYSQLTEST_VARDIR/master-data/mysqltest
+exec chmod 0777 $MYSQLTEST_VARDIR/master-data/mysqltest;
 drop database mysqltest;

--- 1.65/mysql-test/t/rpl_rotate_logs.test	2006-04-24 10:21:04 +02:00
+++ 1.66/mysql-test/t/rpl_rotate_logs.test	2006-12-22 00:38:30 +01:00
@@ -17,8 +17,10 @@
 drop table if exists t1, t2, t3, t4;
 --enable_warnings
 connect (slave,localhost,root,,test,$SLAVE_MYPORT,slave.sock);
-system cat /dev/null > $MYSQLTEST_VARDIR/slave-data/master.info;
-system chmod 000 $MYSQLTEST_VARDIR/slave-data/master.info;
+# Create empty file
+write_file $MYSQLTEST_VARDIR/slave-data/master.info;
+EOF
+chmod 0000 $MYSQLTEST_VARDIR/slave-data/master.info;
 connection slave;
 --disable_warnings
 drop table if exists t1, t2, t3, t4;
@@ -29,7 +31,7 @@
 --replace_result $MYSQL_TEST_DIR TESTDIR
 --error 1105,1105,29
 start slave;
-system chmod 600  $MYSQLTEST_VARDIR/slave-data/master.info;
+chmod 0600  $MYSQLTEST_VARDIR/slave-data/master.info;
 # It will fail again because the file is empty so the slave cannot get valuable
 # info about how to connect to the master from it (failure in
 # init_strvar_from_file() in init_master_info()).
From: kent at mysql dot com
Date: January 2 2007 11:01am
Subject: bk commit into 5.0 tree (kent:1.2357) BUG#25359

Below is the list of changes that have just been committed into a local
5.0 repository of kent. When kent does a push these changes will
be propagated to the main repository and, within 24 hours after the
push, to the public repository.
For information on how to access the public repository
see http://dev.mysql.com/doc/mysql/en/installing-source-tree.html

ChangeSet@stripped, 2007-01-02 11:01:48+01:00, kent@stripped +1 -0
  view.result:
    Temporary work around for bug#25359

  mysql-test/r/view.result@stripped, 2007-01-02 11:01:39+01:00, kent@stripped +4 -4
    Temporary work around for bug#25359

# This is a BitKeeper patch.  What follows are the unified diffs for the
# set of deltas contained in the patch.  The rest of the patch, the part
# that BitKeeper cares about, is below these diffs.
# User:	kent
# Host:	kent-amd64.(none)
# Root:	/home/kent/bk/main/mysql-5.0

--- 1.184/mysql-test/r/view.result	2007-01-02 11:01:54 +01:00
+++ 1.185/mysql-test/r/view.result	2007-01-02 11:01:54 +01:00
@@ -2686,12 +2686,12 @@
 v1	CREATE ALGORITHM=UNDEFINED DEFINER=`root`@`localhost` SQL SECURITY DEFINER VIEW `v1` AS select (year(now()) - year(`t1`.`DOB`)) AS `Age` from `t1` having (`Age` < 75)
 SELECT (year(now())-year(DOB)) AS Age FROM t1 HAVING Age < 75;
 Age
-42
-38
+43
+39
 SELECT * FROM v1;
 Age
-42
-38
+43
+39
 DROP VIEW v1;
 DROP TABLE t1;
 CREATE TABLE t1 (id int NOT NULL PRIMARY KEY, a char(6) DEFAULT 'xxx');
