diff -ruN pgrouting-1.05-orig/CMakeLists.txt pgrouting-1.05/CMakeLists.txt
--- pgrouting-1.05-orig/CMakeLists.txt	2011-01-04 14:12:22.000000000 +0900
+++ pgrouting-1.05/CMakeLists.txt	2011-01-04 14:32:32.000000000 +0900
@@ -26,16 +26,19 @@
   MESSAGE(FATAL ERROR " Please check your Boost installation ") 
 ENDIF(Boost_INCLUDE_DIRS)
 
-EXEC_PROGRAM(pg_config
- ARGS --pkglibdir
- OUTPUT_VARIABLE LIB_DIR)
-
-IF(LIB_DIR)
-  MESSAGE("Output directory for libraries is set to ${LIB_DIR}")
-ELSE(LIB_DIR)
-  SET(LIB_DIR ${PGROUTING_BINARY_DIR}/lib)
-  MESSAGE("pg_config not found, output directory for libraries is set to ${PGROUTING_BINARY_DIR}/lib")
-ENDIF(LIB_DIR)
+#EXEC_PROGRAM(pg_config
+# ARGS --pkglibdir
+# OUTPUT_VARIABLE LIB_DIR)
+#
+#IF(LIB_DIR)
+#  MESSAGE("Output directory for libraries is set to ${LIB_DIR}")
+#ELSE(LIB_DIR)
+#  SET(LIB_DIR ${PGROUTING_BINARY_DIR}/lib)
+#  MESSAGE("pg_config not found, output directory for libraries is set to ${PGROUTING_BINARY_DIR}/lib")
+#ENDIF(LIB_DIR)
+# FINK: if Fink directory is symlinked, it causes an error (Thanks Fangism)
+SET(LIB_DIR ${POSTGRESQL_LIBRARIES})
+MESSAGE("Output directory for libraries is set to ${LIB_DIR}")
 
 if(PGROUTING_CORE_INCLUDE_DIR)
    set(PGROUTING_CORE_FOUND TRUE)
@@ -64,7 +67,8 @@
 
 IF(UNIX)
   SET(LIBRARY_INSTALL_PATH ${LIB_DIR})
-  SET(SQL_INSTALL_PATH /usr/share/postlbs)
+  # SET(SQL_INSTALL_PATH /usr/share/postlbs)
+  SET(SQL_INSTALL_PATH ${CMAKE_INSTALL_PREFIX}/share/postlbs)
   MESSAGE("Installation directory for libraries is set to ${LIBRARY_INSTALL_PATH} and for SQL files is set to ${SQL_INSTALL_PATH}")
 ELSE(UNIX)
   SET(LIBRARY_INSTALL_PATH ${PGROUTING_BINARY_DIR}/lib)
