diff -Naur BitTorrent-3.4.1a.orig/btreannounce.py BitTorrent-3.4.1a/btreannounce.py
--- BitTorrent-3.4.1a.orig/btreannounce.py	Sun Mar 14 19:10:34 2004
+++ BitTorrent-3.4.1a/btreannounce.py	Sun Mar 14 19:11:24 2004
@@ -1,11 +1,11 @@
 #!/usr/bin/env python
-
+
 # Written by Henry 'Pi' James and Bram Cohen
 # see LICENSE.txt for license information
-
-from sys import argv
+
+from sys import argv, exit
 from BitTorrent.bencode import bencode, bdecode
-
+
 if len(argv) < 3:
     print '%s http://new.uri:port/announce file1.torrent file2.torrent' % argv[0]
     print
diff -Naur BitTorrent-3.4.1a.orig/setup.py BitTorrent-3.4.1a/setup.py
--- BitTorrent-3.4.1a.orig/setup.py	Sun Mar 14 19:10:34 2004
+++ BitTorrent-3.4.1a/setup.py	Sun Mar 14 19:13:20 2004
@@ -18,9 +18,9 @@
     
     packages = ["BitTorrent"],
 
-    scripts = ["btdownloadgui.py", "btdownloadheadless.py", "btdownloadlibrary.py", 
+    scripts = ["btdownloadheadless.py", "btdownloadlibrary.py", 
         "bttrack.py", "btmakemetafile.py", "btlaunchmany.py", "btcompletedir.py",
-        "btdownloadcurses.py", "btcompletedirgui.py", "btlaunchmanycurses.py", 
+        "btdownloadcurses.py", "btlaunchmanycurses.py", 
         "btmakemetafile.py", "btreannounce.py", "btrename.py", "btshowmetainfo.py",
         "bttest.py"]
     )
