--- centericq-4.9.8/configure.orig	Sun Nov  9 17:20:54 2003
+++ centericq-4.9.8/configure	Sun Nov  9 18:22:01 2003
@@ -2056,16 +2056,16 @@
 cat > conftest.$ac_ext <<EOF
 #line 2058 "configure"
 #include "confdefs.h"
+#include <iconv.h>
 /* Override any gcc2 internal prototype to avoid an error.  */
 #ifdef __cplusplus
 extern "C"
 #endif
 /* We use char because int might match the return type of a gcc2
     builtin and then its argument prototype would still apply.  */
-char iconv();
 
 int main() {
-iconv()
+iconv(0,0,0,0,0)
 ; return 0; }
 EOF
 if { (eval echo configure:2072: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
@@ -2104,16 +2104,16 @@
 cat > conftest.$ac_ext <<EOF
 #line 2106 "configure"
 #include "confdefs.h"
+#include <iconv.h>
 /* Override any gcc2 internal prototype to avoid an error.  */
 #ifdef __cplusplus
 extern "C"
 #endif
 /* We use char because int might match the return type of a gcc2
     builtin and then its argument prototype would still apply.  */
-char iconv();
 
 int main() {
-iconv()
+iconv(0,0,0,0,0)
 ; return 0; }
 EOF
 if { (eval echo configure:2120: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
