diff -ru dnstop-20060424/Makefile dnstop-20060424-patched/Makefile
--- dnstop-20060424/Makefile	2005-02-03 19:18:13.000000000 +0100
+++ dnstop-20060424-patched/Makefile	2006-05-02 23:25:29.000000000 +0200
@@ -12,7 +12,7 @@
 all: $(PROG)
 
 $(PROG): $(PROG).o
-	$(CC) $(CFLAGS) -o $@ $(PROG).o -lpcap -lcurses $(XTRALIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(PROG).o -lpcap -lcurses
 
 clean:
 	rm -f $(PROG) $(PROG).o $(PROG).core $(PROG).c~
diff -ru dnstop-20060424/dnstop.c dnstop-20060424-patched/dnstop.c
--- dnstop-20060424/dnstop.c	2005-04-05 22:43:53.000000000 +0200
+++ dnstop-20060424-patched/dnstop.c	2006-05-03 00:13:42.000000000 +0200
@@ -25,6 +25,9 @@
 #include <assert.h>
 #include <arpa/inet.h>
 #include <arpa/nameser.h>
+#ifdef __APPLE__
+#include <arpa/nameser_compat.h>
+#endif
 
 #include <sys/socket.h>
 #include <net/if_arp.h>
@@ -1235,6 +1238,12 @@
 	fprintf(stderr, "pcap_setfilter failed\n");
 	exit(1);
     }
+    x = pcap_setnonblock(pcap, 1, errbuf);
+    if (x < 0) {
+	fprintf(stderr, "pcap_setnonblock failed: %s\n", errbuf);
+	exit(1);
+    }
+
     switch (pcap_datalink(pcap)) {
     case DLT_EN10MB:
 	handle_datalink = handle_ether;
@@ -1266,9 +1275,11 @@
     if (interactive) {
 	init_curses();
 	while (0 == Quit) {
-	    if (readfile_state < 2)
-		if (readfile_state || pcap_select(pcap, 1, 0))
-		    x = pcap_dispatch(pcap, 50, handle_pcap, NULL);
+	    if (readfile_state < 2) {
+		if (!readfile_state)
+		    pcap_select(pcap, 1, 0);
+		x = pcap_dispatch(pcap, 50, handle_pcap, NULL);
+	    }
 	    if (0 == x && 1 == readfile_state) {
 		/* block on keyboard until user quits */
 		readfile_state++;
