diff -ru eggdrop1.6.10/configure eggdrop1.6.10-patched/configure
--- eggdrop1.6.10/configure	2002-03-29 20:48:49.000000000 +0100
+++ eggdrop1.6.10-patched/configure	2002-05-03 11:50:36.000000000 +0200
@@ -5257,10 +5257,10 @@
 	tcl7.3 tcl73 tcl7.2 tcl72 tcl7.1 tcl71 tcl7.0 tcl70"
 
 # Tcl library filename suffixes
-tcllibextensions=".so .so.1 .so.1.0 .so.1.2 .a .sl .dll"
+tcllibextensions=".dylib .so .so.1 .so.1.0 .so.1.2 .a .sl .dll"
 
 # Tcl library search paths
-tcllibpaths="/usr/local/lib /usr/local/pkgs/tcl/lib \
+tcllibpaths="${prefix}/lib /usr/local/lib /usr/local/pkgs/tcl/lib \
 	/usr/lib /lib /usr/i486-linuxaout/lib \
 	/beos/system/lib /sys/lib \
 	$HOME/lib $HOME/tcl/lib $HOME"
@@ -5269,7 +5269,7 @@
 tclheadernames="tcl.h"
 
 # Tcl header search paths
-tclheaderpaths="/usr/local/include /usr/local/pkgs/tcl/include \
+tclheaderpaths="${prefix}/include /usr/local/include /usr/local/pkgs/tcl/include \
 	/usr/include /beos/system/include /beos/devel/include \
        /sys/include $HOME/include $HOME/tcl/include $HOME"
 
diff -ru eggdrop1.6.10/src/eggdrop.h eggdrop1.6.10-patched/src/eggdrop.h
--- eggdrop1.6.10/src/eggdrop.h	2002-03-29 20:40:24.000000000 +0100
+++ eggdrop1.6.10-patched/src/eggdrop.h	2002-05-03 12:13:20.000000000 +0200
@@ -123,6 +123,8 @@
 /* NAME_MAX is what POSIX defines, but BSD calls it MAXNAMLEN.
  * Use 255 if we can't find anything else.
  */
+#include <sys/param.h>
+
 #ifndef NAME_MAX
 #  ifdef MAXNAMLEN
 #    define NAME_MAX	MAXNAMLEN
diff -ru eggdrop1.6.10/src/main.h eggdrop1.6.10-patched/src/main.h
--- eggdrop1.6.10/src/main.h	2002-01-02 04:46:35.000000000 +0100
+++ eggdrop1.6.10-patched/src/main.h	2002-05-03 12:12:52.000000000 +0200
@@ -105,12 +105,4 @@
 	(_target)[(_len) - 1] = 0;					\
 } while (0)
 
-#ifdef BORGCUBES
-
-/* For net.c */
-#  define O_NONBLOCK	00000004    /* POSIX non-blocking I/O		   */
-
-#endif				/* BORGUBES */
-
-
 #endif				/* _EGG_MAIN_H */
diff -ru eggdrop1.6.10/src/mod/Makefile.in eggdrop1.6.10-patched/src/mod/Makefile.in
--- eggdrop1.6.10/src/mod/Makefile.in	2000-12-21 21:20:03.000000000 +0100
+++ eggdrop1.6.10-patched/src/mod/Makefile.in	2002-05-03 12:08:21.000000000 +0200
@@ -32,7 +32,7 @@
 
 modconfig = $(top_srcdir)/misc/modconfig --bindir=../.. \
   --top_srcdir=$(top_srcdir)
-egg_ac_parameters = @egg_ac_parameters@
+egg_ac_parameters = --prefix=@prefix@
 
 MAKE_MOD = $(MAKE) 'MAKE=$(MAKE)' 'CC=$(CC)' 'LD=$(LD)' \
 'STRIP=$(STRIP)' 'CFLAGS=$(MOD_CFLAGS)' 'CPPFLAGS=$(MOD_CPPFLAGS)' \
diff -ru eggdrop1.6.10/src/mod/compress.mod/compress.c eggdrop1.6.10-patched/src/mod/compress.mod/compress.c
--- eggdrop1.6.10/src/mod/compress.mod/compress.c	2002-01-02 04:46:37.000000000 +0100
+++ eggdrop1.6.10-patched/src/mod/compress.mod/compress.c	2002-05-03 11:51:36.000000000 +0200
@@ -38,6 +38,7 @@
 
 #include "compress_config.h"
 #ifdef HAVE_MMAP
+#  undef panic
 #  include <sys/types.h>
 #  include <sys/mman.h>
 #  include <sys/stat.h>
