diff -uNr ekg-1.6rc3/configure ekg-1.6rc3-new/configure
--- ekg-1.6rc3/configure	2005-07-18 13:17:58.000000000 -0400
+++ ekg-1.6rc3-new/configure	2005-08-21 15:31:31.000000000 -0400
@@ -4312,7 +4312,7 @@
 		# FIXME: -fPIC is required for -shared on many architectures,
 		# so we specify it here, but the right way would probably be to
 		# properly detect whether it is actually required.
-		CFLAGS="-shared -fPIC -Wl,-z,defs $CFLAGS $PTHREAD_CFLAGS"
+		CFLAGS="-dynamiclib -fPIC $CFLAGS $PTHREAD_CFLAGS"
 		LIBS="$PTHREAD_LIBS $LIBS"
 		CC="$PTHREAD_CC"
 
diff -uNr ekg-1.6rc3/examples/Makefile.in ekg-1.6rc3-new/examples/Makefile.in
--- ekg-1.6rc3/examples/Makefile.in	2003-06-24 16:56:52.000000000 -0400
+++ ekg-1.6rc3-new/examples/Makefile.in	2005-08-21 15:30:36.000000000 -0400
@@ -1,6 +1,6 @@
 # $Id: Makefile.in,v 1.13 2003/06/24 20:56:52 wojtekka Exp $
 
-CC = @CC@
+CC = glibtool --tag=CC --mode=link @CC@
 CFLAGS = @CFLAGS_LIBGADU@ @CFLAGS@ -I.. -I../lib -I../src
 LIBS = @LIBS_LIBGADU@
 MAKE = @MAKE@
diff -uNr ekg-1.6rc3/lib/Makefile.in ekg-1.6rc3-new/lib/Makefile.in
--- ekg-1.6rc3/lib/Makefile.in	2005-06-25 02:41:07.000000000 -0400
+++ ekg-1.6rc3-new/lib/Makefile.in	2005-08-21 15:30:36.000000000 -0400
@@ -25,7 +25,7 @@
 LOBJS = common.lo libgadu.lo pubdir.lo http.lo dcc.lo events.lo pubdir50.lo obsolete.lo
 SRCS = $(OBJS:.o=.c)
 
-SHAREDMM = libgadu.so
+SHAREDMM = libgadu.la
 SHAREDM = $(SHAREDMM).$(VERSION_MAJOR)
 SHARED = $(SHAREDM).$(VERSION_MINOR)
 STATIC = libgadu.a
@@ -51,14 +51,12 @@
 #
 
 %.lo:	%.c
-	$(CC) -c $(CPPFLAGS) $(CFLAGS) -fPIC $< -o $@
+	glibtool --tag=CC --mode=compile $(CC) -c $(CPPFLAGS) $(CFLAGS) -fPIC $< -o $@
 
-shared:	dep $(SHARED)
+shared:	dep $(SHAREDMM)
 
-$(SHARED): $(LOBJS)
-	$(CC) -shared -Wl,-soname,libgadu.so.$(VERSION_MAJOR) -o $(SHARED) $(LOBJS) $(LIBS) -lc
-	$(LN_S) -f $(SHARED) $(SHAREDM)
-	$(LN_S) -f $(SHARED) $(SHAREDMM)
+$(SHAREDMM): $(LOBJS)
+	glibtool --tag=CC --mode=link $(CC) -o $(SHAREDMM) -rpath $(libdir) -version-info $(VERSION_MAJOR):$(VERSION_MINOR) $(LOBJS) $(LIBS)
 
 #
 
@@ -81,15 +79,12 @@
 	$(INSTALL) -m 644 libgadu.pc $(DESTDIR)$(libdir)/pkgconfig
 
 install-shared:	install-common @strip_libgadu_so@
-	$(INSTALL) -m 755 $(SHARED) $(DESTDIR)$(libdir)
-	cd $(DESTDIR)$(libdir) && $(LN_S) -f $(SHARED) $(SHAREDM)
-	cd $(DESTDIR)$(libdir) && $(LN_S) -f $(SHARED) $(SHAREDMM)
+	glibtool --tag=CC --mode=install $(INSTALL) -m 755 $(SHAREDMM) $(DESTDIR)$(libdir)
 
 install-static:	install-common
 	$(INSTALL) -m 755 $(STATIC) $(DESTDIR)$(libdir)
 
 strip-libgadu-so:
-	$(STRIP) $(SHARED)
 
 #
 
diff -uNr ekg-1.6rc3/src/Makefile.in ekg-1.6rc3-new/src/Makefile.in
--- ekg-1.6rc3/src/Makefile.in	2003-05-19 09:36:41.000000000 -0400
+++ ekg-1.6rc3-new/src/Makefile.in	2005-08-21 15:30:36.000000000 -0400
@@ -25,11 +25,11 @@
 
 all:	dep ekg @ioctld@
 
-ekg:	$(OBJS) @libgadu_a@
-	$(CC) $(CFLAGS) -o ekg $(OBJS) -L../lib @lgadu@ @libgadu_a@ $(LDFLAGS) $(LIBS)
+ekg:	$(OBJS) ../lib/libgadu.la
+	glibtool --tag=CC --mode=link $(CC) $(CFLAGS) -o ekg $(OBJS) -L../lib ../lib/libgadu.la $(LDFLAGS) $(LIBS)
 
 ioctld:	ioctld.c
-	$(CC) $(CFLAGS) ioctld.c -o ioctld $(LIBS) @IOCTLD_OBJS@
+	glibtool --tag=CC --mode=link $(CC) $(CFLAGS) ioctld.c -o ioctld $(LIBS) @IOCTLD_OBJS@
 
 dep:	.depend
 
@@ -42,17 +42,15 @@
 
 install-ekg:	@strip_ekg@
 	$(INSTALL) -d $(DESTDIR)$(bindir)
-	$(INSTALL) -m 755 ekg $(DESTDIR)$(bindir)
+	glibtool --tag=CC --mode=install $(INSTALL) -m 755 ekg $(DESTDIR)$(bindir)
 
 install-ioctld:	@strip_ioctld@
 	$(INSTALL) -d $(DESTDIR)$(libexecdir)
-	$(INSTALL) -m 755 ioctld $(DESTDIR)$(libexecdir)
+	glibtool --tag=CC --mode=install $(INSTALL) -m 755 ioctld $(DESTDIR)$(libexecdir)
 
 strip-ekg:
-	$(STRIP) ekg
 
 strip-ioctld:
-	$(STRIP) ioctld
 
 #
 
