? nget.presameserverids
Index: file.cc
===================================================================
RCS file: /cvsroot/nget/nget/file.cc,v
retrieving revision 1.29
diff -u -r1.29 file.cc
--- nget-0.24.2/file.cc	31 Aug 2003 21:45:45 -0000	1.29
+++ nget-0.24.2/file.cc	26 Oct 2003 01:02:56 -0000
@@ -187,13 +187,13 @@
 	fd=-1;
 	return i;
 }
-inline int c_file_fd::isopen(void)const{
+int c_file_fd::isopen(void)const{
 	return (fd>=0);
 }
-inline ssize_t c_file_fd::dowrite(const void *data,size_t len){
+ssize_t c_file_fd::dowrite(const void *data,size_t len){
 	return ::write(fd,(char*)data,len);
 }
-inline ssize_t c_file_fd::doread(void *data,size_t len){
+ssize_t c_file_fd::doread(void *data,size_t len){
 	return ::read(fd,data,len);
 }
 int c_file_fd::seek(int offset, int whence){
@@ -223,13 +223,13 @@
 	fs=NULL;
 	return i;
 }
-inline int c_file_stream::isopen(void)const{
+int c_file_stream::isopen(void)const{
 	return (fs!=0);
 }
-inline ssize_t c_file_stream::dowrite(const void *data,size_t len){
+ssize_t c_file_stream::dowrite(const void *data,size_t len){
 	return fwrite(data,1,len,fs);
 }
-inline ssize_t c_file_stream::doread(void *data,size_t len){
+ssize_t c_file_stream::doread(void *data,size_t len){
 	return fread(data,1,len,fs);
 }
 #endif
@@ -269,14 +269,14 @@
 	sock=SOCK_INVALID;
 	return i;
 }
-inline int c_file_tcp::isopen(void)const{
+int c_file_tcp::isopen(void)const{
 	return sock_isvalid(sock);
 }
-inline ssize_t c_file_tcp::dowrite(const void *data,size_t len){
+ssize_t c_file_tcp::dowrite(const void *data,size_t len){
 	//don't need to use sock_write_ensured since c_file::write handles the looping.
 	return sock_write(sock,(char*)data, len);
 }
-inline ssize_t c_file_tcp::doread(void *data,size_t len){
+ssize_t c_file_tcp::doread(void *data,size_t len){
 	return sock_read(sock,data,len);
 }
 bool c_file_tcp::datawaiting(void) const {
Index: file_gz.cc
===================================================================
RCS file: /cvsroot/nget/nget/file_gz.cc,v
retrieving revision 1.2
diff -u -r1.2 file_gz.cc
--- nget-0.24.2/file_gz.cc	31 Aug 2003 19:35:20 -0000	1.2
+++ nget-0.24.2/file_gz.cc	26 Oct 2003 01:02:56 -0000
@@ -50,13 +50,13 @@
 	gzh=NULL;
 	return i;
 }
-inline int c_file_gz::isopen(void)const{
+int c_file_gz::isopen(void)const{
 	return (gzh!=0);
 }
-inline ssize_t c_file_gz::dowrite(const void *data,size_t len){
+ssize_t c_file_gz::dowrite(const void *data,size_t len){
 	return gzwrite(gzh,(void*)data,len);
 }
-inline ssize_t c_file_gz::doread(void *data,size_t len){
+ssize_t c_file_gz::doread(void *data,size_t len){
 	return gzread(gzh,data,len);
 }
 #endif
