diff -uNbr squid-2.5.STABLE1/src/cf.data.pre squid-2.5.STABLE1-patched/src/cf.data.pre
--- squid-2.5.STABLE1/src/cf.data.pre	Wed Sep  4 09:35:01 2002
+++ squid-2.5.STABLE1-patched/src/cf.data.pre	Mon Oct  7 20:09:04 2002
@@ -2064,6 +2064,9 @@
 #
 # INSERT YOUR OWN RULE(S) HERE TO ALLOW ACCESS FROM YOUR CLIENTS
 
+# Allow proxy connections from localhost
+http_access allow localhost
+
 # Exampe rule allowing access from your local networks. Adapt
 # to list your (internal) IP networks from where browsing should
 # be allowed
