diff -Pur unison-2.9.1.orig/ubase/uprintf.ml unison-2.9.1/ubase/uprintf.ml
--- unison-2.9.1.orig/ubase/uprintf.ml	Thu Apr 11 01:13:23 2002
+++ unison-2.9.1/ubase/uprintf.ml	Tue Aug 24 12:02:22 2004
@@ -10,8 +10,8 @@
 (*                                                                     *)
 (***********************************************************************)
 
-external format_int: string -> int -> string = "format_int"
-external format_float: string -> float -> string = "format_float"
+external format_int: string -> int -> string = "caml_format_int"
+external format_float: string -> float -> string = "caml_format_float"
 
 let fprintf outchan doafter format =
   let format = (Obj.magic format : string) in
