diff -ru xchat-2.0.6/configure xchat-2.0.6-patched/configure
--- xchat-2.0.6/configure	2003-11-28 04:08:56.000000000 +0100
+++ xchat-2.0.6-patched/configure	2003-12-29 20:39:32.000000000 +0100
@@ -23731,7 +23731,7 @@
                         PY_PREFIX=`$pythonpath -c 'import sys; print sys.prefix'`
                         PY_EXEC_PREFIX=`$pythonpath -c 'import sys; print sys.exec_prefix'`
                         if test -f $PY_INC/Python.h; then
-                                PY_LIBS="-L$PY_LIB/config -lpython$PY_VER -lpthread -lutil"
+                                PY_LIBS="-L$PY_LIB/config -lpython$PY_VER"
                                 PY_CFLAGS="-I$PY_INC"
                                 echo "$as_me:$LINENO: result: ok" >&5
 echo "${ECHO_T}ok" >&6
