diff -ru DB_File-1.814.orig/config.in DB_File-1.814/config.in
--- DB_File-1.814.orig/config.in	2005-10-31 17:38:45.000000000 -0500
+++ DB_File-1.814/config.in	2005-11-20 12:06:38.000000000 -0500
@@ -9,7 +9,7 @@
 #    Change the path below to point to the directory where db.h is
 #    installed on your system.
 
-INCLUDE	= /usr/local/BerkeleyDB/include
+INCLUDE	= @PREFIX@/include/db4
 #INCLUDE	= /usr/local/include
 #INCLUDE	= /usr/include
 
@@ -18,7 +18,7 @@
 #    Change the path below to point to the directory where libdb is
 #    installed on your system.
 
-LIB	= /usr/local/BerkeleyDB/lib
+LIB	= @PREFIX@/lib
 #LIB	= /usr/local/lib
 #LIB	= /usr/lib
 
diff -ru DB_File-1.814.orig/version.c DB_File-1.814/version.c
--- DB_File-1.814.orig/version.c	2004-06-20 18:13:45.000000000 -0400
+++ DB_File-1.814/version.c	2005-11-20 12:08:21.000000000 -0500
@@ -50,7 +50,7 @@
 
     /* Check that the versions of db.h and libdb.a are the same */
     if (Major != DB_VERSION_MAJOR || Minor != DB_VERSION_MINOR 
-		|| Patch != DB_VERSION_PATCH)
+)//		|| Patch != DB_VERSION_PATCH)
 	croak("\nDB_File needs compatible versions of libdb & db.h\n\tyou have db.h version %d.%d.%d and libdb version %d.%d.%d\n",  
 		DB_VERSION_MAJOR, DB_VERSION_MINOR, DB_VERSION_PATCH, 
 		Major, Minor, Patch) ;
