diff -ruN Getopt-Long-2.32.orig/Makefile.PL Getopt-Long-2.32/Makefile.PL
--- Getopt-Long-2.32.orig/Makefile.PL	2002-02-20 04:18:07.000000000 -0700
+++ Getopt-Long-2.32/Makefile.PL	2002-11-08 16:32:15.000000000 -0700
@@ -9,42 +9,42 @@
 sub MY::postamble {
     my $ret = "";
     my $mandir = $Config{installman3dir};
-    my $obsolete = "$mandir/Getopt::GetoptLong.3";
-    if ( -e $obsolete ) {
-	print STDERR (<<EOD);
-
-I found a left-over manual page from a previous version of Getopt::Long.
-This manual page has a wrong name, and should be removed.
-I can add instructions to the Makefile to remove it when you
-install this version.
-EOD
-	my $ans = prompt ("Shall I add the remove instructions?");
-	if ( $ans =~ /^y/i ) {
-	    $ret .= <<EOD;
-install ::
-	\$(RM_F) "$obsolete"
-
-EOD
-	}
-    }
+#    my $obsolete = "$mandir/Getopt::GetoptLong.3";
+#    if ( -e $obsolete ) {
+#	print STDERR (<<EOD);
+#
+#I found a left-over manual page from a previous version of Getopt::Long.
+#This manual page has a wrong name, and should be removed.
+#I can add instructions to the Makefile to remove it when you
+#install this version.
+#EOD
+#	my $ans = prompt ("Shall I add the remove instructions?");
+#	if ( $ans =~ /^y/i ) {
+#	    $ret .= <<EOD;
+#install ::
+#	\$(RM_F) "$obsolete"
+#
+#EOD
+#	}
+#    }
     my $libdir = $Config{installprivlib};
-    my $obsolete = "$libdir/auto/Getopt/Long";
-    if ( -d $obsolete ) {
-	print STDERR (<<EOD);
-
-I found left-over files from a previous version of Getopt::Long.
-These should be removed.
-I can add instructions to the Makefile to remove them when you
-install this version.
-EOD
-	my $ans = prompt ("Shall I add the remove instructions?");
-	if ( $ans =~ /^y/i ) {
-	    $ret .= <<EOD;
-install ::
-	\$(RM_RF) "$libdir/auto/Getopt/Long"
-EOD
-	}
-    }
+#    my $obsolete = "$libdir/auto/Getopt/Long";
+#    if ( -d $obsolete ) {
+#	print STDERR (<<EOD);
+#
+#I found left-over files from a previous version of Getopt::Long.
+#These should be removed.
+#I can add instructions to the Makefile to remove them when you
+#install this version.
+#EOD
+#	my $ans = prompt ("Shall I add the remove instructions?");
+#	if ( $ans =~ /^y/i ) {
+#	    $ret .= <<EOD;
+#install ::
+#	\$(RM_RF) "$libdir/auto/Getopt/Long"
+#EOD
+#	}
+#    }
     $ret;
 }
 
