diff -Nurd mysql-5.0.45/mysys/default.c mysql-5.0.45-patched/mysys/default.c
--- mysql-5.0.45/mysys/default.c	2007-07-04 09:06:27.000000000 -0400
+++ mysql-5.0.45-patched/mysys/default.c	2009-06-12 14:05:02.000000000 -0400
@@ -989,6 +989,8 @@
 {
   const char *env, **ptr= default_directories;
 
+  *ptr++= "@PREFIX@/etc/";
+
 #ifdef __WIN__
   *ptr++= "C:/";
 
diff -Nurd mysql-5.0.45/scripts/mysqld_safe.sh mysql-5.0.45-patched/scripts/mysqld_safe.sh
--- mysql-5.0.45/scripts/mysqld_safe.sh	2007-07-04 09:06:09.000000000 -0400
+++ mysql-5.0.45-patched/scripts/mysqld_safe.sh	2009-06-12 14:05:02.000000000 -0400
@@ -133,22 +133,24 @@
 # Second, try to find the data directory
 #
 
-# Try where the binary installs put it
-if test -d $MY_BASEDIR_VERSION/data/mysql
-then
-  DATADIR=$MY_BASEDIR_VERSION/data
-  if test -z "$defaults" -a -r "$DATADIR/my.cnf"
-  then
-    defaults="--defaults-extra-file=$DATADIR/my.cnf"
-  fi
-# Next try where the source installs put it
-elif test -d $MY_BASEDIR_VERSION/var/mysql
-then
-  DATADIR=$MY_BASEDIR_VERSION/var
-# Or just give up and use our compiled-in default
-else
+# Fink's MySQL uses @PREFIX@/var/lib by default..
+
+# # Try where the binary installs put it
+# if test -d $MY_BASEDIR_VERSION/data/mysql
+# then
+#   DATADIR=$MY_BASEDIR_VERSION/data
+#   if test -z "$defaults" -a -r "$DATADIR/my.cnf"
+#   then
+#     defaults="--defaults-extra-file=$DATADIR/my.cnf"
+#   fi
+# # Next try where the source installs put it
+# elif test -d $MY_BASEDIR_VERSION/var/mysql
+# then
+#   DATADIR=$MY_BASEDIR_VERSION/var
+# # Or just give up and use our compiled-in default
+# else
   DATADIR=@localstatedir@
-fi
+# fi
 
 if test -z "$MYSQL_HOME"
 then 
