--- libast-0.5/include/libast/types.h.in.orig	2003-05-04 15:08:44.000000000 -0700
+++ libast-0.5/include/libast/types.h.in	2003-05-04 15:08:57.000000000 -0700
@@ -24,6 +24,8 @@
 #ifndef _LIBAST_TYPES_H_
 #define _LIBAST_TYPES_H_
 
+typedef unsigned int socklen_t;
+
 /* Macros to construct object types from the basenames of the types (obj, str, etc.) */
 #define SPIF_TYPE(type)                  spif_ ## type ## _t
 #define SPIF_CONST_TYPE(type)            spif_const_ ## type ## _t
