diff -uNbr libgsf-1.7.2/gsf/gsf-output-stdio.c libgsf-1.7.2-patched/gsf/gsf-output-stdio.c
--- libgsf-1.7.2/gsf/gsf-output-stdio.c	Fri Jan 24 01:56:36 2003
+++ libgsf-1.7.2-patched/gsf/gsf-output-stdio.c	Sat Apr 26 09:05:13 2003
@@ -19,6 +19,8 @@
  * USA
  */
 
+#undef _POSIX_SOURCE
+
 #include <gsf-config.h>
 #include <gsf/gsf-output-stdio.h>
 #include <gsf/gsf-output-impl.h>
diff -uNbr libgsf-1.7.2/gsf-gnome/gsf-input-bonobo.c libgsf-1.7.2-patched/gsf-gnome/gsf-input-bonobo.c
--- libgsf-1.7.2/gsf-gnome/gsf-input-bonobo.c	Mon Aug 26 15:14:12 2002
+++ libgsf-1.7.2-patched/gsf-gnome/gsf-input-bonobo.c	Sat Apr 26 09:05:15 2003
@@ -19,6 +19,8 @@
  * USA
  */
 
+#undef _POSIX_SOURCE
+
 #include <gsf-config.h>
 #include <gsf-gnome/gsf-input-bonobo.h>
 #include <gsf/gsf-input-impl.h>
diff -uNbr libgsf-1.7.2/gsf-gnome/gsf-output-bonobo.c libgsf-1.7.2-patched/gsf-gnome/gsf-output-bonobo.c
--- libgsf-1.7.2/gsf-gnome/gsf-output-bonobo.c	Mon Sep 16 02:07:05 2002
+++ libgsf-1.7.2-patched/gsf-gnome/gsf-output-bonobo.c	Sat Apr 26 09:05:15 2003
@@ -19,6 +19,8 @@
  * USA
  */
 
+#undef _POSIX_SOURCE
+
 #include <gsf-config.h>
 #include <gsf-gnome/gsf-output-bonobo.h>
 #include <bonobo/bonobo-exception.h>
diff -uNbr libgsf-1.7.2/gsf-gnome/gsf-shared-bonobo-stream.c libgsf-1.7.2-patched/gsf-gnome/gsf-shared-bonobo-stream.c
--- libgsf-1.7.2/gsf-gnome/gsf-shared-bonobo-stream.c	Sun Aug 18 22:46:16 2002
+++ libgsf-1.7.2-patched/gsf-gnome/gsf-shared-bonobo-stream.c	Sat Apr 26 09:05:15 2003
@@ -19,6 +19,8 @@
  * USA
  */
 
+#undef _POSIX_SOURCE
+
 #include <gsf-config.h>
 #include <gsf-gnome/gsf-shared-bonobo-stream.h>
 #include <gsf/gsf-impl-utils.h>
