diff -ur libtecla/configure libtecla.new/configure
--- libtecla/configure	Tue May 28 17:18:08 2002
+++ libtecla.new/configure	Fri Nov 22 10:43:09 2002
@@ -1627,6 +1627,12 @@
   LINK_SHARED='ld -b +h $$(@F) +k +vshlibunsats -o $$@ -c libtecla.map.opt $$(LIB_OBJECTS) $$(LIBS) -lc'
   SHARED_CFLAGS="+z"
   ;;
+*darwin*)
+  SHARED_EXT=".${MAJOR_VER}.${MINOR_VER}.${MICRO_VER}.dylib"
+  SHARED_ALT=".dylib .${MAJOR_VER}.dylib"
+  LINK_SHARED='$(CC) -o $$@ -dynamiclib -flat_namespace -undefined suppress -compatibility_version '${MAJOR_VER}.${MINOR_VER}' -current_version '${MAJOR_VER}.${MINOR_VER}.${MICRO_VER}' -install_name '${libdir}'/$$@ $$(LIB_OBJECTS)'
+  SHARED_CFLAGS=""
+  ;;
 *dec-osf*)
   cat >> confdefs.h <<\EOF
 #define _OSF_SOURCE 1
@@ -1637,7 +1643,10 @@
 
 
 if test "$GCC"_ = "yes"_ && test "$LINK_SHARED"_ != "_" ; then
-  SHARED_CFLAGS="-fpic"
+  case $target_os in
+  darwin*) ;;
+  *) SHARED_CFLAGS="-fpic"
+  esac
   case $target_os in
   sparc*solaris*) SHARED_CFLAGS="$SHARED_CFLAGS -mno-app-regs"
   esac
