diff -Naur neon-0.24.4/Makefile.in neon-0.24.4-patched/Makefile.in
--- neon-0.24.4/Makefile.in	2003-09-22 21:57:02.000000000 +0200
+++ neon-0.24.4-patched/Makefile.in	2003-10-27 16:06:04.000000000 +0100
@@ -21,7 +21,7 @@
 man1dir = $(mandir)/man1
 man3dir = $(mandir)/man3
 datadir = @datadir@
-docdir = $(datadir)/doc/neon-@NEON_VERSION@
+docdir = $(datadir)/doc/neon24-ssl
 includedir = @includedir@
 neonincludes = $(includedir)/neon
 pkgconfigdir = $(libdir)/pkgconfig
