diff -ru readline-4.3/doc/Makefile.in readline-4.3-patched/doc/Makefile.in
--- readline-4.3/doc/Makefile.in	Mon May 20 09:54:51 2002
+++ readline-4.3-patched/doc/Makefile.in	Thu Aug  1 09:59:32 2002
@@ -194,11 +194,11 @@
 		${INSTALL_DATA} $(srcdir)/history.info $(DESTDIR)$(infodir)/history.info; \
 	fi
 	-if $(SHELL) -c 'install-info --version' >/dev/null 2>&1; then \
-		install-info --dir-file=$(DESTDIR)$(infodir)/dir \
+		install-info --infodir=$(DESTDIR)$(infodir)/dir \
 			$(DESTDIR)$(infodir)/readline.info ; \
-		install-info --dir-file=$(DESTDIR)$(infodir)/dir \
+		install-info --infodir=$(DESTDIR)$(infodir)/dir \
 			$(DESTDIR)$(infodir)/history.info ; \
-		install-info --dir-file=$(DESTDIR)$(infodir)/dir \
+		install-info --infodir=$(DESTDIR)$(infodir)/dir \
 			$(DESTDIR)$(infodir)/rluserman.info ; \
 	else true; fi
 	-${INSTALL_DATA} $(srcdir)/readline.3 $(DESTDIR)$(man3dir)/readline$(man3ext)
diff -ru readline-4.3/support/shobj-conf readline-4.3-patched/support/shobj-conf
--- readline-4.3/support/shobj-conf	Wed Apr 17 10:27:46 2002
+++ readline-4.3-patched/support/shobj-conf	Thu Aug  1 10:02:05 2002
@@ -146,9 +146,9 @@
 	SHLIB_LIBSUFF='dylib'
 
 	SHOBJ_LDFLAGS='-dynamic'
-	SHLIB_XLDFLAGS='-arch_only `/usr/bin/arch` -install_name $(libdir)/$@ -current_version $(SHLIB_MAJOR)$(SHLIB_MINOR) -compatibility_version $(SHLIB_MAJOR) -v'
+	SHLIB_XLDFLAGS='-arch_only `/usr/bin/arch` -install_name $(libdir)/`echo $@ | sed "s:\\(.*\\.[0-9]\\)\\.[0-9]:\\1:"` -current_version $(SHLIB_MAJOR)$(SHLIB_MINOR) -compatibility_version 4.2.0 -v'
 
-	SHLIB_LIBS='-lSystem'
+	SHLIB_LIBS='-lSystem -lcc_dynamic -lncurses -L@FINK@/lib'
 	;;
 
 openbsd*)

