diff -Nru jove4.16.0.64.orig/Makefile jove4.16.0.64/Makefile
--- jove4.16.0.64.orig/Makefile	Sun Feb  1 11:32:39 2004
+++ jove4.16.0.64/Makefile	Fri Jul  9 10:04:09 2004
@@ -19,14 +19,14 @@
 #	Must not be "nr".
 #
 # If they don't exist, this makefile will try to create the directories
-# LIBDIR and SHAREDIR.  All others must already exist.
+# BINDIR, LIBDIR, MANDIR and SHAREDIR.  All others must already exist.
 
 JOVEHOME = /usr/local
-SHAREDIR = $(JOVEHOME)/lib/jove
+SHAREDIR = $(JOVEHOME)/share/jove
 LIBDIR = $(JOVEHOME)/lib/jove
 BINDIR = $(JOVEHOME)/bin
 XEXT=
-MANDIR = $(JOVEHOME)/man/man$(MANEXT)
+MANDIR = $(JOVEHOME)/share/man/man$(MANEXT)
 MANEXT = 1
 
 # TMPDIR is where the tmp files get stored, usually /tmp, /var/tmp, or
@@ -38,11 +38,11 @@
 # (in case the system startup salvages tempfiles by moving them,
 # which is probably a good idea).
 
-TMPDIR = /tmp
-RECDIR = /var/preserve
+TMPDIR = $(JOVEHOME)/var/tmp/jove
+RECDIR = $(JOVEHOME)/var/tmp/jove/preserve
 
 # DFLTSHELL is the default shell invoked by JOVE and TEACHJOVE.
-DFLTSHELL = /bin/csh
+DFLTSHELL = /bin/sh
 
 # The install commands of BSD and System V differ in unpleasant ways:
 # -c: copy (BSD); -c dir: destination directory (SysV)
@@ -52,18 +52,20 @@
 # If you know that /bin/install is the BSD program, you can use it.
 # "cp" will work reasonably well, but be aware that any links continue
 # referencing the old file with new contents.
-
+INSTALL=/usr/bin/install
 INSTALLFLAGS = # -g bin -o root
 
 # to install executable files
-XINSTALL=cp
+#XINSTALL=cp
 #XINSTALL=/usr/ucb/install $(INSTALLFLAGS) -c -m 755 # -s
 #CYGWIN32: XINSTALL=install $(INSTALLFLAGS) -c -m 755
+XINSTALL=$(INSTALL) $(INSTALLFLAGS) -c -m 755
 
 # to install text files
-TINSTALL=cp
+#TINSTALL=cp
 #TINSTALL=/usr/ucb/install $(INSTALLFLAGS) -c -m 644
 #CYGWIN32: TINSTALL=install $(INSTALLFLAGS) -c -m 644
+TINSTALL=$(INSTALL) $(INSTALLFLAGS) -c -m 644
 
 # These should all just be right if the above ones are.
 # You will confuse JOVE if you move anything from LIBDIR or SHAREDIR.
@@ -73,6 +75,7 @@
 RECOVER = $(LIBDIR)/recover$(XEXT)
 PORTSRV = $(LIBDIR)/portsrv$(XEXT)
 JOVERC = $(SHAREDIR)/jove.rc
+JOVELOCALRC = $(SHAREDIR)/jove-local.rc
 TERMSDIR = $(SHAREDIR)
 CMDS.DOC = $(SHAREDIR)/cmds.doc
 TEACH-JOVE = $(SHAREDIR)/teach-jove
@@ -146,7 +149,7 @@
 # On DEC OSF/1 and Digital UNIX VV4.0, add -std1 to enable ANSI C features
 # and perhaps -g3 for more debugging info with optimization.
 
-OPTFLAGS = -O
+OPTFLAGS = -Os
 
 # For making dependencies under BSD systems
 DEPENDFLAG = -M
@@ -197,6 +200,7 @@
 #	CC = /opt/SUNWspro/bin/cc
 # For DG AViiON, expect compile errors unless you use the GNU C compiler:
 #	CC=gcc
+CC = gcc
 
 # Load invocation of cc.
 # to use Purify(TM): LDCC = purify $(CC)
@@ -361,14 +365,21 @@
 	 $(TEACH-JOVE) $(CMDS.DOC) $(TERMSDIR)docs \
 	 $(PORTSRVINST) $(RECOVER) $(JOVE) $(TEACHJOVE) $(MANUALS)
 	$(TINSTALL) doc/jove.rc $(JOVERC)
+	$(TINSTALL) doc/jove-local.rc $(JOVELOCALRC)
 	@echo See the README about changes to /etc/rc or /etc/rc.local
 	@echo so that the system recovers jove files on reboot after a crash
 
+$(BINDIR)::
+	test -d $(BINDIR) || mkdir -p $(BINDIR)
+
 $(LIBDIR)::
-	test -d $(LIBDIR) || mkdir $(LIBDIR)
+	test -d $(LIBDIR) || mkdir -p $(LIBDIR)
+
+$(MANDIR)::
+	test -d $(MANDIR) || mkdir -p $(MANDIR)
 
 $(SHAREDIR)::
-	test -d $(SHAREDIR) || mkdir $(SHAREDIR)
+	test -d $(SHAREDIR) || mkdir -p $(SHAREDIR)
 
 $(TEACH-JOVE): doc/teach-jove
 	$(TINSTALL) doc/teach-jove $(TEACH-JOVE)
@@ -397,7 +408,7 @@
 $(RECOVER): recover$(XEXT)
 	$(XINSTALL) recover$(XEXT) $(RECOVER)
 
-$(JOVE): jjove$(XEXT)
+$(JOVE): $(BINDIR) jjove$(XEXT)
 	$(XINSTALL) jjove$(XEXT) $(JOVE)
 
 $(TEACHJOVE): teachjove$(XEXT)
@@ -409,7 +420,7 @@
 	     -e 's;<SHAREDIR>;$(SHAREDIR);' \
 	     -e 's;<SHELL>;$(DFLTSHELL);' doc/jove.nr > doc/jove.$(MANEXT)
 
-$(JOVEM): doc/jove.$(MANEXT)
+$(JOVEM): $(MANDIR) doc/jove.$(MANEXT)
 	$(TINSTALL) doc/jove.$(MANEXT) $(JOVEM)
 
 # doc/jove.doc is the formatted manpage (only needed by DOS)
diff -Nru jove4.16.0.64.orig/doc/jove-local.rc jove4.16.0.64/doc/jove-local.rc
--- jove4.16.0.64.orig/doc/jove-local.rc	Wed Dec 31 16:00:00 1969
+++ jove4.16.0.64/doc/jove-local.rc	Tue Jul  6 15:33:39 2004
@@ -0,0 +1,3 @@
+bind-to-key query-replace-string ^[%
+bind-to-key suspend-jove ^Z
+bind-to-key insert-file ^Xi
