diff -ur sam.bak/include/libc.h sam/include/libc.h
--- sam.bak/include/libc.h	2005-06-05 21:37:51.000000000 -0700
+++ sam/include/libc.h	2005-06-13 16:17:30.000000000 -0700
@@ -5,6 +5,7 @@
 
 typedef unsigned char		uchar;
 typedef	unsigned short		Rune;
+typedef unsigned long		ulong;
 
 #define	sprint				sprintf
 #define	dup(a,b)			dup2(a,b)
diff -ur sam.bak/libXg/Makefile sam/libXg/Makefile
--- sam.bak/libXg/Makefile	2005-06-05 21:37:59.000000000 -0700
+++ sam/libXg/Makefile	2005-06-13 16:14:15.000000000 -0700
@@ -8,7 +8,7 @@
 #	
 #	Additionally, -D_POSIX_SOURCE (or its equivalent) may be specified
 #	if your compiler supports posix-compatible compilation
-OS=-DIRIX -ansiposix
+OS=-DIRIX
 
 #	add -Iincludedir for any include directories that need to be searched
 INCS=-I../include
diff -ur sam.bak/libframe/Makefile sam/libframe/Makefile
--- sam.bak/libframe/Makefile	2005-06-05 21:37:59.000000000 -0700
+++ sam/libframe/Makefile	2005-06-13 16:14:09.000000000 -0700
@@ -8,7 +8,7 @@
 #	
 #	Additionally, -D_POSIX_SOURCE (or its equivalent) may be specified
 #	if your compiler supports posix-compatible compilation
-OS=-DIRIX -ansiposix
+OS=-DIRIX
 
 #	add -Iincludedir for any include directories that need to be searched
 #	for posix header files (for UMIPS, add -I/usr/include/posix)
diff -ur sam.bak/sam/Makefile sam/sam/Makefile
--- sam.bak/sam/Makefile	2005-06-05 21:37:59.000000000 -0700
+++ sam/sam/Makefile	2005-06-13 16:14:24.000000000 -0700
@@ -8,7 +8,7 @@
 #	
 #	Additionally, -D_POSIX_SOURCE (or its equivalent) may be specified
 #	if your compiler supports posix-compatible compilation
-OS=-DIRIX -ansiposix
+OS=-DIRIX
 
 #	add -Iincludedir for any include directories that need to be searched
 #	for posix header files (for UMIPS, add -I/usr/include/posix)
diff -ur sam.bak/samterm/Makefile sam/samterm/Makefile
--- sam.bak/samterm/Makefile	2005-06-05 21:37:59.000000000 -0700
+++ sam/samterm/Makefile	2005-06-13 16:14:27.000000000 -0700
@@ -8,7 +8,7 @@
 #	
 #	Additionally, -D_POSIX_SOURCE (or its equivalent) may be specified
 #	if your compiler supports posix-compatible compilation
-OS=-DIRIX -ansiposix
+OS=-DIRIX
 
 #	add -Iincludedir for any include directories that need to be searched
 #	for posix header files (for UMIPS, add -I/usr/include/posix)
