diff -Naur mell-1.0.0.orig/etc/init-mell.el mell-1.0.0/etc/init-mell.el
--- mell-1.0.0.orig/etc/init-mell.el	Thu Jan  1 09:00:00 1970
+++ mell-1.0.0/etc/init-mell.el	Thu Mar 11 17:09:52 2004
@@ -0,0 +1,17 @@
+;;;; init-mell.el: init file for mell
+;;;; $Id: _init-mell.el.in,v 1.2 2003/06/02 18:13:17 komatsu Exp $
+;;;; 
+;;;; Copyright (C) 2003 Hiroyuki Komatsu <komatsu@taiyaki.org>
+;;;;     All rights reserved.
+;;;;     This is free software with ABSOLUTELY NO WARRANTY.
+;;;; 
+;;;; You can redistribute it and/or modify it under the terms of 
+;;;; the GNU General Public License version 2.
+
+(defconst mell-version '("1.0.0" "2003-06-01"))
+
+(provide 'init-mell)
+;;;; Text properties for Emacs.
+; Local variables:
+; mode: emacs-lisp
+; End:
diff -Naur mell-1.0.0.orig/fink/mell-install mell-1.0.0/fink/mell-install
--- mell-1.0.0.orig/fink/mell-install	Thu Jan  1 09:00:00 1970
+++ mell-1.0.0/fink/mell-install	Thu Mar 11 17:31:37 2004
@@ -0,0 +1,58 @@
+#! /bin/sh -e
+# @PREFIX@/lib/emacsen-common/packages/install/mell
+
+FLAVOR=$1
+VERSION=1.0.0-1
+PACKAGE=mell
+
+if [ "X${FLAVOR}" = "X" ]; then
+    echo "Need argument to determine FLAVOR of emacsen"
+    exit 1
+fi
+if [ "X${PACKAGE}" = "X" ]; then
+    echo "Internal error: need package name"
+    exit 1
+fi
+if [ "X${FLAVOR}" = Xemacs ]; then exit 0; fi
+if [ "X${FLAVOR}" = Xemacs19 ]; then exit 0; fi
+if [ "X${FLAVOR}" = Xmule2 ]; then exit 0; fi
+
+ELDIR=@PREFIX@/share/emacs/site-lisp/${PACKAGE}
+SITELISP=@PREFIX@/share/${FLAVOR}/site-lisp
+ELCDIR=${SITELISP}/${PACKAGE}
+APELDIR=${SITELISP}/apel
+FLAGS="${SITEFLAG} -q -batch -l ./path.el -l ${APELDIR}/poe.elc -f batch-byte-compile"
+STAMP="${ELCDIR}/compile-stamp"
+
+LOG=CompilationLog
+
+echo -n "install/${PACKAGE}: Handling install for emacsen flavor ${FLAVOR}..."
+if [ -e "${STAMP}" ]; then
+    if [ "X${VERSION}" = X"$(cat ${STAMP})" ]; then
+	echo " skipped. (already compiled)" 
+	exit 0
+    else
+	rm -f ${STAMP}
+    fi
+fi
+
+rm -rf ${ELCDIR}
+install -m 755 -d ${ELCDIR}
+cd ${ELDIR}
+FILES=$(echo *.el)
+cp ${FILES} ${ELCDIR}
+cd ${ELCDIR}
+
+cat << EOF > path.el
+(setq load-path (cons "${APELDIR}/" (cons "." load-path)) byte-compile-warnings nil)
+EOF
+${FLAVOR} ${FLAGS} ${FILES} > ${LOG} 2>&1
+gzip -9 ${LOG}
+
+rm -f *.el
+
+echo "${VERSION}" > ${STAMP}
+chmod 644 ${STAMP}
+
+echo " done."
+exit 0
diff -Naur mell-1.0.0.orig/fink/mell-remove mell-1.0.0/fink/mell-remove
--- mell-1.0.0.orig/fink/mell-remove	Thu Jan  1 09:00:00 1970
+++ mell-1.0.0/fink/mell-remove	Thu Mar 11 17:32:33 2004
@@ -0,0 +1,11 @@
+#!/bin/sh -e
+# @PREFIX@/lib/emacsen-common/packages/remove/mell
+
+FLAVOR=$1
+PACKAGE=mell
+
+if [ ${FLAVOR} != emacs ]; then
+
+    echo remove/${PACKAGE}: purging byte-compiled files for ${FLAVOR}
+    rm -rf @PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}
+fi
diff -Naur mell-1.0.0.orig/fink/mell-startup mell-1.0.0/fink/mell-startup
--- mell-1.0.0.orig/fink/mell-startup	Thu Jan  1 09:00:00 1970
+++ mell-1.0.0/fink/mell-startup	Thu Mar 11 17:31:59 2004
@@ -0,0 +1,10 @@
+;; -*-emacs-lisp-*-
+;;
+;; Emacs startup file for Fink mell package
+;;
+;; We have to add this to the load-path:
+(setq load-path (cons (concat "@PREFIX@/share/"
+                              (symbol-name fink-emacs-flavor)
+			      "/site-lisp/mell") load-path))
+
+(require 'init-mell nil t)
