diff -Naur prime-el-1.4.1.orig/fink/prime-el-install prime-el-1.4.1/fink/prime-el-install
--- prime-el-1.4.1.orig/fink/prime-el-install	Thu Jan  1 09:00:00 1970
+++ prime-el-1.4.1/fink/prime-el-install	Tue Mar 23 22:56:07 2004
@@ -0,0 +1,70 @@
+#! /bin/sh -e
+# @PREFIX@/lib/emacsen-common/packages/install/prime-el
+
+FLAVOR=$1
+VERSION=1.4.1-1
+PACKAGE=prime-el
+
+if [ "X${FLAVOR}" = "X" ]; then
+  echo "Need argument to determine FLAVOR of emacsen"
+  exit 1
+fi
+if [ "X${PACKAGE}" = "X" ]; then
+  echo "Internal error: need package name"
+  exit 1
+fi
+
+ELDIR=@PREFIX@/share/emacs/site-lisp/${PACKAGE}
+ELCDIR=@PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}
+FLAGS="${SITEFLAG} -q -batch -l path.el -f batch-byte-compile"
+STAMP="${ELCDIR}/compile-stamp"
+
+LOG=CompilationLog
+
+case "$FLAVOR" in
+    emacs20|emacs21)
+	echo -n "install/${PACKAGE}: Handling install for emacsen flavor ${FLAVOR}..."
+	if [ -e "${STAMP}" ]; then
+	    if [ "X${VERSION}" = X"$(cat ${STAMP})" ]; then
+		echo " skipped. (already compiled)" 
+		exit 0
+	    else
+		rm -f ${STAMP}
+	    fi
+	fi
+	
+	rm -rf ${ELCDIR}
+	install -m 755 -d ${ELCDIR}
+	cd ${ELDIR}
+	FILES=$(find . -name '*.el')
+	cp -R * ${ELCDIR}
+	cd ${ELCDIR}
+	
+	cat << EOF > path.el
+(setq load-path (cons "contrib" (cons "." load-path)) byte-compile-warnings nil)
+EOF
+	${FLAVOR} ${FLAGS} ${FILES} > ${LOG} 2>&1
+	gzip -9 ${LOG}
+	
+	rm -f *.el
+	
+	echo "${VERSION}" > ${STAMP}
+	chmod 644 ${STAMP}
+	
+	echo " done."
+	;;
+    xemacs*)
+	echo -n "install/${PACKAGE}: Handling install for emacsen flavor ${FLAVOR}..."
+	rm -rf ${ELCDIR}
+	install -m 755 -d ${ELCDIR}
+	cd ${ELDIR}
+	cp contrib/*.el *.el ${ELCDIR}
+
+	echo " done. (not compiled)"	
+	;;
+    *)
+        echo "install/$PACKAGE: Ignoring emacsen flavor $FLAVOR."
+        ;;
+esac
+
+exit 0
diff -Naur prime-el-1.4.1.orig/fink/prime-el-remove prime-el-1.4.1/fink/prime-el-remove
--- prime-el-1.4.1.orig/fink/prime-el-remove	Thu Jan  1 09:00:00 1970
+++ prime-el-1.4.1/fink/prime-el-remove	Tue Mar 23 22:42:56 2004
@@ -0,0 +1,11 @@
+#!/bin/sh -e
+# @PREFIX@/lib/emacsen-common/packages/remove/prime-el
+
+FLAVOR=$1
+PACKAGE=prime-el
+
+if [ ${FLAVOR} != emacs ]; then
+
+    echo remove/${PACKAGE}: purging byte-compiled files for ${FLAVOR}
+    rm -rf @PREFIX@/share/${FLAVOR}/site-lisp/${PACKAGE}
+fi
diff -Naur prime-el-1.4.1.orig/fink/prime-el-startup prime-el-1.4.1/fink/prime-el-startup
--- prime-el-1.4.1.orig/fink/prime-el-startup	Thu Jan  1 09:00:00 1970
+++ prime-el-1.4.1/fink/prime-el-startup	Tue Mar 23 22:54:50 2004
@@ -0,0 +1,41 @@
+;;;; 55prime-el.el: init file for prime-el
+;;;; $Id: _init-prime.el.in,v 1.11 2004/01/23 22:45:08 komatsu Exp $
+;;;; 
+;;;; Copyright (C) 2002,2003 Hiroyuki Komatsu <komatsu@taiyaki.org>
+;;;;     All rights reserved.
+;;;;     This is free software with ABSOLUTELY NO WARRANTY.
+;;;; 
+;;;; You can redistribute it and/or modify it under the terms of 
+;;;; the GNU General Public License version 2.
+
+(setq load-path 
+      (cons (concat "@PREFIX@/share/"
+		    (symbol-name fink-emacs-flavor)
+		    "/site-lisp/prime-el")
+	    (cons (concat "@PREFIX@/share/"
+			  (symbol-name fink-emacs-flavor)
+			  "/site-lisp/prime-el/contrib") 
+		  load-path)))
+(defvar prime-init-config-file
+  (expand-file-name (concat "@PREFIX@/etc/prime"
+			    "/Custom_prime-el.el")))
+
+(defconst prime-version '("1.4.1" "2004-03-21"))
+
+(autoload 'prime-mode "prime" "Toggle PRIME mode.
+PRIME is a Japanese Predictive Input Method Editor." t)
+
+(if (functionp 'register-input-method)
+    (register-input-method "japanese-prime" "Japanese"
+			   'prime-mode ""
+			   "PRIME: Japanese PRedictive IME")
+  (global-set-key "\C-\\" 'prime-mode))
+(if (and (boundp 'default-input-method)
+	 (not default-input-method))
+    (setq default-input-method "japanese-prime"))
+
+(provide '55prime-el)
+;;;; Text properties for Emacs.
+; Local variables:
+; mode: emacs-lisp
+; End:
