diff -Naur advancemenu-2.3.0/advance/advance.mak advancemenu-2.3.0-patched/advance/advance.mak
--- advancemenu-2.3.0/advance/advance.mak	Sun Feb  1 15:42:57 2004
+++ advancemenu-2.3.0-patched/advance/advance.mak	Thu Feb  5 22:18:12 2004
@@ -287,21 +287,21 @@
 # Common install
 
 installdirs:
-	-$(INSTALL_DATA_DIR) $(DATADIR)
-	-$(INSTALL_DATA_DIR) $(DATADIR)/doc
-	-$(INSTALL_MAN_DIR) $(PREFIX)/man/man1
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/doc
+	-$(INSTALL_MAN_DIR) $(DESTDIR)$(PREFIX)/share/man/man1
 ifneq ($(wildcard $(EMUSRC)),)
-	-$(INSTALL_DATA_DIR) $(DATADIR)/rom
-	-$(INSTALL_DATA_DIR) $(DATADIR)/sample
-	-$(INSTALL_DATA_DIR) $(DATADIR)/artwork
-	-$(INSTALL_DATA_DIR) $(DATADIR)/image
-	-$(INSTALL_DATA_DIR) $(DATADIR)/crc
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/rom
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/sample
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/artwork
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/image
+	-$(INSTALL_DATA_DIR) $(DESTDIR)$(DATADIR)/crc
 endif
 
 install-data: $(INSTALL_DATAFILES)
 ifdef INSTALL_DATAFILES
 	@for i in $(INSTALL_DATAFILES); do \
-		$(INSTALL_DATA) $$i $(DATADIR); \
+		$(INSTALL_DATA) $$i $(DESTDIR)$(DATADIR); \
 	done
 endif
 
@@ -314,7 +314,7 @@
 
 install-bin: $(INSTALL_BINFILES)
 	@for i in $(INSTALL_BINFILES); do \
-		$(INSTALL_PROGRAM) $$i $(PREFIX)/bin; \
+		$(INSTALL_PROGRAM) $$i $(DESTDIR)$(PREFIX)/bin; \
 	done
 
 uninstall-bin:
@@ -325,7 +325,7 @@
 install-doc: $(DOCOBJ) $(INSTALL_DOCFILES)
 ifdef INSTALL_DOCFILES
 	@for i in $(INSTALL_DOCFILES); do \
-		$(INSTALL_DATA) $$i $(DATADIR)/doc; \
+		$(INSTALL_DATA) $$i $(DESTDIR)$(DATADIR)/doc; \
 	done
 endif
 
@@ -339,7 +339,7 @@
 install-man: $(DOCOBJ) $(INSTALL_MANFILES)
 ifdef INSTALL_MANFILES
 	@for i in $(INSTALL_MANFILES); do \
-		$(INSTALL_DATA) $$i $(PREFIX)/man/man1; \
+		$(INSTALL_DATA) $$i $(DESTDIR)$(PREFIX)/share/man/man1; \
 	done
 endif
 
diff -Naur advancemenu-2.3.0/advance/lib/lcd.c advancemenu-2.3.0-patched/advance/lib/lcd.c
--- advancemenu-2.3.0/advance/lib/lcd.c	Sun Feb  1 15:42:57 2004
+++ advancemenu-2.3.0-patched/advance/lib/lcd.c	Thu Feb  5 22:20:28 2004
@@ -56,6 +56,7 @@
 #include <fcntl.h>
 #endif
 
+#include <netinet/in.h>
 #include <string.h>
 #include <stdlib.h>
 #include <ctype.h>
diff -Naur advancemenu-2.3.0/advance/menu/mconfig.cc advancemenu-2.3.0-patched/advance/menu/mconfig.cc
--- advancemenu-2.3.0/advance/menu/mconfig.cc	Sun Feb  1 15:42:57 2004
+++ advancemenu-2.3.0-patched/advance/menu/mconfig.cc	Thu Feb  5 22:18:13 2004
@@ -456,8 +456,6 @@
 			e = new wmame(a0, a2, a3);
 		} else if (a1 == "dmame") {
 			e = new dmame(a0, a2, a3);
-		} else if (a1 == "xmame") {
-			e = new xmame(a0, a2, a3);
 		} else if (a1 == "dmess") {
 			e = new dmess(a0, a2, a3);
 		} else if (a1 == "draine") {
@@ -984,18 +982,9 @@
 		}
 		if (target_search(path, FILE_MAXPATH, "advpac") == 0) {
 			conf_set(config_context, "", "emulator", "\"advpac\" advpac \"advpac\" \"\"");
-		}
-		if (target_search(path, FILE_MAXPATH, "xmame") == 0) {
-			conf_set(config_context, "", "emulator", "\"xmame\" xmame \"xmame\" \"\"");
 		} else {
-			if (target_search(path, FILE_MAXPATH, "xmame.x11") == 0) {
-				conf_set(config_context, "", "emulator", "\"xmame.x11\" xmame \"xmame.x11\" \"\"");
-			}
 			if (target_search(path, FILE_MAXPATH, "xmame.SDL") == 0) {
 				conf_set(config_context, "", "emulator", "\"xmame.SDL\" xmame \"xmame.SDL\" \"\"");
-			}
-			if (target_search(path, FILE_MAXPATH, "xmame.svgalib") == 0) {
-				conf_set(config_context, "", "emulator", "\"xmame.svgalib\" xmame \"xmame.svgalib\" \"\"");
 			}
 		}
 #endif
