diff -ru tsito-0.8.4-orig/src/Engine.cpp tsito-0.8.4/src/Engine.cpp
--- tsito-0.8.4-orig/src/Engine.cpp	Fri Nov  7 18:19:03 2003
+++ tsito-0.8.4/src/Engine.cpp	Fri Nov  7 18:25:21 2003
@@ -15,7 +15,7 @@
 
 //#include	<ctime>
 
-//#include	<sys/timeb.h>
+#include	<sys/time.h>
 
 #ifdef HAVE_CONFIG_H
 # include	"../config.h"
@@ -200,7 +200,7 @@
   // database, if so then use the move returned (if legal).
   myTimer = Timer::timerForColor(board->sideToMove());
   
-  //struct timeb startTime;
+  struct timeval startTime;
   if (searchState == BETWEEN_SEARCHES)
     {
       searchState = SEARCHING;
@@ -208,7 +208,7 @@
       //myTimer->startTimer();
       transposTable->flush();
       killer1.clear(); killer2.clear();
-      ftime(&startTime);
+      gettimeofday(&startTime, NULL);
       nodeCount = 0;
     }
   if (useOpeningBook && openingBook != NULL && openingBook->valid())
@@ -246,11 +246,11 @@
         principleVariation = iterPV;
       if (displayThinking)
         {
-          struct timeb nowTime;
+          struct timeval nowTime;
           int plyMod = (searchAborted ? -1:0);
-          ftime(&nowTime);
+          gettimeofday(&nowTime, NULL);
           cout << (searchAborted ? "*":"") << (i+plyMod) << "\t" << result << "\t" << nodeCount << "\t"
-               << (100*(nowTime.time - startTime.time)) + ((nowTime.millitm - startTime.millitm)/10) << "\t"
+               << (100*(nowTime.tv_sec - startTime.tv_sec)) + ((nowTime.tv_usec - startTime.tv_usec)/10000) << "\t"
                << variationText(principleVariation) << endl;
           cout.flush();
         }
Only in tsito-0.8.4/src: Engine.cpp~
