diff -Naur gtk-doc-0.9.old/gtkdoc-mkhtml.in gtk-doc-0.9.new/gtkdoc-mkhtml.in
--- gtk-doc-0.9.old/gtkdoc-mkhtml.in	Wed Nov 28 11:53:24 2001
+++ gtk-doc-0.9.new/gtkdoc-mkhtml.in	Tue Dec 10 00:56:23 2002
@@ -24,7 +24,11 @@
       rm -f index.sgml
 fi
 
-jade -t sgml -w no-idref -d $gtkdocdir/gtk-doc.dsl $gtkdocdir/gtk-doc.dcl $2
+openjade -t sgml -w no-idref \
+	-c $prefix/share/sgml/openjade-1.3.2/dsssl/catalog \
+	-c $prefix/share/sgml/dtd/docbook/catalog \
+	-c $prefix/share/sgml/dsssl/docbook-dsssl-nwalsh/catalog \
+	-d $gtkdocdir/gtk-doc.dsl $gtkdocdir/gtk-doc.dcl $2
 sed s%href=\"%href=\"$module/% < index.sgml > index.sgml.tmp && mv index.sgml.tmp index.sgml
 
 echo "timestamp" > ../html.stamp
