diff -Nurd findlib-1.1.2pl1.orig/Makefile findlib-1.1.2pl1/Makefile
--- findlib-1.1.2pl1.orig/Makefile	2006-01-17 08:04:50.000000000 +0900
+++ findlib-1.1.2pl1/Makefile	2007-06-28 18:34:50.000000000 +0900
@@ -22,7 +22,7 @@
 	$(MAKE) install-meta
 	cd src/findlib; $(MAKE) install-num-top
 	$(MAKE) install-config
-	cp tools/safe_camlp4 $(OCAMLFIND_BIN)
+	cp tools/safe_camlp4 $(prefix)$(OCAMLFIND_BIN)
 
 uninstall:
 	$(MAKE) uninstall-doc
