--- NaN/src/Makefile	2012-02-10 04:59:46.000000000 -0500
+++ NaN.patched/src/Makefile	2012-02-12 16:57:39.000000000 -0500
@@ -28,10 +28,10 @@
 # Instead of building "mex shortpath.c" and "mex uigetpath.c", I used empty m-functions within argout=argin; 
 ####################################################
 
-CC  	= gcc
-CXX 	= g++
+CC  	= @FINKPREFIX@/bin/gcc-fsf-@GCC@
+CXX 	= @FINKPREFIX@/bin/g++-fsf-@GCC@
 CFLAGS 	= -fopenmp -Wall -Wextra -Wconversion -O2 -fPIC 
-OCTMEX	= mkoctfile$(OCTAVE_VERSION) --mex
+OCTMEX	= mkoctfile-@OCTVERSION@-gcc@GCC@ --mex
 RM      = rm
 
 MEX_OPTION  += -lgomp CC\#$(CXX) CXX\#$(CXX) CFLAGS\#"$(CFLAGS)" CXXFLAGS\#"$(CFLAGS)"
@@ -79,11 +79,11 @@
 
 train.$(MEX_EXT) predict.$(MEX_EXT): train.c linear.h tron.o linear.o linear_model_matlab.c
 	$(CC) $(CFLAGS) -I $(MATLABDIR)/extern/include -c linear_model_matlab.c
-	$(MATMEX)  -lblas train.c tron.o linear.o linear_model_matlab.o 
+	$(MATMEX)  @BLAS_LIBS@ train.c tron.o linear.o linear_model_matlab.o 
 	#$(MATMEX) -lblas predict.c tron.o linear.o linear_model_matlab.o
 
 train.mex predict.mex: train.c linear.h tron.o linear.o linear_model_matlab.c
-	env CC=$(CXX) $(OCTMEX) -lblas train.c tron.o linear.o linear_model_matlab.c
+	env CC=$(CXX) $(OCTMEX) @BLAS_LIBS@ train.c tron.o linear.o linear_model_matlab.c
 	#env CC=$(CXX) $(OCTMEX) -lblas predict.c tron.o linear.o linear_model_matlab.o
 
 linear.o: linear.cpp linear.h
