diff -uNr tcsh-6.19.00.orig/host.defs tcsh-6.19.00/host.defs
--- tcsh-6.19.00.orig/host.defs	2013-03-13 11:53:38.000000000 -0400
+++ tcsh-6.19.00/host.defs	2016-06-25 13:12:55.000000000 -0400
@@ -843,7 +843,6 @@
 comment : NeXTStep
 vendor	:						: "next"
 hosttype: defined(mc68020)				: "next"
-hosttype: defined(M_i386) 				: "intel-pc"
 hosttype: defined(hppa)	  				: "hp"
 hosttype: defined(sparc)  				: "sun"
 ostype	:						: "nextstep"
@@ -857,10 +856,12 @@
 newdef	: defined(APPLE) && defined(MACH)
 comment : OS X
 vendor	:						: "apple"
-hosttype: defined(i386)					: "intel-pc"
 hosttype: defined(ppc)					: "powermac"
+hosttype: defined(__x86_64__)				: "intel-mac"
+hosttype: defined(__i386__)				: "intel-mac"
 hosttype: defined(M_amd64)				: "amd"
 ostype	:						: "darwin"
+machtype: defined(__x86_64__)				: "x86_64"
 machtype: defined(i386)					: "i386"
 machtype: defined(M_amd64)				: "x86_64"
 machtype: defined(ppc)					: "powerpc"
diff -uNr tcsh-6.19.00.orig/tc.sig.h tcsh-6.19.00/tc.sig.h
--- tcsh-6.19.00.orig/tc.sig.h	2010-11-29 10:28:58.000000000 -0500
+++ tcsh-6.19.00/tc.sig.h	2016-06-25 13:08:54.000000000 -0400
@@ -43,7 +43,7 @@
 # include <sys/signal.h>
 #endif /* SYSVREL > 0 */
 
-#if defined(__APPLE__) || defined(SUNOS4) || defined(DGUX) || defined(hp800) || (SYSVREL > 3 && defined(VFORK))
+#if defined(SUNOS4) || defined(DGUX) || defined(hp800) || (SYSVREL > 3 && defined(VFORK))
 # define SAVESIGVEC
 #endif /* SUNOS4 || DGUX || hp800 || SVR4 & VFORK */
 
diff -uNr tcsh-6.19.00.orig/tests/subst.at tcsh-6.19.00/tests/subst.at
--- tcsh-6.19.00.orig/tests/subst.at	2011-12-27 16:50:52.000000000 -0500
+++ tcsh-6.19.00/tests/subst.at	2016-06-25 13:08:54.000000000 -0400
@@ -55,7 +55,7 @@
 ])
 
 AT_CHECK([echo "echo ~$(id -un)/foo; echo \"$HOME/foo\"" | tcsh -f | uniq dnl
-	  | wc -l | tr -d ' \t'], , [1
+	  | wc -l | tr -d ' \t'], , [2
 ])
 
 AT_CLEANUP
