diff -Nurd zsh-5.9.orig/configure zsh-5.9/configure
--- zsh-5.9.orig/configure	2022-05-14 15:02:47.000000000 -0400
+++ zsh-5.9/configure	2023-07-01 20:57:59.000000000 -0400
@@ -14102,7 +14102,7 @@
     sysv4*|esix*) DLLDFLAGS="${DLLDFLAGS=-G $ldflags}" ;;
     aix*)         DLLDFLAGS="${DLLDFLAGS=-G -bexpall -lc}" ;;
     solaris*|sysv4*|esix*) DLLDFLAGS="${DLLDFLAGS=-G}" ;;
-    darwin*)      DLLDFLAGS="${DLLDFLAGS=-bundle -flat_namespace -undefined suppress}" ;;
+    darwin*)      DLLDFLAGS="${DLLDFLAGS=-bundle -twolevel_namespace -undefined dynamic_lookup}" ;;
     beos*|haiku*) DLLDFLAGS="${DLLDFLAGS=-nostart}" ;;
     openbsd*)
       if test x$zsh_cv_sys_elf = xyes; then
