diff -Nurd -x'*~' kaptain-0.72.orig/doc/kaptain.info kaptain-0.72/doc/kaptain.info
--- kaptain-0.72.orig/doc/kaptain.info	2004-09-11 10:07:54.000000000 -0400
+++ kaptain-0.72/doc/kaptain.info	2010-08-01 13:51:21.000000000 -0400
@@ -5,6 +5,12 @@
 
    Copyright (C) 2000-2004 Zsolt Tere'k
 
+INFO-DIR-SECTION Emacs
+
+START-INFO-DIR-ENTRY
+* Kaptain: (kaptain).           Universal graphical front-end
+END-INFO-DIR-ENTRY
+
 
 File: kaptain.info,  Node: Top,  Next: The Concepts of Kaptain,  Prev: (dir),  Up: (dir)
 
diff -Nurd -x'*~' kaptain-0.72.orig/kaptain/intermediate.h kaptain-0.72/kaptain/intermediate.h
--- kaptain-0.72.orig/kaptain/intermediate.h	2004-09-11 10:07:54.000000000 -0400
+++ kaptain-0.72/kaptain/intermediate.h	2010-08-01 13:52:14.000000000 -0400
@@ -7,6 +7,7 @@
 using namespace std;
 
 class Grammar;
+class Kaptain;
 class Rule;
 class Conjunction;
 class Parameter;
