diff -ruN cdrdao-1.1.7/Makefile.in cdrdao-1.1.7.new/Makefile.in
--- cdrdao-1.1.7/Makefile.in	Sun Oct  6 14:53:55 2002
+++ cdrdao-1.1.7.new/Makefile.in	Tue Oct 29 11:27:01 2002
@@ -41,8 +41,8 @@
 #
 #
 #
-SUBDIRS = @pccts_target@ @scsilib_dir@ trackdb paranoia dao utils @xdao@
-SUBDIRSINST = trackdb dao utils @xdao@
+SUBDIRS = @pccts_target@ @scsilib_dir@ trackdb paranoia dao utils
+SUBDIRSINST = trackdb dao utils
 
 all:
 	for d in $(SUBDIRS) ; do  \
diff -ruN cdrdao-1.1.7/dao/Makefile.in cdrdao-1.1.7.new/dao/Makefile.in
--- cdrdao-1.1.7/dao/Makefile.in	Sun Oct  6 14:53:56 2002
+++ cdrdao-1.1.7.new/dao/Makefile.in	Tue Oct 29 11:27:01 2002
@@ -67,14 +67,14 @@
 all: $(PRG)
 
 install: all installdirs
-	$(INSTALL_PROGRAM) $(PRG) $(bindir)
-	$(INSTALL_DATA) cdrdao.man $(man1dir)/cdrdao.1
-	$(INSTALL_DATA)	cdrdao.drivers $(DRIVER_TABLE_FILE)
+	$(INSTALL_PROGRAM) $(PRG) $(DESTDIR)$(bindir)
+	$(INSTALL_DATA) cdrdao.man $(DESTDIR)$(man1dir)/cdrdao.1
+	$(INSTALL_DATA)	cdrdao.drivers $(DESTDIR)$(DRIVER_TABLE_FILE)
 
 installdirs:
-	$(srcdir)/../mkinstalldirs $(bindir)
-	$(srcdir)/../mkinstalldirs $(man1dir)
-	$(srcdir)/../mkinstalldirs $(CDRDAO_DATA_DIR)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(bindir)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(man1dir)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(CDRDAO_DATA_DIR)
 
 clean:
 	rm -f *.o $(PRG)
diff -ruN cdrdao-1.1.7/dao/main.cc cdrdao-1.1.7.new/dao/main.cc
--- cdrdao-1.1.7/dao/main.cc	Sun Oct  6 14:53:56 2002
+++ cdrdao-1.1.7.new/dao/main.cc	Tue Oct 29 11:30:12 2002
@@ -105,6 +105,11 @@
 static const char *SCSI_DEVICE = "/dev/cdrecorder";
 static int FIFO_BUFFERS = 32;
 
+#elif defined(__APPLE__) && defined(__GNUC__)
+
+static const char *SCSI_DEVICE = "IOCompactDiscServices";
+static int FIFO_BUFFERS = 32;
+
 #else
 
 static const char *SCSI_DEVICE = "0,0,0";
diff -ruN cdrdao-1.1.7/utils/Makefile.in cdrdao-1.1.7.new/utils/Makefile.in
--- cdrdao-1.1.7/utils/Makefile.in	Sun Oct  6 14:53:56 2002
+++ cdrdao-1.1.7.new/utils/Makefile.in	Tue Oct 29 11:27:01 2002
@@ -56,9 +56,9 @@
 	$(INSTALL_PROGRAM) @utils_targets@ $(bindir)
 
 installdirs:
-	$(srcdir)/../mkinstalldirs $(bindir)
-#	$(srcdir)/../mkinstalldirs $(man1dir)
-#	$(srcdir)/../mkinstalldirs $(CDRDAO_DATA_DIR)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(bindir)
+#	$(srcdir)/../mkinstalldirs $(DESTDIR)$(man1dir)
+#	$(srcdir)/../mkinstalldirs $(DESTDIR)$(CDRDAO_DATA_DIR)
 
 clean:
 	rm -f *.o toc2cue toc2mp3
diff -ruN cdrdao-1.1.7/xdao/CdDevice.cc cdrdao-1.1.7.new/xdao/CdDevice.cc
--- cdrdao-1.1.7/xdao/CdDevice.cc	Sun Oct  6 14:53:56 2002
+++ cdrdao-1.1.7.new/xdao/CdDevice.cc	Tue Oct 29 11:27:01 2002
@@ -22,7 +22,11 @@
 #include <stddef.h>
 #include <stdio.h>
 #include <unistd.h>
+
+#ifdef HAVE_MALLOC_H
 #include <malloc.h>
+#endif
+
 #include <errno.h>
 #include <ctype.h>
 #include <assert.h>
diff -ruN cdrdao-1.1.7/xdao/Makefile.in cdrdao-1.1.7.new/xdao/Makefile.in
--- cdrdao-1.1.7/xdao/Makefile.in	Sun Oct  6 14:53:56 2002
+++ cdrdao-1.1.7.new/xdao/Makefile.in	Tue Oct 29 11:27:01 2002
@@ -71,32 +71,32 @@
 all: $(PRG)
 
 install: all installdirs
-	$(INSTALL_PROGRAM) $(PRG) $(bindir)
-	$(INSTALL_DATA) xcdrdao.man $(man1dir)/xcdrdao.1
-	$(INSTALL_DATA) gcdmaster.png $(PIXMAPS_DIR)/gcdmaster.png
-	$(INSTALL_DATA) pixmap_audiocd.png $(PIXMAPS_DIR)/pixmap_audiocd.png
-	$(INSTALL_DATA) pixmap_copycd.png $(PIXMAPS_DIR)/pixmap_copycd.png
-	$(INSTALL_DATA) pixmap_datacd.png $(PIXMAPS_DIR)/pixmap_datacd.png
-	$(INSTALL_DATA) pixmap_dumpcd.png $(PIXMAPS_DIR)/pixmap_dumpcd.png
-	$(INSTALL_DATA) pixmap_help.png $(PIXMAPS_DIR)/pixmap_help.png
-	$(INSTALL_DATA) pixmap_mixedcd.png $(PIXMAPS_DIR)/pixmap_mixedcd.png
-	$(INSTALL_DATA) pixmap_open.png $(PIXMAPS_DIR)/pixmap_open.png
-	$(INSTALL_DATA) pixmap_play-start.xpm $(PIXMAPS_DIR)/pixmap_play-start.xpm
-	$(INSTALL_DATA) pixmap_play-pause.xpm $(PIXMAPS_DIR)/pixmap_play-pause.xpm
-	$(INSTALL_DATA) pixmap_play-stop.xpm $(PIXMAPS_DIR)/pixmap_play-stop.xpm
-	$(INSTALL_DATA) pixmap_cursor-tool.xpm $(PIXMAPS_DIR)/pixmap_cursor-tool.xpm
-	$(INSTALL_DATA) pixmap_zoom-tool.xpm $(PIXMAPS_DIR)/pixmap_zoom-tool.xpm
-	$(INSTALL_DATA) pixmap_zoom-in.xpm $(PIXMAPS_DIR)/pixmap_zoom-in.xpm
-	$(INSTALL_DATA) pixmap_zoom-out.xpm $(PIXMAPS_DIR)/pixmap_zoom-out.xpm
-	$(INSTALL_DATA) pixmap_zoom-selection.xpm $(PIXMAPS_DIR)/pixmap_zoom-selection.xpm
-	$(INSTALL_DATA) pixmap_zoom-fit.xpm $(PIXMAPS_DIR)/pixmap_zoom-fit.xpm
-	$(INSTALL_DATA) gcdmaster.desktop $(datadir)/gnome/apps/Applications/gcdmaster.desktop
+	$(INSTALL_PROGRAM) $(PRG) $(DESTDIR)$(bindir)
+	$(INSTALL_DATA) xcdrdao.man $(DESTDIR)$(man1dir)/xcdrdao.1
+	$(INSTALL_DATA) gcdmaster.png $(DESTDIR)$(PIXMAPS_DIR)/gcdmaster.png
+	$(INSTALL_DATA) pixmap_audiocd.png $(DESTDIR)$(PIXMAPS_DIR)/pixmap_audiocd.png
+	$(INSTALL_DATA) pixmap_copycd.png $(DESTDIR)$(PIXMAPS_DIR)/pixmap_copycd.png
+	$(INSTALL_DATA) pixmap_datacd.png $(DESTDIR)$(PIXMAPS_DIR)/pixmap_datacd.png
+	$(INSTALL_DATA) pixmap_dumpcd.png $(DESTDIR)$(PIXMAPS_DIR)/pixmap_dumpcd.png
+	$(INSTALL_DATA) pixmap_help.png $(DESTDIR)$(PIXMAPS_DIR)/pixmap_help.png
+	$(INSTALL_DATA) pixmap_mixedcd.png $(DESTDIR)$(PIXMAPS_DIR)/pixmap_mixedcd.png
+	$(INSTALL_DATA) pixmap_open.png $(DESTDIR)$(PIXMAPS_DIR)/pixmap_open.png
+	$(INSTALL_DATA) pixmap_play-start.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_play-start.xpm
+	$(INSTALL_DATA) pixmap_play-pause.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_play-pause.xpm
+	$(INSTALL_DATA) pixmap_play-stop.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_play-stop.xpm
+	$(INSTALL_DATA) pixmap_cursor-tool.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_cursor-tool.xpm
+	$(INSTALL_DATA) pixmap_zoom-tool.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_zoom-tool.xpm
+	$(INSTALL_DATA) pixmap_zoom-in.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_zoom-in.xpm
+	$(INSTALL_DATA) pixmap_zoom-out.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_zoom-out.xpm
+	$(INSTALL_DATA) pixmap_zoom-selection.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_zoom-selection.xpm
+	$(INSTALL_DATA) pixmap_zoom-fit.xpm $(DESTDIR)$(PIXMAPS_DIR)/pixmap_zoom-fit.xpm
+	$(INSTALL_DATA) gcdmaster.desktop $(DESTDIR)$(datadir)/gnome/apps/Applications/gcdmaster.desktop
 
 installdirs:
-	$(srcdir)/../mkinstalldirs $(bindir)
-	$(srcdir)/../mkinstalldirs $(man1dir)
-	$(srcdir)/../mkinstalldirs $(PIXMAPS_DIR)
-	$(srcdir)/../mkinstalldirs $(datadir)/gnome/apps/Applications
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(bindir)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(man1dir)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(PIXMAPS_DIR)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(datadir)/gnome/apps/Applications
 
 uninstall:
 	rm -f $(bindir)/$(PRG) $(man1dir)/xcdrdao.1
