diff -u efax-0.9a-001114/fax efax-0.9a-001114-patched/fax
--- efax-0.9a-001114/fax	Mon Oct  2 07:06:08 2000
+++ efax-0.9a-001114-patched/fax	Sun Mar 24 01:02:48 2002
@@ -28,7 +28,7 @@
 # /dev/modem is a link to /dev/cua1, then getty, uucp, kermit,
 # pppd, dip, etc. must *all* use either /dev/modem or /dev/cua1.
 
-DEV=cua1
+DEV=cu.modem
 
 # Your fax number in international format, 20 characters maximum.
 # Use only digits, spaces, and the "+" character.
@@ -81,12 +81,12 @@
 # conventions. Protect with single quotes for delayed evaluation.
 # Add a leading '#' to the file name to use binary format.
 
-LOCK='-x /var/lock/LCK..$DEV' 			# modern systems
+#LOCK='-x /var/lock/LCK..$DEV' 			# modern systems
 # LOCK='-x /usr/spool/uucp/LCK..$DEV'		# older systems
 # LOCK='-x /var/lock/LCK..$DEV -x /var/spool/uucp/LCK..$DEV' # both
 # LOCK='-x #/usr/spool/uucp/LCK..$DEV'		# binary format
 # LOCK='-x /usr/spool/locks/LK.047.040.011'	# SysV style names
-# LOCK=''					# no lock file
+LOCK=''					# no lock file
 
 # Uncomment one of the following lines to force xon/xoff flow
 # control only if you have one of the types of modems listed.
@@ -277,7 +277,7 @@
 # (e.g. because the current user isn't privileged).  Comment this
 # out if it causes problems.
 
-NICE="nice -n -10"
+#NICE="nice -n -10"
 
 # Standard versions of commands that are often aliased.
 
