diff -ruN eroaster-2.0.12/Makefile.in eroaster-2.0.12.new/Makefile.in
--- eroaster-2.0.12/Makefile.in	Thu Oct 25 13:02:36 2001
+++ eroaster-2.0.12.new/Makefile.in	Wed Aug  7 22:17:07 2002
@@ -14,31 +14,31 @@
 
 install:
 	@echo Installing eroaster executables in $(bindir) ...
-	@$(INSTALL) -d $(bindir)
-	@$(INSTALL) -m 755 eroaster $(bindir)
-	@$(INSTALL) -m 755 eroaster-applet $(bindir)
-	@echo Installing pixmaps in $(pixmapdir) ...
-	@$(INSTALL) -d $(pixmapdir)
-	@$(INSTALL) -m 644 pixmaps/*.xpm $(pixmapdir)
-	@$(INSTALL) -d $(deskpixdir)
-	@$(INSTALL) -m 644 pixmaps/applet.xpm $(deskpixdir)/eroaster.xpm
-	@echo Installing supplementary files in $(libdir) ...
-	@$(INSTALL) -d $(libdir)
-	@$(INSTALL) -m 755 ecat.py $(libdir)
-	@$(INSTALL) -m 755 waitreload.py $(libdir)
-	@$(INSTALL) -m 644 constants.py $(libdir)
-	@$(INSTALL) -m 644 main.py $(libdir)
-	@$(INSTALL) -m 644 tools.py $(libdir)
-	@$(INSTALL) -m 644 mp3info.py $(libdir)
-	@$(INSTALL) -m 644 gnometools.py $(libdir)
-	@$(INSTALL) -m 644 log4py.py $(libdir)
-	@$(INSTALL) -m 644 eroaster.shortcuts $(libdir)
-	@echo Installing gnorba file in $(gnorbadir) ...
-	@$(INSTALL) -d $(gnorbadir)
-	@$(INSTALL) -m 644 eroaster.gnorba $(gnorbadir)
-	@echo Installing applet file in $(appletdir) ...
-	@$(INSTALL) -d $(appletdir)
-	@$(INSTALL) -m 644 eroaster.desktop $(appletdir)
+	@$(INSTALL) -d $(DESTDIR)$(bindir)
+	@$(INSTALL) -m 755 eroaster $(DESTDIR)$(bindir)
+	@$(INSTALL) -m 755 eroaster-applet $(DESTDIR)$(bindir)
+	@echo Installing pixmaps in $(DESTDIR)$(pixmapdir) ...
+	@$(INSTALL) -d $(DESTDIR)$(pixmapdir)
+	@$(INSTALL) -m 644 pixmaps/*.xpm $(DESTDIR)$(pixmapdir)
+	@$(INSTALL) -d $(DESTDIR)$(deskpixdir)
+	@$(INSTALL) -m 644 pixmaps/applet.xpm $(DESTDIR)$(deskpixdir)/eroaster.xpm
+	@echo Installing supplementary files in $(DESTDIR)$(libdir) ...
+	@$(INSTALL) -d $(DESTDIR)$(libdir)
+	@$(INSTALL) -m 755 ecat.py $(DESTDIR)$(libdir)
+	@$(INSTALL) -m 755 waitreload.py $(DESTDIR)$(libdir)
+	@$(INSTALL) -m 644 constants.py $(DESTDIR)$(libdir)
+	@$(INSTALL) -m 644 main.py $(DESTDIR)$(libdir)
+	@$(INSTALL) -m 644 tools.py $(DESTDIR)$(libdir)
+	@$(INSTALL) -m 644 mp3info.py $(DESTDIR)$(libdir)
+	@$(INSTALL) -m 644 gnometools.py $(DESTDIR)$(libdir)
+	@$(INSTALL) -m 644 log4py.py $(DESTDIR)$(libdir)
+	@$(INSTALL) -m 644 eroaster.shortcuts $(DESTDIR)$(libdir)
+	@echo Installing gnorba file in $(DESTDIR)$(gnorbadir) ...
+	@$(INSTALL) -d $(DESTDIR)$(gnorbadir)
+	@$(INSTALL) -m 644 eroaster.gnorba $(DESTDIR)$(gnorbadir)
+	@echo Installing applet file in $(DESTDIR)$(appletdir) ...
+	@$(INSTALL) -d $(DESTDIR)$(appletdir)
+	@$(INSTALL) -m 644 eroaster.desktop $(DESTDIR)$(appletdir)
 	@echo Compiling supplementary files ...
 	@$(PYTHON) pycompile
 	@echo Finished ...
