diff -c --recursive source-highlight-1.6-orig/configure.in source-highlight-1.6/configure.in
*** source-highlight-1.6-orig/configure.in	Sun Oct 13 15:38:51 2002
--- source-highlight-1.6/configure.in	Fri Oct 18 13:04:46 2002
***************
*** 112,117 ****
--- 112,118 ----
  AC_CHECK_FUNCS(strdup)
  
  dnl check for getopt in standard library
+ AC_CHECK_LIB(gnugetopt, getopt_long)
  AC_CHECK_FUNCS(getopt_long , , [AC_LIBOBJ([getopt])]) 
  AC_CHECK_FUNCS(getopt_long , , [AC_LIBOBJ([getopt1])]) 
  
