diff -ruN vobcopy-0.5.7.orig/Makefile vobcopy-0.5.7/Makefile
--- vobcopy-0.5.7.orig/Makefile	Fri Mar 14 13:02:27 2003
+++ vobcopy-0.5.7/Makefile	Sun Mar 16 19:51:28 2003
@@ -6,12 +6,12 @@
 #PREFIX += /usr/local
 #BINDIR = ${PREFIX}/bin
 #MANDIR = ${PREFIX}/man
-PREFIX += /usr/local
+PREFIX += @FINKPREFIX@
 BINDIR = ${PREFIX}/bin
-MANDIR = ${PREFIX}/man
+MANDIR = ${PREFIX}/share/man
 LFS    = -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
-CFLAGS += -I/usr/include/
-LDFLAGS += -ldvdread -L/usr/local/lib
+CFLAGS += -I@FINKPREFIX@/include
+LDFLAGS += -L@FINKPREFIX@/lib -ldvdread -lgnugetopt -ldl
 
 #This specifies the conversion from .c to .o 
 .c.o:
@@ -39,9 +39,10 @@
 	rm vobcopy.o dvd.o
 
 install:
-	mkdir -p $(MANDIR)/man1
-	cp vobcopy   $(BINDIR)/vobcopy
-	cp vobcopy.1 $(MANDIR)/man1/vobcopy.1
+	install -d -m 755 $(DESTDIR)/$(BINDIR)
+	install -d -m 755 $(DESTDIR)/$(MANDIR)/man1
+	install -m 755 vobcopy $(DESTDIR)/$(BINDIR)/vobcopy
+	install -m 644 vobcopy.1 $(DESTDIR)/$(MANDIR)/man1/vobcopy.1
 
 uninstall:
 	rm -f $(BINDIR)/vobcopy
diff -ruN vobcopy-0.5.7.orig/dvd.c vobcopy-0.5.7/dvd.c
--- vobcopy-0.5.7.orig/dvd.c	Fri Mar 14 13:02:28 2003
+++ vobcopy-0.5.7/dvd.c	Sun Mar 16 20:30:57 2003
@@ -23,7 +23,7 @@
 #include <unistd.h>
 #include <ctype.h>
 #include <sys/stat.h>
-#if (defined(__unix__) || defined(unix)) && !defined(USG)
+#if ((defined(__unix__) || defined(unix)) && !defined(USG)) || (defined(__APPLE__) && defined(__GNUC__))
 #include <sys/param.h>  
 #include <sys/mount.h>
 #endif
@@ -122,7 +122,7 @@
   char  *pointer;
   char  *k;
   bool  mounted = FALSE;
-#if (defined(BSD) && (BSD >= 199306))
+#if ((defined(BSD) && (BSD >= 199306))) || (defined(__APPLE__) && defined(__GNUC__))
   struct statfs buf;
 #endif
 
@@ -150,7 +150,7 @@
     /*
      *look through /etc/mtab to see if it's actually mounted
      */
-#if (defined(BSD) && (BSD >= 199306)) 
+#if ((defined(BSD) && (BSD >= 199306))) || (defined(__APPLE__) && defined(__GNUC__))
     if(!statfs(path,&buf))
       {
        if(!strcmp(path,buf.f_mntonname))
@@ -325,7 +325,7 @@
 /* returns <0 if error                            */
 int get_device_oyo( char *path, char *device )
 { //oyo
-#if (defined(BSD) && (BSD >= 199306))
+#if ((defined(BSD) && (BSD >= 199306))) || (defined(__APPLE__) && defined(__GNUC__))
   int i, n, cd_count = 0;
   struct statfs *mntbuf;
 
diff -ruN vobcopy-0.5.7.orig/vobcopy.c vobcopy-0.5.7/vobcopy.c
--- vobcopy-0.5.7.orig/vobcopy.c	Fri Mar 14 13:02:28 2003
+++ vobcopy-0.5.7/vobcopy.c	Sun Mar 16 20:01:07 2003
@@ -63,7 +63,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <dirent.h> /*for readdir*/
-#if (defined(__unix__) || defined(unix)) && !defined(USG)
+#if ((defined(__unix__) || defined(unix)) && !defined(USG)) || (defined(__APPLE__) && defined(__GNUC__))
 #include <sys/param.h>
 #else
 #include <sys/vfs.h>
@@ -84,6 +84,10 @@
 #ifdef HAVE_GETOPT_LONG
 #include <getopt.h>
 #endif
+#if defined(__APPLE__) && defined(__GNUC__)
+#define HAVE_GETOPT_LONG
+#include <gnugetopt/getopt.h>
+#endif /* Fink on Darwin */
 
 #include "dvd.h"
 #include "vobcopy.h"
@@ -134,7 +138,7 @@
   /**
    * This case here has to be examined for every system vobcopy shall run under
    */
-#if defined( __linux__ ) || ( defined( BSD ) && ( BSD >= 199306 ))
+#if defined( __linux__ ) || ( defined( BSD ) && ( BSD >= 199306 )) || (defined (__APPLE__) && defined(__GNUC__))
   struct statfs     buf1;
 #elif !defined(__sun__)
   struct statvfs    buf1;
@@ -340,7 +344,7 @@
 	  provided_input_dir_flag = TRUE;
 	  break;
 
-#if defined( __USE_FILE_OFFSET64 ) || ( defined( BSD ) && ( BSD >= 199306 ) )
+#if defined( __USE_FILE_OFFSET64 ) || ( defined( BSD ) && ( BSD >= 199306 ) ) || (defined(__APPLE__) && defined(__GNUC__))
 	  case'l': /*large file output*/
 	    max_filesize_in_blocks = 4500000000000000; 
 	  /* 2^63 / 2048 (not exactly) */
@@ -1437,7 +1441,7 @@
 off_t freespace_getter( char *path, int verbosity_level )
 {
 
-#if defined( __linux__ ) || ( defined( BSD ) && ( BSD >= 199306 ) )
+#if defined( __linux__ ) || ( defined( BSD ) && ( BSD >= 199306 ) ) || (defined(__APPLE__) && defined(__GNUC__))
   struct statfs     buf1;
 #else 
   struct statvfs    buf1; 
@@ -1445,7 +1449,7 @@
 /*   ssize_t temp1, temp2; */
   long temp1, temp2;
   off_t sum;
-#if defined( __linux__ ) || ( defined( BSD ) && ( BSD >= 199306 ) )
+#if defined( __linux__ ) || ( defined( BSD ) && ( BSD >= 199306 ) ) || (defined(__APPLE__) && defined(__GNUC__))
   statfs( path, &buf1 );
   if( verbosity_level >= 1 )
     fprintf( stderr, "Used the linux statfs\n" );
@@ -1474,7 +1478,7 @@
 off_t usedspace_getter( char *path, int verbosity_level )
 {
 
-#if defined( __linux__ ) || ( defined( BSD ) && ( BSD >= 199306 ) )
+#if defined( __linux__ ) || ( defined( BSD ) && ( BSD >= 199306 ) ) || (defined(__APPLE__) && defined(__GNUC__))
   struct statfs     buf2;
 #else 
   struct statvfs    buf2; 
@@ -1482,7 +1486,7 @@
 /*   ssize_t temp1, temp2; */
   long temp1, temp2;
   off_t sum;
-#if defined( __linux__ ) || ( defined( BSD ) && ( BSD >= 199306 ) )
+#if defined( __linux__ ) || ( defined( BSD ) && ( BSD >= 199306 ) ) || (defined(__APPLE__) && defined(__GNUC__))
   statfs( path, &buf2 );
   if( verbosity_level >= 1 )
     fprintf( stderr, "Used the linux statfs\n" );
@@ -1541,7 +1545,7 @@
 	   "[-1/path/to/second/output/dir/] [-2/.../third/..] [-3/../] [-4 /../]\n"
 	   "[-b <skip-size-at-beginning[bkmg]>] [-e <skip-size-at-end[bkmg]>]\n" 
            "[-t <name (can be \"stdout\")>] [-m (mirror)] [-F (fast-switch)]\n", program_name );
-#if defined( __USE_FILE_OFFSET64 ) || ( defined( BSD ) && ( BSD >= 199306 ) )
+#if defined( __USE_FILE_OFFSET64 ) || ( defined( BSD ) && ( BSD >= 199306 ) ) || (defined(__APPLE__) && defined(__GNUC__))
   fprintf( stderr, "[-l (large-file support for files > 2GB)] \n" );
 #endif
   exit( 1 );
diff -ruN vobcopy-0.5.7.orig/vobcopy.h vobcopy-0.5.7/vobcopy.h
--- vobcopy-0.5.7.orig/vobcopy.h	Fri Mar 14 13:02:29 2003
+++ vobcopy-0.5.7/vobcopy.h	Sun Mar 16 20:01:57 2003
@@ -4,7 +4,14 @@
 #define MAX_STRING  81
 #define MAX_DIFFER  2000
 
+#if defined(__APPLE__) && defined(__GNUC__)
+typedef int bool;
+
+#define FALSE 0
+#define TRUE 1
+#else
 typedef enum  { FALSE=0, TRUE=1 }  bool;
+#endif /* Darwin */
 
 void usage(char *);
 int end_slash_adder( char * );
