diff -Nurd -x'*~' elinks-0.11.6.orig/Makefile.config.in elinks-0.11.6/Makefile.config.in
--- elinks-0.11.6.orig/Makefile.config.in	2009-03-21 08:31:54.000000000 -0400
+++ elinks-0.11.6/Makefile.config.in	2009-03-22 16:32:18.000000000 -0400
@@ -160,7 +160,7 @@
 LIBS = @LIBS@
 
 
-INCLUDES = -I$(top_builddir) -I$(top_srcdir)/src
+INCLUDES = -I$(top_builddir) -I$(top_srcdir)/src -I$(top_srcdir)/src/intl/gettext
 
 COMPILE = $(CC) $(DEFS) $(INCLUDES) $(CPPFLAGS) $(CFLAGS)
 
diff -Nurd -x'*~' elinks-0.11.6.orig/configure elinks-0.11.6/configure
--- elinks-0.11.6.orig/configure	2009-03-21 08:50:25.000000000 -0400
+++ elinks-0.11.6/configure	2009-03-22 16:30:21.000000000 -0400
@@ -11479,6 +11479,7 @@
 echo "${ECHO_T}$HAVE_VARIADIC_MACROS" >&6; }
 
 
+if false; then
 { echo "$as_me:$LINENO: checking for -rdynamic" >&5
 echo $ECHO_N "checking for -rdynamic... $ECHO_C" >&6; }
 LDFLAGS_X="$LDFLAGS"
@@ -11529,6 +11530,7 @@
 test "$have_rdynamic" = no && LDFLAGS="$LDFLAGS_X"
 { echo "$as_me:$LINENO: result: $have_rdynamic" >&5
 echo "${ECHO_T}$have_rdynamic" >&6; }
+fi
 
 
 
