diff -uNr jakarta-tomcat-4.1.31/tomcat jakarta-tomcat-4.1.31-new/tomcat
--- jakarta-tomcat-4.1.31/tomcat	1969-12-31 19:00:00.000000000 -0500
+++ jakarta-tomcat-4.1.31-new/tomcat	2006-06-29 21:36:20.000000000 -0400
@@ -0,0 +1,58 @@
+#!/bin/sh
+# wrapper script for tomcat startup
+
+if [ `id -u -n` != 'tomcat' ]; then
+	chown -R tomcat:tomcat @PREFIX@/var/tomcat4/logs >/dev/null 2>&1 || :
+	sudo -u tomcat $0 "$@"
+	exit $?
+fi
+
+for file in @PREFIX@/etc/tomcat4/*.conf; do source "$file"; done
+
+chown -R tomcat:tomcat "$CATALINA_HOME"/work "$CATALINA_HOME"/logs >/dev/null 2>&1
+
+tcproccount() {
+	ps -auxwww | grep TOMCATLAUNCH | grep -v grep | wc -l
+}
+
+case "$1" in
+	start)
+		/bin/echo "Starting Tomcat: \c"
+		$CATALINA_HOME/bin/catalina.sh start >>@PREFIX@/var/tomcat4/logs/tomcat.log 2>&1
+		/bin/echo "ok"
+		;;
+
+	stop)
+		$CATALINA_HOME/bin/catalina.sh stop >>@PREFIX@/var/tomcat4/logs/tomcat.log 2>&1
+		;;
+
+	restart)
+		$0 stop
+
+		# wait 20 seconds and if tomcat hasn't shut down, error out
+		TCDONE=0
+		for i in 1 2 3 4 5 6 7 8 9 10; do
+			if [ "`tcproccount`" -ge 1 ]; then
+				sleep 2
+			else
+				TCDONE=1
+				break
+			fi
+		done
+		if [ "$TCDONE" = "1" ]; then
+			$0 start
+		else
+			/bin/echo ""
+			/bin/echo "Tomcat is not shutting down properly.  Please check"
+			/bin/echo "your logs and try again."
+			/bin/echo ""
+			exit 1
+		fi
+		;;
+
+	*)
+		/bin/echo "Usage: $0 {start|stop|restart}"
+		exit 1
+		;;
+
+esac
diff -uNr jakarta-tomcat-4.1.31/tomcat.conf jakarta-tomcat-4.1.31-new/tomcat.conf
--- jakarta-tomcat-4.1.31/tomcat.conf	1969-12-31 19:00:00.000000000 -0500
+++ jakarta-tomcat-4.1.31-new/tomcat.conf	2006-06-29 21:36:20.000000000 -0400
@@ -0,0 +1,11 @@
+# location of your JDK
+JAVA_HOME=/Library/Java/Home            export JAVA_HOME
+JSSE_HOME=$JAVA_HOME/lib/ext            export JSSE_HOME
+
+# location of Tomcat
+CATALINA_HOME=@PREFIX@/var/tomcat4      export CATALINA_HOME
+
+# any extra options to pass to tomcat startup
+# feel free to add to this, but please don't remove the TOMCATLAUNCH,
+# it is used for finding tomcat processes
+CATALINA_OPTS="-DTOMCATLAUNCH=true"     export CATALINA_OPTS
diff -uNr jakarta-tomcat-4.1.31/webapps/admin.xml jakarta-tomcat-4.1.31-new/webapps/admin.xml
--- jakarta-tomcat-4.1.31/webapps/admin.xml	2004-09-25 10:19:15.000000000 -0400
+++ jakarta-tomcat-4.1.31-new/webapps/admin.xml	2006-06-29 21:36:40.000000000 -0400
@@ -7,7 +7,7 @@
 -->
 
 
-<Context path="/admin" docBase="../server/webapps/admin"
+<Context path="/admin" docBase="@PREFIX@/var/tomcat4/server/webapps/admin"
         debug="0" privileged="true">
 
   <!-- Uncomment this Valve to limit access to the Admin app to localhost
diff -uNr jakarta-tomcat-4.1.31/webapps/manager.xml jakarta-tomcat-4.1.31-new/webapps/manager.xml
--- jakarta-tomcat-4.1.31/webapps/manager.xml	2004-09-25 10:19:11.000000000 -0400
+++ jakarta-tomcat-4.1.31-new/webapps/manager.xml	2006-06-29 21:36:59.000000000 -0400
@@ -7,7 +7,7 @@
 -->
 
 
-<Context path="/manager" docBase="../server/webapps/manager"
+<Context path="/manager" docBase="@PREFIX@/var/tomcat4/server/webapps/manager"
         debug="0" privileged="true">
 
   <!-- Link to the user database we will get roles from -->
