diff -ru ddd-3.3.12/configure ddd-3.3.12-patched/configure
--- ddd-3.3.12/configure	2009-02-11 18:25:52.000000000 +0100
+++ ddd-3.3.12-patched/configure	2009-10-22 23:44:00.000000000 +0200
@@ -4275,7 +4275,7 @@
 rm -f conftest$ac_cv_exeext
 { echo "$as_me:$LINENO: result: $ac_cv_exeext" >&5
 echo "${ECHO_T}$ac_cv_exeext" >&6; }
-
+ac_cv_exeext=".exe"
 rm -f conftest.$ac_ext
 EXEEXT=$ac_cv_exeext
 ac_exeext=$EXEEXT
diff -ru ddd-3.3.12/ddd/Makefile.in ddd-3.3.12-patched/ddd/Makefile.in
--- ddd-3.3.12/ddd/Makefile.in	2009-02-11 18:25:55.000000000 +0100
+++ ddd-3.3.12-patched/ddd/Makefile.in	2009-10-22 23:41:51.000000000 +0200
@@ -2693,7 +2693,7 @@
 
 $(srcdir)/ddd.info.txt.gz.C: $(srcdir)/ddd.info.txt.gz
 	$(MAKE) stringify$(EXEEXT)
-	-./stringify < $? > $@~ && $(MV) $@~ $@
+	-./stringify$(EXEEXT) < $? > $@~ && $(MV) $@~ $@
 
 # License.
 $(srcdir)/COPYING.gz: $(srcdir)/../COPYING
@@ -2701,7 +2701,7 @@
 
 $(srcdir)/COPYING.gz.C: $(srcdir)/COPYING.gz
 	$(MAKE) stringify$(EXEEXT)
-	-./stringify < $? > $@~ && $(MV) $@~ $@
+	-./stringify$(EXEEXT) < $? > $@~ && $(MV) $@~ $@
 
 # News.
 $(srcdir)/NEWS.gz: $(srcdir)/../NEWS
@@ -2709,7 +2709,7 @@
 
 $(srcdir)/NEWS.gz.C: $(srcdir)/NEWS.gz
 	$(MAKE) stringify$(EXEEXT)
-	-./stringify < $? > $@~ && $(MV) $@~ $@
+	-./stringify$(EXEEXT) < $? > $@~ && $(MV) $@~ $@
 
 show.$(OBJEXT): COPYING.gz.C NEWS.gz.C ddd.info.txt.gz.C
 
