--- ppl-0.10.2/demos/ppl_lpsol/examples/ex1.mps.orig        2009-04-17 22:03:19.000000000 -0700
+++ ppl-0.10.2/demos/ppl_lpsol/examples/ex1.mps     2011-04-04 13:03:32.000000000 -0700
@@ -16,4 +16,6 @@
     RHS       C1        5
     RHS       r_000002  5
 BOUNDS
+ PL FOO       x1
+ PL FOO       x2
 ENDATA
--- ppl-0.10.2/demos/ppl_lpsol/examples/unboundedmin.mps.orig       2009-04-17 22:03:19.000000000 -0700
+++ ppl-0.10.2/demos/ppl_lpsol/examples/unboundedmin.mps    2011-04-04 13:03:32.000000000 -0700
@@ -16,4 +16,6 @@
     RHS       C1        0
     RHS       r_000002  0
 BOUNDS
+ PL FOO       x1
+ PL FOO       x2
 ENDATA
