diff -Nur subversion-1.6.0.orig/build/ltmain.sh subversion-1.6.0/build/ltmain.sh
--- subversion-1.6.0.orig/build/ltmain.sh	2009-03-18 00:51:42.000000000 -0400
+++ subversion-1.6.0/build/ltmain.sh	2009-03-22 22:16:07.000000000 -0400
@@ -2924,100 +2924,72 @@
 	    fi
 	    tmp_libs="$tmp_libs $deplib"
 	  done
-
-	  if test "$link_all_deplibs" != no; then
-	    # Add the search paths of all dependency libraries
-	    for deplib in $dependency_libs; do
-	      case $deplib in
-	      -L*) path="$deplib" ;;
-	      *.la)
-		dir=`$echo "X$deplib" | $Xsed -e 's%/[^/]*$%%'`
-		test "X$dir" = "X$deplib" && dir="."
-		# We need an absolute path.
-		case $dir in
-		[\\/]* | [A-Za-z]:[\\/]*) absdir="$dir" ;;
-		*)
-		  absdir=`cd "$dir" && pwd`
-		  if test -z "$absdir"; then
-		    $echo "$modename: warning: cannot determine absolute directory name of \`$dir'" 1>&2
-		    absdir="$dir"
-		  fi
-		  ;;
-		esac
-		if grep "^installed=no" $deplib > /dev/null; then
-		  path="$absdir/$objdir"
-		else
-		  eval libdir=`${SED} -n -e 's/^libdir=\(.*\)$/\1/p' $deplib`
-		  if test -z "$libdir"; then
-		    $echo "$modename: \`$deplib' is not a valid libtool archive" 1>&2
-		    exit $EXIT_FAILURE
-		  fi
-		  if test "$absdir" != "$libdir"; then
-		    $echo "$modename: warning: \`$deplib' seems to be moved" 1>&2
-		  fi
-		  path="$absdir"
-		fi
-		depdepl=
-		case $host in
-		*-*-darwin*)
-		  # we do not want to link against static libs,
-		  # but need to link against shared
-		  eval deplibrary_names=`${SED} -n -e 's/^library_names=\(.*\)$/\1/p' $deplib`
-		  eval deplibdir=`${SED} -n -e 's/^libdir=\(.*\)$/\1/p' $deplib`
-		  if test -n "$deplibrary_names" ; then
-		    for tmp in $deplibrary_names ; do
-		      depdepl=$tmp
-		    done
-		    if test -f "$deplibdir/$depdepl" ; then
-		      depdepl="$deplibdir/$depdepl"
-	      	    elif test -f "$path/$depdepl" ; then
-		      depdepl="$path/$depdepl"
-		    else
-		      # Can't find it, oh well...
-		      depdepl=
-		    fi
-		    # do not add paths which are already there
-		    case " $newlib_search_path " in
-		    *" $path "*) ;;
-		    *) newlib_search_path="$newlib_search_path $path";;
-		    esac
-		  fi
-		  path=""
-		  ;;
-		*)
-		  path="-L$path"
-		  ;;
-		esac
-		;;
-	      -l*)
-		case $host in
-		*-*-darwin*)
-		  # Again, we only want to link against shared libraries
-		  eval tmp_libs=`$echo "X$deplib" | $Xsed -e "s,^\-l,,"`
-		  for tmp in $newlib_search_path ; do
-		    if test -f "$tmp/lib$tmp_libs.dylib" ; then
-		      eval depdepl="$tmp/lib$tmp_libs.dylib"
-		      break
-		    fi
-		  done
-		  path=""
-		  ;;
-		*) continue ;;
-		esac
-		;;
-	      *) continue ;;
-	      esac
-	      case " $deplibs " in
-	      *" $path "*) ;;
-	      *) deplibs="$path $deplibs" ;;
-	      esac
-	      case " $deplibs " in
-	      *" $depdepl "*) ;;
-	      *) deplibs="$depdepl $deplibs" ;;
-	      esac
-	    done
-	  fi # link_all_deplibs != no
-	fi # linkmode = lib
+          if test "$link_all_deplibs" != no; then
+            # Add the search paths of all dependency libraries
+            for deplib in $dependency_libs; do
+              case $deplib in
+              -L*) path="$deplib" ;;
+              *.la)
+#               func_dirname "$deplib" "" "."
+#               dir="$func_dirname_result"
+                case "$deplib" in
+                  */*) dir="${deplib%/*}" ;;
+                  *  ) dir="." ;;
+                esac
+                # We need an absolute path.
+                case $dir in
+                [\\/]* | [A-Za-z]:[\\/]*) absdir="$dir" ;;
+                *)
+                  absdir=`cd "$dir" && pwd`
+                  if test -z "$absdir"; then
+                    $echo "$modename: warning: cannot determine absolute directory name of \`$dir'" 1>&2
+                    absdir="$dir"
+                  fi
+                  ;;
+                esac
+                if grep "^installed=no" $deplib > /dev/null; then
+                case $host in
+                *-*-darwin*)
+                  depdepl=
+                  eval deplibrary_names=`${SED} -n -e 's/^library_names=\(.*\)$/\1/p' $deplib`
+                  if test -n "$deplibrary_names" ; then
+                    for tmp in $deplibrary_names ; do
+                      depdepl=$tmp
+                    done
+                    if test -f "$absdir/$objdir/$depdepl" ; then
+                      depdepl="$absdir/$objdir/$depdepl"
+                      darwin_install_name=`otool -L $depdepl | awk '{if (NR == 2) {print $1;exit}}'`
+                      if test -z "$darwin_install_name"; then
+                          darwin_install_name=`otool64 -L $depdepl  | awk '{if (NR == 2) {print $1;exit}}'`
+                      fi
+                      compiler_flags="$compiler_flags ${wl}-dylib_file ${wl}${darwin_install_name}:${depdepl}"
+                      linker_flags="$linker_flags -dylib_file ${darwin_install_name}:${depdepl}"
+                      path=
+                    fi
+                  fi
+                  ;;
+                *)
+                  path="-L$absdir/$objdir"
+                  ;;
+                esac
+                else
+                  eval libdir=`${SED} -n -e 's/^libdir=\(.*\)$/\1/p' $deplib`
+                  test -z "$libdir" && \
+                    $echo "\`$deplib' is not a valid libtool archive" 1>&2
+                  test -z "$libdir" && exit 1
+                  test "$absdir" != "$libdir" && \
+                    $echo "\`$deplib' seems to be moved" 1>&2
+                  path="-L$absdir"
+                fi
+                ;;
+              esac
+              case " $deplibs " in
+              *" $path "*) ;;
+              *) deplibs="$path $deplibs" ;;
+              esac
+            done
+          fi # link_all_deplibs != no
+        fi # linkmode = lib
       done # for deplib in $libs
       dependency_libs="$newdependency_libs"
       if test "$pass" = dlpreopen; then
diff -Nur subversion-1.6.0.orig/fink/authz_svn.load subversion-1.6.0/fink/authz_svn.load
--- subversion-1.6.0.orig/fink/authz_svn.load	1969-12-31 19:00:00.000000000 -0500
+++ subversion-1.6.0/fink/authz_svn.load	2009-03-22 22:17:01.000000000 -0400
@@ -0,0 +1 @@
+LoadModule authz_svn_module @FINKPREFIX@/lib/apache2/modules/mod_authz_svn.so
diff -Nur subversion-1.6.0.orig/fink/dav_svn.conf subversion-1.6.0/fink/dav_svn.conf
--- subversion-1.6.0.orig/fink/dav_svn.conf	1969-12-31 19:00:00.000000000 -0500
+++ subversion-1.6.0/fink/dav_svn.conf	2009-03-22 22:17:01.000000000 -0400
@@ -0,0 +1,44 @@
+# dav_svn.conf - Example Subversion/Apache configuration
+#
+# For details and further options see the Apache user manual and
+# the Subversion book.
+
+# <Location URL> ... </Location>
+# URL controls how the repository appears to the outside world.
+# In this example clients access the repository as http://hostname/svn/
+<Location /svn>
+
+  # Uncomment this to enable the repository,
+  # DAV svn
+
+  # Set this to the path to your repository
+  # SVNPath @FINKPREFIX@/var/svn
+
+  # The following allows for basic http authentication.  Basic authentication
+  # should not be considered secure for any particularly rigorous definition of
+  # secure.
+
+  # to create a passwd file
+  # # rm -f @FINKPREFIX@/etc/apache2/dav_svn.passwd 
+  # # htpasswd2 -c @FINKPREFIX@/etc/apache2/dav_svn.passwd dwhedon
+  # New password: 
+  # Re-type new password: 
+  # Adding password for user dwhedon
+  # #
+
+  # Uncomment the following 3 lines to enable Basic Authentication
+  # AuthType Basic
+  # AuthName "Subversion Repository"
+  # AuthUserFile @FINKPREFIX@/etc/apache2/dav_svn.passwd
+
+  # Uncomment the following line to enable Authz Authentication
+  # AuthzSVNAccessFile @FINKPREFIX@/etc/apache2/dav_svn.authz
+
+  # The following three lines allow anonymous read, but make
+  # committers authenticate themselves.
+
+  # <LimitExcept GET PROPFIND OPTIONS REPORT>
+  #   Require valid-user
+  # </LimitExcept> 
+
+</Location>
diff -Nur subversion-1.6.0.orig/fink/dav_svn.load subversion-1.6.0/fink/dav_svn.load
--- subversion-1.6.0.orig/fink/dav_svn.load	1969-12-31 19:00:00.000000000 -0500
+++ subversion-1.6.0/fink/dav_svn.load	2009-03-22 22:17:01.000000000 -0400
@@ -0,0 +1,2 @@
+LoadModule dav_svn_module @FINKPREFIX@/lib/apache2/modules/mod_dav_svn.so
+LoadModule authz_svn_module @FINKPREFIX@/lib/apache2/modules/mod_authz_svn.so
diff -Nur subversion-1.6.0.orig/fink/httpd.conf subversion-1.6.0/fink/httpd.conf
--- subversion-1.6.0.orig/fink/httpd.conf	1969-12-31 19:00:00.000000000 -0500
+++ subversion-1.6.0/fink/httpd.conf	2009-03-22 22:17:01.000000000 -0400
@@ -0,0 +1,35 @@
+ServerRoot "@FINKBUILD@/fink/apache2"
+LockFile @FINKBUILD@/fink/apache2/accept.lock
+PidFile @FINKBUILD@/fink/apache2/apache2.pid
+AcceptMutex flock
+ErrorLog @FINKBUILD@/fink/apache2/error.log
+Listen 127.0.0.1:52080
+ServerName localhost:52080
+TypesConfig @FINKPREFIX@/etc/apache2/mime.types
+AddDefaultCharset UTF-8
+Timeout 300
+KeepAlive On
+MaxKeepAliveRequests 100
+KeepAliveTimeout 15
+LoadModule dav_module @FINKPREFIX@/lib/apache2/modules/mod_dav.so
+LoadModule dav_svn_module @FINKPREFIX@/lib/apache2/modules/mod_dav_svn.so
+LoadModule authz_svn_module @FINKPREFIX@/lib/apache2/modules/mod_authz_svn.so
+<Location /svn-test-work/repositories>
+  DAV svn
+  SVNParentPath @FINKBUILD@/subversion/tests/cmdline/svn-test-work/repositories
+  AuthzSVNAccessFile @FINKBUILD@/subversion/tests/cmdline/svn-test-work/authz
+  AuthType Basic
+  AuthName "Subversion Repository"
+  AuthUserFile @FINKBUILD@/tools/test-scripts/svntest/dav_svn.passwd
+  Require valid-user
+</Location>
+
+<Location /svn-test-work/local_tmp/repos>
+  DAV svn
+  SVNPath @FINKBUILD@/subversion/tests/cmdline/svn-test-work/local_tmp/repos
+  AuthzSVNAccessFile @FINKBUILD@/subversion/tests/cmdline/svn-test-work/authz
+  AuthType Basic
+  AuthName "Subversion Repository"
+  AuthUserFile @FINKBUILD@/tools/test-scripts/svntest/dav_svn.passwd
+  Require valid-user
+</Location>
