--- aalib-1.4.0/src/aax.c	Tue Apr 24 14:00:02 2001
+++ aalib-1.4.0-patched/src/aax.c	Wed Oct 31 01:12:58 2001
@@ -1,4 +1,3 @@
-#include <malloc.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>

--- aalib-1.4.0/src/aaxkbd.c	Tue Apr 24 14:00:02 2001
+++ aalib-1.4.0-patched/src/aaxkbd.c	Wed Oct 31 01:23:56 2001
@@ -1,4 +1,3 @@
-#include <malloc.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>

--- aalib-1.4.0/src/aaxmouse.c	Tue Apr 24 14:00:02 2001
+++ aalib-1.4.0-patched/src/aaxmouse.c	Wed Oct 31 01:24:09 2001
@@ -1,4 +1,3 @@
-#include <malloc.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>

--- aalib-1.4.0/src/aarec.c	Tue Apr 24 14:00:02 2001
+++ aalib-1.4.0-patched/src/aarec.c	Thu Nov  1 23:09:07 2001
@@ -1,5 +1,5 @@
 #include <string.h>
-#include <malloc.h>
+#include <stdlib.h>
 #include "aalib.h"
 #include "aaint.h"
 aa_linkedlist *aa_kbdrecommended = NULL, *aa_mouserecommended = NULL,

--- aalib-1.4.0/src/aalib.c	Tue Apr 24 14:00:02 2001
+++ aalib-1.4.0-patched/src/aalib.c	Thu Nov  1 23:10:55 2001
@@ -1,6 +1,6 @@
 #include <stdio.h>
 #include <string.h>
-#include <malloc.h>
+#include <stdlib.h>
 #include "aalib.h"
 #include "aaint.h"

--- aalib-1.4.0/src/aaedit.c	Tue Apr 24 14:00:02 2001
+++ aalib-1.4.0-patched/src/aaedit.c	Thu Nov  1 23:11:23 2001
@@ -1,6 +1,6 @@
 #include <string.h>
 #include <ctype.h>
-#include <malloc.h>
+#include <stdlib.h>
 #include "aalib.h"
 #include "aaint.h"
 static void aa_editdisplay(struct aa_edit *e)

--- aalib-1.4.0/src/aaregist.c	Tue Apr 24 14:00:02 2001
+++ aalib-1.4.0-patched/src/aaregist.c	Thu Nov  1 23:11:47 2001
@@ -1,4 +1,4 @@
-#include <malloc.h>
+#include <stdlib.h>
 #include "config.h"
 #include "aalib.h"
 #include "aaint.h"

--- aalib-1.4.0/src/aakbdreg.c	Tue Apr 24 14:00:02 2001
+++ aalib-1.4.0-patched/src/aakbdreg.c	Thu Nov  1 23:10:02 2001
@@ -1,4 +1,4 @@
-#include <malloc.h>
+#include <stdlib.h>
 #include "config.h"
 #include "aalib.h"
 #include "aaint.h"

--- aalib-1.4.0/src/aamoureg.c	Tue Apr 24 14:00:02 2001
+++ aalib-1.4.0-patched/src/aamoureg.c	Thu Nov  1 23:12:11 2001
@@ -1,4 +1,4 @@
-#include <malloc.h>
+#include <stdlib.h>
 #include "config.h"
 #include "aalib.h"
 #include "aaint.h"
