diff -Naur Apache-Test-1.27-orig/Makefile.PL Apache-Test-1.27/Makefile.PL
--- Apache-Test-1.27-orig/Makefile.PL	2005-12-21 16:45:14.000000000 +0100
+++ Apache-Test-1.27/Makefile.PL	2005-12-21 16:48:39.000000000 +0100
@@ -44,7 +44,7 @@
 # installs the uninstall code
 if (TOP_LEVEL) {
     require "./install-pl";
-    add_Apache__test_target();
+#    add_Apache__test_target();
 }
 
 Apache::TestConfig::custom_config_file_stub_write();
