diff -Nurd -x'*~' Imager-0.91.orig/CountColor/Makefile.PL Imager-0.91/CountColor/Makefile.PL
--- Imager-0.91.orig/CountColor/Makefile.PL	2011-06-06 07:28:32.000000000 -0400
+++ Imager-0.91/CountColor/Makefile.PL	2012-08-06 20:28:09.000000000 -0400
@@ -8,6 +8,7 @@
    VERSION_FROM => 'CountColor.pm',
    OBJECT => 'CountColor.o',
    INC => '-I..',
+   exists $ENV{IM_DFLAGS} ? (DEFINE => $ENV{IM_DFLAGS}) : (),
   );
 my $MM_ver = eval $ExtUtils::MakeMaker::VERSION;
 if ($MM_ver > 6.06) {
diff -Nurd -x'*~' Imager-0.91.orig/DynTest/Makefile.PL Imager-0.91/DynTest/Makefile.PL
--- Imager-0.91.orig/DynTest/Makefile.PL	2011-06-06 07:28:43.000000000 -0400
+++ Imager-0.91/DynTest/Makefile.PL	2012-08-06 20:20:29.000000000 -0400
@@ -7,7 +7,8 @@
    NAME => 'Imager::Filter::DynTest',
    VERSION_FROM => 'DynTest.pm',
    OBJECT => 'DynTest.o linstretch.o',
-   INC => '-I..'
+   INC => '-I..',
+   exists $ENV{IM_DFLAGS} ? (DEFINE => $ENV{IM_DFLAGS}) : (),
   );
 my $MM_ver = eval $ExtUtils::MakeMaker::VERSION;
 if ($MM_ver > 6.06) {
diff -Nurd -x'*~' Imager-0.91.orig/Flines/Makefile.PL Imager-0.91/Flines/Makefile.PL
--- Imager-0.91.orig/Flines/Makefile.PL	2011-06-06 07:29:19.000000000 -0400
+++ Imager-0.91/Flines/Makefile.PL	2012-08-06 20:16:43.000000000 -0400
@@ -7,7 +7,8 @@
    NAME => 'Imager::Filter::Flines',
    VERSION_FROM => 'Flines.pm',
    OBJECT => 'Flines.o',
-   INC => '-I..'
+   INC => '-I..',
+   exists $ENV{IM_DFLAGS} ? (DEFINE => $ENV{IM_DFLAGS}) : (),
   );
 my $MM_ver = eval $ExtUtils::MakeMaker::VERSION;
 if ($MM_ver > 6.06) {
diff -Nurd -x'*~' Imager-0.91.orig/ICO/Makefile.PL Imager-0.91/ICO/Makefile.PL
--- Imager-0.91.orig/ICO/Makefile.PL	2011-06-06 07:29:50.000000000 -0400
+++ Imager-0.91/ICO/Makefile.PL	2012-08-06 20:24:43.000000000 -0400
@@ -9,6 +9,7 @@
    OBJECT => 'ICO.o msicon.o imicon.o',
    INC => '-I..',
    clean => { FILES => 'testout' },
+   exists $ENV{IM_DFLAGS} ? (DEFINE => $ENV{IM_DFLAGS}) : (),
   );
 my $MM_ver = eval $ExtUtils::MakeMaker::VERSION;
 if ($MM_ver > 6.06) {
diff -Nurd -x'*~' Imager-0.91.orig/Makefile.PL Imager-0.91/Makefile.PL
--- Imager-0.91.orig/Makefile.PL	2012-05-24 03:23:35.000000000 -0400
+++ Imager-0.91/Makefile.PL	2012-08-06 19:48:02.000000000 -0400
@@ -259,12 +259,17 @@
   }
 }
 
+my @want_good = (qw/ FT2 GIF JPEG PNG T1 TIFF /);
+my @want_bad = (qw/ Win32 /);
 print "\n";
 print "Libraries found:\n" if @good;
 print "  $_\n" for @good;
+print "(expected: @want_good)\n";
 print "Libraries *not* found:\n" if @bad;
 print "  $_\n" for @bad;
+print "(expected: @want_bad)\n";
 
+exit 1 unless ("@good" eq "@want_good" and "@bad" eq "@want_bad");
 exit;
 
 
@@ -457,7 +462,7 @@
 
 sub init {
 
-  my @definc = qw(/usr/include);
+  my @definc = qw(@PREFIX/include /usr/include);
   @definc{@definc}=(1) x @definc;
   @incs=
     (
@@ -474,12 +479,8 @@
     push @incs, grep -d, split /\Q$Config{path_sep}/, $Config{incpath};
   }
   push @incs, grep -d,
-      qw(/sw/include 
-         /usr/include/freetype2
-         /usr/local/include/freetype2
-         /usr/local/include/freetype1/freetype
-         /usr/include /usr/local/include /usr/include/freetype
-         /usr/local/include/freetype);
+      qw(@PREFIX@/include @PREFIX@/include/freetype2
+         /usr/include);
   if ($Config{ccflags}) {
     my @hidden = map { /^-I(.*)$/ ? ($1) : () } split ' ', $Config{ccflags};
     push @incs, @hidden;
@@ -511,7 +512,7 @@
     # it can't find and it doesn't look for -L in ldflags
     #@deflib{@hidden} = @hidden;
   }
-  push @libs, grep -d, qw(/usr/local/lib);
+  push @libs, grep -d, qw(@PREFIX@/lib);
 
   $formats{'TT-fonts'}=
     {
diff -Nurd -x'*~' Imager-0.91.orig/Mandelbrot/Makefile.PL Imager-0.91/Mandelbrot/Makefile.PL
--- Imager-0.91.orig/Mandelbrot/Makefile.PL	2011-06-06 07:30:33.000000000 -0400
+++ Imager-0.91/Mandelbrot/Makefile.PL	2012-08-06 20:20:11.000000000 -0400
@@ -7,7 +7,8 @@
    NAME => 'Imager::Filter::Mandelbrot',
    VERSION_FROM => 'Mandelbrot.pm',
    OBJECT => 'Mandelbrot.o mandel.o',
-   INC => '-I..'
+   INC => '-I..',
+   exists $ENV{IM_DFLAGS} ? (DEFINE => $ENV{IM_DFLAGS}) : (),
   );
 my $MM_ver = eval $ExtUtils::MakeMaker::VERSION;
 if ($MM_ver > 6.06) {
diff -Nurd -x'*~' Imager-0.91.orig/SGI/Makefile.PL Imager-0.91/SGI/Makefile.PL
--- Imager-0.91.orig/SGI/Makefile.PL	2011-06-06 07:30:59.000000000 -0400
+++ Imager-0.91/SGI/Makefile.PL	2012-08-06 20:19:42.000000000 -0400
@@ -9,6 +9,7 @@
    OBJECT => 'SGI.o imsgi.o',
    INC => '-I..',
    clean => { FILES => 'testout' },
+   exists $ENV{IM_DFLAGS} ? (DEFINE => $ENV{IM_DFLAGS}) : (),
   );
 my $MM_ver = eval $ExtUtils::MakeMaker::VERSION;
 if ($MM_ver > 6.06) {
diff -Nurd -x'*~' Imager-0.91.orig/dynfilt/Makefile.PL Imager-0.91/dynfilt/Makefile.PL
--- Imager-0.91.orig/dynfilt/Makefile.PL	2011-01-10 03:39:18.000000000 -0500
+++ Imager-0.91/dynfilt/Makefile.PL	2012-08-06 20:15:42.000000000 -0400
@@ -17,6 +17,7 @@
     NAME      => 'Imager::plugins',
     SKIP      => [qw(all dynamic static )],
     clean     => {'FILES' => $libstr},
+    exists $ENV{IM_DFLAGS} ? (DEFINE => $ENV{IM_DFLAGS}) : (),
 );
 
 
diff -Nurd -x'*~' Imager-0.91.orig/lib/Imager/Color.pm Imager-0.91/lib/Imager/Color.pm
--- Imager-0.91.orig/lib/Imager/Color.pm	2011-06-06 07:25:29.000000000 -0400
+++ Imager-0.91/lib/Imager/Color.pm	2012-08-06 19:45:01.000000000 -0400
@@ -157,6 +157,9 @@
 
 my @x_search =
   (
+   '/usr/X11/share/X11/rgb.txt',   # OS X modern xorg
+   '/usr/X11R6/share/X11/rgb.txt', # OS X older/compatibility
+
    '/usr/share/X11/rgb.txt', # newer Xorg X11 dists use this
    '/usr/lib/X11/rgb.txt', # seems fairly standard
    '/usr/local/lib/X11/rgb.txt', # seems possible
diff -Nurd -x'*~' Imager-0.91.orig/lib/Imager/Probe.pm Imager-0.91/lib/Imager/Probe.pm
--- Imager-0.91.orig/lib/Imager/Probe.pm	2012-04-28 21:46:37.000000000 -0400
+++ Imager-0.91/lib/Imager/Probe.pm	2012-08-06 20:10:30.000000000 -0400
@@ -62,6 +62,9 @@
 
   $result or return;
 
+  if (exists $ENV{IM_DFLAGS}) {
+      $result->{DEFINE} = $ENV{IM_DFLAGS} . ' ' . $result->{DEFINE};
+  }
   return $result;
 }
 
@@ -394,8 +397,8 @@
      ),
      $^O eq "MSWin32" ? $ENV{LIB} : "",
      $^O eq "cygwin" ? "/usr/lib/w32api" : "",
+     "@PREFIX@/lib",
      "/usr/lib",
-     "/usr/local/lib",
      _gcc_lib_paths(),
     );
 }
@@ -432,8 +435,8 @@
       grep $_,
       @Config{qw/locincpth incpath/}
      ),
+     "@PREFIX@/include",
      "/usr/include",
-     "/usr/local/include",
     );
 
   if ($req->{incsuffix}) {
diff -Nurd -x'*~' Imager-0.91.orig/t/t1000files.t Imager-0.91/t/t1000files.t
--- Imager-0.91.orig/t/t1000files.t	2012-06-04 05:53:32.000000000 -0400
+++ Imager-0.91/t/t1000files.t	2012-08-06 20:37:22.000000000 -0400
@@ -37,7 +37,6 @@
 PERL
   close SCRIPT;
   my $perl = $^X;
-  $perl = qq/"$perl"/ if $perl =~ / /;
   
   print "# script: $test_script\n";
   my $cmd = "$perl -Mblib $test_script t/t1000files.t";
