diff -Nurd -x'*~' gettext-1.04.orig/Makefile.PL gettext-1.04/Makefile.PL
--- gettext-1.04.orig/Makefile.PL	2005-01-30 18:30:22.000000000 -0500
+++ gettext-1.04/Makefile.PL	2005-04-06 13:53:40.000000000 -0400
@@ -7,11 +7,15 @@
 } else {
 	$cc = $Config{'cc'};
 }
-my $libs = '';
+my $inc = $ENV{'CPPFLAGS'};
+my $libs = $ENV{'LDFLAGS'};
+
+open(CONFIG_LOG, ">config.log") or die "could not write to config.log: $!\n";
+close CONFIG_LOG;
 
 unless (conftest("char *x = gettext(\"foo\");", "gettext", 0)) {
 	# try with -lintl
-	$libs = "-lintl";
+	$libs .= " -lintl";
 	unless (conftest("char *x = gettext(\"foo\");", "gettext", 0)) {
 		unlink("conftest.c");
 		unlink("conftest");
@@ -33,6 +37,7 @@
 
 WriteMakefile(
     NAME => "Locale::gettext",
+    INC => $inc,
     LIBS => ($libs eq '') ? [] : [$libs],
     VERSION_FROM => 'gettext.pm', 
 );
@@ -41,6 +46,7 @@
 	my ($testcode, $func, $record) = @_;
 
 	print "checking for ", $func;
+	print(" with ", $inc) if ($inc ne '');
 	print(" in ", $libs) if ($libs ne '');
 	print "...";
 	open(TEST, ">conftest.c");
@@ -49,18 +55,22 @@
 	print TEST "return 0;}\n";
 	close TEST;
 	open(SAVE, ">&STDERR");
-	open(STDERR, ">/dev/null");
-	system($cc . " -o conftest " . $libs . " conftest.c");
+	open(STDERR, ">>config.log");
+	warn "checking for $func\n";
+	my $cmd = "$cc -o conftest $inc $libs conftest.c";
+	warn "$cmd\n";
+	system $cmd;
 	my $exitstatus = $?;
-	open(STDERR, ">&SAVE");
+	warn "status = $exitstatus\n\n";
 	if ($exitstatus != 0) {
+		warn "program that failed was\n", `cat conftest.c`;
 		print " no\n";
-		return 0;
 	} else {
 		print " yes\n";
 		if ($record) {
 			print CONFIG "#define HAVE_", uc($func), "\n";
 		}
-		return 1;
 	}
+	open(STDERR, ">&SAVE");
+	return ($exitstatus == 0);
 }
