diff -Nurd Module-Install-0.99/t/lib/Test.pm Module-Install-0.99-new/t/lib/Test.pm
--- Module-Install-0.99/t/lib/Test.pm	2010-05-31 19:05:38.000000000 -0400
+++ Module-Install-0.99-new/t/lib/Test.pm	2010-08-09 14:26:21.000000000 -0400
@@ -164,7 +164,7 @@
 	local $ENV{X_MYMETA} = $X_MYMETA;
 
 	my @run_params=@{ $params{run_params} || [] };
-	my $ret = system($^X, "-I../../lib", "-I../../blib/lib", "Makefile.PL",@run_params);
+	my $ret = system(split(/\s+/,$ENV{FULLPERL}), "-I../../lib", "-I../../blib/lib", "Makefile.PL",@run_params);
 	chdir $home or return 0;
 	return $ret ? 0 : 1;
 }
@@ -179,7 +179,7 @@
 	local $ENV{X_MYMETA} = $X_MYMETA;
 
 	my $run_params=join(' ',@{ $params{run_params} || [] });
-	my $ret = system("$^X -I../../lib -I../../blib/lib Makefile.PL $run_params");
+	my $ret = system("$ENV{FULLPERL} -I../../lib -I../../blib/lib Makefile.PL $run_params");
 	#my $result=qx();
 	chdir $home or return 0;
 	return $ret ? 0 : 1;
