diff -Nurd -x'*~' Module-Refresh-0.17.orig/Makefile.PL Module-Refresh-0.17/Makefile.PL
--- Module-Refresh-0.17.orig/Makefile.PL	2011-10-26 08:37:08.000000000 -0400
+++ Module-Refresh-0.17/Makefile.PL	2012-07-16 09:33:59.000000000 -0400
@@ -6,8 +6,6 @@
 version_from('lib/Module/Refresh.pm');
 abstract_from('lib/Module/Refresh.pm');
 build_requires ('Test::More' => 0,
-                'Path::Class' => 0,
-                'File::Temp' => "0.19",
                 );
 sign;
 &WriteAll;
