diff -Nurd -x'*~' Perl-Critic-Pulp-80.orig/lib/Perl/Critic/Policy/Compatibility/ConstantLeadingUnderscore.pm Perl-Critic-Pulp-80/lib/Perl/Critic/Policy/Compatibility/ConstantLeadingUnderscore.pm
--- Perl-Critic-Pulp-80.orig/lib/Perl/Critic/Policy/Compatibility/ConstantLeadingUnderscore.pm	2013-03-28 20:11:50.000000000 -0400
+++ Perl-Critic-Pulp-80/lib/Perl/Critic/Policy/Compatibility/ConstantLeadingUnderscore.pm	2014-04-03 13:27:24.000000000 -0400
@@ -70,7 +70,7 @@
     ($inc->module||'') eq 'constant' || next;
 
     if (my $ver = Perl::Critic::Pulp::Utils::include_module_version ($inc)) {
-      $ver = version->new ($ver);
+      $ver = version->new ($ver->literal);
       if (! defined $modver || $ver > $modver) {
         $modver = $ver;
 
diff -Nurd -x'*~' Perl-Critic-Pulp-80.orig/lib/Perl/Critic/Policy/Compatibility/ConstantPragmaHash.pm Perl-Critic-Pulp-80/lib/Perl/Critic/Policy/Compatibility/ConstantPragmaHash.pm
--- Perl-Critic-Pulp-80.orig/lib/Perl/Critic/Policy/Compatibility/ConstantPragmaHash.pm	2013-03-28 20:11:50.000000000 -0400
+++ Perl-Critic-Pulp-80/lib/Perl/Critic/Policy/Compatibility/ConstantPragmaHash.pm	2014-04-03 13:27:24.000000000 -0400
@@ -66,7 +66,7 @@
     ($inc->module||'') eq 'constant' || next;
 
     if (my $ver = Perl::Critic::Pulp::Utils::include_module_version ($inc)) {
-      $ver = version->new ($ver);
+      $ver = version->new ($ver->literal);
       if (! defined $modver || $ver > $modver) {
         $modver = $ver;
 
