diff -Nurd Perl/Makefile.PL.in Perl.new/Makefile.PL.in 
--- Perl/Makefile.PL.in        2006-03-15 07:44:07.000000000 -0500
+++ Perl.new/Makefile.PL.in    2007-02-22 15:04:44.000000000 -0500
@@ -4,7 +4,7 @@
 
 $prefix = "@prefix@";
 $exec_prefix = "@exec_prefix@";
-$libdir = "../../libpisock/.libs";
+$libdir = "${prefix}/lib";
 
 $lib = "-L$libdir -lpisock";
 
@@ -38,7 +38,7 @@
     'NAME'       => 'PDA::Pilot',
     'VERSION'    => '@PILOT_LINK_VERS@.@PILOT_LINK_MAJOR@.@PILOT_LINK_MINOR@@PILOT_LINK_PATCH@',
     'XSPROTOARG' => '-noprototypes',
-    'INC'        => "-I$plincdir",
+    'INC'        => "-I${prefix}/include",
     'DEFINE'     => '-DPERL_POLLUTE',
     'PREFIX'     => "$prefix",
     'INSTALLDIRS'=> 'vendor',
