--- a/lib/Test/Builder/Tester.pm	2013-10-12 19:04:56.000000000 +0000
+++ b/lib/Test/Builder/Tester.pm	2013-10-21 14:14:35.931170825 +0000
@@ -480,7 +480,7 @@
     my( $self, $check ) = @_;
 
     # Since we ship with Test::Builder, calling a private method is safe...ish.
-    return $t->_indent . $check;
+    return ref($check) ? $check : $t->_indent . $check;
 }
 
 sub _translate_Failed_check {
--- a/t/Tester/tbt_01basic.t	2013-09-27 14:09:25.000000000 +0000
+++ b/t/Tester/tbt_01basic.t	2013-10-21 14:11:35.319278517 +0000
@@ -1,6 +1,6 @@
 #!/usr/bin/perl
 
-use Test::Builder::Tester tests => 9;
+use Test::Builder::Tester tests => 10;
 use Test::More;
 
 ok(1,"This is a basic test");
@@ -21,8 +21,12 @@
 ok(2,"two");
 test_test("multiple tests");
 
+test_out(qr/ok 1 - tested\n/);
+ok(1,"tested");
+test_test("regexp matching");
+
 test_out("not ok 1 - should fail");
-test_err("#     Failed test ($0 at line 28)");
+test_err("#     Failed test ($0 at line 32)");
 test_err("#          got: 'foo'");
 test_err("#     expected: 'bar'");
 is("foo","bar","should fail");
@@ -46,7 +50,7 @@
 
 
 test_out("not ok 1 - name # TODO Something");
-test_out("#     Failed (TODO) test ($0 at line 52)");
+test_out("#     Failed (TODO) test ($0 at line 56)");
 TODO: { 
     local $TODO = "Something";
     fail("name");