diff -Nurd kdeutils-3.5.10/kgpg/Makefile.am kdeutils-3.5.10-new/kgpg/Makefile.am
--- kdeutils-3.5.10/kgpg/Makefile.am	2006-01-19 11:49:15.000000000 -0500
+++ kdeutils-3.5.10-new/kgpg/Makefile.am	2010-07-26 18:53:12.000000000 -0400
@@ -11,7 +11,7 @@
 		keyproperties.ui conf_encryption.ui conf_decryption.ui conf_gpg.ui dcopiface.skel \
 		keyinfowidget.cpp groupedit.ui kgpgrevokewidget.ui newkey.ui adduid.ui 	keyexport.ui \
 		sourceselect.ui kgpgsettings.kcfgc conf_servers.ui conf_ui2.ui conf_misc.ui
-kgpg_LDADD   =  $(LIB_KIO) $(LIB_KABC) -lkutils -lkdeprint
+kgpg_LDADD   =  $(LIB_KIO) $(LIB_KABC) -lkutils -lkdeprint $(LIB_QT) $(LIB_KDEUI) -lDCOP $(LIB_KDECORE)
 
 pics_DATA= kgpg_anim.gif kgpg_docked.png  kgpg_docked.gif kgpg_blank.png  kgpg_fill.png
 picsdir=$(kde_datadir)/kgpg/pics
