diff -Nurd -x'*~' kvpnc-0.9.6a-kde4.orig/CMakeLists.txt kvpnc-0.9.6a-kde4/CMakeLists.txt
--- kvpnc-0.9.6a-kde4.orig/CMakeLists.txt	2009-08-20 15:55:48.000000000 -0400
+++ kvpnc-0.9.6a-kde4/CMakeLists.txt	2012-08-17 04:51:30.000000000 -0400
@@ -13,7 +13,7 @@
 add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
 add_definitions(-DQT3_SUPPORT -DQT3_SUPPORT_WARNINGS)
 
-include_directories(${KDE4_INCLUDES} ${KDE4_INCLUDE_DIR} ${QT4_INCLUDES} ${CMAKE_CURRENT_BINARY_DIR}/src)
+include_directories(${KDE4_INCLUDES} ${KDE4_INCLUDE_DIR} ${QT4_INCLUDES} ${LIBGCRYPT_INCLUDE_DIR} ${CMAKE_CURRENT_BINARY_DIR}/src)
 
 set(KVPNC_VERSION_STRING "0.9.2-svn")
 
