diff -ruN nessus-core.orig/nessus-adduser.in nessus-core/nessus-adduser.in
--- nessus-core.orig/nessus-adduser.in	Wed May 29 03:17:54 2002
+++ nessus-core/nessus-adduser.in	Mon Apr 14 13:45:11 2003
@@ -274,17 +274,17 @@
 
 
 if [ "$auth" = "pass" ]; then
-    if [ "$MD5CMD" ]; then
-    	test -c /dev/urandom &&
-	{
-	URANDOM=`dd if=/dev/urandom bs=16 count=16 2>/dev/null|$MD5CMD`
-	}
-	SEED=`(echo $SEED; date; df; ls -l; echo $URANDOM) | $MD5CMD | awk '{print $1}'`
-	H=`echo $Xn $SEED$password$Xc | $MD5CMD | awk '{print $1}'`
-	echo $H $SEED > "$localstatedir/nessus/users/$login/auth/hash"
-    else
+#    if [ "$MD5CMD" ]; then
+#    	test -c /dev/urandom &&
+#	{
+#	URANDOM=`dd if=/dev/urandom bs=16 count=16 2>/dev/null|$MD5CMD`
+#	}
+#	SEED=`(echo $SEED; date; df; ls -l; echo $URANDOM) | $MD5CMD | awk '{print $1}'`
+#	H=`echo $Xn $SEED$password$Xc | $MD5CMD | awk '{print $1}'`
+#	echo $H $SEED > "$localstatedir/nessus/users/$login/auth/hash"
+#    else
 	echo "$password" > "$localstatedir/nessus/users/$login/auth/password"
-    fi
+#    fi
 elif [ "$auth" = "cert" ]; then
 	echo "$dn" > "$localstatedir/nessus/users/$login/auth/dname"
 fi
diff -ruN nessus-core.orig/nessusd.conf nessus-core/nessusd.conf
--- nessus-core.orig/nessusd.conf	Wed Dec 31 17:00:00 1969
+++ nessus-core/nessusd.conf	Mon Apr 14 13:44:43 2003
@@ -0,0 +1,90 @@
+# Nessus Security Scanner, Fink default configuration file
+#
+# Empty lines and those starting with '#' are ignored.
+
+# Directory where plug-ins are to be found
+plugins_folder = @FINKPREFIX@/lib/nessus/plugins
+
+# E-mail address of the admin
+email = root
+
+# Maximum number of threads, i.e. simultaneous tests
+max_threads = 15
+
+# File used to log activity. Set it to 'syslog' if you want to use syslogd.
+logfile = @FINKPREFIX@/var/log/nessus/nessusd.messages
+
+# Log every detail of the attack
+log_whole_attack = yes
+
+# Log the name of the plugins that are loaded by the server
+log_plugins_name_at_load = no
+
+# Dump file for debugging output, use `-' for stdout
+dumpfile = @FINKPREFIX@/var/lib/nessus/nessusd.dump
+
+# File that contains rules
+rules = @FINKPREFIX@/etc/nessus/nessusd.rules
+
+# Users database file
+users = @FINKPREFIX@/etc/nessus/nessusd.users
+
+# CGI paths to check for (cgi-bin:/cgi-aws:/ can do)
+cgi_path = /cgi-bin
+
+# Optimize the test
+optimize_test = yes
+
+# Read timeout for the sockets of the tests
+checks_read_timeout = 15
+
+# Delay in seconds) to pass for between two tests against the same port
+# (to be inetd friendly)
+delay_between_tests = 1
+
+# Remote file that the plugins will try to read:
+test_file = /etc/passwd
+
+# Range of the ports that nmap will scan
+port_range = 1-15000
+
+# Ping hosts before scanning them?
+ping_hosts = yes
+
+# Only test the IPs that can be reversely looked up?
+reverse_lookup = no
+
+# Host expansion:
+# dns:  performs and AXFR on the remote name server
+#       and test the host obtained
+# nfs:  test hosts that have the right to mount the
+#       filesystems exported by the remote host
+# ip:   scan the entire subnet
+host_expansion = dns;ip
+
+subnet_class = C
+
+# Maximum number of hosts (-1 means unlimited)
+max_hosts = -1
+
+scan_level = normal
+outside_firewall = no
+
+# Language to use in plugins.
+language = english
+
+# Public key client server encryption (crypto options)
+peks_username = nessusd
+peks_keylen = 1024
+peks_keyfile = @FINKPREFIX@/etc/nessus/nessusd.private-keys
+peks_usrkeys = @FINKPREFIX@/etc/nessus/nessusd.user-keys
+peks_pwdfail = 5
+track_iothreads = yes
+cookie_logpipe = @FINKPREFIX@/etc/nessus/nessusd.logpipe
+cookie_logpipe_suptmo = 2
+force_pubkey_auth = yes
+
+# Uncomment the following for IO thread debugging
+#track_iothreads = yes
+
+# End of /etc/nessus/nessusd.conf file.
