--- a/src/channels.cxx	2006-03-21 05:42:25.000000000 -0500
+++ b/src/channels.cxx	2012-03-16 12:15:03.000000000 -0400
@@ -1361,7 +1361,7 @@
 
     filterMutex.Wait();
     for (PINDEX i = 0; i < filters.GetSize(); i++)
-      filters[i](frame, (INT)&sendPacket);
+      filters[i](frame, (long INT)&sendPacket);
     filterMutex.Signal();
 
     if (sendPacket || (silent && frame.GetPayloadSize() > 0)) {
