diff -Naur mozilla.orig/config/add-chrome.pl mozilla/config/add-chrome.pl
--- mozilla.orig/config/add-chrome.pl	2005-06-01 10:20:34.000000000 -0400
+++ mozilla/config/add-chrome.pl	2009-01-05 02:28:30.000000000 -0500
@@ -14,7 +14,7 @@
 my $jarFileName = $ARGV[4];
 
 my $win32 = ($^O =~ /((MS)?win32)|msys|cygwin|os2/i) ? 1 : 0;
-my $macos = ($^O =~ /MacOS|darwin/i) ? 1 : 0;
+my $macos = 0;
 my $unix  = !($win32 || $macos) ? 1 : 0;
 
 sub foreignPlatformFile
diff -Naur mozilla.orig/config/autoconf.mk.in mozilla/config/autoconf.mk.in
--- mozilla.orig/config/autoconf.mk.in	2006-09-14 14:07:03.000000000 -0400
+++ mozilla/config/autoconf.mk.in	2009-01-05 02:28:30.000000000 -0500
@@ -57,14 +57,14 @@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-includedir	= @includedir@/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+includedir	= @includedir@/$(MOZ_APP_NAME)
 libdir		= @libdir@
 datadir		= @datadir@
 mandir		= @mandir@
-idldir		= @datadir@/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+idldir		= @datadir@/idl/$(MOZ_APP_NAME)
 
-mozappdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-mredir		= $(libdir)/mre/mre-$(MOZ_APP_VERSION)
+mozappdir	= $(libdir)/$(MOZ_APP_NAME)
+mredir		= $(libdir)/mre/mre
 mrelibdir	= $(mredir)/lib
 
 DIST		= $(DEPTH)/dist
diff -Naur mozilla.orig/directory/c-sdk/config/config.mk mozilla/directory/c-sdk/config/config.mk
--- mozilla.orig/directory/c-sdk/config/config.mk	2006-02-03 09:41:18.000000000 -0500
+++ mozilla/directory/c-sdk/config/config.mk	2009-01-05 02:28:30.000000000 -0500
@@ -64,10 +64,10 @@
 
 NFSPWD		= $(MOD_DEPTH)/config/nfspwd
 
-CFLAGS		= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(OS_CFLAGS)\
-		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
-CCCFLAGS	= $(CCC_ONLY_FLAGS) $(OPTIMIZER) $(OS_CFLAGS)\
-		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
+CFLAGS		= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(XP_DEFINE)\
+		  $(DEFINES) $(INCLUDES) $(XCFLAGS) $(OS_CFLAGS)
+CCCFLAGS	= $(CCC_ONLY_FLAGS) $(OPTIMIZER) $(XP_DEFINE)\
+		  $(DEFINES) $(INCLUDES) $(XCFLAGS) $(OS_CFLAGS)
 # For purify
 NOMD_CFLAGS	= $(CC_ONLY_FLAGS) $(OPTIMIZER) $(NOMD_OS_CFLAGS)\
 		  $(XP_DEFINE) $(DEFINES) $(INCLUDES) $(XCFLAGS)
diff -Naur mozilla.orig/fink/applications/Seamonkey-address.desktop mozilla/fink/applications/Seamonkey-address.desktop
--- mozilla.orig/fink/applications/Seamonkey-address.desktop	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/fink/applications/Seamonkey-address.desktop	2009-01-05 02:28:30.000000000 -0500
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Seamonkey Addressbook
+Comment=Addressbook from Seamonkey Mailnews
+Exec=seamonkey -addressbook
+Icon=seamonkey-addressbook.xpm
+Terminal=false
+MultipleArgs=false
+Type=Application
+Categories=Application;Network
+StartupNotify=true
diff -Naur mozilla.orig/fink/applications/Seamonkey-calendar.desktop mozilla/fink/applications/Seamonkey-calendar.desktop
--- mozilla.orig/fink/applications/Seamonkey-calendar.desktop	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/fink/applications/Seamonkey-calendar.desktop	2009-01-05 02:28:30.000000000 -0500
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Seamonkey Calendar
+Comment=Seamonkey Calendar
+Exec=seamonkey -calendar
+Icon=seamonkey-calendar.xpm
+Terminal=false
+MultipleArgs=false
+Type=Application
+Categories=Application;Network
+StartupNotify=true
diff -Naur mozilla.orig/fink/applications/Seamonkey-chatzilla.desktop mozilla/fink/applications/Seamonkey-chatzilla.desktop
--- mozilla.orig/fink/applications/Seamonkey-chatzilla.desktop	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/fink/applications/Seamonkey-chatzilla.desktop	2009-01-05 02:28:30.000000000 -0500
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Chatzilla
+Comment=Seamonkey Chatzilla IRC client
+Exec=seamonkey -chat
+Icon=seamonkey-chatzilla.xpm
+Terminal=false
+MultipleArgs=false
+Type=Application
+Categories=Application;Network
+StartupNotify=true
diff -Naur mozilla.orig/fink/applications/Seamonkey-editor.desktop mozilla/fink/applications/Seamonkey-editor.desktop
--- mozilla.orig/fink/applications/Seamonkey-editor.desktop	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/fink/applications/Seamonkey-editor.desktop	2009-01-05 02:28:30.000000000 -0500
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Seamonkey Composer
+Comment=HTML editor from Seamonkey project
+Exec=seamonkey -edit
+Icon=seamonkey-editor.xpm
+Terminal=false
+Type=Application
+Categories=Application;Network
+StartupNotify=true
diff -Naur mozilla.orig/fink/applications/Seamonkey-mail-composer.desktop mozilla/fink/applications/Seamonkey-mail-composer.desktop
--- mozilla.orig/fink/applications/Seamonkey-mail-composer.desktop	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/fink/applications/Seamonkey-mail-composer.desktop	2009-01-05 02:28:30.000000000 -0500
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Seamonkey Mail Composer
+Comment=Seamonkey Mail Composer
+Exec=seamonkey -compose
+Icon=seamonkey-mail-composer.xpm
+Terminal=false
+MultipleArgs=false
+Type=Application
+Categories=Application;Network
+StartupNotify=true
diff -Naur mozilla.orig/fink/applications/Seamonkey-mail.desktop mozilla/fink/applications/Seamonkey-mail.desktop
--- mozilla.orig/fink/applications/Seamonkey-mail.desktop	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/fink/applications/Seamonkey-mail.desktop	2009-01-05 02:28:30.000000000 -0500
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Seamonkey Mail
+Comment=Seamonkey Mail
+Exec=seamonkey -mail
+Icon=seamonkey-mailnews.xpm
+Terminal=false
+MultipleArgs=false
+Type=Application
+Categories=Application;Network
+StartupNotify=true
diff -Naur mozilla.orig/fink/applications/Seamonkey-news.desktop mozilla/fink/applications/Seamonkey-news.desktop
--- mozilla.orig/fink/applications/Seamonkey-news.desktop	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/fink/applications/Seamonkey-news.desktop	2009-01-05 02:28:30.000000000 -0500
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Seamonkey News
+Comment=Seamonkey News
+Exec=seamonkey -news
+Icon=seamonkey-mailnews.xpm
+Terminal=false
+MultipleArgs=false
+Type=Application
+Categories=Application;Network
+StartupNotify=true
diff -Naur mozilla.orig/fink/applications/Seamonkey.desktop mozilla/fink/applications/Seamonkey.desktop
--- mozilla.orig/fink/applications/Seamonkey.desktop	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/fink/applications/Seamonkey.desktop	2009-01-05 02:28:30.000000000 -0500
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Seamonkey Web Browser
+Comment=Browse the World Wide Web
+Exec=seamonkey
+Icon=seamonkey.xpm
+Terminal=false
+Type=Application
+Categories=Application;Network
+StartupNotify=true
diff -Naur mozilla.orig/fink/chrome.d/default mozilla/fink/chrome.d/default
--- mozilla.orig/fink/chrome.d/default	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/fink/chrome.d/default	2009-01-05 02:28:30.000000000 -0500
@@ -0,0 +1,2 @@
+skin,install,select,modern/1.0
+locale,install,select,en-US
diff -Naur mozilla.orig/fink/mozilla mozilla/fink/mozilla
--- mozilla.orig/fink/mozilla	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/fink/mozilla	2009-01-05 02:28:30.000000000 -0500
@@ -0,0 +1,98 @@
+#!/bin/sh
+#
+# The contents of this file are subject to the Netscape Public
+# License Version 1.1 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of
+# the License at http://www.mozilla.org/NPL/
+#
+# Software distributed under the License is distributed on an "AS
+# IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# rights and limitations under the License.
+#
+# The Original Code is mozilla.org code.
+#
+# The Initial Developer of the Original Code is Netscape
+# Communications Corporation.  Portions created by Netscape are
+# Copyright (C) 1998 Netscape Communications Corporation. All
+# Rights Reserved.
+#
+# Contributor(s):
+#
+
+##
+## Usage:
+##
+## $ mozilla
+##
+## This script is meant to run a mozilla program from the mozilla
+## rpm installation.
+##
+## The script will setup all the environment voodoo needed to make
+## mozilla work.
+
+##
+## Variables
+##
+## Set MOZILLA_FIVE_HOME
+##
+MOZILLA_FIVE_HOME="@PREFIX@/lib/seamonkey"
+export MOZILLA_FIVE_HOME
+
+MOZ_PROGRAM="${MOZILLA_FIVE_HOME}/seamonkey-bin"
+
+##
+## Set DYLD_LIBRARY_PATH
+##
+DYLD_LIBRARY_PATH=${MOZILLA_FIVE_HOME}:${MOZILLA_FIVE_HOME}/components:${MOZILLA_FIVE_HOME}/plugins:${MOZILLA_FIVE_HOME}/ipc/modules:$DYLD_LIBRARY_PATH
+export DYLD_LIBRARY_PATH
+
+# If there is no command line argument at all then try to open a new
+# window in an already running instance.
+
+if [ -z "$1" ]; then
+  $MOZ_PROGRAM -remote "openurl(about:blank,new-window)" >/dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "2" ]; then
+    $MOZ_PROGRAM ${1+"$@"}
+    RETURN_VAL=$?
+  fi
+  exit $RETURN_VAL
+fi
+
+# If there's a command line argument but it doesn't begin with a -
+# it's probably a url.  Try to send it to a running instance.
+
+USE_EXIST=0
+opt="$1"
+case "$opt" in
+  -*) ;;
+  *) USE_EXIST=1 ;;
+esac
+
+if [ "${USE_EXIST}" -eq "1" ]; then
+  # check to make sure that the command contains at least a :/ in it.
+  echo $opt | grep -e ':/' 2>/dev/null >/dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "1" ]; then
+    # does it begin with a / ?
+    echo $opt | grep -e '^/' >/dev/null
+    RETURN_VAL=$?
+    if [ "$RETURN_VAL" -eq "0" ]; then
+      opt="file:$opt"
+    elif [ -e `pwd`/$opt ]; then
+      opt="`pwd`/$opt"
+    else
+      opt="http://$opt"
+    fi
+  fi
+  $MOZ_PROGRAM -remote "openurl($opt,new-window)" >/dev/null
+  RETURN_VAL=$?
+  if [ "$RETURN_VAL" -eq "2" ]; then
+    $MOZ_PROGRAM ${1+"$@"}
+    RETURN_VAL=$?
+  fi
+  exit $RETURN_VAL
+fi
+
+$MOZ_PROGRAM ${1+"$@"}
diff -Naur mozilla.orig/fink/prefs.js mozilla/fink/prefs.js
--- mozilla.orig/fink/prefs.js	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/fink/prefs.js	2009-01-05 02:28:30.000000000 -0500
@@ -0,0 +1,23 @@
+// XIM style
+pref("xim.input_style", "over-the-spot");
+
+// TrueType
+pref("font.FreeType2.enable", true);
+pref("font.freetype2.shared-library", "/usr/X11R6/lib/libfreetype.6.dylib");
+pref("font.FreeType2.autohinted", true);
+pref("font.FreeType2.unhinted", true);
+pref("font.antialias.min",        10);
+pref("font.embedded_bitmaps.max", 1000000);
+pref("font.scale.tt_bitmap.dark_text.min", 64);
+pref("font.scale.tt_bitmap.dark_text.gain", "0.0");
+pref("font.directory.truetype.1", "/Library/Fonts");
+pref("font.directory.truetype.2", "/Network/Library/Fonts");
+pref("font.directory.truetype.3", "/System/Library/Fonts");
+pref("font.directory.truetype.4", "/usr/X11R6/lib/X11/fonts/TTF");
+pref("font.directory.truetype.5", "@PREFIX@/share/fonts/truetype");
+
+// AA with Bitmap scaling.
+pref("font.scale.aa_bitmap.enable", true);
+pref("font.scale.aa_bitmap.always", false);
+pref("font.scale.aa_bitmap.min", 6);
+
diff -Naur mozilla.orig/fink/update-seamonkey-chrome mozilla/fink/update-seamonkey-chrome
--- mozilla.orig/fink/update-seamonkey-chrome	1969-12-31 19:00:00.000000000 -0500
+++ mozilla/fink/update-seamonkey-chrome	2009-01-05 02:28:30.000000000 -0500
@@ -0,0 +1,19 @@
+#!/bin/sh -e
+
+ulimit -s unlimited || true
+
+MOZILLA_FIVE_HOME=@PREFIX@/lib/seamonkey
+CHROME_DIR=@PREFIX@/var/lib/seamonkey/chrome.d
+
+export DYLD_LIBRARY_PATH=${MOZILLA_FIVE_HOME}:${MOZILLA_FIVE_HOME}/components:${MOZILLA_FIVE_HOME}/plugins:${MOZILLA_FIVE_HOME}/ipc/modules:$DYLD_LIBRARY_PATH
+cd ${MOZILLA_FIVE_HOME}
+TMPFILE=`tempfile`
+cat ${CHROME_DIR}/* >>${TMPFILE}
+mv ${TMPFILE} chrome/installed-chrome.txt
+chmod 644 chrome/installed-chrome.txt
+
+rm -f chrome/{c,o,s}*.rdf components/*.dat
+sudo -H ./regxpcom -a || true
+sudo -H ./regchrome || true
+
+exit 0
diff -Naur mozilla.orig/modules/libpref/src/nsPrefService.cpp mozilla/modules/libpref/src/nsPrefService.cpp
--- mozilla.orig/modules/libpref/src/nsPrefService.cpp	2007-02-15 21:19:25.000000000 -0500
+++ mozilla/modules/libpref/src/nsPrefService.cpp	2009-01-05 02:28:30.000000000 -0500
@@ -751,6 +751,7 @@
 #elif defined(XP_BEOS)
       "beos.js"
 #endif
+      , "fink.js"
   };
 
   rv = pref_LoadPrefsInDir(defaultPrefDir, specialFiles, NS_ARRAY_LENGTH(specialFiles));
diff -Naur mozilla.orig/nsprpub/configure mozilla/nsprpub/configure
--- mozilla.orig/nsprpub/configure	2008-01-29 14:27:43.000000000 -0500
+++ mozilla/nsprpub/configure	2009-01-05 02:28:30.000000000 -0500
@@ -3338,9 +3338,6 @@
 
     # Add Mac OS X support for loading CFM & CFBundle plugins
     if test -f /System/Library/Frameworks/Carbon.framework/Carbon; then
-        cat >> confdefs.h <<\EOF
-#define XP_MACOSX 1
-EOF
 
         OS_TARGET=MacOSX
 
@@ -4964,8 +4961,6 @@
 
 
 case $target in
-*-darwin*)
-    ;;
 *)
     echo $ac_n "checking for dlopen in -ldl""... $ac_c" 1>&6
 echo "configure:4972: checking for dlopen in -ldl" >&5
diff -Naur mozilla.orig/nsprpub/pr/include/md/_darwin.h mozilla/nsprpub/pr/include/md/_darwin.h
--- mozilla.orig/nsprpub/pr/include/md/_darwin.h	2007-02-15 21:16:06.000000000 -0500
+++ mozilla/nsprpub/pr/include/md/_darwin.h	2009-01-05 02:28:30.000000000 -0500
@@ -62,7 +62,8 @@
 
 #undef  HAVE_STACK_GROWING_UP
 #define HAVE_DLL
-#define USE_MACH_DYLD
+/* #define USE_MACH_DYLD */
+#define USE_DLFCN
 #define _PR_HAVE_SOCKADDR_LEN  
 #define _PR_STAT_HAS_ST_ATIMESPEC
 #define _PR_HAVE_LARGE_OFF_T
diff -Naur mozilla.orig/nsprpub/pr/src/Makefile.in mozilla/nsprpub/pr/src/Makefile.in
--- mozilla.orig/nsprpub/pr/src/Makefile.in	2006-12-22 08:47:21.000000000 -0500
+++ mozilla/nsprpub/pr/src/Makefile.in	2009-01-05 02:28:31.000000000 -0500
@@ -204,9 +204,9 @@
 endif
 endif
 
-ifeq ($(OS_TARGET),MacOSX)
-OS_LIBS		= -framework CoreServices -framework CoreFoundation
-endif
+#ifeq ($(OS_TARGET),MacOSX)
+#OS_LIBS		= -framework CoreServices -framework CoreFoundation
+#endif
 
 ifdef GC_LEAK_DETECTOR
 EXTRA_LIBS	= -L$(dist_libdir) -lboehm
diff -Naur mozilla.orig/nsprpub/pr/src/linking/prlink.c mozilla/nsprpub/pr/src/linking/prlink.c
--- mozilla.orig/nsprpub/pr/src/linking/prlink.c	2006-12-22 08:47:22.000000000 -0500
+++ mozilla/nsprpub/pr/src/linking/prlink.c	2009-01-05 02:28:31.000000000 -0500
@@ -158,7 +158,8 @@
 #if defined(SUNOS4) || defined(DARWIN) || defined(NEXTSTEP) \
     || defined(WIN16) || defined(XP_OS2) \
     || ((defined(OPENBSD) || defined(NETBSD)) && !defined(__ELF__))
-#define NEED_LEADING_UNDERSCORE
+/* On MacOSX 10.3, dlsym does not accept symbols with '_'.
+/* #define NEED_LEADING_UNDERSCORE */
 #endif
 
 #define PR_LD_PATHW 0x8000  /* for PR_LibSpec_PathnameU */
diff -Naur mozilla.orig/themes/Makefile.in mozilla/themes/Makefile.in
--- mozilla.orig/themes/Makefile.in	2005-03-11 14:39:35.000000000 -0500
+++ mozilla/themes/Makefile.in	2009-01-05 02:28:31.000000000 -0500
@@ -50,7 +50,9 @@
 
 # select classic as the default skin
 libs::
-	echo skin,install,select,classic/1.0 >> $(DIST)/bin/chrome/installed-chrome.txt
+# fix later
+#	echo skin,install,select,classic/1.0 >> $(DIST)/bin/chrome/installed-chrome.txt
 
 install::
-	echo skin,install,select,classic/1.0 >> $(DESTDIR)$(mozappdir)/chrome/installed-chrome.txt
+# fix later
+#	echo skin,install,select,classic/1.0 >> $(DESTDIR)$(mozappdir)/chrome/installed-chrome.txt
