--- ruby-1.6.7/error.c.orig	Thu Oct 17 06:45:22 2002
+++ ruby-1.6.7/error.c	Thu Oct 17 06:45:54 2002
@@ -468,7 +468,7 @@
 static VALUE *syserr_list;
 #endif
 
-#if !defined(NT) && !defined(__FreeBSD__) && !defined(__NetBSD__) && !defined(__OpenBSD__) && !defined(sys_nerr)
+#if !defined(NT) && !defined(__FreeBSD__) && !defined(__NetBSD__) && !defined(__OpenBSD__) && !defined(sys_nerr) && !defined(__APPLE__)
 extern int sys_nerr;
 #endif
 
--- ruby-1.6.7/configure.orig	Thu Oct 17 06:51:28 2002
+++ ruby-1.6.7/configure	Thu Oct 17 06:52:13 2002
@@ -2428,7 +2428,7 @@
 #define $ac_tr_lib 1
 EOF
 
-  LIBS="-ldl $LIBS"
+  LIBS="$LDFLAGS -ldl $LIBS"
 
 else
   echo "$ac_t""no" 1>&6
@@ -5054,8 +5054,8 @@
     darwin*)
 	LIBRUBY_SO='lib$(RUBY_INSTALL_NAME).$(MAJOR).$(MINOR).$(TEENY).dylib'
 	LIBRUBY_LDSHARED='cc -dynamiclib -undefined suppress -flat_namespace'
-	LIBRUBY_DLDFLAGS='-install_name lib$(RUBY_INSTALL_NAME).dylib -current_version $(MAJOR).$(MINOR).$(TEENY) -compatibility_version $(MAJOR).$(MINOR)'
-	LIBRUBY_ALIASES='lib$(RUBY_INSTALL_NAME).$(MAJOR).$(MINOR).dylib lib$(RUBY_INSTALL_NAME).dylib'
+	LIBRUBY_DLDFLAGS='-install_name ${prefix}/lib/libruby.1.dylib -current_version $(MAJOR).$(MINOR).$(TEENY) -compatibility_version $(MAJOR).$(MINOR)'
+	LIBRUBY_ALIASES='lib$(RUBY_INSTALL_NAME).$(MAJOR).$(MINOR).dylib lib$(RUBY_INSTALL_NAME).dylib libruby.dylib'
 	;;
     *)
 	;;
@@ -5076,7 +5076,7 @@
     	CFLAGS="$CFLAGS -pipe -no-precomp"
 	;;
     darwin*)
-    	CFLAGS="$CFLAGS -pipe -no-precomp"
+    	CFLAGS="$CFLAGS -pipe -no-cpp-precomp"
 	;;
     os2_emx)
     	CFLAGS="$CFLAGS -DOS2"
--- ruby-1.6.7/ext/extmk.rb.in.orig	Thu Oct 17 12:18:45 2002
+++ ruby-1.6.7/ext/extmk.rb.in	Thu Oct 17 12:19:18 2002
@@ -25,8 +25,8 @@
 SRC_EXT = ["c", "cc", "m", "cxx", "cpp", "C"]
 $extlist = []
 
-$includedir = "@includedir@".gsub(/\$\{prefix\}|\$\(prefix\)/,'@prefix@')
-$libdir = "@libdir@".gsub(/\$\{exec_prefix\}|\$\(exec_prefix\)/,'@exec_prefix@')
+$includedir = "@PREFIX@/include"
+$libdir = "@PREFIX@/lib"
 
 $top_srcdir = "@top_srcdir@"
 if $top_srcdir !~ "^/"
--- ruby-1.6.7/ext/tcltklib/extconf.rb.orig	Thu Oct 17 12:22:00 2002
+++ ruby-1.6.7/ext/tcltklib/extconf.rb	Thu Oct 17 12:35:53 2002
@@ -11,14 +11,14 @@
 
 dir_config("tk")
 dir_config("tcl")
-dir_config("X11")
+dir_config("X11", "/usr/X11R6/include", "/usr/X11R6/lib")
 
 tklib = with_config("tklib")
 tcllib = with_config("tcllib")
 stubs = enable_config("tcltk_stubs") || with_config("tcltk_stubs")
 
 def find_tcl(tcllib, stubs)
-  paths = ["/usr/local/lib", "/usr/pkg/lib", "/usr/lib"]
+  paths = ["@PREFIX@/lib", "/usr/local/lib", "/usr/pkg", "/usr/lib"]
   func = stubs ? "Tcl_InitStubs" : "Tcl_FindExecutable"
   if tcllib
     find_library(tcllib, func, *paths)
@@ -40,7 +40,7 @@
 end
 
 def find_tk(tklib, stubs)
-  paths = ["/usr/local/lib", "/usr/pkg/lib", "/usr/lib"]
+  paths = ["@PREFIX@/lib", "/usr/local/lib", "/usr/pkg", "/usr/lib"]
   func = stubs ? "Tk_InitStubs" : "Tk_Init"
   if tklib
     find_library(tklib, func, *paths)
