diff -ruN deadbeef-1.9.6-orig/configure.ac deadbeef-1.9.6/configure.ac
--- deadbeef-1.9.6-orig/configure.ac	2023-11-07 12:10:00.000000000 -0600
+++ deadbeef-1.9.6/configure.ac	2024-04-16 06:18:13.000000000 -0500
@@ -263,14 +263,14 @@
     ])
 ])
 
-AS_IF([test "${enable_staticlink}" != "no"], [
-    ], [
-    AC_CHECK_LIB([dispatch], [main], [HAVE_DISPATCH=yes])
-])
-DISPATCH_CFLAGS=-fblocks
-DISPATCH_LIBS="-ldispatch -lBlocksRuntime"
-AC_SUBST(DISPATCH_CFLAGS)
-AC_SUBST(DISPATCH_LIBS)
+#AS_IF([test "${enable_staticlink}" != "no"], [
+#    ], [
+#    AC_CHECK_LIB([dispatch], [main], [HAVE_DISPATCH=yes])
+#])
+#DISPATCH_CFLAGS=-fblocks
+#DISPATCH_LIBS="-ldispatch -lBlocksRuntime"
+#AC_SUBST(DISPATCH_CFLAGS)
+#AC_SUBST(DISPATCH_LIBS)
 
 AS_IF([test "${enable_gtk2}" = "yes"], [
     AS_IF([test "${enable_staticlink}" != "no"], [
diff -ruN deadbeef-1.9.6-orig/plugins/converter/Makefile.am deadbeef-1.9.6/plugins/converter/Makefile.am
--- deadbeef-1.9.6-orig/plugins/converter/Makefile.am	2023-11-07 12:10:00.000000000 -0600
+++ deadbeef-1.9.6/plugins/converter/Makefile.am	2024-04-16 20:51:19.000000000 -0500
@@ -37,7 +37,8 @@
 convdatadir = $(libdir)/deadbeef/convpresets
 convdata_DATA = $(convdata)
 
-converter_la_CFLAGS =  $(CFLAGS) -I@top_srcdir@/external/mp4p/include -I@top_srcdir@/include -std=c99 -fPIC -DUSE_TAGGING=1
+DEFAULT_INCLUDES = -I@top_srcdir@/include -I@top_srcdir@
+converter_la_CFLAGS =  $(CFLAGS) -I@top_srcdir@/external/mp4p/include -std=c99 -fPIC -DUSE_TAGGING=1
 converter_la_SOURCES = converter.c converter.h
 converter_la_LDFLAGS = -module -avoid-version
 converter_la_LIBADD = $(LDADD) ../../shared/libmp4tagutil.la ../../external/libmp4p.la
diff -ruN deadbeef-1.9.6-orig/plugins/ffmpeg/Makefile.am deadbeef-1.9.6/plugins/ffmpeg/Makefile.am
--- deadbeef-1.9.6-orig/plugins/ffmpeg/Makefile.am	2023-11-07 12:10:00.000000000 -0600
+++ deadbeef-1.9.6/plugins/ffmpeg/Makefile.am	2024-04-16 19:59:44.000000000 -0500
@@ -4,5 +4,6 @@
 ffmpeg_la_LDFLAGS = -module -avoid-version
 
 ffmpeg_la_LIBADD = $(LDADD) $(FFMPEG_DEPS_LIBS)
-ffmpeg_la_CFLAGS = $(CFLAGS) -std=c99 ${FFMPEG_DEPS_CFLAGS} -I@top_srcdir@/include
+ffmpeg_la_CFLAGS = $(CFLAGS) -std=c99 ${FFMPEG_DEPS_CFLAGS}
+DEFAULT_INCLUDES = -I@top_srcdir@/include
 endif
diff -ruN deadbeef-1.9.6-orig/src/Makefile.am deadbeef-1.9.6/src/Makefile.am
--- deadbeef-1.9.6-orig/src/Makefile.am	2023-11-07 12:10:00.000000000 -0600
+++ deadbeef-1.9.6/src/Makefile.am	2024-04-17 05:16:24.000000000 -0500
@@ -35,10 +35,16 @@
 	u8_uc_map.h\
 	vfs.c vfs.h vfs_stdio.c\
 	volume.c volume.h\
-	viz.c viz.h
+	viz.c viz.h\
+	../plugins/coreaudio/coreaudio.c ../plugins/nullout/nullout.c\
+	cocoautil.m cocoautil.h
 
-deadbeef_LDADD = $(LDADD) $(ICONV_LIB) $(DL_LIBS) -lm -lpthread $(DISPATCH_LIBS) $(LTLIBINTL) ../shared/libctmap.la ../plugins/libparser/libparser.la
+deadbeef_LDADD = $(LDADD) $(ICONV_LIB) $(DL_LIBS) -lm -lpthread $(DISPATCH_LIBS) $(LTLIBINTL) $(COREAUDIO_LIBS) ../shared/libctmap.la ../plugins/libparser/libparser.la
+deadbeef_LDFLAGS = -Wl,-framework,IOKit -Wl,-framework,Foundation
 
-deadbeef_CFLAGS = $(DEPS_CFLAGS) $(DISPATCH_CFLAGS) -std=c99 -DLOCALEDIR=\"@localedir@\" -I@top_srcdir@/include -I@top_srcdir@
+deadbeef_CFLAGS = $(DEPS_CFLAGS) $(DISPATCH_CFLAGS) -std=c99 -DLOCALEDIR=\"@localedir@\"
+
+DEFAULT_INCLUDES = -I@top_srcdir@/include -I@top_srcdir@
+deadbeef-streamer.$(OBJEXT) : CPPFLAGS += -I@top_srcdir@/plugins/coreaudio
 
 EXTRA_DIST = sj_to_unicode.h 
diff -ruN deadbeef-1.9.6-orig/src/junklib.c deadbeef-1.9.6/src/junklib.c
--- deadbeef-1.9.6-orig/src/junklib.c	2023-11-07 12:10:00.000000000 -0600
+++ deadbeef-1.9.6/src/junklib.c	2024-04-16 06:18:59.000000000 -0500
@@ -32,7 +32,7 @@
 #include <stdlib.h>
 #include <string.h>
 #if HAVE_ICONV
-  #if !defined(__MINGW32__) && !defined(__OpenBSD__)
+  #if !defined(__MINGW32__) && !defined(__OpenBSD__) && !defined(__APPLE__)
   #define LIBICONV_PLUG
   #endif
   #include <iconv.h>
