diff -ruN jack-audio-connection-kit-0.125.0-orig/config/os/macosx/pThreadUtilities.h jack-audio-connection-kit-0.125.0/config/os/macosx/pThreadUtilities.h
--- jack-audio-connection-kit-0.125.0-orig/config/os/macosx/pThreadUtilities.h	2016-09-14 13:40:49.000000000 -0500
+++ jack-audio-connection-kit-0.125.0/config/os/macosx/pThreadUtilities.h	2019-04-25 05:06:16.000000000 -0500
@@ -79,7 +79,7 @@
 #if defined(MAC_OS_X_VERSION_10_7) && MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_7
 #import <MacTypes.h>
 #else
-#import <CoreServices/../Frameworks/CarbonCore.framework/Headers/MacTypes.h>
+#import <MacTypes.h>
 #endif
 
 static inline UInt32
diff -ruN jack-audio-connection-kit-0.125.0-orig/configure.ac jack-audio-connection-kit-0.125.0/configure.ac
--- jack-audio-connection-kit-0.125.0-orig/configure.ac	2016-09-14 17:00:41.000000000 -0500
+++ jack-audio-connection-kit-0.125.0/configure.ac	2019-04-25 05:17:29.000000000 -0500
@@ -185,7 +185,7 @@
     JACK_CPP_VARARGS_BROKEN=1
     JACK_DO_NOT_MLOCK=1
     JACK_USE_MACH_THREADS=1
-    OS_LDFLAGS="-framework CoreAudio -framework CoreServices -framework AudioUnit"
+    OS_LDFLAGS="-Wl,-framework,CoreAudio -Wl,-framework,CoreServices -Wl,-framework,AudioUnit"
     TRY_POSIX_SHM=yes			# POSIX shm works better
     HOST_DEFAULT_TMP_DIR=/tmp
     ;;
@@ -350,11 +350,6 @@
 
 JACK_CORE_CFLAGS="$JACK_CORE_CFLAGS $JACK_LIBC_HELPER_FLAGS"
 
-case $build_os in
-     # we need weak linkage which appeared in 10.2, but lets ask for 10.4 anyway
-     darwin*) JACK_CORE_CFLAGS="$JACK_CORE_CFLAGS -mmacosx-version-min=10.4" ;;
-esac
-
 AC_ARG_WITH(cpu-target,
             [  --with-cpu-target=cpu-type    explicit, overriding argument for gcc -march= flag])
 
