diff -Nurd libao-0.8.8.orig/configure.ac libao-0.8.8/configure.ac
--- libao-0.8.8.orig/configure.ac	2007-05-24 06:51:05.000000000 -0400
+++ libao-0.8.8/configure.ac	2025-08-25 20:47:45.000000000 -0400
@@ -85,9 +85,9 @@
                 PROFILE="-pg -g -O20 -D__NO_MATH_INLINES -fsigned-char" ;;
         *-darwin*)
                 PLUGIN_LDFLAGS="-module -avoid-version"
-                DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char -Ddlsym=dlsym_auto_underscore"
-                CFLAGS="-D__NO_MATH_INLINES -fsigned-char -Ddlsym=dlsym_auto_underscore"
-                PROFILE="-g -pg -D__NO_MATH_INLINES -fsigned-char -Ddlsym=dlsym_auto_underscore" ;;
+                DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char"
+                CFLAGS="-D__NO_MATH_INLINES -fsigned-char"
+                PROFILE="-g -pg -D__NO_MATH_INLINES -fsigned-char" ;;
         *)
                 PLUGIN_LDFLAGS="-export-dynamic -avoid-version"
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char"
@@ -138,7 +138,7 @@
 	DLOPEN_FLAG='(RTLD_LAZY)'
 	SHARED_LIB_EXT='.sl'
 	;;    
-    *openbsd* | *netbsd* | *solaris2.7)
+    *darwin* | *openbsd* | *netbsd* | *solaris2.7)
 	DLOPEN_FLAG='(RTLD_LAZY)'
 	SHARED_LIB_EXT='.so'
 	;;
diff -Nurd libao-0.8.8.orig/doc/Makefile.am libao-0.8.8/doc/Makefile.am
--- libao-0.8.8.orig/doc/Makefile.am	2007-05-24 05:19:08.000000000 -0400
+++ libao-0.8.8/doc/Makefile.am	2025-08-25 21:05:04.000000000 -0400
@@ -2,7 +2,7 @@
 
 AUTOMAKE_OPTIONS = foreign
 
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 
 # We list all of these as opposed to using a wildcard so that
 # building outside the source directory works.
diff -Nurd libao-0.8.8.orig/src/plugins/macosx/ao_macosx.c libao-0.8.8/src/plugins/macosx/ao_macosx.c
--- libao-0.8.8.orig/src/plugins/macosx/ao_macosx.c	2007-05-24 05:19:07.000000000 -0400
+++ libao-0.8.8/src/plugins/macosx/ao_macosx.c	2025-08-25 20:47:45.000000000 -0400
@@ -173,6 +173,26 @@
 	return 0;
     }
 
+	if (internal->outputStreamBasicDescription.mChannelsPerFrame != format->channels)
+	{
+		internal->outputStreamBasicDescription.mChannelsPerFrame = format->channels;
+		internal->outputStreamBasicDescription.mBytesPerFrame =
+			internal->outputStreamBasicDescription.mChannelsPerFrame * sizeof (float);
+		internal->outputStreamBasicDescription.mBytesPerPacket =
+			internal->outputStreamBasicDescription.mBytesPerFrame *
+			internal->outputStreamBasicDescription.mFramesPerPacket;
+		status = AudioDeviceSetProperty(internal->outputDeviceID, 0, 0, 0,
+			kAudioDevicePropertyStreamFormat,
+			sizeof (internal->outputStreamBasicDescription),
+			&internal->outputStreamBasicDescription);
+		if (status != noErr)
+		{
+			fprintf(stderr, "ao_macosx_open: AudioDeviceSetProperty returned %.*s when setting kAudioDevicePropertyStreamFormat\n",
+				(int) sizeof (status), (char *) &status);
+			return 0;
+		}
+	}
+
     fprintf(stderr, "hardware format...\n");
     fprintf(stderr, "%f mSampleRate\n", internal->outputStreamBasicDescription.mSampleRate);
     fprintf(stderr, "%c%c%c%c mFormatID\n", (int)(internal->outputStreamBasicDescription.mFormatID & 0xff000000) >> 24,
