diff -ruN pulseaudio-14.2-orig/configure pulseaudio-14.2/configure
--- pulseaudio-14.2-orig/configure	2021-01-16 11:37:58
+++ pulseaudio-14.2/configure	2023-08-11 17:35:06
@@ -9364,7 +9364,7 @@
 else
 
   ax_check_save_flags=$CFLAGS
-  CFLAGS="$CFLAGS -pedantic -Werror -std=gnu11"
+  CFLAGS="$CFLAGS -pedantic -std=gnu11"
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -9402,7 +9402,7 @@
 else
 
   ax_check_save_flags=$CFLAGS
-  CFLAGS="$CFLAGS -pedantic -Werror $flag"
+  CFLAGS="$CFLAGS -pedantic $flag"
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
