diff -ruN libvorbis-1.3.7-orig/doc/Makefile.in libvorbis-1.3.7/doc/Makefile.in
--- libvorbis-1.3.7-orig/doc/Makefile.in	2020-07-04 01:20:16.000000000 -0500
+++ libvorbis-1.3.7/doc/Makefile.in	2020-11-26 17:55:50.000000000 -0600
@@ -320,7 +320,7 @@
 builddir = @builddir@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)
+docdir = $(datadir)/doc/$(PACKAGE)
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
diff -ruN libvorbis-1.3.7-orig/doc/vorbisenc/Makefile.in libvorbis-1.3.7/doc/vorbisenc/Makefile.in
--- libvorbis-1.3.7-orig/doc/vorbisenc/Makefile.in	2020-07-04 01:20:16.000000000 -0500
+++ libvorbis-1.3.7/doc/vorbisenc/Makefile.in	2020-11-26 17:56:08.000000000 -0600
@@ -260,7 +260,7 @@
 builddir = @builddir@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)/vorbisenc
+docdir = $(datadir)/doc/$(PACKAGE)/vorbisenc
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
diff -ruN libvorbis-1.3.7-orig/doc/vorbisfile/Makefile.in libvorbis-1.3.7/doc/vorbisfile/Makefile.in
--- libvorbis-1.3.7-orig/doc/vorbisfile/Makefile.in	2020-07-04 01:20:16.000000000 -0500
+++ libvorbis-1.3.7/doc/vorbisfile/Makefile.in	2020-11-26 17:56:15.000000000 -0600
@@ -260,7 +260,7 @@
 builddir = @builddir@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(datadir)/doc/$(PACKAGE)-$(VERSION)/vorbisfile
+docdir = $(datadir)/doc/$(PACKAGE)/vorbisfile
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
