diff -ruN libxmp-4.5.0-orig/Makefile.in libxmp-4.5.0/Makefile.in
--- libxmp-4.5.0-orig/Makefile.in	2021-06-09 19:24:24.000000000 -0500
+++ libxmp-4.5.0/Makefile.in	2021-08-08 17:43:46.000000000 -0500
@@ -5,7 +5,9 @@
 VERSION		= $(VERSION_MAJOR).$(VERSION_MINOR).$(VERSION_RELEASE)
 
 # This dylib will support anything linked against COMPAT_VERSION through VERSION
-COMPAT_VERSION	= $(VERSION_MAJOR)
+# Change back to COMPAT_VERSION from version 4.2.2 to avoid breakage.
+# Eliminate this patch with next VERSION_MAJOR.
+COMPAT_VERSION	= $(VERSION_MAJOR).2
 
 CC		= @CC@
 CFLAGS		= -c @CFLAGS@ @DEFS@ -D_REENTRANT
