diff -ru libxmp-4.2.3.orig/Makefile.in libxmp-4.2.3/Makefile.in
--- libxmp-4.2.3.orig/Makefile.in	2014-01-18 05:53:52.000000000 -0500
+++ libxmp-4.2.3/Makefile.in	2014-01-18 19:50:39.000000000 -0500
@@ -5,7 +5,9 @@
 VERSION		= $(VERSION_MAJOR).$(VERSION_MINOR).$(VERSION_RELEASE)
 
 # This dylib will support anything linked against COMPAT_VERSION through VERSION
-COMPAT_VERSION	= $(VERSION_MAJOR)
+# Change back to COMPAT_VERSION from version 4.2.2 to avoid breakage.
+# Eliminate this patch with next VERSION_MAJOR.
+COMPAT_VERSION	= $(VERSION_MAJOR).2
 
 CC		= @CC@
 CFLAGS		= -c @CFLAGS@ @DEFS@ -D_REENTRANT
@@ -112,7 +114,7 @@
 
 lib/$(DYLIB): $(LOBJS)
 	@mkdir -p lib
-	@CMD='$(LD) $(LDFLAGS) -dynamiclib -Wl,-headerpad_max_install_names,-undefined,dynamic_lookup,-compatibility_version,$(COMPAT_VERSION),-current_version,$(VERSION),-install_name,$(prefix)/lib/$(DYLIB) -o $@ $(LOBJS) $(LIBS)'; \
+	@CMD='$(LD) $(LDFLAGS) -dynamiclib -Wl,-compatibility_version,$(COMPAT_VERSION),-current_version,$(VERSION),-install_name,$(prefix)/lib/$(DYLIB) -o $@ $(LOBJS) $(LIBS)'; \
 	if [ "$(V)" -gt 0 ]; then echo $$CMD; else echo LD $@ ; fi; \
 	eval $$CMD
 	ln -sf $(DYLIB) lib/libxmp.dylib
