diff -ruN moc-2.5.2-orig/decoder_plugins/ffmpeg/ffmpeg.c moc-2.5.2/decoder_plugins/ffmpeg/ffmpeg.c
--- moc-2.5.2-orig/decoder_plugins/ffmpeg/ffmpeg.c	2016-11-15 19:54:37.000000000 -0500
+++ moc-2.5.2/decoder_plugins/ffmpeg/ffmpeg.c	2020-01-06 09:44:28.000000000 -0500
@@ -993,8 +993,8 @@
 #endif
 
 	set_downmixing (data);
-	if (data->codec->capabilities & CODEC_CAP_TRUNCATED)
-		data->enc->flags |= CODEC_FLAG_TRUNCATED;
+	if (data->codec->capabilities & AV_CODEC_CAP_TRUNCATED)
+		data->enc->flags |= AV_CODEC_FLAG_TRUNCATED;
 
 #ifdef HAVE_AVCODEC_OPEN2
 	if (avcodec_open2 (data->enc, data->codec, NULL) < 0)
@@ -1025,7 +1025,7 @@
 
 	data->sample_width = sfmt_Bps (data->fmt);
 
-	if (data->codec->capabilities & CODEC_CAP_DELAY)
+	if (data->codec->capabilities & AV_CODEC_CAP_DELAY)
 		data->delay = true;
 	data->seek_broken = is_seek_broken (data);
 	data->timing_broken = is_timing_broken (data->ic);
diff -ruN moc-2.5.2-orig/keys.c moc-2.5.2/keys.c
--- moc-2.5.2-orig/keys.c	2016-11-15 18:54:37.000000000 -0600
+++ moc-2.5.2/keys.c	2017-03-04 21:56:24.000000000 -0600
@@ -17,7 +17,7 @@
 #include <strings.h>
 #include <assert.h>
 
-#ifdef HAVE_NCURSESW_H
+#ifdef HAVE_NCURSESW_CURSES_H
 # include <ncursesw/curses.h>
 #elif HAVE_NCURSES_H
 # include <ncurses.h>
diff -ruN moc-2.5.2-orig/menu.h moc-2.5.2/menu.h
--- moc-2.5.2-orig/menu.h	2016-11-15 18:54:37.000000000 -0600
+++ moc-2.5.2/menu.h	2017-03-04 21:56:31.000000000 -0600
@@ -1,7 +1,7 @@
 #ifndef MENU_H
 #define MENU_H
 
-#ifdef HAVE_NCURSESW_H
+#ifdef HAVE_NCURSESW_CURSES_H
 # include <ncursesw/curses.h>
 #elif HAVE_NCURSES_H
 # include <ncurses.h>
diff -ruN moc-2.5.2-orig/themes.c moc-2.5.2/themes.c
--- moc-2.5.2-orig/themes.c	2016-11-15 18:54:37.000000000 -0600
+++ moc-2.5.2/themes.c	2017-03-04 21:56:28.000000000 -0600
@@ -13,7 +13,7 @@
 # include "config.h"
 #endif
 
-#ifdef HAVE_NCURSESW_H
+#ifdef HAVE_NCURSESW_CURSES_H
 # include <ncursesw/curses.h>
 #elif HAVE_NCURSES_H
 # include <ncurses.h>
