diff -Nurd -x'*~' phonon-backend-xine-4.4.4.orig/xine/CMakeLists.txt phonon-backend-xine-4.4.4/xine/CMakeLists.txt
--- phonon-backend-xine-4.4.4.orig/xine/CMakeLists.txt	2011-01-19 14:20:53.000000000 -0500
+++ phonon-backend-xine-4.4.4/xine/CMakeLists.txt	2014-07-15 02:42:50.000000000 -0400
@@ -42,7 +42,7 @@
         endif(XINE_BUGFIX_VERSION LESS 9)
     endif(XINE_BUGFIX_VERSION LESS 7)
 else(XINE_MINOR_VERSION LESS 2 AND XINE_BUGFIX_VERSION LESS 90)
-    message(FATAL_ERROR "xine-lib v${XINE_VERSION} was found on your system. This version contains major changes compared to 1.1.x and has not been tested properly for use with this backend. Consider downgrading to version 1.1.9." "http://sourceforge.net/project/showfiles.php?group_id=9655&package_id=9732")
+#   message(FATAL_ERROR "xine-lib v${XINE_VERSION} was found on your system. This version contains major changes compared to 1.1.x and has not been tested properly for use with this backend. Consider downgrading to version 1.1.9." "http://sourceforge.net/project/showfiles.php?group_id=9655&package_id=9732")
 endif(XINE_MINOR_VERSION LESS 2 AND XINE_BUGFIX_VERSION LESS 90)
 
 set(phonon_xine_SRCS
@@ -90,7 +90,7 @@
 
 
 
-include_directories(${XINE_INCLUDE_DIR})
+include_directories(${PHONON_INCLUDES} ${XINE_INCLUDE_DIR})
 
 configure_file(config-xine-widget.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-xine-widget.h )
 configure_file(xine.desktop.cmake ${CMAKE_CURRENT_BINARY_DIR}/xine.desktop)
diff -Nurd -x'*~' phonon-backend-xine-4.4.4.orig/xine/kequalizer_plugin.cpp phonon-backend-xine-4.4.4/xine/kequalizer_plugin.cpp
--- phonon-backend-xine-4.4.4.orig/xine/kequalizer_plugin.cpp	2011-01-19 14:20:53.000000000 -0500
+++ phonon-backend-xine-4.4.4/xine/kequalizer_plugin.cpp	2014-07-15 02:43:43.000000000 -0400
@@ -48,6 +48,8 @@
 #include <xine/xineutils.h>
 #undef this
 
+#include <libavutil/mem.h>
+
 #define KEQUALIZER_MAX_GAIN 12.0
 #define KEQUALIZER_L       2      // Storage for filter taps
 #define KEQUALIZER_KM      10     // Max number of bands 
@@ -281,7 +283,7 @@
     Q_UNUSED(video_target);
     kequalizer_plugin_t *that;
     //deprecated: kequalizer_plugin_t *that = static_cast<kequalizer_plugin_t *>(xine_xmalloc(sizeof(kequalizer_plugin_t)));
-    xine_xmalloc_aligned(2,sizeof(kequalizer_plugin_t),(void**)(&that));
+    that = (kequalizer_plugin_t *)av_mallocz(sizeof(kequalizer_plugin_t));
     post_in_t           *input;
     post_out_t          *output;
     xine_post_in_t      *input_api;
diff -Nurd -x'*~' phonon-backend-xine-4.4.4.orig/xine/mediaobject.cpp phonon-backend-xine-4.4.4/xine/mediaobject.cpp
--- phonon-backend-xine-4.4.4.orig/xine/mediaobject.cpp	2011-01-19 14:20:53.000000000 -0500
+++ phonon-backend-xine-4.4.4/xine/mediaobject.cpp	2014-07-15 02:44:18.000000000 -0400
@@ -443,7 +443,7 @@
     const int lastSize = m_titles.size();
     m_titles.clear();
     int num = 0;
-    char **mrls = xine_get_autoplay_mrls(m_stream->xine(), plugin, &num);
+    const char * const *mrls = xine_get_autoplay_mrls(m_stream->xine(), plugin, &num);
     for (int i = 0; i < num; ++i) {
         if (mrls[i]) {
             debug() << Q_FUNC_INFO << mrls[i];
