diff -ur portaudio/configure portaudio-patched/configure
--- portaudio/configure	2009-12-10 02:37:02.000000000 +0100
+++ portaudio-patched/configure	2011-08-07 16:06:44.000000000 +0200
@@ -21448,6 +21448,7 @@
               mac_sysroot="-isysroot /Developer/SDKs/MacOSX10.4u.sdk"
            fi
         else
+           mac_version_min="-mmacosx-version-min=10.4"
            mac_arches=""
            mac_sysroot=""
            mac_version=""
