diff -Nurd -x'*~' GeoIP-Python-1.2.7.orig/setup.py GeoIP-Python-1.2.7/setup.py
--- GeoIP-Python-1.2.7.orig/setup.py	2011-08-23 14:26:23.000000000 -0400
+++ GeoIP-Python-1.2.7/setup.py	2012-07-10 05:02:27.000000000 -0400
@@ -3,8 +3,7 @@
 module1 = Extension('GeoIP',
 	libraries = ['GeoIP'],
 	sources = ['py_GeoIP.c'],
-	library_dirs = ['/usr/local/lib'],
-	include_dirs = ['/usr/local/include'])
+	)
 
 setup (name = 'GeoIP-Python',
 	version = '1.2.7',
