--- pylibacl-0.2.1.orig/setup.py	2002-12-29 06:49:04.000000000 +0000
+++ pylibacl-0.2.1/setup.py	2005-10-30 20:20:07.000000000 +0000
@@ -14,6 +14,8 @@
 elif u_sysname == "FreeBSD":
     macros.append(("HAVE_FREEBSD", None))
     libs.append("posix1e")
+elif u_sysname == "Darwin":
+	libs.append("pthread")
 else:
     raise ValueError("I don't know your system. Please contact the author")
 
@@ -33,7 +35,7 @@
                              libraries=libs,
                              define_macros=macros,
                              )],
-      data_files=[("/usr/share/doc/pylibacl-%s" % version,
+      data_files=[("share/doc/pylibacl-python2.X",
                    ["README","IMPLEMENTATION", "PLATFORMS",
                     "BENCHMARK",
                     "posix1e.html", "posix1e.txt"])],
