--- apbs-1.1.0-source/tools/python/Makefile.in.org	2009-04-13 18:09:30.000000000 -0400
+++ apbs-1.1.0-source/tools/python/Makefile.in	2009-04-13 18:10:50.000000000 -0400
@@ -52,7 +52,6 @@
 	${top_builddir}/bin/.libs/libapbsmainroutines.a \
 	${top_builddir}/src/aaa_lib/.libs/libapbs.a \
 	${top_builddir}/contrib/lib/libmaloc.a \
-	${top_builddir}/contrib/lib/libapbsblas.a \
 	$(am__DEPENDENCIES_1)
 DEFAULT_INCLUDES = -I. -I$(top_builddir)/src/aaa_inc@am__isrc@
 depcomp =
@@ -235,7 +234,7 @@
 SUBDIRS = vgrid 
 _apbslib_so_SOURCES = apbslib.c
 _apbslib_so_LINK = $(PY_CC) $(PY_LINKFLAGS) -o $@ $(PY_SHARED)
-_apbslib_so_LDADD = ${top_builddir}/bin/.libs/libapbsmainroutines.a ${top_builddir}/src/aaa_lib/.libs/libapbs.a ${top_builddir}/contrib/lib/libmaloc.a ${top_builddir}/contrib/lib/libapbsblas.a $(PY_LDFLAGS)
+_apbslib_so_LDADD = ${top_builddir}/bin/.libs/libapbsmainroutines.a ${top_builddir}/src/aaa_lib/.libs/libapbs.a ${top_builddir}/contrib/lib/libmaloc.a $(PY_LDFLAGS)
 all: all-recursive
 
 .SUFFIXES:
--- apbs-1.1.0-source/configure.org	2009-04-13 18:45:48.000000000 -0400
+++ apbs-1.1.0-source/configure	2009-04-13 18:46:20.000000000 -0400
@@ -22073,7 +22073,7 @@
 			;;
 		darwin**)
 			PY_CC=${F77}
-			PY_LINKFLAGS="-Wl,-framework -Wl,Python -shared"
+			PY_LINKFLAGS="-Wl,-framework -Wl,Python -bundle"
 			PY_LDFLAGS="-Wl,-framework -Wl,Accelerate"
 			;;
 		mingw*)
--- apbs-1.1.0-source/configure.org	2009-05-09 16:22:51.000000000 -0400
+++ apbs-1.1.0-source/configure	2009-05-09 16:25:46.000000000 -0400
@@ -22359,7 +22359,7 @@
 
 SERVICEURL='http://ws.nbcr.net/opal/services/APBS_1.1.0'
 PARALLELSERVICEURL='http://ws.nbcr.net/opal/services/APBS_Parallel_1.1.0'
-SITEPKGS=${prefix}/lib/python${PY_VERSION}/site-packages
+SITEPKGS=@INSTALL_DIR$${prefix}/lib/python${PY_VERSION}/site-packages
 
 sed -e 's%@SITEPKGS@%'${SITEPKGS}'%' \
     -e 's%@PACKAGE_VERSION@%'${PACKAGE_VERSION}'%' \
--- apbs-1.1.0-source/contrib/opal/ZSI/Makefile.in.org	2009-05-09 16:50:42.000000000 -0400
+++ apbs-1.1.0-source/contrib/opal/ZSI/Makefile.in	2009-05-09 16:51:12.000000000 -0400
@@ -529,7 +529,7 @@
 
 install-data-local:
 	-mkdir -p ${SITEPKGS}
-	${py_path} setup.py install --prefix=${prefix}
+	${py_path} setup.py install --prefix=@INSTALL_DIR@${prefix}
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
