diff -urN ATLAS.orig/config.c ATLAS/config.c
--- ATLAS.orig/config.c	Mon Jun 17 10:37:24 2002
+++ ATLAS/config.c	Tue Oct 14 21:29:13 2003
@@ -1955,7 +1955,7 @@
             if (OS == OSOSX && !WANTGCC)
             {
                strcpy(comp, "cc");
-               strcat(flag, " -traditional-cpp -faltivec");
+               strcat(flag, " -no-cpp-precomp -faltivec");
             }
          }
          break;
