--- calc-2.11.5.8/Makefile.orig	Thu Mar 14 09:28:28 2002
+++ calc-2.11.5.8/Makefile	Tue Nov  5 22:40:01 2002
@@ -51,7 +51,7 @@
 #
 # If in doubt, leave TERMCONTROL empty.
 #
-TERMCONTROL=
+TERMCONTROL= -DUSE_TERMIOS
 #TERMCONTROL= -DUSE_TERMIOS
 #TERMCONTROL= -DUSE_TERMIO
 #TERMCONTROL= -DUSE_SGTTY
@@ -556,7 +556,7 @@
 
 #INCDIR= /usr/local/include
 #INCDIR= /dev/env/DJDIR/include
-INCDIR= /usr/include
+INCDIR= @PREFIX@/include
 
 # Where to install calc realted things
 #
@@ -582,15 +582,15 @@
 #
 #BINDIR= /usr/local/bin
 #BINDIR= /dev/env/DJDIR/bin
-BINDIR= /usr/bin
+BINDIR= @PREFIX@/bin
 
 #LIBDIR= /usr/local/lib
 #LIBDIR= /dev/env/DJDIR/lib
-LIBDIR= /usr/lib
+LIBDIR= @PREFIX@/lib
 
 #CALC_SHAREDIR= /usr/local/lib/calc
 #CALC_SHAREDIR= /dev/env/DJDIR/share/calc
-CALC_SHAREDIR= /usr/share/calc
+CALC_SHAREDIR= @PREFIX@/share/calc
 
 # By default, these values are based CALC_SHAREDIR, INCDIR, BINDIR
 # ---------------------------------------------------------------
@@ -619,7 +619,7 @@
 CUSTOMCALDIR= ${CALC_SHAREDIR}/custom
 CUSTOMHELPDIR= ${CALC_SHAREDIR}/custhelp
 CUSTOMINCDIR= ${CALC_INCDIR}/custom
-SCRIPTDIR= ${BINDIR}/cscript
+SCRIPTDIR= ${BINDIR}
 
 # T - top level directory under which calc will be installed
 #
@@ -662,7 +662,7 @@
 #MANDIR=
 #MANDIR= /usr/local/man/man1
 #MANDIR= /usr/man/man1
-MANDIR= /usr/share/man/man1
+MANDIR= @PREFIX@/share/man/man1
 #MANDIR= /dev/env/DJDIR/man/man1
 #MANDIR= /usr/man/u_man/man1
 #MANDIR= /usr/contrib/man/man1
@@ -723,7 +723,7 @@
 # lines below.
 #
 #NROFF= nroff
-NROFF=
+NROFF= nroff
 #NROFF= groff
 NROFF_ARG= -man
 #NROFF_ARG= -mandoc
@@ -767,15 +767,15 @@
 #
 # If in doubt, set USE_READLINE, READLINE_LIB and READLINE_INCLUDE to nothing.
 #
-USE_READLINE=
+USE_READLINE= -DUSE_READLINE
 #USE_READLINE= -DUSE_READLINE
 #
-READLINE_LIB=
+READLINE_LIB= -L@PREFIX@/lib -lreadline -lhistory -lncurses
 #READLINE_LIB= -lreadline -lhistory -lncurses
 #READLINE_LIB= -L/usr/gnu/lib -lreadline -lhistory -lncurses
 #READLINE_LIB= -L/usr/local/lib -lreadline -lhistory -lncurses
 #
-READLINE_INCLUDE=
+READLINE_INCLUDE= -I@PREFIX@/include
 #READLINE_INCLUDE= -I/usr/gnu/include
 #READLINE_INCLUDE= -I/usr/local/include
 
@@ -962,7 +962,7 @@
 LDFLAGS= ${NO_SHARED} ${LD_NO_SHARED}
 ILDFLAGS=
 #
-LCC= gcc
+LCC= cc
 CC= ${PURIFY} ${LCC}
 #
 ###
