diff -ruN coot-0.6.2-pre-1-orig/scheme/group-settings.scm coot-0.6.2-pre-1/scheme/group-settings.scm
--- coot-0.6.2-pre-1-orig/scheme/group-settings.scm	2009-07-04 05:57:36.000000000 -0700
+++ coot-0.6.2-pre-1/scheme/group-settings.scm	2010-03-31 09:52:42.000000000 -0700
@@ -28,6 +28,7 @@
       (set-browser-interface "firefox"))
 
   (if (string=? os-type "Darwin")
+	  (set-display-lists-for-maps 0)
       (set-browser-interface "open")))
 
 ;; York setting for molprobity. 
diff -ruN coot-0.6.2-pre-1-orig/scheme/povray.scm coot-0.6.2-pre-1/scheme/povray.scm
--- coot-0.6.2-pre-1-orig/scheme/povray.scm	2009-07-04 05:57:34.000000000 -0700
+++ coot-0.6.2-pre-1/scheme/povray.scm	2010-03-31 09:54:43.000000000 -0700
@@ -14,7 +14,7 @@
   (lambda ()
     
     ; this is directory that contains colors.inc
-    (list (string-append "+L/sw/share/povray-" povray-version "/include")
+    (list (string-append "+L@PREFIX@/share/povray-" povray-version "/include")
 	  "+FN16")))
 
 ;; Run provray using current displayed image and write .pov file to
diff -ruN coot-0.6.2-pre-1-orig/scheme/raster3d.scm coot-0.6.2-pre-1/scheme/raster3d.scm
--- coot-0.6.2-pre-1-orig/scheme/raster3d.scm	2009-07-04 05:57:34.000000000 -0700
+++ coot-0.6.2-pre-1/scheme/raster3d.scm	2010-03-31 09:55:54.000000000 -0700
@@ -1,7 +1,7 @@
 
 (define coot-r3d-file-name "coot.r3d")
 (define coot-png-file-name "coot.png")
-(define coot-png-display-program "display")
+(define coot-png-display-program "open")
 
 ;; run raster3d
 (define render-image
