--- coq/README.fink~	2010-05-28 13:58:15.000000000 +0200
+++ coq/README.fink	2010-05-28 14:40:12.000000000 +0200
@@ -0,0 +1,4 @@
+This package was originally finkified by Roland Zumkeller
+(roland.zumkeller@polytechnique.fr); it was modified and submitted to the
+fink package submission tracker by Jesse Alama (alama@stanford.edu). It was
+continued from there by Bruno De Fraine (bruno@defraine.net).
--- coq-8.2pl1/Makefile.doc	2009-03-24 09:05:22.000000000 +0100
+++ coq-8.2pl1mod/Makefile.doc	2009-07-20 16:18:58.000000000 +0200
@@ -74,7 +74,7 @@
 
 ### Version
 
-doc/common/version.tex: config/Makefile
+doc/common/version.tex:
 	printf '\\newcommand{\\coqversion}{$(VERSION)}' > doc/common/version.tex
 
 ######################################################################

