diff -urN dx-4.3.2/bin/Makefile.am dx-new/bin/Makefile.am
--- dx-4.3.2/bin/Makefile.am	Fri Jul 11 20:11:37 2003
+++ dx-new/bin/Makefile.am	Sat Oct  2 17:56:12 2004
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
 dxbindir = $(prefix)/dx/bin
-bindir = $(prefix)/bin
+bindir = $(exec_prefix)/bin
 
 EXTRA_SCRIPTS = mdf2c dx
 EXTRA_DIST = url_handler.sh
diff -urN dx-4.3.2/bin/dx.in dx-new/bin/dx.in
--- dx-4.3.2/bin/dx.in	Fri Sep 26 10:35:00 2003
+++ dx-new/bin/dx.in	Sat Oct  2 17:54:27 2004
@@ -60,10 +60,10 @@
 #
 longhelp() {
 
-if test -f "$dxroot/man/catl/dx.l" ; then
-    more "$dxroot/man/catl/dx.l"
+if test -f "$dxroot/../man/catl/dx.l" ; then
+    more "$dxroot/../man/catl/dx.l"
 else
-    echo "cannot find $dxroot/man/catl/dx.l"
+    echo "cannot find $dxroot/../man/catl/dx.l"
     echo "set the DXROOT environment variable to the root of the"
     echo "dx installation tree and try again."
 fi
diff -urN dx-4.3.2/configure.ac dx-new/configure.ac
--- dx-4.3.2/configure.ac	Tue Sep 16 15:48:18 2003
+++ dx-new/configure.ac	Sat Oct  2 17:56:12 2004
@@ -675,7 +675,7 @@
 	AC_MSG_RESULT(no -- some components failed test)
         have_hdf='no (failed tests)'
     else
-	LIB_HDF='-ldf'
+	LIB_HDF='-ldf -ljpeg -lz'
 	LIBS="$LIB_HDF $LIBS"
 	FMT_LIBS="$FMT_LIBS libdf.a"
 	AC_DEFINE(HAVE_LIBDF, 1, [Define to 1 if you have the HDF libraries])
diff -urN dx-4.3.2/include/Makefile.am dx-new/include/Makefile.am
--- dx-4.3.2/include/Makefile.am	Thu Aug 16 16:12:41 2001
+++ dx-new/include/Makefile.am	Sat Oct  2 17:56:12 2004
@@ -1,3 +1,3 @@
-dxincdir = $(prefix)/dx/include
+dxincdir = $(exec_prefix)/include
 SUBDIRS = dx
 dxinc_HEADERS = dxl.h dxconfig.h dxstereo.h
diff -urN dx-4.3.2/include/dx/Makefile.am dx-new/include/dx/Makefile.am
--- dx-4.3.2/include/dx/Makefile.am	Tue May 25 19:35:16 1999
+++ dx-new/include/dx/Makefile.am	Sat Oct  2 17:56:12 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-dxincdir = $(prefix)/dx/include/dx
+dxincdir = $(exec_prefix)/include/dx
  
 dxinc_HEADERS = \
 	UserInteractors.h advanced.h arch.h array.h arrayhandles.h \
diff -urN dx-4.3.2/man/catl/Makefile.am dx-new/man/catl/Makefile.am
--- dx-4.3.2/man/catl/Makefile.am	Mon Apr  5 07:56:00 1999
+++ dx-new/man/catl/Makefile.am	Sat Oct  2 17:56:12 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-catdir = $(prefix)/dx/man/catl
+catdir = $(exec_prefix)/share/man/catl
 
 cat_DATA = dx.l
 EXTRA_DIST = $(cat_DATA)
diff -urN dx-4.3.2/man/manl/Makefile.am dx-new/man/manl/Makefile.am
--- dx-4.3.2/man/manl/Makefile.am	Mon May 20 00:47:06 2002
+++ dx-new/man/manl/Makefile.am	Sat Oct  2 17:56:12 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-mandir = $(prefix)/dx/man
+mandir = $(exec_prefix)/share/man
 
 man_MANS = dx.l
 EXTRA_DIST = $(man_MANS)
