diff -Nurd -x'*~' dynagraph-1.3f.orig/src/glcanvas.c dynagraph-1.3f/src/glcanvas.c
--- dynagraph-1.3f.orig/src/glcanvas.c	2003-10-28 19:54:36.000000000 -0500
+++ dynagraph-1.3f/src/glcanvas.c	2005-12-21 03:03:42.000000000 -0500
@@ -38,6 +38,11 @@
 }
 CSPEC;
 
+/* forward declaration */
+FL_OBJECT *
+fl_create_glcanvas(int type, FL_Coord x, FL_Coord y, FL_Coord w, FL_Coord h,
+		   const char *label);
+
 /*
  * Default GL canvas configuration. The user can modify this default
  * using fl_set_glcanvas_defaults(const int *config) or
