diff -Nurd EMBOSS-6.3.1/README.fixes EMBOSS-6.3.1-patched/README.fixes
--- EMBOSS-6.3.1/README.fixes	1969-12-31 19:00:00.000000000 -0500
+++ EMBOSS-6.3.1-patched/README.fixes	2010-07-21 05:25:27.000000000 -0400
@@ -0,0 +1,9 @@
+The files in this directory are bugfix replacements for files in
+the EMBOSS-6.3.1 distribution. Just drop the replacement files in
+the location shown and redo the 'make install.'
+
+Fix 1. EMBOSS-6.3.1/configure
+       EMBOSS-6.3.1/m4/mysql.m4
+
+21 Jul 2010: Addresses a problem whereby, in some circumstances, inclusion of
+             hpdf support prevented inclusion of MySQL support.
diff -Nurd EMBOSS-6.3.1/ajax/acd/Makefile.in EMBOSS-6.3.1-patched/ajax/acd/Makefile.in
--- EMBOSS-6.3.1/ajax/acd/Makefile.in	2010-07-19 07:55:40.000000000 -0400
+++ EMBOSS-6.3.1-patched/ajax/acd/Makefile.in	2010-07-24 10:57:13.000000000 -0400
@@ -73,7 +73,7 @@
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(libdir)" "$(DESTDIR)$(includedir)"
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libacd_la_LIBADD =
+libacd_la_LIBADD = ../core/libajax.la ../graphics/libajaxg.la ../ajaxdb/libajaxdb.la
 am__objects_1 = ajacd.lo
 am_libacd_la_OBJECTS = $(am__objects_1)
 libacd_la_OBJECTS = $(am_libacd_la_OBJECTS)
diff -Nurd EMBOSS-6.3.1/ajax/ajaxdb/Makefile.in EMBOSS-6.3.1-patched/ajax/ajaxdb/Makefile.in
--- EMBOSS-6.3.1/ajax/ajaxdb/Makefile.in	2010-07-19 07:55:40.000000000 -0400
+++ EMBOSS-6.3.1-patched/ajax/ajaxdb/Makefile.in	2010-07-24 10:57:47.000000000 -0400
@@ -73,7 +73,7 @@
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(libdir)" "$(DESTDIR)$(includedir)"
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libajaxdb_la_LIBADD =
+libajaxdb_la_LIBADD = ../core/libajax.la
 am__objects_1 = ajmart.lo ajseqdb.lo
 am_libajaxdb_la_OBJECTS = $(am__objects_1)
 libajaxdb_la_OBJECTS = $(am_libajaxdb_la_OBJECTS)
diff -Nurd EMBOSS-6.3.1/ajax/core/Makefile.in EMBOSS-6.3.1-patched/ajax/core/Makefile.in
--- EMBOSS-6.3.1/ajax/core/Makefile.in	2010-07-19 07:55:40.000000000 -0400
+++ EMBOSS-6.3.1-patched/ajax/core/Makefile.in	2010-07-24 13:04:41.000000000 -0400
@@ -73,7 +73,7 @@
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(libdir)" "$(DESTDIR)$(includedir)"
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libajax_la_LIBADD =
+libajax_la_LIBADD =  ../pcre/libepcre.la
 am__objects_1 = ajalign.lo ajarr.lo ajassert.lo ajbase.lo ajcall.lo \
 	ajcod.lo ajdan.lo ajdmx.lo ajdom.lo ajdomain.lo ajexcept.lo \
 	ajfeat.lo ajfile.lo ajfiledata.lo ajfileio.lo ajfmt.lo \
diff -Nurd EMBOSS-6.3.1/ajax/graphics/Makefile.in EMBOSS-6.3.1-patched/ajax/graphics/Makefile.in
--- EMBOSS-6.3.1/ajax/graphics/Makefile.in	2010-07-19 07:55:40.000000000 -0400
+++ EMBOSS-6.3.1-patched/ajax/graphics/Makefile.in	2010-07-24 13:05:30.000000000 -0400
@@ -73,7 +73,7 @@
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(libdir)" "$(DESTDIR)$(includedir)"
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libajaxg_la_LIBADD =
+libajaxg_la_LIBADD = ../core/libajax.la ../../plplot/libeplplot.la
 am__objects_1 = ajgraph.lo ajhist.lo
 am_libajaxg_la_OBJECTS = $(am__objects_1)
 libajaxg_la_OBJECTS = $(am_libajaxg_la_OBJECTS)
diff -Nurd EMBOSS-6.3.1/configure EMBOSS-6.3.1-patched/configure
--- EMBOSS-6.3.1/configure	2010-07-19 07:55:36.000000000 -0400
+++ EMBOSS-6.3.1-patched/configure	2010-07-21 05:25:34.000000000 -0400
@@ -14481,8 +14481,8 @@
 
     	    EMBCFLAGS=$CFLAGS
 	    EMBLDFLAGS=$LDFLAGS
-	    CFLAGS=$MYSQL_CFLAGS
-	    LDFLAGS=$MYSQL_LDFLAGS
+	    CFLAGS="$MYSQL_CFLAGS $EMBCFLAGS"
+	    LDFLAGS="$MYSQL_LDFLAGS $EMBLDFLAGS"
 
             cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
diff -Nurd EMBOSS-6.3.1/m4/mysql.m4 EMBOSS-6.3.1-patched/m4/mysql.m4
--- EMBOSS-6.3.1/m4/mysql.m4	2010-05-18 18:12:06.000000000 -0400
+++ EMBOSS-6.3.1-patched/m4/mysql.m4	2010-07-21 05:25:41.000000000 -0400
@@ -96,8 +96,8 @@
 
     	    EMBCFLAGS=$CFLAGS
 	    EMBLDFLAGS=$LDFLAGS
-	    CFLAGS=$MYSQL_CFLAGS
-	    LDFLAGS=$MYSQL_LDFLAGS
+	    CFLAGS="$MYSQL_CFLAGS $EMBCFLAGS"
+	    LDFLAGS="$MYSQL_LDFLAGS $EMBLDFLAGS"
 
             AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <stdio.h>
                                               #include "mysql.h"]],
diff -Nurd EMBOSS-6.3.1/nucleus/Makefile.in EMBOSS-6.3.1-patched/nucleus/Makefile.in
--- EMBOSS-6.3.1/nucleus/Makefile.in	2010-07-19 07:55:42.000000000 -0400
+++ EMBOSS-6.3.1-patched/nucleus/Makefile.in	2010-07-24 13:06:08.000000000 -0400
@@ -73,7 +73,7 @@
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(libdir)" "$(DESTDIR)$(includedir)"
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libnucleus_la_LIBADD =
+libnucleus_la_LIBADD = ../ajax/core/libajax.la ../ajax/graphics/libajaxg.la ../ajax/ajaxdb/libajaxdb.la ../ajax/acd/libacd.la
 am__objects_1 = embaln.lo embcom.lo embcons.lo embdata.lo embdbi.lo \
 	embdmx.lo embdomain.lo embest.lo embexit.lo embgroup.lo \
 	embiep.lo embindex.lo embinit.lo embmat.lo embmisc.lo \
diff -Nurd EMBOSS-6.3.1/plplot/Makefile.in EMBOSS-6.3.1-patched/plplot/Makefile.in
--- EMBOSS-6.3.1/plplot/Makefile.in	2010-07-19 07:55:42.000000000 -0400
+++ EMBOSS-6.3.1-patched/plplot/Makefile.in	2010-07-24 13:06:33.000000000 -0400
@@ -72,7 +72,7 @@
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(libdir)" "$(DESTDIR)$(pkgincludedir)"
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libeplplot_la_LIBADD =
+libeplplot_la_LIBADD = = $(XLIB) -lgd
 am__objects_1 = pdfutils.lo plargs.lo plbox.lo plcont.lo plcore.lo \
 	plctrl.lo plcvt.lo pldtik.lo plfill.lo plhist.lo plline.lo \
 	plmap.lo plot3d.lo plpage.lo plsdef.lo plshade.lo plsym.lo \
