diff -Naur EMBOSS-5.0.0/ajax/Makefile.in EMBOSS-5.0.0-patched/ajax/Makefile.in
--- EMBOSS-5.0.0/ajax/Makefile.in	2007-07-12 06:53:32.000000000 -0400
+++ EMBOSS-5.0.0-patched/ajax/Makefile.in	2007-12-03 20:09:43.000000000 -0500
@@ -55,7 +55,8 @@
 am__installdirs = "$(DESTDIR)$(libdir)" "$(DESTDIR)$(includedir)"
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libajax_la_LIBADD =
+#libajax_la_LIBADD =
+libajaxg_la_LIBADD = libajax.la ../plplot/libeplplot.la
 am__libajax_la_SOURCES_DIST = ajacd.c ajalign.c ajarr.c ajassert.c \
 	ajbase.c ajcall.c ajcod.c ajdan.c ajdmx.c ajdom.c ajdomain.c \
 	ajexcept.c ajfeat.c ajfile.c ajfmt.c ajgraphxml.c ajindex.c \
@@ -85,7 +86,9 @@
 	$(libajax_la_LDFLAGS) $(LDFLAGS) -o $@
 @ISCYGWIN_FALSE@am_libajax_la_rpath = -rpath $(libdir)
 @ISCYGWIN_TRUE@am_libajax_la_rpath = -rpath $(libdir)
-libajaxg_la_LIBADD =
+#libajaxg_la_LIBADD =
+libajaxg_la_LIBADD = libajax.la ../plplot/libeplplot.la
+libajaxg_la_DEPENDENCIES = libajax.la ../plplot/libeplplot.la
 am__libajaxg_la_SOURCES_DIST = ajgraph.c ajhist.c
 @ISCYGWIN_FALSE@am_libajaxg_la_OBJECTS = $(am__objects_2)
 libajaxg_la_OBJECTS = $(am_libajaxg_la_OBJECTS)
diff -Naur EMBOSS-5.0.0/ajax/ajindex.c EMBOSS-5.0.0-patched/ajax/ajindex.c
--- EMBOSS-5.0.0/ajax/ajindex.c	2007-07-06 08:18:48.000000000 -0400
+++ EMBOSS-5.0.0-patched/ajax/ajindex.c	2008-01-09 13:50:33.000000000 -0500
@@ -2516,7 +2516,14 @@
 
 static ajint btreeNumIdCompare(const void *a, const void *b)
 {
-    return (*(AjPBtNumId const *)a)->offset - (*(AjPBtNumId const *)b)->offset;
+  ajlong val;
+
+    val = (*(AjPBtNumId const *)a)->offset - (*(AjPBtNumId const *)b)->offset;
+
+    if(!val)
+      return 0;
+
+    return (val < 0L) ? -1 : 1;
 }
 
 
@@ -11956,8 +11963,14 @@
 
 static ajint btreeOffsetCompare(const void *a, const void *b)
 {
-    return (*(AjPBtId const *)a)->offset -
-		  (*(AjPBtId const *)b)->offset;
+  ajlong val;
+
+    val = (*(AjPBtId const *)a)->offset - (*(AjPBtId const *)b)->offset;
+
+    if(!val)
+      return 0;
+
+    return (val < 0L) ? -1 : 1;
 }
 
 
diff -Naur EMBOSS-5.0.0/emboss/psiphi.c EMBOSS-5.0.0-patched/emboss/psiphi.c
--- EMBOSS-5.0.0/emboss/psiphi.c	2007-07-06 08:15:29.000000000 -0400
+++ EMBOSS-5.0.0-patched/emboss/psiphi.c	2007-07-27 05:28:27.000000000 -0400
@@ -160,7 +160,7 @@
     finishres = ajAcdGetInt("finishresiduenumber");
 
     /* reserve memory for and read in structure */
-    /* JISON */    pdb = ajPdbReadoldNew(pdbfile);
+    /* JISON */    pdb = ajPdbReadNew(pdbfile,0);
     
     /* check and set number of chain to be analysed */
     highest = pdb->Nchn;
@@ -440,6 +440,9 @@
     ajint lowestres = 0;
 
     AjPAtom inlist   = NULL;
+
+    if(!ajListGetLength(pdb->Chains[myindex]->Atoms))
+	ajFatal("Chain %d has no atoms",myindex+1);
     
     /* read first atom in list into memory, but keep it on list */
     ajListPeek(pdb->Chains[myindex]->Atoms,
diff -Naur EMBOSS-5.0.0/nucleus/Makefile.in EMBOSS-5.0.0-patched/nucleus/Makefile.in
--- EMBOSS-5.0.0/nucleus/Makefile.in	2007-07-12 06:53:36.000000000 -0400
+++ EMBOSS-5.0.0-patched/nucleus/Makefile.in	2007-12-03 20:09:04.000000000 -0500
@@ -55,7 +55,9 @@
 am__installdirs = "$(DESTDIR)$(libdir)" "$(DESTDIR)$(includedir)"
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libnucleus_la_LIBADD =
+#libnucleus_la_LIBADD =
+libnucleus_la_LIBADD = ../ajax/libajax.la
+libajaxg_la_DEPENDENCIES = ../ajax/libajax.la
 am__objects_1 = embaln.lo embcom.lo embcons.lo embdata.lo embdbi.lo \
 	embdmx.lo embdomain.lo embest.lo embexit.lo embgroup.lo \
 	embiep.lo embindex.lo embinit.lo embmat.lo embmisc.lo \
diff -Naur EMBOSS-5.0.0/plplot/Makefile.in EMBOSS-5.0.0-patched/plplot/Makefile.in
--- EMBOSS-5.0.0/plplot/Makefile.in	2007-07-12 06:53:36.000000000 -0400
+++ EMBOSS-5.0.0-patched/plplot/Makefile.in	2007-12-03 20:10:52.000000000 -0500
@@ -54,7 +54,8 @@
 am__installdirs = "$(DESTDIR)$(libdir)" "$(DESTDIR)$(pkgincludedir)"
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libeplplot_la_LIBADD =
+#libeplplot_la_LIBADD =
+libeplplot_la_LIBADD = $(X_LIBS) -lX11 -lgd
 am__objects_1 = pdfutils.lo plargs.lo plbox.lo plcont.lo plcore.lo \
 	plctrl.lo plcvt.lo pldtik.lo plfill.lo plhist.lo plline.lo \
 	plmap.lo plot3d.lo plpage.lo plsdef.lo plshade.lo plsym.lo \
