diff -Naur EMBOSS-6.0.1/ajax/ajseqwrite.c EMBOSS-6.0.1-patched/ajax/ajseqwrite.c
--- EMBOSS-6.0.1/ajax/ajseqwrite.c	2008-07-14 10:58:07.000000000 -0400
+++ EMBOSS-6.0.1-patched/ajax/ajseqwrite.c	2008-12-15 05:40:32.000000000 -0500
@@ -3743,7 +3743,7 @@
 	ajFmtPrintF(outseq->File, "DE   %S\n", outseq->Desc);
     
     if(ajStrGetLen(outseq->Tax))
-	ajFmtPrintF(outseq->File, "OS   %S\n", outseq->Tax);
+	ajFmtPrintF(outseq->File, "OS   %S.\n", outseq->Tax);
     
     if(ajListGetLength(outseq->Taxlist) > 1)
     {
diff -Naur EMBOSS-6.0.1/emboss/jaspscan.c EMBOSS-6.0.1-patched/emboss/jaspscan.c
--- EMBOSS-6.0.1/emboss/jaspscan.c	2008-06-26 04:39:09.000000000 -0400
+++ EMBOSS-6.0.1-patched/emboss/jaspscan.c	2008-08-27 03:32:24.000000000 -0400
@@ -293,7 +293,7 @@
 
 	    jaspscan_scan(substr,begin,mfname, cp, thresh, both, hits);
 
-	    ajStrDel(&mfname);
+            ajListPushAppend(flist, (void **)mfname);
 	}
 
 	jaspscan_ReportHits(TabRpt,mattab,hits);
@@ -303,6 +303,10 @@
     }
 
 
+    while(ajListPop(flist,(void **)&mfname))
+        ajStrDel(&mfname);
+
+    
     ajStrDel(&dir);
     ajStrDel(&menu);
     ajStrDel(&excl);
diff -Naur EMBOSS-6.0.1/emboss/vectorstrip.c EMBOSS-6.0.1-patched/emboss/vectorstrip.c
--- EMBOSS-6.0.1/emboss/vectorstrip.c	2008-06-26 04:39:09.000000000 -0400
+++ EMBOSS-6.0.1-patched/emboss/vectorstrip.c	2008-12-15 05:23:57.000000000 -0500
@@ -174,7 +174,7 @@
     vectorlist = ajListNew();
 
     /* data from command line or file? */
-    if(vec == AJTRUE)
+    if(vec && vectorfile)
 	vectorstrip_read_vector_data(vectorfile, vectorlist);
     else
     {
diff -Naur EMBOSS-6.0.1/nucleus/embprop.c EMBOSS-6.0.1-patched/nucleus/embprop.c
--- EMBOSS-6.0.1/nucleus/embprop.c	2008-07-14 18:13:33.000000000 -0400
+++ EMBOSS-6.0.1-patched/nucleus/embprop.c	2008-08-27 03:32:32.000000000 -0400
@@ -880,6 +880,7 @@
 
     if(allpartials)
     {
+        lim = defcnt;
 	for(i=0;i<lim;++i)
 	    for(j=i+1;j<lim;++j)
 	    {
