diff -Naur EMBOSS-6.1.0/ajax/ajfeat.c EMBOSS-6.1.0-patched/ajax/ajfeat.c
--- EMBOSS-6.1.0/ajax/ajfeat.c	2009-07-06 11:48:46.000000000 -0400
+++ EMBOSS-6.1.0-patched/ajax/ajfeat.c	2009-07-30 11:13:23.000000000 -0400
@@ -9517,12 +9517,12 @@
     ajint i;
     
     if(!featRegUfoFmt)
-	featRegUfoFmt = ajRegCompC("^([A-Za-z0-9]+):+(.*)$");
+	featRegUfoFmt = ajRegCompC("^([A-Za-z0-9][A-Za-z0-9]+):+(.*)$");
     /* \1 format */
     /* \2 remainder */
     
     if(!featRegUfoFile)
-	featRegUfoFile = ajRegCompC("^([^:]+)$");
+	featRegUfoFile = ajRegCompC("^(([A-Za-z]:)?[^:]+)$");
     
     /*ajDebug("ajFeatUfoRead UFO '%S'\n", ufo);*/
     
diff -Naur EMBOSS-6.1.0/ajax/ajseqread.c EMBOSS-6.1.0-patched/ajax/ajseqread.c
--- EMBOSS-6.1.0/ajax/ajseqread.c	2009-07-06 19:08:26.000000000 -0400
+++ EMBOSS-6.1.0-patched/ajax/ajseqread.c	2009-07-30 11:13:23.000000000 -0400
@@ -689,7 +689,7 @@
        AJTRUE,  AJTRUE,  seqReadSwiss, AJFALSE, 0},
   {"nbrf",        "NBRF/PIR entry format",
        AJFALSE, AJTRUE,  AJTRUE,  AJTRUE,
-       AJFALSE, AJTRUE,  seqReadNbrf, AJFALSE, 0},	/* test before NCBI */
+       AJTRUE,  AJTRUE,  seqReadNbrf, AJFALSE, 0},	/* test before NCBI */
   {"pir",         "NBRF/PIR entry format (alias)",
        AJTRUE,  AJFALSE, AJTRUE,  AJTRUE,
        AJTRUE,  AJTRUE,  seqReadNbrf, AJFALSE, 0},	/* alias for nbrf */
@@ -712,7 +712,7 @@
   {"ncbi",        "FASTA format including NCBI-style IDs (alias)",
        AJTRUE,  AJFALSE, AJTRUE,  AJTRUE,
        AJFALSE, AJTRUE,  seqReadNcbi, AJFALSE, 0}, /* test before pearson */
-  {"gifasta",     "FASTA format including NCBI-style IDs (alias)",
+  {"gifasta",     "FASTA format including NCBI-style GIs (alias)",
        AJFALSE, AJFALSE, AJTRUE,  AJTRUE,
        AJFALSE, AJTRUE,  seqReadGifasta, AJFALSE, 0}, /* NCBI with GI as ID*/
   {"pearson",     "Plain old fasta format with IDs not parsed further",
@@ -749,11 +749,11 @@
        AJTRUE,  AJFALSE, AJTRUE,  AJFALSE,
        AJTRUE,  AJTRUE,  seqReadRefseq, AJFALSE, 0}, /* alias for genbank */
   {"refseqp",     "Refseq protein entry format",
-       AJFALSE, AJFALSE, AJFALSE, AJTRUE,       /* genbank forwards proteins */
+       AJFALSE, AJFALSE, AJFALSE, AJTRUE,       /* genbank format proteins */
        AJTRUE,  AJTRUE,  seqReadRefseqp, AJFALSE, 0},
   {"genpept",     "Refseq protein entry format (alias)",
-       AJTRUE,  AJFALSE, AJFALSE, AJTRUE,
-       AJTRUE,  AJTRUE,  seqReadGenpept, AJFALSE, 0}, /* alias for refseqp*/
+       AJFALSE, AJFALSE, AJFALSE, AJTRUE,
+       AJTRUE,  AJTRUE,  seqReadGenpept, AJFALSE, 0},
   {"codata",      "Codata entry format",
        AJFALSE, AJTRUE,  AJTRUE,  AJTRUE,
        AJTRUE,  AJTRUE,  seqReadCodata, AJFALSE, 0},
@@ -785,7 +785,7 @@
   {"hennig86",    "Hennig86 output format",
        AJFALSE, AJTRUE,  AJTRUE,  AJTRUE,
        AJFALSE, AJTRUE,  seqReadHennig86, AJFALSE, 0},
-  {"jackknifer",  "Jackknifer output interleaved format",
+  {"jackknifer",  "Jackknifer interleaved and non-interleaved formats",
        AJFALSE, AJTRUE,  AJTRUE,  AJTRUE,
        AJFALSE, AJTRUE,  seqReadJackknifer, AJFALSE, 0},
   {"nexus",       "Nexus/paup interleaved format",
@@ -797,7 +797,7 @@
   {"treecon",     "Treecon output format",
        AJFALSE, AJTRUE,  AJTRUE,  AJTRUE,
        AJFALSE, AJTRUE,  seqReadTreecon, AJFALSE, 0},
-  {"mega",        "Mega interleaved output format",
+  {"mega",        "Mega interleaved and non-interleaved formats",
        AJFALSE, AJTRUE,  AJTRUE,  AJTRUE,
        AJFALSE, AJTRUE,  seqReadMega, AJFALSE, 0},
   {"igstrict",    "Intelligenetics sequence format strict parser",
@@ -813,7 +813,7 @@
        AJFALSE, AJFALSE, AJTRUE,  AJTRUE,
        AJFALSE, AJTRUE,  seqReadText, AJFALSE, 0},/* can read almost anything */
   {"plain",       "Plain text (alias)",
-       AJFALSE, AJFALSE, AJTRUE,  AJTRUE,
+       AJTRUE,  AJFALSE, AJTRUE,  AJTRUE,
        AJFALSE, AJTRUE,  seqReadText, AJFALSE, 0},	/* alias for text */
   {"gff2",         "GFF feature file with sequence in the header",
        AJFALSE, AJTRUE,  AJTRUE,  AJTRUE,
@@ -822,17 +822,17 @@
        AJFALSE, AJTRUE,  AJTRUE,  AJTRUE,
        AJTRUE,  AJTRUE,  seqReadGff3, AJFALSE, 0},
   {"gff",         "GFF3 feature file with sequence",
-       AJTRUE,  AJTRUE,  AJTRUE,  AJTRUE,
+       AJTRUE,  AJFALSE,  AJTRUE,  AJTRUE,
        AJTRUE,  AJTRUE,  seqReadGff3, AJFALSE, 0},
   {"stockholm",   "Stockholm (pfam) format",
        AJFALSE, AJTRUE,  AJTRUE,  AJTRUE,
        AJFALSE, AJTRUE,  seqReadStockholm, AJFALSE, 0},
-  {"selex",       "Selex format",                /* can read almost anything */
-       AJFALSE, AJFALSE, AJTRUE,  AJTRUE,
-       AJFALSE, AJTRUE,  seqReadSelex, AJFALSE, 0},
   {"pfam",        "Stockholm (pfam) format (alias)",
        AJTRUE,  AJTRUE,  AJTRUE,  AJTRUE,
        AJFALSE, AJTRUE,  seqReadStockholm, AJFALSE, 0},
+  {"selex",       "Selex format",                /* can read almost anything */
+       AJFALSE, AJFALSE, AJTRUE,  AJTRUE,
+       AJFALSE, AJTRUE,  seqReadSelex, AJFALSE, 0},
   {"fitch",       "Fitch program format",
        AJFALSE, AJTRUE,  AJTRUE,  AJTRUE,
        AJFALSE, AJTRUE,  seqReadFitch, AJFALSE, 0},
@@ -2627,8 +2627,8 @@
 				 seqin->Text, &thys->TextPtr);
 
     while(ok &&
-          (ajStrGetLen(qualstr) < seqlen) &&
-          (!ajStrPrefixC(seqReadLine, "@")))
+          ((ajStrGetLen(qualstr) < seqlen) ||
+           ajStrGetCharFirst(seqReadLine) != '@'))
     {
         seqqualAppendWarn(&qualstr, seqReadLine);
 
@@ -2790,8 +2790,8 @@
 				 seqin->Text, &thys->TextPtr);
 
     while(ok &&
-          (ajStrGetLen(qualstr) < seqlen) &&
-          (!ajStrPrefixC(seqReadLine, "@")))
+          ((ajStrGetLen(qualstr) < seqlen) ||
+           ajStrGetCharFirst(seqReadLine) != '@'))
     {
         seqqualAppendWarn(&qualstr, seqReadLine);
 
@@ -3157,8 +3157,8 @@
 				 seqin->Text, &thys->TextPtr);
 
     while(ok &&
-          (ajStrGetLen(qualstr) < seqlen) &&
-          (!ajStrPrefixC(seqReadLine, "@")))
+          ((ajStrGetLen(qualstr) < seqlen) ||
+           ajStrGetCharFirst(seqReadLine) != '@'))
     {
         seqqualAppendWarn(&qualstr, seqReadLine);
 
@@ -3352,8 +3352,8 @@
 				 seqin->Text, &thys->TextPtr);
 
     while(ok &&
-          (ajStrGetLen(qualstr) < seqlen) &&
-          (!ajStrPrefixC(seqReadLine, "@")))
+          ((ajStrGetLen(qualstr) < seqlen) ||
+           ajStrGetCharFirst(seqReadLine) != '@'))
     {
         seqqualAppendWarn(&qualstr, seqReadLine);
 
@@ -5254,8 +5254,7 @@
 	return ajFalse;
     }
 
-    ajStrAssignS(&thys->Name, seqReadLine);
-    ajStrCutEnd(&thys->Name, 1);
+    seqSetName(&thys->Name, seqReadLine);
     bufflines++;
 
     while(ajBuffreadLineStore(buff, &seqReadLine,
@@ -5330,8 +5329,7 @@
 	return ajFalse;
     }
 
-    ajStrAssignS(&thys->Name, seqReadLine);
-    ajStrCutEnd(&thys->Name, 1);
+    seqSetName(&thys->Name, seqReadLine);
     bufflines++;
 
     while(ajBuffreadLineStore(buff, &seqReadLine,
@@ -6651,6 +6649,7 @@
 	    if(ajStrIsWhite(phyitem->Name) ||
 	       ajTableFetch(phytable, phyitem->Name))
 	    {
+                ajFilebuffSetBuffered(buff);
 		ajFilebuffResetStore(buff, seqin->Text, &thys->TextPtr);
 		ajDebug("phytable repeated name '%S'\n",
 			phyitem->Name);
@@ -6745,6 +6744,7 @@
 			    ajDebug("phylip format length mismatch at %d "
 				    "(length %d)\n",
 				    maxlen, ilen);
+                            ajFilebuffSetBuffered(buff);
 			    ajFilebuffResetStore(buff,
 						 seqin->Text, &thys->TextPtr);
 			    seqMsfDataDel((SeqPMsfData*) &seqin->Data);
@@ -6771,6 +6771,7 @@
 	    if(!done)
 	    {
 		ajDebug("seqReadPhylip read failed, try seqReadPhylipnon\n");
+                ajFilebuffSetBuffered(buff);
 		ajFilebuffResetStore(buff, seqin->Text, &thys->TextPtr);
 		seqMsfDataDel((SeqPMsfData*) &seqin->Data);
 
@@ -6781,6 +6782,7 @@
 	    {
 		ajDebug("Phylip format %d sequences partly read at end\n",
 			iseq-jseq);
+                ajFilebuffSetBuffered(buff);
 		ajFilebuffResetStore(buff, seqin->Text, &thys->TextPtr);
 		seqMsfDataDel((SeqPMsfData*) &seqin->Data);
 
@@ -10973,6 +10975,8 @@
     if(!ajBuffreadLine(buff, &seqReadLine))
 	return ajFalse;
 
+    ajDebug("++seqReadGenbank first line '%S'\n", seqReadLine);
+
     bufflines++;
 
     ok = ajTrue;
@@ -11030,6 +11034,7 @@
 
     if(nfields == 9) 
     {
+        ajFilebuffSetBuffered(buff);
         ajFilebuffResetStore(buff, seqin->Text, &thys->TextPtr);
 
         return seqReadGenpept(thys,seqin);
@@ -11053,10 +11058,13 @@
 	case 4:
 	    if(ajStrMatchC(token, "aa"))
             {
+                ajFilebuffSetBuffered(buff);
                 ajFilebuffResetStore(buff, seqin->Text, &thys->TextPtr);
                 ajStrDel(&token);
                 ajStrTokenDel(&handle);
 
+                ajDebug("first line %d aa pass to refseqp '%S'\n",
+                        buff->Pos, seqReadLine);
                 return seqReadRefseqp(thys,seqin);
             }
 	    if(!ajStrMatchC(token, "bp"))
@@ -11521,6 +11529,7 @@
 
     if(nfields == 8) 
     {
+        ajFilebuffSetBuffered(buff);
         ajFilebuffResetStore(buff, seqin->Text, &thys->TextPtr);
 
         return seqReadRefseqp(thys,seqin);
@@ -11916,6 +11925,8 @@
     if(!ajBuffreadLine(buff, &seqReadLine))
 	return ajFalse;
 
+    ajDebug("++seqReadRefseqp  %d first line '%S'\n", buff->Pos, seqReadLine);
+
     bufflines++;
 
     ok = ajTrue;
@@ -12303,8 +12314,8 @@
 	    ok = ajBuffreadLineStore(buff,&seqReadLine,
 				    seqin->Text, &thys->TextPtr);
 
-
     ajFilebuffClear(buff, 0);
+    ajDebug("++last line %d '%S'\n", buff->Pos, seqReadLine);
 
     ajStrTokenDel(&handle);
     ajStrDel(&token);
diff -Naur EMBOSS-6.1.0/ajax/ajseqwrite.c EMBOSS-6.1.0-patched/ajax/ajseqwrite.c
--- EMBOSS-6.1.0/ajax/ajseqwrite.c	2009-07-06 11:50:31.000000000 -0400
+++ EMBOSS-6.1.0-patched/ajax/ajseqwrite.c	2009-07-30 11:13:23.000000000 -0400
@@ -231,7 +231,7 @@
 	 AJFALSE, AJFALSE, AJFALSE, AJTRUE,  AJTRUE,
 	 AJFALSE, AJTRUE,  AJFALSE, seqWriteGcg},
     {"gcg8",       "GCG old (version 8) sequence format",
-	 AJFALSE, AJFALSE, AJFALSE, AJTRUE,  AJTRUE,
+	 AJTRUE,  AJFALSE, AJFALSE, AJTRUE,  AJTRUE,
 	 AJFALSE, AJTRUE,  AJFALSE, seqWriteGcg}, /* alias for gcg */
     {"embl",       "EMBL entry format",
 	 AJFALSE, AJFALSE, AJFALSE, AJTRUE, AJFALSE,
@@ -240,7 +240,7 @@
 	 AJTRUE,  AJFALSE, AJFALSE, AJTRUE,  AJFALSE,
 	 AJTRUE,  AJTRUE,  AJFALSE, seqWriteEmbl},
     {"emblold",    "EMBL entry format (alias)",
-	 AJFALSE, AJFALSE, AJFALSE, AJTRUE,  AJFALSE,
+	 AJTRUE,  AJFALSE, AJFALSE, AJTRUE,  AJFALSE,
 	 AJTRUE,  AJTRUE,  AJFALSE, seqWriteEmbl}, /* embl pre-87 format*/
     {"em",         "EMBL entry format (alias)",
 	 AJTRUE,  AJFALSE, AJFALSE, AJTRUE,  AJFALSE,
@@ -255,7 +255,7 @@
 	 AJTRUE,  AJFALSE, AJFALSE, AJFALSE, AJTRUE,
 	 AJTRUE,  AJTRUE,  AJFALSE, seqWriteSwiss},
     {"swissold",   "Swissprot entry format", /* format before 2006 release */
-	 AJFALSE, AJFALSE, AJFALSE, AJFALSE, AJTRUE,
+	 AJTRUE,  AJFALSE, AJFALSE, AJFALSE, AJTRUE,
 	 AJTRUE,  AJTRUE,  AJFALSE, seqWriteSwiss},
     {"swissprotold","Swissprot entry format",
 	 AJTRUE,  AJFALSE, AJFALSE, AJFALSE, AJTRUE,
@@ -347,7 +347,7 @@
     {"msf",        "GCG MSF (mutiple sequence file) file format",
 	 AJFALSE, AJFALSE, AJTRUE,  AJTRUE,  AJTRUE,
 	 AJFALSE, AJTRUE,  AJFALSE, seqWriteMsf},
-    {"clustal",    "Clustalw output format",
+    {"clustal",    "Clustalw multiple alignment format",
 	 AJFALSE, AJFALSE, AJTRUE,  AJTRUE,  AJTRUE,
 	 AJFALSE, AJTRUE,  AJFALSE, seqWriteClustal},
     {"aln",        "Clustalw output format (alias)",
@@ -1360,15 +1360,11 @@
 
 static void seqWriteFastqSanger(AjPSeqout outseq)
 {
-    ajuint i;
     ajuint j;
     ajuint ilen;
-    ajuint jlen;
     AjPStr seq = NULL;
-    ajuint linelen     = 60;
-    ajuint iend;
     AjPStr db = NULL;
-    char qchar;
+    ajint qchar;
 
     ajStrAssignS(&db, outseq->Setoutdb);
     /* ajStrAssignEmptyS(&db, outseq->Db);*/
@@ -1394,37 +1390,27 @@
     ajFmtPrintF(outseq->File, "\n");
     ilen = ajStrGetLen(outseq->Seq);
 
-    for(i=0; i < ilen; i += linelen)
-    {
-	iend = AJMIN(ilen-1, i+linelen-1);
-	ajStrAssignSubS(&seq, outseq->Seq, i, iend);
-	ajFmtPrintF(outseq->File, "%S\n", seq);
-    }
-
-    ajFmtPrintF(outseq->File, "+%S", outseq->Name);
-
-    if(ajStrGetLen(outseq->Sv))
-	ajFmtPrintF(outseq->File, " %S", outseq->Sv);
-    else if(ajStrGetLen(outseq->Acc))
-	ajFmtPrintF(outseq->File, " %S", outseq->Acc);
+    ajFmtPrintF(outseq->File, "%S\n", outseq->Seq);
 
-    ajFmtPrintF(outseq->File, "\n");
+    ajFmtPrintF(outseq->File, "+\n");
 
     ilen = ajStrGetLen(outseq->Seq);
 
     if(outseq->Accuracy)
     {
-        for(i=0; i < ilen; i += linelen)
+        ajStrAssignClear(&seq);
+
+        for(j=0;j<ilen;j++)
         {
-            iend = AJMIN(ilen-1, i+linelen-1);
-            ajStrAssignClear(&seq);
-            for(j=i;j<=iend;j++)
-            {
-                qchar = 33 + (int) outseq->Accuracy[j];
-                ajStrAppendK(&seq, qchar);
-            }
-            ajFmtPrintF(outseq->File, "%S\n", seq);
+            qchar = 33 + (int) (0.5 + outseq->Accuracy[j]);
+            if(qchar > 126)
+                qchar = 126;
+            if(qchar < 33)
+                qchar = 33;
+            ajStrAppendK(&seq, (char) qchar);
         }
+
+        ajFmtPrintF(outseq->File, "%S\n", seq);
     }
 
     else 
@@ -1433,15 +1419,11 @@
         ** default to a score of 1 (0.75 error : 1 base in 4 is right)
         */
 
-        for(i=0; i < ilen; i += linelen)
-        {
-            iend = AJMIN(ilen-1, i+linelen-1);
-            jlen = (iend - i + 1);
-            ajStrAssignClear(&seq);
-            ajStrAppendCountK(&seq,'\"', jlen); 
-            ajFmtPrintF(outseq->File, "%S\n", seq);
-        }
+        ajStrAssignClear(&seq);
+
+        ajStrAppendCountK(&seq,'\"', ilen); 
 
+        ajFmtPrintF(outseq->File, "%S\n", seq);
     }
     
     ajStrDel(&seq);
@@ -1464,15 +1446,11 @@
 
 static void seqWriteFastqIllumina(AjPSeqout outseq)
 {
-    ajuint i;
     ajuint j;
     ajuint ilen;
-    ajuint jlen;
     AjPStr seq = NULL;
-    ajuint linelen     = 60;
-    ajuint iend;
     AjPStr db = NULL;
-    char qchar;
+    ajint qchar;
 
     ajStrAssignS(&db, outseq->Setoutdb);
     /* ajStrAssignEmptyS(&db, outseq->Db);*/
@@ -1498,37 +1476,26 @@
     ajFmtPrintF(outseq->File, "\n");
     ilen = ajStrGetLen(outseq->Seq);
 
-    for(i=0; i < ilen; i += linelen)
-    {
-	iend = AJMIN(ilen-1, i+linelen-1);
-	ajStrAssignSubS(&seq, outseq->Seq, i, iend);
-	ajFmtPrintF(outseq->File, "%S\n", seq);
-    }
-
-    ajFmtPrintF(outseq->File, "+%S", outseq->Name);
-
-    if(ajStrGetLen(outseq->Sv))
-	ajFmtPrintF(outseq->File, " %S", outseq->Sv);
-    else if(ajStrGetLen(outseq->Acc))
-	ajFmtPrintF(outseq->File, " %S", outseq->Acc);
-
-    ajFmtPrintF(outseq->File, "\n");
+    ajFmtPrintF(outseq->File, "%S\n", outseq->Seq);
+    ajFmtPrintF(outseq->File, "+\n");
 
     ilen = ajStrGetLen(outseq->Seq);
 
     if(outseq->Accuracy)
     {
-        for(i=0; i < ilen; i += linelen)
+        ajStrAssignClear(&seq);
+
+        for(j=0;j<ilen;j++)
         {
-            iend = AJMIN(ilen-1, i+linelen-1);
-            ajStrAssignClear(&seq);
-            for(j=i;j<=iend;j++)
-            {
-                qchar = 64 + (int) outseq->Accuracy[j];
-                ajStrAppendK(&seq, qchar);
-            }
-            ajFmtPrintF(outseq->File, "%S\n", seq);
+            qchar = 64 + (int) (0.5 + outseq->Accuracy[j]);
+            if(qchar > 126)
+                qchar = 126;
+            else if(qchar < 33)
+                qchar = 33;
+            ajStrAppendK(&seq, (char) qchar);
         }
+
+        ajFmtPrintF(outseq->File, "%S\n", seq);
     }
 
     else 
@@ -1537,15 +1504,11 @@
         ** default to a score of 1 (0.75 error : 1 base in 4 is right)
         */
 
-        for(i=0; i < ilen; i += linelen)
-        {
-            iend = AJMIN(ilen-1, i+linelen-1);
-            jlen = (iend - i + 1);
-            ajStrAssignClear(&seq);
-            ajStrAppendCountK(&seq,'A', jlen);
-            ajFmtPrintF(outseq->File, "%S\n", seq);
-        }
+        ajStrAssignClear(&seq);
+
+        ajStrAppendCountK(&seq,'A', ilen);
 
+        ajFmtPrintF(outseq->File, "%S\n", seq);
     }
     
     ajStrDel(&seq);
@@ -1568,15 +1531,11 @@
 
 static void seqWriteFastqSolexa(AjPSeqout outseq)
 {
-    ajuint i;
     ajuint j;
     ajuint ilen;
-    ajuint jlen;
     AjPStr seq = NULL;
-    ajuint linelen     = 60;
-    ajuint iend;
     AjPStr db = NULL;
-    char qchar;
+    ajint qchar;
     double sval;
     double pval;
     double qval;
@@ -1605,47 +1564,39 @@
     ajFmtPrintF(outseq->File, "\n");
     ilen = ajStrGetLen(outseq->Seq);
 
-    for(i=0; i < ilen; i += linelen)
-    {
-	iend = AJMIN(ilen-1, i+linelen-1);
-	ajStrAssignSubS(&seq, outseq->Seq, i, iend);
-	ajFmtPrintF(outseq->File, "%S\n", seq);
-    }
+    ajFmtPrintF(outseq->File, "%S\n", outseq->Seq);
 
-    ajFmtPrintF(outseq->File, "+%S", outseq->Name);
-
-    if(ajStrGetLen(outseq->Sv))
-	ajFmtPrintF(outseq->File, " %S", outseq->Sv);
-    else if(ajStrGetLen(outseq->Acc))
-	ajFmtPrintF(outseq->File, " %S", outseq->Acc);
-
-    ajFmtPrintF(outseq->File, "\n");
+    ajFmtPrintF(outseq->File, "+\n");
 
     ilen = ajStrGetLen(outseq->Seq);
 
     if(outseq->Accuracy)
     {
-        for(i=0; i < ilen; i += linelen)
+        ajStrAssignClear(&seq);
+
+        for(j=0;j<ilen;j++)
         {
-            iend = AJMIN(ilen-1, i+linelen-1);
-            ajStrAssignClear(&seq);
-            for(j=i;j<=iend;j++)
-            {
-                sval = outseq->Accuracy[j];
-                pval = 1.0 / pow(10.0, (sval/10.0));
+            sval = outseq->Accuracy[j];
+            pval = 1.0 / pow(10.0, (sval/10.0));
 
-                /*adjust zero phred score to 0.75 error rate */
-                if(pval == 1.0)
-                    pval = 0.75;
-
-                qval = 0.5 + -10.0 * log10(pval/(1.0 - pval));
-                qchar = 64 + (int) qval;
-                ajStrAppendK(&seq, qchar);
-                ajDebug("[%d] aval:%.4f qval:%.4f %d '%c'\n",
-                        j, sval, qval, (int)qval, qchar);
-            }
-            ajFmtPrintF(outseq->File, "%S\n", seq);
+            /*adjust zero phred score to 0.75 error rate */
+            if(pval > 0.75)
+                pval = 0.75;
+
+            qval = -10.0 * log10(pval/(1.0 - pval));
+            if(qval >= 0.0)
+                qchar = 64 + (int) (qval + 0.5);
+            else
+                qchar = 64 + (int) (qval - 0.5);
+            if(qchar > 126)
+                qchar = 126;
+            else if(qchar < 33)
+                qchar = 33;
+            ajStrAppendK(&seq, (char) qchar);
+            ajDebug("[%d] aval:%.4f qval:%.4f %d '%c'\n",
+                    j, sval, qval, (int)qval, qchar);
         }
+        ajFmtPrintF(outseq->File, "%S\n", seq);
     }
 
     else 
@@ -1654,15 +1605,11 @@
         ** default to a score of -5 (0.75 error : 1 base in 4 is right)
         */
 
-        for(i=0; i < ilen; i += linelen)
-        {
-            iend = AJMIN(ilen-1, i+linelen-1);
-            jlen = (iend - i + 1);
-            ajStrAssignClear(&seq);
-            ajStrAppendCountK(&seq,';', jlen); 
-            ajFmtPrintF(outseq->File, "%S\n", seq);
-        }
+        ajStrAssignClear(&seq);
+        
+        ajStrAppendCountK(&seq,';', ilen); 
 
+        ajFmtPrintF(outseq->File, "%S\n", seq);
     }
     
     ajStrDel(&seq);
@@ -7798,6 +7745,7 @@
 
 
 
+
 /* @funcstatic seqSeqFormat ***************************************************
 **
 ** Initialises sequence output formatting parameters.
diff -Naur EMBOSS-6.1.0/jemboss/org/emboss/jemboss/gui/Browser.java EMBOSS-6.1.0-patched/jemboss/org/emboss/jemboss/gui/Browser.java
--- EMBOSS-6.1.0/jemboss/org/emboss/jemboss/gui/Browser.java	2009-06-30 12:18:26.000000000 -0400
+++ EMBOSS-6.1.0-patched/jemboss/org/emboss/jemboss/gui/Browser.java	2009-07-30 11:13:23.000000000 -0400
@@ -68,7 +68,10 @@
 import javax.swing.tree.DefaultMutableTreeNode;
 import javax.swing.tree.TreeSelectionModel;
 
+import org.emboss.jemboss.Jemboss;
 import org.emboss.jemboss.JembossParams;
+import org.emboss.jemboss.server.JembossServer;
+import org.emboss.jemboss.soap.GetVersion;
 
 
 /**
@@ -110,6 +113,8 @@
   private JMenuItem fwdMenu;
   /** Forward button option */
   private JButton fwdBt;
+  /** version of EMBOSS programs */
+  String embossVersion = "6.1";
 
   /**
   *
@@ -178,8 +183,11 @@
       URL pageURL = new URL(initialURL);
       setURL(pageURL,initialURL);
     }
+    if (Jemboss.withSoap){
+    	embossVersion = GetVersion.getVersion(mysettings);
+    }
   }
-
+  
 
   /**
   *
@@ -712,8 +720,13 @@
             loc = "http://emboss.sourceforge.net/Jemboss/";
           else if(selectedValue.equals("Home") && category.equals("EMBOSS"))
             loc = "http://emboss.sourceforge.net/";
-          else if(selectedValue.equals("Apps") && category.equals("EMBOSS"))
-            loc = "http://emboss.sourceforge.net/apps/release/6.0/emboss/apps/";
+          else if(selectedValue.equals("Apps") && category.equals("EMBOSS")){
+        	if (!JembossParams.isJembossServer()){
+        	  	embossVersion = new JembossServer().version();
+        	  	embossVersion = embossVersion.split("\\.")[0]+"."+embossVersion.split("\\.")[1];
+        	}
+            loc = "http://emboss.sourceforge.net/apps/release/"+embossVersion+"/emboss/apps/";
+          }
 
           if(loc != null) 
           {
diff -Naur EMBOSS-6.1.0/jemboss/org/emboss/jemboss/gui/BuildProgramMenu.java EMBOSS-6.1.0-patched/jemboss/org/emboss/jemboss/gui/BuildProgramMenu.java
--- EMBOSS-6.1.0/jemboss/org/emboss/jemboss/gui/BuildProgramMenu.java	2009-07-08 04:45:48.000000000 -0400
+++ EMBOSS-6.1.0-patched/jemboss/org/emboss/jemboss/gui/BuildProgramMenu.java	2009-07-30 11:13:23.000000000 -0400
@@ -141,6 +141,22 @@
           return woss;
       }
 
+      private synchronized void  updateConnectionSettings(){      
+		  splashing.doneSomething("Cannot connect!");
+		  ServerSetup ss = new ServerSetup(mysettings);
+		  splashThread.setInterval(100000);
+		  int sso = JOptionPane.showConfirmDialog(f,ss,
+				  "Check Settings",
+				  JOptionPane.OK_CANCEL_OPTION,
+				  JOptionPane.ERROR_MESSAGE,null);
+		  if(sso == JOptionPane.OK_OPTION){
+			  ss.setNewSettings();
+			  splashThread.setInterval(80);
+		  }
+		  else
+			  System.exit(0);
+	  }
+
       private void constructWithSoap(){
 
           mainMenu.setEnableFileManagers(false);
@@ -157,93 +173,65 @@
             mysettings.updateJembossPropStrings(settings);
           }
 
-          SwingWorker databaseworker = new SwingWorker()
+          try
           {
-            public Object construct()
-            {
-              ShowDB showdb = null;
-              try
-              {
-                showdb  = new ShowDB(mysettings);
-              }
-              catch (Exception ex)
-              {
-                splashing.doneSomething("Cannot connect!");
-                ServerSetup ss = new ServerSetup(mysettings);
-                int sso = JOptionPane.showConfirmDialog(f,ss,
-                           "Check Settings",
-                           JOptionPane.OK_CANCEL_OPTION,
-                           JOptionPane.ERROR_MESSAGE,null);
-                if(sso == JOptionPane.OK_OPTION)
-                  ss.setNewSettings();
-                else
-                  System.exit(0);
+        	  Hashtable hwoss = (new JembossJarUtil("resources/wossname.jar")).getHash();
+        	  if(hwoss.containsKey("wossname.out"))
+        		  woss = new String((byte[])hwoss.get("wossname.out"));
+          }
+          catch (Exception ex){}
+          if(woss.equals(""))
+          {
+        	  boolean connectionmade = false;
+        	  while (!connectionmade){
+        		  try
+        		  {
+        			  GetWossname ewoss = new GetWossname(mysettings);
+        			  woss = ewoss.getWossnameText(); 
+        			  mainMenu.setEnableFileManagers(true);
+        			  mainMenu.setEnableShowResults(true);
+        		  }
+        		  catch (JembossSoapException ex){
+        			  updateConnectionSettings();
+        			  continue;
+        		  }
+        		  connectionmade = true;
+        	  }
+          }
 
-                try
-                {
-                  showdb  = new ShowDB(mysettings);
-                }
-                catch (Exception exp)
-                {
-                  exp.printStackTrace();
-                }
-              }
-              String showdbOut = showdb.getDBText();
+          SwingWorker databaseworker = new SwingWorker()
+          {
 
-              Database d = new Database(showdbOut);
-              db = d.getDB();
-              mainMenu.setEnableFileManagers(true);
-              mainMenu.setEnableShowResults(true);
-              splashing.doneSomething("");
-              splashThread.setInterval(0);
-
-              matrices = showdb.getMatrices();  // get the available matrices
-              codons   = showdb.getCodonUsage();
-
-              /*JLabel jl = */new JLabel("<html>"); // not used but speeds first
-                                                // ACD form loading which
-                                                // uses html
-              return null;
-            }
+        	  public Object construct() {
+        		  boolean connectionmade = false;
+        		  while (!connectionmade) {
+        			  ShowDB showdb = null;
+        			  try {
+        				  showdb = new ShowDB(mysettings);
+        				  String showdbOut = showdb.getDBText();
+
+        				  Database d = new Database(showdbOut);
+        				  db = d.getDB();
+        				  mainMenu.setEnableFileManagers(true);
+        				  mainMenu.setEnableShowResults(true);
+        				  splashing.doneSomething("");
+        				  splashThread.setInterval(0);
+
+        				  matrices = showdb.getMatrices();
+        				  codons = showdb.getCodonUsage();
+        			  } catch (JembossSoapException ex) {
+        				  updateConnectionSettings();
+        				  continue;
+        			  }
+        			  connectionmade = true;
+        		  }
+        		  return null;
+        	  }
           };
           databaseworker.start();
-          
           splashing.doneSomething("");
-
-          try
-          {
-            try
-            {
-              Hashtable hwoss = (new JembossJarUtil("resources/wossname.jar")).getHash();
-              if(hwoss.containsKey("wossname.out"))
-                woss = new String((byte[])hwoss.get("wossname.out"));
-            }
-            catch (Exception ex){}
-
-            if(woss.equals(""))
-            {
-              GetWossname ewoss = new GetWossname(mysettings);
-              woss = ewoss.getWossnameText(); 
-              mainMenu.setEnableFileManagers(true);
-              mainMenu.setEnableShowResults(true);
-            }
-            
-            splashing.doneSomething("");
-          } 
-          catch(Exception e)
-          {
-            splashing.doneSomething("Cannot connect!");
-            ServerSetup ss = new ServerSetup(mysettings);
-            int sso = JOptionPane.showConfirmDialog(f,ss,
-                           "Check Settings",
-                           JOptionPane.OK_CANCEL_OPTION,
-                           JOptionPane.ERROR_MESSAGE,null);
-            if(sso == JOptionPane.OK_OPTION)
-              ss.setNewSettings();
-            else
-              System.exit(0);
-          }        
       }
+      
             
       private void constructWithoutSoap(){
 
@@ -310,7 +298,7 @@
         progs = new ProgList(woss,menuBar);
 
         if(withSoap)
-          splashing.doneSomething("");
+          splashing.doneEverything("");
 
         int npG = progs.getNumPrimaryGroups();
         menuBar.setLayout(new GridLayout(npG,1));
diff -Naur EMBOSS-6.1.0/jemboss/org/emboss/jemboss/gui/form/BuildJembossForm.java EMBOSS-6.1.0-patched/jemboss/org/emboss/jemboss/gui/form/BuildJembossForm.java
--- EMBOSS-6.1.0/jemboss/org/emboss/jemboss/gui/form/BuildJembossForm.java	2009-07-13 12:24:04.000000000 -0400
+++ EMBOSS-6.1.0-patched/jemboss/org/emboss/jemboss/gui/form/BuildJembossForm.java	2009-07-30 11:13:23.000000000 -0400
@@ -100,8 +100,7 @@
   private boolean withSoap;
   private JFrame f;
   private JPanel p2;
-  private String embossBin;
-
+  
   private int numofFields;
   private JembossParams mysettings;
   
@@ -125,7 +124,6 @@
     this.mysettings = mysettings;
     this.withSoap = withSoap;
 
-    embossBin  = mysettings.getEmbossBin();
     this.envp = envp;
     this.applName = applName;
 
@@ -172,10 +170,12 @@
         {
           String urlEmbassyPrefix = parseAcd.getUrlPrefix();
           url = mysettings.getembURL();
+          String version = GetVersion.getVersion(mysettings);
           if(urlEmbassyPrefix != null)
-            url = url + "apps/release/6.0/embassy/" +applName+ "/" ;
+        	 // get the version from server
+            url = url + "apps/release/"+version+"/embassy/"+urlEmbassyPrefix+"/" ;
           else
-            url = url + "apps/release/6.0/emboss/apps/";
+            url = url + "apps/release/"+version+"/emboss/apps/";
 
           url = url+applName+".html";
         }
@@ -893,8 +893,9 @@
       }
       else if ( att.startsWith("float") )
       {
-        if( (textFloat[h].getText() != null) && textFloat[h].isVisible()
-                                             && textFloat[h].isEnabled())
+        if( textFloat[h].getText() != null &&
+        		textFloat[h].getText().length()>0 &&
+        		textFloat[h].isVisible() && textFloat[h].isEnabled())
         {
           options = options.concat(" -" + val + " " + textFloat[h].getValue());
           optionsA.add("-"+val);
diff -Naur EMBOSS-6.1.0/jemboss/org/emboss/jemboss/gui/sequenceChooser/InputSequenceAttributes.java EMBOSS-6.1.0-patched/jemboss/org/emboss/jemboss/gui/sequenceChooser/InputSequenceAttributes.java
--- EMBOSS-6.1.0/jemboss/org/emboss/jemboss/gui/sequenceChooser/InputSequenceAttributes.java	2009-07-13 12:23:30.000000000 -0400
+++ EMBOSS-6.1.0-patched/jemboss/org/emboss/jemboss/gui/sequenceChooser/InputSequenceAttributes.java	2009-07-30 11:13:23.000000000 -0400
@@ -281,7 +281,7 @@
 //  bx.add(Box.createHorizontalGlue());
 
 
-    UFO = new JTextField();                     //sufo
+    UFO = new JTextField();                     //ufo
     UFO.setPreferredSize(new Dimension(100, 30));
     UFO.setMaximumSize(new Dimension(100, 30));
 
@@ -433,8 +433,8 @@
 
   /**
   *
-  * Determine if there is a default -sufo value
-  * @return     true if -sufo is set
+  * Determine if there is a default -ufo value
+  * @return     true if -ufo is set
   *
   */
   public boolean isUFODefault()
@@ -508,7 +508,7 @@
 //                                     sID.getText());
 
     if(!isUFODefault())
-      options = options.concat(" -sufo" + seq + " " +
+      options = options.concat(" -ufo" + seq + " " +
                                        UFO.getText());
 
     return options;
@@ -559,7 +559,7 @@
     }
 
     if(!isUFODefault()){
-      optionsA.add("-sufo"+ seq);
+      optionsA.add("-ufo"+ seq);
       optionsA.add(UFO.getText());
     }
 
diff -Naur EMBOSS-6.1.0/jemboss/org/emboss/jemboss/soap/GetVersion.java EMBOSS-6.1.0-patched/jemboss/org/emboss/jemboss/soap/GetVersion.java
--- EMBOSS-6.1.0/jemboss/org/emboss/jemboss/soap/GetVersion.java	1969-12-31 19:00:00.000000000 -0500
+++ EMBOSS-6.1.0-patched/jemboss/org/emboss/jemboss/soap/GetVersion.java	2009-07-30 11:13:23.000000000 -0400
@@ -0,0 +1,82 @@
+/********************************************************************
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Library General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Library General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Library General Public
+ *  License along with this library; if not, write to the
+ *  Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ *  Boston, MA  02111-1307, USA.
+ *
+ *  @author: Copyright (C) Tim Carver
+ *
+ ********************************************************************/
+
+package org.emboss.jemboss.soap;
+
+import javax.xml.namespace.QName;
+
+import org.apache.axis.client.Call;
+import org.emboss.jemboss.JembossParams;
+
+/**
+ * 
+ * Get the version of EMBOSS programs used by the Jemboss server connected
+ * 
+ */
+/**
+ * @author uludag
+ *
+ */
+public class GetVersion {
+
+	private static String version = null;
+
+	/**
+	 * returns complete version string of EMBOSS connected Jemboss server is using
+	 * @param mysettings
+	 *            jemboss properties
+	 * 
+	 */
+	public static String getVersionComplete(JembossParams mysettings) {
+
+		if (version != null)
+			return version;
+
+		if (!JembossParams.isJembossServer())
+			return null;
+		try {
+			String endpoint = mysettings.getPublicSoapURL();
+			org.apache.axis.client.Service serv = new org.apache.axis.client.Service();
+
+			Call call = (Call) serv.createCall();
+			call.setTargetEndpointAddress(new java.net.URL(endpoint));
+			call.setOperationName(new QName(mysettings.getPublicSoapService(),
+					"version"));
+			call.setReturnType(org.apache.axis.Constants.SOAP_STRING);
+			version = (String) call.invoke(new Object[] {});
+		} catch (Exception jse) {
+			jse.printStackTrace();
+		}
+		return version;
+	}
+	
+	
+	/**
+	 * returns firt two digits of EMBOSS version connected server has
+	 * @param mysettings
+	 * @return version of EMBOSS programs such as 5.0, or 6.1
+	 */
+	public static String getVersion(JembossParams mysettings){
+		getVersionComplete(mysettings);
+		return version.split("\\.")[0]+"."+version.split("\\.")[1];        
+	}
+
+}
diff -Naur EMBOSS-6.1.0/jemboss/org/emboss/jemboss/soap/ShowDB.java EMBOSS-6.1.0-patched/jemboss/org/emboss/jemboss/soap/ShowDB.java
--- EMBOSS-6.1.0/jemboss/org/emboss/jemboss/soap/ShowDB.java	2008-06-12 04:19:07.000000000 -0400
+++ EMBOSS-6.1.0-patched/jemboss/org/emboss/jemboss/soap/ShowDB.java	2009-07-30 11:13:23.000000000 -0400
@@ -50,15 +50,12 @@
   * @param mysettings 	jemboss properties
   *
   */
-  public ShowDB(JembossParams mysettings)
+  public ShowDB(JembossParams mysettings) throws JembossSoapException
   {
  
     PublicRequest dbReq = null;   
-    try
-    {
-      dbReq = new PublicRequest(mysettings, "show_db");
-    }
-    catch (JembossSoapException jse) {}
+
+    dbReq = new PublicRequest(mysettings, "show_db");
 
     statusmsg = dbReq.getVal("msg");
     status = dbReq.getVal("status");
diff -Naur EMBOSS-6.1.0/jemboss/utils/install-jemboss-server.sh EMBOSS-6.1.0-patched/jemboss/utils/install-jemboss-server.sh
--- EMBOSS-6.1.0/jemboss/utils/install-jemboss-server.sh	2009-06-30 04:31:50.000000000 -0400
+++ EMBOSS-6.1.0-patched/jemboss/utils/install-jemboss-server.sh	2009-07-30 11:13:23.000000000 -0400
@@ -694,25 +694,16 @@
 echo "         EMBOSS and Jemboss Server installation script"
 echo "--------------------------------------------------------------"
 echo " "
-echo "Note: any default values are given in square brackets []. "
+echo "Note: any default values are given in square brackets []."
 echo " "
-echo "There are two types of installation see details at: "
+echo "This script installs EMBOSS as well as Jemboss."
+echo "Jemboss is deployed as a Java web application in your tomcat server."
+echo "A script is prepared to run Jemboss client that by default uses the"
+echo "above Jemboss web application."
+echo
+echo "For detailed information on installing Jemboss see: "
 echo "http://emboss.sourceforge.net/Jemboss/install/setup.html"
-echo " "
-echo "(1) CLIENT-SERVER"
-echo "(2) STANDALONE"
-echo "Enter type of installation [1] :"
-read INSTALL_TYPE
-
-if (test "$INSTALL_TYPE" != "1") && (test "$INSTALL_TYPE" != "2"); then
-  INSTALL_TYPE="1"
-fi
-clear
-
 echo
-echo "--------------------------------------------------------------"
-echo "         EMBOSS and Jemboss Server installation script"
-echo "--------------------------------------------------------------"
 echo
 echo "*** This script needs to be run with permissions to be able"
 echo "*** to install EMBOSS in the required directories. This may"
@@ -721,15 +712,11 @@
 echo "Before running this script you should download the latest:"
 echo
 echo "(1) EMBOSS release (contains Jemboss) ftp://emboss.open-bio.org/pub/EMBOSS/"
-
-
-if [ $INSTALL_TYPE = "1" ]; then
-  echo "(2) Tomcat release http://jakarta.apache.org/site/binindex.html"
-  echo "(3) Apache AXIS (SOAP) release   http://xml.apache.org/axis/"
-fi
+echo "(2) Tomcat 5.5 series release http://tomcat.apache.org/"
+echo "(3) Apache AXIS (SOAP) release 1.4   http://ws.apache.org/axis/"
   
 echo
-echo "Has the above been downloaded (y/n)? "
+echo "Have the above been downloaded (y/n)? "
 read DOWNLOADED
 
 if (test "$DOWNLOADED" != "y") && (test "$DOWNLOADED" != "Y"); then
@@ -831,6 +818,9 @@
 fi
 
 SSL="y"
+
+# keep the following variable for now so we have cvs compare for a while
+INSTALL_TYPE="1";
 if [ $INSTALL_TYPE = "1" ]; then
 #
 # localhost name
@@ -1271,17 +1261,6 @@
 #cd $EMBOSS_INSTALL/share/EMBOSS/jemboss
 JEMBOSS=$EMBOSS_INSTALL/share/EMBOSS/jemboss
 
-#---------------------------------------------------------------------------------
-# Exit for standalone installations
-#---------------------------------------------------------------------------------
-if [ $INSTALL_TYPE = "2" ]; then
-  echo
-  echo "To run Jemboss:"
-  echo "cd $JEMBOSS"
-  echo "./runJemboss.sh"
-  echo
-  exit 0
-fi
 
 
 #
