diff -Naur --exclude='*~' E/Makefile E.fink/Makefile
--- E/Makefile	2011-06-25 09:06:19.000000000 -0700
+++ E.fink/Makefile	2011-07-13 09:23:34.000000000 -0700
@@ -133,12 +133,12 @@
 
 install: E
 	-sh -c 'mkdir -p $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/eprover $(EXECPATH) ' 
-	-sh -c 'development_tools/e_install PROVER/epclextract $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/eproof $(EXECPATH)'
-	-sh -c 'development_tools/e_install PROVER/eground $(EXECPATH)'	
-	-sh -c 'development_tools/e_install PROVER/e_ltb_runner $(EXECPATH)'	
-	-sh -c 'development_tools/e_install PROVER/e_axfilter $(EXECPATH)'	
+	-sh -c 'development_tools/e_install PROVER/eprover $(DESTDIR)$(EXECPATH) ' 
+	-sh -c 'development_tools/e_install PROVER/epclextract $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/eproof $(DESTDIR)$(EXECPATH)'
+	-sh -c 'development_tools/e_install PROVER/eground $(DESTDIR)$(EXECPATH)'	
+	-sh -c 'development_tools/e_install PROVER/e_ltb_runner $(DESTDIR)$(EXECPATH)'	
+	-sh -c 'development_tools/e_install PROVER/e_axfilter $(DESTDIR)$(EXECPATH)'	
 	-sh -c 'mkdir -p $(MANPATH)'
 	-sh -c 'development_tools/e_install DOC/man/eprover.1 $(MANPATH)'
 	-sh -c 'development_tools/e_install DOC/man/epclextract.1 $(MANPATH)'
