diff -Nurd -x'*~' genius-1.0.1.orig/configure genius-1.0.1/configure
--- genius-1.0.1.orig/configure	2007-11-15 22:52:16.000000000 -0500
+++ genius-1.0.1/configure	2007-11-17 17:34:32.000000000 -0500
@@ -23920,6 +23920,7 @@
 
   if test "x$use_gtksourceview" = "xyes" ; then
 
+if false; then  # skip gtksourceview2 detection
 pkg_failed=no
 { echo "$as_me:$LINENO: checking for GSV" >&5
 echo $ECHO_N "checking for GSV... $ECHO_C" >&6; }
@@ -23990,7 +23991,7 @@
 echo "${ECHO_T}yes" >&6; }
 	have_gtksourceview2=yes
 fi
-
+fi  # end skip of gtksourceview2 detection
     if test "x$have_gtksourceview2" != "xyes"; then
 
 pkg_failed=no
diff -Nurd -x'*~' genius-1.0.1.orig/ve/Makefile.in genius-1.0.1/ve/Makefile.in
--- genius-1.0.1.orig/ve/Makefile.in	2007-11-17 00:14:48.000000000 -0500
+++ genius-1.0.1/ve/Makefile.in	2007-11-17 18:05:21.000000000 -0500
@@ -63,7 +63,7 @@
 am_test_ve_config_OBJECTS = test-ve-config.$(OBJEXT)
 test_ve_config_OBJECTS = $(am_test_ve_config_OBJECTS)
 am__DEPENDENCIES_1 =
-test_ve_config_DEPENDENCIES = $(am__DEPENDENCIES_1) \
+test_ve_config_DEPENDENCIES = $(top_builddir)/ve/libvicious.a $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 DEFAULT_INCLUDES = -I. -I$(top_builddir)@am__isrc@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
@@ -317,11 +317,10 @@
 	test-ve-config.c
 
 test_ve_config_LDADD = \
+	$(top_builddir)/ve/libvicious.a		\
 	$(VICIOUS_LIBS)				\
 	$(INTLLIBS)				\
-	$(GLIB_LIBS)				\
-	-L$(top_builddir)/ve			\
-	-lvicious
+	$(GLIB_LIBS)
 
 all: all-recursive
 
