diff -Nurd -x'*~' ghemical-1.01.orig/openbabel/Makefile.in ghemical-1.01/openbabel/Makefile.in
--- ghemical-1.01.orig/openbabel/Makefile.in	2002-06-19 17:18:59.000000000 -0400
+++ ghemical-1.01/openbabel/Makefile.in	2007-10-14 23:01:19.000000000 -0400
@@ -8,7 +8,7 @@
 CXX	      = @CXX@
 
 CFLAGS        = @CFLAGS@
-CXXFLAGS      = @CXXFLAGS@ -DDATADIR=\"${pkgdatadir}\" -I$(builddir)
+CXXFLAGS      = -DDATADIR=\"${pkgdatadir}\" -I$(builddir) @CXXFLAGS@
 LIBS	      = @LIBS@
 LIBFLAGS      = -DLIBRARY
 
diff -Nurd -x'*~' ghemical-1.01.orig/openbabel/fileformat.h ghemical-1.01/openbabel/fileformat.h
--- ghemical-1.01.orig/openbabel/fileformat.h	2002-06-19 17:19:00.000000000 -0400
+++ ghemical-1.01/openbabel/fileformat.h	2007-10-14 23:00:56.000000000 -0400
@@ -122,7 +122,7 @@
 bool WriteQChem(std::ostream &,OBMol &);
 bool WriteReport(std::ostream &,OBMol &);
 bool WriteSDFile(std::ostream &,OBMol &,char *dimension="3D");
-bool WriteSmiles(std::ostream &,OBMol &,char *title=NULL);
+bool WriteSmiles(std::ostream &,OBMol &,char *title=(char*)NULL);
 //test
 bool WriteSmiOrderedMol2(std::ostream &,OBMol &,char *dimension="3D");
 //test
diff -Nurd -x'*~' ghemical-1.01.orig/src/common/Makefile.in ghemical-1.01/src/common/Makefile.in
--- ghemical-1.01.orig/src/common/Makefile.in	2002-06-19 17:19:02.000000000 -0400
+++ ghemical-1.01/src/common/Makefile.in	2007-10-14 23:23:27.000000000 -0400
@@ -47,7 +47,7 @@
 #########################################################################
 
 %.o:		%.cpp
-		$(CXX) $(DEFINES) -c $(CXXFLAGS) $(INCS) $(srcdir)/$(@:.o=.cpp)
+		$(CXX) $(DEFINES) -c $(INCS) $(CXXFLAGS) $(srcdir)/$(@:.o=.cpp)
 
 all:		$(TARGET)
 
@@ -60,7 +60,7 @@
 
 depend:
 		-@for i in $(SRC); do \
-			$(CXX) -c -MM $(CXXFLAGS) $(INCS) $(srcdir)/$$i >> depend; \
+			$(CXX) -c -MM $(INCS) $(CXXFLAGS) $(srcdir)/$$i >> depend; \
 		done && test -z "$$fail"
 
 distclean:	clean
diff -Nurd -x'*~' ghemical-1.01.orig/src/common/utility.h ghemical-1.01/src/common/utility.h
--- ghemical-1.01.orig/src/common/utility.h	2001-07-24 12:56:37.000000000 -0400
+++ ghemical-1.01/src/common/utility.h	2007-10-14 23:00:56.000000000 -0400
@@ -149,7 +149,7 @@
 	
 	public:
 	
-	singleton_cleaner(TYPE1 * p1 = NULL) { instance = p1; }
+	singleton_cleaner(TYPE1 * p1 = (TYPE1 *)NULL) { instance = p1; }
 	~singleton_cleaner(void) { if (instance) delete instance; }
 	
 	bool SetInstance(TYPE1 * p1) { if (instance) return false; instance = p1; return true; }
diff -Nurd -x'*~' ghemical-1.01.orig/src/graphics/Makefile.in ghemical-1.01/src/graphics/Makefile.in
--- ghemical-1.01.orig/src/graphics/Makefile.in	2002-06-19 17:19:02.000000000 -0400
+++ ghemical-1.01/src/graphics/Makefile.in	2007-10-14 23:23:12.000000000 -0400
@@ -44,7 +44,7 @@
 ###########################################################################
 
 %.o:		%.cpp
-		$(CXX) $(DEFINES) -c $(CXXFLAGS) $(INCS) $(srcdir)/$(@:.o=.cpp)
+		$(CXX) $(DEFINES) -c $(INCS) $(CXXFLAGS) $(srcdir)/$(@:.o=.cpp)
 
 all:		$(TARGET)
 
@@ -57,7 +57,7 @@
 
 depend:
 		-@for i in $(SRC); do \
-			$(CXX) -c -MM $(CXXFLAGS) $(INCS) $(srcdir)/$$i >> depend; \
+			$(CXX) -c -MM $(INCS) $(CXXFLAGS) $(srcdir)/$$i >> depend; \
 		done && test -z "$$fail"
 
 distclean:	clean
diff -Nurd -x'*~' ghemical-1.01.orig/src/graphics/objects.h ghemical-1.01/src/graphics/objects.h
--- ghemical-1.01.orig/src/graphics/objects.h	2001-12-29 09:56:49.000000000 -0500
+++ ghemical-1.01/src/graphics/objects.h	2007-10-14 23:00:56.000000000 -0400
@@ -177,7 +177,7 @@
 {
 	protected:
 	
-	friend obj_loc_data * dummy_object::GetLocDataRW(void) const;
+	friend class dummy_object;
 	
 	obj_loc_data * data;
 	
diff -Nurd -x'*~' ghemical-1.01.orig/src/target3/Makefile.in ghemical-1.01/src/target3/Makefile.in
--- ghemical-1.01.orig/src/target3/Makefile.in	2002-06-17 05:52:20.000000000 -0400
+++ ghemical-1.01/src/target3/Makefile.in	2007-10-14 23:21:44.000000000 -0400
@@ -33,7 +33,7 @@
 		@GNOMEUI_LIBS@ \
 		@LIBGLADE_LIBS@ \
 		@SC_LIBS@
-		
+
 GHEMICAL_LIBS=	$(top_builddir)/src/target3/glade/libdialogs.a \
 		$(top_builddir)/src/graphics/libgraphics.a \
 		$(top_builddir)/src/common/libcommon.a \
@@ -65,7 +65,7 @@
 		$(CXX) -o $(TARGET) $(OBJ) $(GHEMICAL_LIBS) $(LIBS)
 
 %.o:		%.cpp
-		$(CXX) $(DEFINES) -c $(CXXFLAGS) $(INCS) $(srcdir)/$(@:.o=.cpp)
+		$(CXX) $(DEFINES) -c $(INCS) $(CXXFLAGS) $(srcdir)/$(@:.o=.cpp)
 
 all:		$(TARGET)
 
@@ -73,7 +73,7 @@
 
 depend:
 		-@for i in $(SRC); do \
-			$(CXX) -c -MM $(CXXFLAGS) $(INCS) $(srcdir)/$$i >> depend; \
+			$(CXX) -c -MM $(INCS) $(CXXFLAGS) $(srcdir)/$$i >> depend; \
 		done
 
 distclean:	clean
diff -Nurd -x'*~' ghemical-1.01.orig/src/target3/glade/Makefile.in ghemical-1.01/src/target3/glade/Makefile.in
--- ghemical-1.01.orig/src/target3/glade/Makefile.in	2002-06-07 06:54:57.000000000 -0400
+++ ghemical-1.01/src/target3/glade/Makefile.in	2007-10-14 23:22:44.000000000 -0400
@@ -33,7 +33,7 @@
 SRC =		geomopt_dialog.cpp moldyn_dialog.cpp trajview_dialog.cpp \
 		file_import_dialog.cpp file_export_dialog.cpp \
 		setup_qm1_dialog.cpp setup_mm1_dialog.cpp
-		
+
 OBJ =		$(SRC:.cpp=.o)
 
 TARGET=		libdialogs.a
@@ -41,7 +41,7 @@
 ###########################################################################
 
 %.o:		%.cpp
-		$(CXX) $(DEFINES) -c $(CXXFLAGS) $(INCS) $(srcdir)/$(@:.o=.cpp)
+		$(CXX) $(DEFINES) -c $(INCS) $(CXXFLAGS) $(srcdir)/$(@:.o=.cpp)
 
 all:		$(TARGET)
 
@@ -54,7 +54,7 @@
 
 depend:
 		-@for i in $(SRC); do \
-			$(CXX) -c -MM $(CXXFLAGS) $(INCS) $(srcdir)/$$i >> depend; \
+			$(CXX) -c -MM $(INCS) $(CXXFLAGS) $(srcdir)/$$i >> depend; \
 		done && test -z "$$fail"
 
 distclean:	clean
