diff -ruN --exclude='*~' gildas-src-dec08b/admin/Makefile gildas-src-dec08b.patched/admin/Makefile
--- gildas-src-dec08b/admin/Makefile	2007-12-12 11:23:47.000000000 +0100
+++ gildas-src-dec08b.patched/admin/Makefile	2008-12-06 00:17:24.000000000 +0100
@@ -33,31 +33,31 @@
 integ: integ-message
 
 integ-message:
-	@(echo)
-	@(echo "Compilation succeeded! Bye.")
-	@(echo)
+#	@(echo)
+#	@(echo "Compilation succeeded! Bye.")
+#	@(echo)
 
 default-install: install-message
 
 install-message:
-	@(echo)
-	@(echo "***************************************************************************")
-	@(echo "Installation succeeded!")
-	@(echo)
-	@(echo "bash users: You should add in your ~/.bash_profile the following lines")
-	@(echo "    export GAG_ROOT_DIR=$(gagexedir)")
-	@(echo "    export GAG_EXEC_SYSTEM=$(GAG_EXEC_SYSTEM)")
-	@(echo '    source $$GAG_ROOT_DIR/etc/bash_profile')
-	@(echo)
-	@(echo "csh users:  You should add in your ~/.login the following lines")
-	@(echo "    setenv GAG_ROOT_DIR $(gagexedir)")
-	@(echo "    setenv GAG_EXEC_SYSTEM $(GAG_EXEC_SYSTEM)")
-	@(echo '    source $$GAG_ROOT_DIR/etc/login')
-	@(echo)
-	@(echo "You should then exit this xterm and open a new one to reset your path.")
-	@(echo "GILDAS should be ready for use (e.g. type greg @ gag_demo:demo).")
-	@(echo "Bye.")
-	@(echo "***************************************************************************")
-	@(echo)
+#	@(echo)
+#	@(echo "***************************************************************************")
+#	@(echo "Installation succeeded!")
+#	@(echo)
+#	@(echo "bash users: You should add in your ~/.bash_profile the following lines")
+#	@(echo "    export GAG_ROOT_DIR=$(gagexedir)")
+#	@(echo "    export GAG_EXEC_SYSTEM=$(GAG_EXEC_SYSTEM)")
+#	@(echo '    source $$GAG_ROOT_DIR/etc/bash_profile')
+#	@(echo)
+#	@(echo "csh users:  You should add in your ~/.login the following lines")
+#	@(echo "    setenv GAG_ROOT_DIR $(gagexedir)")
+#	@(echo "    setenv GAG_EXEC_SYSTEM $(GAG_EXEC_SYSTEM)")
+#	@(echo '    source $$GAG_ROOT_DIR/etc/login')
+#	@(echo)
+#	@(echo "You should then exit this xterm and open a new one to reset your path.")
+#	@(echo "GILDAS should be ready for use (e.g. type greg @ gag_demo:demo).")
+#	@(echo "Bye.")
+#	@(echo "***************************************************************************")
+#	@(echo)
 
 ###########################################################################
diff -ruN --exclude='*~' gildas-src-dec08b/admin/Makefile.def gildas-src-dec08b.patched/admin/Makefile.def
--- gildas-src-dec08b/admin/Makefile.def	2008-11-26 16:38:00.000000000 +0100
+++ gildas-src-dec08b.patched/admin/Makefile.def	2008-12-06 14:22:47.000000000 +0100
@@ -542,7 +542,7 @@
 ###########################################################################
 # Kernel libraries
 
-LEGACY_LIBS = -lslatec -llapack -lblas
+LEGACY_LIBS = -lslatec -llapack -lf77blas
 GAG_LIBS = -lggui -lgcore -lgwcs -lgio -lgsys -lginc
 
 GUI_LIBS =
diff -ruN --exclude='*~' gildas-src-dec08b/admin/define-system.sh gildas-src-dec08b.patched/admin/define-system.sh
--- gildas-src-dec08b/admin/define-system.sh	2008-11-25 16:50:43.000000000 +0100
+++ gildas-src-dec08b.patched/admin/define-system.sh	2008-12-06 00:17:24.000000000 +0100
@@ -211,11 +211,11 @@
 		    DEFAULT_COMPILER=g95
 		fi
 	    elif [ `uname -p` = "i386" ]; then
-		if [ `sysctl -n hw.optional.x86_64 2> /dev/null` = 1 ]; then
-		    GAG_MACHINE=x86_64
-		else
+#		if [ `sysctl -n hw.optional.x86_64 2> /dev/null` = 1 ]; then
+#		    GAG_MACHINE=x86_64
+#		else
 		    GAG_MACHINE=pc
-		fi
+#		fi
 		if which ifort > /dev/null 2>&1; then
 		    DEFAULT_COMPILER=ifort
 		else
diff -ruN --exclude='*~' gildas-src-dec08b/admin/gildas-env.sh gildas-src-dec08b.patched/admin/gildas-env.sh
--- gildas-src-dec08b/admin/gildas-env.sh	2008-11-06 14:21:29.000000000 +0100
+++ gildas-src-dec08b.patched/admin/gildas-env.sh	2008-12-06 14:17:20.000000000 +0100
@@ -462,11 +462,11 @@
     INC_PATH=`echo $LIB_PATH | sed -e "s&lib&include&g"`
     #
     # Library locations under 64 bits linux
-    if [ "$GAG_MACHINE" = "x86_64" ]; then
-	if [ "$GAG_COMPILER" != "ifort_32" ]; then
-	    LIB_PATH=`echo $LIB_PATH | sed -e "s&lib&lib64&g"`
-	fi
-    fi
+#    if [ "$GAG_MACHINE" = "x86_64" ]; then
+#	if [ "$GAG_COMPILER" != "ifort_32" ]; then
+#	    LIB_PATH=`echo $LIB_PATH | sed -e "s&lib&lib64&g"`
+#	fi
+#    fi
     #
     OLDIFS=$IFS
     IFS=:
@@ -515,10 +515,10 @@
 	    NEWATM_LIB_DIR=$DIR
 	fi
 	if file_present "liblapack." "${DIR}"; then
-	    LAPACK_PRESENT=yes
+	    LAPACK_PRESENT=yes	    
 	    LAPACK_LIB_DIR=$DIR
 	fi
-	if file_present "libblas." "${DIR}"; then
+	if file_present "libf77blas." "${DIR}"; then
 	    BLAS_PRESENT=yes
 	    BLAS_LIB_DIR=$DIR
 	fi
