diff -ruN --exclude='*~' gildas-src-aug09c/admin/Makefile gildas-src-aug09c.patched/admin/Makefile
--- gildas-src-aug09c/admin/Makefile	2007-12-12 11:23:47.000000000 +0100
+++ gildas-src-aug09c.patched/admin/Makefile	2009-09-27 18:31:46.000000000 +0200
@@ -33,31 +33,31 @@
 integ: integ-message
 
 integ-message:
-	@(echo)
-	@(echo "Compilation succeeded! Bye.")
-	@(echo)
+#	@(echo)
+#	@(echo "Compilation succeeded! Bye.")
+#	@(echo)
 
 default-install: install-message
 
 install-message:
-	@(echo)
-	@(echo "***************************************************************************")
-	@(echo "Installation succeeded!")
-	@(echo)
-	@(echo "bash users: You should add in your ~/.bash_profile the following lines")
-	@(echo "    export GAG_ROOT_DIR=$(gagexedir)")
-	@(echo "    export GAG_EXEC_SYSTEM=$(GAG_EXEC_SYSTEM)")
-	@(echo '    source $$GAG_ROOT_DIR/etc/bash_profile')
-	@(echo)
-	@(echo "csh users:  You should add in your ~/.login the following lines")
-	@(echo "    setenv GAG_ROOT_DIR $(gagexedir)")
-	@(echo "    setenv GAG_EXEC_SYSTEM $(GAG_EXEC_SYSTEM)")
-	@(echo '    source $$GAG_ROOT_DIR/etc/login')
-	@(echo)
-	@(echo "You should then exit this xterm and open a new one to reset your path.")
-	@(echo "GILDAS should be ready for use (e.g. type greg @ gag_demo:demo).")
-	@(echo "Bye.")
-	@(echo "***************************************************************************")
-	@(echo)
+#	@(echo)
+#	@(echo "***************************************************************************")
+#	@(echo "Installation succeeded!")
+#	@(echo)
+#	@(echo "bash users: You should add in your ~/.bash_profile the following lines")
+#	@(echo "    export GAG_ROOT_DIR=$(gagexedir)")
+#	@(echo "    export GAG_EXEC_SYSTEM=$(GAG_EXEC_SYSTEM)")
+#	@(echo '    source $$GAG_ROOT_DIR/etc/bash_profile')
+#	@(echo)
+#	@(echo "csh users:  You should add in your ~/.login the following lines")
+#	@(echo "    setenv GAG_ROOT_DIR $(gagexedir)")
+#	@(echo "    setenv GAG_EXEC_SYSTEM $(GAG_EXEC_SYSTEM)")
+#	@(echo '    source $$GAG_ROOT_DIR/etc/login')
+#	@(echo)
+#	@(echo "You should then exit this xterm and open a new one to reset your path.")
+#	@(echo "GILDAS should be ready for use (e.g. type greg @ gag_demo:demo).")
+#	@(echo "Bye.")
+#	@(echo "***************************************************************************")
+#	@(echo)
 
 ###########################################################################
diff -ruN --exclude='*~' gildas-src-aug09c/admin/Makefile.build gildas-src-aug09c.patched/admin/Makefile.build
--- gildas-src-aug09c/admin/Makefile.build	2009-03-12 19:28:41.000000000 +0100
+++ gildas-src-aug09c.patched/admin/Makefile.build	2009-09-28 13:09:45.000000000 +0200
@@ -306,6 +306,10 @@
 
 -lm:
 
+-lLAPACK:
+
+-lBLAS:
+
 # Required by ASDM libs:
 -lstdc++:
 
diff -ruN --exclude='*~' gildas-src-aug09c/admin/Makefile.def gildas-src-aug09c.patched/admin/Makefile.def
--- gildas-src-aug09c/admin/Makefile.def	2009-05-28 11:17:16.000000000 +0200
+++ gildas-src-aug09c.patched/admin/Makefile.def	2009-09-28 13:00:24.000000000 +0200
@@ -549,7 +549,7 @@
 ###########################################################################
 # Kernel libraries
 
-LEGACY_LIBS = -lslatec -llapack -lblas
+LEGACY_LIBS = -lslatec -lLAPACK -lBLAS
 GAG_LIBS = -lggui -lgcore -lgwcs -lgio -lgsys -lginc
 
 ifeq ($(GAG_USE_XML),yes)
diff -ruN --exclude='*~' gildas-src-aug09c/admin/define-system.sh gildas-src-aug09c.patched/admin/define-system.sh
--- gildas-src-aug09c/admin/define-system.sh	2008-11-25 16:50:43.000000000 +0100
+++ gildas-src-aug09c.patched/admin/define-system.sh	2009-09-27 18:31:46.000000000 +0200
@@ -211,11 +211,11 @@
 		    DEFAULT_COMPILER=g95
 		fi
 	    elif [ `uname -p` = "i386" ]; then
-		if [ `sysctl -n hw.optional.x86_64 2> /dev/null` = 1 ]; then
-		    GAG_MACHINE=x86_64
-		else
+#		if [ `sysctl -n hw.optional.x86_64 2> /dev/null` = 1 ]; then
+#		    GAG_MACHINE=x86_64
+#		else
 		    GAG_MACHINE=pc
-		fi
+#		fi
 		if which ifort > /dev/null 2>&1; then
 		    DEFAULT_COMPILER=ifort
 		else
diff -ruN --exclude='*~' gildas-src-aug09c/admin/gildas-env.sh gildas-src-aug09c.patched/admin/gildas-env.sh
--- gildas-src-aug09c/admin/gildas-env.sh	2009-04-27 11:46:55.000000000 +0200
+++ gildas-src-aug09c.patched/admin/gildas-env.sh	2009-09-28 12:56:43.000000000 +0200
@@ -461,11 +461,11 @@
     INC_PATH=`echo $LIB_PATH:$GAGENV_SEARCH_PATH | sed -e "s&lib&include&g"`
     #
     # Library locations under 64 bits linux
-    if [ "$GAG_MACHINE" = "x86_64" ]; then
-	if [ "$GAG_COMPILER" != "ifort_32" ]; then
-	    LIB_PATH=`echo $LIB_PATH | sed -e "s&lib&lib64&g"`
-	fi
-    fi
+#    if [ "$GAG_MACHINE" = "x86_64" ]; then
+#	if [ "$GAG_COMPILER" != "ifort_32" ]; then
+#	    LIB_PATH=`echo $LIB_PATH | sed -e "s&lib&lib64&g"`
+#	fi
+#    fi
     # Add user's $GAG_SEARCH_PATH, but only after converting lib into lib64.
     LIB_PATH=$LIB_PATH:$GAGENV_SEARCH_PATH
     #
@@ -515,11 +515,11 @@
 	    NEWATM_PRESENT=yes
 	    NEWATM_LIB_DIR=$DIR
 	fi
-	if file_present "liblapack." "${DIR}"; then
-	    LAPACK_PRESENT=yes
+	if file_present "libLAPACK." "${DIR}"; then
+	    LAPACK_PRESENT=yes	    
 	    LAPACK_LIB_DIR=$DIR
 	fi
-	if file_present "libblas." "${DIR}"; then
+	if file_present "libBLAS." "${DIR}"; then
 	    BLAS_PRESENT=yes
 	    BLAS_LIB_DIR=$DIR
 	fi
diff -ruN --exclude='*~' gildas-src-aug09c/packages/clic/lib/Makefile gildas-src-aug09c.patched/packages/clic/lib/Makefile
--- gildas-src-aug09c/packages/clic/lib/Makefile	2008-11-14 10:52:59.000000000 +0100
+++ gildas-src-aug09c.patched/packages/clic/lib/Makefile	2009-09-28 12:59:24.000000000 +0200
@@ -55,7 +55,7 @@
 # Mathematical library dependency
 ifeq ($(LINEAR),lapack)
   LINEAR_LIB_OBJECTS = singular.o mth_lapack.o
-  LINEAR_LIB_DEPENDS = -llapack -lblas
+  LINEAR_LIB_DEPENDS = -lLAPACK -lBLAS
 else 
   ifeq ($(LINEAR),nag)
     LINEAR_LIB_OBJECTS = mth_nag.o
diff -ruN --exclude='*~' gildas-src-aug09c/packages/clic07/lib/Makefile gildas-src-aug09c.patched/packages/clic07/lib/Makefile
--- gildas-src-aug09c/packages/clic07/lib/Makefile	2009-07-10 12:17:56.000000000 +0200
+++ gildas-src-aug09c.patched/packages/clic07/lib/Makefile	2009-09-28 12:59:49.000000000 +0200
@@ -49,7 +49,7 @@
 # Mathematical library dependency
 ifeq ($(LINEAR),lapack)
   LINEAR_LIB_OBJECTS = singular.o mth_lapack.o
-  LINEAR_LIB_DEPENDS = -llapack -lblas
+  LINEAR_LIB_DEPENDS = -lLAPACK -lBLAS
 else 
   ifeq ($(LINEAR),nag)
     LINEAR_LIB_OBJECTS = mth_nag.o
