--- grace-5.1.9/config.cache.orig	Sun Sep  1 18:21:16 2002
+++ grace-5.1.9/config.cache	Sun Sep  1 18:20:12 2002
@@ -0,0 +1 @@
+ice_cv_have_isnan_decl=${ice_cv_have_isnan_decl=yes}
--- grace-5.1.9/doc/Makefile.orig	Sun Sep  1 20:33:33 2002
+++ grace-5.1.9/doc/Makefile	Sun Sep  1 20:32:42 2002
@@ -94,9 +94,9 @@
 	$(MAKE) install
 
 links : $(DESTDIR)$(GRACE_HOME)/doc/grace.1
-	$(MKINSTALLDIRS) $(DESTDIR)$(PREFIX)/man
-	$(MKINSTALLDIRS) $(DESTDIR)$(PREFIX)/man/man1
-	cd $(DESTDIR)$(PREFIX)/man/man1; \
+	$(MKINSTALLDIRS) $(DESTDIR)$(PREFIX)/share/man
+	$(MKINSTALLDIRS) $(DESTDIR)$(PREFIX)/share/man/man1
+	cd $(DESTDIR)$(PREFIX)/share/man/man1; \
 	for i in $(MANPAGES) $(MANLINKS); do $(RM) $$i; $(LN_S) $(GRACE_HOME)/doc/$$i $$i; done
 
 tests : dummy
