diff -Nurd -x'*~' gri-2.12.8.orig/doc/Makefile.in gri-2.12.8/doc/Makefile.in
--- gri-2.12.8.orig/doc/Makefile.in	2004-06-13 12:19:20.000000000 -0400
+++ gri-2.12.8/doc/Makefile.in	2006-07-11 18:01:07.000000000 -0400
@@ -621,7 +619,6 @@
 #	This is handled by gri.spec [2003-may-31 Dan Kelley]
 @OS_IS_LINUX_REDHAT_FALSE@	gzip $(MAN_DIR)/gri_merge.1
 @OS_IS_LINUX_REDHAT_FALSE@	gzip $(MAN_DIR)/gri_unpage.1
-@OS_IS_LINUX_REDHAT_FALSE@	cd  $(INFO_DIR) ; gzip -f --best gri.info*
 
 uninstall-local:
 	$(RM) `ls ../*ps ../license.txt *.html gri.info* $(HTML_DIR)/examples/* $(HTML_DIR)/resources/* $(HTML_DIR)/tst_suite/* $(HTML_DIR)/screenshots/* $(HTML_DIR)/*`
