--- hdf5-1.8.2/config/apple	2008-11-10 16:28:27.000000000 -0500
+++ hdf5-1.8.2.patched/config/apple	2009-01-13 10:23:53.000000000 -0500
@@ -31,10 +31,24 @@
 
 # The default Fortran 90 compiler
 if test "X-" = "X-$FC"; then
-    # Assume G95 compiler
-    FC=g95
-    FC_BASENAME=g95
-fi
+    case $F9X in
+        # The ibm, g95 and absoft
+        xlf*|g95*|f95*)
+            ;;
+    xlf)
+		FC=xlf5
+	    FC_BASENAME=xlf5
+	    ;;
+    g95)
+	    FC=g95
+	    FC_BASENAME=g95
+        ;;
+    f95)
+	    FC=f95
+	    FC_BASENAME=f95
+        ;;
+    esac
+ fi
 
 case $FC_BASENAME in
     xlf) 
