--- gctp/Makefile.orig	Thu May 29 12:35:24 2003
+++ gctp/Makefile	Thu May 29 13:10:59 2003
@@ -4,7 +4,7 @@
 LIBDIR  = 
 SRCDIR  = .
 INCS    = cproj.h proj.h 
-CC	= cc
+CC	= cc -Dunix 
 
 OBJECTS=		\
 		$(ARCHIVE)(alberfor.o) $(ARCHIVE)(alberinv.o) \
@@ -41,6 +41,7 @@
 		$(ARCHIVE)(cproj.o) $(ARCHIVE)(report.o) \
 		$(ARCHIVE)(paksz.o) $(ARCHIVE)(sphdz.o) \
 		$(ARCHIVE)(untfz.o) $(ARCHIVE)(gctp.o) \
+		$(ARCHIVE)(isinusfor.o) $(ARCHIVE)(isinusinv.o) \
 		$(ARCHIVE)(br_gctp.o)
 
 SOURCES = gctp.c alberfor.c alberinv.c alconfor.c alconinv.c azimfor.c \
--- hdfeos5/bin/INSTALL-HDFEOS.orig	Thu May 29 06:12:19 2003
+++ hdfeos5/bin/INSTALL-HDFEOS	Thu May 29 07:48:11 2003
@@ -208,11 +208,11 @@
             UserPrompt "Please enter the HDF5 library files directory: "
         else
             echo "Current value of the HDF5 library directory is: $HDF5LIB"
-            UserPrompt "New value (or Return to accept):"
-        fi
-        if [ "$user_response" != "" ] ; then
-            HDF5LIB="$user_response"
+#           UserPrompt "New value (or Return to accept):"
         fi
+#       if [ "$user_response" != "" ] ; then
+#           HDF5LIB="$user_response"
+#       fi
     fi
 
     ValidHDFLIB 	# make sure it's valid before proceeding
@@ -223,11 +223,11 @@
             UserPrompt "Please enter the HDF5 include files directory: "
         else
             echo "Current value of the HDF5 include directory is: $HDF5INC"
-            UserPrompt "New value (or Return to accept):"
-        fi
-        if [ "$user_response" != "" ] ; then
-            HDF5INC="$user_response"
+#           UserPrompt "New value (or Return to accept):"
         fi
+#       if [ "$user_response" != "" ] ; then
+#           HDF5INC="$user_response"
+#       fi
     fi
 
     ValidHDFINC  	# make sure it's valid before proceeding
@@ -452,7 +452,7 @@
 MACHINE="`uname -m | awk '{print $1}'`"	# needed on Cray & SCO
 
 case "$MACHINE" in
-    i386) OSTYPE=sco386 ;;		# SCO box
+#   i386) OSTYPE=sco386 ;;		# SCO box
     CRAY) OSTYPE=UNICOS ;;		# CRAY
     *) OSTYPE="`uname`"	;;		# everybody else
 esac
@@ -543,6 +543,12 @@
     BRAND=linux
     ;;
 
+  Darwin )
+    PATH=/usr/bin:/bin
+    HOST=`hostname`
+    BRAND=linux
+    ;;
+
   OSF1) 
     PATH=/usr/local/bin:/bin:/usr/bin:/etc:/usr/etc:/usr/ucb:/usr/bin/X11:/usr/sbin
     HOST="`hostname -s`"
