diff -Nurd -x'*~' mathomatic-15.1.5.orig/makefile mathomatic-15.1.5/makefile
--- mathomatic-15.1.5.orig/makefile	2010-07-19 15:33:49.000000000 -0400
+++ mathomatic-15.1.5/makefile	2010-07-20 06:20:46.000000000 -0400
@@ -151,8 +151,6 @@
 	$(INSTALL_DATA) m4/* $(mathdocdir)/m4
 	$(INSTALL_DATA) tests/* $(mathdocdir)/tests
 	$(INSTALL_DATA) factorial/* $(mathdocdir)/factorial
-	@echo
-	@echo Documentation is installed in $(mathdocdir)/html
 
 uninstall:
 	rm -f $(bindir)/$(AOUT)
