diff -Nurd -x'*~' mathomatic-15.0.6.orig/makefile mathomatic-15.0.6/makefile
--- mathomatic-15.0.6.orig/makefile	2010-01-28 15:25:58.000000000 -0500
+++ mathomatic-15.0.6/makefile	2010-04-14 01:15:33.000000000 -0400
@@ -149,8 +149,6 @@
 	$(INSTALL) -m 0644 m4/* $(mathdocdir)/m4
 	$(INSTALL) -m 0644 tests/* $(mathdocdir)/tests
 	$(INSTALL) -m 0644 factorial/* $(mathdocdir)/factorial
-	@echo
-	@echo Documentation is installed in $(mathdocdir)/html
 
 uninstall:
 	rm -f $(bindir)/$(AOUT)
