diff -Nurd -x'*~' mathomatic-15.1.0.orig/makefile mathomatic-15.1.0/makefile
--- mathomatic-15.1.0.orig/makefile	2010-05-14 08:31:54.000000000 -0400
+++ mathomatic-15.1.0/makefile	2010-05-18 23:20:07.000000000 -0400
@@ -143,8 +143,6 @@
 	$(INSTALL) -m 0644 m4/* $(mathdocdir)/m4
 	$(INSTALL) -m 0644 tests/* $(mathdocdir)/tests
 	$(INSTALL) -m 0644 factorial/* $(mathdocdir)/factorial
-	@echo
-	@echo Documentation is installed in $(mathdocdir)/html
 
 uninstall:
 	rm -f $(bindir)/$(AOUT)
