diff -Nurd muparser.orig/Makefile.in muparser/Makefile.in
--- muparser.orig/Makefile.in	2007-07-14 17:21:32.000000000 -0400
+++ muparser/Makefile.in	2008-04-23 01:42:38.000000000 -0400
@@ -270,7 +270,7 @@
 @COND_SHARED_1@	done
 
 @COND_SAMPLES_1@$(top_builddir)/samples/example1/example1$(EXEEXT): $(EXAMPLE1_OBJECTS) $(__muParser_lib___depname)
-@COND_SAMPLES_1@	$(CXX) -o $@ $(EXAMPLE1_OBJECTS) -L$(top_builddir)/lib $(LDFLAGS)  -L$(srcdir)/lib  $(LIBS) -lmuparser$(DEBUG_BUILD_POSTFIX)
+@COND_SAMPLES_1@	$(CXX) -o $@ $(EXAMPLE1_OBJECTS) $(LDFLAGS)  $(LIBS) $(top_builddir)/lib/$(DLLPREFIX)muparser$(DEBUG_BUILD_POSTFIX)$(__muParser_dll___targetsuf3)
 @COND_SAMPLES_1@	
 @COND_SAMPLES_1@	$(__example1___mac_setfilecmd)
 
