diff -Nurd -x'*~' ncbitools-2.2.6.orig/LICENSE ncbitools-2.2.6/LICENSE
--- ncbitools-2.2.6.orig/LICENSE	1969-12-31 19:00:00.000000000 -0500
+++ ncbitools-2.2.6/LICENSE	2005-12-29 00:24:13.000000000 -0500
@@ -0,0 +1,109 @@
+
+Introduction
+
+	The GenInfo Software Toolbox is a set of software and data exchange 
+specifications that are used by NCBI to produce portable, modular software for 
+molecular biology.  We make this software and specifications available to the 
+community for use in its own right, or as a foundation for building other 
+tools with similar properties.  All software produced by NCBI and here 
+provided includes the following copyright notice:
+
+   **************************************************************************
+   *                                                                        *
+   *              National Center for Biotechnology Information             *
+   *       Bldg. 38A, NIH,  8600 Rockville Pike,  Bethesda, MD  20894       *
+   *                                                                        *
+   *                             COPYRIGHT NOTICE                           *
+   *                                                                        *
+   * This software/database is "United States Government Work" under the    *
+   * terms of the United States Copyright Act.  It was written as part of   *
+   * the author's official duties as a Government employee and thus cannot  *
+   * be copyrighted.  This software/database is freely available to the     *
+   * public for use without a copyright notice.  Restrictions cannot be     *
+   * placed on its present or future use.                                   *
+   *                                                                        *
+   * Although all reasonable efforts have been taken to ensure the accuracy *
+   * and reliability of the software and data, the National Library of      *
+   * Medicine (NLM) and the U.S. Government does not and cannot warrant the *
+   * performance or results that may be obtained by using this software or  *
+   * data.  The NLM and the U.S. Government disclaims all warranties as to  *
+   * performance, merchantability or fitness for any particular purpose.    *
+   *                                                                        *
+   * In any work or product derived from this material, proper attribution  *
+   * of the author as the source of the software or data would be           *
+   * appreciated.                                                           *
+   *                                                                        *
+   **************************************************************************
+
+	In those cases where software is made available from other sources,
+other restrictions may apply.
+
+	It is available by anonymous ftp from ftp.ncbi.nih.gov. In the "toolbox"
+directory you will find the following:
+
+Directory Structure
+
+	These files are available for anonymous ftp from "ftp.ncbi.nih.gov".
+The directory structure is as follows:
+
+    toolbox       - top level toolbox directory
+		ncbi_tools		- NCBI software development toolkit and ASN.1 specs
+					most current version (3.xx). FTP the toolkit from
+                            this directory.
+		other_ncbi              - various other tools made by ncbi
+                other_tools             - a few ASN.1 tools not made by ncbi
+		vms_utils		- VMS tools for untarring and uncompressing
+
+ncbi_tools
+
+	This is the NCBI portable software toolkit.  A single compressed tar
+file contains all the code, ASN.1 specifications, and demo programs
+(including complete source code for Entrez).  This file is:
+
+ncbi.tar.Z  (Unix compressed tar file)
+ncbiZ.exe   (DOS, Windows PKZIP file)
+ncbi.hqx    (Mac self extracting archive)
+
+The contents of the files are the same except the UNIX version also includes
+the documentation. Documentation is available in the ncbi_tools/newdoc
+directory as MS Word file. A README in the files above explains installation.
+Hardcopy of the documentation and questions my be directed to
+toolbox@ncbi.nlm.nih.gov
+
+vms_utils
+
+	This contains a set of utilities and instructions for VMS users to 
+untar and uncompress the files in the other directories.  Get this first if
+you are on VMS and read the instructions.  Kindly maintained by Will Gilbert
+of Whitehead Institute.  Also contains other VMS utilities such as drivers
+for ISO9660 CDROMS.
+==========================================================================
+in other_tools directory:
+
+isode
+
+	This is a very large system of tools from ISO for use in prototyping
+OSI applications.  It includes many large tools in one file, and 5 volumes of 
+PostScript documentation in the other file.  The documentation is good for 
+learning about OSI.  There is an ASN.1 compiler which will give you the 
+flavor of what can be done with these tools.  This is not a simple or terribly 
+efficient system.
+
+
+osikit
+
+	This is a set of tools from NIST (formerly the Bureau of Standards) 
+to assist development of OSI applications.  It includes the free value tool,
+which can read and validate an ASN.1 specification and produce an encoder and
+decoder for both the print form and basic encoding rules form.  We find it
+very useful for checking syntax on both specifications and data objects.  The
+limitation is that in the print form, the tool can only accept VisibleStrings
+all on one line, and less than 200 chars in length.
+
+
+asn_pmp
+
+	A parser for ASN.1 print files written by Peter Karp.  It parses print 
+files into S expressions which makes a simple, flexible way to input data 
+objects.  It does no checking on syntax or content.  See the README for more 
+details.
diff -Nurd -x'*~' ncbitools-2.2.6.orig/demo.fasta ncbitools-2.2.6/demo.fasta
--- ncbitools-2.2.6.orig/demo.fasta	1969-12-31 19:00:00.000000000 -0500
+++ ncbitools-2.2.6/demo.fasta	2005-12-29 00:24:13.000000000 -0500
@@ -0,0 +1,4 @@
+>demo polylinker
+CAGGAAACAGCTATGACCATGATTACGAATTCGAAGCTTAAGGCCTCCATGGATCCCGGGTCGACGCGTA
+CGATATCGATGTCTAGATCTCCAGTACTAGTCTCGAGCTCTGCAGGGCCCGCGGTACCATGCATACTGGC
+CGTCGTTTTACAACGTCGTGACTGGGAAAAC
diff -Nurd -x'*~' ncbitools-2.2.6.orig/ncbi/api/salsap.h ncbitools-2.2.6/ncbi/api/salsap.h
--- ncbitools-2.2.6.orig/ncbi/api/salsap.h	1999-11-24 16:24:28.000000000 -0500
+++ ncbitools-2.2.6/ncbi/api/salsap.h	2006-03-30 00:55:23.000000000 -0500
@@ -86,7 +86,7 @@
 
 NLM_EXTERN Pointer LIBCALL FindSeqAlignInSeqEntry (SeqEntryPtr sep, Uint1 choice);
 
-NLM_EXTERN SeqAlignPtr LIBCALL is_salp_in_sap (SeqAnnotPtr sap, Uint1 choice);
+//NLM_EXTERN SeqAlignPtr LIBCALL is_salp_in_sap (SeqAnnotPtr sap, Uint1 choice);
 
 NLM_EXTERN Boolean LIBCALL is_dim1seqalign (SeqAlignPtr salp);
 	
diff -Nurd -x'*~' ncbitools-2.2.6.orig/ncbi/api/salutil.h ncbitools-2.2.6/ncbi/api/salutil.h
--- ncbitools-2.2.6.orig/ncbi/api/salutil.h	1999-09-06 20:19:24.000000000 -0400
+++ ncbitools-2.2.6/ncbi/api/salutil.h	2006-03-30 00:56:29.000000000 -0500
@@ -120,7 +120,7 @@
 NLM_EXTERN SeqIdPtr     ValNodeSeqIdListDup (ValNodePtr id_list);
 NLM_EXTERN CharPtr PNTR SeqIdListToCharArray (SeqIdPtr id_list, Int2 n);
 
-NLM_EXTERN SeqIdPtr     SeqIdReplaceID (SeqIdPtr head, SeqIdPtr pre, SeqIdPtr sip, SeqIdPtr next);
+//NLM_EXTERN SeqIdPtr     SeqIdReplaceID (SeqIdPtr head, SeqIdPtr pre, SeqIdPtr sip, SeqIdPtr next);
 NLM_EXTERN BioseqPtr    BioseqReplaceID (BioseqPtr bsp, SeqIdPtr newsip);
 NLM_EXTERN SeqEntryPtr  SeqEntryReplaceSeqID (SeqEntryPtr source_sep, SeqIdPtr sip);
 /*********************************************************
diff -Nurd -x'*~' ncbitools-2.2.6.orig/ncbi/corelib/ncbimisc.c ncbitools-2.2.6/ncbi/corelib/ncbimisc.c
--- ncbitools-2.2.6.orig/ncbi/corelib/ncbimisc.c	2002-11-06 16:25:10.000000000 -0500
+++ ncbitools-2.2.6/ncbi/corelib/ncbimisc.c	2005-12-29 12:01:28.000000000 -0500
@@ -1123,28 +1123,6 @@
 *****************************************************************************/
 
 #if defined(OS_MAC) || defined(OS_UNIX_DARWIN)
-
-// 2001-03-22:  Joshua Juran
-// C2PStr() and P2CStr() do not exist in Carbon, so we roll our own.
-# if TARGET_API_MAC_CARBON
-static void C2PStr(char *ioStr)
-{
-	size_t len = strlen(ioStr);
-	if (len > 255) {
-		len = 255;
-	}
-	memmove(ioStr + 1, ioStr, len);
-	ioStr[0] = len;
-}
-
-static void P2CStr(StringPtr ioStr)
-{
-	Byte len = ioStr[0];
-	memmove(ioStr, ioStr + 1, len);
-	ioStr[len] = '\0';
-}
-# endif
-
 /*  p_churchill 12/99 removed conditional compilation support for 
  *  THINKC and MPW
  */
diff -Nurd -x'*~' ncbitools-2.2.6.orig/ncbi/corelib/ncbithr.c ncbitools-2.2.6/ncbi/corelib/ncbithr.c
--- ncbitools-2.2.6.orig/ncbi/corelib/ncbithr.c	2003-03-06 10:12:27.000000000 -0500
+++ ncbitools-2.2.6/ncbi/corelib/ncbithr.c	2005-12-29 00:29:02.000000000 -0500
@@ -2554,14 +2554,16 @@
 }
 
 
-NLM_EXTERN Int4 NlmCPUNumber(void)
-{
 #if defined(OS_UNIX_DARWIN)
 
 #include <mach/mach.h>
 #include <mach/mach_host.h>
 #include <mach/host_info.h>
+#endif
 
+NLM_EXTERN Int4 NlmCPUNumber(void)
+{
+#if defined(OS_UNIX_DARWIN)
 host_basic_info_data_t hinfo;
 mach_msg_type_number_t hinfo_count = HOST_BASIC_INFO_COUNT;
 kern_return_t rc;
diff -Nurd -x'*~' ncbitools-2.2.6.orig/ncbi/vibrant/vibwndws.c ncbitools-2.2.6/ncbi/vibrant/vibwndws.c
--- ncbitools-2.2.6.orig/ncbi/vibrant/vibwndws.c	2003-04-09 14:16:53.000000000 -0400
+++ ncbitools-2.2.6/ncbi/vibrant/vibwndws.c	2006-03-30 00:28:36.000000000 -0500
@@ -6328,12 +6328,16 @@
  */
 /* #ifdef OS_MAC */
 #ifdef WIN_MAC
+#if TARGET_RT_LITTLE_ENDIAN
+  sleep(5);
+#else
   KeyMap  keys;
 
   GetKeys (keys);
   if ((keys [1] & 1) != 0) {
     Nlm_GetSet ();
   }
+#endif /* TARGET_RT_LITTLE_ENDIAN */
 #endif
 #ifdef WIN_MSWIN
   if (GetAsyncKeyState(VK_SHIFT) & 0x8000)
