--- netcdf-3.6.2/Makefile.am.orig	2007-02-21 16:25:37.000000000 -0500
+++ netcdf-3.6.2/Makefile.am	2007-03-23 20:45:58.000000000 -0400
@@ -24,6 +24,11 @@
 F77_TEST = nf_test
 endif
 
+# Does the user want C API?
+if BUILD_C
+LIBSRC = libsrc
+endif
+
 # Does the user want C++ API?
 if BUILD_CXX
 CXX_DIR = cxx
@@ -57,7 +62,7 @@
 # This is the list of subdirs for which Makefiles will be constructed
 # and run. ncgen must come before ncdump and cxx, because their tests
 # depend on it. nf_test depends upon ncgen as well.
-SUBDIRS = $(F90_DIR) $(F77_DIR) libsrc $(V2_TEST) $(NCGEN) $(NCDUMP)	\
+SUBDIRS = $(F90_DIR) $(F77_DIR) $(LIBSRC) $(V2_TEST) $(NCGEN) $(NCDUMP)	\
 $(F77_TEST) ${CXX_DIR} nc_test $(MAN) $(EXAMPLES)
 
 # make test should do the same as make check.
--- netcdf-3.6.2/Makefile.in.orig	2007-03-01 17:46:18.000000000 -0500
+++ netcdf-3.6.2/Makefile.in	2007-03-23 20:46:07.000000000 -0400
@@ -263,6 +263,9 @@
 @BUILD_F77_TRUE@F77_DIR = fortran
 @BUILD_F77_TRUE@F77_TEST = nf_test
 
+# Does the user want C API?
+@BUILD_C_TRUE@LIBSRC = libsrc
+
 # Does the user want C++ API?
 @BUILD_CXX_TRUE@CXX_DIR = cxx
 
@@ -283,7 +286,7 @@
 # This is the list of subdirs for which Makefiles will be constructed
 # and run. ncgen must come before ncdump and cxx, because their tests
 # depend on it. nf_test depends upon ncgen as well.
-SUBDIRS = $(F90_DIR) $(F77_DIR) libsrc $(V2_TEST) $(NCGEN) $(NCDUMP)	\
+SUBDIRS = $(F90_DIR) $(F77_DIR) $(LIBSRC) $(V2_TEST) $(NCGEN) $(NCDUMP)	\
 $(F77_TEST) ${CXX_DIR} nc_test $(MAN) $(EXAMPLES)
 
 
--- netcdf-3.6.2/f90/netcdf_f90.3.orig	2006-08-14 15:32:52.000000000 -0400
+++ netcdf-3.6.2/f90/netcdf_f90.3	2007-03-23 23:12:15.000000000 -0400
@@ -456,7 +456,7 @@
 and the name of the variable.
 .RE
 .HP
-\fBfunction nf90_Inqire_Variable(\fIncid\fP, \fIvarid\fP, \fIname\fP, 
+\fBfunction nf90_inquire_variable(\fIncid\fP, \fIvarid\fP, \fIname\fP, 
 \fIxtype\fP, \fIndims\fP, \fIdimids\fP, \fInatts\fP)\fR
 .RS
 .nf
--- netcdf-3.6.2/man/netcdf_f90-man.html.orig	2007-03-01 17:48:13.000000000 -0500
+++ netcdf-3.6.2/man/netcdf_f90-man.html	2007-03-23 23:12:59.000000000 -0400
@@ -713,7 +713,7 @@
 <tr valign="top" align="left">
 <td width="10%"></td>
 <td width="89%">
-<p><b>function nf90_Inqire_Variable(</b><i>ncid</i><b>,</b>
+<p><b>function nf90_inquire_variable(</b><i>ncid</i><b>,</b>
 <i>varid</i><b>,</b> <i>name</i><b>,</b>
 <i>xtype</i><b>,</b> <i>ndims</i><b>,</b>
 <i>dimids</i><b>,</b> <i>natts</i><b>)</b></p></td>
--- netcdf-3.6.2/cxx/Makefile.am.orig	2007-02-27 10:44:33.000000000 -0500
+++ netcdf-3.6.2/cxx/Makefile.am	2007-03-24 15:32:32.000000000 -0400
@@ -68,9 +68,8 @@
 
 else # not USE_NETCDF4
 
-# Point pre-processor and linker to netcdf-3 directory (libsrc).
+# Point pre-processor to netcdf-3 directory (libsrc).
 AM_CPPFLAGS += -I$(top_srcdir)/libsrc
-AM_LDFLAGS = -L. -L$(top_builddir)/libsrc
 
 # Link to the netCDF-3 library and the C++ library.
 LDADD = $(top_builddir)/cxx/libnetcdf_c++.la $(top_builddir)/libsrc/libnetcdf.la 
--- netcdf-3.6.2/cxx/Makefile.in.orig	2007-03-01 17:46:14.000000000 -0500
+++ netcdf-3.6.2/cxx/Makefile.in	2007-03-24 15:33:51.000000000 -0400
@@ -318,7 +318,6 @@
 tst_failure_SOURCES = tst_failure.cpp
 XFAIL_TESTS = tst_failure$(EXEEXT)
 TESTS = $(TESTFILES) $(am__append_1) $(am__append_6)
-@USE_NETCDF4_FALSE@AM_LDFLAGS = -L. -L$(top_builddir)/libsrc
 @USE_NETCDF4_TRUE@AM_LDFLAGS = -L. -L$(top_builddir)/libsrc4 \
 @USE_NETCDF4_TRUE@	$(am__append_4)
 
