diff -Naur octplot-0.4.0/src/figure.cpp octplot-0.4.0.patched/src/figure.cpp
--- octplot-0.4.0/src/figure.cpp	2007-04-12 13:50:07.000000000 -0400
+++ octplot-0.4.0.patched/src/figure.cpp	2007-09-27 20:09:10.000000000 -0400
@@ -16,8 +16,8 @@
 //  
 
 #include <iostream>
-#include "Fl/Fl.H"
-#include "Fl/fl_draw.H"
+#include "FL/Fl.H"
+#include "FL/fl_draw.H"
 
 #include "axes.h"
 #include "property.h"
