diff -ruN pointless-orig/Makefile pointless/Makefile
--- pointless-orig/Makefile	2007-11-23 05:49:31.000000000 -0800
+++ pointless/Makefile	2007-12-11 17:54:06.000000000 -0800
@@ -20,16 +20,20 @@
 #XCPPFLAGS= -ftemplate-depth-120 -fcoalesce-templates -Wno-long-double
 
 # CCTBX locations
-CCTBX=${CLIB}/cctbx
+CCTBX=/@prefix@/share/xtal/cctbx
 CCTBX_sources=${CCTBX}/cctbx_sources
 ## Blank for version 1
 CCTBX_VERSION=
 ## Set for version 2
-##CCTBX_VERSION="\#define CCTBX_VERSION_2"
+##
+CCTBX_VERSION="\#define CCTBX_VERSION_2"
 
 # Clipper location
-CLPR=${CLIB}/clipper
+# CLPR=${CLIB}/clipper
+CLPR=/@prefix@/share/xtal/ccp4-onlylibs-dev-6.0.99/lib/clipper
 
+CLIB=/@prefix@/share/xtal/ccp4-onlylibs-dev-6.0.99/lib
+CCP4=/@prefix@/share/xtal/ccp4-onlylibs-dev-6.0.99
 #<<<<< Includes
 # CCP4
 ICCP4=-I${CLIBS}
@@ -41,6 +45,7 @@
 # cctbx
 ITBX=\
 -I${CCTBX_sources}/boost \
+-I/@prefix@/share/xtal/cctbx/cctbx_build/include \
 -I${CCP4}/include \
 -I${CCTBX_sources}/cctbx/include \
 -I${CCTBX_sources}/scitbx/include
@@ -49,13 +54,13 @@
 
 #<<<<< Libraries
 # CCP4
-LCCP4=-L${CLIB} -lccp4f -lccp4c -lm
+LCCP4=-L${CLIB} -lccp4f -lccp4c -lm -L/@prefix@/lib -ldrfftw -lfftw
 
 # Clipper
 LCLPR=-L${CLIB} -lclipper-ccp4 -lclipper-core 
 
 # cctbx
-LTBX= -L${CLIB} -lcctbx 
+LTBX= -L/@prefix@/share/xtal/cctbx/cctbx_build/lib -lcctbx 
 
 LDLIBS=${LTBX} ${LCLPR} ${LCCP4}
 
