diff -ru polyxmass-bin-0.8.6/configure polyxmass-bin-0.8.6-fink/configure
--- polyxmass-bin-0.8.6/configure	Sat Feb  5 15:51:12 2005
+++ polyxmass-bin-0.8.6-fink/configure	Wed Feb  9 19:50:14 2005
@@ -1844,11 +1844,11 @@
 
 
 GLIB_REQUIRED=2.4.6
-GTK_REQUIRED=2.4.10
+GTK_REQUIRED=2.4.9
 LIB_XML_REQUIRED=2.6.11
 LIBGNOMECANVAS_REQUIRED=2.6.1
-LIBRSVG_REQUIRED=2.7.2
-LIBGLADE_REQUIRED=2.4.0
+LIBRSVG_REQUIRED=2.6.5
+LIBGLADE_REQUIRED=2.3.6
 LIBPOLYXMASS_REQUIRED=0.8.6
 
 
