--- qtiplot-0.9.7.5orig/manual/Makefile	2009-03-05 13:57:29.000000000 +0100
+++ qtiplot-0.9.7.5/manual/Makefile	2009-03-26 12:50:12.000000000 +0100
@@ -1,3 +1,9 @@
 html-pdf:
-	docbook2html -d qtiplot.dsl docbook-en/index.docbook -e no-valid
-	dblatex -o qtiplot-manual-en.pdf -c dblatex.conf -p dblatex-param.xsl docbook-en/index.docbook
+#	docbook2html -d qtiplot.dsl docbook-en/index.docbook -e no-valid
+#	dblatex -o qtiplot-manual-en.pdf -c dblatex.conf -p dblatex-param.xsl docbook-en/index.docbook
+
+install:
+
+clean:
+
+distclean:
--- qtiplot-0.9.7.5orig/qtiplot/qtiplot.pro	2009-03-09 00:28:18.000000000 +0100
+++ qtiplot-0.9.7.5/qtiplot/qtiplot.pro	2009-03-26 12:49:45.000000000 +0100
@@ -1,6 +1,6 @@
 # building without muParser doesn't work yet
 SCRIPTING_LANGS += muParser
-SCRIPTING_LANGS += Python
+#SCRIPTING_LANGS += Python
 
 # a console displaying output of scripts; particularly useful on Windows
 # where running QtiPlot from a terminal is inconvenient
@@ -28,13 +28,13 @@
 #!!! Warning: You must modify these paths according to your computer settings
 #############################################################################
 
-INCLUDEPATH       += ../3rdparty/muparser/include
+INCLUDEPATH       += @PREFIX@/include
 INCLUDEPATH       += ../3rdparty/qwtplot3d/include
 INCLUDEPATH       += ../3rdparty/qwt/src
 INCLUDEPATH       += ../3rdparty/liborigin
-INCLUDEPATH       += ../3rdparty/gsl/include
+INCLUDEPATH       += @PREFIX@/include/gsl
 INCLUDEPATH       += ../3rdparty/zlib
-INCLUDEPATH       += ../3rdparty/boost_1_36_0
+INCLUDEPATH       += @PREFIX@/include/boost
 
 ##################### 3rd PARTY LIBRARIES SECTION ###########################
 #!!! Warning: You must modify these paths according to your computer settings
@@ -43,18 +43,18 @@
 ##################### Linux (Mac OS X) ######################################
 
 # statically link against libraries in 3rdparty
-unix:LIBS         += ../3rdparty/muparser/lib/libmuparser.a
+#unix:LIBS         += ../3rdparty/muparser/lib/libmuparser.a
 unix:LIBS         += ../3rdparty/qwt/lib/libqwt.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgsl.a
-unix:LIBS         += ../3rdparty/gsl/lib/libgslcblas.a
-unix:LIBS         += ../3rdparty/boost_1_36_0/lib/libboost_date_time-gcc43-mt-1_36.a
-unix:LIBS         += ../3rdparty/boost_1_36_0/lib/libboost_thread-gcc43-mt-1_36.a
+#unix:LIBS         += @PREFIX@/lib/libgsl.a
+#unix:LIBS         += @PREFIX@/lib/libgslcblas.a
+#unix:LIBS         += @PREFIX@/lib/libboost_date_time-mt.a
+#unix:LIBS         += @PREFIX@/lib/libboost_thread-mt.a
 
 # dynamically link against dependencies if they are installed system-wide
-#unix:LIBS         += -lmuparser
+unix:LIBS         += -lmuparser
 #unix:LIBS         += -lqwt
-#unix:LIBS         += -lgsl -lgslcblas
-
+unix:LIBS         += -lgsl -lgslcblas
+unix:LIBS         += -lboost_date_time-mt -lboost_thread-mt
 ##################### Windows ###############################################
 
 win32:LIBS        += ../3rdparty/muparser/lib/libmuparser.a
@@ -84,7 +84,7 @@
 	INSTALLS        += documentation
 	unix:INSTALLS        += man
 
-	unix: INSTALLBASE = /usr
+	unix: INSTALLBASE = @PREFIX@
 	win32: INSTALLBASE = C:/QtiPlot
 
 	unix: target.path = $$INSTALLBASE/bin
@@ -210,8 +210,9 @@
 
 contains(CONFIG, HAVE_LIBPNG){
 	DEFINES += GL2PS_HAVE_LIBPNG
-	INCLUDEPATH += ../3rdparty/libpng/
-	LIBS        += ../3rdparty/libpng/libpng.a
+	INCLUDEPATH += @PREFIX@/include
+#	LIBS        += @PREFIX@/lib/libpng.a
+	LIBS        += -lpng
 }
 
 ###############################################################
--- qtiplot-0.9.7.5orig/fitPlugins/exp_saturation/exp_saturation.pro	2009-03-05 13:56:56.000000000 +0100
+++ qtiplot-0.9.7.5/fitPlugins/exp_saturation/exp_saturation.pro	2009-03-27 10:03:07.000000000 +0100
@@ -12,14 +12,14 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+INCLUDEPATH += @PREFIX@/include/
+#LIBS        += @PREFIX@/lib/libgsl.a
+#LIBS        += @PREFIX@/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:LIBS += -L@PREFIX@/lib -lgsl -lgslcblas
 
-target.path=/usr/lib$${libsuff}/qtiplot/plugins
-INSTALLS += target
+#target.path=@PREFIX@/lib$${libsuff}/qtiplot/plugins
+#INSTALLS += target
 
 SOURCES += exp_saturation.c
--- qtiplot-0.9.7.5orig/fitPlugins/explin/explin.pro	2009-03-05 13:56:56.000000000 +0100
+++ qtiplot-0.9.7.5/fitPlugins/explin/explin.pro	2009-03-27 10:03:17.000000000 +0100
@@ -12,14 +12,14 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+INCLUDEPATH += @PREFIX@/include/
+#LIBS        += @PREFIX@/lib/libgsl.a
+#LIBS        += @PREFIX@/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:LIBS += -L@PREFIX@/lib -lgsl -lgslcblas
 
-target.path=/usr/lib$${libsuff}/qtiplot/plugins
-INSTALLS += target
+#target.path=@PREFIX@/lib$${libsuff}/qtiplot/plugins
+#INSTALLS += target
 
 SOURCES += explin.c
--- qtiplot-0.9.7.5orig/fitPlugins/fitRational0/fitRational0.pro	2009-03-05 13:56:56.000000000 +0100
+++ qtiplot-0.9.7.5/fitPlugins/fitRational0/fitRational0.pro	2009-03-27 10:03:23.000000000 +0100
@@ -1,25 +1,25 @@
-QMAKE_PROJECT_DEPTH = 0
-linux-g++-64: libsuff=64
-
-TARGET            = fitRational0
-TEMPLATE          = lib
-
-CONFIG           += thread
-CONFIG           += warn_on 
-CONFIG           += release
-CONFIG           += dll 
-
-DESTDIR           = ../
- 
-# statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
-
-#dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
-
-target.path=/usr/lib$${libsuff}/qtiplot/plugins
-INSTALLS += target
-
+QMAKE_PROJECT_DEPTH = 0
+linux-g++-64: libsuff=64
+
+TARGET            = fitRational0
+TEMPLATE          = lib
+
+CONFIG           += thread
+CONFIG           += warn_on 
+CONFIG           += release
+CONFIG           += dll 
+
+DESTDIR           = ../
+ 
+# statically link against GSL in 3rdparty
+INCLUDEPATH += @PREFIX@/include/
+#LIBS        += @PREFIX@/lib/libgsl.a
+#LIBS        += @PREFIX@/lib/libgslcblas.a
+
+#dynamically link against GSL installed system-wide
+unix:LIBS += -L@PREFIX@/lib -lgsl -lgslcblas
+
+#target.path=@PREFIX@/lib$${libsuff}/qtiplot/plugins
+#INSTALLS += target
+
 SOURCES += fitRational0.cpp
--- qtiplot-0.9.7.5orig/fitPlugins/fitRational1/fitRational1.pro	2009-03-05 13:56:56.000000000 +0100
+++ qtiplot-0.9.7.5/fitPlugins/fitRational1/fitRational1.pro	2009-03-27 10:03:29.000000000 +0100
@@ -12,14 +12,14 @@
 DESTDIR           = ../
  
 # statically link against GSL in 3rdparty
-INCLUDEPATH += ../../3rdparty/gsl/include/
-LIBS        += ../../3rdparty/gsl/lib/libgsl.a
-LIBS        += ../../3rdparty/gsl/lib/libgslcblas.a
+INCLUDEPATH += @PREFIX@/include/
+#LIBS        += @PREFIX@/lib/libgsl.a
+#LIBS        += @PREFIX@/lib/libgslcblas.a
 
 #dynamically link against GSL installed system-wide
-#unix:LIBS += -L /usr/lib$${libsuff} -lgsl -lgslcblas
+unix:LIBS += -L@PREFIX@/lib -lgsl -lgslcblas
 
-target.path=/usr/lib$${libsuff}/qtiplot/plugins
-INSTALLS += target
+#target.path=@PREFIX@/lib$${libsuff}/qtiplot/plugins
+#INSTALLS += target
 
 SOURCES += fitRational1.cpp

--- qtiplot-0.9.7.5orig/3rdparty/qwt/qwt.prf	2009-03-10 12:43:17.000000000 +0100
+++ qtiplot-0.9.7.5/3rdparty/qwt/qwt.prf	2009-03-27 08:35:04.000000000 +0100
@@ -11,7 +11,7 @@
 QwtVersion = 5.2.0
 
 unix {
-    QwtBase    = /usr/local/qwt-$${QwtVersion}-svn
+    QwtBase    = @PREFIX@/share/qwt-$${QwtVersion}-svn
     LIBS        += -L$${QwtBase}/lib -lqwt
 }
 win32 {
--- qtiplot-0.9.7.5orig/3rdparty/qwt/qwtconfig.pri	2009-03-10 12:43:54.000000000 +0100
+++ qtiplot-0.9.7.5/3rdparty/qwt/qwtconfig.pri	2009-03-27 11:39:32.000000000 +0100
@@ -8,15 +8,15 @@
 VERSION      = $${VER_MAJ}.$${VER_MIN}.$${VER_PAT}
 
 unix {
-    INSTALLBASE    = /usr/local/qwt-$$VERSION-svn
+#    INSTALLBASE    = @PREFIX@/share/qwt-$$VERSION-svn
 }
 
 win32 {
     INSTALLBASE    = C:/Qwt-$$VERSION-svn
 }
 
-target.path    = $$INSTALLBASE/lib
-headers.path   = $$INSTALLBASE/include
-doc.path       = $$INSTALLBASE/doc
+#target.path    = $$INSTALLBASE/lib
+#headers.path   = $$INSTALLBASE/include
+#doc.path       = $$INSTALLBASE/doc
 
 ######################################################################
 # qmake internal options