diff -ruN refmac-orig/makefile_fink refmac/makefile_fink
--- refmac-orig/makefile_fink	1969-12-31 16:00:00.000000000 -0800
+++ refmac/makefile_fink	2007-12-29 21:57:57.000000000 -0800
@@ -0,0 +1,18 @@
+VERSION = -5.5.0082
+FC      = gfortran
+FOPTIM  = -O3
+COPTIM  = -O
+XFFLAGS = -fno-second-underscore
+
+XCFLAGS = -DGFORTRAN -DPROTOTYPE
+FFLAGS = $(FOPTIM) $(XFFLAGS)  -static-libgcc
+ 
+CFLAGS = $(COPTIM) $(XCFLAGS)
+
+
+LLIBCIF = 
+LLIBCCP = $(CLIB)/libccp4f.a $(CLIB)/libccp4c.dylib  $(BLAS_DIR)/libLAPACK.dylib  $(BLAS_DIR)/libBLAS.dylib  $(FORTLIB)/libgfortran.a
+LLIBOTHERS = -lstdc++ $(FORTLIB)/libgfortran.a   $(FORTLIB2)/libgfortranbegin.a  $(FORTLIB2)/libgcc.a  
+
+
+include makefile_refmac.inc1
