--- configure	29 May 2007 15:18:03 -0000	1.398
+++ configure	1 Jun 2007 03:34:49 -0000
@@ -2478,21 +2478,21 @@
     # for a system library, then see if we have various headers needed.
     if test "x$enable_builtin_afterimage" = "xyes" ; then
        check_header "jpeglib.h" "" $ASIMAGE $ASIMAGE/include \
-          /usr/local/include /usr/include /opt/include
+          $finkdir/include /usr/local/include /usr/include /opt/include
        asjpegincdir=$found_dir
        check_header "png.h" "" $ASIMAGE $ASIMAGE/include \
-          /usr/local/include /usr/include /usr/local/include/libpng \
+          $finkdir/include /usr/local/include /usr/include /usr/local/include/libpng \
           /opt/include
        aspngincdir=$found_dir
        if test ! "x$enable_astiff" = "xno" ; then
            check_header "tiffio.h" "" $ASIMAGE $ASIMAGE/include \
-               /usr/local/include /usr/include /opt/include
+               $finkdir/include /usr/local/include /usr/include /opt/include
            astiffincdir=$found_dir
        else
            astiffincdir="--with-tiff=no"
        fi
        check_header "gif_lib.h" "" $ASIMAGE $ASIMAGE/include \
-          /usr/local/include /usr/include /opt/include
+          $finkdir/include /usr/local/include /usr/include /opt/include
        asgifincdir=$found_dir
 
        asextralib=""
@@ -2510,7 +2510,7 @@
        fi
        for k in $aslibs ; do
            check_library $k "$enable_shared" "" \
-             $ASIMAGE $ASIMAGE/lib /usr/local/lib /usr/lib /opt/lib
+             $ASIMAGE $ASIMAGE/lib $finkdir/lib /usr/local/lib /usr/lib /opt/lib
            if test "x$k" = "xlibz" ; then
                found_libz=$found_lib
                found_dirz=$found_dir
--- build/unix/makelib.sh	29 Jan 2007 10:34:36 -0000	1.53
+++ build/unix/makelib.sh	1 Jun 2007 03:34:49 -0000
@@ -201,7 +201,7 @@
 	ln -fs $SONAME.$MAJOR        $LIB
     elif [ -f $LIBVERS ]; then
 	# Versioned library has format foo.3.05.so
-	source_file=`echo $SONAME | sed "s/.*\./&${MINOR}./"`
+	source_file=`/usr/bin/basename $SONAME | sed "s/.*\./&${MINOR}./"`
 	LIBNOMAJORMINOR=`echo $LIB|sed 's,\.'$MAJOR'\.'$MINOR',,'`
 	if [ $LIB != $LIBNOMAJORMINOR ]; then
 	    LIBNOMINOR=`echo $LIB|sed 's,\.'$MINOR',,'`
--- config/Makefile.depend	17 May 2007 14:15:26 -0000	1.138
+++ config/Makefile.depend	1 Jun 2007 03:35:26 -0000
@@ -267,68 +267,82 @@
 NETLIBEXTRA             = -Llib -lRIO
 HISTLIBEXTRA            = -Llib -lMatrix
 GRAFLIBEXTRA            = -Llib -lHist -lMatrix -lRIO
-GPADLIBEXTRA            = -Llib -lGraf -lHist
-G3DLIBEXTRA             = -Llib -lGraf -lHist -lGpad
-GEDLIBEXTRA             = -Llib -lHist -lGpad -lGraf -lGui -lTree -lTreePlayer
-POSTSCRIPTLIBEXTRA      = -Llib -lGraf
-GUILIBEXTRA             = -Llib -lGpad -lGraf -lRIO
-HISTPAINTERLIBEXTRA     = -Llib -lGraf -lHist -lMatrix -lGpad
-SPECTRUMPAINTERLIBEXTRA = -Llib -lGraf -lHist
-HTMLLIBEXTRA            = -Llib -lGraf
-MINUITLIBEXTRA          = -Llib -lGraf -lHist -lMatrix
-MINUIT2LIBEXTRA         = -Llib -lGraf -lHist -lMatrix
-FUMILILIBEXTRA          = -Llib -lGraf -lHist
+GPADLIBEXTRA            = -Llib -lGraf -lHist -lMatrix -lRIO
+G3DLIBEXTRA             = -Llib -lGraf -lHist -lGpad -lMatrix -lRIO
+GEDLIBEXTRA             = -Llib -lHist -lGpad -lGraf -lGui -lTree -lTreePlayer \
+                          -lMatrix -lNet -lRIO -lGraf3d
+POSTSCRIPTLIBEXTRA      = -Llib -lGraf -lRIO -lHist -lMatrix
+GUILIBEXTRA             = -Llib -lGpad -lGraf -lRIO -lHist -lMatrix
+HISTPAINTERLIBEXTRA     = -Llib -lGraf -lHist -lMatrix -lGpad -lRIO
+SPECTRUMPAINTERLIBEXTRA = -Llib -lGraf -lHist -lMatrix -lRIO
+HTMLLIBEXTRA            = -Llib -lGraf -lRIO -lHist -lMatrix
+MINUITLIBEXTRA          = -Llib -lGraf -lHist -lMatrix -lRIO
+MINUIT2LIBEXTRA         = -Llib -lGraf -lHist -lMatrix -lRIO
+FUMILILIBEXTRA          = -Llib -lGraf -lHist -lMatrix -lRIO
 TREELIBEXTRA            = -Llib -lNet -lRIO
-TREEPLAYERLIBEXTRA      = -Llib -lTree -lGraf3d -lGraf -lHist -lGpad -lRIO
-TREEVIEWERLIBEXTRA      = -Llib -lTree -lGpad -lGraf -lHist -lGui
+TREEPLAYERLIBEXTRA      = -Llib -lTree -lGraf3d -lGraf -lHist -lGpad -lRIO \
+                          -lMatrix -lNet
+TREEVIEWERLIBEXTRA      = -Llib -lTree -lGpad -lGraf -lHist -lGui -lRIO \
+                          -lMatrix -lNet
 PROOFLIBEXTRA           = -Llib -lNet -lTree -lThread -lRIO
 PROOFPLAYERLIBEXTRA     = -Llib -lProof -lNet -lTree -lGraf3d -lGraf -lGpad \
-                          -lHist -lTreePlayer -lThread -lRIO
-PROOFXLIBEXTRA          = -Llib -lNet -lProof -lThread
-SESSIONVIEWERLIBEXTRA   = -Llib -lProof -lGui -lGpad -lGraf -lTree
+                          -lHist -lTreePlayer -lThread -lRIO -lMatrix
+PROOFXLIBEXTRA          = -Llib -lNet -lProof -lThread -lRIO -lTree
+SESSIONVIEWERLIBEXTRA   = -Llib -lProof -lGui -lGpad -lGraf -lTree \
+                          -lHist -lNet -lThread -lRIO -lMatrix
 PEACLIBEXTRA            = -Llib -lProof -lClarens
 PEACGUILIBEXTRA         = -Llib -lGui
-X3DLIBEXTRA             = -Llib -lGraf3d -lGui
-EGLIBEXTRA              = -Llib -lGraf3d -lGraf -lGpad
-VMCLIBEXTRA             = -Llib -lEG -lGeom
+X3DLIBEXTRA             = -Llib -lGraf3d -lGui -lGraf -lGpad -lHist -lMatrix -lRIO
+EGLIBEXTRA              = -Llib -lGraf3d -lGraf -lGpad -lHist -lMatrix -lRIO
+VMCLIBEXTRA             = -Llib -lEG -lGeom -lGraf3d -lGraf -lGpad -lHist \
+                          -lMatrix -lRIO
 PHYSICSLIBEXTRA         = -Llib -lMatrix
-PYTHIA6LIBEXTRA         = -Llib -lEG -lGraf -lVMC -lPhysics $(FPYTHIA6LIBDIR) \
-                          $(FPYTHIA6LIB)
-X11TTFLIBEXTRA          = -Llib -lGX11 -lGraf
-TABLELIBEXTRA           = -Llib -lTree -lGpad -lGraf3d -lGraf -lHist -lRIO
-GLLIBEXTRA              = -Llib -lGpad -lGraf3d -lGui -lGraf -lHist -lGed
-HBOOKLIBEXTRA           = -Llib -lHist -lMatrix -lTree -lGraf -lTreePlayer -lRIO
+PYTHIA6LIBEXTRA         = -Llib -lEG -lGraf -lVMC -lPhysics -lGraf3d -lGpad -lGeom \
+                          -lHist -lMatrix -lRIO $(FPYTHIA6LIBDIR) $(FPYTHIA6LIB)
+X11TTFLIBEXTRA          = -Llib -lGX11 -lGraf -lHist -lMatrix -lRIO
+TABLELIBEXTRA           = -Llib -lTree -lGpad -lGraf3d -lGraf -lHist -lRIO \
+                          -lMatrix -lNet
+GLLIBEXTRA              = -Llib -lGpad -lGraf3d -lGui -lGraf -lHist -lGed \
+                          -lMatrix -lRIO -lTree -lTreePlayer -lNet
+HBOOKLIBEXTRA           = -Llib -lHist -lMatrix -lTree -lGraf -lTreePlayer \
+                          -lRIO -lNet -lGraf3d -lGpad
 GEOMLIBEXTRA            = -Llib -lRIO
-GEOMPAINTERLIBEXTRA     = -Llib -lGeom -lTree -lGraf3d -lHist -lGpad
-GEOMBUILDERLIBEXTRA     = -Llib -lGeom -lGraf3d -lGpad -lGraf -lGui -lGed
-ASIMAGELIBEXTRA         = -Llib -lGraf
-ASIMAGEGUILIBEXTRA      = -Llib -lGraf -lHist -lGui -lASImage -lRIO
-ASIMAGEGSLIBEXTRA       = -Llib -lGraf -lASImage
+GEOMPAINTERLIBEXTRA     = -Llib -lGeom -lTree -lGraf3d -lHist -lGpad \
+                          -lRIO -lMatrix -lNet -lGraf
+GEOMBUILDERLIBEXTRA     = -Llib -lGeom -lGraf3d -lGpad -lGraf -lGui -lGed \
+                          -lHist -lTree -lTreePlayer -lNet -lMatrix -lRIO
+ASIMAGELIBEXTRA         = -Llib -lGraf -lHist -lGpad -lMatrix -lRIO
+ASIMAGEGUILIBEXTRA      = -Llib -lGraf -lHist -lGui -lASImage -lRIO -lGpad -lMatrix
+ASIMAGEGSLIBEXTRA       = -Llib -lGraf -lASImage -lHist -lRIO -lGpad -lMatrix
 MLPLIBEXTRA             = -Llib -lHist -lMatrix -lTree -lGraf -lGpad \
-                          -lTreePlayer
+                          -lTreePlayer -lNet -lRIO -lGraf3d
 SPECTRUMLIBEXTRA        = -Llib -lHist -lMatrix
 TMVALIBEXTRA            = -Llib -lRIO -lHist -lMatrix -lTree -lGraf -lGpad \
-                          -lTreePlayer -lMLP
-SPLOTLIBEXTRA           = -Llib -lMatrix -lHist -lTree -lTreePlayer -lGraf3d -lGraf
-QTROOTLIBEXTRA          = -Llib -lGui -lGQt
-GQTLIBEXTRA             = -Llib -lGpad -lGraf -lRint
+                          -lTreePlayer -lMLP -lGraf3d -lNet
+SPLOTLIBEXTRA           = -Llib -lMatrix -lHist -lTree -lTreePlayer  -lGpad \
+                          -lNet -lRIO -lGraf3d -lGraf
+QTROOTLIBEXTRA          = -Llib -lGui -lGQt -lGraf -lGpad -lRIO -lHist -lMatrix -lRint
+GQTLIBEXTRA             = -Llib -lGpad -lGraf -lRint -lHist -lMatrix -lRIO
 QUADPLIBEXTRA           = -Llib -lMatrix
-RUBYLIBEXTRA            = -Llib -lHist
-GUIBLDLIBEXTRA          = -Llib -lGui -lGraf
-GUIHTMLLIBEXTRA         = -Llib -lGui -lGraf -lNet
+RUBYLIBEXTRA            = -Llib -lHist -lMatrix
+GUIBLDLIBEXTRA          = -Llib -lGui -lGraf -lGpad -lRIO -lHist -lMatrix
+GUIHTMLLIBEXTRA         = -Llib -lGui -lGraf -lNet -lGpad -lRIO -lHist -lMatrix
 XMLLIBEXTRA             = -Llib -lRIO
-FOAMLIBEXTRA            = -Llib -lHist
+FOAMLIBEXTRA            = -Llib -lHist -lMatrix
 ALIENLIBEXTRA           = -Llib -lXMLIO -lNetx -lTree -lProof
-ROOFITCORELIBEXTRA      = -Llib -lHist -lGraf -lMatrix -lTree -lMinuit -lHtml -lRIO
-ROOFITLIBEXTRA          = -Llib -lRooFitCore -lTree -lRIO
+ROOFITCORELIBEXTRA      = -Llib -lHist -lGraf -lMatrix -lTree -lMinuit -lHtml \
+                          -lRIO -lNet
+ROOFITLIBEXTRA          = -Llib -lRooFitCore -lTree -lRIO -lHist -lGraf -lMatrix \
+                          -lMinuit -lHtml -lNet
 CINT7LIBEXTRA           = -Llib -lReflex
 CINTEXLIBEXTRA          = -Llib -lReflex
 REFLEXDICTLIBEXTRA      = -Llib -lReflex
 RAUTHLIBEXTRA           = -Llib -lNet -lRIO
-KRB5AUTHLIBEXTRA        = -Llib -lRootAuth -lNet
-SRVAUTHLIBEXTRA         = -Llib -lNet
-QTGSILIBEXTRA           = -Llib -lGui -lGpad
-FITPANELLIBEXTRA        = -Llib -lGui -lGpad -lHist -lGraf -lTree
+KRB5AUTHLIBEXTRA        = -Llib -lRootAuth -lNet -lRIO
+SRVAUTHLIBEXTRA         = -Llib -lNet -lRIO
+QTGSILIBEXTRA           = -Llib -lGui -lGpad -lGraf -lRIO -lHist -lMatrix
+FITPANELLIBEXTRA        = -Llib -lGui -lGpad -lHist -lGraf -lTree -lNet \
+                          -lMatrix -lRIO
 RFIOLIBEXTRA            = -Llib -lRIO
 SQLLIBEXTRA             = -Llib -lNet -lRIO
 ODBCLIBEXTRA            = -Llib -lNet -lRIO
@@ -337,8 +351,8 @@
 CASTORLIBEXTRA          = -Llib -lNet -lRIO
 PGSQLLIBEXTRA           = -Llib -lNet -lRIO
 MYSQLLIBEXTRA           = -Llib -lNet -lRIO
-GDMLLIBEXTRA            = -Llib -lGeom -lXMLIO -lHist -lRIO
-UNURANLIBEXTRA          = -Llib -lHist
+GDMLLIBEXTRA            = -Llib -lGeom -lXMLIO -lHist -lRIO -lMatrix
+UNURANLIBEXTRA          = -Llib -lHist -lMatrix
 
 endif
 
--- xrootd/Module.mk	24 Mar 2007 14:06:26 -0000	1.31
+++ xrootd/Module.mk	1 Jun 2007 03:35:27 -0000
@@ -77,6 +77,16 @@
 		   gunzip -c $(XROOTDVERS).src.tgz | tar xf -; \
 		   etag=`basename $(XROOTDETAG)` ; \
 		   touch $$etag ; \
+		   finkdir=`which fink 2>&1 | sed -ne "s/\/bin\/fink//p"` ; \
+		   major=`sed 's|\(.*\)\..*/.*|\1|' < ../../build/version_number` ; \
+		   minor=`sed 's|.*\.\(.*\)/.*|\1|' < ../../build/version_number` ; \
+		   revis=`sed 's|.*\..*/\([0-9]*\).*|\1|' < ../../build/version_number` ; \
+		   /usr/bin/sed -i '.bak' s:libXrdClient.$$\(TYPEDYLIB\):libXrdClient.$$major.$$minor.$$\(TYPEDYLIB\)\ -install_name\ $$finkdir/lib/root/libXrdClient.$$major.$$\(TYPEDYLIB\)\ -compatibility_version\ $$major\ -current_version\ $$major.$$minor.$$revis:g xrootd/src/XrdClient/GNUmakefile ; \
+		   /bin/ln -s libXrdClient.$$major.$$minor.dylib ../../lib/libXrdClient.$$major.dylib ; \
+		   /bin/ln -s libXrdClient.$$major.$$minor.dylib ../../lib/libXrdClient.dylib ; \
+		   /usr/bin/sed -i '.bak' s:libXrdPosix.$$\(TYPEDYLIB\):libXrdPosix.$$major.$$minor.$$\(TYPEDYLIB\)\ -install_name\ $$finkdir/lib/root/libXrdPosix.$$major.$$\(TYPEDYLIB\)\ -compatibility_version\ $$major\ -current_version\ $$major.$$minor.$$revis:g xrootd/src/XrdPosix/GNUmakefile ; \
+		   /bin/ln -s libXrdPosix.$$major.$$minor.dylib ../../lib/libXrdPosix.$$major.dylib ; \
+		   /bin/ln -s libXrdPosix.$$major.$$minor.dylib ../../lib/libXrdPosix.dylib ; \
 		fi)
 ifeq ($(PLATFORM),win32)
 		@(if [ -d $(XROOTDDIRD)/pthreads-win32 ]; then \
--- config/Makefile.macosx	2007-03-05 09:35:31.000000000 -0600
+++ config/Makefile.macosx	2007-03-30 15:34:01.000000000 -0500
@@ -103,12 +103,9 @@
 F77LIBS       =
 endif
 
-# We add libg2c only in case of ppc because then we probably have cernlib
-# compiled with g77. In case of Intel Mac it should be compiled with the
-# same fortran we use.
-ifeq ($(MACOSX_CPU),powerpc)
-F77LIBS += -L$(FINK_DIR)/lib -lg2c
-endif
+# We add the g77 library libg2c.a only if it exists. In this case it
+# is likely that the CERNLIB was compiled with g77.
+F77LIBS += $(shell [ -r $(FINK_DIR)/lib/libg2c.a ] && echo -L$(FINK_DIR)/lib -lg2c)
 
 ifeq ($(shell find $(FINK_DIR)/bin -name $(F77)),)
 BUILDHBOOK   := no
