--- root.old/configure	9 Dec 2006 10:22:16 -0000	1.377
+++ root.new/configure	16 Dec 2006 22:22:38 -0000
@@ -2428,21 +2428,21 @@
     # for a system library, then see if we have various headers needed.
     if test "x$enable_builtin_afterimage" = "xyes" ; then
        check_header "jpeglib.h" "" $ASIMAGE $ASIMAGE/include \
-          /usr/local/include /usr/include /opt/include
+          $finkdir/include /usr/local/include /usr/include /opt/include
        asjpegincdir=$found_dir
        check_header "png.h" "" $ASIMAGE $ASIMAGE/include \
-          /usr/local/include /usr/include /usr/local/include/libpng \
+          $finkdir/include /usr/local/include /usr/include /usr/local/include/libpng \
           /opt/include
        aspngincdir=$found_dir
        if test ! "x$enable_astiff" = "xno" ; then
            check_header "tiffio.h" "" $ASIMAGE $ASIMAGE/include \
-               /usr/local/include /usr/include /opt/include
+               $finkdir/include /usr/local/include /usr/include /opt/include
            astiffincdir=$found_dir
        else
            astiffincdir="--with-tiff=no"
        fi
        check_header "gif_lib.h" "" $ASIMAGE $ASIMAGE/include \
-          /usr/local/include /usr/include /opt/include
+          $finkdir/include /usr/local/include /usr/include /opt/include
        asgifincdir=$found_dir
 
        asextralib=""
@@ -2460,7 +2460,7 @@
        fi
        for k in $aslibs ; do
            check_library $k "$enable_shared" "" \
-             $ASIMAGE $ASIMAGE/lib /usr/local/lib /usr/lib /opt/lib
+             $ASIMAGE $ASIMAGE/lib $finkdir/lib /usr/local/lib /usr/lib /opt/lib
            if test "x$k" = "xlibz" ; then
                found_libz=$found_lib
                found_dirz=$found_dir
--- root.old/config/Makefile.macosx	2007-03-05 09:35:31.000000000 -0600
+++ root.new/config/Makefile.macosx	2007-03-30 15:34:01.000000000 -0500
@@ -103,12 +103,9 @@
 F77LIBS       =
 endif
 
-# We add libg2c only in case of ppc because then we probably have cernlib
-# compiled with g77. In case of Intel Mac it should be compiled with the
-# same fortran we use.
-ifeq ($(MACOSX_CPU),powerpc)
-F77LIBS += -L$(FINK_DIR)/lib -lg2c
-endif
+# We add the g77 library libg2c.a only if it exists. In this case it
+# is likely that the CERNLIB was compiled with g77.
+F77LIBS += $(shell [ -r $(FINK_DIR)/lib/libg2c.a ] && echo -L$(FINK_DIR)/lib -lg2c)
 
 ifeq ($(shell find $(FINK_DIR)/bin -name $(F77)),)
 BUILDHBOOK   := no
--- root.old/config/Makefile.depend	20 Mar 2007 16:16:04 -0000	1.126
+++ root.new/config/Makefile.depend	1 Apr 2007 00:05:01 -0000
@@ -184,132 +184,152 @@
 HISTLIBEXTRA        = -Llib -lMatrix
 GRAFLIBDEP          = $(HISTLIB) $(MATRIXLIB) $(IOLIB)
 GRAFLIBEXTRA        = -Llib -lHist -lMatrix -lRIO
-GPADLIBDEP          = $(GRAFLIB) $(HISTLIB)
-GPADLIBEXTRA        = -Llib -lGraf -lHist
-G3DLIBDEP           = $(GRAFLIB) $(HISTLIB) $(GPADLIB)
-G3DLIBEXTRA         = -Llib -lGraf -lHist -lGpad
+GPADLIBDEP          = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB) $(IOLIB)
+GPADLIBEXTRA        = -Llib -lGraf -lHist -lMatrix -lRIO
+G3DLIBDEP           = $(GRAFLIB) $(HISTLIB) $(GPADLIB) $(MATRIXLIB) $(IOLIB)
+G3DLIBEXTRA         = -Llib -lGraf -lHist -lGpad -lMatrix -lRIO
 GEDLIBDEP           = $(HISTLIB) $(GPADLIB) $(GRAFLIB) $(GUILIB) $(TREELIB) \
-                      $(TREEPLAYERLIB)
-GEDLIBEXTRA         = -Llib -lHist -lGpad -lGraf -lGui -lTree -lTreePlayer
-POSTSCRIPTLIBDEP    = $(GRAFLIB)
-POSTSCRIPTLIBEXTRA  = -Llib -lGraf
-GUILIBDEP           = $(GPADLIB) $(GRAFLIB) $(IOLIB)
-GUILIBEXTRA         = -Llib -lGpad -lGraf -lRIO
-HISTPAINTERLIBDEP   = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB) $(GPADLIB)
-HISTPAINTERLIBEXTRA = -Llib -lGraf -lHist -lMatrix -lGpad
-SPECTRUMPAINTERLIBDEP   = $(GRAFLIB) $(HISTLIB)
-SPECTRUMPAINTERLIBEXTRA = -Llib -lGraf -lHist
-HTMLLIBDEP          = $(GRAFLIB)
-HTMLLIBEXTRA        = -Llib -lGraf
-MINUITLIBDEP        = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
-MINUITLIBEXTRA      = -Llib -lGraf -lHist -lMatrix
-MINUIT2LIBDEP       = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
-MINUIT2LIBEXTRA     = -Llib -lGraf -lHist -lMatrix
-FUMILILIBDEP        = $(GRAFLIB) $(HISTLIB)
-FUMILILIBEXTRA      = -Llib -lGraf -lHist
+                      $(TREEPLAYERLIB) $(MATRIXLIB) $(IOLIB) $(G3DLIB) $(NETLIB)
+GEDLIBEXTRA         = -Llib -lHist -lGpad -lGraf -lGui -lTree -lTreePlayer \
+                      -lMatrix -lRIO -lGraf3d -lNet
+POSTSCRIPTLIBDEP    = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB) $(IOLIB)
+POSTSCRIPTLIBEXTRA  = -Llib -lGraf -lHist -lMatrix -lRIO
+GUILIBDEP           = $(GPADLIB) $(GRAFLIB) $(IOLIB) $(HISTLIB) $(MATRIXLIB)
+GUILIBEXTRA         = -Llib -lGpad -lGraf -lRIO -lHist -lMatrix
+HISTPAINTERLIBDEP   = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB) $(GPADLIB) $(IOLIB)
+HISTPAINTERLIBEXTRA = -Llib -lGraf -lHist -lMatrix -lGpad -lRIO
+SPECTRUMPAINTERLIBDEP   = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB) $(IOLIB)
+SPECTRUMPAINTERLIBEXTRA = -Llib -lGraf -lHist -lMatrix -lRIO
+HTMLLIBDEP          = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB) $(IOLIB)
+HTMLLIBEXTRA        = -Llib -lGraf -lHist -lMatrix -lRIO
+MINUITLIBDEP        = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB) $(IOLIB)
+MINUITLIBEXTRA      = -Llib -lGraf -lHist -lMatrix -lRIO
+MINUIT2LIBDEP       = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB) $(IOLIB)
+MINUIT2LIBEXTRA     = -Llib -lGraf -lHist -lMatrix -lRIO
+FUMILILIBDEP        = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB) $(IOLIB)
+FUMILILIBEXTRA      = -Llib -lGraf -lHist -lMatrix -lRIO
 TREELIBDEP          = $(NETLIB) $(IOLIB)
 TREELIBEXTRA        = -Llib -lNet -lRIO
 TREEPLAYERLIBDEP    = $(TREELIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB) $(GPADLIB) \
-                      $(IOLIB)
-TREEPLAYERLIBEXTRA  = -Llib -lTree -lGraf3d -lGraf -lHist -lGpad -lRIO
-TREEVIEWERLIBDEP    = $(TREELIB) $(GPADLIB) $(GRAFLIB) $(HISTLIB) $(GUILIB)
-TREEVIEWERLIBEXTRA  = -Llib -lTree -lGpad -lGraf -lHist -lGui
+                      $(IOLIB) $(NETLIB) $(MATRIXLIB)
+TREEPLAYERLIBEXTRA  = -Llib -lTree -lGraf3d -lGraf -lHist -lGpad -lRIO -lNet \
+                      -lMatrix
+TREEVIEWERLIBDEP    = $(TREELIB) $(GPADLIB) $(GRAFLIB) $(HISTLIB) $(GUILIB) \
+                      $(IOLIB) $(NETLIB) $(MATRIXLIB)
+TREEVIEWERLIBEXTRA  = -Llib -lTree -lGpad -lGraf -lHist -lGui -lRIO -lNet \
+                      -lMatrix
 PROOFLIBDEP         = $(NETLIB) $(TREELIB) $(THREADLIB) $(IOLIB)
 PROOFLIBEXTRA       = -Llib -lNet -lTree -lThread -lRIO
 PROOFPLAYERLIBDEP   = $(PROOFLIB) $(NETLIB) $(TREELIB) $(G3DLIB) $(GRAFLIB) \
-                      $(HISTLIB) $(GPADLIB) $(TREEPLAYERLIB) $(THREADLIB) \
-                      $(IOLIB)
+                      $(GPADLIB) $(HISTLIB) $(TREEPLAYERLIB) $(THREADLIB) \
+                      $(IOLIB) $(MATRIXLIB)
 PROOFPLAYERLIBEXTRA = -Llib -lProof -lNet -lTree -lGraf3d -lGraf -lGpad \
-                      -lHist -lTreePlayer -lThread -lRIO
-PROOFXLIBDEP        = $(NETLIB) $(PROOFLIB) $(THREADLIB)
-PROOFXLIBEXTRA      = -Llib -lNet -lProof -lThread
-SESSIONVIEWERLIBDEP   = $(PROOFLIB) $(GUILIB) $(GPADLIB) $(GRAFLIB) $(TREELIB)
-SESSIONVIEWERLIBEXTRA = -Llib -lProof -lGui -lGpad -lGraf -lTree
+                      -lHist -lTreePlayer -lThread -lRIO -lMatrix
+PROOFXLIBDEP        = $(NETLIB) $(PROOFLIB) $(THREADLIB) $(IOLIB) $(TREELIB)
+PROOFXLIBEXTRA      = -Llib -lNet -lProof -lThread -lRIO -lTree
+SESSIONVIEWERLIBDEP   = $(PROOFLIB) $(GUILIB) $(GPADLIB) $(GRAFLIB) $(TREELIB) $(NETLIB) \
+                        $(HISTLIB) $(THREADLIB) $(IOLIB) $(MATRIXLIB)
+SESSIONVIEWERLIBEXTRA = -Llib -lProof -lGui -lGpad -lGraf -lTree -lNet \
+                        -lHist -lThread -lRIO -lMatrix
 PEACLIBDEP          = $(PROOFLIB) $(CLARENSLIB)
 PEACLIBEXTRA        = -Llib -lProof -lClarens
 PEACGUILIBDEP       = $(GUILIB)
 PEACGUILIBEXTRA     = -Llib -lGui
-X3DLIBDEP           = $(G3DLIB) $(GUILIB)
-X3DLIBEXTRA         = -Llib -lGraf3d -lGui
-EGLIBDEP            = $(G3DLIB) $(GRAFLIB) $(GPADLIB)
-EGLIBEXTRA          = -Llib -lGraf3d -lGraf -lGpad
-VMCLIBDEP           = $(EGLIB) $(GEOMLIB)
-VMCLIBEXTRA         = -Llib -lEG -lGeom
+X3DLIBDEP           = $(G3DLIB) $(GUILIB) $(GRAFLIB) $(GPADLIB) $(HISTLIB) $(MATRIXLIB) \
+                      $(IOLIB)
+X3DLIBEXTRA         = -Llib -lGraf3d -lGui -lGraf -lGpad -lHist -lMatrix -lRIO
+EGLIBDEP            = $(G3DLIB) $(GRAFLIB) $(GPADLIB) $(HISTLIB) $(MATRIXLIB) $(IOLIB)
+EGLIBEXTRA          = -Llib -lGraf3d -lGraf -lGpad -lHist -lMatrix -lRIO
+VMCLIBDEP           = $(EGLIB) $(GEOMLIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB) $(MATRIXLIB) \
+                      $(IOLIB) $(GPADLIB)
+VMCLIBEXTRA         = -Llib -lEG -lGeom -lGraf3d -lGraf -lHist -lMatrix -lRIO -lGpad
 PHYSICSLIBDEP       = $(MATRIXLIB)
 PHYSICSLIBEXTRA     = -Llib -lMatrix
-PYTHIA6LIBDEP       = $(EGLIB) $(GRAFLIB) $(VMCLIB) $(PHYSICSLIB)
-PYTHIA6LIBEXTRA     = -Llib -lEG -lGraf -lVMC -lPhysics $(FPYTHIA6LIBDIR) \
-                      $(FPYTHIA6LIB)
-X11TTFLIBDEP        = $(X11LIB) $(GRAFLIB)
-X11TTFLIBEXTRA      = -Llib -lGX11 -lGraf
+PYTHIA6LIBDEP       = $(EGLIB) $(GRAFLIB) $(VMCLIB) $(PHYSICSLIB) $(G3DLIB) $(HISTLIB) \
+                      $(MATRIXLIB) $(GPADLIB) $(GEOMLIB) $(IOLIB)
+PYTHIA6LIBEXTRA     = -Llib -lEG -lGraf -lVMC -lPhysics -lGraf3d -lHist -lMatrix \
+                      -lGpad -lGeom -lRIO $(FPYTHIA6LIBDIR) $(FPYTHIA6LIB)
+X11TTFLIBDEP        = $(X11LIB) $(GRAFLIB) $(HISTLIB) $(MATRIXLIB) $(IOLIB)
+X11TTFLIBEXTRA      = -Llib -lGX11 -lGraf -lHist -lMatrix -lRIO
 TABLELIBDEP         = $(TREELIB) $(GPADLIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB) \
-                      $(IOLIB)
-TABLELIBEXTRA       = -Llib -lTree -lGpad -lGraf3d -lGraf -lHist -lRIO
+                      $(IOLIB) $(MATRIXLIB) $(NETLIB)
+TABLELIBEXTRA       = -Llib -lTree -lGpad -lGraf3d -lGraf -lHist -lRIO \
+                      -lMatrix -lNet
 GLLIBDEP            = $(GPADLIB) $(G3DLIB) $(GUILIB) $(GRAFLIB) $(HISTLIB) \
-                      $(GEDLIB)
-GLLIBEXTRA          = -Llib -lGpad -lGraf3d -lGui -lGraf -lHist -lGed
+                      $(GEDLIB) $(MATRIXLIB) $(TREELIB) $(TREEPLAYERLIB) $(NETLIB) \
+                      $(IOLIB)
+GLLIBEXTRA          = -Llib -lGpad -lGraf3d -lGui -lGraf -lHist -lGed \
+                      -lMatrix -lTree -lTreePlayer -lNet -lRIO
 HBOOKLIBDEP         = $(HISTLIB) $(MATRIXLIB) $(TREELIB) $(GRAFLIB) \
-                      $(TREEPLAYERLIB) $(IOLIB)
-HBOOKLIBEXTRA       = -Llib -lHist -lMatrix -lTree -lGraf -lTreePlayer -lRIO
+                      $(TREEPLAYERLIB) $(IOLIB) $(GPADLIB) $(G3DLIB) $(NETLIB)
+HBOOKLIBEXTRA       = -Llib -lHist -lMatrix -lTree -lGraf -lTreePlayer -lRIO -lGpad \
+                      -lGraf3d -lNet
 GEOMLIBDEP          = $(IOLIB)
 GEOMLIBEXTRA        = -Llib -lRIO
-GEOMPAINTERLIBDEP   = $(GEOMLIB) $(TREELIB) $(G3DLIB) $(HISTLIB) $(GPADLIB)
-GEOMPAINTERLIBEXTRA = -Llib -lGeom -lTree -lGraf3d -lHist -lGpad
+GEOMPAINTERLIBDEP   = $(GEOMLIB) $(TREELIB) $(G3DLIB) $(HISTLIB) $(GPADLIB) \
+                      $(GRAFLIB) $(IOLIB) $(MATRIXLIB) $(NETLIB)
+GEOMPAINTERLIBEXTRA = -Llib -lGeom -lTree -lGraf3d -lHist -lGpad -lGraf -lRIO \
+                      -lMatrix -lNet
 GEOMBUILDERLIBDEP   = $(GEOMLIB) $(G3DLIB) $(GPADLIB) $(GRAFLIB) $(GUILIB) \
-		      $(GEDLIB)
-GEOMBUILDERLIBEXTRA = -Llib -lGeom -lGraf3d -lGpad -lGraf -lGui -lGed
-ASIMAGELIBDEP       = $(GRAFLIB) $(HISTLIB) $(GPADLIB)
-ASIMAGELIBEXTRA     = -Llib -lGraf -lHist -lGpad
-ASIMAGEGUILIBDEP    = $(GRAFLIB) $(HISTLIB) $(GUILIB) $(ASIMAGELIB) $(IOLIB)
-ASIMAGEGUILIBEXTRA  = -Llib -lGraf -lHist -lGui -lASImage -lRIO
-ASIMAGEGSLIBDEP     = $(GRAFLIB) $(ASIMAGELIB)
-ASIMAGEGSLIBEXTRA   = -Llib -lGraf -lASImage
+		      $(GEDLIB) $(HISTLIB) $(MATRIXLIB) $(TREEPLAYERLIB) $(IOLIB) \
+		      $(NETLIB) $(TREELIB)
+GEOMBUILDERLIBEXTRA = -Llib -lGeom -lGraf3d -lGpad -lGraf -lGui -lGed \
+                      -lHist -lMatrix -lTreePlayer -lRIO -lNet -lTree
+ASIMAGELIBDEP       = $(GRAFLIB) $(HISTLIB) $(GPADLIB) $(IOLIB) $(MATRIXLIB)
+ASIMAGELIBEXTRA     = -Llib -lGraf -lHist -lGpad -lRIO -lMatrix
+ASIMAGEGUILIBDEP    = $(GRAFLIB) $(HISTLIB) $(GUILIB) $(ASIMAGELIB) $(IOLIB) \
+		      $(GPADLIB) $(MATRIXLIB)
+ASIMAGEGUILIBEXTRA  = -Llib -lGraf -lHist -lGui -lASImage -lRIO -lGpad -lMatrix
+ASIMAGEGSLIBDEP     = $(GRAFLIB) $(ASIMAGELIB) $(HISTLIB) $(GPADLIB) $(MATRIXLIB) $(IOLIB)
+ASIMAGEGSLIBEXTRA   = -Llib -lGraf -lASImage -lHist -lGpad -lMatrix -lRIO
 MLPLIBDEP           = $(HISTLIB) $(MATRIXLIB) $(TREELIB) $(GRAFLIB) \
-                      $(GPADLIB) $(TREEPLAYERLIB)
+                      $(GPADLIB) $(TREEPLAYERLIB) $(IOLIB) $(NETLIB) $(G3DLIB)
 MLPLIBEXTRA         = -Llib -lHist -lMatrix -lTree -lGraf -lGpad \
-                      -lTreePlayer
+                      -lTreePlayer -lRIO -lNet -lGraf3d
 SPECTRUMLIBDEP      = $(HISTLIB) $(MATRIXLIB)
 SPECTRUMLIBEXTRA    = -Llib -lHist -lMatrix
 TMVALIBDEP          = $(IOLIB) $(HISTLIB) $(MATRIXLIB) $(TREELIB) $(GRAFLIB) \
-                      $(GPADLIB) $(TREEPLAYERLIB) $(MLPLIB)
+                      $(GPADLIB) $(TREEPLAYERLIB) $(MLPLIB) $(NETLIB) $(G3DLIB)
 TMVALIBEXTRA        = -Llib -lRIO -lHist -lMatrix -lTree -lGraf -lGpad \
-                      -lTreePlayer -lMLP
+                      -lTreePlayer -lMLP -lNet -lGraf3d
 SPLOTLIBDEP         = $(MATRIXLIB) $(HISTLIB) $(TREELIB) $(TREEPLAYERLIB) \
-                      $(G3DLIB) $(GRAFLIB)
-SPLOTLIBEXTRA       = -Llib -lMatrix -lHist -lTree -lTreePlayer -lGraf3d -lGraf
+                      $(G3DLIB) $(GRAFLIB) $(IOLIB) $(NETLIB) $(GPADLIB)
+SPLOTLIBEXTRA       = -Llib -lMatrix -lHist -lTree -lTreePlayer -lGraf3d -lGraf \
+                      -lRIO -lNet -lGpad
 QTROOTLIBDEP        = $(GUILIB) $(GQTLIB)
 QTROOTLIBEXTRA      = -Llib -lGui -lGQt
 GQTLIBDEP           = $(GPADLIB) $(GRAFLIB) $(RINTLIB)
 GQTLIBEXTRA         = -Llib -lGpad -lGraf -lRint
 QUADPLIBDEP         = $(MATRIXLIB)
 QUADPLIBEXTRA       = -Llib -lMatrix
-RUBYLIBDEP          = $(HISTLIB)
-RUBYLIBEXTRA        = -Llib -lHist
-GUIBLDLIBDEP        = $(GUILIB) $(GRAFLIB)
-GUIBLDLIBEXTRA      = -Llib -lGui -lGraf
+RUBYLIBDEP          = $(HISTLIB) $(MATRIXLIB)
+RUBYLIBEXTRA        = -Llib -lHist -lMatrix
+GUIBLDLIBDEP        = $(GUILIB) $(GRAFLIB) $(HISTLIB) $(GPADLIB) $(MATRIXLIB) $(IOLIB)
+GUIBLDLIBEXTRA      = -Llib -lGui -lGraf -lHist -lGpad -lMatrix -lRIO
 XMLLIBDEP           = $(IOLIB)
 XMLLIBEXTRA         = -Llib -lRIO
-FOAMLIBDEP          = $(HISTLIB)
-FOAMLIBEXTRA        = -Llib -lHist
+FOAMLIBDEP          = $(HISTLIB) $(MATRIXLIB)
+FOAMLIBEXTRA        = -Llib -lHist -lMatrix
 ALIENLIBDEP         = $(XMLLIB) $(NETXLIB) $(TREELIB) $(PROOFLIB)
 ALIENLIBEXTRA       = -Llib -lXMLIO -lNetx -lTree -lProof
 ROOFITLIBDEP        = $(HISTLIB) $(GRAFLIB) $(MATRIXLIB) $(TREELIB) \
-                      $(MINUITLIB) $(HTMLLIB) $(IOLIB)
-ROOFITLIBEXTRA      = -Llib -lHist -lGraf -lMatrix -lTree -lMinuit -lHtml -lRIO
+                      $(MINUITLIB) $(HTMLLIB) $(IOLIB) $(NETLIB)
+ROOFITLIBEXTRA      = -Llib -lHist -lGraf -lMatrix -lTree -lMinuit -lHtml -lRIO -lNet
 CINT7LIBDEP         = $(REFLEXLIB)
 CINT7LIBEXTRA       = -Llib -lReflex
 CINTEXLIBDEP        = $(REFLEXLIB)
 CINTEXLIBEXTRA      = -Llib -lReflex
 RAUTHLIBDEP         = $(NETLIB) $(IOLIB)
 RAUTHLIBEXTRA       = -Llib -lNet -lRIO
-KRB5AUTHLIBDEP      = $(RAUTHLIB) $(NETLIB)
-KRB5AUTHLIBEXTRA    = -Llib -lRootAuth -lNet
-SRVAUTHLIBDEP       = $(NETLIB)
-SRVAUTHLIBEXTRA     = -Llib -lNet
+KRB5AUTHLIBDEP      = $(RAUTHLIB) $(NETLIB) $(IOLIB)
+KRB5AUTHLIBEXTRA    = -Llib -lRootAuth -lNet -lRIO
+SRVAUTHLIBDEP       = $(NETLIB) $(IOLIB)
+SRVAUTHLIBEXTRA     = -Llib -lNet -lRIO
 QTGSILIBDEP         = $(GUILIB) $(GPADLIB)
 QTGSILIBEXTRA       = -Llib -lGui -lGpad
-FITPANELLIBDEP      = $(GUILIB) $(GPADLIB) $(HISTLIB) $(GRAFLIB) $(TREELIB)
-FITPANELLIBEXTRA    = -Llib -lGui -lGpad -lHist -lGraf -lTree
+FITPANELLIBDEP      = $(GUILIB) $(GPADLIB) $(HISTLIB) $(GRAFLIB) $(TREELIB)  \
+                      $(MATRIXLIB) $(NETLIB) $(IOLIB)
+FITPANELLIBEXTRA    = -Llib -lGui -lGpad -lHist -lGraf -lTree -lMatrix -lNet -lRIO
 RFIOLIBDEP          = $(IOLIB)
 RFIOLIBEXTRA        = -Llib -lRIO
 SQLLIBDEP           = $(NETLIB) $(IOLIB)
@@ -326,9 +346,9 @@
 PGSQLLIBEXTRA       = -Llib -lNet -lRIO
 MYSQLLIBDEP         = $(NETLIB) $(IOLIB)
 MYSQLLIBEXTRA       = -Llib -lNet -lRIO
-GDMLLIBDEP          = $(GEOMLIB) $(XMLLIB) $(HISTLIB) $(IOLIB)
-GDMLLIBEXTRA        = -Llib -lGeom -lXMLIO -lHist -lRIO
-UNURANLIBDEP        = $(HISTLIB)
-UNURANLIBEXTRA      = -Llib -lHist
+GDMLLIBDEP          = $(GEOMLIB) $(XMLLIB) $(HISTLIB) $(IOLIB) $(MATRIXLIB)
+GDMLLIBEXTRA        = -Llib -lGeom -lXMLIO -lHist -lRIO -lMatrix
+UNURANLIBDEP        = $(HISTLIB) $(MATRIXLIB)
+UNURANLIBEXTRA      = -Llib -lHist -lMatrix
 
 endif
