--- root.old/configure	9 Dec 2006 10:22:16 -0000	1.377
+++ root.new/configure	16 Dec 2006 22:22:38 -0000
@@ -2392,21 +2392,21 @@
     # for a system library, then see if we have various headers needed.
     if test "x$enable_builtin_afterimage" = "xyes" ; then
        check_header "jpeglib.h" "" $ASIMAGE $ASIMAGE/include \
-          /usr/local/include /usr/include /opt/include
+          $finkdir/include /usr/local/include /usr/include /opt/include
        asjpegincdir=$found_dir
        check_header "png.h" "" $ASIMAGE $ASIMAGE/include \
-          /usr/local/include /usr/include /usr/local/include/libpng \
+          $finkdir/include /usr/local/include /usr/include /usr/local/include/libpng \
           /opt/include
        aspngincdir=$found_dir
        if test ! "x$enable_astiff" = "xno" ; then
            check_header "tiffio.h" "" $ASIMAGE $ASIMAGE/include \
-               /usr/local/include /usr/include /opt/include
+               $finkdir/include /usr/local/include /usr/include /opt/include
            astiffincdir=$found_dir
        else
            astiffincdir="--with-tiff=no"
        fi
        check_header "gif_lib.h" "" $ASIMAGE $ASIMAGE/include \
-          /usr/local/include /usr/include /opt/include
+          $finkdir/include /usr/local/include /usr/include /opt/include
        asgifincdir=$found_dir
 
        asextralib=""
@@ -2424,7 +2424,7 @@
        fi
        for k in $aslibs ; do
            check_library $k "$enable_shared" "" \
-             $ASIMAGE $ASIMAGE/lib /usr/local/lib /usr/lib /opt/lib
+             $ASIMAGE $ASIMAGE/lib $finkdir/lib /usr/local/lib /usr/lib /opt/lib
            if test "x$k" = "xlibz" ; then
                found_libz=$found_lib
                found_dirz=$found_dir
--- root.old/config/Makefile.depend	6 Dec 2006 14:42:26 -0000	1.97
+++ root.new/config/Makefile.depend	16 Dec 2006 22:22:38 -0000
@@ -4,85 +4,97 @@
 
 ifeq ($(PLATFORM),win32)
 
-GDMLLIBDEP          = $(GEOMLIB) $(XMLLIB) $(HISTLIB) 
-GDMLLIBEXTRA        = lib/libGeom.lib lib/libXMLIO.lib lib/libHist.lib 
+GDMLLIBDEP          = $(GEOMLIB) $(XMLLIB) $(HISTLIB) $(MATRIXLIB)
+GDMLLIBEXTRA        = lib/libGeom.lib lib/libXMLIO.lib lib/libHist.lib \
+                      lib/libMatrix.lib
 HISTLIBDEP          = $(MATRIXLIB)
 HISTLIBEXTRA        = lib/libMatrix.lib
 GRAFLIBDEP          = $(HISTLIB) $(MATRIXLIB)
 GRAFLIBEXTRA        = lib/libHist.lib lib/libMatrix.lib
-GPADLIBDEP          = $(GRAFLIB) $(HISTLIB)
-GPADLIBEXTRA        = lib/libGraf.lib lib/libHist.lib
-G3DLIBDEP           = $(GRAFLIB) $(HISTLIB)
-G3DLIBEXTRA         = lib/libGraf.lib lib/libHist.lib
+GPADLIBDEP          = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+GPADLIBEXTRA        = lib/libGraf.lib lib/libHist.lib lib/libMatrix.lib
+G3DLIBDEP           = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+G3DLIBEXTRA         = lib/libGraf.lib lib/libHist.lib lib/libMatrix.lib
 WIN32GDKLIBDEP      = $(GPADLIB) $(GRAFLIB)
 WIN32GDKLIBEXTRA    = lib/libGpad.lib lib/libGraf.lib
-ASIMAGELIBDEP       = $(GRAFLIB) $(HISTLIB) $(GPADLIB)
-ASIMAGELIBEXTRA     = lib/libGraf.lib lib/libHist.lib lib/libGpad.lib
-ASIMAGEGUILIBDEP    = $(GRAFLIB) $(HISTLIB) $(GUILIB) $(ASIMAGELIB)
+ASIMAGELIBDEP       = $(GRAFLIB) $(HISTLIB) $(GPADLIB) $(MATRIXLIB)
+ASIMAGELIBEXTRA     = lib/libGraf.lib lib/libHist.lib lib/libGpad.lib lib/libMatrix.lib
+ASIMAGEGUILIBDEP    = $(GRAFLIB) $(HISTLIB) $(GUILIB) $(ASIMAGELIB) $(GPADLIB) $(MATRIXLIB)
 ASIMAGEGUILIBEXTRA  = lib/libGraf.lib lib/libHist.lib lib/libGui.lib \
-                      lib/libASImage.lib
-ASIMAGEGSLIBDEP     = $(GRAFLIB) $(ASIMAGELIB)
-ASIMAGEGSLIBEXTRA   = lib/libGraf.lib lib/libASImage.lib
+                      lib/libASImage.lib lib/libGpad.lib lib/libMatrix.lib
+ASIMAGEGSLIBDEP     = $(GRAFLIB) $(ASIMAGELIB) $(HISTLIB) $(GPADLIB) $(MATRIXLIB)
+ASIMAGEGSLIBEXTRA   = lib/libGraf.lib lib/libASImage.lib lib/libHist.lib \
+                      lib/libGpad.lib lib/libMatrix.lib
 GEDLIBDEP           = $(HISTLIB) $(GPADLIB) $(GRAFLIB) $(GUILIB) $(TREELIB) \
-                      $(TREEPLAYERLIB)
+                      $(TREEPLAYERLIB) $(MARIXLIB) $(G3DLIB)
 GEDLIBEXTRA         = lib/libHist.lib lib/libGpad.lib lib/libGraf.lib \
-                      lib/libTree.lib lib/libTreePlayer.lib lib/libGui.lib
-POSTSCRIPTLIBDEP    = $(GRAFLIB)
-POSTSCRIPTLIBEXTRA  = lib/libGraf.lib
-GUILIBDEP           = $(GPADLIB) $(GRAFLIB)
-GUILIBEXTRA         = lib/libGpad.lib lib/libGraf.lib
+                      lib/libTree.lib lib/libTreePlayer.lib lib/libGui.lib \
+                      lib/libMatrix.lib lib/libGraf3d.lib
+POSTSCRIPTLIBDEP    = $(GRAFLIB) $(HISTLIB) $(MARIXLIB)
+POSTSCRIPTLIBEXTRA  = lib/libGraf.lib lib/libHist.lib lib/libMatrix.lib
+GUILIBDEP           = $(GPADLIB) $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+GUILIBEXTRA         = lib/libGpad.lib lib/libGraf.lib lib/libHist.lib lib/libMatrix.lib
 HISTPAINTERLIBDEP   = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
 HISTPAINTERLIBEXTRA = lib/libGraf.lib lib/libHist.lib lib/libMatrix.lib
-SPECTRUMPAINTERLIBDEP   = $(GRAFLIB) $(HISTLIB)
-SPECTRUMPAINTERLIBEXTRA = lib/libGraf.lib lib/libHist.lib
-HTMLLIBDEP          = $(GRAFLIB)
-HTMLLIBEXTRA        = lib/libGraf.lib
+SPECTRUMPAINTERLIBDEP   = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+SPECTRUMPAINTERLIBEXTRA = lib/libGraf.lib lib/libHist.lib lib/libMatrix.lib
+HTMLLIBDEP          = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+HTMLLIBEXTRA        = lib/libGraf.lib lib/libHist.lib lib/libMatrix.lib
 MINUITLIBDEP        = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
 MINUITLIBEXTRA      = lib/libGraf.lib lib/libHist.lib lib/libMatrix.lib
 MINUIT2LIBDEP       = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
 MINUIT2LIBEXTRA     = lib/libGraf.lib lib/libHist.lib lib/libMatrix.lib
-FUMILILIBDEP        = $(GRAFLIB) $(HISTLIB)
-FUMILILIBEXTRA      = lib/libGraf.lib lib/libHist.lib
-TREEPLAYERLIBDEP    = $(TREELIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB)
+FUMILILIBDEP        = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+FUMILILIBEXTRA      = lib/libGraf.lib lib/libHist.lib lib/libMatrix.lib
+TREEPLAYERLIBDEP    = $(TREELIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
 TREEPLAYERLIBEXTRA  = lib/libTree.lib lib/libGraf3d.lib \
-                      lib/libGraf.lib lib/libHist.lib
-TREEVIEWERLIBDEP    = $(TREELIB) $(GPADLIB) $(GRAFLIB) $(HISTLIB) $(GUILIB)
+                      lib/libGraf.lib lib/libHist.lib lib/libMatrix.lib
+TREEVIEWERLIBDEP    = $(TREELIB) $(GPADLIB) $(GRAFLIB) $(HISTLIB) $(GUILIB) $(MATRIXLIB)
 TREEVIEWERLIBEXTRA  = lib/libTree.lib lib/libGpad.lib lib/libGraf.lib \
-                      lib/libHist.lib lib/libGui.lib
+                      lib/libHist.lib lib/libGui.lib lib/libMatrix.lib
 PROOFLIBDEP         = $(TREELIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB) \
-                      $(TREEPLAYERLIB) $(THREADLIB)
+                      $(TREEPLAYERLIB) $(THREADLIB) $(MATRIXLIB)
 PROOFLIBEXTRA       = lib/libTree.lib lib/libGraf3d.lib lib/libGraf.lib \
-                      lib/libHist.lib lib/libTreePlayer.lib lib/libThread.lib
-PROOFXLIBDEP        = $(PROOFLIB) $(THREADLIB) $(XPDLIB)
-PROOFXLIBEXTRA      = lib/libProof.lib lib/libThread.lib lib/libXrdProofd.lib
-PROOFGUILIBDEP      = $(PROOFLIB) $(GUILIB) $(GPADLIB) $(GRAFLIB) $(TREELIB)
-PROOFGUILIBEXTRA    = lib/libProof.lib lib/libGui.lib lib/libGpad.lib \
-                      lib/libGraf.lib lib/libTree.lib
+                      lib/libHist.lib lib/libTreePlayer.lib lib/libThread.lib \
+                      lib/libMatrix.lib
+PROOFXLIBDEP        = $(PROOFLIB) $(THREADLIB) $(XPDLIB) $(TREELIB) $(G3DLIB) \
+                      $(GRAFLIB) $(HISTLIB) $(TREEPLAYERLIB) $(MATRIXLIB)
+PROOFXLIBEXTRA      = lib/libProof.lib lib/libThread.lib lib/libXrdProofd.lib \
+                      lib/libTree.lib lib/libGraf3d.lib lib/libGraf.lib \
+                      lib/libHist.lib lib/libTreePlayer.lib lib/libMatrix.lib
+PROOFGUILIBDEP      = $(PROOFLIB) $(GUILIB) $(TREELIB) $(G3DLIB) $(GRAFLIB) $(GPADLIB) \
+                      $(HISTLIB) $(TREEPLAYERLIB) $(THREADLIB) $(MATRIXLIB)
+PROOFGUILIBEXTRA    = lib/libProof.lib lib/libGui.lib lib/libTree.lib lib/libGraf3d.lib \
+                      lib/libGraf.lib lib/libGpad.lib lib/libHist.lib lib/libTreePlayer.lib \
+                      lib/libThread.lib lib/libMatrix.lib
 PEACLIBDEP          = $(PROOFLIB) $(CLARENSLIB)
 PEACLIBEXTRA        = lib/libProof.lib lib/libClarens.lib
 PEACGUILIBDEP       = $(GUILIB)
 PEACGUILIBEXTRA     = lib/libGui.lib
-EGLIBDEP            = $(G3DLIB) $(GRAFLIB)
-EGLIBEXTRA          = lib/libGraf3d.lib lib/libGraf.lib
-VMCLIBDEP           = $(EGLIB) $(GEOMLIB)
-VMCLIBEXTRA         = lib/libEG.lib lib/libGeom.lib
+EGLIBDEP            = $(G3DLIB) $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+EGLIBEXTRA          = lib/libGraf3d.lib lib/libGraf.lib lib/libHist.lib lib/libMatrix.lib
+VMCLIBDEP           = $(EGLIB) $(GEOMLIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+VMCLIBEXTRA         = lib/libEG.lib lib/libGeom.lib \
+                      lib/libGraf3d.lib lib/libGraf.lib lib/libHist.lib lib/libMatrix.lib
 PHYSICSLIBDEP       = $(MATRIXLIB)
 PHYSICSLIBEXTRA     = lib/libMatrix.lib
-PYTHIA6LIBDEP       = $(EGLIB) $(GRAFLIB) $(VMCLIB)
-PYTHIA6LIBEXTRA     = lib/libEG.lib lib/libGraf.lib lib/libVMC.lib
-TABLELIBDEP         = $(TREELIB) $(GPADLIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB)
+PYTHIA6LIBDEP       = $(EGLIB) $(GRAFLIB) $(VMCLIB) $(G3DLIB) $(HISTLIB) $(MATRIXLIB) $(GEOMLIB)
+PYTHIA6LIBEXTRA     = lib/libEG.lib lib/libGraf.lib lib/libVMC.lib lib/libGraf3d.lib \
+                      lib/libHist.lib lib/libMatrix.lib lib/libGeom.lib
+TABLELIBDEP         = $(TREELIB) $(GPADLIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
 TABLELIBEXTRA       = lib/libTree.lib lib/libGpad.lib \
-                      lib/libGraf3d.lib lib/libGraf.lib lib/libHist.lib
-MLPLIBDEP           = $(HISTLIB) $(MATRIXLIB) $(TREELIB) $(GRAFLIB) \
+                      lib/libGraf3d.lib lib/libGraf.lib lib/libHist.lib lib/libMatrix.lib
+MLPLIBDEP           = $(HISTLIB) $(MATRIXLIB) $(TREELIB) $(G3DLIB) $(GRAFLIB) \
                       $(GPADLIB) $(TREEPLAYERLIB)
 MLPLIBEXTRA         = lib/libHist.lib lib/libMatrix.lib lib/libTree.lib \
+                      lib/libGraf3d.lib lib/libGraf.lib lib/libGpad.lib lib/libTreePlayer.lib
                       lib/libGraf.lib lib/libGpad.lib lib/libTreePlayer.lib
 SPECTRUMLIBDEP      = $(HISTLIB) $(MATRIXLIB)
 SPECTRUMLIBEXTRA    = lib/libHist.lib lib/libMatrix.lib
 TMVALIBDEP          = $(HISTLIB) $(MATRIXLIB) $(TREELIB) $(GRAFLIB) \
-                      $(GPADLIB) $(TREEPLAYERLIB) $(MLPLIB)
+                      $(G3DLIB) $(GPADLIB) $(TREEPLAYERLIB) $(MLPLIB)
 TMVALIBEXTRA        = lib/libHist.lib lib/libMatrix.lib lib/libTree.lib \
-                      lib/libGraf.lib lib/libGpad.lib lib/libTreePlayer.lib \
+                      lib/libGraf.lib lib/libGraf3d.lib lib/libGpad.lib lib/libTreePlayer.lib \
                       lib/libMLP.lib
 SPLOTLIBDEP         = $(MATRIXLIB) $(HISTLIB) $(TREELIB) $(TREEPLAYERLIB) \
                       $(G3DLIB) $(GRAFLIB)
@@ -90,30 +102,32 @@
                       lib/libTreePlayer.lib lib/libGraf3d.lib lib/libGraf.lib
 QUADPLIBDEP         = $(MATRIXLIB)
 QUADPLIBEXTRA       = lib/libMatrix.lib
-GLLIBDEP            = $(G3DLIB) $(GUILIB) $(GRAFLIB) $(HISTLIB) $(GEDLIB)
+GLLIBDEP            = $(G3DLIB) $(GUILIB) $(GRAFLIB) $(HISTLIB) $(GEDLIB) \
+                      $(MATRIXLIB) $(TREELIB) $(TREEPLAYERLIB)
 GLLIBEXTRA          = lib/libGraf3d.lib lib/libGui.lib lib/libGraf.lib \
-                      lib/libHist.lib lib/libGed.lib
+                      lib/libHist.lib lib/libGed.lib lib/libMatrix.lib \
+                      lib/libTree.lib lib/libTreePlayer.lib
 HBOOKLIBDEP         = $(HISTLIB) $(MATRIXLIB) $(TREELIB) $(GRAFLIB) \
-                      $(TREEPLAYERLIB)
+                      $(G3DLIB) $(TREEPLAYERLIB)
 HBOOKLIBEXTRA       = lib/libHist.lib lib/libMatrix.lib lib/libTree.lib \
-                      lib/libGraf.lib lib/libTreePlayer.lib
-GEOMPAINTERLIBDEP   = $(GEOMLIB) $(TREELIB) $(G3DLIB) $(HISTLIB)
-GEOMPAINTERLIBEXTRA = lib/libGeom.lib lib/libTree.lib lib/libGraf3d.lib \
-                      lib/libHist.lib
-GEOMBUILDERLIBDEP   = $(GEOMLIB) $(G3DLIB) $(GPADLIB) $(GRAFLIB) $(GUILIB) \
-		      $(GEDLIB)
-GEOMBUILDERLIBEXTRA = lib/libGeom.lib  lib/libGraf3d.lib lib/libGpad.lib \
-                      lib/libGraf.lib lib/libGui.lib lib/libGed.lib
+                      lib/libGraf.lib lib/libGraf3d.lib lib/libTreePlayer.lib
+GEOMBUILDERLIBDEP   = $(GEOMLIB) $(TREELIB) $(G3DLIB) $(GPADLIB) $(GRAFLIB) \
+		      $(GUILIB) $(GEDLIB) $(HISTLIB) $(MATRIXLIB) $(TREEPLAYERLIB)
+GEOMBUILDERLIBEXTRA = lib/libGeom.lib lib/libTree.lib lib/libGraf3d.lib \
+                      lib/libGpad.lib lib/libGraf.lib lib/libGui.lib \
+                      lib/libGed.lib lib/libHist.lib lib/libMatrix.lib \
+                      lib/libTreePlayer.lib
 QTROOTLIBDEP        = $(GUILIB) $(GQTLIB)
 QTROOTLIBEXTRA      = lib/libGui.lib lib/libGQt.lib
 GQTLIBDEP           = $(GPADLIB) $(RINTLIB)
 GQTLIBEXTRA         = lib/libGpad.lib lib/libRint.lib
-RUBYLIBDEP          = $(HISTLIB)
-RUBYLIBEXTRA        = lib/libHist.lib
-GUIBLDLIBDEP        = $(GUILIB) $(GRAFLIB)
-GUIBLDLIBEXTRA      = lib/libGui.lib lib/libGraf.lib
-FOAMLIBDEP          = $(HISTLIB)
-FOAMLIBEXTRA        = lib/libHist.lib
+RUBYLIBDEP          = $(HISTLIB) $(MATRIXLIB)
+RUBYLIBEXTRA        = lib/libHist.lib lib/libMatrix.lib
+GUIBLDLIBDEP        = $(GUILIB) $(GRAFLIB) $(GPADLIB) $(HISTLIB) $(MATRIXLIB)
+GUIBLDLIBEXTRA      = lib/libGui.lib lib/libGraf.lib lib/libGpad.lib \
+                      lib/libHist.lib lib/libMatrix.lib
+FOAMLIBDEP          = $(HISTLIB) $(MATRIXLIB)
+FOAMLIBEXTRA        = lib/libHist.lib lib/libMatrix.lib
 ALIENLIBDEP         = $(XMLLIB) $(NETXLIB)
 ALIENLIBEXTRA       = lib/libXMLIO.lib lib/libNetx.lib
 ROOFITLIBDEP        = $(HISTLIB) $(GRAFLIB) $(MATRIXLIB) $(TREELIB) \
@@ -128,96 +142,106 @@
 KRB5AUTHLIBEXTRA    = lib/libRootAuth.lib
 QTGSILIBDEP         = $(GUILIB) $(GPADLIB)
 QTGSILIBEXTRA       = lib/libGui.lib lib/libGpad.lib
-FITPANELLIBDEP      = $(GUILIB) $(GPADLIB) $(HISTLIB) $(GRAFLIB) $(TREELIB)
+FITPANELLIBDEP      = $(GUILIB) $(GPADLIB) $(HISTLIB) $(GRAFLIB) $(TREELIB) \
+                      $(MATRIXLIB)
 FITPANELLIBEXTRA    = lib/libGui.lib lib/libGpad.lib lib/libHist.lib \
-                      lib/libGraf.lib lib/libTree.lib 
+                      lib/libGraf.lib lib/libTree.lib lib/libMatrix.lib
 
 else
 
-GDMLLIBDEP          = $(GEOMLIB) $(XMLLIB) $(HISTLIB)
-GDMLLIBEXTRA        = -Llib -lGeom -lXMLIO -lHist
+GDMLLIBDEP          = $(GEOMLIB) $(XMLLIB) $(HISTLIB) $(MATRIXLIB)
+GDMLLIBEXTRA        = -Llib -lGeom -lXMLIO -lHist -lMatrix
 HISTLIBDEP          = $(MATRIXLIB)
 HISTLIBEXTRA        = -Llib -lMatrix
 GRAFLIBDEP          = $(HISTLIB) $(MATRIXLIB)
 GRAFLIBEXTRA        = -Llib -lHist -lMatrix
-GPADLIBDEP          = $(GRAFLIB) $(HISTLIB)
-GPADLIBEXTRA        = -Llib -lGraf -lHist
-G3DLIBDEP           = $(GRAFLIB) $(HISTLIB)
-G3DLIBEXTRA         = -Llib -lGraf -lHist
+GPADLIBDEP          = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+GPADLIBEXTRA        = -Llib -lGraf -lHist -lMatrix
+G3DLIBDEP           = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+G3DLIBEXTRA         = -Llib -lGraf -lHist -lMatrix
 GEDLIBDEP           = $(HISTLIB) $(GPADLIB) $(GRAFLIB) $(GUILIB) $(TREELIB) \
-                      $(TREEPLAYERLIB)
-GEDLIBEXTRA         = -Llib -lHist -lGpad -lGraf -lGui -lTree -lTreePlayer
-POSTSCRIPTLIBDEP    = $(GRAFLIB)
-POSTSCRIPTLIBEXTRA  = -Llib -lGraf
-GUILIBDEP           = $(GPADLIB) $(GRAFLIB)
-GUILIBEXTRA         = -Llib -lGpad -lGraf
+                      $(TREEPLAYERLIB) $(MATRIXLIB) $(G3DLIB)
+GEDLIBEXTRA         = -Llib -lHist -lGpad -lGraf -lGui -lTree -lTreePlayer \
+                      -lMatrix -lGraf3d
+POSTSCRIPTLIBDEP    = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+POSTSCRIPTLIBEXTRA  = -Llib -lGraf -lHist -lMatrix
+GUILIBDEP           = $(GPADLIB) $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+GUILIBEXTRA        := -Llib -lGpad -lGraf -lHist -lMatrix $(GUILIBEXTRA)
 HISTPAINTERLIBDEP   = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
 HISTPAINTERLIBEXTRA = -Llib -lGraf -lHist -lMatrix
-SPECTRUMPAINTERLIBDEP   = $(GRAFLIB) $(HISTLIB)
-SPECTRUMPAINTERLIBEXTRA = -Llib -lGraf -lHist
-HTMLLIBDEP          = $(GRAFLIB)
-HTMLLIBEXTRA        = -Llib -lGraf
+SPECTRUMPAINTERLIBDEP   = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+SPECTRUMPAINTERLIBEXTRA = -Llib -lGraf -lHist -lMatrix
+HTMLLIBDEP          = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+HTMLLIBEXTRA        = -Llib -lGraf -lHist -lMatrix
 MINUITLIBDEP        = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
 MINUITLIBEXTRA      = -Llib -lGraf -lHist -lMatrix
 MINUIT2LIBDEP       = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
 MINUIT2LIBEXTRA     = -Llib -lGraf -lHist -lMatrix
-FUMILILIBDEP        = $(GRAFLIB) $(HISTLIB)
-FUMILILIBEXTRA      = -Llib -lGraf -lHist
-TREEPLAYERLIBDEP    = $(TREELIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB)
-TREEPLAYERLIBEXTRA  = -Llib -lTree -lGraf3d -lGraf -lHist
-TREEVIEWERLIBDEP    = $(TREELIB) $(GPADLIB) $(GRAFLIB) $(HISTLIB) $(GUILIB)
-TREEVIEWERLIBEXTRA  = -Llib -lTree -lGpad -lGraf -lHist -lGui
+FUMILILIBDEP        = $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+FUMILILIBEXTRA      = -Llib -lGraf -lHist -lMatrix
+TREEPLAYERLIBDEP    = $(TREELIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+TREEPLAYERLIBEXTRA  = -Llib -lTree -lGraf3d -lGraf -lHist -lMatrix
+TREEVIEWERLIBDEP    = $(TREELIB) $(GPADLIB) $(GRAFLIB) $(HISTLIB) $(GUILIB) $(MATRIXLIB)
+TREEVIEWERLIBEXTRA  = -Llib -lTree -lGpad -lGraf -lHist -lGui -lMatrix
 PROOFLIBDEP         = $(TREELIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB) \
-                      $(TREEPLAYERLIB) $(THREADLIB)
-PROOFLIBEXTRA       = -Llib -lTree -lGraf3d -lGraf -lHist -lTreePlayer -lThread
-PROOFXLIBDEP        = $(PROOFLIB) $(THREADLIB) $(XPDLIB)
-PROOFXLIBEXTRA      = -Llib -lProof -lThread -lXrdProofd
-PROOFGUILIBDEP      = $(PROOFLIB) $(GUILIB) $(GPADLIB) $(GRAFLIB) $(TREELIB)
-PROOFGUILIBEXTRA    = -Llib -lProof -lGui -lGpad -lGraf -lTree
+                      $(TREEPLAYERLIB) $(THREADLIB) $(MATRIXLIB)
+PROOFLIBEXTRA       = -Llib -lTree -lGraf3d -lGraf -lHist -lTreePlayer -lThread \
+                      -lMatrix
+PROOFXLIBDEP        = $(PROOFLIB) $(THREADLIB) $(XPDLIB) $(TREELIB) $(G3DLIB) \
+                      $(GRAFLIB) $(HISTLIB) $(TREEPLAYERLIB) $(MATRIXLIB)
+PROOFXLIBEXTRA      = -Llib -lProof -lThread -lXrdProofd -lTree -lGraf3d -lGraf \
+                      -lHist -lTreePlayer -lMatrix
+PROOFGUILIBDEP      = $(PROOFLIB) $(GUILIB) $(TREELIB) $(G3DLIB) $(GRAFLIB) \
+                      $(GPADLIB) $(HISTLIB) $(TREEPLAYERLIB) $(THREADLIB) $(MATRIXLIB)
+PROOFGUILIBEXTRA    = -Llib -lProof -lGui -lTree -lGraf3d -lGraf -lGpad -lHist \
+                      -lTreePlayer -lThread -lMatrix
 PEACLIBDEP          = $(PROOFLIB) $(CLARENSLIB)
 PEACLIBEXTRA        = -Llib -lProof -lClarens
 PEACGUILIBDEP       = $(GUILIB)
 PEACGUILIBEXTRA     = -Llib -lGui
-X3DLIBDEP           = $(G3DLIB) $(GUILIB)
-X3DLIBEXTRA         = -Llib -lGraf3d -lGui
-EGLIBDEP            = $(G3DLIB) $(GRAFLIB)
-EGLIBEXTRA          = -Llib -lGraf3d -lGraf
-VMCLIBDEP           = $(EGLIB) $(GEOMLIB)
-VMCLIBEXTRA         = -Llib -lEG -lGeom
+X3DLIBDEP           = $(G3DLIB) $(GUILIB) $(GRAFLIB) $(GPADLIB) $(HISTLIB) $(MATRIXLIB)
+X3DLIBEXTRA         = -Llib -lGraf3d -lGui -lGraf -lGpad -lHist -lMatrix
+EGLIBDEP            = $(G3DLIB) $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+EGLIBEXTRA          = -Llib -lGraf3d -lGraf -lHist -lMatrix
+VMCLIBDEP           = $(EGLIB) $(GEOMLIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+VMCLIBEXTRA         = -Llib -lEG -lGeom -lGraf3d -lGraf -lHist -lMatrix
 PHYSICSLIBDEP       = $(MATRIXLIB)
 PHYSICSLIBEXTRA     = -Llib -lMatrix
-PYTHIA6LIBDEP       = $(EGLIB) $(GRAFLIB) $(VMCLIB)
-PYTHIA6LIBEXTRA     = -Llib -lEG -lGraf -lVMC $(FPYTHIA6LIBDIR) $(FPYTHIA6LIB)
-X11TTFLIBDEP        = $(X11LIB) $(GRAFLIB)
-X11TTFLIBEXTRA      = -Llib -lGX11 -lGraf
-TABLELIBDEP         = $(TREELIB) $(GPADLIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB)
-TABLELIBEXTRA       = -Llib -lTree -lGpad -lGraf3d -lGraf -lHist
+PYTHIA6LIBDEP       = $(EGLIB) $(GRAFLIB) $(VMCLIB) $(G3DLIB) $(HISTLIB) $(MATRIXLIB) $(GEOMLIB)
+PYTHIA6LIBEXTRA     = -Llib -lEG -lGraf -lVMC -lGraf3d -lHist -lMatrix -lGeom \
+                      $(FPYTHIA6LIBDIR) $(FPYTHIA6LIB)
+X11TTFLIBDEP        = $(X11LIB) $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+X11TTFLIBEXTRA      = -Llib -lGX11 -lGraf -lHist -lMatrix
+TABLELIBDEP         = $(TREELIB) $(GPADLIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+TABLELIBEXTRA       = -Llib -lTree -lGpad -lGraf3d -lGraf -lHist -lMatrix
 GLLIBDEP            = $(GPADLIB) $(G3DLIB) $(GUILIB) $(GRAFLIB) $(HISTLIB) \
-                      $(GEDLIB)
-GLLIBEXTRA          = -Llib -lGpad -lGraf3d -lGui -lGraf -lHist -lGed
+                      $(GEDLIB) $(MATRIXLIB) $(TREELIB) $(TREEPLAYERLIB)
+GLLIBEXTRA          = -Llib -lGpad -lGraf3d -lGui -lGraf -lHist -lGed \
+                      -lMatrix -lTree -lTreePlayer
 HBOOKLIBDEP         = $(HISTLIB) $(MATRIXLIB) $(TREELIB) $(GRAFLIB) \
-                      $(TREEPLAYERLIB)
-HBOOKLIBEXTRA       = -Llib -lHist -lMatrix -lTree -lGraf -lTreePlayer
-GEOMPAINTERLIBDEP   = $(GEOMLIB) $(TREELIB) $(G3DLIB) $(HISTLIB)
-GEOMPAINTERLIBEXTRA = -Llib -lGeom -lTree -lGraf3d -lHist
-GEOMBUILDERLIBDEP   = $(GEOMLIB) $(G3DLIB) $(GPADLIB) $(GRAFLIB) $(GUILIB) \
-		      $(GEDLIB)
-GEOMBUILDERLIBEXTRA = -Llib -lGeom -lGraf3d -lGpad -lGraf -lGui -lGed
-ASIMAGELIBDEP       = $(GRAFLIB) $(HISTLIB) $(GPADLIB)
-ASIMAGELIBEXTRA     = -Llib -lGraf -lHist -lGpad
-ASIMAGEGUILIBDEP    = $(GRAFLIB) $(HISTLIB) $(GUILIB) $(ASIMAGELIB)
-ASIMAGEGUILIBEXTRA  = -Llib -lGraf -lHist -lGui -lASImage
-ASIMAGEGSLIBDEP     = $(GRAFLIB) $(ASIMAGELIB)
-ASIMAGEGSLIBEXTRA   = -Llib -lGraf -lASImage
-MLPLIBDEP           = $(HISTLIB) $(MATRIXLIB) $(TREELIB) $(GRAFLIB) \
+                      $(G3DLIB) $(TREEPLAYERLIB)
+HBOOKLIBEXTRA       = -Llib -lHist -lMatrix -lTree -lGraf -lGraf3d -lTreePlayer
+GEOMPAINTERLIBDEP   = $(GEOMLIB) $(TREELIB) $(G3DLIB) $(GRAFLIB) $(HISTLIB) $(MATRIXLIB)
+GEOMPAINTERLIBEXTRA = -Llib -lGeom -lTree -lGraf3d -lGraf -lHist -lMatrix
+GEOMBUILDERLIBDEP   = $(GEOMLIB) $(TREELIB) $(G3DLIB) $(GPADLIB) $(GRAFLIB) \
+		      $(GUILIB) $(GEDLIB) $(HISTLIB) $(MATRIXLIB) $(TREEPLAYERLIB)
+GEOMBUILDERLIBEXTRA = -Llib -lGeom -lTree -lGraf3d -lGpad -lGraf -lGui -lGed \
+                      -lHist -lMatrix -lTreePlayer
+ASIMAGELIBDEP       = $(GRAFLIB) $(HISTLIB) $(GPADLIB) $(MATRIXLIB)
+ASIMAGELIBEXTRA     = -Llib -lGraf -lHist -lGpad -lMatrix
+ASIMAGEGUILIBDEP    = $(GRAFLIB) $(HISTLIB) $(GUILIB) $(ASIMAGELIB) $(GPADLIB) $(MATRIXLIB)
+ASIMAGEGUILIBEXTRA  = -Llib -lGraf -lHist -lGui -lASImage -lGpad -lMatrix
+ASIMAGEGSLIBDEP     = $(GRAFLIB) $(ASIMAGELIB) $(HISTLIB) $(GPADLIB) $(MATRIXLIB)
+ASIMAGEGSLIBEXTRA   = -Llib -lGraf -lASImage -lHist -lGpad -lMatrix
+MLPLIBDEP           = $(HISTLIB) $(MATRIXLIB) $(TREELIB) $(G3DLIB) $(GRAFLIB) \
                       $(GPADLIB) $(TREEPLAYERLIB)
-MLPLIBEXTRA         = -Llib -lHist -lMatrix -lTree -lGraf -lGpad \
+MLPLIBEXTRA         = -Llib -lHist -lMatrix -lTree -lGraf3d -lGraf -lGpad \
                       -lTreePlayer
 SPECTRUMLIBDEP      = $(HISTLIB) $(MATRIXLIB)
 SPECTRUMLIBEXTRA    = -Llib -lHist -lMatrix 
 TMVALIBDEP          = $(HISTLIB) $(MATRIXLIB) $(TREELIB) $(GRAFLIB) \
-                      $(GPADLIB) $(TREEPLAYERLIB) $(MLPLIB)
-TMVALIBEXTRA        = -Llib -lHist -lMatrix -lTree -lGraf -lGpad -lTreePlayer \
+                      $(G3DLIB) $(GPADLIB) $(TREEPLAYERLIB) $(MLPLIB)
+TMVALIBEXTRA        = -Llib -lHist -lMatrix -lTree -lGraf3d -lGraf -lGpad -lTreePlayer \
                       -lMLP
 SPLOTLIBDEP         = $(MATRIXLIB) $(HISTLIB) $(TREELIB) $(TREEPLAYERLIB) \
                       $(G3DLIB) $(GRAFLIB)
@@ -228,12 +252,12 @@
 GQTLIBEXTRA         = -Llib -lGpad -lRint
 QUADPLIBDEP         = $(MATRIXLIB)
 QUADPLIBEXTRA       = -Llib -lMatrix
-RUBYLIBDEP          = $(HISTLIB)
-RUBYLIBEXTRA        = -Llib -lHist
-GUIBLDLIBDEP        = $(GUILIB) $(GRAFLIB)
-GUIBLDLIBEXTRA      = -Llib -lGui -lGraf
-FOAMLIBDEP          = $(HISTLIB)
-FOAMLIBEXTRA        = -Llib -lHist
+RUBYLIBDEP          = $(HISTLIB) $(MATRIXLIB)
+RUBYLIBEXTRA        = -Llib -lHist -lMatrix
+GUIBLDLIBDEP        = $(GUILIB) $(GRAFLIB) $(HISTLIB) $(GPADLIB) $(MATRIXLIB)
+GUIBLDLIBEXTRA      = -Llib -lGui -lGraf -lHist -lGpad -lMatrix
+FOAMLIBDEP          = $(HISTLIB) $(MATRIXLIB)
+FOAMLIBEXTRA        = -Llib -lHist -lMatrix
 ALIENLIBDEP         = $(XMLLIB) $(NETXLIB)
 ALIENLIBEXTRA       = -Llib -lXMLIO -lNetx
 ROOFITLIBDEP        = $(HISTLIB) $(GRAFLIB) $(MATRIXLIB) $(TREELIB) \
@@ -247,8 +271,9 @@
 KRB5AUTHLIBEXTRA    = -Llib -lRootAuth
 QTGSILIBDEP         = $(GUILIB) $(GPADLIB)
 QTGSILIBEXTRA       = -Llib -lGui -lGpad
-FITPANELLIBDEP      = $(GUILIB) $(GPADLIB) $(HISTLIB) $(GRAFLIB) $(TREELIB)
-FITPANELLIBEXTRA    = -Llib -lGui -lGpad -lHist -lGraf -lTree 
+FITPANELLIBDEP      = $(GUILIB) $(GPADLIB) $(HISTLIB) $(GRAFLIB) $(TREELIB) \
+                      $(MATRIXLIB)
+FITPANELLIBEXTRA    = -Llib -lGui -lGpad -lHist -lGraf -lTree -lMatrix
 HADDLIBDEP          = $(TREEPLAYERLIB)
 HADDLIBEXTRA        = -lTreePlayer
 
--- root.old/config/Makefile.macosx	2007-03-05 09:35:31.000000000 -0600
+++ root.new/config/Makefile.macosx	2007-03-30 15:34:01.000000000 -0500
@@ -103,12 +103,9 @@
 F77LIBS       =
 endif
 
-# We add libg2c only in case of ppc because then we probably have cernlib
-# compiled with g77. In case of Intel Mac it should be compiled with the
-# same fortran we use.
-ifeq ($(MACOSX_CPU),powerpc)
-F77LIBS += -L$(FINK_DIR)/lib -lg2c
-endif
+# We add the g77 library libg2c.a only if it exists. In this case it
+# is likely that the CERNLIB was compiled with g77.
+F77LIBS += $(shell [ -r $(FINK_DIR)/lib/libg2c.a ] && echo -L$(FINK_DIR)/lib -lg2c)
 
 ifeq ($(shell find $(FINK_DIR)/bin -name $(F77)),)
 BUILDHBOOK   := no
