diff -Nurd -x'*~' ScientificPython-2.7.8.orig/Src/Scientific_netcdf.c ScientificPython-2.7.8/Src/Scientific_netcdf.c
--- ScientificPython-2.7.8.orig/Src/Scientific_netcdf.c	2007-10-31 12:53:18.000000000 -0400
+++ ScientificPython-2.7.8/Src/Scientific_netcdf.c	2008-06-20 17:59:42.000000000 -0400
@@ -19,7 +19,7 @@
 #include "Numeric/arrayobject.h"
 #endif
 #endif
-#include "netcdf.h"
+#include "libnc-dap/netcdf.h"
 
 #define _NETCDF_MODULE
 #include "Scientific/netcdfmodule.h"
diff -Nurd -x'*~' ScientificPython-2.7.8.orig/setup.py ScientificPython-2.7.8/setup.py
--- ScientificPython-2.7.8.orig/setup.py	2007-11-29 13:07:37.000000000 -0500
+++ ScientificPython-2.7.8/setup.py	2008-06-23 18:07:53.000000000 -0400
@@ -91,14 +91,14 @@
     else:
         print "Using netCDF installation in ", netcdf_prefix
         netcdf_include = os.path.join(netcdf_prefix, 'include')
-        netcdf_h_file = os.path.join(netcdf_prefix, 'include', 'netcdf.h')
+        netcdf_h_file = os.path.join(netcdf_prefix, 'include', 'libnc-dap/netcdf.h')
         netcdf_lib = os.path.join(netcdf_prefix, 'lib')
     ext_modules = [Extension('Scientific_netcdf',
                              ['Src/Scientific_netcdf.c'],
                              include_dirs=['Include', netcdf_include]
                                           + arrayobject_h_include,
                              library_dirs=[netcdf_lib],
-                             libraries = ['netcdf'],
+                             libraries = ['nc-dap','dap'],
                              extra_compile_args=extra_compile_args)]
 
 try:
