--- ScientificPython/setup.py.orig	2007-09-22 07:31:15.000000000 -0600
+++ ScientificPython/setup.py	2007-09-22 07:32:39.000000000 -0600
@@ -61,7 +61,7 @@
                              include_dirs=['Include', netcdf_include]
                                           + arrayobject_h_include,
                              library_dirs=[netcdf_lib],
-                             libraries = ['netcdf'],
+                             libraries = ['nc-dap','dap'],
                              extra_compile_args=extra_compile_args)]
 
 cmdclass = {}
--- ScientificPython/Src/Scientific_netcdf.c.orig	2007-09-22 07:33:10.000000000 -0600
+++ ScientificPython/Src/Scientific_netcdf.c	2007-09-22 07:33:23.000000000 -0600
@@ -19,7 +19,7 @@
 #include "Numeric/arrayobject.h"
 #endif
 #endif
-#include "netcdf.h"
+#include "libnc-dap/netcdf.h"
 
 #define _NETCDF_MODULE
 #include "Scientific/netcdfmodule.h"
