--- scipy/Lib/sandbox/setup.py.orig	2006-09-08 10:34:46.000000000 -0600
+++ scipy/Lib/sandbox/setup.py	2006-09-08 10:36:48.000000000 -0600
@@ -32,43 +32,43 @@
     # check it in and screw other people up :-)
 
     # An example package:
-    #config.add_subpackage('exmplpackage')
+    #config.add_subpackage('exmplpackage')
 
     # Monte Carlo package
-    #config.add_subpackage('montecarlo')
+    #config.add_subpackage('montecarlo')
 
     # PySparse fork with NumPy compatibility
-    #config.add_subpackage('pysparse')
+    #config.add_subpackage('pysparse')
 
     # Robert Kern's corner:
-    #config.add_subpackage('rkern')
+    #config.add_subpackage('rkern')
 
     # ODRPACK
-    #config.add_subpackage('odr')
+    #config.add_subpackage('odr')
 
     # Delaunay triangulation and Natural Neighbor interpolation
-    #config.add_subpackage('delaunay')
+    config.add_subpackage('delaunay')
 
     # Gist-based plotting library for X11
-    #config.add_subpackage('xplt')
+    config.add_subpackage('xplt')
 
     # elementwise numerical expressions
-    #config.add_subpackage('numexpr')
+    config.add_subpackage('numexpr')
 
     # Statistical models
-    #config.add_subpackage('models')
+    config.add_subpackage('models')
 
     # Adaptation of Scientific.IO (2.4.9) to use NumPy
-    #config.add_subpackage('netcdf')
+    config.add_subpackage('netcdf')
 
     # Finite Difference Formulae package
-    #config.add_subpackage('fdfpack')
+    config.add_subpackage('fdfpack')
 
     # Package with useful constants and unit-conversions defined
-    #config.add_subpackage('constants')
+    config.add_subpackage('constants')
 
     # Interpolating between sparse samples
-    #config.add_subpackage('buildgrid')
+    config.add_subpackage('buildgrid')
     return config
 
 if __name__ == '__main__':
