--- scipy/Lib/sandbox/setup.py.orig	2006-09-08 10:34:46.000000000 -0600
+++ scipy/Lib/sandbox/setup.py	2006-09-08 10:36:48.000000000 -0600
@@ -47,40 +47,40 @@
     #config.add_subpackage('odr')
 
     # Delaunay triangulation and Natural Neighbor interpolation
-    #config.add_subpackage('delaunay')
+    config.add_subpackage('delaunay')
 
     # Gist-based plotting library for X11
-    #config.add_subpackage('xplt')
+    config.add_subpackage('xplt')
 
     # elementwise numerical expressions
-    #config.add_subpackage('numexpr')
+    config.add_subpackage('numexpr')
 
     # Statistical models
-    #config.add_subpackage('models')
+    #config.add_subpackage('models')
 
     # Adaptation of Scientific.IO (2.4.9) to use NumPy
-    #config.add_subpackage('netcdf')
+    config.add_subpackage('netcdf')
 
     # Finite Difference Formulae package
-    #config.add_subpackage('fdfpack')
+    config.add_subpackage('fdfpack')
 
     # Package with useful constants and unit-conversions defined
-    #config.add_subpackage('constants')
+    config.add_subpackage('constants')
 
     # Interpolating between sparse samples
-    #config.add_subpackage('buildgrid')
+    #config.add_subpackage('buildgrid')
 
     # Package for Support Vector Machine 
-    #config.add_subpackage('svm')
+    config.add_subpackage('svm')
 
     # Package for Gaussian Mixture Models
-    #config.add_subpackage('pyem')
+    config.add_subpackage('pyem')
 
     # David Cournapeau's corner: autocorrelation, lpc, lpc residual
-    #config.add_subpackage('cdavid')
+    #config.add_subpackage('cdavid')
 
     # New spline package (based on scipy.interpolate)
-    #config.add_subpackage('spline')
+    config.add_subpackage('spline')
 
     return config
 
--- scipy/Lib/linalg/setup_atlas_version.py.orig	2007-06-08 05:19:51.000000000 -0600
+++ scipy/Lib/linalg/setup_atlas_version.py	2007-06-08 05:23:30.000000000 -0600
@@ -4,7 +4,7 @@
 
 import os
 from distutils.core import Extension
-from numpy.distutils.misc_util import get_path, default_config_dict
+from numpy.distutils.misc_util import get_path_from_frame, default_config_dict
 from numpy.distutils.system_info import get_info,AtlasNotFoundError
 
 def configuration (parent_package=''):
--- scipy/Lib/odr/setup.py.orig	2007-06-08 05:28:24.000000000 -0600
+++ scipy/Lib/odr/setup.py	2007-06-08 05:23:23.000000000 -0600
@@ -6,7 +6,7 @@
 import warnings
 
 from numpy.distutils.core import Extension
-from numpy.distutils.misc_util import get_path, Configuration, dot_join
+from numpy.distutils.misc_util import Configuration, dot_join
 
 from numpy.distutils.system_info import get_info,dict_append,\
      AtlasNotFoundError,LapackNotFoundError,BlasNotFoundError,\
