--- numpy/numpy/distutils/fcompiler/gnu.py.orig	2006-01-23 06:25:07.000000000 -0700
+++ numpy/numpy/distutils/fcompiler/gnu.py	2006-01-23 06:21:38.000000000 -0700
@@ -29,7 +29,8 @@
         'compiler_f77' : [fc_exe,"-Wall","-fno-second-underscore"],
         'compiler_f90' : None,
         'compiler_fix' : None,
-        'linker_so'    : [fc_exe,"-Wall"],
+#        'linker_so'    : [fc_exe,"-Wall"],
+        'linker_so'    : [fc_exe+" -undefined dynamic_lookup -bundle"],
         'archiver'     : ["ar", "-cr"],
         'ranlib'       : ["ranlib"],
         'linker_exe'   : [fc_exe,"-Wall"]
@@ -64,7 +65,7 @@
             os.environ['MACOSX_DEPLOYMENT_TARGET'] = '%s.%s' % (major,
                 minor)
 
-            opt.extend(['-undefined', 'dynamic_lookup', '-bundle'])
+            #opt.extend(['-undefined', 'dynamic_lookup', '-bundle'])
         else:
             opt.append("-shared")
         if sys.platform[:5]=='sunos':
@@ -108,7 +109,11 @@
         if g2c is not None:
             opt.append(g2c)
         if sys.platform == 'darwin':
-            opt.append('cc_dynamic')
+            #opt.append('cc_dynamic')
+            import platform
+            opt = []
+            if platform.mac_ver()[0] < 10.4:
+                opt.append('cc_dynamic')
         return opt
 
     def get_flags_debug(self):
@@ -211,7 +216,8 @@
 class Gnu95FCompiler(GnuFCompiler):
 
     compiler_type = 'gnu95'
-    version_pattern = r'GNU Fortran 95 \(GCC (?P<version>[^\s*\)]+)'
+    version_pattern = r'GNU Fortran 95 ((\(GCC[^\)]*(\)\)|\)))|)\s*'\
+                      '(?P<version>[^\s*\)]+)'
 
     # 'gfortran --version' results:
     # Debian: GNU Fortran 95 (GCC 4.0.3 20051023 (prerelease) (Debian 4.0.2-3))
@@ -224,7 +230,8 @@
         'compiler_f77' : [fc_exe,"-Wall","-ffixed-form","-fno-second-underscore"],
         'compiler_f90' : [fc_exe,"-Wall","-fno-second-underscore"],
         'compiler_fix' : [fc_exe,"-Wall","-ffixed-form","-fno-second-underscore"],
-        'linker_so'    : [fc_exe,"-Wall"],
+        #'linker_so'    : [fc_exe,"-Wall"],
+        'linker_so'    : [fc_exe+" -undefined dynamic_lookup -bundle"],
         'archiver'     : ["ar", "-cr"],
         'ranlib'       : ["ranlib"],
         'linker_exe'   : [fc_exe,"-Wall"]
--- numpy/numpy/distutils/fcompiler/ibm.py.orig	2006-01-23 06:27:47.000000000 -0700
+++ numpy/numpy/distutils/fcompiler/ibm.py	2006-01-23 06:22:55.000000000 -0700
@@ -4,18 +4,30 @@
 
 from numpy.distutils.fcompiler import FCompiler
 from distutils import log
+from numpy.distutils.exec_command import exec_command, find_executable
 
 class IbmFCompiler(FCompiler):
 
     compiler_type = 'ibm'
     version_pattern =  r'xlf\(1\)\s*IBM XL Fortran (Advanced Edition |)Version (?P<version>[^\s*]*)'
 
+    for fc_exe in map(find_executable,['xlf']):
+        if os.path.isfile(fc_exe):
+              break
+    for f90_exe in map(find_executable,['xlf90']):
+        if os.path.isfile(f90_exe):
+              break
+    for f95_exe in map(find_executable,['xlf95']):
+        if os.path.isfile(f95_exe):
+              break
+
     executables = {
-        'version_cmd'  : ["xlf"],
-        'compiler_f77' : ["xlf"],
-        'compiler_fix' : ["xlf90", "-qfixed"],
-        'compiler_f90' : ["xlf90"],
-        'linker_so'    : ["xlf95"],
+        'version_cmd'  : [fc_exe],
+        'compiler_f77' : [fc_exe],
+        'compiler_fix' : [f90_exe+" -qfixed"],
+        'compiler_f90' : [f90_exe],
+        #'linker_so'    : ["xlf95"],
+        'linker_so'    : [f95_exe+" -undefined dynamic_lookup -bundle"],
         'archiver'     : ["ar", "-cr"],
         'ranlib'       : ["ranlib"]
         }
@@ -31,9 +43,9 @@
             l.sort()
             l.reverse()
             l = [d for d in l if os.path.isfile(os.path.join(xlf_dir,d,'xlf.cfg'))]
-            if not l:
-                from distutils.version import LooseVersion
-                self.version = version = LooseVersion(l[0])
+            #if not l:
+            from distutils.version import LooseVersion
+            self.version = version = LooseVersion(l[0])
         return version
 
     def get_flags(self):
@@ -45,7 +57,8 @@
     def get_flags_linker_so(self):
         opt = []
         if sys.platform=='darwin':
-            opt.append('-Wl,-bundle,-flat_namespace,-undefined,suppress')
+            #opt.append('-Wl,-bundle,-flat_namespace,-undefined,suppress')
+            pass
         else:
             opt.append('-bshared')
         version = self.get_version(ok_status=[0,40])
 
--- numpy/numpy/distutils/fcompiler/g95.py	2006-03-29 12:07:42.000000000 -0700
+++ numpy/numpy/distutils/fcompiler/g95.py.orig	2006-03-29 12:04:01.000000000 -0700
@@ -9,14 +9,16 @@
 class G95FCompiler(FCompiler):
 
     compiler_type = 'g95'
-    version_pattern = r'G95.*\(experimental\) \(g95!\) (?P<version>.*)\).*'
+    version_pattern = r'G95 ((\(GCC[^\)]*(\)\)|\)))|)\s*'\
+                      '(?P<version>[^\s*\)]+)'
 
     executables = {
         'version_cmd'  : ["g95", "--version"],
         'compiler_f77' : ["g95", "-ffixed-form"],
         'compiler_fix' : ["g95", "-ffixed-form"],
         'compiler_f90' : ["g95"],
-        'linker_so'    : ["g95","-shared"],
+        #'linker_so'    : ["g95","-shared"],
+        'linker_so'    : ["g95 -undefined dynamic_lookup -bundle"],
         'archiver'     : ["ar", "-cr"],
         'ranlib'       : ["ranlib"]
         }
@@ -27,7 +29,7 @@
     def get_flags(self):
         return ['-fno-second-underscore']
     def get_flags_opt(self):
-        return ['-O']
+        return ['-O3 -funroll-loops']
     def get_flags_debug(self):
         return ['-g']
 
