diff -ru SNNSv4.2/xgui/sources/ui_config.c SNNSv4.2-patched/xgui/sources/ui_config.c
--- SNNSv4.2/xgui/sources/ui_config.c	1998-09-03 15:39:32.000000000 +0200
+++ SNNSv4.2-patched/xgui/sources/ui_config.c	2004-02-28 21:10:11.000000000 +0100
@@ -492,7 +492,7 @@
 	err = fscanf(filePtr,"%*[^:]: %d\n", &noParams);
 	chkretline(1);
 	lines++;
-	for (i=0; (i<UI_NO_LEARN_PARAMS+UI_NO_OF_CASCADE_PARAMS); i++){
+	for (i=0; (i<UI_NO_LEARN_PARAMS+UI_NO_OF_CASCADE_PARAMS) && (i<noParams); i++){
 	    err = fscanf(filePtr,"%*[^:]: %f\n", &ui_learnParameters[i]);
 	    chkretline(1);
 	    lines++;
diff -ru -x *~ SNNSv4.2/kernel/sources/kr_mem.c SNNSv4.2-patched/kernel/sources/kr_mem.c
--- SNNSv4.2/kernel/sources/kr_mem.c	Thu Sep  3 15:38:16 1998
+++ SNNSv4.2-patched/kernel/sources/kr_mem.c	Thu Sep 13 19:32:29 2001
@@ -27,7 +27,9 @@
 #else
 #include <math.h>
 #endif
+#include <limits.h>
 
+#define MAXSHORT 32767
 
 #include "kr_typ.h"	 /*  Kernel Types and Constants  */
 #include "kernel.h"
diff -ru -x *~ SNNSv4.2/kernel/sources/learn_f.c SNNSv4.2-patched/kernel/sources/learn_f.c
--- SNNSv4.2/kernel/sources/learn_f.c	Thu Sep  3 15:38:22 1998
+++ SNNSv4.2-patched/kernel/sources/learn_f.c	Thu Sep 13 19:41:38 2001
@@ -36,6 +36,8 @@
 #endif
 #include <ctype.h>
 
+#define MAXINT 2147483647
+
 #include "kr_typ.h"		/* Kernel Types and Constants  */
 #include "kr_const.h"		/* Constant Declarators for SNNS-Kernel  */
 #include "kr_def.h"		/* Default Values  */
