diff -ru stellarium-0.10.5.orig/CMakeLists.txt stellarium-0.10.5/CMakeLists.txt
--- stellarium-0.10.5.orig/CMakeLists.txt	2010-06-02 10:23:55.000000000 -0400
+++ stellarium-0.10.5/CMakeLists.txt	2010-07-24 17:51:23.000000000 -0400
@@ -55,12 +55,12 @@
 
 # best way to get to APPLE?
 IF(APPLE)
- SET(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -fno-common -Wall -Wextra -mmacosx-version-min=10.4")
- SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-common -Wall -Wextra -Wno-unused-parameter")
- SET(CMAKE_OSX_DEPLOYMENT_TARGET "10.4")
- SET(CMAKE_OSX_SYSROOT "/Developer/SDKs/MacOSX10.4u.sdk/")
- # Universal binary
- SET(CMAKE_OSX_ARCHITECTURES "i386;ppc")
+ IF(APPLE_APP)
+  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-common -Wall -Wextra -Wno-unused-parameter -DMACOSX_APP")
+ ELSE(APPLE_APP)
+  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-common -Wall -Wextra -Wno-unused-parameter")
+ ENDIF(APPLE_APP)
+SET(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS}")
 ENDIF()
 
 ########### Check functions ###########
@@ -217,9 +217,10 @@
 
 ########### Set some global variables ###########
 IF(UNIX AND NOT WIN32)
-IF(APPLE)
-  SET(CMAKE_INSTALL_PREFIX "${PROJECT_BINARY_DIR}/Stellarium.app/Contents")
-ELSE(APPLE)
+IF(APPLE_APP)
+  SET(INSTALL_DATADIR "${CMAKE_INSTALL_PREFIX}/Resources")
+  SET(INSTALL_LOCALEDIR "${CMAKE_INSTALL_PREFIX}/Resources/locale")
+ELSE(APPLE_APP)
   ADD_DEFINITIONS(-DINSTALL_DATADIR="${CMAKE_INSTALL_PREFIX}/share/stellarium")
   ADD_DEFINITIONS(-DINSTALL_LOCALEDIR="${CMAKE_INSTALL_PREFIX}/share/locale")
 ENDIF()
@@ -285,7 +286,7 @@
 ENDIF()
 
 ########### Macosx Bundling ###############
-IF(APPLE)
+IF(APPLE_APP)
 ADD_CUSTOM_TARGET(macosx_bundle bash util/macosx_bundle.sh ${CMAKE_INSTALL_PREFIX} ${PROJECT_SOURCE_DIR} DEPENDS install WORKING_DIRECTORY ${PROJECT_SOURCE_DIR} COMMENT "making the macosx bundle." VERBATIM)
 ENDIF()
 
diff -ru stellarium-0.10.5.orig/src/core/StelFileMgr.cpp stellarium-0.10.5/src/core/StelFileMgr.cpp
--- stellarium-0.10.5.orig/src/core/StelFileMgr.cpp	2010-06-02 10:23:47.000000000 -0400
+++ stellarium-0.10.5/src/core/StelFileMgr.cpp	2010-07-24 17:55:02.000000000 -0400
@@ -51,7 +51,7 @@
 	{
 		userDir = winApiPath + "\\Stellarium";
 	}
-#elif defined(Q_OS_MAC)
+#elif defined(MACOSX_APP)
 	userDir = QDir::homePath() + "/Library/Application Support/Stellarium";
 #else
 	userDir = QDir::homePath() + "/.stellarium";
@@ -84,13 +84,13 @@
 		qWarning() << "WARNING: could not locate installation directory";
 	}
 
-#if defined(Q_OS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_OS_WIN) || defined(MACOSX_APP)
 	screenshotDir = getDesktopDir();
 #else
 	screenshotDir = QDir::homePath();
 #endif
 
-#ifdef Q_OS_MAC
+#ifdef MACOSX_APP
 	QCoreApplication::addLibraryPath(QCoreApplication::applicationDirPath());
 #endif
 }
@@ -348,7 +348,7 @@
 		return ".";
 	}
 
-#ifdef Q_OS_MAC
+#ifdef MACOSX_APP
 	QFileInfo MacOSdir(QCoreApplication::applicationDirPath() + "/../Resources");
 	QDir ResourcesDir = MacOSdir.dir();
 	ResourcesDir.cd(QString("Resources"));
@@ -398,7 +398,7 @@
 {
 #ifdef ENABLE_NLS
 	QFileInfo localePath;
-#if defined(Q_OS_WIN) || defined(Q_OS_MAC)
+#if defined(Q_OS_WIN) || defined(MACOSX_APP)
 	// Windows and MacOS X have the locale dir in the installation folder
 	localePath = QFileInfo(getInstallationDir() + "/locale");
 	// or MacosxDirs::getApplicationResourcesDirectory().append( "/locale" );
diff -ru stellarium-0.10.5.orig/src/core/external/glues_stel/source/glues.h stellarium-0.10.5/src/core/external/glues_stel/source/glues.h
--- stellarium-0.10.5.orig/src/core/external/glues_stel/source/glues.h	2010-06-02 10:23:47.000000000 -0400
+++ stellarium-0.10.5/src/core/external/glues_stel/source/glues.h	2010-07-24 17:59:49.000000000 -0400
@@ -39,7 +39,7 @@
 
 #ifdef USE_OPENGL_ES2
 # include <GLES2/gl2.h>
-#elif defined(Q_OS_MAC) || defined(__APPLE__) || defined(__APPLE_CC__)
+#elif defined(MACOSX_APP)
 # include <OpenGL/gl.h>
 #elif defined(QT_OPENGL_ES_1) || defined(QT_OPENGL_ES_1_CL)
 # include <GLES/gl.h>
diff -ru stellarium-0.10.5.orig/src/core/external/glues_stel/source/glues_error.c stellarium-0.10.5/src/core/external/glues_stel/source/glues_error.c
--- stellarium-0.10.5.orig/src/core/external/glues_stel/source/glues_error.c	2010-06-02 10:23:47.000000000 -0400
+++ stellarium-0.10.5/src/core/external/glues_stel/source/glues_error.c	2010-07-24 18:00:14.000000000 -0400
@@ -33,7 +33,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-#if defined(Q_OS_MAC) || defined(__APPLE__) || defined(__APPLE_CC__)
+#if defined(MACOSX_APP)
 # include <OpenGL/gl.h>
 #elif defined(QT_OPENGL_ES_1) || defined(QT_OPENGL_ES_1_CL)
 # include <GLES/gl.h>
diff -ru stellarium-0.10.5.orig/util/macosx_bundle.sh stellarium-0.10.5/util/macosx_bundle.sh
--- stellarium-0.10.5.orig/util/macosx_bundle.sh	2010-06-02 10:23:57.000000000 -0400
+++ stellarium-0.10.5/util/macosx_bundle.sh	2010-07-24 18:03:28.000000000 -0400
@@ -17,22 +17,6 @@
 rmdir ${CMAKE_INSTALL_PREFIX}/Resources/stellarium
 
 
-mkdir ${CMAKE_INSTALL_PREFIX}/Frameworks
-/usr/bin/perl util/pkgApp.pl ${CMAKE_INSTALL_PREFIX} MacOS/stellarium Frameworks 
-cp -pr /Developer/Applications/Qt/plugins/{imageformats,iconengines} ${CMAKE_INSTALL_PREFIX}/MacOS
-mkdir ${CMAKE_INSTALL_PREFIX}/MacOS/sqldrivers/
-cp -pr /Developer/Applications/Qt/plugins/sqldrivers/libqsqlite.dylib  ${CMAKE_INSTALL_PREFIX}/MacOS/sqldrivers/
-for f in ${CMAKE_INSTALL_PREFIX}/MacOS/{imageformats,iconengines,sqldrivers}/*.dylib; do
-    fdir=`dirname $f`
-    dir=`basename $fdir`
-    base=`basename $f`
-    #/usr/bin/install_name_tool -id "@executable_path/$dir/$base" $f
-    for qt in `otool -L $f | egrep '	Qt' | cut -f 1 -d ' '`; do
-	newt="@executable_path/../Frameworks/$qt"
-	/usr/bin/install_name_tool -change $qt $newt $f
-    done
-done
-
 cp -pr $PROJECT_SOURCE_DIR/data/Icon.icns $CMAKE_INSTALL_PREFIX/Resources
 cp -pr $PROJECT_SOURCE_DIR/data/{PkgInfo,Info.plist} $CMAKE_INSTALL_PREFIX
-cp -pr $PROJECT_SOURCE_DIR/util/qt.conf $CMAKE_INSTALL_PREFIX/Resources
\ No newline at end of file
+cp -pr $PROJECT_SOURCE_DIR/util/qt.conf $CMAKE_INSTALL_PREFIX/Resources
