--- whatcheck/Makefile_whatcheck	Tue Jun  3 06:56:20 2003
+++ whatcheck/Makefile_whatcheck.new	Sun Aug 29 19:26:07 2004
@@ -2,12 +2,12 @@
 # Automatically created by DO_INSTALL.SH
 
 DEFINES=-DUNIX -DCTEXT -DF2C -DGLX11 -DX11 -DDX4 -DLITE -DWHATCHECK
-OPTFLAGS=-O0 -g -Wall
+OPTFLAGS=-O3
 LIBS= -L/usr/X11R6/lib -lX11 -lm
 ANSICC=cc
 KRCC=cc
-INCLUDES= -I/usr/X11R6/include
-F77=g77 -xf77-cpp-input -Wsurprising
+INCLUDES= -I/usr/X11R6/include 
+F77=g77 -xf77-cpp-input -Wsurprising -I.
 LD=g77
 
 # End-Of-Configuration: Do not delete this line!!!
@@ -55,17 +55,17 @@
 	@echo please ignore error message and re-execute make
 	exit 3
 
-SRCFILES = $(addprefix ../whatif/src/, $(OBJECTS:%.o=%.f))
+SRCFILES = $(addprefix ., $(OBJECTS:%.o=%.f))
 DSTFILES = $(notdir $(SRCFILES:%.f=%.F))
 
-%.F: ../whatif/src/%.f
-	gcc -x c -E $(DEFINES) $< > $@
+%.F: %.f
+	gcc -O3 -x c -E $(DEFINES) $< > $@
 
 obfuscated.f: $(DSTFILES)
 	cat $(DSTFILES) > obfuscated.f
 
-support.o: ../whatif/src/support.c
-	gcc $(DEFINES) -c -o $@ $<
+support.o: support.c
+	gcc -O3 -I /usr/include $(DEFINES) -c $<  -o $@ 
 
 test:
 	echo $(ALLF)
--- whatcheck/WHATIF.FIG	Tue Jun  3 06:59:25 2003
+++ whatcheck/WHATIF.FIG.new	Sun Aug 29 19:29:56 2004
@@ -1,4 +1,4 @@
-/home/vriend/whatcheck/dbdata/
+@FINKPREFIX@/share/whatcheck/dbdata/
 YES DATABASE
-/home/vriend/whatcheck/qualty/
+@FINKPREFIX@/share/whatcheck/qualty/
 IRIF  FFTFFFTFFFFFFFTFFFFF
--- whatcheck/scatterdir/support.c.org	Sun Aug 29 20:00:07 2004
+++ whatcheck/scatterdir/support.c	Sun Aug 29 20:07:41 2004
@@ -15,7 +15,7 @@
 #endif
 #if !defined(ESV)
 #include <unistd.h>
-#include <malloc.h>
+#include <sys/malloc.h>
 #else
 #include "/usr/include/unistd.h"
 #endif
--- whatcheck/support.c.org	Mon Aug 30 21:43:21 2004
+++ whatcheck/support.c	Mon Aug 30 21:43:38 2004
@@ -50,7 +50,7 @@
   #include "/usr/include/unistd.h"
 #else
   #include <unistd.h>
-  #include <malloc.h>
+  #include <sys/malloc.h>
 #endif
 #include <stdio.h>
 #include <stdlib.h>
diff -uNr whatcheck/whatcheck.csh whatcheck.new/whatcheck.csh
--- whatcheck/whatcheck.csh	Wed Dec 31 19:00:00 1969
+++ whatcheck.new/whatcheck.csh	Tue Aug 31 20:28:29 2004
@@ -0,0 +1 @@
+alias whatcheck '@FINKPREFIX@/share/whatcheck/whatcheck'
diff -uNr whatcheck/whatcheck.sh whatcheck.new/whatcheck.sh
--- whatcheck/whatcheck.sh	Wed Dec 31 19:00:00 1969
+++ whatcheck.new/whatcheck.sh	Tue Aug 31 20:28:43 2004
@@ -0,0 +1 @@
+alias whatcheck='@FINKPREFIX@/share/whatcheck/whatcheck'
