diff -Naur xcircuit-3.7.26/Makefile.in xcircuit-3.7.26.patched//Makefile.in
--- xcircuit-3.7.26/Makefile.in	2009-11-06 22:26:53.000000000 -0500
+++ xcircuit-3.7.26.patched//Makefile.in	2011-04-14 08:27:42.000000000 -0400
@@ -638,7 +638,7 @@
 
 lib/tcl/xcircuit${SHDLIB_EXT}: xcwrap.o ${xcircuit_OBJECTS} ${xcircuit_DEPEND}
 	$(RM) lib/xcircuit${SHDLIB_EXT}
-	$(CC) ${CFLAGS} ${SHLIB_CFLAGS} -o $@ ${LDDL_FLAGS} xcwrap.o \
+	$(CC) ${CFLAGS} ${SHLIB_CFLAGS} -o $@ -install_name $(libdir)/xcircuit-@MAJOR_VERSION@/xcircuit${SHDLIB_EXT} ${LDDL_FLAGS} xcwrap.o \
 		${xcircuit_OBJECTS} ${xcircuit_LDADD} ${SHLIB_LIB_SPECS} \
 		${LDFLAGS} ${X_EXTRA_LIBS} ${EXTRA_LIBS} -lc ${LIBS} \
 		${EXTRA_LIB_SPECS}
