diff -Naur xtide-2.11/xtide_program_wrapper xtide-2.11.patched//xtide_program_wrapper
--- xtide-2.11/xtide_program_wrapper	1969-12-31 19:00:00.000000000 -0500
+++ xtide-2.11.patched//xtide_program_wrapper	2011-03-24 13:52:49.000000000 -0400
@@ -0,0 +1,5 @@
+#!/bin/bash
+. PREFIX/bin/init.sh
+export HFILE_PATH="$HFILE_PATH_US:$HFILE_PATH_NON_US:$HFILE_PATH"
+executable=${0##*/}
+exec -a "$executable" "$executable.bin" "$@"
