diff -uNr rubycocoa-0.4.3d2/pre-config.rb rubycocoa-0.4.3d2-patched/pre-config.rb
--- rubycocoa-0.4.3d2/pre-config.rb	2006-02-01 23:46:53.000000000 -0500
+++ rubycocoa-0.4.3d2-patched/pre-config.rb	2006-08-16 13:02:48.000000000 -0400
@@ -25,7 +25,7 @@
 
 # build options
 cflags = '-fno-common -g'
-ldflags = '-undefined suppress -flat_namespace'
+ldflags = ''
 sdkroot = ''
 
 if @config['build-universal'] == 'yes'
