diff -Naur alpine-2.00.orig/imap/Makefile alpine-2.00/imap/Makefile
--- alpine-2.00.orig/imap/Makefile	2009-07-25 18:19:58.000000000 -0700
+++ alpine-2.00/imap/Makefile	2009-07-28 06:41:18.000000000 -0700
@@ -265,6 +265,7 @@
 EXTRASPECIALS=
 SPECIALS=
 
+MAC_OSX_KLUDGE=$(shell if [ -e $(SDKROOT)/usr/include/pam ]; then echo "-DMAC_OSX_KLUDGE=1"; fi)
 
 # Normal commands
 
@@ -418,8 +419,8 @@
 	$(TOUCH) ip6
 	$(BUILD) BUILDTYPE=osx IP=$(IP6) EXTRAAUTHENTICATORS="$(EXTRAAUTHENTICATORS) gss" \
 	PASSWDTYPE=pam \
-	EXTRACFLAGS="$(EXTRACFLAGS) -DMAC_OSX_KLUDGE=1" \
-	SPECIALS="SSLINCLUDE=/usr/include/openssl SSLLIB=/usr/lib SSLCERTS=/System/Library/OpenSSL/certs SSLKEYS=/System/Library/OpenSSL/private GSSINCLUDE=/usr/include GSSLIB=/usr/lib PAMDLFLAGS=-lpam"
+	EXTRACFLAGS="$(EXTRACFLAGS) $(MAC_OSX_KLUDGE)" \
+	SPECIALS="SSLINCLUDE=@PREFIX@/include/openssl SSLLIB=@PREFIX@/lib SSLCERTS=/System/Library/OpenSSL/certs SSLKEYS=/System/Library/OpenSSL/private GSSINCLUDE=@PREFIX@/include GSSLIB=@PREFIX@/lib PAMDLFLAGS=-lpam"
 
 osx:	osxok an
 	$(TOUCH) ip6
