diff -Nurd -x'*~' epiphany-1.6.5.orig/data/epiphany.schemas.in epiphany-1.6.5/data/epiphany.schemas.in
--- epiphany-1.6.5.orig/data/epiphany.schemas.in	2005-07-24 07:33:18.000000000 -0400
+++ epiphany-1.6.5/data/epiphany.schemas.in	2005-11-04 02:33:50.000000000 -0500
@@ -382,6 +382,7 @@
         <owner>epiphany</owner>
         <type>int</type>
         <default>0</default>
+	<locale name="C"/>
       </schema>
       <schema>
         <key>/schemas/apps/epiphany/dialogs/print_file</key>
