diff -uNbwr ettercap-NG-0.7.3/Makefile.in ettercap-NG-0.7.3-new/Makefile.in
--- ettercap-NG-0.7.3/Makefile.in	2005-05-27 11:11:46.000000000 -0400
+++ ettercap-NG-0.7.3-new/Makefile.in	2008-02-19 08:41:22.000000000 -0500
@@ -19,7 +19,7 @@
 top_srcdir = @top_srcdir@
 VPATH = @srcdir@
 pkgdatadir = $(datadir)/@PACKAGE@
-pkglibdir = $(libdir)/@PACKAGE@
+pkglibdir = $(prefix)/share/@PACKAGE@/plugins/
 pkgincludedir = $(includedir)/@PACKAGE@
 top_builddir = .
 am__cd = CDPATH="$${ZSH_VERSION+.}$(PATH_SEPARATOR)" && cd
diff -uNbwr ettercap-NG-0.7.3/configure ettercap-NG-0.7.3-new/configure
--- ettercap-NG-0.7.3/configure	2005-05-27 11:11:45.000000000 -0400
+++ ettercap-NG-0.7.3-new/configure	2008-02-19 08:41:22.000000000 -0500
@@ -28243,7 +28243,7 @@
 
 fi
 
-   elif test "$OS" != "MACOSX" -a "$OS" != "WINDOWS"; then
+   elif test "$OS" != "DARWIN" -a "$OS" != "WINDOWS"; then
       echo "$as_me:$LINENO: checking whether $CC accepts -pthread" >&5
 echo $ECHO_N "checking whether $CC accepts -pthread... $ECHO_C" >&6
       LDFLAGS_store="$LDFLAGS"
@@ -28260,13 +28260,8 @@
 int
 main ()
 {
-
-            int main(int argc, char **argv)
-            {
                pthread_create(NULL, NULL, NULL, NULL);
                return 0;
-            }
-
   ;
   return 0;
 }
@@ -29604,15 +29599,10 @@
 int
 main ()
 {
-
-               int main(int argc, char **argv)
-               {
                   char *q;
                   char p[NS_MAXDNAME];
 
                   dn_expand(q, q, q, p, sizeof(p));
-               }
-
   ;
   return 0;
 }
diff -uNbwr ettercap-NG-0.7.3/src/interfaces/curses/widgets/wdg.h ettercap-NG-0.7.3-new/src/interfaces/curses/widgets/wdg.h
--- ettercap-NG-0.7.3/src/interfaces/curses/widgets/wdg.h	2004-09-28 09:50:37.000000000 -0400
+++ ettercap-NG-0.7.3-new/src/interfaces/curses/widgets/wdg.h	2008-02-19 08:42:09.000000000 -0500
@@ -12,6 +12,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/types.h>
 
 #ifdef OS_WINDOWS
    #include <windows.h>
