diff -uNr evolution-data-server-1.12.0/calendar/libical/src/libical/icaltz-util.c evolution-data-server-1.12.0-new/calendar/libical/src/libical/icaltz-util.c
--- evolution-data-server-1.12.0/calendar/libical/src/libical/icaltz-util.c	2007-09-16 15:11:13.000000000 -0400
+++ evolution-data-server-1.12.0-new/calendar/libical/src/libical/icaltz-util.c	2007-09-19 13:06:26.000000000 -0400
@@ -23,6 +23,8 @@
 #include <string.h>
 #if defined(sun) && defined(__SVR4)
 #include <sys/byteorder.h>
+#elif defined(__APPLE__)
+#include <machine/endian.h>
 #else
 #include <byteswap.h>
 #include <endian.h>
@@ -89,8 +91,12 @@
     if ((BYTE_ORDER == BIG_ENDIAN) && sizeof (int) == 4)
         return *(const int *) ptr;
     else if (BYTE_ORDER == LITTLE_ENDIAN && sizeof (int) == 4)
+#ifdef __APPLE__
+        return OSSwapBigToHostInt32(*(const int *) ptr);
+#else
         return bswap_32 (*(const int *) ptr);
 #endif
+#endif
 	else
 	{
 		const unsigned char *p = ptr;
@@ -105,6 +111,9 @@
 	}
 }
 
+#define ntohl(x)        OSSwapBigToHostInt32(x)
+#define htonl(x)        OSSwapHostToBigInt32(x)
+
 static char *
 zname_from_stridx (char *str, long int idx) 
 {
diff -uNr evolution-data-server-1.12.0/configure evolution-data-server-1.12.0-new/configure
--- evolution-data-server-1.12.0/configure	2007-09-17 02:26:11.000000000 -0400
+++ evolution-data-server-1.12.0-new/configure	2007-09-19 10:13:20.000000000 -0400
@@ -25693,7 +25693,7 @@
 #include <iconv.h>
 #include <stdlib.h>
 #include <string.h>
-#ifdef HAVE_GNU_GET_LIBC_VERSION
+#if 0
 #include <gnu/libc-version.h>
 #endif
 
@@ -25706,7 +25706,7 @@
 	size_t from_len = strlen (from), utf8_len = 20;
 	size_t utf8_real_len = strlen (utf8);
 
-#ifdef HAVE_GNU_GET_LIBC_VERSION
+#if 0
 	/* glibc 2.1.2's iconv is broken in hard to test ways. */
 	if (!strcmp (gnu_get_libc_version (), "2.1.2"))
 		exit (1);
diff -uNr evolution-data-server-1.12.0/libdb/dist/configure evolution-data-server-1.12.0-new/libdb/dist/configure
--- evolution-data-server-1.12.0/libdb/dist/configure	2007-09-16 15:00:15.000000000 -0400
+++ evolution-data-server-1.12.0-new/libdb/dist/configure	2007-09-19 10:13:20.000000000 -0400
@@ -18815,7 +18815,7 @@
 
 
 
-for ac_func in _fstati64 clock_gettime directio gettimeofday getuid
+for ac_func in _fstati64 clock_gettime gettimeofday getuid
 do
 as_ac_var=`echo "ac_cv_func_$ac_func" | $as_tr_sh`
 echo "$as_me:$LINENO: checking for $ac_func" >&5
@@ -18904,7 +18904,7 @@
 
 
 
-for ac_func in pstat_getdynamic sched_yield select strtoul sysconf yield
+for ac_func in sched_yield select strtoul sysconf yield
 do
 as_ac_var=`echo "ac_cv_func_$ac_func" | $as_tr_sh`
 echo "$as_me:$LINENO: checking for $ac_func" >&5
diff -uNr evolution-data-server-1.12.0/libedataserverui/test-name-selector.c evolution-data-server-1.12.0-new/libedataserverui/test-name-selector.c
--- evolution-data-server-1.12.0/libedataserverui/test-name-selector.c	2007-09-16 14:51:04.000000000 -0400
+++ evolution-data-server-1.12.0-new/libedataserverui/test-name-selector.c	2007-09-19 13:20:45.000000000 -0400
@@ -28,6 +28,7 @@
 #include "e-name-selector-model.h"
 #include "e-name-selector-dialog.h"
 #include "e-name-selector-entry.h"
+#include <sys/types.h>
 #include <gtk/gtk.h>
 #include <camel/camel.h>
 
