diff -Naur evolution-data-server-0.0.96-old/libdb/dist/configure evolution-data-server-0.0.96/libdb/dist/configure
--- evolution-data-server-0.0.96-old/libdb/dist/configure	2004-07-19 08:52:49.000000000 -0400
+++ evolution-data-server-0.0.96/libdb/dist/configure	2004-07-22 18:07:35.000000000 -0400
@@ -19856,7 +19856,7 @@
 
 
 
-for ac_func in _fstati64 clock_gettime directio gettimeofday getuid
+for ac_func in _fstati64 clock_gettime gettimeofday getuid
 do
 as_ac_var=`echo "ac_cv_func_$ac_func" | $as_tr_sh`
 echo "$as_me:$LINENO: checking for $ac_func" >&5
@@ -19963,7 +19963,7 @@
 
 
 
-for ac_func in pstat_getdynamic sched_yield select strtoul sysconf yield
+for ac_func in sched_yield select strtoul sysconf yield
 do
 as_ac_var=`echo "ac_cv_func_$ac_func" | $as_tr_sh`
 echo "$as_me:$LINENO: checking for $ac_func" >&5
