diff -uNr evolution-2.8.2.1/mail/mail-vfolder.c evolution-2.8.2.1-new/mail/mail-vfolder.c
--- evolution-2.8.2.1/mail/mail-vfolder.c	2006-06-30 05:46:26.000000000 -0400
+++ evolution-2.8.2.1-new/mail/mail-vfolder.c	2006-12-22 12:46:39.000000000 -0500
@@ -52,7 +52,7 @@
 #define d(x)  /* (printf("%s(%d):%s: ",  __FILE__, __LINE__, __PRETTY_FUNCTION__), (x))*/
 
 static EMVFolderContext *context;	/* context remains open all time */
-CamelStore *vfolder_store; /* the 1 static vfolder store */
+extern CamelStore *vfolder_store; /* the 1 static vfolder store */
 
 /* lock for accessing shared resources (below) */
 static pthread_mutex_t vfolder_lock = PTHREAD_MUTEX_INITIALIZER;
diff -uNr evolution-2.8.2.1/plugins/calendar-weather/Makefile.in evolution-2.8.2.1-new/plugins/calendar-weather/Makefile.in
--- evolution-2.8.2.1/plugins/calendar-weather/Makefile.in	2006-11-20 22:53:20.000000000 -0500
+++ evolution-2.8.2.1-new/plugins/calendar-weather/Makefile.in	2006-12-22 12:53:22.000000000 -0500
@@ -59,7 +59,7 @@
 LTLIBRARIES = $(plugin_LTLIBRARIES)
 am__DEPENDENCIES_1 =
 liborg_gnome_calendar_weather_la_DEPENDENCIES =  \
-	$(top_builddir)/calendar/gui/libevolution-calendar.la \
+	$(top_builddir)/calendar/common/libevolution-calendarprivate.la \
 	$(am__DEPENDENCIES_1)
 am_liborg_gnome_calendar_weather_la_OBJECTS = calendar-weather.lo
 liborg_gnome_calendar_weather_la_OBJECTS =  \
@@ -445,7 +445,7 @@
 liborg_gnome_calendar_weather_la_SOURCES = calendar-weather.c
 liborg_gnome_calendar_weather_la_LDFLAGS = -module -avoid-version $(NO_UNDEFINED)
 liborg_gnome_calendar_weather_la_LIBADD = \
-	$(top_builddir)/calendar/gui/libevolution-calendar.la	\
+	$(top_builddir)/calendar/common/libevolution-calendarprivate.la	\
 	$(EVOLUTION_CALENDAR_LIBS)
 
 EXTRA_DIST = \
diff -uNr evolution-2.8.2.1/plugins/groupwise-features/Makefile.in evolution-2.8.2.1-new/plugins/groupwise-features/Makefile.in
--- evolution-2.8.2.1/plugins/groupwise-features/Makefile.in	2006-11-20 22:53:22.000000000 -0500
+++ evolution-2.8.2.1-new/plugins/groupwise-features/Makefile.in	2006-12-22 13:12:03.000000000 -0500
@@ -62,7 +62,7 @@
 am__DEPENDENCIES_2 =
 liborg_gnome_groupwise_features_la_DEPENDENCIES =  \
 	$(top_builddir)/e-util/libeutil.la \
-	$(top_builddir)/calendar/gui/libevolution-calendar.la \
+	$(top_builddir)/calendar/common/libevolution-calendarprivate.la \
 	$(top_builddir)/widgets/misc/libemiscwidgets.la \
 	$(top_builddir)/filter/libfilter.la $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_2) $(am__DEPENDENCIES_2) \
@@ -472,7 +472,7 @@
 
 liborg_gnome_groupwise_features_la_LIBADD = \
 	$(top_builddir)/e-util/libeutil.la			\
-	$(top_builddir)/calendar/gui/libevolution-calendar.la	\
+	$(top_builddir)/calendar/common/libevolution-calendarprivate.la	\
 	$(top_builddir)/widgets/misc/libemiscwidgets.la		\
 	$(top_builddir)/filter/libfilter.la			\
 	$(NO_UNDEFINED_REQUIRED_LIBS)				\
diff -uNr evolution-2.8.2.1/plugins/import-ics-attachments/Makefile.in evolution-2.8.2.1-new/plugins/import-ics-attachments/Makefile.in
--- evolution-2.8.2.1/plugins/import-ics-attachments/Makefile.in	2006-11-20 22:53:22.000000000 -0500
+++ evolution-2.8.2.1-new/plugins/import-ics-attachments/Makefile.in	2006-12-22 13:21:38.000000000 -0500
@@ -59,7 +59,6 @@
 am__DEPENDENCIES_1 =
 liborg_gnome_evolution_mail_attachments_import_ics_la_DEPENDENCIES =  \
 	$(top_builddir)/e-util/libeutil.la \
-	$(top_builddir)/mail/libevolution-mail.la \
 	$(am__DEPENDENCIES_1)
 am_liborg_gnome_evolution_mail_attachments_import_ics_la_OBJECTS =  \
 	icsimporter.lo
@@ -442,7 +441,6 @@
 liborg_gnome_evolution_mail_attachments_import_ics_la_LDFLAGS = -module -avoid-version $(NO_UNDEFINED)
 liborg_gnome_evolution_mail_attachments_import_ics_la_LIBADD = \
 	$(top_builddir)/e-util/libeutil.la		\
-	$(top_builddir)/mail/libevolution-mail.la	\
 	$(EVOLUTION_MAIL_LIBS)
 
 EXTRA_DIST = org-gnome-evolution-mail-attachments-import-ics.eplug.xml
diff -uNr evolution-2.8.2.1/plugins/itip-formatter/Makefile.in evolution-2.8.2.1-new/plugins/itip-formatter/Makefile.in
--- evolution-2.8.2.1/plugins/itip-formatter/Makefile.in	2006-11-20 22:53:23.000000000 -0500
+++ evolution-2.8.2.1-new/plugins/itip-formatter/Makefile.in	2006-12-22 13:08:10.000000000 -0500
@@ -60,8 +60,7 @@
 am__DEPENDENCIES_1 =
 liborg_gnome_itip_formatter_la_DEPENDENCIES =  \
 	$(top_builddir)/e-util/libeutil.la \
-	$(top_builddir)/calendar/gui/libevolution-calendar.la \
-	$(top_builddir)/mail/libevolution-mail.la \
+	$(top_builddir)/calendar/common/libevolution-calendarprivate.la \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 am_liborg_gnome_itip_formatter_la_OBJECTS = itip-formatter.lo \
 	itip-view.lo
@@ -435,8 +434,7 @@
 liborg_gnome_itip_formatter_la_LDFLAGS = -module -avoid-version $(NO_UNDEFINED)
 liborg_gnome_itip_formatter_la_LIBADD = \
 	$(top_builddir)/e-util/libeutil.la			\
-	$(top_builddir)/calendar/gui/libevolution-calendar.la	\
-	$(top_builddir)/mail/libevolution-mail.la		\
+	$(top_builddir)/calendar/common/libevolution-calendarprivate.la	\
 	$(CAMEL_LIBS)						\
 	$(EVOLUTION_CALENDAR_LIBS)
 
diff -uNr evolution-2.8.2.1/plugins/mail-account-disable/Makefile.in evolution-2.8.2.1-new/plugins/mail-account-disable/Makefile.in
--- evolution-2.8.2.1/plugins/mail-account-disable/Makefile.in	2006-11-20 22:53:23.000000000 -0500
+++ evolution-2.8.2.1-new/plugins/mail-account-disable/Makefile.in	2006-12-22 13:13:59.000000000 -0500
@@ -59,7 +59,6 @@
 am__DEPENDENCIES_1 =
 libmail_account_disable_la_DEPENDENCIES =  \
 	$(top_builddir)/e-util/libeutil.la \
-	$(top_builddir)/mail/libevolution-mail.la \
 	$(am__DEPENDENCIES_1)
 am_libmail_account_disable_la_OBJECTS = mail-account-disable.lo
 libmail_account_disable_la_OBJECTS =  \
@@ -431,7 +430,6 @@
 libmail_account_disable_la_SOURCES = mail-account-disable.c
 libmail_account_disable_la_LDFLAGS = -module -avoid-version $(NO_UNDEFINED)
 libmail_account_disable_la_LIBADD = $(top_builddir)/e-util/libeutil.la		\
-	$(top_builddir)/mail/libevolution-mail.la	\
 	$(EVOLUTION_MAIL_LIBS)
 
 EXTRA_DIST = org-gnome-mail-account-disable.eplug.xml
diff -uNr evolution-2.8.2.1/plugins/mark-all-read/Makefile.in evolution-2.8.2.1-new/plugins/mark-all-read/Makefile.in
--- evolution-2.8.2.1/plugins/mark-all-read/Makefile.in	2006-11-20 22:53:25.000000000 -0500
+++ evolution-2.8.2.1-new/plugins/mark-all-read/Makefile.in	2006-12-22 13:09:54.000000000 -0500
@@ -59,7 +59,6 @@
 am__DEPENDENCIES_1 =
 liborg_gnome_mark_all_read_la_DEPENDENCIES =  \
 	$(top_builddir)/e-util/libeutil.la \
-	$(top_builddir)/mail/libevolution-mail.la \
 	$(am__DEPENDENCIES_1)
 am_liborg_gnome_mark_all_read_la_OBJECTS = mark-all-read.lo
 liborg_gnome_mark_all_read_la_OBJECTS =  \
@@ -431,7 +430,6 @@
 liborg_gnome_mark_all_read_la_LDFLAGS = -module -avoid-version $(NO_UNDEFINED)
 liborg_gnome_mark_all_read_la_LIBADD = \
 	$(top_builddir)/e-util/libeutil.la		\
-	$(top_builddir)/mail/libevolution-mail.la	\
 	$(EVOLUTION_MAIL_LIBS)
 
 EXTRA_DIST = org-gnome-mark-all-read.eplug.xml
diff -uNr evolution-2.8.2.1/plugins/print-message/Makefile.in evolution-2.8.2.1-new/plugins/print-message/Makefile.in
--- evolution-2.8.2.1/plugins/print-message/Makefile.in	2006-11-20 22:53:27.000000000 -0500
+++ evolution-2.8.2.1-new/plugins/print-message/Makefile.in	2006-12-22 13:09:47.000000000 -0500
@@ -59,7 +59,6 @@
 am__DEPENDENCIES_1 =
 liborg_gnome_print_message_la_DEPENDENCIES =  \
 	$(top_builddir)/e-util/libeutil.la \
-	$(top_builddir)/mail/libevolution-mail.la \
 	$(am__DEPENDENCIES_1)
 am_liborg_gnome_print_message_la_OBJECTS = print-message.lo
 liborg_gnome_print_message_la_OBJECTS =  \
@@ -431,7 +430,6 @@
 liborg_gnome_print_message_la_LDFLAGS = -module -avoid-version $(NO_UNDEFINED)
 liborg_gnome_print_message_la_LIBADD = \
 	$(top_builddir)/e-util/libeutil.la		\
-	$(top_builddir)/mail/libevolution-mail.la	\
 	$(EVOLUTION_MAIL_LIBS)
 
 EXTRA_DIST = \
diff -uNr evolution-2.8.2.1/plugins/publish-calendar/Makefile.in evolution-2.8.2.1-new/plugins/publish-calendar/Makefile.in
--- evolution-2.8.2.1/plugins/publish-calendar/Makefile.in	2006-11-20 22:53:27.000000000 -0500
+++ evolution-2.8.2.1-new/plugins/publish-calendar/Makefile.in	2006-12-22 13:17:22.000000000 -0500
@@ -59,7 +59,7 @@
 am__DEPENDENCIES_1 =
 liborg_gnome_publish_calendar_la_DEPENDENCIES =  \
 	$(top_builddir)/e-util/libeutil.la \
-	$(top_builddir)/calendar/gui/libevolution-calendar.la \
+	$(top_builddir)/calendar/common/libevolution-calendarprivate.la \
 	$(am__DEPENDENCIES_1)
 am_liborg_gnome_publish_calendar_la_OBJECTS = publish-calendar.lo \
 	publish-format-fb.lo publish-format-ical.lo \
@@ -447,7 +447,7 @@
 liborg_gnome_publish_calendar_la_LDFLAGS = -module -avoid-version $(NO_UNDEFINED)
 liborg_gnome_publish_calendar_la_LIBADD = \
 	$(top_builddir)/e-util/libeutil.la			\
-	$(top_builddir)/calendar/gui/libevolution-calendar.la	\
+	$(top_builddir)/calendar/common/libevolution-calendarprivate.la	\
 	$(EVOLUTION_CALENDAR_LIBS)
 
 EXTRA_DIST = \
diff -uNr evolution-2.8.2.1/plugins/startup-wizard/Makefile.in evolution-2.8.2.1-new/plugins/startup-wizard/Makefile.in
--- evolution-2.8.2.1/plugins/startup-wizard/Makefile.in	2006-11-20 22:53:29.000000000 -0500
+++ evolution-2.8.2.1-new/plugins/startup-wizard/Makefile.in	2006-12-22 13:08:00.000000000 -0500
@@ -61,8 +61,7 @@
 	$(top_builddir)/e-util/libeutil.la \
 	$(top_builddir)/widgets/e-timezone-dialog/libetimezonedialog.la \
 	$(top_builddir)/widgets/misc/libemiscwidgets.la \
-	$(top_builddir)/calendar/gui/libevolution-calendar.la \
-	$(top_builddir)/mail/libevolution-mail.la \
+	$(top_builddir)/calendar/common/libevolution-calendarprivate.la \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 am_liborg_gnome_evolution_startup_wizard_la_OBJECTS =  \
 	startup-wizard.lo
@@ -437,8 +436,7 @@
 	$(top_builddir)/e-util/libeutil.la				\
 	$(top_builddir)/widgets/e-timezone-dialog/libetimezonedialog.la	\
 	$(top_builddir)/widgets/misc/libemiscwidgets.la			\
-	$(top_builddir)/calendar/gui/libevolution-calendar.la		\
-	$(top_builddir)/mail/libevolution-mail.la			\
+	$(top_builddir)/calendar/common/libevolution-calendarprivate.la		\
 	$(EVOLUTION_CALENDAR_LIBS)					\
 	$(EVOLUTION_MAIL_LIBS)
 
