diff -ruN mozilla-1.9.1-orig/.mozconfig mozilla-1.9.1/.mozconfig
--- mozilla-1.9.1-orig/.mozconfig	1969-12-31 19:00:00.000000000 -0500
+++ mozilla-1.9.1/.mozconfig	2009-04-27 16:56:30.000000000 -0400
@@ -0,0 +1,44 @@
+# sh
+# Build configuration script
+#
+# See https://developer.mozilla.org/en/Configuring_Build_Options for build options.
+#
+
+. $topsrcdir/browser/config/mozconfig
+
+# Options for client.mk.
+# environment variables to set
+
+#mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/obj-@CONFIG_GUESS@
+mk_add_options AUTOCONF=autoconf
+
+# Options for 'configure' (same as command-line options).
+ac_add_options --prefix=@FINKPREFIX@
+ac_add_options --enable-official-branding
+ac_add_options --enable-default-toolkit=cairo-gtk2
+ac_add_options --x-libraries=/usr/X11R6/lib
+ac_add_options --x-includes=/usr/X11R6/include
+#ac_add_options --enable-xft
+#ac_add_options --disable-xft
+#ac_add_options --enable-freetype2
+#ac_add_options --disable-freetype2
+#ac_add_options --enable-xinerama
+#ac_add_options --with-pthreads
+ac_add_options --disable-gnomevfs
+ac_add_options --disable-gnomeui
+ac_add_options --enable-dbus
+ac_add_options --enable-extensions=default,-gnomevfs
+
+ac_add_options --disable-dependency-tracking
+ac_add_options --disable-crashreporter
+#ac_add_options --disable-necko-wifi
+
+ac_add_options --disable-debug
+ac_add_options --disable-tests
+ac_add_options --enable-strip
+ac_add_options --enable-optimize='-O3'
+ac_add_options --enable-macos-target=10.5
+#ac_add_options --with-default-mozilla-five-home=@FINKPREFIX@/lib/firefox3
+ac_add_options --with-system-jpeg=@FINKPREFIX@
+ac_add_options --disable-system-cairo
+ac_add_options --disable-system-sqlite
diff -ruN mozilla-1.9.1-orig/browser/locales/generic/profile/bookmarks.html.in mozilla-1.9.1/browser/locales/generic/profile/bookmarks.html.in
--- mozilla-1.9.1-orig/browser/locales/generic/profile/bookmarks.html.in	2009-04-23 21:46:37.000000000 -0400
+++ mozilla-1.9.1/browser/locales/generic/profile/bookmarks.html.in	2009-04-27 16:56:30.000000000 -0400
@@ -15,6 +15,7 @@
     <DL><p>
         <DT><A HREF="http://www.mozilla.com/@AB_CD@/firefox/central/" ID="rdf:#$GvPhC3">@getting_started@</A>
         <DT><A HREF="http://fxfeeds.mozilla.com/@AB_CD@/firefox/livebookmarks/" FEEDURL="http://fxfeeds.mozilla.com/@AB_CD@/firefox/headlines.xml" ID="rdf:#$HvPhC3">@latest_headlines@</A>
+        <DT><A HREF="http://www.finkproject.org/" ICON_URI="http://www.finkproject.org/favicon.ico" ICON="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAA+0lEQVQ4jWNgGDLg/6sTqv9OVe75uzv43J+TpUv/MzAwEqfxfEfU3y3O///u9P//d5v7/7/HCir/vz8n//9ir9+fffF+eDX/O5yxCqLR4///28u0GRgYGP6uN/sPN/zNBWlMG9+e4Pt7tq0bbutO//9/t3v9///qFQ9e2/7uCZ3/b3fwvX9HMg//PZpz7e9O//9/9ye8+Lc34ubf7V7/f1+e4Irdj3dWVf//f5UNzn91VeLvzoD/f0+U1OK1EadLtrr8/3+uoYcszf+PZHX/OxB/jizNDAwMDP92BX0hW/P/o1k7ydbMwMDA8P98D/ZQJkrz+6sJFNk+oAAAPwyICxl3w8YAAAAASUVORK5CYII=">Fink - Home</A>
     </DL><p>
     <HR>
     <DT><H3 ID="rdf:#$ZvPhC3">@firefox_heading@</H3>
diff -ruN mozilla-1.9.1-orig/config/autoconf.mk.in mozilla-1.9.1/config/autoconf.mk.in
--- mozilla-1.9.1-orig/config/autoconf.mk.in	2009-04-23 21:46:37.000000000 -0400
+++ mozilla-1.9.1/config/autoconf.mk.in	2009-04-27 16:56:30.000000000 -0400
@@ -57,14 +57,14 @@
 prefix		= @prefix@
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
-includedir	= @includedir@/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+includedir	= @includedir@/$(MOZ_APP_NAME)3
 libdir		= @libdir@
 datadir		= @datadir@
 mandir		= @mandir@
-idldir		= $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+idldir		= $(datadir)/idl/$(MOZ_APP_NAME)3
 
-installdir	= $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir		= $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+installdir	= $(libdir)/$(MOZ_APP_NAME)3
+sdkdir		= $(libdir)/$(MOZ_APP_NAME)3-devel
 
 DIST		= $(DEPTH)/dist
 LIBXUL_SDK      = @LIBXUL_SDK@
diff -ruN mozilla-1.9.1-orig/configure mozilla-1.9.1/configure
--- mozilla-1.9.1-orig/configure	2009-04-23 21:51:07.000000000 -0400
+++ mozilla-1.9.1/configure	2009-04-27 16:56:30.000000000 -0400
@@ -5941,11 +5941,11 @@
     ;;
 
 *-darwin*)
-    HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX -DXP_MACOSX -DNO_X11"
+    HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX"
     HOST_NSPR_MDCPUCFG='\"md/_darwin.cfg\"'
     HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-O3}"
     MOZ_FIX_LINK_PATHS='-Wl,-executable_path,$(LIBXUL_DIST)/bin'
-    LIBXUL_LIBS='$(XPCOM_FROZEN_LDOPTS) $(LIBXUL_DIST)/bin/XUL -lobjc'
+    LIBXUL_LIBS='$(XPCOM_FROZEN_LDOPTS) $(LIBXUL_DIST)/bin/libxul.dylib -lobjc'
     ;;
 
 *-linux*|*-kfreebsd*-gnu)
@@ -6230,10 +6230,10 @@
     MKCSHLIB='$(CC) $(CFLAGS) $(DSO_PIC_CFLAGS) $(DSO_LDOPTS) -o $@'
     MOZ_OPTIMIZE_FLAGS="-O2"
     _PEDANTIC=
-    CFLAGS="$CFLAGS -fpascal-strings -fno-common"
-    CXXFLAGS="$CXXFLAGS -fpascal-strings -fno-common"
+    CFLAGS="$CFLAGS -fno-common"
+    CXXFLAGS="$CXXFLAGS -fno-common"
     DLL_SUFFIX=".dylib"
-    DSO_LDOPTS=''
+    DSO_LDOPTS='-framework CoreFoundation'
     STRIP="$STRIP -x -S"
     _PLATFORM_DEFAULT_TOOLKIT='cairo-cocoa'
     MOZ_ENABLE_POSTSCRIPT=
diff -ruN mozilla-1.9.1-orig/fink/applications/firefox.desktop mozilla-1.9.1/fink/applications/firefox.desktop
--- mozilla-1.9.1-orig/fink/applications/firefox.desktop	1969-12-31 19:00:00.000000000 -0500
+++ mozilla-1.9.1/fink/applications/firefox.desktop	2009-04-27 16:56:30.000000000 -0400
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Encoding=UTF-8
+Name=Firefox Web Browser
+Comment=Browse the World Wide Web
+Exec=firefox
+Icon=firefox.png
+Terminal=false
+Type=Application
+Categories=Application;Network
+StartupNotify=true
diff -ruN mozilla-1.9.1-orig/fink/fink.js mozilla-1.9.1/fink/fink.js
--- mozilla-1.9.1-orig/fink/fink.js	1969-12-31 19:00:00.000000000 -0500
+++ mozilla-1.9.1/fink/fink.js	2009-04-27 16:56:30.000000000 -0400
@@ -0,0 +1 @@
+pref("general.useragent.extra.vendorComment", "Fink Community Edition");
diff -ruN mozilla-1.9.1-orig/modules/libjar/nsZipArchive.cpp mozilla-1.9.1/modules/libjar/nsZipArchive.cpp
--- mozilla-1.9.1-orig/modules/libjar/nsZipArchive.cpp	2009-04-23 21:46:48.000000000 -0400
+++ mozilla-1.9.1/modules/libjar/nsZipArchive.cpp	2009-04-27 16:56:30.000000000 -0400
@@ -51,7 +51,7 @@
 
 #ifndef STANDALONE
 
-#include "nsWildCard.h"
+#include "nsWildCard_jar.h"
 #include "nscore.h"
 #include "prmem.h"
 #include "prio.h"
diff -ruN mozilla-1.9.1-orig/modules/libjar/objs.mk mozilla-1.9.1/modules/libjar/objs.mk
--- mozilla-1.9.1-orig/modules/libjar/objs.mk	2009-04-23 21:46:48.000000000 -0400
+++ mozilla-1.9.1/modules/libjar/objs.mk	2009-04-27 16:56:30.000000000 -0400
@@ -37,7 +37,7 @@
 
 MODULES_STANDALONE_LCPPSRCS = \
 		nsZipArchive.cpp \
-		nsWildCard.cpp \
+		nsWildCard_jar.cpp \
 		$(NULL)
 
 MODULES_LIBJAR_LCPPSRCS = \
diff -ruN mozilla-1.9.1-orig/modules/libpr0n/decoders/icon/Makefile.in mozilla-1.9.1/modules/libpr0n/decoders/icon/Makefile.in
--- mozilla-1.9.1-orig/modules/libpr0n/decoders/icon/Makefile.in	2009-04-23 21:46:48.000000000 -0400
+++ mozilla-1.9.1/modules/libpr0n/decoders/icon/Makefile.in	2009-04-27 16:56:30.000000000 -0400
@@ -70,7 +70,7 @@
 PLATFORM = os2
 endif
 
-ifeq ($(OS_ARCH),Darwin)
+ifeq (cocoa,$(MOZ_WIDGET_TOOLKIT))
 PLATFORM = mac
 endif
 
diff -ruN mozilla-1.9.1-orig/netwerk/wifi/src/Makefile.in mozilla-1.9.1/netwerk/wifi/src/Makefile.in
--- mozilla-1.9.1-orig/netwerk/wifi/src/Makefile.in	Tue May 05 21:33:44 2009 +0200
+++ mozilla-1.9.1/netwerk/wifi/src/Makefile.in	Tue May 05 20:50:38 2009 -0400
@@ -57,16 +57,16 @@
 		nsWifiAccessPoint.cpp \
 		$(NULL)
 
-ifeq ($(MOZ_WIDGET_TOOLKIT),cocoa)
+ifeq ($(OS_ARCH),Darwin)
 CPPSRCS += nsWifiScannerMac.cpp
 CMMSRCS = osx_corewlan.mm
 else
-ifeq ($(MOZ_WIDGET_TOOLKIT),windows)
+ifneq (,$(filter WINCE WINNT,$(OS_ARCH)))
 CPPSRCS	+= nsWifiScannerWin.cpp
 else
-ifeq ($(MOZ_WIDGET_TOOLKIT),gtk2)
+ifeq ($(OS_ARCH),Linux)
 CPPSRCS	+= nsWifiScannerUnix.cpp
-endif # gtk
+endif # linux
 endif # windows
 endif # mac
 
--- mozilla-1.9.1-orig/netwerk/wifi/src/nsWifiMonitor.h	2009-10-16 11:14:14.000000000 -0400
+++ mozilla-1.9.1/netwerk/wifi/src/nsWifiMonitor.h	2009-10-28 09:30:59.000000000 -0400
@@ -88,7 +88,7 @@
 
   nsresult DoScan();
 
-#if defined(XP_MACOSX)
+#if defined(XP_UNIX)
   nsresult DoScanWithCoreWLAN();
   nsresult DoScanOld();
 #endif
diff -ruN mozilla-1.9.1-orig/toolkit/library/Makefile.in mozilla-1.9.1/toolkit/library/Makefile.in
--- mozilla-1.9.1-orig/toolkit/library/Makefile.in	2009-04-23 21:47:06.000000000 -0400
+++ mozilla-1.9.1/toolkit/library/Makefile.in	2009-04-27 17:31:35.000000000 -0400
@@ -205,9 +205,12 @@
 	-lcrypto \
 	$(TK_LIBS) \
 	$(NULL)
+endif
 
+ifeq ($(OS_ARCH),Darwin)
 ifdef MOZ_SYDNEYAUDIO
 EXTRA_DSO_LDOPTS += \
+	-framework Carbon \
 	-framework CoreAudio \
 	-framework AudioToolbox \
 	-framework AudioUnit \
diff -ruN mozilla-1.9.1-orig/toolkit/mozapps/installer/packager.mk mozilla-1.9.1/toolkit/mozapps/installer/packager.mk
--- mozilla-1.9.1-orig/toolkit/mozapps/installer/packager.mk	2009-04-23 21:47:07.000000000 -0400
+++ mozilla-1.9.1/toolkit/mozapps/installer/packager.mk	2009-04-27 16:56:30.000000000 -0400
@@ -437,7 +437,7 @@
 # dist/sdk/lib -> prefix/lib/appname-devel-version/lib
 # prefix/lib/appname-devel-version/* symlinks to the above directories
 install:: stage-package
-ifneq (,$(filter WINNT Darwin,$(OS_ARCH)))
+ifneq (,$(filter WINNT,$(OS_ARCH)))
 	$(error "make install" is not supported on this platform. Use "make package" instead.)
 endif
 	$(NSINSTALL) -D $(DESTDIR)$(installdir)
@@ -453,7 +453,7 @@
 	(cd $(DIST)/sdk/include && tar $(TAR_CREATE_FLAGS) - .) | \
 	  (cd $(DESTDIR)$(includedir)/stable && tar -xf -)
 # The dist/include has module subdirectories that we need to flatten
-	find $(DIST)/include -xtype f -exec $(SYSINSTALL) $(IFLAGS1) {} $(DESTDIR)$(includedir)/unstable \;
+	find $(DIST)/include -type f -exec $(SYSINSTALL) $(IFLAGS1) {} $(DESTDIR)$(includedir)/unstable \;
 # IDL directory is stable (dist/sdk/idl) and unstable (dist/idl)
 	$(NSINSTALL) -D $(DESTDIR)$(idldir)/stable 
 	$(NSINSTALL) -D $(DESTDIR)$(idldir)/unstable
diff -ruN mozilla-1.9.1-orig/widget/src/gtk2/nsFilePicker.cpp mozilla-1.9.1/wiwidget/src/gtk2/nsFilePicker.cpp
--- mozilla-1.9.1-orig/widget/src/gtk2/nsFilePicker.cpp 2009-04-03 18:35:03.000000000 -0700
+++ mozilla-1.9.1/widget/src/gtk2/nsFilePicker.cpp 2009-04-03 18:38:58.000000000 -0700
@@ -203,7 +203,11 @@
     _gtk_file_chooser_get_filename = (_gtk_file_chooser_get_filename_fn)func;
   } else {
     // XXX hmm, this seems to fail when gtk 2.4 is already loaded...
+#ifdef __APPLE__
+    mGTK24 = PR_LoadLibrary("@FINKPREFIX@/lib/libgtk-x11-2.0.dylib");
+#else
     mGTK24 = LoadVersionedLibrary("gtk-2", ".4");
+#endif
     if (!mGTK24) {
       return NS_ERROR_NOT_AVAILABLE;
     }
diff -ruN mozilla-1.9.1-orig/xpcom/base/nsStackWalk.cpp mozilla-1.9.1/xpcom/base/nsStackWalk.cpp
--- mozilla-1.9.1-orig/xpcom/base/nsStackWalk.cpp	2009-04-23 21:47:11.000000000 -0400
+++ mozilla-1.9.1/xpcom/base/nsStackWalk.cpp	2009-04-27 16:56:31.000000000 -0400
@@ -1124,7 +1124,7 @@
 #define __USE_GNU
 #endif
 
-#if defined(HAVE_LIBDL) || defined(XP_MACOSX)
+#if defined(HAVE_LIBDL) || defined(XP_UNIX)
 #include <dlfcn.h>
 #endif
 
diff -ruN mozilla-1.9.1-orig/xpcom/stub/Makefile.in mozilla-1.9.1/xpcom/stub/Makefile.in
--- mozilla-1.9.1-orig/xpcom/stub/Makefile.in	2009-04-23 21:47:12.000000000 -0400
+++ mozilla-1.9.1/xpcom/stub/Makefile.in	2009-04-27 16:56:31.000000000 -0400
@@ -108,8 +108,8 @@
 endif
 
 ifeq (Darwin,$(OS_ARCH))
-EXTRA_DSO_LDOPTS += $(DIST)/bin/XUL
-DEPENDENT_LIBS_LIST += XUL
+EXTRA_DSO_LDOPTS += $(DIST)/bin/libxul.dylib
+DEPENDENT_LIBS_LIST += $(LIB_PREFIX)xul$(DLL_SUFFIX)
 else
 EXTRA_DSO_LIBS = xul
 DEPENDENT_LIBS_LIST += $(LIB_PREFIX)xul$(DLL_SUFFIX)
