diff -uNr galeon-2.0.3/mozilla/mozilla-embed-shell.cpp galeon-2.0.3-new/mozilla/mozilla-embed-shell.cpp
--- galeon-2.0.3/mozilla/mozilla-embed-shell.cpp	2006-07-01 16:55:18.000000000 -0400
+++ galeon-2.0.3-new/mozilla/mozilla-embed-shell.cpp	2007-09-25 14:59:03.000000000 -0400
@@ -236,6 +236,9 @@
 
 	g_setenv ("MOZ_PLUGIN_PATH", new_path->str, TRUE);
 
+	/* Work-around Flash Player crash */
+	g_setenv ("XLIB_SKIP_ARGB_VISUALS", "1", FALSE);
+
 	g_string_free (new_path, TRUE);
 }
 
diff -uNr galeon-2.0.3/utils/gul-x11.c galeon-2.0.3-new/utils/gul-x11.c
--- galeon-2.0.3/utils/gul-x11.c	2005-02-10 13:56:17.000000000 -0500
+++ galeon-2.0.3-new/utils/gul-x11.c	2007-09-25 14:57:35.000000000 -0400
@@ -52,7 +52,7 @@
 
        XGetWindowProperty (GDK_DISPLAY_XDISPLAY (display), GDK_WINDOW_XID (root_win),
                            gdk_x11_get_xatom_by_name_for_display (display, "_NET_CURRENT_DESKTOP"),
-                           0, G_MAXLONG,
+                           0, G_MAXLONG - 3,
                            False, XA_CARDINAL, &type, &format, &nitems,
                            &bytes_after, (gpointer)&current_desktop);
        if (type == XA_CARDINAL && format == 32 && nitems > 0)
@@ -85,7 +85,7 @@
 
        XGetWindowProperty (GDK_DISPLAY_XDISPLAY (display), GDK_WINDOW_XID (window),
                            gdk_x11_get_xatom_by_name_for_display (display, "_NET_WM_DESKTOP"),
-                           0, G_MAXLONG,
+                           0, G_MAXLONG - 3,
                            False, XA_CARDINAL, &type, &format, &nitems,
                            &bytes_after, (gpointer)&workspace);
        if (type == XA_CARDINAL && format == 32 && nitems > 0)
