diff -uNr gnucash-2.0.5/configure gnucash-2.0.5-new/configure
--- gnucash-2.0.5/configure	2007-02-18 23:09:58.000000000 -0500
+++ gnucash-2.0.5-new/configure	2007-03-16 11:49:20.000000000 -0400
@@ -29892,23 +29892,23 @@
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        { echo "$as_me:$LINENO: checking for libgtkhtml-3.8 " >&5
-echo $ECHO_N "checking for libgtkhtml-3.8 ... $ECHO_C" >&6; }
+        { echo "$as_me:$LINENO: checking for libgtkhtml-3.14 " >&5
+echo $ECHO_N "checking for libgtkhtml-3.14 ... $ECHO_C" >&6; }
 
-        if $PKG_CONFIG --silence-errors --exists "libgtkhtml-3.8 " ; then
+        if $PKG_CONFIG --silence-errors --exists "libgtkhtml-3.14 " ; then
             { echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
             succeeded=yes
 
             { echo "$as_me:$LINENO: checking GTKHTML_CFLAGS" >&5
 echo $ECHO_N "checking GTKHTML_CFLAGS... $ECHO_C" >&6; }
-            GTKHTML_CFLAGS=`$PKG_CONFIG --cflags "libgtkhtml-3.8 "`
+            GTKHTML_CFLAGS=`$PKG_CONFIG --cflags "libgtkhtml-3.14 "`
             { echo "$as_me:$LINENO: result: $GTKHTML_CFLAGS" >&5
 echo "${ECHO_T}$GTKHTML_CFLAGS" >&6; }
 
             { echo "$as_me:$LINENO: checking GTKHTML_LIBS" >&5
 echo $ECHO_N "checking GTKHTML_LIBS... $ECHO_C" >&6; }
-            GTKHTML_LIBS=`$PKG_CONFIG --libs "libgtkhtml-3.8 "`
+            GTKHTML_LIBS=`$PKG_CONFIG --libs "libgtkhtml-3.14 "`
             { echo "$as_me:$LINENO: result: $GTKHTML_LIBS" >&5
 echo "${ECHO_T}$GTKHTML_LIBS" >&6; }
         else
@@ -29916,7 +29916,7 @@
             GTKHTML_LIBS=""
             ## If we have a custom action on failure, don't print errors, but
             ## do set a variable so people can do so.
-            GTKHTML_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgtkhtml-3.8 " 2>&1`
+            GTKHTML_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgtkhtml-3.14 " 2>&1`
 
         fi
 
diff -uNr gnucash-2.0.5/src/gnome-utils/gnc-html.c gnucash-2.0.5-new/src/gnome-utils/gnc-html.c
--- gnucash-2.0.5/src/gnome-utils/gnc-html.c	2007-02-18 23:10:23.000000000 -0500
+++ gnucash-2.0.5-new/src/gnome-utils/gnc-html.c	2007-03-16 11:48:55.000000000 -0400
@@ -1325,7 +1325,7 @@
     return;
   }
 
-  gtk_html_print(GTK_HTML(html->html), ps->context);
+  gtk_html_print_page(GTK_HTML(html->html), ps->context);
   gnc_print_session_done(ps);
 }
 
