diff -Nurd gnumeric-1.8.0/doc/C/gnumeric-C.omf modgnumeric-1.8.0/doc/gnumeric-C.omf
--- gnumeric-1.8.0/doc/C/gnumeric-C.omf	1969-12-31 19:00:00.000000000 -0500
+++ modgnumeric-1.8.0/doc/C/gnumeric-C.omf	2008-01-11 11:07:48.000000000 -0500
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!DOCTYPE omf PUBLIC "-//OMF//DTD Scrollkeeper OMF Variant V1.0//EN" "http://scrollkeeper.sourceforge.net/dtds/scrollkeeper-omf-1.0/scrollkeeper-omf.dtd">
+<omf>
+  <resource>
+    <creator>
+      miguel@gnome.org (Miguel de Icaza)
+    </creator>
+    <maintainer>
+      jody@gnome.org (Jody Goldberg)
+    </maintainer>
+    <contributor>
+      acuster@gmail.com (Adrian Custer)
+    </contributor>
+    <contributor>
+      mrproper@ximian.com (Kevin Breit)
+    </contributor>
+    <contributor>
+      aguelzow@taliesin.ca (Andreas Guelzow)
+    </contributor>
+    <contributor>
+      aaron@ximian.com (Aaron Weber)
+    </contributor>
+    <contributor>
+      tommydal@ihug.com.au (Thomas Canty)
+    </contributor>
+    <contributor>
+      kirillov@math.sunysb.edu (Alexander Kirillov)
+    </contributor>
+    <contributor>
+      k_wayne@linuxpower.org (Wayne Schuller)
+    </contributor>
+    <contributor>
+      jiivonen@hutcs.cs.hut.fi (Jukka-Pekka Iivonen)
+    </contributor>
+    <contributor>
+      adrian@gimp.or (Adrian Likins)
+    </contributor>
+
+    <title>
+      The Gnumeric Manual, version 1.8.
+    </title>
+
+    <date>
+      2007-12-20
+    </date>
+
+    <version identifier="1.7" date="2007-10-27" 
+      description="Updated for 1.8"/>
+    <version identifier="1.3" date="2004-12-09" 
+      description="Updated for 1.4"/>
+
+    <subject category="GNOME|Applications|Office"/>
+
+    <description>
+      This is the user's manual for Gnumeric, a GNOME spreadsheet.
+    </description>
+
+    <type>
+      User's Guide
+    </type>
+
+    <format mime="text/xml" dtd="-//OASIS//DTD DocBook XML V4.2//EN"/>
+    <identifier url="gnumeric.xml"/>
+    <language code="C"/>
+    <relation seriesid="c5e3db84-64c3-11d6-85a3-a3e50590e6d1"/>
+    <rights type="GNU FDL" license.version="1.1"
+      license="http://www.gnu.org/licenses/fdl.html" 
+      holder="Miguel de Icaza, Jody Goldberg, Thomas Canty, 
+      Jukka-Pekka Iivonen, Andreas J. Guelzow, Wayne Schuller, 
+      Adrian Custer, Kevin Breit, Aaron Weber, Alexander Kirillov"/>
+  </resource>
+</omf>
diff -Nurd gnumeric-1.8.0/plugins/python-loader/gnm-python.c modgnumeric-1.8.0/plugins/python-loader/gnm-python.c
--- gnumeric-1.8.0/plugins/python-loader/gnm-python.c	2007-12-29 01:43:58.000000000 -0500
+++ modgnumeric-1.8.0/plugins/python-loader/gnm-python.c	2008-01-11 20:16:53.000000000 -0500
@@ -7,6 +7,7 @@
 
 #include <gnumeric-config.h>
 #include <Python.h>
+#define NO_IMPORT_PYGOBJECT
 #include <pygobject.h>
 #include <glib.h>
 #include "gnm-py-interpreter.h"
diff -Nurd gnumeric-1.8.0/plugins/python-loader/py-gnumeric.c modgnumeric-1.8.0/plugins/python-loader/py-gnumeric.c
--- gnumeric-1.8.0/plugins/python-loader/py-gnumeric.c	2007-12-29 01:43:54.000000000 -0500
+++ modgnumeric-1.8.0/plugins/python-loader/py-gnumeric.c	2008-01-11 20:19:23.000000000 -0500
@@ -9,6 +9,7 @@
 #include <Python.h>
 #include <gnumeric.h>
 #include <glib.h>
+#define NO_IMPORT_PYGOBJECT
 #include "pygobject.h"
 #include "application.h"
 #include "workbook.h"
diff -Nurd gnumeric-1.8.0/plugins/python-loader/python-loader.c modgnumeric-1.8.0/plugins/python-loader/python-loader.c
--- gnumeric-1.8.0/plugins/python-loader/python-loader.c	2007-12-29 01:43:59.000000000 -0500
+++ modgnumeric-1.8.0/plugins/python-loader/python-loader.c	2008-01-11 22:52:41.000000000 -0500
@@ -10,6 +10,7 @@
 #include "python-loader.h"
 
 #include <Python.h>
+#define NO_IMPORT_PYGOBJECT
 #include "pygobject.h"
 #include "py-gnumeric.h"
 #include "gnm-python.h"
