diff -Nurd -x'*~' irssi-0.8.12.orig/configure irssi-0.8.12/configure
--- irssi-0.8.12.orig/configure	2007-10-06 05:41:38.000000000 -0400
+++ irssi-0.8.12/configure	2007-11-11 14:52:36.000000000 -0500
@@ -24794,7 +24794,7 @@
 
 	    CURSES_INCLUDEDIR="-I$withval/include"
 	    search_ncurses=false
-	    screen_manager=""ncurses on $withval/include""
+	    screen_manager="ncurses on $withval/include"
             cat >>confdefs.h <<\_ACEOF
 #define HAS_CURSES 1
 _ACEOF
