diff -Nurd kdegames-4.2.2/kubrick/src/CMakeLists.txt kdegames-4.2.2-new/kubrick/src/CMakeLists.txt
--- kdegames-4.2.2/kubrick/src/CMakeLists.txt	2009-02-26 09:32:26.000000000 -0500
+++ kdegames-4.2.2-new/kubrick/src/CMakeLists.txt	2009-03-30 10:54:47.000000000 -0400
@@ -1,6 +1,8 @@
 
 ########### next target ###############
 
+include_directories( ${OPENGL_INCLUDE_DIR} )
+
 set(kubrick_SRCS 
    main.cpp 
    kubrick.cpp 
