diff -Nurd kdepim-4.3.98/kleopatra/kwatchgnupg/CMakeLists.txt kdepim-4.3.98-new/kleopatra/kwatchgnupg/CMakeLists.txt
--- kdepim-4.3.98/kleopatra/kwatchgnupg/CMakeLists.txt	2009-05-14 13:27:05.000000000 -0400
+++ kdepim-4.3.98-new/kleopatra/kwatchgnupg/CMakeLists.txt	2010-02-04 13:50:05.000000000 -0500
@@ -23,10 +23,10 @@
 add_definitions( -DKDE_DEFAULT_DEBUG_AREA=5153 )
 
 if ( NOT KDE4_KCMUTILS_LIBS )
-   set ( KDE4_KCMUTILS_LIBS ${KDE4_KUTILS_LIBS} )
+   set ( KDE4_KCMUTILS_LIBS ${KDE4_KUTILS_LIBS} ${QGPGME_LIBRARIES} )
 endif ( NOT KDE4_KCMUTILS_LIBS )
 
-target_link_libraries(kwatchgnupg ${KDE4_KDEUI_LIBS} kleo ${KDEPIMLIBS_GPGMEPP_LIBS})
+target_link_libraries(kwatchgnupg ${KDE4_KDEUI_LIBS} kleo ${QGPGME_LIBRARIES})
 
 install(TARGETS kwatchgnupg ${INSTALL_TARGETS_DEFAULT_ARGS})
 
diff -Nurd kdepim-4.3.98/ksendemail/CMakeLists.txt kdepim-4.3.98-new/ksendemail/CMakeLists.txt
--- kdepim-4.3.98/ksendemail/CMakeLists.txt	2009-10-26 05:15:23.000000000 -0400
+++ kdepim-4.3.98-new/ksendemail/CMakeLists.txt	2010-02-04 13:50:18.000000000 -0500
@@ -9,7 +9,7 @@
 
   qt4_add_dbus_interfaces( ksendemail_SRCS  ${CMAKE_BINARY_DIR}/kmail/org.kde.kmail.kmail.xml )
 
-  kde4_add_executable( ksendemail ${ksendemail_SRCS})
+  kde4_add_executable( ksendemail NOGUI ${ksendemail_SRCS})
   add_dependencies( ksendemail kmail_xml )
   target_link_libraries( ksendemail ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${KDE4_KIO_LIBS} )
 
diff -Nurd kdepim-4.3.98/nepomuk_email_feeder/CMakeLists.txt kdepim-4.3.98-new/nepomuk_email_feeder/CMakeLists.txt
--- kdepim-4.3.98/nepomuk_email_feeder/CMakeLists.txt	2009-12-31 08:28:57.000000000 -0500
+++ kdepim-4.3.98-new/nepomuk_email_feeder/CMakeLists.txt	2010-02-04 13:50:05.000000000 -0500
@@ -4,6 +4,7 @@
   ${Boost_INCLUDE_DIR}
   ${CMAKE_CURRENT_SOURCE_DIR}/shared
   ${SOPRANO_INCLUDE_DIR}
+  ${STRIGI_INCLUDE_DIR}
 )
 
 set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${KDE4_ENABLE_EXCEPTIONS}" )
