diff -Nurd libgda-3.1.2.orig/libgda-report/DocBook/Makefile.am libgda-3.1.2/libgda-report/DocBook/Makefile.am
--- libgda-3.1.2.orig/libgda-report/DocBook/Makefile.am	2007-09-17 03:25:04.000000000 -0400
+++ libgda-3.1.2/libgda-report/DocBook/Makefile.am	2008-02-28 16:09:18.000000000 -0500
@@ -14,8 +14,8 @@
 	$(gda_report_headers)	\
 	gda-report-docbook-document.c	
 
-libgda_report_docbook_3_0_la_LIBADD = $(LIBGDA_LIBS) \
-	$(top_builddir)/libgda/libgda-3.0.la
+libgda_report_docbook_3_0_la_LIBADD = \
+	$(top_builddir)/libgda/libgda-3.0.la $(LIBGDA_LIBS)
 
 gdareportincludedir=$(includedir)/libgda-$(GDA_ABI_MAJOR_VERSION).$(GDA_ABI_MINOR_VERSION)/libgda-report
 gdareportinclude_HEADERS=$(gda_report_headers)
diff -Nurd libgda-3.1.2.orig/libgda-report/RML/Makefile.am libgda-3.1.2/libgda-report/RML/Makefile.am
--- libgda-3.1.2.orig/libgda-report/RML/Makefile.am	2007-09-17 03:25:14.000000000 -0400
+++ libgda-3.1.2/libgda-report/RML/Makefile.am	2008-02-28 16:09:18.000000000 -0500
@@ -16,8 +16,8 @@
 	$(gda_report_headers)	\
 	gda-report-rml-document.c	
 
-libgda_report_rml_3_0_la_LIBADD = $(LIBGDA_LIBS) \
-	$(top_builddir)/libgda/libgda-3.0.la
+libgda_report_rml_3_0_la_LIBADD = \
+	$(top_builddir)/libgda/libgda-3.0.la $(LIBGDA_LIBS)
 
 gdareportincludedir=$(includedir)/libgda-$(GDA_ABI_MAJOR_VERSION).$(GDA_ABI_MINOR_VERSION)/libgda-report
 gdareportinclude_HEADERS=$(gda_report_headers)
diff -Nurd libgda-3.1.2.orig/libgda-report/engine/Makefile.am libgda-3.1.2/libgda-report/engine/Makefile.am
--- libgda-3.1.2.orig/libgda-report/engine/Makefile.am	2007-09-17 03:24:38.000000000 -0400
+++ libgda-3.1.2/libgda-report/engine/Makefile.am	2008-02-28 16:09:18.000000000 -0500
@@ -15,8 +15,8 @@
 	$(gda_report_headers)	\
 	gda-report-engine.c	
 
-libgda_report_engine_3_0_la_LIBADD = $(LIBGDA_LIBS) \
-	$(top_builddir)/libgda/libgda-3.0.la 
+libgda_report_engine_3_0_la_LIBADD = \
+	$(top_builddir)/libgda/libgda-3.0.la $(LIBGDA_LIBS)
 
 gdareportincludedir=$(includedir)/libgda-$(GDA_ABI_MAJOR_VERSION).$(GDA_ABI_MINOR_VERSION)/libgda-report
 gdareportinclude_HEADERS=$(gda_report_headers)
diff -Nurd libgda-3.1.2.orig/libgda-xslt/Makefile.am libgda-3.1.2/libgda-xslt/Makefile.am
--- libgda-3.1.2.orig/libgda-xslt/Makefile.am	2007-09-26 15:48:17.000000000 -0400
+++ libgda-3.1.2/libgda-xslt/Makefile.am	2008-02-28 16:09:18.000000000 -0500
@@ -18,8 +18,8 @@
 	sql_exslt_internal.c \
 	sql_backend.c
 
-libgda_xslt_3_0_la_LIBADD = $(LIBGDA_LIBS) $(LIBXSLT_LIBS)\
-        $(top_builddir)/libgda/libgda-3.0.la
+libgda_xslt_3_0_la_LIBADD = \
+        $(top_builddir)/libgda/libgda-3.0.la $(LIBGDA_LIBS) $(LIBXSLT_LIBS)
 
 gdaxsltincludedir=$(includedir)/libgda-$(GDA_ABI_MAJOR_VERSION).$(GDA_ABI_MINOR_VERSION)/libgda-xslt
 gdaxsltinclude_HEADERS=$(gda_xslt_headers)
diff -Nurd libgda-3.1.2.orig/providers/bdb/Makefile.am libgda-3.1.2/providers/bdb/Makefile.am
--- libgda-3.1.2.orig/providers/bdb/Makefile.am	2007-09-12 05:58:41.000000000 -0400
+++ libgda-3.1.2/providers/bdb/Makefile.am	2008-02-28 16:09:18.000000000 -0500
@@ -23,8 +23,7 @@
 
 gda_bdb_test_SOURCES = gda-bdb-test.c
 gda_bdb_test_LDADD = \
-	$(LIBGDA_LIBS) \
-	$(top_builddir)/libgda/libgda-3.0.la 
+	$(top_builddir)/libgda/libgda-3.0.la $(LIBGDA_LIBS)
 
 xmldir   = $(datadir)/libgda-3.0
 xml_in_files = bdb_specs_dsn.xml.in
diff -Nurd libgda-3.1.2.orig/tests/data-models/Makefile.am libgda-3.1.2/tests/data-models/Makefile.am
--- libgda-3.1.2.orig/tests/data-models/Makefile.am	2007-09-12 09:27:18.000000000 -0400
+++ libgda-3.1.2/tests/data-models/Makefile.am	2008-02-28 16:09:18.000000000 -0500
@@ -11,23 +11,23 @@
 
 check_model_import_SOURCES = $(common_sources) check_model_import.c
 check_model_import_CFLAGS = @CHECK_CFLAGS@
-check_model_import_LDADD = @CHECK_LIBS@ \
+check_model_import_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_virtual_SOURCES = $(common_sources) check_virtual.c
 check_virtual_CFLAGS = @CHECK_CFLAGS@ \
 	-I$(top_srcdir)/libgda/sqlite \
 	-I$(top_srcdir)/libgda/sqlite/virtual
-check_virtual_LDADD = @CHECK_LIBS@ \
+check_virtual_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_data_proxy_SOURCES = $(common_sources) check_data_proxy.c
 check_data_proxy_CFLAGS = @CHECK_CFLAGS@ 
-check_data_proxy_LDADD = @CHECK_LIBS@ \
+check_data_proxy_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 
 EXTRA_DIST = \
diff -Nurd libgda-3.1.2.orig/tests/providers/Makefile.am libgda-3.1.2/tests/providers/Makefile.am
--- libgda-3.1.2.orig/tests/providers/Makefile.am	2007-08-26 11:23:30.000000000 -0400
+++ libgda-3.1.2/tests/providers/Makefile.am	2008-02-28 16:09:18.000000000 -0500
@@ -81,87 +81,87 @@
 
 check_sqlite_SOURCES = $(common_sources) check_sqlite.c
 check_sqlite_CFLAGS = @CHECK_CFLAGS@
-check_sqlite_LDADD = @CHECK_LIBS@ \
+check_sqlite_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_bdb_SOURCES = $(common_sources) check_bdb.c
 check_bdb_CFLAGS = @CHECK_CFLAGS@
-check_bdb_LDADD = @CHECK_LIBS@ \
+check_bdb_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_ibmdb2_SOURCES = $(common_sources) check_ibmdb2.c
 check_ibmdb2_CFLAGS = @CHECK_CFLAGS@
-check_ibmdb2_LDADD = @CHECK_LIBS@ \
+check_ibmdb2_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_firebird_SOURCES = $(common_sources) check_firebird.c
 check_firebird_CFLAGS = @CHECK_CFLAGS@
-check_firebird_LDADD = @CHECK_LIBS@ \
+check_firebird_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_mdb_SOURCES = $(common_sources) check_mdb.c
 check_mdb_CFLAGS = @CHECK_CFLAGS@
-check_mdb_LDADD = @CHECK_LIBS@ \
+check_mdb_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_mysql_SOURCES = $(common_sources) check_mysql.c
 check_mysql_CFLAGS = @CHECK_CFLAGS@
-check_mysql_LDADD = @CHECK_LIBS@ \
+check_mysql_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_msql_SOURCES = $(common_sources) check_msql.c
 check_msql_CFLAGS = @CHECK_CFLAGS@
-check_msql_LDADD = @CHECK_LIBS@ \
+check_msql_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_odbc_SOURCES = $(common_sources) check_odbc.c
 check_odbc_CFLAGS = @CHECK_CFLAGS@
-check_odbc_LDADD = @CHECK_LIBS@ \
+check_odbc_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_oracle_SOURCES = $(common_sources) check_oracle.c
 check_oracle_CFLAGS = @CHECK_CFLAGS@
-check_oracle_LDADD = @CHECK_LIBS@ \
+check_oracle_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_postgres_SOURCES = $(common_sources) check_postgres.c
 check_postgres_CFLAGS = @CHECK_CFLAGS@
-check_postgres_LDADD = @CHECK_LIBS@ \
+check_postgres_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_sybase_SOURCES = $(common_sources) check_sybase.c
 check_sybase_CFLAGS = @CHECK_CFLAGS@
-check_sybase_LDADD = @CHECK_LIBS@ \
+check_sybase_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_freetds_SOURCES = $(common_sources) check_freetds.c
 check_freetds_CFLAGS = @CHECK_CFLAGS@
-check_freetds_LDADD = @CHECK_LIBS@ \
+check_freetds_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_xbase_SOURCES = $(common_sources) check_xbase.c
 check_xbase_CFLAGS = @CHECK_CFLAGS@
-check_xbase_LDADD = @CHECK_LIBS@ \
+check_xbase_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_ldap_SOURCES = $(common_sources) check_ldap.c
 check_ldap_CFLAGS = @CHECK_CFLAGS@
-check_ldap_LDADD = @CHECK_LIBS@ \
+check_ldap_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 EXTRA_DIST = \
 	README \
diff -Nurd libgda-3.1.2.orig/tests/queries/Makefile.am libgda-3.1.2/tests/queries/Makefile.am
--- libgda-3.1.2.orig/tests/queries/Makefile.am	2007-08-29 09:37:52.000000000 -0400
+++ libgda-3.1.2/tests/queries/Makefile.am	2008-02-28 16:51:06.000000000 -0500
@@ -10,15 +10,15 @@
 
 check_delimiter_SOURCES = test-util.h test-util.c check_delimiter.c $(top_srcdir)/libgda/sql-delimiter/gda-sql-delimiter.h
 check_delimiter_CFLAGS = @CHECK_CFLAGS@
-check_delimiter_LDADD = @CHECK_LIBS@ \
+check_delimiter_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 check_gdaquery_SOURCES = test-util.h test-util.c check_gdaquery.c $(top_srcdir)/libsql/sql_parser.h
 check_gdaquery_CFLAGS = @CHECK_CFLAGS@
-check_gdaquery_LDADD = @CHECK_LIBS@  \
+check_gdaquery_LDADD = \
 	$(top_builddir)/libgda/libgda-3.0.la \
-	$(LIBGDA_LIBS)
+	$(LIBGDA_LIBS) @CHECK_LIBS@
 
 EXTRA_DIST = all_sql_tests.xml \
 	app_sql.xml \
