diff -Nurd libgnomedb-3.1.2.orig/configure libgnomedb-3.1.2/configure
--- libgnomedb-3.1.2.orig/configure	2007-10-24 15:44:07.000000000 -0400
+++ libgnomedb-3.1.2/configure	2008-03-04 13:51:04.000000000 -0500
@@ -23288,7 +23288,7 @@
 			LIBGCRYPT_LIBS=""
 
 
-if 0; then
+if false; then
   HAVE_LIBGCRYPT_TRUE=
   HAVE_LIBGCRYPT_FALSE='#'
 else
@@ -23316,7 +23316,7 @@
       	LIBGCRYPT_LIBS=""
 
 
-if 0; then
+if false; then
   HAVE_LIBGCRYPT_TRUE=
   HAVE_LIBGCRYPT_FALSE='#'
 else
@@ -23436,6 +23436,7 @@
 
 GOOCANVAS_MODULES="goocanvas >= 0.9"
 
+if false; then
 pkg_failed=no
 { echo "$as_me:$LINENO: checking for GOOCANVAS" >&5
 echo $ECHO_N "checking for GOOCANVAS... $ECHO_C" >&6; }
@@ -23506,6 +23507,8 @@
 echo "${ECHO_T}yes" >&6; }
 	have_goocanvas=yes
 fi
+fi
+have_goocanvas=no
 
 
 if test x"$have_goocanvas" = "xyes"; then
@@ -23527,6 +23530,7 @@
 
 GRAPHVIZ_MODULES="libgvc"
 
+if false; then
 pkg_failed=no
 { echo "$as_me:$LINENO: checking for GRAPHVIZ" >&5
 echo $ECHO_N "checking for GRAPHVIZ... $ECHO_C" >&6; }
@@ -23597,6 +23601,8 @@
 echo "${ECHO_T}yes" >&6; }
 	have_graphviz=yes
 fi
+fi
+have_graphviz=no
 
 
 if test x"$have_graphviz" = "xyes"; then
@@ -24001,6 +24007,7 @@
 
 
 
+if false; then
 pkg_failed=no
 { echo "$as_me:$LINENO: checking for GTKSOURCEVIEW" >&5
 echo $ECHO_N "checking for GTKSOURCEVIEW... $ECHO_C" >&6; }
@@ -24071,6 +24078,8 @@
 echo "${ECHO_T}yes" >&6; }
 	have_gtksourceview=yes
 fi
+fi
+have_gtksourceview=no
 
 
 if test x"$have_gtksourceview" = "xyes"; then
@@ -24092,6 +24101,7 @@
 
 
 
+if false; then
 pkg_failed=no
 { echo "$as_me:$LINENO: checking for GDA_EVOLUTION" >&5
 echo $ECHO_N "checking for GDA_EVOLUTION... $ECHO_C" >&6; }
@@ -24167,6 +24177,8 @@
 	EVOLUTION_VERSION=`pkg-config --modversion libedataserver-1.2`
 
 fi
+fi
+have_evolution=no
 
 
 if test x"$have_evolution" = "xyes"; then
diff -Nurd libgnomedb-3.1.2.orig/glade3/Makefile.in libgnomedb-3.1.2/glade3/Makefile.in
--- libgnomedb-3.1.2.orig/glade3/Makefile.in	2007-10-24 16:22:18.000000000 -0400
+++ libgnomedb-3.1.2/glade3/Makefile.in	2008-03-04 13:18:29.000000000 -0500
@@ -319,7 +319,7 @@
 @HAVE_LIBGNOMECANVAS_TRUE@graphlib = $(top_builddir)/libgnomedb-graph/libgnomedb_graph-3.0.la
 gladegnomedb_LTLIBRARIES = libgladegnomedb.la
 libgladegnomedb_la_SOURCES = glade3-gnomedb.c
-libgladegnomedb_la_CFLAGS = -g -Wall -I$(top_srcdir) -I$(top_builddir) \
+AM_CPPFLAGS = -g -Wall -I$(top_srcdir) -I$(top_builddir) \
                             -DG_LOG_DOMAIN=\"GnomeDB\" \
                             $(GLADE3_CFLAGS) $(LIBGNOMEDB_CFLAGS) $(LIBGLADE_CFLAGS)
 
