diff -Nurd -x'*~' libgda-3.1.2.orig/libgda-report/DocBook/Makefile.in libgda-3.1.2/libgda-report/DocBook/Makefile.in
--- libgda-3.1.2.orig/libgda-report/DocBook/Makefile.in	2007-10-24 16:17:52.000000000 -0400
+++ libgda-3.1.2/libgda-report/DocBook/Makefile.in	2008-02-28 13:47:27.000000000 -0500
@@ -367,8 +367,9 @@
 	$(gda_report_headers)	\
 	gda-report-docbook-document.c	
 
-libgda_report_docbook_3_0_la_LIBADD = $(LIBGDA_LIBS) \
-	$(top_builddir)/libgda/libgda-3.0.la
+libgda_report_docbook_3_0_la_LIBADD = \
+	$(top_builddir)/libgda/libgda-3.0.la \
+	$(LIBGDA_LIBS)
 
 gdareportincludedir = $(includedir)/libgda-$(GDA_ABI_MAJOR_VERSION).$(GDA_ABI_MINOR_VERSION)/libgda-report
 gdareportinclude_HEADERS = $(gda_report_headers)
diff -Nurd -x'*~' libgda-3.1.2.orig/libgda-report/RML/Makefile.am libgda-3.1.2/libgda-report/RML/Makefile.am
--- libgda-3.1.2.orig/libgda-report/RML/Makefile.am	2007-09-17 03:25:14.000000000 -0400
+++ libgda-3.1.2/libgda-report/RML/Makefile.am	2008-02-28 13:46:24.000000000 -0500
@@ -16,8 +16,9 @@
 	$(gda_report_headers)	\
 	gda-report-rml-document.c	
 
-libgda_report_rml_3_0_la_LIBADD = $(LIBGDA_LIBS) \
-	$(top_builddir)/libgda/libgda-3.0.la
+libgda_report_rml_3_0_la_LIBADD = \
+	$(top_builddir)/libgda/libgda-3.0.la \
+	$(LIBGDA_LIBS)
 
 gdareportincludedir=$(includedir)/libgda-$(GDA_ABI_MAJOR_VERSION).$(GDA_ABI_MINOR_VERSION)/libgda-report
 gdareportinclude_HEADERS=$(gda_report_headers)
diff -Nurd -x'*~' libgda-3.1.2.orig/libgda-report/RML/Makefile.in libgda-3.1.2/libgda-report/RML/Makefile.in
--- libgda-3.1.2.orig/libgda-report/RML/Makefile.in	2007-10-24 16:17:53.000000000 -0400
+++ libgda-3.1.2/libgda-report/RML/Makefile.in	2008-02-28 13:47:02.000000000 -0500
@@ -375,8 +375,9 @@
 	$(gda_report_headers)	\
 	gda-report-rml-document.c	
 
-libgda_report_rml_3_0_la_LIBADD = $(LIBGDA_LIBS) \
-	$(top_builddir)/libgda/libgda-3.0.la
+libgda_report_rml_3_0_la_LIBADD = \
+	$(top_builddir)/libgda/libgda-3.0.la \
+	$(LIBGDA_LIBS)
 
 gdareportincludedir = $(includedir)/libgda-$(GDA_ABI_MAJOR_VERSION).$(GDA_ABI_MINOR_VERSION)/libgda-report
 gdareportinclude_HEADERS = $(gda_report_headers)
diff -Nurd -x'*~' libgda-3.1.2.orig/libgda-report/engine/Makefile.in libgda-3.1.2/libgda-report/engine/Makefile.in
--- libgda-3.1.2.orig/libgda-report/engine/Makefile.in	2007-10-24 16:17:54.000000000 -0400
+++ libgda-3.1.2/libgda-report/engine/Makefile.in	2008-02-28 13:47:17.000000000 -0500
@@ -368,8 +368,9 @@
 	$(gda_report_headers)	\
 	gda-report-engine.c	
 
-libgda_report_engine_3_0_la_LIBADD = $(LIBGDA_LIBS) \
-	$(top_builddir)/libgda/libgda-3.0.la 
+libgda_report_engine_3_0_la_LIBADD = \
+	$(top_builddir)/libgda/libgda-3.0.la \
+	$(LIBGDA_LIBS)
 
 gdareportincludedir = $(includedir)/libgda-$(GDA_ABI_MAJOR_VERSION).$(GDA_ABI_MINOR_VERSION)/libgda-report
 gdareportinclude_HEADERS = $(gda_report_headers)
diff -Nurd -x'*~' libgda-3.1.2.orig/libgda-xslt/Makefile.in libgda-3.1.2/libgda-xslt/Makefile.in
--- libgda-3.1.2.orig/libgda-xslt/Makefile.in	2007-10-24 16:17:54.000000000 -0400
+++ libgda-3.1.2/libgda-xslt/Makefile.in	2008-02-28 13:45:14.000000000 -0500
@@ -372,8 +372,9 @@
 	sql_exslt_internal.c \
 	sql_backend.c
 
-libgda_xslt_3_0_la_LIBADD = $(LIBGDA_LIBS) $(LIBXSLT_LIBS)\
-        $(top_builddir)/libgda/libgda-3.0.la
+libgda_xslt_3_0_la_LIBADD = \
+        $(top_builddir)/libgda/libgda-3.0.la \
+        $(LIBGDA_LIBS) $(LIBXSLT_LIBS)
 
 gdaxsltincludedir = $(includedir)/libgda-$(GDA_ABI_MAJOR_VERSION).$(GDA_ABI_MINOR_VERSION)/libgda-xslt
 gdaxsltinclude_HEADERS = $(gda_xslt_headers)
