diff -Nurd -x'*~' Module-Signature-0.64.orig/Makefile.PL Module-Signature-0.64/Makefile.PL
--- Module-Signature-0.64.orig/Makefile.PL	2010-05-08 18:41:13.000000000 -0400
+++ Module-Signature-0.64/Makefile.PL	2010-06-03 21:49:35.000000000 -0400
@@ -61,12 +61,7 @@
 };
 
 # Is openpgp currently installed
-if ( can_use('Crypt::OpenPGP') ) {
-	# If OpenPGP is already installed, so relist all the
-	# dependencies so they will upgrade as needed.
-	requires( @OPEN_PGP );
-
-} elsif ( locate_gpg() ) {
+if      ( locate_gpg() ) {
 	# We SHOULD have gpg, double-check formally
 	requires_external_bin 'gpg';
 
@@ -113,11 +108,6 @@
 
 	return 1 if grep { /^--installdeps/} @ARGV;
 
-	if ( prompt("Import PAUSE and author keys to GnuPG?", 'y' ) =~ /^y/i) {
-		print 'Importing... ';
-		system 'gpg', '--quiet', '--import', glob('*.pub');
-		print "done.\n";
-	}
 
 	return 1;
 }
