
--- monodevelop-0.15/monodevelop.in	2007-07-31 13:56:11.000000000 -0400
+++ monodevelop-0.15-new/monodevelop.in	2007-09-20 22:25:24.000000000 -0400
@@ -36,13 +36,17 @@
 
 MD_BIN_PATH=@prefix@/lib/monodevelop/bin
 
-if [ -n $LD_LIBRARY_PATH ]; then
-	export LD_LIBRARY_PATH=$MOZILLA_HOME:$LD_LIBRARY_PATH
+if [ -n $DYLD_LIBRARY_PATH ]; then
+	export DYLD_LIBRARY_PATH=$MOZILLA_HOME:$DYLD_LIBRARY_PATH
 else
-	export LD_LIBRARY_PATH=$MOZILLA_HOME
+	export DYLD_LIBRARY_PATH=$MOZILLA_HOME
 fi
 
-MD_PKG_CONFIG_PATH=@gtksharp_prefix@/lib/pkgconfig/:/usr/lib/pkgconfig/:/usr/local/lib/pkgconfig/:/usr/share/pkgconfig/:/usr/local/share/pkgconfig/
+for dir in fontconfig2 freetyp219; do
+	DYLD_LIBRARY_PATH="@FINKPREFIX@/lib/$dir/lib:$DYLD_LIBRARY_PATH"
+done
+
+MD_PKG_CONFIG_PATH=@FINKPREFIX@/lib/fontconfig2/lib/pkgconfig/:@FINKPREFIX@/lib/freetype219/lib/pkgconfig/:@FINKPREFIX@/lib/pango-ft219/lib/pkgconfig/@gtksharp_prefix@/lib/pkgconfig/:/usr/lib/pkgconfig/:/usr/local/lib/pkgconfig/:/usr/share/pkgconfig/:/usr/local/share/pkgconfig/
 
 if [ -n $PKG_CONFIG_PATH ]; then
 	export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$MD_PKG_CONFIG_PATH
