diff -Nurd monodevelop-2.0/monodevelop.in monodevelop-2.0-new/monodevelop.in
--- monodevelop-2.0/monodevelop.in	2009-03-19 19:43:56.000000000 -0400
+++ monodevelop-2.0-new/monodevelop.in	2009-04-07 12:50:46.000000000 -0400
@@ -13,7 +13,7 @@
 #############################
 
 #(NOTE: any changes to this block should be kept in sync with the one in configure.in)
-MOZFILE="libgtkembedmoz.so"
+MOZFILE="libgtkembedmoz.dylib"
 find_libgtkembedmoz ()
 {
 	MOZDIR=$MOZILLA_FIVE_HOME
@@ -48,10 +48,10 @@
 	fi
 fi
 
-if [ -n $LD_LIBRARY_PATH ]; then
-	export LD_LIBRARY_PATH="$MOZILLA_HOME:$LD_LIBRARY_PATH"
+if [ -n $DYLD_FALLBACK_LIBRARY_PATH ]; then
+	export DYLD_FALLBACK_LIBRARY_PATH=$MOZILLA_HOME:$DYLD_FALLBACK_LIBRARY_PATH
 else
-	export LD_LIBRARY_PATH="$MOZILLA_HOME"
+	export DYLD_FALLBACK_LIBRARY_PATH=$MOZILLA_HOME
 fi
 
 export MOZILLA_FIVE_HOME="$MOZILLA_HOME"
@@ -60,6 +60,8 @@
 ## END MOZILLA DETECTION ##
 ###########################
 
+export PKG_CONFIG_PATH="@FINKPREFIX@/lib/fontconfig2/lib/pkgconfig/:@FINKPREFIX@/lib/freetype219/lib/pkgconfig/:@FINKPREFIX@/lib/pango-ft219/lib/pkgconfig/:$PKG_CONFIG_PATH"
+
 for arg in "$@"; do 
 case "x$arg" in
 	x--profile*)
