diff -uNr seahorse-0.9.92/agent/Makefile.am seahorse-0.9.92-new/agent/Makefile.am
--- seahorse-0.9.92/agent/Makefile.am	2006-12-10 10:49:27.000000000 -0500
+++ seahorse-0.9.92-new/agent/Makefile.am	2007-03-12 10:49:50.000000000 -0400
@@ -28,7 +28,7 @@
 						 seahorse-agent-io.c seahorse-agent-status.c $(SSH_SRCS)
 
 seahorse_agent_LDADD = \
-    $(top_builddir)/libseahorse/libseahorse.a \
+    $(top_builddir)/libseahorse/libseahorse.la \
     $(top_builddir)/libcryptui/libcryptui.la \
     $(SEAHORSE_LIBS)
 
diff -uNr seahorse-0.9.92/daemon/Makefile.am seahorse-0.9.92-new/daemon/Makefile.am
--- seahorse-0.9.92/daemon/Makefile.am	2006-12-10 10:39:36.000000000 -0500
+++ seahorse-0.9.92-new/daemon/Makefile.am	2007-03-12 10:49:54.000000000 -0400
@@ -34,7 +34,7 @@
     $(SHARING_SRCS)
 
 seahorse_daemon_LDADD = \
-    $(top_builddir)/libseahorse/libseahorse.a \
+    $(top_builddir)/libseahorse/libseahorse.la \
     $(top_builddir)/libcryptui/libcryptui.la \
     $(SEAHORSE_LIBS)
 
diff -uNr seahorse-0.9.92/libseahorse/Makefile.am seahorse-0.9.92-new/libseahorse/Makefile.am
--- seahorse-0.9.92/libseahorse/Makefile.am	2006-12-21 16:34:20.000000000 -0500
+++ seahorse-0.9.92-new/libseahorse/Makefile.am	2007-03-12 10:49:30.000000000 -0400
@@ -15,7 +15,7 @@
 	-DEXECDIR=\""$(seahorselibexecbindir)"\" \
 	-DLIBCRYPTUI_API_SUBJECT_TO_CHANGE 
 
-noinst_LIBRARIES = libseahorse.a
+noinst_LTLIBRARIES = libseahorse.la
 
 BUILT_SOURCES = \
 	seahorse-marshal.c seahorse-marshal.h
@@ -54,7 +54,7 @@
 SSH_SRCS = 
 endif
 
-libseahorse_a_SOURCES = \
+libseahorse_la_SOURCES = \
 	seahorse-key.c seahorse-key.h \
 	seahorse-key-source.c seahorse-key-source.h \
 	seahorse-pgp-key.c seahorse-pgp-key.h \
@@ -102,7 +102,7 @@
 
 seahorse_ssh_askpass_SOURCES = seahorse-ssh-askpass.c
 seahorse_ssh_askpass_LDADD = \
-	$(top_builddir)/libseahorse/libseahorse.a \
+	$(top_builddir)/libseahorse/libseahorse.la \
 	$(top_builddir)/libcryptui/libcryptui.la \
 	$(SEAHORSE_LIBS)
 
diff -uNr seahorse-0.9.92/plugins/applet/Makefile.am seahorse-0.9.92-new/plugins/applet/Makefile.am
--- seahorse-0.9.92/plugins/applet/Makefile.am	2006-10-31 12:26:59.000000000 -0500
+++ seahorse-0.9.92-new/plugins/applet/Makefile.am	2007-03-12 10:49:58.000000000 -0400
@@ -32,7 +32,7 @@
 
 seahorse_applet_LDADD = \
 	$(GNOME_APPLETS_LIBS) \
-	$(top_builddir)/libseahorse/libseahorse.a \
+	$(top_builddir)/libseahorse/libseahorse.la \
 	$(top_builddir)/libcryptui/libcryptui.la \
 	$(SEAHORSE_LIBS) \
 	$(LIBCRYPTUI_LIBS)
diff -uNr seahorse-0.9.92/plugins/nautilus/Makefile.am seahorse-0.9.92-new/plugins/nautilus/Makefile.am
--- seahorse-0.9.92/plugins/nautilus/Makefile.am	2007-02-20 19:10:41.000000000 -0500
+++ seahorse-0.9.92-new/plugins/nautilus/Makefile.am	2007-03-12 10:50:03.000000000 -0400
@@ -16,7 +16,7 @@
 seahorse_preferences_SOURCES = seahorse-pgp-preferences.c 
 
 seahorse_preferences_LDADD = \
-    $(top_builddir)/libseahorse/libseahorse.a \
+    $(top_builddir)/libseahorse/libseahorse.la \
     $(top_builddir)/libcryptui/libcryptui.la \
     $(SEAHORSE_LIBS)
     
@@ -27,7 +27,7 @@
     seahorse-tool-progress.c
 
 seahorse_tool_LDADD = \
-    $(top_builddir)/libseahorse/libseahorse.a \
+    $(top_builddir)/libseahorse/libseahorse.la \
     $(top_builddir)/libcryptui/libcryptui.la \
     $(SEAHORSE_LIBS) $(LIBCRYPTUI_LIBS)
 
diff -uNr seahorse-0.9.92/src/Makefile.am seahorse-0.9.92-new/src/Makefile.am
--- seahorse-0.9.92/src/Makefile.am	2006-09-16 13:35:02.000000000 -0400
+++ seahorse-0.9.92-new/src/Makefile.am	2007-03-12 10:50:07.000000000 -0400
@@ -58,7 +58,7 @@
     $(SSH_SRCS)
 
 seahorse_LDADD = \
-	$(top_builddir)/libseahorse/libseahorse.a \
+	$(top_builddir)/libseahorse/libseahorse.la \
 	$(top_builddir)/libcryptui/libcryptui.la \
 	$(SEAHORSE_LIBS)
 
