diff -Naur silc-toolkit-0.9.12/Makefile.in silc-toolkit-0.9.12-patched/Makefile.in
--- silc-toolkit-0.9.12/Makefile.in	Sun Feb 29 15:10:00 2004
+++ silc-toolkit-0.9.12-patched/Makefile.in	Tue Aug 31 14:57:16 2004
@@ -770,7 +770,6 @@
 	-rm -rf $(distdir)
 
 install-dirs:
-	-mkdir -p $(etcdir)
 	-mkdir -p $(modulesdir)
 	-mkdir -p $(docdir)
 
@@ -833,7 +832,7 @@
 	-mkdir -p $(docdir)/toolkit/
 	-$(INSTALL_DATA) $(srcdir)/doc/toolkit/* $(docdir)/toolkit
 	-$(INSTALL_DATA) $(srcdir)/lib/doc/*.gif $(docdir)/toolkit
-	-cp -R $(srcdir)/tutorial $(prefix)
+	-cp -R $(srcdir)/tutorial $(DESTDIR)$(datadir)/
 
 examples-install:
 	-mkdir -p $(docdir)/examples/
@@ -846,7 +845,7 @@
 	fi
 
 @SILC_DIST_CLIENT_TRUE@install-data-hook: install-dirs-client sim-install doc-install-client etc-install-client
-@SILC_DIST_CLIENT_FALSE@@SILC_DIST_TOOLKIT_TRUE@install-data-hook: install-dirs-client install-dirs-server sim-install doc-install-client doc-install-server toolkit-install examples-install etc-install-client etc-install-server
+@SILC_DIST_CLIENT_FALSE@@SILC_DIST_TOOLKIT_TRUE@install-data-hook: install-dirs sim-install toolkit-install examples-install
 @SILC_DIST_CLIENT_FALSE@@SILC_DIST_TOOLKIT_FALSE@install-data-hook: install-dirs-server sim-install doc-install-server examples-install etc-install-server generate-server-key
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
diff -Naur silc-toolkit-0.9.12/lib/silcsim/Makefile.in silc-toolkit-0.9.12-patched/lib/silcsim/Makefile.in
--- silc-toolkit-0.9.12/lib/silcsim/Makefile.in	Sun Feb 29 15:10:02 2004
+++ silc-toolkit-0.9.12-patched/lib/silcsim/Makefile.in	Tue Aug 31 14:58:58 2004
@@ -593,7 +593,7 @@
 
 $(SIM_CIPHER_OBJS):
 	@if test '!' -f lib$*.la ; then \
-	  $(LIBTOOL) --mode=link $(CCLD) -rpath $(silc_modulesdir) \
+	  $(LIBTOOL) --mode=link $(CCLD) -rpath $(silc_modulesdir) -module \
 		../silccrypt/$*.lo -o lib$*.la $(LTFLAGS); \
 	  cd $(srcdir) && $(LN_S) -f $(srcdir)/.libs/lib$*.so \
 		$(srcdir)/$*.sim.so; \
@@ -601,7 +601,7 @@
 
 $(SIM_HASH_OBJS):
 	@if test '!' -f lib$*.la ; then \
-	  $(LIBTOOL) --mode=link $(CCLD) -rpath $(silc_modulesdir) \
+	  $(LIBTOOL) --mode=link $(CCLD) -rpath $(silc_modulesdir) -module \
 		../silccrypt/$*.lo -o lib$*.la $(LTFLAGS); \
 	  cd $(srcdir) && $(LN_S) -f $(srcdir)/.libs/lib$*.so \
 		$(srcdir)/$*.sim.so; \
