diff -ru sitecopy-0.16.3/configure sitecopy-0.16.3-patched/configure
--- sitecopy-0.16.3/configure	2006-03-12 11:09:54.000000000 +0100
+++ sitecopy-0.16.3-patched/configure	2006-11-19 00:02:09.000000000 +0100
@@ -13576,7 +13576,7 @@
 
 
     CFLAGS="$CFLAGS -I$neon_bundled_srcdir"
-    NEON_LIBS="-L$neon_bundled_builddir -lneon $NEON_LIBS"
+    NEON_LIBS="$neon_bundled_builddir/libneon.a $NEON_LIBS"
     NEON_NEED_XML_PARSER=yes
     neon_library_message="included libneon (${NEON_VERSION})"
 else
