diff -Naur ximian-connector-1.5.90-old/lib/e2k-autoconfig.c ximian-connector-1.5.90/lib/e2k-autoconfig.c
--- ximian-connector-1.5.90-old/lib/e2k-autoconfig.c	2004-07-07 12:34:34.000000000 -0400
+++ ximian-connector-1.5.90/lib/e2k-autoconfig.c	2004-07-07 12:36:32.000000000 -0400
@@ -31,8 +31,10 @@
 #include <sys/stat.h>
 #include <unistd.h>
 
+#include <sys/types.h>
 #include <netinet/in.h>
-#include <arpa/nameser.h>
+#include <sys/socket.h>
+#include <arpa/nameser_compat.h>
 #include <resolv.h>
 
 #include "e2k-autoconfig.h"
