diff -ruN jakarta-ant-1.4/ant.csh jakarta-ant-1.4-patched/ant.csh
--- jakarta-ant-1.4/ant.csh	Thu Jan  1 01:00:00 1970
+++ jakarta-ant-1.4-patched/ant.csh	Mon Oct  8 22:57:44 2001
@@ -0,0 +1,2 @@
+setenv JAVA_HOME /Library/Java/Home
+setenv ANT_HOME @PREFIX@/lib/ant
diff -ruN jakarta-ant-1.4/ant.sh jakarta-ant-1.4-patched/ant.sh
--- jakarta-ant-1.4/ant.sh	Thu Jan  1 01:00:00 1970
+++ jakarta-ant-1.4-patched/ant.sh	Mon Oct  8 22:58:08 2001
@@ -0,0 +1,4 @@
+JAVA_HOME=/Library/Java/Home
+export JAVA_HOME
+ANT_HOME=@PREFIX@/lib/ant
+export ANT_HOME
diff -ruN jakarta-ant-1.4/build.sh jakarta-ant-1.4-patched/build.sh
--- jakarta-ant-1.4/build.sh	Mon Sep  3 12:48:23 2001
+++ jakarta-ant-1.4-patched/build.sh	Sun Oct 14 12:30:02 2001
@@ -3,6 +3,8 @@
 REALANTHOME=$ANT_HOME
 ANT_HOME=bootstrap
 export ANT_HOME
+JAVA_HOME=/Library/Java/Home
+export JAVA_HOME
 
 if test ! -f bootstrap/lib/ant.jar -o  ! -x bootstrap/bin/ant -o ! -x bootstrap/bin/antRun ; then
   /bin/sh ./bootstrap.sh
diff -ruN jakarta-ant-1.4/build.xml jakarta-ant-1.4-patched/build.xml
--- jakarta-ant-1.4/build.xml	Mon Sep  3 12:48:23 2001
+++ jakarta-ant-1.4-patched/build.xml	Sun Oct 14 12:31:53 2001
@@ -329,7 +329,11 @@
     <fixcrlf srcdir="${dist.bin}" includes="runant.pl" />
 
     <chmod perm="ugo+rx" dir="${dist.dir}" type="dir" includes="**" />
-    <chmod perm="ugo+r" dir="${dist.dir}" type="file" includes="**" />
+    <!--
+	Commented because this cause an ArgumentTooLong error
+	It's done in the install script
+	<chmod perm="ugo+r" dir="${dist.dir}" type="file" includes="**" />
+    -->
     <chmod perm="ugo+x" type="file">
        <fileset dir="${dist.bin}">
          <include name="**/ant" />
