diff -Naur findbugs-0.6.1/bin/findbugs findbugs-0.6.1.fink/bin/findbugs
--- findbugs-0.6.1/bin/findbugs	Tue Jul 22 06:39:39 2003
+++ findbugs-0.6.1.fink/bin/findbugs	Tue Jul 22 14:57:16 2003
@@ -65,34 +65,38 @@
 
 # Find findbugs_home.
 # This code shamelessly stolen from Ant's "bin/ant" script.
+## (Ant code not required for fink, since we know where everything is
+## installed - RL
 
-## resolve links - $0 may be a link to ant's home
-PRG="$0"
-progname=`basename "$0"`
-saveddir=`pwd`
-
-# need this for relative symlinks
-dirname_prg=`dirname "$PRG"`
-cd "$dirname_prg"
+# ## resolve links - $0 may be a link to ant's home
+# PRG="$0"
+# progname=`basename "$0"`
+# saveddir=`pwd`
+
+# # need this for relative symlinks
+# dirname_prg=`dirname "$PRG"`
+# cd "$dirname_prg"
   
-while [ -h "$PRG" ] ; do
-	ls=`ls -ld "$PRG"`
-	link=`expr "$ls" : '.*-> \(.*\)$'`
-	if expr "$link" : '/.*' > /dev/null; then
-		PRG="$link"
-	else
-		PRG=`dirname "$PRG"`"/$link"
-	fi
-done
+# while [ -h "$PRG" ] ; do
+# 	ls=`ls -ld "$PRG"`
+# 	link=`expr "$ls" : '.*-> \(.*\)$'`
+# 	if expr "$link" : '/.*' > /dev/null; then
+# 		PRG="$link"
+# 	else
+# 		PRG=`dirname "$PRG"`"/$link"
+# 	fi
+# done
   
-findbugs_home=`dirname "$PRG"`/..
+# findbugs_home=`dirname "$PRG"`/..
+
+# cd "$saveddir"
 
-cd "$saveddir"
+# # make it fully qualified
+# findbugs_home=`cd "$findbugs_home" && pwd`
 
-# make it fully qualified
-findbugs_home=`cd "$findbugs_home" && pwd`
+# # End of code shamelessly stolen from Ant.
 
-# End of code shamelessly stolen from Ant.
+findbugs_home=@PREFIX@/share/java/findbugs
 
 if [ ! -d "$findbugs_home" ]; then
 	echo "The path $findbugs_home,"
@@ -149,6 +153,6 @@
 fi
 
 # Launch FindBugs!
-java $jvmArgs "-Dfindbugs.home=$findbugs_home" -jar "$findbugs_home/lib/$appJar" $argList
+exec java $jvmArgs "-Dfindbugs.home=$findbugs_home" -jar "$findbugs_home/lib/$appJar" $argList
 
 # vim:ts=4
diff -Naur findbugs-0.6.1/bin/mergeResults findbugs-0.6.1.fink/bin/mergeResults
--- findbugs-0.6.1/bin/mergeResults	Tue Jul 22 06:39:39 2003
+++ findbugs-0.6.1.fink/bin/mergeResults	Tue Jul 22 14:57:06 2003
@@ -57,33 +57,37 @@
 
 # Find findbugs_home.
 # This code shamelessly stolen from Ant's "bin/ant" script.
+## (Ant code not required for fink, since we know where everything is
+## installed - RL
 
-## resolve links - $0 may be a link to ant's home
-PRG="$0"
-progname=`basename "$0"`
-saveddir=`pwd`
-
-# need this for relative symlinks
-dirname_prg=`dirname "$PRG"`
-cd "$dirname_prg"
+# ## resolve links - $0 may be a link to ant's home
+# PRG="$0"
+# progname=`basename "$0"`
+# saveddir=`pwd`
+
+# # need this for relative symlinks
+# dirname_prg=`dirname "$PRG"`
+# cd "$dirname_prg"
   
-while [ -h "$PRG" ] ; do
-	ls=`ls -ld "$PRG"`
-	link=`expr "$ls" : '.*-> \(.*\)$'`
-	if expr "$link" : '/.*' > /dev/null; then
-		PRG="$link"
-	else
-		PRG=`dirname "$PRG"`"/$link"
-	fi
-done
+# while [ -h "$PRG" ] ; do
+# 	ls=`ls -ld "$PRG"`
+# 	link=`expr "$ls" : '.*-> \(.*\)$'`
+# 	if expr "$link" : '/.*' > /dev/null; then
+# 		PRG="$link"
+# 	else
+# 		PRG=`dirname "$PRG"`"/$link"
+# 	fi
+# done
   
-findbugs_home=`dirname "$PRG"`/..
+# findbugs_home=`dirname "$PRG"`/..
 
-cd "$saveddir"
+# cd "$saveddir"
 
-# make it fully qualified
-findbugs_home=`cd "$findbugs_home" && pwd`
+# # make it fully qualified
+# findbugs_home=`cd "$findbugs_home" && pwd`
 
-# End of code shamelessly stolen from Ant.
+# # End of code shamelessly stolen from Ant.
 
-java -Dfindbugs.home=$findbugs_home -classpath $findbugs_home/findbugs.jar:$CLASSPATH edu.umd.cs.findbugs.MergeResults $@
+findbugs_home=@PREFIX@/share/java/findbugs
+
+exec java -Dfindbugs.home=$findbugs_home -classpath $findbugs_home/findbugs.jar:$CLASSPATH edu.umd.cs.findbugs.MergeResults $@
