diff -Naur findbugs-0.6.3/bin/findbugs findbugs-0.6.3.fink/bin/findbugs
--- findbugs-0.6.3/bin/findbugs	Wed Aug  6 04:38:40 2003
+++ findbugs-0.6.3.fink/bin/findbugs	Wed Aug  6 14:30:58 2003
@@ -85,8 +85,13 @@
 		PRG=`dirname "$PRG"`"/$link"
 	fi
 done
-  
-findbugs_home=`dirname "$PRG"`/..
+
+# handle FHS compliant installations (eg fink)
+if [ -d `dirname "$PRG"`/../share/java/findbugs ]; then
+    findbugs_home=`dirname "$PRG"`/../share/java/findbugs
+else
+    findbugs_home=`dirname "$PRG"`/..
+fi
 
 cd "$saveddir"
 
@@ -161,6 +166,6 @@
 fi
 
 # Launch FindBugs!
-$jvmCmd $jvmArgs $debugArg "-Dfindbugs.home=$findbugs_home" -jar "$findbugs_home/lib/$appJar" $argList
+exec $jvmCmd $jvmArgs $debugArg "-Dfindbugs.home=$findbugs_home" -jar "$findbugs_home/lib/$appJar" $argList
 
 # vim:ts=4
diff -Naur findbugs-0.6.3/bin/mergeResults findbugs-0.6.3.fink/bin/mergeResults
--- findbugs-0.6.3/bin/mergeResults	Wed Aug  6 04:38:40 2003
+++ findbugs-0.6.3.fink/bin/mergeResults	Wed Aug  6 14:31:47 2003
@@ -77,7 +77,12 @@
 	fi
 done
   
-findbugs_home=`dirname "$PRG"`/..
+# handle FHS compliant installations (eg fink)
+if [ -d `dirname "$PRG"`/../share/java/findbugs ]; then
+    findbugs_home=`dirname "$PRG"`/../share/java/findbugs
+else
+    findbugs_home=`dirname "$PRG"`/..
+fi
 
 cd "$saveddir"
 
@@ -86,4 +91,6 @@
 
 # End of code shamelessly stolen from Ant.
 
-java -Dfindbugs.home=$findbugs_home -classpath $findbugs_home/findbugs.jar:$CLASSPATH edu.umd.cs.findbugs.MergeResults $@
+exec java -Dfindbugs.home=$findbugs_home \
+          -classpath $findbugs_home/findbugs.jar:$CLASSPATH \
+          edu.umd.cs.findbugs.MergeResults $@
