diff -ruN gengetopt-2.10.orig/src/Makefile.in gengetopt-2.10/src/Makefile.in
--- gengetopt-2.10.orig/src/Makefile.in	Thu May 22 13:17:10 2003
+++ gengetopt-2.10/src/Makefile.in	Sun Jun  8 16:14:41 2003
@@ -166,7 +166,7 @@
 
 gengetopt_SOURCES = parser.yy scanner.ll \
                     argsdef.c cmdline.c gengetopt.cc gm.cc yyerror.cc \
-                    conf_parser_gen.cc
+                    conf_parser_gen.cc getopt.c getopt1.c
 
 
 #confparsergen.cc
@@ -216,13 +216,15 @@
 
 am_gengetopt_OBJECTS = parser.$(OBJEXT) scanner.$(OBJEXT) \
 	argsdef.$(OBJEXT) cmdline.$(OBJEXT) gengetopt.$(OBJEXT) \
-	gm.$(OBJEXT) yyerror.$(OBJEXT) conf_parser_gen.$(OBJEXT)
+	gm.$(OBJEXT) yyerror.$(OBJEXT) conf_parser_gen.$(OBJEXT) \
+	getopt.$(OBJEXT) getopt1.$(OBJEXT)
 gengetopt_OBJECTS = $(am_gengetopt_OBJECTS)
 gengetopt_DEPENDENCIES = @LIBOBJS@ skels/libgen.a
 gengetopt_LDFLAGS =
 am__objects_1 = parser.$(OBJEXT) scanner.$(OBJEXT) argsdef.$(OBJEXT) \
 	cmdline.$(OBJEXT) gengetopt.$(OBJEXT) gm.$(OBJEXT) \
-	yyerror.$(OBJEXT) conf_parser_gen.$(OBJEXT)
+	yyerror.$(OBJEXT) conf_parser_gen.$(OBJEXT) getopt.$(OBJEXT) \
+	getopt1.$(OBJEXT)
 am_gengetopt_efence_OBJECTS = $(am__objects_1)
 gengetopt_efence_OBJECTS = $(am_gengetopt_efence_OBJECTS)
 gengetopt_efence_DEPENDENCIES = @LIBOBJS@ skels/libgen.a
@@ -237,7 +239,8 @@
 @AMDEP_TRUE@	./$(DEPDIR)/conf_parser_gen.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/gengetopt.Po ./$(DEPDIR)/gm.Po \
 @AMDEP_TRUE@	./$(DEPDIR)/parser.Po ./$(DEPDIR)/scanner.Po \
-@AMDEP_TRUE@	./$(DEPDIR)/yyerror.Po
+@AMDEP_TRUE@	./$(DEPDIR)/yyerror.Po \
+@AMDEP_TRUE@	./$(DEPDIR)/getopt.Po ./$(DEPDIR)/getopt1.Po
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
@@ -322,6 +325,8 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/parser.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/scanner.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/yyerror.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/getopt.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/getopt1.Po@am__quote@
 
 distclean-depend:
 	-rm -rf $(DEPDIR) ./$(DEPDIR)
diff -ruN gengetopt-2.10.orig/src/cmdline.c gengetopt-2.10/src/cmdline.c
--- gengetopt-2.10.orig/src/cmdline.c	Mon May 12 13:01:13 2003
+++ gengetopt-2.10/src/cmdline.c	Sun Jun  8 16:07:04 2003
@@ -18,8 +18,11 @@
 #include "config.h"
 #endif
 
+#if defined(__APPLE__) && defined(__GNUC__)
+#include "gnugetopt.h"
+#else
 #include "getopt.h"
-
+#endif
 #include "cmdline.h"
 
 void
diff -ruN gengetopt-2.10.orig/src/getopt.c gengetopt-2.10/src/getopt.c
--- gengetopt-2.10.orig/src/getopt.c	Sun Dec 16 07:25:45 2001
+++ gengetopt-2.10/src/getopt.c	Sun Jun  8 16:18:21 2003
@@ -102,7 +102,11 @@
    GNU application programs can use a third alternative mode in which
    they can distinguish the relative order of options and other arguments.  */
 
+#if defined(__APPLE__) && defined(__GNUC__)
+#include "gnugetopt.h"
+#else
 #include "getopt.h"
+#endif
 
 /* For communication from `getopt' to the caller.
    When `getopt' finds an option that takes an argument,
diff -ruN gengetopt-2.10.orig/src/getopt1.c gengetopt-2.10/src/getopt1.c
--- gengetopt-2.10.orig/src/getopt1.c	Sun Dec 16 07:25:45 2001
+++ gengetopt-2.10/src/getopt1.c	Sun Jun  8 16:18:31 2003
@@ -22,7 +22,11 @@
 #include <config.h>
 #endif
 
+#if defined(__APPLE__) && defined(__GNUC__)  
+#include "gnugetopt.h" 
+#else 
 #include "getopt.h"
+#endif
 
 #if !defined __STDC__ || !__STDC__
 /* This is a separate conditional since some stdc systems
