diff -Naur pkgconfig-0.15.0.old/main.c pkgconfig-0.15.0.new/main.c
--- pkgconfig-0.15.0.old/main.c	Wed Jan 15 14:01:00 2003
+++ pkgconfig-0.15.0.new/main.c	Mon Feb 10 14:34:05 2003
@@ -280,6 +280,10 @@
       g_strfreev (search_dirs);
     }
 
+    /* Add /usr/X11R6/lib/pkgconfig to search path
+     */
+    add_search_dir ("/usr/X11R6/lib/pkgconfig");
+
 #ifdef G_OS_WIN32
   {
     /* Add search directories from the Registry */
