diff -u -r prcs-1.3.1/src/include/prcserror.h prcs-1.3.1-patched/src/include/prcserror.h
--- prcs-1.3.1/src/include/prcserror.h	Sat Oct 30 21:26:02 1999
+++ prcs-1.3.1-patched/src/include/prcserror.h	Sun Mar 17 08:22:20 2002
@@ -27,12 +27,17 @@
 #include <iostream.h>
 #include <iomanip.h>
 #if defined(__GNUG__)
-#include <stdiostream.h>
-#include <strstream.h>
+# if defined(__APPLE__)
+#  include <streambuf.h>
+   typedef filebuf stdiobuf;
+# else
+#  include <stdiostream.h>
+# endif /* if defined(__APPLE__) */
+# include <strstream.h>
 #else
-#include <fstream.h>
-typedef filebuf stdiobuf;
-#include "be-strstream.h"
+# include <fstream.h>
+  typedef filebuf stdiobuf;
+# include "be-strstream.h"
 #endif
 
 
diff -u -r prcs-1.3.1/src/prcserror.cc prcs-1.3.1-patched/src/prcserror.cc
--- prcs-1.3.1/src/prcserror.cc	Sun Feb  3 13:39:38 2002
+++ prcs-1.3.1-patched/src/prcserror.cc	Sun Mar 17 09:44:20 2002
@@ -48,8 +48,13 @@
 int return_if_fail_if_ne_val;
 #endif
 
-stdiobuf stdout_stream(stdout);
-stdiobuf stderr_stream(stderr);
+#if defined(__APPLE__)
+ stdiobuf stdout_stream(STDOUT_FILENO);
+ stdiobuf stderr_stream(STDERR_FILENO);
+#else
+ stdiobuf stdout_stream(stdout);
+ stdiobuf stderr_stream(stderr);
+#endif /* if defined(__APPLE__) */
 strstreambuf query_stream;
 
 static PrettyStreambuf pretty_stdout_stream(&stdout_stream, NULL);
