diff -ur --exclude=CVS --new-file sdcc.CVS/as/mcs51/Makefile.in sdcc/as/mcs51/Makefile.in
--- sdcc.CVS/as/mcs51/Makefile.in	Mon Oct  1 02:55:45 2001
+++ sdcc/as/mcs51/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -93,7 +93,7 @@
 dep: Makefile.dep
 
 Makefile.dep: $(ASSOURCES) $(LKSOURCES) *.h $(PRJDIR)/*.h
-	$(CPP) $(CPPFLAGS) $(M_OR_MM) $(ASSOURCES) $(LKSOURCES) >Makefile.dep
+	$(CC) $(CPPFLAGS) $(M_OR_MM) $(ASSOURCES) $(LKSOURCES) >Makefile.dep
 
 include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/as/z80/Makefile sdcc/as/z80/Makefile
--- sdcc.CVS/as/z80/Makefile	Sat Oct  6 17:44:03 2001
+++ sdcc/as/z80/Makefile	Mon Aug 12 16:40:52 2002
@@ -27,7 +27,7 @@
 dep: Makefile.dep
 
 Makefile.dep: $(SRC) *.h
-	$(CPP) $(CPPFLAGS) $(M_OR_MM) $(SRC) > Makefile.dep
+	$(CC) $(CPPFLAGS) $(M_OR_MM) $(SRC) > Makefile.dep
 
 include Makefile.dep
 
diff -ur --exclude=CVS --new-file sdcc.CVS/as/z80/Makefile.in sdcc/as/z80/Makefile.in
--- sdcc.CVS/as/z80/Makefile.in	Thu Aug 16 14:04:20 2001
+++ sdcc/as/z80/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -76,7 +76,7 @@
 dep: Makefile.dep
 
 Makefile.dep: $(ASSOURCES) *.h $(PRJDIR)/*.h
-	$(CPP) $(CPPFLAGS) $(M_OR_MM) $(ASSOURCES) >Makefile.dep
+	$(CC) $(CPPFLAGS) $(M_OR_MM) $(ASSOURCES) >Makefile.dep
 
 include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/debugger/mcs51/Makefile.in sdcc/debugger/mcs51/Makefile.in
--- sdcc.CVS/debugger/mcs51/Makefile.in	Sat Oct  6 17:44:03 2001
+++ sdcc/debugger/mcs51/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -88,7 +88,7 @@
 dep: Makefile.dep
 
 Makefile.dep: $(SOURCES) *.h $(PRJDIR)/*.h
-	$(CPP) $(CPPFLAGS) $(M_OR_MM) $(SOURCES) >Makefile.dep
+	$(CC) $(CPPFLAGS) $(M_OR_MM) $(SOURCES) >Makefile.dep
 
 include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/device/include/Makefile.in sdcc/device/include/Makefile.in
--- sdcc.CVS/device/include/Makefile.in	Mon Aug 12 09:46:25 2002
+++ sdcc/device/include/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -67,7 +67,7 @@
 dep: #Makefile.dep
 
 #Makefile.dep:
-#	$(CPP) $(CPPFLAGS) $(M_OR_MM) *.c >main.dep
+#	$(CC) $(CPPFLAGS) $(M_OR_MM) *.c >main.dep
 
 #include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/device/lib/.#Makefile.in.1.36 sdcc/device/lib/.#Makefile.in.1.36
--- sdcc.CVS/device/lib/.#Makefile.in.1.36	Wed Dec 31 18:00:00 1969
+++ sdcc/device/lib/.#Makefile.in.1.36	Mon Aug 12 16:40:52 2002
@@ -0,0 +1,247 @@
+#
+#
+#
+
+VERSION         = @VERSION@
+VERSIONHI       = @VERSIONHI@
+VERSIONLO       = @VERSIONLO@
+VERSIONP        = @VERSIONP@
+
+SHELL		= /bin/sh
+CPP		= @CPP@
+INSTALL		= @INSTALL@
+CP              = @CP@
+
+PRJDIR		= ../..
+INCDIR		= $(PRJDIR)/device/include
+
+CC		= $(PRJDIR)/bin/sdcc
+CPP		= $(PRJDIR)/bin/sdcpp
+
+prefix          = @prefix@
+exec_prefix     = @exec_prefix@
+bindir          = @bindir@
+libdir          = @libdir@
+datadir         = @datadir@
+sdcc_datadir    = @sdcc_datadir@
+includedir      = @includedir@
+mandir          = @mandir@
+man1dir         = $(mandir)/man1
+man2dir         = $(mandir)/man2
+infodir         = @infodir@
+srcdir          = @srcdir@
+
+CPPFLAGS        = -I$(INCDIR)
+CFLAGS		= $(MODELFLAGS) --nostdinc --nostdlib
+
+BUILDDIR        = build
+# Default
+PORT            = z80
+PORTDIR         = $(BUILDDIR)/$(PORT)
+
+SOURCES		= _atoi.c _atol.c _autobaud.c _bp.c _schar2fs.c \
+		  _decdptr.c _divsint.c _divslong.c _divuint.c \
+		  _divulong.c _fs2schar.c _fs2sint.c _fs2slong.c \
+		  _fs2uchar.c _fs2uint.c _fs2ulong.c _fsadd.c \
+		  _fsdiv.c _fseq.c _fsgt.c _fslt.c _fsmul.c \
+		  _fsneq.c _fssub.c _gptrget.c _gptrgetc.c _gptrput.c \
+		  _sint2fs.c _iscntrl.c _isdigit.c _isgraph.c \
+		  _islower.c _isprint.c _ispunct.c _isspace.c \
+		  _isupper.c _isxdigit.c _slong2fs.c _memcmp.c \
+		  _memcpy.c _memset.c _modsint.c _modslong.c \
+		  _moduint.c _modulong.c _mulint.c _mullong.c \
+		  _ser.c _setjmp.c \
+		  _spx.c _startup.c _strchr.c _strcmp.c _strcpy.c \
+		  _strcspn.c _strlen.c _strncat.c _strncmp.c \
+		  _strncpy.c _strpbrk.c _strrchr.c _strspn.c \
+		  _strstr.c _strtok.c _uchar2fs.c _uint2fs.c \
+		  _ulong2fs.c malloc.c serial.c ser_ir.c printfl.c \
+		  printf_large.c vprintf.c puts.c gets.c \
+		  assert.c _strcat.c time.c printf_fast.c bpx.c \
+		  fabsf.c frexpf.c ldexpf.c expf.c powf.c sincosf.c sinf.c \
+		  cosf.c logf.c log10f.c sqrtf.c tancotf.c tanf.c cotf.c \
+		  asincosf.c asinf.c acosf.c atanf.c atan2f.c sincoshf.c \
+		  sinhf.c coshf.c tanhf.c floorf.c ceilf.c modff.c
+
+OBJECTS		= $(patsubst %.c,$(PORTDIR)/%.rel,$(SOURCES))
+
+Z80SOURCES      = _atoi.c \
+		  _iscntrl.c _isdigit.c _isgraph.c \
+		  _islower.c _isprint.c _ispunct.c _isspace.c \
+		  _isupper.c _isxdigit.c _memcmp.c \
+		  _memcpy.c _memset.c \
+		  _startup.c _strchr.c _strcmp.c _strcpy.c \
+		  _strcspn.c _strlen.c _strncat.c _strncmp.c \
+		  _strncpy.c _strpbrk.c _strrchr.c _strspn.c \
+		  _strstr.c _strtok.c \
+		  puts.c gets.c \
+		  assert.c _strcat.c \
+		  _modslong.c _modulong.c \
+		  _mullong.c \
+		  _divslong.c _divulong.c \
+		  malloc.c \
+		  _fs2schar.c _fs2sint.c _fs2slong.c \
+		  _fs2uchar.c _fs2uint.c _fs2ulong.c _fsadd.c \
+		  _fsdiv.c _fseq.c _fsgt.c _fslt.c _fsmul.c \
+		  _fsneq.c _fssub.c \
+		  _uchar2fs.c _uint2fs.c \
+		  _ulong2fs.c \
+		  _slong2fs.c _sint2fs.c _schar2fs.c 
+
+Z80OBJECTS      = $(Z80SOURCES:%.c=$(PORTDIR)/%.o)
+
+XA51SOURCES      = _atoi.c _atol.c _schar2fs.c \
+		  _divsint.c _divslong.c _divuint.c \
+		  _divulong.c _fs2schar.c _fs2sint.c _fs2slong.c \
+		  _fs2uchar.c _fs2uint.c _fs2ulong.c _fsadd.c \
+		  _fsdiv.c _fseq.c _fsgt.c _fslt.c _fsmul.c \
+		  _fsneq.c _fssub.c \
+		  _sint2fs.c _iscntrl.c _isdigit.c _isgraph.c \
+		  _islower.c _isprint.c _ispunct.c _isspace.c \
+		  _isupper.c _isxdigit.c _slong2fs.c _memcmp.c \
+		  _memcpy.c _memset.c _modsint.c _modslong.c \
+		  _moduint.c _modulong.c _mulint.c _mullong.c \
+		  _strchr.c _strcmp.c _strcpy.c \
+		  _strcspn.c _strlen.c _strncat.c _strncmp.c \
+		  _strncpy.c _strpbrk.c _strrchr.c _strspn.c \
+		  _strstr.c _strtok.c _uchar2fs.c _uint2fs.c \
+		  _ulong2fs.c malloc.c puts.c gets.c \
+		  printf_large.c puts.c gets.c \
+		  assert.c _strcat.c time.c \
+		  fabsf.c frexpf.c ldexpf.c expf.c powf.c sincosf.c sinf.c \
+		  cosf.c logf.c log10f.c sqrtf.c tancotf.c tanf.c cotf.c \
+		  asincosf.c asinf.c acosf.c atanf.c atan2f.c sincoshf.c \
+		  sinhf.c coshf.c tanhf.c floorf.c ceilf.c modff.c
+
+XA51OBJECTS      = $(XA51SOURCES:%.c=$(PORTDIR)/%.rel)
+
+OEXT            = .rel
+
+include incl.mk
+
+# Compiling entire program or any subproject
+# ------------------------------------------
+all: checkconf models model-ds390 model-z80 model-gbz80
+
+objects: build-dir $(OBJECTS) port-specific-objects lib-files
+
+models:
+	if [ "`grep mcs51 ../../ports.build`" = mcs51 ]; then \
+	  for model in $(MODELS); do \
+	    $(MAKE) MODELFLAGS="--model-$$model" PORT=$$model objects; \
+	  done \
+	fi
+
+model-ds390:
+	if [ "`grep ds390 ../../ports.build`" = ds390 ]; then \
+	  $(MAKE) MODELFLAGS="-mds390" PORT=ds390 objects; \
+	fi
+
+model-xa51:
+	if [ "`grep xa51 ../../ports.build`" = xa51 ]; then \
+	  $(MAKE) MODELFLAGS="-mxa51" PORT=xa51 objects-xa51; \
+	fi
+
+objects-xa51: build-dir $(XA51OBJECTS) port-specific-objects
+	cd $(PORTDIR); ls *$(OEXT) > lib$(PORT).lib
+
+model-z80:
+	if [ "`grep z80 ../../ports.build`" = z80 ]; then \
+	  $(MAKE) MODELFLAGS="-mz80" PORT=z80 objects-z80 OEXT=.o; \
+	fi
+
+model-gbz80:
+	if [ "`grep z80 ../../ports.build`" = z80 ]; then \
+	  $(MAKE) MODELFLAGS="-mgbz80" PORT=gbz80 objects-z80 OEXT=.o; \
+	fi
+
+objects-z80: build-dir $(Z80OBJECTS) port-specific-objects
+	cd $(PORTDIR); ls *$(OEXT) > $(PORT).lib
+
+build-dir: $(PORTDIR)
+
+$(PORTDIR):
+	mkdir -p $(PORTDIR)
+
+port-specific-objects:
+	if [ -d $(PORT) ]; then \
+	  $(MAKE) -C $(PORT); \
+	  cp -f $(PORT)/*.lib $(PORT)/*$(OEXT) $(PORTDIR); \
+	fi
+
+lib-files:
+	cp *.lib $(PORTDIR)
+
+# Compiling and installing everything and runing test
+# ---------------------------------------------------
+install: all installdirs
+	cp -r $(BUILDDIR)/* $(sdcc_datadir)/lib
+	cp -r ds390 gbz80 z80 *.c $(sdcc_datadir)/lib/src
+	rm `find $(sdcc_datadir)/lib/src -name '*.rel' -or -name '*.dump*' -or -name '*.sym' -or -name '*.o' -or -name '*.lst' -or -name '*.asm'`
+
+# Deleting all the installed files
+# --------------------------------
+uninstall:
+	for hdr in ../include/*.h; do rm -f $(sdcc_datadir)/include/$$hdr; done
+	for cfl in *.c; do rm -f $(sdcc_datadir)/lib/$$cfl; done
+	for model in $(MODELS); do \
+	  rm -rf $(sdcc_datadir)/lib/$$model; \
+	done
+	rm -rf $(sdcc_datadir)/lib/ds390
+
+
+# Performing self-test
+# --------------------
+check:
+
+
+# Performing installation test
+# ----------------------------
+installcheck:
+
+
+# Creating installation directories
+# ---------------------------------
+installdirs:
+	[ -d $(sdcc_datadir)/lib ] || mkdir -p $(sdcc_datadir)/lib
+	for model in $(MODELS); do \
+	 [ -d $(sdcc_datadir)/lib/$$model ] || \
+	 mkdir -p $(sdcc_datadir)/lib/$$model; \
+	done
+	[ -d $(sdcc_datadir)/lib/ds390 ] || mkdir -p $(sdcc_datadir)/lib/ds390
+	mkdir -p $(sdcc_datadir)/lib/src
+
+# Creating dependencies
+# ---------------------
+dep: Makefile.dep
+
+Makefile.dep: $(SOURCES) $(INCDIR)/*.h
+	rm -f Makefile.dep
+	for i in $(SOURCES); do \
+	  $(CC) -lang-c++ -M $(CPPFLAGS) $$i >$${i}.dep; \
+	  cat $${i}.dep >>Makefile.dep; \
+	  rm $${i}.dep; \
+	done
+
+include Makefile.dep
+include clean.mk
+
+# My rules
+# --------
+
+.SUFFIXES: .rel .o
+
+$(PORTDIR)/%$(OEXT): %.c
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c $<
+	mv -f `basename $@` $@
+	mv -f `basename $@ $(OEXT)`.asm $(PORTDIR)
+
+# Remaking configuration
+# ----------------------
+checkconf:
+	@if [ -f $(PRJDIR)/devel ]; then\
+	  $(MAKE) -f $(srcdir)/conf.mk srcdir="$(srcdir)" PRJDIR="$(PRJDIR)" \
+	  freshconf;\
+	fi
+
+# End of main_in.mk/main.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/device/lib/Makefile.in sdcc/device/lib/Makefile.in
--- sdcc.CVS/device/lib/Makefile.in	Sun Aug 25 11:28:32 2002
+++ sdcc/device/lib/Makefile.in	Sun Aug 25 11:28:01 2002
@@ -219,7 +219,7 @@
 Makefile.dep: $(SOURCES) $(INCDIR)/*.h
 	rm -f Makefile.dep
 	for i in $(SOURCES); do \
-	  $(CPP) -lang-c++ -M $(CPPFLAGS) $$i >$${i}.dep; \
+	  $(CC) -lang-c++ -M $(CPPFLAGS) $$i >$${i}.dep; \
 	  cat $${i}.dep >>Makefile.dep; \
 	  rm $${i}.dep; \
 	done
diff -ur --exclude=CVS --new-file sdcc.CVS/link/z80/Makefile.in sdcc/link/z80/Makefile.in
--- sdcc.CVS/link/z80/Makefile.in	Thu Aug 16 14:09:35 2001
+++ sdcc/link/z80/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -81,7 +81,7 @@
 dep: Makefile.dep
 
 Makefile.dep: $(LKSOURCES) *.h $(PRJDIR)/*.h
-	$(CPP) $(CPPFLAGS) $(M_OR_MM) $(LKSOURCES) >Makefile.dep
+	$(CC) $(CPPFLAGS) $(M_OR_MM) $(LKSOURCES) >Makefile.dep
 
 include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/sim/ucsim/avr.src/Makefile.in sdcc/sim/ucsim/avr.src/Makefile.in
--- sdcc.CVS/sim/ucsim/avr.src/Makefile.in	Mon Jan 28 04:19:57 2002
+++ sdcc/sim/ucsim/avr.src/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -95,7 +95,7 @@
 dep: Makefile.dep
 
 Makefile.dep: *.cc *.h
-	$(CXXCPP) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
+	$(CXX) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
 
 include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/sim/ucsim/cmd.src/Makefile.in sdcc/sim/ucsim/cmd.src/Makefile.in
--- sdcc.CVS/sim/ucsim/cmd.src/Makefile.in	Mon Jan 28 04:19:57 2002
+++ sdcc/sim/ucsim/cmd.src/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -78,7 +78,7 @@
 dep: Makefile.dep
 
 Makefile.dep: *.cc *.h
-	$(CXXCPP) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
+	$(CXX) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
 
 include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/sim/ucsim/gui.src/Makefile.in sdcc/sim/ucsim/gui.src/Makefile.in
--- sdcc.CVS/sim/ucsim/gui.src/Makefile.in	Mon Jan 28 04:19:58 2002
+++ sdcc/sim/ucsim/gui.src/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -137,7 +137,7 @@
 dep: Makefile.dep
 
 Makefile.dep: $(ALL_SOURCES) *.h
-	$(CXXCPP) -MM $(CPPFLAGS) $(M_OR_MM) $(ALL_SOURCES) >Makefile.dep
+	$(CXX) -MM $(CPPFLAGS) $(M_OR_MM) $(ALL_SOURCES) >Makefile.dep
 
 include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/sim/ucsim/gui.src/obsolete/portmon.src/Makefile.in sdcc/sim/ucsim/gui.src/obsolete/portmon.src/Makefile.in
--- sdcc.CVS/sim/ucsim/gui.src/obsolete/portmon.src/Makefile.in	Mon Apr  2 07:08:48 2001
+++ sdcc/sim/ucsim/gui.src/obsolete/portmon.src/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -78,7 +78,7 @@
 dep: Makefile.dep
 
 Makefile.dep: *.cc *.h $(PRJDIR)/*.h $(PKGDIR)/*.h
-	$(CXXCPP) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
+	$(CXX) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
 
 include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/sim/ucsim/gui.src/serio.src/Makefile.in sdcc/sim/ucsim/gui.src/serio.src/Makefile.in
--- sdcc.CVS/sim/ucsim/gui.src/serio.src/Makefile.in	Tue Apr  9 07:46:57 2002
+++ sdcc/sim/ucsim/gui.src/serio.src/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -80,7 +80,7 @@
 dep: Makefile.dep
 
 Makefile.dep: *.cc *.h *.hh
-	$(CXXCPP) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
+	$(CXX) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
 
 include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/sim/ucsim/main_in.mk sdcc/sim/ucsim/main_in.mk
--- sdcc.CVS/sim/ucsim/main_in.mk	Mon Jan 28 04:19:57 2002
+++ sdcc/sim/ucsim/main_in.mk	Mon Aug 12 16:40:52 2002
@@ -94,7 +94,7 @@
 dep: main.dep
 
 main.dep: $(ALL_SOURCES) *.h
-	$(CXXCPP) $(CPPFLAGS) $(M_OR_MM) $(ALL_SOURCES) >main.dep
+	$(CXX) $(CPPFLAGS) $(M_OR_MM) $(ALL_SOURCES) >main.dep
 
 include main.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/sim/ucsim/s51.src/Makefile.in sdcc/sim/ucsim/s51.src/Makefile.in
--- sdcc.CVS/sim/ucsim/s51.src/Makefile.in	Mon Jan 28 04:19:58 2002
+++ sdcc/sim/ucsim/s51.src/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -101,7 +101,7 @@
 dep: Makefile.dep
 
 Makefile.dep: *.cc *.h
-	$(CXXCPP) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
+	$(CXX) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
 
 include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/sim/ucsim/sim.src/Makefile.in sdcc/sim/ucsim/sim.src/Makefile.in
--- sdcc.CVS/sim/ucsim/sim.src/Makefile.in	Mon Jan 28 04:19:58 2002
+++ sdcc/sim/ucsim/sim.src/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -80,7 +80,7 @@
 dep: main.dep
 
 Makefile.dep: *.cc *.h
-	$(CXXCPP) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
+	$(CXX) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
 
 include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/sim/ucsim/xa.src/Makefile.in sdcc/sim/ucsim/xa.src/Makefile.in
--- sdcc.CVS/sim/ucsim/xa.src/Makefile.in	Mon Jan 28 04:19:58 2002
+++ sdcc/sim/ucsim/xa.src/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -85,7 +85,7 @@
 dep: Makefile.dep
 
 Makefile.dep: *.cc *.h
-	$(CXXCPP) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
+	$(CXX) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
 
 include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/sim/ucsim/z80.src/Makefile.in sdcc/sim/ucsim/z80.src/Makefile.in
--- sdcc.CVS/sim/ucsim/z80.src/Makefile.in	Mon Jan 28 04:19:58 2002
+++ sdcc/sim/ucsim/z80.src/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -102,7 +102,7 @@
 dep: Makefile.dep
 
 Makefile.dep: *.cc *.h
-	$(CXXCPP) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
+	$(CXX) $(CPPFLAGS) $(M_OR_MM) *.cc >Makefile.dep
 
 include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/src/Makefile.in sdcc/src/Makefile.in
--- sdcc.CVS/src/Makefile.in	Mon Oct  1 02:55:45 2001
+++ sdcc/src/Makefile.in	Mon Aug 12 16:40:52 2002
@@ -84,7 +84,7 @@
 dep: Makefile.dep
 
 Makefile.dep: $(SOURCES) $(SPECIAL) *.h $(PRJDIR)/*.h
-	$(CPP) $(CPPFLAGS) $(M_OR_MM) $(SOURCES) >Makefile.dep
+	$(CC) $(CPPFLAGS) $(M_OR_MM) $(SOURCES) >Makefile.dep
 
 include Makefile.dep
 include clean.mk
diff -ur --exclude=CVS --new-file sdcc.CVS/src/port.mk sdcc/src/port.mk
--- sdcc.CVS/src/port.mk	Sun Sep 30 12:42:31 2001
+++ sdcc/src/port.mk	Mon Aug 12 16:40:52 2002
@@ -39,7 +39,7 @@
 dep: Makefile.dep
 
 Makefile.dep: $(PREBUILD) Makefile $(SOURCES) $(SPECIAL) *.h $(PRJDIR)/*.h $(PRJDIR)/src/*.h
-	$(CPP) $(CPPFLAGS) $(M_OR_MM) $(SOURCES) >Makefile.dep
+	$(CC) $(CPPFLAGS) $(M_OR_MM) $(SOURCES) >Makefile.dep
 
 include Makefile.dep
 
