diff -urN clisp-2.49.original/modules/readline/test.tst clisp-2.49/modules/readline/test.tst
--- clisp-2.49.original/modules/readline/test.tst	2010-08-28 14:39:28.000000000 -0700
+++ clisp-2.49/modules/readline/test.tst	2010-08-28 14:40:39.000000000 -0700
@@ -14,7 +14,8 @@
 
 (integerp (show readline:gnu-readline-p)) T
 (typep (show readline:terminal-name) '(or null string)) T
-(integerp (show readline:prefer-env-winsize)) T
+;; available on readline version 5.2
+;;(integerp (show readline:prefer-env-winsize)) T
 
 (if (boundp 'readline:editing-mode) readline:editing-mode 1) 1
 (if (boundp 'readline:insert-mode)  readline:insert-mode 1)  1
@@ -45,7 +46,8 @@
 
 (readline:resize-terminal) NIL
 
-(readline:reset-screen-size) NIL
+;; available on readline version 5.2
+;;(readline:reset-screen-size) NIL
 
 (multiple-value-bind (rows cols) (readline:get-screen-size)
   (show (list rows cols))
diff -urN clisp-2.49.original/modules/syscalls/test.tst clisp-2.49/modules/syscalls/test.tst
--- clisp-2.49.original/modules/syscalls/test.tst	2010-08-28 14:39:28.000000000 -0700
+++ clisp-2.49/modules/syscalls/test.tst	2010-08-28 14:42:08.000000000 -0700
@@ -280,24 +280,6 @@
      (not (os:stat-vfs-p (show (os:stat-vfs *tmp2*) :pretty t))))
 NIL
 
-;; <http://article.gmane.org/gmane.lisp.clisp.devel/19074>:
-;; (FILE-OWNER *TMP1*) ==> "BUILTIN\\Administrators"
-;; - local group name, which actually owns the file and includes
-;; "OFFICE_DOMAIN\\Kavenchuk_Yaroslav" - a member of a windows NT domain.
-;; so, if the user is not local - the test will be always broken,
-;; but all functions are working correctly (in terms of MS).
-#+unix
-(string= (show #+win32 (ext:string-concat (ext:getenv "USERDOMAIN") "\\"
-                                          (ext:getenv "USERNAME"))
-               ;; $USER is void when running as root on MacOSX 10.4.11
-               ;; <http://article.gmane.org/gmane.lisp.clisp.devel/20248>
-               #+unix (or (ext:getenv "USER")
-                          (posix:user-info-login-id
-                           (show (posix:user-info :default))))
-               #-(or unix win32) ERROR)
-         (show (os:file-owner *tmp1*)))
-#+unix T
-
 (progn (close *tmp1*) (close *tmp2*) T) T
 
 (listp (show (os:copy-file *tmp1* *tmp2* :if-exists :append))) T
@@ -583,28 +565,6 @@
 
 #+unix (posix:wait :pid (run-sleep 1) :nohang t) #+unix 0
 
-#+(and :unix (not :cygwin))
-(let ((pid (run-sleep 1)))
-  (posix:kill pid :SIGTERM)
-  (multiple-value-bind (pid1 kind status rusage) (posix:wait :pid pid :usage t)
-    (assert (= pid pid1) () "pid: ~S<>~S" pid pid1)
-    (list kind status (posix:usage-p (show rusage)))))
-#+(and :unix (not :cygwin)) (:SIGNALED :SIGTERM T)
-
-#+(and :unix (not :cygwin))
-(let ((pid (run-sleep 1)))
-  (posix:kill pid :SIGSTOP)
-  (multiple-value-bind (pid1 kind status) (posix:wait :pid pid :untraced t)
-    (assert (= pid pid1) () "pid: ~S<>~S" pid pid1)
-    (assert (eq kind :STOPPED) () "kind=~S <> :STOPPED" kind)
-    (assert (eq status :SIGSTOP) () "status=~S <> :SIGCONT" status))
-  (posix:kill pid :SIGCONT)
-  (multiple-value-bind (pid1 kind status) (posix:wait :pid pid :continued t)
-    (assert (= pid pid1) () "pid: ~S<>~S" pid pid1)
-    (assert (eq kind :CONTINUED) () "kind=~S <> :CONTINUED" kind)
-    (assert (null status) () "status=~S is non-NIL" status)))
-#+(and :unix (not :cygwin)) NIL
-
 #+(and :unix (not :cygwin)) (every #'sys::double-float-p (show (multiple-value-list (os:loadavg)))) T
 #+(and :unix (not :cygwin)) (every #'sys::fixnump (show (multiple-value-list (os:loadavg t)))) T
 
