diff -ruN net-http-new/configure net-http/configure
--- net-http-new/configure	2005-03-27 07:32:49.000000000 -0800
+++ net-http/configure	2005-03-27 07:33:17.000000000 -0800
@@ -522,8 +522,8 @@
 
 
 
-    # Extract the first word of "guile", so it can be a program name with args.
-set dummy guile; ac_word=$2
+    # Extract the first word of "guile-1.8", so it can be a program name with args.
+set dummy guile-1.8; ac_word=$2
 echo $ac_n "checking for $ac_word""... $ac_c" 1>&6
 echo "configure:529: checking for $ac_word" >&5
 if eval "test \"`echo '$''{'ac_cv_path_GUILE'+set}'`\" = set"; then
@@ -562,9 +562,9 @@
 
 
     ## First, let's just see if we can find Guile at all.
-    echo $ac_n "checking "for guile-config"""... $ac_c" 1>&6
-echo "configure:567: checking "for guile-config"" >&5
-    for GUILE_CONFIG in guile-config $target-guile-config $build-guile-config; do
+    echo $ac_n "checking "for guile-1.8-config"""... $ac_c" 1>&6
+echo "configure:567: checking "for guile-1.8-config"" >&5
+    for GUILE_CONFIG in guile-1.8-config $target-guile-1.8-config $build-guile-1.8-config; do
         echo "$ac_t"""$GUILE_CONFIG"" 1>&6
         if ! $GUILE_CONFIG --version > /dev/null 2>&1 ; then
             echo "configure: warning: "cannot execute $GUILE_CONFIG"" 1>&2
@@ -576,7 +576,7 @@
         fi
     done
     if test "$GUILE_CONFIG" = "error"; then
-        { echo "configure: error: "cannot find guile-config; is Guile installed?"" 1>&2; exit 1; }
+        { echo "configure: error: "cannot find guile-1.8-config; is Guile installed?"" 1>&2; exit 1; }
         exit 1
     fi
     
@@ -584,18 +584,6 @@
 
     echo $ac_n "checking "Guile version"""... $ac_c" 1>&6
 echo "configure:587: checking "Guile version"" >&5
-    need_guile_version="1.4"
-    need_guile_version_numeric=100400
-    guile_version=`$GUILE_CONFIG --version 2>&1 | awk '{print $NF}'`
-    guile_version_numeric=`echo $guile_version | awk -F. '
-{if ($3) {last = $3}
-else {last =0}}
-{printf "%s%s%s\n",$1*100, $2*10,last}'`
-    echo "$ac_t"""$guile_version"" 1>&6
-    if test $guile_version_numeric -lt $need_guile_version_numeric
-    then
-        { echo "configure: error: "Guile version "$need_guile_version" or newer is needed"" 1>&2; exit 1; }
-    fi
 
 
     echo $ac_n "checking for Guile site directory""... $ac_c" 1>&6
