--- /tmp/pf/Makefile	Mon Apr  6 17:13:12 1998
+++ ./Makefile	Thu Jan  9 13:50:28 2003
@@ -18,7 +18,6 @@
         -fno-unroll-loops \
         -fpeephole \
         -fno-keep-inline-functions \
-        -x c++ \
         -Wcast-qual \
         -Wall \
         -Wwrite-strings \
@@ -69,7 +68,7 @@
 
 # build basic dictionary by running newly built pforth and including system.fth
 pforth.dic: pforth
-	pforth -i system.fth
+	./pforth -i system.fth
 
 ${SOURCEDIR}/pfdicdat.h: pforth pforth.dic
 	@(echo 'INCLUDE utils/savedicd.fth'; \
