diff -ru libgcrypt-1.1.94/src/ath.h libgcrypt-1.1.94-patched/src/ath.h
--- libgcrypt-1.1.94/src/ath.h	Mon Oct 27 12:37:05 2003
+++ libgcrypt-1.1.94-patched/src/ath.h	Tue Dec 16 18:54:47 2003
@@ -31,6 +31,11 @@
 #include <sys/socket.h>
 #endif
 
+//+crh
+#include <sys/time.h>
+#define socklen_t int
+//-crh
+
 
 /* Define _ATH_EXT_SYM_PREFIX if you want to give all external symbols
    a prefix.  */
