diff -urN --exclude=*.orig dxsamples-4.3.2/callmodule/Makefile.am dxsamples-new/callmodule/Makefile.am
--- dxsamples-4.3.2/callmodule/Makefile.am	Thu Jun 19 01:02:52 2003
+++ dxsamples-new/callmodule/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -7,8 +7,8 @@
 EXTRA_DIST = $(data_DATA)
 
 install-data-hook:
-	cp localmk $(datadir)/makefile
+	cp localmk $(DESTDIR)$(datadir)/makefile
 
 uninstall-local:
-	rm $(datadir)/makefile
+	rm $(DESTDIR)$(datadir)/makefile
 
diff -urN --exclude=*.orig dxsamples-4.3.2/callmodule/Makefile.in dxsamples-new/callmodule/Makefile.in
--- dxsamples-4.3.2/callmodule/Makefile.in	Tue Oct 28 12:24:17 2003
+++ dxsamples-new/callmodule/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -271,7 +271,7 @@
 
 
 install-data-hook:
-	cp localmk $(datadir)/makefile
+	cp localmk $(DESTDIR)$(datadir)/makefile
 
 uninstall-local:
 	rm $(datadir)/makefile
diff -urN --exclude=*.orig dxsamples-4.3.2/program_guide/Makefile.am dxsamples-new/program_guide/Makefile.am
--- dxsamples-4.3.2/program_guide/Makefile.am	Thu Jun 19 01:09:12 2003
+++ dxsamples-new/program_guide/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -19,7 +19,7 @@
 EXTRA_DIST = $(data_DATA) 
 
 install-data-hook:
-	cp localmk $(datadir)/makefile
+	cp localmk $(DESTDIR)$(datadir)/makefile
 
 uninstall-local:
-	rm $(datadir)/makefile
+	rm $(DESTDIR)$(datadir)/makefile
diff -urN --exclude=*.orig dxsamples-4.3.2/program_guide/Makefile.in dxsamples-new/program_guide/Makefile.in
--- dxsamples-4.3.2/program_guide/Makefile.in	Tue Oct 28 12:24:28 2003
+++ dxsamples-new/program_guide/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -286,10 +286,10 @@
 
 
 install-data-hook:
-	cp localmk $(datadir)/makefile
+	cp localmk $(DESTDIR)$(datadir)/makefile
 
 uninstall-local:
-	rm $(datadir)/makefile
+	rm $(DESTDIR)$(datadir)/makefile
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/2D_DATA/Makefile.am dxsamples-new/programs/2D_DATA/Makefile.am
--- dxsamples-4.3.2/programs/2D_DATA/Makefile.am	Tue Apr 30 14:53:26 2002
+++ dxsamples-new/programs/2D_DATA/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/2D_DATA
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/2D_DATA
 
 LINKLIST = \
          AlternateVisualizations.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/2D_DATA/Makefile.in dxsamples-new/programs/2D_DATA/Makefile.in
--- dxsamples-4.3.2/programs/2D_DATA/Makefile.in	Tue Oct 28 12:24:29 2003
+++ dxsamples-new/programs/2D_DATA/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/2D_DATA
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/2D_DATA
 
 LINKLIST = \
          AlternateVisualizations.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/3D_DATA/Makefile.am dxsamples-new/programs/3D_DATA/Makefile.am
--- dxsamples-4.3.2/programs/3D_DATA/Makefile.am	Tue Apr 30 14:53:26 2002
+++ dxsamples-new/programs/3D_DATA/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/3D_DATA
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/3D_DATA
 
 LINKLIST = \
          AnnotationGlyphs.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/3D_DATA/Makefile.in dxsamples-new/programs/3D_DATA/Makefile.in
--- dxsamples-4.3.2/programs/3D_DATA/Makefile.in	Tue Oct 28 12:24:30 2003
+++ dxsamples-new/programs/3D_DATA/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/3D_DATA
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/3D_DATA
 
 LINKLIST = \
          AnnotationGlyphs.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/ANNOTATION/Makefile.am dxsamples-new/programs/ANNOTATION/Makefile.am
--- dxsamples-4.3.2/programs/ANNOTATION/Makefile.am	Tue Apr 30 14:53:25 2002
+++ dxsamples-new/programs/ANNOTATION/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/ANNOTATION
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/ANNOTATION
 
 LINKLIST = \
          AnnotationGlyphs.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/ANNOTATION/Makefile.in dxsamples-new/programs/ANNOTATION/Makefile.in
--- dxsamples-4.3.2/programs/ANNOTATION/Makefile.in	Tue Oct 28 12:24:31 2003
+++ dxsamples-new/programs/ANNOTATION/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/ANNOTATION
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/ANNOTATION
 
 LINKLIST = \
          AnnotationGlyphs.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/CATEGORICAL/Makefile.am dxsamples-new/programs/CATEGORICAL/Makefile.am
--- dxsamples-4.3.2/programs/CATEGORICAL/Makefile.am	Tue Apr 30 14:53:24 2002
+++ dxsamples-new/programs/CATEGORICAL/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/CATEGORICAL
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/CATEGORICAL
 
 LINKLIST = \
          Categorical.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/CATEGORICAL/Makefile.in dxsamples-new/programs/CATEGORICAL/Makefile.in
--- dxsamples-4.3.2/programs/CATEGORICAL/Makefile.in	Tue Oct 28 12:24:31 2003
+++ dxsamples-new/programs/CATEGORICAL/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/CATEGORICAL
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/CATEGORICAL
 
 LINKLIST = \
          Categorical.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/COLORMAP_EDITOR/Makefile.am dxsamples-new/programs/COLORMAP_EDITOR/Makefile.am
--- dxsamples-4.3.2/programs/COLORMAP_EDITOR/Makefile.am	Tue Apr 30 14:53:23 2002
+++ dxsamples-new/programs/COLORMAP_EDITOR/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/COLORMAP_EDITOR
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/COLORMAP_EDITOR
 
 LINKLIST = \
          DataDrivenInteractors.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/COLORMAP_EDITOR/Makefile.in dxsamples-new/programs/COLORMAP_EDITOR/Makefile.in
--- dxsamples-4.3.2/programs/COLORMAP_EDITOR/Makefile.in	Tue Oct 28 12:24:32 2003
+++ dxsamples-new/programs/COLORMAP_EDITOR/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/COLORMAP_EDITOR
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/COLORMAP_EDITOR
 
 LINKLIST = \
          DataDrivenInteractors.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/COMPUTE/Makefile.am dxsamples-new/programs/COMPUTE/Makefile.am
--- dxsamples-4.3.2/programs/COMPUTE/Makefile.am	Tue Apr 30 14:53:23 2002
+++ dxsamples-new/programs/COMPUTE/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/COMPUTE
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/COMPUTE
 
 LINKLIST = \
          Bounce.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/COMPUTE/Makefile.in dxsamples-new/programs/COMPUTE/Makefile.in
--- dxsamples-4.3.2/programs/COMPUTE/Makefile.in	Tue Oct 28 12:24:34 2003
+++ dxsamples-new/programs/COMPUTE/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/COMPUTE
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/COMPUTE
 
 LINKLIST = \
          Bounce.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/DATA_DRIVEN_INTERACTORS/Makefile.am dxsamples-new/programs/DATA_DRIVEN_INTERACTORS/Makefile.am
--- dxsamples-4.3.2/programs/DATA_DRIVEN_INTERACTORS/Makefile.am	Tue Apr 30 14:53:22 2002
+++ dxsamples-new/programs/DATA_DRIVEN_INTERACTORS/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/DATA_DRIVEN_INTERACTORS
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/DATA_DRIVEN_INTERACTORS
 
 LINKLIST = \
          DataDrivenInteractors.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/DATA_DRIVEN_INTERACTORS/Makefile.in dxsamples-new/programs/DATA_DRIVEN_INTERACTORS/Makefile.in
--- dxsamples-4.3.2/programs/DATA_DRIVEN_INTERACTORS/Makefile.in	Tue Oct 28 12:24:35 2003
+++ dxsamples-new/programs/DATA_DRIVEN_INTERACTORS/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/DATA_DRIVEN_INTERACTORS
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/DATA_DRIVEN_INTERACTORS
 
 LINKLIST = \
          DataDrivenInteractors.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/DEBUGGING/Makefile.am dxsamples-new/programs/DEBUGGING/Makefile.am
--- dxsamples-4.3.2/programs/DEBUGGING/Makefile.am	Tue Apr 30 14:53:21 2002
+++ dxsamples-new/programs/DEBUGGING/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/DEBUGGING
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/DEBUGGING
 
 LINKLIST = \
          Verify.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/DEBUGGING/Makefile.in dxsamples-new/programs/DEBUGGING/Makefile.in
--- dxsamples-4.3.2/programs/DEBUGGING/Makefile.in	Tue Oct 28 12:24:35 2003
+++ dxsamples-new/programs/DEBUGGING/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/DEBUGGING
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/DEBUGGING
 
 LINKLIST = \
          Verify.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/DISTRIBUTED_PROCESSING/Makefile.am dxsamples-new/programs/DISTRIBUTED_PROCESSING/Makefile.am
--- dxsamples-4.3.2/programs/DISTRIBUTED_PROCESSING/Makefile.am	Tue Apr 30 14:53:20 2002
+++ dxsamples-new/programs/DISTRIBUTED_PROCESSING/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/DISTRIBUTED_PROCESSING
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/DISTRIBUTED_PROCESSING
 
 LINKLIST = \
          Distributed.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/DISTRIBUTED_PROCESSING/Makefile.in dxsamples-new/programs/DISTRIBUTED_PROCESSING/Makefile.in
--- dxsamples-4.3.2/programs/DISTRIBUTED_PROCESSING/Makefile.in	Tue Oct 28 12:24:36 2003
+++ dxsamples-new/programs/DISTRIBUTED_PROCESSING/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/DISTRIBUTED_PROCESSING
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/DISTRIBUTED_PROCESSING
 
 LINKLIST = \
          Distributed.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/IMAGE_PROCESSING/Makefile.am dxsamples-new/programs/IMAGE_PROCESSING/Makefile.am
--- dxsamples-4.3.2/programs/IMAGE_PROCESSING/Makefile.am	Tue Apr 30 14:53:20 2002
+++ dxsamples-new/programs/IMAGE_PROCESSING/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/IMAGE_PROCESSING
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/IMAGE_PROCESSING
 
 LINKLIST = \
          FFT.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/IMAGE_PROCESSING/Makefile.in dxsamples-new/programs/IMAGE_PROCESSING/Makefile.in
--- dxsamples-4.3.2/programs/IMAGE_PROCESSING/Makefile.in	Tue Oct 28 12:24:37 2003
+++ dxsamples-new/programs/IMAGE_PROCESSING/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/IMAGE_PROCESSING
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/IMAGE_PROCESSING
 
 LINKLIST = \
          FFT.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/IMPORTING/Makefile.am dxsamples-new/programs/IMPORTING/Makefile.am
--- dxsamples-4.3.2/programs/IMPORTING/Makefile.am	Tue Apr 30 14:53:19 2002
+++ dxsamples-new/programs/IMPORTING/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/IMPORTING
+sampleprogdir = $(ESTDIR)$(prefix)/dx/samples/programs/IMPORTING
 
 LINKLIST = \
          Categorical.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/IMPORTING/Makefile.in dxsamples-new/programs/IMPORTING/Makefile.in
--- dxsamples-4.3.2/programs/IMPORTING/Makefile.in	Tue Oct 28 12:24:37 2003
+++ dxsamples-new/programs/IMPORTING/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/IMPORTING
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/IMPORTING
 
 LINKLIST = \
          Categorical.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/INTERFACE_CONTROL/Makefile.am dxsamples-new/programs/INTERFACE_CONTROL/Makefile.am
--- dxsamples-4.3.2/programs/INTERFACE_CONTROL/Makefile.am	Tue Apr 30 14:53:18 2002
+++ dxsamples-new/programs/INTERFACE_CONTROL/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/INTERFACE_CONTROL
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/INTERFACE_CONTROL
 
 LINKLIST = \
          DialogStyle.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/INTERFACE_CONTROL/Makefile.in dxsamples-new/programs/INTERFACE_CONTROL/Makefile.in
--- dxsamples-4.3.2/programs/INTERFACE_CONTROL/Makefile.in	Tue Oct 28 12:24:38 2003
+++ dxsamples-new/programs/INTERFACE_CONTROL/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/INTERFACE_CONTROL
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/INTERFACE_CONTROL
 
 LINKLIST = \
          DialogStyle.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/LOOPING/Makefile.am dxsamples-new/programs/LOOPING/Makefile.am
--- dxsamples-4.3.2/programs/LOOPING/Makefile.am	Tue Apr 30 14:53:17 2002
+++ dxsamples-new/programs/LOOPING/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/LOOPING
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/LOOPING
 
 LINKLIST = \
          Accumulate.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/LOOPING/Makefile.in dxsamples-new/programs/LOOPING/Makefile.in
--- dxsamples-4.3.2/programs/LOOPING/Makefile.in	Tue Oct 28 12:24:39 2003
+++ dxsamples-new/programs/LOOPING/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/LOOPING
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/LOOPING
 
 LINKLIST = \
          Accumulate.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/MISC/Makefile.am dxsamples-new/programs/MISC/Makefile.am
--- dxsamples-4.3.2/programs/MISC/Makefile.am	Tue Apr 30 14:53:13 2002
+++ dxsamples-new/programs/MISC/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/MISC
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/MISC
 
 LINKLIST = \
          CensusData.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/MISC/Makefile.in dxsamples-new/programs/MISC/Makefile.in
--- dxsamples-4.3.2/programs/MISC/Makefile.in	Tue Oct 28 12:24:41 2003
+++ dxsamples-new/programs/MISC/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/MISC
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/MISC
 
 LINKLIST = \
          CensusData.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/PLOTTING/Makefile.am dxsamples-new/programs/PLOTTING/Makefile.am
--- dxsamples-4.3.2/programs/PLOTTING/Makefile.am	Tue Apr 30 14:53:12 2002
+++ dxsamples-new/programs/PLOTTING/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/PLOTTING
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/PLOTTING
 
 LINKLIST = \
          Categorical.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/PLOTTING/Makefile.in dxsamples-new/programs/PLOTTING/Makefile.in
--- dxsamples-4.3.2/programs/PLOTTING/Makefile.in	Tue Oct 28 12:24:42 2003
+++ dxsamples-new/programs/PLOTTING/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/PLOTTING
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/PLOTTING
 
 LINKLIST = \
          Categorical.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/PROBES/Makefile.am dxsamples-new/programs/PROBES/Makefile.am
--- dxsamples-4.3.2/programs/PROBES/Makefile.am	Tue Apr 30 14:53:11 2002
+++ dxsamples-new/programs/PROBES/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/PROBES
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/PROBES
 
 LINKLIST = \
          PlotLine2.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/PROBES/Makefile.in dxsamples-new/programs/PROBES/Makefile.in
--- dxsamples-4.3.2/programs/PROBES/Makefile.in	Tue Oct 28 12:24:43 2003
+++ dxsamples-new/programs/PROBES/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/PROBES
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/PROBES
 
 LINKLIST = \
          PlotLine2.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/RENDERING/Makefile.am dxsamples-new/programs/RENDERING/Makefile.am
--- dxsamples-4.3.2/programs/RENDERING/Makefile.am	Tue Apr 30 14:53:10 2002
+++ dxsamples-new/programs/RENDERING/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/RENDERING
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/RENDERING
 
 LINKLIST = \
          FatLines.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/RENDERING/Makefile.in dxsamples-new/programs/RENDERING/Makefile.in
--- dxsamples-4.3.2/programs/RENDERING/Makefile.in	Tue Oct 28 12:24:44 2003
+++ dxsamples-new/programs/RENDERING/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/RENDERING
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/RENDERING
 
 LINKLIST = \
          FatLines.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/SCATTERED/Makefile.am dxsamples-new/programs/SCATTERED/Makefile.am
--- dxsamples-4.3.2/programs/SCATTERED/Makefile.am	Tue Apr 30 14:53:08 2002
+++ dxsamples-new/programs/SCATTERED/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/SCATTERED
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/SCATTERED
 
 LINKLIST = \
          AnnotationGlyphs.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/SCATTERED/Makefile.in dxsamples-new/programs/SCATTERED/Makefile.in
--- dxsamples-4.3.2/programs/SCATTERED/Makefile.in	Tue Oct 28 12:24:44 2003
+++ dxsamples-new/programs/SCATTERED/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/SCATTERED
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/SCATTERED
 
 LINKLIST = \
          AnnotationGlyphs.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/SEQUENCER/Makefile.am dxsamples-new/programs/SEQUENCER/Makefile.am
--- dxsamples-4.3.2/programs/SEQUENCER/Makefile.am	Tue Apr 30 14:53:07 2002
+++ dxsamples-new/programs/SEQUENCER/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/SEQUENCER
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/SEQUENCER
 
 LINKLIST = \
          ContoursAndCaption.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/SEQUENCER/Makefile.in dxsamples-new/programs/SEQUENCER/Makefile.in
--- dxsamples-4.3.2/programs/SEQUENCER/Makefile.in	Tue Oct 28 12:24:45 2003
+++ dxsamples-new/programs/SEQUENCER/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/SEQUENCER
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/SEQUENCER
 
 LINKLIST = \
          ContoursAndCaption.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/SIMPLE/Makefile.am dxsamples-new/programs/SIMPLE/Makefile.am
--- dxsamples-4.3.2/programs/SIMPLE/Makefile.am	Tue Apr 30 14:53:06 2002
+++ dxsamples-new/programs/SIMPLE/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/SIMPLE
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/SIMPLE
 
 LINKLIST = \
          Arrange.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/SIMPLE/Makefile.in dxsamples-new/programs/SIMPLE/Makefile.in
--- dxsamples-4.3.2/programs/SIMPLE/Makefile.in	Tue Oct 28 12:24:46 2003
+++ dxsamples-new/programs/SIMPLE/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/SIMPLE
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/SIMPLE
 
 LINKLIST = \
          Arrange.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/SUPERVISE/Makefile.am dxsamples-new/programs/SUPERVISE/Makefile.am
--- dxsamples-4.3.2/programs/SUPERVISE/Makefile.am	Tue Apr 30 14:53:01 2002
+++ dxsamples-new/programs/SUPERVISE/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-sampleprogdir = $(prefix)/dx/samples/programs/SUPERVISE
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/SUPERVISE
 
 LINKLIST = \
          Image_wo_UI.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/programs/SUPERVISE/Makefile.in dxsamples-new/programs/SUPERVISE/Makefile.in
--- dxsamples-4.3.2/programs/SUPERVISE/Makefile.in	Tue Oct 28 12:24:46 2003
+++ dxsamples-new/programs/SUPERVISE/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -101,7 +101,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-sampleprogdir = $(prefix)/dx/samples/programs/SUPERVISE
+sampleprogdir = $(DESTDIR)$(prefix)/dx/samples/programs/SUPERVISE
 
 LINKLIST = \
          Image_wo_UI.cfg\
diff -urN --exclude=*.orig dxsamples-4.3.2/stereo/src/Makefile.am dxsamples-new/stereo/src/Makefile.am
--- dxsamples-4.3.2/stereo/src/Makefile.am	Sun Mar 31 21:28:51 2002
+++ dxsamples-new/stereo/src/Makefile.am	Sat Jan 31 13:58:31 2004
@@ -7,7 +7,7 @@
 EXTRA_DIST = $(data_DATA) 
 
 install-data-hook:
-	cp localmk $(datadir)/makefile
+	cp localmk $(DESTDIR)$(datadir)/makefile
 
 uninstall-local:
-	rm $(datadir)/makefile
+	rm $(DESTDIR)$(datadir)/makefile
diff -urN --exclude=*.orig dxsamples-4.3.2/stereo/src/Makefile.in dxsamples-new/stereo/src/Makefile.in
--- dxsamples-4.3.2/stereo/src/Makefile.in	Tue Oct 28 12:24:48 2003
+++ dxsamples-new/stereo/src/Makefile.in	Sat Jan 31 13:58:31 2004
@@ -271,10 +271,10 @@
 
 
 install-data-hook:
-	cp localmk $(datadir)/makefile
+	cp localmk $(DESTDIR)$(datadir)/makefile
 
 uninstall-local:
-	rm $(datadir)/makefile
+	rm $(DESTDIR)$(datadir)/makefile
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -urN --exclude=*.orig dxsamples-4.3.2/user/Makefile.am dxsamples-new/user/Makefile.am
--- dxsamples-4.3.2/user/Makefile.am	Thu Jun 19 01:09:14 2003
+++ dxsamples-new/user/Makefile.am	Sat Jan 31 14:11:05 2004
@@ -1,7 +1,8 @@
+
 ## Process this file with automake to produce Makefile.in
 
 
 datadir = $(prefix)/dx/samples/user
 
-data_DATA =  Makefile_inboard_alphax Makefile_inboard_aviion Makefile_inboard_hp700 Makefile_inboard_ibm6000 Makefile_inboard_ibmpvs Makefile_inboard_intelnt Makefile_inboard_sgi Makefile_inboard_solaris Makefile_inboard_sun4 Makefile_intelnt Makefile_loadable_alphax Makefile_loadable_hp700 Makefile_loadable_ibm6000 Makefile_loadable_intelnt Makefile_loadable_sgi Makefile_loadable_solaris Makefile_os2 Makefile_outboard_alphax Makefile_outboard_aviion Makefile_outboard_hp700 Makefile_outboard_ibm6000 Makefile_outboard_ibmpvs Makefile_outboard_intelnt Makefile_outboard_sgi Makefile_outboard_solaris Makefile_outboard_sun4 README add.c hello.c hello_loadable.mdf makex.c user_inboard.mdf user_loadable.mdf user_outboard.mdf
+data_DATA =  Makefile_inboard_alphax Makefile_inboard_aviion Makefile_inboard_hp700 Makefile_inboard_ibm6000 Makefile_inboard_ibmpvs Makefile_inboard_intelnt Makefile_inboard_macos Makefile_inboard_sgi Makefile_inboard_solaris Makefile_inboard_sun4 Makefile_intelnt Makefile_loadable_alphax Makefile_loadable_hp700 Makefile_loadable_ibm6000 Makefile_loadable_intelnt Makefile_loadable_macos Makefile_loadable_sgi Makefile_loadable_solaris Makefile_os2 Makefile_outboard_alphax Makefile_outboard_aviion Makefile_outboard_hp700 Makefile_outboard_ibm6000 Makefile_outboard_ibmpvs Makefile_outboard_intelnt Makefile_outboard_sgi Makefile_outboard_solaris Makefile_outboard_sun4 README add.c hello.c hello_loadable.mdf makex.c user_inboard.mdf user_loadable.mdf user_outboard.mdf
 EXTRA_DIST = $(data_DATA)
diff -urN --exclude=*.orig dxsamples-4.3.2/user/Makefile.in dxsamples-new/user/Makefile.in
--- dxsamples-4.3.2/user/Makefile.in	Tue Oct 28 12:25:01 2003
+++ dxsamples-new/user/Makefile.in	Sat Jan 31 14:09:13 2004
@@ -102,7 +102,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 
-data_DATA = Makefile_inboard_alphax Makefile_inboard_aviion Makefile_inboard_hp700 Makefile_inboard_ibm6000 Makefile_inboard_ibmpvs Makefile_inboard_intelnt Makefile_inboard_sgi Makefile_inboard_solaris Makefile_inboard_sun4 Makefile_intelnt Makefile_loadable_alphax Makefile_loadable_hp700 Makefile_loadable_ibm6000 Makefile_loadable_intelnt Makefile_loadable_sgi Makefile_loadable_solaris Makefile_os2 Makefile_outboard_alphax Makefile_outboard_aviion Makefile_outboard_hp700 Makefile_outboard_ibm6000 Makefile_outboard_ibmpvs Makefile_outboard_intelnt Makefile_outboard_sgi Makefile_outboard_solaris Makefile_outboard_sun4 README add.c hello.c hello_loadable.mdf makex.c user_inboard.mdf user_loadable.mdf user_outboard.mdf
+data_DATA = Makefile_inboard_alphax Makefile_inboard_aviion Makefile_inboard_hp700 Makefile_inboard_ibm6000 Makefile_inboard_ibmpvs Makefile_inboard_intelnt Makefile_inboard_macos Makefile_inboard_sgi Makefile_inboard_solaris Makefile_inboard_sun4 Makefile_intelnt Makefile_loadable_alphax Makefile_loadable_hp700 Makefile_loadable_ibm6000 Makefile_loadable_intelnt Makefile_loadable_macos Makefile_loadable_sgi Makefile_loadable_solaris Makefile_os2 Makefile_outboard_alphax Makefile_outboard_aviion Makefile_outboard_hp700 Makefile_outboard_ibm6000 Makefile_outboard_ibmpvs Makefile_outboard_intelnt Makefile_outboard_sgi Makefile_outboard_solaris Makefile_outboard_sun4 README add.c hello.c hello_loadable.mdf makex.c user_inboard.mdf user_loadable.mdf user_outboard.mdf
 EXTRA_DIST = $(data_DATA)
 subdir = user
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
diff -urN --exclude=*.orig dxsamples-4.3.2/user/Makefile_inboard_macos dxsamples-new/user/Makefile_inboard_macos
--- dxsamples-4.3.2/user/Makefile_inboard_macos	Wed Dec 31 19:00:00 1969
+++ dxsamples-new/user/Makefile_inboard_macos	Sat Jan 31 14:01:08 2004
@@ -0,0 +1,22 @@
+FILES = user.o add.o makex.o hello.o
+ARCH = macos
+BASE = @PREFIX@/share/dx
+
+LDFLAGS= -L$(BASE)/lib_$(ARCH) -L@PREFIX@/lib -L/usr/X11R6/lib -bind_at_load
+
+CFLAGS = -O -Dmacos -I@PREFIX@/include
+include $(BASE)/lib_$(ARCH)/arch.mak
+LIBS = -lDX $(DXEXECLINKLIBS)
+BIN = $(BASE)/bin
+
+# make the Data Explorer executable
+dxexec:		$(FILES)
+		$(CC) $(LDFLAGS) $(FILES) $(LIBS) -o dxexec
+
+
+# make the user.c file
+user.c:		user_inboard.mdf
+		$(BIN)/mdf2c user_inboard.mdf > user.c
+
+clean:		
+		rm -rf user.c dxexec
diff -urN --exclude=*.orig dxsamples-4.3.2/user/Makefile_loadable_macos dxsamples-new/user/Makefile_loadable_macos
--- dxsamples-4.3.2/user/Makefile_loadable_macos	Wed Dec 31 19:00:00 1969
+++ dxsamples-new/user/Makefile_loadable_macos	Sat Jan 31 14:01:08 2004
@@ -0,0 +1,26 @@
+# This makefile has three targets:
+#    loadablelib creates a "library" of two run-time loadable modules,
+#       makex and add 
+#    hello creates a single runtime loadable executable, the hello module.
+#    all creates both of the above.
+BASE=@PREFIX@/share/dx
+ARCH=macos
+BIN = $(BASE)/bin
+CFLAGS = -O -D$(ARCH) -I$(BASE)/include -I@PREFIX@/include -fno-common
+BUNDLEFLAGS = -bundle -flat_namespace -undefined suppress
+
+.DEFAULT: 	all
+
+loadablelib: 	makex.o add.o
+		$(BIN)/mdf2c -m user_loadable.mdf > user.c
+		cc  $(CFLAGS) -D$(ARCH) -c user.c
+		cc $(BUNDLEFLAGS) -o loadablelib user.o makex.o add.o -e _DXEntry 
+
+hello: 		hello.o
+		$(BIN)/mdf2c -m hello_loadable.mdf > user.c
+		cc $(CFLAGS) -D$(ARCH) -c user.c
+		cc $(BUNDLEFLAGS) -o hello user.o hello.o -e _DXEntry 
+
+all:      	hello loadablelib 
+clean:	
+	rm -rf test.c *.o hello loadablelib
