diff -ruN fastlink/4.1P/src/Makefile fastlink.new/4.1P/src/Makefile
--- fastlink/4.1P/src/Makefile	Sat Jul  3 15:13:38 1999
+++ fastlink.new/4.1P/src/Makefile	Tue Feb 24 14:20:19 2004
@@ -44,7 +44,6 @@
 # Destination bin directory for executables.  Note: this
 # directory *must* exist, or compilation will fail during
 # linkage.
-BINDIR = ./
 
 # Compiler optimization flag.  Comment this out
 # if your compiler 
@@ -70,7 +69,7 @@
 #SYSDEP = -DSUNOS_4_1_X_GCC
 #SYSDEP = -DSOLARIS
 #SYSDEP = -D_POSIX_SOURCE
-#SYSDEP =
+SYSDEP = -Dmac
 
 
 ######################################################
diff -ruN fastlink/4.1P/src/commondefs.h fastlink.new/4.1P/src/commondefs.h
--- fastlink/4.1P/src/commondefs.h	Fri Sep  5 12:26:17 2003
+++ fastlink.new/4.1P/src/commondefs.h	Tue Feb 24 14:08:38 2004
@@ -33,7 +33,7 @@
 /* Shriram: end */
 
 /* cgh */
-#if !defined(vms)
+#if !defined(vms) && !defined(mac) 
 #include <malloc.h>
 #endif
 
diff -ruN fastlink/4.1P/src/unknown.h fastlink.new/4.1P/src/unknown.h
--- fastlink/4.1P/src/unknown.h	Fri Sep  5 12:26:26 2003
+++ fastlink.new/4.1P/src/unknown.h	Tue Feb 24 14:06:09 2004
@@ -17,7 +17,7 @@
 #include <string.h>
 
 /* cgh */
-#if !defined(vms)
+#if !defined(vms) && !defined(mac)
 #include <malloc.h>
 #endif
 
