--- gdl/src/Makefile.in.orig	Sat May 29 05:29:11 2004
+++ gdl/src/Makefile.in	Sat May 29 05:29:24 2004
@@ -134,7 +134,7 @@
 # the library search path.
 gdl_LDFLAGS = $(all_libraries)
 SUBDIRS = antlr
-gdl_LDADD = $(top_builddir)/src/antlr/libantlr.a -L/usr/X11R6/lib -lX11
+gdl_LDADD = $(top_builddir)/src/antlr/libantlr.a -L/usr/X11R6/lib -lX11 -lXext
 noinst_HEADERS = gdlwidget.hpp
 subdir = src
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
--- gdl/configure.orig	Mon Dec 13 23:48:10 2004
+++ gdl/configure	Tue Feb 15 09:28:38 2005
@@ -9215,10 +9215,10 @@
 
 if test "x$with_hdf" != "xno"; then
   if test "x$with_hdf" != "xyes"; then
-    LIBS="$LIBS -L$with_hdf/lib -lmfhdf -ldf -ljpeg -lz"
+    LIBS="$LIBS -L$with_hdf/lib -lmfhdf -ldf -ljpeg -lz -lsz"
     INCLUDES="$INCLUDES -I$with_hdf/include"
   else
-    LIBS="$LIBS -lmfhdf -ldf -ljpeg -lz"
+    LIBS="$LIBS -lmfhdf -ldf -ljpeg -lz -lsz"
   fi
 
   echo "$as_me:$LINENO: checking for Hopen in -ldf" >&5
