diff -ruN ghemical-1.01/src/graphics/objects.h ghemical-1.01-patched/src/graphics/objects.h
--- ghemical-1.01/src/graphics/objects.h	2001-12-29 09:56:49.000000000 -0500
+++ ghemical-1.01-patched/src/graphics/objects.h	2005-03-18 22:02:59.000000000 -0500
@@ -177,7 +177,7 @@
 {
 	protected:
 	
-	friend obj_loc_data * dummy_object::GetLocDataRW(void) const;
+	friend class dummy_object;
 	
 	obj_loc_data * data;
 	
--- ghemical-1.01/openbabel/fileformat.h~	2005-08-12 16:04:19.000000000 +0200
+++ ghemical-1.01/openbabel/fileformat.h	2005-08-12 16:04:56.000000000 +0200
@@ -122,7 +122,7 @@
 bool WriteQChem(std::ostream &,OBMol &);
 bool WriteReport(std::ostream &,OBMol &);
 bool WriteSDFile(std::ostream &,OBMol &,char *dimension="3D");
-bool WriteSmiles(std::ostream &,OBMol &,char *title=NULL);
+bool WriteSmiles(std::ostream &,OBMol &,char *title=(char*)NULL);
 //test
 bool WriteSmiOrderedMol2(std::ostream &,OBMol &,char *dimension="3D");
 //test
--- ghemical-1.01/src/common/utility.h~	2001-07-24 18:56:37.000000000 +0200
+++ ghemical-1.01/src/common/utility.h	2005-08-12 16:17:35.000000000 +0200
@@ -149,7 +149,7 @@
 	
 	public:
 	
-	singleton_cleaner(TYPE1 * p1 = NULL) { instance = p1; }
+	singleton_cleaner(TYPE1 * p1 = (TYPE1 *)NULL) { instance = p1; }
 	~singleton_cleaner(void) { if (instance) delete instance; }
 	
 	bool SetInstance(TYPE1 * p1) { if (instance) return false; instance = p1; return true; }
