diff -ur nodelib-orig/src/xalloc.c nodelib/src/xalloc.c
--- nodelib-orig/src/xalloc.c	Wed Jan  5 03:32:03 2000
+++ nodelib/src/xalloc.c	Wed Jun 26 12:46:22 2002
@@ -3,7 +3,7 @@
 
 #include <stdio.h>
 #include <string.h>
-#include <malloc.h>
+ 
 #include <memory.h>
 
 #undef XALLOC_DEBUG
--- nodelib-orig/src/ulog.c	2002-09-25 18:04:20.000000000 -0700
+++ nodelib/src/ulog.c	2002-09-25 18:04:42.000000000 -0700
@@ -45,7 +45,8 @@
 
 static void expand_pct_m(char *format, int pfsz)
 {
-  extern int errno, sys_nerr;
+  extern int errno;
+  extern const int sys_nerr;
   extern const char *const sys_errlist[];
   const char *err;
   char *buf = ulog_fmt;
