--- numarray/addons.py.orig	Thu Mar 17 13:19:04 2005
+++ numarray/addons.py	Thu Mar 17 13:27:38 2005
@@ -44,6 +44,7 @@
     if os.path.exists('/System/Library/Frameworks/vecLib.framework'):
         # use Apple's optimized BLAS implementation
         lapack_libs = []
+        lapack_dirs = []
         lapack_link_args = ['-framework', 'vecLib']
     elif USE_ABSOFT:  # Absoft Fortran
         lapack_dirs = ['/usr/local/lib/atlas', '/opt/absoft/lib']
@@ -56,7 +57,6 @@
  
     # Set to list directories to be searched for BLAS and LAPACK libraries
     lapack_compile_args = []
-    lapack_link_args = []
     lapack_include_dirs = ["Packages/LinearAlgebra2/Src"]
     
     if os.environ.has_key('LINALG_LIB'):
