--- procheck/nb.c.org   2005-12-09 21:54:59.000000000 -0500
+++ procheck/nb.c       2005-12-09 22:16:12.000000000 -0500
@@ -1987,6 +1987,18 @@
 
 /*============================================================================*/
 
+char *getsafe(char *buffer, int count)
+{
+   char *result = buffer, *np;
+   if ((buffer == NULL) || (count < 1))
+      result = NULL;
+   else if (count == 1)
+      *result = '\0';
+   else if ((result = fgets(buffer, count, stdin)) != NULL)
+      if (np = strchr(buffer, '\n'))
+         *np = '\0';
+   return result;
+}
 
 main()
 {
@@ -2009,7 +2021,7 @@
 /* CHECK v.2.0
     printf("\nEnter table output filename (or '@' for multiple files):");
 
-    gets(tmpstr);
+    getsafe(tmpstr,sizeof(tmpstr));
     sscanf(tmpstr, "%s", tablefn);
     if (tablefn[0] != '@')
     {
@@ -2027,7 +2039,7 @@
         printf("  @file name for a file holding names of Brookhaven ");
         printf("format files.\n");
         printf("If you've had enough, a blank line lets you escape!\n\n");
-        gets(tmpstr);
+        getsafe(tmpstr,sizeof(tmpstr));
         if (tmpstr[0] == '\0' || tmpstr[0] == '\n')
             break;
         if (tmpstr[0] == '@')
@@ -2057,7 +2069,7 @@
 
 /* CHECK v.2.0 */
     printf("\nEnter the name of original coordinates file\n");
-    gets(tmpstr);
+    getsafe(tmpstr,sizeof(tmpstr));
     if (tmpstr[0] != '\0' && tmpstr[0] != '\n')
     {
         ierror = getnam(tmpstr,&istart,&iend);
diff -uNr procheck/Makefile procheck.new/Makefile
--- procheck/Makefile	Thu Jan 11 11:16:16 2001
+++ procheck.new/Makefile	Tue Aug 31 07:37:29 2004
@@ -3,11 +3,11 @@
 #
 # Andrew C.R. Martin, University of Reading
 #
-F77   = f77  # Your FORTRAN compiler
+F77   = g77  # Your FORTRAN compiler
 CC    = cc   # Your C compiler
 CLIBS = -lm  # Your C maths library
-COPTS = -g   # Compile/link options for C
-FOPTS = -g   # Compile/link options for FORTRAN
+COPTS = -O3  # Compile/link options for C
+FOPTS = -O3  # Compile/link options for FORTRAN
 
 ##########################################################################
 # Shouldn't need to alter anything below here
diff -uNr procheck/setup.scr procheck.new/setup.scr
--- procheck/setup.scr	Wed Jan 10 08:26:23 2001
+++ procheck.new/setup.scr	Tue Aug 31 07:39:31 2004
@@ -8,8 +8,8 @@
 
 # PROCHECK environments and aliases
 # ---------------------------------
-set    prodir =  @@_procheck_program_directory_@@
-setenv prodir   '@@_procheck_program_directory_@@'
+set    prodir =  @PREFIX@/share/procheck
+setenv prodir   '@PREFIX@/share/procheck'
 alias procheck      $prodir'/procheck.scr'
 alias procheck_comp $prodir'/procheck_comp.scr'
 alias procheck_nmr  $prodir'/procheck_nmr.scr'
--- procheck/setup.sh   2005-12-12 20:07:07.000000000 -0500
+++ procheck/setup.sh.new       2005-12-12 20:05:40.000000000 -0500
@@ -8,9 +8,9 @@
 
 # PROCHECK environments and aliases
 # ---------------------------------
-prodir=@@_procheck_program_directory_@@
+prodir=@PREFIX@/share/procheck
 export prodir
-alias procheck= $prodir'/procheck.scr'
+alias procheck=$prodir'/procheck.scr'
 alias procheck_comp=$prodir'/procheck_comp.scr'
 alias procheck_nmr=$prodir'/procheck_nmr.scr'
 alias proplot=$prodir'/proplot.scr'

