--- root.orig/configure	2005-09-18 18:09:57.000000000 -0500
+++ root.new/configure	2005-09-21 12:14:53.000000000 -0500
@@ -1895,17 +1895,17 @@
     # for a system library, then see if we have various headers needed.
     if test "x$enable_builtin_afterimage" = "xyes" ; then
        check_header "jpeglib.h" "" $ASIMAGE $ASIMAGE/include \
-          /usr/local/include /usr/include /opt/include
+          /usr/local/include /usr/include /opt/include $finkdir/include
        asjpegincdir=$found_dir
        check_header "png.h" "" $ASIMAGE $ASIMAGE/include \
           /usr/local/include /usr/include /usr/local/include/libpng \
-          /opt/include
+          /opt/include $finkdir/include
        aspngincdir=$found_dir
        check_header "tiffio.h" "" $ASIMAGE $ASIMAGE/include \
-          /usr/local/include /usr/include /opt/include
+          /usr/local/include /usr/include /opt/include $finkdir/include
        astiffincdir=$found_dir
        check_header "gif_lib.h" "" $ASIMAGE $ASIMAGE/include \
-          /usr/local/include /usr/include /opt/include
+          /usr/local/include /usr/include /opt/include $finkdir/include
        asgifincdir=$found_dir
 
        asextralib=""
@@ -1918,7 +1918,7 @@
        #for k in libjpeg libtiff libungif libz libpng ; do
        for k in libjpeg libtiff libz libpng ; do
            check_library $k "$enable_shared" "" \
-             $ASIMAGE $ASIMAGE/lib /usr/local/lib /usr/lib /opt/lib
+             $ASIMAGE $ASIMAGE/lib /usr/local/lib /usr/lib /opt/lib $finkdir/lib
            if test "x$k" = "xlibz" ; then
                found_libz=$found_lib
                found_dirz=$found_dir
--- root.orig/config/Makefile.macosx	2005-05-10 10:02:41.000000000 -0500
+++ root.new/config/Makefile.macosx	2005-11-02 06:59:32.000000000 -0600
@@ -92,8 +92,8 @@
 endif
 
 # Override default in Makefile.config
-INSTALL      := cp -pPR
-INSTALLDATA  := cp -pPR
+INSTALL      := /bin/cp -pPR
+INSTALLDATA  := /bin/cp -pPR
 
 # Extras
 ifeq ($(NEEDFINK),yes)
