--- rpy-0.3.4/setup.py.orig	Thu Jan 29 10:19:27 2004
+++ rpy-0.3.4/setup.py	Thu Jan 29 10:19:36 2004
@@ -61,6 +61,11 @@
     libraries=["Rdll","Advapi32"]
     library_dirs = [r_libs, "C:\rpy\bin"]
     extra_compile_args=[]
+elif sys.platform=='darwin':
+    include_dirs = [ os.path.join(RHOME.strip(), 'include'), 'src' ]
+    libraries=["R"]
+    library_dirs = [r_libs]
+    extra_compile_args=[]
 else:
     include_dirs = [ os.path.join(RHOME.strip(), 'include'), 'src' ]
     libraries=["R"]
