diff -Nurd -x'*~' ghex-2.8.2.orig/Makefile.in ghex-2.8.2/Makefile.in
--- ghex-2.8.2.orig/Makefile.in	2006-03-07 14:54:48.000000000 -0500
+++ ghex-2.8.2/Makefile.in	2006-03-07 17:32:04.000000000 -0500
@@ -800,8 +800,9 @@
 @INTLTOOL_DESKTOP_RULE@
 
 #Install the ghex2 schema file
-install-data-local:
-	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(srcdir)/$(schema_DATA)
+@GCONF_SCHEMAS_INSTALL_TRUE@install-data-local:
+@GCONF_SCHEMAS_INSTALL_TRUE@	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(srcdir)/$(schema_DATA)
+@GCONF_SCHEMAS_INSTALL_FALSE@install-data-local:
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
