diff -Naur mew-5.2.orig/mew-pop.el mew-5.2/mew-pop.el
--- mew-5.2.orig/mew-pop.el	Thu Jan  1 09:00:00 1970
+++ mew-5.2/mew-pop.el	Mon Mar 14 19:29:36 2005
@@ -77,7 +77,7 @@
 		  "Auth'ing"
 		  nil
 		  (mew-pop-secure-p pnm))
-  (if (re-search-forward "<[^<>]+@[^<>]+>" nil t)
+  (if (re-search-forward "<[=!-;?-~]+@[=!-;?-~]+>" nil t)
       (mew-pop-set-key pnm (mew-match-string 0)))
   (mew-pop-process-send-string pro "CAPA"))
 
diff -Naur mew-4.2.orig/fink/mew-install mew-4.2/fink/mew-install
--- mew-4.2.orig/fink/mew-install	Thu Jan  1 09:00:00 1970
+++ mew-4.2/fink/mew-install	Mon Mar 14 19:29:36 2005
@@ -0,0 +1,191 @@
+#!/bin/sh
+# @PREFIX@/lib/emacsen-common/packages/install/mew
+set -e
+
+abort_install() {
+  echo "Byte compile for ${FLAVOR} is failed" >&2
+  echo "Please check ${ELCDIR}/CompilationLog" >&2
+  exit 1
+}
+
+FLAVOR=$1
+VERSION=@VERSION@
+PACKAGE=mew
+
+if [ -z "${FLAVOR}" ]; then exit 0; fi
+if [ "${FLAVOR}" = emacs ]; then exit 0; fi
+if [ "${FLAVOR}" = emacs19 ]; then exit 0; fi
+if [ "${FLAVOR}" = mule2 ]; then exit 0; fi
+
+ELDIR=@PREFIX@/share/emacs/site-lisp/mew
+ELCDIR_BASE=@PREFIX@/share/${FLAVOR}/site-lisp/mew
+FLAGS="-no-site-file -q -batch -l __path.el -f batch-byte-compile"
+
+EMACSEN_MULE=${FLAVOR}
+EMACSEN_NOMULE=""
+
+FLAVOR_SUFFIX=`echo ${FLAVOR} | cut -s -d - -f 2`
+FLAVORTEST=`echo ${FLAVOR} | sed -e s/^s// | cut -c-6`
+if [ "X${FLAVORTEST}" = "Xxemacs" ]; then
+  FLAVOR_BODY=`echo ${FLAVOR} | cut -d - -f 1`
+  if [ "X${FLAVOR_SUFFIX}" = "Xmule" ]; then
+    EMACSEN_MULE=`ls @PREFIX@/bin/${FLAVOR_BODY}*-mule* 2>/dev/null | head -n 1`
+    if [ -n "${EMACSEN_MULE}" ]; then
+      EMACSEN_MULE=`basename ${EMACSEN_MULE}`
+    fi
+    EMACSEN_NOMULE=""
+  elif [ "X${FLAVOR_SUFFIX}" = "Xnomule" ]; then
+    EMACSEN_MULE=""
+    EMACSEN_NOMULE=`ls @PREFIX@/bin/${FLAVOR_BODY}*-nomule* 2>/dev/null | head -n 1`
+    if [ -n "${EMACSEN_NOMULE}" ]; then
+      EMACSEN_NOMULE=`basename ${EMACSEN_NOMULE}`
+    fi
+  else
+    FLAVOR_SUFFIX=""
+    EMACSEN_MULE=`ls @PREFIX@/bin/${FLAVOR}*-mule* 2>/dev/null | head -n 1`
+    if [ -n "${EMACSEN_MULE}" ]; then
+      EMACSEN_MULE=`basename ${EMACSEN_MULE}`
+    fi
+    EMACSEN_NOMULE=`ls @PREFIX@/bin/${FLAVOR}*-nomule* 2>/dev/null | head -n 1`
+    if [ -n "${EMACSEN_NOMULE}" ]; then
+      EMACSEN_NOMULE=`basename ${EMACSEN_NOMULE}`
+    fi
+  fi
+fi
+
+COMPILED=no
+for EMACSEN in "${EMACSEN_MULE}" "${EMACSEN_NOMULE}"; do
+  if [ -n "${EMACSEN}" ]; then
+    ELCDIR=${ELCDIR_BASE}
+    if [ "X${FLAVORTEST}" = "Xxemacs" -a -z "${FLAVOR_SUFFIX}" ]; then
+      if [ "X${EMACSEN}" = "X${EMACSEN_MULE}" ]; then
+	/bin/echo "install/${PACKAGE}: Byte-compiling for ${FLAVOR} (mule) ...\c"
+        ELCDIR=${ELCDIR}/mule
+      else
+	/bin/echo "install/${PACKAGE}: Byte-compiling for ${FLAVOR} (nomule) ...\c"
+        ELCDIR=${ELCDIR}/nomule
+      fi
+    else
+      /bin/echo "install/${PACKAGE}: Byte-compiling for ${FLAVOR} ...\c"
+    fi
+    STAMP="${ELCDIR}/compile-stamp"
+    if [ -e "${STAMP}" ]; then
+      if [ "X${VERSION}" = X"`cat ${STAMP}`" ]; then
+	echo " skipped. (already compiled)" 
+      else
+	rm -f ${STAMP}
+      fi
+    fi
+    if [ ! -e "${STAMP}" ]; then
+      if [ "X${FLAVORTEST}" = "Xxemacs" -a -z "${FLAVOR_SUFFIX}" ]; then
+	rm -f ${ELCDIR_BASE}/*.el ${ELCDIR_BASE}/*.elc
+      fi
+      rm -rf ${ELCDIR}
+      install -m 755 -d ${ELCDIR}
+      cd ${ELDIR}
+      cp *.el contrib/*.el ${ELCDIR}/
+      cd ${ELCDIR}
+      if [ "X${EMACSEN}" != "X${EMACSEN_MULE}" ]; then
+	rm -f mew-lang-jp.el mew-mule3.el
+      fi
+      rm -f auxiliary.el
+      FILES=`ls -1 *.el | grep -v '^mew-nmz'`
+      cat << EOF > __path.el
+(setq load-path (cons "." load-path))
+(setq byte-compile-warnings nil)
+(setq mew-compiling t)
+EOF
+      ${EMACSEN} ${FLAGS} ${FILES} >CompilationLog 2>&1 || abort_install
+      if [ "X${EMACSEN}" != "X${EMACSEN_MULE}" ]; then
+	rm -f mew-nmz*.el
+      else
+	# compile mew-nmz*.el if possible
+	if [ -e @PREFIX@/share/${FLAVOR}/site-lisp/w3m/w3m-namazu.elc ]; then
+	  cat << EOF >> __path.el
+(setq load-path (cons "@PREFIX@/share/${FLAVOR}/site-lisp/w3m" load-path))
+EOF
+	  ${EMACSEN} ${FLAGS} mew-nmz*.el >>CompilationLog 2>&1 || abort_install
+	  rm -f mew-nmz*.el
+	else
+	  rm -f mew-nmz*.el
+	  ln -sf ${ELDIR}/mew-nmz*.el .
+	  ln -sf ${ELDIR}/contrib/mew-nmz*.el .
+	  rm -f 'mew-nmz*.el'
+	fi
+      fi
+      rm -f etc
+      ln -sf ${ELDIR}/etc .
+      rm -f ${FILES} __path.el
+      ( for d in "${ELDIR}" "${ELDIR}/contrib"; do
+	  if [ -d ${d} ]; then
+	    cd ${d}
+	    for f in *.el; do
+	      if [ -f ${ELCDIR}/${f}c ]; then
+		ln -sf ${d}/${f} ${ELCDIR}/${f}
+	      fi
+	    done
+	  fi
+	done
+      )
+      gzip -9 CompilationLog
+      chmod 644 CompilationLog.gz
+      if [ -f mew.elc ]; then
+        chmod 644 *.elc
+	echo "${VERSION}" > ${STAMP}
+        chmod 644 ${STAMP}
+	COMPILED=yes
+      fi
+      if [ "X${FLAVORTEST}" = "Xxemacs" -a -z "${FLAVOR_SUFFIX}" ]; then
+	if [ -e ${ELCDIR_BASE}/mule/mew.elc ]; then
+	  ln -sf ${ELCDIR_BASE}/mule/mew* ${ELCDIR_BASE}/
+	elif [ -e ${ELCDIR_BASE}/nomule/mew.elc ]; then
+	  ln -sf ${ELCDIR_BASE}/nomule/mew* ${ELCDIR_BASE}/
+	fi
+      fi
+      echo " done."
+    fi
+  fi
+done
+
+if [ "X${COMPILED}" = Xyes ]; then
+  RECOMPILE_MHC=no
+  RECOMPILE_W3MEL=no
+  if [ -d @PREFIX@/share/${FLAVOR}/site-lisp/mhc ]; then
+    RECOMPILE_MHC=yes
+  fi
+  if [ -d @PREFIX@/share/${FLAVOR}/site-lisp/w3m ]; then
+    RECOMPILE_W3MEL=yes
+  fi
+  if [ "X${RECOMPILE_MHC}" = Xyes ]; then
+    # remove mhc before removing emacs-w3m to prevent redundant byte-compiling
+    pkg=mhc
+    f=@PREFIX@/lib/emacsen-common/packages/remove/${pkg}
+    if [ -f ${f} ]; then
+      ${f} ${FLAVOR}
+    fi
+  fi
+  if [ "X${RECOMPILE_W3MEL}" = Xyes ]; then
+    pkg=emacs-w3m
+    f=@PREFIX@/lib/emacsen-common/packages/remove/${pkg}
+    if [ -f ${f} ]; then
+      ${f} ${FLAVOR}
+    fi
+    # install emacs-w3m before installing mhc to prevent redundant byte-compiling
+    pkg=emacs-w3m
+    f=@PREFIX@/lib/emacsen-common/packages/install/${pkg}
+    if [ -f ${f} ]; then
+      ${f} ${FLAVOR}
+    fi
+  fi
+  if [ "X${RECOMPILE_MHC}" = Xyes ]; then
+    pkg=mhc
+    if [ ! -d @PREFIX@/share/${FLAVOR}/site-lisp/${pkg} ]; then
+      f=@PREFIX@/lib/emacsen-common/packages/install/${pkg}
+      if [ -f ${f} ]; then
+	${f} ${FLAVOR}
+      fi
+    fi
+  fi
+fi
+
+exit 0
diff -Naur mew-4.2.orig/fink/mew-remove mew-4.2/fink/mew-remove
--- mew-4.2.orig/fink/mew-remove	Thu Jan  1 09:00:00 1970
+++ mew-4.2/fink/mew-remove	Mon Mar 14 17:00:31 2005
@@ -0,0 +1,19 @@
+#!/bin/sh
+# @PREFIX@/lib/emacsen-common/packages/remove/mew
+set -e
+
+FLAVOR=$1
+PACKAGE=mew
+
+if [ -z "${FLAVOR}" ]; then exit 0; fi
+if [ "${FLAVOR}" = emacs ]; then exit 0; fi
+if [ "${FLAVOR}" = emacs19 ]; then exit 0; fi
+if [ "${FLAVOR}" = mule2 ]; then exit 0; fi
+
+ELCDIR=@PREFIX@/share/${FLAVOR}/site-lisp/mew
+
+/bin/echo "remove/${PACKAGE}: Handling removal of emacsen flavor ${FLAVOR} ...\c"
+rm -rf ${ELCDIR}
+echo " done."
+
+exit 0
diff -Naur mew-4.2.orig/fink/mew-startup mew-4.2/fink/mew-startup
--- mew-4.2.orig/fink/mew-startup	Thu Jan  1 09:00:00 1970
+++ mew-4.2/fink/mew-startup	Thu Mar 24 16:13:05 2005
@@ -0,0 +1,39 @@
+;;; 50mew.el --- Fink mew startup file  -*-mode: emacs-lisp;-*-
+
+;;; Code:
+
+(let ((elc-dir (concat "@PREFIX@/share/" (symbol-name fink-emacs-flavor) "/site-lisp/mew")))
+  (if (or (not (file-exists-p "@PREFIX@/lib/emacsen-common/packages/install/mew"))
+	  (not (file-directory-p elc-dir)))
+      ()
+    (if (featurep 'xemacs)
+	(if (featurep 'mule)
+	    (and (file-directory-p (concat elc-dir "/mule"))
+		 (setq elc-dir (concat elc-dir "/mule")))
+	  (and (file-directory-p (concat elc-dir "/nomule"))
+	       (setq elc-dir (concat elc-dir "/nomule")))))
+    (if (and nil (fboundp 'fink-pkg-add-load-path-item)) ; fink-pkg-* is not
+	(fink-pkg-add-load-path-item elc-dir)            ; correctly working.
+      (setq load-path (cons elc-dir load-path)))
+    (autoload 'mew-user-agent-compose "mew" nil t)
+    (if (fboundp 'define-mail-user-agent)
+	(define-mail-user-agent
+	  'mew-user-agent
+	  'mew-user-agent-compose
+	  'mew-draft-send-message
+	  'mew-draft-kill
+	  'mew-send-hook))
+    (autoload 'mew "mew" nil t)
+    (autoload 'mew-send "mew" nil t)
+    (if (featurep 'xemacs)
+	(eval-after-load "mew" '(require 'timer)))
+    (setq mew-icon-directory "@PREFIX@/share/pixmaps/mew")
+    (if (file-executable-p "@PREFIX@/sbin/stunnel4")
+	(setq mew-prog-ssl "@PREFIX@/sbin/stunnel4")
+      (setq mew-prog-ssl "@PREFIX@/sbin/stunnel"))
+    (setq mew-ssl-cert-directory "@PREFIX@/etc/ssl/certs")
+    (setq mew-ssl-verify-level 2)
+    ;;
+    ))
+
+;;; 50mew.el ends here
