diff -Nurd tea-14.2-orig/Makefile.am tea-14.2/Makefile.am
--- tea-14.2-orig/Makefile.am	2005-05-10 10:09:39.000000000 +0200
+++ tea-14.2/Makefile.am	2006-08-17 01:13:11.000000000 +0200
@@ -6,11 +6,35 @@
 	autogen.sh \
 	doc
      
+if FINK
+pkgdocdir = $(datadir)/doc/@PACKAGE@
+endif
+
 if DEB
 install-data-local:
 	@$(NORMAL_INSTALL)
 	$(mkinstalldirs) $(DESTDIR)$(pkgdatadir)/doc; 
 else
+if FINK
+install-data-local:
+	@$(NORMAL_INSTALL)
+	$(mkinstalldirs) $(DESTDIR)$(pkgdocdir); \
+	cp -r $(srcdir)/doc/* $(DESTDIR)$(pkgdocdir); \
+	cp $(srcdir)/AUTHORS $(DESTDIR)$(pkgdocdir); \
+	cp $(srcdir)/ChangeLog $(DESTDIR)$(pkgdocdir); \
+	cp $(srcdir)/README $(DESTDIR)$(pkgdocdir); \
+	cp $(srcdir)/NEWS $(DESTDIR)$(pkgdocdir); \
+	cp $(srcdir)/COPYING $(DESTDIR)$(pkgdocdir); \
+	cp $(srcdir)/TODO $(DESTDIR)$(pkgdocdir); \
+	if test -d $(srcdir)/pixmaps; then \
+	  $(mkinstalldirs) $(DESTDIR)$(pkgdatadir)/pixmaps; \
+	  for pixmap in $(srcdir)/pixmaps/*; do \
+	    if test -f $$pixmap; then \
+	      $(INSTALL_DATA) $$pixmap $(DESTDIR)$(pkgdatadir)/pixmaps; \
+	    fi \
+	  done \
+	fi 
+else 
 install-data-local:
 	@$(NORMAL_INSTALL)
 	$(mkinstalldirs) $(DESTDIR)$(pkgdatadir)/doc; \
@@ -30,6 +54,7 @@
 	  done \
 	fi 
 endif 
+endif
 
 dist-hook:
 	if test -d $(srcdir)/pixmaps; then \
@@ -41,6 +66,11 @@
 	  done \
 	fi
 	
+if FINK
+uninstall-local:
+	rm -f -r -d $(DESTDIR)$(pkgdatadir); \
+	rm -f -r -d $(DESTDIR)$(pkgdocdir);
+else
 uninstall-local:
 	rm -f -r -d $(DESTDIR)$(pkgdatadir);
-
+endif
diff -Nurd tea-14.2-orig/configure.in tea-14.2/configure.in
--- tea-14.2-orig/configure.in	2006-08-15 21:14:01.000000000 +0200
+++ tea-14.2/configure.in	2006-08-18 16:02:33.000000000 +0200
@@ -35,13 +35,15 @@
 
 AM_CONDITIONAL(DEB, test "$enable_debian" = yes)
 
+AM_CONDITIONAL(FINK, test "x$OSTYPE" = "xDARWIN")
+
 AC_CHECK_HEADER(aspell.h, 
                AC_CHECK_LIB(aspell, new_aspell_config, 
                LIBS="$LIBS -laspell"
                AC_DEFINE(HAVE_LIBASPELL, 1, [use aspell])
                ))
 
-CFLAGS="$CFLAGS -w -s -pipe"
+dnl CFLAGS="$CFLAGS -w -s -pipe"
 dnl CFLAGS="$CFLAGS -s -pipe"
 
 dnl Add DARWIN flag for Mac OS X
@@ -51,13 +53,29 @@
    CFLAGS="$CFLAGS -w -s -pipe"
 fi
 
+AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
+if test "x$PKG_CONFIG" = "xno" ; then
+	AC_MSG_ERROR([pkg-config not found please install pkg-config])
+fi
 
 dnl pkg_modules="gthread-2.0 >= 2.0.0"
 dnl PKG_CHECK_MODULES(PACKAGE, [$pkg_modules])
 
-pkg_modules="gtk+-2.0 >= 2.4.0"
-PKG_CHECK_MODULES(PACKAGE, [$pkg_modules])
+dnl pkg_modules="gtk+-2.0 >= 2.4.0"
+dnl PKG_CHECK_MODULES(PACKAGE, [$pkg_modules])
 
+echo -n "checking for gtk+-2.0..."
+if pkg-config --exists gtk+-2.0 ; then
+	if pkg-config gtk+-2.0 --atleast-version=2.4.0 ; then
+		LIBS="$LIBS `pkg-config --libs gtk+-2.0`"
+		CFLAGS="$CFLAGS `pkg-config --cflags gtk+-2.0`"
+		echo "yes"
+	else
+		echo "no"
+	fi
+else
+	echo "no"
+fi
 
 dnl check for IPv6 option
 dnl AC_ARG_ENABLE(ipv6,
@@ -76,8 +94,18 @@
 
         AC_DEFINE(CHAI_MODE, 1, [chai mode])
 
-	pkg_modules="gtksourceview-1.0 >= 1.0"
-	PKG_CHECK_MODULES(PACKAGE, [$pkg_modules])
+	echo -n "checking for gtksourceview-1.0..."
+	if pkg-config --exists gtksourceview-1.0 ; then
+		if pkg-config gtksourceview-1.0 --atleast-version=1.0 ; then
+			LIBS="$LIBS `pkg-config --libs gtksourceview-1.0`"
+			CFLAGS="$CFLAGS `pkg-config --cflags gtksourceview-1.0`"
+			echo "yes"
+		else
+			echo "no"
+		fi
+	else
+		echo "no"
+	fi
 
 	echo -n "checking for gnome-vfs-2.0... "
 	if pkg-config --exists gnome-vfs-2.0 ; then
@@ -113,6 +141,10 @@
 AC_SUBST(PACKAGE_CFLAGS)
 AC_SUBST(PACKAGE_LIBS)
 
+dnl Add the languages which your application supports here.
+ALL_LINGUAS="ru ja uk sr pl de cs lv es fr"
+AM_GLIB_GNU_GETTEXT
+
 if test -n "$enable_debian"; then
 GETTEXT_PACKAGE=teaed
 else
@@ -122,10 +154,6 @@
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE",1)
 
-dnl Add the languages which your application supports here.
-ALL_LINGUAS="ru ja uk sr pl de cs lv es fr"
-AM_GLIB_GNU_GETTEXT
-
 AC_OUTPUT([
 Makefile
 src/Makefile
diff -Nurd tea-14.2-orig/po/fr.po tea-14.2/po/fr.po
--- tea-14.2-orig/po/fr.po	2006-08-15 21:24:45.000000000 +0200
+++ tea-14.2/po/fr.po	2006-08-18 01:00:21.000000000 +0200
@@ -9,7 +9,7 @@
 "Project-Id-Version: tea 14.1.1\n"
 "Report-Msgid-Bugs-To: \n"
 "POT-Creation-Date: 2006-08-15 22:24+0300\n"
-"PO-Revision-Date: 2006-08-14 09:48+0200\n"
+"PO-Revision-Date: 2006-08-18 00:59+0200\n"
 "Last-Translator: Michèle Garoche <michele.garoche@easyconnect.fr>\n"
 "Language-Team: français <fr@li.org>\n"
 "MIME-Version: 1.0\n"
@@ -38,7 +38,7 @@
 
 #: src/interface.c:163
 msgid "Close a current document"
-msgstr "Fermer le document courant"
+msgstr "Ferme le document courant"
 
 #: src/interface.c:167
 msgid "File"
@@ -1077,7 +1077,7 @@
 
 #: src/interface.c:924
 msgid "Self check"
-msgstr "Autovérification"
+msgstr "Vérification automatique"
 
 #: src/interface.c:995
 msgid "Click here to search this through the text"
@@ -1238,7 +1238,7 @@
 
 #: src/callbacks.c:5039 src/callbacks.c:5204
 msgid "Create new"
-msgstr "Créer un nouveau document"
+msgstr "Créer un nouveau répertoire"
 
 #: src/callbacks.c:5116
 msgid "Save as..."
@@ -1718,7 +1718,7 @@
 
 #: src/tea_options.c:534
 msgid "Defaults"
-msgstr "Tables de caractère à utiliser par défaut"
+msgstr "Jeux de caractères à utiliser par défaut"
 
 #: src/tea_options.c:542
 msgid "Default charset for File open"
@@ -1742,7 +1742,7 @@
 
 #: src/tea_options.c:557
 msgid "Autodetect"
-msgstr "Autodétection"
+msgstr "Détection automatique"
 
 #: src/tea_options.c:566
 msgid "CP 1251"
@@ -1811,15 +1811,15 @@
 
 #: src/tea_options.c:589
 msgid "Available"
-msgstr "Disponible"
+msgstr "Disponibles"
 
 #: src/tea_options.c:595
 msgid "Selected"
-msgstr "Sélection"
+msgstr "Choisis"
 
 #: src/tea_options.c:598
 msgid "Configure charsets"
-msgstr "Configuration des encodages"
+msgstr "Configuration des jeux de caractères"
 
 #: src/tea_options.c:617
 msgid "External image editor"
@@ -1879,7 +1879,7 @@
 
 #: src/tea_options.c:679
 msgid "Show the main toolbar"
-msgstr "Afficher la barre d'icône principale"
+msgstr "Afficher la barre d'outils principale"
 
 #: src/tea_options.c:680
 msgid "Show the common close button"
@@ -1887,7 +1887,7 @@
 
 #: src/tea_options.c:681
 msgid "Show the cursor position at statusbar"
-msgstr ""
+msgstr "Afficher la position du curseur dans la barre de statut"
 
 #: src/tea_options.c:683
 msgid "Start in the full-screen mode"
diff -Nurd tea-14.2-orig/src/tea_defs.h tea-14.2/src/tea_defs.h
--- tea-14.2-orig/src/tea_defs.h	2006-06-30 13:05:04.000000000 +0200
+++ tea-14.2/src/tea_defs.h	2006-08-15 23:35:31.000000000 +0200
@@ -116,7 +116,16 @@
 
 #else
 
+#if defined(DARWIN)
+
+#define TEA_DOC_DIR PACKAGE_DATA_DIR"/doc/tea/"
+
+#else
+
 #define TEA_DOC_DIR PACKAGE_DATA_DIR"/tea/doc/"
+
+#endif
+
 #define TEA_PIX_MAINLOGO PACKAGE_DATA_DIR"/tea/pixmaps/tea_logo.jpg"
 #define TEA_PIX_MAINICON PACKAGE_DATA_DIR"/tea/pixmaps/tea_icon_v2.png"
 #define TEA_PIX_DIR PACKAGE_DATA_DIR"/tea/pixmaps/"
@@ -124,4 +133,4 @@
 
 #endif
 
-#define doc_get_buf(x) get_text_buf(x)
\ No newline at end of file
+#define doc_get_buf(x) get_text_buf(x)
diff -Nurd tea-14.2-orig/src/tea_funx.h tea-14.2/src/tea_funx.h
--- tea-14.2-orig/src/tea_funx.h	2006-08-15 13:35:38.000000000 +0200
+++ tea-14.2/src/tea_funx.h	2006-08-16 01:37:31.000000000 +0200
@@ -71,3 +71,8 @@
 gchar* file_replace_path (const gchar *filename, const gchar *new_dir);
 gchar* file_combine_path (const gchar *dir, const gchar *filename);
 gchar* add_slash_if_need (const gchar *dir);
+gboolean is_markup (const gchar *f);
+gboolean is_c (const gchar *f);
+gboolean is_pascal (const gchar *f);
+gchar* compose_browser_cmd (const gchar *filename);
+GList* image_formats_fill (void);
diff -Nurd tea-14.2-orig/src/tea_options.c tea-14.2/src/tea_options.c
--- tea-14.2-orig/src/tea_options.c	2006-08-15 13:35:38.000000000 +0200
+++ tea-14.2/src/tea_options.c	2006-08-18 04:06:35.000000000 +0200
@@ -333,7 +333,17 @@
   glist_strings_free (gl_enc_choosed);
   gl_enc_choosed = l;
 
-  glist_save_to_file (gl_enc_choosed, confile.tea_choosed_charsets);
+  // Need to test whether the list is empty or not
+  // If it is empty, remove the charsets file in home/.config/tea
+  // If it is not empty, save the new charsets set
+  if (g_list_length(gl_enc_choosed) == 0)
+  {
+	  gchar *cm = g_strconcat ("rm -f ", confile.tea_choosed_charsets, NULL);
+	  system (cm);
+	  g_free (cm);
+  }
+  else
+	  glist_save_to_file (gl_enc_choosed, confile.tea_choosed_charsets);
   get_iconv_sup ();
   tv_fill_with_glist (tv_choosed_charsets, gl_enc_choosed);
 }
diff -Nurd tea-14.2-orig/src/tea_text_document.c tea-14.2/src/tea_text_document.c
--- tea-14.2-orig/src/tea_text_document.c	2006-08-15 13:46:35.000000000 +0200
+++ tea-14.2/src/tea_text_document.c	2006-08-19 00:07:58.000000000 +0200
@@ -4501,27 +4501,36 @@
       gnome_vfs_file_info_unref (info);
       return;
      }
+  
+  GtkSourceLanguage *l = NULL;
 
-//what a dirty hack!
+#if defined(DARWIN)
+  // GnomeVFS see xml files type as application/xml
+  // But GtkSourceView wants them as text/xml
+  // at least with gnomevfs 1.0.5 and gtksourceview 1.4.2
+  // So we change the string on the fly
+  // And get syntax hightlighting with xml files :-)
   gchar *m2 = g_strdup (m);
-
-  if (strcmp (m2, "application/xhtml+xml") == 0)
-     m2 = ch_str (m2, "text/html");
-//
-
-  GtkSourceLanguage *l = NULL;
+  
+  if (strcmp (m2, "application/xml") == 0)
+	  m2 = ch_str (m2, "text/html");
 
   if (m2)
-     l = gtk_source_languages_manager_get_language_from_mime_type (source_languages_manager, m2);
-
+	  l = gtk_source_languages_manager_get_language_from_mime_type (source_languages_manager, m2);
+#else  
+  l = gtk_source_languages_manager_get_language_from_mime_type (source_languages_manager, m);
+#endif
+  
   if (l)
      {
       gtk_source_buffer_set_language (doc->text_buffer, l);
       gtk_source_buffer_set_highlight (doc->text_buffer, TRUE);
      }
 
+#if defined(DARWIN)
   g_free (m2);
-
+#endif
+  
   gnome_vfs_file_info_unref (info);
 }
 
diff -Nurd tea-14.2-orig/src/tea_text_document.h tea-14.2/src/tea_text_document.h
--- tea-14.2-orig/src/tea_text_document.h	2006-08-15 13:46:26.000000000 +0200
+++ tea-14.2/src/tea_text_document.h	2006-08-16 01:55:41.000000000 +0200
@@ -260,3 +260,14 @@
 		gpointer       user_data);
 void doc_spaces_to_tabs (t_note_page *doc, gint tabsize);
 void doc_tabs_to_spaces (t_note_page *doc, gint tabsize);
+void create_tags (void);
+void prepare_hl_all (void);
+gchar* get_c_url (t_note_page *doc);
+void doc_move_to_pos_bw_quote (t_note_page* doc);
+gboolean doc_search_f (t_note_page *doc, const gchar *text);
+void apply_hl (t_note_page *doc);
+void doc_toggle_images_visibility (t_note_page *p);
+void remove_tags (t_note_page *doc);
+gboolean doc_search_f_next (t_note_page *doc);
+gint find_index_by_page (t_note_page *page);
+
