*** src/os_mac.h.orig	Mon Feb 16 01:20:52 2004
--- src/os_mac.h	Mon Feb 16 01:22:29 2004
***************
*** 119,123 ****
  # undef COLON_AS_PATHSEP
  # define USE_UNIXFILENAME
- # define USE_TRANSPARENCY
  /* Use OS depended message localization mechanism. */
  # ifdef USE_MCH_GETTEXT
--- 119,122 ----
*** src/auto/configure.orig	Mon Feb 16 01:19:07 2004
--- src/auto/configure	Mon Feb 16 01:22:07 2004
***************
*** 1284,1288 ****
    tt=`echo "$CPPFLAGS" | sed -e 's+-I/usr/local/include ++g' -e 's+-I/usr/local/include$++g'`
    if test "$tt" = "$CPPFLAGS"; then
!     CPPFLAGS="$CPPFLAGS -I/usr/local/include"
    fi
  fi
--- 1284,1288 ----
    tt=`echo "$CPPFLAGS" | sed -e 's+-I/usr/local/include ++g' -e 's+-I/usr/local/include$++g'`
    if test "$tt" = "$CPPFLAGS"; then
!     CPPFLAGS="$CPPFLAGS"
    fi
  fi
